--- valgrind/include/pub_tool_basics.h.jj	2011-10-26 23:24:32.000000000 +0200
+++ valgrind/include/pub_tool_basics.h	2011-12-09 17:42:50.828630546 +0100
@@ -297,7 +297,7 @@ static inline Bool sr_EQ ( SysRes sr1, S
 #define VG_BUGS_TO "www.valgrind.org"
 
 /* Branch prediction hints. */
-#if defined(__GNUC__)
+#if defined(__GNUC__) && (__GNUC__ >= 3)
 #  define LIKELY(x)   __builtin_expect(!!(x), 1)
 #  define UNLIKELY(x) __builtin_expect(!!(x), 0)
 #else
