diff -p -up ./coregrind/m_debuginfo/readdwarf.c.tv3 ./coregrind/m_debuginfo/readdwarf.c
--- ./coregrind/m_debuginfo/readdwarf.c.tv3	2012-08-27 14:35:14.115878372 +0000
+++ ./coregrind/m_debuginfo/readdwarf.c	2012-08-27 14:55:11.769749239 +0000
@@ -1230,7 +1230,10 @@ void ML_(read_debuginfo_dwarf3)
       read_unitinfo_dwarf2( &ui, block_img, 
                                  debug_abbv_img, debug_abbv_sz, debug_str_img,
                                  debug_str_alt_img );
+      /* Ignore blocks without a name */
+      if (ui.name == NULL)
+         break;
       if (0)
          VG_(printf)( "   => LINES=0x%llx    NAME=%s     DIR=%s\n", 
                       ui.stmt_list, ui.name, ui.compdir );
       
