--- valgrind/memcheck/mc_translate.c	(revision 12883)
+++ valgrind/memcheck/mc_translate.c	(revision 12884)
@@ -3313,6 +3313,8 @@
       case Iop_DivModS128to64:
          return mkLazy2(mce, Ity_I128, vatom1, vatom2);
 
+      case Iop_8HLto16:
+         return assignNew('V', mce, Ity_I16, binop(op, vatom1, vatom2));
       case Iop_16HLto32:
          return assignNew('V', mce, Ity_I32, binop(op, vatom1, vatom2));
       case Iop_32HLto64:
