--- valgrind/memcheck/tests/filter_memcpy	(revision 0)
+++ valgrind/memcheck/tests/filter_memcpy	(working copy)
@@ -0,0 +1,5 @@
+#! /bin/sh
+
+# mc_replace_strmem.c intercepts various memcpy glibc versions.
+./filter_stderr "$@" |
+perl -p -e "s/: memcpy\@\@?GLIBC_[.1-9]+ \(mc_replace_strmem.c:...\)/: memcpy \(mc_replace_strmem.c:...\)/"

--- valgrind/memcheck/tests/overlap.vgtest	(revision 13010)
+++ valgrind/memcheck/tests/overlap.vgtest	(working copy)
@@ -1,2 +1,3 @@
 prog: overlap
 vgopts: -q
+stderr_filter: filter_memcpy
--- valgrind/memcheck/tests/Makefile.am	2012-09-25 10:02:30.798819187 +0200
+++ valgrind/memcheck/tests/Makefile.am	2012-09-25 10:03:00.358251560 +0200
@@ -44,7 +44,8 @@
 	filter_leak_cases_possible \
 	filter_stderr filter_xml \
 	filter_varinfo3 \
-	filter_memcheck
+	filter_memcheck \
+	filter_memcpy
 
 noinst_HEADERS = leak.h
 
--- valgrind/memcheck/tests/Makefile.in	2012-09-25 10:02:30.797819171 +0200
+++ valgrind/memcheck/tests/Makefile.in	2012-09-25 10:03:28.620664967 +0200
@@ -971,7 +971,8 @@
 	filter_leak_cases_possible \
 	filter_stderr filter_xml \
 	filter_varinfo3 \
-	filter_memcheck
+	filter_memcheck \
+	filter_memcpy
 
 noinst_HEADERS = leak.h
 EXTRA_DIST = \
