--- valgrind/tests/x86_amd64_features.c	2012-07-11 15:30:47.564508460 +0200
+++ valgrind/tests/x86_amd64_features.c	2012-09-23 21:21:40.826638636 +0200
@@ -123,7 +123,7 @@
      cmask = 1 << 20;
    } else if ( strcmp( cpu, "amd64-avx" ) == 0 ) {
      level = 1;
-     cmask = (1 << 20) | (1 << 28);
+     cmask = (1 << 27) | (1 << 28); /* osxsave and avx */
      require_xgetbv = True;
 #endif
    } else {
