--- valgrind/configure.in	2012-03-05 15:16:23.000000000 -0500
+++ valgrind/configure.in	2012-03-05 13:49:04.000000000 -0500
@@ -186,7 +186,7 @@ case "${host_cpu}" in
         ARCH_MAX="s390x"
         ;;
 
-     armv7*)
+     armv[57]*)
 	AC_MSG_RESULT([ok (${host_cpu})])
 	ARCH_MAX="arm"
 	;;
--- valgrind/configure	2012-03-05 15:16:23.000000000 -0500
+++ valgrind/configure	2012-03-05 13:49:04.000000000 -0500
@@ -5335,7 +5335,7 @@
         ARCH_MAX="s390x"
         ;;
 
-     armv7*)
+     armv[57]*)
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: ok (${host_cpu})" >&5
 $as_echo "ok (${host_cpu})" >&6; }
 	ARCH_MAX="arm"
