From 77eecf82aa8a7c3b9f25c857ba988cd633485967 Mon Sep 17 00:00:00 2001
From: Igor Gevka <igor.gevka@gmail.com>
Date: Sun, 6 May 2012 02:12:50 +0200
Subject: [PATCH] Quick Google search from context menu

---
 data/guake.glade |   20 ++++++++++++++++++++
 src/guake        |   37 +++++++++++++++++++++++++++++++++++++
 2 files changed, 57 insertions(+)

diff --git a/data/guake.glade b/data/guake.glade
index 08124d7..f3be900 100644
--- a/data/guake.glade
+++ b/data/guake.glade
@@ -215,6 +215,26 @@
       </widget>
     </child>
     <child>
+      <widget class="GtkImageMenuItem" id="context_websearch_selection">
+        <property name="label" translatable="no">Google</property>
+        <property name="visible">True</property>
+        <property name="use_stock">False</property>
+        <signal name="activate" handler="websearch_selection"/>
+        <child internal-child="image">
+          <widget class="GtkImage" id="websearch_icon">
+            <property name="visible">True</property>
+            <property name="stock">gtk-find</property>
+            <property name="icon-size">1</property>
+          </widget>
+        </child>
+      </widget>
+    </child>
+    <child>
+      <widget class="GtkSeparatorMenuItem" id="websearch_separator">
+        <property name="visible">True</property>
+      </widget>
+    </child>
+    <child>
       <widget class="GtkImageMenuItem" id="context_close_tab">
         <property name="label" translatable="yes">Close Tab</property>
         <property name="visible">True</property>
diff --git a/src/guake b/src/guake
index 33ad7fd..d325f47 100644
--- a/src/guake
+++ b/src/guake
@@ -41,6 +41,7 @@ import signal
 from thread import start_new_thread
 from time import sleep
 import posix
+from urllib import quote_plus
 from urllib import url2pathname
 from urlparse import urlsplit
 
@@ -719,6 +720,24 @@ class Guake(SimpleGladeApp):
             self.get_widget('context_paste').set_sensitive(False)
         else:
             self.get_widget('context_paste').set_sensitive(True)
+
+        current_selection = ''
+        current_term = self.term_list[self.notebook.get_current_page()]
+        if current_term.get_has_selection():
+            primary_clipboard = gtk.Clipboard(selection="PRIMARY")
+            current_selection = primary_clipboard.wait_for_text().rstrip()
+            #uncomment following 2 lines if you want shortened context menu label
+            #if len(current_selection) > 15:
+            #    current_selection = current_selection[:12] + "..."
+
+        if current_selection:
+            self.get_widget('context_websearch_selection').set_label("Google '%s'" % current_selection)
+            self.get_widget('context_websearch_selection').set_visible(True)
+            self.get_widget('websearch_separator').set_visible(True)
+        else:
+            self.get_widget('context_websearch_selection').set_visible(False)
+            self.get_widget('websearch_separator').set_visible(False)
+
         context_menu = self.get_widget('context-menu')
         context_menu.popup(None, None, None, 3, gtk.get_current_event_time())
         return True
@@ -949,6 +968,24 @@ class Guake(SimpleGladeApp):
         self.term_list[pos].paste_clipboard()
         return True
 
+    def websearch_selection(self, *args):
+        """Callback to perform a websearch for selected text. Useful for
+        error message lookups. Called by the context menu
+        """
+        current_term = self.term_list[self.notebook.get_current_page()]
+
+        clipboard = gtk.Clipboard(selection="PRIMARY")
+        if clipboard.wait_is_text_available():
+            selection = clipboard.wait_for_text()
+            search_query = quote_plus(selection)
+            if search_query:
+                request_url = "http://www.google.com/search?q="+search_query
+                gtk.show_uri(current_term.window.get_screen(),
+                             request_url,
+                             gtk.gdk.x11_get_server_time(current_term.window))
+
+        return True
+
     def accel_toggle_fullscreen(self, *args):
         """Callback toggle the fullscreen status of the main
         window. It uses the toolbar_visible_in_fullscreen variable
-- 
1.7.9.5

