From fde6c99bd1956e5db632499cf39c05720f3d2c99 Mon Sep 17 00:00:00 2001
From: Dom Sekotill <kodo@gmx.co.uk>
Date: Thu, 29 Sep 2011 14:18:53 +0100
Subject: [PATCH 2/2] Added start in fullscreen to the gconf schema.

---
 data/guake.schemas |   12 ++++++++++++
 1 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/data/guake.schemas b/data/guake.schemas
index 403bf1d..4c614a2 100644
--- a/data/guake.schemas
+++ b/data/guake.schemas
@@ -105,6 +105,18 @@
             </locale>
         </schema>
 
+        <schema>
+            <key>/schemas/apps/guake/general/start_fullscreen</key>
+            <applyto>/apps/guake/general/start_fullscreen</applyto>
+            <owner>guake</owner>
+            <type>bool</type>
+            <default>false</default>
+            <locale name="C">
+                <short>Start fullscreen.</short>
+                <long>When true, the program will start in fullscreen mode.</long>
+            </locale>
+        </schema>
+
 	<schema>
             <key>/schemas/apps/guake/general/window_width</key>
             <applyto>/apps/guake/general/window_width</applyto>
-- 
1.7.4.1

