--- guake-0.4.0-orig/data/guake.glade	2009-03-24 21:53:19.000000000 +0200
+++ guake-0.4.0-notify/data/guake.glade	2009-08-28 22:38:53.000000000 +0300
@@ -247,6 +247,14 @@
   <widget class="GtkMenu" id="tab-menu">
     <property name="visible">True</property>
     <child>
+      <widget class="GtkCheckMenuItem" id="ignore_output">
+        <property name="visible">True</property>
+        <property name="label" translatable="yes">Ignore Output</property>
+        <property name="use_underline">True</property>
+        <signal name="toggled" handler="on_ignore_output_toggled"/>
+      </widget>
+    </child>
+    <child>
       <widget class="GtkImageMenuItem" id="rename">
         <property name="label" translatable="yes">Rename</property>
         <property name="visible">True</property>
