Index: dvi2tty-5.3.4/Makefile
===================================================================
--- dvi2tty-5.3.4.orig/Makefile
+++ dvi2tty-5.3.4/Makefile
@@ -33,6 +33,9 @@ clean:
 	rm -f dvi2tty disdvi *.o
 
 install:	all
-	install dvi2tty disdvi $(bindir)
-	install dvi2tty.1 $(mandir)
-	install disdvi.1 $(mandir)
+	install -d ${bindir}
+	install -m 555 dvi2tty disdvi $(bindir)
+	install -d ${mandir}
+	install -m 444 dvi2tty.1 $(mandir)
+	install -m 444 disdvi.1 $(mandir)
+
