diff -Nur pyvorbis-1.4-old/test/ogg123.py pyvorbis-1.4/test/ogg123.py
--- pyvorbis-1.4-old/test/ogg123.py	2007-03-06 15:36:48.737601230 +0200
+++ pyvorbis-1.4/test/ogg123.py	2007-03-06 15:37:13.048905350 +0200
@@ -98,7 +98,7 @@
     def __init__(self, id=None):
         import ao
         if id is None:
-            id = ao.driver_id('esd')
+            id = ao.driver_id('alsa')
         self.dev = ao.AudioDevice(id)
 
     def write(self, buff, bytes):
diff -Nur pyvorbis-1.4-old/test/short.py pyvorbis-1.4/test/short.py
--- pyvorbis-1.4-old/test/short.py	2007-03-06 15:36:48.737601230 +0200
+++ pyvorbis-1.4/test/short.py	2007-03-06 15:37:13.048905350 +0200
@@ -6,7 +6,7 @@
 import ao
 
 filename = 'test.ogg'
-device = 'esd'
+device = 'alsa'
 SIZE = 4096
 
 vf = ogg.vorbis.VorbisFile(filename)
