diff -Naur extensions/launchXmoto.inx extensions_patched/launchXmoto.inx
--- extensions/launchXmoto.inx	2009-02-12 00:21:52.000000000 +0200
+++ extensions_patched/launchXmoto.inx	2010-06-15 01:22:36.713402584 +0300
@@ -6,7 +6,7 @@
 	<_param name="dummy" type="description">If the Xmoto executable is present in your path,</_param>
 	<_param name="dummy" type="description">you don't have to fill in the value.</_param>
 	<_param name="dummy" type="description">Else enter the Xmoto executable with its full path.</_param>
-	<param _gui-text="Xmoto executable:" name="xmoto" type="string">C:\Program Files\Xmoto\xmoto.exe</param>
+	<param _gui-text="Xmoto executable:" name="xmoto" type="string">/usr/bin/xmoto</param>
     <effect>
 		<object-type>path</object-type>
 		<effects-menu>
