--- easy-rsa/2.0/openssl.cnf.orig	2010-01-14 16:27:51.977369194 +0200
+++ easy-rsa/2.0/openssl.cnf	2010-01-14 16:29:54.090381299 +0200
@@ -283,9 +283,9 @@
 #pkcs11 = pkcs11_section
 
 [ pkcs11_section ]
-engine_id = pkcs11
-dynamic_path = /usr/lib/engines/engine_pkcs11.so
-MODULE_PATH = $ENV::PKCS11_MODULE_PATH
-PIN = $ENV::PKCS11_PIN
-init = 0
+#engine_id = pkcs11
+#dynamic_path = /usr/lib/engines/engine_pkcs11.so
+#MODULE_PATH = $ENV::PKCS11_MODULE_PATH
+#PIN = $ENV::PKCS11_PIN
+#init = 0
 
