--- src/modules/graphic/ssggraph/Makefile.orig	2009-07-16 16:41:59.535354464 +0300
+++ src/modules/graphic/ssggraph/Makefile	2009-07-16 16:41:28.656707230 +0300
@@ -24,7 +24,7 @@
 
 MODULEDIR = modules/graphic
 
-LIBS      = -lplibsl -lplibsm -lplibsg -lplibssg -lplibul -lplibssgaux -lopenal
+LIBS      = -lplibsl -lplibsm -lplibsg -lplibssg -lplibul -lplibssgaux -lopenal -lalut
 
 SHIPDIR   = config
 
