--- src/linux/Makefile.orig	2012-02-06 09:16:06.990826504 +0100
+++ src/linux/Makefile	2012-02-06 09:16:48.215927184 +0100
@@ -25,9 +25,7 @@ SOURCES = main.cpp linuxspec.cpp
 
 LDFLAGS = -rdynamic
 
-SOLIBS  =  -lracescreens  \
-	   -lrobottools   \
-	   -lclient       \
+SOLIBS  =  -lclient       \
 	   -lconfscreens  \
 	   -ltgf          \
 	   -ltgfclient    \
@@ -35,6 +33,8 @@ SOLIBS  =  -lracescreens  \
 	   -lplibul	  \
 	   -lraceengine   \
 	   -lmusicplayer  \
+	   -lracescreens  \
+	   -lrobottools   \
 	   -llearning
 
 EXPDIR  = include
