diff -urN torcs-1.3.4/Make-config torcs-1.3.4.new/Make-config
--- torcs-1.3.4/Make-config	2012-10-19 16:05:48.000000000 +0400
+++ torcs-1.3.4.new/Make-config	2012-12-08 17:52:44.624446485 +0400
@@ -28,7 +28,7 @@
 bindir = ${exec_prefix}/bin
 libdir = ${exec_prefix}/lib64/torcs
 datarootdir = ${prefix}/share
-datadir = ${prefix}/share/games/torcs
+datadir = ${prefix}/share/torcs
 vardir = ${prefix}/var/games/torcs
 
 top_builddir = .
diff -urN torcs-1.3.4/Make-config.in torcs-1.3.4.new/Make-config.in
--- torcs-1.3.4/Make-config.in	2012-10-19 16:05:48.000000000 +0400
+++ torcs-1.3.4.new/Make-config.in	2012-12-08 17:52:44.626446443 +0400
@@ -28,7 +28,7 @@
 bindir = @bindir@
 libdir = @libdir@/torcs
 datarootdir = @datarootdir@
-datadir = @datarootdir@/games/torcs
+datadir = @datarootdir@/torcs
 vardir = @localstatedir@/games/torcs
 
 top_builddir = .
diff -urN torcs-1.3.4/src/linux/torcs torcs-1.3.4.new/src/linux/torcs
--- torcs-1.3.4/src/linux/torcs	2012-10-19 16:05:44.000000000 +0400
+++ torcs-1.3.4.new/src/linux/torcs	2012-12-08 17:52:44.626446443 +0400
@@ -24,7 +24,7 @@
 BINDIR=${exec_prefix}/bin
 LIBDIR=${exec_prefix}/lib64/torcs
 DATAROOTDIR=${prefix}/share
-DATADIR=$DATAROOTDIR/games/torcs
+DATADIR=$DATAROOTDIR/torcs
 VARDIR=${prefix}/var/games/torcs
 
 LOCAL_CONF=$HOME/.torcs
diff -urN torcs-1.3.4/src/linux/torcs.in torcs-1.3.4.new/src/linux/torcs.in
--- torcs-1.3.4/src/linux/torcs.in	2012-10-19 16:05:44.000000000 +0400
+++ torcs-1.3.4.new/src/linux/torcs.in	2012-12-08 17:52:44.628446387 +0400
@@ -24,7 +24,7 @@
 BINDIR=@bindir@
 LIBDIR=@libdir@/torcs
 DATAROOTDIR=@datarootdir@
-DATADIR=$DATAROOTDIR/games/torcs
+DATADIR=$DATAROOTDIR/torcs
 VARDIR=@localstatedir@/games/torcs
 
 LOCAL_CONF=$HOME/.torcs
diff -urN torcs-1.3.4/src/tools/accc/accc torcs-1.3.4.new/src/tools/accc/accc
--- torcs-1.3.4/src/tools/accc/accc	2012-10-19 16:05:45.000000000 +0400
+++ torcs-1.3.4.new/src/tools/accc/accc	2012-12-08 17:52:44.629446356 +0400
@@ -24,7 +24,7 @@
 BINDIR=${exec_prefix}/bin
 LIBDIR=${exec_prefix}/lib64/torcs
 DATAROOTDIR=${prefix}/share
-DATADIR=$DATAROOTDIR/games/torcs
+DATADIR=$DATAROOTDIR/torcs
 VARDIR=${prefix}/var/games/torcs
 
 export LD_LIBRARY_PATH=${LD_LIBRARY_PATH:+$LD_LIBRARY_PATH:}$LIBDIR/lib
diff -urN torcs-1.3.4/src/tools/accc/accc.in torcs-1.3.4.new/src/tools/accc/accc.in
--- torcs-1.3.4/src/tools/accc/accc.in	2012-10-19 16:05:45.000000000 +0400
+++ torcs-1.3.4.new/src/tools/accc/accc.in	2012-12-08 17:52:44.629446356 +0400
@@ -24,7 +24,7 @@
 BINDIR=@bindir@
 LIBDIR=@libdir@/torcs
 DATAROOTDIR=@datarootdir@
-DATADIR=$DATAROOTDIR/games/torcs
+DATADIR=$DATAROOTDIR/torcs
 VARDIR=@localstatedir@/games/torcs
 
 export LD_LIBRARY_PATH=${LD_LIBRARY_PATH:+$LD_LIBRARY_PATH:}$LIBDIR/lib
diff -urN torcs-1.3.4/src/tools/nfs2ac/nfs2ac torcs-1.3.4.new/src/tools/nfs2ac/nfs2ac
--- torcs-1.3.4/src/tools/nfs2ac/nfs2ac	2012-10-19 16:05:45.000000000 +0400
+++ torcs-1.3.4.new/src/tools/nfs2ac/nfs2ac	2012-12-08 17:52:44.630446324 +0400
@@ -24,7 +24,7 @@
 BINDIR=${exec_prefix}/bin
 LIBDIR=${exec_prefix}/lib64/torcs
 DATAROOTDIR=${prefix}/share
-DATADIR=$DATAROOTDIR/games/torcs
+DATADIR=$DATAROOTDIR/torcs
 VARDIR=${prefix}/var/games/torcs
 
 export LD_LIBRARY_PATH=${LD_LIBRARY_PATH:+$LD_LIBRARY_PATH:}$LIBDIR/lib
diff -urN torcs-1.3.4/src/tools/nfs2ac/nfs2ac.in torcs-1.3.4.new/src/tools/nfs2ac/nfs2ac.in
--- torcs-1.3.4/src/tools/nfs2ac/nfs2ac.in	2012-10-19 16:05:45.000000000 +0400
+++ torcs-1.3.4.new/src/tools/nfs2ac/nfs2ac.in	2012-12-08 17:52:44.630446324 +0400
@@ -24,7 +24,7 @@
 BINDIR=@bindir@
 LIBDIR=@libdir@/torcs
 DATAROOTDIR=@datarootdir@
-DATADIR=$DATAROOTDIR/games/torcs
+DATADIR=$DATAROOTDIR/torcs
 VARDIR=@localstatedir@/games/torcs
 
 export LD_LIBRARY_PATH=${LD_LIBRARY_PATH:+$LD_LIBRARY_PATH:}$LIBDIR/lib
diff -urN torcs-1.3.4/src/tools/nfsperf/nfsperf torcs-1.3.4.new/src/tools/nfsperf/nfsperf
--- torcs-1.3.4/src/tools/nfsperf/nfsperf	2012-10-19 16:05:45.000000000 +0400
+++ torcs-1.3.4.new/src/tools/nfsperf/nfsperf	2012-12-08 17:52:44.632446260 +0400
@@ -24,7 +24,7 @@
 BINDIR=${exec_prefix}/bin
 LIBDIR=${exec_prefix}/lib64/torcs
 DATAROOTDIR=${prefix}/share
-DATADIR=$DATAROOTDIR/games/torcs
+DATADIR=$DATAROOTDIR/torcs
 VARDIR=${prefix}/var/games/torcs
 
 export LD_LIBRARY_PATH=${LD_LIBRARY_PATH:+$LD_LIBRARY_PATH:}$LIBDIR/lib
diff -urN torcs-1.3.4/src/tools/nfsperf/nfsperf.in torcs-1.3.4.new/src/tools/nfsperf/nfsperf.in
--- torcs-1.3.4/src/tools/nfsperf/nfsperf.in	2012-10-19 16:05:45.000000000 +0400
+++ torcs-1.3.4.new/src/tools/nfsperf/nfsperf.in	2012-12-08 17:52:44.632446260 +0400
@@ -24,7 +24,7 @@
 BINDIR=@bindir@
 LIBDIR=@libdir@/torcs
 DATAROOTDIR=@datarootdir@
-DATADIR=$DATAROOTDIR/games/torcs
+DATADIR=$DATAROOTDIR/torcs
 VARDIR=@localstatedir@/games/torcs
 
 export LD_LIBRARY_PATH=${LD_LIBRARY_PATH:+$LD_LIBRARY_PATH:}$LIBDIR/lib
diff -urN torcs-1.3.4/src/tools/texmapper/texmapper torcs-1.3.4.new/src/tools/texmapper/texmapper
--- torcs-1.3.4/src/tools/texmapper/texmapper	2012-10-19 16:05:45.000000000 +0400
+++ torcs-1.3.4.new/src/tools/texmapper/texmapper	2012-12-08 17:52:44.632446260 +0400
@@ -24,7 +24,7 @@
 BINDIR=${exec_prefix}/bin
 LIBDIR=${exec_prefix}/lib64/torcs
 DATAROOTDIR=${prefix}/share
-DATADIR=$DATAROOTDIR/games/torcs
+DATADIR=$DATAROOTDIR/torcs
 VARDIR=${prefix}/var/games/torcs
 
 export LD_LIBRARY_PATH=${LD_LIBRARY_PATH:+$LD_LIBRARY_PATH:}$LIBDIR/lib
diff -urN torcs-1.3.4/src/tools/texmapper/texmapper.in torcs-1.3.4.new/src/tools/texmapper/texmapper.in
--- torcs-1.3.4/src/tools/texmapper/texmapper.in	2012-10-19 16:05:45.000000000 +0400
+++ torcs-1.3.4.new/src/tools/texmapper/texmapper.in	2012-12-08 17:52:44.634446194 +0400
@@ -24,7 +24,7 @@
 BINDIR=@bindir@
 LIBDIR=@libdir@/torcs
 DATAROOTDIR=@datarootdir@
-DATADIR=$DATAROOTDIR/games/torcs
+DATADIR=$DATAROOTDIR/torcs
 VARDIR=@localstatedir@/games/torcs
 
 export LD_LIBRARY_PATH=${LD_LIBRARY_PATH:+$LD_LIBRARY_PATH:}$LIBDIR/lib
diff -urN torcs-1.3.4/src/tools/trackgen/trackgen torcs-1.3.4.new/src/tools/trackgen/trackgen
--- torcs-1.3.4/src/tools/trackgen/trackgen	2012-10-19 16:05:45.000000000 +0400
+++ torcs-1.3.4.new/src/tools/trackgen/trackgen	2012-12-08 17:52:44.634446194 +0400
@@ -24,7 +24,7 @@
 BINDIR=${exec_prefix}/bin
 LIBDIR=${exec_prefix}/lib64/torcs
 DATAROOTDIR=${prefix}/share
-DATADIR=$DATAROOTDIR/games/torcs
+DATADIR=$DATAROOTDIR/torcs
 VARDIR=${prefix}/var/games/torcs
 
 export LD_LIBRARY_PATH=${LD_LIBRARY_PATH:+$LD_LIBRARY_PATH:}$LIBDIR/lib
diff -urN torcs-1.3.4/src/tools/trackgen/trackgen.in torcs-1.3.4.new/src/tools/trackgen/trackgen.in
--- torcs-1.3.4/src/tools/trackgen/trackgen.in	2012-10-19 16:05:45.000000000 +0400
+++ torcs-1.3.4.new/src/tools/trackgen/trackgen.in	2012-12-08 17:52:44.634446194 +0400
@@ -24,7 +24,7 @@
 BINDIR=@bindir@
 LIBDIR=@libdir@/torcs
 DATAROOTDIR=@datarootdir@
-DATADIR=$DATAROOTDIR/games/torcs
+DATADIR=$DATAROOTDIR/torcs
 VARDIR=@localstatedir@/games/torcs
 
 export LD_LIBRARY_PATH=${LD_LIBRARY_PATH:+$LD_LIBRARY_PATH:}$LIBDIR/lib
