Index: transmission-2.31/gtk/transmission-gtk.desktop.in
===================================================================
--- transmission-2.31.orig/gtk/transmission-gtk.desktop.in
+++ transmission-2.31/gtk/transmission-gtk.desktop.in
@@ -1,5 +1,5 @@
 [Desktop Entry]
-_Name=Transmission
+_Name=Transmission Gtk
 _GenericName=BitTorrent Client
 _X-GNOME-FullName=Transmission BitTorrent Client
 _Comment=Download and share files over BitTorrent
