Index: kvkbd-0.6/src/kvkbd.desktop
===================================================================
--- kvkbd-0.6.orig/src/kvkbd.desktop
+++ kvkbd-0.6/src/kvkbd.desktop
@@ -21,6 +21,7 @@ Comment[pt_BR]=Um teclado virtual para o
 Comment[ro]=Tastatură virtuală pentru KDE
 Comment[sv]=Ett virtuellt tangentbord för KDE
 Comment[uk]=Віртуальна клавіатура для KDE
+Comment[tr]=KDE için sanal klavye
 Comment[x-test]=xxA virtual keyboard for KDExx
 GenericName=Virtual Keyboard
 GenericName[el]=Εικονικό πληκτρολόγιο
@@ -33,6 +34,7 @@ GenericName[pt]=Teclado Virtual
 GenericName[pt_BR]=Teclado Virtual
 GenericName[sv]=Virtuellt tangentbord
 GenericName[uk]=Віртуальна клавіатура
+GenericName[tr]=Sanal Klavye
 GenericName[x-test]=xxVirtual Keyboardxx
 Categories=Qt;KDE;Utility;Accessibility;
 Terminal=false
