diff -Nur dxflib-2.0.4.8-2.src.orig/src/dl_writer_ascii.cpp dxflib-2.0.4.8-2.src/src/dl_writer_ascii.cpp
--- dxflib-2.0.4.8-2.src.orig/src/dl_writer_ascii.cpp	2008-05-29 18:01:54.000000000 +0300
+++ dxflib-2.0.4.8-2.src/src/dl_writer_ascii.cpp	2008-05-29 18:04:02.000000000 +0300
@@ -30,6 +30,7 @@
 #endif // _MSC_VER > 1000
 
 #include <stdio.h>
+#include <string.h>
 
 #include "dl_writer_ascii.h"
 #include "dl_exception.h"
diff -Nur dxflib-2.0.4.8-2.src.orig/src/dl_writer.h dxflib-2.0.4.8-2.src/src/dl_writer.h
--- dxflib-2.0.4.8-2.src.orig/src/dl_writer.h	2008-05-29 18:01:54.000000000 +0300
+++ dxflib-2.0.4.8-2.src/src/dl_writer.h	2008-05-29 18:03:47.000000000 +0300
@@ -37,6 +37,7 @@
 #endif
 
 #include <iostream>
+#include <strings.h>
 
 #include "dl_attributes.h"
 #include "dl_codes.h"
