Index: pycairo-1.8.10/src/Makefile.in
===================================================================
--- pycairo-1.8.10.orig/src/Makefile.in
+++ pycairo-1.8.10/src/Makefile.in
@@ -439,13 +439,6 @@ install-pycairoexecPYTHON: $(pycairoexec
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(pycairoexecdir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(pycairoexecdir)" || exit $$?; \
 	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pycairoexecdir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pycairoexecdir)" $$dlist; \
-	  fi; \
-	else :; fi
 
 uninstall-pycairoexecPYTHON:
 	@$(NORMAL_UNINSTALL)
