diff -Nuar redland-1.0.10.orig/src/rdf_digest.c redland-1.0.10/src/rdf_digest.c
--- redland-1.0.10.orig/src/rdf_digest.c	2010-03-07 01:30:22.588657802 +0200
+++ redland-1.0.10/src/rdf_digest.c	2010-03-07 01:37:17.716408728 +0200
@@ -209,6 +209,9 @@
 {
   librdf_digest* d;
 
+  if (!factory)
+      return NULL;
+
   librdf_world_open(world);
 
   d=(librdf_digest*)LIBRDF_CALLOC(librdf_digest, 1, sizeof(librdf_digest));
