diff -Naur redland-1.0.9-orig/configure.ac redland-1.0.9/configure.ac
--- redland-1.0.9-orig/configure.ac	2009-07-28 14:59:04.361703816 +0300
+++ redland-1.0.9/configure.ac	2009-07-28 14:59:16.608454149 +0300
@@ -977,8 +977,6 @@
 
 AC_CONFIG_FILES([Makefile
 redland.rdf
-docs/Makefile
-docs/version.xml
 examples/Makefile
 src/Makefile
 src/redland.spec
@@ -990,9 +988,6 @@
 AC_CONFIG_FILES([redland-config], [chmod +x redland-config])
 AC_CONFIG_FILES([redland-src-config], [chmod +x redland-src-config])
 
-dnl Check for gtk-doc and docbook
-GTK_DOC_CHECK([1.3])
-
 AC_OUTPUT
 
 AC_MSG_RESULT([
diff -Naur redland-1.0.9-orig/Makefile.am redland-1.0.9/Makefile.am
--- redland-1.0.9-orig/Makefile.am	2009-07-28 14:59:04.365626697 +0300
+++ redland-1.0.9/Makefile.am	2009-07-28 14:59:24.224454025 +0300
@@ -34,7 +34,7 @@
 DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-doc
 
 # Subdirectories to build/install/distribute etc.
-SUBDIRS=$(subdirs) src examples utils demos docs data
+SUBDIRS=$(subdirs) src examples utils demos data
 
 EXTRA_DIST=ChangeLog ChangeLog.1 ChangeLog.2 ChangeLog.3 ChangeLog.4 \
 ChangeLog.5 ChangeLog.6 ChangeLog.7 \
