Index: smb4k-0.10.90/smb4k/smb4k.desktop
===================================================================
--- smb4k-0.10.90.orig/smb4k/smb4k.desktop
+++ smb4k-0.10.90/smb4k/smb4k.desktop
@@ -33,6 +33,7 @@ GenericName[pt_BR]=Navegador da vizinhan
 GenericName[sv]=Avancerad bläddrare för nätverksgrannskapet
 GenericName[ug]=ئالىي تور قوشنا كۆرگۈ
 GenericName[uk]=Переглядач мережевого оточення з додатковими можливостями
+GenericName[tr]=Samba Paylaşım Yöneticisi
 GenericName[x-test]=xxAdvanced Network Neighborhood Browserxx
 Comment=Advanced Network Neighborhood Browser
 Comment[da]=Den avancerede browser til netværksnabolag
@@ -50,6 +51,7 @@ Comment[pt_BR]=Navegador da vizinhança
 Comment[sv]=Avancerad bläddrare för nätverksgrannskapet
 Comment[ug]=ئالىي تور قوشنا كۆرگۈ
 Comment[uk]=Переглядач мережевого оточення з додатковими можливостями
+Comment[tr]=Gelişmiş Çevre Ağ Tarayıcısı
 Comment[x-test]=xxAdvanced Network Neighborhood Browserxx
 Comment[zh_CN]=高级网上邻居浏览器
 Exec=smb4k
@@ -57,6 +59,6 @@ Icon=smb4k
 X-DocPath=smb4k/index.html
 Type=Application
 Terminal=false
-Categories=Qt;KDE;Utility;
+Categories=Qt;KDE;Network;Utility;
 X-DBUS-StartupType=Unique
 X-DBUS-ServiceName=de.berlios.smb4k
