Index: setup.py
===================================================================
--- setup.py	(revision 27404)
+++ setup.py	(revision 27405)
@@ -76,7 +76,9 @@
 setup(
     name='piksemel',
     version=version,
-    ext_modules=[Extension('piksemel', ['src/iksemel.c', 'src/pyiks.c'])],
+    ext_modules=[Extension('piksemel',
+                            ['src/iksemel.c', 'src/pyiks.c'],
+                            extra_compile_args=["-fvisibility=hidden"])],
     cmdclass = {
         'install' : Install
     }
Index: src/pyiks.c
===================================================================
--- src/pyiks.c	(revision 27404)
+++ src/pyiks.c	(revision 27405)
@@ -994,6 +994,7 @@
 	{ NULL, NULL, 0, NULL }
 };
 
+__attribute__((visibility("default")))
 PyMODINIT_FUNC
 initpiksemel(void)
 {
