Index: NetworkManager-openvpn-0.8.5.91/po/POTFILES.in
===================================================================
--- NetworkManager-openvpn-0.8.5.91.orig/po/POTFILES.in
+++ NetworkManager-openvpn-0.8.5.91/po/POTFILES.in
@@ -5,7 +5,7 @@ auth-dialog/main.c
 nm-openvpn.desktop.in
 properties/auth-helpers.c
 properties/nm-openvpn.c
-properties/nm-openvpn-dialog.ui
+[type: gettext/glade]properties/nm-openvpn-dialog.ui
 src/nm-openvpn-service.c
 src/nm-openvpn-service-openvpn-helper.c
 
