diff -Naur python-ldap-2.3.5.orig/setup.cfg python-ldap-2.3.5/setup.cfg
--- python-ldap-2.3.5.orig/setup.cfg	2009-01-16 10:22:32.000000000 +0200
+++ python-ldap-2.3.5/setup.cfg	2009-01-16 10:23:15.000000000 +0200
@@ -2,8 +2,8 @@
 extra_objects = 
 extra_compile_args = 
 libs = ldap_r lber sasl2 ssl crypto
-library_dirs = /usr/local/openldap-2.3/lib
-include_dirs = /usr/local/openldap-2.3/include /usr/include/sasl
+library_dirs = /usr/lib
+include_dirs = /usr/include /usr/include/sasl
 
 [egg_info]
 tag_build = 
@@ -11,8 +11,8 @@
 tag_svn_revision = 0
 
 [install]
-compile = 1
-optimize = 1
+compile = 0
+optimize = 0
 
 [bdist_rpm]
 doc_files = CHANGES README INSTALL TODO Demo/
