diff -Naur python-ldap-2.3.5.orig//Modules/errors.c python-ldap-2.3.5/Modules/errors.c
--- python-ldap-2.3.5.orig//Modules/errors.c	2011-02-11 09:34:41.277000104 +0200
+++ python-ldap-2.3.5/Modules/errors.c	2011-02-11 09:34:52.976000104 +0200
@@ -18,7 +18,7 @@
 
 #ifdef LDAP_ASSERTION_FAILED
 #define LDAP_ERROR_MAX          LDAP_ASSERTION_FAILED
-#elif
+#else
 #define LDAP_ERROR_MAX          LDAP_OTHER
 #endif
 
