Index: libgnomevfs/gnome-vfs-filesystem-type.c
===================================================================
--- libgnomevfs/gnome-vfs-filesystem-type.c.orig
+++ libgnomevfs/gnome-vfs-filesystem-type.c
@@ -37,7 +37,7 @@ struct FSInfo {
  * documentation! */
 static struct FSInfo fs_data[] = {
 	{ "affs"     , N_("AFFS Volume"), 0},
-	{ "afs"      , N_("AFS Network Volume"), 0 },
+	{ "afs"      , N_("AFS Network Volume"), 1 },
 	{ "auto"     , N_("Auto-detected Volume"), 0 },
 	{ "cd9660"   , N_("CD-ROM Drive"), 0 },
 	{ "cdda"     , N_("CD Digital Audio"), 0 },
