diff -ur squirrel.OSmanOS/sqstate.h squirrel/sqstate.h
--- squirrel.OSmanOS/sqstate.h	2007-10-14 23:21:35.000000000 +0300
+++ squirrel/sqstate.h	2013-03-13 19:55:01.572967261 +0200
@@ -137,7 +137,4 @@
 
 bool CompileTypemask(SQIntVec &res,const SQChar *typemask);
 
-void *sq_vm_malloc(SQUnsignedInteger size);
-void *sq_vm_realloc(void *p,SQUnsignedInteger oldsize,SQUnsignedInteger size);
-void sq_vm_free(void *p,SQUnsignedInteger size);
 #endif //_SQSTATE_H_
diff -ur squirrel.OSmanOS/squtils.h squirrel/squtils.h
--- squirrel.OSmanOS/squtils.h	2008-10-14 17:36:01.000000000 +0300
+++ squirrel/squtils.h	2013-03-13 19:56:51.979975432 +0200
@@ -8,6 +8,9 @@
 #define SQ_FREE(__ptr,__size) sq_vm_free((__ptr),(__size));
 #define SQ_REALLOC(__ptr,__oldsize,__size) sq_vm_realloc((__ptr),(__oldsize),(__size));
 
+void *sq_vm_malloc(SQUnsignedInteger size);
+void *sq_vm_realloc(void *p,SQUnsignedInteger oldsize,SQUnsignedInteger size);
+void sq_vm_free(void *p,SQUnsignedInteger size);
 //sqvector mini vector class, supports objects by value
 template<typename T> class sqvector
 {
