--- pulse/pulse.c.orig	2008-10-29 14:42:13.000000000 +0200
+++ pulse/pulse.c	2008-10-31 12:16:02.000000000 +0200
@@ -268,7 +268,10 @@
 	return 0;
 
       error:
-	SNDERR("PulseAudio: Unable to connect: %s\n",
+	SNDERR("*** PulseAudio: Unable to connect: %s\n"
+           "*** Is your sound server running?\n"
+           "*** See: http://www.pulseaudio.org/wiki/Troubleshooting\n"
+           "*** If you think it's a bug, report it: http://bugs.pardus.org.tr\n",
 		pa_strerror(pa_context_errno(p->context)));
 
 	pa_threaded_mainloop_unlock(p->mainloop);
