diff -Nuar eooorg.orig//module.desktop.in eooorg/module.desktop.in
--- eooorg.orig//module.desktop.in	2011-02-16 22:39:52.000000000 +0200
+++ eooorg/module.desktop.in	2011-03-31 16:39:34.834013001 +0300
@@ -5,11 +5,13 @@
 Name[fr]=Démarrage rapide OpenOffice.org
 Name[it]=Avvio rapido OpenOffice.org
 Name[pt]=Início rápido do OpenOffice.org
+Name[tr]=OpenOffice.org Otomatik Başlat
 Icon=e-module-eooorg
 Comment=This module preloads the openoffice process to save start time with the costs of some memory.
 Comment[ru]=Этот модуль подгружает процесс OpenOffice.org для убыстрения запуска, ценой некоторого объема памяти.
 Comment[fr]=Préchargement de OpenOffice.org pour accélérer son lancement, au prix d'un peu de mémoire vive.
 Comment[it]=Questo modulo precarica il processo di OpenOffice.org per risparmiare tempo<br>nell'avvio al costo di un po' di memoria.
 Comment[pt]=Este módulo faz o carregamento automático do openoffice para minimizar o tempo de arranque.
+Comment[tr]=Openoffice sürecini ön yükleyerek başlangıç zamanından tasarruf sağlamanıza olanak tanır.
 #Several different types available here. NEED TO DOC THESE.
 X-Enlightenment-ModuleType=utils
