diff -up vsftpd-2.2.0/ssl.c.openssl vsftpd-2.2.0/ssl.c
--- vsftpd-2.2.0/ssl.c.openssl	2009-01-09 21:47:05.000000000 +0100
+++ vsftpd-2.2.0/ssl.c	2009-08-22 20:41:58.000000000 +0200
@@ -537,7 +537,11 @@ get_ssl(struct vsf_session* p_sess, int 
   if (tunable_debug_ssl)
   {
     const char* p_ssl_version = SSL_get_cipher_version(p_ssl);
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+    const SSL_CIPHER* p_ssl_cipher = SSL_get_current_cipher(p_ssl);
+#else
     SSL_CIPHER* p_ssl_cipher = SSL_get_current_cipher(p_ssl);
+#endif
     const char* p_cipher_name = SSL_CIPHER_get_name(p_ssl_cipher);
     X509* p_ssl_cert = SSL_get_peer_certificate(p_ssl);
     int reused = SSL_session_reused(p_ssl);
