--- vsftpd-2.0.5.orig/sysdeputil.c  2007-01-05 17:58:28.000000000 +0000
+++ vsftpd-2.0.5/sysdeputil.c   2007-01-05 17:55:30.000000000 +0000
@@ -154,11 +154,8 @@
 #include <sys/capability.h>
 
 #if defined(VSF_SYSDEP_HAVE_CAPABILITIES) && !defined(VSF_SYSDEP_HAVE_LIBCAP)
-#include <linux/unistd.h>
 #include <linux/capability.h>
-#include <errno.h>
-#include <syscall.h>
-_syscall2(int, capset, cap_user_header_t, header, const cap_user_data_t, data)
+#include <sys/syscall.h>
 /* Gross HACK to avoid warnings - linux headers overlap glibc headers */
 #undef __NFDBITS
 #undef __FDMASK
