diff -up vsftpd-2.1.0/Makefile.libs vsftpd-2.1.0/Makefile
--- vsftpd-2.1.0/Makefile.libs	2009-01-08 18:55:39.000000000 +0100
+++ vsftpd-2.1.0/Makefile	2009-01-08 18:55:57.000000000 +0100
@@ -5,7 +5,7 @@ IFLAGS  = -idirafter dummyinc
 #CFLAGS = -g
 CFLAGS	=	-O2 -Wall -W -Wshadow #-pedantic -Werror -Wconversion
 
-LIBS	=	`./vsf_findlibs.sh`
+LIBS	=	-lwrap -lnsl -lpam -lcap -ldl -lcrypto
 LINK	=	-Wl,-s
 
 OBJS	=	main.o utility.o prelogin.o ftpcmdio.o postlogin.o privsock.o \
