diff -urN lxdm-0.2.99_git20100706.orig/src/lxdm.c lxdm-0.2.99_git20100706/src/lxdm.c
--- lxdm-0.2.99_git20100706.orig/src/lxdm.c	2010-07-08 10:38:58.000000000 +0200
+++ lxdm-0.2.99_git20100706/src/lxdm.c	2010-07-08 10:42:58.000000000 +0200
@@ -1140,7 +1140,7 @@
 		if(!strcasecmp(name,"LXDE"))
 			exec = g_strdup("startlxde");
 		else if( !strcasecmp(name, "GNOME") )
-			exec = g_strdup("gnome-session");
+			exec = g_strdup("gnome");
 		else if( !strcasecmp(name, "KDE") )
 			exec = g_strdup("startkde");
 		else if( !strcasecmp(name, "XFCE") || !strcasecmp(name, "xfce4"))
