diff -uNr old-lxdm-0.4.0.git.1310644223//src/lxdm.c lxdm-0.4.0.git.1310644223//src/lxdm.c
--- old-lxdm-0.4.0.git.1310644223//src/lxdm.c	2011-07-15 15:14:21.000000000 +0200
+++ lxdm-0.4.0.git.1310644223//src/lxdm.c	2011-07-15 15:15:27.939044540 +0200
@@ -1267,7 +1267,7 @@
 		if(!strcasecmp(name,"LXDE"))
 			exec = g_strdup("startlxde");
 		else if( !strcasecmp(name, "GNOME") )
-			exec = g_strdup("gnome-session");
+			exec = g_strdup("gnome");
 		else if( !strcasecmp(name, "KDE") )
 			exec = g_strdup("startkde");
 		else if( !strcasecmp(name, "XFCE") || !strcasecmp(name, "xfce4"))
