diff -uNr old-lxdm-0.4.0.git.1310644223//data/LoginReady.in lxdm-0.4.0.git.1310644223//data/LoginReady.in
--- old-lxdm-0.4.0.git.1310644223//data/LoginReady.in	2011-07-15 15:17:02.000000000 +0200
+++ lxdm-0.4.0.git.1310644223//data/LoginReady.in	2011-07-15 15:18:16.341044121 +0200
@@ -1,3 +1,2 @@
 #!/bin/sh
-#
-# Note: this is a sample and will not be run as is.
+test -x /etc/X11/xdm/Xsetup && exec /etc/X11/xdm/Xsetup
diff -uNr old-lxdm-0.4.0.git.1310644223//data/PostLogout.in lxdm-0.4.0.git.1310644223//data/PostLogout.in
--- old-lxdm-0.4.0.git.1310644223//data/PostLogout.in	2011-07-15 15:17:02.000000000 +0200
+++ lxdm-0.4.0.git.1310644223//data/PostLogout.in	2011-07-15 15:18:16.341044121 +0200
@@ -1,3 +1,2 @@
 #!/bin/sh
-#
-# Note: this is a sample and will not be run as is.
+test -x /etc/X11/xdm/Xreset && exec /etc/X11/xdm/Xreset
diff -uNr old-lxdm-0.4.0.git.1310644223//data/PreLogin.in lxdm-0.4.0.git.1310644223//data/PreLogin.in
--- old-lxdm-0.4.0.git.1310644223//data/PreLogin.in	2011-07-15 15:17:02.000000000 +0200
+++ lxdm-0.4.0.git.1310644223//data/PreLogin.in	2011-07-15 15:18:16.342044142 +0200
@@ -1,3 +1,2 @@
 #!/bin/sh
-#
-# Note: this is a sample and will not be run as is.
+test -x /etc/X11/xdm/Xstartup && exec /etc/X11/xdm/Xstartup
diff -uNr old-lxdm-0.4.0.git.1310644223//data/Xsession lxdm-0.4.0.git.1310644223//data/Xsession
--- old-lxdm-0.4.0.git.1310644223//data/Xsession	2011-07-15 15:17:02.000000000 +0200
+++ lxdm-0.4.0.git.1310644223//data/Xsession	2011-07-15 15:19:16.391044441 +0200
@@ -1,46 +1,26 @@
 #!/bin/bash
 
-# use bash for "exec -l", howto run login shell by /bin/sh ?
+failsafe="xterm -ls -T Failsafe -geometry 80x24+0+0"
+trap "exec $failsafe" EXIT SIGHUP SIGINT SIGPIPE SIGTERM SIGIO
+shopt -s execfail
 
-if [ $# -eq 1 -a -n "$1" ]; then
-	LXSESSION=$1
-else
 # default session
-	LXSESSION=/usr/bin/startlxde
-fi
+LXSESSION=/usr/bin/startlxde
+test $# -eq 1 -a -n "$1" && LXSESSION=$1
 
-[ -f /etc/xprofile ] && . /etc/xprofile
-[ -f ~/.xprofile ] && . ~/.xprofile
+export WINDOWMANAGER=$LXSESSION
+test -x /etc/X11/xdm/Xsession  && exec /etc/X11/xdm/Xsession "${LXSESSION}"
+test -x /etc/X11/xinit/xinitrc && exec -l $SHELL -c /etc/X11/xinit/xinitrc
 
-if [ -f /etc/X11/xinit/xinitrc-common ]; then
-# fedora
-	. /etc/X11/xinit/xinitrc-common
-	if ! [ -z "$XDG_SESSION_COOKIE" ]; then
-		CK_XINIT_SESSION=
-	elif [ -x /usr/bin/ck-launch-session -a -z "$CK_XINIT_SESSION" ]; then
-		CK_XINIT_SESSION="/usr/bin/ck-launch-session"
-	fi
-	exec -l $SHELL -c "$CK_XINIT_SESSION \"$LXSESSION\""
-elif [ -x /etc/X11/xinit/Xsession ]; then
-# fedora
-	exec /etc/X11/xinit/Xsession "$LXSESSION"
-elif [ -x /etc/X11/Xsession ]; then
-# mandriva, debian, ubuntu
-	exec /etc/X11/Xsession "$LXSESSION"
-elif [ -x /etc/X11/xinit/xinitrc ]; then
-#suse
-	export WINDOWMANAGER=$LXSESSION
-	exec -l $SHELL -c /etc/X11/xinit/xinitrc
-else
 # unknown, user should custom /etc/lxdm/xinitrc self
-	if [ -x /etc/lxdm/xinitrc ]; then
-		. /etc/lxdm/xinitrc
-	fi
-	if ! [ -z "$XDG_SESSION_COOKIE" ]; then
-		CK_XINIT_SESSION=
-	elif [ -x /usr/bin/ck-launch-session ]; then
-		CK_XINIT_SESSION="/usr/bin/ck-launch-session"
-	fi
-	exec -l $SHELL -c "$CK_XINIT_SESSION \"$LXSESSION\""
+test -s /etc/lxdm/xinitrc && . /etc/lxdm/xinitrc
+
+if ! test -z "$XDG_SESSION_COOKIE" ; then
+	CK_XINIT_SESSION=
+elif test -x /usr/bin/ck-launch-session ; then
+	CK_XINIT_SESSION="/usr/bin/ck-launch-session"
 fi
+exec -l $SHELL -c "$CK_XINIT_SESSION \"$LXSESSION\""
 
+# failsafe
+exit 1
