diff -uNr old-lxdm-0.4.0.git.1310644223//data/lxdm.conf.in lxdm-0.4.0.git.1310644223//data/lxdm.conf.in
--- old-lxdm-0.4.0.git.1310644223//data/lxdm.conf.in	2011-07-15 15:14:21.000000000 +0200
+++ lxdm-0.4.0.git.1310644223//data/lxdm.conf.in	2011-07-15 15:16:39.410044395 +0200
@@ -27,7 +27,7 @@
 gtk_theme=Clearlooks
 
 ## background of the greeter
-bg=/usr/share/backgrounds/default.png
+bg=/usr/share/wallpapers/default-1600x1200.png
 
 ## if show bottom pane
 bottom_pane=1
