diff -up GraphicsMagick-1.3.16/Magick++/bin/GraphicsMagick++-config.in.multilib GraphicsMagick-1.3.16/Magick++/bin/GraphicsMagick++-config.in
--- GraphicsMagick-1.3.16/Magick++/bin/GraphicsMagick++-config.in.multilib	2011-11-12 14:39:22.000000000 -0600
+++ GraphicsMagick-1.3.16/Magick++/bin/GraphicsMagick++-config.in	2012-06-24 11:25:12.603862643 -0500
@@ -33,13 +33,13 @@ while test $# -gt 0; do
       echo '@MAGICK_API_CPPFLAGS@'
       ;;
     --cxxflags)
-      echo '@CXXFLAGS@'
+      echo ''
       ;;
     --ldflags)
-      echo '@MAGICK_API_LDFLAGS@'
+      echo ''
       ;;
     --libs)
-      echo '-lGraphicsMagick++ @MAGICK_API_LIBS@'
+      echo '-lGraphicsMagick++'
       ;;
     *)
       echo "${usage}" 1>&2
diff -up GraphicsMagick-1.3.16/magick/GraphicsMagick-config.in.multilib GraphicsMagick-1.3.16/magick/GraphicsMagick-config.in
--- GraphicsMagick-1.3.16/magick/GraphicsMagick-config.in.multilib	2011-11-12 14:39:26.000000000 -0600
+++ GraphicsMagick-1.3.16/magick/GraphicsMagick-config.in	2012-06-24 11:14:55.947571850 -0500
@@ -30,16 +30,16 @@ while test $# -gt 0; do
       echo @PACKAGE_VERSION@
       ;;
     --cflags)
-      echo '@CFLAGS@'
+      echo ''
       ;;
     --cppflags)
       echo '@MAGICK_API_CPPFLAGS@'
       ;;
     --ldflags)
-      echo '@MAGICK_API_LDFLAGS@'
+      echo ''
       ;;
     --libs)
-      echo '@MAGICK_API_LIBS@'
+      echo '-lGraphicsMagick'
       ;;
     *)
       echo "${usage}" 1>&2
