--- configure.good	2007-03-24 13:06:30.000000000 -0800
+++ configure	2007-03-24 13:07:49.000000000 -0800
@@ -22501,39 +22501,39 @@
 
 
 
-  { echo "$as_me:$LINENO: checking for python module egg.trayicon" >&5
-echo $ECHO_N "checking for python module egg.trayicon... $ECHO_C" >&6; }
-
-  prog="
-import sys
-
-try:
-
-    import egg.trayicon
-
-    sys.exit(0)
-except ImportError:
-    sys.exit(1)
-except SystemExit:
-    raise
-except Exception, e:
-    print '  Error while trying to import egg.trayicon:'
-    print '    %r: %s' % (e, e)
-    sys.exit(1)"
-
-if $PYTHON -c "$prog" 1>&5 2>&5
-then
+#  { echo "$as_me:$LINENO: checking for python module egg.trayicon" >&5
+#echo $ECHO_N "checking for python module egg.trayicon... $ECHO_C" >&6; }
+#
+#  prog="
+#import sys
+#
+#try:
+#
+#    import egg.trayicon
+#
+#    sys.exit(0)
+#except ImportError:
+#    sys.exit(1)
+#except SystemExit:
+#    raise
+#except Exception, e:
+#    print '  Error while trying to import egg.trayicon:'
+#    print '    %r: %s' % (e, e)
+#    sys.exit(1)"
+#
+#if $PYTHON -c "$prog" 1>&5 2>&5
+#then
     { echo "$as_me:$LINENO: result: found" >&5
 echo "${ECHO_T}found" >&6; }
-    :
-else
-    { echo "$as_me:$LINENO: result: not found" >&5
-echo "${ECHO_T}not found" >&6; }
-    { { echo "$as_me:$LINENO: error: Unable to import egg.trayicon -- check your PYTHONPATH?" >&5
-echo "$as_me: error: Unable to import egg.trayicon -- check your PYTHONPATH?" >&2;}
-   { (exit 1); exit 1; }; }
-    EGG_SUPPORTED=no
-fi
+#    :
+#else
+#    { echo "$as_me:$LINENO: result: not found" >&5
+#echo "${ECHO_T}not found" >&6; }
+#    { { echo "$as_me:$LINENO: error: Unable to import egg.trayicon -- check your PYTHONPATH?" >&5
+#echo "$as_me: error: Unable to import egg.trayicon -- check your PYTHONPATH?" >&2;}
+#   { (exit 1); exit 1; }; }
+#    EGG_SUPPORTED=no
+#fi
 
 
 
