--- istanbul/main/screencast.py	2007-02-23 02:01:12.000000000 -0900
+++ istanbul/main/screencast.py	2008-05-02 21:40:39.000000000 -0800
@@ -135,7 +135,8 @@
             m.run()
             m.destroy()
             self.finished_callback(message)
-        elif message.type == gst.MESSAGE_EOS:
+# FIXME: should be fixed in oggmux in gstreamer not worked around
+        elif message.type == gst.MESSAGE_EOS or message.type == gst.MESSAGE_CLOCK_LOST:
             self._pipeline.set_state(gst.STATE_NULL)
             self.on_eos()
 
