Index: istanbul-0.2.2/istanbul/main/window_select.py
===================================================================
--- istanbul-0.2.2.orig/istanbul/main/window_select.py
+++ istanbul-0.2.2/istanbul/main/window_select.py
@@ -85,7 +85,7 @@ class WindowSelector:
                 return child
 
         for child in children:
-            result = self._get_client_window_check_children(window)
+            result = self._get_client_window_check_children(child)
             if result != None:
                 return result
 
