--- notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c
+++ notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c
@@ -499,7 +499,7 @@ static void
 draw_pie(GtkWidget *pie, WindowData *windata, cairo_t *cr)
 {
 	gdouble arc_angle = 1.0 - (gdouble)windata->remaining / (gdouble)windata->timeout;
-	cairo_set_source_rgba (cr, 1.0, 0.4, 0.0, 0.3);
+	cairo_set_source_rgba (cr, 0.6, 0.4, 0.4, 0.5);
 	cairo_move_to(cr, PIE_RADIUS, PIE_RADIUS);
 	cairo_arc_negative(cr, PIE_RADIUS, PIE_RADIUS, PIE_RADIUS,
 					-M_PI/2, (-0.25 + arc_angle)*2*M_PI);
