--- src/flyback.desktop.orig	2013-01-15 17:52:26.000000000 +0300
+++ src/flyback.desktop	2013-01-15 19:06:26.962630142 +0300
@@ -1,13 +1,15 @@
 [Desktop Entry]
 Version=0.5.0
 Encoding=UTF-8
-Name=Flyback Backup Manager
+Name=Flyback
 Type=Application
 GenericName=Backup Utility
+GenericName[tr]=Yedekleme Aracı
 Comment=Apple's Time Machine for Linux
-Icon=/usr/share/icons/gnome/scalable/actions/stock_save.svg
+Comment[tr]=Linux için Apple Time Machine 
+Icon=flyback
 FilePattern=flyback;
-Exec=/usr/share/flyback/flyback.py
+Exec=flyback
 Terminal=false
 StartupNotify=false
 Categories=Application;System
