diff -Naur kdebase-runtime-4.2.3-orig/knotify/kde.notifyrc kdebase-runtime-4.2.3/knotify/kde.notifyrc
--- kdebase-runtime-4.2.3-orig/knotify/kde.notifyrc	2009-05-22 19:03:32.692478771 +0300
+++ kdebase-runtime-4.2.3/knotify/kde.notifyrc	2009-05-22 19:10:44.681230582 +0300
@@ -1499,7 +1499,7 @@
 Comment[x-test]=xxKDE is exitingxx
 Comment[zh_CN]=KDE 正在退出
 Comment[zh_TW]=KDE 結束中
-Action=Sound
+Action=
 Sound=KDE-Sys-Log-Out.ogg
 
 [Event/cancellogout]
