commit e1940bcb4de9b52d47101df824c9c8dec8749a64
Author: Dawit Alemayehu <adawit@kde.org>
Date:   Fri Nov 4 01:48:12 2011 -0400

    Added maxInstances and maxInstancesPerHost properties.
    
    BUG: 250882
    FIXED-IN: 4.7.4

diff --git a/kioslave/smb/smb.protocol b/kioslave/smb/smb.protocol
index f749d67..654bcfb 100644
--- a/kioslave/smb/smb.protocol
+++ b/kioslave/smb/smb.protocol
@@ -10,5 +10,7 @@ makedir=true
 deleting=true
 opening=true
 moving=true
+maxInstances=20
+maxInstancesPerHost=5
 X-DocPath=kioslave/smb/index.html
 Icon=network-workgroup
