Index: nepomuk/services/strigi/strigiserviceconfig.cpp
===================================================================
--- nepomuk/services/strigi/strigiserviceconfig.cpp.orig
+++ nepomuk/services/strigi/strigiserviceconfig.cpp
@@ -72,6 +72,10 @@ QStringList Nepomuk::StrigiServiceConfig
     return fl;
 }
 
+bool Nepomuk::StrigiServiceConfig::systrayEnabled() const
+{
+    return m_config.group( "General" ).readEntry( "enable systray applet", false );
+}
 
 QStringList Nepomuk::StrigiServiceConfig::excludeFolders() const
 {
Index: nepomuk/services/strigi/strigiserviceconfig.h
===================================================================
--- nepomuk/services/strigi/strigiserviceconfig.h.orig
+++ nepomuk/services/strigi/strigiserviceconfig.h
@@ -60,6 +60,8 @@ namespace Nepomuk {
 
         QStringList excludeFilters() const;
 
+        bool systrayEnabled() const;
+
         QList<QRegExp> excludeFilterRegExps() const { return m_excludeFilterRegExpCache; }
 
         bool indexHiddenFolders() const;
Index: nepomuk/services/strigi/strigiservice.cpp
===================================================================
--- nepomuk/services/strigi/strigiservice.cpp.orig
+++ nepomuk/services/strigi/strigiservice.cpp
@@ -86,8 +86,11 @@ Nepomuk::StrigiService::StrigiService( Q
         // create the status widget (hidden)
         StatusWidget* sw = new StatusWidget( mainModel(), this );
 
-        // create the systray
-        new SystemTray( this, sw );
+        if (StrigiServiceConfig::self()->systrayEnabled())
+        {
+            // create the systray
+            new SystemTray( this, sw );
+        }
 
         // setup status connections
         connect( m_indexScheduler, SIGNAL( indexingStarted() ),
