diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kdebase-runtime-4.2.1-orig/knetattach/knetattach.cpp kdebase-runtime-4.2.1/knetattach/knetattach.cpp
--- kdebase-runtime-4.2.1-orig/knetattach/knetattach.cpp	2009-03-06 14:32:40.000000000 +0200
+++ kdebase-runtime-4.2.1/knetattach/knetattach.cpp	2009-03-06 15:02:34.000000000 +0200
@@ -65,7 +65,14 @@
     }
     _encoding->clear();
     _encoding->addItems(KGlobal::charsets()->descriptiveEncodingNames());
-    _encoding->setCurrentIndex(0);    
+
+    int utfIndex = _encoding->findText("UTF-8", Qt::MatchContains);
+
+    if (utfIndex == -1)
+        _encoding->setCurrentIndex(0);
+    else
+        _encoding->setCurrentIndex(utfIndex);
+
 }
 
 void KNetAttach::slotHelpClicked()
