Index: tig-0.17/tig.h
===================================================================
--- tig-0.17.orig/tig.h
+++ tig-0.17/tig.h
@@ -133,6 +133,20 @@ name(type **mem, size_t size, size_t inc
 	strncmp(str1, str2, STRING_SIZE(str2))
 
 static inline int
+ascii_toupper(int c) {
+    if (c >= 'a' && c <= 'z')
+        c &= ~0x20;
+    return c;
+}
+
+static inline int
+ascii_tolower(int c) {
+    if (c >= 'A' && c <= 'Z')
+        c |= ~0x20;
+    return c;
+}
+
+static inline int
 suffixcmp(const char *str, int slen, const char *suffix)
 {
 	size_t len = slen >= 0 ? slen : strlen(str);
@@ -245,7 +259,7 @@ string_enum_compare(const char *str1, co
 
 	/* Diff-Header == DIFF_HEADER */
 	for (i = 0; i < len; i++) {
-		if (toupper(str1[i]) == toupper(str2[i]))
+		if (ascii_toupper(str1[i]) == ascii_toupper(str2[i]))
 			continue;
 
 		if (string_enum_sep(str1[i]) &&
@@ -268,7 +282,7 @@ enum_map_name(const char *name, size_t n
 	int bufpos;
 
 	for (bufpos = 0; bufpos <= namelen; bufpos++) {
-		buf[bufpos] = tolower(name[bufpos]);
+		buf[bufpos] = ascii_tolower(name[bufpos]);
 		if (buf[bufpos] == '_')
 			buf[bufpos] = '-';
 	}
