diff --git a/src/rtcpcompoundpacketbuilder.cpp b/src/rtcpcompoundpacketbuilder.cpp
index 8172007..8fd4510 100644
--- a/src/rtcpcompoundpacketbuilder.cpp
+++ b/src/rtcpcompoundpacketbuilder.cpp
@@ -30,6 +30,8 @@
 
 */
 
+#include <cstring>
+
 #include "rtcpcompoundpacketbuilder.h"
 #include "rtcpsrpacket.h"
 #include "rtcprrpacket.h"
diff --git a/src/rtppacket.cpp b/src/rtppacket.cpp
index b6d5fda..8c516c7 100644
--- a/src/rtppacket.cpp
+++ b/src/rtppacket.cpp
@@ -30,6 +30,8 @@
 
 */
 
+#include <cstring>
+
 #include "rtppacket.h"
 #include "rtpstructs.h"
 #include "rtpdefines.h"

