diff -Nuar ois-v1-3.OSmanOS/demos/FFConsoleDemo.cpp ois-v1-3/demos/FFConsoleDemo.cpp
--- ois-v1-3.OSmanOS/demos/FFConsoleDemo.cpp	2010-04-01 06:38:04.000000000 +0300
+++ ois-v1-3/demos/FFConsoleDemo.cpp	2013-03-10 08:37:33.727940247 +0200
@@ -7,6 +7,7 @@
 #include <ios>
 #include <sstream>
 #include <vector>
+#include <unistd.h>
 
 using namespace std;
 
diff -Nuar ois-v1-3.OSmanOS/demos/OISConsole.cpp ois-v1-3/demos/OISConsole.cpp
--- ois-v1-3.OSmanOS/demos/OISConsole.cpp	2010-08-16 01:51:34.000000000 +0300
+++ ois-v1-3/demos/OISConsole.cpp	2013-03-10 08:38:15.940937658 +0200
@@ -12,6 +12,7 @@
 #include <iostream>
 #include <vector>
 #include <sstream>
+#include <unistd.h>
 
 ////////////////////////////////////Needed Windows Headers////////////
 #if defined OIS_WIN32_PLATFORM
@@ -77,7 +78,7 @@
 	}
 	bool keyReleased( const KeyEvent &arg ) {
 		if( arg.key == KC_ESCAPE || arg.key == KC_Q )
-			appRunning = false;
+			appRunning = false;
 		std::cout << "KeyReleased {" << ((Keyboard*)(arg.device))->getAsString(arg.key) << "}\n";
 		return true;
 	}
diff -Nuar ois-v1-3.OSmanOS/src/linux/LinuxForceFeedback.cpp ois-v1-3/src/linux/LinuxForceFeedback.cpp
--- ois-v1-3.OSmanOS/src/linux/LinuxForceFeedback.cpp	2010-04-01 06:06:55.000000000 +0300
+++ ois-v1-3/src/linux/LinuxForceFeedback.cpp	2013-03-10 08:36:14.635945097 +0200
@@ -26,6 +26,7 @@
 #include <cstdlib>
 #include <errno.h>
 #include <memory.h>
+#include <unistd.h>
 
 using namespace OIS;
 
diff -Nuar ois-v1-3.OSmanOS/src/linux/LinuxJoyStickEvents.cpp ois-v1-3/src/linux/LinuxJoyStickEvents.cpp
--- ois-v1-3.OSmanOS/src/linux/LinuxJoyStickEvents.cpp	2010-07-26 04:28:41.000000000 +0300
+++ ois-v1-3/src/linux/LinuxJoyStickEvents.cpp	2013-03-10 08:38:50.739935524 +0200
@@ -33,6 +33,7 @@
 #include <fcntl.h>        //Needed to Open a file descriptor
 #include <cassert>	
 #include <linux/input.h>
+#include <unistd.h>
 
 
 #include <sstream>
