--- rocksndiamonds-3.2.2/src/libgame/setup.c.hs	2006-08-20 14:05:41.000000000 +0200
+++ rocksndiamonds-3.2.2/src/libgame/setup.c	2006-10-14 14:56:13.000000000 +0200
@@ -126,7 +126,11 @@
 static char *getScoreDir(char *level_subdir)
 {
   static char *score_dir = NULL;
+#ifdef USE_USERDATADIR_FOR_COMMONDATA
+  char *data_dir = getUserGameDataDir();
+#else
   char *data_dir = getCommonDataDir();
+#endif
   char *score_subdir = SCORES_DIRECTORY;
 
   checked_free(score_dir);
@@ -781,7 +785,11 @@
 
 void InitScoreDirectory(char *level_subdir)
 {
+#ifdef USE_USERDATADIR_FOR_COMMONDATA
+  createDirectory(getUserGameDataDir(), "user data", PERMS_PRIVATE);
+#else
   createDirectory(getCommonDataDir(), "common data", PERMS_PUBLIC);
+#endif
   createDirectory(getScoreDir(NULL), "main score", PERMS_PUBLIC);
   createDirectory(getScoreDir(level_subdir), "level score", PERMS_PUBLIC);
 }
