--- usr/lib/byobu/wifi_quality.orig	2010-03-21 14:00:16.031543814 +0200
+++ usr/lib/byobu/wifi_quality	2010-03-21 14:00:43.119527288 +0200
@@ -21,11 +21,11 @@
 color 2>/dev/null || color() { true; }
 
 if [ "$1" = "--detail" ]; then
-	/sbin/iwconfig 2>/dev/null
+	/usr/sbin/iwconfig 2>/dev/null
 	exit 0
 fi
 
-iwconfig=`/sbin/iwconfig 2>/dev/null`
+iwconfig=`/usr/sbin/iwconfig 2>/dev/null`
 bitrate=`echo "$iwconfig" | grep "Bit Rate." | sed -e "s/^.*Bit Rate.//" -e "s/ .*$//g"`
 [ -z "$bitrate" ] && exit 0
 quality=`echo "$iwconfig" | grep "Link Quality." | sed -e "s/^.*Link Quality.//" -e "s/ .*$//g" | awk -F/ '{printf "%.0f", 100*$1/$2}'`
