--- usr/lib/byobu/reboot_required.orig	2011-01-02 00:41:20.176000033 +0200
+++ usr/lib/byobu/reboot_required	2011-01-02 00:43:21.385000027 +0200
@@ -21,7 +21,7 @@
 DATA="$HOME/.$PKG"
 color 2>/dev/null || color() { true; }
 
-reboot="/var/run/reboot-required"
+reboot="/var/lib/pisi/info/needsreboot"
 [ -d "/var/run/screen/S-$USER" ] && DIR="/var/run/screen/S-$USER" || DIR="$DATA"
 reload="$DIR/$PKG.reload-required"
 case "$1" in
