From ca98a014e25e60c547d9fe7692ea4e58a15bad4a Mon Sep 17 00:00:00 2001
From: Erich Hoover <ehoover@mines.edu>
Date: Thu, 20 Dec 2012 09:52:19 -0700
Subject: ws2_32: Implement SIO_ADDRESS_LIST_CHANGE with NotifyAddrChange.

---
 dlls/ws2_32/socket.c |   14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/dlls/ws2_32/socket.c b/dlls/ws2_32/socket.c
index eeb5265..55fe615 100644
--- a/dlls/ws2_32/socket.c
+++ b/dlls/ws2_32/socket.c
@@ -3386,10 +3386,18 @@ INT WINAPI WSAIoctl(SOCKET s, DWORD code, LPVOID in_buff, DWORD in_size, LPVOID
        }
 
    case WS_SIO_ADDRESS_LIST_CHANGE:
-       FIXME("-> SIO_ADDRESS_LIST_CHANGE request: stub\n");
-       /* FIXME: error and return code depend on whether socket was created
-        * with WSA_FLAG_OVERLAPPED, but there is no easy way to get this */
+   {
+       HANDLE handle;
+
+       TRACE("-> SIO_ADDRESS_LIST_CHANGE request\n");
+
+       if (overlapped || _is_blocking(s))
+           status = NotifyAddrChange(&handle, overlapped);
+       else
+           status = WSAEWOULDBLOCK;
+       overlapped = NULL; /* managed by NotifyAddrChange */
        break;
+   }
 
    case WS_SIO_ADDRESS_LIST_QUERY:
    {
-- 
1.7.9.5

