diff -up enblend-enfuse-4.1.1/doc/Makefile.am.doc enblend-enfuse-4.1.1/doc/Makefile.am
diff -up enblend-enfuse-4.1.1/doc/Makefile.in.doc enblend-enfuse-4.1.1/doc/Makefile.in
--- enblend-enfuse-4.1.1/doc/Makefile.in.doc	2013-02-15 08:14:30.000000000 -0600
+++ enblend-enfuse-4.1.1/doc/Makefile.in	2013-03-14 12:18:42.002825015 -0500
@@ -387,12 +387,12 @@ smooth-edge.gp: $(top_builddir)/config.s
 	rm -rf $$backupdir; exit $$rc
 
 .texi.dvi:
-	TEXINPUTS="$(am__TEXINFO_TEX_DIR)$(PATH_SEPARATOR)$$TEXINPUTS" \
+	TEXINPUTS="$(am__TEXINFO_TEX_DIR)$(PATH_SEPARATOR)$$TEXINPUTS:" \
 	MAKEINFO='$(MAKEINFO) $(AM_MAKEINFOFLAGS) $(MAKEINFOFLAGS) -I $(srcdir)' \
 	$(TEXI2DVI) $<
 
 .texi.pdf:
-	TEXINPUTS="$(am__TEXINFO_TEX_DIR)$(PATH_SEPARATOR)$$TEXINPUTS" \
+	TEXINPUTS="$(am__TEXINFO_TEX_DIR)$(PATH_SEPARATOR)$$TEXINPUTS:" \
 	MAKEINFO='$(MAKEINFO) $(AM_MAKEINFOFLAGS) $(MAKEINFOFLAGS) -I $(srcdir)' \
 	$(TEXI2PDF) $<
 
@@ -450,7 +450,7 @@ mostlyclean-1:
 maintainer-clean-1:
 	-rm -f $(srcdir)/stamp-1 $(srcdir)/versenfuse.texi
 .dvi.ps:
-	TEXINPUTS="$(am__TEXINFO_TEX_DIR)$(PATH_SEPARATOR)$$TEXINPUTS" \
+	TEXINPUTS="$(am__TEXINFO_TEX_DIR)$(PATH_SEPARATOR)$$TEXINPUTS:" \
 	$(DVIPS) -o $@ $<
 
 uninstall-dvi-am:
@@ -788,7 +788,7 @@ uninstall-am: uninstall-dvi-am uninstall
 	uninstall-dvi-am uninstall-html-am uninstall-info-am \
 	uninstall-pdf-am uninstall-ps-am
 
-export TEXINPUTS=$(top_builddir):$(srcdir)
+export TEXINPUTS=$(top_builddir):$(srcdir):
 
 # Phony Targets
  
