Index: enblend-enfuse-4.0-753b534c819d/configure.in
===================================================================
--- enblend-enfuse-4.0-753b534c819d.orig/configure.in
+++ enblend-enfuse-4.0-753b534c819d/configure.in
@@ -220,7 +220,7 @@ AS_IF([test "x$with_boost_filesystem" = 
                               AC_MSG_NOTICE([Boost "filesystem" header is missing.]))
               if test x$ac_cv_header_boost_filesystem_hpp = xyes; then
                   found_boost_filesystem_lib=no
-                  candidates="-lboost_filesystem"
+                  candidates="-lboost_filesystem -lboost_system"
                   if test x$acx_pthread_ok = xyes; then
                       candidates="-lboost_filesystem-mt $candidates"
                   fi
