Index: enblend-enfuse-4.0-753b534c819d/configure.in
===================================================================
--- enblend-enfuse-4.0-753b534c819d.orig/configure.in
+++ enblend-enfuse-4.0-753b534c819d/configure.in
@@ -313,116 +313,10 @@ else
     HELP2MAN=:
 fi
 
-can_build_doc=yes
-AX_PROG_PERL_MODULES(File::Basename, [],
-                     [AC_MSG_WARN(missing Perl module File::Basename)
-                      can_build_doc=no
-                      missing_for_doc="$missing_for_doc File::Basename"])
-AX_PROG_PERL_MODULES(IO::File, [],
-                     [AC_MSG_WARN(missing Perl module IO::File)
-                      can_build_doc=no
-                      missing_for_doc="$missing_for_doc IO::File"])
-AX_PROG_PERL_MODULES(IO::Handle, [],
-                     [AC_MSG_WARN(missing Perl module IO::Handle)
-                      can_build_doc=no
-                      missing_for_doc="$missing_for_doc IO::Handle"])
-
-AX_WITH_PROG(GNUPLOT, gnuplot, false, [])
-if test "x$GNUPLOT" = xfalse; then
-    AC_MSG_WARN(cannot find gnuplot; will not be able to build documentation)
-    can_build_doc=no
-    missing_for_doc="$missing_for_doc gnuplot"
-fi
-
-AC_ARG_WITH([ttf-path],
-            AC_HELP_STRING([--with-ttf-path=<PATH>],
-                           [set path to TrueType fonts]),
-            [GDFONTPATH="$withval"],
-            [GDFONTPATH="/opt/local/share/fonts/freefont-ttf:/sw/share/fonts/freefont-ttf:/usr/X11/lib/X11/fonts/TTF:/usr/X11R6/lib/X11/fonts/TTF:/usr/share/fonts/truetype/freefont"])
-AC_ARG_VAR(GDFONTPATH, [path to TrueType fonts])
-have_freesans_font=no
-orig_ifs="$IFS"
-IFS=':'
-for dir in $GDFONTPATH; do \
-    if test -f "$dir/FreeSans.ttf"; then \
-        have_freesans_font=yes; \
-        break; \
-    fi; \
-done
-IFS="$orig_ifs"
-if test $have_freesans_font != yes; then
-    AC_MSG_WARN(cannot find font "FreeSans.ttf" anywhere in GDFONTPATH; text in the documentation's figures could look strange)
-fi
-
-AC_CHECK_PROG(FIG2DEV,
-              fig2dev,
-              fig2dev,
-              false)
-if test "x$FIG2DEV" = xfalse; then
-    AC_MSG_WARN(cannot find fig2dev; will not be able to build documentation)
-    can_build_doc=no
-    missing_for_doc="$missing_for_doc fig2dev"
-fi
+can_build_doc=no
 
 AC_PROG_SED
 
-# These variables are mentioned in the AutoMake documentation as being
-# influential to the documentation build process.
-AC_CHECK_PROG(MAKEINFO,
-              makeinfo,
-              makeinfo,
-              false)
-if test "x$MAKEINFO" = xfalse; then
-    AC_MSG_WARN(cannot find makeinfo; will not be able to build documentation)
-    can_build_doc=no
-    missing_for_doc="$missing_for_doc makeinfo"
-fi
-
-dnl AC_CHECK_PROG(TEXI2DVI,
-dnl               texi2dvi,
-dnl               texi2dvi,
-dnl               false)
-dnl if test "x$TEXI2DVI" = xfalse; then
-dnl   AC_MSG_WARN(cannot find texi2dvi; will not be able to build documentation)
-dnl   can_build_doc=no
-dnl   missing_for_doc="$missing_for_doc texi2dvi"
-dnl fi
-
-dnl AC_CHECK_PROG(TEXI2PDF,
-dnl               texi2pdf,
-dnl               texi2pdf,
-dnl               false)
-dnl if test "x$TEXI2PDF" = xfalse; then
-dnl   AC_MSG_WARN(cannot find texi2pdf; will not be able to build documentation)
-dnl   can_build_doc=no
-dnl   missing_for_doc="$missing_for_doc texi2pdf"
-dnl fi
-
-dnl AC_CHECK_PROG(DVIPS,
-dnl               dvips,
-dnl               dvips,
-dnl               false)
-dnl if test "x$DVIPS" = xfalse; then
-dnl   AC_MSG_WARN(cannot find dvips; will not be able to build documentation)
-dnl   can_build_doc=no
-dnl   missing_for_doc="$missing_for_doc dvips"
-dnl fi
-
-AC_CHECK_PROG(TIDY,
-              tidy,
-              tidy,
-              false)
-if test "x$TIDY" = xfalse; then
-    AC_MSG_WARN(cannot find tidy; will not be able to build XHTML documentation)
-    can_build_doc=no
-    missing_for_doc="$missing_for_doc tidy"
-fi
-
-AM_CONDITIONAL([BUILD_DOC], [test $can_build_doc = yes])
-if test $can_build_doc = no; then
-    no_doc_reason=", because of missing$missing_for_doc"
-fi
-
 AC_MSG_CHECKING([checking whether split documentation files])
 split_doc_default="yes"
 AC_ARG_ENABLE([split-doc],
@@ -511,21 +405,13 @@ else
     enable_openmp=yes
 fi
 
-AC_CONFIG_FILES([doc/entropy.gp
-                 doc/entropy-cutoff.gp
-                 doc/gaussian.gp
-                 doc/laplacian-of-gaussian.gp
-                 doc/sharp-edge.gp
-                 doc/smooth-edge.gp])
-
 AC_CONFIG_FILES([Makefile
                  include/Makefile
                  include/vigra/Makefile
                  include/vigra_ext/Makefile
                  src/Makefile
                  src/vigra_impex/Makefile
-                 src/win32helpers/Makefile
-                 doc/Makefile])
+                 src/win32helpers/Makefile])
 AC_OUTPUT
 
 # AC_OUTPUT has created "config.h"
Index: enblend-enfuse-4.0-753b534c819d/Makefile.am
===================================================================
--- enblend-enfuse-4.0-753b534c819d.orig/Makefile.am
+++ enblend-enfuse-4.0-753b534c819d/Makefile.am
@@ -1,26 +1,9 @@
 EXTRA_DIST = VIGRA_LICENSE README.txt VERSION
 
-if BUILD_DOC
-  MAYBE_DOC = doc
-endif
-
-SUBDIRS = include src $(MAYBE_DOC)
-DIST_SUBDIRS = include src doc
+SUBDIRS = include src
+DIST_SUBDIRS = include src
 
 ACLOCAL_AMFLAGS = -I m4
 
 # created by configure(1)
 DISTCLEANFILES = config-h.texi
-
-
-.PHONY: xhtml
-xhtml:
-	$(MAKE) --directory=doc $@
-
-.PHONY: install-xhtml
-install-xhtml:
-	$(MAKE) --directory=doc $@
-
-.PHONY: uninstall-xhtml
-uninstall-xhtml:
-	$(MAKE) --directory=doc $@
