--- source/common/unicode/rbbi.h	2006-08-11 08:46:40.000000000 +0300
+++ source/common/unicode/rbbi.h	2007-11-15 04:08:17.000000000 +0200
@@ -611,12 +611,14 @@
     virtual int32_t getBreakType() const;
 #endif
 
+public:
     /**
       * Set the type of the break iterator.
       * @internal
       */
     virtual void setBreakType(int32_t type);
 
+protected:
     /**
       * Common initialization function, used by constructors and bufferClone.
       *   (Also used by DictionaryBasedBreakIterator::createBufferClone().)
