diff -urN libwnck-2.16.3/libwnck/window.c libwnck/libwnck/window.c
--- libwnck-2.16.3/libwnck/window.c     2007-01-29 15:56:52.000000000 -0600
+++ libwnck/libwnck/window.c    2007-01-30 05:54:39.000000000 -0600
@@ -2204,6 +2204,12 @@
     window->priv->actions |=
         WNCK_WINDOW_ACTION_MAXIMIZE   |
         WNCK_WINDOW_ACTION_UNMAXIMIZE;
+
+  /* There's no _NET_WM_ACTION_ABOVE hint so always add it if
+     window manager claims to support_NET_WM_STATE_ABOVE */
+  if (wnck_screen_net_wm_supports (wnck_window_get_screen (window),
+                                   "_NET_WM_STATE_ABOVE"))
+      window->priv->actions |= WNCK_WINDOW_ACTION_ABOVE;
 }
 
 static void