diff -ur pnm2ppa-1.12/Makefile pnm2ppa-1.12.new/Makefile
--- Makefile	2002-11-15 18:29:29.000000000 -0500
+++ Makefile	2002-12-11 17:39:27.000000000 -0500
@@ -5,13 +5,13 @@
 # paths to locations where the pnm2ppa binary, the configuration file,
 # and the man page should be installed.   On some systems (like Solaris)
 # the install command requires the syntax BINDIR="-f /usr/local/bin".
-BINDIR="/usr/local/bin"
+BINDIR="/usr/bin"
 CONFDIR="/etc"
-MANDIR="/usr/local/man/man1"
+MANDIR="/usr/share/man/man1"
 
 #compiler and linker options:
 CC=gcc
-CFLAGS=-Wall -pedantic -O2  
+CFLAGS=-Wall -pedantic -O3
 LDFLAGS=-lm
 
 INSTALL=install
diff -ur pnm2ppa-1.12/lpd/ascii.if pnm2ppa-1.12.new/lpd/ascii.if
--- lpd/ascii.if	2000-10-28 10:58:07.000000000 -0400
+++ lpd/ascii.if	2002-12-11 17:39:55.000000000 -0500
@@ -1,6 +1,6 @@
 #! /bin/sh
 # for printing black and white text (ascii) files.
-FILTER="/usr/local/bin/bw.if"
+FILTER="/usr/bin/bw.if"
 #FILTER="/usr/local/bin/bweco.if"
 #------enscript command line--------------------------
 enscript -2rj -p- | $FILTER
diff -ur pnm2ppa-1.12/lpd/bw.if pnm2ppa-1.12.new/lpd/bw.if
--- lpd/bw.if	2000-10-28 10:58:07.000000000 -0400
+++ lpd/bw.if	2002-12-11 17:40:14.000000000 -0500
@@ -6,7 +6,7 @@
 #PAPER=a4
 #PAPER=a3
 #PPAOPTS="-B 2"
-PNM2PPA="/usr/local/bin/pnm2ppa"
+PNM2PPA="/usr/bin/pnm2ppa"
 #------------------dont change below this line----------------------
 gs -sDEVICE=pbmraw -sPAPERSIZE=$PAPER -q -dNOPAUSE -r600 -sOutputFile=- - | \
 $PNM2PPA --bw $PPAOPTS -i - -o -
diff -ur pnm2ppa-1.12/lpd/bweco.if pnm2ppa-1.12.new/lpd/bweco.if
--- lpd/bweco.if	2000-10-28 10:58:07.000000000 -0400
+++ lpd/bweco.if	2002-12-11 17:40:23.000000000 -0500
@@ -6,7 +6,7 @@
 #PAPER=a4
 #PAPER=a3
 #PPAOPTS="-B 2"
-PNM2PPA="/usr/local/bin/pnm2ppa"
+PNM2PPA="/usr/bin/pnm2ppa"
 #------------------dont change below this line----------------------
 gs -sDEVICE=pbmraw -sPAPERSIZE=$PAPER -q -dNOPAUSE -r600 -sOutputFile=- - | \
 $PNM2PPA --bw --eco $PPAOPTS -i - -o -
diff -ur pnm2ppa-1.12/lpd/color.if pnm2ppa-1.12.new/lpd/color.if
--- lpd/color.if	2000-10-28 10:58:07.000000000 -0400
+++ lpd/color.if	2002-12-11 17:40:31.000000000 -0500
@@ -6,7 +6,7 @@
 #PAPER=a4
 #PAPER=a3
 #PPAOPTS="-p"
-PNM2PPA="/usr/local/bin/pnm2ppa"
+PNM2PPA="/usr/bin/pnm2ppa"
 #------------------dont change below this line----------------------
 gs -sDEVICE=ppmraw -sPAPERSIZE=$PAPER -q -dNOPAUSE -r600 -sOutputFile=- - | \
 $PNM2PPA $PPAOPTS -i - -o -
diff -ur pnm2ppa-1.12/lpd/coloreco.if pnm2ppa-1.12.new/lpd/coloreco.if
--- lpd/coloreco.if	2000-10-28 10:58:07.000000000 -0400
+++ lpd/coloreco.if	2002-12-11 17:40:45.000000000 -0500
@@ -6,7 +6,7 @@
 #PAPER=a4
 #PAPER=a3
 #PPAOPTS="--fd"
-PNM2PPA="/usr/local/bin/pnm2ppa"
+PNM2PPA="/usr/bin/pnm2ppa"
 #------------------dont change below this line----------------------
 gs -sDEVICE=ppmraw -sPAPERSIZE=$PAPER -q -dNOPAUSE -r600 -sOutputFile=- - | \
 $PNM2PPA --eco $PPAOPTS -i - -o -
diff -ur pnm2ppa-1.12/lpd/printcap pnm2ppa-1.12.new/lpd/printcap
--- lpd/printcap	2000-10-27 14:12:23.000000000 -0400
+++ lpd/printcap	2002-12-11 17:41:35.000000000 -0500
@@ -6,7 +6,7 @@
         :sd=/var/spool/lpd/coloreco:\
         :lf=/var/spool/lpd/coloreco/log:\
         :af=/var/spool/lpd/coloreco/acct:\
-        :if=/usr/local/bin/coloreco.if:\
+        :if=/usr/bin/coloreco.if:\
         :la:mx#0:\
         :sh:sf:
 
@@ -16,7 +16,7 @@
         :sd=/var/spool/lpd/color:\
         :lf=/var/spool/lpd/color/log:\
         :af=/var/spool/lpd/color/acct:\
-        :if=/usr/local/bin/color.if:\
+        :if=/usr/bin/color.if:\
         :la:mx#0:\
         :sh:sf:
 
@@ -26,7 +26,7 @@
         :sd=/var/spool/lpd/bw:\
         :lf=/var/spool/lpd/bw/log:\
         :af=/var/spool/lpd/bw/acct:\
-        :if=/usr/local/bin/bw.if:\
+        :if=/usr/bin/bw.if:\
         :la:mx#0:\
         :sh:sf:
 
@@ -36,7 +36,7 @@
         :sd=/var/spool/lpd/bweco:\
         :lf=/var/spool/lpd/bweco/log:\
         :af=/var/spool/lpd/bweco/acct:\
-        :if=/usr/local/bin/bweco.if:\
+        :if=/usr/bin/bweco.if:\
         :la:mx#0:\
         :sh:sf:
 
@@ -46,7 +46,7 @@
         :sd=/var/spool/lpd/ascii:\
         :lf=/var/spool/lpd/ascii/log:\
         :af=/var/spool/lpd/ascii/acct:\
-        :if=/usr/local/bin/ascii.if:\
+        :if=/usr/bin/ascii.if:\
         :la:mx#0:\
         :sh:sf:
 
