--- plasma-widget-message-indicator-0.5.8-orig/CMakeLists.txt	2011-08-31 19:21:30.000000000 +0300
+++ plasma-widget-message-indicator-0.5.8/CMakeLists.txt	2013-01-20 16:46:19.248862640 +0200
@@ -6,7 +6,6 @@
 find_package(KDE4 4.4 REQUIRED)
 include(KDE4Defaults)
 find_package(PkgConfig REQUIRED)
-pkg_check_modules(INDICATE REQUIRED indicate-0.6)
 pkg_check_modules(INDICATEQT REQUIRED indicate-qt>=0.2.5)
 pkg_check_modules(DBUSMENUQT REQUIRED dbusmenu-qt>=0.3.0)
 
 
