--- php-5.2.14.orig/ext/zip/php_zip.c	2011-01-14 16:12:17.886000067 +0200
+++ php-5.2.14/ext/zip/php_zip.c	2011-01-14 16:14:50.233000067 +0200
@@ -1954,6 +1954,9 @@
 
 	PHP_ZIP_STAT_INDEX(intern, index, 0, sb);
 	comment = zip_get_file_comment(intern, index, &comment_len, (int)flags);
+   if(comment==NULL) {
+       RETURN_FALSE;
+   }
 	RETURN_STRINGL((char *)comment, (long)comment_len, 1);
 }
 /* }}} */
