Index: clustalx-2.0.12/ClustalQtParams.h
===================================================================
--- ClustalQtParams.h	2009-09-18 16:11:15.000000000 +0200
+++ ClustalQtParams_mod.h	2009-11-01 19:25:29.429596067 +0100
@@ -13,9 +13,9 @@
 #include "clustalW/alignment/Alignment.h"
 #include <vector>
 
-static QString psColorFile = "colprot.xml";
-static QString psDNAFile = "coldna.xml";
-static QString psPrintFile = "colprint.xml";
+static QString psColorFile = "/usr/share/clustalx/colprot.xml";
+static QString psDNAFile = "/usr/share/clustalx/coldna.xml";
+static QString psPrintFile = "/usr/share/clustalx/colprint.xml";
 
 static const int BlackAndWhite = 1;
 static const int ColorsBackGround = 2;
