Index: clustalx-2.0.12/HelpDisplayWidget.cpp
===================================================================
--- HelpDisplayWidget.cpp	2009-11-21 21:58:44.250522045 +0100
+++ HelpDisplayWidget_mod.cpp	2009-11-21 21:57:40.317237004 +0100
@@ -19,7 +19,7 @@
 #include <QtGui>
 #include "Resources.h"
 
-const string HelpDisplayWidget::helpFileName = "clustalx.hlp";
+const string HelpDisplayWidget::helpFileName = "/usr/share/clustalx/clustalx.hlp";
 
 HelpDisplayWidget::HelpDisplayWidget(char helpPointer)
 {
