Index: gambas3-2.99.2/gb.sdl/src/SDLfont.cpp
===================================================================
--- gambas3-2.99.2.orig/gb.sdl/src/SDLfont.cpp
+++ gambas3-2.99.2/gb.sdl/src/SDLfont.cpp
@@ -33,7 +33,7 @@
 #include <ctype.h>
 
 #define DEFAULT_FONT_SIZE 20
-#define DEFAULT_FONT "DejaVuSans.ttf"
+#define DEFAULT_FONT "dejavu/DejaVuSans.ttf"
 /*
 #ifndef TTF_STYLE_STRIKETHROUGH
 #define TTF_STYLE_STRIKETHROUGH	0x08
@@ -149,7 +149,7 @@ SDLfont::SDLfont(const char *fontfile)
 	if (!fontfile)
 	{
 		hfontname = GB.System.Path();
-		hfontname += "/share/gambas" GAMBAS_VERSION_STRING "/gb.sdl/" DEFAULT_FONT;
+		hfontname += "/share/fonts/" DEFAULT_FONT;
 	}
 	else
 		hfontname = fontfile;
