Index: kdelibs-4.6.5/plasma/theme.cpp
===================================================================
--- kdelibs-4.6.5.orig/plasma/theme.cpp
+++ kdelibs-4.6.5/plasma/theme.cpp
@@ -52,8 +52,8 @@
 namespace Plasma
 {
 
-#define DEFAULT_WALLPAPER_THEME "Horos"
-#define DEFAULT_WALLPAPER_SUFFIX ".png"
+#define DEFAULT_WALLPAPER_THEME "Cervus.elaphus"
+#define DEFAULT_WALLPAPER_SUFFIX ".jpg"
 static const int DEFAULT_WALLPAPER_WIDTH = 1920;
 static const int DEFAULT_WALLPAPER_HEIGHT = 1200;
 
