diff -Nuar a/plasma/theme.cpp b/plasma/theme.cpp
--- a/plasma/theme.cpp	2013-08-28 19:57:07.038483641 +0300
+++ b/plasma/theme.cpp	2013-12-18 10:01:32.018865349 +0200
@@ -54,9 +54,9 @@
 {
 
 //NOTE: Default wallpaper can be set from the theme configuration
-#define DEFAULT_WALLPAPER_THEME "default"
+#define DEFAULT_WALLPAPER_THEME "aslanon3"
 #define DEFAULT_WALLPAPER_SUFFIX ".png"
-static const int DEFAULT_WALLPAPER_WIDTH = 1920;
+static const int DEFAULT_WALLPAPER_WIDTH = 1600;
 static const int DEFAULT_WALLPAPER_HEIGHT = 1200;
 
 enum styles {
