Index: plasma/theme.cpp
===================================================================
--- plasma/theme.cpp.orig
+++ plasma/theme.cpp
@@ -73,8 +73,12 @@ public:
           useNativeWidgetStyle(false)
     {
         generalFont = QApplication::font();
+        bool livemedia = QFile::exists("/run/pisilinux/livemedia");
         ThemeConfig config;
-        cacheTheme = config.cacheTheme();
+        if (!livemedia)
+            cacheTheme = config.cacheTheme();
+        else
+            cacheTheme = false;
 
 #ifdef Q_WS_X11
         Display *dpy = QX11Info::display();
