diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kdelibs-4.2.2-orig/kfile/kdiroperator.cpp kdelibs-4.2.2/kfile/kdiroperator.cpp
--- kdelibs-4.2.2-orig/kfile/kdiroperator.cpp	2009-04-12 06:44:58.000000000 +0300
+++ kdelibs-4.2.2/kfile/kdiroperator.cpp	2009-04-12 07:11:27.000000000 +0300
@@ -2001,7 +2001,7 @@
         KConfigGroup cg(KGlobal::config(), QLatin1String("KDE"));
         const bool del = !d->currUrl.isLocalFile() ||
                          (QApplication::keyboardModifiers() & Qt::ShiftModifier) ||
-                         cg.readEntry("ShowDeleteCommand", false);
+                         cg.readEntry("ShowDeleteCommand", true);
         if (del) {
             d->actionMenu->addAction(d->actionCollection->action("delete"));
         }
