Index: kdelibs-4.4.3/plasma/widgets/iconwidget.cpp
===================================================================
--- kdelibs-4.4.3.orig/plasma/widgets/iconwidget.cpp
+++ kdelibs-4.4.3/plasma/widgets/iconwidget.cpp
@@ -603,9 +603,7 @@ QString IconWidget::svg() const
 
 QSizeF IconWidget::sizeHint(Qt::SizeHint which, const QSizeF & constraint) const
 {
-    if (which == Qt::PreferredSize) {
-        return sizeFromIconSize(KIconLoader::SizeMedium);
-    } else if (which == Qt::MinimumSize) {
+    if (which == Qt::MinimumSize) {
         return sizeFromIconSize(KIconLoader::SizeSmall);
     } else {
         return QGraphicsWidget::sizeHint(which, constraint);
