Index: chromium/src/base/debug/stack_trace.h
===================================================================
--- base/debug/stack_trace.h
+++ base/debug/stack_trace.h
@@ -5,6 +5,7 @@
 #ifndef BASE_DEBUG_STACK_TRACE_H_
 #define BASE_DEBUG_STACK_TRACE_H_
 
+#include <stdint.h>
 #include <iosfwd>
 #include <string>
 
Index: chromium/src/third_party/skia/src/ports/SkFontHost_fontconfig.cpp
===================================================================
--- third_party/skia/src/ports/SkFontHost_fontconfig.cpp
+++ third_party/skia/src/ports/SkFontHost_fontconfig.cpp
@@ -14,6 +14,7 @@
 #include "SkStream.h"
 #include "SkTypeface.h"
 #include "SkTypefaceCache.h"
+#include <unistd.h>

 ///////////////////////////////////////////////////////////////////////////////
 ///////////////////////////////////////////////////////////////////////////////

