Index: src/third_party/WebKit/Source/wtf/text/TextCodecICU.cpp
===================================================================
--- third_party/WebKit/Source/wtf/text/TextCodecICU.cpp	(revision 101168)
+++ third_party/WebKit/Source/wtf/text/TextCodecICU.cpp	(working copy)
@@ -203,7 +203,11 @@
     registrar("iso_8859-6", "ISO-8859-6");
     registrar("iso_8859-7", "ISO-8859-7");
     registrar("iso_8859-8", "ISO-8859-8");
+    registrar("ISO_8859-8-I", "ISO-8859-8-I");
     registrar("iso_8859-9", "windows-1254");
+    registrar("ISO_8859-10", "ISO-8859-10");
+    registrar("ISO_8859-13", "ISO-8859-13");
+    registrar("ISO_8859-14", "ISO-8859-14");
     registrar("iso_8859-15", "ISO-8859-15");
     registrar("koi8_r", "KOI8-R");
     registrar("x-cp1253", "windows-1253");

