--- build/common.gypi	2014-08-29 13:04:19.000000000 +0200
+++ build/common.gypi	2014-08-29 13:04:19.000000000 +0200
@@ -2440,7 +2440,7 @@
         }],
       ],
       'clang_warning_flags': [
-        '-Wheader-hygiene',
+        # '-Wheader-hygiene',
 
         # Don't die on dtoa code that uses a char as an array index.
         # This is required solely for base/third_party/dmg_fp/dtoa.cc.
@@ -2456,7 +2456,7 @@
         '-Wno-covered-switch-default',
 
         # Warns when a const char[] is converted to bool.
-        '-Wstring-conversion',
+        # '-Wstring-conversion',
 
         # C++11-related flags:
 
