diff -uprN sed-4.1.1.orig/sed/compile.c sed-4.1.1/sed/compile.c
--- sed-4.1.1.orig/sed/compile.c	2004-08-18 20:29:58.742844320 +0900
+++ sed-4.1.1/sed/compile.c	2004-08-18 20:32:53.064343424 +0900
@@ -1309,6 +1309,7 @@ compile_program(vector)
 
                 memset(&cur_stat, 0, sizeof(mbstate_t));
                 idx = 0;
+		len = dest_len;
 
                 /* trans_pairs = {src(0), dest(0), src(1), dest(1), ..., NULL}
                      src(i) : pointer to i-th source character.
