diff --git a/ChangeLog b/ChangeLog
index 06956a7..30d9fca 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,7 @@
+2009-10-14  Yuri G. Kudryashov  <urkud.urkud@gmail.com> (tiny change)
+
+	* testsuite/Makefile.tests: Override LC_ALL, not LANG.
+
 2009-06-27  Paolo Bonzini  <bonzini@gnu.org>
 
 	* configure.ac: Bump version.
diff --git a/testsuite/Makefile.tests b/testsuite/Makefile.tests
index 1f04290..c0910dc 100644
--- a/testsuite/Makefile.tests
+++ b/testsuite/Makefile.tests
@@ -34,16 +34,16 @@ y-bracket y-newline insert brackets::
 	@$(RM) $@.out 
 
 badenc::
-	LANG=ru_RU.UTF-8 $(TIME) $(SED) -nf $(srcdir)/$@.sed \
+	LC_ALL=ru_RU.UTF-8 $(TIME) $(SED) -nf $(srcdir)/$@.sed \
 		< $(srcdir)/$@.inp | $(TR) -d \\r > $@.out 
 	$(CMP) $(srcdir)/$@.good $@.out 
-	LANG=it_IT.UTF-8 $(TIME) $(SED) -nf $(srcdir)/$@.sed \
+	LC_ALL=it_IT.UTF-8 $(TIME) $(SED) -nf $(srcdir)/$@.sed \
 		< $(srcdir)/$@.inp | $(TR) -d \\r > $@.out 
 	$(CMP) $(srcdir)/$@.good $@.out 
-	LANG=en_US.UTF-8 $(TIME) $(SED) -nf $(srcdir)/$@.sed \
+	LC_ALL=en_US.UTF-8 $(TIME) $(SED) -nf $(srcdir)/$@.sed \
 		< $(srcdir)/$@.inp | $(TR) -d \\r > $@.out 
 	$(CMP) $(srcdir)/$@.good $@.out 
-	LANG=en_GB.UTF-8 $(TIME) $(SED) -nf $(srcdir)/$@.sed \
+	LC_ALL=en_GB.UTF-8 $(TIME) $(SED) -nf $(srcdir)/$@.sed \
 		< $(srcdir)/$@.inp | $(TR) -d \\r > $@.out 
 	$(CMP) $(srcdir)/$@.good $@.out 
 	@$(RM) $@.out 
@@ -51,10 +51,10 @@ badenc::
 # Try with ru_RU.UTF-8.  If it is presumably not installed, see if the current
 # locale is UTF-8 and run it in the current locale.
 utf8-1 utf8-2 utf8-3 utf8-4::
-	echo "LANG=ru_RU.UTF-8" \
+	echo "LC_ALL=ru_RU.UTF-8" \
 	  "$(TIME) $(SED) -f $(srcdir)/$@.sed" \
 	  "< $(srcdir)/$@.inp | $(TR) -d \\r > $@.out"; \
-	LANG=ru_RU.UTF-8 \
+	LC_ALL=ru_RU.UTF-8 \
 	  $(TIME) $(SED) -f $(srcdir)/$@.sed \
 		< $(srcdir)/$@.inp | $(TR) -d \\r > $@.out; \
 	$(CMP) $(srcdir)/$@.good $@.out && exit 0; \
