Index: sysinfo-2.3.1/src/Makefile.am
===================================================================
--- sysinfo-2.3.1.orig/src/Makefile.am
+++ sysinfo-2.3.1/src/Makefile.am
@@ -1,12 +1,12 @@
 SUBDIRS = . about
 
-INCLUDES = $(all_includes) `pkg-config --cflags hal`
+INCLUDES = $(all_includes)
 
 kde_module_LTLIBRARIES = kio_sysinfo.la libksysinfopart.la
 
 kio_sysinfo_la_SOURCES = sysinfo.cpp
 kio_sysinfo_la_LIBADD = $(LIB_KIO) -lhd -lGL
-kio_sysinfo_la_LDFLAGS = -module -avoid-version $(all_libraries) $(KDE_PLUGIN) $$(pkg-config --libs hal)
+kio_sysinfo_la_LDFLAGS = -module -avoid-version $(all_libraries) $(KDE_PLUGIN)
 
 kde_services_DATA = sysinfo.protocol ksysinfopart.desktop
 
Index: sysinfo-2.3.1/src/sysinfo.cpp
===================================================================
--- sysinfo-2.3.1.orig/src/sysinfo.cpp
+++ sysinfo-2.3.1/src/sysinfo.cpp
@@ -46,7 +46,6 @@
 #include <sys/vfs.h>
 #include <string.h>
 #include <sys/utsname.h>
-#include <hal/libhal.h>
 
 #include <kapplication.h>
 #include <kdebug.h>
