From a6683cf28a5f90b3bff983fa93899af9859cd716 Mon Sep 17 00:00:00 2001
From: Stefan Metzmacher <metze@samba.org>
Date: Tue, 31 Aug 2010 08:42:32 +0200
Subject: [PATCH 01/12] server/recoverd: do takeover_run after verifying the reclock file

metze
---
 server/ctdb_recoverd.c |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/server/ctdb_recoverd.c b/server/ctdb_recoverd.c
index ebcca1b..116825c 100644
--- a/server/ctdb_recoverd.c
+++ b/server/ctdb_recoverd.c
@@ -2979,10 +2979,6 @@ again:
 			rec->reallocate_callers = NULL;
 		}
 	}
-	/* if there are takeovers requested, perform it and notify the waiters */
-	if (rec->reallocate_callers) {
-		process_ipreallocate_requests(ctdb, rec);
-	}
 
 	if (rec->recmaster == (uint32_t)-1) {
 		DEBUG(DEBUG_NOTICE,(__location__ " Initial recovery master set - forcing election\n"));
@@ -3169,6 +3165,11 @@ again:
 		}
 	}
 
+	/* if there are takeovers requested, perform it and notify the waiters */
+	if (rec->reallocate_callers) {
+		process_ipreallocate_requests(ctdb, rec);
+	}
+
 	/* get the nodemap for all active remote nodes
 	 */
 	remote_nodemaps = talloc_array(mem_ctx, struct ctdb_node_map *, nodemap->num);
-- 
1.7.4

