From f60496197e417c1e0fbac4ffe7e1c3ac64364f93 Mon Sep 17 00:00:00 2001
From: Stefan Metzmacher <metze@samba.org>
Date: Mon, 30 Aug 2010 18:25:28 +0200
Subject: [PATCH 02/12] server/recoverd: if we can't get the recovery lock, ban ourself

metze
---
 server/ctdb_recoverd.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/server/ctdb_recoverd.c b/server/ctdb_recoverd.c
index 116825c..378472d 100644
--- a/server/ctdb_recoverd.c
+++ b/server/ctdb_recoverd.c
@@ -1337,8 +1337,10 @@ static int do_recovery(struct ctdb_recoverd *rec,
 		DEBUG(DEBUG_ERR,("Taking out recovery lock from recovery daemon\n"));
 		start_time = timeval_current();
 		if (!ctdb_recovery_lock(ctdb, true)) {
-			ctdb_set_culprit(rec, pnn);
-			DEBUG(DEBUG_ERR,("Unable to get recovery lock - aborting recovery\n"));
+			DEBUG(DEBUG_ERR,("Unable to get recovery lock - aborting recovery "
+					 "and ban ourself for %u seconds\n",
+					 ctdb->tunable.recovery_ban_period));
+			ctdb_ban_node(rec, pnn, ctdb->tunable.recovery_ban_period);
 			return -1;
 		}
 		ctdb_ctrl_report_recd_lock_latency(ctdb, CONTROL_TIMEOUT(), timeval_elapsed(&start_time));
-- 
1.7.4

