From 10b3632331d785886a9b60fbd89dae3118cc7a79 Mon Sep 17 00:00:00 2001
From: Stefan Metzmacher <metze@samba.org>
Date: Tue, 19 Oct 2010 19:21:23 +0200
Subject: [PATCH 04/12] events.d/11.routing: handle "updateip" event

metze
---
 config/events.d/11.routing |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/config/events.d/11.routing b/config/events.d/11.routing
index c265c38..9a1fdef 100755
--- a/config/events.d/11.routing
+++ b/config/events.d/11.routing
@@ -27,6 +27,17 @@ case "$1" in
 	done
 	;;
 
+    updateip)
+	oiface=$2
+	niface=$3
+	cat $CTDB_BASE/static-routes | egrep "^$niface " | while read IFACE DEST GW; do
+	    ip route add $DEST via $GW dev $IFACE >/dev/null 2>/dev/null
+	done
+	cat $CTDB_BASE/static-routes | egrep "^$oiface " | while read IFACE DEST GW; do
+	    ip route add $DEST via $GW dev $IFACE >/dev/null 2>/dev/null
+	done
+	;;
+
     *)
 	ctdb_standard_event_handler "$@"
 	;;
-- 
1.7.4

