From f8ec9382cb885c108be3c38b0e78cc7f705665f0 Mon Sep 17 00:00:00 2001
From: Stefan Metzmacher <metze@samba.org>
Date: Mon, 12 Jul 2010 14:11:41 +0200
Subject: [PATCH 11/12] config/interface_modify.sh: do the echo before running the script

metze
Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>
---
 config/interface_modify.sh |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/config/interface_modify.sh b/config/interface_modify.sh
index 69601dd..27e06ea 100755
--- a/config/interface_modify.sh
+++ b/config/interface_modify.sh
@@ -71,12 +71,12 @@ delete_ip_from_iface()
 			test -x "$_s_script" || {
 				continue
 			}
+			echo "call $_s_script '$_iface' '$_s_ip' '$_s_maskbits'"
 			$_s_script "$_iface" "$_s_ip" "$_s_maskbits" || {
 				ret=$?
 				echo "$_s_script '$_iface' '$_s_ip' '$_s_maskbits' - failed - $ret"
 				_failed=1
 			}
-			echo "call $_s_script '$_iface' '$_s_ip' '$_s_maskbits'"
 		done
 
 	    done
-- 
1.7.4

