diff -Nur nvidia-settings-260.19.12-old//src/XF86Config-parser/Device.c nvidia-settings-260.19.12/src/XF86Config-parser/Device.c
--- nvidia-settings-260.19.12-old//src/XF86Config-parser/Device.c	2010-10-21 10:44:27.683000043 +0300
+++ nvidia-settings-260.19.12/src/XF86Config-parser/Device.c	2010-10-21 10:45:33.802000043 +0300
@@ -365,7 +365,7 @@
                      "is required.");
         return (FALSE);
     }
-
+#if 0
     while (device) {
         if (!device->driver) {
             xconfigErrorMsg(ValidationErrorMsg, UNDEFINED_DRIVER_MSG,
@@ -375,6 +375,8 @@
     device = device->next;
     }
     return (TRUE);
+#endif
+
 }
 
 XConfigDevicePtr
