diff -Nuar zlib-1.2.7.orig/Makefile.in zlib-1.2.7/Makefile.in
--- zlib-1.2.7.orig/Makefile.in	2012-06-04 00:28:45.000000000 +0300
+++ zlib-1.2.7/Makefile.in	2012-06-04 00:32:57.830953700 +0300
@@ -201,7 +201,7 @@
 	  rm -f $(DESTDIR)$(sharedlibdir)/$(SHAREDLIB) $(DESTDIR)$(sharedlibdir)/$(SHAREDLIBM); \
 	  ln -s $(SHAREDLIBV) $(DESTDIR)$(sharedlibdir)/$(SHAREDLIB); \
 	  ln -s $(SHAREDLIBV) $(DESTDIR)$(sharedlibdir)/$(SHAREDLIBM); \
-	  ($(LDCONFIG) || true)  >/dev/null 2>&1; \
+	  #~ ($(LDCONFIG) || true)  >/dev/null 2>&1; \
 	fi
 	cp zlib.3 $(DESTDIR)$(man3dir)
 	chmod 644 $(DESTDIR)$(man3dir)/zlib.3
