Index: zlib-1.2.5/zlib.h
===================================================================
--- zlib-1.2.5.orig/zlib.h
+++ zlib-1.2.5/zlib.h
@@ -1256,7 +1256,12 @@ ZEXTERN int ZEXPORT gzwrite OF((gzFile f
    error.
 */
 
-ZEXTERN int ZEXPORTVA gzprintf OF((gzFile file, const char *format, ...));
+ZEXTERN int ZEXPORTVA gzprintf OF((gzFile file, const char *format, ...))
+#ifdef __GNUC__
+	__attribute__((__format__(__printf__,2,3)))
+#endif
+;
+
 /*
      Converts, formats, and writes the arguments to the compressed file under
    control of the format string, as in fprintf.  gzprintf returns the number of
