From 3024bb0cbf359f3e14b5386ed14ecfb168e8a267 Mon Sep 17 00:00:00 2001
From: Evgeniy Khramtsov <ekhramtsov@process-one.net>
Date: Thu, 19 Aug 2010 16:28:31 +1000
Subject: [PATCH 09/11] fixes ampersand escaping (EJAB-1258)

---
 src/eldap/eldap_filter.erl |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/eldap/eldap_filter.erl b/src/eldap/eldap_filter.erl
index 51dac5e..341fba1 100644
--- a/src/eldap/eldap_filter.erl
+++ b/src/eldap/eldap_filter.erl
@@ -171,7 +171,7 @@ do_sub(S, {RegExp, New, Times}, Iter) ->
     end.
 
 replace_amps(String) ->
-    lists:map(
+    lists:flatmap(
       fun($&) -> "\\&";
-	 (Chr) -> Chr
+	 (Chr) -> [Chr]
       end, String).
-- 
1.7.2.1

