From c9ff37027805d1d72446b5bcbc1a60da0f2aaae4 Mon Sep 17 00:00:00 2001
From: Evgeniy Khramtsov <ekhramtsov@process-one.net>
Date: Wed, 4 Aug 2010 19:23:52 +1000
Subject: [PATCH 01/11] Loading ASN.1 driver explicitly to avoid races in LDAP (EJAB-1284)

---
 src/ejabberd_app.erl |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/ejabberd_app.erl b/src/ejabberd_app.erl
index 97ae7fe..2227904 100644
--- a/src/ejabberd_app.erl
+++ b/src/ejabberd_app.erl
@@ -57,6 +57,8 @@ start(normal, _Args) ->
     ejabberd_config:start(),
     ejabberd_check:config(),
     connect_nodes(),
+    %% Loading ASN.1 driver explicitly to avoid races in LDAP
+    catch asn1rt:load_driver(),
     Sup = ejabberd_sup:start_link(),
     ejabberd_rdbms:start(),
     ejabberd_auth:start(),
-- 
1.7.2.1

