Index: trunk/src/common/algorithm.c
===================================================================
--- trunk/src/common/algorithm.c	(revision 404)
+++ trunk/src/common/algorithm.c	(revision 429)
@@ -52,5 +52,5 @@
 }
 
-ALGDIGEST *Alg_get_digest_by_name(ALGORITHM_TYPE hash)
+const ALGDIGEST *Alg_get_digest_by_name(ALGORITHM_TYPE hash)
 {
     return HASH_GetHashObjectByOidTag(hash);
