--- pam_pkcs11-0.6.1/src/pam_pkcs11/pam_config.c
+++ pam_pkcs11-0.6.1/src/pam_pkcs11/pam_config.c
@@ -45,7 +45,7 @@
         0,				/* int card_only; */
         0,				/* int wait_for_card; */
         "default", 			/* const char *pkcs11_module; */
-        CONFDIR "/pkcs11_module.so",/* const char *pkcs11_module_path; */
+        NULL,				/* const char *pkcs11_module_path; */
         NULL,                           /* screen savers */
         NULL,			/* slot_description */
         -1,				/* int slot_num; */
