Index: pam_pkcs11-0.6.2/src/common/cert_vfy.c
===================================================================
--- pam_pkcs11-0.6.2.orig/src/common/cert_vfy.c
+++ pam_pkcs11-0.6.2/src/common/cert_vfy.c
@@ -28,6 +28,12 @@ int verify_certificate(X509 * x509, cert
     SECStatus rv;
     CERTCertDBHandle *handle;
 
+    /* if neither ca nor crl check are requested skip */
+    if ( (policy->ca_policy==0) && (policy->crl_policy==CRLP_NONE) ) {
+        DBG("Neither CA nor CRL check requested. CertVrfy() skipped");
+        return 1;
+    }
+
     handle = CERT_GetDefaultCertDB();
 
     /* NSS already check all the revocation info with OCSP and crls */
