Index: 2008-devel/arch/x86/boot/video.c
===================================================================
--- 2008-devel.orig/arch/x86/boot/video.c
+++ 2008-devel/arch/x86/boot/video.c
@@ -469,6 +469,7 @@ static void restore_screen(void)
 void set_video(void)
 {
 	u16 mode = boot_params.hdr.vid_mode;
+	int vesa_failed = 0;
 
 	RESET_HEAP();
 
@@ -483,8 +484,15 @@ void set_video(void)
 		if (!set_mode(mode))
 			break;
 
-		printf("Undefined video mode number: %x\n", mode);
-		mode = ASK_VGA;
+		if (!vesa_failed) {
+			/* fallback is vesa 640x480 16bit */
+			printf("Could not set mode: %x, trying 640x480-16\n", mode);
+			mode = 0x311;
+			vesa_failed = 1;
+		} else {
+			printf("Undefined video mode number: %x, falling back to text mode\n", mode);
+			mode = VIDEO_80x25;
+		}
 	}
 	vesa_store_edid();
 	store_mode_params();
