--- subtitlecomposer-0.5.3/src/main/CMakeLists.txt.orig	2009-04-17 21:09:12.000000000 +0200
+++ subtitlecomposer-0.5.3/src/main/CMakeLists.txt	2011-04-29 23:18:14.633227475 +0200
@@ -47,6 +47,7 @@
 	${KDE4_KHTML_LIBS}
 	${KDE4_KFILE_LIBS}
 	${KDE4_KROSSUI_LIBS}
+	${KDE4_KROSSCORE_LIBS}
 	${common_LIBS}
 	${config_LIBS}
 	${core_LIBS}
--- subtitlecomposer-0.5.3/src/player/xine/CMakeLists.txt.link	2012-07-02 06:59:06.000000000 +0000
+++ subtitlecomposer-0.5.3/src/player/xine/CMakeLists.txt	2012-07-02 06:59:25.000000000 +0000
@@ -36,6 +36,7 @@
 	)
 	SET( players_xine_LIBS
 		${LIBXCB_LIBRARIES}
+		libX11
 		${players_xine_LIBS}
 		CACHE INTERNAL EXPORTEDVARIABLE
 	)
--- subtitlecomposer-0.5.3/src/player/gstreamer/CMakeLists.txt.link	2012-07-02 06:59:39.000000000 +0000
+++ subtitlecomposer-0.5.3/src/player/gstreamer/CMakeLists.txt	2012-07-02 07:01:50.000000000 +0000
@@ -1,5 +1,6 @@
 # TODO WHY AREN'T THESE INCLUDED BY MACRO_OPTIONAL_FIND_PACKAGE( GStreamer )?
 FIND_PACKAGE( GLIB2 REQUIRED )
+FIND_PACKAGE( GObject REQUIRED )
 FIND_PACKAGE( LibXml2 REQUIRED )
 
 SET( players_gstreamer_INCLUDE_DIR
