diff -Nuar lxterminal-0.1.9.orig//data/lxterminal.conf lxterminal-0.1.9//data/lxterminal.conf
--- lxterminal-0.1.9.orig//data/lxterminal.conf	2011-02-08 16:05:30.911000025 +0200
+++ lxterminal-0.1.9//data/lxterminal.conf	2011-02-08 16:07:17.233000026 +0200
@@ -2,3 +2,4 @@
 fontname=Monospace 10
 selchars=-A-Za-z0-9,./?%&#:_
 scrollback=1000
+bgalpha=65535
diff -Nuar lxterminal-0.1.9.orig//data/lxterminal.conf.in lxterminal-0.1.9//data/lxterminal.conf.in
--- lxterminal-0.1.9.orig//data/lxterminal.conf.in	2011-02-08 16:05:30.911000025 +0200
+++ lxterminal-0.1.9//data/lxterminal.conf.in	2011-02-08 16:07:23.682000026 +0200
@@ -2,3 +2,4 @@
 fontname=Monospace 10
 selchars=-A-Za-z0-9,./?%&#:_
 scrollback=1000
+bgalpha=65535
