--- CMakeLists.txt.orig	2011-01-26 13:40:14.590000041 +0200
+++ CMakeLists.txt	2011-01-26 13:41:31.745000040 +0200
@@ -321,7 +321,7 @@
   CACHE STRING "The folder where CGAL CMake modules will be installed, relative to CMAKE_INSTALL_PREFIX" 
   )
 
-set ( CGAL_INSTALL_DOC_DIR "share/doc/${CGAL_VERSION_DIR}"  
+set ( CGAL_INSTALL_DOC_DIR "share/doc/CGAL"
   CACHE STRING "The folder where CGAL documentation and license files will be installed, relative to CMAKE_INSTALL_PREFIX" 
   )
 
