Index: vlock-2.2.2/modules/Makefile
===================================================================
--- vlock-2.2.2.orig/modules/Makefile
+++ vlock-2.2.2/modules/Makefile
@@ -27,9 +27,9 @@ all.o: all.c ../src/console_switch.h
 
 # special installation rules
 
-install-new.so : MODULE_GROUP=$(VLOCK_GROUP)
+install-new.so : MODULE_GROUP=$(ROOT_GROUP)
 install-new.so : MODULE_MODE=$(VLOCK_MODULE_MODE)
-install-nosysrq.so : MODULE_GROUP=$(VLOCK_GROUP)
+install-nosysrq.so : MODULE_GROUP=$(ROOT_GROUP)
 install-nosysrq.so : MODULE_MODE=$(VLOCK_MODULE_MODE)
 
 # generic installation rule
Index: vlock-2.2.2/configure
===================================================================
--- vlock-2.2.2.orig/configure
+++ vlock-2.2.2/configure
@@ -230,7 +230,7 @@ set_defaults() {
   ENABLE_PLUGINS="yes"
   SCRIPTS=""
 
-  VLOCK_GROUP="vlock"
+  VLOCK_GROUP="root"
   VLOCK_MODULE_MODE="0750"
 
   BOURNE_SHELL="/bin/sh"
