Index: paman-0.9.4/src/paman.desktop
===================================================================
--- paman-0.9.4.orig/src/paman.desktop
+++ paman-0.9.4/src/paman.desktop
@@ -1,7 +1,10 @@
 [Desktop Entry]
 Encoding=UTF-8
-Name=PulseAudio Manager
-Comment=Manager your PulseAudio sound server.
+Name=paman
+Comment=Manage your PulseAudio sound server.
+Comment[tr]=PulseAudio ses sunucunuzu yönetin.
+GenericName=PulseAudio Manager
+GenericName[tr]=PulseAudio Yöneticisi
 Exec=paman
 Icon=audio-card
 StartupNotify=true
