--- src/cf.data.pre	2007-11-16 00:33:05.000000000 +0100
+++ src/cf.data.pre.oden	2008-01-27 20:37:58.000000000 +0100
@@ -3189,6 +3189,9 @@ DOC_START
 	This value is the lifetime to set for all open descriptors
 	during shutdown mode.  Any active clients after this many
 	seconds will receive a 'timeout' message.
+NOCOMMENT_START
+shutdown_lifetime 5 seconds
+NOCOMMENT_END
 DOC_END
 
 COMMENT_START
