Index: chromium-bsu-0.9.15/misc/chromium-bsu.desktop
===================================================================
--- chromium-bsu-0.9.15.orig/misc/chromium-bsu.desktop
+++ chromium-bsu-0.9.15/misc/chromium-bsu.desktop
@@ -1,10 +1,13 @@
 [Desktop Entry]
 Name=Chromium B.S.U.
+Name[tr]=Chromium B.S.U.
 GenericName=Shoot'em up space shooter
 GenericName[sv]=Rymdskjutspel (Shoot'em up)
+GenericName[tr]=Uzayda "önüne geleni vur" oyunu
 Comment=Scrolling space shooter
 Comment[ro]=O navă în spațiu trebuie să împuște inamicii
 Comment[sv]=Vertikalrullande skjutspel i rymdmiljö (Shoot'em up)
+Comment[tr]=Karşınıza çıkan düşman gemilerini yok edin
 Exec=chromium-bsu
 Icon=chromium-bsu
 Terminal=false
