diff -Nur urbanterror-4.1-old//code/q3_ui/ui_mods.c urbanterror-4.1/code/q3_ui/ui_mods.c
--- urbanterror-4.1-old//code/q3_ui/ui_mods.c	2011-01-20 12:09:32.040000025 +0200
+++ urbanterror-4.1/code/q3_ui/ui_mods.c	2011-01-20 12:10:41.974000033 +0200
@@ -125,7 +125,7 @@
 
 	// always start off with baseq3
 	s_mods.list.numitems = 1;
-	s_mods.list.itemnames[0] = s_mods.descriptionList[0] = "Quake III Arena";
+	s_mods.list.itemnames[0] = s_mods.descriptionList[0] = "Urban Terror";
 	s_mods.fs_gameList[0] = "";
 
 	numdirs = trap_FS_GetFileList( "$modlist", "", dirlist, sizeof(dirlist) );
diff -Nur urbanterror-4.1-old//code/qcommon/files.c urbanterror-4.1/code/qcommon/files.c
--- urbanterror-4.1-old//code/qcommon/files.c	2011-01-20 12:09:31.938000023 +0200
+++ urbanterror-4.1/code/qcommon/files.c	2011-01-20 12:11:47.893000069 +0200
@@ -2823,7 +2823,7 @@
 		homePath = fs_basepath->string;
 	}
 	fs_homepath = Cvar_Get ("fs_homepath", homePath, CVAR_INIT );
-	fs_gamedirvar = Cvar_Get ("fs_game", "q3ut4", CVAR_INIT|CVAR_SYSTEMINFO );
+	fs_gamedirvar = Cvar_Get ("fs_game", "baseut4", CVAR_INIT|CVAR_SYSTEMINFO );
 
 	// add search path elements in reverse priority order
 	if (fs_basepath->string[0]) {
diff -Nur urbanterror-4.1-old//code/qcommon/q_shared.h urbanterror-4.1/code/qcommon/q_shared.h
--- urbanterror-4.1-old//code/qcommon/q_shared.h	2011-01-20 12:09:31.933000023 +0200
+++ urbanterror-4.1/code/qcommon/q_shared.h	2011-01-20 12:12:13.483000001 +0200
@@ -28,7 +28,7 @@
 
 #ifdef STANDALONE
   #define PRODUCT_NAME				"ioq3-urt"
-  #define BASEGAME					"q3ut4"
+  #define BASEGAME					"baseut4"
   #define CLIENT_WINDOW_TITLE     	"ioq3-urt"
   #define CLIENT_ICONIFIED_TITLE		"ioq3-urt"
   #define GAMENAME_FOR_MASTER		"ioq3-urt"	// must NOT contain whitespaces
diff -Nur urbanterror-4.1-old//code/sys/sys_unix.c urbanterror-4.1/code/sys/sys_unix.c
--- urbanterror-4.1-old//code/sys/sys_unix.c	2011-01-20 12:09:32.040000025 +0200
+++ urbanterror-4.1/code/sys/sys_unix.c	2011-01-20 12:12:59.710999983 +0200
@@ -60,7 +60,7 @@
 			Q_strcat( homePath, sizeof( homePath ),
 					"/Library/Application Support/Quake3" );
 #else
-			Q_strcat( homePath, sizeof( homePath ), "/.q3a" );
+			Q_strcat( homePath, sizeof( homePath ), "/.urbanterror" );
 #endif
 		}
 	}
