--- openssh-4.3p2/scp.c.fromto-remote	2006-01-31 12:11:38.000000000 +0100
+++ openssh-4.3p2/scp.c	2006-04-14 10:09:56.000000000 +0200
@@ -446,7 +446,11 @@
 				addargs(&alist, "-v");
 			addargs(&alist, "-x");
 			addargs(&alist, "-oClearAllForwardings yes");
-			addargs(&alist, "-n");
+			if (isatty(fileno(stdin))) {
+				addargs(&alist, "-t");
+			} else {
+				addargs(&alist, "-n");
+			}
 
 			*src++ = 0;
 			if (*src == 0)
