diff -up openssh-5.8p1/packet.c.packet openssh-5.8p1/packet.c
--- openssh-5.8p1/packet.c.packet	2011-04-05 13:29:06.998648899 +0200
+++ openssh-5.8p1/packet.c	2011-04-05 13:30:32.967648596 +0200
@@ -294,6 +294,8 @@ packet_connection_is_on_socket(void)
 	struct sockaddr_storage from, to;
 	socklen_t fromlen, tolen;
 
+	if (!active_state)
+		return 0;
 	/* filedescriptors in and out are the same, so it's a socket */
 	if (active_state->connection_in == active_state->connection_out)
 		return 1;
