--- openssh-4.3p2/contrib/gnome-ssh-askpass2.c.grab-info	2006-07-17 15:10:11.000000000 +0200
+++ openssh-4.3p2/contrib/gnome-ssh-askpass2.c	2006-07-17 15:25:04.000000000 +0200
@@ -65,9 +65,12 @@
 	err = gtk_message_dialog_new(NULL, 0,
 				     GTK_MESSAGE_ERROR,
 				     GTK_BUTTONS_CLOSE,
-				     "Could not grab %s. "
-				     "A malicious client may be eavesdropping "
-				     "on your session.", what);
+				     "SSH password dialog could not grab the %s input.\n"
+				     "This might be caused by application such as screensaver, "
+				     "however it could also mean that someone may be eavesdropping "
+				     "on your session.\n"
+				     "Either close the application which grabs the %s or "
+				     "log out and log in again to prevent this from happening.", what, what);
 	gtk_window_set_position(GTK_WINDOW(err), GTK_WIN_POS_CENTER);
 	gtk_label_set_line_wrap(GTK_LABEL((GTK_MESSAGE_DIALOG(err))->label),
 				TRUE);
