diff -up openssh-5.3p1/entropy.c.randclean openssh-5.3p1/entropy.c
--- openssh-5.3p1/entropy.c.randclean	2010-01-21 09:26:30.000000000 +0100
+++ openssh-5.3p1/entropy.c	2010-01-21 09:26:37.000000000 +0100
@@ -159,6 +159,9 @@ init_rng(void)
 		fatal("OpenSSL version mismatch. Built against %lx, you "
 		    "have %lx", OPENSSL_VERSION_NUMBER, SSLeay());
 
+	/* clean the PRNG status when exiting the program */
+	atexit(RAND_cleanup);
+
 #ifndef OPENSSL_PRNG_ONLY
 	original_uid = getuid();
 	original_euid = geteuid();
