diff -p -up a2ps-4.14/etc/a2ps_cfg.in.org a2ps-4.14/etc/a2ps_cfg.in
--- a2ps-4.14/etc/a2ps_cfg.in.org	2008-04-22 11:24:00.000000000 -0300
+++ a2ps-4.14/etc/a2ps_cfg.in	2008-04-22 11:24:31.000000000 -0300
@@ -183,7 +183,7 @@ Variable: del.imagick @convert@ -page #w
 # We give -dNOPLATFONTS so that real fonts are used
 # (Strange results appear with Ogonkify's fonts for instance).
 @COM_ghostview@Variable: ghostview @ghostview@ --arguments=-dNOPLATFONTS
-@COM_gv@Variable: ghostview @gv@ -antialias --arguments=-dNOPLATFONTS
+Variable: ghostview @gv@ -antialias --arguments=-dNOPLATFONTS
 
 # Used for the `pdf' printer.
 # Use as `#{ps2pdf} INPUT OUTPUT'.
@@ -376,7 +376,7 @@ UserOption: simplex -SDuplex:false
 UserOption: display -1 --prolog=color -Pdisplay
 
 # It is better not to pipe in ghostview
-@COM_DISPLAY@Printer: display | cat > #f0 && #{ghostview} #f0
+Printer: display | cat > #f0 && #{ghostview} #f0
 
 # Save into a file named after the input files.
 # E.g. a2ps src/foo.c => foo.ps
