diff -Nur libdvdcss-1.2.9-old/test/Makefile.am libdvdcss-1.2.9/test/Makefile.am
--- libdvdcss-1.2.9-old/test/Makefile.am	2009-04-23 18:56:44.000000000 +0300
+++ libdvdcss-1.2.9/test/Makefile.am	2009-04-23 18:56:53.000000000 +0300
@@ -1,5 +1,5 @@
 
-noinst_PROGRAMS = csstest #dvd_region
+check_PROGRAMS = csstest #dvd_region
 
 csstest_SOURCES = csstest.c
 csstest_LDADD = $(top_builddir)/src/libdvdcss.la $(bsdi_libadd)
