diff -up openct-0.6.16/etc/openct.udev.in.nosleep openct-0.6.16/etc/openct.udev.in
--- openct-0.6.16/etc/openct.udev.in.nosleep	2009-02-01 09:26:41.000000000 +0100
+++ openct-0.6.16/etc/openct.udev.in	2009-06-15 22:31:00.000000000 +0200
@@ -17,7 +17,8 @@ ACTION!="add", GOTO="openct_usb_rules_en
 KERNEL=="[0-9]*:*", WAIT_FOR_SYSFS="bInterfaceProtocol"
 
 # sleep for 100ms - the wait_for_sysfs might not be enough
-PROGRAM="/bin/sleep 0.1"
+# Disabled in this package - see https://bugzilla.redhat.com/287871
+#PROGRAM="/bin/sleep 0.1"
 
 # ccid
 ENV{DEVTYPE}=="usb_interface", SYSFS{bInterfaceClass}=="0b", SYSFS{bInterfaceSubClass}=="00", SYSFS{bInterfaceProtocol}=="00" RUN+="@udevdir@/openct_usb"
