diff -up openct-0.6.16/etc/openct_usb.in.udevadm openct-0.6.16/etc/openct_usb.in
--- openct-0.6.16/etc/openct_usb.in.udevadm	2008-12-27 16:51:56.000000000 +0100
+++ openct-0.6.16/etc/openct_usb.in	2009-06-15 22:11:06.000000000 +0200
@@ -10,10 +10,10 @@ if [ -z "$DEVNAME" ]; then
 	# Guess udev info interface.
 	# Newer udev uses udevadm
 	#
-	if which udevinfo > /dev/null 2>&1; then
-		UDEVINFO="udevinfo"
-	else
+	if which udevadm > /dev/null 2>&1; then
 		UDEVINFO="udevadm info"
+	else
+		UDEVINFO="udevinfo"
 	fi
 	DEVNAME=/dev/$($UDEVINFO --query=name --path=$(dirname $DEVPATH))
 fi
