diff -Naur extensions/spellcheck-orig//hunspell/src/mozHunspell.h extensions/spellcheck/hunspell/src/mozHunspell.h
--- extensions/spellcheck-orig//hunspell/src/mozHunspell.h	2011-01-13 19:50:02.000000000 +0000
+++ extensions/spellcheck/hunspell/src/mozHunspell.h	2011-01-15 15:02:35.262000385 +0000
@@ -71,7 +71,7 @@
 #include "nsWeakReference.h"
 #include "nsCycleCollectionParticipant.h"
 
-#define MOZ_HUNSPELL_CONTRACTID "@mozilla.org/spellchecker/engine;1"
+#define MOZ_HUNSPELL_CONTRACTID "@mozilla.org/spellchecker/hunspell;1"
 #define MOZ_HUNSPELL_CID         \
 /* 56c778e4-1bee-45f3-a689-886692a97fe7 */   \
 { 0x56c778e4, 0x1bee, 0x45f3, \
diff -Naur extensions/spellcheck-orig//Makefile.in extensions/spellcheck/Makefile.in
--- extensions/spellcheck-orig//Makefile.in	2011-01-13 19:50:02.000000000 +0000
+++ extensions/spellcheck/Makefile.in	2011-01-15 15:02:35.184000384 +0000
@@ -47,7 +47,7 @@
 ifeq (camino,$(MOZ_BUILD_APP))
 DIRS		+= osxspell
 else
-DIRS		+= hunspell
+DIRS		+= hunspell zemberek
 endif
 
 DIRS += src
