diff -Naur connectagram-1.0.1.orig/icons/connectagram.desktop connectagram-1.0.1/icons/connectagram.desktop
--- connectagram-1.0.1.orig/icons/connectagram.desktop	2009-08-29 03:21:24.000000000 +0300
+++ connectagram-1.0.1/icons/connectagram.desktop	2009-09-08 01:06:17.976822358 +0300
@@ -1,5 +1,11 @@
 [Desktop Entry]
+Encoding=UTF-8
 Name=Connectagram
+Name[tr]=Connectagram
+GenericName=Word Unscrambling Game
+GenericName[tr]=Sözcük Tahmin Etme Oyunu
+Comment=A fantastic word unscrambling game
+Comment[tr]=Eğlenceli bir sözcük tahmin etme oyunu
 Exec=connectagram
 Icon=connectagram
 Terminal=false
