Index: xchat-2.8.8/configure.in
===================================================================
--- xchat-2.8.8.orig/configure.in
+++ xchat-2.8.8/configure.in
@@ -8,7 +8,7 @@ AC_COPYRIGHT([Copyright (C) 1998-2010 Pe
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_SRCDIR([configure.in])
 
-AM_INIT_AUTOMAKE([1.9 dist-bzip2 subdir-objects no-define])
+AM_INIT_AUTOMAKE([1.9 foreign dist-bzip2 subdir-objects no-define])
 
 AM_MAINTAINER_MODE
 AC_PROG_CC
