diff -ur xchat-2.8.6/src/common/servlist.c xchat-2.8.6-default-utf8/src/common/servlist.c
--- xchat-2.8.6/src/common/servlist.c	2008-04-01 10:22:34.000000000 +0200
+++ xchat-2.8.6-default-utf8/src/common/servlist.c	2008-06-15 14:43:35.000000000 +0200
@@ -892,7 +892,7 @@
 		if (def[i].network)
 		{
 			net = servlist_net_add (def[i].network, def[i].host, FALSE);
-			net->encoding = strdup ("IRC (Latin/Unicode Hybrid)");
+			net->encoding = strdup ("UTF-8 (Unicode)");
 			if (def[i].channel)
 				net->autojoin = strdup (def[i].channel);
 			if (def[i].charset)
diff -ur xchat-2.8.6/src/fe-gtk/servlistgui.c xchat-2.8.6-default-utf8/src/fe-gtk/servlistgui.c
--- xchat-2.8.6/src/fe-gtk/servlistgui.c	2008-06-08 10:04:48.000000000 +0200
+++ xchat-2.8.6-default-utf8/src/fe-gtk/servlistgui.c	2008-06-15 14:36:03.000000000 +0200
@@ -267,7 +267,7 @@
 	ircnet *net;
 
 	net = servlist_net_add (_("New Network"), "", TRUE);
-	net->encoding = strdup ("IRC (Latin/Unicode Hybrid)");
+	net->encoding = strdup ("UTF-8 (Unicode)");
 	servlist_server_add (net, "newserver/6667");
 
 	store = (GtkListStore *)gtk_tree_view_get_model (treeview);
