--- main.cpp.orig	2010-05-11 11:19:38.692337335 +0300
+++ main.cpp	2010-05-11 11:19:50.474467287 +0300
@@ -450,7 +450,10 @@
     if( !allArgumentsUsed())
         return false;
     KApplication::updateUserTimestamp( 0 ); // TODO
-    if( !mime.isEmpty())
+
+    KMimeType::Ptr mimeType = KMimeType::mimeType( mime);
+
+    if( !mime.isEmpty() && mimeType && KMimeTypeTrader::self()->preferredService( mimeType->name()))
         {
         return KRun::runUrl( url, mime, NULL ); // TODO parent
         }
