Index: kmozillahelper/CMakeLists.txt
===================================================================
--- kmozillahelper.orig/CMakeLists.txt
+++ kmozillahelper/CMakeLists.txt
@@ -17,6 +17,6 @@ kde4_add_executable( kmozillahelper ${km
 target_link_libraries( kmozillahelper ${KDE4_KFILE_LIBS} )
 
 # TODO hardcoded path?
-install( TARGETS kmozillahelper DESTINATION /usr/lib/mozilla/ )
+install( TARGETS kmozillahelper DESTINATION /usr/libexec/ )
 
 install( FILES kmozillahelper.notifyrc  DESTINATION  ${DATA_INSTALL_DIR}/kmozillahelper )
Index: kmozillahelper/main.cpp
===================================================================
--- kmozillahelper.orig/main.cpp
+++ kmozillahelper/main.cpp
@@ -530,7 +530,7 @@ bool App::handleIsDefaultBrowser()
         return false;
     QString browser = KConfigGroup( KSharedConfig::openConfig( "kdeglobals" ), "General" )
         .readEntry( "BrowserApplication" );
-    return browser == "MozillaFirefox" || browser == "MozillaFirefox.desktop"
+    return browser == "mozillafirefox" || browser == "mozillafirefox.desktop" || browser == "firefox"
         || browser == "!firefox" || browser == "!/usr/bin/firefox";
     }
 
