diff -Nur kmozillahelper-0.6.3.orig/CMakeLists.txt kmozillahelper-0.6.3/CMakeLists.txt
--- kmozillahelper-0.6.3.orig/CMakeLists.txt	2013-03-19 02:20:02.000000000 +0200
+++ kmozillahelper-0.6.3/CMakeLists.txt	2013-03-19 02:42:05.119349166 +0200
@@ -17,6 +17,6 @@
 target_link_libraries( kmozillahelper ${KDE4_KFILE_LIBS} )
 
 # TODO hardcoded path?
-install( TARGETS kmozillahelper DESTINATION /usr/lib/mozilla/ )
+install( TARGETS kmozillahelper DESTINATION /usr/libexec/ )
 
 install( FILES kmozillahelper.notifyrc  DESTINATION  ${DATA_INSTALL_DIR}/kmozillahelper )
diff -Nur kmozillahelper-0.6.3.orig/main.cpp kmozillahelper-0.6.3/main.cpp
--- kmozillahelper-0.6.3.orig/main.cpp	2013-03-19 02:20:02.000000000 +0200
+++ kmozillahelper-0.6.3/main.cpp	2013-03-19 02:43:28.115350512 +0200
@@ -535,7 +535,7 @@
         return false;
     QString browser = KConfigGroup( KSharedConfig::openConfig( "kdeglobals" ), "General" )
         .readEntry( "BrowserApplication" );
-    return browser == "MozillaFirefox" || browser == "MozillaFirefox.desktop"
+    return browser == "mozillafirefox" || browser == "mozillafirefox.desktop" || browser == "firefox"
         || browser == "!firefox" || browser == "!/usr/bin/firefox"
         || browser == "firefox" || browser == "firefox.desktop";
     }
