From 86f220c11a6f05468682e940d066346a80168dd2 Mon Sep 17 00:00:00 2001
From: Frederic Crozat <fcrozat@mandriva.com>
Date: Wed, 8 Apr 2009 18:32:13 +0200
Subject: [PATCH] kernel header 2.6.29

---
 src/VBox/Devices/Storage/swab.h |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/VBox/Devices/Storage/swab.h b/src/VBox/Devices/Storage/swab.h
index 35db597..b0170ba 100644
--- a/src/VBox/Devices/Storage/swab.h
+++ b/src/VBox/Devices/Storage/swab.h
@@ -34,6 +34,8 @@
 
 #define _LINUX_BYTEORDER_SWAB_H
 #define _LINUX_BYTEORDER_SWABB_H
+#define _LINUX_SWAB_H
+
 
 #include <asm/types.h>
 
-- 
1.6.2.2

