diff -p -up VirtualBox-1.6.0_OSE/src/VBox/Runtime/r3/linux/semmutex-linux.cpp.futex VirtualBox-1.6.0_OSE/src/VBox/Runtime/r3/linux/semmutex-linux.cpp
--- VirtualBox-1.6.0_OSE/src/VBox/Runtime/r3/linux/semmutex-linux.cpp.futex	2008-05-09 02:59:00.000000000 +0200
+++ VirtualBox-1.6.0_OSE/src/VBox/Runtime/r3/linux/semmutex-linux.cpp	2008-05-09 02:59:17.000000000 +0200
@@ -81,7 +81,7 @@ struct RTSEMMUTEXINTERNAL
 static long sys_futex(int32_t volatile *uaddr, int op, int val, struct timespec *utime, int32_t *uaddr2, int val3)
 {
     errno = 0;
-    long rc = syscall(__NR_futex, uaddr, op, val, utime, uaddr2, val3);
+    long rc = syscall(SYS_futex, uaddr, op, val, utime, uaddr2, val3);
     if (rc < 0)
     {
         Assert(rc == -1);
