--- Makefile.orig	2007-12-29 12:49:31.000000000 +0200
+++ Makefile	2007-12-29 12:52:08.000000000 +0200
@@ -98,16 +98,16 @@
 clean:
 	rm -f *.o stormbaancoureur
 
-GAMEDIR=$(DESTDIR)/usr/share/games/stormbaancoureur
+GAMEDIR=$(DESTDIR)/usr/share/stormbaancoureur
 install: stormbaancoureur
 	# Directories
-	mkdir -p $(DESTDIR)/usr/games/
+	mkdir -p $(DESTDIR)/usr/bin/
 	mkdir -p $(GAMEDIR)/images/
 	mkdir -p $(GAMEDIR)/sounds/
 	mkdir -p $(GAMEDIR)/shaders/
 	mkdir -p $(GAMEDIR)/models/
 	# Binary
-	install stormbaancoureur $(DESTDIR)/usr/games/stormbaancoureur
+	install stormbaancoureur $(DESTDIR)/usr/bin/stormbaancoureur
 	# Images
 	install -m 644 images/engine.tga $(GAMEDIR)/images/engine.tga
 	install -m 644 images/info_carpet_smooth.rgb $(GAMEDIR)/images/info_carpet_smooth.rgb
