--- main.cxx.orig	2007-12-28 22:09:41.000000000 +0200
+++ main.cxx	2007-12-29 13:23:48.000000000 +0200
@@ -1176,11 +1176,11 @@
   fprintf(stderr,"OpenDE is (c) by Russel L. Smith\n");
 
   char *bindirname  = dirname(argv[0]);
-  if (!strcmp(bindirname,"."))
-    dirprefix="/usr/share/games/stormbaancoureur";
+  if (!strcmp(bindirname,"/usr/bin"))
+    dirprefix="/usr/share/stormbaancoureur";
   else
   {
-    dirprefix = dirname(bindirname) + std::string("/share/games/stormbaancoureur");
+    dirprefix = std::string("/usr/share/stormbaancoureur");
   }
   if (getenv("PLODE_DATADIR"))
     dirprefix = getenv("PLODE_DATADIR");
