--- Makefile.orig	2007-12-28 23:55:42.000000000 +0200
+++ Makefile	2007-12-29 12:49:31.000000000 +0200
@@ -43,15 +43,14 @@
 
 
 LIBS=\
-  $(ODEPREFIX)/$(LIBDIRNAME)/libode.a \
   -lplibssgaux -lplibssg -lplibsg -lplibpu -lplibfnt -lplibul \
-  -lglut -lGLU -lGL -lasound
+  -lglut -lGLU -lGL -lasound -lode
 
 
 all: stormbaancoureur
 
 
-stormbaancoureur: $(OBJS) $(ODEPREFIX)/$(LIBDIRNAME)/libode.a
+stormbaancoureur: $(OBJS)
 	$(CXX) -o stormbaancoureur $(OBJS) $(LFLAGS) $(LIBS)
 
 staticworldobject.o: ../src-common/staticworldobject.cxx ../src-common/staticworldobject.h ../src-common/worldobject.h
