Index: LVM2.2.02.70/libdm/regex/parse_rx.c
===================================================================
--- LVM2.2.02.70.orig/libdm/regex/parse_rx.c
+++ LVM2.2.02.70/libdm/regex/parse_rx.c
@@ -67,7 +67,7 @@ static void _regex_print(struct rx_node 
 			printf("[");
 		for (i = 0; i < 256; i++)
 			if (dm_bit(rx->charset, i)) {
-				if isprint(i)
+				if (isprint(i))
 					printf("%c", (char) i);
 				else if (i == HAT_CHAR)
 					printf("^");
Index: LVM2.2.02.70/configure.in
===================================================================
--- LVM2.2.02.70.orig/configure.in
+++ LVM2.2.02.70/configure.in
@@ -30,7 +30,7 @@ case "$host_os" in
 	linux*)
 		CFLAGS="$CFLAGS"
 		COPTIMISE_FLAG="-O2"
-		CLDFLAGS="$CLDFLAGS -Wl,--version-script,.export.sym"
+#		CLDFLAGS="$CLDFLAGS -Wl,--version-script,.export.sym"
 		CLDWHOLEARCHIVE="-Wl,-whole-archive"
 		CLDNOWHOLEARCHIVE="-Wl,-no-whole-archive"
 		LDDEPS="$LDDEPS .export.sym"
