--- LVM2.2.02.45/lib/commands/toolcontext.c	2009-02-22 22:14:38.000000000 +0100
+++ LVM2.2.02.45-diet/lib/commands/toolcontext.c	2009-03-18 12:18:23.000000000 +0100
@@ -1006,10 +1006,10 @@
 #ifdef M_MMAP_MAX
 	mallopt(M_MMAP_MAX, 0);
 #endif
-
+#ifdef __dietlibc__
 	if (!setlocale(LC_ALL, ""))
 		log_very_verbose("setlocale failed");
-
+#endif
 #ifdef INTL_PACKAGE
 	bindtextdomain(INTL_PACKAGE, LOCALEDIR);
 #endif
--- LVM2.2.02.45/lib/locking/file_locking.c	2008-11-12 10:30:52.000000000 +0100
+++ LVM2.2.02.45-diet/lib/locking/file_locking.c	2009-03-18 12:18:52.000000000 +0100
@@ -30,6 +30,10 @@
 #include <fcntl.h>
 #include <signal.h>
 
+#ifdef __dietlibc__
+typedef void (*sig_t) (int);
+#endif
+
 struct lock_list {
 	struct dm_list list;
 	int lf;
--- LVM2.2.02.45/libdm/libdevmapper.h	2008-11-04 16:07:45.000000000 +0100
+++ LVM2.2.02.45-diet/libdm/libdevmapper.h	2009-03-18 12:17:27.000000000 +0100
@@ -160,6 +160,10 @@
 /*
  * Control read_ahead.
  */
+#ifdef __dietlibc__
+# define UINT32_MAX             (4294967295U)
+#endif
+
 #define DM_READ_AHEAD_AUTO UINT32_MAX	/* Use kernel default readahead */
 #define DM_READ_AHEAD_NONE 0		/* Disable readahead */
 
--- LVM2.2.02.45/lib/config/config.c	2008-11-03 23:14:27.000000000 +0100
+++ LVM2.2.02.45-diet/lib/config/config.c	2009-03-18 12:22:31.000000000 +0100
@@ -28,6 +28,10 @@
 #include <fcntl.h>
 #include <ctype.h>
 
+#ifdef __dietlibc__
+typedef char* caddr_t __attribute_dontuse__;
+#endif
+
 #define SECTION_B_CHAR '{'
 #define SECTION_E_CHAR '}'
 
--- LVM2.2.02.45/lib/misc/lib.h	2008-11-03 19:59:58.000000000 +0100
+++ LVM2.2.02.45-diet/lib/misc/lib.h	2009-03-18 12:24:30.000000000 +0100
@@ -45,4 +45,20 @@
 #include <unistd.h>
 #include <sys/stat.h>
 
+#ifdef __dietlibc__
+# include <strings.h>
+# include <limits.h>
+# if __WORDSIZE == 64
+#  define INT64_C(c)    c ## L
+# else
+#  define INT64_C(c)    c ## LL
+# endif
+# define UINT32_C(c)    c ## U
+# if __WORDSIZE == 64
+#  define UINT64_C(c)   c ## UL
+# else
+#  define UINT64_C(c)   c ## ULL
+# endif
+#endif
+
 #endif
--- LVM2.2.02.45/tools/tools.h	2008-11-04 15:57:06.000000000 +0100
+++ LVM2.2.02.45-diet/tools/tools.h	2009-03-18 12:24:58.000000000 +0100
@@ -55,6 +55,21 @@
 #include <stdarg.h>
 #include <sys/types.h>
 
+#ifdef __dietlibc__
+# include <limits.h>
+# if __WORDSIZE == 64
+#  define INT64_C(c)    c ## L
+# else
+#  define INT64_C(c)    c ## LL
+# endif
+# define UINT32_C(c)    c ## U
+# if __WORDSIZE == 64
+#  define UINT64_C(c)   c ## UL
+# else
+#  define UINT64_C(c)   c ## ULL
+# endif
+#endif
+
 #define CMD_LEN 256
 #define MAX_ARGS 64
 
--- LVM2.2.02.45/lib/report/report.c	2009-02-09 10:45:49.000000000 +0100
+++ LVM2.2.02.45-diet/lib/report/report.c	2009-03-18 12:45:23.000000000 +0100
@@ -24,6 +24,11 @@
 #include "str_list.h"
 #include "lvmcache.h"
 
+#ifdef __dietlibc__
+#define __UINT64_C(c) c ## ULL
+# define UINT64_MAX             (__UINT64_C(18446744073709551615))
+#endif
+
 struct lvm_report_object {
 	struct volume_group *vg;
 	struct logical_volume *lv;
--- LVM2.2.02.45/tools/dmsetup.c	2008-10-30 18:52:06.000000000 +0100
+++ LVM2.2.02.45-diet/tools/dmsetup.c	2009-03-18 12:47:46.000000000 +0100
@@ -40,6 +40,21 @@
 #include <fcntl.h>
 #include <sys/stat.h>
 
+#ifdef __dietlibc__
+# include <limits.h>
+# if __WORDSIZE == 64
+#  define INT64_C(c)    c ## L
+# else
+#  define INT64_C(c)    c ## LL
+# endif
+# define UINT32_C(c)    c ## U
+# if __WORDSIZE == 64
+#  define UINT64_C(c)   c ## UL
+# else
+#  define UINT64_C(c)   c ## ULL
+# endif
+#endif
+
 /* FIXME Unused so far */
 #undef HAVE_SYS_STATVFS_H

--- LVM2.2.02.65/tools/Makefile.in~	2010-05-18 13:18:54.344217166 +0200
+++ LVM2.2.02.65/tools/Makefile.in	2010-05-18 13:21:19.303172002 +0200
@@ -122,17 +122,17 @@
 device-mapper: $(TARGETS_DM)
 
 dmsetup: dmsetup.o $(top_builddir)/libdm/libdevmapper.$(LIB_SUFFIX)
-	$(CC) $(CFLAGS) $(LDFLAGS) -L$(top_builddir)/libdm \
+	$(CC) $(CFLAGS) -L$(interfacedir) -L$(top_builddir)/libdm $(LDFLAGS) \
 	      -o $@ dmsetup.o -ldevmapper $(LIBS)
 
 dmsetup.static: dmsetup.o $(interfacebuilddir)/libdevmapper.a
-	$(CC) $(CFLAGS) $(LDFLAGS) -static -L$(interfacebuilddir) \
+	$(CC) $(CFLAGS) -static -L$(interfacebuilddir) $(LDFLAGS) \
 	      -o $@ dmsetup.o -ldevmapper $(STATIC_LIBS) $(LIBS)
 
 all: device-mapper
 
 lvm: $(OBJECTS) lvm.o $(top_builddir)/lib/liblvm-internal.a
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(OBJECTS) lvm.o \
+	$(CC) $(CFLAGS) -L$(interfacedir) $(LDFLAGS) -o $@ $(OBJECTS) lvm.o \
 		$(LVMLIBS) $(READLINE_LIBS) $(LIBS) -rdynamic
 
 lvm.static: $(OBJECTS) lvm-static.o $(top_builddir)/lib/liblvm-internal.a  $(interfacebuilddir)/libdevmapper.a

--- LVM2.2.02.65/tools/Makefile.in~	2010-05-18 13:39:46.797633173 +0200
+++ LVM2.2.02.65/tools/Makefile.in	2010-05-18 13:46:01.629690313 +0200
@@ -137,7 +137,7 @@
 
 lvm.static: $(OBJECTS) lvm-static.o $(top_builddir)/lib/liblvm-internal.a  $(interfacebuilddir)/libdevmapper.a
 	$(CC) $(CFLAGS) $(LDFLAGS) -static -L$(interfacebuilddir) -o $@ \
-	      $(OBJECTS) lvm-static.o $(LVMLIBS) $(STATIC_LIBS) $(LIBS)
+	      $(OBJECTS) lvm-static.o $(LVMLIBS) $(STATIC_LIBS) $(LIBS) $(DIETLIBC_LIBS)
 
 liblvm2cmd.a: $(top_builddir)/lib/liblvm-internal.a $(OBJECTS) lvmcmdlib.o lvm2cmd.o
 	cat $(top_builddir)/lib/liblvm-internal.a > $@
