diff -ur subversion-1.4.0.orig/Makefile.in subversion-1.4.0/Makefile.in
--- subversion-1.4.0.orig/Makefile.in	2006-11-06 16:36:17.000000000 +0200
+++ subversion-1.4.0/Makefile.in	2006-11-06 16:50:14.000000000 +0200
@@ -209,10 +209,6 @@
       $(INSTALL_DATA) "$$i" $(DESTDIR)$(swig_pydir); \
     done; \
   fi; \
-  $(PYTHON) -c 'import compileall; \
-    compileall.compile_dir("$(DESTDIR)$(swig_pydir)", 1, "$(swig_pydir)"); \
-    compileall.compile_dir("$(DESTDIR)$(swig_pydir_extra)", 1, \
-    "$(swig_pydir_extra)");'
 
 # The path to generated and complementary source files for the SWIG
 # bindings.
