Index: subversion-1.6.16/build/ac-macros/kwallet.m4
===================================================================
--- subversion-1.6.16.orig/build/ac-macros/kwallet.m4
+++ subversion-1.6.16/build/ac-macros/kwallet.m4
@@ -48,7 +48,7 @@ AC_DEFUN(SVN_LIB_KWALLET,
                   done
                   qt_include_dirs="`$PKG_CONFIG --cflags-only-I QtCore QtDBus QtGui`"
                   kde_dir="`$kde4_config --prefix`"
-                  SVN_KWALLET_INCLUDES="$DBUS_CPPFLAGS $qt_include_dirs -I$kde_dir/include"
+                  SVN_KWALLET_INCLUDES="$DBUS_CPPFLAGS $qt_include_dirs -I`$kde4_config --path include`"
                   qt_libs_other_options="`$PKG_CONFIG --libs-only-other QtCore QtDBus QtGui`"
                   SVN_KWALLET_LIBS="$DBUS_LIBS -lQtCore -lQtDBus -lQtGui -lkdecore -lkdeui $qt_libs_other_options"
                   CXXFLAGS="$CXXFLAGS $SVN_KWALLET_INCLUDES"
