---
 build/ac-macros/neon.m4 |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/build/ac-macros/neon.m4
+++ b/build/ac-macros/neon.m4
@@ -153,7 +153,7 @@ AC_DEFUN(SVN_NEON_CONFIG,
            test "$svn_allowed_neon" = "any"; then
             svn_allowed_neon_on_system="yes"
             SVN_NEON_INCLUDES=[`$neon_config --cflags | sed -e 's/-D[^ ]*//g'`]
-            NEON_LIBS=`$neon_config --la-file`
+            NEON_LIBS=`$neon_config --libs`
             CFLAGS=["$CFLAGS `$neon_config --cflags | sed -e 's/-I[^ ]*//g'`"]
             svn_lib_neon="yes"
             break
