--- xscreensaver-5.12/hacks/flame.c.debug	2010-05-24 14:39:51.000000000 +0900
+++ xscreensaver-5.12/hacks/flame.c	2010-10-13 23:16:00.000000000 +0900
@@ -173,7 +173,7 @@
 static int
 recurse (struct state *st, double x, double y, int l, Display *dpy, Window win)
 {
-  int /*xp, yp,*/ i;
+  int i;
   double nx, ny;
 
   if (l == st->max_levels)
@@ -184,8 +184,8 @@
 
       if (x > -1.0 && x < 1.0 && y > -1.0 && y < 1.0)
 	{
-/*	  xp = st->points[st->num_points].x = (int) ((st->width / 2) * (x + 1.0));
-	  yp = st->points[st->num_points].y = (int) ((st->height / 2) * (y + 1.0));*/
+	  st->points[st->num_points].x = (int) ((st->width / 2) * (x + 1.0));
+	  st->points[st->num_points].y = (int) ((st->height / 2) * (y + 1.0));
 	  st->num_points++;
 	  if (st->num_points >= POINT_BUFFER_SIZE)
 	    {
