--- xscreensaver-5.12/hacks/Makefile.in.list	2010-10-13 18:02:21.000000000 +0900
+++ xscreensaver-5.12/hacks/Makefile.in	2010-10-13 18:28:53.000000000 +0900
@@ -118,7 +118,7 @@
 # Programs that are mentioned in XScreenSaver.ad, and that have XML files,
 # but that are not shipped with xscreensaver itself.
 #
-EXTERNALS	= cosmos electricsheep extrusion fireflies goban \
+EXTERNALS	= cosmos electricsheep fireflies goban \
 		  sphereeversion ssystem xaos xdaliclock xearth xfishtank \
 		  xmountains xplanet xsnow
 
