--- xscreensaver-5.12/driver/xscreensaver.c.debug	2010-09-15 17:43:27.000000000 +0900
+++ xscreensaver-5.12/driver/xscreensaver.c	2010-09-20 01:33:26.000000000 +0900
@@ -1382,6 +1382,7 @@
   struct passwd *spasswd;
   int i;
 
+#if 0
 #ifdef ENABLE_NLS
   if (!setlocale (LC_ALL, ""))
     fprintf (stderr, "locale not supported by C library\n");
@@ -1389,6 +1390,7 @@
   bindtextdomain (GETTEXT_PACKAGE, LOCALEDIR);
   textdomain (GETTEXT_PACKAGE);
 #endif /* ENABLE_NLS */
+#endif
 
   memset(si, 0, sizeof(*si));
   global_si_kludge = si;	/* I hate C so much... */
