--- xscreensaver-5.11/Makefile.in.debug	2010-04-12 13:49:19.000000000 +0900
+++ xscreensaver-5.11/Makefile.in	2010-04-12 20:36:05.000000000 +0900
@@ -99,8 +99,8 @@
 # or later, feel free.  Personally, I can't be bothered.
 #
 configure::
-	autoconf263
-	autoheader263
+	autoconf
+	autoheader
 	@TMP=configure.$$$$ ;						     \
 	echo "munging configure's --help message..." ;			     \
 	( perl -e '							     \
--- xscreensaver-5.11/configure.in.debug	2010-01-02 20:28:15.000000000 +0900
+++ xscreensaver-5.11/configure.in	2010-04-12 20:40:46.000000000 +0900
@@ -315,7 +315,7 @@
 ###############################################################################
 
 AC_DEFUN(AC_PROG_CC_ANSI,
- [AC_PROG_CC
+ [AC_REQUIRE([AC_PROG_CC])
 
   if test -z "$GCC"; then
     # not using GCC
