diff -p -up ./src/xfce-backdrop.c.orig ./src/xfce-backdrop.c
--- ./src/xfce-backdrop.c.orig	2012-03-06 22:51:16.000000000 +0000
+++ ./src/xfce-backdrop.c	2012-03-07 18:32:32.720764864 +0000
@@ -345,12 +345,12 @@ xfce_backdrop_init(XfceBackdrop *backdro
     backdrop->priv->backdrop_list = NULL;
 
     /* color defaults */
-    backdrop->priv->color1.red = 0x1515;
-    backdrop->priv->color1.green = 0x2222;
-    backdrop->priv->color1.blue = 0x3333;
-    backdrop->priv->color2.red = 0x1515;
-    backdrop->priv->color2.green = 0x2222;
-    backdrop->priv->color2.blue = 0x3333;
+    backdrop->priv->color1.red = 0x3b00;
+    backdrop->priv->color1.green = 0x5b00;
+    backdrop->priv->color1.blue = 0x8900;
+    backdrop->priv->color2.red = 0x2800;
+    backdrop->priv->color2.green = 0x5400;
+    backdrop->priv->color2.blue = 0xb000;
 }
 
 static void

