From 18b8fd4603876baa4d1fe546d9199ac95c30bfc3 Mon Sep 17 00:00:00 2001
From: Eric Koegel <eric.koegel@gmail.com>
Date: Sun, 3 Jun 2012 09:11:29 +0300
Subject: [PATCH] Single click continues to work after logout (Bug #8750)

---
 src/xfdesktop-icon-view.c |   14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/src/xfdesktop-icon-view.c b/src/xfdesktop-icon-view.c
index ef5e869..6c52f6c 100644
--- a/src/xfdesktop-icon-view.c
+++ b/src/xfdesktop-icon-view.c
@@ -63,6 +63,8 @@
 #define SCREEN_MARGIN     8
 #define DEFAULT_RUBBERBAND_ALPHA  64
 
+#define SINGLE_CLICK_PROPERTY "/desktop-icons/single-click"
+
 #if defined(DEBUG) && DEBUG > 0
 #define DUMP_GRID_LAYOUT(icon_view) \
 {\
@@ -3372,16 +3374,22 @@ GtkWidget *
 xfdesktop_icon_view_new(XfdesktopIconViewManager *manager)
 {
     XfdesktopIconView *icon_view;
+    XfconfChannel *channel = NULL;
     
     g_return_val_if_fail(XFDESKTOP_IS_ICON_VIEW_MANAGER(manager), NULL);
     
     icon_view = g_object_new(XFDESKTOP_TYPE_ICON_VIEW, NULL);
     icon_view->priv->manager = manager;
 
-    icon_view->priv->channel = xfconf_channel_get(XFDESKTOP_CHANNEL);
+    channel = xfconf_channel_get(XFDESKTOP_CHANNEL);
+    icon_view->priv->channel = channel;
 
-    xfconf_g_property_bind(icon_view->priv->channel,
-                           "/desktop-icons/single-click",
+    /* Set the single_click property before we bind to it */
+    icon_view->priv->single_click = xfconf_channel_get_bool(channel,
+                                                            SINGLE_CLICK_PROPERTY,
+                                                            FALSE);
+    xfconf_g_property_bind(channel,
+                           SINGLE_CLICK_PROPERTY,
                            G_TYPE_BOOLEAN,
                            G_OBJECT(icon_view),
                            "single_click");
-- 
1.7.9.5

