--- gnome/gretl.desktop.orig	2008-06-25 06:45:37.000000000 +0300
+++ gnome/gretl.desktop	2008-08-06 23:09:35.000000000 +0300
@@ -1,9 +1,12 @@
 [Desktop Entry]
-Name=gretl
-Comment=Econometrics program
+Version=1.0
+Encoding=UTF-8
+Name=Gretl
+GenericName=Econometrics program
+GenericName[tr]=Ekonometri programı
 Exec=gretl %f
-Icon=gretl.xpm
-Terminal=0
+Icon=gretl
+Terminal=false
+StartupNotify=True
 Type=Application
 Categories=Application;Science;Econometrics
-
