From b4735104164bb5b5b23f885caa5011f8506eb773 Mon Sep 17 00:00:00 2001
From: Alexey Gladkov <gladkov.alexey@gmail.com>
Date: Tue, 6 Jul 2010 01:44:00 +0400
Subject: [PATCH 3/6] Do not set prefer_unicode for iso-8859-1

Signed-off-by: Alexey Gladkov <gladkov.alexey@gmail.com>
---
 src/ksyms.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/src/ksyms.c b/src/ksyms.c
index 85eaa49..a7b7394 100644
--- a/src/ksyms.c
+++ b/src/ksyms.c
@@ -1701,6 +1701,12 @@ set_charset(const char *charset) {
 					syms[0].table[i] = p->name;
 			}
 			strcpy(chosen_charset, charset);
+
+			/* Unicode: The first 256 code points were made
+			   identical to the content of ISO 8859-1 */
+			if (prefer_unicode && !strcasecmp(charset, "iso-8859-1"))
+				prefer_unicode = 0;
+
 			return 0;
 		}
 	}
-- 
1.7.3.4

