Index: dev86-0.16.17/libc/Config_sh
===================================================================
--- dev86-0.16.17.orig/libc/Config_sh
+++ dev86-0.16.17/libc/Config_sh
@@ -46,8 +46,8 @@ main()
    do
       display
       echo 
-      echon 'Select config option to flip [or quit] >'
-      read n
+      echon 'Do not select config option, I can do it for you ;)'
+      n="q" # read n
       v=""
       case "$n" in
       [qQ]* ) RUNNING=0
