Index: dev86-0.16.17/libc/i386sys/Makefile
===================================================================
--- dev86-0.16.17.orig/libc/i386sys/Makefile
+++ dev86-0.16.17/libc/i386sys/Makefile
@@ -20,6 +20,10 @@ CFLAGS=$(ARCH) $(CCFLAGS) $(DEFS)
 all: $(SYSCALLS) $(LIBC)($(OBJ))
 	@$(RM) $(OBJ)
 
+# when OBJ is null at $(LIBC)($(OBJ)), make crashes
+$(LIBC)():
+	@:
+
 syscalls: syscall.mak
 	$(MAKE) -f syscall.mak LIBC="$(LIBC)" CFLAGS="$(CFLAGS)"
 
