Index: src/PlanetProperties.cpp
===================================================================
--- src/PlanetProperties.cpp
+++ src/PlanetProperties.cpp	2009-05-25 12:31:55.000000000 +0200
@@ -1,6 +1,8 @@
 #include <algorithm>
 #include <cctype>
 #include <cstring>
+#include <cstdio>
+#include <cstdlib>
 #include <string>
 #include <vector>
 using namespace std;
Index: src/Satellite.cpp
===================================================================
--- src/Satellite.cpp
+++ src/Satellite.cpp	2009-05-25 11:44:36.000000000 +0200
@@ -1,5 +1,6 @@
 #include <clocale>
 #include <cstring>
+#include <cstdio>
 #include <iostream>
 using namespace std;
 
Index: src/Separation.cpp
===================================================================
--- src/Separation.cpp
+++ src/Separation.cpp	2009-05-25 11:50:11.000000000 +0200
@@ -1,5 +1,6 @@
 #include <cmath>
 #include <iostream>
+#include <cstdio>
 
 #include "Separation.h"
 #include "View.h"
