Index: libupnp-1.6.6/m4/acx_pthread.m4
===================================================================
--- libupnp-1.6.6.orig/m4/acx_pthread.m4
+++ libupnp-1.6.6/m4/acx_pthread.m4
@@ -146,7 +146,7 @@ acx_pthread_flags="pthreads none -Kthrea
 # pthread-config: use pthread-config program (for GNU Pth library)
 
 case "${host_cpu}-${host_os}" in
-        *solaris*)
+        *solaris*|*linux*)
 
         # On Solaris (at least, for some versions), libc contains stubbed
         # (non-functional) versions of the pthreads routines, so link-based
