Index: pm-utils-1.4.1/pm/sleep.d/55NetworkManager
===================================================================
--- pm-utils-1.4.1.orig/pm/sleep.d/55NetworkManager	2010-09-20 11:12:23.750155840 +0200
+++ pm-utils-1.4.1/pm/sleep.d/55NetworkManager	2010-09-20 11:12:25.610155850 +0200
@@ -13,6 +13,7 @@
 	# Tell NetworkManager to shut down networking
         printf "Having NetworkManager put all interaces to sleep..."
 	dbus_send --system                         \
+                --print-reply --reply-timeout=200      \
 		--dest=org.freedesktop.NetworkManager  \
 		/org/freedesktop/NetworkManager        \
 		org.freedesktop.NetworkManager.sleep && \
@@ -25,6 +26,7 @@
 	printf "Having NetworkManager wake interfaces back up..."
         dbus_send --system                        \
 		--dest=org.freedesktop.NetworkManager \
+                --print-reply --reply-timeout=200      \
 		/org/freedesktop/NetworkManager       \
 		org.freedesktop.NetworkManager.wake && \
 	    echo Done. || echo Failed.
