diff -urN 4Suite-XML/test/Xml/XPath/Core/test_location_path.py 4Suite-XML-1.0.2/test/Xml/XPath/Core/test_location_path.py
--- 4Suite-XML/test/Xml/XPath/Core/test_location_path.py	2004-10-05 00:55:01.000000000 +0200
+++ 4Suite-XML-1.0.2/test/Xml/XPath/Core/test_location_path.py	2010-02-26 16:37:33.916336763 +0100
@@ -15,8 +15,8 @@
     DomTree = tester.test_data['tree']
 
     nt = ParsedNodeTest.ParsedNameTest('*')
-    as = ParsedAxisSpecifier.ParsedAxisSpecifier('child')
-    step = ParsedStep.ParsedStep(as,nt,None)
+    as_ = ParsedAxisSpecifier.ParsedAxisSpecifier('child')
+    step = ParsedStep.ParsedStep(as_,nt,None)
     # [(expression, context, expected)...]
     tests = [(ParsedAbbreviatedAbsoluteLocationPath(step),
               Context.Context(DomTree.CHILD2, 1, 1),
diff -urN 4Suite-XML/test/Xml/XPath/Core/test_step.py 4Suite-XML-1.0.2/test/Xml/XPath/Core/test_step.py
--- 4Suite-XML/test/Xml/XPath/Core/test_step.py	2001-09-15 20:46:11.000000000 +0200
+++ 4Suite-XML-1.0.2/test/Xml/XPath/Core/test_step.py	2010-02-26 16:38:10.654334053 +0100
@@ -21,38 +21,38 @@
     tests = []
 
     # Test 1
-    as = ParsedAxisSpecifier.ParsedAxisSpecifier('ancestor')
+    as_ = ParsedAxisSpecifier.ParsedAxisSpecifier('ancestor')
     nt = ParsedNodeTest.ParsedNameTest('*')
-    s = ParsedStep.ParsedStep(as, nt)
+    s = ParsedStep.ParsedStep(as_, nt)
     tests.append((s, [DomTree.ROOT]))
     
     # Test 2
-    as = ParsedAxisSpecifier.ParsedAxisSpecifier('ancestor-or-self')
-    s = ParsedStep.ParsedStep(as, nt, None)
+    as_ = ParsedAxisSpecifier.ParsedAxisSpecifier('ancestor-or-self')
+    s = ParsedStep.ParsedStep(as_, nt, None)
     tests.append((s, [DomTree.ROOT, DomTree.CHILD1]))
 
     # Test 3
-    as = ParsedAxisSpecifier.ParsedAxisSpecifier('descendant-or-self')
+    as_ = ParsedAxisSpecifier.ParsedAxisSpecifier('descendant-or-self')
     nt = ParsedNodeTest.ParsedNameTest('GCHILD')
-    s = ParsedStep.ParsedStep(as, nt)
+    s = ParsedStep.ParsedStep(as_, nt)
     tests.append((s, DomTree.GCHILDREN1))
     
     # Test 4
-    as = ParsedAxisSpecifier.ParsedAxisSpecifier('child')
+    as_ = ParsedAxisSpecifier.ParsedAxisSpecifier('child')
     nt = ParsedNodeTest.ParsedNameTest('GCHILD')
     left = ParsedExpr.ParsedFunctionCallExpr('position', [])
     right = ParsedExpr.ParsedNLiteralExpr('1')
     exp = ParsedExpr.ParsedEqualityExpr('=', left, right)
     pl = ParsedPredicateList.ParsedPredicateList([exp])
-    s = ParsedStep.ParsedStep(as, nt, pl)
+    s = ParsedStep.ParsedStep(as_, nt, pl)
     tests.append((s, [DomTree.GCHILD11]))
 
     # Test 5
     right = ParsedExpr.ParsedNLiteralExpr('1')
     pl = ParsedPredicateList.ParsedPredicateList([right])
-    as = ParsedAxisSpecifier.ParsedAxisSpecifier('child')
+    as_ = ParsedAxisSpecifier.ParsedAxisSpecifier('child')
     nt = ParsedNodeTest.ParsedNameTest('GCHILD')
-    s = ParsedStep.ParsedStep(as,nt,pl)
+    s = ParsedStep.ParsedStep(as_,nt,pl)
     tests.append((s, [DomTree.GCHILD11]))
     
     tester.testDone()
