Index: hplip-3.10.5/systray.py
===================================================================
--- hplip-3.10.5.orig/systray.py
+++ hplip-3.10.5/systray.py
@@ -48,6 +48,10 @@ if __name__ == '__main__':
     if os.getpgid(os.getpid()) != os.getpid():
         os.setsid()
 
+    if "-session" in sys.argv:
+        si = sys.argv.index("-session")
+        del sys.argv[si:si+2]
+
     mod = module.Module(__mod__, __title__, __version__, __doc__, None,
                        (GUI_MODE,), (UI_TOOLKIT_QT4, UI_TOOLKIT_QT3))
 
