Index: hplip-3.9.6b/base/module.py
===================================================================
--- hplip-3.9.6b.orig/base/module.py
+++ hplip-3.9.6b/base/module.py
@@ -149,8 +149,8 @@ class Module(object):
         #log.debug("Default ui-toolkit: %s" % self.default_ui_toolkit)
         #log.debug("Default mode: %s" % self.default_mode)
 
-        if os.getuid() == 0 and not run_as_root_ok:
-            log.warn("%s should not be run as root/superuser." % mod)
+        #if os.getuid() == 0 and not run_as_root_ok:
+        #    log.warn("%s should not be run as root/superuser." % mod)
 
 
     def setUsage(self, include_flags=0, extra_options=None,
