Fedora-specific makefile patch.

--- iputils-s20121221-dist/Makefile     2013-01-07 13:37:04.913224305 +0100
+++ iputils-s20121221-new/Makefile      2013-01-07 13:42:56.542052062 +0100
@@ -27,7 +27,7 @@
 # sysfs support (with libsysfs - deprecated) [no|yes|static]
 USE_SYSFS=no
 # IDN support (experimental) [no|yes|static]
-USE_IDN=no
+USE_IDN=yes
 
 # Do not use getifaddrs [no|yes|static]
 WITHOUT_IFADDRS=no
@@ -35,7 +35,7 @@
 ARPING_DEFAULT_DEVICE=
 
 # GNU TLS library for ping6 [yes|no|static]
-USE_GNUTLS=yes
+USE_GNUTLS=no
 # Crypto library for ping6 [shared|static]
 USE_CRYPTO=shared
 # Resolv library for ping6 [yes|static]
@@ -49,7 +49,7 @@
 # -------------------------------------
 # What a pity, all new gccs are buggy and -Werror does not work. Sigh.
 # CCOPT=-fno-strict-aliasing -Wstrict-prototypes -Wall -Werror -g
-CCOPT=-fno-strict-aliasing -Wstrict-prototypes -Wall -g
+CCOPT=-fno-strict-aliasing -Wstrict-prototypes -Wall -g -Werror
 CCOPTOPT=-O3
 GLIBCFIX=-D_GNU_SOURCE
 DEFINES=
@@ -110,7 +110,7 @@
 IPV6_TARGETS=tracepath6 traceroute6 ping6
 TARGETS=$(IPV4_TARGETS) $(IPV6_TARGETS)
 
-CFLAGS=$(CCOPTOPT) $(CCOPT) $(GLIBCFIX) $(DEFINES)
+CFLAGS+=$(RPM_OPT_FLAGS) $(CCOPTOPT) $(CCOPT) $(GLIBCFIX) $(DEFINES)
 LDLIBS=$(LDLIB) $(ADDLIB)
 
 UNAME_N:=$(shell uname -n)
@@ -187,6 +187,7 @@
 
 # -------------------------------------
 # ninfod
+ninfod: CFLAGS:=$(filter-out -Werror,$(CFLAGS))
 ninfod:
        @set -e; \
                if [ ! -f ninfod/Makefile ]; then \
