diff -up doxygen-1.8.6/addon/doxywizard/Makefile.in.config doxygen-1.8.6/addon/doxywizard/Makefile.in
--- doxygen-1.8.6/addon/doxywizard/Makefile.in.config	2013-12-26 12:17:57.281368574 -0700
+++ doxygen-1.8.6/addon/doxywizard/Makefile.in	2013-12-26 12:20:44.673513404 -0700
@@ -10,7 +10,6 @@
 # See the GNU General Public License for more details.
 # 
 
-QMAKE=$(QTDIR)/bin/qmake $(MKSPECS)
 INCBUFSIZE=$(PYTHON) ../../src/increasebuffer.py
 
 all: Makefile.doxywizard  
diff -up doxygen-1.8.6/configure.config doxygen-1.8.6/configure
--- doxygen-1.8.6/configure.config	2013-12-24 09:14:46.000000000 -0700
+++ doxygen-1.8.6/configure	2013-12-26 12:20:20.542636493 -0700
@@ -326,9 +326,10 @@ if test "$f_wizard" = YES; then
   if test -z "$QTDIR"; then
     echo "  QTDIR environment variable not set!"
     printf "  Checking for Qt..."
-    for d in /usr/{lib,share,qt}/{qt-4,qt4,qt,qt*,4} /usr; do
+    for d in /usr/{lib64,lib,share,qt}/{qt-4,qt4,qt,qt*,4} /usr; do
       if test -x "$d/bin/qmake"; then
         QTDIR=$d
+        QMAKE=$d/bin/qmake
         break 2
       fi
     done
@@ -704,6 +705,8 @@ INSTTOOL  = $f_insttool
 DOXYDOCS  = ..
 DOCDIR    = $f_docdir
 QTDIR     = $QTDIR
+QMAKE     = $QMAKE
+MAN1DIR    = share/man/man1
 EOF
 
 if test "$f_dot" != NO; then
diff -up doxygen-1.8.6/Makefile.in.config doxygen-1.8.6/Makefile.in
--- doxygen-1.8.6/Makefile.in.config	2013-12-26 12:17:57.283368564 -0700
+++ doxygen-1.8.6/Makefile.in	2013-12-26 12:19:17.796956667 -0700
@@ -78,8 +78,6 @@ realclean: clean
 
 DATE=$(shell date "+%B %Y")
 
-MAN1DIR = man/man1
-
 install: doxywizard_install doxysearch_install
 	$(INSTTOOL) -d $(DESTDIR)$(INSTALL)/bin
 	$(INSTTOOL) -m 755 bin/doxygen        $(DESTDIR)$(INSTALL)/bin
diff -up doxygen-1.8.6/tmake/lib/linux-g++/tmake.conf.config doxygen-1.8.6/tmake/lib/linux-g++/tmake.conf
--- doxygen-1.8.6/tmake/lib/linux-g++/tmake.conf.config	2013-11-02 08:18:53.000000000 -0600
+++ doxygen-1.8.6/tmake/lib/linux-g++/tmake.conf	2013-12-26 12:18:38.268158455 -0700
@@ -11,7 +11,7 @@ TMAKE_CC		= gcc
 TMAKE_CFLAGS		= -pipe -fsigned-char
 TMAKE_CFLAGS_WARN_ON	= -Wall -W 
 TMAKE_CFLAGS_WARN_OFF	=
-TMAKE_CFLAGS_RELEASE	= -O2
+TMAKE_CFLAGS_RELEASE	= $(RPM_OPT_FLAGS)
 TMAKE_CFLAGS_DEBUG	= -g
 TMAKE_CFLAGS_SHLIB	= -fPIC
 TMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
@@ -27,12 +27,12 @@ TMAKE_CXXFLAGS_YACC	= $$TMAKE_CFLAGS_YAC
 
 TMAKE_INCDIR		=
 TMAKE_LIBDIR		=
-TMAKE_INCDIR_X11	= /usr/X11R6/include
-TMAKE_LIBDIR_X11	= /usr/X11R6/lib
-TMAKE_INCDIR_QT		= $(QTDIR)/include
-TMAKE_LIBDIR_QT		= $(QTDIR)/lib
-TMAKE_INCDIR_OPENGL	= /usr/X11R6/include
-TMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
+TMAKE_INCDIR_X11	=
+TMAKE_LIBDIR_X11	=
+TMAKE_INCDIR_QT		=
+TMAKE_LIBDIR_QT		=
+TMAKE_INCDIR_OPENGL	=
+TMAKE_LIBDIR_OPENGL	=
 
 TMAKE_LINK		= g++
 TMAKE_LINK_SHLIB	= g++
