diff -up doxygen-1.8.1.1/src/configoptions.cpp.html_timestamp_off.patch doxygen-1.8.1.1/src/configoptions.cpp
--- doxygen-1.8.1.1/src/configoptions.cpp.html_timestamp_off.patch	2012-06-09 06:41:04.000000000 -0500
+++ doxygen-1.8.1.1/src/configoptions.cpp	2012-06-13 13:31:15.291500215 -0500
@@ -1270,7 +1270,7 @@ void addConfigOptions(Config *cfg)
                  "If the HTML_TIMESTAMP tag is set to YES then the footer of each generated HTML\n"
                  "page will contain the date and time when the page was generated. Setting\n"
                  "this to NO can help when comparing the output of multiple runs.",
-                 TRUE
+                 FALSE 
                 );
   cb->addDependency("GENERATE_HTML");
   //----
