Index: virt-manager-1.0.1/virt-clone
===================================================================
--- virt-manager-1.0.1.orig/virt-clone
+++ virt-manager-1.0.1/virt-clone
@@ -1,4 +1,4 @@
-#!/usr/bin/python2 -tt
+#!/usr/bin/python2.7 -tt
 #
 # Copyright(c) FUJITSU Limited 2007.
 #
Index: virt-manager-1.0.1/virt-convert
===================================================================
--- virt-manager-1.0.1.orig/virt-convert
+++ virt-manager-1.0.1/virt-convert
@@ -1,4 +1,4 @@
-#!/usr/bin/python2 -tt
+#!/usr/bin/python2.7 -tt
 #
 # Copyright 2008, 2013, 2014  Red Hat, Inc.
 # Joey Boggs <jboggs@redhat.com>
Index: virt-manager-1.0.1/virt-image
===================================================================
--- virt-manager-1.0.1.orig/virt-image
+++ virt-manager-1.0.1/virt-image
@@ -1,4 +1,4 @@
-#!/usr/bin/python2 -tt
+#!/usr/bin/python2.7 -tt
 #
 # Create a virtual machine from an XML image description
 #
Index: virt-manager-1.0.1/virt-install
===================================================================
--- virt-manager-1.0.1.orig/virt-install
+++ virt-manager-1.0.1/virt-install
@@ -1,4 +1,4 @@
-#!/usr/bin/python2 -tt
+#!/usr/bin/python2.7 -tt
 #
 # Copyright 2005-2014 Red Hat, Inc.
 #
Index: virt-manager-1.0.1/virt-manager
===================================================================
--- virt-manager-1.0.1.orig/virt-manager
+++ virt-manager-1.0.1/virt-manager
@@ -1,4 +1,4 @@
-#!/usr/bin/python2 -tt
+#!/usr/bin/python2.7 -tt
 #
 # Copyright (C) 2006, 2014 Red Hat, Inc.
 # Copyright (C) 2006 Daniel P. Berrange <berrange@redhat.com>
Index: virt-manager-1.0.1/virt-xml
===================================================================
--- virt-manager-1.0.1.orig/virt-xml
+++ virt-manager-1.0.1/virt-xml
@@ -1,4 +1,4 @@
-#!/usr/bin/python2 -tt
+#!/usr/bin/python2.7 -tt
 #
 # Copyright 2013-2014 Red Hat, Inc.
 # Cole Robinson <crobinso@redhat.com>
