--- bin/mudur.py~	2013-06-26 21:18:14.642465426 +0200
+++ bin/mudur.py	2013-06-26 21:18:29.641464878 +0200
@@ -17,6 +17,7 @@
 import os
 import re
 import sys
+import stat
 import time
 import signal
 import gettext
--- bin/mudur.py~	2013-09-10 23:35:51.819342944 +0200
+++ bin/mudur.py	2013-09-10 23:38:30.406337150 +0200
@@ -838,6 +838,7 @@
     # When these files are missing, lots of trouble happens
     # so we double check their existence
     create_directory("/run/shm")
+    if not stat.S_IMODE(os.stat("./test").st_mode) == 511: os.chmod("./test", 511)
     create_link("/run/shm", "/dev/shm")
 
     # Start udev daemon
