--- a/setup.py	(revision 33872)
+++ b/setup.py	(revision 33873)
@@ -25,7 +25,7 @@
     po/*.po
 """
 
-i18n_source_list = ["bin/mudur.py", "bin/service.py", "bin/network.py"]
+i18n_source_list = ["bin/mudur.py", "bin/service.py"]
 
 def update_messages():
     os.system("xgettext -o po/mudur.pot %s" % " ".join(i18n_source_list))
@@ -80,7 +80,6 @@
     install_file("bin/update-fstab.py", prefix, "sbin/update-fstab")
     install_file("bin/compat.py", prefix, "etc/init.d/compat.py")
     install_file("bin/service.py", prefix, "bin/service")
-    install_file("bin/network.py", prefix, "bin/network")
     install_file("bin/adduser.py", prefix, "sbin/adduser")
     install_file("bin/deluser.py", prefix, "sbin/deluser")
     install_file("etc/mudur.conf", prefix, "etc/conf.d/mudur")
