--- bin/mudur_tmpfiles.py~	2014-02-13 08:18:20.039993198 +0100
+++ bin/mudur_tmpfiles.py	2014-02-13 08:46:07.496654410 +0100
@@ -24,7 +24,10 @@
 from pwd import getpwnam
 from grp import getgrnam
 
-DEFAULT_CONFIG_DIRS = ["/etc/tmpfiles.d", "/run/tmpfiles.d", "/usr/lib/tmpfiles.d"]
+# search files order
+DEFAULT_CONFIG_DIRS_SO = ["/etc/tmpfiles.d", "/run/tmpfiles.d", "/usr/lib/tmpfiles.d"]
+# execution order
+DEFAULT_CONFIG_DIRS_EO = ["/run/tmpfiles.d", "/usr/lib/tmpfiles.d", "/etc/tmpfiles.d"]
 
 def read_file(path):
     with open(path) as f:
@@ -64,7 +67,7 @@
 %s
 \tparsing .conf files in:
 \t%s
-""" % (sys.argv[0], sys.argv[0], "; ".join(DEFAULT_CONFIG_DIRS))
+""" % (sys.argv[0], sys.argv[0], "; ".join(DEFAULT_CONFIG_DIRS_SO))
     
 def usage():
     print USAGE
@@ -92,15 +95,22 @@
             add_config_file(head, tail)
     else:
         all_files_names = []
-        for head in DEFAULT_CONFIG_DIRS:
+        for head in DEFAULT_CONFIG_DIRS_SO:
             if not os.path.isdir(head): continue
-            for tail in os.listdir(head):
+            ls = os.listdir(head)
+            for tail in ls:
                 # only .conf files and don't override (previous paths have higer priority)
                 if not tail.endswith(".conf") or tail in all_files_names: continue
                 all_files_names.append(tail)
                 add_config_file(head, tail)
+            if ls and "baselayout.conf" in config_files[head]:
+                config_files[head].insert(0, config_files[head].pop(config_files[head].index("baselayout.conf")))
 
-    for d, fs in config_files.items():
+    for d in DEFAULT_CONFIG_DIRS_EO:
+        try:
+            fs = config_files[d]
+        except KeyError:
+            continue
         for f in fs:
             conf = read_file(os.path.join(d, f))
             # parse config file
@@ -135,4 +145,4 @@
                 # 
                 if len(errors) == cerr: create(*fields)
 
-    print "\n".join(errors)
\ No newline at end of file
+    print "\n".join(errors)
