--- bin/mudur.py~	2014-02-13 17:32:01.289997663 +0100
+++ bin/mudur.py	2014-02-13 17:24:52.000000000 +0100
@@ -1013,6 +1013,19 @@
     UI.info(_("Mounting local filesystems"))
     run("/bin/mount", "-at", "noproc,nocifs,nonfs,nonfs4,noncpfs")
 
+@skip_for_lxc_guests
+@plymouth_update_milestone
+def mount_tmpfs_run():
+    df = os.popen('df 2>/dev/null', 'r')
+    for line in df.readlines():
+        line = line.strip()
+        if line.startswith("tmpfs") and line.endswith("/run"):
+            UI.info(_("Unmounting /run"))
+            run("/bin/umount", "/run")
+            break
+    df.close()
+    run_full("/bin/mount", "-t", "tmpfs", "-o", "nodev,nosuid,size=10%,mode=755", "tmpfs", "/run")
+
 def mount_remote_filesystems():
     """Mounts remote filesystems."""
     from pardus.fstabutils import Fstab
@@ -1362,6 +1375,7 @@
 
         # Mount local filesystems
         mount_local_filesystems()
+        mount_tmpfs_run()
 
         # Activate swap space
         enable_swap()
