Index: bin/update-fstab.py
===================================================================
--- bin/update-fstab.py	(revision 34591)
+++ bin/update-fstab.py	(working copy)
@@ -29,7 +29,7 @@
 
 default_mount_dir = "/mnt"
 
-excluded_file_systems = ("proc", "tmpfs", "sysfs", "swap", "nfs", "nfs4", "cifs")
+excluded_file_systems = ("proc", "tmpfs", "sysfs", "linux-swap", "swap", "nfs", "nfs4", "cifs")
 
 pardus_labels = ("PARDUS_ROOT", "PARDUS_HOME", "PARDUS_SWAP")
 
@@ -49,18 +49,18 @@
     return devices
 
 def blockPartitions(dev):
-    pdev = parted.PedDevice.get(dev)
+    pdev = parted.getDevice(dev)
     try:
-        disk = parted.PedDisk.new(pdev)
+        disk = parted.Disk(device=pdev)
     except:
         # FIXME: replace with what exception could we get here, bare except sucks
-        disk = pdev.disk_new_fresh(parted.disk_type_get("msdos"))
+        disk = parted.freshDisk(pdev, parted.diskType['msdos'])
 
-    part = disk.next_partition()
+    part = disk.getFirstPartition()
     while part:
-        if part.fs_type and part.fs_type.name != "linux-swap":
-            yield dev + str(part.num), part.fs_type.name
-        part = disk.next_partition(part)
+        if part.fileSystem and part.fileSystem.type != "linux-swap":
+            yield part.path, part.fileSystem.type
+        part = part.nextPartition()
 
 def blockNameByLabel(label):
     path = os.path.join("/dev/disk/by-label/%s" % label)
