--- mudur/bin/mudur.py	2012-10-24 17:43:41.000000000 +0300
+++ mudur.orig/bin/mudur.py	2013-03-02 15:44:00.000000000 +0200
@@ -368,7 +368,7 @@
 
         # Fill in the options
         self.options["live"] = options.has_key("thin") or \
-                               os.path.exists("/var/run/pardus/livemedia")
+                               os.path.exists("/run/pisilinux/livemedia")
 
         for k in [_k for _k in options.keys() if _k not in ("thin")]:
             self.options[k] = options[k] if options[k] else True
@@ -490,12 +490,12 @@
     def greet(self):
         """Dump release information, sets unicode mode."""
         print self.UNICODE_MAGIC
-        if os.path.exists("/etc/pardus-release"):
-            release = load_file("/etc/pardus-release").rstrip("\n")
-            print "\x1b[1m  %s  \x1b[0;36mhttp://www.pardus.org.tr\x1b[0m" \
+        if os.path.exists("/etc/pisilinux-release"):
+            release = load_file("/etc/pisilinux-release").rstrip("\n")
+            print "\x1b[1m  %s  \x1b[0;36mhttp://www.pisilinux.org\x1b[0m" \
                     % release
         else:
-            self.error(_("Cannot find /etc/pardus-release"))
+            self.error(_("Cannot find /etc/pisilinux-release"))
         print
 
     def info(self, msg):
@@ -1011,7 +1011,7 @@
 
 def mount_remote_filesystems():
     """Mounts remote filesystems."""
-    from pardus.fstabutils import Fstab
+    from pisilinux.fstabutils import Fstab
     fstab = Fstab()
     if fstab.contains_remote_mounts():
         UI.info(_("Mounting remote filesystems"))
@@ -1047,8 +1047,8 @@
         # kernel already got a hostname (pxeboot or something)
         host = khost
     else:
-        # if nothing found, use the default hostname 'pardus'
-        host = uhost if uhost else "pardus"
+        # if nothing found, use the default hostname 'pisinux'
+        host = uhost if uhost else "pisilinux"
 
     if uhost and host != uhost:
         i = data.find('HOSTNAME="')
@@ -1290,7 +1290,7 @@
     """Hook that intercepts and handles exceptions."""
     import traceback
     print
-    print _("An internal error occured. Please report to the bugs.pardus.org.tr"
+    print _("An internal error occured. Please report to the bugs.pisilinux.org"
             "with following information:").encode("utf-8")
     print
     print e_type, e_value
@@ -1386,7 +1386,7 @@
 
         create_directory("/run/lock")
         create_directory("/run/lock/subsys")
-        create_directory("/run/pardus")
+        create_directory("/run/pisilinux")
 
         run("/bin/chgrp", "utmp", "/run/utmp", "/var/log/wtmp")
 
