--- bin/mudur.py~	2013-09-10 21:16:45.235647911 +0200
+++ bin/mudur.py	2013-09-10 21:17:01.843647304 +0200
@@ -1351,9 +1351,6 @@
         # Minimize dmesg noise
         minimize_printk_log_level()
 
-        # Start udev and event triggering
-        start_udev()
-
         # Check root file system
         check_root_filesystem()
 
@@ -1361,6 +1358,9 @@
         mount_root_filesystem()
         SPLASH.rootfs_is_now_rw()
 
+        # Start udev and event triggering
+        start_udev()
+
         # Grab persistent rules and udev.log file from /dev
         copy_udev_rules()
 
