--- bin/mudur_tmpfiles.py~      2014-08-03 17:48:07.461463934 +0200
+++ bin/mudur_tmpfiles.py       2014-08-03 17:49:06.951461761 +0200
@@ -63,6 +63,7 @@
             os.makedirs(os.path.dirname(path))
             os.chown(os.path.dirname(path), uid, gid)
         write_file(path, arg, mode = "a" if type == "f" else "w")
+        os.chmod(path, mode)
         os.chown(path, uid, gid)
 
 USAGE = """\
