--- bin/mudur.py.orig	2012-12-02 21:55:18.318333761 +0200
+++ bin/mudur.py	2012-12-02 21:55:38.094340932 +0200
@@ -795,6 +795,8 @@
     """Trigger only the events which are failed during a previous run."""
     if os.path.exists("/dev/.udev/failed"):
         run("/sbin/udevadm", "trigger", "--type=failed", "--action=add")
+    #Very hacky code, we have to find a proper solution to trigger devices smoothly
+    run("/sbin/udevadm", "trigger")
 
 @skip_for_lxc_guests
 def copy_udev_rules():
