--- bin/mudur.py.old	2014-01-18 21:01:03.000000000 +0200
+++ bin/mudur.py	2014-02-10 21:06:42.443323201 +0200
@@ -1374,7 +1374,10 @@
 
         # Set the system language
         set_system_language()
-
+        
+        # Cleanup run directory, fixes D-Bus starting issue
+        cleanup_run()
+        
         # When we exit this runlevel, init will write a boot record to utmp
         write_to_file("/run/utmp")
         touch("/var/log/wtmp")
