diff -Naur kde-workspace-4.8.1.orig/systemsettings/app/SettingsBase.cpp kde-workspace-4.8.1/systemsettings/app/SettingsBase.cpp
--- kde-workspace-4.8.1.orig/systemsettings/app/SettingsBase.cpp	2012-03-03 00:11:07.000000000 +0200
+++ kde-workspace-4.8.1/systemsettings/app/SettingsBase.cpp	2012-03-10 16:05:24.433529492 +0200
@@ -34,6 +34,7 @@
 #include <KCModuleInfo>
 #include <KXMLGUIFactory>
 #include <KStandardAction>
+#include <KStandardDirs>
 #include <KActionCollection>
 #include <KServiceTypeTrader>
 
@@ -60,6 +61,12 @@
 
     spacerWidget = new QWidget( this );
     spacerWidget->setSizePolicy( QSizePolicy::MinimumExpanding, QSizePolicy::Maximum );
+
+    // Brand Icon
+    QString brandIcon = KStandardDirs::locate("appdata", "brandicon.png");
+    brandIconWidget = new QLabel( this );
+    brandIconWidget->setPixmap( QPixmap(brandIcon) );
+
     // Initalise the window so we don't flicker
     initToolBar();
     // We can now launch the delayed loading safely
@@ -134,13 +141,19 @@
     spacerAction = new KAction( this );
     spacerAction->setDefaultWidget(spacerWidget);
     actionCollection()->addAction( "spacer", spacerAction );
-    // Finally the search line-edit
+    // Search line-edit
     searchAction = new KAction( this );
     searchAction->setDefaultWidget(searchText);
     searchAction->setShortcut(KShortcut(QKeySequence(Qt::CTRL + Qt::Key_F)));
     connect( searchAction, SIGNAL(triggered(Qt::MouseButtons,Qt::KeyboardModifiers)),
          searchText, SLOT(setFocus()));
     actionCollection()->addAction( "searchText", searchAction );
+
+    // Finally Brand Icon
+    brandIconAction = new KAction( this );
+    brandIconAction->setDefaultWidget(brandIconWidget);
+    actionCollection()->addAction( "brandIcon", brandIconAction );
+
     // Initialise the Window
     setupGUI(Save|Create,QString());
     menuBar()->hide();
@@ -366,7 +379,7 @@
     guiFactory()->unplugActionList( this, "quit" );
     if ( BaseMode::Search & toolbar ) {
         QList<QAction*> searchBarActions;
-        searchBarActions << spacerAction << searchAction;
+        searchBarActions << searchAction;
         guiFactory()->plugActionList( this, "search", searchBarActions );
     }
     if ( BaseMode::Configure & toolbar ) {
diff -Naur kde-workspace-4.8.1.orig/systemsettings/app/SettingsBase.h kde-workspace-4.8.1/systemsettings/app/SettingsBase.h
--- kde-workspace-4.8.1.orig/systemsettings/app/SettingsBase.h	2012-03-03 00:11:07.000000000 +0200
+++ kde-workspace-4.8.1/systemsettings/app/SettingsBase.h	2012-03-10 16:07:40.180424873 +0200
@@ -27,6 +27,7 @@
 
 #include <QMap>
 #include <QStackedWidget>
+#include <QLabel>
 
 #include <KService>
 #include <KLineEdit>
@@ -70,6 +71,8 @@
     // The search bar
     KLineEdit * searchText;
     QWidget * spacerWidget;
+    QLabel * brandIconWidget;
+    KAction * brandIconAction;
     // The toolbar
     KAction * searchAction;
     KAction * spacerAction;
diff -Naur kde-workspace-4.8.1.orig/systemsettings/app/systemsettingsui.rc kde-workspace-4.8.1/systemsettings/app/systemsettingsui.rc
--- kde-workspace-4.8.1.orig/systemsettings/app/systemsettingsui.rc	2012-03-03 00:11:07.000000000 +0200
+++ kde-workspace-4.8.1/systemsettings/app/systemsettingsui.rc	2012-03-10 16:08:26.017038899 +0200
@@ -18,7 +18,9 @@
     <Separator />
     <ActionList name="configure" />
     <Action name="help_toolbar_menu" />
+    <Action name="spacer" />
     <ActionList name="quit" />
     <ActionList name="search" />
+    <Action name="brandIcon" />
   </ToolBar>
 </gui>
