Index: kdebase-workspace-4.5.5/kdm/kfrontend/kdmshutdown.cpp
===================================================================
--- kdebase-workspace-4.5.5.orig/kdm/kfrontend/kdmshutdown.cpp
+++ kdebase-workspace-4.5.5/kdm/kfrontend/kdmshutdown.cpp
@@ -41,6 +41,7 @@ Foundation, Inc., 51 Franklin Street, Fi
 #include <QCheckBox>
 #include <QComboBox>
 #include <QDateTime>
+#include <QFile>
 #include <QFrame>
 #include <QGroupBox>
 #include <QHeaderView>
@@ -612,7 +613,7 @@ void KDMSlimShutdown::slotSuspend(QActio
   // would require string changes (coolo)
   QList<DpySpec> sess;
   kDebug() << "slotSuspend" << _allowSuspend;
-  if (!sess.isEmpty() || _allowSuspend == SHUT_ROOT)
+  if ((!sess.isEmpty() || _allowSuspend == SHUT_ROOT) && !QFile::exists("/run/pisilinux/livemedia"))
   {
       int ret = KDMConfShutdown( -1, sess, SHUT_SUSPEND, 0 ).exec();
       if ( !ret )
@@ -685,7 +686,7 @@ KDMSlimShutdown::checkShutdown(int type,
 {
     reject();
     QList<DpySpec> sess = fetchSessions(lstRemote | lstTTY);
-    if (sess.isEmpty() && _allowShutdown != SHUT_ROOT)
+    if ((sess.isEmpty() && _allowShutdown != SHUT_ROOT) || QFile::exists("/run/pisilinux/livemedia") )
         return true;
     int ret = KDMConfShutdown(-1, sess, type, os).exec();
     if (ret == Schedule) {
@@ -699,7 +700,7 @@ void
 KDMSlimShutdown::externShutdown(int type, const QString &os, int uid, bool ask)
 {
     QList<DpySpec> sess = fetchSessions(lstRemote | lstTTY);
-    if (ask || !sess.isEmpty() || (uid && _allowShutdown == SHUT_ROOT)) {
+    if (!QFile::exists("/run/pisilinux/livemedia") && (ask || !sess.isEmpty() || (uid && _allowShutdown == SHUT_ROOT))) {
         int ret = KDMConfShutdown(uid, sess, type, os).exec();
         if (ret == Schedule) {
             KDMShutdown(uid).exec();
Index: kdebase-workspace-4.5.5/kdm/kfrontend/kgdialog.cpp
===================================================================
--- kdebase-workspace-4.5.5.orig/kdm/kfrontend/kgdialog.cpp
+++ kdebase-workspace-4.5.5/kdm/kfrontend/kgdialog.cpp
@@ -33,6 +33,7 @@ Foundation, Inc., 51 Franklin Street, Fi
 
 #include <QAction>
 #include <QMenu>
+#include <QFile>
 
 KGDialog::KGDialog(bool themed) : inherited(0, !themed)
 {
