diff -Naur kwin-orig/effects/dimscreen/dimscreen.desktop kwin/effects/dimscreen/dimscreen.desktop
--- kwin-orig/effects/dimscreen/dimscreen.desktop	2009-08-24 09:13:15.139137470 +0300
+++ kwin/effects/dimscreen/dimscreen.desktop	2009-08-24 09:13:35.330025508 +0300
@@ -126,6 +126,6 @@
 X-KDE-PluginInfo-Category=Focus
 X-KDE-PluginInfo-Depends=
 X-KDE-PluginInfo-License=GPL
-X-KDE-PluginInfo-EnabledByDefault=false
+X-KDE-PluginInfo-EnabledByDefault=true
 X-KDE-Library=kwin4_effect_builtins
 X-Ordering=50
