--- kde-workspace-4.8.1.orig/plasma/desktop/applets/kickoff/core/favoritesmodel.cpp	2012-03-03 00:11:07.000000000 +0200
+++ kde-workspace-4.8.1/plasma/desktop/applets/kickoff/core/favoritesmodel.cpp	2012-03-10 16:13:43.144449827 +0200
@@ -105,12 +105,12 @@
     static QList<QString> defaultFavorites()
     {
         QList<QString> applications;
-        applications << "konqbrowser" << "KMail2" << "systemsettings" << "dolphin";
+        applications << "mozillafirefox" << "dolphin" << "libreoffice-startcenter" << "clementine" << "systemsettings" << "package-manager";
 
         QList<QString> desktopFiles;
 
         foreach (const QString& application, applications) {
-            KService::Ptr service = KService::serviceByStorageId("kde4-" + application + ".desktop");
+            KService::Ptr service = KService::serviceByStorageId(application + ".desktop");
             if (service) {
                 desktopFiles << service->entryPath();
             }
