diff -Nuar a/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js b/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
--- a/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js	2013-06-28 20:10:44.508410537 +0300
+++ b/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js	2013-09-22 20:19:12.382851452 +0300
@@ -7,6 +7,7 @@
 panel.height = screenGeometry(panel.screen).height > 1024 ? 35 : 27
 panel.addWidget("launcher")
 panel.addWidget("org.kde.showActivityManager")
+panel.addWidget("showdesktop")
 pager = panel.addWidget("pager")
 pager.writeConfig("hideWhenSingleDesktop", "true")
 tasks = panel.addWidget("tasks")
