--- kde-workspace-4.8.1.orig/freespacenotifier/freespacenotifier.cpp	2012-03-03 00:11:07.000000000 +0200
+++ kde-workspace-4.8.1/freespacenotifier/freespacenotifier.cpp	2012-03-10 15:14:04.015672960 +0200
@@ -45,7 +45,10 @@
     FreeSpaceNotifierSettings::setEnableNotification( true );
 
     connect( &timer, SIGNAL(timeout()), SLOT(checkFreeDiskSpace()) );
-    timer.start( 1000 * 60 /* 1 minute */ );
+
+    //start timer if we're not working on a live media
+    if(!QFile::exists("/run/pisilinux/livemedia"))
+        timer.start( 1000 * 60 /* 1 minute */ );
 }
 
 FreeSpaceNotifier::~FreeSpaceNotifier()
