diff -up 0ad-0.0.15-alpha/source/network/NetServer.cpp.orig 0ad-0.0.15-alpha/source/network/NetServer.cpp
--- 0ad-0.0.15-alpha/source/network/NetServer.cpp.orig	2014-04-18 12:08:43.415641350 -0300
+++ 0ad-0.0.15-alpha/source/network/NetServer.cpp	2014-04-18 12:09:27.221643028 -0300
@@ -274,7 +274,7 @@ void* CNetServerWorker::SetupUPnP(void*)
 	// Check that the port was actually forwarded.
 	ret = UPNP_GetSpecificPortMappingEntry(urls.controlURL,
 									 data.first.servicetype,
-									 psPort, protocall,
+									 psPort, protocall, externalIPAddress,
 									 intClient, intPort, NULL/*desc*/,
 									 NULL/*enabled*/, duration);
 