--- 0ad-0.0.12-alpha-orig/build/resources/0ad.desktop	2010-10-08 17:08:54.000000000 +0300
+++ 0ad-0.0.12-alpha/build/resources/0ad.desktop	2013-02-06 06:08:07.899793417 +0200
@@ -1,10 +1,14 @@
 [Desktop Entry]
 Version=1.0
-Name=0 A.D.
+Encoding=UTF-8
+Name=0AD
+GenericName=0AD strategy game
+GenericName[tr]=0AD strateji oyunu
 Comment=A real-time strategy game of ancient warfare
 Comment[it]=Videogioco strategico in tempo reale di guerre antiche
-Exec=/usr/bin/0ad
-Icon=0ad
+Comment[tr]= Antik dönemler gerçek zamanlı strateji oyunu
+Exec=0ad
+Icon=0ad.png
 Terminal=false
 Type=Application
 Categories=Game;StrategyGame;
