diff -Nuar 0ad-r09786-alpha.orig//source/graphics/MapGenerator.h 0ad-r09786-alpha/source/graphics/MapGenerator.h
--- 0ad-r09786-alpha.orig//source/graphics/MapGenerator.h	2011-05-15 22:51:33.000000000 +0300
+++ 0ad-r09786-alpha/source/graphics/MapGenerator.h	2011-07-13 19:09:01.445000170 +0300
@@ -15,6 +15,8 @@
  * along with 0 A.D.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <boost/random/linear_congruential.hpp>
+
 #ifndef INCLUDED_MAPGENERATOR
 #define INCLUDED_MAPGENERATOR
 
@@ -22,8 +24,6 @@
 #include "ps/ThreadUtil.h"
 #include "scriptinterface/ScriptInterface.h"
 
-#include <boost/random/linear_congruential.hpp>
-
 #include <set>
 
 class CMapGeneratorWorker;
diff -Nuar 0ad-r09786-alpha.orig//source/graphics/MapReader.cpp 0ad-r09786-alpha/source/graphics/MapReader.cpp
--- 0ad-r09786-alpha.orig//source/graphics/MapReader.cpp	2011-06-27 00:45:58.000000000 +0300
+++ 0ad-r09786-alpha/source/graphics/MapReader.cpp	2011-07-13 19:09:01.445000170 +0300
@@ -15,6 +15,8 @@
  * along with 0 A.D.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "graphics/MapGenerator.h"
+
 #include "precompiled.h"
 
 #include "MapReader.h"
@@ -23,7 +25,6 @@
 #include "graphics/CinemaTrack.h"
 #include "graphics/Entity.h"
 #include "graphics/GameView.h"
-#include "graphics/MapGenerator.h"
 #include "graphics/Patch.h"
 #include "graphics/Terrain.h"
 #include "graphics/TerrainTextureEntry.h"
diff -Nuar 0ad-r09786-alpha.orig//source/graphics/MapReader.h 0ad-r09786-alpha/source/graphics/MapReader.h
--- 0ad-r09786-alpha.orig//source/graphics/MapReader.h	2011-06-14 02:32:41.000000000 +0300
+++ 0ad-r09786-alpha/source/graphics/MapReader.h	2011-07-13 19:09:01.453000163 +0300
@@ -15,6 +15,8 @@
  * along with 0 A.D.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "scriptinterface/ScriptInterface.h"
+
 #ifndef INCLUDED_MAPREADER
 #define INCLUDED_MAPREADER
 
@@ -23,7 +25,6 @@
 #include "ps/CStr.h"
 #include "LightEnv.h"
 #include "ps/FileIo.h"
-#include "scriptinterface/ScriptInterface.h"
 #include "simulation2/system/Entity.h"
 
 class CObjectEntry;
diff -Nuar 0ad-r09786-alpha.orig//source/scriptinterface/ScriptInterface.cpp 0ad-r09786-alpha/source/scriptinterface/ScriptInterface.cpp
--- 0ad-r09786-alpha.orig//source/scriptinterface/ScriptInterface.cpp	2011-06-28 02:27:25.000000000 +0300
+++ 0ad-r09786-alpha/source/scriptinterface/ScriptInterface.cpp	2011-07-13 19:09:01.455000169 +0300
@@ -15,6 +15,8 @@
  * along with 0 A.D.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <boost/random/linear_congruential.hpp>
+
 #include "precompiled.h"
 
 #include "ScriptInterface.h"
@@ -34,7 +36,6 @@
 #define BOOST_MULTI_INDEX_DISABLE_SERIALIZATION
 #include <boost/preprocessor/punctuation/comma_if.hpp>
 #include <boost/preprocessor/repetition/repeat.hpp>
-#include <boost/random/linear_congruential.hpp>
 #include <boost/random/uniform_real.hpp>
 #include <boost/flyweight.hpp>
 #include <boost/flyweight/key_value.hpp>
diff -Nuar 0ad-r09786-alpha.orig//source/scriptinterface/tests/test_ScriptInterface.h 0ad-r09786-alpha/source/scriptinterface/tests/test_ScriptInterface.h
--- 0ad-r09786-alpha.orig//source/scriptinterface/tests/test_ScriptInterface.h	2011-03-23 18:32:44.000000000 +0200
+++ 0ad-r09786-alpha/source/scriptinterface/tests/test_ScriptInterface.h	2011-07-13 19:09:01.458000169 +0300
@@ -15,13 +15,14 @@
  * along with 0 A.D.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <boost/random/linear_congruential.hpp>
+
 #include "lib/self_test.h"
 
 #include "scriptinterface/ScriptInterface.h"
 
 #include "ps/CLogger.h"
 
-#include <boost/random/linear_congruential.hpp>
 
 class TestScriptInterface : public CxxTest::TestSuite
 {
diff -Nuar 0ad-r09786-alpha.orig//source/simulation2/components/ICmpAIInterface.cpp 0ad-r09786-alpha/source/simulation2/components/ICmpAIInterface.cpp
--- 0ad-r09786-alpha.orig//source/simulation2/components/ICmpAIInterface.cpp	2011-01-12 14:29:00.000000000 +0200
+++ 0ad-r09786-alpha/source/simulation2/components/ICmpAIInterface.cpp	2011-07-13 19:09:01.459000168 +0300
@@ -15,12 +15,13 @@
  * along with 0 A.D.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "simulation2/scripting/ScriptComponent.h"
+
 #include "precompiled.h"
 
 #include "ICmpAIInterface.h"
 
 #include "simulation2/system/InterfaceScripted.h"
-#include "simulation2/scripting/ScriptComponent.h"
 
 BEGIN_INTERFACE_WRAPPER(AIInterface)
 END_INTERFACE_WRAPPER(AIInterface)
diff -Nuar 0ad-r09786-alpha.orig//source/simulation2/components/ICmpGuiInterface.cpp 0ad-r09786-alpha/source/simulation2/components/ICmpGuiInterface.cpp
--- 0ad-r09786-alpha.orig//source/simulation2/components/ICmpGuiInterface.cpp	2011-03-21 19:53:13.000000000 +0200
+++ 0ad-r09786-alpha/source/simulation2/components/ICmpGuiInterface.cpp	2011-07-13 19:09:01.460000168 +0300
@@ -15,12 +15,13 @@
  * along with 0 A.D.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "simulation2/scripting/ScriptComponent.h"
+
 #include "precompiled.h"
 
 #include "ICmpGuiInterface.h"
 
 #include "simulation2/system/InterfaceScripted.h"
-#include "simulation2/scripting/ScriptComponent.h"
 
 BEGIN_INTERFACE_WRAPPER(GuiInterface)
 END_INTERFACE_WRAPPER(GuiInterface)
diff -Nuar 0ad-r09786-alpha.orig//source/simulation2/components/ICmpIdentity.cpp 0ad-r09786-alpha/source/simulation2/components/ICmpIdentity.cpp
--- 0ad-r09786-alpha.orig//source/simulation2/components/ICmpIdentity.cpp	2011-05-05 23:07:51.000000000 +0300
+++ 0ad-r09786-alpha/source/simulation2/components/ICmpIdentity.cpp	2011-07-13 19:09:01.460000168 +0300
@@ -15,12 +15,13 @@
  * along with 0 A.D.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "simulation2/scripting/ScriptComponent.h"
+
 #include "precompiled.h"
 
 #include "ICmpIdentity.h"
 
 #include "simulation2/system/InterfaceScripted.h"
-#include "simulation2/scripting/ScriptComponent.h"
 
 
 BEGIN_INTERFACE_WRAPPER(Identity)
diff -Nuar 0ad-r09786-alpha.orig//source/simulation2/components/ICmpMotion.cpp 0ad-r09786-alpha/source/simulation2/components/ICmpMotion.cpp
--- 0ad-r09786-alpha.orig//source/simulation2/components/ICmpMotion.cpp	2010-01-09 21:20:14.000000000 +0200
+++ 0ad-r09786-alpha/source/simulation2/components/ICmpMotion.cpp	2011-07-13 19:09:01.460000168 +0300
@@ -15,12 +15,13 @@
  * along with 0 A.D.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "simulation2/scripting/ScriptComponent.h"
+
 #include "precompiled.h"
 
 #include "ICmpMotion.h"
 
 #include "simulation2/system/InterfaceScripted.h"
-#include "simulation2/scripting/ScriptComponent.h"
 
 BEGIN_INTERFACE_WRAPPER(Motion)
 END_INTERFACE_WRAPPER(Motion)
diff -Nuar 0ad-r09786-alpha.orig//source/simulation2/components/ICmpPlayer.cpp 0ad-r09786-alpha/source/simulation2/components/ICmpPlayer.cpp
--- 0ad-r09786-alpha.orig//source/simulation2/components/ICmpPlayer.cpp	2011-06-17 00:21:33.000000000 +0300
+++ 0ad-r09786-alpha/source/simulation2/components/ICmpPlayer.cpp	2011-07-13 19:09:01.461000168 +0300
@@ -15,13 +15,14 @@
  * along with 0 A.D.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "simulation2/scripting/ScriptComponent.h"
+
 #include "precompiled.h"
 
 #include "ICmpPlayer.h"
 
 #include "maths/FixedVector3D.h"
 #include "simulation2/system/InterfaceScripted.h"
-#include "simulation2/scripting/ScriptComponent.h"
 
 #include "ps/Overlay.h"
 
diff -Nuar 0ad-r09786-alpha.orig//source/simulation2/components/ICmpPlayerManager.cpp 0ad-r09786-alpha/source/simulation2/components/ICmpPlayerManager.cpp
--- 0ad-r09786-alpha.orig//source/simulation2/components/ICmpPlayerManager.cpp	2011-03-26 22:17:21.000000000 +0200
+++ 0ad-r09786-alpha/source/simulation2/components/ICmpPlayerManager.cpp	2011-07-13 19:09:01.461000168 +0300
@@ -15,12 +15,13 @@
  * along with 0 A.D.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "simulation2/scripting/ScriptComponent.h"
+
 #include "precompiled.h"
 
 #include "ICmpPlayerManager.h"
 
 #include "simulation2/system/InterfaceScripted.h"
-#include "simulation2/scripting/ScriptComponent.h"
 
 BEGIN_INTERFACE_WRAPPER(PlayerManager)
 END_INTERFACE_WRAPPER(PlayerManager)
diff -Nuar 0ad-r09786-alpha.orig//source/simulation2/components/ICmpUnitMotion.cpp 0ad-r09786-alpha/source/simulation2/components/ICmpUnitMotion.cpp
--- 0ad-r09786-alpha.orig//source/simulation2/components/ICmpUnitMotion.cpp	2011-06-09 22:44:40.000000000 +0300
+++ 0ad-r09786-alpha/source/simulation2/components/ICmpUnitMotion.cpp	2011-07-13 19:09:39.922000169 +0300
@@ -15,12 +15,13 @@
  * along with 0 A.D.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "simulation2/scripting/ScriptComponent.h"
+
 #include "precompiled.h"
 
 #include "ICmpUnitMotion.h"
 
 #include "simulation2/system/InterfaceScripted.h"
-#include "simulation2/scripting/ScriptComponent.h"
 
 BEGIN_INTERFACE_WRAPPER(UnitMotion)
 DEFINE_INTERFACE_METHOD_4("MoveToPointRange", bool, ICmpUnitMotion, MoveToPointRange, entity_pos_t, entity_pos_t, entity_pos_t, entity_pos_t)
diff -Nuar 0ad-r09786-alpha.orig//source/simulation2/components/ICmpUnknownScript.cpp 0ad-r09786-alpha/source/simulation2/components/ICmpUnknownScript.cpp
--- 0ad-r09786-alpha.orig//source/simulation2/components/ICmpUnknownScript.cpp	2010-07-01 19:58:44.000000000 +0300
+++ 0ad-r09786-alpha/source/simulation2/components/ICmpUnknownScript.cpp	2011-07-13 19:09:01.461000168 +0300
@@ -15,12 +15,13 @@
  * along with 0 A.D.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "simulation2/scripting/ScriptComponent.h"
+
 #include "precompiled.h"
 
 #include "ICmpUnknownScript.h"
 
 #include "simulation2/system/InterfaceScripted.h"
-#include "simulation2/scripting/ScriptComponent.h"
 
 BEGIN_INTERFACE_WRAPPER(UnknownScript)
 END_INTERFACE_WRAPPER(UnknownScript)
diff -Nuar 0ad-r09786-alpha.orig//source/simulation2/system/ComponentManager.h 0ad-r09786-alpha/source/simulation2/system/ComponentManager.h
--- 0ad-r09786-alpha.orig//source/simulation2/system/ComponentManager.h	2011-03-22 00:59:00.000000000 +0200
+++ 0ad-r09786-alpha/source/simulation2/system/ComponentManager.h	2011-07-13 19:09:01.462000168 +0300
@@ -15,6 +15,8 @@
  * along with 0 A.D.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <boost/random/linear_congruential.hpp>
+
 #ifndef INCLUDED_COMPONENTMANAGER
 #define INCLUDED_COMPONENTMANAGER
 
@@ -23,7 +25,6 @@
 #include "scriptinterface/ScriptInterface.h"
 #include "simulation2/helpers/Player.h"
 
-#include <boost/random/linear_congruential.hpp>
 #include <boost/unordered_map.hpp>
 
 #include <map>
