--- libgammu/gsmphones.c.orig	2011-12-13 13:33:39.000000000 +0400
+++ libgammu/gsmphones.c	2011-12-20 14:44:20.000000000 +0400
@@ -901,7 +901,7 @@
 	{"Serias E",	"Serias E",	"",				   {F_OBEX, F_IRMC_LEVEL_2, F_MODE22, F_PBK_UNICODE, 0}},
 	{"E815m",	"E815m",	"",				   {F_OBEX, F_IRMC_LEVEL_2, F_MODE22, F_PBK_UNICODE, 0}},
 	{"V635",	"V635",	"",					   {F_OBEX, F_IRMC_LEVEL_2, F_MODE22, F_PBK_UNICODE, 0}},
-	{"L7",		"L7", "",					   {F_NO_ATOBEX, F_CKPD_NO_UNICODE, F_USE_SMSTEXTMODE, 0}},
+	{"L7",		"L7", "",					   {F_NO_ATOBEX, F_CKPD_NO_UNICODE, 0}},
 	{"L7cm",	"L7cm", "",					   {F_NO_ATOBEX, F_CKPD_NO_UNICODE, F_USE_SMSTEXTMODE, 0}},
 	{"K1",		"K1", "",					   {F_NO_ATOBEX, F_PBK_UNICODE, F_USE_SMSTEXTMODE, 0}},
 	{"K1v",		"K1v", "",					   {F_NO_ATOBEX, F_PBK_UNICODE, 0}},
