commit 9b6f96a38b8c84cb87bb07165b503acbe5ea81ff
Author: Christoph Feck <christoph@maxiom.de>
Date:   Fri Mar 23 23:05:19 2012 +0100

    Fix includes

diff --git a/kmymoney/mymoney/storage/mymoneydatabasemgrtest.cpp b/kmymoney/mymoney/storage/mymoneydatabasemgrtest.cpp
index ff217e3..d31bb02 100644
--- a/kmymoney/mymoney/storage/mymoneydatabasemgrtest.cpp
+++ b/kmymoney/mymoney/storage/mymoneydatabasemgrtest.cpp
@@ -16,6 +16,7 @@
 
 #include "mymoneydatabasemgrtest.h"
 #include <pwd.h>
+#include <unistd.h>
 #include <iostream>
 
 #include <QtTest/QtTest>
