diff -Nur ipython-0.8.2-old/IPython/UserConfig/ipythonrc ipython-0.8.2/IPython/UserConfig/ipythonrc
--- ipython-0.8.2-old/IPython/UserConfig/ipythonrc	2008-04-16 10:00:21.000000000 +0300
+++ ipython-0.8.2/IPython/UserConfig/ipythonrc	2008-04-16 10:01:45.000000000 +0300
@@ -629,4 +629,8 @@
 
 # alias 
 
+# use cls to clear screen, as clear is used for history cleanup
+alias cls clear
+
+
 #************************* end of file <ipythonrc> ************************
