=== modified file 'IPython/iplib.py'
--- IPython/iplib.py	2008-09-20 09:34:38 +0000
+++ IPython/iplib.py	2009-02-24 11:45:58 +0000
@@ -2028,7 +2028,7 @@
         # this allows execution of indented pasted code. It is tempting
         # to add '\n' at the end of source to run commands like ' a=1'
         # directly, but this fails for more complicated scenarios
-        source=source.encode(self.stdin_encoding)
+        #source=source.encode(self.stdin_encoding)
         if source[:1] in [' ', '\t']:
             source = 'if 1:\n%s' % source
 

