diff -Nur enigma-1.01-old/src/Makefile.am enigma-1.01/src/Makefile.am
--- enigma-1.01-old/src/Makefile.am	2009-08-10 16:42:28.899429483 +0300
+++ enigma-1.01/src/Makefile.am	2009-08-10 16:42:55.816468389 +0300
@@ -10,7 +10,7 @@
 	enigma.doxygen
 
 
-CXXFLAGS = @CXXFLAGS@\
+AM_CXXFLAGS = @CXXFLAGS@\
 	-DSYSTEM_DATA_DIR=\"$(datadir)/enigma\" \
 	-DLOCALEDIR=\"/usr/share/locale\" \
 	-I$(top_srcdir)/lib-src/zipios++  \
