--- enigma-1.01/configure.ac
+++ enigma-1.01/configure.ac
@@ -260,7 +260,6 @@
 	CXXFLAGS="$CXXFLAGS -O2 -ffast-math -fomit-frame-pointer"
 	AC_MSG_RESULT(yes)
 else
-	CXXFLAGS="$CXXFLAGS -g"
 	AC_MSG_RESULT(no)
 fi
 
@@ -321,7 +320,6 @@
 dnl Configure libraries
 dnl ----------------------------------------------------------
 AC_CONFIG_SUBDIRS(lib-src/zipios++)
-AC_CONFIG_SUBDIRS(lib-src/enet)
 
 AC_CONFIG_FILES([Makefile m4/Makefile  intl/Makefile 
            data/Makefile
--- enigma-1.01/etc/Makefile.am
+++ enigma-1.01/etc/Makefile.am
@@ -1,14 +1,14 @@
 # if !MINGW32
 # Comply with xdg icon theme specification
-icondir = $(prefix)/share/icons/hicolor/48x48/apps
+icondir = /usr/share/icons/hicolor/48x48/apps
 icon_DATA = enigma.png
 
 # Backwards-compatible global search location for images
-pixmapdir = $(prefix)/share/pixmaps
+pixmapdir = /usr/share/pixmaps
 pixmap_DATA = enigma.png
 
 # Install .desktop file in xdg standard location
-desktopdir = $(prefix)/share/applications
+desktopdir = /usr/share/applications
 desktop_DATA = enigma.desktop
 # endif !MINGW32
 
--- enigma-1.01/lib-src/Makefile.am
+++ enigma-1.01/lib-src/Makefile.am
@@ -1 +1 @@
-SUBDIRS = oxydlib lua zipios++ enet enigma-core
+SUBDIRS = oxydlib lua zipios++ enigma-core
--- enigma-1.01/Makefile.am
+++ enigma-1.01/Makefile.am
@@ -3,11 +3,9 @@
 #   tools -> lib-src/lua
 #   src   -> tools, lib-src
 #
-SUBDIRS = m4 lib-src tools intl src data doc po etc
+SUBDIRS = m4 lib-src tools intl src data po etc
 EXTRA_DIST = enigma CHANGES ACKNOWLEDGEMENTS
 
-docdir = @datadir@/doc/enigma
-doc_DATA = README CHANGES COPYING ACKNOWLEDGEMENTS
 
 ACLOCAL_AMFLAGS = -I m4
 
--- enigma-1.01/po/Makefile.in.in
+++ enigma-1.01/po/Makefile.in.in
@@ -24,12 +24,12 @@
 prefix = @prefix@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
-localedir = $(datadir)/locale
+localedir = /usr/share/locale
 gettextsrcdir = $(datadir)/gettext/po
 
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
-MKINSTALLDIRS = @MKINSTALLDIRS@ 
+MKINSTALLDIRS = $(top_srcdir)/mkinstalldirs
 #install_sh@ -d
 mkinstalldirs = $(SHELL) $(MKINSTALLDIRS)
 
--- enigma-1.01/src/Makefile.am
+++ enigma-1.01/src/Makefile.am
@@ -12,12 +12,11 @@
 
 CXXFLAGS = @CXXFLAGS@\
 	-DSYSTEM_DATA_DIR=\"$(datadir)/enigma\" \
-	-DLOCALEDIR=\"$(localedir)\" \
+	-DLOCALEDIR=\"/usr/share/locale\" \
 	-I$(top_srcdir)/lib-src/zipios++  \
 	-I$(top_builddir)/lib-src/zipios++ \
 	-I$(top_srcdir)/lib-src/lua \
 	-I$(top_srcdir)/lib-src/enigma-core \
-	-I$(top_srcdir)/lib-src/enet/include \
 	-I$(top_srcdir)/lib-src \
 	-I$(top_builddir)/intl @SDL_CFLAGS@
 
@@ -193,7 +192,7 @@
 
 else # not MINGW32
 
-  localedir = $(datadir)/locale
+  localedir = /usr/share/locale
   mingw_ldadd =
 
 endif
@@ -202,7 +201,7 @@
 	@LIBINTL@ \
 	$(top_builddir)/lib-src/enigma-core/libecl.a \
 	$(top_builddir)/lib-src/lua/liblua.a \
-	$(top_builddir)/lib-src/enet/libenet.a \
+	-lenet \
 	$(top_builddir)/lib-src/oxydlib/liboxyd.a \
 	$(top_builddir)/lib-src/zipios++/src/libzipios.a \
 	-lz \
