--- lib-src/enigma-core/ecl_util.hh
+++ lib-src/enigma-core/ecl_util.hh
@@ -20,6 +20,7 @@
 #define ECL_UTIL_HH_INCLUDED
 
 #include <string>
+#include <algorithm>
 
 /* hide GNU extensions for non-gnu compilers: */
 #ifndef __GNU__
--- src/Utf8ToXML.cc
+++ src/Utf8ToXML.cc
@@ -21,6 +21,7 @@
 #include "main.hh"
 #include <xercesc/util/XMLString.hpp>
 #include <xercesc/util/TransService.hpp>
+#include <cstring>
 
 XERCES_CPP_NAMESPACE_USE
 
--- src/file.hh
+++ src/file.hh
@@ -22,6 +22,7 @@
 #include <iosfwd>
 #include <vector>
 #include <list>
+#include <memory>
 #include "ecl_error.hh"
 
 namespace enigma
--- src/gui/widgets.cc
+++ src/gui/widgets.cc
@@ -28,6 +28,7 @@
 #include <cassert>
 #include <algorithm>
 #include <iostream>
+#include <climits>
 
 using namespace enigma::gui;
 using namespace ecl;
