diff -Nuar kmix-4.14.0.orig/gui/kmixdockwidget.cpp kmix-4.14.0/gui/kmixdockwidget.cpp
--- kmix-4.14.0.orig/gui/kmixdockwidget.cpp	2014-07-16 11:24:17.000000000 +0300
+++ kmix-4.14.0/gui/kmixdockwidget.cpp	2014-08-29 00:40:18.689065814 +0300
@@ -30,12 +30,15 @@
 #include <kwindowsystem.h>
 #include <kactioncollection.h>
 #include <ktoggleaction.h>
+#include <kprocess.h>
+#include <kmessagebox.h>
 
 #include <QDBusConnection>
 #include <QDBusConnectionInterface>
 #include <QDesktopWidget>
 #include <QApplication>
 #include <QTextDocument>
+#include <QFile>
 
 #include "apps/kmix.h"
 #include "core/ControlManager.h"
@@ -80,7 +83,7 @@
 		QString(), // All mixers (as the Global master Mixer might change)
 		(ControlChangeType::Type) (ControlChangeType::Volume | ControlChangeType::MasterChanged), this,
 		QString("KMixDockWidget"));
-	 
+
 	      // Refresh in all cases. When there is no Golbal Master we still need
      // to initialize correctly (e.g. for showin 0% or hiding it)
      refreshVolumeLevels();
@@ -145,6 +148,16 @@
         menu->addAction( action );
     }
 
+  if (QFile::exists("/usr/bin/pavucontrol"))
+    {
+    // Add pavucontrol shortcut to the context menu
+    QAction *paction = actionCollection()->addAction( "run_pavucontrol" );
+    paction->setText( i18n ("Advanced Sound Control") );
+    connect(paction, SIGNAL(triggered(bool) ), SLOT (runPavucontrol() ));
+    QAction *a1 = actionCollection()->action ("run_pavucontrol" );
+    if ( a1 ) menu->addAction( a1 );
+    }
+
     // Put "Select Master Channel" dialog in context menu
     QAction *action = actionCollection()->addAction( "select_master" );
     action->setText( i18n("Select Master Channel...") );
@@ -156,6 +169,18 @@
     menu->addAction(_kmixMainWindow->actionCollection()->action("launch_kdesoundsetup"));
 }
 
+void KMixDockWidget::runPavucontrol()
+{
+    KProcess *pulse_proc = new KProcess();
+    pulse_proc->setProgram("/usr/bin/pavucontrol");
+
+    if ( pulse_proc->startDetached() == 0 ) {
+        // Error
+        KMessageBox::error(0, i18n("Failed launching pavucontrol, make sure the pavucontrol package is correctly installed."),
+                              i18n("Error launching pavucontrol"));
+    }
+}
+
 void
 KMixDockWidget::setVolumeTip()
 {
@@ -218,7 +243,7 @@
       // Pixmap must be changed => do so
       switch ( newPixmapType ) {
          case 'e': setIconByName( "kmixdocked_error" ); break;
-         case 'm': 
+         case 'm':
          case '0': setIconByName( "audio-volume-muted"  ); break;
          case '1': setIconByName( "audio-volume-low"  ); break;
          case '2': setIconByName( "audio-volume-medium" ); break;
@@ -373,7 +398,7 @@
 }
 
 void KMixDockWidget::updateDockMuteAction ( KToggleAction* dockMuteAction )
-{  
+{
     shared_ptr<MixDevice> md = Mixer::getGlobalMasterMD();
     if ( md && dockMuteAction != 0 )
     {
diff -Nuar kmix-4.14.0.orig/gui/kmixdockwidget.h kmix-4.14.0/gui/kmixdockwidget.h
--- kmix-4.14.0.orig/gui/kmixdockwidget.h	2014-07-16 11:24:17.000000000 +0300
+++ kmix-4.14.0/gui/kmixdockwidget.h	2014-08-29 00:40:51.837066702 +0300
@@ -74,6 +74,7 @@
  private slots:
    void dockMute();
    void trayWheelEvent(int delta,Qt::Orientation);
+   void runPavucontrol();
    void contextMenuAboutToShow();
 };
 
