--- python-bugzilla-0.8.0.orig/bin/bugzilla	2013-02-14 23:06:21.000000000 +0200
+++ python-bugzilla-0.8.0/bin/bugzilla	2013-11-05 19:51:22.605206157 +0200
@@ -24,7 +24,7 @@
 
 import bugzilla
 
-default_bz = 'https://bugzilla.redhat.com/xmlrpc.cgi'
+default_bz = 'https://bugzilla.pardus.org.tr/xmlrpc.cgi'
 
 _is_unittest = bool(os.getenv("__BUGZILLA_UNITTEST"))
 cmdlist = ['login', 'new', 'query', 'modify', 'attach', 'info']
