From dec007d6154e98c96b9bb74e5305ea9c7967db64 Mon Sep 17 00:00:00 2001
From: Maarten Lankhorst <maarten.lankhorst@canonical.com>
Date: Sun, 9 Jun 2013 11:20:28 +0200
Subject: [PATCH 04/38] avrt: Add realtime to stub

---
 dlls/avrt/main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlls/avrt/main.c b/dlls/avrt/main.c
index 5c9ef4a..43e8bd8 100644
--- a/dlls/avrt/main.c
+++ b/dlls/avrt/main.c
@@ -80,6 +80,7 @@ HANDLE WINAPI AvSetMmThreadCharacteristicsW(LPCWSTR TaskName, LPDWORD TaskIndex)
         SetLastError(ERROR_INVALID_HANDLE);
         return NULL;
     }
+    SetThreadPriority(GetCurrentThread(), THREAD_PRIORITY_TIME_CRITICAL);
     return (HANDLE)0x12345678;
 }
 
-- 
1.8.3

