From 73bf265e32532c3522fe40153ec5fcf7ec66e9f2 Mon Sep 17 00:00:00 2001
From: Maarten Lankhorst <maarten.lankhorst@canonical.com>
Date: Sun, 9 Jun 2013 11:20:28 +0200
Subject: [PATCH 05/38] server: Bump priority on server to process messages
 faster

---
 server/main.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/server/main.c b/server/main.c
index 7aed338..6c257b4 100644
--- a/server/main.c
+++ b/server/main.c
@@ -28,6 +28,7 @@
 #include <stdlib.h>
 #include <sys/time.h>
 #include <unistd.h>
+#include <sys/syscall.h>
 #ifdef HAVE_GETOPT_H
 # include <getopt.h>
 #endif
@@ -38,6 +39,8 @@
 #include "request.h"
 #include "wine/library.h"
 
+extern int rtkit_make_realtime(pid_t process, pid_t thread, int priority);
+
 /* command-line options */
 int debug_level = 0;
 int foreground = 0;
@@ -145,6 +148,7 @@ int main( int argc, char *argv[] )
     init_signals();
     init_directories();
     init_registry();
+    rtkit_make_realtime(getpid(), syscall( SYS_gettid ), 2);
     main_loop();
     return 0;
 }
-- 
1.8.3

