diff -Nuar kitchensync-0.22.0.orig//CMakeLists.txt kitchensync-0.22.0//CMakeLists.txt
--- kitchensync-0.22.0.orig//CMakeLists.txt	2011-02-09 15:12:54.458000020 +0200
+++ kitchensync-0.22.0//CMakeLists.txt	2011-02-09 15:13:46.545000025 +0200
@@ -18,7 +18,8 @@
 find_package(Opensync REQUIRED)
 
 include_directories( ${CMAKE_CURRENT_SOURCE_DIR} ${QT_QTCORE_INCLUDE_DIR}
-                     ${OPENSYNC_INCLUDE_DIRS} ${OSENGINE_INCLUDE_DIRS} )
+                     ${OPENSYNC_INCLUDE_DIRS} ${OSENGINE_INCLUDE_DIRS} 
+                     /usr/include/kde4 )
 
 ADD_SUBDIRECTORY( libqopensync ) 
 ADD_SUBDIRECTORY( src ) 
