diff -urN kitchensync-0.22.0/CMakeLists.txt kitchensync-0.22.0-patched/CMakeLists.txt
--- kitchensync-0.22.0/CMakeLists.txt	2010-09-22 03:38:55.000000000 +1100
+++ kitchensync-0.22.0-patched/CMakeLists.txt	2012-03-07 22:22:51.133427836 +1100
@@ -17,7 +17,7 @@
 
 find_package(Opensync REQUIRED)
 
-include_directories( ${CMAKE_CURRENT_SOURCE_DIR} ${QT_QTCORE_INCLUDE_DIR}
+include_directories( ${CMAKE_CURRENT_SOURCE_DIR} ${QT_QTCORE_INCLUDE_DIR} ${QT_INCLUDE_DIR}
                      ${OPENSYNC_INCLUDE_DIRS} ${OSENGINE_INCLUDE_DIRS} )
 
 ADD_SUBDIRECTORY( libqopensync ) 
