Index: resume.c
===================================================================
--- resume.c.orig
+++ resume.c
@@ -421,10 +421,10 @@ int main(int argc, char *argv[])
 	if (error)
 		return -error;
 
-	if (splash_param != 'y' && splash_param != 'Y')
-		splash_param = 0;
-	else
+	if (splash_param != 'n' && splash_param != 'N')
 		splash_param = SPL_RESUME;
+	else
+		splash_param = 0;
 
 	get_page_and_buffer_sizes();
 
Index: suspend.c
===================================================================
--- suspend.c.orig
+++ suspend.c
@@ -1782,10 +1782,10 @@ int main(int argc, char *argv[])
 	if (do_encrypt != 'y' && do_encrypt != 'Y')
 		do_encrypt = 0;
 #endif
-	if (splash_param != 'y' && splash_param != 'Y')
-		splash_param = 0;
-	else
+	if (splash_param != 'n' && splash_param != 'N')
 		splash_param = SPL_SUSPEND;
+	else
+		splash_param = 0;
 
 	if (early_writeout != 'n' && early_writeout != 'N')
 		early_writeout = 1;
Index: conf/suspend.conf
===================================================================
--- conf/suspend.conf.orig
+++ conf/suspend.conf
@@ -40,7 +40,7 @@
 ## will most of the time speed up overall writing time (default y)
 #early writeout = n
 #
-## use splash picture? (default n)
+## use splash picture? (default y)
 #splash = y
 #
 ## shutdown method:
