diff -Naur transfig.3.2.5/fig2dev/bound.c transfig.3.2.5-patched/fig2dev/bound.c
--- transfig.3.2.5/fig2dev/bound.c	2005-11-24 00:49:31.000000000 +0100
+++ transfig.3.2.5-patched/fig2dev/bound.c	2008-08-23 13:33:15.000000000 +0200
@@ -72,7 +72,6 @@
 		   { 5, 1, 0, True, True, False, 1.5, {{-0.75,0.5},{0,0},{-0.75,-0.5},{-1.0,0},{-0.75,0.5}}},
 		   /* type 3b filled convex spearhead */
 		   { 5, 1, 0, True, True, False, 1.5, {{-0.75,0.5},{0,0},{-0.75,-0.5},{-1.0,0},{-0.75,0.5}}},
-#ifdef NEWARROWTYPES
 		   /* type 4a diamond */
 		   { 5, 1, 0, True, True, False, 1.15, {{-0.5,0.5},{0,0},{-0.5,-0.5},{-1.0,0},{-0.5,0.5}}},
 		   /* type 4b filled diamond */
@@ -120,7 +119,6 @@
 		   { 4, 0, 0, True, True, False, -1.0, {{0,0.5},{-1.0,0.5},{-1.0,-0.5},{0,-0.5}}},
 		   /* type 14b backward two-prong fork */
 		   { 4, 1, 0, True, True, False, 0.0, {{-1.0,0.5,},{0,0.5},{0,-0.5},{-1.0,-0.5}}},
-#endif /* NEWARROWTYPES */
 		};
 
 void
diff -Naur transfig.3.2.5/transfig/sys.c transfig.3.2.5-patched/transfig/sys.c
--- transfig.3.2.5/transfig/sys.c	2005-10-27 20:23:42.000000000 +0200
+++ transfig.3.2.5-patched/transfig/sys.c	2008-08-23 13:38:15.614628350 +0200
@@ -15,6 +15,8 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include "transfig.h"
 
 #define MAXSYS 10000
diff -Naur transfig.3.2.5/transfig/transfig.c transfig.3.2.5-patched/transfig/transfig.c
--- transfig.3.2.5/transfig/transfig.c	2003-04-15 20:31:10.000000000 +0200
+++ transfig.3.2.5-patched/transfig/transfig.c	2008-08-23 13:37:18.706627399 +0200
@@ -25,6 +25,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include "patchlevel.h"
 #include "transfig.h"
 
diff -Naur transfig.3.2.5/transfig/txfile.c transfig.3.2.5-patched/transfig/txfile.c
--- transfig.3.2.5/transfig/txfile.c	2003-10-31 19:26:27.000000000 +0100
+++ transfig.3.2.5-patched/transfig/txfile.c	2008-08-23 13:37:33.530628914 +0200
@@ -15,6 +15,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include "transfig.h"
 
 /*
