--- bash-2.05b/lib/readline/readline.c.oom	2002-03-13 23:10:46.000000000 +0100
+++ bash-2.05b/lib/readline/readline.c	2002-08-07 12:02:04.000000000 +0200
@@ -567,7 +567,7 @@
 	{
 	  /* Special case rl_do_lowercase_version (). */
 	  if (func == rl_do_lowercase_version)
-	    return (_rl_dispatch (_rl_to_lower (key), map));
+	    return (_rl_dispatch (_rl_to_lower ((unsigned char)key), map));
 
 	  rl_executing_keymap = map;
 
