--- bash-3.1/Makefile.in.xcc	2005-10-25 19:37:52.000000000 +0100
+++ bash-3.1/Makefile.in	2005-12-23 16:11:09.000000000 +0000
@@ -68,6 +68,7 @@
 ARFLAGS = @ARFLAGS@
 RANLIB = @RANLIB@
 SIZE = @SIZE@
+STRIP = strip
 
 INSTALL = @INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
@@ -535,7 +536,7 @@
 	@chmod a+rx bashbug
 
 strip:	$(Program) .made
-	strip $(Program)
+	$(STRIP) $(Program)
 	ls -l $(Program)
 	-$(SIZE) $(Program)
 
