diff -up bash-4.2-rc2/tests/exec.right.tty_tests bash-4.2-rc2/tests/exec.right
--- bash-4.2-rc2/tests/exec.right.tty_tests	2011-02-09 10:42:48.000000000 +0100
+++ bash-4.2-rc2/tests/exec.right	2011-02-09 10:42:59.000000000 +0100
@@ -50,7 +50,6 @@ this is ohio-state
 0
 1
 testb
-expand_aliases 	on
 1
 1
 1
diff -up bash-4.2-rc2/tests/execscript.tty_tests bash-4.2-rc2/tests/execscript
--- bash-4.2-rc2/tests/execscript.tty_tests	2010-12-27 22:01:02.000000000 +0100
+++ bash-4.2-rc2/tests/execscript	2011-02-09 10:42:34.000000000 +0100
@@ -107,8 +107,6 @@ ${THIS_SH} ./exec6.sub
 # checks for properly deciding what constitutes an executable file
 ${THIS_SH} ./exec7.sub
 
-${THIS_SH} -i ./exec8.sub
-
 ${THIS_SH} ./exec9.sub
 
 true | `echo true` &
diff -up bash-4.2-rc2/tests/read.right.tty_tests bash-4.2-rc2/tests/read.right
--- bash-4.2-rc2/tests/read.right.tty_tests	2010-12-21 16:49:00.000000000 +0100
+++ bash-4.2-rc2/tests/read.right	2011-02-09 10:42:34.000000000 +0100
@@ -33,14 +33,6 @@ a = abcdefg
 a = xyz
 a = -xyz 123-
 a = abc
-timeout 1: ok
-
-timeout 2: ok
-
-./read2.sub: line 23: read: -3: invalid timeout specification
-1
-
-abcde
 ./read3.sub: line 4: read: -1: invalid number
 abc
 ab
diff -up bash-4.2-rc2/tests/read.tests.tty_tests bash-4.2-rc2/tests/read.tests
--- bash-4.2-rc2/tests/read.tests.tty_tests	2008-09-06 19:09:11.000000000 +0200
+++ bash-4.2-rc2/tests/read.tests	2011-02-09 10:42:34.000000000 +0100
@@ -82,9 +82,6 @@ echo " foo" | { IFS=$':' ; read line; re
 # test read -d delim behavior
 ${THIS_SH} ./read1.sub
 
-# test read -t timeout behavior
-${THIS_SH} ./read2.sub
-
 # test read -n nchars behavior
 ${THIS_SH} ./read3.sub
 
