--- bash-2.05b/aclocal.m4.pgrp_sync	2002-06-25 14:45:43.000000000 +0100
+++ bash-2.05b/aclocal.m4	2003-01-15 18:17:35.000000000 +0000
@@ -1255,7 +1255,7 @@
 	wait(&status);
 	exit(ok ? 0 : 5);
 }
-], bash_cv_pgrp_pipe=no,bash_cv_pgrp_pipe=yes,
+], bash_cv_pgrp_pipe=yes,bash_cv_pgrp_pipe=yes,
    [AC_MSG_WARN(cannot check pgrp synchronization if cross compiling -- defaulting to no)
     bash_cv_pgrp_pipe=no])
 ])
