Index: cellwriter-1.3.4_20110312/src/window.c
===================================================================
--- cellwriter-1.3.4_20110312.orig/src/window.c
+++ cellwriter-1.3.4_20110312/src/window.c
@@ -70,7 +70,7 @@ void key_widget_resize(KeyWidget *key_wi
 GtkWidget *window;
 GtkTooltips *tooltips;
 int window_force_x = -1, window_force_y = -1, training_block = 0,
-    window_docked = WINDOW_UNDOCKED, window_force_docked = -1,
+    window_docked = WINDOW_DOCKED_BOTTOM, window_force_docked = -1,
     window_button_labels = TRUE, window_force_show = FALSE,
     window_force_hide = FALSE, style_colors = TRUE, window_embedded = FALSE,
     window_struts = FALSE;
