Index: cellwriter-1.3.4_20110312/src/window.c
===================================================================
--- cellwriter-1.3.4_20110312.orig/src/window.c
+++ cellwriter-1.3.4_20110312/src/window.c
@@ -331,7 +331,6 @@ void window_set_docked(int mode)
            FIXME This allegedly solves docking problems with some window
                  managers but only seems to cause more problems for
                  Compiz and Metacity. */
-        /*
         gtk_widget_hide(window);
         if (mode == WINDOW_UNDOCKED)
                 gtk_window_set_type_hint(GTK_WINDOW(window),
@@ -340,7 +339,6 @@ void window_set_docked(int mode)
                 gtk_window_set_type_hint(GTK_WINDOW(window),
                                          GDK_WINDOW_TYPE_HINT_DOCK);
         gtk_widget_show(window);
-        */
 
         /* Restore the old window position */
         if (!mode) {
