Index: lockdev-1.0.3/src/sample.c
===================================================================
--- lockdev-1.0.3.orig/src/sample.c
+++ lockdev-1.0.3/src/sample.c
@@ -1,5 +1,7 @@
 #include <stdio.h>
 #include <stdlib.h>
+#include <sys/types.h>
+#include <unistd.h>
 #include "lockdev.h"
 
 void
@@ -14,7 +16,7 @@ int
 main (int   argc,
       char *argv[])
 {
-	int i, chld;
+	int i;
 	char *p = NULL, *dev = NULL, ch;
 
 	ch = '\0';
