--- lockdev-1.0.0/src/baudboy.h	Sat Apr 20 15:42:16 2002
+++ lockdev-1.0.0-/src/baudboy.h	Sat Apr 20 15:41:49 2002
@@ -34,11 +34,11 @@
 
 #define	LOCKDEV_PATH	"/usr/sbin/lockdev"
 
-static inline int doit(const char * argv[])
+static inline int _doit(const char * argv[])
 {
     pid_t child;
     int status;
-    void (*osig) (int) = signal(SIGCHLD, SIG_IGN);
+    void (*osig) (int) = signal(SIGCHLD, SIG_DFL);
     int rc;
 
     if (!(child = fork())) {
@@ -104,21 +104,21 @@
 {
     const char * argv[] = { LOCKDEV_PATH, "-l", NULL, NULL};
     argv[2] = devname;
-    return doit(argv);
+    return _doit(argv);
 }
 
 static inline int ttyunlock(const char * devname)
 {
     const char * argv[] = { LOCKDEV_PATH, "-u", NULL, NULL};
     argv[2] = devname;
-    return doit(argv);
+    return _doit(argv);
 }
 
 static inline int ttylocked(const char * devname)
 {
     const char * argv[] = { LOCKDEV_PATH, NULL, NULL};
     argv[1] = devname;
-    return doit(argv);
+    return _doit(argv);
 }
 
 static inline int ttywait(const char * devname)
