--- volpack-1.0b3.orig/src/volpack.h
+++ volpack-1.0b3/src/volpack.h
@@ -348,7 +348,9 @@
  * Global variables.                                               *
  *******************************************************************/
 
-EXTERN char *vpCompilerOptions;
+/* Seems not to be used any more
+   EXTERN char *vpCompilerOptions;
+ */
 
 /*******************************************************************
  * Function declarations.                                          *

