diff -up ffDiaporama/src/ffDiaporama/engine/cDeviceModelDef.h.ffmpeg-2_4 ffDiaporama/src/ffDiaporama/engine/cDeviceModelDef.h
--- ffDiaporama/src/ffDiaporama/engine/cDeviceModelDef.h.ffmpeg-2_4	2014-04-12 06:41:33.000000000 +0200
+++ ffDiaporama/src/ffDiaporama/engine/cDeviceModelDef.h	2014-11-15 01:31:09.021441450 +0100
@@ -84,7 +84,7 @@ extern "C" {
 #elif (LIBAVUTIL_VERSION_MICRO>=100)&&(LIBAVCODEC_VERSION_MICRO>=100)&&(LIBAVFORMAT_VERSION_MICRO>=100)&&(LIBAVDEVICE_VERSION_MICRO>=100)&&(LIBAVFILTER_VERSION_MICRO>=100)&&(LIBSWSCALE_VERSION_MICRO>=100)
     #define FFMPEG
     #include "libswresample/swresample.h"
-    #define RESAMPLE_MAX_CHANNELS SWR_CH_MAX
+    #define RESAMPLE_MAX_CHANNELS 32
     #include "libavfilter/avcodec.h"
     #include "libavfilter/buffersink.h"
     #include "libavfilter/buffersrc.h"