--- openoffice.org.orig/desktop/scripts/unopkg.sh	2008-01-14 15:55:26.000000000 +0000
+++ openoffice.org/desktop/scripts/unopkg.sh	2008-02-14 10:52:10.000000000 +0000
@@ -62,6 +62,33 @@
     ;;
 esac
 
+isnotuser=0
+for arg in $@
+do
+if [ "$arg" = "--shared" -o "$arg" = "--bundled" ]; then 
+    isnotuser=1
+fi
+done
+if [ $isnotuser -eq 1 ]; then
+    echo $@ | grep -q env:JFW_PLUGIN_DO_NOT_CHECK_ACCESSIBILITY
+    if [ $? -ne 0 ]; then
+        set -- $@ '-env:JFW_PLUGIN_DO_NOT_CHECK_ACCESSIBILITY=1'
+    fi
+    echo $@ | grep -q env:UNO_JAVA_JFW_INSTALL_DATA
+    if [ $? -ne 0 -a -w $sd_prog/../share/config/javasettingsunopkginstall.xml ]; then
+        set -- $@ '-env:UNO_JAVA_JFW_INSTALL_DATA=$$ORIGIN/../share/config/javasettingsunopkginstall.xml'
+    fi
+    echo $@ | grep -q env:UserInstallation
+    if [ $? -ne 0 ]; then
+        INSTDIR=`/bin/mktemp -d --tmpdir unoinstall.XXXXXX`
+        if [ $? -ne 0 ]; then
+            echo "Could not create tmp dir" >&2
+            exit 1
+        fi
+        set -- $@ '-env:UserInstallation=file://'$INSTDIR
+    fi
+fi
+
 #collect all bootstrap variables specified on the command line
 #so that they can be passed as arguments to javaldx later on
 for arg in $@
@@ -110,6 +137,8 @@
 # SAL_NO_XINITTHREADS=true; export SAL_NO_XINITTHREADS
 
 # execute binary
-exec "$sd_prog/unopkg.bin" "$@" \
+"$sd_prog/unopkg.bin" "$@" \
     "-env:INIFILENAME=vnd.sun.star.pathname:$sd_prog/redirectrc"
-
+if [ -n "$INSTDIR" ]; then
+   rm -rf $INSTDIR
+fi
