Index: bin/modules/installer/scriptitems.pm
===================================================================
RCS file: /cvs/tools/solenv/bin/modules/installer/scriptitems.pm,v
retrieving revision 1.17
diff -u -p -r1.17 scriptitems.pm
--- openoffice.org.orig/solenv/bin/modules/installer/scriptitems.pm	24 Feb 2005 16:21:15 -0000	1.17
+++ openoffice.org/solenv/bin/modules/installer/scriptitems.pm	18 Mar 2005 22:39:42 -0000
@@ -1356,11 +1356,10 @@

             if ( ! $installer::globals::languagepack && !$installer::globals::helppack)
             {
-                $infoline = "ERROR: Removing file $filename from file list.\n";
+                $infoline = "WARNING: Removing file $filename from file list.\n";
                 push( @installer::globals::logfileinfo, $infoline);
 
-                push(@missingfiles, "ERROR: File not found: $filename\n");
-                $error_occurred = 1;
+                push(@missingfiles, "WARNING: File not found: $filename\n");
 
                 next;   # removing this file from list, if sourcepath is empty
             }
@@ -1368,11 +1367,10 @@
             {
                 if (( $onefile->{'ismultilingual'} ) || ( $styles =~ /\bFORCELANGUAGEPACK\b/ ))
                 {
-                    $infoline = "ERROR: Removing file $filename from file list.\n";
+                    $infoline = "WARNING: Removing file $filename from file list.\n";
                     push( @installer::globals::logfileinfo, $infoline);
 
-                    push(@missingfiles, "ERROR: File not found: $filename\n");
-                    $error_occurred = 1;
+                    push(@missingfiles, "WARNING: File not found: $filename\n");
 
                     next;   # removing this file from list, if sourcepath is empty
                 }
@@ -1390,11 +1388,10 @@
             {
                 if (( $onefile->{'ismultilingual'} ) || ( $styles =~ /\bFORCEHELPPACK\b/ ))
                 {
-                    $infoline = "ERROR: Removing file $filename from file list.\n";
+                    $infoline = "WARNING: Removing file $filename from file list.\n";
                     push( @installer::globals::logfileinfo, $infoline);
 
-                    push(@missingfiles, "ERROR: File not found: $filename\n");
-                    $error_occurred = 1;
+                    push(@missingfiles, "WARNING: File not found: $filename\n");
 
                     next;   # removing this file from list, if sourcepath is empty
                 }
--- openoffice.org.orig/solenv/bin/modules/installer/simplepackage.pm	2010-07-12 10:27:26.000000000 +0100
+++ openoffice.org/solenv/bin/modules/installer/simplepackage.pm	2010-07-12 10:27:54.000000000 +0100
@@ -53,7 +53,7 @@
         ( $installer::globals::packageformat eq "archive" ))
     {
         $installer::globals::is_simple_packager_project = 1;
-        $installer::globals::patch_user_dir = 1;
+        $installer::globals::patch_user_dir = 1;
     }
     elsif( $installer::globals::packageformat eq "dmg" )
     {
