Index: bluefish-2.0.2/src/plugin_about/po/Makefile.in.in
===================================================================
--- bluefish-2.0.2.orig/src/plugin_about/po/Makefile.in.in
+++ bluefish-2.0.2/src/plugin_about/po/Makefile.in.in
@@ -9,7 +9,7 @@
 # General Public License and is *not* in the public domain.
 #
 # Origin: gettext-0.17
-GETTEXT_MACRO_VERSION = 0.17
+GETTEXT_MACRO_VERSION = 0.18
 
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
Index: bluefish-2.0.2/src/plugin_charmap/po/Makefile.in.in
===================================================================
--- bluefish-2.0.2.orig/src/plugin_charmap/po/Makefile.in.in
+++ bluefish-2.0.2/src/plugin_charmap/po/Makefile.in.in
@@ -9,7 +9,7 @@
 # General Public License and is *not* in the public domain.
 #
 # Origin: gettext-0.17
-GETTEXT_MACRO_VERSION = 0.17
+GETTEXT_MACRO_VERSION = 0.18
 
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
Index: bluefish-2.0.2/src/plugin_entities/po/Makefile.in.in
===================================================================
--- bluefish-2.0.2.orig/src/plugin_entities/po/Makefile.in.in
+++ bluefish-2.0.2/src/plugin_entities/po/Makefile.in.in
@@ -9,7 +9,7 @@
 # General Public License and is *not* in the public domain.
 #
 # Origin: gettext-0.17
-GETTEXT_MACRO_VERSION = 0.17
+GETTEXT_MACRO_VERSION = 0.18
 
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
Index: bluefish-2.0.2/src/plugin_htmlbar/po/Makefile.in.in
===================================================================
--- bluefish-2.0.2.orig/src/plugin_htmlbar/po/Makefile.in.in
+++ bluefish-2.0.2/src/plugin_htmlbar/po/Makefile.in.in
@@ -9,7 +9,7 @@
 # General Public License and is *not* in the public domain.
 #
 # Origin: gettext-0.17
-GETTEXT_MACRO_VERSION = 0.17
+GETTEXT_MACRO_VERSION = 0.18
 
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
Index: bluefish-2.0.2/src/plugin_infbrowser/po/Makefile.in.in
===================================================================
--- bluefish-2.0.2.orig/src/plugin_infbrowser/po/Makefile.in.in
+++ bluefish-2.0.2/src/plugin_infbrowser/po/Makefile.in.in
@@ -9,7 +9,7 @@
 # General Public License and is *not* in the public domain.
 #
 # Origin: gettext-0.17
-GETTEXT_MACRO_VERSION = 0.17
+GETTEXT_MACRO_VERSION = 0.18
 
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
Index: bluefish-2.0.2/src/plugin_snippets/po/Makefile.in.in
===================================================================
--- bluefish-2.0.2.orig/src/plugin_snippets/po/Makefile.in.in
+++ bluefish-2.0.2/src/plugin_snippets/po/Makefile.in.in
@@ -9,7 +9,7 @@
 # General Public License and is *not* in the public domain.
 #
 # Origin: gettext-0.17
-GETTEXT_MACRO_VERSION = 0.17
+GETTEXT_MACRO_VERSION = 0.18
 
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
Index: bluefish-2.0.2/src/plugin_zencoding/po/Makefile.in.in
===================================================================
--- bluefish-2.0.2.orig/src/plugin_zencoding/po/Makefile.in.in
+++ bluefish-2.0.2/src/plugin_zencodingo/Makefile.in.in
@@ -9,7 +9,7 @@
 # General Public License and is *not* in the public domain.
 #
 # Origin: gettext-0.17
-GETTEXT_MACRO_VERSION = 0.17
+GETTEXT_MACRO_VERSION = 0.18
 
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
