From d88b91cd4fa9f93b91824b00a61b64ecac752bb1 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Mon, 10 Dec 2007 11:26:57 -0500
Subject: [PATCH 01/17] Build libxf86config with -fPIC.

---
 hw/xfree86/parser/Makefile.am |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/hw/xfree86/parser/Makefile.am b/hw/xfree86/parser/Makefile.am
index caf7079..655e73f 100644
--- a/hw/xfree86/parser/Makefile.am
+++ b/hw/xfree86/parser/Makefile.am
@@ -32,7 +32,7 @@ libxf86config_internal_la_SOURCES = \
 
 libxf86config_a_SOURCES = \
 	$(INTERNAL_SOURCES)
-libxf86config_a_CFLAGS = $(AM_CFLAGS)
+libxf86config_a_CFLAGS = $(AM_CFLAGS) -fPIC -fvisibility=hidden
 
 AM_CFLAGS = $(DIX_CFLAGS) $(XORG_CFLAGS) \
 	-DSYSCONFDIR=\"$(sysconfdir)\" \
-- 
1.7.1

