From 6f9b0496f919178720a7f99049d8ab9ff16d3575 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Mon, 13 Sep 2010 11:34:28 -0400
Subject: [PATCH] vbe: Refuse to believe tiny (or negative) panel sizes from PanelID

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
 hw/xfree86/vbe/vbe.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/hw/xfree86/vbe/vbe.c b/hw/xfree86/vbe/vbe.c
index 0b34648..1defd9a 100644
--- a/hw/xfree86/vbe/vbe.c
+++ b/hw/xfree86/vbe/vbe.c
@@ -1088,6 +1088,11 @@ VBEInterpretPanelID(int scrnIndex, struct vbePanelID *data)
     if (pScrn->monitor->nHsync || pScrn->monitor->nVrefresh)
 	return;
 
+    if (data->hsize < 320 || data->vsize < 240) {
+	xf86DrvMsg(scrnIndex, X_INFO, "... which I refuse to believe\n");
+	return;
+    }
+
     mode = xf86CVTMode(data->hsize, data->vsize, PANEL_HZ, 1, 0);
 
     pScrn->monitor->nHsync = 1;
-- 
1.7.2.2

