From e3824df6768a3d254d02de65b514cdf561838bb8 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Tue, 7 Oct 2008 11:09:14 -0400
Subject: [PATCH 05/17] Force ModeDebug on.

---
 hw/xfree86/modes/xf86Crtc.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/hw/xfree86/modes/xf86Crtc.c b/hw/xfree86/modes/xf86Crtc.c
index 85d499a..883bf3b 100644
--- a/hw/xfree86/modes/xf86Crtc.c
+++ b/hw/xfree86/modes/xf86Crtc.c
@@ -2363,8 +2363,7 @@ xf86InitialConfiguration (ScrnInfoPtr scrn, Bool canGrow)
     xf86ProcessOptions (scrn->scrnIndex,
 			scrn->options,
 			config->options);
-    config->debug_modes = xf86ReturnOptValBool (config->options,
-						OPTION_MODEDEBUG, FALSE);
+    config->debug_modes = TRUE;
 
     if (scrn->display->virtualX)
 	width = scrn->display->virtualX;
-- 
1.7.1

