diff --git a/configure.ac b/configure.ac
index df0ade5..f360a31 100644
--- a/configure.ac
+++ b/configure.ac
@@ -101,6 +101,8 @@ XDT_CHECK_PACKAGE([DBUS_GLIB], [dbus-glib-1], [dbus_glib_minimum_version])
 XDT_CHECK_PACKAGE([LIBXFCE4GUI], [libxfcegui4-1.0],[libxfcegui4_minimum_version])
 XDT_CHECK_PACKAGE([LIBXFCE4UTIL], [libxfce4util-1.0],[libxfce4util_minimum_version])
 
+XDT_CHECK_LIBX11()
+
 #=======================================================#
 #           Check for availability of libnotify         #
 #=======================================================#
diff --git a/src/Makefile.am b/src/Makefile.am
index b2928a8..795e6ac 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -20,7 +20,8 @@ INCLUDES =					\
         $(LIBXFCE4GUI_CFLAGS)                   \
 	$(GST_VIDEO_CFLAGS)			\
 	$(GST_INTERFACES_CFLAGS)		\
-	$(TAGLIBC_CFLAGS)
+	$(TAGLIBC_CFLAGS)			\
+	$(LIBX11_CFLAGS)
 
 GENERATED_FILES =				\
 	gmarshal.c				\
@@ -85,7 +86,8 @@ parole_LDADD =					\
         $(LIBXFCE4GUI_LIBS)                     \
 	$(GST_VIDEO_LIBS)			\
 	$(GST_INTERFACES_LIBS)			\
-	$(TAGLIBC_LIBS)
+	$(TAGLIBC_LIBS)				\
+	$(LIBX11_LIBS)
 	
 parole_glib_enum_headers =			\
 	parole-conf.h				\
