Index: linux-2.6.37/drivers/mmc/host/rts_pstor/define.debug
===================================================================
--- linux-2.6.37.orig/drivers/mmc/host/rts_pstor/define.debug
+++ /dev/null
@@ -1,30 +0,0 @@
-/* Driver for Realtek PCI-Express card reader
- * Header file
- *
- * Copyright(c) 2009 Realtek Semiconductor Corp. All rights reserved.  
- *
- * This program is free software; you can redistribute it and/or modify it
- * under the terms of the GNU General Public License as published by the
- * Free Software Foundation; either version 2, or (at your option) any
- * later version.
- *
- * This program is distributed in the hope that it will be useful, but
- * WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- * General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License along
- * with this program; if not, see <http://www.gnu.org/licenses/>.
- *
- * Author:
- *   wwang (wei_wang@realsil.com.cn)
- *   No. 450, Shenhu Road, Suzhou Industry Park, Suzhou, China
- */
-
-#ifndef __REALTEK_RTSX_DEFINE_H
-#define __REALTEK_RTSX_DEFINE_H
-
-#define DBG	1
-
-#endif  // __REALTEK_RTSX_DEFINE_H
-
Index: linux-2.6.37/drivers/mmc/host/rts_pstor/define.release
===================================================================
--- linux-2.6.37.orig/drivers/mmc/host/rts_pstor/define.release
+++ /dev/null
@@ -1,30 +0,0 @@
-/* Driver for Realtek PCI-Express card reader
- * Header file
- *
- * Copyright(c) 2009 Realtek Semiconductor Corp. All rights reserved.  
- *
- * This program is free software; you can redistribute it and/or modify it
- * under the terms of the GNU General Public License as published by the
- * Free Software Foundation; either version 2, or (at your option) any
- * later version.
- *
- * This program is distributed in the hope that it will be useful, but
- * WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- * General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License along
- * with this program; if not, see <http://www.gnu.org/licenses/>.
- *
- * Author:
- *   wwang (wei_wang@realsil.com.cn)
- *   No. 450, Shenhu Road, Suzhou Industry Park, Suzhou, China
- */
-
-#ifndef __REALTEK_RTSX_DEFINE_H
-#define __REALTEK_RTSX_DEFINE_H
-
-#define DBG	0
-
-#endif  // __REALTEK_RTSX_DEFINE_H
-
Index: linux-2.6.37/drivers/mmc/host/rts_pstor/Makefile
===================================================================
--- linux-2.6.37.orig/drivers/mmc/host/rts_pstor/Makefile
+++ linux-2.6.37/drivers/mmc/host/rts_pstor/Makefile
@@ -1,48 +1,4 @@
-# Driver for Realtek PCI-Express card reader
-#
-# Copyright(c) 2009 Realtek Semiconductor Corp. All rights reserved.  
-#
-# This program is free software; you can redistribute it and/or modify it
-# under the terms of the GNU General Public License as published by the
-# Free Software Foundation; either version 2, or (at your option) any
-# later version.
-#
-# This program is distributed in the hope that it will be useful, but
-# WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-# General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License along
-# with this program; if not, see <http://www.gnu.org/licenses/>.
-#
-# Author:
-#   wwang (wei_wang@realsil.com.cn)
-#   No. 450, Shenhu Road, Suzhou Industry Park, Suzhou, China
-#
-# Makefile for the PCI-Express Card Reader drivers.
-#
-
-TARGET_MODULE := rts_pstor
-
-EXTRA_CFLAGS := -Idrivers/scsi 
-
-obj-m += $(TARGET_MODULE).o
-
-$(TARGET_MODULE)-objs := rtsx.o rtsx_chip.o rtsx_transport.o rtsx_scsi.o rtsx_card.o \
-			 general.o sd.o xd.o ms.o spi.o
-
-default:
-	sed "s/RTSX_MK_TIME/`date +%y.%m.%d.%H.%M`/" timestamp.in > timestamp.h
-	cp -f ./define.release ./define.h
-	make -C /lib/modules/$(shell uname -r)/build/ SUBDIRS=$(PWD) modules
-debug:
-	sed "s/RTSX_MK_TIME/`date +%y.%m.%d.%H.%M`/" timestamp.in > timestamp.h
-	cp -f ./define.debug ./define.h
-	make -C /lib/modules/$(shell uname -r)/build/ SUBDIRS=$(PWD) modules
-install:
-	cp $(TARGET_MODULE).ko /lib/modules/$(shell uname -r)/kernel/drivers/scsi -f
-clean:
-	rm -f *.o *.ko
-	rm -f $(TARGET_MODULE).mod.c
-
+rts_pstor-objs 	= rtsx.o rtsx_chip.o rtsx_transport.o rtsx_scsi.o rtsx_card.o \
+					general.o sd.o xd.o ms.o spi.o
 
+obj-$(CONFIG_RTS_PSTOR) 	+= rts_pstor.o
Index: linux-2.6.37/drivers/mmc/host/rts_pstor/README.txt
===================================================================
--- linux-2.6.37.orig/drivers/mmc/host/rts_pstor/README.txt
+++ /dev/null
@@ -1,15 +0,0 @@
-General Information
-===================
-
-Linux driver for Realtek PCI-Express card reader chip.
-
-
-Build Steps
-===========
-
-1) make
-2) make install
-3) depmod
-4) reboot your computer
-
-Note: Root privilege is required in step 2 and 3
\ Dosya sonunda yenisatır yok.
Index: linux-2.6.37/drivers/mmc/host/rts_pstor/rtsx.c
===================================================================
--- linux-2.6.37.orig/drivers/mmc/host/rts_pstor/rtsx.c
+++ linux-2.6.37/drivers/mmc/host/rts_pstor/rtsx.c
@@ -36,8 +36,6 @@
 #include "sd.h"
 #include "xd.h"
 
-#include "timestamp.h"
-
 #define DRIVER_VERSION 		"v1.10"
 
 MODULE_DESCRIPTION("Realtek PCI-Express card reader driver");
@@ -154,7 +152,6 @@ static int proc_info (struct Scsi_Host *
 	SPRINTF("       Vendor: Realtek Corp.\n");
 	SPRINTF("      Product: PCIE Card Reader\n");
 	SPRINTF("      Version: %s\n", DRIVER_VERSION);
-	SPRINTF("        Build: %s\n", DRIVER_MAKE_TIME);
 
 	/*
 	 * Calculate start of next buffer, and return value.
@@ -920,8 +917,6 @@ static int __devinit rtsx_probe(struct p
 	int err = 0;
 	struct task_struct *th;
 
-	printk(KERN_INFO "--- %s ---\n", DRIVER_MAKE_TIME);
-
 	err = pci_enable_device(pci);
 	if (err < 0) {
 		printk(KERN_ERR "PCI enable device failed!\n");
Index: linux-2.6.37/drivers/mmc/host/rts_pstor/timestamp.in
===================================================================
--- linux-2.6.37.orig/drivers/mmc/host/rts_pstor/timestamp.in
+++ /dev/null
@@ -1,30 +0,0 @@
-/* Driver for Realtek PCI-Express card reader
- * Header file
- *
- * Copyright(c) 2009 Realtek Semiconductor Corp. All rights reserved.  
- *
- * This program is free software; you can redistribute it and/or modify it
- * under the terms of the GNU General Public License as published by the
- * Free Software Foundation; either version 2, or (at your option) any
- * later version.
- *
- * This program is distributed in the hope that it will be useful, but
- * WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- * General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License along
- * with this program; if not, see <http://www.gnu.org/licenses/>.
- *
- * Author:
- *   wwang (wei_wang@realsil.com.cn)
- *   No. 450, Shenhu Road, Suzhou Industry Park, Suzhou, China
- */
-
-#ifndef __RTSX_VER_H
-#define __RTSX_VER_H
-
-#define DRIVER_MAKE_TIME	"RTSX_MK_TIME"
-
-#endif // __RTSX_VER_H
-
Index: linux-2.6.37/drivers/mmc/host/Kconfig
===================================================================
--- linux-2.6.37.orig/drivers/mmc/host/Kconfig
+++ linux-2.6.37/drivers/mmc/host/Kconfig
@@ -488,3 +488,5 @@ config MMC_USHC
 
 	  Note: These controllers only support SDIO cards and do not
 	  support MMC or SD memory cards.
+
+source "drivers/mmc/host/rts_pstor/Kconfig"
Index: linux-2.6.37/drivers/mmc/host/Makefile
===================================================================
--- linux-2.6.37.orig/drivers/mmc/host/Makefile
+++ linux-2.6.37/drivers/mmc/host/Makefile
@@ -48,3 +48,4 @@ sdhci-of-$(CONFIG_MMC_SDHCI_OF_HLWD)	+=
 ifeq ($(CONFIG_CB710_DEBUG),y)
 	CFLAGS-cb710-mmc	+= -DDEBUG
 endif
+obj-$(CONFIG_RTS_PSTOR)		+= rts_pstor/
Index: linux-2.6.37/drivers/mmc/host/rts_pstor/Kconfig
===================================================================
--- /dev/null
+++ linux-2.6.37/drivers/mmc/host/rts_pstor/Kconfig
@@ -0,0 +1,5 @@
+config RTS_PSTOR
+	tristate "Realtek PCIE Card Reader support"
+	help
+	  This selects the Realtek PCIE Card Reader driver found in
+	  HP Compaq CQ-10 netbooks.
Index: linux-2.6.37/drivers/mmc/host/rts_pstor/define.h
===================================================================
--- /dev/null
+++ linux-2.6.37/drivers/mmc/host/rts_pstor/define.h
@@ -0,0 +1,30 @@
+/* Driver for Realtek PCI-Express card reader
+ * Header file
+ *
+ * Copyright(c) 2009 Realtek Semiconductor Corp. All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2, or (at your option) any
+ * later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author:
+ *   wwang (wei_wang@realsil.com.cn)
+ *   No. 450, Shenhu Road, Suzhou Industry Park, Suzhou, China
+ */
+
+#ifndef __REALTEK_RTSX_DEFINE_H
+#define __REALTEK_RTSX_DEFINE_H
+
+#define DBG 0
+
+#endif  // __REALTEK_RTSX_DEFINE_H
+
