--- linux-3.8-orig/arch/x86/boot/video.c	2013-02-19 01:58:34.000000000 +0200
+++ linux-3.8/arch/x86/boot/video.c	2013-03-02 20:31:02.611424949 +0200
@@ -315,6 +315,7 @@
 void set_video(void)
 {
 	u16 mode = boot_params.hdr.vid_mode;
+	int vesa_failed = 0;
 
 	RESET_HEAP();
 
@@ -329,8 +330,15 @@
 		if (!set_mode(mode))
 			break;
 
-		printf("Undefined video mode number: %x\n", mode);
-		mode = ASK_VGA;
+		if (!vesa_failed) {
+			/* fallback is vesa 640x480 16bit */
+			printf("Could not set mode: %x, trying 640x480-16\n", mode);
+			mode = 0x311;
+			vesa_failed = 1;
+		} else {
+			printf("Undefined video mode number: %x, falling back to text mode\n", mode);
+			mode = VIDEO_80x25;
+		}
 	}
 	boot_params.hdr.vid_mode = mode;
 	vesa_store_edid();
