diff -Naur a/arch/x86/boot/video.c b/arch/x86/boot/video.c
--- a/arch/x86/boot/video.c	2013-07-01 01:13:29.000000000 +0300
+++ b/arch/x86/boot/video.c	2013-07-02 17:16:42.682684939 +0300
@@ -329,8 +329,15 @@
 		if (!set_mode(mode))
 			break;
 
-		printf("Undefined video mode number: %x\n", mode);
-		mode = ASK_VGA;
+		if (!vesa_failed) {
+			/* fallback is vesa 640x480 16bit */
+			printf("Could not set mode: %x, trying 640x480-16\n", mode);
+			mode = 0x311;
+			vesa_failed = 1;
+		} else {
+			printf("Undefined video mode number: %x, falling back to text mode\n", mode);
+			mode = VIDEO_80x25;
+		}
 	}
 	boot_params.hdr.vid_mode = mode;
 	vesa_store_edid();
