Index: xserver-xorg-input-synaptics/src/synaptics.c
===================================================================
--- xserver-xorg-input-synaptics.orig/src/synaptics.c	2011-01-25 21:55:50.000000000 +1100
+++ xserver-xorg-input-synaptics/src/synaptics.c	2011-01-25 22:06:10.496826002 +1100
@@ -703,6 +703,9 @@
 	return BadAlloc;
     }
 
+    if (xf86Screens[0]->vtSema == FALSE)
+	return !Success;
+
     /* may change pInfo->options */
     SetDeviceAndProtocol(pInfo);
