Index: xserver-xorg-input-synaptics/src/synaptics.c
===================================================================
--- xserver-xorg-input-synaptics.orig/src/synaptics.c	2009-01-26 09:52:14.000000000 +0200
+++ xserver-xorg-input-synaptics/src/synaptics.c	2009-01-26 09:52:27.000000000 +0200
@@ -403,10 +403,10 @@
 	palmMinWidth = 10;
     }
 
-    /* Enable tap if we don't have a phys left button */
-    tapButton1 = priv->has_left ? 0 : 1;
-    tapButton2 = priv->has_left ? 0 : 3;
-    tapButton3 = priv->has_left ? 0 : 2;
+    /* Enable tap */
+    tapButton1 = 1;
+    tapButton2 = 2;
+    tapButton3 = 3;
 
     /* Enable multifinger-click if we don't have right/middle button,
        otherwise clickFinger is always button 1. */
