Index: xserver-xorg-input-synaptics/src/synaptics.c
===================================================================
--- xserver-xorg-input-synaptics.orig/src/synaptics.c	2009-01-26 09:52:44.000000000 +0200
+++ xserver-xorg-input-synaptics/src/synaptics.c	2009-01-26 09:52:48.000000000 +0200
@@ -413,8 +413,8 @@
     clickFinger2 = 2;
     clickFinger3 = 3;
 
-    /* Enable vert edge scroll if we can't detect doubletap */
-    vertEdgeScroll = priv->has_double ? FALSE : TRUE;
+    /* Enable vert edge scroll */
+    vertEdgeScroll = TRUE;
     horizEdgeScroll = FALSE;
 
     /* Enable twofinger scroll if we can detect doubletap */
