--- gnuchess-5.07/src/common.h	2009-09-10 18:01:34.000000000 +0200
+++ gnuchess-5.07_getline/src/common.h	2009-09-10 17:57:40.000000000 +0200
@@ -745,7 +745,7 @@
  * Input routine, initialized to one of the specific
  * input routines. The given argument is the prompt.
  */
-void (*getline) (char *);
+void (*getline_chess) (char *);
 
 #define MAXSTR 128
 extern char inputstr[MAXSTR];
--- gnuchess-5.07/src/input.c	2003-07-01 19:06:43.000000000 +0200
+++ gnuchess-5.07_getline/src/input.c	2009-09-10 18:00:53.000000000 +0200
@@ -127,7 +127,7 @@
 	      (RealGameCnt+1)/2 + 1 );
     }
     pthread_mutex_lock(&input_mutex);
-    getline(prompt);
+    getline_chess(prompt);
     input_status = INPUT_AVAILABLE;
     pthread_cond_signal(&input_cond);
     pthread_mutex_unlock(&input_mutex);
@@ -173,13 +173,13 @@
 {
 #ifdef HAVE_LIBREADLINE
   if (isatty(STDIN_FILENO)) {
-    getline = getline_readline;
+    getline_chess = getline_readline;
     using_history();
   } else {
-    getline = getline_standard;
+    getline_chess = getline_standard;
   }
 #else
-  getline = getline_standard;
+  getline_chess = getline_standard;
 #endif
   /* Start input thread */
   pthread_create(&input_thread, NULL, input_func, NULL);
