diff -Nur gnuchess-5.07-old/src/input.c gnuchess-5.07/src/input.c
--- gnuchess-5.07-old/src/input.c	2008-12-28 21:04:24.000000000 +0200
+++ gnuchess-5.07/src/input.c	2008-12-28 21:10:29.000000000 +0200
@@ -81,7 +81,26 @@
 }
 #endif /* HAVE_LIBREADLINE */
 
-/* The generic input routine */
+void rinput (char str[], int n, FILE *stream) {
+
+  /* Jens Hamacher's input function, a little fix by Onur Küçük - read in up to n-1
+     characters from stream, or until we encounter a \n or an EOF. Appends a null
+     character at the end of the string, and stores the string in str[] */
+
+  int ch, i = 0;
+
+  while ((ch = getc (stream)) != (int) '\n' && ch != EOF) {
+    if (i < n-1) {
+      str[i++] = ch;
+    }
+  }
+
+  if (ch == EOF) {
+    cmd_quit();
+  }
+
+  str [i] = '\0';
+}
 
 void getline_standard(char *p)
 {
@@ -89,7 +108,7 @@
     fputs(p, stdout);
     fflush(stdout);
   }
-  fgets(inputstr, MAXSTR, stdin);
+  rinput(inputstr, MAXSTR, stdin);
 }
 
 static pthread_t input_thread;
