diff -p -up ./src/libcmis-c/Makefile.am.0000 ./src/libcmis-c/Makefile.am
--- ./src/libcmis-c/Makefile.am.0000	2012-09-12 15:52:37.000000000 +0200
+++ ./src/libcmis-c/Makefile.am	2012-12-17 02:25:08.071356584 +0100
@@ -38,6 +38,7 @@ libcmis_c_@LIBCMIS_API_VERSION@_la_LDFLA
 libcmis_c_@LIBCMIS_API_VERSION@_la_LIBADD = \
 	$(XML2_LIBS) \
 	$(CURL_LIBS) \
+	$(top_srcdir)/src/libcmis/libcmis-@LIBCMIS_API_VERSION@.la \
 	$(BOOST_SMART_PTR_LIBS) \
 	$(BOOST_DATE_TIME_LIBS)
 
diff -p -up ./src/libcmis-c/Makefile.in.0000 ./src/libcmis-c/Makefile.in
--- ./src/libcmis-c/Makefile.in.0000	2012-12-17 02:24:47.471783337 +0100
+++ ./src/libcmis-c/Makefile.in	2012-12-17 02:24:56.419467339 +0100
@@ -364,6 +364,7 @@ libcmis_c_@LIBCMIS_API_VERSION@_la_LDFLA
 libcmis_c_@LIBCMIS_API_VERSION@_la_LIBADD = \
 	$(XML2_LIBS) \
 	$(CURL_LIBS) \
+	$(top_srcdir)/src/libcmis/libcmis-@LIBCMIS_API_VERSION@.la \
 	$(BOOST_SMART_PTR_LIBS) \
 	$(BOOST_DATE_TIME_LIBS)
 
