diff -Nur nvidia-settings-260.19.12-old//src/gtk+-2.x/ctkdisplayconfig-utils.c nvidia-settings-260.19.12/src/gtk+-2.x/ctkdisplayconfig-utils.c
--- nvidia-settings-260.19.12-old//src/gtk+-2.x/ctkdisplayconfig-utils.c	2010-10-19 15:18:45.690001322 +0300
+++ nvidia-settings-260.19.12/src/gtk+-2.x/ctkdisplayconfig-utils.c	2010-10-19 15:20:00.147000214 +0300
@@ -3267,7 +3267,7 @@
     if (tmp_filename) {
         filename = g_strdup(tmp_filename);
     } else {
-        filename = g_strdup("");
+        filename = g_strdup("/etc/X11/xorg.conf");
     }
     xconfigCloseConfigFile();
 
