diff -Nur nvidia-settings-260.19.12-old//src/XF86Config-parser/Generate.c nvidia-settings-260.19.12/src/XF86Config-parser/Generate.c
--- nvidia-settings-260.19.12-old//src/XF86Config-parser/Generate.c	2010-10-22 11:37:31.615000042 +0300
+++ nvidia-settings-260.19.12/src/XF86Config-parser/Generate.c	2010-10-22 11:37:59.976000043 +0300
@@ -71,9 +71,9 @@
 
     /* add files, fonts, and modules */
 
-    config->files = xconfigAlloc(sizeof(XConfigFilesRec));
-    add_font_path(gop, config);
-    add_modules(gop, config);
+    config->files = NULL;
+    /*add_font_path(gop, config);
+    add_modules(gop, config); */
 
     /* add the keyboard and mouse */
 
@@ -82,6 +82,8 @@
 
     /* add the layout */
 
+    config->inputs = NULL;
+
     add_layout(gop, config);
 
     return config;
@@ -1042,7 +1044,7 @@
     xconfigAddNewOption(&input->options, "ZAxisMapping", "4 5");
 
     input->next = config->inputs;
-    config->inputs = input;
+    config->inputs = NULL;
 
     return TRUE;
 
@@ -1303,7 +1305,7 @@
     }
 
     input->next = config->inputs;
-    config->inputs = input;
+    config->inputs = NULL;
 
     return TRUE;
 
diff -Nur nvidia-settings-260.19.12-old//src/XF86Config-parser/Generate.c nvidia-settings-260.19.12/src/XF86Config-parser/Generate.c
--- nvidia-settings-260.19.12-old//src/XF86Config-parser/Generate.c	2010-10-21 11:11:06.251000043 +0300
+++ nvidia-settings-260.19.12/src/XF86Config-parser/Generate.c	2010-10-21 11:32:59.793000043 +0300
@@ -571,6 +571,12 @@
 {
     XConfigInputrefPtr inputRef;
 
+    XConfigInputPtr input;
+    input = xconfigFindInput(name, config->inputs);
+
+    if(!input)
+        return;
+
     inputRef = xconfigAlloc(sizeof(XConfigInputrefRec));
     inputRef->input_name = xconfigStrdup(name);
     inputRef->input = xconfigFindInput(inputRef->input_name, config->inputs);
