--- src/k3b.desktop	2011-01-15 21:47:30.000000000 +0100
+++ src/k3b.desktop	2011-01-16 12:08:05.698000050 +0100
@@ -137,7 +137,7 @@
 Name[x-test]=xxK3bxx
 Name[zh_CN]=K3b
 Name[zh_TW]=K3b
-MimeType=application/x-k3b;application/x-iso;application/x-cd-image;application/x-cue;
+MimeType=application/x-k3b;application/x-iso;application/x-cd-image;application/x-cue;x-content/blank-cd;x-content/blank-dvd;x-content/blank-bd;x-content/blank-hddvd;
 X-KDE-StartupNotify=true
 X-KDE-NativeMimeType=application/x-k3b;
 Categories=KDE;Qt;AudioVideo;DiscBurning;System;Filesystem;
