diff -up wicd-1.7.0/in/man=wicd-curses.8.in.orig wicd-1.7.0/in/man=wicd-curses.8.in
--- wicd-1.7.0/in/man=wicd-curses.8.in.orig	2010-01-14 18:49:11.000000000 -1000
+++ wicd-1.7.0/in/man=wicd-curses.8.in	2010-05-19 09:20:00.000000000 -1000
@@ -57,10 +57,6 @@ Bring up instructions on how to edit the
 .BR O
 Raise the Ad-Hoc network creation dialog
 
-.SH "FILES"
-.TP
-.I ~/.wicd/WHEREAREMYFILES
-Reminder that your network configuration files are not here ;-)
 .PP
 .SH "SEE ALSO"
 .BR wicd-client (1),
diff -up wicd-1.7.0/in/scripts=wicd-client.in.orig wicd-1.7.0/in/scripts=wicd-client.in
--- wicd-1.7.0/in/scripts=wicd-client.in.orig	2010-01-14 18:49:11.000000000 -1000
+++ wicd-1.7.0/in/scripts=wicd-client.in	2010-05-19 09:20:25.000000000 -1000
@@ -6,10 +6,6 @@ NC=$(tput sgr0)
 if [ ! -d "$HOME/.wicd" ]; then
     mkdir -p "$HOME/.wicd"
 fi
-# Make sure the user knows WHEREAREMYFILES ;-)
-if [ -e "%VARLIB%WHEREAREMYFILES" ] && [ ! -L "$HOME/.wicd/WHEREAREMYFILES" ]; then
-    ln -s "%VARLIB%WHEREAREMYFILES" "$HOME/.wicd/WHEREAREMYFILES"
-fi
 if [ "$DISPLAY" = "" ]; then
     if [ -x "%BIN%wicd-curses" ]; then
         if [ ! -f "$HOME/.wicd/CLIENT_CURSES_WARNING" ]; then
diff -up wicd-1.7.0/in/scripts=wicd-curses.in.orig wicd-1.7.0/in/scripts=wicd-curses.in
--- wicd-1.7.0/in/scripts=wicd-curses.in.orig	2010-01-14 18:49:11.000000000 -1000
+++ wicd-1.7.0/in/scripts=wicd-curses.in	2010-05-19 09:20:36.000000000 -1000
@@ -2,8 +2,5 @@
 if [ ! -d "$HOME/.wicd" ]; then
     mkdir -p "$HOME/.wicd"
 fi
-if [ -e "%VARLIB%WHEREAREMYFILES" ] && [ ! -L "$HOME/.wicd/WHEREAREMYFILES" ]; then
-    ln -s "%VARLIB%WHEREAREMYFILES" "$HOME/.wicd/WHEREAREMYFILES"
-fi
 
 exec %PYTHON% -O %SHARE%curses/wicd-curses.py $@
diff -up wicd-1.7.0/in/scripts=wicd-gtk.in.orig wicd-1.7.0/in/scripts=wicd-gtk.in
--- wicd-1.7.0/in/scripts=wicd-gtk.in.orig	2010-01-14 18:49:11.000000000 -1000
+++ wicd-1.7.0/in/scripts=wicd-gtk.in	2010-05-19 09:20:44.000000000 -1000
@@ -4,9 +4,5 @@
 if [ ! -d "$HOME/.wicd" ]; then
     mkdir -p "$HOME/.wicd"
 fi
-# Make sure the user knows WHEREAREMYFILES ;-)
-if [ -e "/var/lib/wicd/WHEREAREMYFILES" ] && [ ! -L "$HOME/.wicd/WHEREAREMYFILES" ]; then
-    ln -s "/var/lib/wicd/WHEREAREMYFILES" "$HOME/.wicd/WHEREAREMYFILES"
-fi
 
 exec %PYTHON% -O %SHARE%gtk/wicd-client.py $@
