Index: librecad-1.0.0_20111012/desktop/librecad.desktop
===================================================================
--- librecad-1.0.0_20111012.orig/desktop/librecad.desktop
+++ librecad-1.0.0_20111012/desktop/librecad.desktop
@@ -2,10 +2,13 @@
 Version=1.0
 Type=Application
 Name=LibreCAD
+GenericName=A professional CAD System
+GenericName[tr]=Profesyonel bir CAD Sistemi
 Comment=A professional CAD System
 Comment[es]=Un sistema CAD profesional
+Comment[tr]=Profesyonel bir CAD Sistemi
 TryExec=librecad
 Exec=librecad %F
 Icon=librecad
 Categories=Graphics;Engineering;
-MimeType=image/vnd.dxf
+MimeType=image/vnd.dxf;
