diff -U 3 -H -d -r -N -- a/src/lib/engine/rs_ellipse.cpp b/src/lib/engine/rs_ellipse.cpp
--- a/src/lib/engine/rs_ellipse.cpp	2011-08-14 07:28:00.000000000 +0200
+++ b/src/lib/engine/rs_ellipse.cpp	2011-08-15 12:13:58.000000000 +0200
@@ -25,6 +25,7 @@
 **********************************************************************/
 
 
+#include <algorithm>
 #include "rs_ellipse.h"
 
 #include "rs_graphic.h"
 
