Index: libkopete/private/kopeteviewmanager.cpp
===================================================================
--- kdenetwork/kopete/libkopete/private/kopeteviewmanager.cpp	(wersja 1155637)
+++ kdenetwork/kopete/libkopete/private/kopeteviewmanager.cpp	(kopia robocza)
@@ -305,8 +305,10 @@
 
 		if ( appendMessageEvent || showNotification )
 		{
-			if ( msg.from() && d->eventList.isEmpty() ) // may happen for internal messages
-				showNotification = true;
+			if (! ( msg.from() && d->eventList.isEmpty() ) ) // may happen for internal messages
+				//Changing this variable onwards should not break the notfications not 
+				//showing while Away
+				showNotification = false;
 
 			event = new Kopete::MessageEvent( msg, session );
 			d->eventList.append( event );
