diff -Nur mtools-4.0.10-old/mtools.c mtools-4.0.10/mtools.c
--- mtools-4.0.10-old/mtools.c	2009-06-15 14:59:21.345435652 +0300
+++ mtools-4.0.10/mtools.c	2009-06-15 14:59:45.263597397 +0300
@@ -64,7 +64,7 @@
 
 #ifdef HAVE_SETLOCALE
 	char *locale;
-	locale=setlocale(LC_ALL, "");
+	locale=setlocale(LC_ALL, "en_US.UTF-8");
 	if(locale == NULL || !strcmp(locale, "C"))
 		setlocale(LC_ALL, "en_US");
 #endif
