--- mformat.c
+++ mformat.c
@@ -3,6 +3,7 @@
  */
 
 #define DONT_NEED_WAIT
+#define DONT_NEED_IN
 
 #include "sysincludes.h"
 #include "msdos.h"
@@ -31,7 +32,7 @@
 
 #define _LINUX_STRING_H_
 #define kdev_t int
-#include "linux/fs.h"
+#include "sys/mount.h"
 #undef _LINUX_STRING_H_
 
 #endif
--- mpartition.c
+++ mpartition.c
@@ -2,6 +2,7 @@
  * mformat.c
  */
 #define DONT_NEED_WAIT
+#define DONT_NEED_IN
 
 #include "sysincludes.h"
 #include "msdos.h"
@@ -20,7 +21,7 @@
 
 #define _LINUX_STRING_H_
 #define kdev_t int
-#include "linux/fs.h"
+#include "sys/mount.h"
 #undef _LINUX_STRING_H_
 
 #endif
--- sysincludes.h
+++ sysincludes.h
@@ -268,12 +268,16 @@
 #endif
 
 #ifdef HAVE_NETINET_IN_H
+#ifndef DONT_NEED_IN
 #include <netinet/in.h>
 #endif
+#endif
 
 #ifdef HAVE_ARPA_INET_H
+#ifndef DONT_NEED_IN
 #include <arpa/inet.h>
 #endif
+#endif
 
 #ifdef HAVE_NETDB_H
 #include <netdb.h>
