--- configure.in
+++ configure.in
@@ -95,10 +95,10 @@
 dnl
 AC_MSG_CHECKING(whether llseek declared in unistd.h)
 AC_CACHE_VAL(mtools_cv_have_llseek_prototype,
-        AC_TRY_COMPILE(
+        [AC_TRY_COMPILE(
 [#include <unistd.h>], [extern int llseek(int);],
         [mtools_cv_have_llseek_prototype=no],
-        [mtools_cv_have_llseek_prototype=yes]))
+        [mtools_cv_have_llseek_prototype=yes])])
 AC_MSG_RESULT($mtools_cv_have_llseek_prototype)
 if test "$mtools_cv_have_llseek_prototype" = yes; then
    AC_DEFINE([HAVE_LLSEEK_PROTOTYPE],1,[Define when you have an LLSEEK prototype])
@@ -106,13 +106,13 @@
 
 AC_MSG_CHECKING(whether lseek64 declared in unistd.h)
 AC_CACHE_VAL(mtools_cv_have_lseek64_prototype,
-        AC_TRY_COMPILE(
+        [AC_TRY_COMPILE(
 [
 #include "sysincludes.h"
 #include <unistd.h>
 ], [extern int lseek64(int);],
         [mtools_cv_have_lseek64_prototype=no],
-        [mtools_cv_have_lseek64_prototype=yes]))
+        [mtools_cv_have_lseek64_prototype=yes])])
 AC_MSG_RESULT($mtools_cv_have_lseek64_prototype)
 if test "$mtools_cv_have_lseek64_prototype" = yes; then
    AC_DEFINE([HAVE_LSEEK64_PROTOTYPE],1,[Define when you have an LSEEK64 prototype])
@@ -145,12 +145,12 @@
 dnl
 AC_DEFUN(SFS_CHECK_OFF_T_64,
 [AC_CACHE_CHECK(for 64-bit off_t, sfs_cv_off_t_64,
-AC_TRY_COMPILE([
+[AC_TRY_COMPILE([
 #include <unistd.h>
 #include <sys/types.h>
 ],[
 switch (0) case 0: case (sizeof (off_t) <= 4):;
-], sfs_cv_off_t_64=no, sfs_cv_off_t_64=yes))
+], sfs_cv_off_t_64=no, sfs_cv_off_t_64=yes)])
 if test $sfs_cv_off_t_64 = yes; then
         AC_DEFINE([HAVE_OFF_T_64],1,[Define when the system has a 64 bit off_t type])
 fi])
