--- mtools.conf
+++ mtools.conf
@@ -14,25 +14,20 @@
 # You should have received a copy of the GNU General Public License
 # along with Mtools.  If not, see <http://www.gnu.org/licenses/>.
 
-# Example mtools.conf files.  Uncomment the lines which correspond to
-# your architecture and comment out the "SAMPLE FILE" line below
-SAMPLE FILE
+drive a: file="/dev/fd0" exclusive mformat_only cylinders=80 heads=2 sectors=18
+drive b: file="/dev/fd1" exclusive mformat_only cylinders=80 heads=2 sectors=18
 
-# # Linux floppy drives
-# drive a: file="/dev/fd0" exclusive
-# drive b: file="/dev/fd1" exclusive
+# First SCSI hard disk partition
+drive c: file="/dev/sda1"
 
-# # First SCSI hard disk partition
-# drive c: file="/dev/sda1"
+# First IDE hard disk partition
+#drive c: file="/dev/hda1"
 
-# # First IDE hard disk partition
-# drive c: file="/dev/hda1"
+#dosemu floppy image
+drive m: file="/var/lib/dosemu/diskimage"
 
-# # dosemu floppy image
-# drive m: file="/var/lib/dosemu/diskimage"
-
-# # dosemu hdimage
-# drive n: file="/var/lib/dosemu/diskimage" offset=3840
+#dosemu hdimage
+drive n: file="/var/lib/dosemu/hdimage.first" MTOOLS_SKIP_CHECK=1 MTOOLS_LOWER_CASE=1 MTOOLS_NO_VFAT=1 partition=1 offset=128
 
 # # Atari ramdisk image
 # drive o: file="/tmp/atari_rd" offset=136
@@ -60,21 +55,4 @@
 # ZIP drive on SCSI ID 6
 #drive z: file="/dev/scsi_disk_060" offset=16384 fat_bits=16                        
 
-# SCO Unix 3.2v4
-# # Floppy disk drives
-#
-# drive a: file="/dev/install" exclusive
-# drive b: file="/dev/install1" exclusive
-#  
-# # SCSI hard disk partitions
-#  
-# drive c: file="/dev/dsk/0sC"
-# drive d: file="/dev/dsk/0sD"
-# drive e: file="/dev/dsk/0sE"
-# drive f: file="/dev/dsk/0sF"
-# drive g: file="/dev/dsk/0sG"
-# drive h: file="/dev/dsk/0sH"
-
-# # uncomment the following line to display all file names in lower
-# # case by default
-# mtools_lower_case=1
+mtools_lower_case=1
