diff -Nuar enet-1.2.3.orig//include/enet/enet.h enet-1.2.3/include/enet/enet.h
--- enet-1.2.3.orig//include/enet/enet.h	2011-02-09 21:15:22.000000000 +0200
+++ enet-1.2.3/include/enet/enet.h	2011-03-07 12:53:52.762000014 +0200
@@ -13,15 +13,15 @@
 #include <stdlib.h>
 
 #ifdef WIN32
-#include "enet/win32.h"
+#include "./win32.h"
 #else
-#include "enet/unix.h"
+#include "./unix.h"
 #endif
 
-#include "enet/types.h"
-#include "enet/protocol.h"
-#include "enet/list.h"
-#include "enet/callbacks.h"
+#include "./types.h"
+#include "./protocol.h"
+#include "./list.h"
+#include "./callbacks.h"
 
 #define ENET_VERSION_MAJOR 1
 #define ENET_VERSION_MINOR 2
diff -Nuar enet-1.2.3.orig//include/enet/protocol.h enet-1.2.3/include/enet/protocol.h
--- enet-1.2.3.orig//include/enet/protocol.h	2011-02-09 21:09:20.000000000 +0200
+++ enet-1.2.3/include/enet/protocol.h	2011-03-07 12:54:04.352000014 +0200
@@ -5,7 +5,7 @@
 #ifndef __ENET_PROTOCOL_H__
 #define __ENET_PROTOCOL_H__
 
-#include "enet/types.h"
+#include "./types.h"
 
 enum
 {
