Index: system-config-printer-1.1.11/system-config-printer.py
===================================================================
--- system-config-printer-1.1.11.orig/system-config-printer.py
+++ system-config-printer-1.1.11/system-config-printer.py
@@ -1428,7 +1428,7 @@ class GUI(GtkGUI, monitor.Watcher):
                 def_emblem = None
                 emblem = None
                 if name == self.default_printer:
-                    def_emblem = 'emblem-default'
+                    def_emblem = 'dialog-ok'
                 elif name == userdef:
                     def_emblem = 'emblem-favorite'
 
