Index: libvisual-plugins-0.4.0/configure.ac
===================================================================
--- libvisual-plugins-0.4.0.orig/configure.ac
+++ libvisual-plugins-0.4.0/configure.ac
@@ -515,7 +515,7 @@ fi
 AC_SUBST(OPT_CFLAGS)
 
 AC_SUBST(CFLAGS, "${CFLAGS} ${DEBUG_CFLAGS} ${OPT_CFLAGS} ${GFORCE_CFLAGS}")
-AC_SUBST(CXXFLAGS, "${CFLAGS} ${DEBUG_CFLAGS} ${OPT_CFLAGS} ${GFORCE_CFLAGS}")
+AC_SUBST(CXXFLAGS, "${CXXFLAGS} ${DEBUG_CFLAGS} ${OPT_CFLAGS} ${GFORCE_CFLAGS}")
 
 dnl Plugin installation paths
 
