--- libvisual-plugins-0.4.0/plugins/actor/JESS/Makefile.am~	2006-02-23 09:03:24.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/JESS/Makefile.am	2006-12-27 02:28:08.000000000 +0000
@@ -2,7 +2,7 @@
 
 actor_plugin_LTLIBRARIES = actor_JESS.la
 
-LIBS += $(LIBVISUAL_LIBS)
+actor_JESS_la_LIBADD = $(LIBVISUAL_LIBS)
 
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/actor/gstreamer/Makefile.am~	2006-02-23 09:03:27.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/gstreamer/Makefile.am	2006-12-27 02:31:19.000000000 +0000
@@ -2,7 +2,7 @@
 
 actor_plugin_LTLIBRARIES = actor_gstreamer.la
 
-LIBS += $(LIBVISUAL_LIBS) $(GSTREAMER_LIBS)
+actor_gstreamer_la_LIBADD = $(LIBVISUAL_LIBS) $(GSTREAMER_LIBS)
 
 AM_CFLAGS = $(LIBVISUAL_CFLAGS) $(GSTREAMER_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/actor/G-Force/unix/libvisual/Makefile.am~	2006-03-08 13:06:58.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/G-Force/unix/libvisual/Makefile.am	2006-12-27 02:25:38.000000000 +0000
@@ -4,7 +4,7 @@
 
 GFORCE_TOP_SRCDIR = $(top_srcdir)/plugins/actor/G-Force
 
-LIBS = $(LIBVISUAL_LIBS)
+libactor_gforce_la_LIBADD = $(LIBVISUAL_LIBS)
 
 AM_CXXFLAGS = $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/actor/G-Force/unix/libmfl/Makefile.am~	2006-03-08 13:06:56.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/G-Force/unix/libmfl/Makefile.am	2006-12-27 02:25:52.000000000 +0000
@@ -2,7 +2,7 @@
 
 noinst_LTLIBRARIES = libmfl.la
 
-LIBS = $(LIBVISUAL_LIBS)
+libmfl_la_LIBADD = $(LIBVISUAL_LIBS)
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
 libmfl_la_SOURCES = $(ALLSOURCES)
--- libvisual-plugins-0.4.0/plugins/actor/G-Force/Common/math/Makefile.am~	2006-03-08 13:06:56.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/G-Force/Common/math/Makefile.am	2006-12-27 02:26:39.000000000 +0000
@@ -15,7 +15,7 @@
 
 ALLSOURCES = ExprArray.cpp ExprVirtualMachine.cpp Expression.cpp ExpressionDict.cpp FourierAnalyzer.cpp R3Matrix.cpp V3.cpp 
 
-LIBS = $(LIBVISUAL_LIBS)
+libmath_la_LIBADD = $(LIBVISUAL_LIBS)
 AM_CXXFLAGS = $(LIBVISUAL_CFLAGS)
 
 noinst_LTLIBRARIES = libmath.la
--- libvisual-plugins-0.4.0/plugins/actor/G-Force/Makefile.am~	2006-02-23 10:42:50.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/G-Force/Makefile.am	2006-12-27 02:26:52.000000000 +0000
@@ -4,7 +4,7 @@
 
 actor_plugin_LTLIBRARIES = actor_gforce.la
 
-LIBS = $(LIBVISUAL_LIBS)
+actor_gforce_la_LIBADD = $(LIBVISUAL_LIBS)
 
 INCLUDES = $(all_includes) -I$(top_srcdir)
 
--- libvisual-plugins-0.4.0/plugins/actor/G-Force/GForceCommon/Makefile.am~	2006-03-08 13:06:56.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/G-Force/GForceCommon/Makefile.am	2006-12-27 02:27:06.000000000 +0000
@@ -17,7 +17,7 @@
 
 DATADIR = $(LIBVISUAL_PLUGINS_DATA_DIR)/actor/actor_gforce
 
-LIBS = $(LIBVISUAL_LIBS)
+libgforce_la_LIBADD = $(LIBVISUAL_LIBS)
 AM_CXXFLAGS = -DDATADIR=\"$(DATADIR)\" $(LIBVISUAL_CFLAGS)
 
 
--- libvisual-plugins-0.4.0/plugins/actor/nastyfft/Makefile.am~	2006-02-24 11:24:50.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/nastyfft/Makefile.am	2006-12-27 02:27:24.000000000 +0000
@@ -4,5 +4,5 @@
 
 actor_nastyfft_la_CFLAGS = -I$(top_srcdir) $(LIBVISUAL_CFLAGS) $(X_CFLAGS)
 actor_nastyfft_la_LDFLAGS = -module -avoid-version 
-actor_nastyfft_la_LIBS = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU
+actor_nastyfft_la_LIBADD = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU
 actor_nastyfft_la_SOURCES = actor_nastyfft.c
--- libvisual-plugins-0.4.0/plugins/actor/lv_analyzer/Makefile.am~	2006-02-23 09:03:28.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/lv_analyzer/Makefile.am	2006-12-27 02:28:03.000000000 +0000
@@ -2,7 +2,7 @@
 
 actor_plugin_LTLIBRARIES = actor_lv_analyzer.la
 
-LIBS += $(LIBVISUAL_LIBS)
+actor_lv_analyzer_la_LIBADD = $(LIBVISUAL_LIBS)
 
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/actor/madspin/Makefile.am~	2006-02-24 11:24:49.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/madspin/Makefile.am	2006-12-27 02:29:13.000000000 +0000
@@ -10,5 +10,5 @@
 
 actor_madspin_la_CFLAGS = -I$(top_srcdir) $(LIBVISUAL_CFLAGS) -DSTAR_DIR=\""$(imagesdir)/"\"
 actor_madspin_la_LDFLAGS = -module -avoid-version 
-actor_madspin_la_LIBS = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL 
+actor_madspin_la_LIBADD = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL 
 actor_madspin_la_SOURCES = madspin.c
--- libvisual-plugins-0.4.0/plugins/actor/bumpscope/Makefile.am~	2006-02-23 09:03:24.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/bumpscope/Makefile.am	2006-12-27 02:29:28.000000000 +0000
@@ -2,7 +2,7 @@
 
 actor_plugin_LTLIBRARIES = actor_bumpscope.la
 
-LIBS += $(LIBVISUAL_LIBS)
+actor_bumpscope_la_LIBADD = $(LIBVISUAL_LIBS)
 
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/actor/pseudotoad_flower/Makefile.am~	2006-03-08 13:06:58.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/pseudotoad_flower/Makefile.am	2006-12-27 02:29:50.000000000 +0000
@@ -4,7 +4,7 @@
 
 actor_flower_la_LDFLAGS = -module -avoid-version 
 actor_flower_la_CFLAGS = -I$(top_srcdir) $(LIBVISUAL_CFLAGS) $(X_CFLAGS)
-actor_flower_la_LIBS = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU 
+actor_flower_la_LIBADD = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU 
 
 actor_flower_la_SOURCES = \
 	actor_flower.c \
--- libvisual-plugins-0.4.0/plugins/actor/lv_gltest/Makefile.am~	2006-02-24 11:24:49.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/lv_gltest/Makefile.am	2006-12-27 02:30:08.000000000 +0000
@@ -4,5 +4,5 @@
 
 actor_lv_gltest_la_CFLAGS = -I$(top_srcdir) $(LIBVISUAL_CFLAGS) $(X_CFLAGS)
 actor_lv_gltest_la_LDFLAGS = -module -avoid-version 
-actor_lv_gltest_la_LIBS = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU 
+actor_lv_gltest_la_LIBADD = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU 
 actor_lv_gltest_la_SOURCES = actor_lv_gltest.c
--- libvisual-plugins-0.4.0/plugins/actor/corona/Makefile.am~	2006-02-23 09:03:24.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/corona/Makefile.am	2006-12-27 02:30:22.000000000 +0000
@@ -2,7 +2,7 @@
 
 actor_plugin_LTLIBRARIES = actor_corona.la
 
-LIBS += $(LIBVISUAL_LIBS)
+actor_corona_la_LIBADD = $(LIBVISUAL_LIBS)
 
 AM_CXXFLAGS = $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/actor/jakdaw/Makefile.am~	2006-02-23 09:03:28.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/jakdaw/Makefile.am	2006-12-27 02:30:50.000000000 +0000
@@ -2,7 +2,7 @@
 
 actor_plugin_LTLIBRARIES = actor_jakdaw.la
 
-LIBS += $(LIBVISUAL_LIBS)
+actor_jakdaw_la_LIBADD = $(LIBVISUAL_LIBS)
 
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/actor/gdkpixbuf/Makefile.am~	2006-02-23 13:34:48.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/gdkpixbuf/Makefile.am	2006-12-27 02:31:08.000000000 +0000
@@ -2,7 +2,7 @@
 
 actor_plugin_LTLIBRARIES = actor_gdkpixbuf.la
 
-LIBS += $(LIBVISUAL_LIBS) $(GTK_LIBS)
+actor_gdkpixbuf_la_LIBADD = $(LIBVISUAL_LIBS) $(GTK_LIBS)
 
 AM_CFLAGS = $(LIBVISUAL_CFLAGS) $(GTK_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/actor/lv_scope/Makefile.am~	2006-02-23 09:03:29.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/lv_scope/Makefile.am	2006-12-27 02:31:29.000000000 +0000
@@ -2,7 +2,7 @@
 
 actor_plugin_LTLIBRARIES = actor_lv_scope.la
 
-LIBS += $(LIBVISUAL_LIBS)
+actor_lv_scope_la_LIBADD = $(LIBVISUAL_LIBS)
 
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/actor/oinksie/Makefile.am~	2006-02-23 09:03:31.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/oinksie/Makefile.am	2006-12-27 02:31:40.000000000 +0000
@@ -2,7 +2,7 @@
 
 actor_plugin_LTLIBRARIES = actor_oinksie.la
 
-LIBS += $(LIBVISUAL_LIBS)
+actor_oinksie_la_LIBADD = $(LIBVISUAL_LIBS)
 
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/actor/infinite/Makefile.am~	2006-02-23 09:03:27.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/actor/infinite/Makefile.am	2006-12-27 02:31:53.000000000 +0000
@@ -2,7 +2,7 @@
 
 actor_plugin_LTLIBRARIES = actor_infinite.la
 
-LIBS += $(LIBVISUAL_LIBS)
+actor_infinite_la_LIBADD = $(LIBVISUAL_LIBS)
 
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/input/esd/Makefile.am~	2006-02-23 09:03:37.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/input/esd/Makefile.am	2006-12-27 02:32:04.000000000 +0000
@@ -2,7 +2,7 @@
 
 input_plugin_LTLIBRARIES = input_esd.la
 
-LIBS += $(LIBESD_LIBS) $(LIBVISUAL_LIBS)
+input_esd_la_LIBADD = $(LIBESD_LIBS) $(LIBVISUAL_LIBS)
 
 AM_CFLAGS = $(LIBESD_CFLAGS) $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/input/alsa/Makefile.am~	2006-02-23 09:03:35.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/input/alsa/Makefile.am	2006-12-27 02:32:15.000000000 +0000
@@ -2,7 +2,7 @@
 
 input_plugin_LTLIBRARIES = input_alsa.la
 
-LIBS += $(ALSA_LIBS) $(LIBVISUAL_LIBS)
+input_alsa_la_LIBADD = $(ALSA_LIBS) $(LIBVISUAL_LIBS)
 
 AM_CFLAGS = $(ALSA_CFLAGS) $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/input/jack/Makefile.am~	2006-02-23 09:03:37.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/input/jack/Makefile.am	2006-12-27 02:32:26.000000000 +0000
@@ -2,7 +2,7 @@
 
 input_plugin_LTLIBRARIES = input_jack.la
 
-LIBS += $(LIBJACK_LIBS) $(LIBVISUAL_LIBS)
+input_jack_la_LIBADD = $(LIBJACK_LIBS) $(LIBVISUAL_LIBS)
 
 AM_CFLAGS = $(LIBJACK_CFLAGS) $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/input/debug/Makefile.am~	2006-02-23 09:03:36.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/input/debug/Makefile.am	2006-12-27 02:32:37.000000000 +0000
@@ -2,7 +2,7 @@
 
 input_plugin_LTLIBRARIES = input_debug.la
 
-LIBS += $(LIBVISUAL_LIBS)
+input_debug_la_LIBADD = $(LIBVISUAL_LIBS)
 
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/input/mplayer/Makefile.am~	2006-02-23 09:03:38.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/input/mplayer/Makefile.am	2006-12-27 02:32:46.000000000 +0000
@@ -2,7 +2,7 @@
 
 input_plugin_LTLIBRARIES = input_mplayer.la
 
-LIBS += $(LIBVISUAL_LIBS)
+input_mplayer_la_LIBADD = $(LIBVISUAL_LIBS)
 
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/morph/flash/Makefile.am~	2006-02-23 09:03:39.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/morph/flash/Makefile.am	2006-12-27 02:33:03.000000000 +0000
@@ -2,7 +2,7 @@
 
 morph_plugin_LTLIBRARIES = morph_flash.la
 
-LIBS += $(LIBVISUAL_LIBS)
+morph_flash_la_LIBADD = $(LIBVISUAL_LIBS)
 
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/morph/slide/Makefile.am~	2006-02-23 09:03:39.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/morph/slide/Makefile.am	2006-12-27 02:33:11.000000000 +0000
@@ -2,7 +2,7 @@
 
 morph_plugin_LTLIBRARIES = morph_slide.la
 
-LIBS += $(LIBVISUAL_LIBS)
+morph_slide_la_LIBADD = $(LIBVISUAL_LIBS)
 
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/morph/tentacle/Makefile.am~	2006-02-23 09:03:39.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/morph/tentacle/Makefile.am	2006-12-27 02:33:22.000000000 +0000
@@ -2,7 +2,7 @@
 
 morph_plugin_LTLIBRARIES = morph_tentacle.la
 
-LIBS += $(LIBVISUAL_LIBS)
+morph_tentacle_la_LIBADD = $(LIBVISUAL_LIBS)
 
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
--- libvisual-plugins-0.4.0/plugins/morph/alphablend/Makefile.am~	2006-02-23 09:03:38.000000000 +0000
+++ libvisual-plugins-0.4.0/plugins/morph/alphablend/Makefile.am	2006-12-27 02:33:31.000000000 +0000
@@ -2,7 +2,7 @@
 
 morph_plugin_LTLIBRARIES = morph_alphablend.la
 
-LIBS += $(LIBVISUAL_LIBS)
+morph_alphablend_la_LIBADD = $(LIBVISUAL_LIBS)
 
 AM_CFLAGS = $(LIBVISUAL_CFLAGS)
 
