https://bugs.gentoo.org/show_bug.cgi?id=179629

Index: libvisual-plugins-0.4.0/plugins/actor/G-Force/Common/math/Expression.cpp
===================================================================
--- libvisual-plugins-0.4.0.orig/plugins/actor/G-Force/Common/math/Expression.cpp
+++ libvisual-plugins-0.4.0/plugins/actor/G-Force/Common/math/Expression.cpp
@@ -4,6 +4,8 @@
 #include <gettext.h>
 #include <stdlib.h>
 
+#include <stdint.h>
+
 #include "Expression.h"
 
 #include "ExpressionDict.h"
