diff -Naupr pigment-0.3.17.orig/pgm/Makefile.am pigment-0.3.17/pgm/Makefile.am
--- pigment-0.3.17.orig/pgm/Makefile.am	2009-05-20 19:08:30.000000000 +0200
+++ pigment-0.3.17/pgm/Makefile.am	2009-05-31 13:03:21.745556855 +0200
@@ -128,7 +128,6 @@ libpigment_@PGM_MAJORMINOR@_includedir =
 libpigment_@PGM_MAJORMINOR@_include_HEADERS = \
 	$(source_h) \
 	pgmmarshal.h \
-	pgmenumtypes.h \
 	pgmversion.h
 
 EXTRA_DIST = \
