--- plugins/opengles/pgmglesviewport.c.OSmanOS	2009-05-20 20:08:30.000000000 +0300
+++ plugins/opengles/pgmglesviewport.c			2013-05-07 23:10:44.214736406 +0300
@@ -1418,22 +1418,6 @@
 void
 pgm_gles_viewport_update_drawable_projection (PgmGlesViewport *glesviewport)
 {
-  g_mutex_lock (glesviewport->layer_lock);
-
-  /* Adapt position and size of all the drawables */
-  g_list_foreach (glesviewport->near_layer,
-                  (GFunc) pgm_gles_drawable_update_projection, NULL);
-  g_list_foreach (glesviewport->middle_layer,
-                  (GFunc) pgm_gles_drawable_update_projection, NULL);
-  g_list_foreach (glesviewport->far_layer,
-                  (GFunc) pgm_gles_drawable_update_projection, NULL);
-
-  g_mutex_unlock (glesviewport->layer_lock);
-}
-
-void
-pgm_gles_viewport_update_drawable_projection (PgmGlesViewport *glesviewport)
-{
   GList *near_layer, *middle_layer, *far_layer;
 
   /* Get copies of the layers */
