--- CMakeLists.txt	2015-02-07 21:56:30.000000000 +0200
+++ CMakeLists.txt	2015-03-08 00:15:42.450350197 +0200
@@ -133,7 +133,7 @@
 find_package(Qt5X11Extras REQUIRED QUIET)
 find_package(Qt5LinguistTools REQUIRED QUIET)
 find_package(Qt5Xdg REQUIRED QUIET)
-find_package(KF5WindowSystem REQUIRED QUIET)
+
 message(STATUS "Building with Qt ${Qt5Core_VERSION_STRING}")
 
 QT5_WRAP_UI(UIS ${FORMS})
@@ -169,7 +169,7 @@
 set(LXQT_RELATIVE_TRANSLATIONS_DIR "${LXQT_LIBRARY_NAME}/translations")
 set(LXQT_TRANSLATIONS_DIR "${LXQT_SHARE_DIR}/translations")
 
-set(LXQT_INTREE_INCLUDE_DIR "${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/include")
+set(LXQT_INTREE_INCLUDE_DIR "${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/usr/include")
 set(LXQT_INTREE_TARGETS_FILE "${CMAKE_BINARY_DIR}/${LXQT_LIBRARY_NAME}-targets.cmake")
 
 set(LXQT_INSTALL_INCLUDE_DIR "${CMAKE_INSTALL_FULL_INCLUDEDIR}/${LXQT_LIBRARY_NAME}")
@@ -301,7 +301,6 @@
 )
 
 target_link_libraries(${LXQT_LIBRARY_NAME}
-    KF5::WindowSystem
     Qt5::Widgets
     Qt5::DBus
     Qt5::X11Extras
