--- lxqtsingleapplication.cpp	2015-02-07 21:56:30.000000000 +0200
+++ lxqtsingleapplication.cpp	2015-03-08 12:54:50.056306830 +0200
@@ -27,8 +27,8 @@
 
 #include "lxqtsingleapplication.h"
 #include "singleapplicationadaptor.h"
-#include <KWindowSystem/KWindowSystem>
-#include <KWindowSystem/NETWM>
+#include <KF5/KWindowSystem/KWindowSystem>
+#include <KF5/KWindowSystem/NETWM>
 #include <QDBusMessage>
 #include <QWidget>
 #include <QDebug>
