diff -up irrlicht-1.7.1/source/Irrlicht/CZipReader.cpp.libaesgm irrlicht-1.7.1/source/Irrlicht/CZipReader.cpp
--- irrlicht-1.7.1/source/Irrlicht/CZipReader.cpp.libaesgm	2010-01-31 16:03:28.000000000 -0500
+++ irrlicht-1.7.1/source/Irrlicht/CZipReader.cpp	2010-05-24 11:15:18.856512375 -0400
@@ -20,7 +20,7 @@
 	#endif
 
 	#ifdef _IRR_COMPILE_WITH_ZIP_ENCRYPTION_
-	#include "aesGladman/fileenc.h"
+	#include <aes/fileenc.h>
 	#endif
 	#ifdef _IRR_COMPILE_WITH_BZIP2_
 	#ifndef _IRR_USE_NON_SYSTEM_BZLIB_
--- irrlicht-1.7.1/source/Irrlicht/Makefile.libaesgm	2010-05-24 11:18:35.966512547 -0400
+++ irrlicht-1.7.1/source/Irrlicht/Makefile	2010-05-24 11:18:10.368387943 -0400
@@ -54,7 +54,7 @@ LINKOBJ = $(IRRMESHOBJ) $(IRROBJ) $(IRRP
 ###############
 #Compiler flags
 CXXINCS = -I../../include
-CPPFLAGS += $(CXXINCS) -DIRRLICHT_EXPORTS=1 $(RPM_OPT_FLAGS)
+CPPFLAGS += $(CXXINCS) -DIRRLICHT_EXPORTS=1 -DUSE_SHA1 $(RPM_OPT_FLAGS)
 CXXFLAGS += $(RPM_OPT_FLAGS) -Wall -pipe -fno-exceptions -fno-rtti -fstrict-aliasing
 ifndef NDEBUG
 CXXFLAGS += -g -D_DEBUG
