diff -Nur cluttermm-0.9.4-old/Makefile.am cluttermm-0.9.4/Makefile.am
--- cluttermm-0.9.4-old/Makefile.am	2009-10-04 02:36:07.129885011 +0300
+++ cluttermm-0.9.4/Makefile.am	2009-10-04 02:36:20.169135720 +0300
@@ -1,6 +1,8 @@
 SUBDIRS		= tools clutter docs examples tests
 DIST_SUBDIRS	= $(SUBDIRS) scripts
 
+ACLOCAL_AMFLAGS = -Iscripts
+
 EXTRA_DIST	=  \
 		  build_shared/Makefile_build.am_fragment \
 		  build_shared/Makefile_build_gensrc.am_fragment \
