Index: paprefs-0.9.8/src/paprefs.desktop.in
===================================================================
--- paprefs-0.9.8.orig/src/paprefs.desktop.in
+++ paprefs-0.9.8/src/paprefs.desktop.in
@@ -5,7 +5,7 @@ _Name=PulseAudio Preferences
 _GenericName=Sound Server Preferences
 _Comment=View and modify the configuration of the local sound server
 Exec=paprefs
-Icon=preferences-desktop
+Icon=paprefs
 StartupNotify=true
 Type=Application
-Categories=Settings;
+Categories=Application;AudioVideo;Audio;
