Index: paprefs-0.9.9/src/paprefs.glade
===================================================================
--- paprefs-0.9.9.orig/src/paprefs.glade
+++ paprefs-0.9.9/src/paprefs.glade
@@ -5,7 +5,7 @@
   <widget class="GtkWindow" id="mainWindow">
     <property name="title" translatable="yes">PulseAudio Preferences</property>
     <property name="resizable">False</property>
-    <property name="icon_name">preferences-desktop</property>
+    <property name="icon_name">paprefs</property>
     <child>
       <widget class="GtkVBox" id="vbox20">
         <property name="visible">True</property>
