diff -Nur libfame-0.9.1-old/configure libfame-0.9.1/configure
--- libfame-0.9.1-old/configure	2008-04-03 15:21:40.000000000 +0300
+++ libfame-0.9.1/configure	2008-04-03 15:29:59.000000000 +0300
@@ -4511,12 +4511,12 @@
             ;;
     i486-*-*)
             if test x$ac_cv_c_compiler_gnu = xyes; then
-                CFLAGS="$CFLAGS -march=i486 -DHAS_BSWAP"
+                CFLAGS="$CFLAGS -DHAS_BSWAP"
             fi
             ;;
     i?86-*-*)
             if test x$ac_cv_c_compiler_gnu = xyes; then
-                CFLAGS="$CFLAGS -march=i586 -DHAS_BSWAP"
+                CFLAGS="$CFLAGS -DHAS_BSWAP"
             fi
             ;;
 esac
