diff -p -up ekiga-3.1.0/src/gui/main.cpp~ ekiga-3.1.0/src/gui/main.cpp
--- ekiga-3.1.0/src/gui/main.cpp~	2009-01-05 18:31:38.000000000 +0100
+++ ekiga-3.1.0/src/gui/main.cpp	2009-01-06 15:05:38.000000000 +0100
@@ -603,7 +603,7 @@ static gboolean on_stats_refresh_cb (gpo
                            mw->priv->current_call->get_received_video_bandwidth (),
                            videooutput_stats.tx_fps,
                            videooutput_stats.rx_fps);
-    ekiga_main_window_flash_message (mw, msg);
+    ekiga_main_window_flash_message (mw, "%s", msg);
     ekiga_main_window_set_call_duration (mw, mw->priv->current_call->get_duration ().c_str ());
     g_free (msg);
 
