--- hpwd.c
+++ hpwd.c
@@ -24,6 +24,7 @@
 # endif
 
 # include <stdio.h>
+# include <string.h>
 
 # include "hfs.h"
 # include "hcwd.h"
--- libhfs/os/unix.c
+++ libhfs/os/unix.c
@@ -53,7 +53,7 @@
  */
 int os_open(void **priv, const char *path, int mode)
 {
-  int fd;
+  long int fd;
   struct flock lock;
 
   switch (mode)
@@ -100,7 +100,7 @@
  */
 int os_close(void **priv)
 {
-  int fd = (int) *priv;
+  long int fd = (long int) *priv;
 
   *priv = (void *) -1;
 
@@ -119,7 +119,7 @@
  */
 int os_same(void **priv, const char *path)
 {
-  int fd = (int) *priv;
+  long int fd = (long int) *priv;
   struct stat fdev, dev;
 
   if (fstat(fd, &fdev) == -1 ||
@@ -139,7 +139,7 @@
  */
 unsigned long os_seek(void **priv, unsigned long offset)
 {
-  int fd = (int) *priv;
+  long int fd = (long int) *priv;
   off_t result;
 
   /* offset == -1 special; seek to last block of device */
@@ -164,7 +164,7 @@
  */
 unsigned long os_read(void **priv, void *buf, unsigned long len)
 {
-  int fd = (int) *priv;
+  long int fd = (long int) *priv;
   ssize_t result;
 
   result = read(fd, buf, len << HFS_BLOCKSZ_BITS);
@@ -184,7 +184,7 @@
  */
 unsigned long os_write(void **priv, const void *buf, unsigned long len)
 {
-  int fd = (int) *priv;
+  long int fd = (long int) *priv;
   ssize_t result;
 
   result = write(fd, buf, len << HFS_BLOCKSZ_BITS);
