--- ushare-1.1a/src/ushare.c	2007-12-09 14:03:36.000000000 +0100
+++ ushare-1.1a/src/ushare.c.new	2008-03-09 09:58:06.000000000 +0100
@@ -424,8 +424,6 @@
     if ((itf->ifa_flags & IFF_UP)
         && !strncmp (itf->ifa_name, interface, IFNAMSIZ))
     {
-      log_error (_("Interface %s is down.\n"), interface);
-      log_error (_("Recheck uShare's configuration and try again !\n"));
       freeifaddrs (itflist);
       return true;
     }
