diff -up kphotoalbum-4.3/MainWindow/WelcomeDialog.cpp.xdg_Pictures kphotoalbum-4.3/MainWindow/WelcomeDialog.cpp
--- kphotoalbum-4.3/MainWindow/WelcomeDialog.cpp.xdg_Pictures	2012-09-25 10:40:43.000000000 -0500
+++ kphotoalbum-4.3/MainWindow/WelcomeDialog.cpp	2012-11-28 15:09:35.061085929 -0600
@@ -112,7 +112,7 @@ FileDialog::FileDialog( QWidget* parent
     QLabel* label = new QLabel( i18n("<h1>KPhotoAlbum database creation</h1>"
                                      "<p>You need to show where the photos and videos are for KPhotoAlbum to "
                                      "find them. They all need to be under one root directory, for example "
-                                     "/home/user/Images. In this directory you can have as many subdirectories as you "
+                                     "/home/user/Pictures. In this directory you can have as many subdirectories as you "
                                      "want, KPhotoAlbum will find them all for you.</p>"
                                      "<p>Feel safe, KPhotoAlbum will not modify or edit any of your images, so you can "
                                      "simply point KPhotoAlbum to the directory where you already have all your "
@@ -128,7 +128,7 @@ FileDialog::FileDialog( QWidget* parent
     lay2->addWidget( label );
 
     _lineEdit = new KLineEdit( top );
-    _lineEdit->setText( QString::fromLatin1( "~/Images" ) );
+    _lineEdit->setText( KGlobalSettings::picturesPath() );
     lay2->addWidget( _lineEdit );
 
     QPushButton* button = new QPushButton( QString::fromLatin1("..."), top ); 
