Index: suspend-0.8_20100831/resume.c
===================================================================
--- suspend-0.8_20100831.orig/resume.c
+++ suspend-0.8_20100831/resume.c
@@ -471,19 +471,9 @@ int main(int argc, char *argv[])
 
 	while (stat(resume_dev_name, &stat_buf)) {
 		fprintf(stderr, 
-			"%s: Could not stat the resume device file '%s'\n"
-			"\tPlease type in the full path name to try again\n"
-			"\tor press ENTER to boot the system: ", my_name,
-			resume_dev_name);
-		fgets(resume_dev_name, MAX_STR_LEN - 1, stdin);
-		n = strlen(resume_dev_name) - 1;
-		if (n <= 0) {
-			error = EINVAL;
-			goto Free;
-		}
-
-		if (resume_dev_name[n] == '\n')
-			resume_dev_name[n] = '\0';
+			"%s: Could not stat the resume device file '%s'\n", my_name, resume_dev_name);
+		error = EINVAL;
+		goto Free;
 	}
 
 	setvbuf(stdout, NULL, _IONBF, 0);
