diff -Nur gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpegdemux.c gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegdemux.c
--- gst-ffmpeg-0.10.4-old/ext/ffmpeg/gstffmpegdemux.c	2008-06-02 12:04:46.000000000 +0300
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegdemux.c	2008-06-02 12:05:41.000000000 +0300
@@ -136,9 +136,6 @@
   const gchar *message = NULL;
 
   switch (av_errno) {
-    case AVERROR_UNKNOWN:
-      message = "Unknown error";
-      break;
     case AVERROR_IO:
       message = "Input/output error";
       break;
