diff -ur oprofile-0.9.3.orig/libutil++/file_manip.cpp oprofile-0.9.3/libutil++/file_manip.cpp
--- oprofile-0.9.3.orig/libutil++/file_manip.cpp	2008-05-08 23:59:23.000000000 +0300
+++ oprofile-0.9.3/libutil++/file_manip.cpp	2008-05-09 00:00:44.000000000 +0300
@@ -39,7 +39,7 @@
 	if (stat(source.c_str(), &buf))
 		return false;
 
-	int fd = open(destination.c_str(), O_RDWR|O_CREAT);
+	int fd = open(destination.c_str(), O_RDWR);
 	if (fd < 0)
 		return false;
 	close(fd);
