--- libregex/tests/java_test.cpp.old	2008-09-03 00:57:19.357004497 +0200
+++ libregex/tests/java_test.cpp	2008-09-03 00:57:38.234025373 +0200
@@ -16,6 +16,7 @@
 
 #include <iostream>
 #include <fstream>
+#include <cstdlib>
 
 using namespace std;
 
--- libregex/tests/regex_test.cpp.old	2008-09-03 00:55:18.564005266 +0200
+++ libregex/tests/regex_test.cpp	2008-09-03 00:55:43.576001018 +0200
@@ -20,6 +20,7 @@
 
 #include <iostream>
 #include <fstream>
+#include <cstdlib>
 
 using namespace std;
 
--- libabi/tests/abi_test.cpp.old	2005-08-17 21:15:42.000000000 +0200
+++ libabi/tests/abi_test.cpp	2008-09-03 01:00:04.091003916 +0200
@@ -17,6 +17,8 @@
 
 #include <fstream>
 #include <iostream>
+#include <cstring>
+#include <cstdlib>
 
 using namespace std;
 
