--- oprofile-0.4/gui/oprof_start_config.cpp.guess2	2002-10-01 18:32:31.000000000 -0400
+++ oprofile-0.4/gui/oprof_start_config.cpp	2002-12-06 14:45:19.000000000 -0500
@@ -98,8 +98,7 @@
 		perror("oprof_start: Unable to determine OS release.");
 	} else {
 		string const version(info.release);
-		string const vmlinux_path("/lib/modules/" + version
-					 + "/build/vmlinux");
+		std::string const vmlinux_path("/usr/lib/debug/lib/modules/" + version + "/vmlinux");
 		kernel_filename = vmlinux_path;
 	}
 }
