Index: oprofile-0.9.4/configure.in
===================================================================
--- oprofile-0.9.4.orig/configure.in
+++ oprofile-0.9.4/configure.in
@@ -96,6 +96,9 @@ ORIG_SAVE_LIBS="$LIBS"
 dnl advanced glibc features which we need but may not be present
 AC_CHECK_FUNCS(sched_setaffinity perfmonctl)
 
+AC_USE_SYSTEM_EXTENSIONS
+AC_CHECK_DECLS_ONCE([basename])
+
 AC_CHECK_LIB(popt, poptGetContext,, AC_MSG_ERROR([popt library not found]))
 
 AX_BINUTILS
