diff -Nur oprofile-0.9.4-old/configure.in oprofile-0.9.4/configure.in
--- oprofile-0.9.4-old/configure.in	2009-06-16 19:19:09.098644953 +0300
+++ oprofile-0.9.4/configure.in	2009-06-16 19:21:57.769541950 +0300
@@ -139,8 +139,8 @@
 
 dnl finally restore the original libs setting
 LIBS="$ORIG_SAVE_LIBS"
-LIBERTY_LIBS="-liberty $DL_LIB $INTL_LIB"
-BFD_LIBS="-lbfd -liberty $DL_LIB $INTL_LIB"
+LIBERTY_LIBS="-lz /usr/lib/libiberty.a $DL_LIB $INTL_LIB"
+BFD_LIBS="-lz /usr/lib/libbfd.a /usr/lib/libiberty.a $DL_LIB $INTL_LIB"
 POPT_LIBS="-lpopt"
 AC_SUBST(LIBERTY_LIBS)
 AC_SUBST(BFD_LIBS)
diff -Nur oprofile-0.9.4-old/m4/binutils.m4 oprofile-0.9.4/m4/binutils.m4
--- oprofile-0.9.4-old/m4/binutils.m4	2009-06-16 19:19:09.124761759 +0300
+++ oprofile-0.9.4/m4/binutils.m4	2009-06-16 19:20:31.654785266 +0300
@@ -8,11 +8,11 @@
 AC_CHECK_FUNCS(xmemdup)
 AC_CHECK_LIB(dl, dlopen, LIBS="$LIBS -ldl"; DL_LIB="-ldl", DL_LIB="")
 AC_CHECK_LIB(intl, main, LIBS="$LIBS -lintl"; INTL_LIB="-lintl", INTL_LIB="")
-AC_CHECK_LIB(bfd, bfd_openr,, AC_MSG_ERROR([bfd library not found]))
+dnl AC_CHECK_LIB(bfd, bfd_openr, LIBS="$LIBS -lz /usr/lib/libbfd.a /usr/lib/libiberty.a", AC_MSG_ERROR([bfd library not found]))
 
 AC_LANG_PUSH(C)
 SAVE_LIBS=$LIBS
-LIBS=" -lbfd -liberty "
+LIBS=" -lz /usr/lib/libbfd.a /usr/lib/libiberty.a "
 # Determine if bfd_get_synthetic_symtab macro is available
 OS="`uname`"
 if test "$OS" = "Linux"; then
diff -Nur oprofile-0.9.4-old/m4/cellspubfdsupport.m4 oprofile-0.9.4/m4/cellspubfdsupport.m4
--- oprofile-0.9.4-old/m4/cellspubfdsupport.m4	2009-06-16 19:19:09.124761759 +0300
+++ oprofile-0.9.4/m4/cellspubfdsupport.m4	2009-06-16 19:20:31.733947365 +0300
@@ -12,7 +12,7 @@
 
 AC_LANG_PUSH(C)
 SAVE_LIBS=$LIBS
-LIBS=" -lbfd -liberty "
+LIBS=" -lz /usr/lib/libbfd.a /usr/lib/libiberty.a "
 
 AC_CHECK_LIB(bfd, bfd_openr_iovec,
 	[bfd_openr_iovec_exists="yes"],
