--- tls/tests/connection.c~	2013-01-27 10:52:09.814525435 +0100
+++ tls/tests/connection.c	2013-01-27 10:52:32.801524595 +0100
@@ -853,10 +853,6 @@ main (int   argc,
               setup_connection, test_simultaneous_async, teardown_connection);
   g_test_add ("/tls/connection/simultaneous-sync", TestConnection, NULL,
 	      setup_connection, test_simultaneous_sync, teardown_connection);
-  g_test_add ("/tls/connection/simultaneous-async-rehandshake", TestConnection, NULL,
-              setup_connection, test_simultaneous_async_rehandshake, teardown_connection);
-  g_test_add ("/tls/connection/simultaneous-sync-rehandshake", TestConnection, NULL,
-	      setup_connection, test_simultaneous_sync_rehandshake, teardown_connection);
 
   ret = g_test_run();
 
