--- kaptan-5.0.5/po/tr.po	2011-10-19 14:58:27.410216504 +0300
+++ tr.po	2011-10-19 15:02:55.562499393 +0300
@@ -471,8 +471,8 @@
 msgstr "Güncellemeleri kontrol etme sıklığı"
 
 #: .tmp/ui_scrPackage.py:135
-msgid " hours"
-msgstr " saat"
+msgid " minute(s)"
+msgstr " dakika"
 
 #: .tmp/ui_scrPackage.py:136
 msgid "To install and remove programs to Pardus, use Package-Manager. Package-Manager also can automatically install the latest updates available. You can set the period for checking new updates."
diff -Nuar kaptan-5.0.5/src/kaptan/screens/scrPackage.py /home/meltem/Desktop/kaptan-5.0.5//src/kaptan/screens/scrPackage.py
--- kaptan-5.0.5/src/kaptan/screens/scrPackage.py       2011-01-03 11:59:05.000000000 +0200
+++ /home/meltem/Desktop/kaptan-5.0.5//src/kaptan/screens/scrPackage.py 2011-08-22 08:46:20.533000420 +0300
@@ -28,17 +28,11 @@
     title = ki18n("Packages")
     desc = ki18n("Install / Remove Programs")

-    # min update time
-    updateTime = 12
-
     def __init__(self, *args):
         QtGui.QWidget.__init__(self,None)
         self.ui = Ui_packageWidget()
         self.ui.setupUi(self)

-        # set updateTime
-        self.ui.updateInterval.setValue(self.updateTime)
-
         # set initial checkbox visibility
         self.ui.checkUpdate.setVisible(False)
         self.ui.updateInterval.setVisible(False)
@@ -66,9 +60,9 @@
     def applySettings(self):
         # write selected configurations to future package-managerrc
         config = PMConfig()
-        config.setSystemTray(QVariant(self.ui.showTray.isChecked()))
-        config.setUpdateCheck(QVariant(self.ui.checkUpdate.isChecked()))
-        config.setUpdateCheckInterval(QVariant(self.ui.updateInterval.value() * 60))
+        config.setSystemTray(self.ui.showTray.isChecked())
+        config.setUpdateCheck(self.ui.checkUpdate.isChecked())
+        config.setUpdateCheckInterval(self.ui.updateInterval.value())

         if self.ui.showTray.isChecked():
             p = subprocess.Popen(["package-manager"], stdout=subprocess.PIPE)
diff -Nuar kaptan-5.0.5/ui/ui_scrPackage.ui /home/meltem/Desktop/kaptan-5.0.5//ui/ui_scrPackage.ui
--- kaptan-5.0.5/ui/ui_scrPackage.ui    2011-01-03 11:59:05.000000000 +0200
+++ /home/meltem/Desktop/kaptan-5.0.5//ui/ui_scrPackage.ui      2011-08-19 14:34:10.862000421 +0300
@@ -108,13 +108,16 @@
              <string notr="true">color: rgb(30, 43, 51)</string>
             </property>
             <property name="suffix">
-             <string> hours</string>
+             <string> minute(s)</string>
             </property>
             <property name="minimum">
-             <number>2</number>
+             <number>1</number>
+            </property>
+            <property name="maximum">
+             <number>10000</number>
             </property>
             <property name="value">
-             <number>2</number>
+             <number>1</number>
             </property>
            </widget>
           </item>
