--- src/kaptan/screens/scrPackage.py.orig       2013-12-03 22:16:44.811087187 +0200
+++ src/kaptan/screens/scrPackage.py    2013-12-03 22:20:40.525093500 +0200
@@ -62,7 +62,7 @@
         config = PMConfig()
         config.setSystemTray(self.ui.showTray.isChecked())
         config.setUpdateCheck(self.ui.checkUpdate.isChecked())
-        config.setUpdateCheckInterval(self.ui.updateInterval.value())
+        config.setUpdateCheckInterval(self.ui.updateInterval.value() * 60)
 
         if self.ui.showTray.isChecked():
             p = subprocess.Popen(["package-manager"], stdout=subprocess.PIPE)
 
