--- kaptan-5.0.5-orig/src/kaptan/screens/scrGoodbye.py	2010-08-31 21:28:53.000000000 +0300
+++ kaptan-5.0.5/src/kaptan/screens/scrGoodbye.py	2013-01-15 00:44:19.468006000 +0200
@@ -31,9 +31,9 @@
 
         lang = KGlobal.locale().language()
         if lang == "tr":
-            self.helpPageUrl = "http://www.pardus.org.tr/destek"
+            self.helpPageUrl = "http://www.pardus-anka.org/"
         else:
-            self.helpPageUrl = "http://www.pardus.org.tr/eng/support"
+            self.helpPageUrl = "http://www.pardus-anka.org/"
 
         self.smoltUrl = "http://smolt.pardus.org.tr:8090"
 
