--- src/kaptan/screens/scrGoodbye.py.orig	2013-10-13 00:59:06.268829177 +0300
+++ src/kaptan/screens/scrGoodbye.py	2013-10-13 00:59:52.977829934 +0300
@@ -39,7 +39,7 @@
 
     def on_buttonSystemSettings_clicked(self):
         self.procSettings = QProcess()
-        self.procSettings.start("systemsettings")
+        self.procSettings.startDetached("systemsettings")
 
     def on_buttonHelpPages_clicked(self):
         self.procSettings = QProcess()
