diff -Naur econcentration-0.0.1_55225.orig//data/desktop/econcentration.desktop econcentration-0.0.1_55225/data/desktop/econcentration.desktop
--- econcentration-0.0.1_55225.orig//data/desktop/econcentration.desktop	2011-03-01 13:59:43.000000000 +0200
+++ econcentration-0.0.1_55225/data/desktop/econcentration.desktop	2011-03-11 22:55:23.067000039 +0200
@@ -3,6 +3,7 @@
 Name=Econcentration
 Comment=Game
 Comment[pt]=Jogo
+Comment[tr]=Oyun
 Exec=econcentration
 Icon=econcentration
 Terminal=false
