Index: blobwars-1.19/makefile
===================================================================
--- blobwars-1.19.orig/makefile
+++ blobwars-1.19/makefile
@@ -9,8 +9,8 @@ RELEASE ?= 0
 USEPAK ?= 0
 
 PREFIX ?= /usr
-BINDIR ?= $(PREFIX)/games/
-DATADIR ?= $(PREFIX)/share/games/blobwars/
+BINDIR ?= $(PREFIX)/bin/
+DATADIR ?= $(PREFIX)/share/blobwars/
 DOCDIR ?= $(PREFIX)/share/doc/$(PROG)/
 ICONDIR ?= $(PREFIX)/share/icons/hicolor/
 DESKTOPDIR ?= $(PREFIX)/share/applications/
