--- cutecom.desktop.orig	2009-09-18 15:23:11.935027115 +0300
+++ cutecom.desktop	2009-09-18 15:41:35.818777775 +0300
@@ -2,12 +2,16 @@
 Encoding=UTF-8
 BinaryPattern=
 Name=CuteCom
+Name[tr]=CuteCom
 MimeType=
 GenericName=Serial Terminal
+GenericName[tr]=Seri Terminali
 Exec=cutecom
-Icon=openterm
+Icon=utilities-terminal
 TerminalOptions=
 Path=
 Type=Application
-Terminal=0
+Terminal=false
 X-KDE-StartupNotify=false
+Categories=Qt;Development;Utility;X-KDE-Utilities-Peripherals
+
