Link against static libnettle

There's no libnettle udeb.

Index: xorg-server/configure.ac
===================================================================
--- xorg-server.orig/configure.ac	2013-08-02 20:15:25.471466837 +0200
+++ xorg-server/configure.ac	2013-08-02 20:17:12.875999417 +0200
@@ -1433,7 +1433,7 @@
 if test "x$with_sha1" = xlibnettle; then
 	AC_DEFINE([HAVE_SHA1_IN_LIBNETTLE], [1],
 	          [Use libnettle SHA1 functions])
-	SHA1_LIBS=-lnettle
+	SHA1_LIBS=-l:libnettle.a
 fi
 AC_CHECK_LIB([gcrypt], [gcry_md_open], [HAVE_LIBGCRYPT=yes])
 if test "x$with_sha1" = x && test "x$HAVE_LIBGCRYPT" = xyes; then
