diff -Nur gmock-1.5.0-old//Makefile.am gmock-1.5.0/Makefile.am
--- gmock-1.5.0-old//Makefile.am	2011-04-05 16:32:41.691001439 +0300
+++ gmock-1.5.0/Makefile.am	2011-04-05 16:35:57.779001526 +0300
@@ -28,6 +28,7 @@
 lib_LTLIBRARIES = lib/libgmock.la lib/libgmock_main.la
 
 lib_libgmock_la_SOURCES = src/gmock-all.cc
+lib_libgmock_la_LIBADD = $(GTEST_LIBS)
 
 pkginclude_HEADERS = include/gmock/gmock.h \
                      include/gmock/gmock-actions.h \
