diff -Nur gmock-1.5.0-old//build-aux/ltmain.sh gmock-1.5.0/build-aux/ltmain.sh
--- gmock-1.5.0-old//build-aux/ltmain.sh	2011-04-05 16:18:08.123000906 +0300
+++ gmock-1.5.0/build-aux/ltmain.sh	2011-04-05 16:19:11.846001458 +0300
@@ -1647,7 +1647,7 @@
 	continue
 	;;
 
-     -mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe|-threads)
+     -mt|-mthreads|-kthread|-Kthread|-lpthread|-pthread|-pthreads|--thread-safe|-threads)
 	compiler_flags="$compiler_flags $arg"
 	compile_command="$compile_command $arg"
 	finalize_command="$finalize_command $arg"
@@ -2150,7 +2150,7 @@
 	lib=
 	found=no
 	case $deplib in
-	-mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe|-threads)
+	-mt|-mthreads|-kthread|-Kthread|-lpthread|-pthread|-pthreads|--thread-safe|-threads)
 	  if test "$linkmode,$pass" = "prog,link"; then
 	    compile_deplibs="$deplib $compile_deplibs"
 	    finalize_deplibs="$deplib $finalize_deplibs"
diff -Nur gmock-1.5.0-old//gtest/m4/acx_pthread.m4 gmock-1.5.0/gtest/m4/acx_pthread.m4
--- gmock-1.5.0-old//gtest/m4/acx_pthread.m4	2011-04-05 16:18:08.133001175 +0300
+++ gmock-1.5.0/gtest/m4/acx_pthread.m4	2011-04-05 16:18:49.919001484 +0300
@@ -99,7 +99,7 @@
 # which indicates that we try without any flags at all, and "pthread-config"
 # which is a program returning the flags for the Pth emulation library.
 
-acx_pthread_flags="pthreads none -Kthread -kthread lthread -pthread -pthreads -mthreads pthread --thread-safe -mt pthread-config"
+acx_pthread_flags="pthreads none -Kthread -kthread lthread -lpthread -pthread -pthreads -mthreads pthread --thread-safe -mt pthread-config"
 
 # The ordering *is* (sometimes) important.  Some notes on the
 # individual items follow:
