--- libmatecanvas-1.2.0/docs/reference/Makefile.am.orig	2012-02-20 18:32:58.000000000 +0200
+++ libmatecanvas-1.2.0/docs/reference/Makefile.am	2012-07-24 00:56:28.171610496 +0300
@@ -29,8 +29,6 @@
 # CFLAGS and LDFLAGS for compiling scan program. Only needed if your app/lib
 # contains GtkObjects/GObjects and you want to document signals and properties.
 INCLUDES = -I$(top_srcdir) $(LIBMATECANVAS_CFLAGS)
-GTKDOC_LIBS = $(top_builddir)/libmatecanvas/libmatecanvas-2.la \
-  $(LIBMATECANVAS_LIBS)
 
 # Extra options to supply to gtkdoc-mkdb.
 MKDB_OPTIONS=--sgml-mode --output-format=xml
@@ -44,4 +42,3 @@
 # Extra options to supply to gtkdoc-fixref.
 FIXXREF_OPTIONS=
 
-include $(top_srcdir)/gtk-doc.make
