Index: anjuta-3.6.1/manuals/anjuta-manual/Makefile.in
===================================================================
--- anjuta-3.6.1.orig/manuals/anjuta-manual/Makefile.in
+++ anjuta-3.6.1/manuals/anjuta-manual/Makefile.in
@@ -576,7 +576,7 @@ uninstall-am:
 	maintainer-clean-generic mostlyclean mostlyclean-generic \
 	mostlyclean-libtool pdf pdf-am ps ps-am uninstall uninstall-am
 
-@YELP_HELP_RULES@
+	@YELP_HELP_RULES@
 
 -include $(top_srcdir)/git.mk
 
Index: anjuta-3.6.1/manuals/anjuta-manual/Makefile.am
===================================================================
--- anjuta-3.6.1.orig/manuals/anjuta-manual/Makefile.am
+++ anjuta-3.6.1/manuals/anjuta-manual/Makefile.am
@@ -1,4 +1,4 @@
-@YELP_HELP_RULES@
+	@YELP_HELP_RULES@
 
 HELP_ID = anjuta-manual
 
Index: anjuta-3.6.1/manuals/anjuta-faqs/Makefile.in
===================================================================
--- anjuta-3.6.1.orig/manuals/anjuta-faqs/Makefile.in
+++ anjuta-3.6.1/manuals/anjuta-faqs/Makefile.in
@@ -13,7 +13,7 @@
 # even the implied warranty of MERCHANTABILITY or FITNESS FOR A
 # PARTICULAR PURPOSE.
 
-@SET_MAKE@
+	@SET_MAKE@
 VPATH = @srcdir@
 pkgdatadir = $(datadir)/@PACKAGE@
 pkgincludedir = $(includedir)/@PACKAGE@
@@ -514,7 +514,7 @@ uninstall-am:
 	maintainer-clean-generic mostlyclean mostlyclean-generic \
 	mostlyclean-libtool pdf pdf-am ps ps-am uninstall uninstall-am
 
-@YELP_HELP_RULES@
+	@YELP_HELP_RULES@
 
 -include $(top_srcdir)/git.mk
 
Index: anjuta-3.6.1/manuals/anjuta-faqs/Makefile.am
===================================================================
--- anjuta-3.6.1.orig/manuals/anjuta-faqs/Makefile.am
+++ anjuta-3.6.1/manuals/anjuta-faqs/Makefile.am
@@ -1,4 +1,4 @@
-@YELP_HELP_RULES@
+	@YELP_HELP_RULES@
 
 HELP_ID = anjuta-faqs
 
Index: anjuta-3.6.1/manuals/anjuta-build-tutorial/Makefile.am
===================================================================
--- anjuta-3.6.1.orig/manuals/anjuta-build-tutorial/Makefile.am
+++ anjuta-3.6.1/manuals/anjuta-build-tutorial/Makefile.am
@@ -1,4 +1,4 @@
-@YELP_HELP_RULES@
+	@YELP_HELP_RULES@
 
 HELP_ID = anjuta-build-tutorial
 
Index: anjuta-3.6.1/manuals/anjuta-build-tutorial/Makefile.in
===================================================================
--- anjuta-3.6.1.orig/manuals/anjuta-build-tutorial/Makefile.in
+++ anjuta-3.6.1/manuals/anjuta-build-tutorial/Makefile.in
@@ -13,7 +13,7 @@
 # even the implied warranty of MERCHANTABILITY or FITNESS FOR A
 # PARTICULAR PURPOSE.
 
-@SET_MAKE@
+	@SET_MAKE@
 VPATH = @srcdir@
 pkgdatadir = $(datadir)/@PACKAGE@
 pkgincludedir = $(includedir)/@PACKAGE@
@@ -518,7 +518,7 @@ uninstall-am:
 	maintainer-clean-generic mostlyclean mostlyclean-generic \
 	mostlyclean-libtool pdf pdf-am ps ps-am uninstall uninstall-am
 
-@YELP_HELP_RULES@
+	@YELP_HELP_RULES@
 
 -include $(top_srcdir)/git.mk
 
