diff -Naur keepassx-0.4.1.orig/src/translations/keepassx-tr_TR.ts keepassx-0.4.1/src/translations/keepassx-tr_TR.ts
--- keepassx-0.4.1.orig/src/translations/keepassx-tr_TR.ts	2009-10-14 00:23:38.893047236 +0300
+++ keepassx-0.4.1/src/translations/keepassx-tr_TR.ts	2009-10-14 00:54:06.440047072 +0300
@@ -390,7 +390,7 @@
     </message>
     <message>
         <source>System Language</source>
-        <translation type="unfinished">Sistem Dili</translation>
+        <translation>Sistem Dili</translation>
     </message>
 </context>
 <context>
@@ -771,7 +771,7 @@
     </message>
     <message>
         <source>Repeat:</source>
-        <translation type="unfinished"></translation>
+        <translation>Tekrar:</translation>
     </message>
 </context>
 <context>
@@ -1060,34 +1060,34 @@
     </message>
     <message>
         <source>Use following character groups:</source>
-        <translation type="unfinished"></translation>
+        <translation>Aşağıdaki karakter gruplarını kullanın:</translation>
     </message>
     <message>
         <source>Custom</source>
-        <translation type="unfinished"></translation>
+        <translation>Özel</translation>
     </message>
     <message>
         <source>Use the following characters:</source>
-        <translation type="unfinished"></translation>
+        <translation>Aşağıdaki karakterleri kullanın:</translation>
     </message>
 </context>
 <context>
     <name>HelpDlg</name>
     <message>
         <source>Previous Page</source>
-        <translation type="unfinished"></translation>
+        <translation>Önceki Sayfa</translation>
     </message>
     <message>
         <source>Next Page</source>
-        <translation type="unfinished"></translation>
+        <translation>Sonraki Sayfa</translation>
     </message>
     <message>
         <source>First Page</source>
-        <translation type="unfinished"></translation>
+        <translation>İlk Sayfa</translation>
     </message>
     <message>
         <source>Help Contents</source>
-        <translation type="unfinished"></translation>
+        <translation>Yardım İçeriği</translation>
     </message>
 </context>
 <context>
@@ -1264,15 +1264,15 @@
     </message>
     <message>
         <source>Unable to initialize the twofish algorithm.</source>
-        <translation type="unfinished"></translation>
+        <translation>Twofish algoritması yüklenemiyor.</translation>
     </message>
     <message>
         <source>Unknown encryption algorithm.</source>
-        <translation type="unfinished"></translation>
+        <translation>Bilinmeyen şifreleme algoritması.</translation>
     </message>
     <message>
         <source>The database has been opened read-only.</source>
-        <translation type="unfinished"></translation>
+        <translation>Veritabanı salt okunur olarak açıldı.</translation>
     </message>
 </context>
 <context>
@@ -1599,11 +1599,14 @@
 This means that either someone else has opened the file or KeePassX crashed last time it opened the database.
 
 Do you want to open it anyway?</source>
-        <translation type="unfinished"></translation>
+        <translation>Açmaya çalıştığınız veritabanı kilitli
+Bu dosya başka bir uygulama tarafından açılmış olabilir ya da KeePassX düzgün kapatılmamış olabilir
+
+Yine de bu veritabanını açmak ister misiniz?</translation>
     </message>
     <message>
         <source>Couldn&apos;t create database lock file.</source>
-        <translation type="unfinished">Veritabanı kilit dosyası oluşturulamadı.</translation>
+        <translation>Veritabanı kilit dosyası oluşturulamadı.</translation>
     </message>
     <message>
         <source>The current file was modified.
@@ -1617,15 +1620,15 @@
     </message>
     <message>
         <source>Open read-only</source>
-        <translation type="unfinished"></translation>
+        <translation>Salt okunur olarak aç</translation>
     </message>
     <message>
         <source>Couldn&apos;t create lock file. Opening the database read-only.</source>
-        <translation type="unfinished"></translation>
+        <translation>Veritabanı kilit dosyası oluşturulamadı. Veritabanı salt okunur olarak açılıyor.</translation>
     </message>
     <message>
         <source>locked</source>
-        <translation type="unfinished"></translation>
+        <translation>kilitli</translation>
     </message>
 </context>
 <context>
@@ -1915,19 +1918,19 @@
     </message>
     <message>
         <source>&amp;Entries</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Girdiler</translation>
     </message>
     <message>
         <source>&amp;Groups</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Gruplar</translation>
     </message>
     <message>
         <source>Sort groups</source>
-        <translation type="unfinished"></translation>
+        <translation>Grupları sırala</translation>
     </message>
     <message>
         <source>&amp;Group</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Grup</translation>
     </message>
 </context>
 <context>
@@ -2278,7 +2281,7 @@
     </message>
     <message>
         <source>You can disable several features of KeePassX here according to your needs in order to keep the user interface slim.</source>
-        <translation type="unfinished"></translation>
+        <translation>Arayüzü sadeleştirmek için bir çok KeePassX özelliğini ihtiyaçlarınıza göre kapatabilirsiniz.</translation>
     </message>
     <message>
         <source>Bookmarks</source>
@@ -2318,7 +2321,7 @@
     </message>
     <message>
         <source>Time between the activation of an auto-type action by the user and the first simulated key stroke.</source>
-        <translation type="unfinished"></translation>
+        <translation>Otomatik yazma eylemi aktivasyonu ve ilk olarak simüle edilen anahtar çifti arasındaki süre.</translation>
     </message>
     <message>
         <source>ms</source>
@@ -2334,7 +2337,7 @@
     </message>
     <message>
         <source>Delay between two simulated key strokes. Increase this if Auto-Type is randomly skipping characters.</source>
-        <translation type="unfinished"></translation>
+        <translation>Simüle edilen iki anahtar çifti arasındaki gecikme. Eğer Otomatik Yazma karakterleri düzensiz olarak atlıyorsa bu değeri arttırın.</translation>
     </message>
     <message>
         <source>Custom Browser Command</source>
@@ -2358,7 +2361,7 @@
     </message>
     <message>
         <source>Enable this if you want to use your bookmarks and the last opened file independet from their absolute paths. This is especially useful when using KeePassX portably and therefore with changing mount points in the file system.</source>
-        <translation type="unfinished"></translation>
+        <translation>Yer imlerinizi ve son kullanılan veritabanını son konumlarından bağımsız olarak kullanmak istiyorsanız bunu seçin. Özellikle KeePassX&apos;i taşınabilir olarak kullanıyorsanız ya da dosya sisteminizdeki bağlanma noktaları değişiyorsa yararlıdır.</translation>
     </message>
     <message>
         <source>Save relative paths (bookmarks and last file)</source>
@@ -2442,7 +2445,7 @@
     </message>
     <message>
         <source>Show window always on top</source>
-        <translation type="unfinished"></translation>
+        <translation>Pencereyi her zaman üstte göster</translation>
     </message>
 </context>
 <context>
@@ -2488,7 +2491,8 @@
     <message>
         <source>To specify the target window, either select an existing currently-opened window
 from the drop-down list, or enter the window title manually:</source>
-        <translation type="unfinished"></translation>
+        <translation>Hedef pencereyi seçmek için,açılır menüden şu anda açık olan bir pencereyi seçin
+ya da pencere başlığını girin:</translation>
     </message>
 </context>
 <context>
