Index: frostwire-5.1.4.noarch/frostwire.desktop
===================================================================
--- frostwire-5.1.4.noarch.orig/frostwire.desktop
+++ frostwire-5.1.4.noarch/frostwire.desktop
@@ -1,9 +1,11 @@
 [Desktop Entry]
 Name=FrostWire
-GenericName=P2P Gnutella/Bittorrent client
-Comment=Search and share all kinds of files on the Gnutella network
+GenericName=Bittorrent client
+GenericName[tr]=Bittorrent istemcisi
+Comment=Search and share all kinds of files using BitTorrent
+Comment[tr]=BitTorrent kullanarak her türlü dosyayı bulun ve paylaşın
 Exec=/usr/bin/frostwire
-Icon=frostwire.png
+Icon=frostwire
 Terminal=false
 Type=Application
-Categories=Application;Network;
\ Dosya sonunda yenisatır yok.
+Categories=Network;P2P;
