--- frostwire.desktop.OSmanOS	2013-02-28 20:36:48.000000000 +0200
+++ frostwire.desktop	2013-09-15 10:15:31.191880197 +0300
@@ -1,14 +1,16 @@
 [Desktop Entry]
 Version=1.0
 Name=FrostWire
-GenericName=P2P Bittorrent client
-GenericName[es]=P2P cliente Bittorrent
+GenericName=Bittorrent client
+GenericName[tr]=Bittorrent istemcisi
+Comment=Search and share all kinds of files using BitTorrent
+Comment[tr]=BitTorrent kullanarak her türlü dosyayı bulun ve paylaşın
 Comment=Search and explore all kinds of files on the Bittorrent network
 Comment[es]=Busque y explore todo tipo de archivos en la red Bittorrent
 Exec=/usr/bin/frostwire %u
 Icon=frostwire
 Terminal=false
 Type=Application
-Categories=Application;Network;
+Categories=Network;P2P;
 MimeType=application/x-bittorrent;x-scheme-handler/magnet;
 
