diff -Nur ufoai-2.3-source-old/src/client/cl_language.c ufoai-2.3-source/src/client/cl_language.c
--- ufoai-2.3-source-old/src/client/cl_language.c	2010-08-16 10:09:40.053267768 +0300
+++ ufoai-2.3-source/src/client/cl_language.c	2010-08-16 10:10:45.813907956 +0300
@@ -162,7 +162,7 @@
 #ifdef LOCALEDIR
 		Com_sprintf(languagePath, sizeof(languagePath), LOCALEDIR);
 #else
-		Com_sprintf(languagePath, sizeof(languagePath), "%s/"BASEDIRNAME"/i18n/", FS_GetCwd());
+		Com_sprintf(languagePath, sizeof(languagePath), "%s/"BASEDIRNAME"/i18n/", "/usr/share/ufoai");
 #endif
 	Com_DPrintf(DEBUG_CLIENT, "CL_LanguageTest: using mo files from '%s'\n", languagePath);
 	Q_strcat(languagePath, localeID, sizeof(languagePath));
diff -Nur ufoai-2.3-source-old/src/client/cl_main.c ufoai-2.3-source/src/client/cl_main.c
--- ufoai-2.3-source-old/src/client/cl_main.c	2010-08-16 10:09:40.057267607 +0300
+++ ufoai-2.3-source/src/client/cl_main.c	2010-08-16 10:11:38.392900605 +0300
@@ -1114,7 +1114,7 @@
 	if (fs_i18ndir->string[0] != '\0')
 		Q_strncpyz(languagePath, fs_i18ndir->string, sizeof(languagePath));
 	else
-		Com_sprintf(languagePath, sizeof(languagePath), "%s/"BASEDIRNAME"/i18n/", FS_GetCwd());
+		Com_sprintf(languagePath, sizeof(languagePath), "%s/"BASEDIRNAME"/i18n/", "/usr/share/ufoai");
 	Com_DPrintf(DEBUG_CLIENT, "...using mo files from %s\n", languagePath);
 	bindtextdomain(TEXT_DOMAIN, languagePath);
 	bind_textdomain_codeset(TEXT_DOMAIN, "UTF-8");
