diff -Nur ufoai-2.3-source-old/src/common/common.h ufoai-2.3-source/src/common/common.h
--- ufoai-2.3-source-old/src/common/common.h	2010-08-16 09:23:19.929268115 +0300
+++ ufoai-2.3-source/src/common/common.h	2010-08-16 09:23:47.335022917 +0300
@@ -43,7 +43,7 @@
 #  endif
 
 #elif defined __linux__
-#  define BUILDSTRING_OS "Linux"
+#  define BUILDSTRING_OS "Pardus Linux RELEASE"
 #  ifndef SHARED_EXT
 #    define SHARED_EXT "so"
 #  endif
