diff -Nur ufoai-2.3.1-source-old/src/client/cl_main.c ufoai-2.3.1-source/src/client/cl_main.c
--- ufoai-2.3.1-source-old/src/client/cl_main.c	2011-01-17 21:55:38.058940096 +0200
+++ ufoai-2.3.1-source/src/client/cl_main.c	2011-01-17 22:07:14.467641549 +0200
@@ -1100,6 +1100,7 @@
 	/* i18n through gettext */
 	char languagePath[MAX_OSPATH];
 	cvar_t *fs_i18ndir;
+	char *localetmp;
 
 	if (sv_dedicated->integer)
 		return;					/* nothing running on the client */
@@ -1109,7 +1110,9 @@
 	fs_i18ndir = Cvar_Get("fs_i18ndir", "", 0, "System path to language files");
 	/* i18n through gettext */
 	setlocale(LC_ALL, "C");
-	setlocale(LC_MESSAGES, "");
+	localetmp = setlocale(LC_MESSAGES, "");
+	if (strncmp(localetmp, "tr_TR.UTF-8", strlen("tr_TR.UTF-8")) == 0)
+		setlocale(LC_MESSAGES, "en_US.UTF-8");
 	/* use system locale dir if we can't find in gamedir */
 	if (fs_i18ndir->string[0] != '\0')
 		Q_strncpyz(languagePath, fs_i18ndir->string, sizeof(languagePath));
diff -Nur ufoai-2.3.1-source-old/src/ports/unix/unix_main.c ufoai-2.3.1-source/src/ports/unix/unix_main.c
--- ufoai-2.3.1-source-old/src/ports/unix/unix_main.c	2011-01-17 21:55:37.988940376 +0200
+++ ufoai-2.3.1-source/src/ports/unix/unix_main.c	2011-01-17 22:10:39.044175185 +0200
@@ -161,7 +161,7 @@
 {
 	/* Calling with NULL param should return current system settings. */
 	const char *currentLocale = setlocale(LC_MESSAGES, NULL);
-	if (currentLocale != NULL && currentLocale[0] != '\0')
+	if (currentLocale != NULL && currentLocale[0] != '\0' && strncmp(currentLocale, "tr_TR.UTF-8", strlen("tr_TR.UTF-8") != 0))
 		return currentLocale;
 	else
 		return "C";
