diff -up digikam-3.5.0/extra/kipi-plugins/panorama/importwizard/lastpage.cpp.panorama-crasher digikam-3.5.0/extra/kipi-plugins/panorama/importwizard/lastpage.cpp
--- digikam-3.5.0/extra/kipi-plugins/panorama/importwizard/lastpage.cpp.panorama-crasher	2013-10-01 08:23:09.000000000 +0200
+++ digikam-3.5.0/extra/kipi-plugins/panorama/importwizard/lastpage.cpp	2013-12-12 11:50:14.199493306 +0100
@@ -91,11 +91,7 @@ LastPage::LastPage(Manager* const mngr,
 
     QLabel *fileTemplateLabel = new QLabel(i18n("File name template:"), d->saveSettingsGroupBox);
     formatVBox->addWidget(fileTemplateLabel);
-    QString first = d->mngr->itemsList().front().fileName();
-    QString last = d->mngr->itemsList().back().fileName();
-//     QString file = "panorama";
-    QString file = QString("%1-%2").arg(first.left(first.lastIndexOf('.'))).arg(last.left(last.lastIndexOf('.')));
-    d->fileTemplateKLineEdit  = new KLineEdit(file, d->saveSettingsGroupBox);
+    d->fileTemplateKLineEdit  = new KLineEdit("panorama", d->saveSettingsGroupBox);
     d->fileTemplateKLineEdit->setToolTip(i18n("Name of the panorama file (without its extension)."));
     d->fileTemplateKLineEdit->setWhatsThis(i18n("<b>File name template</b>: Set here the base name of the files that "
                                                 "will be saved. For example, if your template is <i>panorama</i> and if "
@@ -146,6 +142,11 @@ LastPage::~LastPage()
 
 void LastPage::resetTitle()
 {
+    QString first = d->mngr->itemsList().front().fileName();
+    QString last = d->mngr->itemsList().back().fileName();
+    QString file = QString("%1-%2").arg(first.left(first.lastIndexOf('.'))).arg(last.left(last.lastIndexOf('.')));
+    d->fileTemplateKLineEdit->setText(file);
+
     slotTemplateChanged(d->fileTemplateKLineEdit->text());
     checkFiles();
 }
