Index: digikam-1.6.0/cmake/modules/FindCLAPACK.cmake
===================================================================
--- digikam-1.6.0.orig/cmake/modules/FindCLAPACK.cmake
+++ digikam-1.6.0/cmake/modules/FindCLAPACK.cmake
@@ -17,11 +17,11 @@ if(CLAPACK_INCLUDE_DIR AND CLAPACK_LIBRA
 
 else(CLAPACK_INCLUDE_DIR AND CLAPACK_LIBRARY)
 
-  find_path(CLAPACK_INCLUDE_DIR clapack.h PATHS /usr/include/clapack NO_DEFAULT_PATH)
+  find_path(CLAPACK_INCLUDE_DIR NAMES f2c.h  PATHS /usr/include/clapack NO_DEFAULT_PATH)
   if(CLAPACK_INCLUDE_DIR)
      message(STATUS "Found clapack includes: ${CLAPACK_INCLUDE_DIR}")
   else(CLAPACK_INCLUDE_DIR)
-     find_path(CLAPACK_INCLUDE_DIR clapack.h PATHS /usr/include/clapack)
+     find_path(CLAPACK_INCLUDE_DIR NAMES f2c.h PATHS /usr/include/clapack)
      if(CLAPACK_INCLUDE_DIR)
         message(STATUS "Found clapack includes: ${CLAPACK_INCLUDE_DIR}")
      endif(CLAPACK_INCLUDE_DIR)
