diff -Naur libshout-2.2.2.orig/src/Makefile.am libshout-2.2.2/src/Makefile.am
--- libshout-2.2.2.orig/src/Makefile.am	2006-01-31 03:42:42.000000000 +0200
+++ libshout-2.2.2/src/Makefile.am	2008-10-08 17:51:20.000000000 +0300
@@ -26,7 +26,7 @@
 AM_CFLAGS = @XIPH_CFLAGS@
 
 libshout_la_LIBADD = net/libicenet.la timing/libicetiming.la avl/libiceavl.la\
-		httpp/libicehttpp.la $(MAYBE_THREAD_LIB) $(THEORA_LIBS) $(VORBIS_LIBS)
+		httpp/libicehttpp.la $(MAYBE_THREAD_LIB) $(THEORA_LIBS) $(VORBIS_LIBS) $(SPEEX_LIBS)
 
 INCLUDES = -I$(top_builddir)/include
 
diff -Naur libshout-2.2.2.orig/src/Makefile.in libshout-2.2.2/src/Makefile.in
--- libshout-2.2.2.orig/src/Makefile.in	2006-06-19 23:36:32.000000000 +0300
+++ libshout-2.2.2/src/Makefile.in	2008-10-08 17:52:02.000000000 +0300
@@ -261,7 +261,7 @@
 libshout_la_SOURCES = shout.c util.c ogg.c vorbis.c mp3.c $(MAYBE_THEORA) $(MAYBE_SPEEX)
 AM_CFLAGS = @XIPH_CFLAGS@
 libshout_la_LIBADD = net/libicenet.la timing/libicetiming.la avl/libiceavl.la\
-		httpp/libicehttpp.la $(MAYBE_THREAD_LIB) $(THEORA_LIBS) $(VORBIS_LIBS)
+		httpp/libicehttpp.la $(MAYBE_THREAD_LIB) $(THEORA_LIBS) $(VORBIS_LIBS) $(SPEEX_LIBS)
 
 INCLUDES = -I$(top_builddir)/include
 all: all-recursive
@@ -324,7 +324,7 @@
 	  echo "rm -f \"$${dir}/so_locations\""; \
 	  rm -f "$${dir}/so_locations"; \
 	done
-libshout.la: $(libshout_la_OBJECTS) $(libshout_la_DEPENDENCIES) 
+libshout.la: $(libshout_la_OBJECTS) $(libshout_la_DEPENDENCIES)
 	$(LINK) -rpath $(libdir) $(libshout_la_LDFLAGS) $(libshout_la_OBJECTS) $(libshout_la_LIBADD) $(LIBS)
 
 mostlyclean-compile:
