--- lximage-qt.desktope.in	2015-02-18 02:27:45.000000000 +0200
+++ data/lximage-qt.desktop.in	2015-04-14 00:23:04.235455873 +0300
@@ -1,11 +1,13 @@
 [Desktop Entry]
 Type=Application
 Name=LXImage
+Name[tr]=LXImage Resim Görüntüleyici
 GenericName=Image Viewer
 Comment=The LXQt image viewer
+Comment[tr]=LXQt resim görüntüleyici
 Icon=lximage-qt
 Exec=lximage-qt
 StartupNotify=true
 Terminal=false
-Categories=Graphics;Utility;Core;Qt;Viewer;RasterGraphics;2DGraphics;Photography;
+Categories=Graphics;Qt;
 MimeType=image/bmp;image/gif;image/jpeg;image/jpg;image/png;image/tiff;image/x-bmp;image/x-pcx;image/x-tga;image/x-portable-pixmap;image/x-portable-bitmap;image/x-targa;image/x-portable-greymap;application/pcx;image/svg+xml;image/svg-xml;
