--- trackballs-1.1.4/share/icons/Makefile.in~	2007-06-05 18:57:33.000000000 +0200
+++ trackballs-1.1.4/share/icons/Makefile.in	2007-06-05 18:57:33.000000000 +0200
@@ -374,12 +374,12 @@
 	echo "Exec=$(bindir)/trackballs" >> trackballs.desktop
 
 install-iconsDATA:
-	$(mkinstalldirs) $(iconsdir)/32x32/apps
-	$(INSTALL_DATA) $(srcdir)/trackballs-32x32.png $(iconsdir)/32x32/apps/trackballs.png
-	$(mkinstalldirs) $(iconsdir)/48x48/apps
-	$(INSTALL_DATA) $(srcdir)/trackballs-48x48.png $(iconsdir)/48x48/apps/trackballs.png
-	$(mkinstalldirs) $(iconsdir)/64x64/apps
-	$(INSTALL_DATA) $(srcdir)/trackballs-64x64.png $(iconsdir)/64x64/apps/trackballs.png
+	$(mkinstalldirs) $(DESTDIR)$(iconsdir)/32x32/apps
+	$(INSTALL_DATA) $(srcdir)/trackballs-32x32.png $(DESTDIR)$(iconsdir)/32x32/apps/trackballs.png
+	$(mkinstalldirs) $(DESTDIR)$(iconsdir)/48x48/apps
+	$(INSTALL_DATA) $(srcdir)/trackballs-48x48.png $(DESTDIR)$(iconsdir)/48x48/apps/trackballs.png
+	$(mkinstalldirs) $(DESTDIR)$(iconsdir)/64x64/apps
+	$(INSTALL_DATA) $(srcdir)/trackballs-64x64.png $(DESTDIR)$(iconsdir)/64x64/apps/trackballs.png
 
 uninstall-iconsDATA:
 	-rm -f $(iconsdir)/32x32/apps/trackballs.png
--- trackballs-1.1.4/po/Makefile.in.in~	2007-06-05 19:31:45.000000000 +0200
+++ trackballs-1.1.4/po/Makefile.in.in	2007-06-05 19:31:45.000000000 +0200
@@ -27,7 +27,7 @@
 
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
-MKINSTALLDIRS = @MKINSTALLDIRS@
+MKINSTALLDIRS = $(SHELL) $(top_srcdir)/mkinstalldirs
 mkinstalldirs = ../mkinstalldirs
 ###$(top_builddir)/./mkinstalldirs
 ### mkinstalldirs = $(SHELL) `case "$(MKINSTALLDIRS)" in /*) echo "$(MKINSTALLDIRS)" ;; *) echo "$(top_builddir)/$(MKINSTALLDIRS)" ;; esac`
