--- configure.ac~	2009-05-11 23:30:56.613365223 +0300
+++ configure.ac	2009-05-11 23:30:56.614364505 +0300
@@ -114,8 +114,6 @@
 ])
 AC_SUBST(CCACHE_DISABLE)
 
-# check for gtk-doc
-GTK_DOC_CHECK(1.9)
 AC_CONFIG_MACRO_DIR(m4)
 
 AC_OUTPUT
--- docs/reference/libnice/Makefile.am~ 2009-05-11 23:33:58.339370823 +0300
+++ docs/reference/libnice/Makefile.am  2009-05-11 23:33:58.343367633 +0300
@@ -82,11 +82,11 @@
 
 
 # This includes the standard gtk-doc make rules, copied by gtkdocize.
-include $(top_srcdir)/gtk-doc.make
+# include $(top_srcdir)/gtk-doc.make
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
-EXTRA_DIST +=
+EXTRA_DIST =
 
 # Files not to distribute
 # for --rebuild-types in $(SCAN_OPTIONS), e.g. $(DOC_MODULE).types
