diff -up kdenetwork-4.2.98/krdc/CMakeLists.txt.me kdenetwork-4.2.98/krdc/CMakeLists.txt
--- kdenetwork-4.2.98/krdc/CMakeLists.txt.me	2009-06-17 22:08:12.000000000 +0200
+++ kdenetwork-4.2.98/krdc/CMakeLists.txt	2009-07-24 14:26:34.000000000 +0200
@@ -73,6 +73,8 @@ kde4_add_ui_files(krdc_SRCS
     config/general.ui
 )
 
+kde4_add_app_icon(krdc_SRCS "hi*-apps-krdc.png")
+
 kde4_add_executable(krdc ${krdc_SRCS})
 
 target_link_libraries(krdc
@@ -93,3 +95,5 @@ install(FILES pointcursor.png pointcurso
 if(NOT INSIDE_KDENETWORK)
     macro_display_feature_log()
 endif(NOT INSIDE_KDENETWORK)
+
+kde4_install_icons( ${ICON_INSTALL_DIR}   )
