diff -urNad ghostscript-8.64.dfsg.1~svn9323~/base/Makefile.in ghostscript-8.64.dfsg.1~svn9323/base/Makefile.in
--- ghostscript-8.64.dfsg.1~svn9323~/base/Makefile.in	2009-01-05 19:08:07.000000000 +0900
+++ ghostscript-8.64.dfsg.1~svn9323/base/Makefile.in	2009-01-05 19:43:07.000000000 +0900
@@ -71,8 +71,8 @@
 gssharedir = @libdir@/ghostscript/$(GS_DOT_VERSION)
 gsincludedir = @includedir@/ghostscript/
 
-docdir=$(gsdatadir)/doc
-exdir=$(gsdatadir)/examples
+docdir=$(prefix)/share/doc/ghostscript
+exdir=$(prefix)/share/doc/ghostscript/examples
 GS_DOCDIR=$(docdir)
 
 # Define install location for 'cups' device/filter support
