Author: Bo Yang <boyang@novell.com>
Subject: Handle child smbd crash
Upstream: No
Index: source3/passdb/lookup_sid.c
===================================================================
--- a/source3/passdb/lookup_sid.c.orig
+++ b/source3/passdb/lookup_sid.c
@@ -423,9 +423,17 @@ static bool wb_lookup_rids(TALLOC_CTX *m
 
 	if (!winbind_lookup_rids(tmp_ctx, domain_sid, num_rids, rids,
 				 domain_name, &my_names, &my_types)) {
-		*domain_name = "";
+		*domain_name = talloc_strdup(mem_ctx, "");
+		if (!(*domain_name)) {
+			TALLOC_FREE(tmp_ctx);
+			return False;
+		}
 		for (i=0; i<num_rids; i++) {
-			names[i] = "";
+			names[i] = talloc_strdup(names, "");
+			if (!names[i]) {
+				TALLOC_FREE(tmp_ctx);
+				return False;
+			}
 			types[i] = SID_NAME_UNKNOWN;
 		}
 		TALLOC_FREE(tmp_ctx);
