Index: xvidcap.desktop
===================================================================
--- xvidcap.desktop.orig
+++ xvidcap.desktop
@@ -1,12 +1,13 @@
 [Desktop Entry]
 Type=Application
-Encoding=UTF-8
-Name=XVidCap Screen Capture
+Name=XVidCap
 GenericName=X11 Screencam
+GenericName[tr]=Görüntü Yakalama Aracı
 Comment=X11 Screencam
 Comment[de]=X11 Videoaufnahme
-Icon=xvidcap.png
+Comment[tr]=Görüntü Yakalama Aracı
+Icon=xvidcap
 TryExec=xvidcap
 Exec=xvidcap
 Terminal=false
-Categories=GTK;Application;AudioVideo;Video;
+Categories=GTK;AudioVideo;Video;
