--- Makefile.am	2007-04-25 23:07:22.000000000 +0300
+++ Makefile.am	2007-04-25 23:19:02.000000000 +0300
@@ -6,10 +6,6 @@
 	po \
 	debian

-pkgdata_DATA = xvidcap.desktop \
-	xvidcap.png \
-    ppm2mpeg.sh
-
 xvidcapdocdir = ${datadir}/doc/xvidcap
 xvidcapdoc_DATA = \
 	README\
--- Makefile.am	2007-04-25 23:26:45.000000000 +0300
+++ Makefile.am	2007-04-25 23:30:06.000000000 +0300
@@ -45,9 +45,6 @@
 	mkdir $(DESTDIR)$(datadir)/applications ; \
 	cp $(top_srcdir)/xvidcap.desktop $(DESTDIR)$(datadir)/applications/

-install-data-hook:
-	chmod 755 $(DESTDIR)$(datadir)/xvidcap/ppm2mpeg.sh
-
 uninstall-hook:
 	rm $(DESTDIR)$(datadir)/pixmaps/xvidcap.png ; \
 	rm $(DESTDIR)$(datadir)/applications/xvidcap.desktop ; \
--- Makefile.am	2007-04-25 23:26:45.000000000 +0300
+++ Makefile.am	2007-04-25 23:34:42.000000000 +0300
@@ -44,6 +44,8 @@
 	cp $(top_srcdir)/xvidcap.png $(DESTDIR)$(datadir)/pixmaps/ ; \
 	mkdir $(DESTDIR)$(datadir)/applications ; \
 	cp $(top_srcdir)/xvidcap.desktop $(DESTDIR)$(datadir)/applications/
+	cp $(top_srcdir)/ppm2mpeg.sh $(DESTDIR)$(bindir)
+	chmod 755 $(DESTDIR)$(bindir)/ppm2mpeg.sh
 
 install-data-hook:
 	chmod 755 $(DESTDIR)$(datadir)/xvidcap/ppm2mpeg.sh
