--- xvidcap-1.1.7/configure.in.2	2008-06-13 00:25:11.000000000 +0300
+++ xvidcap-1.1.7/configure.in	2009-07-12 00:50:28.000000000 +0300
@@ -352,7 +352,7 @@
 
 test x$ac_cv_lib_mp3lame_lame_init = xyes && ac_my_ffmpeg_cfg_lib_switch="--enable-libmp3lame"
 dnl test x$ac_cv_lib_vorbisenc_vorbis_encode_init = xyes && ac_my_ffmpeg_cfg_lib_switch="${ac_my_ffmpeg_cfg_lib_switch} --enable-libvorbis"
-test x$ac_cv_lib_theora_theora_encode_init = xyes && ac_my_ffmpeg_cfg_lib_switch="${ac_my_ffmpeg_cfg_lib_switch} --enable-libogg --enable-libtheora"
+test x$ac_cv_lib_theora_theora_encode_init = xyes && ac_my_ffmpeg_cfg_lib_switch="${ac_my_ffmpeg_cfg_lib_switch} --enable-libtheora"
 
 # if cache empty (not tested or failed) and want static avcodec (then it failed) ...
 if ( `test \( \( x${ac_cv_lib_avutil_av_rescale+set} = x \) -a \( x$static_avutil = xyes \) \)` ) ; then
