--- configure.in~	2013-01-19 20:20:37.462750378 +0800
+++ configure.in	2013-01-19 20:21:05.023470697 +0800
@@ -5,7 +5,7 @@
 #########################################################
 AC_INIT(configure.in)
 AM_INIT_AUTOMAKE(xvidcap, 1.1.7)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AM_MAINTAINER_MODE
 
 IT_PROG_INTLTOOL
@@ -14,7 +14,6 @@
 
 AC_ISC_POSIX
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_HEADER_STDC
 AC_PROG_MAKE_SET
 AC_PROG_RANLIB
