diff -p -up ./scripts/xinitrc.in.in.0000 ./scripts/xinitrc.in.in
--- ./scripts/xinitrc.in.in.0000	2012-03-30 18:48:01.000000000 +0000
+++ ./scripts/xinitrc.in.in	2012-04-01 19:14:14.875368141 +0000
@@ -44,6 +44,23 @@ if which xdg-user-dirs-update >/dev/null
     xdg-user-dirs-update
 fi
 
+# Create Thunar user preferences if they are missing
+for i in Thunar; do
+     if [ ! -d $XDG_CONFIG_HOME/$i ]; then
+        mkdir -p $XDG_CONFIG_HOME/$i
+        cp -r /etc/xdg/$i $XDG_CONFIG_HOME
+     fi ;
+done
+
+
+# Create panel and xfconf user preferences if they are missing
+for i in panel xfconf terminal; do
+     if [ ! -d $XDG_CONFIG_HOME/xfce4/$i ]; then
+        mkdir -p $XDG_CONFIG_HOME/xfce4/$i
+        cp -r /etc/xdg/xfce4/$i $XDG_CONFIG_HOME/xfce4
+     fi ;
+done
+
 # Modify libglade and glade environment variables so that
 # it will find the files installed by Xfce
 GLADE_CATALOG_PATH="$GLADE_CATALOG_PATH:@XFCE_GLADE_CATALOG_PATH@"
 
