From 499a719019e03da90b6901bb0f8a89d06ae167ee Mon Sep 17 00:00:00 2001
From: Nick Schermer <nick@xfce.org>
Date: Sun, 22 Jul 2012 09:44:49 +0000
Subject: Fix logic of the --fast action (bug #8857).

---
diff --git a/xfce4-session-logout/main.c b/xfce4-session-logout/main.c
index 5af54c6..19a5c9a 100644
--- a/xfce4-session-logout/main.c
+++ b/xfce4-session-logout/main.c
@@ -48,7 +48,7 @@ gboolean opt_halt = FALSE;
 gboolean opt_reboot = FALSE;
 gboolean opt_suspend = FALSE;
 gboolean opt_hibernate = FALSE;
-gboolean allow_save = FALSE;
+gboolean opt_fast = FALSE;
 gboolean opt_version = FALSE;
 
 enum
@@ -83,7 +83,7 @@ static GOptionEntry option_entries[] =
     N_("Hibernate without displaying the logout dialog"),
     NULL
   },
-  { "fast", 'f', G_OPTION_FLAG_IN_MAIN, G_OPTION_ARG_NONE, &allow_save,
+  { "fast", 'f', G_OPTION_FLAG_IN_MAIN, G_OPTION_ARG_NONE, &opt_fast,
     N_("Log out quickly; don't save the session"),
     NULL
   },
@@ -123,6 +123,7 @@ main (int argc, char **argv)
   gboolean         show_dialog;
   gboolean         result = FALSE;
   guint            shutdown_type;
+  gboolean         allow_save;
 
   xfce_textdomain (GETTEXT_PACKAGE, PACKAGE_LOCALE_DIR, "UTF-8");
 
@@ -149,6 +150,9 @@ main (int argc, char **argv)
       return EXIT_FAILURE;
     }
 
+  /* save the session, unless fast is provided */
+  allow_save = !opt_fast;
+
   /* create messsage */
   proxy = dbus_g_proxy_new_for_name_owner (conn,
                                            "org.xfce.SessionManager",
--
cgit v0.9.0.3
 
