diff -urN lxmenu-data-0.1.1.orig/layout/lxde-applications.menu lxmenu-data-0.1.1/layout/lxde-applications.menu
--- lxmenu-data-0.1.1.orig/layout/lxde-applications.menu	2009-05-19 18:52:11.000000000 +0200
+++ lxmenu-data-0.1.1/layout/lxde-applications.menu	2010-05-14 13:56:00.000000000 +0200
@@ -19,13 +19,14 @@
 		<Directory>lxde-utility.directory</Directory>
 		<Include>
 			<And>
-				<Category>Utility</Category>
+			<Category>Utility</Category>
 	<!-- Accessibility spec must have either the Utility or Settings
 			 category, and we display an accessibility submenu already for
 			 the ones that do not have Settings, so don't display accessibility
 			 applications here -->
 				<Not><Category>Accessibility</Category></Not>
 				<Not><Category>System</Category></Not>
+				<Not><Category>Settings</Category></Not>
 			</And>
 		</Include>
 	</Menu> <!-- End Accessories -->
@@ -129,6 +130,7 @@
 			<And>
 				<Category>System</Category>
 				<Not><Category>Settings</Category></Not>
+				<Not><Category>SystemSetup</Category></Not>
 				<Not><Category>PackageManager</Category></Not>
 			</And>
 		</Include>
@@ -149,17 +151,56 @@
 		</Menu> <!-- End Other -->
 
 	<Menu>
-		<Name>DesktopSettings</Name>
-		<Directory>lxde-settings.directory</Directory>
-		<OnlyUnallocated/>
+		<Name>Settings</Name>
+		<Directory>lxde-menu-system.directory</Directory>
+		<Menu>
+			<Name>Preferences</Name>
+			<Directory>lxde-settings.directory</Directory>
+			<Include>
+				<And>
+					<Category>Settings</Category>
+					<Not>
+						<Or>
+							<Category>System</Category>
+							<Filename>gnomecc.desktop</Filename>
+							<Filename>lxcc.desktop</Filename>
+						</Or>
+					</Not>
+				</And>
+			</Include>
+		</Menu>
+
+		<!-- System Settings -->
+		<Menu>
+			<Name>Administration</Name>
+			<Directory>lxde-settings-system.directory</Directory>
+			<Include>
+				<And>
+					<Or>
+						<Category>PackageManager</Category>
+						<Category>SystemSetup</Category>
+						<And>
+							<Category>Settings</Category>
+							<Category>System</Category>
+						</And>
+					</Or>
+					<Not>
+						<Or>
+							<Category>X-SuSE-YaST</Category>
+							<Filename>lxcc.desktop</Filename>
+						</Or>
+					</Not>
+				</And>
+			</Include>
+		</Menu>		 <!-- End System Settings -->
+
 		<Include>
-			<Or>
-				<Category>Settings</Category>
-				<Category>PackageManager</Category>
-				<Category>System</Category>
-			</Or>
+			<Filename>lxcc.desktop</Filename>
 		</Include>
+
 		<Layout>
+			<Menuname>Preferences</Menuname>
+			<Menuname>Administration</Menuname>
 			<Merge type="menus"/>
 			<Merge type="files"/>
 		</Layout>
@@ -170,7 +211,7 @@
 		<Merge type="files"/>
 		<Merge type="menus"/>
 		<Separator/>
-		<Menuname>DesktopSettings</Menuname>
+		<Menuname>Settings</Menuname>
 	</Layout>
 
 </Menu> <!-- End Applications -->
