diff -up libqzeitgeist-0.8.0/src/CMakeLists.txt.linking libqzeitgeist-0.8.0/src/CMakeLists.txt
--- libqzeitgeist-0.8.0/src/CMakeLists.txt.linking	2011-10-27 09:08:58.000000000 -0500
+++ libqzeitgeist-0.8.0/src/CMakeLists.txt	2011-10-31 08:04:24.037330288 -0500
@@ -70,7 +70,7 @@ add_custom_command(OUTPUT ${CMAKE_CURREN
                    )
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/manifestation.h ${CMAKE_CURRENT_BINARY_DIR}/interpretation.h DESTINATION include/QZeitgeist)
 
-target_link_libraries(qzeitgeist ${QT_LIBRARIES} ${QDBUS_LDFLAGS})
+target_link_libraries(qzeitgeist ${QT_QTDBUS_LIBRARIES} ${QT_QTGUI_LIBRARIES} ${QDBUS_LDFLAGS})
 set_target_properties(qzeitgeist PROPERTIES VERSION "${QZEITGEIST_VERSION}"
                       VERSION ${QZEITGEIST_VERSION}
                       SOVERSION ${QZEITGEIST_LIB_SOVERSION}
