diff -up mhash-0.9.9.9/src/keygen_test.c.BAD mhash-0.9.9.9/src/keygen_test.c
--- mhash-0.9.9.9/src/keygen_test.c.BAD	2009-07-22 18:01:59.636042665 -0400
+++ mhash-0.9.9.9/src/keygen_test.c	2009-07-22 18:04:53.608292727 -0400
@@ -121,7 +121,7 @@ int main()
 	
 	mhash_keygen_ext(KEYGEN_S2K_SALTED, data, key, keysize, password, passlen);
 
-	mutils_memset(tmp, 0, keysize * 2);
+	// mutils_memset(tmp, 0, keysize * 2);
 
 	tmp = mutils_asciify(key, keysize);
 
