diff -urN a/mateconf/default.path.in b/mateconf/default.path.in
--- a/mateconf/default.path.in	2012-02-22 15:10:58.000000000 +0000
+++ b/mateconf/default.path.in	2012-06-19 11:46:03.583966454 +0100
@@ -5,31 +5,17 @@
 
 # See the MateConf manual for details
 
-# Look first in systemwide mandatory settings directory
-xml:readonly:@sysmateconfdir@/mateconf.xml.mandatory
+xml:readwrite:$(HOME)/.mateconf
 
-# To read in any mandatory settings that the Sys Admin may have created
-# prior to a desktop system upgrade. The SysAdmin can stick read-only system
-# wide sources in this file.
-include @sysmateconfdir@/2/local-mandatory.path
+xml:readonly:@sysmateconfdir@/mateconf.xml.defaults
 
-# Now see where users want us to look - basically the user can stick arbitrary 
-# sources in a ~/.mateconf.path file and they're inserted here
 include "$(HOME)/.mateconf.path"
 
-# Give users a default storage location, ~/.mateconf
-xml:readwrite:$(HOME)/.mateconf
+xml:readonly:@sysmateconfdir@/mateconf.xml.mandatory
+
+include @sysmateconfdir@/2/local-mandatory.path
 
-# Location for system-wide settings that are set by the defaults mechanism
 xml:readonly:/etc/mateconf/mateconf.xml.system
 
-# To read in any defaults settings that the Sys Admin may have created
-# prior to a desktop system upgrade. The SysAdmin can stick default values
-# system-wide in this file.
 include @sysmateconfdir@/2/local-defaults.path
 
-# Finally, look at the systemwide defaults
-xml:readonly:@sysmateconfdir@/mateconf.xml.defaults
-
-
-
 
