Index: qtemu-1.0.5/CMakeLists.txt
===================================================================
--- qtemu-1.0.5.orig/CMakeLists.txt
+++ qtemu-1.0.5/CMakeLists.txt
@@ -114,7 +114,7 @@ ADD_CUSTOM_TARGET(translations ALL
                    DEPENDS ${QM_FILES})
 
 INSTALL (FILES ${QM_FILES}
-         DESTINATION translations)
+         DESTINATION share/qtemu/translations)
 # translation end
 
 SET(QT_USE_QTXML)
@@ -146,13 +146,13 @@ TARGET_LINK_LIBRARIES(qtemu
 )
 
 FILE(GLOB helpfiles "${CMAKE_CURRENT_SOURCE_DIR}/help/*.png")
-INSTALL(FILES ${helpfiles} DESTINATION help)
+INSTALL(FILES ${helpfiles} DESTINATION share/qtemu/help)
 
-INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/help/main.htm DESTINATION help)
+INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/help/main.htm DESTINATION share/qtemu/help)
 
 FILE(GLOB helpfiles_de "${CMAKE_CURRENT_SOURCE_DIR}/help/de/*.png")
-INSTALL(FILES ${helpfiles_de} DESTINATION help/de)
+INSTALL(FILES ${helpfiles_de} DESTINATION share/qtemu/help/de)
 
-INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/help/de/main.htm DESTINATION help/de)
+INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/help/de/main.htm DESTINATION share/qtemu/help/de)
 
 INSTALL(TARGETS qtemu DESTINATION bin)
Index: qtemu-1.0.5/helpwindow.cpp
===================================================================
--- qtemu-1.0.5.orig/helpwindow.cpp
+++ qtemu-1.0.5/helpwindow.cpp
@@ -66,21 +66,32 @@ HelpWindow::HelpWindow(QWidget *parent)
 QUrl HelpWindow::getHelpFile()
 {
     QSettings settings("QtEmu", "QtEmu");
-    QString locale = settings.value("language", QString(QLocale::system().name())).toString();
-    if (locale != "en")
+    QString locale = settings.value("language", QLocale::system().name().replace("_", "-")).toString();
+    QString language = locale.left(locale.indexOf("-"));
+
+    if (language != "en")
     { //check first if there is a language specific help available
         //check for case when qtemu executable is in same dir (windows)
-        QUrl testUrl("help/" + locale + "/main.htm");
+        QUrl testUrl("help/" + language + "/main.htm");
         if (QFile::exists(testUrl.toString()))
             return testUrl;
     
         //check for case when qtemu executable is in same dir (linux)
-        testUrl = QUrl(QCoreApplication::applicationDirPath()+"/help/" + locale + "/main.htm");
+        testUrl = QUrl(QCoreApplication::applicationDirPath()+"/help/" + language + "/main.htm");
         if (QFile::exists(testUrl.toString()))
             return testUrl;
     
         //check for case when qtemu executable is in bin/ (installed on linux)
-        testUrl = QUrl(QCoreApplication::applicationDirPath()+"/../help/" + locale + "/main.htm");
+        testUrl = QUrl(QCoreApplication::applicationDirPath()+"/../help/" + language + "/main.htm");
+        if (QFile::exists(testUrl.toString()))
+            return testUrl;
+
+        //check for case when FHS is supported (installed on linux)
+        testUrl = QUrl("/usr/share/qtemu/help/" + language + "/main.htm");
+        if (QFile::exists(testUrl.toString()))
+            return testUrl;
+
+        testUrl = QUrl("/usr/share/qtemu/help/" + locale + "/main.htm");
         if (QFile::exists(testUrl.toString()))
             return testUrl;
     }
@@ -100,6 +111,11 @@ QUrl HelpWindow::getHelpFile()
     if (QFile::exists(testUrl.toString()))
         return testUrl;
 
+    //check for case when FHS is supported (installed on linux)
+    testUrl = QUrl("/usr/share/qtemu/help/main.htm");
+    if (QFile::exists(testUrl.toString()))
+        return testUrl;
+
     //qDebug(testUrl.toString().toLocal8Bit().constData());
     QMessageBox::critical(this, tr("Help not found"),
                                 tr("Help not found. It is probably not installed."));
Index: qtemu-1.0.5/main.cpp
===================================================================
--- qtemu-1.0.5.orig/main.cpp
+++ qtemu-1.0.5/main.cpp
@@ -37,18 +37,32 @@ int main(int argc, char *argv[])
 
     //load translation
     QSettings settings("QtEmu", "QtEmu");
-    QString locale = settings.value("language", QString(QLocale::system().name())).toString();
+    QString locale = settings.value("language", QLocale::system().name()).toString();
+    QString language = locale.left(locale.indexOf("_"));
+    QString locale2 = locale.replace("_", "-");
+
     QTranslator translator;
     QString path;
     path = QCoreApplication::applicationDirPath()+"/translations/qtemu_" + locale + ".qm";
     if (QFile::exists(path))
         translator.load(path);
-    else
-    {
-        path = QCoreApplication::applicationDirPath()+"/../translations/qtemu_" + locale + ".qm";
-        if (QFile::exists(path))
-            translator.load(path);
-    }
+    path = QCoreApplication::applicationDirPath()+"/translations/qtemu_" + locale2 + ".qm";
+    if (QFile::exists(path))
+        translator.load(path);
+    path = QCoreApplication::applicationDirPath()+"/translations/qtemu_" + language + ".qm";
+    if (QFile::exists(path))
+        translator.load(path);
+    //check for case when FHS is supported (installed on linux)
+    path = "/usr/share/qtemu/translations/qtemu_" + locale + ".qm";
+    if (QFile::exists(path))
+        translator.load(path);
+    path = "/usr/share/qtemu/translations/qtemu_" + locale2 + ".qm";
+    if (QFile::exists(path))
+        translator.load(path);
+    path = "/usr/share/qtemu/translations/qtemu_" + language + ".qm";
+    if (QFile::exists(path))
+        translator.load(path);
+
     app.installTranslator(&translator);
 
     MainWindow mainWin;
