Index: libproxy-0.4.0/CMakeLists.txt
===================================================================
--- libproxy-0.4.0.orig/CMakeLists.txt
+++ libproxy-0.4.0/CMakeLists.txt
@@ -56,7 +56,7 @@ if (WIN32)
   add_definitions(-D_CRT_SECURE_NO_WARNINGS=1)
 else()
   add_definitions(-D_POSIX_C_SOURCE=1)
-  set(CMAKE_CXX_FLAGS "-g -Wall -Werror -fvisibility=hidden ${CMAKE_CXX_FLAGS}")
+  set(CMAKE_CXX_FLAGS "-fvisibility=hidden ${CMAKE_CXX_FLAGS}")
 endif()
 
 ### Subdirectories
