Index: libproxy-0.4.0/CMakeLists.txt
===================================================================
--- libproxy-0.4.0.orig/CMakeLists.txt
+++ libproxy-0.4.0/CMakeLists.txt
@@ -38,7 +38,7 @@ if(${WIN32})
 else()
   set(rlibdir lib${LIB_SUFFIX}/)
   set(rmoduledir  ${rlibdir}/${CMAKE_PROJECT_NAME}/${CMAKE_PROJECT_VERSION}/modules/)
-  set(rlibexecdir libexec/)
+  set(rlibexecdir lib/${CMAKE_PROJECT_NAME})
   set(rbindir     bin/)
   set(rincludedir include/)
 endif()
