--- /dev/null	2006-10-25 13:55:08.864481206 +0300
+++ veusz.desktop	2006-10-25 14:33:38.000000000 +0300
@@ -0,0 +1,12 @@
+[Desktop Entry]
+Name=Veusz
+Name[tr]=Veusz
+Comment=Vector graphics tool
+comment[tr]=Vektör grafik aracı
+GenericName=Vector graphics tool
+GenericName[tr]=Vektör grafik aracı
+Encoding=UTF-8
+Exec=veusz
+Icon=veusz.png
+Type=Application
+Categories=Application;Education;Science;


