--- lftp-3.7.7/lftp.conf.agent	2008-08-07 14:27:47.000000000 +0200
+++ lftp-3.7.7/lftp.conf	2009-01-12 14:21:24.000000000 +0100
@@ -13,7 +13,7 @@
 ## Uncomment the following two lines to make switch cls and ls, making
 ## cls the default.
 #alias ls command cls
-#alias hostls command ls
+alias hostls command ls
 
 ## default protocol selection
 #set default-protocol/ftp.*	ftp
@@ -31,7 +31,7 @@
 set auto-sync-mode "icrosoft FTP Service|MadGoat|MikroTik"
 
 ## if default ftp passive mode does not work, try this:
-# set ftp:passive-mode off
+set ftp:passive-mode on
 
 ## Set this to follow http redirections
 set xfer:max-redirections 10
@@ -87,4 +87,4 @@
 ## If you don't like advertising lftp or servers hate it, set this:
 # set ftp:anon-pass "mozilla@"
 # set ftp:client ""
-# set http:user-agent "Mozilla/4.7 [en] (WinNT; I)"
+# set http:user-agent "Mozilla/4.7 [en] (Linux; U)"
