diff -Naurp mate-desktop-1.6.0/schemas/org.mate.background.gschema.xml.in.in mate-desktop-1.6.0./schemas/org.mate.background.gschema.xml.in.in
--- mate-desktop-1.6.0/schemas/org.mate.background.gschema.xml.in.in	2013-04-01 03:27:50.000000000 +0300
+++ mate-desktop-1.6.0./schemas/org.mate.background.gschema.xml.in.in	2013-04-08 03:37:45.669260650 +0300
@@ -34,7 +34,7 @@
       <_description>Determines how the image set by wallpaper_filename is rendered.  Possible values are "wallpaper", "centered", "scaled", "stretched", "zoom", "spanned".</_description>
     </key>
     <key name="picture-filename" type="s">
-      <default>'@datadir@/backgrounds/mate/desktop/Stripes.png'</default>
+      <default>'@datadir@/backgrounds/mate/desktop/rengarenk.png'</default>
       <_summary>Picture Filename</_summary>
       <_description>File to use for the background image.</_description>
     </key>
