--- schemas/org.mate.interface.gschema.xml.in.in.OSmanOS	2014-01-01 14:54:50.000000000 +0200
+++ schemas/org.mate.interface.gschema.xml.in.in	        2014-02-07 19:25:22.809273460 +0200
@@ -61,17 +61,17 @@
       <_description>Length of the cursor blink cycle, in milliseconds.</_description>
     </key>
     <key name="icon-theme" type="s">
-      <default>'menta'</default>
+      <default>'matefaenzadark'</default>
       <_summary>Icon Theme</_summary>
       <_description>Icon theme to use for the panel, caja etc.</_description>
     </key>
     <key name="gtk-theme" type="s">
-      <default>'Menta'</default>
+      <default>'MediterraneanLightDarkest'</default>
       <_summary>Gtk+ Theme</_summary>
       <_description>Basename of the default theme used by gtk+.</_description>
     </key>
     <key name="gtk-key-theme" type="s">
-      <default>'Default'</default>
+      <default>'MediterraneanLightDarkest'</default>
       <_summary>Gtk+ Theme</_summary>
       <_description>Basename of the default theme used by gtk+.</_description>
     </key>
