diff -up libutempter-1.1.6/Makefile.pierelro libutempter-1.1.6/Makefile
--- libutempter-1.1.6/Makefile.pierelro	2010-11-04 18:14:53.000000000 +0100
+++ libutempter-1.1.6/Makefile	2013-01-24 15:58:06.241865564 +0100
@@ -52,11 +52,12 @@ all: $(TARGETS)
 	$(COMPILE.c) -fPIC $< $(OUTPUT_OPTION)
 
 $(PROJECT): utempter.c
-	$(LINK.c) -Wl,-z,now,-stats $(LDLIBS) $< $(OUTPUT_OPTION)
+	$(LINK.c) -pie -fPIE -Wl,-z,relro,-z,now,-stats $(LDLIBS) \
+		$< $(OUTPUT_OPTION)
 
 $(SHAREDLIB): iface.os $(MAP)
 	$(LINK.o) -shared \
-		-Wl,-soname,$(SONAME),--version-script=$(MAP),-z,defs,-stats \
+		-Wl,-soname,$(SONAME),--version-script=$(MAP),-z,relro,-z,defs,-stats \
 		-lc $< $(OUTPUT_OPTION)
 
 $(STATICLIB): iface.o
