--- w3c-libwww-5.3.2/Library/src/Makefile.am~	Sat Mar  2 06:40:12 2002
+++ w3c-libwww-5.3.2/Library/src/Makefile.am	Sat Mar  2 06:40:12 2002
@@ -47,8 +47,7 @@
 	perl $(top_srcdir)/config/winConfigure.pl $(srcdir)
     endif
 
-icondir = @icondir@
-CPPFLAGS := $(CPPFLAGS) -DW3C_ICONS=\"$(icondir)\"
+CPPFLAGS = @CPPFLAGS@ -DW3C_ICONS=\"$(icondir)\"
 
 inc: $(BUILT_SOURCES)
 	@echo Generated built sources
@@ -539,7 +539,7 @@
 
 pkginclude_HEADERS = $(BUILT_SOURCES)
 
-DOCS :=	$(wildcard *.html)
+DOCS =	$(wildcard *.html)
 
 EXTRA_DIST = \
 	$(BUILT_SOURCES) \
