--- w3c-libwww-5.4.0/Makefile.am.orig	2002-06-12 11:32:59.000000000 +0200
+++ w3c-libwww-5.4.0/Makefile.am	2003-10-23 00:32:42.756648688 +0200
@@ -42,7 +42,6 @@
 	chmod -R a+r $(distdir)
 	zip -rq $(distdir).zip $(distdir)
 	$(TAR) chofz $(distdir).tgz $(distdir)
-	# gzip -f $(distdir).tar
 
 # Tag before making distribution.  Also, don't make a distribution if
 # checks fail.
--- w3c-libwww-5.4.0/Library/User/Makefile.am.orig	2002-06-06 16:59:37.000000000 +0200
+++ w3c-libwww-5.4.0/Library/User/Makefile.am	2003-10-23 00:57:54.843776504 +0200
@@ -11,9 +11,6 @@
 
 SUBDIRS = Patch Architecture Using Guide Style Platform
 
-BUILT_SOURCES = \
-	Extrnals.html
-
 SUFFIXES = .html
 
 DEPENDENCIES = \
