--- w3c-libwww-5.4.0/Library/src/Makefile.am.orig	2007-04-10 19:22:56.706644924 +0200
+++ w3c-libwww-5.4.0/Library/src/Makefile.am	2007-04-10 19:23:14.659668008 +0200
@@ -6,23 +6,23 @@
 	libwwwutils.la \
 	libwwwcore.la \
 	libwwwtrans.la \
-	libwwwstream.la \
-	libwwwcache.la \
+	libwwwhtml.la \
 	libwwwdir.la \
 	libwwwfile.la \
+	libwwwstream.la \
+	libwwwcache.la \
 	libwwwftp.la \
 	libwwwgopher.la \
 	libwwwmime.la \
 	libwwwhttp.la \
 	libwwwnews.la \
 	libwwwtelnet.la \
-	libwwwhtml.la \
-	libwwwapp.la \
-	libwwwinit.la \
 	libwwwmux.la \
         @WWWDAV@ \
 	@WWWXML@ \
 	@WWWZIP@ \
+	libwwwapp.la \
+	libwwwinit.la \
 	@WWWWAIS@ \
 	@WWWSQL@
 
@@ -33,7 +33,7 @@
         libwwwdav.la \
 	libwwwsql.la
 
-SUBDIRS = SSL windows vms
+SUBDIRS = . SSL windows vms
 
 #	.h files are distributed but originally are made from the
 #	self-documenting hypertext files.
@@ -143,6 +143,9 @@
 	HTWWWStr.h \
 	HTWWWStr.c
 
+libwwwcore_la_LIBADD = \
+	libwwwutils.la
+
 libwwwtrans_la_SOURCES = \
 	WWWTrans.h \
 	HTANSI.h \
@@ -158,6 +161,8 @@
 	HTWriter.h \
 	HTWriter.c
 
+libwwwtrans_la_LIBADD = libwwwutils.la libwwwcore.la
+
 libwwwstream_la_SOURCES = \
 	WWWStream.h \
 	HTConLen.h \
@@ -181,11 +186,15 @@
 	HTXParse.h \
 	HTXParse.c
 
+libwwwstream_la_LIBADD = libwwwfile.la libwwwutils.la libwwwcore.la
+
 libwwwcache_la_SOURCES = \
 	WWWCache.h \
 	HTCache.h \
 	HTCache.c
 
+libwwwcache_la_LIBADD = libwwwutils.la libwwwcore.la libwwwtrans.la
+
 libwwwfile_la_SOURCES = \
 	WWWFile.h \
 	HTBInit.h \
@@ -197,6 +206,8 @@
 	HTMulti.h \
 	HTMulti.c
 
+libwwwfile_la_LIBADD = libwwwtrans.la libwwwdir.la libwwwutils.la libwwwcore.la
+
 libwwwftp_la_SOURCES = \
 	WWWFTP.h \
 	HTFTP.h \
@@ -204,11 +215,15 @@
 	HTFTPDir.h \
 	HTFTPDir.c
 
+libwwwftp_la_LIBADD = libwwwutils.la libwwwcore.la libwwwdir.la libwwwfile.la
+
 libwwwgopher_la_SOURCES = \
 	WWWGophe.h \
 	HTGopher.h \
 	HTGopher.c
 
+libwwwgopher_la_LIBADD = libwwwutils.la libwwwcore.la libwwwhtml.la libwwwdir.la
+
 libwwwmime_la_SOURCES = \
 	WWWMIME.h \
 	HTBound.h \
@@ -225,6 +240,8 @@
 	HTMIMImp.c \
 	HTMulpar.h
 
+libwwwmime_la_LIBADD = libwwwcache.la libwwwstream.la libwwwutils.la libwwwcore.la
+
 libwwwhttp_la_SOURCES = \
 	WWWHTTP.h \
 	HTTPUtil.h \
@@ -254,6 +271,9 @@
 libwwwhttp_la_DEPENDENCIES = \
 	../../modules/md5/libmd5.la
 
+libwwwhttp_la_LIBADD = \
+	../../modules/md5/libmd5.la libwwwmime.la libwwwutils.la libwwwcore.la
+
 libwwwnews_la_SOURCES = \
 	WWWNews.h \
 	HTNDir.h \
@@ -265,11 +285,15 @@
 	HTNewsRq.h \
 	HTNewsRq.c
 
+libwwwnews_la_LIBADD = libwwwstream.la libwwwutils.la libwwwcore.la libwwwmime.la
+
 libwwwtelnet_la_SOURCES = \
 	WWWTelnt.h \
 	HTTelnet.h \
 	HTTelnet.c
 
+libwwwtelnet_la_LIBADD = libwwwutils.la libwwwcore.la
+
 libwwwwais_la_SOURCES = \
 	WWWWAIS.h \
 	HTWAIS.h \
@@ -301,6 +325,8 @@
 	HTStyle.h \
 	HTStyle.c
 
+libwwwhtml_la_LIBADD = libwwwutils.la libwwwcore.la
+
 libwwwdir_la_SOURCES = \
 	WWWDir.h \
 	HTIcons.h \
@@ -310,6 +336,9 @@
 	HTDir.h \
 	HTDir.c
 
+libwwwdir_la_LIBADD = \
+	libwwwhtml.la libwwwutils.la libwwwcore.la
+
 libwwwapp_la_SOURCES = \
 	WWWApp.h \
 	HTAccess.h \
@@ -333,11 +362,18 @@
 	HTRules.h \
 	HTRules.c
 
+libwwwapp_la_LIBADD = \
+	libwwwmime.la libwwwdir.la libwwwhttp.la libwwwnews.la libwwwftp.la libwwwxml.la libwwwzip.la \
+	libwwwgopher.la libwwwtelnet.la libwwwstream.la
+
 libwwwinit_la_SOURCES = \
 	WWWInit.h \
 	HTProfil.h \
 	HTProfil.c
 
+libwwwinit_la_LIBADD = \
+	libwwwfile.la libwwwhtml.la libwwwapp.la
+
 libwwwmux_la_SOURCES = \
 	WWWMux.h \
 	HTDemux.h \
@@ -348,12 +384,16 @@
 	HTMuxTx.h \
 	HTMuxTx.c
 
+libwwwmux_la_LIBADD = libwwwutils.la libwwwstream.la libwwwcore.la
+
 libwwwdav_la_SOURCES = \
         HTDAV.h \
         HTDAV.c
 
 libwwwdav_la_LDFLAGS = -rpath $(libdir)
 
+libwwwdav_la_LIBADD = libwwwcore.la libwwwapp.la
+
 libwwwxml_la_SOURCES = \
 	WWWXML.h \
 	HTRDF.h \
@@ -363,7 +403,7 @@
 
 libwwwxml_la_LDFLAGS = -rpath $(libdir)
 
-libwwwxml_la_LIBADD = -lexpat
+libwwwxml_la_LIBADD = -lexpat libwwwcore.la
 
 libwwwzip_la_SOURCES = \
 	WWWZip.h \
@@ -371,6 +411,8 @@
 	HTZip.h \
 	HTZip.c
 
+libwwwzip_la_LIBADD = libwwwutils.la libwwwcore.la
+
 libwwwzip_la_LDFLAGS = -rpath $(libdir)
 
 libwwwsql_la_SOURCES = \
