diff -Nuar apolkit-0.107/configure.ac polkit-0.107/configure.ac
--- apolkit-0.107/configure.ac	2012-07-06 16:54:35.000000000 +0300
+++ polkit-0.107/configure.ac	2012-09-20 12:08:05.612322550 +0300
@@ -47,7 +47,7 @@
 fi
 AM_CONDITIONAL(MAN_PAGES_ENABLED, test x$enable_man_pages = xyes)
 
-GTK_DOC_CHECK([1.3])
+
 
 #### gcc warning flags
 
@@ -473,11 +473,6 @@
 src/polkitagent/Makefile
 src/programs/Makefile
 src/examples/Makefile
-docs/version.xml
-docs/extensiondir.xml
-docs/Makefile
-docs/polkit/Makefile
-docs/man/Makefile
 po/Makefile.in
 test/Makefile
 test/polkit/Makefile
diff -Nuar apolkit-0.107/Makefile.in polkit-0.107/Makefile.in
--- apolkit-0.107/Makefile.in	2012-07-06 16:54:39.000000000 +0300
+++ polkit-0.107/Makefile.in	2012-09-20 12:08:39.781895373 +0300
@@ -296,7 +296,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = actions data src docs po test
+SUBDIRS = actions data src po test
 NULL = 
 EXTRA_DIST = \
 	HACKING 		\
diff -Nuar apolkit-0.107/Makefile.am polkit-0.107/Makefile.am
--- apolkit-0.107/Makefile.am   2012-06-04 20:47:39.000000000 +0300
+++ polkit-0.107/Makefile.am    2012-09-22 02:42:01.108243818 +0300
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS = actions data src docs po test
+SUBDIRS = actions data src po test
