diff --git a/lenses/json.aug b/lenses/json.aug
index 6ceab09..2645806 100644
--- a/lenses/json.aug
+++ b/lenses/json.aug
@@ -37,9 +37,9 @@ let const (r:regexp) = [ label "const" . store r . ws ]
 let value0 = str | number | const /true|false|null/
 
 let fix_value (value:lens) =
-  let array = [ label "array" . lbrack . Build.opt_list value comma . rbrack ] in
+  let array = [ label "array" . lbrack . (Build.opt_list value comma)? . rbrack ] in
   let pair = [ label "entry" . str_store . colon . value ] in
-  let obj = [ label "dict" . lbrace . Build.opt_list pair comma . rbrace ] in
+  let obj = [ label "dict" . lbrace . (Build.opt_list pair comma)? . rbrace ] in
   (str | number | obj | array | const /true|false|null/)
 
 (* Typecheck finitely deep nesting *)
diff --git a/lenses/tests/test_json.aug b/lenses/tests/test_json.aug
index d8b7fa8..aec7d4c 100644
--- a/lenses/tests/test_json.aug
+++ b/lenses/tests/test_json.aug
@@ -30,6 +30,11 @@ test lns get "{ \"0\": true, \"1\":false }" =
 test lns get "{\"menu\": \"entry one\"}" =
   { "dict" { "entry" = "menu" { "string" = "entry one" } } }
 
+test lns get "[ ]" =
+  { "array" }
+
+test lns get "{}" =
+  { "dict" }
 
 let s = "{\"menu\": {
   \"id\": \"file\",