diff -Nur augeas-0.8.0-old//lenses/grub.aug augeas-0.8.0/lenses/grub.aug
--- augeas-0.8.0-old//lenses/grub.aug	2011-03-17 21:02:21.132000011 +0200
+++ augeas-0.8.0/lenses/grub.aug	2011-03-17 21:54:59.760000114 +0200
@@ -70,6 +70,7 @@
                      | kw_menu_arg "timeout"
                      | kw_menu_arg "splashimage"
                      | kw_menu_arg "gfxmenu"
+                     | kw_menu_arg "background"
                      | serial
                      | terminal
                      | password_arg
@@ -113,6 +114,9 @@
     let savedefault =
       [ command "savedefault" "\t" . (spc . store Rx.integer)? . eol ]
 
+    let configfile =
+      [ command "configfile" "\t" . spc . store Rx.no_spaces . eol ]
+
     let boot_setting = kw_boot_arg "root"
                      | kernel
                      | kw_boot_arg "initrd"
@@ -122,6 +126,7 @@
                      | kw_boot_arg "findroot"  (* Solaris extension *)
                      | kw_pres "quiet"  (* Seems to be a Ubuntu extension *)
                      | savedefault
+                     | configfile
                      | module_line
                      | map_line
 
@@ -166,4 +171,6 @@
 
     let lns = (comment | empty | menu_setting | boot | debian)*
     let xfm = transform lns (incl "/boot/grub/menu.lst"
+                           . incl "/boot/grub/grub-alt.conf"
+                           . incl "/boot/grub/grub.conf"
                            . incl "/etc/grub.conf")
