diff -Nur augeas-0.8.0-old//lenses/grub.aug augeas-0.8.0/lenses/grub.aug
--- augeas-0.8.0-old//lenses/grub.aug	2011-03-17 21:56:04.540000037 +0200
+++ augeas-0.8.0/lenses/grub.aug	2011-03-17 23:14:53.699999322 +0200
@@ -7,6 +7,7 @@
 
     let value_to_eol = store /[^= \t\n][^\n]*[^= \t\n]|[^= \t\n]/
     let eol = Util.eol
+    let indent = Util.indent
     let del_to_eol = del /[^ \t\n]*/ ""
     let spc = Util.del_ws_spc
     let opt_ws = Util.del_opt_ws ""
@@ -64,6 +65,8 @@
           |[ spc . switch_arg /timeout|lines/ ])* .
           [ spc . key /console|serial|hercules/ ]* . eol ]
 
+    let quit = [ indent . key "quit" . eol ]
+
     let menu_setting = kw_menu_arg "default"
                      | kw_menu_arg "fallback"
                      | kw_pres "hiddenmenu"
@@ -71,6 +74,9 @@
                      | kw_menu_arg "splashimage"
                      | kw_menu_arg "gfxmenu"
                      | kw_menu_arg "background"
+                     | kw_menu_arg "root"
+                     | kw_menu_arg "setup"
+                     | quit
                      | serial
                      | terminal
                      | password_arg
