--- tls1.6/tclconfig/tcl.m4	2007-06-22 14:19:12.000000000 -0700
+++ tls1.6/tclconfig/tcl.m4.new	2008-10-15 20:22:52.000000000 -0700
@@ -1460,7 +1460,7 @@
 	    DL_OBJS="tclLoadDl.o"
 	    DL_LIBS="-ldl"
 	    LDFLAGS="$LDFLAGS -Wl,--export-dynamic"
-	    CC_SEARCH_FLAGS='-Wl,-rpath,${LIB_RUNTIME_DIR}'
+	    CC_SEARCH_FLAGS='-Wl'
 	    LD_SEARCH_FLAGS=${CC_SEARCH_FLAGS}
 	    if test "`uname -m`" = "alpha" ; then
 		CFLAGS="$CFLAGS -mieee"
