Subject: Enigmail breaks Thunderbird packaging
Reference:
https://bug886095.bugzilla.mozilla.org/attachment.cgi?id=773916

diff --git a/mailnews/extensions/enigmail/genxpi b/mailnews/extensions/enigmail/genxpi
--- a/mailnews/extensions/enigmail/genxpi
+++ b/mailnews/extensions/enigmail/genxpi
@@ -73,16 +73,17 @@ if [ ${platform} = OS2_x86-gcc3 ]; then
   enigDllFile=platform/${platform}/components/${enigmimeDll}
 else
   enigDllFile=platform/${platform}/components/${libPrefix}enigmime-${xpcomAbi}${dllSuffix}
 fi
 
 subprocessDll=${libPrefix}subprocess${dllSuffix}
 spDllFile=platform/${platform}/lib/${libPrefix}subprocess-${xpcomAbi}${dllSuffix}
 
+cp ${targetDir}/chrome.manifest{,.save}
 # Prepare chrome.manifest
 cat  ${srcDir}/package/chrome.manifest | \
 sed  's/##ENIGMIMEDLL-PLACEHOLDER##/binary-component platform\/'${platform}'\/components\/'`basename ${enigDllFile}`' ABI='${platform}'/' \
 > ${targetDir}/chrome.manifest
 
 # Prepare languages other than en-US
 
 if [ -s ${srcDir}/lang/current-languages.txt ]; then
@@ -146,9 +147,10 @@ zip ${xpiFile} \
     chrome/${xpiModule}.jar \
     ${localeJar} \
     components/enigmime.xpt \
     ${enigDllFile} \
     ${spDllFile} \
     chrome.manifest \
     install.rdf
 
-rm install.rdf chrome.manifest
+rm install.rdf
+mv chrome.manifest{.save,} 
