--- newt-0.52.6/newt.c.orig	2007-05-28 14:02:40.000000000 -0300
+++ newt-0.52.6/newt.c	2007-05-28 14:11:02.000000000 -0300
@@ -46,29 +46,29 @@
 ;
 
 const struct newtColors newtDefaultColorPalette = {
-	"white", "blue", 			/* root fg, bg */
-	"black", "lightgray",			/* border fg, bg */
-	"black", "lightgray",			/* window fg, bg */
-	"white", "black",			/* shadow fg, bg */
-	"red", "lightgray",			/* title fg, bg */
-	"lightgray", "red",			/* button fg, bg */
-	"red", "lightgray",			/* active button fg, bg */
-	"lightgray", "blue",			/* checkbox fg, bg */
-	"lightgray", "red",			/* active checkbox fg, bg */
-	"lightgray", "blue",			/* entry box fg, bg */
-	"blue", "lightgray",			/* label fg, bg */
-	"black", "lightgray",			/* listbox fg, bg */
-	"lightgray", "blue",			/* active listbox fg, bg */
-	"black", "lightgray",			/* textbox fg, bg */
-	"lightgray", "red",			/* active textbox fg, bg */
-	"white", "blue",			/* help line */
-	"lightgray", "blue",			/* root text */
+	"cyan", "black",		/* root fg, bg */
+	"black", "blue",		/* border fg, bg */
+	"white", "blue",		/* window fg, bg */
+	"white", "black",		/* shadow fg, bg */
+	"white", "blue",		/* title fg, bg */
+	"black", "cyan",		/* button fg, bg */
+	"yellow", "cyan",		/* active button fg, bg */
+	"yellow", "blue",		/* checkbox fg, bg */
+	"blue", "brown",		/* active checkbox fg, bg */
+	"yellow", "blue",		/* entry box fg, bg */
+	"white", "blue",		/* label fg, bg */
+	"black", "cyan",		/* listbox fg, bg */
+	"yellow", "cyan",		/* active listbox fg, bg */
+	"white", "blue",		/* textbox fg, bg */
+	"cyan", "black",		/* active textbox fg, bg */
+	"white", "blue",		/* help line */
+	"yellow", "blue",		/* root text */
 	"blue",					/* scale full */
 	"red",					/* scale empty */
-	"blue", "lightgray",			/* disabled entry fg, bg */
-	"black", "lightgray",			/* compact button fg, bg */
-	"lightgray", "red",			/* active & sel listbox */
-	"black", "brown"			/* selected listbox */
+	"blue", "cyan",			/* disabled entry fg, bg */
+	"white", "blue",		/* compact button fg, bg */
+	"yellow", "red",		/* active & sel listbox */
+	"black", "brown"		/* selected listbox */
 };
 
 static const struct keymap keymap[] = {
