diff -Nuar slimevolley.orig/CMakeLists.txt slimevolley/CMakeLists.txt
--- slimevolley.orig/CMakeLists.txt	2010-07-17 23:41:41.000000000 +0300
+++ slimevolley/CMakeLists.txt	2010-11-07 20:34:23.394000034 +0200
@@ -104,7 +104,7 @@
 			SET(DATA_DIR "data")
 		ELSE(WINDOWS)
 			SET(DATA_DIR 
-"${CMAKE_INSTALL_PREFIX}/share/games/slimevolley")
+"${CMAKE_INSTALL_PREFIX}/share/slimevolley")
 		ENDIF(WINDOWS)
 	ENDIF(NOT DATA_DIR)
 	MESSAGE(STATUS "Slime data will be installed in ${DATA_DIR}.")
diff -Nuar slimevolley.orig/debian/slimevolley.desktop slimevolley/debian/slimevolley.desktop
--- slimevolley.orig/debian/slimevolley.desktop	2010-07-17 23:41:42.000000000 +0300
+++ slimevolley/debian/slimevolley.desktop	2010-11-07 20:29:45.929000035 +0200
@@ -1,9 +1,14 @@
 [Desktop Entry]
+Type=Application
+Version=1.0
+Encoding=UTF-8
 Name=Slime Volley
+GenericName=Volleyball Game
+GenericName[tr]=Voleybol Oyunu
 Comment=Unrealistic 2D volleyball simulation
 Comment[fr]=Simulation non réaliste de volley en 2D
+Comment[fr]=Gerçekçi olmayan bir 2B voleybol simülasyonu
 Exec=slimevolley
 Icon=slimevolley
 Terminal=false
-Type=Application
 Categories=Game;SportsGame;
