--- trunk/scripts/CMakeLists.txt	2011/10/31 15:33:42	12018
+++ trunk/scripts/CMakeLists.txt	2012/01/04 15:45:07	12116
@@ -30,9 +30,9 @@
   if(PLD_pstex)
     set(scripts_SCRIPTS ${scripts_SCRIPTS} pstex2eps)
   endif(PLD_pstex)
-  if (NOT ${scripts_SCRIPTS} STREQUAL "")
+  if (NOT "${scripts_SCRIPTS}" STREQUAL "")
     install(PROGRAMS ${scripts_SCRIPTS} DESTINATION ${BIN_DIR})
-  endif (NOT ${scripts_SCRIPTS} STREQUAL "")
+  endif (NOT "${scripts_SCRIPTS}" STREQUAL "")
 endif(HAVE_BASH)
 
 if(ENABLE_tcl OR ENABLE_tk)
