--- plplot_test/CMakeLists.txt.orig	2011-11-20 16:10:04.000000000 +0000
+++ plplot_test/CMakeLists.txt	2011-11-20 16:10:24.000000000 +0000
@@ -387,12 +387,6 @@
       )
   endif(PLD_pdfqt)
 
-  if(PLD_pdf)
-    add_test(examples_pdf
-      ${SH_EXECUTABLE} -c "${TEST_ENVIRONMENT} ${TEST_SCRIPT} --front-end=c --device=pdf"
-      )
-  endif(PLD_pdf)
-
   if(CMP_EXECUTABLE OR DIFF_EXECUTABLE AND TAIL_EXECUTABLE)
     configure_file(
       test_diff.sh.in
