--- a/src/wl/sys/wl_linux.c	2014-05-01 17:00:42.882111597 +0200
+++ b/src/wl/sys/wl_linux.c	2014-05-01 16:59:01.866015434 +0200
@@ -2042,8 +2042,11 @@
 void
 wl_dump_ver(wl_info_t *wl, struct bcmstrbuf *b)
 {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdate-time"
 	bcm_bprintf(b, "wl%d: %s %s version %s\n", wl->pub->unit,
 		__DATE__, __TIME__, EPI_VERSION_STR);
+#pragma GCC diagnostic pop
 }
 
 #if defined(BCMDBG) 
