diff -Nuar a/src/wl/sys/wl_linux.c b/src/wl/sys/wl_linux.c
--- a/src/wl/sys/wl_linux.c	2011-10-22 19:56:55.000000000 +0300
+++ b/src/wl/sys/wl_linux.c	2013-07-04 15:33:42.992777649 +0300
@@ -40,7 +40,8 @@
 #include <linux/pci_ids.h>
 #define WLC_MAXBSSCFG		1	
 
-#include <asm/system.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,10,0)
+#endif
 #include <asm/io.h>
 #include <asm/irq.h>
 #include <asm/pgtable.h>
@@ -245,6 +246,7 @@
 
 static int nompc = 0;
 module_param(nompc, int, 0);
+MODULE_LICENSE("Mixed/Proprietary");
 
 #ifdef quote_str
 #undef quote_str
@@ -385,7 +387,11 @@
 #endif
 	.ndo_get_stats = wl_get_stats,
 	.ndo_set_mac_address = wl_set_mac_address,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 10, 0)
+	.ndo_set_rx_mode = wl_set_multicast_list,
+#else
 	.ndo_set_multicast_list = wl_set_multicast_list,
+#endif
 	.ndo_do_ioctl = wl_ioctl
 };
 
@@ -1579,11 +1585,7 @@
 	}
 
 	WL_LOCK(wl);
-	if (!capable(CAP_NET_ADMIN)) {
-		bcmerror = BCME_EPERM;
-	} else {
-		bcmerror = wlc_ioctl(wl->wlc, ioc.cmd, buf, ioc.len, wlif->wlcif);
-	}
+	bcmerror = wlc_ioctl(wl->wlc, ioc.cmd, buf, ioc.len, wlif->wlcif);
 	WL_UNLOCK(wl);
 
 done1:
@@ -3172,19 +3174,3 @@
 	}
 	return length;
 }
-
-static int
-wl_reg_proc_entry(wl_info_t *wl)
-{
-	char tmp[32];
-	sprintf(tmp, "%s%d", HYBRID_PROC, wl->pub->unit);
-	if ((wl->proc_entry = create_proc_entry(tmp, 0644, NULL)) == NULL) {
-		WL_ERROR(("%s: create_proc_entry %s failed\n", __FUNCTION__, tmp));
-		ASSERT(0);
-		return -1;
-	}
-	wl->proc_entry->read_proc = wl_proc_read;
-	wl->proc_entry->write_proc = wl_proc_write;
-	wl->proc_entry->data = wl;
-	return 0;
-}
