diff --git a/hw/xfree86/modes/xf86RandR12.c b/hw/xfree86/modes/xf86RandR12.c
index a773c34..f9ebb5f 100644
--- a/hw/xfree86/modes/xf86RandR12.c
+++ b/hw/xfree86/modes/xf86RandR12.c
@@ -927,7 +927,8 @@ xf86RandR12SetRotations(ScreenPtr pScreen, Rotation rotations)
     for (c = 0; c < config->num_crtc; c++) {
         xf86CrtcPtr crtc = config->crtc[c];
 
-        RRCrtcSetRotations(crtc->randr_crtc, rotations);
+        if (crtc != NULL)
+            RRCrtcSetRotations(crtc->randr_crtc, rotations);
     }
 #endif
     randrp->supported_rotations = rotations;
