Index: xorg-server/hw/xfree86/common/xf86AutoConfig.c
===================================================================
--- xorg-server.orig/hw/xfree86/common/xf86AutoConfig.c	2009-12-02 19:30:15.000000000 +0200
+++ xorg-server/hw/xfree86/common/xf86AutoConfig.c	2009-12-02 23:32:39.000000000 +0200
@@ -179,6 +179,7 @@
 	case 0x1002:		    driverList[0] = "ati"; break;
 	case 0x102c:		    driverList[0] = "chips"; break;
 	case 0x1013:		    driverList[0] = "cirrus"; break;
+	case 0x1234:		    driverList[0] = "vesa"; break; /* qemu device not supported by cirrus */
 	case 0x3d3d:		    driverList[0] = "glint"; break;
 	case 0x105d:		    driverList[0] = "i128"; break;
 	case 0x8086:
