Index: xorg-server/hw/xfree86/modes/xf86EdidModes.c
===================================================================
--- xorg-server.orig/hw/xfree86/modes/xf86EdidModes.c	2009-12-02 19:30:15.000000000 +0200
+++ xorg-server/hw/xfree86/modes/xf86EdidModes.c	2009-12-02 23:37:43.000000000 +0200
@@ -175,6 +175,11 @@
 	DDC->vendor.prod_id == 1157)
 	return TRUE;
 
+    /* LP #380009: HP Compaq NC8430 LP154W01-TLA8 */
+    if (memcmp (DDC->vendor.name, "LPL", 4) == 0 &&
+        DDC->vendor.prod_id == 5750)
+        return TRUE;
+
     return FALSE;
 }
 
