=== modified file 'hw/vfb/InitOutput.c'
--- a/hw/vfb/InitOutput.c	2010-02-02 16:34:02 +0000
+++ b/hw/vfb/InitOutput.c	2010-02-02 16:36:42 +0000
@@ -912,6 +912,12 @@
 	fbPictureInit (pScreen, 0, 0);
 #endif
 
+    if (!miRandRInit(pScreen))
+        /* Not sure how to emit warnings in xorg. Should warn here though:
+	 * ("Could not initialise RANDR\n");
+	 */
+        return FALSE;
+
     miInitializeBackingStore(pScreen);
 
     /*

