=== modified file 'hw/vfb/InitOutput.c'
--- a/hw/vfb/InitOutput.c	2009-06-11 20:34:06 +0000
+++ b/hw/vfb/InitOutput.c	2010-02-02 06:28:05 +0000
@@ -249,7 +249,7 @@
     ErrorF("-screen scrn WxHxD     set screen's width, height, depth\n");
     ErrorF("-pixdepths list-of-int support given pixmap depths\n");
 #ifdef RENDER
-    ErrorF("+/-render		   turn on/of RENDER extension support"
+    ErrorF("+/-render		   turn on/off RENDER extension support"
 	   "(default on)\n");
 #endif
     ErrorF("-linebias n            adjust thin line pixelization\n");

