diff -Naur dansguardian-2.10.1.1/configs/contentscanners/clamdscan.conf.in dansguardian-2.10.1.1.oden/configs/contentscanners/clamdscan.conf.in
--- dansguardian-2.10.1.1/configs/contentscanners/clamdscan.conf.in	2008-11-18 12:27:04.000000000 +0100
+++ dansguardian-2.10.1.1.oden/configs/contentscanners/clamdscan.conf.in	2009-06-10 15:31:53.000000000 +0200
@@ -1,7 +1,7 @@
 plugname = 'clamdscan'
 
 # edit this to match the location of your ClamD UNIX domain socket
-#clamdudsfile = '/var/run/clamav/clamd.sock'
+clamdudsfile = '@localstatedir@/clamav/clamd.socket'
 
 # If this string is set, the text it contains shall be removed from the
 # beginning of filenames when passing them to ClamD.
diff -Naur dansguardian-2.10.1.1/configs/dansguardian.conf.in dansguardian-2.10.1.1.oden/configs/dansguardian.conf.in
--- dansguardian-2.10.1.1/configs/dansguardian.conf.in	2009-04-03 10:32:13.000000000 +0200
+++ dansguardian-2.10.1.1.oden/configs/dansguardian.conf.in	2009-06-10 15:32:24.000000000 +0200
@@ -62,7 +62,7 @@
 # Log file location
 # 
 # Defines the log directory and filename.
-#loglocation = '@DGLOGLOCATION@/access.log'
+loglocation = '@DGLOGLOCATION@/access.log'
 
 
 # Statistics log file location
@@ -72,7 +72,7 @@
 # Once every 3 minutes, the current number of IPs in the cache, and the most
 # that have been in the cache since the daemon was started, are written to this
 # file. IPs persist in the cache for 7 days.
-#statlocation = '@DGLOGLOCATION@/stats'
+statlocation = '@DGLOGLOCATION@/stats'
 
 
 # Network Settings
@@ -105,7 +105,7 @@
 #
 # See http://www.kb.cert.org/vuls/id/435052
 # on (default) | off
-@ORIGIPSUPPORT@originalip = on
+@ORIGIPSUPPORT@originalip = off
 
 # accessdeniedaddress is the address of your web server to which the cgi
 # dansguardian reporting script was copied. Only used in reporting levels 1 and 2.
@@ -348,7 +348,7 @@
 # File cache dir
 # Where DG will download files to be scanned if too large for the
 # RAM cache.
-filecachedir = '/tmp'
+filecachedir = '@localstatedir@/dansguardian/tmp'
 
 
 
@@ -565,24 +565,24 @@
 # IPC filename
 # 
 # Defines IPC server directory and filename used to communicate with the log process.
-ipcfilename = '/tmp/.dguardianipc'
+ipcfilename = '@localstatedir@/dansguardian/.dguardianipc'
 
 # URL list IPC filename
 # 
 # Defines URL list IPC server directory and filename used to communicate with the URL
 # cache process.
-urlipcfilename = '/tmp/.dguardianurlipc'
+urlipcfilename = '@localstatedir@/dansguardian/.dguardianurlipc'
 
 # IP list IPC filename
 #
 # Defines IP list IPC server directory and filename, for communicating with the client
 # IP cache process.
-ipipcfilename = '/tmp/.dguardianipipc'
+ipipcfilename = '@localstatedir@/dansguardian/.dguardianipipc'
 
 # PID filename
 # 
 # Defines process id directory and filename.
-#pidfilename = '@DGPIDDIR@/dansguardian.pid'
+pidfilename = '@DGPIDDIR@/dansguardian.pid'
 
 # Disable daemoning
 # If enabled the process will not fork into the background.
@@ -609,8 +609,8 @@
 # Temp files created during virus scanning are given owner and group read
 # permissions; to use content scanners based on external processes, such as
 # clamdscan, the two processes must run with either the same group or user ID.
-#daemonuser = '@DGPROXYUSER@'
-#daemongroup = '@DGPROXYGROUP@'
+daemonuser = '@DGPROXYUSER@'
+daemongroup = '@DGPROXYGROUP@'
 
 # Soft restart
 # When on this disables the forced killing off all processes in the process group.
