Index: dansguardian-2.10.1.1/configure.ac
===================================================================
--- dansguardian-2.10.1.1.orig/configure.ac
+++ dansguardian-2.10.1.1/configure.ac
@@ -333,7 +333,7 @@ clamav,
 	CLAMAVSHM="!! Not supported on this platform !!"
 else
 	AC_MSG_RESULT(yes)
-	PKG_CHECK_MODULES([CLAMAV],[libclamav >= 4])
+	PKG_CHECK_MODULES([CLAMAV],[libclamav >= 0.96])
 
 	AC_DEFINE([HAVE_CLAMAV],[],[Define to enable ClamAV content scanner])
 	
