--- dansguardian-2.9.8.2/configs/lists/downloadmanagers/managedmimetypelist.org	    2007-03-02 15:44:09.000000000 +0200
+++ dansguardian-2.9.8.2/configs/lists/downloadmanagers/managedmimetypelist	        2007-03-02 15:45:40.000000000 +0200
@@ -95,6 +95,10 @@
 application/bzip2
 application/x-bz2
 application/x-bzip
+# pisi
+application/zip
+application/x-zip
+multipart/x-zip
 # sit
 application/stuffit
 application/x-sit
--- dansguardian-2.9.8.2/configs/lists/downloadmanagers/managedextensionlist.org	2007-03-02 15:44:02.000000000 +0200
+++ dansguardian-2.9.8.2/configs/lists/downloadmanagers/managedextensionlist	    2007-03-02 15:48:05.000000000 +0200
@@ -26,6 +26,7 @@
 .gz
 .tar
 .bz2
+.pisi
 .sit
 .bin
 .hqx
