Index: cloudcity/XBar/menubar.cpp
===================================================================
--- cloudcity.orig/XBar/menubar.cpp
+++ cloudcity/XBar/menubar.cpp
@@ -261,7 +261,7 @@ MenuBar::initStyleOption(QStyleOptionMen
         int lightX = QFontMetrics(option->font).xHeight();
         option->font.setWeight(QFont::Black);
 //         option->font.setCapitalization ( QFont::AllUppercase );
-        option->font.setPointSize((option->font.pointSize()*lightX)/QFontMetrics(option->font).xHeight());
+//        option->font.setPointSize((option->font.pointSize()*lightX)/QFontMetrics(option->font).xHeight());
     }
     option->fontMetrics = QFontMetrics(option->font);
 
