--- ./scripts/exim_install	2012-10-25 06:37:38.000000000 +0300
+++ ./scripts/exim_install	2012-11-28 22:41:39.000000000 +0200
@@ -217,15 +217,7 @@
   # The exim binary is handled specially
 
   if [ $name = exim${EXE} ]; then
-    version=exim-`./exim -bV -C /dev/null | \
-      awk '/Exim version/ { OFS=""; print $3,"-",substr($4,2,length($4)-1) }'`${EXE}
-
-    if [ "${version}" = "exim-${EXE}" ]; then
-      echo $com ""
-      echo $com "*** Could not run ./exim to find version number ***"
-      echo $com "*** Exim installation ${ver}failed ***"
-      exit 1
-    fi
+    version=exim
 
     # Do something only if newer than existing file, or no existing file
 
@@ -267,72 +259,6 @@
         echo $com "$CHMOD u+s omitted: -no_chown was specified"
       fi
 
-      # Now sort out the "exim" alias, unless NO_SYMLINK is set.
-
-      if [ "X$NO_SYMLINK" = "X" ] && [ "$do_symlink" != "no" ] ; then
-
-        #  First check whether "exim" exists in the directory.
-        if [ -f ${BIN_DIRECTORY}/exim ]; then
-
-          # If it's not a symbolic link, make a copy with the old version number
-          if [ `ls -l ${BIN_DIRECTORY}/exim | cut -c1-1` != 'l' ]; then
-            oldversion=exim-`${BIN_DIRECTORY}/exim -bV -C /dev/null | \
-              awk '/Exim version/ { OFS=""; print $3,"-",substr($4,2,length($4)-1) }'`${EXE}
-            if [ "${version}" = "${oldversion}" ] ; then
-              echo $com ""
-              echo $com "*** Existing file called exim has the same version and compile number ***"
-              echo $com "*** Exim installation ${ver}failed ***"
-              exit 1
-            fi
-            echo ${CP} ${BIN_DIRECTORY}/exim ${BIN_DIRECTORY}/${oldversion}
-            ${real} ${CP} ${BIN_DIRECTORY}/exim ${BIN_DIRECTORY}/${oldversion}
-            if [ $? -ne 0 ]; then
-              echo $com ""
-              echo $com "*** Exim installation ${ver}failed ***"
-              exit 1
-            fi
-          fi
-
-          # Now we can move the name "exim" to be a symbolic link to the new
-          # version, atomically.
-
-          echo \(cd ${BIN_DIRECTORY}\; ${LN} -s ${version} temporary_exim\)
-          (${real} cd ${BIN_DIRECTORY}; ${real} ${LN} -s ${version} temporary_exim)
-          if [ $? -ne 0 ]; then
-            echo $com ""
-            echo $com "*** Exim installation ${ver}failed ***"
-            exit 1
-          fi
-
-          echo ${MV} -f ${BIN_DIRECTORY}/temporary_exim ${BIN_DIRECTORY}/exim
-          ${real} ${MV} -f ${BIN_DIRECTORY}/temporary_exim ${BIN_DIRECTORY}/exim
-          if [ $? -ne 0 ]; then
-            echo $com ""
-            echo $com "*** Exim installation ${ver}failed ***"
-            exit 1
-          fi
-
-        # If "exim" does not already exist just create a symbolic link.
-
-        else
-          echo \(cd ${BIN_DIRECTORY}\; ${LN} -s ${version} exim\)
-          (${real} cd ${BIN_DIRECTORY}; ${real} ${LN} -s ${version} exim)
-          if [ $? -ne 0 ]; then
-            echo $com ""
-            echo $com "*** Exim installation ${ver}failed ***"
-            exit 1
-          fi
-        fi
-
-      else
-        echo $com "creation of symlink omitted"
-        if [ "X$NO_SYMLINK" != "X" ] ; then
-          echo $com "(NO_SYMLINK is specified in Local/Makefile)"
-        else
-          echo $com "(-no_symlink was specified)"
-        fi
-      fi
-
     # New binary is not newer than the installed file
 
     else
