Index: wxGTK-2.8.11/configure.in
===================================================================
--- wxGTK-2.8.11.orig/configure.in
+++ wxGTK-2.8.11/configure.in
@@ -5418,7 +5418,7 @@ if test "$TOOLKIT" != "MSW" -a "$USE_OS2
         dnl define the list of the thread options to try in the loop below
         dnl with the convention that anything starting with '-' is a cpp flag
         dnl while anything else is a library (i.e. there is an implicit "-l")
-        THREAD_OPTS="-pthread"
+        THREAD_OPTS="pthread"
         case "${host}" in
           *-*-solaris2* | *-*-sunos4* )
                 if test "x$GCC" = "xyes"; then
Index: wxGTK-2.8.11/configure
===================================================================
--- wxGTK-2.8.11.orig/configure
+++ wxGTK-2.8.11/configure
@@ -39579,7 +39579,7 @@ echo "$as_me: WARNING: BeOS threads are 
     if test "$wxUSE_THREADS" = "yes" ; then
 
 
-                                THREAD_OPTS="-pthread"
+                                THREAD_OPTS="pthread"
         case "${host}" in
           *-*-solaris2* | *-*-sunos4* )
                 if test "x$GCC" = "xyes"; then
