diff -uNr matplotlib-1.0.1.orig/lib/matplotlib/sphinxext/plot_directive.py matplotlib-1.0.1/lib/matplotlib/sphinxext/plot_directive.py
--- matplotlib-1.0.1.orig/lib/matplotlib/sphinxext/plot_directive.py	2011-01-23 05:42:08.000000000 +0900
+++ matplotlib-1.0.1/lib/matplotlib/sphinxext/plot_directive.py	2011-01-23 05:44:48.000000000 +0900
@@ -346,7 +346,7 @@
         del options['nofigs']
 
     formats = setup.config.plot_formats
-    if type(formats) == str:
+    if type(formats) == str or type(formats) == unicode:
         formats = eval(formats)
 
     fname = os.path.basename(plot_path)
