--- setupext.py.bak	2012-09-21 14:22:08.752827333 -0400
+++ setupext.py	2012-09-21 14:23:43.760172391 -0400
@@ -366,9 +366,12 @@
 def add_base_flags(module):
     incdirs = filter(os.path.exists,
                      [os.path.join(p, 'include') for p in basedirlist ])
-    libdirs = filter(os.path.exists,
-                     [os.path.join(p, 'lib')     for p in basedirlist ]+
-                     [os.path.join(p, 'lib64')     for p in basedirlist ] )
+    if sys.arch == 'x86_64':
+        libdirs = filter(os.path.exists,
+                         [os.path.join(p, 'lib64')     for p in basedirlist ])
+    else:
+        libdirs = filter(os.path.exists,
+                         [os.path.join(p, 'lib')     for p in basedirlist ])
 
     module.include_dirs.extend(incdirs)
     module.include_dirs.append('.')
