--- setupext.py.bak	2011-10-11 09:28:30.923854941 -0400
+++ setupext.py	2011-10-11 09:30:23.644095529 -0400
@@ -912,10 +912,10 @@
     return tcl_lib, tcl_inc, 'tcl' + tk_ver, tk_lib, tk_inc, 'tk' + tk_ver
 
 def hardcoded_tcl_config():
-    tcl_inc = "/usr/local/include"
-    tk_inc = "/usr/local/include"
-    tcl_lib = "/usr/local/lib"
-    tk_lib = "/usr/local/lib"
+    tcl_inc = "/usr/include"
+    tk_inc = "/usr/include"
+    tcl_lib = "/usr/lib"
+    tk_lib = "/usr/lib"
     return tcl_lib, tcl_inc, 'tcl', tk_lib, tk_inc, 'tk'
 
 def add_tk_flags(module):
