Index: work/kpartx/Makefile
===================================================================
--- work.orig/kpartx/Makefile
+++ work/kpartx/Makefile
@@ -6,7 +6,7 @@ include ../Makefile.inc
 
 CFLAGS += -I. -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
 
-LDFLAGS = -ldevmapper
+LIBADD = -ldevmapper
 OBJS = bsd.o dos.o kpartx.o solaris.o unixware.o dasd.o sun.o \
        gpt.o mac.o crc32.o lopart.o xstrncpy.o devmapper.o
 EXEC = kpartx
@@ -14,7 +14,7 @@ EXEC = kpartx
 all: $(EXEC)
 
 $(EXEC): $(OBJS)
-	$(CC) $(OBJS) -o $(EXEC) $(LDFLAGS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(OBJS) $(LIBADD) -o $(EXEC)
 	$(GZIP) $(EXEC).8 > $(EXEC).8.gz
 	
 install: $(EXEC) $(EXEC).8
Index: work/multipathd/Makefile
===================================================================
--- work.orig/multipathd/Makefile
+++ work/multipathd/Makefile
@@ -6,8 +6,7 @@ include ../Makefile.inc
 # basic flags setting
 #
 CFLAGS += -I$(multipathdir)
-LDFLAGS += -lpthread -ldevmapper -lreadline -lncurses -ldl \
-	   -lmultipath -L$(multipathdir)
+LIBADD = -lpthread -ldevmapper -lreadline -lncurses -ldl -lmultipath
 
 #
 # debuging stuff
@@ -28,7 +27,7 @@ OBJS = main.o pidfile.o uxlsnr.o uxclnt.
 all : $(EXEC)
 
 $(EXEC): $(OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $(EXEC) $(OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -L$(multipathdir) $(OBJS) $(LIBADD) -o $(EXEC)
 	$(GZIP) $(EXEC).8 > $(EXEC).8.gz
 
 install:
Index: work/multipath/Makefile
===================================================================
--- work.orig/multipath/Makefile
+++ work/multipath/Makefile
@@ -7,14 +7,13 @@ include ../Makefile.inc
 OBJS = main.o
 
 CFLAGS += -I$(multipathdir)
-LDFLAGS += -lpthread -ldevmapper -ldl -lmultipath -L$(multipathdir)
-
+LIBADD = -lpthread -ldevmapper -ldl -lmultipath
 EXEC = multipath
 
 all: $(EXEC)
 
 $(EXEC): $(OBJS)
-	$(CC) $(CFLAGS) $(OBJS) -o $(EXEC) $(LDFLAGS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -L$(multipathdir) $(OBJS) $(LIBADD) -o $(EXEC)
 	$(GZIP) $(EXEC).8 > $(EXEC).8.gz
 	$(GZIP) $(EXEC).conf.5 > $(EXEC).conf.5.gz
 
