Index: hplip-3.9.6b/plugin.py
===================================================================
--- hplip-3.9.6b.orig/plugin.py
+++ hplip-3.9.6b/plugin.py
@@ -406,6 +406,14 @@ else: # INTERACTIVE_MODE
 
         tui.header("INSTALLING PLUG-IN")
 
+        # Remove udev rules
+        import glob
+        for rule in glob.glob("/etc/udev/rules.d/86-hpmud*"):
+            try:
+                os.unlink(rule)
+            except IOError:
+                pass
+
         core.run_plugin(mode, plugin_install_callback)
 
         cups_devices = device.getSupportedCUPSDevices(['hp']) #, 'hpfax'])
