--- istanbul/main/save_window.py	
+++ istanbul/main/save_window.py	
@@ -143,8 +143,11 @@ class SaveWindow(gtk.Window):
             yoptions=0)
         table.set_row_spacing(0,6)
         table.set_row_spacing(1,6)
+        def init_videowidget( widget):
+            widget._xid = widget.window.xid
+            self.player.pause()
         self.videowidget.connect_after('realize',
-                                       lambda *x: self.player.pause() )
+            init_videowidget )
         self.videowidget.set_size_request(width, height)
 
         # create save, edit, cancel buttons
