--- istanbul/main/save_window.py	2007-02-23 02:01:12.000000000 -0900
+++ ../save_window.py.new	2009-11-25 21:28:31.000000000 -0900
@@ -289,6 +289,15 @@
                         dialog.hide()
                     else:
                         save_succesful = True
+            except Exception:
+                dialog = gtk.MessageDialog(self,
+                    gtk.DIALOG_MODAL | gtk.DIALOG_DESTROY_WITH_PARENT,
+                    gtk.MESSAGE_QUESTION,
+                    gtk.BUTTONS_OK,
+                    _("There was an unknown error writing to \"%s\".  Please try a different file or location.") % (
+                    self.filechooser.get_uri()))
+                dialog.run()
+                dialog.hide()
             else:
                 save_succesful = True
         else:
