diff -Nuar e-module-snow-0.0.1_58042.orig//module.desktop.in e-module-snow-0.0.1_58042/module.desktop.in
--- e-module-snow-0.0.1_58042.orig//module.desktop.in	2011-02-16 22:39:52.000000000 +0200
+++ e-module-snow-0.0.1_58042/module.desktop.in	2011-03-31 15:21:31.527179802 +0300
@@ -9,6 +9,7 @@
 Name[fr]=Neige
 Name[el]=Χιόνι
 Name[pt]=Neve
+Name[tr]=Kar
 Icon=e-module-snow
 X-Enlightenment-ModuleType=look
 Comment=Module to display snow on the desktop.
@@ -21,3 +22,4 @@
 Comment[fr]=Un module montrant de la neige sur le bureau.
 Comment[el]=Άρθρωμα για την προβολή χιονιού στην επιφάνεια εργασίας.
 Comment[pt]=Um módulo para exibir neve na área de trabalho
+Comment[tr]=Masaüstüne kar yağmasını sağlayan modüldür.
