diff -up ./CMakeLists.txt.stageplugin ./CMakeLists.txt
--- ./CMakeLists.txt.stageplugin	2011-05-29 22:19:52.280070049 -0400
+++ ./CMakeLists.txt	2011-05-29 22:20:21.776210856 -0400
@@ -194,9 +194,9 @@ ADD_SUBDIRECTORY(assets)
 ADD_SUBDIRECTORY(worlds)
 ADD_SUBDIRECTORY(avonstage)		 
 
-#IF ( BUILD_PLAYER_PLUGIN AND PLAYER_FOUND )
-#  ADD_SUBDIRECTORY(libstageplugin)
-#ENDIF ( BUILD_PLAYER_PLUGIN AND PLAYER_FOUND )	 
+IF ( BUILD_PLAYER_PLUGIN AND PLAYER_FOUND )
+  ADD_SUBDIRECTORY(libstageplugin)
+ENDIF ( BUILD_PLAYER_PLUGIN AND PLAYER_FOUND )	 
 
 
 # generate a cpack config file used to create packaged tarballs
