diff -ur e2fsprogs-1.40.8.orig/lib/blkid/probe.c e2fsprogs-1.40.8/lib/blkid/probe.c
--- e2fsprogs-1.40.8.orig/lib/blkid/probe.c	2008-03-14 00:20:46.000000000 +0200
+++ e2fsprogs-1.40.8/lib/blkid/probe.c	2008-05-26 14:37:08.000000000 +0300
@@ -1105,7 +1105,7 @@
 static struct blkid_magic type_array[] = {
 /*  type     kboff   sboff len  magic			probe */
   { "oracleasm", 0,	32,  8, "ORCLDISK",		probe_oracleasm },
-  { "ntfs",	 0,	 3,  8, "NTFS    ",		probe_ntfs },
+  { "ntfs-3g",	 0,	 3,  8, "NTFS    ",		probe_ntfs },
   { "jbd",	 1,   0x38,  2, "\123\357",		probe_jbd },
   { "ext4dev",	 1,   0x38,  2, "\123\357",		probe_ext4dev },
   { "ext4",	 1,   0x38,  2, "\123\357",		probe_ext4 },
