Index: vimchant-1.1/autoload/vimchant.vim
===================================================================
--- vimchant-1.1.orig/autoload/vimchant.vim
+++ vimchant-1.1/autoload/vimchant.vim
@@ -108,7 +108,7 @@ function! s:CheckSpelling() "{{{1
 	let content_string = substitute(content_string,'\v\s[[:punct:]]*''(\k)',' \1','g')
 	let content_string = substitute(content_string,'\v(\k)''[[:punct:]]*\s','\1 ','g')
 	let spelling_errors = system(lang.s:spellcheck_prg,content_string)
-	if v:shell_error != 0
+	if v:shell_error == 0
 		if spelling_errors =~ '\cCouldn''t create a dictionary for'
 			call vimchant#SpellCheckSwitch('Off',1)
 			echohl WarningMsg
