--- samba-3.5.1/source3/configure.in~	2010-03-08 20:34:39.000000000 +0100
+++ samba-3.5.1/source3/configure.in	2010-04-06 16:05:40.000000000 +0100
@@ -1712,7 +1712,7 @@
 # platforms unresolved symbols might be intended, so we might have to add
 # platform specific exceptions here.
 
-for flags in "-Wl,-z,defs" "-error_unresolved" "-Wl,-error_unresolved" ; do
+for flags in "-Wl,--no-undefined" "-Wl,-z,defs" "-error_unresolved" "-Wl,-error_unresolved" ; do
 	saved_ldflags="$LDFLAGS"
 	AC_MSG_CHECKING([if $flags works])
 	LDFLAGS="$flags $saved_ldflags"
--- samba-3.5.1/source3/Makefile.in~	2010-03-08 20:34:39.000000000 +0100
+++ samba-3.5.1/source3/Makefile.in	2010-04-08 15:35:20.000000000 +0100
@@ -1432,6 +1432,7 @@
 	@echo "      LDFLAGS    = $(LDFLAGS)"
 	@echo "      DYNEXP     = $(DYNEXP)"
 	@echo "      LDSHFLAGS  = $(LDSHFLAGS)"
+	@echo "LDSHFLAGS_MODULES= $(LDSHFLAGS_MODULES)"
 	@echo "      SHLIBEXT   = @SHLIBEXT@"
 	@echo "      SONAMEFLAG = @SONAMEFLAG@"
  
