diff -p -up samba-3.5.3/nsswitch/wins.c.wins_log samba-3.5.3/nsswitch/wins.c
--- samba-3.5.3/nsswitch/wins.c.wins_log	2010-05-17 11:51:23.000000000 +0000
+++ samba-3.5.3/nsswitch/wins.c	2010-06-06 15:48:54.401949879 +0000
@@ -96,7 +96,7 @@ static void nss_wins_init(void)
 	AllowDebugChange = False;
 
 	TimeInit();
-	setup_logging("nss_wins",False);
+	setup_logging(NULL,False);
 	load_case_tables();
 	lp_load(get_dyn_CONFIGFILE(),True,False,False,True);
 	load_interfaces();
diff -p -up samba-3.5.3/source3/lib/debug.c.wins_log samba-3.5.3/source3/lib/debug.c
--- samba-3.5.3/source3/lib/debug.c.wins_log	2010-05-17 12:51:23.000000000 +0100
+++ samba-3.5.3/source3/lib/debug.c	2010-06-07 20:20:56.000000000 +0100
@@ -589,7 +589,7 @@
 		x_setbuf( x_stdout, NULL );
 	}
 #ifdef WITH_SYSLOG
-	else {
+	else if (pname) {
 		const char *p = strrchr_m( pname,'/' );
 		if (p)
 			pname = p + 1; 
