Index: kradio4-4.0.2/kradio4.desktop
===================================================================
--- kradio4-4.0.2.orig/kradio4.desktop
+++ kradio4-4.0.2/kradio4.desktop
@@ -1,9 +1,11 @@
 [Desktop Entry]
 Type=Application
-Name=kradio4
+Name=KRadio
 Exec=kradio4 -caption "%c"
-Icon=kradio4.svg
+Icon=kradio4.png
 X-DocPath=kradio4/index.html
 Comment=V4L/V4L2-Radio Application for KDE4
+GenericName[tr]=Radyo Uygulaması
+Comment[tr]=KDE4 için Radyo Uygulaması
 Terminal=false
-Categories=Qt;KDE;AudioVideo;
+Categories=Qt;KDE;AudioVideo;Audio;Player;
