--- /dev/null   1970-01-01 02:00:00.000000000 +0200
+++ presets/turkey/izmir-antenna.krp	2006-10-08 15:05:29.499508019 +0300
@@ -0,0 +1,500 @@
+<kradiorc>
+    <stationlist>
+        <format>kradio-1.0</format>
+        <info>
+            <creator>kradio-snapshot_2005_12_04</creator>
+            <maintainer>leopartux, &lt;leopartux@yahoo.com.tr&gt;</maintainer>
+            <changed>2006-09-24T16:19:46</changed>
+            <country>Turkiye</country>
+            <city>İzmir</city>
+            <media>antenna</media>
+            <comments></comments>
+        </info>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607FA7E2E4A53A41AE958DB98159D59BEEBD842C5DF5F6322151D6BBB2A029A07D9</stationID>
+            <name>RADYO KARADENİZ</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-0.01</volumepreset>
+            <frequency>87.7</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607B40A26E67591AE45B92DC144480D83B900716DBCD0608DC88818AD7C74264ADA</stationID>
+            <name>TRT-3</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>88</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607DC9FC312AF54A5820F54F6D69F13B6DA7834E3EB8449B1001039ABD906FFAB6F</stationID>
+            <name>MORAL FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>88.3</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607EFF83345BFCEFBAEFAF1881B9FEB912ED88EBB5F9C4C28450B4A1C3430EDDCBF</stationID>
+            <name>RADYO 35</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>89</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>115910560775C8F2F2262D92720199F1766DC82E20C507D29B926D7F027CBF7F7040CEEC62</stationID>
+            <name>ALEM - FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>89.3</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>11591056078312289400695553F47805AC33DDE809782DE802697526527568315202F9FAED</stationID>
+            <name>SHOW RADYO</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>89.6</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607B5E20EC323A8E43E9553520088A61D9B0BFB32B8E61EB1A089FEF16654F633A9</stationID>
+            <name>İMBAT FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>90</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607EC1A0C9EF7B06E83D93436764017E649C1B4F90CDE9C09E1CD414EBBCD4CCCCC</stationID>
+            <name>NUMBER ONE FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>90.5</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607FAE384F5B5F1E3B604B5612D64BCD9450BF8385755621A2DEAEEA0B1DE1AE04D</stationID>
+            <name>SÜPER FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>90.8</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607E505526A4CAF0C67C9579B9DE0ADAF67664515CA85DC1780BC78EAA24A23075C</stationID>
+            <name>TRT-FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>91.2</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>115910560756282CAC18B01D0EC3397BF5B75167B3A0B249DB684EAE9336C14A4FB71BAEE5</stationID>
+            <name>TATLISES</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>91.5</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>115910560734851591640F1FB0B7CC04962AD8C2EF2CF8097700FCB7B1567A3630B9031AD0</stationID>
+            <name>TGRT FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>92</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607B8C8EE6D45D70363E9D5FDEEA005097D8CFE64BCFB17C55CF4997EA999CF5DF6</stationID>
+            <name>RADYO EGE</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>92.7</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>115910560798C8FD6E5B3CFFCDAE6FF62B05835EBA661CDAD465511335D4CE222E5EDA796C</stationID>
+            <name>S SBS TÜRK</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>93</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>11591056070F3750DB792C0D8E7A82C170894000493F167FC34457EA674E4C5E918F36A3C7</stationID>
+            <name>CAN RADYO</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>93.3</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>115910560752926AA64A6B4EB4BF08023558E67F3934CA10086150EED0E6D724AB2D0D7F61</stationID>
+            <name>BATI RADYO</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>93.6</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607C2B3A02FE39ED42C6C500CC73BC8F3348B3B0384180C28440220F9BCFBC95EB0</stationID>
+            <name>RADYO HERKÜL</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>94.1</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>11591056073C37C0249646B9579684894FCBF257164C13EC62E959F45959314281D8292F84</stationID>
+            <name>DİNİ</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>94.4</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>115910560748518A040B4D49C702A8234BA8B2F7F7535D057275B82E35B774E7448977B1A8</stationID>
+            <name>TRT</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>94.7</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607E4D0917B5CC0C5C73B223F6F22D7027E88EC628B76F5714628215FDA0DC0B455</stationID>
+            <name>ROMANTİK RADYO</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>95.2</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>11591056071310D735CBD17F34FC9E0796AD7224AA70652439B4373BCA9FCCDC28C5C8F02A</stationID>
+            <name>NTV RADYO</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>95.7</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607C63D293FFC07F5D40C466CBC59E902DF12D5053986DF725ECA46E9C1E6A2B5B9</stationID>
+            <name>RADYO KLAS</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>96</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607E2AA531318CED2328BEFD30BE86DD8AB6C938D99A12E140D9DAFBF5FFFB5CA16</stationID>
+            <name>KRAL FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>96.2</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607F7EAB9637AC2DEC25EF87FC9C4BA3E6053E105E3267439970F74E23271C9A94F</stationID>
+            <name>KORDON FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>96.5</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>11591056075AE767CC1742EE410890A3434BF841DE4CCF8E04384C668F48FCF7A295EBB241</stationID>
+            <name>AKRA</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>96.9</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607E23C551FF00DA5D3B6900CFC96A53967A5A3D06B321E20F272B10D52B66D3A9D</stationID>
+            <name>METRO FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>97.2</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607C29AEC7E64F2BD94D8F90B69DDD92AEF6F8219916B6A4C1D6370AC3986BFE78A</stationID>
+            <name>RADYO ÖZMEN</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>97.7</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>11591056070AE78D419999974A013C8B1DE244AE6F8AB229050B4BFC3EF2A3C8E1818AECDD</stationID>
+            <name>RADYO ÇAĞ</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>98</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607A0554A722F050281FB640527CD582CD086A783D05314DF7DA99868FB51027C6E</stationID>
+            <name>BEST FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>98.4</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607261BF429647FEC9AF5F182BECE1ABFDE5E6841AB4751779F19AA4E44A48C2064</stationID>
+            <name>TRT RADYO 3</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>99.1</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607DC02FAA3C60B19BF7D463FE4E57FF6D1754E336F2EB5C09269A0C4CC9FADED7F</stationID>
+            <name>KAPİTAL RADYO</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>99.5</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>115910560718F41B55DA2C22C8CAFC7E646FB818DD5F3FA899E4B716D0B081E9783EF2EB1E</stationID>
+            <name>POWER FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>99.8</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607AC13C9B2E73B51979ADC9E07FE4E975B0EB34CEACEEDCE31338BD8CEAFC1565E</stationID>
+            <name>POWER FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>100</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>11591056070DE8C2C54FC76A0CE7A5804E004F6843E0993FF66783850FD14B5A27B9B06B00</stationID>
+            <name>TRT</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>100.5</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607B5377E3FDAE59A8798658C4F60F97AF71D8B83DCF21BB780E63EB31AAB6A9F6A</stationID>
+            <name>FM İZMİR 101</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>101</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607A2D18FDB42A8746E13B11517D5399E0C3E97810CA2137BE97264E6174A1C5F2A</stationID>
+            <name>RADYO 7</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>101.3</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>11591056070AE74CCF17E01A748B57E7FB9B90E1CBD7832D22D52808F73DB4B74FF73A1B90</stationID>
+            <name>BUCA FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>101.8</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>11591056078A9AA84174307D34A82FB4934AF5AA9FE8D127E21ED075EEDC81D3B10EF660C3</stationID>
+            <name>YAYIN</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>102</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607866A5463E5CDE0B629E0448E2019F9DF6BF9FF6CAB3E5CC01371F184743D6161</stationID>
+            <name>YAYIN</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>102.5</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607F161CDA7E8B7716584D06C42DCBC31ADEA54E41A89E5533C7C755FBD01457233</stationID>
+            <name>ÇAĞRI FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>102.8</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607E3DB104F21C6D584C2470DA6757609FCB5F041BC8DF37995D178CD501CD6E3E7</stationID>
+            <name>RADYO KLAS</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>103.1</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>11591056074FBE7B317764E18E47444F0C5605B7A4AA0E6E8C48E995846ED16A9D447F5DF4</stationID>
+            <name>KISS FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>104.3</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607813B68DE1187BC92E8443057F779259412E0F5C5A19B6F48234809F58ED80E74</stationID>
+            <name>RADYO SAHİL GÜVENLİK</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>104.7</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607FDD595CBC65F79B5308FA63D07B3492AF46849A8892EECF9D50907E5A49F75C3</stationID>
+            <name>RADYO MEGA</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>105</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607FD0B95759823566EF1414449C2B4A747605E9C9994A559E3DD3AE72B603ED397</stationID>
+            <name>YILDIZ - FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>105.3</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>11591056072E1066F35321AE34E12FEEF7B0EDBBB65CB33830D9026DFC936049285A0F7F90</stationID>
+            <name>RADYO İZMİR  1</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>105.5</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607F87CE144CC085AE076248CE7689A592571F142D16E26261229BE68BD99E69C0F</stationID>
+            <name>POWER TÜRK</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>105.7</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607957DCEA71BC3720EC830BA271A93149816AF306870516E1B4E5F62EFD22DC213</stationID>
+            <name>RADYO MAYDANOZ</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>106</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>115910560765128416B0A5B64C0F1ACD2416FA82F09BC57C729FA982D1333C43016761F906</stationID>
+            <name>RADYO 5</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>106.3</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607ABADBA5D050666CD5490144F32CC52C8DFCB1697610B885FA4B3E4ACA0489C5D</stationID>
+            <name>MARMARA FM</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>106.7</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607B430DF02D6E1DF350808F9BB1259A2DA0C0610EF236A642DC8EB6015DE926E2B</stationID>
+            <name>RADİO SPORT</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>107</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607DEFBE0652474F7C9279D33917322527B9B49AB91D0F5BA571B38BCFCEC5746D4</stationID>
+            <name>DEMOKRAT RADYO</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>107.3</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607D69CD982A9991158EF55CF55931CE256E36CB987D47EE8F96BAB7599DF323996</stationID>
+            <name>RADYO GÖKKUŞAĞI</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>107.6</frequency>
+        </FrequencyRadioStation>
+        <FrequencyRadioStation>
+            
+<stationID>1159105607C5C1733FC47A4D8400C1FE885FFF45C80A378DC50971AB47AA459209E7B188B9</stationID>
+            <name>RADYO DOKUZ EYLÜL</name>
+            <shortname></shortname>
+            <icon></icon>
+            <volumepreset>-1</volumepreset>
+            <frequency>108</frequency>
+        </FrequencyRadioStation>
+    </stationlist>
+</kradiorc>
