diff -Nur vbetool-1.1-old/vbetool.c vbetool-1.1/vbetool.c
--- vbetool-1.1-old/vbetool.c	2008-05-27 15:41:04.000000000 +0300
+++ vbetool-1.1/vbetool.c	2008-05-27 15:41:15.000000000 +0300
@@ -529,7 +529,7 @@
   r.edi = (unsigned long)(id-LRMI_base_addr()) & 0xf;
 
   if(sizeof(struct panel_id) != 32)
-    return fprintf(stderr, "oops: panel_id, sizeof struct panel_id != 32, it's %ld...\n", sizeof(struct panel_id)), 7;
+    return fprintf(stderr, "oops: panel_id, sizeof struct panel_id != 32, it's %u...\n", sizeof(struct panel_id)), 7;
 
   if(real_mode_int(0x10, &r))
     return fprintf(stderr, "Can't get panel id (vm86 failure)\n"), 8;
