diff -Nur urbanterror-4.1-1807-old//code/client/cl_curl.h urbanterror-4.1-1807/code/client/cl_curl.h
--- urbanterror-4.1-1807-old//code/client/cl_curl.h	2011-01-20 14:23:31.435000036 +0200
+++ urbanterror-4.1-1807/code/client/cl_curl.h	2011-01-20 14:26:43.533000271 +0200
@@ -36,7 +36,7 @@
 #define DEFAULT_CURL_LIB "libcurl.dylib"
 #else
 #define DEFAULT_CURL_LIB "libcurl.so.4"
-#define ALTERNATE_CURL_LIB "libcurl.so.3"
+#define ALTERNATE_CURL_LIB "libcurl.so"
 #endif
  
 #ifdef USE_LOCAL_HEADERS
