--- client/Makefile.orig	2009-05-30 21:46:49.098866808 +0300
+++ client/Makefile	2009-05-30 21:56:20.413989888 +0300
@@ -262,7 +262,7 @@
   SHLIBLDFLAGS=-shared $(LDFLAGS)
 
   THREAD_LDFLAGS=-lpthread
-  LDFLAGS=-ldl -lm
+  LDFLAGS+=-ldl -lm
 
   ifeq ($(USE_SDL),1)
     CLIENT_LDFLAGS=$(shell sdl-config --libs)
@@ -546,7 +546,7 @@
 
   THREAD_LDFLAGS=-lpthread
   # don't need -ldl (FreeBSD)
-  LDFLAGS=-lm
+  LDFLAGS+=-lm
 
   CLIENT_LDFLAGS =
 
@@ -1128,13 +1128,13 @@
 
 $(B)/ioUrbanTerror.$(ARCH)$(BINEXT): $(Q3OBJ) $(Q3POBJ) $(LIBSDLMAIN)
 	$(echo_cmd) "LD $@"
-	$(Q)$(CC) -o $@ $(Q3OBJ) $(Q3POBJ) $(CLIENT_LDFLAGS) \
-		$(LDFLAGS) $(LIBSDLMAIN) -lcurl
+	$(Q)$(CC) $(LDFLAGS) $(Q3OBJ) $(Q3POBJ) $(CLIENT_LDFLAGS) \
+		 $(LIBSDLMAIN) -lcurl -o $@
 
 $(B)/ioUrbanTerror-smp.$(ARCH)$(BINEXT): $(Q3OBJ) $(Q3POBJ_SMP) $(LIBSDLMAIN)
 	$(echo_cmd) "LD $@"
-	$(Q)$(CC) -o $@ $(Q3OBJ) $(Q3POBJ_SMP) $(CLIENT_LDFLAGS) \
-		$(THREAD_LDFLAGS) $(LDFLAGS) $(LIBSDLMAIN) -lcurl
+	$(Q)$(CC) $(LDFLAGS) $(Q3OBJ) $(Q3POBJ_SMP) $(CLIENT_LDFLAGS) \
+		$(THREAD_LDFLAGS) $(LIBSDLMAIN) -lcurl -o $@
 
 ifneq ($(strip $(LIBSDLMAIN)),)
 ifneq ($(strip $(LIBSDLMAINSRC)),)
