diff -Nur urbanterror-4.1-1807-old//code/ioq3-urt/ioq3-urt_Init.c urbanterror-4.1-1807/code/ioq3-urt/ioq3-urt_Init.c
--- urbanterror-4.1-1807-old//code/ioq3-urt/ioq3-urt_Init.c	2011-01-20 13:48:13.364000026 +0200
+++ urbanterror-4.1-1807/code/ioq3-urt/ioq3-urt_Init.c	2011-01-20 13:48:22.739999556 +0200
@@ -1,6 +1,6 @@
 //under GPL v2
 
-#include "SDL.h"
+#include <SDL/SDL.h>
 #include "ioq3-urt.h"
 #include "ioq3-urt_Init.h"
 #include "../client/snd_local.h" // for MAX_CHANNELS
