diff -Naur urbanterror-4.1.orig/client/code/q3_ui/ui_mods.c urbanterror-4.1/client/code/q3_ui/ui_mods.c
--- urbanterror-4.1.orig/client/code/q3_ui/ui_mods.c	2007-10-09 02:47:14.000000000 +0300
+++ urbanterror-4.1/client/code/q3_ui/ui_mods.c	2007-12-21 14:44:17.000000000 +0200
@@ -161,7 +161,7 @@
 
 	// always start off with baseq3
 	s_mods.list.numitems = 1;
-	s_mods.list.itemnames[0] = s_mods.descriptionList[0] = "Quake III Arena";
+	s_mods.list.itemnames[0] = s_mods.descriptionList[0] = "Urban Terror";
 	s_mods.fs_gameList[0] = "";
 
 	numdirs = trap_FS_GetFileList( "$modlist", "", dirlist, sizeof(dirlist) );
diff -Naur urbanterror-4.1.orig/client/code/qcommon/files.c urbanterror-4.1/client/code/qcommon/files.c
--- urbanterror-4.1.orig/client/code/qcommon/files.c	2007-12-16 17:27:50.000000000 +0200
+++ urbanterror-4.1/client/code/qcommon/files.c	2007-12-21 14:44:17.000000000 +0200
@@ -2799,7 +2799,7 @@
 		homePath = fs_basepath->string;
 	}
 	fs_homepath = Cvar_Get ("fs_homepath", homePath, CVAR_INIT );
-	fs_gamedirvar = Cvar_Get ("fs_game", "q3ut4", CVAR_INIT|CVAR_SYSTEMINFO );
+	fs_gamedirvar = Cvar_Get ("fs_game", "baseut4", CVAR_INIT|CVAR_SYSTEMINFO );
 	fs_restrict = Cvar_Get ("fs_restrict", "", CVAR_INIT );
 
 	// add search path elements in reverse priority order
diff -Naur urbanterror-4.1.orig/client/code/qcommon/q_shared.h urbanterror-4.1/client/code/qcommon/q_shared.h
--- urbanterror-4.1.orig/client/code/qcommon/q_shared.h	2007-12-16 17:37:24.000000000 +0200
+++ urbanterror-4.1/client/code/qcommon/q_shared.h	2007-12-21 14:44:17.000000000 +0200
@@ -36,7 +36,7 @@
 #define CONSOLE_WINDOW_ICON   "iourbanterror console"
 // 1.32 released 7-10-2002
 
-#define BASEGAME              "q3ut4"
+#define BASEGAME              "baseut4"
 
 #define MAX_TEAMNAME 32
 
diff -Naur urbanterror-4.1.orig/client/code/unix/unix_shared.c urbanterror-4.1/client/code/unix/unix_shared.c
--- urbanterror-4.1.orig/client/code/unix/unix_shared.c	2007-10-09 02:47:16.000000000 +0300
+++ urbanterror-4.1/client/code/unix/unix_shared.c	2007-12-21 14:44:17.000000000 +0200
@@ -409,7 +409,7 @@
 #ifdef MACOS_X
 		Q_strcat(homePath, sizeof(homePath), "/Library/Application Support/Quake3");
 #else
-		Q_strcat(homePath, sizeof(homePath), "/.q3a");
+		Q_strcat(homePath, sizeof(homePath), "/.urbanterror");
 #endif
 		if (mkdir(homePath, 0777)) {
 			if (errno != EEXIST) 
diff -Naur urbanterror-4.1.orig/server/code/q3_ui/ui_mods.c urbanterror-4.1/server/code/q3_ui/ui_mods.c
--- urbanterror-4.1.orig/server/code/q3_ui/ui_mods.c	2007-11-18 19:53:48.000000000 +0200
+++ urbanterror-4.1/server/code/q3_ui/ui_mods.c	2007-12-21 14:46:30.000000000 +0200
@@ -125,7 +125,7 @@
 
 	// always start off with baseq3
 	s_mods.list.numitems = 1;
-	s_mods.list.itemnames[0] = s_mods.descriptionList[0] = "Quake III Arena";
+	s_mods.list.itemnames[0] = s_mods.descriptionList[0] = "Urban Terror";
 	s_mods.fs_gameList[0] = "";
 
 	numdirs = trap_FS_GetFileList( "$modlist", "", dirlist, sizeof(dirlist) );
diff -Naur urbanterror-4.1.orig/server/code/qcommon/files.c urbanterror-4.1/server/code/qcommon/files.c
--- urbanterror-4.1.orig/server/code/qcommon/files.c	2007-12-16 19:01:20.000000000 +0200
+++ urbanterror-4.1/server/code/qcommon/files.c	2007-12-21 14:46:51.000000000 +0200
@@ -2728,7 +2728,7 @@
 		homePath = fs_basepath->string;
 	}
 	fs_homepath = Cvar_Get ("fs_homepath", homePath, CVAR_INIT );
-	fs_gamedirvar = Cvar_Get ("fs_game", "q3ut4", CVAR_INIT|CVAR_SYSTEMINFO );
+	fs_gamedirvar = Cvar_Get ("fs_game", "baseut4", CVAR_INIT|CVAR_SYSTEMINFO );
 
 	// add search path elements in reverse priority order
 	if (fs_basepath->string[0]) {
diff -Naur urbanterror-4.1.orig/server/code/qcommon/q_shared.h urbanterror-4.1/server/code/qcommon/q_shared.h
--- urbanterror-4.1.orig/server/code/qcommon/q_shared.h	2007-12-16 19:01:20.000000000 +0200
+++ urbanterror-4.1/server/code/qcommon/q_shared.h	2007-12-21 14:47:06.000000000 +0200
@@ -39,7 +39,7 @@
 #define CLIENT_WINDOW_MIN_TITLE "ioUrT"
 // 1.32 released 7-10-2002
 
-#define BASEGAME              "q3ut4"
+#define BASEGAME              "baseut4"
 
 #define MAX_TEAMNAME 32
 
diff -Naur urbanterror-4.1.orig/server/code/sys/sys_unix.c urbanterror-4.1/server/code/sys/sys_unix.c
--- urbanterror-4.1.orig/server/code/sys/sys_unix.c	2007-12-03 02:37:34.000000000 +0200
+++ urbanterror-4.1/server/code/sys/sys_unix.c	2007-12-21 14:47:47.000000000 +0200
@@ -55,7 +55,7 @@
 #ifdef MACOS_X
 			Q_strcat( homePath, sizeof( homePath ), "/Library/Application Support/Quake3" );
 #else
-			Q_strcat( homePath, sizeof( homePath ), "/.q3a" );
+			Q_strcat( homePath, sizeof( homePath ), "/.urbanterror" );
 #endif
 			if( mkdir( homePath, 0777 ) )
 			{
