diff -up firefox-29.0/mozilla-release/browser/app/profile/firefox.js.addons firefox-29.0/mozilla-release/browser/app/profile/firefox.js
--- firefox-29.0/mozilla-release/browser/app/profile/firefox.js.addons	2014-04-22 07:27:34.000000000 +0200
+++ firefox-29.0/mozilla-release/browser/app/profile/firefox.js	2014-04-22 15:49:04.953173027 +0200
@@ -65,7 +65,8 @@ pref("extensions.hotfix.certs.1.sha1Fing
 
 // Disable add-ons that are not installed by the user in all scopes by default.
 // See the SCOPE constants in AddonManager.jsm for values to use here.
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 0);
+pref("extensions.showMismatchUI", false);
 
 // Dictionary download preference
 pref("browser.dictionaries.download.url", "https://addons.mozilla.org/%LOCALE%/firefox/dictionaries/");
diff -up firefox-29.0/mozilla-release/toolkit/mozapps/extensions/XPIProvider.jsm.addons firefox-29.0/mozilla-release/toolkit/mozapps/extensions/XPIProvider.jsm
--- firefox-29.0/mozilla-release/toolkit/mozapps/extensions/XPIProvider.jsm.addons	2014-04-22 15:49:04.955173021 +0200
+++ firefox-29.0/mozilla-release/toolkit/mozapps/extensions/XPIProvider.jsm	2014-04-22 15:50:34.711891888 +0200
@@ -2051,8 +2051,8 @@ var XPIProvider = {
         this.showUpgradeUI();
         flushCaches = true;
       }
-      else if (aAppChanged === undefined) {
-        // For new profiles we will never need to show the add-on selection UI
+      else {
+        // For all profiles we will never need to show the add-on selection UI
         Services.prefs.setBoolPref(PREF_SHOWN_SELECTION_UI, true);
       }
 

