--- mediastreamer2/src/videofilters/nowebcam.c.image	2009-07-13 11:25:33.000000000 +0200
+++ mediastreamer2/src/videofilters/nowebcam.c	2009-09-19 03:16:14.000000000 +0200
@@ -1668,9 +1668,9 @@
 mblk_t *ms_load_nowebcam(MSVideoSize *reqsize, int idx){
 	char tmp[256];
 	if (idx<0)
-		snprintf(tmp, sizeof(tmp), "%s/images/%s.jpg", PACKAGE_DATA_DIR, NOWEBCAM_JPG);
+		snprintf(tmp, sizeof(tmp), "%s/images/linphone/%s.jpg", PACKAGE_DATA_DIR, NOWEBCAM_JPG);
 	else
-		snprintf(tmp, sizeof(tmp), "%s/images/%s%i.jpg", PACKAGE_DATA_DIR, NOWEBCAM_JPG, idx);
+		snprintf(tmp, sizeof(tmp), "%s/images/linphone/%s%i.jpg", PACKAGE_DATA_DIR, NOWEBCAM_JPG, idx);
 	return ms_load_jpeg_as_yuv(tmp,reqsize);
 }
 
--- mediastreamer2/src/Makefile.am.image-dir	2013-05-30 16:09:06.000000000 +0800
+++ mediastreamer2/src/Makefile.am	2013-06-28 11:42:41.023215683 +0800
@@ -363,7 +363,7 @@
 
 AM_OBJCFLAGS=$(AM_CFLAGS)
 
-imgdir=$(datadir)/images/
+imgdir=$(datadir)/images/linphone/
 
 img_DATA=voip/nowebcamCIF.jpg
 
