diff -up libjingle-0.5.1/talk/base/unixfilesystem.cc.missingheader libjingle-0.5.1/talk/base/unixfilesystem.cc
--- libjingle-0.5.1/talk/base/unixfilesystem.cc.missingheader	2010-11-19 10:59:11.465967001 -0500
+++ libjingle-0.5.1/talk/base/unixfilesystem.cc	2010-11-19 11:02:12.027967004 -0500
@@ -25,9 +25,8 @@
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include "talk/base/unixfilesystem.h"
-
 #include <errno.h>
+#include <time.h>
 #include <fcntl.h>
 #include <stdlib.h>
 #include <unistd.h>
@@ -56,6 +55,7 @@
 #include <algorithm>
 #endif
 
+#include "talk/base/unixfilesystem.h"
 #include "talk/base/fileutils.h"
 #include "talk/base/pathutils.h"
 #include "talk/base/stream.h"
