diff -Naur libjingle-0.3.12/talk/base/Makefile.am libjingle-0.3.12.oden/talk/base/Makefile.am
--- libjingle-0.3.12/talk/base/Makefile.am	2008-09-16 15:55:22.000000000 +0200
+++ libjingle-0.3.12.oden/talk/base/Makefile.am	2008-10-17 21:40:42.000000000 +0200
@@ -31,7 +31,7 @@
                             signalthread.cc \
                             socketpool.cc
 
-libjinglebase_la_LIBADD = -lrt
+libjinglebase_la_LIBADD = -lrt -lpthread -lssl -lcrypto
 
 libjinglebase_includedir = $(includedir)/libjingle-@LIBJINGLE_MAJORMINOR@/talk/base
 libjinglebase_include_HEADERS = asyncfile.h \
diff -Naur libjingle-0.3.12/talk/p2p/base/Makefile.am libjingle-0.3.12.oden/talk/p2p/base/Makefile.am
--- libjingle-0.3.12/talk/p2p/base/Makefile.am	2008-09-16 15:55:22.000000000 +0200
+++ libjingle-0.3.12.oden/talk/p2p/base/Makefile.am	2008-10-17 21:42:23.000000000 +0200
@@ -38,7 +38,7 @@
 								   sessionresponsemessage.h \
 								   sessionmessagefactory.h
 
-libjinglep2pbase_la_LIBADD = -lssl -lcrypto
+libjinglep2pbase_la_LIBADD = ../../base/libjinglebase.la -lpthread
 
 AM_CPPFLAGS = -DPOSIX
 
diff -Naur libjingle-0.3.12/talk/p2p/client/Makefile.am libjingle-0.3.12.oden/talk/p2p/client/Makefile.am
--- libjingle-0.3.12/talk/p2p/client/Makefile.am	2008-09-16 15:55:22.000000000 +0200
+++ libjingle-0.3.12.oden/talk/p2p/client/Makefile.am	2008-10-17 21:50:37.000000000 +0200
@@ -5,6 +5,8 @@
 	socketclient.cc \
 	httpportallocator.cc
 
+libjinglep2pclient_la_LIBADD = ../../base/libjinglebase.la -lpthread
+
 libjinglep2pclient_includedir  = $(includedir)/libjingle-@LIBJINGLE_MAJORMINOR@/talk/p2p/client
 libjinglep2pclient_include_HEADERS = basicportallocator.h \
 	socketmonitor.h \
