diff -up libjingle-0.6.14/configure.ac.SANITY libjingle-0.6.14/configure.ac
--- libjingle-0.6.14/configure.ac.SANITY	2012-06-11 15:19:25.240335579 -0400
+++ libjingle-0.6.14/configure.ac	2012-06-11 15:23:05.590338885 -0400
@@ -0,0 +1,114 @@
+AC_INIT([libjingle], [0.6.0], [google-talk-open@googlegroups.com])
+AC_CANONICAL_SYSTEM
+AM_CONFIG_HEADER(config.h)
+AM_INIT_AUTOMAKE([dist-zip])
+AC_PROG_CC
+AC_PROG_CXX
+AM_PROG_LIBTOOL
+LIBTOOL="$LIBTOOL --silent"
+AC_PROG_INSTALL
+AC_DEFINE(PRODUCTION_BUILD, 1, [Build as a production build])
+AC_DEFINE(PRODUCTION, 1, [Build as a production build])
+AC_DEFINE(POSIX, 1, [If we're using configure, we're on POSIX])
+AC_DEFINE(FEATURE_ENABLE_VOICEMAIL, 1, [voice mail])
+AC_DEFINE(LOGGING, 1, [Logging])
+
+LIBJINGLE_MAJORMINOR=0.6
+AC_SUBST(LIBJINGLE_MAJORMINOR)
+
+LIBJINGLE_LIBRARY_VERSION=1:0:0
+AC_SUBST(LIBJINGLE_LIBRARY_VERSION)
+
+HAVE_EXPAT=no
+AC_CHECK_LIB(expat, XML_ParserCreate, HAVE_EXPAT="yes")
+if test "x$HAVE_EXPAT" = xyes ; then
+  EXPAT_LIBS="-lexpat"
+  AC_SUBST(EXPAT_LIBS)
+else
+  AC_ERROR([Expat is required to build libjingle. You can get it from http://expat.sourceforge.net/])
+fi
+
+if test `uname -s` = Linux ; then
+  AC_DEFINE(LINUX, 1, [Building on Linux])
+  HAVE_ALSA=no
+  AC_CHECK_LIB(asound, snd_pcm_open, HAVE_ALSA="yes")
+  if test "x$HAVE_ALSA" = xyes ; then
+    ALSA_LIBS="-lasound"
+    AC_SUBST(ALSA_LIBS)
+    AC_DEFINE(HAVE_ALSA, 1, [Using ALSA])
+  else
+    AC_ERROR([libasound is required to build libjingle. You can get it from http://www.alsa-project.org/])
+  fi
+
+  AC_CHECK_HEADER(openssl/ssl.h, HAVE_OPENSSL_SSL_H=yes, HAVE_OPENSSL_SSL_H=no)
+  if test x$HAVE_OPENSSL_SSL_H = xyes; then
+    AC_DEFINE(HAVE_OPENSSL_SSL_H, 1, [Found openssl/ssl.h])
+    PKG_CHECK_MODULES(OPENSSL, openssl >= 0.9.7g, HAVE_OPENSSL=yes, HAVE_OPENSSL=no)
+    AC_SUBST(OPENSSL_CFLAGS)
+    AC_SUBST(OPENSSL_LIBS)
+    AC_DEFINE(SSL_USE_OPENSSL, 1, [Using OpenSSL])
+    AC_DEFINE(FEATURE_ENABLE_SSL, 1, [SSL Enabled])
+    CPPFLAGS+="$OPENSSL_CFLAGS"
+    LDFLAGS+=" $OPENSSL_LIBS"
+  else
+    AC_DEFINE(HAVE_OPENSSL_SSL_H, 0, [Did not find openssl/ssl.h])
+  fi
+
+  PKG_CHECK_MODULES(GTK2, gtk+-2.0, enable_gtk=yes, enable_gtk=no)
+  if test x$enable_gtk = xno ; then
+    AC_ERROR([GTK2 is required to build libjingle.])
+  else
+    AC_SUBST(GTK2_CFLAGS)
+    AC_SUBST(GTK2_LIBS)
+    AC_DEFINE(HAVE_GTK2, 1, [GTK2 present])
+  fi
+fi
+
+PKG_CHECK_MODULES(LIBSRTP, libsrtp >= 1.4.4, enable_srtp=yes, enable_srtp=no)
+if test x$enable_srtp = xno ; then
+        HAVE_SRTP=no
+else
+        AC_SUBST(LIBSRTP_CFLAGS)
+       	AC_SUBST(LIBSRTP_LIBS)
+        AC_DEFINE(HAVE_SRTP, 1, [libSRTP provides SRTP supprt])
+fi
+
+PKG_CHECK_MODULES(XRENDER, xrender, enable_xrender=yes, enable_xrender=no)
+if test x$enable_xrender = xno ; then
+        AC_ERROR([xrender is required to build libjingle.])
+else
+        AC_SUBST(XRENDER_CFLAGS)
+        AC_SUBST(XRENDER_LIBS)
+        AC_DEFINE(HAVE_XRENDER, 1, [xrender present])
+fi
+
+PKG_CHECK_MODULES(XCOMPOSITE, xcomposite, enable_xcomposite=yes, enable_xcomposite=no)
+if test x$enable_xcomposite = xno ; then
+        AC_ERROR([xcomposite is required to build libjingle.])
+else
+	AC_SUBST(XCOMPOSITE_CFLAGS)
+	AC_SUBST(XCOMPOSITE_LIBS)
+	AC_DEFINE(HAVE_XCOMPOSITE, 1, [xcomposite present])
+fi
+
+AC_OUTPUT([Makefile
+          talk/Makefile
+          talk/base/Makefile
+          talk/examples/Makefile
+          talk/examples/login/Makefile
+          talk/examples/call/Makefile
+          talk/p2p/Makefile
+          talk/p2p/base/Makefile
+          talk/p2p/client/Makefile
+          talk/session/Makefile
+          talk/session/tunnel/Makefile
+          talk/session/phone/Makefile
+          talk/sound/Makefile
+          talk/xmllite/Makefile
+          talk/xmpp/Makefile
+          pkgconfig/Makefile
+          pkgconfig/jinglebase.pc
+          pkgconfig/jinglep2p.pc
+          pkgconfig/jinglesessiontunnel.pc
+          pkgconfig/jinglesessionphone.pc
+        ])
diff -up libjingle-0.6.14/Makefile.am.SANITY libjingle-0.6.14/Makefile.am
--- libjingle-0.6.14/Makefile.am.SANITY	2012-06-11 15:19:25.240335579 -0400
+++ libjingle-0.6.14/Makefile.am	2012-06-11 15:19:25.240335579 -0400
@@ -0,0 +1,4 @@
+SUBDIRS=talk pkgconfig
+
+configdir = $(includedir)/libjingle-@LIBJINGLE_MAJORMINOR@/talk
+config_HEADERS = config.h
diff -up libjingle-0.6.14/pkgconfig/jinglebase.pc.in.SANITY libjingle-0.6.14/pkgconfig/jinglebase.pc.in
--- libjingle-0.6.14/pkgconfig/jinglebase.pc.in.SANITY	2012-06-11 15:19:25.241335579 -0400
+++ libjingle-0.6.14/pkgconfig/jinglebase.pc.in	2012-06-11 15:19:25.241335579 -0400
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: jinglebase
+Description: Jingle Base Library
+Requires:
+Version: @VERSION@
+Libs: -L${libdir} -lpthread -ljinglebase
+Cflags: -I${includedir}/libjingle-@LIBJINGLE_MAJORMINOR@ -DPOSIX
+
diff -up libjingle-0.6.14/pkgconfig/jinglep2p.pc.in.SANITY libjingle-0.6.14/pkgconfig/jinglep2p.pc.in
--- libjingle-0.6.14/pkgconfig/jinglep2p.pc.in.SANITY	2012-06-11 15:19:25.241335579 -0400
+++ libjingle-0.6.14/pkgconfig/jinglep2p.pc.in	2012-06-11 15:19:25.241335579 -0400
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: jinglep2p
+Description: Jingle P2P Library
+Requires:
+Version: @VERSION@
+Libs: -L${libdir} -ljinglep2pbase -ljinglep2pclient -ljinglexmllite -ljinglexmpp -lexpat
+Cflags: -I${includedir}/libjingle-@LIBJINGLE_MAJORMINOR@ -DPOSIX
+
diff -up libjingle-0.6.14/pkgconfig/jinglesessionphone.pc.in.SANITY libjingle-0.6.14/pkgconfig/jinglesessionphone.pc.in
--- libjingle-0.6.14/pkgconfig/jinglesessionphone.pc.in.SANITY	2012-06-11 15:19:25.241335579 -0400
+++ libjingle-0.6.14/pkgconfig/jinglesessionphone.pc.in	2012-06-11 15:19:25.241335579 -0400
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: jinglesessionphone
+Description: Jingle Session Phone Library
+Requires:
+Version: @VERSION@
+Libs: -L${libdir} -lpthread -ljinglesessionphone -lsrtp
+Cflags: -I${includedir}/libjingle-@LIBJINGLE_MAJORMINOR@ -DPOSIX
+
diff -up libjingle-0.6.14/pkgconfig/jinglesessiontunnel.pc.in.SANITY libjingle-0.6.14/pkgconfig/jinglesessiontunnel.pc.in
--- libjingle-0.6.14/pkgconfig/jinglesessiontunnel.pc.in.SANITY	2012-06-11 15:19:25.242335579 -0400
+++ libjingle-0.6.14/pkgconfig/jinglesessiontunnel.pc.in	2012-06-11 15:19:25.242335579 -0400
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: jinglesessiontunnel
+Description: Jingle Session Tunnel Library
+Requires:
+Version: @VERSION@
+Libs: -L${libdir} -lpthread -ljinglesessiontunnel
+Cflags: -I${includedir}/libjingle-@LIBJINGLE_MAJORMINOR@ -DPOSIX
+
diff -up libjingle-0.6.14/pkgconfig/Makefile.am.SANITY libjingle-0.6.14/pkgconfig/Makefile.am
--- libjingle-0.6.14/pkgconfig/Makefile.am.SANITY	2012-06-11 15:19:25.242335579 -0400
+++ libjingle-0.6.14/pkgconfig/Makefile.am	2012-06-11 15:19:25.242335579 -0400
@@ -0,0 +1,20 @@
+### all of the standard pc files we need to generate
+pcfiles = jinglebase-@LIBJINGLE_MAJORMINOR@.pc jinglep2p-@LIBJINGLE_MAJORMINOR@.pc jinglesessiontunnel-@LIBJINGLE_MAJORMINOR@.pc jinglesessionphone-@LIBJINGLE_MAJORMINOR@.pc
+
+all-local: $(pcfiles)
+
+### how to generate pc files
+%-@LIBJINGLE_MAJORMINOR@.pc: %.pc
+	cp $< $@
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = $(pcfiles)
+
+EXTRA_DIST =							\
+	jinglebase.pc.in					\
+	jinglep2p.pc.in						\
+	jinglesessionphone.pc.in				\
+	jinglesessiontunnel.pc.in
+
+CLEANFILES = $(pcfiles)
+
diff -up libjingle-0.6.14/talk/base/Makefile.am.SANITY libjingle-0.6.14/talk/base/Makefile.am
--- libjingle-0.6.14/talk/base/Makefile.am.SANITY	2012-06-11 15:19:25.243335579 -0400
+++ libjingle-0.6.14/talk/base/Makefile.am	2012-06-11 15:19:25.243335579 -0400
@@ -0,0 +1,208 @@
+lib_LTLIBRARIES = libjinglebase.la
+
+libjinglebase_la_SOURCES = asyncfile.cc \
+                            asynchttprequest.cc \
+                            asyncsocket.cc \
+                            asynctcpsocket.cc \
+                            asyncudpsocket.cc \
+                            autodetectproxy.cc \
+                            bandwidthsmoother.cc \
+                            base64.cc \
+                            basicpacketsocketfactory.cc \
+                            bytebuffer.cc \
+                            checks.cc \
+                            common.cc \
+                            cpuid.cc \
+                            cpumonitor.cc \
+                            dbus.cc \
+                            diskcache.cc \
+                            event.cc \
+                            filelock.cc \
+                            fileutils.cc \
+                            firewallsocketserver.cc \
+                            flags.cc \
+                            helpers.cc \
+                            host.cc \
+                            httpbase.cc \
+                            httpclient.cc \
+                            httpcommon.cc \
+                            httprequest.cc \
+                            httpserver.cc \
+                            ipaddress.cc \
+                            latebindingsymboltable.cc \
+                            libdbusglibsymboltable.cc \
+                            linux.cc \
+                            linuxfdwalk.cc \
+                            linuxwindowpicker.cc \
+                            logging.cc \
+                            md5c.c \
+                            messagehandler.cc \
+                            messagequeue.cc \
+                            multipart.cc \
+                            natserver.cc \
+                            natsocketfactory.cc \
+                            nattypes.cc \
+                            nethelpers.cc \
+                            network.cc \
+                            openssladapter.cc \
+                            opensslidentity.cc \
+                            opensslstreamadapter.cc \
+                            optionsfile.cc \
+                            pathutils.cc \
+                            physicalsocketserver.cc \
+                            posix.cc \
+                            proxydetect.cc \
+                            proxyinfo.cc \
+                            proxyserver.cc \
+                            ratetracker.cc \
+                            sharedexclusivelock.cc \
+                            signalthread.cc \
+                            socketadapters.cc \
+                            socketaddress.cc \
+                            socketaddresspair.cc \
+                            socketpool.cc \
+                            socketstream.cc \
+                            ssladapter.cc \
+                            sslidentity.cc \
+                            sslsocketfactory.cc \
+                            sslstreamadapter.cc \
+                            stream.cc \
+                            stringdigest.cc \
+                            stringencode.cc \
+                            stringutils.cc \
+                            systeminfo.cc \
+                            task.cc \
+                            taskparent.cc \
+                            taskrunner.cc \
+                            testclient.cc \
+                            thread.cc \
+                            timeutils.cc \
+                            timing.cc \
+                            transformadapter.cc \
+                            unixfilesystem.cc \
+                            urlencode.cc \
+                            versionparsing.cc \
+                            virtualsocketserver.cc \
+                            worker.cc
+
+libjinglebase_la_LIBADD = -lrt $(XRENDER_LIBS) $(XCOMPOSITE_LIBS)
+
+libjinglebase_la_LDFLAGS = -version-info $(LIBJINGLE_LIBRARY_VERSION)
+
+libjinglebase_includedir = $(includedir)/libjingle-@LIBJINGLE_MAJORMINOR@/talk/base
+
+libjinglebase_include_HEADERS = asyncfile.h \
+                            asynchttprequest.h \
+                            asyncpacketsocket.h \
+                            asyncsocket.h \
+                            asynctcpsocket.h \
+                            asyncudpsocket.h \
+                            atomicops.h \
+                            autodetectproxy.h \
+                            bandwidthsmoother.h \
+                            base64.h \
+                            basicdefs.h \
+                            basicpacketsocketfactory.h \
+                            basictypes.h \
+                            buffer.h \
+                            bytebuffer.h \
+                            byteorder.h \
+                            checks.h \
+                            common.h \
+                            constructormagic.h \
+                            cpuid.h \
+                            cpumonitor.h \
+                            criticalsection.h \
+                            cryptstring.h \
+                            dbus.h \
+                            diskcache.h \
+                            event.h \
+                            fakenetwork.h \
+                            faketaskrunner.h \
+                            filelock.h \
+                            fileutils.h \
+                            fileutils_mock.h \
+                            firewallsocketserver.h \
+                            flags.h \
+                            gunit.h \
+                            helpers.h \
+                            host.h \
+                            httpbase.h \
+                            httpclient.h \
+                            httpcommon-inl.h \
+                            httpcommon.h \
+                            httprequest.h \
+                            httpserver.h \
+                            ipaddress.h \
+                            latebindingsymboltable.h \
+                            libdbusglibsymboltable.h \
+                            linked_ptr.h \
+                            linux.h \
+                            linuxfdwalk.h \
+                            logging.h \
+                            mathutils.h \
+                            md5.h \
+                            messagehandler.h \
+                            messagequeue.h \
+                            multipart.h \
+                            natserver.h \
+                            natsocketfactory.h \
+                            nattypes.h \
+                            nethelpers.h \
+                            network.h \
+                            openssladapter.h \
+                            opensslidentity.h \
+                            opensslstreamadapter.h \
+                            packetsocketfactory.h \
+                            pathutils.h \
+                            physicalsocketserver.h \
+                            posix.h \
+                            proxydetect.h \
+                            proxyinfo.h \
+                            proxyserver.h \
+                            ratetracker.h \
+                            refcount.h \
+                            referencecountedsingletonfactory.h \
+                            rollingaccumulator.h \
+                            scoped_autorelease_pool.h \
+                            scoped_ptr.h \
+                            scoped_ref_ptr.h \
+                            sec_buffer.h \
+                            sharedexclusivelock.h \
+                            signalthread.h \
+                            sigslot.h \
+                            sigslotrepeater.h \
+                            socket.h \
+                            socketadapters.h \
+                            socketaddress.h \
+                            socketaddresspair.h \
+                            socketfactory.h \
+                            socketpool.h \
+                            socketserver.h \
+                            socketstream.h \
+                            ssladapter.h \
+                            sslidentity.h \
+                            sslroots.h \
+                            sslsocketfactory.h \
+                            sslstreamadapter.h \
+                            stream.h \
+                            stringdigest.h \
+                            stringencode.h \
+                            stringutils.h \
+                            systeminfo.h \
+                            task.h \
+                            taskparent.h \
+                            taskrunner.h \
+                            thread.h \
+                            timeutils.h \
+                            timing.h \
+                            transformadapter.h \
+                            unixfilesystem.h \
+                            urlencode.h \
+                            versionparsing.h \
+                            virtualsocketserver.h \
+                            window.h \
+                            windowpicker.h \
+                            worker.h
+
+AM_CXXFLAGS = -DPOSIX -DLINUX -DFEATURE_ENABLE_VOICEMAIL $(XRENDER_CFLAGS) $(XCOMPOSITE_CFLAGS)
diff -up libjingle-0.6.14/talk/examples/call/Makefile.am.SANITY libjingle-0.6.14/talk/examples/call/Makefile.am
--- libjingle-0.6.14/talk/examples/call/Makefile.am.SANITY	2012-06-11 15:19:25.243335579 -0400
+++ libjingle-0.6.14/talk/examples/call/Makefile.am	2012-06-11 15:19:25.243335579 -0400
@@ -0,0 +1,23 @@
+bin_PROGRAMS = call
+call_CXXFLAGS = $(AM_CXXFLAGS)
+call_SOURCES = call_main.cc \
+               callclient.cc \
+               console.cc \
+               discoitemsquerytask.cc \
+               friendinvitesendtask.cc \
+               mucinviterecvtask.cc \
+               mucinvitesendtask.cc \
+               presenceouttask.cc \
+               presencepushtask.cc \
+               voicemailjidrequester.cc
+noinst_HEADERS = callclient.h  console.h
+call_LDADD = $(top_srcdir)/talk/examples/login/libxmpphelp.la \
+               $(top_srcdir)/talk/session/phone/libjinglesessionphone.la \
+               $(top_srcdir)/talk/p2p/client/libjinglep2pclient.la \
+               $(top_srcdir)/talk/p2p/base/libjinglep2pbase.la \
+               $(top_srcdir)/talk/xmpp/libjinglexmpp.la \
+               $(top_srcdir)/talk/xmllite/libjinglexmllite.la \
+               $(top_srcdir)/talk/base/libjinglebase.la \
+               $(EXPAT_LIBS) $(OPENSSL_LIBS) $(ALSA_LIBS) $(LIBSRTP_LIBS) -lpthread
+AM_CPPFLAGS  = -DPOSIX
+
diff -up libjingle-0.6.14/talk/examples/login/Makefile.am.SANITY libjingle-0.6.14/talk/examples/login/Makefile.am
--- libjingle-0.6.14/talk/examples/login/Makefile.am.SANITY	2012-06-11 15:19:25.243335579 -0400
+++ libjingle-0.6.14/talk/examples/login/Makefile.am	2012-06-11 15:19:25.243335579 -0400
@@ -0,0 +1,24 @@
+noinst_LTLIBRARIES= libxmpphelp.la
+libxmpphelp_la_SOURCES = xmppsocket.cc \
+                         xmppauth.cc \
+                         xmpppump.cc
+
+noinst_HEADERS = xmppauth.h \
+                 xmpppump.h \
+                 xmppsocket.h \
+                 xmppthread.h \
+                 jingleinfotask.h \
+                 presenceouttask.h \
+                 presencepushtask.h \
+                 status.h
+
+bin_PROGRAMS = login
+login_CXXFLAGS = $(AM_CXXFLAGS)
+login_SOURCES = login_main.cc xmppthread.cc
+login_LDADD =  $(top_srcdir)/talk/xmpp/libjinglexmpp.la \
+               $(top_srcdir)/talk/xmllite/libjinglexmllite.la \
+               $(top_srcdir)/talk/base/libjinglebase.la \
+               $(top_srcdir)/talk/examples/login/libxmpphelp.la \
+               $(EXPAT_LIBS) $(OPENSSL_LIBS) -lpthread
+AM_CPPFLAGS  = -DPOSIX
+
diff -up libjingle-0.6.14/talk/examples/Makefile.am.SANITY libjingle-0.6.14/talk/examples/Makefile.am
--- libjingle-0.6.14/talk/examples/Makefile.am.SANITY	2012-06-11 15:19:25.243335579 -0400
+++ libjingle-0.6.14/talk/examples/Makefile.am	2012-06-11 15:19:25.243335579 -0400
@@ -0,0 +1 @@
+SUBDIRS=login call
diff -up libjingle-0.6.14/talk/Makefile.am.SANITY libjingle-0.6.14/talk/Makefile.am
--- libjingle-0.6.14/talk/Makefile.am.SANITY	2012-06-11 15:19:25.243335579 -0400
+++ libjingle-0.6.14/talk/Makefile.am	2012-06-11 15:19:25.243335579 -0400
@@ -0,0 +1 @@
+SUBDIRS=base xmllite xmpp p2p session sound
diff -up libjingle-0.6.14/talk/p2p/base/Makefile.am.SANITY libjingle-0.6.14/talk/p2p/base/Makefile.am
--- libjingle-0.6.14/talk/p2p/base/Makefile.am.SANITY	2012-06-11 15:19:25.243335579 -0400
+++ libjingle-0.6.14/talk/p2p/base/Makefile.am	2012-06-11 15:19:25.243335579 -0400
@@ -0,0 +1,82 @@
+lib_LTLIBRARIES = libjinglep2pbase.la
+
+libjinglep2pbase_la_SOURCES =  constants.cc \
+                               p2ptransport.cc \
+                               p2ptransportchannel.cc \
+                               parsing.cc \
+                               port.cc \
+                               portallocator.cc \
+                               portallocatorsessionproxy.cc \
+                               portproxy.cc \
+                               pseudotcp.cc \
+                               relayport.cc \
+                               relayserver.cc \
+                               rawtransport.cc \
+                               rawtransportchannel.cc \
+                               session.cc \
+                               sessiondescription.cc \
+                               sessionmanager.cc \
+                               sessionmessages.cc \
+                               stun.cc \
+                               stunport.cc \
+                               stunrequest.cc \
+                               stunserver.cc \
+                               tcpport.cc \
+                               transport.cc \
+                               transportchannel.cc \
+                               transportchannelproxy.cc \
+                               udpport.cc
+
+libjinglep2pbase_includedir  = $(includedir)/libjingle-@LIBJINGLE_MAJORMINOR@/talk/p2p/base
+libjinglep2pbase_include_HEADERS = candidate.h \
+                               common.h \
+                               constants.h \
+                               fakesession.h \
+                               p2ptransportchannel.h \
+                               p2ptransport.h \
+                               parsing.h \
+                               portallocator.h \
+                               portallocatorsessionproxy.h \
+                               port.h \
+                               portproxy.h \
+                               pseudotcp.h \
+                               rawtransportchannel.h \
+                               rawtransport.h \
+                               relayport.h \
+                               relayserver.h \
+                               sessionclient.h \
+                               sessiondescription.h \
+                               session.h \
+                               sessionid.h \
+                               sessionmanager.h \
+                               sessionmessages.h \
+                               stun.h \
+                               stunport.h \
+                               stunrequest.h \
+                               stunserver.h \
+                               tcpport.h \
+                               testrelayserver.h \
+                               teststunserver.h \
+                               transportchannel.h \
+                               transportchannelimpl.h \
+                               transportchannelproxy.h \
+                               transport.h \
+                               udpport.h
+
+libjinglep2pbase_la_LDFLAGS = -version-info $(LIBJINGLE_LIBRARY_VERSION)
+libjinglep2pbase_la_LIBADD = $(top_srcdir)/talk/xmllite/libjinglexmllite.la \
+                             $(top_srcdir)/talk/xmpp/libjinglexmpp.la
+
+AM_CPPFLAGS = -DPOSIX -DENABLE_DEBUG -D_DEBUG -g -DFEATURE_ENABLE_VOICEMAIL
+
+P2PLIBS = libjinglep2pbase.la ../../base/libjinglebase.la -lpthread
+XMLLIBS = ../../xmllite/libjinglexmllite.la ../../xmpp/libjinglexmpp.la $(EXPAT_LIBS)
+
+bin_PROGRAMS = relayserver stunserver
+relayserver_SOURCES = relayserver.cc relayserver_main.cc
+relayserver_LDADD = $(P2PLIBS)
+relayserver_CPPFLAGS = $(AM_CPPFLAGS)
+
+stunserver_SOURCES = stunserver.cc stunserver_main.cc
+stunserver_LDADD = $(P2PLIBS)
+stunserver_CPPFLAGS = $(AM_CPPFLAGS)
diff -up libjingle-0.6.14/talk/p2p/client/Makefile.am.SANITY libjingle-0.6.14/talk/p2p/client/Makefile.am
--- libjingle-0.6.14/talk/p2p/client/Makefile.am.SANITY	2012-06-11 15:19:25.243335579 -0400
+++ libjingle-0.6.14/talk/p2p/client/Makefile.am	2012-06-11 15:19:25.243335579 -0400
@@ -0,0 +1,20 @@
+lib_LTLIBRARIES = libjinglep2pclient.la
+
+libjinglep2pclient_la_SOURCES =  basicportallocator.cc \
+                                  connectivitychecker.cc \
+                                  httpportallocator.cc \
+                                  socketmonitor.cc
+
+libjinglep2pclient_includedir =  $(includedir)/libjingle-@LIBJINGLE_MAJORMINOR@/talk/p2p/client
+
+libjinglep2pclient_include_HEADERS = basicportallocator.h \
+                                  connectivitychecker.h \
+                                  socketmonitor.h \
+                                  sessionmanagertask.h \
+                                  sessionsendtask.h \
+                                  httpportallocator.h
+
+libjinglep2pclient_la_LDFLAGS = -version-info $(LIBJINGLE_LIBRARY_VERSION)
+
+AM_CPPFLAGS = -I../../.. -DLINUX -DPOSIX -DINTERNAL_BUILD -DFEATURE_ENABLE_VOICEMAIL
+
diff -up libjingle-0.6.14/talk/p2p/Makefile.am.SANITY libjingle-0.6.14/talk/p2p/Makefile.am
--- libjingle-0.6.14/talk/p2p/Makefile.am.SANITY	2012-06-11 15:19:25.243335579 -0400
+++ libjingle-0.6.14/talk/p2p/Makefile.am	2012-06-11 15:19:25.243335579 -0400
@@ -0,0 +1 @@
+SUBDIRS=base client
diff -up libjingle-0.6.14/talk/session/Makefile.am.SANITY libjingle-0.6.14/talk/session/Makefile.am
--- libjingle-0.6.14/talk/session/Makefile.am.SANITY	2012-06-11 15:19:25.243335579 -0400
+++ libjingle-0.6.14/talk/session/Makefile.am	2012-06-11 15:19:25.243335579 -0400
@@ -0,0 +1 @@
+SUBDIRS = tunnel phone
diff -up libjingle-0.6.14/talk/session/phone/Makefile.am.SANITY libjingle-0.6.14/talk/session/phone/Makefile.am
--- libjingle-0.6.14/talk/session/phone/Makefile.am.SANITY	2012-06-11 15:19:25.244335579 -0400
+++ libjingle-0.6.14/talk/session/phone/Makefile.am	2012-06-11 15:19:25.244335579 -0400
@@ -0,0 +1,93 @@
+lib_LTLIBRARIES = libjinglesessionphone.la
+
+libjinglesessionphone_la_SOURCES = audiomonitor.cc \
+                                    call.cc \
+                                    channel.cc \
+                                    channelmanager.cc \
+                                    codec.cc \
+                                    currentspeakermonitor.cc \
+                                    devicemanager.cc \
+                                    filemediaengine.cc \
+                                    filevideocapturer.cc \
+                                    gtkvideorenderer.cc \
+                                    libudevsymboltable.cc \
+                                    linuxdevicemanager.cc \
+                                    mediaengine.cc \
+                                    mediamessages.cc \
+                                    mediamonitor.cc \
+                                    mediarecorder.cc \
+                                    mediasession.cc \
+                                    mediasessionclient.cc \
+                                    rtcpmuxfilter.cc \
+                                    rtpdump.cc \
+                                    rtputils.cc \
+                                    soundclip.cc \
+                                    srtpfilter.cc \
+                                    ssrcmuxfilter.cc \
+                                    v4llookup.cc \
+                                    videoadapter.cc \
+                                    videocapturer.cc \
+                                    videocommon.cc \
+                                    videoframe.cc
+
+libjinglesessionphone_includedir = $(includedir)/libjingle-@LIBJINGLE_MAJORMINOR@/talk/session/phone
+libjinglesessionphone_include_HEADERS = audioframe.h \
+                                    audiomonitor.h \
+                                    call.h \
+                                    channel.h \
+                                    channelmanager.h \
+                                    codec.h \
+                                    cryptoparams.h \
+                                    currentspeakermonitor.h \
+                                    devicemanager.h \
+                                    fakedevicemanager.h \
+                                    fakemediaengine.h \
+                                    fakemediaprocessor.h \
+                                    fakenetworkinterface.h \
+                                    fakertp.h \
+                                    fakevideocapturer.h \
+                                    fakevideorenderer.h \
+                                    fakewebrtccommon.h \
+                                    fakewebrtcdeviceinfo.h \
+                                    fakewebrtcvcmfactory.h \
+                                    fakewebrtcvideocapturemodule.h \
+                                    fakewebrtcvideoengine.h \
+                                    fakewebrtcvoiceengine.h \
+                                    filemediaengine.h \
+                                    gtkvideorenderer.h \
+                                    libudevsymboltable.h \
+                                    linuxdevicemanager.h \
+                                    mediachannel.h \
+                                    mediacommon.h \
+                                    mediaengine.h \
+                                    mediamessages.h \
+                                    mediamonitor.h \
+                                    mediarecorder.h \
+                                    mediasessionclient.h \
+                                    mediasession.h \
+                                    mediasink.h \
+                                    nullvideorenderer.h \
+                                    rtcpmuxfilter.h \
+                                    rtpdump.h \
+                                    rtputils.h \
+                                    screencastid.h \
+                                    soundclip.h \
+                                    srtpfilter.h \
+                                    ssrcmuxfilter.h \
+                                    streamparams.h \
+                                    v4llookup.h \
+                                    videoadapter.h \
+                                    videocapturer.h \
+                                    videocommon.h \
+                                    videoframe.h \
+                                    videoprocessor.h \
+                                    videorenderer.h \
+                                    videorendererfactory.h \
+                                    voicechannel.h \
+                                    voiceprocessor.h
+
+libjinglesessionphone_la_LDFLAGS = -version-info $(LIBJINGLE_LIBRARY_VERSION)
+
+libjinglesessionphone_la_LIBADD = $(LIBSRTP_LIBS) $(GTK2_LIBS)
+
+AM_CPPFLAGS := -DPOSIX $(LIBSRTP_CFLAGS) $(GTK2_CFLAGS) -DFEATURE_ENABLE_VOICEMAIL
diff -up libjingle-0.6.14/talk/session/tunnel/Makefile.am.SANITY libjingle-0.6.14/talk/session/tunnel/Makefile.am
--- libjingle-0.6.14/talk/session/tunnel/Makefile.am.SANITY	2012-06-11 15:19:25.244335579 -0400
+++ libjingle-0.6.14/talk/session/tunnel/Makefile.am	2012-06-11 15:19:25.244335579 -0400
@@ -0,0 +1,16 @@
+lib_LTLIBRARIES = libjinglesessiontunnel.la
+
+libjinglesessiontunnel_la_SOURCES = tunnelsessionclient.cc \
+                                     securetunnelsessionclient.cc \
+                                     pseudotcpchannel.cc
+
+libjinglesessiontunnel_includedir = $(includedir)/libjingle-@LIBJINGLE_MAJORMINOR@/talk/session/tunnel
+
+libjinglesessiontunnel_include_HEADERS = tunnelsessionclient.h \
+                                          pseudotcpchannel.h \
+                                          securetunnelsessionclient.h
+
+libjinglesessiontunnel_la_LDFLAGS = -version-info $(LIBJINGLE_LIBRARY_VERSION)
+
+AM_CXXFLAGS = -DPOSIX -DFEATURE_ENABLE_VOICEMAIL
+
diff -up libjingle-0.6.14/talk/sound/Makefile.am.SANITY libjingle-0.6.14/talk/sound/Makefile.am
--- libjingle-0.6.14/talk/sound/Makefile.am.SANITY	2012-06-11 15:19:25.244335579 -0400
+++ libjingle-0.6.14/talk/sound/Makefile.am	2012-06-11 15:19:25.244335579 -0400
@@ -0,0 +1,35 @@
+lib_LTLIBRARIES = libjinglesound.la
+
+libjinglesound_la_SOURCES = alsasoundsystem.cc \
+                            alsasymboltable.cc \
+                            linuxsoundsystem.cc \
+                            nullsoundsystem.cc \
+                            platformsoundsystem.cc \
+                            platformsoundsystemfactory.cc \
+                            pulseaudiosoundsystem.cc \
+                            pulseaudiosymboltable.cc \
+                            soundsysteminterface.cc \
+                            soundsystemproxy.cc
+
+libjinglesound_includedir = $(includedir)/libjingle-@LIBJINGLE_MAJORMINOR@/talk/sound
+
+libjinglesound_include_HEADERS = alsasoundsystem.h \
+                                 alsasymboltable.h \
+                                 automaticallychosensoundsystem.h \
+                                 linuxsoundsystem.h \
+                                 nullsoundsystem.h \
+                                 nullsoundsystemfactory.h \
+                                 platformsoundsystem.h \
+                                 platformsoundsystemfactory.h \
+                                 pulseaudiosoundsystem.h \
+                                 pulseaudiosymboltable.h \
+                                 sounddevicelocator.h \
+                                 soundinputstreaminterface.h \
+                                 soundoutputstreaminterface.h \
+                                 soundsystemfactory.h \
+                                 soundsysteminterface.h \
+                                 soundsystemproxy.h
+
+libjinglesound_la_LDFLAGS = -version-info $(LIBJINGLE_LIBRARY_VERSION)
+
+AM_CXXFLAGS = -DPOSIX -DFEATURE_ENABLE_VOICEMAIL
diff -up libjingle-0.6.14/talk/xmllite/Makefile.am.SANITY libjingle-0.6.14/talk/xmllite/Makefile.am
--- libjingle-0.6.14/talk/xmllite/Makefile.am.SANITY	2012-06-11 15:19:25.244335579 -0400
+++ libjingle-0.6.14/talk/xmllite/Makefile.am	2012-06-11 15:19:25.244335579 -0400
@@ -0,0 +1,21 @@
+lib_LTLIBRARIES = libjinglexmllite.la
+
+libjinglexmllite_la_SOURCES = qname.cc \
+                               xmlbuilder.cc \
+                               xmlconstants.cc \
+                               xmlelement.cc \
+                               xmlnsstack.cc \
+                               xmlparser.cc \
+                               xmlprinter.cc
+
+libjinglexmllite_includedir = $(includedir)/libjingle-@LIBJINGLE_MAJORMINOR@/talk/xmllite
+libjinglexmllite_include_HEADERS = qname.h \
+                               xmlbuilder.h \
+                               xmlconstants.h \
+                               xmlelement.h \
+                               xmlnsstack.h \
+                               xmlparser.h \
+                               xmlprinter.h
+AM_CPPFLAGS = -DPOSIX
+
+libjinglexmllite_la_LDFLAGS = -version-info $(LIBJINGLE_LIBRARY_VERSION) $(EXPAT_LIBS)
diff -up libjingle-0.6.14/talk/xmpp/Makefile.am.SANITY libjingle-0.6.14/talk/xmpp/Makefile.am
--- libjingle-0.6.14/talk/xmpp/Makefile.am.SANITY	2012-06-11 15:19:25.244335579 -0400
+++ libjingle-0.6.14/talk/xmpp/Makefile.am	2012-06-11 15:19:25.244335579 -0400
@@ -0,0 +1,57 @@
+lib_LTLIBRARIES = libjinglexmpp.la
+
+libjinglexmpp_la_SOURCES =  constants.cc \
+                            hangoutpubsubclient.cc \
+                            iqtask.cc \
+                            jid.cc \
+                            moduleimpl.cc \
+                            mucroomconfigtask.cc \
+                            mucroomhistorytask.cc \
+                            mucroomlookuptask.cc \
+                            pubsubclient.cc \
+                            pubsub_task.cc \
+                            pubsubtasks.cc \
+                            receivetask.cc \
+                            saslmechanism.cc \
+                            xmppclient.cc \
+                            xmppengineimpl.cc \
+                            xmppengineimpl_iq.cc \
+                            xmpplogintask.cc \
+                            xmppstanzaparser.cc \
+                            xmpptask.cc
+
+libjinglexmpp_includedir = $(includedir)/libjingle-@LIBJINGLE_MAJORMINOR@/talk/xmpp
+libjinglexmpp_include_HEADERS = asyncsocket.h \
+                                chatroommodule.h \
+                                constants.h \
+                                fakexmppclient.h \
+                                hangoutpubsubclient.h \
+                                iqtask.h \
+                                jid.h \
+                                module.h \
+                                moduleimpl.h \
+                                mucroomconfigtask.h \
+                                mucroomhistorytask.h \
+                                mucroomlookuptask.h \
+                                plainsaslhandler.h \
+                                prexmppauth.h \
+                                pubsubclient.h \
+                                pubsub_task.h \
+                                pubsubtasks.h \
+                                receivetask.h \
+                                rostermodule.h \
+                                saslcookiemechanism.h \
+                                saslhandler.h \
+                                saslmechanism.h \
+                                saslplainmechanism.h \
+                                xmppclient.h \
+                                xmppclientsettings.h \
+                                xmppengine.h \
+                                xmppengineimpl.h \
+                                xmpplogintask.h \
+                                xmppstanzaparser.h \
+                                xmpptask.h
+
+libjinglexmpp_la_LDFLAGS = -version-info $(LIBJINGLE_LIBRARY_VERSION)
+
+AM_CPPFLAGS = -DPOSIX -DFEATURE_ENABLE_VOICEMAIL
