diff -up libjingle-0.6.14/talk/session/phone/devicemanager.cc.alsa libjingle-0.6.14/talk/session/phone/devicemanager.cc
--- libjingle-0.6.14/talk/session/phone/devicemanager.cc.alsa	2012-02-28 19:22:42.000000000 -0500
+++ libjingle-0.6.14/talk/session/phone/devicemanager.cc	2012-06-11 14:16:26.367278877 -0400
@@ -25,6 +25,8 @@
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include "config.h"
+
 #include "talk/session/phone/devicemanager.h"
 
 #include "talk/base/fileutils.h"
