diff -up libjingle-0.6.6/talk/base/openssladapter.cc.unistd libjingle-0.6.6/talk/base/openssladapter.cc
--- libjingle-0.6.6/talk/base/openssladapter.cc.unistd	2011-12-14 10:58:28.000000000 -0500
+++ libjingle-0.6.6/talk/base/openssladapter.cc	2012-01-05 14:17:27.519862129 -0500
@@ -45,6 +45,8 @@
 #include "talk/base/sslroots.h"
 #include "talk/base/stringutils.h"
 
+#include <unistd.h>
+
 // TODO: Use a nicer abstraction for mutex.
 
 #if defined(WIN32)
diff -up libjingle-0.6.6/talk/session/phone/v4llookup.cc.unistd libjingle-0.6.6/talk/session/phone/v4llookup.cc
--- libjingle-0.6.6/talk/session/phone/v4llookup.cc.unistd	2012-01-05 14:24:09.442099202 -0500
+++ libjingle-0.6.6/talk/session/phone/v4llookup.cc	2012-01-05 14:24:16.253035437 -0500
@@ -15,6 +15,7 @@
 #include <sys/ioctl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <unistd.h>
 
 #include <cstring>
 
diff -up libjingle-0.6.10/talk/base/rollingaccumulator.h.gcc47 libjingle-0.6.10/talk/base/rollingaccumulator.h
--- libjingle-0.6.10/talk/base/rollingaccumulator.h.gcc47	2012-02-13 21:59:27.755326807 -0500
+++ libjingle-0.6.10/talk/base/rollingaccumulator.h	2012-02-13 21:59:37.185231575 -0500
@@ -29,6 +29,7 @@
 #define TALK_BASE_ROLLINGACCUMULATOR_H_
 
 #include <vector>
+#include <unistd.h>
 
 #include "talk/base/common.h"
 
