diff -up libjingle-0.3.12/talk/base/openssladapter.cc.openssl libjingle-0.3.12/talk/base/openssladapter.cc
--- libjingle-0.3.12/talk/base/openssladapter.cc.openssl    2008-09-16 15:55:22.000000000 +0200
+++ libjingle-0.3.12/talk/base/openssladapter.cc    2009-08-25 17:26:25.000000000 +0200
@@ -641,7 +641,11 @@ OpenSSLAdapter::SSLPostConnectionCheck(S
     int extension_nid = OBJ_obj2nid(X509_EXTENSION_get_object(extension));
 
     if (extension_nid == NID_subject_alt_name) {
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+      const X509V3_EXT_METHOD* meth = X509V3_EXT_get(extension);
+#else
       X509V3_EXT_METHOD* meth = X509V3_EXT_get(extension);
+#endif
       if (!meth)
         break;
 

