diff -Nuar qwtplot3d.orig/qwtplot3d.pro qwtplot3d/qwtplot3d.pro
--- qwtplot3d.orig/qwtplot3d.pro	2010-03-24 11:09:56.301853388 +0200
+++ qwtplot3d/qwtplot3d.pro	2010-03-24 11:21:00.477089410 +0200
@@ -3,7 +3,7 @@
 
 TARGET            = qwtplot3d
 TEMPLATE          = lib
-CONFIG           += qt warn_on opengl thread zlib debug
+CONFIG           += qt warn_on opengl thread zlib
 MOC_DIR           = tmp
 OBJECTS_DIR       = tmp
 INCLUDEPATH       = include
@@ -21,7 +21,7 @@
 win32:CONFIG -= zlib
 
 linux-g++:TMAKE_CXXFLAGS += -fno-exceptions
-unix:VERSION = 0.2.6
+unix:VERSION = 0.2.7
 
 # Input
 SOURCES += src/qwt3d_axis.cpp \
@@ -80,16 +80,20 @@
            include/qwt3d_graphplot.h \
            include/qwt3d_multiplot.h
 
-# gl2ps support
-HEADERS+=3rdparty/gl2ps/gl2ps.h \
-         include/qwt3d_io_gl2ps.h
+# gl2ps support taken from system libs
+HEADERS+=include/qwt3d_io_gl2ps.h
          
-SOURCES+=src/qwt3d_io_gl2ps.cpp \
-         3rdparty/gl2ps/gl2ps.c
-
+SOURCES+=src/qwt3d_io_gl2ps.cpp
+unix:LIBS += -lgl2ps
 # zlib support for gl2ps
 zlib {
   DEFINES += GL2PS_HAVE_ZLIB
   win32:LIBS += zlib.lib
 	unix:LIBS  += -lz
-}
\ Dosya sonunda yenisatır yok.
+}
+
+target.path = /usr/lib
+headers.path = /usr/include/qwtplot3d
+headers.files = $$HEADERS
+INSTALLS = target headers
+
diff -Nuar qwtplot3d.orig/src/qwt3d_io_gl2ps.cpp qwtplot3d/src/qwt3d_io_gl2ps.cpp
--- qwtplot3d.orig/src/qwt3d_io_gl2ps.cpp	2010-03-24 11:09:56.303838766 +0200
+++ qwtplot3d/src/qwt3d_io_gl2ps.cpp	2010-03-24 11:17:51.267127880 +0200
@@ -4,7 +4,7 @@
 
 #include <time.h>
 #include "qwt3d_openglhelper.h"
-#include "../3rdparty/gl2ps/gl2ps.h"
+#include <gl2ps.h>
 #include "qwt3d_io_gl2ps.h"
 #include "qwt3d_plot.h"
 
