diff -up bind-9.7.0rc2/lib/bind9/Makefile.in.nonexec bind-9.7.0rc2/lib/bind9/Makefile.in
--- bind-9.7.0rc2/lib/bind9/Makefile.in.nonexec	2009-12-06 00:31:40.000000000 +0100
+++ bind-9.7.0rc2/lib/bind9/Makefile.in	2010-01-28 12:13:33.406696161 +0100
@@ -78,7 +78,7 @@ installdirs:
 	$(SHELL) ${top_srcdir}/mkinstalldirs ${DESTDIR}${libdir}
 
 install:: timestamp installdirs
-	${LIBTOOL_MODE_INSTALL} ${INSTALL_DATA} libbind9.@A@ ${DESTDIR}${libdir}
+	${LIBTOOL_MODE_INSTALL} ${INSTALL_PROGRAM} libbind9.@A@ ${DESTDIR}${libdir}
 
 clean distclean::
 	rm -f libbind9.@A@ timestamp
diff -up bind-9.7.0rc2/lib/dns/Makefile.in.nonexec bind-9.7.0rc2/lib/dns/Makefile.in
--- bind-9.7.0rc2/lib/dns/Makefile.in.nonexec	2009-12-06 00:31:40.000000000 +0100
+++ bind-9.7.0rc2/lib/dns/Makefile.in	2010-01-28 12:13:33.406696161 +0100
@@ -131,7 +131,7 @@ installdirs:
 	$(SHELL) ${top_srcdir}/mkinstalldirs ${DESTDIR}${libdir}
 
 install:: timestamp installdirs
-	${LIBTOOL_MODE_INSTALL} ${INSTALL_DATA} libdns.@A@ ${DESTDIR}${libdir}
+	${LIBTOOL_MODE_INSTALL} ${INSTALL_PROGRAM} libdns.@A@ ${DESTDIR}${libdir}
 
 clean distclean::
 	rm -f libdns.@A@ timestamp
diff -up bind-9.7.0rc2/lib/isccc/Makefile.in.nonexec bind-9.7.0rc2/lib/isccc/Makefile.in
--- bind-9.7.0rc2/lib/isccc/Makefile.in.nonexec	2009-12-06 00:31:41.000000000 +0100
+++ bind-9.7.0rc2/lib/isccc/Makefile.in	2010-01-28 12:13:33.406696161 +0100
@@ -80,7 +80,7 @@ installdirs:
 	$(SHELL) ${top_srcdir}/mkinstalldirs ${DESTDIR}${libdir}
 
 install:: timestamp installdirs
-	${LIBTOOL_MODE_INSTALL} ${INSTALL_DATA} libisccc.@A@ ${DESTDIR}${libdir}
+	${LIBTOOL_MODE_INSTALL} ${INSTALL_PROGRAM} libisccc.@A@ ${DESTDIR}${libdir}
 
 clean distclean::
 	rm -f libisccc.@A@ timestamp
diff -up bind-9.7.0rc2/lib/isccfg/Makefile.in.nonexec bind-9.7.0rc2/lib/isccfg/Makefile.in
--- bind-9.7.0rc2/lib/isccfg/Makefile.in.nonexec	2009-12-06 00:31:41.000000000 +0100
+++ bind-9.7.0rc2/lib/isccfg/Makefile.in	2010-01-28 12:13:33.406696161 +0100
@@ -77,7 +77,7 @@ installdirs:
 	$(SHELL) ${top_srcdir}/mkinstalldirs ${DESTDIR}${libdir}
 
 install:: timestamp installdirs
-	${LIBTOOL_MODE_INSTALL} ${INSTALL_DATA} libisccfg.@A@ ${DESTDIR}${libdir}
+	${LIBTOOL_MODE_INSTALL} ${INSTALL_PROGRAM} libisccfg.@A@ ${DESTDIR}${libdir}
 
 clean distclean::
 	rm -f libisccfg.@A@ timestamp
diff -up bind-9.7.0rc2/lib/isc/Makefile.in.nonexec bind-9.7.0rc2/lib/isc/Makefile.in
--- bind-9.7.0rc2/lib/isc/Makefile.in.nonexec	2009-12-18 05:09:55.000000000 +0100
+++ bind-9.7.0rc2/lib/isc/Makefile.in	2010-01-28 12:13:53.566696766 +0100
@@ -121,7 +121,7 @@ installdirs:
 	$(SHELL) ${top_srcdir}/mkinstalldirs ${DESTDIR}${libdir}
 
 install:: timestamp installdirs
-	${LIBTOOL_MODE_INSTALL} ${INSTALL_DATA} libisc.@A@ ${DESTDIR}${libdir}
+	${LIBTOOL_MODE_INSTALL} ${INSTALL_PROGRAM} libisc.@A@ ${DESTDIR}${libdir}
 
 clean distclean::
 	rm -f libisc.@A@ libisc-nosymtbl.@A@ libisc.la \
diff -up bind-9.7.0rc2/lib/lwres/Makefile.in.nonexec bind-9.7.0rc2/lib/lwres/Makefile.in
--- bind-9.7.0rc2/lib/lwres/Makefile.in.nonexec	2007-06-20 01:47:22.000000000 +0200
+++ bind-9.7.0rc2/lib/lwres/Makefile.in	2010-01-28 12:13:33.406696161 +0100
@@ -78,7 +78,7 @@ installdirs:
 	$(SHELL) ${top_srcdir}/mkinstalldirs ${DESTDIR}${libdir}
 
 install:: timestamp installdirs
-	${LIBTOOL_MODE_INSTALL} ${INSTALL_DATA} liblwres.@A@ ${DESTDIR}${libdir}
+	${LIBTOOL_MODE_INSTALL} ${INSTALL_PROGRAM} liblwres.@A@ ${DESTDIR}${libdir}
 
 clean distclean::
 	rm -f liblwres.@A@ liblwres.la timestamp
