Index: Xfce/xfce4-notifyd-config/xfce4-notifyd-config.desktop.in
===================================================================
--- Xfce.orig/xfce4-notifyd-config/xfce4-notifyd-config.desktop.in
+++ Xfce/xfce4-notifyd-config/xfce4-notifyd-config.desktop.in
@@ -8,5 +8,5 @@ Terminal=false
 StartupNotify=true
 Type=Application
 OnlyShowIn=XFCE;
-Categories=X-XFCE;Settings;DesktopSettings;
+Categories=X-XFCE;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-PersonalSettings;
 X-XfcePluggable=true
