--- /dev/null	2009-12-03 13:12:21.275958023 -0200
+++ comm-1.9.1/mozilla/xpinstall/packager/packages-unix	2009-12-09 14:44:07.000000000 -0200
@@ -0,0 +1,13 @@
+[enigmail]
+bin/components/enigmail.js
+bin/components/enigmail.xpt
+bin/components/enigprefs-service.js
+bin/chrome/enigmail.jar
+bin/chrome/enigmail-skin.jar
+bin/defaults/pref/enigmail.js
+
+[enigmime]
+bin/components/enigmime.xpt
+bin/components/ipc.xpt
+bin/components/libenigmime.so
+bin/chrome/enigmime.jar
