diff -p -up comm-1.9.2/mail/components/shell/nsMailGNOMEIntegration.cpp.thunderbird-appname comm-1.9.2/mail/components/shell/nsMailGNOMEIntegration.cpp
--- comm-1.9.2/mail/components/shell/nsMailGNOMEIntegration.cpp.thunderbird-appname	2010-07-13 18:22:45.000000000 -0300
+++ comm-1.9.2/mail/components/shell/nsMailGNOMEIntegration.cpp	2010-07-20 17:37:39.595057298 -0300
@@ -98,6 +98,12 @@ nsMailGNOMEIntegration::Init()
   NS_ENSURE_SUCCESS(rv, rv);
 
   rv = appPath->GetNativePath(mAppPath);
+
+  /* This path of thunderbird is hardcoded for the default path setting 
+   * And the above code is not used.
+   */
+  mAppPath.Assign ("/usr/bin/thunderbird");
+
   return rv;
 }
 
