--- bin/hsqldb.orig	2011-01-24 11:59:14.356301132 +0200
+++ bin/hsqldb	2011-01-24 12:00:06.278430417 +0200
@@ -41,7 +41,7 @@
 # use this file.
 
 # This stuff will be ignored by systems that don't use chkconfig.
-# chkconfig: 345 87 13
+# chkconfig: - 87 13
 # description: Hsqldb, A High Performance Java Database
 # pidfile: /var/run/hsqldb.pid
 # config: /etc/sysconfig/hsqldb
@@ -51,7 +51,6 @@
 # Provides:          HSQLDB-Server
 # Required-Start:    $syslog $remote_fs $network $named
 # Required-Stop:
-# Default-Start:     3 5
 # Default-Stop:      0 1 2 6
 # Short-Description: HSQLDB-Server
 # Description:       Hsqldb, A High Performance Java Database Server
@@ -369,7 +368,7 @@ running with pid $HSQLDB_PID."
             # Some OSes choke if there are newlines in this string.
             # N.b.!!!  The shell of the -c command is the target user's default
             # login shell, so keep this command shell-independent!
-            nohup su "$HSQLDB_OWNER" -c "cd '$SERVER_HOME' && echo "'$$'" > '$PIDFILE' && exec '$JAVA_EXECUTABLE' $SERVER_JVMARGS $TLS_SWITCHES -classpath '${SERVER_ADDL_CLASSPATH}${HSQLDB_JAR_PATH}' org.hsqldb.util.MainInvoker org.hsqldb.$TARGET_CLASS $INVOC_ADDL_ARGS" >> "$LOGFILE" 2>&1 &
+            nohup su "$HSQLDB_OWNER" -s /bin/sh -c "cd '$SERVER_HOME' && echo "'$$'" > '$PIDFILE' && exec '$JAVA_EXECUTABLE' $SERVER_JVMARGS $TLS_SWITCHES -classpath '${SERVER_ADDL_CLASSPATH}${HSQLDB_JAR_PATH}' org.hsqldb.util.MainInvoker org.hsqldb.$TARGET_CLASS $INVOC_ADDL_ARGS" >> "$LOGFILE" 2>&1 &
         else
             cd "$SERVER_HOME" || {
                 echo "Failed to cd to '$SERVER_HOME'" 1>&2
@@ -464,7 +463,7 @@ See log file '$LOGFILE'." 1>&2
         echo "Successful shutdown ${SHUTDOWN_OPTION} (for the $TARGET_CLASS process)!"
         exit 0
     ;;
-    restart|restartcompacted)
+    restart|restartcompacted|reload)
         STOP_COMMAND=stop
         [ "$COMMAND" = restartcompacted ] && STOP_COMMAND=stopcompact
         "${INVOC_PATH}"hsqldb $STOP_COMMAND || exit $?