--- bin/runUtil.sh.orig	2004-08-10 16:04:01.000000000 +0200
+++ bin/runUtil.sh	2004-08-10 16:05:43.000000000 +0200
@@ -31,7 +31,7 @@
 progname=`basename $0`
 
 # ../ will lead us to the home
-dbhome="$progdir/.."
+dbhome=/var/lib/hsqldb
 
 # absolutize dbhome
 
@@ -40,7 +40,7 @@
 #
 # bring in needed functions
 
-. ${dbhome}/lib/functions
+. /var/lib/hsqldb/lib/functions
 
 #--------------------------------------------------------------------------------------------------------------
 pre_main