diff -ur hsqldb.orig/bin/hsqldb hsqldb/bin/hsqldb
--- hsqldb.orig/bin/hsqldb	2006-09-05 10:37:12.000000000 -0400
+++ hsqldb/bin/hsqldb	2006-09-05 10:50:45.000000000 -0400
@@ -110,7 +110,10 @@
 SU_ECHO_SECS=1
 # File used as semaphore.  If file is removed, a running pid checker
 # process will exit.
-PIDCHECKER_FLAGFILE=/tmp/pidchecker.run
+PIDCHECKER_FLAGFILE=`mktemp /tmp/hsqldb.XXXXXXXX`
+if [ x"$PIDCHECKER_FLAGFILE" = "x" ] ; then
+	echo "unable to create tmp file"; exit 1
+fi
 # The following settings get overridden by optional setting in the config file.
 # Time to allow for JVM to die after all HSQLDB instances stopped.
 MAX_TERMINATE_SECS=1