Index: colorgcc-1.3.2/colorgcc
===================================================================
--- colorgcc-1.3.2.orig/colorgcc
+++ colorgcc-1.3.2/colorgcc
@@ -101,6 +101,8 @@ sub initDefaults
    $compilerPaths{"g77"} = "/usr/bin/gfortran";
    $compilerPaths{"f77"} = "/usr/bin/gfortran";
    $compilerPaths{"gcj"} = "/usr/bin/gcj";
+   $compilerPaths{"i686-pc-linux-gnu-g++"} = "/usr/bin/i686-pc-linux-gnu-g++";
+   $compilerPaths{"i686-pc-linux-gnu-gcc"} = "/usr/bin/i686-pc-linux-gnu-gcc";
 
    $nocolor{"dumb"} = "true";
 
@@ -210,7 +212,7 @@ if (-f $configFile)
 $0 =~ m%.*/(.*)$%;
 $progName = $1 || $0;
 
-$compiler = $compilerPaths{$progName} || $compilerPaths{"gcc"};
+$compiler = $compilerPaths{$progName} || "/usr/bin/$progName";
 
 # Check that we don't reference self
 die "$compiler is self-referencing"
