Index: htmldoc-1.8.27/data/Makefile
===================================================================
--- htmldoc-1.8.27.orig/data/Makefile
+++ htmldoc-1.8.27/data/Makefile
@@ -53,11 +53,11 @@ all:
 #
 
 install:
-	if [ ! -d $(datadir)/htmldoc/data ]; then\
-		$(MKDIR) $(datadir)/htmldoc/data;\
+	if [ ! -d $(DESTDIR)$(datadir)/htmldoc/data ]; then\
+		$(MKDIR) $(DESTDIR)$(datadir)/htmldoc/data;\
 	fi
-	$(CP) $(FILES) $(datadir)/htmldoc/data
-	$(CHMOD) ugo+r $(datadir)/htmldoc/data/*
+	$(CP) $(FILES) $(DESTDIR)$(datadir)/htmldoc/data
+	$(CHMOD) ugo+r $(DESTDIR)$(datadir)/htmldoc/data/*
 
 
 #
Index: htmldoc-1.8.27/doc/Makefile
===================================================================
--- htmldoc-1.8.27.orig/doc/Makefile
+++ htmldoc-1.8.27/doc/Makefile
@@ -57,16 +57,16 @@ all: $(DOCUMENTS)
 #
 
 install: $(DOCUMENTS)
-	if [ ! -d $(datadir)/doc/htmldoc ]; then\
-		$(MKDIR) $(datadir)/doc/htmldoc;\
+	if [ ! -d $(DESTDIR)$(datadir)/doc/htmldoc ]; then\
+		$(MKDIR) $(DESTDIR)$(datadir)/doc/htmldoc;\
 	fi
-	$(CP) $(DOCFILES) $(datadir)/doc/htmldoc
-	$(CHMOD) ugo+r $(datadir)/doc/htmldoc/*
-	if [ ! -d $(mandir)/man1 ]; then\
-		$(MKDIR) $(mandir)/man1;\
+	$(CP) $(DOCFILES) $(DESTDIR)$(datadir)/doc/htmldoc
+	$(CHMOD) ugo+r $(DESTDIR)$(datadir)/doc/htmldoc/*
+	if [ ! -d $(DESTDIR)$(mandir)/man1 ]; then\
+		$(MKDIR) $(DESTDIR)$(mandir)/man1;\
 	fi
-	$(CP) htmldoc.man $(mandir)/man1/htmldoc.1
-	$(CHMOD) ugo+r $(mandir)/man1/htmldoc.1
+	$(CP) htmldoc.man $(DESTDIR)$(mandir)/man1/htmldoc.1
+	$(CHMOD) ugo+r $(DESTDIR)$(mandir)/man1/htmldoc.1
 
 
 #
Index: htmldoc-1.8.27/htmldoc/Makefile
===================================================================
--- htmldoc-1.8.27.orig/htmldoc/Makefile
+++ htmldoc-1.8.27/htmldoc/Makefile
@@ -50,11 +50,11 @@ all:	htmldoc$(EXEEXT)
 #
 
 install:	all
-	if [ ! -d $(bindir) ]; then\
-		$(MKDIR) $(bindir);\
+	if [ ! -d $(DESTDIR)$(bindir) ]; then\
+		$(MKDIR) $(DESTDIR)$(bindir);\
 	fi
-	cp htmldoc$(EXEEXT) $(bindir)
-	chmod ugo+rx $(bindir)/htmldoc$(EXEEXT)
+	cp htmldoc$(EXEEXT) $(DESTDIR)$(bindir)
+	chmod ugo+rx $(DESTDIR)$(bindir)/htmldoc$(EXEEXT)
 
 
 #
Index: htmldoc-1.8.27/fonts/Makefile
===================================================================
--- htmldoc-1.8.27.orig/fonts/Makefile
+++ htmldoc-1.8.27/fonts/Makefile
@@ -74,15 +74,15 @@ all:
 #
 
 install:
-	echo "Installing font files in $(datadir)/htmldoc/fonts..."
-	if [ ! -d $(datadir)/htmldoc/fonts ]; then\
-		$(MKDIR) $(datadir)/htmldoc/fonts;\
+	echo "Installing font files in $(DESTDIR)$(datadir)/htmldoc/fonts..."
+	if [ ! -d $(DESTDIR)$(datadir)/htmldoc/fonts ]; then\
+		$(MKDIR) $(DESTDIR)$(datadir)/htmldoc/fonts;\
 	fi
 	for font in $(FONTS); do \
-		$(CP) $$font.afm $(datadir)/htmldoc/fonts; \
-		$(CP) $$font.pfa $(datadir)/htmldoc/fonts; \
+		$(CP) $$font.afm $(DESTDIR)$(datadir)/htmldoc/fonts; \
+		$(CP) $$font.pfa $(DESTDIR)$(datadir)/htmldoc/fonts; \
 	done
-	$(CHMOD) ugo+r $(datadir)/htmldoc/fonts/*
+	$(CHMOD) ugo+r $(DESTDIR)$(datadir)/htmldoc/fonts/*
 
 
 #
