--- mate-menus-1.2.0/libmenu/Makefile.am.orig	2012-02-24 00:13:03.000000000 +0200
+++ mate-menus-1.2.0/libmenu/Makefile.am	2012-07-31 21:57:04.590700045 +0300
@@ -37,7 +37,7 @@
 libmate_menu_la_LDFLAGS = \
 	-version-info $(LIB_MENU_LT_VERSION) \
 	-no-undefined \
-	-export-symbols-regex matemenu_tree
+	-export-symbols-regex matemenu_tree $(LDFLAGS)
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libmate-menu.pc
