--- mate-menus-1.2.0/python/Makefile.am.orig	2012-02-24 00:13:03.000000000 +0200
+++ mate-menus-1.2.0/python/Makefile.am	2012-07-31 22:00:34.945678507 +0300
@@ -15,6 +15,6 @@
 
 matemenu_la_SOURCES = matemenu.c
 matemenu_la_LDFLAGS = -module -avoid-version -fPIC -export-symbols-regex initmatemenu
-matemenu_la_LIBADD = $(GLIB_LIBS) $(PYTHON_LIBS) $(top_builddir)/libmenu/libmate-menu.la
+matemenu_la_LIBADD = $(GLIB_LIBS) $(PYTHON_LIBS) $(top_builddir)/libmenu/libmate-menu.la -lpython2.7
 
 -include $(top_srcdir)/git.mk
