diff -Nuar nature.OSmanOS/Makefile.am nature/Makefile.am
--- nature.OSmanOS/Makefile.am	2013-12-31 17:20:19.000000000 +0200
+++ nature/Makefile.am	2014-02-05 22:54:17.975301175 +0200
@@ -1,6 +1,15 @@
 backgrounddir = $(datadir)/backgrounds/mate/nature
 
 background_DATA =\
+	1.jpg \
+	2.jpg \
+	3.jpg \
+	4.jpg \
+	5.jpg \
+	6.jpg \
+	7.jpg \
+	8.jpg \
+	pisi.jpg \
 	Aqua.jpg \
 	Blinds.jpg \
 	Dune.jpg \
diff -Nuar nature.OSmanOS/Makefile.in nature/Makefile.in
--- nature.OSmanOS/Makefile.in	2013-12-31 17:20:21.000000000 +0200
+++ nature/Makefile.in	2014-02-05 22:54:11.785301002 +0200
@@ -255,6 +255,15 @@
 top_srcdir = @top_srcdir@
 backgrounddir = $(datadir)/backgrounds/mate/nature
 background_DATA = \
+	1.jpg \
+	2.jpg \
+	3.jpg \
+	4.jpg \
+	5.jpg \
+	6.jpg \
+	7.jpg \
+	8.jpg \
+	pisi.jpg \
 	Aqua.jpg \
 	Blinds.jpg \
 	Dune.jpg \
diff -Nuar nature.OSmanOS/mate-nature.xml.in.in nature/mate-nature.xml.in.in
--- nature.OSmanOS/mate-nature.xml.in.in	2013-12-31 17:20:19.000000000 +0200
+++ nature/mate-nature.xml.in.in	2014-02-05 22:54:05.828634154 +0200
@@ -61,6 +61,51 @@
 		<options>zoom</options>
 	</wallpaper>
 	<wallpaper deleted="false">
+		<_name>1</_name>
+		<filename>@datadir@/backgrounds/mate/nature/1.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
+	<wallpaper deleted="false">
+		<_name>2</_name>
+		<filename>@datadir@/backgrounds/mate/nature/2.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
+	<wallpaper deleted="false">
+		<_name>3</_name>
+		<filename>@datadir@/backgrounds/mate/nature/3.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
+	<wallpaper deleted="false">
+		<_name>4</_name>
+		<filename>@datadir@/backgrounds/mate/nature/4.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
+	<wallpaper deleted="false">
+		<_name>5</_name>
+		<filename>@datadir@/backgrounds/mate/nature/5.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
+	<wallpaper deleted="false">
+		<_name>6</_name>
+		<filename>@datadir@/backgrounds/mate/nature/6.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
+	<wallpaper deleted="false">
+		<_name>7</_name>
+		<filename>@datadir@/backgrounds/mate/nature/7.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
+	<wallpaper deleted="false">
+		<_name>8</_name>
+		<filename>@datadir@/backgrounds/mate/nature/8.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
+	<wallpaper deleted="false">
+		<_name>Pisi</_name>
+		<filename>@datadir@/backgrounds/mate/nature/pisi.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
+	<wallpaper deleted="false">
 		<_name>Wood</_name>
 		<filename>@datadir@/backgrounds/mate/nature/Wood.jpg</filename>
 		<options>zoom</options>
