diff -Naur comar-api-2.4.9.orig/comar/network.py comar-api-2.4.9/comar/network.py
--- comar-api-2.4.9.orig/comar/network.py	2010-11-11 14:29:05.000000000 +0100
+++ comar-api-2.4.9/comar/network.py	2014-08-19 20:06:27.584878014 +0200
@@ -146,12 +146,12 @@
         # Start service
         startService(command="/usr/sbin/ifplugd",
                      args="%s -i %s" % (args, device),
-                     pidfile="/var/run/ifplugd.%s.pid" % device,
+                     pidfile="/run/ifplugd.%s.pid" % device,
                      detach=True,
                      donotify=False)
     else:
         # Stop service
-        stopService(pidfile="/var/run/ifplugd.%s.pid" % device, donotify=False)
+        stopService(pidfile="/run/ifplugd.%s.pid" % device, donotify=False)
 
 def plugCheck(device):
     # Return true if ifplugd is missing
diff -Naur comar-api-2.4.9.orig/comar/utility.py comar-api-2.4.9/comar/utility.py
--- comar-api-2.4.9.orig/comar/utility.py	2010-11-11 14:29:05.000000000 +0100
+++ comar-api-2.4.9/comar/utility.py	2014-08-19 20:06:52.055877120 +0200
@@ -31,7 +31,7 @@
     """
     class Handler:
         def handler(self, *args, **kwargs):
-            lock = FileLock("/var/lock/subsys/%s.comar" % script())
+            lock = FileLock("/run/lock/subsys/%s.comar" % script())
             lock.lock()
             self.myfunc(*args, **kwargs)
             lock.unlock()
