diff -ur libindi-0.6.2/CMakeLists.txt libindi-0.6.2.new/CMakeLists.txt
--- libindi-0.6.2/CMakeLists.txt	2010-07-29 12:26:03.996058345 +0200
+++ libindi-0.6.2.new/CMakeLists.txt	2010-07-29 12:31:02.808099234 +0200
@@ -29,6 +29,7 @@
 include (MacroLogFeature)
 include (MacroBoolTo01)
 include (CheckIncludeFiles)
+include (FindPkgConfig)
 
 find_package(ZLIB REQUIRED)
 
@@ -36,6 +37,7 @@
 macro_log_feature(LIBUSB_FOUND "libusb" "User level access to USB devices" "http://indi.sf.net" FALSE "" "Provides support for USB based drivers in INDI.")
 
 macro_optional_find_package(CFITSIO)
+pkg_check_modules(CFITSIO REQUIRED cfitsio)
 
 if (NOT CFITSIO_FOUND OR CFITSIO_VERSION_MAJOR LESS 3)
   message(FATAL_ERROR "CFITSIO version too old, Please install cfitsio 3.x and try again. http://indi.sf.net")
@@ -64,7 +66,7 @@
 include_directories( ${CMAKE_SOURCE_DIR}/libs/webcam)
 
 if (CFITSIO_FOUND)
-  include_directories(${CFITSIO_INCLUDE_DIR})
+  include_directories(${CFITSIO_INCLUDE_DIRS})
 endif (CFITSIO_FOUND)
 
 if (NOVA_FOUND)
