--- tools/compiler.c.str	2009-05-20 04:00:40.000000000 +0200
+++ tools/compiler.c	2009-05-20 04:01:05.000000000 +0200
@@ -294,15 +294,15 @@
 	    break;
 	case '|':
 	    if (*cexpr == '|') { cexpr++; tok = OR; }
-	    else { (void) sprintf (err_msg, badop); return (ERR); }
+	    else { (void) sprintf (err_msg, "%s", badop); return (ERR); }
 	    break;
 	case '&':
 	    if (*cexpr == '&') { cexpr++; tok = AND; }
-	    else { (void) sprintf (err_msg, badop); return (ERR); }
+	    else { (void) sprintf (err_msg, "%s", badop); return (ERR); }
 	    break;
 	case '=':
 	    if (*cexpr == '=') { cexpr++; tok = EQ; }
-	    else { (void) sprintf (err_msg, badop); return (ERR); }
+	    else { (void) sprintf (err_msg, "%s", badop); return (ERR); }
 	    break;
 	case '!':
 	    if (*cexpr == '=') { cexpr++; tok = NE; } else { tok = NOT; }
