diff -rupN ZynAddSubFX-2.4.2.old/ExternalPrograms/Controller/Makefile ZynAddSubFX-2.4.2/ExternalPrograms/Controller/Makefile
--- ZynAddSubFX-2.4.2.old/ExternalPrograms/Controller/Makefile	2012-02-26 13:37:11.000000000 -0500
+++ ZynAddSubFX-2.4.2/ExternalPrograms/Controller/Makefile	2012-03-08 21:32:16.843988193 -0500
@@ -1,17 +1,17 @@
 controller: main.o Controller.o ControllerUI.o
-	gcc -o controller *.o `fltk-config --ldflags` -lasound -lpthread -lm -lstdc++
+	gcc $(CFLAGS) -o controller *.o `fltk-config --ldflags` -lasound -lpthread -lm -lstdc++
 
 ControllerUI.cxx ControllerUI.h: ControllerUI.fl
 	fluid -c ControllerUI.fl
 
 ControllerUI.o: ControllerUI.cxx ControllerUI.h
-	gcc `fltk-config --cflags` -c ControllerUI.cxx -o ControllerUI.o
+	gcc $(CFLAGS) `fltk-config --cflags` -c ControllerUI.cxx -o ControllerUI.o
 
 Controller.o: Controller.C Controller.h
-	gcc `fltk-config --cflags` -c Controller.C -o Controller.o
+	gcc $(CFLAGS) `fltk-config --cflags` -c Controller.C -o Controller.o
 
 main.o: main.C Controller.h ControllerUI.h
-	gcc `fltk-config --cflags` -c main.C -o main.o
+	gcc $(CFLAGS) `fltk-config --cflags` -c main.C -o main.o
 
 clean:
 	rm -f *.o controller ControllerUI.cxx ControllerUI.h 
diff -rupN ZynAddSubFX-2.4.2.old/ExternalPrograms/Spliter/Makefile ZynAddSubFX-2.4.2/ExternalPrograms/Spliter/Makefile
--- ZynAddSubFX-2.4.2.old/ExternalPrograms/Spliter/Makefile	2012-02-26 13:37:11.000000000 -0500
+++ ZynAddSubFX-2.4.2/ExternalPrograms/Spliter/Makefile	2012-03-08 21:33:39.109571039 -0500
@@ -1,17 +1,17 @@
 spliter: main.o Spliter.o SpliterUI.o
-	gcc -o spliter *.o `fltk-config --ldflags` -lasound -lstdc++
+	gcc $(CFLAGS) -o spliter *.o `fltk-config --ldflags` -lasound -lstdc++ -lpthread
 
 SplitterUI.cxx SpliterUI.h: SpliterUI.fl
 	fluid -c SpliterUI.fl
 
 SpliterUI.o: SplitterUI.cxx SpliterUI.h
-	gcc -g `fltk-config --cflags` -c SpliterUI.cxx -o SpliterUI.o
+	gcc $(CFLAGS) -g `fltk-config --cflags` -c SpliterUI.cxx -o SpliterUI.o
 
 Spliter.o: Spliter.C Spliter.h
-	gcc -g `fltk-config --cflags` -c Spliter.C -o Spliter.o
+	gcc $(CFLAGS) -g `fltk-config --cflags` -c Spliter.C -o Spliter.o
 
 main.o: main.C Spliter.h SpliterUI.h
-	gcc -g `fltk-config --cflags` -c main.C -o main.o
+	gcc $(CFLAGS) -g `fltk-config --cflags` -c main.C -o main.o
 
 clean:
 	rm -f *.o spliter SpliterUI.cxx SpliterUI.h 
diff -rupN ZynAddSubFX-2.4.2.old/src/CMakeLists.txt ZynAddSubFX-2.4.2/src/CMakeLists.txt
--- ZynAddSubFX-2.4.2.old/src/CMakeLists.txt	2012-02-26 13:37:11.000000000 -0500
+++ ZynAddSubFX-2.4.2/src/CMakeLists.txt	2012-03-08 21:23:26.111658165 -0500
@@ -70,9 +70,14 @@ endif(NOT PKG_CONFIG_FOUND)
 option (BuildForAMD_X86_64 "Build for AMD x86_64 system" OFF)
 option (BuildForCore2_X86_64 "Build for Intel Core2 x86_64 system" OFF)
 option (BuildForDebug "Include gdb debugging support" OFF)
+option (X86Build "Build for X86 family" OFF)
 
 set(CMAKE_BUILD_TYPE "Release")
 
+if (X86Build)
+   set (AsmOption "-DASM_F2I_YES")
+endif ()
+
 set (BuildOptions_x86_64AMD
     "-O3 -march=athlon64 -m64 -Wall -ffast-math -fno-finite-math-only -fomit-frame-pointer"
   CACHE STRING "X86_64 compiler options"
@@ -84,7 +89,7 @@ set (BuildOptions_X86_64Core2
 )
 
 set (BuildOptionsBasic
-    "-O3 -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer"
+    "-ffast-math -fno-finite-math-only -fomit-frame-pointer ${AsmOption}"
     CACHE STRING "basic X86 complier options"
 )
 
@@ -133,26 +138,19 @@ endif()
 # other include directories
 include_directories(${ZLIB_INCLUDE_DIRS} ${MXML_INCLUDE_DIRS})
 
-add_definitions(
-	 -DASM_F2I_YES
-	 -g #TODO #todo put in a better location
-	 -Wall
-	 -Wextra
-	 )
-
 if (BuildForDebug)
 	set (CMAKE_BUILD_TYPE "Debug")
-	set (CMAKE_CXX_FLAGS_DEBUG ${BuildOptionsDebug})
+	set (CMAKE_CXX_FLAGS_DEBUG "${BuildOptionsDebug} ${CMAKE_CXX_FLAGS}")
 	message (STATUS "Building for ${CMAKE_BUILD_TYPE}, flags: ${CMAKE_CXX_FLAGS_DEBUG}")
 else (BuildForDebug)
 	set (CMAKE_BUILD_TYPE "Release")
 	if (BuildForAMD_X86_64)
-		set (CMAKE_CXX_FLAGS_RELEASE ${BuildOptions_x86_64AMD})
+		set (CMAKE_CXX_FLAGS_RELEASE "${BuildOptions_x86_64AMD} ${CMAKE_CXX_FLAGS}")
 	else (BuildForAMD_X86_64)
 		if (BuildForCore2_X86_64)
-			set (CMAKE_CXX_FLAGS_RELEASE ${BuildOptions_X86_64Core2})
+			set (CMAKE_CXX_FLAGS_RELEASE "${BuildOptions_X86_64Core2} ${CMAKE_CXX_FLAGS}")
 		else (BuildForCore2_X86_64)
-			set (CMAKE_CXX_FLAGS_RELEASE ${BuildOptionsBasic})
+			set (CMAKE_CXX_FLAGS_RELEASE "${BuildOptionsBasic} ${CMAKE_CXX_FLAGS}")
 		endif (BuildForCore2_X86_64)
 	endif (BuildForAMD_X86_64)
 	message (STATUS "Building for ${CMAKE_BUILD_TYPE}, flags: ${CMAKE_CXX_FLAGS_RELEASE}")

