diff -Nuar libofx-0.8.3.orig/dtd/Makefile.in libofx-0.8.3/dtd/Makefile.in
--- libofx-0.8.3.orig/dtd/Makefile.in	2007-01-09 06:29:25.000000000 +0200
+++ libofx-0.8.3/dtd/Makefile.in	2007-08-23 18:17:52.000000000 +0300
@@ -54,7 +54,7 @@
     *) f=$$p;; \
   esac;
 am__strip_dir = `echo $$p | sed -e 's|^.*/||'`;
-am__installdirs = "$(DESTDIR)$(docdir)"
+am__installdirs = "$(DESTDIR)$(LIBOFX_DTD_DIR)"
 docDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(doc_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
@@ -247,20 +247,20 @@
 uninstall-info-am:
 install-docDATA: $(doc_DATA)
 	@$(NORMAL_INSTALL)
-	test -z "$(docdir)" || $(mkdir_p) "$(DESTDIR)$(docdir)"
+	test -z "$(docdir)" || $(mkdir_p) "$(DESTDIR)$(LIBOFX_DTD_DIR)"
 	@list='$(doc_DATA)'; for p in $$list; do \
 	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
 	  f=$(am__strip_dir) \
-	  echo " $(docDATA_INSTALL) '$$d$$p' '$(DESTDIR)$(docdir)/$$f'"; \
-	  $(docDATA_INSTALL) "$$d$$p" "$(DESTDIR)$(docdir)/$$f"; \
+	  echo " $(docDATA_INSTALL) '$$d$$p' '$(DESTDIR)$(LIBOFX_DTD_DIR)/$$f'"; \
+	  $(docDATA_INSTALL) "$$d$$p" "$(DESTDIR)$(LIBOFX_DTD_DIR)/$$f"; \
 	done
 
 uninstall-docDATA:
 	@$(NORMAL_UNINSTALL)
 	@list='$(doc_DATA)'; for p in $$list; do \
 	  f=$(am__strip_dir) \
-	  echo " rm -f '$(DESTDIR)$(docdir)/$$f'"; \
-	  rm -f "$(DESTDIR)$(docdir)/$$f"; \
+	  echo " rm -f '$(DESTDIR)$(LIBOFX_DTD_DIR)/$$f'"; \
+	  rm -f "$(DESTDIR)$(LIBOFX_DTD_DIR)/$$f"; \
 	done
 tags: TAGS
 TAGS:
@@ -300,7 +300,7 @@
 check: check-am
 all-am: Makefile $(DATA)
 installdirs:
-	for dir in "$(DESTDIR)$(docdir)"; do \
+	for dir in "$(DESTDIR)$(LIBOFX_DTD_DIR)"; do \
 	  test -z "$$dir" || $(mkdir_p) "$$dir"; \
 	done
 install: install-am
