Description: Add colord profile fix
Author: Alexey Galakhov <agalakhov@gmail.com>
Bug-Debian: http://bugs.debian.org/711868
Last-Update: 2013-06-10

--- cups-1.6.2.orig/scheduler/colorman.c	2013-02-20 05:53:01.000000000 +0600
+++ cups-1.6.2/scheduler/colorman.c	2013-06-10 18:50:49.000000000 +0600
@@ -84,10 +84,13 @@
 #  define COLORD_KIND_PRINTER	"printer"
 					/* printing output device */
 
-#  define COLORD_DBUS_MSG(p,m)	dbus_message_new_method_call(\
+#  define COLORD_DBUS_MSG_PARM(p,x,m)	dbus_message_new_method_call(\
 					"org.freedesktop.ColorManager", (p),\
-                                        "org.freedesktop.ColorManager", (m))
+                                        "org.freedesktop.ColorManager" x, (m))
                                         /* Macro to make new colord messages */
+#  define COLORD_DBUS_MSG(p,m)	COLORD_DBUS_MSG_PARM(p,"",m)
+                                        /* Macro to make new colord messages */
+
 #  define COLORD_DBUS_PATH	"/org/freedesktop/ColorManager"
 					/* Path for color management system */
 #  define COLORD_DBUS_TIMEOUT	5000	/* Timeout for connecting to colord in ms */
@@ -1177,7 +1180,7 @@
   DBusError	error;			/* D-Bus error */
 
 
-  message = COLORD_DBUS_MSG(device_path, "AddProfile");
+  message = COLORD_DBUS_MSG_PARM(device_path, ".Device", "AddProfile");
 
   dbus_message_iter_init_append(message, &args);
   dbus_message_iter_append_basic(&args, DBUS_TYPE_STRING, &relation);
