Author: till.kamppeter@gmail.com

Index: cups-1.6.0/config-scripts/cups-defaults.m4
===================================================================
--- cups-1.6.0.orig/config-scripts/cups-defaults.m4	2012-04-23 19:26:57.000000000 +0200
+++ cups-1.6.0/config-scripts/cups-defaults.m4	2012-07-27 18:08:04.090143383 +0200
@@ -100,7 +100,7 @@
 	default_local_protocols="$withval",
 	default_local_protocols="default")
 
-if test x$with_local_protocols != xno; then
+if test "x$with_local_protocols" != "xno"; then
 	if test "x$default_local_protocols" = "xdefault"; then
 		if test "x$DNSSD_BACKEND" != "x"; then
 			CUPS_BROWSE_LOCAL_PROTOCOLS="dnssd"
