Author: till.kamppeter@gmail.com

--- a/config-scripts/cups-defaults.m4
+++ b/config-scripts/cups-defaults.m4
@@ -100,7 +100,7 @@
 	default_local_protocols="$withval",
 	default_local_protocols="default")
 
-if test x$with_local_protocols != xno; then
+if test "x$with_local_protocols" != "xno"; then
 	if test "x$default_local_protocols" = "xdefault"; then
 		if test "x$DNSSD_BACKEND" != "x"; then
 			CUPS_BROWSE_LOCAL_PROTOCOLS="dnssd"
