Author: till.kamppeter@gmail.com

Index: cups-1.6.0/scheduler/client.c
===================================================================
--- cups-1.6.0.orig/scheduler/client.c	2012-05-08 00:41:30.000000000 +0200
+++ cups-1.6.0/scheduler/client.c	2012-07-27 18:05:28.194145474 +0200
@@ -406,7 +406,7 @@
   {
     if (httpAddrLocalhost(&temp))
       strlcpy(con->servername, "localhost", sizeof(con->servername));
-    else if (HostNameLookups || RemotePort)
+    else if (HostNameLookups)
       httpAddrLookup(&temp, con->servername, sizeof(con->servername));
     else
       httpAddrString(&temp, con->servername, sizeof(con->servername));
Index: cups-1.6.0/scheduler/conf.c
===================================================================
--- cups-1.6.0.orig/scheduler/conf.c	2012-07-27 17:15:59.550185245 +0200
+++ cups-1.6.0/scheduler/conf.c	2012-07-27 18:05:28.194145474 +0200
@@ -834,7 +834,7 @@
     cupsdAddAlias(ServerAlias, temp);
     cupsdLogMessage(CUPSD_LOG_DEBUG, "Added auto ServerAlias %s", temp);
 
-    if (HostNameLookups || RemotePort)
+    if (HostNameLookups)
     {
       struct hostent	*host;		/* Host entry to get FQDN */
 
