Description: Provide a slightly more verbose HTTP_ERROR handling
Author: Michael Sweet <msweet@apple.com>
Origin: upstream
Bug-Debian: http://bugs.debian.org/645436
Last-Update: 2013-06-25
--- a/cups/http-support.c
+++ b/cups/http-support.c
@@ -1268,6 +1268,9 @@
 
   switch (status)
   {
+    case HTTP_ERROR :
+        s = strerror(errno);
+        break;
     case HTTP_CONTINUE :
         s = _("Continue");
 	break;
