Author: till.kamppeter@gmail.com

Index: cups-1.6.0/conf/mime.convs.in
===================================================================
--- cups-1.6.0.orig/conf/mime.convs.in	2011-08-27 11:23:01.000000000 +0200
+++ cups-1.6.0/conf/mime.convs.in	2012-07-27 18:26:48.762128316 +0200
@@ -38,7 +38,7 @@
 # PostScript filters
 #
 
-application/postscript		application/vnd.cups-postscript	66	pstops
+application/postscript		application/vnd.cups-postscript	65	pstops
 
 ########################################################################
 #
