Index: cups-1.4.2/scheduler/printers.c
===================================================================
--- cups-1.4.2.orig/scheduler/printers.c
+++ cups-1.4.2/scheduler/printers.c
@@ -4398,6 +4398,9 @@ load_ppd(cupsd_printer_t *p)		/* I - Pri
     }
 
 #ifdef HAVE_DNSSD
+	/* silence Avahi Bonjour compat warning, since this can cause SIGPIPE
+	 * crashes if stderr is not available */
+	setenv("AVAHI_COMPAT_NOWARN", "1", 0);
     cupsdSetString(&p->product, ppd->product);
 #endif /* HAVE_DNSSD */
 
