Description: Do not write VCS tags into installed conffiles
Bug: http://www.cups.org/str.php?L2369
Author: Kenshi Muto <kmuto@debian.org>
Author: Didier Raboud <odyx@debian.org>

Last-Update: 2012-11-27

--- a/conf/Makefile
+++ b/conf/Makefile
@@ -75,9 +75,13 @@
 			$(INSTALL_CONFIG) -g $(CUPS_GROUP) $$file $(SERVERROOT)/$$file.N ; \
 		else \
 			$(INSTALL_CONFIG) -g $(CUPS_GROUP) $$file $(SERVERROOT) ; \
+			grep -v "\$$Id[:\$$]" $(SERVERROOT)/$$file > $(SERVERROOT)/$$file.n ; \
+			mv $(SERVERROOT)/$$file.n $(SERVERROOT)/$$file ; \
 		fi ; \
 	done
 	$(INSTALL_CONFIG) -g $(CUPS_GROUP) cupsd.conf $(SERVERROOT)/cupsd.conf.default
+	grep -v "\$$Id[:\$$]" $(SERVERROOT)/cupsd.conf.default > $(SERVERROOT)/cupsd.conf.default.n ; \
+	mv $(SERVERROOT)/cupsd.conf.default.n $(SERVERROOT)/cupsd.conf.default ; \
 	$(INSTALL_DIR) -m 755 $(DATADIR)/mime
 	for file in $(REPLACE); do \
 		if test -r $(DATADIR)/mime/$$file ; then \
