Description: Do not write VCS tags into installed conffiles
Bug: http://www.cups.org/str.php?L2369
Author: Kenshi Muto <kmuto@debian.org>

--- cups-1.4.0~svn8773~/conf/Makefile	2009-08-23 12:15:52.000000000 +0200
+++ cups-1.4.0~svn8773/conf/Makefile	2009-08-23 17:11:44.000000000 +0200
@@ -75,6 +75,8 @@
 			$(INSTALL_CONFIG) -g $(CUPS_GROUP) $$file $(SERVERROOT)/$$file.N ; \
 		else \
 			$(INSTALL_CONFIG) -g $(CUPS_GROUP) $$file $(SERVERROOT) ; \
+			grep -v "\$$Id:" $(SERVERROOT)/$$file > $(SERVERROOT)/$$file.n ; \
+			mv $(SERVERROOT)/$$file.n $(SERVERROOT)/$$file ; \
 		fi ; \
 	done
 	$(INSTALL_CONFIG) -g $(CUPS_GROUP) cupsd.conf $(SERVERROOT)/cupsd.conf.default
