Author: till.kamppeter@gmail.com

Index: cups-1.6.0/Makedefs.in
===================================================================
--- cups-1.6.0.orig/Makedefs.in	2012-07-27 17:40:53.000000000 +0200
+++ cups-1.6.0/Makedefs.in	2012-07-27 18:23:23.950131064 +0200
@@ -235,7 +235,6 @@
 # Rules...
 #
 
-.SILENT:
 .SUFFIXES:	.1 .1.gz .1m .1m.gz .3 .3.gz .5 .5.gz .7 .7.gz .8 .8.gz .a .c .cxx .h .man .o .gz
 
 .c.o:
