--- cups-1.3.0~rc2~/scheduler/conf.c	2007-07-30 17:47:10.000000000 +0100
+++ cups-1.3.0~rc2/scheduler/conf.c	2007-08-01 15:33:55.000000000 +0100
@@ -548,7 +548,7 @@
   cupsdClearString(&BrowseLocalOptions);
   cupsdClearString(&BrowseRemoteOptions);
 
-  cupsdSetString(&ErrorPolicy, "stop-printer");
+  cupsdSetString(&ErrorPolicy, "retry-job");
 
 #ifdef HAVE_LDAP
   cupsdClearString(&BrowseLDAPBindDN);
