diff -uNr libopenraw-0.0.8.old/configure.in libopenraw-0.0.8/configure.in
--- libopenraw-0.0.8.old/configure.in	2009-05-15 22:21:53.000000000 +0200
+++ libopenraw-0.0.8/configure.in	2010-10-24 23:20:03.000000000 +0200
@@ -135,8 +135,8 @@
    if test x$HAVE_LIBGLIB = xyes ; then
       PKG_CHECK_MODULES(LIBGDKPIXBUF, gdk-pixbuf-2.0 >= $LIBGDKPIXBUF_REQUIRED,
                      [AC_DEFINE(HAVE_LIBGDKPIXBUF, 1, [Define to 1 to enable libgdkpixbuf support])
-		      GTK_VERSION=`$PKG_CONFIG --variable=gtk_binary_version gtk+-2.0`
-		      GDK_PIXBUF_DIR="gtk-2.0/$GTK_VERSION/loaders"
+		      GDK_PIXBUF_VERSION=`$PKG_CONFIG --variable=gdk_pixbuf_binary_version gdk-pixbuf-2.0`
+                      GDK_PIXBUF_DIR="gdk-pixbuf-2.0/$GDK_PIXBUF_VERSION/loaders"
 		      AC_SUBST(GDK_PIXBUF_DIR)
 		      HAVE_LIBGDKPIXBUF=yes],
                      [HAVE_LIBGDKPIXBUF=no])
