diff -Naur klavaro-1.2.0.rc3.orig/src/translation.c klavaro-1.2.0.rc3/src/translation.c
--- klavaro-1.2.0.rc3.orig/src/translation.c	2009-05-15 23:59:19.000000000 +0300
+++ klavaro-1.2.0.rc3/src/translation.c	2009-05-16 00:03:26.000000000 +0300
@@ -338,7 +338,7 @@
 	if (lang_ok == FALSE)
 	{
 		g_free (tmp_code);
-		tmp_code = g_strdup ("eo");
+		tmp_code = g_strdup ("en_GB");
 		g_message ("as your prefered locales aren't available, "
 			       "we are using this neutral one: \"%s\"", tmp_code);
 	}
