--- hexchat-2.9.6.1.orig/share/icons/Makefile.am	2013-09-12 01:18:19.000000000 +0300
+++ hexchat-2.9.6.1/share/icons/Makefile.am	2014-01-07 15:46:51.006817974 +0200
@@ -4,10 +4,3 @@
 hicolor_DATA = hexchat.svg
 hicolordir = $(datadir)/icons/hicolor/scalable/apps
 
-UPDATE_ICON_CACHE = gtk-update-icon-cache -f -t $(datadir)/icons/hicolor || :
-
-install-data-hook:
-	$(UPDATE_ICON_CACHE);
-
-uninstall-hook:
-	$(UPDATE_ICON_CACHE);
--- hexchat-2.9.6.1.orig/share/icons/Makefile.in	2013-09-12 01:31:20.000000000 +0300
+++ hexchat-2.9.6.1/share/icons/Makefile.in	2014-01-07 15:56:02.933496818 +0200
@@ -360,7 +360,6 @@
 icondir = $(datadir)/icons/hicolor/48x48/apps
 hicolor_DATA = hexchat.svg
 hicolordir = $(datadir)/icons/hicolor/scalable/apps
-UPDATE_ICON_CACHE = gtk-update-icon-cache -f -t $(datadir)/icons/hicolor || :
 all: all-am
 
 .SUFFIXES:
@@ -536,9 +535,6 @@
 
 info-am:
 
-install-data-am: install-hicolorDATA install-iconDATA
-	@$(NORMAL_INSTALL)
-	$(MAKE) $(AM_MAKEFLAGS) install-data-hook
 install-dvi: install-dvi-am
 
 install-dvi-am:
--- hexchat-2.9.6.1.orig/share/misc/hexchat.desktop	2013-09-12 01:18:19.000000000 +0300
+++ hexchat-2.9.6.1/share/misc/hexchat.desktop	2014-01-07 16:31:53.343544262 +0200
@@ -3,6 +3,7 @@
 Name[zh_TW]=網路清談
 GenericName=IRC Client
 GenericName[de]=IRC-Client
+GenericName[tr]=IRC istemcisi
 GenericName[es]=Aplicación de IRC
 GenericName[fi]=IRC-sovellus
 GenericName[fr]=Client IRC
--- hexchat-2.9.6.1.orig/src/fe-gtk/joind.c	2013-09-12 01:18:19.000000000 +0300
+++ hexchat-2.9.6.1/src/fe-gtk/joind.c	2014-01-07 16:54:51.830241343 +0200
@@ -246,7 +246,7 @@
 	if (serv->network)
 		if (g_ascii_strcasecmp(((ircnet*)serv->network)->name, "freenode") == 0)
 		{
-			gtk_entry_set_text (GTK_ENTRY (entry1), "#hexchat");
+			gtk_entry_set_text (GTK_ENTRY (entry1), "#pisi-linux");
 			gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON(radiobutton2), TRUE);
 		}
 
