diff -Nur mc-4.6.2-pre1-old/ChangeLog mc-4.6.2-pre1/ChangeLog
--- mc-4.6.2-pre1-old/ChangeLog	2008-01-26 23:01:58.674393701 +0200
+++ mc-4.6.2-pre1/ChangeLog	2008-01-26 23:02:28.550851790 +0200
@@ -1,4 +1,9 @@
 2007-09-10  Pavel Tsekov  <ptsekov@gmx.net>
+2007-03-19  Martin Petricek  <tux@centrum.cz>
+
+	* src/boxes.c: Dialog for selecting listing mode is up to 15
+	characters wider if size of the screen allows it.
+
 
 	* configure.ac: Update the version number to 4.6.2-pre1.
 	* doc/mc.1.in: Likewise.
diff -Nur mc-4.6.2-pre1-old/src/boxes.c mc-4.6.2-pre1/src/boxes.c
--- mc-4.6.2-pre1-old/src/boxes.c	2008-01-26 23:01:58.658396132 +0200
+++ mc-4.6.2-pre1/src/boxes.c	2008-01-26 23:02:28.551851639 +0200
@@ -143,6 +143,7 @@
     const char *cancel_button = _("&Cancel");
 
     static int button_start = 30;
+    int extra_x = 0;
 
     displays_status = _status;
 
@@ -182,20 +183,27 @@
 
 	i18n_displays_flag = 1;
     }
-    dd = create_dlg (0, 0, DISPLAY_Y, DISPLAY_X, dialog_colors,
+
+    /* Widen the dialog by up to 15 characters if screen is large enough */
+    if (COLS > DISPLAY_X) {
+        extra_x = COLS - DISPLAY_X;
+	if (extra_x > 15) extra_x = 15;
+    }
+    
+    dd = create_dlg (0, 0, DISPLAY_Y, DISPLAY_X + extra_x, dialog_colors,
 		     display_callback, "[Listing Mode...]", display_title,
 		     DLG_CENTER | DLG_REVERSE);
 
     add_widget (dd,
-		button_new (4, button_start, B_CANCEL, NORMAL_BUTTON,
+		button_new (4, button_start + extra_x, B_CANCEL, NORMAL_BUTTON,
 			    cancel_button, 0));
 
     add_widget (dd,
-		button_new (3, button_start, B_ENTER, DEFPUSH_BUTTON,
+		button_new (3, button_start + extra_x, B_ENTER, DEFPUSH_BUTTON,
 			    ok_button, 0));
 
     status =
-	input_new (10, 9, INPUT_COLOR, DISPLAY_X - 14, _status[radio_sel],
+	input_new (10, 9, INPUT_COLOR, DISPLAY_X + extra_x - 14, _status[radio_sel],
 		   "mini-input");
     add_widget (dd, status);
     input_set_point (status, 0);
@@ -205,7 +213,7 @@
     add_widget (dd, check_status);
 
     user =
-	input_new (7, 9, INPUT_COLOR, DISPLAY_X - 14, init_text,
+	input_new (7, 9, INPUT_COLOR, DISPLAY_X + extra_x - 14, init_text,
 		   "user-fmt-input");
     add_widget (dd, user);
     input_set_point (user, 0);
