--- mc-4.6.1a/lib/cedit.menu.changelog	2005-05-27 05:35:12.000000000 +0200
+++ mc-4.6.1a/lib/cedit.menu	2006-01-30 10:46:11.000000000 +0100
@@ -449,6 +449,13 @@
         EMAIL="<$REPLYTO>"
         echo "$DATE  $AUTHOR  $EMAIL" >%b
 
+S       Insert `Spec-file Changelog' string
+        DATE="`date +\"%%a %%b %%e %%Y\"`"
+	MY_UID="`id | sed 's/^.*uid=\([^(]*\).*$/\1/'`"
+	AUTHOR="`awk -F: '$3 == '$MY_UID' {print $5}' /etc/passwd`"
+        EMAIL="<$REPLYTO>"
+        echo "* $DATE $AUTHOR $EMAIL" >%b
+
 s       Invoke `shell'
 	sh
 
