diff -Nurwd mc-4.6.2~pre1.orig/vfs/ftpfs.c mc-4.6.2~pre1/vfs/ftpfs.c
--- mc-4.6.2~pre1.orig/vfs/ftpfs.c	2008-02-13 16:10:32.000000000 +0100
+++ mc-4.6.2~pre1/vfs/ftpfs.c	2008-02-13 16:54:31.000000000 +0100
@@ -1525,7 +1525,7 @@
     char buf[BUF_SMALL];
 
     g_snprintf(buf, sizeof(buf), "SITE CHMOD %4.4o /%%s", mode & 07777);
-    return ftpfs_send_command(me, path, buf, OPT_FLUSH);
+    return ftpfs_send_command(me, path, buf, OPT_FLUSH | OPT_IGNORE_ERROR);
 }
 
 static int ftpfs_chown (struct vfs_class *me, const char *path, int owner, int group)
