diff -Nur mc-4.6.2-old/lib/mc.ext.in mc-4.6.2/lib/mc.ext.in
--- mc-4.6.2-old/lib/mc.ext.in	2009-03-24 12:50:30.000000000 +0200
+++ mc-4.6.2/lib/mc.ext.in	2009-03-24 12:51:38.000000000 +0200
@@ -162,7 +162,7 @@
 # rar
 regex/\.[rR]([aA][rR]|[0-9][0-9])$
 	Open=%cd %p#urar
-	View=%view{ascii} rar v -c- %f
+	View=%view{ascii} unrar v -c- %f
 
 # ALZip
 regex/\.(alz|ALZ)$
diff -Nur mc-4.6.2-old/vfs/extfs/urar.in mc-4.6.2/vfs/extfs/urar.in
--- mc-4.6.2-old/vfs/extfs/urar.in	2009-03-24 12:50:30.000000000 +0200
+++ mc-4.6.2/vfs/extfs/urar.in	2009-03-24 12:50:34.000000000 +0200
@@ -22,6 +22,10 @@
     fi
 done
 
+if [ ! -x $UNRAR -a -x $RAR ]; then
+    UNRAR=$RAR
+fi
+
 mcrarfs_list ()
 {
      $UNRAR v -c- "$1" | @AWK@ -v uid=`id -u` -v gid=`id -g` '
