--- mc-4.6.1a/src/file.c.delcheck	2006-04-28 13:46:33.000000000 +0200
+++ mc-4.6.1a/src/file.c	2006-04-28 13:52:48.000000000 +0200
@@ -1755,6 +1755,12 @@
     free_linklist (&linklist);
     free_linklist (&dest_dirs);
 
+    /* Update panel contents to avoid actions on deleted files */
+    if (!panel->is_panelized) {
+    	update_panels (UP_RELOAD, UP_KEEPSEL);
+    	repaint_screen ();
+    }
+
     if (single_entry) {
 	if (force_single) {
 	    source = selection (panel)->fname;
