diff -Nur mc-4.6.1-old/syntax/ebuild.syntax mc-4.6.1/syntax/ebuild.syntax
--- mc-4.6.1-old/syntax/ebuild.syntax	1970-01-01 02:00:00.000000000 +0200
+++ mc-4.6.1/syntax/ebuild.syntax	2006-05-17 19:41:46.000000000 +0300
@@ -0,0 +1,87 @@
+context default
+    keyword linestart HOMEPAGE brightcyan
+    keyword linestart DESCRIPTION brightcyan
+    keyword linestart SRC_URI brightcyan
+    keyword linestart LICENSE brightcyan
+    keyword linestart SLOT brightcyan
+    keyword linestart KEYWORDS brightcyan
+    keyword linestart IUSE brightcyan
+    keyword linestart RESTRICT brightcyan
+    keyword linestart DEPEND brightcyan
+    keyword linestart RDEPEND brightcyan
+    keyword linestart PROVIDE brightcyan
+    keyword linestart S brightcyan
+
+    keyword whole local yellow
+    keyword whole use yellow
+    keyword whole has_version yellow
+    keyword whole best_version yellow
+    keyword whole use_with yellow
+    keyword whole use_enable yellow
+    keyword whole check_KV yellow
+    keyword whole keepdir yellow
+    keyword whole econf yellow
+    keyword whole einstall yellow
+    keyword whole die yellow
+    keyword whole einfo yellow
+    keyword whole emake yellow
+    keyword whole eerror yellow
+    keyword whole epatch yellow
+    keyword whole ebegin yellow
+    keyword whole eend yellow
+    keyword whole ewarn yellow
+    keyword whole unpack yellow
+
+    keyword whole diropts yellow
+    keyword whole dobin yellow
+    keyword whole docinto yellow
+    keyword whole dodir yellow
+    keyword whole dodoc yellow
+    keyword whole doexe yellow
+    keyword whole dohard yellow
+    keyword whole dohtml yellow
+    keyword whole doinfo yellow
+    keyword whole doins yellow
+    keyword whole dolib.a yellow
+    keyword whole dolib.so yellow
+    keyword whole dolib yellow
+    keyword whole doman yellow
+    keyword whole dosbin yellow
+    keyword whole dosym yellow
+    keyword whole exeinto yellow
+    keyword whole exeopts yellow
+    keyword whole fowners yellow
+    keyword whole fperms yellow
+    keyword whole insinto yellow
+    keyword whole insopts yellow
+    keyword whole into yellow
+    keyword whole libopts yellow
+    keyword whole newbin yellow
+    keyword whole newdoc yellow
+    keyword whole newexe yellow
+    keyword whole newins yellow
+    keyword whole newman yellow
+    keyword whole newsbin yellow
+    keyword whole prepall yellow
+    keyword whole prepalldocs yellow
+    keyword whole prepallinfo yellow
+    keyword whole prepallman yellow
+
+    keyword = yellow
+    keyword : yellow
+    keyword ${+} brightgreen
+
+    keyword linestart abcdefghijklmnopqrstuvwxyz_\(\) brightmagenta
+
+    keyword *() brightcyan
+
+    keyword { magenta
+    keyword } magenta
+
+context linestart inherit \n magenta
+    keyword whole inherit yellow
+
+context # \n red
+
+context " " green
+    keyword wholeright ${+} brightgreen
diff -Nur mc-4.6.1-old/syntax/Makefile.am mc-4.6.1/syntax/Makefile.am
--- mc-4.6.1-old/syntax/Makefile.am	2006-05-17 19:41:38.000000000 +0300
+++ mc-4.6.1/syntax/Makefile.am	2006-05-17 19:42:01.000000000 +0300
@@ -8,6 +8,7 @@
 	changelog.syntax	\
 	diff.syntax		\
 	dos.syntax		\
+	ebuild.syntax		\
 	eiffel.syntax		\
 	fortran.syntax		\
 	html.syntax		\
diff -Nur mc-4.6.1-old/syntax/Makefile.in mc-4.6.1/syntax/Makefile.in
--- mc-4.6.1-old/syntax/Makefile.in	2006-05-17 19:41:38.000000000 +0300
+++ mc-4.6.1/syntax/Makefile.in	2006-05-17 19:42:19.000000000 +0300
@@ -222,6 +222,7 @@
 	changelog.syntax	\
 	diff.syntax		\
 	dos.syntax		\
+	ebuild.syntax		\
 	eiffel.syntax		\
 	fortran.syntax		\
 	html.syntax		\
diff -Nur mc-4.6.1-old/syntax/Syntax mc-4.6.1/syntax/Syntax
--- mc-4.6.1-old/syntax/Syntax	2006-05-17 19:41:38.000000000 +0300
+++ mc-4.6.1/syntax/Syntax	2006-05-17 19:43:30.000000000 +0300
@@ -144,5 +144,8 @@
 file Don_t_match_me Mail\sfolder ^From\s
 include mail.syntax
 
+file .\*\\.(ebuild|eclass)$ Gentoo\sEbuild
+include ebuild.syntax
+
 file .\* unknown
 include unknown.syntax
