--- mc-2006-08-12-18/src/command.c.exit	2005-06-07 16:16:19.000000000 +0200
+++ mc-2006-08-12-18/src/command.c	2006-08-15 12:48:12.000000000 +0200
@@ -214,6 +214,11 @@
 	size_t i, j, cmd_len;
 
 	if (!vfs_current_is_local ()) {
+		if (strcmp (cmd, "exit") == 0) {
+			quiet_quit_cmd ();
+			return MSG_HANDLED;
+		}
+
 	    message (1, MSG_ERROR,
 		     _
 		     (" Cannot execute commands on non-local filesystems"));
