--- src/key.c
+++ src/key.c
@@ -234,6 +234,11 @@
     { KEY_F(9),   ESC_STR "[20~", MCKEY_NOACTION },
     { KEY_F(10),  ESC_STR "[21~", MCKEY_NOACTION },
 
+    { KEY_M_SHIFT | KEY_F(1),   ESC_STR "[1;2P",   MCKEY_NOACTION },
+    { KEY_M_SHIFT | KEY_F(2),   ESC_STR "[1;2Q",   MCKEY_NOACTION },
+    { KEY_M_SHIFT | KEY_F(3),   ESC_STR "[1;2R",   MCKEY_NOACTION },
+    { KEY_M_SHIFT | KEY_F(4),   ESC_STR "[1;2S",   MCKEY_NOACTION },
+
     /* old xterm Shift-arrows */
     { KEY_M_SHIFT | KEY_UP,    ESC_STR "O2A",   MCKEY_NOACTION },
     { KEY_M_SHIFT | KEY_DOWN,  ESC_STR "O2B",   MCKEY_NOACTION },
