diff -up mc-4.6.2-pre1/src/util.c.hintchk mc-4.6.2-pre1/src/util.c
--- mc-4.6.2-pre1/src/util.c.hintchk	2008-03-27 12:39:54.000000000 +0100
+++ mc-4.6.2-pre1/src/util.c	2008-03-27 12:46:39.000000000 +0100
@@ -995,6 +995,11 @@ load_mc_home_file (const char *filename,
     if (hintfile != hintfile_base)
 	g_free (hintfile_base);
 
+    if (!data) {
+	g_free(hintfile);
+	return NULL;
+    }
+
     if (allocated_filename)
 	*allocated_filename = hintfile;
     else
