--- mc-4.6.2/vfs/extfs/u7z
+++ mc-4.6.2/vfs/extfs/u7z
@@ -2,6 +2,7 @@
 #
 # extfs support for p7zip
 # Written by Pavel Roskin <proski@gnu.org>
+# Some Bugfixes/workarounds by Sergiy Niskorodov <sgh@mail.zp.ua>
 # Licensed under GNU GPL version 2 or later version.
 
 P7ZIP=7za
@@ -19,7 +20,10 @@
 
 mcu7zip_copyout ()
 {
-	$P7ZIP e -so "$1" "$2" > "$3" 2>/dev/null
+	#first we check if we have old p7zip archive with prefix ./ in filename
+	$P7ZIP l "$1" "$2" | grep -q "0 files" && \
+	EXFNAME=*./"$2" || EXFNAME="$2"
+	$P7ZIP e -so "$1" "$EXFNAME" > "$3" 2>/dev/null
 }
 
 mcu7zip_copyin ()
@@ -31,19 +35,27 @@
 {
 	dir=`mktemp -d "${MC_TMPDIR:-/tmp}/mctmpdir-u7z.XXXXXX"` || exit 1
 	mkdir -p "$dir"/"$2"
-	$P7ZIP a -w"$dir" "$1" "$2" >/dev/null 2>&1
+	$P7ZIP a -w"$dir" "$1" "$dir"/"$2" >/dev/null 2>&1
 	rm -rf "$dir"
 }
 
 mcu7zip_rm ()
 {
 	# NOTE: Version 4.20 fails to delete files in subdirectories
-	$P7ZIP d "$1" "$2" >/dev/null 2>&1
+	#first we check if we have old p7zip archive with prefix ./ in filename
+	$P7ZIP l "$1" "$2" | grep -q "0 files" && \
+	EXFNAME=*./"$2" || EXFNAME="$2"
+	$P7ZIP d "$1" "$EXFNAME" 2>&1 | grep -q E_NOTIMPL > /dev/null 2>&1 && \
+	{ printf "Function not implemented...\n7z cannot delete from solid archive.\n" >&2 ; exit 1 ; }
 }
 
 mcu7zip_rmdir ()
 {
-	$P7ZIP d "$1" "$2"/ >/dev/null 2>&1
+	#first we check if we have old p7zip archive with prefix ./ in filename
+	$P7ZIP l "$1" "$2" | grep -q "0 files" && \
+	EXFNAME=*./"$2" || EXFNAME="$2"
+	$P7ZIP d "$1" "$EXFNAME"/ 2>&1 | grep -q E_NOTIMPL > /dev/null 2>&1  && \
+	{ printf "Function not implemented...\n7z cannot delete from solid archive.\n" >&2 ; exit 1 ; }
 }
 
 # override any locale for dates
