diff -Nur mc-4.6.2-old/lib/mc-wrapper.sh.in mc-4.6.2/lib/mc-wrapper.sh.in
--- mc-4.6.2-old/lib/mc-wrapper.sh.in	2009-03-17 19:36:13.000000000 +0200
+++ mc-4.6.2/lib/mc-wrapper.sh.in	2009-03-17 19:42:15.000000000 +0200
@@ -1,5 +1,6 @@
 MC_USER=`id | sed 's/[^(]*(//;s/).*//'`
 MC_PWD_FILE="${TMPDIR-/tmp}/mc-$MC_USER/mc.pwd.$$"
+test ! -d $HOME/.mc && mkdir $HOME/.mc
 @bindir@/mc -P "$MC_PWD_FILE" "$@"
 
 if test -r "$MC_PWD_FILE"; then
