diff -Nur mc-4.6.2-pre1-old/lib/mc.ext.in mc-4.6.2-pre1/lib/mc.ext.in
--- mc-4.6.2-pre1-old/lib/mc.ext.in	2008-05-12 06:07:22.000000000 +0300
+++ mc-4.6.2-pre1/lib/mc.ext.in	2008-05-12 06:16:19.000000000 +0300
@@ -492,14 +492,17 @@
 	Open=(if test -n "$DISPLAY"; then (firefox file://%d/%p &) 1>&2; else links -force-html %f || ${PAGER:-more} %f; fi) 2>/dev/null
 	View=%view{ascii} links -dump -force-html %f
 
-# StarOffice 5.2
-shell/.sdw
-	Open=(ooffice %f &)
-
-# StarOffice 6 and OpenOffice.org formats
-regex/\.(odt|ott|sxw|stw|ods|ots|sxc|stc|odp|otp|sxi|sti|odg|otg|sxd|std|odb|odf|sxm|odm|sxg)$
-	Open=(ooffice %f &)
-	View=%view{ascii} odt2txt %f
+# Openoffice Writer
+regex/\.(odt|ott|sxw|stw|doc|rtf|pdb|fodt|psw)$
+    Open=(oowriter %f &>/dev/null &)
+
+# Openoffice Calc
+regex/\.(ods|ots|sxc|stc|dif|dbs|xls|xlt|slk|fods|pxl)$
+    Open=(oocalc %f &>/dev/null &)
+
+# Openoffice Impress
+regex/\.(odp|otp|sxi|sti|ppt|pot|sxd|fodp|odg)$
+    Open=(ooimpress %f &>/dev/null &)

 # AbiWord
 shell/.abw
@@ -527,7 +530,7 @@
 
 # Use OpenOffice.org to open any MS Office documents
 type/^Microsoft\ Office\ Document
-	Open=(ooffice %f &)
+	Open=(oowriter %f &)
 
 # Framemaker
 type/^FrameMaker
