diff -Nur mc-4.6.2-pre1-old/lib/mc.ext.in mc-4.6.2-pre1/lib/mc.ext.in
--- mc-4.6.2-pre1-old/lib/mc.ext.in	2008-05-21 17:58:47.000000000 +0300
+++ mc-4.6.2-pre1/lib/mc.ext.in	2008-05-21 18:01:50.000000000 +0300
@@ -414,8 +414,56 @@
 regex/\.([oO][gG][mM])$
 	Include=video
 
+regex/\.([dD][aA][tT])$
+	Include=video
+
+regex/\.([fF][lL][iIvVcC])$
+	Include=video
+
+regex/\.([nN][uU][tT])$
+	Include=video
+
+regex/\.([sS][wW][fF])$
+	Include=video
+
+regex/\.([mM][kK42][vV])$
+	Include=video
+
+regex/\.([dD][vV])$
+	Include=video
+
+regex/\.([nN][sS][vV])$
+	Include=video
+
+regex/\.([yY][uU][vV]?4)$
+	Include=video
+
+regex/\.(3[gG][pP])$
+	Include=video
+
+regex/\.([mM][pP][cC])$
+	Include=video
+
+regex/\.([mM][jJ][pP][gG])$
+	Include=video
+
+regex/\.([mM][jJ][pP][eE][gG])$
+	Include=video
+
+regex/\.([mM][xX][fF])$
+	Include=video
+
+#regex/\.([rR][aA]?[mM])$
+#	Open=(realplay %f >/dev/null 2>&1 &)
+
+regex/\.([rR][mM][vV][bB])$
+	Include=video
+
 regex/\.([rR][aA]?[mM])$
-	Open=(realplay %f >/dev/null 2>&1 &)
+	Include=video
+
+regex/\.(3[gG]?[pP])$
+	Include=video
 
 include/video
 	Open=(mplayer %f >/dev/null 2>&1 &)
