diff -Nur mc-4.6.1-old/lib/mc.ext.in mc-4.6.1/lib/mc.ext.in
--- mc-4.6.1-old/lib/mc.ext.in	2006-12-12 13:03:52.000000000 +0200
+++ mc-4.6.1/lib/mc.ext.in	2006-12-12 13:04:18.000000000 +0200
@@ -334,7 +334,8 @@
 	View=sxpm %f
 
 include/image
-	Open=if [ "$DISPLAY" = "" ]; then zgv %f; else (gqview %f &); fi
+	# Open=if [ "$DISPLAY" = "" ]; then zgv %f; else (gqview %f &); fi
+	Open=if [ "$DISPLAY" = "" ]; then zgv %f; else (gwenview %f &>/dev/null &); fi
 	View=%view{ascii} identify %f
 	#View=%view{ascii} asciiview %f
 
