diff -Nur mc-4.6.1-old/lib/mc.ext.in mc-4.6.1/lib/mc.ext.in
--- mc-4.6.1-old/lib/mc.ext.in	2006-12-12 13:12:29.000000000 +0200
+++ mc-4.6.1/lib/mc.ext.in	2006-12-12 13:13:27.000000000 +0200
@@ -458,12 +458,13 @@
 
 # Postscript
 type/^PostScript
-	Open=(gv %f &)
+	Open=(kpdf %f &)
 	View=%view{ascii} ps2ascii %f
 
 # PDF
 type/^PDF
-	Open=(xpdf %f &)
+	Open=(kpdf %f &)
+	#Open=(xpdf %f &)
 	#Open=(acroread %f &)
 	#Open=(ghostview %f &)
 	View=%view{ascii} pdftotext %f -
