diff -Nur mc-4.6.1-old/lib/mc.ext.in mc-4.6.1/lib/mc.ext.in
--- mc-4.6.1-old/lib/mc.ext.in	2006-12-12 13:07:59.000000000 +0200
+++ mc-4.6.1/lib/mc.ext.in	2006-12-12 13:10:52.000000000 +0200
@@ -520,7 +520,8 @@
 
 # DVI
 regex/\.([dD][vV][iI])$
-	Open=if [ x$DISPLAY = x ]; then dvisvga %f; else (xdvi %f &); fi
+	# Open=if [ x$DISPLAY = x ]; then dvisvga %f; else (xdvi %f &); fi
+	Open=if [ x$DISPLAY = x ]; then dvisvga %f; else (kdvi %f &); fi
 	View=%view{ascii} dvi2tty %f
 
 # TeX
