--- mc-2006-11-14-16/src/main.c.newlinedir	2006-11-20 14:54:21.000000000 +0100
+++ mc-2006-11-14-16/src/main.c	2006-11-20 14:47:52.000000000 +0100
@@ -647,7 +647,7 @@
     }
     directory = *new_dir ? new_dir : home_dir;
 
-    if (mc_chdir (directory) == -1) {
+    if (strchr(directory,'\n') || mc_chdir (directory) == -1) {
 	strcpy (panel->cwd, olddir);
 	g_free (olddir);
 	g_free (translated_url);
