diff -Nurwd mc-4.6.2~pre1/vfs/cpio.c mc-4.6.2~pre1/vfs/cpio.c
--- mc-4.6.2~pre1/vfs/cpio.c	2005-05-27 05:35:16.000000000 +0200
+++ mc-4.6.2~pre1/vfs/cpio.c	2008-01-24 23:07:35.000000000 +0100
@@ -90,7 +90,7 @@
 };
 
 /* FIXME: should be off_t instead of int. */
-static int cpio_position;
+static off_t cpio_position;
 
 static int cpio_find_head(struct vfs_class *me, struct vfs_s_super *super);
 static int cpio_read_bin_head(struct vfs_class *me, struct vfs_s_super *super);
@@ -114,7 +114,7 @@
     return l;
 }
 
-static int cpio_skip_padding(struct vfs_s_super *super)
+static off_t cpio_skip_padding(struct vfs_s_super *super)
 {
     switch(super->u.arch.type) {
     case CPIO_BIN:
diff -Nurwd mc-4.6.2~pre1/vfs/direntry.c mc-4.6.2~pre1/vfs/direntry.c
--- mc-4.6.2~pre1/vfs/direntry.c	2006-03-30 15:04:50.000000000 +0200
+++ mc-4.6.2~pre1/vfs/direntry.c	2008-01-24 23:09:01.000000000 +0100
@@ -849,7 +849,7 @@
     return 0;
 }
 
-static int
+static off_t
 vfs_s_lseek (void *fh, off_t offset, int whence)
 {
     off_t size = FH->ino->st.st_size;
@@ -858,7 +858,7 @@
         vfs_die ("cannot lseek() after linear_read!");
 
     if (FH->handle != -1){	/* If we have local file opened, we want to work with it */
-	int retval = lseek (FH->handle, offset, whence);
+	off_t retval = lseek (FH->handle, offset, whence);
 	if (retval == -1)
 	    FH->ino->super->me->verrno = errno;
 	return retval;
diff -Nurwd mc-4.6.2~pre1/vfs/extfs.c mc-4.6.2~pre1/vfs/extfs.c
--- mc-4.6.2~pre1/vfs/extfs.c	2006-03-23 16:00:50.000000000 +0100
+++ mc-4.6.2~pre1/vfs/extfs.c	2008-01-24 23:06:15.000000000 +0100
@@ -1140,7 +1140,7 @@
     return 0;
 }
 
-static int extfs_lseek (void *data, off_t offset, int whence)
+static off_t extfs_lseek (void *data, off_t offset, int whence)
 {
     struct pseudofile *file = (struct pseudofile *) data;
 
diff -Nurwd mc-4.6.2~pre1/vfs/local.c mc-4.6.2~pre1/vfs/local.c
--- mc-4.6.2~pre1/vfs/local.c	2005-07-05 23:01:57.000000000 +0200
+++ mc-4.6.2~pre1/vfs/local.c	2008-01-24 23:06:15.000000000 +0100
@@ -226,7 +226,7 @@
     return chdir (path);
 }
 
-int
+off_t
 local_lseek (void *data, off_t offset, int whence)
 {
     int fd = * (int *) data;
diff -Nurwd mc-4.6.2~pre1/vfs/local.h mc-4.6.2~pre1/vfs/local.h
--- mc-4.6.2~pre1/vfs/local.h	2005-07-05 23:01:57.000000000 +0200
+++ mc-4.6.2~pre1/vfs/local.h	2008-01-24 23:09:40.000000000 +0100
@@ -11,6 +11,6 @@
 extern int local_read (void *data, char *buffer, int count);
 extern int local_fstat (void *data, struct stat *buf);
 extern int local_errno (struct vfs_class *me);
-extern int local_lseek (void *data, off_t offset, int whence);
+extern off_t local_lseek (void *data, off_t offset, int whence);
 
 #endif
diff -Nurwd mc-4.6.2~pre1/vfs/mcfs.c mc-4.6.2~pre1/vfs/mcfs.c
--- mc-4.6.2~pre1/vfs/mcfs.c	2005-06-08 02:34:55.000000000 +0200
+++ mc-4.6.2~pre1/vfs/mcfs.c	2008-01-24 23:06:15.000000000 +0100
@@ -1068,7 +1068,7 @@
     return 0;
 }
 
-static int
+static off_t
 mcfs_lseek (void *data, off_t offset, int whence)
 {
     mcfs_handle *info = (mcfs_handle *) data;
diff -Nurwd mc-4.6.2~pre1/vfs/smbfs.c mc-4.6.2~pre1/vfs/smbfs.c
--- mc-4.6.2~pre1/vfs/smbfs.c	2005-11-11 03:22:02.000000000 +0100
+++ mc-4.6.2~pre1/vfs/smbfs.c	2008-01-24 23:06:15.000000000 +0100
@@ -1624,7 +1624,7 @@
 
 #define smbfs_lstat smbfs_stat	/* no symlinks on smb filesystem? */
 
-static int
+static off_t
 smbfs_lseek (void *data, off_t offset, int whence)
 {
     smbfs_handle *info = (smbfs_handle *) data;
diff -Nurwd mc-4.6.2~pre1/vfs/tar.c mc-4.6.2~pre1/vfs/tar.c
--- mc-4.6.2~pre1/vfs/tar.c	2006-04-18 09:27:21.000000000 +0200
+++ mc-4.6.2~pre1/vfs/tar.c	2008-01-24 23:06:15.000000000 +0100
@@ -209,7 +209,7 @@
 }
 
 /* As we open one archive at a time, it is safe to have this static */
-static int current_tar_position = 0;
+static off_t current_tar_position = 0;
 
 /* Returns fd of the open tar file */
 static int
@@ -521,7 +521,7 @@
 	struct stat st;
 	struct vfs_s_entry *entry;
 	struct vfs_s_inode *inode, *parent;
-	long data_position;
+	off_t data_position;
 	char *q;
 	int len;
 	char *current_file_name, *current_link_name;
@@ -750,8 +750,9 @@
     int fd = FH_SUPER->u.arch.fd;
     struct vfs_class *me = FH_SUPER->me;
 
-    if (mc_lseek (fd, begin + FH->pos, SEEK_SET) != 
-        begin + FH->pos) ERRNOR (EIO, -1);
+
+    off_t o = mc_lseek(fd, begin + FH->pos, SEEK_SET);
+    if ( o != begin + FH->pos) ERRNOR (EIO, -1);
 
     count = MIN(count, FH->ino->st.st_size - FH->pos);
 
diff -Nurwd mc-4.6.2~pre1/vfs/undelfs.c mc-4.6.2~pre1/vfs/undelfs.c
--- mc-4.6.2~pre1/vfs/undelfs.c	2005-05-27 05:35:16.000000000 +0200
+++ mc-4.6.2~pre1/vfs/undelfs.c	2008-01-24 23:06:15.000000000 +0100
@@ -648,7 +648,7 @@
 }
 
 /* this has to stay here for now: vfs layer does not know how to emulate it */
-static int
+static off_t
 undelfs_lseek(void *vfs_info, off_t offset, int whence)
 {
     return -1;
diff -Nurwd mc-4.6.2~pre1/vfs/vfs.c mc-4.6.2~pre1/vfs/vfs.c
--- mc-4.6.2~pre1/vfs/vfs.c	2005-07-05 23:01:57.000000000 +0200
+++ mc-4.6.2~pre1/vfs/vfs.c	2008-01-24 23:06:15.000000000 +0100
@@ -627,14 +627,14 @@
 off_t mc_lseek (int fd, off_t offset, int whence)
 {
     struct vfs_class *vfs;
-    int result;
+    off_t result;
 
     if (fd == -1)
 	return -1;
 
     vfs = vfs_op (fd);
     result = vfs->lseek ? (*vfs->lseek)(vfs_info (fd), offset, whence) : -1;
-    if (result == -1)
+    if (result == (off_t)-1)
         errno = vfs->lseek ? ferrno (vfs) : E_NOTSUPP;
     return result;
 }
diff -Nurwd mc-4.6.2~pre1/vfs/vfs-impl.h mc-4.6.2~pre1/vfs/vfs-impl.h
--- mc-4.6.2~pre1/vfs/vfs-impl.h	2005-11-15 23:27:45.000000000 +0100
+++ mc-4.6.2~pre1/vfs/vfs-impl.h	2008-01-24 23:06:15.000000000 +0100
@@ -70,7 +70,7 @@
     int (*rename) (struct vfs_class *me, const char *p1, const char *p2);
     int (*chdir) (struct vfs_class *me, const char *path);
     int (*ferrno) (struct vfs_class *me);
-    int (*lseek) (void *vfs_info, off_t offset, int whence);
+    off_t (*lseek) (void *vfs_info, off_t offset, int whence);
     int (*mknod) (struct vfs_class *me, const char *path, int mode, int dev);
 
     vfsid (*getid) (struct vfs_class *me, const char *path);
diff -Nurwd mc-4.6.2~pre1/vfs/xdirentry.h mc-4.6.2~pre1/vfs/xdirentry.h
--- mc-4.6.2~pre1/vfs/xdirentry.h	2006-01-27 23:16:45.000000000 +0100
+++ mc-4.6.2~pre1/vfs/xdirentry.h	2008-01-24 23:06:15.000000000 +0100
@@ -94,7 +94,7 @@
     char *linkname;		/* Symlink's contents */
     char *localname;		/* Filename of local file, if we have one */
     struct timeval timestamp;	/* Subclass specific */
-    long data_offset;		/* Subclass specific */
+    off_t data_offset;		/* Subclass specific */
 };
 
 /* Data associated with an open file */
