--- mc-4.6.2-pre1/lib/mc.menu
+++ mc-4.6.2-pre1/lib/mc.menu
@@ -186,6 +186,33 @@
 	fi
 	echo "Please test the output file before deleting anything"
 
++ & t r & ! t t
+m       Install as MEMO on palm pilot
+	echo "Installing MEMO..."
+	install-memo %f
+
+=+ f \.pdb$ | f \.prc$ | f \.pqa$ | f \.PDB$ | f \.PRC$ | f \.PQA$ & t r & ! t t
+p       Install on palm pilot (programs or databases)
+        echo "Installing file on PALM"
+	pilot-xfer -i %f
+
++ t t
+M       Install tagged files as MEMOs on palm pilot
+        for i in %t
+        do
+	  echo "Installing MEMO: $i"
+        done
+	install-memo -t %t
+
++ t t
+P       Install tagged files on palm pilot (programs or databases)
+        echo "Installing files on PALM"
+        for i in %t
+        do
+	  echo "Installing file: $i"
+        done
+	pilot-xfer -i %t
+
 =+ f \.tar\.gz$ | f \.tar\.z$ | f \.tgz$ | f \.tpz$ | f \.tar\.Z$| f \.tar\.bz2$ & t r
 x       Extract the contents of a compressed tar file
 	unset EXT
