diff -Nur mc-4.6.2-old/src/Makefile.in mc-4.6.2/src/Makefile.in
--- mc-4.6.2-old/src/Makefile.in	2009-03-15 00:47:14.000000000 +0200
+++ mc-4.6.2/src/Makefile.in	2009-03-15 00:47:19.000000000 +0200
@@ -70,7 +70,7 @@
 PROGRAMS = $(bin_PROGRAMS) $(noinst_PROGRAMS) $(pkglibexec_PROGRAMS)
 am_cons_saver_OBJECTS = cons.saver.$(OBJEXT)
 cons_saver_OBJECTS = $(am_cons_saver_OBJECTS)
-cons_saver_LDADD = $(LDADD)
+cons_saver_LDADD = $(LDADD) -Wl,-z,now
 am_ecs_test_OBJECTS = ecs-test.$(OBJEXT) ecs.$(OBJEXT)
 ecs_test_OBJECTS = $(am_ecs_test_OBJECTS)
 am__DEPENDENCIES_1 =
