--- syntax/spec.syntax
+++ syntax/spec.syntax
@@ -11,6 +11,7 @@
     keyword whole Description: green    
     keyword whole Distribution: green    
     keyword whole Doc\{Dd\}ir: green    
+    keyword whole Enhances: green
     keyword whole Epoch: green
     keyword whole Exclude\{Aa\}rch: green
     keyword whole Exclusive\{Aa\}rch: green
@@ -27,16 +28,19 @@
     keyword whole Prefix: green
     keyword whole Pre\{Rr\}eq: green    
     keyword whole Provides: green    
+    keyword whole Recommends: green
     keyword whole Release: green    
     keyword whole Requires: green    
     keyword whole Requires(\[abcdefghijklmnopqrstuvwxyz\]): green
     keyword whole Root: green    
     keyword whole Serial: white
     keyword whole Source\[0123456789\]: green
+    keyword whole Suggests: green
     keyword whole Summary: green
     keyword whole Summary(\[abcdefghijklmnopqrstuvwxyz\]): green
     keyword whole Summary(\[abcdefghijklmnopqrstuvwxyz\]_\[ABCDEFGHIJKLMNOPQRSTUVWXYZ\]): green
     keyword whole Summary(\[abcdefghijklmnopqrstuvwxyz\]_\[ABCDEFGHIJKLMNOPQRSTUVWXYZ\].\[ABCDEFGHIJKLMNOPQRSTUVWXYZ-1234567890\]): green
+    keyword whole Supplements: green
     keyword whole Vendor: green
     keyword whole Version: green
     keyword whole U\{Rr\}\{Ll\}: green
--- vfs/extfs/rpm
+++ vfs/extfs/rpm
@@ -118,6 +118,14 @@
     if test "$MCFASTRPM" != 0 ; then
     $RPM -qp --qf "[%{REQUIRENAME}\n]" "$f" | grep "(none)" > /dev/null ||
 	echo "$FILEPREF 0 $DATE INFO/REQUIRENAME"
+    $RPM -qp --qf "[%|ENHANCESFLAGS:depflag_strong?{}:{%{ENHANCESNAME} %{ENHANCESFLAGS:depflags} %{ENHANCESVERSION}\n}|]" "$f" | grep "(none)" > /dev/null ||
+	echo "$FILEPREF 0 $DATE INFO/ENHANCES"
+    $RPM -qp --qf "[%|SUGGESTSFLAGS:depflag_strong?{}:{%{SUGGESTSNAME} %{SUGGESTSFLAGS:depflags} %{SUGGESTSVERSION}\n}|]" "$f" | grep "(none)" > /dev/null ||
+	echo "$FILEPREF 0 $DATE INFO/SUGGESTS"
+    $RPM -qp --qf "[%|SUGGESTSFLAGS:depflag_strong?{%{SUGGESTSNAME} %{SUGGESTSFLAGS:depflags} %{SUGGESTSVERSION}\n}|]" "$f" | grep "(none)" > /dev/null ||
+	echo "$FILEPREF 0 $DATE INFO/ENHANCES"
+    $RPM -qp --qf "[%|ENHANCESFLAGS:depflag_strong?{%{ENHANCESNAME} %{ENHANCESFLAGS:depflags} %{ENHANCESVERSION}\n}|]" "$f" | grep "(none)" > /dev/null ||
+	echo "$FILEPREF 0 $DATE INFO/SUGGESTS"
     $RPM -qp --qf "[%{OBSOLETES}\n]" "$f" | grep "(none)" > /dev/null ||
 	echo "$FILEPREF 0 $DATE INFO/OBSOLETES"
     $RPM -qp --qf "[%{PROVIDES}\n]" "$f" | grep "(none)" > /dev/null ||
@@ -128,8 +136,10 @@
        echo "$FILEPREF 0 $DATE INFO/CHANGELOG"
     else 
 	echo "$FILEPREF 0 $DATE INFO/REQUIRENAME"
-	echo "$FILEPREF 0 $DATE INFO/OBSOLETES"
-	echo "$FILEPREF 0 $DATE INFO/PROVIDES"
+	echo "$FILEPREF 0 $DATE INFO/ENHANCES"
+	echo "$FILEPREF 0 $DATE INFO/SUGGESTS"
+	echo "$FILEPREF 0 $DATE INFO/RECOMMENDS"
+	echo "$FILEPREF 0 $DATE INFO/SUPPLEMENTS"
 	echo "$FILEPREF 0 $DATE INFO/CONFLICTS"
         echo "$FILEPREF 0 $DATE INFO/CHANGELOG"
     fi
@@ -162,6 +172,10 @@
 	INFO/LICENSE)		$RPM -qp --qf "%{LICENSE}\n" "$f" >"$3"; exit 0;;
 	INFO/RPMVERSION)	$RPM -qp --qf "%{RPMVERSION}\n" "$f" >"$3"; exit 0;;
 	INFO/REQUIRENAME)	$RPM -qp --qf "[%{REQUIRENAME} %{REQUIREFLAGS:depflags} %{REQUIREVERSION}\n]" "$f" >"$3"; exit 0;;
+	INFO/ENHANCES)		$RPM -qp --qf "[%|ENHANCESFLAGS:depflag_strong?{}:{%{ENHANCESNAME} %{ENHANCESFLAGS:depflags} %{ENHANCESVERSION}\n}|]" "$f" >"$3"; exit 0;;
+	INFO/SUGGESTS)		$RPM -qp --qf "[%|SUGGESTSFLAGS:depflag_strong?{}:{%{SUGGESTSNAME} %{SUGGESTSFLAGS:depflags} %{SUGGESTSVERSION}\n}|]" "$f" >"$3"; exit 0;;
+	INFO/RECOMMENDS)	$RPM -qp --qf "[%|SUGGESTSFLAGS:depflag_strong?{%{SUGGESTSNAME} %{SUGGESTSFLAGS:depflags} %{SUGGESTSVERSION}\n}|]" "$f" >"$3"; exit 0;;
+	INFO/SUPPLEMENTS)	$RPM -qp --qf "[%|ENHANCESFLAGS:depflag_strong?{%{ENHANCESNAME} %{ENHANCESFLAGS:depflags} %{ENHANCESVERSION}\n}|]" "$f" >"$3"; exit 0;;
 	INFO/PROVIDES)		$RPM -qp --qf "[%{PROVIDES}\n]" "$f" >"$3"; exit 0;;
 	INFO/SCRIPTS/PREIN)	$RPM -qp --qf "%{RPMTAG_PREIN}\n" "$f" >"$3"; exit 0;;
 	INFO/SCRIPTS/POSTIN)	$RPM -qp --qf "%{RPMTAG_POSTIN}\n" "$f" >"$3"; exit 0;;
