diff -Nur mc-4.6.1-old/vfs/fish.c mc-4.6.1/vfs/fish.c
--- mc-4.6.1-old/vfs/fish.c	2007-05-13 16:25:10.495271460 +0300
+++ mc-4.6.1/vfs/fish.c	2007-05-13 16:25:17.761166876 +0300
@@ -209,7 +209,8 @@
 fish_open_archive_int (struct vfs_class *me, struct vfs_s_super *super)
 {
     {
-	const char *argv[10];
+	char gbuf[10];
+	const char *argv[10];	/* All of 10 is used now */
 	const char *xsh = (SUP.flags == FISH_FLAG_RSH ? "rsh" : "ssh");
 	int i = 0;
 
@@ -219,6 +220,11 @@
 #endif
 	if (SUP.flags == FISH_FLAG_COMPRESSED)
 	    argv[i++] = "-C";
+	if (SUP.flags > FISH_FLAG_RSH) {
+	    argv[i++] = "-p";
+	    g_snprintf (gbuf, sizeof (gbuf), "%d", SUP.flags);
+	    argv[i++] = gbuf;
+	}
 	argv[i++] = "-l";
 	argv[i++] = SUP.user;
 	argv[i++] = SUP.host;
@@ -316,7 +322,7 @@
     SUP.user = user;
     SUP.flags = flags;
     if (!strncmp (op, "rsh:", 4))
-	SUP.flags |= FISH_FLAG_RSH;
+	SUP.flags = FISH_FLAG_RSH;
     SUP.cwdir = NULL;
     if (password)
 	SUP.password = password;
@@ -870,22 +876,23 @@
 fish_fill_names (struct vfs_class *me, fill_names_f func)
 {
     struct vfs_s_super *super = MEDATA->supers;
-    const char *flags;
     char *name;
-    
+    char gbuf[10];
+
     while (super){
-	switch (SUP.flags & (FISH_FLAG_RSH | FISH_FLAG_COMPRESSED)) {
+	const char *flags = "";
+	switch (SUP.flags) {
 	case FISH_FLAG_RSH:
 		flags = ":r";
 		break;
 	case FISH_FLAG_COMPRESSED:
 		flags = ":C";
 		break;
-	case FISH_FLAG_RSH | FISH_FLAG_COMPRESSED:
-		flags = "";
-		break;
 	default:
-		flags = "";
+		if (SUP.flags > FISH_FLAG_RSH) {
+		    g_snprintf (gbuf, sizeof (gbuf), ":%d", SUP.flags);
+		    flags = gbuf;
+		}
 		break;
 	}
 
