--- src/setup.c	26 Sep 2007 10:22:25 -0000	1.93
+++ src/setup.c	27 Feb 2008 18:48:37 -0000
@@ -216,6 +216,8 @@
     { "editor_option_typewriter_wrap", &option_typewriter_wrap },
     { "editor_edit_confirm_save", &edit_confirm_save },
     { "editor_syntax_highlighting", &option_syntax_highlighting },
+    { "editor_visible_tabs", &visible_tabs },
+    { "editor_visible_spaces", &visible_tws },
 #endif /* USE_INTERNAL_EDIT */
 
     { "nice_rotating_dash", &nice_rotating_dash },
--- edit/edit.h	24 Sep 2007 09:28:35 -0000	1.88
+++ edit/edit.h	27 Feb 2008 18:48:37 -0000
@@ -298,6 +298,9 @@
 extern int edit_confirm_save;
 extern int column_highlighting;
 
+extern int visible_tabs;
+extern int visible_tws;
+
 /* File names */
 #define EDIT_DIR           ".mc" PATH_SEP_STR "cedit"
 #define SYNTAX_FILE        EDIT_DIR PATH_SEP_STR "Syntax"
