Index: gdal-1.10.0/frmts/netcdf/gmtdataset.cpp
===================================================================
--- gdal-1.10.0.orig/frmts/netcdf/gmtdataset.cpp
+++ gdal-1.10.0/frmts/netcdf/gmtdataset.cpp
@@ -29,7 +29,7 @@
 
 #include "gdal_pam.h"
 #include "gdal_frmts.h"
-#include "netcdf.h"
+#include "netcdf/netcdf.h"
 #include "cpl_multiproc.h"
 
 CPL_CVSID("$Id: gmtdataset.cpp 25593 2013-02-02 13:03:03Z rouault $");
Index: gdal-1.10.0/frmts/netcdf/netcdfdataset.h
===================================================================
--- gdal-1.10.0.orig/frmts/netcdf/netcdfdataset.h
+++ gdal-1.10.0/frmts/netcdf/netcdfdataset.h
@@ -36,7 +36,7 @@
 #include "gdal_frmts.h"
 #include "cpl_string.h"
 #include "ogr_spatialref.h"
-#include "netcdf.h"
+#include "netcdf/netcdf.h"
 
 
 /************************************************************************/
