diff -up gdal-1.9.2-fedora/swig/include/java/gdalconst_java.i.jni gdal-1.9.2-fedora/swig/include/java/gdalconst_java.i
--- gdal-1.9.2-fedora/swig/include/java/gdalconst_java.i.jni	2012-10-08 18:58:28.000000000 -0600
+++ gdal-1.9.2-fedora/swig/include/java/gdalconst_java.i	2013-06-12 14:40:22.858828066 -0600
@@ -22,7 +22,7 @@
 
   static {
     try {
-      System.loadLibrary("gdalconstjni");
+      System.load("/usr/lib/java/gdal/libgdalconstjni.so");
       available = true;
     } catch (UnsatisfiedLinkError e) {
       available = false;
diff -up gdal-1.9.2-fedora/swig/include/java/gdal_java.i.jni gdal-1.9.2-fedora/swig/include/java/gdal_java.i
--- gdal-1.9.2-fedora/swig/include/java/gdal_java.i.jni	2012-10-08 18:58:28.000000000 -0600
+++ gdal-1.9.2-fedora/swig/include/java/gdal_java.i	2013-06-12 14:40:39.142749339 -0600
@@ -22,7 +22,7 @@
 
   static {
     try {
-      System.loadLibrary("gdaljni");
+      System.load("/usr/lib/java/gdal/libgdaljni.so");
       available = true;
       
       if (gdal.HasThreadSupport() == 0)
diff -up gdal-1.9.2-fedora/swig/include/java/ogr_java.i.jni gdal-1.9.2-fedora/swig/include/java/ogr_java.i
--- gdal-1.9.2-fedora/swig/include/java/ogr_java.i.jni	2012-10-08 18:58:28.000000000 -0600
+++ gdal-1.9.2-fedora/swig/include/java/ogr_java.i	2013-06-12 14:39:47.946997388 -0600
@@ -24,7 +24,7 @@
 
   static {
     try {
-      System.loadLibrary("ogrjni");
+      System.load("/usr/lib/java/gdal/libogrjni.so");
       available = true;
       
       if (org.gdal.gdal.gdal.HasThreadSupport() == 0)
diff -up gdal-1.9.2-fedora/swig/include/java/osr_java.i.jni gdal-1.9.2-fedora/swig/include/java/osr_java.i
--- gdal-1.9.2-fedora/swig/include/java/osr_java.i.jni	2012-10-08 18:58:28.000000000 -0600
+++ gdal-1.9.2-fedora/swig/include/java/osr_java.i	2013-06-12 14:41:10.097554983 -0600
@@ -26,7 +26,7 @@
 
   static {
     try {
-      System.loadLibrary("osrjni");
+      System.load("/usr/lib/java/gdal/libosrjni.so");
       available = true;
     } catch (UnsatisfiedLinkError e) {
       available = false;
