diff -Nur gdal-1.9.1-fedora/configure gdal-1.9.1-fedora-dods3.11.3/configure
--- gdal-1.9.1-fedora/configure	2012-05-17 05:30:56.000000000 +0200
+++ gdal-1.9.1-fedora-dods3.11.3/configure	2012-05-20 10:47:10.032831845 +0200
@@ -24320,8 +24320,8 @@
     as_fn_error $? "--with-dods-root requires path to dods root as argument" "$LINENO" 5
 else
     DODS_LIB=$with_dods_root/lib
-    DODS_INC="-I$with_dods_root/include -I$with_dods_root/include/libdap -I$with_dods_root/include/dap"
     DODS_BIN=$with_dods_root/bin
+    DODS_INC="-I$with_dods_root/include -I$with_dods_root/include/libdap -I$with_dods_root/include/dap `$DODS_BIN/xml2-config --cflags`"
 
 rm -f islibdappost310.*
 echo '#include "Connect.h"' > islibdappost310.cpp
Binary files gdal-1.9.1-fedora/.configure.swp and gdal-1.9.1-fedora-dods3.11.3/.configure.swp differ
