--- ogr/ogrsf_frmts/mysql/ogr_mysql.h.orig	2008-10-23 03:04:33.000000000 +0300
+++ ogr/ogrsf_frmts/mysql/ogr_mysql.h	2009-02-02 22:47:05.000000000 +0200
@@ -31,6 +31,8 @@
 #ifndef _OGR_MYSQL_H_INCLUDED
 #define _OGR_MYSQL_H_INCLUDED
 
+#include "ogrsf_frmts.h"
+
 #include <my_global.h>
 #include <mysql.h>
 
@@ -38,8 +40,6 @@
 #undef bool
 #endif
 
-#include "ogrsf_frmts.h"
-
 /************************************************************************/
 /*                            OGRMySQLLayer                             */
 /************************************************************************/
