diff -Nuar a/swig/python/GNUmakefile b/swig/python/GNUmakefile
--- a/swig/python/GNUmakefile	2013-04-24 21:55:26.000000000 +0300
+++ b/swig/python/GNUmakefile	2013-08-02 01:12:51.334398963 +0300
@@ -67,10 +67,10 @@
 	
 install:
 
-ifeq ($(PY_HAVE_SETUPTOOLS),1)
-	$(PYTHON) setup.py install 
+ifeq ($(PY_HAVE_SETUPTOOLS),0)
+	$(PYTHON) setup.py install --root=$(DESTDIR) 
 else
-	$(PYTHON) setup.py install --prefix=$(DESTDIR)$(prefix)
+	$(PYTHON) setup.py install --root=$(DESTDIR) --prefix=$(prefix)
 endif
 
 	for f in $(SCRIPTS) ; do $(INSTALL) ./scripts/$$f $(DESTDIR)$(INST_BIN) ; done
