diff -Nuar a/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp b/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp
--- a/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp	2013-04-24 21:55:19.000000000 +0300
+++ b/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp	2013-08-02 04:07:33.063036051 +0300
@@ -138,24 +138,6 @@
     int  nArgc = 1;
     const char *client_groups[] = {"client", "ogr", NULL };
 
-    my_init(); // I hope there is no problem with calling this multiple times!
-    load_defaults( "my", client_groups, &nArgc, &papszArgv );
-
-    for( i = 0; i < nArgc; i++ )
-    {
-        if( EQUALN(papszArgv[i],"--user=",7) )
-            oUser = papszArgv[i] + 7;
-        else if( EQUALN(papszArgv[i],"--host=",7) )
-            oHost = papszArgv[i] + 7;
-        else if( EQUALN(papszArgv[i],"--password=",11) )
-            oPassword = papszArgv[i] + 11;
-        else if( EQUALN(papszArgv[i],"--port=",7) )
-            nPort = atoi(papszArgv[i] + 7);
-    }
-
-    // cleanup
-    free_defaults( papszArgv );
-
 /* -------------------------------------------------------------------- */
 /*      Parse out connection information.                               */
 /* -------------------------------------------------------------------- */
