--- ./libparted/fs/reiserfs/reiserfs.c
+++ ./libparted/fs/reiserfs/reiserfs.c
@@ -85,7 +85,7 @@
 FCLASS int (FPTR libreiserfs_exception_option) (reiserfs_exception_t *);
 FCLASS char *(FPTR libreiserfs_exception_message) (reiserfs_exception_t *);
 FCLASS void (FPTR libreiserfs_exception_set_handler)
-		(int(FPTR)(reiserfs_exception_t *));
+		(int(*)(reiserfs_exception_t *));
 
 FCLASS void (FPTR dal_realize) (dal_t *);
 FCLASS size_t (FPTR dal_block_size) (dal_t *);
