Index: parted-1.8.8/libparted/arch/linux.c
===================================================================
--- parted-1.8.8.orig/libparted/arch/linux.c	2009-03-09 16:16:58.000000000 +0100
+++ parted-1.8.8/libparted/arch/linux.c	2009-03-09 16:24:20.000000000 +0100
@@ -1983,7 +1983,7 @@
                         || dev->type == PED_DEVICE_DM
 #endif
                         || isdigit (dev->path[path_len - 1]))
-                snprintf (result, result_len, "%sp%d", dev->path, num);
+                snprintf (result, result_len, "%s_part%d", dev->path, num);
         else
                 snprintf (result, result_len, "%s%d", dev->path, num);
 
