Index: parted-1.8.8/libparted/arch/linux.c
===================================================================
--- parted-1.8.8.orig/libparted/arch/linux.c
+++ parted-1.8.8/libparted/arch/linux.c
@@ -2120,7 +2120,7 @@ _blkpg_add_partition (PedDisk* disk, Ped
         linux_part.start = part->geom.start * disk->dev->sector_size;
         /* see fs/partitions/msdos.c:msdos_partition(): "leave room for LILO" */
         if (part->type & PED_PARTITION_EXTENDED)
-                linux_part.length = part->geom.length == 1 ? 512 : 1024;
+                linux_part.length = PED_SECTOR_SIZE_DEFAULT;
         else
                 linux_part.length = part->geom.length * disk->dev->sector_size;
         linux_part.pno = part->num;
