diff -up parted-1.8.8/include/parted/Makefile.am.noinst parted-1.8.8/include/parted/Makefile.am
--- parted-1.8.8/include/parted/Makefile.am.noinst	2007-08-09 05:20:33.000000000 -1000
+++ parted-1.8.8/include/parted/Makefile.am	2007-12-18 12:30:34.000000000 -1000
@@ -6,9 +6,7 @@ endif
 
 partedincludedir      =	$(includedir)/parted
 
-partedinclude_HEADERS = gnu.h		\
-			linux.h		\
-			constraint.h	\
+partedinclude_HEADERS = constraint.h	\
 			debug.h		\
 			device.h	\
 			disk.h		\
@@ -18,10 +16,12 @@ partedinclude_HEADERS = gnu.h		\
 			natmath.h	\
 			timer.h		\
 			unit.h		\
-			parted.h        \
-			$(S390_HDRS)
+			parted.h
 
 noinst_HEADERS	      = crc32.h		\
-			endian.h
+			endian.h		\
+			gnu.h		\
+			linux.h		\
+			$(S390_HDRS)
 
 MAINTAINERCLEANFILES = Makefile.in
