diff -up parted-1.8.8/parted/parted.c.xvd parted-1.8.8/parted/parted.c
--- parted-1.8.8/parted/parted.c.xvd	2007-08-09 05:20:33.000000000 -1000
+++ parted-1.8.8/parted/parted.c	2007-12-18 12:25:16.000000000 -1000
@@ -1264,9 +1264,10 @@ do_print (PedDevice** dev)
         int             has_free_arg = 0;
         int             has_list_arg = 0;
         int             has_num_arg = 0;
-        char*           transport[13] = {"unknown", "scsi", "ide", "dac960",
+        char*           transport[14] = {"unknown", "scsi", "ide", "dac960",
                                          "cpqarray", "file", "ataraid", "i2o",
-                                         "ubd", "dasd", "viodasd", "sx8", "dm"};
+                                         "ubd", "dasd", "viodasd", "sx8", "dm",
+                                         "xvd"};
         char*           peek_word;
         char*           start;
         char*           end;
