diff -up parted-1.8.8/include/parted/device.h.devmapper parted-1.8.8/include/parted/device.h
--- parted-1.8.8/include/parted/device.h.devmapper	2007-07-31 09:03:53.000000000 -1000
+++ parted-1.8.8/include/parted/device.h	2007-12-18 12:27:10.000000000 -1000
@@ -45,9 +45,7 @@ typedef enum {
         PED_DEVICE_DASD         = 9,
         PED_DEVICE_VIODASD      = 10,
         PED_DEVICE_SX8          = 11,
-#ifdef ENABLE_DEVICE_MAPPER
         PED_DEVICE_DM           = 12,
-#endif
         PED_DEVICE_XVD          = 13
 } PedDeviceType;
 
