diff -up parted-1.8.8/libparted/labels/dos.c.swap parted-1.8.8/libparted/labels/dos.c
--- parted-1.8.8/libparted/labels/dos.c.swap	2009-03-23 17:21:21.000000000 -1000
+++ parted-1.8.8/libparted/labels/dos.c	2009-03-23 17:24:57.000000000 -1000
@@ -1411,6 +1411,12 @@ msdos_partition_set_flag (PedPartition* 
 		dos_data->prep = state;
 		return ped_partition_set_system (part, part->fs_type);
 
+	case PED_PARTITION_SWAP:
+		if (state) {
+			return ped_partition_set_system (part, ped_file_system_type_get("linux-swap"));
+		}
+		//break;
+
 	default:
 		return 0;
 	}
diff -up parted-1.8.8/libparted/labels/gpt.c.swap parted-1.8.8/libparted/labels/gpt.c
diff -up parted-1.8.8/libparted/labels/mac.c.swap parted-1.8.8/libparted/labels/mac.c
