diff -Nuar a/src/client/cl_language.c b/src/client/cl_language.c
--- a/src/client/cl_language.c	2012-04-24 08:04:43.000000000 +0300
+++ b/src/client/cl_language.c	2013-10-01 08:38:36.499444670 +0300
@@ -175,7 +175,7 @@
 #ifdef LOCALEDIR
 		Com_sprintf(languagePath, sizeof(languagePath), LOCALEDIR);
 #else
-		Com_sprintf(languagePath, sizeof(languagePath), "%s/"BASEDIRNAME"/i18n/", FS_GetCwd());
+		Com_sprintf(languagePath, sizeof(languagePath), "%s/"BASEDIRNAME"/i18n/", "/usr/share/ufoai");
 #endif
 	Com_DPrintf(DEBUG_CLIENT, "CL_LanguageTest: using mo files from '%s'\n", languagePath);
 	Q_strcat(languagePath, localeID, sizeof(languagePath));
diff -Nuar a/src/client/cl_main.c b/src/client/cl_main.c
--- a/src/client/cl_main.c	2012-04-24 08:04:43.000000000 +0300
+++ b/src/client/cl_main.c	2013-10-01 08:39:30.731442793 +0300
@@ -1180,7 +1180,7 @@
 	if (fs_i18ndir->string[0] != '\0')
 		Q_strncpyz(languagePath, fs_i18ndir->string, sizeof(languagePath));
 	else
-		Com_sprintf(languagePath, sizeof(languagePath), "%s/"BASEDIRNAME"/i18n/", FS_GetCwd());
+		Com_sprintf(languagePath, sizeof(languagePath), "%s/"BASEDIRNAME"/i18n/", "/usr/share/ufoai");
 	Com_DPrintf(DEBUG_CLIENT, "...using mo files from %s\n", languagePath);
 	bindtextdomain(TEXT_DOMAIN, languagePath);
 	bind_textdomain_codeset(TEXT_DOMAIN, "UTF-8");
