diff -Nuar a/src/client/cl_main.c b/src/client/cl_main.c
--- a/src/client/cl_main.c	2012-04-24 08:04:43.000000000 +0300
+++ b/src/client/cl_main.c	2013-10-01 09:31:33.421334675 +0300
@@ -1166,6 +1166,7 @@
 	/* i18n through gettext */
 	char languagePath[MAX_OSPATH];
 	cvar_t *fs_i18ndir;
+	char *localetmp;
 
 	if (sv_dedicated->integer)
 		return;					/* nothing running on the client */
diff -Nuar a/src/ports/unix/unix_shared.c b/src/ports/unix/unix_shared.c
--- a/src/ports/unix/unix_shared.c	2012-04-24 08:04:43.000000000 +0300
+++ b/src/ports/unix/unix_shared.c	2013-10-01 09:34:54.547327711 +0300
@@ -126,7 +126,7 @@
 {
 	/* Calling with NULL param should return current system settings. */
 	const char *currentLocale = setlocale(LC_MESSAGES, NULL);
-	if (currentLocale != NULL && currentLocale[0] != '\0')
+	if (currentLocale != NULL && currentLocale[0] != '\0' && strncmp(currentLocale, "tr_TR.UTF-8", strlen("tr_TR.UTF-8") != 0))
 		return currentLocale;
 	else
 		return "C";
