diff -Nuar telepathy-python-0.15.15.orig/Makefile.in telepathy-python-0.15.15/Makefile.in
--- telepathy-python-0.15.15.orig/Makefile.in	2010-01-20 15:47:58.000000000 +0200
+++ telepathy-python-0.15.15/Makefile.in	2010-02-16 14:34:28.333117263 +0200
@@ -34,7 +34,7 @@
 subdir = .
 DIST_COMMON = README $(am__configure_deps) $(srcdir)/Makefile.am \
 	$(srcdir)/Makefile.in $(top_srcdir)/configure AUTHORS COPYING \
-	ChangeLog INSTALL NEWS install-sh missing py-compile
+	ChangeLog INSTALL NEWS install-sh missing
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/configure.ac
 am__configure_deps = $(am__aclocal_m4_deps) $(CONFIGURE_DEPENDENCIES) \
diff -Nuar telepathy-python-0.15.15.orig/src/client/Makefile.in telepathy-python-0.15.15/src/client/Makefile.in
--- telepathy-python-0.15.15.orig/src/client/Makefile.in	2010-01-20 15:47:58.000000000 +0200
+++ telepathy-python-0.15.15/src/client/Makefile.in	2010-02-16 14:38:54.061356515 +0200
@@ -71,7 +71,6 @@
   sed '$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;s/\n/ /g' | \
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(clientdir)"
-py_compile = $(top_srcdir)/py-compile
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 ACLOCAL = @ACLOCAL@
 AMTAR = @AMTAR@
@@ -218,14 +217,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(clientdir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(clientdir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(clientdir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(clientdir)" $$dlist; \
-	  fi; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-clientPYTHON:
 	@$(NORMAL_UNINSTALL)
diff -Nuar telepathy-python-0.15.15.orig/src/Makefile.in telepathy-python-0.15.15/src/Makefile.in
--- telepathy-python-0.15.15.orig/src/Makefile.in	2010-01-20 15:47:58.000000000 +0200
+++ telepathy-python-0.15.15/src/Makefile.in	2010-02-16 14:38:41.068107412 +0200
@@ -79,7 +79,6 @@
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(generateddir)" \
 	"$(DESTDIR)$(versiondir)" "$(DESTDIR)$(telepathydir)"
-py_compile = $(top_srcdir)/py-compile
 RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
   distclean-recursive maintainer-clean-recursive
 AM_RECURSIVE_TARGETS = $(RECURSIVE_TARGETS:-recursive=) \
@@ -288,14 +287,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(generateddir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(generateddir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(generateddir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(generateddir)" $$dlist; \
-	  fi; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-nodist_generatedPYTHON:
 	@$(NORMAL_UNINSTALL)
@@ -326,14 +318,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(versiondir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(versiondir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(versiondir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(versiondir)" $$dlist; \
-	  fi; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-nodist_versionPYTHON:
 	@$(NORMAL_UNINSTALL)
@@ -364,14 +349,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(telepathydir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(telepathydir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(telepathydir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(telepathydir)" $$dlist; \
-	  fi; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-telepathyPYTHON:
 	@$(NORMAL_UNINSTALL)
diff -Nuar telepathy-python-0.15.15.orig/src/server/Makefile.in telepathy-python-0.15.15/src/server/Makefile.in
--- telepathy-python-0.15.15.orig/src/server/Makefile.in	2010-01-20 15:47:58.000000000 +0200
+++ telepathy-python-0.15.15/src/server/Makefile.in	2010-02-16 14:39:09.238088477 +0200
@@ -71,7 +71,6 @@
   sed '$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;s/\n/ /g' | \
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(serverdir)"
-py_compile = $(top_srcdir)/py-compile
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 ACLOCAL = @ACLOCAL@
 AMTAR = @AMTAR@
@@ -222,14 +221,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(serverdir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(serverdir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(serverdir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(serverdir)" $$dlist; \
-	  fi; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-serverPYTHON:
 	@$(NORMAL_UNINSTALL)
