diff -Nuar alarm.orig//module.desktop.in alarm//module.desktop.in
--- alarm.orig//module.desktop.in	2011-02-16 22:39:52.000000000 +0200
+++ alarm//module.desktop.in	2011-03-31 14:48:11.676012993 +0300
@@ -28,3 +28,4 @@
 Comment[el]=Ένα άρθρωμα για δημιουργείτε υπενθυμίσεις.
 Comment[pt]=Um módulo para a definição de alertas
 Comment[de]=Modul zum Anzeigen von Erinnerungen als Popups
+Comment[tr]=Açılır pencere hatırlatıcıları ayarlamaya olanak sağlayan bir modül.
