diff -Nur libnids-1.23-old/src/Makefile.in libnids-1.23/src/Makefile.in
--- libnids-1.23-old/src/Makefile.in	2009-08-12 16:18:35.155793051 +0300
+++ libnids-1.23/src/Makefile.in	2009-08-12 16:19:23.200791156 +0300
@@ -65,7 +65,7 @@
 	ar -cr $@ $(OBJS)
 	$(RANLIB) $@
 $(LIBSHARED): $(OBJS_SHARED)
-	$(CC) -shared -Wl,-soname,$(LIBSHARED) -o $(LIBSHARED) $(OBJS_SHARED) $(LIBS) $(LNETLIB) $(PCAPLIB)
+	$(CC) -shared -Wl,-soname,$(LIBSHARED) $(LDFLAGS) -o $(LIBSHARED) $(OBJS_SHARED) $(LIBS) $(LNETLIB) $(PCAPLIB)
 
 _install install: $(LIBSTATIC)
 	../mkinstalldirs $(install_prefix)$(libdir)
