--- bin/mudur.py~	2014-06-14 14:28:45.136961885 +0200
+++ bin/mudur.py	2014-06-14 14:29:12.000000000 +0200
@@ -1406,7 +1406,7 @@
         UI.info(_("Creating tmpfiles"))
         if not os.path.isdir("/run/tmpfiles.d"): create_directory("/run/tmpfiles.d")
         run("/usr/bin/kmod", "static-nodes", "--format=tmpfiles", "--output=/run/tmpfiles.d/kmod.conf")
-        out = capture("/sbin/mudur_tmpfiles.py")[0].strip().split("\n")
+        out = [line for line in capture("/sbin/mudur_tmpfiles.py")[0].split("\n") if line.strip()]
         if out: LOGGER.log("Errors during tmpfiles creation.\n\t%s" % "\n\t".join(out))
         run("mount", "-t", "tmpfs", "tmpfs", "/dev/shm")
 
