--- bin/mudur_tmpfiles.py~	2014-05-16 03:56:41.000000000 +0200
+++ bin/mudur_tmpfiles.py	2014-06-11 21:54:58.333421994 +0200
@@ -37,6 +37,10 @@
     open(path, mode).write(content)
 
 def create(type, path, mode, uid, gid, age, arg):
+    if type == "d" and \
+        os.path.isdir(path) and \
+        (not uid == os.stat(path).st_uid or not gid == os.stat(path).st_gid):
+        type = "D"
     if type == "L":
         if not os.path.islink(path): os.symlink(arg, path)
         return
