--- kdecore/localization/kuitformats.cpp.orig	2009-02-11 12:44:43.000000000 +0200
+++ kdecore/localization/kuitformats.cpp	2009-02-11 12:45:18.000000000 +0200
@@ -59,7 +59,7 @@
     easternArabicDecSep = QString::fromUtf8(",")[0];
 }
 
-K_GLOBAL_STATIC(KuitFormatsStaticData, staticData)
+K_GLOBAL_STATIC(KuitFormatsStaticData, staticFormatData)
 
 static QString insertIntegerSeparators (const QString &istr,
                                         const QChar &sep, int ngrp)
@@ -140,7 +140,7 @@
 // by Youssef Chahibi <chahibi@gmail.com>
 QString KuitFormats::toNumberEArab (const QString &numstr)
 {
-    KuitFormatsStaticData *s = staticData;
+    KuitFormatsStaticData *s = staticFormatData;
 
     const int power = 3;
 
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kdelibs-4.2.0-orşg/kate/completion/katecompletionwidget.cpp kdelibs-4.2.0/kate/completion/katecompletionwidget.cpp
--- kate/completion/katecompletionwidget.cpp	2009-02-11 14:02:07.000000000 +0200
+++ kate/completion/katecompletionwidget.cpp	2009-02-11 14:03:56.000000000 +0200
@@ -54,7 +54,10 @@
 
 //#include "modeltest.h"
 
+#ifndef _ENABLE_FINAL_
+#define _ENABLE_FINAL_
 Q_DECLARE_METATYPE(KTextEditor::Cursor)
+#endif
 
 KTextEditor::CodeCompletionModelControllerInterface* modelController(KTextEditor::CodeCompletionModel *model)
 {
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kdelibs-4.2.0-orşg/kate/script/katescript.cpp kdelibs-4.2.0/kate/script/katescript.cpp
--- kate/script/katescript.cpp	2009-02-11 14:02:07.000000000 +0200
+++ kate/script/katescript.cpp	2009-02-11 14:04:45.000000000 +0200
@@ -36,7 +36,10 @@
 /**
  * metatype register
  */
+#ifndef _ENABLE_FINAL_
+#define _ENABLE_FINAL_
 Q_DECLARE_METATYPE(KTextEditor::Cursor)
+#endif
 
 /**
  * conversion functions
