Index: kdelibs-4.2.1/kdecore/services/kmimetypefactory.cpp
===================================================================
--- kdelibs-4.2.1.orig/kdecore/services/kmimetypefactory.cpp
+++ kdelibs-4.2.1/kdecore/services/kmimetypefactory.cpp
@@ -25,6 +25,8 @@
 #include <kshell.h>
 #include <kdebug.h>
 
+#include <magic.h>
+
 ////
 
 KMimeTypeFactory::KMimeTypeFactory()
@@ -363,7 +365,50 @@ KMimeType::Ptr KMimeTypeFactory::findFro
 
     // Do fallback code so that we never return 0 - unless we were only looking for HighPriorityRules
     if (whichPriority != HighPriorityRules) {
+
+        //Nothing worked, try libmagic to find out mimetype
+        magic_t handle = magic_open(MAGIC_MIME_TYPE);
+        if (handle == NULL)
+        {
+            kDebug(7009) << "Can not initialize libmagic";
+
+            if (accuracy)
+                *accuracy = 0;
+            return KMimeType::defaultMimeTypePtr();
+        }
+
+        magic_load(handle, NULL);
+
+        if (device && !device->isOpen() ) {
+             if ( !device->open(QIODevice::ReadOnly) ) {
+                if (accuracy)
+                    *accuracy = 0;
+                return KMimeType::defaultMimeTypePtr();
+              }
+         }
+
+        QByteArray buf = device->read(1024);
+        const char *result = magic_buffer(handle, (const void *)(buf.constData()), buf.size());
+
+        if (result == NULL)
+        {
+            const char *err = magic_error(handle);
+            if (err != NULL)
+                kDebug(7009) << "error: " << magic_error(handle);
+
+            if (accuracy)
+                *accuracy = 0;
+
+            magic_close(handle);
+            return KMimeType::defaultMimeTypePtr();
+        }
+
+        magic_close(handle);
+        *accuracy = 90;
+        return findMimeTypeByName(QString(result));
+
         // Nothing worked, check if the file contents looks like binary or text
+        /*
         if (!KMimeType::isBufferBinaryData(beginning)) {
             if (accuracy)
                 *accuracy = 5;
@@ -372,8 +417,10 @@ KMimeType::Ptr KMimeTypeFactory::findFro
         if (accuracy)
             *accuracy = 0;
         return KMimeType::defaultMimeTypePtr();
+        */
     }
 
+    *accuracy = 0;
     return KMimeType::Ptr();
 }
 
Index: kdelibs-4.2.1/kdecore/CMakeLists.txt
===================================================================
--- kdelibs-4.2.1.orig/kdecore/CMakeLists.txt
+++ kdelibs-4.2.1/kdecore/CMakeLists.txt
@@ -275,7 +275,7 @@ endif (NOT Q_WS_X11 AND NOT Q_WS_QWS)
 
 kde4_add_library(kdecore SHARED ${kdecore_LIB_SRCS})
 
-target_link_libraries(kdecore ${QT_QTCORE_LIBRARY} ${QT_QTNETWORK_LIBRARY} ${QT_QTDBUS_LIBRARY} ${QT_QTXML_LIBRARY} ${ZLIB_LIBRARY} ${kdecore_OPTIONAL_LIBS})
+target_link_libraries(kdecore ${QT_QTCORE_LIBRARY} ${QT_QTNETWORK_LIBRARY} ${QT_QTDBUS_LIBRARY} ${QT_QTXML_LIBRARY} ${ZLIB_LIBRARY} ${kdecore_OPTIONAL_LIBS} magic)
 
 if(WIN32)
   set(kdecore_LINK_INTERFACE_LIBRARIES  ${QT_QTDBUS_LIBRARY} ${QT_QTCORE_LIBRARY} ${KDEWIN32_LIBRARIES})
