diff -Nuar e-module-screenshot-0.0.1_58042.orig//module.desktop.in e-module-screenshot-0.0.1_58042/module.desktop.in
--- e-module-screenshot-0.0.1_58042.orig//module.desktop.in	2011-03-14 20:17:38.000000000 +0200
+++ e-module-screenshot-0.0.1_58042/module.desktop.in	2011-03-31 16:37:42.702043554 +0300
@@ -8,6 +8,7 @@
 Name[fr]=Capture
 Name[el]=Στιγμιότυπο Οθόνης
 Name[pt]=Captura de ecrã
+Name[tr]=Ekran görüntüsü
 Icon=e-module-screenshot
 X-Enlightenment-ModuleType=utils
 Comment=Gadget to take screenshots using emprint.
@@ -19,3 +20,4 @@
 Comment[fr]=Un gadget qui permet de faire des captures d'écran en utilisant Emprint.
 Comment[el]=Συστατικό για τη λήψη στιγμιοτύπων της οθόνης με το emprint.
 Comment[pt]=Um "gadget" para obter capturas do ecrã utilizando o emprint
+Comment[tr]=emprint kullanarak ekran görüntüsü alan modüldür.
