Index: b/src/event_race_select.cpp
===================================================================
--- a/src/event_race_select.cpp
+++ b/src/event_race_select.cpp
@@ -479,13 +479,13 @@
 			UIMgr.setDirty();
 			return true;
 		case 'c': 
-	    	mp_conditionsSSBtn->simulateMouseClick();
+	 //   	mp_conditionsSSBtn->simulateMouseClick();
 	    	return true;
 		case 'w': 
-	    	mp_windSSBtn->simulateMouseClick();
+	  //  	mp_windSSBtn->simulateMouseClick();
 	    	return true;
 		case 'm':
-	    	mp_mirrorSSBtn->simulateMouseClick();
+	  //  	mp_mirrorSSBtn->simulateMouseClick();
 	    	return true;
 		default:
 			return false;
