diff -Nur extreme-tuxracer-0.5beta-old/src/Makefile.in extreme-tuxracer-0.5beta/src/Makefile.in
--- extreme-tuxracer-0.5beta-old/src/Makefile.in	2009-11-23 14:25:29.238069525 +0200
+++ extreme-tuxracer-0.5beta/src/Makefile.in	2009-11-23 14:25:52.252193862 +0200
@@ -223,7 +223,7 @@
 install_sh = @install_sh@
 libdir = @libdir@
 libexecdir = @libexecdir@
-localedir = $(datadir)/locale
+localedir =@localedir@
 localstatedir = @localstatedir@
 mandir = @mandir@
 mkdir_p = @mkdir_p@
