diff -up kdepim-runtime-4.4.93/kcm/kcm_akonadi.desktop.show-akonadi-kcm kdepim-runtime-4.4.93/kcm/kcm_akonadi.desktop
--- kdepim-runtime-4.4.93/kcm/kcm_akonadi.desktop.show-akonadi-kcm  2010-08-31 08:04:08.000000000 -0500
+++ kdepim-runtime-4.4.93/kcm/kcm_akonadi.desktop   2010-09-08 08:52:40.689479608 -0500
@@ -8,9 +8,10 @@ X-KDE-ModuleType=Library
 X-KDE-Library=kcm_akonadi
 X-KDE-FactoryName=kcm_akonadi
 X-KDE-HasReadOnlyMode=false
-X-KDE-ParentApp=
+X-KDE-ParentApp=kcontrol
 X-KDE-ParentComponents=
 X-KDE-CfgDlgHierarchy=
+X-KDE-System-Settings-Parent-Category=workspace-appearance-and-behavior
 
 Name=Akonadi Configuration
 Name[ar]=ضبط اكوندا 
