diff -up mhgui-0.2/include/mhgui/ImageData.h.png15 mhgui-0.2/include/mhgui/ImageData.h
--- mhgui-0.2/include/mhgui/ImageData.h.png15	2007-12-12 22:07:20.000000000 +0100
+++ mhgui-0.2/include/mhgui/ImageData.h	2012-01-21 15:10:13.179933285 +0100
@@ -67,7 +67,7 @@ class ImageData
 private:
   char         *image_data; /* raw png image data */
   char         *end_bufferPtr;
-  unsigned long width, height;
+  png_uint_32   width, height;
   unsigned int  bytesPerRow;
   bool          alpha;
 
