Index: irrlicht-1.8/source/Irrlicht/COpenGLDriver.cpp
===================================================================
--- irrlicht-1.8.orig/source/Irrlicht/COpenGLDriver.cpp
+++ irrlicht-1.8/source/Irrlicht/COpenGLDriver.cpp
@@ -4148,7 +4148,8 @@ bool COpenGLDriver::setRenderTarget(vide
 		if (texture)
 		{
 			// we want to set a new target. so do this.
-			glViewport(0, 0, texture->getSize().Width, texture->getSize().Height);
+			//glViewport(0, 0, texture->getSize().Width, texture->getSize().Height);
+		        setViewPort(core::rect<s32>(0, 0, texture->getSize().Width, texture->getSize().Height));
 			RenderTargetTexture = static_cast<COpenGLTexture*>(texture);
 			// calls glDrawBuffer as well
 			RenderTargetTexture->bindRTT();
@@ -4157,7 +4158,8 @@ bool COpenGLDriver::setRenderTarget(vide
 		}
 		else
 		{
-			glViewport(0,0,ScreenSize.Width,ScreenSize.Height);
+		  // glViewport(0,0,ScreenSize.Width,ScreenSize.Height);
+                        setViewPort(core::rect<s32>(0,0,ScreenSize.Width,ScreenSize.Height));
 			RenderTargetTexture = 0;
 			CurrentRendertargetSize = core::dimension2d<u32>(0,0);
 			CurrentTarget=ERT_FRAME_BUFFER;
