diff -Nuar pyparted-2.1.1.orig/src/parted/Makefile.in pyparted-2.1.1/src/parted/Makefile.in
--- pyparted-2.1.1.orig/src/parted/Makefile.in	2009-08-05 01:37:23.000000000 +0300
+++ pyparted-2.1.1/src/parted/Makefile.in	2009-09-03 12:30:40.007434912 +0300
@@ -283,13 +283,13 @@
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(parteddir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(parteddir)" || exit $$?; \
 	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(parteddir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(parteddir)" $$dlist; \
-	  fi; \
-	else :; fi
+#	if test -n "$$dlist"; then \
+#	  if test -z "$(DESTDIR)"; then \
+#	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(parteddir)" $$dlist; \
+#	  else \
+#	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(parteddir)" $$dlist; \
+#	  fi; \
+#	else :; fi
 
 uninstall-partedPYTHON:
 	@$(NORMAL_UNINSTALL)
