diff -ur pyparted-1.8.9.orig/Makefile pyparted-1.8.9/Makefile
--- pyparted-1.8.9.orig/Makefile	2007-08-10 23:49:53.000000000 +0300
+++ pyparted-1.8.9/Makefile	2007-12-14 08:50:47.000000000 +0200
@@ -36,7 +36,7 @@
 # Build with libparted (parted-1.8.3 and higher have a pkg-config file)
 LDFLAGS += $(shell pkg-config --libs libparted)
 
-libdir ?= $(DESTDIR)$(shell rpm --eval "%{_libdir}")/python$(PYVER)/site-packages
+libdir ?= $(DESTDIR)/usr/lib/python$(PYVER)/site-packages
 
 all: partedmodule.so
 
