--- configure.ac.orig	2014-01-27 23:45:42.000000000 +0200
+++ configure.ac	2014-05-25 19:23:54.610573002 +0300
@@ -268,15 +268,6 @@
   AC_MSG_ERROR("You need the tool `rapper' from the `raptor-utils' package in order to compile Zeitgeist")
 fi
 
-# check for python-rdflib
-AC_MSG_CHECKING([for python-rdflib])
-echo "import rdflib" | python - 2>/dev/null
-if test $? -ne 0 ; then
-  AC_MSG_FAILURE([failed.  Please install the python-rdflib package.])
-else
-  AC_MSG_RESULT([yes])
-fi
-
 #################################################
 # Summary
 #################################################
