diff -up openmpi-1.7/configure.removed openmpi-1.7/configure
--- openmpi-1.7/configure.removed	2013-03-28 14:38:48.000000000 -0600
+++ openmpi-1.7/configure	2013-04-16 21:43:00.165986877 -0600
@@ -72177,9 +72177,6 @@ $as_echo "$compile_mode" >&6; }
     # try to configure the component.  pay no attention to
     # --enable-dist, since we'll always have makefiles.
 
-    ac_config_files="$ac_config_files opal/mca/backtrace/darwin/Makefile"
-
-
     for ac_func in vm_read_overwrite
 do :
   ac_fn_c_check_func "$LINENO" "vm_read_overwrite" "ac_cv_func_vm_read_overwrite"
@@ -194246,7 +194243,6 @@ do
     "opal/mca/backtrace/Makefile") CONFIG_FILES="$CONFIG_FILES opal/mca/backtrace/Makefile" ;;
     "opal/mca/backtrace/printstack/Makefile") CONFIG_FILES="$CONFIG_FILES opal/mca/backtrace/printstack/Makefile" ;;
     "opal/mca/backtrace/execinfo/Makefile") CONFIG_FILES="$CONFIG_FILES opal/mca/backtrace/execinfo/Makefile" ;;
-    "opal/mca/backtrace/darwin/Makefile") CONFIG_FILES="$CONFIG_FILES opal/mca/backtrace/darwin/Makefile" ;;
     "opal/mca/backtrace/none/Makefile") CONFIG_FILES="$CONFIG_FILES opal/mca/backtrace/none/Makefile" ;;
     "opal/mca/compress/Makefile") CONFIG_FILES="$CONFIG_FILES opal/mca/compress/Makefile" ;;
     "opal/mca/compress/gzip/Makefile") CONFIG_FILES="$CONFIG_FILES opal/mca/compress/gzip/Makefile" ;;
diff -up openmpi-1.7/Makefile.in.removed openmpi-1.7/Makefile.in
--- openmpi-1.7/Makefile.in.removed	2013-03-28 14:34:54.000000000 -0600
+++ openmpi-1.7/Makefile.in	2013-04-16 21:42:00.734597103 -0600
@@ -226,7 +226,6 @@ am__aclocal_m4_deps = $(top_srcdir)/ompi
 	$(top_srcdir)/opal/mca/memory/configure.m4 \
 	$(top_srcdir)/opal/mca/pstat/configure.m4 \
 	$(top_srcdir)/opal/mca/timer/configure.m4 \
-	$(top_srcdir)/opal/mca/backtrace/darwin/configure.m4 \
 	$(top_srcdir)/opal/mca/backtrace/none/configure.m4 \
 	$(top_srcdir)/opal/mca/backtrace/printstack/configure.m4 \
 	$(top_srcdir)/opal/mca/backtrace/execinfo/configure.m4 \
