Index: VirtualBox-3.2.8_OSE/LocalConfig.kmk
===================================================================
--- /dev/null
+++ VirtualBox-3.2.8_OSE/LocalConfig.kmk
@@ -0,0 +0,15 @@
+# Disable Registration and update
+VBOX_WITH_REGISTRATION_REQUEST =
+VBOX_WITH_UPDATE_REQUEST =
+
+# Do not build tests
+VBOX_WITH_TESTCASES         :=
+VBOX_WITH_TESTSUITE         :=
+VBOX_WITH_WARNINGS_AS_ERRORS:=
+
+VBOX_WITH_ORIGIN            :=
+VBOX_WITH_RUNPATH           := __VBOXLIBDIR__
+VBOX_WITH_SHARED_LIBS       := __VBOXLIBDIR__
+VBOX_PATH_APP_PRIVATE_ARCH  := __VBOXLIBDIR__
+VBOX_PATH_APP_PRIVATE       := __VBOXDATADIR__
+VBOX_PATH_APP_DOCS          := __VBOXDATADIR__
