Index: VirtualBox-4.1.2_OSE/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp
===================================================================
--- VirtualBox-4.1.2_OSE.orig/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp
+++ VirtualBox-4.1.2_OSE/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp
@@ -4623,6 +4623,9 @@ void VBoxGlobal::showRegistrationDialog
  */
 void VBoxGlobal::showUpdateDialog (bool aForce)
 {
+    // Updates are handled trough our package manager
+    return ;
+
     /* Silently check in one day after current time-stamp */
     QTimer::singleShot (24 /* hours */   * 60   /* minutes */ *
                         60 /* seconds */ * 1000 /* milliseconds */,
Index: VirtualBox-4.1.2_OSE/src/VBox/Frontends/VirtualBox/src/runtime/UIMachineMenuBar.cpp
===================================================================
--- VirtualBox-4.1.2_OSE.orig/src/VBox/Frontends/VirtualBox/src/runtime/UIMachineMenuBar.cpp
+++ VirtualBox-4.1.2_OSE/src/VBox/Frontends/VirtualBox/src/runtime/UIMachineMenuBar.cpp
@@ -275,13 +275,7 @@ void UIMachineMenuBar::prepareMenuHelp(Q
 #if defined(Q_WS_MAC) && (QT_VERSION < 0x040700)
     if (m_fIsFirstTime)
 # endif
-        pMenu->addAction(gActionPool->action(UIActionIndex_Simple_Update));
-#ifndef Q_WS_MAC
-    pMenu->addSeparator();
-#endif /* !Q_WS_MAC */
-#if defined(Q_WS_MAC) && (QT_VERSION < 0x040700)
-    if (m_fIsFirstTime)
-# endif
+
         pMenu->addAction(gActionPool->action(UIActionIndex_Simple_About));
 
 
Index: VirtualBox-4.1.2_OSE/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp
===================================================================
--- VirtualBox-4.1.2_OSE.orig/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp
+++ VirtualBox-4.1.2_OSE/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp
@@ -540,6 +540,9 @@ bool UISettingsDialogGlobal::isPageAvail
 #endif /* !VBOX_WITH_NETFLT */
             break;
         }
+        case GLSettingsPage_Update:
+            return false;
+
         default:
             break;
     }
Index: VirtualBox-4.1.2_OSE/src/VBox/Frontends/VirtualBox/src/VBoxUpdateDlg.cpp
===================================================================
--- VirtualBox-4.1.2_OSE.orig/src/VBox/Frontends/VirtualBox/src/VBoxUpdateDlg.cpp
+++ VirtualBox-4.1.2_OSE/src/VBox/Frontends/VirtualBox/src/VBoxUpdateDlg.cpp
@@ -105,7 +105,7 @@ QStringList VBoxUpdateData::list()
 
 VBoxUpdateData::VBoxUpdateData(const QString &strData)
     : m_strData(strData)
-    , m_periodIndex(Period1Day)
+    , m_periodIndex(PeriodNever)
     , m_branchIndex(BranchStable)
 {
     decode();
