Index: VirtualBox-4.0.2_OSE/src/VBox/VMM/VMMR3/VM.cpp
===================================================================
--- VirtualBox-4.0.2_OSE.orig/src/VBox/VMM/VMMR3/VM.cpp
+++ VirtualBox-4.0.2_OSE/src/VBox/VMM/VMMR3/VM.cpp
@@ -361,10 +361,8 @@ VMMR3DECL(int)   VMR3Create(uint32_t cCp
             {
                 case VERR_VM_DRIVER_LOAD_ERROR:
 #ifdef RT_OS_LINUX
-                    pszError = N_("VirtualBox kernel driver not loaded. The vboxdrv kernel module "
-                                  "was either not loaded or /dev/vboxdrv is not set up properly. "
-                                  "Re-setup the kernel module by executing "
-                                  "'/etc/init.d/vboxdrv setup' as root");
+                    pszError = N_("VirtualBox kernel driver not loaded. "
+                                  "Please install the module-virtualbox package.");
 #else
                     pszError = N_("VirtualBox kernel driver not loaded");
 #endif
@@ -403,10 +401,8 @@ VMMR3DECL(int)   VMR3Create(uint32_t cCp
                 case VERR_INVALID_HANDLE: /** @todo track down and fix this error. */
                 case VERR_VM_DRIVER_NOT_INSTALLED:
 #ifdef RT_OS_LINUX
-                    pszError = N_("VirtualBox kernel driver not installed. The vboxdrv kernel module "
-                                  "was either not loaded or /dev/vboxdrv was not created for some "
-                                  "reason. Re-setup the kernel module by executing "
-                                  "'/etc/init.d/vboxdrv setup' as root");
+                    pszError = N_("VirtualBox kernel driver not installed. "
+                                  "Please install the module-virtualbox package.");
 #else
                     pszError = N_("VirtualBox kernel driver not installed");
 #endif
@@ -416,9 +412,9 @@ VMMR3DECL(int)   VMR3Create(uint32_t cCp
                     break;
                 case VERR_VERSION_MISMATCH:
                 case VERR_VM_DRIVER_VERSION_MISMATCH:
-                    pszError = N_("The VirtualBox support driver which is running is from a different "
-                                  "version of VirtualBox.  You can correct this by stopping all "
-                                  "running instances of VirtualBox and reinstalling the software.");
+                    pszError = N_("The version of the VirtualBox kernel modules and the version of "
+                                  "VirtualBox application are not matching. You can correct this by "
+                                  "updating virtualbox and module-virtualbox packages.");
                     break;
                 default:
                     pszError = N_("Unknown error initializing kernel driver");
