Index: VirtualBox-3.2.8_OSE/configure
===================================================================
--- VirtualBox-3.2.8_OSE.orig/configure
+++ VirtualBox-3.2.8_OSE/configure
@@ -1843,7 +1843,7 @@ EOF
   found=
 # For Solaris we use libpython2.4 for compatibility with Solaris 10 and passing IPS pkg audit
   if [ "$OS" != "solaris" ]; then
-    SUPPYTHONLIBS="python2.6 python2.5 python2.4 python2.3"
+    SUPPYTHONLIBS="python2.7 python2.6 python2.5 python2.4 python2.3"
   else
     SUPPYTHONLIBS="python2.4"
   fi
