Index: blobwars-1.19/icons/blobwars.desktop
===================================================================
--- blobwars-1.19.orig/icons/blobwars.desktop
+++ blobwars-1.19/icons/blobwars.desktop
@@ -1,9 +1,11 @@
 [Desktop Entry]
-Encoding=UTF-8
-Categories=Game;ActionGame;
-Name=Blob Wars : Metal Blob Solid
+Name=Blobwars
+GenericName=Blob Wars: Metal Blob Solid
+GenericName[tr]=Blob Wars: Metal Blob Solid
 Comment=Mission and Objective based 2D Platform Game
+Comment[tr]=Görev ve Hedef Tabanlı 2B Platform Oyunu
 Icon=blobwars
 Exec=blobwars
 Terminal=false
 Type=Application
+Categories=Game;ActionGame;
