Index: blobwars-1.19/src/main.cpp
===================================================================
--- blobwars-1.19.orig/src/main.cpp
+++ blobwars-1.19/src/main.cpp
@@ -104,7 +104,11 @@ int main(int argc, char *argv[])
 	#endif
 
 	#if RELEASE
-	chdir(PAKLOCATION);
+	if (chdir(PAKLOCATION))
+	{
+	    perror("Could not chdir to " PAKLOCATION ":");
+	    return 1;
+	}
 	#endif
 	
 	config.engine = &engine;
