diff -Nur clutter-imcontext-0.1.4-old/configure.ac clutter-imcontext-0.1.4/configure.ac
--- clutter-imcontext-0.1.4-old/configure.ac	2009-10-03 01:35:32.507865609 +0300
+++ clutter-imcontext-0.1.4/configure.ac	2009-10-03 01:36:42.390868433 +0300
@@ -38,7 +38,6 @@
 AC_SUBST(CLUTTER_CFLAGS)
 AC_SUBST(CLUTTER_LIBS)
 
-GTK_DOC_CHECK([1.9])
 
 # Add --enable-debug arg
 AC_ARG_ENABLE(debug,
@@ -54,9 +53,6 @@
 clutter-imcontext/clutter-imcontext.pc
 tools/Makefile
 tests/Makefile
-doc/Makefile
-doc/reference/Makefile
-doc/reference/version.xml
 ])
 
 echo ""
diff -Nur clutter-imcontext-0.1.4-old/Makefile.am clutter-imcontext-0.1.4/Makefile.am
--- clutter-imcontext-0.1.4-old/Makefile.am	2009-10-03 01:35:32.506865063 +0300
+++ clutter-imcontext-0.1.4/Makefile.am	2009-10-03 01:36:55.693939081 +0300
@@ -1,5 +1,5 @@
 EXTRA_DIST = autogen.sh
-SUBDIRS = clutter-imcontext tools tests doc
+SUBDIRS = clutter-imcontext tools tests
 
 DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-doc
 
