diff -Nur hping3-20051105-old/Makefile.in hping3-20051105/Makefile.in
--- hping3-20051105-old/Makefile.in	2009-09-14 21:54:18.979453681 +0300
+++ hping3-20051105/Makefile.in	2009-09-14 21:55:00.476204226 +0300
@@ -50,7 +50,7 @@
 	$(RANLIB) $@
 
 hping3: byteorder.h $(OBJ)
-	$(CC) -o hping3 $(CCOPT) $(DEBUG) $(OBJ) -L/usr/local/lib $(PCAP) @SOLARISLIB@ @TCL_LIB@
+	$(CC) -o hping3 $(CCOPT) $(LDFLAGS) $(DEBUG) $(OBJ) -L/usr/local/lib $(PCAP) @SOLARISLIB@ @TCL_LIB@
 	@echo
 
 hping3-static: byteorder.h $(OBJ)
