--- cpio-2.9/src/copyout.c.exitCode	2007-06-28 10:54:43.000000000 +0200
+++ cpio-2.9/src/copyout.c	2007-07-16 16:23:05.000000000 +0200
@@ -297,7 +297,7 @@ to_ascii (char *where, uintmax_t v, size
 static void
 field_width_error (const char *filename, const char *fieldname)
 {
-  error (0, 0, _("%s: field width not sufficient for storing %s"),
+  error (1, 0, _("%s: field width not sufficient for storing %s"),
 	 filename, fieldname);
 }
 
