--- cpio-2.9/doc/cpio.info.rh	2007-06-28 15:25:08.000000000 +0200
+++ cpio-2.9/doc/cpio.info	2007-07-16 15:10:20.000000000 +0200
@@ -266,7 +266,8 @@ File: cpio.info,  Node: Options,  Prev: 
      Set the I/O block size to BLOCK-SIZE * 512 bytes.
 
 `-c'
-     Use the old portable (ASCII) archive format.
+     Identical to "-H newc", use the new (SVR4) portable format.
+     If you wish the old portable (ASCII) archive format, use "-H odc" instead.
 
 `-C IO-SIZE'
 `--io-size=IO-SIZE'
--- cpio-2.9/src/main.c.rh	2007-06-28 12:46:41.000000000 +0200
+++ cpio-2.9/src/main.c	2007-07-16 15:09:10.000000000 +0200
@@ -111,7 +111,7 @@ static struct argp_option options[] = {
   {"block-size", BLOCK_SIZE_OPTION, N_("BLOCK-SIZE"), 0,
    N_("Set the I/O block size to BLOCK-SIZE * 512 bytes"), GRID+1 },
   {NULL, 'c', NULL, 0,
-   N_("Use the old portable (ASCII) archive format"), GRID+1 },
+   N_("Identical to \"-H newc\", use the new (SVR4) portable format.If you wish the old portable (ASCII) archive format, use \"-H odc\" instead."), GRID+1 },
   {"dot", 'V', NULL, 0, 
    N_("Print a \".\" for each file processed"), GRID+1 },
   {"io-size", 'C', N_("NUMBER"), 0,
@@ -338,6 +338,7 @@ parse_opt (int key, char *arg, struct ar
     case 'c':		/* Use the old portable ASCII format.  */
       if (archive_format != arf_unknown)
 	error (0, EXIT_FAILURE, _("Archive format multiply defined"));
+#define SVR4_COMPAT
 #ifdef SVR4_COMPAT
       archive_format = arf_newascii; /* -H newc.  */
 #else
