--- client/config.py.orig	2011-01-09 14:27:51.791000017 +0200
+++ client/config.py	2011-01-09 14:49:32.558000025 +0200
@@ -3,17 +3,17 @@
 import commands
 import os_detect
 
-SMOON_URL = "http://www.smolts.org/"
+SMOON_URL = "http://smolt.pardus.org.tr:8090"
 SECURE = 0
 
 
 #Only a fool would edit what lays beyond here
 #Are you that fool?
 
-HW_UUID = "/etc/smolt/hw-uuid"
+HW_UUID = "/etc/hw-uuid"
 PUB_UUID = "/etc/smolt/pub-uuid"
-#UUID_DB = "/etc/smolt/uuiddb.cfg"
-#ADMIN_TOKEN = "/etc/sysconfig/smolt-token"
+UUID_DB = "/etc/smolt/uuiddb.cfg"
+ADMIN_TOKEN = "/etc/smolt/smolt-token"
 
 
 #These are the defaults taken from the source code.
