--- client/os_detect.py.orig	2009-06-13 17:07:27.000000000 +0300
+++ client/os_detect.py	2009-06-13 17:08:32.000000000 +0300
@@ -1,4 +1,5 @@
 import os
+import subprocess
 from UserDict import UserDict
 
 class odict(UserDict):
@@ -101,7 +102,8 @@
       full_path_to_executable = os.path.join(path, executable)
       if os.path.exists(full_path_to_executable):
         command = executable + params
-        child_stdin, child_stdout = os.popen2(command)
+        p = subprocess.Popen(command, shell=True, bufsize=-1, stdin=subprocess.PIPE, stdout=subprocess.PIPE, close_fds=True)
+        (child_stdin, child_stdout) = (p.stdin, p.stdout)
         output = child_stdout.readline().strip()
         child_stdout.close()
         child_stdin.close()
