diff -Nur pingus-0.7.1-old/src/pingus_main.cpp pingus-0.7.1/src/pingus_main.cpp
--- pingus-0.7.1-old/src/pingus_main.cpp	2007-09-24 14:23:04.595977458 +0300
+++ pingus-0.7.1/src/pingus_main.cpp	2007-09-24 16:34:18.103022450 +0300
@@ -578,7 +578,7 @@
         }
 
       char* exe_path = br_find_exe_dir(".");
-      path_manager.add_path(exe_path + std::string("/../share/pingus/data/"));
+      path_manager.add_path(std::string("/usr/share/pingus/"));
       path_manager.add_path(exe_path + std::string("/data"));
       path_manager.add_path("data"); // assume game is run from source dir, without any magic
       free(exe_path);
