--- src/input/evdev_device.cpp.old	2009-04-03 12:14:25.027452635 +0200
+++ src/input/evdev_device.cpp	2009-04-03 12:14:31.672453084 +0200
@@ -17,6 +17,7 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
+#include <cstdio>
 #include <assert.h>
 #include <linux/input.h>
 #include <sys/ioctl.h>
--- src/tinygettext/po_file_reader.cpp.old	2009-04-03 12:13:30.095452649 +0200
+++ src/tinygettext/po_file_reader.cpp	2009-04-03 12:13:38.117452492 +0200
@@ -17,6 +17,7 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
+#include <cstdio>
 #include <config.h>
 #include <vector>
 #include <iconv.h>
