diff -up hdf5-1.8.8/tools/lib/h5diff.c.ph5diff hdf5-1.8.8/tools/lib/h5diff.c
--- hdf5-1.8.8/tools/lib/h5diff.c.ph5diff	2011-11-07 15:11:35.000000000 -0700
+++ hdf5-1.8.8/tools/lib/h5diff.c	2012-02-09 12:14:59.788422152 -0700
@@ -1391,7 +1391,7 @@ hsize_t diff_match(hid_t file1_id, const
                 options->not_cmp = options->not_cmp | nFoundbyWorker.not_cmp;
                 busyTasks--;
             } /* end if */
-            else if(Status.MPI_TAG == MPI_TAG_TOK_RETURN)
+            else if(Status.MPI_TAG == MPI_TAG_DONE)
             {
                 MPI_Recv(&nFoundbyWorker, sizeof(nFoundbyWorker), MPI_BYTE, Status.MPI_SOURCE, MPI_TAG_DONE, MPI_COMM_WORLD, &Status);
                 nfound += nFoundbyWorker.nfound;
