diff -Naur hdf5-1.8.5.orig/hl/src/H5LTanalyze.c hdf5-1.8.5/hl/src/H5LTanalyze.c
--- hdf5-1.8.5.orig/hl/src/H5LTanalyze.c	2010-06-04 17:24:24.000000000 +0300
+++ hdf5-1.8.5/hl/src/H5LTanalyze.c	2010-07-08 10:12:29.096888065 +0300
@@ -2329,7 +2329,7 @@
    return ret;
 }
 
-int H5LTyyerror(char *msg)
+void H5LTyyerror(char *msg)
 {
    printf("ERROR: %s before \"%s\".\n", msg, yytext);
 }
diff -Naur hdf5-1.8.5.orig/hl/src/H5LTanalyze.l hdf5-1.8.5/hl/src/H5LTanalyze.l
--- hdf5-1.8.5.orig/hl/src/H5LTanalyze.l	2010-06-04 17:24:24.000000000 +0300
+++ hdf5-1.8.5/hl/src/H5LTanalyze.l	2010-07-08 10:12:48.804013698 +0300
@@ -181,7 +181,7 @@
    return ret;
 }
 
-int H5LTyyerror(char *msg)
+void H5LTyyerror(char *msg)
 {
    printf("ERROR: %s before \"%s\".\n", msg, yytext);
 }
diff -Naur hdf5-1.8.5.orig/hl/src/H5LTparse.c hdf5-1.8.5/hl/src/H5LTparse.c
--- hdf5-1.8.5.orig/hl/src/H5LTparse.c	2010-06-04 17:24:24.000000000 +0300
+++ hdf5-1.8.5/hl/src/H5LTparse.c	2010-07-08 10:13:12.994013752 +0300
@@ -63,7 +63,7 @@
 #include<hdf5.h>
 
 extern int yylex();
-extern int yyerror(char *);
+extern void yyerror(char *);
 
 #define STACK_SIZE      16
 
diff -Naur hdf5-1.8.5.orig/test/dt_arith.c hdf5-1.8.5/test/dt_arith.c
--- hdf5-1.8.5.orig/test/dt_arith.c	2010-06-04 17:24:13.000000000 +0300
+++ hdf5-1.8.5/test/dt_arith.c	2010-07-08 10:13:36.339012831 +0300
@@ -3409,6 +3409,7 @@
     else if(run_test==TEST_DENORM || run_test==TEST_SPECIAL)
         return 1;
 #endif
+    return 1;
 }
 
 
diff -Naur hdf5-1.8.5.orig/tools/h5dump/h5dump.c hdf5-1.8.5/tools/h5dump/h5dump.c
--- hdf5-1.8.5.orig/tools/h5dump/h5dump.c	2010-06-04 17:24:19.000000000 +0300
+++ hdf5-1.8.5/tools/h5dump/h5dump.c	2010-07-08 10:14:24.672013257 +0300
@@ -628,6 +628,7 @@
  *
  *-------------------------------------------------------------------------
  */
+static void leave(int ret) __attribute__ ((__noreturn__));
 static void
 leave(int ret)
 {
diff -Naur hdf5-1.8.5.orig/tools/h5ls/h5ls.c hdf5-1.8.5/tools/h5ls/h5ls.c
--- hdf5-1.8.5.orig/tools/h5ls/h5ls.c	2010-06-04 17:24:22.000000000 +0300
+++ hdf5-1.8.5/tools/h5ls/h5ls.c	2010-07-08 10:14:49.131888061 +0300
@@ -2144,6 +2144,7 @@
  *
  *-------------------------------------------------------------------------
  */
+static void leave(int ret) __attribute__ ((__noreturn__));
 static void
 leave(int ret)
 {
diff -Naur hdf5-1.8.5.orig/tools/h5stat/h5stat.c hdf5-1.8.5/tools/h5stat/h5stat.c
--- hdf5-1.8.5.orig/tools/h5stat/h5stat.c	2010-06-04 17:24:22.000000000 +0300
+++ hdf5-1.8.5/tools/h5stat/h5stat.c	2010-07-08 10:15:11.562013746 +0300
@@ -194,6 +194,7 @@
     { NULL, 0, '\0' }
 };
 
+static void leave(int ret) __attribute__ ((__noreturn__));
 static void
 leave(int ret)
 {
