diff -ur gnome-menus-2.26.1.orig/python/Makefile.am gnome-menus-2.26.1/python/Makefile.am
--- gnome-menus-2.26.1.orig/python/Makefile.am	2009-04-14 01:00:59.000000000 +0300
+++ gnome-menus-2.26.1/python/Makefile.am	2009-06-04 13:37:37.211819945 +0300
@@ -15,5 +15,5 @@
 gmenu_LTLIBRARIES = gmenu.la
 
 gmenu_la_LDFLAGS = -module -avoid-version -fPIC -export-symbols-regex initgmenu
-gmenu_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/libmenu/libgnome-menu.la
+gmenu_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/libmenu/libgnome-menu.la -lpython2.7
 gmenu_la_SOURCES = gmenu.c
