Index: d-feet-0.1.12/dfeet/settings.py
===================================================================
--- d-feet-0.1.12.orig/dfeet/settings.py
+++ d-feet-0.1.12/dfeet/settings.py
@@ -19,7 +19,7 @@ class Settings:
     general =   {
                 "windowheight" : 550,
                 "windowwidth" : 900,
-                "windowstate" : None,
+                "windowstate" : "None",
                 "bustabs_list" : []
                 }
     
