diff -Nur syslinux-3.63-old/bcopy32.inc syslinux-3.63/bcopy32.inc
--- syslinux-3.63-old/bcopy32.inc	2008-07-04 14:19:34.000000000 +0300
+++ syslinux-3.63/bcopy32.inc	2008-07-04 14:25:54.000000000 +0300
@@ -336,6 +336,12 @@
 		out 060h, al
 		call empty_8042_uncond
 
+		; UHCI spec assumes that A20 toggle ends with a null command
+		; this must fix some usb keyboard locks
+		mov al,0FFh			; Null command
+		out 064h, al
+		call empty_8042_uncond
+
 		; Verify that A20 actually is enabled.  Do that by
 		; observing a word in low memory and the same word in
 		; the HMA until they are no longer coherent.  Note that
@@ -455,6 +461,9 @@
 		mov al,0DDh		; A20 off
 		out 060h, al
 		call empty_8042_uncond
+		mov al,0FFh		; Null command/synchronization
+		out 064h, al
+		call empty_8042_uncond
 		; Wait a bit for it to take effect
 a20d_snooze:
 		push cx
diff -Nur syslinux-3.63-old/memdisk/memdisk16.asm syslinux-3.63/memdisk/memdisk16.asm
--- syslinux-3.63-old/memdisk/memdisk16.asm	2008-07-04 14:19:34.000000000 +0300
+++ syslinux-3.63/memdisk/memdisk16.asm	2008-07-04 14:29:15.000000000 +0300
@@ -302,6 +302,12 @@
 		out 060h, al
 		call empty_8042_uncond
 
+		; UHCI spec assumes that A20 toggle ends with a null command
+		; this must fix some usb keyboard locks
+		mov al,0FFh			; Null command
+		out 064h, al
+		call empty_8042_uncond
+
 		; Verify that A20 actually is enabled.  Do that by
 		; observing a word in low memory and the same word in
 		; the HMA until they are no longer coherent.  Note that
@@ -430,6 +436,9 @@
 		mov al,0DDh		; A20 off
 		out 060h, al
 		call empty_8042_uncond
+		mov al,0FFh		; Null command/synchronization
+		out 064h, al
+		call empty_8042_uncond
 		; Wait a bit for it to take effect
 a20d_snooze:
 		push cx
