--- utils/Makefile~ 2013-01-09 20:50:53.234544145 +0200                                                                                                                                            
+++ utils/Makefile    2013-01-09 20:52:20.056545552 +0200                                                                                                                                            
@@ -53,9 +53,6 @@                                                                                                                                                                              
 isohybrid: isohybrid.o isohdpfx.o                                                                                                                                                             
        $(CC) $(LDFLAGS) -o $@ $^ -luuid                                                                                                                                                       
 
-gethostip: gethostip.o
-       $(CC) $(LDFLAGS) -o $@ $^
-
 memdiskfind: memdiskfind.o
        $(CC) $(LDFLAGS) -o $@ $^
 
