diff -Nur syslinux-4.04-old/Makefile syslinux-4.04/Makefile > myown-4.04-remove-win32-from-build.patch
--- syslinux-4.04-old/Makefile	2011-04-18 23:24:17.000000000 +0200
+++ syslinux-4.04/Makefile	2011-08-08 12:41:53.000000000 +0200
@@ -42,7 +42,6 @@
 	mbr/*.bin \
 	core/pxelinux.0 core/isolinux.bin core/isolinux-debug.bin \
 	gpxe/gpxelinux.0 dos/syslinux.com \
-	win32/syslinux.exe win64/syslinux64.exe \
 	dosutil/*.com dosutil/*.sys \
 	$(MODULES)
 
@@ -53,7 +52,7 @@
 # files that depend only on the B phase, but may have to be regenerated
 # for "make installer".
 BSUBDIRS = codepage com32 lzo core memdisk modules mbr memdump gpxe sample \
-	   diag libinstaller dos win32 win64 dosutil
+	   diag libinstaller dos dosutil
 ITARGET  =
 IOBJECTS = $(ITARGET) \
 	utils/gethostip utils/isohybrid utils/mkdiskimage \
@@ -69,7 +68,7 @@
 		core/isolinux.bin core/isolinux-debug.bin \
 		dos/syslinux.com \
 		mbr/*.bin $(MODULES)
-INSTALL_AUX_OPT = win32/syslinux.exe win64/syslinux64.exe
+INSTALL_AUX_OPT =
 INSTALL_DIAG  =	diag/mbr/handoff.bin \
 		diag/geodsp/geodsp1s.img.xz diag/geodsp/geodspms.img.xz
 
