From 200f54ec3498c9d53c530b863ee43b42ff258fea Mon Sep 17 00:00:00 2001
From: Frederic Crozat <fcrozat@mandriva.com>
Date: Thu, 5 Mar 2009 11:44:35 +0100
Subject: [PATCH] Fix format security

---
 exp_clib.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/exp_clib.c b/exp_clib.c
index 79e6309..1ae6073 100644
--- a/exp_clib.c
+++ b/exp_clib.c
@@ -1936,8 +1936,8 @@ expDiagLogU(str)
      char *str;
 {
   if (exp_is_debugging) {
-    fprintf(stderr,str);
-    if (exp_logfile) fprintf(exp_logfile,str);
+    fprintf(stderr,"%s",str);
+    if (exp_logfile) fprintf(exp_logfile,"%s",str);
   }
 }
 
-- 
1.6.1.3

