diff -up libjingle-0.5.8/talk/session/phone/srtpfilter.cc.system-srtp libjingle-0.5.8/talk/session/phone/srtpfilter.cc
--- libjingle-0.5.8/talk/session/phone/srtpfilter.cc.system-srtp	2011-07-25 11:56:10.950441446 -0400
+++ libjingle-0.5.8/talk/session/phone/srtpfilter.cc	2011-07-25 11:56:21.599308183 -0400
@@ -58,7 +58,7 @@
 #ifdef SRTP_RELATIVE_PATH
 #include "srtp.h"  // NOLINT
 #else
-#include "third_party/libsrtp/include/srtp.h"
+#include <srtp.h>
 #endif  // SRTP_RELATIVE_PATH
 #ifdef _DEBUG
 extern "C" debug_module_t mod_srtp;
