diff -up libjingle-0.5.8/talk/xmllite/xmlbuilder.h.system-expat libjingle-0.5.8/talk/xmllite/xmlbuilder.h
--- libjingle-0.5.8/talk/xmllite/xmlbuilder.h.system-expat	2011-07-25 11:45:52.604310202 -0400
+++ libjingle-0.5.8/talk/xmllite/xmlbuilder.h	2011-07-25 11:46:04.055287491 -0400
@@ -36,7 +36,7 @@
 #ifdef EXPAT_RELATIVE_PATH
 #include "expat.h"
 #else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
+#include <expat.h>
 #endif  // EXPAT_RELATIVE_PATH
 
 namespace buzz {
diff -up libjingle-0.5.8/talk/xmllite/xmlparser.h.system-expat libjingle-0.5.8/talk/xmllite/xmlparser.h
--- libjingle-0.5.8/talk/xmllite/xmlparser.h.system-expat	2011-07-25 11:47:21.209287907 -0400
+++ libjingle-0.5.8/talk/xmllite/xmlparser.h	2011-07-25 11:47:32.917530191 -0400
@@ -34,7 +34,7 @@
 #ifdef EXPAT_RELATIVE_PATH
 #include "expat.h"
 #else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
+#include <expat.h>
 #endif  // EXPAT_RELATIVE_PATH
 
 struct XML_ParserStruct;
diff -up libjingle-0.5.8/talk/xmpp/xmppstanzaparser.cc.system-expat libjingle-0.5.8/talk/xmpp/xmppstanzaparser.cc
--- libjingle-0.5.8/talk/xmpp/xmppstanzaparser.cc.system-expat	2011-07-25 11:45:19.013287946 -0400
+++ libjingle-0.5.8/talk/xmpp/xmppstanzaparser.cc	2011-07-25 11:45:30.391424695 -0400
@@ -33,7 +33,7 @@
 #ifdef EXPAT_RELATIVE_PATH
 #include "expat.h"
 #else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
+#include <expat.h>
 #endif
 
 namespace buzz {
