diff -up libjingle-0.6.0/talk/p2p/base/pseudotcp.h.size_t libjingle-0.6.0/talk/p2p/base/pseudotcp.h
--- libjingle-0.6.0/talk/p2p/base/pseudotcp.h.size_t	2011-09-13 18:47:00.000000000 -0400
+++ libjingle-0.6.0/talk/p2p/base/pseudotcp.h	2011-09-27 11:15:48.274865394 -0400
@@ -29,6 +29,7 @@
 #define TALK_P2P_BASE_PSEUDOTCP_H_
 
 #include <list>
+#include <cstdlib>
 
 #include "talk/base/basictypes.h"
 #include "talk/base/stream.h"
