diff -up libjingle-0.5.8/talk/base/physicalsocketserver.cc.statfix libjingle-0.5.8/talk/base/physicalsocketserver.cc
--- libjingle-0.5.8/talk/base/physicalsocketserver.cc.statfix	2011-07-25 11:39:12.301403847 -0400
+++ libjingle-0.5.8/talk/base/physicalsocketserver.cc	2011-07-25 11:39:25.723411866 -0400
@@ -34,6 +34,7 @@
 #ifdef POSIX
 #include <string.h>
 #include <errno.h>
+#include <time.h>
 #include <fcntl.h>
 #include <sys/time.h>
 #include <unistd.h>
diff -up libjingle-0.5.8/talk/base/stream.cc.statfix libjingle-0.5.8/talk/base/stream.cc
--- libjingle-0.5.8/talk/base/stream.cc.statfix	2011-07-25 11:39:48.101311105 -0400
+++ libjingle-0.5.8/talk/base/stream.cc	2011-07-25 11:39:55.260452171 -0400
@@ -26,6 +26,7 @@
  */
 
 #if defined(POSIX)
+#include <time.h>
 #include <sys/file.h>
 #endif  // POSIX
 #include <sys/types.h>
