diff -up libjingle-0.6.6/talk/base/ratetracker.cc.compilefix libjingle-0.6.6/talk/base/ratetracker.cc
--- libjingle-0.6.6/talk/base/ratetracker.cc.compilefix	2011-12-14 10:58:28.000000000 -0500
+++ libjingle-0.6.6/talk/base/ratetracker.cc	2012-01-05 13:38:54.590475337 -0500
@@ -25,8 +25,8 @@
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include "talk/base/ratetracker.h"
 #include "talk/base/timeutils.h"
+#include "talk/base/ratetracker.h"
 
 namespace talk_base {
 
