diff -up libjingle-0.5.8/talk/base/linux.cc.config libjingle-0.5.8/talk/base/linux.cc
--- libjingle-0.5.8/talk/base/linux.cc.config	2011-07-01 18:57:06.000000000 -0400
+++ libjingle-0.5.8/talk/base/linux.cc	2011-07-25 12:05:40.084014155 -0400
@@ -25,9 +25,9 @@
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#if defined(LINUX) || defined(ANDROID)
 #include "talk/base/linux.h"
 
+#if defined(LINUX) || defined(ANDROID)
 #include <ctype.h>
 
 #include <errno.h>
diff -up libjingle-0.5.8/talk/base/linux.h.config libjingle-0.5.8/talk/base/linux.h
--- libjingle-0.5.8/talk/base/linux.h.config	2011-07-01 18:57:06.000000000 -0400
+++ libjingle-0.5.8/talk/base/linux.h	2011-07-25 12:05:53.314954072 -0400
@@ -28,6 +28,8 @@
 #ifndef TALK_BASE_LINUX_H_
 #define TALK_BASE_LINUX_H_
 
+#include "config.h"
+
 #if defined(LINUX) || defined(ANDROID)
 #include <string>
 #include <map>
