diff -up libjingle-0.5.1/talk/base/thread.cc.timefix libjingle-0.5.1/talk/base/thread.cc
--- libjingle-0.5.1/talk/base/thread.cc.timefix	2010-11-19 10:53:41.509966999 -0500
+++ libjingle-0.5.1/talk/base/thread.cc	2010-11-19 10:53:51.317967002 -0500
@@ -25,14 +25,13 @@
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include "talk/base/thread.h"
-
 #if defined(WIN32)
 #include <comdef.h>
 #elif defined(POSIX)
 #include <time.h>
 #endif
 
+#include "talk/base/thread.h"
 #include "talk/base/common.h"
 #include "talk/base/logging.h"
 #include "talk/base/stringutils.h"
