diff -up libjingle-0.5.1/talk/base/basictypes.h.linkage libjingle-0.5.1/talk/base/basictypes.h
--- libjingle-0.5.1/talk/base/basictypes.h.linkage	2010-11-18 16:20:19.365761996 -0500
+++ libjingle-0.5.1/talk/base/basictypes.h	2010-11-18 16:20:28.934762001 -0500
@@ -80,8 +80,10 @@ typedef int socklen_t;
 #endif
 
 namespace talk_base {
+extern "C++" {
   template<class T> inline T _min(T a, T b) { return (a > b) ? b : a; }
   template<class T> inline T _max(T a, T b) { return (a < b) ? b : a; }
+}
 
   // For wait functions that take a number of milliseconds, kForever indicates
   // unlimited time.
