diff -up libjingle-0.5.1/talk/base/md5.h.uint32 libjingle-0.5.1/talk/base/md5.h
--- libjingle-0.5.1/talk/base/md5.h.uint32	2010-11-18 16:31:58.492762006 -0500
+++ libjingle-0.5.1/talk/base/md5.h	2010-11-18 16:32:06.322761999 -0500
@@ -22,7 +22,7 @@
 extern "C" {
 #endif
 
-typedef long unsigned int uint32;
+typedef unsigned int uint32;
 typedef struct MD5Context MD5_CTX;
 
 #define md5byte unsigned char
