--- plotutils-2.5.1/configure.ac.orig	2008-07-16 02:00:33.000000000 +0200
+++ plotutils-2.5.1/configure.ac	2008-07-24 22:29:43.896744273 +0200
@@ -58,6 +58,9 @@
   CXXFLAGS_NOT_SET_BY_INSTALLER="no"
 fi
 
+AC_PROG_CC
+AC_PROG_CXX
+
 # We now use libtool to make and installed a shared library.  This should
 # invoke the tests AC_PROG_CC, AC_OBJEXT, AC_PROG_INSTALL, and
 # AC_PROG_MAKE_SET, or equivalents, so we don't perform them explicitly.
