Index: apbs-1.3-source/configure.ac
===================================================================
--- apbs-1.3-source.orig/configure.ac
+++ apbs-1.3-source/configure.ac
@@ -1223,7 +1223,7 @@ dnl # COMPILATIONS
 dnl # -----------------------
 top_srcdir="`pwd`"
 base_list="${FLIBS} ${tree_lib} ${vmetis_lib} ${vor_lib} ${mcx_lib} ${mc_lib} \ 
-	   ${thread_lib} ${mpi_lib} ${math_lib} ${rl_lib} ${socket_lib} ${nsl_lib} ${blas_lib} ${z_lib}"
+	   ${thread_lib} ${mpi_lib} ${math_lib} ${rl_lib} ${socket_lib} ${nsl_lib} ${z_lib}"
 
 case $host_os in
 
@@ -1259,7 +1259,7 @@ if test -n "${mc_found}"; then
 	case $host_os in
 
 	*)
-		extra_list3="-lsuperlu -lumfpack -lamd -lpunc -lblas"
+		extra_list3="-lsuperlu -lumfpack -lamd -lpunc"
 		;;
 	
 	esac
@@ -1288,7 +1288,7 @@ else
 			${top_apbsdir}/src/aaa_lib/libapbs.la \
 			${top_apbsdir}/contrib/pmgZ/libapbspmgz.la \
 			${top_apbsdir}/contrib/aqua/libapbsaqua.la \
-			${maloc_lib} ${blas_lib} ${lib_list}"
+			${maloc_lib} ${lib_list}"
 fi
 AC_SUBST(apbs_build_libraries)
 
