--- gettext-tools/gnulib-lib/gl_array_list.c
+++ gettext-tools/gnulib-lib/gl_array_list.c
@@ -421,7 +421,7 @@
   result.count = list->count;
   result.p = list->elements + 0;
   result.q = list->elements + list->count;
-#ifdef lint
+#if defined(lint) || defined(gcc_is_lint)
   result.i = 0;
   result.j = 0;
 #endif
@@ -442,7 +442,7 @@
   result.count = list->count;
   result.p = list->elements + start_index;
   result.q = list->elements + end_index;
-#ifdef lint
+#if defined(lint) || defined(gcc_is_lint)
   result.i = 0;
   result.j = 0;
 #endif
