--- gettext-tools/misc/gettextize.in
+++ gettext-tools/misc/gettextize.in
@@ -1070,20 +1070,6 @@
   echo "It is a wrapper around <libintl.h> that implements the configure --disable-nls"
   echo "option."
   echo
-  count=`echo "$please" | grep '^$' | wc -l`
-  count=`echo "$count" | sed -e 's/[ 	]//g'`
-  case "$count" in
-    1) count="paragraph";;
-    2) count="two paragraphs";;
-    3) count="three paragraphs";;
-    4) count="four paragraphs";;
-    5) count="five paragraphs";;
-    *) count="$count paragraphs";;
-  esac
-  echo "Press Return to acknowledge the previous $count."
-  # Read from /dev/tty, not stdin, so that gettextize cannot be abused by
-  # non-interactive tools.
-  read dummy < /dev/tty
 fi
 
 exit 0
