--- gettext-0.18.2/gettext-tools/libgettextpo/Makefile.am.link	2012-12-25 13:16:50.000000000 +0800
+++ gettext-0.18.2/gettext-tools/libgettextpo/Makefile.am	2012-12-26 12:51:16.213604572 +0800
@@ -98,11 +98,11 @@
 # How to build libgettextpo.la.
 # Don't make it depend on libgettextsrc.la or libgettextlib.la, since they
 # define an uncontrolled amount of symbols.
-libgettextpo_la_LIBADD = libgnu.la $(WOE32_LIBADD) $(LTLIBUNISTRING)
+libgettextpo_la_LIBADD = libgnu.la $(WOE32_LIBADD) $(LTLIBUNISTRING) @LTLIBINTL@ @LTLIBICONV@ -lc
 libgettextpo_la_LDFLAGS = \
   -version-info $(LTV_CURRENT):$(LTV_REVISION):$(LTV_AGE) \
   -rpath $(libdir) \
-  @LTLIBINTL@ @LTLIBICONV@ -lc -no-undefined
+  -no-undefined
 
 # Tell the mingw or Cygwin linker which symbols to export.
 if WOE32DLL
--- gettext-0.18.2/gettext-tools/src/Makefile.am.link	2012-12-25 07:00:48.000000000 +0800
+++ gettext-0.18.2/gettext-tools/src/Makefile.am	2012-12-26 12:52:04.704871906 +0800
@@ -237,8 +237,9 @@
 # Need @LTLIBICONV@ because po-charset.c, po-lex.c, msgl-iconv.c, write-po.c
 # use iconv().
 libgettextsrc_la_LDFLAGS = \
-  -release @VERSION@ \
-  ../gnulib-lib/libgettextlib.la $(LTLIBUNISTRING) @LTLIBINTL@ @LTLIBICONV@ -lc -no-undefined
+  -release @VERSION@ -no-undefined
+libgettextsrc_la_LIBADD = \
+  ../gnulib-lib/libgettextlib.la $(LTLIBUNISTRING) @LTLIBINTL@ @LTLIBICONV@ -lc
 
 # Tell the mingw or Cygwin linker which symbols to export.
 if WOE32DLL
