diff -up gettext-0.18.2/gettext-tools/tests/Makefile.am.shsh gettext-0.18.2/gettext-tools/tests/Makefile.am
--- gettext-0.18.2/gettext-tools/tests/Makefile.am.shsh	2012-12-04 07:28:58.000000000 +0100
+++ gettext-0.18.2/gettext-tools/tests/Makefile.am	2013-01-12 01:27:54.873524715 +0100
@@ -187,8 +187,7 @@ TESTS_ENVIRONMENT = top_srcdir=$(top_src
 		    LOCALE_FR_UTF8='@LOCALE_FR_UTF8@' \
 		    LOCALE_JA='@LOCALE_JA@' \
 		    host_os='@host_os@' \
-		    CONFIG_SHELL='$(SHELL)' \
-		    $(SHELL)
+		    CONFIG_SHELL='$(SHELL)'
 
 # Update the expected test results.
 update-expected: xg-c-1.ok.po
