--- gettext-tools/gnulib-tests/test-xvasprintf.c.str	2012-12-25 13:29:46.000000000 +0800
+++ gettext-tools/gnulib-tests/test-xvasprintf.c	2012-12-26 12:49:39.391074145 +0800
@@ -95,7 +95,7 @@
   {
     /* Silence gcc warning about zero-length format string.  */
     const char *empty = "";
-    result = xasprintf (empty);
+    result = xasprintf ("%s", empty);
     ASSERT (result != NULL);
     ASSERT (strcmp (result, "") == 0);
     free (result);
--- gettext-tools/libgrep/m-fgrep.c.str	2012-12-04 14:28:58.000000000 +0800
+++ gettext-tools/libgrep/m-fgrep.c	2012-12-26 12:49:00.220050303 +0800
@@ -106,7 +106,7 @@
       for (lim = beg; lim < pattern + pattern_size && *lim != '\n'; ++lim)
         ;
       if ((err = kwsincr (ckwset->kwset, beg, lim - beg)) != NULL)
-        error (exit_failure, 0, err);
+        error (exit_failure, 0, "%s", err);
       if (lim < pattern + pattern_size)
         ++lim;
       beg = lim;
@@ -114,7 +114,7 @@
   while (beg < pattern + pattern_size);
 
   if ((err = kwsprep (ckwset->kwset)) != NULL)
-    error (exit_failure, 0, err);
+    error (exit_failure, 0, "%s", err);
   return ckwset;
 }
 
--- gettext-tools/libgrep/m-regex.c.str	2012-12-04 14:28:58.000000000 +0800
+++ gettext-tools/libgrep/m-regex.c	2012-12-26 12:49:00.220050303 +0800
@@ -106,7 +106,7 @@
 
         if ((err = re_compile_pattern (motif, len,
                                        &cregex->patterns[cregex->pcount].regexbuf)) != NULL)
-          error (exit_failure, 0, err);
+          error (exit_failure, 0, "%s", err);
         cregex->pcount++;
 
         motif = sep;
