--- gettext-tools/gnulib-lib/gl_anylinked_list2.h
+++ gettext-tools/gnulib-lib/gl_anylinked_list2.h
@@ -35,6 +35,12 @@
 # define ASYNCSAFE(type)
 #endif
 
+#if (__GNUC__ > 4)  || (__GNUC__ == 4 && __GNUC_MINOR__ == 1 && __GNUC_PATCHLEVEL__ >= 3) || (__GNUC__ == 4 && __GNUC_MINOR__ > 1)
+# ifndef lint
+#  define lint
+# endif
+#endif
+
 /* -------------------------- gl_list_t Data Type -------------------------- */
 
 static gl_list_t
