diff -p -up gettext-0.18.1.1/gettext-tools/gnulib-lib/libcroco/cr-statement.c~ gettext-0.18.1.1/gettext-tools/gnulib-lib/libcroco/cr-statement.c
--- gettext-0.18.1.1/gettext-tools/gnulib-lib/libcroco/cr-statement.c~	2009-02-04 10:33:45.000000000 +0100
+++ gettext-0.18.1.1/gettext-tools/gnulib-lib/libcroco/cr-statement.c	2009-02-04 12:13:19.000000000 +0100
@@ -2611,7 +2611,7 @@ cr_statement_dump_ruleset (CRStatement c
         g_return_if_fail (a_fp && a_this);
         str = cr_statement_ruleset_to_string (a_this, a_indent);
         if (str) {
-                fprintf (a_fp, str);
+                fprintf (a_fp, "%s", str);
                 g_free (str);
                 str = NULL;
         }
@@ -2662,7 +2662,7 @@ cr_statement_dump_charset (CRStatement c
         str = cr_statement_charset_to_string (a_this,
                                               a_indent) ;
         if (str) {
-                fprintf (a_fp, str) ;
+                fprintf (a_fp, "%s", str) ;
                 g_free (str) ;
                 str = NULL ;
         }
@@ -2689,7 +2689,7 @@ cr_statement_dump_page (CRStatement cons
 
         str = cr_statement_at_page_rule_to_string (a_this, a_indent) ;
         if (str) {
-                fprintf (a_fp, str);
+                fprintf (a_fp, "%s", str);
                 g_free (str) ;
                 str = NULL ; 
         }
@@ -2715,7 +2715,7 @@ cr_statement_dump_media_rule (CRStatemen
 
         str = cr_statement_media_rule_to_string (a_this, a_indent) ;
         if (str) {
-                fprintf (a_fp, str) ;
+                fprintf (a_fp, "%s", str) ;
                 g_free (str) ;
                 str = NULL ;
         }
@@ -2741,7 +2741,7 @@ cr_statement_dump_import_rule (CRStateme
 
         str = cr_statement_import_rule_to_string (a_this, a_indent) ;
         if (str) {
-                fprintf (a_fp, str) ;
+                fprintf (a_fp, "%s", str) ;
                 g_free (str) ;
                 str = NULL ;
         }
diff -p -up gettext-0.18.1.1/gnulib-local/lib/libcroco/cr-statement.c~ gettext-0.18.1.1/gnulib-local/lib/libcroco/cr-statement.c
--- gettext-0.18.1.1/gnulib-local/lib/libcroco/cr-statement.c~	2009-02-04 10:33:45.000000000 +0100
+++ gettext-0.18.1.1/gnulib-local/lib/libcroco/cr-statement.c	2009-02-04 12:13:19.000000000 +0100
@@ -2611,7 +2611,7 @@ cr_statement_dump_ruleset (CRStatement c
         g_return_if_fail (a_fp && a_this);
         str = cr_statement_ruleset_to_string (a_this, a_indent);
         if (str) {
-                fprintf (a_fp, str);
+                fprintf (a_fp, "%s", str);
                 g_free (str);
                 str = NULL;
         }
@@ -2662,7 +2662,7 @@ cr_statement_dump_charset (CRStatement c
         str = cr_statement_charset_to_string (a_this,
                                               a_indent) ;
         if (str) {
-                fprintf (a_fp, str) ;
+                fprintf (a_fp, "%s", str) ;
                 g_free (str) ;
                 str = NULL ;
         }
@@ -2689,7 +2689,7 @@ cr_statement_dump_page (CRStatement cons
 
         str = cr_statement_at_page_rule_to_string (a_this, a_indent) ;
         if (str) {
-                fprintf (a_fp, str);
+                fprintf (a_fp, "%s", str);
                 g_free (str) ;
                 str = NULL ; 
         }
@@ -2715,7 +2715,7 @@ cr_statement_dump_media_rule (CRStatemen
 
         str = cr_statement_media_rule_to_string (a_this, a_indent) ;
         if (str) {
-                fprintf (a_fp, str) ;
+                fprintf (a_fp, "%s", str) ;
                 g_free (str) ;
                 str = NULL ;
         }
@@ -2741,7 +2741,7 @@ cr_statement_dump_import_rule (CRStateme
 
         str = cr_statement_import_rule_to_string (a_this, a_indent) ;
         if (str) {
-                fprintf (a_fp, str) ;
+                fprintf (a_fp, "%s", str) ;
                 g_free (str) ;
                 str = NULL ;
         }

