--- a/src/mfile.orig
+++ b/src/mfile
@@ -429,22 +429,22 @@ check:
 
 
 install:
-	mkdir -p -m 755 $(INCLUDEDIR)
-	rm -rf $(INCLUDEDIR)/NTL
-	mkdir -m 755 $(INCLUDEDIR)/NTL
-	cp -p ../include/NTL/*.h $(INCLUDEDIR)/NTL
-	- chmod -R a+r $(INCLUDEDIR)/NTL
-	mkdir -p -m 755 $(DOCDIR)
-	rm -rf $(DOCDIR)/NTL
-	mkdir -m 755 $(DOCDIR)/NTL
-	cp -p ../doc/*.txt $(DOCDIR)/NTL
-	cp -p ../doc/*.html $(DOCDIR)/NTL
-	cp -p ../doc/*.gif $(DOCDIR)/NTL
-	- chmod -R a+r $(DOCDIR)/NTL
-	mkdir -p -m 755 $(LIBDIR)
-@{LSTAT}	cp -p ntl.a $(LIBDIR)/libntl.a #LSTAT
-@{LSTAT}	- chmod a+r $(LIBDIR)/libntl.a #LSTAT
-@{LSHAR}	$(LIBTOOL) --mode=install cp -p libntl.la $(LIBDIR) #LSHAR
+	mkdir -p -m 755 $(DESTDIR)$(INCLUDEDIR)
+	rm -rf $(DESTDIR)$(INCLUDEDIR)/NTL
+	mkdir -m 755 $(DESTDIR)$(INCLUDEDIR)/NTL
+	cp -p ../include/NTL/*.h $(DESTDIR)$(INCLUDEDIR)/NTL
+	- chmod -R a+r $(DESTDIR)$(INCLUDEDIR)/NTL
+	mkdir -p -m 755 $(DESTDIR)$(DOCDIR)
+	rm -rf $(DESTDIR)$(DOCDIR)/NTL
+	mkdir -m 755 $(DESTDIR)$(DOCDIR)/NTL
+	cp -p ../doc/*.txt $(DESTDIR)$(DOCDIR)/NTL
+	cp -p ../doc/*.html $(DESTDIR)$(DOCDIR)/NTL
+	cp -p ../doc/*.gif $(DESTDIR)$(DOCDIR)/NTL
+	- chmod -R a+r $(DESTDIR)$(DOCDIR)/NTL
+	mkdir -p -m 755 $(DESTDIR)$(LIBDIR)
+@{LSTAT}	cp -p ntl.a $(DESTDIR)$(LIBDIR)/libntl.a #LSTAT
+@{LSTAT}	- chmod a+r $(DESTDIR)$(LIBDIR)/libntl.a #LSTAT
+@{LSHAR}	$(LIBTOOL) --mode=install cp -p libntl.la $(DESTDIR)$(LIBDIR) #LSHAR
 
 
 uninstall:
