Index: ntl-5.5.1/src/mfile
===================================================================
--- ntl-5.5.1.orig/src/mfile
+++ ntl-5.5.1/src/mfile
@@ -314,8 +314,8 @@ SFILES=$(SFI3)
 NTL_INCLUDE = -I../include -I.
 # NTL needs this to find its include files
 
-COMPILE = $(CC) $(NTL_INCLUDE) $(CPPFLAGS) $(CFLAGS) -c
-COMPILE_CXX = $(CXX) $(NTL_INCLUDE) $(CPPFLAGS) $(CXXFLAGS) -c
+COMPILE = $(CC) -shared $(NTL_INCLUDE) $(CPPFLAGS) $(CFLAGS) -c
+COMPILE_CXX = $(CXX) -shared $(NTL_INCLUDE) $(CPPFLAGS) $(CXXFLAGS) -c
 
 LINK = $(CC) $(NTL_INCLUDE) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS)
 LINK_CXX = $(CXX) $(NTL_INCLUDE) $(CPPFLAGS) $(CXXFLAGS) $(LDFLAGS_CXX)
