--- a/src/TestScript.orig	2007-02-10 23:18:14.000000000 +0100
+++ b/src/TestScript	2007-02-10 23:19:56.000000000 +0100
@@ -143,14 +143,14 @@
 echo "---------------------------------"
 echo "making subset"
 make subset
-echo "subset 40 40 20 10 999999 f"
+echo "subset 10 10 20 10 999999 f"
 ./subset <<here
-  40 40 20 10 999999 f
-here
-echo "subset 40 40 20 10 999999 q"
-./subset <<here
-  40 40 20 10 999999 q
+  10 10 20 10 999999 f
 here
+#echo "subset 40 40 20 10 999999 q"
+#./subset <<here
+#  40 40 20 10 999999 q
+#here
 
 sh RemoveProg subset
 
