diff -up wxPython-src-2.8.12.0/wxPython/contrib/gizmos/gtk/gizmos_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/contrib/gizmos/gtk/gizmos_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/contrib/gizmos/gtk/gizmos_wrap.cpp.format    2014-03-14 11:35:34.581008618 +0100
+++ wxPython-src-2.8.12.0/wxPython/contrib/gizmos/gtk/gizmos_wrap.cpp   2014-03-14 11:35:46.035843300 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/contrib/glcanvas/gtk/glcanvas_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/contrib/glcanvas/gtk/glcanvas_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/contrib/glcanvas/gtk/glcanvas_wrap.cpp.format    2014-03-14 11:34:19.451092820 +0100
+++ wxPython-src-2.8.12.0/wxPython/contrib/glcanvas/gtk/glcanvas_wrap.cpp   2014-03-14 11:34:32.157909458 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/contrib/stc/gtk/stc_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/contrib/stc/gtk/stc_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/contrib/stc/gtk/stc_wrap.cpp.format  2014-03-14 11:34:48.078679712 +0100
+++ wxPython-src-2.8.12.0/wxPython/contrib/stc/gtk/stc_wrap.cpp 2014-03-14 11:34:59.102520628 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/_core_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/_core_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/src/gtk/_core_wrap.cpp.format    2014-03-14 11:27:46.008769964 +0100
+++ wxPython-src-2.8.12.0/wxPython/src/gtk/_core_wrap.cpp   2014-03-14 11:28:51.805819727 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/_gdi_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/_gdi_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/src/gtk/_gdi_wrap.cpp.format 2014-03-14 11:27:46.008769964 +0100
+++ wxPython-src-2.8.12.0/wxPython/src/gtk/_gdi_wrap.cpp    2014-03-14 11:28:51.805819727 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/animate_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/animate_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/src/gtk/animate_wrap.cpp.format  2014-03-14 11:27:46.008769964 +0100
+++ wxPython-src-2.8.12.0/wxPython/src/gtk/animate_wrap.cpp 2014-03-14 11:28:51.805819727 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/aui_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/aui_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/src/gtk/aui_wrap.cpp.format  2014-03-14 11:27:46.013769892 +0100
+++ wxPython-src-2.8.12.0/wxPython/src/gtk/aui_wrap.cpp 2014-03-14 11:29:04.018643338 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/calendar_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/calendar_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/src/gtk/calendar_wrap.cpp.format 2014-03-14 11:27:46.015769863 +0100
+++ wxPython-src-2.8.12.0/wxPython/src/gtk/calendar_wrap.cpp    2014-03-14 11:29:17.674446166 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/combo_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/combo_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/src/gtk/combo_wrap.cpp.format    2014-03-14 11:27:46.018769820 +0100
+++ wxPython-src-2.8.12.0/wxPython/src/gtk/combo_wrap.cpp   2014-03-14 11:29:29.227279558 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/_controls_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/_controls_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/src/gtk/_controls_wrap.cpp.format    2014-03-14 11:27:46.025769719 +0100
+++ wxPython-src-2.8.12.0/wxPython/src/gtk/_controls_wrap.cpp   2014-03-14 11:28:04.778498903 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/_core_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/_core_wrap.cpp
diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/_gdi_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/_gdi_wrap.cpp
diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/grid_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/grid_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/src/gtk/grid_wrap.cpp.format 2014-03-14 11:27:46.044769444 +0100
+++ wxPython-src-2.8.12.0/wxPython/src/gtk/grid_wrap.cpp    2014-03-14 11:29:40.914111013 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/html_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/html_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/src/gtk/html_wrap.cpp.format 2014-03-14 11:27:46.048769387 +0100
+++ wxPython-src-2.8.12.0/wxPython/src/gtk/html_wrap.cpp    2014-03-14 11:29:52.878938455 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/media_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/media_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/src/gtk/media_wrap.cpp.format    2014-03-14 11:27:46.050769358 +0100
+++ wxPython-src-2.8.12.0/wxPython/src/gtk/media_wrap.cpp   2014-03-14 11:30:04.953764306 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/_misc_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/_misc_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/src/gtk/_misc_wrap.cpp.format    2014-03-14 11:27:46.055769285 +0100
+++ wxPython-src-2.8.12.0/wxPython/src/gtk/_misc_wrap.cpp   2014-03-14 11:28:26.677182645 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/richtext_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/richtext_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/src/gtk/richtext_wrap.cpp.format 2014-03-14 11:27:46.060769213 +0100
+++ wxPython-src-2.8.12.0/wxPython/src/gtk/richtext_wrap.cpp    2014-03-14 11:30:15.843607244 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/webkit_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/webkit_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/src/gtk/webkit_wrap.cpp.format   2014-03-14 11:27:46.063769170 +0100
+++ wxPython-src-2.8.12.0/wxPython/src/gtk/webkit_wrap.cpp  2014-03-14 11:30:31.108387076 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/_windows_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/_windows_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/src/gtk/_windows_wrap.cpp.format 2014-03-14 11:27:46.068769098 +0100
+++ wxPython-src-2.8.12.0/wxPython/src/gtk/_windows_wrap.cpp    2014-03-14 11:28:41.342970838 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/wizard_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/wizard_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/src/gtk/wizard_wrap.cpp.format   2014-03-14 11:27:46.070769069 +0100
+++ wxPython-src-2.8.12.0/wxPython/src/gtk/wizard_wrap.cpp  2014-03-14 11:30:42.440223630 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }

diff -up wxPython-src-2.8.12.0/wxPython/src/gtk/xrc_wrap.cpp.format wxPython-src-2.8.12.0/wxPython/src/gtk/xrc_wrap.cpp
--- wxPython-src-2.8.12.0/wxPython/src/gtk/xrc_wrap.cpp.format  2014-03-14 11:27:46.073769026 +0100
+++ wxPython-src-2.8.12.0/wxPython/src/gtk/xrc_wrap.cpp 2014-03-14 11:30:55.391036827 +0100
@@ -861,7 +861,7 @@ SWIG_Python_AddErrorMsg(const char* mesg
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
