--- ogre-1.6.1.dfsg1/OgreMain/src/nedmalloc/malloc.c.h.orig	2009-01-05 12:07:43.000000000 +0000
+++ ogre-1.6.1.dfsg1/OgreMain/src/nedmalloc/malloc.c.h	2009-05-15 18:45:23.000000000 +0100
@@ -1839,7 +1839,7 @@
 /* skipped internal declaration from pthread.h */
 #ifdef linux
 #ifndef PTHREAD_MUTEX_RECURSIVE
-extern int pthread_mutexattr_setkind_np __P ((pthread_mutexattr_t *__attr,
+extern "C" int pthread_mutexattr_setkind_np __P ((pthread_mutexattr_t *__attr,
 					   int __kind));
 #define PTHREAD_MUTEX_RECURSIVE PTHREAD_MUTEX_RECURSIVE_NP
 #define pthread_mutexattr_settype(x,y) pthread_mutexattr_setkind_np(x,y)
