--- CMakeLists.orig.txt	2010-03-23 15:13:30.838325329 +0200
+++ CMakeLists.txt	2010-03-23 15:13:03.968450007 +0200
@@ -325,7 +325,7 @@
 
 # Set the XDG mime stuff
 SET(MIMEDIR "share/mime/packages/")
-SET(MIMELNKDIR "share/mimelnk/application/")
+SET(MIMELNKDIR "share/applications/")
 
 ##############################################################################################################
 ########## Build Setup                                                                              ##########
