--- src/main/subtitlecomposer.desktop~	2007-12-16 04:40:25.000000000 +0200
+++ src/main/subtitlecomposer.desktop	2008-01-30 00:55:55.000000000 +0200
@@ -1,11 +1,14 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=Subtitle Composer
+Name[tr]=Altyazı Düzenleyici
+GenericName=Subtitle Composer
+GenericName[tr]=Altyazı Düzenleyici
 Exec=subtitlecomposer %f -caption "%c"
 MimeType=subtitle/x-subrip;subtitle/x-microdvd;subtitle/x-mplayer;subtitle/x-substationalpha;subtitle/x-advsubstationalpha;
 Icon=subtitlecomposer
 Type=Application
 DocPath=subtitlecomposer/subtitlecomposer.html
 Comment=KDE subtitles editor
+Comment[tr]=KDE altyazı düzenleyici
 Terminal=false
 Categories=Qt;KDE;Application;AudioVideo;
