diff -Nur nexuiz-2.5.2-old/darkplaces/fs.c nexuiz-2.5.2/darkplaces/fs.c
--- nexuiz-2.5.2/darkplaces/fs.c	2009-10-06 16:28:51.973969396 +0300
+++ nexuiz-2.5.2/darkplaces/fs.c	2009-10-06 16:32:52.566183285 +0300
@@ -1210,10 +1210,10 @@
 	FS_ClearSearchPath();
 
 	// add the game-specific paths
-	// gamedirname1 (typically id1)
-	FS_AddGameHierarchy (gamedirname1);
+	// "nexuiz" (typically id1)
+	FS_AddGameHierarchy ("nexuiz");
 	// update the com_modname (used for server info)
-	strlcpy(com_modname, gamedirname1, sizeof(com_modname));
+	strlcpy(com_modname, "nexuiz", sizeof(com_modname));
 
 	// add the game-specific path, if any
 	// (only used for mission packs and the like, which should set fs_modified)
@@ -1243,7 +1243,7 @@
 
 	// set the default screenshot name to either the mod name or the
 	// gamemode screenshot name
-	if (strcmp(com_modname, gamedirname1))
+	if (strcmp(com_modname, "nexuiz"))
 		Cvar_SetQuick (&scr_screenshot_name, com_modname);
 	else
 		Cvar_SetQuick (&scr_screenshot_name, gamescreenshotname);
@@ -1599,7 +1599,7 @@
 #endif
 	}
 
-	strlcpy(fs_gamedir, "", sizeof(fs_gamedir));
+	strlcpy(fs_gamedir, "/usr/share/quake1", sizeof(fs_gamedir));
 
 // If the base directory is explicitly defined by the compilation process
 #ifdef DP_FS_BASEDIR
@@ -1644,7 +1644,7 @@
 
 	p = FS_CheckGameDir(gamedirname1);
 	if(!p || p == fs_checkgamedir_missing)
-		Con_Printf("WARNING: base gamedir %s%s/ not found!\n", fs_basedir, gamedirname1);
+		Con_Printf("WARNING: base gamedir %s%s/ not found!\n", fs_basedir, "nexuiz");
 
 	if(gamedirname2)
 	{
