diff -Nur nexuiz-2.5.1-old/darkplaces/makefile.inc nexuiz-2.5.1/darkplaces/makefile.inc
--- nexuiz-2.5.2/darkplaces/makefile.inc	2009-06-01 10:16:30.000000000 +0300
+++ nexuiz-2.5.2/darkplaces/makefile.inc	2009-06-01 10:18:59.000000000 +0300
@@ -3,7 +3,7 @@
 CHECKLEVEL2 = @if [ "$(LEVEL)" != 2 ]; then $(MAKE) help; false; fi
 
 # Choose the compiler you want to use
-CC?=gcc
+CC?=pardusCC
 
 # athlon optimizations
 #CPUOPTIMIZATIONS?=-march=athlon
@@ -18,7 +18,7 @@
 # 686 (Pentium Pro/II) optimizations
 #CPUOPTIMIZATIONS?=-march=i686
 # No specific CPU (386 compatible)
-CPUOPTIMIZATIONS?=
+CPUOPTIMIZATIONS?=pardusCFLAGS
 # Experimental
 #CPUOPTIMIZATIONS?=-fno-math-errno -ffinite-math-only -fno-rounding-math -fno-signaling-nans -fassociative-math -freciprocal-math -fno-signed-zeros -fno-trapping-math
 # NOTE: *never* *ever* use the -ffast-math or -funsafe-math-optimizations flag
@@ -51,7 +51,7 @@
 
 # Advanced Linux Sound Architecture (Linux)
 OBJ_SND_ALSA=$(OBJ_SND_COMMON) snd_alsa.o
-LIB_SND_ALSA=-lasound $(LIB_SND_MODPLUG)
+LIB_SND_ALSA=`pkg-config --libs alsa` $(LIB_SND_MODPLUG)
 
 # Core Audio (Mac OS X)
 OBJ_SND_COREAUDIO=$(OBJ_SND_COMMON) snd_coreaudio.o
@@ -190,7 +190,7 @@
 
 OBJ_GLX= builddate.c sys_linux.o vid_glx.o $(OBJ_SOUND) $(OBJ_CD) $(OBJ_COMMON)
 
-LDFLAGS_UNIXCOMMON=-lm
+LDFLAGS_UNIXCOMMON=pardusLDFLAGS -lm
 LDFLAGS_UNIXCL=-L$(UNIX_X11LIBPATH) -lX11 -lXpm -lXext -lXxf86dga -lXxf86vm $(LIB_SOUND)
 LDFLAGS_UNIXCL_PRELOAD=-lz -ljpeg -lpng -logg -ltheora -lvorbis -lvorbisenc -lvorbisfile -lcurl -lmodplug
 LDFLAGS_UNIXSV_PRELOAD=-lz -lcurl
@@ -416,7 +416,7 @@
 		DP_MAKE_TARGET=$(DP_MAKE_TARGET) DP_SOUND_API=$(DP_SOUND_API) \
 		CFLAGS="$(CFLAGS_COMMON) $(CFLAGS_EXTRA) $(CFLAGS_RELEASE) $(OPTIM_RELEASE)"\
 		LDFLAGS="$(LDFLAGS_RELEASE) $(LDFLAGS_COMMON)" LEVEL=2
-	$(STRIP) $(EXE)
+	# $(STRIP) $(EXE)
 
 snd_modplug.o: snd_modplug.c
 	$(CHECKLEVEL2)
