diff -ur kdelibs-4.11.3/kinit/klauncher_main.cpp kdelibs-4.11.3-klauncher-no-glib/kinit/klauncher_main.cpp
--- kdelibs-4.11.3/kinit/klauncher_main.cpp	2013-06-28 19:03:41.000000000 +0200
+++ kdelibs-4.11.3-klauncher-no-glib/kinit/klauncher_main.cpp	2013-12-09 00:32:12.000000000 +0100
@@ -75,10 +75,21 @@
    // WABA: Make sure not to enable session management.
    putenv(strdup("SESSION_MANAGER="));
 
+   // Disable the GLib event loop (rh#983110)
+   bool wasQtNoGlibSet = (getenv("QT_NO_GLIB") != 0);
+   if (!wasQtNoGlibSet) {
+      setenv("QT_NO_GLIB", "1", true);
+   }
+
    // We need a QCoreApplication to get a DBus event loop
    QCoreApplication app(argc, argv);
    app.setApplicationName( componentData.componentName() );
 
+   // Now get rid of QT_NO_GLIB again so launched processes don't inherit it
+   if (!wasQtNoGlibSet) {
+      unsetenv("QT_NO_GLIB");
+   }
+
    int maxTry = 3;
    while(true)
    {
