diff -up kdelibs-4.9.2/cmake/modules/FindSamba.cmake.samba4 kdelibs-4.9.2/cmake/modules/FindSamba.cmake
--- kdelibs-4.9.2/cmake/modules/FindSamba.cmake.samba4	2012-09-27 19:09:04.000000000 -0500
+++ kdelibs-4.9.2/cmake/modules/FindSamba.cmake	2012-10-02 07:11:38.137381941 -0500
@@ -17,10 +17,22 @@ if(SAMBA_INCLUDE_DIR AND SAMBA_LIBRARIES
     set(Samba_FIND_QUIETLY TRUE)
 endif(SAMBA_INCLUDE_DIR AND SAMBA_LIBRARIES)
 
-find_path(SAMBA_INCLUDE_DIR NAMES libsmbclient.h )
+IF (NOT WIN32)
+   # use pkg-config to get the directories and then use these values
+   # in the FIND_PATH() and FIND_LIBRARY() calls
+   find_package(PkgConfig)
+   pkg_check_modules(PC_LIBSMBCLIENT QUIET smbclient)
+ENDIF (NOT WIN32)
 
-find_library(SAMBA_LIBRARIES NAMES smbclient )
+find_path(SAMBA_INCLUDE_DIR NAMES libsmbclient.h 
+          PATHS
+          ${PC_LIBSMBCLIENT_INCLUDEDIR}
+          ${PC_LIBSMBCLIENT_INCLUDE_DIRS})
 
+find_library(SAMBA_LIBRARIES NAMES smbclient 
+             PATHS
+             ${PC_LIBSMBCLIENT_LIBDIR}
+             ${PC_LIBSMBCLIENT_LIB_DIRS})
 
 if(SAMBA_INCLUDE_DIR AND SAMBA_LIBRARIES)
    set(SAMBA_FOUND TRUE)
