diff -up kdelibs-4.10.0/solid/solid/managerbase.cpp.SOLID_UPNP kdelibs-4.10.0/solid/solid/managerbase.cpp
--- kdelibs-4.10.0/solid/solid/managerbase.cpp.SOLID_UPNP	2013-01-23 15:44:27.000000000 -0600
+++ kdelibs-4.10.0/solid/solid/managerbase.cpp	2013-01-31 07:48:05.058342162 -0600
@@ -98,7 +98,11 @@ void Solid::ManagerBasePrivate::loadBack
 #        endif
 
 #        if defined (HUPNP_FOUND)
-            m_backends << new Solid::Backends::UPnP::UPnPDeviceManager(0);
+            bool solidUpnpEnabled
+                = QString::fromLocal8Bit(qgetenv("SOLID_UPNP")).toInt()==1;
+            if (solidUpnpEnabled) {
+                m_backends << new Solid::Backends::UPnP::UPnPDeviceManager(0);
+            }
 #        endif
     }
 }
