diff -up kdelibs-4.9.90/cmake/modules/FindSamba.cmake.FindSamba-samba4 kdelibs-4.9.90/cmake/modules/FindSamba.cmake
--- kdelibs-4.9.90/cmake/modules/FindSamba.cmake.FindSamba-samba4	2012-11-14 09:58:29.000000000 -0600
+++ kdelibs-4.9.90/cmake/modules/FindSamba.cmake	2012-12-03 07:29:54.103057454 -0600
@@ -17,10 +17,20 @@ if(SAMBA_INCLUDE_DIR AND SAMBA_LIBRARIES
     set(Samba_FIND_QUIETLY TRUE)
 endif(SAMBA_INCLUDE_DIR AND SAMBA_LIBRARIES)
 
-find_path(SAMBA_INCLUDE_DIR NAMES libsmbclient.h )
+# use pkg-config to get the directories and then use these values
+# in the FIND_PATH() and FIND_LIBRARY() calls
+find_package(PkgConfig)
+pkg_check_modules(PC_LIBSMBCLIENT QUIET smbclient)
 
-find_library(SAMBA_LIBRARIES NAMES smbclient )
+find_path(SAMBA_INCLUDE_DIR NAMES libsmbclient.h 
+          HINTS 
+          ${PC_LIBSMBCLIENT_INCLUDEDIR}
+          ${PC_LIBSMBCLIENT_INCLUDE_DIRS})
 
+find_library(SAMBA_LIBRARIES NAMES smbclient 
+             HINTS 
+             ${PC_LIBSMBCLIENT_LIBDIR}
+             ${PC_LIBSMBCLIENT_LIB_DIRS})
 
 if(SAMBA_INCLUDE_DIR AND SAMBA_LIBRARIES)
    set(SAMBA_FOUND TRUE)
