diff -up kdelibs-4.9.3/kio/kssl/kcm/cacertificatespage.cpp.orig kdelibs-4.9.3/kio/kssl/kcm/cacertificatespage.cpp
--- kdelibs-4.9.3/kio/kssl/kcm/cacertificatespage.cpp.orig	2012-11-29 15:37:07.458858688 +0100
+++ kdelibs-4.9.3/kio/kssl/kcm/cacertificatespage.cpp	2012-11-29 13:52:05.243926802 +0100
@@ -291,7 +291,7 @@ void CaCertificatesPage::removeSelection
 void CaCertificatesPage::addCertificateClicked()
 {
     QStringList certFiles
-      = KFileDialog::getOpenFileNames(KUrl(), QLatin1String("application/x-x509-ca-cert"),
+      = KFileDialog::getOpenFileNames(KUrl(), QLatin1String("*.pem *.cert *.crt *.der"),
                                       this, i18n("Pick Certificates"));
 
     QList<QSslCertificate> certs;
