diff -up kdelibs-4.10.0/cmake/modules/FindKDE4Internal.cmake.xxcmake kdelibs-4.10.0/cmake/modules/FindKDE4Internal.cmake
--- kdelibs-4.10.0/cmake/modules/FindKDE4Internal.cmake.xxcmake	2013-01-31 07:45:31.958256176 -0600
+++ kdelibs-4.10.0/cmake/modules/FindKDE4Internal.cmake	2013-01-31 07:45:31.989255789 -0600
@@ -955,6 +955,7 @@ endif(WIN32)
 # CMake generators if no build type is set.
 if (NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
    set(CMAKE_BUILD_TYPE RelWithDebInfo)
+   set(CMAKE_CONFIGURATION_TYPES ${CMAKE_BUILD_TYPE})
 endif (NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
 
 
