diff -up kdelibs-4.10.0/kio/kio/kprotocolmanager.cpp.branding kdelibs-4.10.0/kio/kio/kprotocolmanager.cpp
--- kdelibs-4.10.0/kio/kio/kprotocolmanager.cpp.branding	2013-01-23 15:44:24.000000000 -0600
+++ kdelibs-4.10.0/kio/kio/kprotocolmanager.cpp	2013-01-31 07:41:09.248540500 -0600
@@ -743,6 +743,7 @@ QString KProtocolManager::defaultUserAge
     d->useragent += QString::number(KDE::versionMajor());
     d->useragent += QL1C('.');
     d->useragent += QString::number(KDE::versionMinor());
+    d->useragent += QL1S(" Fedora/@@VERSION_RELEASE@@"); 
   }
   else
   {
