diff -up kdelibs-4.10.0/kdecore/kernel/kstandarddirs.cpp.kstandarddirs kdelibs-4.10.0/kdecore/kernel/kstandarddirs.cpp
--- kdelibs-4.10.0/kdecore/kernel/kstandarddirs.cpp.kstandarddirs	2013-01-31 07:44:58.336676504 -0600
+++ kdelibs-4.10.0/kdecore/kernel/kstandarddirs.cpp	2013-01-31 07:44:58.340676454 -0600
@@ -1153,7 +1153,9 @@ QStringList KStandardDirs::KStandardDirs
                  pit != prefixList->end();
                  ++pit)
             {
-            if((*pit)!=installprefix||installdir.isEmpty())
+            // "exe" never has a custom install path, and the check triggers
+            // a false positive due to the libexecdir patch
+            if((*pit)!=installprefix||installdir.isEmpty()||!strcmp("exe", type))
             {
                     for (QStringList::ConstIterator it = dirs.constBegin();
                          it != dirs.constEnd(); ++it)
@@ -1171,6 +1173,11 @@ QStringList KStandardDirs::KStandardDirs
                         if ((local || testdir.exists()) && !candidates.contains(path))
                             candidates.append(path);
                     }
+                    // special-case "config" (forward porting Chris Cheney's
+                    // hack) - we want /etc/kde after the local config paths
+                    // and before the ones in /usr (including kde-profile)
+                    if (local && !strcmp("config", type))
+                       candidates.append(QLatin1String("/etc/kde/"));
                     local = false;
                 }
             else
