diff -up kdelibs-4.3.90/doc/common/CMakeLists.txt.all-css kdelibs-4.3.90/doc/common/CMakeLists.txt
--- kdelibs-4.3.90/doc/common/CMakeLists.txt.all-css	2010-01-06 10:58:53.000000000 -0600
+++ kdelibs-4.3.90/doc/common/CMakeLists.txt	2010-01-06 13:33:35.920884922 -0600
@@ -8,6 +8,8 @@ install(FILES
     kde-docs.css
     doxygen.css
     tabs.css
+    kde.css flat.css print.css
+    Doxyfile.global
     header.html
     footer.html
     mainheader.html
