diff -up kdelibs-4.10.0/kdeui/icons/kicontheme.cpp.kde149705 kdelibs-4.10.0/kdeui/icons/kicontheme.cpp
--- kdelibs-4.10.0/kdeui/icons/kicontheme.cpp.kde149705	2013-01-23 15:44:19.000000000 -0600
+++ kdelibs-4.10.0/kdeui/icons/kicontheme.cpp	2013-01-31 07:40:19.163166653 -0600
@@ -527,7 +527,7 @@ QString KIconTheme::current()
     }
 
     KConfigGroup cg(KGlobal::config(), "Icons");
-    *_theme = cg.readEntry("Theme", defaultThemeName());
+    *_theme = cg.readEntry("Theme4", cg.readEntry("Theme", defaultThemeName()));
     if ( *_theme == QLatin1String("hicolor") ) {
         *_theme = defaultThemeName();
     }
diff -up kdelibs-4.10.0/kdeui/kernel/kglobalsettings.cpp.kde149705 kdelibs-4.10.0/kdeui/kernel/kglobalsettings.cpp
--- kdelibs-4.10.0/kdeui/kernel/kglobalsettings.cpp.kde149705	2013-01-23 15:44:19.000000000 -0600
+++ kdelibs-4.10.0/kdeui/kernel/kglobalsettings.cpp	2013-01-31 07:40:19.163166653 -0600
@@ -942,7 +942,7 @@ void KGlobalSettings::Private::applyGUIS
     if (kde_overrideStyle.isEmpty()) {
         const QString &defaultStyle = KStyle::defaultStyle();
         const KConfigGroup pConfig(KGlobal::config(), "General");
-        const QString &styleStr = pConfig.readEntry("widgetStyle", defaultStyle);
+        const QString &styleStr = pConfig.readEntry("widgetStyle4", pConfig.readEntry("widgetStyle", defaultStyle));
 
         if (styleStr.isEmpty() ||
                 // check whether we already use the correct style to return then
diff -up kdelibs-4.10.0/kutils/kdeglobals.kcfg.kde149705 kdelibs-4.10.0/kutils/kdeglobals.kcfg
--- kdelibs-4.10.0/kutils/kdeglobals.kcfg.kde149705	2013-01-23 15:44:26.000000000 -0600
+++ kdelibs-4.10.0/kutils/kdeglobals.kcfg	2013-01-31 07:40:19.164166641 -0600
@@ -24,6 +24,10 @@
    <whatsthis>The name of the widget style, for example &quot;keramik&quot; or &quot;plastik&quot;. Without quotes.</whatsthis>
    <default>keramik</default>
   </entry>
+  <entry key="widgetStyle4" type="String" >
+   <label>Widget style to use for KDE 4</label>
+   <whatsthis>The name of the widget style, for example &quot;oxygen&quot;. Without quotes. Defaults to widgetStyle.</whatsthis>
+  </entry>
   <entry key="windowBackground" type="String" />
   <entry key="windowForeground" type="String" />
   <entry key="UseSystemBell" type="Bool" >
