diff -ur pygtksourceview-2.6.0.orig/configure.ac pygtksourceview-2.6.0/configure.ac
--- pygtksourceview-2.6.0.orig/configure.ac	2009-03-21 22:06:40.000000000 +0200
+++ pygtksourceview-2.6.0/configure.ac	2010-09-01 20:44:29.047624299 +0300
@@ -124,6 +124,9 @@
 fi
 AC_SUBST(CODEGEN)
 
+MODULE_LIBS=`python-config --libs`
+AC_SUBST(MODULE_LIBS)
+
 JH_ADD_CFLAG([-Wall])
 JH_ADD_CFLAG([-fno-strict-aliasing])
 
diff -ur pygtksourceview-2.6.0.orig/Makefile.am pygtksourceview-2.6.0/Makefile.am
--- pygtksourceview-2.6.0.orig/Makefile.am	2008-11-10 23:57:58.000000000 +0200
+++ pygtksourceview-2.6.0/Makefile.am	2010-09-01 20:45:20.793652643 +0300
@@ -27,7 +27,7 @@
 
 # pygtksourceview2 module
 pyexec_LTLIBRARIES = gtksourceview2.la
-gtksourceview2_la_LDFLAGS = -module -avoid-version -export-symbols-regex initgtksourceview2 $(no_undefined) $(pyd_ext)
+gtksourceview2_la_LDFLAGS = -module -avoid-version -export-symbols-regex initgtksourceview2 $(no_undefined) $(pyd_ext) $(MODULE_LIBS)
 gtksourceview2_la_CFLAGS = $(PYGTKSOURCEVIEW_CFLAGS)
 gtksourceview2_la_LIBADD = $(PYGTKSOURCEVIEW_LIBS) $(PYTHON_LIBS)
 gtksourceview2_la_SOURCES = gtksourceview2module.c 
