--- octave-3.4.2/liboctave/mach-info.cc.orig	2011-07-14 15:57:43.573464558 -0300
+++ octave-3.4.2/liboctave/mach-info.cc	2011-07-14 15:59:01.247897894 -0300
@@ -124,7 +124,7 @@ get_float_format (void)
   int i = 0;
   do
     {
-      if (equiv_compare (fp[i].fp_par, mach_fp_par, 4))
+      if (equiv_compare (fp[i].fp_par, mach_fp_par, 2))
         {
           retval = fp[i].fp_fmt;
           break;
