--- octave-3.4.2/configure.ac.ori      2011-06-27 17:36:26.000000000 +0000
+++ octave-3.4.2/configure.ac  2011-06-27 17:36:55.000000000 +0000
@@ -312,7 +312,7 @@ AC_SUBST(DEPEND_EXTRA_SED_PATTERN)
 AX_PTHREAD
 ### Include pthread libs and flags here in case other tests need them.
 ### They seem to be required for the OpenGL tests on Debian systems.
-LIBS="$PTHREAD_LIBS $LIBS"
+LIBS="$PTHREAD_LIBS $LIBS -lgfortran -llapack -lGLU -lGL -lpthread"
 CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
 
 AC_PATH_X
