diff -up joe-3.7/rc/joerc.in.joerc joe-3.7/rc/joerc.in
--- joe-3.7/rc/joerc.in.joerc	2008-11-02 04:29:43.000000000 +0100
+++ joe-3.7/rc/joerc.in	2009-02-03 12:26:34.000000000 +0100
@@ -718,6 +718,8 @@ msg		^[ ^H		Display a message
 nextw		^K N		Goto next window
 nextw		^K ^N
 nextw		^K n
+nextw		^[ [ 1 ; 3 C	alt right in (newer) xterm
+nextw		^[ [ 3 C	alt right in gnome-terminal
 pgdn		.kN		Screen down
 pgdn		^V
 pgdn		^[ [ 6 ~
@@ -730,6 +732,8 @@ play		^K 0 TO 9	Execute macro
 prevw		^K P		Goto previous window
 prevw		^K ^P
 prevw		^K p
+prevw		^[ [ 1 ; 3 D	alt left in (newer) xterm
+prevw		^[ [ 3 D	alt left in gnome-terminal
 query		^K ?		Macro query insert
 record		^K [		Record a macro
 retype		^R		Refresh screen
@@ -763,8 +767,8 @@ if,"char==65",then,"it's an A",else,"it'
 :def ispellword psh,nextword,markk,prevword,markb,filt,"cat >ispell.tmp;ispell ispell.tmp </dev/tty >/dev/tty;tr -d <ispell.tmp '\\012';/bin/rm ispell.tmp",rtn,retype,nextword
 
  Aspell
-:def aspellfile filt,"SPLTMP=ispell.tmp;cat >$SPLTMP;aspell --lang=",language,".",charset," -x -c $SPLTMP </dev/tty >/dev/tty;cat $SPLTMP;/bin/rm $SPLTMP",rtn,retype
-:def aspellword psh,nextword,markk,prevword,markb,filt,"SPLTMP=ispell.tmp;cat >$SPLTMP;aspell --lang=",language,".",charset," -x -c $SPLTMP </dev/tty >/dev/tty;tr -d <$SPLTMP '\\012';/bin/rm $SPLTMP",rtn,retype,nextword
+:def aspellfile filt,"SPLTMP=`mktemp -t joespell.XXXXXXXXXX`;cat >$SPLTMP;aspell --lang=",language,".",charset," -x -c $SPLTMP </dev/tty >/dev/tty;cat $SPLTMP;/bin/rm $SPLTMP",rtn,retype
+:def aspellword psh,nextword,markk,prevword,markb,filt,"SPLTMP=`mktemp -t joespell.XXXXXXXXXX`;cat >$SPLTMP;aspell --lang=",language,".",charset," -x -c $SPLTMP </dev/tty >/dev/tty;tr -d <$SPLTMP '\\012';/bin/rm $SPLTMP",rtn,retype,nextword
 
 
 aspellfile	^[ l
@@ -1015,9 +1019,13 @@ markk		^K k
 menu,"root",rtn	^T
 nextpos		^K =		Goto next position in position history
 nextword	^X		Goto next word
+nextword	^[ [ 1 ; 5 C	ctrl right in (newer) xterm
+nextword	^[ [ 5 C	ctrl right in gnome-terminal
 open		^]		Split line
 prevpos		^K -		Previous position in history
 prevword	^Z		Previous word
+prevword	^[ [ 1 ; 5 D	ctrl left in (newer) xterm
+prevword	^[ [ 5 D	ctrl left in gnome-terminal
 reload		^K `		Revert file
 redo		^^		Redo changes
 rindent		^K .		Indent to right
