commit da4d42fd7c0c444874bd27488f4609f72bf9d1e7
Author: Cole Robinson <crobinso@redhat.com>
Date:   Tue Apr 5 11:20:53 2011 -0400

    Guest: Don't parse spice <channel> XML as a <channel> device

diff --git a/virtinst/Guest.py b/virtinst/Guest.py
index 4c0ef0b..02937b8 100644
--- a/virtinst/Guest.py
+++ b/virtinst/Guest.py
@@ -790,7 +790,8 @@ class Guest(XMLBuilderDomain.XMLBuilderDomain):
             if node.name != "devices":
                 continue
 
-            children = filter(lambda x: x.name in device_mappings,
+            children = filter(lambda x: (x.name in device_mappings and
+                                         x.parent == node),
                               node.children)
             for devnode in children:
                 objclass = device_mappings.get(devnode.name)
