diff -Nur skey-1.1.5-old//skeyprune.pl skey-1.1.5/skeyprune.pl
--- skey-1.1.5-old//skeyprune.pl	2010-09-12 15:07:36.778000090 +0300
+++ skey-1.1.5/skeyprune.pl	2010-09-12 15:08:05.152998914 +0300
@@ -37,7 +37,7 @@
 
 while (<OLD>) {
     # Ignore commented out entries
-    if ( ! /^#[^\s#]+\s+(MD[0-9]+\s+)?[0-9]+\s+[A-z0-9_-]+\s+[a-f0-9]+\s+(Jan|Feb|Mar|Apr|May|Ju[nl]|Aug|Sep|Oct|Nov|Dec)\s+[0-9]+,\s*[0-9]+\s+[0-9]+:[0-9]+:[0-9]+$/ ) {
+    if ( ! /^#[^\s#]+\s+(MD[0-9]+\s+)?[0-9]+\s+[A-z0-9_-]+\s+[a-f0-9]+\s+(Jan|Feb|Mar|Apr|May|Ju[nl]|Aug|Sep|Oct|Nov|Dec)\s+[0-9]+,\s*[0-9]+\s+[0-9]+:[0-9]+:[0-9]+$/i ) {
 	/((Jan|Feb|Mar|Apr|May|Ju[nl]|Aug|Sep|Oct|Nov|Dec)\s+[0-9]+,\s*[0-9]+\s+[0-9]+:[0-9]+:[0-9]+)$/;
 
 	# Prune out old entries if asked to
