--- src/partitionmanager.desktop.OSmanOS	2011-05-11 10:11:32.000000000 +0300
+++ src/partitionmanager.desktop		2013-01-02 22:47:00.335717680 +0200
@@ -37,7 +37,7 @@
 Name[sr@latin]=KDE‑ov menadžer particija
 Name[sv]=KDE:s partitionshanterare
 Name[th]=เครื่องมือจัดการพาร์ทิชันของ KDE
-Name[tr]=KDE Partition Manager
+Name[tr]=KDE Disk Bölümü Yöneticisi
 Name[uk]=Керування розділами диска KDE
 Name[x-test]=xxKDE Partition Managerxx
 Name[zh_CN]=KDE 分区管理器
