Index: partitionmanager-1.0.3/src/kcm/kcm_partitionmanager.desktop
===================================================================
--- partitionmanager-1.0.3.orig/src/kcm/kcm_partitionmanager.desktop
+++ partitionmanager-1.0.3/src/kcm/kcm_partitionmanager.desktop
@@ -7,7 +7,7 @@ X-KDE-Library=kcm_partitionmanager
 X-KDE-RootOnly=true
 X-KDE-ServiceTypes=KCModule
 X-KDE-ParentApp=kcontrol
-X-KDE-System-Settings-Parent-Category=system
+X-KDE-System-Settings-Parent-Category=system-administration
 X-KDE-SubstituteUID=true
 
 Name=Partition Manager
@@ -31,6 +31,7 @@ Name[pt]=Gestor de Partições
 Name[pt_BR]=Gerenciador de partições
 Name[ro]=Gestionar de partiții
 Name[sv]=Partitionshanterare
+Name[tr]=Disk Bölümü Yöneticisi
 Name[uk]=Керування розділами диска
 Name[x-test]=xxPartition Managerxx
 Name[zh_TW]=分割區管理員
Index: partitionmanager-1.0.3/src/kpart/partitionmanagerpart.desktop
===================================================================
--- partitionmanager-1.0.3.orig/src/kpart/partitionmanagerpart.desktop
+++ partitionmanager-1.0.3/src/kpart/partitionmanagerpart.desktop
@@ -27,6 +27,7 @@ Name[pt_BR]=Gerenciador de partições d
 Name[ro]=Gestionarul de partiții KDE
 Name[ru]=Управление разделами KDE
 Name[sv]=KDE:s partitionshanterare
+Name[tr]=KDE Disk Bölümü Yöneticisi
 Name[uk]=Керування розділами диска KDE
 Name[x-test]=xxKDE Partition Managerxx
 Name[zh_CN]=KDE 分区管理器
