Index: partitionmanager-1.0.3/src/gui/listdevices.cpp
===================================================================
--- partitionmanager-1.0.3.orig/src/gui/listdevices.cpp
+++ partitionmanager-1.0.3/src/gui/listdevices.cpp
@@ -49,8 +49,8 @@ void ListDevices::updateDevices()
 
 	foreach(const Device* d, pmWidget().previewDevices())
 	{
-		const QString shortText = d->deviceNode() + " (" + Capacity(*d).toString() + ')';
-		const QString longText = d->deviceNode() + " (" + Capacity(*d).toString() + ", " + d->name() + ')';
+		const QString shortText = d->name() + " (" + Capacity(*d).toString() + ')';
+		const QString longText = d->deviceNode();
 		QListWidgetItem* item = new QListWidgetItem(DesktopIcon("drive-harddisk"), shortText);
 		item->setToolTip(longText);
 		item->setSizeHint(QSize(0, 32));
