diff -Naur zvbi-0.2.31/src/Makefile.am zvbi-0.2.31.oden/src/Makefile.am
--- zvbi-0.2.31/src/Makefile.am	2008-02-19 01:35:14.000000000 +0100
+++ zvbi-0.2.31.oden/src/Makefile.am	2008-11-08 13:11:22.000000000 +0100
@@ -119,6 +119,8 @@
 	proxy-client.c \
 	proxy-msg.c
 
+libzvbi_chains_la_LIBADD = -ldl
+
 if MAINTAINER_MODE
 
 hammgen: hammgen.c
diff -Naur zvbi-0.2.31/src/Makefile.in zvbi-0.2.31.oden/src/Makefile.in
--- zvbi-0.2.31/src/Makefile.in	2008-07-25 08:20:51.000000000 +0200
+++ zvbi-0.2.31.oden/src/Makefile.in	2008-11-08 13:11:07.000000000 +0100
@@ -62,7 +62,7 @@
 	"$(DESTDIR)$(libzvbiincludedir)"
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES) $(noinst_LTLIBRARIES)
-libzvbi_chains_la_LIBADD =
+libzvbi_chains_la_LIBADD = -ldl
 am_libzvbi_chains_la_OBJECTS = chains.lo io.lo misc.lo proxy-client.lo \
 	proxy-msg.lo
 libzvbi_chains_la_OBJECTS = $(am_libzvbi_chains_la_OBJECTS)
