diff -Nuar a/tools/wine.desktop b/tools/wine.desktop
--- a/tools/wine.desktop	2013-07-12 21:37:04.000000000 +0300
+++ b/tools/wine.desktop	2013-07-14 14:29:50.640704433 +0300
@@ -23,6 +23,7 @@
 Name[sr@latin]=Wine - dizač Windows programa
 Name[hr]=Wine - dizač Windows programa
 Name[he]=Wine — מריץ תכניות Windows
++Name[tr]=Wine Windows Programı Yükleyicisi
 Name[ja]=Wine Windowsプログラムローダー
 Exec=wine start /unix %f
 MimeType=application/x-ms-dos-executable;application/x-msi;application/x-ms-shortcut;
