--- wine-1.7.19.orig/configure.ac	2014-05-16 22:06:48.000000000 +0300
+++ wine-1.7.19/configure.ac	2014-05-29 14:46:49.482117199 +0300
@@ -1356,7 +1356,7 @@
 CURSES_LIBS=""
 if test "$ac_cv_header_ncurses_h" = "yes"
 then
-    WINE_CHECK_SONAME(ncurses,waddch,[CURSES_LIBS="-lncurses"])
+    WINE_CHECK_SONAME(ncurses,waddch,[CURSESLIBS="-lncurses"],,,[[libncursesw.*]])
 elif test "$ac_cv_header_curses_h" = "yes"
 then
     WINE_CHECK_SONAME(curses,waddch,[CURSES_LIBS="-lcurses"])
