diff -Nuar 0.5.1.orig//libbitdht/src/libbitdht.pro 0.5.1/libbitdht/src/libbitdht.pro
--- 0.5.1.orig//libbitdht/src/libbitdht.pro	2011-02-25 22:15:07.000000000 +0200
+++ 0.5.1/libbitdht/src/libbitdht.pro	2011-03-06 14:20:52.489000002 +0200
@@ -2,7 +2,7 @@
 CONFIG += staticlib
 CONFIG -= qt
 TARGET = bitdht
-QMAKE_CXXFLAGS *= -Wall -DBE_DEBUG
+QMAKE_CXXFLAGS *= -Wall -DBE_DEBUG -Wno-unused-parameter
 
 profiling {
 	QMAKE_CXXFLAGS -= -fomit-frame-pointer
diff -Nuar 0.5.1.orig//libretroshare/src/libretroshare.pro 0.5.1/libretroshare/src/libretroshare.pro
--- 0.5.1.orig//libretroshare/src/libretroshare.pro	2011-02-25 22:15:07.000000000 +0200
+++ 0.5.1/libretroshare/src/libretroshare.pro	2011-03-06 14:22:35.105000007 +0200
@@ -149,7 +149,7 @@
 	isEmpty(LIB_DIR) { LIB_DIR = $${PREFIX}/lib/ }
 
 	DESTDIR = lib
-	QMAKE_CXXFLAGS *= -Wall -D_FILE_OFFSET_BITS=64
+	QMAKE_CXXFLAGS *= -Wall -D_FILE_OFFSET_BITS=64 -Wno-unused-parameter
 	QMAKE_CC = g++
 
 	SSL_DIR = /usr/include/openssl
diff -Nuar 0.5.1.orig//retroshare-gui/src/RetroShare.pro 0.5.1/retroshare-gui/src/RetroShare.pro
--- 0.5.1.orig//retroshare-gui/src/RetroShare.pro	2011-02-25 22:15:07.000000000 +0200
+++ 0.5.1/retroshare-gui/src/RetroShare.pro	2011-03-06 14:23:52.813000008 +0200
@@ -27,7 +27,11 @@
 # Put lib dir in QMAKE_LFLAGS so it appears before -L/usr/lib
 linux-* {
 	#CONFIG += version_detail_bash_script
-	QMAKE_CXXFLAGS *= -D_FILE_OFFSET_BITS=64
+	QMAKE_CXXFLAGS *= -D_FILE_OFFSET_BITS=64 -Wno-unused-parameter
+
+	isEmpty(PREFIX) {
+		PREFIX = /usr
+	}
 
 	system(which gpgme-config >/dev/null 2>&1) {
 		INCLUDEPATH += $$system(gpgme-config --cflags | sed -e "s/-I//g")
@@ -41,6 +45,9 @@
 	LIBS += -lssl -lgpgme -lupnp -lXss -lgnome-keyring
 	DEFINES *= HAVE_XSS # for idle time, libx screensaver extensions
 	DEFINES *= UBUNTU
+	
+	target.path = $${PREFIX}/bin
+	INSTALLS *= target
 }
 
 linux-g++ {
diff -Nuar 0.5.1.orig//retroshare-nogui/src/retroshare-nogui.pro 0.5.1/retroshare-nogui/src/retroshare-nogui.pro
--- 0.5.1.orig//retroshare-nogui/src/retroshare-nogui.pro	2011-02-12 14:45:03.000000000 +0200
+++ 0.5.1/retroshare-nogui/src/retroshare-nogui.pro	2011-03-06 14:24:57.134000008 +0200
@@ -7,6 +7,10 @@
 	#CONFIG += version_detail_bash_script
 	QMAKE_CXXFLAGS *= -D_FILE_OFFSET_BITS=64
 
+	isEmpty(PREFIX) {
+		PREFIX = /usr
+	}
+	
 	system(which gpgme-config >/dev/null 2>&1) {
 		INCLUDEPATH += $$system(gpgme-config --cflags | sed -e "s/-I//g")
 	} else {
@@ -15,6 +19,9 @@
 
 	LIBS += ../../libretroshare/src/lib/libretroshare.a
 	LIBS += -lssl -lgpgme -lupnp -lgnome-keyring
+	
+	target.path = $${PREFIX}/bin
+	INSTALLS *= target
 }
 
 linux-g++ {
