--- gtkdatabox-0.9.1.1.orig/gtk/Makefile.am	2011-04-17 05:32:32.031350930 -0400
+++ gtkdatabox-0.9.1.1/gtk/Makefile.am	2011-04-17 06:41:44.391132493 -0400
@@ -16,8 +16,9 @@
 			gtkdatabox_ruler.c
 
 libgtkdatabox_la_LDFLAGS= \
-		        -version-info $(LT_CURRENT)\
-			-release $(LT_RELEASE)\
+			-version-info $(LT_CURRENT)\
+			-release $(LT_RELEASE)
+libgtkdatabox_la_LIBADD = -lm \
 			@GTK_LIBS@
 
 
