diff -up gtkdatabox-0.9.1.1/examples/addremove.c.gtk222 gtkdatabox-0.9.1.1/examples/addremove.c
--- gtkdatabox-0.9.1.1/examples/addremove.c.gtk222	2010-09-03 20:48:04.000000000 +0200
+++ gtkdatabox-0.9.1.1/examples/addremove.c	2010-09-03 20:48:14.000000000 +0200
@@ -203,7 +203,7 @@ create_addremove (void)
    g_signal_connect_swapped (GTK_OBJECT (close_button), "clicked",
 			     G_CALLBACK (gtk_main_quit), GTK_OBJECT (box));
    gtk_box_pack_start (GTK_BOX (box2), close_button, TRUE, TRUE, 0);
-   GTK_WIDGET_SET_FLAGS (close_button, GTK_CAN_DEFAULT);
+   gtk_widget_set_can_default(close_button, TRUE);
    gtk_widget_grab_default (close_button);
 
    gtk_widget_show_all (window);
diff -up gtkdatabox-0.9.1.1/examples/basics2.c.gtk222 gtkdatabox-0.9.1.1/examples/basics2.c
--- gtkdatabox-0.9.1.1/examples/basics2.c.gtk222	2009-04-19 23:40:55.000000000 +0200
+++ gtkdatabox-0.9.1.1/examples/basics2.c	2010-09-03 20:44:36.000000000 +0200
@@ -225,7 +225,7 @@ create_basics (void)
    g_signal_connect_swapped (GTK_OBJECT (close_button), "clicked",
 			     G_CALLBACK (gtk_main_quit), GTK_OBJECT (box));
    gtk_box_pack_start (GTK_BOX (vbox), close_button, FALSE, FALSE, 0);
-   GTK_WIDGET_SET_FLAGS (close_button, GTK_CAN_DEFAULT);
+   gtk_widget_set_can_default(close_button, TRUE);
    gtk_widget_grab_default (close_button);
    gtk_widget_grab_focus (close_button);
 
diff -up gtkdatabox-0.9.1.1/examples/basics.c.gtk222 gtkdatabox-0.9.1.1/examples/basics.c
--- gtkdatabox-0.9.1.1/examples/basics.c.gtk222	2009-04-17 23:15:02.000000000 +0200
+++ gtkdatabox-0.9.1.1/examples/basics.c	2010-09-03 20:44:36.000000000 +0200
@@ -113,7 +113,7 @@ create_basics (void)
    g_signal_connect_swapped (GTK_OBJECT (close_button), "clicked",
 			     G_CALLBACK (gtk_main_quit), GTK_OBJECT (box));
    gtk_box_pack_start (GTK_BOX (vbox), close_button, FALSE, FALSE, 0);
-   GTK_WIDGET_SET_FLAGS (close_button, GTK_CAN_DEFAULT);
+   gtk_widget_set_can_default(close_button, TRUE);
    gtk_widget_grab_default (close_button);
    gtk_widget_grab_focus (close_button);
 
diff -up gtkdatabox-0.9.1.1/examples/colors.c.gtk222 gtkdatabox-0.9.1.1/examples/colors.c
--- gtkdatabox-0.9.1.1/examples/colors.c.gtk222	2009-04-17 23:15:02.000000000 +0200
+++ gtkdatabox-0.9.1.1/examples/colors.c	2010-09-03 20:44:36.000000000 +0200
@@ -194,7 +194,7 @@ create_colors (void)
    g_signal_connect (G_OBJECT (close_button), "clicked",
 		     G_CALLBACK (gtk_main_quit), (gpointer) NULL);
    gtk_box_pack_start (GTK_BOX (box2), close_button, TRUE, TRUE, 0);
-   GTK_WIDGET_SET_FLAGS (close_button, GTK_CAN_DEFAULT);
+   gtk_widget_set_can_default(close_button, TRUE);
    gtk_widget_grab_default (close_button);
 
 
diff -up gtkdatabox-0.9.1.1/examples/enable_disable.c.gtk222 gtkdatabox-0.9.1.1/examples/enable_disable.c
--- gtkdatabox-0.9.1.1/examples/enable_disable.c.gtk222	2010-09-03 20:47:44.000000000 +0200
+++ gtkdatabox-0.9.1.1/examples/enable_disable.c	2010-09-03 20:47:54.000000000 +0200
@@ -145,7 +145,7 @@ create_show_hide (void)
    g_signal_connect_swapped (GTK_OBJECT (close_button), "clicked",
 			     G_CALLBACK (gtk_main_quit), GTK_OBJECT (box));
    gtk_box_pack_start (GTK_BOX (box2), close_button, TRUE, TRUE, 0);
-   GTK_WIDGET_SET_FLAGS (close_button, GTK_CAN_DEFAULT);
+   gtk_widget_set_can_default(close_button, TRUE);
    for (i = 0; i < noEnableSets; ++i)
    {
       GtkWidget *vbox = gtk_vbox_new (FALSE, 10);
diff -up gtkdatabox-0.9.1.1/examples/grid_array.c.gtk222 gtkdatabox-0.9.1.1/examples/grid_array.c
--- gtkdatabox-0.9.1.1/examples/grid_array.c.gtk222	2010-09-03 20:47:00.000000000 +0200
+++ gtkdatabox-0.9.1.1/examples/grid_array.c	2010-09-03 20:47:09.000000000 +0200
@@ -134,7 +134,7 @@ create_grid (void)
 			     G_CALLBACK (gtk_main_quit),
 			     GTK_OBJECT (box));
    gtk_box_pack_start (GTK_BOX (box2), close_button, TRUE, TRUE, 0);
-   GTK_WIDGET_SET_FLAGS (close_button, GTK_CAN_DEFAULT);
+   gtk_widget_set_can_default(close_button, TRUE);
    gtk_widget_grab_default (close_button);
 
    gtk_widget_show_all (window);
diff -up gtkdatabox-0.9.1.1/examples/grid.c.gtk222 gtkdatabox-0.9.1.1/examples/grid.c
--- gtkdatabox-0.9.1.1/examples/grid.c.gtk222	2010-09-03 20:46:41.000000000 +0200
+++ gtkdatabox-0.9.1.1/examples/grid.c	2010-09-03 20:46:49.000000000 +0200
@@ -133,7 +133,7 @@ create_grid (void)
 			     G_CALLBACK (gtk_main_quit),
 			     GTK_OBJECT (box));
    gtk_box_pack_start (GTK_BOX (box2), close_button, TRUE, TRUE, 0);
-   GTK_WIDGET_SET_FLAGS (close_button, GTK_CAN_DEFAULT);
+   gtk_widget_set_can_default(close_button, TRUE);
    gtk_widget_grab_default (close_button);
 
    gtk_widget_show_all (window);
diff -up gtkdatabox-0.9.1.1/examples/keycontrol.c.gtk222 gtkdatabox-0.9.1.1/examples/keycontrol.c
--- gtkdatabox-0.9.1.1/examples/keycontrol.c.gtk222	2009-04-17 23:15:02.000000000 +0200
+++ gtkdatabox-0.9.1.1/examples/keycontrol.c	2010-09-03 20:44:36.000000000 +0200
@@ -133,7 +133,8 @@ create_basics (void)
    gtk_databox_create_box_with_scrollbars_and_rulers (&box, &table,
 						      TRUE, TRUE, TRUE, TRUE);
    gtk_widget_add_events (box, GDK_KEY_PRESS_MASK);
-   GTK_WIDGET_SET_FLAGS (box, GTK_CAN_FOCUS | GTK_CAN_DEFAULT);
+   gtk_widget_set_can_default(box, TRUE);
+   gtk_widget_set_can_focus(box, TRUE);
 
    gtk_box_pack_start (GTK_BOX (box1), table, TRUE, TRUE, 0);
 
diff -up gtkdatabox-0.9.1.1/examples/lissajous.c.gtk222 gtkdatabox-0.9.1.1/examples/lissajous.c
--- gtkdatabox-0.9.1.1/examples/lissajous.c.gtk222	2010-09-03 20:47:22.000000000 +0200
+++ gtkdatabox-0.9.1.1/examples/lissajous.c	2010-09-03 20:47:32.000000000 +0200
@@ -154,7 +154,7 @@ create_lissajous (void)
 			     G_CALLBACK (gtk_main_quit), G_OBJECT (box));
 
    gtk_box_pack_start (GTK_BOX (box2), close_button, TRUE, TRUE, 0);
-   GTK_WIDGET_SET_FLAGS (close_button, GTK_CAN_DEFAULT);
+   gtk_widget_set_can_default(close_button, TRUE);
    gtk_widget_grab_default (close_button);
    lissajous_idle = g_idle_add ((GSourceFunc) lissajous_idle_func, box);
 
diff -up gtkdatabox-0.9.1.1/examples/logarithmic.c.gtk222 gtkdatabox-0.9.1.1/examples/logarithmic.c
--- gtkdatabox-0.9.1.1/examples/logarithmic.c.gtk222	2009-04-19 19:58:21.000000000 +0200
+++ gtkdatabox-0.9.1.1/examples/logarithmic.c	2010-09-03 20:44:36.000000000 +0200
@@ -207,7 +207,7 @@ create_logarithmic (void)
    g_signal_connect (GTK_OBJECT (close_button), "clicked",
 		     G_CALLBACK (gtk_main_quit), NULL);
    gtk_box_pack_start (GTK_BOX (vbox), close_button, FALSE, FALSE, 0);
-   GTK_WIDGET_SET_FLAGS (close_button, GTK_CAN_DEFAULT);
+   gtk_widget_set_can_default(close_button, TRUE);
    gtk_widget_grab_default (close_button);
    gtk_widget_grab_focus (close_button);
 
diff -up gtkdatabox-0.9.1.1/examples/markers.c.gtk222 gtkdatabox-0.9.1.1/examples/markers.c
--- gtkdatabox-0.9.1.1/examples/markers.c.gtk222	2010-09-03 20:45:36.000000000 +0200
+++ gtkdatabox-0.9.1.1/examples/markers.c	2010-09-03 20:46:02.000000000 +0200
@@ -220,7 +220,7 @@ create_markerss (void)
    g_signal_connect_swapped (GTK_OBJECT (close_button), "clicked",
 			     G_CALLBACK (gtk_main_quit), GTK_OBJECT (box));
    gtk_box_pack_start (GTK_BOX (box2), close_button, TRUE, TRUE, 0);
-   GTK_WIDGET_SET_FLAGS (close_button, GTK_CAN_DEFAULT);
+   gtk_widget_set_can_default(close_button, TRUE);
    gtk_widget_grab_default (close_button);
 
    gtk_widget_show_all (window);
diff -up gtkdatabox-0.9.1.1/examples/signals.c.gtk222 gtkdatabox-0.9.1.1/examples/signals.c
--- gtkdatabox-0.9.1.1/examples/signals.c.gtk222	2010-09-03 20:46:10.000000000 +0200
+++ gtkdatabox-0.9.1.1/examples/signals.c	2010-09-03 20:46:28.000000000 +0200
@@ -287,7 +287,7 @@ create_signals (void)
    g_signal_connect_swapped (G_OBJECT (close_button), "clicked",
 			     G_CALLBACK (gtk_main_quit), G_OBJECT (box));
    gtk_box_pack_start (GTK_BOX (box2), close_button, TRUE, TRUE, 0);
-   GTK_WIDGET_SET_FLAGS (close_button, GTK_CAN_DEFAULT);
+   gtk_widget_set_can_default(close_button, TRUE);
    gtk_widget_grab_default (close_button);
 
    g_signal_connect (G_OBJECT (box), "zoomed",
diff -up gtkdatabox-0.9.1.1/gtk/gtkdatabox.c.gtk222 gtkdatabox-0.9.1.1/gtk/gtkdatabox.c
--- gtkdatabox-0.9.1.1/gtk/gtkdatabox.c.gtk222	2010-09-03 20:44:36.000000000 +0200
+++ gtkdatabox-0.9.1.1/gtk/gtkdatabox.c	2010-09-03 20:44:36.000000000 +0200
@@ -587,7 +587,7 @@ gtk_databox_realize (GtkWidget * widget)
    gint attributes_mask;
 
    box = GTK_DATABOX (widget);
-   GTK_WIDGET_SET_FLAGS (box, GTK_REALIZED);
+   gtk_widget_set_realized(box, TRUE);
 
    attributes.window_type = GDK_WINDOW_CHILD;
    attributes.x = widget->allocation.x;
diff -up gtkdatabox-0.9.1.1/gtk/gtkdatabox_ruler.c.gtk222 gtkdatabox-0.9.1.1/gtk/gtkdatabox_ruler.c
--- gtkdatabox-0.9.1.1/gtk/gtkdatabox_ruler.c.gtk222	2010-09-03 20:44:36.000000000 +0200
+++ gtkdatabox-0.9.1.1/gtk/gtkdatabox_ruler.c	2010-09-03 20:44:36.000000000 +0200
@@ -842,7 +842,7 @@ gtk_databox_ruler_realize (GtkWidget * w
    gint attributes_mask;
 
    ruler = GTK_DATABOX_RULER (widget);
-   GTK_WIDGET_SET_FLAGS (ruler, GTK_REALIZED);
+   gtk_widget_set_realized(ruler, TRUE);
 
    attributes.window_type = GDK_WINDOW_CHILD;
    attributes.x = widget->allocation.x;
