Description: Fix pkg-config paths
Author: Sebastian Dröge <slomo@debian.org>
Forwarded: not-needed
--- notify-sharp-0.4.0~r2998/notify-sharp.pc.in.old	2008-03-23 00:22:07.000000000 +0100
+++ notify-sharp-0.4.0~r2998/notify-sharp.pc.in	2008-03-23 00:22:17.000000000 +0100
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=${prefix}
 libdir=${exec_prefix}/lib
-pkglibdir=${libdir}/mono/notify-sharp
+pkglibdir=${libdir}/cli/notify-sharp-0.4
 
 Name: notify-sharp
 Description: C# client library for notification-daemon
