Index: avogadro-1.0.3/avogadro/src/avogadro.desktop
===================================================================
--- avogadro-1.0.3.orig/avogadro/src/avogadro.desktop
+++ avogadro-1.0.3/avogadro/src/avogadro.desktop
@@ -2,6 +2,7 @@
 Version=1.0
 Name=Avogadro
 Comment=Advanced molecular editor
+Comment[tr]=Gelişmiş molekül düzenleyicisi
 Exec=avogadro %f
 Icon=avogadro-icon
 Terminal=false
