--- doc-build/omf.make.orig	2007-01-01 00:52:19.000000000 +0200
+++ doc-build/omf.make	2009-02-07 23:36:37.000000000 +0200
@@ -45,7 +45,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(DESTDIR)$(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)

 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
