--- mozilla/security/nss/lib/ssl/sslsock.c.transitional	2011-10-22 18:45:40.000000000 +0200
+++ mozilla/security/nss/lib/ssl/sslsock.c	2011-11-10 14:17:36.000000000 +0100
@@ -181,7 +181,7 @@
     PR_FALSE,   /* noLocks            */
     PR_FALSE,   /* enableSessionTickets */
     PR_FALSE,   /* enableDeflate      */
-    2,          /* enableRenegotiation (default: requires extension) */
+    3,          /* enableRenegotiation (default: requires extension) */
     PR_FALSE,   /* requireSafeNegotiation */
     PR_FALSE,   /* enableFalseStart   */
     PR_TRUE     /* cbcRandomIV        */
