Index: arptools-1.0.2/src/arpflood.c
===================================================================
--- arptools-1.0.2.orig/src/arpflood.c
+++ arptools-1.0.2/src/arpflood.c
@@ -51,7 +51,7 @@ char *hw_ntoa(struct libnet_ether_addr *
     static int i;
     char *str;
 
-    if ((str = malloc(6*2+5+1)) == NULL) {
+    if ((str = malloc(6*2+5+1+1)) == NULL) {
 	fprintf(stderr, "%s\n", strerror(errno));
 	exit(EXIT_FAILURE);
     }
