diff -Nur arptools-1.0.2-old/src/arpdiscover.c arptools-1.0.2/src/arpdiscover.c
--- arptools-1.0.2-old/src/arpdiscover.c	2009-10-21 16:06:54.078821072 +0300
+++ arptools-1.0.2/src/arpdiscover.c	2009-10-21 16:07:06.754571280 +0300
@@ -51,7 +51,7 @@
     static int i;
     char *str;
 
-    if ((str = malloc(6*2+5+1)) == NULL) {
+    if ((str = malloc(6*2+5+1+1)) == NULL) {
 	fprintf(stderr, "%s\n", strerror(errno));
 	exit(EXIT_FAILURE);
     }
