Index: omniORB-4.1.2/sample.cfg
===================================================================
--- omniORB-4.1.2.orig/sample.cfg
+++ omniORB-4.1.2/sample.cfg
@@ -338,6 +338,9 @@ abortOnNativeException = 0
 #
 #   The default for the set of initial references is empty.
 
+# resolve the omniNames running on localhost
+InitRef = NameService=corbaname::localhost
+
 ############################################################################
 # DefaultInitRef
 #
