diff -p -up omniORB-4.1.4/src/tool/omniidl/cxx/cccp/cccp.c.orig omniORB-4.1.4/src/tool/omniidl/cxx/cccp/cccp.c
--- omniORB-4.1.4/src/tool/omniidl/cxx/cccp/cccp.c.orig	2010-02-17 19:51:28.045065641 -0200
+++ omniORB-4.1.4/src/tool/omniidl/cxx/cccp/cccp.c	2010-02-17 19:52:22.278060504 -0200
@@ -8034,7 +8034,7 @@ macroexpand (hp, op)
       else
 	parse_error = macarg (NULL_PTR, 0);
       if (parse_error) {
-	error_with_line (line_for_error (start_line), parse_error);
+	error_with_line (line_for_error (start_line), "%s", parse_error);
 	break;
       }
       i++;
