--- amsn-0.98.4.orig/amsn-remote	2010-12-29 23:12:57.553000249 +0200
+++ amsn-0.98.4/amsn-remote	2010-12-29 23:32:25.263000241 +0200
@@ -1,9 +1,9 @@
-#!/usr/bin/wish
+#!/usr/bin/wish8.5
 ###
 ### Remote controller program
 ###
 
-source amsncore.tcl
+source /usr/share/amsn/amsncore.tcl
 
 set initialize_amsn 1
 

--- amsn-0.98.4.orig/amsn-remote-CLI	2010-12-29 23:12:57.515000253 +0200
+++ amsn-0.98.4/amsn-remote-CLI	2010-12-29 23:35:18.202000256 +0200
@@ -1,11 +1,11 @@
-#!/usr/bin/tclsh
+#!/usr/bin/wish8.5
 ###
 ### Remote controller program
 ###
 
 set initialize_amsn 1
 
-source amsncore.tcl
+source /usr/share/amsn/amsncore.tcl
 
 proc readsocket { } {
 	global sock
@@ -229,7 +229,7 @@
    }
 }
 
-source [file join $program_dir migmd5.tcl]
+source [file join $program_dir /usr/share/amsn/migmd5.tcl]
 
 set sockDS 0
 
