--- amsn-0.98.4.orig/login_screen.tcl	2010-12-30 12:31:23.793281007 +0200
+++ amsn-0.98.4/login_screen.tcl	2010-12-30 21:51:53.464151034 +0200
@@ -43,9 +43,6 @@
 	component forgot_pass_link
 	component new_account_link
 
-	component check_ver_icon
-	component check_ver_text
-
 	variable more_label
 
 	variable remember_me 0
@@ -88,7 +85,6 @@
 		contentmanager add group login_screen main checkboxes auto_login -orient horizontal	-widget $self	-pady 2
 		contentmanager add group login_screen main login		-orient horizontal	-widget $self	-align center	-pady 8
 		contentmanager add group login_screen main links		-orient vertical	-pady 32	-widget $self	-align left
-		contentmanager add group login_screen main check_ver		-orient horizontal	-pady 8		-widget $self	-align center
 		contentmanager add group login_screen main more			-orient horizontal	-pady 8		-widget $self	-align right
 
 		# Create widgets
@@ -156,9 +152,6 @@
 		set forgot_pass_link [$self create text 0 0 -anchor nw -text [trans forgot_pass]]
 		# New account
 		set new_account_link [$self create text 0 0 -anchor nw -text [trans new_account]]
-		# Check for newer amsn version
-		set check_version_icon [$self create image 0 0 -anchor nw -image [::skin::loadPixmap download]]
-		set check_version_text [$self create text 0 0 -anchor nw -text [trans checkver] -fill [::skin::getKey loginurlfg]]
 
 		set more_label [$self create text 0 0 -anchor nw -text [trans more]]
 
@@ -190,8 +183,6 @@
 		# Links
 		contentmanager add element login_screen main links forgot_pass -widget $self -tag $forgot_pass_link -pady 2
 		contentmanager add element login_screen main links new_account -widget $self -tag $new_account_link -pady 2
-		contentmanager add element login_screen main check_ver icon -widget $self -tag $check_version_icon -padx 4 -valign middle
-		contentmanager add element login_screen main check_ver text -widget $self -tag $check_version_text -padx 4 -valign middle
 		contentmanager add element login_screen main more label	-widget $self	-tag $more_label
 
 		# Set font for canvas all text items
@@ -236,7 +227,6 @@
 		$self CanvasTextToLink login_screen main more label "$self ShowMore"
 		$self CanvasTextToLink login_screen main links forgot_pass [list launch_browser "https://accountservices.passport.net/uiresetpw.srf?lc=1033"]
 		$self CanvasTextToLink login_screen main links new_account [list launch_browser "https://accountservices.passport.net/reg.srf?sl=1&lc=1033"]
-		$self CanvasTextToLink login_screen main check_ver text "::autoupdate::check_version"
 
 		$self FillLoginFields
 
@@ -450,7 +440,6 @@
  		contentmanager show login_screen main checkboxes
  		contentmanager show login_screen main login
  		contentmanager show login_screen main links
- 		contentmanager show login_screen main check_ver
  		contentmanager show login_screen main more
 
 		# We sort to get the sizes of each element..
@@ -465,7 +454,6 @@
  		contentmanager hide login_screen main checkboxes
  		contentmanager hide login_screen main login
  		contentmanager hide login_screen main links
- 		contentmanager hide login_screen main check_ver
  		contentmanager show login_screen main more
 
  		set max [winfo height $self]
@@ -507,12 +495,8 @@
 					if { $current < $max } {
 						contentmanager show login_screen lang
 
-						incr current [contentmanager height login_screen main check_ver]
-						incr current [contentmanager cget login_screen main check_ver -pady]
-						incr current [contentmanager cget login_screen main check_ver -pady]
 						
 						if { $current < $max } {
-							contentmanager show login_screen main check_ver
 
 							if { [::config::getKey show_login_screen_links 1] } {
 								incr current [contentmanager height login_screen main links]
@@ -560,7 +544,6 @@
  		contentmanager show login_screen main checkboxes
  		contentmanager show login_screen main login
  		contentmanager show login_screen main links
- 		contentmanager show login_screen main check_ver
  		contentmanager show login_screen main more
 
 		# We sort to get the sizes of each element..
