--- amsn-0.98.4.orig/config.tcl	2010-12-29 23:12:57.628000249 +0200
+++ amsn-0.98.4/config.tcl	2010-12-29 23:56:21.738000251 +0200
@@ -236,7 +236,7 @@
 		::config::setKey autoconnect 0			;#Automatically connect when amsn starts
 		::config::setKey libtls_temp ""			;#TLS
 		::config::setKey lineflushlog 1			;#Flush log files after each line
-		::config::setKey autocheckver 1			;#Automatically check for newer versions on startup
+		::config::setKey autocheckver 0			;#Automatically check for newer versions on startup
 		::config::setKey truncatenames 1		;#Truncate nicknames longer than window width in windows' title
 		::config::setKey truncatenicks 0		;#Truncate nicknames longer than window width in chat windows
 		::config::setKey keepalive 1				;#Keep alive connection (ping every minute)
@@ -247,7 +247,7 @@
 		::config::setKey start_ns_server "messenger.hotmail.com:1863"
 		::config::setKey default_gateway_server "gateway.messenger.hotmail.com"
 		::config::setKey start_gateway_server "gateway.messenger.hotmail.com"		
-		::config::setKey activeautoupdate 1		;#Active the auto update
+		::config::setKey activeautoupdate 0		;#Active the auto update
 		::config::setKey allowbadwords 1		;#Disable censure on nicks
 		::config::setKey enablebanner 1		;#Show or Hide AMSN Banner (By default Show)
 		::config::setKey startontray 0		;#Start amsn on tray icon only (hide contact list)
