--- amsn-0.98.4.orig/config.tcl	2010-12-30 12:31:23.793281007 +0200
+++ amsn-0.98.4/config.tcl	2010-12-30 12:47:51.468699167 +0200
@@ -160,8 +160,8 @@
 
 		::config::setKey encoding auto						;#ANSN encoding
 
-		::config::setKey textsize 2							;#User text size
-		::config::setKey mychatfont "{Helvetica} {} 000000"	;#User chat font
+		::config::setKey textsize 0							;#User text size
+		::config::setKey mychatfont "{DeJavu\ Sans} {} 000000"	;#User chat font
 		::config::setKey winchatsize "350x320"		;#Default chat window size
 		::config::setKey winchatoutheight "200"		;#Default chat window output height
 		::config::setKey savechatwinsize 1			;#Save chat window sizes when changed?
@@ -252,7 +252,7 @@
 		::config::setKey enablebanner 1		;#Show or Hide AMSN Banner (By default Show)
 		::config::setKey startontray 0		;#Start amsn on tray icon only (hide contact list)
 		::config::setKey storename 1			;#Store original nick in a variable when go to custom states to revert it when go back
-		::config::setKey strictfonts 0		;#Use strict fonts' size in _ALL_ AMSN's fonts (Disabled by default)
+		::config::setKey strictfonts 1		;#Use strict fonts' size in _ALL_ AMSN's fonts (Disabled by default)
 		if {[OnMaemo] } {
 			::config::setKey sngdblclick 1		;#Use single or double click to open a message window (0 double, 1 single)
 		} else {
@@ -589,7 +589,7 @@
 		} elseif { [OnWin] } {
 			setGlobalKey basefont [list Arial 10 normal]
 		} else {
-			setGlobalKey basefont [list Helvetica 11 normal]	;#AMSN base font
+			setGlobalKey basefont [list DeJavu\ Sans 10 normal]	;#AMSN base font
 		}
 		setGlobalKey dpi 0	;# The screen's DPI if we want to force it. 0 to let it use default
 
