--- amsn-0.98.4.orig/config.tcl	2010-12-29 23:12:57.628000249 +0200
+++ amsn-0.98.4/config.tcl	2010-12-30 01:09:55.986000261 +0200
@@ -579,7 +579,7 @@
 
 		setGlobalKey last_client_version ""
 		setGlobalKey language [detect_language "en"]	;#Default language
-		setGlobalKey skin "default"			;#AMSN skin
+		setGlobalKey skin "Oxygen"			;#AMSN skin
 		setGlobalKey disableprofiles 0 ;#Disable profiles (useful for cybercafes or similar)
 
