Index: prelink/ltconfig
===================================================================
--- prelink.orig/ltconfig
+++ prelink/ltconfig
@@ -660,7 +660,6 @@ no_builtin_flag=
 
 if test "$with_gcc" = yes; then
   wl='-Wl,'
-  link_static_flag='-static'
 
   case "$host_os" in
   beos* | irix5* | irix6* | osf3* | osf4* | osf5*)
@@ -950,21 +949,6 @@ if test -n "$special_shlib_compile_flags
   fi
 fi
 
-echo $ac_n "checking if $compiler static flag $link_static_flag works... $ac_c" 1>&6
-$rm conftest*
-echo 'main(){return(0);}' > conftest.c
-save_LDFLAGS="$LDFLAGS"
-LDFLAGS="$LDFLAGS $link_static_flag"
-echo "$progname:958: checking if $compiler static flag $link_static_flag works" >&5
-if { (eval echo $progname:959: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest; then
-  echo "$ac_t$link_static_flag" 1>&6
-else
-  echo "$ac_t"none 1>&6
-  link_static_flag=
-fi
-LDFLAGS="$save_LDFLAGS"
-$rm conftest*
-
 if test -z "$LN_S"; then
   # Check to see if we can use ln -s, or we need hard links.
   echo $ac_n "checking whether ln -s works... $ac_c" 1>&6
Index: prelink/src/Makefile.am
===================================================================
--- prelink.orig/src/Makefile.am
+++ prelink/src/Makefile.am
@@ -21,7 +21,7 @@ prelink_SOURCES = cache.c conflict.c cxx
 		  verify.c canonicalize.c md5.c md5.h sha.c sha.h 	     \
 		  $(common_SOURCES) $(arch_SOURCES)
 prelink_LDADD = @LIBGELF@
-prelink_LDFLAGS = -all-static
+prelink_LDFLAGS = -shared
 
 execstack_SOURCES = execstack.c $(common_SOURCES) $(arch_SOURCES)
 
