diff -Nur aalib-1.4.0-old/src/aados.c aalib-1.4.0/src/aados.c
--- aalib-1.4.0-old/src/aados.c	2008-03-25 03:27:18.805769361 +0200
+++ aalib-1.4.0/src/aados.c	2008-03-25 03:27:24.188950993 +0200
@@ -5,7 +5,7 @@
 #include <go32.h>
 #include <dpmi.h>
 #include <sys/nearptr.h>
-#include <malloc.h>
+#include <stdlib.h>
 #include "aalib.h"
 #include "aaint.h"
 int __use_nearptr_hack;
diff -Nur aalib-1.4.0-old/src/aadoskbd.c aalib-1.4.0/src/aadoskbd.c
--- aalib-1.4.0-old/src/aadoskbd.c	2008-03-25 03:27:18.805769361 +0200
+++ aalib-1.4.0/src/aadoskbd.c	2008-03-25 03:27:24.199949321 +0200
@@ -4,7 +4,7 @@
 #include <go32.h>
 #include <dpmi.h>
 #include <sys/nearptr.h>
-#include <malloc.h>
+#include <stdlib.h>
 #include <stdlib.h>
 #include <dos.h>
 #include <dpmi.h>
diff -Nur aalib-1.4.0-old/src/aaedit.c aalib-1.4.0/src/aaedit.c
--- aalib-1.4.0-old/src/aaedit.c	2008-03-25 03:27:18.806769209 +0200
+++ aalib-1.4.0/src/aaedit.c	2008-03-25 03:27:24.200949169 +0200
@@ -1,6 +1,6 @@
 #include <string.h>
 #include <ctype.h>
-#include <malloc.h>
+#include <stdlib.h>
 #include "aalib.h"
 #include "aaint.h"
 static void aa_editdisplay(struct aa_edit *e)
diff -Nur aalib-1.4.0-old/src/aakbdreg.c aalib-1.4.0/src/aakbdreg.c
--- aalib-1.4.0-old/src/aakbdreg.c	2008-03-25 03:27:18.806769209 +0200
+++ aalib-1.4.0/src/aakbdreg.c	2008-03-25 03:27:24.207948105 +0200
@@ -1,4 +1,4 @@
-#include <malloc.h>
+#include <stdlib.h>
 #include "config.h"
 #include "aalib.h"
 #include "aaint.h"
diff -Nur aalib-1.4.0-old/src/aalib.c aalib-1.4.0/src/aalib.c
--- aalib-1.4.0-old/src/aalib.c	2008-03-25 03:27:18.806769209 +0200
+++ aalib-1.4.0/src/aalib.c	2008-03-25 03:27:24.208947953 +0200
@@ -1,6 +1,6 @@
 #include <stdio.h>
 #include <string.h>
-#include <malloc.h>
+#include <stdlib.h>
 #include "aalib.h"
 #include "aaint.h"
 
diff -Nur aalib-1.4.0-old/src/aalinux.c aalib-1.4.0/src/aalinux.c
--- aalib-1.4.0-old/src/aalinux.c	2008-03-25 03:27:18.807769057 +0200
+++ aalib-1.4.0/src/aalinux.c	2008-03-25 03:27:24.209947801 +0200
@@ -1,7 +1,7 @@
 #include "config.h"
 #ifdef LINUX_DRIVER
 #include <sys/stat.h>
-#include <malloc.h>
+#include <stdlib.h>
 #include <sys/ioctl.h>
 #include <linux/kd.h>
 #ifdef GPM_MOUSEDRIVER
diff -Nur aalib-1.4.0-old/src/aamoureg.c aalib-1.4.0/src/aamoureg.c
--- aalib-1.4.0-old/src/aamoureg.c	2008-03-25 03:27:18.806769209 +0200
+++ aalib-1.4.0/src/aamoureg.c	2008-03-25 03:27:24.212947345 +0200
@@ -1,4 +1,4 @@
-#include <malloc.h>
+#include <stdlib.h>
 #include "config.h"
 #include "aalib.h"
 #include "aaint.h"
diff -Nur aalib-1.4.0-old/src/aarec.c aalib-1.4.0/src/aarec.c
--- aalib-1.4.0-old/src/aarec.c	2008-03-25 03:27:18.807769057 +0200
+++ aalib-1.4.0/src/aarec.c	2008-03-25 03:27:24.216946737 +0200
@@ -1,5 +1,5 @@
 #include <string.h>
-#include <malloc.h>
+#include <stdlib.h>
 #include "aalib.h"
 #include "aaint.h"
 aa_linkedlist *aa_kbdrecommended = NULL, *aa_mouserecommended = NULL,
diff -Nur aalib-1.4.0-old/src/aaregist.c aalib-1.4.0/src/aaregist.c
--- aalib-1.4.0-old/src/aaregist.c	2008-03-25 03:27:18.806769209 +0200
+++ aalib-1.4.0/src/aaregist.c	2008-03-25 03:27:24.217946585 +0200
@@ -1,4 +1,4 @@
-#include <malloc.h>
+#include <stdlib.h>
 #include "config.h"
 #include "aalib.h"
 #include "aaint.h"
diff -Nur aalib-1.4.0-old/src/aax.c aalib-1.4.0/src/aax.c
--- aalib-1.4.0-old/src/aax.c	2008-03-25 03:27:18.807769057 +0200
+++ aalib-1.4.0/src/aax.c	2008-03-25 03:27:24.229944761 +0200
@@ -1,4 +1,4 @@
-#include <malloc.h>
+#include <stdlib.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
diff -Nur aalib-1.4.0-old/src/aaxkbd.c aalib-1.4.0/src/aaxkbd.c
--- aalib-1.4.0-old/src/aaxkbd.c	2008-03-25 03:27:18.807769057 +0200
+++ aalib-1.4.0/src/aaxkbd.c	2008-03-25 03:27:24.229944761 +0200
@@ -1,4 +1,4 @@
-#include <malloc.h>
+#include <stdlib.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
diff -Nur aalib-1.4.0-old/src/aaxmouse.c aalib-1.4.0/src/aaxmouse.c
--- aalib-1.4.0-old/src/aaxmouse.c	2008-03-25 03:27:18.807769057 +0200
+++ aalib-1.4.0/src/aaxmouse.c	2008-03-25 03:27:24.229944761 +0200
@@ -1,4 +1,4 @@
-#include <malloc.h>
+#include <stdlib.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
