--- speex.pyx	2011-02-18 13:52:44.810198522 +0100
+++ speex.pyx-new	2011-02-18 13:52:36.046865189 +0100
@@ -522,7 +522,7 @@
       
                     # Copy from float to short (16 bits) for output
                     for i from 0 <= i < self.encFramesPerBlock:
-                      decShorts1[i] = decFloats1[i]
+                      decShorts1[i] = int(decFloats1[i])
       
                     self.decPhase = 0 #  back to awaiting LSB of count header
                     self.decNumBytes = 0
