--- tuxmath.desktop.OSmanOS	2010-01-27 00:40:39.000000000 +0200
+++ tuxmath.desktop	        2013-02-17 18:37:46.874223995 +0200
@@ -4,11 +4,13 @@
 GenericName[de]=Mathe Spiel
 GenericName[ru]=Образовательная игра
 GenericName[fr]=Jeu mathématique
+GenericName[tr]=Eğitsel matematik oyunu
 Comment=Tux Math - Learn math with Tux!
 Comment[pl]=Tux Math - Ucz siê matematyki z Tuxem!
 Comment[de]=Tux Math - Lerne Mathe mit Tux!
 Comment[ru]=Обучение арифметике с пингвином в главной роли
 Comment[fr]=Tux Math - Apprenez les mathématiques avec Tux!
+Comment[tr]=TuxMath Matematik Oyunu
 Exec=tuxmath
 Icon=tuxmath
 Type=Application
