Index: glib-2.25.14/gio/tests/Makefile.in
===================================================================
--- glib-2.25.14.orig/gio/tests/Makefile.in
+++ glib-2.25.14/gio/tests/Makefile.in
@@ -45,7 +45,7 @@ noinst_PROGRAMS = $(am__EXEEXT_5) $(am__
 @OS_UNIX_TRUE@	gdbus-threading gdbus-export gdbus-error \
 @OS_UNIX_TRUE@	gdbus-peer gdbus-exit-on-close gdbus-non-socket \
 @OS_UNIX_TRUE@	application testapps appinfo contenttype file \
-@OS_UNIX_TRUE@	$(NULL) live-g-file desktop-app-info unix-fd \
+@OS_UNIX_TRUE@	$(NULL) live-g-file unix-fd \
 @OS_UNIX_TRUE@	unix-streams
 @OS_UNIX_TRUE@am__append_2 = gdbus-example-unix-fd-client
 @OS_WIN32_TRUE@am__append_3 = win32-streams
@@ -76,7 +76,7 @@ am__EXEEXT_1 =
 @OS_UNIX_TRUE@	testapps$(EXEEXT) appinfo$(EXEEXT) \
 @OS_UNIX_TRUE@	contenttype$(EXEEXT) file$(EXEEXT) \
 @OS_UNIX_TRUE@	$(am__EXEEXT_1) live-g-file$(EXEEXT) \
-@OS_UNIX_TRUE@	desktop-app-info$(EXEEXT) unix-fd$(EXEEXT) \
+@OS_UNIX_TRUE@	unix-fd$(EXEEXT) \
 @OS_UNIX_TRUE@	unix-streams$(EXEEXT)
 @OS_WIN32_TRUE@am__EXEEXT_3 = win32-streams$(EXEEXT)
 @HAVE_DBUS1_TRUE@am__EXEEXT_4 = gdbus-serialization$(EXEEXT)
@@ -145,9 +145,6 @@ data_input_stream_DEPENDENCIES = $(progs
 am_data_output_stream_OBJECTS = data-output-stream.$(OBJEXT)
 data_output_stream_OBJECTS = $(am_data_output_stream_OBJECTS)
 data_output_stream_DEPENDENCIES = $(progs_ldadd)
-am_desktop_app_info_OBJECTS = desktop-app-info.$(OBJEXT)
-desktop_app_info_OBJECTS = $(am_desktop_app_info_OBJECTS)
-desktop_app_info_DEPENDENCIES = $(progs_ldadd)
 am_echo_server_OBJECTS = echo-server.$(OBJEXT)
 echo_server_OBJECTS = $(am_echo_server_OBJECTS)
 echo_server_DEPENDENCIES = $(progs_ldadd) \
@@ -739,8 +736,6 @@ buffered_output_stream_SOURCES = buffere
 buffered_output_stream_LDADD = $(progs_ldadd)
 live_g_file_SOURCES = live-g-file.c
 live_g_file_LDADD = $(progs_ldadd)
-desktop_app_info_SOURCES = desktop-app-info.c
-desktop_app_info_LDADD = $(progs_ldadd)
 unix_streams_SOURCES = unix-streams.c
 unix_streams_LDADD = $(progs_ldadd) \
 	$(top_builddir)/gthread/libgthread-2.0.la
@@ -1013,9 +1008,6 @@ data-input-stream$(EXEEXT): $(data_input
 data-output-stream$(EXEEXT): $(data_output_stream_OBJECTS) $(data_output_stream_DEPENDENCIES) 
 	@rm -f data-output-stream$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(data_output_stream_OBJECTS) $(data_output_stream_LDADD) $(LIBS)
-desktop-app-info$(EXEEXT): $(desktop_app_info_OBJECTS) $(desktop_app_info_DEPENDENCIES) 
-	@rm -f desktop-app-info$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(desktop_app_info_OBJECTS) $(desktop_app_info_LDADD) $(LIBS)
 echo-server$(EXEEXT): $(echo_server_OBJECTS) $(echo_server_DEPENDENCIES) 
 	@rm -f echo-server$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(echo_server_OBJECTS) $(echo_server_LDADD) $(LIBS)
@@ -1187,7 +1179,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/converter-stream.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/data-input-stream.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/data-output-stream.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/desktop-app-info.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/echo-server.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/file.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/filter-cat.Po@am__quote@
