--- xdelta-1.1.3/xdelta.c.sopwith	Thu Jan 24 21:27:24 2002
+++ xdelta-1.1.3/xdelta.c	Thu Jan 24 21:29:19 2002
@@ -230,6 +230,15 @@
 /* $Format: "#define XDELTA_REQUIRED_VERSION \"$ReleaseMajorVersion$.$ReleaseMinorVersion$.\"" $ */
 #define XDELTA_REQUIRED_VERSION "1.1."
 
+void
+xdp_generator_free (XdeltaGenerator *xg)
+{
+  xdp_source_free(xg->data_source);
+  g_ptr_array_free(xg->sources, TRUE);
+
+  g_free(xg);
+}
+
 XdeltaGenerator*
 __xdp_generator_new (const char* version)
 {
