Index: libvmime-0.7.1/Makefile.in
===================================================================
--- libvmime-0.7.1.orig/Makefile.in
+++ libvmime-0.7.1/Makefile.in
@@ -38,7 +38,7 @@ NORMAL_UNINSTALL = :
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
 host_triplet = @host@
-DIST_COMMON = autotools/config.guess autotools/config.sub INSTALL NEWS \
+DIST_COMMON = autotools/config.guess autotools/config.sub NEWS \
 	README AUTHORS ChangeLog $(srcdir)/Makefile.in \
 	$(srcdir)/Makefile.am $(top_srcdir)/configure \
 	$(am__configure_deps) $(srcdir)/config.h.in \
@@ -209,8 +209,8 @@ AUTOMAKE_OPTIONS = no-dist
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = $(GENERIC_VERSIONED_LIBRARY_NAME).pc
 EXTRA_DIST = SConstruct bootstrap
-doc_DATA = AUTHORS ChangeLog COPYING INSTALL NEWS README
-docdir = $(datadir)/doc/$(GENERIC_LIBRARY_NAME)
+doc_DATA = AUTHORS ChangeLog COPYING NEWS README
+docdir = $(datadir)/doc/$(LIBRARY_NAME)
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
