--- squid-3.1.0.18-20100316/src/cf.data.pre	2010-03-16 02:35:01.000000000 +0100
+++ squid-3.1.0.18-20100316.dlucio/src/cf.data.pre	2010-03-21 22:42:28.000000000 +0100
@@ -868,7 +868,7 @@
 http_access allow localhost
 
 # And finally deny all other access to this proxy
-http_access deny all
+http_access allow localhost
 NOCOMMENT_END
 DOC_END
 
@@ -2706,7 +2706,7 @@
 
 NAME: logfile_rotate
 TYPE: int
-DEFAULT: 10
+DEFAULT: 0
 LOC: Config.Log.rotateNumber
 DOC_START
 	Specifies the number of logfile rotations to make when you
@@ -4022,7 +4022,7 @@
 
 NAME: cache_mgr
 TYPE: string
-DEFAULT: webmaster
+DEFAULT: root
 LOC: Config.adminEmail
 DOC_START
 	Email-address of local cache manager who will receive
@@ -5161,10 +5161,10 @@
 NAME: icon_directory
 TYPE: string
 LOC: Config.icons.directory
-DEFAULT: @DEFAULT_ICON_DIR@
+DEFAULT: /usr/share/squid/icons
 DOC_START
 	Where the icons are stored. These are normally kept in
-	@DEFAULT_ICON_DIR@
+	/usr/share/squid/icons
 DOC_END
 
 NAME: global_internal_static
@@ -5206,8 +5206,10 @@
 DOC_START
 	If you wish to create your own versions of the default
 	error files to customize them to suit your company copy
-	the error/template files to another directory and point
-	this tag at them.
+	/usr/share/squid/errors contains sets of error files
+        in different languages. The default error directory
+        is /etc/squid/errors, which is a link to one of these
+        error sets.
 
 	WARNING: This option will disable multi-language support
 	         on error pages if used.
