diff -Naur '--exclude=cf.data.pre.dlucio' squid-3.3.5/doc/Makefile.am squid-3.3.5.dlucio/doc/Makefile.am
--- squid-3.3.5/doc/Makefile.am	2013-05-20 07:48:55.000000000 -0400
+++ squid-3.3.5.dlucio/doc/Makefile.am	2013-05-21 20:23:50.738888762 -0400
@@ -3,7 +3,7 @@
 
 DEFAULT_CONFIG_FILE = $(sysconfdir)/squid.conf
 DEFAULT_MIME_TABLE      = $(sysconfdir)/mime.conf
-DEFAULT_ERROR_DIR       = $(datadir)/errors
+DEFAULT_ERROR_DIR       = $(datadir)/squid/errors/@ERR_DEFAULT_LANGUAGE@
 
 EXTRA_DIST = \
 	debug-sections.txt
diff -Naur '--exclude=cf.data.pre.dlucio' squid-3.3.5/errors/Makefile.am squid-3.3.5.dlucio/errors/Makefile.am
--- squid-3.3.5/errors/Makefile.am	2013-05-20 07:48:55.000000000 -0400
+++ squid-3.3.5.dlucio/errors/Makefile.am	2013-05-21 20:23:21.885249482 -0400
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-errordir = $(datadir)/errors
+errordir = $(datadir)/squid/errors
 
 ##dist-hook eveything
 
diff -Naur '--exclude=cf.data.pre.dlucio' squid-3.3.5/icons/Makefile.am squid-3.3.5.dlucio/icons/Makefile.am
--- squid-3.3.5/icons/Makefile.am	2013-05-20 07:48:55.000000000 -0400
+++ squid-3.3.5.dlucio/icons/Makefile.am	2013-05-21 20:24:21.187508102 -0400
@@ -1,6 +1,6 @@
 include $(top_srcdir)/icons/list
 
-icondir = $(datadir)/icons/silk
+icondir = $(datadir)/squid/icons/silk
 icon_DATA = $(ICONS)
 EXTRA_DIST = $(ICONS) list SN.png
 DISTCLEANFILES = 
diff -Naur '--exclude=cf.data.pre.dlucio' squid-3.3.5/src/Makefile.am squid-3.3.5.dlucio/src/Makefile.am
--- squid-3.3.5/src/Makefile.am	2013-05-20 07:48:55.000000000 -0400
+++ squid-3.3.5.dlucio/src/Makefile.am	2013-05-21 20:25:09.665902039 -0400
@@ -931,8 +931,8 @@
 DEFAULT_UNLINKD		= $(libexecdir)/`echo unlinkd | sed '$(transform);s/$$/$(EXEEXT)/'`
 DEFAULT_LOGFILED	= $(libexecdir)/`echo log_file_daemon | sed '$(transform);s/$$/$(EXEEXT)/'`
 DEFAULT_DISKD		= $(libexecdir)/`echo diskd | sed '$(transform);s/$$/$(EXEEXT)/'`
-DEFAULT_ICON_DIR	= $(datadir)/icons
-DEFAULT_ERROR_DIR	= $(datadir)/errors
+DEFAULT_ICON_DIR       = $(datadir)/squid/icons
+DEFAULT_ERROR_DIR      = $(datadir)/squid/errors/@ERR_DEFAULT_LANGUAGE@
 
 # Make location configure settings available to the code
 DEFS += -DDEFAULT_CONFIG_FILE=\"$(DEFAULT_CONFIG_FILE)\" -DDEFAULT_SQUID_DATA_DIR=\"$(datadir)\" -DDEFAULT_SQUID_CONFIG_DIR=\"$(sysconfdir)\"
