--- src/cf.data.pre	2013-05-21 20:17:28.649665535 -0400
+++ src/cf.data.pre.dlucio	2013-05-21 20:20:07.046685300 -0400
@@ -5353,11 +5353,14 @@ DOC_START
 	UID/GID to the user specified below.  The default is to change
 	to UID of @DEFAULT_CACHE_EFFECTIVE_USER@.
 	see also; cache_effective_group
+NOCOMMENT_START
+cache_effective_user @DEFAULT_CACHE_EFFECTIVE_USER@
+NOCOMMENT_END
 DOC_END
 
 NAME: cache_effective_group
 TYPE: string
-DEFAULT: none
+DEFAULT: @DEFAULT_CACHE_EFFECTIVE_USER@
 DEFAULT_DOC: Use system group memberships of the cache_effective_user account
 LOC: Config.effectiveGroup
 DOC_START
@@ -5376,6 +5379,9 @@ DOC_START
 	This option is not recommended by the Squid Team.
 	Our preference is for administrators to configure a secure
 	user account for squid with UID/GID matching system policies.
+NOCOMMENT_START
+cache_effective_group @DEFAULT_CACHE_EFFECTIVE_USER@
+NOCOMMENT_END
 DOC_END
 
 NAME: httpd_suppress_version_string
