diff -up ghostscript-9.07/cups/gdevcups.c.gdevcups-debug-uninit ghostscript-9.07/cups/gdevcups.c
--- ghostscript-9.07/cups/gdevcups.c.gdevcups-debug-uninit	2013-02-14 07:58:16.000000000 +0000
+++ ghostscript-9.07/cups/gdevcups.c	2013-02-14 17:16:59.677951232 +0000
@@ -830,6 +830,7 @@ cups_get_matrix(gx_device *pdev,	/* I -
   dmprintf4(pdev->memory, "DEBUG2: PageSize = [ %d %d ], HWResolution = [ %d %d ]\n",
             cups->header.PageSize[0], cups->header.PageSize[1],
             cups->header.HWResolution[0], cups->header.HWResolution[1]);
+  if (size_set)
   dmprintf4(pdev->memory, "DEBUG2: HWMargins = [ %.3f %.3f %.3f %.3f ]\n",
             pdev->HWMargins[0], pdev->HWMargins[1], pdev->HWMargins[2],
             pdev->HWMargins[3]);
