diff -up ghostscript-9.12/base/gsicc_manage.c.icc-missing-check ghostscript-9.12/base/gsicc_manage.c
--- ghostscript-9.12/base/gsicc_manage.c.icc-missing-check	2014-03-26 11:31:11.296570797 +0000
+++ ghostscript-9.12/base/gsicc_manage.c	2014-03-26 11:32:10.384912584 +0000
@@ -613,6 +613,8 @@ gsicc_set_srcgtag_struct(gsicc_manager_t
                             icc_profile =
                                 gsicc_profile_new(str, mem, curr_ptr, strlen(curr_ptr));
                             code = sfclose(str);
+                        }
+                        if (str != NULL && icc_profile != NULL) {
                             gsicc_init_profile_info(icc_profile);
                             cmm = gsCMM_DEFAULT;
                             /* Check if this object is a devicelink profile.
