diff -up ghostscript-9.12/base/gsicc.c.gs694154 ghostscript-9.12/base/gsicc.c
--- ghostscript-9.12/base/gsicc.c.gs694154	2014-03-25 08:52:16.000000000 +0000
+++ ghostscript-9.12/base/gsicc.c	2014-03-26 11:32:31.406034840 +0000
@@ -468,6 +468,8 @@ gx_remap_ICC_imagelab(const gs_client_co
     cmm_dev_profile_t *dev_profile;
 
     code = dev_proc(dev, get_profile)(dev, &dev_profile);
+    if (dev_profile == NULL)
+        return gs_throw(gs_error_Fatal, "Attempting to do ICC remap with no profile");
     num_des_comps = gsicc_get_device_profile_comps(dev_profile);
     rendering_params.black_point_comp = pis->blackptcomp;
     rendering_params.graphics_type_tag = dev->graphics_type_tag;
