--- scipy/sparse/linalg/dsolve/umfpack/setup.py.bak	2008-03-11 11:22:32.000000000 -0400
+++ scipy/sparse/linalg/dsolve/umfpack/setup.py	2008-03-11 11:23:04.000000000 -0400
@@ -10,7 +10,8 @@
     config.add_data_dir('tests')
 
     umf_info = get_info( 'umfpack', notfound_action = 1 )
-
+    umf_info['include_dirs'] = ['/usr/include/suitesparse']
+    
     umfpack_i_file = config.paths('umfpack.i')[0]
     def umfpack_i(ext, build_dir):
         if umf_info:
