diff -up libmusicbrainz-3.0.2/src/utils_private.cpp.gcc44 libmusicbrainz-3.0.2/src/utils_private.cpp
--- libmusicbrainz-3.0.2/src/utils_private.cpp.gcc44	2008-07-02 01:57:10.000000000 -0500
+++ libmusicbrainz-3.0.2/src/utils_private.cpp	2009-02-25 11:07:46.000000000 -0600
@@ -23,6 +23,7 @@
 #include <cstdlib>
 #include <cstring>
 #include <cstdarg>
+#include <cstdio>
 #include <iostream>
 #include <map>
 #include <ne_uri.h> 
