Index: emesene-emesene-93a12ab/emesene/gui/base/Handler.py
===================================================================
--- emesene-emesene-93a12ab.orig/emesene/gui/base/Handler.py
+++ emesene-emesene-93a12ab/emesene/gui/base/Handler.py
@@ -64,6 +64,8 @@ EMESENE_CONTRIBUTORS = [ # TODO: Load th
 'Gabriele "Whisky" Visconti <visconti.gabriele@gmail.com>',
 ]

+PARDUS_BUG_WEBSITE = "http://bugs.pardus.org.tr/"
+
 class MenuHandler(object):
     '''this handler contains all the handlers needed to handle all the
     menu items
@@ -201,9 +203,9 @@ class HelpHandler(object):

     def notify_update(self, ver):
         self.dialog.information(_("New version available: %(ver)s\nVisit the "
-                    "<a href=\"%(web)s\">emesene website</a> for the "
-                    "latest informations about it.") % {"ver" : ver,
-                    "web" : EMESENE_WEBSITE})
+                    "<a href=\"%(web)s\">Pardus Bugzilla Website</a> and open a new update "
+                    "report to inform the package maintainer.") % {"ver" : ver,
+                    "web" : PARDUS_BUG_WEBSITE})

     def on_check_update_selected(self):
         ''' checks if a new stable version of emesene is available '''
