--- procps-3.2.5/w.c
+++ procps-3.2.5/w.c
@@ -285,10 +285,16 @@
 
     if (ioctl(1, TIOCGWINSZ, &win) != -1 && win.ws_col > 0)
 	maxcmd = win.ws_col;
+#if 0
+/* I have no idea why I shouldn't be allowed to see the result
+   with a line wrap. The other solution would be to create a
+   wcat alias, which does w|cat to work around this, but this 
+   would be stupid in my opinion. */
     if (maxcmd < 71) {
 	fprintf(stderr, "%d column window is too narrow\n", maxcmd);
 	exit(1);
     }
+#endif
     maxcmd -= 29 + (from ? 16 : 0) + (longform ? 20 : 0);
     if (maxcmd < 3)
 	fprintf(stderr, "warning: screen width %d suboptimal.\n", win.ws_col);
