diff -up procps-3.2.7/top.c.p6 procps-3.2.7/top.c
--- procps-3.2.7/top.c.p6	2006-06-25 08:41:48.000000000 +0200
+++ procps-3.2.7/top.c	2008-09-01 11:28:15.000000000 +0200
@@ -3331,12 +3331,15 @@ static void frame_make (void)
 
 int main (int dont_care_argc, char *argv[])
 {
+   struct stat isproc;
    (void)dont_care_argc;
    before(*argv);
                                         //                 +-------------+
    windows_stage1();                    //                 top (sic) slice
    configs_read();                      //                 > spread etc, <
    parse_args(&argv[1]);                //                 > lean stuff, <
+   if (stat("/proc/self", &isproc)==-1)
+      std_err("/proc is not mounted, required for output data");
    whack_terminal();                    //                 > onions etc. <
    windows_stage2();                    //                 as bottom slice
                                         //                 +-------------+
