diff -up procps-3.2.7/ps/parser.c.p3 procps-3.2.7/ps/parser.c
--- procps-3.2.7/ps/parser.c.p3	2008-09-01 11:11:50.000000000 +0200
+++ procps-3.2.7/ps/parser.c	2008-09-01 11:11:50.000000000 +0200
@@ -1238,7 +1238,7 @@ try_bsd:
   // as SysV options... and you're screwed if you've been patching
   // out the friendly warning. Cut-over is likely to be in 2005.
   if(!(personality & PER_FORCE_BSD))
-    fprintf(stderr, "Warning: bad ps syntax, perhaps a bogus '-'? See http://procps.sf.net/faq.html\n");
+    fprintf(stderr, "Warning: bad syntax, perhaps a bogus '-'? See /usr/share/doc/procps-%s/FAQ\n",procps_number_version);
   // Remember: contact albert@users.sf.net or procps-feedback@lists.sf.net
   // if you should feel tempted. Be damn sure you understand all
   // the issues. The same goes for other stuff too, BTW. Please ask.
diff -up procps-3.2.7/proc/version.h.p3 procps-3.2.7/proc/version.h
--- procps-3.2.7/proc/version.h.p3	2002-12-15 01:08:32.000000000 +0100
+++ procps-3.2.7/proc/version.h	2008-09-01 11:11:50.000000000 +0200
@@ -16,6 +16,7 @@ EXTERN_C_BEGIN
 
 extern void display_version(void);	/* display suite version */
 extern const char procps_version[];		/* global buf for suite version */
+extern const char procps_number_version[];	/* global buf for suite number version */
 
 extern int linux_version_code;		/* runtime version of LINUX_VERSION_CODE
 					   in /usr/include/linux/version.h */
diff -up procps-3.2.7/proc/library.map.p3 procps-3.2.7/proc/library.map
--- procps-3.2.7/proc/library.map.p3	2005-03-14 05:32:40.000000000 +0100
+++ procps-3.2.7/proc/library.map	2008-09-01 11:11:50.000000000 +0200
@@ -9,7 +9,7 @@ global:
   escape_str; escape_strlist;
   openproc; closeproc;
   tty_to_dev; dev_to_tty; open_psdb_message; open_psdb; lookup_wchan;
-  display_version; procps_version; linux_version_code;
+  display_version; procps_version; procps_number_version; linux_version_code;
   Hertz; smp_num_cpus; have_privs;
   sprint_uptime; uptime; user_from_uid; print_uptime; loadavg;
   pretty_print_signals; print_given_signals; unix_print_signals; signal_name_to_number; signal_number_to_name;
diff -up procps-3.2.7/proc/version.c.p3 procps-3.2.7/proc/version.c
--- procps-3.2.7/proc/version.c.p3	2003-01-29 02:11:43.000000000 +0100
+++ procps-3.2.7/proc/version.c	2008-09-01 11:11:50.000000000 +0200
@@ -16,8 +16,10 @@
 
 #ifdef MINORVERSION
 const char procps_version[] = "procps version " VERSION "." SUBVERSION "." MINORVERSION;
+const char procps_number_version[] = VERSION "." SUBVERSION "." MINORVERSION;
 #else
 const char procps_version[] = "procps version " VERSION "." SUBVERSION;
+const char procps_number_version[] = VERSION "." SUBVERSION;
 #endif
 
 void display_version(void) {
