diff -up procps-3.2.8/Makefile.make procps-3.2.8/Makefile
--- procps-3.2.8/Makefile.make	2010-09-08 18:16:34.000000000 +0200
+++ procps-3.2.8/Makefile	2010-09-08 18:54:41.668727996 +0200
@@ -174,7 +174,8 @@ INSTALL := $(BINFILES) $(MANFILES)
 # want this rule first, use := on ALL, and ALL not filled in yet
 all: do_all
 
--include */module.mk
+-include proc/module.mk
+-include ps/module.mk
 
 do_all:    $(ALL)
 
