diff -up procps-3.2.7/vmstat.c.sm procps-3.2.7/vmstat.c
--- procps-3.2.7/vmstat.c.sm	2010-04-01 14:02:55.000000000 +0200
+++ procps-3.2.7/vmstat.c	2010-04-01 14:13:27.000000000 +0200
@@ -645,16 +645,16 @@ int main(int argc, char *argv[]) {
 	}
         break;
       case 'S':
-	if (argv[1]){
-	      ++argv;
-	 	if (!strcmp(*argv, "k")) dataUnit=UNIT_k;
-	 	else if (!strcmp(*argv, "K")) dataUnit=UNIT_K;
-	 	else if (!strcmp(*argv, "m")) dataUnit=UNIT_m;
-	 	else if (!strcmp(*argv, "M")) dataUnit=UNIT_M;
+	if (argv[1] || argv[0][1]){
+	        char * a = argv[0][1] ? argv[0]+1 : *(++argv);
+	 	if (!strcmp(a, "k")) dataUnit=UNIT_k;
+	 	else if (!strcmp(a, "K")) dataUnit=UNIT_K;
+	 	else if (!strcmp(a, "m")) dataUnit=UNIT_m;
+	 	else if (!strcmp(a, "M")) dataUnit=UNIT_M;
 		else {fprintf(stderr, "-S requires k, K, m or M (default is kb)\n");
 		     exit(EXIT_FAILURE);
 		}
-		strcpy(szDataUnit, *argv);
+		strcpy(szDataUnit, a);
 	 }else {fprintf(stderr, "-S requires an argument\n");
 		exit(EXIT_FAILURE);
 	 }
