diff -up procps-3.2.7/vmstat.c.vmstat procps-3.2.7/vmstat.c
--- procps-3.2.7/vmstat.c.vmstat	2009-02-23 13:44:27.000000000 +0100
+++ procps-3.2.7/vmstat.c	2009-02-23 13:45:03.000000000 +0100
@@ -312,7 +312,7 @@ static int diskpartition_format(const ch
     struct disk_stat *disks;
     struct partition_stat *partitions, *current_partition=NULL;
     unsigned long ndisks, j, k, npartitions;
-    const char format[] = "%20u %10llu %10u %10u\n";
+    const char format[] = "%20u %10llu %10u %10llu\n";
 
     fDiskstat=fopen("/proc/diskstats","rb");
     if(!fDiskstat){
