--- procps-3.2.7/ps/ps.1.eip64	2007-06-14 11:08:35.000000000 +0200
+++ procps-3.2.7/ps/ps.1	2007-06-14 11:03:28.000000000 +0200
@@ -1156,6 +1156,14 @@
 or\ a\ decimal representation otherwise.
 T}
 
+rip	RIP	T{
+64-bit instruction pointer.
+T}
+
+rsp	RSP	T{
+64-bit stack pointer.
+T}
+
 rss	RSS	T{
 resident set size, the non\-swapped physical memory that
 a task has used (in\ kiloBytes).
--- procps-3.2.7/ps/output.c.eip64	2007-06-14 11:08:24.000000000 +0200
+++ procps-3.2.7/ps/output.c	2007-06-14 10:58:21.000000000 +0200
@@ -764,10 +764,18 @@
     return snprintf(outbuf, COLWID, "%08x", (unsigned)(pp->kstk_esp));
 }
 
+static int pr_rsp(char *restrict const outbuf, const proc_t *restrict const pp){
+    return snprintf(outbuf, COLWID, "%016lx", (unsigned long)(pp->kstk_esp));
+}
+
 static int pr_eip(char *restrict const outbuf, const proc_t *restrict const pp){
     return snprintf(outbuf, COLWID, "%08x", (unsigned)(pp->kstk_eip));
 }
 
+static int pr_rip(char *restrict const outbuf, const proc_t *restrict const pp){
+    return snprintf(outbuf, COLWID, "%016lx", (unsigned long)(pp->kstk_eip));
+}
+
 /* This function helps print old-style time formats */
 static int old_time_helper(char *dst, unsigned long long t, unsigned long long rel) {
   if(!t)            return snprintf(dst, COLWID, "    -");
@@ -1403,7 +1411,9 @@
 {"resident",  "RES",     pr_nop,      sr_resident, 5,MEM,    LNX, PO|RIGHT},
 {"rgid",      "RGID",    pr_rgid,     sr_rgid,    5,   0,    XXX, ET|RIGHT},
 {"rgroup",    "RGROUP",  pr_rgroup,   sr_rgroup,  8, GRP,    U98, ET|USER}, /* was 8 wide */
+{"rip",       "RIP",     pr_rip,      sr_kstk_eip, 16, 0,    LNX, TO|RIGHT},
 {"rlink",     "RLINK",   pr_nop,      sr_nop,     8,   0,    BSD, AN|RIGHT},
+{"rsp",       "RSP",     pr_rsp,      sr_kstk_eip, 16, 0,    LNX, TO|RIGHT},
 {"rss",       "RSS",     pr_rss,      sr_rss,     5,   0,    XXX, PO|RIGHT}, /* was 5 wide */
 {"rssize",    "RSS",     pr_rss,      sr_vm_rss,  5,   0,    DEC, PO|RIGHT}, /*rsz*/
 {"rsz",       "RSZ",     pr_rss,      sr_vm_rss,  5,   0,    BSD, PO|RIGHT}, /*rssize*/
