diff -up procps-3.2.7/vmstat.c.jx procps-3.2.7/vmstat.c
--- procps-3.2.7/vmstat.c.jx	2009-01-07 14:13:52.000000000 -0500
+++ procps-3.2.7/vmstat.c	2009-01-07 14:17:07.000000000 -0500
@@ -150,7 +150,7 @@ static int format_1000(unsigned long lon
 ////////////////////////////////////////////////////////////////////////////
 
 static void new_header(void){
-  printf("procs -----------memory---------- ---swap-- -----io---- --system-- -----cpu------\n");
+  printf("procs -----------memory---------- ---swap-- -----io---- --system-- -----cpu-----\n");
   printf(
     "%2s %2s %6s %6s %6s %6s %4s %4s %5s %5s %4s %4s %2s %2s %2s %2s %2s\n",
     "r","b",
