diff -up procps-3.2.7/w.c.p4 procps-3.2.7/w.c
--- procps-3.2.7/w.c.p4	2006-06-17 12:40:47.000000000 +0200
+++ procps-3.2.7/w.c	2008-09-01 11:26:33.000000000 +0200
@@ -163,6 +163,10 @@ static const proc_t *getproc(const utmp_
 	if(best && tmp->start_time <= best->start_time) continue;
     	best = tmp;
     }
+    /* It is there but SELinux wouldn't allow us to know the detail. Really
+       w should just be given rights */
+    if(!kill(u->ut_pid, 0) || errno != ESRCH)
+    	*found_utpid = 1;
     return best ? best : secondbest;
 }
 
