diff -up procps-3.2.7/top.c.cpuint procps-3.2.7/top.c
--- procps-3.2.7/top.c.cpuint	2008-01-11 08:32:44.000000000 +0100
+++ procps-3.2.7/top.c	2008-01-11 08:32:52.000000000 +0100
@@ -1824,7 +1824,7 @@ static void configs_read (void)
    if(Rc.mode_irixps && smp_num_cpus>1){
       // good for 100 CPUs per process
       pcpu_max_value = 9999.0;
-      Fieldstab[P_CPU].fmts = " %4.0f";
+      Fieldstab[P_CPU].fmts = " %4.1f";
    }
 
    // lastly, establish the true runtime secure mode and delay time
