--- procps-3.2.7/ps/parser.c.selinux	2006-06-25 09:14:58.000000000 +0200
+++ procps-3.2.7/ps/parser.c	2006-07-10 10:14:22.000000000 +0200
@@ -231,7 +231,7 @@
     // In the meantime, please do not add to it. The list is
     // intended to ONLY contain flags defined by the POSIX and UNIX
     // standards published by The Open Group, IEEE, and ISO.
-    if(!strchr("aAdefgGlnoptuU", *flagptr)) not_pure_unix = 1;  // dude, -Z ain't in POSIX
+    if(!strchr("aAdefgGlnoptuUZ", *flagptr)) not_pure_unix = 1;  // dude, -Z ain't in POSIX
 
     switch(*flagptr){
     case 'A':
--- procps-3.2.7/ps/ps.1.selinux	2005-10-30 03:44:03.000000000 +0100
+++ procps-3.2.7/ps/ps.1	2006-07-10 10:08:12.000000000 +0200
@@ -450,6 +450,9 @@
 Do not show flags; show rss in place of addr.
 This option can only be used with \fB\-l\fR.
 
+.opt \-Z
+display security context format (SELinux, etc.)
+
 .opt \-\-format \ format
 user\-defined format.  Identical to \fB\-o\fR and \fBo\fR.
 
