From 5fcb3b018797457c3099a86c6c1257b6e4e89ceb Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Tue, 25 Sep 2012 08:54:36 -0400
Subject: [PATCH] Remove mibstore.h

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
 src/alp_driver.c | 5 -----
 src/lg_driver.c  | 5 -----
 2 files changed, 10 deletions(-)

diff --git a/src/alp_driver.c b/src/alp_driver.c
index e38c613..36f2039 100644
--- a/src/alp_driver.c
+++ b/src/alp_driver.c
@@ -39,9 +39,6 @@
 /* All drivers initialising the SW cursor need this */
 #include "mipointer.h"
 
-/* All drivers implementing backing store need this */
-#include "mibstore.h"
-
 #include "micmap.h"
 
 /* Needed by the Shadow Framebuffer */
@@ -1617,8 +1614,6 @@ AlpScreenInit(SCREEN_INIT_ARGS_DECL)
 	if (init_picture)
 		fbPictureInit (pScreen, 0, 0);
     
-	miInitializeBackingStore(pScreen);
-
 	/*
 	 * Set initial black & white colourmap indices.
 	 */
diff --git a/src/lg_driver.c b/src/lg_driver.c
index 13f222d..c20bbd0 100644
--- a/src/lg_driver.c
+++ b/src/lg_driver.c
@@ -46,9 +46,6 @@
 /* need this for inputInfo */
 #include "inputstr.h"
 
-/* All drivers implementing backing store need this */
-#include "mibstore.h"
-
 #include "micmap.h"
 
 /* Needed by the Shadow Framebuffer */
@@ -1373,8 +1370,6 @@ LgScreenInit(SCREEN_INIT_ARGS_DECL)
 
 	fbPictureInit(pScreen, 0, 0);
 
-	miInitializeBackingStore(pScreen);
-
 	/*
 	 * Set initial black & white colourmap indices.
 	 */
-- 
1.8.0.2

