diff -p -up pilot-link-0.12.3/configure.ac~ pilot-link-0.12.3/configure.ac
--- pilot-link-0.12.3/configure.ac~	2007-11-09 20:36:37.000000000 +0100
+++ pilot-link-0.12.3/configure.ac	2009-02-26 14:56:33.000000000 +0100
@@ -58,16 +58,6 @@ AC_SUBST(PISYNC_CURRENT)
 AC_SUBST(PISYNC_REVISION)
 AC_SUBST(PISYNC_AGE)
 
-dnl Eat -Werror so configure will run properly, if the user provided it
-enable_werror=no
-save_CFLAGS="$CFLAGS"
-CFLAGS=`echo $save_CFLAGS | sed -e s/-Werror//g`
-CXXFLAGS=`echo $save_CXXFLAGS | sed -e s/-Werror//g`
-if test "x$CFLAGS" != "x$save_CFLAGS"; then
-	dnl -Werror was set; treat it as implicit --enable-werror below
-	enable_werror="yes"
-fi
-
 
 dnl Check host type
 AC_CANONICAL_HOST
