Index: kdepim-4.7.2/kmail/kmail_part.rc
===================================================================
--- kdepim-4.7.2.orig/kmail/kmail_part.rc
+++ kdepim-4.7.2/kmail/kmail_part.rc
@@ -256,7 +256,6 @@
  <ToolBar noMerge="1" name="mainToolBar" fullWidth="true" >
   <text>Main Toolbar</text>
   <Action name="new_message" />
-  <Action name="file_print" />
   <Separator/>
   <Action name="check_mail_in" />
   <Separator/>
Index: kdepim-4.7.2/kmail/kmreadermainwin.rc
===================================================================
--- kdepim-4.7.2.orig/kmail/kmreadermainwin.rc
+++ kdepim-4.7.2/kmail/kmreadermainwin.rc
@@ -70,8 +70,6 @@
  </MenuBar>
  <ToolBar noMerge="1" name="mainToolBar" fullWidth="true" >
   <text>Main Toolbar</text>
-  <Action name="file_print" />
-  <Separator/>
   <Action name="message_reply_menu" />
   <Action name="message_forward"/>
   <Separator/>
