Index: kdepim-4.4.4/kmail/mailinglistpropertiesdialog.cpp
===================================================================
--- kdepim-4.4.4.orig/kmail/mailinglistpropertiesdialog.cpp
+++ kdepim-4.4.4/kmail/mailinglistpropertiesdialog.cpp
@@ -96,6 +96,7 @@ MailingListFolderPropertiesDialog::Maili
            label, SLOT(setEnabled(bool)) );
   groupLayout->addWidget( label, 4, 0 );
   mMLId = new QLabel( "", mlGroup );
+  mMLId->setWordWrap( true );
   mMLId->setBuddy( label );
   groupLayout->addWidget( mMLId, 4, 1, 1, 2 );
   mMLId->setEnabled( false );
