diff -Nur gxine-0.5.905.orig/configure.ac gxine-0.5.905/configure.ac
--- gxine-0.5.905.orig/configure.ac	2010-09-24 14:39:45.029625945 +0300
+++ gxine-0.5.905/configure.ac	2010-09-24 14:40:20.945627608 +0300
@@ -336,7 +336,7 @@
 AM_PATH_XINE([$XINE_LIB_MIN_VER],, AC_MSG_ERROR([*** Please install xine-lib (devel) first ***]))
 
 dnl Check for xine-list-* (MIME types)
-XINE_LIST="$(xine-config --xine-list 2>/dev/null)"
+XINE_LIST=""
 AM_CONDITIONAL([XINE_LIST], [test "$XINE_LIST" != ''])
 AC_SUBST(XINE_LIST)
 
