diff -up gxine-0.5.902/src/ui.c.non-separate-toolbar gxine-0.5.902/src/ui.c
--- gxine-0.5.902/src/ui.c.non-separate-toolbar	2008-02-25 18:14:46.000000000 +0100
+++ gxine-0.5.902/src/ui.c	2008-04-22 22:57:14.000000000 +0200
@@ -838,10 +838,10 @@ void ui_preferences_register (xine_t *th
      NULL, 0, NULL, NULL);
 
   xine_config_register_bool
-    (this, "gui.windowed_mode.separate_toolbar", CONFIG_VERSION_ATLEAST (0, 6, 0),
+    (this, "gui.windowed_mode.separate_toolbar", 0,
      _("In windowed mode, whether the toolbar is in a separate window"), NULL,
      0, NULL, NULL);
-  config_update_default ("gui.windowed_mode.separate_toolbar", 1);
+  config_update_default ("gui.windowed_mode.separate_toolbar", 0);
 
   xine_config_register_bool
     (this, "gui.windowed_mode.unblank", CONFIG_VERSION_ATLEAST (0, 6, 0),
