diff -up gxine-0.5.905/configure.lirc gxine-0.5.905/configure
--- gxine-0.5.905/configure.lirc	2010-01-01 21:17:59.000000000 +0100
+++ gxine-0.5.905/configure	2010-01-01 21:18:40.000000000 +0100
@@ -17178,10 +17178,10 @@ fi
      if test "$have_lirc" = "yes"; then
 
         if test x"$LIRC_PREFIX" != "x"; then
-           lirc_libprefix="$LIRC_PREFIX/lib"
+           lirc_libprefix="$LIRC_PREFIX/${libdir}"
 	   LIRC_INCLUDE="-I$LIRC_PREFIX/include"
         fi
-        for llirc in $lirc_libprefix /lib /usr/lib /usr/local/lib; do
+        for llirc in $lirc_libprefix ${libdir}; do
           as_ac_File=`$as_echo "ac_cv_file_"$llirc/liblirc_client.so"" | $as_tr_sh`
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for \"$llirc/liblirc_client.so\"" >&5
 $as_echo_n "checking for \"$llirc/liblirc_client.so\"... " >&6; }
