diff -up gxine-0.5.905/Makefile.in.desktop gxine-0.5.905/Makefile.in
--- gxine-0.5.905/Makefile.in.desktop	2010-01-01 20:53:14.000000000 +0100
+++ gxine-0.5.905/Makefile.in	2010-01-01 20:53:23.000000000 +0100
@@ -887,7 +887,7 @@ translator: dist-pre
 	$(MAKE) -C $(top_builddir)/misc/po update-po
 
 @XINE_LIST_TRUE@gxine.desktop: gxine.desktop.in @XINE_LIST@
-@XINE_LIST_TRUE@	exec > $@; $(XINE_MIME_PROCESS) $<; @XINE_LIST@; cat $(srcdir)/mime.extra
+@XINE_LIST_TRUE@	exec > $@; $(XINE_MIME_PROCESS) $<; @XINE_LIST@; cat $(srcdir)/mime.extra | awk 'BEGIN { s = "%s;" } { printf s, $$1; s = "%s;" }'
 @XINE_LIST_FALSE@gxine.desktop: gxine.desktop.in $(MIME)
 @XINE_LIST_FALSE@	exec > $@; \
 @XINE_LIST_FALSE@	$(XINE_MIME_PROCESS) $<; \
