diff -p -up gxine-0.5.901/src/wizards.c~ gxine-0.5.901/src/wizards.c
--- gxine-0.5.901/src/wizards.c~	2008-03-09 20:03:46.000000000 +0100
+++ gxine-0.5.901/src/wizards.c	2008-03-09 20:04:33.000000000 +0100
@@ -256,8 +256,10 @@ static void desktop_integration (void)
   di_registration_flush ();
   if (gtk_toggle_button_get_active (cb.mailcap))
     di_register_mailcap();
+  /*
   if (gtk_toggle_button_get_active (cb.gnome))
     di_register_gnome();
+  */
   if (gtk_toggle_button_get_active (cb.kde))
     di_register_kde();
   if (gtk_toggle_button_get_active (cb.moz))
