diff -p -up gxine-0.5.905/gxine.desktop.in.fix-mime-types gxine-0.5.905/gxine.desktop.in
--- gxine-0.5.905/gxine.desktop.in.fix-mime-types	2009-12-19 17:10:03.000000000 +0100
+++ gxine-0.5.905/gxine.desktop.in	2010-01-02 16:08:22.000000000 +0100
@@ -14,5 +14,4 @@ Icon=gxine
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=AudioVideo;GTK;Player;TV;
-MimeType=x-content/video-dvd;x-content/audio-cdda;x-content/video-vcd;x-content/video-svcd;
\ Kein Zeilenumbruch am Dateiende.
+Categories=AudioVideo;GTK;Player;TV;Video;
diff -p -up gxine-0.5.905/mime.extra.fix-mime-types gxine-0.5.905/mime.extra
--- gxine-0.5.905/mime.extra.fix-mime-types	2009-12-19 17:10:04.000000000 +0100
+++ gxine-0.5.905/mime.extra	2010-01-02 16:08:22.000000000 +0100
@@ -1,3 +1 @@
-x-content/video-dvd
-x-content/video-svcd
-x-content/video-vcd
+x-content/video-dvd;x-content/video-svcd;x-content/video-vcd;
