Index: gxine-0.5.905/gxine.desktop.in
===================================================================
--- gxine-0.5.905.orig/gxine.desktop.in
+++ gxine-0.5.905/gxine.desktop.in
@@ -5,9 +5,11 @@ GenericName=Video Player
 GenericName[ca]=Reproductor de peŀlícules
 GenericName[de]=Video-Player
 GenericName[es]=Reproductor de películas
+GenericName[tr]=Medya Oynatıcı
 Comment=Watch videos or TV and listen to audio in a variety of formats
 Comment[ca]=Reproduïu vídeos i cançons
 Comment[es]=Reproduce vídeos y canciones
+Comment[tr]=Video veya TV oynatabilir, müzik dinleyebilirsiniz
 Exec=gxine
 X-MultipleArgs=true
 Icon=gxine
