Index: notify-python-0.1.1/src/Makefile.am
===================================================================
--- notify-python-0.1.1.orig/src/Makefile.am
+++ notify-python-0.1.1/src/Makefile.am
@@ -17,7 +17,7 @@ pkgpypynotify_PYTHON = __init__.py
 pypynotifydir = $(pkgpyexecdir)/pynotify
 pypynotify_LTLIBRARIES = _pynotify.la
 
-_pynotify_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_pynotify
+_pynotify_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_pynotify `python-config --libs`
 _pynotify_la_LIBADD = $(NOTIFY_PYTHON_LIBS)
 _pynotify_la_SOURCES = pynotifymodule.c pynotify.c
 #nodist_pynotify_la_SOURCES = pynotify.c
