Index: task-1.9.4/src/Config.cpp
===================================================================
--- task-1.9.4.orig/src/Config.cpp
+++ task-1.9.4/src/Config.cpp
@@ -558,15 +558,15 @@ void Config::createDefaultRC (const std:
            << defaults.substr (0, loc + 14)
            << data
            << "\n\n# Color theme (uncomment one to use)\n"
-           << "#include /usr/local/share/doc/task/rc/light-16.theme\n"
-           << "#include /usr/local/share/doc/task/rc/light-256.theme\n"
-           << "#include /usr/local/share/doc/task/rc/dark-16.theme\n"
-           << "#include /usr/local/share/doc/task/rc/dark-256.theme\n"
-           << "#include /usr/local/share/doc/task/rc/dark-red-256.theme\n"
-           << "#include /usr/local/share/doc/task/rc/dark-green-256.theme\n"
-           << "#include /usr/local/share/doc/task/rc/dark-blue-256.theme\n"
-           << "#include /usr/local/share/doc/task/rc/dark-violets-256.theme\n"
-           << "#include /usr/local/share/doc/task/rc/dark-yellow-green.theme\n"
+           << "#include /usr/share/doc/task/rc/light-16.theme\n"
+           << "#include /usr/share/doc/task/rc/light-256.theme\n"
+           << "#include /usr/share/doc/task/rc/dark-16.theme\n"
+           << "#include /usr/share/doc/task/rc/dark-256.theme\n"
+           << "#include /usr/share/doc/task/rc/dark-red-256.theme\n"
+           << "#include /usr/share/doc/task/rc/dark-green-256.theme\n"
+           << "#include /usr/share/doc/task/rc/dark-blue-256.theme\n"
+           << "#include /usr/share/doc/task/rc/dark-violets-256.theme\n"
+           << "#include /usr/share/doc/task/rc/dark-yellow-green.theme\n"
            << "\n";
 
   // Write out the new file.
