diff -Naur font-manager-0.5.7/configure.ac font-manager-0.5.7.orig/configure.ac
--- font-manager-0.5.7/configure.ac	2010-12-15 03:10:18.000000000 +0200
+++ font-manager-0.5.7.orig/configure.ac	2011-03-24 14:20:04.722000071 +0200
@@ -38,7 +38,7 @@
   *) AC_MSG_ERROR([bad value ${enableval} for --enable-bytecode]) ;;
 esac],
 [bytecode=true])
-AM_CONDITIONAL([BYTECODE], [test x$bytecode = xtrue])
+AM_CONDITIONAL([BYTECODE], [test x$bytecode = xfalse])
 
 AC_ARG_ENABLE([nls],
 [AS_HELP_STRING([--enable-nls], [Include translations])],
