diff -Naur font-manager-0.5.7.orig/src/font-manager.desktop font-manager-0.5.7/src/font-manager.desktop
--- font-manager-0.5.7.orig/src/font-manager.desktop	2010-09-06 22:28:20.000000000 +0300
+++ font-manager-0.5.7/src/font-manager.desktop	2011-03-24 15:36:34.054000073 +0200
@@ -2,7 +2,11 @@
 Version=1.0
 Name=Font Manager
 Name[en_US]=Font Manager
+Name[tr]=Font Manager
+GenericName=Font Manager
+GenericName[tr]=Yazıtipi Yöneticisi
 Comment=Preview, compare and manage fonts
+Comment[tr]=Yazıtiplerini görüntüleyin ve yönetin.
 Type=Application
 Exec=font-manager
 Terminal=false
