Index: gtk+2.0-2.21.5/gtk/gtksettings.c
===================================================================
--- gtk+2.0-2.21.5.orig/gtk/gtksettings.c	2010-07-12 18:23:45.000000000 +0200
+++ gtk+2.0-2.21.5/gtk/gtksettings.c	2010-07-22 17:55:16.033053648 +0200
@@ -310,7 +310,7 @@
                                              g_param_spec_string ("gtk-fallback-icon-theme",
 								  P_("Fallback Icon Theme Name"),
 								  P_("Name of a icon theme to fall back to"),
-								  NULL,
+								  "gnome",
 								  GTK_PARAM_READWRITE),
                                              NULL);
   g_assert (result == PROP_FALLBACK_ICON_THEME);
