--- gtk+-3.4.1/gtk/gtksettings.c~	2012-04-04 03:06:04.000000000 +0200
+++ gtk+-3.4.1/gtk/gtksettings.c	2012-04-14 13:53:25.306887626 +0200
@@ -410,7 +410,7 @@
 #ifdef G_OS_WIN32
 								  _gtk_win32_theme_get_default (),
 #else
-                                                                  "Raleigh",
+                                                                  "oxygen-gtk",
 #endif
                                                                   GTK_PARAM_READWRITE),
                                              NULL);
 
