diff -ur gtk+-2.22.1.orig//gdk/Makefile.am gtk+-2.22.1/gdk/Makefile.am
--- gtk+-2.22.1.orig//gdk/Makefile.am	2010-11-15 18:49:19.000000000 +0200
+++ gtk+-2.22.1/gdk/Makefile.am	2010-12-03 00:44:17.230000183 +0200
@@ -239,7 +239,7 @@
         x11/gdkx.h
 
 GdkX11-2.0.gir: $(gdktargetlib) Gdk-2.0.gir Makefile
-GdkX11_2_0_gir_SCANNERFLAGS = --warn-all --strip-prefix=Gdk
+GdkX11_2_0_gir_SCANNERFLAGS = --warn-all --identifier-prefix=Gdk --symbol-prefix=gdk
 GdkX11_2_0_gir_INCLUDES = Gio-2.0 Gdk-2.0 GdkPixbuf-2.0 Pango-1.0 xlib-2.0
 GdkX11_2_0_gir_LIBS = $(gdktargetlib)
 GdkX11_2_0_gir_FILES = $(x11_introspection_files)
