From ae657537cc1d144facb65cecfb3fe3d060fef3ec Mon Sep 17 00:00:00 2001
From: Stefano Karapetsas <stefano@karapetsas.com>
Date: Thu, 22 Nov 2012 16:28:24 +0000
Subject: Merge pull request #16 from bhull2010/master

Fix evince -> atril in configure
---
diff --git a/configure.ac b/configure.ac
index e8d347b..f1994ef 100644
--- a/configure.ac
+++ b/configure.ac
@@ -713,7 +713,7 @@ if test "x$enable_pixbuf" = "xyes"; then
     ATRIL_MIME_TYPES="${ATRIL_MIME_TYPES}image/*;"
 fi
 if test "x$enable_xps" = "xyes"; then
-    EVINCE_MIME_TYPES="${EVINCE_MIME_TYPES}application/oxps;application/vnd.ms-xpsdocument;"
+    ATRIL_MIME_TYPES="${ATRIL_MIME_TYPES}application/oxps;application/vnd.ms-xpsdocument;"
 fi
 if test "x$enable_impress" = "xyes"; then
     ATRIL_MIME_TYPES="${ATRIL_MIME_TYPES}application/vnd.sun.xml.impress;application/vnd.oasis.opendocument.presentation;"
--
cgit v0.9.0.3-65-g4555
