--- a/data/atril.desktop.in.in	2012-06-12 19:31:46.269149080 +0200
+++ b/data/atril.desktop.in.in	2012-06-12 19:32:21.353151758 +0200
@@ -13,5 +13,5 @@ X-MATE-Bugzilla-Bugzilla=MATE
 X-MATE-Bugzilla-Product=atril
 X-MATE-Bugzilla-Component=BugBuddyBugs
 X-MATE-Bugzilla-Version=@VERSION@
-Categories=MATE;GTK;Graphics;VectorGraphics;Viewer;
+Categories=MATE;GTK;Office;Viewer;Graphics;VectorGraphics;
 MimeType=@ATRIL_MIME_TYPES@

