--- backintime-1.0.8/common/backintime
+++ backintime-1.0.8/common/backintime
@@ -17,11 +17,7 @@ 
 #    with this program; if not, write to the Free Software Foundation, Inc.,
 #    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-if [ -f backintime.py ]; then
-	APP_PATH="."
-else
-	APP_PATH="/usr/share/backintime/common"
-fi
+APP_PATH="/usr/share/backintime/common"
 
 python ${APP_PATH}/backintime.py "$@"
 
--- backintime-1.0.8/gnome/backintime-gnome
+++ backintime-1.0.8/gnome/backintime-gnome
@@ -17,11 +17,7 @@ 
 #    with this program; if not, write to the Free Software Foundation, Inc.,
 #    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-if [ -f app.py ]; then
-	APP_PATH="."
-else
-	APP_PATH="/usr/share/backintime/gnome"
-fi
+APP_PATH="/usr/share/backintime/gnome"
 
 python ${APP_PATH}/app.py "$@"
 
--- backintime-1.0.8/kde4/backintime-kde4
+++ backintime-1.0.8/kde4/backintime-kde4
@@ -17,11 +17,7 @@ 
 #    with this program; if not, write to the Free Software Foundation, Inc.,
 #    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-if [ -f app.py ]; then
-	APP_PATH="."
-else
-	APP_PATH="/usr/share/backintime/kde4"
-fi
+APP_PATH="/usr/share/backintime/kde4"
 
 python ${APP_PATH}/app.py "$@"
 
