Index: pcsc-lite-1.7.0/src/hotplug_libudev.c
===================================================================
--- pcsc-lite-1.7.0.orig/src/hotplug_libudev.c
+++ pcsc-lite-1.7.0/src/hotplug_libudev.c
@@ -275,8 +275,8 @@ static LONG HPReadBundleValues(void)
 		}
 	}
 
-	/* if we found a specific driver */
-	if (driver)
+	/* if we found a specific driver other than openct */
+	if (driver && driver->CFBundleName && ((0 != strcmp(driver->CFBundleName, "openct-ifd") || !classdriver)))
 		return driver;
 
 	/* else return the Class driver (if any) */
