diff -Naur fluxbox-1.3.1.orig/util/startfluxbox.in fluxbox-1.3.1/util/startfluxbox.in
--- fluxbox-1.3.1.orig/util/startfluxbox.in	2009-10-04 22:45:41.000000000 +0300
+++ fluxbox-1.3.1/util/startfluxbox.in	2011-03-12 14:47:33.040000000 +0200
@@ -49,6 +49,9 @@
 # wmnd &
 # wmsmixer -w &
 # idesk &
+/usr/libexec/polkit-gnome-authentication-agent-1 &
+gvtray &
+nm-applet &

 # And last but not least we start fluxbox.
 # Because it is the last app you have to run it with ''exec'' before it.
