diff -Nuar fluxbox-1.3.1.orig//data/init.in fluxbox-1.3.1/data/init.in
--- fluxbox-1.3.1.orig//data/init.in	2011-03-12 15:11:57.776000002 +0200
+++ fluxbox-1.3.1/data/init.in	2011-03-12 15:13:07.733000001 +0200
@@ -3,5 +3,5 @@
 
 session.menuFile:	~/.@pkgprefix@fluxbox@pkgsuffix@/menu
 session.keyFile: ~/.@pkgprefix@fluxbox@pkgsuffix@/keys
-session.styleFile: @default_style@
+session.styleFile: /usr/share/fluxbox/styles/Operation
 session.configVersion:	13
