diff -up TiMidity++-2.13.2/timidity/timidity.c.dynlibroot TiMidity++-2.13.2/timidity/timidity.c
--- TiMidity++-2.13.2/timidity/timidity.c.dynlibroot	2008-02-21 12:41:09.000000000 +0100
+++ TiMidity++-2.13.2/timidity/timidity.c	2008-02-21 12:49:55.000000000 +0100
@@ -554,7 +554,7 @@ MAIN_INTERFACE struct URL_module *url_mo
 #ifndef SHARED_LIB_PATH
 #define SHARED_LIB_PATH PKGLIBDIR
 #endif /* SHARED_LIB_PATH */
-static char *dynamic_lib_root = SHARED_LIB_PATH;
+static char *dynamic_lib_root = NULL;
 #endif /* IA_DYNAMIC */
 
 #ifndef MAXPATHLEN
@@ -2970,8 +2970,7 @@ static inline int parse_opt_d(const char
 {
 	/* dynamic lib root */
 #ifdef IA_DYNAMIC
-	if (dynamic_lib_root)
-		free(dynamic_lib_root);
+	free(dynamic_lib_root);
 	dynamic_lib_root = safe_strdup(arg);
 	return 0;
 #else
@@ -3893,6 +3892,8 @@ static inline int parse_opt_h(const char
 				cmp->id_character, cmp->id_name);
 #endif	/* IA_DYNAMIC */
 #ifdef IA_DYNAMIC
+	if (!dynamic_lib_root)
+		dynamic_lib_root = safe_strdup(SHARED_LIB_PATH);
 	fprintf(fp, "Supported dynamic load interfaces (%s):" NLS,
 			dynamic_lib_root);
 	memset(mark, 0, sizeof(mark));
@@ -3997,6 +3998,8 @@ static inline char *dynamic_interface_in
 	int fd, n;
 	char *nl;
 	
+	if (!dynamic_lib_root)
+		dynamic_lib_root = safe_strdup(SHARED_LIB_PATH);
 	sprintf(libinfo, "%s" PATH_STRING "interface_%c.txt",
 			dynamic_lib_root, id);
 	if ((fd = open(libinfo, 0)) < 0)
@@ -4021,6 +4024,8 @@ char *dynamic_interface_module(int id)
 	static char shared_library[MAXPATHLEN];
 	int fd;
 	
+	if (!dynamic_lib_root)
+		dynamic_lib_root = safe_strdup(SHARED_LIB_PATH);
 	sprintf(shared_library, "%s" PATH_STRING "interface_%c%s",
 			dynamic_lib_root, id, SHARED_LIB_EXT);
 	if ((fd = open(shared_library, 0)) < 0)
