--- TiMidity++-2.13.0/timidity/sysdep.h.64bit	2004-07-01 16:40:02.681491184 +0200
+++ TiMidity++-2.13.0/timidity/sysdep.h	2004-07-01 16:40:51.139793001 +0200
@@ -107,7 +107,8 @@
 /* DEC MMS has 64 bit long words */
 /* Linux-Axp has also 64 bit long words */
 #if defined(DEC) || defined(__alpha__) \
-		|| defined(__ia64__) || defined (__x86_64__) || defined(__ppc64__)
+		|| defined(__ia64__) || defined (__x86_64__) \
+		|| defined(__ppc64__) || defined(__s390x__)
 typedef          int   int32;
 typedef unsigned int  uint32;
 typedef          long  int64;
