--- TiMidity++-2.13.0/timidity/playmidi.c.warnings	2004-07-01 16:30:33.245320437 +0200
+++ TiMidity++-2.13.0/timidity/playmidi.c	2004-07-01 16:30:38.867542215 +0200
@@ -8099,7 +8099,7 @@
 
 static int play_midi(MidiEvent *eventlist, int32 samples)
 {
-    int rc;
+    int rc = RC_NONE;
     static int play_count = 0;
 
     if (play_mode->id_character == 'M') {
--- TiMidity++-2.13.0/libunimod/mloader.c.warnings	2004-07-01 16:31:15.590458998 +0200
+++ TiMidity++-2.13.0/libunimod/mloader.c	2004-07-01 16:31:28.669648536 +0200
@@ -636,14 +636,14 @@
 	  s->sample->flags = (s->sample->flags & ~SF_FORMATMASK) | s->outfmt;
 	  if (s->sample->data == NULL)
 	    {
-	      FreeSampleList (musiclist);
+	      FreeSampleList ();
 	      return 1;
 	    }
 	}
       s = s->next;
     }
 
-  FreeSampleList (musiclist);
+  FreeSampleList ();
   return 0;
 }
 
