diff -up TiMidity++-2.13.2/configure~ TiMidity++-2.13.2/configure
--- TiMidity++-2.13.2/configure~	2008-04-03 15:54:20.000000000 +0200
+++ TiMidity++-2.13.2/configure	2008-04-03 15:54:20.000000000 +0200
@@ -6983,8 +6983,8 @@ main ()
   for (i = 0; i < 256; i++)
     if (XOR (islower (i), ISLOWER (i))
 	|| toupper (i) != TOUPPER (i))
-      exit(2);
-  exit (0);
+      return (2);
+  return (0);
 }
 _ACEOF
 rm -f conftest$ac_exeext
@@ -9583,8 +9583,8 @@ else
       int main ()
       {
 	if (fork() < 0)
-	  exit (1);
-	exit (0);
+	  return (1);
+	return (0);
       }
 _ACEOF
 rm -f conftest$ac_exeext
diff -up TiMidity++-2.13.2/timidity/speex_a.c~ TiMidity++-2.13.2/timidity/speex_a.c
--- TiMidity++-2.13.2/timidity/speex_a.c~	2008-04-03 16:07:23.000000000 +0200
+++ TiMidity++-2.13.2/timidity/speex_a.c	2008-04-03 16:08:14.000000000 +0200
@@ -26,6 +26,8 @@
 #include "config.h"
 #endif /* HAVE_CONFIG_H */
 #include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
 #if HAVE_UNISTD_H
 #include <unistd.h>
 #endif
@@ -35,6 +37,7 @@
 
 #include <speex/speex.h>
 #include <speex/speex_header.h>
+#include <speex/speex_stereo.h>
 #include <ogg/ogg.h>
 
 #include "timidity.h"
@@ -45,6 +48,7 @@
 #include "playmidi.h"
 #include "readmidi.h"
 #include "miditrace.h"
+#include "common.h"
 
 static int open_output(void); /* 0=success, 1=warning, -1=fatal error */
 static void close_output(void);
diff -up TiMidity++-2.13.2/timidity/esd_a.c~ TiMidity++-2.13.2/timidity/esd_a.c
--- TiMidity++-2.13.2/timidity/esd_a.c~	2008-04-03 16:05:37.000000000 +0200
+++ TiMidity++-2.13.2/timidity/esd_a.c	2008-04-03 16:05:37.000000000 +0200
@@ -28,6 +28,7 @@
 #endif /* HAVE_CONFIG_H */
 #define _GNU_SOURCE
 #include <stdio.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <fcntl.h>
 #include <dlfcn.h>
diff -up TiMidity++-2.13.2/timidity/oss_a.c~ TiMidity++-2.13.2/timidity/oss_a.c
--- TiMidity++-2.13.2/timidity/oss_a.c~	2008-04-03 16:03:36.000000000 +0200
+++ TiMidity++-2.13.2/timidity/oss_a.c	2008-04-03 16:03:36.000000000 +0200
@@ -39,6 +39,7 @@
 
 #if defined(HAVE_SYS_SOUNDCARD_H)
 #include <sys/soundcard.h>
+#include <sys/ioctl.h>
 #elif defined(linux)
 #include <sys/ioctl.h> /* new with 1.2.0? Didn't need this under 1.1.64 */
 #include <linux/soundcard.h>
diff -up TiMidity++-2.13.2/timidity/flac_a.c~ TiMidity++-2.13.2/timidity/flac_a.c
--- TiMidity++-2.13.2/timidity/flac_a.c~	2008-04-03 16:06:00.000000000 +0200
+++ TiMidity++-2.13.2/timidity/flac_a.c	2008-04-03 16:06:36.000000000 +0200
@@ -30,10 +30,10 @@
 #include <unistd.h>
 #endif /* HAVE_UNISTD_H */
 #include <fcntl.h>
+#include <time.h>
 
 #ifdef __W32__
 #include <io.h>
-#include <time.h>
 #endif
 
 #if defined(AU_FLAC_DLL) || defined(AU_OGGFLAC_DLL)
@@ -61,6 +61,7 @@
 #include "playmidi.h"
 #include "readmidi.h"
 #include "miditrace.h"
+#include "common.h"
 
 static int open_output(void); /* 0=success, 1=warning, -1=fatal error */
 static void close_output(void);
diff -up TiMidity++-2.13.2/timidity/effect.c~ TiMidity++-2.13.2/timidity/effect.c
--- TiMidity++-2.13.2/timidity/effect.c~	2008-04-03 15:59:16.000000000 +0200
+++ TiMidity++-2.13.2/timidity/effect.c	2008-04-03 15:59:16.000000000 +0200
@@ -29,6 +29,8 @@
 #include "config.h"
 #endif /* HAVE_CONFIG_H */
 
+#include <stdlib.h>
+
 #ifndef NO_STRING_H
 #include <string.h>
 #else
diff -up TiMidity++-2.13.2/timidity/sndfont.c~ TiMidity++-2.13.2/timidity/sndfont.c
--- TiMidity++-2.13.2/timidity/sndfont.c~	2008-04-03 16:02:29.000000000 +0200
+++ TiMidity++-2.13.2/timidity/sndfont.c	2008-04-03 16:02:29.000000000 +0200
@@ -55,6 +55,7 @@
 #include "output.h"
 #include "filter.h"
 #include "resample.h"
+#include "freq.h"
 
 #define FILENAME_NORMALIZE(fname) url_expand_home_dir(fname)
 #define FILENAME_REDUCED(fname)   url_unexpand_home_dir(fname)
