diff -Nuar firewall-manager-3.0.0//AUTHORS managers/firewall-manager//AUTHORS
--- firewall-manager-3.0.0//AUTHORS	1970-01-01 02:00:00.000000000 +0200
+++ managers/firewall-manager//AUTHORS	2011-03-29 16:03:27.899000000 +0300
@@ -0,0 +1,2 @@
+Gökmen Göksel <gokmen@pardus.org.tr>
+Bahadır Kandemir <bahadir@pardus.org.tr>
diff -Nuar firewall-manager-3.0.0//data/firewall-manager.desktop managers/firewall-manager//data/firewall-manager.desktop
--- firewall-manager-3.0.0//data/firewall-manager.desktop	2011-01-16 01:14:41.000000000 +0200
+++ managers/firewall-manager//data/firewall-manager.desktop	2011-03-29 16:03:27.748000000 +0300
@@ -8,11 +8,13 @@
 Name[hu]=Tűzfal kezelő
 Name[it]=Firewall Manager
 Name[nl]=Firewallbeheer
+Name[pl]=Menadżer zapory sieciowej.
 Name[pt_BR]=Gerenciador do Firewall
 Name[ru]=Firewall Manager
 Name[sr]=Менаџер ватреног зида
 Name[sv]=Brandväggshanteraren
 Name[tr]=Güvenlik Duvarı Yöneticisi
+Name[uk]=Керування мережевої безпеки.
 GenericName=Firewall Manager
 GenericName[az]=Təhlükəsizlik Divarı İdarəçisi
 GenericName[da]=Håndtering af firewall
@@ -22,11 +24,13 @@
 GenericName[hu]=Tűzfal kezelő
 GenericName[it]=Firewall Manager
 GenericName[nl]=Firewallbeheer
+GenericName[pl]=Menadżer zapory sieciowej.
 GenericName[pt_BR]=Gerenciador do Firewall
 GenericName[ru]=Firewall Manager
 GenericName[sr]=Менаџер ватреног зида
 GenericName[sv]=Brandväggshanteraren
 GenericName[tr]=Güvenlik Duvarı Yöneticisi
+GenericName[uk]=Керування мережевої безпеки.
 Comment=An application for managing firewall.
 Comment[az]=Təhlükəsizlik divarı idarəçilik vasitəsi
 Comment[de]=Konfiguration der Firewall.
@@ -35,11 +39,13 @@
 Comment[hu]=Tűzfal kezelő alkalmazás
 Comment[it]=Un'applicazione per la gestione del firewall.
 Comment[nl]=Een toepassing voor het beheren van de firewall.
+Comment[pl]=Aplikacja do zarządzania zaporą sieciową.
 Comment[pt_BR]=Um aplicativo para gerenciar seu Firewall
 Comment[ru]=Приложение для управления брандмауэром.
 Comment[sr]=Апликација за управљање ватреним зидом.
 Comment[sv]=Ett program för hantering av brandvägg.
 Comment[tr]=Güvenlik duvarı yönetim aracı.
+Comment[uk]=Програма для керування мережевої безпеки.
 Exec=firewall-manager
 Icon=security-high
 Type=Application
diff -Nuar firewall-manager-3.0.0//data/kcm_firewall-manager.desktop managers/firewall-manager//data/kcm_firewall-manager.desktop
--- firewall-manager-3.0.0//data/kcm_firewall-manager.desktop	2011-01-16 01:14:41.000000000 +0200
+++ managers/firewall-manager//data/kcm_firewall-manager.desktop	2011-03-29 16:03:27.747000000 +0300
@@ -20,11 +20,13 @@
 Name[hu]=Tűzfal kezelő
 Name[it]=Firewall Manager
 Name[nl]=Firewallbeheer
+Name[pl]=Menadżer zapory sieciowej.
 Name[pt_BR]=Gerenciador do Firewall
 Name[ru]=Firewall Manager
 Name[sr]=Менаџер ватреног зида
 Name[sv]=Brandväggshanteraren
 Name[tr]=Güvenlik Duvarı Yöneticisi
+Name[uk]=Керування мережевої безпеки.
 Comment=An application for managing firewall.
 Comment[az]=Təhlükəsizlik divarı idarəçilik vasitəsi
 Comment[de]=Konfiguration der Firewall.
@@ -33,9 +35,11 @@
 Comment[hu]=Tűzfal kezelő alkalmazás
 Comment[it]=Un'applicazione per la gestione del firewall.
 Comment[nl]=Een toepassing voor het beheren van de firewall.
+Comment[pl]=Aplikacja do zarządzania zaporą sieciową.
 Comment[pt_BR]=Um aplicativo para gerenciar seu Firewall
 Comment[ru]=Приложение для управления брандмауэром.
 Comment[sr]=Апликација за управљање ватреним зидом.
 Comment[sv]=Ett program för hantering av brandvägg.
 Comment[tr]=Güvenlik duvarı yönetim aracı.
+Comment[uk]=Програма для керування мережевої безпеки.
 X-KDE-Keywords=python security firewall managers
diff -Nuar firewall-manager-3.0.0//po/az.po managers/firewall-manager//po/az.po
--- firewall-manager-3.0.0//po/az.po	2010-11-29 23:17:55.000000000 +0200
+++ managers/firewall-manager//po/az.po	2011-03-29 16:03:27.893000000 +0300
@@ -76,6 +76,10 @@
 msgid "Access denied."
 msgstr "Qadağan olunmuş keçid"
 
+#: src/firewallmanager/main.py:94
+msgid "Error"
+msgstr "Xata"
+
 #: src/firewallmanager/pagedialog.py:31 src/firewallmanager/pagedialog.py:35
 msgid "Settings"
 msgstr "Ayarlar"
@@ -111,3 +115,28 @@
 #: ui/ui_main.py:82
 msgid "Add New"
 msgstr "Yenini əlavə et"
+
+#: ui/ui_settingsitem:78
+msgid "Add"
+msgstr "Əlavə et"
+
+#: ui/ui_settingsitem:79
+msgid "Remove"
+msgstr "Sil"
+
+#: ui/ui_settingsitem:80
+msgid "Move Up"
+msgstr "Yuxarı Daşı"
+
+#: ui/ui_settingsitem:81
+msgid "Move Down"
+msgstr "Aşağı Daşı"
+
+#: src/firewallmanager/pagedialog.py:67
+msgid "Ok"
+msgstr "Əvət"
+
+#: src/firewallmanager/pagedialog.py:71
+msgid "Cancel"
+msgstr "ləğv etmək"
+
diff -Nuar firewall-manager-3.0.0//po/da.po managers/firewall-manager//po/da.po
--- firewall-manager-3.0.0//po/da.po	2010-11-29 23:17:55.000000000 +0200
+++ managers/firewall-manager//po/da.po	2011-03-29 16:03:27.895000000 +0300
@@ -78,6 +78,10 @@
 msgid "Access denied."
 msgstr "Adgang nægtet."
 
+#: src/firewallmanager/main.py:94
+msgid "Error"
+msgstr "Fejl"
+
 #: src/firewallmanager/pagedialog.py:31 src/firewallmanager/pagedialog.py:35
 msgid "Settings"
 msgstr "Indstillinger"
@@ -113,3 +117,28 @@
 #: ui/ui_main.py:82
 msgid "Add New"
 msgstr "Tilføj ny"
+
+#: ui/ui_settingsitem:78
+msgid "Add"
+msgstr "Tilføj"
+
+#: ui/ui_settingsitem:79
+msgid "Remove"
+msgstr "Fjern"
+
+#: ui/ui_settingsitem:80
+msgid "Move Up"
+msgstr "Flyt op"
+
+#: ui/ui_settingsitem:81
+msgid "Move Down"
+msgstr "Flyt ned"
+
+#: src/firewallmanager/pagedialog.py:67
+msgid "Ok"
+msgstr "Ok"
+
+#: src/firewallmanager/pagedialog.py:71
+msgid "Cancel"
+msgstr "annullere"
+
diff -Nuar firewall-manager-3.0.0//po/de.po managers/firewall-manager//po/de.po
--- firewall-manager-3.0.0//po/de.po	2011-01-04 22:52:32.000000000 +0200
+++ managers/firewall-manager//po/de.po	2011-03-29 16:03:27.896000000 +0300
@@ -78,6 +78,10 @@
 msgid "Access denied."
 msgstr "Zugriff verweigert."
 
+#: src/firewallmanager/main.py:94
+msgid "Error"
+msgstr "Fehler"
+
 #: src/firewallmanager/pagedialog.py:31 src/firewallmanager/pagedialog.py:35
 msgid "Settings"
 msgstr "Einstellungen"
@@ -114,5 +118,29 @@
 msgid "Add New"
 msgstr "Neu hinzufügen"
 
+#: ui/ui_settingsitem:78
+msgid "Add"
+msgstr "Hinzufügen"
+
+#: ui/ui_settingsitem:79
+msgid "Remove"
+msgstr "Entfernen"
+
+#: ui/ui_settingsitem:80
+msgid "Move Up"
+msgstr "Nach oben"
+
+#: ui/ui_settingsitem:81
+msgid "Move Down"
+msgstr "Nach unten"
+
+#: src/firewallmanager/pagedialog.py:67
+msgid "Ok"
+msgstr "Ok"
+
+#: src/firewallmanager/pagedialog.py:71
+msgid "Cancel"
+msgstr "stornieren"
+
 #~ msgid " "
 #~ msgstr " "
diff -Nuar firewall-manager-3.0.0//po/es.po managers/firewall-manager//po/es.po
--- firewall-manager-3.0.0//po/es.po	2010-12-20 21:01:31.000000000 +0200
+++ managers/firewall-manager//po/es.po	2011-03-29 16:03:27.892000000 +0300
@@ -80,6 +80,10 @@
 msgid "Access denied."
 msgstr "Acceso denegado."
 
+#: src/firewallmanager/main.py:94
+msgid "Error"
+msgstr "Error"
+
 #: src/firewallmanager/pagedialog.py:31 src/firewallmanager/pagedialog.py:35
 msgid "Settings"
 msgstr "Configuración"
@@ -117,5 +121,29 @@
 msgid "Add New"
 msgstr "Agregar Nuevo"
 
+#: ui/ui_settingsitem:78
+msgid "Add"
+msgstr "Agregar"
+
+#: ui/ui_settingsitem:79
+msgid "Remove"
+msgstr "Eliminar"
+
+#: ui/ui_settingsitem:80
+msgid "Move Up"
+msgstr "Subir"
+
+#: ui/ui_settingsitem:81
+msgid "Move Down"
+msgstr "Bajar"
+
+#: src/firewallmanager/pagedialog.py:67
+msgid "Ok"
+msgstr "Bien"
+
+#: src/firewallmanager/pagedialog.py:71
+msgid "Cancel"
+msgstr "Cancelar"
+
 #~ msgid " "
 #~ msgstr " "
diff -Nuar firewall-manager-3.0.0//po/firewall-manager.pot managers/firewall-manager//po/firewall-manager.pot
--- firewall-manager-3.0.0//po/firewall-manager.pot	2010-11-29 23:17:55.000000000 +0200
+++ managers/firewall-manager//po/firewall-manager.pot	2011-03-29 16:03:27.897000000 +0300
@@ -75,6 +75,10 @@
 msgid "Access denied."
 msgstr ""
 
+#:src/firewallmanager/main.py:94
+msgid "Error"
+msgstr ""
+
 #: src/firewallmanager/pagedialog.py:31 src/firewallmanager/pagedialog.py:35
 msgid "Settings"
 msgstr ""
@@ -110,3 +114,28 @@
 #: ui/ui_main.py:82
 msgid "Add New"
 msgstr ""
+
+#:ui/ui_settingsitem.py:78
+msgid "Add"
+msgstr ""
+
+#:ui/ui_settingsitem.py:78
+msgid "Remove"
+msgstr ""
+
+#:ui/ui_settingsitem.py:78
+msgid "Move Up"
+msgstr ""
+
+#:ui/ui_settingsitem.py:78
+msgid "Move Down"
+msgstr ""
+
+#: src/firewallmanager/pagedialog.py:67
+msgid "Ok"
+msgstr ""
+
+#: src/firewallmanager/pagedialog.py:71
+msgid "Cancel"
+msgstr ""
+
diff -Nuar firewall-manager-3.0.0//po/fr.po managers/firewall-manager//po/fr.po
--- firewall-manager-3.0.0//po/fr.po	2010-11-29 23:17:55.000000000 +0200
+++ managers/firewall-manager//po/fr.po	2011-03-29 16:03:27.893000000 +0300
@@ -1,3 +1,5 @@
+#ifndef FR.PO
+#define FR.PO
 # Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
 # This file is distributed under the same license as the PACKAGE package.
 # 
@@ -77,6 +79,10 @@
 msgid "Access denied."
 msgstr "Accès refusé."
 
+#: src/firewallmanager/main.py:94
+msgid "Error"
+msgstr "Erreur"
+
 #: src/firewallmanager/pagedialog.py:31 src/firewallmanager/pagedialog.py:35
 msgid "Settings"
 msgstr "Paramètres"
@@ -113,5 +119,31 @@
 msgid "Add New"
 msgstr "Ajouter un nouveau"
 
+#: ui/ui_settingsitem:78
+msgid "Add"
+msgstr "Ajouter"
+
+#: ui/ui_settingsitem:79
+msgid "Remove"
+msgstr "Supprimer"
+
+#: ui/ui_settingsitem:80
+msgid "Move Up"
+msgstr "Déplacer vers le haut"
+
+#: ui/ui_settingsitem:81
+msgid "Move Down"
+msgstr "Descendre"
+
+#: src/firewallmanager/pagedialog.py:67
+msgid "Ok"
+msgstr "Bien"
+
+#: src/firewallmanager/pagedialog.py:71
+msgid "Cancel"
+msgstr "Annuler"
+
 #~ msgid " "
+
 #~ msgstr " "
+#endif // FR.PO
diff -Nuar firewall-manager-3.0.0//po/hu.po managers/firewall-manager//po/hu.po
--- firewall-manager-3.0.0//po/hu.po	2010-11-29 23:17:55.000000000 +0200
+++ managers/firewall-manager//po/hu.po	2011-03-29 16:03:27.894000000 +0300
@@ -79,6 +79,10 @@
 msgid "Access denied."
 msgstr "Hozzáférés megtagadva."
 
+#: src/firewallmanager/main.py:94
+msgid "Error"
+msgstr "Hiba"
+
 #: src/firewallmanager/pagedialog.py:31 src/firewallmanager/pagedialog.py:35
 msgid "Settings"
 msgstr "Beállítások"
@@ -115,5 +119,29 @@
 msgid "Add New"
 msgstr "Új hozzáadása"
 
+#: ui/ui_settingsitem:78
+msgid "Add"
+msgstr "Hozzáadása"
+
+#: ui/ui_settingsitem:79
+msgid "Remove"
+msgstr "Eltávolítás"
+
+#: ui/ui_settingsitem:80
+msgid "Move Up"
+msgstr "felköltözik"
+
+#: ui/ui_settingsitem:81
+msgid "Move Down"
+msgstr "Lefelé"
+
+#: src/firewallmanager/pagedialog.py:67
+msgid "Ok"
+msgstr "rendben"
+
+#: src/firewallmanager/pagedialog.py:71
+msgid "Cancel"
+msgstr "megszakításához"
+
 #~ msgid " "
 #~ msgstr " "
diff -Nuar firewall-manager-3.0.0//po/it.po managers/firewall-manager//po/it.po
--- firewall-manager-3.0.0//po/it.po	2010-12-16 23:42:07.000000000 +0200
+++ managers/firewall-manager//po/it.po	2011-03-29 16:03:27.894000000 +0300
@@ -82,6 +82,10 @@
 msgid "Access denied."
 msgstr "Accesso negato."
 
+#: src/firewallmanager/main.py:94
+msgid "Error"
+msgstr "Errore"
+
 #: src/firewallmanager/pagedialog.py:31 src/firewallmanager/pagedialog.py:35
 msgid "Settings"
 msgstr "Impostazioni"
@@ -118,5 +122,29 @@
 msgid "Add New"
 msgstr "Aggiungi"
 
+#: ui/ui_settingsitem:78
+msgid "Add"
+msgstr "Aggiungi"
+
+#: ui/ui_settingsitem:79
+msgid "Remove"
+msgstr "Rimuovere"
+
+#: ui/ui_settingsitem:80
+msgid "Move Up"
+msgstr "Rimontare"
+
+#: ui/ui_settingsitem:81
+msgid "Move Down"
+msgstr "Abbassare"
+
+#: src/firewallmanager/pagedialog.py:67
+msgid "Ok"
+msgstr "va bene"
+
+#: src/firewallmanager/pagedialog.py:71
+msgid "Cancel"
+msgstr "Annulla"
+
 #~ msgid " "
 #~ msgstr " "
diff -Nuar firewall-manager-3.0.0//po/nl.po managers/firewall-manager//po/nl.po
--- firewall-manager-3.0.0//po/nl.po	2010-11-29 23:17:55.000000000 +0200
+++ managers/firewall-manager//po/nl.po	2011-03-29 16:03:27.893000000 +0300
@@ -78,6 +78,10 @@
 msgid "Access denied."
 msgstr "Toegang geweigerd."
 
+#: src/firewallmanager/main.py:94
+msgid "Error"
+msgstr "Fout"
+
 #: src/firewallmanager/pagedialog.py:31 src/firewallmanager/pagedialog.py:35
 msgid "Settings"
 msgstr "Instellingen"
@@ -114,5 +118,29 @@
 msgid "Add New"
 msgstr "Nieuwe toevoegen"
 
+#: ui/ui_settingsitem:78
+msgid "Add"
+msgstr "Toe te voegen"
+
+#: ui/ui_settingsitem:79
+msgid "Remove"
+msgstr "Verwijder"
+
+#: ui/ui_settingsitem:80
+msgid "Move Up"
+msgstr "overgaan"
+
+#: ui/ui_settingsitem:81
+msgid "Move Down"
+msgstr "Naar beneden"
+
+#: src/firewallmanager/pagedialog.py:67
+msgid "Ok"
+msgstr "Okay"
+
+#: src/firewallmanager/pagedialog.py:71
+msgid "Cancel"
+msgstr "Annuleren"
+
 #~ msgid " "
 #~ msgstr " "
diff -Nuar firewall-manager-3.0.0//po/pt_BR.po managers/firewall-manager//po/pt_BR.po
--- firewall-manager-3.0.0//po/pt_BR.po	2010-12-20 21:01:31.000000000 +0200
+++ managers/firewall-manager//po/pt_BR.po	2011-03-29 16:03:27.892000000 +0300
@@ -77,6 +77,10 @@
 msgid "Access denied."
 msgstr "Acesso negado."
 
+#: src/firewallmanager/main.py:94
+msgid "Error"
+msgstr "Erro"
+
 #: src/firewallmanager/pagedialog.py:31 src/firewallmanager/pagedialog.py:35
 msgid "Settings"
 msgstr "Configurações"
@@ -113,5 +117,29 @@
 msgid "Add New"
 msgstr "Adicionar Novo"
 
+#: ui/ui_settingsitem:78
+msgid "Add"
+msgstr "Adicionar"
+
+#: ui/ui_settingsitem:79
+msgid "Remove"
+msgstr "Remover"
+
+#: ui/ui_settingsitem:80
+msgid "Move Up"
+msgstr "Mover para cima"
+
+#: ui/ui_settingsitem:81
+msgid "Move Down"
+msgstr "Descer"
+
+#: src/firewallmanager/pagedialog.py:67
+msgid "Ok"
+msgstr "Ok"
+
+#: src/firewallmanager/pagedialog.py:71
+msgid "Cancel"
+msgstr "cancelar"
+
 #~ msgid " "
 #~ msgstr " "
diff -Nuar firewall-manager-3.0.0//po/ru.po managers/firewall-manager//po/ru.po
--- firewall-manager-3.0.0//po/ru.po	2010-11-29 23:17:55.000000000 +0200
+++ managers/firewall-manager//po/ru.po	2011-03-29 16:03:27.896000000 +0300
@@ -76,6 +76,10 @@
 msgid "Access denied."
 msgstr "Отказано в доступе."
 
+#: src/firewallmanager/main.py:94
+msgid "Error"
+msgstr "Ошибка"
+
 #: src/firewallmanager/pagedialog.py:31 src/firewallmanager/pagedialog.py:35
 msgid "Settings"
 msgstr "Настройки"
@@ -111,3 +115,28 @@
 #: ui/ui_main.py:82
 msgid "Add New"
 msgstr "Добавить"
+
+#: ui/ui_settingsitem:78
+msgid "Add"
+msgstr "Добавить"
+
+#: ui/ui_settingsitem:79
+msgid "Remove"
+msgstr "Удалить"
+
+#: ui/ui_settingsitem:80
+msgid "Move Up"
+msgstr "придвигаться"
+
+#: ui/ui_settingsitem:81
+msgid "Move Down"
+msgstr "спускать"
+
+#: src/firewallmanager/pagedialog.py:67
+msgid "Ok"
+msgstr "хорошо"
+
+#: src/firewallmanager/pagedialog.py:71
+msgid "Cancel"
+msgstr "Отменить"
+
diff -Nuar firewall-manager-3.0.0//po/sr.po managers/firewall-manager//po/sr.po
--- firewall-manager-3.0.0//po/sr.po	2010-11-29 23:17:55.000000000 +0200
+++ managers/firewall-manager//po/sr.po	2011-03-29 16:03:27.895000000 +0300
@@ -77,6 +77,10 @@
 msgid "Access denied."
 msgstr "Приступ забрањен."
 
+#: src/firewallmanager/main.py:94
+msgid "Error"
+msgstr "Грешка"
+
 #: src/firewallmanager/pagedialog.py:31 src/firewallmanager/pagedialog.py:35
 msgid "Settings"
 msgstr "Подешавања"
@@ -112,3 +116,28 @@
 #: ui/ui_main.py:82
 msgid "Add New"
 msgstr "Додај нови"
+
+#: ui/ui_settingsitem:78
+msgid "Add"
+msgstr "Додај"
+
+#: ui/ui_settingsitem:79
+msgid "Remove"
+msgstr "Уклони"
+
+#: ui/ui_settingsitem:80
+msgid "Move Up"
+msgstr "Напредовати"
+
+#: ui/ui_settingsitem:81
+msgid "Move Down"
+msgstr "Премести надоле"
+
+#: src/firewallmanager/pagedialog.py:67
+msgid "Ok"
+msgstr "у реду"
+
+#: src/firewallmanager/pagedialog.py:71
+msgid "Cancel"
+msgstr "отказати"
+
diff -Nuar firewall-manager-3.0.0//po/sv.po managers/firewall-manager//po/sv.po
--- firewall-manager-3.0.0//po/sv.po	2010-11-29 23:17:55.000000000 +0200
+++ managers/firewall-manager//po/sv.po	2011-03-29 16:03:27.897000000 +0300
@@ -77,6 +77,10 @@
 msgid "Access denied."
 msgstr "Åtkomst nekas."
 
+#: src/firewallmanager/main.py:94
+msgid "Error"
+msgstr "Fel"
+
 #: src/firewallmanager/pagedialog.py:31 src/firewallmanager/pagedialog.py:35
 msgid "Settings"
 msgstr "Anpassa"
@@ -113,5 +117,29 @@
 msgid "Add New"
 msgstr "Lägg till ny"
 
+#: ui/ui_settingsitem:78
+msgid "Add"
+msgstr "Lägg till"
+
+#: ui/ui_settingsitem:79
+msgid "Remove"
+msgstr "Ta bort"
+
+#: ui/ui_settingsitem:80
+msgid "Move Up"
+msgstr "Flytta upp"
+
+#: ui/ui_settingsitem:81
+msgid "Move Down"
+msgstr "Flytta ned"
+
+#: src/firewallmanager/pagedialog.py:67
+msgid "Ok"
+msgstr "okej"
+
+#: src/firewallmanager/pagedialog.py:71
+msgid "Cancel"
+msgstr "avbryta"
+
 #~ msgid " "
 #~ msgstr " "
diff -Nuar firewall-manager-3.0.0//po/tr.po managers/firewall-manager//po/tr.po
--- firewall-manager-3.0.0//po/tr.po	2010-11-29 23:17:55.000000000 +0200
+++ managers/firewall-manager//po/tr.po	2011-03-29 16:03:27.896000000 +0300
@@ -80,6 +80,10 @@
 msgid "Access denied."
 msgstr "Erişim reddedildi."
 
+#: src/firewallmanager/main.py:94
+msgid "Error"
+msgstr "Hata"
+
 #: src/firewallmanager/pagedialog.py:31 src/firewallmanager/pagedialog.py:35
 msgid "Settings"
 msgstr "Ayarlar"
@@ -116,5 +120,29 @@
 msgid "Add New"
 msgstr "Yeni Ekle"
 
+#: ui/ui_settingsitem:78
+msgid "Add"
+msgstr "Ekle"
+
+#: ui/ui_settingsitem:79
+msgid "Remove"
+msgstr "Sil"
+
+#: ui/ui_settingsitem:80
+msgid "Move Up"
+msgstr "Yukarı Taşı"
+
+#: ui/ui_settingsitem:81
+msgid "Move Down"
+msgstr "Asağı Taşı"
+
+#: src/firewallmanager/pagedialog.py:67
+msgid "Ok"
+msgstr "Tamam"
+
+#: src/firewallmanager/pagedialog.py:71
+msgid "Cancel"
+msgstr "İptal"
+
 #~ msgid " "
 #~ msgstr " "
diff -Nuar firewall-manager-3.0.0//setup.py managers/firewall-manager//setup.py
--- firewall-manager-3.0.0//setup.py	2010-11-29 23:17:55.000000000 +0200
+++ managers/firewall-manager//setup.py	2011-03-29 16:03:27.900000000 +0300
@@ -26,6 +26,12 @@
 from distutils.command.install import install
 
 PROJECT = about.appName
+FOR_KDE_4=False
+
+if 'kde4' in sys.argv:
+    sys.argv.remove('kde4')
+    FOR_KDE_4=True
+    print 'UI files will be created for KDE 4.. '
 
 def makeDirs(directory):
     if not os.path.exists(directory):
@@ -48,7 +54,10 @@
     # Collect UI files
     filelist = []
     for filename in glob.glob1("ui", "*.ui"):
-        os.system("pykde4uic -o ui/ui_%s.py ui/%s" % (filename.split(".")[0], filename))
+        if FOR_KDE_4:
+            os.system("pykde4uic -o ui/ui_%s.py ui/%s -g %s" % (filename.split(".")[0], filename, PROJECT))
+        else:
+            os.system("pyuic4 -o ui/ui_%s.py ui/%s -g %s" % (filename.split(".")[0], filename, PROJECT))
 
     # Collect headers for desktop files
     for filename in glob.glob("data/*.desktop.in"):
@@ -65,12 +74,7 @@
                         --keyword=N_ \
                         --keyword=i18n \
                         --keyword=ki18n \
-                        --kde \
-                        -ci18n -ki18n:1 -ki18nc:1c,2 -ki18np:1,2 -ki18ncp:1c,2,3 -ktr2i18n:1 \
-                        -kI18N_NOOP:1 -kI18N_NOOP2:1c,2 -kaliasLocale -kki18n:1 -kki18nc:1c,2 \
-                        -kki18np:1,2 -kki18ncp:1c,2,3 \
-                        --files-from=%s \
-                        -o po/%s.pot" % (PROJECT, files, PROJECT))
+                        -o po/%s.pot" % (PROJECT, files))
 
     # Update PO files
     for item in glob.glob1("po", "*.po"):
@@ -105,7 +109,10 @@
 
         print "Generating UIs..."
         for filename in glob.glob1("ui", "*.ui"):
-            os.system("pykde4uic -o build/firewallmanager/ui_%s.py ui/%s" % (filename.split(".")[0], filename))
+            if FOR_KDE_4:
+                os.system("pykde4uic -o build/firewallmanager/ui_%s.py ui/%s -g %s" % (filename.split(".")[0], filename, PROJECT))
+            else:
+                os.system("pyuic4 -o build/firewallmanager/ui_%s.py ui/%s -g %s" % (filename.split(".")[0], filename, PROJECT))
 
         print "Generating RCs..."
         for filename in glob.glob1("data", "*.qrc"):
@@ -128,9 +135,9 @@
             bin_dir = "/usr/bin"
 
         locale_dir = os.path.join(root_dir, "locale")
-        apps_dir = os.path.join(root_dir, "applications/kde4")
-        services_dir = os.path.join(root_dir, "kde4/services")
-        project_dir = os.path.join(root_dir, "kde4/apps", PROJECT)
+        apps_dir = os.path.join(root_dir, "applications")
+        #services_dir = os.path.join(root_dir, "kde4/services")
+        project_dir = os.path.join(root_dir, PROJECT)
 
         # Make directories
         print "Making directories..."
@@ -138,13 +145,14 @@
         makeDirs(locale_dir)
         makeDirs(apps_dir)
         makeDirs(project_dir)
-        makeDirs(services_dir)
+        #makeDirs(services_dir)
 
         # Install desktop files
         print "Installing desktop files..."
-
-        shutil.copy("data/%s.desktop" % PROJECT, apps_dir)
-        shutil.copy("data/kcm_%s.desktop" % PROJECT, services_dir)
+        if FOR_KDE_4:
+            shutil.copy("data/kcm_%s.desktop" % PROJECT, apps_dir)
+        else:
+            shutil.copy("data/%s.desktop" % PROJECT, apps_dir)
         shutil.rmtree('build/data')
 
         # Install codes
@@ -192,14 +200,14 @@
         bin_dir = "/usr/bin"
 
         locale_dir = os.path.join(root_dir, "locale")
-        apps_dir = os.path.join(root_dir, "applications/kde4")
-        services_dir = os.path.join(root_dir, "kde4/services")
-        project_dir = os.path.join(root_dir, "kde4/apps", PROJECT)
+        apps_dir = os.path.join(root_dir, "applications")
+        #services_dir = os.path.join(root_dir, "kde4/services")
+        project_dir = os.path.join(root_dir, PROJECT)
 
         print 'Uninstalling ...'
         remove(project_dir)
         remove(apps_dir +"/%s.desktop" % PROJECT)
-        remove(services_dir +"/kcm_%s.desktop" % PROJECT)
+        #remove(services_dir +"/kcm_%s.desktop" % PROJECT)
         for filename in glob.glob1('po', '*.po'):
             lang = filename.rsplit(".", 1)[0]
             remove(os.path.join(locale_dir, "%s/LC_MESSAGES" % lang, "%s.mo" % PROJECT))
@@ -225,8 +233,8 @@
       description       = unicode(about.PACKAGE),
       license           = unicode('GPL'),
       author            = "Pardus Developers",
-      author_email      = about.bugEmail,
-      url               = about.homePage,
+      author_email      = "bugs@pardus.org.tr",
+      url               = "http://www.pardus.org.tr/eng/projects",
       packages          = [''],
       package_dir       = {'': ''},
       data_files        = [],
diff -Nuar firewall-manager-3.0.0//src/firewallmanager/about.py managers/firewall-manager//src/firewallmanager/about.py
--- firewall-manager-3.0.0//src/firewallmanager/about.py	2010-12-26 17:47:22.000000000 +0200
+++ managers/firewall-manager//src/firewallmanager/about.py	2011-03-29 16:03:27.698000000 +0300
@@ -10,30 +10,32 @@
 #
 # Please read the COPYING file.
 #
-
-# PyKDE
-from PyKDE4.kdecore import KAboutData, ki18n, ki18nc
+import context as ctx
 
 PACKAGE = "Firewall Manager"
-
-# Application Data
-appName     = "firewall-manager"
-modName     = "firewallmanager"
-programName = ki18n(PACKAGE)
-version     = "3.0.0"
-description = ki18n(PACKAGE)
-license     = KAboutData.License_GPL
-copyright   = ki18n("(c) 2006-2010 TUBITAK/UEKAE")
-text        = ki18n(None)
+appName = "firewall-manager"
+version = "3.0.0"
 homePage    = "http://developer.pardus.org.tr/projects/firewall-manager"
 bugEmail    = "bugs@pardus.org.tr"
-catalog     = appName
-aboutData   = KAboutData(appName, catalog, programName, version, description, license, copyright, text, homePage, bugEmail)
+if ctx.Pds.session == ctx.pds.Kde4:
+    
+    # PyKDE
+    from PyKDE4.kdecore import KAboutData, ki18n, ki18nc
+
+    # Application Data
+    modName     = "firewallmanager"
+    programName = ki18n(PACKAGE)
+    description = ki18n(PACKAGE)
+    license     = KAboutData.License_GPL
+    copyright   = ki18n("(c) 2006-2010 TUBITAK/UEKAE")
+    text        = ki18n(None)
+    catalog     = appName
+    aboutData   = KAboutData(appName, catalog, programName, version, description, license, copyright, text, homePage, bugEmail)
 
-# Author(s)
-aboutData.addAuthor(ki18n("Gökmen Göksel"), ki18n("Current Maintainer"))
-aboutData.addAuthor(ki18n("Bahadır Kandemir"), ki18n("First Developer"))
-aboutData.setTranslator(ki18nc("NAME OF TRANSLATORS", "Your names"), ki18nc("EMAIL OF TRANSLATORS", "Your emails"))
+    # Author(s)
+    aboutData.addAuthor(ki18n("Gökmen Göksel"), ki18n("Current Maintainer"))
+    aboutData.addAuthor(ki18n("Bahadır Kandemir"), ki18n("First Developer"))
+    aboutData.setTranslator(ki18nc("NAME OF TRANSLATORS", "Your names"), ki18nc("EMAIL OF TRANSLATORS", "Your emails"))
 
-# Use this if icon name is different than appName
-aboutData.setProgramIconName("security-high")
+    # Use this if icon name is different than appName
+    aboutData.setProgramIconName("security-high")
diff -Nuar firewall-manager-3.0.0//src/firewallmanager/config.py managers/firewall-manager//src/firewallmanager/config.py
--- firewall-manager-3.0.0//src/firewallmanager/config.py	2010-07-06 14:16:34.000000000 +0300
+++ managers/firewall-manager//src/firewallmanager/config.py	2011-03-29 16:03:27.702000000 +0300
@@ -11,7 +11,6 @@
 # Please read the COPYING file.
 #
 
-from PyKDE4.kdecore import i18n
 
 # Animation
 ANIM_SHOW, ANIM_HIDE = range(2)
diff -Nuar firewall-manager-3.0.0//src/firewallmanager/context.py managers/firewall-manager//src/firewallmanager/context.py
--- firewall-manager-3.0.0//src/firewallmanager/context.py	1970-01-01 02:00:00.000000000 +0200
+++ managers/firewall-manager//src/firewallmanager/context.py	2011-03-29 16:18:18.327000000 +0300
@@ -0,0 +1,36 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+import pds
+import traceback
+from time import time
+from pds.qiconloader import QIconLoader
+from PyQt4.QtGui import QMessageBox
+from context import *
+
+Pds = pds.Pds('firewall-manager', debug = True)
+# Force to use Default Session for testing
+#Pds.session = pds.DefaultDe
+print 'Current session is : %s %s' % (Pds.session.Name, Pds.session.Version)
+
+i18n = Pds.i18n
+KIconLoader = QIconLoader(Pds,forceCache=True)
+KIcon = KIconLoader.icon
+
+time_counter = 0
+start_time = time()
+last_time = time()
+
+def _time():
+    global last_time, time_counter
+    trace = list(traceback.extract_stack())
+    diff = time() - start_time
+    print ('%s ::: %s:%s' % (time_counter, trace[-2][0].split('/')[-1], trace[-2][1])), diff, diff - last_time
+    last_time = diff
+    time_counter += 1
+
+def createMessage(self,errorTitle, errorMessage):
+    errorTitle = i18n(errorTitle)
+    errorMessage= i18n(errorMessage)
+    self.messageBox = QMessageBox(errorTitle, errorMessage, QMessageBox.Critical, QMessageBox.Ok, 0, 0)
+    self.messageBox.show()
diff -Nuar firewall-manager-3.0.0//src/firewallmanager/edit.py managers/firewall-manager//src/firewallmanager/edit.py
--- firewall-manager-3.0.0//src/firewallmanager/edit.py	2010-07-06 14:16:34.000000000 +0300
+++ managers/firewall-manager//src/firewallmanager/edit.py	2011-03-29 16:03:27.702000000 +0300
@@ -15,6 +15,3 @@
 from PyQt4 import QtCore
 from PyQt4 import QtGui
 
-# PyKDE
-from PyKDE4 import kdeui
-from PyKDE4 import kdecore
diff -Nuar firewall-manager-3.0.0//src/firewallmanager/item.py managers/firewall-manager//src/firewallmanager/item.py
--- firewall-manager-3.0.0//src/firewallmanager/item.py	2010-07-06 14:16:34.000000000 +0300
+++ managers/firewall-manager//src/firewallmanager/item.py	2011-03-29 16:03:27.700000000 +0300
@@ -16,7 +16,7 @@
 from PyQt4 import QtGui
 
 # PyKDE
-from PyKDE4.kdeui import KIcon
+from context import *
 
 # UI
 from firewallmanager.ui_item import Ui_ItemWidget
diff -Nuar firewall-manager-3.0.0//src/firewallmanager/main.py managers/firewall-manager//src/firewallmanager/main.py
--- firewall-manager-3.0.0//src/firewallmanager/main.py	2010-07-06 14:16:34.000000000 +0300
+++ managers/firewall-manager//src/firewallmanager/main.py	2011-03-29 16:03:27.701000000 +0300
@@ -14,11 +14,6 @@
 # PyQt
 from PyQt4 import QtCore
 from PyQt4 import QtGui
-
-# PyKDE
-from PyKDE4 import kdeui
-from PyKDE4 import kdecore
-
 # UI
 from firewallmanager.ui_main import Ui_MainWidget
 
@@ -37,6 +32,9 @@
 # Page Dialog
 from firewallmanager.pagedialog import PageDialog
 
+#Context 
+import context as ctx
+from context import *
 
 class MainWidget(QtGui.QWidget, Ui_MainWidget):
     def __init__(self, parent, embed=False):
@@ -93,7 +91,7 @@
             Check if there are packages that provide required backend.
         """
         if not len(self.iface.getPackages()):
-            kdeui.KMessageBox.error(self, kdecore.i18n("There are no packages that provide backend for this application.\nPlease make sure that packages are installed and configured correctly."))
+            createMessage(self,"Error","There are no packages that provide backend for this application.\nPlease make sure that packages are installed and configured correctly.")
             return False
         return True
 
@@ -149,7 +147,7 @@
         type_ = ""
 
         # Build widget and widget item
-        widget = self.makeItemWidget(id_, name, description, type_, kdeui.KIcon(icon), state)
+        widget = self.makeItemWidget(id_, name, description, type_,KIcon(icon), state)
         widgetItem = ItemListWidgetItem(self.listItems, widget)
 
         # Rules are static
@@ -197,7 +195,7 @@
             Builds item filter.
         """
         self.comboFilter.clear()
-        self.comboFilter.addItem(kdecore.i18n("All"), QtCore.QVariant("all"))
+        self.comboFilter.addItem(i18n("All"), QtCore.QVariant("all"))
 
     def buildMenu(self):
         """
@@ -208,7 +206,7 @@
         self.pushNew.setMenu(menu)
 
         # New item
-        action = QtGui.QAction(kdecore.i18n("Action"), self)
+        action = QtGui.QAction(i18n("Action"), self)
         action.setData(QtCore.QVariant("action"))
         menu.addAction(action)
 
@@ -254,9 +252,9 @@
             self.iface.setModuleState(widget.getId(), state == QtCore.Qt.Checked)
         except Exception, e:
             if "Comar.PolicyKit" in e._dbus_error_name:
-                kdeui.KMessageBox.error(self, kdecore.i18n("Access denied."))
+                createMessage(self,"Error","Access denied.")
             else:
-                kdeui.KMessageBox.error(self, unicode(e))
+                createMessage(self,"Error", unicode(e))
             self.buildItemList()
 
     def slotItemEdit(self):
@@ -277,9 +275,9 @@
                 self.iface.setModuleParameters(widget.getId(), dialog.getValues())
             except Exception, e:
                 if "Comar.PolicyKit" in e._dbus_error_name:
-                    kdeui.KMessageBox.error(self, kdecore.i18n("Access denied."))
+                    createMessage(self,"Error","Access denied.")
                 else:
-                    kdeui.KMessageBox.error(self, unicode(e))
+                    createMessage(self,"Error", unicode(e))
 
     def slotItemDelete(self):
         """
@@ -314,9 +312,9 @@
             self.iface.setState(state)
         except Exception, e:
             if "Comar.PolicyKit" in e._dbus_error_name:
-                kdeui.KMessageBox.error(self, kdecore.i18n("Access denied."))
+                createMessage(self,"Error", "Access denied.")
             else:
-                kdeui.KMessageBox.error(self, unicode(e))
+                createMessage(self,"Error", unicode(e))
         self.widgetService.setEnabled(True)
 
     def slotAnimate(self, frame):
diff -Nuar firewall-manager-3.0.0//src/firewallmanager/pagedialog.py managers/firewall-manager//src/firewallmanager/pagedialog.py
--- firewall-manager-3.0.0//src/firewallmanager/pagedialog.py	2010-07-06 14:16:34.000000000 +0300
+++ managers/firewall-manager//src/firewallmanager/pagedialog.py	2011-03-29 16:03:27.699000000 +0300
@@ -15,30 +15,77 @@
 from PyQt4 import QtCore
 from PyQt4 import QtGui
 
-# PyKDE
-from PyKDE4 import kdeui
-from PyKDE4 import kdecore
-
 # Settings item widget
 from firewallmanager.settingsitem import SettingsItemWidget
-
-
-class PageDialog(kdeui.KPageDialog):
-    def __init__(self, parent, parameters, savedParameters):
-        kdeui.KPageDialog.__init__(self, parent)
-
-        self.setFaceType(kdeui.KPageDialog.Tabbed)
-        self.setCaption(kdecore.i18n("Settings"))
-
-
-        self.page_widget = PageWidget(self, parameters, savedParameters)
-        self.page_item = kdeui.KPageWidgetItem(self.page_widget, kdecore.i18n("Settings"))
-
-        self.addPage(self.page_item)
-
-    def getValues(self):
-        return self.page_widget.getValues()
-
+#Context
+from context import *
+import context as ctx
+
+# Config
+from firewallmanager.config import ANIM_SHOW, ANIM_HIDE, ANIM_TARGET, ANIM_DEFAULT, ANIM_TIME
+
+
+if ctx.Pds.session == ctx.pds.Kde4:
+
+    # PyKDE
+    from PyKDE4 import kdeui
+    from PyKDE4 import kdecore
+
+    class PageDialog(kdeui.KPageDialog):
+        def __init__(self, parent, parameters, savedParameters):
+            kdeui.KPageDialog.__init__(self, parent)
+
+            self.setFaceType(kdeui.KPageDialog.Tabbed)
+            self.setCaption(kdecore.i18n("Settings"))
+
+
+            self.page_widget = PageWidget(self, parameters, savedParameters)
+            self.page_item = kdeui.KPageWidgetItem(self.page_widget, kdecore.i18n("Settings"))
+
+            self.addPage(self.page_item)
+
+        def getValues(self):
+            return self.page_widget.getValues()
+
+else :
+
+    class PageDialog(QtGui.QDialog):
+        def __init__(self, parent, parameters, savedParameters):
+            self.animationLast = ANIM_HIDE
+            QtGui.QDialog.__init__(self,parent)
+            self.setWindowTitle(i18n("Settings"))
+            self.resize(548,180)
+            self.page_widget = PageWidget(self, parameters,savedParameters)
+            self.tab=QtGui.QTabWidget(self)
+            self.tab.addTab(self.page_widget,i18n("Settings"))
+            self.buttonBox = QtGui.QDialogButtonBox(self)
+            self.buttonBox.setGeometry(QtCore.QRect(4, 152, 540, 25))
+            self.buttonBox.setStandardButtons(QtGui.QDialogButtonBox.Cancel|QtGui.QDialogButtonBox.Ok)
+            self.layout=QtGui.QVBoxLayout(self)
+            self.layout.addWidget(self.tab)
+            self.layout.addWidget(self.buttonBox)
+            self.buttonBox.setObjectName(i18n("buttonBox"))
+            QtCore.QObject.connect(self.buttonBox, QtCore.SIGNAL(i18n("accepted()")), self.accept)
+            QtCore.QObject.connect(self.buttonBox, QtCore.SIGNAL(i18n("rejected()")),self.reject)
+            QtCore.QMetaObject.connectSlotsByName(self)
+
+        def getValues(self):
+            return self.page_widget.getValues()
+
+        def hideEditBox(self):
+            if self.animationLast == ANIM_SHOW:
+                self.animationLast = ANIM_HIDE
+                # Set range
+                self.animator.setFrameRange(self.frameEdit.height(), ANIM_TARGET)
+                # Go go go!
+                self.animator.start()
+
+        def slotCancelEdit(self):
+            self.hideEditBox()
+        
+        def slotSaveEdit(self):
+            # Hide edit box
+            self.hideEditBox()
 
 class PageWidget(QtGui.QWidget):
     def __init__(self, parent, parameters=[], saved={}):
diff -Nuar firewall-manager-3.0.0//src/firewallmanager/service.py managers/firewall-manager//src/firewallmanager/service.py
--- firewall-manager-3.0.0//src/firewallmanager/service.py	2010-07-06 14:16:34.000000000 +0300
+++ managers/firewall-manager//src/firewallmanager/service.py	2011-03-29 16:03:27.701000000 +0300
@@ -15,10 +15,8 @@
 from PyQt4 import QtCore
 from PyQt4 import QtGui
 
-# PyKDE
-from PyKDE4 import kdeui
-from PyKDE4 import kdecore
-
+#Context
+from context import *
 # UI
 from firewallmanager.ui_service import Ui_ServiceWidget
 
@@ -35,15 +33,15 @@
     def setState(self, state):
         self.state = state
         if state:
-            self.labelStatus.setText(kdecore.i18n("Firewall is activated."))
-            self.labelIcon.setPixmap(kdeui.KIcon("document-encrypt").pixmap(48, 48))
-            self.pushToggle.setIcon(kdeui.KIcon("media-playback-stop"))
-            self.pushToggle.setText(kdecore.i18n("Stop"))
+            self.labelStatus.setText(i18n("Firewall is activated."))
+            self.labelIcon.setPixmap(KIcon("document-encrypt").pixmap(48, 48))
+            self.pushToggle.setIcon(KIcon("media-playback-stop"))
+            self.pushToggle.setText(i18n("Stop"))
         else:
-            self.labelStatus.setText(kdecore.i18n("Firewall is deactivated."))
-            self.labelIcon.setPixmap(kdeui.KIcon("document-decrypt").pixmap(48, 48))
-            self.pushToggle.setIcon(kdeui.KIcon("media-playback-start"))
-            self.pushToggle.setText(kdecore.i18n("Start"))
+            self.labelStatus.setText(i18n("Firewall is deactivated."))
+            self.labelIcon.setPixmap(KIcon("document-decrypt").pixmap(48, 48))
+            self.pushToggle.setIcon(KIcon("media-playback-start"))
+            self.pushToggle.setText(i18n("Start"))
 
     def getState(self):
         return self.state
diff -Nuar firewall-manager-3.0.0//src/firewallmanager/settingsitem.py managers/firewall-manager//src/firewallmanager/settingsitem.py
--- firewall-manager-3.0.0//src/firewallmanager/settingsitem.py	2010-07-06 14:16:34.000000000 +0300
+++ managers/firewall-manager//src/firewallmanager/settingsitem.py	2011-03-29 16:03:27.700000000 +0300
@@ -1,3 +1,5 @@
+#ifndef SETTINGSITEM.PY
+#define SETTINGSITEM.PY
 #!/usr/bin/python
 # -*- coding: utf-8 -*-
 #
@@ -14,10 +16,8 @@
 # PyQt
 from PyQt4 import QtCore
 from PyQt4 import QtGui
-
-# PyKDE
-#from PyKDE4.kdeui import KIcon
-
+from context import *
+from PyQt4.QtCore import SIGNAL
 # UI
 from firewallmanager.ui_settingsitem import Ui_SettingsItemWidget
 
@@ -33,7 +33,7 @@
         self.lineItem.hide()
         self.comboItems.hide()
         self.listItems.hide()
-
+        self.setDisabledAll()
         if type_ == "combo":
             self.comboItems.show()
         elif type_ == "editlist":
@@ -41,6 +41,99 @@
         elif type_ == "text":
             self.lineItem.show()
 
+        self.connect(self.pushAdd, QtCore.SIGNAL("clicked()"), self.addItemToList)
+        self.connect(self.pushDelete, QtCore.SIGNAL("clicked()"), self.removeItemToList)
+        self.connect(self.pushUp, QtCore.SIGNAL("clicked()"), self.funcpushUp)
+        self.connect(self.pushDown, QtCore.SIGNAL("clicked()"), self.funcpushDown)
+        QtCore.QObject.connect(self.listWidget, QtCore.SIGNAL(("currentItemChanged(QListWidgetItem*,QListWidgetItem*)")), self.HideButtons)
+        QtCore.QObject.connect(self.lineEdit, QtCore.SIGNAL(("textChanged(QString)")), self.hideAdd)
+        self.pushAdd.setIcon(KIcon("list-add"))
+        self.pushDelete.setIcon(KIcon("list-remove"))
+        self.pushUp.setIcon(KIcon("arrow-up"))
+        self.pushDown.setIcon(KIcon("arrow-down"))
+
+    def setDisabledAll(self):
+        self.pushAdd.setEnabled(0)
+        self.pushDelete.setEnabled(0)
+        self.pushUp.setEnabled(0)
+        self.pushDown.setEnabled(0)
+
+    def hideAdd(self):
+        if not(self.lineEdit.text()==""):
+            self.pushAdd.setEnabled(1)
+        else :
+            self.pushAdd.setEnabled(0)
+
+    def alreadyInList(self):
+        for i in range(self.listWidget.count()):
+            if (self.lineEdit.text()==self.listWidget.item(i).text()):
+                return False 
+        return True
+
+    def addItemToList(self):
+        if (self.alreadyInList()):
+            if self.listWidget.currentItem():
+                self.listWidget.currentItem().setText(self.lineEdit.text())
+            else:
+                if not(self.lineEdit.text()== ""):
+                    self.listWidget.insertItem(0,self.lineEdit.text())
+        self.lineEdit.setText("")
+        self.listWidget.setCurrentItem(None)
+        self.setDisabledAll()
+
+    def removeItemToList(self):
+        self.listWidget.takeItem(self.listWidget.currentRow())
+        if (self.listWidget.count()==0):
+            self.pushDelete.setEnabled(0)
+        if (self.listWidget.count()<2):
+            self.pushUp.setEnabled(0)
+            self.pushDown.setEnabled(0)
+
+    def listToLineEdit(self):
+        if (self.listWidget.currentItem()):
+            self.lineEdit.setText(self.listWidget.currentItem().text())
+        else:
+            self.lineEdit.setText("")
+
+    def HideButtons(self):
+        self.listToLineEdit()
+        if not(self.lineEdit.text()==""):
+            self.pushAdd.setEnabled(1)
+        if self.listWidget.currentRow() == 0 :
+            self.pushUp.setEnabled(0)
+            self.pushDown.setEnabled(1)
+        elif self.listWidget.currentRow() == self.listWidget.count()-1:
+            self.pushUp.setEnabled(1)
+            self.pushDown.setEnabled(0)
+        else:
+            self.pushDown.setEnabled(1)
+            self.pushUp.setEnabled(1)
+        if self.listWidget.currentRow :
+            self.pushDelete.setEnabled(1)
+        if (self.listWidget.count()<2):
+            self.pushUp.setEnabled(0)
+            self.pushDown.setEnabled(0)
+
+    #TODO Create backend file for listWidget 
+    def funcpushDown(self):
+        self.listWidget.setCurrentRow(self.listWidget.currentRow()+1)
+        degisken = self.listWidget.currentItem().text()
+        self.listWidget.setCurrentRow(self.listWidget.currentRow()-1)
+        degisken_ = self.listWidget.currentItem().text()
+        self.listWidget.currentItem().setText(degisken)
+        self.listWidget.setCurrentRow(self.listWidget.currentRow()+1)
+        self.listWidget.currentItem().setText(degisken_)
+        self.listToLineEdit()
+
+    def funcpushUp(self):
+        self.listWidget.setCurrentRow(self.listWidget.currentRow()-1)
+        degisken = self.listWidget.currentItem().text()
+        self.listWidget.setCurrentRow(self.listWidget.currentRow()+1)
+        degisken_ = self.listWidget.currentItem().text()
+        self.listWidget.currentItem().setText(degisken)
+        self.listWidget.setCurrentRow(self.listWidget.currentRow()-1)
+        self.listWidget.currentItem().setText(degisken_)
+        self.listToLineEdit()
     def setTitle(self, title):
         self.labelTitle.setText(unicode(title))
 
@@ -51,7 +144,7 @@
                     name, label = item.split("\t")
                     self.comboItems.addItem(label, QtCore.QVariant(name))
             elif key == "format" and self.type in ["editlist", "text"]:
-                editor = self.listItems.lineEdit()
+                editor = self.lineEdit
                 validator = QtGui.QRegExpValidator(QtCore.QRegExp(value), self)
                 editor.setValidator(validator)
 
@@ -64,7 +157,7 @@
             self.comboItems.setCurrentIndex(index)
         elif self.type == "editlist":
             for item in value.split():
-                self.listItems.insertItem(unicode(item))
+                self.listWidget.insertItem(0,unicode(item))
         elif self.type == "text":
             self.lineItem.setText(unicode(value))
 
@@ -74,8 +167,9 @@
             return unicode(self.comboItems.itemData(index).toString())
         elif self.type == "editlist":
             items = []
-            for index in range(self.listItems.count()):
-                items.append(unicode(self.listItems.text(index)))
+            for index in range(self.listWidget.count()):
+                items.append(unicode(self.listWidget.item(index).text()))
             return " ".join(items)
         elif self.type == "text":
             return unicode(self.lineItem.text())
+#endif // SETTINGSITEM.PY
diff -Nuar firewall-manager-3.0.0//src/firewallmanager/standalone.py managers/firewall-manager//src/firewallmanager/standalone.py
--- firewall-manager-3.0.0//src/firewallmanager/standalone.py	1970-01-01 02:00:00.000000000 +0200
+++ managers/firewall-manager//src/firewallmanager/standalone.py	2011-03-29 16:03:27.698000000 +0300
@@ -0,0 +1,29 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# PyKDE4 Stuff
+from PyKDE4.kdeui import *
+from PyKDE4.kdecore import KGlobal
+from main import * 
+class FirewallManager(KMainWindow):
+    
+    def __init__(self, parent=None):
+        KMainWindow.__init__(self, parent)
+        self.setWindowIcon(KIcon("security-high"))
+        widget = MainWidget(self)
+        KGlobal.locale().insertCatalog("firewall-manager")
+        self.resize(widget.size())
+        self.setCentralWidget(widget)
+
+class Module(KCModule):
+    def __init__(self, component_data, parent):
+        KCModule.__init__(self, component_data, parent)
+
+        KGlobal.locale().insertCatalog(catalog)
+
+        if not dbus.get_default_main_loop():
+            from dbus.mainloop.qt import DBusQtMainLoop
+            DBusQtMainLoop(set_as_default=True)
+
+        MainWidget(self, embed=True)
+
diff -Nuar firewall-manager-3.0.0//src/firewall-manager.py managers/firewall-manager//src/firewall-manager.py
--- firewall-manager-3.0.0//src/firewall-manager.py	2010-07-06 14:16:34.000000000 +0300
+++ managers/firewall-manager//src/firewall-manager.py	2011-03-29 16:03:27.706000000 +0300
@@ -11,52 +11,67 @@
 # Please read the COPYING file.
 #
 
+#SyStem
+
 import sys
 import dbus
 
-from PyQt4 import QtGui
-from PyQt4 import QtCore
-
-from PyKDE4.kdeui import KMainWindow, KApplication, KCModule, KIcon
-from PyKDE4.kdecore import KCmdLineArgs, KGlobal
+import firewallmanager.context as ctx
+from firewallmanager.context import *
 
-from firewallmanager.about import aboutData, catalog
-from firewallmanager.main import MainWidget
+#Qt
 
-class MainWindow(KMainWindow):
+from PyQt4 import QtGui
+from PyQt4 import QtCore
+from PyQt4.QtCore import *
+class MainWindow(QtGui.QMainWindow):
     def __init__(self, parent=None):
-        KMainWindow.__init__(self, parent)
+        QtGui.QMainWindow.__init__(self, parent)
         widget = MainWidget(self)
         self.resize(widget.size())
         self.setCentralWidget(widget)
-
-
+        self.qtrans=QtCore.QTranslator()
 if __name__ == "__main__":
 
-    KCmdLineArgs.init(sys.argv, aboutData)
-    app = KApplication()
 
     if not dbus.get_default_main_loop():
         from dbus.mainloop.qt import DBusQtMainLoop
         DBusQtMainLoop(set_as_default=True)
 
-    window = MainWindow()
-    window.show()
-
-    app.exec_()
-
-
-class Module(KCModule):
-    def __init__(self, component_data, parent):
-        KCModule.__init__(self, component_data, parent)
-
-        KGlobal.locale().insertCatalog(catalog)
-
-        if not dbus.get_default_main_loop():
-            from dbus.mainloop.qt import DBusQtMainLoop
-            DBusQtMainLoop(set_as_default=True)
-
-        MainWidget(self, embed=True)
+    if ctx.Pds.session == ctx.pds.Kde4:
+        
+        #PyKde4
+        from PyKDE4.kdeui import KMainWindow, KApplication, KCModule, KIcon
+        from PyKDE4.kdecore import KCmdLineArgs, KGlobal
+        from firewallmanager.about import aboutData, catalog
+        from firewallmanager.standalone import FirewallManager
+        #Set Commandline arguments
+        KCmdLineArgs.init(sys.argv, aboutData)
+        #Create a Kapplication instance
+        app = KApplication()
+        window = FirewallManager()
+        window.show()
+
+        app.exec_()
+    else:
+
+        import gettext
+        
+        __trans = gettext.translation('firewall-manager', fallback=True)
+        i18n = __trans.ugettext
+
+        from firewallmanager.main import MainWidget
+        from pds.quniqueapp import QUniqueApplication
+
+        app = QUniqueApplication(sys.argv, catalog="firewall-manager")
+
+        mainWindow = MainWidget(None)
+        mainWindow.show()
+        mainWindow.resize(640, 480)
+        mainWindow.setWindowTitle(i18n("Firewall Manager"))
+        mainWindow.setWindowIcon(KIcon("security-high"))
+        app.connect(app, SIGNAL('lastWindowClosed()'), app.quit)
+        app.exec_()
 
 def CreatePlugin(widget_parent, parent, component_data):
     return Module(component_data, parent)
diff -Nuar firewall-manager-3.0.0//ui/main.ui managers/firewall-manager//ui/main.ui
--- firewall-manager-3.0.0//ui/main.ui	2010-07-06 14:16:34.000000000 +0300
+++ managers/firewall-manager//ui/main.ui	2011-03-29 16:03:27.757000000 +0300
@@ -55,7 +55,7 @@
          </spacer>
         </item>
         <item>
-         <widget class="KComboBox" name="comboFilter"/>
+         <widget class="QComboBox" name="comboFilter"/>
         </item>
        </layout>
       </item>
@@ -97,8 +97,8 @@
           <rect>
            <x>0</x>
            <y>0</y>
-           <width>528</width>
-           <height>192</height>
+           <width>526</width>
+           <height>189</height>
           </rect>
          </property>
          <layout class="QGridLayout" name="gridLayout">
@@ -121,15 +121,7 @@
    </item>
   </layout>
  </widget>
- <customwidgets>
-  <customwidget>
-   <class>KComboBox</class>
-   <extends>QComboBox</extends>
-   <header>kcombobox.h</header>
-  </customwidget>
- </customwidgets>
  <tabstops>
-  <tabstop>comboFilter</tabstop>
   <tabstop>listItems</tabstop>
   <tabstop>buttonBox</tabstop>
  </tabstops>
diff -Nuar firewall-manager-3.0.0//ui/settingsitem.ui managers/firewall-manager//ui/settingsitem.ui
--- firewall-manager-3.0.0//ui/settingsitem.ui	2010-07-06 14:16:34.000000000 +0300
+++ managers/firewall-manager//ui/settingsitem.ui	2011-03-29 16:03:27.758000000 +0300
@@ -6,52 +6,96 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>530</width>
-    <height>256</height>
+    <width>546</width>
+    <height>432</height>
    </rect>
   </property>
   <property name="windowTitle">
    <string notr="true">SettingsItemWidget</string>
   </property>
-  <layout class="QHBoxLayout" name="horizontalLayout">
-   <item>
-    <widget class="QLabel" name="labelTitle">
-     <property name="minimumSize">
-      <size>
-       <width>120</width>
-       <height>0</height>
-      </size>
-     </property>
-     <property name="text">
-      <string notr="true">labelTitle:</string>
-     </property>
-     <property name="wordWrap">
-      <bool>true</bool>
-     </property>
-    </widget>
-   </item>
-   <item>
-    <layout class="QVBoxLayout" name="verticalLayout">
-     <item>
-      <widget class="QLineEdit" name="lineItem"/>
-     </item>
+  <layout class="QFormLayout" name="formLayout">
+   <property name="fieldGrowthPolicy">
+    <enum>QFormLayout::ExpandingFieldsGrow</enum>
+   </property>
+   <item row="0" column="0">
+    <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
-      <widget class="KEditListBox" name="listItems"/>
+      <widget class="QLabel" name="labelTitle">
+       <property name="minimumSize">
+        <size>
+         <width>120</width>
+         <height>0</height>
+        </size>
+       </property>
+       <property name="text">
+        <string notr="true">labelTitle:</string>
+       </property>
+       <property name="wordWrap">
+        <bool>true</bool>
+       </property>
+      </widget>
      </item>
      <item>
-      <widget class="QComboBox" name="comboItems"/>
+      <layout class="QVBoxLayout" name="verticalLayout01">
+       <item>
+        <widget class="QLineEdit" name="lineItem"/>
+       </item>
+       <item>
+        <widget class="QGroupBox" name="listItems">
+         <property name="title">
+          <string/>
+         </property>
+         <layout class="QFormLayout" name="formLayout_2">
+          <item row="0" column="0" colspan="2">
+           <widget class="QLineEdit" name="lineEdit"/>
+          </item>
+          <item row="1" column="0">
+           <widget class="QListWidget" name="listWidget"/>
+          </item>
+          <item row="1" column="1">
+           <layout class="QVBoxLayout" name="verticalLayout_02">
+            <item>
+             <widget class="QPushButton" name="pushAdd">
+              <property name="text">
+               <string>Add</string>
+              </property>
+             </widget>
+            </item>
+            <item>
+             <widget class="QPushButton" name="pushDelete">
+              <property name="text">
+               <string>Remove</string>
+              </property>
+             </widget>
+            </item>
+            <item>
+             <widget class="QPushButton" name="pushUp">
+              <property name="text">
+               <string>Move Up</string>
+              </property>
+             </widget>
+            </item>
+            <item>
+             <widget class="QPushButton" name="pushDown">
+              <property name="text">
+               <string>Move Down</string>
+              </property>
+             </widget>
+            </item>
+           </layout>
+          </item>
+         </layout>
+        </widget>
+       </item>
+       <item>
+        <widget class="QComboBox" name="comboItems"/>
+       </item>
+      </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
- <customwidgets>
-  <customwidget>
-   <class>KEditListBox</class>
-   <extends>QGroupBox</extends>
-   <header>keditlistbox.h</header>
-  </customwidget>
- </customwidgets>
  <resources/>
  <connections/>
 </ui>
