diff -up lxml-2.2.6/src/lxml/html/__init__.py.fix-2to3 lxml-2.2.6/src/lxml/html/__init__.py
--- lxml-2.2.6/src/lxml/html/__init__.py.fix-2to3   2010-07-26 13:36:42.333672181 -0400
+++ lxml-2.2.6/src/lxml/html/__init__.py    2010-07-26 13:36:52.379671732 -0400
@@ -823,7 +823,8 @@ def submit_form(form, extra_values=None,
 def open_http_urllib(method, url, values):
     ## FIXME: should test that it's not a relative URL or something
     try:
-        from urllib import urlencode, urlopen
+        from urllib import urlencode
+        from urllib import urlopen
     except ImportError: # Python 3
         from urllib.request import urlopen
         from urllib.parse import urlencode
