---
 Makefile.in    |   10 +++++-----
 doc/quilt.1.in |    2 +-
 2 files changed, 6 insertions(+), 6 deletions(-)

--- quilt-0.48.orig/Makefile.in
+++ quilt-0.48/Makefile.in
@@ -266,7 +266,7 @@ $(patsubst %.in,%,$(wildcard bin/*.in qu
 		-e 's:@VERSION''@:$(VERSION):g'				\
 		-e 's:@RELEASE''@:$(RELEASE):g'				\
 		-e 's:@LOCALEDIR''@:$(localedir):g'			\
-		-e 's:@DOCSUBDIR''@:$(docdir)/$(PACKAGE)-$(VERSION):g'	\
+		-e 's:@DOCSUBDIR''@:$(docdir)/$(PACKAGE):g'		\
 		$< > $@
 	@$(if $(filter-out $<,$(NON_EXEC_IN)),chmod +x $@)
 
@@ -325,11 +325,11 @@ endif
 	$(INSTALL) -d $(BUILD_ROOT)$(libdir)/$(PACKAGE)
 	$(INSTALL) -m 755 $(LIB:%=lib/%) $(BUILD_ROOT)$(libdir)/$(PACKAGE)/
 
-	$(INSTALL) -d $(BUILD_ROOT)$(docdir)/$(PACKAGE)-$(VERSION)/
+	$(INSTALL) -d $(BUILD_ROOT)$(docdir)/$(PACKAGE)/
 	$(INSTALL) -m 644 doc/README					\
-		   $(BUILD_ROOT)$(docdir)/$(PACKAGE)-$(VERSION)/
+		   $(BUILD_ROOT)$(docdir)/$(PACKAGE)/
 	$(INSTALL) -m 644 doc/quilt.pdf doc/README.MAIL			\
-		   $(BUILD_ROOT)$(docdir)/$(PACKAGE)-$(VERSION)/
+		   $(BUILD_ROOT)$(docdir)/$(PACKAGE)/
 
 	$(INSTALL) -d $(BUILD_ROOT)$(mandir)/man1
 	$(INSTALL) -m 644 $(MAN1) $(BUILD_ROOT)$(mandir)/man1/
@@ -367,7 +367,7 @@ uninstall ::
 		   $(notdir $(MAN1)))					\
 	       $(BUILD_ROOT)$(etcdir)/bash_completion.d/quilt		\
 	       $(BUILD_ROOT)$(etcdir)/quilt.quiltrc			\
-	       $(BUILD_ROOT)$(docdir)/$(PACKAGE)-$(VERSION)/		\
+	       $(BUILD_ROOT)$(docdir)/$(PACKAGE)/			\
 	       $(BUILD_ROOT)$(emacsdir)/quilt.el
 
 check: $(TESTS:test/%.test=test/.%.ok)
--- quilt-0.48.orig/doc/quilt.1.in
+++ quilt-0.48/doc/quilt.1.in
@@ -217,7 +217,7 @@ the pdf documentation, and in the help m
 
 .SH SEE ALSO
 
-The pdf documentation, which should be under /usr/share/doc/quilt/quilt.pdf.
+The pdf documentation, which should be under /usr/share/doc/packages/quilt/quilt.pdf.
 Note that some distributors compress this file. 
 .BR zxpdf ( 1 )
 can be used to display compressed pdf files.
