---
 Makefile.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- quilt-0.48.orig/Makefile.in	2009-01-31 18:09:47.000000000 +0100
+++ quilt-0.48/Makefile.in	2009-06-10 14:29:07.000000000 +0200
@@ -123,7 +123,7 @@ SRC +=		changes2changelog
 
 NON_EXEC_IN :=	doc/quilt.1 doc/README quilt/scripts/patchfns
 
-GIT_DESC :=	$(shell ./git-desc)
+GIT_DESC :=	$(shell test ! -x ./git-desc || ./git-desc)
 
 TESTS :=	$(filter-out test/patch-wrapper.test,$(wildcard test/*.test))
 DIRT +=		test/.depend $(wildcard test/.*.ok)
