Index: libcanberra-0.12/src/Makefile.am
===================================================================
--- libcanberra-0.12.orig/src/Makefile.am
+++ libcanberra-0.12/src/Makefile.am
@@ -58,7 +58,8 @@ libcanberra_la_CFLAGS = \
 	$(AM_CFLAGS) \
 	$(VORBIS_CFLAGS)
 libcanberra_la_LIBADD = \
-	$(VORBIS_LIBS)
+	$(VORBIS_LIBS) \
+	$(PTHREAD_LIBS)
 libcanberra_la_LDFLAGS = \
 	-export-dynamic \
 	-version-info $(LIBCANBERRA_VERSION_INFO)
@@ -290,6 +291,7 @@ libcanberra_gtk_la_CFLAGS = \
 	$(GTK_CFLAGS)
 libcanberra_gtk_la_LIBADD = \
 	$(GTK_LIBS) \
+	-lX11 \
 	libcanberra.la
 libcanberra_gtk_la_LDFLAGS = \
 	-export-dynamic -version-info $(LIBCANBERRA_GTK_VERSION_INFO)
