--- tv-viewer-0.8.2b1-orig/configure.tcl	2011-09-28 22:52:02.000000000 +0300
+++ tv-viewer-0.8.2b1/configure.tcl	2013-01-30 07:38:55.000000000 +0200
@@ -1,7 +1,7 @@
 #!/usr/bin/env tclsh
 
 #       configure.tcl
-#       © Copyright 2007-2011 Christian Rapp <christianrapp@users.sourceforge.net>
+#       © Copyright 2007-2013 Christian Rapp <christianrapp@users.sourceforge.net>
 #       
 #       This program is free software; you can redistribute it and/or modify
 #       it under the terms of the GNU General Public License as published by
@@ -174,7 +174,7 @@
 	puts "
 tv-viewer configure tcl script version [lindex $option(release_version) 0]"
 	puts "
-© Copyright 2007-2011 Christian Rapp <christianrapp@users.sourceforge.net>
+© Copyright 2007-2013 Christian Rapp <christianrapp@users.sourceforge.net>
 
 This script is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
@@ -332,7 +332,7 @@
 		exit 1
 	}
 	
-	set dependencies [dict create ivtv-tune ivtv-utils v4l2-ctl ivtv-utils mplayer MPlayer xdg-email xdg-utils xdg-open xdg-utils xdg-screensaver xdg-utils]
+	set dependencies [dict create ivtv-tune ivtv-utils v4l2-ctl ivtv-utils mplayer MPlayer xdg-email xdg-utils xdg-open xdg-utils xdg-screensaver xdg-utils sqlite3 sqlite]
 	
 	foreach {key elem} [dict get $dependencies] {
 		puts -nonewline $::printchan "$key "
@@ -357,6 +357,28 @@
 		}
 	}
 	
+	puts -nonewline $::printchan "sqlite3 Tcl extension "
+	set status_sql [catch {package require sqlite3} version_sql]
+	set i 0
+	while { $i != 3 } {
+		puts -nonewline $::printchan "*"
+		flush stdout
+		after 50
+		incr i
+	}
+	if {$status_sql == 0} {
+		puts $::printchan "\033\[0;1;32m OK\033\[0m"
+		puts $log "sqlite3 Tcl extension $version_sql OK"
+	} else {
+		puts $log "sqlite3 Tcl FAILED" 
+		puts $::printchan "\033\[0;1;31m FAILED\033\[0m"
+		puts $::printchan "
+TV-Viewer needs sqlite3 Tcl extension
+see the README for system requirements
+EXIT 1"
+		exit 1
+	}
+	
 	puts $::printchan "
 checking for optional dependencies
 "
@@ -481,7 +503,7 @@
 	after 250
 	
 	if {$tclkit == 1} {
-		configure_writeTclkitStarter "$where_is" "$prefix" "$eprefix" "$bindir" "$datadir" "$tclkit" "$log" "$tclkitbin"
+		configure_writeTclkitStarter "$where_is" "$prefix" "$eprefix" "$bindir" "$bintarget" "$datadir" "$tclkit" "$log" "$tclkitbin"
 	}
 	
 	if {[file exists $where_is/installer.tcl]} {
@@ -558,18 +580,18 @@
 	exit 0
 }
 
-proc configure_writeTclkitStarter {where_is prefix eprefix bindir datadir tclkit log tclkitbin} {
+proc configure_writeTclkitStarter {where_is prefix eprefix bindir bintarget datadir tclkit log tclkitbin} {
 	array set opt {
-		"##@@tv-viewer_sym" {if \[ "\$0" == "$bindir/tv-viewer" \]}
-		"##@@tv-viewer" {$datadir/tv-viewer/extensions/tclkit/$tclkitbin $datadir/tv-viewer/data/tv-viewer_main.tcl \$@ &}
-		"##@@tv-viewer_diag_sym" {if \[ "\$0" == "$bindir/tv-viewer_diag" \]}
-		"##@@tv-viewer_diag" {$datadir/tv-viewer/extensions/tclkit/$tclkitbin $datadir/tv-viewer/data/diag_runtime.tcl \$@ &}
-		"##@@tv-viewer_lirc_sym" {if \[ "\$0" == "$bindir/tv-viewer_lirc" \]}
-		"##@@tv-viewer_lirc" {$datadir/tv-viewer/extensions/tclkit/$tclkitbin $datadir/tv-viewer/data/lirc_emitter.tcl \$@ &}
-		"##@@tv-viewer_recext_sym" {if \[ "\$0" == "$bindir/tv-viewer_recext" \]}
-		"##@@tv-viewer_recext" {$datadir/tv-viewer/extensions/tclkit/$tclkitbin $datadir/tv-viewer/data/record_external.tcl \$@ &}
-		"##@@tv-viewer_scheduler_sym" {if \[ "\$0" == "$bindir/tv-viewer_scheduler" \]}
-		"##@@tv-viewer_scheduler" {$datadir/tv-viewer/extensions/tclkit/$tclkitbin $datadir/tv-viewer/data/scheduler.tcl \$@ &}
+		"##@@tv-viewer_sym" {if \[ "\${0##*/}" == "tv-viewer" \]}
+		"##@@tv-viewer" {$bintarget/extensions/tclkit/$tclkitbin $bintarget/data/tv-viewer_main.tcl \$@ &}
+		"##@@tv-viewer_diag_sym" {if \[ "\${0##*/}" == "tv-viewer_diag" \]}
+		"##@@tv-viewer_diag" {$bintarget/extensions/tclkit/$tclkitbin $bintarget/data/diag_runtime.tcl \$@ &}
+		"##@@tv-viewer_lirc_sym" {if \[ "\${0##*/}" == "tv-viewer_lirc" \]}
+		"##@@tv-viewer_lirc" {$bintarget/extensions/tclkit/$tclkitbin $bintarget/data/lirc_emitter.tcl \$@ &}
+		"##@@tv-viewer_recext_sym" {if \[ "\${0##*/}" == "tv-viewer_recext" \]}
+		"##@@tv-viewer_recext" {$bintarget/extensions/tclkit/$tclkitbin $bintarget/data/record_external.tcl \$@ &}
+		"##@@tv-viewer_scheduler_sym" {if \[ "\${0##*/}" == "tv-viewer_scheduler" \]}
+		"##@@tv-viewer_scheduler" {$bintarget/extensions/tclkit/$tclkitbin $bintarget/data/scheduler.tcl \$@ &}
 		"#tclkitstarter.sh.in @@" "#!/bin/bash"
 	}
 	
