diff -Naur guake-0.4.2/src/guake guake-0.4.2-patched/src/guake
--- guake-0.4.2/src/guake	2010-06-23 20:13:01.000000000 +0300
+++ guake-0.4.2-patched/src/guake	2011-10-19 15:20:41.647240729 +0300
@@ -430,6 +430,16 @@
             self.accel_group.connect_group(key, mask, gtk.ACCEL_VISIBLE,
                                            self.guake.accel_toggle_fullscreen)
 
+        key, mask = gtk.accelerator_parse('<Shift>Up')
+        if key > 0:
+            self.accel_group.connect_group(key, mask, gtk.ACCEL_VISIBLE,
+                                           self.guake.accel_up)
+
+        key, mask = gtk.accelerator_parse('<Shift>Down')
+        if key > 0:
+            self.accel_group.connect_group(key, mask, gtk.ACCEL_VISIBLE,
+                                           self.guake.accel_down)
+
 class GuakeTerminal(vte.Terminal):
     """Just a vte.Terminal with some properties already set.
     """
@@ -972,6 +982,26 @@
             self.client.notify(KEY('/general/show_resizer'))
 
         return True
+        
+    def accel_up(self, *args):
+        """Callback to override vte.terminal default scrolling on Arrow Up
+        """
+        terminal = self.term_list[self.notebook.get_current_page()]
+        adjustment = terminal.get_adjustment()
+        adjustment.set_value(adjustment.get_value() - 1)
+
+        return True
+
+    def accel_down(self, *args):
+        """Callback to override vte.terminal default scrolling on Arrow Down
+        """
+        terminal = self.term_list[self.notebook.get_current_page()]
+        adjustment = terminal.get_adjustment()
+
+        value = min(adjustment.get_value() + 1, adjustment.upper - adjustment.page_size)
+        adjustment.set_value(value)
+
+        return True
 
     # -- callbacks --
 
