--- data/guake.desktop.in.OSmanOS	2012-07-26 11:47:08.000000000 +0300
+++ data/guake.desktop.in		2012-11-02 17:38:16.105881804 +0200
@@ -5,11 +5,14 @@
 Name[pt_BR]=Guake Terminal
 Name[fr]=Guake Terminal
 Name[fr_FR]=Guake Terminal
+Name[tr]=Guake
+GenericName[tr]=Guake Uçbirim
 Comment=Use the command line in a Quake-like terminal
 Comment[pt]=Utilizar a linha de comando em um terminal estilo Quake
 Comment[pt_BR]=Utilizar a linha de comando em um terminal estilo Quake
 Comment[fr]=Utilisez la ligne de commande comme dans un terminal quake
 Comment[fr_FR]=Utilisez la ligne de commande comme dans un terminal quake
+Comment[tr]=Quake benzeri uçbirimde komut satırı kullanın
 TryExec=guake
 Exec=guake
 Icon=guake