--- guake-0.4.2/data/guake.schemas.old	2012-05-19 23:28:43.234945651 -0500
+++ guake-0.4.2/data/guake.schemas	2012-05-19 23:27:34.901616869 -0500
@@ -426,6 +426,42 @@
             </locale>
         </schema>
 
+	<schema>
+            <key>/schemas/apps/guake/keybindings/local/fontsize_decrease</key>
+            <applyto>/apps/guake/keybindings/local/fontsize_decrease</applyto>
+            <owner>guake</owner>
+            <type>string</type>
+            <default>&lt;Control&gt;-</default>
+            <locale name="C">
+                <short>Decrease fontsize.</short>
+                <long>Decrease fontsize by 2 units.</long>
+            </locale>
+        </schema>
+
+	<schema>
+            <key>/schemas/apps/guake/keybindings/local/fontsize_increase</key>
+            <applyto>/apps/guake/keybindings/local/fontsize_increase</applyto>
+            <owner>guake</owner>
+            <type>string</type>
+            <default>&lt;Control&gt;+</default>
+            <locale name="C">
+                <short>Increase fontsize.</short>
+                <long>Increase fontsize by 2 units.</long>
+            </locale>
+        </schema>
+
+	<schema>
+            <key>/schemas/apps/guake/keybindings/local/fontsize_reset</key>
+            <applyto>/apps/guake/keybindings/local/fontsize_reset</applyto>
+            <owner>guake</owner>
+            <type>string</type>
+            <default>&lt;Control&gt;0</default>
+            <locale name="C">
+                <short>Set the default fontsize.</short>
+                <long>Set the default fontsize.</long>
+            </locale>
+        </schema>
+
         <!-- These options are not shown in guake-prefs window to
              avoid bloating that pretty ui =) -->
 
