From afe87841d1149de2c28c5a7905fc74a865f53e13 Mon Sep 17 00:00:00 2001
From: Thomas Hunkapiller <furydevoid@gmail.com>
Date: Fri, 29 Jun 2012 15:21:40 -0500
Subject: [PATCH] fix ignored ctrl key in global hotkeys

---
 src/globalhotkeys/eggaccelerators.c |   20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/src/globalhotkeys/eggaccelerators.c b/src/globalhotkeys/eggaccelerators.c
index 6f96411..f53eb4b 100644
--- a/src/globalhotkeys/eggaccelerators.c
+++ b/src/globalhotkeys/eggaccelerators.c
@@ -129,6 +129,20 @@ is_control (const gchar *string)
 }
 
 static inline gboolean
+is_new_control (const gchar *string)
+{
+  return ((string[0] == '<') &&
+	  (string[1] == 'p' || string[1] == 'P') &&
+	  (string[2] == 'r' || string[2] == 'R') &&
+	  (string[3] == 'i' || string[3] == 'I') &&
+	  (string[4] == 'm' || string[4] == 'M') &&
+	  (string[5] == 'a' || string[5] == 'A') &&
+	  (string[6] == 'r' || string[6] == 'R') &&
+	  (string[7] == 'y' || string[7] == 'Y') &&
+	  (string[8] == '>'));
+}
+
+static inline gboolean
 is_release (const gchar *string)
 {
   return ((string[0] == '<') &&
@@ -250,6 +264,12 @@ egg_accelerator_parse_virtual (const gchar            *accelerator,
 	      len -= 9;
 	      mods |= EGG_VIRTUAL_CONTROL_MASK;
 	    }
+	  else if (len >= 9 && is_new_control (accelerator))
+	    {
+	      accelerator += 9;
+	      len -= 9;
+	      mods |= EGG_VIRTUAL_CONTROL_MASK;
+	    }
 	  else if (len >= 7 && is_shift (accelerator))
 	    {
 	      accelerator += 7;
-- 
1.7.10

