diff --git a/data/guake.glade b/data/guake.glade
index 08124d7..89f5564 100644
--- a/data/guake.glade
+++ b/data/guake.glade
@@ -5,6 +5,7 @@
   <widget class="GtkWindow" id="window-root">
     <property name="has_focus">True</property>
     <property name="title" translatable="yes">Guake!</property>
+    <property name="type_hint">dock</property>
     <property name="skip_taskbar_hint">True</property>
     <property name="skip_pager_hint">True</property>
     <property name="urgency_hint">True</property>
