From 42bdc1fb0f98c09760474ed3a9a705da9a1913c9 Mon Sep 17 00:00:00 2001
From: Lukas Elsner <open@mindrunner.de>
Date: Wed, 18 Jul 2012 10:17:37 +0200
Subject: [PATCH] fixed window height

---
 src/prefs.py |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/src/prefs.py b/src/prefs.py
index 9b833e4..a45c4fc 100644
--- a/src/prefs.py
+++ b/src/prefs.py
@@ -532,6 +532,9 @@ class PrefsDialog(SimpleGladeApp):
         value = self.client.get_int(KEY('/style/background/transparency'))
         self.get_widget('background_transparency').set_value(value)
 
+        value = self.client.get_int(KEY('/general/window_height'))
+        self.get_widget('window_height').set_value(value)
+
         # it's a separated method, to be reused.
         self.reload_erase_combos()
 
-- 
1.7.8.6

