Index: File-RsyncP-0.70/FileList/Makefile.PL
===================================================================
--- File-RsyncP-0.70.orig/FileList/Makefile.PL
+++ File-RsyncP-0.70/FileList/Makefile.PL
@@ -4,7 +4,6 @@ use ExtUtils::MakeMaker;
 WriteMakefile(
     'NAME'	    => 'File::RsyncP::FileList',
     'VERSION_FROM'  => 'FileList.pm', # finds $VERSION
-    'LIBS'	    => ['-lm'], # e.g., '-lm'
     'DEFINE'	    => '-DPERL_BYTEORDER=$(BYTEORDER)',
     'INC'	    => '',     # e.g., '-I/usr/include/other' 
     'NORECURS'      => 1,
