Index: antiword-0.37/Makefile.Linux
===================================================================
--- antiword-0.37.orig/Makefile.Linux
+++ antiword-0.37/Makefile.Linux
@@ -2,8 +2,8 @@
 # Makefile for antiword (Linux version)
 #
 
-CC	= gcc
-LD	= gcc
+CC	= $(CC_Pardus)
+LD	= $(CC_Pardus)
 
 INSTALL		= cp -f
 INSTALL_PROGRAM	= $(INSTALL)
@@ -16,8 +16,8 @@ OPT	= -O2
 
 LDLIBS	=
 
-CFLAGS	= -Wall -pedantic $(OPT) -D$(DB)
-LDFLAGS	=
+CFLAGS	= $(CFLAGS_Pardus) $(OPT) -D$(DB)
+LDFLAGS	= $(LDFLAGS_Pardus)
 
 OBJS	=\
 	main_u.o asc85enc.o blocklist.o chartrans.o datalist.o depot.o\
