diff -Naur kding-0.5.2.orig/resources/kding.desktop kding-0.5.2/resources/kding.desktop
--- kding-0.5.2.orig/resources/kding.desktop	2009-09-08 22:03:03.761352659 +0300
+++ kding-0.5.2/resources/kding.desktop	2009-09-08 22:05:39.699353368 +0300
@@ -4,8 +4,10 @@
 Type=Application
 Name=KDing
 GenericName=German-English Translator
+GenericName[tr]=Almanca-İngilizce Çeviri Uygulaması
 GenericName[de]=Deutsch-Englisch Übersetzer
 Comment=Translator
+Comment[tr]=Çeviri Uygulaması
 Comment[de]=Übersetzungsprogramm
 Exec=kding
 Icon=kding
