Index: knemo-0.6.3/src/knemod/interfacetray.cpp
===================================================================
--- knemo-0.6.3.orig/src/knemod/interfacetray.cpp
+++ knemo-0.6.3/src/knemod/interfacetray.cpp
@@ -159,8 +159,8 @@ QString InterfaceTray::toolTipData()
     const BackendData * data = mInterface->getData();
     if ( !data )
         return QString();
-    QString leftTags = "<tr><td style='padding-right:1em; white-space:nowrap;'>";
-    QString centerTags = "</td><td style='white-space:nowrap;'>";
+    QString leftTags = "<tr><td style='padding-right:1em;'>";
+    QString centerTags = "</td><td style='white-space:nowrap;'>";
     QString rightTags = "</td></tr>";
 
     tipData = "<table cellspacing='2'>";
