diff -Nuar pitivi-0.13.4.orig/Makefile.in pitivi-0.13.4/Makefile.in
--- pitivi-0.13.4.orig/Makefile.in	2010-03-14 20:32:42.632136368 +0200
+++ pitivi-0.13.4/Makefile.in	2010-03-14 20:33:10.001880166 +0200
@@ -36,8 +36,7 @@
 DIST_COMMON = README $(am__configure_deps) $(srcdir)/Makefile.am \
 	$(srcdir)/Makefile.in $(srcdir)/pitivi.desktop.in.in \
 	$(srcdir)/pitivi.spec.in $(top_srcdir)/configure ABOUT-NLS \
-	AUTHORS COPYING ChangeLog INSTALL NEWS install-sh missing \
-	py-compile
+	AUTHORS COPYING ChangeLog INSTALL NEWS install-sh missing
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/common/m4/as-ac-expand.m4 \
 	$(top_srcdir)/common/m4/as-python.m4 \
diff -Nuar pitivi-0.13.4.orig/pitivi/elements/Makefile.in pitivi-0.13.4/pitivi/elements/Makefile.in
--- pitivi-0.13.4.orig/pitivi/elements/Makefile.in	2010-03-14 20:32:42.648125625 +0200
+++ pitivi-0.13.4/pitivi/elements/Makefile.in	2010-03-14 20:34:42.732868240 +0200
@@ -69,7 +69,6 @@
   sed '$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;s/\n/ /g' | \
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(elementsdir)"
-py_compile = $(top_srcdir)/py-compile
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 ACLOCAL = @ACLOCAL@
 ACLOCAL_AMFLAGS = @ACLOCAL_AMFLAGS@
@@ -266,14 +265,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(elementsdir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(elementsdir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(elementsdir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(elementsdir)" $$dlist; \
-	  fi; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-elementsPYTHON:
 	@$(NORMAL_UNINSTALL)
diff -Nuar pitivi-0.13.4.orig/pitivi/factories/Makefile.in pitivi-0.13.4/pitivi/factories/Makefile.in
--- pitivi-0.13.4.orig/pitivi/factories/Makefile.in	2010-03-14 20:32:42.646125370 +0200
+++ pitivi-0.13.4/pitivi/factories/Makefile.in	2010-03-14 20:34:14.762866832 +0200
@@ -69,7 +69,6 @@
   sed '$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;s/\n/ /g' | \
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(factoriesdir)"
-py_compile = $(top_srcdir)/py-compile
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 ACLOCAL = @ACLOCAL@
 ACLOCAL_AMFLAGS = @ACLOCAL_AMFLAGS@
@@ -263,14 +262,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(factoriesdir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(factoriesdir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(factoriesdir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(factoriesdir)" $$dlist; \
-	  fi; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-factoriesPYTHON:
 	@$(NORMAL_UNINSTALL)
diff -Nuar pitivi-0.13.4.orig/pitivi/formatters/Makefile.in pitivi-0.13.4/pitivi/formatters/Makefile.in
--- pitivi-0.13.4.orig/pitivi/formatters/Makefile.in	2010-03-14 20:32:42.650135098 +0200
+++ pitivi-0.13.4/pitivi/formatters/Makefile.in	2010-03-14 20:35:14.084875062 +0200
@@ -69,7 +69,6 @@
   sed '$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;s/\n/ /g' | \
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(formattersdir)"
-py_compile = $(top_srcdir)/py-compile
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 ACLOCAL = @ACLOCAL@
 ACLOCAL_AMFLAGS = @ACLOCAL_AMFLAGS@
@@ -262,14 +261,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(formattersdir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(formattersdir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(formattersdir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(formattersdir)" $$dlist; \
-	  fi; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-formattersPYTHON:
 	@$(NORMAL_UNINSTALL)
diff -Nuar pitivi-0.13.4.orig/pitivi/log/Makefile.in pitivi-0.13.4/pitivi/log/Makefile.in
--- pitivi-0.13.4.orig/pitivi/log/Makefile.in	2010-03-14 20:32:42.649143422 +0200
+++ pitivi-0.13.4/pitivi/log/Makefile.in	2010-03-14 20:34:58.565117219 +0200
@@ -69,7 +69,6 @@
   sed '$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;s/\n/ /g' | \
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(logdir)"
-py_compile = $(top_srcdir)/py-compile
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 ACLOCAL = @ACLOCAL@
 ACLOCAL_AMFLAGS = @ACLOCAL_AMFLAGS@
@@ -262,14 +261,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(logdir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(logdir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(logdir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(logdir)" $$dlist; \
-	  fi; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-logPYTHON:
 	@$(NORMAL_UNINSTALL)
diff -Nuar pitivi-0.13.4.orig/pitivi/Makefile.in pitivi-0.13.4/pitivi/Makefile.in
--- pitivi-0.13.4.orig/pitivi/Makefile.in	2010-03-14 20:32:42.642130169 +0200
+++ pitivi-0.13.4/pitivi/Makefile.in	2010-03-14 20:33:55.000884668 +0200
@@ -76,7 +76,6 @@
   sed '$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;s/\n/ /g' | \
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(pitividir)"
-py_compile = $(top_srcdir)/py-compile
 RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
   distclean-recursive maintainer-clean-recursive
 AM_RECURSIVE_TARGETS = $(RECURSIVE_TARGETS:-recursive=) \
@@ -339,14 +338,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(pitividir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(pitividir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pitividir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pitividir)" $$dlist; \
-	  fi; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-pitiviPYTHON:
 	@$(NORMAL_UNINSTALL)
diff -Nuar pitivi-0.13.4.orig/pitivi/timeline/Makefile.in pitivi-0.13.4/pitivi/timeline/Makefile.in
--- pitivi-0.13.4.orig/pitivi/timeline/Makefile.in	2010-03-14 20:32:42.646125370 +0200
+++ pitivi-0.13.4/pitivi/timeline/Makefile.in	2010-03-14 20:34:29.560881439 +0200
@@ -69,7 +69,6 @@
   sed '$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;s/\n/ /g' | \
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(timelinedir)"
-py_compile = $(top_srcdir)/py-compile
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 ACLOCAL = @ACLOCAL@
 ACLOCAL_AMFLAGS = @ACLOCAL_AMFLAGS@
@@ -262,14 +261,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(timelinedir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(timelinedir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(timelinedir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(timelinedir)" $$dlist; \
-	  fi; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-timelinePYTHON:
 	@$(NORMAL_UNINSTALL)
diff -Nuar pitivi-0.13.4.orig/pitivi/ui/Makefile.in pitivi-0.13.4/pitivi/ui/Makefile.in
--- pitivi-0.13.4.orig/pitivi/ui/Makefile.in	2010-03-14 20:32:42.652145620 +0200
+++ pitivi-0.13.4/pitivi/ui/Makefile.in	2010-03-14 20:35:25.097906508 +0200
@@ -69,7 +69,6 @@
   sed '$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;s/\n/ /g' | \
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(uidir)" "$(DESTDIR)$(uidir)"
-py_compile = $(top_srcdir)/py-compile
 DATA = $(ui_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 ACLOCAL = @ACLOCAL@
@@ -308,14 +307,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(uidir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(uidir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(uidir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(uidir)" $$dlist; \
-	  fi; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-uiPYTHON:
 	@$(NORMAL_UNINSTALL)
