--- mate-doc-utils-1.4.0/tools/mate-doc-utils.make.orig	2012-02-22 17:20:58.000000000 +0200
+++ mate-doc-utils-1.4.0/tools/mate-doc-utils.make	2012-07-21 02:46:00.677408532 +0300
@@ -613,7 +613,6 @@
 	done
 	@if test "x$(_ENABLE_SK)" = "xtrue"; then \
 	  echo "scrollkeeper-update -p $(DESTDIR)$(_sklocalstatedir) -o $(DESTDIR)$(OMF_DIR)/$(_doc_install_dir)"; \
-	  scrollkeeper-update -p "$(DESTDIR)$(_sklocalstatedir)" -o "$(DESTDIR)$(OMF_DIR)/$(_doc_install_dir)"; \
 	fi;
 
 install-doc-dsk:
