--- gtk+-2.2.4/gtk/gtkrc.c.lib64	2003-08-24 23:01:54.000000000 +0200
+++ gtk+-2.2.4/gtk/gtkrc.c	2004-07-16 12:27:11.000000000 +0200
@@ -357,8 +357,13 @@ gtk_rc_get_im_module_file (void)
     {
       if (im_module_file)
 	result = g_strdup (im_module_file);
-      else
-	result = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", "gtk.immodules", NULL);
+      else {
+	result = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", "gtk.immodules." GTK_LIB, NULL);
+	if (! result) {
+	  g_free (result);
+	  result = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", "gtk.immodules", NULL);
+	}
+      }
     }
 
   return result;
--- gtk+-2.2.4/configure.ac.lib64	2004-07-16 12:18:05.000000000 +0200
+++ gtk+-2.2.4/configure.ac	2004-07-16 12:26:06.000000000 +0200
@@ -1510,6 +1510,39 @@ AM_CONDITIONAL(ENABLE_GTK_DOC, test x$en
 
 
 ##################################################
+# Determine the target libdir name
+##################################################
+
+AC_CACHE_CHECK([for target libdir name],
+  ac_cv_gtk_libname, [
+  case $host in
+  *-*-linux*)
+    # Test if the compiler is 64bit
+    echo 'int i;' > conftest.$ac_ext
+    gtk_cv_cc_64bit_output=no
+    if AC_TRY_EVAL(ac_compile); then
+      case `/usr/bin/file conftest.$ac_objext` in
+      *"ELF 64"*)
+        gtk_cv_cc_64bit_output=yes
+        ;;
+      esac
+    fi
+    rm -rf conftest*
+    ;;
+  esac
+  case $host_cpu:$gtk_cv_cc_64bit_output in
+  powerpc64:yes | s390x:yes | sparc64:yes | x86_64:yes)
+    ac_cv_gtk_libname="lib64"
+    ;;
+  *:*)
+    ac_cv_gtk_libname="lib"
+    ;;
+  esac
+])
+AC_DEFINE_UNQUOTED(GTK_LIB, "${ac_cv_gtk_libname}","")
+			
+
+##################################################
 # Output commands
 ##################################################
 
