From 2b98b29a01382325bb989407a36eefe959b8cd4c Mon Sep 17 00:00:00 2001
From: Pascal Terjan <pterjan@mandriva.com>
Date: Tue, 6 Oct 2009 11:47:56 +0200
Subject: [PATCH 3/3] [Bug 596977] Mark bg_pixmap as unset when it is

Fixes a crash

Signed-off-by: Pascal Terjan <pterjan@mandriva.com>
---
 gdk/x11/gdkwindow-x11.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/gdk/x11/gdkwindow-x11.c b/gdk/x11/gdkwindow-x11.c
index c65a0d4..2c94909 100644
--- a/gdk/x11/gdkwindow-x11.c
+++ b/gdk/x11/gdkwindow-x11.c
@@ -224,9 +224,11 @@ tmp_unset_bg (GdkWindow *window)
 
   impl->no_bg = TRUE;
 
-  if (obj->bg_pixmap != GDK_NO_BG)
+  if (obj->bg_pixmap != GDK_NO_BG) {
     XSetWindowBackgroundPixmap (GDK_DRAWABLE_XDISPLAY (window),
 				GDK_DRAWABLE_XID (window), None);
+    obj->bg_pixmap = GDK_NO_BG;
+  }
 }
 
 static void
-- 
1.6.4.4

