From 2ce2d3c748c3723fa692633ce285f80d70ceda22 Mon Sep 17 00:00:00 2001
From: Frederic Crozat <fcrozat@mandriva.com>
Date: Tue, 26 Aug 2008 07:59:40 +0000
Subject: [PATCH 3/5] Default themes for graphics and sound

---
 gtk/gtksettings.c |   23 ++++++++++++++++++-----
 1 files changed, 18 insertions(+), 5 deletions(-)

diff --git a/gtk/gtksettings.c b/gtk/gtksettings.c
index 4917263..e725f8e 100644
--- a/gtk/gtksettings.c
+++ b/gtk/gtksettings.c
@@ -295,7 +308,7 @@ gtk_settings_class_init (GtkSettingsClas
 #ifdef G_OS_WIN32
 								  "MS-Windows",
 #else
-								  "Raleigh",
+								  "oxygen-gtk",
 #endif
 								  GTK_PARAM_READWRITE),
                                              NULL);
@@ -919,7 +932,7 @@ gtk_settings_class_init (GtkSettingsClass *class)
                                                                    /* Translators: this means sounds that are played as feedback to user input */
 								   P_("Audible Input Feedback"),
 								   P_("Whether to play event sounds as feedback to user input"),
-								   TRUE,
+								   FALSE,
 								   GTK_PARAM_READWRITE),
                                              NULL);
   g_assert (result == PROP_ENABLE_INPUT_FEEDBACK_SOUNDS);
@@ -941,7 +954,7 @@ gtk_settings_class_init (GtkSettingsClass *class)
                                              g_param_spec_boolean ("gtk-enable-event-sounds",
 								   P_("Enable Event Sounds"),
 								   P_("Whether to play any event sounds at all"),
-								   TRUE,
+								   FALSE,
 								   GTK_PARAM_READWRITE),
                                              NULL);
   g_assert (result == PROP_ENABLE_EVENT_SOUNDS);
-- 
1.6.4.4

