Index: xdelta-1.1.4/Makefile.am
===================================================================
--- xdelta-1.1.4.orig/Makefile.am
+++ xdelta-1.1.4/Makefile.am
@@ -28,7 +28,7 @@ m4datadir = $(datadir)/aclocal
 m4data_DATA = xdelta.m4
 
 ## $Format: "libxdelta_la_LDFLAGS = -version-info $LibCurrent$:$LibRevision$:$LibAge$" $
-libxdelta_la_LDFLAGS = -version-info 2:0:0
+libxdelta_la_LDFLAGS = -version-info 2:0:0 -Llibedsio/.libs -ledsio
 
 #
 # Rules for the generated code
