diff -Nur /var/pisi/xdelta-1.1.4-8/work/xdelta-1.1.4.orig/libedsio/edsio.c /var/pisi/xdelta-1.1.4-8/work/xdelta-1.1.4/libedsio/edsio.c
--- xdelta-1.1.4.orig/libedsio/edsio.c	2008-04-27 12:50:53.000000000 +0300
+++ xdelta-1.1.4/libedsio/edsio.c	2008-04-27 12:58:27.000000000 +0300
@@ -1490,8 +1490,8 @@
 {
   GQueue *q = g_new (GQueue, 1);
 
-  q->list = q->list_end = NULL;
-  q->list_size = 0;
+  q->head = q->tail = NULL;
+  q->length = 0;
 
   return q;
 }
@@ -1502,8 +1502,8 @@
 {
   if (q)
     {
-      if (q->list)
-        g_list_free (q->list);
+      if (q->head)
+        g_list_free (q->head);
       g_free (q);
     }
 }
@@ -1512,7 +1512,7 @@
 guint
 g_queue_get_size (GQueue *q)
 {
-  return (q == NULL) ? 0 : q->list_size;
+  return (q == NULL) ? 0 : q->length;
 }
 
 
@@ -1521,12 +1521,12 @@
 {
   if (q)
     {
-      q->list = g_list_prepend (q->list, data);
+      q->head = g_list_prepend (q->head, data);
 
-      if (q->list_end == NULL)
-        q->list_end = q->list;
+      if (q->tail == NULL)
+        q->tail = q->head;
 
-      q->list_size++;
+      q->length++;
     }
 }
 
@@ -1536,14 +1536,14 @@
 {
   if (q)
     {
-      q->list_end = g_list_append (q->list_end, data);
+      q->tail = g_list_append (q->tail, data);
 
-      if (! q->list)
-        q->list = q->list_end;
+      if (! q->head)
+        q->head = q->tail;
       else
-        q->list_end = q->list_end->next;
+        q->tail = q->tail->next;
 
-      q->list_size++;
+      q->length++;
     }
 }
 
@@ -1553,23 +1553,23 @@
 {
   gpointer data = NULL;
 
-  if ((q) && (q->list))
+  if ((q) && (q->head))
     {
       GList *node;
 
-      node = q->list;
+      node = q->head;
       data = node->data;
 
       if (! node->next)
         {
-          q->list = q->list_end = NULL;
-          q->list_size = 0;
+          q->head = q->tail = NULL;
+          q->length = 0;
         }
       else
         {
-          q->list = node->next;
-          q->list->prev = NULL;
-          q->list_size--;
+          q->head = node->next;
+          q->head->prev = NULL;
+          q->length--;
         }
 
       g_list_free_1 (node);
@@ -1584,23 +1584,23 @@
 {
   gpointer data = NULL;
 
-  if ((q) && (q->list))
+  if ((q) && (q->head))
     {
       GList *node;
 
-      node = q->list_end;
+      node = q->tail;
       data = node->data;
 
       if (! node->prev)
 	{
-          q->list = q->list_end = NULL;
-          q->list_size = 0;
+          q->head = q->tail = NULL;
+          q->length = 0;
         }
       else
 	{
-          q->list_end = node->prev;
-          q->list_end->next = NULL;
-          q->list_size--;
+          q->tail = node->prev;
+          q->tail->next = NULL;
+          q->length--;
         }
 
       g_list_free_1 (node);
