From fd9157fc6f7258d14b8dee753184c410d3c90293 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ralf=20Cors=C3=A9pius?= <corsepiu@fedoraproject.org>
Date: Wed, 14 Aug 2013 12:36:32 +0200
Subject: [PATCH 3/3] Activate osgviewerWX.

---
 examples/CMakeLists.txt | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index 22b58b7..dffc586 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -200,11 +200,9 @@ IF(DYNAMIC_OPENSCENEGRAPH)
         ADD_SUBDIRECTORY(osgviewerFOX)
     ENDIF(FOX_FOUND)
 
-    IF   (wxWidgets_FOUND AND CMAKE_BUILD_TYPE)
-        IF (${CMAKE_BUILD_TYPE} STREQUAL "Release")
+    IF   (wxWidgets_FOUND)
             ADD_SUBDIRECTORY(osgviewerWX)
-        ENDIF()
-    ENDIF()
+    ENDIF(wxWidgets_FOUND)
 
     IF   ( (QT4_FOUND OR Qt5Widgets_FOUND) AND NOT OSG_GLES1_AVAILABLE AND NOT OSG_GLES2_AVAILABLE AND NOT OSG_GL3_AVAILABLE)
 
-- 
1.8.3.1

