diff -Nuar a/libk3bdevice/k3bdevice_mmc.cpp b/libk3bdevice/k3bdevice_mmc.cpp
--- a/libk3bdevice/k3bdevice_mmc.cpp	2011-01-15 22:47:30.000000000 +0200
+++ b/libk3bdevice/k3bdevice_mmc.cpp	2013-09-18 02:02:28.708185636 +0300
@@ -322,7 +322,7 @@
     // DVD-DL: 48 (MMC5)
     // CD:     36 (MMC2)
     //
-    if( dataLen <= 4 ) {
+    if( dataLen <= 6 ) {
         int m = mediaType();
         if( m & (MEDIA_DVD_R_DL|MEDIA_DVD_R_DL_SEQ|MEDIA_DVD_R_DL_JUMP) )
             dataLen = 48;