--- lives-2.0.5/lives-plugins/plugins/decoders/mkv_decoder.h~	2012-09-17 22:20:55.000000000 +0800
+++ lives-2.0.5/lives-plugins/plugins/decoders/mkv_decoder.h	2013-07-15 12:42:54.465105597 +0800
@@ -244,12 +244,12 @@
 
 typedef struct CodecTags{
     char str[20];
-    enum CodecID id;
+    enum AVCodecID id;
 }CodecTags;
 
 typedef struct CodecMime{
     char str[32];
-    enum CodecID id;
+    enum AVCodecID id;
 }CodecMime;
 
 /* max. depth in the EBML tree structure */
--- lives-2.0.5/lives-plugins/plugins/decoders/asf_decoder.c~	2012-09-17 22:20:55.000000000 +0800
+++ lives-2.0.5/lives-plugins/plugins/decoders/asf_decoder.c	2013-07-15 12:46:04.380524620 +0800
@@ -63,7 +63,7 @@
 #include "asf_decoder.h"
 
 
-static enum CodecID ff_codec_get_id(const AVCodecTag *tags, unsigned int tag)
+static enum AVCodecID ff_codec_get_id(const AVCodecTag *tags, unsigned int tag)
 {
   int i;
   for(i=0; tags[i].id != CODEC_ID_NONE;i++) {
--- lives-2.0.5/lives-plugins/plugins/decoders/mkv_decoder.c~	2013-06-17 09:25:33.000000000 +0800
+++ lives-2.0.5/lives-plugins/plugins/decoders/mkv_decoder.c	2013-07-15 12:46:48.151772711 +0800
@@ -97,7 +97,7 @@
 
 
 
-static enum CodecID ff_codec_get_id(const AVCodecTag *tags, unsigned int tag)
+static enum AVCodecID ff_codec_get_id(const AVCodecTag *tags, unsigned int tag)
 {
   int i;
   for(i=0; tags[i].id != CODEC_ID_NONE;i++) {
@@ -1318,7 +1318,7 @@
   for (i=0; i < matroska->tracks.nb_elem; i++) {
 
     MatroskaTrack *track = &tracks[i];
-    enum CodecID codec_id = CODEC_ID_NONE;
+    enum AVCodecID codec_id = CODEC_ID_NONE;
     EbmlList *encodings_list = &track->encodings;
     MatroskaTrackEncoding *encodings = encodings_list->elem;
     uint8_t *extradata = NULL;
--- lives-2.0.5/lives-plugins/plugins/decoders/mpegts_decoder.c~	2013-03-13 11:33:33.000000000 +0800
+++ lives-2.0.5/lives-plugins/plugins/decoders/mpegts_decoder.c	2013-07-15 12:47:03.542213014 +0800
@@ -383,7 +383,7 @@
 
 //extern AVInputFormat ff_mpegts_demuxer;
 
-static enum CodecID ff_codec_get_id(const AVCodecTag *tags, unsigned int tag)
+static enum AVCodecID ff_codec_get_id(const AVCodecTag *tags, unsigned int tag)
 {
   int i;
   for(i=0; tags[i].id != CODEC_ID_NONE;i++) {
@@ -882,7 +882,7 @@
 typedef struct {
   uint32_t stream_type;
   enum AVMediaType codec_type;
-  enum CodecID codec_id;
+  enum AVCodecID codec_id;
 } StreamType;
 
 static const StreamType ISO_types[] = {
