diff -up blender-2.64a/source/blender/avi/intern/avi_endian.c.big blender-2.64a/source/blender/avi/intern/avi_endian.c
--- blender-2.64a/source/blender/avi/intern/avi_endian.c.big	2012-10-29 09:49:54.000000000 +0100
+++ blender-2.64a/source/blender/avi/intern/avi_endian.c	2012-10-29 09:50:01.000000000 +0100
@@ -49,7 +49,7 @@
 #ifdef __BIG_ENDIAN__
 
 /* copied from BLI_endian_switch_inline.h */
-static void invert(int *num)
+static void invert(int *val)
 {
 	int tval = *val;
 	*val = ((tval >> 24))             |
