diff -up blender-2.67/source/blender/blenfont/intern/blf_lang.c.syspath blender-2.67/source/blender/blenfont/intern/blf_lang.c
--- blender-2.67/source/blender/blenfont/intern/blf_lang.c.syspath	2013-03-22 15:31:03.000000000 +0100
+++ blender-2.67/source/blender/blenfont/intern/blf_lang.c	2013-05-08 16:51:05.091202907 +0200
@@ -78,7 +78,7 @@ static void free_locales(void)
 
 static void fill_locales(void)
 {
-	const char * const languages_path = BLI_get_folder(BLENDER_DATAFILES, "locale");
+	const char * const languages_path = "/usr/share/blender";
 	char languages[FILE_MAX];
 	LinkNode *lines = NULL, *line;
 	char *str;
