diff -up blender-2.64/intern/cycles/util/util_cuda.h.64bit blender-2.64/intern/cycles/util/util_cuda.h
--- blender-2.64/intern/cycles/util/util_cuda.h.64bit	2012-10-05 17:34:55.000000000 +0200
+++ blender-2.64/intern/cycles/util/util_cuda.h	2012-10-05 17:35:15.000000000 +0200
@@ -38,7 +38,7 @@ CCL_NAMESPACE_END
 
 #define CUDA_VERSION 3020
 
-#if defined(__x86_64) || defined(AMD64) || defined(_M_AMD64)
+#if defined(__x86_64) || defined(AMD64) || defined(_M_AMD64) || defined(__LP64__)
 typedef unsigned long long CUdeviceptr;
 #else
 typedef unsigned int CUdeviceptr;
