diff -up blender-2.64/source/blender/imbuf/intern/jp2.c.openjpeg_stdbool blender-2.64/source/blender/imbuf/intern/jp2.c
--- blender-2.64/source/blender/imbuf/intern/jp2.c.openjpeg_stdbool	2012-10-03 16:43:39.000000000 +0200
+++ blender-2.64/source/blender/imbuf/intern/jp2.c	2012-10-03 20:00:25.238000524 +0200
@@ -39,6 +39,7 @@
 #include "IMB_colormanagement.h"
 #include "IMB_colormanagement_intern.h"
 
+#include <stdbool.h>
 #include "openjpeg.h"
 
 #define JP2_FILEHEADER_SIZE 14
