diff -up blender-2.66a/source/blender/modifiers/intern/MOD_meshcache_pc2.c.bigendian blender-2.66a/source/blender/modifiers/intern/MOD_meshcache_pc2.c
--- blender-2.66a/source/blender/modifiers/intern/MOD_meshcache_pc2.c.bigendian	2013-03-11 17:17:10.088435806 +0100
+++ blender-2.66a/source/blender/modifiers/intern/MOD_meshcache_pc2.c	2013-03-11 17:17:35.941532159 +0100
@@ -62,7 +62,7 @@ static bool meshcache_read_pc2_head(FILE
 	}
 
 #ifdef __BIG_ENDIAN__
-	BLI_endian_switch_int32_array(&pc2_head->huh, (sizeof(*pc2_head) - sizeof(pc2_head->header)) / sizeof(int));
+    BLI_endian_switch_int32_array(&pc2_head->file_version, (sizeof(*pc2_head) - sizeof(pc2_head->header)) / sizeof(int));
 #endif
 
 	if (pc2_head->verts_tot != verts_tot) {