diff -up blender-2.49/config/linux2-config.py.org blender-2.49/config/linux2-config.py
--- blender-2.49/config/linux2-config.py.org	2009-05-18 21:13:48.000000000 +0200
+++ blender-2.49/config/linux2-config.py	2009-06-01 20:16:59.000000000 +0200
@@ -181,6 +181,12 @@ CCFLAGS = ['-pipe','-fPIC','-funsigned-c
 
 CPPFLAGS = ['-DXP_UNIX']
 CXXFLAGS = ['-pipe','-fPIC','-funsigned-char','-fno-strict-aliasing']
+
+import os
+RPM_OPT_FLAGS = os.environ['RPM_OPT_FLAGS'].split()
+CCFLAGS += RPM_OPT_FLAGS
+CXXFLAGS += RPM_OPT_FLAGS
+
 REL_CFLAGS = ['-O2']
 REL_CCFLAGS = ['-O2']
 ##BF_DEPEND = True
diff -up blender-2.49/extern/SConscript.org blender-2.49/extern/SConscript
--- blender-2.49/extern/SConscript.org	2009-02-26 15:21:17.000000000 +0100
+++ blender-2.49/extern/SConscript	2009-06-01 20:20:48.000000000 +0200
@@ -2,8 +2,6 @@
 
 Import('env')
 
-SConscript(['glew/SConscript'])
-
 if env['WITH_BF_GAMEENGINE']:
 	if env['WITH_BF_SOLID']:
 		SConscript(['qhull/SConscript', 'solid/SConscript'])
diff -up blender-2.49/SConstruct.org blender-2.49/SConstruct
--- blender-2.49/SConstruct.org	2009-04-21 21:06:24.000000000 +0200
+++ blender-2.49/SConstruct	2009-06-01 20:18:42.000000000 +0200
@@ -223,7 +223,7 @@ if env['OURPLATFORM'] == 'linux2' :
 
 		def CheckFreeAlut(context,env):
 			context.Message( B.bc.OKGREEN + "Linux platform detected:\n  checking for FreeAlut... " + B.bc.ENDC )
-			env['LIBS'] = 'alut'
+			env['LIBS'] = 'alut, openal'
 			result = context.TryLink(mylib_test_source_file, '.c')
 			context.Result(result)
 			return result
