diff -ruN blender.orig/source/blender/src/language.c blender/source/blender/src/language.c
--- blender.orig/source/blender/src/language.c	2005-06-01 10:59:36.000000000 +0200
+++ blender/source/blender/src/language.c	2005-06-01 11:07:15.000000000 +0200
@@ -358,9 +358,9 @@
 		strcpy(name, bundlePath);
 		strcat(name, "/Contents/Resources/.Blanguages");
 #else
-		/* Check the CWD. Takes care of the case where users
-		 * unpack blender tarball; cd blender-dir; ./blender */
-		strcpy(name, ".blender/.Blanguages");
+		/* Check the standar location. Takes care of the case 
+		 * in which the distribution installs blender */
+		strcpy(name, "/usr/share/blender/.Blanguages");
 #endif
 		lines= BLI_read_file_as_lines(name);
 
