diff -up blender-2.65/source/blender/blenfont/intern/blf_lang.c.syspath blender-2.65/source/blender/blenfont/intern/blf_lang.c
--- blender-2.65/source/blender/blenfont/intern/blf_lang.c.syspath	2012-12-10 20:25:45.000000000 +0100
+++ blender-2.65/source/blender/blenfont/intern/blf_lang.c	2012-12-15 08:39:00.897062402 +0100
@@ -83,7 +83,7 @@ static void free_locales(void)
 
 static void fill_locales(void)
 {
-	char *languages_path = BLI_get_folder(BLENDER_DATAFILES, "locale");
+        char languages_path[FILE_MAX] = "/usr/share/blender";
 	LinkNode *lines = NULL, *line;
 	char *str;
 	int idx = 0;
@@ -184,7 +184,7 @@ EnumPropertyItem *BLF_RNA_lang_enum_prop
 
 void BLF_lang_init(void)
 {
-	char *messagepath = BLI_get_folder(BLENDER_DATAFILES, "locale");
+        char *messagepath = "/usr/share/locale";
 
 	if (messagepath) {
 		bl_locale_init(messagepath, TEXT_DOMAIN_NAME);
diff -up blender-2.65/source/blender/blenlib/intern/path_util.c.syspath blender-2.65/source/blender/blenlib/intern/path_util.c
--- blender-2.65/source/blender/blenlib/intern/path_util.c.syspath	2012-12-10 20:25:44.000000000 +0100
+++ blender-2.65/source/blender/blenlib/intern/path_util.c	2012-12-15 08:34:18.594027763 +0100
@@ -983,38 +983,6 @@ static int get_path_system(char *targetp
 	char system_path[FILE_MAX];
 	const char *system_base_path;
 
-
-	/* first allow developer only overrides to the system path
-	 * these are only used when running blender from source */
-	char cwd[FILE_MAX];
-	char relfolder[FILE_MAX];
-
-	if (folder_name) {
-		if (subfolder_name) {
-			BLI_join_dirfile(relfolder, sizeof(relfolder), folder_name, subfolder_name);
-		}
-		else {
-			BLI_strncpy(relfolder, folder_name, sizeof(relfolder));
-		}
-	}
-	else {
-		relfolder[0] = '\0';
-	}
-
-	/* try CWD/release/folder_name */
-	if (BLI_current_working_dir(cwd, sizeof(cwd))) {
-		if (test_path(targetpath, cwd, "release", relfolder)) {
-			return 1;
-		}
-	}
-
-	/* try EXECUTABLE_DIR/release/folder_name */
-	if (test_path(targetpath, bprogdir, "release", relfolder))
-		return 1;
-	/* end developer overrides */
-
-
-
 	system_path[0] = '\0';
 
 	if (test_env_path(system_path, envvar)) {
