diff -up blender-2.66a/intern/cycles/device/device_cpu.cpp.nosse blender-2.66a/intern/cycles/device/device_cpu.cpp
diff -up blender-2.66a/intern/cycles/util/util_system.cpp.nosse blender-2.66a/intern/cycles/util/util_system.cpp
--- blender-2.66a/intern/cycles/util/util_system.cpp.nosse	2013-03-11 18:08:55.107881563 +0100
+++ blender-2.66a/intern/cycles/util/util_system.cpp	2013-03-11 18:08:44.299259325 +0100
@@ -199,6 +199,16 @@ bool system_cpu_support_sse3()
 
 #else
 
+bool system_cpu_support_sse2()
+{
+    return false;
+}
+
+bool system_cpu_support_sse3()
+{
+    return false;
+}
+
 bool system_cpu_support_optimized()
 {
 	return false;