--- release/freedesktop/blender.desktop.orig	2009-07-25 14:54:34.268969281 +0300
+++ release/freedesktop/blender.desktop			2009-07-25 14:58:00.331967159 +0300
@@ -2,8 +2,11 @@
 Encoding=UTF-8
 Name=Blender
 Comment=3D modeling, animation, rendering and post-production
+Comment[tr]=3B modelleme, animasyon, rendering ve üretim sonrası uygulaması
+GenericName=3D modelling/rendering
+GenericName[tr]=3D modelleme/rendering programı
 Exec=blender
-Icon=blender.png
+Icon=blender
 Terminal=false
 Type=Application
 Categories=Graphics;3DGraphics;
