diff -ur m4.orig/ost_ssl.m4 m4/ost_ssl.m4
--- m4.orig/ost_ssl.m4	2010-11-01 00:01:26.000000000 +0200
+++ m4/ost_ssl.m4	2013-05-13 23:40:20.025378117 +0300
@@ -41,7 +41,7 @@
 #       ])
 #   ],[
         AC_CHECK_HEADER(openssl/ssl.h,[
-            SSL_LIBS="-lssl"
+            SSL_LIBS="-lssl -lcrypto"
             ost_cv_ssl=true
             AC_DEFINE(CCXX_OPENSSL, [1], [define openssl])
             AC_DEFINE(CCXX_SSL, [OPENSSL], [defines ssl])
