diff -Nur secure_delete-3.1-old//srm.c secure_delete-3.1/srm.c
--- secure_delete-3.1-old//srm.c	2011-06-16 14:36:13.119001280 +0300
+++ secure_delete-3.1/srm.c	2011-06-16 14:36:22.264001036 +0300
@@ -166,8 +166,10 @@
             return 3;
         }
 
-        if (sdel_overwrite(mode, fd, 0, bufsize, filestat.st_size > 0 ? filestat.st_size : 1, zero) == 0)
+        if (sdel_overwrite(mode, fd, 0, bufsize, filestat.st_size > 0 ? filestat.st_size : 1, zero) == 0) {
+            close(fd);
             return sdel_unlink(filename, 0, 1, slow);
+		}
     } /* end IS_REG() */
     else {
         if (S_ISDIR(filestat.st_mode)) {
