diff -up stardict-3.0.1/src/conf.cpp.orig stardict-3.0.1/src/conf.cpp
--- stardict-3.0.1/src/conf.cpp.orig	2009-12-27 03:34:19.643362105 +1000
+++ stardict-3.0.1/src/conf.cpp	2009-12-27 03:34:35.408307165 +1000
@@ -52,7 +52,7 @@ AppConf::AppConf() : 
 #else
 	add_entry("/apps/stardict/preferences/dictionary/use_custom_font", false);
 #endif
-	add_entry("/apps/stardict/preferences/network/enable_netdict", true);
+	add_entry("/apps/stardict/preferences/network/enable_netdict", false);
 	// Default stardictd server.
 	add_entry("/apps/stardict/preferences/network/server", std::string(_("dict.stardict.org")));
 	add_entry("/apps/stardict/preferences/network/port", 2628);
diff -up stardict-3.0.1/src/prefsdlg.cpp.orig stardict-3.0.1/src/prefsdlg.cpp
--- stardict-3.0.1/src/prefsdlg.cpp.orig	2009-12-27 03:35:32.412305578 +1000
+++ stardict-3.0.1/src/prefsdlg.cpp	2009-12-27 03:36:47.272583142 +1000
@@ -1037,6 +1037,10 @@ void PrefsDlg::setup_network_netdict()
     GtkWidget *hbox1 = gtk_hbox_new(FALSE, 6);
     gtk_box_pack_start(GTK_BOX(hbox1),button,false,false,0);
     gtk_box_pack_start(GTK_BOX(vbox1),hbox1,false,false,0);
+
+    label = gtk_label_new(_("Warning: Requests to remote StarDict server are sent over the network in an unencrypted form. Do not enable this if you are translating sensitive documents."));
+    gtk_label_set_line_wrap(GTK_LABEL(label), TRUE);
+    gtk_box_pack_start(GTK_BOX(vbox1),label,false,false,0);
 }
 
 void PrefsDlg::on_setup_mainwin_searchWhileTyping_ckbutton_toggled(GtkToggleButton *button, PrefsDlg *oPrefsDlg)
