--- src/lib/data.cpp.orig	2009-04-25 00:54:05.000000000 +0200
+++ src/lib/data.cpp	2009-04-25 00:54:23.000000000 +0200
@@ -23,6 +23,7 @@
 #endif
 
 #include "kmp.h"
+#include <cstdio>
 
 #include "data.hpp"
 #include "getuint32.h"
--- src/lib/stardict_client.cpp.orig	2009-04-25 01:07:27.000000000 +0200
+++ src/lib/stardict_client.cpp	2009-04-25 01:07:41.000000000 +0200
@@ -28,6 +28,7 @@
 #include "sockets.hpp"
 #include "md5.h"
 #include "getuint32.h"
+#include <cstdio>
 
 #include "stardict_client.hpp"
 
