diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur libical-0.41/configure.in libical-0.41-orig/configure.in
--- libical-0.41/configure.in	2008-11-21 13:23:16.000000000 +0200
+++ libical-0.41-orig/configure.in	2008-11-21 13:22:37.000000000 +0200
@@ -307,7 +307,7 @@
 	  AC_CHECK_HEADER([python$PY_VERSION/Python.h],[],[AC_MSG_ERROR([Can't find python header files])])
 	  if test -f $PY_PREFIX/include/python$PY_VERSION/Python.h; then
 		  PY_LIBS="python$PY_VERSION"
-		  if test -d '$PY_EXEC_PREFIX/lib/python$PY_VERSION/config'; then		  
+		  if test -d "$PY_EXEC_PREFIX/lib/python$PY_VERSION/config"; then		  
 		  	  PY_LIB_LOC="-L$PY_EXEC_PREFIX/lib/python$PY_VERSION/config"
 			  PY_MAKEFILE="$PY_EXEC_PREFIX/lib/python$PY_VERSION/config/Makefile"
 		  else  	  
diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur libical-0.41-orig/src/python/Makefile.am libical-0.41/src/python/Makefile.am
--- libical-0.41-orig/src/python/Makefile.am	2008-11-21 13:25:42.000000000 +0200
+++ libical-0.41/src/python/Makefile.am	2008-11-21 13:29:40.000000000 +0200
@@ -1,7 +1,7 @@
 
-lib_LTLIBRARIES = libLibicalWrap.la
+lib_LTLIBRARIES = libicalwrap.la
 
-libLibicalWrap_la_SOURCES = LibicalWrap.c
+libicalwrap_la_SOURCES = LibicalWrap.c
 
 INCLUDES =				\
 	-I$(top_builddir)		\
@@ -22,8 +22,8 @@
 # This part should be done with libtool, but I don't know how to do
 # it. Libtool needs to generate a shared library in this directory
 # regardless of the value of AM_DISABLE_SHARED
-LibicalWrap.so: LibicalWrap.c 
-	ld -shared -o LibicalWrap.so LibicalWrap.o ../libical/.libs/libical.a  ../libicalss/.libs/libicalss.a 
+LibicalWrap.so: LibicalWrap.o
+	ld -shared -o libicalwrap.so LibicalWrap.o ../libical/.libs/libical.so  ../libicalss/.libs/libicalss.so
 
 CLEANFILES = LibicalWrap.c LibicalWrap_wrap.doc  Libical.pyc LibicalWrap.so
 
