Index: libical-0.44/src/libical/Makefile.am
===================================================================
--- libical-0.44.orig/src/libical/Makefile.am
+++ libical-0.44/src/libical/Makefile.am
@@ -119,6 +119,8 @@ libical_cxx_la_SOURCES =	\
 	icalvalue_cxx.h		\
 	icalvalue_cxx.cpp       \
 	icptrholder.h
+
+libical_cxx_la_LIBADD = $(PTHREAD_LIBS)
 endif
 
 libical_includedir = $(includedir)/libical
Index: libical-0.44/src/libicalss/Makefile.am
===================================================================
--- libical-0.44.orig/src/libicalss/Makefile.am
+++ libical-0.44/src/libicalss/Makefile.am
@@ -64,7 +64,7 @@ INCLUDES =			\
 	$(BDB_INCLUDE)
 
 libicalss_la_LDFLAGS = -version-info 44:0:44
-libicalss_la_LIBADD = $(BDB_LIBFILES) 
+libicalss_la_LIBADD = $(BDB_LIBFILES) ../libical/libical.la
 
 if DEV
 libicalss_la_DEVSOURCES = icalsslexer.l icalssyacc.y
Index: libical-0.44/src/libicalss/Makefile.am
===================================================================
--- libical-0.44.orig/src/libicalss/Makefile.am
+++ libical-0.44/src/libicalss/Makefile.am
@@ -119,6 +119,7 @@ libicalss_cxx_la_SOURCES =	\
 	icalspanlist_cxx.h	\
 	icalspanlist_cxx.cpp
 
+libicalss_cxx_la_LIBADD = ../libical/libical_cxx.la
 endif
 
 icalss.h: $(COMBINEDHEADERS)
