diff -p -up libical-0.43/examples/access_properties_and_parameters.c.format-security libical-0.43/examples/access_properties_and_parameters.c
--- libical-0.43/examples/access_properties_and_parameters.c.format-security	2009-01-15 14:33:10.000000000 +0100
+++ libical-0.43/examples/access_properties_and_parameters.c	2009-01-15 14:33:34.000000000 +0100
@@ -44,7 +44,7 @@ void get_required_attendees(icalcomponen
 	    /* Remember, the caller does not own this string, so you
                should strdup it if you want to change it. */
 	    const char *attendee = icalproperty_get_attendee(p);
-	    printf(attendee);
+	    printf("%s", attendee);
 	}
     }
 
