Index: soprano-2.6.0/backends/virtuoso/odbcconnectionpool.cpp
===================================================================
--- soprano-2.6.0.orig/backends/virtuoso/odbcconnectionpool.cpp
+++ soprano-2.6.0/backends/virtuoso/odbcconnectionpool.cpp
@@ -32,13 +32,18 @@
 #include <QtCore/QMutexLocker>
 
 #include <sql.h>
+#include <locale.h>
 
 
 Soprano::ODBC::Connection* Soprano::ODBC::ConnectionPoolPrivate::createConnection()
 {
+    char *tmpLocale = setlocale(LC_CTYPE, "");
+    setlocale(LC_CTYPE, "C");
+
     Environment* env = Environment::createEnvironment();
     if ( !env ) {
         setError( "Unable to create ODBC environment." );
+        setlocale(LC_CTYPE, tmpLocale);
         return 0;
     }
 
@@ -48,6 +53,7 @@ Soprano::ODBC::Connection* Soprano::ODBC
     if ( SQLAllocConnect( env->henv(), &hdbc ) != SQL_SUCCESS ) {
         setError( "Failed to allocate SQL handle" );
         delete env;
+        setlocale(LC_CTYPE, tmpLocale);
         return 0;
     }
 
@@ -72,6 +78,7 @@ Soprano::ODBC::Connection* Soprano::ODBC
         setError( Virtuoso::convertSqlError( SQL_HANDLE_DBC, hdbc ) );
         SQLFreeHandle( SQL_HANDLE_DBC, hdbc );
         delete env;
+        setlocale(LC_CTYPE, tmpLocale);
         return 0;
     }
 
@@ -87,10 +94,12 @@ Soprano::ODBC::Connection* Soprano::ODBC
         if ( conn->executeCommand( command ) != Error::ErrorNone ) {
             setError( conn->lastError() );
             delete conn;
+            setlocale(LC_CTYPE, tmpLocale);
             return 0;
         }
     }
 
+    setlocale(LC_CTYPE, tmpLocale);
     return conn;
 }
 
