diff -up soprano-2.5.63/CMakeLists.txt.rpath soprano-2.5.63/CMakeLists.txt
--- soprano-2.5.63/CMakeLists.txt.rpath	2010-11-22 10:36:57.000000000 -0600
+++ soprano-2.5.63/CMakeLists.txt	2011-01-24 14:19:19.169375389 -0600
@@ -136,7 +136,11 @@ set(PKGCONFIG_INSTALL_PREFIX "${CMAKE_IN
 # RPATH directories outside the current CMAKE_BINARY_DIR and also the library
 # install directory. Alex
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH  TRUE)
-set(CMAKE_INSTALL_RPATH                ${LIB_DESTINATION} )
+list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${LIB_DESTINATION}" _isSystem)
+if("${_isSystem}" LESS 0)
+  list(APPEND CMAKE_INSTALL_RPATH "${LIB_DESTINATION}")
+endif()
+
 
 if(APPLE)
    set(CMAKE_INSTALL_NAME_DIR ${LIB_DESTINATION})
