diff -Nuar wicd-1.7.0.orig/other/wicd.desktop wicd-1.7.0/other/wicd.desktop
--- wicd-1.7.0.orig/other/wicd.desktop	2010-01-17 23:06:38.229526532 +0200
+++ wicd-1.7.0/other/wicd.desktop	2010-01-17 23:08:56.592777894 +0200
@@ -2,12 +2,12 @@
 Categories=Application;Network;
 Exec=wicd-gtk --no-tray
 GenericName=Network Manager
+GenericName[tr]=Ağ Yöneticisi
 Icon=wicd-gtk
-Icon[en_US]=wicd-gtk
-Name=Wicd Network Manager
-Name[en_US]=Wicd Network Manager
+Name=Wicd
 Comment=Start the Wicd client without system tray icon
 Comment[en_US]=Start the Wicd client without system tray icon
+Comment[tr]=Sistem çekmecesine yerleştirmeden Wicd'i başlat
 Terminal=false
 Type=Application
 Version=1.0
diff -Nuar wicd-1.7.0.orig/other/wicd-tray.desktop wicd-1.7.0/other/wicd-tray.desktop
--- wicd-1.7.0.orig/other/wicd-tray.desktop	2010-01-17 23:06:38.229526532 +0200
+++ wicd-1.7.0/other/wicd-tray.desktop	2010-01-17 23:09:25.655777371 +0200
@@ -2,12 +2,12 @@
 Categories=Application;Network;
 Exec=wicd-gtk
 GenericName=Network Manager
+GenericName[tr]=Ağ Yöneticisi
 Icon=wicd-gtk
-Icon[en_US]=wicd-gtk
-Name=Wicd Network Manager Tray
-Name[en_US]=Wicd Network Manager Tray
+Name=Wicd
 Comment=Display network connection status in the system tray
 Comment[en_US]=Display network connection status in the system tray
+Comment[tr]=Sistem çekmecesine yerleştirmeden Wicd'i başlat
 Terminal=false
 Type=Application
 Version=1.0
