diff -Nuar wicd-1.7.0.orig/wicd/dbusmanager.py wicd-1.7.0/wicd/dbusmanager.py
--- wicd-1.7.0.orig/wicd/dbusmanager.py	2010-01-17 23:06:38.209527407 +0200
+++ wicd-1.7.0/wicd/dbusmanager.py	2010-01-17 23:44:17.545544215 +0200
@@ -76,6 +76,11 @@
     
     def connect_to_dbus(self):
         """ Connects to wicd's dbus interfaces and loads them into a dict. """
+        # For Pardus
+        from wicd import startservice as pardus
+        if pardus.infoService("wicd") in ["stopped", "off"]:
+            pardus.startService("wicd")
+
         proxy_obj = self._bus.get_object("org.wicd.daemon", '/org/wicd/daemon')
         daemon = dbus.Interface(proxy_obj, 'org.wicd.daemon')
         
@@ -90,4 +95,4 @@
         self._dbus_ifaces = {"daemon" : daemon, "wireless" : wireless,
                              "wired" : wired}
         
-DBUS_MANAGER = DBusManager()
\ Dosya sonunda yenisatır yok.
+DBUS_MANAGER = DBusManager()
