diff -up wicd-1.7.1/cli/wicd-cli.py.dbus-failure wicd-1.7.1/cli/wicd-cli.py
--- wicd-1.7.1/cli/wicd-cli.py.dbus-failure	2012-01-15 09:19:03.000000000 -0500
+++ wicd-1.7.1/cli/wicd-cli.py	2012-02-10 15:26:31.709169719 -0500
@@ -47,6 +47,10 @@ if not daemon:
     print 'Error connecting to wicd via D-Bus.  Please make sure the wicd service is running.'
     sys.exit(3)
 
+if daemon is None:
+	print 'Error connecting to wicd via D-Bus.  Please make sure the wicd service is running.'
+	sys.exit(3)
+
 parser = optparse.OptionParser()
 
 parser.add_option('--network', '-n', type='int', default=-1)
diff -up wicd-1.7.1/curses/wicd-curses.py.dbus-failure wicd-1.7.1/curses/wicd-curses.py
--- wicd-1.7.1/curses/wicd-curses.py.dbus-failure	2012-02-02 11:38:26.000000000 -0500
+++ wicd-1.7.1/curses/wicd-curses.py	2012-02-10 15:26:31.709169719 -0500
@@ -1037,6 +1037,10 @@ def setup_dbus(force=True):
         print 'Error connecting to wicd via D-Bus.  Please make sure the wicd service is running.'
         sys.exit(3)
 
+    if daemon is None:
+        print 'Error connecting to wicd via D-Bus.  Please make sure the wicd service is running.'
+        sys.exit(3)
+
     netentry_curses.dbus_init(dbus_ifaces)
     return True
 
diff -up wicd-1.7.1/gtk/gui.py.dbus-failure wicd-1.7.1/gtk/gui.py
--- wicd-1.7.1/gtk/gui.py.dbus-failure	2012-01-15 09:24:23.000000000 -0500
+++ wicd-1.7.1/gtk/gui.py	2012-02-10 15:26:31.711170031 -0500
@@ -158,6 +158,17 @@ class appGui(object):
             d.run()
             sys.exit(1)
 
+        if daemon is None:
+            errmsg = "Error connecting to wicd service via D-Bus." + \
+                     "Please ensure the wicd service is running."
+            d = gtk.MessageDialog(parent=None,
+                                  flags=gtk.DIALOG_MODAL,
+                                  type=gtk.MESSAGE_ERROR,
+                                  buttons=gtk.BUTTONS_OK,
+                                  message_format=errmsg)
+            d.run()
+            sys.exit(1)
+
         self.tray = tray
 
         gladefile = os.path.join(wpath.gtk, "wicd.ui")
