diff -up wicd-1.7.1/wicd/wicd-daemon.py.orig wicd-1.7.1/wicd/wicd-daemon.py
--- wicd-1.7.1/wicd/wicd-daemon.py.orig	2012-02-02 11:38:26.000000000 -0500
+++ wicd-1.7.1/wicd/wicd-daemon.py	2012-03-22 10:21:38.735140300 -0400
@@ -812,7 +812,7 @@ class WicdDaemon(dbus.service.Object):
         pass
 
     @dbus.service.signal(dbus_interface='org.wicd.daemon', signature='')
-    def DaemonClosing(self):
+    def DaemonClosing(self, arg1=None, arg2=None):
         """ Emits a signal indicating the daemon will be closing. """
         pass
 
