diff -up wicd-1.7.0/in/other=wicd.conf.in.orig wicd-1.7.0/in/other=wicd.conf.in
--- wicd-1.7.0/in/other=wicd.conf.in.orig	2010-01-14 23:49:11.000000000 -0500
+++ wicd-1.7.0/in/other=wicd.conf.in	2011-08-11 16:57:20.818332542 -0400
@@ -7,41 +7,24 @@
                 <allow own="org.wicd.daemon"/>
                 <allow send_destination="org.wicd.daemon"/>
                 <allow send_interface="org.wicd.daemon"/>
-		<allow send_destination="org.wicd.daemon.wireless"/>
-		<allow send_interface="org.wicd.daemon.wireless"/>
-		<allow send_destination="org.wicd.daemon.wired"/>
-		<allow send_interface="org.wicd.daemon.wired"/>
+                <allow send_destination="org.wicd.daemon.wireless"/>
+                <allow send_interface="org.wicd.daemon.wireless"/>
+                <allow send_destination="org.wicd.daemon.wired"/>
+                <allow send_interface="org.wicd.daemon.wired"/>
         </policy>
 
         <policy context="default">
                 <deny own="org.wicd.daemon"/>
         </policy>
 
-	<!-- This Unix group will have permission to use Wicd's gui -->
-        <policy group="%WICDGROUP%">
-                <allow send_destination="org.wicd.daemon"/>
-                <allow send_interface="org.wicd.daemon"/>
-                <allow send_interface="org.freedesktop.DBus.Introspectable"/>
-                <!-- The Introspectable allow shouldn't be needed here, but
-                     it seems that a few distributions aren't yet allowing
-                     it in their hal configs, and we need it here, so... -->
-        </policy>
-
-       <!-- Comment the block below if you do not want all users logged in 
-            locally to have permission to use wicd-client.  This ignores the 
-            group based permission model defined above for the "%WICDGROUP%"
-            group. Note that this only applies if you are using ConsoleKit -
-            if you do not have ConsoleKit installed and in use, then this
-            block makes no difference either way.  -->
-
         <policy at_console="true">
                 <allow send_destination="org.wicd.daemon"/>
                 <allow send_interface="org.wicd.daemon"/>
-		<allow send_destination="org.wicd.daemon.wireless"/>
-		<allow send_interface="org.wicd.daemon.wireless"/>
-		<allow send_destination="org.wicd.daemon.wired"/>
-		<allow send_interface="org.wicd.daemon.wired"/>
-		<allow send_interface="org.freedesktop.DBus.Introspectable"/>
+                <allow send_destination="org.wicd.daemon.wireless"/>
+                <allow send_interface="org.wicd.daemon.wireless"/>
+                <allow send_destination="org.wicd.daemon.wired"/>
+                <allow send_interface="org.wicd.daemon.wired"/>
+                <allow send_interface="org.freedesktop.DBus.Introspectable"/>
         </policy>
 
 </busconfig>
