Index: pavucontrol-0.9.8/src/pavucontrol.desktop.in
===================================================================
--- pavucontrol-0.9.8.orig/src/pavucontrol.desktop.in
+++ pavucontrol-0.9.8/src/pavucontrol.desktop.in
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Version=1.0
 Encoding=UTF-8
-_Name=PulseAudio Volume Control
-_GenericName=Volume Control
+_Name=pavucontrol
+_GenericName=PulseAudio Volume Control
 _Comment=Adjust the volume level
 Exec=pavucontrol
 Icon=multimedia-volume-control
