diff -up gvfs-1.3.2/Makefile.am.archive-integration gvfs-1.3.2/Makefile.am
--- gvfs-1.3.2/Makefile.am.archive-integration	2009-06-25 19:55:39.000000000 -0400
+++ gvfs-1.3.2/Makefile.am	2009-07-13 22:51:52.480693853 -0400
@@ -1,5 +1,19 @@
 NULL =
 
+@INTLTOOL_DESKTOP_RULE@
+
+desktop_in_files = mount-archive.desktop.in
+
+mount-archive.desktop.in: mount-archive.desktop.in.in
+	sed -e "s|\@libexecdir\@|$(libexecdir)|" $< > $@
+
+desktopdir       = $(datadir)/applications
+if HAVE_ARCHIVE
+desktop_DATA     = mount-archive.desktop
+else
+desktop_DATA     =
+endif
+
 SUBDIRS = \
 	common \
 	metadata \
@@ -19,6 +33,8 @@ EXTRA_DIST = \
 	gvfs.doap		\
 	README.commits		\
 	MAINTAINERS		\
+	mount-archive.desktop.in.in	\
+	$(desktop_in_files)	\
 	m4			\
 	$(NULL)
 
diff -up /dev/null gvfs-1.3.2/mount-archive.desktop.in.in
--- /dev/null	2009-07-13 16:21:55.049012465 -0400
+++ gvfs-1.3.2/mount-archive.desktop.in.in	2009-07-13 22:39:14.844931097 -0400
@@ -0,0 +1,13 @@
+[Desktop Entry]
+_Name=Archive Mounter
+Exec=@libexecdir@/gvfsd-archive file=%u
+X-Gnome-Vfs-System=gio
+MimeType=application/x-cd-image;application/x-bzip-compressed-tar;application/x-compressed-tar;application/x-tar;application/x-cpio;application/x-zip;application/zip;application/x-lzma-compressed-tar;application/x-xz-compressed-tar;
+Terminal=false
+StartupNotify=false
+Type=Application
+NoDisplay=true
+X-GNOME-Bugzilla-Bugzilla=GNOME
+X-GNOME-Bugzilla-Product=gvfs
+X-GNOME-Bugzilla-Component=archive-backend
+X-GNOME-Bugzilla-Version=@VERSION@
