--- client/os_detect.py.orig	2009-12-12 13:06:25.918513404 +0200
+++ client/os_detect.py	2009-12-12 13:06:32.589767480 +0200
@@ -70,6 +70,7 @@
 distro_info['Linux from Scratch']= '/etc/lfs-release'
 distro_info['Mandrake Linux']= '/etc/mandrake-release'
 distro_info['Mandriva Linux']= '/etc/mandriva-release'
+distro_info['Pisi Linux']= '/etc/pisilinux-release'
 distro_info['Slackware Linux']= '/etc/slackware-release'
 distro_info['Solaris/Sparc']= '/etc/release'
 distro_info['Sun JDS']= '/etc/sun-release'
