--- smolt-1.4.2.2-orig/client/config.py	2010-03-19 16:08:22.000000000 +0200
+++ smolt-1.4.2.2/client/config.py	2013-01-14 02:26:02.791714365 +0200
@@ -10,10 +10,10 @@
 #Only a fool would edit what lays beyond here
 #Are you that fool?
 
-HW_UUID = "/etc/smolt/hw-uuid"
+HW_UUID = "/etc/hw-uuid"
 PUB_UUID = "/etc/smolt/pub-uuid"
-#UUID_DB = "/etc/smolt/uuiddb.cfg"
-#ADMIN_TOKEN = "/etc/sysconfig/smolt-token"
+UUID_DB = "/etc/smolt/uuiddb.cfg"
+ADMIN_TOKEN = "/etc/smolt/smolt-token"
 
 
 #These are the defaults taken from the source code.
