Index: librtmp/librtmp.pc.in
===================================================================
--- a/librtmp/librtmp.pc.in	(revision 546)
+++ b/librtmp/librtmp.pc.in	(working copy)
@@ -6,7 +6,8 @@
 Name: librtmp
 Description: RTMP implementation
 Version: @VERSION@
-Requires: @CRYPTO_REQ@
+Requires.private: @CRYPTO_REQ@
 URL: http://rtmpdump.mplayerhq.hu
-Libs: -L${libdir} -lrtmp -lz
+Libs: -L${libdir} -lrtmp
+Libs.private: -lz
 Cflags: -I${incdir}
