Index: librtmp/librtmp.pc.in
===================================================================
--- a/librtmp/librtmp.pc.in	(revision 546)
+++ b/librtmp/librtmp.pc.in	(working copy)
@@ -1,6 +1,6 @@
 prefix=@prefix@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+libdir=@libdir@
 incdir=${prefix}/include
 
 Name: librtmp
Index: librtmp/Makefile
===================================================================
--- a/librtmp/Makefile	(revision 546)
+++ b/librtmp/Makefile	(working copy)
@@ -80,7 +80,8 @@
 parseurl.o: parseurl.c rtmp.h rtmp_sys.h log.h Makefile
 
 librtmp.pc: librtmp.pc.in Makefile
-	sed -e "s;@prefix@;$(prefix);" -e "s;@VERSION@;$(VERSION);" \
+	sed -e "s;@prefix@;$(prefix);" -e "s;@libdir@;$(libdir);" \
+		-e "s;@VERSION@;$(VERSION);" \
 		-e "s;@CRYPTO_REQ@;$(CRYPTO_REQ);" librtmp.pc.in > $@
 
 install:	install_base $(SO_INST)
