diff -Nurpa -x '*~' -x '*.orig' -x '*.rej' -x '*.swp' rtmpdump-2.3//Makefile newdir/Makefile
--- rtmpdump-2.3//Makefile	2010-06-30 22:58:35.000000000 +0300
+++ rtmpdump-2.3/Makefile	2010-07-19 23:57:27.582603859 +0300
@@ -24,11 +24,9 @@ CFLAGS=-Wall $(XCFLAGS) $(INC) $(DEF) $(
 LDFLAGS=-Wall $(XLDFLAGS)
 
 bindir=$(prefix)/bin
-sbindir=$(prefix)/sbin
 mandir=$(prefix)/man
 
 BINDIR=$(DESTDIR)$(bindir)
-SBINDIR=$(DESTDIR)$(sbindir)
 MANDIR=$(DESTDIR)$(mandir)
 
 LIBS_posix=
@@ -52,9 +50,8 @@ all:	$(LIBRTMP) progs
 progs:	rtmpdump rtmpgw rtmpsrv rtmpsuck
 
 install:	progs
-	-mkdir -p $(BINDIR) $(SBINDIR) $(MANDIR)/man1 $(MANDIR)/man8
-	cp rtmpdump$(EXT) $(BINDIR)
-	cp rtmpgw$(EXT) rtmpsrv$(EXT) rtmpsuck$(EXT) $(SBINDIR)
+	-mkdir -p $(BINDIR) $(MANDIR)/man1 $(MANDIR)/man8
+	cp rtmpdump$(EXT) rtmpgw$(EXT) rtmpsrv$(EXT) rtmpsuck$(EXT) $(BINDIR)
 	cp rtmpdump.1 $(MANDIR)/man1
 	cp rtmpgw.8 $(MANDIR)/man8
 	@cd librtmp; $(MAKE) install
