diff -ru libfreebob-1.0.11.orig/src/debugmodule/debugmodule.cpp libfreebob-1.0.11/src/debugmodule/debugmodule.cpp
--- libfreebob-1.0.11.orig/src/debugmodule/debugmodule.cpp	2009-03-05 15:31:17.000000000 +0000
+++ libfreebob-1.0.11/src/debugmodule/debugmodule.cpp	2009-03-05 16:19:27.000000000 +0000
@@ -24,6 +24,7 @@
 #include <netinet/in.h>
 
 #include <iostream>
+#include <cstdio>
 
 using namespace std;
 
diff -ru libfreebob-1.0.11.orig/src/libfreebobavc/avc_extended_plug_info.cpp libfreebob-1.0.11/src/libfreebobavc/avc_extended_plug_info.cpp
--- libfreebob-1.0.11.orig/src/libfreebobavc/avc_extended_plug_info.cpp	2009-03-05 15:31:17.000000000 +0000
+++ libfreebob-1.0.11/src/libfreebobavc/avc_extended_plug_info.cpp	2009-03-05 15:52:09.000000000 +0000
@@ -23,6 +23,8 @@
 #include "ieee1394service.h"
 
 #include <cstdlib>
+#include <cstdio>
+#include <cstring>
 #include <netinet/in.h>
 #include <iostream>
 
Only in libfreebob-1.0.11/src/libfreebobavc: avc_extended_plug_info.cpp.orig
Only in libfreebob-1.0.11/src/libfreebobavc: avc_extended_plug_info.cpp.rej
diff -ru libfreebob-1.0.11.orig/src/libfreebobavc/avc_extended_stream_format.cpp libfreebob-1.0.11/src/libfreebobavc/avc_extended_stream_format.cpp
--- libfreebob-1.0.11.orig/src/libfreebobavc/avc_extended_stream_format.cpp	2009-03-05 15:31:17.000000000 +0000
+++ libfreebob-1.0.11/src/libfreebobavc/avc_extended_stream_format.cpp	2009-03-05 15:51:53.000000000 +0000
@@ -23,6 +23,7 @@
 #include "ieee1394service.h"
 
 #include <netinet/in.h>
+#include <cstdio>
 
 ///////////////////////////////////////////////////////////
 std::ostream& operator<<( std::ostream& stream, StreamFormatInfo info )
diff -ru libfreebob-1.0.11.orig/src/libfreebobavc/avc_generic.cpp libfreebob-1.0.11/src/libfreebobavc/avc_generic.cpp
--- libfreebob-1.0.11.orig/src/libfreebobavc/avc_generic.cpp	2009-03-05 15:31:17.000000000 +0000
+++ libfreebob-1.0.11/src/libfreebobavc/avc_generic.cpp	2009-03-05 16:10:32.000000000 +0000
@@ -24,6 +24,8 @@
 
 #include <cstdlib>
 #include <netinet/in.h>
+#include <cstring>
+#include <cstdio>
 
 #define DEBUG_EXTRA_VERBOSE 2
 
diff -ru libfreebob-1.0.11.orig/src/libfreebobavc/ieee1394service.cpp libfreebob-1.0.11/src/libfreebobavc/ieee1394service.cpp
--- libfreebob-1.0.11.orig/src/libfreebobavc/ieee1394service.cpp	2009-03-05 15:31:17.000000000 +0000
+++ libfreebob-1.0.11/src/libfreebobavc/ieee1394service.cpp	2009-03-05 16:13:59.000000000 +0000
@@ -25,6 +25,7 @@
 #include <netinet/in.h>
 
 #include <iostream>
+#include <cstdio>
 
 using namespace std;
 
diff -ru libfreebob-1.0.11.orig/src/libfreebobavc/serialize.cpp libfreebob-1.0.11/src/libfreebobavc/serialize.cpp
--- libfreebob-1.0.11.orig/src/libfreebobavc/serialize.cpp	2009-03-05 15:31:17.000000000 +0000
+++ libfreebob-1.0.11/src/libfreebobavc/serialize.cpp	2009-03-05 16:14:22.000000000 +0000
@@ -22,6 +22,7 @@
 
 #include <iostream>
 #include <iomanip>
+#include <cstdio>
 
 #include <netinet/in.h>
 
diff -ru libfreebob-1.0.11.orig/tests/test-extplugcmd.cpp libfreebob-1.0.11/tests/test-extplugcmd.cpp
--- libfreebob-1.0.11.orig/tests/test-extplugcmd.cpp	2009-03-05 15:31:17.000000000 +0000
+++ libfreebob-1.0.11/tests/test-extplugcmd.cpp	2009-03-05 15:51:53.000000000 +0000
@@ -24,6 +24,7 @@
 #include "libfreebobavc/ieee1394service.h"
 
 #include <argp.h>
+#include <cstdlib>
 
 using namespace std;
 
diff -ru libfreebob-1.0.11.orig/tests/test-volume.cpp libfreebob-1.0.11/tests/test-volume.cpp
--- libfreebob-1.0.11.orig/tests/test-volume.cpp	2009-03-05 15:31:17.000000000 +0000
+++ libfreebob-1.0.11/tests/test-volume.cpp	2009-03-05 16:28:29.000000000 +0000
@@ -22,6 +22,9 @@
 #include "libfreebobavc/serialize.h"
 #include "libfreebobavc/ieee1394service.h"
 
+#include <cstdlib>
+#include <cstdio>
+
 const bool bVerbose = false;
 
 short int
