diff -aur user-manager~/usermanager/context.py user-manager/usermanager/context.py
--- src~/usermanager/context.py	2014-01-13 23:04:06.000000000 +0100
+++ src/usermanager/context.py	2014-01-27 20:15:09.899975716 +0100
@@ -5,8 +5,8 @@
 
 Pds = pds.Pds('user-manager', debug = True)
 # Force to use Default Session for testing
-Pds.session = pds.DefaultDe
-# print 'Current session is : %s %s' % (Pds.session.Name, Pds.session.Version)
+#Pds.session = pds.DefaultDe
+print 'Current session is : %s %s' % (Pds.session.Name, Pds.session.Version)
 
 i18n = Pds.i18n
 KIconLoader = QIconLoader(Pds,forceCache=True)
