--- ./Makefile.in	2011-05-30 22:13:39.000000000 +0300
+++ ./Makefile.in	2012-11-21 11:02:34.000000000 +0200
@@ -73,11 +73,11 @@
 
 install:
 	@if [ ! -d ${prefix} ]; then \
-		${base_dir}/mkdir-sh "${prefix}" "0755" "root" ; \
+		mkdir -v -p -m 755 ${prefix}; \
 		fi
 
 	@if [ ! -d ${mandir} ]; then \
-		${base_dir}/mkdir-sh "${mandir}" "0755" "root" ; \
+		mkdir -v -p -m 755 ${mandir}; \
 	fi
 
 	@if [ ! -d ${mandir}/man8 ]; then \
