--- ./popper/Makefile.in	2011-05-30 22:13:39.000000000 +0300
+++ ./popper/Makefile.in	2012-11-21 11:07:26.000000000 +0200
@@ -231,7 +231,7 @@
 
 install: popper ${pop_auth}
 	@if [ ! -d ${installdir} ]; then \
-		${base_dir}/mkdir-sh "${installdir}" "0755" "root:wheel" ; \
+		mkdir -v -p -m 755 ${installdir}; \
 	fi
 
 	@${INSTALL} -s -m 0755 -o root popper ${installdir}/popper
@@ -248,7 +248,7 @@
 	          "with uid ${apop_uid}";                           \
 	    if [ "x${apop_dir}" != "x" ]; then                      \
 	        if [ ! -d "${apop_dir}" ]; then                     \
-	            ${base_dir}/mkdir-sh "${apop_dir}" "0700" "${apop_uid}:0" ; \
+	            mkdir -v -p -m 700 ${apop_dir}; \
 	            echo "Created directory ${apop_dir}" ;          \
 	        fi                                                  \
 	    fi ;                                                    \
