diff -up MyGUI_3.2.0/Common/CMakeLists.txt.fixup MyGUI_3.2.0/Common/CMakeLists.txt
--- MyGUI_3.2.0/Common/CMakeLists.txt.fixup	2013-04-25 14:47:11.128845465 -0400
+++ MyGUI_3.2.0/Common/CMakeLists.txt	2013-04-25 14:47:16.497845545 -0400
@@ -82,6 +82,7 @@ elseif(MYGUI_SAMPLES_INPUT EQUAL 3)
 endif()
 
 add_library(${PROJECTNAME} ${HEADER_FILES} ${SOURCE_FILES})
+set_target_properties(${PROJECTNAME} PROPERTIES VERSION ${MYGUI_VERSION} SOVERSION ${MYGUI_VERSION_MAJOR})
 
 if(MYGUI_RENDERSYSTEM EQUAL 1)
 	add_dependencies(${PROJECTNAME} MyGUI.DirectXPlatform)
