diff -upr caja-1.2.2-orig/data/caja-autorun-software.desktop.in.in caja-1.2.2/data/caja-autorun-software.desktop.in.in
--- caja-1.2.2-orig/data/caja-autorun-software.desktop.in.in	2012-05-09 18:15:51.000000000 +0200
+++ caja-1.2.2/data/caja-autorun-software.desktop.in.in	2012-06-16 16:56:00.190767210 +0200
@@ -2,6 +2,7 @@
 _Name=Autorun Prompt
 TryExec=caja-autorun-software
 Exec=caja-autorun-software %u
+X-Mate-Vfs-System=gio
 Icon=application-x-executable
 NoDisplay=true
 Terminal=false
diff -upr caja-1.2.2-orig/data/caja-browser.desktop.in.in caja-1.2.2/data/caja-browser.desktop.in.in
--- caja-1.2.2-orig/data/caja-browser.desktop.in.in	2012-05-09 18:15:51.000000000 +0200
+++ caja-1.2.2/data/caja-browser.desktop.in.in	2012-06-16 16:56:37.892154304 +0200
@@ -3,6 +3,7 @@ _Name=File Browser
 _Comment=Browse the file system with the file manager
 TryExec=caja
 Exec=caja --no-desktop --browser %U
+X-Mate-Vfs-System=gio
 Icon=system-file-manager
 Terminal=false
 StartupNotify=true
diff -upr caja-1.2.2-orig/data/caja-computer.desktop.in.in caja-1.2.2/data/caja-computer.desktop.in.in
--- caja-1.2.2-orig/data/caja-computer.desktop.in.in	2012-05-09 18:15:51.000000000 +0200
+++ caja-1.2.2/data/caja-computer.desktop.in.in	2012-06-16 16:58:07.072079271 +0200
@@ -3,6 +3,7 @@ _Name=Computer
 _Comment=Browse all local and remote disks and folders accessible from this computer
 TryExec=caja
 Exec=caja --no-desktop computer:
+X-Mate-Vfs-System=gio
 Icon=computer
 Terminal=false
 StartupNotify=true
diff -upr caja-1.2.2-orig/data/caja.desktop.in.in caja-1.2.2/data/caja.desktop.in.in
--- caja-1.2.2-orig/data/caja.desktop.in.in	2012-05-09 18:15:51.000000000 +0200
+++ caja-1.2.2/data/caja.desktop.in.in	2012-06-16 16:59:10.203741474 +0200
@@ -1,6 +1,7 @@
 [Desktop Entry]
 _Name=File Manager
 Exec=caja
+X-Mate-Vfs-System=gio
 Icon=system-file-manager
 Terminal=false
 Type=Application
diff -upr caja-1.2.2-orig/data/caja-file-management-properties.desktop.in.in caja-1.2.2/data/caja-file-management-properties.desktop.in.in
--- caja-1.2.2-orig/data/caja-file-management-properties.desktop.in.in	2012-05-09 18:15:51.000000000 +0200
+++ caja-1.2.2/data/caja-file-management-properties.desktop.in.in	2012-06-16 16:59:48.574146790 +0200
@@ -2,6 +2,7 @@
 _Name=File Management
 _Comment=Change the behaviour and appearance of file manager windows
 Exec=caja-file-management-properties
+X-Mate-Vfs-System=gio
 Icon=system-file-manager
 Terminal=false
 Type=Application
diff -upr caja-1.2.2-orig/data/caja-folder-handler.desktop.in.in caja-1.2.2/data/caja-folder-handler.desktop.in.in
--- caja-1.2.2-orig/data/caja-folder-handler.desktop.in.in	2012-05-09 18:15:51.000000000 +0200
+++ caja-1.2.2/data/caja-folder-handler.desktop.in.in	2012-06-16 17:00:26.916553872 +0200
@@ -2,6 +2,7 @@
 _Name=Open Folder
 TryExec=caja
 Exec=caja --no-desktop %U
+X-Mate-Vfs-System=gio
 NoDisplay=true
 Terminal=false
 Icon=folder-open
diff -upr caja-1.2.2-orig/data/caja-home.desktop.in.in caja-1.2.2/data/caja-home.desktop.in.in
--- caja-1.2.2-orig/data/caja-home.desktop.in.in	2012-05-09 18:15:51.000000000 +0200
+++ caja-1.2.2/data/caja-home.desktop.in.in	2012-06-16 17:00:51.412814953 +0200
@@ -3,6 +3,7 @@ _Name=Home Folder
 _Comment=Open your personal folder
 TryExec=caja
 Exec=caja --no-desktop
+X-Mate-Vfs-System=gio
 Icon=user-home
 Terminal=false
 StartupNotify=true
diff -upr caja-1.2.2-orig/src/mate-network-scheme.desktop.in caja-1.2.2/src/mate-network-scheme.desktop.in
--- caja-1.2.2-orig/src/mate-network-scheme.desktop.in	2012-05-09 18:15:51.000000000 +0200
+++ caja-1.2.2/src/mate-network-scheme.desktop.in	2012-06-16 17:02:06.121616002 +0200
@@ -3,6 +3,7 @@ _Name=Network
 _Comment=Browse bookmarked and local network locations
 TryExec=caja
 Exec=caja --no-desktop network:
+X-Mate-Vfs-System=gio
 Terminal=false
 StartupNotify=true
 Type=Application