diff -urN a/configure.in b/configure.in
--- a/configure.in	2012-05-09 17:15:51.000000000 +0100
+++ b/configure.in	2012-06-16 21:48:15.829913790 +0100
@@ -211,6 +211,8 @@
 
 dnl libexif checking
 
+PKG_CHECK_MODULES(GMODULE, [gmodule-2.0])
+
 PKG_CHECK_MODULES(EXIF, libexif > exif_minver,
     [AM_CONDITIONAL(HAVE_EXIF, true)
      AC_DEFINE(HAVE_EXIF, 1, [Define to enable EXIF support])
diff -urN a/libcaja-private/Makefile.am b/libcaja-private/Makefile.am
--- a/libcaja-private/Makefile.am	2012-05-09 17:15:51.000000000 +0100
+++ b/libcaja-private/Makefile.am	2012-06-16 21:47:36.716913853 +0100
@@ -34,6 +34,7 @@
 	$(top_builddir)/eel/libeel-2.la \
 	$(top_builddir)/libcaja-extension/libcaja-extension.la \
 	$(CORE_LIBS) \
+        $(GMODULE_LIBS) \
 	$(NULL)
 
 marshal_sources = \
 
