diff -ur net-tools-1.60.orig/lib/fddi.c net-tools-1.60/lib/fddi.c
--- net-tools-1.60.orig/lib/fddi.c	2000-03-05 13:26:02.000000000 +0200
+++ net-tools-1.60/lib/fddi.c	2007-01-13 23:20:29.774729400 +0200
@@ -27,6 +27,10 @@
 #error "Disable HW Type FDDI"
 #endif
 #if __GLIBC__ >= 2
+#define __be32 u_int32_t
+#define __le32 u_int32_t
+#define __be16 u_int16_t
+#define __le16 u_int16_t
 #include <netinet/if_fddi.h>
 #else
 #include <linux/if_fddi.h>
net-tools-1.60.orig/lib/fddi.o ve net-tools-1.60/lib/fddi.o dosyaları birbirinden farklı
net-tools-1.60.orig/lib/libnet-tools.a ve net-tools-1.60/lib/libnet-tools.a dosyaları birbirinden farklı
