diff -Nurp a/configure b/configure
--- a/configure	2006-10-07 13:22:08.000000000 +0300
+++ b/configure	2013-08-13 17:56:41.889843586 +0300
@@ -20102,7 +20102,7 @@ _ACEOF
 
 
 
-ALL_LINGUAS=""
+ALL_LINGUAS="tr"
 if test -n "$ac_tool_prefix"; then
   # Extract the first word of "${ac_tool_prefix}ranlib", so it can be a program name with args.
 set dummy ${ac_tool_prefix}ranlib; ac_word=$2
diff -Nurp a/configure.in b/configure.in
--- a/configure.in	2006-10-07 13:24:17.000000000 +0300
+++ b/configure.in	2013-08-13 17:57:31.487841869 +0300
@@ -29,7 +29,7 @@ AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GE
 
 
 dnl Languages which your application supports
-ALL_LINGUAS=""
+ALL_LINGUAS="tr"
 AM_GNU_GETTEXT
 AM_GNU_GETTEXT_VERSION(0.10.40)
 
diff -Nurp a/Makefile.in b/Makefile.in
--- a/Makefile.in	2005-02-19 17:45:31.000000000 +0200
+++ b/Makefile.in	2013-08-13 18:04:00.978828384 +0300
@@ -47,8 +47,8 @@ AUTOHEADER = @AUTOHEADER@
 AUTOMAKE = @AUTOMAKE@
 AWK = @AWK@
 BUILD_INCLUDED_LIBINTL = @BUILD_INCLUDED_LIBINTL@
-CATALOGS = @CATALOGS@
-CATOBJEXT = @CATOBJEXT@
+CATALOGS = tr.gmo
+CATOBJEXT = .gmo
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
 CFLAGS = @CFLAGS@
@@ -73,8 +73,8 @@ FFLAGS = @FFLAGS@
 GENCAT = @GENCAT@
 GETTEXT_PACKAGE = @GETTEXT_PACKAGE@
 GLIBC21 = @GLIBC21@
-GMOFILES = @GMOFILES@
-GMSGFMT = @GMSGFMT@
+GMOFILES = tr.gmo
+GMSGFMT = /usr/bin/msgfmt
 GTK_CFLAGS = @GTK_CFLAGS@
 GTK_LIBS = @GTK_LIBS@
 INSTALL_DATA = @INSTALL_DATA@
@@ -115,8 +115,8 @@ PACKAGE_TARNAME = @PACKAGE_TARNAME@
 PACKAGE_VERSION = @PACKAGE_VERSION@
 PATH_SEPARATOR = @PATH_SEPARATOR@
 PKG_CONFIG = @PKG_CONFIG@
-POFILES = @POFILES@
-POSUB = @POSUB@
+POFILES = tr.po
+POSUB = po
 RANLIB = @RANLIB@
 SET_MAKE = @SET_MAKE@
 SHELL = @SHELL@
diff -Nurp a/po/Makefile.in.in b/po/Makefile.in.in
--- a/po/Makefile.in.in	2005-02-19 17:44:26.000000000 +0200
+++ b/po/Makefile.in.in	2013-08-13 18:06:52.814822434 +0300
@@ -45,14 +45,14 @@ INCLUDES = -I.. -I$(top_srcdir)/intl
 
 COMPILE = $(CC) -c $(DEFS) $(INCLUDES) $(CPPFLAGS) $(CFLAGS) $(XCFLAGS)
 
-POFILES = @POFILES@
-GMOFILES = @GMOFILES@
+POFILES = tr.po
+GMOFILES = tr.gmo
 DISTFILES = ChangeLog Makefile.in.in POTFILES.in $(PACKAGE).pot \
 $(POFILES) $(GMOFILES)
 
 POTFILES = \
 
-CATALOGS = @CATALOGS@
+CATALOGS = tr.gmo
 
 .SUFFIXES:
 .SUFFIXES: .c .o .po .pox .gmo .mo
