--- desktop/mypaint.desktop.orig	2011-11-22 22:27:03.000000000 +0200
+++ desktop/mypaint.desktop	2012-08-29 07:28:38.000000000 +0300
@@ -5,6 +5,7 @@
 TryExec=mypaint
 Exec=mypaint %f
 Comment=Painting program for digital artists
+Comment[tr]=Görüntüleri Sıfırdan Boyama
 Comment[ro]=Software de artă plastică digitală
 Comment[fr]=Logiciel de peinture numérique pour artistes
 Comment[nn_NO]=Måleprogram for digitale artistar
@@ -14,9 +15,10 @@
 Comment[zh_HK]=藝術家的电脑绘画
 GenericName=Raster Graphics Editor
 GenericName[fr]=Éditeur d'Image Matricielle
+GenericName[tr]=Resim Düzenleyici
 MimeType=image/openraster;image/png;image/jpeg;
 Type=Application
-Icon=mypaint
+Icon=/usr/share/icons/hicolor/32x32/apps/mypaint.png
 StartupNotify=true
 Categories=Graphics;GTK;2DGraphics;RasterGraphics;
 Terminal=false
