Index: guile-1.8.6/acinclude.m4
===================================================================
--- guile-1.8.6.orig/acinclude.m4
+++ guile-1.8.6/acinclude.m4
@@ -176,7 +176,7 @@ acx_pthread_flags="pthreads none -Kthrea
 # pthread-config: use pthread-config program (for GNU Pth library)
 
 case "${host_cpu}-${host_os}" in
-        *solaris*)
+        *solaris*|*linux*)
 
         # On Solaris (at least, for some versions), libc contains stubbed
         # (non-functional) versions of the pthreads routines, so link-based
Index: guile-1.8.6/guile-1.8.pc.in
===================================================================
--- guile-1.8.6.orig/guile-1.8.pc.in
+++ guile-1.8.6/guile-1.8.pc.in
@@ -11,5 +11,5 @@ libguileinterface=@LIBGUILE_INTERFACE@
 Name: GNU Guile
 Description: GNU's Ubiquitous Intelligent Language for Extension
 Version: @GUILE_VERSION@
-Libs: -L${libdir} -lguile @GUILE_LIBS@
+Libs: -L${libdir} @GUILE_LIBS@ -lguile
 Cflags: -I${includedir} @GUILE_CFLAGS@
