diff -up guile-1.8.7/test-suite/tests/popen.test.testsuite guile-1.8.7/test-suite/tests/popen.test
--- guile-1.8.7/test-suite/tests/popen.test.testsuite	2009-07-04 00:19:00.000000000 +0200
+++ guile-1.8.7/test-suite/tests/popen.test	2009-07-16 17:09:57.000000000 +0200
@@ -168,7 +168,7 @@
 	   (port (with-error-to-port (cdr c2p)
 		   (lambda ()
 		     (open-output-pipe
-		      "exec 0</dev/null; while true; do echo closed 1>&2; done")))))
+		      "exec 0</dev/null; echo closed 1>&2; sleep 3")))))
       (close-port (cdr c2p))   ;; write side
       (with-epipe
        (lambda ()
