Index: libdvdread4/dvd_input.c
===================================================================
--- libdvdread4/dvd_input.c	(Revision 1166)
+++ libdvdread4/dvd_input.c	(Arbeitskopie)
@@ -50,12 +50,7 @@
 #else
 
 /* dlopening libdvdcss */
-#ifdef HAVE_DLFCN_H
 #include <dlfcn.h>
-#else
-/* Only needed on MINGW at the moment */
-#include "../../msvc/contrib/dlfcn.c"
-#endif
 
 typedef struct dvdcss_s *dvdcss_handle;
 static dvdcss_handle (*DVDcss_open)  (const char *);
