Index: libmpcodecs/vd_ffmpeg.c
===================================================================
--- libmpcodecs/vd_ffmpeg.c	(Revision 29857)
+++ libmpcodecs/vd_ffmpeg.c	(Arbeitskopie)
@@ -118,7 +118,10 @@
 // to set/get/query special features/parameters
 static int control(sh_video_t *sh, int cmd, void *arg, ...){
     vd_ffmpeg_ctx *ctx = sh->context;
-    AVCodecContext *avctx = ctx->avctx;
+    AVCodecContext *avctx;
+    
+    if (!ctx) return CONTROL_UNKNOWN;
+    avctx = ctx->avctx;
     switch(cmd){
     case VDCTRL_QUERY_FORMAT:
     {
