--- mplayer/gui/interface.c.format~	2012-01-10 00:40:55.977159082 +0100
+++ mplayer/gui/interface.c	2012-01-10 00:40:58.037155989 +0100
@@ -1206,7 +1206,7 @@ void gmp_msg(int mod, int lev, const cha
     vsnprintf(msg, sizeof(msg), format, va);
     va_end(va);
 
-    mp_msg(mod, lev, msg);
+    mp_msg(mod, lev, "%s", msg);
 
     if (mp_msg_test(mod, lev))
         gtkMessageBox(GTK_MB_FATAL, msg);
--- mplayer/mplayer.c.format~	2011-12-11 15:44:36.000000000 +0100
+++ mplayer/mplayer.c	2012-01-10 00:38:44.212356938 +0100
@@ -2890,7 +2890,7 @@ int main(int argc, char *argv[])
 
     if (!filename && !player_idle_mode && !use_gui) {
         // no file/vcd/dvd -> show HELP:
-        mp_msg(MSGT_CPLAYER, MSGL_INFO, help_text);
+        mp_msg(MSGT_CPLAYER, MSGL_INFO, "%s", help_text);
         exit_player_with_rc(EXIT_NONE, 0);
     }
 
