--- mplayer/fmt-conversion.c.ffmp~	2012-05-17 02:14:22.204820469 +0200
+++ mplayer/fmt-conversion.c	2012-05-17 02:10:58.848585265 +0200
@@ -65,8 +65,10 @@ static const struct {
     {IMGFMT_RGBA,    PIX_FMT_RGB0},
     {IMGFMT_RGB64LE, PIX_FMT_RGBA64LE},
     {IMGFMT_RGB64BE, PIX_FMT_RGBA64BE},
+    /*
     {IMGFMT_422A,    PIX_FMT_YUVA422P},
     {IMGFMT_444A,    PIX_FMT_YUVA444P},
+    */
 #endif
 #if LIBAVUTIL_VERSION_INT >= AV_VERSION_INT(51, 20, 1)
     {IMGFMT_GBR24P,  PIX_FMT_GBRP},
--- mplayer/libmpcodecs/vf_spp.c.ffmp~	2012-05-17 03:35:04.710553909 +0200
+++ mplayer/libmpcodecs/vf_spp.c	2012-05-17 03:35:08.653578743 +0200
@@ -579,7 +579,7 @@ static int vf_open(vf_instance_t *vf, ch
     init_avcodec();
 
     vf->priv->avctx= avcodec_alloc_context3(NULL);
-    ff_dsputil_init(&vf->priv->dsp, vf->priv->avctx);
+    dsputil_init(&vf->priv->dsp, vf->priv->avctx);
 
     vf->priv->log2_count= 3;
 
--- mplayer/libmpdemux/mp_taglists.c.ffmp~	2012-05-17 02:12:08.470008223 +0200
+++ mplayer/libmpdemux/mp_taglists.c	2012-05-17 05:07:11.200231555 +0200
@@ -125,7 +125,7 @@ static const struct AVCodecTag mp_bmp_ta
     { CODEC_ID_BMV_VIDEO,         MKTAG('B', 'M', 'V', 'V')},
     { CODEC_ID_C93,               MKTAG('C', '9', '3', 'V')},
     { CODEC_ID_CDGRAPHICS,        MKTAG('C', 'D', 'G', 'R')},
-    { CODEC_ID_CDXL,              MKTAG('C', 'D', 'X', 'L')},
+    /*{ CODEC_ID_CDXL,              MKTAG('C', 'D', 'X', 'L')},*/
     { CODEC_ID_CMV,               MKTAG('M', 'V', 'I', 'f')},
     { CODEC_ID_DFA,               MKTAG('C', 'D', 'F', 'A')},
     { CODEC_ID_DNXHD,             MKTAG('A', 'V', 'd', 'n')},
