--- clanbomber-1.05/clanbomber/ClanBomber.h~	2001-11-12 03:44:10.000000000 +0100
+++ clanbomber-1.05/clanbomber/ClanBomber.h	2004-06-26 05:02:31.246085869 +0200
@@ -210,7 +210,7 @@
  #endif
 #endif
 
-#ifndef WIN32
+#if 0 //ndef WIN32
 	#define min(a,b) ((a<b) ? (a) : (b))
 	#define max(a,b) ((a>b) ? (a) : (b))
 #endif
--- clanbomber-1.05/clanbomber/GameStatus.cpp.bak	2004-01-08 19:10:39.000000000 +0100
+++ clanbomber-1.05/clanbomber/GameStatus.cpp	2004-06-26 12:56:07.542504805 +0200
@@ -96,7 +96,7 @@
 					{
 						Resources::Gamestatus_tools()->put_screen(int(320+i*43-20*scalefactor), int(170+nr*43-20*scalefactor),scalefactor,scalefactor, 0 );
 					}
-					Resources::Gamestatus_tools()->put_screen( int(max(277+bomber_object_counter()->get_points()*43-20*scalefactor, (int)pokal_scroll_in)), int(170+nr*43-20*scalefactor), scalefactor,scalefactor, 0 );
+					Resources::Gamestatus_tools()->put_screen( int(max(277+bomber_object_counter()->get_points()*43-20*scalefactor, (float)(int)pokal_scroll_in)), int(170+nr*43-20*scalefactor), scalefactor,scalefactor, 0 );
 				}
 			}
 			else
