--- clanbomber/ClanBomber.h
+++ clanbomber/ClanBomber.h
@@ -20,5 +20,7 @@
 #define ClanBomber_h
 
+#include <math.h>
+#include <assert.h>
 
 #include "Resources.h"
 
--- clanbomber/ClanBomber.cpp
+++ clanbomber/ClanBomber.cpp
@@ -61,6 +61,9 @@
 
 #include <time.h>
 
+using std::cout;
+using std::endl;
+
 ClanBomberApplication app;
 
 CL_String ClanBomberApplication::map_path;
@@ -661,13 +664,13 @@
 		{
 			try
 			{
-				wav = CL_Streamed_MikModSample::create("/usr/share/clanbomber/bud.mod", NULL, true);
+				wav = CL_Streamed_MikModSample::create("/usr/local/share/clanbomber/bud.mod", NULL, true);
 				ses = new CL_SoundBuffer_Session( wav->prepare() );
 				ses->set_volume(1.2f);
 			}
 			catch (CL_Error err)
 			{
-				wav = CL_Streamed_MikModSample::create("/usr/local/share/clanbomber/bud.mod", NULL, true);
+				wav = CL_Streamed_MikModSample::create("/usr/share/clanbomber/bud.mod", NULL, true);
 				ses = new CL_SoundBuffer_Session( wav->prepare() );
 				ses->set_volume(1.2f);
 			}
--- clanbomber/Config.cpp
+++ clanbomber/Config.cpp
@@ -16,6 +16,9 @@
  ***************************************************************************/
 // $Id: Config.cpp,v 1.5 2004/01/13 12:34:48 xmassx Exp $
 
+#include <fstream>
+#include <cstdio>
+
 #include <ClanLib/Display/Input/input.h>
 
 #include "ClanBomber.h"
@@ -23,9 +26,9 @@
 
 #include "Controller.h"
 
-#include <fstream>
-#include <cstdio>
 
+using std::endl;
+
 CL_String Config::filename	= "clanbomber.cfg";
 
 int	Config::round_time       = 90;
@@ -524,7 +526,7 @@
 		bomber[7].set_name( "You" );
 	}
 
-	ofstream configfile(filename);
+	std::ofstream configfile(filename);
 
 	configfile << "27" << endl; // version
 
--- clanbomber/Map.cpp
+++ clanbomber/Map.cpp
@@ -20,6 +20,7 @@
 #include <ClanLib/Core/System/cl_assert.h>
 #include <ClanLib/Display/Display/surface.h>
 #include <ClanLib/Display/Font/font.h>
+#include <iostream>
 #include <fstream>
 #include <cstdio>
 #include <cstring>
@@ -43,6 +44,10 @@
 	#include <dirent.h>
 #endif
 
+using std::ostream;
+using std::ofstream;
+using std::endl;
+
 CL_List<MapEntry> Map::map_list;
 
 Map::Map( ClanBomberApplication* _app )
--- clanbomber/Map.h
+++ clanbomber/Map.h
@@ -19,6 +19,7 @@
 #ifndef Map_h
 #define Map_h
 
+#include <iostream>
 #include "MapEntry.h"
 
 class ClanBomberApplication;
--- clanbomber/Resources.cpp
+++ clanbomber/Resources.cpp
@@ -105,25 +105,24 @@
 
 void Resources::init()
 {
-	try
-	{
-		res = new CL_ResourceManager(PKGDATADIR "clanbomber.dat", true);
-	}
-	catch (CL_Error err)
-	{
-	    	try
-		{
-			res = new CL_ResourceManager(
-				"/usr/share/clanbomber/clanbomber.dat",
-				true);
-		}
-		catch (CL_Error err)
-		{
-			res = new CL_ResourceManager(
-				"/usr/local/share/clanbomber/clanbomber.dat",
-				true);
-		}
-	}
+    try
+    {
+	res = new CL_ResourceManager("clanbomber.dat", true);
+    } catch (CL_Error err) { try {
+	res = new CL_ResourceManager(
+	    "/usr/share/clanbomber/clanbomber.dat", true);
+    } catch (CL_Error err) { try {
+	res = new CL_ResourceManager(
+	    "/usr/local/share/clanbomber/clanbomber.dat", true);
+    } catch (CL_Error err) { try {
+	res = new CL_ResourceManager("clanbomber.scr", false);
+    } catch (CL_Error err) { try {
+	res = new CL_ResourceManager(
+	    "/usr/share/clanbomber/clanbomber.scr", false);
+    } catch (CL_Error err) {
+	res = new CL_ResourceManager(
+	    "/usr/local/share/clanbomber/clanbomber.scr", false);
+    }}}}}
 }
 
 void Resources::load_all()
