diff -Nuar user-manager-3.0.2.OSmanOS/ui/edituser.ui user-manager-3.0.2/ui/edituser.ui
--- user-manager-3.0.2.OSmanOS/ui/edituser.ui	2014-04-15 19:52:30.000000000 +0300
+++ user-manager-3.0.2/ui/edituser.ui	2010-12-17 18:39:57.000000000 +0200
@@ -6,8 +6,8 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>715</width>
-    <height>593</height>
+    <width>518</width>
+    <height>423</height>
    </rect>
   </property>
   <property name="sizePolicy">
@@ -100,19 +100,6 @@
      </layout>
     </widget>
    </item>
-   <item row="1" column="0">
-    <widget class="QPushButton" name="pushAdvanced">
-     <property name="text">
-      <string>Advanced</string>
-     </property>
-     <property name="checkable">
-      <bool>true</bool>
-     </property>
-     <property name="flat">
-      <bool>false</bool>
-     </property>
-    </widget>
-   </item>
    <item row="2" column="0" rowspan="2">
     <widget class="QGroupBox" name="advancedGroup">
      <property name="sizePolicy">
@@ -139,6 +126,13 @@
         </property>
        </widget>
       </item>
+      <item row="2" column="0">
+       <widget class="QLabel" name="label_7">
+        <property name="text">
+         <string>Shell:</string>
+        </property>
+       </widget>
+      </item>
       <item row="2" column="1" colspan="2">
        <widget class="QComboBox" name="comboShell">
         <property name="enabled">
@@ -198,16 +192,22 @@
         </property>
        </widget>
       </item>
-      <item row="2" column="0">
-       <widget class="QLabel" name="label_7">
-        <property name="text">
-         <string>Shell:</string>
-        </property>
-       </widget>
-      </item>
      </layout>
     </widget>
    </item>
+   <item row="1" column="0">
+    <widget class="QPushButton" name="pushAdvanced">
+     <property name="text">
+      <string>Advanced</string>
+     </property>
+     <property name="checkable">
+      <bool>true</bool>
+     </property>
+     <property name="flat">
+      <bool>false</bool>
+     </property>
+    </widget>
+   </item>
    <item row="4" column="0">
     <spacer name="verticalSpacer">
      <property name="orientation">
@@ -491,7 +438,7 @@
    <hints>
     <hint type="sourcelabel">
      <x>46</x>
-     <y>168</y>
+     <y>186</y>
     </hint>
     <hint type="destinationlabel">
      <x>46</x>
diff -Nuar user-manager-3.0.2.OSmanOS/ui/main.ui user-manager-3.0.2/ui/main.ui
--- user-manager-3.0.2.OSmanOS/ui/main.ui	2014-04-15 19:52:30.000000000 +0300
+++ user-manager-3.0.2/ui/main.ui	2010-07-06 14:16:21.000000000 +0300
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>666</width>
-    <height>534</height>
+    <height>561</height>
    </rect>
   </property>
   <property name="windowTitle">
@@ -98,7 +98,7 @@
            <x>0</x>
            <y>0</y>
            <width>644</width>
-           <height>218</height>
+           <height>231</height>
           </rect>
          </property>
          <layout class="QGridLayout" name="gridLayout">
