diff -Nuar user-manager-3.0.2.OSmanOS/ui/edituser.ui user-manager-3.0.2/ui/edituser.ui
--- user-manager-3.0.2.OSmanOS/ui/edituser.ui	2014-04-15 19:52:30.000000000 +0300
+++ user-manager-3.0.2/ui/edituser.ui	2010-12-17 18:39:57.000000000 +0200
@@ -6,8 +6,8 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>715</width>
-    <height>593</height>
+    <width>518</width>
+    <height>423</height>
    </rect>
   </property>
   <property name="sizePolicy">
@@ -100,19 +100,6 @@
      </layout>
     </widget>
    </item>
-   <item row="1" column="0">
-    <widget class="QPushButton" name="pushAdvanced">
-     <property name="text">
-      <string>Advanced</string>
-     </property>
-     <property name="checkable">
-      <bool>true</bool>
-     </property>
-     <property name="flat">
-      <bool>false</bool>
-     </property>
-    </widget>
-   </item>
    <item row="2" column="0" rowspan="2">
     <widget class="QGroupBox" name="advancedGroup">
      <property name="sizePolicy">
@@ -139,6 +126,13 @@
         </property>
        </widget>
       </item>
+      <item row="2" column="0">
+       <widget class="QLabel" name="label_7">
+        <property name="text">
+         <string>Shell:</string>
+        </property>
+       </widget>
+      </item>
       <item row="2" column="1" colspan="2">
        <widget class="QComboBox" name="comboShell">
         <property name="enabled">
@@ -198,16 +192,22 @@
         </property>
        </widget>
       </item>
-      <item row="2" column="0">
-       <widget class="QLabel" name="label_7">
-        <property name="text">
-         <string>Shell:</string>
-        </property>
-       </widget>
-      </item>
      </layout>
     </widget>
    </item>
+   <item row="1" column="0">
+    <widget class="QPushButton" name="pushAdvanced">
+     <property name="text">
+      <string>Advanced</string>
+     </property>
+     <property name="checkable">
+      <bool>true</bool>
+     </property>
+     <property name="flat">
+      <bool>false</bool>
+     </property>
+    </widget>
+   </item>
    <item row="4" column="0">
     <spacer name="verticalSpacer">
      <property name="orientation">
@@ -266,214 +266,161 @@
       <attribute name="title">
        <string>Authorizations</string>
       </attribute>
-      <widget class="QLineEdit" name="filterAuthorizations">
-       <property name="geometry">
-        <rect>
-         <x>4</x>
-         <y>4</y>
-         <width>291</width>
-         <height>23</height>
-        </rect>
-       </property>
-      </widget>
-      <widget class="QTreeWidget" name="treeAuthorizations">
-       <property name="geometry">
-        <rect>
-         <x>4</x>
-         <y>69</y>
-         <width>311</width>
-         <height>271</height>
-        </rect>
-       </property>
-       <property name="sizePolicy">
-        <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
-         <horstretch>0</horstretch>
-         <verstretch>0</verstretch>
-        </sizepolicy>
-       </property>
-       <property name="headerHidden">
-        <bool>true</bool>
-       </property>
-       <property name="columnCount">
-        <number>1</number>
-       </property>
-       <column>
-        <property name="text">
-         <string notr="true">Policy</string>
-        </property>
-       </column>
-      </widget>
-      <widget class="QWidget" name="authGroup" native="true">
-       <property name="enabled">
-        <bool>false</bool>
-       </property>
-       <property name="geometry">
-        <rect>
-         <x>4</x>
-         <y>443</y>
-         <width>152</width>
-         <height>71</height>
-        </rect>
-       </property>
-       <layout class="QVBoxLayout" name="verticalLayout">
-        <property name="margin">
-         <number>0</number>
-        </property>
-        <item>
-         <widget class="QRadioButton" name="radioAuthDefault">
+      <layout class="QGridLayout" name="gridLayout_4">
+       <item row="1" column="0">
+        <widget class="QTreeWidget" name="treeAuthorizations">
+         <property name="sizePolicy">
+          <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
+           <horstretch>0</horstretch>
+           <verstretch>0</verstretch>
+          </sizepolicy>
+         </property>
+         <property name="headerHidden">
+          <bool>true</bool>
+         </property>
+         <property name="columnCount">
+          <number>1</number>
+         </property>
+         <column>
           <property name="text">
-           <string>Use system settings</string>
+           <string notr="true">Policy</string>
           </property>
-          <property name="checked">
-           <bool>true</bool>
+         </column>
+        </widget>
+       </item>
+       <item row="5" column="0">
+        <widget class="Line" name="line_3">
+         <property name="orientation">
+          <enum>Qt::Horizontal</enum>
+         </property>
+        </widget>
+       </item>
+       <item row="6" column="0">
+        <widget class="QPushButton" name="pushAuth">
+         <property name="text">
+          <string>Apply this policy to all actions</string>
+         </property>
+        </widget>
+       </item>
+       <item row="2" column="0">
+        <widget class="QWidget" name="authGroup" native="true">
+         <property name="enabled">
+          <bool>false</bool>
+         </property>
+         <layout class="QVBoxLayout" name="verticalLayout">
+          <property name="margin">
+           <number>0</number>
           </property>
-         </widget>
-        </item>
-        <item>
-         <widget class="QRadioButton" name="radioAuthYes">
-          <property name="text">
-           <string>Authorize user</string>
-          </property>
-         </widget>
-        </item>
-        <item>
-         <widget class="QRadioButton" name="radioAuthNo">
-          <property name="text">
-           <string>Deny user</string>
-          </property>
-         </widget>
-        </item>
-       </layout>
-      </widget>
-      <widget class="Line" name="line_3">
-       <property name="geometry">
-        <rect>
-         <x>4</x>
-         <y>518</y>
-         <width>339</width>
-         <height>16</height>
-        </rect>
-       </property>
-       <property name="orientation">
-        <enum>Qt::Horizontal</enum>
-       </property>
-      </widget>
-      <widget class="QPushButton" name="pushAuth">
-       <property name="geometry">
-        <rect>
-         <x>4</x>
-         <y>525</y>
-         <width>210</width>
-         <height>24</height>
-        </rect>
-       </property>
-       <property name="text">
-        <string>Apply this policy to all actions</string>
-       </property>
-      </widget>
-      <widget class="QPushButton" name="pushButton">
-       <property name="geometry">
-        <rect>
-         <x>220</x>
-         <y>30</y>
-         <width>71</width>
-         <height>24</height>
-        </rect>
-       </property>
-       <property name="text">
-        <string>Clear</string>
-       </property>
-      </widget>
+          <item>
+           <widget class="QRadioButton" name="radioAuthDefault">
+            <property name="text">
+             <string>Use system settings</string>
+            </property>
+            <property name="checked">
+             <bool>true</bool>
+            </property>
+           </widget>
+          </item>
+          <item>
+           <widget class="QRadioButton" name="radioAuthYes">
+            <property name="text">
+             <string>Authorize user</string>
+            </property>
+           </widget>
+          </item>
+          <item>
+           <widget class="QRadioButton" name="radioAuthNo">
+            <property name="text">
+             <string>Deny user</string>
+            </property>
+           </widget>
+          </item>
+         </layout>
+        </widget>
+       </item>
+       <item row="0" column="0">
+        <widget class="KTreeWidgetSearchLine" name="filterAuthorizations">
+         <property name="clickMessage">
+          <string>Start typing to filter</string>
+         </property>
+        </widget>
+       </item>
+      </layout>
      </widget>
      <widget class="QWidget" name="tabGroups">
       <attribute name="title">
        <string>Groups</string>
       </attribute>
-      <widget class="QListWidget" name="listGroups">
-       <property name="geometry">
-        <rect>
-         <x>4</x>
-         <y>59</y>
-         <width>256</width>
-         <height>261</height>
-        </rect>
-       </property>
-       <property name="sizePolicy">
-        <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
-         <horstretch>0</horstretch>
-         <verstretch>0</verstretch>
-        </sizepolicy>
-       </property>
-      </widget>
-      <widget class="QLabel" name="label_8">
-       <property name="geometry">
-        <rect>
-         <x>40</x>
-         <y>340</y>
-         <width>100</width>
-         <height>16</height>
-        </rect>
-       </property>
-       <property name="sizePolicy">
-        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
-         <horstretch>0</horstretch>
-         <verstretch>0</verstretch>
-        </sizepolicy>
-       </property>
-       <property name="minimumSize">
-        <size>
-         <width>100</width>
-         <height>0</height>
-        </size>
-       </property>
-       <property name="text">
-        <string>Main Group:</string>
-       </property>
-      </widget>
-      <widget class="QComboBox" name="comboMainGroup">
-       <property name="geometry">
-        <rect>
-         <x>150</x>
-         <y>340</y>
-         <width>78</width>
-         <height>24</height>
-        </rect>
-       </property>
-       <property name="sizePolicy">
-        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
-         <horstretch>0</horstretch>
-         <verstretch>0</verstretch>
-        </sizepolicy>
-       </property>
-      </widget>
-      <widget class="QLineEdit" name="filtergroups">
-       <property name="geometry">
-        <rect>
-         <x>4</x>
-         <y>4</y>
-         <width>251</width>
-         <height>23</height>
-        </rect>
-       </property>
-      </widget>
-      <widget class="QPushButton" name="btnClear">
-       <property name="geometry">
-        <rect>
-         <x>200</x>
-         <y>30</y>
-         <width>58</width>
-         <height>24</height>
-        </rect>
-       </property>
-       <property name="text">
-        <string>Clear</string>
-       </property>
-      </widget>
+      <layout class="QGridLayout" name="gridLayout_3">
+       <item row="1" column="0" colspan="2">
+        <widget class="QListWidget" name="listGroups">
+         <property name="sizePolicy">
+          <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
+           <horstretch>0</horstretch>
+           <verstretch>0</verstretch>
+          </sizepolicy>
+         </property>
+        </widget>
+       </item>
+       <item row="2" column="0">
+        <widget class="QLabel" name="label_8">
+         <property name="sizePolicy">
+          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
+           <horstretch>0</horstretch>
+           <verstretch>0</verstretch>
+          </sizepolicy>
+         </property>
+         <property name="minimumSize">
+          <size>
+           <width>100</width>
+           <height>0</height>
+          </size>
+         </property>
+         <property name="text">
+          <string>Main Group:</string>
+         </property>
+        </widget>
+       </item>
+       <item row="2" column="1">
+        <widget class="QComboBox" name="comboMainGroup">
+         <property name="sizePolicy">
+          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
+           <horstretch>0</horstretch>
+           <verstretch>0</verstretch>
+          </sizepolicy>
+         </property>
+        </widget>
+       </item>
+       <item row="0" column="0" colspan="2">
+        <widget class="KListWidgetSearchLine" name="filterGroups">
+         <property name="clickMessage">
+          <string>Start typing to filter</string>
+         </property>
+        </widget>
+       </item>
+      </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
+ <customwidgets>
+  <customwidget>
+   <class>KTreeWidgetSearchLine</class>
+   <extends>KLineEdit</extends>
+   <header>ktreewidgetsearchline.h</header>
+  </customwidget>
+  <customwidget>
+   <class>KLineEdit</class>
+   <extends>QLineEdit</extends>
+   <header>klineedit.h</header>
+  </customwidget>
+  <customwidget>
+   <class>KListWidgetSearchLine</class>
+   <extends>KLineEdit</extends>
+   <header>klistwidgetsearchline.h</header>
+  </customwidget>
+ </customwidgets>
  <tabstops>
   <tabstop>lineFullname</tabstop>
   <tabstop>linePassword</tabstop>
@@ -491,7 +438,7 @@
    <hints>
     <hint type="sourcelabel">
      <x>46</x>
-     <y>168</y>
+     <y>186</y>
     </hint>
     <hint type="destinationlabel">
      <x>46</x>
@@ -499,73 +446,5 @@
     </hint>
    </hints>
   </connection>
-  <connection>
-   <sender>btnClear</sender>
-   <signal>clicked()</signal>
-   <receiver>filtergroups</receiver>
-   <slot>clear()</slot>
-   <hints>
-    <hint type="sourcelabel">
-     <x>621</x>
-     <y>73</y>
-    </hint>
-    <hint type="destinationlabel">
-     <x>452</x>
-     <y>44</y>
-    </hint>
-   </hints>
-  </connection>
-  <connection>
-   <sender>filtergroups</sender>
-   <signal>textChanged(QString)</signal>
-   <receiver>EditUserWidget</receiver>
-   <slot>searchListWidget()</slot>
-   <hints>
-    <hint type="sourcelabel">
-     <x>466</x>
-     <y>45</y>
-    </hint>
-    <hint type="destinationlabel">
-     <x>285</x>
-     <y>0</y>
-    </hint>
-   </hints>
-  </connection>
-  <connection>
-   <sender>filterAuthorizations</sender>
-   <signal>textChanged(QString)</signal>
-   <receiver>EditUserWidget</receiver>
-   <slot>searchTreeListWidget()</slot>
-   <hints>
-    <hint type="sourcelabel">
-     <x>476</x>
-     <y>37</y>
-    </hint>
-    <hint type="destinationlabel">
-     <x>351</x>
-     <y>-6</y>
-    </hint>
-   </hints>
-  </connection>
-  <connection>
-   <sender>pushButton</sender>
-   <signal>clicked()</signal>
-   <receiver>filterAuthorizations</receiver>
-   <slot>clear()</slot>
-   <hints>
-    <hint type="sourcelabel">
-     <x>613</x>
-     <y>72</y>
-    </hint>
-    <hint type="destinationlabel">
-     <x>555</x>
-     <y>43</y>
-    </hint>
-   </hints>
-  </connection>
  </connections>
- <slots>
-  <slot>searchListWidget()</slot>
-  <slot>searchTreeListWidget()</slot>
- </slots>
 </ui>
diff -Nuar user-manager-3.0.2.OSmanOS/ui/main.ui user-manager-3.0.2/ui/main.ui
--- user-manager-3.0.2.OSmanOS/ui/main.ui	2014-04-15 19:52:30.000000000 +0300
+++ user-manager-3.0.2/ui/main.ui	2010-07-06 14:16:21.000000000 +0300
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>666</width>
-    <height>534</height>
+    <height>561</height>
    </rect>
   </property>
   <property name="windowTitle">
@@ -55,7 +55,7 @@
          </spacer>
         </item>
         <item>
-         <widget class="QComboBox" name="comboFilter"/>
+         <widget class="KComboBox" name="comboFilter"/>
         </item>
        </layout>
       </item>
@@ -98,7 +98,7 @@
            <x>0</x>
            <y>0</y>
            <width>644</width>
-           <height>218</height>
+           <height>231</height>
           </rect>
          </property>
          <layout class="QGridLayout" name="gridLayout">
@@ -121,6 +121,13 @@
    </item>
   </layout>
  </widget>
+ <customwidgets>
+  <customwidget>
+   <class>KComboBox</class>
+   <extends>QComboBox</extends>
+   <header>kcombobox.h</header>
+  </customwidget>
+ </customwidgets>
  <tabstops>
   <tabstop>comboFilter</tabstop>
   <tabstop>listItems</tabstop>
