--- ui/edituser.ui.old	2014-01-31 02:50:31.000000000 +0400
+++ ui/edituser.ui	2014-02-01 00:37:34.000000000 +0400
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>629</width>
-    <height>492</height>
+    <height>449</height>
    </rect>
   </property>
   <property name="sizePolicy">
@@ -180,9 +180,9 @@
        <property name="geometry">
         <rect>
          <x>4</x>
-         <y>70</y>
+         <y>60</y>
          <width>331</width>
-         <height>271</height>
+         <height>241</height>
         </rect>
        </property>
        <property name="sizePolicy">
@@ -224,8 +224,8 @@
       <widget class="Line" name="line_3">
        <property name="geometry">
         <rect>
-         <x>10</x>
-         <y>423</y>
+         <x>0</x>
+         <y>380</y>
          <width>339</width>
          <height>16</height>
         </rect>
@@ -237,8 +237,8 @@
       <widget class="QPushButton" name="pushAuth">
        <property name="geometry">
         <rect>
-         <x>4</x>
-         <y>430</y>
+         <x>0</x>
+         <y>390</y>
          <width>331</width>
          <height>24</height>
         </rect>
@@ -256,9 +256,9 @@
       <widget class="QPushButton" name="pushButton">
        <property name="geometry">
         <rect>
-         <x>260</x>
+         <x>250</x>
          <y>30</y>
-         <width>71</width>
+         <width>81</width>
          <height>24</height>
         </rect>
        </property>
@@ -269,8 +269,8 @@
       <widget class="QRadioButton" name="radioAuthYes">
        <property name="geometry">
         <rect>
-         <x>10</x>
-         <y>375</y>
+         <x>6</x>
+         <y>335</y>
          <width>321</width>
          <height>21</height>
         </rect>
@@ -288,8 +288,8 @@
       <widget class="QRadioButton" name="radioAuthDefault">
        <property name="geometry">
         <rect>
-         <x>10</x>
-         <y>350</y>
+         <x>6</x>
+         <y>310</y>
          <width>321</width>
          <height>21</height>
         </rect>
@@ -310,8 +310,8 @@
       <widget class="QRadioButton" name="radioAuthNo">
        <property name="geometry">
         <rect>
-         <x>10</x>
-         <y>400</y>
+         <x>6</x>
+         <y>360</y>
          <width>321</width>
          <height>21</height>
         </rect>
@@ -336,7 +336,7 @@
         <rect>
          <x>4</x>
          <y>59</y>
-         <width>256</width>
+         <width>331</width>
          <height>261</height>
         </rect>
        </property>
@@ -352,7 +352,7 @@
         <rect>
          <x>40</x>
          <y>340</y>
-         <width>100</width>
+         <width>151</width>
          <height>16</height>
         </rect>
        </property>
@@ -375,9 +375,9 @@
       <widget class="QComboBox" name="comboMainGroup">
        <property name="geometry">
         <rect>
-         <x>150</x>
+         <x>210</x>
          <y>340</y>
-         <width>78</width>
+         <width>121</width>
          <height>24</height>
         </rect>
        </property>
@@ -393,7 +393,7 @@
         <rect>
          <x>4</x>
          <y>4</y>
-         <width>251</width>
+         <width>331</width>
          <height>23</height>
         </rect>
        </property>
@@ -401,9 +401,9 @@
       <widget class="QPushButton" name="btnClear">
        <property name="geometry">
         <rect>
-         <x>200</x>
+         <x>250</x>
          <y>30</y>
-         <width>58</width>
+         <width>81</width>
          <height>24</height>
         </rect>
        </property>
@@ -439,25 +439,20 @@
       <string/>
      </property>
      <layout class="QGridLayout" name="gridLayout_5">
-      <item row="6" column="1" colspan="2">
-       <widget class="QComboBox" name="comboShell">
-        <property name="enabled">
-         <bool>true</bool>
-        </property>
-        <property name="sizePolicy">
-         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
-          <horstretch>0</horstretch>
-          <verstretch>0</verstretch>
-         </sizepolicy>
-        </property>
-        <property name="acceptDrops">
-         <bool>true</bool>
+      <item row="6" column="0">
+       <widget class="QLabel" name="label_7">
+        <property name="text">
+         <string>Shell:</string>
         </property>
-        <property name="editable">
-         <bool>false</bool>
+       </widget>
+      </item>
+      <item row="1" column="0">
+       <widget class="QLabel" name="label">
+        <property name="locale">
+         <locale language="Turkish" country="Turkey"/>
         </property>
-        <property name="sizeAdjustPolicy">
-         <enum>QComboBox::AdjustToContentsOnFirstShow</enum>
+        <property name="text">
+         <string>User Id:</string>
         </property>
        </widget>
       </item>
@@ -468,30 +463,38 @@
         </property>
        </widget>
       </item>
-      <item row="1" column="0">
-       <widget class="QLabel" name="label">
-        <property name="locale">
-         <locale language="Turkish" country="Turkey"/>
+      <item row="3" column="0" colspan="3">
+       <widget class="QLineEdit" name="lineHomeDir">
+        <property name="sizePolicy">
+         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
         </property>
         <property name="text">
-         <string>User Id:</string>
+         <string notr="true"/>
         </property>
        </widget>
       </item>
-      <item row="1" column="2">
-       <widget class="QCheckBox" name="checkAutoId">
-        <property name="text">
-         <string>Auto</string>
+      <item row="6" column="1" colspan="2">
+       <widget class="QComboBox" name="comboShell">
+        <property name="enabled">
+         <bool>true</bool>
         </property>
-        <property name="checked">
+        <property name="sizePolicy">
+         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
+        </property>
+        <property name="acceptDrops">
          <bool>true</bool>
         </property>
-       </widget>
-      </item>
-      <item row="6" column="0">
-       <widget class="QLabel" name="label_7">
-        <property name="text">
-         <string>Shell:</string>
+        <property name="editable">
+         <bool>false</bool>
+        </property>
+        <property name="sizeAdjustPolicy">
+         <enum>QComboBox::AdjustToContentsOnFirstShow</enum>
         </property>
        </widget>
       </item>
@@ -518,16 +521,13 @@
         </property>
        </widget>
       </item>
-      <item row="3" column="0" colspan="3">
-       <widget class="QLineEdit" name="lineHomeDir">
-        <property name="sizePolicy">
-         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
-          <horstretch>0</horstretch>
-          <verstretch>0</verstretch>
-         </sizepolicy>
-        </property>
+      <item row="8" column="0">
+       <widget class="QCheckBox" name="checkAutoId">
         <property name="text">
-         <string notr="true"/>
+         <string>Auto</string>
+        </property>
+        <property name="checked">
+         <bool>true</bool>
         </property>
        </widget>
       </item>
@@ -539,6 +539,9 @@
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
+     <property name="sizeType">
+      <enum>QSizePolicy::Minimum</enum>
+     </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
