--- edituser.ui	2010-12-17 19:39:57.000000000 +0300
+++ edituser.ui.new	2014-01-29 00:20:06.000000000 +0400
@@ -6,7 +6,7 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>518</width>
+    <width>533</width>
     <height>423</height>
    </rect>
   </property>
@@ -297,6 +297,12 @@
        </item>
        <item row="6" column="0">
         <widget class="QPushButton" name="pushAuth">
+         <property name="sizePolicy">
+          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
+           <horstretch>0</horstretch>
+           <verstretch>0</verstretch>
+          </sizepolicy>
+         </property>
          <property name="text">
           <string>Apply this policy to all actions</string>
          </property>
@@ -313,6 +319,12 @@
           </property>
           <item>
            <widget class="QRadioButton" name="radioAuthDefault">
+            <property name="sizePolicy">
+             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
+              <horstretch>0</horstretch>
+              <verstretch>0</verstretch>
+             </sizepolicy>
+            </property>
             <property name="text">
              <string>Use system settings</string>
             </property>
@@ -323,6 +335,12 @@
           </item>
           <item>
            <widget class="QRadioButton" name="radioAuthYes">
+            <property name="sizePolicy">
+             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
+              <horstretch>0</horstretch>
+              <verstretch>0</verstretch>
+             </sizepolicy>
+            </property>
             <property name="text">
              <string>Authorize user</string>
             </property>
@@ -330,6 +348,12 @@
           </item>
           <item>
            <widget class="QRadioButton" name="radioAuthNo">
+            <property name="sizePolicy">
+             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
+              <horstretch>0</horstretch>
+              <verstretch>0</verstretch>
+             </sizepolicy>
+            </property>
             <property name="text">
              <string>Deny user</string>
             </property>
@@ -406,9 +430,9 @@
  </widget>
  <customwidgets>
   <customwidget>
-   <class>KTreeWidgetSearchLine</class>
+   <class>KListWidgetSearchLine</class>
    <extends>KLineEdit</extends>
-   <header>ktreewidgetsearchline.h</header>
+   <header>klistwidgetsearchline.h</header>
   </customwidget>
   <customwidget>
    <class>KLineEdit</class>
@@ -416,9 +440,9 @@
    <header>klineedit.h</header>
   </customwidget>
   <customwidget>
-   <class>KListWidgetSearchLine</class>
+   <class>KTreeWidgetSearchLine</class>
    <extends>KLineEdit</extends>
-   <header>klistwidgetsearchline.h</header>
+   <header>ktreewidgetsearchline.h</header>
   </customwidget>
  </customwidgets>
  <tabstops>
