diff -up virtuoso-opensource-6.1.0/binsrc/samples/demo/Makefile.am.nodemos_buildfix virtuoso-opensource-6.1.0/binsrc/samples/demo/Makefile.am
--- virtuoso-opensource-6.1.0/binsrc/samples/demo/Makefile.am.nodemos_buildfix	2009-08-31 03:08:47.000000000 -0500
+++ virtuoso-opensource-6.1.0/binsrc/samples/demo/Makefile.am	2010-02-04 11:51:33.650603771 -0600
@@ -164,6 +164,7 @@ install-data-local:
 	chmod +x install_ini.sh
 	mkdir -p "$(DESTDIR)$(dbdir)"
 	$(SHELL) install_ini.sh database
+if WITH_DEMO_VAD
 if WITH_HTMLDOC
 	mkdir -p "$(DESTDIR)$(htmldocdir)"
 	$(INSTALL_DATA) $(top_srcdir)/docsrc/html_virt/*.html "$(DESTDIR)$(htmldocdir)"
@@ -173,6 +174,7 @@ if WITH_HTMLDOC
 	(cd $(top_srcdir)/docsrc/images; find . -name \*.png -print | cpio -dump "$(DESTDIR)$(htmldocdir)/../images")
 	(cd $(top_srcdir)/docsrc/images; find . -name \*.ico -print | cpio -dump "$(DESTDIR)$(htmldocdir)/../images")
 endif
+endif
 
 uninstall-local:
 	rm -f "$(DESTDIR)$(dbdir)/virtuoso.ini"
diff -up virtuoso-opensource-6.1.0/binsrc/vsp/Makefile.am.nodemos_buildfix virtuoso-opensource-6.1.0/binsrc/vsp/Makefile.am
--- virtuoso-opensource-6.1.0/binsrc/vsp/Makefile.am.nodemos_buildfix	2008-04-15 16:47:43.000000000 -0500
+++ virtuoso-opensource-6.1.0/binsrc/vsp/Makefile.am	2010-02-04 11:57:45.894591421 -0600
@@ -24,6 +24,7 @@
 #dist_httproot_DATA = *.html *.css *.txt
 
 install-data-local:
+if WITH_DEMO_VAD
 	@echo Installing in "$(httprootdir)" 
 	mkdir -p "$(DESTDIR)$(httprootdir)"
 	mkdir -p "$(DESTDIR)$(httprootdir)"/images
@@ -37,6 +38,7 @@ install-data-local:
 	$(INSTALL_DATA) images/*.png "$(DESTDIR)$(httprootdir)"/images
 	$(INSTALL_DATA) images/*.jpg "$(DESTDIR)$(httprootdir)"/images
 	$(INSTALL_DATA) vsmx/*.* "$(DESTDIR)$(httprootdir)"/vsmx
+endif
 
 
 uninstall-local:
