Index: configure.in
===================================================================
--- configure.in
+++ configure.in	2010-02-04 09:31:49.000000000 +0100
@@ -1833,7 +1833,7 @@
 #
 if test "x$virt_do_iodbc" = "xyes"
 then
-    iodbc_CPPFLAGS="-I$virt_iodbc_dir/include"
+    iodbc_CPPFLAGS=`iodbc-config --cflags`
 
     saved_CPPFLAGS="$CPPFLAGS"
     CPPFLAGS="$iodbc_CPPFLAGS $CPPFLAGS"
@@ -1842,7 +1842,7 @@
 
     ccincl="$iodbc_CPPFLAGS $ccincl"
 
-    iodbc_LDFLAGS="-L$virt_iodbc_dir/lib -liodbc -liodbcinst $LIBDL"
+    iodbc_LDFLAGS=`iodbc-config --libs`
     AC_CHECK_LIB(iodbc, SQLTables,
 		 [
 		  virt_do_iodbc="yes"
