diff -Nur KoboDeluxe-0.4pre10-old/graphics/window.cpp KoboDeluxe-0.4pre10/graphics/window.cpp
--- KoboDeluxe-0.4pre10-old/graphics/window.cpp	2006-11-01 23:44:36.636766254 +0200
+++ KoboDeluxe-0.4pre10/graphics/window.cpp	2006-11-01 23:47:58.330104205 +0200
@@ -243,7 +243,7 @@
 }
 
 
-void window_t::center_token(int _x, int _y, const char *txt, char token)
+void window_t::center_token(int _x, int _y, const char *txt, signed char token)
 {
 	center_token_fxp(PIXEL2CS(_x), PIXEL2CS(_y), txt, token);
 }
@@ -286,7 +286,7 @@
 }
 
 
-void window_t::center_token_fxp(int _x, int _y, const char *txt, char token)
+void window_t::center_token_fxp(int _x, int _y, const char *txt, signed char token)
 {
 	_x = CS2PIXEL(_x*xsc);
 	_y = CS2PIXEL(_y*ysc);
diff -Nur KoboDeluxe-0.4pre10-old/graphics/window.h KoboDeluxe-0.4pre10/graphics/window.h
--- KoboDeluxe-0.4pre10-old/graphics/window.h	2006-11-01 23:44:36.637766102 +0200
+++ KoboDeluxe-0.4pre10/graphics/window.h	2006-11-01 23:47:58.330104205 +0200
@@ -202,10 +202,10 @@
 	void font(int fnt);
 	void string(int _x, int _y, const char *txt);
 	void center(int _y, const char *txt);
-	void center_token(int _x, int _y, const char *txt, char token = 0);
+	void center_token(int _x, int _y, const char *txt, signed char token = 0);
 	void string_fxp(int _x, int _y, const char *txt);
 	void center_fxp(int _y, const char *txt);
-	void center_token_fxp(int _x, int _y, const char *txt, char token = 0);
+	void center_token_fxp(int _x, int _y, const char *txt, signed char token = 0);
 	int textwidth(const char *txt, int min = 0, int max = 255);
 	int fontheight();
 
