--- console/Subdir.mk.orig	2011-09-05 20:26:02.000000000 +0300
+++ console/Subdir.mk	2011-09-10 14:13:33.887999993 +0300
@@ -88,6 +88,10 @@
 console/scantv   : LDFLAGS := $(DLFLAGS)
 console/streamer : LDFLAGS := $(DLFLAGS)
 console/webcam   : LDFLAGS := $(DLFLAGS)
+console/v4l-conf : LDFLAGS := $(DLFLAGS) -pie
+
+# compile flags
+console/v4l-conf : CFLAGS += -fPIE
 
 # global targets
 all:: $(TARGETS-console) $(TARGETS-v4l-conf)

