diff -Nuar a/dolphin/src/dolphinmainwindow.cpp b/dolphin/src/dolphinmainwindow.cpp
--- a/dolphin/src/dolphinmainwindow.cpp	2013-08-28 20:00:18.309273734 +0300
+++ b/dolphin/src/dolphinmainwindow.cpp	2013-09-11 09:41:30.612792135 +0300
@@ -220,7 +220,7 @@
     showFilterBarAction->setChecked(generalSettings->filterBar());
 
     if (firstRun) {
-        menuBar()->setVisible(false);
+        menuBar()->setVisible(true);
         // Assure a proper default size if Dolphin runs the first time
         resize(750, 500);
     }
diff -Nuar a/dolphin/src/dolphinui.rc b/dolphin/src/dolphinui.rc
--- a/dolphin/src/dolphinui.rc	2013-06-28 20:13:51.543229897 +0300
+++ b/dolphin/src/dolphinui.rc	2013-09-11 09:56:02.000000000 +0300
@@ -88,14 +88,21 @@
         <text context="@title:menu">Main Toolbar</text>
         <Action name="go_back" />
         <Action name="go_forward" />
+        <Action name="go_up" />
         <Separator name="separator_1" />
         <Action name="icons" />
         <Action name="compact" />
         <Action name="details" />
         <Separator name="separator_0" />
+        <Action name="show_preview"/>
+        <Action name="split_view"/>
+        <Action name="edit_cut"/>
+        <Action name="edit_copy"/>
+        <Action name="edit_paste"/>
+        <Action name="move_to_trash"/>
+        <Action name="edit_undo"/>
+        <Action name="open_terminal"/>
         <Action name="edit_find"/>
-        <Action name="show_preview" />
-        <Action name="split_view" />
     </ToolBar>
     <ActionProperties scheme="Default">
         <Action priority="0" name="go_back"/>
diff -Nuar a/dolphin/src/settings/dolphin_generalsettings.kcfg b/dolphin/src/settings/dolphin_generalsettings.kcfg
--- a/dolphin/src/settings/dolphin_generalsettings.kcfg	2013-06-28 20:13:51.557229226 +0300
+++ b/dolphin/src/settings/dolphin_generalsettings.kcfg	2013-09-11 08:45:15.931908977 +0300
@@ -39,7 +39,7 @@
         </entry>
         <entry name="FilterBar" type="Bool">
             <label>Should the filter bar be shown</label>
-            <default>false</default>
+            <default>true</default>
         </entry>
         <entry name="GlobalViewProps" type="Bool">
             <label>Should the view properties be used for all directories</label>
@@ -51,7 +51,7 @@
         </entry>
         <entry name="ConfirmClosingMultipleTabs" type="Bool">
             <label>Ask for confirmation when closing windows with multiple tabs.</label>
-            <default>true</default>
+            <default>false</default>
         </entry>
         <entry name="RenameInline" type="Bool">
             <label>Rename inline</label>
@@ -63,11 +63,11 @@
         </entry>
         <entry name="ShowToolTips" type="Bool">
             <label>Show tooltips</label>
-            <default>false</default>
+            <default>true</default>
         </entry>
         <entry name="ShowCopyMoveMenu" type="Bool">
             <label>Show 'Copy To' and 'Move To' commands in context menu</label>
-            <default>false</default>
+            <default>true</default>
         </entry>
         <entry name="ViewPropsTimestamp" type="DateTime" >
             <label>Timestamp since when the view properties are valid</label>
@@ -82,7 +82,7 @@
         </entry>
         <entry name="ShowSpaceInfo" type="Bool">
             <label>Show the space information in the statusbar</label>
-            <default>false</default>
+            <default>true</default>
         </entry>
         <entry name="LockPanels" type="Bool">
             <label>Lock the layout of the panels</label>
