diff -Naur aaudit-2.2.1/tools/aulast/Makefile.am audit-2.2.1/tools/aulast/Makefile.am
--- aaudit-2.2.1/tools/aulast/Makefile.am	2012-03-23 14:42:43.000000000 +0200
+++ audit-2.2.1/tools/aulast/Makefile.am	2012-09-16 12:11:04.697652353 +0300
@@ -24,7 +24,7 @@
 AUTOMAKE_OPTIONS = no-dependencies
 EXTRA_DIST = $(man_MANS)
 INCLUDES = -I${top_srcdir} -I${top_srcdir}/lib -I${top_srcdir}/auparse
-LIBS = -L${top_builddir}/auparse -lauparse
+LDADD = ${top_builddir}/auparse/libauparse.la
 AM_CFLAGS = -D_GNU_SOURCE
 bin_PROGRAMS = aulast
 noinst_HEADERS = aulast-llist.h
diff -Naur aaudit-2.2.1/tools/aulast/Makefile.in audit-2.2.1/tools/aulast/Makefile.in
--- aaudit-2.2.1/tools/aulast/Makefile.in	2012-03-23 14:42:55.000000000 +0200
+++ audit-2.2.1/tools/aulast/Makefile.in	2012-09-16 12:14:00.013369255 +0300
@@ -75,6 +75,7 @@
 am_aulast_OBJECTS = aulast.$(OBJEXT) aulast-llist.$(OBJEXT)
 aulast_OBJECTS = $(am_aulast_OBJECTS)
 aulast_LDADD = $(LDADD)
+aulast_DEPENDENCIES = ${top_builddir}/auparse/libauparse.la
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
 depcomp =
 am__depfiles_maybe =
@@ -153,7 +154,7 @@
 LIBOBJS = @LIBOBJS@
 LIBPRELUDE_CFLAGS = @LIBPRELUDE_CFLAGS@
 LIBPRELUDE_LDFLAGS = @LIBPRELUDE_LDFLAGS@
-LIBS = -L${top_builddir}/auparse -lauparse
+LIBS = @LIBS@
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBWRAP_LIBS = @LIBWRAP_LIBS@
@@ -251,6 +252,7 @@
 AUTOMAKE_OPTIONS = no-dependencies
 EXTRA_DIST = $(man_MANS)
 INCLUDES = -I${top_srcdir} -I${top_srcdir}/lib -I${top_srcdir}/auparse
+LDADD = ${top_builddir}/auparse/libauparse.la
 AM_CFLAGS = -D_GNU_SOURCE
 noinst_HEADERS = aulast-llist.h
 man_MANS = aulast.8
diff -Naur aaudit-2.2.1/tools/aulastlog/Makefile.am audit-2.2.1/tools/aulastlog/Makefile.am
--- aaudit-2.2.1/tools/aulastlog/Makefile.am	2012-03-23 14:42:43.000000000 +0200
+++ audit-2.2.1/tools/aulastlog/Makefile.am	2012-09-16 12:14:48.133108566 +0300
@@ -24,7 +24,7 @@
 AUTOMAKE_OPTIONS = no-dependencies
 EXTRA_DIST = $(man_MANS)
 INCLUDES = -I${top_srcdir} -I${top_srcdir}/auparse
-LIBS = -L${top_builddir}/auparse -lauparse
+LDADD = ${top_builddir}/auparse/libauparse.la
 AM_CFLAGS = -D_GNU_SOURCE
 bin_PROGRAMS = aulastlog
 noinst_HEADERS = aulastlog-llist.h
diff -Naur aaudit-2.2.1/tools/aulastlog/Makefile.in audit-2.2.1/tools/aulastlog/Makefile.in
--- aaudit-2.2.1/tools/aulastlog/Makefile.in	2012-03-23 14:42:55.000000000 +0200
+++ audit-2.2.1/tools/aulastlog/Makefile.in	2012-09-16 12:16:27.045906044 +0300
@@ -75,6 +75,7 @@
 am_aulastlog_OBJECTS = aulastlog.$(OBJEXT) aulastlog-llist.$(OBJEXT)
 aulastlog_OBJECTS = $(am_aulastlog_OBJECTS)
 aulastlog_LDADD = $(LDADD)
+aulastlog_DEPENDENCIES = ${top_builddir}/auparse/libauparse.la
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
 depcomp =
 am__depfiles_maybe =
@@ -153,7 +154,7 @@
 LIBOBJS = @LIBOBJS@
 LIBPRELUDE_CFLAGS = @LIBPRELUDE_CFLAGS@
 LIBPRELUDE_LDFLAGS = @LIBPRELUDE_LDFLAGS@
-LIBS = -L${top_builddir}/auparse -lauparse
+LIBS = @LIBS@
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBWRAP_LIBS = @LIBWRAP_LIBS@
@@ -251,6 +252,7 @@
 AUTOMAKE_OPTIONS = no-dependencies
 EXTRA_DIST = $(man_MANS)
 INCLUDES = -I${top_srcdir} -I${top_srcdir}/auparse
+LDADD = ${top_builddir}/auparse/libauparse.la
 AM_CFLAGS = -D_GNU_SOURCE
 noinst_HEADERS = aulastlog-llist.h
 man_MANS = aulastlog.8
diff -Naur aaudit-2.2.1/tools/ausyscall/Makefile.am audit-2.2.1/tools/ausyscall/Makefile.am
--- aaudit-2.2.1/tools/ausyscall/Makefile.am	2012-03-23 14:42:43.000000000 +0200
+++ audit-2.2.1/tools/ausyscall/Makefile.am	2012-09-16 12:18:22.608613319 +0300
@@ -24,7 +24,7 @@
 AUTOMAKE_OPTIONS = no-dependencies
 EXTRA_DIST = $(man_MANS)
 INCLUDES = -I${top_srcdir} -I${top_srcdir}/lib
-LIBS = -L${top_builddir}/lib -laudit
+LDADD = ${top_builddir}/lib/libaudit.la
 bin_PROGRAMS = ausyscall
 man_MANS = ausyscall.8
 
diff -Naur aaudit-2.2.1/tools/ausyscall/Makefile.in audit-2.2.1/tools/ausyscall/Makefile.in
--- aaudit-2.2.1/tools/ausyscall/Makefile.in	2012-03-23 14:42:56.000000000 +0200
+++ audit-2.2.1/tools/ausyscall/Makefile.in	2012-09-16 12:19:50.881468438 +0300
@@ -73,6 +73,7 @@
 am_ausyscall_OBJECTS = ausyscall-ausyscall.$(OBJEXT)
 ausyscall_OBJECTS = $(am_ausyscall_OBJECTS)
 ausyscall_LDADD = $(LDADD)
+ausyscall_DEPENDENCIES = ${top_builddir}/lib/libaudit.la
 ausyscall_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(ausyscall_CFLAGS) \
 	$(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
@@ -153,7 +154,7 @@
 LIBOBJS = @LIBOBJS@
 LIBPRELUDE_CFLAGS = @LIBPRELUDE_CFLAGS@
 LIBPRELUDE_LDFLAGS = @LIBPRELUDE_LDFLAGS@
-LIBS = -L${top_builddir}/lib -laudit
+LIBS = @LIBS@
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBWRAP_LIBS = @LIBWRAP_LIBS@
@@ -251,6 +252,7 @@
 AUTOMAKE_OPTIONS = no-dependencies
 EXTRA_DIST = $(man_MANS)
 INCLUDES = -I${top_srcdir} -I${top_srcdir}/lib
+LDADD = ${top_builddir}/lib/libaudit.la
 man_MANS = ausyscall.8
 ausyscall_SOURCES = ausyscall.c
 ausyscall_CFLAGS = -g -D_GNU_SOURCE
diff -Naur aaudit-2.2.1/tools/auvirt/Makefile.am audit-2.2.1/tools/auvirt/Makefile.am
--- aaudit-2.2.1/tools/auvirt/Makefile.am	2012-03-23 14:42:43.000000000 +0200
+++ audit-2.2.1/tools/auvirt/Makefile.am	2012-09-16 12:43:08.973894322 +0300
@@ -28,7 +28,7 @@
 		   -I${top_srcdir}/lib \
 		   -I${top_srcdir}/auparse \
 		   -I${top_srcdir}/src
-LIBS = -L${top_builddir}/auparse -lauparse
+LDADD = ${top_builddir}/auparse/libauparse.la
 AM_CFLAGS = -D_GNU_SOURCE
 bin_PROGRAMS = auvirt
 noinst_HEADERS = auvirt-list.h
diff -Naur aaudit-2.2.1/tools/auvirt/Makefile.in audit-2.2.1/tools/auvirt/Makefile.in
--- aaudit-2.2.1/tools/auvirt/Makefile.in	2012-03-23 14:42:56.000000000 +0200
+++ audit-2.2.1/tools/auvirt/Makefile.in	2012-09-16 12:46:09.382916962 +0300
@@ -77,6 +77,7 @@
 	ausearch-time.$(OBJEXT)
 auvirt_OBJECTS = $(am_auvirt_OBJECTS)
 auvirt_LDADD = $(LDADD)
+auvirt_DEPENDENCIES = ${top_builddir}/auparse/libauparse.la
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
 depcomp =
 am__depfiles_maybe =
@@ -155,7 +156,7 @@
 LIBOBJS = @LIBOBJS@
 LIBPRELUDE_CFLAGS = @LIBPRELUDE_CFLAGS@
 LIBPRELUDE_LDFLAGS = @LIBPRELUDE_LDFLAGS@
-LIBS = -L${top_builddir}/auparse -lauparse
+LIBS = @LIBS@
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBWRAP_LIBS = @LIBWRAP_LIBS@
@@ -256,7 +257,7 @@
 		   -I${top_srcdir}/lib \
 		   -I${top_srcdir}/auparse \
 		   -I${top_srcdir}/src
-
+LDADD = ${top_builddir}/auparse/libauparse.la
 AM_CFLAGS = -D_GNU_SOURCE
 noinst_HEADERS = auvirt-list.h
 man_MANS = auvirt.8
