diff -up ppp-2.4.5/scripts/ip-up.local.add.ppp_resolv ppp-2.4.5/scripts/ip-up.local.add
--- ppp-2.4.5/scripts/ip-up.local.add.ppp_resolv	2010-07-13 10:29:23.227943994 +0200
+++ ppp-2.4.5/scripts/ip-up.local.add	2010-07-13 10:32:27.729695487 +0200
@@ -18,6 +18,9 @@ if [ -n "$USEPEERDNS" -a -f /run/ppp
 		rscf=/run/ppp/resolv.new
 		grep domain /run/ppp/resolv.prev > $rscf
 		grep search /run/ppp/resolv.prev >> $rscf
+		if [ -f /run/ppp/resolv.conf ]; then
+			cat /run/ppp/resolv.conf >> $rscf
+		fi
 		change_resolv_conf $rscf
 		rm -f $rscf
 	else
