diff -up ppp-2.4.4/configure.local ppp-2.4.4/configure
--- ppp-2.4.4/configure.local	2005-07-08 20:23:05.000000000 -0400
+++ ppp-2.4.4/configure	2008-08-28 17:38:04.000000000 -0400
@@ -2,7 +2,7 @@
 # $Id: configure,v 1.38 2008/06/15 07:08:49 paulus Exp $
 
 # Where to install stuff by default
-DESTDIR=/usr/local
+DESTDIR=/usr
 SYSCONF=/etc
 
 #  if [ -d /NextApps ]; then
diff -up ppp-2.4.4/pppd/Makefile.linux.local ppp-2.4.4/pppd/Makefile.linux
--- ppp-2.4.4/pppd/Makefile.linux.local	2008-08-28 17:37:33.000000000 -0400
+++ ppp-2.4.4/pppd/Makefile.linux	2008-08-28 17:37:33.000000000 -0400
@@ -97,8 +97,8 @@ endif
 
 # EAP SRP-SHA1
 ifdef USE_SRP
-CFLAGS	+= -DUSE_SRP -DOPENSSL -I/usr/local/ssl/include
-LIBS	+= -lsrp -L/usr/local/ssl/lib -lcrypto
+CFLAGS	+= -DUSE_SRP -DOPENSSL -I/usr/include/openssl
+LIBS	+= -lsrp -L/usr/lib -lcrypto
 TARGETS	+= srp-entry
 EXTRAINSTALL = $(INSTALL) -c -m 555 srp-entry $(BINDIR)/srp-entry
 MANPAGES += srp-entry.8
diff -up ppp-2.4.4/scripts/ppp-on-rsh.local ppp-2.4.4/scripts/ppp-on-rsh
--- ppp-2.4.4/scripts/ppp-on-rsh.local	2000-04-15 05:49:28.000000000 -0400
+++ ppp-2.4.4/scripts/ppp-on-rsh	2008-08-28 17:37:33.000000000 -0400
@@ -26,7 +26,7 @@ PPPD_RHOST=myremotehost
 # For this example, we assume that pppd on both local and remote
 # machines reside in the same place, /usr/local/bin/pppd
 #
-PPPD_LOC=/usr/local/bin/pppd
+PPPD_LOC=/usr/sbin/pppd
 
 #
 # The location of local options file (where rsh client is running).
diff -up ppp-2.4.4/scripts/ppp-on-ssh.local ppp-2.4.4/scripts/ppp-on-ssh
--- ppp-2.4.4/scripts/ppp-on-ssh.local	2000-04-15 05:49:42.000000000 -0400
+++ ppp-2.4.4/scripts/ppp-on-ssh	2008-08-28 17:37:33.000000000 -0400
@@ -26,7 +26,7 @@ PPPD_RHOST=myremotehost
 # For this example, we assume that pppd on both local and remote
 # machines reside in the same place, /usr/local/bin/pppd
 #
-PPPD_LOC=/usr/local/bin/pppd
+PPPD_LOC=/usr/sbin/pppd
 
 #
 # The location of local options file (where ssh client is running).
@@ -52,7 +52,7 @@ PPPD_REM_OPT=/etc/ppp/options-ssh-rem
 #
 # The location of ssh client on the local machine
 #
-SSH_LOC=/usr/local/bin/ssh
+SSH_LOC=/usr/bin/ssh
 
 export PPPD_LOC PPPD_LOC_OPT PPPD_REM_OPT PPPD_RHOST SSH_LOC
 
diff -up ppp-2.4.4/scripts/secure-card.local ppp-2.4.4/scripts/secure-card
--- ppp-2.4.4/scripts/secure-card.local	2004-02-01 22:36:46.000000000 -0500
+++ ppp-2.4.4/scripts/secure-card	2008-08-28 17:37:33.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/local/bin/expect -f
+#!/usr/bin/expect -f
 #
 # This  script was  written  by  Jim Isaacson  <jcisaac@crl.com>.  It is
 # designed to work  as a script to use the  SecureCARD(tm) device.  This
