Index: gtk+-2.16.5/gdk/x11/gdkxid.c
===================================================================
--- gtk+-2.16.5.orig/gdk/x11/gdkxid.c
+++ gtk+-2.16.5/gdk/x11/gdkxid.c
@@ -57,8 +57,10 @@ _gdk_xid_table_insert (GdkDisplay *displ
     display_x11->xid_ht = g_hash_table_new ((GHashFunc) gdk_xid_hash,
 					    (GEqualFunc) gdk_xid_equal);
 
+  /*
   if (g_hash_table_lookup (display_x11->xid_ht, xid))
     g_warning ("XID collision, trouble ahead");
+    */
 
   g_hash_table_insert (display_x11->xid_ht, xid, data);
 }
