diff -Naur --exclude '*.orig' core/FusionSound.git/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c core2/FusionSound.git/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c
--- core/FusionSound.git/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c	2008-03-11 18:03:00.000000000 +0100
+++ core2/FusionSound.git/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c	2009-03-04 21:15:00.000000000 +0100
@@ -41,8 +41,8 @@
 
 #include <misc/sound_util.h>
 
-#include <avcodec.h>
-#include <avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 
 static DirectResult
 Probe( IFusionSoundMusicProvider_ProbeContext *ctx );
