diff -Naur core/FusionSound.git/drivers/alsa.c core-fix-format-errors/FusionSound.git/drivers/alsa.c
--- core/FusionSound.git/drivers/alsa.c	2008-03-11 18:03:00.000000000 +0100
+++ core-fix-format-errors/FusionSound.git/drivers/alsa.c	2009-03-04 21:21:09.000000000 +0100
@@ -327,6 +327,7 @@
           if (snd_ctl_card_info( ctl, info ) == 0) {
                snprintf( device_info->name,
                          FS_SOUND_DEVICE_INFO_NAME_LENGTH,
+                         "%s",
                          snd_ctl_card_info_get_name( info ) );
           }
 
