diff -Nur core-old/FusionSound.git/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c core/FusionSound.git/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c
--- core-old/FusionSound.git/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c	2009-05-10 00:18:30.000000000 +0300
+++ core/FusionSound.git/interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c	2009-05-10 00:19:18.000000000 +0300
@@ -123,8 +123,8 @@
      return len;
 }
 
-static offset_t
-av_seek_callback( void *opaque, offset_t offset, int whence )
+static int64_t
+av_seek_callback( void *opaque, int64_t offset, int whence )
 {
      IFusionSoundMusicProvider_FFmpeg_data *data = opaque;
      unsigned int                           pos  = 0;
