Index: hunspell-1.2.12/src/tools/hunspell.cxx
===================================================================
--- hunspell-1.2.12.orig/src/tools/hunspell.cxx
+++ hunspell-1.2.12/src/tools/hunspell.cxx
@@ -70,6 +70,7 @@
     ".openoffice.org2.0/user/wordbook"
 #define OOODIR \
     "/opt/openoffice.org/basis3.0/share/dict/ooo:" \
+    "/opt/OpenOffice.org/lib/ooo-3.2/basis-link/share/dict/ooo:" \
     "/usr/lib/openoffice.org/basis3.0/share/dict/ooo:" \
     "/opt/openoffice.org2.4/share/dict/ooo:" \
     "/usr/lib/openoffice.org2.4/share/dict/ooo:" \
