=== modified file 'src/tools/Makefile.am'
Index: src/tools/Makefile.am
===================================================================
--- src/tools/Makefile.am.orig
+++ src/tools/Makefile.am
@@ -1,4 +1,4 @@
-bin_PROGRAMS=analyze chmorph hunspell munch unmunch hzip hunzip
+bin_PROGRAMS=hunspell-analyze hunspell-chmorph hunspell hunspell-munch hunspell-unmunch hzip hunzip
 
 INCLUDES=-I${top_srcdir}/src/hunspell -I${top_srcdir}/src/parsers
 
@@ -6,8 +6,8 @@ hzip_SOURCES=hzip.c
 hunzip_SOURCES=hunzip.cxx
 hunzip_LDADD = ../hunspell/libhunspell-1.2.la
 
-munch_SOURCES=munch.c munch.h
-unmunch_SOURCES=unmunch.c unmunch.h
+hunspell_munch_SOURCES=munch.c munch.h
+hunspell_unmunch_SOURCES=unmunch.c unmunch.h
 
 example_SOURCES=example.cxx
 example_LDADD = ../hunspell/libhunspell-1.2.la
@@ -16,11 +16,11 @@ hunspell_SOURCES=hunspell.cxx
 hunspell_LDADD = @LIBINTL@ @LIBICONV@ ../parsers/libparsers.a \
 	../hunspell/libhunspell-1.2.la @CURSESLIB@ @READLINELIB@
 
-analyze_SOURCES=analyze.cxx
-analyze_LDADD = ../hunspell/libhunspell-1.2.la
+hunspell_analyze_SOURCES=analyze.cxx
+hunspell_analyze_LDADD = ../hunspell/libhunspell-1.2.la
 
-chmorph_SOURCES=chmorph.cxx
-chmorph_LDADD = ../parsers/libparsers.a ../hunspell/libhunspell-1.2.la
+hunspell_chmorph_SOURCES=chmorph.cxx
+hunspell_chmorph_LDADD = ../parsers/libparsers.a ../hunspell/libhunspell-1.2.la
 
 noinst_PROGRAMS=example
 
Index: src/tools/example.cxx
===================================================================
--- src/tools/example.cxx.orig
+++ src/tools/example.cxx
@@ -17,8 +17,8 @@ main(int argc, char** argv)
     /* first parse the command line options */
 
     if (argc < 4) {
-        fprintf(stderr,"example (now it works with more dictionary files):\n"); 
-        fprintf(stderr,"example affix_file dictionary_file(s) file_of_words_to_check\n");
+        fprintf(stderr,"hunspell-example (now it works with more dictionary files):\n"); 
+        fprintf(stderr,"hunspell-example affix_file dictionary_file(s) file_of_words_to_check\n");
         exit(1);
     }
   
Index: src/tools/munch.c
===================================================================
--- src/tools/munch.c.orig
+++ src/tools/munch.c
@@ -42,14 +42,14 @@ int main(int argc, char** argv)
        wf = mystrdup(argv[1]);
   } else {
     fprintf(stderr,"correct syntax is:\n"); 
-    fprintf(stderr,"munch word_list_file affix_file\n");
+    fprintf(stderr,"hunspell-munch word_list_file affix_file\n");
     exit(1);
   }
   if (argv[2]) {
        af = mystrdup(argv[2]);
   } else {
     fprintf(stderr,"correct syntax is:\n"); 
-    fprintf(stderr,"munch word_list_file affix_file\n");
+    fprintf(stderr,"hunspell-munch word_list_file affix_file\n");
     exit(1);
   }
 
Index: src/tools/unmunch.c
===================================================================
--- src/tools/unmunch.c.orig
+++ src/tools/unmunch.c
@@ -39,14 +39,14 @@ int main(int argc, char** argv)
        wf = mystrdup(argv[1]);
   } else {
     fprintf(stderr,"correct syntax is:\n"); 
-    fprintf(stderr,"unmunch dic_file affix_file\n");
+    fprintf(stderr,"hunspell-unmunch dic_file affix_file\n");
     exit(1);
   }
   if (argv[2]) {
        af = mystrdup(argv[2]);
   } else {
     fprintf(stderr,"correct syntax is:\n"); 
-    fprintf(stderr,"unmunch dic_file affix_file\n");
+    fprintf(stderr,"hunspell-unmunch dic_file affix_file\n");
     exit(1);
   }
 
Index: src/tools/analyze.cxx
===================================================================
--- src/tools/analyze.cxx.orig
+++ src/tools/analyze.cxx
@@ -24,7 +24,7 @@ int main(int argc, char **argv)
 
     for (i = 1; i < 3; i++)
 	if (!argv[i]) {
-	    fprintf(stderr, "correct syntax is:\nanalyze affix_file");
+	    fprintf(stderr, "correct syntax is:\nhunspell-analyze affix_file");
 	    fprintf(stderr, " dictionary_file file_of_words_to_check\n");
 	    fprintf(stderr, "use two words per line for morphological generation\n");
 	    exit(1);
Index: tests/test.sh
===================================================================
--- tests/test.sh.orig
+++ tests/test.sh
@@ -34,7 +34,7 @@ fi
 shopt -s expand_aliases
 
 alias hunspell='../libtool --mode=execute -dlopen ../src/hunspell/.libs/libhunspell*.la ../src/tools/hunspell'
-alias analyze='../libtool --mode=execute -dlopen ../src/hunspell/.libs/libhunspell*.la ../src/tools/analyze'
+alias analyze='../libtool --mode=execute -dlopen ../src/hunspell/.libs/libhunspell*.la ../src/tools/hunspell-analyze'
 
 if [ "$VALGRIND" != "" ]; then
   rm -f $TEMPDIR/test.pid*
@@ -43,7 +43,7 @@ if [ "$VALGRIND" != "" ]; then
   fi
 
   alias hunspell='../libtool --mode=execute -dlopen ../src/hunspell/.libs/libhunspell*.la valgrind --tool=$VALGRIND --leak-check=yes --show-reachable=yes --log-file=$TEMPDIR/test.pid ../src/tools/hunspell'
-  alias analyze='../libtool --mode=execute -dlopen ../src/hunspell/.libs/libhunspell*.la valgrind --tool=$VALGRIND --leak-check=yes --show-reachable=yes --log-file=$TEMPDIR/test.pid ../src/tools/analyze'
+  alias analyze='../libtool --mode=execute -dlopen ../src/hunspell/.libs/libhunspell*.la valgrind --tool=$VALGRIND --leak-check=yes --show-reachable=yes --log-file=$TEMPDIR/test.pid ../src/tools/hunspell-analyze'
 fi
 
 # Tests good words
