--- hunspell-1.2.6.orig/src/hunspell/affixmgr.cxx
+++ hunspell-1.2.6/src/hunspell/affixmgr.cxx
@@ -148,11 +148,11 @@
        sStart[j] = NULL;
   }
 
-  if (keystring) free(keystring);
+  if (keystring && keystring != SPELL_KEYSTRING) free(keystring);
   keystring=NULL;
   if (trystring) free(trystring);
   trystring=NULL;
-  if (encoding) free(encoding);
+  if (encoding && encoding != SPELL_ENCODING) free(encoding);
   encoding=NULL;
   if (maptable) {  
      for (int j=0; j < nummap; j++) {
@@ -2971,7 +2971,9 @@
 // return text encoding of dictionary
 char * AffixMgr::get_encoding()
 {
-  if (! encoding ) encoding = mystrdup(SPELL_ENCODING);
+  if (! encoding ) {
+      encoding = SPELL_ENCODING;
+  }
   return mystrdup(encoding);
 }
 
@@ -3020,7 +3022,7 @@
 // return the keyboard string for suggestions
 char * AffixMgr::get_key_string()
 {
-  if (! keystring ) keystring = mystrdup(SPELL_KEYSTRING);
+  if (! keystring ) keystring = SPELL_KEYSTRING;
   return mystrdup(keystring);
 }
 
