--- src/tests/Makefile.am.link	2012-09-21 22:03:49.025758248 +0800
+++ src/tests/Makefile.am	2012-09-21 22:04:22.436746810 +0800
@@ -20,4 +20,5 @@
 fm_path_SOURCES = test-fm-path.c
 fm_path_LDADD= \
 	../libfm.la \
+	$(GIO_LIBS) \
 	$(NULL)
--- src/tests/Makefile.in.link	2012-09-21 22:03:55.065937004 +0800
+++ src/tests/Makefile.in	2012-09-21 22:04:46.327453406 +0800
@@ -286,6 +286,7 @@
 fm_path_SOURCES = test-fm-path.c
 fm_path_LDADD = \
 	../libfm.la \
+	$(GIO_LIBS) \
 	$(NULL)
 
 all: all-am
