diff -Nur licq-1.5.0-old//src/CMakeLists.txt licq-1.5.0/src/CMakeLists.txt
--- licq-1.5.0-old//src/CMakeLists.txt	2011-08-10 22:27:59.981000188 +0300
+++ licq-1.5.0/src/CMakeLists.txt	2011-08-10 22:29:07.861000211 +0300
@@ -114,7 +114,7 @@
 set_target_properties(licq PROPERTIES LINK_FLAGS "${CMAKE_THREAD_LIBS_INIT}")
 
 # Must export symbols from daemon so that plugins can use them
-list(APPEND CMAKE_EXE_LINKER_FLAGS ${CMAKE_EXE_EXPORTS_CXX_FLAG})
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${CMAKE_EXE_EXPORTS_CXX_FLAG}")
 
 # To find public headers
 include_directories(BEFORE ${PROJECT_SOURCE_DIR}/include)
