--- Makefile.am.OSmanOS         2012-11-08 04:25:58.000000000 +0200
+++ Makefile.am                 2013-02-14 18:27:39.272866825 +0200
@@ -198,7 +198,7 @@
 python_libusermodule_la_SOURCES = python/libusermodule.c python/common.h \
        python/debug.h
 python_libusermodule_la_CPPFLAGS = $(AM_CPPFLAGS) $(PYTHON_CPPFLAGS)
-python_libusermodule_la_LDFLAGS = -module -avoid-version $(GOBJECT_LIBS)
+python_libusermodule_la_LDFLAGS = -module -avoid-version $(GOBJECT_LIBS) -lpython$(PYTHON_VERSION)
 python_libusermodule_la_LIBADD = lib/libuser.la
 
 samples_enum_CPPFLAGS = $(AM_CPPFLAGS) $(LOCALEDIR_CPPFLAGS)
