--- modules/files.c.str	2008-12-27 08:27:58.000000000 +0100
+++ modules/files.c	2008-12-27 08:28:24.000000000 +0100
@@ -281,7 +281,7 @@
 						 string, &err);
 	if (ret == FALSE) {
 		g_assert(err != NULL);
-		g_warning(lu_strerror(err));
+		g_warning("%s",lu_strerror(err));
 		lu_error_free(&err);
 	}
 	return ret;
--- modules/ldap.c.str	2008-12-27 08:28:41.000000000 +0100
+++ modules/ldap.c	2008-12-27 08:29:03.000000000 +0100
@@ -677,7 +677,7 @@
 						(&value, attr, val, &error);
 					if (ok == FALSE) {
 						g_assert(error != NULL);
-						g_warning(lu_strerror(error));
+						g_warning("%s",lu_strerror(error));
 						lu_error_free(&error);
 					} else {
 						lu_ent_add_current(ent, attr,
