Index: cups-1.4.2/backend/Makefile
===================================================================
--- cups-1.4.2.orig/backend/Makefile
+++ cups-1.4.2/backend/Makefile
@@ -91,6 +91,7 @@ install-exec:
 	done
 	$(RM) $(SERVERBIN)/backend/http
 	$(LN) ipp $(SERVERBIN)/backend/http
+	$(LN) ipp $(SERVERBIN)/backend/https
 	if test "x$(DNSSD_BACKEND)" != x; then \
 		$(RM) $(SERVERBIN)/backend/mdns; \
 		$(LN) $(DNSSD_BACKEND) $(SERVERBIN)/backend/mdns; \
