--- libtiff/tif_jpeg.c	2010-09-30 13:43:31.045146032 +0200
+++ libtiff/tif_jpeg.c.oden	2010-09-30 13:44:59.005141117 +0200
@@ -850,6 +850,7 @@ JPEGPreDecode(TIFF* tif, tsample_t s)
 	if (downsampled_output) {
 		/* Need to use raw-data interface to libjpeg */
 		sp->cinfo.d.raw_data_out = TRUE;
+		sp->cinfo.d.do_fancy_upsampling = FALSE;
 		tif->tif_decoderow = JPEGDecodeRaw;
 		tif->tif_decodestrip = JPEGDecodeRaw;
 		tif->tif_decodetile = JPEGDecodeRaw;
