Index: src/mime/codeblocks.desktop
===================================================================
--- src/mime/codeblocks.desktop.orig
+++ src/mime/codeblocks.desktop
@@ -3,6 +3,7 @@ Version=1.0
 Type=Application
 Name=Code::Blocks IDE
 Comment=Configurable and extensible IDE
+Comment[tr]=Yapılandırılabilir ve eklentilerle genişletilebilir IDE
 Exec=codeblocks %F
 Icon=codeblocks
 Terminal=false
