--- codeblocks-12.11release8629-orig/src/plugins/debuggergdb/parsewatchvalue.cpp	2012-11-23 20:58:15.000000000 +0200
+++ codeblocks-12.11release8629/src/plugins/debuggergdb/parsewatchvalue.cpp	2013-01-22 22:03:44.591542259 +0200
@@ -64,7 +64,7 @@
 };
 
 wxRegEx regexRepeatedChars(wxT("^((\\\\'.{1,6}\\\\')|('.{1,6}'))[ \\t](<repeats[ \\t][0-9]+[ \\t]times>)"),
-#ifndef __WXMAC__
+#ifdef wxHAS_REGEX_ADVANCED
                            wxRE_ADVANCED);
 #else
                            wxRE_EXTENDED);
--- codeblocks-12.11release8629-orig/src/plugins/debuggergdb/gdb_commands.h	2012-11-23 20:58:15.000000000 +0200
+++ codeblocks-12.11release8629/src/plugins/debuggergdb/gdb_commands.h	2013-01-22 22:22:52.220502525 +0200
@@ -201,13 +201,13 @@
 //static wxRegEx reStepI(_T("\x1a\x1a.*?:([0-9]*):([0-9]*):(.*?):(.*)"));
 //static wxRegEx reStepI(_T("\x1a\x1a(([a-zA-Z]:)?.*?):([0-9]*):([0-9]*):(middle|beg):(.*)"));
 static wxRegEx reStepI(wxT("(((.*)[a-zA-Z]:)?.*)?:(\\d+):(middle|beg):(.*)"),
-#ifndef __WXMAC__
+#ifdef wxHAS_REGEX_ADVANCED
                        wxRE_ADVANCED);
 #else
                        wxRE_EXTENDED);
 #endif
 static wxRegEx reStepI2(_T("\\A(0x[A-Fa-f0-9]+)\\s+(\\d+)\\s+in (.*)"),
-#ifndef __WXMAC__
+#ifdef wxHAS_REGEX_ADVANCED
                        wxRE_ADVANCED);
 #else
                        wxRE_EXTENDED);
@@ -216,7 +216,7 @@
 static wxRegEx reStepI4(_T("^(0x[A-Fa-f0-9]+) in (.*)? at (.*)"));
 
 static wxRegEx reNextI(_T("\x1a\x1a(([a-zA-Z]:)?.*?):([0-9]*):([0-9]*):(middle|beg):(.*)"),
-#ifndef __WXMAC__
+#ifdef wxHAS_REGEX_ADVANCED
                        wxRE_ADVANCED);
 #else
                        wxRE_EXTENDED);
@@ -627,36 +627,6 @@
 };
 
 /**
-  * Command to setup an exception breakpoint (for a throw or a catch).
-  */
-class GdbCmd_SetCatch : public DebuggerCmd
-{
-        wxString m_type;
-        int *m_resultIndex;
-        wxRegEx m_regExp;
-    public:
-        GdbCmd_SetCatch(DebuggerDriver *driver, const wxString &type, int *resultIndex) :
-            DebuggerCmd(driver),
-            m_type(type),
-            m_resultIndex(resultIndex),
-            m_regExp(wxT("^Catchpoint[ \\t]([0-9]+)[ \\t]\\(") + type + wxT("\\)$"), wxRE_ADVANCED)
-        {
-            m_Cmd = wxT("catch ") + type;
-        }
-
-        void ParseOutput(const wxString& output)
-        {
-            if (m_regExp.Matches(output))
-            {
-                long index;
-                m_regExp.GetMatch(output, 1).ToLong(&index);
-                *m_resultIndex = index;
-            }
-        }
-};
-
-
-/**
   * Command that notifies the debugger plugin that the debuggee has been continued
   */
 class GdbCmd_Continue : public DebuggerContinueBaseCmd
