diff -up hplip-3.10.5/prnt/ps/hp-laserjet_8150_mfp-ps.ppd.ImageableArea hplip-3.10.5/prnt/ps/hp-laserjet_8150_mfp-ps.ppd
--- hplip-3.10.5/prnt/ps/hp-laserjet_8150_mfp-ps.ppd.ImageableArea	2010-05-13 01:48:11.000000000 +0200
+++ hplip-3.10.5/prnt/ps/hp-laserjet_8150_mfp-ps.ppd	2010-06-03 17:01:33.000000000 +0200
@@ -4977,7 +4977,7 @@ restore
 *CloseUI: *PageRegion
 
 *DefaultImageableArea: Letter
-*ImageableArea Letter/Letter:							"4.00 3.00 606.00 786.00"
+*ImageableArea Letter/Letter: "12.00 12.00 599.76 779.76"
 *da.ImageableArea Letter/Letter: ""
 *de.ImageableArea Letter/Letter: ""
 *es.ImageableArea Letter/Letter: ""
@@ -5009,7 +5009,7 @@ restore
 *zh_CN.ImageableArea LetterSmall/信纸 (小): ""
 *zh_TW.ImageableArea LetterSmall/Letter (小): ""
 
-*ImageableArea Executive/Executive:						"3.00 3.00 516.00 750.00"
+*ImageableArea Executive/Executive: "12.00 12.00 509.76 743.76"
 *da.ImageableArea Executive/Executive: ""
 *de.ImageableArea Executive/Executive: ""
 *es.ImageableArea Executive/Exec: ""
@@ -5025,7 +5025,7 @@ restore
 *zh_CN.ImageableArea Executive/Executive: ""
 *zh_TW.ImageableArea Executive/Executive: ""
 
-*ImageableArea Legal/Legal:								"64.00 54.00 606.00 1002.00"
+*ImageableArea Legal/Legal: "12.00 12.00 599.76 995.76"
 *da.ImageableArea Legal/Legal: ""
 *de.ImageableArea Legal/Legal: ""
 *es.ImageableArea Legal/Legal: ""
@@ -5041,7 +5041,7 @@ restore
 *zh_CN.ImageableArea Legal/Legal: ""
 *zh_TW.ImageableArea Legal/Legal: ""
 
-*ImageableArea LegalSmall/Legal (Small):				"3.00 3.00 548.00 954.00"
+*ImageableArea LegalSmall/Legal (Small): "64.00 54.00 548.00 954.00"
 *da.ImageableArea LegalSmall/Legal (lille): ""
 *de.ImageableArea LegalSmall/Legal (Klein): ""
 *es.ImageableArea LegalSmall/Legal (pequeño): ""
@@ -5057,7 +5057,7 @@ restore
 *zh_CN.ImageableArea LegalSmall/Legal (小): ""
 *zh_TW.ImageableArea LegalSmall/Legal (小): ""
 
-*ImageableArea Tabloid/11x17:							"3.00 3.00 786.00 1218.00"
+*ImageableArea Tabloid/11x17: "12.00 12.00 779.76 1211.76"
 *da.ImageableArea Tabloid/11x17: ""
 *de.ImageableArea Tabloid/11x17 Zoll: ""
 *es.ImageableArea Tabloid/11x17: ""
@@ -5073,7 +5073,7 @@ restore
 *zh_CN.ImageableArea Tabloid/11x17: ""
 *zh_TW.ImageableArea Tabloid/11x17 : ""
 
-*ImageableArea w842h1274/11x17 (Oversize 11.7x17.7):	"3.00 3.00 836.00 1268.00"
+*ImageableArea w842h1274/11x17 (Oversize 11.7x17.7): "12.00 12.00 829.68 1261.68"
 *da.ImageableArea w842h1274/11x17 (overstr.  297 x 450 mm): ""
 *de.ImageableArea w842h1274/11x17 Zoll (Übergröße 11,7x17,7 Zoll): ""
 *es.ImageableArea w842h1274/11x17 (Extra 11,7x17,7) : ""
@@ -5089,7 +5089,7 @@ restore
 *zh_CN.ImageableArea w842h1274/11x17 (超大尺寸 11.7x17.7): ""
 *zh_TW.ImageableArea w842h1274/11x17  (Oversize 11.7x17.7): ""
 
-*ImageableArea A3/A3:									"3.00 3.00 836.00 1185.00"
+*ImageableArea A3/A3: "12.00 12.00 829.44 1178.16"
 *da.ImageableArea A3/A3: ""
 *de.ImageableArea A3/A3: ""
 *es.ImageableArea A3/A3: ""
@@ -5105,7 +5105,7 @@ restore
 *zh_CN.ImageableArea A3/A3: ""
 *zh_TW.ImageableArea A3/A3: ""
 
-*ImageableArea A4/A4:									"4.00 3.00 586.00 836.00"
+*ImageableArea A4/A4: "12.00 12.00 582.96 829.44"
 *da.ImageableArea A4/A4: ""
 *de.ImageableArea A4/A4: ""
 *es.ImageableArea A4/A4: ""
@@ -5137,7 +5137,7 @@ restore
 *zh_CN.ImageableArea A4Small/A4 (小): ""
 *zh_TW.ImageableArea A4Small/A4 (小): ""
 
-*ImageableArea A5/A5:									"3.00 3.00 414.00 589.00"
+*ImageableArea A5/A5: "12.00 12.00 407.28 582.96"
 *da.ImageableArea A5/A5: ""
 *de.ImageableArea A5/A5: ""
 *es.ImageableArea A5/A5: ""
@@ -5153,7 +5153,7 @@ restore
 *zh_CN.ImageableArea A5/A5: ""
 *zh_TW.ImageableArea A5/A5: ""
 
-*ImageableArea B4/JIS B4:								"3.00 3.00 723.00 1026.00"
+*ImageableArea B4/JIS B4: "12.00 12.00 716.16 1019.52"
 *da.ImageableArea B4/JIS B4: ""
 *de.ImageableArea B4/B4 (JIS): ""
 *es.ImageableArea B4/JIS B4: ""
@@ -5169,7 +5169,7 @@ restore
 *zh_CN.ImageableArea B4/JIS B4: ""
 *zh_TW.ImageableArea B4/JIS B4: ""
 
-*ImageableArea B5/JIS B5:								"3.00 3.00 510.00 723.00"
+*ImageableArea B5/JIS B5: "12.00 12.00 503.52 716.16"
 *da.ImageableArea B5/JIS B5: ""
 *de.ImageableArea B5/JIS B5: ""
 *es.ImageableArea B5/JIS B5: ""
@@ -5185,7 +5185,7 @@ restore
 *zh_CN.ImageableArea B5/JIS B5: ""
 *zh_TW.ImageableArea B5/JIS B5: ""
 
-*ImageableArea DoublePostcard/Double Postcard (JIS):	"3.00 3.00 413.50 561.00"
+*ImageableArea DoublePostcard/Double Postcard (JIS): "12.00 12.00 407.28 554.64"
 *da.ImageableArea DoublePostcard/Dobbelt postkort (JIS): ""
 *de.ImageableArea DoublePostcard/Doppelte Postkarte (JIS): ""
 *es.ImageableArea DoublePostcard/Tarjeta postal doble (JIS): ""
@@ -5201,7 +5201,7 @@ restore
 *zh_CN.ImageableArea DoublePostcard/大号明信片 (JIS): ""
 *zh_TW.ImageableArea DoublePostcard/雙聯明信片(JIS): ""
 
-*ImageableArea w612h935/Executive (JIS):				"3.00 3.00 606.00 929.00"
+*ImageableArea w612h935/Executive (JIS): "12.00 12.00 599.76 922.76"
 *da.ImageableArea w612h935/Executive (JIS): ""
 *de.ImageableArea w612h935/Executive (JIS): ""
 *es.ImageableArea w612h935/Exec (JIS): ""
@@ -5217,7 +5217,7 @@ restore
 *zh_CN.ImageableArea w612h935/Executive (JIS): ""
 *zh_TW.ImageableArea w612h935/Executive (JIS): ""
 
-*ImageableArea w558h774/16K:							"3.00 3.00 552.00 768.00"
+*ImageableArea w558h774/16K: "12.00 12.00 545.76 761.76"
 *da.ImageableArea w558h774/16K: ""
 *de.ImageableArea w558h774/16K: ""
 *es.ImageableArea w558h774/16K: ""
@@ -5233,7 +5233,7 @@ restore
 *zh_CN.ImageableArea w558h774/16K: ""
 *zh_TW.ImageableArea w558h774/16K: ""
 
-*ImageableArea w774h1116/8K:							"3.00 3.00 768.00 1110.00"
+*ImageableArea w774h1116/8K: "12.00 12.00 761.76 1103.76"
 *da.ImageableArea w774h1116/8K: ""
 *de.ImageableArea w774h1116/8K: ""
 *es.ImageableArea w774h1116/8K: ""
@@ -5249,7 +5249,7 @@ restore
 *zh_CN.ImageableArea w774h1116/8K: ""
 *zh_TW.ImageableArea w774h1116/8K: ""
 
-*ImageableArea Env10/Env Comm10:						"3.00 3.00 291.00 678.00"
+*ImageableArea Env10/Env Comm10: "12.00 12.00 284.64 671.76"
 *da.ImageableArea Env10/Konvolut Comm10: ""
 *de.ImageableArea Env10/Umschlag Comm10: ""
 *es.ImageableArea Env10/Sobre Comm10: ""
@@ -5265,7 +5265,7 @@ restore
 *zh_CN.ImageableArea Env10/Comm10号信封: ""
 *zh_TW.ImageableArea Env10/Comm10 信封: ""
 
-*ImageableArea EnvMonarch/Env Monarch:					"3.00 3.00 273.00 534.00"
+*ImageableArea EnvMonarch/Env Monarch: "12.00 12.00 266.64 527.76"
 *da.ImageableArea EnvMonarch/Konvolut Monarch: ""
 *de.ImageableArea EnvMonarch/Umschlag Monarch: ""
 *es.ImageableArea EnvMonarch/Sobre Monarch: ""
@@ -5281,7 +5281,7 @@ restore
 *zh_CN.ImageableArea EnvMonarch/Monarch号信封: ""
 *zh_TW.ImageableArea EnvMonarch/Monarch 信封: ""
 
-*ImageableArea EnvDL/Env DL:							"3.00 3.00 306.00 618.00"
+*ImageableArea EnvDL/Env DL: "12.00 12.00 299.52 611.28"
 *da.ImageableArea EnvDL/Konvolut DL: ""
 *de.ImageableArea EnvDL/Umschlag DL: ""
 *es.ImageableArea EnvDL/Sobre DL: ""
@@ -5297,7 +5297,7 @@ restore
 *zh_CN.ImageableArea EnvDL/DL号信封: ""
 *zh_TW.ImageableArea EnvDL/DL 信封: ""
 
-*ImageableArea EnvC5/Env C5:							"3.00 3.00 453.00 643.00"
+*ImageableArea EnvC5/Env C5: "12.00 12.00 446.88 636.72"
 *da.ImageableArea EnvC5/Konvolut C5: ""
 *de.ImageableArea EnvC5/Umschlag C5: ""
 *es.ImageableArea EnvC5/Sobre C5: ""
@@ -5313,7 +5313,7 @@ restore
 *zh_CN.ImageableArea EnvC5/C5号信封: ""
 *zh_TW.ImageableArea EnvC5/C5 信封: ""
 
-*ImageableArea EnvISOB5/Env ISO B5:						"3.00 3.00 493.00 703.00"
+*ImageableArea EnvISOB5/Env ISO B5: "12.00 12.00 486.48 696.24"
 *da.ImageableArea EnvISOB5/Konvolut ISO B5: ""
 *de.ImageableArea EnvISOB5/Umschlag ISO B5: ""
 *es.ImageableArea EnvISOB5/Sobre ISO B5: ""
diff -up hplip-3.10.5/prnt/ps/hp-laserjet_9000_mfp-ps.ppd.ImageableArea hplip-3.10.5/prnt/ps/hp-laserjet_9000_mfp-ps.ppd
--- hplip-3.10.5/prnt/ps/hp-laserjet_9000_mfp-ps.ppd.ImageableArea	2010-05-13 01:48:11.000000000 +0200
+++ hplip-3.10.5/prnt/ps/hp-laserjet_9000_mfp-ps.ppd	2010-06-03 17:01:32.000000000 +0200
@@ -4344,7 +4344,7 @@ currentpagedevice /MediaProcessing known
 *CloseUI: *PageRegion
 
 *DefaultImageableArea: Letter
-*ImageableArea Letter/Letter:							"6.00 6.00 606.00 786.00"
+*ImageableArea Letter/Letter: "12.00 12.00 599.76 779.76"
 *da.ImageableArea Letter/Letter: ""
 *de.ImageableArea Letter/Letter: ""
 *es.ImageableArea Letter/Letter: ""
@@ -4376,7 +4376,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea LetterSmall/信纸 (小): ""
 *zh_TW.ImageableArea LetterSmall/Letter (小): ""
 
-*ImageableArea Executive/Executive:						"6.00 6.00 516.00 750.00"
+*ImageableArea Executive/Executive: "12.00 12.00 509.76 743.76"
 *da.ImageableArea Executive/Executive: ""
 *de.ImageableArea Executive/Executive: ""
 *es.ImageableArea Executive/Exec: ""
@@ -4392,7 +4392,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea Executive/Executive: ""
 *zh_TW.ImageableArea Executive/Executive: ""
 
-*ImageableArea Legal/Legal:								"6.00 6.00 606.00 1002.00"
+*ImageableArea Legal/Legal: "12.00 12.00 599.76 995.76"
 *da.ImageableArea Legal/Legal: ""
 *de.ImageableArea Legal/Legal: ""
 *es.ImageableArea Legal/Legal: ""
@@ -4424,7 +4424,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea LegalSmall/Legal (小): ""
 *zh_TW.ImageableArea LegalSmall/Legal (小): ""
 
-*ImageableArea Tabloid/11x17:							"6.00 6.00 786.00 1218.00"
+*ImageableArea Tabloid/11x17: "12.00 12.00 779.76 1211.76"
 *da.ImageableArea Tabloid/11x17: ""
 *de.ImageableArea Tabloid/11x17 Zoll: ""
 *es.ImageableArea Tabloid/11x17: ""
@@ -4440,7 +4440,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea Tabloid/11x17: ""
 *zh_TW.ImageableArea Tabloid/11x17 : ""
 
-*ImageableArea w842h1274/11x17 (Oversize 11.7x17.7):	"6.00 6.00 836.00 1268.00"
+*ImageableArea w842h1274/11x17 (Oversize 11.7x17.7): "12.00 12.00 829.68 1261.68"
 *da.ImageableArea w842h1274/11x17 (overstr.  297 x 450 mm): ""
 *de.ImageableArea w842h1274/11x17 Zoll (Übergröße 11,7x17,7 Zoll): ""
 *es.ImageableArea w842h1274/11x17 (Extra 11,7x17,7) : ""
@@ -4456,7 +4456,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea w842h1274/11x17 (超大尺寸 11.7x17.7): ""
 *zh_TW.ImageableArea w842h1274/11x17  (Oversize 11.7x17.7): ""
 
-*ImageableArea A3/A3:									"6.00 6.00 836.00 1185.00"
+*ImageableArea A3/A3: "12.00 12.00 829.44 1178.16"
 *da.ImageableArea A3/A3: ""
 *de.ImageableArea A3/A3: ""
 *es.ImageableArea A3/A3: ""
@@ -4472,7 +4472,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea A3/A3: ""
 *zh_TW.ImageableArea A3/A3: ""
 
-*ImageableArea A4/A4:									"6.00 6.00 589.00 836.00"
+*ImageableArea A4/A4: "12.00 12.00 582.96 829.44"
 *da.ImageableArea A4/A4: ""
 *de.ImageableArea A4/A4: ""
 *es.ImageableArea A4/A4: ""
@@ -4504,7 +4504,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea A4Small/A4 (小): ""
 *zh_TW.ImageableArea A4Small/A4 (小): ""
 
-*ImageableArea A5/A5:									"6.00 6.00 414.00 589.00"
+*ImageableArea A5/A5: "12.00 12.00 407.28 582.96"
 *da.ImageableArea A5/A5: ""
 *de.ImageableArea A5/A5: ""
 *es.ImageableArea A5/A5: ""
@@ -4520,7 +4520,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea A5/A5: ""
 *zh_TW.ImageableArea A5/A5: ""
 
-*ImageableArea B4/JIS B4:								"6.00 6.00 723.00 1026.00"
+*ImageableArea B4/JIS B4: "12.00 12.00 716.16 1019.52"
 *da.ImageableArea B4/JIS B4: ""
 *de.ImageableArea B4/B4 (JIS): ""
 *es.ImageableArea B4/JIS B4: ""
@@ -4536,7 +4536,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea B4/JIS B4: ""
 *zh_TW.ImageableArea B4/JIS B4: ""
 
-*ImageableArea B5/JIS B5:								"6.00 6.00 510.00 723.00"
+*ImageableArea B5/JIS B5: "12.00 12.00 503.52 716.16"
 *da.ImageableArea B5/JIS B5: ""
 *de.ImageableArea B5/JIS B5: ""
 *es.ImageableArea B5/JIS B5: ""
@@ -4552,7 +4552,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea B5/JIS B5: ""
 *zh_TW.ImageableArea B5/JIS B5: ""
 
-*ImageableArea DoublePostcard/Double Postcard (JIS):	"6.00 6.00 413.50 561.00"
+*ImageableArea DoublePostcard/Double Postcard (JIS): "12.00 12.00 407.28 554.64"
 *da.ImageableArea DoublePostcard/Dobbelt postkort (JIS): ""
 *de.ImageableArea DoublePostcard/Doppelte Postkarte (JIS): ""
 *es.ImageableArea DoublePostcard/Tarjeta postal doble (JIS): ""
@@ -4568,7 +4568,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea DoublePostcard/大号明信片 (JIS): ""
 *zh_TW.ImageableArea DoublePostcard/雙聯明信片(JIS): ""
 
-*ImageableArea w612h935/Executive (JIS):				"6.00 6.00 606.00 929.00"
+*ImageableArea w612h935/Executive (JIS): "12.00 12.00 599.76 922.76"
 *da.ImageableArea w612h935/Executive (JIS): ""
 *de.ImageableArea w612h935/Executive (JIS): ""
 *es.ImageableArea w612h935/Exec (JIS): ""
@@ -4584,7 +4584,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea w612h935/Executive (JIS): ""
 *zh_TW.ImageableArea w612h935/Executive (JIS): ""
 
-*ImageableArea w558h774/16K:							"6.00 6.00 552.00 768.00"
+*ImageableArea w558h774/16K: "12.00 12.00 545.76 761.76"
 *da.ImageableArea w558h774/16K: ""
 *de.ImageableArea w558h774/16K: ""
 *es.ImageableArea w558h774/16K: ""
@@ -4600,7 +4600,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea w558h774/16K: ""
 *zh_TW.ImageableArea w558h774/16K: ""
 
-*ImageableArea w774h1116/8K:							"6.00 6.00 768.00 1110.00"
+*ImageableArea w774h1116/8K: "12.00 12.00 761.76 1103.76"
 *da.ImageableArea w774h1116/8K: ""
 *de.ImageableArea w774h1116/8K: ""
 *es.ImageableArea w774h1116/8K: ""
@@ -4616,7 +4616,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea w774h1116/8K: ""
 *zh_TW.ImageableArea w774h1116/8K: ""
 
-*ImageableArea Env10/Env Comm10:						"11.00 11.00 286.00 673.00"
+*ImageableArea Env10/Env Comm10: "12.00 12.00 284.64 671.76"
 *da.ImageableArea Env10/Konvolut Comm10: ""
 *de.ImageableArea Env10/Umschlag Comm10: ""
 *es.ImageableArea Env10/Sobre Comm10: ""
diff -up hplip-3.10.5/prnt/ps/hp-laserjet_9000_series-ps.ppd.ImageableArea hplip-3.10.5/prnt/ps/hp-laserjet_9000_series-ps.ppd
--- hplip-3.10.5/prnt/ps/hp-laserjet_9000_series-ps.ppd.ImageableArea	2010-05-13 01:48:11.000000000 +0200
+++ hplip-3.10.5/prnt/ps/hp-laserjet_9000_series-ps.ppd	2010-06-03 17:01:35.000000000 +0200
@@ -5140,7 +5140,7 @@ currentpagedevice /MediaProcessing known
 *CloseUI: *PageRegion
 
 *DefaultImageableArea: Letter
-*ImageableArea Letter/Letter:							"6.00 6.00 606.00 786.00"
+*ImageableArea Letter/Letter: "12.00 12.00 599.76 779.76"
 *da.ImageableArea Letter/Letter: ""
 *de.ImageableArea Letter/Letter: ""
 *es.ImageableArea Letter/Letter: ""
@@ -5172,7 +5172,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea LetterSmall/信纸 (小): ""
 *zh_TW.ImageableArea LetterSmall/Letter (小): ""
 
-*ImageableArea Executive/Executive:						"6.00 6.00 516.00 750.00"
+*ImageableArea Executive/Executive: "12.00 12.00 509.76 743.76"
 *da.ImageableArea Executive/Executive: ""
 *de.ImageableArea Executive/Executive: ""
 *es.ImageableArea Executive/Exec: ""
@@ -5188,7 +5188,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea Executive/Executive: ""
 *zh_TW.ImageableArea Executive/Executive: ""
 
-*ImageableArea Legal/Legal:								"6.00 6.00 606.00 1002.00"
+*ImageableArea Legal/Legal: "12.00 12.00 599.76 995.76"
 *da.ImageableArea Legal/Legal: ""
 *de.ImageableArea Legal/Legal: ""
 *es.ImageableArea Legal/Legal: ""
@@ -5220,7 +5220,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea LegalSmall/Legal (小): ""
 *zh_TW.ImageableArea LegalSmall/Legal (小): ""
 
-*ImageableArea Tabloid/11x17:							"6.00 6.00 786.00 1218.00"
+*ImageableArea Tabloid/11x17: "12.00 12.00 779.76 1211.76"
 *da.ImageableArea Tabloid/11x17: ""
 *de.ImageableArea Tabloid/11x17 Zoll: ""
 *es.ImageableArea Tabloid/11x17: ""
@@ -5236,7 +5236,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea Tabloid/11x17: ""
 *zh_TW.ImageableArea Tabloid/11x17 : ""
 
-*ImageableArea w842h1274/11x17 (Oversize 11.7x17.7):	"6.00 6.00 836.00 1268.00"
+*ImageableArea w842h1274/11x17 (Oversize 11.7x17.7): "12.00 12.00 829.68 1261.68"
 *da.ImageableArea w842h1274/11x17 (overstr.  297 x 450 mm): ""
 *de.ImageableArea w842h1274/11x17 Zoll (Übergröße 11,7x17,7 Zoll): ""
 *es.ImageableArea w842h1274/11x17 (Extra 11,7x17,7) : ""
@@ -5252,7 +5252,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea w842h1274/11x17 (超大尺寸 11.7x17.7): ""
 *zh_TW.ImageableArea w842h1274/11x17  (Oversize 11.7x17.7): ""
 
-*ImageableArea A3/A3:									"6.00 6.00 836.00 1185.00"
+*ImageableArea A3/A3: "12.00 12.00 829.44 1178.16"
 *da.ImageableArea A3/A3: ""
 *de.ImageableArea A3/A3: ""
 *es.ImageableArea A3/A3: ""
@@ -5268,7 +5268,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea A3/A3: ""
 *zh_TW.ImageableArea A3/A3: ""
 
-*ImageableArea A4/A4:									"6.00 6.00 589.00 836.00"
+*ImageableArea A4/A4: "12.00 12.00 582.96 829.44"
 *da.ImageableArea A4/A4: ""
 *de.ImageableArea A4/A4: ""
 *es.ImageableArea A4/A4: ""
@@ -5300,7 +5300,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea A4Small/A4 (小): ""
 *zh_TW.ImageableArea A4Small/A4 (小): ""
 
-*ImageableArea A5/A5:									"6.00 6.00 414.00 589.00"
+*ImageableArea A5/A5: "12.00 12.00 407.28 582.96"
 *da.ImageableArea A5/A5: ""
 *de.ImageableArea A5/A5: ""
 *es.ImageableArea A5/A5: ""
@@ -5316,7 +5316,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea A5/A5: ""
 *zh_TW.ImageableArea A5/A5: ""
 
-*ImageableArea B4/JIS B4:								"6.00 6.00 723.00 1026.00"
+*ImageableArea B4/JIS B4: "12.00 12.00 716.16 1019.52"
 *da.ImageableArea B4/JIS B4: ""
 *de.ImageableArea B4/B4 (JIS): ""
 *es.ImageableArea B4/JIS B4: ""
@@ -5332,7 +5332,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea B4/JIS B4: ""
 *zh_TW.ImageableArea B4/JIS B4: ""
 
-*ImageableArea B5/JIS B5:								"6.00 6.00 510.00 723.00"
+*ImageableArea B5/JIS B5: "12.00 12.00 503.52 716.16"
 *da.ImageableArea B5/JIS B5: ""
 *de.ImageableArea B5/JIS B5: ""
 *es.ImageableArea B5/JIS B5: ""
@@ -5348,7 +5348,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea B5/JIS B5: ""
 *zh_TW.ImageableArea B5/JIS B5: ""
 
-*ImageableArea DoublePostcard/Double Postcard (JIS):	"6.00 6.00 413.50 561.00"
+*ImageableArea DoublePostcard/Double Postcard (JIS): "12.00 12.00 407.28 554.64"
 *da.ImageableArea DoublePostcard/Dobbelt postkort (JIS): ""
 *de.ImageableArea DoublePostcard/Doppelte Postkarte (JIS): ""
 *es.ImageableArea DoublePostcard/Tarjeta postal doble (JIS): ""
@@ -5364,7 +5364,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea DoublePostcard/大号明信片 (JIS): ""
 *zh_TW.ImageableArea DoublePostcard/雙聯明信片(JIS): ""
 
-*ImageableArea w612h935/Executive (JIS):				"6.00 6.00 606.00 929.00"
+*ImageableArea w612h935/Executive (JIS): "12.00 12.00 599.76 922.76"
 *da.ImageableArea w612h935/Executive (JIS): ""
 *de.ImageableArea w612h935/Executive (JIS): ""
 *es.ImageableArea w612h935/Exec (JIS): ""
@@ -5380,7 +5380,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea w612h935/Executive (JIS): ""
 *zh_TW.ImageableArea w612h935/Executive (JIS): ""
 
-*ImageableArea w558h774/16K:							"6.00 6.00 552.00 768.00"
+*ImageableArea w558h774/16K: "12.00 12.00 545.76 761.76"
 *da.ImageableArea w558h774/16K: ""
 *de.ImageableArea w558h774/16K: ""
 *es.ImageableArea w558h774/16K: ""
@@ -5396,7 +5396,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea w558h774/16K: ""
 *zh_TW.ImageableArea w558h774/16K: ""
 
-*ImageableArea w774h1116/8K:							"6.00 6.00 768.00 1110.00"
+*ImageableArea w774h1116/8K: "12.00 12.00 761.76 1103.76"
 *da.ImageableArea w774h1116/8K: ""
 *de.ImageableArea w774h1116/8K: ""
 *es.ImageableArea w774h1116/8K: ""
@@ -5412,7 +5412,7 @@ currentpagedevice /MediaProcessing known
 *zh_CN.ImageableArea w774h1116/8K: ""
 *zh_TW.ImageableArea w774h1116/8K: ""
 
-*ImageableArea Env10/Env Comm10:						"11.00 11.00 286.00 673.00"
+*ImageableArea Env10/Env Comm10: "12.00 12.00 284.64 671.76"
 *da.ImageableArea Env10/Konvolut Comm10: ""
 *de.ImageableArea Env10/Umschlag Comm10: ""
 *es.ImageableArea Env10/Sobre Comm10: ""
