diff -up hplip-3.12.11/base/utils.py.ui-optional hplip-3.12.11/base/utils.py
--- hplip-3.12.11/base/utils.py.ui-optional	2012-11-20 10:55:15.000000000 +0100
+++ hplip-3.12.11/base/utils.py	2012-11-27 10:03:21.909512264 +0100
@@ -667,6 +667,13 @@ def checkPyQtImport(): # qt3
         if os.getenv('DISPLAY') and os.getenv('STARTED_FROM_MENU'):
             no_qt_message_gtk()
 
+        # hplip-gui sub-package (Fedora)
+        try:
+            import ui
+        except ImportError:
+            log.error("hplip-gui not installed.  GUI not available. Exiting.")
+            return False
+
         log.error("PyQt not installed. GUI not available. Exiting.")
         return False
 
@@ -712,7 +719,8 @@ def checkPyQtImport4():
         import PyQt4
         import ui4
     except ImportError:
-        log.error("HPLIP is not installed properly or is installed without graphical support. Please reinstall HPLIP")
+        # hplip-gui sub-package (Fedora) requires PyQt4
+        log.error("Install the hplip-gui package for graphical support.")
         return False
     else:
         return True
