diff --git a/src/plugins/contrib/codesnippets/Makefile.am b/src/plugins/contrib/codesnippets/Makefile.am
index 220912a..3da33a8 100644
--- a/src/plugins/contrib/codesnippets/Makefile.am
+++ b/src/plugins/contrib/codesnippets/Makefile.am
@@ -13,7 +13,7 @@ pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libcodesnippets.la
 
-libcodesnippets_la_LDFLAGS = $(WX_GTK2_LIBS) -module -version-info 0:1:0 -shared -no-undefined -avoid-version
+libcodesnippets_la_LDFLAGS = $(WX_GTK2_LIBS) -lX11 -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
 libcodesnippets_la_LIBADD = $(PLUGIN_WX_LIBS) ../../../sdk/libcodeblocks.la
 
diff --git a/src/plugins/contrib/codesnippets/resources/Makefile.am b/src/plugins/contrib/codesnippets/resources/Makefile.am
index a83a6bb..3710602 100644
--- a/src/plugins/contrib/codesnippets/resources/Makefile.am
+++ b/src/plugins/contrib/codesnippets/resources/Makefile.am
@@ -18,7 +18,7 @@ codesnippets_LDFLAGS = $(WX_GTK2_LIBS)
 ##codesnippets_LDADD = $(PLUGIN_WX_LIBS) ../../../../sdk/libcodeblocks.la
 codesnippets_LDADD = -ltinyxml \
             -L../../../../sdk ../../../../sdk/libcodeblocks.la \
-			$(PLUGIN_WX_LIBS) $(WX_LIBS)
+			$(PLUGIN_WX_LIBS) $(WX_LIBS) -lX11
 
 if ! HAVE_TINYXML
 INCLUDES += -I$(top_srcdir)/src/include/tinyxml
