From 91086f2ba50998b9ec286dad6420194806868ab9 Mon Sep 17 00:00:00 2001
From: Dan Horak <dan@danny.cz>
Date: Wed, 13 May 2009 10:21:19 +0200
Subject: [PATCH 01/22] add check for tinyxml into configure

---
 configure.in |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/configure.in b/configure.in
index 17362a0..91f69bc 100644
--- a/configure.in
+++ b/configure.in
@@ -54,9 +54,18 @@ AC_SEARCH_LIBS(gzopen, z, [HAVE_ZLIB=yes], [HAVE_ZLIB=no])
 LIBS=$save_LIBS
 AC_SEARCH_LIBS(BZ2_bzopen, bz2, [HAVE_BZIP2=yes], [HAVE_BZIP2=no])
 LIBS=$save_LIBS
+save_CC=$CC
+save_CPP=$CPP
+CC=$CXX
+CPP=$CXXCPP
+AC_CHECK_HEADER(tinyxml.h, [HAVE_TINYXML=yes], [HAVE_TINYXML=no])
+CC=$save_CC
+CPP=$save_CPP
+
 
 AM_CONDITIONAL([HAVE_ZLIB], [test "$HAVE_ZLIB" = yes])
 AM_CONDITIONAL([HAVE_BZIP2], [test "$HAVE_BZIP2" = yes])
+AM_CONDITIONAL([HAVE_TINYXML], [test "$HAVE_TINYXML" = yes])
 
 dnl versioning info for libtool
 dnl Note this is the ABI version which is not the same as our actual library version
-- 
1.6.6.1

From 3816818f35986a545163228079358adefe62d8ec Mon Sep 17 00:00:00 2001
From: Dan Horak <dan@danny.cz>
Date: Wed, 13 May 2009 16:53:29 +0200
Subject: [PATCH 02/22] move tinywxuni.h header from tinyxml into sdk

---
 src/include/Makefile.am                            |    1 +
 src/include/tinywxuni.h                            |   15 +++++++++++++++
 src/include/tinyxml/Makefile.am                    |    2 +-
 src/include/tinyxml/tinywxuni.h                    |   15 ---------------
 .../contrib/BrowseTracker/BrowseTrackerLayout.cpp  |    2 +-
 .../contrib/lib_finder/librarydetectionmanager.cpp |    2 +-
 .../contrib/wxSmith/wxsversionconverter.cpp        |    2 +-
 .../contrib/wxSmith/wxwidgets/wxsitemresdata.cpp   |    2 +-
 .../wxSmith/wxwidgets/wxwidgetsresfactory.cpp      |    2 +-
 src/sdk/configmanager.cpp                          |    2 +-
 src/sdk/globals.cpp                                |    2 +-
 src/sdk/projectlayoutloader.cpp                    |    2 +-
 src/sdk/projectloader.cpp                          |    2 +-
 src/sdk/workspaceloader.cpp                        |    2 +-
 src/tools/cb_share_config/mainframe.cpp            |    2 +-
 15 files changed, 28 insertions(+), 27 deletions(-)
 create mode 100644 src/include/tinywxuni.h
 delete mode 100644 src/include/tinyxml/tinywxuni.h

diff --git a/src/include/Makefile.am b/src/include/Makefile.am
index 07658aa..9f754da 100644
--- a/src/include/Makefile.am
+++ b/src/include/Makefile.am
@@ -121,6 +121,7 @@ ALLHEADERS = annoyingdialog.h \
 				selecttargetdlg.h \
 				settings.h \
 				templatemanager.h \
+				tinywxuni.h \
 				toolsmanager.h \
 				uservarmanager.h \
 				virtualbuildtargetsdlg.h \
diff --git a/src/include/tinywxuni.h b/src/include/tinywxuni.h
new file mode 100644
index 0000000..9ebf3a5
--- /dev/null
+++ b/src/include/tinywxuni.h
@@ -0,0 +1,15 @@
+#ifndef TINYWXUNI_H
+#define TINYWXUNI_H
+
+class wxString;
+class TiXmlDocument;
+
+namespace TinyXML
+{
+    bool LoadDocument(const wxString& filename, TiXmlDocument *doc);
+    TiXmlDocument* LoadDocument(const wxString& filename);
+
+    bool SaveDocument(const wxString& filename, TiXmlDocument* doc);
+}
+
+#endif
diff --git a/src/include/tinyxml/Makefile.am b/src/include/tinyxml/Makefile.am
index fdfa2c5..c22e938 100644
--- a/src/include/tinyxml/Makefile.am
+++ b/src/include/tinyxml/Makefile.am
@@ -1,3 +1,3 @@
 sdkdir=$(includedir)/codeblocks/tinyxml
 
-sdk_HEADERS = tinystr.h tinyxml.h tinywxuni.h
+sdk_HEADERS = tinystr.h tinyxml.h
diff --git a/src/include/tinyxml/tinywxuni.h b/src/include/tinyxml/tinywxuni.h
deleted file mode 100644
index 9ebf3a5..0000000
--- a/src/include/tinyxml/tinywxuni.h
+++ /dev/null
@@ -1,15 +0,0 @@
-#ifndef TINYWXUNI_H
-#define TINYWXUNI_H
-
-class wxString;
-class TiXmlDocument;
-
-namespace TinyXML
-{
-    bool LoadDocument(const wxString& filename, TiXmlDocument *doc);
-    TiXmlDocument* LoadDocument(const wxString& filename);
-
-    bool SaveDocument(const wxString& filename, TiXmlDocument* doc);
-}
-
-#endif
diff --git a/src/plugins/contrib/BrowseTracker/BrowseTrackerLayout.cpp b/src/plugins/contrib/BrowseTracker/BrowseTrackerLayout.cpp
index 7eeb514..30f38d4 100644
--- a/src/plugins/contrib/BrowseTracker/BrowseTrackerLayout.cpp
+++ b/src/plugins/contrib/BrowseTracker/BrowseTrackerLayout.cpp
@@ -47,7 +47,7 @@
 
 #include <wx/tokenzr.h>
 #include "tinyxml/tinyxml.h"
-#include "tinyxml/tinywxuni.h"
+#include "tinywxuni.h"
 
 #include "BrowseTrackerLayout.h"
 #include "BrowseMarks.h"
diff --git a/src/plugins/contrib/lib_finder/librarydetectionmanager.cpp b/src/plugins/contrib/lib_finder/librarydetectionmanager.cpp
index db01cf2..b61adf8 100644
--- a/src/plugins/contrib/lib_finder/librarydetectionmanager.cpp
+++ b/src/plugins/contrib/lib_finder/librarydetectionmanager.cpp
@@ -22,7 +22,7 @@
 */
 
 #include <tinyxml/tinyxml.h>
-#include <tinyxml/tinywxuni.h>
+#include <tinywxuni.h>
 
 #include <wx/arrstr.h>
 #include <wx/dir.h>
diff --git a/src/plugins/contrib/wxSmith/wxsversionconverter.cpp b/src/plugins/contrib/wxSmith/wxsversionconverter.cpp
index b03ceaa..8c2ca85 100644
--- a/src/plugins/contrib/wxSmith/wxsversionconverter.cpp
+++ b/src/plugins/contrib/wxSmith/wxsversionconverter.cpp
@@ -26,7 +26,7 @@
 
 #include <globals.h>
 #include <wx/string.h>
-#include <tinyxml/tinywxuni.h>
+#include <tinywxuni.h>
 
 // TODO: This is same as in wxsproject.h, make one set instead of two
 namespace
diff --git a/src/plugins/contrib/wxSmith/wxwidgets/wxsitemresdata.cpp b/src/plugins/contrib/wxSmith/wxwidgets/wxsitemresdata.cpp
index 77b4e69..d241a5c 100644
--- a/src/plugins/contrib/wxSmith/wxwidgets/wxsitemresdata.cpp
+++ b/src/plugins/contrib/wxSmith/wxwidgets/wxsitemresdata.cpp
@@ -31,7 +31,7 @@
 #include <globals.h>
 #include <logmanager.h>
 #include <wx/clipbrd.h>
-#include <tinyxml/tinywxuni.h>
+#include <tinywxuni.h>
 
 #if defined(__WXMSW__) && defined(LoadImage)
     // Fix Windows winuser.h Header define of LoadImage.
diff --git a/src/plugins/contrib/wxSmith/wxwidgets/wxwidgetsresfactory.cpp b/src/plugins/contrib/wxSmith/wxwidgets/wxwidgetsresfactory.cpp
index 9517cf3..24d5479 100644
--- a/src/plugins/contrib/wxSmith/wxwidgets/wxwidgetsresfactory.cpp
+++ b/src/plugins/contrib/wxSmith/wxwidgets/wxwidgetsresfactory.cpp
@@ -31,7 +31,7 @@
 #include "../wxsmith.h"
 
 #include <wx/choicdlg.h>
-#include <tinyxml/tinywxuni.h>
+#include <tinywxuni.h>
 #include <sqplus.h>
 #include <sc_base_types.h>
 
diff --git a/src/sdk/configmanager.cpp b/src/sdk/configmanager.cpp
index 2fc8726..c893274 100644
--- a/src/sdk/configmanager.cpp
+++ b/src/sdk/configmanager.cpp
@@ -43,7 +43,7 @@
 #include <CoreFoundation/CFURL.h>
 #endif
 
-#include "tinyxml/tinywxuni.h"
+#include "tinywxuni.h"
 #include <stdlib.h>
 
 template<> CfgMgrBldr* Mgr<CfgMgrBldr>::instance = 0;
diff --git a/src/sdk/globals.cpp b/src/sdk/globals.cpp
index 15da26b..cceb1e9 100644
--- a/src/sdk/globals.cpp
+++ b/src/sdk/globals.cpp
@@ -32,7 +32,7 @@
 #include <algorithm>
 #include <string>
 #include "filefilters.h"
-#include "tinyxml/tinywxuni.h"
+#include "tinywxuni.h"
 #include "filegroupsandmasks.h"
 
 #ifndef __WXMSW__
diff --git a/src/sdk/projectlayoutloader.cpp b/src/sdk/projectlayoutloader.cpp
index 9cb870a..472b053 100644
--- a/src/sdk/projectlayoutloader.cpp
+++ b/src/sdk/projectlayoutloader.cpp
@@ -23,7 +23,7 @@
 
 #include "projectlayoutloader.h"
 #include "tinyxml/tinyxml.h"
-#include "tinyxml/tinywxuni.h"
+#include "tinywxuni.h"
 
 ProjectLayoutLoader::ProjectLayoutLoader(cbProject* project)
     : m_pProject(project),
diff --git a/src/sdk/projectloader.cpp b/src/sdk/projectloader.cpp
index bf89f25..aa4f54f 100644
--- a/src/sdk/projectloader.cpp
+++ b/src/sdk/projectloader.cpp
@@ -31,7 +31,7 @@
 #include "projectloader_hooks.h"
 #include "annoyingdialog.h"
 #include "configmanager.h"
-#include "tinyxml/tinywxuni.h"
+#include "tinywxuni.h"
 
 ProjectLoader::ProjectLoader(cbProject* project)
     : m_pProject(project),
diff --git a/src/sdk/workspaceloader.cpp b/src/sdk/workspaceloader.cpp
index d44b0b4..f2624be 100644
--- a/src/sdk/workspaceloader.cpp
+++ b/src/sdk/workspaceloader.cpp
@@ -26,7 +26,7 @@
 
 
 #include "tinyxml/tinyxml.h"
-#include "tinyxml/tinywxuni.h"
+#include "tinywxuni.h"
 
 WorkspaceLoader::WorkspaceLoader()
 {
diff --git a/src/tools/cb_share_config/mainframe.cpp b/src/tools/cb_share_config/mainframe.cpp
index 02803b6..f43b15c 100644
--- a/src/tools/cb_share_config/mainframe.cpp
+++ b/src/tools/cb_share_config/mainframe.cpp
@@ -24,7 +24,7 @@
   #include <shlobj.h>
 #endif
 
-#include "tinyxml/tinywxuni.h"
+#include "tinywxuni.h"
 
 //***********************************************************************
 
-- 
1.6.6.1

From bc052dd162594f68126177d523092f43937d1936 Mon Sep 17 00:00:00 2001
From: Dan Horak <dan@danny.cz>
Date: Thu, 14 May 2009 09:42:37 +0200
Subject: [PATCH 03/22] move tinywxuni.cpp implementation from tinyxml into sdk

---
 src/base/tinyxml/Makefile.am   |    2 +-
 src/base/tinyxml/tinywxuni.cpp |   62 ----------------------------------------
 src/sdk/Makefile.am            |    1 +
 src/sdk/tinywxuni.cpp          |   62 ++++++++++++++++++++++++++++++++++++++++
 4 files changed, 64 insertions(+), 63 deletions(-)
 delete mode 100644 src/base/tinyxml/tinywxuni.cpp
 create mode 100644 src/sdk/tinywxuni.cpp

diff --git a/src/base/tinyxml/Makefile.am b/src/base/tinyxml/Makefile.am
index bc3e89d..c480f0c 100644
--- a/src/base/tinyxml/Makefile.am
+++ b/src/base/tinyxml/Makefile.am
@@ -5,4 +5,4 @@ INCLUDES = $(WX_CXXFLAGS) \
 
 noinst_LTLIBRARIES = libtinyxml.la
 
-libtinyxml_la_SOURCES = tinystr.cpp tinyxml.cpp tinyxmlerror.cpp tinyxmlparser.cpp tinywxuni.cpp
+libtinyxml_la_SOURCES = tinystr.cpp tinyxml.cpp tinyxmlerror.cpp tinyxmlparser.cpp
diff --git a/src/base/tinyxml/tinywxuni.cpp b/src/base/tinyxml/tinywxuni.cpp
deleted file mode 100644
index 5cfc5ee..0000000
--- a/src/base/tinyxml/tinywxuni.cpp
+++ /dev/null
@@ -1,62 +0,0 @@
-#include "sdk_precomp.h"
-
-#ifndef CB_PRECOMP
-#include <wx/file.h>
-#include <wx/string.h>
-#include "filemanager.h"
-#include "manager.h"
-#endif
-#include "tinywxuni.h"
-#include "tinyxml.h"
-
-bool TinyXML::LoadDocument(const wxString& filename, TiXmlDocument *doc)
-{
-
-    if(!doc || !wxFile::Access(filename, wxFile::read))
-        return false;
-
-    wxFile file(filename);
-    size_t len = file.Length();
-
-    char *input = new char[len+1];
-    input[len] = '\0';
-    file.Read(input, len);
-
-    doc->Parse(input);
-    delete[] input;
-    return true;
-}
-
-TiXmlDocument* TinyXML::LoadDocument(const wxString& filename)
-{
-    TiXmlDocument* doc = new TiXmlDocument();
-
-    if(TinyXML::LoadDocument(filename, doc))
-    {
-        return doc;
-    }
-    else
-    {
-        delete doc;
-        return 0;
-    }
-}
-
-bool TinyXML::SaveDocument(const wxString& filename, TiXmlDocument* doc)
-{
-    if (!doc)
-        return false;
-
-    TiXmlPrinter printer;
-    printer.SetIndent("\t");
-    doc->Accept(&printer);
-
-    return Manager::Get()->GetFileManager()->Save(filename, printer.CStr(), printer.Size());
-
-//    wxTempFile file(filename);
-//    if(file.IsOpened())
-//        if(file.Write(Printer.CStr(), Printer.Size()) && file.Commit())
-//            return true;
-//    return false;
-}
-
diff --git a/src/sdk/Makefile.am b/src/sdk/Makefile.am
index 0e84492..66a2620 100644
--- a/src/sdk/Makefile.am
+++ b/src/sdk/Makefile.am
@@ -101,6 +101,7 @@ libcodeblocks_la_SOURCES = annoyingdialog.cpp \
 						searchresultslog.cpp \
 						selecttargetdlg.cpp \
 						templatemanager.cpp \
+						tinywxuni.cpp \
 						toolsmanager.cpp \
 						uservarmanager.cpp \
 						virtualbuildtargetsdlg.cpp \
diff --git a/src/sdk/tinywxuni.cpp b/src/sdk/tinywxuni.cpp
new file mode 100644
index 0000000..5cfc5ee
--- /dev/null
+++ b/src/sdk/tinywxuni.cpp
@@ -0,0 +1,62 @@
+#include "sdk_precomp.h"
+
+#ifndef CB_PRECOMP
+#include <wx/file.h>
+#include <wx/string.h>
+#include "filemanager.h"
+#include "manager.h"
+#endif
+#include "tinywxuni.h"
+#include "tinyxml.h"
+
+bool TinyXML::LoadDocument(const wxString& filename, TiXmlDocument *doc)
+{
+
+    if(!doc || !wxFile::Access(filename, wxFile::read))
+        return false;
+
+    wxFile file(filename);
+    size_t len = file.Length();
+
+    char *input = new char[len+1];
+    input[len] = '\0';
+    file.Read(input, len);
+
+    doc->Parse(input);
+    delete[] input;
+    return true;
+}
+
+TiXmlDocument* TinyXML::LoadDocument(const wxString& filename)
+{
+    TiXmlDocument* doc = new TiXmlDocument();
+
+    if(TinyXML::LoadDocument(filename, doc))
+    {
+        return doc;
+    }
+    else
+    {
+        delete doc;
+        return 0;
+    }
+}
+
+bool TinyXML::SaveDocument(const wxString& filename, TiXmlDocument* doc)
+{
+    if (!doc)
+        return false;
+
+    TiXmlPrinter printer;
+    printer.SetIndent("\t");
+    doc->Accept(&printer);
+
+    return Manager::Get()->GetFileManager()->Save(filename, printer.CStr(), printer.Size());
+
+//    wxTempFile file(filename);
+//    if(file.IsOpened())
+//        if(file.Write(Printer.CStr(), Printer.Size()) && file.Commit())
+//            return true;
+//    return false;
+}
+
-- 
1.6.6.1

From 7362315ce2418e4fd00513f966b661463141bb55 Mon Sep 17 00:00:00 2001
From: Dan Horak <dan@danny.cz>
Date: Thu, 14 May 2009 10:04:35 +0200
Subject: [PATCH 04/22] update C::B projects with new locations of tinywxuni

---
 src/CodeBlocks-unix.cbp |    4 ++--
 src/CodeBlocks.cbp      |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/CodeBlocks-unix.cbp b/src/CodeBlocks-unix.cbp
index eb2d9da..11ee731 100644
--- a/src/CodeBlocks-unix.cbp
+++ b/src/CodeBlocks-unix.cbp
@@ -400,7 +400,7 @@
 		<Unit filename="base/tinyxml/tinystr.cpp">
 			<Option target="tinyXML" />
 		</Unit>
-		<Unit filename="base/tinyxml/tinywxuni.cpp">
+		<Unit filename="sdk/tinywxuni.cpp">
 			<Option target="sdk" />
 		</Unit>
 		<Unit filename="base/tinyxml/tinyxml.cpp">
@@ -896,7 +896,7 @@
 		<Unit filename="include/tinyxml/tinystr.h">
 			<Option target="sdk" />
 		</Unit>
-		<Unit filename="include/tinyxml/tinywxuni.h">
+		<Unit filename="include/tinywxuni.h">
 			<Option target="sdk" />
 		</Unit>
 		<Unit filename="include/tinyxml/tinyxml.h">
diff --git a/src/CodeBlocks.cbp b/src/CodeBlocks.cbp
index 47a2196..c35e699 100644
--- a/src/CodeBlocks.cbp
+++ b/src/CodeBlocks.cbp
@@ -505,7 +505,7 @@
 		<Unit filename="base\tinyxml\tinystr.cpp">
 			<Option target="tinyXML" />
 		</Unit>
-		<Unit filename="base\tinyxml\tinywxuni.cpp">
+		<Unit filename="sdk\tinywxuni.cpp">
 			<Option target="sdk" />
 		</Unit>
 		<Unit filename="base\tinyxml\tinyxml.cpp">
@@ -998,7 +998,7 @@
 		<Unit filename="include\tinyxml\tinystr.h">
 			<Option target="tinyXML" />
 		</Unit>
-		<Unit filename="include\tinyxml\tinywxuni.h">
+		<Unit filename="include\tinywxuni.h">
 			<Option target="sdk" />
 		</Unit>
 		<Unit filename="include\tinyxml\tinyxml.h">
-- 
1.6.6.1

From a2ee4412e4dbb135841fe9cf16e1272e3318bd3f Mon Sep 17 00:00:00 2001
From: Dan Horak <dan@danny.cz>
Date: Thu, 14 May 2009 10:05:42 +0200
Subject: [PATCH 05/22] convert autorevision to new tinyxml handling

---
 src/build_tools/autorevision/Makefile.am      |    5 ++++-
 src/build_tools/autorevision/autorevision.cpp |    3 +--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/build_tools/autorevision/Makefile.am b/src/build_tools/autorevision/Makefile.am
index f0b09b7..7de8ca4 100644
--- a/src/build_tools/autorevision/Makefile.am
+++ b/src/build_tools/autorevision/Makefile.am
@@ -1,9 +1,12 @@
 noinst_PROGRAMS = auto_revision
 auto_revision_SOURCES = autorevision.cpp
 
+if ! HAVE_TINYXML
 INCLUDES = -I$(top_srcdir)/src/include
-
 auto_revision_LDADD = ../../base/tinyxml/libtinyxml.la
+else
+auto_revision_LDADD = -ltinyxml
+endif
 
 all-local: auto_revision
 	./auto_revision +int +wx +t $(top_srcdir) $(top_srcdir)/src/include/autorevision.h
diff --git a/src/build_tools/autorevision/autorevision.cpp b/src/build_tools/autorevision/autorevision.cpp
index bdc0c88..ea6388b 100644
--- a/src/build_tools/autorevision/autorevision.cpp
+++ b/src/build_tools/autorevision/autorevision.cpp
@@ -11,8 +11,7 @@
 #include <string>
 #include <fstream>
 
-#include "tinyxml/tinystr.h"
-#include "tinyxml/tinyxml.h"
+#include <tinyxml.h>
 
 using namespace std;
 
-- 
1.6.6.1

From b370d63c271cb1ef8eac45a561ce21b7a7733268 Mon Sep 17 00:00:00 2001
From: Dan Horak <dan@danny.cz>
Date: Thu, 14 May 2009 17:33:02 +0200
Subject: [PATCH 06/22] convert sdk to new tinyxml handling

---
 src/include/configmanager.h         |    3 +--
 src/include/ibaseloader.h           |    2 +-
 src/include/projecttemplateloader.h |    2 +-
 src/sdk/Makefile.am                 |    9 +++++++--
 src/sdk/globals.cpp                 |    2 +-
 src/sdk/pluginmanager.cpp           |    2 +-
 src/sdk/projectlayoutloader.cpp     |    2 +-
 src/sdk/tinywxuni.cpp               |    2 +-
 src/sdk/workspaceloader.cpp         |    2 +-
 9 files changed, 15 insertions(+), 11 deletions(-)

diff --git a/src/include/configmanager.h b/src/include/configmanager.h
index 951dd5d..9d9de79 100644
--- a/src/include/configmanager.h
+++ b/src/include/configmanager.h
@@ -12,8 +12,7 @@
 #include "settings.h"
 #include "globals.h"
 
-#include "tinyxml/tinystr.h"
-#include "tinyxml/tinyxml.h"
+#include <tinyxml.h>
 #include "manager.h"
 #include "base64.h"
 
diff --git a/src/include/ibaseloader.h b/src/include/ibaseloader.h
index 95b3892..c0b98fb 100644
--- a/src/include/ibaseloader.h
+++ b/src/include/ibaseloader.h
@@ -8,7 +8,7 @@
 
 #include <wx/string.h>
 #include "settings.h"
-#include "tinyxml/tinyxml.h"
+#include <tinyxml.h>
 
 /** Base interface for project loaders. */
 class DLLIMPORT IBaseLoader
diff --git a/src/include/projecttemplateloader.h b/src/include/projecttemplateloader.h
index 82b5c68..9add87c 100644
--- a/src/include/projecttemplateloader.h
+++ b/src/include/projecttemplateloader.h
@@ -6,7 +6,7 @@
 #ifndef PROJECTTEMPLATELOADER_H
 #define PROJECTTEMPLATELOADER_H
 
-#include "tinyxml/tinyxml.h"
+#include <tinyxml.h>
 #include "settings.h"
 
 struct FileSetFile
diff --git a/src/sdk/Makefile.am b/src/sdk/Makefile.am
index 66a2620..29f99d6 100644
--- a/src/sdk/Makefile.am
+++ b/src/sdk/Makefile.am
@@ -3,7 +3,6 @@ SUBDIRS = scripting wxscintilla resources
 INCLUDES = $(WX_CXXFLAGS) \
 			-I$(top_srcdir)/src/include \
 			-I$(top_srcdir)/src/sdk/wxscintilla/include \
-			-I$(top_srcdir)/src/include/tinyxml \
 			-I$(top_srcdir)/src/include/scripting/include \
  			-I$(top_srcdir)/src/include/scripting/sqplus \
 			-I$(top_srcdir)/src/include/mozilla_chardet
@@ -17,9 +16,15 @@ libcodeblocks_la_LIBADD =  $(WX_LIBS) \
 			scripting/sqplus/libsqplus.la \
 			scripting/sqstdlib/libsqstdlib.la \
 			scripting/squirrel/libsquirrel.la \
-			../base/tinyxml/libtinyxml.la \
 			wxscintilla/libwxscintilla.la
 
+if ! HAVE_TINYXML
+INCLUDES +=		-I$(top_srcdir)/src/include/tinyxml
+libcodeblocks_la_LIBADD += ../base/tinyxml/libtinyxml.la
+else
+libcodeblocks_la_LIBADD += -ltinyxml
+endif
+
 libcodeblocks_la_SOURCES = annoyingdialog.cpp \
 						autodetectcompilers.cpp \
 						base64.cpp \
diff --git a/src/sdk/globals.cpp b/src/sdk/globals.cpp
index cceb1e9..bb66a59 100644
--- a/src/sdk/globals.cpp
+++ b/src/sdk/globals.cpp
@@ -23,7 +23,7 @@
     #include "projectmanager.h"
 #endif
 
-#include "tinyxml/tinyxml.h"
+#include <tinyxml.h>
 #include <wx/filefn.h>
 #include <wx/tokenzr.h>
 #include <wx/dirdlg.h>
diff --git a/src/sdk/pluginmanager.cpp b/src/sdk/pluginmanager.cpp
index d6c9344..90cc953 100644
--- a/src/sdk/pluginmanager.cpp
+++ b/src/sdk/pluginmanager.cpp
@@ -42,7 +42,7 @@
 #include <wx/txtstrm.h>
 
 #include "filefilters.h"
-#include "tinyxml/tinyxml.h"
+#include <tinyxml.h>
 
 #include "annoyingdialog.h"
 #include "pluginsconfigurationdlg.h"
diff --git a/src/sdk/projectlayoutloader.cpp b/src/sdk/projectlayoutloader.cpp
index 472b053..07cd159 100644
--- a/src/sdk/projectlayoutloader.cpp
+++ b/src/sdk/projectlayoutloader.cpp
@@ -22,7 +22,7 @@
 #endif
 
 #include "projectlayoutloader.h"
-#include "tinyxml/tinyxml.h"
+#include <tinyxml.h>
 #include "tinywxuni.h"
 
 ProjectLayoutLoader::ProjectLayoutLoader(cbProject* project)
diff --git a/src/sdk/tinywxuni.cpp b/src/sdk/tinywxuni.cpp
index 5cfc5ee..1e33cd6 100644
--- a/src/sdk/tinywxuni.cpp
+++ b/src/sdk/tinywxuni.cpp
@@ -7,7 +7,7 @@
 #include "manager.h"
 #endif
 #include "tinywxuni.h"
-#include "tinyxml.h"
+#include <tinyxml.h>
 
 bool TinyXML::LoadDocument(const wxString& filename, TiXmlDocument *doc)
 {
diff --git a/src/sdk/workspaceloader.cpp b/src/sdk/workspaceloader.cpp
index f2624be..f502519 100644
--- a/src/sdk/workspaceloader.cpp
+++ b/src/sdk/workspaceloader.cpp
@@ -25,7 +25,7 @@
 
 
 
-#include "tinyxml/tinyxml.h"
+#include <tinyxml.h>
 #include "tinywxuni.h"
 
 WorkspaceLoader::WorkspaceLoader()
-- 
1.6.6.1

From 99f0de7b8d7ff02020ab019df365ca2681c27f2e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Sat, 16 May 2009 12:36:33 +0200
Subject: [PATCH 07/22] update codecompletion plugin

---
 src/plugins/codecompletion/Makefile.am      |    4 ++++
 src/plugins/codecompletion/nativeparser.cpp |    2 +-
 2 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/src/plugins/codecompletion/Makefile.am b/src/plugins/codecompletion/Makefile.am
index 842148d..02dbbd8 100644
--- a/src/plugins/codecompletion/Makefile.am
+++ b/src/plugins/codecompletion/Makefile.am
@@ -4,6 +4,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libcodecompletion.la
diff --git a/src/plugins/codecompletion/nativeparser.cpp b/src/plugins/codecompletion/nativeparser.cpp
index a1c5ecc..174898e 100644
--- a/src/plugins/codecompletion/nativeparser.cpp
+++ b/src/plugins/codecompletion/nativeparser.cpp
@@ -26,7 +26,7 @@
 #include <cbstyledtextctrl.h>
 #include <compilerfactory.h>
 #include <projectloader_hooks.h>
-#include <tinyxml/tinyxml.h>
+#include <tinyxml.h>
 
 #include "cbauibook.h"
 #include <wx/log.h> // for wxSafeShowMessage()
-- 
1.6.6.1

From a944d02321b7ad449580947e8257fb5fc1fcafc2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Sat, 16 May 2009 12:38:01 +0200
Subject: [PATCH 08/22] update codesnippets plugin

---
 src/plugins/contrib/codesnippets/Makefile.am       |    4 ++++
 .../contrib/codesnippets/codesnippetstreectrl.cpp  |    1 -
 .../contrib/codesnippets/codesnippetstreectrl.h    |    2 +-
 .../contrib/codesnippets/codesnippetswindow.cpp    |    2 +-
 4 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/plugins/contrib/codesnippets/Makefile.am b/src/plugins/contrib/codesnippets/Makefile.am
index 220912a..b44850c 100644
--- a/src/plugins/contrib/codesnippets/Makefile.am
+++ b/src/plugins/contrib/codesnippets/Makefile.am
@@ -9,6 +9,10 @@ INCLUDES = $(WX_CXXFLAGS) \
  		-I$(top_srcdir)/src/plugins/contrib/codesnippets/editor \
 		-I$(top_srcdir)/src/include/mozilla_chardet
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libcodesnippets.la
diff --git a/src/plugins/contrib/codesnippets/codesnippetstreectrl.cpp b/src/plugins/contrib/codesnippets/codesnippetstreectrl.cpp
index c6f8c14..191a4ca 100644
--- a/src/plugins/contrib/codesnippets/codesnippetstreectrl.cpp
+++ b/src/plugins/contrib/codesnippets/codesnippetstreectrl.cpp
@@ -41,7 +41,6 @@
 //-#else
 //-#endif
 
-#include <tinyxml/tinyxml.h>
 #include "snippetitemdata.h"
 #include "codesnippetstreectrl.h"
 #include "codesnippetswindow.h"
diff --git a/src/plugins/contrib/codesnippets/codesnippetstreectrl.h b/src/plugins/contrib/codesnippets/codesnippetstreectrl.h
index 00c3084..9e17819 100644
--- a/src/plugins/contrib/codesnippets/codesnippetstreectrl.h
+++ b/src/plugins/contrib/codesnippets/codesnippetstreectrl.h
@@ -30,7 +30,7 @@ class TiXmlElement;
 #include "wx/mimetype.h"
 
 #include "snippetitemdata.h"
-#include <tinyxml/tinyxml.h>
+#include <tinyxml.h>
 #include "snippetproperty.h"
 #include "codesnippetsevent.h"
 #include "snippetsconfig.h"
diff --git a/src/plugins/contrib/codesnippets/codesnippetswindow.cpp b/src/plugins/contrib/codesnippets/codesnippetswindow.cpp
index 634c7d0..5efeeab 100644
--- a/src/plugins/contrib/codesnippets/codesnippetswindow.cpp
+++ b/src/plugins/contrib/codesnippets/codesnippetswindow.cpp
@@ -63,7 +63,7 @@
 #include "version.h"
 #include "codesnippetswindow.h"
 #include "snippetitemdata.h"
-#include <tinyxml/tinyxml.h>
+#include <tinyxml.h>
 #include "snippetsconfig.h"
 #include "snippetsimages.h"
 #include "codesnippetstreectrl.h"
-- 
1.6.6.1

From e40d8c9f7f166adc2fca6706dac330ad232db260 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Sat, 16 May 2009 12:39:05 +0200
Subject: [PATCH 09/22] update envvars plugin

---
 src/plugins/contrib/envvars/Makefile.am |    4 ++++
 src/plugins/contrib/envvars/envvars.cpp |    2 +-
 2 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/src/plugins/contrib/envvars/Makefile.am b/src/plugins/contrib/envvars/Makefile.am
index 8d38e27..f782f9b 100644
--- a/src/plugins/contrib/envvars/Makefile.am
+++ b/src/plugins/contrib/envvars/Makefile.am
@@ -2,6 +2,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 pkgdata_DATA = envvars.zip
 
diff --git a/src/plugins/contrib/envvars/envvars.cpp b/src/plugins/contrib/envvars/envvars.cpp
index 5a9e0ba..8c3e920 100644
--- a/src/plugins/contrib/envvars/envvars.cpp
+++ b/src/plugins/contrib/envvars/envvars.cpp
@@ -15,7 +15,7 @@
   #include <wx/menu.h>
   #include <wx/toolbar.h>
 
-  #include <tinyxml/tinyxml.h>
+  #include <tinyxml.h>
 
   #include "cbproject.h"
   #include "globals.h"
-- 
1.6.6.1

From 3a1f5a276af4999c028f2daeec6fe2661c3d4326 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Sat, 16 May 2009 12:40:11 +0200
Subject: [PATCH 10/22] update BrowseTracker plugin

---
 .../contrib/BrowseTracker/BrowseTrackerLayout.cpp  |    2 +-
 src/plugins/contrib/BrowseTracker/Makefile.am      |    4 ++++
 2 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/src/plugins/contrib/BrowseTracker/BrowseTrackerLayout.cpp b/src/plugins/contrib/BrowseTracker/BrowseTrackerLayout.cpp
index 30f38d4..189eaf7 100644
--- a/src/plugins/contrib/BrowseTracker/BrowseTrackerLayout.cpp
+++ b/src/plugins/contrib/BrowseTracker/BrowseTrackerLayout.cpp
@@ -46,7 +46,7 @@
 #endif
 
 #include <wx/tokenzr.h>
-#include "tinyxml/tinyxml.h"
+#include <tinyxml.h>
 #include "tinywxuni.h"
 
 #include "BrowseTrackerLayout.h"
diff --git a/src/plugins/contrib/BrowseTracker/Makefile.am b/src/plugins/contrib/BrowseTracker/Makefile.am
index a6a98c9..e8a941e 100644
--- a/src/plugins/contrib/BrowseTracker/Makefile.am
+++ b/src/plugins/contrib/BrowseTracker/Makefile.am
@@ -2,6 +2,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libBrowseTracker.la
-- 
1.6.6.1

From cd4ad531af8b685cf2ad3bafc42eeb2810f40de1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Sat, 16 May 2009 12:41:03 +0200
Subject: [PATCH 11/22] update Valgrind plugin

---
 src/plugins/contrib/Valgrind/Makefile.am  |    4 ++++
 src/plugins/contrib/Valgrind/Valgrind.cpp |    2 +-
 2 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/src/plugins/contrib/Valgrind/Makefile.am b/src/plugins/contrib/Valgrind/Makefile.am
index 69c7e33..db897f5 100644
--- a/src/plugins/contrib/Valgrind/Makefile.am
+++ b/src/plugins/contrib/Valgrind/Makefile.am
@@ -4,6 +4,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libValgrind.la
diff --git a/src/plugins/contrib/Valgrind/Valgrind.cpp b/src/plugins/contrib/Valgrind/Valgrind.cpp
index 4f13b77..f2a5ec3 100644
--- a/src/plugins/contrib/Valgrind/Valgrind.cpp
+++ b/src/plugins/contrib/Valgrind/Valgrind.cpp
@@ -21,7 +21,7 @@
 #include "projectmanager.h"
 #endif
 #include <wx/filefn.h>
-#include "tinyxml/tinyxml.h"
+#include <tinyxml.h>
 #include "loggers.h"
 #include "Valgrind.h"
 #include "ValgrindListLog.h"
-- 
1.6.6.1

From b2736c42db392d3fd2b2259bfa5b8644bcd8b93f Mon Sep 17 00:00:00 2001
From: Dan Horak <dan@danny.cz>
Date: Sat, 16 May 2009 12:04:35 +0200
Subject: [PATCH 12/22] convert AutoVersioning plugin

---
 .../contrib/AutoVersioning/AutoVersioning.cpp      |    2 +-
 src/plugins/contrib/AutoVersioning/Makefile.am     |    4 ++++
 .../contrib/AutoVersioning/avSvnRevision.cpp       |    2 +-
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/plugins/contrib/AutoVersioning/AutoVersioning.cpp b/src/plugins/contrib/AutoVersioning/AutoVersioning.cpp
index 09db296..2a18834 100644
--- a/src/plugins/contrib/AutoVersioning/AutoVersioning.cpp
+++ b/src/plugins/contrib/AutoVersioning/AutoVersioning.cpp
@@ -17,7 +17,7 @@
 #include <manager.h>
 #include <projectbuildtarget.h>
 #include <projectmanager.h>
-#include <tinyxml/tinyxml.h>
+#include <tinyxml.h>
 #endif
 
 #include "projectloader_hooks.h"
diff --git a/src/plugins/contrib/AutoVersioning/Makefile.am b/src/plugins/contrib/AutoVersioning/Makefile.am
index 36a8072..eb85fd6 100644
--- a/src/plugins/contrib/AutoVersioning/Makefile.am
+++ b/src/plugins/contrib/AutoVersioning/Makefile.am
@@ -2,6 +2,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libAutoVersioning.la
diff --git a/src/plugins/contrib/AutoVersioning/avSvnRevision.cpp b/src/plugins/contrib/AutoVersioning/avSvnRevision.cpp
index 17335ad..f549150 100644
--- a/src/plugins/contrib/AutoVersioning/avSvnRevision.cpp
+++ b/src/plugins/contrib/AutoVersioning/avSvnRevision.cpp
@@ -3,7 +3,7 @@
 #include <wx/utils.h>
 #include <wx/string.h>
 #include <wx/arrstr.h>
-#include <tinyxml/tinyxml.h>
+#include <tinyxml.h>
 #include <globals.h>
 
 bool QuerySvn(const wxString& workingDir, wxString& revision, wxString& date)
-- 
1.6.6.1

From 1fcda0ed358cf7e25cd0f0bc9668fb1fabec8d52 Mon Sep 17 00:00:00 2001
From: Dan Horak <dan@danny.cz>
Date: Sat, 16 May 2009 12:10:45 +0200
Subject: [PATCH 13/22] convert lib_finder plugin

---
 src/plugins/contrib/lib_finder/Makefile.am         |    4 ++++
 src/plugins/contrib/lib_finder/lib_finder.h        |    2 +-
 .../contrib/lib_finder/librarydetectionmanager.cpp |    2 +-
 .../contrib/lib_finder/projectconfiguration.h      |    2 +-
 4 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/plugins/contrib/lib_finder/Makefile.am b/src/plugins/contrib/lib_finder/Makefile.am
index 2448840..2a30765 100644
--- a/src/plugins/contrib/lib_finder/Makefile.am
+++ b/src/plugins/contrib/lib_finder/Makefile.am
@@ -10,6 +10,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/include/scripting/bindings \
 		-I$(top_srcdir)/src/include/mozilla_chardet
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = liblib_finder.la
diff --git a/src/plugins/contrib/lib_finder/lib_finder.h b/src/plugins/contrib/lib_finder/lib_finder.h
index 4ee49e8..8aaa51a 100644
--- a/src/plugins/contrib/lib_finder/lib_finder.h
+++ b/src/plugins/contrib/lib_finder/lib_finder.h
@@ -27,7 +27,7 @@
 #include <cbplugin.h>
 #include <settings.h>
 #include <sdk_events.h>
-#include <tinyxml/tinyxml.h>
+#include <tinyxml.h>
 
 #include "librarydetectionmanager.h"
 #include "resultmap.h"
diff --git a/src/plugins/contrib/lib_finder/librarydetectionmanager.cpp b/src/plugins/contrib/lib_finder/librarydetectionmanager.cpp
index b61adf8..594d960 100644
--- a/src/plugins/contrib/lib_finder/librarydetectionmanager.cpp
+++ b/src/plugins/contrib/lib_finder/librarydetectionmanager.cpp
@@ -21,7 +21,7 @@
 * $HeadURL: svn+ssh://byo@svn.berlios.de/svnroot/repos/codeblocks/trunk/src/plugins/contrib/wxSmith/plugin/wxsmithpluginregistrants.cpp $
 */
 
-#include <tinyxml/tinyxml.h>
+#include <tinyxml.h>
 #include <tinywxuni.h>
 
 #include <wx/arrstr.h>
diff --git a/src/plugins/contrib/lib_finder/projectconfiguration.h b/src/plugins/contrib/lib_finder/projectconfiguration.h
index 3172492..2eaf313 100644
--- a/src/plugins/contrib/lib_finder/projectconfiguration.h
+++ b/src/plugins/contrib/lib_finder/projectconfiguration.h
@@ -27,7 +27,7 @@
 #include <wx/arrstr.h>
 #include <wx/string.h>
 #include <wx/hashmap.h>
-#include <tinyxml/tinyxml.h>
+#include <tinyxml.h>
 #include <cbproject.h>
 
 /** \brief Configuration of one project */
-- 
1.6.6.1

From ca8c55a67091aaa83c761077f746505c2b6ec1b1 Mon Sep 17 00:00:00 2001
From: Dan Horak <dan@danny.cz>
Date: Sat, 16 May 2009 12:20:55 +0200
Subject: [PATCH 14/22] convert wxSmith plugin

---
 src/plugins/contrib/wxSmith/Makefile.am            |    4 ++++
 src/plugins/contrib/wxSmith/properties/Makefile.am |    4 ++++
 .../contrib/wxSmith/properties/wxsproperty.h       |    2 +-
 src/plugins/contrib/wxSmith/wxsgui.h               |    2 +-
 src/plugins/contrib/wxSmith/wxsmith.h              |    2 +-
 src/plugins/contrib/wxSmith/wxsproject.h           |    2 +-
 src/plugins/contrib/wxSmith/wxsversionconverter.h  |    2 +-
 src/plugins/contrib/wxSmith/wxwidgets/Makefile.am  |    4 ++++
 src/plugins/contrib/wxSmith/wxwidgets/wxsevents.h  |    2 +-
 .../wxSmith/wxwidgets/wxsitemresdataobject.h       |    2 +-
 10 files changed, 19 insertions(+), 7 deletions(-)

diff --git a/src/plugins/contrib/wxSmith/Makefile.am b/src/plugins/contrib/wxSmith/Makefile.am
index e1abc9f..c4634c3 100644
--- a/src/plugins/contrib/wxSmith/Makefile.am
+++ b/src/plugins/contrib/wxSmith/Makefile.am
@@ -12,6 +12,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/plugins/contrib/wxSmith/propgrid/contrib/include \
 		-I$(top_srcdir)/src/include/mozilla_chardet
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 lib_LTLIBRARIES = libwxsmithlib.la
 
 libwxsmithlib_la_LDFLAGS = -version-info 0:1:0 -shared
diff --git a/src/plugins/contrib/wxSmith/properties/Makefile.am b/src/plugins/contrib/wxSmith/properties/Makefile.am
index b7fc223..6ef1ed2 100644
--- a/src/plugins/contrib/wxSmith/properties/Makefile.am
+++ b/src/plugins/contrib/wxSmith/properties/Makefile.am
@@ -5,6 +5,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include \
 		-I$(top_srcdir)/src/plugins/contrib/wxSmith/propgrid/contrib/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 noinst_LTLIBRARIES = libwxsmith_properties.la
 
 libwxsmith_properties_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
diff --git a/src/plugins/contrib/wxSmith/properties/wxsproperty.h b/src/plugins/contrib/wxSmith/properties/wxsproperty.h
index 1e2d321..de36cee 100644
--- a/src/plugins/contrib/wxSmith/properties/wxsproperty.h
+++ b/src/plugins/contrib/wxSmith/properties/wxsproperty.h
@@ -34,7 +34,7 @@
 #include <wx/propgrid/propgrid.h>
 #include <wx/propgrid/manager.h>
 
-#include <tinyxml/tinyxml.h>
+#include <tinyxml.h>
 #include "wxspropertystream.h"
 
 class wxsPropertyContainer;
diff --git a/src/plugins/contrib/wxSmith/wxsgui.h b/src/plugins/contrib/wxSmith/wxsgui.h
index 7694750..48a06e3 100644
--- a/src/plugins/contrib/wxSmith/wxsgui.h
+++ b/src/plugins/contrib/wxSmith/wxsgui.h
@@ -24,7 +24,7 @@
 #define WXSGUI_H
 
 #include <configurationpanel.h>
-#include <tinyxml/tinyxml.h>
+#include <tinyxml.h>
 #include "wxsguifactory.h"
 #include "wxsproject.h"
 
diff --git a/src/plugins/contrib/wxSmith/wxsmith.h b/src/plugins/contrib/wxSmith/wxsmith.h
index ff311c9..2a442b5 100644
--- a/src/plugins/contrib/wxSmith/wxsmith.h
+++ b/src/plugins/contrib/wxSmith/wxsmith.h
@@ -29,7 +29,7 @@
 #include <cbplugin.h>
 #include <settings.h>
 #include <sdk_events.h>
-#include <tinyxml/tinyxml.h>
+#include <tinyxml.h>
 
 #include "wxsstoringsplitterwindow.h"
 #include "wxsresourcetree.h"
diff --git a/src/plugins/contrib/wxSmith/wxsproject.h b/src/plugins/contrib/wxSmith/wxsproject.h
index 4ffcab1..de0336a 100644
--- a/src/plugins/contrib/wxSmith/wxsproject.h
+++ b/src/plugins/contrib/wxSmith/wxsproject.h
@@ -24,7 +24,7 @@
 #define WXSPROJECT_H
 
 #include <cbproject.h>
-#include <tinyxml/tinyxml.h>
+#include <tinyxml.h>
 #include "wxsresourcetree.h"
 
 class wxSmith;
diff --git a/src/plugins/contrib/wxSmith/wxsversionconverter.h b/src/plugins/contrib/wxSmith/wxsversionconverter.h
index ac2b79c..0673bb5 100644
--- a/src/plugins/contrib/wxSmith/wxsversionconverter.h
+++ b/src/plugins/contrib/wxSmith/wxsversionconverter.h
@@ -23,7 +23,7 @@
 #ifndef WXSVERSIONCONVERTER_H
 #define WXSVERSIONCONVERTER_H
 
-#include <tinyxml/tinyxml.h>
+#include <tinyxml.h>
 #include <wx/string.h>
 
 class wxsProject;
diff --git a/src/plugins/contrib/wxSmith/wxwidgets/Makefile.am b/src/plugins/contrib/wxSmith/wxwidgets/Makefile.am
index 7238394..7a37296 100644
--- a/src/plugins/contrib/wxSmith/wxwidgets/Makefile.am
+++ b/src/plugins/contrib/wxSmith/wxwidgets/Makefile.am
@@ -11,6 +11,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/include/scripting/bindings \
 		-I$(top_srcdir)/src/plugins/contrib/wxSmith/propgrid/contrib/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 noinst_LTLIBRARIES = libwxsmith_wxwidgets.la
 
 libwxsmith_wxwidgets_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
diff --git a/src/plugins/contrib/wxSmith/wxwidgets/wxsevents.h b/src/plugins/contrib/wxSmith/wxwidgets/wxsevents.h
index 4e5d044..18571a4 100644
--- a/src/plugins/contrib/wxSmith/wxwidgets/wxsevents.h
+++ b/src/plugins/contrib/wxSmith/wxwidgets/wxsevents.h
@@ -24,7 +24,7 @@
 #define WXSEVENTS_H
 
 #include <wx/arrstr.h>
-#include <tinyxml/tinyxml.h>
+#include <tinyxml.h>
 #include <wx/arrstr.h>
 
 #include "wxscodercontext.h"
diff --git a/src/plugins/contrib/wxSmith/wxwidgets/wxsitemresdataobject.h b/src/plugins/contrib/wxSmith/wxwidgets/wxsitemresdataobject.h
index 142d6dd..e17f529 100644
--- a/src/plugins/contrib/wxSmith/wxwidgets/wxsitemresdataobject.h
+++ b/src/plugins/contrib/wxSmith/wxwidgets/wxsitemresdataobject.h
@@ -24,7 +24,7 @@
 #define WXSITEMRESDATAOBJECT_H
 
 #include <wx/dataobj.h>
-#include <tinyxml/tinyxml.h>
+#include <tinyxml.h>
 
 #define wxsDF_WIDGET   _T("wxSmith XML")
 
-- 
1.6.6.1

From 3bcf565a2da2c0133612c4865d48cbad4b92d5cc Mon Sep 17 00:00:00 2001
From: Dan Horak <dan@danny.cz>
Date: Sat, 16 May 2009 12:23:54 +0200
Subject: [PATCH 15/22] convert cb_share_config tool

---
 src/tools/cb_share_config/Makefile.am |    9 +++++++--
 src/tools/cb_share_config/mainframe.h |    3 +--
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/src/tools/cb_share_config/Makefile.am b/src/tools/cb_share_config/Makefile.am
index fa88563..3eb9e5c 100644
--- a/src/tools/cb_share_config/Makefile.am
+++ b/src/tools/cb_share_config/Makefile.am
@@ -1,13 +1,18 @@
 bin_PROGRAMS = cb_share_config
 
 INCLUDES = $(WX_CXXFLAGS) \
-			-I$(top_srcdir)/src/include \
-			-I$(top_srcdir)/src/include/tinyxml
+			-I$(top_srcdir)/src/include
 
 cb_share_config_LDFLAGS = 
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
 cb_share_config_LDADD =  -L../../base/tinyxml -ltinyxml \
 			$(WX_LIBS)
+else
+cb_share_config_LDADD = -ltinyxml $(WX_LIBS)
+endif
+
 
 cb_share_config_SOURCES = app.cpp \
 			mainframe.cpp
diff --git a/src/tools/cb_share_config/mainframe.h b/src/tools/cb_share_config/mainframe.h
index c49e790..b658bad 100644
--- a/src/tools/cb_share_config/mainframe.h
+++ b/src/tools/cb_share_config/mainframe.h
@@ -20,8 +20,7 @@
 #include <wx/arrstr.h>
 #include <vector>
 
-#include "tinyxml/tinystr.h"
-#include "tinyxml/tinyxml.h"
+#include <tinyxml.h>
 
 class MainFrame: public wxFrame
 {
-- 
1.6.6.1

From 3557cbc3393d2cce972ce7fdc29d960cc018641b Mon Sep 17 00:00:00 2001
From: Dan Horak <dan@danny.cz>
Date: Sat, 16 May 2009 12:26:31 +0200
Subject: [PATCH 16/22] build local copy of tinyxml library only when no system tinyxml library exists

---
 src/base/Makefile.am |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/base/Makefile.am b/src/base/Makefile.am
index 62a6da1..8f8347f 100644
--- a/src/base/Makefile.am
+++ b/src/base/Makefile.am
@@ -1 +1,3 @@
+if ! HAVE_TINYXML
 SUBDIRS = tinyxml
+endif
-- 
1.6.6.1

From 969c42a03f0c76807098785c919617850bbc0667 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Sat, 16 May 2009 13:07:12 +0200
Subject: [PATCH 17/22] fix include dirs for the main app when system tinyxml is not present

---
 src/plugins/autosave/Makefile.am                   |    4 ++++
 src/plugins/contrib/HexEditor/Makefile.am          |    4 ++++
 src/plugins/contrib/ThreadSearch/Makefile.am       |    4 ++++
 src/plugins/contrib/byogames/Makefile.am           |    4 ++++
 .../contrib/codesnippets/resources/Makefile.am     |    4 ++++
 src/plugins/contrib/dragscroll/Makefile.am         |    4 ++++
 src/plugins/contrib/headerfixup/Makefile.am        |    4 ++++
 src/plugins/contrib/keybinder/Makefile.am          |    4 ++++
 src/plugins/contrib/wxSmith/plugin/Makefile.am     |    4 ++++
 .../contrib/wxSmith/wxwidgets/defitems/Makefile.am |    4 ++++
 .../wxSmith/wxwidgets/properties/Makefile.am       |    4 ++++
 .../contrib/wxSmithContribItems/Makefile.am        |    4 ++++
 src/plugins/openfileslist/Makefile.am              |    4 ++++
 src/plugins/projectsimporter/Makefile.am           |    3 +++
 src/plugins/scriptedwizard/Makefile.am             |    4 ++++
 src/src/Makefile.am                                |    4 ++++
 16 files changed, 63 insertions(+), 0 deletions(-)

diff --git a/src/plugins/autosave/Makefile.am b/src/plugins/autosave/Makefile.am
index 2ee7459..d184342 100644
--- a/src/plugins/autosave/Makefile.am
+++ b/src/plugins/autosave/Makefile.am
@@ -2,6 +2,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libautosave.la
diff --git a/src/plugins/contrib/HexEditor/Makefile.am b/src/plugins/contrib/HexEditor/Makefile.am
index 1d30d9b..a2e2ccc 100644
--- a/src/plugins/contrib/HexEditor/Makefile.am
+++ b/src/plugins/contrib/HexEditor/Makefile.am
@@ -2,6 +2,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libHexEditor.la
diff --git a/src/plugins/contrib/ThreadSearch/Makefile.am b/src/plugins/contrib/ThreadSearch/Makefile.am
index 68b9a25..843a6a6 100644
--- a/src/plugins/contrib/ThreadSearch/Makefile.am
+++ b/src/plugins/contrib/ThreadSearch/Makefile.am
@@ -6,6 +6,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/include/mozilla_chardet \
 		-I$(top_srcdir)/src/plugins/contrib/wxSmithContribItems/wxthings/wxthings/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libThreadSearch.la
diff --git a/src/plugins/contrib/byogames/Makefile.am b/src/plugins/contrib/byogames/Makefile.am
index d66aa15..e329f54 100644
--- a/src/plugins/contrib/byogames/Makefile.am
+++ b/src/plugins/contrib/byogames/Makefile.am
@@ -2,6 +2,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libbyogames.la
diff --git a/src/plugins/contrib/codesnippets/resources/Makefile.am b/src/plugins/contrib/codesnippets/resources/Makefile.am
index a83a6bb..dbbb4b3 100644
--- a/src/plugins/contrib/codesnippets/resources/Makefile.am
+++ b/src/plugins/contrib/codesnippets/resources/Makefile.am
@@ -7,6 +7,10 @@ INCLUDES = $(WX_CXXFLAGS) \
  		-I$(top_srcdir)/src/plugins/contrib/codesnippets/editor \
 		-I$(top_srcdir)/src/include/mozilla_chardet
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 ##pluginlib_LTLIBRARIES = codesnippets
diff --git a/src/plugins/contrib/dragscroll/Makefile.am b/src/plugins/contrib/dragscroll/Makefile.am
index a6520fb..c4aa031 100644
--- a/src/plugins/contrib/dragscroll/Makefile.am
+++ b/src/plugins/contrib/dragscroll/Makefile.am
@@ -3,6 +3,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/src \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libdragscroll.la
diff --git a/src/plugins/contrib/headerfixup/Makefile.am b/src/plugins/contrib/headerfixup/Makefile.am
index 2983fcd..503c3d7 100644
--- a/src/plugins/contrib/headerfixup/Makefile.am
+++ b/src/plugins/contrib/headerfixup/Makefile.am
@@ -2,6 +2,10 @@ INCLUDES = $(WX_CXXFLAGS) \
            -I$(top_srcdir)/src/include \
            -I$(top_srcdir)/src/sdk/wxscintilla/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 CXXFLAGS = @CXXFLAGS@  -O0
 
 pluginlibdir = $(pkglibdir)/plugins
diff --git a/src/plugins/contrib/keybinder/Makefile.am b/src/plugins/contrib/keybinder/Makefile.am
index d9116af..5da5553 100644
--- a/src/plugins/contrib/keybinder/Makefile.am
+++ b/src/plugins/contrib/keybinder/Makefile.am
@@ -2,6 +2,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libkeybinder.la
diff --git a/src/plugins/contrib/wxSmith/plugin/Makefile.am b/src/plugins/contrib/wxSmith/plugin/Makefile.am
index cf27c1a..ec395b7 100644
--- a/src/plugins/contrib/wxSmith/plugin/Makefile.am
+++ b/src/plugins/contrib/wxSmith/plugin/Makefile.am
@@ -5,6 +5,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include \
 		-I$(top_srcdir)/src/plugins/contrib/propgrid/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlib_LTLIBRARIES = libwxsmith.la
 
 pluginlibdir = $(pkglibdir)/plugins
diff --git a/src/plugins/contrib/wxSmith/wxwidgets/defitems/Makefile.am b/src/plugins/contrib/wxSmith/wxwidgets/defitems/Makefile.am
index d340b90..4289516 100644
--- a/src/plugins/contrib/wxSmith/wxwidgets/defitems/Makefile.am
+++ b/src/plugins/contrib/wxSmith/wxwidgets/defitems/Makefile.am
@@ -5,6 +5,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include \
 		-I$(top_srcdir)/src/plugins/contrib/wxSmith/propgrid/contrib/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 noinst_LTLIBRARIES = libwxsmith_wxwidgets_defitems.la
 
 libwxsmith_wxwidgets_defitems_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
diff --git a/src/plugins/contrib/wxSmith/wxwidgets/properties/Makefile.am b/src/plugins/contrib/wxSmith/wxwidgets/properties/Makefile.am
index 3b157f0..7950a67 100644
--- a/src/plugins/contrib/wxSmith/wxwidgets/properties/Makefile.am
+++ b/src/plugins/contrib/wxSmith/wxwidgets/properties/Makefile.am
@@ -5,6 +5,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include \
 		-I$(top_srcdir)/src/plugins/contrib/wxSmith/propgrid/contrib/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 noinst_LTLIBRARIES = libwxsmith_wxwidgets_properties.la
 
 libwxsmith_wxwidgets_properties_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
diff --git a/src/plugins/contrib/wxSmithContribItems/Makefile.am b/src/plugins/contrib/wxSmithContribItems/Makefile.am
index 3d71f45..04c8b26 100644
--- a/src/plugins/contrib/wxSmithContribItems/Makefile.am
+++ b/src/plugins/contrib/wxSmithContribItems/Makefile.am
@@ -11,6 +11,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(srcdir)/wxflatnotebook/wxFlatNotebook/include \
 		-I$(srcdir)/wxthings/wxthings/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libwxsmithcontribitems.la
diff --git a/src/plugins/openfileslist/Makefile.am b/src/plugins/openfileslist/Makefile.am
index 1b9f299..f9a737a 100644
--- a/src/plugins/openfileslist/Makefile.am
+++ b/src/plugins/openfileslist/Makefile.am
@@ -2,6 +2,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libopenfileslist.la
diff --git a/src/plugins/projectsimporter/Makefile.am b/src/plugins/projectsimporter/Makefile.am
index 318c823..ee3ead4 100644
--- a/src/plugins/projectsimporter/Makefile.am
+++ b/src/plugins/projectsimporter/Makefile.am
@@ -5,6 +5,9 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include \
 		-I$(top_srcdir)/src/include/mozilla_chardet
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
 
 pluginlibdir = $(pkglibdir)/plugins
 
diff --git a/src/plugins/scriptedwizard/Makefile.am b/src/plugins/scriptedwizard/Makefile.am
index eabbcd9..5ef7f7a 100644
--- a/src/plugins/scriptedwizard/Makefile.am
+++ b/src/plugins/scriptedwizard/Makefile.am
@@ -6,6 +6,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/include/scripting/include \
 		-I$(top_srcdir)/src/include/scripting/sqplus
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libscriptedwizard.la
diff --git a/src/src/Makefile.am b/src/src/Makefile.am
index 74f03f3..5a838ec 100644
--- a/src/src/Makefile.am
+++ b/src/src/Makefile.am
@@ -7,6 +7,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 			-I$(top_srcdir)/src/include/scripting/sqplus \
 			-I$(top_srcdir)/src/include/mozilla_chardet
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 if CODEBLOCKS_LINUX
 AM_CPPFLAGS = -DENABLE_BINRELOC -DAPP_PREFIX="\"@prefix@\""
 endif
-- 
1.6.6.1

From 5f83ee6539b9dbd67742da5e92362d37afbb099c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Tue, 18 Aug 2009 13:21:38 +0200
Subject: [PATCH 18/22] use/install local timyxml headers only when a system-wide instance doesn't exist

---
 src/include/Makefile.am |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/include/Makefile.am b/src/include/Makefile.am
index 9f754da..63add9a 100644
--- a/src/include/Makefile.am
+++ b/src/include/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = scripting tinyxml
+SUBDIRS = scripting
 
 if PRECOMPILE_HEADERS
 BUILT_SOURCES = $(top_srcdir)/src/include/sdk.h.gch \
@@ -10,11 +10,15 @@ endif
 INCLUDES = $(WX_CXXFLAGS) \
 			-I$(top_srcdir)/src/include \
 			-I$(top_srcdir)/src/sdk/wxscintilla/include \
-			-I$(top_srcdir)/src/include/tinyxml \
 			-I$(top_srcdir)/src/include/scripting/include \
  			-I$(top_srcdir)/src/include/scripting/sqplus \
 			-I$(top_srcdir)/src/include/mozilla_chardet
 
+if ! HAVE_TINYXML
+SUBDIRS += tinyxml
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 # Because we are not generating any programs/libraries with this makefile,
 # $(CXXCOMPILE) is empty.
 # Create the basic one here:
-- 
1.6.6.1

From c231728469c34e3de91ad98555cc4ad88546983b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Tue, 18 Aug 2009 13:56:15 +0200
Subject: [PATCH 19/22] fix linking with tinyxml in the codesnippets plugin

---
 .../contrib/codesnippets/resources/Makefile.am     |   11 ++++++-----
 1 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/plugins/contrib/codesnippets/resources/Makefile.am b/src/plugins/contrib/codesnippets/resources/Makefile.am
index dbbb4b3..0fded23 100644
--- a/src/plugins/contrib/codesnippets/resources/Makefile.am
+++ b/src/plugins/contrib/codesnippets/resources/Makefile.am
@@ -7,10 +7,6 @@ INCLUDES = $(WX_CXXFLAGS) \
  		-I$(top_srcdir)/src/plugins/contrib/codesnippets/editor \
 		-I$(top_srcdir)/src/include/mozilla_chardet
 
-if ! HAVE_TINYXML
-INCLUDES += -I$(top_srcdir)/src/include/tinyxml
-endif
-
 pluginlibdir = $(pkglibdir)/plugins
 
 ##pluginlib_LTLIBRARIES = codesnippets
@@ -20,10 +16,15 @@ codesnippets_LDFLAGS = $(WX_GTK2_LIBS)
 
 ##codesnippets_LIBADD = $(PLUGIN_WX_LIBS) ../../../../sdk/libcodeblocks.la
 ##codesnippets_LDADD = $(PLUGIN_WX_LIBS) ../../../../sdk/libcodeblocks.la
-codesnippets_LDADD =  -L../../../../base/tinyxml -ltinyxml \
+codesnippets_LDADD = -ltinyxml \
             -L../../../../sdk ../../../../sdk/libcodeblocks.la \
 			$(PLUGIN_WX_LIBS) $(WX_LIBS)
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+codesnippets_LDADD += -L../../../../base/tinyxml
+endif
+
 codesnippets_SOURCES =	$(srcdir)/../version.cpp \
 		$(srcdir)/../Search/DirectoryParamsPanel.cpp \
 		$(srcdir)/../Search/InsertIndexManager.cpp \
-- 
1.6.6.1

From 6b92be5a3a50f7b3182b18d319c7e83f45cddb8d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Mon, 15 Feb 2010 21:34:45 +0100
Subject: [PATCH 20/22] fix tinyxml includes in the help plugin

---
 src/plugins/contrib/help_plugin/Makefile.am |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/plugins/contrib/help_plugin/Makefile.am b/src/plugins/contrib/help_plugin/Makefile.am
index 7b3dd26..219437b 100644
--- a/src/plugins/contrib/help_plugin/Makefile.am
+++ b/src/plugins/contrib/help_plugin/Makefile.am
@@ -40,6 +40,10 @@ else
 libhelp_plugin_la_LIBADD += -lz
 endif
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 libhelp_plugin_la_SOURCES = help_common.cpp \
 			HelpConfigDialog.cpp \
 			help_plugin.cpp \
-- 
1.6.6.1

From 59462f3d197a498cee9dc3d153a9211f78d19197 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Mon, 15 Feb 2010 21:37:48 +0100
Subject: [PATCH 21/22] fix tinyxml includes in the wxSmithAui plugin

---
 src/plugins/contrib/wxSmithAui/Makefile.am |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/plugins/contrib/wxSmithAui/Makefile.am b/src/plugins/contrib/wxSmithAui/Makefile.am
index d56f2d7..808ec2e 100644
--- a/src/plugins/contrib/wxSmithAui/Makefile.am
+++ b/src/plugins/contrib/wxSmithAui/Makefile.am
@@ -9,6 +9,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(srcdir)/wxAuiNotebook \
 		-I$(srcdir)/wxAuiToolBar
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libwxSmithAui.la
-- 
1.6.6.1

From d715e995644be51817787048f6deb7dec50e7bcb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Mon, 15 Feb 2010 21:39:14 +0100
Subject: [PATCH 22/22] fix tinyxml includes in the MouseSap plugin

---
 src/plugins/contrib/MouseSap/Makefile.am |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/plugins/contrib/MouseSap/Makefile.am b/src/plugins/contrib/MouseSap/Makefile.am
index 2c38f84..b185834 100644
--- a/src/plugins/contrib/MouseSap/Makefile.am
+++ b/src/plugins/contrib/MouseSap/Makefile.am
@@ -3,6 +3,10 @@ INCLUDES = $(WX_CXXFLAGS) \
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
+if ! HAVE_TINYXML
+INCLUDES += -I$(top_srcdir)/src/include/tinyxml
+endif
+
 pluginlibdir = $(pkglibdir)/plugins
 
 pluginlib_LTLIBRARIES = libMouseSap.la
-- 
1.6.6.1

