Index: src/sdk/macrosmanager.cpp
===================================================================
--- src/sdk/macrosmanager.cpp	(revision 7665)
+++ src/sdk/macrosmanager.cpp	(working copy)
@@ -86,19 +86,19 @@
     m_RE_IfSp.Compile(_T("[^=!<>]+|(([^=!<>]+)[ ]*(=|==|!=|>|<|>=|<=)[ ]*([^=!<>]+))"), wxRE_EXTENDED | wxRE_NEWLINE);
     m_RE_Script.Compile(_T("(\\[\\[(.*)\\]\\])"), wxRE_EXTENDED | wxRE_NEWLINE);
     m_RE_ToAbsolutePath.Compile(_T("\\$TO_ABSOLUTE_PATH{([^}]*)}"),
-#ifndef __WXMAC__
+#ifdef wxHAS_REGEX_ADVANCED
                                 wxRE_ADVANCED);
 #else
                                 wxRE_EXTENDED);
 #endif
     m_RE_To83Path.Compile(_T("\\$TO_83_PATH{([^}]*)}"),
-#ifndef __WXMAC__
+#ifdef wxHAS_REGEX_ADVANCED
                                 wxRE_ADVANCED);
 #else
                                 wxRE_EXTENDED);
 #endif
     m_RE_RemoveQuotes.Compile(_T("\\$REMOVE_QUOTES{([^}]*)}"),
-#ifndef __WXMAC__
+#ifdef wxHAS_REGEX_ADVANCED
                                 wxRE_ADVANCED);
 #else
                                 wxRE_EXTENDED);
Index: src/plugins/contrib/ToolsPlus/PipedProcessCtrl.cpp
===================================================================
--- src/plugins/contrib/ToolsPlus/PipedProcessCtrl.cpp	(revision 7665)
+++ src/plugins/contrib/ToolsPlus/PipedProcessCtrl.cpp	(working copy)
@@ -206,7 +206,12 @@
 
 void PipedProcessCtrl::ParseLinks(int lineno, int lastline)
 {
-    wxRegEx re(m_linkregex,wxRE_ADVANCED|wxRE_NEWLINE);
+    #ifdef wxHAS_REGEX_ADVANCED
+        wxRegEx re(m_linkregex,wxRE_ADVANCED|wxRE_NEWLINE);
+    #else
+        wxRegEx re(m_linkregex,wxRE_EXTENDED|wxRE_NEWLINE);
+    #endif
+
     while(lineno<lastline)
     {
         int col=0;
@@ -305,7 +310,11 @@
     wxString text=m_textctrl->GetTextRange(start,end+1);
 
     //retrieve the file and line number parts of the link
-    wxRegEx re(m_linkregex,wxRE_ADVANCED|wxRE_NEWLINE);
+    #ifdef wxHAS_REGEX_ADVANCED
+        wxRegEx re(m_linkregex,wxRE_ADVANCED|wxRE_NEWLINE);
+    #else
+        wxRegEx re(m_linkregex,wxRE_EXTENDED|wxRE_NEWLINE);
+    #endif
     wxString file;
     long line;
     if(!re.Matches(text))
Index: src/plugins/contrib/DoxyBlocks/Expressions.h
===================================================================
--- src/plugins/contrib/DoxyBlocks/Expressions.h	(revision 7665)
+++ src/plugins/contrib/DoxyBlocks/Expressions.h	(working copy)
@@ -84,7 +84,7 @@
     "([^)]*)?"                      // The function's parameters.
     "\\)"                           // The closing parenthesis.
     ),
-#ifndef __WXMAC__
+#ifdef wxHAS_REGEX_ADVANCED
     wxRE_ADVANCED);
 #else
     wxRE_EXTENDED);
@@ -104,7 +104,7 @@
     "([^)]*)?"                      // The function's parameters.
     "\\)"                           // The closing parenthesis.
     ),
-#ifndef __WXMAC__
+#ifdef wxHAS_REGEX_ADVANCED
     wxRE_ADVANCED);
 #else
     wxRE_EXTENDED);
