diff -U 3 -H -d -r -N -- digikam-4.6.0/extra/libkface/libkface/CMakeLists.txt digikam-4.6.0.install-libs-dev/extra/libkface/libkface/CMakeLists.txt
--- digikam-4.6.0/extra/libkface/libkface/CMakeLists.txt	2014-12-16 08:54:40.000000000 +0100
+++ digikam-4.6.0.install-libs-dev/extra/libkface/libkface/CMakeLists.txt	2015-01-03 21:03:21.172580963 +0100
@@ -63,9 +63,6 @@
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/database/dbconfig.xml
         DESTINATION ${DATA_INSTALL_DIR}/libkface/database )
 
-# Check if compilation is done through digiKam SC or not. If yes, do not install header files.
-if(NOT DIGIKAMSC_COMPILE_LIBKFACE)
-
     install(FILES facedetector.h
                   recognitiondatabase.h
                   identity.h
@@ -76,9 +73,7 @@
            )
 
     if(NOT WIN32)
-        configure_file(${CMAKE_SOURCE_DIR}/cmake/templates/libkface.pc.cmake.in  ${CMAKE_CURRENT_BINARY_DIR}/libkface.pc)
-        configure_file(${CMAKE_SOURCE_DIR}/cmake/templates/libkface.lsm.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/libkface.lsm)
+        configure_file(${CMAKE_CURRENT_SOURCE_DIR}/../cmake/templates/libkface.pc.cmake.in  ${CMAKE_CURRENT_BINARY_DIR}/libkface.pc)
+        configure_file(${CMAKE_CURRENT_SOURCE_DIR}/../cmake/templates/libkface.lsm.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/libkface.lsm)
         install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libkface.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
     endif()
-
-endif()
diff -U 3 -H -d -r -N -- digikam-4.6.0/extra/libkgeomap/libkgeomap/CMakeLists.txt digikam-4.6.0.install-libs-dev/extra/libkgeomap/libkgeomap/CMakeLists.txt
--- digikam-4.6.0/extra/libkgeomap/libkgeomap/CMakeLists.txt	2014-12-16 08:54:40.000000000 +0100
+++ digikam-4.6.0.install-libs-dev/extra/libkgeomap/libkgeomap/CMakeLists.txt	2015-01-03 21:03:38.204621741 +0100
@@ -73,9 +73,6 @@
 
 install(TARGETS kgeomap ${INSTALL_TARGETS_DEFAULT_ARGS})
 
-# Check if compilation is done through digiKam SC or not. If yes, do not install header files.
-if(NOT DIGIKAMSC_COMPILE_LIBKGEOMAP)
-
     install(FILES libkgeomap_export.h
                   abstractmarkertiler.h
                   dragdrophandler.h
@@ -94,14 +91,12 @@
            )
 
     if(NOT WIN32)
-        configure_file(${CMAKE_SOURCE_DIR}/cmake/templates/libkgeomap.pc.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/libkgeomap.pc)
-        configure_file(${CMAKE_SOURCE_DIR}/cmake/templates/libkgeomap.lsm.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/libkgeomap.lsm)
+        configure_file(${CMAKE_CURRENT_SOURCE_DIR}/../cmake/templates/libkgeomap.pc.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/libkgeomap.pc)
+        configure_file(${CMAKE_CURRENT_SOURCE_DIR}/../cmake/templates/libkgeomap.lsm.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/libkgeomap.lsm)
         install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libkgeomap.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
     endif()
 
     # Install find cmake script to the system for client applications.
 
-    install(FILES ${CMAKE_SOURCE_DIR}/cmake/modules/FindKGeoMap.cmake
+    install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/../cmake/modules/FindKGeoMap.cmake
             DESTINATION ${DATA_INSTALL_DIR}/cmake/modules)
-
-endif()
diff -U 3 -H -d -r -N -- digikam-4.6.0/extra/libmediawiki/libmediawiki/CMakeLists.txt digikam-4.6.0.install-libs-dev/extra/libmediawiki/libmediawiki/CMakeLists.txt
--- digikam-4.6.0/extra/libmediawiki/libmediawiki/CMakeLists.txt	2014-12-16 08:54:40.000000000 +0100
+++ digikam-4.6.0.install-libs-dev/extra/libmediawiki/libmediawiki/CMakeLists.txt	2015-01-03 21:03:05.940544498 +0100
@@ -40,9 +40,6 @@
 
 install(TARGETS mediawiki ${INSTALL_TARGETS_DEFAULT_ARGS})
 
-# Check if compilation is done through digiKam SC or not. If yes, do not install header files.
-if(NOT DIGIKAMSC_COMPILE_LIBMEDIAWIKI)
-
     install(FILES image.h
                   imageinfo.h
                   mediawiki.h
@@ -68,14 +65,12 @@
         DESTINATION ${INCLUDE_INSTALL_DIR}/libmediawiki COMPONENT Devel)
 
     if(NOT WIN32)
-        configure_file(${CMAKE_SOURCE_DIR}/cmake/templates/libmediawiki.pc.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/libmediawiki.pc)
-        configure_file(${CMAKE_SOURCE_DIR}/cmake/templates/libmediawiki.lsm.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/libmediawiki.lsm)
+        configure_file(${CMAKE_CURRENT_SOURCE_DIR}/../cmake/templates/libmediawiki.pc.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/libmediawiki.pc)
+        configure_file(${CMAKE_CURRENT_SOURCE_DIR}/../cmake/templates/libmediawiki.lsm.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/libmediawiki.lsm)
         install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libmediawiki.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig )
     endif()
 
     # Install find cmake script to the system for client applications.
 
-    install(FILES ${CMAKE_SOURCE_DIR}/cmake/modules/FindMediawiki.cmake
+    install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/../cmake/modules/FindMediawiki.cmake
             DESTINATION ${DATA_INSTALL_DIR}/cmake/modules)
-
-endif()
 
