diff -up qtsoap-2.7_1-opensource/qtsoap.pro.install-pub-headers qtsoap-2.7_1-opensource/qtsoap.pro
--- qtsoap-2.7_1-opensource/qtsoap.pro.install-pub-headers	2009-12-16 11:43:54.000000000 +0100
+++ qtsoap-2.7_1-opensource/qtsoap.pro	2011-02-22 14:16:53.872495328 +0100
@@ -3,3 +3,10 @@ CONFIG += ordered
 include(common.pri)
 qtsoap-uselib:SUBDIRS=buildlib
 SUBDIRS+=examples
+
+INSTALLS += PUB_HEADERS
+
+PUB_HEADERS.files = src/QtSoapArray src/QtSoapHttpTransport src/QtSoapNamespaces src/QtSoapSimpleType src/QtSoapStructIterator \
+                    src/QtSoapTypeConstructor src/QtSoapTypeFactory src/QtSoapArrayIterator src/QtSoapMessage src/QtSoapQName \
+                    src/QtSoapStruct src/QtSoapType src/QtSoapTypeConstructorBase src/qtsoap.h
+PUB_HEADERS.path = $$[QT_INSTALL_HEADERS]/QtSoap
