diff -ur facile-1.1/Makefile facile-1.1-makefile-fixes/Makefile
--- facile-1.1/Makefile	2004-09-08 11:51:02.000000000 +0200
+++ facile-1.1-makefile-fixes/Makefile	2008-03-19 03:04:09.000000000 +0100
@@ -7,11 +7,15 @@
 
 install:
 	if test -d $(FACILEDIR); then : ; else mkdir $(FACILEDIR); fi
-	cp src/facile.cmi src/facile.cma src/facile.cmxa src/facile.a $(FACILEDIR)
-	chmod a+r $(FACILEDIR)/facile.cmi
-	chmod a+r $(FACILEDIR)/facile.cma
-	chmod a+r $(FACILEDIR)/facile.cmxa
-	chmod a+r $(FACILEDIR)/facile.a
+	install -m 644 src/*.mli $(FACILEDIR)
+	if [ -e src/facile.cmi ] ; then cp src/facile.cmi $(FACILEDIR) ; fi
+	if [ -e src/facile.cma ] ; then cp src/facile.cma $(FACILEDIR) ; fi
+	if [ -e src/facile.a ] ; then cp src/facile.a $(FACILEDIR) ; fi
+	if [ -e src/facile.cmxa ] ; then cp src/facile.cmxa $(FACILEDIR) ; fi
+	if [ -e $(FACILEDIR)/facile.cmi ] ; then chmod a+r $(FACILEDIR)/facile.cmi ; fi
+	if [ -e $(FACILEDIR)/facile.cma ] ; then chmod a+r $(FACILEDIR)/facile.cma ; fi
+	if [ -e $(FACILEDIR)/facile.cmxa ] ; then chmod a+r $(FACILEDIR)/facile.cmxa ; fi
+	if [ -e $(FACILEDIR)/facile.a ] ; then chmod a+r $(FACILEDIR)/facile.a ; fi
 
 clean:
 	cd src; make clean
diff -ur facile-1.1/src/Makefile facile-1.1-makefile-fixes/src/Makefile
--- facile-1.1/src/Makefile	2004-09-08 11:51:02.000000000 +0200
+++ facile-1.1-makefile-fixes/src/Makefile	2008-03-19 03:04:17.000000000 +0100
@@ -15,8 +15,13 @@
 CSTRCMO = $(CSTR:.ml=.cmo)
 CSTRCMX = $(CSTR:.ml=.cmx)
 
+CMXA =
+ifneq ($(wildcard /usr/bin/ocamlopt),)
+CMXA = facile.cmxa
+endif
+
 # Default target: libraries and toplevel
-all : facile.cma facile.cmxa facile
+all : facile.cma $(CMXA) facile
 
 # Libraries
 facile.cma : $(CSTRCMO)
@@ -31,7 +36,7 @@
 	make facile.cmxa OPTOPT=-p
 	rm -fr $(CSTRCMX)
 	mv facile.cmxa $@
-	mv facile.a facile.p.a
+	if [ -e facile.a ] ;  then mv facile.a facile.p.a ; fi
 
 # Toplevel (to be run with the "-I +facile" option)
 facile : facile.cma
@@ -45,7 +50,7 @@
 	make facile.cmxa OPTOPT="-pp inline_functors"
 	rm -fr $(CSTRCMX)
 	mv facile.cmxa $@
-	mv facile.a facile.exp.a
+	if [ -e facile.a ] ; then mv facile.a facile.exp.a ; fi
 
 .SUFFIXES:
 .SUFFIXES: .ml .mli .mly .mll .cmi .cmo .cmx .p.cmx .s
