Index: tbb30_20101215oss/build/linux.gcc.inc
===================================================================
--- tbb30_20101215oss.orig/build/linux.gcc.inc
+++ tbb30_20101215oss/build/linux.gcc.inc
@@ -42,17 +42,17 @@ LIBDL = -ldl
 
 TBB_NOSTRICT = 1
 
-CPLUS = g++
-CONLY = gcc
-LIB_LINK_FLAGS = -shared -Wl,-soname=$(BUILDING_LIBRARY)
+CPLUS = $(CXX)
+CONLY = $(CC)
+LIB_LINK_FLAGS = $(LDFLAGS) -shared -Wl,-soname=$(BUILDING_LIBRARY)
 LIBS = -lpthread -lrt
-C_FLAGS = $(CPLUS_FLAGS)
+C_FLAGS = $(CFLAGS)
 
 ifeq ($(cfg), release)
-        CPLUS_FLAGS = -DDO_ITT_NOTIFY -O2 -DUSE_PTHREAD
+        CPLUS_FLAGS = $(CXXFLAGS) -DDO_ITT_NOTIFY -O2 -DUSE_PTHREAD
 endif
 ifeq ($(cfg), debug)
-        CPLUS_FLAGS = -DTBB_USE_DEBUG -DDO_ITT_NOTIFY -g -O0 -DUSE_PTHREAD
+        CPLUS_FLAGS = $(CXXFLAGS) -DTBB_USE_DEBUG -DDO_ITT_NOTIFY -g -O0 -DUSE_PTHREAD
 endif
 
 ifneq (0,$(cpp0x))
