diff -Nuar SuiteSparse.orig//Makefile SuiteSparse/Makefile
--- SuiteSparse.orig//Makefile	2010-12-27 13:53:37.133000046 +0200
+++ SuiteSparse/Makefile	2010-12-27 22:38:11.452000113 +0200
@@ -11,13 +11,13 @@
 #	( cd metis-4.0 ; $(MAKE) )
 	( cd AMD ; $(MAKE) )
 	( cd CAMD ; $(MAKE) )
+	( cd CCOLAMD ; $(MAKE) )
 	( cd COLAMD ; $(MAKE) )
 	( cd BTF ; $(MAKE) )
 	( cd KLU ; $(MAKE) )
 	( cd LDL ; $(MAKE) )
-	( cd CCOLAMD ; $(MAKE) )
-	( cd UMFPACK ; $(MAKE) )
 	( cd CHOLMOD ; $(MAKE) )
+	( cd UMFPACK ; $(MAKE) )
 	( cd CSparse ; $(MAKE) )
 	( cd CXSparse ; $(MAKE) )
 	( cd SPQR ; $(MAKE) )
