diff -Nuar SuiteSparse.orig//AMD/Lib/GNUmakefile SuiteSparse/AMD/Lib/GNUmakefile
--- SuiteSparse.orig//AMD/Lib/GNUmakefile	2010-12-03 09:48:29.707137174 +0200
+++ SuiteSparse/AMD/Lib/GNUmakefile	2010-12-03 14:56:22.144896718 +0200
@@ -2,7 +2,7 @@
 # AMD Makefile for compiling on Unix systems (for GNU make only)
 #-------------------------------------------------------------------------------
 
-default: libamd.a
+default: libamd.a libamd.so.3.5.0
 
 include ../../UFconfig/UFconfig.mk
 
@@ -26,18 +26,27 @@
 AMDI = $(addsuffix .o, $(subst amd_,amd_i_,$(AMD)))
 AMDL = $(addsuffix .o, $(subst amd_,amd_l_,$(AMD)))
 
+AMDI_SL = $(addsuffix .oo, $(subst amd_,amd_i_,$(AMD)))
+AMDL_SL = $(addsuffix .oo, $(subst amd_,amd_l_,$(AMD)))
+
 #-------------------------------------------------------------------------------
 # compile each int and long routine (with no real/complex version)
 #-------------------------------------------------------------------------------
 
 amd_global.o: ../Source/amd_global.c $(INC)
 	$(C) -c $< -o $@
+amd_global.oo: ../Source/amd_global.c $(INC)
+	$(C) -fPIC -c $< -o $@
 
 amd_i_%.o: ../Source/amd_%.c $(INC)
 	$(C) -DDINT -c $< -o $@
+amd_i_%.oo: ../Source/amd_%.c $(INC)
+	$(C) -fPIC -DDINT -c $< -o $@
 
 amd_l_%.o: ../Source/amd_%.c $(INC)
 	$(C) -DDLONG -c $< -o $@
+amd_l_%.oo: ../Source/amd_%.c $(INC)
+	$(C) -fPIC -DDLONG -c $< -o $@
 
 #-------------------------------------------------------------------------------
 # Create the libamd.a library (C versions only)
@@ -47,6 +56,11 @@
 	$(AR) libamd.a $^
 	- $(RANLIB) libamd.a
 
+libamd.so.3.5.0: amd_global.oo $(AMDI_SL) $(AMDL_SL)
+	$(CC) -shared $^ -lm -Wl,-soname -Wl,$@ -o $@
+	ln -s $@ libamd.so
+
+
 #-------------------------------------------------------------------------------
 # compile the Fortran versions and the libamdf77.a library
 #-------------------------------------------------------------------------------
@@ -71,6 +85,9 @@
 
 clean:
 	- $(RM) $(CLEAN)
+	- $(RM) *.oo
+	- $(RM) *.a 
+	- $(RM) *.so*
 
 purge: distclean