diff -Nuar SuiteSparse.orig//BTF/Lib/Makefile SuiteSparse/BTF/Lib/Makefile
--- SuiteSparse.orig//BTF/Lib/Makefile	2010-12-27 13:53:37.132000046 +0200
+++ SuiteSparse/BTF/Lib/Makefile	2010-12-27 14:43:29.399000046 +0200
@@ -15,38 +15,56 @@
 
 all: library
 
-library: libbtf.a
+library: libbtf.a libbtf.so.3.5.0
 
 OBJ = btf_order.o btf_maxtrans.o btf_strongcomp.o \
     btf_l_order.o btf_l_maxtrans.o btf_l_strongcomp.o
 
+OBJ_SL = $(subst .o,.oo,$(OBJ)) 
 libbtf.a: $(OBJ)
 	$(AR) libbtf.a $(OBJ)
 	$(RANLIB) libbtf.a
 
-$(OBJ): $(INC)
+libbtf.so.3.5.0: $(OBJ_SL)
+	$(CC) -shared $^ -Wl,-soname -Wl,$@ -o $@
+	ln -s $@ libbtf.so
+
+$(OBJ) $(OBJ_SL): $(INC)
+
 
 #-------------------------------------------------------------------------------
 
 btf_order.o: ../Source/btf_order.c
 	$(C) -c $(I) $< -o $@
+btf_order.oo: ../Source/btf_order.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 btf_maxtrans.o: ../Source/btf_maxtrans.c
 	$(C) -c $(I) $< -o $@
+btf_maxtrans.oo: ../Source/btf_maxtrans.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 btf_strongcomp.o: ../Source/btf_strongcomp.c
 	$(C) -c $(I) $< -o $@
+btf_strongcomp.oo: ../Source/btf_strongcomp.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 #-------------------------------------------------------------------------------
 
 btf_l_order.o: ../Source/btf_order.c
 	$(C) -c $(I) -DDLONG $< -o $@
+btf_l_order.oo: ../Source/btf_order.c
+	$(C) -fPIC -c $(I) -DDLONG $< -o $@
 
 btf_l_maxtrans.o: ../Source/btf_maxtrans.c
 	$(C) -c $(I) -DDLONG $< -o $@
+btf_l_maxtrans.oo: ../Source/btf_maxtrans.c
+	$(C) -fPIC -c $(I) -DDLONG $< -o $@
 
 btf_l_strongcomp.o: ../Source/btf_strongcomp.c
 	$(C) -c $(I) -DDLONG $< -o $@
+btf_l_strongcomp.oo: ../Source/btf_strongcomp.c
+	$(C) -fPIC -c $(I) -DDLONG $< -o $@
 
 #-------------------------------------------------------------------------------
 
@@ -57,3 +75,6 @@
 
 clean:
 	- $(RM) $(CLEAN)
+	- $(RM) *.oo
+	- $(RM) *.a
+	- $(RM) *.so*
