diff -Nuar SuiteSparse.orig//CAMD/Lib/GNUmakefile SuiteSparse/CAMD/Lib/GNUmakefile
--- SuiteSparse.orig//CAMD/Lib/GNUmakefile	2010-12-27 13:53:37.134000046 +0200
+++ SuiteSparse/CAMD/Lib/GNUmakefile	2010-12-27 14:50:10.925000053 +0200
@@ -2,7 +2,7 @@
 # CAMD Makefile for compiling on Unix systems (for GNU make only)
 #-------------------------------------------------------------------------------
 
-default: libcamd.a
+default: libcamd.a libcamd.so.3.5.0
 
 include ../../UFconfig/UFconfig.mk
 
@@ -26,18 +26,27 @@
 CAMDI = $(addsuffix .o, $(subst camd_,camd_i_,$(CAMD)))
 CAMDL = $(addsuffix .o, $(subst camd_,camd_l_,$(CAMD)))
 
+CAMDI_SL = $(addsuffix .oo, $(subst camd_,camd_i_,$(CAMD)))
+CAMDL_SL = $(addsuffix .oo, $(subst camd_,camd_l_,$(CAMD)))
+
 #-------------------------------------------------------------------------------
 # compile each int and long routine (with no real/complex version)
 #-------------------------------------------------------------------------------
 
 camd_global.o: ../Source/camd_global.c $(INC)
 	$(C) -c $< -o $@
+camd_global.oo: ../Source/camd_global.c $(INC)
+	$(C) -fPIC -c $< -o $@
 
 camd_i_%.o: ../Source/camd_%.c $(INC)
 	$(C) -DDINT -c $< -o $@
+camd_i_%.oo: ../Source/camd_%.c $(INC)
+	$(C) -fPIC -DDINT -c $< -o $@
 
 camd_l_%.o: ../Source/camd_%.c $(INC)
 	$(C) -DDLONG -c $< -o $@
+camd_l_%.oo: ../Source/camd_%.c $(INC)
+	$(C) -fPIC -DDLONG -c $< -o $@
 
 #-------------------------------------------------------------------------------
 # Create the libcamd.a library (C versions only)
@@ -46,6 +55,9 @@
 libcamd.a: camd_global.o $(CAMDI) $(CAMDL)
 	$(AR) libcamd.a $^
 	- $(RANLIB) libcamd.a
+libcamd.so.3.5.0: camd_global.oo $(CAMDI_SL) $(CAMDL_SL)
+	$(CC) -shared $^ -lm -Wl,-soname -Wl,$@ -o $@
+	ln -s $@ libcamd.so
 
 #-------------------------------------------------------------------------------
 # Remove all but the files in the original distribution
@@ -58,3 +70,6 @@
 
 distclean: clean
 	- $(RM) libcamd.a
+	- $(RM) *.oo
+	- $(RM) *.a
+	- $(RM) *.so*
