diff -Nuar SuiteSparse.orig//CCOLAMD/Lib/Makefile SuiteSparse/CCOLAMD/Lib/Makefile
--- SuiteSparse.orig//CCOLAMD/Lib/Makefile	2010-12-27 13:53:37.200000046 +0200
+++ SuiteSparse/CCOLAMD/Lib/Makefile	2010-12-27 14:52:53.400000056 +0200
@@ -2,7 +2,7 @@
 # CCOLAMD Makefile
 #-------------------------------------------------------------------------------
 
-default: libccolamd.a
+default: libccolamd.a libccolamd.so.3.5.0
 
 include ../../UFconfig/UFconfig.mk
 
@@ -19,6 +19,13 @@
 	$(CC) $(CFLAGS) $(I) -c ../Source/ccolamd.c -DDLONG -o ccolamd_l.o
 	$(AR) libccolamd.a ccolamd.o ccolamd_l.o ccolamd_global.o
 
+libccolamd.so.3.5.0:  $(SRC) $(INC)
+	$(CC) -fPIC $(CFLAGS) $(I) -c ../Source/ccolamd_global.c -o ccolamd_global.oo
+	$(CC) -fPIC $(CFLAGS) $(I) -c ../Source/ccolamd.c -o ccolamd.oo
+	$(CC) -fPIC $(CFLAGS) $(I) -c ../Source/ccolamd.c -DDLONG -o ccolamd_l.oo
+	$(CC) -shared *.oo -lm -Wl,-soname -Wl,$@ -o $@
+	ln -s $@ libccolamd.so
+
 ccode: libccolamd.a
 
 library: libccolamd.a
@@ -30,3 +37,6 @@
 
 distclean: clean
 	- $(RM) libccolamd.a
+	- $(RM) *.oo
+	- $(RM) *.a
+	- $(RM) *.so*
