diff -Nuar SuiteSparse.orig//COLAMD/Lib/Makefile SuiteSparse/COLAMD/Lib/Makefile
--- SuiteSparse.orig//COLAMD/Lib/Makefile	2010-12-27 13:53:37.173000046 +0200
+++ SuiteSparse/COLAMD/Lib/Makefile	2010-12-27 15:03:31.601000049 +0200
@@ -2,7 +2,7 @@
 # COLAMD Makefile
 #-------------------------------------------------------------------------------
 
-default: libcolamd.a
+default: libcolamd.a libcolamd.so.3.5.0
 
 include ../../UFconfig/UFconfig.mk
 
@@ -19,6 +19,13 @@
 	$(CC) $(CFLAGS) $(I) -c ../Source/colamd.c -DDLONG -o colamd_l.o
 	$(AR) libcolamd.a colamd.o colamd_l.o colamd_global.o
 
+libcolamd.so.3.5.0:  $(SRC) $(INC)
+	$(CC) -fPIC $(CFLAGS) $(I) -c ../Source/colamd_global.c -o colamd_global.oo
+	$(CC) -fPIC $(CFLAGS) $(I) -c ../Source/colamd.c -o colamd.oo
+	$(CC) -fPIC $(CFLAGS) $(I) -c ../Source/colamd.c -DDLONG -o colamd_l.oo
+	$(CC) -shared *.oo -lm -Wl,-soname -Wl,$@ -o $@
+	ln -s $@ libcolamd.so
+
 ccode: libcolamd.a
 
 library: libcolamd.a
@@ -30,3 +37,6 @@
 
 distclean: clean
 	- $(RM) libcolamd.a
+	- $(RM) *.oo
+	- $(RM) *.a
+	- $(RM) *.so*
