diff -Nuar SuiteSparse.orig//CSparse/Lib/Makefile SuiteSparse/CSparse/Lib/Makefile
--- SuiteSparse.orig//CSparse/Lib/Makefile	2010-12-03 09:48:29.616137549 +0200
+++ SuiteSparse/CSparse/Lib/Makefile	2010-12-06 08:27:35.039465648 +0200
@@ -1,11 +1,12 @@
 # Modify the "-O" optimization option for best performance (-O3 on Linux):
-CC = cc
-CFLAGS = -O -I../Include
+#CC = cc
+#CFLAGS = -O -I../Include
 
+C = $(CC) -O -I../Include
 AR = ar cr
 RANLIB = ranlib
 
-all: libcsparse.a
+all: libcsparse.a libcsparse.so.3.5.0
 
 CS = cs_add.o cs_amd.o cs_chol.o cs_cholsol.o cs_counts.o cs_cumsum.o \
 	cs_droptol.o cs_dropzeros.o cs_dupl.o cs_entry.o \
@@ -16,18 +17,31 @@
 	cs_transpose.o cs_compress.o cs_usolve.o cs_utsolve.o cs_scc.o \
 	cs_maxtrans.o cs_dmperm.o cs_updown.o cs_print.o cs_norm.o cs_load.o \
 	cs_dfs.o cs_reach.o cs_spsolve.o cs_ereach.o cs_leaf.o cs_randperm.o
+CS_SL = $(subst .o,.oo,$(CS))
 
 $(CS): ../Include/cs.h Makefile
+$(CS_SL): ../Include/cs.h Makefile
 
 %.o: ../Source/%.c ../Include/cs.h
-	$(CC) $(CFLAGS) -c $<
+	$(C) -c $< 
+
+%.oo: ../Source/%.c ../Include/cs.h
+	$(C) -fPIC -c $< -o $@
 
 libcsparse.a: $(CS)
 	$(AR) libcsparse.a $(CS)
 	$(RANLIB) libcsparse.a
 
+libcsparse.so.3.5.0: $(CS_SL)
+	$(CC) -shared $^ -lm -Wl,-soname -Wl,$@ -o $@
+	ln -s $@ libcsparse.so
+
+
 clean:
-	rm -f *.o
+	rm -f *.o *.oo
+	- $(RM) *.a
+	- $(RM) *.so*
+
 
 purge: distclean
 