diff -Nuar SuiteSparse.orig//KLU/Lib/Makefile SuiteSparse/KLU/Lib/Makefile
--- SuiteSparse.orig//KLU/Lib/Makefile	2010-12-03 09:48:29.702137377 +0200
+++ SuiteSparse/KLU/Lib/Makefile	2010-12-03 15:11:16.016664149 +0200
@@ -17,132 +17,201 @@
 
 all: library
 
-library: libklu.a
+library: libklu.a libklu.so.3.5.0
 
 KLU_D = klu_d.o klu_d_kernel.o klu_d_dump.o \
     klu_d_factor.o klu_d_free_numeric.o klu_d_solve.o \
     klu_d_scale.o klu_d_refactor.o \
     klu_d_tsolve.o klu_d_diagnostics.o klu_d_sort.o klu_d_extract.o
+KLU_D_SL = $(subst .o,.oo,$(KLU_D))
 
 KLU_Z = klu_z.o klu_z_kernel.o klu_z_dump.o \
     klu_z_factor.o klu_z_free_numeric.o klu_z_solve.o \
     klu_z_scale.o klu_z_refactor.o \
     klu_z_tsolve.o klu_z_diagnostics.o klu_z_sort.o klu_z_extract.o
+KLU_Z_SL = $(subst .o,.oo,$(KLU_Z))
 
 KLU_L = klu_l.o klu_l_kernel.o klu_l_dump.o \
     klu_l_factor.o klu_l_free_numeric.o klu_l_solve.o \
     klu_l_scale.o klu_l_refactor.o \
     klu_l_tsolve.o klu_l_diagnostics.o klu_l_sort.o klu_l_extract.o
+KLU_L_SL = $(subst .o,.oo,$(KLU_L))
 
 KLU_ZL = klu_zl.o klu_zl_kernel.o klu_zl_dump.o \
     klu_zl_factor.o klu_zl_free_numeric.o klu_zl_solve.o \
     klu_zl_scale.o klu_zl_refactor.o \
     klu_zl_tsolve.o klu_zl_diagnostics.o klu_zl_sort.o klu_zl_extract.o
+KLU_ZL_SL = $(subst .o,.oo,$(KLU_ZL))
 
 COMMON = \
     klu_free_symbolic.o klu_defaults.o klu_analyze_given.o \
     klu_analyze.o klu_memory.o \
     klu_l_free_symbolic.o klu_l_defaults.o klu_l_analyze_given.o \
     klu_l_analyze.o klu_l_memory.o
+COMMON_SL = $(subst .o,.oo,$(COMMON))
 
 OBJ = $(COMMON) $(KLU_D) $(KLU_Z) $(KLU_L) $(KLU_ZL)
+OBJ_SL = $(COMMON_SL) $(KLU_D_SL) $(KLU_Z_SL) $(KLU_L_SL) $(KLU_ZL_SL)
 
 libklu.a: $(OBJ)
 	$(AR) libklu.a $(OBJ)
 	$(RANLIB) libklu.a
 
-$(OBJ): $(INC)
+libklu.so.3.5.0: $(OBJ_SL)
+	$(CC) -shared $^ ../../AMD/Lib/libamd.so ../../BTF/Lib/libbtf.so ../../COLAMD/Lib/libcolamd.so -Wl,-soname -Wl,$@ -o $@
+	ln -s $@ libklu.so
+
+$(OBJ) $(OBJ_SL): $(INC)
+
 
 #-------------------------------------------------------------------------------
 
 klu_d.o: ../Source/klu.c
 	$(C) -c $(I) $< -o $@
+klu_d.oo: ../Source/klu.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 klu_z.o: ../Source/klu.c
 	$(C) -c -DCOMPLEX $(I) $< -o $@
+klu_z.oo: ../Source/klu.c
+	$(C) -fPIC -c -DCOMPLEX $(I) $< -o $@
 
 klu_d_kernel.o: ../Source/klu_kernel.c
 	$(C) -c $(I) $< -o $@
+klu_d_kernel.oo: ../Source/klu_kernel.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 klu_z_kernel.o: ../Source/klu_kernel.c
 	$(C) -c -DCOMPLEX $(I) $< -o $@
+klu_z_kernel.oo: ../Source/klu_kernel.c
+	$(C) -fPIC -c -DCOMPLEX $(I) $< -o $@
 
 klu_d_sort.o: ../Source/klu_sort.c
 	$(C) -c $(I) $< -o $@
+klu_d_sort.oo: ../Source/klu_sort.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 klu_z_sort.o: ../Source/klu_sort.c
 	$(C) -c -DCOMPLEX $(I) $< -o $@
+klu_z_sort.oo: ../Source/klu_sort.c
+	$(C) -fPIC -c -DCOMPLEX $(I) $< -o $@
 
 klu_d_diagnostics.o: ../Source/klu_diagnostics.c
 	$(C) -c $(I) $< -o $@
+klu_d_diagnostics.oo: ../Source/klu_diagnostics.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 klu_z_diagnostics.o: ../Source/klu_diagnostics.c
 	$(C) -c -DCOMPLEX $(I) $< -o $@
+klu_z_diagnostics.oo: ../Source/klu_diagnostics.c
+	$(C) -fPIC -c -DCOMPLEX $(I) $< -o $@
 
 klu_d_dump.o: ../Source/klu_dump.c
 	$(C) -c $(I) $< -o $@
+klu_d_dump.oo: ../Source/klu_dump.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 klu_z_dump.o: ../Source/klu_dump.c
 	$(C) -c -DCOMPLEX $(I) $< -o $@
+klu_z_dump.oo: ../Source/klu_dump.c
+	$(C) -fPIC -c -DCOMPLEX $(I) $< -o $@
 
 klu_d_factor.o: ../Source/klu_factor.c
 	$(C) -c $(I) $< -o $@
+klu_d_factor.oo: ../Source/klu_factor.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 klu_z_factor.o: ../Source/klu_factor.c
 	$(C) -c -DCOMPLEX $(I) $< -o $@
+klu_z_factor.oo: ../Source/klu_factor.c
+	$(C) -fPIC -c -DCOMPLEX $(I) $< -o $@
 
 klu_d_free_numeric.o: ../Source/klu_free_numeric.c
 	$(C) -c $(I) $< -o $@
+klu_d_free_numeric.oo: ../Source/klu_free_numeric.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 klu_z_free_numeric.o: ../Source/klu_free_numeric.c
 	$(C) -c -DCOMPLEX $(I) $< -o $@
+klu_z_free_numeric.oo: ../Source/klu_free_numeric.c
+	$(C) -fPIC -c -DCOMPLEX $(I) $< -o $@
 
 klu_d_extract.o: ../Source/klu_extract.c
 	$(C) -c $(I) $< -o $@
+klu_d_extract.oo: ../Source/klu_extract.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 klu_z_extract.o: ../Source/klu_extract.c
 	$(C) -c -DCOMPLEX $(I) $< -o $@
+klu_z_extract.oo: ../Source/klu_extract.c
+	$(C) -fPIC -c -DCOMPLEX $(I) $< -o $@
 
 klu_d_refactor.o: ../Source/klu_refactor.c
 	$(C) -c $(I) $< -o $@
+klu_d_refactor.oo: ../Source/klu_refactor.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 klu_z_refactor.o: ../Source/klu_refactor.c
 	$(C) -c -DCOMPLEX $(I) $< -o $@
+klu_z_refactor.oo: ../Source/klu_refactor.c
+	$(C) -fPIC -c -DCOMPLEX $(I) $< -o $@
 
 klu_d_scale.o: ../Source/klu_scale.c
 	$(C) -c $(I) $< -o $@
+klu_d_scale.oo: ../Source/klu_scale.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 klu_z_scale.o: ../Source/klu_scale.c
 	$(C) -c -DCOMPLEX $(I) $< -o $@
+klu_z_scale.oo: ../Source/klu_scale.c
+	$(C) -fPIC -c -DCOMPLEX $(I) $< -o $@
 
 klu_d_solve.o: ../Source/klu_solve.c
 	$(C) -c $(I) $< -o $@
+klu_d_solve.oo: ../Source/klu_solve.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 klu_z_solve.o: ../Source/klu_solve.c
 	$(C) -c -DCOMPLEX $(I) $< -o $@
+klu_z_solve.oo: ../Source/klu_solve.c
+	$(C) -fPIC -c -DCOMPLEX $(I) $< -o $@
 
 klu_d_tsolve.o: ../Source/klu_tsolve.c
 	$(C) -c $(I) $< -o $@
+klu_d_tsolve.oo: ../Source/klu_tsolve.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 klu_z_tsolve.o: ../Source/klu_tsolve.c
 	$(C) -c -DCOMPLEX $(I) $< -o $@
+klu_z_tsolve.oo: ../Source/klu_tsolve.c
+	$(C) -fPIC -c -DCOMPLEX $(I) $< -o $@
 
 #-------------------------------------------------------------------------------
 
 klu_analyze.o: ../Source/klu_analyze.c
 	$(C) -c $(I) $< -o $@
+klu_analyze.oo: ../Source/klu_analyze.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 klu_analyze_given.o: ../Source/klu_analyze_given.c
 	$(C) -c $(I) $< -o $@
+klu_analyze_given.oo: ../Source/klu_analyze_given.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 klu_defaults.o: ../Source/klu_defaults.c
 	$(C) -c $(I) $< -o $@
+klu_defaults.oo: ../Source/klu_defaults.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 klu_free_symbolic.o: ../Source/klu_free_symbolic.c
 	$(C) -c $(I) $< -o $@
+klu_free_symbolic.oo: ../Source/klu_free_symbolic.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 klu_memory.o: ../Source/klu_memory.c
 	$(C) -c $(I) $< -o $@
+klu_memory.oo: ../Source/klu_memory.c
+	$(C) -fPIC -c $(I) $< -o $@
 
 #-------------------------------------------------------------------------------
 
@@ -153,96 +222,158 @@
 
 clean:
 	- $(RM) $(CLEAN)
+	- $(RM) *.oo
+	- $(RM) *.a
+	- $(RM) *.so*
+
 
 #-------------------------------------------------------------------------------
 
 klu_l.o: ../Source/klu.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l.oo: ../Source/klu.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 klu_zl.o: ../Source/klu.c
 	$(C) -c -DCOMPLEX -DDLONG $(I) $< -o $@
+klu_zl.oo: ../Source/klu.c
+	$(C) -fPIC -c -DCOMPLEX -DDLONG $(I) $< -o $@
 
 klu_l_kernel.o: ../Source/klu_kernel.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l_kernel.oo: ../Source/klu_kernel.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 klu_zl_kernel.o: ../Source/klu_kernel.c
 	$(C) -c -DCOMPLEX -DDLONG $(I) $< -o $@
+klu_zl_kernel.oo: ../Source/klu_kernel.c
+	$(C) -fPIC -c -DCOMPLEX -DDLONG $(I) $< -o $@
 
 klu_l_sort.o: ../Source/klu_sort.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l_sort.oo: ../Source/klu_sort.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 klu_zl_sort.o: ../Source/klu_sort.c
 	$(C) -c -DCOMPLEX -DDLONG $(I) $< -o $@
+klu_zl_sort.oo: ../Source/klu_sort.c
+	$(C) -fPIC -c -DCOMPLEX -DDLONG $(I) $< -o $@
 
 klu_l_diagnostics.o: ../Source/klu_diagnostics.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l_diagnostics.oo: ../Source/klu_diagnostics.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 klu_zl_diagnostics.o: ../Source/klu_diagnostics.c
 	$(C) -c -DCOMPLEX -DDLONG $(I) $< -o $@
+klu_zl_diagnostics.oo: ../Source/klu_diagnostics.c
+	$(C) -fPIC -c -DCOMPLEX -DDLONG $(I) $< -o $@
 
 klu_l_dump.o: ../Source/klu_dump.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l_dump.oo: ../Source/klu_dump.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 klu_zl_dump.o: ../Source/klu_dump.c
 	$(C) -c -DCOMPLEX -DDLONG $(I) $< -o $@
+klu_zl_dump.oo: ../Source/klu_dump.c
+	$(C) -fPIC -c -DCOMPLEX -DDLONG $(I) $< -o $@
 
 klu_l_factor.o: ../Source/klu_factor.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l_factor.oo: ../Source/klu_factor.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 klu_zl_factor.o: ../Source/klu_factor.c
 	$(C) -c -DCOMPLEX -DDLONG $(I) $< -o $@
+klu_zl_factor.oo: ../Source/klu_factor.c
+	$(C) -fPIC -c -DCOMPLEX -DDLONG $(I) $< -o $@
 
 klu_l_free_numeric.o: ../Source/klu_free_numeric.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l_free_numeric.oo: ../Source/klu_free_numeric.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 klu_zl_free_numeric.o: ../Source/klu_free_numeric.c
 	$(C) -c -DCOMPLEX -DDLONG $(I) $< -o $@
+klu_zl_free_numeric.oo: ../Source/klu_free_numeric.c
+	$(C) -fPIC -c -DCOMPLEX -DDLONG $(I) $< -o $@
 
 klu_l_extract.o: ../Source/klu_extract.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l_extract.oo: ../Source/klu_extract.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 klu_zl_extract.o: ../Source/klu_extract.c
 	$(C) -c -DCOMPLEX -DDLONG $(I) $< -o $@
+klu_zl_extract.oo: ../Source/klu_extract.c
+	$(C) -fPIC -c -DCOMPLEX -DDLONG $(I) $< -o $@
 
 klu_l_refactor.o: ../Source/klu_refactor.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l_refactor.oo: ../Source/klu_refactor.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 klu_zl_refactor.o: ../Source/klu_refactor.c
 	$(C) -c -DCOMPLEX -DDLONG $(I) $< -o $@
+klu_zl_refactor.oo: ../Source/klu_refactor.c
+	$(C) -fPIC -c -DCOMPLEX -DDLONG $(I) $< -o $@
 
 klu_l_scale.o: ../Source/klu_scale.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l_scale.oo: ../Source/klu_scale.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 klu_zl_scale.o: ../Source/klu_scale.c
 	$(C) -c -DCOMPLEX -DDLONG $(I) $< -o $@
+klu_zl_scale.oo: ../Source/klu_scale.c
+	$(C) -fPIC -c -DCOMPLEX -DDLONG $(I) $< -o $@
 
 klu_l_solve.o: ../Source/klu_solve.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l_solve.oo: ../Source/klu_solve.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 klu_zl_solve.o: ../Source/klu_solve.c
 	$(C) -c -DCOMPLEX -DDLONG $(I) $< -o $@
+klu_zl_solve.oo: ../Source/klu_solve.c
+	$(C) -fPIC -c -DCOMPLEX -DDLONG $(I) $< -o $@
 
 klu_l_tsolve.o: ../Source/klu_tsolve.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l_tsolve.oo: ../Source/klu_tsolve.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 klu_zl_tsolve.o: ../Source/klu_tsolve.c
 	$(C) -c -DCOMPLEX -DDLONG $(I) $< -o $@
+klu_zl_tsolve.oo: ../Source/klu_tsolve.c
+	$(C) -fPIC -c -DCOMPLEX -DDLONG $(I) $< -o $@
 
 #-------------------------------------------------------------------------------
 
 klu_l_analyze.o: ../Source/klu_analyze.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l_analyze.oo: ../Source/klu_analyze.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 klu_l_analyze_given.o: ../Source/klu_analyze_given.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l_analyze_given.oo: ../Source/klu_analyze_given.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 klu_l_defaults.o: ../Source/klu_defaults.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l_defaults.oo: ../Source/klu_defaults.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 klu_l_free_symbolic.o: ../Source/klu_free_symbolic.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l_free_symbolic.oo: ../Source/klu_free_symbolic.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 klu_l_memory.o: ../Source/klu_memory.c
 	$(C) -c -DDLONG $(I) $< -o $@
+klu_l_memory.oo: ../Source/klu_memory.c
+	$(C) -fPIC -c -DDLONG $(I) $< -o $@
 
 #-------------------------------------------------------------------------------