diff -Nuar SuiteSparse.orig//LDL/Lib/Makefile SuiteSparse/LDL/Lib/Makefile
--- SuiteSparse.orig//LDL/Lib/Makefile	2010-12-03 09:48:29.656137745 +0200
+++ SuiteSparse/LDL/Lib/Makefile	2010-12-03 15:12:09.112651598 +0200
@@ -10,7 +10,7 @@
 
 C = $(CC) $(CFLAGS) $(I)
 
-all: libldl.a
+all: libldl.a libldl.so.3.5.0
 
 #-------------------------------------------------------------------------------
 # the ldl library:
@@ -21,11 +21,20 @@
 	$(C) -DLDL_LONG -c ../Source/ldl.c -o ldll.o
 	$(AR) libldl.a ldl.o ldll.o
 	- $(RANLIB) libldl.a
+libldl.so.3.5.0: ../Source/ldl.c ../Include/ldl.h
+	$(C) -fPIC -c ../Source/ldl.c -o ldl.oo
+	$(C) -fPIC -DLDL_LONG -c ../Source/ldl.c -o ldll.oo
+	$(C) -fPIC -shared *.oo -Wl,-soname -Wl,$@ -o $@
+	ln -s $@ libldl.so
+
 
 distclean: purge
 
 purge: clean
 	- $(RM) libldl.a
+	- $(RM) *.oo 
+	- $(RM) *.a
+	- $(RM) *.so* 
 
 clean:
 	- $(RM) $(CLEAN)