diff -Nuar SuiteSparse.orig//UFconfig/UFconfig.mk SuiteSparse/UFconfig/UFconfig.mk
--- SuiteSparse.orig//UFconfig/UFconfig.mk	2010-12-27 13:53:37.068000046 +0200
+++ SuiteSparse/UFconfig/UFconfig.mk	2010-12-27 15:27:36.309000057 +0200
@@ -33,8 +33,8 @@
 # C compiler and compiler flags:  These will normally not give you optimal
 # performance.  You should select the optimization parameters that are best
 # for your system.  On Linux, use "CFLAGS = -O3 -fexceptions" for example.
-CC = cc
-CFLAGS = -O3 -fexceptions
+CC = gcc
+CFLAGS = ${SYSTEM_FLAGS}
 
 # C++ compiler (also uses CFLAGS)
 CPLUSPLUS = g++
@@ -127,8 +127,8 @@
 # The path is relative to where it is used, in CHOLMOD/Lib, CHOLMOD/MATLAB, etc.
 # You may wish to use an absolute path.  METIS is optional.  Compile
 # CHOLMOD with -DNPARTITION if you do not wish to use METIS.
-METIS_PATH = ../../metis-4.0
-METIS = ../../metis-4.0/libmetis.a
+METIS_PATH = /usr/include/metis
+METIS = /usr/lib/libmetis.so
 
 # If you use CHOLMOD_CONFIG = -DNPARTITION then you must use the following
 # options:
