diff -Nuar SuiteSparse.orig//UMFPACK/Lib/GNUmakefile SuiteSparse/UMFPACK/Lib/GNUmakefile
--- SuiteSparse.orig//UMFPACK/Lib/GNUmakefile	2010-12-27 13:53:37.116000046 +0200
+++ SuiteSparse/UMFPACK/Lib/GNUmakefile	2010-12-28 00:59:18.986000094 +0200
@@ -2,7 +2,7 @@
 # UMFPACK Makefile for compiling on Unix systems (for GNU Make)
 #-------------------------------------------------------------------------------
 
-default: libumfpack.a
+default: libumfpack.a libumfpack.so.3.5.0
 
 include ../../UFconfig/UFconfig.mk
 
@@ -81,56 +81,85 @@
 #-------------------------------------------------------------------------------
 
 DI = $(addsuffix .o, $(subst umf_,umf_di_,$(UMF)) $(subst umfpack_,umfpack_di_,$(USER)))
+DI_SL = $(addsuffix .oo, $(subst umf_,umf_di_,$(UMF)) $(subst umfpack_,umfpack_di_,$(USER)))
 DL = $(addsuffix .o, $(subst umf_,umf_dl_,$(UMF)) $(subst umfpack_,umfpack_dl_,$(USER)))
+DL_SL = $(addsuffix .oo, $(subst umf_,umf_dl_,$(UMF)) $(subst umfpack_,umfpack_dl_,$(USER)))
 ZI = $(addsuffix .o, $(subst umf_,umf_zi_,$(UMF)) $(subst umfpack_,umfpack_zi_,$(USER)))
+ZI_SL = $(addsuffix .oo, $(subst umf_,umf_zi_,$(UMF)) $(subst umfpack_,umfpack_zi_,$(USER)))
 ZL = $(addsuffix .o, $(subst umf_,umf_zl_,$(UMF)) $(subst umfpack_,umfpack_zl_,$(USER)))
+ZL_SL = $(addsuffix .oo, $(subst umf_,umf_zl_,$(UMF)) $(subst umfpack_,umfpack_zl_,$(USER)))
 II = $(addsuffix .o, $(subst umf_,umf_i_,$(UMFINT)))
+II_SL = $(addsuffix .oo, $(subst umf_,umf_i_,$(UMFINT)))
 LL = $(addsuffix .o, $(subst umf_,umf_l_,$(UMFINT)))
+LL_SL = $(addsuffix .oo, $(subst umf_,umf_l_,$(UMFINT)))
 GN = $(addsuffix .o, $(subst umfpack_,umfpack_gn_,$(GENERIC)))
-
+GN_SL = $(addsuffix .oo, $(subst umfpack_,umfpack_gn_,$(GENERIC)))
 #-------------------------------------------------------------------------------
 # compile each int and UF_long routine (with no real/complex version)
 #-------------------------------------------------------------------------------
 
 umf_i_%.o: ../Source/umf_%.c $(INC)
 	$(C) -DDINT -c $< -o $@
+umf_i_%.oo: ../Source/umf_%.c $(INC)
+	$(C) -fPIC -DDINT -c $< -o $@
 
 umf_l_%.o: ../Source/umf_%.c $(INC)
 	$(C) -DDLONG -c $< -o $@
-
+umf_l_%.oo: ../Source/umf_%.c $(INC)
+	$(C) -fPIC -DDLONG -c $< -o $@
 #-------------------------------------------------------------------------------
 # compile each routine in the DI version
 #-------------------------------------------------------------------------------
 
 umf_di_%.o: ../Source/umf_%.c $(INC)
 	$(C) -DDINT -c $< -o $@
+umf_di_%.oo: ../Source/umf_%.c $(INC)
+	$(C) -fPIC -DDINT -c $< -o $@
 
 umf_di_%hsolve.o: ../Source/umf_%tsolve.c $(INC)
 	$(C) -DDINT -DCONJUGATE_SOLVE -c $< -o $@
+umf_di_%hsolve.oo: ../Source/umf_%tsolve.c $(INC)
+	$(C) -fPIC -DDINT -DCONJUGATE_SOLVE -c $< -o $@
 
 umf_di_triplet_map_x.o: ../Source/umf_triplet.c $(INC)
 	$(C) -DDINT -DDO_MAP -DDO_VALUES -c $< -o $@
+umf_di_triplet_map_x.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DDINT -DDO_MAP -DDO_VALUES -c $< -o $@
 
 umf_di_triplet_map_nox.o: ../Source/umf_triplet.c $(INC)
 	$(C) -DDINT -DDO_MAP -c $< -o $@
+umf_di_triplet_map_nox.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DDINT -DDO_MAP -c $< -o $@
 
 umf_di_triplet_nomap_x.o: ../Source/umf_triplet.c $(INC)
 	$(C) -DDINT -DDO_VALUES -c $< -o $@
+umf_di_triplet_nomap_x.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DDINT -DDO_VALUES -c $< -o $@
 
 umf_di_triplet_nomap_nox.o: ../Source/umf_triplet.c $(INC)
 	$(C) -DDINT -c $< -o $@
+umf_di_triplet_nomap_nox.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DDINT -c $< -o $@
 
 umf_di_assemble_fixq.o: ../Source/umf_assemble.c $(INC)
 	$(C) -DDINT -DFIXQ -c $< -o $@
+umf_di_assemble_fixq.oo: ../Source/umf_assemble.c $(INC)
+	$(C) -fPIC -DDINT -DFIXQ -c $< -o $@
 
 umf_di_store_lu_drop.o: ../Source/umf_store_lu.c $(INC)
 	$(C) -DDINT -DDROP -c $< -o $@
+umf_di_store_lu_drop.oo: ../Source/umf_store_lu.c $(INC)
+	$(C) -fPIC -DDINT -DDROP -c $< -o $@
 
 umfpack_di_wsolve.o: ../Source/umfpack_solve.c $(INC)
 	$(C) -DDINT -DWSOLVE -c $< -o $@
+umfpack_di_wsolve.oo: ../Source/umfpack_solve.c $(INC)
+	$(C) -fPIC -DDINT -DWSOLVE -c $< -o $@
 
 umfpack_di_%.o: ../Source/umfpack_%.c $(INC)
 	$(C) -DDINT -c $< -o $@
+umfpack_di_%.oo: ../Source/umfpack_%.c $(INC)
+	$(C) -fPIC -DDINT -c $< -o $@
 
 #-------------------------------------------------------------------------------
 # compile each routine in the DL version
@@ -138,33 +167,53 @@
 
 umf_dl_%.o: ../Source/umf_%.c $(INC)
 	$(C) -DDLONG -c $< -o $@
+umf_dl_%.oo: ../Source/umf_%.c $(INC)
+	$(C) -fPIC -DDLONG -c $< -o $@
 
 umf_dl_%hsolve.o: ../Source/umf_%tsolve.c $(INC)
 	$(C) -DDLONG -DCONJUGATE_SOLVE -c $< -o $@
+umf_dl_%hsolve.oo: ../Source/umf_%tsolve.c $(INC)
+	$(C) -fPIC -DDLONG -DCONJUGATE_SOLVE -c $< -o $@
 
 umf_dl_triplet_map_x.o: ../Source/umf_triplet.c $(INC)
 	$(C) -DDLONG -DDO_MAP -DDO_VALUES -c $< -o $@
+umf_dl_triplet_map_x.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DDLONG -DDO_MAP -DDO_VALUES -c $< -o $@
 
 umf_dl_triplet_map_nox.o: ../Source/umf_triplet.c $(INC)
 	$(C) -DDLONG -DDO_MAP -c $< -o $@
+umf_dl_triplet_map_nox.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DDLONG -DDO_MAP -c $< -o $@
 
 umf_dl_triplet_nomap_x.o: ../Source/umf_triplet.c $(INC)
 	$(C) -DDLONG -DDO_VALUES -c $< -o $@
+umf_dl_triplet_nomap_x.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DDLONG -DDO_VALUES -c $< -o $@
 
 umf_dl_triplet_nomap_nox.o: ../Source/umf_triplet.c $(INC)
 	$(C) -DDLONG -c $< -o $@
+umf_dl_triplet_nomap_nox.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DDLONG -c $< -o $@
 
 umf_dl_assemble_fixq.o: ../Source/umf_assemble.c $(INC)
 	$(C) -DDLONG -DFIXQ -c $< -o $@
+umf_dl_assemble_fixq.oo: ../Source/umf_assemble.c $(INC)
+	$(C) -fPIC -DDLONG -DFIXQ -c $< -o $@
 
 umf_dl_store_lu_drop.o: ../Source/umf_store_lu.c $(INC)
 	$(C) -DDLONG -DDROP -c $< -o $@
+umf_dl_store_lu_drop.oo: ../Source/umf_store_lu.c $(INC)
+	$(C) -fPIC -DDLONG -DDROP -c $< -o $@
 
 umfpack_dl_wsolve.o: ../Source/umfpack_solve.c $(INC)
 	$(C) -DDLONG -DWSOLVE -c $< -o $@
+umfpack_dl_wsolve.oo: ../Source/umfpack_solve.c $(INC)
+	$(C) -fPIC -DDLONG -DWSOLVE -c $< -o $@
 
 umfpack_dl_%.o: ../Source/umfpack_%.c $(INC)
 	$(C) -DDLONG -c $< -o $@
+umfpack_dl_%.oo: ../Source/umfpack_%.c $(INC)
+	$(C) -fPIC -DDLONG -c $< -o $@
 
 #-------------------------------------------------------------------------------
 # compile each routine in the ZI version
@@ -172,33 +221,58 @@
 
 umf_zi_%.o: ../Source/umf_%.c $(INC)
 	$(C) -DZINT -c $< -o $@
+umf_zi_%.oo: ../Source/umf_%.c $(INC)
+	$(C) -fPIC -DZINT -c $< -o $@
 
 umf_zi_%hsolve.o: ../Source/umf_%tsolve.c $(INC)
 	$(C) -DZINT -DCONJUGATE_SOLVE -c $< -o $@
+umf_zi_%hsolve.oo: ../Source/umf_%tsolve.c $(INC)
+	$(C) -fPIC -DZINT -DCONJUGATE_SOLVE -c $< -o $@
 
 umf_zi_triplet_map_x.o: ../Source/umf_triplet.c $(INC)
 	$(C) -DZINT -DDO_MAP -DDO_VALUES -c $< -o $@
+umf_zi_triplet_map_x.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DZINT -DDO_MAP -DDO_VALUES -c $< -o $@
 
 umf_zi_triplet_map_nox.o: ../Source/umf_triplet.c $(INC)
 	$(C) -DZINT -DDO_MAP -c $< -o $@
+umf_zi_triplet_map_x.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DZINT -DDO_MAP -DDO_VALUES -c $< -o $@
+
+umf_zi_triplet_map_nox.o: ../Source/umf_triplet.c $(INC)
+	$(C) -DZINT -DDO_MAP -c $< -o $@
+umf_zi_triplet_map_nox.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DZINT -DDO_MAP -c $< -o $@
 
 umf_zi_triplet_nomap_x.o: ../Source/umf_triplet.c $(INC)
 	$(C) -DZINT -DDO_VALUES -c $< -o $@
+umf_zi_triplet_nomap_x.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DZINT -DDO_VALUES -c $< -o $@
 
 umf_zi_triplet_nomap_nox.o: ../Source/umf_triplet.c $(INC)
 	$(C) -DZINT -c $< -o $@
+umf_zi_triplet_nomap_nox.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DZINT -c $< -o $@
 
 umf_zi_assemble_fixq.o: ../Source/umf_assemble.c $(INC)
 	$(C) -DZINT -DFIXQ -c $< -o $@
+umf_zi_assemble_fixq.oo: ../Source/umf_assemble.c $(INC)
+	$(C) -fPIC -DZINT -DFIXQ -c $< -o $@
 
 umf_zi_store_lu_drop.o: ../Source/umf_store_lu.c $(INC)
 	$(C) -DZINT -DDROP -c $< -o $@
+umf_zi_store_lu_drop.oo: ../Source/umf_store_lu.c $(INC)
+	$(C) -fPIC -DZINT -DDROP -c $< -o $@
 
 umfpack_zi_wsolve.o: ../Source/umfpack_solve.c $(INC)
 	$(C) -DZINT -DWSOLVE -c $< -o $@
+umfpack_zi_wsolve.oo: ../Source/umfpack_solve.c $(INC)
+	$(C) -fPIC -DZINT -DWSOLVE -c $< -o $@
 
 umfpack_zi_%.o: ../Source/umfpack_%.c $(INC)
 	$(C) -DZINT -c $< -o $@
+umfpack_zi_%.oo: ../Source/umfpack_%.c $(INC)
+	$(C) -fPIC -DZINT -c $< -o $@
 
 #-------------------------------------------------------------------------------
 # compile each routine in the ZL version
@@ -206,33 +280,53 @@
 
 umf_zl_%.o: ../Source/umf_%.c $(INC)
 	$(C) -DZLONG -c $< -o $@
+umf_zl_%.oo: ../Source/umf_%.c $(INC)
+	$(C) -fPIC -DZLONG -c $< -o $@
 
 umf_zl_%hsolve.o: ../Source/umf_%tsolve.c $(INC)
 	$(C) -DZLONG -DCONJUGATE_SOLVE -c $< -o $@
+umf_zl_%hsolve.oo: ../Source/umf_%tsolve.c $(INC)
+	$(C) -fPIC -DZLONG -DCONJUGATE_SOLVE -c $< -o $@
 
 umf_zl_triplet_map_x.o: ../Source/umf_triplet.c $(INC)
 	$(C) -DZLONG -DDO_MAP -DDO_VALUES -c $< -o $@
+umf_zl_triplet_map_x.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DZLONG -DDO_MAP -DDO_VALUES -c $< -o $@
 
 umf_zl_triplet_map_nox.o: ../Source/umf_triplet.c $(INC)
 	$(C) -DZLONG -DDO_MAP -c $< -o $@
+umf_zl_triplet_map_nox.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DZLONG -DDO_MAP -c $< -o $@
 
 umf_zl_triplet_nomap_x.o: ../Source/umf_triplet.c $(INC)
 	$(C) -DZLONG -DDO_VALUES -c $< -o $@
+umf_zl_triplet_nomap_x.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DZLONG -DDO_VALUES -c $< -o $@
 
 umf_zl_triplet_nomap_nox.o: ../Source/umf_triplet.c $(INC)
 	$(C) -DZLONG -c $< -o $@
+umf_zl_triplet_nomap_nox.oo: ../Source/umf_triplet.c $(INC)
+	$(C) -fPIC -DZLONG -c $< -o $@
 
 umf_zl_assemble_fixq.o: ../Source/umf_assemble.c $(INC)
 	$(C) -DZLONG -DFIXQ -c $< -o $@
+umf_zl_assemble_fixq.oo: ../Source/umf_assemble.c $(INC)
+	$(C) -fPIC -DZLONG -DFIXQ -c $< -o $@
 
 umf_zl_store_lu_drop.o: ../Source/umf_store_lu.c $(INC)
 	$(C) -DZLONG -DDROP -c $< -o $@
+umf_zl_store_lu_drop.oo: ../Source/umf_store_lu.c $(INC)
+	$(C) -fPIC -DZLONG -DDROP -c $< -o $@
 
 umfpack_zl_wsolve.o: ../Source/umfpack_solve.c $(INC)
 	$(C) -DZLONG -DWSOLVE -c $< -o $@
+umfpack_zl_wsolve.oo: ../Source/umfpack_solve.c $(INC)
+	$(C) -fPIC -DZLONG -DWSOLVE -c $< -o $@
 
 umfpack_zl_%.o: ../Source/umfpack_%.c $(INC)
 	$(C) -DZLONG -c $< -o $@
+umfpack_zl_%.oo: ../Source/umfpack_%.c $(INC)
+	$(C) -fPIC -DZLONG -c $< -o $@
 
 #-------------------------------------------------------------------------------
 # Create the generic routines (GN) using a generic rule
@@ -240,6 +334,8 @@
 
 umfpack_gn_%.o: ../Source/umfpack_%.c $(INC)
 	$(C) -c $< -o $@
+umfpack_gn_%.oo: ../Source/umfpack_%.c $(INC)
+	$(C) -fPIC -c $< -o $@
 
 #-------------------------------------------------------------------------------
 # Create the libumfpack.a library
@@ -249,6 +345,10 @@
 	$(AR) libumfpack.a $^
 	- $(RANLIB) libumfpack.a
 
+libumfpack.so.3.5.0: $(II_SL) $(LL_SL) $(GN_SL) $(DI_SL) $(DL_SL) $(ZI_SL) $(ZL_SL)
+	$(CC) -shared $^ -lm -lblas ../../CHOLMOD/Lib/libcholmod.so ../../AMD/Lib/libamd.so -Wl,-soname -Wl,$@ -o $@
+	ln -s $@ libumfpack.so
+
 so: $(II) $(LL) $(GN) $(DI) $(DL) $(ZI) $(ZL)
 	gcc -shared -Wl,-soname,libumfpack.so -o libumfpack.so $^
 
@@ -261,3 +361,7 @@
 
 clean:
 	- $(RM) $(CLEAN)
+	- $(RM) *.oo
+	- $(RM) *.a
+	- $(RM) *.so*
+
