diff -Nuar SuiteSparse.orig//AMD/Makefile SuiteSparse/AMD/Makefile
--- SuiteSparse.orig//AMD/Makefile	2009-11-07 19:37:22.000000000 +0200
+++ SuiteSparse/AMD/Makefile	2010-11-26 15:34:56.563336279 +0200
@@ -12,13 +12,11 @@
 # Do not compile the FORTRAN versions, or MATLAB interface.
 demos:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 
 # Compile all C code, including the C-callable routine and the mexFunctions.
 # Do not compile the FORTRAN versions.
 all:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 	( cd MATLAB ; $(MAKE) )
 
 # compile just the C-callable libraries (not mexFunctions or Demos)
diff -Nuar SuiteSparse.orig//CAMD/Makefile SuiteSparse/CAMD/Makefile
--- SuiteSparse.orig//CAMD/Makefile	2009-11-07 19:37:38.000000000 +0200
+++ SuiteSparse/CAMD/Makefile	2010-11-26 15:35:27.203336876 +0200
@@ -12,13 +12,11 @@
 # Do not compile the FORTRAN versions, or MATLAB interface.
 demos:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 
 # Compile all C code, including the C-callable routine and the mexFunctions.
 # Do not compile the FORTRAN versions.
 all:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 	( cd MATLAB ; $(MAKE) )
 
 # compile just the C-callable libraries (not mexFunctions or Demos)
diff -Nuar SuiteSparse.orig//CCOLAMD/Makefile SuiteSparse/CCOLAMD/Makefile
--- SuiteSparse.orig//CCOLAMD/Makefile	2009-11-07 19:37:40.000000000 +0200
+++ SuiteSparse/CCOLAMD/Makefile	2010-11-26 15:35:41.571336425 +0200
@@ -12,12 +12,10 @@
 # Do not the MATLAB interface.
 demos:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 
 # Compile all C code, including the C-callable routine and the mexFunctions.
 all:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 	( cd MATLAB ; $(MAKE) )
 
 # compile just the C-callable libraries (not mexFunctions or Demos)
diff -Nuar SuiteSparse.orig//CHOLMOD/Makefile SuiteSparse/CHOLMOD/Makefile
--- SuiteSparse.orig//CHOLMOD/Makefile	2009-11-07 19:37:45.000000000 +0200
+++ SuiteSparse/CHOLMOD/Makefile	2010-11-26 15:36:02.067337607 +0200
@@ -14,7 +14,6 @@
 # Compile the C-callable libraries and the Demo programs.
 all:
 	( cd Lib ; $(MAKE) )
-	( cd Demo ; $(MAKE) )
 
 # Compile the C-callable libraries only.
 library:
diff -Nuar SuiteSparse.orig//COLAMD/Makefile SuiteSparse/COLAMD/Makefile
--- SuiteSparse.orig//COLAMD/Makefile	2009-11-07 19:37:50.000000000 +0200
+++ SuiteSparse/COLAMD/Makefile	2010-11-26 15:36:15.923336904 +0200
@@ -12,12 +12,10 @@
 # Do not the MATLAB interface.
 demos:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 
 # Compile all C code, including the C-callable routine and the mexFunctions.
 all:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 	( cd MATLAB ; $(MAKE) )
 
 # compile just the C-callable libraries (not mexFunctions or Demos)
diff -Nuar SuiteSparse.orig//CSparse/Makefile SuiteSparse/CSparse/Makefile
--- SuiteSparse.orig//CSparse/Makefile	2009-11-07 19:33:25.000000000 +0200
+++ SuiteSparse/CSparse/Makefile	2010-11-26 15:36:34.395345268 +0200
@@ -4,7 +4,6 @@
 
 C:
 	( cd Lib ; $(MAKE) )
-	( cd Demo ; $(MAKE) )
 
 all: C cov
 
diff -Nuar SuiteSparse.orig//CXSparse/Makefile SuiteSparse/CXSparse/Makefile
--- SuiteSparse.orig//CXSparse/Makefile	2009-11-07 21:31:28.000000000 +0200
+++ SuiteSparse/CXSparse/Makefile	2010-11-26 15:36:49.459592117 +0200
@@ -8,7 +8,6 @@
 
 C:
 	( cd Lib ; $(MAKE) )
-	( cd Demo ; $(MAKE) )
 
 all: C cov
 
diff -Nuar SuiteSparse.orig//CXSparse_newfiles/Makefile SuiteSparse/CXSparse_newfiles/Makefile
--- SuiteSparse.orig//CXSparse_newfiles/Makefile	2009-11-07 21:31:28.000000000 +0200
+++ SuiteSparse/CXSparse_newfiles/Makefile	2010-11-26 15:37:09.611336595 +0200
@@ -8,7 +8,6 @@
 
 C:
 	( cd Lib ; $(MAKE) )
-	( cd Demo ; $(MAKE) )
 
 all: C cov
 
diff -Nuar SuiteSparse.orig//KLU/Makefile SuiteSparse/KLU/Makefile
--- SuiteSparse.orig//KLU/Makefile	2009-11-07 19:37:57.000000000 +0200
+++ SuiteSparse/KLU/Makefile	2010-11-26 15:37:26.875347378 +0200
@@ -6,7 +6,6 @@
 
 demos:
 	( cd Lib ; $(MAKE) )
-	( cd Demo ; $(MAKE) )
 
 library:
 	( cd Lib ; $(MAKE) )
diff -Nuar SuiteSparse.orig//LDL/Makefile SuiteSparse/LDL/Makefile
--- SuiteSparse.orig//LDL/Makefile	2009-11-07 19:38:00.000000000 +0200
+++ SuiteSparse/LDL/Makefile	2010-11-26 15:37:41.363336786 +0200
@@ -11,12 +11,10 @@
 # Compile all C code, including the C-callable routine.
 demos:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 
 # Compile all C code, including the C-callable routine and the mexFunctions.
 all:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 	( cd MATLAB ; $(MAKE) )
 
 # compile just the C-callable libraries (not mexFunctions or Demos)
diff -Nuar SuiteSparse.orig//RBio/Makefile SuiteSparse/RBio/Makefile
--- SuiteSparse.orig//RBio/Makefile	2009-11-07 19:38:03.000000000 +0200
+++ SuiteSparse/RBio/Makefile	2010-11-26 15:41:31.667337006 +0200
@@ -11,7 +11,6 @@
 # Compile the C-callable libraries and the Demo programs.
 all:
 	( cd Lib ; $(MAKE) )
-	( cd Demo ; $(MAKE) )
 
 # Compile the C-callable libraries only.
 library:
diff -Nuar SuiteSparse.orig//SPQR/Makefile SuiteSparse/SPQR/Makefile
--- SuiteSparse.orig//SPQR/Makefile	2010-06-02 18:39:56.000000000 +0300
+++ SuiteSparse/SPQR/Makefile	2010-11-26 15:41:51.267336597 +0200
@@ -10,7 +10,6 @@
 
 all:
 	( cd Lib ; $(MAKE) )
-	( cd Demo ; $(MAKE) )
 
 library:
 	( cd Lib ; $(MAKE) )
diff -Nuar SuiteSparse.orig//UMFPACK/Makefile SuiteSparse/UMFPACK/Makefile
--- SuiteSparse.orig//UMFPACK/Makefile	2010-06-02 18:41:06.000000000 +0300
+++ SuiteSparse/UMFPACK/Makefile	2010-11-26 15:42:53.003336698 +0200
@@ -15,7 +15,6 @@
 all:
 	( cd ../AMD ; $(MAKE) library )
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 	- cat Doc/License
 
 # compile just the C-callable libraries and demo programs (not mexFunctions)
