diff -Nur SuiteSparse/CHOLMOD/Lib/Makefile SuiteSparse-patched/CHOLMOD/Lib/Makefile
--- SuiteSparse/CHOLMOD/Lib/Makefile	2007-05-05 17:36:56.000000000 +0300
+++ SuiteSparse-patched/CHOLMOD/Lib/Makefile	2008-07-26 11:27:57.000000000 +0300
@@ -116,7 +116,7 @@
 $(OBJ): $(INC)
 
 I = -I../../AMD/Include -I../../AMD/Source -I../../COLAMD/Include \
-	-I$(METIS_PATH)/Lib -I../../CCOLAMD/Include -I../../CAMD/Include \
+	-I/usr/include/metis -I../../CCOLAMD/Include -I../../CAMD/Include \
 	-I../Include -I../../UFconfig
 
 #-------------------------------------------------------------------------------
