--- SDL-1.2.14/include/SDL_events.h.enums	2009-10-13 01:07:19.000000000 +0200
+++ SDL-1.2.14/include/SDL_events.h	2009-11-08 14:38:30.000000000 +0100
@@ -49,7 +49,7 @@
 /*@}*/
 
 /** Event enumerations */
-typedef enum {
+typedef enum SDL_Events {
        SDL_NOEVENT = 0,			/**< Unused (do not remove) */
        SDL_ACTIVEEVENT,			/**< Application loses/gains visibility */
        SDL_KEYDOWN,			/**< Keys pressed */
@@ -85,7 +85,7 @@
 /** @name Predefined event masks */
 /*@{*/
 #define SDL_EVENTMASK(X)	(1<<(X))
-typedef enum {
+typedef enum SDL_EventMasks {
 	SDL_ACTIVEEVENTMASK	= SDL_EVENTMASK(SDL_ACTIVEEVENT),
 	SDL_KEYDOWNMASK		= SDL_EVENTMASK(SDL_KEYDOWN),
 	SDL_KEYUPMASK		= SDL_EVENTMASK(SDL_KEYUP),
