Index: libsdl1.2-1.2.10/build-scripts/makedep.sh
===================================================================
--- libsdl1.2-1.2.10.orig/build-scripts/makedep.sh	2006-05-26 10:20:17.000000000 +0200
+++ libsdl1.2-1.2.10/build-scripts/makedep.sh	2006-05-26 10:21:00.000000000 +0200
@@ -65,7 +65,7 @@
         ;;
         asm) cat >>${output}.new <<__EOF__
 
-	\$(LIBTOOL) --tag=CC --mode=compile \$(auxdir)/strip_fPIC.sh \$(NASM) $src -o \$@
+	\$(LIBTOOL) --tag=CC --mode=compile \$(auxdir)/strip_fPIC.sh \$(NASM) -I\$(srcdir)/src/hermes/ $src -o \$@
 
 __EOF__
         ;;
