Index: libsdl1.2-1.2.13/src/video/x11/SDL_x11events.c
===================================================================
--- libsdl1.2-1.2.13.orig/src/video/x11/SDL_x11events.c	2008-01-02 03:19:45.000000000 +0100
+++ libsdl1.2-1.2.13/src/video/x11/SDL_x11events.c	2008-01-02 03:44:38.000000000 +0100
@@ -1096,7 +1096,13 @@
 	KeySym xsym;
 	SDLKey key;
 
-	xsym = XKeycodeToKeysym(display, kc, 0);
+//	if ( xkey ) {
+//		/* Until we handle the IM protocol, use XLookupString() */
+//		unsigned char keybuf[32];
+//		XLookupString(xkey, keybuf, sizeof(keybuf), &xsym, NULL);
+//	} else {
+		xsym = XKeycodeToKeysym(display, kc, 0);
+//	}
 #ifdef DEBUG_KEYS
 	fprintf(stderr, "Translating key code %d -> 0x%.4x\n", kc, xsym);
 #endif
