Index: plymouth-0.8.3_20100929/src/libply/ply-logger.h
===================================================================
--- plymouth-0.8.3_20100929.orig/src/libply/ply-logger.h
+++ plymouth-0.8.3_20100929/src/libply/ply-logger.h
@@ -91,8 +91,8 @@ do
         ply_logger_flush (logger);                                             \
         errno = _old_errno;                                                    \
         ply_logger_inject (logger,                                             \
-                           "[%s] %45.45s:" format "\r\n",                      \
-                           __FILE__, __func__, ##args);                        \
+                           "" format "\n",                      \
+                            ##args);                        \
         ply_logger_flush (logger);                                             \
         errno = _old_errno;                                                    \
       }                                                                        \
