---
 lib/Xm/TextF.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/lib/Xm/TextF.c
+++ b/lib/Xm/TextF.c
@@ -3919,7 +3919,7 @@ InsertChar(Widget w,
 					    nextPos, insert_string,
 					    insert_length, True);
     } else {
-      char stack_cache[100];
+      wchar_t stack_cache[100];
       insert_string[insert_length] = '\0'; /* NULL terminate for mbstowcs */
       wc_insert_string = (wchar_t*)XmStackAlloc((Cardinal)(insert_length+1) *
 						sizeof(wchar_t), stack_cache);
@@ -6981,7 +6981,7 @@ ValidateString(XmTextFieldWidget tf,
   
   int str_len = 0;
   int i, j;
-  char stack_cache[400];
+  wchar_t stack_cache[400];
   
   if (!is_wchar) {
     char *temp_str, *curr_str, *start_temp;
