* don't do automagic guessing of paths

--- configure.ac
+++ configure.ac
@@ -183,21 +183,8 @@
 
 case "$host_os" in
     freebsd*) CFLAGS="$CFLAGS -DCSRG_BASED -DXNO_MTSAFE_API -DXNO_MTSAFE_PWDAPI";
-        LDFLAGS="$LDFLAGS -L/usr/local/lib";;
+        ;;
     cygwin*) CFLAGS="$CFLAGS -DXNO_MTSAFE_DIRENTDAPI";;
-    solaris* | sunos*) X_LIBS="$X_LIBS -L/usr/X/lib" &&
-        if test x$enable_motif22_compatibility = x
-        then
-            enable_motif22_compatibility="no"
-        fi;
-        if test x$enable_utf8 = x
-        then
-            enable_utf8="no"
-        fi;
-        if test x$enable_xft = x
-        then
-            enable_xft="no"
-        fi;;
 esac
 
 if test x$enable_motif22_compatibility = x
