diff -aur libopensync-0.22.orig/opensync/Makefile.am libopensync-0.22/opensync/Makefile.am
--- libopensync-0.22.orig/opensync/Makefile.am	2007-03-27 13:49:18.000000000 +0200
+++ libopensync-0.22/opensync/Makefile.am	2013-02-05 22:59:33.612901389 +0100
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-AM_CFLAGS = -Wall @XML_CFLAGS@ -Werror @GCOV_CFLAGS@ -DOPENSYNC_PLUGINDIR=\"@OPENSYNC_PLUGINDIR@\" -DOPENSYNC_CONFIGDIR=\"@OPENSYNC_CONFIGDIR@\" -DOPENSYNC_FORMATSDIR=\"@OPENSYNC_FORMATSDIR@\"
+AM_CFLAGS = -Wall @XML_CFLAGS@ @GCOV_CFLAGS@ -DOPENSYNC_PLUGINDIR=\"@OPENSYNC_PLUGINDIR@\" -DOPENSYNC_CONFIGDIR=\"@OPENSYNC_CONFIGDIR@\" -DOPENSYNC_FORMATSDIR=\"@OPENSYNC_FORMATSDIR@\"
 
 INCLUDES = @PACKAGE_CFLAGS@
 
diff -aur libopensync-0.22.orig/opensync/Makefile.in libopensync-0.22/opensync/Makefile.in
--- libopensync-0.22.orig/opensync/Makefile.in	2007-03-27 14:22:04.000000000 +0200
+++ libopensync-0.22/opensync/Makefile.in	2013-02-05 23:14:18.145940854 +0100
@@ -237,7 +237,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-AM_CFLAGS = -Wall @XML_CFLAGS@ -Werror @GCOV_CFLAGS@ -DOPENSYNC_PLUGINDIR=\"@OPENSYNC_PLUGINDIR@\" -DOPENSYNC_CONFIGDIR=\"@OPENSYNC_CONFIGDIR@\" -DOPENSYNC_FORMATSDIR=\"@OPENSYNC_FORMATSDIR@\"
+AM_CFLAGS = -Wall @XML_CFLAGS@ @GCOV_CFLAGS@ -DOPENSYNC_PLUGINDIR=\"@OPENSYNC_PLUGINDIR@\" -DOPENSYNC_CONFIGDIR=\"@OPENSYNC_CONFIGDIR@\" -DOPENSYNC_FORMATSDIR=\"@OPENSYNC_FORMATSDIR@\"
 INCLUDES = @PACKAGE_CFLAGS@
 lib_LTLIBRARIES = libopensync.la
 opensyncincludedir = @OPENSYNC_HEADERDIR@
diff -aur libopensync-0.22.orig/osengine/Makefile.am libopensync-0.22/osengine/Makefile.am
--- libopensync-0.22.orig/osengine/Makefile.am	2007-03-27 13:49:26.000000000 +0200
+++ libopensync-0.22/osengine/Makefile.am	2013-02-05 23:18:28.884869812 +0100
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-AM_CFLAGS = -Werror -Wall @GCOV_CFLAGS@ @XML_CFLAGS@
+AM_CFLAGS = -Wall @GCOV_CFLAGS@ @XML_CFLAGS@
 
 INCLUDES = -I$(top_srcdir) @PACKAGE_CFLAGS@ 
 
diff -aur libopensync-0.22.orig/osengine/Makefile.in libopensync-0.22/osengine/Makefile.in
--- libopensync-0.22.orig/osengine/Makefile.in	2007-03-27 14:22:05.000000000 +0200
+++ libopensync-0.22/osengine/Makefile.in	2013-02-05 23:18:42.329238494 +0100
@@ -235,7 +235,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-AM_CFLAGS = -Werror -Wall @GCOV_CFLAGS@ @XML_CFLAGS@
+AM_CFLAGS = -Wall @GCOV_CFLAGS@ @XML_CFLAGS@
 INCLUDES = -I$(top_srcdir) @PACKAGE_CFLAGS@ 
 @BUILD_ENGINE_TRUE@lib_LTLIBRARIES = libosengine.la
 @BUILD_ENGINE_TRUE@osengineincludedir = @OPENSYNC_ENGINEHEADERDIR@
diff -aur libopensync-0.22.orig/tools/Makefile.am libopensync-0.22/tools/Makefile.am
--- libopensync-0.22.orig/tools/Makefile.am	2007-03-27 13:49:19.000000000 +0200
+++ libopensync-0.22/tools/Makefile.am	2013-02-05 23:23:42.372383588 +0100
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-AM_CFLAGS = -Wall -Werror @GCOV_CFLAGS@
+AM_CFLAGS = -Wall @GCOV_CFLAGS@
 
 INCLUDES = -I$(top_srcdir) @PACKAGE_CFLAGS@ @XML_CFLAGS@ -I$(top_srcdir)/osengine
 
diff -aur libopensync-0.22.orig/tools/Makefile.in libopensync-0.22/tools/Makefile.in
--- libopensync-0.22.orig/tools/Makefile.in	2007-03-27 14:22:07.000000000 +0200
+++ libopensync-0.22/tools/Makefile.in	2013-02-05 23:23:57.823968693 +0100
@@ -240,7 +240,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-AM_CFLAGS = -Wall -Werror @GCOV_CFLAGS@
+AM_CFLAGS = -Wall @GCOV_CFLAGS@
 INCLUDES = -I$(top_srcdir) @PACKAGE_CFLAGS@ @XML_CFLAGS@ -I$(top_srcdir)/osengine
 osyncdump_SOURCES = osyncdump.c
 osyncdump_LDFLAGS = @PACKAGE_LIBS@ @XML_LIBS@ -R $(libdir) @GCOV_LDFLAGS@
diff -aur libopensync-0.22.orig/tools/osynctest.c libopensync-0.22/tools/osynctest.c
--- libopensync-0.22.orig/tools/osynctest.c	2007-03-27 13:49:19.000000000 +0200
+++ libopensync-0.22/tools/osynctest.c	2013-02-05 22:59:08.713321264 +0100
@@ -81,8 +81,7 @@
 double _second()     /* note that some compilers like AIX xlf do not require the trailing  '_' */
 {
     struct timeval tp;
-    int rtn;
-    rtn=gettimeofday(&tp, NULL);
+    gettimeofday(&tp, NULL);
 
     return ((double)tp.tv_sec+(1.e-6)*tp.tv_usec);
 }
diff -aur libopensync-0.22.orig/wrapper/Makefile.am libopensync-0.22/wrapper/Makefile.am
--- libopensync-0.22.orig/wrapper/Makefile.am	2007-03-27 13:49:09.000000000 +0200
+++ libopensync-0.22/wrapper/Makefile.am	2013-02-05 23:01:32.534859659 +0100
@@ -1,10 +1,10 @@
 
-AM_CFLAGS = -Werror
+AM_CFLAGS =
 
 INCLUDES= @PACKAGE_CFLAGS@ $(PYTHON_INCLUDES) -I$(top_srcdir) -I$(top_srcdir)/opensync @XML_CFLAGS@
 
 opensync_wrap.c: $(srcdir)/opensync.i
-	$(SWIG) -Werror -python -modern -I$(top_srcdir)/opensync $(PYTHON_INCLUDES) -o opensync_wrap.c $(srcdir)/opensync.i
+	$(SWIG) -python -modern -I$(top_srcdir)/opensync $(PYTHON_INCLUDES) -o opensync_wrap.c $(srcdir)/opensync.i
 
 EXTRA_DIST = opensync.i
 
diff -aur libopensync-0.22.orig/wrapper/Makefile.in libopensync-0.22/wrapper/Makefile.in
--- libopensync-0.22.orig/wrapper/Makefile.in	2007-03-27 14:22:08.000000000 +0200
+++ libopensync-0.22/wrapper/Makefile.in	2013-02-05 23:01:23.639049984 +0100
@@ -228,7 +228,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-AM_CFLAGS = -Werror
+AM_CFLAGS =
 INCLUDES = @PACKAGE_CFLAGS@ $(PYTHON_INCLUDES) -I$(top_srcdir) -I$(top_srcdir)/opensync @XML_CFLAGS@
 EXTRA_DIST = opensync.i
 CLEANFILES = \
@@ -548,7 +548,7 @@
 
 
 opensync_wrap.c: $(srcdir)/opensync.i
-	$(SWIG) -Werror -python -modern -I$(top_srcdir)/opensync $(PYTHON_INCLUDES) -o opensync_wrap.c $(srcdir)/opensync.i
+	$(SWIG) -python -modern -I$(top_srcdir)/opensync $(PYTHON_INCLUDES) -o opensync_wrap.c $(srcdir)/opensync.i
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
