diff -Naur dispatch/__init__.py dispatch__/__init__.py
--- RuleDispatch.orig/src/dispatch/__init__.py	2007-03-11 18:02:11.000000000 +0200
+++ RuleDispatch/src/dispatch/__init__.py	2009-08-01 05:30:49.000000000 +0300
@@ -95,7 +95,7 @@
     return decorate_assignment(callback)
 
 
-def as(*decorators):
+def as_(*decorators):
     """Use Python 2.4 decorators w/Python 2.2+
 
     Example:
diff -Naur dispatch/predicates.py dispatch__/predicates.py
--- RuleDispatch.orig/src/dispatch/predicates.py	2007-04-21 16:09:43.000000000 +0300
+++ RuleDispatch/src/dispatch/predicates.py	2009-08-01 05:33:55.000000000 +0300
@@ -236,7 +236,7 @@
 
         return or_, (EXPR_GETTER_ID,)
 
-    [as(classmethod)]
+    [as_(classmethod)]
     def immediate(klass,seq):
         for item in seq:
             if item:
@@ -260,7 +260,7 @@
 
         return and_, (EXPR_GETTER_ID,)
 
-    [as(classmethod)]
+    [as_(classmethod)]
     def immediate(klass,seq):
         for item in seq:
             if not item:
@@ -278,7 +278,7 @@
             return get(argIds[2])
         return ifelse, (EXPR_GETTER_ID,)
 
-    [as(classmethod)]
+    [as_(classmethod)]
     def immediate(klass,seq):
         if seq[1]: return seq[0]
         return seq[2]
