--- sash-3.7.orig/sash.c
+++ sash-3.7/sash.c
@@ -399,12 +399,14 @@
 	const char *	commandFile;
 	BOOL		quietFlag;
 	BOOL		aliasFlag;
+        BOOL		interactiveFlag;
 	char		buf[PATH_LEN];
 
 	singleCommand = NULL;
 	commandFile = NULL;
 	quietFlag = FALSE;
 	aliasFlag = FALSE;
+        interactiveFlag = FALSE;
 
 	/*
 	 * Look for options.
@@ -419,11 +421,17 @@
 
 		while (*cp) switch (*cp++)
 		{
+                        case '-':
+                                /*
+                                 * Ignore.  This is so that we can be
+                                 * run from login.
+                                 */
+                                break;
 			case 'c':
 				/*
 				 * Execute specified command.
 				 */
-				if ((argc != 1) || singleCommand)
+				if ((argc != 1) || singleCommand || interactiveFlag)
 					usage();
 
 				singleCommand = *argv++;
@@ -446,6 +454,18 @@
 
 				break;
 
+			case 'i':
+				/*
+				 * Be an interactive shell
+				 * ..is a no-op, but some contexts require this
+				 * ..interactiveFlag is to avoid -ic as a legacy
+				 */
+				 if (singleCommand)
+				 	usage();
+				 
+				 interactiveFlag = TRUE;
+				 break;
+				 
 			case 'p':
 				/*
 				 * Set the prompt string.
@@ -1152,13 +1172,6 @@
 
 	name = argv[1];
 
-	if (access(name, 4))
-	{
-		perror(name);
-
-		return;
-	}
-
 	while (--sourceCount >= 0)
 	{
 		if (sourcefiles[sourceCount] != stdin)
@@ -1168,7 +1181,7 @@
 	argv[argc] = NULL;
 
 	execvp(name, (char **) argv + 1);
-	exit(1);
+	perror(name);
 }
 
 
@@ -1270,7 +1283,7 @@
 {
 	fprintf(stderr, "Stand-alone shell (version %s)\n", version);
 	fprintf(stderr, "\n");
-	fprintf(stderr, "Usage: sash [-a] [-q] [-f fileName] [-c command] [-p prompt]\n");
+	fprintf(stderr, "Usage: sash [-a] [-q] [-f fileName] [-c command] [-p prompt] [-i]\n");
 
 	exit(1);
 }
--- sash-3.7.orig/Makefile
+++ sash-3.7/Makefile
@@ -13,13 +13,12 @@
 HAVE_BSD_MOUNT		= 0
 MOUNT_TYPE		= '"ext3"'
 
-
-CFLAGS = -O3 -Wall -Wmissing-prototypes \
-	-DHAVE_GZIP=$(HAVE_GZIP) \
+DEFS = -DHAVE_GZIP=$(HAVE_GZIP) \
 	-DHAVE_LINUX_ATTR=$(HAVE_LINUX_ATTR) \
 	-DHAVE_LINUX_MOUNT=$(HAVE_LINUX_MOUNT) \
 	-DHAVE_BSD_MOUNT=$(HAVE_BSD_MOUNT) \
 	-DMOUNT_TYPE=$(MOUNT_TYPE)
+CFLAGS = -O3 -Wall -Wmissing-prototypes -g $(DEFS)
 
 LDFLAGS = -static -s
 LIBS = -lz
@@ -43,4 +42,7 @@
 	cp sash $(BINDIR)/sash
 	cp sash.1 $(MANDIR)/sash.1
 
+%.o: %.c
+	$(CC) $(CFLAGS) $(DEFS) -c $<
+
 $(OBJS):	sash.h
--- sash-3.7.orig/cmds.c
+++ sash-3.7/cmds.c
@@ -337,12 +337,9 @@
 		if (fd >= 0)
 		{
 			close(fd);
-
-			continue;
-		}
-
-		if (utime(name, &now) < 0)
+		} else {
 			perror(name);
+		}
 	}
 }
 
--- sash-3.7.orig/cmd_chattr.c
+++ sash-3.7/cmd_chattr.c
@@ -12,7 +12,7 @@
 
 #include <sys/ioctl.h>
 #include <sys/types.h>
-#include <linux/ext2_fs.h>
+#include <ext2fs/ext2_fs.h>
 
 #include "sash.h"
 
