Index: xmoto-0.5.8/src/Makefile.am
===================================================================
--- xmoto-0.5.8.orig/src/Makefile.am
+++ xmoto-0.5.8/src/Makefile.am
@@ -320,7 +320,7 @@ common_SOURCES += $(drawlib_SOURCES)
 common_SOURCES += $(states_SOURCES)
 common_SOURCES += $(thread_SOURCES)
 
-common_LDADD = $(GL_LIBS) $(SDL_LIBS) $(LIBINTL)
+common_LDADD = $(GL_LIBS) $(SDL_LIBS) $(LIBINTL) -lxml2
 if WIN32
 common_LDADD += $(MINGW32_LIBS)
 endif
