Index: warmux-11.04/data/warmux.desktop
===================================================================
--- warmux-11.04.orig/data/warmux.desktop
+++ warmux-11.04/data/warmux.desktop
@@ -8,7 +8,9 @@ Exec=warmux
 Terminal=false
 StartupNotify=false
 GenericName=Arcade Game
+GenericName[fr_FR]=Arcade Game
 GenericName[pl]=Gra zręcznościowa
+GenericName[tr]=Eğlenceli Bir Oyun
 Comment=A 2D game where funny characters launch funny weapons
 Comment[de]=Ein 2D Spiel mit lustigen Figuren und Waffen
 Comment[es]=Un juego en 2D donde extraños personajes se destripan con alegría
@@ -18,4 +20,3 @@ Comment[pl]=Gra 2D, w której zabawne po
 Comment[ro]=Un joc în 2D în care personaje amuzante lansează arme amuzante
 Comment[tr]=Sevimli karakterlerin sevimli silahlar kullandığı 2B bir oyun
 Categories=Game;ArcadeGame;
-GenericName[fr_FR]=Arcade Game
Index: warmux-11.04/data/warmux_files.desktop
===================================================================
--- warmux-11.04.orig/data/warmux_files.desktop
+++ warmux-11.04/data/warmux_files.desktop
@@ -2,7 +2,9 @@
 Version=1.0
 Type=Application
 Name=Warmux folder for replays, teams and maps
+Name[tr]=Takımlar, haritalar ve oyun kayıtlarını içeren Warmux klasörü
 Comment=Location where WarMUX stores replays, chat logs and expect custom teamps and maps
+Comment[tr]=WarMUX'un kaydettiği oyun kayıtları, mesajlaşmalar, takımlar, haritalar vs. konumu
 Icon=folder
 Categories=Game;ArcadeGame;
 Exec=xdg-open ".local/share/wormux"
