diff -ru libicns-0.7.1/icnsutils/png2icns.c libicns-0.7.1_changed/icnsutils/png2icns.c
--- libicns-0.7.1/icnsutils/png2icns.c	2009-05-11 08:17:16.000000000 +0200
+++ libicns-0.7.1_changed/icnsutils/png2icns.c	2010-10-24 16:10:30.686666714 +0200
@@ -70,7 +70,7 @@
 	switch (color_type)
 	{
 		case PNG_COLOR_TYPE_GRAY:
-			png_set_gray_1_2_4_to_8(png_ptr);
+			png_set_expand_gray_1_2_4_to_8(png_ptr);
 
 			if (bit_depth == 16) {
 				png_set_strip_16(png_ptr);
@@ -82,7 +82,7 @@
 			break;
 
 		case PNG_COLOR_TYPE_GRAY_ALPHA:
-			png_set_gray_1_2_4_to_8(png_ptr);
+			png_set_expand_gray_1_2_4_to_8(png_ptr);
 
 			if (bit_depth == 16) {
 				png_set_strip_16(png_ptr);
