--- tools/qdbus/qdbusxml2cpp/qdbusxml2cpp.cpp.orig
+++ tools/qdbus/qdbusxml2cpp/qdbusxml2cpp.cpp
@@ -549,9 +549,11 @@ static void writeProxy(const QString &fi
     } else {
         includeGuard = QLatin1String("QDBUSXML2CPP_PROXY");
     }
+#if 0
     includeGuard = QString(QLatin1String("%1_%2"))
                    .arg(includeGuard)
                    .arg(QDateTime::currentDateTime().toTime_t());
+#endif
     hs << "#ifndef " << includeGuard << endl
        << "#define " << includeGuard << endl
        << endl;
@@ -855,9 +857,11 @@ static void writeAdaptor(const QString &
     } else {
         includeGuard = QLatin1String("QDBUSXML2CPP_ADAPTOR");
     }
+#if 0
     includeGuard = QString(QLatin1String("%1_%2"))
                    .arg(includeGuard)
                    .arg(QDateTime::currentDateTime().toTime_t());
+#endif
     hs << "#ifndef " << includeGuard << endl
        << "#define " << includeGuard << endl
        << endl;
