Index: qt-everywhere-opensource-src-4.8.4/src/corelib/arch/qatomic_generic.h
===================================================================
--- qt-everywhere-opensource-src-4.8.4.orig/src/corelib/arch/qatomic_generic.h
+++ qt-everywhere-opensource-src-4.8.4/src/corelib/arch/qatomic_generic.h
@@ -109,6 +109,16 @@ Q_CORE_EXPORT bool QBasicAtomicPointer_t
 Q_CORE_EXPORT void *QBasicAtomicPointer_fetchAndStoreOrdered(void * volatile *, void *);
 Q_CORE_EXPORT void *QBasicAtomicPointer_fetchAndAddOrdered(void * volatile *, qptrdiff);
 
+inline bool QBasicAtomicPointer_testAndSetOrdered(void * volatile *_q_value, const void *expected_value, const void *newValue)
+{
+    return QBasicAtomicPointer_testAndSetOrdered(_q_value, const_cast<void *>(expected_value), const_cast<void *>(newValue));
+}
+
+inline void *QBasicAtomicPointer_fetchAndStoreOrdered(void * volatile *_q_value, const void *newValue)
+{
+    return QBasicAtomicPointer_fetchAndStoreOrdered(_q_value, const_cast<void *>(newValue));
+}
+
 // Reference counting
 
 inline bool QBasicAtomicInt::ref()
