diff -Nurp linux-3.17-rc5/tools/perf/util/util.h linux-3.17-rc5.fix/tools/perf/util/util.h
--- linux-3.17-rc5/tools/perf/util/util.h	2014-08-04 01:25:02.000000000 +0300
+++ linux-3.17-rc5.fix/tools/perf/util/util.h	2014-09-15 16:38:11.432987481 +0300
@@ -38,7 +38,7 @@
 #define decimal_length(x)	((int)(sizeof(x) * 2.56 + 0.5) + 1)
 
 #define _ALL_SOURCE 1
-#define _BSD_SOURCE 1
+#define _DEFAULT_SOURCE 1
 #define HAS_BOOL
 
 #include <unistd.h>
