diff -Naur gtkglextmm-1.2.0.orig/examples/Makefile.am gtkglextmm-1.2.0/examples/Makefile.am
--- gtkglextmm-1.2.0.orig/examples/Makefile.am	2003-08-15 14:39:33.000000000 +0300
+++ gtkglextmm-1.2.0/examples/Makefile.am	2011-05-02 18:53:52.090000030 +0300
@@ -11,10 +11,6 @@
 	-I$(top_builddir)/gtkglext	\
 	$(GTKGLEXTMM_DEBUG_FLAGS)	\
 	$(GTKGLEXTMM_DEP_CFLAGS)	\
-	-DG_DISABLE_DEPRECATED		\
-	-DGDK_DISABLE_DEPRECATED	\
-	-DGDK_PIXBUF_DISABLE_DEPRECATED	\
-	-DGTK_DISABLE_DEPRECATED
 
 DEPS = \
 	$(top_builddir)/gdkglext/gdkmm/gl/$(gdkglextmm_targetlib)	\
diff -Naur gtkglextmm-1.2.0.orig/examples/Makefile.in gtkglextmm-1.2.0/examples/Makefile.in
--- gtkglextmm-1.2.0.orig/examples/Makefile.in	2006-02-28 03:45:06.000000000 +0200
+++ gtkglextmm-1.2.0/examples/Makefile.in	2011-05-02 18:54:04.830000028 +0300
@@ -208,10 +208,6 @@
 	-I$(top_builddir)/gtkglext	\
 	$(GTKGLEXTMM_DEBUG_FLAGS)	\
 	$(GTKGLEXTMM_DEP_CFLAGS)	\
-	-DG_DISABLE_DEPRECATED		\
-	-DGDK_DISABLE_DEPRECATED	\
-	-DGDK_PIXBUF_DISABLE_DEPRECATED	\
-	-DGTK_DISABLE_DEPRECATED
 
 
 DEPS = \
diff -Naur gtkglextmm-1.2.0.orig/gdkglext/gdkmm/gl/Makefile.am gtkglextmm-1.2.0/gdkglext/gdkmm/gl/Makefile.am
--- gtkglextmm-1.2.0.orig/gdkglext/gdkmm/gl/Makefile.am	2004-05-18 09:27:09.000000000 +0300
+++ gtkglextmm-1.2.0/gdkglext/gdkmm/gl/Makefile.am	2011-05-02 18:53:52.099000030 +0300
@@ -7,9 +7,6 @@
 sublib_cflags = \
 	$(GDKGLEXTMM_DEP_CFLAGS)	\
 	-DGDKMM_GL_COMPILATION		\
-	-DG_DISABLE_DEPRECATED		\
-	-DGDK_DISABLE_DEPRECATED	\
-	-DGDK_PIXBUF_DISABLE_DEPRECATED
 
 files_extra_h = \
 	defs.h		\
diff -Naur gtkglextmm-1.2.0.orig/gdkglext/gdkmm/gl/Makefile.in gtkglextmm-1.2.0/gdkglext/gdkmm/gl/Makefile.in
--- gtkglextmm-1.2.0.orig/gdkglext/gdkmm/gl/Makefile.in	2006-02-28 03:45:07.000000000 +0200
+++ gtkglextmm-1.2.0/gdkglext/gdkmm/gl/Makefile.in	2011-05-02 18:54:04.878000028 +0300
@@ -210,9 +210,6 @@
 sublib_cflags = \
 	$(GDKGLEXTMM_DEP_CFLAGS)	\
 	-DGDKMM_GL_COMPILATION		\
-	-DG_DISABLE_DEPRECATED		\
-	-DGDK_DISABLE_DEPRECATED	\
-	-DGDK_PIXBUF_DISABLE_DEPRECATED
 
 
 files_extra_h = \
diff -Naur gtkglextmm-1.2.0.orig/gtkglext/gtkmm/gl/Makefile.am gtkglextmm-1.2.0/gtkglext/gtkmm/gl/Makefile.am
--- gtkglextmm-1.2.0.orig/gtkglext/gtkmm/gl/Makefile.am	2004-05-18 09:27:09.000000000 +0300
+++ gtkglextmm-1.2.0/gtkglext/gtkmm/gl/Makefile.am	2011-05-02 18:53:52.093000030 +0300
@@ -9,10 +9,6 @@
 	-I$(top_srcdir)/gdkglext	\
 	$(GTKGLEXTMM_DEP_CFLAGS)	\
 	-DGTKMM_GL_COMPILATION		\
-	-DG_DISABLE_DEPRECATED		\
-	-DGDK_DISABLE_DEPRECATED	\
-	-DGDK_PIXBUF_DISABLE_DEPRECATED	\
-	-DGTK_DISABLE_DEPRECATED
 
 files_extra_h = \
 	defs.h		\
diff -Naur gtkglextmm-1.2.0.orig/gtkglext/gtkmm/gl/Makefile.in gtkglextmm-1.2.0/gtkglext/gtkmm/gl/Makefile.in
--- gtkglextmm-1.2.0.orig/gtkglext/gtkmm/gl/Makefile.in	2006-02-28 03:45:09.000000000 +0200
+++ gtkglextmm-1.2.0/gtkglext/gtkmm/gl/Makefile.in	2011-05-02 18:54:04.854000028 +0300
@@ -208,10 +208,6 @@
 	-I$(top_srcdir)/gdkglext	\
 	$(GTKGLEXTMM_DEP_CFLAGS)	\
 	-DGTKMM_GL_COMPILATION		\
-	-DG_DISABLE_DEPRECATED		\
-	-DGDK_DISABLE_DEPRECATED	\
-	-DGDK_PIXBUF_DISABLE_DEPRECATED	\
-	-DGTK_DISABLE_DEPRECATED
 
 
 files_extra_h = \
diff -Naur gtkglextmm-1.2.0.orig/Makefile.in gtkglextmm-1.2.0/Makefile.in
--- gtkglextmm-1.2.0.orig/Makefile.in	2006-02-28 03:45:05.000000000 +0200
+++ gtkglextmm-1.2.0/Makefile.in	2011-05-02 18:54:04.870000028 +0300
@@ -197,7 +197,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-SUBDIRS = tools gdkglext gtkglext m4macros docs
+SUBDIRS = tools gdkglext gtkglext m4macros 
 DIST_SUBDIRS = $(SUBDIRS) examples
 
 EXTRA_DIST = \
