Index: sip-4.8.2/specs/linux-g++
===================================================================
--- sip-4.8.2.orig/specs/linux-g++
+++ sip-4.8.2/specs/linux-g++
@@ -83,7 +83,7 @@ QMAKE_COPY_DIR          = $(COPY) -r
 QMAKE_MOVE		= mv -f
 QMAKE_DEL_FILE		= rm -f
 QMAKE_DEL_DIR		= rmdir
-QMAKE_STRIP             = strip
+QMAKE_STRIP             =
 QMAKE_STRIPFLAGS_LIB 	+= --strip-unneeded
 QMAKE_CHK_DIR_EXISTS	= test -d
 QMAKE_MKDIR		= mkdir -p
Index: sip-4.8.2/specs/linux-lsb
===================================================================
--- sip-4.8.2.orig/specs/linux-lsb
+++ sip-4.8.2/specs/linux-lsb
@@ -83,7 +83,7 @@ QMAKE_COPY_DIR          = $(COPY) -r
 QMAKE_MOVE		= mv -f
 QMAKE_DEL_FILE		= rm -f
 QMAKE_DEL_DIR		= rmdir
-QMAKE_STRIP             = strip
+QMAKE_STRIP             =
 QMAKE_STRIPFLAGS_LIB 	+= --strip-unneeded
 QMAKE_CHK_DIR_EXISTS	= test -d
 QMAKE_MKDIR		= mkdir -p
