--- Makefile.orig.org	2007-07-03 16:12:23.000000000 -0300
+++ Makefile	2007-07-03 16:13:33.000000000 -0300
@@ -6,18 +6,18 @@
 	cd src; make

 install:
-	if test -d $(FACILEDIR); then : ; else mkdir $(FACILEDIR); fi
-	cp src/facile.cmi src/facile.cma src/facile.cmxa src/facile.a $(FACILEDIR)
-	chmod a+r $(FACILEDIR)/facile.cmi
-	chmod a+r $(FACILEDIR)/facile.cma
-	chmod a+r $(FACILEDIR)/facile.cmxa
-	chmod a+r $(FACILEDIR)/facile.a
+	if test -d $(DESTDIR)/$(FACILEDIR); then : ; else mkdir -p $(DESTDIR)/$(FACILEDIR); fi
+	cp src/facile.cmi src/facile.cma src/facile.cmxa src/facile.a $(DESTDIR)/$(FACILEDIR)
+	chmod a+r $(DESTDIR)/$(FACILEDIR)/facile.cmi
+	chmod a+r $(DESTDIR)/$(FACILEDIR)/facile.cma
+	chmod a+r $(DESTDIR)/$(FACILEDIR)/facile.cmxa
+	chmod a+r $(DESTDIR)/$(FACILEDIR)/facile.a

 clean:
 	cd src; make clean

 distclean uninstall:
-	rm -fr $(FACILEDIR)
+	rm -fr $(DESTDIR)/$(FACILEDIR)

 check:
 	cd examples; make FACILEDIR=../src queens.opt; ./queens.opt 8