From 066d61eaecdf0731be99782649340ed141b1d1be Mon Sep 17 00:00:00 2001
From: Mario Ceresa <mrceresa@gmail.com>
Date: Sun, 20 Mar 2011 13:11:54 +0100
Subject: [PATCH 3/6] Removed bundled libcharl reference in dcmjpls

---
 dcmjpls/CMakeLists.txt |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/dcmjpls/CMakeLists.txt b/dcmjpls/CMakeLists.txt
index d89eedf..a6faf10 100644
--- a/dcmjpls/CMakeLists.txt
+++ b/dcmjpls/CMakeLists.txt
@@ -2,4 +2,4 @@
 PROJECT(dcmjpls)
 
 # recurse into subdirectories
-SUBDIRS(libsrc libcharls apps include/dcmtk/dcmjpls)
+SUBDIRS(libsrc apps include/dcmtk/dcmjpls)
-- 
1.7.4

