--- samba-3.6.0/source3/configure.in~	2011-08-09 13:17:47.000000000 +0200
+++ samba-3.6.0/source3/configure.in	2011-10-17 21:10:03.939564051 +0200
@@ -526,7 +526,7 @@
 	  			AC_MSG_RESULT([no large file support])
 				;;
 			5.*)
-			LDFLAGS="$LDFLAGS -lthread"
+			LDFLAGS="$LDFLAGS -lpthread"
 			CPPFLAGS="$CPPFLAGS -D_REENTRANT"
 			AC_MSG_RESULT([enabling large file support])
 			if test "$ac_cv_prog_gcc" = yes; then