diff -urN samba-3.0.25a.org/examples/libsmbclient/smbwrapper/Makefile samba-3.0.25a/examples/libsmbclient/smbwrapper/Makefile
--- samba-3.0.25a.org/examples/libsmbclient/smbwrapper/Makefile	2006-01-25 00:46:42.000000000 +0100
+++ samba-3.0.25a/examples/libsmbclient/smbwrapper/Makefile	2007-06-17 17:43:30.088457851 +0200
@@ -3,7 +3,7 @@
 
 CFLAGS = -I$(SAMBA_INCL) $(EXTLIB_INCL)
 
-LDFLAGS = -L/usr/local/samba/lib -L../../../source3/bin
+LDFLAGS = -L../../../source3/bin
 
 SMBINCLUDE = -I../../../source/include
 CFLAGS= -fpic -g -O0 $(DEFS) $(SMBINCLUDE)
diff -urN samba-3.0.25a.org/examples/libsmbclient/smbwrapper/smbw.c samba-3.0.25a/examples/libsmbclient/smbwrapper/smbw.c
--- samba-3.0.25a.org/examples/libsmbclient/smbwrapper/smbw.c	2006-01-25 00:46:42.000000000 +0100
+++ samba-3.0.25a/examples/libsmbclient/smbwrapper/smbw.c	2007-06-17 17:38:36.620376591 +0200
@@ -548,7 +548,7 @@
 /***************************************************** 
 a wrapper for readlink() - needed for correct errno setting
 *******************************************************/
-int smbw_readlink(const char *fname, char *buf, size_t bufsize)
+ssize_t smbw_readlink(const char *fname, char *buf, size_t bufsize)
 {
 	struct SMBW_stat st;
 	int ret;
diff -urN samba-3.0.25a.org/examples/libsmbclient/smbwrapper/smbw.h samba-3.0.25a/examples/libsmbclient/smbwrapper/smbw.h
--- samba-3.0.25a.org/examples/libsmbclient/smbwrapper/smbw.h	2006-01-25 00:46:42.000000000 +0100
+++ samba-3.0.25a/examples/libsmbclient/smbwrapper/smbw.h	2007-06-17 17:38:36.620376591 +0200
@@ -79,7 +79,7 @@
 int smbw_close(int fd);
 int smbw_fcntl(int fd, int cmd, long arg);
 int smbw_access(const char *name, int mode);
-int smbw_readlink(const char *path, char *buf, size_t bufsize);
+ssize_t smbw_readlink(const char *path, char *buf, size_t bufsize);
 int smbw_unlink(const char *fname);
 int smbw_rename(const char *oldname, const char *newname);
 int smbw_utime(const char *fname, void *buf);
diff -urN samba-3.0.25a.org/examples/libsmbclient/smbwrapper/wrapper.c samba-3.0.25a/examples/libsmbclient/smbwrapper/wrapper.c
--- samba-3.0.25a.org/examples/libsmbclient/smbwrapper/wrapper.c	2006-01-25 00:46:42.000000000 +0100
+++ samba-3.0.25a/examples/libsmbclient/smbwrapper/wrapper.c	2007-06-17 17:39:16.085693163 +0200
@@ -488,7 +490,7 @@
          * function and smbw.c et al.  Functions that return other than an
          * integer value, however, MUST have their return value defined.
          */
-        off64_t         smbw_lseek();
+        /* off64_t         smbw_lseek(); */
         
         if (smbw_fd(fd)) {
 		return (off_t) smbw_lseek(fd, offset, whence);
@@ -518,7 +520,7 @@
          * function and smbw.c et al.  Functions that return other than an
          * integer value, however, MUST have their return value defined.
          */
-        off64_t         smbw_lseek();
+        /* off64_t         smbw_lseek(); */
         
 	if (smbw_fd(fd))
 		ret = smbw_lseek(fd, offset, whence);
diff -urN samba-3.0.25a.org/examples/libsmbclient/smbwrapper/wrapper.h samba-3.0.25a/examples/libsmbclient/smbwrapper/wrapper.h
--- samba-3.0.25a.org/examples/libsmbclient/smbwrapper/wrapper.h	2006-01-25 00:46:42.000000000 +0100
+++ samba-3.0.25a/examples/libsmbclient/smbwrapper/wrapper.h	2007-06-17 17:38:36.620376591 +0200
@@ -123,7 +123,7 @@
         int (* unlink)(char *name);
         int (* utime)(char *name, struct utimbuf *tvp);
         int (* utimes)(char *name, struct timeval *tvp);
-        int (* readlink)(char *path, char *buf, size_t bufsize);
+        ssize_t (* readlink)(char *path, char *buf, size_t bufsize);
         int (* rename)(char *oldname, char *newname);
         int (* rmdir)(char *name);
         int (* symlink)(char *topath, char *frompath);
--- samba-3.5.9/examples/libsmbclient/smbwrapper/Makefile~	2011-07-25 13:34:46.000000000 +0200
+++ samba-3.5.9/examples/libsmbclient/smbwrapper/Makefile	2011-07-25 13:50:38.081681589 +0200
@@ -22,7 +22,7 @@
 	$(CC) -g \
 		-Wl,-init=smbw_initialize \
 		-shared \
-		--export-all-symbols \
+		-Wl,-fexport-all-symbols \
 		-o $(BIN)/smbwrapper.so \
 		$(SMBWRAPPER_OBJS) \
 		$(LDFLAGS) \ 
