diff -urN samba-3.2.1.org/source3/Makefile.in samba-3.2.1/source3/Makefile.in
--- samba-3.2.1.org/source3/Makefile.in	2008-08-05 08:22:33.000000000 +0200
+++ samba-3.2.1/source3/Makefile.in	2008-08-17 23:24:32.085914644 +0200
@@ -1430,7 +1430,7 @@
 	@$(CC) -o $@ $(LDFLAGS) $(PASSDB_LIBS) $(RPCCLIENT_OBJ) \
 		$(DYNEXP) $(TERMLDFLAGS) $(TERMLIBS) $(LIBS) $(POPT_LIBS) \
 		$(KRB5LIBS) $(LDAP_LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS) \
-		$(LIBWBCLIENT_LIBS) $(ZLIB_LIBS) $(PASSDB_LIBS)
+		$(LIBWBCLIENT_LIBS) $(ZLIB_LIBS) $(PASSDB_LIBS) $(NSCD_LIBS)
 
 bin/smbclient@EXEEXT@: $(BINARY_PREREQS) $(CLIENT_OBJ) @BUILD_POPT@ @LIBTALLOC_TARGET@ @LIBTDB_TARGET@ @LIBWBCLIENT_TARGET@
 	@echo Linking $@ 
