diff -Naur font-manager-0.5.7.orig/src/constants.py.in font-manager-0.5.7/src/constants.py.in
--- font-manager-0.5.7.orig/src/constants.py.in	2010-12-17 07:27:59.000000000 +0200
+++ font-manager-0.5.7/src/constants.py.in	2011-04-19 12:39:01.845000009 +0300
@@ -90,7 +90,7 @@
 USER_ACTIONS_CONFIG         =   join(APP_CONFIG_DIR, 'actions.conf')
 
 STANDARD_TEXT = pango_get_sample_string('xx')
-LOCALIZED_TEXT = pango_get_sample_string(LANG.lower().replace('_', '-'))
+LOCALIZED_TEXT = 'Dağ başını duman almış, Gümüş dere durmaz akar.' 
 COMPARE_TEXT = "{0}"
 
 PREVIEW_TEXT = """{0}
