--- blockout-2.4/BlockOut/Utils.cpp	2008-01-09 21:29:06 +0300
+++ blockout-2.4/BlockOut/Utils.cpp	2008-01-15 21:33:37 +0300
@@ -94,8 +94,9 @@
 
   static char ret[32];
   if( time>0 ) {
-#ifdef LOCALTIME32
-    struct tm *ts = _localtime32((__time32_t *)&time);
+#ifdef LOCALTIME64
+    time_t d = (time_t)time;
+    struct tm *ts = localtime(&d);
 #else
     struct tm *ts = localtime((time_t *)&time);
 #endif
@@ -117,10 +118,11 @@
 
   static char ret[32];
   if( time>0 ) {
-#ifdef LOCALTIME32
-    struct tm *ts = _localtime32((__time32_t *)&time);
+#ifdef LOCALTIME64
+    time_t d = (time_t)time;
+    struct tm *ts = localtime(&d);
 #else
-    struct tm *ts = localtime((time_t *)&time);
+     struct tm *ts = localtime((time_t *)&time);
 #endif
     sprintf(ret,"%02d-%02d-%04d",ts->tm_mday,ts->tm_mon+1,ts->tm_year+1900);
   } else {
