--- Python-2.5/Lib/distutils/command/install.py.egginfo	2006-12-06 17:12:57.000000000 -0500
+++ Python-2.5/Lib/distutils/command/install.py	2006-12-06 17:13:10.000000000 -0500
@@ -601,7 +601,7 @@
                     ('install_headers', has_headers),
                     ('install_scripts', has_scripts),
                     ('install_data',    has_data),
-                    ('install_egg_info', lambda self:True),
+                    ('install_egg_info', lambda self:False),
                    ]
 
 # class install
