Index: Lib/platform.py
===================================================================
--- Lib/platform.py	(revision 86735)
+++ Lib/platform.py	(working copy)
@@ -259,7 +259,7 @@
 _supported_dists = (
     'SuSE', 'debian', 'fedora', 'redhat', 'centos',
     'mandrake', 'mandriva', 'rocks', 'slackware', 'yellowdog', 'gentoo',
-    'UnitedLinux', 'turbolinux')
+    'UnitedLinux', 'turbolinux', 'pisilinux')
 
 def _parse_release_file(firstline):
 
