--- freeradius-server-3.0.0.orig/raddb/mods-config/sql/main/postgresql/setup.sql	2013-10-07 15:49:47.000000000 -0400
+++ freeradius-server-3.0.0/raddb/mods-config/sql/main/postgresql/setup.sql	2013-12-13 12:58:38.261559789 -0500
@@ -29,4 +29,7 @@
  * The server can write to the accounting and post-auth logging table.
  */
 GRANT SELECT, INSERT, UPDATE on radacct TO radius;
+GRANT SELECT, USAGE on radacct_radacctid_seq TO radius;
+
 GRANT SELECT, INSERT, UPDATE on radpostauth TO radius;
+GRANT SELECT, USAGE on radpostauth_id_seq TO radius;
