Index: sane-backends-1.0.20/frontend/Makefile.am
===================================================================
--- sane-backends-1.0.20.orig/frontend/Makefile.am
+++ sane-backends-1.0.20/frontend/Makefile.am
@@ -21,7 +21,7 @@ scanimage_LDADD = ../lib/liblib.la ../li
 
 saned_SOURCES = saned.c
 saned_LDADD = ../sanei/libsanei.la ../lib/liblib.la ../lib/libfelib.la \
-  ../backend/libsane.la @SYSLOG_LIBS@
+  ../backend/libsane.la @SYSLOG_LIBS@ @AVAHI_LIBS@
 
 test_SOURCES = test.c
 test_LDADD = ../lib/liblib.la ../lib/libfelib.la ../backend/libsane.la
