diff -up readline-6.2/chardefs.h.orig readline-6.2/chardefs.h
--- readline-6.2/chardefs.h.orig	2011-08-31 09:24:01.852751855 +0200
+++ readline-6.2/chardefs.h	2011-08-31 09:24:07.143685706 +0200
@@ -72,7 +72,7 @@
 #  define IN_CTYPE_DOMAIN(c) isascii(c)
 #endif
 
-#if !defined (isxdigit) && !defined (HAVE_ISXDIGIT)
+#if !defined (isxdigit) && !defined (HAVE_ISXDIGIT) && !defined (__cplusplus) 
 #  define isxdigit(c)   (isdigit((c)) || ((c) >= 'a' && (c) <= 'f') || ((c) >= 'A' && (c) <= 'F'))
 #endif
 
