diff -up readline-6.2/shlib/Makefile.in.shlib readline-6.2/shlib/Makefile.in
--- readline-6.2/shlib/Makefile.in.shlib	2010-12-28 21:56:21.000000000 +0100
+++ readline-6.2/shlib/Makefile.in	2011-02-14 17:34:38.821537496 +0100
@@ -170,7 +170,7 @@ $(SHARED_READLINE):	$(SHARED_OBJ)
 
 $(SHARED_HISTORY):	$(SHARED_HISTOBJ) xmalloc.so xfree.so
 	$(RM) $@
-	$(SHOBJ_LD) ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} -o $@ $(SHARED_HISTOBJ) xmalloc.so xfree.so $(SHLIB_LIBS)
+	$(SHOBJ_LD) ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} -o $@ $(SHARED_HISTOBJ) xmalloc.so xfree.so
 
 # Since tilde.c is shared between readline and bash, make sure we compile 
 # it with the right flags when it's built as part of readline
diff -up readline-6.2/support/shlib-install.shlib readline-6.2/support/shlib-install
--- readline-6.2/support/shlib-install.shlib	2009-10-28 14:30:18.000000000 +0100
+++ readline-6.2/support/shlib-install	2011-02-14 17:33:07.611768164 +0100
@@ -72,7 +72,7 @@ fi
 case "$host_os" in
 hpux*|darwin*|macosx*|linux*)
 	if [ -z "$uninstall" ]; then
-		chmod 555 ${INSTALLDIR}/${LIBNAME}
+		chmod 755 ${INSTALLDIR}/${LIBNAME}
 	fi ;;
 cygwin*|mingw*)
 	IMPLIBNAME=`echo ${LIBNAME} \
diff -up readline-6.2/support/shobj-conf.shlib readline-6.2/support/shobj-conf
--- readline-6.2/support/shobj-conf.shlib	2009-10-28 14:20:21.000000000 +0100
+++ readline-6.2/support/shobj-conf	2011-02-14 17:33:07.612767986 +0100
@@ -112,10 +112,11 @@ sunos5*|solaris2*)
 linux*-*|gnu*-*|k*bsd*-gnu-*|freebsd*-gentoo)
 	SHOBJ_CFLAGS=-fPIC
 	SHOBJ_LD='${CC}'
-	SHOBJ_LDFLAGS='-shared -Wl,-soname,$@'
+ 	SHOBJ_LDFLAGS='$(CFLAGS) -shared -Wl,-soname,$@'
 
-	SHLIB_XLDFLAGS='-Wl,-rpath,$(libdir) -Wl,-soname,`basename $@ $(SHLIB_MINOR)`'
+	SHLIB_XLDFLAGS='-Wl,-soname,`basename $@ $(SHLIB_MINOR)`'
 	SHLIB_LIBVERSION='$(SHLIB_LIBSUFF).$(SHLIB_MAJOR)$(SHLIB_MINOR)'
+ 	SHLIB_LIBS='-ltinfo'
 	;;
 
 freebsd2*)
