diff -up ./complete.c.old ./complete.c
--- ./complete.c.old	2012-01-11 14:26:23.610354454 +0100
+++ ./complete.c	2012-01-11 14:31:44.073348115 +0100
@@ -485,7 +485,7 @@ get_y_or_n (for_pager)
      driven functions.  Have to wait until next major version to add new
      state definition, since it will change value of RL_STATE_DONE. */
 #if defined (READLINE_CALLBACKS)
-  if (RL_ISSTATE (RL_STATE_CALLBACK))
+  if (RL_ISSTATE (RL_STATE_CALLBACK) && (! RL_ISSTATE (RL_STATE_FEDORA_GDB)))
     return 1;
 #endif
 
diff -up ./readline.h.old ./readline.h
--- ./readline.h.old	2012-01-11 14:25:55.049711510 +0100
+++ ./readline.h	2012-01-11 14:28:22.854863691 +0100
@@ -840,6 +840,8 @@ extern int rl_inhibit_completion;
 
 #define RL_STATE_DONE		0x1000000	/* done; accepted line */
 
+#define RL_STATE_FEDORA_GDB		0x2000000	/* exception for fedora gdb */
+
 #define RL_SETSTATE(x)		(rl_readline_state |= (x))
 #define RL_UNSETSTATE(x)	(rl_readline_state &= ~(x))
 #define RL_ISSTATE(x)		(rl_readline_state & (x))
