Index: glade3-3.5.2/data/glade-3.desktop.in.in
===================================================================
--- glade3-3.5.2.orig/data/glade-3.desktop.in.in
+++ glade3-3.5.2/data/glade-3.desktop.in.in
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-_Name=Glade Interface Designer
+_Name=Glade
 _GenericName=User Interface Designer
 _Comment=Create or open user interface designs for GTK+ applications
 Exec=glade-3 %F
