diff -up openbabel-2.2.1b3-20090215-r2890/scripts/Makefile.am.r openbabel-2.2.1b3-20090215-r2890/scripts/Makefile.am
diff -up openbabel-2.2.1b3-20090215-r2890/scripts/perl/Makefile.PL.r openbabel-2.2.1b3-20090215-r2890/scripts/perl/Makefile.PL
--- openbabel-2.2.1b3-20090215-r2890/scripts/perl/Makefile.PL.r	2008-12-01 18:41:42.000000000 +0100
+++ openbabel-2.2.1b3-20090215-r2890/scripts/perl/Makefile.PL	2009-03-01 10:43:01.000000000 +0100
@@ -3,8 +3,8 @@ use ExtUtils::MakeMaker;
 use lib 'inc';
 use Devel::CheckLib;
 
-check_lib_or_exit( lib => 'openbabel' , libpath =>
-@ENV{"LD_LIBRARY_PATH"});
+#check_lib_or_exit( lib => 'openbabel' , libpath =>
+#@ENV{"LD_LIBRARY_PATH"});
 
 $CC = 'g++';
 $srcdir = '../../src';
 
