--- plugins/terminal/terminal/TerminalPlugin.h~	2011-03-20 23:00:12.000000000 +0100
+++ plugins/terminal/terminal/TerminalPlugin.h	2013-01-10 22:02:37.072454220 +0100
@@ -23,8 +23,9 @@ class Preferences : public QWidget, Ui::
     Q_OBJECT
 
     public:
-        Preferences(QTermWidget * term, const QObject * parent,
-                    int colorIX, const QString & emulation,
+        Preferences(const QObject * parent,
+                    const QString &color,
+                    const QString & emulation,
                     const QFont & font);
 };
 
@@ -48,9 +49,9 @@ public:
         void applySettings();
 
 public slots:
-        void colorSchemaChanged(int val);
-        void emulationChanged(const QString & val);
-        void fontChanged(const QFont & val);
+        void colorSchemaChanged(const QString &val);
+        void emulationChanged(const QString &val);
+        void fontChanged(const QFont &val);
         void fontSizeChanged(int val);
 
 private slots:
@@ -61,7 +62,7 @@ private:
 	QAction* termAct_;
 
         QTermWidget * m_term;
-        int prefColorScheme;
+        QString prefColorScheme;
         QString prefEmulation;
         QFont prefFont;
         int prefFontSize;
@@ -70,3 +71,4 @@ private:
 };
 
 #endif
+
--- plugins/terminal/terminal/TerminalPlugin.cpp~	2011-03-20 23:00:12.000000000 +0100
+++ plugins/terminal/terminal/TerminalPlugin.cpp	2013-01-10 22:01:56.751455693 +0100
@@ -10,23 +10,22 @@
 
 #include "qtermwidget.h"
 
-Preferences::Preferences(QTermWidget * term, const QObject * parent,
-                         int colorIX, const QString & emulation,
-                         const QFont & font)
+
+Preferences::Preferences(const QObject *parent,
+                         const QString &color,
+                         const QString &emulation,
+                         const QFont &font)
     : QWidget()
 {
     setupUi(this);
 
-    QStringList colorSchemes;
-    colorSchemes << "White On Black" << \
-                    "Green On Black" << \
-                    "Black On Light Yellow";
-    colorSchemaCombo->addItems(colorSchemes);
-    colorSchemaCombo->setCurrentIndex(colorIX);
-    connect(colorSchemaCombo, SIGNAL(currentIndexChanged(int)),
-            parent, SLOT(colorSchemaChanged(int)));
+    colorSchemaCombo->addItems(QTermWidget::availableColorSchemes());
+    int cix = colorSchemaCombo->findText(color);
+    colorSchemaCombo->setCurrentIndex(cix != -1 ? cix : 0);
+    connect(colorSchemaCombo, SIGNAL(currentIndexChanged(const QString&)),
+            parent, SLOT(colorSchemaChanged(const QString &)));
 
-    emulationComboBox->addItems(term->availableKeyBindings());
+    emulationComboBox->addItems(QTermWidget::availableKeyBindings());
     int eix = emulationComboBox->findText(emulation);
     emulationComboBox->setCurrentIndex(eix != -1 ? eix : 0 );
     connect(emulationComboBox, SIGNAL(currentIndexChanged(const QString&)),
@@ -43,7 +42,6 @@ Preferences::Preferences(QTermWidget * t
 
 
 TerminalPlugin::TerminalPlugin() : QObject(), JuffPlugin() {
-//	w_ = new QWidget();
 	w_ = new QTermWidget();
 	w_->setScrollBarPosition(QTermWidget::ScrollBarRight);
 	w_->setWindowTitle(tr("Terminal"));
@@ -58,7 +56,7 @@ void TerminalPlugin::init()
 {
     prefFont.fromString(PluginSettings::getString(this, "TerminalFont"));
     prefFontSize = prefFont.pointSize();
-    prefColorScheme = PluginSettings::getInt(this, "TerminalColorScheme");
+    prefColorScheme = PluginSettings::getString(this, "TerminalColorScheme");
     prefEmulation = PluginSettings::getString(this, "TerminalEmulation");
     w_->setColorScheme(prefColorScheme);
     w_->setTerminalFont(prefFont);
@@ -98,7 +96,7 @@ Juff::ActionList TerminalPlugin::mainMen
 
 QWidget * TerminalPlugin::settingsPage() const
 {
-    return new Preferences(w_, this, prefColorScheme, prefEmulation, prefFont);
+    return new Preferences(this, prefColorScheme, prefEmulation, prefFont);
 }
 
 void TerminalPlugin::applySettings()
@@ -112,9 +110,9 @@ void TerminalPlugin::applySettings()
     w_->setKeyBindings(prefEmulation);
 }
 
-void TerminalPlugin::colorSchemaChanged(int val)
+void TerminalPlugin::colorSchemaChanged(const QString & val)
 {
-    prefColorScheme = val + 1;
+    prefColorScheme = val;
 }
 
 void TerminalPlugin::emulationChanged(const QString & val)
@@ -153,3 +151,4 @@ void TerminalPlugin::showTerminal()
 }
 
 Q_EXPORT_PLUGIN2(terminal, TerminalPlugin)
+
