Index: src/lib/Document.cpp
===================================================================
--- src/lib/Document.cpp	(revision 1167)
+++ src/lib/Document.cpp	(working copy)
@@ -122,7 +122,7 @@
 bool Document::isNoname(const QString& fileName, bool* ok) {
 	if ( fileName.isEmpty() ) {
 		if ( ok != NULL ) {
-			ok = false;
+            *ok = false;
 		}
 		return true;
 	}
Index: src/3rd_party/qtsingleapplication/qtlocalpeer.cpp
===================================================================
--- src/3rd_party/qtsingleapplication/qtlocalpeer.cpp	(revision 1167)
+++ src/3rd_party/qtsingleapplication/qtlocalpeer.cpp	(working copy)
@@ -57,6 +57,7 @@
 #endif
 #if defined(Q_OS_UNIX)
 #include <time.h>
+#include <unistd.h>
 #endif
 
 namespace QtLP_Private {
