--- plugins/cmake/FindQTermWidget.cmake.OSmanOS	2010-11-04 10:26:03.000000000 +0200
+++ plugins/cmake/FindQTermWidget.cmake		2013-01-09 23:26:37.848983769 +0200
@@ -12,13 +12,14 @@
 
 if (QTERMWIDGET_INCLUDE_DIR AND QTERMWIDGET_LIBRARIES)
   # Already in cache, be silent
-  set(PostgreSQL_FIND_QUIETLY TRUE)
+  set(QTERMWIDGET_FIND_QUIETLY TRUE)
 endif (QTERMWIDGET_INCLUDE_DIR AND QTERMWIDGET_LIBRARIES)
 
 
 find_path(QTERMWIDGET_INCLUDE_DIR qtermwidget.h
    ${QTERMWIDGET_PATH_INCLUDES}/
    /usr/include/
+   /usr/include/qtermwidget/
    /usr/local/include/
    /opt/local/include/
 )
