diff -uNBwp CMakeLists.txt.orig CMakeLists.txt
--- CMakeLists.txt.orig	2011-02-01 12:07:04.856019962 +0100
+++ CMakeLists.txt	2011-02-01 12:09:23.424026968 +0100
@@ -33,7 +33,7 @@ string ( STRIP "${JUFFED_VERSION}" JUFFE
 string ( STRIP "${BASE_VERSION}"   BASE_VERSION )
 
 # flag indicating that this is a development version
-set ( DEV 1 )
+set ( DEV 0 )
 
 if ( DEV )
 	# SVN revision
@@ -53,7 +53,7 @@ message ( STATUS "${JUFFED} version: ${J
 # Directories
 ####################
 
-include ( cmake/LibSuffix.cmake )
+set (LIB_SUFFIX "")
 
 set (BIN_INSTALL_DIR
 	"${PREFIX}/bin"
diff -uNBwp plugins/cmake/FindQTermWidget.cmake.orig plugins/cmake/FindQTermWidget.cmake
--- plugins/cmake/FindQTermWidget.cmake.orig	2011-02-03 11:04:06.896999753 +0100
+++ plugins/cmake/FindQTermWidget.cmake	2011-02-03 11:15:02.833000021 +0100
@@ -12,13 +12,14 @@
 
 if (QTERMWIDGET_INCLUDE_DIR AND QTERMWIDGET_LIBRARIES)
   # Already in cache, be silent
-  set(PostgreSQL_FIND_QUIETLY TRUE)
+  set(QTERMWIDGET_FIND_QUIETLY TRUE)
 endif (QTERMWIDGET_INCLUDE_DIR AND QTERMWIDGET_LIBRARIES)
 
 
 find_path(QTERMWIDGET_INCLUDE_DIR qtermwidget.h
    ${QTERMWIDGET_PATH_INCLUDES}/
    /usr/include/
+   /usr/include/qtermwidget/
    /usr/local/include/
    /opt/local/include/
 )
