--- vte-0.28.0/python/Makefile.am.link	2011-04-10 18:37:45.000000000 +0200
+++ vte-0.28.0/python/Makefile.am	2011-04-10 18:37:56.000000000 +0200
@@ -22,7 +22,7 @@
 # we do this to suport building with -Wl,-z,defs which errors on
 # vtemodule.so as we cannot include $(PYTHON_LIBS) due to bug 410986.
 vtemodule_la_LDFLAGS = -module -avoid-version -export-symbols-regex initvte $(PYTHON_LDFLAGS) $(AM_LDFLAGS)
-vtemodule_la_LIBADD = $(top_builddir)/src/libvte.la $(LIBS) $(PYGTK_LIBS) $(VTE_LIBS) $(X_LIBS)
+vtemodule_la_LIBADD = $(top_builddir)/src/libvte.la $(LIBS) $(PYGTK_LIBS) $(VTE_LIBS) $(X_LIBS) $(PYTHON_LIBS)
 nodist_vtemodule_la_SOURCES = vte.c
 
 vte.c: vte.defs vte.override
--- vte-0.28.0/python/Makefile.in.link	2011-04-10 18:38:01.000000000 +0200
+++ vte-0.28.0/python/Makefile.in	2011-04-10 18:38:12.000000000 +0200
@@ -339,7 +339,7 @@
 # we do this to suport building with -Wl,-z,defs which errors on
 # vtemodule.so as we cannot include $(PYTHON_LIBS) due to bug 410986.
 @BUILD_PYTHON_MODULES_TRUE@vtemodule_la_LDFLAGS = -module -avoid-version -export-symbols-regex initvte $(PYTHON_LDFLAGS) $(AM_LDFLAGS)
-@BUILD_PYTHON_MODULES_TRUE@vtemodule_la_LIBADD = $(top_builddir)/src/libvte.la $(LIBS) $(PYGTK_LIBS) $(VTE_LIBS) $(X_LIBS)
+@BUILD_PYTHON_MODULES_TRUE@vtemodule_la_LIBADD = $(top_builddir)/src/libvte.la $(LIBS) $(PYGTK_LIBS) $(VTE_LIBS) $(X_LIBS) $(PYTHON_LIBS)
 @BUILD_PYTHON_MODULES_TRUE@nodist_vtemodule_la_SOURCES = vte.c
 all: all-am
 