--- src/Makefile.am.orig	2010-01-31 21:06:52.000000000 +0200
+++ src/Makefile.am	2010-02-07 02:51:56.656372942 +0200
@@ -1,5 +1,5 @@
-AM_CFLAGS = $(libglib2_CFLAGS) $(libfuse_CFLAGS) $(libimobiledevice_CFLAGS) -g
-AM_LDFLAGS =  $(libglib2_LIBS) $(libfuse_LIBS) $(libimobiledevice_LIBS)
+INCLUDES = $(libimobiledevice_CFLAGS) $(libfuse_CFLAGS) $(libglib2_CFLAGS)
+ifuse_LDADD =  $(libimobiledevice_LIBS) $(libfuse_LIBS) $(libglib2_LIBS)
 
 bin_PROGRAMS = ifuse
 