Index: jboss-6.0.0.Final/bin/run.sh
===================================================================
--- jboss-6.0.0.Final.orig/bin/run.sh
+++ jboss-6.0.0.Final/bin/run.sh
@@ -96,7 +96,7 @@ if [ "$cygwin" = "false" ]; then
     if [ "$?" -eq 0 ]; then
         # Darwin does not allow RLIMIT_INFINITY on file soft limit
         if [ "$darwin" = "true" -a "$MAX_FD_LIMIT" = "unlimited" ]; then
-            MAX_FD_LIMIT=`/usr/sbin/sysctl -n kern.maxfilesperproc`
+            MAX_FD_LIMIT=`/sbin/sysctl -n kern.maxfilesperproc`
         fi
 
     if [ "$MAX_FD" = "maximum" -o "$MAX_FD" = "max" ]; then
