Index: jboss-6.0.0.Final/bin/run.sh
===================================================================
--- jboss-6.0.0.Final.orig/bin/run.sh
+++ jboss-6.0.0.Final/bin/run.sh
@@ -24,6 +24,10 @@ PROGNAME=`basename "$PRG"`
 
 GREP="grep"
 
+JBOSS_PIDFILE=/var/run/jboss6.pid
+LAUNCH_JBOSS_IN_BACKGROUND=1
+
+
 # Use the maximum available, or set MAX_FD != -1 to use that
 MAX_FD="maximum"
 
@@ -259,6 +263,7 @@ while true; do
          -classpath \"$JBOSS_CLASSPATH\" \
          org.jboss.Main "$@" "&"
       JBOSS_PID=$!
+	  echo $JBOSS_PID > $JBOSS_PIDFILE
       # Trap common signals and relay them to the jboss process
       trap "kill -HUP  $JBOSS_PID" HUP
       trap "kill -TERM $JBOSS_PID" INT
