--- xf86-video-tseng-1.1.0/src/tseng_driver.c.jx	2006-04-07 17:43:17.000000000 -0400
+++ xf86-video-tseng-1.1.0/src/tseng_driver.c	2007-03-28 12:47:45.000000000 -0400
@@ -1069,7 +1069,7 @@
      * Our default depth is 8, so pass it to the helper function.
      * Our preference for depth 24 is 24bpp, so tell it that too.
      */
-    if (!xf86SetDepthBpp(pScrn, 8, 8, 8, Support24bppFb | Support32bppFb |
+    if (!xf86SetDepthBpp(pScrn, 0, 0, 0, Support24bppFb | Support32bppFb |
                          SupportConvert32to24 | PreferConvert32to24)) {
 	return FALSE;
     } else {
