diff -up nfs-utils-1.2.1/utils/statd/sm-notify.man.save nfs-utils-1.2.1/utils/statd/sm-notify.man
--- nfs-utils-1.2.1/utils/statd/sm-notify.man.save	2010-01-15 11:10:25.096874609 -0500
+++ nfs-utils-1.2.1/utils/statd/sm-notify.man	2010-01-15 11:14:12.636873981 -0500
@@ -186,7 +186,7 @@ where NSM state information resides.
 If this option is not specified,
 .B sm-notify
 uses
-.I /var/lib/nfs
+.I /var/lib/nfs/statd
 by default.
 .IP
 After starting,
@@ -287,13 +287,13 @@ Currently, the
 command supports sending notification only via datagram transport protocols.
 .SH FILES
 .TP 2.5i
-.I /var/lib/nfs/sm
+.I /var/lib/nfs/statd/sm
 directory containing monitor list
 .TP 2.5i
-.I /var/lib/nfs/sm.bak
+.I /var/lib/nfs/statd/sm.bak
 directory containing notify list
 .TP 2.5i
-.I /var/lib/nfs/state
+.I /var/lib/nfs/statd/state
 NSM state number for this host
 .TP 2.5i
 .I /proc/sys/fs/nfs/nsm_local_state
diff -up nfs-utils-1.2.1/utils/statd/statd.man.save nfs-utils-1.2.1/utils/statd/statd.man
--- nfs-utils-1.2.1/utils/statd/statd.man.save	2010-01-15 11:10:25.098906325 -0500
+++ nfs-utils-1.2.1/utils/statd/statd.man	2010-01-15 11:11:03.874769717 -0500
@@ -232,7 +232,7 @@ where NSM state information resides.
 If this option is not specified,
 .B rpc.statd
 uses
-.I /var/lib/nfs
+.I /var/lib/nfs/statd
 by default.
 .IP
 After starting,
@@ -368,13 +368,13 @@ As long as at least one network transpor
 will operate.
 .SH FILES
 .TP 2.5i
-.I /var/lib/nfs/sm
+.I /var/lib/nfs/statd/sm
 directory containing monitor list
 .TP 2.5i
-.I /var/lib/nfs/sm.bak
+.I /var/lib/nfs/statd/sm.bak
 directory containing notify list
 .TP 2.5i
-.I /var/lib/nfs/state
+.I /var/lib/nfs/statd/state
 NSM state number for this host
 .TP 2.5i
 .I /run/run.statd.pid
