=== modified file 'gtk/gtk.symbols'
Index: gtk+-2.23.90/gtk/gtk.symbols
===================================================================
--- gtk+-2.23.90.orig/gtk/gtk.symbols	2011-01-07 12:30:26.293891001 +1100
+++ gtk+-2.23.90/gtk/gtk.symbols	2011-01-07 12:30:28.913891001 +1100
@@ -5327,6 +5327,7 @@
 gtk_widget_get_realized
 gtk_widget_set_mapped
 gtk_widget_get_mapped
+ubuntu_gtk_widget_set_has_grab
 #endif
 #endif
 
Index: gtk+-2.23.90/gtk/gtkwidget.c
===================================================================
--- gtk+-2.23.90.orig/gtk/gtkwidget.c	2011-01-07 12:30:26.303891001 +1100
+++ gtk+-2.23.90/gtk/gtkwidget.c	2011-01-07 12:30:28.913891001 +1100
@@ -11412,5 +11412,15 @@
   return res;
 }
 
+void
+ubuntu_gtk_widget_set_has_grab (GtkWidget *widget,
+                                gboolean   has_grab)
+{
+  if (has_grab)
+    GTK_OBJECT_FLAGS (widget) |= GTK_HAS_GRAB;
+  else
+    GTK_OBJECT_FLAGS (widget) &= ~(GTK_HAS_GRAB);
+}
+
 #define __GTK_WIDGET_C__
 #include "gtkaliasdef.c"
Index: gtk+-2.23.90/gtk/gtkwidget.h
===================================================================
--- gtk+-2.23.90.orig/gtk/gtkwidget.h	2011-01-07 12:30:26.313891001 +1100
+++ gtk+-2.23.90/gtk/gtkwidget.h	2011-01-07 12:30:28.913891001 +1100
@@ -1346,6 +1346,9 @@
 						       GtkWidget *toplevel,
 						       gpointer   user_data);
 
+void   ubuntu_gtk_widget_set_has_grab  (GtkWidget *widget,
+                                        gboolean   has_grab);
+
 G_END_DECLS
 
 #endif /* __GTK_WIDGET_H__ */
