diff -Nuar a/src/configuration.cpp b/src/configuration.cpp
--- a/src/configuration.cpp	2013-01-15 01:54:07.000000000 +0200
+++ b/src/configuration.cpp	2013-10-04 09:18:55.750776480 +0300
@@ -123,12 +123,12 @@
 	// this should be enabled on all systems by default
 	war_SetVsync(ini.value("vsync", true).toBool());
 	// 640x480 is minimum that we will support
-	int width = ini.value("width", 640).toInt();
-	int height = ini.value("height", 480).toInt();
-	if (width < 640 || height < 480)	// sanity check
+	int width = ini.value("width", 1024).toInt();
+	int height = ini.value("height", 768).toInt();
+	if (width < 1024 || height < 768)	// sanity check
 	{
-		width = 640;
-		height = 480;
+		width = 1024;
+		height = 768;
 	}
 	pie_SetVideoBufferWidth(width);
 	pie_SetVideoBufferHeight(height);
