diff -Nur warzone2100-2.1.0-old/src/configuration.c warzone2100-2.1.0/src/configuration.c
--- warzone2100-2.1.0-old/src/configuration.c	2008-12-24 16:36:29.000000000 +0200
+++ warzone2100-2.1.0/src/configuration.c	2008-12-24 16:36:43.000000000 +0200
@@ -601,8 +601,8 @@
 	{
 		// If we have an invalid or incomplete resolution specified
 		// fall back to the defaults.
-		war_SetWidth(640);
-		war_SetHeight(480);
+		war_SetWidth(1024);
+		war_SetHeight(768);
 	}
 
 	if (getWarzoneKeyNumeric("bpp", &val))
