diff -Naur icecast-2.3.2.orig/conf/icecast.xml.in icecast-2.3.2/conf/icecast.xml.in
--- icecast-2.3.2.orig/conf/icecast.xml.in	2008-04-04 18:14:10.000000000 +0300
+++ icecast-2.3.2/conf/icecast.xml.in	2008-09-28 01:01:51.000000000 +0300
@@ -7,8 +7,8 @@
         <client-timeout>30</client-timeout>
         <header-timeout>15</header-timeout>
         <source-timeout>10</source-timeout>
-        <!-- If enabled, this will provide a burst of data when a client 
-             first connects, thereby significantly reducing the startup 
+        <!-- If enabled, this will provide a burst of data when a client
+             first connects, thereby significantly reducing the startup
              time for listeners that do substantial buffering. However,
              it also significantly increases latency between the source
              client and listening client.  For low-latency setups, you
@@ -102,7 +102,7 @@
         <hidden>1</hidden>
         <no-yp>1</no-yp>
         <authentication type="htpasswd">
-                <option name="filename" value="myauth"/>
+                <option name="filename" value="/usr/share/icecast/myauth"/>
                 <option name="allow_duplicate_users" value="0"/>
         </authentication>
         <on-connect>/home/icecast/bin/stream-start</on-connect>
@@ -129,10 +129,10 @@
 
         <!-- Note that if <chroot> is turned on below, these paths must both
              be relative to the new root, not the original root -->
-        <logdir>@localstatedir@/log/@PACKAGE@</logdir>
+        <logdir>/var/log/icecast</logdir>
         <webroot>@pkgdatadir@/web</webroot>
         <adminroot>@pkgdatadir@/admin</adminroot>
-        <!-- <pidfile>@pkgdatadir@/icecast.pid</pidfile> -->
+        <pidfile>/run/icecast/icecast.pid</pidfile>
 
         <!-- Aliases: treat requests for 'source' path as being for 'dest' path
              May be made specific to a port or bound address using the "port"
@@ -164,11 +164,10 @@
 
     <security>
         <chroot>0</chroot>
-        <!--
+        <!--test for logrotate soon-->
         <changeowner>
-            <user>nobody</user>
-            <group>nogroup</group>
+            <user>icecast</user>
+            <group>icecast</group>
         </changeowner>
-        -->
     </security>
 </icecast>
