diff -Nuar a/src/freetuxtv-tv-channels-list.c b/src/freetuxtv-tv-channels-list.c
--- a/src/freetuxtv-tv-channels-list.c	2012-05-08 20:33:43.000000000 +0300
+++ b/src/freetuxtv-tv-channels-list.c	2013-07-28 17:25:56.812658147 +0300
@@ -134,12 +134,12 @@
 
 	if(!bFound){
 		if(bNoneIcon){
-			szImgFile = g_build_filename(szUserImgChannelsDir, "_none.png", NULL);	
+			szImgFile = g_build_filename(szUserImgChannelsDir, "default.png", NULL);	
 			if(g_file_test(szImgFile, G_FILE_TEST_EXISTS)){
 				szImgFile = NULL;
 			}else{
 				g_free(szImgFile);
-				szImgFile = g_build_filename(app->paths.datadir, "images", "channels", "_none.png", NULL);
+				szImgFile = g_build_filename(app->paths.datadir, "images", "channels", "default.png", NULL);
 			}
 		}
 	}
