diff -Nuar a/data/channels_groups.xml.in b/data/channels_groups.xml.in
--- a/data/channels_groups.xml.in	2012-05-08 20:33:45.000000000 +0300
+++ b/data/channels_groups.xml.in	2013-07-27 07:46:57.852618214 +0300
@@ -108,7 +108,7 @@
 		<channels_group _name="Web TV (Thai)" uri="http://database.freetuxtv.net/playlists/playlist_webtv_th.m3u" />
 	</language>
 	<language id="tr" _lang="Turkish Web TV and Radio">
-		<channels_group _name="Web TV (Turkish)" uri="http://database.freetuxtv.net/playlists/playlist_webtv_tr.m3u" />
+		<channels_group _name="Web TV (Turkish)" uri="https://github.com/circass/notobs/raw/master/playlist_webtv_tr.m3u" />
 	</language>
 	<language id="none" _lang="No lang Web TV and Radio">
 		<channels_group _name="Web TV (No lang)" uri="http://database.freetuxtv.net/playlists/playlist_webtv_none.m3u" />
diff -Nuar a/data/images/channels/.directory b/data/images/channels/.directory
--- a/data/images/channels/.directory	1970-01-01 02:00:00.000000000 +0200
+++ b/data/images/channels/.directory	2013-07-27 07:48:00.794616035 +0300
@@ -0,0 +1,4 @@
+[Dolphin]
+PreviewsShown=true
+Timestamp=2013,7,27,7,48,0
+Version=3
diff -Nuar a/data/images/channels/Makefile.am b/data/images/channels/Makefile.am
--- a/data/images/channels/Makefile.am	2012-05-08 20:33:45.000000000 +0300
+++ b/data/images/channels/Makefile.am	2013-07-27 07:52:23.894606925 +0300
@@ -68,6 +68,7 @@
   canalsavoir.png \
   canaltv.png \
   cap24-paris.png \
+  capultv.png \
   cashtv.png \
   cctv1.png \
   cctv2.png \
diff -Nuar a/data/images/channels/Makefile.in b/data/images/channels/Makefile.in
--- a/data/images/channels/Makefile.in	2012-05-08 20:34:05.000000000 +0300
+++ b/data/images/channels/Makefile.in	2013-07-27 07:53:07.035605432 +0300
@@ -317,6 +317,7 @@
   canalsavoir.png \
   canaltv.png \
   cap24-paris.png \
+  capultv.png \
   cashtv.png \
   cctv1.png \
   cctv2.png \
diff -Nuar a/data/tv_channels.xml b/data/tv_channels.xml
--- a/data/tv_channels.xml	2012-05-08 20:33:45.000000000 +0300
+++ b/data/tv_channels.xml	2013-07-27 07:57:34.273596179 +0300
@@ -208,6 +208,9 @@
 	<tvchannel name="CAP 24">
 		<logo_filename>cap24-paris.png</logo_filename>
 	</tvchannel>
+	<tvchannel name="ÇAPUL TV">
+		<logo_filename>capultv.png</logo_filename>
+	</tvchannel>
 	<tvchannel name="Cash TV">
 		<logo_filename>cashtv.png</logo_filename>
 	</tvchannel>
