Index: phonon-4.4.2/CMakeLists.txt
===================================================================
--- phonon-4.4.2.orig/CMakeLists.txt
+++ phonon-4.4.2/CMakeLists.txt
@@ -267,10 +267,10 @@ if (Q_WS_WIN)
    add_subdirectory(ds9)
 endif (Q_WS_WIN)
 
-if (Q_WS_X11)
-   add_subdirectory(gstreamer)
-   add_subdirectory(xine)
-endif (Q_WS_X11)
+#if (Q_WS_X11)
+#   add_subdirectory(gstreamer)
+#   add_subdirectory(xine)
+#endif (Q_WS_X11)
 
 if(NOT WIN32) # pkgconfig file
    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/phonon.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/phonon.pc @ONLY)
