diff -up phonon-4.6.0/qt_phonon.pri.me phonon-4.6.0/qt_phonon.pri
--- phonon-4.6.0/qt_phonon.pri.me	2012-03-28 18:22:06.623520519 +0200
+++ phonon-4.6.0/qt_phonon.pri	2012-03-28 18:22:15.886519552 +0200
@@ -3,6 +3,6 @@
 # the next build of Qt. So pretend we're not there in this case.
 # This file is loaded by qt_config.prf, before .qmake.cache has been loaded.
 # Consequently, we have to do some stunts to get values out of the cache.
-if(!exists($$_QMAKE_CACHE_)| \
-   !contains($$list($$fromfile($$_QMAKE_CACHE_, CONFIG)), QTDIR_build)): \
+!exists($$_QMAKE_CACHE_)| \
+   !contains($$list($$fromfile($$_QMAKE_CACHE_, CONFIG)), QTDIR_build): \
     QT_CONFIG += phonon
