diff -up phonon-4.4.4/cmake/FindPhononInternal.cmake.no_rpath phonon-4.4.4/cmake/FindPhononInternal.cmake
--- phonon-4.4.4/cmake/FindPhononInternal.cmake.no_rpath	2011-01-19 13:20:00.000000000 -0600
+++ phonon-4.4.4/cmake/FindPhononInternal.cmake	2011-01-21 09:02:08.242131811 -0600
@@ -232,8 +232,8 @@ endif(APPLE)
 # These two options below make it set the RPATH of the installed targets to all
 # RPATH directories outside the current CMAKE_BINARY_DIR and also the library 
 # install directory. Alex
-set(CMAKE_INSTALL_RPATH_USE_LINK_PATH  TRUE)
-set(CMAKE_INSTALL_RPATH "${LIB_INSTALL_DIR}")
+#set(CMAKE_INSTALL_RPATH_USE_LINK_PATH  TRUE)
+#set(CMAKE_INSTALL_RPATH "${LIB_INSTALL_DIR}")
 
 
 # Uninstall Target
