diff -Nur mkvtoolnix-4.2.0-old//share/desktop/mkvinfo.desktop mkvtoolnix-4.2.0/share/desktop/mkvinfo.desktop
--- mkvtoolnix-4.2.0-old//share/desktop/mkvinfo.desktop	2010-09-05 03:01:41.178000113 +0300
+++ mkvtoolnix-4.2.0/share/desktop/mkvinfo.desktop	2010-09-05 03:03:35.821002914 +0300
@@ -4,6 +4,7 @@
 GenericName=Show information about MKV files
 GenericName[de]=Informationen über MKV-Dateien anzeigen
 GenericName[es]=Mostrar información sobre ficheros MKV
+GenericName[tr]=MKV dosyaları için bilgi göster
 Icon=mkvinfo
 Exec=mkvinfo -g
 Terminal=false
diff -Nur mkvtoolnix-4.2.0-old//share/desktop/mkvmergeGUI.desktop mkvtoolnix-4.2.0/share/desktop/mkvmergeGUI.desktop
--- mkvtoolnix-4.2.0-old//share/desktop/mkvmergeGUI.desktop	2010-09-05 03:01:41.178000113 +0300
+++ mkvtoolnix-4.2.0/share/desktop/mkvmergeGUI.desktop	2010-09-05 03:03:30.996999988 +0300
@@ -5,6 +5,7 @@
 GenericName[de]=MKV-Dateien erstellen
 GenericName[es]=Creador de ficheros MKV
 GenericName[fr]=Créateur de fichiers MKV
+GenericName[tr]=MKV dosyası üretici
 Icon=mkvmergeGUI
 Exec=mmg
 Terminal=false
