--- a/source3/VERSION.orig	2008-10-04 13:57:11.000000000 +0300
+++ b/source3/VERSION	2008-10-04 13:57:21.000000000 +0300
@@ -95,5 +95,5 @@
 # e.g. SAMBA_VERSION_VENDOR_SUFFIX=vendor_version()    #
 #  ->  "CVS 3.0.0rc2-VendorVersion"                    #
 ########################################################
-SAMBA_VERSION_VENDOR_SUFFIX=
+SAMBA_VERSION_VENDOR_SUFFIX="Pardus"
 SAMBA_VERSION_VENDOR_PATCH=
