Index: samba-3.5.8/lib/util/params.c
===================================================================
--- samba-3.5.8.orig/lib/util/params.c
+++ samba-3.5.8/lib/util/params.c
@@ -508,7 +508,7 @@ static myFILE *OpenConfFile( const char
   ret->buf = file_load(FileName, &ret->size, 0, ret);
   if( NULL == ret->buf )
     {
-    DEBUG( 1,
+    DEBUG( 2,
       ("%s Unable to open configuration file \"%s\":\n\t%s\n",
       func, FileName, strerror(errno)) );
     talloc_free(ret);
