Index: samba-3.4.3/source3/Makefile.in
===================================================================
--- samba-3.4.3.orig/source3/Makefile.in
+++ samba-3.4.3/source3/Makefile.in
@@ -202,9 +202,8 @@ ROOT_SBIN_PROGS = @CIFSMOUNT_PROGS@
 
 BIN_PROGS1 = bin/smbclient@EXEEXT@ bin/net@EXEEXT@ bin/smbspool@EXEEXT@ \
 	bin/testparm@EXEEXT@ bin/smbstatus@EXEEXT@ bin/smbget@EXEEXT@
-BIN_PROGS2 = bin/smbcontrol@EXEEXT@ bin/smbtree@EXEEXT@ bin/tdbbackup@EXEEXT@ \
-	bin/nmblookup@EXEEXT@ bin/pdbedit@EXEEXT@ bin/tdbdump@EXEEXT@ \
-	bin/tdbtool@EXEEXT@
+BIN_PROGS2 = bin/smbcontrol@EXEEXT@ bin/smbtree@EXEEXT@ \
+	bin/nmblookup@EXEEXT@ bin/pdbedit@EXEEXT@
 BIN_PROGS3 = bin/smbpasswd@EXEEXT@ bin/rpcclient@EXEEXT@ bin/smbcacls@EXEEXT@ \
 	bin/profiles@EXEEXT@ bin/ntlm_auth@EXEEXT@ bin/sharesec@EXEEXT@ \
 	bin/smbcquotas@EXEEXT@ bin/eventlogadm@EXEEXT@
@@ -1824,9 +1823,9 @@ uninstalllibtdb::
 	-rm -f $(DESTDIR)${prefix}/include/tdb.h
 
 libs:: libtdb
-installlibs:: @INSTALL_LIBTDB@
-uninstalllibs:: @UNINSTALL_LIBTDB@
-cleanlibs:: cleanlibtdb
+installlibs::
+uninstalllibs::
+cleanlibs::
 shlibs test_shlibs: @LIBTDB_SHARED@
 
 showlibtdb:
