--- src/playlist/soundcloud_playlist_plugin.c.orig	2012-08-01 05:04:32.000000000 +0300
+++ src/playlist/soundcloud_playlist_plugin.c	2012-08-27 23:33:25.000000000 +0300
@@ -288,7 +288,6 @@
 
 		if (done) {
 #ifdef HAVE_YAJL1
-			stat = yajl_parse_complete(hand);
 #else
 			stat = yajl_complete_parse(hand);
 #endif
@@ -297,7 +296,6 @@
 
 		if (stat != yajl_status_ok
 #ifdef HAVE_YAJL1
-		    && stat != yajl_status_insufficient_data
 #endif
 		    )
 		{
@@ -381,7 +379,7 @@
 	data.title = NULL;
 	data.stream_url = NULL;
 #ifdef HAVE_YAJL1
-	hand = yajl_alloc(&parse_callbacks, NULL, NULL, (void *) &data);
+
 #else
 	hand = yajl_alloc(&parse_callbacks, NULL, (void *) &data);
 #endif
