diff -Nur google-perftools-1.6-old//Makefile.am google-perftools-1.6/Makefile.am
--- google-perftools-1.6-old//Makefile.am	2011-01-31 14:07:18.025000015 +0200
+++ google-perftools-1.6/Makefile.am	2011-01-31 14:07:52.391999991 +0200
@@ -99,7 +99,7 @@
 nodist_googleinclude_HEADERS = src/google/tcmalloc.h
 noinst_HEADERS = src/google/tcmalloc.h.in
 
-docdir = $(prefix)/share/doc/$(PACKAGE)-$(VERSION)
+docdir = $(prefix)/share/doc/$(PACKAGE)
 # This is for HTML and other documentation you want to install.
 # Add your documentation files (in doc/) in addition to these
 # top-level boilerplate files.  Also add a TODO file if you have one.
