--- Makefile.am	2010-01-19 20:43:25.000000000 +0100
+++ Makefile.am.oden	2010-02-16 13:07:22.078970698 +0100
@@ -355,11 +355,7 @@
                               src/base/thread_annotations.h \
                               src/malloc_hook-inl.h \
                               src/maybe_threads.h
-SG_TCMALLOC_MINIMAL_INCLUDES = src/google/malloc_hook.h \
-                               src/google/malloc_hook_c.h \
-                               src/google/malloc_extension.h \
-                               src/google/malloc_extension_c.h \
-                               src/google/stacktrace.h
+
 TCMALLOC_MINIMAL_INCLUDES = $(S_TCMALLOC_MINIMAL_INCLUDES) $(SG_TCMALLOC_MINIMAL_INCLUDES)
 googleinclude_HEADERS += $(SG_TCMALLOC_MINIMAL_INCLUDES)
 
@@ -1027,8 +1023,7 @@
                           src/base/sysinfo.h \
                           $(SPINLOCK_INCLUDES) \
                           $(LOGGING_INCLUDES)
-SG_CPU_PROFILER_INCLUDES = src/google/profiler.h \
-                           src/google/stacktrace.h
+SG_CPU_PROFILER_INCLUDES = src/google/profiler.h
 CPU_PROFILER_INCLUDES = $(S_CPU_PROFILER_INCLUDES) $(SG_CPU_PROFILER_INCLUDES)
 googleinclude_HEADERS += $(SG_CPU_PROFILER_INCLUDES)
 
