diff -up google-perftools-1.7/src/google/malloc_hook_c.h.BAD google-perftools-1.7/src/google/malloc_hook_c.h
--- google-perftools-1.7/src/google/malloc_hook_c.h.BAD	2011-02-18 13:16:03.389422008 -0500
+++ google-perftools-1.7/src/google/malloc_hook_c.h	2011-02-18 13:16:12.825421817 -0500
@@ -40,7 +40,7 @@
 #include <stddef.h>
 #include <sys/types.h>
 
-// Annoying stuff for windows -- makes sure clients can import these functions
+/* Annoying stuff for windows -- makes sure clients can import these functions */
 #ifndef PERFTOOLS_DLL_DECL
 # ifdef _WIN32
 #   define PERFTOOLS_DLL_DECL  __declspec(dllimport)
