diff -Naur radiotray-0.6.orig/data/radiotray.desktop radiotray-0.6/data/radiotray.desktop
--- radiotray-0.6.orig/data/radiotray.desktop	2010-06-16 23:35:24.067541639 +0300
+++ radiotray-0.6/data/radiotray.desktop	2010-06-16 23:37:04.547541889 +0300
@@ -1,5 +1,9 @@
 [Desktop Entry]
 Name=Radio Tray
+GenericName=Streaming Radio Player
+GenericName[tr]=Çevrimiçi Radyo Çalıcı
+Comment=A simple yet powerful streaming music player
+Comment[tr]=Basit ve güçlü bir çevrimiçi rayo çalıcısı
 Terminal=false
 Exec=radiotray
 Icon=radiotray
