--- tcsh-6.14.00/tw.parse.c.old	2007-04-24 16:46:32.000000000 +0200
+++ tcsh-6.14.00/tw.parse.c	2007-04-24 17:02:28.000000000 +0200
@@ -2065,7 +2065,8 @@
 	     (Strchr(val, 'x') != NULL);
 
     for (i = 0; i < count; i++)	{ /* find widest string */
-	maxwidth = max(maxwidth, (unsigned int) NLSStringWidth(items[i]));
+        if (NLSStringWidth(items[i]) != -1)
+	    maxwidth = max(maxwidth, (unsigned int) NLSStringWidth(items[i]));
     }
 
     maxwidth += no_file_suffix ? 1 : 2;	/* for the file tag and space */
