--- tcsh-6.15.00/sh.h
+++ tcsh-6.15.00/sh.h
@@ -222,6 +222,8 @@ typedef int eChar;
 # define DEV_DEV_COMPARE(x,y)   ((x) == (y))
 #endif /* DEV_DEV_COMPARE */
 
+#define POSIX
+
 #ifdef _SEQUENT_
 # include <sys/procstats.h>
 #endif /* _SEQUENT_ */
@@ -267,12 +269,10 @@ typedef int eChar;
 #  else
 #   include <termio.h>
 #  endif /* _UWIN */
-#  if SYSVREL > 3
 #   undef TIOCGLTC	/* we don't need those, since POSIX has them */
 #   undef TIOCSLTC
 #   undef CSWTCH
 #   define CSWTCH _POSIX_VDISABLE	/* So job control works */
-#  endif /* SYSVREL > 3 */
 # endif /* POSIX */
 #endif /* WINNT_NATIVE */
 
@@ -321,6 +321,8 @@ typedef int eChar;
 #if !((defined(SUNOS4) || defined(_MINIX) /* || defined(DECOSF1) */) && defined(TERMIO))
 # if !defined(_VMS_POSIX) && !defined(WINNT_NATIVE)
 #  include <sys/ioctl.h>
+#  undef TIOCGLTC	/* we don't need those, since POSIX has them */
+#  undef TIOCSLTC
 # endif
 #endif 
 
