diff -Naur arora-0.8.0.orig/src/browsermainwindow.cpp arora-0.8.0/src/browsermainwindow.cpp
--- arora-0.8.0.orig/src/browsermainwindow.cpp	2009-07-20 23:22:39.786451590 +0300
+++ arora-0.8.0/src/browsermainwindow.cpp	2009-07-20 23:28:23.485326774 +0300
@@ -1328,7 +1328,7 @@
 {
     QSettings settings;
     settings.beginGroup(QLatin1String("MainWindow"));
-    QString home = settings.value(QLatin1String("home"), QLatin1String("about:home")).toString();
+    QString home = settings.value(QLatin1String("home"), QLatin1String("http://pardus.org.tr")).toString();
     tabWidget()->loadString(home);
 }
 
diff -Naur arora-0.8.0.orig/src/settings.cpp arora-0.8.0/src/settings.cpp
--- arora-0.8.0.orig/src/settings.cpp	2009-07-20 23:22:39.780452153 +0300
+++ arora-0.8.0/src/settings.cpp	2009-07-20 23:29:01.923326728 +0300
@@ -135,7 +135,7 @@
 {
     QSettings settings;
     settings.beginGroup(QLatin1String("MainWindow"));
-    QString defaultHome = QLatin1String("about:home");
+    QString defaultHome = QLatin1String("http://pardus-anka.org");
     homeLineEdit->setText(settings.value(QLatin1String("home"), defaultHome).toString());
     startupBehavior->setCurrentIndex(settings.value(QLatin1String("startupBehavior"), 0).toInt());
     settings.endGroup();
