diff -Naur arora-0.10.0.orig/src/data/defaultbookmarks.xbel arora-0.10.0/src/data/defaultbookmarks.xbel
--- arora-0.10.0.orig/src/data/defaultbookmarks.xbel	2009-10-01 18:53:12.442307919 +0300
+++ arora-0.10.0/src/data/defaultbookmarks.xbel	2009-10-01 19:00:47.235435498 +0300
@@ -3,11 +3,29 @@
 <xbel version="1.0">
     <folder folded="yes">
         <title>Bookmarks Bar</title>
-        <bookmark href="http://arora-browser.org/">
-            <title>Arora</title>
+        <bookmark href="http://www.pisilinux.org">
+            <title>PisiLinux.Org</title>
+        </bookmark>
+        <bookmark href="http://team.pisilinux.org">
+            <title>PisiLinux Team</title>
+        </bookmark>
+        <bookmark href="http://bugs.pisilinux.org">
+            <title>PisiLinux Hata Takip</title>
+        </bookmark>
+        <bookmark href="https://forum.pisilinux.org">
+            <title>Pisi Forum</title>
+        </bookmark>
+        <bookmark href="http://wiki.pisilinux.org">
+            <title>Pisi Wiki</title>
         </bookmark>
     </folder>
     <folder folded="yes">
         <title>Bookmarks Menu</title>
+        <bookmark href="http://arora-browser.org/">
+            <title>Arora</title>
+        </bookmark>
     </folder>
 </xbel>
