Index: R-2.11.1/src/nmath/standalone/Makefile.in
===================================================================
--- R-2.11.1.orig/src/nmath/standalone/Makefile.in
+++ R-2.11.1/src/nmath/standalone/Makefile.in
@@ -64,7 +64,7 @@ libRmath_la = libRmath$(DYLIB_EXT)
 Rexeclibdir_LTLIBRARIES = $(libRmath_la)
 libRmath_la_SOURCES = $(SOURCES)
 libRmath_la_OBJECTS = $(OBJECTS:.o=.lo)
-libRmath_la_LDFLAGS =
+libRmath_la_LDFLAGS = -lm
 
 CLEANFILES = Makedeps *.d *.o *.lo test $(SOURCES)
 DISTCLEANFILES = Makefile $(Rexeclibdir_LIBRARIES) $(Rexeclibdir_LTLIBRARIES)
@@ -116,7 +116,7 @@ libRmath.a: $(libRmath_a_OBJECTS)
 	$(RANLIB) $@
 
 $(libRmath_la): $(libRmath_la_OBJECTS)
-	$(DYLIB_LINK) -o $@ $(libRmath_la_LDFLAGS) $(libRmath_la_OBJECTS)
+	$(DYLIB_LINK) -o $@ $(libRmath_la_OBJECTS) $(libRmath_la_LDFLAGS)
 
 test: $(srcdir)/test.c
 	$(CC) -o $@ $(ALL_CPPFLAGS) $(ALL_CFLAGS) $(srcdir)/test.c \
