diff -Nuar R-2.9.0.orig/src/modules/lapack/Makefile.in R-2.9.0/src/modules/lapack/Makefile.in
--- R-2.9.0.orig/src/modules/lapack/Makefile.in	2009-01-21 15:01:22.000000000 +0200
+++ R-2.9.0/src/modules/lapack/Makefile.in	2009-05-07 09:01:46.000000000 +0300
@@ -37,10 +37,10 @@
 lapack_la_OBJECTS = $(OBJECTS)
 ## Used for passing '+s' to SHLIB_LINK on HP-UX.
 lapack_la_LDFLAGS = @LAPACK_LDFLAGS@
-lapack_la_LIBADD = $(LIBR) @DYLIB_UNDEFINED_ALLOWED_FALSE@ $(LIBINTL)
+lapack_la_LIBADD = $(LIBR) @DYLIB_UNDEFINED_ALLOWED_FALSE@ $(LIBINTL) -lgfortran
 
 Rlapack_la_OBJECTS = $(LIBOBJECTS)
-Rlapack_la_LIBADD = @DYLIB_UNDEFINED_ALLOWED_FALSE@ $(FLIBS) $(LIBR)
+Rlapack_la_LIBADD = @DYLIB_UNDEFINED_ALLOWED_FALSE@ $(FLIBS) $(LIBR) -lgfortran
 
 ALL_CFLAGS = $(ALL_CFLAGS_LO)
 ALL_FFLAGS = $(ALL_FFLAGS_LO)
