diff -Nuar R-2.7.1.orig/src/unix/Makefile.in R-2.7.1/src/unix/Makefile.in
--- R-2.7.1.orig/src/unix/Makefile.in	2008-06-23 12:38:27.000000000 +0300
+++ R-2.7.1/src/unix/Makefile.in	2008-06-24 16:40:23.000000000 +0300
@@ -66,7 +66,7 @@
 
 ## we need to build at install time to capture the correct 'rhome'
 install-Rscript:
-	$(CC) $(ALL_CPPFLAGS) $(ALL_CFLAGS) -DR_HOME='"$(rhome)"' -o Rscript \
+	$(CC) $(ALL_CPPFLAGS) $(ALL_CFLAGS) -DR_HOME='"/usr/lib/R/"' -o Rscript \
 	  $(srcdir)/Rscript.c
 	@$(MKINSTALLDIRS) "$(DESTDIR)$(Rexecbindir)"
 	@$(INSTALL_PROGRAM) Rscript "$(DESTDIR)$(Rexecbindir)/Rscript"
