diff -Nur vdrift-08-05-08-old/bullet-2.66/mk/jam/variant.jam vdrift-08-05-08/bullet-2.66/mk/jam/variant.jam
--- vdrift-08-05-08-old/bullet-2.66/mk/jam/variant.jam	2008-09-25 14:16:16.097292503 +0300
+++ vdrift-08-05-08/bullet-2.66/mk/jam/variant.jam	2008-09-25 14:16:33.579043633 +0300
@@ -30,7 +30,7 @@
 
 # Set modus related flags
 CCFLAGS += $(COMPILER.CFLAGS) $(COMPILER.CFLAGS.$(VARIANT)) ;
-C++FLAGS += $(COMPILER.CFLAGS) $(COMPILER.C++FLAGS)
+C++FLAGS +=  $(COMPILER.C++FLAGS)
 	    $(COMPILER.CFLAGS.$(VARIANT)) $(COMPILER.C++FLAGS.$(VARIANT)) ;
 LINKLIBS += $(COMPILER.LFLAGS) $(COMPILER.LFLAGS.$(VARIANT)) ;
 LOCATE.OBJECTS = $(LOCATE.OBJECTS)/$(VARIANT) ;
