diff -Nur vdrift-2010-06-30-old//src/texture.cpp vdrift-2010-06-30/src/texture.cpp
--- vdrift-2010-06-30-old//src/texture.cpp	2011-02-09 20:19:16.378000022 +0200
+++ vdrift-2010-06-30/src/texture.cpp	2011-02-09 20:20:26.573999909 +0200
@@ -278,7 +278,7 @@
 		glTexParameteri(GL_TEXTURE_CUBE_MAP,GL_TEXTURE_MIN_FILTER,GL_LINEAR_MIPMAP_LINEAR);
 		glTexParameteri(GL_TEXTURE_CUBE_MAP,GL_TEXTURE_MAG_FILTER,GL_LINEAR);
 		
-		glGenerateMipmap(GL_TEXTURE_CUBE_MAP);
+		glGenerateMipmapEXT(GL_TEXTURE_CUBE_MAP);
 	}
 
 	loaded = true;
