Index: vdrift-2010-06-30/include/fbobject.h
===================================================================
--- vdrift-2010-06-30.orig/include/fbobject.h
+++ vdrift-2010-06-30/include/fbobject.h
@@ -6,6 +6,7 @@
 #include <OpenGL/gl.h>
 #else
 #include <GL/glew.h>
+#define GL_FRAMEBUFFER_INCOMPLETE_LAYER_TARGETS 0x8DA8
 #include <GL/gl.h>
 #endif
 
Index: vdrift-2010-06-30/include/fbtexture.h
===================================================================
--- vdrift-2010-06-30.orig/include/fbtexture.h
+++ vdrift-2010-06-30/include/fbtexture.h
@@ -6,6 +6,7 @@
 #include <OpenGL/gl.h>
 #else
 #include <GL/glew.h>
+#define GL_TEXTURE_RECTANGLE 0x84F5
 #include <GL/gl.h>
 #endif
 
