diff -Naur e-module-news-0.0.1_55225.orig//src/news_menu.c e-module-news-0.0.1_55225/src/news_menu.c
--- e-module-news-0.0.1_55225.orig//src/news_menu.c	2010-10-03 14:34:00.000000000 +0300
+++ e-module-news-0.0.1_55225/src/news_menu.c	2011-03-11 14:57:31.291000032 +0200
@@ -61,10 +61,10 @@
      e_menu_item_icon_file_set(mi, f->icon);
    e_menu_item_callback_set(mi, _cb_configure_feed, f);
 
-   mi = e_menu_item_new(mn);
+   mi = e_menu_item_new(ma);
    e_menu_item_separator_set(mi, 1);
 
-   _menu_append(mn, f->item);
+   _menu_append(ma, f->item);
 
 
    return 1;
@@ -166,7 +166,7 @@
    e_util_menu_item_theme_icon_set(mi, "preferences-system");
    e_menu_item_submenu_set(mi, mn_conf);
 
-   e_gadcon_client_util_menu_items_append(ni->gcc, mn, 0);
+   e_gadcon_client_util_menu_items_append(ni->gcc, mn,mn_conf, 0);
 }
 
 static void
