diff -Nuar a/src/fbobject.h b/src/fbobject.h
--- a/src/fbobject.h	2012-07-21 18:03:21.000000000 +0300
+++ b/src/fbobject.h	2013-09-26 06:42:05.750499055 +0300
@@ -9,6 +9,7 @@
 #include <OpenGL/gl.h>
 #else
 #include <GL/glew.h>
+#define GL_FRAMEBUFFER_INCOMPLETE_LAYER_TARGETS 0x8DA8
 #include <GL/gl.h>
 #endif
 
diff -Nuar a/src/fbtexture.h b/src/fbtexture.h
--- a/src/fbtexture.h	2012-07-21 18:03:21.000000000 +0300
+++ b/src/fbtexture.h	2013-09-26 06:43:00.313497166 +0300
@@ -9,6 +9,7 @@
 #include <OpenGL/gl.h>
 #else
 #include <GL/glew.h>
+#define GL_TEXTURE_RECTANGLE 0x84F5
 #include <GL/gl.h>
 #endif
 
