--- SConstruct~	2010-07-01 16:07:10.000000000 -0500
+++ SConstruct	2010-07-01 16:15:43.529291853 -0500
@@ -138,7 +138,7 @@
         CPPPATH = ['#include', '#bullet'],
         CCFLAGS = ['-Wall', '-Wextra', '-Wno-unused-parameter', '-pthread'],
         LIBPATH = ['.', '#lib'],
-        LINKFLAGS = ['-pthread'],
+        LINKFLAGS = ['-pthread','-lGLU','-lGL'],
         CC = 'gcc', CXX = 'g++',
         options = opts)
     check_headers = ['asio.hpp', 'boost/bind.hpp', 'GL/gl.h', 'GL/glu.h', 'SDL/SDL.h', 'SDL/SDL_image.h', 'SDL/SDL_rotozoom.h', 'vorbis/vorbisfile.h', 'GL/glew.h']
