--- gst-plugins-base-0.10.15/ext/cdparanoia/gstcdparanoiasrc.c_orig	2008-02-02 21:22:47.000000000 +0100
+++ gst-plugins-base-0.10.15/ext/cdparanoia/gstcdparanoiasrc.c	2008-02-02 21:23:23.000000000 +0100
@@ -46,7 +46,7 @@
   PROP_GENERIC_DEVICE
 };
 
-#define DEFAULT_READ_SPEED              -1
+#define DEFAULT_READ_SPEED               4 /* avoid mad spinning ad noise */
 #define DEFAULT_SEARCH_OVERLAP          -1
 #define DEFAULT_PARANOIA_MODE            PARANOIA_MODE_FRAGMENT
 #define DEFAULT_GENERIC_DEVICE           NULL
