--- pkg-config-0.21/parse.c.compat-loop	2007-01-29 16:31:15.000000000 -0500
+++ pkg-config-0.21/parse.c	2007-01-29 16:32:08.000000000 -0500
@@ -1186,61 +1186,7 @@
 
   pkg->path_position = G_MAXINT;
   
-  if (strcmp (name, "glib") == 0)
-    {
-      char *output;
-
-      debug_spew ("Calling glib-config\n");
-      
-      pkg->version = backticks ("glib-config --version");
-      if (pkg->version == NULL)
-        {
-          g_free (pkg);
-          return NULL;
-        }
-      
-      pkg->name = g_strdup ("GLib");
-      pkg->key = g_strdup ("glib");
-      pkg->description = g_strdup ("C Utility Library");
-
-      output = backticks ("glib-config --libs");
-      parse_libs (pkg, output, "glib-config");
-      g_free (output);
-
-      output = backticks ("glib-config --cflags");
-      parse_cflags (pkg, output, "glib-config");
-      g_free (output);
-
-      return pkg;
-    }
-  else if (strcmp (name, "gtk+") == 0)
-    {
-      char *output;
-
-      debug_spew ("Calling gtk-config\n");
-      
-      pkg->version = backticks ("gtk-config --version");
-      if (pkg->version == NULL)
-        {
-          g_free (pkg);
-          return NULL;
-        }
-      
-      pkg->name = g_strdup ("GTK+");
-      pkg->key = g_strdup ("gtk+");
-      pkg->description = g_strdup ("GIMP Tool Kit");
-
-      output = backticks ("gtk-config --libs");
-      parse_libs (pkg, output, "gtk-config");
-      g_free (output);
-
-      output = backticks ("gtk-config --cflags");
-      parse_cflags (pkg, output, "gtk-config");
-      g_free (output);
-
-      return pkg;
-    }
-  else if (strcmp (name, "libgnomevfs") == 0)
+  if (strcmp (name, "libgnomevfs") == 0)
     {
       char *output;
 
