--- pkg-config-0.25/main.c.biarch	2010-07-30 20:58:53.000000000 +0200
+++ pkg-config-0.25/main.c	2010-07-30 21:02:30.000000000 +0200
@@ -29,6 +29,11 @@
 #include <string.h>
 #include <ctype.h>
 #include <stdio.h>
+#if defined(__linux__)
+#include <sys/stat.h>
+#include <sys/syscall.h>
+#include <sys/personality.h>
+#endif
 
 #ifdef G_OS_WIN32
 #define STRICT
@@ -171,6 +176,17 @@
   printf("%s\n", (gchar*)key);
 }
 
+#if defined(__linux__)
+static int
+is_dir(const char *path)
+{
+  struct stat st;
+  if (stat(path, &st) < 0)
+    return 0;
+  return S_ISDIR(st.st_mode);
+}
+#endif
+
 int
 main (int argc, char **argv)
 {
@@ -207,6 +223,9 @@
   char **search_dirs;
   char **iter;
   gboolean need_newline;
+#if defined(__linux__)
+  long personality;
+#endif
   FILE *log = NULL;
   const char *pkgname;
   Package *pkg;
@@ -308,6 +327,12 @@
     }
   else
     {
+#if defined(__linux__)
+      personality = syscall(SYS_personality, 0xffffffff);
+      if ((personality & PER_MASK) == PER_LINUX32 && is_dir(LIBDIR_PATH32))
+	      add_search_dirs(PKG_CONFIG_PC_PATH32, G_SEARCHPATH_SEPARATOR_S);
+      else
+#endif
       add_search_dirs(PKG_CONFIG_PC_PATH, G_SEARCHPATH_SEPARATOR_S);
     }
 
--- pkg-config-0.25/configure.in.biarch	2010-07-30 21:02:41.000000000 +0200
+++ pkg-config-0.25/configure.in	2010-07-30 21:03:12.000000000 +0200
@@ -29,6 +29,10 @@
  AC_SUBST([pc_path])
 
 ])
+libdir_path32="$libdir/pkgconfig/32"
+AC_DEFINE_UNQUOTED(LIBDIR_PATH32, ["$libdir_path32"], [libdir pkgconfig path for 32 personality])
+pc_path32="$libdir/pkgconfig/32:$datadir/pkgconfig"
+AC_DEFINE_UNQUOTED(PKG_CONFIG_PC_PATH32, ["$pc_path32"], [path for 32 personality])
 
 PKG_CONFIG_FIND_PC_PATH
 
