diff -Nuar exaile-0.3.2.1.orig//Makefile exaile-0.3.2.1/Makefile
--- exaile-0.3.2.1.orig//Makefile	2011-02-13 17:17:38.000000000 +0200
+++ exaile-0.3.2.1/Makefile	2011-03-05 22:00:57.506000029 +0200
@@ -8,7 +8,7 @@
 
 .PHONY: dist test coverage clean sanitycheck
 
-all: compile locale manpage
+all: locale manpage
 	@echo "Ready to install..."
 
 # The no_locale stuff is by request of BSD people, please ensure
@@ -16,11 +16,6 @@
 all_no_locale: compile
 	@echo "Ready to install..."
 
-compile:
-	python -m compileall -q xl xlgui
-	-python -O -m compileall -q xl xlgui
-	$(MAKE) -C plugins compile
-
 make-install-dirs:
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
 	mkdir -p $(EXAILELIBDIR)
