diff -Nuar exaile-0.3.2.1.orig//tools/generate-launcher exaile-0.3.2.1/tools/generate-launcher
--- exaile-0.3.2.1.orig//tools/generate-launcher	2011-02-13 17:17:38.000000000 +0200
+++ exaile-0.3.2.1/tools/generate-launcher	2011-03-05 22:05:48.611000027 +0200
@@ -26,7 +26,7 @@
 EXAILE_PREFIX=$PREFIX
 
 if test -z "\$XDG_DATA_DIRS"; then
-	XDG_DATA_DIRS="/usr/local/share/:/usr/share/"
+	XDG_DATA_DIRS="/usr/share/"
 fi
 XDG_DATA_DIRS="\$EXAILE_PREFIX/share/:\$XDG_DATA_DIRS"
 export XDG_DATA_DIRS
