--- pam_pkcs11-0.6.2.orig/etc/pam_pkcs11.conf.example.in
+++ pam_pkcs11-0.6.2/etc/pam_pkcs11.conf.example.in
@@ -24,7 +24,15 @@ pam_pkcs11 {
   use_authtok = false;
 
   # Filename of the PKCS #11 module. The default value is "default"
-  use_pkcs11_module = opensc;
+  use_pkcs11_module = nss;
+
+  # NSS (Network Security Service) config
+  # Note: You should add nss_dir = "path_to_nssdb_directory" into the pkcs11_module section
+  # you want to use.
+  pkcs11_module nss {
+    nss_dir = /etc/ssl/nssdb;
+    crl_policy = none;
+  }
 
   pkcs11_module opensc {
     module = /usr/lib/opensc-pkcs11.so;
@@ -100,12 +108,6 @@ pam_pkcs11 {
     cert_policy = ca,signature;
   }
 
-  # NSS (Network Security Service) config
-  pkcs11_module nss {
-    nss_dir = /etc/ssl/nssdb;
-    crl_policy = none;
-  }
-
   # Default pkcs11 module
   pkcs11_module default {
     module = /usr/lib/pam_pkcs11/pkcs11_module.so;
