diff -ur pciutils-3.0.0.orig/lib/configure pciutils-3.0.0/lib/configure
--- pciutils-3.0.0.orig/lib/configure	2008-04-10 22:15:47.000000000 +0300
+++ pciutils-3.0.0/lib/configure	2008-04-11 12:45:41.000000000 +0300
@@ -125,10 +125,7 @@
 	echo "$ZLIB (auto-detected)"
 fi
 if [ "$ZLIB" = yes ] ; then
-	echo >>$c '#define PCI_COMPRESSED_IDS'
-	echo >>$c '#define PCI_IDS "pci.ids.gz"'
-	echo >>$m 'LIBZ=-lz'
-	echo >>$m 'WITH_LIBS+=$(LIBZ)'
+	echo >>$c '#define PCI_IDS "pci.ids"'
 else
 	echo >>$c '#define PCI_IDS "pci.ids"'
 fi
