diff -Nur linux-atm-2.5.0-old/src/arpd/arp.c linux-atm-2.5.0/src/arpd/arp.c
--- linux-atm-2.5.0-old/src/arpd/arp.c	2008-04-25 03:34:35.000000000 +0300
+++ linux-atm-2.5.0/src/arpd/arp.c	2008-04-25 03:34:51.000000000 +0300
@@ -15,7 +15,6 @@
 #include <sys/types.h>
 #include <sys/socket.h> /* for linux/if_arp.h */
 #include <netinet/in.h> /* for ntohs, etc. */
-#define _LINUX_NETDEVICE_H /* very crude hack for glibc2 */
 #include <linux/types.h>
 #include <linux/if_arp.h>
 #include <linux/if_ether.h>
diff -Nur linux-atm-2.5.0-old/src/arpd/io.c linux-atm-2.5.0/src/arpd/io.c
--- linux-atm-2.5.0-old/src/arpd/io.c	2008-04-25 03:34:35.000000000 +0300
+++ linux-atm-2.5.0/src/arpd/io.c	2008-04-25 03:34:51.000000000 +0300
@@ -21,7 +21,6 @@
 #include <atm.h>
 #include <linux/atmclip.h> /* for CLIP_DEFAULT_IDLETIMER */
 #include <linux/atmarp.h>
-#define _LINUX_NETDEVICE_H /* glibc2 */
 #include <linux/types.h>
 #include <linux/if_arp.h>
 
diff -Nur linux-atm-2.5.0-old/src/arpd/itf.c linux-atm-2.5.0/src/arpd/itf.c
--- linux-atm-2.5.0-old/src/arpd/itf.c	2008-04-25 03:34:35.000000000 +0300
+++ linux-atm-2.5.0/src/arpd/itf.c	2008-04-25 03:34:51.000000000 +0300
@@ -12,7 +12,6 @@
 #include <sys/types.h>
 #include <linux/atmclip.h>
 #include <sys/socket.h>
-#define _LINUX_NETDEVICE_H /* glibc2 */
 #include <linux/types.h>
 #include <linux/if_arp.h>
 
