Author: Richard Nelson <unixabg@gmail.com>
Description: Use background image from desktop-base (Closes: #634004).

diff -Naurp lxde-common.orig/pcmanfm/pcmanfm.conf.in lxde-common/pcmanfm/pcmanfm.conf.in
--- lxde-common.orig/pcmanfm/pcmanfm.conf.in	2012-04-19 00:07:05.131258426 +0200
+++ lxde-common/pcmanfm/pcmanfm.conf.in	2012-04-19 00:56:39.291706654 +0200
@@ -4,7 +4,7 @@
 
 [desktop]
 wallpaper_mode=1
-wallpaper=@prefix@/share/lxde/wallpapers/lxde_blue.jpg
+wallpaper=/usr/share/lxde/wallpapers/pardus-blue.png
 desktop_bg=#000000
 desktop_fg=#ffffff
 desktop_shadow=#000000
