--- lxde-common-0.5.5.orig/desktop.conf	2011-07-29 17:04:03.000000000 +0300
+++ lxde-common-0.5.5/desktop.conf	2014-01-06 19:58:34.486679053 +0200
@@ -2,8 +2,8 @@
 window_manager=openbox-lxde
 
 [GTK]
-sNet/ThemeName=Clearlooks
-sNet/IconThemeName=nuoveXT2
+sNet/ThemeName=MediterraneanTributeDark
+sNet/IconThemeName=elementary
 sGtk/FontName=Sans 10
 iGtk/ToolbarStyle=3
 iGtk/ButtonImages=1
--- lxde-common-0.5.5.orig/LXDE.desktop.in	2011-07-29 17:04:03.000000000 +0300
+++ lxde-common-0.5.5/LXDE.desktop.in	2014-01-06 21:25:13.223394122 +0200
@@ -1,12 +1,14 @@
 [Desktop Entry]
 Encoding=UTF-8
 # The names/descriptions should really be better
-Name=LXDE
+Name=LXDE Desktop Environment
+Name[tr]=LXDE Pencere Yöneticisi
 Comment=LXDE - Lightweight X11 desktop environment
 Comment[de]=LXDE - Leichtgewichtige X11 Desktop-Umgebung
 Comment[fi]=LXDE - kevyt X11-työpöytäympäristö
 Comment[ja]=LXDE - 軽量な X11 デスクトップ環境
 Comment[pt]=LXDE - Ambiente de trabalho X11 leve
+Comment[tr]=LXDE - Hafif bir X11 masaüstü ortamı
 Comment[zh_TW]=LXDE - 輕量級的 X11 桌面環境
 Exec=@prefix@/bin/startlxde
 # Icon=
--- lxde-common-0.5.5.orig/lxpanel/panel.in	2011-07-29 17:04:03.000000000 +0300
+++ lxde-common-0.5.5/lxpanel/panel.in	2014-01-06 20:48:35.333345629 +0200
@@ -15,7 +15,7 @@
     setpartialstrut=1
     usefontcolor=1
     fontcolor=#ffffff
-    background=1
+    background=0
     backgroundfile=@prefix@/share/lxpanel/images/background.png
 }
 
@@ -29,7 +29,7 @@
 Plugin {
     type = menu
     Config {
-        image=@prefix@/share/lxde/images/lxde-icon.png
+        image=/usr/share/lxde/images/pisilinuxLogo.png
         system {
         }
         separator {
@@ -53,7 +53,7 @@
             id=pcmanfm.desktop
         }
         Button {
-            id=firefox.desktop
+            id=/usr/share/applications/mozillafirefox.desktop
         }
     }
 }
@@ -110,15 +113,19 @@
 }
 
 Plugin {
-    type = netstat
+    type = tray
 }
 
 Plugin {
-    type = cpu
+    type = volume
 }
 
 Plugin {
-    type = tray
+    type = netstat
+}
+
+Plugin {
+    type = cpu
 }
 
 Plugin {
--- lxde-common-0.5.5.orig/pcmanfm/pcmanfm.conf.in	2011-07-29 17:04:03.000000000 +0300
+++ lxde-common-0.5.5/pcmanfm/pcmanfm.conf.in	2014-01-06 21:19:51.493387024 +0200
@@ -4,7 +4,7 @@
 
 [desktop]
 wallpaper_mode=1
-wallpaper=@prefix@/share/lxde/wallpapers/lxde_blue.jpg
+wallpaper=/usr/share/lxde/wallpapers/pisilinux3d.jpg
 desktop_bg=#000000
 desktop_fg=#ffffff
 desktop_shadow=#000000
