diff -Nuar lxde-common-0.5.0.orig/LXDE.desktop.in lxde-common-0.5.0/LXDE.desktop.in
--- lxde-common-0.5.0.orig/LXDE.desktop.in	2010-10-31 00:44:18.937494710 +0300
+++ lxde-common-0.5.0/LXDE.desktop.in	2010-10-31 00:48:11.698227546 +0300
@@ -7,6 +7,7 @@
 Comment[fi]=LXDE - kevyt X11-työpöytäympäristö
 Comment[ja]=LXDE - 軽量な X11 デスクトップ環境
 Comment[pt]=LXDE - Ambiente de trabalho X11 leve
+Comment[tr]=LXDE - Hafif bir X11 masaüstü ortamı
 Comment[zh_TW]=LXDE - 輕量級的 X11 桌面環境
 Exec=@prefix@/bin/startlxde
 # Icon=
