Author: Erdinç Gültekin <erdincgultekin@gmail.com>

diff -Nuar lxde-common-0.5.5.orig//desktop.conf lxde-common-0.5.5//desktop.conf
--- lxde-common-0.5.5.orig//desktop.conf	2012-07-01 13:15:11.693000004 +0200
+++ lxde-common-0.5.5//desktop.conf	2012-07-01 14:14:11.392000316 +0200
@@ -2,8 +2,8 @@
 window_manager=openbox-lxde
 
 [GTK]
-sNet/ThemeName=Clearlooks
-sNet/IconThemeName=nuoveXT2
+sNet/ThemeName=Oxygen
+sNet/IconThemeName=elementary
 sGtk/FontName=Sans 10
 iGtk/ToolbarStyle=3
 iGtk/ButtonImages=1
diff -Nuar a/lxpanel/panel.in b/lxpanel/panel.in
--- a/lxpanel/panel.in	2011-07-29 17:04:03.000000000 +0300
+++ b/lxpanel/panel.in	2013-08-14 11:55:01.000000000 +0300
@@ -11,12 +11,18 @@
     transparent=0
     tintcolor=#000000
     alpha=0
+    autohide=0
+    heightwhenhidden=2
     setdocktype=1
     setpartialstrut=1
     usefontcolor=1
+    fontsize=10
     fontcolor=#ffffff
+    usefontsize=0
     background=1
-    backgroundfile=@prefix@/share/lxpanel/images/background.png
+    backgroundfile=/usr/share/lxpanel/images/background-pisilinux.png
+    iconsize=24
+    loglevel=2
 }
 
 Plugin {
@@ -29,7 +35,7 @@
 Plugin {
     type = menu
     Config {
-        image=@prefix@/share/lxde/images/lxde-icon.png
+        image=/usr/share/lxde/images/pisilinuxLogo.png
         system {
         }
         separator {
@@ -53,7 +59,10 @@
             id=pcmanfm.desktop
         }
         Button {
-            id=firefox.desktop
+            id=/usr/share/applications/mozillafirefox.desktop
+        }
+        Button {
+            id=/usr/share/applications/kde4/package-manager.desktop
         }
     }
 }
@@ -68,8 +77,10 @@
 Plugin {
     type = wincmd
     Config {
+        image=window-manager
         Button1=iconify
         Button2=shade
+        Toggle=0
     }
 }
 
@@ -97,24 +108,31 @@
     Config {
         tooltips=1
         IconsOnly=0
-        AcceptSkipPager=1
-        ShowIconified=1
-        ShowMapped=1
         ShowAllDesks=0
         UseMouseWheel=1
         UseUrgencyHint=1
         FlatButton=0
         MaxTaskWidth=150
         spacing=1
+        GroupedTasks=0
     }
 }
 
 Plugin {
-    type = netstat
+    type = cpu
 }
 
 Plugin {
-    type = cpu
+    type = launchbar
+    Config {
+        Button {
+            id=/usr/share/applications/pavucontrol.desktop
+        }
+    }
+}
+
+Plugin {
+    type = netstat
 }
 
 Plugin {
@@ -126,16 +144,9 @@
     Config {
         ClockFmt=%R
         TooltipFmt=%A %x
-        BoldFont=0
-    }
-}
-
-Plugin {
-    type = launchbar
-    Config {
-        Button {
-            id=lxde-logout.desktop
-        }
+        BoldFont=1
+        IconOnly=0
+        CenterText=0
     }
 }
 
