diff -uNr old-lxde-common-0.5.0//openbox/rc.xml.in lxde-common-0.5.0/openbox/rc.xml.in
--- old-lxde-common-0.5.0//openbox/rc.xml.in	2008-04-26 04:32:26.000000000 +0200
+++ lxde-common-0.5.0/openbox/rc.xml.in	2010-03-23 08:43:40.534105730 +0100
@@ -271,12 +271,19 @@
       </action>
   </keybind>
 
-  <keybind key="C-Escape">
+  <keybind key="C-A-Escape">
       <action name="Execute">
-          <command>lxpanelctl menu</command>
+          <command>xkill</command>
       </action>
   </keybind>
 
+  <keybind key="Super_L">
+     <action name="Execute">
+         <command>lxpanelctl menu</command>
+     </action>
+  </keybind>
+
+
   <keybind key="F11">
      <action name="ToggleFullscreen"/>
   </keybind>
@@ -288,6 +295,13 @@
       </action>
   </keybind>
 
+  <!-- Launch Mtpaint for screenshots -->
+  <keybind key="Print">
+      <action name="Execute">
+          <command>mtpaint -s</command>
+      </action>
+  </keybind>
+
 </keyboard>
 
 
