Author: Erdinç Gültekin <erdincgultekin@gmail.com>

diff -Nuar lxde-common-0.5.5.orig//desktop.conf lxde-common-0.5.5//desktop.conf
--- lxde-common-0.5.5.orig//desktop.conf	2012-07-01 13:15:11.693000004 +0200
+++ lxde-common-0.5.5//desktop.conf	2012-07-01 14:14:11.392000316 +0200
@@ -2,8 +2,8 @@
 window_manager=openbox-lxde
 
 [GTK]
-sNet/ThemeName=Clearlooks
-sNet/IconThemeName=nuoveXT2
+sNet/ThemeName=Oxygen
+sNet/IconThemeName=elementary
 sGtk/FontName=Sans 10
 iGtk/ToolbarStyle=3
 iGtk/ButtonImages=1
diff -ur old-lxde-common-0.5.5/lxpanel/panel.in lxde-common-0.5.5/lxpanel/panel.in
--- old-lxde-common-0.5.5/lxpanel/panel.in	2011-07-29 17:04:03.000000000 +0300
+++ lxde-common-0.5.5/lxpanel/panel.in	2012-10-15 17:32:44.901000018 +0300
@@ -11,12 +11,17 @@
     transparent=0
     tintcolor=#000000
     alpha=0
+    autohide=0
+    heightwhenhidden=2
     setdocktype=1
     setpartialstrut=1
     usefontcolor=1
+    fontsize=10
     fontcolor=#ffffff
+    usefontsize=0
     background=1
-    backgroundfile=@prefix@/share/lxpanel/images/background.png
+    backgroundfile=/usr/share/lxpanel/images/background-anka.png
+    iconsize=24
 }
 
 Plugin {
@@ -29,7 +34,7 @@
 Plugin {
     type = menu
     Config {
-        image=@prefix@/share/lxde/images/lxde-icon.png
+        image=/usr/share/lxde/images/PardusAnkaLogo.png
         system {
         }
         separator {
@@ -53,7 +58,10 @@
             id=pcmanfm.desktop
         }
         Button {
-            id=firefox.desktop
+            id=/usr/share/applications/mozillafirefox.desktop
+        }
+        Button {
+            id=/usr/share/applications/kde4/package-manager.desktop
         }
     }
 }
@@ -68,8 +76,10 @@
 Plugin {
     type = wincmd
     Config {
+        image=window-manager
         Button1=iconify
         Button2=shade
+        Toggle=0
     }
 }
 
@@ -97,15 +107,13 @@
     Config {
         tooltips=1
         IconsOnly=0
-        AcceptSkipPager=1
-        ShowIconified=1
-        ShowMapped=1
         ShowAllDesks=0
         UseMouseWheel=1
         UseUrgencyHint=1
         FlatButton=0
         MaxTaskWidth=150
         spacing=1
+        GroupedTasks=0
     }
 }
 
@@ -114,10 +122,6 @@
 }
 
 Plugin {
-    type = cpu
-}
-
-Plugin {
     type = tray
 }
 
@@ -126,16 +130,21 @@
     Config {
         ClockFmt=%R
         TooltipFmt=%A %x
-        BoldFont=0
+        BoldFont=1
+        IconOnly=0
+        CenterText=0
     }
 }
 
 Plugin {
+    type = cpu
+}
+
+Plugin {
     type = launchbar
     Config {
         Button {
             id=lxde-logout.desktop
         }
     }
-}
 
