Index: aMule-2.3.1/amule.desktop
===================================================================
--- aMule-2.3.1.orig/amule.desktop
+++ aMule-2.3.1/amule.desktop
@@ -5,6 +5,6 @@ Icon=amule
 Terminal=false
 Type=Application
 Categories=Network;P2P;
-Comment=A client for the eD2k network
+Comment=Clone of eMule file sharing application
+Comment[tr]=eMule klonu dosya paylaşım programı
 Comment[fr]=Un client pour le réseau eD2k
-Comment[tr]=eD2k ağı için istemci
