Compile with lm_sensors3.

diff -up net-snmp-5.5/agent/mibgroup/hardware/sensors.h.orig net-snmp-5.5/agent/mibgroup/hardware/sensors.h
--- net-snmp-5.5/agent/mibgroup/hardware/sensors.h.orig	2009-04-24 00:53:26.000000000 +0200
+++ net-snmp-5.5/agent/mibgroup/hardware/sensors.h	2009-09-29 12:00:18.000000000 +0200
@@ -7,7 +7,7 @@ config_require(hardware/sensors/picld_se
 config_require(hardware/sensors/kstat_sensors)
 # endif
 #else
-config_require(hardware/sensors/lmsensors_v2)
+config_require(hardware/sensors/lmsensors_v3)
 #endif
 
 /* config_require(hardware/sensors/dummy_sensors) */
