diff -rupN cairo-dock-plugins-3.0.2/alsaMixer/src/applet-init.c cairo-dock-plugins-3.0.2_mod/alsaMixer/src/applet-init.c
--- cairo-dock-plugins-3.0.2/alsaMixer/src/applet-init.c	2012-06-15 21:32:58.000000000 +0200
+++ cairo-dock-plugins-3.0.2_mod/alsaMixer/src/applet-init.c	2012-09-13 16:22:35.178259521 +0200
@@ -52,6 +52,7 @@ static gboolean _cd_mixer_on_enter (GtkW
 	{
 		gtk_widget_show (myData.pScale);
 	}
+	return CAIRO_DOCK_INTERCEPT_NOTIFICATION;
 }
 gboolean _cd_mixer_on_leave (GtkWidget* pWidget,
 	GdkEventCrossing* pEvent,
@@ -62,6 +63,7 @@ gboolean _cd_mixer_on_leave (GtkWidget*
 		if (! myDesklet->container.bInside)
 			gtk_widget_hide (myData.pScale);
 	}
+	return CAIRO_DOCK_INTERCEPT_NOTIFICATION;
 }
 
 CD_APPLET_INIT_BEGIN
diff -rupN cairo-dock-plugins-3.0.2/Dbus/src/interface-main-methods.c cairo-dock-plugins-3.0.2_mod/Dbus/src/interface-main-methods.c
--- cairo-dock-plugins-3.0.2/Dbus/src/interface-main-methods.c	2012-06-15 21:32:58.000000000 +0200
+++ cairo-dock-plugins-3.0.2_mod/Dbus/src/interface-main-methods.c	2012-09-13 16:21:03.949721153 +0200
@@ -1103,6 +1103,7 @@ gboolean cd_dbus_main_reload_icon (dbusM
 static gboolean _on_icon_deleted (GList *ic, Icon *pIcon)
 {
 	ic->data = NULL;
+	return CAIRO_DOCK_INTERCEPT_NOTIFICATION;
 }
 gboolean cd_dbus_main_remove_icon (dbusMainObject *pDbusCallback, gchar *cIconQuery, GError **error)
 {
diff -rupN cairo-dock-plugins-3.0.2/dock-rendering/src/rendering-parabole.c cairo-dock-plugins-3.0.2_mod/dock-rendering/src/rendering-parabole.c
--- cairo-dock-plugins-3.0.2/dock-rendering/src/rendering-parabole.c	2012-06-15 21:32:58.000000000 +0200
+++ cairo-dock-plugins-3.0.2_mod/dock-rendering/src/rendering-parabole.c	2012-09-13 16:24:07.577804792 +0200
@@ -491,7 +491,7 @@ static double cd_rendering_project_curso
 	while (y_sup - y_inf > 1);
 	return (y_inf + y_sup) / 2;
 }
-static double cd_rendering_project_cursor_on_curve (double x0, double y0, double lambda, double alpha, double *fXOnCurve, double *fYOnCurve)
+static void cd_rendering_project_cursor_on_curve (double x0, double y0, double lambda, double alpha, double *fXOnCurve, double *fYOnCurve)
 {
 	double xM, yM;
 	if (my_fParaboleRatio > 1)
diff -rupN cairo-dock-plugins-3.0.2/dustbin/src/applet-notifications.c cairo-dock-plugins-3.0.2_mod/dustbin/src/applet-notifications.c
--- cairo-dock-plugins-3.0.2/dustbin/src/applet-notifications.c	2012-06-15 21:32:58.000000000 +0200
+++ cairo-dock-plugins-3.0.2_mod/dustbin/src/applet-notifications.c	2012-09-13 16:26:01.116474822 +0200
@@ -96,6 +96,7 @@ static gboolean _display_result (CDShare
 	}
 	cairo_dock_discard_task (myData.pInfoTask);
 	myData.pInfoTask = NULL;
+	return TRUE;
 }
 static void _cd_dustbin_show_info (GtkMenuItem *menu_item, CairoDockModuleInstance *myApplet)
 {

