Index: xorg-docs-1.4_20090613/sgml/core/Xserver-spec.sgml
===================================================================
--- xorg-docs-1.4_20090613.orig/sgml/core/Xserver-spec.sgml
+++ xorg-docs-1.4_20090613/sgml/core/Xserver-spec.sgml
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="ISO-8859-1"?>
-<!DOCTYPE article PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN" "http://www.oasis-open.org/docbook/xml/4.3/docbookx.dtd" [
+<!DOCTYPE article PUBLIC "-//OASIS//DTD DocBook XML V4.3//EN" [
 ]>
 
 <article>
@@ -5176,4 +5176,4 @@ mi and fb implementations.</para>
 	  </tgroup>
 	</table>
 </section>
-</article>
\ No newline at end of file
+</article>
Index: xorg-docs-1.4_20090613/sgml/security/XACE-Spec.sgml
===================================================================
--- xorg-docs-1.4_20090613.orig/sgml/security/XACE-Spec.sgml
+++ xorg-docs-1.4_20090613/sgml/security/XACE-Spec.sgml
@@ -1,6 +1,5 @@
 <?xml version="1.0" encoding="ISO-8859-1"?>
-<!DOCTYPE article PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"
-                      "http://www.oasis-open.org/docbook/xml/4.3/docbookx.dtd"
+<!DOCTYPE article PUBLIC "-//OASIS//DTD DocBook XML V4.3//EN"
 []>
 
 <article>
