Index: kdelibs-4.7.1/kio/kio/kprotocolmanager.cpp
===================================================================
--- kdelibs-4.7.1.orig/kio/kio/kprotocolmanager.cpp
+++ kdelibs-4.7.1/kio/kio/kprotocolmanager.cpp
@@ -632,7 +632,7 @@ QString KProtocolManager::defaultUserAge
     }
 
     // Full format: Mozilla/5.0 (Linux
-    d->useragent = QL1S("Mozilla/5.0 (");
+    d->useragent = QL1S("Mozilla/5.0 (PisiLinux");
     d->useragent += supp;
     d->useragent += QL1S(") KHTML/");
     d->useragent += QString::number(KDE::versionMajor());
