Index: kate/document/katedocument.cpp
===================================================================
--- kate/document/katedocument.cpp.orig
+++ kate/document/katedocument.cpp
@@ -2159,8 +2159,7 @@ bool KateDocument::openFile()
     // this file can't be saved again without killing it
     setReadWrite( false );
 
-    if (!suppressOpeningErrorDialogs())
-      KMessageBox::information (parentWidget
+    KMessageBox::information (parentWidget
         , i18n ("The file %1 was opened with %2 encoding but contained invalid characters."
                 " It is set to read-only mode, as saving might destroy its content."
                 " Either reopen the file with the correct encoding chosen or enable the read-write mode again in the menu to be able to edit it.", this->url().pathOrUrl(),
