Index: qtermwidget/lib/lib.pro
===================================================================
--- qtermwidget.orig/lib/lib.pro
+++ qtermwidget/lib/lib.pro
@@ -4,7 +4,7 @@ DESTDIR 	= ..
 
 TARGET		= qtermwidget
 
-CONFIG		+= qt debug_and_release warn_on build_all staticlib #dll
+CONFIG		+= qt debug_and_release warn_on build_all #dll
 
 QT += core gui
 
@@ -42,7 +42,3 @@ SOURCES 	= TerminalCharacterDecoder.cpp 
 		Session.cpp ShellCommand.cpp \
 		qtermwidget.cpp
 
-	    
-
-
-	
\ Dosya sonunda yenisatır yok.
Index: qtermwidget/src/src.pro
===================================================================
--- qtermwidget.orig/src/src.pro
+++ qtermwidget/src/src.pro
@@ -10,11 +10,11 @@ MOC_DIR 	= ../.moc
 CONFIG(debug, debug|release) {
     OBJECTS_DIR = ../.objs_d
     TARGET 	= consoleq_d
-    LIBS 	+= -L.. ../libqtermwidget_d.a
+    LIBS 	+= -L.. ../libqtermwidget_d.so
 } else {
     OBJECTS_DIR = ../.objs
     TARGET 	= consoleq
-    LIBS 	+= -L.. ../libqtermwidget.a
+    LIBS 	+= -L.. ../libqtermwidget.so
 }
 
 SOURCES 	= main.cpp 
@@ -24,4 +24,3 @@ INCLUDEPATH 	= ../lib
 #LIBS 		+= -L.. -lqtermwidget
 
 
-	
\ Dosya sonunda yenisatır yok.
