Index: env.c
===================================================================
RCS file: /home/cvs/courtesan/sudo/env.c,v
retrieving revision 1.101
diff -u -r1.101 env.c
--- env.c	11 Mar 2009 23:01:10 -0000	1.101
+++ env.c	24 Apr 2009 15:07:33 -0000
@@ -257,6 +257,9 @@
 	return(-1);
     }
 
+    if (val == NULL)
+	val = "";		/* glibc treats a NULL val as "" */
+
     esize = strlen(var) + 1 + strlen(val) + 1;
     estring = emalloc(esize);
 
