diff -Nru libindicate-0.6.92.orig/configure.ac libindicate-0.6.92/configure.ac
--- libindicate-0.6.92.orig/configure.ac	2012-06-05 13:23:17.038530346 -0400
+++ libindicate-0.6.92/configure.ac	2012-06-05 15:44:41.811635384 -0400
@@ -213,7 +213,7 @@
 )
 
 # libpyglib-2.0-python* doesn't have a pkgconfig unfortunately
-INDICATE_PYTHON_LIBS+=" -lpyglib-2.0-python$PYTHON_VERSION"
+INDICATE_PYTHON_LIBS+=" -lpyglib-2.0-python"
 
 AC_SUBST(INDICATE_PYTHON_CFLAGS)
 AC_SUBST(INDICATE_PYTHON_LIBS)
