Index: fltk-1.1.10/configure.in
===================================================================
--- fltk-1.1.10.orig/configure.in
+++ fltk-1.1.10/configure.in
@@ -202,10 +202,7 @@ if test x$enable_shared = xyes; then
             FLDSONAME="libfltk_forms.so.$FL_API_VERSION"
             GLDSONAME="libfltk_gl.so.$FL_API_VERSION"
             IMGDSONAME="libfltk_images.so.$FL_API_VERSION"
-            DSOCOMMAND="\$(CXX) \$(DSOFLAGS) -Wl,-soname,\$@ \$(LDLIBS) -shared -fPIC $DEBUGFLAG -o"
-	    if test "x$libdir" != "x/usr/lib" -a "x$libdir" != "x/usr/lib32" -a "x$libdir" != "x/usr/lib64"; then
-		    DSOLINK="-Wl,-rpath,$libdir"
-	    fi
+            DSOCOMMAND="\$(CXX) \$(DSOFLAGS) \$(LDFLAGS) -Wl,-soname,\$@ -shared -fPIC $DEBUGFLAG -o"
 	    ;;
 	AIX*)
             DSONAME="libfltk_s.a"
Index: fltk-1.1.10/src/Makefile
===================================================================
--- fltk-1.1.10.orig/src/Makefile
+++ fltk-1.1.10/src/Makefile
@@ -210,13 +210,13 @@ $(LIBNAME): $(OBJECTS)
 
 libfltk.so.1.1: $(OBJECTS)
 	echo $(DSOCOMMAND) $@ ...
-	$(DSOCOMMAND) $@ $(OBJECTS)
+	$(DSOCOMMAND) $@ $(OBJECTS) $(LDLIBS)
 	$(RM) libfltk.so
 	$(LN) libfltk.so.1.1 libfltk.so
 
 libfltk.sl.1.1: $(OBJECTS)
 	echo $(DSOCOMMAND) $@ ...
-	$(DSOCOMMAND) $@ $(OBJECTS)
+	$(DSOCOMMAND) $@ $(OBJECTS) $(LDLIBS)
 	$(RM) libfltk.sl
 	$(LN) libfltk.sl.1.1 libfltk.sl
 
@@ -232,7 +232,7 @@ libfltk.1.1.dylib: $(OBJECTS)
 
 libfltk_s.a: $(OBJECTS)
 	echo $(DSOCOMMAND) libfltk_s.o ...
-	$(DSOCOMMAND) libfltk_s.o $(OBJECTS) $(IMAGELIBS)
+	$(DSOCOMMAND) libfltk_s.o $(OBJECTS) $(IMAGELIBS) $(LDLIBS)
 	echo $(LIBCOMMAND) libfltk_s.a libfltk_s.o
 	$(RM) $@
 	$(LIBCOMMAND) libfltk_s.a libfltk_s.o
@@ -246,13 +246,13 @@ $(FLLIBNAME): $(FLOBJECTS)
 
 libfltk_forms.so.1.1: $(FLOBJECTS) libfltk.so.1.1
 	echo $(DSOCOMMAND) $@ ...
-	$(DSOCOMMAND) $@ $(FLOBJECTS) -L. -lfltk
+	$(DSOCOMMAND) $@ $(FLOBJECTS) $(LDLIBS) -L. -lfltk
 	$(RM) libfltk_forms.so
 	$(LN) libfltk_forms.so.1.1 libfltk_forms.so
 
 libfltk_forms.sl.1.1: $(FLOBJECTS) libfltk.sl.1.1
 	echo $(DSOCOMMAND) $@ ...
-	$(DSOCOMMAND) $@ $(FLOBJECTS) -L. -lfltk
+	$(DSOCOMMAND) $@ $(FLOBJECTS) $(LDLIBS) -L. -lfltk
 	$(RM) libfltk_forms.sl
 	$(LN) libfltk_forms.sl.1.1 libfltk_forms.sl
 
@@ -268,7 +268,7 @@ libfltk_forms.1.1.dylib: $(FLOBJECTS) li
 
 libfltk_forms_s.a: $(FLOBJECTS)
 	echo $(DSOCOMMAND) libfltk_forms_s.o ...
-	$(DSOCOMMAND) libfltk_forms_s.o $(FLOBJECTS)
+	$(DSOCOMMAND) libfltk_forms_s.o $(FLOBJECTS) $(LDLIBS)
 	echo $(LIBCOMMAND) libfltk_forms_s.a libfltk_forms_s.o
 	$(RM) $@
 	$(LIBCOMMAND) libfltk_forms_s.a libfltk_forms_s.o
@@ -282,13 +282,13 @@ $(GLLIBNAME): $(GLOBJECTS)
 
 libfltk_gl.so.1.1: $(GLOBJECTS) libfltk.so.1.1
 	echo $(DSOCOMMAND) $@ ...
-	$(DSOCOMMAND) $@ $(GLOBJECTS) -L. $(GLDLIBS) -lfltk
+	$(DSOCOMMAND) $@ $(GLOBJECTS) $(LDLIBS) -lGL -L. $(GLDLIBS) -lfltk
 	$(RM) libfltk_gl.so
 	$(LN) libfltk_gl.so.1.1 libfltk_gl.so
 
 libfltk_gl.sl.1.1: $(GLOBJECTS) libfltk.sl.1.1
 	echo $(DSOCOMMAND) $@ ...
-	$(DSOCOMMAND) $@ $(GLOBJECTS) -L. -lfltk
+	$(DSOCOMMAND) $@ $(GLOBJECTS) $(LDLIBS) -L. -lfltk
 	$(RM) libfltk_gl.sl
 	$(LN) libfltk_gl.sl.1.1 libfltk_gl.sl
 
@@ -304,7 +304,7 @@ libfltk_gl.1.1.dylib: $(GLOBJECTS) libfl
 
 libfltk_gl_s.a: $(GLOBJECTS)
 	echo $(DSOCOMMAND) libfltk_gl_s.o ...
-	$(DSOCOMMAND) libfltk_gl_s.o $(GLOBJECTS)
+	$(DSOCOMMAND) libfltk_gl_s.o $(GLOBJECTS) $(LDLIBS)
 	echo $(LIBCOMMAND) libfltk_gl_s.a libfltk_gl_s.o
 	$(RM) $@
 	$(LIBCOMMAND) libfltk_gl_s.a libfltk_gl_s.o
@@ -318,13 +318,13 @@ $(IMGLIBNAME): $(IMGOBJECTS)
 
 libfltk_images.so.1.1: $(IMGOBJECTS) libfltk.so.1.1
 	echo $(DSOCOMMAND) $@ ...
-	$(DSOCOMMAND) $@ $(IMGOBJECTS) -L. $(IMAGELIBS) -lfltk
+	$(DSOCOMMAND) $@ $(IMGOBJECTS) -L. $(IMAGELIBS) $(LDLIBS) -lfltk
 	$(RM) libfltk_images.so
 	$(LN) libfltk_images.so.1.1 libfltk_images.so
 
 libfltk_images.sl.1.1: $(IMGOBJECTS) libfltk.sl.1.1
 	echo $(DSOCOMMAND) $@ ...
-	$(DSOCOMMAND) $@ $(IMGOBJECTS) -L. $(IMAGELIBS) -lfltk
+	$(DSOCOMMAND) $@ $(IMGOBJECTS) -L. $(IMAGELIBS) $(LDLIBS) -lfltk
 	$(RM) libfltk_images.sl
 	$(LN) libfltk_images.sl.1.1 libfltk_images.sl
 
@@ -340,7 +340,7 @@ libfltk_images.1.1.dylib: $(IMGOBJECTS) 
 
 libfltk_images_s.a: $(IMGOBJECTS)
 	echo $(DSOCOMMAND) libfltk_images_s.o ...
-	$(DSOCOMMAND) libfltk_images_s.o $(IMGOBJECTS)
+	$(DSOCOMMAND) libfltk_images_s.o $(IMGOBJECTS) $(LDLIBS)
 	echo $(LIBCOMMAND) libfltk_images_s.a libfltk_images_s.o
 	$(RM) $@
 	$(LIBCOMMAND) libfltk_images_s.a libfltk_images_s.o
