diff -up fltk-1.1.9/configure.in.rpath fltk-1.1.9/configure.in
--- fltk-1.1.9/configure.in.rpath	2008-04-11 11:53:01.000000000 -0500
+++ fltk-1.1.9/configure.in	2009-05-13 10:20:15.159163817 -0500
@@ -180,9 +180,9 @@ if test x$enable_shared = xyes; then
             GLDSONAME="libfltk_gl.so.$FL_API_VERSION"
             IMGDSONAME="libfltk_images.so.$FL_API_VERSION"
             DSOCOMMAND="\$(CXX) \$(DSOFLAGS) -Wl,-soname,\$@ \$(LDLIBS) -shared -fPIC $DEBUGFLAG -o"
-	    if test "x$libdir" != "x/usr/lib"; then
-		    DSOLINK="-Wl,-rpath,$libdir"
-            fi
+	    if test "x$libdir" != "x/usr/lib" -a "x$libdir" != "x/usr/lib32" -a "x$libdir" != "x/usr/lib64"; then
+		    DSOLINK="-Wl,-rpath,$libdir"
+	    fi
 	    ;;
 	AIX*)
             DSONAME="libfltk_s.a"
diff -up fltk-1.1.9/configure.rpath fltk-1.1.9/configure
--- fltk-1.1.9/configure.rpath	2008-04-27 03:28:21.000000000 -0500
+++ fltk-1.1.9/configure	2009-05-13 10:20:15.164164455 -0500
@@ -1930,9 +1930,9 @@ if test x$enable_shared = xyes; then
             GLDSONAME="libfltk_gl.so.$FL_API_VERSION"
             IMGDSONAME="libfltk_images.so.$FL_API_VERSION"
             DSOCOMMAND="\$(CXX) \$(DSOFLAGS) -Wl,-soname,\$@ \$(LDLIBS) -shared -fPIC $DEBUGFLAG -o"
-	    if test "x$libdir" != "x/usr/lib"; then
+	    if test "x$libdir" != "x/usr/lib" -a "x$libdir" != "x/usr/lib32" -a "x$libdir" != "x/usr/lib64"; then
 		    DSOLINK="-Wl,-rpath,$libdir"
-            fi
+	    fi
 	    ;;
 	AIX*)
             DSONAME="libfltk_s.a"
