diff -up fltk-1.1.8/fluid/fluid.desktop.dt fltk-1.1.8/fluid/fluid.desktop
--- fltk-1.1.8/fluid/fluid.desktop.dt	2008-03-16 07:07:41.000000000 -0500
+++ fltk-1.1.8/fluid/fluid.desktop	2008-03-29 17:07:49.000000000 -0500
@@ -2,9 +2,9 @@
 Name=FLUID
 Comment=FLTK GUI Designer
 TryExec=fluid
-Exec=cd %D && fluid %F
+Exec=fluid %F
 Icon=fluid
 Terminal=false
 Type=Application
-MimeType=application/x-fluid
+MimeType=application/x-fluid;
 Categories=Development;GUIDesigner;
