diff -up fltk-1.1.9/fltk-config.in.ldstaticflags fltk-1.1.9/fltk-config.in
--- fltk-1.1.9/fltk-config.in.ldstaticflags	2008-02-20 12:10:13.000000000 -0600
+++ fltk-1.1.9/fltk-config.in	2009-05-13 10:23:15.071926233 -0500
@@ -57,7 +57,7 @@ ARCHFLAGS="@ARCHFLAGS@"
 CFLAGS="@CFLAGS@ @LARGEFILE@ @PTHREAD_FLAGS@"
 CXXFLAGS="@CXXFLAGS@ @LARGEFILE@ @PTHREAD_FLAGS@"
 LDFLAGS="@LDFLAGS@"
-LDLIBS="@LIBS@"
+LDSTATICLIBS="@LIBS@"
 OPTIM="@OPTIM@"
 
 # Check for local invocation, and update paths accordingly...
@@ -216,7 +216,7 @@ else
     includes=
 fi
 
-if test "$libdir" != /usr/lib -a "$libdir" != /usr/lib32; then
+if test "$libdir" != /usr/lib -a "$libdir" != /usr/lib32 -a "$libdir" != /usr/lib64; then
     libs=-L$libdir
 else
     libs=
@@ -240,7 +240,7 @@ if test x$use_images = xyes; then
 fi
 
 LDLIBS="$DSOLINK $LDFLAGS $libs $LDLIBS"
-LDSTATIC="$LDFLAGS $LDSTATIC"
+LDSTATIC="$LDFLAGS $LDSTATIC $LDSTATICLIBS"
 
 # Answer to user requests
 if test -n "$echo_help"; then
