diff -Nur python-xlib-0.14rc1-old/Xlib/protocol/display.py python-xlib-0.14rc1/Xlib/protocol/display.py
--- python-xlib-0.14rc1-old/Xlib/protocol/display.py	2007-06-16 17:05:29.196129122 +0300
+++ python-xlib-0.14rc1/Xlib/protocol/display.py	2007-06-16 17:06:26.629397937 +0300
@@ -534,7 +534,7 @@
                 # We're the recieving thread, parse the data
                 if recieving:
                     try:
-                        bytes_recv = self.socket.recv(2048)
+                        bytes_recv = self.socket.recv(4096)
                     except socket.error, err:
                         self.close_internal('server: %s' % err[1])
                         raise self.socket_error
