diff -Naur --exclude '*.orig' autofs-5.0.4-228-separate-config-files/lib/defaults.c autofs-5.0.4-228-rename-configuration-file/lib/defaults.c
--- autofs-5.0.4-228-separate-config-files/lib/defaults.c	2009-03-09 13:38:11.000000000 +0100
+++ autofs-5.0.4-228-rename-configuration-file/lib/defaults.c	2009-03-09 13:41:59.000000000 +0100
@@ -23,7 +23,7 @@
 #include "log.h"
 #include "automount.h"
 
-#define DEFAULTS_CONFIG_FILE		AUTOFS_CONF_DIR "/autofs"
+#define DEFAULTS_CONFIG_FILE		AUTOFS_CONF_DIR "/autofs.conf"
 #define MAX_LINE_LEN			256
 
 #define ENV_NAME_MASTER_MAP		"MASTER_MAP_NAME"
diff -Naur --exclude '*.orig' autofs-5.0.4-228-separate-config-files/samples/Makefile autofs-5.0.4-228-rename-configuration-file/samples/Makefile
--- autofs-5.0.4-228-separate-config-files/samples/Makefile	2009-03-09 13:38:29.000000000 +0100
+++ autofs-5.0.4-228-rename-configuration-file/samples/Makefile	2009-03-09 13:41:59.000000000 +0100
@@ -51,8 +51,8 @@
 		echo "Installed package default configuration file as \"autofs.conf.new\"." ; \
 	else \
 		install -v autofs.conf.default -m 644 $(CONFIG) \
-				$(INSTALLROOT)$(autofsconfdir)/autofs ; \
-		echo "Installed package configuration configuration as \"autofs\"." ; \
+				$(INSTALLROOT)$(autofsconfdir)/autofs.conf ; \
+		echo "Installed package configuration configuration as \"autofs.conf\"." ; \
 		if test -z "$(CEXISTS)" ; \
 		then \
 			echo "Backup of existing configuration made to \"autofs.orig\"." ; \
