--- a/redhat/autofs.init.in
+++ b/redhat/autofs.init.in
@@ -129,9 +129,16 @@ function restart() {
 	status autofs > /dev/null 2>&1
 	if [ $? -eq 0 ]; then
 		stop
-		while [ -n "`pidof $prog`" ] ; do
-			sleep 5
-		done
+		if [ -n "`pidof $prog`" ]; then
+			# If we failed to stop, try at least one more time
+			# after waiting a little while
+			sleep 20
+			stop
+			auto_pid=`pidof $prog`
+			if [ -n "$auto_pid" ]; then
+				kill -9 $auto_pid
+			fi
+		fi
 	fi
 	start
 }