diff -Nuar gpac-4288.orig/modules/ffmpeg_in/ffmpeg_decode.c gpac-4288/modules/ffmpeg_in/ffmpeg_decode.c
--- gpac-4288.orig/modules/ffmpeg_in/ffmpeg_decode.c	2013-03-12 03:43:22.000000000 +0200
+++ gpac-4288/modules/ffmpeg_in/ffmpeg_decode.c	2014-05-22 15:48:46.693734171 +0300
@@ -38,7 +38,9 @@
 #undef USE_AVCODEC2
 #endif
 
-#if (LIBAVCODEC_VERSION_MAJOR >= 54) && (LIBAVCODEC_VERSION_MINOR >= 35)
+#if (LIBAVCODEC_VERSION_MAJOR >= 55) 
+#define USE_AVCTX3
+#elif (LIBAVCODEC_VERSION_MAJOR >= 54) && (LIBAVCODEC_VERSION_MINOR >= 35)
 #define USE_AVCTX3
 #endif
 
@@ -625,10 +627,12 @@
 		if (ffd->frame_start>inBufferLength) ffd->frame_start = 0;
 
 redecode:
-		gotpic = AVCODEC_MAX_AUDIO_FRAME_SIZE;
+		
 #ifdef USE_AVCODEC2
+       gotpic = 192000;
 		len = avcodec_decode_audio3(ctx, (short *)ffd->audio_buf, &gotpic, &pkt);
 #else
+        gotpic = AVCODEC_MAX_AUDIO_FRAME_SIZE;
 		len = avcodec_decode_audio2(ctx, (short *)ffd->audio_buf, &gotpic, inBuffer + ffd->frame_start, inBufferLength - ffd->frame_start);
 #endif
 		if (len<0) { ffd->frame_start = 0; return GF_NON_COMPLIANT_BITSTREAM; }
diff -Nuar gpac-4288.orig/modules/ffmpeg_in/ffmpeg_in.h gpac-4288/modules/ffmpeg_in/ffmpeg_in.h
--- gpac-4288.orig/modules/ffmpeg_in/ffmpeg_in.h	2013-03-12 03:43:22.000000000 +0200
+++ gpac-4288/modules/ffmpeg_in/ffmpeg_in.h	2014-05-22 15:49:25.863735873 +0300
@@ -120,7 +120,7 @@
 
 	/*for audio packed frames*/
 	u32 frame_start;
-	char audio_buf[AVCODEC_MAX_AUDIO_FRAME_SIZE];
+    char audio_buf[192000];
 	Bool check_h264_isma;
 
 	u32 base_ES_ID;
