--- configure.orig	2012-11-03 19:41:39.000000000 +0200
+++ configure	2012-11-03 19:57:23.214752513 +0200
@@ -62,7 +62,7 @@
 
 #GPAC module config
 js_flags="XP_UNIX"
-js_lib="-ljs"
+js_lib="-lmozjs185"
 lm_lib=""
 has_mingw_directx="no"
 has_js="no"
@@ -686,7 +686,7 @@
     #try local
     js_inc="$local_inc/js"
     js_flags="-DXP_UNIX -I$local_inc/js"
-    if $cc -o $TMPO $TMPC $js_flags -L$local_lib -ljs -lpthread 2> /dev/null  ; then
+    if $cc -o $TMPO $TMPC $js_flags -L$local_lib -lmozjs185 -lpthread 2> /dev/null  ; then
         has_js="local"
         #dc added
     else
@@ -721,17 +721,17 @@
             #try prefix (DC)
             js_inc="$prefix/include/js"
             js_flags="-DXP_UNIX -I$prefix/include/js"
-            if $cc -o $TMPO $TMPC $js_flags -L$prefix/lib -ljs -lpthread 2> /dev/null  ; then
+            if $cc -o $TMPO $TMPC $js_flags -L$prefix/lib -lmozjs185 -lpthread 2> /dev/null  ; then
                 has_js="prefix"
                 #dc added end
             else
-                if $cc -o $TMPO $TMPC $js_flags $LDFLAGS -ljs -lpthread 2> /dev/null  ; then
+                if $cc -o $TMPO $TMPC $js_flags $LDFLAGS -lmozjs185 -lpthread 2> /dev/null  ; then
                     js_inc="/usr/include"
                     has_js="system"
-                elif $cc -o $TMPO $TMPC -DXP_UNIX -I$alt_macosx_dir/include/js -L$alt_macosx_dir/lib $LDFLAGS -ljs -lpthread 2> /dev/null ; then
+                elif $cc -o $TMPO $TMPC -DXP_UNIX -I$alt_macosx_dir/include/js -L$alt_macosx_dir/lib $LDFLAGS -lmozjs185 -lpthread 2> /dev/null ; then
                     has_js="system"
                     js_flags="-DXP_UNIX -I$alt_macosx_dir/include/js"
-                    js_lib="-L$alt_macosx_dir/lib -ljs"
+                    js_lib="-L$alt_macosx_dir/lib -lmozjs185"
                     js_inc="$alt_macosx_dir/include/js"
                 else
                     #debian spidermonkey (smjs)
@@ -775,14 +775,14 @@
     if test "$cpu" = "sh4" ; then
         lm_lib="-lm"
     fi
-    if $cc -o $TMPO $TMPC $js_flags $lm_lib -L$local_lib -ljs 2> /dev/null  ; then
+    if $cc -o $TMPO $TMPC $js_flags $lm_lib -L$local_lib -lmozjs185 2> /dev/null  ; then
         has_js="local"
         #dc added
     else
         #try prefix (DC)
         js_inc="$prefix/include/js"
         js_flags="-DXP_UNIX -I$prefix/include/js"
-        if $cc -o $TMPO $TMPC $js_flags -L$prefix/lib -ljs 2> /dev/null  ; then
+        if $cc -o $TMPO $TMPC $js_flags -L$prefix/lib -lmozjs185 2> /dev/null  ; then
             has_js="prefix"
             #dc added end
         else
@@ -795,13 +795,13 @@
                 fi
             fi
             if test "$has_js" = "no" ; then
-                if $cc -o $TMPO $TMPC $js_flags $LDFLAGS -ljs 2> /dev/null  ; then
+                if $cc -o $TMPO $TMPC $js_flags $LDFLAGS -lmozjs185 2> /dev/null  ; then
                     js_inc="/usr/include"
                     has_js="system"
-                elif $cc -o $TMPO $TMPC -DXP_UNIX -I$alt_macosx_dir/include/js -L$alt_macosx_dir/lib $LDFLAGS -ljs 2> /dev/null ; then
+                elif $cc -o $TMPO $TMPC -DXP_UNIX -I$alt_macosx_dir/include/js -L$alt_macosx_dir/lib $LDFLAGS -lmozjs185 2> /dev/null ; then
                     has_js="system"
                     js_flags="-DXP_UNIX -I$alt_macosx_dir/include/js"
-                    js_lib="-L$alt_macosx_dir/lib -ljs"
+                    js_lib="-L$alt_macosx_dir/lib -lmozjs185"
                     js_inc="$alt_macosx_dir/include/js"
                 else
                     #debian spidermonkey (smjs)
@@ -2572,7 +2572,7 @@
 else
     if test "$has_js" = "local" ; then
         js_flags="-DXP_UNIX -I$local_inc/js"
-        js_lib="-ljs"
+        js_lib="-lmozjs185"
     fi
     echo "JS_FLAGS=$js_flags" >> config.mak
     echo "JS_LIBS=$js_lib" >> config.mak
