diff -up kremotecontrol-4.7.90/libkremotecontrol/CMakeLists.txt.namelink_skip kremotecontrol-4.7.90/libkremotecontrol/CMakeLists.txt
--- kremotecontrol-4.7.90/libkremotecontrol/CMakeLists.txt.namelink_skip	2011-09-26 04:46:44.000000000 -0500
+++ kremotecontrol-4.7.90/libkremotecontrol/CMakeLists.txt	2011-12-08 12:10:07.003458850 -0600
@@ -42,7 +42,7 @@ set_target_properties(libkremotecontrol
 target_link_libraries(libkremotecontrol ${KDE4_KDEUI_LIBS} ${KDE4_KIO_LIBS} ${QT_QTXMLPATTERNS_LIBRARY} ${QT_QTSCRIPT_LIBRARY} ${X11_X11_LIB} ${X11_XTest_LIB})
 
 
-install(TARGETS libkremotecontrol ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS libkremotecontrol ${INSTALL_TARGETS_DEFAULT_ARGS} LIBRARY NAMELINK_SKIP)
 install( FILES kremotecontrolmanager.desktop DESTINATION ${SERVICETYPES_INSTALL_DIR}  )
 
 add_subdirectory(backends)
