diff -up freeglut-2.8.0/progs/demos/CallbackMaker/Makefile.in.fixld freeglut-2.8.0/progs/demos/CallbackMaker/Makefile.in
--- freeglut-2.8.0/progs/demos/CallbackMaker/Makefile.in.fixld	2012-01-02 13:18:07.000000000 +0100
+++ freeglut-2.8.0/progs/demos/CallbackMaker/Makefile.in	2012-01-03 11:04:07.711906872 +0100
@@ -110,7 +110,7 @@ LDFLAGS = @LDFLAGS@
 LIBM = @LIBM@
 LIBOBJS = @LIBOBJS@
 LIBRARY = @LIBRARY@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lGL
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBUSBHID = @LIBUSBHID@
diff -up freeglut-2.8.0/progs/demos/Fractals/Makefile.in.fixld freeglut-2.8.0/progs/demos/Fractals/Makefile.in
--- freeglut-2.8.0/progs/demos/Fractals/Makefile.in.fixld	2012-01-02 13:18:07.000000000 +0100
+++ freeglut-2.8.0/progs/demos/Fractals/Makefile.in	2012-01-03 11:04:07.711906872 +0100
@@ -110,7 +110,7 @@ LDFLAGS = @LDFLAGS@
 LIBM = @LIBM@
 LIBOBJS = @LIBOBJS@
 LIBRARY = @LIBRARY@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lGL
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBUSBHID = @LIBUSBHID@
diff -up freeglut-2.8.0/progs/demos/Fractals_random/Makefile.in.fixld freeglut-2.8.0/progs/demos/Fractals_random/Makefile.in
--- freeglut-2.8.0/progs/demos/Fractals_random/Makefile.in.fixld	2012-01-02 13:18:07.000000000 +0100
+++ freeglut-2.8.0/progs/demos/Fractals_random/Makefile.in	2012-01-03 11:04:07.719906936 +0100
@@ -111,7 +111,7 @@ LDFLAGS = @LDFLAGS@
 LIBM = @LIBM@
 LIBOBJS = @LIBOBJS@
 LIBRARY = @LIBRARY@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lGL
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBUSBHID = @LIBUSBHID@
diff -up freeglut-2.8.0/progs/demos/Lorenz/Makefile.in.fixld freeglut-2.8.0/progs/demos/Lorenz/Makefile.in
--- freeglut-2.8.0/progs/demos/Lorenz/Makefile.in.fixld	2012-01-02 13:18:07.000000000 +0100
+++ freeglut-2.8.0/progs/demos/Lorenz/Makefile.in	2012-01-03 11:04:07.720907364 +0100
@@ -110,7 +110,7 @@ LDFLAGS = @LDFLAGS@
 LIBM = @LIBM@
 LIBOBJS = @LIBOBJS@
 LIBRARY = @LIBRARY@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lGL -lm
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBUSBHID = @LIBUSBHID@
diff -up freeglut-2.8.0/progs/demos/One/Makefile.in.fixld freeglut-2.8.0/progs/demos/One/Makefile.in
--- freeglut-2.8.0/progs/demos/One/Makefile.in.fixld	2012-01-02 13:18:07.000000000 +0100
+++ freeglut-2.8.0/progs/demos/One/Makefile.in	2012-01-03 11:04:07.721907192 +0100
@@ -109,7 +109,7 @@ LDFLAGS = @LDFLAGS@
 LIBM = @LIBM@
 LIBOBJS = @LIBOBJS@
 LIBRARY = @LIBRARY@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lGL
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBUSBHID = @LIBUSBHID@
diff -up freeglut-2.8.0/progs/demos/shapes/Makefile.in.fixld freeglut-2.8.0/progs/demos/shapes/Makefile.in
--- freeglut-2.8.0/progs/demos/shapes/Makefile.in.fixld	2012-01-02 13:18:07.000000000 +0100
+++ freeglut-2.8.0/progs/demos/shapes/Makefile.in	2012-01-03 11:04:07.722907340 +0100
@@ -110,7 +110,7 @@ LDFLAGS = @LDFLAGS@
 LIBM = @LIBM@
 LIBOBJS = @LIBOBJS@
 LIBRARY = @LIBRARY@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lGL
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBUSBHID = @LIBUSBHID@
diff -up freeglut-2.8.0/progs/demos/smooth_opengl3/Makefile.in.fixld freeglut-2.8.0/progs/demos/smooth_opengl3/Makefile.in
--- freeglut-2.8.0/progs/demos/smooth_opengl3/Makefile.in.fixld	2012-01-02 13:18:07.000000000 +0100
+++ freeglut-2.8.0/progs/demos/smooth_opengl3/Makefile.in	2012-01-03 11:04:07.723914882 +0100
@@ -110,7 +110,7 @@ LDFLAGS = @LDFLAGS@
 LIBM = @LIBM@
 LIBOBJS = @LIBOBJS@
 LIBRARY = @LIBRARY@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lGL
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBUSBHID = @LIBUSBHID@
diff -up freeglut-2.8.0/progs/demos/spaceball/Makefile.in.fixld freeglut-2.8.0/progs/demos/spaceball/Makefile.in
--- freeglut-2.8.0/progs/demos/spaceball/Makefile.in.fixld	2012-01-02 13:18:07.000000000 +0100
+++ freeglut-2.8.0/progs/demos/spaceball/Makefile.in	2012-01-03 11:04:07.723914882 +0100
@@ -111,7 +111,7 @@ LDFLAGS = @LDFLAGS@
 LIBM = @LIBM@
 LIBOBJS = @LIBOBJS@
 LIBRARY = @LIBRARY@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lm -lGL
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBUSBHID = @LIBUSBHID@
