diff --git a/config_unix.py b/config_unix.py
index c888c28..942a750 100644
--- a/config_unix.py
+++ b/config_unix.py
@@ -147,7 +147,7 @@ def main():
         Dependency('JPEG', 'jpeglib.h', 'libjpeg', ['jpeg']),
         Dependency('SCRAP', '', 'libX11', ['X11']),
         Dependency('PORTMIDI', 'portmidi.h', 'libportmidi.so', ['portmidi']),
-        Dependency('PORTTIME', 'porttime.h', 'libporttime.so', ['porttime']),
+        Dependency('PORTTIME', 'porttime.h', 'libportmidi.so', ['portmidi']),
         #Dependency('GFX', 'SDL_gfxPrimitives.h', 'libSDL_gfx.so', ['SDL_gfx']),
     ]
     if not DEPS[0].found:
diff --git a/Setup.in b/Setup.in
index 4bb6c1c..c90174f 100644
--- a/Setup.in
+++ b/Setup.in
@@ -16,7 +16,7 @@ PNG = -lpng
 JPEG = -ljpeg
 SCRAP = -lX11
 PORTMIDI = -lportmidi
-PORTTIME = -lporttime
+PORTTIME = 
 #--EndConfig
 
 #DEBUG = -C-W -C-Wall
