--- qstardict/popupwindow.cpp	2008-03-31 08:55:46.000000000 +0300
+++ qstardict/popupwindow.cpp	2008-05-14 22:05:05.000000000 +0300
@@ -64,7 +64,7 @@
     setWindowOpacity(config.value("PopupWindow/opacity", 1.0).toDouble());
     setTimeoutBeforeHide(config.value("PopupWindow/timeoutBeforeHide", 500).toInt());
     setDefaultSize(config.value("PopupWindow/defaultSize", QSize(320, 240)).toSize());
-    setPronounceWord(config.value("PopupWindow/pronounceWord", true).toBool());
+    setPronounceWord(config.value("PopupWindow/pronounceWord", false).toBool());
     setDefaultStyleSheet(config.value("PopupWindow/defaultStyleSheet", defaultStyleSheet()).toString());
 }
 
