--- /dev/null	2008-11-23 16:57:26.413001560 +0200
+++ qstardict/translations/qstardict-tr_TR.ts	2008-11-23 23:28:45.000000000 +0200
@@ -0,0 +1,592 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE TS><TS version="1.1" language="tr_TR">
+<defaultcodec></defaultcodec>
+<context>
+    <name>QStarDict::CSSEdit</name>
+    <message>
+        <location filename="../cssedit.ui" line="16"/>
+        <source>Preview</source>
+        <translation>Önizleme</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="30"/>
+        <source>Element</source>
+        <translation>Öge</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="37"/>
+        <source>Foreground</source>
+        <translation>Önplan</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="44"/>
+        <source>Background</source>
+        <translation>Arkaplan</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="51"/>
+        <source>Select element</source>
+        <translation>Ögeyi seç</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="68"/>
+        <source>Click to select color</source>
+        <translation>Rengi seçmek için tıklayın</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="71"/>
+        <source>#000000</source>
+        <translation>#000000</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="78"/>
+        <source>Font</source>
+        <translation>Yazıtipi</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="85"/>
+        <source>Size</source>
+        <translation>Boyut</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="92"/>
+        <source>Select font</source>
+        <translation>Yazı tipi seç</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="99"/>
+        <source>Select font size</source>
+        <translation>Yazı tipi boyutunu seç</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="102"/>
+        <source>pt</source>
+        <translation>pt</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="127"/>
+        <source>Bold</source>
+        <translation>Kalın</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="130"/>
+        <source>B</source>
+        <translation>K</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="147"/>
+        <source>Italic</source>
+        <translation>Yatık</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="150"/>
+        <source>I</source>
+        <translation>Y</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="167"/>
+        <source>Underline</source>
+        <translation>Altı çizili</translation>
+    </message>
+    <message>
+        <location filename="../cssedit.ui" line="170"/>
+        <source>U</source>
+        <translation>U</translation>
+    </message>
+</context>
+<context>
+    <name>QStarDict::DictBrowser</name>
+    <message>
+        <location filename="../dictbrowser.cpp" line="72"/>
+        <source>The word &lt;b&gt;%1&lt;/b&gt; is not found.</source>
+        <translation>&lt;b&gt;%1&lt;/b&gt; sözcüğü bulunamadı.</translation>
+    </message>
+</context>
+<context>
+    <name>QStarDict::DictWidget</name>
+    <message>
+        <location filename="../dictwidget.cpp" line="114"/>
+        <source>Save translation</source>
+        <translation>Çeviriyi kaydet</translation>
+    </message>
+    <message>
+        <location filename="../dictwidget.cpp" line="129"/>
+        <source>HTML files (*.html, *.htm)</source>
+        <translation>HTML dosyaları (*.html, *.htm)</translation>
+    </message>
+    <message>
+        <location filename="../dictwidget.cpp" line="116"/>
+        <source>Text files (*.txt)</source>
+        <translation>Metin dosyaları (*.txt)</translation>
+    </message>
+    <message>
+        <location filename="../dictwidget.cpp" line="123"/>
+        <source>Error</source>
+        <translation>Hata</translation>
+    </message>
+    <message>
+        <location filename="../dictwidget.cpp" line="124"/>
+        <source>Cannot save translation as %1</source>
+        <translation>Çeviri %1 olarak kaydedilemedi</translation>
+    </message>
+    <message>
+        <location filename="../dictwidget.cpp" line="75"/>
+        <source>Go to &amp;previous translation</source>
+        <translation>&amp;Önceki çeviriye git</translation>
+    </message>
+    <message>
+        <location filename="../dictwidget.cpp" line="80"/>
+        <source>Go to &amp;next translation</source>
+        <translation>&amp;Sonraki çeviriye git</translation>
+    </message>
+    <message>
+        <location filename="../dictwidget.cpp" line="85"/>
+        <source>&amp;Save to file</source>
+        <translation>&amp;Dosyaya kaydet</translation>
+    </message>
+    <message>
+        <location filename="../dictwidget.cpp" line="91"/>
+        <source>Speak &amp;word</source>
+        <translation>Sözcüğü &amp;telaffuz et</translation>
+    </message>
+    <message>
+        <location filename="../dictwidget.cpp" line="88"/>
+        <source>Prin&amp;t translation</source>
+        <translation>Çeviriyi &amp;yazdır</translation>
+    </message>
+</context>
+<context>
+    <name>QStarDict::MainWindow</name>
+    <message>
+        <location filename="../mainwindow.cpp" line="114"/>
+        <source>About QStarDict</source>
+        <translation>QStarDict Hakkında</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.cpp" line="115"/>
+        <source>&lt;b&gt;QStarDict %1 &lt;/b&gt; - Qt version of StarDict&lt;br&gt;</source>
+        <translation>&lt;b&gt;QStarDict %1 &lt;/b&gt; - StarDict uygulamasının Qt sürümü&lt;br&gt;</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.cpp" line="117"/>
+        <source>Copyright (C) 2007 Alexander Rodin &lt;a href=&quot;http://qstardict.ylsoftware.com&quot;&gt;http://qstardict.ylsoftware.com&lt;/a&gt;</source>
+        <translation>Telif hakları (C) 2007 Alexander Rodin &lt;a href=&quot;http://qstardict.ylsoftware.com&quot;&gt;http://qstardict.ylsoftware.com&lt;/a&gt;</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.ui" line="13"/>
+        <source>QStarDict</source>
+        <translation>QStarDict</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.cpp" line="183"/>
+        <source>%1 - QStarDict</source>
+        <translation>%1 - QStarDict</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.ui" line="58"/>
+        <source>Clear the search box</source>
+        <translation>Arama kutusunu temizle</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.ui" line="61"/>
+        <source>Clear</source>
+        <translation>Temizle</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.ui" line="77"/>
+        <source>Fuzzy query</source>
+        <translation>Belirsiz sorgu</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.ui" line="80"/>
+        <source>Search</source>
+        <translation>Ara</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.ui" line="105"/>
+        <source>Words list</source>
+        <translation>Sözcük Listesi</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.ui" line="147"/>
+        <source>&amp;File</source>
+        <translation>&amp;Dosya</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.ui" line="153"/>
+        <source>&amp;Help</source>
+        <translation>&amp;Yardım</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.ui" line="160"/>
+        <source>&amp;Settings</source>
+        <translation>&amp;Ayarlar</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.ui" line="174"/>
+        <source>&amp;Quit</source>
+        <translation>&amp;Çık</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.ui" line="177"/>
+        <source>Ctrl+Q</source>
+        <translation>Ctrl+Q</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.ui" line="182"/>
+        <source>&amp;About</source>
+        <translation>&amp;Hakkında</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.ui" line="187"/>
+        <source>About &amp;Qt</source>
+        <translation>&amp;Qt Hakkında</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.ui" line="195"/>
+        <source>&amp;Configure QStarDict</source>
+        <translation>&amp;QStarDict Uygulamasını Yapılandır</translation>
+    </message>
+    <message>
+        <location filename="../mainwindow.ui" line="203"/>
+        <source>&amp;Scan</source>
+        <translation>&amp;Tara</translation>
+    </message>
+</context>
+<context>
+    <name>QStarDict::SettingsDialog</name>
+    <message>
+        <location filename="../settingsdialog.cpp" line="68"/>
+        <source>Enabled</source>
+        <translation>Etkin</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="68"/>
+        <source>Name</source>
+        <translation>İsim</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="68"/>
+        <source>Plugin</source>
+        <translation>Eklenti</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="261"/>
+        <source>Information about dictionary &quot;%1&quot;</source>
+        <translation>&quot;%1&quot; sözlüğü hakkında bilgiler</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="279"/>
+        <source>&lt;b&gt;Name:&lt;/b&gt; %1&lt;br&gt;</source>
+        <translation>&lt;b&gt;Adı:&lt;/b&gt; %1&lt;br&gt;</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="263"/>
+        <source>&lt;b&gt;Plugin:&lt;/b&gt; %1&lt;br&gt;</source>
+        <translation>&lt;b&gt;Eklenti:&lt;/b&gt; %1&lt;br&gt;</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="265"/>
+        <source>unknown</source>
+        <translation>Bilinmeyen</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="283"/>
+        <source>&lt;b&gt;Description:&lt;/b&gt; %1</source>
+        <translation>&lt;b&gt;Açıklama:&lt;/b&gt; %1&lt;br&gt;</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="278"/>
+        <source>Information about %1 plugin</source>
+        <translation>%1 eklentisi hakkında bilgiler</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="280"/>
+        <source>&lt;b&gt;Version:&lt;/b&gt; %1&lt;br&gt;</source>
+        <translation>&lt;b&gt;Sürüm:&lt;/b&gt; %1&lt;br&gt;</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="281"/>
+        <source>&lt;b&gt;Authors:&lt;/b&gt; %1&lt;br&gt;</source>
+        <translation>&lt;b&gt;Yazarlar:&lt;/b&gt; %1&lt;br&gt;</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="281"/>
+        <source>&lt;br&gt;</source>
+        <translation>&lt;br&gt;</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="282"/>
+        <source>&lt;b&gt;Can search similar words:&lt;/b&gt; %1&lt;br&gt;</source>
+        <translation>&lt;b&gt;Benzer sözcükler aranamadı:&lt;/b&gt; %1&lt;br&gt;</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="282"/>
+        <source>yes</source>
+        <translation>evet</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="282"/>
+        <source>no</source>
+        <translation>hayır</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="13"/>
+        <source>QStarDict Settings</source>
+        <translation>QStarDict Ayarları</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="16"/>
+        <source>Show information about dictionary</source>
+        <translation>Sözlük hakkındaki bilgileri göster</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="57"/>
+        <source>Global settings</source>
+        <translation>Genel ayarlar</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="63"/>
+        <source>Instant search</source>
+        <translation>Hızlı arama</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="87"/>
+        <source>Pronounce words using this command:</source>
+        <translation>Sözcüğü bu komutu kullanarak telaffuz et:</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="94"/>
+        <source>Enter cmd for the speaching program.&lt;br&gt;If cmd contains &quot;%s&quot; it will be replaced to word, else word will be writen to stdin of speech process.</source>
+        <translation>Telaffuz etme uygulaması için komutu girin.&lt;br&gt;Eğer komut &quot;%s&quot; ise sözcük ile değiştirilecek ve sözcük telaffuz sürecine girdi olarak yazılacak.</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="123"/>
+        <source>Dictionaries</source>
+        <translation>Sözlükler</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="147"/>
+        <source>Move up</source>
+        <translation>Yukarı Taşı</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="150"/>
+        <source>Up</source>
+        <translation>Yukarı</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="160"/>
+        <source>Move down</source>
+        <translation>Aşağı Taşı</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="163"/>
+        <source>Down</source>
+        <translation>Aşağı</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="173"/>
+        <source>Show info</source>
+        <translation>Bilgileri göster</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="207"/>
+        <source>Plugins</source>
+        <translation>Eklentiler</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="234"/>
+        <source>Show information about plugin</source>
+        <translation>Eklenti bilgileini göster</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="237"/>
+        <source>Info</source>
+        <translation>Bilgiler</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="247"/>
+        <source>Configure plugin</source>
+        <translation>Eklentiyi yapılandır</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="250"/>
+        <source>Configure</source>
+        <translation>Yapılandır</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="279"/>
+        <source>Popup window</source>
+        <translation>Yüzen pencere</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="303"/>
+        <source>Behavior</source>
+        <translation>Davranış</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="344"/>
+        <source>Pronounce the word</source>
+        <translation>Sözcüğü telaffuz et</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="370"/>
+        <source>Timeout before hide after mouse over</source>
+        <translation>Fare üzerinden çekildikten sonra gizlemek için geçecek süre</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="377"/>
+        <source>Don&apos;t hide</source>
+        <translation>Gizleme</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="380"/>
+        <source> sec</source>
+        <translation> sn</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="424"/>
+        <source>Show only if modifier pressed</source>
+        <translation>Sadece hızlandırma tuşu baslıysa göster</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="435"/>
+        <source>Alt</source>
+        <translation>Alt</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="440"/>
+        <source>Control</source>
+        <translation>Control</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="445"/>
+        <source>Shift</source>
+        <translation>Shift</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="450"/>
+        <source>Win</source>
+        <translation>Win</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="460"/>
+        <source>Scan selection</source>
+        <translation>Seçimi tara</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="470"/>
+        <source>Show if word not found</source>
+        <translation>Sözcük bulunmazsa göster</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="644"/>
+        <source>Apperance</source>
+        <translation>Görünüm</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="521"/>
+        <source>Opacity</source>
+        <translation>Matlık</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="528"/>
+        <source>%</source>
+        <translation>%</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="566"/>
+        <source>Default width</source>
+        <translation>Öntanımlı genişlik</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.ui" line="605"/>
+        <source>Default height</source>
+        <translation>Öntanımlı yükseklik</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="264"/>
+        <source>&lt;b&gt;Author:&lt;/b&gt; %1&lt;br&gt;</source>
+        <translation>&lt;b&gt;Çeviren:&lt;/b&gt; %1&lt;br&gt;</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="265"/>
+        <source>&lt;b&gt;Words count:&lt;/b&gt; %1&lt;br&gt;</source>
+        <translation>&lt;b&gt;Sözcük sayısı:&lt;/b&gt; %1&lt;br&gt;</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="113"/>
+        <source>All translation</source>
+        <translation>Tüm çeviri</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="114"/>
+        <source>Dictionary name</source>
+        <translation>Sözlük adı</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="115"/>
+        <source>Title</source>
+        <translation>Başlık</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="116"/>
+        <source>Explanation</source>
+        <translation>Açıklama</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="117"/>
+        <source>Abbreviation</source>
+        <translation>Kısaltma</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="118"/>
+        <source>Example</source>
+        <translation>Örnek</translation>
+    </message>
+    <message>
+        <location filename="../settingsdialog.cpp" line="119"/>
+        <source>Transcription</source>
+        <translation>Uyarlama</translation>
+    </message>
+</context>
+<context>
+    <name>QStarDict::TrayIcon</name>
+    <message>
+        <location filename="../trayicon.cpp" line="36"/>
+        <source>QStarDict</source>
+        <translation>QStarDict</translation>
+    </message>
+    <message>
+        <location filename="../trayicon.cpp" line="37"/>
+        <source>&amp;Scan</source>
+        <translation>&amp;Tara</translation>
+    </message>
+    <message>
+        <location filename="../trayicon.cpp" line="47"/>
+        <source>&amp;Configure QStarDict</source>
+        <translation>&amp;QStarDict Uygulamasını Yapılandır</translation>
+    </message>
+    <message>
+        <location filename="../trayicon.cpp" line="50"/>
+        <source>&amp;Quit</source>
+        <translation>&amp;Çık</translation>
+    </message>
+    <message>
+        <location filename="../trayicon.cpp" line="84"/>
+        <source>QStarDict: scanning is %1</source>
+        <translation>QStarDict: Sözcük arama %1</translation>
+    </message>
+    <message>
+        <location filename="../trayicon.cpp" line="84"/>
+        <source>enabled</source>
+        <translation>etkin</translation>
+    </message>
+    <message>
+        <location filename="../trayicon.cpp" line="84"/>
+        <source>disabled</source>
+        <translation>etkin değil</translation>
+    </message>
+</context>
+</TS>
