Index: thunderbird/l10n/tr/mail/chrome/messenger/smtpEditOverlay.dtd
===================================================================
--- l10n/tr/mail/chrome/messenger/smtpEditOverlay.dtd
+++ l10n/tr/mail/chrome/messenger/smtpEditOverlay.dtd
@@ -1,5 +1,5 @@
 <!ENTITY settings.caption "Ayarlar">
-<!ENTITY security.caption "Güvenlik ve kimlik doğrulama">
+<!ENTITY security.caption "Güvenlik ve Kimlik Doğrulama">
 <!ENTITY serverName.label "Sunucu adı:">
 <!ENTITY serverName.accesskey "S">
 <!ENTITY serverDescription.label "Tanım:">
@@ -11,10 +11,10 @@
 <!ENTITY connectionSecurity.label "Bağlantı güvenliği:">
 <!ENTITY connectionSecurity.accesskey "n">
 <!ENTITY connectionSecurityType-0.label "Hiçbiri">
-<!ENTITY connectionSecurityType-1.label "STARTTLS, mevcutsa">
+<!ENTITY connectionSecurityType-1.label "Varsa STARTTLS">
 <!ENTITY connectionSecurityType-2.label "STARTTLS">
 <!ENTITY connectionSecurityType-3.label "SSL/TLS">
 <!ENTITY smtpEditTitle.label "SMTP sunucusu">
 <!ENTITY serverPortDefault.label "Öntanımlı:">
-<!ENTITY authMethod.label "Authentication method:">
+<!ENTITY authMethod.label "Kimlik Doğrulama Yöntemi:">
 <!ENTITY authMethod.accesskey "i">
--- l10n/tr/editor/ui/chrome/composer/editorOverlay.dtd	2010-03-17 21:02:39.000000000 +0200
+++ l10n/tr/editor/ui/chrome/composer/editorOverlay.dtd	2010-04-12 16:00:40.403730799 +0300
@@ -174,7 +174,7 @@
 <!ENTITY fontfaceMenu.accesskey "Y">
 <!ENTITY fontVarWidth.label "Değişken genişlik">
 <!ENTITY fontVarWidth.accesskey "D">
-<!ENTITY fontFixedWidth.label "Sabit en">
+<!ENTITY fontFixedWidth.label "Sabit genişlik">
 <!ENTITY fontFixedWidth.accesskey "S">
 <!ENTITY fontFixedWidth.key "T">
 <!ENTITY fontHelvetica.label "Helvetica, Arial">
--- l10n/tr/mail/chrome/messenger/messengercompose/composeMsgs.properties
+++ l10n/tr/mail/chrome/messenger/messengercompose/composeMsgs.properties
@@ -325,8 +325,8 @@ sendMessageCheckSendButtonLabel=Gönder
 
 ## reply header in composeMsg
 ## <author> wrote:
-mailnews.reply_header_authorwrote=%s yazmış
-mailnews.reply_header_ondate=%s
+mailnews.reply_header_authorwrote=%s yazdı
+mailnews.reply_header_ondate=%s tarihinde
 
 ## reply header in composeMsg
 ## user specified 
