diff -Naur comm-esr17.orig/mail/app/profile/mimeTypes.rdf comm-esr17/mail/app/profile/mimeTypes.rdf
--- comm-esr17.orig/mail/app/profile/mimeTypes.rdf	2012-04-21 00:32:37.000000000 +0200
+++ comm-esr17/mail/app/profile/mimeTypes.rdf	2012-04-30 11:18:39.344000038 +0200
@@ -79,4 +79,49 @@
       </RDF:Seq>
     </NC:MIME-types> 
   </RDF:Description> 
+  <RDF:Description RDF:about="urn:scheme:ftp"
+                   NC:value="ftp">
+    <NC:handlerProp RDF:resource="urn:scheme:handler:ftp"/>
+  </RDF:Description>
+  <RDF:Description RDF:about="urn:schemes">
+    <NC:Protocol-Schemes RDF:resource="urn:schemes:root"/>
+  </RDF:Description>
+  <RDF:Seq RDF:about="urn:mimetypes:root">
+  </RDF:Seq>
+  <RDF:Description RDF:about="urn:scheme:handler:ftp"
+                   NC:alwaysAsk="false">
+    <NC:externalApplication RDF:resource="urn:scheme:externalApplication:ftp"/>
+  </RDF:Description>
+  <RDF:Description RDF:about="urn:scheme:externalApplication:http"
+                   NC:prettyName="firefox"
+                   NC:path="/usr/bin/firefox" />
+  <RDF:Seq RDF:about="urn:schemes:root">
+    <RDF:li RDF:resource="urn:scheme:http"/>
+    <RDF:li RDF:resource="urn:scheme:https"/>
+    <RDF:li RDF:resource="urn:scheme:ftp"/>
+  </RDF:Seq>
+  <RDF:Description RDF:about="urn:scheme:externalApplication:ftp"
+                   NC:prettyName="firefox"
+                   NC:path="/usr/bin/firefox" />
+  <RDF:Description RDF:about="urn:scheme:handler:http"
+                   NC:alwaysAsk="false">
+    <NC:externalApplication RDF:resource="urn:scheme:externalApplication:http"/>
+  </RDF:Description>
+  <RDF:Description RDF:about="urn:scheme:handler:https"
+                   NC:alwaysAsk="false">
+    <NC:externalApplication RDF:resource="urn:scheme:externalApplication:https"/>
+  </RDF:Description>
+  <RDF:Description RDF:about="urn:root"
+                   NC:en-US_defaultHandlersVersion="-1" />
+  <RDF:Description RDF:about="urn:scheme:http"
+                   NC:value="http">
+    <NC:handlerProp RDF:resource="urn:scheme:handler:http"/>
+  </RDF:Description>
+  <RDF:Description RDF:about="urn:scheme:https"
+                   NC:value="https">
+    <NC:handlerProp RDF:resource="urn:scheme:handler:https"/>
+  </RDF:Description>
+  <RDF:Description RDF:about="urn:scheme:externalApplication:https"
+                   NC:prettyName="firefox"
+                   NC:path="/usr/bin/firefox" />
 </RDF:RDF>
diff -Naur comm-esr17.orig/mozilla/build/unix/run-mozilla.sh comm-esr17/mozilla/build/unix/run-mozilla.sh
--- comm-esr17.orig/mozilla/build/unix/run-mozilla.sh	2012-04-21 00:39:55.000000000 +0200
+++ comm-esr17/mozilla/build/unix/run-mozilla.sh	2012-04-30 11:21:34.226000039 +0200
@@ -108,7 +108,7 @@
 	done="no"
 	for d in $debuggers
 	do
-		moz_test_binary /bin/which
+		moz_test_binary /usr/bin/which
 		if [ $? -eq 1 ]
 		then
 			dpath=`which ${d}`	
@@ -154,7 +154,7 @@
 	fi
 	if [ -n "$moz_debugger" ]
 	then
-		moz_test_binary /bin/which
+		moz_test_binary /usr/bin/which
 		if [ $? -eq 1 ]
 		then	
 			debugger=`which $moz_debugger` 
