diff -Naur comm-release.orig/config/autoconf.mk.in comm-release/config/autoconf.mk.in
--- comm-release.orig/config/autoconf.mk.in	2012-04-21 00:32:37.000000000 +0200
+++ comm-release/config/autoconf.mk.in	2012-04-30 10:11:01.229999963 +0200
@@ -67,8 +67,8 @@
 mandir		= @mandir@
 idldir		= $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
 
-installdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir	= $(libdir)/$(MOZ_APP_NAME)
+sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel
 
 MOZDEPTH	= $(DEPTH)/mozilla
 DIST		= $(MOZDEPTH)/dist
diff -Naur comm-release.orig/mozilla/config/autoconf.mk.in comm-release/mozilla/config/autoconf.mk.in
--- comm-release.orig/mozilla/config/autoconf.mk.in	2012-04-21 00:39:55.000000000 +0200
+++ comm-release/mozilla/config/autoconf.mk.in	2012-04-30 10:14:30.472000098 +0200
@@ -74,8 +74,8 @@
 mandir		= @mandir@
 idldir		= $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
 
-installdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir	= $(libdir)/$(MOZ_APP_NAME)
+sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel
 
 DIST		= $(DEPTH)/dist
 LIBXUL_SDK      = @LIBXUL_SDK@
diff -Naur comm-release.orig/mozilla/js/src/config/autoconf.mk.in comm-release/mozilla/js/src/config/autoconf.mk.in
--- comm-release.orig/mozilla/js/src/config/autoconf.mk.in	2012-04-21 00:40:07.000000000 +0200
+++ comm-release/mozilla/js/src/config/autoconf.mk.in	2012-04-30 10:16:52.530000013 +0200
@@ -60,8 +60,8 @@
 datadir		= @datadir@
 mandir		= @mandir@
 
-installdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir	= $(libdir)/$(MOZ_APP_NAME)
+sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel
 
 TOP_DIST	= @TOP_DIST@
 ifneq (,$(filter /%,$(TOP_DIST)))
