Index: gtk-doc-1.15/gtkdoc-fixxref.in
===================================================================
--- gtk-doc-1.15.orig/gtkdoc-fixxref.in
+++ gtk-doc-1.15/gtkdoc-fixxref.in
@@ -461,7 +461,7 @@ sub HighlightSourceVim {
     close (NEWFILE);
 
     # format source
-    system "echo 'let html_number_lines=0|let html_use_css=1|let use_xhtml=1|syn on|e $temp_source_file|run! syntax/2html.vim|wa!|qa!' | @HIGHLIGHT@ -n -e -u /dev/null -T xterm >/dev/null";
+    system "echo 'let html_number_lines=0|let html_use_css=1|let use_xhtml=1|syn on|e $temp_source_file|run! syntax/2html.vim|w! $temp_source_file.html|qa!' | @HIGHLIGHT@ -n -e -u /dev/null -T xterm >/dev/null";
 
     my $highlighted_source;
     {
