diff -Nur gdb-7.1-old/gdb/elfread.c gdb-7.1/gdb/elfread.c
--- gdb-7.1-old/gdb/elfread.c	2010-08-29 23:42:28.986000095 +0300
+++ gdb-7.1/gdb/elfread.c	2010-08-29 23:44:38.488003346 +0300
@@ -1195,6 +1195,10 @@
 void
 debug_print_missing (const char *binary, const char *debug)
 {
+
+  /* Don't mess people with debug messages */
+  return;
+
   size_t binary_len0 = strlen (binary) + 1;
   size_t debug_len0 = debug ? strlen (debug) + 1 : 0;
   struct missing_filepair missing_filepair_find;
