Index: gdb-7.0/gdb/linux-thread-db.c
===================================================================
--- gdb-7.0.orig/gdb/linux-thread-db.c
+++ gdb-7.0/gdb/linux-thread-db.c
@@ -894,7 +894,7 @@ thread_db_load (void)
 	      return 1;
 	  }
 	warning (_("Unable to find libthread_db matching inferior's thread"
-		   " library, thread debugging will not be available."));
+		   " library, thread debugging will not be available. Try installing glibc-dbginfo package."));
 	return 0;
     }
   /* Either this executable isn't using libpthread at all, or it is
