diff -Nur gdb-7.1-old/gdb/config.in gdb-7.1/gdb/config.in
--- gdb-7.1-old/gdb/config.in	2010-08-30 01:04:13.961998035 +0300
+++ gdb-7.1/gdb/config.in	2010-08-30 01:04:33.558999570 +0300
@@ -224,6 +224,9 @@
 /* Define if Python 2.6 is being used. */
 #undef HAVE_LIBPYTHON2_6
 
+/* Define if Python 2.7 is being used. */
+#undef HAVE_LIBPYTHON2_7
+
 /* Define if libunwind library is being used. */
 #undef HAVE_LIBUNWIND
 
diff -Nur gdb-7.1-old/gdb/configure gdb-7.1/gdb/configure
--- gdb-7.1-old/gdb/configure	2010-08-30 01:03:21.151000167 +0300
+++ gdb-7.1/gdb/configure	2010-08-30 01:02:26.896000095 +0300
@@ -9545,6 +9545,44 @@
   have_libpython=no
   if test "${have_libpython}" = no; then
 
+  version=python2.7
+
+
+  HAVE_LIBPYTHON2_7=no
+  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for ${version}" >&5
+$as_echo_n "checking for ${version}... " >&6; }
+  save_LIBS=$LIBS
+  LIBS="$LIBS -l${version}"
+  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+#include "${version}/Python.h"
+int
+main ()
+{
+Py_Initialize ();
+  ;
+  return 0;
+}
+_ACEOF
+if ac_fn_c_try_link "$LINENO"; then :
+  HAVE_LIBPYTHON2_7=yes
+                  have_libpython=yes
+else
+  LIBS=$save_LIBS
+fi
+rm -f core conftest.err conftest.$ac_objext \
+    conftest$ac_exeext conftest.$ac_ext
+  { $as_echo "$as_me:${as_lineno-$LINENO}: result: $HAVE_LIBPYTHON2_7" >&5
+$as_echo "$HAVE_LIBPYTHON2_7" >&6; }
+
+    if test "${HAVE_LIBPYTHON2_7}" = yes; then
+
+$as_echo "#define HAVE_LIBPYTHON2_7 1" >>confdefs.h
+
+    fi
+  fi
+  if test "${have_libpython}" = no; then
+
   version=python2.6
 
 
diff -Nur gdb-7.1-old/gdb/configure.ac gdb-7.1/gdb/configure.ac
--- gdb-7.1-old/gdb/configure.ac	2010-08-30 01:03:21.153000146 +0300
+++ gdb-7.1/gdb/configure.ac	2010-08-30 01:02:26.897000095 +0300
@@ -683,6 +683,12 @@
   LIBS="$LIBS ${python_libs}"
   have_libpython=no
   if test "${have_libpython}" = no; then
+    AC_TRY_LIBPYTHON(python2.7, have_libpython)
+    if test "${HAVE_LIBPYTHON2_7}" = yes; then
+      AC_DEFINE(HAVE_LIBPYTHON2_7, 1, [Define if Python 2.7 is being used.])
+    fi
+  fi
+  if test "${have_libpython}" = no; then
     AC_TRY_LIBPYTHON(python2.6, have_libpython)
     if test "${HAVE_LIBPYTHON2_6}" = yes; then
       AC_DEFINE(HAVE_LIBPYTHON2_6, 1, [Define if Python 2.6 is being used.])
diff -Nur gdb-7.1-old/gdb/python/python-internal.h gdb-7.1/gdb/python/python-internal.h
--- gdb-7.1-old/gdb/python/python-internal.h	2010-08-30 01:03:21.153000146 +0300
+++ gdb-7.1/gdb/python/python-internal.h	2010-08-30 01:02:26.898000095 +0300
@@ -45,6 +45,9 @@
 #elif HAVE_LIBPYTHON2_6
 #include "python2.6/Python.h"
 #include "python2.6/frameobject.h"
+#elif HAVE_LIBPYTHON2_7
+#include "python2.7/Python.h"
+#include "python2.7/frameobject.h"
 #else
 #error "Unable to find usable Python.h"
 #endif
