Index: libffado-2.1.0/SConstruct
===================================================================
--- libffado-2.1.0.orig/SConstruct
+++ libffado-2.1.0/SConstruct
@@ -84,6 +84,30 @@ buildenv=os.environ
 
 env = Environment( tools=['default','scanreplace','pyuic','pyuic4','dbus','doxygen','pkgconfig'], toolpath=['admin'], ENV = buildenv, options=opts )
 
+
+if os.environ.has_key('CC'):
+	env['CC'] = os.environ['CC']
+
+if os.environ.has_key('CXX'):
+	env['CXX'] = os.environ['CXX']
+
+# grab OS CFLAGS / CCFLAGS
+if os.environ.has_key('CFLAGS'):
+	env['OS_CFLAGS'] = os.environ['CFLAGS']
+	env['CFLAGS'] = os.environ['CFLAGS']
+
+if os.environ.has_key('CCFLAGS'):
+	env['OS_CCFLAGS'] = os.environ['CCFLAGS']
+	env['CCFLAGS'] = os.environ['CCFLAGS']
+
+if os.environ.has_key('CXXFLAGS'):
+	env['OS_CCFLAGS'] = os.environ['CXXFLAGS']
+	env['CXXFLAGS'] = os.environ['CXXFLAGS']
+
+if os.environ.has_key('CPPFLAGS'):
+	env['OS_CPPFLAGS'] = os.environ['CPPFLAGS']
+	env['CPPFLAGS'] = os.environ['CPPFLAGS']
+
 if env.has_key('COMPILE_FLAGS') and len(env['COMPILE_FLAGS']) > 0:
     print '''
  * Usage of additional flags is not supported by the ffado-devs.
