--- libffado.old/src/libutil/Configuration.cpp	2010-06-19 14:31:27.000000000 -0400
+++ libffado/src/libutil/Configuration.cpp	2010-07-14 01:37:57.000000000 -0400
@@ -333,7 +333,8 @@ Configuration::getValueForDeviceSetting(
     libconfig::Setting *s = getDeviceSetting( vendor_id, model_id );
     if(s) {
         try {
-            return s->lookupValue(setting, ref);
+            long long int refverylong = ref;
+            return s->lookupValue(setting, refverylong);
         } catch (...) {
             debugOutput(DEBUG_LEVEL_VERBOSE, "Setting %s not found\n", setting.c_str());
             return false;
