--- a/src/bebob/bebob_dl_mgr.h	2012-10-20 21:03:46.961310677 +0200
+++ b/src/bebob/bebob_dl_mgr.h	2012-10-20 21:04:04.148310049 +0200
@@ -32,6 +32,7 @@
 #include "debugmodule/debugmodule.h"
 
 #include <iostream>
+#include <unistd.h>
 
 class Ieee1394Service;
 class ConfigRom;
--- a/src/ffadodevice.h	2012-10-20 21:08:21.043300663 +0200
+++ b/src/ffadodevice.h	2012-10-20 21:08:35.614300131 +0200
@@ -34,6 +34,7 @@
 
 #include <vector>
 #include <string>
+#include <unistd.h>
 
 class DeviceManager;
 class ConfigRom;
--- a/src/libutil/PosixSharedMemory.h	2012-10-20 21:18:47.183277783 +0200
+++ b/src/libutil/PosixSharedMemory.h	2012-10-20 21:19:05.616277112 +0200
@@ -27,6 +27,7 @@
 #include "debugmodule/debugmodule.h"
 
 #include <string>
+#include <unistd.h>
 
 namespace Util {
 
--- a/support/dbus/test-dbus.cpp	2012-10-20 21:18:47.183277783 +0200
+++ b/support/dbus/test-dbus.cpp	2012-10-20 21:19:05.616277112 +0200
@@ -24,6 +24,7 @@
 #include <argp.h>
 #include <stdlib.h>
 #include <iostream>
+#include <unistd.h>
 #include <signal.h>

 #include "controlclient.h"

