Index: xine-ui-0.99.6_20110320/misc/Makefile.am
===================================================================
--- xine-ui-0.99.6_20110320.orig/misc/Makefile.am
+++ xine-ui-0.99.6_20110320/misc/Makefile.am
@@ -41,4 +41,4 @@ xine-bugreport: xine-check
 	test -f xine-bugreport || $(LN_S) xine-check xine-bugreport
 
 install-data-hook:
-	test "x$(DESTDIR)" != "x" || update-mime-database $(mimedir)/..
+	test "x$(DESTDIR)" != "x" || /bin/true
