diff -Naur catfish-0.3/catfish.glade catfish-0.3.tpg/catfish.glade
--- catfish-0.3/catfish.glade	2007-10-03 06:44:45.000000000 +0000
+++ catfish-0.3.tpg/catfish.glade	2007-12-18 02:24:25.000000000 +0000
@@ -19,7 +19,7 @@
             <property name="border_width">5</property>
             <child>
               <widget class="GtkScrolledWindow" id="scrolled_findbar">
-                <property name="width_request">180</property>
+                <property name="width_request">200</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="hscrollbar_policy">GTK_POLICY_AUTOMATIC</property>
