--- smbldap-tools-0.9.5/smbldap_tools.pm	2008-05-22 12:18:05.000000000 +0100
+++ smbldap-tools-0.9.5/smbldap_tools.pm	2008-05-22 15:50:58.000000000 +0100
@@ -826,7 +826,7 @@
         foreach my $attr ( $entry->attributes ) {
             my @vals = $entry->get_value($attr);
             foreach my $val (@vals) {
-                $val = "**UNPRINTABLE**" if ( $val =~ /[^[:print:]]/ );
+                $val = "**UNPRINTABLE**" if ( $val =~ /\P{IsPrint}/ );
             }
             $lines .= $attr . ": " . join( ',', @vals ) . "\n";
         }
@@ -854,7 +854,7 @@
         foreach my $attr ( $entry->attributes ) {
             my @vals = $entry->get_value($attr);
             foreach my $val (@vals) {
-                $val = "**UNPRINTABLE**" if ( $val =~ /[^[:print:]]/ );
+                $val = "**UNPRINTABLE**" if ( $val =~ /\P{IsPrint}/ );
             }
             if (   $attr eq "sambaPwdLastSet"
                 or $attr eq "sambaPwdCanChange"
@@ -1213,7 +1213,7 @@
 
     return to_utf8(
         -string  => $arg,
-        -charset => 'ISO-8859-1',
+        -charset => 'UTF8',
     );
 }
 
@@ -1222,7 +1222,7 @@
 
     return from_utf8(
         -string  => $arg,
-        -charset => 'ISO-8859-1',
+        -charset => 'UTF8',
     );
 }
 
