--- smbldap-tools-0.9.5/smbldap.conf	2008-04-22 09:13:29.000000000 +0100
+++ smbldap-tools-0.9.5/smbldap.conf	2008-04-23 17:00:18.000000000 +0100
@@ -34,12 +34,12 @@
 
 # Put your own SID. To obtain this number do: "net getlocalsid".
 # If not defined, parameter is taking from "net getlocalsid" return
-SID="S-1-5-21-2252255531-4061614174-2474224977"
+#SID="S-1-5-21-2252255531-4061614174-2474224977"
 
 # Domain name the Samba server is in charged.
 # If not defined, parameter is taking from smb.conf configuration file
 # Ex: sambaDomain="IDEALX-NT"
-sambaDomain="DOMSMB"
+#sambaDomain="DOMSMB"
 
 ##############################################################################
 #
@@ -71,7 +71,6 @@
 
 # Master LDAP port
 # If not defined, parameter is set to "389"
-#masterPort="389"
 masterPort="389"
 
 # Use TLS for LDAP
@@ -92,15 +91,15 @@
 
 # CA certificate
 # see "man Net::LDAP" in start_tls section for more details
-cafile="/etc/smbldap-tools/ca.pem"
+cafile="/etc/pki/tls/certs/ldapserverca.pem"
 
 # certificate to use to connect to the ldap server
 # see "man Net::LDAP" in start_tls section for more details
-clientcert="/etc/smbldap-tools/smbldap-tools.iallanis.info.pem"
+clientcert="/etc/pki/tls/certs/ldapclient.pem"
 
 # key certificate to use to connect to the ldap server
 # see "man Net::LDAP" in start_tls section for more details
-clientkey="/etc/smbldap-tools/smbldap-tools.iallanis.info.key"
+clientkey="/etc/pki/tls/certs/ldapclientkey.pem"
 
 # LDAP Suffix
 # Ex: suffix=dc=IDEALX,dc=ORG
@@ -109,7 +108,7 @@
 # Where are stored Users
 # Ex: usersdn="ou=Users,dc=IDEALX,dc=ORG"
 # Warning: if 'suffix' is not set here, you must set the full dn for usersdn
-usersdn="ou=Users,${suffix}"
+usersdn="ou=People,${suffix}"
 
 # Where are stored Computers
 # Ex: computersdn="ou=Computers,dc=IDEALX,dc=ORG"
@@ -119,10 +118,10 @@
 # Where are stored Groups
 # Ex: groupsdn="ou=Groups,dc=IDEALX,dc=ORG"
 # Warning: if 'suffix' is not set here, you must set the full dn for groupsdn
-groupsdn="ou=Groups,${suffix}"
+groupsdn="ou=Group,${suffix}"
 
 # Where are stored Idmap entries (used if samba is a domain member server)
-# Ex: groupsdn="ou=Idmap,dc=IDEALX,dc=ORG"
+# Ex: idmapdn="ou=Idmap,dc=IDEALX,dc=ORG"
 # Warning: if 'suffix' is not set here, you must set the full dn for idmapdn
 idmapdn="ou=Idmap,${suffix}"
 
