--- smbldap-tools-0.9.5/smbldap-useradd	2008-04-22 09:13:29.000000000 +0100
+++ smbldap-tools-0.9.5/smbldap-useradd	2008-05-23 11:26:00.000000000 +0100
@@ -467,7 +467,7 @@
                 system "mkdir $userHomeDirectory 2>/dev/null";
             }
             system
-"chown -R $userName:$userGidNumber $userHomeDirectory 2>/dev/null";
+"chown -R $userUidNumber:$userGidNumber $userHomeDirectory 2>/dev/null";
             if ( defined $config{userHomeDirectoryMode} ) {
                 system
 "chmod $config{userHomeDirectoryMode} $userHomeDirectory 2>/dev/null";
