diff -Nur pyclutter-gst-old/Makefile.am pyclutter-gst/Makefile.am
--- pyclutter-gst-old/Makefile.am	2009-10-09 22:18:02.975880095 +0300
+++ pyclutter-gst/Makefile.am	2009-10-09 22:18:34.097132369 +0300
@@ -1,5 +1,7 @@
 SUBDIRS = build clutter-gst examples
 
+ACLOCAL_AMFLAGS = -I build/autotools
+
 pyclutter-gst-$(PYCLUTTERGST_MAJOR_VERSION).$(PYCLUTTERGST_MINOR_VERSION).pc: pyclutter-gst.pc
 	@cp -f $< $(@F)
 
