diff -Nur pyclutter-gst-old/configure.ac pyclutter-gst/configure.ac
--- pyclutter-gst-old/configure.ac	2009-11-07 00:54:38.914520199 +0200
+++ pyclutter-gst/configure.ac	2009-11-07 00:55:03.838256855 +0200
@@ -4,8 +4,8 @@
 # An odd micro number indicates in-progress development, (eg. from svn)
 # An even micro number indicates a released version.
 m4_define(pycluttergst_version_major, 0)
-m4_define(pycluttergst_version_minor, 9)
-m4_define(pycluttergst_version_micro, 3)
+m4_define(pycluttergst_version_minor, 90)
+m4_define(pycluttergst_version_micro, 1)
 m4_define(pycluttergst_version, pycluttergst_version_major.pycluttergst_version_minor.pycluttergst_version_micro)
 
 m4_define(clutter_required_version,      1.0.0)
@@ -112,7 +112,7 @@
 )
 
 dnl check for clutter-gst
-PKG_CHECK_MODULES(CLUTTERGST, clutter-gst-0.10 >= cluttergst_required_version)
+PKG_CHECK_MODULES(CLUTTERGST, clutter-gst-0.90 >= cluttergst_required_version)
 AC_SUBST(CLUTTERGST_CFLAGS)
 AC_SUBST(CLUTTERGST_LIBS)
 AS_IF([test -n "$export_dynamic"],
