--- configure.ac.orig	2008-11-22 14:30:48.000000000 +0100
+++ configure.ac	2008-11-22 14:36:00.000000000 +0100
@@ -208,8 +208,8 @@
 	[enable_lirc=$enableval],
 	[enable_lirc=no])
 if test x$enable_lirc = xyes; then
-	PKG_CHECK_MODULES([LIBLIRCCLIENT], [liblircclient0],,
-		[AC_MSG_ERROR([liblircclient0 not found])])
+	AC_CHECK_LIB(lirc_client, lirc_init, ,
+		[AC_MSG_ERROR([lirc_client not found])])
 	AC_DEFINE([ENABLE_LIRC], [1], [Enable LIRC support])
 fi
 
