diff -Nuar mozilla-beta.OSmanOS/config/baseconfig.mk mozilla-beta/config/baseconfig.mk
--- mozilla-beta.OSmanOS/config/baseconfig.mk   2013-11-22 19:52:55.000000000 +0200
+++ mozilla-beta/config/baseconfig.mk   2013-11-25 07:05:59.715959204 +0200
@@ -1,7 +1,7 @@
-includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-idldir = $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+includedir := $(includedir)/$(MOZ_APP_NAME)
+idldir = $(datadir)/idl/$(MOZ_APP_NAME)
+installdir = $(libdir)/$(MOZ_APP_NAME)
+sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel
 DIST = $(DEPTH)/dist
 
 # We do magic with OBJ_SUFFIX in config.mk, the following ensures we don't
diff -Nuar mozilla-beta.OSmanOS/js/src/config/baseconfig.mk mozilla-beta/js/src/config/baseconfig.mk
--- mozilla-beta.OSmanOS/js/src/config/baseconfig.mk    2013-11-22 19:53:06.000000000 +0200
+++ mozilla-beta/js/src/config/baseconfig.mk    2013-11-25 07:06:31.429958267 +0200
@@ -1,5 +1,5 @@
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir = $(libdir)/$(MOZ_APP_NAME)
+sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel
 
 ifneq (,$(filter /%,$(TOP_DIST)))
 DIST = $(TOP_DIST)
