diff -Nuar tn5250-0.16.5.orig/linux/Makefile.in tn5250-0.16.5/linux/Makefile.in
--- tn5250-0.16.5.orig/linux/Makefile.in	2005-11-30 17:49:37.000000000 +0200
+++ tn5250-0.16.5/linux/Makefile.in	2005-11-30 17:54:58.000000000 +0200
@@ -253,10 +253,6 @@
 	  if [ "$$(uname -s)" = "Linux" ]; then \
             if which tic >/dev/null 2>&1 ; then \
 	      if [ "`whoami`" = "root" ]; then \
-		rm -f /usr/share/terminfo/x/xterm-5250 ; \
-		rm -f /usr/share/terminfo/5/5250 ; \
-		rm -f /usr/lib/terminfo/x/xterm-5250 ; \
-		rm -f /usr/lib/terminfo/5/5250 ; \
 		good=yes ; \
 		tic $(srcdir)/5250.terminfo || good=no ; \
 	      else \
