--- procps-3.2.5/top.c
+++ procps-3.2.5/top.c
@@ -3285,7 +3285,7 @@
          if (rc <= 0) {
             // EOF is pretty much a "can't happen" except for a kernel bug.
             // We should quickly die via SIGHUP, and thus not spin here.
-            // if (rc == 0) end_pgm(0); /* EOF from terminal */
+            if (rc == 0) end_pgm(0); /* EOF from terminal */
             fcntl(STDIN_FILENO, F_SETFL, file_flags);
             select(1, &fs, NULL, NULL, &tv);
             fcntl(STDIN_FILENO, F_SETFL, O_NONBLOCK|file_flags);
