--- procps-3.2.7/ps/output.c.libselinux	2007-08-27 08:34:13.000000000 +0200
+++ procps-3.2.7/ps/output.c	2007-08-27 08:34:30.000000000 +0200
@@ -1141,7 +1141,7 @@
   char *context;
 
   if(!ps_getpidcon && !tried_load){
-    void *handle = dlopen("libselinux.so", RTLD_NOW);
+    void *handle = dlopen("libselinux.so.1", RTLD_NOW);
     if(handle){
       dlerror();
       ps_getpidcon = dlsym(handle, "getpidcon");
