Description: Patch to set modplug support on by default.
Origin: http://anonscm.debian.org/gitweb/?p=pkg-games/alure.git;a=tree;f=debian/patches

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -169,7 +169,7 @@
 OPTION(FLAC       "FLAC support (for FLAC and Ogg FLAC)" ON)
 OPTION(MPG123     "MPG123 support (for MP1/MP2/MP3)" ON)
 OPTION(DUMB       "DUMB support (for IT/XM/S3M/MOD)" ON)
-OPTION(MODPLUG    "ModPlug support (for IT/XM/S3M/MOD)" OFF)
+OPTION(MODPLUG    "ModPlug support (for IT/XM/S3M/MOD)" ON)
 OPTION(FLUIDSYNTH "FluidSynth support (for MID)" ON)
 
 IF(WIN32)
