Index: alure-1.2/CMakeLists.txt
===================================================================
--- alure-1.2.orig/CMakeLists.txt
+++ alure-1.2/CMakeLists.txt
@@ -144,10 +144,10 @@ CHECK_INCLUDE_FILE(windows.h HAVE_WINDOW
 IF(HAVE_WINDOWS_H)
     CHECK_INCLUDE_FILE(ddk/ntddcdrm.h HAVE_DDK_NTDDCDRM_H)
 ELSE(HAVE_WINDOWS_H)
-    CHECK_C_COMPILER_FLAG(-pthread HAVE_PTHREAD)
+    CHECK_C_COMPILER_FLAG(-lpthread HAVE_PTHREAD)
     IF(HAVE_PTHREAD)
-        ADD_DEFINITIONS(-pthread)
-        SET(EXTRA_LIBS ${EXTRA_LIBS} -pthread)
+        ADD_DEFINITIONS(-lpthread)
+        SET(EXTRA_LIBS ${EXTRA_LIBS} -lpthread)
     ENDIF(HAVE_PTHREAD)
 
     # We need pthreads outside of Windows
