diff -Nuar a/data/default/panels/panel.in b/data/default/panels/panel.in
--- a/data/default/panels/panel.in	2012-05-21 00:55:13.000000000 +0300
+++ b/data/default/panels/panel.in	2013-08-14 19:36:39.000000000 +0300
@@ -2,23 +2,27 @@
 # Use preference dialog in lxpanel to adjust config when you can.
 
 Global {
-    edge=bottom
-    allign=left
+    edge=top
+    allign=center
     margin=0
     widthtype=percent
     width=100
     height=26
-    transparent=0
+    transparent=1
     tintcolor=#000000
     alpha=0
+    autohide=0
+    heightwhenhidden=2
     setdocktype=1
     setpartialstrut=1
     usefontcolor=1
+    fontsize=10
     fontcolor=#ffffff
     usefontsize=0
-    fontsize=10    
-    background=1
-    backgroundfile=@prefix@/share/lxpanel/images/background.png
+    background=0
+    backgroundfile=/usr/share/lxpanel/images/background-pisilinux.png
+    iconsize=24
+    loglevel=2
 }
 
 Plugin {
@@ -31,7 +35,7 @@
 Plugin {
     type = menu
     Config {
-        image=@prefix@/share/lxpanel/images/my-computer.png
+        image=/usr/share/lxde/images/pisi_striped_blue.png
         system {
         }
         separator {
@@ -55,10 +59,10 @@
             id=pcmanfm.desktop
         }
         Button {
-            id=gnome-terminal.desktop
+            id=/usr/share/applications/mozillafirefox.desktop
         }
         Button {
-            id=firefox.desktop
+            id=/usr/share/applications/kde4/package-manager.desktop
         }
     }
 }
@@ -73,8 +77,10 @@
 Plugin {
     type = wincmd
     Config {
+        image=window-manager
         Button1=iconify
         Button2=shade
+        Toggle=0
     }
 }
 
@@ -102,15 +108,13 @@
     Config {
         tooltips=1
         IconsOnly=0
-        AcceptSkipPager=1
-        ShowIconified=1
-        ShowMapped=1
         ShowAllDesks=0
         UseMouseWheel=1
         UseUrgencyHint=1
         FlatButton=0
         MaxTaskWidth=150
         spacing=1
+        GroupedTasks=0
     }
 }
 
@@ -119,6 +123,19 @@
 }
 
 Plugin {
+    type = launchbar
+    Config {
+        Button {
+            id=/usr/share/applications/pavucontrol.desktop
+        }
+    }
+}
+
+Plugin {
+    type = netstat
+}
+
+Plugin {
     type = tray
 }
 
@@ -127,7 +144,9 @@
     Config {
         ClockFmt=%R
         TooltipFmt=%A %x
-        BoldFont=0
+        BoldFont=1
+        IconOnly=0
+        CenterText=0
     }
 }
 
