diff -ruN lxpanel-0.5.9.orig/src/plugins/batt/batt.c lxpanel-0.5.9/src/plugins/batt/batt.c
--- lxpanel-0.5.9.orig/src/plugins/batt/batt.c	2012-04-08 01:27:42.000000000 +0200
+++ lxpanel-0.5.9/src/plugins/batt/batt.c	2012-05-28 15:45:35.376081097 +0200
@@ -428,7 +428,7 @@
 
     /* Apply more default options */
     if (! lx_b->alarmCommand)
-        lx_b->alarmCommand = g_strdup("xmessage Battery low");
+        lx_b->alarmCommand = g_strdup("notify-send 'Battery low' 'Your battery has dropped below the configured level'");
     if (! lx_b->backgroundColor)
         lx_b->backgroundColor = g_strdup("black");
     if (! lx_b->chargingColor1)
