diff -ruN lxpanel-0.5.10.orig/autogen.sh lxpanel-0.5.10/autogen.sh
--- lxpanel-0.5.10.orig/autogen.sh	2012-04-08 01:27:42.000000000 +0200
+++ lxpanel-0.5.10/autogen.sh	2012-08-25 15:51:27.543668289 +0200
@@ -5,7 +5,8 @@
 AM_INSTALLED_VERSION=$($AUTOMAKE --version | sed -e '2,$ d' -e 's/.* \([0-9]*\.[0-9]*\).*/\1/')
 
 if [ "$AM_INSTALLED_VERSION" != "1.10" \
-    -a "$AM_INSTALLED_VERSION" != "1.11" ];then
+    -a "$AM_INSTALLED_VERSION" != "1.11" \
+    -a "$AM_INSTALLED_VERSION" != "1.12" ];then
 	echo
 	echo "You must have automake > 1.10 or 1.11 installed to compile lxpanel."
 	echo "Install the appropriate package for your distribution,"
