diff -burN lxpanel-0.5.1.orig/src/plugins/Makefile.am lxpanel-0.5.1/src/plugins/Makefile.am
--- lxpanel-0.5.1.orig/src/plugins/Makefile.am	2009-07-09 07:02:10.000000000 +0200
+++ lxpanel-0.5.1/src/plugins/Makefile.am	2009-08-04 20:42:03.000000000 +0200
@@ -47,6 +47,3 @@
 	icon.xpm
 
 install-exec-hook:
-	rm -f $(DESTDIR)$(libdir)/lxpanel/plugins/*.la
-	rm -f $(DESTDIR)$(libdir)/lxpanel/plugins/*.a
-	rm -f $(DESTDIR)$(libdir)/lxpanel/plugins/*.
diff -burN lxpanel-0.5.1.orig/src/plugins/netstatus/Makefile.am lxpanel-0.5.1/src/plugins/netstatus/Makefile.am
--- lxpanel-0.5.1.orig/src/plugins/netstatus/Makefile.am	2009-08-02 04:29:06.000000000 +0200
+++ lxpanel-0.5.1/src/plugins/netstatus/Makefile.am	2009-08-04 20:42:36.000000000 +0200
@@ -34,4 +34,3 @@
 	$(LXPANEL_MODULE)
 
 install-exec-hook:
-	rm -f "$(moduledir)\*.la" *.a
