--- lxpanel-0.5.6/src/plugins/netstatus/netstatus-iface.c.orig	2011-07-09 19:40:20.176801266 +0200
+++ lxpanel-0.5.6/src/plugins/netstatus/netstatus-iface.c	2011-07-09 19:41:02.074801250 +0200
@@ -408,7 +408,7 @@
   va_start (args, format);
 
   error_message = g_strdup_vprintf (format, args);
-  error = g_error_new (NETSTATUS_ERROR, code, error_message);
+  error = g_error_new (NETSTATUS_ERROR, code, "%s",error_message);
 
   dprintf (POLLING, "ERROR: %s\n", error->message);
   netstatus_iface_set_error (iface, error);
