diff -upNr dbus-c++.orign/src/Makefile.am dbus-c++/src/Makefile.am
--- dbus-c++.orign/src/Makefile.am	2009-01-08 21:58:42.000000000 +0100
+++ dbus-c++/src/Makefile.am	2009-07-31 11:29:22.871072420 +0200
@@ -24,7 +24,6 @@ HEADER_FILES = \
 	$(HEADER_DIR)/object.h \
 	$(HEADER_DIR)/pendingcall.h \
 	$(HEADER_DIR)/server.h \
-	$(HEADER_DIR)/debug.h \
 	$(HEADER_DIR)/util.h \
 	$(HEADER_DIR)/refptr_impl.h \
 	$(HEADER_DIR)/introspection.h \
