diff -up libmsn-4.0-beta7/msntest/msntest.cpp.openssl libmsn-4.0-beta7/msntest/msntest.cpp
--- libmsn-4.0-beta7/msntest/msntest.cpp.openssl	2009-07-22 22:09:36.000000000 +0200
+++ libmsn-4.0-beta7/msntest/msntest.cpp	2009-08-25 17:34:39.000000000 +0200
@@ -261,7 +261,11 @@ int main()
                     if(mySocketsSsl[i].isSSL && !mySocketsSsl[i].isConnected)
                     {
                         BIO *bio_socket_new;
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+                        const SSL_METHOD *meth=NULL;
+#else
                         SSL_METHOD *meth=NULL;
+#endif
                         meth=SSLv23_client_method();
                         SSLeay_add_ssl_algorithms();
                         mySocketsSsl[i].ctx = SSL_CTX_new(meth);
