diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur kfax-3.3.6-kde4.2.0-orig/kfax.desktop kfax-3.3.6-kde4.2.0/kfax.desktop
--- kfax-3.3.6-kde4.2.0-orig/kfax.desktop	2009-03-17 13:16:02.000000000 +0200
+++ kfax-3.3.6-kde4.2.0/kfax.desktop	2009-03-17 13:16:39.000000000 +0200
@@ -53,4 +53,4 @@
 Terminal=false
 X-KDE-StartupNotify=true
 X-DBUS-StartupType=Multi
-Categories=Qt;KDE;Graphics;X-KDE-More;
+Categories=Qt;KDE;Graphics;
