diff -Naur recode-3.6-orig/Makefile.am recode-3.6/Makefile.am
--- recode-3.6-orig/Makefile.am	2009-05-15 14:31:12.463908436 +0300
+++ recode-3.6/Makefile.am	2009-05-15 14:31:31.540660383 +0300
@@ -18,7 +18,7 @@
 # 02111-1307, USA.
 
 AUTOMAKE_OPTIONS = gnits
-ACLOCAL = ./aclocal.sh @ACLOCAL@
+ACLOCAL = /usr/bin/aclocal
 ACLOCAL_AMFLAGS = -I m4
 
 BABYL = ../rmail/* ../*/rmail/* ../*/*/RMAIL ../*/*/rmail/*
