--- Makefile.orig	2007-12-06 14:57:40.000000000 +0200
+++ Makefile	2009-05-31 01:05:22.691990387 +0300
@@ -242,7 +242,7 @@
   SHLIBLDFLAGS=-shared $(LDFLAGS)
 
   THREAD_LDFLAGS=-lpthread
-  LDFLAGS=-ldl -lm
+  LDFLAGS+=-ldl -lm
   LDFLAGS += -L/emul/linux/x86/usr/lib
   
   CLIENT_LDFLAGS=$(shell sdl-config --libs) -lGL
@@ -549,7 +549,7 @@
 
   THREAD_LDFLAGS=-lpthread
   # don't need -ldl (FreeBSD)
-  LDFLAGS=-lm
+  LDFLAGS+=-lm
 
   CLIENT_LDFLAGS =
 
@@ -1319,13 +1319,13 @@
 
 $(B)/wop-engine.$(ARCH)$(BINEXT): $(Q3OBJ) $(Q3POBJ) $(LIBSDLMAIN)
 	$(echo_cmd) "LD $@"
-	$(Q)$(CC) -o $@ $(Q3OBJ) $(Q3POBJ) $(CLIENT_LDFLAGS) \
-		$(LDFLAGS) $(LIBSDLMAIN)
+	$(Q)$(CC) $(LDFLAGS) $(Q3OBJ) $(Q3POBJ) \
+		$(CLIENT_LDFLAGS) $(LIBSDLMAIN) -o $@
 
 $(B)/wop-smp.$(ARCH)$(BINEXT): $(Q3OBJ) $(Q3POBJ_SMP) $(LIBSDLMAIN)
 	$(echo_cmd) "LD $@"
-	$(Q)$(CC) -o $@ $(Q3OBJ) $(Q3POBJ_SMP) $(CLIENT_LDFLAGS) \
-		$(THREAD_LDFLAGS) $(LDFLAGS) $(LIBSDLMAIN)
+	$(Q)$(CC) $(LDFLAGS) $(THREAD_LDFLAGS) $(Q3OBJ) $(Q3POBJ_SMP) \
+		$(CLIENT_LDFLAGS) $(LIBSDLMAIN) -o $@
 
 ifneq ($(strip $(LIBSDLMAIN)),)
 ifneq ($(strip $(LIBSDLMAINSRC)),)
@@ -1451,7 +1451,7 @@
 
 $(B)/wopded.$(ARCH)$(BINEXT): $(Q3DOBJ)
 	$(echo_cmd) "LD $@"
-	$(Q)$(CC) -o $@ $(Q3DOBJ) $(LDFLAGS)
+	$(Q)$(CC) $(LDFLAGS) $(Q3DOBJ) -ldl -lm -o $@
 
 
 
