--- code/client/snd_dma.c.orig	2007-06-15 18:26:43.000000000 +0300
+++ code/client/snd_dma.c	2007-06-15 18:26:53.000000000 +0300
@@ -1454,7 +1454,7 @@
 		return qfalse;
 	}
 
-	s_khz = Cvar_Get ("s_khz", "22", CVAR_ARCHIVE);
+	s_khz = Cvar_Get ("s_khz", "48", CVAR_ARCHIVE);
 	Cvar_Get("s_compression","0",CVAR_ARCHIVE); // show highquality in menu
 	s_mixahead = Cvar_Get ("s_mixahead", "0.2", CVAR_ARCHIVE);
 	s_mixPreStep = Cvar_Get ("s_mixPreStep", "0.05", CVAR_ARCHIVE);
