diff -Nur lame-398-2-old/configure.in lame-398-2/configure.in
--- lame-398-2-old/configure.in	2009-04-23 02:12:46.000000000 +0300
+++ lame-398-2/configure.in	2009-04-23 02:15:06.000000000 +0300
@@ -385,7 +385,7 @@
 
 dnl configure use of features
 
-AM_PATH_GTK(1.2.0, HAVE_GTK="yes", HAVE_GTK="no")
+HAVE_GTK=no
 
 dnl ElectricFence malloc debugging
 AC_MSG_CHECKING(use of ElectricFence malloc debugging)
diff -Nur lame-398-2-old/frontend/Makefile.am lame-398-2/frontend/Makefile.am
--- lame-398-2-old/frontend/Makefile.am	2009-04-23 02:12:46.000000000 +0300
+++ lame-398-2/frontend/Makefile.am	2009-04-23 02:14:07.000000000 +0300
@@ -51,7 +51,7 @@
 mp3x_SOURCES = mp3x.c gtkanal.c gpkplotting.c $(common_sources)
 endif
 
-CFLAGS = @CFLAGS@ @GTK_CFLAGS@ @FRONTEND_CFLAGS@ @SNDFILE_CFLAGS@
+CFLAGS = @CFLAGS@ @FRONTEND_CFLAGS@ @SNDFILE_CFLAGS@
 LDFLAGS = @LDFLAGS@ @FRONTEND_LDFLAGS@ @SNDFILE_LIBS@
 
 INCLUDES = -I$(top_srcdir)/libmp3lame -I$(top_srcdir)/include -I$(top_builddir)
