diff -urp file-roller-2.26.2.orig/src/fr-enum-types.c file-roller-2.26.2/src/fr-enum-types.c
--- file-roller-2.26.2.orig/src/fr-enum-types.c	2009-04-30 12:03:57.000000000 +0300
+++ file-roller-2.26.2/src/fr-enum-types.c	2009-06-09 22:59:51.038872767 +0300
@@ -2,9 +2,8 @@
 /* Generated data (by glib-mkenums) */
 
 #include <glib-object.h>
+#include <typedefs.h>
 
-/* enumerations from "/home/paolo/workspace/file-roller/src/typedefs.h" */
-#include "/home/paolo/workspace/file-roller/src/typedefs.h"
 GType
 fr_compression_get_type (void)
 {
