--- spamc/configure.orig	2011-02-01 01:46:27.170000038 +0200
+++ spamc/configure	2011-02-01 01:46:59.100000040 +0200
@@ -3730,7 +3730,7 @@
 echo "$as_me:$LINENO: result: $ac_cv_lib_crypto_CRYPTO_lock" >&5
 echo "${ECHO_T}$ac_cv_lib_crypto_CRYPTO_lock" >&6
 if test $ac_cv_lib_crypto_CRYPTO_lock = yes; then
-  SSLLIBS="-lcrypto $SSLLIBS"
+  SSLLIBS="$SSLLIBS"
 fi
 
 	echo "$as_me:$LINENO: checking for SSL_CTX_free in -lssl" >&5
@@ -4041,7 +4041,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-ldl  $LIBS"
+LIBS="$LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
