--- fluidsynth/src/sfloader/fluid_defsfont.c (revision 410)
+++ fluidsynth/src/sfloader/fluid_defsfont.c (revision 428)
@@ -108,4 +108,10 @@
 {
   return fluid_defsfont_get_name((fluid_defsfont_t*) sfont->data);
+}
+
+fluid_sample_t* fluid_defsfont_get_sample(fluid_defsfont_t* sfont, char *s)
+{
+  /* This function is here just to avoid an ABI/SONAME bump, see ticket #98. Should never be used. */
+  return NULL;
 }
 
