diff -Nur liquidwar-5.6.4-old/Makefile.in liquidwar-5.6.4/Makefile.in
--- liquidwar-5.6.4-old/Makefile.in	2007-10-30 13:24:50.041841091 +0200
+++ liquidwar-5.6.4/Makefile.in	2007-10-30 13:26:04.916458419 +0200
@@ -7,22 +7,22 @@
 infodir =         @infodir@
 
 BINDIR =          $(bindir)
-GAMEDIR =         $(exec_prefix)/games
-DATADIR =         $(datadir)/games/liquidwar
-PIXDIR =          $(datadir)/pixmaps
-DOCDIR =          $(datadir)/doc/liquidwar
-DESKTOPDIR =           $(datadir)/applications
+GAMEDIR =         $(bindir)
+DATADIR =         $(datadir)/liquidwar
+PIXDIR = /usr/share/pixmaps
+DOCDIR = /usr/share/doc/$(PF)
+DESKTOPDIR = /usr/share/applnk/Games/
 # We don't use the autoconf default mandir=$(prefix)/man man
 # directory since the FHS (File Hierarchy Standard recommends
 # to place man pages in $(datadir)/man (ie /usr/share/man).
 # MANDIR =          $(mandir)/man6
-MANDIR =          $(datadir)/man/man6
+MANDIR = $(mandir)/man6
 # We don't use the autoconf default infodir=$(prefix)/info info
 # directory since the FHS (File Hierarchy Standard recommends
 # to place info pages in $(datadir)/info (ie /usr/share/info).
 # INFODDIR =        $(infodir)
-INFODIR =         $(datadir)/info
-INFODIRDIR =      $(infodir)
+INFODIR = $(infodir)
+INFODIRDIR = $(infodir)
 
 ifeq (@TARGET_OPT@,yes)
 	TARGET =          @target@
@@ -80,23 +80,23 @@
 all: build_bin build_data build_doc
 	@echo
 	@echo "Build complete."
-	@echo "Type \"$(GMAKE) install\" to install Liquid War $(VERSION)."
+	@echo "Type \"$(MAKE) install\" to install Liquid War $(VERSION)."
 
 world: clean world_clean all package_source package_win32 package_dos package_binary package_source_rpm package_binary_rpm
 
 build_bin:
-	@$(GMAKE) -C $(MAKE_BIN_DIR)
+	@$(MAKE) -C $(MAKE_BIN_DIR)
 
 build_data:
-	@$(GMAKE) -C $(MAKE_UTILS_DIR)
-	@$(GMAKE) -C $(MAKE_DATA_DIR)
+	@$(MAKE) -C $(MAKE_UTILS_DIR)
+	@$(MAKE) -C $(MAKE_DATA_DIR)
 
 build_doc:	
-	@$(GMAKE) -C $(MAKE_DOC_DIR)
+	@$(MAKE) -C $(MAKE_DOC_DIR)
 
 install: install_link install_nolink
 
-# the install_nolink is for distribs where $(prefix)/games is in the PATH,
+# the install_nolink is for distribs where $(prefix) is in the PATH,
 # for instance Debian, where a link in $(prefix)/bin is pretty much useless
 install_nolink: install_bin install_data install_custom_map install_custom_texture install_custom_music install_icon install_doc install_gpl install_desktop install_readme
 	@echo 
@@ -109,9 +109,9 @@
 install_bin: src/liquidwar src/liquidwar-server src/liquidwar-mapgen
 	@echo "Installing liquidwar binaries in $(GAMEDIR)."
 	@install -d $(DESTDIR)$(GAMEDIR)
-	@install -c -s src/liquidwar $(DESTDIR)$(GAMEDIR)
-	@install -c -s src/liquidwar-server $(DESTDIR)$(GAMEDIR)
-	@install -c -s src/liquidwar-mapgen $(DESTDIR)$(GAMEDIR)
+	@install -c src/liquidwar $(DESTDIR)$(GAMEDIR)
+	@install -c src/liquidwar-server $(DESTDIR)$(GAMEDIR)
+	@install -c src/liquidwar-mapgen $(DESTDIR)$(GAMEDIR)
 
 install_link:
 	@echo "Installing link in $(BINDIR)."
@@ -147,9 +147,9 @@
 	@install -c -m 0644 custom/music/*.* $(DESTDIR)$(DATADIR)/music
 
 install_doc:
-	@if [ -f $(MAKE_DOC_DIR)/txt/rules.txt ]; then echo "Installing liquidwar plain text doc in $(DOCDIR)/txt."; install -d $(DESTDIR)$(DOCDIR)/txt; install -c -m 0644 $(MAKE_DOC_DIR)/txt/*.txt $(DESTDIR)$(DOCDIR)/txt; fi
+	@if [ -f $(MAKE_DOC_DIR)/txt/rules.txt ]; then echo "Installing liquidwar plain text doc in $(DOCDIR)."; install -d $(DESTDIR)$(DOCDIR); install -c -m 0644 $(MAKE_DOC_DIR)/txt/*.txt $(DESTDIR)$(DOCDIR); fi
 	@if [ -f $(MAKE_DOC_DIR)/html/rules.html ]; then echo "Installing liquidwar HTML doc in $(DOCDIR)/html."; install -d $(DESTDIR)$(DOCDIR)/html; install -c -m 0644 $(MAKE_DOC_DIR)/html/*.html $(DESTDIR)$(DOCDIR)/html; fi
-	@if [ -f $(MAKE_DOC_DIR)/man/liquidwar.6.gz ]; then echo "Installing liquidwar man page in $(MANDIR)."; install -d $(DESTDIR)$(MANDIR); install -c -m 0644 $(MAKE_DOC_DIR)/man/liquidwar.6.gz $(DESTDIR)$(MANDIR); ln -s -f liquidwar.6.gz $(DESTDIR)$(MANDIR)/liquidwar-server.6.gz; ln -s -f liquidwar.6.gz $(DESTDIR)$(MANDIR)/liquidwar-mapgen.6.gz; fi
+	@if [ -f $(MAKE_DOC_DIR)/man/liquidwar.6.gz ]; then echo "Installing liquidwar man page in $(MANDIR)."; install -d $(DESTDIR)$(MANDIR); install -c -m 0644 $(MAKE_DOC_DIR)/man/liquidwar.6.gz $(DESTDIR)$(MANDIR); ln -f liquidwar.6.gz $(DESTDIR)$(MANDIR)/liquidwar-server.6.gz; ln -s -f liquidwar.6.gz $(DESTDIR)$(MANDIR)/liquidwar-mapgen.6.gz; fi
 	@if [ -f $(MAKE_DOC_DIR)/info/liquidwar.info.gz ]; then echo "Installing info page $(INFODIR)/liquidwar.info.gz."; install -d $(DESTDIR)$(INFODIR); install -c -m 644 $(MAKE_DOC_DIR)/info/liquidwar.info* $(DESTDIR)$(INFODIR); if which install-info; then if test -d $(DESTDIR)$(INFODIRDIR); then if test -f $(DESTDIR)$(INFODIRDIR)/dir; then echo "Installing liquidwar info entry."; install-info --info-dir=$(DESTDIR)$(INFODIRDIR) $(MAKE_DOC_DIR)/info/liquidwar.info.gz; fi; fi; fi; fi
 	@if [ -f $(MAKE_DOC_DIR)/ps/liquidwar.ps ]; then echo "Installing liquidwar PostScript doc in $(DOCDIR)/ps."; install -d $(DESTDIR)$(DOCDIR)/ps; install -c -m 0644 $(MAKE_DOC_DIR)/ps/*.ps $(DESTDIR)$(DOCDIR)/ps; fi
 	@if [ -f $(MAKE_DOC_DIR)/pdf/liquidwar.pdf ]; then echo "Installing liquidwar PDF doc in $(DOCDIR)/pdf."; install -d $(DESTDIR)$(DOCDIR)/pdf; install -c -m 0644 $(MAKE_DOC_DIR)/pdf/*.pdf $(DESTDIR)$(DOCDIR)/pdf; fi
@@ -205,7 +205,7 @@
 	@rm -f $(DESTDIR)$(BINDIR)/liquidwar-server
 
 clean:
-	@for d in $(MAKE_DIRS); do $(GMAKE) -C $$d clean; done
+	@for d in $(MAKE_DIRS); do $(MAKE) -C $$d clean; done
 	@find . -name "*~" -o -name "*#*" -o -name ".*#*" | xargs rm -f 	
 
 world_clean:
@@ -228,21 +228,21 @@
 	@rm -rf LW5 lwdos*exe lwwin*exe _tmpfile.arg *.log vc60.* *.idb *.pch
 	@rm -rf data/lwwin* data/lwdos*
 	@rm -rf starter
-	@for d in $(MAKE_DIRS); do $(GMAKE) -C $$d distclean; done
+	@for d in $(MAKE_DIRS); do $(MAKE) -C $$d distclean; done
 
 distclean: packageclean
 	@rm -rf $(PACKAGE_TMP)
 
 config:
-	@for d in $(MAKE_DIRS); do $(GMAKE) -C $$d config; done
+	@for d in $(MAKE_DIRS); do $(MAKE) -C $$d config; done
 
 check:
-	@for d in $(MAKE_DIRS); do $(GMAKE) -C $$d check; done
+	@for d in $(MAKE_DIRS); do $(MAKE) -C $$d check; done
 
 dep:
-	@for d in $(MAKE_DIRS); do $(GMAKE) -C $$d dep; done
+	@for d in $(MAKE_DIRS); do $(MAKE) -C $$d dep; done
 	@echo
-	@echo "Type \"$(GMAKE)\" to build Liquid War $(VERSION)."
+	@echo "Type \"$(MAKE)\" to build Liquid War $(VERSION)."
 
 # ----------------------------------------------------------------------
 # The following targets are used to produce various packages of LW, ie
@@ -269,7 +269,7 @@
 	@echo "Copying files."
 	@find . -maxdepth 1 \! -path "*$(PACKAGE_TMP)*" \! -name "." -exec cp -r \{\} $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR) \;
 	@echo "Cleaning up."
-	@$(GMAKE) -C $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR) packageclean
+	@$(MAKE) -C $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR) packageclean
 	@touch $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR)/.cvsignore
 	@find $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR) -name .cvsignore -o -name CVS -o -name .arch-ids -o -name "{arch}" | xargs rm -rf
 	@cd $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR) && ./fix.sh unix
@@ -285,9 +285,9 @@
 
 $(PACKAGE_DOS_ZIP): $(PACKAGE_DOS_EXE) $(PACKAGE_DOS_CWSDPMI)
 	@echo "Building data."
-	@$(GMAKE) -C data
+	@$(MAKE) -C data
 	@echo "Building doc."
-	@$(GMAKE) -C doc
+	@$(MAKE) -C doc
 	@echo "Creating temporary directory."
 	@rm -rf $(PACKAGE_TMP)
 	@install -d $(PACKAGE_TMP)/$(PACKAGE_DOS_DIR)
@@ -324,9 +324,9 @@
 
 $(PACKAGE_WIN32_ZIP): $(PACKAGE_WIN32_EXE) $(PACKAGE_WIN32_ALLEGDLL)
 	@echo "Building data."
-	@$(GMAKE) -C data
+	@$(MAKE) -C data
 	@echo "Building doc."
-	@$(GMAKE) -C doc
+	@$(MAKE) -C doc
 	@echo "Creating temporary directory."
 	@rm -rf $(PACKAGE_TMP)
 	@install -d $(PACKAGE_TMP)/$(PACKAGE_WIN32_DIR)
@@ -385,17 +385,17 @@
 	@cd $(PACKAGE_TMP) && tar xzf $(PACKAGE_SOURCE_TARGZ)
 	@cd $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR) && ./configure --enable-static --disable-doc-info --target=$(TARGET) --prefix=$(prefix)
 	@echo "Compiling."
-	@$(GMAKE) -C $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR)
+	@$(MAKE) -C $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR)
 	@echo "Cleaning up install directories."
-	@$(GMAKE) -C $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR) uninstall
+	@$(MAKE) -C $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR) uninstall
 	@echo "Installing."
-	@$(GMAKE) -C $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR) install
+	@$(MAKE) -C $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR) install
 	@echo "Creating tarball."
 	@tar cfP $(PACKAGE_TMP)/$(PACKAGE_BINARY_TAR) $(GAMEDIR)/liquidwar $(GAMEDIR)/liquidwar-server $(GAMEDIR)/liquidwar-mapgen $(BINDIR)/liquidwar $(BINDIR)/liquidwar-server $(BINDIR)/liquidwar-mapgen $(DOCDIR) $(DATADIR) $(MANDIR)/liquidwar.6.gz $(MANDIR)/liquidwar-server.6.gz $(MANDIR)/liquidwar-mapgen.6.gz $(PIXDIR)/liquidwar.xpm $(DESKTOPDIR)/liquidwar.desktop
 	@echo "Compressing."
 	@gzip -c -9 $(PACKAGE_TMP)/$(PACKAGE_BINARY_TAR) > $(PACKAGE_BINARY_TGZ)
 	@echo "Cleaning up install directories."
-	@$(GMAKE) -C $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR) uninstall
+	@$(MAKE) -C $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR) uninstall
 	@echo "Deleting temporary directory."
 	@rm -rf $(PACKAGE_TMP) 
 
@@ -432,7 +432,7 @@
 	#@cd $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR) && ./configure --enable-static --disable-doc-info --target=$(TARGET) --prefix=$(prefix)
 	@cd $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR) && ./configure --disable-doc-info --target=$(TARGET) --prefix=$(prefix)
 	@echo "Compiling."
-	@$(GMAKE) -C $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR)
+	@$(MAKE) -C $(PACKAGE_TMP)/$(PACKAGE_SOURCE_DIR)
 	@echo "Creating directory structure."
 	@install -d "$(PACKAGE_TMP)/Liquid War.app"
 	@install -d "$(PACKAGE_TMP)/Liquid War.app/Contents"
