diff -Nur liquidwar-5.6.4-old/src/Makefile.in liquidwar-5.6.4/src/Makefile.in
--- liquidwar-5.6.4-old/src/Makefile.in	2007-10-30 13:28:00.366907283 +0200
+++ liquidwar-5.6.4/src/Makefile.in	2007-10-30 13:28:13.717877627 +0200
@@ -3,8 +3,8 @@
 exec_prefix =     @exec_prefix@
 datadir =         @datadir@
 
-DATADIR =         $(datadir)/games/liquidwar
-GAMEDIR =         $(exec_prefix)/games
+DATADIR =         $(datadir)/liquidwar
+GAMEDIR =         @bindir@
 
 TARGET_OPT =      @TARGET_OPT@
 STATIC =          @STATIC@
