Index: configure.ac
===================================================================
--- configure.ac.orig
+++ configure.ac
@@ -290,7 +290,7 @@ fi
 
 AC_CHECK_HEADER(libpng12/png.h, HAVE_LIBPNG12_PNG_H=1)
 AC_CHECK_HEADER(libpng14/png.h, HAVE_LIBPNG14_PNG_H=1)
-if test x${HAVE_LIBPNG12_PNG_H} = x1 || test x${HAVE_LIBPNG_14_PNG_H} = x1; then
+if test x${HAVE_LIBPNG12_PNG_H} = x1 || test x${HAVE_LIBPNG14_PNG_H} = x1; then
    if test x${HAVE_LIBPNG12_PNG_H} = x1; then
       LIBPNG_1X_H=libpng12/png.h
    fi
