--- liquidwar-old/src/glouglou.s
+++ liquidwar/src/glouglou.s
@@ -240,3 +240,6 @@
 
    ret
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- liquidwar-old/src/monster.s
+++ liquidwar/src/monster.s
@@ -972,3 +972,6 @@
    ret                              # retour au c
 
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- liquidwar-old/src/spread.s
+++ liquidwar/src/spread.s
@@ -957,3 +957,6 @@
 
 
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
