--- smpeg-0.4.5/Makefile.am.libsupc++	2006-05-01 04:56:06.000000000 +0300
+++ smpeg-0.4.5/Makefile.am	2011-02-05 19:15:59.707558762 +0200
@@ -64,6 +64,13 @@
         -release $(LT_RELEASE)	\
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
+# (gb) hackish to force link with gcc so that we don't get a
+# dependence on libstdc++.so.*. Do add necessary supc++ hereunder.
+CXXLD = $(CC)
+
+libsmpeg_la_LIBADD =		\
+	-lsupc++ -lgcc_s
+
 EXTRA_DIST =			\
 	CHANGES			\
 	COPYING			\
