--- Sources/Display/Display/DirectFB/target_directfb.cpp.orig	2011-01-30 20:35:49.233000043 +0200
+++ Sources/Display/Display/DirectFB/target_directfb.cpp	2011-01-30 20:36:28.674000043 +0200
@@ -158,8 +158,6 @@
     {
     case DSPF_A8:
       return 0;
-    case DSPF_RGB15:
-      return 0x007c00;
     case DSPF_RGB16:
       return 0x00f800;
     case DSPF_RGB24:
@@ -181,8 +179,6 @@
     {
     case DSPF_A8:
       return 0;
-    case DSPF_RGB15:
-      return 0x0003e0;
     case DSPF_RGB16:
       return 0x0007e0;
     case DSPF_RGB24:
@@ -204,7 +200,6 @@
     {
     case DSPF_A8:
       return 0;
-    case DSPF_RGB15:
     case DSPF_RGB16:
       return 0x00001f;
     case DSPF_RGB24:
@@ -226,7 +221,6 @@
     {
     case DSPF_A8:
       return 0xff;
-    case DSPF_RGB15:
     case DSPF_RGB16:
     case DSPF_RGB24:
     case DSPF_RGB32:
