Index: Sources/API/GL1/opengl1_defines.h
===================================================================
--- Sources/API/GL1/opengl1_defines.h	(revision 8911)
+++ Sources/API/GL1/opengl1_defines.h	(revision 8912)
@@ -1324,8 +1324,8 @@
 	// ARB_shader_objects:
 
 	GL_PROGRAM_OBJECT                      = 0x8B40,
-	GL_TYPE                                = 0x8B4E,
-	GL_SUBTYPE                             = 0x8B4F,
+	//GL_TYPE                                = 0x8B4E,
+	//GL_SUBTYPE                             = 0x8B4F,
 #endif
 #ifndef GL_DELETE_STATUS 
 	GL_DELETE_STATUS                       = 0x8B80,
