--- ClanLib-0.6.5/Sources/Display/Input/DirectFB/keyboard_directfb.cpp.directfb-fixes	2002-06-07 21:36:33.000000000 +0200
+++ ClanLib-0.6.5/Sources/Display/Input/DirectFB/keyboard_directfb.cpp	2006-05-10 16:27:33.000000000 +0200
@@ -243,7 +243,6 @@
     case DIKI_SHIFT_R:     return CL_KEY_RSHIFT;
     case DIKI_ALT_L:       return CL_KEY_ALT;
     case DIKI_ALT_R:       return CL_KEY_ALT;
-    case DIKI_ALTGR:       return CL_KEY_ALTGR;
     case DIKI_TAB:         return CL_KEY_TAB;
     case DIKI_ENTER:       return CL_KEY_ENTER;
     case DIKI_SPACE:       return CL_KEY_SPACE;
@@ -337,7 +336,7 @@
     case CL_KEY_LSHIFT:    return DIKI_SHIFT_L;
     case CL_KEY_RSHIFT:    return DIKI_SHIFT_R;
     case CL_KEY_ALT:       return DIKI_ALT_L;
-    case CL_KEY_ALTGR:     return DIKI_ALTGR;
+    case CL_KEY_ALTGR:     return DIKI_ALT_R;
     case CL_KEY_TAB:       return DIKI_TAB;
     case CL_KEY_ENTER:     return DIKI_ENTER;
     case CL_KEY_SPACE:     return DIKI_SPACE;
--- ClanLib-0.6.5/Sources/Display/Display/DirectFB/target_directfb.cpp.directfb-fixes	2006-05-10 16:23:50.000000000 +0200
+++ ClanLib-0.6.5/Sources/Display/Display/DirectFB/target_directfb.cpp	2006-05-10 16:23:51.000000000 +0200
@@ -17,6 +17,17 @@
 #ifdef USE_DIRECTFB
 
 #include <Display/Display/DirectFB/target_directfb.h>
+#include <directfb_version.h>
+
+#define DIRECTFB_PREREQ(X, Y, Z) \
+		(((DIRECTFB_MAJOR_VERSION << 16) + \
+		  (DIRECTFB_MINOR_VERSION <<  8) + \
+		   DIRECTFB_MICRO_VERSION) >= (((X) << 16) + ((Y) << 8) + (Z)))
+
+#if DIRECTFB_PREREQ(0,9,21)
+/* From 0.9.21 NEWS: Removed DSPF_RGB15 in the flavor of DSPF_ARGB1555. */
+#define DSPF_RGB15 DSPF_ARGB1555
+#endif
 
 CL_Target_DirectFB::CL_Target_DirectFB (IDirectFBSurface *surface)
 {
