--- ./Setup/Unix/Makefile.conf.in.gege	2002-04-06 14:54:58.000000000 +0200
+++ ./Setup/Unix/Makefile.conf.in	2003-01-20 11:12:52.000000000 +0100
@@ -33,19 +33,19 @@
 	@echo "Compiling $<"
 	-@if [ ! -d Libs ]; then mkdir Libs; fi
 	-@if [ ! -d Libs/Intermediate ]; then mkdir Libs/Intermediate; fi
-	@$(CXX_COMMAND) $< -o $@
+	$(CXX_COMMAND) $< -o $@
 
 Libs/Intermediate/%.o : %.c
 	@echo "Compiling $<"
 	-@if [ ! -d Libs ]; then mkdir Libs; fi
 	-@if [ ! -d Libs/Intermediate ]; then mkdir Libs/Intermediate; fi
-	@$(CC_COMMAND) -c $< -o $@
+	$(CC_COMMAND) -c $< -o $@
 
 Libs/Intermediate/%.o : %.s
 	@echo "Assembling $<"
 	-@if [ ! -d Libs ]; then mkdir Libs; fi
 	-@if [ ! -d Libs/Intermediate ]; then mkdir Libs/Intermediate; fi
-	@$(AS_COMMAND) $< -o $@
+	$(AS_COMMAND) $< -o $@
 
 build_info:
 	@echo Compile command .cpp = $(CXX_COMMAND)
--- ./Setup/Unix/Makefile.core.in.gege	2002-01-08 22:58:35.000000000 +0100
+++ ./Setup/Unix/Makefile.core.in	2003-01-20 11:12:54.000000000 +0100
@@ -92,12 +92,12 @@
 
 Libs/libclanCore.so: $(OBJF_CORE_ALL)
 	@echo "Building Libs/libclanCore.so"
-	@$(LINK_COMMAND) -Wl,-soname=libclanCore.so.$(D_VERSION_MAJOR) -o Libs/libclanCore.so.$(D_VERSION_MINOR) $(OBJF_CORE_ALL) $(LINK_CORE)
-	@ln -s -f libclanCore.so.$(D_VERSION_MINOR) Libs/libclanCore.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanCore.so.$(D_VERSION_MAJOR) Libs/libclanCore.so
+	$(LINK_COMMAND) -Wl,-soname=libclanCore.so.$(D_VERSION_MAJOR) -o Libs/libclanCore.so.$(D_VERSION_MINOR) $(OBJF_CORE_ALL) $(LINK_CORE)
+	ln -s -f libclanCore.so.$(D_VERSION_MINOR) Libs/libclanCore.so.$(D_VERSION_MAJOR)
+	ln -s -f libclanCore.so.$(D_VERSION_MAJOR) Libs/libclanCore.so
 
 Libs/libclanCore.a: $(OBJF_CORE_ALL)
 	@echo "Building Libs/libclanCore.a"
 	-@if [ -e Libs/libclanCore.a ]; then rm -f Libs/libclanCore.a; fi
-	@ar qc Libs/libclanCore.a $(OBJF_CORE_ALL)
+	ar qc Libs/libclanCore.a $(OBJF_CORE_ALL)
 
