diff -Nur ClanLib-0.6.5-old//Setup/Unix/Makefile.conf.in ClanLib-0.6.5/Setup/Unix/Makefile.conf.in
--- ClanLib-0.6.5-old//Setup/Unix/Makefile.conf.in	2011-01-31 16:11:42.757000625 +0200
+++ ClanLib-0.6.5/Setup/Unix/Makefile.conf.in	2011-01-31 16:15:38.251000389 +0200
@@ -21,7 +21,7 @@
 
 OBJF_NONDYN   = @objf_nondyn@
 
-LINK_COMMAND  = @link_command@ @extra_libraries@
+LINK_COMMAND  = @link_command@ $(LDFLAGS) @extra_libraries@
 
 CXX_COMMAND = @CXX@ @CXXFLAGS@ $(COMP_OPTIONS) -c 
 
