--- engine/include/stratagus.h.orig	2007-07-04 00:57:44.000000000 +0300
+++ engine/include/stratagus.h	2007-08-20 15:54:16.000000000 +0300
@@ -186,7 +186,7 @@
 extern std::string NameLine;
 
 #ifndef STRATAGUS_LIB_PATH
-#define STRATAGUS_LIB_PATH  "."      /// Where to find the data files
+#define STRATAGUS_LIB_PATH  "/usr/share/boswars"      /// Where to find the data files
 #endif
 
 #ifndef STRATAGUS_HOME_PATH
