--- gpm-1.20.7/doc/Makefile.in.out_of_source~	2012-10-31 08:09:47.136931965 +0100
+++ gpm-1.20.7/doc/Makefile.in	2012-10-31 08:10:03.633131205 +0100
@@ -32,7 +32,7 @@ include $(top_builddir)/Makefile.include
 
 MANPAGES = gpm.8 mev.1 gpm-root.1 gpm-types.7 mouse-test.1
 
-all: $(srcdir)/gpm.info gpm.man
+all: gpm.info gpm.man
 
 # HTML (texi2html)
 %.html: %.texinfo
--- gpm-1.20.7/Makefile.in.out_of_source~	2012-10-31 08:01:43.161121617 +0100
+++ gpm-1.20.7/Makefile.in	2012-10-31 08:01:43.170121722 +0100
@@ -39,24 +39,6 @@ uninstall:	do-uninstall
 clean: do-clean
 	rm -f config.status.lineno
 
-aclocal.m4: acinclude.m4
-	$(MKDIR) config
-	aclocal -I config
-
-configure: configure.ac aclocal.m4 $(versionfiles)
-	libtoolize --force --copy `libtoolize -n --install >/dev/null 2>&1 && echo --install`
-	autoheader
-	autoconf
-
-config.status:	configure
-	if [ -f config.status ]; then $(SHELL) ./config.status --recheck; \
-	else $(SHELL) ./configure; fi
-
-Makefile: config.status         $(srcdir)/Makefile.in Makefile.include
-	./config.status
-Makefile.include: config.status $(srcdir)/Makefile.include.in
-	./config.status
-
 ### INSTALL
 
 install:	installdirs do-install
--- gpm-1.20.7/src/Makefile.in.out_of_source~	2012-10-31 08:01:43.161121617 +0100
+++ gpm-1.20.7/src/Makefile.in	2012-10-31 08:01:43.170121722 +0100
@@ -28,6 +28,7 @@ GSRC = mice.c twiddler.c synaptics.c
 		daemon/startup.c daemon/wait_text.c
 
 GOBJ = $(GSRC:.c=.o) report.o tools.o
+GOBJ := $(patsubst %,$(srcdir)/%,$(GOBJ))
 
 LSRC = lib/liblow.c lib/libhigh.c lib/libxtra.c lib/report-lib.c
 
@@ -139,14 +140,9 @@ uninstall:
 		rm -f $(bindir)/$$i ;\
 	done
 
-# Configure & unconfigure
-
-Makefile: Makefile.in
-	$(top_builddir)/config.status Makefile
-
 # Subdirectory prog/
 prog/gpm-root.c:	$(srcdir)/prog/gpm-root.y
-	$(YACC) $(YFLAGS) $< && mv y.tab.c prog/gpm-root.c
+	$(YACC) $(YFLAGS) $< && mv y.tab.c $(srcdir)/prog/gpm-root.c
 
 # gpm-root needs an own rule, because gpm-root.c is not in $(srcdir)
 prog/gpm-root: prog/gpm-root.c lib/libgpm.so.@abi_lev@
