--- gpm-1.20.1/doc/Makefile.in.multilib	2002-12-24 17:57:16.000000000 -0500
+++ gpm-1.20.1/doc/Makefile.in	2004-10-20 17:25:11.000000000 -0400
@@ -97,7 +97,7 @@
 
 # Main portion
 
-all: $(srcdir)/gpm.info $(MANPAGES)
+all: gpm.info $(MANPAGES)
 
 # why gpmdoc.ps and gpm.ps??
 # there is no gpm.ps in my tree and no rule to generate gpm.ps.
@@ -131,8 +131,8 @@
 	#i keep all my infopages compressed and i'm tired to do it by
 	#hand, so check if there are any compressed pages and do this
 	#one too
-	-ls $(infodir)/*[-.]info.gz >/dev/null 2>&1 \
-		&& gzip -f $(infodir)/gpm.info
+	#-ls $(infodir)/*[-.]info.gz >/dev/null 2>&1 \
+	#	&& gzip -f $(infodir)/gpm.info
 	# Hmm.... shouldn't man pages be compressed too?
    # maybe they should, but at least at my system they are not.
 
--- gpm-1.20.1/contrib/Makefile.in.multilib	2002-12-24 17:57:16.000000000 -0500
+++ gpm-1.20.1/contrib/Makefile.in	2004-10-20 17:11:38.000000000 -0400
@@ -9,10 +9,10 @@
 
 include $(top_builddir)/Makefile.include
 
-all: $(srcdir)/$(ELISP)
+all: $(filter-out %.elc,$(srcdir)/$(ELISP))
 
 install: all
-	if [ -n "$(ELISP)" ]; then for i in `echo $(ELISP)`; do \
+	if [ -n "$(filter-out %.elc,$(ELISP))" ]; then for i in `echo $(filter-out %.elc,$(ELISP))`; do \
 		$(INSTALL_DATA) $(srcdir)/$$i $(lispdir)/`basename $$i` ;\
 	done; fi
 
@@ -20,7 +20,7 @@
 	$(EMACS) -batch -l $(srcdir)/emacs/exec.el -exec '(byte-compile-file "$<")'
 
 uninstall:
-	if [ -n "$(ELISP)" ]; then for i in `echo $(ELISP)`; do \
+	if [ -n "$(filter-out %.elc,$(ELISP))" ]; then for i in `echo $(filter-out %.elc,$(ELISP))`; do \
 		rm -f $(lispdir)/$$i ;\
 	done; fi 
 
@@ -28,4 +28,4 @@
 	$(CP) -r $(srcdir) $(top_builddir)/gpm-$(release)/
 
 clean distclean:
-	$(RM) -f $(srcdir)/emacs/*.elc Makefile
+	$(RM) $(srcdir)/emacs/*.elc Makefile
