diff -Nur AssaultCube-old/source/src/protocol.h AssaultCube/source/src/protocol.h
--- AssaultCube-old/source/src/protocol.h	2007-02-12 19:38:08.764306604 +0200
+++ AssaultCube/source/src/protocol.h	2007-02-12 19:39:29.854978948 +0200
@@ -1,5 +1,5 @@
 #define MAXCLIENTS 256                  // in a multiplayer game, can be arbitrarily changed
-#define DEFAULTCLIENTS 6
+#define DEFAULTCLIENTS 16
 #define MAXTRANS 5000                   // max amount of data to swallow in 1 go
 #define CUBE_SERVER_PORT 28763
 #define CUBE_SERVINFO_PORT 28764
