From 89b696fcb9fb4b34e79f4cf97b42057fe0c4c2ab Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 29 May 2012 20:43:34 +0100
Subject: [PATCH 4/6] Don't add rpaths to libraries.

---
 tools/Makefile.shared |    3 ---
 1 file changed, 3 deletions(-)

diff --git a/tools/Makefile.shared b/tools/Makefile.shared
index f6818d3..aadd7e2 100644
--- a/tools/Makefile.shared
+++ b/tools/Makefile.shared
@@ -114,9 +114,6 @@ ocamlmklib.ml: ocamlmklib.mlp ../config/Makefile
 	sed -e "s|%%BINDIR%%|$(BINDIR)|" \
 	    -e "s|%%SUPPORTS_SHARED_LIBRARIES%%|$(SUPPORTS_SHARED_LIBRARIES)|" \
 	    -e "s|%%MKSHAREDLIB%%|$(MKSHAREDLIB)|" \
-	    -e "s|%%BYTECCRPATH%%|$(BYTECCRPATH)|" \
-	    -e "s|%%NATIVECCRPATH%%|$(NATIVECCRPATH)|" \
-	    -e "s|%%MKSHAREDLIBRPATH%%|$(MKSHAREDLIBRPATH)|" \
 	    -e "s|%%RANLIB%%|$(RANLIB)|" \
 	  ocamlmklib.mlp >> ocamlmklib.ml
 
-- 
1.7.10.4

