Index: kdeartwork-4.5.5/kscreensaver/kxsconfig/CMakeLists.txt
===================================================================
--- kdeartwork-4.5.5.orig/kscreensaver/kxsconfig/CMakeLists.txt
+++ kdeartwork-4.5.5/kscreensaver/kxsconfig/CMakeLists.txt
@@ -39,6 +39,7 @@ foreach(_currentdesktopfile ${_desktopfi
 	#MESSAGE(STATUS "newItem <${_newitem}>")
 	GET_FILENAME_COMPONENT(_screensaverName ${_newitem} NAME_WE)
 	#MESSAGE(STATUS "name without extension <${_screensaverName}>")
+    unset(XSCREENSAVER_FILE_FOUND CACHE)
 	find_file(XSCREENSAVER_FILE_FOUND ${_screensaverName}.xml PATHS ${XSCREENSAVER_CONFIG_DIR} )
 	if(XSCREENSAVER_FILE_FOUND)
 		MESSAGE(STATUS "xscreensaver name ${_screensaverName} found")
