diff -N -r -U5 kdeartwork-4.3.75svn1048496/kscreensaver/kxsconfig/ScreenSavers/xscreensaver.template kdeartwork-4.3.75svn1048496.template/kscreensaver/kxsconfig/ScreenSavers/xscreensaver.template
--- kdeartwork-4.3.75svn1048496/kscreensaver/kxsconfig/ScreenSavers/xscreensaver.template	2009-09-16 07:43:46.000000000 -0400
+++ kdeartwork-4.3.75svn1048496.template/kscreensaver/kxsconfig/ScreenSavers/xscreensaver.template	2009-11-22 02:35:11.000000000 -0500
@@ -10,16 +10,185 @@
 Name=@LABEL@
 
 [Desktop Action Setup]
 Exec=kxsconfig @NAME@
 Name=Setup...
+Name[af]=Opstelling...
+Name[bg]=Настройки...
+Name[br]=Kefluniadur ...
+Name[ca]=Arranjament...
+Name[cs]=Nastavení...
+Name[cy]=Gosod ...
+Name[da]=Opsætning...
+Name[de]=Einrichtung ...
+Name[el]=Ρύθμιση...
+Name[eo]=Agordo...
+Name[es]=Configuración...
+Name[et]=Seadistamine...
+Name[eu]=Konfiguratu...
+Name[fa]=برپایی...
+Name[fi]=Asetukset...
+Name[fr]=Configuration...
+Name[fy]=Opset...
+Name[ga]=Socrú...
+Name[gl]=Configuración...
+Name[he]=הגדרות...
+Name[hne]=सेटअप...
+Name[hu]=Beállítás...
+Name[is]=Stillingar...
+Name[it]=Impostazioni...
+Name[ja]=設定...
+Name[lt]=Sąranka ...
+Name[lv]=Iestatījumi...
+Name[mai]=सेटअप...
+Name[mk]=Поставување...
+Name[ml]=ഒരുക്കം...
+Name[ms]=Setkan...
+Name[nb]=Innstillinger …
+Name[nds]=Instellen...
+Name[ne]=सेटअप...
+Name[nl]=Instellingen...
+Name[nn]=Oppsett …
+Name[pa]=ਸੈਟਅੱਪ...
+Name[pl]=Ustawienia...
+Name[pt]=Configurar...
+Name[pt_BR]=Configurar...
+Name[ro]=Configurare...
+Name[ru]=Настройка...
+Name[sk]=Nastavenie...
+Name[sl]=Nastavi ...
+Name[sr]=Подеси...
+Name[sr@latin]=Podesi...
+Name[sv]=Inställningar...
+Name[ta]=அமைப்பு...
+Name[tg]=Барпосозӣ...
+Name[th]=ตั้งค่า...
+Name[tr]=Kurulum...
+Name[uk]=Встановити...
+Name[uz]=Oʻrnatish
+Name[uz@cyrillic]=Ўрнатиш
+Name[vi]=Thiết lập...
+Name[xh]=Iyacwangciswa...
+Name[x-test]=xxSetup...xx
+Name[zh_CN]=设置...
+Name[zh_TW]=設定...
 
 [Desktop Action InWindow]
 Exec=kxsrun @NAME@ -- -window-id %w
 Name=Display in Specified Window
+Name[bg]=Показване в избран прозорец
+Name[br]=Diskwel er prenestr spisaet
+Name[ca]=Mostra a la finestra especificada
+Name[cs]=Zobrazit v určeném okně
+Name[cy]=Dangos mewn Ffenestr Benodol
+Name[da]=Visning i angivet vindue
+Name[de]=In bestimmten Fenster anzeigen
+Name[el]=Προβολή στο καθορισμένο παράθυρο
+Name[eo]=Montri en indikita fenestro
+Name[es]=Mostrar en la ventana especificada
+Name[et]=Määratud aknas näitamine
+Name[eu]=Bistaratu zehaztutako lehioan
+Name[fa]=نمایش در پنجرۀ مشخص‌شده
+Name[fi]=Näytä määrätyssä ikkunassa
+Name[fr]=Affichage dans la fenêtre spécifiée
+Name[fy]=Werjefte yn in definiearre finster
+Name[ga]=Taispeáin san Fhuinneog Sonraithe
+Name[gl]=Mostrar na fiestra indicada
+Name[he]=הצג בחלון המצוין
+Name[hne]=निरधारित विंडो मं देखाव
+Name[hu]=Megjelenítés a megadott ablakban
+Name[is]=Birta í völdum glugga
+Name[it]=Mostra nella finestra specificata
+Name[ja]=指定したウィンドウで表示
+Name[lt]=Rodyti nurodytame lange
+Name[lv]=Rādīt norādītajā logā
+Name[mai]=निर्दिष्ट विंडो मे प्रदर्शित करू
+Name[mk]=Прикажи во наведениот прозорец
+Name[ml]=പ്രത്യേകം പറഞ്ഞിരിക്കുന്ന ജാലകത്തില്‍ പ്രദര്‍ശിപ്പിക്കുക
+Name[ms]=Papar dalam Tetingkap Yang Dinyatakan
+Name[nb]=Vis i angitt vindu
+Name[nds]=In utsöcht Finster wiesen
+Name[ne]=निर्दिष्ट सञ्झ्यालमा प्रदर्शन
+Name[nl]=Weergeven In een opgegeven venster
+Name[nn]=Vis i oppgjeve vindauge
+Name[pa]=ਦੱਸੇ ਝਰੋਖੇ ਵਿੱਚ ਵੇਖਾਓ
+Name[pl]=Wyświetl w zadanym oknie
+Name[pt]=Mostrar na Janela Indicada
+Name[pt_BR]=Exibir na janela especificada
+Name[ro]=Afișează în fereastra specificată
+Name[ru]=Показывать в указанном окне
+Name[sk]=Zobraziť v zadanom okne
+Name[sl]=Prikaz v določenemu oknu
+Name[sr]=Прикажи у наведеном прозору
+Name[sr@latin]=Prikaži u navedenom prozoru
+Name[sv]=Visa i angivet fönster
+Name[ta]=குறித்த சாளரத்தில் காட்டு
+Name[tg]=Дар тирезаи нишондода намоиш намоед
+Name[th]=แสดงในหน้าต่างที่กำหนด
+Name[tr]=Belirtilen Pencerede Göster
+Name[uk]=Відобразити у вказаному вікні
+Name[vi]=Hiển thị trong Cửa sổ Đã định
+Name[x-test]=xxDisplay in Specified Windowxx
+Name[zh_CN]=在指定的窗口中显示
+Name[zh_TW]=在指定的視窗中顯示
 NoDisplay=true
 
 [Desktop Action Root]
 Exec=kxsrun @NAME@ -- -root
 Name=Display in Root Window
+Name[bg]=Показване в главния прозорец
+Name[br]=Diskwel er prenestr gwrizienn
+Name[ca]=Mostra a la finestra arrel
+Name[cs]=Zobrazit na pozadí plochy
+Name[cy]=Dangos mewn Ffenestr Wraidd
+Name[da]=Visning i root-vindue
+Name[de]=In Hintergrundfenster anzeigen
+Name[el]=Προβολή στο βασικό παράθυρο
+Name[eo]=Montri en radika fenestro
+Name[es]=Mostrar en la ventana raíz
+Name[et]=Juuraknas näitamine
+Name[eu]=Bistaratu erro lehioan
+Name[fa]=نمایش در پنجرۀ ریشه
+Name[fi]=Näytä juuri-ikkunassa
+Name[fr]=Affichage dans la fenêtre racine
+Name[fy]=Werjefte yn haadfinster
+Name[ga]=Taispeáin sa Phríomhfhuinneog
+Name[gl]=Mostrar na fiestra raíz
+Name[he]=הצג בחלון השורש
+Name[hne]=रूट विंडो मं देखाव
+Name[hu]=Megjelenítés a gyökérablakban
+Name[is]=Birta í rótarglugganum
+Name[it]=Mostra sullo sfondo
+Name[ja]=ルートウィンドウで表示
+Name[lt]=Rodyti root lange
+Name[lv]=Rādīt saknes logā
+Name[mai]=रूट विंडो मे प्रदर्शित करू
+Name[mk]=Прикажи во коренскиот прозорец
+Name[ml]=മൂല ജാലകത്തില്‍ പ്രദര്‍ശിപ്പിക്കുക
+Name[ms]=Papar dalam Tetingkap Root
+Name[nb]=Vis i root-vindu
+Name[nds]=In Hööftfinster wiesen
+Name[ne]=मूल सञ्झ्यालमा प्रदर्शन
+Name[nl]=Weergeven in hoofdvenster
+Name[nn]=Vis i rotvindauget
+Name[pa]=Root ਝਰੋਖੇ ਵਿੱਚ ਵੇਖਾਓ
+Name[pl]=Wyświetl w oknie głównym
+Name[pt]=Mostrar na Janela de Fundo
+Name[pt_BR]=Exibir na janela raiz
+Name[ro]=Afișează în fereastra rădăcină
+Name[ru]=Показывать в корневом окне
+Name[sk]=Zobraziť v koreňovom okne
+Name[sl]=Prikaz v korenskem oknu
+Name[sr]=Прикажи у кореном прозору
+Name[sr@latin]=Prikaži u korenom prozoru
+Name[sv]=Visa i rotfönstret
+Name[ta]=ஆரம்ப சாளரத்தில் காட்டு
+Name[tg]=Дар тирезаи решагӣ намоиш намоед
+Name[th]=แสดงในหน้าต่างราก
+Name[tr]=Kök Pencerede Göster
+Name[uk]=Відобразити в кореневому вікні
+Name[vi]=Hiển thị trong Cửa sổ Gốc
+Name[x-test]=xxDisplay in Root Windowxx
+Name[zh_CN]=在根窗口中显示
+Name[zh_TW]=在根視窗中顯示
 NoDisplay=true
 
diff -N -r -U5 kdeartwork-4.3.75svn1048496/kscreensaver/kxsconfig/screensaver.sh kdeartwork-4.3.75svn1048496.template/kscreensaver/kxsconfig/screensaver.sh
--- kdeartwork-4.3.75svn1048496/kscreensaver/kxsconfig/screensaver.sh	1969-12-31 19:00:00.000000000 -0500
+++ kdeartwork-4.3.75svn1048496.template/kscreensaver/kxsconfig/screensaver.sh	2009-11-22 02:33:10.000000000 -0500
@@ -0,0 +1,24 @@
+#! /bin/sh
+
+[ -n "$XSCREENSAVER_CONFIG_DIR" ] || XSCREENSAVER_CONFIG_DIR=/usr/share/xscreensaver/config/
+[ -n "$XSCREENSAVER_DIR" ] || XSCREENSAVER_DIR=/usr/libexec/xscreensaver/
+
+d1="${XSCREENSAVER_CONFIG_DIR}"
+d2="${XSCREENSAVER_DIR}"
+d3="kscreensaver/kxsconfig/ScreenSavers/"
+
+hacks=`cd $d1 && ls -1 | grep -v config`
+for name in $hacks; do
+  fa=${name/.xml}
+  echo check screeensaver $name
+  if test -f $d2/$fa && test -f $d1/$name && test ! -f $d3/$fa.desktop; then
+      label=`grep '<screensaver.*_label' $d1/$name  | sed -e 's#^.*_label=\"\(.*\)\".*$#\1#'`
+      f=$d3/$fa.desktop
+      sed -e "s#@NAME@#$fa#; s#@LABEL@#$label#" $d3/xscreensaver.template > $f
+      if ldd $d2/$fa 2>&1 | grep libGL; then
+         sed -e "s,@GL1@,X-KDE-Type=OpenGL,; s,@GL2@,X-KDE-Category=OpenGL Screen Savers," $f > $f.new && mv $f.new $f
+      else
+         grep -v @GL $f > $f.new && mv $f.new $f
+      fi
+  fi
+done
