diff -up grep-2.9/gnulib-tests/test-update-copyright.sh.orig grep-2.9/gnulib-tests/test-update-copyright.sh
--- grep-2.9/gnulib-tests/test-update-copyright.sh.orig	2011-01-03 21:57:35.000000000 +0100
+++ grep-2.9/gnulib-tests/test-update-copyright.sh	2011-07-11 11:39:18.000000000 +0200
@@ -231,7 +231,7 @@ Copyright (C) 1990-2005, 2007-2009 Acme,
 # Software Foundation, Inc.
 EOF
 
-rm $TMP*
+rm -f $TMP*
 
 ## -------------- ##
 ## Current year.  ##
@@ -257,7 +257,7 @@ compare /dev/null $TMP-stderr || exit 1
 compare - $TMP <<EOF || exit 1
 '\" Copyright (C) 2006, $YEAR Free Software Foundation, Inc.
 EOF
-rm $TMP*
+rm -f $TMP*
 
 ## ------------------ ##
 ## Surrounding text.  ##
@@ -281,7 +281,7 @@ dnl Undisturbed text.
 dnl Copyright (C) 1989, 2010 Free Software Foundation, Inc.
 dnl   Undisturbed text.
 EOF
-rm $TMP*
+rm -f $TMP*
 
 ## --------------- ##
 ## Widest prefix.  ##
@@ -310,7 +310,7 @@ compare - $TMP <<EOF || exit 1
 #### Copyright (C) 1976-1988, 1999-2008, 2010-2011 Free Software
 #### Foundation, Inc.
 EOF
-rm $TMP*
+rm -f $TMP*
 
 ## ------------------- ##
 ## Prefix too large.  ##
@@ -333,7 +333,7 @@ compare - $TMP <<EOF || exit 1
 ####  1986, 1987, 1988, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007,
 ####  2008 Free Software Foundation, Inc.
 EOF
-rm $TMP*
+rm -f $TMP*
 
 ## ------------- ##
 ## Blank lines.  ##
@@ -368,7 +368,7 @@ Copyright (C) 1976, 1977, 1978, 1979, 19
 1986, 1987, 1988, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007,
 2008 Free Software Foundation, Inc.
 EOF
-rm $TMP*
+rm -f $TMP*
 
 ## -------------- ##
 ## Leading tabs.  ##
@@ -397,7 +397,7 @@ compare - $TMP <<EOF || exit 1
 	Copyright (C) 1987-1988, 1991-2011 Free Software Foundation,
 	Inc.
 EOF
-rm $TMP*
+rm -f $TMP*
 
 ## -------------------- ##
 ## Unusual whitespace.  ##
@@ -427,7 +427,7 @@ compare - $TMP <<EOF || exit 1
 		# Copyright (C) 1987-1988, 1991-2011 Free Software
 		# Foundation, Inc.
 EOF
-rm $TMP*
+rm -f $TMP*
 
 ## --------- ##
 ## DOS EOL.  ##
@@ -449,7 +449,7 @@ Rem 1998, 1999, 2000, 2001, 2002, 2003, 
 Rem 2009, 2010 Free Software Foundation, Inc.@
 EOF
 compare $TMP-exp $TMP || exit 1
-rm $TMP*
+rm -f $TMP*
 
 ## --------------- ##
 ## Omitted "(C)".  ##
@@ -470,7 +470,7 @@ compare - $TMP <<EOF || exit 1
   1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010
   Free Software Foundation, Inc.
 EOF
-rm $TMP*
+rm -f $TMP*
 
 ## ------------------ ##
 ## C-style comments.  ##
@@ -532,6 +532,6 @@ compare - $TMP.extra-text-space <<EOF ||
     Inc. ***
     * End of comments. */
 EOF
-rm $TMP*
+rm -f $TMP*
 
 exit 0
diff -up grep-2.9/gnulib-tests/test-version-etc.sh.orig grep-2.9/gnulib-tests/test-version-etc.sh
--- grep-2.9/gnulib-tests/test-version-etc.sh.orig	2011-01-03 21:57:35.000000000 +0100
+++ grep-2.9/gnulib-tests/test-version-etc.sh	2011-07-11 11:40:12.706055333 +0200
@@ -38,6 +38,6 @@ EOT
  tr -d '\015' |
  diff -c $TMP - || ERR=1
 
-rm $TMP
+rm -f $TMP
 
 exit $ERR
