diff -Nrup ../../italc-1.0.9-rc1.orig/lib/include/qt_features.h ../lib/include/qt_features.h
--- ../../italc-1.0.9-rc1.orig/lib/include/qt_features.h    2007-11-24 21:06:10.000000000 +0100
+++ ../lib/include/qt_features.h    2008-05-09 20:08:20.000000000 +0200
@@ -35,9 +35,9 @@
 
 #else
 
-#ifdef BUILD_ICA
+/*#ifdef BUILD_ICA
 #define SYSTEMTRAY_SUPPORT
-#endif
+#endif*/
 
 #ifdef BUILD_LINUX
 #define NATIVE_VIRTUAL_KEY_SUPPORT

