--- a/erts/emulator/hipe/hipe_x86_signal.c.orig      2007-06-11 15:56:21.000000000 +0300
+++ b/erts/emulator/hipe/hipe_x86_signal.c   2009-05-15 16:00:54.000000000 +0300
@@ -55,7 +55,7 @@
 #define INIT() do { if( !init_done() ) do_init(); } while(0)
 #endif /* glibc 2.3 */

-#if __GLIBC__ == 2 && (__GLIBC_MINOR__ == 2 /*|| __GLIBC_MINOR__ == 3*/)
+#if __GLIBC__ == 2 && __GLIBC_MINOR__ >= 3
 /*
  * __libc_sigaction() is the core routine.
  * Without libpthread, sigaction() and __sigaction() are both aliases
