diff -Nuar gvtray-1.1.orig//gvtray.py gvtray-1.1/gvtray.py
--- gvtray-1.1.orig//gvtray.py	2013-04-05 00:55:45.286000091 +0200
+++ gvtray-1.1/gvtray.py	2013-04-05 00:55:59.609000119 +0200
@@ -48,7 +48,8 @@
         self.icon.add(self.tray_eventbox)
         
 	# get the actual Master volume
-        self.volume = alsa.Mixer('Master').getvolume()[0]
+        #self.volume = alsa.Mixer('Master').getvolume()[0]
+        self.volume = 42
         
 	# create a label for viewing actual sound volume and add it to the eventbox obj. from the trayicon obj.
         self.label = gtk.Label(str(self.volume))
