Index: mozilla/security/nss/lib/Makefile
===================================================================
--- mozilla.orig/security/nss/lib/Makefile
+++ mozilla/security/nss/lib/Makefile
@@ -62,11 +62,11 @@ ifndef USE_SYSTEM_ZLIB
 ZLIB_SRCDIR = zlib  # Add the zlib directory to DIRS.
 endif
 
-ifndef MOZILLA_CLIENT
-ifndef NSS_USE_SYSTEM_SQLITE
-SQLITE_SRCDIR = sqlite  # Add the sqlite directory to DIRS.
-endif
-endif
+#ifndef MOZILLA_CLIENT
+#ifndef NSS_USE_SYSTEM_SQLITE
+#SQLITE_SRCDIR = sqlite  # Add the sqlite directory to DIRS.
+#endif
+#endif
 
 ifndef MOZILLA_CLIENT
 ifeq ($(OS_ARCH),Linux)
Index: mozilla/security/nss/lib/softoken/manifest.mn
===================================================================
--- mozilla.orig/security/nss/lib/softoken/manifest.mn
+++ mozilla/security/nss/lib/softoken/manifest.mn
@@ -47,9 +47,9 @@ MAPFILE = $(OBJDIR)/softokn.def
 
 DEFINES += -DSHLIB_SUFFIX=\"$(DLL_SUFFIX)\" -DSHLIB_PREFIX=\"$(DLL_PREFIX)\" -DSOFTOKEN_LIB_NAME=\"$(notdir $(SHARED_LIBRARY))\" -DSHLIB_VERSION=\"$(LIBRARY_VERSION)\"
 
-ifdef SQLITE_INCLUDE_DIR
-INCLUDES += -I$(SQLITE_INCLUDE_DIR)
-endif
+#ifdef SQLITE_INCLUDE_DIR
+#INCLUDES += -I$(SQLITE_INCLUDE_DIR)
+#endif
 
 EXPORTS = \
 	secmodt.h \
