--- phoebe-gui.desktop.orig	2010-08-30 16:04:56.077411456 +0300
+++ phoebe-gui.desktop	2010-08-30 16:12:50.873397000 +0300
@@ -1,12 +1,14 @@
 [Desktop Entry]
+Encoding=UTF-8
 Type=Application
-Name=PHOEBE
+Name=PHysics Of Eclipsing BinariEs
+Name[tr]=Örten Değişen Çift Yıldız Eğrisi Analiz Programı
 Version=0.31
 Comment=PHysics Of Eclipsing BinariEs
 Terminal=no
 TryExec=phoebe-gui
 Exec=phoebe-gui %f
 Icon=phoebe-gui.png
-Categories=Education;Science;Astronomy;DataVisualization;NumericalAnalysis;Physics;GTK
+Categories=Application;Education;Science;Astronomy;
 MimeType=application/x-phoebe;
 
