From 53ca60964b3691172f7c95e68b6d754ba55e90ac Mon Sep 17 00:00:00 2001
From: Luca Bigliardi <shammash@artha.org>
Date: Mon, 09  Feb  2009  12:34:29  +0000
Subject: Explicitly link against libm.

---
Index: frei0r-plugins-1.2/src/Makefile.am
===================================================================
--- frei0r-plugins-1.2.orig/src/Makefile.am
+++ frei0r-plugins-1.2/src/Makefile.am
@@ -150,7 +150,7 @@ RGB_la_SOURCES = mixer3/RGB/RGB.c
 
 AM_CPPFLAGS = -I@top_srcdir@/include 
 AM_CFLAGS = -I@top_srcdir@/include 
-AM_LDFLAGS = -module -avoid-version -lm -export-dynamic
+AM_LDFLAGS = -module -avoid-version -lm -export-dynamic -lm
 AM_LIBTOOLFLAGS = --tag=disable-static
 
 plugindir = @libdir@/frei0r-1
