--- coreutils-5.92/src/date.c.langinfo	2005-09-16 09:06:57.000000000 +0100
+++ coreutils-5.92/src/date.c	2005-10-24 18:09:16.000000000 +0100
@@ -451,14 +451,7 @@
       format = DATE_FMT_LANGINFO ();
       if (! *format)
         {
-          /* Do not wrap the following literal format string with _(...).
-             For example, suppose LC_ALL is unset, LC_TIME=POSIX,
-             and LANG="ko_KR".  In that case, POSIX says that LC_TIME
-             determines the format and contents of date and time strings
-             written by date, which means "date" must generate output
-             using the POSIX locale; but adding _() would cause "date"
-             to use a Korean translation of the format.  */
-          format = "%a %b %e %H:%M:%S %Z %Y";
+          format = dcgettext(NULL, N_("%a %b %e %H:%M:%S %Z %Y"), LC_TIME);
         }
     }
 
