diff -urNp coreutils-8.13-orig/src/stty.c coreutils-8.13/src/stty.c
--- coreutils-8.13-orig/src/stty.c	2011-07-28 12:38:27.000000000 +0200
+++ coreutils-8.13/src/stty.c	2011-09-09 10:18:57.526687209 +0200
@@ -1005,7 +1005,7 @@ main (int argc, char **argv)
          spurious difference in an uninitialized portion of the structure.  */
       static struct termios new_mode;
 
-      if (tcsetattr (STDIN_FILENO, TCSADRAIN, &mode))
+      if (tcsetattr (STDIN_FILENO, TCSANOW, &mode))
         error (EXIT_FAILURE, errno, "%s", device_name);
 
       /* POSIX (according to Zlotnick's book) tcsetattr returns zero if
