diff -urNp coreutils-6.12-orig/src/md5sum.c coreutils-6.12/src/md5sum.c
--- coreutils-6.12-orig/src/md5sum.c	2008-05-26 08:40:33.000000000 +0200
+++ coreutils-6.12/src/md5sum.c	2008-10-21 16:07:28.000000000 +0200
@@ -175,6 +175,9 @@ With no FILE, or when FILE is -, read st
         fputs (_("\
   -t, --text           read in text mode (default)\n\
 "), stdout);
+  fputs (_("\
+  Note: There is no difference between binary and text mode option on GNU system.\n\
+"), stdout);
       fputs (_("\
 \n\
 The following three options are useful only when verifying checksums:\n\
