diff -up acpid-2.0.15/Makefile.am.mk acpid-2.0.15/Makefile.am
--- acpid-2.0.15/Makefile.am.mk	2012-02-17 04:37:12.109907134 +0100
+++ acpid-2.0.15/Makefile.am	2012-03-16 11:27:15.127573044 +0100
@@ -20,7 +20,7 @@ acpi_listen_SOURCES = acpi_listen.c log.
 
 dist_man8_MANS = acpid.8 acpi_listen.8
 
-dist_doc_DATA = COPYING Changelog README TESTPLAN TODO
+dist_doc_DATA = 
 
 # Just docs to add to the dist.
 # One problem with this is that it includes all files in the samples dir 
diff -up acpid-2.0.15/Makefile.in.mk acpid-2.0.15/Makefile.in
--- acpid-2.0.15/Makefile.in.mk	2012-03-16 11:28:53.568421905 +0100
+++ acpid-2.0.15/Makefile.in	2012-03-16 11:29:21.679378750 +0100
@@ -276,7 +276,7 @@ acpid_SOURCES = acpid.c acpi_ids.c conne
 
 acpi_listen_SOURCES = acpi_listen.c log.c ud_socket.c
 dist_man8_MANS = acpid.8 acpi_listen.8
-dist_doc_DATA = COPYING Changelog README TESTPLAN TODO
+dist_doc_DATA = 
 
 # Just docs to add to the dist.
 # One problem with this is that it includes all files in the samples dir