diff -Nur nut-old/Makefile nut/Makefile
--- nut-old/Makefile	2008-07-30 12:17:38.000000000 +0300
+++ nut/Makefile	2008-07-30 12:17:55.000000000 +0300
@@ -25,7 +25,7 @@
 
 $(NUTUTILS_PROGS): CFLAGS += -Ilibnut
 
-install: install-libnut install-nututils
+install: install-libnut-shared install-nututils
 
 install-libnut: libnut install-libnut-headers
 	install -d $(prefix)/lib
@@ -43,7 +43,7 @@
 	install -d $(prefix)/bin
 	install -m 755 $(NUTUTILS_PROGS) $(prefix)/bin
 
-uninstall: uninstall-libnut uninstall-nututils
+uninstall: uninstall-libnut-shared uninstall-nututils
 
 uninstall-libnut:
 	rm -f $(prefix)/lib/libnut.a
