diff -Nur telepathy-butterfly-0.5.7.orig/butterfly/channel/Makefile.in telepathy-butterfly-0.5.7/butterfly/channel/Makefile.in
--- telepathy-butterfly-0.5.7.orig/butterfly/channel/Makefile.in	2010-03-22 21:27:54.692338442 +0200
+++ telepathy-butterfly-0.5.7/butterfly/channel/Makefile.in	2010-03-22 21:28:56.000326418 +0200
@@ -71,7 +71,6 @@
   sed '$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;s/\n/ /g' | \
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(channeldir)"
-py_compile = $(top_srcdir)/py-compile
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 ACLOCAL = @ACLOCAL@
 AMTAR = @AMTAR@
@@ -215,14 +214,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(channeldir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(channeldir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(channeldir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(channeldir)" $$dlist; \
-	  fi; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-channelPYTHON:
 	@$(NORMAL_UNINSTALL)
diff -Nur telepathy-butterfly-0.5.7.orig/butterfly/Makefile.in telepathy-butterfly-0.5.7/butterfly/Makefile.in
--- telepathy-butterfly-0.5.7.orig/butterfly/Makefile.in	2010-03-22 21:27:54.684324014 +0200
+++ telepathy-butterfly-0.5.7/butterfly/Makefile.in	2010-03-22 21:28:14.198325231 +0200
@@ -78,7 +78,6 @@
   sed '$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;s/\n/ /g' | \
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(butterflydir)"
-py_compile = $(top_srcdir)/py-compile
 RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
   distclean-recursive maintainer-clean-recursive
 AM_RECURSIVE_TARGETS = $(RECURSIVE_TARGETS:-recursive=) \
@@ -262,14 +261,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(butterflydir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(butterflydir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(butterflydir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(butterflydir)" $$dlist; \
-	  fi; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-butterflyPYTHON:
 	@$(NORMAL_UNINSTALL)
diff -Nur telepathy-butterfly-0.5.7.orig/butterfly/media/Makefile.in telepathy-butterfly-0.5.7/butterfly/media/Makefile.in
--- telepathy-butterfly-0.5.7.orig/butterfly/media/Makefile.in	2010-03-22 21:27:54.684324014 +0200
+++ telepathy-butterfly-0.5.7/butterfly/media/Makefile.in	2010-03-22 21:28:26.231584145 +0200
@@ -71,7 +71,6 @@
   sed '$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;s/\n/ /g' | \
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(mediadir)"
-py_compile = $(top_srcdir)/py-compile
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 ACLOCAL = @ACLOCAL@
 AMTAR = @AMTAR@
@@ -211,14 +210,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(mediadir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(mediadir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(mediadir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(mediadir)" $$dlist; \
-	  fi; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-mediaPYTHON:
 	@$(NORMAL_UNINSTALL)
diff -Nur telepathy-butterfly-0.5.7.orig/butterfly/util/Makefile.in telepathy-butterfly-0.5.7/butterfly/util/Makefile.in
--- telepathy-butterfly-0.5.7.orig/butterfly/util/Makefile.in	2010-03-22 21:27:54.692338442 +0200
+++ telepathy-butterfly-0.5.7/butterfly/util/Makefile.in	2010-03-22 21:28:39.885323534 +0200
@@ -71,7 +71,6 @@
   sed '$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;$$!N;s/\n/ /g' | \
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(utildir)"
-py_compile = $(top_srcdir)/py-compile
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 ACLOCAL = @ACLOCAL@
 AMTAR = @AMTAR@
@@ -209,14 +208,7 @@
 	while read files; do \
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(utildir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(utildir)" || exit $$?; \
-	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(utildir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(utildir)" $$dlist; \
-	  fi; \
-	else :; fi
+	done || exit $$?;
 
 uninstall-utilPYTHON:
 	@$(NORMAL_UNINSTALL)
