diff -Nuar drawer.orig//module.desktop.in drawer/module.desktop.in
--- drawer.orig//module.desktop.in	2011-02-16 22:39:52.000000000 +0200
+++ drawer/module.desktop.in	2011-03-31 15:40:34.976013000 +0300
@@ -6,6 +6,7 @@
 Name[el]=Συρτάρι
 Name[eo]=Entenilo
 Name[fr]=Tiroir
+Name[tr]=Çizici
 Icon=e-module-drawer
 Comment=A Gadget that acts as a container for data.
 Comment[ru]=Гаджет, который действует как контейнер для данных.
@@ -16,4 +17,5 @@
 Comment[fr]=Un gadget faisant office de réceptacle.
 Comment[it]=Un modulo che fa da contenitore per dati.
 Comment[pt]=Um "gadget" utilizado para armazenar dados
+Comment[tr]=Veri için bir kap gibi davranan programcık.
 X-Enlightenment-ModuleType=utils
