diff -Naur kaudiocreator-orig/encoderconfigimp.cpp kaudiocreator/encoderconfigimp.cpp
--- kaudiocreator-orig/encoderconfigimp.cpp	2009-06-09 21:41:48.659338337 +0300
+++ kaudiocreator/encoderconfigimp.cpp	2009-06-09 21:50:04.975563580 +0300
@@ -82,7 +82,7 @@
 	// Make sure that the default encoder is valid.
 	QString groupName = QString("Encoder_").append(Prefs::defaultEncoder());
 	if (!list.contains(groupName)) {
-		Prefs::setDefaultEncoder("WAV");
+		Prefs::setDefaultEncoder("MP3 Lame Standard");
 		Prefs::self()->writeConfig();
 	}
 	kcfg_defaultEncoder->setText(Prefs::defaultEncoder());
diff -Naur kaudiocreator-orig/kaudiocreator.cpp kaudiocreator/kaudiocreator.cpp
--- kaudiocreator-orig/kaudiocreator.cpp	2009-06-09 21:41:48.658337896 +0300
+++ kaudiocreator/kaudiocreator.cpp	2009-06-09 21:51:17.735347780 +0300
@@ -248,7 +248,7 @@
 
 	QString groupName = QString("Encoder_").append(Prefs::defaultEncoder());
 	if (!list.contains(groupName)) {
-		Prefs::setDefaultEncoder("WAV");
+		Prefs::setDefaultEncoder("MP3 Lame Standard");
 		Prefs::self()->writeConfig();
 	}
 }
diff -Naur kaudiocreator-orig/kaudiocreator.kcfg kaudiocreator/kaudiocreator.kcfg
--- kaudiocreator-orig/kaudiocreator.kcfg	2009-06-09 21:41:48.659338337 +0300
+++ kaudiocreator/kaudiocreator.kcfg	2009-06-09 21:49:25.063315869 +0300
@@ -59,7 +59,7 @@
   <group name="Encoder">
     <entry name="defaultEncoder" type="String">
       <label>Current default encoder</label>
-      <default>WAV</default>
+      <default>MP3 Lame Standard</default>
     </entry>
     <entry name="inputTypesList" type="StringList">
       <label>List of available input types</label>
