Index: rxvt-unicode-9.11/doc/Makefile.in
===================================================================
--- rxvt-unicode-9.11.orig/doc/Makefile.in
+++ rxvt-unicode-9.11/doc/Makefile.in
@@ -74,7 +74,6 @@ tags allbin:
 
 alldoc: rxvt.1.man.in rxvt.7.man.in rxvtc.1.man.in rxvtd.1.man.in rxvtperl.3.man.in \
 	../README.configure ../README.FAQ
-	tic -C $(srcdir)/etc/rxvt-unicode.terminfo >$(srcdir)/etc/rxvt-unicode.termcap
 
 clean:
 	rm -f rxvt.1.man rxvt.7.man rxvtc.1.man rxvtd.1.man rxvtperl.3.man
@@ -98,7 +97,6 @@ install: all
 	$(INSTALL_DATA) rxvt.7.man  $(DESTDIR)$(man7dir)/$(RXVTNAME).$(man7ext)
 @IF_PERL@	$(INSTALL) -d $(DESTDIR)$(man3dir)
 @IF_PERL@	$(INSTALL_DATA) rxvtperl.3.man $(DESTDIR)$(man3dir)/$(RXVTNAME)perl.$(man3ext)
-	@TIC@ $(srcdir)/etc/rxvt-unicode.terminfo
 
 distdepend: alldoc
 
