diff -up rxvt-unicode-9.10/src/perl/tabbed.orig rxvt-unicode-9.10/src/perl/tabbed
--- rxvt-unicode-9.10/src/perl/tabbed.orig	2011-02-08 17:52:33.000000000 +0200
+++ rxvt-unicode-9.10/src/perl/tabbed	2011-02-08 17:53:34.000000000 +0200
@@ -333,6 +333,10 @@ sub tab_key_press {
          $self->make_current ($self->{tabs}[$idx2]);
 
          return 1;
+      } elsif ($keysym == 0x74) {
+         $self->new_tab;
+
+         return 1;
       }
    }
 
