Index: libvirt-0.8.7/configure.ac
===================================================================
--- libvirt-0.8.7.orig/configure.ac
+++ libvirt-0.8.7/configure.ac
@@ -173,7 +173,7 @@ AC_ARG_WITH([html-dir], [AC_HELP_STRING(
 AC_ARG_WITH([html-subdir], [AC_HELP_STRING([--with-html-subdir=path],
             [directory used under html-dir, default $PACKAGE-$VERSION/html])],
             [test "x$withval" != "x" && HTML_DIR="$HTML_DIR/$withval"],
-            [HTML_DIR="$HTML_DIR/\$(PACKAGE)-\$(VERSION)/html"])
+            [HTML_DIR="$HTML_DIR/\$(PACKAGE)/html"])
 AC_SUBST([HTML_DIR])
 
 dnl Specific XML catalog file for validation of generated html
Index: libvirt-0.8.7/python/Makefile.am
===================================================================
--- libvirt-0.8.7.orig/python/Makefile.am
+++ libvirt-0.8.7/python/Makefile.am
@@ -9,7 +9,7 @@ INCLUDES = \
 	-I$(top_builddir)/include \
 	-I$(top_builddir)/$(subdir)
 
-DOCS_DIR = $(datadir)/doc/libvirt-python-$(LIBVIRT_VERSION)
+DOCS_DIR = $(datadir)/doc/libvirt/python
 
 DOCS = ${srcdir}/TODO
 
Index: libvirt-0.8.7/python/Makefile.in
===================================================================
--- libvirt-0.8.7.orig/python/Makefile.in
+++ libvirt-0.8.7/python/Makefile.in
@@ -1140,7 +1140,7 @@ INCLUDES = \
 	-I$(top_builddir)/include \
 	-I$(top_builddir)/$(subdir)
 
-DOCS_DIR = $(datadir)/doc/libvirt-python-$(LIBVIRT_VERSION)
+DOCS_DIR = $(datadir)/doc/libvirt/python
 DOCS = ${srcdir}/TODO
 CLASSES_EXTRA = \
 	libvirt-override-virConnect.py \
Index: libvirt-0.8.7/python/tests/Makefile.am
===================================================================
--- libvirt-0.8.7.orig/python/tests/Makefile.am
+++ libvirt-0.8.7/python/tests/Makefile.am
@@ -1,4 +1,4 @@
-EXAMPLE_DIR = $(datadir)/doc/libvirt-python-$(LIBVIRT_VERSION)/examples
+EXAMPLE_DIR = $(datadir)/doc/libvirt/python/examples
 
 PYTESTS=		\
 	basic.py	\
Index: libvirt-0.8.7/python/tests/Makefile.in
===================================================================
--- libvirt-0.8.7.orig/python/tests/Makefile.in
+++ libvirt-0.8.7/python/tests/Makefile.in
@@ -1028,7 +1028,7 @@ target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-EXAMPLE_DIR = $(datadir)/doc/libvirt-python-$(LIBVIRT_VERSION)/examples
+EXAMPLE_DIR = $(datadir)/doc/libvirt/python/examples
 PYTESTS = \
 	basic.py	\
 	create.py	\
