--- src/qemu/qemu_capabilities.c.orig	2011-04-06 12:34:36.000000000 +0400
+++ src/qemu/qemu_capabilities.c	2011-05-27 02:08:29.265000014 +0400
@@ -74,9 +74,9 @@
 
 /* The archicture tables for supported QEMU archs */
 static const struct qemu_arch_info const arch_info_hvm[] = {
-    {  "i686",   32, NULL, "qemu",
+    {  "i686",   32, NULL, "qemu-kvm",
        "qemu-system-x86_64", arch_info_i686_flags, 4 },
-    {  "x86_64", 64, NULL, "qemu-system-x86_64",
+    {  "x86_64", 64, NULL, "qemu-kvm",
        NULL, arch_info_x86_64_flags, 2 },
     {  "arm",    32, NULL, "qemu-system-arm",    NULL, NULL, 0 },
     {  "mips",   32, NULL, "qemu-system-mips",   NULL, NULL, 0 },
