Index: libvirt-0.8.2/daemon/libvirtd.policy-0
===================================================================
--- libvirt-0.8.2.orig/daemon/libvirtd.policy-0
+++ libvirt-0.8.2/daemon/libvirtd.policy-0
@@ -18,7 +18,9 @@ file are instantly applied.
 <policyconfig>
     <action id="org.libvirt.unix.monitor">
       <description>Monitor local virtualized systems</description>
+      <description xml:lang="tr">Sanallaştırılmış yerel sistemleri izle</description>
       <message>System policy prevents monitoring of local virtualized systems</message>
+      <message xml:lang="tr">Sistem politikası sanallaştırılmış yerel sistemlerin izlenmesini engelliyor</message>
       <defaults>
         <!-- Any program can use libvirt in read-only mode for monitoring,
              even if not part of a session -->
@@ -30,7 +32,9 @@ file are instantly applied.
 
     <action id="org.libvirt.unix.manage">
       <description>Manage local virtualized systems</description>
+      <description xml:lang="tr">Sanallaştırılmış yerel sistemleri yönet</description>
       <message>System policy prevents management of local virtualized systems</message>
+      <message xml:lang="tr">Sistem politikası sanallaştırılmış yerel sistemlerin yönetilmesini engelliyor</message>
       <defaults>
         <!-- Only a program in the active host session can use libvirt in
              read-write mode for management, and we require user password -->
Index: libvirt-0.8.2/daemon/libvirtd.policy-1
===================================================================
--- libvirt-0.8.2.orig/daemon/libvirtd.policy-1
+++ libvirt-0.8.2/daemon/libvirtd.policy-1
@@ -18,7 +18,9 @@ file are instantly applied.
 <policyconfig>
     <action id="org.libvirt.unix.monitor">
       <description>Monitor local virtualized systems</description>
+      <description xml:lang="tr">Sanallaştırılmış yerel sistemleri izle</description>
       <message>System policy prevents monitoring of local virtualized systems</message>
+      <message xml:lang="tr">Sistem politikası sanallaştırılmış yerel sistemlerin izlenmesini engelliyor</message>
       <defaults>
         <!-- Any program can use libvirt in read-only mode for monitoring,
              even if not part of a session -->
@@ -30,7 +32,9 @@ file are instantly applied.
 
     <action id="org.libvirt.unix.manage">
       <description>Manage local virtualized systems</description>
+      <description xml:lang="tr">Sanallaştırılmış yerel sistemleri yönet</description>
       <message>System policy prevents management of local virtualized systems</message>
+      <message xml:lang="tr">Sistem politikası sanallaştırılmış yerel sistemlerin yönetilmesini engelliyor</message>
       <defaults>
         <!-- Only a program in the active host session can use libvirt in
              read-write mode for management, and we require user password -->
