Index: umplayer-0.0_180/src/basegui.cpp
===================================================================
--- umplayer-0.0_180.orig/src/basegui.cpp
+++ umplayer-0.0_180/src/basegui.cpp
@@ -270,10 +270,10 @@ void BaseGui::initializeGui() {
         resizeFrameStep = new QTimer(this);
         resizeFrameStep->setInterval(300);
         connect(resizeFrameStep, SIGNAL(timeout()), this, SLOT(frameStepNeeded()));
-        if(pref->playCount == 10)
-        {
-            QTimer::singleShot(0, this, SLOT(facebookAlert()));
-        }
+        //if(pref->playCount == 10)
+        //{
+            //QTimer::singleShot(0, this, SLOT(facebookAlert()));
+        //}
 }
 
 void BaseGui::frameStepNeeded()
