--- darkplaces/image_png.c.orig	2011-01-06 12:02:10.658000104 +0200
+++ darkplaces/image_png.c	2011-01-06 12:04:32.002000081 +0200
@@ -110,7 +110,9 @@
 #elif defined(MACOSX)
 		"libpng12.0.dylib",
 #else
-		"libpng12.so.0",
+		"libpng14.so",
+                "libpng14.so.14",
+                "libpng14.so.14.3.0",
 		"libpng.so", // FreeBSD
 #endif
 		NULL
--- darkplaces/jpeg.c.orig	2011-01-06 12:08:12.859000090 +0200
+++ darkplaces/jpeg.c	2011-01-06 12:09:53.102000090 +0200
@@ -69,7 +69,7 @@
 typedef struct jpeg_compress_struct *j_compress_ptr;
 typedef struct jpeg_decompress_struct *j_decompress_ptr;
 
-#define JPEG_LIB_VERSION  62  // Version 6b
+#define JPEG_LIB_VERSION  80  // Version 6b
 
 typedef enum
 {
@@ -497,7 +497,8 @@
 #elif defined(MACOSX)
 		"libjpeg.62.dylib",
 #else
-		"libjpeg.so.62",
+		"libjpeg.so.8.0.2",
+                "libjpeg.so.8",
 		"libjpeg.so",
 #endif
 		NULL
