--- nexuiz-2.5.2/darkplaces/gl_rmain.c	(revision 9279)
+++ nexuiz-2.5.2/darkplaces/gl_rmain.c	(revision 9280)
@@ -3709,14 +3709,14 @@
 		r_waterstate.textureheight = textureheight;
 	}
 
-	// when doing a reduced render (HDR) we want to use a smaller area
-	r_waterstate.waterwidth = (int)bound(1, r_refdef.view.width * r_water_resolutionmultiplier.value, r_refdef.view.width);
-	r_waterstate.waterheight = (int)bound(1, r_refdef.view.height * r_water_resolutionmultiplier.value, r_refdef.view.height);
-
 	if (r_waterstate.waterwidth)
 	{
 		r_waterstate.enabled = true;
 
+		// when doing a reduced render (HDR) we want to use a smaller area
+		r_waterstate.waterwidth = (int)bound(1, r_refdef.view.width * r_water_resolutionmultiplier.value, r_refdef.view.width);
+		r_waterstate.waterheight = (int)bound(1, r_refdef.view.height * r_water_resolutionmultiplier.value, r_refdef.view.height);
+
 		// set up variables that will be used in shader setup
 		r_waterstate.screenscale[0] = 0.5f * (float)r_waterstate.waterwidth / (float)r_waterstate.texturewidth;
 		r_waterstate.screenscale[1] = 0.5f * (float)r_waterstate.waterheight / (float)r_waterstate.textureheight;
