--- darkplaces/image_png.c.orig	2011-01-06 12:20:26.272000088 +0200
+++ darkplaces/image_png.c	2011-01-06 12:23:03.231000110 +0200
@@ -36,7 +36,7 @@
 static void*			(*qpng_create_info_struct)	(void*);
 static void				(*qpng_read_info)			(void*, void*);
 static void				(*qpng_set_expand)			(void*);
-static void				(*qpng_set_gray_1_2_4_to_8)	(void*);
+static void				(*qpng_set_expand_gray_1_2_4_to_8)	(void*);
 static void				(*qpng_set_palette_to_rgb)	(void*);
 static void				(*qpng_set_tRNS_to_alpha)	(void*);
 static void				(*qpng_set_gray_to_rgb)		(void*);
@@ -61,7 +61,7 @@
 	{"png_create_info_struct",	(void **) &qpng_create_info_struct},
 	{"png_read_info",			(void **) &qpng_read_info},
 	{"png_set_expand",			(void **) &qpng_set_expand},
-	{"png_set_gray_1_2_4_to_8",	(void **) &qpng_set_gray_1_2_4_to_8},
+	{"png_set_expand_gray_1_2_4_to_8",	(void **) &qpng_set_expand_gray_1_2_4_to_8},
 	{"png_set_palette_to_rgb",	(void **) &qpng_set_palette_to_rgb},
 	{"png_set_tRNS_to_alpha",	(void **) &qpng_set_tRNS_to_alpha},
 	{"png_set_gray_to_rgb",		(void **) &qpng_set_gray_to_rgb},
@@ -313,7 +313,7 @@
 	{
 		qpng_set_gray_to_rgb(png);
 		if (my_png.BitDepth < 8)
-			qpng_set_gray_1_2_4_to_8(png);
+			qpng_set_expand_gray_1_2_4_to_8(png);
 	}
 
 	if (qpng_get_valid(png, pnginfo, PNG_INFO_tRNS))

--- darkplaces/makefile.orig	2011-01-06 14:30:54.020000084 +0200
+++ darkplaces/makefile	2011-01-06 14:31:17.202000090 +0200
@@ -1,3 +1,5 @@
+DP_LINK_TO_LIBJPEG=1
+
 #####  DP_MAKE_TARGET autodetection and arch specific variables #####
 
 ifndef DP_MAKE_TARGET
