diff -Naur slang-2.1.2.orig/slsh/Makefile.in slang-2.1.2/slsh/Makefile.in
--- slang-2.1.2.orig/slsh/Makefile.in	2007-09-09 10:39:18.000000000 +0900
+++ slang-2.1.2/slsh/Makefile.in	2007-09-19 01:31:41.000000000 +0900
@@ -74,7 +74,7 @@
 slsh_exe: slsh.o readline.o
 	$(CC) $(CFLAGS) slsh.o readline.o -o slsh_exe $(LDFLAGS) $(SRC_LIBS)
 slsh: slsh.o readline.o
-	$(CC) $(CFLAGS) slsh.o readline.o -o slsh $(LDFLAGS) $(INST_LIBS)
+	$(CC) $(CFLAGS) slsh.o readline.o -o slsh $(LDFLAGS) $(SRC_LIBS)
 slsh.o: slsh.c slsh.h config.h Makefile
 	$(CC) -c $(CFLAGS) $(SLANG_SRCINC) $(DEFS) slsh.c
 readline.o: readline.c slsh.h config.h Makefile
