diff -Nur slang-2.1.4-old/src/slang.h slang-2.1.4/src/slang.h
--- slang-2.1.4-old/src/slang.h	2009-02-03 00:36:22.000000000 +0200
+++ slang-2.1.4/src/slang.h	2009-02-03 00:39:53.000000000 +0200
@@ -1683,6 +1683,9 @@
 SL_EXTERN char *SLtt_tigetent (SLFUTURE_CONST char *);
 SL_EXTERN char *SLtt_tigetstr (SLFUTURE_CONST char *, char **);
 SL_EXTERN int SLtt_tigetnum (SLFUTURE_CONST char *, char **);
+SL_EXTERN int SLtt_tgetent(char *name);
+SL_EXTERN char *SLtt_tgoto(char *cap, int col, int row);
+SL_EXTERN int SLtt_tputs(char *str, int affcnt, int (*putc)(int));
 # endif
 #endif
 
diff -Nur slang-2.1.4-old/src/sldisply.c slang-2.1.4/src/sldisply.c
--- slang-2.1.4-old/src/sldisply.c	2009-02-03 00:36:22.000000000 +0200
+++ slang-2.1.4/src/sldisply.c	2009-02-03 00:37:55.000000000 +0200
@@ -2315,6 +2315,29 @@
 }
 #endif
 
+int Sltt_tgetent(char *term)
+{
+   return SLtt_initialize(term) == 0;
+}
+
+int SLtt_tputs(char *str, int affcnt, int (*p)(int))
+{
+   while (*str) (*p)(*str++);
+   return 0;
+}
+
+char *SLtt_tgoto(char *cap, int col, int row)
+{
+  static char buf[64];
+
+  /* beware of overflows. 2^64 is 20 bytes printed */
+  if (strlen(cap) > 23)
+	strcpy(buf, "capability too long");
+  else
+	tt_sprintf(buf, sizeof(buf), cap, row, col);
+  return buf;
+}
+
 static int Vt100_Like = 0;
 
 void SLtt_get_terminfo (void)
