diff -up xf86-video-ati-6.12.2/src/drmmode_display.c.da xf86-video-ati-6.12.2/src/drmmode_display.c
--- xf86-video-ati-6.12.2/src/drmmode_display.c.da	2009-06-22 12:21:21.000000000 +1000
+++ xf86-video-ati-6.12.2/src/drmmode_display.c	2009-06-22 12:21:30.000000000 +1000
@@ -593,7 +593,7 @@ drmmode_output_get_modes(xf86OutputPtr o
 			max_vrefresh = max(max_vrefresh, 60.0);
                         max_vrefresh *= (1 + SYNC_TOLERANCE);
  
-                        m = xf86GetDefaultModes(FALSE, FALSE);
+                        m = xf86GetDefaultModes();
  
                         xf86ValidateModesSize(output->scrn, m, max_x, max_y, 0);
                         for (i = m; i; i = i->next) {
