diff -up xf86-video-ati-6.12.2/src/radeon_kms.c.da xf86-video-ati-6.12.2/src/radeon_kms.c
--- xf86-video-ati-6.12.2/src/radeon_kms.c.da	2009-08-04 15:53:14.000000000 +1000
+++ xf86-video-ati-6.12.2/src/radeon_kms.c	2009-08-04 15:53:48.000000000 +1000
@@ -331,6 +331,9 @@ Bool RADEONPreInit_KMS(ScrnInfoPtr pScrn
     if (pScrn->numEntities != 1) return FALSE;
     if (!RADEONGetRec(pScrn)) return FALSE;
 
+    /* kms bg root enable */
+    pScrn->canDoBGNoneRoot = 1;
+
     info               = RADEONPTR(pScrn);
     info->MMIO         = NULL;
     info->IsSecondary  = FALSE;
