diff -up xf86-video-ati-20120206/src/radeon_probe.c.jx xf86-video-ati-20120206/src/radeon_probe.c
--- xf86-video-ati-20120206/src/radeon_probe.c.jx	2012-02-01 09:41:05.000000000 -0500
+++ xf86-video-ati-20120206/src/radeon_probe.c	2012-03-28 15:41:00.658773763 -0400
@@ -183,16 +183,7 @@ radeon_get_scrninfo(int entity_num, void
       pScrn->ValidMode     = RADEONValidMode;
     } else 
 #endif 
-    {
-      pScrn->PreInit       = RADEONPreInit;
-      pScrn->ScreenInit    = RADEONScreenInit;
-      pScrn->SwitchMode    = RADEONSwitchMode;
-      pScrn->AdjustFrame   = RADEONAdjustFrame;
-      pScrn->EnterVT       = RADEONEnterVT;
-      pScrn->LeaveVT       = RADEONLeaveVT;
-      pScrn->FreeScreen    = RADEONFreeScreen;
-      pScrn->ValidMode     = RADEONValidMode;
-    }
+      return FALSE;
 
     pEnt = xf86GetEntityInfo(entity_num);
 
