Index: util-macros-1.10.0/Makefile.am
===================================================================
--- util-macros-1.10.0.orig/Makefile.am
+++ util-macros-1.10.0/Makefile.am
@@ -21,7 +21,6 @@
 
 aclocaldir = $(datadir)/aclocal
 aclocal_DATA = xorg-macros.m4
-dist_pkgdata_DATA = INSTALL
 
 # Previous versions used to install xorgversion.m4, now integrated
 # into xorg-macros.m4.  Explicitly remove that old file in order not
