diff -Nuar a/configure.ac b/configure.ac
--- a/configure.ac	2013-07-12 21:37:04.000000000 +0300
+++ b/configure.ac	2013-07-14 14:33:22.048697113 +0300
@@ -202,9 +202,9 @@
     enable_fonts=${enable_fonts:-no}
     enable_server=${enable_server:-no}
     enable_tools=${enable_tools:-no}
-elif test "x$enable_win64" = "xyes"
+elif test "x$enable_win64" = "xno"
 then
-    test "x$libdir" != "x\${exec_prefix}/lib" || libdir="\${exec_prefix}/lib64"
+    test "x$libdir" != "x\${exec_prefix}/lib" || libdir="\${exec_prefix}/lib32"
 fi
 
 AC_CACHE_CHECK([for the directory containing the Wine tools], wine_cv_toolsdir,
