--- tools/wine.inf.in.orig	2008-10-29 06:43:52.000000000 +0100
+++ tools/wine.inf.in	2008-10-29 06:44:00.000000000 +0100
@@ -297,7 +297,10 @@
 HKLM,%FontSubStr%,"Courier New TUR,162",,"Courier New,162"
 HKLM,%FontSubStr%,"Helv",,"MS Sans Serif"
 HKLM,%FontSubStr%,"Helvetica",,"Arial"
+HKLM,%FontSubStr%,"MingLiU","AR PL Uming TW"
 HKLM,%FontSubStr%,"MS Shell Dlg 2",,"Tahoma"
+HKLM,%FontSubStr%,"PMingLiU","AR PL Uming TW"
+HKLM,%FontSubStr%,"Simsun","AR PL UMing CN"
 HKLM,%FontSubStr%,"Times",,"Times New Roman"
 HKLM,%FontSubStr%,"Times New Roman Baltic,186",,"Times New Roman,186"
 HKLM,%FontSubStr%,"Times New Roman CE,238",,"Times New Roman,238"
