Index: mozilla/toolkit/xre/nsKDEUtils.cpp
===================================================================
--- mozilla.orig/toolkit/xre/nsKDEUtils.cpp
+++ mozilla/toolkit/xre/nsKDEUtils.cpp
@@ -51,7 +51,7 @@
 #define KMOZILLAHELPER "kmozillahelper"
 #else
 // not need for lib64, it's a binary
-#define KMOZILLAHELPER "/usr/lib/mozilla/kmozillahelper"
+#define KMOZILLAHELPER "/usr/libexec/kmozillahelper"
 #endif
 
 #define KMOZILLAHELPER_VERSION 6
