Index: samba-3.4.3/source3/Makefile.in
===================================================================
--- samba-3.4.3.orig/source3/Makefile.in
+++ samba-3.4.3/source3/Makefile.in
@@ -128,6 +128,8 @@ DATADIR = @datadir@
 INSTALLPERMS_BIN = 0755
 INSTALLPERMS_DATA = 0644
 
+LAZYLDFLAGS=-Wl,-z,now
+
 # set these to where to find various files
 # These can be overridden by command line switches (see smbd(8))
 # or in smb.conf (see smb.conf(5))
@@ -1459,11 +1461,11 @@ bin/smbspool@EXEEXT@: $(BINARY_PREREQS) 
 
 bin/mount.cifs@EXEEXT@: $(BINARY_PREREQS) $(CIFS_MOUNT_OBJ)
 	@echo Linking $@
-	@$(CC) -o $@ $(CIFS_MOUNT_OBJ) $(DYNEXP) $(LDFLAGS)
+	@$(CC) -o $@ $(CIFS_MOUNT_OBJ) $(DYNEXP) $(LDFLAGS) $(POPT_LIBS) $(LAZYLDFLAGS)
 
 bin/umount.cifs@EXEEXT@: $(BINARY_PREREQS) $(CIFS_UMOUNT_OBJ)
 	@echo Linking $@
-	@$(CC) -o $@ $(CIFS_UMOUNT_OBJ) $(DYNEXP) $(LDFLAGS)
+	@$(CC) -o $@ $(CIFS_UMOUNT_OBJ) $(DYNEXP) $(LDFLAGS) $(POPT_LIBS) $(LAZYLDFLAGS)
 
 bin/cifs.upcall@EXEEXT@: $(BINARY_PREREQS) $(CIFS_UPCALL_OBJ) $(LIBSMBCLIENT_OBJ1) @LIBTALLOC_TARGET@ @LIBTDB_TARGET@ @LIBWBCLIENT_TARGET@
 	@echo Linking $@
