--- samba-3.5.10.orig/client/mount.cifs.c	2011-09-09 11:24:57.060000115 +0300
+++ samba-3.5.10/client/mount.cifs.c	2011-09-09 11:29:23.451000115 +0300
@@ -1222,8 +1222,7 @@
 static int check_mtab(const char *progname, const char *devname,
 			const char *dir)
 {
-	if (check_newline(progname, devname) == -1 ||
-	    check_newline(progname, dir) == -1)
+	if (check_newline(progname, devname) || check_newline(progname, dir))
 		return EX_USAGE;
 	return 0;
 }
