--- samba-3.5.8/source3/lib/netapi/examples/Makefile.in~	2011-03-06 20:48:05.000000000 +0200
+++ samba-3.5.8/source3/lib/netapi/examples/Makefile.in	2011-05-02 15:22:07.000000000 +0200
@@ -3,7 +3,7 @@
 
 KRB5LIBS=@KRB5_LIBS@
 LDAP_LIBS=@LDAP_LIBS@
-LIBS=@LIBS@ -lnetapi -ltdb -ltalloc
+LIBS=@LIBS@ -lnetapi -ltdb -ltalloc -lwbclient
 DEVELOPER_CFLAGS=@DEVELOPER_CFLAGS@
 CCFLAGS=-I../ -L../../../bin @CFLAGS@ $(GTK_FLAGS)
 CC=@CC@
