--- lxdream-0.9.1-orig/src/aica/audio.c	2009-06-24 04:48:00.000000000 +0200
+++ lxdream-0.9.1/src/aica/audio.c	2012-05-01 08:22:21.590378228 +0200
@@ -19,7 +19,7 @@

 #include "aica/aica.h"
 #include "aica/audio.h"
-#include <glib/gmem.h>
+#include <glib-2.0/glib.h>
 #include "dream.h"
 #include <assert.h>
 #include <string.h>
--- lxdream-0.9.1-orig/src/aica/audio.h	2009-06-24 04:48:00.000000000 +0200
+++ lxdream-0.9.1/src/aica/audio.h	2012-05-01 08:22:18.040378091 +0200
@@ -20,7 +20,7 @@

 #include <stdint.h>
 #include <stdio.h>
-#include <glib/gtypes.h>
+#include <glib-2.0/glib.h>
 #include "gettext.h"
 #include "plugin.h"

--- lxdream-0.9.1-orig/src/config.c	2009-06-26 07:47:06.000000000 +0200
+++ lxdream-0.9.1/src/config.c	2012-05-01 08:24:55.063717779 +0200
@@ -21,8 +21,7 @@
 #include <errno.h>
 #include <stdlib.h>
 #include <string.h>
-#include <glib/gmem.h>
-#include <glib/gstrfuncs.h>
+#include <glib-2.0/glib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include "dream.h"
--- lxdream-0.9.1-orig/src/config.h	2009-06-26 07:47:06.000000000 +0200
+++ lxdream-0.9.1/src/config.h	2012-05-01 08:21:16.127042242 +0200
@@ -19,8 +19,7 @@
 #ifndef lxdream_config_H
 #define lxdream_config_H 1

-#include <glib/gtypes.h>
-#include <glib/glist.h>
+#include <glib-2.0/glib.h>
 #include "gettext.h"

 #ifdef __cplusplus
--- lxdream-0.9.1-orig/src/drivers/input_lirc.c	2009-06-28 12:29:06.000000000 +0200
+++ lxdream-0.9.1/src/drivers/input_lirc.c	2012-05-01 08:24:46.470384100 +0200
@@ -28,8 +28,7 @@
 #include <dirent.h>
 #include <ctype.h>

-#include <glib/giochannel.h>
-#include <glib.h>
+#include <glib-2.0/glib.h>

 #include <lirc/lirc_client.h>

--- lxdream-0.9.1-orig/src/drivers/joy_linux.c	2009-06-24 04:48:00.000000000 +0200
+++ lxdream-0.9.1/src/drivers/joy_linux.c	2012-05-01 08:24:38.783717121 +0200
@@ -32,8 +32,7 @@
 #include <ctype.h>

 #include <linux/joystick.h>
-#include <glib/giochannel.h>
-#include <glib.h>
+#include <glib-2.0/glib.h>

 #include "lxdream.h"
 #include "display.h"
--- lxdream-0.9.1-orig/src/gdlist.c	2009-06-24 08:06:43.000000000 +0200
+++ lxdream-0.9.1/src/gdlist.c	2012-05-01 08:21:28.383709407 +0200
@@ -19,7 +19,7 @@

 #include <string.h>
 #include <stdlib.h>
-#include <glib/gstrfuncs.h>
+#include <glib-2.0/glib.h>
 #include <libgen.h>
 #include "gettext.h"
 #include "gdrom/gdrom.h"
--- lxdream-0.9.1-orig/src/gdrom/gddriver.h	2009-06-24 04:48:00.000000000 +0200
+++ lxdream-0.9.1/src/gdrom/gddriver.h	2012-05-01 08:24:08.917049245 +0200
@@ -24,7 +24,7 @@
 #include <stdio.h>
 #include "lxdream.h"
 #include "gdrom/gdrom.h"
-#include <glib/gstrfuncs.h>
+#include <glib-2.0/glib.h>

 #ifdef __cplusplus
 extern "C" {
--- lxdream-0.9.1-orig/src/gdrom/gdi.c	2009-06-24 04:48:00.000000000 +0200
+++ lxdream-0.9.1/src/gdrom/gdi.c	2012-05-01 08:23:49.087048443 +0200
@@ -23,7 +23,7 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <sys/stat.h>
-#include <glib/gutils.h>
+#include <glib-2.0/glib.h>
 #include "gdrom/gddriver.h"


--- lxdream-0.9.1-orig/src/gdrom/gdrom.h	2009-06-24 04:48:00.000000000 +0200
+++ lxdream-0.9.1/src/gdrom/gdrom.h	2012-05-01 08:23:56.167048729 +0200
@@ -22,7 +22,7 @@

 #include "lxdream.h"
 #include "hook.h"
-#include <glib/glist.h>
+#include <glib-2.0/glib.h>

 #ifdef __cplusplus
 extern "C" {
--- lxdream-0.9.1-orig/src/gdrom/gdrom.c        2009-06-24 04:48:00.000000000 +0200
+++ lxdream-0.9.1/src/gdrom/gdrom.c     2012-05-01 08:30:08.593561432 +0200
@@ -20,7 +20,7 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <ctype.h>
-#include <glib/gutils.h>
+#include <glib-2.0/glib.h>
 #include "gdrom/ide.h"
 #include "gdrom/gdrom.h"
 #include "gdrom/gddriver.h"
--- lxdream-0.9.1-orig/src/gdrom/nrg.c	2009-06-24 04:48:00.000000000 +0200
+++ lxdream-0.9.1/src/gdrom/nrg.c	2012-05-01 08:24:02.607048993 +0200
@@ -20,7 +20,7 @@
 #include <assert.h>
 #include <stdio.h>
 #include <errno.h>
-#include <glib/gtypes.h>
+#include <glib-2.0/glib.h>
 #include "gdrom/gddriver.h"
 #include "dream.h"

--- lxdream-0.9.1-orig/src/gui.h	2009-06-26 07:47:06.000000000 +0200
+++ lxdream-0.9.1/src/gui.h	2012-05-01 08:22:30.870378608 +0200
@@ -19,7 +19,7 @@
 #ifndef lxdream_gui_H
 #define lxdream_gui_H

-#include <glib/gtypes.h>
+#include <glib-2.0/glib.h>

 #ifdef __cplusplus
 extern "C" {
--- lxdream-0.9.1-orig/src/loader.h	2009-06-24 04:48:00.000000000 +0200
+++ lxdream-0.9.1/src/loader.h	2012-05-01 08:25:24.630385649 +0200
@@ -20,7 +20,7 @@
 #define lxdream_loader_H 1

 #include <stdio.h>
-#include <glib/gtypes.h>
+#include <glib-2.0/glib.h>

 #ifdef __cplusplus
 extern "C" {
--- lxdream-0.9.1-orig/src/lxdream.h	2009-06-26 07:47:06.000000000 +0200
+++ lxdream-0.9.1/src/lxdream.h	2012-05-01 08:23:02.663713230 +0200
@@ -20,7 +20,7 @@
 #define lxdream_lxdream_H 1

 #include <stdint.h>
-#include <glib/gtypes.h>
+#include <glib-2.0/glib.h>

 #include "../config.h"

--- lxdream-0.9.1-orig/src/lxpaths.c	2009-06-29 01:04:40.000000000 +0200
+++ lxdream-0.9.1/src/lxpaths.c	2012-05-01 08:21:06.390375184 +0200
@@ -19,8 +19,7 @@
 #include <ctype.h>
 #include <unistd.h>
 #include <wordexp.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gutils.h>
+#include <glib-2.0/glib.h>

 #include "gui.h"
 #include "config.h"
--- lxdream-0.9.1-orig/src/maple/maple.c	2009-06-24 04:48:00.000000000 +0200
+++ lxdream-0.9.1/src/maple/maple.c	2012-05-01 08:22:52.383712814 +0200
@@ -18,7 +18,7 @@
 #define MODULE maple_module

 #include <assert.h>
-#include <glib/gstrfuncs.h>
+#include <glib-2.0/glib.h>
 #include "dream.h"
 #include "mem.h"
 #include "asic.h"
--- lxdream-0.9.1-orig/src/mem.c	2009-06-24 04:48:00.000000000 +0200
+++ lxdream-0.9.1/src/mem.c	2012-05-01 08:25:31.833719273 +0200
@@ -20,7 +20,7 @@
 #include <sys/types.h>
 #include <sys/mman.h>
 #include <sys/stat.h>
-#include <glib/gstrfuncs.h>
+#include <glib-2.0/glib.h>
 #include <assert.h>
 #include <stdint.h>
 #include <stdlib.h>
--- lxdream-0.9.1-orig/src/paths_unix.c	2009-06-25 03:15:28.000000000 +0200
+++ lxdream-0.9.1/src/paths_unix.c	2012-05-01 08:23:10.263713534 +0200
@@ -18,7 +18,7 @@

 #include <string.h>
 #include <stdlib.h>
-#include <glib/gstrfuncs.h>
+#include <glib-2.0/glib.h>

 #include "lxdream.h"
 #include "config.h"
--- lxdream-0.9.1-orig/src/plugin.c	2009-06-26 07:47:07.000000000 +0200
+++ lxdream-0.9.1/src/plugin.c	2012-05-01 08:22:03.577044166 +0200
@@ -20,8 +20,7 @@
 #include <dirent.h>
 #include <dlfcn.h>
 #include <string.h>
-#include <glib/gmem.h>
-#include <glib/gstrfuncs.h>
+#include <glib-2.0/glib.h>
 #include "plugin.h"
 #include "lxpaths.h"

--- lxdream-0.9.1-orig/src/pvr2/glutil.c	2009-06-24 04:48:00.000000000 +0200
+++ lxdream-0.9.1/src/pvr2/glutil.c	2012-05-01 08:25:09.783718374 +0200
@@ -17,7 +17,7 @@
  */

 #include <string.h>
-#include <glib/gstrfuncs.h>
+#include <glib-2.0/glib.h>
 #include "pvr2/glutil.h"

 gboolean isGLSecondaryColorSupported()
--- lxdream-0.9.1-orig/src/sh4/sh4core.h	2009-06-24 04:48:00.000000000 +0200
+++ lxdream-0.9.1/src/sh4/sh4core.h	2012-05-01 08:21:45.847043453 +0200
@@ -19,7 +19,7 @@
 #ifndef lxdream_sh4core_H
 #define lxdream_sh4core_H 1

-#include <glib/gtypes.h>
+#include <glib-2.0/glib.h>
 #include <stdint.h>
 #include <stdio.h>
 #include "mem.h"
--- lxdream-0.9.1-orig/src/syscall.h	2009-06-24 04:48:00.000000000 +0200
+++ lxdream-0.9.1/src/syscall.h	2012-05-01 08:25:37.880386184 +0200
@@ -20,7 +20,7 @@
 #define lxdream_syscall_H 1

 #include <stdint.h>
-#include <glib/gtypes.h>
+#include <glib-2.0/glib.h>

 #ifdef __cplusplus
 extern "C" {
--- lxdream-0.9.1-orig/src/tools/actparse.c	2009-06-24 04:48:00.000000000 +0200
+++ lxdream-0.9.1/src/tools/actparse.c	2012-05-01 08:18:37.360369145 +0200
@@ -21,7 +21,7 @@
 #include <string.h>
 #include <ctype.h>
 #include <sys/stat.h>
-#include <glib/gstrfuncs.h>
+#include <glib-2.0/glib.h>
 #include "tools/gendec.h"

 static int add_action( struct action *actions, struct ruleset *rules, char *operation, const char *file, int line, char *action )
--- lxdream-0.9.1-orig/src/tools/gendec.c	2009-06-24 04:48:00.000000000 +0200
+++ lxdream-0.9.1/src/tools/gendec.c	2012-05-01 08:18:35.510369015 +0200
@@ -23,7 +23,7 @@
 #include <getopt.h>
 #include <errno.h>
 #include <ctype.h>
-#include <glib/gstrfuncs.h>
+#include <glib-2.0/glib.h>
 #include <assert.h>
 #include "tools/gendec.h"

--- lxdream-0.9.1-orig/src/vmu/vmulist.c	2009-06-24 07:10:27.000000000 +0200
+++ lxdream-0.9.1/src/vmu/vmulist.c	2012-05-01 08:24:20.270383038 +0200
@@ -18,8 +18,7 @@

 #include <string.h>
 #include <stdlib.h>
-#include <glib/glist.h>
-#include <glib/gstrfuncs.h>
+#include <glib-2.0/glib.h>
 #include "vmulist.h"
 #include "config.h"

--- lxdream-0.9.1-orig/src/vmu/vmuvol.c	2009-06-29 01:04:40.000000000 +0200
+++ lxdream-0.9.1/src/vmu/vmuvol.c	2012-05-01 08:24:24.047049852 +0200
@@ -16,8 +16,7 @@
  * GNU General Public License for more details.
  */

-#include <glib/gmem.h>
-#include <glib/gstrfuncs.h>
+#include <glib-2.0/glib.h>
 #include <string.h>
 #include <unistd.h>
 #include <stdio.h>
