--- configure.in~       2013-12-10 15:19:22.000000000 +0200
+++ configure.in        2013-12-10 21:53:37.258654408 +0200
@@ -74,20 +74,6 @@

 dnl For starters, do we have a working objective-c compiler?
 AC_HAVE_OBJC([
-   AC_CHECK_HEADER([Cocoa/Cocoa.h], [
-      HAVE_COCOA='yes'
-      APPLE_BUILD='yes'
-      LIBS="$LIBS -framework AppKit -framework Carbon"
-      LIBS="$LIBS -Wl,-dylib_file,/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib:/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib -framework OpenGL"
-      LXDREAMCPPFLAGS="-DMAC_OS_X_VERSION_MIN_REQUIRED=MAC_OS_X_VERSION_10_4 -mdynamic-no-pic"
-      LDFLAGS="$LDFLAGS -Wl,-headerpad_max_install_names"
-      AC_DEFINE(HAVE_COCOA,[1],[Have Cocoa framework])
-      AC_DEFINE(APPLE_BUILD,[1],[Building on an apple platform. Things are different...])
-      if test "x$with_gtk" = "xx11"; then
-        with_gtk=no
-      fi
-   ], [], [AC_INCLUDES_DEFAULT])
-
    AC_OBJC_VERSION([ OBJCOPT="-O2 -msse2 -mfpmath=sse"
                      OBJCFLAGS="$OBJCFLAGS -fno-strict-aliasing" ],
                    [ OBJCOPT="-O2" ], [] )
