--- configure.in~       2013-12-09 22:39:34.905038638 +0200
+++ configure.in        2013-12-09 22:40:59.966034179 +0200
@@ -191,9 +191,8 @@
 PKG_CHECK_MODULES(LIBPNG, [libpng] )

 dnl Implied by libpng, but check explicitly just in case
-AC_CHECK_LIB(z, uncompress, [], [
-    echo "Zlib (libz.so) could not be found, but is required."
-    exit 1])
+PKG_CHECK_EXISTS([zlib], [ZLIB="zlib"],
+    [AC_CHECK_LIB([z], [uncompress], [], [AC_MSG_ERROR([Unable to find zlib.])])])

 if test "x$with_gtk" = "xno"; then
    dnl Check for GLIB only 
