Index: dietlibc-0.30/dietfeatures.h
===================================================================
--- dietlibc-0.30.orig/dietfeatures.h
+++ dietlibc-0.30/dietfeatures.h
@@ -99,14 +99,14 @@
 /* WARNING: this appears to break with some binutils versions.  Works
  * for me with binutils 2.15.  The symptom is an error message that
  * `main' can not be found. */
-/* #define WANT_STACKGAP */
+#define WANT_STACKGAP
 
 /* Include support for ProPolice/SSP, calls guard_setup */
 /* ProPolice is part of gcc 4.1 and up, there were patches for earlier
  * versions.  To make use of this, compile your application with
  * -fstack-protector. */
 #if (__GNUC__>4) || ((__GNUC__==4) && (__GNUC_MINOR__>=1))
-#define WANT_SSP
+//#define WANT_SSP
 #endif
 
 
