diff -Naupr tagpy-0.94.8.orig/aksetup_helper.py tagpy-0.94.8/aksetup_helper.py
--- tagpy-0.94.8.orig/aksetup_helper.py	2010-07-20 04:38:24.000000000 +0200
+++ tagpy-0.94.8/aksetup_helper.py	2011-07-31 17:01:34.975381171 +0200
@@ -502,7 +502,7 @@ class Libraries(StringListOption):
 class BoostLibraries(Libraries):
     def __init__(self, lib_base_name):
         Libraries.__init__(self, "BOOST_%s" % lib_base_name.upper(), 
-                ["boost_%s-${BOOST_COMPILER}-mt" % lib_base_name],
+                ["boost_%s" % lib_base_name],
                 help="Library names for Boost C++ %s library (without lib or .so)" 
                     % humanize(lib_base_name))
 
 
