diff -rupN qtiocompressor-2.3_1-opensource.old/qtiocompressor.pro qtiocompressor-2.3_1-opensource/qtiocompressor.pro
--- qtiocompressor-2.3_1-opensource.old/qtiocompressor.pro  2009-12-16 05:40:11.000000000 -0500
+++ qtiocompressor-2.3_1-opensource/qtiocompressor.pro  2010-07-05 13:12:33.000000000 -0400
@@ -2,4 +2,3 @@ TEMPLATE=subdirs
 CONFIG += ordered
 include(common.pri)
 qtiocompressor-uselib:SUBDIRS=buildlib
-SUBDIRS+=examples

