--- swig/python/Makefile.in.orig	2010-06-23 11:42:52.317894087 +0300
+++ swig/python/Makefile.in	2010-06-23 11:43:01.848121356 +0300
@@ -459,13 +459,6 @@
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(pkgpythondir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(pkgpythondir)" || exit $$?; \
 	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pkgpythondir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pkgpythondir)" $$dlist; \
-	  fi; \
-	else :; fi
 
 uninstall-pkgpythonPYTHON:
 	@$(NORMAL_UNINSTALL)
