--- blueman-1.23/blueman/plugins/applet/StatusIcon.py.orig	2012-04-26 23:52:01.900955192 +0200
+++ blueman-1.23/blueman/plugins/applet/StatusIcon.py	2012-04-26 23:52:34.882160450 +0200
@@ -68,7 +68,7 @@
 		entry.connect("changed", self.on_entry_changed, ic, image)
 		
 	__options__ = {"icon": {"type": str,
-							"default": "blueman-tray",
+							"default": "bluetooth",
 							"name": _("Icon Name"),
 							"desc": _("Custom icon to use for the notification area"),
 							"decorator": widget_decorator
@@ -166,7 +166,7 @@
 		self.QueryVisibility()
 		
 	def on_status_icon_resized(self):
-		self.icon = "blueman-tray"
+		self.icon = "bluetooth"
 		
 		#p = gtk.gdk.Pixbuf(gtk.gdk.COLORSPACE_RGB, True, 8, 22, 22)
 		#p.fill(0)