Index: blueman-1.10/blueman/gui/manager/ManagerMenu.py
===================================================================
--- blueman-1.10.orig/blueman/gui/manager/ManagerMenu.py
+++ blueman-1.10/blueman/gui/manager/ManagerMenu.py
@@ -55,12 +55,12 @@ class ManagerMenu:
 		self.item_help.set_submenu(help_menu)
 		help_menu.show()
 		
-		item = create_menuitem(_("Get Help Online..."), get_icon("gnome-help", 16))
+		item = create_menuitem(_("Get Help Online..."), get_icon("help-browser", 16))
 		item.connect("activate", lambda x: spawn(["xdg-open", WEBSITE+"/forum"], True))
 		help_menu.append(item)
 		item.show()
 		
-		item = create_menuitem(_("Translate This Application..."), get_icon("config-language", 16))
+		item = create_menuitem(_("Translate This Application..."), get_icon("preferences-desktop-locale", 16))
 		item.connect("activate", lambda x: spawn(["xdg-open", "https://translations.launchpad.net/blueman"], True))
 		help_menu.append(item)
 		item.show()
