diff -Nuar vigra-Version-1-10-0.orig/CMakeLists.txt vigra-Version-1-10-0/CMakeLists.txt
--- vigra-Version-1-10-0.orig/CMakeLists.txt	2013-11-18 18:48:16.000000000 +0200
+++ vigra-Version-1-10-0/CMakeLists.txt	2014-05-30 20:08:32.469162810 +0300
@@ -128,7 +128,6 @@
 #
 ##################################################
 
-ADD_CUSTOM_TARGET(doc)
 
 # Automatically push the latest documentation to github's gh-pages.
 # This only works when ${DOCDIR} refers to a VIGRA repository that is 
@@ -163,7 +162,6 @@
 
 ADD_SUBDIRECTORY(src)
 ADD_SUBDIRECTORY(test)
-ADD_SUBDIRECTORY(docsrc)
 
 IF(WITH_VIGRANUMPY)
     ADD_SUBDIRECTORY(vigranumpy)
diff -Nuar vigra-Version-1-10-0.orig/vigranumpy/CMakeLists.txt vigra-Version-1-10-0/vigranumpy/CMakeLists.txt
--- vigra-Version-1-10-0.orig/vigranumpy/CMakeLists.txt	2013-11-18 18:48:16.000000000 +0200
+++ vigra-Version-1-10-0/vigranumpy/CMakeLists.txt	2014-05-30 20:13:22.309173635 +0300
@@ -30,7 +30,6 @@
     ADD_SUBDIRECTORY(src)
     ADD_SUBDIRECTORY(lib)
     ADD_SUBDIRECTORY(test)
-    ADD_SUBDIRECTORY(docsrc)
 
     IF(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/private/CMakeLists.txt)
         ADD_SUBDIRECTORY(private EXCLUDE_FROM_ALL)
