Index: vigra-1.7.0-src/CMakeLists.txt
===================================================================
--- vigra-1.7.0-src.orig/CMakeLists.txt
+++ vigra-1.7.0-src/CMakeLists.txt
@@ -74,7 +74,6 @@ ENABLE_TESTING()
 #
 ##################################################
 
-ADD_CUSTOM_TARGET(doc)
 
 ##################################################
 #
@@ -93,7 +92,6 @@ ADD_CUSTOM_TARGET(experiments)
 INCLUDE_DIRECTORIES(${PROJECT_BINARY_DIR}/include)
 ADD_SUBDIRECTORY(src)
 ADD_SUBDIRECTORY(test)
-ADD_SUBDIRECTORY(docsrc)
 
 IF(WITH_VIGRANUMPY)
     ADD_SUBDIRECTORY(vigranumpy)
Index: vigra-1.7.0-src/vigranumpy/CMakeLists.txt
===================================================================
--- vigra-1.7.0-src.orig/vigranumpy/CMakeLists.txt
+++ vigra-1.7.0-src/vigranumpy/CMakeLists.txt
@@ -24,7 +24,6 @@ IF(VIGRANUMPY_DEPENDENCIES_FOUND)
     ADD_SUBDIRECTORY(src)
     ADD_SUBDIRECTORY(lib)
     ADD_SUBDIRECTORY(test)
-    ADD_SUBDIRECTORY(docsrc)
 
     IF(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/private)
         ADD_SUBDIRECTORY(private EXCLUDE_FROM_ALL)
