--- scripts/checkelf~   2014-09-21 15:38:00.978702106 +0200
+++ scripts/checkelf    2014-09-21 15:32:49.437713489 +0200
@@ -157,7 +157,7 @@
     result_deps = []
     for elf_files, paths_and_deps in result_dependencies.items():
         for data in paths_and_deps:
-            if not data[1] == "broken" and not data[1] == package_name and not data[1].replace("-32bit", "") == "glibc":
+            if not data[1] == "broken" and not data[1] == package_name and not data[1] == "glibc":
                 if (data[1].startswith("xorg-video-nvidia") or data[1].startswith("xorg-video-fglrx")) \
                     and not (package_name.startswith("xorg-video-nvidia") or package_name.startswith("xorg-video-fglrx")):
                     result_deps.append("mesa")
