Index: libfprint-0.1.0-pre2/libfprint/Makefile.am
===================================================================
--- libfprint-0.1.0-pre2.orig/libfprint/Makefile.am
+++ libfprint-0.1.0-pre2/libfprint/Makefile.am
@@ -54,7 +54,7 @@ NBIS_SRC = \
 
 libfprint_la_CFLAGS = -fvisibility=hidden -I$(srcdir)/nbis/include $(LIBUSB_CFLAGS) $(GLIB_CFLAGS) $(CRYPTO_CFLAGS) $(AM_CFLAGS)
 libfprint_la_LDFLAGS = -version-info @lt_major@:@lt_revision@:@lt_age@
-libfprint_la_LIBADD = -lm $(LIBUSB_LIBS) $(GLIB_LIBS) $(CRYPTO_LIBS)
+libfprint_la_LIBADD = -lm -lrt $(LIBUSB_LIBS) $(GLIB_LIBS) $(CRYPTO_LIBS)
 
 fprint_list_hal_info_SOURCES = fprint-list-hal-info.c
 fprint_list_hal_info_CFLAGS = -fvisibility=hidden -I$(srcdir)/nbis/include $(LIBUSB_CFLAGS) $(GLIB_CFLAGS) $(IMAGEMAGICK_CFLAGS) $(CRYPTO_CFLAGS) $(AM_CFLAGS)
