--- plasma-widget-message-indicator-0.5.8-orig/src/listenermodel.cpp	2011-08-31 19:21:30.000000000 +0300
+++ plasma-widget-message-indicator-0.5.8/src/listenermodel.cpp	2013-01-20 17:01:48.368830471 +0200
@@ -26,7 +26,7 @@
 #include <KIconLoader>
 
 // libindicate
-#include <libindicate/indicator-messages.h>
+#include <libindicate-0.7/libindicate/indicator-messages.h>
 
 // libindicate-qt
 #include <qindicatedecode.h>
