Index: plasma-widget-message-indicator-0.5.4/src/plasma-applet-indicatordisplay.desktop
===================================================================
--- plasma-widget-message-indicator-0.5.4.orig/src/plasma-applet-indicatordisplay.desktop
+++ plasma-widget-message-indicator-0.5.4/src/plasma-applet-indicatordisplay.desktop
@@ -1,6 +1,8 @@
 [Desktop Entry]
 Name=Indicator Display
+Name[tr]=İleti Göstergesi
 Comment=Show new messages from various applications
+Comment[tr]=Çeşitli uygulamaların yeni iletilerini göster
 Icon=mail-message-new
 Type=Service
 X-KDE-ServiceTypes=Plasma/Applet
Index: plasma-widget-message-indicator-0.5.4/src/plasma-applet-message-indicator.desktop
===================================================================
--- plasma-widget-message-indicator-0.5.4.orig/src/plasma-applet-message-indicator.desktop
+++ plasma-widget-message-indicator-0.5.4/src/plasma-applet-message-indicator.desktop
@@ -1,6 +1,8 @@
 [Desktop Entry]
 Name=Message Indicator
+Name[tr]=İleti Göstergesi
 Comment=Show new messages from various applications
+Comment[tr]=Çeşitli uygulamaların yeni iletilerini göster
 Icon=mail-message-new
 Type=Service
 X-KDE-ServiceTypes=Plasma/Applet
