Index: Makefile.in
===================================================================
--- Makefile.in.orig
+++ Makefile.in
@@ -1009,7 +1009,7 @@ man_MANS = src/bluetoothd.8
 @DFUTOOL_TRUE@tools_dfutool_LDADD = @USB_LIBS@
 @USB_TRUE@tools_dfubabel_LDADD = @USB_LIBS@
 @USB_TRUE@tools_avctrl_LDADD = @USB_LIBS@
-@CUPS_TRUE@cupsdir = $(libdir)/cups/backend
+@CUPS_TRUE@cupsdir = `cups-config --serverbin`/backend
 @CUPS_TRUE@cups_bluetooth_SOURCES = $(gdbus_sources) cups/main.c cups/cups.h \
 @CUPS_TRUE@					cups/sdp.c cups/spp.c cups/hcrp.c
 
