diff -Naur tiff-3.9.1/Makefile.am tiff-3.9.1.oden/Makefile.am
--- tiff-3.9.1/Makefile.am	2009-08-21 18:46:01.000000000 +0200
+++ tiff-3.9.1.oden/Makefile.am	2009-08-30 12:43:13.000000000 +0200
@@ -47,7 +47,7 @@
 
 dist_doc_DATA = $(docfiles)
 
-SUBDIRS = port libtiff tools build contrib test man html
+SUBDIRS = port libtiff tools build test man html
 
 release:
 	(rm -f $(top_srcdir)/RELEASE-DATE && echo $(LIBTIFF_RELEASE_DATE) > $(top_srcdir)/RELEASE-DATE)
diff -Naur tiff-3.9.1/Makefile.in tiff-3.9.1.oden/Makefile.in
--- tiff-3.9.1/Makefile.in	2009-08-28 20:41:13.000000000 +0200
+++ tiff-3.9.1.oden/Makefile.in	2009-08-30 12:43:33.000000000 +0200
@@ -331,7 +331,7 @@
 	nmake.opt
 
 dist_doc_DATA = $(docfiles)
-SUBDIRS = port libtiff tools build contrib test man html
+SUBDIRS = port libtiff tools build test man html
 all: all-recursive
 
 .SUFFIXES:
