--- Makefile.orig	2008-08-09 23:22:37.000000000 +0300
+++ Makefile	2009-05-30 23:05:28.574864877 +0300
@@ -283,7 +283,7 @@
   SHLIBLDFLAGS=-shared $(LDFLAGS)
 
   THREAD_LDFLAGS=-lpthread
-  LDFLAGS=-ldl -lm
+  LDFLAGS+=-ldl -lm
 
   CLIENT_LDFLAGS=$(shell sdl-config --libs) -lGL
 
@@ -560,7 +560,7 @@
 
   THREAD_LDFLAGS=-lpthread
   # don't need -ldl (FreeBSD)
-  LDFLAGS=-lm
+  LDFLAGS+=-lm
 
   CLIENT_LDFLAGS =
 
@@ -1452,13 +1452,13 @@
 
 $(B)/openarena.$(ARCH)$(BINEXT): $(Q3OBJ) $(Q3POBJ) $(LIBSDLMAIN)
 	$(echo_cmd) "LD $@"
-	$(Q)$(CC) -o $@ $(Q3OBJ) $(Q3POBJ) $(CLIENT_LDFLAGS) \
-		$(LDFLAGS) $(LIBSDLMAIN)
+	$(Q)$(CC) $(LDFLAGS) $(Q3OBJ) $(Q3POBJ) $(CLIENT_LDFLAGS) \
+		$(LIBSDLMAIN) -o $@
 
 $(B)/openarena-smp.$(ARCH)$(BINEXT): $(Q3OBJ) $(Q3POBJ_SMP) $(LIBSDLMAIN)
 	$(echo_cmd) "LD $@"
-	$(Q)$(CC) -o $@ $(Q3OBJ) $(Q3POBJ_SMP) $(CLIENT_LDFLAGS) \
-		$(THREAD_LDFLAGS) $(LDFLAGS) $(LIBSDLMAIN)
+	$(Q)$(CC) $(LDFLAGS) $(Q3OBJ) $(Q3POBJ_SMP) $(CLIENT_LDFLAGS) \
+		$(THREAD_LDFLAGS) $(LIBSDLMAIN) -o $@
 
 ifneq ($(strip $(LIBSDLMAIN)),)
 ifneq ($(strip $(LIBSDLMAINSRC)),)
@@ -1584,7 +1584,7 @@
 
 $(B)/oa_ded.$(ARCH)$(BINEXT): $(Q3DOBJ)
 	$(echo_cmd) "LD $@"
-	$(Q)$(CC) -o $@ $(Q3DOBJ) $(LDFLAGS)
+	$(Q)$(CC) $(LDFLAGS) $(Q3DOBJ) -lm -ldl -o $@
 
 
 
