diff -Nur ppracer-0.5alpha-old/data/config.nut.in ppracer-0.5alpha/data/config.nut.in
--- ppracer-0.5alpha-old/data/config.nut.in	2006-08-23 17:35:59.766872184 +0300
+++ ppracer-0.5alpha/data/config.nut.in	2006-08-23 17:36:09.520389424 +0300
@@ -120,7 +120,7 @@
 ui_language_description = @"
 Sets the language for the ui
 ",
-ui_language = "en_GB",
+ui_language = "tr_TR",
 
 always_save_event_race_data_description = @"
 only for cheating purpose
