diff -Nur ppracer-0.5alpha-old/data/Makefile.am ppracer-0.5alpha/data/Makefile.am
--- ppracer-0.5alpha-old/data/Makefile.am	2007-01-13 20:55:47.592509266 +0200
+++ ppracer-0.5alpha/data/Makefile.am	2007-01-13 20:56:13.708539025 +0200
@@ -207,6 +207,7 @@
 		translations/uk_UA.nut	\
 		translations/hu_HU.nut	\
 		translations/sv_SE.nut	\
+		translations/tr_TR.nut	\
 		translations/languages.nut	\
 		sounds/tux_hit_tree1.wav	\
 		sounds/tux_on_snow1.wav	\
diff -Nur ppracer-0.5alpha-old/data/Makefile.in ppracer-0.5alpha/data/Makefile.in
--- ppracer-0.5alpha-old/data/Makefile.in	2007-01-13 20:55:47.592509266 +0200
+++ ppracer-0.5alpha/data/Makefile.in	2007-01-13 20:56:13.709538873 +0200
@@ -383,6 +383,7 @@
 		translations/uk_UA.nut	\
 		translations/hu_HU.nut	\
 		translations/sv_SE.nut	\
+		translations/tr_TR.nut	\
 		translations/languages.nut	\
 		sounds/tux_hit_tree1.wav	\
 		sounds/tux_on_snow1.wav	\
diff -Nur ppracer-0.5alpha-old/data/translations/languages.nut ppracer-0.5alpha/data/translations/languages.nut
--- ppracer-0.5alpha-old/data/translations/languages.nut	2007-01-13 20:55:47.592509266 +0200
+++ ppracer-0.5alpha/data/translations/languages.nut	2007-01-13 20:56:13.746533249 +0200
@@ -24,3 +24,4 @@
 Translation.addLanguage("ru_RU", "Русский");
 Translation.addLanguage("uk_UA", "Українська");
 Translation.addLanguage("sv_SE", "Svenska");
+Translation.addLanguage("tr_TR", "Türkçe");
diff -Nur ppracer-0.5alpha-old/data/translations/tr_TR.nut ppracer-0.5alpha/data/translations/tr_TR.nut
--- ppracer-0.5alpha-old/data/translations/tr_TR.nut	1970-01-01 02:00:00.000000000 +0200
+++ ppracer-0.5alpha/data/translations/tr_TR.nut	2007-01-13 21:47:44.856613089 +0200
@@ -0,0 +1,143 @@
+
+//Translator: Uğur Çetin - jnmbk@users.sourceforge.net
+//Translation update and fixes: Onur Küçük - onur@pardus.org.tr
+
+//common strings
+Translation.set("Back","Geri");
+Translation.set("PRESS ANY KEY TO START","BAŞLAMAK İÇİN BİR TUŞA BASIN");
+
+//game_type_select
+Translation.set("Enter an event","Turnuva");
+Translation.set("Practice","Alıştırma");
+Translation.set("Configuration","Ayarlar");
+Translation.set("Credits","Hakkında");
+Translation.set("Quit","Çıkış");
+
+//configuration
+Translation.set("General","Genel");
+Translation.set("Graphics","Grafik");
+Translation.set("Video","Görüntü");
+Translation.set("Audio","Ses");
+Translation.set("Keyboard","Klavye");
+Translation.set("Joystick","Joystick");
+
+//config mode
+Translation.set("Cancel","İptal");
+Translation.set("Ok","Tamam");
+
+//general configuration
+Translation.set("General Configuration","Genel Ayarlar");
+Translation.set("Language:","Dil:");
+Translation.set("Show UI Snow:","Kar:");
+Translation.set("Display FPS:","FPS Göster:");
+Translation.set("Display Progress Bar:","İlerleme Çubuğu:");
+
+//video configuration
+Translation.set("Video Configuration","Görüntü Ayarları");
+Translation.set("Resolution:","Çözünürlük:");
+Translation.set("Bits Per Pixel:","Renk Derinliği:");
+Translation.set("Fullscreen:","Tam Ekran:");
+Translation.set("Stencil Buffer:","Palet Belleği:");
+Translation.set("Experimental (needs restart)","Deneysel (baştan başlatma gerektirir)");
+Translation.set("Enable FSAA:","FSAA Etkin:");
+Translation.set("FSAA Multisamples:","FSAA Örnek Adedi:");
+
+//graphics configuration
+Translation.set("Graphics Configuration","Grafik Ayarları");
+Translation.set("Draw Fog:","Sis:");
+Translation.set("Reflections:","Yansımalar:");
+Translation.set("Shadows:","Gölgeler:");
+Translation.set("Detail settings","Detay ayarları");
+Translation.set("Terrain:","Arkaplan:");
+Translation.set("Textures:","Kaplamalar:");
+Translation.set("Player Model:","Oyuncu Modeli:");
+
+//audio configuration
+Translation.set("Audio Configuration","Ses Ayarları");
+Translation.set("Sound Volume:","Ses Seviyesi:");
+Translation.set("Music Volume:","Müzik Seviyesi:");
+Translation.set("(needs restart)","(baştan başlatma gerektirir)");
+Translation.set("Disable Audio:","Sesi Kapat:");
+Translation.set("Stereo:","Stereo:");
+Translation.set("Bits Per Sample:","Örnekleme Başına Bit:");
+Translation.set("Samples Per Second:","Saniye Başına Örnekleme:");
+
+//keyboard configuration
+Translation.set("Keyboard Configuration","Klavye Ayarları");
+Translation.set("Turn left:","Sola dön:");
+Translation.set("Turn right:","Sağa dön:");
+Translation.set("Paddle:","Ayakları oynat:");
+Translation.set("Brake:","Fren:");
+Translation.set("Jump:","Zıpla:");
+Translation.set("Trick:","Numara yap:");
+Translation.set("Reset:","Piste dön:");
+Translation.set("Pause:","Duraklat:");
+
+//joystick configuration
+Translation.set("Joystick Configuration","Joystick ayarları");
+Translation.set("Enable Joystick","Joystick etkin");
+
+//race select
+Translation.set("Race!","Başla!");
+Translation.set("Select a race","Yarış seç");
+Translation.set("Contributed by:","Pisti yapan:");
+Translation.set("Unknown","Bilinmiyor");
+Translation.set("Time:","Süre:");
+Translation.set("Herring:","Balık:");
+Translation.set("Score:","Puan:");
+
+//event select
+Translation.set("Continue","Devam et");
+Translation.set("Select event and cup","Turnuva ve yarış seçin");
+Translation.set("Event:","Yarış:");
+Translation.set("Cup:","Turnuva:");
+Translation.set("You've won this cup!","Bu turnuvayı kazandınız!");
+Translation.set("You must complete this cup next","Şimdi bu turnuvayı kazanmalısınız");
+Translation.set("You cannot enter this cup yet","Bu turnuvaya katılamazsınız");
+
+//event race select
+Translation.set("You don't have any lives left.","Hiç canınız kalmadı");
+Translation.set("Race won! Your result:","Kazandınız! Sonuçlar:");
+Translation.set("Needed to advance:","Gerekli olan:");
+Translation.set("You can't enter this race yet.","Bu yarışa henüz katılamazsınız.");
+
+//loading
+Translation.set("Loading, Please Wait...","Yükleniyor, Lütfen Bekleyin");
+
+//paused
+Translation.set("Resume","Devam Et");
+Translation.set("Paused","Duraklatıldı");
+
+//race over
+Translation.set("Race Over","Yarış Bitti");
+Translation.set("Time: %02d:%02d.%02d","Süre: %02d:%02d.%02d");
+Translation.set("Herring: %3d","Balık: %3d");
+Translation.set("Score: %6d","Puan: %6d");
+Translation.set("Max speed: %3d km/h","En fazla hız: %3d km/s");
+Translation.set("Was flying: %.01f %% of time","Uçma yüzdesi: %% %.01f");
+Translation.set("Race aborted","Yarış iptal edildi");
+Translation.set("You beat your best score!","En iyi puanınızı geçtiniz!");
+Translation.set("Congratulations! You won the event!","Tebrikler, yarışı kazandınız!");
+Translation.set("Congratulations! You won the cup!","Tebrikler, kupa sizin!");
+Translation.set("You advanced to the next race!","Sonraki yarışa katılmaya hak kazandınız!");
+Translation.set("You didn't advance.","Sonraki yarışa katılamazsınız.");
+
+//HUD
+//Strings are prefixed to prevent collisions with the UI
+Translation.set("H|FPS: %.1f","FPS: %.1f");
+Translation.set("H|Time:","Süre:");
+Translation.set("H|km/h","km/s");
+//herring counter
+Translation.set("H|%03d","%03d");
+//time counter
+Translation.set("H|%02d:%02d:%02d","%02d:%02d:%02d");
+//speed counter
+Translation.set("H|%.0f","%.0f");
+
+//Credits
+Translation.set("Development Team","Geliştirme Ekibi");
+Translation.set("Translators","Çeviriciler");
+Translation.set("Contributors","Katkıcılar");
+Translation.set("Tux Racer Development Team","Tux Racer Geliştirme Ekibi");
+Translation.set("Music","Müzik");
+Translation.set("Thanks To:","Teşekkürler:");
diff -Nur ppracer-0.5alpha-old/src/credits.cpp ppracer-0.5alpha/src/credits.cpp
--- ppracer-0.5alpha-old/src/credits.cpp	2007-01-13 20:55:47.589509721 +0200
+++ ppracer-0.5alpha/src/credits.cpp	2007-01-13 21:49:18.241416440 +0200
@@ -59,6 +59,8 @@
 	{false, "credits_text", "Sergej Kotliar" },	
 	{false, "credits_text", "Theo Snelleman" },	
 	{false, "credits_text", "Trygve B. Wiig" },
+	{false, "credits_text", "Uğur Çetin" },
+	{false, "credits_text", "Onur Küçük" },
     {false, "credits_text", "" },
 	{true,  "credits_h2", "Contributors" },
 	{false, "credits_text", "Peter Reichel" },
