--- webkitgtk-1.11.1/configure.ac~	2012-11-07 01:11:10.290000095 +0200
+++ webkitgtk-1.11.1/configure.ac	2012-11-07 01:11:10.292000112 +0200
@@ -338,24 +338,18 @@
 
 AC_MSG_CHECKING([the GStreamer version to use])
 AC_ARG_WITH([gstreamer],
-        [AS_HELP_STRING([--with-gstreamer=0.10|1.0], [the GStreamer version to use (default: 0.10)])],
+        [AS_HELP_STRING([--with-gstreamer=1.0], [the GStreamer version to use (default: 1.0)])],
         [case "$with_gstreamer" in
-        0.10|1.0) ;;
+        1.0) ;;
         *) AC_MSG_ERROR([invalid GStreamer version specified]) ;;
         esac],
-        [with_gstreamer=0.10])
+        [with_gstreamer=1.0])
 AC_MSG_RESULT([$with_gstreamer])
 
-GSTREAMER_0_10_REQUIRED_VERSION=0.10
-GSTREAMER_0_10_PLUGINS_BASE_REQUIRED_VERSION=0.10.30
 GSTREAMER_1_0_REQUIRED_VERSION=0.11.90
 GSTREAMER_1_0_PLUGINS_BASE_REQUIRED_VERSION=0.11.90
 
 case "$with_gstreamer" in
-     0.10) GSTREAMER_REQUIRED_VERSION=$GSTREAMER_0_10_REQUIRED_VERSION
-           GSTREAMER_PLUGINS_BASE_REQUIRED_VERSION=$GSTREAMER_0_10_PLUGINS_BASE_REQUIRED_VERSION
-           GST_API_VERSION=0.10
-           ;;
      1.0) GSTREAMER_REQUIRED_VERSION=$GSTREAMER_1_0_REQUIRED_VERSION
            GSTREAMER_PLUGINS_BASE_REQUIRED_VERSION=$GSTREAMER_1_0_PLUGINS_BASE_REQUIRED_VERSION
            GST_API_VERSION=1.0
--- webkitgtk-1.11.1/configure~	2012-11-07 01:18:37.757999787 +0200
+++ webkitgtk-1.11.1/configure	2012-11-07 01:18:37.793000111 +0200
@@ -17713,22 +17713,16 @@
         *) as_fn_error $? "invalid GStreamer version specified" "$LINENO" 5 ;;
         esac
 else
-  with_gstreamer=0.10
+  with_gstreamer=1.0
 fi
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $with_gstreamer" >&5
 $as_echo "$with_gstreamer" >&6; }
 
-GSTREAMER_0_10_REQUIRED_VERSION=0.10
-GSTREAMER_0_10_PLUGINS_BASE_REQUIRED_VERSION=0.10.30
 GSTREAMER_1_0_REQUIRED_VERSION=0.11.90
 GSTREAMER_1_0_PLUGINS_BASE_REQUIRED_VERSION=0.11.90
 
 case "$with_gstreamer" in
-     0.10) GSTREAMER_REQUIRED_VERSION=$GSTREAMER_0_10_REQUIRED_VERSION
-           GSTREAMER_PLUGINS_BASE_REQUIRED_VERSION=$GSTREAMER_0_10_PLUGINS_BASE_REQUIRED_VERSION
-           GST_API_VERSION=0.10
-           ;;
      1.0) GSTREAMER_REQUIRED_VERSION=$GSTREAMER_1_0_REQUIRED_VERSION
            GSTREAMER_PLUGINS_BASE_REQUIRED_VERSION=$GSTREAMER_1_0_PLUGINS_BASE_REQUIRED_VERSION
            GST_API_VERSION=1.0
