diff -ur webkit-1.7.2/Source/WebKit/gtk/JSCore.gir.in webkit-1.7.2.patched/Source/WebKit/gtk/JSCore.gir.in
--- webkit-1.7.2/Source/WebKit/gtk/JSCore.gir.in	2011-10-27 22:51:16.000000000 +0200
+++ webkit-1.7.2.patched/Source/WebKit/gtk/JSCore.gir.in	2011-12-09 16:08:52.268130983 +0100
@@ -2,7 +2,7 @@
 <repository version="1.2"
             xmlns="http://www.gtk.org/introspection/core/1.0"
             xmlns:c="http://www.gtk.org/introspection/c/1.0">
-  <namespace name="JSCore" version="@WEBKITGTK_API_VERSION@" shared-library="webkitgtk-@WEBKITGTK_API_VERSION@">
+  <namespace name="JSCore" version="@WEBKITGTK_API_VERSION@" shared-library="libwebkitgtk-@WEBKITGTK_API_VERSION@.so.0">
     <alias name="GlobalContextRef" target="none">
       <type name="JSGlobalContextRef" c:type="JSGlobalContextRef"/>
     </alias>
