Index: Source/WebKit/gtk/GNUmakefile.am
===================================================================
--- Source/WebKit/gtk/GNUmakefile.am.orig
+++ Source/WebKit/gtk/GNUmakefile.am
@@ -71,7 +71,7 @@ libwebkitgtk_@WEBKITGTK_API_MAJOR_VERSIO
 
 libwebkitgtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la_LDFLAGS = \
 	-version-info @LIBWEBKITGTK_VERSION@ \
-	$(version_script) \
+	$(version_script) -Wl,--reduce-memory-overheads -Wl,--no-keep-memory \
 	$(no_undefined)
 
 libwebkitgtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la_LIBADD = \
Index: Source/WebKit2/GNUmakefile.am
===================================================================
--- Source/WebKit2/GNUmakefile.am.orig
+++ Source/WebKit2/GNUmakefile.am
@@ -151,7 +151,7 @@ endif
 
 libwebkit2gtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la_LDFLAGS = \
 	-version-info @LIBWEBKIT2GTK_VERSION@ \
-	$(version_script) \
+	$(version_script) -Wl,--reduce-memory-overheads -Wl,--no-keep-memory \
 	$(no_undefined)
 
 libwebkit2gtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la_LIBADD = \
Index: GNUmakefile.in
===================================================================
--- GNUmakefile.in.orig
+++ GNUmakefile.in
@@ -22263,7 +22263,7 @@ libwebkitgtk_@WEBKITGTK_API_MAJOR_VERSIO
 
 libwebkitgtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la_LDFLAGS = \
 	-version-info @LIBWEBKITGTK_VERSION@ \
-	$(version_script) \
+	$(version_script) -Wl,--reduce-memory-overheads -Wl,--no-keep-memory \
 	$(no_undefined)
 
 libwebkitgtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la_LIBADD = \
@@ -22805,7 +22805,7 @@ libwebkit2gtk_@WEBKITGTK_API_MAJOR_VERSI
 
 libwebkit2gtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la_LDFLAGS = \
 	-version-info @LIBWEBKIT2GTK_VERSION@ \
-	$(version_script) \
+	$(version_script) -Wl,--reduce-memory-overheads -Wl,--no-keep-memory \
 	$(no_undefined)
 
 libwebkit2gtk_@WEBKITGTK_API_MAJOR_VERSION@_@WEBKITGTK_API_MINOR_VERSION@_la_LIBADD =  \
