--- webkitgtk-1.10.0/GNUmakefile.in.link	2012-09-24 15:48:23.000000000 +0800
+++ webkitgtk-1.10.0/GNUmakefile.in	2012-09-29 19:47:47.000000000 +0800
@@ -1380,7 +1380,8 @@
 	$(Libraries_libTestRunnerInjectedBundle_la_LDFLAGS) $(LDFLAGS) \
 	-o $@
 @ENABLE_WEBKIT2_TRUE@am_Libraries_libTestRunnerInjectedBundle_la_rpath =
-Libraries_libTestWebKitAPIInjectedBundle_la_LIBADD =
+Libraries_libTestWebKitAPIInjectedBundle_la_LIBADD = libWTF.la \
+	libwebkit2gtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la $(GTK_LIBS)
 am__Libraries_libTestWebKitAPIInjectedBundle_la_SOURCES_DIST =  \
 	Tools/TestWebKitAPI/InjectedBundleController.cpp \
 	Tools/TestWebKitAPI/InjectedBundleController.h \
@@ -1474,7 +1475,7 @@
 	$(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=link $(CXXLD) \
 	$(Libraries_libgtest_la_CXXFLAGS) $(CXXFLAGS) $(AM_LDFLAGS) \
 	$(LDFLAGS) -o $@
-TestNetscapePlugin_libtestnetscapeplugin_la_LIBADD =
+TestNetscapePlugin_libtestnetscapeplugin_la_LIBADD = -lX11
 am__TestNetscapePlugin_libtestnetscapeplugin_la_SOURCES_DIST = Tools/DumpRenderTree/TestNetscapePlugIn/Tests/DocumentOpenInDestroyStream.cpp \
 	Tools/DumpRenderTree/TestNetscapePlugIn/Tests/EvaluateJSAfterRemovingPluginElement.cpp \
 	Tools/DumpRenderTree/TestNetscapePlugIn/Tests/FormValue.cpp \
@@ -22833,10 +22834,9 @@
 @ENABLE_PLUGIN_PROCESS_TRUE@	$(webkit2_plugin_process_sources)
 
 @ENABLE_PLUGIN_PROCESS_TRUE@Programs_WebKitPluginProcess_LDADD = \
-@ENABLE_PLUGIN_PROCESS_TRUE@	-lpthread \
-@ENABLE_PLUGIN_PROCESS_TRUE@	libjavascriptcoregtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la \
-@ENABLE_PLUGIN_PROCESS_TRUE@	libWebCore.la \
+@ENABLE_PLUGIN_PROCESS_TRUE@	libWebCoreModules.la libWebCore.la\
 @ENABLE_PLUGIN_PROCESS_TRUE@	libWebCoreGtk2.la \
+@ENABLE_PLUGIN_PROCESS_TRUE@	libjavascriptcoregtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la \
 @ENABLE_PLUGIN_PROCESS_TRUE@	$(CAIRO_LIBS) \
 @ENABLE_PLUGIN_PROCESS_TRUE@	$(COVERAGE_LDFLAGS) \
 @ENABLE_PLUGIN_PROCESS_TRUE@	$(GEOCLUE_LIBS) \
@@ -23145,7 +23145,8 @@
 	Tools/WebKitTestRunner/InjectedBundle/TextInputController.h
 
 Libraries_libTestRunnerInjectedBundle_la_LIBADD = \
-	libWebCoreInternals.la
+	libWebCoreInternals.la \
+	libjavascriptcoregtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la libwebkit2gtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la $(GTK_LIBS) -lfontconfig
 
 Libraries_libTestRunnerInjectedBundle_la_LDFLAGS = \
 	-rpath ${shell pwd}/$(top_builddir)/../unix/TestNetscapePlugin/.libs \
--- webkitgtk-1.10.0/Source/WebKit2/GNUmakefile.am.link	2012-08-23 21:21:56.000000000 +0800
+++ webkitgtk-1.10.0/Source/WebKit2/GNUmakefile.am	2012-09-29 19:48:12.000000000 +0800
@@ -472,10 +472,9 @@
 	$(webkit2_plugin_process_sources)
 
 Programs_WebKitPluginProcess_LDADD = \
-	-lpthread \
-	libjavascriptcoregtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la \
-	libWebCore.la \
+	libWebCoreModules.la libWebCore.la \
 	libWebCoreGtk2.la \
+	libjavascriptcoregtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la \
 	$(CAIRO_LIBS) \
 	$(COVERAGE_LDFLAGS) \
 	$(GEOCLUE_LIBS) \
--- webkitgtk-1.10.0/Tools/GNUmakefile.am.link	2012-09-24 15:43:18.000000000 +0800
+++ webkitgtk-1.10.0/Tools/GNUmakefile.am	2012-09-29 19:47:29.000000000 +0800
@@ -246,6 +246,8 @@
 	-avoid-version \
 	-module
 
+TestNetscapePlugin_libtestnetscapeplugin_la_LIBADD = -lX11
+
 CLEANFILES += TestNetscapePlugin/libtestnetscapeplugin.la
 endif
 
--- webkitgtk-1.10.0/Tools/TestWebKitAPI/GNUmakefile.am.link	2012-09-24 15:43:18.000000000 +0800
+++ webkitgtk-1.10.0/Tools/TestWebKitAPI/GNUmakefile.am	2012-09-29 19:47:29.000000000 +0800
@@ -196,6 +196,9 @@
 	-avoid-version \
 	-module
 
+Libraries_libTestWebKitAPIInjectedBundle_la_LIBADD = libWTF.la \
+	libwebkit2gtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la $(GTK_LIBS)
+
 Libraries_libTestWebKitAPIInjectedBundle_la_CPPFLAGS = \
 	$(Libraries_libTestWebKitAPIMain_la_CPPFLAGS) \
 	-I$(top_builddir)/DerivedSources/InjectedBundle \
--- webkitgtk-1.10.0/Tools/WebKitTestRunner/GNUmakefile.am.link	2012-09-24 15:43:18.000000000 +0800
+++ webkitgtk-1.10.0/Tools/WebKitTestRunner/GNUmakefile.am	2012-09-29 19:47:29.000000000 +0800
@@ -110,7 +110,8 @@
 	Tools/WebKitTestRunner/InjectedBundle/TextInputController.h
 
 Libraries_libTestRunnerInjectedBundle_la_LIBADD = \
-	libWebCoreInternals.la
+	libWebCoreInternals.la \
+	libjavascriptcoregtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la libwebkit2gtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la $(GTK_LIBS) -lfontconfig
 
 Libraries_libTestRunnerInjectedBundle_la_LDFLAGS = \
 	-rpath ${shell pwd}/$(top_builddir)/../unix/TestNetscapePlugin/.libs \
