diff -up webkit-1.6.1/Source/WebKit/gtk/webkit/webkitwebsettings.cpp.allowScriptsToCloseWindows webkit-1.6.1/Source/WebKit/gtk/webkit/webkitwebsettings.cpp
--- webkit-1.6.1/Source/WebKit/gtk/webkit/webkitwebsettings.cpp.allowScriptsToCloseWindows	2011-09-26 22:55:01.000000000 +0200
+++ webkit-1.6.1/Source/WebKit/gtk/webkit/webkitwebsettings.cpp	2011-09-28 01:10:09.719364984 +0200
@@ -103,6 +103,7 @@ enum {
     PROP_ENABLE_FRAME_FLATTENING,
     PROP_USER_AGENT,
     PROP_JAVASCRIPT_CAN_OPEN_WINDOWS_AUTOMATICALLY,
+    PROP_ALLOW_SCRIPTS_TO_CLOSE_WINDOWS,
     PROP_JAVASCRIPT_CAN_ACCESS_CLIPBOARD,
     PROP_ENABLE_OFFLINE_WEB_APPLICATION_CACHE,
     PROP_EDITING_BEHAVIOR,
@@ -641,6 +642,20 @@ static void webkit_web_settings_class_in
                                                          flags));
 
     /**
+     * WebKitWebSettings: allow-scripts-to-close-windows
+     *
+     * Whether JavaScript can close windows automatically without user intervention.
+     *
+     */
+    g_object_class_install_property(gobject_class,
+				    PROP_ALLOW_SCRIPTS_TO_CLOSE_WINDOWS,
+				    g_param_spec_boolean("allow-scripts-to-close-windows",
+							 _("JavaScript can close windows automatically"),
+							 _("Whether JavaScript can close windows automatically"),
+							 FALSE,
+							 flags));
+
+    /**
     * WebKitWebSettings:javascript-can-access-clipboard
     *
     * Whether JavaScript can access Clipboard.
@@ -1045,6 +1060,9 @@ static void webkit_web_settings_set_prop
     case PROP_JAVASCRIPT_CAN_OPEN_WINDOWS_AUTOMATICALLY:
         priv->javascriptCanOpenWindowsAutomatically = g_value_get_boolean(value);
         break;
+    case PROP_ALLOW_SCRIPTS_TO_CLOSE_WINDOWS:
+        priv->allowScriptsToCloseWindows = g_value_get_boolean(value);
+        break;
     case PROP_JAVASCRIPT_CAN_ACCESS_CLIPBOARD:
         priv->javascriptCanAccessClipboard = g_value_get_boolean(value);
         break;
@@ -1204,6 +1222,9 @@ static void webkit_web_settings_get_prop
     case PROP_JAVASCRIPT_CAN_OPEN_WINDOWS_AUTOMATICALLY:
         g_value_set_boolean(value, priv->javascriptCanOpenWindowsAutomatically);
         break;
+    case PROP_ALLOW_SCRIPTS_TO_CLOSE_WINDOWS:
+        g_value_set_boolean(value, priv->allowScriptsToCloseWindows);
+        break;
     case PROP_JAVASCRIPT_CAN_ACCESS_CLIPBOARD:
         g_value_set_boolean(value, priv->javascriptCanAccessClipboard);
         break;
diff -up webkit-1.6.1/Source/WebKit/gtk/webkit/webkitwebsettingsprivate.h.allowScriptsToCloseWindows webkit-1.6.1/Source/WebKit/gtk/webkit/webkitwebsettingsprivate.h
--- webkit-1.6.1/Source/WebKit/gtk/webkit/webkitwebsettingsprivate.h.allowScriptsToCloseWindows	2011-09-26 22:55:01.000000000 +0200
+++ webkit-1.6.1/Source/WebKit/gtk/webkit/webkitwebsettingsprivate.h	2011-09-28 01:10:01.751597802 +0200
@@ -62,6 +62,7 @@ struct _WebKitWebSettingsPrivate {
     gboolean enableFrameFlattening;
     CString userAgent;
     gboolean javascriptCanOpenWindowsAutomatically;
+    gboolean allowScriptsToCloseWindows;
     gboolean javascriptCanAccessClipboard;
     gboolean enableOfflineWebApplicationCache;
     WebKitEditingBehavior editingBehavior;
diff -up webkit-1.6.1/Source/WebKit/gtk/webkit/webkitwebview.cpp.allowScriptsToCloseWindows webkit-1.6.1/Source/WebKit/gtk/webkit/webkitwebview.cpp
--- webkit-1.6.1/Source/WebKit/gtk/webkit/webkitwebview.cpp.allowScriptsToCloseWindows	2011-09-26 22:55:01.000000000 +0200
+++ webkit-1.6.1/Source/WebKit/gtk/webkit/webkitwebview.cpp	2011-09-28 00:09:10.911574812 +0200
@@ -3220,6 +3220,7 @@ static void webkit_web_view_update_setti
     coreSettings->setSpatialNavigationEnabled(settingsPrivate->enableSpatialNavigation);
     coreSettings->setFrameFlatteningEnabled(settingsPrivate->enableFrameFlattening);
     coreSettings->setJavaScriptCanOpenWindowsAutomatically(settingsPrivate->javascriptCanOpenWindowsAutomatically);
+    coreSettings->setAllowScriptsToCloseWindows(settingsPrivate->allowScriptsToCloseWindows);
     coreSettings->setJavaScriptCanAccessClipboard(settingsPrivate->javascriptCanAccessClipboard);
     coreSettings->setOfflineWebApplicationCacheEnabled(settingsPrivate->enableOfflineWebApplicationCache);
     coreSettings->setEditingBehaviorType(static_cast<WebCore::EditingBehaviorType>(settingsPrivate->editingBehavior));
@@ -3334,6 +3335,8 @@ static void webkit_web_view_settings_not
         settings->setFrameFlatteningEnabled(g_value_get_boolean(&value));
     else if (name == g_intern_string("javascript-can-open-windows-automatically"))
         settings->setJavaScriptCanOpenWindowsAutomatically(g_value_get_boolean(&value));
+    else if (name == g_intern_string("allow-scripts-to-close-windows"))
+	settings->setAllowScriptsToCloseWindows(g_value_get_boolean(&value));
     else if (name == g_intern_string("javascript-can-access-clipboard"))
         settings->setJavaScriptCanAccessClipboard(g_value_get_boolean(&value));
     else if (name == g_intern_string("enable-offline-web-application-cache"))
