--- webkit-1.7.90/Tools/gtk/common.py.docdir	2012-02-22 21:12:52.000000000 +0100
+++ webkit-1.7.90/Tools/gtk/common.py	2012-03-01 16:23:33.000000000 +0100
@@ -73,6 +73,10 @@
     if is_valid_build_directory(build_dir):
         return build_dir
 
+    build_dir = os.getcwd()
+    if is_valid_build_directory(build_dir):
+        return build_dir
+
     print 'Could not determine build directory.'
     sys.exit(1)
 
