--- setup.py_orig	2011-04-02 20:28:44.435995002 -0700
+++ setup.py	2011-04-02 20:29:51.052047049 -0700
@@ -156,7 +156,7 @@
 #               ("_PYMOL_VMD_PLUGINS",None)
 		("NO_MMLIBS",None),
                 ]
-    ext_comp_args=["-ffast-math","-funroll-loops","-O3"]
+    ext_comp_args=[]
     ext_link_args=[]
   
 setup ( # Distribution meta-data
@@ -181,6 +181,7 @@
                 'pymol/opengl/glu',
                 'pymol/opengl/glut',
                 'pymol/wizard',
+                'pymol2',
 		'web',
 		'web/examples',
 		'web/javascript', ],
