diff -Nuar e-module-weather-0.0.1_58042.orig//module.desktop.in e-module-weather-0.0.1_58042/module.desktop.in
--- e-module-weather-0.0.1_58042.orig//module.desktop.in	2011-02-16 22:39:52.000000000 +0200
+++ e-module-weather-0.0.1_58042/module.desktop.in	2011-03-31 16:11:18.919043556 +0300
@@ -9,6 +9,7 @@
 Name[fr]=Météo
 Name[el]=Καιρός
 Name[pt]=Meteorologia
+Name[tr]=Hava Durumu
 Icon=e-module-weather
 X-Enlightenment-ModuleType=utils
 Comment=A weather gadget.
@@ -21,3 +22,4 @@
 Comment[fr]=Affiche la météo actuelle.
 Comment[el]=Ένα συστατικό που δείχνει τις καιρικές συνθήκες.
 Comment[pt]=Um "gadget" meteorológico
+Comment[tr]=Hava durumu modülüdür.
