diff -up webkitgtk-1.11.90/Source/JavaScriptCore/llint/LowLevelInterpreter.cpp.double2ints webkitgtk-1.11.90/Source/JavaScriptCore/llint/LowLevelInterpreter.cpp
--- webkitgtk-1.11.90/Source/JavaScriptCore/llint/LowLevelInterpreter.cpp.double2ints	2013-02-12 17:22:38.000000000 +0100
+++ webkitgtk-1.11.90/Source/JavaScriptCore/llint/LowLevelInterpreter.cpp	2013-03-07 18:04:39.778807991 +0100
@@ -117,6 +118,17 @@ static double Ints2Double(uint32_t lo, u
     u.ival64 = (static_cast<uint64_t>(hi) << 32) | lo;
     return u.dval;
 }
+
+static void Double2Ints(double input, intptr_t& lo, intptr_t& hi)
+{
+    union {
+        double dval;
+        int64_t ival64;
+    } u;
+    u.dval = input;
+    hi = static_cast<intptr_t>(u.ival64 >> 32);
+    lo = static_cast<intptr_t>(u.ival64);
+}
 #endif // USE(JSVALUE32_64)
 
 } // namespace LLint
