diff -up webkitgtk-1.11.2/Source/WTF/wtf/Platform.h.orig webkitgtk-1.11.2/Source/WTF/wtf/Platform.h
--- webkitgtk-1.11.2/Source/WTF/wtf/Platform.h.orig	2012-12-16 21:27:29.000000000 +0100
+++ webkitgtk-1.11.2/Source/WTF/wtf/Platform.h	2012-12-16 23:16:19.000000000 +0100
@@ -1001,7 +1001,7 @@
 #define ENABLE_REGEXP_TRACING 0
 
 /* Yet Another Regex Runtime - turned on by default for JIT enabled ports. */
-#if !defined(ENABLE_YARR_JIT) && (ENABLE(JIT) || ENABLE(LLINT_C_LOOP)) && !PLATFORM(CHROMIUM) && !(OS(QNX) && PLATFORM(QT))
+#if !defined(ENABLE_YARR_JIT) && ENABLE(JIT) && !PLATFORM(CHROMIUM) && !(OS(QNX) && PLATFORM(QT))
 #define ENABLE_YARR_JIT 1
 
 /* Setting this flag compares JIT results with interpreter results. */
