diff -up webkitgtk-1.11.2/Source/JavaScriptCore/runtime/ArrayConstructor.h.includes webkitgtk-1.11.2/Source/JavaScriptCore/runtime/ArrayConstructor.h
--- webkitgtk-1.11.2/Source/JavaScriptCore/runtime/ArrayConstructor.h.includes	2012-12-16 23:12:06.000000000 +0100
+++ webkitgtk-1.11.2/Source/JavaScriptCore/runtime/ArrayConstructor.h	2012-12-16 23:12:21.000000000 +0100
@@ -21,6 +21,7 @@
 #ifndef ArrayConstructor_h
 #define ArrayConstructor_h
 
+#include "ArrayAllocationProfile.h"
 #include "InternalFunction.h"
 
 namespace JSC {
