--- webkit-1.1.17/Source/WebCore/plugins/PluginDatabase.cpp.old	2009-12-01 15:22:10.000000000 +0000
+++ webkit-1.1.17/Source/WebCore/plugins/PluginDatabase.cpp	2009-12-01 15:22:21.000000000 +0000
@@ -357,6 +357,9 @@ Vector<String> PluginDatabase::defaultPl
     paths.append("/usr/lib64/mozilla/plugins");
     paths.append("/usr/lib/nsbrowser/plugins");
     paths.append("/usr/lib64/nsbrowser/plugins");
+    /* And for nspluginwrapper */
+    paths.append("/usr/lib64/mozilla/plugins-wrapped");
+    paths.append("/usr/lib/mozilla/plugins-wrapped");
 
     String mozHome(getenv("MOZILLA_HOME"));
     mozHome.append("/plugins");
