Index: rtkit-0.6/m4/acx_pthread.m4
===================================================================
--- rtkit-0.6.orig/m4/acx_pthread.m4
+++ rtkit-0.6/m4/acx_pthread.m4
@@ -114,7 +114,7 @@ fi
 # which indicates that we try without any flags at all, and "pthread-config"
 # which is a program returning the flags for the Pth emulation library.
 
-acx_pthread_flags="pthreads none -Kthread -kthread lthread -pthread -pthreads -mthreads pthread --thread-safe -mt pthread-config"
+acx_pthread_flags=""
 
 # The ordering *is* (sometimes) important.  Some notes on the
 # individual items follow:
