diff -Nur nogravity-old/src/Linux/configure.in nogravity/src/Linux/configure.in
--- nogravity-old/src/Linux/configure.in	2006-09-03 00:59:38.978611944 +0300
+++ nogravity/src/Linux/configure.in	2006-09-03 01:04:50.271288264 +0300
@@ -93,17 +93,6 @@
   AC_CHECK_LIB([SDL_mixer], [Mix_OpenAudio], , AC_MSG_ERROR([Can't find SDL_mixer library for sound.]))
 fi
 
-# Checks for libpng.
-AC_CHECK_PROG(HAVE_LIBPNG_CONFIG, [libpng-config], yes)
-if test "$HAVE_LIBPNG_CONFIG" = "yes"
-then
-  CFLAGS="`libpng-config --cflags` $CFLAGS"
-  LDFLAGS="`libpng-config --ldflags` $LDFLAGS"
-else
-  AC_MSG_WARN([Can't find libpng-config, guessing required headers and libraries.])
-fi
-AC_CHECK_HEADERS([png.h], , AC_MSG_ERROR([Can't find libpng header.]))
-AC_CHECK_LIB([png], [png_decompress_chunk], , AC_MSG_ERROR([Can't find libpng library.]))
 
 # Checks for Ogg.
 AC_CHECK_LIB([ogg], [ogg_stream_init], , AC_MSG_ERROR([Can't find Ogg library.]))
