diff -Nur nogravity-old/src/Linux/configure.in nogravity/src/Linux/configure.in
--- nogravity-old/src/Linux/configure.in	2009-07-29 22:14:19.652157926 +0300
+++ nogravity/src/Linux/configure.in	2009-07-29 22:16:54.623909708 +0300
@@ -101,5 +101,11 @@
 AC_CHECK_HEADERS([vorbis/codec.h], , AC_MSG_ERROR([Can't find Vorbis header.]))
 AC_CHECK_LIB([vorbis], [vorbis_synthesis_init], , AC_MSG_ERROR([Can't find Vorbis library.]))
 
+# Checks for zlib
+AC_CHECK_LIB([z], [main], , AC_MSG_ERROR([Can't find libz.]))
+
+# Check for libpng
+AC_CHECK_LIB([png], [main], , AC_MSG_ERROR([Can't find libpng.]))
+
 # Output it all.
 AC_OUTPUT
