diff -BrauN nx-X11.orig/config/cf/X11.tmpl nx-X11/config/cf/X11.tmpl
--- nx-X11.orig/config/cf/X11.tmpl	2005-12-14 23:41:52.000000000 +0000
+++ nx-X11/config/cf/X11.tmpl	2008-01-26 21:00:56.000000000 +0000
@@ -512,7 +512,7 @@
 #define XprtServer		(BuildServer && BuildXprint)
 #endif
 #ifndef BuildXprintLib
-#define BuildXprintLib		(!BuildServersOnly && BuildXprint)
+#define BuildXprintLib		YES
 #endif
 #ifndef BuildXprintClients
 #define BuildXprintClients	BuildXprintLib
diff -BrauN nx-X11.orig/lib/X11/Imakefile nx-X11/lib/X11/Imakefile
--- nx-X11.orig/lib/X11/Imakefile	2007-06-04 08:19:12.000000000 +0000
+++ nx-X11/lib/X11/Imakefile	2008-01-26 18:56:23.000000000 +0000
@@ -83,7 +83,7 @@
 #define DoExtraLib SharedLibX11
 #define DoDebugLib DebugLibX11
 #define DoProfileLib ProfileLibX11
-#define LibName X11
+#define LibName X11-nx
 #define SoRev SOXLIBREV
 #define HugeLibrary YES
 #define IncSubdir X11
diff -BrauN nx-X11.orig/lib/X11/xlibi18n/im/ximcp/Imakefile nx-X11/lib/X11/xlibi18n/im/ximcp/Imakefile
--- nx-X11.orig/lib/X11/xlibi18n/im/ximcp/Imakefile	2004-04-23 18:43:35.000000000 +0000
+++ nx-X11/lib/X11/xlibi18n/im/ximcp/Imakefile	2008-01-26 19:00:36.000000000 +0000
@@ -15,7 +15,7 @@
 
    CONN_DEFINES = ConnectionFlags
    XTRANS_XIM_DEFINES = -DXIM_t -DTRANS_CLIENT
-   REQUIREDLIBS = SharedXlibi18nReqs
+   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11-nx -lc
 
 LinkSourceFile(imCallbk.c, ../../..)
 LinkSourceFile(imDefFlt.c, ../../..)
diff -BrauN nx-X11.orig/lib/X11/xlibi18n/lc/Utf8/Imakefile nx-X11/lib/X11/xlibi18n/lc/Utf8/Imakefile
--- nx-X11.orig/lib/X11/xlibi18n/lc/Utf8/Imakefile	2004-04-23 18:43:35.000000000 +0000
+++ nx-X11/lib/X11/xlibi18n/lc/Utf8/Imakefile	2008-01-26 19:02:10.000000000 +0000
@@ -6,7 +6,7 @@
       XI18NLIBNAME = xlibi18n
            SRCS = lcUTF8Load.c
            OBJS = ${SRCS:.c=.o}
-   REQUIREDLIBS = SharedXlibi18nReqs
+   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11-nx -lc
 
 LinkSourceFile(lcUTF8Load.c, ../../..)
 
diff -BrauN nx-X11.orig/lib/X11/xlibi18n/lc/def/Imakefile nx-X11/lib/X11/xlibi18n/lc/def/Imakefile
--- nx-X11.orig/lib/X11/xlibi18n/lc/def/Imakefile	2004-04-23 18:43:35.000000000 +0000
+++ nx-X11/lib/X11/xlibi18n/lc/def/Imakefile	2008-01-26 19:01:24.000000000 +0000
@@ -6,7 +6,7 @@
       XI18NLIBNAME = xlcDef
            SRCS = lcDefConv.c
            OBJS = ${SRCS:.c=.o}
-   REQUIREDLIBS = SharedXlibi18nReqs
+   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11-nx -lc
 
 LinkSourceFile(lcDefConv.c, ../../..)
 
diff -BrauN nx-X11.orig/lib/X11/xlibi18n/lc/gen/Imakefile nx-X11/lib/X11/xlibi18n/lc/gen/Imakefile
--- nx-X11.orig/lib/X11/xlibi18n/lc/gen/Imakefile	2004-04-23 18:43:36.000000000 +0000
+++ nx-X11/lib/X11/xlibi18n/lc/gen/Imakefile	2008-01-26 19:01:35.000000000 +0000
@@ -6,7 +6,7 @@
       XI18NLIBNAME = xlibi18n
            SRCS = lcGenConv.c
            OBJS = ${SRCS:.c=.o}
-   REQUIREDLIBS = SharedXlibi18nReqs
+   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11-nx -lc
 
 LinkSourceFile(lcGenConv.c, ../../..)
 
diff -BrauN nx-X11.orig/lib/X11/xlibi18n/lc/xlocale/Imakefile nx-X11/lib/X11/xlibi18n/lc/xlocale/Imakefile
--- nx-X11.orig/lib/X11/xlibi18n/lc/xlocale/Imakefile	2004-04-23 18:43:36.000000000 +0000
+++ nx-X11/lib/X11/xlibi18n/lc/xlocale/Imakefile	2008-01-26 19:01:03.000000000 +0000
@@ -6,7 +6,7 @@
       XI18NLIBNAME = xlocale
            SRCS = lcEuc.c lcSjis.c lcJis.c
            OBJS = ${SRCS:.c=.o}
-   REQUIREDLIBS = SharedXlibi18nReqs
+   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11-nx -lc
 
 LinkSourceFile(lcEuc.c, ../../..)
 LinkSourceFile(lcSjis.c, ../../..)
diff -BrauN nx-X11.orig/lib/X11/xlibi18n/om/generic/Imakefile nx-X11/lib/X11/xlibi18n/om/generic/Imakefile
--- nx-X11.orig/lib/X11/xlibi18n/om/generic/Imakefile	2004-04-23 18:43:36.000000000 +0000
+++ nx-X11/lib/X11/xlibi18n/om/generic/Imakefile	2008-01-26 18:59:45.000000000 +0000
@@ -7,7 +7,7 @@
 		  omTextEsc.c omTextExt.c omTextPer.c omXChar.c
 
            OBJS = ${SRCS:.c=.o}
-   REQUIREDLIBS = SharedXlibi18nReqs
+   REQUIREDLIBS = -L$(XENVLIBDIR) -lX11-nx -lc
 
 LinkSourceFile(omDefault.c, ../../..)
 LinkSourceFile(omGeneric.c, ../../..)
diff -BrauN nx-X11.orig/lib/Xaw/Imakefile nx-X11/lib/Xaw/Imakefile
--- nx-X11.orig/lib/Xaw/Imakefile	2004-09-02 01:10:29.000000000 +0000
+++ nx-X11/lib/Xaw/Imakefile	2008-01-26 18:51:53.000000000 +0000
@@ -198,6 +198,9 @@
 	XawI18n.o \
 	XawInit.o
 
+#undef DefaultCCOptions
+#define DefaultCCOptions GccWarningOptions
+
 #include <Library.tmpl>
 
 #if DoSharedLib && SharedDataSeparation
diff -BrauN nx-X11.orig/lib/Xaw6/Imakefile nx-X11/lib/Xaw6/Imakefile
--- nx-X11.orig/lib/Xaw6/Imakefile	2004-04-23 18:43:40.000000000 +0000
+++ nx-X11/lib/Xaw6/Imakefile	2008-01-26 18:52:31.000000000 +0000
@@ -115,6 +115,8 @@
 
 INCLUDES = -I$(AWIDGETSRC)
 
+#undef DefaultCCOptions
+#define DefaultCCOptions GccWarningOptions
 #include <Library.tmpl>
 
 #if DoSharedLib && SharedDataSeparation
diff -BrauN nx-X11.orig/lib/Xext/Imakefile nx-X11/lib/Xext/Imakefile
--- nx-X11.orig/lib/Xext/Imakefile	2004-08-16 15:02:59.000000000 +0000
+++ nx-X11/lib/Xext/Imakefile	2008-01-26 18:53:33.000000000 +0000
@@ -10,14 +10,14 @@
 #define DoExtraLib SharedLibXext
 #define DoDebugLib DebugLibXext
 #define DoProfileLib ProfileLibXext
-#define LibName Xext
+#define LibName Xext-nx
 #define SoRev SOXEXTREV
 #define LibHeaders NO
 
 #include <Threads.tmpl>
 
 #ifdef SharedXextReqs
-REQUIREDLIBS = SharedXextReqs
+REQUIREDLIBS = -L../../exports/lib/ -lX11-nx SharedXextReqs
 #endif
 
 #if HasShm
diff -BrauN nx-X11.orig/lib/Xrender/Imakefile nx-X11/lib/Xrender/Imakefile
--- nx-X11.orig/lib/Xrender/Imakefile	2004-08-07 00:40:40.000000000 +0000
+++ nx-X11/lib/Xrender/Imakefile	2008-01-26 18:55:39.000000000 +0000
@@ -74,7 +74,7 @@
 #define DoSharedLib SharedLibXrender
 #define DoDebugLib DebugLibXrender
 #define DoProfileLib ProfileLibXrender
-#define LibName Xrender
+#define LibName Xrender-nx
 #define SoRev SOXRENDERREV
 
 #ifdef XBuildIncDir
@@ -89,7 +89,7 @@
 #include <Threads.tmpl>
 
 #ifdef SharedXrenderReqs
-REQUIREDLIBS = SharedXrenderReqs
+REQUIREDLIBS = $(LDPRELIB) -L../../exports/lib/ -lXext-nx -lX11-nx
 #endif
 
 #if Malloc0ReturnsNull
diff -BrauN nx-X11.orig/lib/Xxf86dga/Imakefile nx-X11/lib/Xxf86dga/Imakefile
--- nx-X11.orig/lib/Xxf86dga/Imakefile	2005-10-18 14:58:52.000000000 +0000
+++ nx-X11/lib/Xxf86dga/Imakefile	2008-01-26 19:02:50.000000000 +0000
@@ -34,6 +34,9 @@
 
 #define IncludeSharedObjectInNormalLib
 
+#undef DefaultCCOptions
+#define DefaultCCOptions GccWarningOptions
+
 #include <Library.tmpl>
 
 InstallGenManPage(XDGA,$(LIBMANDIR),$(LIBMANSUFFIX))
diff -BrauN nx-X11.orig/programs/Xserver/Imakefile nx-X11/programs/Xserver/Imakefile
--- nx-X11.orig/programs/Xserver/Imakefile	2007-09-03 17:20:32.000000000 +0000
+++ nx-X11/programs/Xserver/Imakefile	2008-01-26 19:06:48.000000000 +0000
@@ -997,7 +997,7 @@
 XPMLIB =  -lXpm
 NXAGENT = hw/nxagent/LibraryTargetName(nxagent)
 NXAGENTLIBS = PreFbLibs $(NXAGENT) FbPostFbLibs $(NXAGENT) $(MI)
-NXAGENTSYSLIBS = $(FONTLIBS) $(LDPRELIBS) $(XLIB) $(SYSLIBS) $(XPMLIB)
+NXAGENTSYSLIBS = $(FONTLIBS) $(LDPRELIBS) -lX11-nx -lXext-nx -lXrender-nx $(SYSLIBS) $(XPMLIB)
 #if HasParallelMake
 MakeMutex($(NXAGENTDIRS) $(NXAGENTOBJS) $(NXAGENTLIBS) $(NXAGENTSYSLIBS))
 #endif
