Index: work/common/usr/share/doc/fglrx/examples/etc/acpi/ati-powermode.sh
===================================================================
--- work.orig/common/usr/share/doc/fglrx/examples/etc/acpi/ati-powermode.sh
+++ work/common/usr/share/doc/fglrx/examples/etc/acpi/ati-powermode.sh
@@ -4,10 +4,12 @@
 # Control script for ACPI lid state and AC adapter state
 #
 
+test -d /sys/module/fglrx || exit 0
+
 getXuser() {
-        user=`finger| grep -m1 ":$displaynum " | awk '{print $1}'`
+        user=`who | grep -m1 ":$displaynum " | awk '{print $1}'`
         if [ x"$user" = x"" ]; then
-                user=`finger| grep -m1 ":$displaynum" | awk '{print $1}'`
+                user=`who | grep -m1 ":$displaynum" | awk '{print $1}'`
         fi
         if [ x"$user" != x"" ]; then
                 userhome=`getent passwd $user | cut -d: -f6`
