
Already defined in mesa

Signed-off-by: Thomas Backlund <tmb@mageia.org>

diff -Nurp fglrx-13.250.18.orig/common/usr/include/GL/glATI.h fglrx-13.250.18/common/usr/include/GL/glATI.h
--- fglrx-13.250.18.orig/common/usr/include/GL/glATI.h	2013-12-06 13:46:50.339162515 +0200
+++ fglrx-13.250.18/common/usr/include/GL/glATI.h	2013-12-06 13:51:18.621140440 +0200
@@ -5367,7 +5367,9 @@ typedef GLvoid (APIENTRY * PFNGLTEXRENDE
 #define GL_SAMPLE_MASK_VALUE                               0x8E52
 #define GL_MAX_SAMPLE_MASK_WORDS                           0x8E59
 #define GL_TEXTURE_2D_MULTISAMPLE                          0x9100
+#ifndef GL_PROXY_TEXTURE_2D_MULTISAMPLE
 #define GL_PROXY_TEXTURE_2D_MULTISAMPLE                    0X9101
+#endif
 #define GL_TEXTURE_2D_MULTISAMPLE_ARRAY                    0x9102
 #define GL_PROXY_TEXTURE_2D_MULTISAMPLE_ARRAY              0x9103
 #define GL_MAX_COLOR_TEXTURE_SAMPLES                       0x910E
@@ -5385,11 +5387,13 @@ typedef GLvoid (APIENTRY * PFNGLTEXRENDE
 #define GL_UNSIGNED_INT_SAMPLER_2D_MULTISAMPLE_ARRAY       0x910D
 
 typedef GLvoid (APIENTRY * PFNGLSAMPLEMASKIPROC) (GLuint index, GLbitfield value);
+#if 0
 typedef GLvoid (APIENTRY * PFNGLTEXIMAGE2DMULTISAMPLEPROC) (GLenum target, GLsizei samples, GLint internalformat, 
                                                             GLsizei width, GLsizei height, GLboolean fixedsamplelocations);
+
 typedef GLvoid (APIENTRY * PFNGLTEXIMAGE3DMULTISAMPLEPROC) (GLenum target, GLsizei samples, GLint internalformat, 
                                                             GLsizei width, GLsizei height, GLsizei depth, GLboolean fixedsamplelocations); 
-
+#endif
 // GL_EXT_timer_query
 #define GL_TIME_ELAPSED_EXT               0x88BF
 
@@ -6006,6 +6010,7 @@ typedef GLvoid (APIENTRY * PFNGLDEBUGMES
 typedef GLvoid (APIENTRY * PFNGLDEBUGMESSAGECALLBACKARB)(GLvoid* callback, GLvoid* userParam);
 typedef GLuint (APIENTRY * PFNGLGETDEBUGMESSAGELOGARB)(GLuint count, GLsizei bufsize, GLenum* sources, GLenum* types, GLuint* ids, GLenum* severities, GLsizei* lengths, GLchar* messageLog);
 
+#if 0
 typedef void (APIENTRY *GLDEBUGPROCARB)(GLenum source,
                                         GLenum type,
                                         GLuint id,
@@ -6014,6 +6019,7 @@ typedef void (APIENTRY *GLDEBUGPROCARB)(
                                         const GLchar* message,
                                         GLvoid* userParam);
 // end ARB_debug_output
+#endif
 
 
 /* ARB_debug_group (KHR_debug) */
