diff -Nur abicheck-1.2-orig/abicheck.pl abicheck-1.2/abicheck.pl
--- abicheck-1.2-orig/abicheck.pl	2003-08-26 21:57:51.000000000 +0200
+++ abicheck-1.2/abicheck.pl	2005-04-11 12:59:04.000000000 +0200
@@ -1584,7 +1584,7 @@
 		$ld_debug_match = q/^\s*\d+:/;
 		$lddstub_match = '';
 		$binding_match =
-		    q/binding file (.*) to (.*): .*symbol `(\w+)'(.*)$/;
+		    q/binding file (.*) \[.*\] to (.*) \[.*\]: .*symbol `(\w+)'(.*)$/;
 		$filter_match  =
 		    q/file=(.*);  filtered by (.*)$/; # XXX not checked
 		$unbound_match =
