diff -Nur abicheck-1.2-orig/test/Makefile abicheck-1.2/test/Makefile
--- abicheck-1.2-orig/test/Makefile	2003-08-26 21:57:51.000000000 +0200
+++ abicheck-1.2/test/Makefile	2004-12-13 13:01:19.827315472 +0100
@@ -9,7 +9,7 @@
 # /usr/bin/ld: BFD 2.11.93.0.2 20020207 assertion fail elf-strtab.c:262
 # but it still creates the binary properly.
 libc_a: libc_a.c
-	cc -o libc_a libc_a.c -lm /usr/lib/libc.a
+	cc -o libc_a libc_a.c -static -lm /usr/lib/libc.a
 
 private1: private1.c
 	cc -D`uname` -o private1 private1.c
diff -Nur abicheck-1.2-orig/test/libc_a.c abicheck-1.2/test/libc_a.c
--- abicheck-1.2-orig/test/libc_a.c	2003-08-26 21:57:51.000000000 +0200
+++ abicheck-1.2/test/libc_a.c	2007-05-24 11:11:20.000000000 +0200
@@ -5,6 +5,7 @@
 
 /* This is used to catch libc.a linking. See Makefile & run_tests */
 
+#include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
 
diff -Nur abicheck-1.2-orig/test/private1.c abicheck-1.2/test/private1.c
--- abicheck-1.2-orig/test/private1.c	2002-06-18 20:16:07.000000000 +0200
+++ abicheck-1.2/test/private1.c	2007-05-24 11:40:25.000000000 +0200
@@ -6,6 +6,7 @@
 /* This is for simple catching of some private symbol usage. See run_tests */
 
 #include <stdio.h>
+#include <string.h>
 
 #if	defined(SunOS)
 #define OS "Solaris"
diff -Nur abicheck-1.2-orig/test/run_tests abicheck-1.2/test/run_tests
--- abicheck-1.2-orig/test/run_tests	2002-02-07 15:40:20.000000000 +0100
+++ abicheck-1.2/test/run_tests	2007-05-24 11:40:52.000000000 +0200
@@ -19,7 +19,7 @@
 # libc_a:
 
 name="libc.a static link test on libc_a"
-if abicheck ./libc_a 2>/dev/null | grep 'STATIC_LINK:.*libc' > /dev/null; then
+if abicheck ./libc_a 2>/dev/null | grep 'STATIC_LINK:.*\(libc\|completely statically linked\)' > /dev/null; then
 	pass
 else
 	fail
diff -Nur abicheck-1.2-orig/test/run_tests abicheck-1.2/test/run_tests
--- abicheck-1.2-orig/test/run_tests	2002-02-07 15:40:20.000000000 +0100
+++ abicheck-1.2/test/run_tests	2007-05-24 12:33:02.000000000 +0200
@@ -39,7 +39,7 @@
 # private1:
 
 name="private1: calls private sym in libc"
-if abicheck ./private1 2>/dev/null | egrep 'PRIVATE:.*libc.*(__open|__nanosleep)' > /dev/null; then
+if abicheck ./private1 2>/dev/null | grep 'PRIVATE:.*libc.*\(__open\|__nanosleep\)' > /dev/null; then
 	pass
 else
 	fail
diff -Nur abicheck-1.2-orig/test/run_tests abicheck-1.2/test/run_tests
--- abicheck-1.2-orig/test/run_tests	2002-02-07 15:40:20.000000000 +0100
+++ abicheck-1.2/test/run_tests	2007-05-24 13:00:58.000000000 +0200
@@ -29,7 +29,7 @@
 # public1:
 
 name="public1: only calls public syms"
-if [ "`abicheck ./public1 2>/dev/null`" = "./public1: OK" ]; then
+if [ "`abicheck ./public1 2>/dev/null`" == "./public1: OK" ]; then
 	pass
 else
 	fail
