diff -Nur abicheck-1.2-orig/abicheck.pl abicheck-1.2/abicheck.pl
--- abicheck-1.2-orig/abicheck.pl	2003-08-26 21:57:51.000000000 +0200
+++ abicheck-1.2/abicheck.pl	2005-08-15 21:10:15.000000000 +0200
@@ -3498,6 +3498,19 @@
 			libc.so.6:__ctype_b_loc
 			libc.so.6:__ctype_tolower_loc
 			libc.so.6:__ctype_toupper_loc
+
+			libgcc_s.so.1:__divdi3
+			libgcc_s.so.1:__register_frame
+			libgcc_s.so.1:__frame_state_for
+			libgcc_s.so.1:__umoddi3
+			libgcc_s.so.1:__udivdi3
+			libgcc_s.so.1:__register_frame_table
+			libgcc_s.so.1:__deregister_frame_info
+			libgcc_s.so.1:__deregister_frame
+			libgcc_s.so.1:__cxa_finalize
+			libgcc_s.so.1:__moddi3
+			libgcc_s.so.1:__register_frame_info_table
+			libgcc_s.so.1:__register_frame_info
 		);
 
 		@{$private} = qw(
