diff -Nur abicheck-1.2-orig/abicheck.pl abicheck-1.2/abicheck.pl
--- abicheck-1.2-orig/abicheck.pl	2003-08-26 21:57:51.000000000 +0200
+++ abicheck-1.2/abicheck.pl	2004-12-13 13:00:40.321321296 +0100
@@ -1592,7 +1592,7 @@
 		$dtneeded_match =
 		    q/^\s*(\S+)\s+=>\s+(not found|\S+)/;
 		$file_match =
-		    q/^(\s*\d+:).*needed by (.*)$/;
+		    q/^(\s*\d+:).*needed by (.*) (\[\d+\])*$/;
 		#
 		# n.b. there is a PID mismatch in the GNU linker output if
 		# we watch for the more sensible:
