diff -Nur abicheck-1.2-old/abicheck.pl abicheck-1.2/abicheck.pl
--- abicheck-1.2-old/abicheck.pl	2008-10-06 03:44:53.000000000 +0300
+++ abicheck-1.2/abicheck.pl	2008-10-06 03:55:33.000000000 +0300
@@ -3566,5 +3566,82 @@
 			libnsl.so.1:readColdStartFile
 			libnsl.so.1:writeColdStartFile
 		);
+
+		#
+		# These are _FORTIFY_SOURCE symbols.
+		#
+		push(@{$public}, qw(
+			libc.so.6:__chk_fail
+			libc.so.6:__confstr_chk
+			libc.so.6:__fgets_chk
+			libc.so.6:__fgets_unlocked_chk
+			libc.so.6:__fgetws_chk
+			libc.so.6:__fgetws_unlocked_chk
+			libc.so.6:__fprintf_chk
+			libc.so.6:__fread_chk
+			libc.so.6:__fread_unlocked_chk
+			libc.so.6:__fwprintf_chk
+			libc.so.6:__getcwd_chk
+			libc.so.6:__getdomainname_chk
+			libc.so.6:__getgroups_chk
+			libc.so.6:__gethostname_chk
+			libc.so.6:__getlogin_r_chk
+			libc.so.6:__gets_chk
+			libc.so.6:__getwd_chk
+			libc.so.6:__mbsnrtowcs_chk
+			libc.so.6:__mbsrtowcs_chk
+			libc.so.6:__mbstowcs_chk
+			libc.so.6:__memcpy_chk
+			libc.so.6:__memmove_chk
+			libc.so.6:__mempcpy_chk
+			libc.so.6:__memset_chk
+			libc.so.6:__pread64_chk
+			libc.so.6:__pread_chk
+			libc.so.6:__printf_chk
+			libc.so.6:__ptsname_r_chk
+			libc.so.6:__read_chk
+			libc.so.6:__readlinkat_chk
+			libc.so.6:__readlink_chk
+			libc.so.6:__realpath_chk
+			libc.so.6:__recv_chk
+			libc.so.6:__recvfrom_chk
+			libc.so.6:__snprintf_chk
+			libc.so.6:__sprintf_chk
+			libc.so.6:__stack_chk_fail
+			libc.so.6:__stpcpy_chk
+			libc.so.6:__stpncpy_chk
+			libc.so.6:__strcat_chk
+			libc.so.6:__strcpy_chk
+			libc.so.6:__strncat_chk
+			libc.so.6:__strncpy_chk
+			libc.so.6:__swprintf_chk
+			libc.so.6:__syslog_chk
+			libc.so.6:__ttyname_r_chk
+			libc.so.6:__vfprintf_chk
+			libc.so.6:__vfwprintf_chk
+			libc.so.6:__vprintf_chk
+			libc.so.6:__vsnprintf_chk
+			libc.so.6:__vsprintf_chk
+			libc.so.6:__vswprintf_chk
+			libc.so.6:__vsyslog_chk
+			libc.so.6:__vwprintf_chk
+			libc.so.6:__wcpcpy_chk
+			libc.so.6:__wcpncpy_chk
+			libc.so.6:__wcrtomb_chk
+			libc.so.6:__wcscat_chk
+			libc.so.6:__wcscpy_chk
+			libc.so.6:__wcsncat_chk
+			libc.so.6:__wcsncpy_chk
+			libc.so.6:__wcsnrtombs_chk
+			libc.so.6:__wcsrtombs_chk
+			libc.so.6:__wcstombs_chk
+			libc.so.6:__wctomb_chk
+			libc.so.6:__wmemcpy_chk
+			libc.so.6:__wmemmove_chk
+			libc.so.6:__wmempcpy_chk
+			libc.so.6:__wmemset_chk
+			libc.so.6:__wprintf_chk
+			)
+		);
 	}
 }
