--- dietlibc-0.30/include/sys/mount.h.relatime	2004-05-12 19:55:24.000000000 +0400
+++ dietlibc-0.30/include/sys/mount.h	2007-08-11 20:25:41.000000000 +0400
@@ -41,6 +41,8 @@ enum
 #define MS_NODIRATIME	MS_NODIRATIME
   MS_BIND = 4096,		/* Bind directory at different place.  */
 #define MS_BIND		MS_BIND
+  MS_RELATIME = (1<<21),	/* Update atime relative to mtime/ctime.  */
+#define MS_RELATIME	MS_RELATIME
 };
 
 /* Flags that can be altered by MS_REMOUNT  */
