--- Makefile	2008-06-08 12:43:14.000000000 -0400
+++ Makefile.oden	2008-06-08 12:47:40.000000000 -0400
@@ -104,10 +104,15 @@ libs: $(LIBS)
 ifeq (,$(ARCH32))
 all_32:
 else
+ifeq (,$(CROSS))
 all_32:
 	$(MAKE) MYARCH=$(ARCH32) CC="$(CC) -m32" libs
+else
+all_32:
+endif
 endif
 	
+
 profiling: $(OBJDIR)/libgmon.a $(OBJDIR)/pstart.o
 
 CFLAGS=-pipe -nostdinc
@@ -373,9 +378,9 @@ install-bin: $(OBJDIR)/diet-i
 	$(INSTALL) -d $(DESTDIR)$(MAN1DIR) $(DESTDIR)$(BINDIR)
 ifeq ($(MYARCH),$(ARCH))
 	$(INSTALL) $(OBJDIR)/diet-i $(DESTDIR)$(BINDIR)/diet
+	$(INSTALL) -m 644 diet.1 $(DESTDIR)$(MAN1DIR)/diet.1
 	-$(INSTALL) $(PICODIR)/diet-dyn-i $(DESTDIR)$(BINDIR)/diet-dyn
 endif
-	$(INSTALL) -m 644 diet.1 $(DESTDIR)$(MAN1DIR)/diet.1
 
 install-profiling:
 	-$(INSTALL) $(OBJDIR)/pstart.o $(OBJDIR)/libgmon.a $(OBJDIR)/dyn_start.o $(OBJDIR)/dyn_stop.o $(DESTDIR)$(ILIBDIR)
@@ -392,13 +397,19 @@ install-pic:
 	if test -f $(PICODIR)/libc.so -a ! -f $(DESTDIR)/etc/diet.ld.conf; then echo "$(ILIBDIR)" > $(DESTDIR)/etc/diet.ld.conf; fi
 
 install-headers:
+ifeq ($(MYARCH),$(ARCH))
 	for i in `find include -name \*.h`; do install -m 644 -D $$i $(DESTDIR)$(INSTALLHOME)/$$i; done
+endif
 
 ifeq (,$(ARCH32))
 install-32:
 else
+ifeq (,$(CROSS))
 install-32:
 	$(MAKE) MYARCH=$(ARCH32) CC="$(CC) -m32" install-lib
+else
+install-32:
+endif
 endif
 
 install: install-32 install-lib install-bin install-profiling install-pic install-headers
