diff -Naurp dietlibc-0.32-20090113/test/Makefile dietlibc-0.32-20090113.oden/test/Makefile
--- dietlibc-0.32-20090113/test/Makefile	2008-02-23 01:02:19.000000000 +0100
+++ dietlibc-0.32-20090113.oden/test/Makefile	2009-01-28 22:28:14.000000000 +0100
@@ -1,4 +1,4 @@
-all: direnttest inet stdiotests stdlibtests stringtests timetests \
+all: direnttest inettests stdiotests stdlibtests stringtests timetests \
 test ffs getdelim stdlib 
 
 DIET=diet
@@ -33,7 +33,7 @@ testandset: testandset.c
 direnttest:
 	 $(MAKE) -C dirent CC="$(CC)" CFLAGS="$(CFLAGS)" DIET=$(DIET) LCOMPAT=$(LCOMPAT)
 
-inettest:
+inettests:
 	$(MAKE) -C inet CC="$(CC)" CFLAGS="$(CFLAGS)" DIET=$(DIET) LCOMPAT=$(LCOMPAT)
 
 stdiotests:
