diff -Naurp dietlibc-0.32-20090113/include/rpc/types.h dietlibc-0.32-20090113.oden/include/rpc/types.h
--- dietlibc-0.32-20090113/include/rpc/types.h	2003-08-19 18:58:18.000000000 +0200
+++ dietlibc-0.32-20090113.oden/include/rpc/types.h	2009-01-28 22:24:30.000000000 +0100
@@ -63,6 +63,16 @@ typedef unsigned long rpcport_t;
 #ifndef makedev /* ie, we haven't already included it */
 #include <sys/types.h>
 #endif
+
+# ifndef __dietlibc_u_char_defined
+typedef char* caddr_t;
+typedef uint32_t u_long;
+typedef unsigned char u_char;
+typedef unsigned short u_short;
+typedef unsigned int u_int;
+# define __dietlibc_u_char_defined
+#endif
+
 #include <sys/time.h>
 #include <sys/param.h>
 
diff -Naurp dietlibc-0.32-20090113/include/sys/types.h dietlibc-0.32-20090113.oden/include/sys/types.h
--- dietlibc-0.32-20090113/include/sys/types.h	2007-10-18 23:08:02.000000000 +0200
+++ dietlibc-0.32-20090113.oden/include/sys/types.h	2009-01-28 22:24:30.000000000 +0100
@@ -115,14 +115,6 @@ typedef uint16_t sa_family_t;
 
 #ifdef _BSD_SOURCE
 /* die, BSD, die! */
-typedef char* caddr_t __attribute_dontuse__;
-typedef unsigned char u_char __attribute_dontuse__;
-typedef unsigned short u_short __attribute_dontuse__;
-typedef unsigned int u_int __attribute_dontuse__;
-typedef unsigned char uchar __attribute_dontuse__;
-typedef unsigned short ushort __attribute_dontuse__;
-typedef unsigned int uint __attribute_dontuse__;
-typedef uint32_t u_long __attribute_dontuse__;
 typedef uint32_t n_long __attribute_dontuse__;
 typedef uint32_t n_time __attribute_dontuse__;
 typedef uint16_t n_short __attribute_dontuse__;
@@ -130,6 +122,17 @@ typedef uint32_t ulong __attribute_dontu
 typedef uint32_t nlong __attribute_dontuse__;
 typedef uint32_t ntime __attribute_dontuse__;
 typedef uint16_t nshort __attribute_dontuse__;
+#ifndef __dietlibc_u_char_defined
+typedef char* caddr_t __attribute_dontuse__;
+typedef unsigned char u_char __attribute_dontuse__;
+typedef unsigned short u_short __attribute_dontuse__;
+typedef uint32_t u_long __attribute_dontuse__;
+typedef unsigned int u_int __attribute_dontuse__;
+#define __dietlibc_u_char_defined
+#endif
+typedef unsigned char uchar __attribute_dontuse__;
+typedef unsigned short ushort __attribute_dontuse__;
+typedef unsigned int uint __attribute_dontuse__;
 
 /* never heard of these two, but dump uses them */
 #if !defined(__STRICT_ANSI__) || __STDC_VERSION__ + 0 >= 199900L
