--- dietlibc-0.29/test/cycles.c.sparc_rdtsc	2006-04-20 14:29:22.000000000 -0400
+++ dietlibc-0.29/test/cycles.c	2006-04-20 14:29:55.000000000 -0400
@@ -22,6 +22,10 @@
     : "=r" (tbu), "=r" (tbl), "=r" (chk) ); \
   dst = ((uint64_t)tbu << 32) | tbl; \
 } while (0)
+#elif defined (__sparcv9__)
+#define RDTSC(dst) asm volatile ("rd %%tick, %%0":"=r"(dst))
+#elif defined (__sparc__)
+#define RDTSC(dst) NULL
 #else
 #error "Unimplemented rdtsc"
 #endif
