--- dietlibc-0.27/include/linux/types.h.kernel2.6-types	2001-07-30 15:34:41.000000000 +0200
+++ dietlibc-0.27/include/linux/types.h	2005-01-29 15:47:46.416375731 +0100
@@ -3,4 +3,28 @@
 
 #include <asm/types.h>
 
+/*
+ * The type of an index into the pagecache.  Use a #define so asm/types.h
+ * can override it.
+ * XXX: temporary solution
+ */
+#ifndef pgoff_t
+#define pgoff_t unsigned long
+#endif
+
+#ifdef __CHECKER__
+#define __bitwise __attribute__((bitwise))
+#else
+#define __bitwise
+#endif
+
+typedef __u16 __bitwise __le16;
+typedef __u16 __bitwise __be16;
+typedef __u32 __bitwise __le32;
+typedef __u32 __bitwise __be32;
+#if defined(__GNUC__)
+typedef __u64 __bitwise __le64;
+typedef __u64 __bitwise __be64;
+#endif
+
 #endif
