--- dietlibc-0.29/dietfeatures.h.features	2005-05-07 19:08:59.000000000 -0400
+++ dietlibc-0.29/dietfeatures.h	2006-04-20 12:30:47.000000000 -0400
@@ -43,11 +43,11 @@
 /* 20040118: enabling this breaks User Mode Linux!  It's their fault. */
 #define WANT_SYSENTER
 
-#define WANT_LINKER_WARNINGS
+#undef WANT_LINKER_WARNINGS
 
 /* you need to define this if you want to run your programs with large
  * file support on kernel 2.2 or 2.0 */
-#define WANT_LARGEFILE_BACKCOMPAT
+#undef WANT_LARGEFILE_BACKCOMPAT
 
 /* do you want localtime(3) to read /etc/localtime?
  * Needed for daylight saving time etc. */
@@ -59,7 +59,7 @@
 #define WANT_FULL_RESOLV_CONF
 
 /* do you want IPv6 transport support in the DNS resolver? */
-#define WANT_IPV6_DNS
+#undef WANT_IPV6_DNS
 
 /* do you want gethostbyname and friends to consult /etc/hosts? */
 #define WANT_ETC_HOSTS
@@ -69,7 +69,7 @@
 #define WANT_INET_ADDR_DNS
 
 /* do you want math functions high precision rather than fast/small? */
-#define WANT_HIGH_PRECISION_MATH
+#undef WANT_HIGH_PRECISION_MATH
 
 /* do you want support for matherr? */
 #define WANT_MATHERR
@@ -79,12 +79,12 @@
 
 /* do you want diet to include a safeguard dependency to make linking
  * against glibc fail?  This may fail with older binutils. */
-#define WANT_SAFEGUARD
+#undef WANT_SAFEGUARD
 
 /* This enables zeroconf DNS aka Rendezvous aka Bonjour. */
 /* This code will try zeroconf DNS if you ask for host.local or if you
  * ask for an unqualified hostname */
-#define WANT_PLUGPLAY_DNS
+#undef WANT_PLUGPLAY_DNS
 
 /* do you want that malloc(0) return a pointer to a "zero-length" object
  * that is realloc-able; means realloc(..,size) gives a NEW object (like a
