--- dietlibc-0.29/sparc/strcmp.S.sparc_weak_asm	2006-04-20 15:38:58.000000000 -0400
+++ dietlibc-0.29/sparc/strcmp.S	2006-04-20 15:39:34.000000000 -0400
@@ -11,6 +11,8 @@
 
 .text
 .globl strcmp
+.weak strcoll
+strcoll:
 strcmp:
 	clr %o4
 .Lloop:
--- dietlibc-0.29/sparc/strchr.S.sparc_weak_asm	2006-04-20 15:38:58.000000000 -0400
+++ dietlibc-0.29/sparc/strchr.S	2006-04-20 15:39:34.000000000 -0400
@@ -11,6 +11,8 @@
 
 .text
 .globl strchr
+.weak index
+index:
 strchr:
 	ldub [%o0], %o2
 	cmp %o2, %o1
