diff -u src-org/analogue.cpp src/analogue.cpp
--- src-org/analogue.cpp    Thu Oct 28 19:22:56 2004
+++ src/analogue.cpp    Thu Oct 28 19:24:43 2004
@@ -24,7 +24,7 @@
 
 /*****************************************************************************/
 
-#include <malloc.h>
+#include <unistd.h>
 #include <math.h>
 #include <stdlib.h>
 #include "cmt.h"
diff -u src-org/canyondelay.cpp src/canyondelay.cpp
--- src-org/canyondelay.cpp Thu Oct 28 19:22:56 2004
+++ src/canyondelay.cpp Thu Oct 28 19:24:43 2004
@@ -25,7 +25,7 @@
 /*****************************************************************************/
 
 
-#include <malloc.h>
+#include <unistd.h>
 #include <math.h>
 #include "cmt.h"
 
Common subdirectories: src-org/freeverb and src/freeverb
diff -u src-org/lofi.cpp src/lofi.cpp
--- src-org/lofi.cpp    Thu Oct 28 19:22:56 2004
+++ src/lofi.cpp    Thu Oct 28 19:24:43 2004
@@ -24,7 +24,7 @@
 
 /*****************************************************************************/
 
-#include <malloc.h>
+#include <unistd.h>
 #include <math.h>
 #include <stdlib.h>
 #include "cmt.h"
--- src-org/phasemod.cpp    Thu Oct 28 19:22:56 2004
+++ src/phasemod.cpp    Thu Oct 28 19:24:43 2004
@@ -24,7 +24,7 @@
 
 /*****************************************************************************/
 
-#include <malloc.h>
+#include <unistd.h>
 #include <math.h>
 #include <stdlib.h>
 #include "cmt.h"
diff -u src-org/syndrum.cpp src/syndrum.cpp
--- src-org/syndrum.cpp Thu Oct 28 19:22:56 2004
+++ src/syndrum.cpp Thu Oct 28 19:24:43 2004
@@ -24,7 +24,7 @@
 
 /*****************************************************************************/
 
-#include <malloc.h>
+#include <unistd.h>
 #include <math.h>
 #include "cmt.h"
 
diff -u src-org/vcf303.cpp src/vcf303.cpp
--- src-org/vcf303.cpp  Thu Oct 28 19:22:56 2004
+++ src/vcf303.cpp  Thu Oct 28 19:24:43 2004
@@ -26,7 +26,7 @@
 /*****************************************************************************/
 
 
-#include <malloc.h>
+#include <unistd.h>
 #include <math.h>
 #include "cmt.h"
