diff -up gzip-1.3.12/zforce.in.zforce gzip-1.3.12/zforce.in
--- gzip-1.3.12/zforce.in.zforce	2007-02-05 21:54:26.000000000 +0100
+++ gzip-1.3.12/zforce.in	2008-09-01 14:17:26.000000000 +0200
@@ -58,7 +58,7 @@ for i do
     continue
   fi
 
-  if gzip -lv < "$i" 2>/dev/null | grep '^defl' > /dev/null; then
+  if gzip -l < "$i" 2>/dev/null | grep '^compressed' > /dev/null; then
 
     new="$i.gz"
     if mv "$i" "$new"; then
