--- zmore.in
+++ zmore.in
@@ -52,11 +52,33 @@
    trap 'stty $ncb echo 2>/dev/null; exit' 0 2 3 5 10 13 15
 fi
 
+opt=
+pager ()
+{
+    eval ${PAGER-more} \$opt
+    cat > /dev/null
+}
+
+while :; do
+  case $1 in
+    --)
+      shift
+      break
+      ;;
+    [-+]*)
+      opt="$opt $1"
+      shift
+      ;;
+    *)
+      break
+      ;;
+  esac
+done
 if test $# = 0; then
     if test -t 0; then
 	echo "$usage"
     else
-	gzip -cdfq | eval ${PAGER-more}
+	gzip -cdfq | pager
     fi
 else
     FIRST=1
@@ -80,7 +102,7 @@
 	fi
 	if test "$ANS" != 's'; then
 		echo "------> $FILE <------"
-		gzip -cdfq -- "$FILE" | eval ${PAGER-more}
+		gzip -cdfq -- "$FILE" | pager
 	fi
 	if test -t 1; then
 		FIRST=0
