diff -Nuar asymptote-2.21.orig/doc/Makefile.in asymptote-2.21/doc/Makefile.in
--- asymptote-2.21.orig/doc/Makefile.in	2013-05-08 21:43:10.569087529 +0300
+++ asymptote-2.21/doc/Makefile.in	2013-05-08 22:04:43.813122161 +0300
@@ -9,7 +9,7 @@
 infodir = $(DESTDIR)@infodir@
 datarootdir = @datarootdir@
 INSTALL = @INSTALL@
-TEXI2DVI = @TEXI2DVI@
+TEXI2DVI = TEX=tex @TEXI2DVI@
 
 export docdir infodir INSTALL
 
