diff -Nur ngrep-1.45-old/ngrep.c ngrep-1.45/ngrep.c
--- ngrep-1.45-old/ngrep.c	2009-09-15 01:30:15.095452744 +0300
+++ ngrep-1.45/ngrep.c	2009-09-15 01:30:44.403205366 +0300
@@ -197,7 +197,7 @@
     signal(SIGWINCH, update_windowsize);
 #endif
 
-    setlocale(LC_ALL, "");
+    setlocale(LC_ALL, "en_US.UTF-8");
 
     while ((c = getopt(argc, argv, "LNhXViwqpevxlDtTRMs:n:c:d:A:I:O:S:P:F:W:")) != EOF) {
         switch (c) {
