Index: meanwhile-1.0.2/samples/Makefile.am
===================================================================
--- meanwhile-1.0.2.orig/samples/Makefile.am
+++ meanwhile-1.0.2/samples/Makefile.am
@@ -19,7 +19,7 @@ sample_DATA = \
 	$(SAMPLES_SRC)
 
 
-sampledir = $(datadir)/doc/@PACKAGE@-doc-@VERSION@/samples
+sampledir = $(datadir)/doc/@PACKAGE@/samples
 
 
 EXTRA_DIST = $(sample_DATA) $(sample_SCRIPTS)
