diff -Naur --exclude '*~' XML-LibXML-1.69/perl-libxml-sax.c XML-LibXML-1.69-fix-format-errors/perl-libxml-sax.c
--- XML-LibXML-1.69/perl-libxml-sax.c	2008-10-30 22:21:04.000000000 +0100
+++ XML-LibXML-1.69-fix-format-errors/perl-libxml-sax.c	2009-05-19 21:19:41.000000000 +0200
@@ -242,7 +242,7 @@
     sv_2mortal(rv);
     if (SvTRUE(ERRSV)) {
         STRLEN n_a;
-        croak(SvPV(ERRSV, n_a));
+        croak("%s", SvPV(ERRSV, n_a));
     }
     FREETMPS ;
     LEAVE ;
@@ -287,7 +287,7 @@
     sv_2mortal(rv);
     if (SvTRUE(ERRSV)) {
         STRLEN n_a;
-        croak(SvPV(ERRSV, n_a));
+        croak("%s", SvPV(ERRSV, n_a));
     }
     
     FREETMPS ;
@@ -662,7 +662,7 @@
         call_method( "start_document", G_SCALAR | G_EVAL | G_DISCARD );
         if (SvTRUE(ERRSV)) {
             STRLEN n_a;
-            croak(SvPV(ERRSV, n_a));
+            croak("%s", SvPV(ERRSV, n_a));
         }
         
         SPAGAIN;
@@ -697,7 +697,7 @@
         sv_2mortal(rv);
         if (SvTRUE(ERRSV)) {
             STRLEN n_a;
-            croak(SvPV(ERRSV, n_a));
+            croak("%s", SvPV(ERRSV, n_a));
         }
         
         FREETMPS ;
@@ -726,7 +726,7 @@
     call_pv( "XML::LibXML::_SAXParser::end_document", G_SCALAR | G_EVAL | G_DISCARD );
     if (SvTRUE(ERRSV)) {
         STRLEN n_a;
-        croak(SvPV(ERRSV, n_a));
+        croak("%s", SvPV(ERRSV, n_a));
     }
     
     FREETMPS ;
@@ -776,7 +776,7 @@
 
     if (SvTRUE(ERRSV)) {
         STRLEN n_a;
-        croak(SvPV(ERRSV, n_a));
+        croak("%s", SvPV(ERRSV, n_a));
     }
     
     FREETMPS ;
@@ -813,7 +813,7 @@
     
     if (SvTRUE(ERRSV)) {
         STRLEN n_a;
-        croak(SvPV(ERRSV, n_a));
+        croak("%s", SvPV(ERRSV, n_a));
     }
     
     FREETMPS ;
@@ -862,7 +862,7 @@
 
         if (SvTRUE(ERRSV)) {
 	  STRLEN n_a;
-	  croak(SvPV(ERRSV, n_a));
+	  croak("%s", SvPV(ERRSV, n_a));
 	}
         
         FREETMPS ;
@@ -904,7 +904,7 @@
 
         if (SvTRUE(ERRSV)) {
             STRLEN n_a;
-            croak(SvPV(ERRSV, n_a));
+            croak("%s", SvPV(ERRSV, n_a));
         }
         
         FREETMPS ;
@@ -938,7 +938,7 @@
         call_method( "start_cdata", G_SCALAR | G_EVAL | G_DISCARD );
         if (SvTRUE(ERRSV)) {
             STRLEN n_a;
-            croak(SvPV(ERRSV, n_a));
+            croak("%s", SvPV(ERRSV, n_a));
         }
         
         SPAGAIN;        
@@ -954,7 +954,7 @@
         call_method( "characters", G_SCALAR | G_EVAL | G_DISCARD);
         if (SvTRUE(ERRSV)) {
             STRLEN n_a;
-            croak(SvPV(ERRSV, n_a));
+            croak("%s", SvPV(ERRSV, n_a));
         }
         
         SPAGAIN;        
@@ -968,7 +968,7 @@
         
         if (SvTRUE(ERRSV)) {
             STRLEN n_a;
-            croak(SvPV(ERRSV, n_a));
+            croak("%s", SvPV(ERRSV, n_a));
         }
         
         FREETMPS ;
@@ -1011,7 +1011,7 @@
 
         if (SvTRUE(ERRSV)) {
             STRLEN n_a;
-            croak(SvPV(ERRSV, n_a));
+            croak("%s", SvPV(ERRSV, n_a));
         }
         
         FREETMPS ;
@@ -1056,7 +1056,7 @@
 
         if (SvTRUE(ERRSV)) {
             STRLEN n_a;
-            croak(SvPV(ERRSV, n_a));
+            croak("%s", SvPV(ERRSV, n_a));
         }
 
         PUSHMARK(SP) ;
@@ -1173,7 +1173,7 @@
 
     if (SvTRUE(ERRSV)) {
         STRLEN n_a;
-        croak(SvPV(ERRSV, n_a));
+        croak("%s", SvPV(ERRSV, n_a));
     }
     
     FREETMPS ;
@@ -1236,7 +1236,7 @@
 #endif
     if (SvTRUE(ERRSV)) {
         STRLEN n_a;
-        croak(SvPV(ERRSV, n_a));
+        croak("%s", SvPV(ERRSV, n_a));
     }
     
     FREETMPS ;
@@ -1284,7 +1284,7 @@
     call_pv( "XML::LibXML::_SAXParser::fatal_error", G_SCALAR | G_EVAL | G_DISCARD );
     if (SvTRUE(ERRSV)) {
         STRLEN n_a;
-        croak(SvPV(ERRSV, n_a));
+        croak("%s", SvPV(ERRSV, n_a));
     }
     
     FREETMPS ;
