--- XML-LibXML-1.63/Makefile.PL.org	    2007-04-24 22:04:35.000000000 +0300
+++ XML-LibXML-1.63/Makefile.PL	        2007-04-24 22:08:35.000000000 +0300
@@ -315,16 +315,16 @@
    package MY;
    my $script = shift->SUPER::install(@_);
    unless ( $::skipsaxinstall ) {
-     $script =~ s/install :: (.*)$/install :: $1 install_sax_driver/m;
-     $script .= <<"INSTALL";
-
-install_sax_driver :
-\t\@\$(PERL) -I\$(INSTALLSITELIB) -MXML::SAX -e "XML::SAX->add_parser(q(XML::LibXML::SAX::Parser))->save_parsers()"
-\t\@\$(PERL) -I\$(INSTALLSITELIB) -MXML::SAX -e "XML::SAX->add_parser(q(XML::LibXML::SAX))->save_parsers()"
-
-INSTALL
-   } else {
-     warn "Note: 'make install' will skip XML::LibXML::SAX registration with XML::SAX!\n";
+#     $script =~ s/install :: (.*)$/install :: $1 install_sax_driver/m;
+#     $script .= <<"INSTALL";
+#
+#install_sax_driver :
+#\t\@\$(PERL) -I\$(INSTALLSITELIB) -MXML::SAX -e "XML::SAX->add_parser(q(XML::LibXML::SAX::Parser))->save_parsers()"
+#\t\@\$(PERL) -I\$(INSTALLSITELIB) -MXML::SAX -e "XML::SAX->add_parser(q(XML::LibXML::SAX))->save_parsers()"
+#
+#INSTALL
+#   } else {
+#     warn "Note: 'make install' will skip XML::LibXML::SAX registration with XML::SAX!\n";
    }
    return $script;
 }
