--- nikto.conf.orig	2010-07-12 16:11:57.234501404 +0300
+++ nikto.conf	2010-07-12 16:14:14.530251214 +0300
@@ -67,10 +67,10 @@
 CHECKMETHODS=HEAD GET
 
 # If you want to specify the location of any of the files, specify them here
-# EXECDIR=/opt/nikto
-# PLUGINDIR=/opt/nikto/plugins
-# TEMPLATEDIR=/opt/nikto/templates
-# DOCDIR=/opt/nikto/docs
+EXECDIR=/usr/bin
+PLUGINDIR=/usr/share/nikto/plugins
+TEMPLATEDIR=/usr/share/nikto/templates
+DOCDIR=/usr/share/doc
 
 # Default plugin macros
 @@MUTATE=dictionary;subdomain
