Index: sysstat-9.0.6.1/configure.in
===================================================================
--- sysstat-9.0.6.1.orig/configure.in
+++ sysstat-9.0.6.1/configure.in
@@ -204,7 +204,7 @@ AC_ARG_VAR([conf_dir],[sysstat configura
 if test x$conf_dir != x""; then
    SYSCONFIG_DIR=$conf_dir
 else
-   SYSCONFIG_DIR=/etc/sysconfig
+   SYSCONFIG_DIR=/etc/sysstat
 fi
 AC_MSG_RESULT($SYSCONFIG_DIR)
 if test ! -d $SYSCONFIG_DIR; then
Index: sysstat-9.0.6.1/Makefile.in
===================================================================
--- sysstat-9.0.6.1.orig/Makefile.in
+++ sysstat-9.0.6.1/Makefile.in
@@ -5,7 +5,7 @@
 VERSION = @PACKAGE_VERSION@
 PACKAGE = @PACKAGE_NAME@
 # Needed by docdir
-PACKAGE_TARNAME = $(PACKAGE)-$(VERSION)
+PACKAGE_TARNAME = $(PACKAGE)
 
 # Compiler to use
 CC = @CC@
