diff -up sysstat-9.0.6.1/sa_common.c.pom sysstat-9.0.6.1/sa_common.c
--- sysstat-9.0.6.1/sa_common.c.pom	2009-10-17 15:08:07.000000000 +0200
+++ sysstat-9.0.6.1/sa_common.c	2010-07-19 08:37:43.000000000 +0200
@@ -207,7 +207,7 @@ int next_slice(unsigned long long uptime
 	pt1 = (entry / interval) * interval;
 	pt2 = ((entry / interval) + 1) * interval;
 
-	return (((pt1 >= min) && (pt1 < max)) || ((pt2 >= min) && (pt2 < max)));
+	return (((pt1 >= min) && (pt1 <= max)) || ((pt2 >= min) && (pt2 <= max)));
 }
 
 /*
