diff -up sysstat-10.0.0/Makefile.in.pom sysstat-10.0.0/Makefile.in
--- sysstat-10.0.0/Makefile.in.pom	2011-03-06 16:46:23.000000000 +0100
+++ sysstat-10.0.0/Makefile.in	2011-04-04 10:23:02.000000000 +0200
@@ -196,8 +196,8 @@ librdsensors.a: librdsensors.a(rd_sensor
 
 sadc.o: sadc.c sa.h version.h common.h ioconf.h sysconfig.h rd_stats.h rd_sensors.h
 
-sadc: LFLAGS += $(LFSENSORS)
 sadc: sadc.o act_sadc.o sa_wrap.o sa_common.o librdstats.a librdsensors.a libsyscom.a
+	$(CC) -o $@ $(CFLAGS) $^ $(LFLAGS) -lsensors
 
 sar.o: sar.c sa.h version.h common.h ioconf.h pr_stats.h sysconfig.h
 
