--- src/main.cpp.orig	2007-11-12 14:27:55.000000000 +0200
+++ src/main.cpp	2007-11-12 18:40:30.000000000 +0200
@@ -78,7 +78,7 @@
 	std::string separator = PHYSFS_getDirSeparator();
 	// Game should be playable out of the source package on all 
 	// platforms
-	PHYSFS_addToSearchPath("data", 1);
+	PHYSFS_addToSearchPath("/usr/share/blobby", 1);
 	PHYSFS_addToSearchPath("data/gfx.zip", 1);
 	PHYSFS_addToSearchPath("data/sounds.zip", 1);
 	PHYSFS_addToSearchPath("data/scripts.zip", 1);
