diff -Naur qt-recordmydesktop-0.3.8.orig/src/Makefile.in qt-recordmydesktop-0.3.8/src/Makefile.in
--- qt-recordmydesktop-0.3.8.orig/src/Makefile.in	2008-11-23 17:13:57.000000000 +0200
+++ qt-recordmydesktop-0.3.8/src/Makefile.in	2008-12-15 22:11:51.000000000 +0200
@@ -56,7 +56,7 @@
   esac;
 am__strip_dir = `echo $$p | sed -e 's|^.*/||'`;
 qt_recordMyDesktopPYTHON_INSTALL = $(INSTALL_DATA)
-py_compile = $(top_srcdir)/py-compile
+# py_compile = $(top_srcdir)/py-compile
 desktopDATA_INSTALL = $(INSTALL_DATA)
 pixmapDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(desktop_DATA) $(pixmap_DATA)
@@ -283,13 +283,13 @@
 	    $(qt_recordMyDesktopPYTHON_INSTALL) "$$b$$p" "$(DESTDIR)$(qt_recordMyDesktopdir)/$$f"; \
 	  else :; fi; \
 	done; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(qt_recordMyDesktopdir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(qt_recordMyDesktopdir)" $$dlist; \
-	  fi; \
-	else :; fi
+# 	if test -n "$$dlist"; then \
+# 	  if test -z "$(DESTDIR)"; then \
+# 	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(qt_recordMyDesktopdir)" $$dlist; \
+# 	  else \
+# 	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(qt_recordMyDesktopdir)" $$dlist; \
+# 	  fi; \
+# 	else :; fi
 
 uninstall-qt_recordMyDesktopPYTHON:
 	@$(NORMAL_UNINSTALL)
