diff -Nur fontconfig-2.6.0/conf.d/30-metric-aliases.conf fontconfig-2.6.0-old/conf.d/30-metric-aliases.conf
--- fontconfig-2.6.0/conf.d/30-metric-aliases.conf	2008-09-22 10:45:32.000000000 +0300
+++ fontconfig-2.6.0-old/conf.d/30-metric-aliases.conf	2008-09-22 10:45:35.000000000 +0300
@@ -46,6 +46,12 @@
 
 		3) Alias each generic to its specifics.
 		   eg. Arial to Liberation Sans, Albany, and Albany AMT
+
+                   However, the AMT fonts are quite ugly and do not
+                   render well on-screen, so we avoid matching them if
+                   the application said `anymetrics'; in that case, a
+                   more generic font with different metrics but better
+                   appearance will be used.
 	-->
 
 <!-- Map specifics to generics -->
@@ -185,7 +191,7 @@
 	  <accept>
           <family>Liberation Sans</family>
 	  <family>Albany</family>
-	  <family>Albany AMT</family>
+	  <!-- <family>Albany AMT</family> -->
 	  </accept>
 	</alias>
 
@@ -194,7 +200,7 @@
 	  <accept>
           <family>Liberation Serif</family>
 	  <family>Thorndale</family>
-	  <family>Thorndale AMT</family>
+	  <!-- <family>Thorndale AMT</family> -->
 	  </accept>
 	</alias>
 
@@ -203,9 +209,42 @@
 	  <accept>
           <family>Liberation Mono</family>
 	  <family>Cumberland</family>
-	  <family>Cumberland AMT</family>
+	  <!-- <family>Cumberland AMT</family> -->
 	  </accept>
 	</alias>
 
+       	<match target="pattern">
+          <test name="family">
+            <string>Times New Roman</string>
+          </test>
+          <test name="anymetrics" qual="all" compare="not_eq">
+            <bool>true</bool>
+          </test>
+          <edit name="family" mode="append">
+            <string>Thorndale AMT</string>
+          </edit>
+     	</match>
+       	<match target="pattern">
+          <test name="family">
+            <string>Arial</string>
+          </test>
+          <test name="anymetrics" qual="all" compare="not_eq">
+            <bool>true</bool>
+          </test>
+          <edit name="family" mode="append">
+            <string>Albany AMT</string>
+          </edit>
+     	</match>
+       	<match target="pattern">
+          <test name="family">
+            <string>Courier New</string>
+          </test>
+          <test name="anymetrics" qual="all" compare="not_eq">
+            <bool>true</bool>
+          </test>
+          <edit name="family" mode="append">
+            <string>Cumberland AMT</string>
+          </edit>
+     	</match>
 
 </fontconfig>
