diff -Nur fontconfig-2.6.0/fontconfig/fontconfig.h fontconfig-2.6.0-old/fontconfig/fontconfig.h
--- fontconfig-2.6.0/fontconfig/fontconfig.h	2008-09-22 10:45:32.000000000 +0300
+++ fontconfig-2.6.0-old/fontconfig/fontconfig.h	2008-09-22 10:45:35.000000000 +0300
@@ -112,6 +112,7 @@
 #define FC_EMBEDDED_BITMAP  "embeddedbitmap"	/* Bool - true to enable embedded bitmaps */
 #define FC_DECORATIVE	    "decorative"	/* Bool - true if style is a decorative variant */
 #define FC_LCD_FILTER	    "lcdfilter"		/* Int */
+#define FC_ANY_METRICS      "anymetrics"	/* Bool - true to allow metrics which do not match the requested font */
 
 #define FC_CACHE_SUFFIX		    ".cache-"FC_CACHE_VERSION
 #define FC_DIR_CACHE_FILE	    "fonts.cache-"FC_CACHE_VERSION
diff -Nur fontconfig-2.6.0/src/fcint.h fontconfig-2.6.0-old/src/fcint.h
--- fontconfig-2.6.0/src/fcint.h	2008-09-22 10:45:32.000000000 +0300
+++ fontconfig-2.6.0-old/src/fcint.h	2008-09-22 10:45:35.000000000 +0300
@@ -849,7 +849,8 @@
 #define FC_EMBEDDED_BITMAP_OBJECT	39
 #define FC_DECORATIVE_OBJECT	40
 #define FC_LCD_FILTER_OBJECT	41
-#define FC_MAX_BASE_OBJECT	FC_LCD_FILTER_OBJECT
+#define FC_ANY_METRICS_OBJECT	42
+#define FC_MAX_BASE_OBJECT	FC_ANY_METRICS_OBJECT
 
 FcPrivate FcBool
 FcNameBool (const FcChar8 *v, FcBool *result);
diff -Nur fontconfig-2.6.0/src/fcname.c fontconfig-2.6.0-old/src/fcname.c
--- fontconfig-2.6.0/src/fcname.c	2008-09-22 10:45:32.000000000 +0300
+++ fontconfig-2.6.0-old/src/fcname.c	2008-09-22 10:45:35.000000000 +0300
@@ -76,6 +76,7 @@
     { FC_EMBEDDED_BITMAP,   FcTypeBool },
     { FC_DECORATIVE,	FcTypeBool },
     { FC_LCD_FILTER,	FcTypeInteger }, /* 41 */
+    { FC_ANY_METRICS,	FcTypeBool }, /* 42 */
 };
 
 #define NUM_OBJECT_TYPES    (sizeof _FcBaseObjectTypes / sizeof _FcBaseObjectTypes[0])
@@ -440,6 +441,7 @@
     { (FcChar8 *) "lcddefault",	    "lcdfilter",    FC_LCD_DEFAULT },
     { (FcChar8 *) "lcdlight",	    "lcdfilter",    FC_LCD_LIGHT },
     { (FcChar8 *) "lcdlegacy",	    "lcdfilter",    FC_LCD_LEGACY },
+    { (FcChar8 *) "anymetrics",	    "anymetrics",   FcTrue },
 };
 
 #define NUM_FC_CONSTANTS   (sizeof _FcBaseConstants/sizeof _FcBaseConstants[0])
