diff -Nur fontconfig-2.6.0/conf.d/40-nonlatin.conf fontconfig-2.6.0-old/conf.d/40-nonlatin.conf
--- fontconfig-2.6.0/conf.d/40-nonlatin.conf	2008-09-22 10:45:32.000000000 +0300
+++ fontconfig-2.6.0-old/conf.d/40-nonlatin.conf	2008-09-22 10:45:35.000000000 +0300
@@ -51,6 +51,7 @@
 		<family>Baekmuk Dotum</family>
 		<family>SimSun</family>
 		<family>MgOpen Modata</family>
+		<family>MgOpen Moderna</family>
 		<family>Sazanami Gothic</family>
 		<family>AR PL ShanHeiSun Uni</family>
 		<family>ZYSong18030</family>
