Index: fontconfig-2.7.0/conf.d/Makefile.am
===================================================================
--- fontconfig-2.7.0.orig/conf.d/Makefile.am
+++ fontconfig-2.7.0/conf.d/Makefile.am
@@ -25,7 +25,12 @@ DOC_FILES= \
 	README
 
 CONF_FILES = \
+	10-antialias.conf \
 	10-autohint.conf \
+	10-hinting.conf \
+	10-hinting-full.conf \
+	10-hinting-medium.conf \
+	10-hinting-slight.conf \
 	10-no-sub-pixel.conf \
 	10-sub-pixel-bgr.conf \
 	10-sub-pixel-rgb.conf \
@@ -42,6 +47,7 @@ CONF_FILES = \
 	49-sansserif.conf \
 	50-user.conf \
 	51-local.conf \
+	53-monospace-lcd-filter.conf \
 	60-latin.conf \
 	65-fonts-persian.conf \
 	65-khmer.conf \
@@ -53,6 +59,10 @@ CONF_FILES = \
 	90-synthetic.conf
 
 CONF_LINKS = \
+	10-antialias.conf \
+	10-hinting.conf \
+	10-hinting-full.conf \
+	10-no-sub-pixel.conf \
 	20-fix-globaladvance.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
@@ -62,6 +72,7 @@ CONF_LINKS = \
 	49-sansserif.conf \
 	50-user.conf \
 	51-local.conf \
+	53-monospace-lcd-filter.conf \
 	60-latin.conf \
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
