Index: fontconfig-2.7.0/conf.d/11-lcd-filter-lcddefault.conf
===================================================================
--- /dev/null
+++ fontconfig-2.7.0/conf.d/11-lcd-filter-lcddefault.conf
@@ -0,0 +1,11 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!-- conf.d/lcd-filter-lcddefault.conf -->
+<fontconfig>
+<!--  Use lcddefault as default for LCD filter -->
+  <match target="font">
+    <edit mode="assign" name="lcdfilter">
+      <const>lcddefault</const>
+    </edit>
+  </match>
+</fontconfig>
Index: fontconfig-2.7.0/conf.d/Makefile.am
===================================================================
--- fontconfig-2.7.0.orig/conf.d/Makefile.am
+++ fontconfig-2.7.0/conf.d/Makefile.am
@@ -37,6 +37,7 @@ CONF_FILES = \
 	10-sub-pixel-vbgr.conf \
 	10-sub-pixel-vrgb.conf \
 	10-unhinted.conf \
+	11-lcd-filter-lcddefault.conf \
 	20-fix-globaladvance.conf \
 	20-unhint-small-vera.conf \
 	25-unhint-nonlatin.conf \
@@ -63,6 +64,7 @@ CONF_LINKS = \
 	10-hinting.conf \
 	10-hinting-medium.conf \
 	10-no-sub-pixel.conf \
+	11-lcd-filter-lcddefault.conf \
 	20-fix-globaladvance.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
