--- unix/Makefile.orig	2007-02-13 11:00:36.000000000 +0200
+++ unix/Makefile	2007-12-22 02:35:28.000000000 +0200
@@ -22,7 +22,7 @@
 #
 # Name of raytracing library file to create
 #
-RAYLIB=${ARCHDIR}/libtachyon.a
+RAYLIB=${ARCHDIR}/libtachyon.so.0.0
 RAYLIBDIR=${ARCHDIR}
 
 #
@@ -255,9 +255,11 @@
 	${OBJDIR}/winbmp.o
 
 ${RAYLIB} : ${RAYOBJS}
-	rm -f ${RAYLIB}
-	${AR} ${ARFLAGS} ${RAYLIB} ${RAYOBJS}
-	${RANLIB} ${RAYLIB}
+	rm -f ${RAYLIB} ${RAYLIB}.0
+	$(CC) $(CFLAGS) $(LDFLAGS) -fPIC -DPIC -shared -Wl,-soname=libtachyon.so.0 -o ${RAYLIB} ${RAYOBJS} ${JPEGLIB} ${PNGLIB} -lpthread -lm
+	cd ${ARCHDIR}
+	ln -s ${RAYLIB} libtachyon.so.0
+	ln -s ${RAYLIB} libtachyon.so
 
 ${OBJDIR}/vol.o : ${SRCDIR}/vol.c ${OBJDEPS}
 	${CC} ${CFLAGS} -c ${SRCDIR}/vol.c -o ${OBJDIR}/vol.o
