Index: netcat-openbsd-1.89/netcat.c
===================================================================
--- netcat-openbsd-1.89.orig/netcat.c	2008-01-25 13:14:34.000000000 -0500
+++ netcat-openbsd-1.89/netcat.c	2008-01-25 13:15:49.000000000 -0500
@@ -966,7 +967,7 @@
 	\t-x addr[:port]\tSpecify proxy address and port\n\
 	\t-z		Zero-I/O mode [used for scanning]\n\
 	Port numbers can be individual or ranges: lo-hi [inclusive]\n");
-	exit(1);
+	exit(0);
 }
 
 void
