Index: ejabberd-2.1.5/src/ejabberdctl.template
===================================================================
--- ejabberd-2.1.5.orig/src/ejabberdctl.template
+++ ejabberd-2.1.5/src/ejabberdctl.template
@@ -62,7 +62,7 @@ EJID=`id -g $INSTALLUSER`
 EXEC_CMD="false"
 for GID in $GIDS; do
     if [ $GID -eq 0 ] ; then
-	EXEC_CMD="su ${INSTALLUSER} -p -c"
+	EXEC_CMD="su ${INSTALLUSER} -s /bin/sh -p -c"
     fi
 done
 if [ "$ID" -eq "$EJID" ] ; then
