diff -Nuar AMDOverdriveCtrl.orig/create_deb/AMDOverdriveCtrl.desktop AMDOverdriveCtrl/create_deb/AMDOverdriveCtrl.desktop
--- AMDOverdriveCtrl.orig/create_deb/AMDOverdriveCtrl.desktop	2013-12-14 01:12:10.000000000 +0200
+++ AMDOverdriveCtrl/create_deb/AMDOverdriveCtrl.desktop	2014-04-20 13:18:59.116742777 +0300
@@ -4,11 +4,13 @@
 Name[de_DE]=AMDOverdriveCtrl
 GenericName=AMD Overdrive Control
 GenericName[ru]=Разгон видеокарт AMD
-Exec=AMDOverdriveCtrl
-Icon=/usr/share/AMDOverdriveCtrl/images/icon.xpm
+GenericName[tr]=AMD Overdrive 
+Exec=AMDOverdriveCtrl
+Icon=AMDOverdriveCtrl.xpm
 Terminal=false
 Type=Application
 Categories=System;Settings;HardwareSettings;
 StartupNotify=true
 Comment=Control AMD Catalyst Overdrive settings
+Comment[tr]=AMD Catalyst Sürücüsü Kontrol Uygulaması
 Comment[ru]=Управляет разгоном видеокарт AMD, позволяет контролировать температуру, регулировать частоты и скорость вращения вентилятора
\ Dosya sonunda yenisatır yok.
