diff -Nur KoboDeluxe-0.4.1-old/kobo.cpp KoboDeluxe-0.4.1/kobo.cpp
--- KoboDeluxe-0.4.1-old/kobo.cpp	2007-12-03 16:13:21.685555675 +0200
+++ KoboDeluxe-0.4.1/kobo.cpp	2007-12-03 16:14:56.181190154 +0200
@@ -136,22 +136,26 @@
 	/*
 	 * Graphics data
 	 */
-	/* Current dir; from within the build tree */
+	/* Current dir; from within the build tree
 	fmap->addpath("GFX", "./data/gfx");
+    */
 	/* Real data dir */
 	fmap->addpath("GFX", "DATA>>gfx");
-	/* Current dir */
+	/* Current dir
 	fmap->addpath("GFX", "./gfx");
+    */
 
 	/*
 	 * Sound data
 	 */
-	/* Current dir; from within the build tree */
+	/* Current dir; from within the build tree
 	fmap->addpath("SFX", "./data/sfx");
+    */
 	/* Real data dir */
 	fmap->addpath("SFX", "DATA>>sfx");
-	/* Current dir */
+	/* Current dir
 	fmap->addpath("SFX", "./sfx");
+    */
 
 	/*
 	 * Score files (user and global)
@@ -168,8 +172,9 @@
 	fmap->addpath("CONFIG", KOBO_CONFIG_DIR);
 	/* System local */
 	fmap->addpath("CONFIG", SYSCONF_DIR);
-	/* In current dir (last resort) */
+	/* In current dir (last resort)
 	fmap->addpath("CONFIG", "./");
+    */
 }
 
 
