diff -uNr Cheetah-2.4.2.1.orig/cheetah/Template.py Cheetah-2.4.2.1/cheetah/Template.py
--- Cheetah-2.4.2.1.orig/cheetah/Template.py	2010-02-07 23:17:23.000000000 -0500
+++ Cheetah-2.4.2.1/cheetah/Template.py	2010-10-18 14:06:54.439935114 -0400
@@ -32,7 +32,7 @@
 
 filetype = None
 
-if isinstance(sys.version_info, tuple):
+if isinstance(sys.version_info[:], tuple):
     # Python 2.xx
     filetype = types.FileType
     def createMethod(func, cls):
diff -uNr Cheetah-2.4.2.1.orig/cheetah/Tests/SyntaxAndOutput.py Cheetah-2.4.2.1/cheetah/Tests/SyntaxAndOutput.py
--- Cheetah-2.4.2.1.orig/cheetah/Tests/SyntaxAndOutput.py	2010-10-18 15:00:44.668935001 -0400
+++ Cheetah-2.4.2.1/cheetah/Tests/SyntaxAndOutput.py	2010-10-18 14:43:56.500934997 -0400
@@ -847,17 +847,17 @@
     def test10(self):
         r"""func placeholder - with ('''\nstring\n''')"""
         self.verify("$aFunc('''\naoeu\n''')",
-                    "\naoeu\n")
+                    "\naoeu\n", convertEOLs=False)
 
     def test11(self):
         r"""func placeholder - with ('''\nstring'\n''')"""
         self.verify("$aFunc('''\naoeu'\n''')",
-                    "\naoeu'\n")
+                    "\naoeu'\n", convertEOLs=False)
 
     def test12(self):
         r'''func placeholder - with ("""\nstring\n""")'''
         self.verify('$aFunc("""\naoeu\n""")',
-                    "\naoeu\n")
+                    "\naoeu\n", convertEOLs=False)
 
     def test13(self):
         """func placeholder - with (string*int)"""
