Index: frostwire-5.1.4.noarch/frostwire
===================================================================
--- frostwire-5.1.4.noarch.orig/frostwire
+++ frostwire-5.1.4.noarch/frostwire
@@ -132,12 +132,12 @@ export LD_LIBRARY_PATH=`pwd`
 CLASSPATH="."
 
 #add all the lw-xxx jars first
-for JAR in $(ls -1 lw*jar); do
+for JAR in $(ls -1 /usr/share/frostwire/lw*jar); do
   CLASSPATH="${CLASSPATH}${PS}${JAR}"
 done
 
 #add all the jars.
-for JAR in $(ls -1 *jar | grep -v lw-); do
+for JAR in $(ls -1 /usr/share/frostwire/*jar | grep -v lw-); do
   CLASSPATH="${CLASSPATH}${PS}${JAR}"
 done
 
