diff -Nur goonies-1.4.1528-old/src/GLTManager.cpp goonies-1.4.1528/src/GLTManager.cpp
--- goonies-1.4.1528-old/src/GLTManager.cpp	2009-07-22 15:05:14.629655466 +0300
+++ goonies-1.4.1528/src/GLTManager.cpp	2009-07-22 15:05:18.464780785 +0300
@@ -71,7 +71,7 @@
         char filename[256];
         GLTile *t;
 
-        sprintf(filename, "graphics/%s.png", name->get());
+        sprintf(filename, "/usr/share/goonies/graphics/%s.png", name->get());
         t = new GLTile(filename);
 
 		if (t->get_n_parts()==0) {
@@ -179,7 +179,7 @@
             n->m_tile = t = new GLTile(s);
         } // if
 
-        sprintf(filename, "graphics/%s.nfo", name->get
+        sprintf(filename, "/usr/share/goonies/graphics/%s.nfo", name->get
                 ());
         {
             FILE *fp;
@@ -238,7 +238,7 @@
         char filename[256];
         GLTile *t;
 
-        sprintf(filename, "graphics/%s.png", name->get());
+        sprintf(filename, "/usr/share/goonies/graphics/%s.png", name->get());
         t = new GLTile(filename);
 
 		if (t->get_n_parts()==0) {
@@ -254,7 +254,7 @@
         n->m_name = new Symbol(name);
         m_hash_smooth[h].Add(n);
 
-        sprintf(filename, "graphics/%s.nfo", name->get
+        sprintf(filename, "/usr/share/goonies/graphics/%s.nfo", name->get
                 ());
         {
             float fx = 1.0f, fy = 1.0f;
diff -Nur goonies-1.4.1528-old/src/TheGooniesApp.cpp goonies-1.4.1528/src/TheGooniesApp.cpp
--- goonies-1.4.1528-old/src/TheGooniesApp.cpp	2009-07-22 15:05:14.625530151 +0300
+++ goonies-1.4.1528/src/TheGooniesApp.cpp	2009-07-22 15:06:15.020533580 +0300
@@ -136,9 +136,9 @@
 
 //    font_extract("font", "graphics/font.png", 10 + 13 + 13 + 10 + 14 + 8, "1234567890abcdefghijklmnopqrstuvwxyzXXXXXXXXXX-./:\"#$%!?:;.,'`[]{|}ñ");
 //    font_extract("font_hl", "graphics/font_highlighted.png", 10 + 13 + 13 + 10 + 14 + 8, "1234567890abcdefghijklmnopqrstuvwxyzXXXXXXXXXX-./:\"#$%!?:;.,'`[]{|}ñ");
-    font_extract("font", "graphics/font.png", 10 + 13 + 13 + 10 + 14 + 8, "1234567890abcdefghijklmnopqrstuvwxyz-!?:;.,'/\"#%[]{|}ñ");
-    font_extract("font_hl", "graphics/font_highlighted.png", 10 + 13 + 13 + 10 + 14 + 8, "1234567890abcdefghijklmnopqrstuvwxyz-!?:;.,'/\"#%[]{|}ñ");
-    font_extract("font_hud", "graphics/font_hud.png", 11, "0123456789-");
+    font_extract("font", "/usr/share/goonies/graphics/font.png", 10 + 13 + 13 + 10 + 14 + 8, "1234567890abcdefghijklmnopqrstuvwxyz-!?:;.,'/\"#%[]{|}ñ");
+    font_extract("font_hl", "/usr/share/goonies/graphics/font_highlighted.png", 10 + 13 + 13 + 10 + 14 + 8, "1234567890abcdefghijklmnopqrstuvwxyz-!?:;.,'/\"#%[]{|}ñ");
+    font_extract("font_hud", "/usr/share/goonies/graphics/font_hud.png", 11, "0123456789-");
 
     m_test_game = 0;
     load_configuration();
