diff -Nur goonies-1.4.1528-old/src/TheGoonies.cpp goonies-1.4.1528/src/TheGoonies.cpp
--- goonies-1.4.1528-old/src/TheGoonies.cpp	2009-07-22 14:34:44.452530361 +0300
+++ goonies-1.4.1528/src/TheGoonies.cpp	2009-07-22 14:35:55.812530250 +0300
@@ -95,7 +95,7 @@
     // Load the room/music table:
     {
         FILE *fp;
-        fp = fopen("maps/musics.txt", "r");
+        fp = fopen("/usr/share/goonies/maps/musics.txt", "r");
         if (fp != 0)
         {
             int n;
@@ -123,7 +123,7 @@
     // Load the room/fratelli table:
     {
         FILE *fp;
-        fp = fopen("maps/fratellies.txt", "r");
+        fp = fopen("/usr/share/goonies/maps/fratellies.txt", "r");
         if (fp != 0)
         {
             int n;
@@ -226,7 +226,7 @@
     // Load the room/fratelli table:
     {
         FILE *fp;
-        fp = fopen("maps/fratellies.txt", "r");
+        fp = fopen("/usr/share/goonies/maps/fratellies.txt", "r");
         if (fp != 0)
         {
             int n;
@@ -277,7 +277,7 @@
 
 void TheGoonies::map_filename(char *fn, int level, int roomx, int roomy)
 {
-    sprintf(fn, "maps/level%i-%i-%i.gmp", level, roomx, roomy);
+    sprintf(fn, "/usr/share/goonies/maps/level%i-%i-%i.gmp", level, roomx, roomy);
 }
 
 bool TheGoonies::cycle(VirtualController *v, GLTManager *GLTM, SFXManager *SFXM, MusicManager *MusicM)
