Index: omniORB-4.1.4/src/services/mklib/dir.mk
===================================================================
--- omniORB-4.1.4.orig/src/services/mklib/dir.mk
+++ omniORB-4.1.4/src/services/mklib/dir.mk
@@ -121,8 +121,8 @@ imps := $(patsubst $(DLLDebugSearchPatte
 dynimps := $(skshared) $(patsubst $(DLLDebugSearchPattern),$(DLLNoDebugSearchPattern), \
          $(OMNIORB_LIB))
 else
-imps := $(OMNIORB_LIB_NODYN)
-dynimps := $(OMNIORB_LIB)
+imps := -lomniDynamic$(OMNIORB_MAJOR_VERSION) $(OMNIORB_LIB_NODYN)
+dynimps := $(skshared) -lomniConnectionMgmt$(OMNIORB_MAJOR_VERSION) $(OMNIORB_LIB)
 endif
 
 ifdef AIX
