--- src/lib/omniORB/orbcore/ssl/sslContext.cc.ssl	2010-04-20 06:19:01.000000000 +0200
+++ src/lib/omniORB/orbcore/ssl/sslContext.cc	2010-04-20 06:20:49.000000000 +0200
@@ -199,7 +199,11 @@
 /////////////////////////////////////////////////////////////////////////
 SSL_METHOD*
 sslContext::set_method() {
+#if OPENSSL_VERSION_NUMBER >= 0x1000000fL
+  return (SSL_METHOD*) SSLv23_method();
+#else
   return SSLv23_method();
+#endif
 }
 
 /////////////////////////////////////////////////////////////////////////
