Index: NetworkManager-openconnect-0.8.4/po/POTFILES.in
===================================================================
--- NetworkManager-openconnect-0.8.4.orig/po/POTFILES.in
+++ NetworkManager-openconnect-0.8.4/po/POTFILES.in
@@ -3,5 +3,5 @@
 auth-dialog/main.c
 properties/auth-helpers.c
 properties/nm-openconnect.c
-properties/nm-openconnect-dialog.ui
+[type: gettext/glade]properties/nm-openconnect-dialog.ui
 src/nm-openconnect-service.c
