diff -Nur abicheck-1.2-old/abicheck.pl abicheck-1.2/abicheck.pl
--- abicheck-1.2-old/abicheck.pl	2008-10-06 02:43:41.000000000 +0300
+++ abicheck-1.2/abicheck.pl	2008-10-06 02:44:26.000000000 +0300
@@ -1,6 +1,4 @@
-  #!/bin/sh -- # A comment mentioning perl.
-eval 'exec perl -S $0 ${1+"$@"}'
-	if 0;
+#!/usr/bin/perl
 #
 # Copyright (c) 2001-2003 by Sun Microsystems, Inc.  All rights reserved.
 # Use is subject to license terms.
