--- libsyncml/parser/sml_wbxml_internals.h.OSmanOS	2012-11-18 21:13:38.000000000 +0200
+++ libsyncml/parser/sml_wbxml_internals.h		2012-11-18 21:14:39.338618933 +0200
@@ -21,7 +21,7 @@
 #ifndef _SML_WBXML_INTERNALS_H_
 #define _SML_WBXML_INTERNALS_H_

-#include <wbxml.h>
+#include <libwbxml-1.0/wbxml/wbxml.h>

 SmlBool smlWbxmlConvertTo(WBXMLConvXML2WBXMLParams *params, const char *input, char **output, unsigned int *outputLen, SmlError** error);
 SmlBool smlWbxmlConvertFrom(WBXMLConvWBXML2XMLParams *params, const char *input, unsigned int inputLen, char **output, SmlError** error);
