Index: api/comar/service.py
===================================================================
--- api/comar/service.py	(revision 33513)
+++ api/comar/service.py	(revision 33514)
@@ -113,6 +113,18 @@
         except:
             pass
 
+def stopDependencies(*services):
+    """Stop dependencies.
+
+    Arguments are service names.
+    """
+    for service in services:
+        try:
+            call(service, "System.Service", "stop")
+        except:
+            pass
+
+
 # Service control utilities
 
 def _getPid(pidfile):
