diff -Naur comar-api-2.4.9.orig/comar/network.py comar-api-2.4.9/comar/network.py
--- comar-api-2.4.9.orig/comar/network.py	2014-08-19 20:10:23.926869379 +0200
+++ comar-api-2.4.9/comar/network.py	2014-08-19 20:11:42.129866522 +0200
@@ -15,8 +15,8 @@
 import os
 import subprocess
 
-from pardus import iniutils
-from pardus import netutils
+from pisilinux import iniutils
+from pisilinux import netutils
 
 from comar.service import startService, stopService, loadConfig
 
diff -Naur comar-api-2.4.9.orig/setup.py comar-api-2.4.9/setup.py
--- comar-api-2.4.9.orig/setup.py	2010-11-11 14:29:05.000000000 +0100
+++ comar-api-2.4.9/setup.py	2014-08-19 20:12:26.807864889 +0200
@@ -50,9 +50,9 @@
 class Install(install):
     def finalize_options(self):
         # NOTE: for Pardus distribution
-        if os.path.exists("/etc/pardus-release"):
-            self.install_platlib = '$base/lib/pardus'
-            self.install_purelib = '$base/lib/pardus'
+        if os.path.exists("/etc/pisilinux-release"):
+            self.install_platlib = '$base/lib/pisilinux'
+            self.install_purelib = '$base/lib/pisilinux'
         install.finalize_options(self)
     
     def run(self):
