Index: dev86-0.16.17/bcc/bcc.c
===================================================================
--- dev86-0.16.17.orig/bcc/bcc.c
+++ dev86-0.16.17/bcc/bcc.c
@@ -226,12 +226,21 @@ char ** argv;
 
    } else {
       /* Relative paths to normal PREFIX directory */
+#ifdef __x86_64__
+      default_include = build_libpath("-I", "/lib64/bcc/include", "");
+      default_libdir  = build_libpath("-L", "/lib64/bcc", libdir_suffix);
+      optim_rules     = build_libpath("-d", "/lib64/bcc", libdir_suffix);
+
+      build_prefix("/lib64/bcc", libdir_suffix, "");
+      build_prefix("/lib64/bcc", "", "");
+#else
       default_include = build_libpath("-I", "/lib/bcc/include", "");
       default_libdir  = build_libpath("-L", "/lib/bcc", libdir_suffix);
       optim_rules     = build_libpath("-d", "/lib/bcc", libdir_suffix);
 
       build_prefix("/lib/bcc", libdir_suffix, "");
       build_prefix("/lib/bcc", "", "");
+#endif
    }
 
    build_prefix("/bin", "", "");
