diff -Nuar e-module-cpu-0.0.1_55225.orig//module.desktop.in Masaüstü/e-module-cpu-0.0.1_55225//module.desktop.in
--- e-module-cpu-0.0.1_55225.orig//module.desktop.in	2011-02-23 14:13:07.046544149 +0200
+++ e-module-cpu-0.0.1_55225//module.desktop.in	2011-02-23 14:24:33.724682653 +0200
@@ -19,3 +19,4 @@
 Comment[fr]=Affiche l'utilisation du CPU.
 Comment[el]=Για την παρατήρηση της δραστηριότητας του επεξεργαστή.
 Comment[pt]=Utilizado para vigiar a utilização do processador
+Comment[tr]=Cpu kullanımını gözlemlemek için kullanılır.
