diff -Nuar librsvg-2.39.0.orig/configure.in librsvg-2.39.0/configure.in
--- librsvg-2.39.0.orig/configure.in	2013-08-16 15:45:23.000000000 +0300
+++ librsvg-2.39.0/configure.in	2013-10-10 02:45:06.365034233 +0300
@@ -284,9 +284,6 @@
 
 GOBJECT_INTROSPECTION_CHECK([0.10.8])
 
-# Vala bindings
-VAPIGEN_CHECK([0.17.1.26],,,[no])
-
 dnl ===========================================================================
 
 m4_copy([AC_DEFUN],[glib_DEFUN])
diff -Nuar librsvg-2.39.0.orig/Makefile.am librsvg-2.39.0/Makefile.am
--- librsvg-2.39.0.orig/Makefile.am	2012-08-30 21:04:49.000000000 +0300
+++ librsvg-2.39.0/Makefile.am	2013-10-10 02:48:51.022040249 +0300
@@ -252,24 +252,6 @@
 
 CLEANFILES += $(nodist_gir_DATA) $(nodist_typelibs_DATA)
 
-if ENABLE_VAPIGEN
-include $(VAPIGEN_MAKEFILE)
-
-librsvg-$(RSVG_API_VERSION).vapi: Rsvg-$(RSVG_API_VERSION).gir
-
-VAPIGEN_VAPIS = librsvg-$(RSVG_API_VERSION).vapi
-
-librsvg_@RSVG_API_VERSION_U@_vapi_DEPS = gio-2.0 cairo
-librsvg_@RSVG_API_VERSION_U@_vapi_METADATADIRS = $(srcdir)
-librsvg_@RSVG_API_VERSION_U@_vapi_FILES = Rsvg-$(RSVG_API_VERSION).gir Rsvg-$(RSVG_API_VERSION)-custom.vala
-
-vapidir = $(datadir)/vala/vapi
-vapi_DATA = $(VAPIGEN_VAPIS)
-
-CLEANFILES += $(VAPIGEN_VAPIS)
-
-endif # ENABLE_VAPIGEN
-
 endif # HAVE_INTROSPECTION
 
 # ChangeLog generation
diff -Nuar librsvg-2.39.0.orig/Makefile.in librsvg-2.39.0/Makefile.in
--- librsvg-2.39.0.orig/Makefile.in	2013-08-16 15:45:57.000000000 +0300
+++ librsvg-2.39.0/Makefile.in	2013-10-10 02:47:52.009038669 +0300
@@ -86,7 +86,6 @@
 @HAVE_GTK_3_TRUE@am__append_2 = rsvg-view-3
 @OS_WIN32_TRUE@am__append_3 = -mwindows
 @HAVE_INTROSPECTION_TRUE@am__append_4 = $(nodist_gir_DATA) $(nodist_typelibs_DATA)
-@ENABLE_VAPIGEN_TRUE@@HAVE_INTROSPECTION_TRUE@am__append_5 = $(VAPIGEN_VAPIS)
 subdir = .
 DIST_COMMON = $(srcdir)/Makefile.in $(srcdir)/Makefile.am \
 	$(top_srcdir)/configure $(am__configure_deps) \
@@ -139,7 +138,6 @@
 am__installdirs = "$(DESTDIR)$(libdir)" "$(DESTDIR)$(bindir)" \
 	"$(DESTDIR)$(man1dir)" "$(DESTDIR)$(girdir)" \
 	"$(DESTDIR)$(typelibsdir)" "$(DESTDIR)$(pkgconfigdir)" \
-	"$(DESTDIR)$(vapidir)" "$(DESTDIR)$(librsvgincdir)"
 LTLIBRARIES = $(lib_LTLIBRARIES)
 am__DEPENDENCIES_1 =
 librsvg_@RSVG_API_MAJOR_VERSION@_la_DEPENDENCIES =  \
@@ -259,8 +257,7 @@
 man1dir = $(mandir)/man1
 NROFF = nroff
 MANS = $(dist_man_MANS)
-DATA = $(nodist_gir_DATA) $(nodist_typelibs_DATA) $(pkgconfig_DATA) \
-	$(vapi_DATA)
+DATA = $(nodist_gir_DATA) $(nodist_typelibs_DATA) $(pkgconfig_DATA)
 HEADERS = $(librsvginc_HEADERS)
 RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
   distclean-recursive maintainer-clean-recursive
@@ -447,9 +444,6 @@
 SET_MAKE = @SET_MAKE@
 SHELL = @SHELL@
 STRIP = @STRIP@
-VAPIGEN = @VAPIGEN@
-VAPIGEN_MAKEFILE = @VAPIGEN_MAKEFILE@
-VAPIGEN_VAPIDIR = @VAPIGEN_VAPIDIR@
 VERSION = @VERSION@
 abs_builddir = @abs_builddir@
 abs_srcdir = @abs_srcdir@
@@ -696,12 +690,6 @@
 @HAVE_INTROSPECTION_TRUE@nodist_gir_DATA = $(INTROSPECTION_GIRS)
 @HAVE_INTROSPECTION_TRUE@typelibsdir = $(libdir)/girepository-1.0
 @HAVE_INTROSPECTION_TRUE@nodist_typelibs_DATA = $(INTROSPECTION_GIRS:.gir=.typelib)
-@ENABLE_VAPIGEN_TRUE@@HAVE_INTROSPECTION_TRUE@VAPIGEN_VAPIS = librsvg-$(RSVG_API_VERSION).vapi
-@ENABLE_VAPIGEN_TRUE@@HAVE_INTROSPECTION_TRUE@librsvg_@RSVG_API_VERSION_U@_vapi_DEPS = gio-2.0 cairo
-@ENABLE_VAPIGEN_TRUE@@HAVE_INTROSPECTION_TRUE@librsvg_@RSVG_API_VERSION_U@_vapi_METADATADIRS = $(srcdir)
-@ENABLE_VAPIGEN_TRUE@@HAVE_INTROSPECTION_TRUE@librsvg_@RSVG_API_VERSION_U@_vapi_FILES = Rsvg-$(RSVG_API_VERSION).gir Rsvg-$(RSVG_API_VERSION)-custom.vala
-@ENABLE_VAPIGEN_TRUE@@HAVE_INTROSPECTION_TRUE@vapidir = $(datadir)/vala/vapi
-@ENABLE_VAPIGEN_TRUE@@HAVE_INTROSPECTION_TRUE@vapi_DATA = $(VAPIGEN_VAPIS)
 all: $(BUILT_SOURCES) config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
@@ -1255,27 +1243,6 @@
 	@list='$(pkgconfig_DATA)'; test -n "$(pkgconfigdir)" || list=; \
 	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
 	dir='$(DESTDIR)$(pkgconfigdir)'; $(am__uninstall_files_from_dir)
-install-vapiDATA: $(vapi_DATA)
-	@$(NORMAL_INSTALL)
-	@list='$(vapi_DATA)'; test -n "$(vapidir)" || list=; \
-	if test -n "$$list"; then \
-	  echo " $(MKDIR_P) '$(DESTDIR)$(vapidir)'"; \
-	  $(MKDIR_P) "$(DESTDIR)$(vapidir)" || exit 1; \
-	fi; \
-	for p in $$list; do \
-	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
-	  echo "$$d$$p"; \
-	done | $(am__base_list) | \
-	while read files; do \
-	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(vapidir)'"; \
-	  $(INSTALL_DATA) $$files "$(DESTDIR)$(vapidir)" || exit $$?; \
-	done
-
-uninstall-vapiDATA:
-	@$(NORMAL_UNINSTALL)
-	@list='$(vapi_DATA)'; test -n "$(vapidir)" || list=; \
-	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
-	dir='$(DESTDIR)$(vapidir)'; $(am__uninstall_files_from_dir)
 install-librsvgincHEADERS: $(librsvginc_HEADERS)
 	@$(NORMAL_INSTALL)
 	@list='$(librsvginc_HEADERS)'; test -n "$(librsvgincdir)" || list=; \
@@ -1593,7 +1560,7 @@
 
 installdirs: installdirs-recursive
 installdirs-am:
-	for dir in "$(DESTDIR)$(libdir)" "$(DESTDIR)$(bindir)" "$(DESTDIR)$(man1dir)" "$(DESTDIR)$(girdir)" "$(DESTDIR)$(typelibsdir)" "$(DESTDIR)$(pkgconfigdir)" "$(DESTDIR)$(vapidir)" "$(DESTDIR)$(librsvgincdir)"; do \
+	for dir in "$(DESTDIR)$(libdir)" "$(DESTDIR)$(bindir)" "$(DESTDIR)$(man1dir)" "$(DESTDIR)$(girdir)" "$(DESTDIR)$(typelibsdir)" "$(DESTDIR)$(pkgconfigdir)" "$(DESTDIR)$(librsvgincdir)"; do \
 	  test -z "$$dir" || $(MKDIR_P) "$$dir"; \
 	done
 install: $(BUILT_SOURCES)
@@ -1655,7 +1622,7 @@
 
 install-data-am: install-librsvgincHEADERS install-man \
 	install-nodist_girDATA install-nodist_typelibsDATA \
-	install-pkgconfigDATA install-vapiDATA
+	install-pkgconfigDATA 
 
 install-dvi: install-dvi-recursive
 
@@ -1706,7 +1673,7 @@
 uninstall-am: uninstall-binPROGRAMS uninstall-libLTLIBRARIES \
 	uninstall-librsvgincHEADERS uninstall-man \
 	uninstall-nodist_girDATA uninstall-nodist_typelibsDATA \
-	uninstall-pkgconfigDATA uninstall-vapiDATA
+	uninstall-pkgconfigDATA 
 
 uninstall-man: uninstall-man1
 
@@ -1728,15 +1695,14 @@
 	install-librsvgincHEADERS install-man install-man1 \
 	install-nodist_girDATA install-nodist_typelibsDATA install-pdf \
 	install-pdf-am install-pkgconfigDATA install-ps install-ps-am \
-	install-strip install-vapiDATA installcheck installcheck-am \
+	install-strip installcheck installcheck-am \
 	installdirs installdirs-am maintainer-clean \
 	maintainer-clean-generic mostlyclean mostlyclean-compile \
 	mostlyclean-generic mostlyclean-libtool pdf pdf-am ps ps-am \
 	tags tags-am uninstall uninstall-am uninstall-binPROGRAMS \
 	uninstall-libLTLIBRARIES uninstall-librsvgincHEADERS \
 	uninstall-man uninstall-man1 uninstall-nodist_girDATA \
-	uninstall-nodist_typelibsDATA uninstall-pkgconfigDATA \
-	uninstall-vapiDATA
+	uninstall-nodist_typelibsDATA uninstall-pkgconfigDATA 
 
 
 librsvg-enum-types.h: s-enum-types-h
@@ -1772,9 +1738,6 @@
 
 @HAVE_INTROSPECTION_TRUE@Rsvg-@RSVG_API_VERSION@.gir: librsvg-@RSVG_API_MAJOR_VERSION@.la
 
-@ENABLE_VAPIGEN_TRUE@@HAVE_INTROSPECTION_TRUE@include $(VAPIGEN_MAKEFILE)
-
-@ENABLE_VAPIGEN_TRUE@@HAVE_INTROSPECTION_TRUE@librsvg-$(RSVG_API_VERSION).vapi: Rsvg-$(RSVG_API_VERSION).gir
 
 # ChangeLog generation
 
