diff -up librsvg-2.37.0/configure.in.vala librsvg-2.37.0/configure.in
--- librsvg-2.37.0/configure.in.vala	2013-01-15 21:07:11.264576371 -0500
+++ librsvg-2.37.0/configure.in	2013-01-15 21:07:17.638576143 -0500
@@ -285,7 +285,7 @@ fi
 GOBJECT_INTROSPECTION_CHECK([0.10.8])
 
 # Vala bindings
-VAPIGEN_CHECK([0.17.1.26],,,[no])
+VAPIGEN_CHECK([0.18.1],,,[auto])
 
 dnl ===========================================================================
 
diff -up librsvg-2.37.0/Rsvg-2.0-custom.vala librsvg-2.37.0/Rsvg-2.0-custom
