--- libunwind-1.0.1/src/Makefile.am.orig	2011-09-15 13:57:09.000000000 +0200
+++ libunwind-1.0.1/src/Makefile.am	2011-09-15 13:57:44.000000000 +0200
@@ -8,7 +8,7 @@ COMMON_SO_LDFLAGS =	-XCClinker -nostartf
 lib_LIBRARIES =
 lib_LTLIBRARIES =
 if !REMOTE_ONLY
-lib_LIBRARIES += libunwind-ptrace.a
+lib_LTLIBRARIES += libunwind-ptrace.la
 lib_LTLIBRARIES += libunwind.la
 endif
 
@@ -16,7 +16,7 @@ noinst_HEADERS =
 noinst_LTLIBRARIES =
 
 ### libunwind-ptrace:
-libunwind_ptrace_a_SOURCES =						  \
+libunwind_ptrace_la_SOURCES =						  \
 	ptrace/_UPT_elf.c						  \
 	ptrace/_UPT_accessors.c ptrace/_UPT_access_fpreg.c		  \
 	ptrace/_UPT_access_mem.c ptrace/_UPT_access_reg.c		  \
--- libunwind-20110424git1e10c293-orig/tests/Makefile.am	2011-05-09 15:46:44.000000000 +0200
+++ libunwind-20110424git1e10c293/tests/Makefile.am	2011-05-09 15:52:46.000000000 +0200
@@ -108,7 +108,7 @@ Gtest_trace_SOURCES = Gtest-trace.c iden
 Ltest_trace_SOURCES = Ltest-trace.c ident.c
 
 LIBUNWIND = $(top_builddir)/src/libunwind-$(arch).la
-LIBUNWIND_ptrace = $(top_builddir)/src/libunwind-ptrace.a
+LIBUNWIND_ptrace = $(top_builddir)/src/libunwind-ptrace.la
 
 if USE_ELF32
 LIBUNWIND_ELF = $(top_builddir)/src/libunwind-elf32.la
