Index: gnucap-2009-12-07/doc/Makefile.am
===================================================================
--- gnucap-2009-12-07.orig/doc/Makefile.am
+++ gnucap-2009-12-07/doc/Makefile.am
@@ -3,7 +3,7 @@
 
 dist_man_MANS=	gnucap.1 gnucap-ibis.1
 
-pkgdocsdir= ${pkgdatadir}/doc
+pkgdocsdir= ${docdir}
 dist_pkgdocs_DATA= \
 	acs-tutorial COPYING history INSTALL relnotes.029 \
 	relnotes.030 relnotes.031 relnotes.032 relnotes.033 relnotes.034 whatisit 
Index: gnucap-2009-12-07/examples/Makefile.am
===================================================================
--- gnucap-2009-12-07.orig/examples/Makefile.am
+++ gnucap-2009-12-07/examples/Makefile.am
@@ -1,7 +1,7 @@
 ## $Id$
 ##
 
-pkgexamplesdir=	${pkgdatadir}/examples
+pkgexamplesdir=	${docdir}/examples
 dist_pkgexamples_DATA= ${EXFILES}
 
 ##EXFILES= *.c *.doc *.ckt runall runall.out
