Index: gnucap-2009-12-07/modelgen/ap.h
===================================================================
--- gnucap-2009-12-07.orig/modelgen/ap.h
+++ gnucap-2009-12-07/modelgen/ap.h
@@ -22,6 +22,7 @@
  * stuff for the "ap" family of parsing functions
  */
 //testing=script,sparse 2006.07.17
+#include <cstring>
 #ifndef AP_H
 #define AP_H
 #include "md.h"
Index: gnucap-2009-12-07/modelgen/l_lib.h
===================================================================
--- gnucap-2009-12-07.orig/modelgen/l_lib.h
+++ gnucap-2009-12-07/modelgen/l_lib.h
@@ -20,6 +20,7 @@
  * 02110-1301, USA.
  */
 //testing=script 2006.07.13
+#include <cstring>
 #ifndef L_LIB_H
 #define L_LIB_H
 #include "md.h"
Index: gnucap-2009-12-07/src/ap.h
===================================================================
--- gnucap-2009-12-07.orig/src/ap.h
+++ gnucap-2009-12-07/src/ap.h
@@ -22,6 +22,7 @@
  * stuff for the "ap" family of parsing functions
  */
 //testing=script,sparse 2006.07.17
+#include <cstring>
 #ifndef AP_H
 #define AP_H
 #include "md.h"
Index: gnucap-2009-12-07/src/e_base.h
===================================================================
--- gnucap-2009-12-07.orig/src/e_base.h
+++ gnucap-2009-12-07/src/e_base.h
@@ -22,6 +22,7 @@
  * real base for anything to do with a circuit
  */
 //testing=obsolete
+#include <cstring>
 #ifndef E_BASE_H
 #define E_BASE_H
 #include "md.h"
Index: gnucap-2009-12-07/src/io_findf.cc
===================================================================
--- gnucap-2009-12-07.orig/src/io_findf.cc
+++ gnucap-2009-12-07/src/io_findf.cc
@@ -32,6 +32,7 @@
  * PATHSEP, ENDDIR are system dependent, defined in md.h
  */
 //testing=script,sparse 2006.07.17
+#include <cstring>
 #include "l_lib.h"
 /*--------------------------------------------------------------------------*/
 std::string findfile(const std::string& filename, const std::string& path,
Index: gnucap-2009-12-07/src/l_ftos.cc
===================================================================
--- gnucap-2009-12-07.orig/src/l_ftos.cc
+++ gnucap-2009-12-07/src/l_ftos.cc
@@ -37,6 +37,7 @@
  * after that, the space is overwritten, every POOLSIZE calls
  */
 //testing=script 2005.10.11
+#include <cstring>
 #include "l_lib.h"
 #include "constant.h"
 /*--------------------------------------------------------------------------*/
Index: gnucap-2009-12-07/src/l_lib.h
===================================================================
--- gnucap-2009-12-07.orig/src/l_lib.h
+++ gnucap-2009-12-07/src/l_lib.h
@@ -20,6 +20,7 @@
  * 02110-1301, USA.
  */
 //testing=script 2006.07.13
+#include <cstring>
 #ifndef L_LIB_H
 #define L_LIB_H
 #include "md.h"
Index: gnucap-2009-12-07/src/l_timer.cc
===================================================================
--- gnucap-2009-12-07.orig/src/l_timer.cc
+++ gnucap-2009-12-07/src/l_timer.cc
@@ -22,6 +22,7 @@
  * Time a command, or whatever
  */
 //testing=script,sparse 2006.07.13
+#include <cstring>
 #include "l_timer.h"
 /*--------------------------------------------------------------------------*/
 //		TIMER::TIMER();
Index: gnucap-2009-12-07/modelgen/ap_construct.cc
===================================================================
--- gnucap-2009-12-07.orig/modelgen/ap_construct.cc
+++ gnucap-2009-12-07/modelgen/ap_construct.cc
@@ -22,6 +22,7 @@
  * construction, copy, etc.
  */
 //testing=script,sparse 2006.07.17
+#include <cstring>
 #include "u_opt.h"
 #include "ap.h"
 
Index: gnucap-2009-12-07/src/ap_construct.cc
===================================================================
--- gnucap-2009-12-07.orig/src/ap_construct.cc
+++ gnucap-2009-12-07/src/ap_construct.cc
@@ -22,6 +22,7 @@
  * construction, copy, etc.
  */
 //testing=script,sparse 2006.07.17
+#include <cstring>
 #include "u_opt.h"
 #include "ap.h"
 
Index: gnucap-2009-12-07/src/main.cc
===================================================================
--- gnucap-2009-12-07.orig/src/main.cc
+++ gnucap-2009-12-07/src/main.cc
@@ -23,6 +23,7 @@
  * it all starts here
  */
 //testing=script 2006.07.14
+#include <cstring>
 #include "e_cardlist.h"
 #include "u_lang.h"
 #include "ap.h"
