http://bugs.gentoo.org/219621

fix building with gcc-4.3

--- SDL_gfx-2.0.16/SDL_imageFilter.c
+++ SDL_gfx-2.0.16/SDL_imageFilter.c
@@ -84 +84 @@
-      ".L1010:                \n\t" "movq    (%%eax), %%mm1 \n\t"	/* load 8 bytes from Src1 into mm1 */
+      "1: movq (%%eax), %%mm1 \n\t"    	/* load 8 bytes from Src1 into mm1 */
@@ -90 +90 @@
-      "jnz             .L1010 \n\t"	/* check loop termination, proceed if required */
+      "jnz             1b     \n\t"     /* check loop termination, proceed if required */
@@ -161 +161 @@
-      ".L21011:                \n\t" 
+      "1:                      \n\t"
@@ -177 +177 @@
-      "jnz             .L21011 \n\t"	/* check loop termination, proceed if required */
+      "jnz                 1b \n\t"     /* check loop termination, proceed if required */
@@ -244 +244 @@
-      ".L1012:         \n\t" "movq    (%%eax), %%mm1 \n\t"	/* load 8 bytes from Src1 into mm1 */
+      "1: movq (%%eax), %%mm1 \n\t"     /* load 8 bytes from Src1 into mm1 */
@@ -250 +250 @@
-      "jnz .L1012    \n\t"	/* check loop termination, proceed if required */
+      "jnz 1b         \n\t"     /* check loop termination, proceed if required */
@@ -316 +316 @@
-      ".L1013:         \n\t" "movq    (%%eax), %%mm1 \n\t"	/* load 8 bytes from Src1 into mm1 */
+      "1: movq (%%eax), %%mm1 \n\t"     /* load 8 bytes from Src1 into mm1 */
@@ -325 +325 @@
-      "jnz .L1013    \n\t"	/* check loop termination, proceed if required */
+      "jnz 1b        \n\t"      /* check loop termination, proceed if required */
@@ -391 +391 @@
-      ".L1014:         \n\t" "movq    (%%eax), %%mm1 \n\t"	/* load 8 bytes from Src1 into mm1 */
+      "1: movq (%%eax), %%mm1 \n\t"     /* load 8 bytes from Src1 into mm1 */
@@ -415 +415 @@
-      "jnz .L1014    \n\t"	/* check loop termination, proceed if required */
+      "jnz 1b        \n\t"      /* check loop termination, proceed if required */
@@ -484 +484 @@
-      ".L10141:        \n\t" "mov  (%%edx), %%al \n\t"	/* load a byte from Src1 */
+      "1:mov  (%%edx), %%al \n\t"      /* load a byte from Src1 */
@@ -486 +486 @@
-       ".L10142:           \n\t" "mov %%al, (%%edi)  \n\t"	/* move a byte result to Dest */
+       "mov %%al, (%%edi)  \n\t"       /* move a byte result to Dest */
@@ -490 +490 @@
-      "jnz .L10141    \n\t"	/* check loop termination, proceed if required */
+      "jnz 1b         \n\t"     /* check loop termination, proceed if required */
@@ -560 +560 @@
-      ".L1015:         \n\t" "movq    (%%eax), %%mm1 \n\t"	/* load 8 bytes from Src1 into mm1 */
+      "1: movq (%%eax), %%mm1 \n\t"	/* load 8 bytes from Src1 into mm1 */
@@ -577 +577 @@
-      "jnz .L1015    \n\t"	/* check loop termination, proceed if required */
+      "jnz 1b        \n\t"	/* check loop termination, proceed if required */
@@ -645 +645 @@
-      ".L1016:         \n\t" "movq    (%%eax), %%mm1 \n\t"	/* load 8 bytes from Src1 into mm1 */
+      "1: movq (%%eax), %%mm1 \n\t"	/* load 8 bytes from Src1 into mm1 */
@@ -664 +664 @@
-      "jnz .L1016    \n\t"	/* check loop termination, proceed if required */
+      "jnz 1b        \n\t"	/* check loop termination, proceed if required */
@@ -731 +731 @@
-      ".L1017:         \n\t" "movq    (%%eax), %%mm1 \n\t"	/* load 8 bytes from Src1 into mm1 */
+      "1: movq (%%eax), %%mm1 \n\t"	/* load 8 bytes from Src1 into mm1 */
@@ -737 +737 @@
-      "jnz .L1017    \n\t"	/* check loop termination, proceed if required */
+      "jnz 1b        \n\t"	/* check loop termination, proceed if required */
@@ -803 +803 @@
-      ".L91017:        \n\t" "movq    (%%eax), %%mm1 \n\t"	/* load 8 bytes from Src1 into mm1 */
+      "1: movq (%%eax), %%mm1 \n\t"	/* load 8 bytes from Src1 into mm1 */
@@ -809 +809 @@
-      "jnz .L91017   \n\t"	/* check loop termination, proceed if required */
+      "jnz 1b        \n\t"	/* check loop termination, proceed if required */
@@ -871 +871 @@
-      ".L10191:      \n\t" "mov  (%%esi), %%bl  \n\t"	/* load a byte from Src2 */
+      "1: mov (%%esi), %%bl  \n\t"	/* load a byte from Src2 */
@@ -873,2 +873,2 @@
-      "jnz .L10192         \n\t" "movb  $255, (%%edi) \n\t"	/* division by zero = 255 !!! */
-      "jmp  .L10193        \n\t" ".L10192:            \n\t" "xor   %%ah, %%ah    \n\t"	/* prepare AX, zero AH register */
+      "jnz 2f              \n\t" "movb  $255, (%%edi) \n\t"	/* division by zero = 255 !!! */
+      "jmp 3f              \n\t" "2:                  \n\t" "xor   %%ah, %%ah    \n\t"	/* prepare AX, zero AH register */
@@ -878 +878 @@
-       ".L10193:            \n\t" "inc %%edx \n\t"	/* increment Src1, Src2, Dest */
+      "3: inc %%edx        \n\t"	/* increment Src1, Src2, Dest */
@@ -881 +881 @@
-      "jnz .L10191  \n\t"	/* check loop termination, proceed if required */
+      "jnz 1b       \n\t"	/* check loop termination, proceed if required */
@@ -942 +942 @@
-      ".L91117:        \n\t" "movq    (%%eax), %%mm0 \n\t"	/* load 8 bytes from Src1 into mm1 */
+      "1: movq (%%eax), %%mm0 \n\t"	/* load 8 bytes from Src1 into mm1 */
@@ -947 +947 @@
-      "jnz .L91117   \n\t"	/* check loop termination, proceed if required */
+      "jnz 1b        \n\t"	/* check loop termination, proceed if required */
@@ -1015 +1015 @@
-      ".L1021:                \n\t" 
+      "1:                     \n\t" 
@@ -1022 +1022 @@
-      "jnz             .L1021 \n\t"	/* check loop termination, proceed if required */
+      "jnz                 1b \n\t"	/* check loop termination, proceed if required */
@@ -1094 +1094 @@
-      ".L11023:                \n\t" 
+      "1:                     \n\t" 
@@ -1101 +1101 @@
-      "jnz             .L11023 \n\t"	/* check loop termination, proceed if required */
+      "jnz                 1b \n\t"	/* check loop termination, proceed if required */
@@ -1189 +1189 @@
-      ".L1022:                \n\t" 
+      "1:                     \n\t" 
@@ -1199 +1199 @@
-      "jnz             .L1022 \n\t"	/* check loop termination, proceed if required */
+      "jnz                  1b \n\t"	/* check loop termination, proceed if required */
@@ -1278 +1278 @@
-      ".L1023:                \n\t" "movq    (%%eax), %%mm0 \n\t"	/* load 8 bytes from SrcDest into MM0 */
+      "1: movq (%%eax), %%mm0 \n\t"	/* load 8 bytes from SrcDest into MM0 */
@@ -1284 +1284 @@
-      "jnz             .L1023 \n\t"	/* check loop termination, proceed if required */
+      "jnz                 1b \n\t"	/* check loop termination, proceed if required */
@@ -1357 +1357 @@
-      ".L11024:                \n\t" "movq    (%%eax), %%mm0 \n\t"	/* load 8 bytes from SrcDest into MM0 */
+      "1: movq (%%eax), %%mm0 \n\t"	/* load 8 bytes from SrcDest into MM0 */
@@ -1363 +1363 @@
-      "jnz             .L11024 \n\t"	/* check loop termination, proceed if required */
+      "jnz                  1b \n\t"	/* check loop termination, proceed if required */
@@ -1440 +1440 @@
-       ".L10240:               \n\t"	/* ** Prepare proper bit-Mask in MM1 ** */
+       "1:                     \n\t"	/* ** Prepare proper bit-Mask in MM1 ** */
@@ -1445 +1445 @@
-      "jnz            .L10240 \n\t"	/* check loop termination, proceed if required */
+      "jnz                 1b \n\t"	/* check loop termination, proceed if required */
@@ -1452 +1452 @@
-      ".L10241:               \n\t" 
+      "2:                     \n\t" 
@@ -1461 +1461 @@
-      "jnz            .L10241 \n\t"	/* check loop termination, proceed if required */
+      "jnz                 2b \n\t"	/* check loop termination, proceed if required */
@@ -1530 +1530 @@
-      ".L13023:                \n\t" "movq    (%%eax), %%mm0 \n\t"	/* load 8 bytes from SrcDest into MM0 */
+      "1: movq (%%eax), %%mm0 \n\t"	/* load 8 bytes from SrcDest into MM0 */
@@ -1536 +1536 @@
-      "jnz             .L13023 \n\t"	/* check loop termination, proceed if required */
+      "jnz                 1b \n\t"	/* check loop termination, proceed if required */
@@ -1616,2 +1616,2 @@
-      "jg             .L10251 \n\t" ".align 16              \n\t"	/* 16 byte allignment of the loop entry */
-      ".L10250:               \n\t" "movq    (%%eax), %%mm3 \n\t"	/* load 8 bytes from Src1 into MM3 */
+      "jg                  2f \n\t" ".align 16              \n\t"	/* 16 byte allignment of the loop entry */
+      "1: movq (%%eax), %%mm3 \n\t"	/* load 8 bytes from Src1 into MM3 */
@@ -1628,3 +1628,3 @@
-      "jnz            .L10250 \n\t"	/* check loop termination, proceed if required */
-      "jmp            .L10252 \n\t" ".align 16              \n\t"	/* 16 byte allignment of the loop entry */
-      ".L10251:               \n\t" "movq    (%%eax), %%mm3 \n\t"	/* load 8 bytes from Src1 into MM3 */
+      "jnz                 1b \n\t"	/* check loop termination, proceed if required */
+      "jmp                 3f \n\t" ".align 16              \n\t"	/* 16 byte allignment of the loop entry */
+      "2: movq (%%eax), %%mm3 \n\t"	/* load 8 bytes from Src1 into MM3 */
@@ -1650,2 +1650,2 @@
-      "jnz            .L10251 \n\t"	/* check loop termination, proceed if required */
-       ".L10252:               \n\t" "emms                   \n\t"	/* exit MMX state */
+      "jnz                 2b \n\t"	/* check loop termination, proceed if required */
+       "3: emms               \n\t"	/* exit MMX state */
@@ -1730,2 +1730,2 @@
-       ".align 16              \n\t"	/* 16 byte allignment of the loop entry */
-      ".L1026:                \n\t" "movq    (%%eax), %%mm3 \n\t"	/* load 8 bytes from Src1 into MM3 */
+       ".align 16             \n\t"	/* 16 byte allignment of the loop entry */
+      "1: movq (%%eax), %%mm3 \n\t"	/* load 8 bytes from Src1 into MM3 */
@@ -1744 +1744 @@
-      "jnz             .L1026 \n\t"	/* check loop termination, proceed if required */
+      "jnz                 1b \n\t"	/* check loop termination, proceed if required */
@@ -1819 +1819 @@
-       ".L10270:               \n\t"	/* ** Prepare proper bit-Mask in MM1 ** */
+       "1:                     \n\t"	/* ** Prepare proper bit-Mask in MM1 ** */
@@ -1823 +1823 @@
-      "jnz            .L10270 \n\t"	/* check loop termination, proceed if required */
+      "jnz                 1b \n\t"	/* check loop termination, proceed if required */
@@ -1830 +1830 @@
-      ".L10271:               \n\t" "movq    (%%eax), %%mm0 \n\t"	/* load 8 bytes from Src1 into MM0 */
+      "2: movq (%%eax), %%mm0 \n\t"	/* load 8 bytes from Src1 into MM0 */
@@ -1837 +1837 @@
-      "jnz            .L10271 \n\t"	/* check loop termination, proceed if required */
+      "jnz                 2b \n\t"	/* check loop termination, proceed if required */
@@ -1905 +1905 @@
-      ".L12023:                \n\t" "movq    (%%eax), %%mm0 \n\t"	/* load 8 bytes from SrcDest into MM0 */
+      "1: movq (%%eax), %%mm0 \n\t"	/* load 8 bytes from SrcDest into MM0 */
@@ -1911 +1911 @@
-      "jnz             .L12023 \n\t"	/* check loop termination, proceed if required */
+      "jnz                 1b \n\t"	/* check loop termination, proceed if required */
@@ -1984,2 +1984,2 @@
-      "jg             .L10281 \n\t" ".align 16              \n\t"	/* 16 byte allignment of the loop entry */
-      ".L10280:               \n\t" "movq    (%%eax), %%mm3 \n\t"	/* load 8 bytes from Src1 into MM3 */
+      "jg                  2f \n\t" ".align 16              \n\t"	/* 16 byte allignment of the loop entry */
+      "1: movq (%%eax), %%mm3 \n\t"	/* load 8 bytes from Src1 into MM3 */
@@ -1996,3 +1996,3 @@
-      "jnz            .L10280 \n\t"	/* check loop termination, proceed if required */
-      "jmp            .L10282 \n\t" ".align 16              \n\t"	/* 16 byte allignment of the loop entry */
-      ".L10281:               \n\t" "movq    (%%eax), %%mm3 \n\t"	/* load 8 bytes from Src1 into MM3 */
+      "jnz                 1b \n\t"	/* check loop termination, proceed if required */
+      "jmp                 3f \n\t" ".align 16              \n\t"	/* 16 byte allignment of the loop entry */
+      "2: movq (%%eax), %%mm3 \n\t"	/* load 8 bytes from Src1 into MM3 */
@@ -2018,2 +2018,2 @@
-      "jnz            .L10281 \n\t"	/* check loop termination, proceed if required */
-       ".L10282:               \n\t" "emms                   \n\t"	/* exit MMX state */
+      "jnz                 2b \n\t"	/* check loop termination, proceed if required */
+       "3: emms                \n\t"	/* exit MMX state */
@@ -2098 +2098 @@
-      ".L1029:                \n\t" 
+      "1:                     \n\t" 
@@ -2106 +2106 @@
-      "jnz             .L1029 \n\t"	/* check loop termination, proceed if required */
+      "jnz                 1b \n\t"	/* check loop termination, proceed if required */
@@ -2189 +2189 @@
-      ".L1030:                \n\t" 
+      "1:                     \n\t" 
@@ -2198 +2198 @@
-      "jnz             .L1030 \n\t"	/* check loop termination, proceed if required */
+      "jnz                 1b \n\t"	/* check loop termination, proceed if required */
@@ -2266 +2266 @@
-      "jz             .L10311 \n\t"	/* check division by zero */
+      "jz                  1f \n\t"	/* check division by zero */
@@ -2269,2 +2269,2 @@
-      "jmp            .L10312 \n\t" ".L10311:               \n\t" "mov         $255, %%ax \n\t"	/* if div by zero, assume result max. byte value */
-       ".L10312:               \n\t"	/* ** Duplicate AX in 4 words of MM0 ** */
+      "jmp                 2f \n\t" "1:                     \n\t" "mov         $255, %%ax \n\t"	/* if div by zero, assume result max. byte value */
+       "2:                    \n\t"	/* ** Duplicate AX in 4 words of MM0 ** */
@@ -2299 +2299 @@
-      ".L1031:                \n\t" 
+      "1:                     \n\t" 
@@ -2324 +2324 @@
-      "jnz             .L1031 \n\t"	/* check loop termination, proceed if required */
+      "jnz                 1b \n\t"	/* check loop termination, proceed if required */
