diff -Nur pingus-0.7.1-old/src/globals.cpp pingus-0.7.1/src/globals.cpp
--- pingus-0.7.1-old/src/globals.cpp	2007-09-24 20:54:23.548174138 +0300
+++ pingus-0.7.1/src/globals.cpp	2007-09-24 20:55:38.913716836 +0300
@@ -34,8 +34,8 @@
 bool        drag_drop_scrolling             = false;
 int         tile_size                       = 32;
 
-int         screen_width                    = 800;
-int         screen_height                   = 600;
+int         screen_width                    = 1024;
+int         screen_height                   = 768;
 bool        draw_collision_map              = false;
 bool        swcursor_enabled                = false;
 bool        action_help                     = true;
@@ -53,7 +53,7 @@
 
 unsigned int pingus_debug_flags              = 0;
 
-bool        fullscreen_enabled              = false;
+bool        fullscreen_enabled              = true;
 
 bool        resize_enabled                  = false;
 
