--- SConstruct	2008-04-04 14:47:04.000000000 +0300
+++ SConstruct.orig	2008-04-04 14:46:44.000000000 +0300
@@ -363,8 +363,8 @@ def CheckSDLLib(context, sdllib):
    lib = "SDL_%s" % sdllib
    context.Message('Checking for %s...' % lib)
    text = """
-#include "SDL.h"
-#include "%s.h"
+#include <SDL.h>
+#include <%s.h>
 int main(int argc, char* argv[]) { return 0; }
 """ % lib
    context.AppendLIBS(lib)
pingus-0.7.2-orig/src/input/controller_description.o ve pingus-0.7.2/src/input/controller_description.o dosyaları birbirinden farklı
pingus-0.7.2-orig/src/input/controller.o ve pingus-0.7.2/src/input/controller.o dosyaları birbirinden farklı
pingus-0.7.2-orig/src/input/core_driver.o ve pingus-0.7.2/src/input/core_driver.o dosyaları birbirinden farklı
diff -pruN pingus-0.7.2-orig/src/input/evdev_device.cpp pingus-0.7.2/src/input/evdev_device.cpp
--- src/input/evdev_device.cpp.orig	2008-04-04 15:06:50.000000000 +0300
+++ src/input/evdev_device.cpp	2008-04-04 15:07:27.000000000 +0300
@@ -25,6 +25,7 @@
 #include <stdexcept>
 #include <iostream>
 #include <errno.h>
+#include <cstring>
 
 #include "evdev_device.hpp"
 
