--- Makefile.OSmanOS	2011-12-24 23:46:47.000000000 +0200
+++ Makefile	2013-05-26 23:31:41.818655430 +0300
@@ -15,7 +15,7 @@
 ##  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 DESTDIR = 
-PREFIX  = /usr/local
+PREFIX  = /usr
 DATADIR = $(PREFIX)/share/pingus
 MANDIR  = $(PREFIX)/share/man
 BINDIR  = $(PREFIX)/bin
@@ -33,13 +33,13 @@
 	rm -f pingus
 	scons -c
 
-install: install-exec install-data install-man
+install: install-exec install-man
 
 install-exec: build/pingus
 	install -d "$(DESTDIR)$(BINDIR)"
 
 	install -D build/pingus "$(DESTDIR)$(BINDIR)/pingus.bin"
-	echo -e "#!/bin/sh\nexec \"$(BINDIR)/pingus.bin\" --datadir \"$(DATADIR)\"" > "$(DESTDIR)$(BINDIR)/pingus"
+	echo -e "#!/bin/sh\nexec \"$(BINDIR)/pingus.bin\" --datadir \"$(DATADIR)\"" > "$(DESTDIR)$(DATADIR)"
 	chmod 755 "$(DESTDIR)$(BINDIR)/pingus"
 
 install-data:
@@ -68,6 +68,6 @@
 install-man:
 	install -D doc/man/pingus.6 "$(DESTDIR)$(MANDIR)/man1/pingus.6"
 
-.PHONY : clean install install-exec install-data install-man
+.PHONY : clean install install-exec install-man
 
 # EOF #
