--- Makefile~	2011-10-31 08:40:11.000000000 -0500
+++ Makefile	2011-10-31 08:40:34.000000000 -0500
@@ -39,7 +39,7 @@
 	install -d "$(DESTDIR)$(BINDIR)"
 
 	install -D build/pingus "$(DESTDIR)$(BINDIR)/pingus.bin"
-	echo "#!/bin/sh\nexec \"$(BINDIR)/pingus.bin\" --datadir \"$(DATADIR)\"" > "$(DESTDIR)$(BINDIR)/pingus"
+	echo -e "#!/bin/sh\nexec \"$(BINDIR)/pingus.bin\" --datadir \"$(DATADIR)\"" > "$(DESTDIR)$(BINDIR)/pingus"
 	chmod 755 "$(DESTDIR)$(BINDIR)/pingus"
 
 install-data:

