--- ./CMakeLists.txt	2012-06-26 02:48:57.000000000 +0300
+++ ./CMakeLists.txt	2012-12-24 17:59:01.465291356 +0200
@@ -158,7 +158,7 @@
   LIBRARY DESTINATION lib${LIB_SUFFIX}
   ARCHIVE DESTINATION lib${LIB_SUFFIX}
 )
-install (FILES
+install (FILES bsdqueue.h codelength.h connecthostport.h minisoap.h minissdpc.h miniupnpcstrings.h miniupnpctypes.h minixml.h portlistingparse.h receivedata.h
 	miniupnpc.h
   miniwget.h
   upnpcommands.h
