Index: kdenlive-0.8/src/kdenlive.desktop
===================================================================
--- kdenlive-0.8.orig/src/kdenlive.desktop
+++ kdenlive-0.8/src/kdenlive.desktop
@@ -4,6 +4,7 @@ Name=Kdenlive
 GenericName=Video Editor
 GenericName[de]=Video-Editor
 GenericName[it]=Editor Video
+GenericName[tr]=Video Düzenleyici
 GenericName[ru]=Видеоредактор
 GenericName[uk]=Редактор відео
 GenericName[es]=Editor de video
@@ -11,6 +12,7 @@ Comment=Nonlinear video editor for KDE
 Comment[de]=Nicht-linearer Video-Editor für KDE
 Comment[it]=Editor video non lineare per KDE
 Comment[ru]=Редактор нелинейного видеомонтажа для KDE
+Comment[tr]=KDE için Nonlineer video düzenleyici
 Comment[uk]=Нелінійний редактор відео для KDE
 Comment[es]=Editor no lineal de video para KDE
 Type=Application
