--- kross-interpreters/ruby/CMakeLists.txt	2013-03-20 07:23:12.323170891 +0100
+++ new/ruby/CMakeLists.txt	2013-03-25 14:05:41.357343483 +0100
@@ -1,11 +1,11 @@
 find_package(Ruby 1.9 QUIET)
 
-if( RUBY_VERSION VERSION_GREATER "1.10" )
-    message(STATUS "Found Ruby: ${RUBY_EXECUTABLE} (found version \"${RUBY_VERSION}\") but only 1.9.x supported")
-    set(RUBY_FOUND False)
-else()
+#if( RUBY_VERSION VERSION_GREATER "1.10" )
+#    message(STATUS "Found Ruby: ${RUBY_EXECUTABLE} (found version \"${RUBY_VERSION}\") but only 1.9.x supported")
+#    set(RUBY_FOUND False)
+#else()
     message(STATUS "Found Ruby: ${RUBY_EXECUTABLE} (found version \"${RUBY_VERSION}\")")
-endif()
+#endif()
 
 macro_log_feature(RUBY_FOUND
   "Ruby" "Ruby libraries" "http://www.ruby-lang.org" FALSE "1.9.x" "Needed to compile the Ruby bindings")
