diff -up pulseaudio-4.0-266-gf81e3/src/daemon/pulseaudio-kde.desktop.in.kde_autostart_phase pulseaudio-4.0-266-gf81e3/src/daemon/pulseaudio-kde.desktop.in
--- pulseaudio-4.0-266-gf81e3/src/daemon/pulseaudio-kde.desktop.in.kde_autostart_phase	2013-10-10 17:02:01.000000000 -0500
+++ pulseaudio-4.0-266-gf81e3/src/daemon/pulseaudio-kde.desktop.in	2013-12-07 13:15:24.034110082 -0600
@@ -9,3 +9,4 @@ Type=Application
 Categories=
 GenericName=
 OnlyShowIn=KDE;
+X-KDE-autostart-phase=1
diff -up pulseaudio-4.0-266-gf81e3/src/daemon/pulseaudio.desktop.in.kde_autostart_phase pulseaudio-4.0-266-gf81e3/src/daemon/pulseaudio.desktop.in
--- pulseaudio-4.0-266-gf81e3/src/daemon/pulseaudio.desktop.in.kde_autostart_phase	2013-10-10 17:02:01.000000000 -0500
+++ pulseaudio-4.0-266-gf81e3/src/daemon/pulseaudio.desktop.in	2013-12-07 13:15:17.579180709 -0600
@@ -9,3 +9,4 @@ Type=Application
 Categories=
 GenericName=
 X-GNOME-Autostart-Phase=Initialization
+X-KDE-autostart-phase=1
