From d075bfb8a5207f933a5a704becd7f64e8521dc28 Mon Sep 17 00:00:00 2001
From: zanee <zanee>
Date: Tue, 4 May 2010 18:47:08 +0000
Subject: [PATCH 5/5] Updating ChangeLog with relevant changes

Signed-off-by: Kamil Dudka <kdudka@redhat.com>
---
 ChangeLog |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 618654d..885c8b0 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,14 @@
+Version 7.19.2
+--------------
+
+        * Cleaned up website
+
+        * Fix pycurl.reset() (patch by <johansen at sun.com>).
+
+        * Fix install routine in setup.py where
+          certain platforms (Solaris, Mac OSX, etc)
+          would search for a static copy of libcurl (dbp)
+
 Version 7.19.1 [requires libcurl-7.19.0 or better]
 --------------
 
-- 
1.7.1

