diff -Nur matplotlib-0.98.0-old/setup.cfg matplotlib-0.98.0/setup.cfg
--- matplotlib-0.98.0-old/setup.cfg	1970-01-01 02:00:00.000000000 +0200
+++ matplotlib-0.98.0/setup.cfg	2008-06-08 03:45:41.000000000 +0300
@@ -0,0 +1,18 @@
+
+[provide_packages]
+## Date/timezone support:
+pytz = False
+dateutil = False
+
+## Experimental config package support:
+enthought.traits = False
+configobj = False
+
+[gui_support]
+gtk = True
+gtkagg = True
+tkagg = False
+wxagg = False
+
+
+
