diff -urNp xmlto-0.0.23-orig/format/fo/dvi xmlto-0.0.23/format/fo/dvi
--- xmlto-0.0.23-orig/format/fo/dvi	2009-09-18 12:06:23.000000000 +0200
+++ xmlto-0.0.23/format/fo/dvi	2009-10-13 14:08:46.000000000 +0200
@@ -13,7 +13,10 @@ post-process)
   fi
   # Work around stupid tetex bug with '_' in filenames
   # Also work around stupid tetex limitation with long lines (bug #101055)
-  sed -e "s,/>,\n/>,g" "$XSLT_PROCESSED" >tmp.fo
+  # and workaround passivetex limitation for chapter titles starting with L
+  # (#526273)
+  sed -e "s,/>,/>\n,g" \
+      -e "s,block>,&\n,g" "$XSLT_PROCESSED" >tmp.fo
   OUT=output
   TEXINPUTS="$(dirname "$INPUT_FILE")::$SEARCHPATH"
   export TEXINPUTS
diff -urNp xmlto-0.0.23-orig/format/fo/pdf xmlto-0.0.23/format/fo/pdf
--- xmlto-0.0.23-orig/format/fo/pdf	2009-09-18 12:06:23.000000000 +0200
+++ xmlto-0.0.23/format/fo/pdf	2009-10-13 14:09:08.000000000 +0200
@@ -15,7 +15,10 @@ DEFAULT|DBLATEX)
 	  fi
 	  # Work around stupid tetex bug with '_' in filenames
 	  # Also work around stupid tetex limitation with long lines (bug #101055)
-	  sed -e "s,/>,\n/>,g" "$XSLT_PROCESSED" >tmp.fo
+    # and workaround passivetex limitation for chapter titles starting with L
+    # (#526273)
+	  sed -e "s,/>,/>\n,g" \
+        -e "s,block>,&\n,g" "$XSLT_PROCESSED" >tmp.fo
 	  OUT=output
 	  TEXINPUTS="$(dirname "$INPUT_FILE")::$SEARCHPATH"
 	  export TEXINPUTS
diff -urNp xmlto-0.0.23-orig/format/fo/ps xmlto-0.0.23/format/fo/ps
--- xmlto-0.0.23-orig/format/fo/ps	2009-03-17 17:39:52.000000000 +0100
+++ xmlto-0.0.23/format/fo/ps	2009-10-13 14:01:39.000000000 +0200
@@ -10,7 +10,10 @@ DEFAULT|DBLATEX)
 	  fi
 	  # Work around stupid tetex bug with '_' in filenames
 	  # Also work around stupid tetex limitation with long lines (bug #101055)
-	  sed -e "s,/>,\n/>,g" "$XSLT_PROCESSED" >tmp.fo
+    # and workaround passivetex limitation for chapter titles starting with L
+    # (#526273)
+	  sed -e "s,/>,/>\n,g" \
+        -e "s,block>,&\n,g" "$XSLT_PROCESSED" >tmp.fo
 	  OUT=output
 	  TEXINPUTS="$(dirname "$INPUT_FILE")::$SEARCHPATH"
 	  export TEXINPUTS
