--- chromium-browser-42.0.2311.82/third_party/angle/src/angle.gyp~	2015-04-09 00:14:43.000000000 +0300
+++ chromium-browser-42.0.2311.82/third_party/angle/src/angle.gyp	2015-04-21 22:02:35.668680845 +0300
@@ -104,8 +104,7 @@
                         {
                             'action_name': 'Generate ANGLE Commit ID Header',
                             'message': 'Generating ANGLE Commit ID',
-                            # reference the git index as an input, so we rebuild on changes to the index
-                            'inputs': [ '<(angle_id_script)', '<(angle_path)/.git/index' ],
+                            'inputs': [ '<(angle_id_script)' ],
                             'outputs': [ '<(angle_id_header)' ],
                             'msvs_cygwin_shell': 0,
                             'action': 
