--- chromium-browser-31.0.1650.57/build/linux/unbundle/remove_bundled_libraries.py	2013-11-16 22:18:32.618209221 +0200
+++ chromium-browser-31.0.1650.57/build/linux/unbundle/remove_bundled_libraries.py	2013-11-16 23:34:12.718095629 +0200
@@ -27,6 +27,7 @@
 
   parser = optparse.OptionParser()
   parser.add_option('--do-remove', action='store_true')
+  parser.add_option('--do-print', action='store_true')
 
   options, args = parser.parse_args(argv)
 
@@ -62,6 +63,8 @@
 
       if options.do_remove:
         # Delete the file - best way to ensure it's not used during build.
+        if options.do_print:
+            print relpath
         os.remove(path)
       else:
         # By default just print paths that would be removed.
@@ -74,7 +77,7 @@
   for exclusion, used in exclusion_used.iteritems():
     if not used:
       print '%s does not exist' % exclusion
-      exit_code = 1
+      exit_code = 0
 
   if not options.do_remove:
     print ('To actually remove files printed above, please pass ' +
 
