diff -Nuar a/code/client/cl_curl.h b/code/client/cl_curl.h
--- a/code/client/cl_curl.h	2009-05-26 00:12:35.000000000 +0300
+++ b/code/client/cl_curl.h	2013-09-16 02:29:34.734551257 +0300
@@ -39,7 +39,7 @@
 #endif
 
 #ifdef USE_LOCAL_HEADERS
-  #include "../libcurl/curl/curl.h"
+  #include <curl/curl.h>
 #else
   #include <curl/curl.h>
 #endif
diff -Nuar a/code/qcommon/ioapi.c b/code/qcommon/ioapi.c
--- a/code/qcommon/ioapi.c	2011-01-22 02:55:16.000000000 +0200
+++ b/code/qcommon/ioapi.c	2013-09-16 02:27:57.007554640 +0300
@@ -11,7 +11,7 @@
 #include <string.h>
 
 #ifdef USE_LOCAL_HEADERS
-#include "../zlib/zlib.h"
+#include <zlib.h>
 #else
 #include <zlib.h>
 #endif
diff -Nuar a/code/qcommon/unzip.h b/code/qcommon/unzip.h
--- a/code/qcommon/unzip.h	2011-01-22 02:55:16.000000000 +0200
+++ b/code/qcommon/unzip.h	2013-09-16 02:25:10.209560415 +0300
@@ -50,7 +50,7 @@
 #endif
 
 #ifdef USE_LOCAL_HEADERS
-  #include "../zlib/zlib.h"
+  #include <zlib.h>
 #else
   #include <zlib.h>
 #endif
