diff -Naur worldofpadman-1.6_svn2178-src.orig/Makefile worldofpadman-1.6_svn2178-src/Makefile
--- worldofpadman-1.6_svn2178-src.orig/Makefile	2011-12-17 16:07:14.000000000 +0100
+++ worldofpadman-1.6_svn2178-src/Makefile	2012-05-27 22:42:02.000000000 +0200
@@ -345,6 +345,7 @@
   
   ifeq ($(USE_CIN_THEORA),1)
     CLIENT_LDFLAGS += -ltheora
+    CLIENT_LIBS += -ltheora
   endif
   
   ifeq ($(USE_CODEC_MP3),1)
