diff -up kde-workspace-4.7.80/powerdevil/daemon/BackendConfig.cmake.no_HAL kde-workspace-4.7.80/powerdevil/daemon/BackendConfig.cmake
--- kde-workspace-4.7.80/powerdevil/daemon/BackendConfig.cmake.no_HAL	2011-10-20 22:52:49.000000000 +0200
+++ kde-workspace-4.7.80/powerdevil/daemon/BackendConfig.cmake	2011-11-21 15:59:36.707272566 +0100
@@ -39,18 +39,7 @@ install(TARGETS backlighthelper DESTINAT
 kde4_install_auth_helper_files(backlighthelper org.kde.powerdevil.backlighthelper root)
 kde4_install_auth_actions(org.kde.powerdevil.backlighthelper ${CMAKE_CURRENT_SOURCE_DIR}/backends/upower/backlight_helper_actions.actions)
 
-########################## HAL Backend #####################################
-
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/backends/hal)
-
-set(powerdevilhalbackend_SRCS
-    backends/hal/halsuspendjob.cpp
-    backends/hal/powerdevilhalbackend.cpp
-)
-
-set(powerdevilhalbackend_LIBS ${KDE4_SOLID_LIBS})
-
 ########################## Daemon variables ################################
 
-set(POWERDEVIL_BACKEND_SRCS ${powerdevilupowerbackend_SRCS} ${powerdevilhalbackend_SRCS})
-set(POWERDEVIL_BACKEND_LIBS ${powerdevilupowerbackend_LIBS} ${powerdevilhalbackend_LIBS})
+set(POWERDEVIL_BACKEND_SRCS ${powerdevilupowerbackend_SRCS})
+set(POWERDEVIL_BACKEND_LIBS ${powerdevilupowerbackend_LIBS})
