diff -up kde-workspace-4.7.80/kwin/workspace.h.kde#171685 kde-workspace-4.7.80/kwin/workspace.h
--- kde-workspace-4.7.80/kwin/workspace.h.kde#171685	2011-11-17 21:55:50.000000000 +0100
+++ kde-workspace-4.7.80/kwin/workspace.h	2011-11-21 15:00:14.772410822 +0100
@@ -155,7 +155,7 @@ public:
     void setShouldGetFocus(Client*);
     bool activateNextClient(Client* c);
     bool focusChangeEnabled() {
-        return block_focus == 0;
+        return block_focus < 1;
     }
 
     void updateColormap();
