diff -up kde-workspace-4.9.90/ksmserver/screenlocker/CMakeLists.txt.kscreenlocker_rename kde-workspace-4.9.90/ksmserver/screenlocker/CMakeLists.txt
--- kde-workspace-4.9.90/ksmserver/screenlocker/CMakeLists.txt.kscreenlocker_rename	2012-11-14 10:06:18.000000000 -0600
+++ kde-workspace-4.9.90/ksmserver/screenlocker/CMakeLists.txt	2012-12-03 16:45:24.987934352 -0600
@@ -31,4 +31,4 @@ target_link_libraries(screenlocker_stati
 # Needed to compile on Arm target.
 set_target_properties(screenlocker_static PROPERTIES COMPILE_FLAGS "-fPIC")
 
-install(FILES kscreenlocker.notifyrc  DESTINATION ${DATA_INSTALL_DIR}/ksmserver/ksmserver.notifyrc)
+install(FILES kscreenlocker.notifyrc  DESTINATION ${DATA_INSTALL_DIR}/ksmserver/ RENAME ksmserver.notifyrc)
