diff -up kde-workspace-4.8.0/plasma/generic/applets/systemtray/ui/applet.cpp.systray_ktp kde-workspace-4.8.0/plasma/generic/applets/systemtray/ui/applet.cpp
--- kde-workspace-4.8.0/plasma/generic/applets/systemtray/ui/applet.cpp.systray_ktp	2012-01-18 14:08:42.000000000 -0600
+++ kde-workspace-4.8.0/plasma/generic/applets/systemtray/ui/applet.cpp	2012-02-21 15:42:27.074108968 -0600
@@ -730,6 +730,10 @@ void Applet::checkDefaultApplets()
         engines->unloadEngine("powermanagement");
     }
 
+    if (!applets.contains("ktp_presence")) {
+        s_manager->addApplet("ktp_presence", this);
+    }
+
     config().writeEntry("DefaultAppletsAdded", true);
 }
 
