diff -ur kdebase-workspace-4.5.80/plasma/generic/applets/battery/battery.cpp kdebase-workspace-4.5.80-battery-plasmoid-showremainingtime/plasma/generic/applets/battery/battery.cpp
--- kdebase-workspace-4.5.80/plasma/generic/applets/battery/battery.cpp	2010-11-19 11:51:21.000000000 +0100
+++ kdebase-workspace-4.5.80-battery-plasmoid-showremainingtime/plasma/generic/applets/battery/battery.cpp	2010-11-20 20:39:11.000000000 +0100
@@ -712,7 +712,11 @@
             m_acInfoLabel->setText(i18n("<b>Not plugged in</b>"));
         }
 
-        if (batteryCount && showRemainingTime && m_remainingMSecs > 0) {
+        // Always show the remaining time in the popup.
+        // We don't just set m_showRemainingTime to true by default because
+        // that'd try to use the time also for the icon. That won't fit on
+        // small panels.
+        if (batteryCount && m_remainingMSecs > 0) {
             m_remainingTimeLabel->show();
             m_remainingInfoLabel->show();
             // we don't have too much accuracy so only give hours and minutes
