diff -up kdebase-workspace-4.6.90/startkde.cmake.multilib_qt_plugin_path kdebase-workspace-4.6.90/startkde.cmake
--- kdebase-workspace-4.6.90/startkde.cmake.multilib_qt_plugin_path	2011-06-28 23:13:28.000000000 +0200
+++ kdebase-workspace-4.6.90/startkde.cmake	2011-06-28 23:17:05.000000000 +0200
@@ -221,6 +221,11 @@ done
 
 # Set the path for Qt plugins provided by KDE
 QT_PLUGIN_PATH=${QT_PLUGIN_PATH+$QT_PLUGIN_PATH:}`kde4-config --path qtplugins`
+
+# hack in multilib support
+if ! echo ${QT_PLUGIN_PATH} | /bin/grep -q /usr/lib/kde4/plugins ; then
+  QT_PLUGIN_PATH=${QT_PLUGIN_PATH}:/usr/lib/kde4/plugins
+fi
 export QT_PLUGIN_PATH
 
 # Set a left cursor instead of the standard X11 "X" cursor, since I've heard
