diff -up kde-workspace-4.6.80/kcontrol/krdb/krdb.cpp.krdb kde-workspace-4.6.80/kcontrol/krdb/krdb.cpp
--- kde-workspace-4.6.80/kcontrol/krdb/krdb.cpp.krdb	2011-05-20 22:32:06.000000000 +0200
+++ kde-workspace-4.6.80/kcontrol/krdb/krdb.cpp	2011-05-27 17:05:16.251235301 +0200
@@ -259,6 +259,7 @@ static void createGtkrc( bool exportColo
     if ( 2==version ) {  // we should maybe check for MacOS settings here
         t << endl;
         t << "gtk-alternative-button-order = 1" << endl;
+        t << "gtk-fallback-icon-theme = \"gnome\"" << endl;
         t << endl;
     }
 
