diff -up kdebase-workspace-4.2.0/klipper/klipperrc.desktop.klipper-arora kdebase-workspace-4.2.0/klipper/klipperrc.desktop
--- kdebase-workspace-4.2.0/klipper/klipperrc.desktop.klipper-arora	2009-01-17 11:59:33.000000000 +0100
+++ kdebase-workspace-4.2.0/klipper/klipperrc.desktop	2009-02-24 14:31:28.000000000 +0100
@@ -2065,7 +2065,7 @@ Enabled=true
 
 [General]
 KeepClipboardContents=true
-No Actions for WM_CLASS=Navigator,navigator:browser,konqueror,keditbookmarks,mozilla-bin,Mozilla,Opera main window,opera,gnumeric,Gnumeric,Galeon,kcontrol,ksirc,MozillaFirebird-bin,firefox-bin,Firefox-bin,klipper,Gecko,gecko
+No Actions for WM_CLASS=Navigator,navigator:browser,konqueror,keditbookmarks,mozilla-bin,Mozilla,Opera main window,opera,gnumeric,Gnumeric,Galeon,kcontrol,ksirc,MozillaFirebird-bin,firefox-bin,Firefox-bin,klipper,Gecko,gecko,arora,Arora
 Number of Actions=7
 PopupAtMousePosition=false
 URLGrabberEnabled=false
