--- kde-workspace-4.7.2/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js	2011-05-20 23:32:08.000000000 +0300
+++ kde-workspace-4.7.2/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js	2011-10-21 21:08:57.028169349 +0300
@@ -8,10 +8,7 @@
 panel.addWidget("launcher")
 panel.addWidget("org.kde.showActivityManager")
 panel.addWidget("pager")
-tasks = panel.addWidget("tasks")
+panel.addWidget("tasks")
 panel.addWidget("systemtray")
 panel.addWidget("digital-clock")
 
-tasks.currentConfigGroup = new Array("Launchers")
-tasks.writeConfig("browser", "preferred://browser, , , ")
-tasks.writeConfig("filemanager", "preferred://filemanager, , , ")
