diff -up kdebase-workspace-4.5.90/powerdevil/daemon/powerdevilbackendloader.cpp.no_HAL2 kdebase-workspace-4.5.90/powerdevil/daemon/powerdevilbackendloader.cpp
--- kdebase-workspace-4.5.90/powerdevil/daemon/powerdevilbackendloader.cpp.no_HAL2	2010-12-10 09:18:05.000000000 -0600
+++ kdebase-workspace-4.5.90/powerdevil/daemon/powerdevilbackendloader.cpp	2010-12-22 13:36:36.476705060 -0600
@@ -21,7 +21,7 @@
 #include "powerdevilbackendloader.h"
 
 #include "backends/upower/powerdevilupowerbackend.h"
-#include "backends/hal/powerdevilhalbackend.h"
+// #include "backends/hal/powerdevilhalbackend.h"
 
 #include <KDebug>
 
@@ -39,6 +39,7 @@ BackendInterface* loadBackend(QObject *p
 
     kDebug() << "Failed!";
 
+#if 0
     // If we are here, try HAL
     kDebug() << "Loading HAL backend...";
     if (PowerDevilHALBackend::isAvailable()) {
@@ -48,6 +49,8 @@ BackendInterface* loadBackend(QObject *p
 
     // Fail...
     kDebug() << "Failed!";
+#endif
+
     return 0;
 }
 
