diff -up kdebase-workspace-4.2.85/klipper/klipperrc.desktop.klipper-url kdebase-workspace-4.2.85/klipper/klipperrc.desktop
--- kdebase-workspace-4.2.85/klipper/klipperrc.desktop.klipper-url	2009-04-05 22:00:13.000000000 +0200
+++ kdebase-workspace-4.2.85/klipper/klipperrc.desktop	2009-05-11 22:56:51.000000000 +0200
@@ -409,7 +409,7 @@ Description[zh_TW]=使用 &Konqueror 開
 Enabled=true
 
 [Action_1/Command_2]
-Commandline=ps x |grep -q '[m]ozilla' && "mozilla -remote openURL(%s, new-window)" || mozilla %s
+Commandline=ps x |grep -q '[m]ozilla' && mozilla -remote 'openURL(%s, new-window)' || mozilla %s
 Icon=mozilla
 Description=Open with &Mozilla
 Description[af]=Maak oop met Mozilla
@@ -581,7 +581,7 @@ Description[zh_TW]=送出 &URL
 Enabled=true
 
 [Action_1/Command_4]
-Commandline=ps x |grep -q '[f]irefox' && "firefox -remote openURL(%s, new-window)" || firefox "%s"
+Commandline=ps x |grep -q '[f]irefox' && firefox -remote 'openURL(%s, new-window)' || firefox "%s"
 Icon=firefox
 Description=Open with &Firefox
 Description[af]=Maak oop met Firefox
@@ -1892,7 +1892,7 @@ Description[zh_TW]=使用 &Konqueror 開
 Enabled=true
 
 [Action_6/Command_2]
-Commandline=mozilla -remote openURL(%s, new-window)
+Commandline=mozilla -remote 'openURL(%s, new-window)'
 Description=Open with &Mozilla
 Description[af]=Maak oop met Mozilla
 Description[ar]=افتح بواسطة &موزيلا
