diff -up kdebase-workspace-4.4.92/plasma/desktop/applets/kickoff/core/applicationmodel.cpp.show_systemsettings kdebase-workspace-4.4.92/plasma/desktop/applets/kickoff/core/applicationmodel.cpp
diff -up kdebase-workspace-4.4.92/systemsettings/app/systemsettings.desktop.show_systemsettings kdebase-workspace-4.4.92/systemsettings/app/systemsettings.desktop
--- kdebase-workspace-4.4.92/systemsettings/app/systemsettings.desktop.show_systemsettings	2010-04-28 03:01:06.000000000 -0500
+++ kdebase-workspace-4.4.92/systemsettings/app/systemsettings.desktop	2010-07-07 17:32:00.204202460 -0500
@@ -169,3 +169,4 @@ Name[zh_TW]=系統設定
 
 X-DBUS-StartupType=Unique
 Categories=Qt;KDE;Settings;
+OnlyShowIn=KDE;
