diff -ur kdebase-workspace-4.3.98/qguiplatformplugin_kde/qguiplatformplugin_kde.cpp kdebase-workspace-4.3.98-platformplugin-widgetstyle4/qguiplatformplugin_kde/qguiplatformplugin_kde.cpp
--- kdebase-workspace-4.3.98/qguiplatformplugin_kde/qguiplatformplugin_kde.cpp	2009-12-17 17:35:19.000000000 +0100
+++ kdebase-workspace-4.3.98-platformplugin-widgetstyle4/qguiplatformplugin_kde/qguiplatformplugin_kde.cpp	2010-02-02 01:24:26.000000000 +0100
@@ -159,7 +159,7 @@
     {
         const QString defaultStyle = KStyle::defaultStyle();
         const KConfigGroup pConfig(KGlobal::config(), "General");
-        return pConfig.readEntry("widgetStyle", defaultStyle);
+        return pConfig.readEntry("widgetStyle4", pConfig.readEntry("widgetStyle", defaultStyle));
     }
     virtual QPalette palette()
     {
