diff -Nuar a/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js b/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
--- a/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js	2013-06-28 20:10:44.508410537 +0300
+++ b/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js	2013-09-02 23:39:17.620676882 +0300
@@ -10,6 +10,7 @@
 pager = panel.addWidget("pager")
 pager.writeConfig("hideWhenSingleDesktop", "true")
 tasks = panel.addWidget("tasks")
+panel.addWidget("nazar")
 panel.addWidget("systemtray")
 panel.addWidget("digital-clock")
 
