Index: kdebase-workspace-4.5.5/kcontrol/kdm/kdm-gen.cpp
===================================================================
--- kdebase-workspace-4.5.5.orig/kdm/kcm/kdm-gen.cpp
+++ kdebase-workspace-4.5.5/kdm/kcm/kdm-gen.cpp
@@ -35,6 +35,8 @@
 #include <QFormLayout>
 #include <QVBoxLayout>
 
+#include <locale.h>
+
 extern KConfig *config;
 
 KDMGeneralWidget::KDMGeneralWidget(QWidget *parent)
@@ -205,7 +207,21 @@ void KDMGeneralWidget::load()
     colcombo->setCurrentId(configGrp.readEntry("ColorScheme"));
 
     // get the language
-    langcombo->setCurrentItem(configGrp.readEntry("Language", "C"));
+    QString langEntry = configGrp.readEntry("Language", "C");
+    if ((langEntry == "C") || !langcombo->contains(langEntry))
+    {
+        char *locale = setlocale(LC_MESSAGES, NULL);
+
+        QString lang, country, modifier, charset;
+        KLocale::splitLocale(locale, lang, country, modifier, charset);
+
+        if (lang.toLower() == country.toLower())
+            langEntry =  lang;
+        else
+            langEntry = lang + "_" + country;
+    }
+
+    langcombo->setCurrentItem(langEntry);
 
     // Read the fonts
     QFont font = stdFontChooser->font();
