diff -Nuar a/ksmserver/screenlocker/greeter/themes/org.kde.passworddialog/contents/ui/main.qml b/ksmserver/screenlocker/greeter/themes/org.kde.passworddialog/contents/ui/main.qml
--- a/ksmserver/screenlocker/greeter/themes/org.kde.passworddialog/contents/ui/main.qml	2013-06-28 20:10:44.306419390 +0300
+++ b/ksmserver/screenlocker/greeter/themes/org.kde.passworddialog/contents/ui/main.qml	2013-08-29 09:50:27.000000000 +0300
@@ -42,7 +42,7 @@
     Image {
         id: background
         anchors.fill: parent
-        source: theme.wallpaperPathForSize(parent.width, parent.height)
+        source: "/usr/share/kde4/apps/kdm/themes/pisilinux/background.png"
         smooth: true
     }
 
