diff -Nuar kdm.orig/config.def kdm/config.def
--- kdm.orig/config.def	2013-04-12 22:57:35.140103398 +0300
+++ kdm/config.def	2013-04-12 22:58:19.921104597 +0300
@@ -1259,10 +1259,10 @@
  Grub2/BO_GRUB2: Grub2 boot manager
  Burg/BO_BURG: Burg boot manager
  Lilo/BO_LILO: Lilo boot manager (Linux on i386 &amp; x86-64 only)
-Default: None
+Default: Grub2
 User: core
 User: greeter
-Instance: #Grub
+Instance: Grub2
 Merge: kdm:UseLilo(P_UseLilo)
 Comment: &
 Description:
@@ -1348,7 +1348,7 @@

 Key: ServerTimeout
 Type: int
-Default: 30
+Default: 60
 User: core
 Instance: #:*/
 Comment:
@@ -1476,7 +1476,7 @@

 Key: TerminateServer
 Type: bool
-Default: false
+Default: true
 User: core
 Instance: #:*/!
 Merge: xdm
@@ -1553,7 +1553,7 @@
 Key: Resources
 # XXX strictly speaking this is supposed to be a string list, i think.
 Type: string
-Default: ""
+Default: "/etc/X11/xdm/Xresources"
 User: core
 Instance: #*/""
 Update: cp_resources
@@ -2171,7 +2171,7 @@
 # This needs to come _in front_ of the font settings to be effective!
 Key: AntiAliasing
 Type: bool
-Default: false
+Default: true
 User: greeter
 Instance: #*/!
 Comment: &
@@ -2180,8 +2180,8 @@

 Key: GreetFont
 Type: string
-Default: "Serif,20,-1,5,50,0,0,0,0,0"
-CDefault: "Serif 20pt bold"
+Default: "DejaVu Sans,20,-1,5,50,0,0,0,0,0"
+CDefault: "DejaVu Sans 20pt bold"
 User: greeter:font
 Instance: #*/"Serif,20,-1,5,50,0,0,0,0,0"
 Comment: &
@@ -2190,20 +2190,20 @@

 Key: StdFont
 Type: string
-Default: "Sans Serif,10,-1,5,50,0,0,0,0,0"
-CDefault: "Sans Serif 10pt"
+Default: "DejaVu Sans,10,-1,5,50,0,0,0,0,0"
+CDefault: "DejaVu Sans 10pt"
 User: greeter(normalFont):font
-Instance: #*/"Sans Serif,10,-1,5,50,0,0,0,0,0"
+Instance: #*/"DejaVu Sans,10,-1,5,50,0,0,0,0,0"
 Comment: &
 Description:
  The normal font used in the greeter. The value is encoded.

 Key: FailFont
 Type: string
-Default: "Sans Serif,10,-1,5,75,0,0,0,0,0"
-CDefault: "Sans Serif 10pt bold"
+Default: "DejaVu Sans,10,-1,5,75,0,0,0,0,0"
+CDefault: "DejaVu Sans 10pt bold"
 User: greeter:font
-Instance: #*/"Sans Serif,10,-1,5,75,0,0,0,0,0"
+Instance: #*/"DejaVu Sans,10,-1,5,75,0,0,0,0,0"
 Comment: &
 Description:
  The font used for the <quote>Login Failed</quote> message. The value is encoded.
@@ -2350,9 +2350,9 @@
  PreferAdmin/FACE_PREFER_ADMIN: prefer &lt;<option>FaceDir</option>&gt;, fallback on $<envar>HOME</envar>
  PreferUser/FACE_PREFER_USER: ... and the other way round
  UserOnly/FACE_USER_ONLY: from the user's <filename>$<envar>HOME</envar>/.face[.icon]</filename>
-Default: AdminOnly
+Default: PreferUser
 User: greeter
-Instance: #*/PreferUser
+Instance: */PreferUser
 Comment:
  Specify, where the users' pictures should be taken from.
 Description:
@@ -2702,7 +2702,7 @@

 Key: UseTheme
 Type: bool
-Default: false
+Default: true
 User: greeter
 Instance: */true
 Comment:
@@ -2717,7 +2717,7 @@
 Type: string
 Default: ""
 User: greeter
-Instance: */KDMDATA "/themes/elarun"
+Instance: */KDMDATA "/themes/pisilinux"
 Comment: &
 Description:
  The theme to use for the greeter. Can point to either a directory or an XML
diff -Nuar kdm.orig/kfrontend/genkdmconf.c kdm/kfrontend/genkdmconf.c
--- kdm.orig/kfrontend/genkdmconf.c	2013-04-12 22:57:35.140103398 +0300
+++ kdm/kfrontend/genkdmconf.c	2013-04-12 22:59:08.650105902 +0300
@@ -1641,8 +1641,8 @@
         return;
     buf[fread(buf, 1, sizeof(buf) - 1, fp)] = 0;
     pclose(fp);
-    if (strstr(buf, "\n-br "))
-        addStr(&sa, "-br");
+    if (strstr(buf, "\n-nr "))
+        addStr(&sa, "-nr");
     if (strstr(buf, "\n-novtswitch "))
         addStr(&sa, "-novtswitch");
     if (strstr(buf, "\n-quiet "))
