Index: kdebase-workspace-4.6.5/plasma/generic/applets/icon/icon.h
===================================================================
--- kdebase-workspace-4.6.5.orig/plasma/generic/applets/icon/icon.h
+++ kdebase-workspace-4.6.5/plasma/generic/applets/icon/icon.h
@@ -46,6 +46,7 @@ class IconApplet : public Plasma::Applet
         void constraintsEvent(Plasma::Constraints constraints);
         void setDisplayLines(int displayLines);
         int displayLines();
+        QString cleanFileName(KUrl destination);
         QPainterPath shape() const;
 
     public slots:
Index: kdebase-workspace-4.6.5/plasma/generic/applets/icon/icon.cpp
===================================================================
--- kdebase-workspace-4.6.5.orig/plasma/generic/applets/icon/icon.cpp
+++ kdebase-workspace-4.6.5/plasma/generic/applets/icon/icon.cpp
@@ -153,7 +153,7 @@ void IconApplet::setUrl(const KUrl& url)
         m_text = f.readName();
         //corrupted desktop file?
         if (m_text.isNull()) {
-            m_text = m_url.fileName();
+            m_text = cleanFileName(m_url);
         }
         m_icon->setIcon(f.readIcon());
 
@@ -161,7 +161,8 @@ void IconApplet::setUrl(const KUrl& url)
 
         connect(m_watcher, SIGNAL(dirty(const QString &)), this, SLOT(updateDesktopFile()));
     } else {
-        m_text = m_url.fileName();
+        m_text = cleanFileName(m_url);
+
         m_service = KService::serviceByStorageId(m_text);
         connect(KSycoca::self(), SIGNAL(databaseChanged(QStringList)),
                 this, SLOT(checkService(QStringList)));
@@ -330,6 +331,20 @@ int IconApplet::displayLines()
     return 0;
 }
 
+// Remove "trashid-" prefix from the file name when restoring a file with drag and drop action or copy/cut paste
+// KDE bug #267221 and #262881
+QString IconApplet::cleanFileName(KUrl destination)
+{
+    if (destination.protocol() == QLatin1String("trash")) {
+        QString fileName = m_url.fileName();
+        int dashPos = fileName.indexOf(QLatin1Char('-'), 0) + 1;
+        QString newFileName = fileName.remove(0, dashPos);
+
+        return newFileName;
+    }
+    return destination.fileName();
+}
+
 void IconApplet::acceptedPropertiesDialog()
 {
     KConfigGroup cg = config();
