diff -Nuar libfli1-1.7.orig//CMakeLists.txt libfli1-1.7/CMakeLists.txt
--- libfli1-1.7.orig//CMakeLists.txt	2010-11-29 12:24:08.012171850 +0200
+++ libfli1-1.7/CMakeLists.txt	2010-11-29 12:24:26.521163114 +0200
@@ -13,7 +13,7 @@
 set_target_properties(fli PROPERTIES VERSION 1.7 SOVERSION 1)
 
 #need to link to some other libraries ? just add them here
-TARGET_LINK_LIBRARIES(fli)
+TARGET_LINK_LIBRARIES(fli m)
 
 #add an install target here
 INSTALL(FILES libfli.h DESTINATION include)
