--- conf.d/50-user.conf.orig	2013-04-09 15:15:06.170861118 +0300
+++ conf.d/50-user.conf	2013-04-09 15:15:37.679861962 +0300
@@ -9,7 +9,4 @@
 	-->
 	<include ignore_missing="yes" prefix="xdg">fontconfig/conf.d</include>
 	<include ignore_missing="yes" prefix="xdg">fontconfig/fonts.conf</include>
-	<!-- the following elements will be removed in the future -->
-	<include ignore_missing="yes" deprecated="yes">~/.fonts.conf.d</include>
-	<include ignore_missing="yes" deprecated="yes">~/.fonts.conf</include>
 </fontconfig>
