diff -pruN fontconfig-2.8.90.orig/fc-cache/fc-cache.c fontconfig-2.8.90/fc-cache/fc-cache.c
--- fontconfig-2.8.90.orig/fc-cache/fc-cache.c	2012-02-20 12:49:18.000000000 +0900
+++ fontconfig-2.8.90/fc-cache/fc-cache.c	2012-03-06 12:28:26.000000000 +0900
@@ -465,7 +465,7 @@ main (int argc, char **argv)
     FcConfigDestroy (config);
     FcFini ();
     if (changed)
-	sleep (2);
+	sleep (1);
     if (verbose)
 	printf ("%s: %s\n", argv[0], ret ? "failed" : "succeeded");
     return ret;
