Index: src/pilot-read-notepad.c
===================================================================
--- src/pilot-read-notepad.c.orig
+++ src/pilot-read-notepad.c
@@ -38,9 +38,11 @@
 #include "pi-userland.h"
 
 #ifdef HAVE_PNG
-#include "png.h"
-#if (PNG_LIBPNG_VER < 10201)
+#include <png.h>
+#ifndef png_voidp_NULL
  #define png_voidp_NULL (png_voidp)NULL
+#endif
+#ifndef png_error_ptr_NULL
  #define png_error_ptr_NULL (png_error_ptr)NULL
 #endif
 #endif
Index: src/pilot-read-palmpix.c
===================================================================
--- src/pilot-read-palmpix.c.orig
+++ src/pilot-read-palmpix.c
@@ -41,9 +41,11 @@
 #include "pi-userland.h"
 
 #ifdef HAVE_PNG
-#include "png.h"
-#if (PNG_LIBPNG_VER < 10201)
+#include <png.h>
+#ifndef png_voidp_NULL
  #define png_voidp_NULL (png_voidp)NULL
+#endif
+#ifndef png_error_ptr_NULL
  #define png_error_ptr_NULL (png_error_ptr)NULL
 #endif
 #endif
Index: src/pilot-read-screenshot.c
===================================================================
--- src/pilot-read-screenshot.c.orig
+++ src/pilot-read-screenshot.c
@@ -39,9 +39,11 @@
 #include "pi-userland.h"
 
 #ifdef HAVE_PNG
-# include "png.h"
-# if (PNG_LIBPNG_VER < 10201)
+# include <png.h>
+#ifndef png_voidp_NULL
 #  define png_voidp_NULL (png_voidp)NULL
+#endif
+#ifndef png_error_ptr_NULL
 #  define png_error_ptr_NULL (png_error_ptr)NULL
 # endif
 #endif
Index: src/pilot-read-veo.c
===================================================================
--- src/pilot-read-veo.c.orig
+++ src/pilot-read-veo.c
@@ -40,9 +40,11 @@
 #include "pi-userland.h"
 
 #ifdef HAVE_PNG
-# include "png.h"
-# if (PNG_LIBPNG_VER < 10201)
+# include <png.h>
+#ifndef png_voidp_NULL
 #  define png_voidp_NULL (png_voidp)NULL
+#endif
+#ifndef png_error_ptr_NULL
 #  define png_error_ptr_NULL (png_error_ptr)NULL
 # endif
 #endif
