diff -p -up pilot-link-0.12.3/src/pilot-dlpsh.c~ pilot-link-0.12.3/src/pilot-dlpsh.c
--- pilot-link-0.12.3/src/pilot-dlpsh.c~	2007-02-02 12:35:44.000000000 +0100
+++ pilot-link-0.12.3/src/pilot-dlpsh.c	2009-02-26 14:59:58.000000000 +0100
@@ -389,7 +389,7 @@ int time_fn(int sd, int argc, const char
 
 	strftime(timebuf, 80, "Now setting Palm time from desktop to: "
 			      "%a %b %d %H:%M:%S %Z %Y\n", tm_ptr);
-	printf(timebuf);
+	printf("%s", timebuf);
 	gettimeofday(&tv, 0);
 	ltime = tv.tv_sec + 1;
 	usleep(1000000 - tv.tv_usec);
diff -p -up pilot-link-0.12.3/src/pilot-hinotes.c~ pilot-link-0.12.3/src/pilot-hinotes.c
--- pilot-link-0.12.3/src/pilot-hinotes.c~	2006-10-12 16:21:21.000000000 +0200
+++ pilot-link-0.12.3/src/pilot-hinotes.c	2009-02-26 15:00:39.000000000 +0100
@@ -80,7 +80,7 @@ void write_memo_mbox(struct PilotUser Us
 	else
 		printf("\n");
 	printf("\n");
-	printf(m.text);
+	printf("%s", m.text);
 	printf("\n");
 }
 
diff -p -up pilot-link-0.12.3/src/pilot-xfer.c~ pilot-link-0.12.3/src/pilot-xfer.c
--- pilot-link-0.12.3/src/pilot-xfer.c~	2006-10-12 16:21:21.000000000 +0200
+++ pilot-link-0.12.3/src/pilot-xfer.c	2009-02-26 15:02:22.000000000 +0100
@@ -2469,7 +2469,7 @@ main(int argc, const char *argv[])
 					fprintf(stderr, "   ERROR: '%s' is not a directory or does not exist.\n"
 							"   Please supply a directory name when performing a "
 							"backup or restore and try again.\n\n", dirname);
-					fprintf(stderr,gracias);
+					fprintf(stderr,"%s", gracias);
 					return 1;
 				}
 			}
@@ -2479,13 +2479,13 @@ main(int argc, const char *argv[])
 			if (rargc > 0)
 			{
 				fprintf(stderr,"   ERROR: Do not pass additional arguments to -busrlLC.\n");
-				fprintf(stderr,gracias);
+				fprintf(stderr,"%s", gracias);
 				return 1;
 			}
 			break;
 		case palm_op_noop:
 			fprintf(stderr,"   ERROR: Must specify one of -bursimfdlC.\n");
-			fprintf(stderr,gracias);
+			fprintf(stderr,"%s", gracias);
 			return 1;
 			break;
 		case palm_op_merge:
