Index: pilot-link-0.12.3/m4/threads.m4
===================================================================
--- pilot-link-0.12.3.orig/m4/threads.m4
+++ pilot-link-0.12.3/m4/threads.m4
@@ -62,7 +62,7 @@ AC_DEFUN([ACX_PTHREAD], [
 	# which indicates that we try without any flags at all, and "pthread-config"
 	# which is a program returning the flags for the Pth emulation library.
 
-	acx_pthread_flags="pthreads none -Kthread -kthread lthread -pthread -pthreads -mthreads pthread --thread-safe -mt pthread-config"
+	acx_pthread_flags="-lpthread pthreads none -Kthread -kthread lthread -pthread -pthreads -mthreads pthread --thread-safe -mt pthread-config"
 
 	# The ordering *is* (sometimes) important.  Some notes on the
 	# individual items follow:
