diff -Naur uqm-0.6.2-orig/sc2/build.vars uqm-0.6.2/sc2/build.vars
--- uqm-0.6.2-orig/sc2/build.vars	1970-01-01 02:00:00.000000000 +0200
+++ uqm-0.6.2/sc2/build.vars	2007-02-02 16:08:33.000000000 +0200
@@ -0,0 +1,54 @@
+# Variables for the build procedure
+# Copyright (c) 2002 Serge van den Boom
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+#  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+BUILD_SYSTEM='Linux'
+HOST_SYSTEM='Linux'
+PREPROC_C='gcc -E '
+MKDEP_C='gcc -MM '
+COMPILE_C='gcc -c '
+PREPROC_OBJC=''
+MKDEP_OBJC=''
+COMPILE_OBJC=''
+LINK='gcc '
+MAKE='make'
+ECHON='echo -n'
+SED='sed'
+REZ=''
+WINDRES=''
+uqm_CFLAGS=' pardusCFLAGS -I/usr/include/SDL -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL -D_GNU_SOURCE=1 -DGFXMODULE_SDL -DHAVE_OPENGL  -DSOUNDMODULE_SDL -pthread  -DNETPLAY=NETPLAY_FULL -DHAVE_JOYSTICK -DHAVE_ZIP=1  -DUSE_PLATFORM_ACCEL -I "."'
+uqm_LDFLAGS=' pardusLDFLAGS -L/usr/lib -lSDL -L/usr/lib -lSDL -lSDL_image -lGL -L/usr/lib -lmikmod -ldl -L/usr/lib -lm -lvorbisfile -lvorbis -lm -logg   -lz'
+uqm_INSTALL_BINDIR='/usr/bin/'
+uqm_INSTALL_LIBDIR='/usr/lib/'
+uqm_INSTALL_SHAREDIR='/usr/share/'
+
+# Exported variables are available from all the Makeinfo files
+# Non-exported files only where build.vars is explicitely included.
+uqm_SOUNDMODULE='mixsdl'
+uqm_USE_INTERNAL_MIKMOD=''
+uqm_HAVE_REGEX='1'
+uqm_HAVE_OPENGL='1'
+uqm_USE_ZIP_IO='1'
+uqm_USE_PLATFORM_ACCEL='1'
+uqm_NETPLAY='FULL'
+uqm_USE_WINSOCK=''
+DEBUG='0'
+MACRO_WIN32=''
+MACRO___MINGW32__=''
+export BUILD_SYSTEM HOST_SYSTEM DEBUG
+export MACRO_WIN32 MACRO___MINGW32__
+export uqm_SOUNDMODULE uqm_USE_INTERNAL_MIKMOD uqm_HAVE_REGEX uqm_USE_WINSOCK
+export uqm_HAVE_OPENGL uqm_USE_ZIP_IO uqm_USE_PLATFORM_ACCEL uqm_NETPLAY
diff -Naur uqm-0.6.2-orig/sc2/config.state uqm-0.6.2/sc2/config.state
--- uqm-0.6.2-orig/sc2/config.state	1970-01-01 02:00:00.000000000 +0200
+++ uqm-0.6.2/sc2/config.state	2007-02-02 16:08:32.000000000 +0200
@@ -0,0 +1,13 @@
+CHOICE_debug_VALUE='nodebug'
+CHOICE_graphics_VALUE='opengl'
+CHOICE_sound_VALUE='mixsdl'
+CHOICE_mikmod_VALUE='external'
+CHOICE_ovcodec_VALUE='standard'
+CHOICE_netplay_VALUE='full'
+CHOICE_joystick_VALUE='enabled'
+CHOICE_ioformat_VALUE='stdio_zip'
+CHOICE_accel_VALUE='asm'
+INPUT_install_prefix_VALUE='/usr'
+INPUT_install_bindir_VALUE='$prefix/bin'
+INPUT_install_libdir_VALUE='$prefix/lib'
+INPUT_install_sharedir_VALUE='$prefix/share'
diff -Naur uqm-0.6.2-orig/sc2/config_unix.h uqm-0.6.2/sc2/config_unix.h
--- uqm-0.6.2-orig/sc2/config_unix.h	1970-01-01 02:00:00.000000000 +0200
+++ uqm-0.6.2/sc2/config_unix.h	2007-02-02 16:08:32.000000000 +0200
@@ -0,0 +1,57 @@
+/* This file contains some compile-time configuration options for *nix
+ * systems.
+ * config_unix.h is generated from config_unix.h.in by build.sh
+ * When building on MS Windows using build.sh (MinGW, Cygwin),
+ * config_win.h is generated from src/config_win.h.in.
+ * When using MSVC on MS Windows, you'll have to edit src/msvc++/config.h
+ * manually if you want anything else than the defaults.
+ */
+
+#ifndef _CONFIG_UNIX_H
+#define _CONFIG_UNIX_H
+
+/* Directory where the UQM game data is located */
+#define CONTENTDIR "/usr/share/uqm/content"
+
+/* Directory where game data will be stored */
+#define USERDIR "~/.uqm/"
+
+/* Directory where config files will be stored */
+#define CONFIGDIR USERDIR
+
+/* Directory where supermelee teams will be stored */
+#define MELEEDIR "${UQM_CONFIG_DIR}/teams/"
+
+/* Directory where save games will be stored */
+#define SAVEDIR "${UQM_CONFIG_DIR}/save/"
+
+/* Defined if words are stored with the most significant byte first */
+#undef WORDS_BIGENDIAN
+
+/* Defined if your system has readdir_r of its own */
+#define HAVE_READDIR_R
+
+/* Defined if your system has setenv of its own */
+#define HAVE_SETENV
+
+/* Defined if your system has strupr of its own */
+#undef HAVE_STRUPR
+
+/* Defined if your system has stricmp of its own */
+#undef HAVE_STRICMP
+
+/* Defined if your system has getopt_long */
+#define HAVE_GETOPT_LONG
+
+/* Defined if your system has iswgraph of its own*/
+#define HAVE_ISWGRAPH
+
+/* Defined if your system has wchar_t of its own */
+#define HAVE_WCHAR_T
+
+/* Defined if your system has wint_t of its own */
+#define HAVE_WINT_T
+
+#endif  /* _CONFIG_UNIX_H */
+
+
