--- tuxpaint-0.9.21/src/tuxpaint.c~	2013-03-18 02:51:58.000000000 +0200
+++ tuxpaint-0.9.21/src/tuxpaint.c	2013-03-18 02:51:58.524684273 +0100
@@ -12898,12 +12898,8 @@
       {
 	png_init_io(png_ptr, fi);
 
-	info_ptr->width = surf->w;
-	info_ptr->height = surf->h;
-	info_ptr->bit_depth = 8;
-	info_ptr->color_type = PNG_COLOR_TYPE_RGB;
-	info_ptr->interlace_type = 1;
-	info_ptr->valid = 0;	/* will be updated by various png_set_FOO() functions */
+	png_set_IHDR(png_ptr, info_ptr, surf->w, surf->h, 8, PNG_COLOR_TYPE_RGB,
+			1, PNG_COMPRESSION_TYPE_BASE, PNG_FILTER_TYPE_BASE);
 
 	png_set_sRGB_gAMA_and_cHRM(png_ptr, info_ptr,
 				   PNG_sRGB_INTENT_PERCEPTUAL);
--- tuxpaint-0.9.21/Makefile~	2012-02-11 11:14:52.538332123 +0100
+++ tuxpaint-0.9.21/Makefile	2012-02-11 11:19:26.806386279 +0100
@@ -72,8 +72,7 @@
 ARCH_LIBS:=$($(OS)_ARCH_LIBS)
 
 PAPER_LIB:=$(call linktest,-lpaper,)
-PNG:=$(call linktest,-lpng,)
+PNG:=$(call linktest,-lpng15,)
-PNG:=$(if $(PNG),$(PNG),$(call linktest,-lpng12,))
 
 FRIBIDI_LIB:=$(shell pkg-config --libs fribidi)
 FRIBIDI_CFLAGS:=$(shell pkg-config --cflags fribidi)
@@ -81,7 +81,7 @@
 windows_ARCH_LINKS:=-lintl $(PNG) -lwinspool -lshlwapi $(FRIBIDI_LIB)
 osx_ARCH_LINKS:=$(PAPER_LIB) $(FRIBIDI_LIB)
 beos_ARCH_LINKS:="-lintl $(PNG) -lz -lbe -liconv $(FRIBIDI_LIB)"
-linux_ARCH_LINKS:=$(PAPER_LIB) $(FRIBIDI_LIB)
+linux_ARCH_LINKS:=$(PAPER_LIB) $(FRIBIDI_LIB) $(PNG)
 ARCH_LINKS:=$($(OS)_ARCH_LINKS)
 
 windows_ARCH_HEADERS:=src/win32_print.h 
