diff -Naur --exclude '*.orig' autofs-5.0.4/configure.in autofs-5.0.4-link-with-kerberos-lib/configure.in
--- autofs-5.0.4/configure.in	2008-11-04 02:36:48.000000000 +0100
+++ autofs-5.0.4-link-with-kerberos-lib/configure.in	2009-01-30 15:26:25.000000000 +0100
@@ -256,7 +256,7 @@
 if test -z "$HAVE_SASL" -o "$HAVE_SASL" != "0" -a "$HAVE_LIBXML" == "1"
 then
 	HAVE_SASL=0
-	AC_CHECK_LIB(sasl2, sasl_client_start, HAVE_SASL=1 LIBSASL="$LIBSASL -lsasl2", , -lsasl2 $LIBS)
+	AC_CHECK_LIB(sasl2, sasl_client_start, HAVE_SASL=1 LIBSASL="$LIBSASL -lsasl2 -lkrb5", , -lsasl2 $LIBS)
 	if test "$HAVE_SASL" == "1"; then
 		AC_DEFINE(WITH_SASL,1,
 			[Define if using SASL authentication with the LDAP module])
