--- test/globals.orig	2006-04-29 19:19:59.000000000 +0200
+++ test/globals	2006-04-29 19:24:35.000000000 +0200
@@ -14,8 +14,9 @@
 
 error=0
 
-bad_globals=`nm -g --defined-only $builddir/../libmpeg2/*.o |\
-    awk '{if ($3) print $3}' | grep -v '^_\?mpeg2_'`
+bad_globals=`readelf -s $builddir/../libmpeg2/*.o |\
+	awk '$5=="GLOBAL" && $6!="HIDDEN" && $7!="UND" {print $8}' |\
+	grep -v '^_\?mpeg2_'`
 
 if test x"$bad_globals" != x""; then
     echo BAD GLOBAL SYMBOLS:
@@ -23,8 +24,9 @@
     error=1
 fi
 
-bad_globals=`nm -g --defined-only $builddir/../libmpeg2/convert/*.o |\
-    awk '{if ($3) print $3}' | grep -v '^_\?mpeg2convert_'`
+bad_globals=`readelf -s $builddir/../libmpeg2/convert/*.o |\
+	awk '$5=="GLOBAL" && $6!="HIDDEN" && $7!="UND" {print $8}' |\
+	grep -v '^_\?mpeg2convert_'`
 
 if test x"$bad_globals" != x""; then
     echo BAD GLOBAL SYMBOLS:
