From 5fdb6557be8b5758856a91e8172acfddc9d04382 Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Tue, 16 Feb 2010 16:03:38 +0300
Subject: [PATCH 01/10] Fix PAM service example name to match actual one

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 src/ejabberd.cfg.example |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/ejabberd.cfg.example b/src/ejabberd.cfg.example
index 4087947..b351cfc 100644
--- a/src/ejabberd.cfg.example
+++ b/src/ejabberd.cfg.example
@@ -235,7 +235,7 @@
 %% Authentication using PAM
 %%
 %%{auth_method, pam}.
-%%{pam_service, "pamservicename"}.
+%%{pam_service, "ejabberd"}.
 
 %%
 %% Authentication using LDAP
-- 
1.7.2.1

