From 52209eb6efd83b91d5ad49623d99c53a27ffcbe9 Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Tue, 16 Feb 2010 16:30:05 +0300
Subject: [PATCH 05/10] Fedora-specific changes to ejabberdctl

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 src/ejabberdctl.template |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/ejabberdctl.template b/src/ejabberdctl.template
index fecdecc..bceec28 100644
--- a/src/ejabberdctl.template
+++ b/src/ejabberdctl.template
@@ -9,10 +9,10 @@ ERL_MAX_ETS_TABLES=1400
 
 # define default environment variables
 NODE=ejabberd
-HOST=localhost
+HOST=`hostname -s 2>/dev/null || echo "localhost"`
 ERLANG_NODE=$NODE@$HOST
 ERL=@erl@
-INSTALLUSER=@installuser@
+INSTALLUSER=ejabberd
 
 # parse command line parameters
 ARGS=
@@ -46,7 +46,7 @@ if [ "$LOGS_DIR" = "" ] ; then
     LOGS_DIR=@LOCALSTATEDIR@/log/ejabberd
 fi
 if [ "$SPOOLDIR" = "" ] ; then
-    SPOOLDIR=@LOCALSTATEDIR@/lib/ejabberd
+    SPOOLDIR=@LOCALSTATEDIR@/lib/ejabberd/spool
 fi
 if [ "$EJABBERD_DOC_PATH" = "" ] ; then
     EJABBERD_DOC_PATH=@DOCDIR@
1.7.2.1

