From ca74de7045b7393b09b53dd02e548e76cab9c849 Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Tue, 16 Feb 2010 16:05:53 +0300
Subject: [PATCH 02/10] Mention mod_ctlextra as an ejabberd module

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 src/ejabberd.app |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/ejabberd.app b/src/ejabberd.app
index 0ebb1aa..3a8d604 100644
--- a/src/ejabberd.app
+++ b/src/ejabberd.app
@@ -70,6 +70,7 @@
 	     mod_caps,
 	     mod_configure2,
 	     mod_configure,
+	     mod_ctlextra,
 	     mod_disco,
 	     mod_echo,
 	     mod_http_bind,
-- 
1.7.2.1

