From c91ddeb5735d2a4cebda364df6419637679a0dbe Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Sat, 12 Jun 2010 14:14:52 +0400
Subject: [PATCH 07/10] Install *.so objects with 0755 permissions

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 src/Makefile.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/Makefile.in b/src/Makefile.in
index cf600c3..c9492f3 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -230,7 +230,7 @@ install: all
 	#
 	# Binary system libraries
 	install -d $(SODIR)
-	install -m 644 *.so $(SODIR)
+	install -m 755 *.so $(SODIR)
 	#
 	# Translated strings
 	install -d $(MSGSDIR)
-- 
1.7.2.1

