Index: src/Makefile.am
===================================================================
--- src/Makefile.am.orig
+++ src/Makefile.am
@@ -72,3 +72,11 @@ libdumb_la_LDFLAGS = \
 libaldmb_la_LDFLAGS = \
 	-release $(VERSION)
 
+libdumb_la_LIBADD = \
+	-lm
+
+libaldmb_la_LIBADD = \
+	libdumb.la $(allegro_LIBS)
+
+libaldmb_la_DEPENDENCIES = \
+	libdumb.la
Index: src/Makefile.in
===================================================================
--- src/Makefile.in.orig
+++ src/Makefile.in
@@ -216,12 +216,12 @@ mkinstalldirs = $(SHELL) $(top_srcdir)/c
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(lib_LTLIBRARIES)
 
-libaldmb_la_LIBADD =
+libaldmb_la_LIBADD = libdumb.la $(allegro_LIBS)
 am_libaldmb_la_OBJECTS = datduh.lo datunld.lo packfile.lo alplay.lo \
 	datit.lo datxm.lo dats3m.lo datmod.lo datitq.lo datxmq.lo \
 	dats3mq.lo datmodq.lo
 libaldmb_la_OBJECTS = $(am_libaldmb_la_OBJECTS)
-libdumb_la_LIBADD =
+libdumb_la_LIBADD = -lm
 am_libdumb_la_OBJECTS = atexit.lo duhlen.lo duhtag.lo dumbfile.lo \
 	loadduh.lo makeduh.lo rawsig.lo readduh.lo register.lo \
 	rendduh.lo rendsig.lo unload.lo clickrem.lo memfile.lo \
@@ -309,7 +309,7 @@ clean-libLTLIBRARIES:
 	  echo "rm -f \"$${dir}/so_locations\""; \
 	  rm -f "$${dir}/so_locations"; \
 	done
-libaldmb.la: $(libaldmb_la_OBJECTS) $(libaldmb_la_DEPENDENCIES) 
+libaldmb.la: $(libaldmb_la_OBJECTS) libdumb.la 
 	$(LINK) -rpath $(libdir) $(libaldmb_la_LDFLAGS) $(libaldmb_la_OBJECTS) $(libaldmb_la_LIBADD) $(LIBS)
 libdumb.la: $(libdumb_la_OBJECTS) $(libdumb_la_DEPENDENCIES) 
 	$(LINK) -rpath $(libdir) $(libdumb_la_LDFLAGS) $(libdumb_la_OBJECTS) $(libdumb_la_LIBADD) $(LIBS)
