Description:Fix some misbehaviour in upstream desktop files
Author: Alf Gaida <agaida@siduction.org>

---
Bug: <url in upstream bugtracker>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Last-Update: 2014-10-18

--- a/lxqt-config-appearance/lxqt-config-appearance.desktop.in
+++ lxqt-config-0.8.0//lxqt-config-appearance/lxqt-config-appearance.desktop.in
@@ -6,6 +6,6 @@
 Exec=lxqt-config-appearance
 Icon=preferences-desktop-theme
 Categories=Settings;DesktopSettings;Qt;LXQt;
-OnlyShowIn=LXDE;LXQt;
+OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
--- a/lxqt-config-input/lxqt-config-input.desktop.in
+++ lxqt-config-0.8.0//lxqt-config-input/lxqt-config-input.desktop.in
@@ -6,6 +6,6 @@
 Exec=lxqt-config-input
 Icon=input-keyboard
 Categories=Qt;Settings;HardwareSettings;DesktopSettings;LXQt;
-OnlyShowIn=LXDE;LXQt;
+OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
--- a/src/lxqt-config.desktop.in
+++ lxqt-config-0.8.0//src/lxqt-config.desktop.in
@@ -6,6 +6,6 @@
 Exec=lxqt-config
 Icon=preferences-system
 Categories=Settings;DesktopSettings;Qt;LXQt;
-OnlyShowIn=LXDE;LXQt;
+OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
--- a/lxqt-config-monitor/lxqt-config-monitor.desktop.in
+++ lxqt-config-0.8.0//lxqt-config-monitor/lxqt-config-monitor.desktop.in
@@ -5,7 +5,7 @@
 Comment=Configure monitors
 Exec=lxqt-config-monitor
 Icon=preferences-desktop-display
-Categories=Settings;DesktopSettings;Qt;LXQt;;HardwareSettings;
+Categories=Settings;DesktopSettings;Qt;LXQt;HardwareSettings;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
 
