--- interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c.PiSiLinux        2013-11-30 23:11:39.000000000 +0200
+++ interfaces/IFusionSoundMusicProvider/ifusionsoundmusicprovider_ffmpeg.c  2013-11-30 23:17:44.289784993 +0200
@@ -45,7 +45,7 @@
 #include <libavcodec/avcodec.h>
 #include <libavformat/avformat.h>
 
-
+#define AVCODEC_MAX_AUDIO_FRAME_SIZE 192000
 #if (LIBAVFORMAT_VERSION_MAJOR >= 53)
 #include <libavutil/mathematics.h>
 struct AVDictionary {
@@ -1294,7 +1294,7 @@
           return DR_FAILURE;
      }
 
-     if (av_find_stream_info( data->ctx ) < 0) {
+     if (avformat_find_stream_info( data->ctx, NULL ) < 0) {
           D_ERROR( "IFusionSoundMusicProvider_FFmpeg: couldn't find stream info!\n" );
           IFusionSoundMusicProvider_FFmpeg_Destruct( thiz );
           return DR_FAILURE;
@@ -1319,7 +1319,7 @@
 
      data->codec = data->st->codec;
      c = avcodec_find_decoder( data->codec->codec_id );
-     if (!c || avcodec_open( data->codec, c ) < 0) {
+     if (!c || avcodec_open2( data->codec, c, NULL ) < 0) {
           D_ERROR( "IFusionSoundMusicProvider_FFmpeg: couldn't find audio decoder!\n" );
           IFusionSoundMusicProvider_FFmpeg_Destruct( thiz );
           return DR_FAILURE;
