diff -up ntp-4.2.6p1/ntpd/ntp_config.c.logdefault ntp-4.2.6p1/ntpd/ntp_config.c
--- ntp-4.2.6p1/ntpd/ntp_config.c.logdefault	2010-01-24 11:01:45.000000000 +0100
+++ ntp-4.2.6p1/ntpd/ntp_config.c	2010-03-09 17:44:09.000000000 +0100
@@ -3794,7 +3794,7 @@ getconfig(
 
 #endif /* SYS_WINNT */
 	res_fp = NULL;
-	ntp_syslogmask = NLOG_SYNCMASK; /* set more via logconfig */
+	ntp_syslogmask = NLOG_SYNCMASK | NLOG_EVENT | NLOG_STATUS; /* set more via logconfig */
 
 	/*
 	 * install a non default variable with this daemon version
