diff -up ntp-4.2.4p7/ntpstat-0.2/ntpstat.c.ntpstat ntp-4.2.4p7/ntpstat-0.2/ntpstat.c
--- ntpstat-0.2/ntpstat.c.ntpstat	2002-06-10 08:02:12.000000000 +0200
+++ ntpstat-0.2/ntpstat.c	2009-07-20 12:22:35.000000000 +0200
@@ -187,7 +187,7 @@ int main (void) {
     else
       printf("unknown source");
 
-    if (!strncmp(clksrcname[clksrc],clksrcname[6],sizeof(clksrcname[6]))) {
+    if (clksrc == 6) {
       // source of sync is another NTP server so check the IP address
       strncpy(buff, ntpmsg.payload, sizeof(buff));
       if ((newstr = strstr (buff, REFID))) {
