--- share/functions/__fish_config_interactive.fish.orig	2010-10-19 05:02:20.082000012 +0300
+++ share/functions/__fish_config_interactive.fish	2010-10-19 05:02:33.558000009 +0300
@@ -213,7 +213,7 @@
 	__fish_reload_key_bindings
 
 	# Repaint screen when window changes size
-	function __fish_winch_handler --on-signal winch
+	function __fish_winch_handler --on-signal WINCH
 		commandline -f repaint
 	end
 
