--- solfege/soundcard/Makefile.link	2011-06-19 12:30:53.000000000 +0200
+++ solfege/soundcard/Makefile	2011-06-19 12:31:27.000000000 +0200
@@ -50,10 +50,10 @@
 endif
 
 solfege/soundcard/solfege_c_midimodule.so: $(OBJECTS)
-	$(CC) $(LDFLAGS) -fPIC -shared $(OBJECTS) -o $@ -lc
+	$(CC) $(LDFLAGS) -fPIC -shared $(OBJECTS) -o $@ -lc -lpython$(PYTHON_VERSION)
 
 solfege/soundcard/_solfege_c_midi.so: $(OBJECTS)
-	$(CC) $(LDFLAGS) -fPIC -shared $(OBJECTS) -o $@ -lc 
+	$(CC) $(LDFLAGS) -fPIC -shared $(OBJECTS) -o $@ -lc -lpython$(PYTHON_VERSION)
 
 solfege/soundcard/solfege_c_midi_wrap.o: solfege/soundcard/solfege_c_midi_wrap.c solfege/soundcard/solfege_c_midi.i
 	(cd solfege/soundcard; $(CC) $(CFLAGS) -c solfege_c_midi_wrap.c $(PYTHON_INCLUDES))