--- a/setup.py	2008-06-30 16:41:43.000000000 +0300
+++ b/setup.py	2008-06-30 16:44:45.000000000 +0300
@@ -209,6 +209,8 @@
             args['compiler_so'] = compiler + ' ' + ccshared + ' ' + cflags
         self.compiler.set_executables(**args)
 
+        self.make_args = args
+
         build_ext.build_extensions(self)
 
         longest = max([len(e.name) for e in self.extensions])
@@ -241,9 +243,18 @@
 
     def build_extension(self, ext):
 
+        nexecf = " -Wa,--noexecstack"
         if ext.name == '_ctypes':
             if not self.configure_ctypes(ext):
                 return
+            # Add -Wa,--noexecstack to CFLAGS
+            if 'compiler_so' in self.make_args:
+                if not self.make_args['compiler_so'].endswith(nexecf):
+                    self.make_args['compiler_so'] += nexecf
+        elif 'compiler_so' in self.make_args:
+            self.make_args['compiler_so'] = \
+                    self.make_args['compiler_so'].replace(nexecf,'')
+        self.compiler.set_executables(**self.make_args)
 
         try:
             build_ext.build_extension(self, ext)
