diff -ur gnome-python-2.26.1.orig/gnomecanvas/Makefile.am gnome-python-2.26.1/gnomecanvas/Makefile.am
--- gnome-python-2.26.1.orig/gnomecanvas/Makefile.am	2009-03-14 20:56:53.000000000 +0200
+++ gnome-python-2.26.1/gnomecanvas/Makefile.am	2009-06-02 23:08:38.524463472 +0300
@@ -8,13 +8,13 @@
 pkgpythondir = $(pyexecdir)/gtk-2.0
 pkgpyexecdir = $(pyexecdir)/gtk-2.0
 
-common_ldflags = -module -avoid-version $(PYTHON_LDFLAGS)
+common_ldflags = -module -avoid-version -lpython2.6
 
 # libgnomecanvas binding
 pkgpyexec_LTLIBRARIES = gnomecanvas.la
 gnomecanvas_la_CFLAGS = $(GNOMECANVAS_CFLAGS)
 gnomecanvas_la_LDFLAGS = $(common_ldflags) -export-symbols-regex initgnomecanvas
-gnomecanvas_la_LIBADD = $(GNOMECANVAS_LIBS) $(PYTHON_LIBS)
+gnomecanvas_la_LIBADD = $(GNOMECANVAS_LIBS) -lpython2.6
 gnomecanvas_la_SOURCES = canvasmodule.c
 nodist_gnomecanvas_la_SOURCES = canvas.c
 CLEANFILES = canvas.c
