--- a/ui.c	2004-02-05 23:53:19.000000000 +0100
+++ b/ui.c	2005-09-28 18:41:13.000000000 +0200
@@ -263,7 +263,7 @@
             char s[40], *p;
             int x;
             /* This 1024 vs 1000 stuff is just plain evil */
-            readable_size(i, s, sizeof s, options.log_scale ? 1000 : 1024, 0);
+            readable_size(i, s, sizeof s, options.log_scale ? 1000 : 1024, options.bandwidth_in_bytes);
             p = s + strspn(s, " ");
             x = get_bar_length(i * 8);
             mvaddch(*y + 1, x, ACS_BTEE);
