diff -up pidgin-libnotify-0.14/src/pidgin-libnotify.c.libnotify070 pidgin-libnotify-0.14/src/pidgin-libnotify.c
--- pidgin-libnotify-0.14/src/pidgin-libnotify.c.libnotify070	2010-11-08 21:28:01.981121257 +0000
+++ pidgin-libnotify-0.14/src/pidgin-libnotify.c	2010-11-08 21:28:50.027245840 +0000
@@ -286,7 +286,7 @@ notify (const gchar *title,
 		g_free (tr_body);
 		return;
 	}
-	notification = notify_notification_new (title, tr_body, NULL, NULL);
+	notification = notify_notification_new (title, tr_body, NULL);
 	purple_debug_info (PLUGIN_ID, "notify(), new: "
 					 "title: '%s', body: '%s', buddy: '%s'\n",
 					 title, tr_body, best_name (buddy)); 
