diff --git a/src/tests/Makefile.am b/src/tests/Makefile.am
index 43b3eb1..2e28800 100644
--- a/src/tests/Makefile.am
+++ b/src/tests/Makefile.am
@@ -71,7 +71,6 @@ if WITH_TESTS
 check-local: test-modem-helpers
 	$(abs_builddir)/test-modem-helpers
 	$(abs_builddir)/test-charsets
-	$(abs_builddir)/test-qcdm-serial-port
 	$(abs_builddir)/test-sms
 
 endif
diff --git a/src/tests/Makefile.in b/src/tests/Makefile.in
index d6c4d1a..bd216e2 100644
--- a/src/tests/Makefile.in
+++ b/src/tests/Makefile.in
@@ -720,7 +720,6 @@ uninstall-am:
 @WITH_TESTS_TRUE@check-local: test-modem-helpers
 @WITH_TESTS_TRUE@	$(abs_builddir)/test-modem-helpers
 @WITH_TESTS_TRUE@	$(abs_builddir)/test-charsets
-@WITH_TESTS_TRUE@	$(abs_builddir)/test-qcdm-serial-port
 @WITH_TESTS_TRUE@	$(abs_builddir)/test-sms
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
