--- a/doc/README-pulse	2009-04-26 19:59:50.000000000 +0300
+++ b/doc/README-pulse	2009-04-26 20:01:07.000000000 +0300
@@ -1,6 +1,15 @@
 PulseAudio <--> ALSA plugins
 ============================
 
+Requirements
+=============
+
+In order to be able to use this plugin, you have to install
+alsa-plugins-pulseaudio package in Pisi Linux.
+
+Introduction
+=============
+
 This plugin allows any program that uses the ALSA API to access a PulseAudio
 sound daemon. In other words, native ALSA applications can play and record
 sound across a network.
