Index: alsa-plugins/arcam-av/Makefile.am
===================================================================
--- alsa-plugins.orig/arcam-av/Makefile.am
+++ alsa-plugins/arcam-av/Makefile.am
@@ -6,4 +6,4 @@ AM_CFLAGS = -Wall -g @ALSA_CFLAGS@
 AM_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined
 
 libasound_module_ctl_arcam_av_la_SOURCES = ctl_arcam_av.c arcam_av.c arcam_av.h
-libasound_module_ctl_arcam_av_la_LIBADD = @ALSA_LIBS@
+libasound_module_ctl_arcam_av_la_LIBADD = @ALSA_LIBS@ -lpthread
