--- src/odbc/Makefile.am.orig	2006-07-26 14:05:55.000000000 +0200
+++ src/odbc/Makefile.am	2006-07-26 14:06:02.000000000 +0200
@@ -15,7 +15,7 @@
 				$(patsubst %.c, %.lo, $(SQLSOURCES))) \
 				$(patsubst %, $(MDBDIR)/%, \
 				$(patsubst %.c, %.lo, $(MDBSOURCES)))
-LIBS	=	@LEXLIB@ $(GLIB_LIBS) $(ODBC_LIBS)
+LIBS	=	@LEXLIB@ $(GLIB_LIBS) $(ODBC_LIBS) -lm
 unittest_LDADD	=	libmdbodbc.la ../libmdb/libmdb.la ../sql/libmdbsql.la
 
 ## Need blank statement to avoid compiling odbc.c
