--- mdbtools-cvs-20050624/src/odbc/odbc.c.orig	2010-05-24 10:20:59.000000000 -0400
+++ mdbtools-cvs-20050624/src/odbc/odbc.c	2010-05-25 17:28:05.000000000 -0400
@@ -210,7 +210,7 @@
     SQLHSTMT           hstmt,
     SQLUSMALLINT       ipar,
     SQLSMALLINT FAR   *pfSqlType,
-    SQLUINTEGER FAR   *pcbParamDef,
+    SQLULEN FAR   *pcbParamDef,
     SQLSMALLINT FAR   *pibScale,
     SQLSMALLINT FAR   *pfNullable)
 {
@@ -221,8 +221,8 @@
 SQLRETURN SQL_API SQLExtendedFetch(
     SQLHSTMT           hstmt,
     SQLUSMALLINT       fFetchType,
-    SQLINTEGER         irow,
-    SQLUINTEGER FAR   *pcrow,
+    SQLLEN         irow,
+    SQLULEN FAR   *pcrow,
     SQLUSMALLINT FAR  *rgfRowStatus)
 {
 struct _hstmt *stmt = (struct _hstmt *) hstmt;
@@ -297,8 +297,8 @@
 
 SQLRETURN SQL_API SQLParamOptions(
     SQLHSTMT           hstmt,
-    SQLUINTEGER        crow,
-    SQLUINTEGER FAR   *pirow)
+    SQLULEN        crow,
+    SQLULEN FAR   *pirow)
 {
 	TRACE("SQLParamOptions");
 	return SQL_SUCCESS;
@@ -347,7 +347,7 @@
 
 SQLRETURN SQL_API SQLSetPos(
     SQLHSTMT           hstmt,
-    SQLUSMALLINT       irow,
+    SQLSETPOSIROW       irow,
     SQLUSMALLINT       fOption,
     SQLUSMALLINT       fLock)
 {
@@ -398,11 +398,11 @@
     SQLSMALLINT        fParamType,
     SQLSMALLINT        fCType,
     SQLSMALLINT        fSqlType,
-    SQLUINTEGER        cbColDef,
+    SQLULEN        cbColDef,
     SQLSMALLINT        ibScale,
     SQLPOINTER         rgbValue,
-    SQLINTEGER         cbValueMax,
-    SQLINTEGER FAR    *pcbValue)
+    SQLLEN         cbValueMax,
+    SQLLEN FAR    *pcbValue)
 {
 struct _hstmt *stmt;
 
@@ -502,8 +502,8 @@
     SQLUSMALLINT       icol,
     SQLSMALLINT        fCType,
     SQLPOINTER         rgbValue,
-    SQLINTEGER         cbValueMax,
-    SQLINTEGER FAR    *pcbValue)
+    SQLLEN         cbValueMax,
+    SQLLEN FAR    *pcbValue)
 {
 	struct _hstmt *stmt = (struct _hstmt *) hstmt;
 	struct _sql_bind_info *cur, *newitem;
@@ -594,7 +594,7 @@
     SQLSMALLINT        cbColNameMax,
     SQLSMALLINT FAR   *pcbColName,
     SQLSMALLINT FAR   *pfSqlType,
-    SQLUINTEGER FAR   *pcbColDef, /* precision */
+    SQLULEN FAR   *pcbColDef, /* precision */
     SQLSMALLINT FAR   *pibScale,
     SQLSMALLINT FAR   *pfNullable)
 {
@@ -649,7 +649,7 @@
     SQLPOINTER         rgbDesc,
     SQLSMALLINT        cbDescMax,
     SQLSMALLINT FAR   *pcbDesc,
-    SQLINTEGER FAR    *pfDesc)
+    SQLLEN FAR    *pfDesc)
 {
 	int namelen, i;
 	struct _hstmt *stmt;
@@ -969,7 +969,7 @@
 
 SQLRETURN SQL_API SQLRowCount(
     SQLHSTMT           hstmt,
-    SQLINTEGER FAR    *pcrow)
+    SQLLEN FAR    *pcrow)
 {
 struct _hstmt *stmt=(struct _hstmt *)hstmt;
 
@@ -1002,10 +1002,10 @@
     SQLUSMALLINT       ipar,
     SQLSMALLINT        fCType,
     SQLSMALLINT        fSqlType,
-    SQLUINTEGER        cbParamDef,
+    SQLULEN        cbParamDef,
     SQLSMALLINT        ibScale,
     SQLPOINTER         rgbValue,
-    SQLINTEGER FAR     *pcbValue)
+    SQLLEN FAR     *pcbValue)
 {
 	TRACE("SQLSetParam");
 	return SQL_SUCCESS;
@@ -1125,8 +1125,8 @@
     SQLUSMALLINT       icol,
     SQLSMALLINT        fCType,
     SQLPOINTER         rgbValue,
-    SQLINTEGER         cbValueMax,
-    SQLINTEGER FAR    *pcbValue)
+    SQLLEN         cbValueMax,
+    SQLLEN FAR    *pcbValue)
 {
 	struct _hstmt *stmt;
 	struct _hdbc *dbc;
@@ -1457,7 +1457,7 @@
 SQLRETURN SQL_API SQLPutData(
     SQLHSTMT           hstmt,
     SQLPOINTER         rgbValue,
-    SQLINTEGER         cbValue)
+    SQLLEN         cbValue)
 {
 	TRACE("SQLPutData");
 	return SQL_SUCCESS;
@@ -1466,7 +1466,7 @@
 SQLRETURN SQL_API SQLSetConnectOption(
     SQLHDBC            hdbc,
     SQLUSMALLINT       fOption,
-    SQLUINTEGER        vParam)
+    SQLULEN        vParam)
 {
 	TRACE("SQLSetConnectOption");
 	return SQL_SUCCESS;
@@ -1475,7 +1475,7 @@
 SQLRETURN SQL_API SQLSetStmtOption(
     SQLHSTMT           hstmt,
     SQLUSMALLINT       fOption,
-    SQLUINTEGER        vParam)
+    SQLULEN        vParam)
 {
 	TRACE("SQLSetStmtOption");
 	return SQL_SUCCESS;
