--- liba52/imdct.c	2002-07-28 04:52:07.000000000 +0300
+++ liba52/imdct.c	2007-05-01 17:41:11.000000000 +0300
@@ -418,15 +418,12 @@
     }
 
 #ifdef LIBA52_DJBFFT
-    if (mm_accel & MM_ACCEL_DJBFFT) {
 	fprintf (stderr, "Using djbfft for IMDCT transform\n");
 	ifft128 = (void (*) (complex_t *)) fftc4_un128;
 	ifft64 = (void (*) (complex_t *)) fftc4_un64;
-    } else
-#endif
-    {
+#else
 	fprintf (stderr, "No accelerated IMDCT transform found\n");
 	ifft128 = ifft128_c;
 	ifft64 = ifft64_c;
-    }
+#endif
 }
