diff -up kbd-1.15/src/paths.h.orig kbd-1.15/src/paths.h
--- kbd-1.15/src/paths.h.orig	2010-06-07 15:26:21.787315778 +0200
+++ kbd-1.15/src/paths.h	2010-06-07 15:26:49.353336391 +0200
@@ -19,7 +19,7 @@
 # define DEFMAP "sunkeymap.kmap"
 # define KERNDIR "/usr/src/linux/drivers/sbus/char"
 #else
-# define DEFMAP "defkeymap.kmap"
+# define DEFMAP "defkeymap.map"
 # define KERNDIR "/usr/src/linux/drivers/char"
 #endif
 
