diff -up kbd-1.15/configure_old kbd-1.15/configure
--- kbd-1.15/configure_old	2009-01-08 15:11:04.000000000 +0100
+++ kbd-1.15/configure	2009-01-08 15:11:45.000000000 +0100
@@ -7637,8 +7637,9 @@ fi
 
 
 case $host_cpu in
-	i?86*) RESIZECONS_PROGS=yes ;;
-	*)     RESIZECONS_PROGS=no ;;
+	i?86*)   RESIZECONS_PROGS=yes ;;
+	x86_64*) RESIZECONS_PROGS=yes ;;
+	*)       RESIZECONS_PROGS=no ;;
 esac
  if test "$RESIZECONS_PROGS" = "yes"; then
   RESIZECONS_PROGS_TRUE=
