diff -up dillo-3.0.3/src/Makefile.am.orig dillo-3.0.3/src/Makefile.am
--- dillo-3.0.3/src/Makefile.am.orig	2013-01-27 18:26:38.000000000 +0100
+++ dillo-3.0.3/src/Makefile.am	2013-05-30 08:32:10.559199750 +0200
@@ -19,7 +19,7 @@ dillo_LDADD = \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
 	@LIBJPEG_LIBS@ @LIBPNG_LIBS@ @LIBFLTK_LIBS@ @LIBZ_LIBS@ \
-	@LIBICONV_LIBS@ @LIBPTHREAD_LIBS@
+	@LIBICONV_LIBS@ @LIBPTHREAD_LIBS@ @LIBSSL_LIBS@ -lX11
 
 dillo_SOURCES = \
 	dillo.cc \
diff -up dillo-3.0.3/test/Makefile.am.orig dillo-3.0.3/test/Makefile.am
--- dillo-3.0.3/test/Makefile.am.orig	2013-01-28 16:58:56.000000000 +0100
+++ dillo-3.0.3/test/Makefile.am	2013-05-30 08:33:42.415220646 +0200
@@ -33,7 +33,7 @@ dw_anchors_test_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 dw_example_SOURCES = dw_example.cc
 dw_example_LDADD = \
@@ -41,7 +41,7 @@ dw_example_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 dw_find_test_SOURCES = dw_find_test.cc
 dw_find_test_LDADD = \
@@ -49,7 +49,7 @@ dw_find_test_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 dw_links_SOURCES = dw_links.cc
 dw_links_LDADD = \
@@ -57,7 +57,7 @@ dw_links_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 dw_links2_SOURCES = dw_links2.cc
 dw_links2_LDADD = \
@@ -65,7 +65,7 @@ dw_links2_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 dw_images_simple_SOURCES = dw_images_simple.cc
 dw_images_simple_LDADD = \
@@ -73,7 +73,7 @@ dw_images_simple_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 dw_images_scaled_SOURCES = dw_images_scaled.cc
 dw_images_scaled_LDADD = \
@@ -81,7 +81,7 @@ dw_images_scaled_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 dw_images_scaled2_SOURCES = dw_images_scaled2.cc
 dw_images_scaled2_LDADD = \
@@ -89,7 +89,7 @@ dw_images_scaled2_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 dw_lists_SOURCES = dw_lists.cc
 dw_lists_LDADD = \
@@ -97,7 +97,7 @@ dw_lists_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 dw_table_aligned_SOURCES = dw_table_aligned.cc
 dw_table_aligned_LDADD = \
@@ -105,7 +105,7 @@ dw_table_aligned_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 dw_table_SOURCES = dw_table.cc
 dw_table_LDADD = \
@@ -113,7 +113,7 @@ dw_table_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 dw_border_test_SOURCES = dw_border_test.cc
 dw_border_test_LDADD = \
@@ -121,7 +121,7 @@ dw_border_test_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 
 dw_imgbuf_mem_test_SOURCES = dw_imgbuf_mem_test.cc
@@ -130,7 +130,7 @@ dw_imgbuf_mem_test_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 dw_resource_test_SOURCES = dw_resource_test.cc
 dw_resource_test_LDADD = \
@@ -138,7 +138,7 @@ dw_resource_test_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 dw_ui_test_SOURCES = \
 	dw_ui_test.cc \
@@ -149,7 +149,7 @@ dw_ui_test_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 shapes_SOURCES = shapes.cc
 shapes_LDADD = \
@@ -171,7 +171,7 @@ liang_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 trie_SOURCES = trie.cc
 
@@ -180,7 +180,7 @@ trie_LDADD = \
 	$(top_builddir)/dw/libDw-fltk.a \
 	$(top_builddir)/dw/libDw-core.a \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
 
 notsosimplevector_SOURCES = notsosimplevector.cc
 
@@ -190,4 +190,4 @@ unicode_test_SOURCES = unicode_test.cc
 
 unicode_test_LDADD = \
 	$(top_builddir)/lout/liblout.a \
-	@LIBFLTK_LIBS@
+	@LIBFLTK_LIBS@ -lX11
