diff -Nur libsrcs/angelscript/angelSVN/sdk/angelscript/projects/gnuc/makefile libsrcs/angelscript/angelSVN/sdk/angelscript/projects/gnuc/makefile
--- libsrcs/angelscript/angelSVN/sdk/angelscript/projects/gnuc/makefile	2011-02-07 13:25:12.098744321 +0200
+++ libsrcs/angelscript/angelSVN/sdk/angelscript/projects/gnuc/makefile	2011-02-07 13:28:07.441684273 +0200
@@ -12,10 +12,10 @@
 INCDIR = ../../include
 OBJDIR = obj
 
-CXX ?= gcc
+CXX ?= g++
 AR ?= ar
 RANLIB ?= ranlib
-CXXFLAGS_AS = $(CXXFLAGS) -g -Wall -fPIC
+CXXFLAGS_AS = -fno-strict-aliasing -fno-omit-frame-pointer -g -Wall -fPIC
 DELETER = rm -f
 COPIER = cp
 
diff -Nur source/Makefile source/Makefile
--- source/Makefile	2011-02-07 13:25:12.089690328 +0200
+++ source/Makefile	2011-02-07 13:36:22.667679067 +0200
@@ -48,7 +48,7 @@
 else
 CC?=gcc
 CXX?=g++
-LD=gcc
+LD=$(CC)
 AR=ar
 RANLIB=ranlib
 
@@ -166,7 +166,7 @@
 LOCALBASE?=/usr/local
 X11BASE?=/usr/X11R6
 
-CFLAGS_COMMON=$(CFLAGS) -pipe -I. -I$(LOCALBASE)/include -I$(X11BASE)/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -Wall -Wno-unused-function
+CFLAGS_COMMON=$(CFLAGS) -pipe -I. -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -Wall -Wno-unused-function
 CFLAGS_RELEASE=-O2 -fno-strict-aliasing -ffast-math -funroll-loops -DNDEBUG
 CFLAGS_DEBUG=-O0 -ggdb -D_DEBUG
 ifeq ($(DEBUG_BUILD),YES)
@@ -184,14 +184,14 @@
 ifeq ($(OS),FreeBSD)
 LDFLAGS_COMMON+= -L/usr/local/lib -lm -pthread
 else
-LDFLAGS_COMMON+= -L/usr/local/lib -ldl -lm -Wl -O1 --as-needed
+LDFLAGS_COMMON+= -ldl -lm
 endif
 ifneq ($(DEBUG_BUILD),YES)
 LDFLAGS_COMMON+= -s
 endif
 
 ifeq ($(ARCH),x86_64)
-LIB=lib64
+LIB=lib
 else
 LIB=lib
 endif
@@ -477,16 +477,16 @@
 
 all: client openal qf ded cgame game ui irc angelwrap tv_server
 
-client: $(BUILDDIRS) message-client compile-client link-client start-script-client
+client: $(BUILDDIRS) message-client compile-client link-client
 openal: $(BUILDDIRS) message-openal compile-openal link-openal
 qf: $(BUILDDIRS) message-qf compile-qf link-qf
-ded: $(BUILDDIRS) message-ded compile-ded link-ded start-script-ded
+ded: $(BUILDDIRS) message-ded compile-ded link-ded
 cgame: $(BUILDDIRS) message-cgame compile-cgame link-cgame
 game: $(BUILDDIRS) message-game compile-game link-game
 ui: $(BUILDDIRS) message-ui compile-ui link-ui
 irc: $(BUILDDIRS) message-irc compile-irc link-irc
 angelwrap: $(BUILDDIRS) message-angelwrap compile-angelwrap link-angelwrap
-tv_server: $(BUILDDIRS) message-tv_server compile-tv_server link-tv_server start-script-tv_server
+tv_server: $(BUILDDIRS) message-tv_server compile-tv_server link-tv_server
 
 clean: clean-msg clean-depend clean-client clean-openal clean-qf clean-ded clean-cgame clean-game clean-ui clean-irc clean-angelwrap clean-tv_server
 
