diff -Nur source-old/qcommon/files.c source/qcommon/files.c
--- source-old/qcommon/files.c	2009-09-11 16:00:54.662795432 +0300
+++ source/qcommon/files.c	2009-09-11 16:05:54.424045084 +0300
@@ -3733,7 +3733,7 @@
 	// set basepaths
 	//
 	fs_cdpath = Cvar_Get( "fs_cdpath", "", CVAR_NOSET );
-	fs_basepath = Cvar_Get( "fs_basepath", ".", CVAR_NOSET );
+	fs_basepath = Cvar_Get( "fs_basepath", "/usr/share/warsow", CVAR_NOSET );
 	if( Sys_FS_GetHomeDirectory() != NULL )
 #ifdef PUBLIC_BUILD
 		fs_usehomedir = Cvar_Get( "fs_usehomedir", "1", CVAR_NOSET );
