--- pyclutter-1.0.2/clutter/Makefile.am.link	2010-11-06 03:03:25.000000000 +0000
+++ pyclutter-1.0.2/clutter/Makefile.am	2010-11-06 03:03:57.000000000 +0000
@@ -1,6 +1,7 @@
 include $(top_srcdir)/Makefile.tmpl
 
 NULL =
+LIBS += -lpython$(PYTHON_VERSION)
 
 CLEANFILES =
 EXTRA_DIST = 
--- pyclutter-1.0.2/clutter/Makefile.in.link	2010-11-06 03:04:04.000000000 +0000
+++ pyclutter-1.0.2/clutter/Makefile.in	2010-11-06 03:04:22.000000000 +0000
@@ -190,7 +190,7 @@
 LD = @LD@
 LDFLAGS = @LDFLAGS@
 LIBOBJS = @LIBOBJS@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lpython$(PYTHON_VERSION)
 LIBTOOL = @LIBTOOL@
 LIPO = @LIPO@
 LN_S = @LN_S@
