diff -up alienarena-7.45/source/ref_gl/r_shadowmaps.c.BAD alienarena-7.45/source/ref_gl/r_shadowmaps.c
--- alienarena-7.45/source/ref_gl/r_shadowmaps.c.BAD	2010-08-02 14:46:28.251171881 -0400
+++ alienarena-7.45/source/ref_gl/r_shadowmaps.c	2010-08-02 14:46:54.482305473 -0400
@@ -128,14 +128,14 @@ void generateShadowFBO()
 	//FBO for capturing stencil volumes
 
 	//must check for abilit to blit(Many old ATI drivers do not support)
-	if(gl_state.hasFBOblit) {
-		if(!qglBlitFramebufferEXT) {
-			Com_Printf("qglBlitFramebufferEXT not found...\n");
+	// if(gl_state.hasFBOblit) {
+	//	if(!qglBlitFramebufferEXT) {
+	//		Com_Printf("qglBlitFramebufferEXT not found...\n");
 			//no point in continuing on
-			gl_state.hasFBOblit = false;
-			return;
-		}
-	}
+	//		gl_state.hasFBOblit = false;
+	//		return;
+	//	}
+	// }
 
     qglBindTexture(GL_TEXTURE_2D, r_colorbuffer->texnum);
     qglTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, vid.width, vid.height, 0, GL_RGBA, GL_UNSIGNED_BYTE, 0);
