Index: alienarena-7.50/source/qcommon/qcommon.h
===================================================================
--- alienarena-7.50.orig/source/qcommon/qcommon.h
+++ alienarena-7.50/source/qcommon/qcommon.h
@@ -43,7 +43,7 @@ Foundation, Inc., 59 Temple Place - Suit
 
 #if !defined USER_GAMEDATA
 // default for COR_GAME environment variable
-#define USER_GAMEDATA ".codered"
+#define USER_GAMEDATA ".alienarena"
 #endif
 /*----------------------------------------*/
 
Index: alienarena-7.50/Tools/LinuxScripts/launch-server
===================================================================
--- alienarena-7.50.orig/Tools/LinuxScripts/launch-server
+++ alienarena-7.50/Tools/LinuxScripts/launch-server
@@ -22,7 +22,7 @@
 if [ $COR_GAME ] ; then
 aadir=$COR_GAME
 else
-aadir=~/.codered
+aadir=~/.alienarena
 fi
 
 ### Set your AA Binary Name or command string ($1 is the argument
Index: alienarena-7.50/Tools/LinuxScripts/launchservers
===================================================================
--- alienarena-7.50.orig/Tools/LinuxScripts/launchservers
+++ alienarena-7.50/Tools/LinuxScripts/launchservers
@@ -10,7 +10,7 @@ execname=crx-ded
 if [ ${COR_GAME} ]; then
 cfgpath=${COR_GAME}/arena
 else
-cfgpath=~/.codered/arena
+cfgpath=~/.alienarena/arena
 fi
 
 # use to generate core dumps after crashs
