diff -Nuar a/razorqt-panel/panel/resources/panel.conf b/razorqt-panel/panel/resources/panel.conf
--- a/razorqt-panel/panel/resources/panel.conf	2013-08-27 00:45:56.000000000 +0300
+++ b/razorqt-panel/panel/resources/panel.conf	2013-08-27 22:34:43.695409787 +0300
@@ -12,13 +12,11 @@
 type=desktopswitch
 
 [quicklaunch]
+alignment=Left
 type=quicklaunch
-apps/size=5
-apps/1/desktop=firefox.desktop
-apps/2/desktop=chromium-browser.desktop
-apps/3/desktop=pcmanfm-qt.desktop
-apps/4/desktop=KMail.desktop
-apps/5/desktop=razor-config.desktop
+apps\1\desktop=/usr/share/applications/mozillafirefox.desktop
+apps\size=5
+apps\2\desktop=/usr/share/applications/kde4/dolphin.desktop
 
 [taskbar]
 type=taskbar
