diff -up ./server/drivers/shell/readlog.cc.gzfile ./server/drivers/shell/readlog.cc
--- ./server/drivers/shell/readlog.cc.gzfile	2012-06-30 14:19:53.612450930 -0400
+++ ./server/drivers/shell/readlog.cc	2012-06-30 14:20:29.446824252 -0400
@@ -665,7 +665,7 @@ void ReadLog::Main()
       // back up to the beginning of the file
 #if HAVE_Z
       if (this->gzfile)
-        ret = gzseek(this->file,0,SEEK_SET);
+        ret = gzseek((gzFile)this->file,0,SEEK_SET);
       else
         ret = fseek(this->file,0,SEEK_SET);
 #else
@@ -711,7 +711,7 @@ void ReadLog::Main()
       // compared to fgets (on uncompressed files), so use the latter.
 #if HAVE_Z
       if (this->gzfile)
-        ret = (gzgets(this->file, this->line, this->line_size) == NULL);
+        ret = (gzgets((gzFile)this->file, this->line, this->line_size) == NULL);
       else
         ret = (fgets(this->line, this->line_size, (FILE*) this->file) == NULL);
 #else

