diff -up bind-9.7.2-P2/isc-config.sh.in.exportlib bind-9.7.2-P2/isc-config.sh.in
--- bind-9.7.2-P2/isc-config.sh.in.exportlib	2007-06-20 01:46:59.000000000 +0200
+++ bind-9.7.2-P2/isc-config.sh.in	2010-10-20 14:05:25.423861548 +0200
@@ -20,6 +20,8 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 exec_prefix_set=
+libdir=@libdir@
+includedir=@includedir@
 
 usage()
 {
@@ -115,14 +117,14 @@ if test x"$echo_exec_prefix" = x"true" ;
 	echo $exec_prefix
 fi
 if test x"$echo_cflags" = x"true"; then
-	includes="-I${exec_prefix}/include"
+	includes="-I${includedir}"
 	if test x"$libisc" = x"true"; then
 		includes="$includes @ALWAYS_DEFINES@ @STD_CINCLUDES@ @STD_CDEFINES@ @CCOPT@"
 	fi
 	echo $includes
 fi
 if test x"$echo_libs" = x"true"; then
-	libs=-L${exec_prefix}/lib
+	libs="-L${libdir}"
 	if test x"$liblwres" = x"true" ; then
 		libs="$libs -llwres"
 	fi
diff -up bind-9.7.2-P2/lib/export/dns/Makefile.in.exportlib bind-9.7.2-P2/lib/export/dns/Makefile.in
--- bind-9.7.2-P2/lib/export/dns/Makefile.in.exportlib	2010-06-10 01:49:43.000000000 +0200
+++ bind-9.7.2-P2/lib/export/dns/Makefile.in	2010-10-20 14:08:58.123772859 +0200
@@ -35,9 +35,9 @@ CDEFINES =	-DUSE_MD5 @USE_OPENSSL@ @USE_
 
 CWARNINGS =
 
-ISCLIBS =	../isc/libisc.@A@
+ISCLIBS =	../isc/libisc-export.@A@
 
-ISCDEPLIBS =	../isc/libisc.@A@
+ISCDEPLIBS =	../isc/libisc-export.@A@
 
 LIBS =		@LIBS@
 
@@ -114,29 +114,29 @@ version.@O@: ${srcdir}/version.c
 		-DLIBAGE=${LIBAGE} \
 		-c ${srcdir}/version.c
 
-libdns.@SA@: ${OBJS}
+libdns-export.@SA@: ${OBJS}
 	${AR} ${ARFLAGS} $@ ${OBJS}
 	${RANLIB} $@
 
-libdns.la: ${OBJS}
+libdns-export.la: ${OBJS}
 	${LIBTOOL_MODE_LINK} \
-		${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libdns.la \
+		${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libdns-export.la \
 		-rpath ${export_libdir} \
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
 		${OBJS} ${ISCLIBS} @DNS_CRYPTO_LIBS@ ${LIBS}
 
-timestamp: libdns.@A@
+timestamp: libdns-export.@A@
 	touch timestamp
 
 installdirs:
 	$(SHELL) ${top_srcdir}/mkinstalldirs ${DESTDIR}${export_libdir}
 
 install:: timestamp installdirs
-	${LIBTOOL_MODE_INSTALL} ${INSTALL_DATA} libdns.@A@ \
+	${LIBTOOL_MODE_INSTALL} ${INSTALL_PROGRAM} libdns-export.@A@ \
 	${DESTDIR}${export_libdir}/
 
 clean distclean::
-	rm -f libdns.@A@ timestamp
+	rm -f libdns-export.@A@ timestamp
 	rm -f gen code.h include/dns/enumtype.h include/dns/enumclass.h
 	rm -f include/dns/rdatastruct.h
 
diff -up bind-9.7.2-P2/lib/export/irs/Makefile.in.exportlib bind-9.7.2-P2/lib/export/irs/Makefile.in
--- bind-9.7.2-P2/lib/export/irs/Makefile.in.exportlib	2009-12-06 00:31:40.000000000 +0100
+++ bind-9.7.2-P2/lib/export/irs/Makefile.in	2010-10-20 14:10:39.893717488 +0200
@@ -42,9 +42,9 @@ SRCS =		context.c \
 		gai_sterror.c getaddrinfo.c getnameinfo.c \
 		resconf.c
 
-ISCLIBS =	../isc/libisc.@A@
-DNSLIBS =	../dns/libdns.@A@
-ISCCFGLIBS =	../isccfg/libisccfg.@A@
+ISCLIBS =	../isc/libisc-export.@A@
+DNSLIBS =	../dns/libdns-export.@A@
+ISCCFGLIBS =	../isccfg/libisccfg-export.@A@
 
 LIBS =		@LIBS@
 
@@ -61,26 +61,26 @@ version.@O@: ${srcdir}/version.c
 		-DLIBAGE=${LIBAGE} \
 		-c ${srcdir}/version.c
 
-libirs.@SA@: ${OBJS} version.@O@
+libirs-export.@SA@: ${OBJS} version.@O@
 	${AR} ${ARFLAGS} $@ ${OBJS} version.@O@
 	${RANLIB} $@
 
-libirs.la: ${OBJS} version.@O@
+libirs-export.la: ${OBJS} version.@O@
 	${LIBTOOL_MODE_LINK} \
-		${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libirs.la \
+		${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libirs-export.la \
 		-rpath ${export_libdir} \
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
 		${OBJS} version.@O@ ${LIBS} ${ISCCFGLIBS} ${DNSLIBS} ${ISCLIBS}
 
-timestamp: libirs.@A@
+timestamp: libirs-export.@A@
 	touch timestamp
 
 installdirs:
 	$(SHELL) ${top_srcdir}/mkinstalldirs ${DESTDIR}${export_libdir}
 
 install:: timestamp installdirs
-	${LIBTOOL_MODE_INSTALL} ${INSTALL_DATA} libirs.@A@ \
+	${LIBTOOL_MODE_INSTALL} ${INSTALL_PROGRAM} libirs-export.@A@ \
 	${DESTDIR}${export_libdir}/
 
 clean distclean::
-	rm -f libirs.@A@ libirs.la timestamp
+	rm -f libirs-export.@A@ libirs-export.la timestamp
diff -up bind-9.7.2-P2/lib/export/isccfg/Makefile.in.exportlib bind-9.7.2-P2/lib/export/isccfg/Makefile.in
--- bind-9.7.2-P2/lib/export/isccfg/Makefile.in.exportlib	2009-12-06 00:31:41.000000000 +0100
+++ bind-9.7.2-P2/lib/export/isccfg/Makefile.in	2010-10-20 14:10:14.593741247 +0200
@@ -30,11 +30,11 @@ CINCLUDES =	-I. ${DNS_INCLUDES} -I${expo
 CDEFINES =
 CWARNINGS =
 
-ISCLIBS =	../isc/libisc.@A@
-DNSLIBS =	../dns/libdns.@A@
+ISCLIBS =	../isc/libisc-export.@A@
+DNSLIBS =	../dns/libdns-export.@A@
 
 ISCDEPLIBS =	../../lib/isc/libisc.@A@
-ISCCFGDEPLIBS =	libisccfg.@A@
+ISCCFGDEPLIBS =	libisccfg-export.@A@
 
 LIBS =		@LIBS@
 
@@ -58,26 +58,26 @@ version.@O@: ${srcdir}/version.c
 		-DLIBAGE=${LIBAGE} \
 		-c ${srcdir}/version.c
 
-libisccfg.@SA@: ${OBJS}
+libisccfg-export.@SA@: ${OBJS}
 	${AR} ${ARFLAGS} $@ ${OBJS}
 	${RANLIB} $@
 
-libisccfg.la: ${OBJS}
+libisccfg-export.la: ${OBJS}
 	 ${LIBTOOL_MODE_LINK} \
-		${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libisccfg.la \
+		${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libisccfg-export.la \
 		-rpath ${export_libdir} \
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
 		${OBJS} ${LIBS} ${DNSLIBS} ${ISCLIBS}
 
-timestamp: libisccfg.@A@
+timestamp: libisccfg-export.@A@
 	touch timestamp
 
 installdirs:
 	$(SHELL) ${top_srcdir}/mkinstalldirs ${DESTDIR}${export_libdir}
 
 install:: timestamp installdirs
-	${LIBTOOL_MODE_INSTALL} ${INSTALL_DATA} libisccfg.@A@ \
+	${LIBTOOL_MODE_INSTALL} ${INSTALL_PROGRAM} libisccfg-export.@A@ \
 	${DESTDIR}${export_libdir}/
 
 clean distclean::
-	rm -f libisccfg.@A@ timestamp
+	rm -f libisccfg-export.@A@ timestamp
diff -up bind-9.7.2-P2/lib/export/isc/Makefile.in.exportlib bind-9.7.2-P2/lib/export/isc/Makefile.in
--- bind-9.7.2-P2/lib/export/isc/Makefile.in.exportlib	2010-06-10 01:49:43.000000000 +0200
+++ bind-9.7.2-P2/lib/export/isc/Makefile.in	2010-10-20 14:05:25.433861543 +0200
@@ -101,6 +101,10 @@ SRCS =		@ISC_EXTRA_SRCS@ \
 
 LIBS =		@LIBS@
 
+# Note: the order of SUBDIRS is important.
+# Attempt to disable parallel processing.
+.NOTPARALLEL:
+.NO_PARALLEL:
 SUBDIRS =	include unix nls @ISC_THREAD_DIR@
 TARGETS =	timestamp
 
@@ -114,26 +118,26 @@ version.@O@: ${srcdir}/version.c
 		-DLIBAGE=${LIBAGE} \
 		-c ${srcdir}/version.c
 
-libisc.@SA@: ${OBJS}
+libisc-export.@SA@: ${OBJS}
 	${AR} ${ARFLAGS} $@ ${OBJS}
 	${RANLIB} $@
 
-libisc.la: ${OBJS}
+libisc-export.la: ${OBJS}
 	${LIBTOOL_MODE_LINK} \
-		${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libisc.la \
+		${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libisc-export.la \
 		-rpath ${export_libdir} \
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
 		${OBJS} ${LIBS}
 
-timestamp: libisc.@A@
+timestamp: libisc-export.@A@
 	touch timestamp
 
 installdirs:
 	$(SHELL) ${top_srcdir}/mkinstalldirs ${DESTDIR}${export_libdir}
 
 install:: timestamp installdirs
-	${LIBTOOL_MODE_INSTALL} ${INSTALL_DATA} libisc.@A@ \
+	${LIBTOOL_MODE_INSTALL} ${INSTALL_PROGRAM} libisc-export.@A@ \
 	${DESTDIR}${export_libdir}
 
 clean distclean::
-	rm -f libisc.@A@ libisc.la timestamp
+	rm -f libisc-export.@A@ libisc-export.la timestamp
diff -up bind-9.7.2-P2/lib/export/samples/Makefile.in.exportlib bind-9.7.2-P2/lib/export/samples/Makefile.in
--- bind-9.7.2-P2/lib/export/samples/Makefile.in.exportlib	2009-12-06 00:31:41.000000000 +0100
+++ bind-9.7.2-P2/lib/export/samples/Makefile.in	2010-10-20 14:05:25.433861543 +0200
@@ -30,15 +30,15 @@ CINCLUDES =	-I${srcdir}/include -I../dns
 CDEFINES =
 CWARNINGS =
 
-DNSLIBS =	../dns/libdns.@A@ @DNS_CRYPTO_LIBS@
-ISCLIBS =	../isc/libisc.@A@
-ISCCFGLIBS =	../isccfg/libisccfg.@A@
-IRSLIBS =	../irs/libirs.@A@
-
-DNSDEPLIBS =	../dns/libdns.@A@
-ISCDEPLIBS =	../isc/libisc.@A@
-ISCCFGDEPLIBS =	../isccfg/libisccfg.@A@
-IRSDEPLIBS =	../irs/libirs.@A@
+DNSLIBS =	../dns/libdns-export.@A@ @DNS_CRYPTO_LIBS@
+ISCLIBS =	../isc/libisc-export.@A@
+ISCCFGLIBS =	../isccfg/libisccfg-export.@A@
+IRSLIBS =	../irs/libirs-export.@A@
+
+DNSDEPLIBS =	../dns/libdns-export.@A@
+ISCDEPLIBS =	../isc/libisc-export.@A@
+ISCCFGDEPLIBS =	../isccfg/libisccfg-export.@A@
+IRSDEPLIBS =	../irs/libirs-export.@A@
 
 DEPLIBS =	${DNSDEPLIBS} ${ISCCFGDEPLIBS} ${ISCDEPLIBS}
 
