diff -up bind-9.9.0b2/lib/dns/include/dns/Makefile.in.forward bind-9.9.0b2/lib/dns/include/dns/Makefile.in
--- bind-9.9.0b2/lib/dns/include/dns/Makefile.in.forward	2011-12-07 16:17:50.822438237 +0100
+++ bind-9.9.0b2/lib/dns/include/dns/Makefile.in	2011-12-07 16:18:00.374455261 +0100
@@ -31,7 +31,8 @@ HEADERS =	acl.h adb.h byaddr.h cache.h c
 		rdataslab.h rdatatype.h request.h resolver.h result.h \
 		rootns.h rpz.h sdb.h sdlz.h secalg.h secproto.h soa.h ssu.h \
 		tcpmsg.h time.h tkey.h tsig.h ttl.h types.h \
-		validator.h version.h view.h xfrin.h zone.h zonekey.h zt.h
+		validator.h version.h view.h xfrin.h zone.h zonekey.h zt.h \
+		forward.h
 
 GENHEADERS =	enumclass.h enumtype.h rdatastruct.h
 
