diff -up bind-9.5.0b1/lib/dns/Makefile.in.parallel bind-9.5.0b1/lib/dns/Makefile.in
--- bind-9.5.0b1/lib/dns/Makefile.in.parallel	2008-01-17 18:27:38.000000000 +0100
+++ bind-9.5.0b1/lib/dns/Makefile.in	2008-01-17 18:27:45.000000000 +0100
@@ -19,10 +19,6 @@ srcdir =	@srcdir@
 VPATH =		@srcdir@
 top_srcdir =	@top_srcdir@
 
-# Attempt to disable parallel processing.
-.NOTPARALLEL:
-.NO_PARALLEL:
-
 @BIND9_VERSION@
 
 @LIBDNS_API@
