diff -up bind-9.6.0b1/configure.in.libtool2 bind-9.6.0b1/configure.in
--- bind-9.6.0b1/configure.in.libtool2	2008-11-24 13:05:37.000000000 +0100
+++ bind-9.6.0b1/configure.in	2008-11-24 13:05:56.000000000 +0100
@@ -27,6 +27,8 @@ AC_CONFIG_SUBDIRS(lib/bind)
 
 AC_CONFIG_HEADER(config.h)
 
+AC_CONFIG_MACRO_DIR([m4])
+
 AC_CANONICAL_HOST
 
 AC_PROG_MAKE_SET
diff -up bind-9.6.0b1/Makefile.in.libtool2 bind-9.6.0b1/Makefile.in
