diff -up bind-9.7.0/configure.in.rh478718 bind-9.7.0/configure.in
--- bind-9.7.0/configure.in.rh478718	2010-03-01 14:50:02.331207076 +0100
+++ bind-9.7.0/configure.in	2010-03-01 14:50:21.501207488 +0100
@@ -2540,6 +2540,10 @@ main() {
 	AC_MSG_RESULT($arch)
 fi
 
+if test ! "$arch" = "x86_64" -a "$have_xaddq" = "yes"; then
+	AC_MSG_ERROR([XADDQ present but disabled by Fedora patch!])
+fi
+
 if test "$have_atomic" = "yes"; then
 	AC_MSG_CHECKING([compiler support for inline assembly code])
 
diff -up bind-9.7.0/lib/isc/include/isc/platform.h.in.rh478718 bind-9.7.0/lib/isc/include/isc/platform.h.in
--- bind-9.7.0/lib/isc/include/isc/platform.h.in.rh478718	2010-03-01 14:50:31.421207522 +0100
+++ bind-9.7.0/lib/isc/include/isc/platform.h.in	2010-03-01 14:50:40.313707286 +0100
@@ -255,7 +255,11 @@
  * If the "xaddq" operation (64bit xadd) is available on this architecture,
  * ISC_PLATFORM_HAVEXADDQ will be defined.
  */
-@ISC_PLATFORM_HAVEXADDQ@
+#ifdef __x86_64__
+#define ISC_PLATFORM_HAVEXADDQ 1
+#else
+#undef ISC_PLATFORM_HAVEXADDQ
+#endif
 
 /*
  * If the "atomic swap" operation is available on this architecture,
