diff -up bind-9.9.2/lib/isc/unix/include/isc/Makefile.in.stat bind-9.9.2/lib/isc/unix/include/isc/Makefile.in
--- bind-9.9.2/lib/isc/unix/include/isc/Makefile.in.stat	2012-10-11 15:52:08.597838751 +0200
+++ bind-9.9.2/lib/isc/unix/include/isc/Makefile.in	2012-10-11 15:52:16.339836619 +0200
@@ -21,7 +21,7 @@ top_srcdir =	@top_srcdir@
 
 @BIND9_VERSION@
 
-HEADERS =	dir.h int.h net.h netdb.h offset.h stdtime.h \
+HEADERS =	dir.h int.h net.h netdb.h offset.h stat.h stdtime.h \
 		syslog.h time.h
 
 SUBDIRS =
