diff -up bind-9.9.2/bin/tests/system/Makefile.in.buildfix bind-9.9.2/bin/tests/system/Makefile.in
--- bind-9.9.2/bin/tests/system/Makefile.in.buildfix	2012-09-27 02:35:19.000000000 +0200
+++ bind-9.9.2/bin/tests/system/Makefile.in	2012-10-11 13:33:21.245382076 +0200
@@ -21,7 +21,7 @@ top_srcdir =	@top_srcdir@
 
 @BIND9_MAKE_INCLUDES@
 
-SUBDIRS =	dlzexternal filter-aaaa lwresd rpz rsabigexponent tkey tsiggss
+SUBDIRS =	filter-aaaa lwresd rpz rsabigexponent tkey tsiggss
 TARGETS =
 
 @BIND9_MAKE_RULES@
