diff -up bind-9.5.1b1/bin/dig/dighost.c.rh461409 bind-9.5.1b1/bin/dig/dighost.c
--- bind-9.5.1b1/bin/dig/dighost.c.rh461409	2008-09-16 14:04:03.000000000 +0200
+++ bind-9.5.1b1/bin/dig/dighost.c	2008-09-16 14:06:06.000000000 +0200
@@ -3665,6 +3665,15 @@ output_filter (isc_buffer_t *buffer, uns
 	(void) strcpy (tmp1, tmp2);
 	free (tmp2);
 
+	tmp2 = stringprep_utf8_to_locale (tmp1);
+	if (tmp2 == NULL) {
+		debug ("output_filter: stringprep_utf8_to_locale failed");
+		return ISC_R_SUCCESS;
+	}
+
+	(void) strcpy (tmp1, tmp2);
+	free (tmp2);
+
         tolen = strlen (tmp1);
         if (absolute && !end_with_dot && tmp1[tolen - 1] == '.')
                 tolen--;
