diff -up bind-9.7.0b1/configure.in.dist-pkcs11 bind-9.7.0b1/configure.in
--- bind-9.7.0b1/configure.in.dist-pkcs11	2009-11-03 11:37:25.481430279 +0100
+++ bind-9.7.0b1/configure.in	2009-11-03 11:41:04.573930858 +0100
@@ -721,15 +721,16 @@ AC_ARG_WITH(pkcs11,
 case "$use_pkcs11" in
 	no|'')
 		AC_MSG_RESULT(disabled)
-		USE_PKCS11=''
 		PKCS11_TOOLS=''
 		;;
 	yes|*)
 		AC_MSG_RESULT(using OpenSSL with PKCS11 support)
-		USE_PKCS11='-DUSE_PKCS11'
 		PKCS11_TOOLS=pkcs11
 		;;
 esac
+# We don't want to automatically load engine because it needs to have openct,
+# opencryptoki and friends installed.
+USE_PKCS11=''
 AC_SUBST(USE_PKCS11)
 AC_SUBST(PKCS11_TOOLS)
 
