--- panda/panda.py~	2013-12-22 18:35:44.986644221 +0100
+++ panda/panda.py	2013-12-23 00:47:08.696656055 +0100
@@ -7,6 +7,7 @@
 import pisi
 import shutil
 from subprocess import call
+from distutils import util
 
 sysdir = "/sys/bus/pci/devices/"
 driversDB = "/usr/share/X11/DriversDB"
@@ -30,13 +31,15 @@
         self.driver_packages = {"fglrx": ["module-fglrx",
                                      "module-pae-fglrx",
                                      "module-fglrx-userspace",
-                                     "xorg-video-fglrx"],
+                                     "xorg-video-fglrx",
+                                     "%s"],
                            "nvidia-current": ["module-nvidia-current",
                                               "module-pae-nvidia-current",
                                               "module-nvidia-current-userspace",
                                               "xorg-video-nvidia-current",
                                               "nvidia-xconfig",
-                                              "nvidia-settings"] ,
+                                              "nvidia-settings"
+                                              "%s"],
                            "nvidia96": ["module-nvidia96",
                                         "module-pae-nvidia96",
                                         "module-nvidia96-userspace",
@@ -53,7 +56,8 @@
                                          "module-nvidia304-userspace",
                                          "xorg-video-nvidia304",
                                          "nvidia-xconfig",
-                                         "nvidia-settings"]}
+                                         "nvidia-settings"]
+                                % ("xorg-video-fglrx-32bit", "xorg-video-nvidia-current-32bit" if util.get_platform() == "linux-x86_64" else "", "")}
 
     def __get_primary_driver(self):
         '''Get driver name for the working primary device'''
