--- panda.py~	2013-12-22 18:35:44.986644221 +0100
+++ panda.py	2013-12-23 10:50:00.433218873 +0100
@@ -7,6 +7,7 @@
 import pisi
 import shutil
 from subprocess import call
+from distutils import util
 
 sysdir = "/sys/bus/pci/devices/"
 driversDB = "/usr/share/X11/DriversDB"
@@ -36,7 +37,7 @@
                                               "module-nvidia-current-userspace",
                                               "xorg-video-nvidia-current",
                                               "nvidia-xconfig",
-                                              "nvidia-settings"] ,
+                                              "nvidia-settings"],
                            "nvidia96": ["module-nvidia96",
                                         "module-pae-nvidia96",
                                         "module-nvidia96-userspace",
@@ -54,6 +55,9 @@
                                          "xorg-video-nvidia304",
                                          "nvidia-xconfig",
                                          "nvidia-settings"]}
+        if util.get_platform() == "linux-x86_64":
+            self.driver_packages["fglrx"].append("xorg-video-fglrx-32bit")
+            self.driver_packages["nvidia-current"].append("xorg-video-nvidia-current-32bit")
 
     def __get_primary_driver(self):
         '''Get driver name for the working primary device'''
