--- panda.py~	2013-06-26 23:14:48.709209877 +0200
+++ panda.py	2013-06-26 23:15:23.467208607 +0200
@@ -269,10 +269,10 @@
 
         with open(grub_default_file) as grub_default:
             for line in grub_default:
-                if "GRUB_CMDLINE_LINUX=" in line:
+                if "GRUB_CMDLINE_LINUX_DEFAULT=" in line:
                     params = {} if line.startswith("#") else dict((k, v.split(','))
                                                                   for k,v in (item.split('=') if "=" in item else (item, '')
-                                                                              for item in line.replace('"', '').split("LINUX=")[1].split()))
+                                                                              for item in line.replace('"', '').replace("'", "").split("DEFAULT=")[1].split()))
                     old_line = " ".join(["%s%s%s" % (k, "=" if l[0] else "", ",".join([v for v in l])) for k, l in sorted(params.items())])
                     if arg == "os":
                         try :
@@ -285,10 +285,6 @@
                             if not params["xorg"] or not params["xorg"][0]: del params["xorg"] 
                         except KeyError:
                             pass
-                        try :
-                            del params["vga"] 
-                        except KeyError:
-                            pass
                         status = "os"
 
                     elif arg == "vendor":
@@ -302,7 +298,6 @@
                             if not params["xorg"] or not params["xorg"][0]: del params["xorg"]
                         except KeyError:
                             pass
-                        if self.driver_name in ["nvidia-current"]: params["vga"] = ["794"]
                         status = "vendor"
 
                     elif arg == "generic":
@@ -430,4 +425,3 @@
 #    print p.update_grub_entries("vendor")
     print p.update_system_files("vendor")
     print p.get_needed_driver_packages(installable=False)
-
--- panda.py~	2013-07-12 12:33:05.029649344 +0200
+++ panda.py	2013-07-12 12:33:09.728649172 +0200
@@ -309,7 +309,7 @@
 
                     new_line = " ".join(["%s%s" % ("%s=" % k if l[0] else k, ",".join([v for v in l])) for k, l in sorted(params.items())])
                     configured = old_line != new_line
-                    new_line = 'GRUB_CMDLINE_LINUX="%s"\n' % new_line
+                    new_line = 'GRUB_CMDLINE_LINUX_DEFAULT="%s"\n' % new_line
                     grub_tmp.write(new_line)
                 else:
                     grub_tmp.write(line)
