--- panda.py~   2011-02-10 08:53:21.000000000 +0200
+++ panda.py    2013-03-17 06:12:39.322933331 +0200
@@ -10,9 +10,9 @@
 sysdir = "/sys/bus/pci/devices/"
 driversDB = "/usr/share/X11/DriversDB"
 
-grub_file = "/boot/grub/grub.conf"
-grub_new = "/boot/grub/grub.conf.new"
-grub_back = "/boot/grub/grub.conf.back"
+grub_file = "/boot/grub2/grub.conf"
+grub_new = "/boot/grub2/grub.conf.new"
+grub_back = "/boot/grub2/grub.conf.back"
 kernel_file = "/etc/kernel/kernel"
 kernel_file_pae = "/etc/kernel/kernel-pae"
 
@@ -215,7 +215,7 @@
                     else:
                         return "os"
 
-        return "Cannot parse grub.conf"
+        return "Cannot parse %s" % grub_file
 
     ######################################
     # Grub parsing and writing functions #
@@ -273,6 +273,8 @@
                     new_line = self.update_parameter_in_line(line, "xorg", xorg_param)
                     new_line = self.update_parameter_in_line(new_line, "nomodeset", nomodeset_param)
                     new_line = self.update_parameter_in_line(new_line, "blacklist", blacklist)
+                    # keep indentation
+                    new_line = line.split("linux")[0] + new_line 
                     grub_tmp.write(new_line)
                     configured = line != new_line
 
