--- panda.py~	2014-04-12 20:47:04.213380223 +0200
+++ panda.py	2014-04-12 20:51:07.980049491 +0200
@@ -30,7 +30,6 @@
         self.os_driver = None
         self.driver_packages = {"fglrx": ["module-fglrx",
                                      "module-pae-fglrx",
-                                     "module-fglrx-userspace",
                                      "xorg-video-fglrx"],
                            "nvidia-current": ["module-nvidia-current",
                                               "module-pae-nvidia-current",
@@ -179,7 +178,6 @@
                 open(nvidia_blacklist_file, "w").write("blacklist nouveau\n")
             elif self.driver_name in ["nvidia-current", "nvidia96", "nvidia173", "nvidia304"] and os.path.isfile(nvidia_blacklist_file):
                 os.remove(nvidia_blacklist_file)
-                
 
         return status
 
