--- libogg-1.1/ogg.m4.orig	2003-03-07 21:44:29.000000000 -0500
+++ libogg-1.1/ogg.m4	2003-12-11 15:08:01.000000000 -0500
@@ -19,7 +19,7 @@
   elif test "x$ogg_prefix" != "x" ; then
     OGG_LIBS="-L$ogg_prefix/lib"
   elif test "x$prefix" != "xNONE" ; then
-    OGG_LIBS="-L$prefix/lib"
+    OGG_LIBS="-L$libdir"
   fi
 
   OGG_LIBS="$OGG_LIBS -logg"
@@ -29,7 +29,7 @@
   elif test "x$ogg_prefix" != "x" ; then
     OGG_CFLAGS="-I$ogg_prefix/include"
   elif test "x$prefix" != "xNONE"; then
-    OGG_CFLAGS="-I$prefix/include"
+    OGG_CFLAGS=""
   fi
 
   AC_MSG_CHECKING(for Ogg)
