Index: ufraw-0.17/ufraw.desktop
===================================================================
--- ufraw-0.17.orig/ufraw.desktop
+++ ufraw-0.17/ufraw.desktop
@@ -15,6 +15,7 @@ GenericName[pl]=Konwerter zdjęć RAW
 GenericName[pt]=Conversor de imagens cruas
 GenericName[ru]=Проявитель цифровых негативов
 GenericName[sr]=Преводилац Raw слика
+GenericName[tr]=Ham Görüntü Dönüştürücü
 GenericName[sr@latin]=Prevodilac Raw slika
 GenericName[zh_CN]=Raw 图像转换器
 
@@ -34,6 +35,7 @@ Comment[pt]=Programa para ler e manipula
 Comment[ru]=Проявка цифровых негативов и сохранение в TIFF/JPEG
 Comment[sr]=Алат за читање и обраду raw слика из дигиталних камера
 Comment[sr@latin]=Alat za čitanje i obradu raw slika iz digitalnih kamera
+Comment[tr]=Sayısal fotoğraf makinesi ham görüntülerini işlemek için araç
 Comment[zh_CN]=一个从数码相机中读取并处理原始图像的工具
 
 Version=1.0
