diff -up ufraw-0.18/ufraw_ufraw.c.darkframe ufraw-0.18/ufraw_ufraw.c
--- ufraw-0.18/ufraw_ufraw.c.darkframe	2011-02-20 08:56:15.000000000 +0100
+++ ufraw-0.18/ufraw_ufraw.c	2011-03-10 11:48:13.936442587 +0100
@@ -293,6 +293,8 @@ int ufraw_load_darkframe(ufraw_data *uf)
     }
     dark->conf = g_new(conf_data, 1);
     conf_init(dark->conf);
+    /* initialize ufobject member */
+    dark->conf->ufobject = ufraw_image_new();
     /* disable all auto settings on darkframe */
     dark->conf->autoExposure = disabled_state;
     dark->conf->autoBlack = disabled_state;
