diff -Nuar Thunar-1.0.1.orig/plugins/thunar-apr/Makefile.in Thunar-1.0.1/plugins/thunar-apr/Makefile.in
--- Thunar-1.0.1.orig/plugins/thunar-apr/Makefile.in	2009-06-08 15:21:41.000000000 +0300
+++ Thunar-1.0.1/plugins/thunar-apr/Makefile.in	2009-06-08 16:20:54.000000000 +0300
@@ -374,10 +374,12 @@
 	-avoid-version							\
 	-export-dynamic							\
 	-module								\
-	$(PLATFORM_LDFLAGS)
+	$(PLATFORM_LDFLAGS) `pkg-config --libs gtk+-2.0`
 
 thunar_apr_la_LIBADD = \
-	$(EXIF_LIBS)
+	$(top_builddir)/thunarx/libthunarx-$(THUNAR_VERSION_API).la \
+	$(EXIF_LIBS) \
+	$(EXO_LIBS)
 
 EXTRA_DIST = \
 	README
diff -Nuar Thunar-1.0.1.orig/plugins/thunar-sbr/Makefile.in Thunar-1.0.1/plugins/thunar-sbr/Makefile.in
--- Thunar-1.0.1.orig/plugins/thunar-sbr/Makefile.in	2009-06-08 15:21:41.000000000 +0300
+++ Thunar-1.0.1/plugins/thunar-sbr/Makefile.in	2009-06-08 17:00:43.000000000 +0300
@@ -384,11 +384,14 @@
 	-avoid-version							\
 	-export-dynamic							\
 	-module								\
-	$(PLATFORM_LDFLAGS)
+	$(PLATFORM_LDFLAGS) `pkg-config --libs gtk+-2.0`
 
 thunar_sbr_la_LIBADD = \
+	$(top_builddir)/thunarx/libthunarx-$(THUNAR_VERSION_API).la \
+	$(top_builddir)/thunar-vfs/libthunar-vfs-$(THUNAR_VERSION_API).la \
 	$(PCRE_LIBS)							\
-	$(EXIF_LIBS)
+	$(EXIF_LIBS) \
+	$(EXO_LIBS)
 
 EXTRA_DIST = \
 	README
diff -Nuar Thunar-1.0.1.orig/plugins/thunar-uca/Makefile.in Thunar-1.0.1/plugins/thunar-uca/Makefile.in
--- Thunar-1.0.1.orig/plugins/thunar-uca/Makefile.in	2009-06-08 15:21:41.000000000 +0300
+++ Thunar-1.0.1/plugins/thunar-uca/Makefile.in	2009-06-08 17:24:55.000000000 +0300
@@ -56,7 +56,11 @@
 	"$(DESTDIR)$(defaultsdir)"
 extensionsLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(extensions_LTLIBRARIES)
-thunar_uca_la_LIBADD =
+thunar_uca_la_LIBADD = \
+	$(top_builddir)/thunar-vfs/libthunar-vfs-$(THUNAR_VERSION_API).la \
+	$(top_builddir)/thunarx/libthunarx-$(THUNAR_VERSION_API).la \
+	$(EXO_LIBS)
+
 am_thunar_uca_la_OBJECTS = thunar_uca_la-thunar-uca-chooser.lo \
 	thunar_uca_la-thunar-uca-context.lo \
 	thunar_uca_la-thunar-uca-editor.lo \
@@ -381,7 +385,7 @@
 	-avoid-version							\
 	-export-dynamic							\
 	-module								\
-	$(PLATFORM_LDFLAGS)
+	$(PLATFORM_LDFLAGS) `pkg-config --libs gtk+-2.0`
 
 defaultsdir = $(sysconfdir)/xdg/Thunar
 defaults_in_files = uca.xml.in
diff -Nuar Thunar-1.0.1.orig/plugins/thunar-wallpaper/Makefile.in Thunar-1.0.1/plugins/thunar-wallpaper/Makefile.in
--- Thunar-1.0.1.orig/plugins/thunar-wallpaper/Makefile.in	2009-06-08 15:21:41.000000000 +0300
+++ Thunar-1.0.1/plugins/thunar-wallpaper/Makefile.in	2009-06-08 17:40:39.000000000 +0300
@@ -52,7 +52,11 @@
 am__installdirs = "$(DESTDIR)$(extensionsdir)"
 extensionsLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(extensions_LTLIBRARIES)
-thunar_wallpaper_plugin_la_LIBADD =
+thunar_wallpaper_plugin_la_LIBADD = \
+	$(top_builddir)/thunar-vfs/libthunar-vfs-$(THUNAR_VERSION_API).la \
+	$(top_builddir)/thunarx/libthunarx-$(THUNAR_VERSION_API).la \
+	$(LIBX11_LIBS)
+	
 am_thunar_wallpaper_plugin_la_OBJECTS =  \
 	thunar_wallpaper_plugin_la-twp-provider.lo \
 	thunar_wallpaper_plugin_la-thunar-wallpaper-plugin.lo
@@ -365,7 +369,7 @@
 	-module									\
 	$(PLATFORM_LDFLAGS)						\
 	$(THUNARX_LDFLAGS)						\
-	$(THUNAR_VFS_LDFLAGS)
+	$(THUNAR_VFS_LDFLAGS) `pkg-config --libs gtk+-2.0`
 
 all: all-am
 
diff -Nuar Thunar-1.0.1.orig/thunar-vfs/Makefile.in Thunar-1.0.1/thunar-vfs/Makefile.in
--- Thunar-1.0.1.orig/thunar-vfs/Makefile.in	2009-06-08 15:21:41.000000000 +0300
+++ Thunar-1.0.1/thunar-vfs/Makefile.in	2009-06-08 15:31:06.000000000 +0300
@@ -649,7 +649,7 @@
 	$(PLATFORM_LDFLAGS)
 
 libthunar_vfs_1_la_LIBADD = $(EXO_LIBS) $(GTHREAD_LIBS) $(LIBFAM_LIBS) \
-	$(LIBJPEG_LIBS) $(LIBPNG_LIBS) $(LIBSTARTUP_NOTIFICATION_LIBS) \
+	$(LIBJPEG_LIBS) $(LIBPNG_LIBS) $(LIBSTARTUP_NOTIFICATION_LIBS) $(LIBX11_LIBS) \
 	$(am__append_5)
 thunar_vfs_mime_cleaner_1_SOURCES = \
 	thunar-vfs-mime-cleaner.c
