Index: Thunar-1.4.0/plugins/thunar-uca/uca.xml.in
===================================================================
--- Thunar-1.4.0.orig/plugins/thunar-uca/uca.xml.in
+++ Thunar-1.4.0/plugins/thunar-uca/uca.xml.in
@@ -36,7 +36,7 @@
 <actions>
 
   <action>
-    <icon>Terminal</icon>
+    <icon>utilities-terminal</icon>
     <patterns>*</patterns>
     <_name>Open Terminal Here</_name>
     <command>exo-open --working-directory %f --launch TerminalEmulator</command>
Index: Thunar-1.4.0/thunar/thunar-preferences-dialog.c
===================================================================
--- Thunar-1.4.0.orig/thunar/thunar-preferences-dialog.c
+++ Thunar-1.4.0/thunar/thunar-preferences-dialog.c
@@ -187,7 +187,7 @@ thunar_preferences_dialog_init (ThunarPr
 
   /* configure the dialog properties */
   gtk_dialog_set_has_separator (GTK_DIALOG (dialog), FALSE);
-  gtk_window_set_icon_name (GTK_WINDOW (dialog), "xfce-filemanager");
+  gtk_window_set_icon_name (GTK_WINDOW (dialog), "system-file-manager");
   gtk_window_set_resizable (GTK_WINDOW (dialog), FALSE);
   gtk_window_set_title (GTK_WINDOW (dialog), _("File Manager Preferences"));
 
