diff --git a/plugins/thunar-tpa/thunar-tpa-icon.c b/plugins/thunar-tpa/thunar-tpa-icon.c
index da38304..740000a 100644
--- a/plugins/thunar-tpa/thunar-tpa-icon.c
+++ b/plugins/thunar-tpa/thunar-tpa-icon.c
@@ -175,7 +175,7 @@ thunar_tpa_icon_init (ThunarTpaIcon *icon)
   gtk_widget_show (icon->button);
 
   /* setup the image for the trash icon */
-  icon->image = gtk_image_new_from_icon_name ("gnome-fs-trash-empty", GTK_ICON_SIZE_BUTTON);
+  icon->image = gtk_image_new_from_icon_name ("user-trash", GTK_ICON_SIZE_BUTTON);
   gtk_container_add (GTK_CONTAINER (icon->button), icon->image);
   gtk_widget_show (icon->image);
 
@@ -287,7 +287,7 @@ thunar_tpa_icon_state (ThunarTpaIcon *icon,
   gtk_tooltips_set_tip (icon->tooltips, icon->button, full ? _("Trash contains files") : _("Trash is empty"), NULL);
 
   /* setup the appropriate icon */
-  gtk_image_set_from_icon_name (GTK_IMAGE (icon->image), full ? "gnome-fs-trash-full" : "gnome-fs-trash-empty", GTK_ICON_SIZE_BUTTON);
+  gtk_image_set_from_icon_name (GTK_IMAGE (icon->image), full ? "user-trash-full" : "user-trash", GTK_ICON_SIZE_BUTTON);
 
   /* apply the new state */
   icon->full = full;
diff --git a/plugins/thunar-tpa/thunar-tpa.desktop.in.in b/plugins/thunar-tpa/thunar-tpa.desktop.in.in
index 6d848f6..f647521 100644
--- a/plugins/thunar-tpa/thunar-tpa.desktop.in.in
+++ b/plugins/thunar-tpa/thunar-tpa.desktop.in.in
@@ -3,6 +3,6 @@ Type=X-XFCE-PanelPlugin
 Encoding=UTF-8
 _Name=Trash Applet
 _Comment=Display the trash can
-Icon=gnome-fs-trash-empty
+Icon=user-trash
 X-XFCE-Exec=@libexecdir@/xfce4/panel-plugins/thunar-tpa
 
diff --git a/thunar-vfs/thunar-vfs-io-local.c b/thunar-vfs/thunar-vfs-io-local.c
index 615a36c..e2d75e7 100644
--- a/thunar-vfs/thunar-vfs-io-local.c
+++ b/thunar-vfs/thunar-vfs-io-local.c
@@ -390,7 +390,7 @@ _thunar_vfs_io_local_get_info (ThunarVfsPath *path,
       if (G_UNLIKELY (filename[0] == G_DIR_SEPARATOR && filename[1] == '\0'))
         {
           /* root folder gets a special custom icon... */
-          info->custom_icon = g_strdup ("gnome-dev-harddisk");
+          info->custom_icon = g_strdup ("drive-harddisk");
 
           /* ...and a special display name */
           info->display_name = g_strdup (_("File System"));
diff --git a/thunar-vfs/thunar-vfs-io-trash.c b/thunar-vfs/thunar-vfs-io-trash.c
index 8ca0370..29f333d 100644
--- a/thunar-vfs/thunar-vfs-io-trash.c
+++ b/thunar-vfs/thunar-vfs-io-trash.c
@@ -1119,7 +1119,7 @@ _thunar_vfs_io_trash_get_info (ThunarVfsPath *path,
       info->ctime = mtime;
       info->mime_info = thunar_vfs_mime_info_ref (_thunar_vfs_mime_inode_directory);
       info->path = thunar_vfs_path_ref (_thunar_vfs_path_trash_root);
-      info->custom_icon = g_strdup (empty ? "gnome-fs-trash-empty" : "gnome-fs-trash-full");
+      info->custom_icon = g_strdup (empty ? "user-trash" : "user-trash-full");
       info->display_name = g_strdup (_("Trash"));
       info->ref_count = 1;
     }
diff --git a/thunar-vfs/thunar-vfs-mime-info.c b/thunar-vfs/thunar-vfs-mime-info.c
index 7c8fe0c..c3c07e1 100644
--- a/thunar-vfs/thunar-vfs-mime-info.c
+++ b/thunar-vfs/thunar-vfs-mime-info.c
@@ -47,18 +47,18 @@ static const struct
   const gchar icon[19];
 } GNOME_INODE_ICONNAMES[] =
 {
-  { "blockdevice", "gnome-fs-blockdev"  },
-  { "chardevice",  "gnome-fs-chardev"   },
-  { "directory",   "gnome-fs-directory" },
-  { "fifo",        "gnome-fs-fifo"      },
-  { "socket",      "gnome-fs-socket"    },
+  { "blockdevice", "inode-blockdev"  },
+  { "chardevice",  "inode-chardev"   },
+  { "directory",   "folder" },
+  { "fifo",        "inode-fifo"      },
+  { "socket",      "inode-socket"    },
 };
 
 /* first fallback gnome icon name */
 static const gchar GNOME_FS_REGULAR[] = "gnome-fs-regular";
 
 /* second fallback gnome icon name */
-static const gchar GNOME_MIME_APPLICATION_OCTET_STREAM[] = "gnome-mime-application-octet-stream";
+static const gchar GNOME_MIME_APPLICATION_OCTET_STREAM[] = "application-x-executable";
 
 /* generic standard icon names (Icon Naming Spec 0.8) */
 static const gchar XDG_GENERIC_ICONNAMES[][18] =
diff --git a/thunar-vfs/thunar-vfs-volume.c b/thunar-vfs/thunar-vfs-volume.c
index 95bdc99..9d5661d 100644
--- a/thunar-vfs/thunar-vfs-volume.c
+++ b/thunar-vfs/thunar-vfs-volume.c
@@ -398,83 +398,76 @@ thunar_vfs_volume_lookup_icon_name (ThunarVfsVolume *volume,
   kind = thunar_vfs_volume_get_kind (volume);
   switch (kind)
     {
-cdrom:
-    case THUNAR_VFS_VOLUME_KIND_CDROM:
-      if (gtk_icon_theme_has_icon (icon_theme, "gnome-dev-cdrom"))
-        return "gnome-dev-cdrom";
-      break;
-
-    case THUNAR_VFS_VOLUME_KIND_CDR:
-      if (gtk_icon_theme_has_icon (icon_theme, "gnome-dev-disc-cdr"))
-        return "gnome-dev-disc-cdr";
-      goto cdrom;
-
-    case THUNAR_VFS_VOLUME_KIND_CDRW:
-      if (gtk_icon_theme_has_icon (icon_theme, "gnome-dev-disc-cdrw"))
-        return "gnome-dev-disc-cdrw";
-      goto cdrom;
-
-dvdrom:
-    case THUNAR_VFS_VOLUME_KIND_DVDROM:
-      if (gtk_icon_theme_has_icon (icon_theme, "gnome-dev-disc-dvdrom"))
-        return "gnome-dev-disc-dvdrom";
-      else if (gtk_icon_theme_has_icon (icon_theme, "gnome-dev-dvd"))
-        return "gnome-dev-dvd";
-      goto cdrom;
-
-    case THUNAR_VFS_VOLUME_KIND_DVDRAM:
-      if (gtk_icon_theme_has_icon (icon_theme, "gnome-dev-disc-dvdram"))
-        return "gnome-dev-disc-dvdram";
+    case THUNAR_VFS_VOLUME_KIND_DVDPLUSRW:
+      if (gtk_icon_theme_has_icon (icon_theme, "media-optical-dvd-rw-plus"))
+        return "media-optical-dvd-rw-plus";
+    case THUNAR_VFS_VOLUME_KIND_DVDRW:
+      if (gtk_icon_theme_has_icon (icon_theme, "media-optical-dvd-rw"))
+        return "media-optical-dvd-rw";
       goto dvdrom;
-
-dvdr:
+    case THUNAR_VFS_VOLUME_KIND_DVDPLUSR:
+      if (gtk_icon_theme_has_icon (icon_theme, "media-optical-dvd-r-plus"))
+        return "media-optical-dvd-r-plus";
     case THUNAR_VFS_VOLUME_KIND_DVDR:
-      if (gtk_icon_theme_has_icon (icon_theme, "gnome-dev-disc-dvdr"))
-        return "gnome-dev-disc-dvdr";
+      if (gtk_icon_theme_has_icon (icon_theme, "media-optical-dvd-r"))
+        return "media-optical-dvd-r";
       goto dvdrom;
-
-    case THUNAR_VFS_VOLUME_KIND_DVDRW:
-    case THUNAR_VFS_VOLUME_KIND_DVDPLUSRW:
-      if (gtk_icon_theme_has_icon (icon_theme, "gnome-dev-disc-dvdrw"))
-        return "gnome-dev-disc-dvdrw";
+    case THUNAR_VFS_VOLUME_KIND_DVDRAM:
+      if (gtk_icon_theme_has_icon (icon_theme, "media-optical-dvd-ram"))
+        return "media-optical-dvd-ram";
       goto dvdrom;
-
-    case THUNAR_VFS_VOLUME_KIND_DVDPLUSR:
-      if (gtk_icon_theme_has_icon (icon_theme, "gnome-dev-disc-dvdr-plus"))
-        return "gnome-dev-disc-dvdr-plus";
-      goto dvdr;
+dvdrom:
+    case THUNAR_VFS_VOLUME_KIND_DVDROM:
+      if (gtk_icon_theme_has_icon (icon_theme, "media-optical-dvd"))
+        return "media-optical-dvd";
+      goto optical;
+      
+    case THUNAR_VFS_VOLUME_KIND_CDRW:
+      if (gtk_icon_theme_has_icon (icon_theme, "media-optical-cd-rw"))
+        return "media-optical-cd-rw";
+      goto cdrom;
+    case THUNAR_VFS_VOLUME_KIND_CDR:
+      if (gtk_icon_theme_has_icon (icon_theme, "media-optical-cd-r"))
+        return "media-optical-cd-r";
+      goto cdrom;
+cdrom:
+    case THUNAR_VFS_VOLUME_KIND_CDROM:
+      if (gtk_icon_theme_has_icon (icon_theme, "media-optical-cd"))
+        return "media-optical-cd";
+      goto optical;
+    case THUNAR_VFS_VOLUME_KIND_AUDIO_CD:
+      if (gtk_icon_theme_has_icon (icon_theme, "media-optical-audio"))
+        return "media-optical-audio";
+      goto optical;
+optical:
+      if (gtk_icon_theme_has_icon (icon_theme, "media-optical"))
+        return "media-optical";
+      break;
 
     case THUNAR_VFS_VOLUME_KIND_FLOPPY:
-      if (gtk_icon_theme_has_icon (icon_theme, "gnome-dev-floppy"))
-        return "gnome-dev-floppy";
+      if (gtk_icon_theme_has_icon (icon_theme, "media-floppy"))
+        return "media-floppy";
       break;
-
     case THUNAR_VFS_VOLUME_KIND_HARDDISK:
-      if (gtk_icon_theme_has_icon (icon_theme, "gnome-dev-harddisk"))
-        return "gnome-dev-harddisk";
+      if (gtk_icon_theme_has_icon (icon_theme, "drive-harddisk"))
+        return "drive-harddisk";
       break;
 
     case THUNAR_VFS_VOLUME_KIND_USBSTICK:
-      if (gtk_icon_theme_has_icon (icon_theme, "gnome-dev-removable-usb"))
-        return "gnome-dev-removable-usb";
-      else if (gtk_icon_theme_has_icon (icon_theme, "gnome-dev-harddisk-usb"))
-        return "gnome-dev-harddisk-usb";
-      break;
-
+      if (gtk_icon_theme_has_icon (icon_theme, "drive-removable-media-usb"))
+        return "drive-removable-media-usb";
+      goto removable_media;
     case THUNAR_VFS_VOLUME_KIND_AUDIO_PLAYER:
-      if (gtk_icon_theme_has_icon (icon_theme, "gnome-dev-ipod"))
-        return "gnome-dev-ipod";
-      break;
+      if (gtk_icon_theme_has_icon (icon_theme, "drive-removable-media-ipod"))
+        return "drive-removable-media-ipod";
+      goto removable_media;
 
-    case THUNAR_VFS_VOLUME_KIND_AUDIO_CD:
-      if (gtk_icon_theme_has_icon (icon_theme, "gnome-dev-cdrom-audio"))
-        return "gnome-dev-cdrom-audio";
-      goto cdrom;
       
     case THUNAR_VFS_VOLUME_KIND_MEMORY_CARD:
     case THUNAR_VFS_VOLUME_KIND_REMOVABLE_DISK:
-      if (gtk_icon_theme_has_icon (icon_theme, "gnome-dev-removable"))
-        return "gnome-dev-removable";
+removable_media:
+      if (gtk_icon_theme_has_icon (icon_theme, "drive-removable-media"))
+        return "drive-removable-media";
       break;
 
     default:
diff --git a/thunar/thunar-application.c b/thunar/thunar-application.c
index c30d43a..80c9784 100644
--- a/thunar/thunar-application.c
+++ b/thunar/thunar-application.c
@@ -1166,7 +1166,7 @@ thunar_application_copy_to (ThunarApplication *application,
   _thunar_return_if_fail (THUNAR_IS_APPLICATION (application));
 
   /* launch the operation */
-  thunar_application_launch (application, parent, "stock_folder-copy",
+  thunar_application_launch (application, parent, "gtk-copy",
                              _("Copying files..."), thunar_vfs_copy_files,
                              source_path_list, target_path_list, new_files_closure);
 }
@@ -1199,7 +1199,7 @@ thunar_application_copy_into (ThunarApplication *application,
   _thunar_return_if_fail (target_path != NULL);
 
   /* collect the target paths and launch the job */
-  thunar_application_collect_and_launch (application, parent, "stock_folder-copy",
+  thunar_application_collect_and_launch (application, parent, "gtk-copy",
                                          _("Copying files..."), thunar_vfs_copy_files,
                                          source_path_list, target_path, new_files_closure);
 }
@@ -1233,7 +1233,7 @@ thunar_application_link_into (ThunarApplication *application,
   _thunar_return_if_fail (target_path != NULL);
 
   /* collect the target paths and launch the job */
-  thunar_application_collect_and_launch (application, parent, "stock_link",
+  thunar_application_collect_and_launch (application, parent, "insert-link",
                                          _("Creating symbolic links..."),
                                          thunar_vfs_link_files, source_path_list,
                                          target_path, new_files_closure);
@@ -1273,12 +1273,15 @@ thunar_application_move_into (ThunarApplication *application,
   /* determine the appropriate message text and the icon based on the target_path */
   if (thunar_vfs_path_get_scheme (target_path) == THUNAR_VFS_PATH_SCHEME_TRASH)
     {
-      icon = "gnome-fs-trash-full";
+      icon = "user-trash-full";
       text = _("Moving files into the trash...");
     }
   else
     {
-      icon = "stock_folder-move";
+      /* What kind of icon-name should we use according to the 
+       * icon-naming-spec?
+       */
+      icon = "folder-move";
       text = _("Moving files...");
     }
 
@@ -1393,7 +1396,7 @@ thunar_application_unlink_files (ThunarApplication *application,
       if (G_LIKELY (response == GTK_RESPONSE_YES))
         {
           /* launch the "Delete" operation */
-          thunar_application_launch (application, parent, "stock_delete",
+          thunar_application_launch (application, parent, "edit-delete",
                                       _("Deleting files..."), unlink_stub,
                                       path_list, path_list, NULL);
         }
@@ -1445,7 +1448,7 @@ thunar_application_creat (ThunarApplication *application,
   _thunar_return_if_fail (THUNAR_IS_APPLICATION (application));
 
   /* launch the operation */
-  thunar_application_launch (application, parent, "stock_new",
+  thunar_application_launch (application, parent, "document-new",
                              _("Creating files..."), creat_stub,
                              path_list, path_list, new_files_closure);
 }
@@ -1485,7 +1488,7 @@ thunar_application_mkdir (ThunarApplication *application,
   _thunar_return_if_fail (THUNAR_IS_APPLICATION (application));
 
   /* launch the operation */
-  thunar_application_launch (application, parent, "stock_folder",
+  thunar_application_launch (application, parent, "folder-new",
                              _("Creating directories..."), mkdir_stub,
                              path_list, path_list, new_files_closure);
 }
@@ -1549,7 +1552,7 @@ thunar_application_empty_trash (ThunarApplication *application,
       path_list.prev = NULL;
 
       /* launch the operation */
-      thunar_application_launch (application, parent, "gnome-fs-trash-empty",
+      thunar_application_launch (application, parent, "user-trash",
                                   _("Emptying the Trash..."),
                                   unlink_stub, &path_list, NULL, NULL);
 
@@ -1685,7 +1688,7 @@ thunar_application_restore_files (ThunarApplication *application,
   else if (G_LIKELY (response == GTK_RESPONSE_YES))
     {
       /* launch the operation */
-      thunar_application_launch (application, parent, "stock_folder-move",
+      thunar_application_launch (application, parent, "folder-move",
                                  _("Restoring files..."), thunar_vfs_move_files,
                                  source_path_list, target_path_list, new_files_closure);
     }
diff --git a/thunar/thunar-dialogs.c b/thunar/thunar-dialogs.c
index 0e36ee3..570efbe 100644
--- a/thunar/thunar-dialogs.c
+++ b/thunar/thunar-dialogs.c
@@ -602,7 +602,7 @@ thunar_dialogs_show_job_ask_replace (GtkWindow     *parent,
   gtk_box_pack_start (GTK_BOX (GTK_DIALOG (dialog)->vbox), table, TRUE, TRUE, 0);
   gtk_widget_show (table);
 
-  image = gtk_image_new_from_icon_name ("stock_folder-copy", GTK_ICON_SIZE_BUTTON);
+  image = gtk_image_new_from_icon_name ("gtk-copy", GTK_ICON_SIZE_BUTTON);
   gtk_misc_set_alignment (GTK_MISC (image), 0.5f, 0.0f);
   gtk_misc_set_padding (GTK_MISC (image), 6, 6);
   gtk_table_attach (GTK_TABLE (table), image, 0, 1, 0, 1, GTK_FILL, GTK_EXPAND | GTK_FILL, 0, 0);
diff --git a/thunar/thunar-dnd.c b/thunar/thunar-dnd.c
index 4fb278c..ddf0a64 100644
--- a/thunar/thunar-dnd.c
+++ b/thunar/thunar-dnd.c
@@ -68,7 +68,7 @@ thunar_dnd_ask (GtkWidget    *widget,
 {
   static const GdkDragAction dnd_action_items[] = { GDK_ACTION_COPY, GDK_ACTION_MOVE, GDK_ACTION_LINK };
   static const gchar        *dnd_action_names[] = { N_ ("_Copy here"), N_ ("_Move here"), N_ ("_Link here") };
-  static const gchar        *dnd_action_icons[] = { "stock_folder-copy", "stock_folder-move", NULL };
+  static const gchar        *dnd_action_icons[] = { "gtk-copy", "folder-move", NULL };
 
   ThunarxProviderFactory *factory;
   GdkDragAction           dnd_action = 0;
diff --git a/thunar/thunar-file.c b/thunar/thunar-file.c
index c049213..45063fe 100644
--- a/thunar/thunar-file.c
+++ b/thunar/thunar-file.c
@@ -1619,16 +1619,16 @@ thunar_file_get_icon_name (const ThunarFile   *file,
 
   /* special icon for the home node */
   if (G_UNLIKELY (thunar_file_is_home (file))
-      && gtk_icon_theme_has_icon (icon_theme, "gnome-fs-home"))
+      && gtk_icon_theme_has_icon (icon_theme, "user-home"))
     {
-      return "gnome-fs-home";
+      return "user-home";
     }
 
   /* special icon for the desktop node */
   if (G_UNLIKELY (thunar_file_is_desktop (file))
-      && gtk_icon_theme_has_icon (icon_theme, "gnome-fs-desktop"))
+      && gtk_icon_theme_has_icon (icon_theme, "user-desktop"))
    {
-     return "gnome-fs-desktop";
+     return "user-desktop";
    }
 
   /* try to be smart when determining icons for executable files
@@ -1648,10 +1648,10 @@ thunar_file_get_icon_name (const ThunarFile   *file,
 
   /* check if we have an accept icon for the icon we found */
   if ((icon_state == THUNAR_FILE_ICON_STATE_DROP || icon_state == THUNAR_FILE_ICON_STATE_OPEN)
-      && strcmp (icon_name, "gnome-fs-directory") == 0
-      && gtk_icon_theme_has_icon (icon_theme, "gnome-fs-directory-accept"))
+      && strcmp (icon_name, "folder") == 0
+      && gtk_icon_theme_has_icon (icon_theme, "folder-drag-accept"))
     {
-      return "gnome-fs-directory-accept";
+      return "folder-drag-accept";
     }
 
   return icon_name;
diff --git a/thunar/thunar-icon-factory.c b/thunar/thunar-icon-factory.c
index 0cdbe03..22630cd 100644
--- a/thunar/thunar-icon-factory.c
+++ b/thunar/thunar-icon-factory.c
@@ -1036,7 +1036,7 @@ again:
       if (G_UNLIKELY (thumb_state == THUNAR_FILE_THUMB_STATE_LOADING))
         {
           /* check if the icon theme supports the loading icon */
-          icon = thunar_icon_factory_lookup_icon (factory, "gnome-fs-loading-icon", icon_size, FALSE);
+          icon = thunar_icon_factory_lookup_icon (factory, "image-loading", icon_size, FALSE);
           if (G_LIKELY (icon != NULL))
             return icon;
         }
diff --git a/thunar/thunar-stock.c b/thunar/thunar-stock.c
index 9586ed1..c777b5f 100644
--- a/thunar/thunar-stock.c
+++ b/thunar/thunar-stock.c
@@ -45,12 +45,12 @@ typedef struct
 /* keep in sync with thunar-stock.h */
 static const ThunarStockIcon thunar_stock_icons[] =
 {
-  { THUNAR_STOCK_DESKTOP,     "gnome-fs-desktop",       },
-  { THUNAR_STOCK_HOME,        "gnome-fs-home",          },
+  { THUNAR_STOCK_DESKTOP,     "user-desktop",       },
+  { THUNAR_STOCK_HOME,        "user-home",          },
   { THUNAR_STOCK_SHORTCUTS,   "stock_thunar-shortcuts", },
-  { THUNAR_STOCK_TEMPLATES,   "stock_thunar-templates", },
-  { THUNAR_STOCK_TRASH_EMPTY, "gnome-fs-trash-empty",   },
-  { THUNAR_STOCK_TRASH_FULL,  "gnome-fs-trash-full",    },
+  { THUNAR_STOCK_TEMPLATES,   "text-x-generic-template", },
+  { THUNAR_STOCK_TRASH_EMPTY, "user-trash",   },
+  { THUNAR_STOCK_TRASH_FULL,  "user-trash-full",    },
   { THUNAR_STOCK_DOWNLOADS,   "emblem-downloads",       },
   { THUNAR_STOCK_DOCUMENTS,   "emblem-documents",       },
   { THUNAR_STOCK_MUSIC,       "audio-x-generic",        },
diff --git a/thunar/thunar-stock.h b/thunar/thunar-stock.h
index 775b3e0..3f0b937 100644
--- a/thunar/thunar-stock.h
+++ b/thunar/thunar-stock.h
@@ -26,9 +26,9 @@ G_BEGIN_DECLS;
 #define THUNAR_STOCK_HOME         "thunar-home"         /* see ThunarWindow */
 #define THUNAR_STOCK_SHORTCUTS    "thunar-shortcuts"    /* see ThunarShortcutsPane */
 #define THUNAR_STOCK_TEMPLATES    "thunar-templates"    /* see ThunarWindow */
-#define THUNAR_STOCK_TRASH_EMPTY  "thunar-trash-empty"  /* see ThunarTrashAction */
+#define THUNAR_STOCK_TRASH_EMPTY  "thunar-trash"  /* see ThunarTrashAction */
 #define THUNAR_STOCK_TRASH_FULL   "thunar-trash-full"   /* see ThunarTrashAction */
-#define THUNAR_STOCK_DIRECTORY    "gnome-fs-directory"
+#define THUNAR_STOCK_DIRECTORY    "gtk-directory"
 #define THUNAR_STOCK_DOWNLOADS    "thunar-downloads"
 #define THUNAR_STOCK_DOCUMENTS    "thunar-documents"
 #define THUNAR_STOCK_MUSIC        "thunar-music"
