--- examples/dumbplay.c
+++ examples/dumbplay.c
@@ -147,6 +147,12 @@
 	DUH *duh;          /* Encapsulates the music file. */
 	AL_DUH_PLAYER *dp; /* Holds the current playback state. */
 
+	/* Tell Allegro where to find configuration data. This means you can
+	 * put any settings for Allegro in dumb.ini. See Allegro's
+	 * documentation for more information.
+	 */
+	set_config_file("dumb.ini");
+
 	/* Initialise Allegro */
 	if (allegro_init())
 		return EXIT_FAILURE;
@@ -155,12 +161,6 @@
 	if (argc != 2)
 		usage(argv[0]);
 
-	/* Tell Allegro where to find configuration data. This means you can
-	 * put any settings for Allegro in dumb.ini. See Allegro's
-	 * documentation for more information.
-	 */
-	set_config_file("dumb.ini");
-
 	/* Initialise Allegro's keyboard input. */
 	if (install_keyboard()) {
 		allegro_message("Failed to initialise keyboard driver!\n");
