Index: b/conf/suspend.conf
===================================================================
--- a/conf/suspend.conf
+++ b/conf/suspend.conf
@@ -1,9 +1,62 @@
-snapshot device = /dev/snapshot
-resume device = <path_to_resume_device_file>
+#############################################################################
+##
+## note:
+## using pm-utils or powersaved, this file (/etc/suspend.conf) only serves as
+## a template, image_size and resume_device are filled in dynamically
+## and the generated /var/lib/s2disk.conf is used to suspend.
+## _If_ you enter stuff here, it will be copied to that file unchanged,
+## but this might skip some features and sanity checks.
+##
+#############################################################################
+##
+## your snapshot device. You should not need to change this.
+# snapshot device = /dev/snapshot
+#
+## enter your swap device here. Read the warning on pm-utils above, please!
+#resume device = <path_to_resume_device_file>
+#
+## image size will also be filled in by pm-utils
 #image size = 350000000
+#
 #suspend loglevel = 2
+#max loglevel =
+#
+## compute checksum will slow down suspend and resume.
+## Debugging option, default n
 #compute checksum = y
-#compress = y
+#
+## compression will often speed up suspend and resume (default n)
+#compress = n
+#
+## encryption support is rather basic right now - e.g. USB keyboards will not
+## work to enter the key in the standard initrd, also beware of
+## non-US keyboard layouts. Only use this if you know what you are doing.
 #encrypt = y
-#early writeout = y
+#
+## RSA key file that is used for encryption
+#RSA key file = /etc/suspend.key
+#
+## start writing out the image early, before buffers are full.
+## will most of the time speed up overall writing time (default y)
+#early writeout = n
+#
+## use splash picture? (default n)
 #splash = y
+#
+## shutdown method:
+## platform - go through ACPI BIOS to power off the machine (default on
+##            machines that support it)
+## shutdown - just power off like after a shutdown
+## reboot   - reboot instead of powering off. For debugging only.
+#shutdown method = platform
+#
+## resume offset: for use with swapfiles, use "swap-offset" to find out.
+#resume offset = 12345
+#
+## pause after resume for n seconds, so that the timing information can
+## actually be read (default 0 => don't pause)
+#resume pause = 2
+#
+## use threads for suspend? (default n)
+## this hugely speeds up encryption and also compression on mulitcore machines
+#threads = y
