Index: suspend.c
===================================================================
--- suspend.c.orig
+++ suspend.c
@@ -1768,13 +1768,15 @@ int main(int argc, char *argv[])
 	if (compute_checksum != 'y' && compute_checksum != 'Y')
 		compute_checksum = 0;
 #ifdef CONFIG_COMPRESS
-	if (do_compress != 'y' && do_compress != 'Y') {
+	if (do_compress != 'n' && do_compress != 'N') {
+		do_compress = 1;
+		if (lzo_init() != LZO_E_OK) {
+			suspend_error("Failed to initialize LZO. "
+					"Compression disabled.\n");
+			do_compress = 0;
+		}
+	} else
 		do_compress = 0;
-	} else if (lzo_init() != LZO_E_OK) {
-		suspend_error("Failed to initialize LZO. "
-				"Compression disabled.\n");
-		do_compress = 0;
-	}
 #endif
 #ifdef CONFIG_ENCRYPT
 	if (do_encrypt != 'y' && do_encrypt != 'Y')
Index: conf/suspend.conf
===================================================================
--- conf/suspend.conf.orig
+++ conf/suspend.conf
@@ -25,7 +25,7 @@
 ## Debugging option, default n
 #compute checksum = y
 #
-## compression will often speed up suspend and resume (default n)
+## compression will often speed up suspend and resume (default y)
 #compress = n
 #
 ## encryption support is rather basic right now - e.g. USB keyboards will not
