Index: colordiff-1.0.9/Makefile
===================================================================
--- colordiff-1.0.9.orig/Makefile
+++ colordiff-1.0.9/Makefile
@@ -36,10 +36,7 @@ install:
 	else \
 	  install -d ${DESTDIR}${ETC_DIR}; \
 	fi
-	cp colordiffrc ${DESTDIR}${ETC_DIR}/colordiffrc
-	-chown root.root ${DESTDIR}${ETC_DIR}/colordiffrc
-	chmod 644 ${DESTDIR}${ETC_DIR}/colordiffrc
-
+	install -m 644 colordiffrc $(DESTDIR)${ETC_DIR}/colordiffrc
 uninstall:
 	rm -f ${DESTDIR}${INSTALL_DIR}/colordiff
 	rm -f ${DESTDIR}${ETC_DIR}/colordiffrc
