--- colordiff-1.0.6/colordiff.pl~	2006-09-13 22:37:24.000000000 +0300
+++ colordiff-1.0.6/colordiff.pl	2007-04-04 19:09:01.000000000 +0300
@@ -55,11 +55,11 @@
 $colour{off}         = "\033[0;0m";
 
 # Default colours if /etc/colordiffrc or ~/.colordiffrc do not exist
-my $plain_text = $colour{white};
+my $plain_text = $colour{darkwhite};
 my $file_old   = $colour{red};
 my $file_new   = $colour{blue};
 my $diff_stuff = $colour{magenta};
-my $cvs_stuff  = $colour{green};
+my $cvs_stuff  = $colour{darkyellow};
 
 # Locations for personal and system-wide colour configurations
 my $HOME   = $ENV{HOME};
--- colordiff-1.0.6/colordiffrc~	2006-07-31 21:36:22.000000000 +0300
+++ colordiff-1.0.6/colordiffrc	2007-04-04 19:09:30.000000000 +0300
@@ -19,8 +19,8 @@
 # aliases for the same thing, namely "don't colour highlight
 # this, use the default output colour"
 #
-plain=off
+plain=darkwhite
 newtext=blue
 oldtext=red
 diffstuff=magenta
-cvsstuff=green
+cvsstuff=darkyellow
--- colordiff-1.0.6/colordiffrc-lightbg~	2006-07-31 21:37:11.000000000 +0300
+++ colordiff-1.0.6/colordiffrc-lightbg	2007-04-04 19:08:08.000000000 +0300
@@ -19,8 +19,8 @@
 # aliases for the same thing, namely "don't colour highlight
 # this, use the default output colour"
 #
-plain=off
-newtext=blue
-oldtext=red
+plain=black
+newtext=darkblue
+oldtext=darkred
 diffstuff=darkgreen
-cvsstuff=darkmagenta
+cvsstuff=darkwhite
