diff -up colordiff-1.0.9/cdiff.sh~ colordiff-1.0.9/cdiff.sh
--- colordiff-1.0.9/cdiff.sh~	2009-01-26 22:17:47.000000000 +0200
+++ colordiff-1.0.9/cdiff.sh	2009-07-29 21:36:42.000000000 +0300
@@ -2,7 +2,7 @@
 
 # cdiff.sh - Convenience wrapper for colordiff
 #
-# Copyright (C) 2003-2004 Ville Skyttä <ville.skytta@iki.fi>
+# Copyright (C) 2003-2009 Ville Skyttä <ville.skytta@iki.fi>
 # Based on cdiff version 1.4 by eivind@FreeBSD.org
 #
 # This program is free software; you can redistribute it and/or
@@ -23,9 +23,9 @@
     while [ "$1" != "" ]; do
         file=`echo "$1" | perl -pe 's|^file:/+|/|i'`
         case "$file" in
-            *.bz2)   cat="bzip2 -dcf" ;;
-            *.lzma)  cat="lzma -dc"   ;;
-            *)       cat="gzip -dcf"  ;;
+            *.bz2)       cat="bzip2 -dcf" ;;
+            *.xz|*.lzma) cat="xz -dc"     ;;
+            *)           cat="gzip -dcf"  ;;
         esac
         case `echo "$file" | perl -ne 'print lc $_'` in
         http:*|https:*|ftp:*)
diff -up colordiff-1.0.9/cdiff.xml~ colordiff-1.0.9/cdiff.xml
--- colordiff-1.0.9/cdiff.xml~	2009-01-26 22:16:44.000000000 +0200
+++ colordiff-1.0.9/cdiff.xml	2009-07-29 21:37:08.000000000 +0300
@@ -35,9 +35,9 @@ in a URL or a file, e.g.</para>
 cdiff http://some.url.com/foo/thing.patch
 </screen>
 
-<para>It also adds support for reading gzip, bzip2 and lzma compressed diffs,
-and like colordiff, also supports reading diffs from standard input if URL
-or a filename is not specified.  &cdiff; pipes colordiff's output through
+<para>It also adds support for reading gzip, bzip2, xz and lzma compressed
+diffs, and like colordiff, also supports reading diffs from standard input if
+URL or a filename is not specified.  &cdiff; pipes colordiff's output through
 less -R.</para>
 
 </refsect1>
