Index: colordiff-1.0.9/Makefile
===================================================================
--- colordiff-1.0.9.orig/Makefile
+++ colordiff-1.0.9/Makefile
@@ -10,10 +10,10 @@ TARBALL=${TMPDIR}.tar.gz
 
 
 doc: colordiff.xml cdiff.xml
-	xmlto -vv man colordiff.xml
-	xmlto -vv man cdiff.xml
-	xmlto -vv txt colordiff.xml
-	xmlto -vv html-nochunks colordiff.xml
+	xmlto --with-fop --skip-validation -vv man colordiff.xml
+	xmlto --with-fop --skip-validation -vv man cdiff.xml
+	xmlto --with-fop --skip-validation -vv txt colordiff.xml
+	xmlto --with-fop --skip-validation -vv html-nochunks colordiff.xml
 	mv colordiff.txt README
 	perl -p -i -e 's#<head>#<head><link rel=\"stylesheet\" type=\"text/css\" href=\"colordiff.css\">#' colordiff.html
 	perl -p -i -e 's#</body>#</div></body>#' colordiff.html
