diff -ur old-unalz/UnAlz.cpp unalz/UnAlz.cpp
--- old-unalz/UnAlz.cpp	2009-04-01 09:00:27.000000000 +0300
+++ unalz/UnAlz.cpp	2012-11-04 01:49:42.247000093 +0200
@@ -1907,7 +1907,7 @@
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 UINT32 CUnAlz::CRC32(UINT32 l, BYTE c)
 {
-	const unsigned long *CRC_TABLE = get_crc_table();
+	const z_crc_t *CRC_TABLE = get_crc_table();
 	return CRC_TABLE[(l ^ c) & 0xff] ^ (l >> 8);
 }
 
