Index: privoxy-3.0.16-stable/privoxy-generic.init
===================================================================
--- privoxy-3.0.16-stable.orig/privoxy-generic.init
+++ privoxy-3.0.16-stable/privoxy-generic.init
@@ -46,12 +46,12 @@
 # logfile is writable by $P_USER (logfile is defined in config), and that 
 # there is suitable write access for $P_PIDFILE.
 
-PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/xpg4/bin:/usr/bin:/sbin:/bin
+PATH=/usr/bin:/bin:/sbin:/usr/sbin
 P_NAME=Privoxy
 # Path to executable.
 P_DAEMON=privoxy
 # Full path to location of Privoxy config file. 
-P_CONF_FILE=/usr/local/etc/privoxy/config
+P_CONF_FILE=/etc/privoxy/config
 # Full path to PID file location. Location must be writable by 
 # whoever runs this script and by Privoxy itself.
 P_PIDFILE=/var/run/privoxy.pid
Index: privoxy-3.0.16-stable/config
===================================================================
--- privoxy-3.0.16-stable.orig/config
+++ privoxy-3.0.16-stable/config
@@ -267,7 +267,7 @@
 #
 #      No trailing "/", please.
 #
-confdir .
+confdir /etc/privoxy
 #
 #
 #  2.2. templdir
@@ -324,7 +324,7 @@ confdir .
 #
 #      No trailing "/", please.
 #
-logdir .
+logdir /var/log/privoxy
 #
 #
 #  2.4. actionsfile
