--- Makefile.orig	2011-04-11 13:03:18.392000690 +0300
+++ Makefile	2011-04-11 13:03:41.574000635 +0300
@@ -18,7 +18,7 @@
 ICONDIR = $(PREFIX)/share/pixmaps
 LOCALEDIR = $(PREFIX)/share/locale
 # CONFIGURE_IN = sed -e 's!prefix!$(PREFIX)!g'
-CONFIGURE_IN = sed -e "s!prefix = '/usr'!prefix = '$(prefix)'!g"
+CONFIGURE_IN = sed -e "s!prefix = '/usr'!prefix = '/usr'!g"
 
 LANGUAGES = `find locale/ -maxdepth 1 -mindepth 1 -type d -printf "%f "`
 
