--- po/POTFILES.in	2006-11-20 03:18:36.000000000 +0200
+++ po/POTFILES.in.new	2008-02-18 23:29:18.000000000 +0200
@@ -2,6 +2,15 @@
 #src/kino.glade.h
 #src/magick.glade.h
 
+ffmpeg/libavcodec/ps2/idct_mmi.c
+ffmpeg/libavcodec/sparc/dsputil_vis.c
+ffmpeg/libavcodec/sparc/vis.h
+ffmpeg/libavutil/bswap.h
+ffmpeg/libswscale/yuv2rgb_template.c
+src/export.h
+src/message.cc
+src/page_bttv.cc
+
 src/main.c
 src/support.c
 src/audio_filters.cc
