diff -Naur kino-1.3.3.orig/src/frame.cc kino-1.3.3/src/frame.cc
--- kino-1.3.3.orig/src/frame.cc	2008-12-26 07:27:01.000000000 +0200
+++ kino-1.3.3/src/frame.cc	2009-08-01 03:54:21.015857787 +0300
@@ -1121,14 +1121,14 @@
 	avcodec_decode_video( libavcodec, frame, &got_picture, data, GetFrameSize() );
 	if ( got_picture )
 	{
-		avpicture_fill( &output, static_cast<uint8_t*>( yuv ), PIX_FMT_YUV422, GetWidth(), GetHeight() );
+		avpicture_fill( &output, static_cast<uint8_t*>( yuv ), PIX_FMT_YUYV422, GetWidth(), GetHeight() );
 #if defined(HAVE_SWSCALE)
 		if ( !imgConvertYuvCtx )
 			imgConvertYuvCtx = sws_getContext( libavcodec->width, libavcodec->height, libavcodec->pix_fmt,
-				GetWidth(), GetHeight(), PIX_FMT_YUV422, SWS_FAST_BILINEAR, NULL, NULL, NULL );
+				GetWidth(), GetHeight(), PIX_FMT_YUYV422, SWS_FAST_BILINEAR, NULL, NULL, NULL );
 		sws_scale( imgConvertYuvCtx, frame->data, frame->linesize, 0, libavcodec->height, output.data, output.linesize );
 #else
-		img_convert( &output, PIX_FMT_YUV422, (AVPicture *)frame, libavcodec->pix_fmt, GetWidth(), GetHeight() );
+		img_convert( &output, PIX_FMT_YUYV422, (AVPicture *)frame, libavcodec->pix_fmt, GetWidth(), GetHeight() );
 #endif
 	}
 	av_free( frame );
