diff -up blender-2.67b/source/blender/makesrna/intern/rna_access.c.rna blender-2.67b/source/blender/makesrna/intern/rna_access.c
--- blender-2.67b/source/blender/makesrna/intern/rna_access.c.rna	2013-05-16 12:43:18.000000000 +0200
+++ blender-2.67b/source/blender/makesrna/intern/rna_access.c	2013-06-14 16:35:39.263926932 +0200
@@ -1273,13 +1273,15 @@ void RNA_property_enum_items_gettexted(b
 			int totitem = 0;
 
 			/* count */
-			for (i = 0; (*item)[i].identifier; i++)
-				totitem++;
+			if (*item)
+			  for (i = 0; (*item)[i].identifier; i++)
+			    totitem++;
 
 			nitem = MEM_callocN(sizeof(EnumPropertyItem) * (totitem + 1), "enum_items_gettexted");
 
-			for (i = 0; (*item)[i].identifier; i++)
-				nitem[i] = (*item)[i];
+			if (*item)
+			  for (i = 0; (*item)[i].identifier; i++)
+			    nitem[i] = (*item)[i];
 
 			*free = TRUE;
 		}
