--- src/modules/rtp/rtp.c.orig	2012-05-20 09:23:37.081609264 +0200
+++ src/modules/rtp/rtp.c	2012-05-20 09:24:08.991608129 +0200
@@ -286,7 +286,7 @@
 
     if (!found_tstamp) {
         pa_log_warn("Couldn't find SCM_TIMESTAMP data in auxiliary recvmsg() data!");
-        memset(tstamp, 0, sizeof(tstamp));
+        memset(tstamp, 0, sizeof(struct timeval));
     }
 
     return 0;
--- src/modules/echo-cancel/adrian-aec.c.orig	2012-05-20 09:24:20.836607707 +0200
+++ src/modules/echo-cancel/adrian-aec.c	2012-05-20 09:26:07.641603903 +0200
@@ -163,7 +163,7 @@
     } else if (1 == a->hangover) {
       --(a->hangover);
       // My Leaky NLMS is to erase vector w when hangover expires
-      memset(a->w, 0, sizeof(a->w));
+      memset(a->w, 0, sizeof(REAL));
     }
   }
 }
