---
 src/pulsecore/socket-client.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: pulseaudio-0.99.2/src/pulsecore/socket-client.c
===================================================================
--- pulseaudio-0.99.2.orig/src/pulsecore/socket-client.c
+++ pulseaudio-0.99.2/src/pulsecore/socket-client.c
@@ -253,7 +253,7 @@ static int sockaddr_prepare(pa_socket_cl
     c->local = pa_socket_address_is_local(sa);
 
     if ((c->fd = pa_socket_cloexec(sa->sa_family, SOCK_STREAM, 0)) < 0) {
-        pa_log("socket(): %s", pa_cstrerror(errno));
+        /*pa_log("socket(): %s", pa_cstrerror(errno));*/
         return -1;
     }
 
