diff -Nur dsniff-2.4-old/arp.c dsniff-2.4/arp.c
--- dsniff-2.4-old/arp.c	2009-09-15 00:11:20.339202117 +0300
+++ dsniff-2.4/arp.c	2009-09-15 00:11:20.598451630 +0300
@@ -34,6 +34,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <string.h>
 
 #include "arp.h"
 
diff -Nur dsniff-2.4-old/buf.c dsniff-2.4/buf.c
--- dsniff-2.4-old/buf.c	2009-09-15 00:11:17.163202943 +0300
+++ dsniff-2.4/buf.c	2009-09-15 00:11:20.598451630 +0300
@@ -17,6 +17,7 @@
 #include <unistd.h>
 #include <ctype.h>
 #include <err.h>
+#include <string.h>
 
 #include "buf.h"
 
diff -Nur dsniff-2.4-old/decode_nntp.c dsniff-2.4/decode_nntp.c
--- dsniff-2.4-old/decode_nntp.c	2009-09-15 00:11:17.165203268 +0300
+++ dsniff-2.4/decode_nntp.c	2009-09-15 00:11:20.598451630 +0300
@@ -15,6 +15,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "base64.h"
 #include "decode.h"
diff -Nur dsniff-2.4-old/decode_pop.c dsniff-2.4/decode_pop.c
--- dsniff-2.4-old/decode_pop.c	2009-09-15 00:11:17.164203350 +0300
+++ dsniff-2.4/decode_pop.c	2009-09-15 00:11:20.598451630 +0300
@@ -14,6 +14,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "base64.h"
 #include "options.h"
diff -Nur dsniff-2.4-old/decode_rlogin.c dsniff-2.4/decode_rlogin.c
--- dsniff-2.4-old/decode_rlogin.c	2009-09-15 00:11:17.168204279 +0300
+++ dsniff-2.4/decode_rlogin.c	2009-09-15 00:11:20.598451630 +0300
@@ -14,6 +14,8 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcpy.h>
+#include <strlcat.h>
 
 #include "options.h"
 #include "decode.h"
diff -Nur dsniff-2.4-old/decode_smb.c dsniff-2.4/decode_smb.c
--- dsniff-2.4-old/decode_smb.c	2009-09-15 00:11:17.163202943 +0300
+++ dsniff-2.4/decode_smb.c	2009-09-15 00:11:20.598451630 +0300
@@ -15,6 +15,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "decode.h"
 
diff -Nur dsniff-2.4-old/decode_smtp.c dsniff-2.4/decode_smtp.c
--- dsniff-2.4-old/decode_smtp.c	2009-09-15 00:11:17.158475246 +0300
+++ dsniff-2.4/decode_smtp.c	2009-09-15 00:11:20.599452177 +0300
@@ -14,6 +14,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "base64.h"
 #include "options.h"
diff -Nur dsniff-2.4-old/decode_sniffer.c dsniff-2.4/decode_sniffer.c
--- dsniff-2.4-old/decode_sniffer.c	2009-09-15 00:11:17.164203350 +0300
+++ dsniff-2.4/decode_sniffer.c	2009-09-15 00:11:20.599452177 +0300
@@ -15,6 +15,8 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
+#include <strlcpy.h>
 
 #include "base64.h"
 #include "decode.h"
diff -Nur dsniff-2.4-old/decode_socks.c dsniff-2.4/decode_socks.c
--- dsniff-2.4-old/decode_socks.c	2009-09-15 00:11:17.165203268 +0300
+++ dsniff-2.4/decode_socks.c	2009-09-15 00:11:20.599452177 +0300
@@ -14,6 +14,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "decode.h"
 
diff -Nur dsniff-2.4-old/decode_tds.c dsniff-2.4/decode_tds.c
--- dsniff-2.4-old/decode_tds.c	2009-09-15 00:11:17.167204361 +0300
+++ dsniff-2.4/decode_tds.c	2009-09-15 00:11:20.599452177 +0300
@@ -18,6 +18,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "decode.h"
 
diff -Nur dsniff-2.4-old/decode_telnet.c dsniff-2.4/decode_telnet.c
--- dsniff-2.4-old/decode_telnet.c	2009-09-15 00:11:17.164203350 +0300
+++ dsniff-2.4/decode_telnet.c	2009-09-15 00:11:20.600452304 +0300
@@ -14,6 +14,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcpy.h>
 
 #include "options.h"
 #include "decode.h"
diff -Nur dsniff-2.4-old/decode_x11.c dsniff-2.4/decode_x11.c
--- dsniff-2.4-old/decode_x11.c	2009-09-15 00:11:17.164203350 +0300
+++ dsniff-2.4/decode_x11.c	2009-09-15 00:11:20.600452304 +0300
@@ -14,6 +14,8 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
+#include <strlcpy.h>
 
 #include "decode.h"
 
diff -Nur dsniff-2.4-old/dnsspoof.c dsniff-2.4/dnsspoof.c
--- dsniff-2.4-old/dnsspoof.c	2009-09-15 00:11:20.443202659 +0300
+++ dsniff-2.4/dnsspoof.c	2009-09-15 00:11:20.600452304 +0300
@@ -20,6 +20,7 @@
 #include <stdlib.h>
 #include <signal.h>
 #include <string.h>
+#include <strlcpy.h>
 #include <resolv.h>
 #include <err.h>
 #include <libnet.h>
diff -Nur dsniff-2.4-old/magic.c dsniff-2.4/magic.c
--- dsniff-2.4-old/magic.c	2009-09-15 00:11:17.164203350 +0300
+++ dsniff-2.4/magic.c	2009-09-15 00:11:20.600452304 +0300
@@ -36,6 +36,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <strlcpy.h>
 #include <ctype.h>
 #include <time.h>
 #include <err.h>
diff -Nur dsniff-2.4-old/missing/strlcat.h dsniff-2.4/missing/strlcat.h
--- dsniff-2.4-old/missing/strlcat.h	1970-01-01 02:00:00.000000000 +0200
+++ dsniff-2.4/missing/strlcat.h	2009-09-15 00:11:20.601451803 +0300
@@ -0,0 +1 @@
+size_t strlcat(char *dst, const char *src, size_t siz);
diff -Nur dsniff-2.4-old/missing/strlcpy.h dsniff-2.4/missing/strlcpy.h
--- dsniff-2.4-old/missing/strlcpy.h	1970-01-01 02:00:00.000000000 +0200
+++ dsniff-2.4/missing/strlcpy.h	2009-09-15 00:11:20.601451803 +0300
@@ -0,0 +1 @@
+size_t strlcpy(char *dst, const char *src, size_t siz);
diff -Nur dsniff-2.4-old/sshmitm.c dsniff-2.4/sshmitm.c
--- dsniff-2.4-old/sshmitm.c	2009-09-15 00:11:20.444202787 +0300
+++ dsniff-2.4/sshmitm.c	2009-09-15 00:11:20.601451803 +0300
@@ -24,6 +24,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <strlcat.h>
 
 #include "buf.h"
 #include "record.h"
