diff -Nur dsniff-2.4-old/decode.c dsniff-2.4/decode.c
--- dsniff-2.4-old/decode.c	2009-09-15 00:11:17.167204361 +0300
+++ dsniff-2.4/decode.c	2009-09-15 00:11:20.744452006 +0300
@@ -63,7 +63,8 @@
 	{ "http",	decode_http },
 	{ "ospf",	decode_ospf },
 	{ "poppass",	decode_poppass },
-	{ "pop",	decode_pop },
+	{ "pop2",	decode_pop },
+	{ "pop3",	decode_pop },
 	{ "nntp",	decode_nntp },
 	{ "smb",	decode_smb },
 	{ "imap",	decode_imap },
diff -Nur dsniff-2.4-old/dsniff.services dsniff-2.4/dsniff.services
--- dsniff-2.4-old/dsniff.services	2009-09-15 00:11:17.164203350 +0300
+++ dsniff-2.4/dsniff.services	2009-09-15 00:11:20.744452006 +0300
@@ -10,8 +10,8 @@
 ospf		89/ip
 http		98/tcp
 poppass		106/tcp
-pop		109/tcp
-pop		110/tcp
+pop2		109/tcp
+pop3		110/tcp
 portmap		111/tcp
 portmap		-111/tcp
 portmap		111/udp
