--- openslp-1.2.1/configure.in.optflags	2005-02-15 14:06:02.000000000 -0600
+++ openslp-1.2.1/configure.in	2005-03-04 06:53:02.000000000 -0600
@@ -130,7 +130,7 @@
        CFLAGS="$CFLAGS -Werror"
     fi
     if test X"$debug" = X"no"; then 
-       OPTFLAGS="-O3"
+       OPTFLAGS="-O3 $RPM_OPT_FLAGS"
     fi
     
 elif $CC -V 2>&1 | grep "WorkShop Compilers"; then
--- openslp-1.2.1/configure.optflags	2005-03-02 12:00:28.000000000 -0600
+++ openslp-1.2.1/configure	2005-03-04 06:53:04.000000000 -0600
@@ -20596,7 +20596,7 @@
        CFLAGS="$CFLAGS -Werror"
     fi
     if test X"$debug" = X"no"; then
-       OPTFLAGS="-O3"
+       OPTFLAGS="-O3 $RPM_OPT_FLAGS"
     fi
 
 elif $CC -V 2>&1 | grep "WorkShop Compilers"; then
