Index: openshot-1.1.1/xdg/openshot.desktop
===================================================================
--- openshot-1.1.1.orig/xdg/openshot.desktop
+++ openshot-1.1.1/xdg/openshot.desktop
@@ -1,12 +1,16 @@
 [Desktop Entry]
-Name=OpenShot Video Editor
-X-GNOME-FullName=OpenShot Video Editor
+Encoding=UTF-8
+Type=Application
+Version=1.0
+Name=OpenShot
+GenericName=OpenShot Non-linear Video Editor
+GenericName[tr]=OpenShot Video Düzenleyici
 Comment=Create and edit videos and movies
+Comment[tr]=Videolar, filmler yaratın ve düzenleyin
+Icon=openshot
 Exec=openshot %F
 Terminal=false
-Type=Application
-Icon=openshot
 Categories=GNOME;GTK;AudioVideo;Video;AudioVideoEditing;
 MimeType=application/vnd.openshot-project;
 X-AppInstall-Package=openshot
-X-Ubuntu-Gettext-Domain=OpenShot
\ Dosya sonunda yenisatır yok.
+X-Ubuntu-Gettext-Domain=OpenShot
