From 2a0365a6264d81917091b302ec0e215e421ec19c Mon Sep 17 00:00:00 2001
From: Mete Alpaslan <mete@pardus.org.tr>
Date: Mon, 30 May 2011 16:23:25 +0300
Subject: [PATCH 1/4] Fix adding resume kernel parameter if no swap device
 defined

---
 yali/storage/bootloader.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/yali/storage/bootloader.py b/yali/storage/bootloader.py
index 2ae4744..482e5db 100644
--- a/yali/storage/bootloader.py
+++ b/yali/storage/bootloader.py
@@ -182,7 +182,7 @@ class BootLoader(object):
         self._swapDevice = device
 
     def _getSwapDevice(self):
-        if not self._swapDevice:
+        if not self._swapDevice and self.storage.swaps:
             self._swapDevice = self.storage.swaps[0]
 
         return self._swapDevice
-- 
1.7.5.2

