From fc0d1c830fcc1213d70e0f085c9da0b34d04149f Mon Sep 17 00:00:00 2001
From: Mete Alpaslan <mete@pardus.org.tr>
Date: Mon, 30 May 2011 16:25:34 +0300
Subject: [PATCH 2/4] Fix importing other linux systems boot loader conf

---
 yali/storage/bootloader.py |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/yali/storage/bootloader.py b/yali/storage/bootloader.py
index 482e5db..f642d2a 100644
--- a/yali/storage/bootloader.py
+++ b/yali/storage/bootloader.py
@@ -316,7 +316,11 @@ class BootLoader(object):
             self.grubConf = grubutils.grubConf()
             self.grubConf.parseConf(os.path.join(ctx.consts.target_dir, self._conf))
             for entry in additional_conf.entries:
-                stage2Device = entry.getCommand("uuid").value if entry.getCommand("uuid") else entry.getCommand("root").value
+                stage2Device = None
+                if entry.getCommand("uuid"):
+                    stage2Device = entry.getCommand("uuid").value
+                elif entry.getCommand("root"):
+                    stage2Device = entry.getCommand("root").value
                 if stage2Device:
                     entry.title = entry.title + " [ %s ]" % device
                     self.grubConf.addEntry(entry)
-- 
1.7.5.2

