From ac8481e60a1df48e79f9ff6f342f2172dfb00c14 Mon Sep 17 00:00:00 2001
From: Mete Alpaslan <mete@pardus.org.tr>
Date: Mon, 30 May 2011 16:26:57 +0300
Subject: [PATCH 3/4] Allow passing screen if only disk layout warning raised

---
 yali/gui/ScrPartitionManual.py |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/yali/gui/ScrPartitionManual.py b/yali/gui/ScrPartitionManual.py
index b9c3090..cbe6cbc 100644
--- a/yali/gui/ScrPartitionManual.py
+++ b/yali/gui/ScrPartitionManual.py
@@ -112,8 +112,6 @@ class Widget(QWidget, ScreenWidget):
             rc = self.intf.detailedMessageWindow(_("Partitioning Warnings"),
                                                   detailed, comments, type="custom", customIcon="warning",
                                                   customButtons=[_("Ok"), _("Cancel")], default=1)
-            if rc:
-                return False
 
         formatWarnings = getPreExistFormatWarnings(self.storage)
         if formatWarnings:
-- 
1.7.5.2

