From 3bf708842fa346a4d71d0606e5bd514393d2beaa Mon Sep 17 00:00:00 2001
From: Mete Alpaslan <mete@pardus.org.tr>
Date: Mon, 30 May 2011 16:29:39 +0300
Subject: [PATCH 4/4] add padding on help screen

---
 yali/gui/Ui/help.ui |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/yali/gui/Ui/help.ui b/yali/gui/Ui/help.ui
index bc9241a..88d386d 100644
--- a/yali/gui/Ui/help.ui
+++ b/yali/gui/Ui/help.ui
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>600</width>
-    <height>57</height>
+    <height>85</height>
    </rect>
   </property>
   <property name="minimumSize">
@@ -34,6 +34,9 @@
    </property>
    <item row="0" column="0">
     <widget class="QWidget" name="helpContentFrame" native="true">
+     <property name="styleSheet">
+      <string notr="true">padding:15px</string>
+     </property>
      <layout class="QGridLayout" name="gridLayout">
       <property name="margin">
        <number>20</number>
-- 
1.7.5.2

