--- yali~/postinstall.py	2013-04-04 13:41:05.000000000 +0200
+++ yali/postinstall.py	2013-07-20 23:17:19.067562593 +0200
@@ -269,6 +269,11 @@
 
 def writeBootLooder():
     ctx.logger.info("Generating grub configuration file")
+
+    swapDevices = ctx.storage.storageset.swapDevices
+    if swapDevices:
+        yali.util.chroot("""sed -e 's/\(GRUB_CMDLINE_LINUX_DEFAULT.*\)\(['"'"'"]\)\s*$/\\1 resume=%s\\2/' /etc/default/grub""" % swapDevices[0].path)
+
     if ctx.storage.storageset.active:
         yali.util.chroot("grub2-mkconfig -o /boot/grub2/grub.cfg")
         return True
