--- yali/gui/ScrGoodbye.py~	2013-06-17 23:09:56.083453247 +0200
+++ yali/gui/ScrGoodbye.py	2013-06-17 23:09:13.000000000 +0200
@@ -89,6 +89,8 @@
             postInstallOperations.append(yali.postinstall.Operation(_("Writing bootloader config..."), yali.postinstall.writeBootLooder))
             postInstallOperations.append(yali.postinstall.Operation(_("Stopping to D-Bus..."), yali.util.stop_dbus))
             postInstallOperations.append(yali.postinstall.Operation(_("Teardown storage subsystem..."), yali.postinstall.teardownStorage))
+        elif ctx.flags.install_type == ctx.STEP_BASE or ctx.flags.install_type == ctx.STEP_DEFAULT:
+            postInstallOperations.append(yali.postinstall.Operation(_("Writing bootloader config..."), yali.postinstall.writeBootLooder))
 
         if ctx.flags.install_type == ctx.STEP_DEFAULT or ctx.flags.install_type == ctx.STEP_BASE or ctx.flags.install_type == ctx.STEP_RESCUE:
             postInstallOperations.append(yali.postinstall.Operation(_("Stopping to D-Bus..."), yali.util.stop_dbus))
