--- yali/gui/Ui/bootloaderwidget.ui~	2013-01-18 14:02:04.149351496 +0100
+++ yali/gui/Ui/bootloaderwidget.ui	2013-01-18 14:02:55.149349632 +0100
@@ -185,20 +185,13 @@
                 <property name="verticalSpacing">
                  <number>5</number>
                 </property>
-                <item row="3" column="0" colspan="2">
+                <item row="2" column="0" colspan="2">
                  <widget class="QRadioButton" name="noInstall">
                   <property name="text">
                    <string>Don't install bootloader</string>
                   </property>
                  </widget>
                 </item>
-                <item row="2" column="0" colspan="2">
-                 <widget class="QRadioButton" name="installPartition">
-                  <property name="text">
-                   <string>The partition where Pardus is installed</string>
-                  </property>
-                 </widget>
-                </item>
                 <item row="0" column="0">
                  <widget class="QRadioButton" name="installMBR">
                   <property name="sizePolicy">
--- yali/gui/ScrBootloader.py~	2013-01-18 10:15:06.000000000 +0100
+++ yali/gui/ScrBootloader.py	2013-01-18 15:13:11.985419513 +0100
@@ -35,7 +35,7 @@
 
         self.connect(self.ui.defaultSettings, SIGNAL("toggled(bool)"), self.showDefaultSettings)
         self.connect(self.ui.noInstall, SIGNAL("toggled(bool)"), self.deactivateBootloader)
-        self.connect(self.ui.installPartition, SIGNAL("toggled(bool)"), self.activateInstallPartition)
+#        self.connect(self.ui.installPartition, SIGNAL("toggled(bool)"), self.activateInstallPartition)
         self.connect(self.ui.drives, SIGNAL("currentIndexChanged(int)"), self.currentDeviceChanged)
 
         self.ui.advancedSettingsBox.show()
@@ -68,8 +68,8 @@
 
         if self.ui.noInstall.isChecked():
             self.bootloader.bootType = BOOT_TYPE_NONE
-        elif self.ui.installPartition.isChecked():
-            self.bootloader.bootType = BOOT_TYPE_PARTITION
+#        elif self.ui.installPartition.isChecked():
+#            self.bootloader.bootType = BOOT_TYPE_PARTITION
         elif self.ui.installMBR.isChecked():
             self.bootloader.bootType = BOOT_TYPE_MBR
 
@@ -98,19 +98,19 @@
             if choice == BOOT_TYPE_MBR:
                 self.ui.installMBR.setText(_("The first sector of"))
                 self.boot_disk = self.bootloader.choices[BOOT_TYPE_MBR][0]
-            elif choice == BOOT_TYPE_RAID:
-                self.ui.installPartition.setText("The RAID array where Pardus is installed")
-                self.boot_partition = self.bootloader.choices[BOOT_TYPE_RAID][0]
-            elif choice == BOOT_TYPE_PARTITION:
-                self.ui.installPartition.setText(_("The partition where Pardus is installed"))
-                self.boot_partition = self.bootloader.choices[BOOT_TYPE_PARTITION][0]
+#            elif choice == BOOT_TYPE_RAID:
+#                self.ui.installPartition.setText("The RAID array where Pardus is installed")
+#                self.boot_partition = self.bootloader.choices[BOOT_TYPE_RAID][0]
+#            elif choice == BOOT_TYPE_PARTITION:
+#                self.ui.installPartition.setText(_("The partition where Pardus is installed"))
+#                self.boot_partition = self.bootloader.choices[BOOT_TYPE_PARTITION][0]
 
         if self.boot_disk:
             self.default = self.boot_disk
             self.ui.installMBR.setChecked(True)
-        else:
-            self.default = self.boot_partition
-            self.ui.installPartition.setChecked(True)
+#        else:
+#            self.default = self.boot_partition
+#            self.ui.installPartition.setChecked(True)
 
     def deactivateBootloader(self):
         self.device = None
