--- yali/storage/operations.py~	2013-02-07 12:11:40.354491158 +0100
+++ yali/storage/operations.py	2013-03-25 09:20:08.687127003 +0100
@@ -7,6 +7,7 @@
 from devices.partition import Partition
 from formats import getFormat
 from udev import udev_get_block_device, udev_device_get_uuid
+import yali.context as ctx
 
 OPERATION_TYPE_NONE = 0
 OPERATION_TYPE_DESTROY = 1000
@@ -234,6 +235,12 @@
 
             self.device.disk.format.commitToDisk()
 
+        info = udev_get_block_device(self.device.sysfsPath)
+        try:
+            olduuid = udev_device_get_uuid(info)
+        except AttributeError:
+            olduuid = 0
+        ctx.logger.info(">>>INFO: %s" % info)
         self.device.format.create(intf=intf,
                                   device=self.device.path,
                                   options=self.device.formatArgs)
@@ -241,8 +248,12 @@
         # Get the UUID now that the format is created
         yali.baseudev.udev_settle()
         self.device.updateSysfsPath()
-        info = udev_get_block_device(self.device.sysfsPath)
-        self.device.format.uuid = udev_device_get_uuid(info)
+        newuuid = olduuid
+        # wait for new uuid
+        while newuuid == olduuid:
+            info = udev_get_block_device(self.device.sysfsPath)
+            newuuid = udev_device_get_uuid(info)
+        self.device.format.uuid = newuuid
 
     def cancel(self):
         self.device.format = self.origFormat
