diff -aur yali/storage/operations.py~ yali/storage/operations.py
--- yali/storage/operations.py~	2011-05-30 01:47:45.000000000 +0200
+++ yali/storage/operations.py	2013-02-08 12:39:52.449824872 +0100
@@ -7,6 +7,7 @@
 from devices.partition import Partition
 from formats import getFormat
 from udev import udev_get_block_device, udev_device_get_uuid
+import yali.context as ctx
 
 OPERATION_TYPE_NONE = 0
 OPERATION_TYPE_DESTROY = 1000
@@ -234,6 +235,9 @@
 
             self.device.disk.format.commitToDisk()
 
+        info = udev_get_block_device(self.device.sysfsPath)
+        olduuid = udev_device_get_uuid(info)
+        ctx.logger.info("old uuid: %s" % olduuid)
         self.device.format.create(intf=intf,
                                   device=self.device.path,
                                   options=self.device.formatArgs)
@@ -241,8 +245,16 @@
         # Get the UUID now that the format is created
         yali.baseudev.udev_settle()
         self.device.updateSysfsPath()
-        info = udev_get_block_device(self.device.sysfsPath)
-        self.device.format.uuid = udev_device_get_uuid(info)
+        newuuid = olduuid
+        count = 0
+        # wait for new uuid
+        while newuuid == olduuid:
+            count += 1
+            info = udev_get_block_device(self.device.sysfsPath)
+            newuuid = udev_device_get_uuid(info)
+        self.device.format.uuid = newuuid
+        ctx.logger.info("new uuid: %s" % newuuid)
+        ctx.logger.info("count: %d" % count)
 
     def cancel(self):
         self.device.format = self.origFormat
