Index: yali-2.4.1_pre20110530/yali/gui/Ui/installprogress.ui
===================================================================
--- yali-2.4.1_pre20110530.orig/yali/gui/Ui/installprogress.ui
+++ yali-2.4.1_pre20110530/yali/gui/Ui/installprogress.ui
@@ -19,6 +19,18 @@
    </property>
    <item>
     <widget class="QWidget" name="InstallProgressFrame" native="true">
+     <property name="minimumSize">
+      <size>
+       <width>640</width>
+       <height>113</height>
+      </size>
+     </property>
+     <property name="maximumSize">
+      <size>
+       <width>640</width>
+       <height>113</height>
+      </size>
+     </property>
      <property name="styleSheet">
       <string notr="true">#InstallProgressFrame{
 	background-color: rgba(0, 0, 0, 100);
@@ -51,9 +63,27 @@
           <height>16777215</height>
          </size>
         </property>
+        <property name="styleSheet">
+         <string notr="true"/>
+        </property>
         <property name="value">
          <number>0</number>
         </property>
+        <property name="alignment">
+         <set>Qt::AlignCenter</set>
+        </property>
+        <property name="textVisible">
+         <bool>true</bool>
+        </property>
+        <property name="orientation">
+         <enum>Qt::Horizontal</enum>
+        </property>
+        <property name="invertedAppearance">
+         <bool>false</bool>
+        </property>
+        <property name="textDirection">
+         <enum>QProgressBar::TopToBottom</enum>
+        </property>
        </widget>
       </item>
       <item row="0" column="0">
@@ -64,6 +94,17 @@
           <height>40</height>
          </size>
         </property>
+        <property name="maximumSize">
+         <size>
+          <width>16777215</width>
+          <height>40</height>
+         </size>
+        </property>
+        <property name="font">
+         <font>
+          <pointsize>8</pointsize>
+         </font>
+        </property>
         <property name="text">
          <string>Installing Package: </string>
         </property>
