diff -aur yali~/storage/partitioning.py yali/storage/partitioning.py
--- yali~/storage/partitioning.py	2011-05-30 01:47:45.000000000 +0200
+++ yali/storage/partitioning.py	2013-07-16 14:37:32.540499702 +0200
@@ -1404,6 +1404,7 @@
     disks.extend(whole)
     disks.sort(key=lambda d: d.name)
     for disk in disks:
+        if not hasattr(disk.format, "freePartitions"): continue
         for partition in disk.format.freePartitions:
             if partition.getSize(unit="MB") > 1:
                 freeSpaces = True
