--- yali/pisiiface.py~	2013-06-11 09:14:12.128373788 +0200
+++ yali/pisiiface.py	2013-06-11 09:14:22.940373393 +0200
@@ -219,7 +219,7 @@
         packages = mergePackagesWithRepoPath(getCollectionPackages(collectionIndex, kernels=ignoreKernels))
     else:
         # Get packages with their full paths
-        packages = glob.glob('%s/repo/*.pisi' % ctx.consts.source_dir)
+        packages = glob.glob('%s/repo/*/*/*.pisi' % ctx.consts.source_dir)
 
     return packages
 
