--- yali/gui/ScrKeyboard.py~	2011-05-30 01:47:45.000000000 +0200
+++ yali/gui/ScrKeyboard.py	2013-06-11 20:37:44.000000000 +0200
@@ -55,6 +55,8 @@
 
 
         self.ui.keyboard_list.setCurrentIndex(self.default_layout_index)
+        # apply default layout
+        self.slotLayoutChanged()
 
         self.connect(self.ui.keyboard_list, SIGNAL("currentIndexChanged(int)"),
                 self.slotLayoutChanged)
