--- yali/storage/operations.py~	2013-04-06 17:51:00.000000000 +0200
+++ yali/storage/operations.py	2013-04-06 17:54:31.000000000 +0200
@@ -240,7 +240,7 @@
             olduuid = udev_device_get_uuid(info)
         except AttributeError:
             olduuid = ""
-        ctx.logger.info(">>>INFO: %s" % info)
+
         self.device.format.create(intf=intf,
                                   device=self.device.path,
                                   options=self.device.formatArgs)
@@ -248,12 +248,16 @@
         # Get the UUID now that the format is created
         yali.baseudev.udev_settle()
         self.device.updateSysfsPath()
-        newuuid = olduuid
-        # wait for new uuid
-        while newuuid == olduuid:
+        if olduuid:
+            newuuid = olduuid
+            # wait for new uuid
+            while newuuid == olduuid:
+                info = udev_get_block_device(self.device.sysfsPath)
+                newuuid = udev_device_get_uuid(info)
+            self.device.format.uuid = newuuid
+        else:
             info = udev_get_block_device(self.device.sysfsPath)
-            newuuid = udev_device_get_uuid(info)
-        self.device.format.uuid = newuuid
+            self.device.format.uuid = udev_device_get_uuid(info)
 
     def cancel(self):
         self.device.format = self.origFormat
