--- yali.orig/gui/ScrGoodbye.py	2011-05-30 01:47:45.000000000 +0200
+++ yali/gui/ScrGoodbye.py	2013-01-17 12:34:36.000000000 +0100
@@ -86,9 +86,9 @@
             (ctx.flags.install_type == ctx.STEP_RESCUE and ctx.installData.rescueMode == ctx.RESCUE_GRUB)) and \
             ctx.bootloader.stage1Device:
             postInstallOperations.append(yali.postinstall.Operation(_("Writing bootloader..."), yali.postinstall.writeBootLooder))
+            postInstallOperations.append(yali.postinstall.Operation(_("Installing bootloader..."), yali.postinstall.installBootloader))
             postInstallOperations.append(yali.postinstall.Operation(_("Stopping to D-Bus..."), yali.util.stop_dbus))
             postInstallOperations.append(yali.postinstall.Operation(_("Teardown storage subsystem..."), yali.postinstall.teardownStorage))
-            postInstallOperations.append(yali.postinstall.Operation(_("Installing bootloader..."), yali.postinstall.installBootloader))
 
         if ctx.flags.install_type == ctx.STEP_DEFAULT or ctx.flags.install_type == ctx.STEP_BASE or ctx.flags.install_type == ctx.STEP_RESCUE:
             postInstallOperations.append(yali.postinstall.Operation(_("Stopping to D-Bus..."), yali.util.stop_dbus))
--- yali.orig/postinstall.py	2011-05-30 01:47:45.000000000 +0200
+++ yali/postinstall.py	2013-01-17 13:05:39.000000000 +0100
@@ -1,3 +1,4 @@
+
 #
 # Copyright (C) 2005-2010 TUBITAK/UEKAE
 #
@@ -269,18 +270,18 @@
 def writeBootLooder():
     ctx.logger.info("Generating grub configuration file")
     if ctx.storage.storageset.active:
-        ctx.bootloader.write()
+        yali.util.chroot("grub2-mkconfig -o /boot/grub2/grub.cfg")
         return True
 
     ctx.logger.debug("writeBootLooder:StorageSet not activated")
     return False
 
 def installBootloader():
-    if len(ctx.mountCount):
-        ctx.logger.debug("StorageSet is already active. Bootloader installBootloader failed")
-        return False
+    #if len(ctx.mountCount):
+    #    ctx.logger.debug("StorageSet is already active. Bootloader installBootloader failed")
+    #    return False
 
-    rc = ctx.bootloader.install()
+    rc = ctx.bootloader.install2()
     if rc:
         ctx.logger.debug("Bootloader installation failed")
         return False
--- yali.orig/storage/bootloader.py	2013-01-17 15:07:04.865462774 +0100
+++ yali/storage/bootloader.py	2013-01-17 13:12:35.000000000 +0100
@@ -372,3 +372,7 @@
         rc = yali.util.run_batch("grub", ["--no-floppy", "--device-map=/tmp/device.map", "--batch < ", "/tmp/batch"])[0]
         yali.util.sync()
         return rc
+
+    def install2(self):
+        stage1Devices = get_physical_devices(self.storage, self.storage.devicetree.getDeviceByName(self.stage1Device))
+        yali.util.chroot("grub2-install --recheck %s" % stage1Devices[0].path)
