--- yali/gui/Ui/bootloaderwidget.py~	2013-06-21 21:58:17.037695929 +0200
+++ yali/gui/Ui/bootloaderwidget.py	2013-06-21 21:59:27.227693365 +0200
@@ -101,9 +101,9 @@
         self.noInstall = QtGui.QRadioButton(self.advancedSettingsBox)
         self.noInstall.setObjectName(_fromUtf8("noInstall"))
         self.gridLayout_2.addWidget(self.noInstall, 3, 0, 1, 2)
-        self.installPartition = QtGui.QRadioButton(self.advancedSettingsBox)
-        self.installPartition.setObjectName(_fromUtf8("installPartition"))
-        self.gridLayout_2.addWidget(self.installPartition, 2, 0, 1, 2)
+#        self.installPartition = QtGui.QRadioButton(self.advancedSettingsBox)
+#        self.installPartition.setObjectName(_fromUtf8("installPartition"))
+#        self.gridLayout_2.addWidget(self.installPartition, 2, 0, 1, 2)
         self.installMBR = QtGui.QRadioButton(self.advancedSettingsBox)
         sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Fixed, QtGui.QSizePolicy.Preferred)
         sizePolicy.setHorizontalStretch(0)
@@ -141,6 +141,6 @@
         self.advancedSettings.setText(i18n("Advanced settings"))
         self.defaultSettings.setText(i18n("Default settings"))
         self.noInstall.setText(i18n("Don\'t install bootloader"))
-        self.installPartition.setText(i18n("The partition where Pisi Linux is installed"))
+#        self.installPartition.setText(i18n("The partition where Pisi Linux is installed"))
         self.installMBR.setText(i18n("The first sector of"))
 
--- yali/gui/Ui/bootloaderwidget.ui~	2013-06-21 21:58:22.491695730 +0200
+++ yali/gui/Ui/bootloaderwidget.ui	2013-06-21 21:58:52.585694630 +0200
@@ -189,20 +189,13 @@
                 <property name="verticalSpacing">
                  <number>5</number>
                 </property>
-                <item row="3" column="0" colspan="2">
+                <item row="1" column="0" colspan="2">
                  <widget class="QRadioButton" name="noInstall">
                   <property name="text">
                    <string>Don't install bootloader</string>
                   </property>
                  </widget>
                 </item>
-                <item row="2" column="0" colspan="2">
-                 <widget class="QRadioButton" name="installPartition">
-                  <property name="text">
-                   <string>The partition where Pisi Linux is installed</string>
-                  </property>
-                 </widget>
-                </item>
                 <item row="0" column="0">
                  <widget class="QRadioButton" name="installMBR">
                   <property name="sizePolicy">
