diff --git a/yali/gui/YaliDialog.py b/yali/gui/YaliDialog.py
index f53c958..1e21752 100644
--- a/yali/gui/YaliDialog.py
+++ b/yali/gui/YaliDialog.py
@@ -184,6 +184,7 @@ class MessageWindow:
         elif customIcon == "info":
             icon = "info"
 
+        text = "<qt>%s</qt>" % text.replace("\n", " ")
         self.msgBox.setText(text)
         if detailed:
             self.msgBox.setDetailedText(unicode(longText))
