--- a/yali/gui/ScrDateTime.py	2011-05-30 01:47:45.000000000 +0200
+++ b/yali/gui/ScrDateTime.py	2013-01-02 21:02:26.226488301 +0100
@@ -208,7 +208,8 @@
                 self.pds_messagebox.setMessage(_("Storage Devices initialising..."))
                 self.pds_messagebox.animate(start=MIDCENTER, stop=MIDCENTER)
                 ctx.mainScreen.step_increment = 0
-                self.pthread.start()
+#                self.pthread.start()
+                self.startInit()
                 QTimer.singleShot(2, self.startStorageInitialize)
                 return False
 
--- a/yali/gui/ScrAdmin.py	2013-01-02 21:17:57.333468841 +0100
+++ b/yali/gui/ScrAdmin.py	2013-01-02 21:18:11.973468306 +0100
@@ -96,7 +96,8 @@
                 self.pds_messagebox.setMessage(_("Storage Devices initialising..."))
                 self.pds_messagebox.animate(start=MIDCENTER, stop=MIDCENTER)
                 ctx.mainScreen.step_increment = 0
-                self.pthread.start()
+#                self.pthread.start()
+                self.startInit()
                 QTimer.singleShot(2, self.startStorageInitialize)
                 return False
 
