Index: yali-2.4.0_20101121/yali/constants.py
===================================================================
--- yali-2.4.0_20101121.orig/yali/constants.py
+++ yali-2.4.0_20101121/yali/constants.py
@@ -79,8 +79,8 @@ class Constants:
         self.__c.cd_repo_name = "pardus-cd"
         self.__c.dvd_repo_name = "pardus-dvd"
         self.__c.cd_repo_uri = os.path.join(self.__c.source_dir, "repo/pisi-index.xml.bz2")
-        self.__c.pardus_repo_name = "pardus-2009"
-        self.__c.pardus_repo_uri = "http://packages.pardus.org.tr/pardus-2009/pisi-index.xml.bz2"
+        self.__c.pardus_repo_name = "pardus-anka"
+        self.__c.pardus_repo_uri = "http://anka.pardus-linux.org/2013/anka/repos/alpha/pisi-index.xml.xz"
         self.__c.pisi_index_file = os.path.join(self.__c.data_dir,"data/pisi-index.xml.bz2")
         self.__c.pisi_index_file_sum = os.path.join(self.__c.data_dir,"data/pisi-index.xml.bz2.sha1sum")
         self.__c.lang = locale.getdefaultlocale()[0][:2]
