diff --git a/yali/gui/lvm_gui.py b/yali/gui/lvm_gui.py
index 63dc4fe..5b480a0 100644
--- a/yali/gui/lvm_gui.py
+++ b/yali/gui/lvm_gui.py
@@ -673,17 +673,18 @@ class LogicalVolumeEditor:
 
 
             if not self.origrequest.exists:
-                badsize = 0
+                size = 0
                 try:
                     size = long(widget.sizeSpin.value())
                 except:
                     size = 1
 
-                if badsize or size <= 0:
+                if mountpoint and mountpoint == '/' and size < ctx.consts.min_root_size:
                     self.intf.messageWindow(_("Illegal size"),
-                                                   _("The requested size as entered is "
-                                                     "not a valid number greater than 0."),
-                                                   customIcon="error")
+                                            _("The requested size as entered is "
+                                              "not a valid number greater than %s.")
+                                            % ctx.consts.min_root_size,
+                                            customIcon="error")
                     continue
             else:
                 size = self.origrequest.size
diff --git a/yali/gui/partition_gui.py b/yali/gui/partition_gui.py
index 1110e7e..65f8062 100644
--- a/yali/gui/partition_gui.py
+++ b/yali/gui/partition_gui.py
@@ -91,6 +91,13 @@ class PartitionEditor:
                     primary = None
 
                 size = widget.sizeSpin.value()
+                if mountpoint and mountpoint == '/' and size < ctx.consts.min_root_size:
+                    self.intf.messageWindow(_("Illegal size"),
+                                            _("The requested size as entered is "
+                                              "not a valid number greater than %s.")
+                                            % ctx.consts.min_root_size,
+                                            customIcon="error")
+                    continue
 
                 formatType = str(widget.filesystemMenu.currentText())
                 format = formats.getFormat(formatType, mountpoint=mountpoint)
