Index: hplip-3.9.10/configure.in
===================================================================
--- hplip-3.9.10.orig/configure.in
+++ hplip-3.9.10/configure.in
@@ -438,9 +438,7 @@ AM_CONDITIONAL(POLICYKIT_INSTALL, test x
 
 # Determine where the PolicyKit policy file goes
 if test "$policykit" = "yes"; then
-   if test -d /usr/share/polkit-1; then
-      policykit_dir=/usr/share/polkit-1/actions
-   elif test -d /usr/share/PolicyKit; then
+   if test -d /usr/share/PolicyKit; then
       policykit_dir=/usr/share/PolicyKit/policy
    else
       policykit_dir=/usr/share/PolicyKit/policy
