--- setup.py.orig	2010-05-15 17:17:03.479224260 +0300
+++ setup.py	2010-05-15 17:17:47.094223002 +0300
@@ -75,7 +75,7 @@
 	  scripts = ['rdiff-backup', 'rdiff-backup-statistics'],
 	  data_files = [('share/man/man1', ['rdiff-backup.1',
 										'rdiff-backup-statistics.1']),
-					('share/doc/rdiff-backup-%s' % (version_string,),
+					('share/doc/rdiff-backup',
 					 ['CHANGELOG', 'COPYING', 'README', 'FAQ.html'])],
 					**extra_options)
 
