diff -Nuar qwtplot3d.OSmanOS/include/qwt3d_function.h qwtplot3d/include/qwt3d_function.h
--- qwtplot3d.OSmanOS/include/qwt3d_function.h	2005-07-10 16:54:52.000000000 +0300
+++ qwtplot3d/include/qwt3d_function.h	2013-05-11 20:49:59.044810804 +0300
@@ -2,6 +2,7 @@
 #define qwt3d_function_h__2004_03_05_13_51_begin_guarded_code
 
 #include "qwt3d_gridmapping.h"
+#include <cstdio>
 
 namespace Qwt3D
 {
diff -Nuar qwtplot3d.OSmanOS/include/qwt3d_helper.h qwtplot3d/include/qwt3d_helper.h
--- qwtplot3d.OSmanOS/include/qwt3d_helper.h	2005-07-19 17:40:28.000000000 +0300
+++ qwtplot3d/include/qwt3d_helper.h	2013-05-11 20:49:07.572812324 +0300
@@ -5,6 +5,7 @@
 #include <float.h>
 #include <vector>
 #include <algorithm>
+#include <GL/glu.h>
 
 namespace
 {
diff -Nuar qwtplot3d.OSmanOS/include/qwt3d_openglhelper.h qwtplot3d/include/qwt3d_openglhelper.h
--- qwtplot3d.OSmanOS/include/qwt3d_openglhelper.h	2005-07-19 17:40:28.000000000 +0300
+++ qwtplot3d/include/qwt3d_openglhelper.h	2013-05-11 20:42:28.085824128 +0300
@@ -8,6 +8,8 @@
 #include <QtOpenGL/qgl.h>
 #endif
 
+#include <GL/glu.h>
+
 namespace Qwt3D
 {
 
