diff -up libnl-1.1/include/netlink-local.h.limits libnl-1.1/include/netlink-local.h
--- libnl-1.1/include/netlink-local.h.limits	2008-02-22 11:31:18.000000000 -0500
+++ libnl-1.1/include/netlink-local.h	2008-02-22 11:32:57.000000000 -0500
@@ -22,6 +22,7 @@
 #include <time.h>
 #include <stdarg.h>
 #include <ctype.h>
+#include <limits.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <inttypes.h>
