Index: policykitlistener.cpp
===================================================================
--- policykitlistener.cpp.orig
+++ policykitlistener.cpp
@@ -22,6 +22,7 @@
 #include "AuthDialog.h"
 
 #include <KDebug>
+#include <KApplication>
 
 #include <PolkitQt1/Agent/Listener>
 #include <PolkitQt1/Agent/Session>
@@ -67,6 +68,8 @@ void PolicyKitListener::initiateAuthenti
         const PolkitQt1::Identity::List &identities,
         PolkitQt1::Agent::AsyncResult* result)
 {
+    kapp->updateUserTimestamp( 0 ); // HACK make sure it gets focus when shown
+
     kDebug() << "Initiating authentication";
 
     if (m_inProgress) {
