Index: polkit-kde-agent-1-0.99.0/AuthDialog.cpp
===================================================================
--- polkit-kde-agent-1-0.99.0.orig/AuthDialog.cpp
+++ polkit-kde-agent-1-0.99.0/AuthDialog.cpp
@@ -26,8 +26,9 @@
 #include <QtCore/QProcess>
 #include <QtGui/QPainter>
 #include <QtGui/QStandardItemModel>
-#include <KDebug>
+#include <QTimer>
 
+#include <KDebug>
 #include <KToolInvocation>
 #include <KUser>
 
@@ -124,6 +125,8 @@ AuthDialog::AuthDialog(const QString &ac
                 this, SLOT(on_userCB_currentIndexChanged(int)));
 
         createUserCB(identities);
+        QTimer::singleShot(0, this, SLOT(selectUser()));
+
     } else {
         userCB->addItem("", QVariant(identities[0].toString()));
         userCB->setCurrentIndex(0);
@@ -134,6 +137,25 @@ AuthDialog::~AuthDialog()
 {
 }
 
+void AuthDialog::selectUser()
+{
+    char *userName = getenv("USER");
+    int currentUserIndex = -1;
+
+    if (userName)
+        currentUserIndex = userCB->findData(qVariantFromValue<QString> (QString("unix-user:") + userName), Qt::UserRole);
+
+    if (currentUserIndex == -1)
+    {
+        qDebug() << "Current user not found and will not be selected in combobox.";
+    }
+    else
+    {
+        qDebug() << "Current user is " << QString(userName) << ", and will be selected.";
+        userCB->setCurrentIndex(currentUserIndex);
+    }
+}
+
 void AuthDialog::accept()
 {
     // Do nothing, do not close the dialog. This is needed so that the dialog stays
Index: polkit-kde-agent-1-0.99.0/AuthDialog.h
===================================================================
--- polkit-kde-agent-1-0.99.0.orig/AuthDialog.h
+++ polkit-kde-agent-1-0.99.0/AuthDialog.h
@@ -63,6 +63,7 @@ public slots:
 
 private slots:
     void on_userCB_currentIndexChanged(int index);
+    void selectUser();
     void notificationActivated(unsigned int action);
 
 private:
