From 47c076d6bbe8b4d131351642c8ba796271be74fe Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Mon, 3 Feb 2014 21:52:11 +0100
Subject: [PATCH] disable firebird unit test

---
 dbaccess/Module_dbaccess.mk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/dbaccess/Module_dbaccess.mk b/dbaccess/Module_dbaccess.mk
index b9d7d56..824f3f5 100644
--- a/dbaccess/Module_dbaccess.mk
+++ b/dbaccess/Module_dbaccess.mk
@@ -36,7 +36,6 @@ $(eval $(call gb_Module_add_l10n_targets,dbaccess,\
 
 ifeq ($(ENABLE_FIREBIRD_SDBC),TRUE)
 $(eval $(call gb_Module_add_check_targets,dbaccess,\
-    CppunitTest_dbaccess_firebird_test \
 ))
 endif
 
-- 
1.8.4.2

