diff --git a/sc/inc/dpmacros.hxx b/sc/inc/calcmacros.hxx
similarity index 86%
rename from sc/inc/dpmacros.hxx
rename to sc/inc/calcmacros.hxx
index 72659cc..03ca590 100644
--- a/sc/inc/dpmacros.hxx
+++ b/sc/inc/calcmacros.hxx
@@ -7,9 +7,10 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-#ifndef __SC_DPMACROS_HXX__
-#define __SC_DPMACROS_HXX__
+#ifndef SC_CALCMACROS_HXX
+#define SC_CALCMACROS_HXX
 
+#define DEBUG_COLUMN_STORAGE 0
 #define DEBUG_PIVOT_TABLE 0
 
 #if DEBUG_PIVOT_TABLE
diff --git a/sc/inc/dpcache.hxx b/sc/inc/dpcache.hxx
index 5a75463..707749c 100644
--- a/sc/inc/dpcache.hxx
+++ b/sc/inc/dpcache.hxx
@@ -21,7 +21,7 @@
 
 #include "global.hxx"
 #include "dpnumgroupinfo.hxx"
-#include "dpmacros.hxx"
+#include "calcmacros.hxx"
 #include "tools/date.hxx"
 
 #include <boost/noncopyable.hpp>
diff --git a/sc/inc/dpfilteredcache.hxx b/sc/inc/dpfilteredcache.hxx
index c65b95c..bff1e18 100644
--- a/sc/inc/dpfilteredcache.hxx
+++ b/sc/inc/dpfilteredcache.hxx
@@ -24,7 +24,7 @@
 #include "osl/mutex.hxx"
 #include "global.hxx"
 #include "dpitemdata.hxx"
-#include "dpmacros.hxx"
+#include "calcmacros.hxx"
 
 #include <vector>
 #include <boost/unordered_set.hpp>
diff --git a/sc/inc/dpitemdata.hxx b/sc/inc/dpitemdata.hxx
index 9137e2a..823a6c1 100644
--- a/sc/inc/dpitemdata.hxx
+++ b/sc/inc/dpitemdata.hxx
@@ -16,7 +16,7 @@
 #include "sal/types.h"
 #include "tools/solar.h"
 #include "rtl/ustring.hxx"
-#include "dpmacros.hxx"
+#include "calcmacros.hxx"
 #include "dpglobal.hxx"
 
 /**
diff --git a/sc/inc/dpnumgroupinfo.hxx b/sc/inc/dpnumgroupinfo.hxx
index 523b792..6989fff 100644
--- a/sc/inc/dpnumgroupinfo.hxx
+++ b/sc/inc/dpnumgroupinfo.hxx
@@ -11,7 +11,7 @@
 #define __SC_DPNUMGROUPINFO_HXX__
 
 #include "scdllapi.h"
-#include "dpmacros.hxx"
+#include "calcmacros.hxx"
 
 struct ScDPNumGroupInfo
 {
diff --git a/sc/inc/dpobject.hxx b/sc/inc/dpobject.hxx
index 735b033..bbd3ba8 100644
--- a/sc/inc/dpobject.hxx
+++ b/sc/inc/dpobject.hxx
@@ -26,7 +26,7 @@
 #include "dpoutput.hxx"
 #include "dptypes.hxx"
 #include "pivot.hxx"
-#include "dpmacros.hxx"
+#include "calcmacros.hxx"
 
 #include <com/sun/star/sheet/XDimensionsSupplier.hpp>
 
diff --git a/sc/inc/dptabdat.hxx b/sc/inc/dptabdat.hxx
index 22a8ffb..3239a80 100644
--- a/sc/inc/dptabdat.hxx
+++ b/sc/inc/dptabdat.hxx
@@ -24,7 +24,7 @@
 #include "dpoutput.hxx"
 #include "dpfilteredcache.hxx"
 #include "dpcache.hxx"
-#include "dpmacros.hxx"
+#include "calcmacros.hxx"
 
 #include "svl/zforlist.hxx"
 
diff --git a/sc/inc/dptabres.hxx b/sc/inc/dptabres.hxx
index e6289be..cab2da8 100644
--- a/sc/inc/dptabres.hxx
+++ b/sc/inc/dptabres.hxx
@@ -22,7 +22,7 @@
 
 #include "global.hxx"
 #include "dpfilteredcache.hxx"
-#include "dpmacros.hxx"
+#include "calcmacros.hxx"
 
 #include <tools/string.hxx>
 #include <com/sun/star/sheet/MemberResult.hpp>
diff --git a/sc/inc/mtvelements.hxx b/sc/inc/mtvelements.hxx
index 0ebf8be..ad65418 100644
--- a/sc/inc/mtvelements.hxx
+++ b/sc/inc/mtvelements.hxx
@@ -14,8 +14,7 @@
 #include "formulacell.hxx"
 #include "svl/broadcast.hxx"
 #include "editeng/editobj.hxx"
-
-#define DEBUG_COLUMN_STORAGE 0
+#include "calcmacros.hxx"
 
 #if DEBUG_COLUMN_STORAGE
 #ifdef NDEBUG
diff --git a/sc/inc/pivot.hxx b/sc/inc/pivot.hxx
index 85e19e0..12ed637 100644
--- a/sc/inc/pivot.hxx
+++ b/sc/inc/pivot.hxx
@@ -40,7 +40,7 @@
 #include "global.hxx"
 #include "address.hxx"
 #include "dpglobal.hxx"
-#include "dpmacros.hxx"
+#include "calcmacros.hxx"
 
 #include <vector>
 #include <boost/ptr_container/ptr_vector.hpp>
diff --git a/sc/source/core/data/dptabsrc.cxx b/sc/source/core/data/dptabsrc.cxx
index c38ee38..76ec9f5 100644
--- a/sc/source/core/data/dptabsrc.cxx
+++ b/sc/source/core/data/dptabsrc.cxx
@@ -45,7 +45,7 @@
 #include "dpitemdata.hxx"
 #include "dputil.hxx"
 #include "dpresfilter.hxx"
-#include "dpmacros.hxx"
+#include "calcmacros.hxx"
 
 #include <com/sun/star/beans/PropertyAttribute.hpp>
 #include <com/sun/star/sheet/DataPilotFieldFilter.hpp>
diff --git a/sc/source/ui/dbgui/fieldwnd.cxx b/sc/source/ui/dbgui/fieldwnd.cxx
index 3701081..8673e7f 100644
--- a/sc/source/ui/dbgui/fieldwnd.cxx
+++ b/sc/source/ui/dbgui/fieldwnd.cxx
@@ -48,7 +48,7 @@
 
 #include "pvlaydlg.hxx"
 #include "dpuiglobal.hxx"
-#include "dpmacros.hxx"
+#include "calcmacros.hxx"
 #include "AccessibleDataPilotControl.hxx"
 #include "scresid.hxx"
 #include "pivot.hrc"