From 5b03bc8a4d92fee2fdfdca4917b321985feb930a Mon Sep 17 00:00:00 2001
From: Matúš Kukan <matus.kukan@collabora.com>
Date: Mon, 02 Dec 2013 19:53:54 +0000
Subject: ugly build-fix for libmerged

Change-Id: I6c107ea5e780ea1178759e2ee827f9bd6d5ede3a
---
diff --git a/Library_merged.mk b/Library_merged.mk
index 1e96111..3490a8a 100644
--- a/Library_merged.mk
+++ b/Library_merged.mk
@@ -31,6 +31,8 @@ $(eval $(call gb_Library_use_libraries,merged,\
 ))
 
 $(eval $(call gb_Library_use_externals,merged,\
+	icui18n \
+	icuuc \
 	boostdatetime \
 	$(call gb_Helper_optional,DESKTOP,clucene) \
 	cups \
@@ -43,8 +45,6 @@ $(eval $(call gb_Library_use_externals,merged,\
 	$(if $(ENABLE_GTK),gtk) \
 	harfbuzz \
 	hunspell \
-	icui18n \
-	icuuc \
 	jpeg \
 	lcms2 \
 	$(call gb_Helper_optional,LIBATOMIC_OPS,libatomic_ops) \
--
cgit v0.9.0.2-2-gbebe
