--- a/shell/Module_shell.mk	2013-02-13 15:48:02.880128705 +0000
+++ b/shell/Module_shell.mk	2013-02-13 16:37:33.766353120 +0000
@@ -32,14 +32,6 @@
 	Library_localebe \
 ))
 
-ifeq ($(OS),LINUX)
-ifeq ($(ENABLE_GIO),TRUE)
-$(eval $(call gb_Module_add_targets,shell,\
-	Library_losessioninstall \
-))
-endif
-endif
-
 ifeq ($(ENABLE_GCONF),TRUE)
 $(eval $(call gb_Module_add_targets,shell,\
 	Library_gconfbe \
diff --git a/Repository.mk b/Repository.mk
index d0de1d8..526fea3 100644
--- a/Repository.mk
+++ b/Repository.mk
@@ -229,7 +229,6 @@ $(eval $(call gb_Helper_register_libraries_for_install,OOOLIBS,GNOME, \
     $(if $(ENABLE_GTK3),vclplug_gtk3) \
     $(if $(ENABLE_GCONF),gconfbe1) \
     $(if $(ENABLE_SYSTRAY_GTK),qstart_gtk) \
-    $(if $(ENABLE_GIO),losessioninstall) \
     $(if $(ENABLE_GIO),ucpgio1) \
     $(if $(ENABLE_GNOMEVFS),ucpgvfs1) \
 ))
diff --git a/postprocess/Rdb_services.mk b/postprocess/Rdb_services.mk
index 936aac2..7d26eb4 100644
--- a/postprocess/Rdb_services.mk
+++ b/postprocess/Rdb_services.mk
@@ -197,7 +197,6 @@ $(eval $(call gb_Rdb_add_components,services,\
 		shell/source/backends/gconfbe/gconfbe1 \
 	) \
 	$(if $(ENABLE_GIO), \
-		shell/source/sessioninstall/losessioninstall \
 		ucb/source/ucp/gio/ucpgio \
 	) \
 	$(if $(ENABLE_GNOMEVFS), \
diff --git a/configure.ac b/configure.ac
index 059188f..23503da 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7113,17 +7113,6 @@ if test "$with_system_jpeg" = "yes"; then
         [AC_MSG_ERROR(jpeg.h not found. install libjpeg)], [])
     AC_CHECK_LIB(jpeg, jpeg_resync_to_restart, [ JPEG3RDLIB=-ljpeg ],
     [AC_MSG_CHECKING(jpeg library not found or fuctional)], [])
-    AC_MSG_CHECKING([[libjpeg supports JERR_BAD_CROP_SPEC (jpeg-7 API)]])
-	AC_COMPILE_IFELSE([ AC_LANG_SOURCE(
-			  [[
-#include <stdio.h>
-#include <jerror.h>
-int main(int c, char**v) { printf("%d\n", JERR_BAD_CROP_SPEC); return 0; }
-			  ]]) ],
-			  [AC_MSG_RESULT(yes)],
-		          [AC_MSG_RESULT(no)
-                           AC_MSG_ERROR(jpeg library version >= 7 or jpeg-turbo version >= 1.1 required)
-                          ])
     libo_MINGW_CHECK_DLL([libjpeg])
 else
     AC_MSG_RESULT([internal])
diff --git a/vcl/source/filter/jpeg/transupp.c b/vcl/source/filter/jpeg/transupp.c
index 33da623..4cae59e 100644
--- a/vcl/source/filter/jpeg/transupp.c
+++ b/vcl/source/filter/jpeg/transupp.c
@@ -1006,7 +1006,7 @@ jtransform_request_workspace (j_decompress_ptr srcinfo,
       info->crop_yoffset = 0;   /* default to +0 */
     if (info->crop_xoffset >= info->output_width ||
     info->crop_yoffset >= info->output_height)
-      ERREXIT(srcinfo, JERR_BAD_CROP_SPEC);
+      ERREXIT(srcinfo, JERR_CONVERSION_NOTIMPL);
     if (info->crop_width_set == JCROP_UNSET)
       info->crop_width = info->output_width - info->crop_xoffset;
     if (info->crop_height_set == JCROP_UNSET)
@@ -1016,7 +1016,7 @@ jtransform_request_workspace (j_decompress_ptr srcinfo,
     info->crop_height <= 0 || info->crop_height > info->output_height ||
     info->crop_xoffset > info->output_width - info->crop_width ||
     info->crop_yoffset > info->output_height - info->crop_height)
-      ERREXIT(srcinfo, JERR_BAD_CROP_SPEC);
+      ERREXIT(srcinfo, JERR_CONVERSION_NOTIMPL);
     /* Convert negative crop offsets into regular offsets */
     if (info->crop_xoffset_set == JCROP_NEG)
       xoffset = info->output_width - info->crop_width - info->crop_xoffset;
