--- a/configure.ac	2011-11-10 13:22:02.062474990 +0000
+++ b/configure.ac	2011-11-10 13:22:35.072472880 +0000
@@ -9659,16 +9659,6 @@ if test "$_os" != Android -a "$_os" != iOS -a "$ENABLE_PDFIMPORT" != FALSE; then
       AC_MSG_RESULT([external])
       SYSTEM_POPPLER=YES
       PKG_CHECK_MODULES( POPPLER, poppler >= 0.8.0 )
-      AC_LANG_PUSH([C++])
-      save_CXXFLAGS=$CXXFLAGS
-      save_CPPFLAGS=$CPPFLAGS
-      CXXFLAGS="$CXXFLAGS $POPPLER_CFLAGS"
-      CPPFLAGS="$CPPFLAGS $POPPLER_CFLAGS"
-      AC_CHECK_HEADER([cpp/poppler-version.h], [],
-                   [AC_MSG_ERROR([cpp/poppler-version.h not found. Install poppler])], [])
-      CXXFLAGS=$save_CXXFLAGS
-      CPPFLAGS=$save_CPPFLAGS
-      AC_LANG_POP([C++])
       libo_MINGW_CHECK_DLL([libpoppler])
   else
       AC_MSG_RESULT([internal])
--- a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx	2011-11-10 13:22:11.949473646 +0000
+++ b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx	2011-11-10 13:23:24.889473889 +0000
@@ -64,16 +64,7 @@
 class GfxPath;
 class GfxFont;
 class PDFDoc;
-#ifndef SYSTEM_POPPLER
 #define POPPLER_CHECK_VERSION(major,minor,micro) (0)
-typedef GString GooString;
-#else
-#include <cpp/poppler-version.h>
-#define POPPLER_CHECK_VERSION(major,minor,micro) \
-  (POPPLER_VERSION_MAJOR > (major) || \
-   (POPPLER_VERSION_MAJOR == (major) && POPPLER_VERSION_MINOR > (minor)) || \
-   (POPPLER_VERSION_MAJOR == (major) && POPPLER_VERSION_MINOR == (minor) && POPPLER_VERSION_MICRO >= (micro)))
-#endif
 
 namespace pdfi
 {
--- a/sdext/Module_sdext.mk
+++ b/sdext/Module_sdext.mk
@@ -32,11 +32,6 @@ $(eval $(call gb_Module_add_targets,sdext,\
     StaticLibrary_pdfimport_s \
 ))
 
-$(eval $(call gb_Module_add_check_targets,sdext,\
-    CppunitTest_pdfimport \
-    Executable_pdf2xml \
-    Executable_pdfunzip \
-))
 endif
 
 # vim:set noet sw=4 ts=4:
