--- a/scp2/source/ooo/common_brand.scp
+++ b/scp2/source/ooo/common_brand.scp
@@ -1252,6 +1252,7 @@
     Section = "Bootstrap";
     Key = "URE_MORE_JAVA_CLASSPATH_URLS";
     Value = "";
+    ValueList1 = "$ORIGIN/../ure-link/lib";
 #ifdef SYSTEM_HSQLDB
     ValueList1 = HSQLDB_JAR;
 #endif
--- a/odk/pack/gendocu/makefile.mk
+++ b/odk/pack/gendocu/makefile.mk
@@ -60,9 +60,6 @@
         $(JAVA_SRC_DIR)$/com$/sun$/star$/beans$/OfficeWindow.java
 
 JAVADOCPARAMS= -use -splitindex -windowtitle "Java UNO Runtime Reference" -header $(JAVADOCREFNAME) -d $(DESTDIRGENJAVAREF) -sourcepath $(JAVA_SRC_DIR) -classpath $(SOLARBINDIR)$/ridl.jar -linkoffline ../../common/ref ./uno -linkoffline http://java.sun.com/j2se/1.5/docs/api ./java $(JAVA_PACKAGES)
-.IF "$(JAVADOCISGJDOC)" != "yes"
-JAVADOCPARAMS+= -notimestamp
-.ENDIF
 
 JAVADOCLOG = $(MISC)$/javadoc_log.txt
 
