--- a/shell/Module_shell.mk	2013-02-13 15:48:02.880128705 +0000
+++ b/shell/Module_shell.mk	2013-02-13 16:37:33.766353120 +0000
@@ -32,14 +32,6 @@
 	Library_localebe \
 ))
 
-ifeq ($(OS),LINUX)
-ifeq ($(ENABLE_GIO),TRUE)
-$(eval $(call gb_Module_add_targets,shell,\
-	Library_losessioninstall \
-))
-endif
-endif
-
 ifeq ($(ENABLE_GCONF),TRUE)
 $(eval $(call gb_Module_add_targets,shell,\
 	Library_gconfbe \
--- a/scp2/source/gnome/file_gnome.scp	2013-02-13 16:38:19.475352623 +0000
+++ b/scp2/source/gnome/file_gnome.scp	2013-02-13 16:38:28.799352829 +0000
@@ -33,8 +33,6 @@
     Dir = SCP2_OOO_BIN_DIR;
     Name = STRING(CONCAT2(ucpgio1.uno,UNXSUFFIX));
 End
-
-SPECIAL_COMPONENT_LIB_FILE(gid_File_Lib_Losessioninstall, losessioninstall.uno)
 #endif
 
 #ifdef ENABLE_GNOMEVFS
--- a/scp2/source/gnome/module_gnome.scp	2013-02-13 16:38:19.475352623 +0000
+++ b/scp2/source/gnome/module_gnome.scp	2013-02-13 16:38:41.308353064 +0000
@@ -55,7 +55,6 @@
 #endif
 #ifdef ENABLE_GIO
       gid_File_Lib_GIO,
-      gid_File_Lib_Losessioninstall,
 #endif
 #ifdef ENABLE_GCONF
       gid_File_Lib_Gconfbe,
--- a/postprocess/packcomponents/makefile.mk	2013-02-15 11:29:08.647352823 +0000
+++ b/postprocess/packcomponents/makefile.mk	2013-02-15 11:29:13.594376055 +0000
@@ -246,7 +246,6 @@
 
 .IF "$(ENABLE_GIO)" != ""
 my_components += \
-    component/shell/source/sessioninstall/losessioninstall \
     component/ucb/source/ucp/gio/ucpgio
 .END
 
