--- a/solenv/bin/modules/installer/worker.pm	2010-11-20 13:36:38.000000000 +0000
+++ b/solenv/bin/modules/installer/worker.pm	2010-11-20 13:36:42.000000000 +0000
@@ -204,7 +204,6 @@
 
         $installdir = installer::systemactions::create_directories("install", $languageref);
         installer::logger::print_message( "... creating installation set in $installdir ...\n" );
-        remove_old_installation_sets($installdir);
         my $inprogressinstalldir = $installdir . "_inprogress";
         installer::systemactions::rename_directory($installdir, $inprogressinstalldir);
         $installdir = $inprogressinstalldir;
