Index: configuration/ppds/SGENPRT.PS
===================================================================
RCS file: /cvs/external/psprint_config/configuration/ppds/SGENPRT.PS,v
retrieving revision 1.3
diff -u -p -u -r1.3 SGENPRT.PS
--- openoffice.org.orig/psprint_config/configuration/ppds/SGENPRT.PS	26 Nov 2004 16:10:35 -0000	1.3
+++ openoffice.org/psprint_config/configuration/ppds/SGENPRT.PS	30 Aug 2005 11:17:51 -0000
@@ -58,6 +58,7 @@
 *ColorDevice: True
 *DefaultColorSpace: RGB
 *LanguageLevel: "2"
+*TTRasterizer: Type42
 
 *% --- For None Color or old PostScript(R) printers use following lines ---
 *% *ColorDevice: False
