--- sw/Module_sw.mk~    2013-05-14 08:06:42.893439456 +0300
+++ sw/Module_sw.mk     2013-05-14 08:07:13.399439165 +0300
@@ -44,7 +44,6 @@
 $(eval $(call gb_Module_add_slowcheck_targets,sw,\
     CppunitTest_sw_macros_test \
     CppunitTest_sw_subsequent_ooxmlexport \
-    CppunitTest_sw_subsequent_ooxmlimport \
     CppunitTest_sw_subsequent_ww8export \
     CppunitTest_sw_subsequent_ww8import \
     CppunitTest_sw_subsequent_rtfexport \
