--- vcl/qa/cppunit/graphicfilter/filters-test.cxx~	2013-05-13 21:43:31.244496437 +0200
+++ vcl/qa/cppunit/graphicfilter/filters-test.cxx	2013-05-13 21:44:06.167495161 +0200
@@ -90,10 +90,6 @@
         rtl::OUString());
 
     testDir(rtl::OUString(),
-        getURLFromSrc("/vcl/qa/cppunit/graphicfilter/data/jpg/"),
-        rtl::OUString());
-
-    testDir(rtl::OUString(),
         getURLFromSrc("/vcl/qa/cppunit/graphicfilter/data/gif/"),
         rtl::OUString());
 
