diff -p -up tzdata/tzdata2012c/backward.orig tzdata/tzdata2012c/backward
--- tzdata/tzdata2012f/backward.orig	2010-11-01 11:18:22.000000000 -0200
+++ tzdata/tzdata2012f/backward	2010-11-03 13:10:42.657695694 -0200
@@ -116,3 +116,18 @@ Link	Etc/UTC			UTC
 Link	Etc/UTC			Universal
 Link	Europe/Moscow		W-SU
 Link	Etc/UTC			Zulu
+
+# links requested by users (to access timezones by the name of main cities)
+# pablo@mandriva.com
+# Canada/Saskatchewan 
+Link	America/Regina	America/Saskatoon
+# Canada/Mountain (Alberta)
+Link	America/Edmonton	America/Calgary
+# Canada/Atlantic (New Brunswick)
+Link	America/Halifax		America/Fredericton
+# America/Eastern (Ontario)
+Link	America/Montreal	America/Ontario
+# PRC
+Link	Asia/Shanghai		Asia/Beijing
+# South Africa
+Link	Africa/Johannesburg	Africa/Pretoria
