From c4d89dd11e2c33f6f5797fb82910b97d483fb599 Mon Sep 17 00:00:00 2001
From: Olivier Fourdan <fourdan@xfce.org>
Date: Wed, 18 Jul 2012 20:30:36 +0000
Subject: Revert part of git commit 8637c3a as this breaks deletetion

of current workspace (bug 8827)
---
diff --git a/src/workspaces.c b/src/workspaces.c
index fb19ab0..f57eec9 100644
--- a/src/workspaces.c
+++ b/src/workspaces.c
@@ -492,7 +492,8 @@ workspaceInsert (ScreenInfo * screen_info, guint position)
 void
 workspaceDelete (ScreenInfo * screen_info, guint position)
 {
-    guint count;
+    Client *c;
+    guint i, count;
 
     g_return_if_fail (screen_info != NULL);
 
@@ -504,6 +505,14 @@ workspaceDelete (ScreenInfo * screen_info, guint position)
         return;
     }
 
+    for (c = screen_info->clients, i = 0; i < screen_info->client_count; c = c->next, i++)
+    {
+        if (c->win_workspace > position)
+        {
+            clientSetWorkspace (c, c->win_workspace - 1, TRUE);
+        }
+    }
+
     workspaceSetCount(screen_info, count - 1);
 }
 
--
cgit v0.9.0.3
 
