From 28649d37c3103a5879aaea3c8bd7a4eb30820dc9 Mon Sep 17 00:00:00 2001
From: Jérôme Guelfucci <jeromeg@xfce.org>
Date: Tue, 06 Nov 2012 20:24:27 +0000
Subject: Fix wrong debug message (bug #7547).

---
diff --git a/src/placement.c b/src/placement.c
index 248d339..f926de3 100644
--- a/src/placement.c
+++ b/src/placement.c
@@ -627,7 +627,7 @@ static void
 mousePlacement (Client * c, int full_x, int full_y, int full_w, int full_h, int mx, int my)
 {
     g_return_if_fail (c != NULL);
-    TRACE ("entering centerPlacement");
+    TRACE ("entering mousePlacement");
 
     c->x = mx + frameLeft(c) - frameWidth(c) / 2;
     c->y = my + frameTop(c) - frameHeight(c) / 2;
--
cgit v0.9.0.3
 
