diff -Nuar a/src/vidalia/config/TorSettings.cpp b/src/vidalia/config/TorSettings.cpp
--- a/src/vidalia/config/TorSettings.cpp	2012-02-28 01:25:30.000000000 +0200
+++ b/src/vidalia/config/TorSettings.cpp	2013-08-05 02:50:06.805988511 +0300
@@ -77,10 +77,10 @@
   else
     setDefault(SETTING_TOR_EXECUTABLE, programFiles + "\\Tor\\tor.exe");
 #else
-  setDefault(SETTING_TOR_EXECUTABLE, "tor");
+  setDefault(SETTING_TOR_EXECUTABLE, "/usr/bin/tor");
 #endif
 
   setDefault(SETTING_TORRC,         Vidalia::dataDirectory() + "/torrc");
   setDefault(SETTING_CONTROL_ADDR,  "127.0.0.1");
   setDefault(SETTING_CONTROL_PORT,  9051);
