--- src/Makefile.orig	2009-12-28 15:34:36.896267754 +0200
+++ src/Makefile	2009-12-29 10:02:13.522666423 +0200
@@ -40,16 +40,16 @@
 	$(MKDLL)$@ $(COBJS:$(XO)=.d$(XO)) $(GLLIBS) $(OCAMLDLL)
 lablgl.cma: liblablgl$(XA) dlllablgl.dll $(LIBOBJS) $(CONFIG)
 	$(LINKER) -a -o $@ $(LIBOBJS) \
-	  -cclib -llablgl -dllib -llablgl \
+	  -cclib -llablgl -dllib -llablgl -ltcl8.5 -ltk8.5 -L/usr/lib/ocaml -lcamlrun_shared\
 	  -cclib "$(GLLIBS)"
 lablgl.cmxa: liblablgl$(XA) $(OPTOBJS) $(CONFIG)
-	$(OPTLINK) -a -o $@ $(OPTOBJS) -cclib -llablgl \
+	$(OPTLINK) -a -o $@ $(OPTOBJS) -cclib -llablgl -ltcl8.5 -ltk8.5 -L/usr/lib/ocaml -lcamlrun_shared\
 	  -cclib "$(GLLIBS)"
 else
 liblablgl$(XA) lablgl.cma: $(COBJS) $(LIBOBJS) $(CONFIG)
-	$(LIBRARIAN) -o lablgl $(COBJS) $(LIBOBJS) $(GLLIBS) $(XLIBS)
+	$(LIBRARIAN) -o lablgl $(COBJS) $(LIBOBJS) $(GLLIBS) $(XLIBS) -L/usr/lib/ocaml -lcamlrun_shared
 lablgl.cmxa: $(COBJS) $(OPTOBJS) $(CONFIG)
-	$(LIBRARIAN) -o lablgl $(COBJS) $(OPTOBJS) $(GLLIBS) $(XLIBS)
+	$(LIBRARIAN) -o lablgl $(COBJS) $(OPTOBJS) $(GLLIBS) $(XLIBS) -L/usr/lib/ocaml -lcamlrun_shared
 endif
 
 gl_tags.c: gl_tags.var
@@ -93,3 +93,4 @@
 ml_glu$(XO) ml_glutess$(XO) : ml_gl.h ml_glu.h glu_tags.h glu_tags.c
 ml_raw$(XO): raw_tags.h ml_raw.h
 include .depend
+
--- Togl/src/Makefile.orig	2009-12-28 15:24:10.224029329 +0200
+++ Togl/src/Makefile	2009-12-29 09:49:42.198666248 +0200
@@ -5,7 +5,7 @@
 # Composite options
 INCLUDES = -I$(SRCDIR) -I$(TOGLDIR) \
 	$(TKINCLUDES) -I. $(GLINCLUDES) $(XINCLUDES)
-LIBS = $(TKLIBS) $(GLLIBS) $(XLIBS)
+LIBS = $(TKLIBS) $(GLLIBS) $(XLIBS) -ltcl8.5 -ltk8.5 -L/usr/lib/ocaml -lcamlrun_shared
 LIBDIRS =
 
 OCAMLINC=-I +labltk -I $(SRCDIR)
@@ -15,7 +15,7 @@
 
 # Extra rules
 .cmx.opt: $(TOPDIR)/src/lablgl.cmxa togl.cmxa
-	$(OPTLINK) -o $@ $(OCAMLINC) -ccopt -L. \
+	$(OPTLINK) -o $@ $(OCAMLINC) -ccopt -L. -ltcl8.5 -ltk8.5 -L/usr/lib/ocaml -lcamlrun_shared\
 	    unix.cmxa labltk.cmxa ../../lablgl.cmxa togl.cmxa $<
 
 all: lablgltop$(XE) lablgl$(XB)
@@ -23,9 +23,9 @@
 opt: togl.cmxa
 
 libtogl.a togl.cma: togl.cmo $(TOGLOBJS) $(CONFIG)
-	$(LIBRARIAN) -o togl togl.cmo $(TOGLOBJS) $(GLLIBS) $(TKLIBS) $(XLIBS)
+	$(LIBRARIAN) -o togl togl.cmo $(TOGLOBJS) $(GLLIBS) $(TKLIBS) $(XLIBS) -ltcl8.5 -ltk8.5 -L/usr/lib/ocaml -lcamlrun_shared
 togl.cmxa: togl.cmx $(TOGLOBJS) $(CONFIG)
-	$(LIBRARIAN) -o togl togl.cmx $(TOGLOBJS) $(GLLIBS) $(TKLIBS) $(XLIBS)
+	$(LIBRARIAN) -o togl togl.cmx $(TOGLOBJS) $(GLLIBS) $(TKLIBS) $(XLIBS) -ltcl8.5 -ltk8.5 -L/usr/lib/ocaml -lcamlrun_shared
 
 $(TOGLDIR)/togl.o:
 	cd $(TOGLDIR) && $(MAKE) togl.o CC="$(CAMLC) -verbose" CFLAGS='-c -ccopt "$(COPTS) $(INCLUDES) $(TK_FLAGS)"'
--- LablGlut/src/Makefile.orig	2009-12-29 09:52:09.685947278 +0200
+++ LablGlut/src/Makefile	2009-12-29 09:57:00.240666050 +0200
@@ -29,16 +29,16 @@
 	$(MKDLL)$@ $(COBJS:$(XO)=.d$(XO)) $(GLUTLIBS) $(GLLIBS) $(OCAMLDLL)
 lablglut.cma: liblablglut$(XA) dlllablglut.dll $(LIBOBJS) ../../Makefile.config
 	$(LINKER) -a -o $@ $(LIBOBJS) \
-	  -cclib -llablglut -dllib -llablglut \
+	  -cclib -llablglut -dllib -llablglut -L/usr/lib/ocaml -lcamlrun_shared\
 	  -cclib "$(GLLIBS)" -cclib "$(GLUTLIBS)"
 lablglut.cmxa: liblablglut$(XA) $(OPTOBJS) ../../Makefile.config
-	$(OPTLINK) -a -o $@ $(OPTOBJS) -cclib -llablglut \
+	$(OPTLINK) -a -o $@ $(OPTOBJS) -cclib -llablglut -L/usr/lib/ocaml -lcamlrun_shared\
 	  -cclib "$(GLLIBS)" -cclib "$(GLUTLIBS)"
 else
 liblablglut$(XA) lablglut.cma: $(COBJS) $(LIBOBJS) ../../Makefile.config
-	$(LIBRARIAN) -o lablglut $(COBJS) $(LIBOBJS) $(GLUTLIBS) $(GLLIBS) $(XLIBS)
+	$(LIBRARIAN) -o lablglut $(COBJS) $(LIBOBJS) $(GLUTLIBS) $(GLLIBS) $(XLIBS) -L/usr/lib/ocaml -lcamlrun_shared
 lablglut.cmxa: $(COBJS) $(OPTOBJS) ../../Makefile.config
-	$(LIBRARIAN) -o lablglut $(COBJS) $(OPTOBJS) $(GLUTLIBS) $(GLLIBS) $(XLIBS)
+	$(LIBRARIAN) -o lablglut $(COBJS) $(OPTOBJS) $(GLUTLIBS) $(GLLIBS) $(XLIBS) -L/usr/lib/ocaml -lcamlrun_shared
 endif
 
 lablgluttop$(XE): lablglut.cma 
