--- src/qemu/qemu_capabilities.c~	2014-01-10 20:27:38.383264024 +0100
+++ src/qemu/qemu_capabilities.c	2014-01-10 20:28:02.696597278 +0100
@@ -638,14 +638,14 @@ virQEMUCapsFindBinaryForArch(virArch hos
     if (guestarch == VIR_ARCH_I686 &&
         !ret &&
         hostarch == VIR_ARCH_X86_64) {
-        ret = virFindFileInPath("qemu-system-x86_64");
+        ret = virFindFileInPath("qemu-kvm");
         if (ret && !virFileIsExecutable(ret))
             VIR_FREE(ret);
     }
 
     if (guestarch == VIR_ARCH_I686 &&
         !ret) {
-        ret = virFindFileInPath("qemu");
+        ret = virFindFileInPath("qemu-kvm");
         if (ret && !virFileIsExecutable(ret))
             VIR_FREE(ret);
     }
