Index: qtoctave-0.9.1/CMakeLists.txt
===================================================================
--- qtoctave-0.9.1.orig/CMakeLists.txt
+++ qtoctave-0.9.1/CMakeLists.txt
@@ -28,7 +28,7 @@ SET(QT_USE_QTXML 1)
 
 include(UseQt4)
 
-if( NOT ${QT_VERSION_MAJOR} EQUAL 4 OR NOT ${QT_VERSION_MINOR} EQUAL 6   )
+if( NOT ${QT_VERSION_MAJOR} EQUAL 4 OR ${QT_VERSION_MINOR} LESS 5   )
 	MESSAGE (FATAL_ERROR 
 			"Qt version 4.6 required.\n"
 			"Qt version installed is: ${QT_VERSION_MAJOR}.${QT_VERSION_MINOR}\n"
