--- qtoctave-0.9.1/qtoctave/src/CMakeLists.txt	2010-01-25 15:19:48.000000000 +0200
+++ qtoctave-0.9.1/qtoctave/src/CMakeLists.txt	2010-04-21 00:03:03.000000000 +0300
@@ -52,7 +52,7 @@
 
 install (
 	DIRECTORY qtoctave_doc
-	DESTINATION share/doc/qtoctave
+	DESTINATION share/qtoctave
 	PATTERN ".svn" EXCLUDE
 	)
 
--- qtoctave-0.9.1/qtoctave/src/configure.h.in	2010-01-25 15:19:50.000000000 +0200
+++ qtoctave-0.9.1/qtoctave/src/configure.h.in	2010-04-21 00:03:12.000000000 +0300
@@ -5,7 +5,7 @@
 #define HELP_PATH "${CMAKE_INSTALL_PREFIX}/share/doc/octave-html/octave_doc/index.html"
 #define ICON_PATH "${CMAKE_INSTALL_PREFIX}/share/qtoctave/images"
 #define LANG_PATH "${CMAKE_INSTALL_PREFIX}/share/qtoctave/lang"
-#define QTOCTAVE_HELP_PATH "${CMAKE_INSTALL_PREFIX}/share/doc/qtoctave/qtoctave_doc/index.html"
+#define QTOCTAVE_HELP_PATH "${CMAKE_INSTALL_PREFIX}/share/qtoctave/qtoctave_doc/index.html"
 
 #define PKG_CMD_PATH "${CMAKE_INSTALL_PREFIX}/share/qtoctave/pkg-commands.list"
 #endif
