--- virt-wrapper.py~	2013-02-01 21:29:54.545523923 +0100
+++ virt-wrapper.py	2013-02-01 21:30:39.649522275 +0100
@@ -130,7 +130,7 @@
         if mod.replace("-", "_") in loaded:
             link = comar.Link()
             try:
-                link.Boot.Modules["module_init_tools"].unload(mod)
+                link.Boot.Modules["kmod"].unload(mod)
             except dbus.exceptions.DBusException:
                 return False
     else:
@@ -168,7 +168,7 @@
     failure = False
     for mod in mods:
         try:
-            link.Boot.Modules["module_init_tools"].load(mod, "")
+            link.Boot.Modules["kmod"].load(mod, "")
         except dbus.exceptions.DBusException:
             failure = True
 
