diff -Nuar eektool.orig//module.desktop.in eektool/module.desktop.in
--- eektool.orig//module.desktop.in	2011-02-16 22:39:52.000000000 +0200
+++ eektool/module.desktop.in	2011-03-31 15:24:47.779013001 +0300
@@ -17,3 +17,4 @@
 Comment[el]=Εκτελέστε μια εντολή κελύφους και να εμφανίσει την έξοδο.
 Comment[eo]=Plenumu ŝelo komando kaj montri la eligo.
 Comment[pt]=Executa un comando shell e mostra a saída
+Comment[tr]=Bir kabuk komutu çalıştırır ve çıktısını gösterir.
