--- src/gtk-recordmydesktop.desktop	2007-04-27 06:22:52.000000000 +0300
+++ src/gtk-recordmydesktop.desktop	2007-10-14 20:57:09.000000000 +0300
@@ -2,8 +2,11 @@
 Version=1.0
 Encoding=UTF-8
 Name=gtk-recordMyDesktop
+GenericName=Frontend for recordMyDesktop
+GenericName[tr]=recordMyDesktop arayüzü
 Comment=Frontend for recordMyDesktop
 Comment[fr]=Interface pour recordmydesktop
+Comment[tr]=recordMyDesktop için arayüz
 Exec=gtk-recordMyDesktop
 Terminal=false
 Type=Application
