Index: flac-1.2.1/src/libFLAC/crc.c
===================================================================
--- flac-1.2.1.orig/src/libFLAC/crc.c
+++ flac-1.2.1/src/libFLAC/crc.c
@@ -74,7 +74,7 @@ FLAC__byte const FLAC__crc8_table[256] =
 
 /* CRC-16, poly = x^16 + x^15 + x^2 + x^0, init = 0 */
 
-unsigned FLAC__crc16_table[256] = {
+const unsigned FLAC__crc16_table[256] = {
 	0x0000,  0x8005,  0x800f,  0x000a,  0x801b,  0x001e,  0x0014,  0x8011,
 	0x8033,  0x0036,  0x003c,  0x8039,  0x0028,  0x802d,  0x8027,  0x0022,
 	0x8063,  0x0066,  0x006c,  0x8069,  0x0078,  0x807d,  0x8077,  0x0072,
Index: flac-1.2.1/src/libFLAC/format.c
===================================================================
--- flac-1.2.1.orig/src/libFLAC/format.c
+++ flac-1.2.1/src/libFLAC/format.c
@@ -57,13 +57,13 @@
 #endif
 
 /* VERSION should come from configure */
-FLAC_API const char *FLAC__VERSION_STRING = VERSION;
+FLAC_API const char *const FLAC__VERSION_STRING = VERSION;
 
 #if defined _MSC_VER || defined __BORLANDC__ || defined __MINW32__
 /* yet one more hack because of MSVC6: */
-FLAC_API const char *FLAC__VENDOR_STRING = "reference libFLAC 1.2.1 20070917";
+FLAC_API const char *const FLAC__VENDOR_STRING = "reference libFLAC 1.2.1 20070917";
 #else
-FLAC_API const char *FLAC__VENDOR_STRING = "reference libFLAC " VERSION " 20070917";
+FLAC_API const char *const FLAC__VENDOR_STRING = "reference libFLAC " VERSION " 20070917";
 #endif
 
 FLAC_API const FLAC__byte FLAC__STREAM_SYNC_STRING[4] = { 'f','L','a','C' };
Index: flac-1.2.1/src/libFLAC/include/private/crc.h
===================================================================
--- flac-1.2.1.orig/src/libFLAC/include/private/crc.h
+++ flac-1.2.1/src/libFLAC/include/private/crc.h
@@ -48,7 +48,7 @@ FLAC__uint8 FLAC__crc8(const FLAC__byte 
 ** polynomial = x^16 + x^15 + x^2 + x^0
 ** init = 0
 */
-extern unsigned FLAC__crc16_table[256];
+extern const unsigned FLAC__crc16_table[256];
 
 #define FLAC__CRC16_UPDATE(data, crc) (((((crc)<<8) & 0xffff) ^ FLAC__crc16_table[((crc)>>8) ^ (data)]))
 /* this alternate may be faster on some systems/compilers */
Index: flac-1.2.1/src/libFLAC/stream_decoder.c
===================================================================
--- flac-1.2.1.orig/src/libFLAC/stream_decoder.c
+++ flac-1.2.1/src/libFLAC/stream_decoder.c
@@ -79,7 +79,7 @@
 
 
 /* technically this should be in an "export.c" but this is convenient enough */
-FLAC_API int FLAC_API_SUPPORTS_OGG_FLAC =
+FLAC_API const int FLAC_API_SUPPORTS_OGG_FLAC =
 #if FLAC__HAS_OGG
 	1
 #else
Index: flac-1.2.1/include/FLAC/format.h
===================================================================
--- flac-1.2.1.orig/include/FLAC/format.h
+++ flac-1.2.1/include/FLAC/format.h
@@ -158,13 +158,13 @@ extern "C" {
  * This does not correspond to the shared library version number, which
  * is used to determine binary compatibility.
  */
-extern FLAC_API const char *FLAC__VERSION_STRING;
+extern FLAC_API const char *const FLAC__VERSION_STRING;
 
 /** The vendor string inserted by the encoder into the VORBIS_COMMENT block.
  *  This is a NUL-terminated ASCII string; when inserted into the
  *  VORBIS_COMMENT the trailing null is stripped.
  */
-extern FLAC_API const char *FLAC__VENDOR_STRING;
+extern FLAC_API const char *const FLAC__VENDOR_STRING;
 
 /** The byte string representation of the beginning of a FLAC stream. */
 extern FLAC_API const FLAC__byte FLAC__STREAM_SYNC_STRING[4]; /* = "fLaC" */
Index: flac-1.2.1/include/FLAC/export.h
===================================================================
--- flac-1.2.1.orig/include/FLAC/export.h
+++ flac-1.2.1/include/FLAC/export.h
@@ -80,7 +80,7 @@ extern "C" {
 #endif
 
 /** \c 1 if the library has been compiled with support for Ogg FLAC, else \c 0. */
-extern FLAC_API int FLAC_API_SUPPORTS_OGG_FLAC;
+extern FLAC_API const int FLAC_API_SUPPORTS_OGG_FLAC;
 
 #ifdef __cplusplus
 }
