--- src/amd64/acpus.s
+++ src/amd64/acpus.s
@@ -128,3 +128,6 @@
    popq %rbp
    ret
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/iblit16.s
+++ src/i386/iblit16.s
@@ -743,3 +743,6 @@
 
 
 #endif      /* ifdef ALLEGRO_COLOR16 */
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/iblit24.s
+++ src/i386/iblit24.s
@@ -372,3 +372,6 @@
 
 
 #endif      /* ifdef ALLEGRO_COLOR24 */
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/iblit32.s
+++ src/i386/iblit32.s
@@ -357,3 +357,6 @@
 
 
 #endif      /* ifdef ALLEGRO_COLOR32 */
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/iblit8.s
+++ src/i386/iblit8.s
@@ -787,3 +787,6 @@
 
 
 #endif      /* ifdef ALLEGRO_COLOR8 */
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/icpus.s
+++ src/i386/icpus.s
@@ -213,3 +213,6 @@
    popl %ebp
    ret
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/igfx15.s
+++ src/i386/igfx15.s
@@ -436,3 +436,6 @@
 
 #endif      /* ifdef ALLEGRO_COLOR16 */
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/igfx16.s
+++ src/i386/igfx16.s
@@ -487,3 +487,6 @@
 
 #endif      /* ifdef ALLEGRO_COLOR16 */
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/igfx24.s
+++ src/i386/igfx24.s
@@ -581,3 +581,6 @@
 
 #endif      /* ifdef ALLEGRO_COLOR24 */
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/igfx32.s
+++ src/i386/igfx32.s
@@ -467,3 +467,6 @@
 
 #endif      /* ifdef ALLEGRO_COLOR32 */
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/igfx8.s
+++ src/i386/igfx8.s
@@ -530,3 +530,6 @@
 
 #endif      /* ifdef ALLEGRO_COLOR8 */
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/imisc.s
+++ src/i386/imisc.s
@@ -600,3 +600,6 @@
    popl %ebp
    ret                           /* end of fixhypot() */
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/iscanmmx.s
+++ src/i386/iscanmmx.s
@@ -1672,3 +1672,6 @@
 
 
 #endif /* MMX */
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/iscan.s
+++ src/i386/iscan.s
@@ -2777,3 +2777,6 @@
    ret                           /* end of _poly_scanline_ptex_mask_trans24() */
 
 #endif /* COLOR24 */
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/ispr15.s
+++ src/i386/ispr15.s
@@ -494,3 +494,6 @@
 
 #endif      /* ifdef ALLEGRO_COLOR16 */
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/ispr16.s
+++ src/i386/ispr16.s
@@ -889,3 +889,6 @@
 
 #endif      /* ifdef ALLEGRO_COLOR16 */
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/ispr24.s
+++ src/i386/ispr24.s
@@ -913,3 +913,6 @@
 
 #endif      /* ifdef ALLEGRO_COLOR24 */
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/ispr32.s
+++ src/i386/ispr32.s
@@ -700,3 +700,6 @@
 
 #endif      /* ifdef ALLEGRO_COLOR32 */
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/ispr8.s
+++ src/i386/ispr8.s
@@ -1046,3 +1046,6 @@
 
 
 #endif      /* ifdef ALLEGRO_COLOR8 */
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/i386/izbuf.s
+++ src/i386/izbuf.s
@@ -2328,3 +2328,6 @@
    ret                           /* end of _poly_zbuf_ptex_mask_trans24() */
 
 #endif /* COLOR24 */
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/linux/svgalibs.s
+++ src/linux/svgalibs.s
@@ -65,3 +65,6 @@
 
 #endif
 #endif
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/misc/icolconv.s
+++ src/misc/icolconv.s
@@ -3288,3 +3288,6 @@
 
 #endif  /* ALLEGRO_NO_COLORCOPY */
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/misc/modexgfx.s
+++ src/misc/modexgfx.s
@@ -1688,3 +1688,6 @@
 
 
 #endif        /* (!defined ALLEGRO_LINUX) || ((defined ALLEGRO_LINUX_VGA) && ... */
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/misc/vbeafs.s
+++ src/misc/vbeafs.s
@@ -188,3 +188,6 @@
 
 
 #endif      /* ifdef VBE/AF is cool on this platform */
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/x/xdga2s.s
+++ src/x/xdga2s.s
@@ -51,3 +51,6 @@
 #endif
 #endif
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
--- src/x/xwins.s
+++ src/x/xwins.s
@@ -51,3 +51,6 @@
 
 #endif
 
+#ifdef __ELF__
+.section .note.GNU-stack,"",%progbits
+#endif
