--- CMakeLists.txt
+++ CMakeLists.txt
@@ -376,6 +376,10 @@
 # Platform drivers
 #
 
+option(WANT_OSS "Build OSS support" on)
+option(WANT_ALSA "Build ALSA support" on)
+option(WANT_JACK "Build JACK support" on)
+
 set(PLATFORM_SOURCES)
 set(PLATFORM_LIBS)              # always link with these libraries
 set(PLATFORM_LIBS_NON_MODULES)  # only link with these if modules disabled
@@ -391,13 +395,16 @@
     list(APPEND PLATFORM_LIBS m ${CMAKE_THREAD_LIBS_INIT})
     list(APPEND PLATFORM_SOURCES ${ALLEGRO_SRC_UNIX_FILES})
 
+    if(WANT_OSS)
     include(AllegroFindOSS)
     if(OSS_FOUND)
         set(ALLEGRO_WITH_OSSDIGI 1)
         set(ALLEGRO_WITH_OSSMIDI 1)
         include_directories(SYSTEM ${OSS_INCLUDE_DIR})
     endif(OSS_FOUND)
+    endif(WANT_OSS)
 
+    if(WANT_ALSA)
     pkg_check_modules(ALSA alsa)
     if(ALSA_FOUND)
         # ALSA 0.5 is beyond obsolete.
@@ -409,7 +416,9 @@
         add_our_module(alleg-alsadigi src/unix/alsa9.c ${ALSA_LIBRARIES})
         add_our_module(alleg-alsamidi src/unix/alsamidi.c ${ALSA_LIBRARIES})
     endif(ALSA_FOUND)
+    endif(WANT_ALSA)
 
+    if(WANT_JACK)
     pkg_check_modules(JACK jack)
     if(JACK_FOUND)
         set(ALLEGRO_WITH_JACKDIGI 1)
@@ -417,6 +426,7 @@
         list(APPEND PLATFORM_LIBS_NON_MODULES ${JACK_LIBRARIES})
         add_our_module(alleg-jack src/unix/jack.c ${JACK_LIBRARIES})
     endif(JACK_FOUND)
+    endif(WANT_JACK)
 
     # This is untested.
     check_library_exists(audio alOpenPort "" SGIAUDIO_FOUND)
