--- mpgtx-1.3.1/configure.old	2010-01-23 14:41:46.081368114 +0200
+++ mpgtx-1.3.1/configure	2010-01-23 14:48:08.962350635 +0200
@@ -6,20 +6,19 @@
 #                                               #
 #################################################
 
-prefix="/usr/local"
+prefix="/usr"
 manprefix='$(PREFIX)'
 #compiler flags
-CC="g++"
 common_cflags="-Wall"
 devel_cflags="-g -Werror -fno-builtin"
-optimization_cflags="-O3"
+optimization_cflags="${CXXFLAGS}"
 release_cflags=$optimization_cflags
 OFLAGS="-ansi"
 cflags=$common_cflags
 
 #linker flags
 
-lflags=""
+lflags="-lm"
 
 #state vars
 debug="no"
@@ -36,7 +35,6 @@
 
 if [ "${HOSTNAME}" = "powerpc-apple-darwin6.0" ]; then
 #compiler flags
-CC="c++"
 common_cflags="-pedantic -Wall"
 #devel_cflags="-g -Werror -fno-builtin"
 devel_cflags="-g -fno-builtin"
@@ -165,7 +163,7 @@
 		return (int)(fseeko(stdin,0,SEEK_SET));
 		}
 EOF
-	$CC __LFStest.cpp -o __LFStestPASSED -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE 1>__LFSout 2>__LFSerror
+	${CXX} __LFStest.cpp -o __LFStestPASSED -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE 1>__LFSout 2>__LFSerror
 	if test -f __LFStestPASSED; then
 		echo "PASSED";
 		cflags="$cflags -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE";
@@ -293,15 +291,15 @@
 	@echo ""
 
 mpgtx : commandline.cxx mpegOut.o mpeg.o chunkTab.o id3command.o common.hh
-	$CC \$(CFLAGS) \$(OFLAGS) -o mpgtx commandline.cxx mpegOut.o mpeg.o chunkTab.o id3command.o \$(LFLAGS)
+	$CXX \$(CFLAGS) \$(OFLAGS) \$(LDFLAGS) -o mpgtx commandline.cxx mpegOut.o mpeg.o chunkTab.o id3command.o \$(LFLAGS)
 mpegOut.o : mpegOut.cxx mpegOut.hh mpeg.hh common.hh
-	$CC \$(CFLAGS) \$(OFLAGS) -c mpegOut.cxx 
+	$CXX \$(CFLAGS) \$(OFLAGS) -c mpegOut.cxx 
 mpeg.o : mpeg.cxx mpeg.hh mpegOut.hh common.hh
-	$CC \$(CFLAGS) \$(OFLAGS) -c mpeg.cxx 
+	$CXX \$(CFLAGS) \$(OFLAGS) -c mpeg.cxx 
 chunkTab.o : chunkTab.cxx chunkTab.hh common.hh
-	$CC \$(CFLAGS) \$(OFLAGS) -c chunkTab.cxx
+	$CXX \$(CFLAGS) \$(OFLAGS) -c chunkTab.cxx
 id3command.o : id3command.cxx id3command.hh common.hh
-	$CC \$(CFLAGS) \$(OFLAGS) -c id3command.cxx
+	$CXX \$(CFLAGS) \$(OFLAGS) -c id3command.cxx
 
 
 
