diff -Nur clutter-gtkmm-0.9.4-old/Makefile.am clutter-gtkmm-0.9.4/Makefile.am
--- clutter-gtkmm-0.9.4-old/Makefile.am	2009-10-04 03:03:03.820157549 +0300
+++ clutter-gtkmm-0.9.4/Makefile.am	2009-10-04 03:03:36.658135434 +0300
@@ -1,6 +1,8 @@
 SUBDIRS		= tools clutter-gtk docs examples
 DIST_SUBDIRS	= $(SUBDIRS) scripts
 
+ACLOCAL_AMFLAGS = -Iscripts
+
 EXTRA_DIST	=  \
 		  build_shared/Makefile_build.am_fragment \
 		  build_shared/Makefile_build_gensrc.am_fragment \
