diff -Nur clutter-gtkmm-0.9.4-old/configure.in clutter-gtkmm-0.9.4/configure.in
--- clutter-gtkmm-0.9.4-old/configure.in	2009-11-07 00:58:25.956497084 +0200
+++ clutter-gtkmm-0.9.4/configure.in	2009-11-07 00:59:02.097256023 +0200
@@ -4,8 +4,8 @@
 #  Version and initialization
 #########################################################################
 CLUTTER_GTKMM_MAJOR_VERSION=0
-CLUTTER_GTKMM_MINOR_VERSION=9
-CLUTTER_GTKMM_MICRO_VERSION=4
+CLUTTER_GTKMM_MINOR_VERSION=90
+CLUTTER_GTKMM_MICRO_VERSION=1
 
 #
 # +1 : ? : +1  == new interface that does not break old one
@@ -74,7 +74,7 @@
 PKG_CHECK_MODULES(CLUTTER_GTKMM,
                   [gtkmm-2.4
                    cluttermm-0.9
-                   clutter-gtk-0.10 >= 0.10.2])
+                   clutter-gtk-0.90 >= 0.10.2])
 AC_SUBST(CLUTTER_GTKMM_CFLAGS)
 AC_SUBST(CLUTTER_GTKMM_LIBS)
 
