diff -Nur clutter-gtkmm-0.9.4-old/build_shared/Makefile_build_gensrc.am_fragment clutter-gtkmm-0.9.4/build_shared/Makefile_build_gensrc.am_fragment
--- clutter-gtkmm-0.9.4-old/build_shared/Makefile_build_gensrc.am_fragment	2009-10-04 03:05:38.888135918 +0300
+++ clutter-gtkmm-0.9.4/build_shared/Makefile_build_gensrc.am_fragment	2009-10-04 03:06:39.885887860 +0300
@@ -21,6 +21,6 @@
 
 files_built_cc		= $(files_hg:.hg=.cc) wrap_init.cc
 files_built_h		= $(files_hg:.hg=.h)
-files_extra_h		+= wrap_init.h
+#files_extra_h		+= wrap_init.h
 
 include $(top_srcdir)/build_shared/Makefile_build.am_fragment
