diff -Nur freealut-1.1.0-old/src/Makefile.am freealut-1.1.0/src/Makefile.am
--- freealut-1.1.0-old/src/Makefile.am	2006-11-17 14:08:18.000000000 +0200
+++ freealut-1.1.0/src/Makefile.am	2006-11-17 14:08:21.000000000 +0200
@@ -20,7 +20,7 @@
 	alutWaveform.c
 
 # We use sin and floor, so we might need -lm, autoconf takes care of this.
-libalut_la_LIBADD = $(LIBM)
+libalut_la_LIBADD = $(LIBM) -lopenal -lpthread
 
 # The following libtool flag is neccessary for building DLLs on Windows.
 libalut_la_LDFLAGS = -no-undefined
