diff -up yp-tools-2.11/src/yppasswd.c.shadow yp-tools-2.11/src/yppasswd.c
--- yp-tools-2.11/src/yppasswd.c.shadow	2010-04-20 15:31:59.000000000 +0200
+++ yp-tools-2.11/src/yppasswd.c	2010-11-19 18:37:14.931766315 +0100
@@ -449,6 +449,7 @@ verifypassword (struct passwd *pwd, char
 
   passwdlen = get_passwd_len (pwd->pw_passwd);
   if (pwd->pw_passwd[0]
+      && 0 != strcmp (pwd->pw_passwd, "x") /* don't check shadow passwords */
       && !strncmp (pwd->pw_passwd, crypt (pwdstr, pwd->pw_passwd), passwdlen)
       && uid)
     {
