diff -ur gnome-python-extras-2.25.3.orig//gtkmozembed/Makefile.am gnome-python-extras-2.25.3/gtkmozembed/Makefile.am
--- gnome-python-extras-2.25.3.orig//gtkmozembed/Makefile.am	2008-12-12 03:06:29.000000000 +0200
+++ gnome-python-extras-2.25.3/gtkmozembed/Makefile.am	2011-01-12 19:53:17.426000002 +0200
@@ -12,8 +12,7 @@
 # gtkmozembed module
 pkgpyexec_LTLIBRARIES = gtkmozembed.la
 gtkmozembed_la_LDFLAGS = -module -avoid-version	\
-	-export-symbols-regex initgtkmozembed	\
-	-R$(MOZILLA_HOME) -L$(MOZILLA_HOME)
+	-export-symbols-regex initgtkmozembed
 gtkmozembed_la_LIBADD = $(GTKMOZEMBED_LIBS)
 gtkmozembed_la_SOURCES = gtkmozembedmodule.c
 nodist_gtkmozembed_la_SOURCES = gtkmozembed.c
Yalnızca gnome-python-extras-2.25.3/gtkmozembed'da: Makefile.am~
