Index: qdevelop-0.25/src/toolscontrolimpl.cpp
===================================================================
--- qdevelop-0.25.orig/src/toolscontrolimpl.cpp
+++ qdevelop-0.25/src/toolscontrolimpl.cpp
@@ -61,7 +61,7 @@ ToolsControlImpl::ToolsControlImpl( QWid
 #else
     make->setText( settings.value("m_makeName", "/usr/bin/make").toString() );
     gdb->setText( settings.value("m_gdbName", "/usr/bin/gdb").toString() );
-    ctags->setText( settings.value("m_ctagsName", "/usr/bin/ctags").toString() );
+    ctags->setText( settings.value("m_ctagsName", "/usr/bin/exuberant-ctags").toString() );
 #endif
     checkEnvironmentOnStartup->setChecked( settings.value("m_checkEnvironmentOnStartup", true).toBool() );
     settings.endGroup();
