Index: qdevelop-v0.28/CMakeLists.txt
===================================================================
--- qdevelop-v0.28.orig/CMakeLists.txt
+++ qdevelop-v0.28/CMakeLists.txt
@@ -280,7 +280,7 @@ IF (AUTOPLUGINS)
 				INSTALL (FILES ${PLG_QM} DESTINATION translations/${PLG_NAME})
 			ELSE (NOT WIN32)
 				INSTALL (TARGETS ${PLG_NAME} DESTINATION lib/qdevelop/plugins)
-				INSTALL (FILES ${PLG_QM} DESTINATION lib/qdevelop/translations/${PLG_NAME})
+				INSTALL (FILES ${PLG_QM} DESTINATION share/qdevelop/translations/${PLG_NAME})
 			ENDIF (WIN32)
 		ENDIF (IS_EMPTY)
 	ENDFOREACH (PLG)
@@ -290,5 +290,5 @@ INSTALL (TARGETS ${QDEV} RUNTIME DESTINA
 IF (WIN32)
 	INSTALL (FILES ${qdevelop_QM} DESTINATION translations)
 ELSE (NOT WIN32)
-	INSTALL (FILES ${qdevelop_QM} DESTINATION lib/qdevelop/translations)
+	INSTALL (FILES ${qdevelop_QM} DESTINATION share/qdevelop/translations)
 ENDIF (WIN32)
Index: qdevelop-v0.28/plugins/formatting-astyle/astyleplugin.cpp
===================================================================
--- qdevelop-v0.28.orig/plugins/formatting-astyle/astyleplugin.cpp
+++ qdevelop-v0.28/plugins/formatting-astyle/astyleplugin.cpp
@@ -37,7 +37,7 @@ QString AStylePlugin::menuName() const
 		#ifdef Q_OS_WIN32
 		QString defaultTranslationsPath = "/../translations/formatting-astyle";
 		#else
-		QString defaultTranslationsPath = "/../lib/qdevelop/translations/formatting-astyle";
+		QString defaultTranslationsPath = "/../share/qdevelop/translations/formatting-astyle";
 		#endif
 		QDir translationsDir(QCoreApplication::applicationDirPath() + defaultTranslationsPath);
 		
Index: qdevelop-v0.28/plugins/tools-regexp-planner/replugin.cpp
===================================================================
--- qdevelop-v0.28.orig/plugins/tools-regexp-planner/replugin.cpp
+++ qdevelop-v0.28/plugins/tools-regexp-planner/replugin.cpp
@@ -37,7 +37,7 @@ QString RePlugin::menuName() const
 		#ifdef Q_OS_WIN32
 		QString defaultTranslationsPath = "/../translations/tools-regexp-planner";
 		#else
-		QString defaultTranslationsPath = "/../lib/qdevelop/translations/tools-regexp-planner";
+		QString defaultTranslationsPath = "/../share/qdevelop/translations/tools-regexp-planner";
 		#endif
 		QDir translationsDir(QCoreApplication::applicationDirPath() + defaultTranslationsPath);
 		
Index: qdevelop-v0.28/src/main.cpp
===================================================================
--- qdevelop-v0.28.orig/src/main.cpp
+++ qdevelop-v0.28/src/main.cpp
@@ -79,7 +79,7 @@ int main(int argc, char *argv[])
 	#ifdef Q_OS_WIN32
 	QString defaultTranslationsPath = "/translations";
 	#else
-	QString defaultTranslationsPath = "/../lib/qdevelop/translations";
+	QString defaultTranslationsPath = "/../share/qdevelop/translations";
 	#endif
 	QDir translationsDir(QCoreApplication::applicationDirPath() + defaultTranslationsPath);
 	// load & install QDevelop translation
