Index: qdevelop-v0.28/qdevelop.desktop
===================================================================
--- qdevelop-v0.28.orig/qdevelop.desktop
+++ qdevelop-v0.28/qdevelop.desktop
@@ -5,6 +5,7 @@ Comment=A development environment dedica
 Comment[ru]=Среда разработки для Qt4
 Comment[fr]=Un environnement de développement intégré pour Qt4. 
 Comment[de]=Eine Entwicklungsumgebung für Qt4
+Comment[tr]=Qt4 için Geliştirme Ortamı
 Exec=qdevelop
 Icon=qdevelop
 Terminal=false
