diff -ur jamin-0.95.0.orig/data/jamin.desktop.in jamin-0.95.0/data/jamin.desktop.in
--- jamin-0.95.0.orig/data/jamin.desktop.in	2004-12-29 06:25:32.000000000 +0200
+++ jamin-0.95.0/data/jamin.desktop.in	2007-12-30 17:37:57.000000000 +0200
@@ -1,11 +1,13 @@
 [Desktop Entry]
 Encoding=UTF-8
-_Name=JAMin
-_GenericName=
-_Comment=JACK Audio Mastering interface
+Name=JAMin
+Name[ru]=JAMin
+GenericName=
+Comment=JACK Audio Mastering interface
+Comment[ru]=JAMin -- приложение для мастеринга звука
 Categories=Application;AudioVideo;
 Exec=jamin
-Icon=jamin.svg
+Icon=/usr/share/icons/jamin.svg
 MimeType=application/x-jamin;
 StartupNotify=true
 Terminal=false
