commit f9e2d539599345b03dde495a36d4ec8bd90d78a5
Author: Alan T. DeKok <aland@freeradius.org>
Date:   Thu May 15 10:31:23 2014 -0400

    Use the correct data type.  Closes #634

diff --git a/src/main/mainconfig.c b/src/main/mainconfig.c
index 86757df..4564778 100644
--- a/src/main/mainconfig.c
+++ b/src/main/mainconfig.c
@@ -83,7 +83,7 @@ static char const *radlog_dest = NULL;
  */
 static char const	*localstatedir = NULL;
 static char const	*prefix = NULL;
-static char		my_name;
+static char const	*my_name = NULL;
 static char const	*sbindir = NULL;
 static char const	*run_dir = NULL;
 static char		*syslog_facility = NULL;
