--- ircp/dirtraverse.c.orig
+++ ircp/dirtraverse.c
@@ -47,6 +47,7 @@ int visit_dir(char *path, visit_cb cb, v
 		else {
 			snprintf(t, MAXPATHLEN, "%s/%s", path, dirent->d_name);
 			if(lstat(t, &statbuf) < 0) {
+			    	closedir(dir);
 				return -1;
 			}
 			else if(S_ISREG(statbuf.st_mode)) {
@@ -76,6 +77,7 @@ int visit_dir(char *path, visit_cb cb, v
 	}
 
 out:
+	closedir(dir);
 	return ret;
 
 #else