--- openobex-1.4/acinclude.m4.old	2008-11-07 13:36:09.000000000 +0100
+++ openobex-1.4/acinclude.m4	2008-11-18 14:39:41.000000000 +0100
@@ -158,7 +158,7 @@
 		;;
 	*)
 		PKG_CHECK_MODULES(USB, libusb, usb_lib_found=yes, AC_MSG_RESULT(no))
-		AC_CHECK_FILE(${prefix}/lib/pkgconfig/libusb.pc, REQUIRES="libusb")
+		AC_CHECK_FILE(${libdir}/pkgconfig/libusb.pc, REQUIRES="$REQUIRES libusb")
 		;;
 	esac
 	AC_SUBST(USB_CFLAGS)
