From 09e0f9ec22f680c759feea2e90913128ce2a249d Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 28 Apr 2010 14:35:59 +0400
Subject: [PATCH 7/7] Install missing emacs files

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 lib/tools/emacs/Makefile |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/lib/tools/emacs/Makefile b/lib/tools/emacs/Makefile
index 7249263..17978de 100644
--- a/lib/tools/emacs/Makefile
+++ b/lib/tools/emacs/Makefile
@@ -38,7 +38,10 @@ MAN_FILES= \
 
 EMACS_FILES= \
 	erlang-start \
+	erlang-skels-old \
+	erlang-skels \
 	erlang-eunit \
+	erlang_appwiz \
 	erlang
 
 README_FILES= README
-- 
1.6.6.1

