--- src/gnome-terminal.schemas.in	2008-03-20 12:16:25.000000000 +0200
@@ -153,7 +153,7 @@
       <applyto>/apps/gnome-terminal/profiles/Default/foreground_color</applyto>
       <owner>gnome-terminal</owner>
       <type>string</type>
-      <default>#000000</default>
+      <default>#AAAAAA</default>
       <locale name="C">
          <short>Default color of text in the terminal</short>
          <long>
@@ -169,7 +169,7 @@
       <applyto>/apps/gnome-terminal/profiles/Default/background_color</applyto>
       <owner>gnome-terminal</owner>
       <type>string</type>
-      <default>#FFFFDD</default>
+      <default>#000000</default>
       <locale name="C">
          <short>Default color of terminal background</short>
          <long>
@@ -569,7 +569,7 @@
       <applyto>/apps/gnome-terminal/profiles/Default/use_theme_colors</applyto>
       <owner>gnome-terminal</owner>
       <type>bool</type>
-      <default>true</default>
+      <default>false</default>
       <locale name="C">
          <short>Whether to use the colors from the theme for the terminal widget</short>
          <long>
