--- src/obj_block.cxx.seg	2005-04-20 04:35:20.000000000 +0200
+++ src/obj_block.cxx	2009-04-07 10:01:53.000000000 +0200
@@ -1515,6 +1515,12 @@
 		delete[] tex_coords;
     return;
   }
+  if (strstr((char*) renderer, "865G 20021115")) {
+    cerr << "**********\nWARNING:"
+      "disabling call to 1d texturing on 865G 20021115 renderer "
+      "since it segfaults\n**********" << endl;
+    return;
+  }
 
   glTexImage1D(GL_TEXTURE_1D, 0, GL_LUMINANCE, 32, GL_FALSE, GL_LUMINANCE,
    GL_UNSIGNED_BYTE, texture);
