--- wordlist.orig/scowl/r/special/proper-names	2008-02-08 11:53:27.000000000 +0000
+++ wordlist/scowl/r/special/proper-names	2008-02-08 12:06:00.000000000 +0000
@@ -3,7 +3,9 @@
 Barack
 Bernanke
 Biden
+Camino
 Bugzilla
+ChatZilla
 Debian
 Emacs
 Gentoo
@@ -11,10 +13,14 @@
 ISO
 Ispell
 LyX
+Mandriva
 Mozilla
+MySpell
 PayPal
+SeaMonkey
 SUSE
 Slackware
+Sunbird
 Thunderbird
 Ubuntu
 XEmacs
--- wordlist.orig/scowl/speller/en.aff	2008-02-08 20:28:24.000000000 +0000
+++ wordlist/scowl/speller/en.aff	2008-02-08 20:28:45.000000000 +0000
@@ -110,13 +110,14 @@
 SFX L Y 1
 SFX L   0     ment       .
 
-REP 88
+REP 89
 REP a ei
 REP ei a
 REP a ey
 REP ey a
 REP ai ie
 REP ie ai
+REP alot a_lot
 REP are air
 REP are ear
 REP are eir
