--- sysvinit-2.86/src/init.c.single	2005-12-21 12:21:04.000000000 -0500
+++ sysvinit-2.86/src/init.c	2005-12-21 12:21:40.000000000 -0500
@@ -1153,7 +1153,7 @@
 		 *	See if we have a single user entry.
 		 */
 		for(old = newFamily; old; old = old->next)
-			if (strpbrk(old->rlevel, "S")) break;
+			if (strpbrk(old->rlevel, "S") && !ISPOWER(old->action)) break;
 		if (old == NULL)
 			snprintf(buf, sizeof(buf), "~~:S:wait:%s -l\n", SHELL);
 		else
