--- sysvinit-2.86/src/init.c.foo	2006-09-21 14:46:26.000000000 -0400
+++ sysvinit-2.86/src/init.c	2006-09-21 14:47:00.000000000 -0400
@@ -959,7 +959,7 @@
 		 *	for any wait-type subprocess we _force_ the console
 		 *	to be its controlling tty.
 		 */
-  		if (strchr("*#sS", runlevel) && ch->flags & WAITING) {
+  		if ((strchr("*#sS", runlevel) || (ch->id[0] == 'l' && isdigit(ch->id[1]))) && ch->flags & WAITING) {
 			/*
 			 *	We fork once extra. This is so that we can
 			 *	wait and change the process group and session
