--- cmake/modules/FindGpgme.cmake
+++ cmake/modules/FindGpgme.cmake
@@ -226,10 +226,8 @@
 	  set( _gpgme_config_pthread_libs )
 	endif()
 
-        exec_program( ${_GPGMECONFIG_EXECUTABLE} ARGS --thread=pth     --libs OUTPUT_VARIABLE _gpgme_config_pth_libs     RETURN_VALUE _ret )
-	if ( _ret )
+     # disable pth
 	  set( _gpgme_config_pth_libs )
-	endif()
 
         # append -lgpg-error to the list of libraries, if necessary
         foreach ( _flavour vanilla pthread pth )
