diff -up kde-runtime-4.7.90/nepomuk/controller/nepomukcontroller.desktop.onlyshowin_kde kde-runtime-4.7.90/nepomuk/controller/nepomukcontroller.desktop
--- kde-runtime-4.7.90/nepomuk/controller/nepomukcontroller.desktop.onlyshowin_kde	2011-12-02 15:17:38.000000000 -0600
+++ kde-runtime-4.7.90/nepomuk/controller/nepomukcontroller.desktop	2011-12-09 14:40:33.763287346 -0600
@@ -98,3 +98,4 @@ X-DBUS-StartupType=Unique
 X-KDE-StartupNotify=false
 X-KDE-UniqueApplet=true
 X-KDE-autostart-condition=nepomukserverrc:Basic Settings:Start Nepomuk:true
+OnlyShowIn=KDE;
diff -up kde-runtime-4.7.90/nepomuk/services/backupsync/gui/nepomukbackup.desktop.onlyshowin_kde kde-runtime-4.7.90/nepomuk/services/backupsync/gui/nepomukbackup.desktop
--- kde-runtime-4.7.90/nepomuk/services/backupsync/gui/nepomukbackup.desktop.onlyshowin_kde	2011-09-26 04:43:12.000000000 -0500
+++ kde-runtime-4.7.90/nepomuk/services/backupsync/gui/nepomukbackup.desktop	2011-12-09 14:40:49.410091736 -0600
@@ -57,3 +57,4 @@ Exec=nepomukbackup
 Icon=nepomuk
 Type=Application
 Categories=Qt;KDE;System;Archiving;
+OnlyShowIn=KDE;
