diff -up kdebase-runtime-4.0.80/kurifilter-plugins/ikws/searchproviders/bugzilla.desktop.orig kdebase-runtime-4.0.80/kurifilter-plugins/ikws/searchproviders/bugzilla.desktop
--- kdebase-runtime-4.0.80/kurifilter-plugins/ikws/searchproviders/bugzilla.desktop.orig	2008-06-05 13:19:46.000000000 +0200
+++ kdebase-runtime-4.0.80/kurifilter-plugins/ikws/searchproviders/bugzilla.desktop	2008-06-05 13:21:39.000000000 +0200
@@ -0,0 +1,7 @@
+[Desktop Entry]
+Hidden=false
+Type=Service
+Name=Red Hat bug tracking system
+ServiceTypes=SearchProvider
+Keys=rhbug,bz
+Query=http://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=\\1
diff -up kdebase-runtime-4.0.80/kurifilter-plugins/ikws/searchproviders/fedora.desktop.orig kdebase-runtime-4.0.5/kurifilter-plugins/ikws/searchproviders/fedora.desktop
--- kdebase-runtime-4.0.80/kurifilter-plugins/ikws/searchproviders/fedora.desktop.orig	2008-06-05 13:20:19.000000000 +0200
+++ kdebase-runtime-4.0.80/kurifilter-plugins/ikws/searchproviders/fedora.desktop	2008-06-05 13:20:47.000000000 +0200
@@ -0,0 +1,7 @@
+[Desktop Entry]
+Hidden=false
+Keys=fp,fedora
+Name=fedoraproject
+Query=http://fedoraproject.org/wiki/?action=fullsearch&context=180&value=\\{@}&titlesearch=Titles
+ServiceTypes=SearchProvider
+Type=Service
