diff -up kdebase-runtime-4.3.3/kioslave/desktop/Home.desktop.home_onlyshowin_kde kdebase-runtime-4.3.3/kioslave/desktop/Home.desktop
--- kdebase-runtime-4.3.3/kioslave/desktop/Home.desktop.home_onlyshowin_kde	2009-10-02 03:18:57.000000000 -0500
+++ kdebase-runtime-4.3.3/kioslave/desktop/Home.desktop	2009-11-05 08:16:19.637144099 -0600
@@ -174,4 +174,4 @@ GenericName[zh_TW]=個人檔案
 URL[$e]=$HOME
 Icon=user-home
 Type=Link
-
+OnlyShowIn=KDE;
