diff -up kdebase-runtime-4.3.4/kioslave/man/kio_man.cpp.manpath kdebase-runtime-4.3.4/kioslave/man/kio_man.cpp
--- kdebase-runtime-4.3.4/kioslave/man/kio_man.cpp.manpath	2011-02-03 18:30:23.000000000 +0100
+++ kdebase-runtime-4.3.4/kioslave/man/kio_man.cpp	2011-02-03 18:27:09.000000000 +0100
@@ -902,6 +902,7 @@ void MANProtocol::constructPath(QStringL
 
     // Default paths
     static const char * const manpaths[] = {
+        "/usr/share/man/overrides",
         "/usr/X11/man",
         "/usr/X11R6/man",
         "/usr/man",
