diff -ur kdebase-runtime-4.5.95/kcontrol/componentchooser/windowmanagers/compiz.desktop kdebase-runtime-4.5.95-compiz/kcontrol/componentchooser/windowmanagers/compiz.desktop
--- kdebase-runtime-4.5.95/kcontrol/componentchooser/windowmanagers/compiz.desktop	2010-11-05 10:46:36.000000000 +0000
+++ kdebase-runtime-4.5.95-compiz/kcontrol/componentchooser/windowmanagers/compiz.desktop	2011-01-20 09:57:24.134453897 +0000
@@ -78,7 +78,7 @@
 Name[x-test]=xxCompizxx
 Name[zh_CN]=Compiz
 Name[zh_TW]=Compiz
-Exec=compiz ccp
+Exec=compiz-manager
 TryExec=compiz
-X-KDE-WindowManagerConfigure=simple-ccsm
+X-KDE-WindowManagerConfigure=ccsm
 X-KDE-WindowManagerRestartArgument=--replace
