diff -Nuar mate-backgrounds-1.6.1.OSmanOS/nature/Makefile.am mate-backgrounds-1.6.1/nature/Makefile.am
--- mate-backgrounds-1.6.1.OSmanOS/nature/Makefile.am	2013-12-31 17:20:19.000000000 +0200
+++ mate-backgrounds-1.6.1/nature/Makefile.am	2014-02-08 10:33:04.843370771 +0200
@@ -1,6 +1,14 @@
 backgrounddir = $(datadir)/backgrounds/mate/nature
 
 background_DATA =\
+	1.jpg \
+	2.jpg \
+	3.jpg \
+	4.jpg \
+	5.jpg \
+	6.jpg \
+	7.jpg \
+	8.jpg \
 	Aqua.jpg \
 	Blinds.jpg \
 	Dune.jpg \
diff -Nuar mate-backgrounds-1.6.1.OSmanOS/nature/Makefile.in mate-backgrounds-1.6.1/nature/Makefile.in
--- mate-backgrounds-1.6.1.OSmanOS/nature/Makefile.in	2013-12-31 17:20:21.000000000 +0200
+++ mate-backgrounds-1.6.1/nature/Makefile.in	2014-02-08 10:32:54.386704014 +0200
@@ -255,6 +255,14 @@
 top_srcdir = @top_srcdir@
 backgrounddir = $(datadir)/backgrounds/mate/nature
 background_DATA = \
+	1.jpg \
+	2.jpg \
+	3.jpg \
+	4.jpg \
+	5.jpg \
+	6.jpg \
+	7.jpg \
+	8.jpg \
 	Aqua.jpg \
 	Blinds.jpg \
 	Dune.jpg \
diff -Nuar mate-backgrounds-1.6.1.OSmanOS/nature/mate-nature.xml.in.in mate-backgrounds-1.6.1/nature/mate-nature.xml.in.in
--- mate-backgrounds-1.6.1.OSmanOS/nature/mate-nature.xml.in.in	2013-12-31 17:20:19.000000000 +0200
+++ mate-backgrounds-1.6.1/nature/mate-nature.xml.in.in	2014-02-08 09:50:02.400015155 +0200
@@ -65,4 +65,44 @@
 		<filename>@datadir@/backgrounds/mate/nature/Wood.jpg</filename>
 		<options>zoom</options>
 	</wallpaper>
+	<wallpaper deleted="false">
+		<_name>1</_name>
+		<filename>@datadir@/backgrounds/mate/nature/1.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
+	<wallpaper deleted="false">
+		<_name>2</_name>
+		<filename>@datadir@/backgrounds/mate/nature/2.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
+	<wallpaper deleted="false">
+		<_name>3</_name>
+		<filename>@datadir@/backgrounds/mate/nature/3.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
+	<wallpaper deleted="false">
+		<_name>4</_name>
+		<filename>@datadir@/backgrounds/mate/nature/4.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
+	<wallpaper deleted="false">
+		<_name>5</_name>
+		<filename>@datadir@/backgrounds/mate/nature/5.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
+	<wallpaper deleted="false">
+		<_name>6</_name>
+		<filename>@datadir@/backgrounds/mate/nature/6.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
+	<wallpaper deleted="false">
+		<_name>7</_name>
+		<filename>@datadir@/backgrounds/mate/nature/7.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
+	<wallpaper deleted="false">
+		<_name>8</_name>
+		<filename>@datadir@/backgrounds/mate/nature/8.jpg</filename>
+		<options>zoom</options>
+	</wallpaper>
 </wallpapers>
