---
 lib/dejagnu.exp |   18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

Index: dejagnu-1.4.4.cvs20060709/lib/dejagnu.exp
===================================================================
--- dejagnu-1.4.4.cvs20060709.orig/lib/dejagnu.exp	2007-04-01 18:06:32.000000000 -0400
+++ dejagnu-1.4.4.cvs20060709/lib/dejagnu.exp	2007-04-01 18:13:06.000000000 -0400
@@ -133,49 +133,50 @@ proc host_execute {args} {
     # test case.
     # spawn -noecho -open [open "|./${executable}" "r"]
     spawn -noecho "./${executable}" ${params}
+    set prefix "\[^\r\n\]*"
     expect {
-	-re "\[0-9\]\[0-9\]:..:..:${text}\r\n" {
+	-re "^$prefix\[0-9\]\[0-9\]:..:..:${text}*\r\n" {
 	    regsub "\[\n\r\t\]*NOTE: $text\r\n" $expect_out(0,string) "" output
 	    verbose "$output" 3
 	    set timetol 0
 	    exp_continue
 	}
-	-re "NOTE:${text}*" {
+	-re "^$prefix\tNOTE:${text}*" {
 	    regsub "\[\n\r\t\]*NOTE: $text\r\n" $expect_out(0,string) "" output
 	    set output [string range $output 6 end]
 	    verbose "$output" 2
 	    set timetol 0
 	    exp_continue
 	}
-	-re "PASSED:${text}*" {
+	-re "^$prefix\tPASSED:${text}*" {
 	    regsub "\[\n\r\t\]*PASSED: $text\r\n" $expect_out(0,string) "" output
 	    set output [string range $output 8 end]
 	    pass "$output"
 	    set timetol 0
 	    exp_continue
 	}
-	-re "FAILED:${text}*" {
+	-re "^$prefix\tFAILED:${text}*" {
 	    regsub "\[\n\r\t\]*FAILED: $text\r\n" $expect_out(0,string) "" output
 	    set output [string range $output 8 end]
 	    fail "$output"
 	    set timetol 0
 	    exp_continue
 	}
-	-re "UNTESTED:${text}*" {
+	-re "^$prefix\tUNTESTED:${text}*" {
 	    regsub "\[\n\r\t\]*TESTED: $text\r\n" $expect_out(0,string) "" output
 	    set output [string range $output 8 end]
 	    untested "$output"
 	    set timetol 0
 	    exp_continue
 	}
-	-re "UNRESOLVED:${text}*" {
+	-re "^$prefix\tUNRESOLVED:${text}*" {
 	    regsub "\[\n\r\t\]*UNRESOLVED: $text\r\n" $expect_out(0,string) "" output
 	    set output [string range $output 8 end]
 	    unresolved "$output"
 	    set timetol 0
 	    exp_continue
 	}
-	-re "Totals" {
+	-re "^Totals" {
 	    verbose "All done" 2
 	}
 	eof {
@@ -193,6 +194,9 @@ proc host_execute {args} {
 		return "Timed out executing test case"
 	    }
 	}
+	-re "^$prefix\r\n" {
+	    exp_continue
+	}
     }
 
     # force a close of the executable to be safe.
