From 4a1072e0bcce9165a262dc76a12baa2680caa0e3 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Mon, 19 Jul 2010 01:27:15 +0200
Subject: [PATCH 173/174] udev: fix hex decoding

---
 src/modules/udev-util.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/modules/udev-util.c b/src/modules/udev-util.c
index eee5409..52df103 100644
--- a/src/modules/udev-util.c
+++ b/src/modules/udev-util.c
@@ -63,10 +63,10 @@ static int dehex(char x) {
         return x - '0';
 
     if (x >= 'A' && x <= 'F')
-        return x - 'A';
+        return x - 'A' + 10;
 
     if (x >= 'a' && x <= 'f')
-        return x - 'a';
+        return x - 'a' + 10;
 
     return -1;
 }
-- 
1.7.2.1

