From 7cf5792832cf21687798836114bb3546d9370756 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <development@colin.guthr.ie>
Date: Wed, 8 Oct 2008 20:13:15 +0100
Subject: [PATCH 500/503] Customise startup so we can easily disable PA

---
 src/daemon/start-pulseaudio-x11.in |   16 +++++++++++++++-
 1 files changed, 15 insertions(+), 1 deletions(-)

diff --git a/src/daemon/start-pulseaudio-x11.in b/src/daemon/start-pulseaudio-x11.in
index c57c489..a12021c 100755
--- a/src/daemon/start-pulseaudio-x11.in
+++ b/src/daemon/start-pulseaudio-x11.in
@@ -21,7 +21,19 @@ set -e
 
 [ -z "$PULSE_SERVER" ]
 
-@PA_BINARY@ --start "$@"
+# Some Mandriva customisations
+CONFIG=/etc/sound/profiles/current/profile.conf
+
+PULSE_ARGS=" --log-target=syslog"
+[ -r "$CONFIG" ] && . $CONFIG
+
+if [ "$SOUNDPROFILE" != "pulse" ]; then
+  echo "Pulseaudio startup disabled due to soundprofile configuration" >&2
+else
+
+@PA_BINARY@ --start ${PULSE_ARGS} "$@"
+
+# End Mandriva customisations (except for the "fi" below)
 
 if [ x"$DISPLAY" != x ] ; then
 
@@ -32,3 +44,5 @@ if [ x"$DISPLAY" != x ] ; then
 	@PACTL_BINARY@ load-module module-x11-xsmp "display=$DISPLAY session_manager=$SESSION_MANAGER" > /dev/null
     fi
 fi
+
+fi
-- 
1.6.4

