From 14a46c2bcfc126de65aae0387adc8f18bc2ff594 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <development@colin.guthr.ie>
Date: Tue, 5 Aug 2008 20:43:41 +0100
Subject: [PATCH 502/503] Change the default resample method to speex-fixed-0 (Mandriva bug #36084)

---
 src/daemon/daemon.conf.in |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/src/daemon/daemon.conf.in b/src/daemon/daemon.conf.in
index d8b58d8..b685190 100644
--- a/src/daemon/daemon.conf.in
+++ b/src/daemon/daemon.conf.in
@@ -50,7 +50,12 @@
 ; log-time = no
 ; log-backtrace = 0
 
+# for (supposed) better quality
 ; resample-method = speex-float-3
+# for better speed (and lower quality)
+; resample-method = trivial
+# good compromise
+resample-method = speex-float-0
 ; enable-remixing = yes
 ; enable-lfe-remixing = no
 
-- 
1.6.4

