From c968dfd2782c83851f382ce65469ca7731df4be9 Mon Sep 17 00:00:00 2001
From: Frederic Crozat <fcrozat@mandriva.com>
Date: Tue, 21 Apr 2009 11:12:31 +0200
Subject: [PATCH 503/503] start PA earlier in GNOME : Mdv bug #47594

---
 src/daemon/pulseaudio.desktop.in |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/daemon/pulseaudio.desktop.in b/src/daemon/pulseaudio.desktop.in
index 99bdbd0..b75b378 100644
--- a/src/daemon/pulseaudio.desktop.in
+++ b/src/daemon/pulseaudio.desktop.in
@@ -8,3 +8,4 @@ Terminal=false
 Type=Application
 Categories=
 GenericName=
+X-GNOME-Autostart-Phase=Initialization
-- 
1.6.3.3

