From 9bc7d657112cafbe9f022518ed4e52f71f35561f Mon Sep 17 00:00:00 2001
From: Colin Guthrie <cguthrie@mandriva.org>
Date: Mon, 4 Jan 2010 19:23:26 +0000
Subject: [PATCH 506/506] Fix div-by-zero in mcalign (patch from Ubuntu)

---
 src/pulsecore/mcalign.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/pulsecore/mcalign.c b/src/pulsecore/mcalign.c
index a03d5ae..b289bf2 100644
--- a/src/pulsecore/mcalign.c
+++ b/src/pulsecore/mcalign.c
@@ -206,7 +206,7 @@ size_t pa_mcalign_csize(pa_mcalign *m, size_t l) {
     if (m->leftover.memblock)
         l += m->leftover.length;
 
-    return (l/m->base)*m->base;
+    return (m->base == 0) ? 0 : (l/m->base)*m->base;
 }
 
 void pa_mcalign_flush(pa_mcalign *m) {
-- 
1.6.6

