Index: graphviz-2.26.3/configure.ac
===================================================================
--- graphviz-2.26.3.orig/configure.ac
+++ graphviz-2.26.3/configure.ac
@@ -97,15 +97,6 @@ AC_SUBST(GRAPHVIZ_VERSION_BUILD)
 AC_SUBST(GRAPHVIZ_VERSION_REVISION)
 
 LIBPOSTFIX=
-case "${host_os}" in
-  *linux* )
-    case "${host_cpu}" in
-      powerpc64 | s390x | x86_64 )
-        LIBPOSTFIX="64"
-        ;;
-    esac
-    ;;
-esac
 
 UWIN=no
 CYGWIN=no
