diff -urN a/capplets/common/mate-theme-info.c b/capplets/common/mate-theme-info.c
--- a/capplets/common/mate-theme-info.c	2012-03-26 00:04:17.000000000 +0100
+++ b/capplets/common/mate-theme-info.c	2012-06-17 22:16:47.282911480 +0100
@@ -1499,7 +1499,7 @@
 
 		for (l = engines; l; l = l->next)
 		{
-			#if 1 // set to 0 if you can not compile with the follow code
+			#if 0 // set to 0 if you can not compile with the follow code
 			GtkThemeEngine* a = gtk_theme_engine_get((const gchar*) l->data);
 
 			if (!a)
 
