diff -urN a/capplets/common/Makefile.am b/capplets/common/Makefile.am
--- a/capplets/common/Makefile.am	2012-03-26 00:04:17.000000000 +0100
+++ b/capplets/common/Makefile.am	2012-06-17 01:00:07.810983992 +0100
@@ -45,6 +45,7 @@
 
 libcommon_la_LIBADD =							\
 	$(top_builddir)/libwindow-settings/libmate-window-settings.la  \
+        $(GMODULE_LIBS) \
 	$(MARCO_LIBS)						\
 	$(DBUS_LIBS)							\
 	$(MATE_DESKTOP_LIBS)						\
@@ -55,6 +56,7 @@
 
 mate_theme_test_LDADD = 						\
 	libcommon.la							\
+        $(GMODULE_LIBS) \
 	$(MATECC_CAPPLETS_LIBS)
 
 noinst_PROGRAMS = \
diff -urN a/configure.ac b/configure.ac
--- a/configure.ac	2012-03-26 00:04:17.000000000 +0100
+++ b/configure.ac	2012-06-17 01:01:04.632983782 +0100
@@ -60,6 +60,8 @@
 AC_SUBST(TYPING_BREAK)
 AC_SUBST(SCREENSAVER_LIBS)
 
+PKG_CHECK_MODULES(GMODULE, [gmodule-2.0])
+
 PKG_CHECK_MODULES(LIBCANBERRA_GTK, libcanberra-gtk, [
   CANBERRA_GTK=1
   AC_SUBST(LIBCANBERRA_GTK_CFLAGS)
 
