diff -Nur rp-pppoe-3.5-old/scripts/adsl-connect.in rp-pppoe-3.5/scripts/adsl-connect.in
--- rp-pppoe-3.5-old/scripts/pppoe-connect.in	2005-10-22 21:55:13.000000000 +0300
+++ rp-pppoe-3.5/scripts/pppoe-connect.in	2005-10-22 21:56:16.000000000 +0300
@@ -155,7 +155,7 @@
 fi
 
 # If DNSTYPE is SERVER, we must use "usepeerdns" option to pppd.
-if test "$DNSTYPE" = "SERVER" ; then
+if test "$DNSTYPE" = "SERVER" -o "$DNSTYPE" = "COMAR"; then
     PEERDNS=yes
     USEPEERDNS=yes
 fi
