diff -up ypbind-mt-1.32/src/serv_list.c.typo ypbind-mt-1.32/src/serv_list.c
--- ypbind-mt-1.32/src/serv_list.c.typo	2010-11-29 11:28:51.701891385 +0100
+++ ypbind-mt-1.32/src/serv_list.c	2010-11-29 11:30:29.682891306 +0100
@@ -830,7 +830,7 @@ ping_all (struct binding *list)
       if (s_in.sin_port == 0)
 	{
 	  if (verbose_flag && list->active == i)
-		log_msg (LOG_NOTICE, "NIS server '%s' not repsonding "
+		log_msg (LOG_NOTICE, "NIS server '%s' not responding "
 		    "for domain '%s'", list->server[i].host, list->domain);
 
 	  if (debug_flag)
