diff -Nur libnfo-1.0.0-old/configure libnfo-1.0.0/configure
--- libnfo-1.0.0-old/configure	2010-04-01 15:37:35.629291526 +0300
+++ libnfo-1.0.0/configure	2010-04-01 15:38:18.254291842 +0300
@@ -706,7 +706,7 @@
 #################################################
 if enabled shared; then
   case "$arch" in
-    x86_64|ia64|alpha|sparc*|power*) add_cflags "-fPIC" ;;
+    x86_64|x86_32|ia64|alpha|sparc*|power*) add_cflags "-fPIC" ;;
   esac
 fi
 
