diff -Naur tolua++-1.0.93.orig/config_linux.py tolua++-1.0.93/config_linux.py
--- tolua++-1.0.93.orig/config_linux.py	2005-12-01 00:31:27.000000000 +0200
+++ tolua++-1.0.93/config_linux.py	2009-06-02 00:04:21.000000000 +0300
@@ -4,7 +4,7 @@
 
 # Compiler flags (based on Debian's installation of lua)
 #LINKFLAGS = ['-g']
-CCFLAGS = ['-I/usr/include/lua50', '-O2', '-ansi', '-Wall']
+CCFLAGS = ['-I/usr/include', '-O2', '-ansi', '-Wall']
 #CCFLAGS = ['-I/usr/include/lua50', '-g']
 
 # this is the default directory for installation. Files will be installed on
@@ -15,8 +15,8 @@
 #
 # You can see more 'generic' options for POSIX systems on config_posix.py
 
-prefix = '/usr/local'
+prefix = '/usr'
 
 # libraries (based on Debian's installation of lua)
-LIBS = ['lua50', 'lualib50', 'dl', 'm']
+LIBS = ['lua', 'dl', 'm']
 
diff -Naur tolua++-1.0.93.orig/config_posix.py tolua++-1.0.93/config_posix.py
--- tolua++-1.0.93.orig/config_posix.py	2006-01-13 07:37:25.000000000 +0200
+++ tolua++-1.0.93/config_posix.py	2009-06-02 00:05:13.000000000 +0300
@@ -13,10 +13,10 @@
 # You can also specify this directory on the command line with the 'prefix'
 # option
 
-prefix = '/usr/local'
+prefix = '/usr'
 
 # libraries
-LIBS = ['lua', 'lualib', 'm']
+LIBS = ['lua', 'm']
 
 
 
diff -Naur tolua++-1.0.93.orig/custom-5.1.py tolua++-1.0.93/custom-5.1.py
--- tolua++-1.0.93.orig/custom-5.1.py	2008-04-21 03:05:35.000000000 +0300
+++ tolua++-1.0.93/custom-5.1.py	2009-06-02 00:06:33.000000000 +0300
@@ -1,9 +1,9 @@
-CCFLAGS = ['-I/usr/include/lua5.1', '-O2', '-ansi']
+CCFLAGS = ['-I/usr/include', '-O2', '-ansi']
 #LIBPATH = ['/usr/local/lib']
-LIBS = ['lua5.1', 'dl', 'm']
-prefix = '/mingw'
+LIBS = ['lua', 'dl', 'm']
+prefix = '/usr'
 #build_dev=1
-tolua_bin = 'tolua++5.1'
-tolua_lib = 'tolua++5.1'
-TOLUAPP = 'tolua++5.1'
+tolua_bin = 'tolua++'
+tolua_lib = 'tolua++'
+TOLUAPP = 'tolua++'
 
