diff -up cyrus-sasl-2.1.22/lib/auxprop.c.warnings cyrus-sasl-2.1.22/lib/auxprop.c
--- cyrus-sasl-2.1.22/lib/auxprop.c.warnings	2006-03-14 15:23:55.000000000 +0100
+++ cyrus-sasl-2.1.22/lib/auxprop.c	2008-09-10 14:31:01.000000000 +0200
@@ -46,6 +46,7 @@
 #include <sasl.h>
 #include <prop.h>
 #include <ctype.h>
+#include <stdio.h>
 #include "saslint.h"
 
 struct proppool 
diff -up cyrus-sasl-2.1.22/lib/server.c.warnings cyrus-sasl-2.1.22/lib/server.c
--- cyrus-sasl-2.1.22/lib/server.c.warnings	2006-05-17 18:46:13.000000000 +0200
+++ cyrus-sasl-2.1.22/lib/server.c	2008-09-10 14:39:02.000000000 +0200
@@ -517,7 +517,7 @@ static int load_config(const sasl_callba
             goto done;
         }
 
-        snprintf(config_filename, len, "%.*s%c%s.conf", path_len, path_to_config, 
+        snprintf(config_filename, len, "%.*s%c%s.conf", (int)path_len, path_to_config, 
 	        HIER_DELIMITER, global_callbacks.appname);
 
         /* Ask the application if it's safe to use this file */
diff -up cyrus-sasl-2.1.22/plugins/gssapi.c.warnings cyrus-sasl-2.1.22/plugins/gssapi.c
--- cyrus-sasl-2.1.22/plugins/gssapi.c.warnings	2004-07-21 16:39:06.000000000 +0200
+++ cyrus-sasl-2.1.22/plugins/gssapi.c	2008-09-10 14:20:59.000000000 +0200
@@ -190,7 +190,8 @@ sasl_gss_seterror_(const sasl_utils_t *u
     OM_uint32 msg_ctx;
     int ret;
     char *out = NULL;
-    size_t len, curlen = 0;
+    size_t len;
+    unsigned curlen = 0;
     const char prefix[] = "GSSAPI Error: ";
     
     len = sizeof(prefix);
diff -up cyrus-sasl-2.1.22/plugins/ldapdb.c.warnings cyrus-sasl-2.1.22/plugins/ldapdb.c
--- cyrus-sasl-2.1.22/plugins/ldapdb.c.warnings	2005-02-19 03:26:31.000000000 +0100
+++ cyrus-sasl-2.1.22/plugins/ldapdb.c	2008-09-10 14:52:33.000000000 +0200
@@ -21,6 +21,7 @@
 
 #include "plugin_common.h"
 
+#define LDAP_DEPRECATED 1
 #include <ldap.h>
 
 static char ldapdb[] = "ldapdb";
diff -up cyrus-sasl-2.1.22/saslauthd/lak.c.warnings cyrus-sasl-2.1.22/saslauthd/lak.c
--- cyrus-sasl-2.1.22/saslauthd/lak.c.warnings	2005-05-15 07:49:51.000000000 +0200
+++ cyrus-sasl-2.1.22/saslauthd/lak.c	2008-09-10 14:52:51.000000000 +0200
@@ -55,6 +55,7 @@
 #include <openssl/des.h>
 #endif
 
+#define LDAP_DEPRECATED 1
 #include <ldap.h>
 #include <lber.h>
 #include <sasl.h>
diff -up cyrus-sasl-2.1.22/saslauthd/auth_httpform.c.warnings cyrus-sasl-2.1.22/saslauthd/auth_httpform.c
--- cyrus-sasl-2.1.22/saslauthd/auth_httpform.c.warnings	2006-04-19 21:51:13.000000000 +0200
+++ cyrus-sasl-2.1.22/saslauthd/auth_httpform.c	2008-09-10 14:45:57.000000000 +0200
@@ -552,7 +552,7 @@ auth_httpform (
               "Content-Type: application/x-www-form-urlencoded" CRLF
               "Content-Length: %d" TWO_CRLF
               "%s",
-              r_uri, r_host, r_port, strlen(escreq), escreq);
+              r_uri, r_host, r_port, (int)strlen(escreq), escreq);
 
     if (flags & VERBOSE) {
         syslog(LOG_DEBUG, "auth_httpform: sending %s %s %s",
diff -up cyrus-sasl-2.1.22/saslauthd/auth_shadow.c.warnings cyrus-sasl-2.1.22/saslauthd/auth_shadow.c
--- cyrus-sasl-2.1.22/saslauthd/auth_shadow.c.warnings	2006-04-19 21:36:36.000000000 +0200
+++ cyrus-sasl-2.1.22/saslauthd/auth_shadow.c	2008-09-10 14:47:47.000000000 +0200
@@ -63,6 +63,10 @@
 #  include <shadow.h>
 # endif /* ! HAVE_GETUSERPW */
 
+# ifdef HAVE_CRYPT_H
+#  include <crypt.h>
+# endif
+
 # include "auth_shadow.h"
 # include "globals.h"
 /* END PUBLIC DEPENDENCIES */
