diff -up cyrus-sasl-2.1.23/plugins/ntlm.c.ntlm cyrus-sasl-2.1.23/plugins/ntlm.c
--- cyrus-sasl-2.1.23/plugins/ntlm.c.ntlm	2011-05-25 04:07:22.923341599 +0200
+++ cyrus-sasl-2.1.23/plugins/ntlm.c	2011-05-25 04:08:02.648341353 +0200
@@ -1572,7 +1572,7 @@ static int ntlm_server_mech_step2(server
 	    sparams->utils->log(NULL, SASL_LOG_DEBUG,
 				"calculating NTv2 response");
 	    V2(resp, password, authid, domain, text->nonce,
-	       lm_resp + MD5_DIGEST_LENGTH, nt_resp_len - MD5_DIGEST_LENGTH,
+	       nt_resp + MD5_DIGEST_LENGTH, nt_resp_len - MD5_DIGEST_LENGTH,
 	       sparams->utils, &text->out_buf, &text->out_buf_len,
 	       &result);
 
