--- cyrus-sasl-2.1.18/cmulocal/cyrus.m4	2004-05-13 14:32:14.341038898 +0200
+++ cyrus-sasl-2.1.18/cmulocal/cyrus.m4	2004-05-13 14:34:18.471968788 +0200
@@ -32,14 +32,5 @@
 dnl runpath initialization
 AC_DEFUN([CMU_GUESS_RUNPATH_SWITCH], [
    # CMU GUESS RUNPATH SWITCH
-  AC_CACHE_CHECK(for runpath switch, andrew_runpath_switch, [
-    # first, try -R
-    SAVE_LDFLAGS="${LDFLAGS}"
-    LDFLAGS="-R /usr/lib"
-    AC_TRY_LINK([],[],[andrew_runpath_switch="-R"], [
-  	LDFLAGS="-Wl,-rpath,/usr/lib"
-    AC_TRY_LINK([],[],[andrew_runpath_switch="-Wl,-rpath,"],
-    [andrew_runpath_switch="none"])
-    ])
-  LDFLAGS="${SAVE_LDFLAGS}"
-  ])])
+    andrew_runpath_switch="none"
+  ])
