diff -up cyrus-sasl-2.1.22/config/kerberos_v4.m4.krb4 cyrus-sasl-2.1.22/config/kerberos_v4.m4
--- cyrus-sasl-2.1.22/config/kerberos_v4.m4.krb4	2005-05-07 06:14:55.000000000 +0200
+++ cyrus-sasl-2.1.22/config/kerberos_v4.m4	2008-08-14 23:41:26.000000000 +0200
@@ -102,7 +102,6 @@ AC_DEFUN([SASL_KERBEROS_V4_CHK], [
        if test -n "${cyrus_krbinclude}"; then
          CPPFLAGS="$CPPFLAGS -I${cyrus_krbinclude}"
        fi
-       LDFLAGS="$LDFLAGS -L$krb4/lib"
     fi
 
     if test "$with_des" != no; then
diff -up cyrus-sasl-2.1.22/plugins/kerberos4.c.krb4 cyrus-sasl-2.1.22/plugins/kerberos4.c
--- cyrus-sasl-2.1.22/plugins/kerberos4.c.krb4	2005-01-10 08:08:53.000000000 +0100
+++ cyrus-sasl-2.1.22/plugins/kerberos4.c	2008-08-14 23:36:33.000000000 +0200
@@ -49,11 +49,7 @@
 #include <krb.h>
 
 #ifdef WITH_DES
-# ifdef WITH_SSL_DES
-#  include <openssl/des.h>
-# else
 #  include <des.h>
-# endif /* WITH_SSL_DES */
 #endif /* WITH_DES */
 
 #ifdef WIN32
