diff -Nur pyv4l-0.5.0-old/v4l.c pyv4l-0.5.0/v4l.c
--- pyv4l-0.5.0-old/v4l.c	2007-03-03 20:30:47.352201530 +0200
+++ pyv4l-0.5.0/v4l.c	2007-03-03 20:30:58.079570722 +0200
@@ -229,6 +229,11 @@
 "\n"
 "Returns channel information.";
 
+static char getchannelext_doc[] =
+"getChannelExt(channel) -> (int channel, string name, int tuners, int type, int norm)"
+"\n"
+"Returns full channel information.";
+
 static char getoverlay_doc[] =
 "getOverlay() -> (int x, int y, int width, int height, int chromakey, int flags, int clipcount)\n"
 "\n"
@@ -249,8 +254,6 @@
 "\n"
 "Returns tuner information.";
 
-
-
 static char getvbi_doc[] =
 "getVbi() -> (int sampling_rate, int samples_per_line,\n\
     int sample_format, (int start1, int start2),\n\
@@ -259,9 +262,9 @@
 "Returns VBI information.";
 
 static char setchannel_doc[] =
-"setChannel(channel) -> None\n"
+"setChannel(channel,norm) -> None\n"
 "\n"
-"Set Channel.";
+"Set Channel. Optional norm can be specified for channel";
 
 static char setoverlay_doc[] =
 "setOverlay(x, y, width, height) -> None\n"
@@ -377,22 +380,43 @@
 }
 
 
+static PyObject *
+v4l_getChannelExt(v4lobject *self, PyObject *args) {
+  int chan;
+  if (!PyArg_ParseTuple(args, "i", &chan)) return NULL;
+  struct video_channel vc;
+  vc.channel = chan;
+  PyObject *vidchan;
+  if (ioctl(self->fd, VIDIOCGCHAN, &vc) < 0) {
+    PyErr_SetString(V4lError, "Error retriving video channel.");
+    return NULL;
+  }
+  vidchan = Py_BuildValue("(isiii)",
+			  vc.channel,
+			  vc.name,
+			  vc.tuners,
+			  vc.type,
+			  vc.norm
+			  );
+  return vidchan;
+}
 
 
 static PyObject *
 v4l_setChannel(v4lobject *self, PyObject *args) {
     int chan;
+    int norm;
+    if (!PyArg_ParseTuple(args, "i|i", &chan, &norm)) 
+      return NULL;
     struct video_channel vc;
-    if (!PyArg_ParseTuple(args, "i", &chan)) return NULL;
-    //printf("Setting Video Channel: %d\n", chan);
     vc.channel = chan;
     if (ioctl(self->fd, VIDIOCGCHAN, &vc) < 0) {
 	PyErr_SetString(V4lError, "Error retriving video channel.");
 	return NULL;
     }
+    if ( PySequence_Length(args) > 1) // norm is requested to be set
+      vc.norm = norm;
    
-    //strcpy(vc.name, "Television");
-    //vc.tuners = 1;
     if (ioctl(self->fd, VIDIOCSCHAN, &vc) < 0) {
 	PyErr_SetString(V4lError, "Error setting channel.");
 	return NULL;
@@ -1218,6 +1242,7 @@
     {"getCapabilities", (PyCFunction) v4l_getCapabilities, METH_VARARGS, getcapabilities_doc},
     {"getChannel", (PyCFunction) v4l_getChannel, METH_VARARGS, getchannel_doc},
     {"setChannel", (PyCFunction) v4l_setChannel, METH_VARARGS, setchannel_doc},
+    {"getChannelExt", (PyCFunction) v4l_getChannelExt, METH_VARARGS, getchannelext_doc},
     {"setFrequency", (PyCFunction) v4l_setFrequency, METH_VARARGS, setfreq_doc},
     {"getFrequency", (PyCFunction) v4l_getFrequency, METH_NOARGS, getfreq_doc},
     {"setupImage", (PyCFunction) v4l_setupImage, METH_VARARGS, setupimage_doc},
