diff -Nur pyv4l-0.5.0-old/v4l.c pyv4l-0.5.0/v4l.c
--- pyv4l-0.5.0-old/v4l.c	2007-03-03 20:01:21.266687338 +0200
+++ pyv4l-0.5.0/v4l.c	2007-03-03 20:01:45.287035690 +0200
@@ -819,7 +819,8 @@
     if (ioctl(self->fd, VIDIOCGMBUF, &self->mbuf) < 0) return -1;
     //fprintf(stderr,"  mbuf: size=%d frames=%d\n", self->mbuf.size,self->mbuf.frames);
     //self->map = mmap(0, self->mbuf.size, PROT_READ|PROT_WRITE, MAP_SHARED, self->fd, 0);
-    self->map = mmap(0, self->mbuf.size, PROT_READ, MAP_PRIVATE, self->fd, 0);
+    //self->map = mmap(0, self->mbuf.size, PROT_READ, MAP_PRIVATE, self->fd, 0);
+    self->map = mmap(0, self->mbuf.size, PROT_READ|PROT_WRITE, MAP_SHARED, self->fd, 0);
     if ((self->map == NULL) | (self->map == MAP_FAILED)) {
 	PyErr_SetString(VideoError, "Error Memory Mapping device.");
     	return -1;
