diff -up expect-5.43/exp_log.c.log_file expect-5.43/exp_log.c
--- expect-5.43/exp_log.c.log_file	2004-08-18 22:54:35.000000000 +0200
+++ expect-5.43/exp_log.c	2008-09-25 12:24:42.000000000 +0200
@@ -471,7 +471,7 @@ expLogChannelOpen(interp,filename,append
 	Tcl_DStringAppend(&tsdPtr->logFilename,filename,-1);
     }
 
-    tsdPtr->logChannel = Tcl_OpenFileChannel(interp,newfilename,mode,0777);
+    tsdPtr->logChannel = Tcl_OpenFileChannel(interp,newfilename,mode,0666);
     if (!tsdPtr->logChannel) {
 	Tcl_DStringFree(&tsdPtr->logFilename);
 	return TCL_ERROR;
