--- configure~	2007-11-26 21:47:43.000000000 +0100
+++ configure	2008-10-20 03:54:58.000000000 +0200
@@ -610,9 +610,9 @@
 fi
 
 echolog "Checking for libavformat ..."
-check_lib ffmpeg/avformat.h av_register_all -lavformat || die "Error, can't find libavformat !"
+check_lib libavformat/avformat.h av_register_all -lavformat || die "Error, can't find libavformat !"
 echolog "Checking for libavcodec ..."
-check_lib ffmpeg/avcodec.h avcodec_register_all -lavcodec || die "Error, can't find libavcodec !"
+check_lib libavcodec/avcodec.h avcodec_register_all -lavcodec || die "Error, can't find libavcodec !"
 
 #################################################
 #   version
--- src/containers.c~	2007-11-26 21:47:43.000000000 +0100
+++ src/containers.c	2008-10-20 03:56:34.000000000 +0200
@@ -4,7 +4,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 
 #include "containers.h"
 #include "profiles.h"
--- src/profiles.h~	2007-11-26 21:47:43.000000000 +0100
+++ src/profiles.h	2008-10-20 03:56:05.000000000 +0200
@@ -22,8 +22,8 @@
 #ifndef _PROFILES_H_
 #define _PROFILES_H_
 
-#include <ffmpeg/avcodec.h>
-#include <ffmpeg/avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 
 #include "dlna_internals.h"
 #include "containers.h"
--- src/av_mpeg4_part10.c~	2007-11-26 21:47:43.000000000 +0100
+++ src/av_mpeg4_part10.c	2008-10-20 03:57:28.000000000 +0200
@@ -26,7 +26,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 
 #include "dlna_internals.h"
 #include "profiles.h"
--- src/av_mpeg4_part2.c~	2007-11-26 21:47:43.000000000 +0100
+++ src/av_mpeg4_part2.c	2008-10-20 03:57:03.000000000 +0200
@@ -26,7 +26,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 
 #include "dlna_internals.h"
 #include "profiles.h"
