--- ufoai-2.4-source/Makefile~	2012-05-03 20:26:43.234788947 +0200
+++ ufoai-2.4-source/Makefile	2012-05-03 21:08:59.808214435 +0200
@@ -24,7 +24,7 @@ ifeq ($(DISABLE_DEPENDENCY_TRACKING),)
 endif
 
 INSTALL         ?= install
-INSTALL_PROGRAM ?= $(INSTALL) -m 755 -s
+INSTALL_PROGRAM ?= $(INSTALL) -m 755
 INSTALL_SCRIPT  ?= $(INSTALL) -m 755
 INSTALL_DIR     ?= $(INSTALL) -d
 INSTALL_MAN     ?= $(INSTALL) -m 444
