diff -Naur agg-2.5-orig/src/platform/X11/Makefile.am agg-2.5/src/platform/X11/Makefile.am
--- agg-2.5-orig/src/platform/X11/Makefile.am	2009-05-14 14:27:38.537878975 +0300
+++ agg-2.5/src/platform/X11/Makefile.am	2009-05-14 14:27:59.432631216 +0300
@@ -1,7 +1,7 @@
 if ENABLE_X11
 lib_LTLIBRARIES = libaggplatformX11.la
 
-libaggplatformX11_la_LDFLAGS = -version-info @AGG_LIB_VERSION@  -L@x_libraries@
+libaggplatformX11_la_LDFLAGS = -version-info @AGG_LIB_VERSION@
 libaggplatformX11_la_SOURCES = agg_platform_support.cpp
 libaggplatformX11_la_CXXFLAGS =  -I$(top_srcdir)/include -I@x_includes@
 libaggplatformX11_la_LIBADD = -lX11
