Index: ppd/Makefile
===================================================================
--- ppd/Makefile.orig
+++ ppd/Makefile
@@ -75,25 +75,16 @@ install:
 	$(Q)install -d -m 755 ${DESTDIR}${CUPSPPD}/samsung
 	$(Q)for filename in ${SAMSUNG}; do \
 		install -m 644 $$filename.${DRIVERSEXT} ${DESTDIR}${CUPSPPD}/samsung;\
-		for lang in ${LANGUAGES}; do \
-			install -m 644 $$filename$$lang.${DRIVERSEXT} ${DESTDIR}${CUPSPPD}/samsung;\
-		done; \
 	done;
 	$(call printCmd, $(cmd_install_xerox))
 	$(Q)install -d -m 755 ${DESTDIR}${CUPSPPD}/xerox
 	$(Q)for filename in ${XEROX}; do \
 		install -m 644 $$filename.${DRIVERSEXT} ${DESTDIR}${CUPSPPD}/xerox;\
-		for lang in ${LANGUAGES}; do \
-			install -m 644 $$filename$$lang.${DRIVERSEXT} ${DESTDIR}${CUPSPPD}/xerox;\
-		done; \
 	done;
 	$(call printCmd, $(cmd_install_dell))
 	$(Q)install -d -m 755 ${DESTDIR}${CUPSPPD}/dell
 	$(Q)for filename in ${DELL}; do \
 		install -m 644 $$filename.${DRIVERSEXT} ${DESTDIR}${CUPSPPD}/dell;\
-		for lang in ${LANGUAGES}; do \
-			install -m 644 $$filename$$lang.${DRIVERSEXT} ${DESTDIR}${CUPSPPD}/dell;\
-		done; \
 	done;
 
 .PHONY: clean distclean
