diff -up splix/rules.mk.nostrip splix/rules.mk
--- splix/rules.mk.nostrip	2011-06-27 20:38:45.000000000 +0200
+++ splix/rules.mk	2011-11-18 18:09:14.438351922 +0100
@@ -21,9 +21,9 @@ cmd_install_cms		= INSTALL           col
 install: $(rastertoqpdl_TARGET) $(pstoqpdl_TARGET)
 	$(Q)mkdir -p $(DESTDIR)${CUPSFILTER}
 	$(call printCmd, $(cmd_install_raster))
-	$(Q)install -m 755 -s $(rastertoqpdl_TARGET) $(DESTDIR)${CUPSFILTER}
+	$(Q)install -m 755 $(rastertoqpdl_TARGET) $(DESTDIR)${CUPSFILTER}
 	$(call printCmd, $(cmd_install_ps))
-	$(Q)install -m 755 -s $(pstoqpdl_TARGET) $(DESTDIR)${CUPSFILTER}
+	$(Q)install -m 755 $(pstoqpdl_TARGET) $(DESTDIR)${CUPSFILTER}
 	$(Q)$(MAKE) --no-print-directory -C ppd install Q=$(Q) \
 		DESTDIR=$(abspath $(DESTDIR)) DISABLE_JBIG=$(DISABLE_JBIG)
 	@echo ""
