Index: splix-2.0.0/Makefile
===================================================================
--- splix-2.0.0.orig/Makefile
+++ splix-2.0.0/Makefile
@@ -55,8 +55,6 @@
 # +--------------------------------------------------------------------------+
 LANGUAGES	:= cpp c
 
-CC		:= gcc
-CXX		:= g++
 RM		:= rm -f
 AR		:= ar crs
 LEX		:= flex
@@ -72,10 +70,10 @@ empty           :=
 space           := $(empty) $(empty)
 comma           := ,
 
-DEBUG_CFLAGS    := -O0 -g
-DEBUG_CXXFLAGS  := -O0 -g
-OPTIM_CFLAGS	:= -O2
-OPTIM_CXXFLAGS	:= -O2
+DEBUG_CFLAGS    := $(OS_CFLAGS)
+DEBUG_CXXFLAGS  := $(OS_CXXFLAGS)
+OPTIM_CFLAGS	:= $(OS_CFLAGS)
+OPTIM_CXXFLAGS	:= $(OS_CXXFLAGS)
 
 ARCHI           := $(shell uname -s)
 
Index: splix-2.0.0/module.mk
===================================================================
--- splix-2.0.0.orig/module.mk
+++ splix-2.0.0/module.mk
@@ -26,13 +26,13 @@ DISABLE_BLACKOPTIM	?= 0
 
 
 # Flags
-CXXFLAGS		+= `cups-config --cflags` -Iinclude -Wall -I/opt/local/include
-DEBUG_CXXFLAGS		+= -DDEBUG  -DDUMP_CACHE
-OPTIM_CXXFLAGS 		+= -g
-rastertoqpdl_LDFLAGS	:= `cups-config --ldflags` -L/opt/local/lib
-rastertoqpdl_LIBS	:= `cups-config --libs` -lcupsimage
-pstoqpdl_LDFLAGS	:= `cups-config --ldflags`
-pstoqpdl_LIBS		:= `cups-config --libs` -lcupsimage
+CXXFLAGS		+=  `cups-config --cflags` -Iinclude $(OS_CXXFLAGS)
+DEBUG_CXXFLAGS		+=  $(OS_CXXFLAGS)
+OPTIM_CXXFLAGS 		+=  $(OS_CXXFLAGS)
+rastertoqpdl_LDFLAGS    :=  `cups-config --ldflags` $(OS_LDFLAGS) -L/opt/local/lib
+rastertoqpdl_LIBS  := `cups-config --libs` -lcupsimage
+pstoqpdl_LDFLAGS   := `cups-config --ldflags` $(OS_LDFLAGS)
+pstoqpdl_LIBS   := `cups-config --libs` -lcupsimage
 
 
 # Update compilation flags with defined options
