From 361a2ce3f2fc0b6ef02ac11d31bf77a0a9354e70 Mon Sep 17 00:00:00 2001
From: Ayan George <ayan.george@canonical.com>
Date: Tue, 03 May 2011 06:47:34 +0000
Subject: Mark Realtek PCI flash card reader as such

https://launchpad.net/bugs/759963

Signed-off-by: Martin Pitt <martin.pitt@ubuntu.com>
---
diff --git a/data/80-udisks.rules b/data/80-udisks.rules
index 94119b5..2be210d 100644
--- a/data/80-udisks.rules
+++ b/data/80-udisks.rules
@@ -135,6 +135,9 @@ SUBSYSTEMS=="usb", ATTRS{idVendor}=="05e3", ATTRS{idProduct}=="070e", ENV{ID_INS
 #
 SUBSYSTEMS=="usb", ATTRS{idVendor}=="05ac", ATTRS{idProduct}=="8403", ENV{ID_DRIVE_FLASH_SD}="1"
 
+# Realtek card reader
+SUBSYSTEMS=="pci", ATTRS{vendor}=="0x10ec", ATTRS{device}=="0x5208", ENV{ID_DRIVE_FLASH}="1"
+
 # Common theme
 #
 SUBSYSTEMS=="usb", ENV{ID_MODEL}=="*SD_Reader*", ENV{ID_DRIVE_FLASH_SD}="1"
--
cgit v0.8.3-6-g21f6
