diff --git a/gweather/gweather-applet.c b/gweather/gweather-applet.c
index 3768ded..aa8e0ed 100644
--- a/gweather/gweather-applet.c
+++ b/gweather/gweather-applet.c
@@ -394,7 +394,6 @@ update_finish (WeatherInfo *info, gpointer data)
     static int gw_fault_counter = 0;
 #ifdef HAVE_LIBNOTIFY
     char *message, *detail;
-    GConfClient *conf;
 #endif
     char *s;
     GWeatherApplet *gw_applet = (GWeatherApplet *)data;
diff --git a/gweather/gweather-pref.c b/gweather/gweather-pref.c
index 9366e3a..ecf4de0 100644
--- a/gweather/gweather-pref.c
+++ b/gweather/gweather-pref.c
@@ -294,11 +294,9 @@ compare_location (GtkTreeModel *model,
 
 static void load_locations (GWeatherPref *pref)
 {
-    GWeatherApplet *gw_applet = pref->priv->applet;
     GtkTreeView *tree = GTK_TREE_VIEW(pref->priv->tree);
     GtkTreeViewColumn *column;
     GtkCellRenderer *cell_renderer;
-    WeatherLocation *current_location;
         
     /* Add a colum for the locations */
     cell_renderer = gtk_cell_renderer_text_new ();