Index: QtCurve-Gtk2-0.69.2/style/qt_settings.c
===================================================================
--- QtCurve-Gtk2-0.69.2.orig/style/qt_settings.c
+++ QtCurve-Gtk2-0.69.2/style/qt_settings.c
@@ -205,7 +205,7 @@ struct QtData
 #define DEFAULT_KDE_FONT      "Sans Serif"
 // Seems to be 9pt on KUbuntu, but 10.0 on openSUSE ???
 //#define DEFAULT_KDE_FONT_SIZE (qtSettings.qt4 ? 9.0 : 10.0)
-#define DEFAULT_KDE_FONT_SIZE 10.0
+#define DEFAULT_KDE_FONT_SIZE 9.0
 #define MAX_LINE_LEN 1024
 
 struct QtData qtSettings;
