diff -up foomatic-filters-4.0.0/configure.ac.libdir foomatic-filters-4.0.0/configure.ac
--- foomatic-filters-4.0.0/configure.ac.libdir	2009-01-14 19:23:15.000000000 +0000
+++ foomatic-filters-4.0.0/configure.ac	2009-01-15 10:57:55.000000000 +0000
@@ -43,7 +43,7 @@ AC_CHECK_FUNCS([dup2 getcwd gethostname 
 
 
 ETCSEARCHPATH=${prefix}/etc:/usr/etc:/usr/local/etc
-LIBSEARCHPATH=${prefix}/lib:${prefix}/lib64:/usr/lib:/usr/lib64:/usr/local/lib:${prefix}/libexec:/usr/libexec:/usr/local/libexec
+LIBSEARCHPATH=${libdir}:/usr/lib:/usr/lib64:/usr/local/lib:${prefix}/libexec:/usr/libexec:/usr/local/libexec
 
 uname=`uname`
 uversion=`uname -r | sed -e '1,$s/[[^0-9]]//g'`
