diff --git a/tcsh.man b/tcsh.man
index 0c9c3b6..3e389f4 100644
--- a/tcsh.man
+++ b/tcsh.man
@@ -1035,7 +1035,7 @@ Substitute \fIl\fR for \fIr\fR.
 \fIl\fR is simply a string like \fIr\fR, not a regular expression as in
 the eponymous \fIed\fR(1) command.
 Any character may be used as the delimiter in place of `/';
-a `\\' can be used to quote the delimiter inside \fIl\fR and \fIr\fR.
+a `\\' can be used to quote the delimiter expect `(', `)', `|' and `>' inside \fIl\fR and \fIr\fR.
 The character `&' in the \fIr\fR is replaced by \fIl\fR; `\\' also quotes `&'.
 If \fIl\fR is empty (``''), the \fIl\fR from a previous substitution or the
 \fIs\fR from a previous search or event number in event specification is used.
