--- ivtv-utils-1.4.1.orig/debian/compat
+++ ivtv-utils-1.4.1/debian/compat
@@ -0,0 +1 @@
+5
--- ivtv-utils-1.4.1.orig/debian/copyright
+++ ivtv-utils-1.4.1/debian/copyright
@@ -0,0 +1,48 @@
+This package was debianized by Ian Campbell <ijc@hellion.org.uk> on
+Fri, 22 Apr 2005 20:03:56 +0100.
+
+It was downloaded from http://dl.ivtvdriver.org
+
+Copyright Holders:
+
+	Copyright (C) 2005	Adam Forsyth <agforsyth@gmail.com>
+	Copyright (C) 2004	Brian Jackson
+	Copyright (C) 2004	Chris Kennedy <c@groovy.org>
+	Copyright (C) 1998	Dave Perks <dperks@ibm.net>
+	Copyright (C) 1998-99	Frodo Looijaard <frodol@dds.nl>
+	Copyright (C) 2003	Hans-Peter Nilsson.
+	Copyright (C) 2004-2005	Hans Verkuil <hverkuil at xs4all.nl>
+	Copyright (C) 2003-2004	James A. Pattie <james@pcxperience.com>
+	Copyright (C) 2005	Jelte van der Hoek.
+	Copyright (C) 2001-2003	Kees Cook
+	Copyright (C) 2003-2004	Kevin Thayer <nufan_wfk at yahoo dot com>
+        Copyright (C) 2006	Maksym Veremeyenko (verem@m1stereo.tv)
+	Copyright (C) 2004	Mark Rafn
+	Copyright (C) 2004	Matthias Badaire
+	Copyright (c) 2003	Matt T. Yourst <yourst@yourst.com>
+	Copyright (C) 2002	Maxim Yevtyushkin <max@linuxmedialabs.com>
+	Copyright (C) 2002	Pontscho (pontscho@makacs.poliod.hu)
+	Copyright (C) 2000 	Ralph  Metzler <ralph@xxxxxxxxxxxxxx>
+	Copyright 		Gerard Lantau  (see http://ffmpeg.sf.net)
+	Copyright (C) 1995-97	Simon G. Vogl
+	Copyright (C) 2004	Stephane Zermatten
+	Copyright (C) 2004	Steven Fuerst & Rob Caelers
+	Copyright (C) 2005	T. Adachi <tadachi@tadachi-net.com>
+	Copyright (C) 2005	Tyler Trafford <ttrafford@gmail.com>
+
+License:
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License with
+the Debian GNU/Linux distribution in file /usr/share/common-licenses/GPL;
+if not, write to the Free Software Foundation, 51 Franklin St, Fifth Floor, Boston,
+MA 02110-1301, USA.
--- ivtv-utils-1.4.1.orig/debian/watch
+++ ivtv-utils-1.4.1/debian/watch
@@ -0,0 +1,3 @@
+# See uscan(1) for format
+version=3
+http://dl.ivtvdriver.org/ivtv/stable/ivtv-utils-(.*)\.tar\.gz debian svn-upgrade
--- ivtv-utils-1.4.1.orig/debian/control
+++ ivtv-utils-1.4.1/debian/control
@@ -0,0 +1,19 @@
+Source: ivtv-utils
+Section: contrib/x11
+Priority: extra
+Maintainer: Debian MythTV Team <pkg-mythtv-maintainers@lists.alioth.debian.org>
+Uploaders: Ian Campbell <ijc@hellion.org.uk>, Mark Purcell <msp@debian.org>
+Build-Depends: debhelper, bzip2, dpatch
+Standards-Version: 3.8.1.0
+Vcs-Svn: svn://svn.debian.org/pkg-mythtv/ivtv/trunk/
+Vcs-Browser: http://svn.debian.org/wsvn/pkg-mythtv/ivtv/trunk/
+Homepage: http://ivtvdriver.org
+
+Package: ivtv-utils
+Architecture: any
+Depends: ${shlibs:Depends}, ${perl:Depends}, libconfig-inifiles-perl, libvideo-ivtv-perl, libvideo-capture-v4l-perl
+Recommends: perl-tk, devscripts, fakeroot, unzip
+Description: utilities for use with the ivtv kernel driver
+ The IVTV project develops a kernel driver for Linux and a driver for
+ X11 for hardware based on Conexant's CX23415/CX23416 codec chip such
+ as the Hauppauge PVR 150/250/350/500 models and other supported hardware. 
--- ivtv-utils-1.4.1.orig/debian/docs
+++ ivtv-utils-1.4.1/debian/docs
@@ -0,0 +1,11 @@
+README
+doc/modules.txt
+doc/README.devices
+doc/README.install
+doc/README.ivtvfb
+doc/README.lirc
+doc/README.radio
+doc/README.tda9887
+doc/README.utils
+doc/README.vbi
+doc/video-quality.txt
--- ivtv-utils-1.4.1.orig/debian/README.Debian
+++ ivtv-utils-1.4.1/debian/README.Debian
@@ -0,0 +1,8 @@
+Firmware
+--------
+
+The IVTV drivers require firmware which can be obtained by installing
+the firmware-ivtv package from non-free. (previously the firmware package was
+called ivtv-firmware.)
+
+ -- Ian Campbell <ijc@hellion.org.uk>, Mon Feb 19 21:08:29 GMT 2007
--- ivtv-utils-1.4.1.orig/debian/NEWS
+++ ivtv-utils-1.4.1/debian/NEWS
@@ -0,0 +1,22 @@
+ivtv (1.3.0-1) unstable; urgency=low
+
+  * The IVTV driver has been included in the upstream kernel since v2.6.22
+    (framebuffer driver since v2.6.24) therefore the ivtv-source component
+    of this package is no longer available or necessary and there is no
+    need to build your own modules.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Wed, 15 Apr 2009 20:32:13 +0100
+
+ivtv (1.0.3-1) unstable; urgency=low
+
+  * The IVTV framebuffer driver module has been renamed by upstream from
+    "ivtv-fb.ko" to "ivtvfb.ko".
+  
+ -- Ian Campbell <ijc@hellion.org.uk>  Sun, 04 Nov 2007 09:45:23 +0000
+
+ivtv (0.6.0-1)
+
+  * The IVTV devices have been renumbered starting with 1 instead of
+    zero. i.e. "Tuner 0" has become "Tuner 1" etc.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Wed, 22 Mar 2006 19:07:08 +0000
--- ivtv-utils-1.4.1.orig/debian/dirs
+++ ivtv-utils-1.4.1/debian/dirs
@@ -0,0 +1 @@
+usr/bin
--- ivtv-utils-1.4.1.orig/debian/changelog
+++ ivtv-utils-1.4.1/debian/changelog
@@ -0,0 +1,571 @@
+ivtv-utils (1.4.1-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Mon, 12 Apr 2010 07:43:30 +0100
+
+ivtv-utils (1.4.0-1) unstable; urgency=low
+
+  * New upstream release.
+  * Update debian/watch to reflect new source package name.
+  * Builds against newer kernel again, drop 06_use-own-videodev2.h.dpatch.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Tue, 09 Jun 2009 19:49:36 +0100
+
+ivtv-utils (1.3.0-1) unstable; urgency=low
+
+  [ Mark Purcell ]
+  * New upstream release
+  * Add Homepage:
+
+  [ Ian Campbell ]
+  * New upstream release.
+  - The source package name is now ivtv-utils. The driver modules are all
+    in the kernel proper included so the ivtv-source package has been
+    removed.
+  - Simplified build since we now build all subdirectories.
+  - Patches which are no longer required:
+    - 01_no-depmod
+    - 02_build-with-make-kpkg
+    - 04_system-include-videodev2.h.dpatch
+    - 05_local-include-ivtv.h.dpatch
+  - New patch:
+    - 06_use-own-videodev2.h.dpatch
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Wed, 20 May 2009 22:56:08 +0100
+
+ivtv (1.2.0-1) unstable; urgency=low
+
+  * New upstream release.
+  - The majority of the drivers are now merged with the upstream
+    kernel. Only saa717x.ko in the i2c-drivers sub-directory remains.
+  - This should resolve the build issues with 2.6.24+ (Closes: 464112)
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Tue, 27 May 2008 22:33:19 +0100
+
+ivtv (1.0.3-2) unstable; urgency=low
+
+  [ Ian Campbell ]
+  * Vcs-Browser and Vcs-Svn are now supported, so drop the XS- prefix.
+  * Added NEWS note about ivtv-fb -> ivtvfb rename in previous
+    release (Closes: #449208).
+  * Corrected the name of debian/NEWS so it actually gets installed and
+    include NEWS and ChangeLogs in modules packages.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Sun, 04 Nov 2007 18:12:50 +0000
+
+ivtv (1.0.3-1) unstable; urgency=low
+
+  [ Ian Campbell ]
+  * New upstream release
+    - Updated 01_no-depmod.dpatch and 04_r4006-improve-detection.dpatch.
+  * Do not install ivtv.h to /usr/include/linux/. Nothing requires it and
+    now that the driver is in the upstream kernel the header should come
+    from there anyway, new patch is
+    03_do-not-install-ivtv.h.dpatch. (Closes: #443165)
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Thu, 18 Oct 2007 20:24:04 +0100
+
+ivtv (1.0.2-2) unstable; urgency=low
+
+  [ Ian Campbell ]
+  * Improve 04_r4006-improve-detection.dpatch the kernel with the changed ABI
+    has bumped the ABI version to 2 so we can reliably detect based on that.
+  * Remove drivers/Module.symvers before build since a stray file from a
+    previous build can cause us to build unusable modules. (Closes: #440519)
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Sun, 02 Sep 2007 16:32:10 +0100
+
+ivtv (1.0.2-1) unstable; urgency=low
+
+  [ Ian Campbell ]
+  * New upstream release.
+  * Add r4006 from upstream SVN to fix build issues with 2.6.22.2+.
+    s/`uname -r`/$(KVER)/ in the patch to work with our rules file.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Wed, 22 Aug 2007 08:01:03 +0100
+
+ivtv (1.0.1-1) experimental; urgency=low
+
+  [ Ian Campbell ]
+  * New upstream release (Closes: #437298).
+    - generated module package fails to build on 2.6.22.x (Closes:
+    #438783)
+
+  [ Mark Purcell ]
+  * debian/rules [ ! -f Makefile ] || : lintian debian-rules-ignores-
+    make-clean-error
+
+ -- Mark Purcell <msp@debian.org>  Sun, 19 Aug 2007 18:52:31 +0100
+
+ivtv (0.10.5-2) unstable; urgency=low
+
+  * Only build try and build a module package if CONFIG_PCI is set in the
+    kernel configuration. If CONFIG_PCI is not enabled print a nice
+    message instead of #erroring during the build. (Closes: #431411)
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Thu, 26 Jul 2007 19:45:06 +0100
+
+ivtv (0.10.5-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Mark Purcell <msp@debian.org>  Thu, 19 Jul 2007 18:26:57 +0100
+
+ivtv (0.10.4-1) UNRELEASED; urgency=low
+
+  * New upstream release.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Tue, 17 Jul 2007 19:21:11 +0100
+
+ivtv (0.10.3-1) unstable; urgency=low
+
+  * New upstream release
+  * Update debian/control
+    - update IVTV project description (Closes: #426187)
+
+ -- Mark Purcell <msp@debian.org>  Fri, 08 Jun 2007 23:37:57 +0100
+
+ivtv (0.10.2-1) unstable; urgency=low
+
+  [ Ian Campbell ]
+  * New upstream release.
+  * The ivtv-firmware package has been integrated with firmware-nonfree
+    package. The naming scheme for firmware-nonfree requires the package to be
+    firmware-ivtv so update debian/control.modules.in and debian/README.Debian
+    accordingly.
+
+  [ Mark Purcell ]
+  * debian/control cleanup 0.8, add other PVR models and upstream URL
+  * Recommends: firmware-ivtv | ivtv-firmware (first is given priority)
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Thu, 17 May 2007 21:01:33 +0100
+
+ivtv (0.10.1-2) unstable; urgency=low
+
+  [ Ian Campbell ]
+  * Update debian/control to corectly refer to the 0.10 release.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Fri, 30 Mar 2007 07:29:03 +0100
+
+ivtv (0.10.1-1) experimental; urgency=low
+
+  * New upstream release
+    - create seperate source packages for different major releases
+    (Closes: #380155)
+    - FTBFS with 2.6.19-1 (Closes: #405561)
+  * Build without debian/patches/03_include_own_videodev2.h.dpatch
+
+ -- Mark Purcell <msp@debian.org>  Thu, 29 Mar 2007 23:02:56 +0100
+
+ivtv (0.8.2-3) unstable; urgency=low
+
+  [ Ian Campbell ]
+  * Stop providing ivtv-make-fwpkg now that we have ivtv-firmware in
+    non-free. Drop depends on wget and tar.
+  * Move unzip from Depends to Recommends since it is only needed for
+    ivtvfwextract and we have ivtv-firmware packaged now.  
+  * Miscellaneous cleanups
+    - Update debhelper build depends in control.modules.in to match
+      control and fix type (s/familiy/family).
+    - Trailing whitespace.
+    - Added .dpatch suffix to debian/patches/03_include_own_videodev2.h
+    - Removed unused MAKE_DEBIAN_KPKG from debian/rules.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Sun, 18 Feb 2007 10:54:06 +0000
+
+ivtv (0.8.2-2) unstable; urgency=low
+
+  * Backout help2man as it was causing buildd segfaults :-(
+
+ -- Mark Purcell <msp@debian.org>  Sun,  7 Jan 2007 14:21:09 +0000
+
+ivtv (0.8.2-1) unstable; urgency=low
+
+  [ Ian Campbell ]
+  * New upstream release. This release is purely a bugfix release.
+    - Fix handling of corrupt tveeprom.
+    - Fix busy wait loop.
+
+  [ Mark Purcell ]
+  * Remove -source from /usr/src tarball to allow ivtv to ship with
+    linux-modules-contrib-2.6 conglomeration package. (Closes: #362857)
+  * Use help2man for 'easy' man pages ivtv-tune.1 cx25840ctl.1 ivtvctl.1
+
+ -- Mark Purcell <msp@debian.org>  Sun,  7 Jan 2007 12:15:28 +0000
+
+ivtv (0.8.1-2) unstable; urgency=low
+
+  * Remove depmod call from i2c-drivers/Makefile this was causing the modules
+    package build to fail when not run as root.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Sat, 25 Nov 2006 15:41:27 +0000
+
+ivtv (0.8.1-1) unstable; urgency=low
+
+  * New Upstream Release
+
+ -- Mark Purcell <msp@debian.org>  Sat,  4 Nov 2006 11:45:10 +0000
+
+ivtv (0.8.0-1) unstable; urgency=low
+
+  * New upstream release. (Closes: #393689)
+  * Use a local copy of videodev2.h instead of the current kernel's copy.
+  * All of the firmware related documentation has been removed upstream
+    because it has been added to the kernel source so stop trying to include
+    it in our package.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Tue, 24 Oct 2006 19:04:02 +0100
+
+ivtv (0.7.1-1) unstable; urgency=low
+
+  [ Ian Campbell ]
+  * Update control.modules.in to refer to version 0.7 not 0.6.
+    Thanks to Michael Prokop for noticing (Closes: #389368)
+  * Updated debian/rules to cope with ~ in version.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Wed, 27 Sep 2006 10:50:56 +0100
+
+ivtv (0.7.0-2) unstable; urgency=low
+
+  [ Ian Campbell ]
+  * Remove /lib/modules/*/*.modules which are sometimes created during the
+    module build. These are unecessary since they are created by depmod at
+    install time. (Closes: 385454)
+
+  [ Mark Purcell ]
+  * broken tar command in debian/rules (Closes: #387162)
+
+ -- Mark Purcell <msp@debian.org>  Tue, 12 Sep 2006 21:24:31 +0100
+
+ivtv (0.7.0-1) unstable; urgency=low
+
+  * New upstream release.
+  * Remove postinst.modules.in since dh_installmodules already adds
+    the required depmod calls and there is nothing else left.
+    (Closes: #381752)
+  * Patch 50_non-i386-portability has been merged upstream.
+  * Update debian/watch for 0.7.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Tue, 22 Aug 2006 08:33:01 +0100
+
+ivtv (0.6.3-2) unstable; urgency=low
+
+  [ Ian Campbell ]
+  * ivtv-source should depend on dpatch. (Closes: #379377)
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Wed, 26 Jul 2006 11:19:18 +0100
+
+ivtv (0.6.3-1) unstable; urgency=low
+
+  * New upstream release
+  * Replaced 50_non-i386-portability.dpatch with an improved version which has
+    been submitted upstream. This also removes the need to set $(X86) in
+    debian/rules.
+  * 75_svn-r3315-initialize-struct-v4l2_frequency.type.dpatch was included
+    upstream.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Sat,  1 Jul 2006 09:52:06 +0100
+
+ivtv (0.6.2-2) unstable; urgency=low
+
+  [ Ian Campbell ]
+  * Convert build system to use dpatch.
+  * Nothing uses $DEBIAN_MAKE_KPKG anymore so drop it.
+  * Backport r3315 from svn which ensures that the tuner is set to TV and not
+    radio mode when initializing the driver.
+  * Depend on debhelper >= 5.0.37 so as to get a dh_installmodules which
+    calls depmod correctly for us. Add call to dh_installmodules when
+    building module packages.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Mon, 12 Jun 2006 07:12:56 +0100
+
+ivtv (0.6.2-1) unstable; urgency=low
+
+  * New upstream release.
+  * Standards-Version: 3.7.2 (no packaging changes required).
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Mon,  8 May 2006 07:31:13 +0100
+
+ivtv (0.6.1-5) unstable; urgency=low
+
+  * Only compile ivtvplay, ivtv-mpegindex and ivtv-encoder on i386 and amd64
+    since they use non-portable assembly code.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Thu,  4 May 2006 19:56:38 +0100
+
+ivtv (0.6.1-4) unstable; urgency=low
+
+  [ Ian Campbell ]
+  * Change package to Priority: extra rather than Priority: optional due to
+    dependency on packages which are Priority: extra, as per policy section
+    2.5.
+  * Fix spelling error in ivtv-utils description. (Closes: #363981)
+  * Do not attempt to clean the drivers subdirectory during a normal
+    build. The files in this directory are only built by kdist_modules and
+    so should only be cleaned by kdist_clean. The necessary kernel files
+    may not be present when running the regular clean target leading to
+    FTBFS problems.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Fri, 14 Apr 2006 18:12:29 +0100
+
+ivtv (0.6.1-3) unstable; urgency=low
+
+  * Disable prep DMA (-p) option to ivtvfbctl on non-i386 architectures since
+    it uses i386 specific inline assembly. (Closes: #361195)
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Wed, 12 Apr 2006 22:12:06 +0100
+
+ivtv (0.6.1-2) unstable; urgency=low
+
+  [ Ian Campbell ]
+  * Undo changes to use own copy of videodev2.h. Upstream release 0.6.1
+    included a fix for this problem.
+  * Remove incorrect provides/conflicts from the modules package left over
+    from non-Debian packaging.
+  * Move to contrib as the driver requires non-free firmware.
+    (Closes: #360539)
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Tue,  4 Apr 2006 07:13:51 +0100
+
+ivtv (0.6.1-1) unstable; urgency=low
+
+  * New Upstream Release
+  * Update debian/watch for svn-upgrade
+  * Inject to svn.debian.org
+  * Maintainer: Debian MythTV Team <pkg-mythtv-
+    maintainers@lists.alioth.debian.org>
+  * Initial release. (Closes: #308692: ITP: ivtv -- drivers and
+    utilities for WinTV PVR cards - Debian Bug report logs)
+
+ -- Mark Purcell <msp@debian.org>  Mon, 27 Mar 2006 22:23:16 +0100
+
+ivtv (0.6.0-1) unstable; urgency=low
+
+  [ Ian Campbell ]
+  * New upstream release.
+  * Include own copy of linux/videodev2.h until the one in
+    /usr/include/linux is updated.
+  * Rearrange -source package to include the i2c-drivers subdirectory.
+  * Updated debian/copyright.
+  * Included doc/* in the -utils package.
+
+  [ Mark Purcell ]
+  * Initial release. (Closes: #308692: ITP: ivtv -- drivers and
+    utilities for WinTV PVR cards - Debian Bug report logs)
+  * fix old-fsf-address-in-copyright-file
+
+ -- Mark Purcell <msp@debian.org>  Fri, 24 Mar 2006 20:46:08 +0000
+
+ivtv (0.4.3-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Thu, 16 Feb 2006 07:07:33 +0000
+
+ivtv (0.4.2-2) unstable; urgency=low
+
+  * Fix bugs reported by Abram Catalano:
+    - Change references to DEBMAIL in docs to DEBEMAIL.
+    - ivtv-utils package Recommends devscripts (for debuild) and fakeroot.
+  
+ -- Ian Campbell <ijc@hellion.org.uk>  Wed,  1 Feb 2006 06:59:50 +0000
+
+ivtv (0.4.2-1) unstable; urgency=low
+
+  * New upstream release.
+  * Install firmware mpg image as v4l-cx2341x-init.mpg, keep the old name
+    as a symbolic link for backwards compatability.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Sun, 15 Jan 2006 20:45:58 +0000
+
+ivtv (0.4.1-3) unstable; urgency=low
+
+  * Fix Makefile2.4. Kbuild needs to be included for both branches of the
+    ifneq.
+  * Changes to ivtv-make-fwpkg suggested by Marc Haber:
+    - The environment variable containing the maintainer e-mail address
+      is DEBEMAIL, not DEBMAIL.
+    - The build script should abort after "Failed to determine firmware
+      version."
+    - The build script should abort with a clear error message if no
+      firmware file is given on the command line.
+  * Remove a skeletal ivtvctl.1 manpage from the diff, it's worse than
+    useless.
+  * Clean in driver directory.
+  * IVTV_WARN is no longer used.
+  
+ -- Ian Campbell <ijc@hellion.org.uk>  Tue,  3 Jan 2006 20:59:52 +0000
+
+ivtv (0.4.1-2) unstable; urgency=low
+
+  * Fix kernel version suffix support by forcing KERNELVERSION when make
+    is called.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Mon,  2 Jan 2006 10:51:30 +0000
+
+ivtv (0.4.1-1) unstable; urgency=low
+
+  * New upstream release.
+  * driver/compat.h was moved to driver/ivtv-compat.h so move the changes
+    over. Also include videodev.h in the module source package.
+  * Handle renames in debian/rules and debian/ivtv-make-fwpkg.in:
+    - ivtv-fw-enc.bin    -> v4l-cx2341x-enc.fw
+    - ivtv-fw-dec.bin    -> v4l-cx2341x-dec.fw
+    - HcwMakoA.ROM       -> v4l-cx25840.fw
+    - ivtv_init_mpeg.bin -> v4l-cx2341x-init-mpeg.bin
+  * Remove depmod call from driver/Makefile, postinst will do it instead.
+  
+ -- Ian Campbell <ijc@hellion.org.uk>  Wed, 21 Dec 2005 21:39:23 +0000
+
+ivtv (0.4.0-1) unstable; urgency=low
+
+  * New upstream release.
+  * ptune.pl and ptune-ui.pl have been moved to utils/perl, update
+    debian/rules accordingly.
+  * utils/ now installs to $(PREFIX) not $(INSTALLDIR)
+  * Include ivtv_init_mpeg.bin in utils package. I'm unsure how this will
+    work with a 2.4 kernel -- a symbolic link from /lib/modules might also
+    be required.
+  * Added IVTV_WARN in driver/ivtv-driver.h
+  
+ -- Ian Campbell <ijc@hellion.org.uk>  Mon, 10 Oct 2005 14:09:25 +0100
+
+ivtv (0.3.8-1) unstable; urgency=low
+
+  * New upstream release.
+  * Update debian/watch and debian/copyright for move to
+    http://dl.ivtvdriver.org/ivtv/
+  * Hack compat.h so that videodev.2h can be included in the
+    ivtv-source package.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Tue, 13 Sep 2005 19:11:51 +0100
+
+ivtv (0.3.7k-2) unstable; urgency=low
+
+  * Fix module loader support by passing HP_FWLOAD=1 when building
+    modules.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Sun, 28 Aug 2005 16:27:53 +0100
+
+ivtv (0.3.7k-1) unstable; urgency=low
+
+  * New upstream release.
+  * Added ivtv-make-fwpkg script.
+  
+ -- Ian Campbell <ijc@hellion.org.uk>  Sun, 28 Aug 2005 15:35:20 +0100
+
+ivtv (0.3.7i-1) unstable; urgency=low
+
+  * New upstream release.
+  * Rework package versioning to make it suitable for a Debian upload.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Tue, 23 Aug 2005 21:35:33 +0100
+
+ivtv (0.3.7b-1) unstable; urgency=low
+
+  * New upstream release.
+  * Add dependency on unzip for ivtvfwextract.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Mon,  8 Aug 2005 11:37:01 +0100
+
+ivtv (0.3.7a-2) unstable; urgency=low
+
+  * Added ptune and ptune-ui to the utils packages.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Sun, 31 Jul 2005 15:50:22 +0100
+
+ivtv (0.3.7a-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Mon, 25 Jul 2005 12:42:17 +0100
+
+ivtv (0.3.7-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Wed, 20 Jul 2005 06:48:25 +0100
+
+ivtv (0.3.6z-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Fri, 15 Jul 2005 08:56:32 +0100
+
+ivtv (0.3.6x-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Tue,  5 Jul 2005 20:24:12 +0100
+
+ivtv (0.3.6p-2) unstable; urgency=low
+
+  * Include the ivtvfwextract.pl script.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Sun, 26 Jun 2005 18:25:08 +0100
+
+ivtv (0.3.6p-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Tue, 21 Jun 2005 09:39:07 +0100
+
+ivtv (0.3.6g-1) unstable; urgency=low
+
+  * New upstream release.
+  * Following Tyler Trafford's lead
+    "Instead of creating 'module_install' I moved the header file
+     installation to utils/Makefile." Also made install-sane just install.
+  * Include /usr/include/linux/ivtv.h
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Fri, 10 Jun 2005 20:09:46 +0100
+
+ivtv (0.3.6b-2) unstable; urgency=low
+
+  * Install a link in /usr/share/modass/packages/ so module assistant support
+    works ok.
+  * Modules packages should be architecture: any
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Tue,  7 Jun 2005 19:28:17 +0100
+
+ivtv (0.3.6b-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Mon,  6 Jun 2005 17:47:50 +0100
+
+ivtv (0.3.5n-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Mon, 30 May 2005 14:53:54 +0100
+
+ivtv (0.3.4w-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Ian Campbell <ijc@hellion.org.uk>  Wed, 18 May 2005 13:32:32 +0100
+
+ivtv (0.3.4q-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Ian Campbell <ijc@cthulhu.hellion.org.uk>  Sun, 15 May 2005 18:35:48 +0100
+
+ivtv (0.3.4j-1) unstable; urgency=low
+
+  * New upstream release.
+
+ -- Ian Campbell <ijc@cthulhu.hellion.org.uk>  Sun, 15 May 2005 18:33:29 +0100
+
+ivtv (0.3.4f-1) unstable; urgency=low
+
+  * Initial Release.
+  * Remove dependency on kernel from utils/Makefile
+  * Separate module and header install in driver/Makefile2.4 and
+    driver/Makefile2.6
+  * Make install-sane target in utils/Makefile actually perform the
+    install instead of just echoing commands.
+  
+ -- Ian Campbell <ijc@hellion.org.uk>  Sun,  8 May 2005 18:59:32 +0100
--- ivtv-utils-1.4.1.orig/debian/ivtv-utils.docs
+++ ivtv-utils-1.4.1/debian/ivtv-utils.docs
@@ -0,0 +1,9 @@
+doc/conversion.txt
+doc/fw-readme.txt
+doc/README.devices
+doc/README.install
+doc/README.lirc
+doc/README.radio
+doc/README.utils
+doc/README.vbi
+doc/video-quality.txt
--- ivtv-utils-1.4.1.orig/debian/rules
+++ ivtv-utils-1.4.1/debian/rules
@@ -0,0 +1,104 @@
+#!/usr/bin/make -f
+# -*- makefile -*-
+
+# Uncomment this to turn on verbose mode.
+#export DH_VERBOSE=1
+
+# This has to be exported to make some magic below work.
+export DH_OPTIONS
+
+include /usr/share/dpatch/dpatch.make
+
+PKG := ivtv
+
+# prefix of the target package name
+PACKAGE=$(PKG)-modules
+
+CFLAGS = -Wall -g
+
+CURDIR := $(shell pwd)
+
+DEB_BUILD_ARCH := $(shell dpkg-architecture -qDEB_BUILD_ARCH)
+
+ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
+	CFLAGS += -O0
+else
+	CFLAGS += -O2
+endif
+
+DEBVERSION:=$(shell head -n 1 debian/changelog \
+                    | sed -e 's/^[^(]*(\([^)]*\)).*/\1/')
+UPVERSION:=$(shell echo $(DEBVERSION) | sed -e 's/^.*://' -e 's/-[0-9.]*\(~unreleased\)\?$$//' -e 's/.dfsg$$//')
+
+FILENAME := ivtv-utils_$(UPVERSION).dfsg.orig.tar.gz
+UPFILENAME := ivtv-utils_$(UPVERSION).orig.tar.gz
+URL := http://dl.ivtvdriver.org/ivtv/stable/ivtv-utils-$(UPVERSION).tar.gz
+
+configure: patch configure-stamp
+configure-stamp:
+	dh_testdir
+	touch configure-stamp
+
+build: build-stamp
+build-stamp: configure
+	$(MAKE) all
+	touch build-arch-stamp
+
+clean: clean-patched unpatch
+clean-patched:
+	dh_testdir
+	dh_testroot
+	rm -f build-arch-stamp configure-stamp
+
+	$(MAKE) clean
+
+	dh_clean 
+
+install:
+	dh_testdir
+	dh_testroot
+	dh_clean -k -s 
+	dh_installdirs -s
+
+	$(MAKE) install DESTDIR=$(CURDIR)/debian/$(PKG)-utils PREFIX=/usr
+	cp utils/ivtvfwextract.pl debian/$(PKG)-utils/usr/bin/ivtvfwextract
+	cp utils/perl/ptune.pl  debian/$(PKG)-utils/usr/bin/ptune
+	cp utils/perl/ptune-ui.pl  debian/$(PKG)-utils/usr/bin/ptune-ui
+
+# Build architecture dependant packages.
+binary-arch: build install
+	dh_testdir -a
+	dh_testroot -a
+	dh_installchangelogs -a ChangeLog
+	dh_installdocs -a
+	dh_installexamples -a
+	dh_installman -a
+	dh_link -a
+	dh_strip -a
+	dh_perl -a
+	dh_compress -a
+	dh_fixperms -a
+	dh_makeshlibs -a
+	dh_installdeb -a
+	dh_shlibdeps -a
+	dh_gencontrol -a
+	dh_md5sums -a
+	dh_builddeb -a
+
+# There are no architecture independant packages.
+binary-indep:
+
+binary: binary-arch binary-indep
+.PHONY: build clean binary-indep binary-arch binary install install-indep install-arch configure
+
+print-version:
+	@@echo "Debian version:          $(DEBVERSION)"
+	@@echo "Upstream version:        $(UPVERSION)"
+
+get-orig-source:
+	@@dh_testdir
+	@@[ -d ../tarballs/. ]||mkdir -p ../tarballs
+	@@echo Downloading $(UPFILENAME) from $(URL) ...
+	@@wget -nv -T10 -t3 -O ../tarballs/$(UPFILENAME) $(URL)
+
+
--- ivtv-utils-1.4.1.orig/debian/patches/00list
+++ ivtv-utils-1.4.1/debian/patches/00list
@@ -0,0 +1,2 @@
+03_do-not-install-ivtv.h.dpatch
+
--- ivtv-utils-1.4.1.orig/debian/patches/03_do-not-install-ivtv.h.dpatch
+++ ivtv-utils-1.4.1/debian/patches/03_do-not-install-ivtv.h.dpatch
@@ -0,0 +1,38 @@
+#! /bin/sh /usr/share/dpatch/dpatch-run
+## 03_do-not-install-ivtv.h.dpatch by Ian Campbell <ijc@hellion.org.uk>
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: Do not install ivtv.h, nothing uses it and it should come from the
+## DP: kernel/linux-libc-dev package anyway.
+
+@DPATCH@
+Index: ivtv-utils-1.3.0/utils/Makefile
+===================================================================
+--- ivtv-utils-1.3.0.orig/utils/Makefile	2008-08-19 21:03:34.000000000 +0100
++++ ivtv-utils-1.3.0/utils/Makefile	2009-04-15 22:30:52.000000000 +0100
+@@ -1,6 +1,5 @@
+ PREFIX = /usr/local
+ BINDIR = $(PREFIX)/bin
+-HDRDIR = /usr/include/linux
+ 
+ X86  := $(shell if echo - | $(CC) -E -dM - | egrep -e '__i386__|__x86_64__' >/dev/null; \
+ 	then echo y; else echo n; fi)
+@@ -13,9 +12,6 @@
+ 
+ BIN := $(EXES) ivtv-tune/ivtv-tune cx25840ctl/cx25840ctl
+ 
+-
+-HEADERS := linux/ivtv.h linux/ivtvfb.h
+-
+ CFLAGS = -D_GNU_SOURCE -O2 -Wall -g -I.
+ CXXFLAGS = $(CFLAGS)
+ 
+@@ -42,8 +38,6 @@
+ 	$(CC) -lpthread -o $@ $^
+ 
+ install: all
+-	install -d $(DESTDIR)/$(HDRDIR)
+-	install -m 0644 $(HEADERS) $(DESTDIR)/$(HDRDIR)
+ 	install -d $(DESTDIR)/$(BINDIR)
+ 	install -m 0755 $(BIN) $(DESTDIR)/$(BINDIR)
+ 
