diff --git a/invest-applet/invest/chart.py b/invest-applet/invest/chart.py
index e13e068..0899351 100644
--- a/invest-applet/invest/chart.py
+++ b/invest-applet/invest/chart.py
@@ -210,7 +210,7 @@ def on_refresh_chart(self, from_timer=False):
 				a += "%s%s," % (name[1:], param)
 		
 		# Create the image URL -----------------------------------------------------
-		chart_base_url = "http://ichart.europe.yahoo.com/z?s=%(s)s&t=%(t)s&q=%(q)s&l=%(l)s&z=%(z)s&p=%(p)s&a=%(a)s%(opt)s"
+		chart_base_url = "http://chart.finance.yahoo.com/z?s=%(s)s&t=%(t)s&q=%(q)s&l=%(l)s&z=%(z)s&p=%(p)s&a=%(a)s%(opt)s"
 		url = chart_base_url % {
 			"s": tickers[0],
 			"t": self.time_ranges[self.ui.get_object("t").get_active()],
diff --git a/invest-applet/invest/quotes.py b/invest-applet/invest/quotes.py
index 44943b1..448c13b 100644
--- a/invest-applet/invest/quotes.py
+++ b/invest-applet/invest/quotes.py
@@ -287,7 +287,7 @@ def populate(self, quotes):
 					self.add_balance_change(balance, change, val["currency"])
 
 				if len(ticker.split('.')) == 2:
-					url = 'http://ichart.europe.yahoo.com/h?s=%s' % ticker
+					url = 'http://chart.finance.yahoo.com/h?s=%s' % ticker
 				else:
 					url = 'http://ichart.yahoo.com/h?s=%s' % ticker
 
