From 4a7f182d70f31ce39f00261b43b02ec6f30962a6 Mon Sep 17 00:00:00 2001
From: Stefano Karapetsas <stefano@karapetsas.com>
Date: Sun, 09 Dec 2012 17:09:39 +0000
Subject: multiload: add GIO in Makefile

---
diff --git a/multiload/Makefile.am b/multiload/Makefile.am
index fd75c8f..c749701 100644
--- a/multiload/Makefile.am
+++ b/multiload/Makefile.am
@@ -5,7 +5,8 @@ INCLUDES =			\
 	-DMULTILOAD_MENU_UI_DIR=\""$(uidir)"\" \
 	$(MATE_APPLETS4_CFLAGS)	\
 	$(GTOP_APPLETS_CFLAGS)	\
-	$(MATEDESKTOP_CFLAGS)
+	$(MATEDESKTOP_CFLAGS)	\
+	$(GIO_CFLAGS)
 
 libexec_PROGRAMS = multiload-applet-2
 
@@ -24,8 +25,9 @@ multiload_applet_2_SOURCES = \
 multiload_applet_2_LDADD = \
 				$(MATE_APPLETS4_LIBS) 	\
 				$(GTOP_APPLETS_LIBS)	\
-	$(MATEDESKTOP_LIBS) \
-	-lm
+				$(MATEDESKTOP_LIBS)		\
+				$(GIO_LIBS)				\
+				-lm
 
 @INTLTOOL_XML_NOMERGE_RULE@
 gsettings_SCHEMAS = org.mate.panel.applet.multiload.gschema.xml
--
cgit v0.9.0.3-65-g4555
