diff --git a/battstat/battstat-upower.c b/battstat/battstat-upower.c
index 36acb40..5592949 100644
--- a/battstat/battstat-upower.c
+++ b/battstat/battstat-upower.c
@@ -86,10 +86,12 @@
   GCancellable *cancellable = g_cancellable_new();
   GError *gerror;
 
+#if !UP_CHECK_VERSION(0, 99, 0)
   if (! up_client_enumerate_devices_sync( upc, cancellable, &gerror ) ) {
     sprintf(error_str, "Unable to enumerate upower devices: %s\n", gerror->message);
     goto error_shutdownclient;
   }
+#endif
 
   g_signal_connect_after( upc, "device-changed", device_cb, NULL );
   g_signal_connect_after( upc, "device-added", device_cb, NULL );
