diff -Nuar gluon-gluon.orig//creator/gluon-creator.desktop gluon-gluon/creator/gluon-creator.desktop
--- gluon-gluon.orig//creator/gluon-creator.desktop	2010-10-21 08:17:39.898000001 -0400
+++ gluon-gluon/creator/gluon-creator.desktop	2010-10-21 08:18:48.950000000 -0400
@@ -53,7 +53,7 @@
 GenericName[pt_BR]=Game Creation Tool
 GenericName[ro]=Game Creation Tool
 GenericName[sv]=Game Creation Tool
-GenericName[tr]=Game Creation Tool
+GenericName[tr]=Oyun Oluşturma Aracı
 GenericName[uk]=Game Creation Tool
 GenericName[x-test]=xxGame Creation Toolxx
 GenericName[zh_CN]=Game Creation Tool
