diff -up qwt-6.0.1/qwtbuild.pri.install-qt qwt-6.0.1/qwtbuild.pri
--- qwt-6.0.1/qwtbuild.pri.install-qt	2011-08-01 09:33:53.000000000 -0500
+++ qwt-6.0.1/qwtbuild.pri	2012-08-14 11:04:38.898276331 -0500
@@ -56,7 +56,7 @@ RCC_DIR      = resources
 
 unix {
 
-    exists( $${QMAKE_LIBDIR_QT}/libqwt.* ) {
+    exists( $${QMAKE_LIBDIR_QT}/libqwt.so ) {
 
         # On some Linux distributions the Qwt libraries are installed 
         # in the same directory as the Qt libraries. Unfortunately
diff -up qwt-6.0.1/qwtconfig.pri.install-qt qwt-6.0.1/qwtconfig.pri
--- qwt-6.0.1/qwtconfig.pri.install-qt	2011-08-01 09:33:53.000000000 -0500
+++ qwt-6.0.1/qwtconfig.pri	2012-08-14 11:04:38.898276331 -0500
@@ -18,6 +18,13 @@ QWT_VERSION      = $${QWT_VER_MAJ}.$${QW
 
 QWT_INSTALL_PREFIX = $$[QT_INSTALL_PREFIX]
 
+CONFIG(install-qt) {
+QWT_INSTALL_DOCS      = $$[QT_INSTALL_DOCS]/html
+QWT_INSTALL_HEADERS   = $$[QT_INSTALL_HEADERS]/qwt
+QWT_INSTALL_LIBS      = $$[QT_INSTALL_LIBS]
+QWT_INSTALL_PLUGINS   = $$[QT_INSTALL_PLUGINS]/designer 
+QWT_INSTALL_FEATURES  = $$[QMAKE_MKSPECS]/features 
+} else {
 unix {
     QWT_INSTALL_PREFIX    = /usr/local/qwt-$$QWT_VERSION
 }
@@ -51,6 +58,7 @@ QWT_INSTALL_PLUGINS   = $${QWT_INSTALL_P
 
 QWT_INSTALL_FEATURES  = $${QWT_INSTALL_PREFIX}/features
 # QWT_INSTALL_FEATURES  = $${QT_INSTALL_PREFIX}/features
+}
 
 ######################################################################
 # Build the static/shared libraries.
diff -up qwt-6.0.1/src/src.pro.install-qt qwt-6.0.1/src/src.pro
--- qwt-6.0.1/src/src.pro.install-qt	2011-08-01 09:33:53.000000000 -0500
+++ qwt-6.0.1/src/src.pro	2012-08-14 11:07:38.498031025 -0500
@@ -16,6 +16,15 @@ include( $${QWT_ROOT}/qwtbuild.pri )
 TEMPLATE          = lib
 TARGET            = $$qtLibraryTarget(qwt)
 
+# pkgconfig support
+CONFIG     += create_pc create_prl no_install_prl
+QMAKE_PKGCONFIG_NAME = qwt
+QMAKE_PKGCONFIG_DESCRIPTION = Qt Widgets for Technical Applications
+QMAKE_PKGCONFIG_LIBDIR = $${QWT_INSTALL_LIBS}
+QMAKE_PKGCONFIG_INCDIR = $${QWT_INSTALL_HEADERS}
+QMAKE_PKGCONFIG_REQUIRES = QtGui QtSvg
+QMAKE_PKGCONFIG_DESTDIR = pkgconfig
+
 DESTDIR           = $${QWT_ROOT}/lib
 
 contains(QWT_CONFIG, QwtDll) {
