diff -up ghostscript-8.64/psi/int.mak.noopt ghostscript-8.64/psi/int.mak
--- ghostscript-8.64/psi/int.mak.noopt	2008-12-26 07:25:31.000000000 +0000
+++ ghostscript-8.64/psi/int.mak	2009-02-04 11:33:48.000000000 +0000
@@ -111,7 +111,7 @@ $(PSOBJ)igc.$(OBJ) : $(PSSRC)igc.c $(GH)
 $(PSOBJ)igcref.$(OBJ) : $(PSSRC)igcref.c $(GH) $(memory__h)\
  $(gsexit_h) $(gsstruct_h)\
  $(iastate_h) $(idebug_h) $(igc_h) $(iname_h) $(ipacked_h) $(store_h)
-	$(PSCC) $(PSO_)igcref.$(OBJ) $(C_) $(PSSRC)igcref.c
+	$(PSCC) -O0 $(PSO_)igcref.$(OBJ) $(C_) $(PSSRC)igcref.c
 
 $(PSOBJ)igcstr.$(OBJ) : $(PSSRC)igcstr.c $(GH) $(memory__h)\
  $(gsmdebug_h) $(gsstruct_h) $(iastate_h) $(igcstr_h)
