diff -up ghostscript-8.64/base/gxclrast.c.pksmraw ghostscript-8.64/base/gxclrast.c
--- ghostscript-8.64/base/gxclrast.c.pksmraw	2008-12-13 20:05:37.000000000 +0000
+++ ghostscript-8.64/base/gxclrast.c	2009-02-04 11:36:25.000000000 +0000
@@ -2685,8 +2685,6 @@ static int apply_create_compositor(gx_de
         rc_increment(tdev);
         *ptarget = tdev;
     }
-    if (code < 0)
-        return code;
 
     /* Perform any updates for the clist device required */
     code = pcomp->type->procs.clist_compositor_read_update(pcomp,
