diff -up ghostscript-8.71/Resource/Init/gs_fonts.ps.runlibfileifexists ghostscript-8.71/Resource/Init/gs_fonts.ps
--- ghostscript-8.71/Resource/Init/gs_fonts.ps.runlibfileifexists	2010-09-13 15:00:22.566812041 +0100
+++ ghostscript-8.71/Resource/Init/gs_fonts.ps	2010-09-13 15:03:24.520814577 +0100
@@ -96,10 +96,19 @@ userdict /Fontmap .FontDirectory maxleng
                 % stack: dict file fontname filename|aliasname
       1 index type /stringtype eq
       1 index type /nametype eq and 1 index xcheck and
-      1 index /run eq 2 index /.runlibfile eq or and {
+      1 index /run eq 2 index /.runlibfile eq 3 index /.runlibfileifexists eq or or and {
                 % This is an inclusion entry.
-        pop findlibfile { exch pop } { file } ifelse
-        2 index exch .readFontmap pop
+        0 index /.runlibfileifexists eq {
+          pop findlibfile {
+            exch pop
+            2 index exch .readFontmap pop
+          } {
+            pop
+          } ifelse
+        } {
+          pop findlibfile { exch pop } { file } ifelse
+          2 index exch .readFontmap pop
+        } ifelse
       } {
                 % This is a real entry.
                 % Read and pop tokens until a semicolon.
diff -up ghostscript-8.64/Resource/Init/gs_init.ps.runlibfileifexists ghostscript-8.64/Resource/Init/gs_init.ps
--- ghostscript-8.64/Resource/Init/gs_init.ps.runlibfileifexists	2009-01-08 09:17:18.000000000 +0000
+++ ghostscript-8.64/Resource/Init/gs_init.ps	2009-02-04 11:35:19.000000000 +0000
@@ -672,6 +672,14 @@ systemdict /internaldict dup .makeintern
 	   { /undefinedfilename signalerror }
 	  ifelse
 	} bind def
+/runlibfileifexists
+	{		% We don't want to bind 'run' into this procedure,
+			% since run may get redefined.
+	  findlibfile
+	   { exch pop /run .systemvar exec }
+	   { pop }
+	  ifelse
+	} bind def
 /selectdevice
 	{ finddevice setdevice .setdefaultscreen } bind def
 /signalerror		% <object> <errorname> signalerror -
@@ -840,6 +848,7 @@ userdict /.currentresourcefile //null pu
 	} bind def
 % Temporarily substitute it for the real runlibfile.
 /.runlibfile /runlibfile load def
+/.runlibfileifexists /runlibfileifexists load def
 /runlibfile /runlibfile0 load def
 
 % Create the error handling machinery.
