diff -urNp ghostscript-9.01.old/ijs/Makefile.am ghostscript-9.01/ijs/Makefile.am
--- ghostscript-9.01.old/ijs/Makefile.am	2011-03-08 10:09:33.055722164 +0000
+++ ghostscript-9.01/ijs/Makefile.am	2011-03-08 10:09:37.390722163 +0000
@@ -24,7 +24,7 @@
 
 ## Process this file with automake to produce Makefile.in.
 
-AUTOMAKE_OPTIONS = 1.6 foreign dist-bzip2 no-dependencies 
+AUTOMAKE_OPTIONS = 1.8 foreign dist-bzip2 no-dependencies 
 
 @SET_MAKE@
 
