diff -up ghostscript-8.70/lib/bdftops.gs-executable ghostscript-8.70/lib/bdftops
--- ghostscript-8.70/lib/bdftops.gs-executable	2007-07-05 20:04:29.000000000 +0100
+++ ghostscript-8.70/lib/bdftops	2009-08-03 17:53:20.569875831 +0100
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -dBATCH -dNODISPLAY -- bdftops.ps "$@"
+exec "$gs" -q -dBATCH -dNODISPLAY -- bdftops.ps "$@"
diff -up ghostscript-8.70/lib/dumphint.gs-executable ghostscript-8.70/lib/dumphint
--- ghostscript-8.70/lib/dumphint.gs-executable	2009-01-04 20:11:57.000000000 +0000
+++ ghostscript-8.70/lib/dumphint	2009-08-03 17:53:20.570875283 +0100
@@ -7,9 +7,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 
 OPTIONS="-dSAFER -dDELAYSAFER"
@@ -27,4 +26,4 @@ if [ $# -ne 1 ]; then
 	exit 1
 fi
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY $OPTIONS -- "`dirname $0`/dumphint.ps" "$1"
+exec "$gs" -q -dNODISPLAY $OPTIONS -- "`dirname $0`/dumphint.ps" "$1"
diff -up ghostscript-8.70/lib/eps2eps.gs-executable ghostscript-8.70/lib/eps2eps
--- ghostscript-8.70/lib/eps2eps.gs-executable	2009-01-10 22:11:18.000000000 +0000
+++ ghostscript-8.70/lib/eps2eps	2009-08-03 17:53:20.573877995 +0100
@@ -7,9 +7,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 OPTIONS="-dDEVICEWIDTH=250000 -dDEVICEHEIGHT=250000"
 while true
@@ -26,4 +25,4 @@ if [ $# -ne 2 ]; then
 	exit 1
 fi
 
-exec "$GS_EXECUTABLE" -q -sDEVICE=epswrite -sstdout=%stderr "-sOutputFile=$2" -dNOPAUSE -dBATCH -dSAFER $OPTIONS "$1"
+exec "$gs" -q -sDEVICE=epswrite -sstdout=%stderr "-sOutputFile=$2" -dNOPAUSE -dBATCH -dSAFER $OPTIONS "$1"
diff -up ghostscript-8.70/lib/font2c.gs-executable ghostscript-8.70/lib/font2c
--- ghostscript-8.70/lib/font2c.gs-executable	2007-07-05 20:04:29.000000000 +0100
+++ ghostscript-8.70/lib/font2c	2009-08-03 17:53:20.576874823 +0100
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY -dWRITESYSTEMDICT -- font2c.ps "$@"
+exec "$gs" -q -dNODISPLAY -dWRITESYSTEMDICT -- font2c.ps "$@"
diff -up ghostscript-8.70/lib/gsbj.gs-executable ghostscript-8.70/lib/gsbj
--- ghostscript-8.70/lib/gsbj.gs-executable	2007-07-05 20:04:29.000000000 +0100
+++ ghostscript-8.70/lib/gsbj	2009-08-03 17:53:20.579874829 +0100
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -sDEVICE=bj10e -r180 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
+exec "$gs" -q -sDEVICE=bj10e -r180 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
diff -up ghostscript-8.70/lib/gsdj500.gs-executable ghostscript-8.70/lib/gsdj500
--- ghostscript-8.70/lib/gsdj500.gs-executable	2007-07-05 20:04:29.000000000 +0100
+++ ghostscript-8.70/lib/gsdj500	2009-08-03 17:53:20.582874574 +0100
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -sDEVICE=djet500 -r300 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
+exec "$gs" -q -sDEVICE=djet500 -r300 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
diff -up ghostscript-8.70/lib/gsdj.gs-executable ghostscript-8.70/lib/gsdj
--- ghostscript-8.70/lib/gsdj.gs-executable	2007-07-05 20:04:29.000000000 +0100
+++ ghostscript-8.70/lib/gsdj	2009-08-03 17:53:20.585875161 +0100
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -sDEVICE=deskjet -r300 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
+exec "$gs" -q -sDEVICE=deskjet -r300 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
diff -up ghostscript-8.70/lib/gslj.gs-executable ghostscript-8.70/lib/gslj
--- ghostscript-8.70/lib/gslj.gs-executable	2007-07-05 20:04:29.000000000 +0100
+++ ghostscript-8.70/lib/gslj	2009-08-03 17:53:20.588874966 +0100
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -sDEVICE=laserjet -r300 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
+exec "$gs" -q -sDEVICE=laserjet -r300 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
diff -up ghostscript-8.70/lib/gslp.gs-executable ghostscript-8.70/lib/gslp
--- ghostscript-8.70/lib/gslp.gs-executable	2007-07-05 20:04:29.000000000 +0100
+++ ghostscript-8.70/lib/gslp	2009-08-03 17:53:20.591874666 +0100
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -sDEVICE=epson -r180 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
+exec "$gs" -q -sDEVICE=epson -r180 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
diff -up ghostscript-8.70/lib/gsnd.gs-executable ghostscript-8.70/lib/gsnd
--- ghostscript-8.70/lib/gsnd.gs-executable	2007-07-05 20:04:29.000000000 +0100
+++ ghostscript-8.70/lib/gsnd	2009-08-03 17:53:20.594874682 +0100
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -dNODISPLAY "$@"
+exec "$gs" -dNODISPLAY "$@"
diff -up ghostscript-8.70/lib/pdf2dsc.gs-executable ghostscript-8.70/lib/pdf2dsc
--- ghostscript-8.70/lib/pdf2dsc.gs-executable	2007-07-05 20:04:29.000000000 +0100
+++ ghostscript-8.70/lib/pdf2dsc	2009-08-03 17:53:20.597874954 +0100
@@ -11,9 +11,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 me=`basename $0`
 
@@ -31,5 +30,5 @@ pdffile=$1
 dscfile=$2
 : ${dscfile:=`echo $pdffile | sed 's,\.[^/.]*,,'`.dsc}
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY -dSAFER -dDELAYSAFER\
+exec "$gs" -q -dNODISPLAY -dSAFER -dDELAYSAFER\
     -sPDFname="$pdffile" -sDSCname="$dscfile" pdf2dsc.ps -c quit
diff -up ghostscript-8.70/lib/pdf2ps.gs-executable ghostscript-8.70/lib/pdf2ps
--- ghostscript-8.70/lib/pdf2ps.gs-executable	2007-07-05 20:04:29.000000000 +0100
+++ ghostscript-8.70/lib/pdf2ps	2009-08-03 17:53:20.601874677 +0100
@@ -7,9 +7,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 OPTIONS=""
 while true
@@ -35,4 +34,4 @@ fi
 # Doing an initial 'save' helps keep fonts from being flushed between pages.
 # We have to include the options twice because -I only takes effect if it
 # appears before other options.
-exec "$GS_EXECUTABLE" $OPTIONS -q -dNOPAUSE -dBATCH -dSAFER -sDEVICE=pswrite "-sOutputFile=$outfile" $OPTIONS -c save pop -f "$1"
+exec "$gs" $OPTIONS -q -dNOPAUSE -dBATCH -dSAFER -sDEVICE=pswrite "-sOutputFile=$outfile" $OPTIONS -c save pop -f "$1"
diff -up ghostscript-8.70/lib/pdfopt.gs-executable ghostscript-8.70/lib/pdfopt
--- ghostscript-8.70/lib/pdfopt.gs-executable	2008-05-25 03:17:14.000000000 +0100
+++ ghostscript-8.70/lib/pdfopt	2009-08-03 17:53:20.604874898 +0100
@@ -7,9 +7,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 OPTIONS="-dSAFER -dDELAYSAFER"
 while true
@@ -26,4 +25,4 @@ if [ $# -ne 2 ]; then
 	exit 1
 fi
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY $OPTIONS -- pdfopt.ps "$1" "$2"
+exec "$gs" -q -dNODISPLAY $OPTIONS -- pdfopt.ps "$1" "$2"
diff -up ghostscript-8.70/lib/pf2afm.gs-executable ghostscript-8.70/lib/pf2afm
--- ghostscript-8.70/lib/pf2afm.gs-executable	2007-07-05 20:04:29.000000000 +0100
+++ ghostscript-8.70/lib/pf2afm	2009-08-03 17:53:20.608874740 +0100
@@ -10,8 +10,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY -dSAFER -dDELAYSAFER  -- pf2afm.ps "$@"
+exec "$gs" -q -dNODISPLAY -dSAFER -dDELAYSAFER  -- pf2afm.ps "$@"
diff -up ghostscript-8.70/lib/pfbtopfa.gs-executable ghostscript-8.70/lib/pfbtopfa
--- ghostscript-8.70/lib/pfbtopfa.gs-executable	2007-07-05 20:04:29.000000000 +0100
+++ ghostscript-8.70/lib/pfbtopfa	2009-08-03 17:53:20.612874691 +0100
@@ -7,9 +7,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 if [ $# -eq 2 ] 
 then
@@ -22,4 +21,4 @@ else
     exit 1
 fi
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY -- pfbtopfa.ps "$1" "$outfile"
+exec "$gs" -q -dNODISPLAY -- pfbtopfa.ps "$1" "$outfile"
diff -up ghostscript-8.70/lib/pphs.gs-executable ghostscript-8.70/lib/pphs
--- ghostscript-8.70/lib/pphs.gs-executable	2007-07-05 20:04:29.000000000 +0100
+++ ghostscript-8.70/lib/pphs	2009-08-03 17:53:20.616874647 +0100
@@ -9,8 +9,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY -- pphs.ps "$@"
+exec "$gs" -q -dNODISPLAY -- pphs.ps "$@"
diff -up ghostscript-8.70/lib/printafm.gs-executable ghostscript-8.70/lib/printafm
--- ghostscript-8.70/lib/printafm.gs-executable	2007-07-05 20:04:29.000000000 +0100
+++ ghostscript-8.70/lib/printafm	2009-08-03 17:53:20.619874766 +0100
@@ -9,8 +9,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY -- printafm.ps "$@"
+exec "$gs" -q -dNODISPLAY -- printafm.ps "$@"
diff -up ghostscript-8.70/lib/ps2epsi.gs-executable ghostscript-8.70/lib/ps2epsi
--- ghostscript-8.70/lib/ps2epsi.gs-executable	2009-04-07 11:20:02.000000000 +0100
+++ ghostscript-8.70/lib/ps2epsi	2009-08-03 17:54:48.355750266 +0100
@@ -6,9 +6,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 # try to create a temporary file securely
 if test -z "$TMPDIR"; then
@@ -50,7 +49,7 @@ else
 	outfile=$2
 fi
 
-"$GS_EXECUTABLE" -q -dBATCH -dNOPAUSE -dSAFER -dDELAYSAFER -sDEVICE=bbox -sOutputFile=/dev/null "${infile}" 2>${outfile}
+"$gs" -q -dBATCH -dNOPAUSE -dSAFER -dDELAYSAFER -sDEVICE=bbox -sOutputFile=/dev/null "${infile}" 2>${outfile}
 
 ls -l "${infile}" |
 awk 'F==1	{
@@ -92,7 +91,7 @@ awk 'F==1	{
 		}
 	' F=1 - F=2 "${outfile}" >>"$tmpfile"
 
-"$GS_EXECUTABLE" -q -dNOPAUSE -dSAFER -dDELAYSAFER -r72 -sDEVICE=bit -sOutputFile=/dev/null "$tmpfile" ps2epsi.ps "$tmpfile" <"${infile}" 1>&2
+"$gs" -q -dNOPAUSE -dSAFER -dDELAYSAFER -r72 -sDEVICE=bit -sOutputFile=/dev/null "$tmpfile" ps2epsi.ps "$tmpfile" <"${infile}" 1>&2
 rm -f "$tmpfile"
 rm -rf "$tmpdir"
 
diff -up ghostscript-8.70/lib/ps2pdfwr.gs-executable ghostscript-8.70/lib/ps2pdfwr
--- ghostscript-8.70/lib/ps2pdfwr.gs-executable	2009-01-10 22:11:18.000000000 +0000
+++ ghostscript-8.70/lib/ps2pdfwr	2009-08-03 17:53:20.626874926 +0100
@@ -7,9 +7,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 OPTIONS="-dSAFER"
 while true
@@ -42,4 +41,4 @@ fi
 
 # We have to include the options twice because -I only takes effect if it
 # appears before other options.
-exec "$GS_EXECUTABLE" $OPTIONS -q -dNOPAUSE -dBATCH -sDEVICE=pdfwrite -sstdout=%stderr "-sOutputFile=$outfile" $OPTIONS -c .setpdfwrite -f "$infile"
+exec "$gs" $OPTIONS -q -dNOPAUSE -dBATCH -sDEVICE=pdfwrite -sstdout=%stderr "-sOutputFile=$outfile" $OPTIONS -c .setpdfwrite -f "$infile"
diff -up ghostscript-8.70/lib/ps2ps.gs-executable ghostscript-8.70/lib/ps2ps
--- ghostscript-8.70/lib/ps2ps.gs-executable	2009-01-10 22:11:18.000000000 +0000
+++ ghostscript-8.70/lib/ps2ps	2009-08-03 17:53:20.630875037 +0100
@@ -7,9 +7,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 OPTIONS="-dSAFER"
 while true
@@ -27,4 +26,4 @@ if [ $# -ne 2 ]; then
 	exit 1
 fi
 
-exec "$GS_EXECUTABLE" -q -sDEVICE=pswrite -sstdout=%stderr "-sOutputFile=$2" -dNOPAUSE -dBATCH $OPTIONS "$1"
+exec "$gs" -q -sDEVICE=pswrite -sstdout=%stderr "-sOutputFile=$2" -dNOPAUSE -dBATCH $OPTIONS "$1"
diff -up ghostscript-8.70/lib/wftopfa.gs-executable ghostscript-8.70/lib/wftopfa
--- ghostscript-8.70/lib/wftopfa.gs-executable	2007-07-05 20:04:29.000000000 +0100
+++ ghostscript-8.70/lib/wftopfa	2009-08-03 17:53:20.633874780 +0100
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY -- wftopfa.ps "$@"
+exec "$gs" -q -dNODISPLAY -- wftopfa.ps "$@"
