diff -up ghostscript-9.05/cups/cups.mak.cups-filters ghostscript-9.05/cups/cups.mak
--- ghostscript-9.05/cups/cups.mak.cups-filters	2012-02-08 08:48:50.000000000 +0000
+++ ghostscript-9.05/cups/cups.mak	2012-02-09 11:42:22.211731587 +0000
@@ -56,9 +56,9 @@ install-cups: cups
 	    $(INSTALL_PROGRAM) $(GSTORASTER_XE) $(DESTDIR)$(CUPSSERVERBIN)/filter; \
 	fi
 	$(INSTALL_PROGRAM) cups/gstopxl $(DESTDIR)$(CUPSSERVERBIN)/filter
-	-mkdir -p $(DESTDIR)$(CUPSSERVERROOT)
+	-mkdir -p $(DESTDIR)$(CUPSDATA)/mime
 	if [ "$(CUPSPDFTORASTER)" = "1" ]; then \
-	    $(INSTALL_DATA) cups/gstoraster.convs $(DESTDIR)$(CUPSSERVERROOT); \
+	    $(INSTALL_DATA) cups/gstoraster.convs $(DESTDIR)$(CUPSDATA)/mime; \
 	fi
 	-mkdir -p $(DESTDIR)$(CUPSDATA)/model
 	$(INSTALL_DATA) cups/pxlcolor.ppd $(DESTDIR)$(CUPSDATA)/model
