diff -up ghostscript-9.02/cups/cups.mak.cups-filters ghostscript-9.02/cups/cups.mak
--- ghostscript-9.02/cups/cups.mak.cups-filters	2011-02-28 22:31:28.000000000 +0000
+++ ghostscript-9.02/cups/cups.mak	2011-04-04 12:39:23.691844891 +0100
@@ -56,10 +56,8 @@ install-cups: cups
 	    $(INSTALL_PROGRAM) $(GSTORASTER_XE) $(DESTDIR)$(CUPSSERVERBIN)/filter; \
 	fi
 	$(INSTALL_PROGRAM) cups/pstopxl $(DESTDIR)$(CUPSSERVERBIN)/filter
-	-mkdir -p $(DESTDIR)$(CUPSSERVERROOT)
-	if [ "$(CUPSPDFTORASTER)" = "1" ]; then \
-	    $(INSTALL_DATA) cups/gstoraster.convs $(DESTDIR)$(CUPSSERVERROOT); \
-	fi
+	-mkdir -p $(DESTDIR)$(CUPSDATA)/mime
+	$(INSTALL_DATA) cups/gstoraster.convs $(DESTDIR)$(CUPSDATA)/mime
 	-mkdir -p $(DESTDIR)$(CUPSDATA)/model
 	$(INSTALL_DATA) cups/pxlcolor.ppd $(DESTDIR)$(CUPSDATA)/model
 	$(INSTALL_DATA) cups/pxlmono.ppd $(DESTDIR)$(CUPSDATA)/model
