--- src/config.cpp.orig	2004-09-27 12:35:48.000000000 +0300
+++ src/config.cpp	2008-04-21 23:38:01.000000000 +0300
@@ -29,6 +29,8 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <cstring>
+
 #include "config.h"
 #include "init.h"
 
--- src/enemy.cpp.orig	2004-09-27 12:28:22.000000000 +0300
+++ src/enemy.cpp	2008-04-21 23:37:56.000000000 +0300
@@ -28,6 +28,8 @@
 *************************************************************************/
 
 #include <stdio.h>
+#include <cstring>
+
 #include "SDL.h"
 #include "SDL_opengl.h"
 #include "SDL_image.h"
