Index: kdebase-runtime-4.4.80/knetattach/knetattach.cpp
===================================================================
--- kdebase-runtime-4.4.80.orig/knetattach/knetattach.cpp	2010-02-24 07:52:17.000000000 -0500
+++ kdebase-runtime-4.4.80/knetattach/knetattach.cpp	2010-05-24 14:48:49.046936849 -0400
@@ -176,7 +176,7 @@
       KConfig config("kio_fishrc");
       KConfigGroup cg(&config, _host->text().trimmed());
       cg.writeEntry("Charset", KGlobal::charsets()->encodingForName(_encoding->currentText()));
-	  url.setProtocol("fish");
+	  url.setProtocol("sftp");
 	  url.setPort(_port->value());
       } else if (_type == "FTP") {
 	  url.setProtocol("ftp");
