Index: kde-runtime-4.7.1/knetattach/knetattach.cpp
===================================================================
--- kde-runtime-4.7.1.orig/knetattach/knetattach.cpp
+++ kde-runtime-4.7.1/knetattach/knetattach.cpp
@@ -49,7 +49,7 @@ KNetAttach::KNetAttach( QWidget* parent
     connect(_createIcon, SIGNAL(toggled(bool)), this, SLOT(updateFinishButtonText(bool)));
     connect( this, SIGNAL(helpRequested()), this, SLOT( slotHelpClicked() ) );
     connect( this, SIGNAL(currentIdChanged(int)), this, SLOT( slotPageChanged(int) ) );
-    setWindowIcon(KIcon("knetattach"));
+    setWindowIcon(KIcon("folder-new"));
     setOption(HaveHelpButton, true);
     //setResizeMode(Fixed); FIXME: make the wizard fixed-geometry
     button(FinishButton)->setEnabled(false);
Index: kde-runtime-4.7.1/knetattach/knetattach.desktop
===================================================================
--- kde-runtime-4.7.1.orig/knetattach/knetattach.desktop
+++ kde-runtime-4.7.1/knetattach/knetattach.desktop
@@ -179,7 +179,7 @@ GenericName[zh_CN]=网络文件夹向导
 GenericName[zh_TW]=網路資料夾精靈
 Exec=knetattach
 InitialPreference=6
-Icon=knetattach
+Icon=folder-new
 Type=Application
 Terminal=false
 X-KDE-StartupNotify=true
