diff --git a/Makefile b/Makefile
index 1b40530..dfaf3f7 100644
--- a/Makefile
+++ b/Makefile
@@ -12,12 +12,12 @@ VERSION := $(shell awk '/Version:/ { print $$2 }' python-pyblock.spec)
 RELEASE := $(shell awk -F '[ %]' '/Release:/ { print $$2 }' python-pyblock.spec)
 USESELINUX = 1
 
-dm_LIBS = dmraid devmapper
+dm_LIBS = devmapper python2.7
 ifeq (1, $(USESELINUX))
 dm_LIBS += selinux
 CFLAGS += -DUSESELINUX=1
 endif
-dmraid_LIBS = dmraid devmapper
+dmraid_LIBS = dmraid python2.7
 
 PYFILES=__init__.py maps.py device.py
 LIBS = dmmodule.so dmraidmodule.so
