diff -ur libgdata-0.13.3-orig/Makefile.am libgdata-0.13.3/Makefile.am
--- libgdata-0.13.3-orig/Makefile.am	2013-02-26 01:53:38.000000000 +0200
+++ libgdata-0.13.3/Makefile.am	2013-03-10 21:33:36.847566915 +0200
@@ -539,11 +539,6 @@
 endif
 endif
 
-# Code coverage
-@GNOME_CODE_COVERAGE_RULES@
-CODE_COVERAGE_DIRECTORY = $(top_builddir)/gdata
-CODE_COVERAGE_LCOV_OPTIONS = --base-directory $(abs_top_srcdir)
-
 # Demo programs
 noinst_PROGRAMS =
 
diff -ur libgdata-0.13.3-orig/Makefile.in libgdata-0.13.3/Makefile.in
--- libgdata-0.13.3-orig/Makefile.in	2013-02-26 02:05:28.000000000 +0200
+++ libgdata-0.13.3/Makefile.in	2013-03-10 21:33:58.042566181 +0200
@@ -3350,9 +3350,6 @@
 @ENABLE_VAPIGEN_TRUE@@HAVE_INTROSPECTION_TRUE@		echo $$pkg >> $@; \
 @ENABLE_VAPIGEN_TRUE@@HAVE_INTROSPECTION_TRUE@	done
 
-# Code coverage
-@GNOME_CODE_COVERAGE_RULES@
-
 # ChangeLog
 ChangeLog: $(srcdir)/ChangeLog
 $(srcdir)/ChangeLog:
