diff -Nur cddlib-094f-old/Makefile.am cddlib-094f/Makefile.am
--- cddlib-094f-old/Makefile.am	2009-09-03 19:07:05.815018166 +0300
+++ cddlib-094f/Makefile.am	2009-09-03 19:07:53.261928175 +0300
@@ -16,5 +16,5 @@
 #---------------------------------------------
 # End of site customization.
 ##############################################
-SUBDIRS      = lib-src src lib-src-gmp src-gmp
+SUBDIRS      = lib-src lib-src-gmp
 EXTRA_DIST   = examples/*.* examples-ine/*.* examples-ine3d/*.* examples-ml/*.* examples-ext/*.*
