--- a/aclocal.m4
+++ b/aclocal.m4
@@ -234,7 +234,7 @@ AC_DEFUN([AF_WITH_SYSTEMD],
 [if test "$withval" = yes; then
   if test -z "$systemddir"; then
     AC_MSG_CHECKING([location of the systemd unit files directory])
-    for systemd_d in /lib/systemd/system; do
+    for systemd_d in /usr/lib/systemd/system /usr/lib64/systemd/system /lib/systemd/system /lib64/systemd/system; do
       if test -z "$systemddir"; then
         if test -d "$systemd_d"; then
           systemddir="$systemd_d"
diff --git a/configure b/configure
index ba3bba6..3722a46 100755
--- a/configure
+++ b/configure
@@ -2157,7 +2157,7 @@ if test "${with_systemd+set}" = set; then :
   if test -z "$systemddir"; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking location of the systemd unit files directory" >&5
 $as_echo_n "checking location of the systemd unit files directory... " >&6; }
-    for systemd_d in /lib/systemd/system; do
+    for systemd_d in /usr/lib/systemd/system /usr/lib64/systemd/system /lib/systemd/system /lib64/systemd/system; do
       if test -z "$systemddir"; then
         if test -d "$systemd_d"; then
           systemddir="$systemd_d"