--- a/redhat/autofs.init.in
+++ b/redhat/autofs.init.in
@@ -167,6 +167,19 @@ function usage_message() {
 
 RETVAL=0
 
+# allow non-root users to read status / usage
+
+case "$1" in
+	status)
+		status -p @@autofspiddir@@/autofs.pid -l autofs $prog
+		exit 0;
+		;;
+	usage)
+		usage_message
+		exit 0;
+		;;
+esac
+
 # Only the root user may change the service status
 if [ `id -u` -ne 0 ] && [ "$1" != "status" ]; then
 	echo "insufficient privilege to change service status"
@@ -184,9 +197,6 @@ case "$1" in
 	stop)
 		stop
 		;;
-	status)
-		status -p @@autofspiddir@@/autofs.pid -l autofs $prog
-		;;
 	restart|force-reload)
 		restart
 		;;
@@ -202,10 +212,6 @@ case "$1" in
 			restart
 		fi
 		;;
-	usage)
-		usage_message
-		exit 0
-		;;
 	*)
 		usage_message
 		exit 2