--- a/modules/mount_nfs.c
+++ b/modules/mount_nfs.c
@@ -263,13 +263,14 @@ int mount_mount(struct autofs_point *ap, const char *root, const char *name, int
 			} else
 				strcpy(loc, n_addr);
 		} else {
-			loc = malloc(strlen(this->name) + strlen(this->path) + 2);
+			char *host = this->name ? this->name : "localhost";
+			loc = malloc(strlen(host) + strlen(this->path) + 2);
 			if (!loc) {
 				char *estr = strerror_r(errno, buf, MAX_ERR_BUF);
 				error(ap->logopt, "malloc: %s", estr);
 				goto forced_fail;
 			}
-			strcpy(loc, this->name);
+			strcpy(loc, host);
 		}
 		strcat(loc, ":");
 		strcat(loc, this->path);