diff -Nuar gnubg.orig//pixmaps/Makefile.am gnubg/pixmaps/Makefile.am
--- gnubg.orig//pixmaps/Makefile.am	2011-02-07 05:50:04.000000000 +0200
+++ gnubg/pixmaps/Makefile.am	2011-02-18 21:53:25.189000125 +0200
@@ -1,5 +1,5 @@
-gnubgiconsdir = $(datadir)
-gnubgpixmapsdir = $(pkgdatadir)/pixmaps
+gnubgiconsdir = /usr/share
+gnubgpixmapsdir = /usr/share/pixmaps
 
 ICONS_16_SCALABLE = icons/hicolor/16x16/apps/scalable/gnubg.svg
 ICONS_16 = icons/hicolor/16x16/apps/gnubg.png
