Index: e2fsprogs-1.41.8/lib/ss/Makefile.in
===================================================================
--- e2fsprogs-1.41.8/lib/ss/Makefile.in
+++ e2fsprogs-1.41.8/lib/ss/Makefile.in	2009-07-15 22:26:05.000000000 +0300
@@ -124,6 +124,10 @@
 	$(Q) $(SUBSTITUTE) $(srcdir)/mk_cmds.sh.in mk_cmds
 	$(Q) $(CHMOD) +x mk_cmds
 
+mk_cmds.install: mk_cmds
+	@perl -ne '/^# BUILD start/ ... /^# BUILD end/ or print' $< > $@
+	@$(CHMOD) +x $@
+
 ss.pc: $(srcdir)/ss.pc.in $(top_builddir)/config.status
 	$(E) "	CONFIG.STATUS $@"
 	$(Q) cd $(top_builddir); CONFIG_FILES=lib/ss/ss.pc ./config.status
@@ -134,7 +138,7 @@
 		$(DESTDIR)$(includedir)/ss $(DESTDIR)$(datadir)/ss \
 		$(DESTDIR)$(bindir) $(DESTDIR)$(libdir)/pkgconfig
 
-install:: libss.a $(INSTALL_HFILES) installdirs ss_err.h mk_cmds ss.pc
+install:: libss.a $(INSTALL_HFILES) installdirs ss_err.h mk_cmds ss.pc mk_cmds.install
 	$(E) "	INSTALL_DATA $(DESTDIR)$(libdir)/libss.a"
 	$(Q) $(INSTALL_DATA) libss.a $(DESTDIR)$(libdir)/libss.a
 	-$(Q) $(RANLIB) $(DESTDIR)$(libdir)/libss.a
@@ -151,7 +155,7 @@
 		$(INSTALL_DATA) $(srcdir)/$$i $(DESTDIR)$(datadir)/ss/$$i; \
 	done
 	$(E) "	INSTALL $(bindir)/mk_cmds"
-	$(Q) $(INSTALL) mk_cmds $(DESTDIR)$(bindir)/mk_cmds
+	$(Q) $(INSTALL) mk_cmds.install $(DESTDIR)$(bindir)/mk_cmds
 	$(E) "	INSTALL_DATA $(man1dir)/mk_cmds.1"
 	$(Q) $(INSTALL_DATA) $(srcdir)/mk_cmds.1 $(DESTDIR)$(man1dir)/mk_cmds.1
 	$(E) "	INSTALL_DATA $(libdir)/pkgconfig/ss.pc"
Index: e2fsprogs-1.41.8/lib/ss/mk_cmds.sh.in
===================================================================
--- e2fsprogs-1.41.8/lib/ss/mk_cmds.sh.in
+++ e2fsprogs-1.41.8/lib/ss/mk_cmds.sh.in	2009-07-15 22:22:12.000000000 +0300
@@ -4,7 +4,9 @@
 
 datarootdir=@datarootdir@
 DIR="${DIR-@datadir@/ss}"
+# BUILD start
 SS_DIR="@SS_DIR@"
+# BUILD end
 AWK=@AWK@
 SED=@SED@
 
@@ -30,12 +32,16 @@
 fi
 
 if test ! -f $DIR/ct_c.sed || test ! -f $DIR/ct_c.awk ; then
+# BUILD start
     DIR="$SS_DIR"
 #    echo "Falling back to $DIR..."
     if test ! -f "$DIR/ct_c.sed" || test ! -f "$DIR/ct_c.awk" ; then
+# BUILD end
 	echo "mk_cmds: Couldn't find mk_cmds's template files."
 	exit 1
+# BUILD start
     fi
+# BUILD end
 fi
 
 FILE="$1"
Index: e2fsprogs-1.41.8/lib/et/Makefile.in
===================================================================
--- e2fsprogs-1.41.8/lib/et/Makefile.in
+++ e2fsprogs-1.41.8/lib/et/Makefile.in	2009-07-15 22:27:03.000000000 +0300
@@ -59,6 +59,10 @@
 	$(Q) $(SUBSTITUTE) $(srcdir)/compile_et.sh.in compile_et
 	$(Q) $(CHMOD) +x compile_et
 
+compile_et.install: compile_et
+	@perl -ne '/^# BUILD start/ ... /^# BUILD end/ or print' $< > $@
+	@$(CHMOD) +x $@
+
 com_err.ps : com_err.dvi
 com_err.dvi: com_err.texinfo
 
@@ -80,7 +84,7 @@
 		$(DESTDIR)$(bindir) $(DESTDIR)$(man1dir) \
 		$(DESTDIR)$(man3dir) $(DESTDIR)$(libdir)/pkgconfig
 
-install:: compile_et libcom_err.a $(HFILES) installdirs com_err.pc
+install:: compile_et libcom_err.a $(HFILES) installdirs com_err.pc compile_et.install
 	$(E) "	INSTALL_DATA $(libdir)/libcom_err.a"
 	$(Q) $(INSTALL_DATA) libcom_err.a $(DESTDIR)$(libdir)/libcom_err.a
 	-$(Q) $(RANLIB) $(DESTDIR)$(libdir)/libcom_err.a
@@ -94,7 +98,7 @@
 		$(INSTALL_DATA) $(srcdir)/$$i $(DESTDIR)$(datadir)/et/$$i; \
 	done
 	$(E) "	INSTALL_SCRIPT $(bindir)/compile_et"
-	$(Q) $(INSTALL_SCRIPT) compile_et $(DESTDIR)$(bindir)/compile_et
+	$(Q) $(INSTALL_SCRIPT) compile_et.install $(DESTDIR)$(bindir)/compile_et
 	$(E) "	INSTALL_DATA $(man3dir)/com_err.3"
 	$(Q) $(INSTALL_DATA) $(srcdir)/com_err.3 $(DESTDIR)$(man3dir)/com_err.3
 	$(E) "	INSTALL_DATA $(man1dir)/compile_et.1"
Index: e2fsprogs-1.41.8/lib/et/compile_et.sh.in
===================================================================
--- e2fsprogs-1.41.8/lib/et/compile_et.sh.in
+++ e2fsprogs-1.41.8/lib/et/compile_et.sh.in	2009-07-15 22:22:12.000000000 +0300
@@ -5,12 +5,14 @@
 datarootdir=@datarootdir@
 AWK=@AWK@
 DIR="${DIR-@datadir@/et}"
+# BUILD start
 ET_DIR="@ET_DIR@"
 
 if test "$1" = "--build-tree" ; then
     shift;
     DIR="$ET_DIR"
 fi
+# BUILD end
 
 if test "x$1" = x ; then
     echo "Usage: compile_et file"
@@ -30,12 +32,16 @@
 done
 
 if test ! -f "$DIR/et_h.awk" || test ! -f "$DIR/et_c.awk" ; then
+# BUILD start
     DIR="$ET_DIR"
 #    echo "Falling back to $DIR..."
     if test ! -f "$DIR/et_h.awk" || test ! -f "$DIR/et_c.awk" ; then
+# BUILD end
 	echo "compile_et: Couldn't find compile_et's template files."
 	exit 1
+# BUILD start
     fi
+# BUILD end
 fi
 
 ROOT=`echo $1 | sed -e s/.et$//`
