Author: Paul Brossier <piem@piem.org>
Description: do not strip externals
diff -Nur -x '*.orig' puredata-0.42.5/build-tree/pd-0.42-5/extra/makefile puredata-0.42.5.new/build-tree/pd-0.42-5/extra/makefile
--- pd-0.42-5/extra/makefile	2008-12-10 19:30:53.000000000 +0100
+++ pd-0.42-5/extra/makefile	2009-11-14 11:41:14.000000000 +0100
@@ -38,7 +38,6 @@
 .c.pd_linux:
 	$(CC) $(LINUXCFLAGS) $(LINUXINCLUDE) -o $*.o -c $*.c
 	$(CC) -shared -o $*.pd_linux $*.o -lc -lm
-	strip --strip-unneeded $*.pd_linux
 	rm -f $*.o
 
 # ----------------------- Mac OSX -----------------------
diff -Nur -x '*.orig' puredata-0.42.5/build-tree/pd-0.42-5/extra/expr~/makefile puredata-0.42.5.new/build-tree/pd-0.42-5/extra/expr~/makefile
--- pd-0.42-5/extra/expr~/makefile	2008-12-10 19:30:53.000000000 +0100
+++ pd-0.42-5/extra/expr~/makefile	2009-11-14 11:41:28.000000000 +0100
@@ -89,7 +89,6 @@
 
 expr.pd_linux: $(LINUXOBJ)
 	$(CC) -export_dynamic  -shared -o expr.pd_linux $(LINUXOBJ) -lc -lm
-	strip --strip-unneeded expr.pd_linux
 
 expr~.pd_linux: expr.pd_linux
 	-ln -s expr.pd_linux expr~.pd_linux
diff -Nur -x '*.orig' puredata-0.42.5/build-tree/pd-0.42-5/extra/pd~/makefile puredata-0.42.5.new/build-tree/pd-0.42-5/extra/pd~/makefile
--- pd-0.42-5/extra/pd~/makefile	2008-12-10 19:30:53.000000000 +0100
+++ pd-0.42-5/extra/pd~/makefile	2009-11-14 11:41:25.000000000 +0100
@@ -11,5 +11,4 @@
 pdsched.pd_linux: pdsched.c
 	$(CC) $(LINUXCFLAGS) $(LINUXINCLUDE) -o $*.o -c $*.c
 	$(CC) -export_dynamic -shared -o $*.pd_linux $*.o -lc -lm
-	strip --strip-unneeded $*.pd_linux
 	rm -f $*.o
