diff -p -up ./src/configure.in.linking ./src/configure.in
--- ./src/configure.in.linking	2010-12-15 17:42:03.000000000 +0100
+++ ./src/configure.in	2010-12-15 17:42:03.000000000 +0100
@@ -211,7 +211,7 @@ dnl This should be fixed so Pd can use A
     then
         SYSSRC=$SYSSRC" s_audio_alsa.c s_audio_alsamm.c s_midi_alsa.c"
         CPPFLAGS=$CPPFLAGS" -DPA_USE_ALSA -DUSEAPI_ALSA"
-        LDFLAGS=$LDFLAGS" -lasound"
+        LIB=$LIB" -lasound"
     fi
 
     if test x$portaudio = "xyes";
@@ -264,11 +264,11 @@ dnl This should be fixed so Pd can use A
     fi
     if test x$jack = "xyes";
     then
-        LDFLAGS=$LDFLAGS" -lrt -ljack"
+        LIB=$LIB" -lrt -ljack"
     fi
     if test x$jack = "xrun";
     then
-        LDFLAGS=$LDFLAGS" -lrt -ljack"
+        LIB=$LIB" -lrt -ljack"
     fi
 
     OSNUMBER=0
@@ -416,7 +416,7 @@ fi
 if test x$fftw = "xyes";
 then
     SYSSRC=$SYSSRC" d_fft_fftw.c d_fftroutine.c"
-    LDFLAGS=$LDFLAGS" -lfftw"
+    LIB=$LIB" -lfftw"
 else
     SYSSRC=$SYSSRC" d_fft_mayer.c d_fftroutine.c"
 fi
diff -p -up ./src/makefile.in.linking ./src/makefile.in
--- ./src/makefile.in.linking	2010-04-09 05:42:33.000000000 +0200
+++ ./src/makefile.in	2010-12-15 17:45:07.000000000 +0100
@@ -121,10 +121,10 @@ $(BIN_DIR)/pdreceive: u_pdreceive.c $(BI
 	$(CC) $(CFLAGS)  $(STRIPFLAG) -o $(BIN_DIR)/pdreceive u_pdreceive.c
 
 $(PDEXEC): $(OBJ) $(BIN_DIR)
-	cd ../obj;  $(CC) $(LDFLAGS) $(DBG_CFLAGS) -o $(PDEXEC) $(OBJ) $(LIB)
+	cd ../obj;  $(CC) $(LDFLAGS) $(DBG_CFLAGS) -o $(PDEXEC) $(OBJ) $(LIB) -ljack
 
 $(BIN_DIR)/pd-gui: $(GOBJ) $(GSRC)
-	cd ../obj; $(CC) $(INCLUDE) -o $(BIN_DIR)/$(GUINAME) $(GOBJ) $(GLIB)
+	cd ../obj; $(CC) $(LDFLAGS) $(INCLUDE) -o $(BIN_DIR)/$(GUINAME) $(GOBJ) $(GLIB)
 
 $(BIN_DIR)/pd.tk: u_main.tk $(BIN_DIR)
 	cp u_main.tk $(BIN_DIR)/pd.tk
