diff -Nur john-1.7.6-old/src/Makefile john-1.7.6/src/Makefile
--- john-1.7.6-old/src/Makefile	2010-09-05 01:46:00.584269654 +0300
+++ john-1.7.6/src/Makefile	2010-09-05 01:46:02.676269508 +0300
@@ -21,12 +21,12 @@
 #OMPFLAGS = -fopenmp
 # Sun Studio with OpenMP (set the OMP_NUM_THREADS env var at runtime)
 #OMPFLAGS = -xopenmp
-CFLAGS = -c -Wall -O2 -fomit-frame-pointer -I/usr/local/include $(OMPFLAGS)
+CFLAGS = -c -Wall -O2 -fomit-frame-pointer  $(OMPFLAGS)
 # -DHAVE_SKEY
 ASFLAGS = -c $(OMPFLAGS)
-LDFLAGS = -s -L/usr/local/lib -L/usr/local/ssl/lib -lcrypto -lm $(OMPFLAGS)
+override LDFLAGS +=    -lcrypto -lm $(OMPFLAGS)
 # -lskey
-LDFLAGS_MKV = -s -lm
+LDFLAGS_MKV =  -lm
 OPT_NORMAL = -funroll-loops
 OPT_INLINE = -finline-functions
 
@@ -430,7 +430,7 @@
 		JOHN_OBJS="$(JOHN_OBJS) c3_fmt.o" \
 		CC=cc \
 		CFLAGS="-c -fast -xarch=native64 -DHAVE_CRYPT $(OMPFLAGS)" \
-		LDFLAGS="-s -xarch=native64 $(OMPFLAGS) -lc -lrt" \
+		LDFLAGS=" -xarch=native64 $(OMPFLAGS) -lc -lrt" \
 		OPT_NORMAL="" \
 		OPT_INLINE="-xinline=s1,s2,s3,s4,s5,s6,s7,s8"
 
@@ -447,7 +447,7 @@
 		JOHN_OBJS="$(JOHN_OBJS) c3_fmt.o" \
 		CC=cc \
 		CFLAGS="-c -xO4 -xarch=v8plusa -xchip=ultra -DHAVE_CRYPT $(OMPFLAGS)" \
-		LDFLAGS="-s $(OMPFLAGS) -lc -lrt" \
+		LDFLAGS=" $(OMPFLAGS) -lc -lrt" \
 		OPT_NORMAL="" \
 		OPT_INLINE="-xinline=s1,s2,s3,s4,s5,s6,s7,s8"
 
@@ -457,7 +457,7 @@
 		JOHN_OBJS="$(JOHN_OBJS) c3_fmt.o" \
 		CC=cc \
 		CFLAGS="-c -xO4 -xarch=v8 -DHAVE_CRYPT $(OMPFLAGS)" \
-		LDFLAGS="-s $(OMPFLAGS) -lc -lrt" \
+		LDFLAGS=" $(OMPFLAGS) -lc -lrt" \
 		OPT_NORMAL="" \
 		OPT_INLINE="-xinline=s1,s2,s3,s4,s5,s6,s7,s8"
 
@@ -475,7 +475,7 @@
 		CC=cc \
 		CFLAGS="-c -fast -xarch=native64 -DHAVE_CRYPT $(OMPFLAGS)" \
 		ASFLAGS="-c -xarch=native64 $(OMPFLAGS)" \
-		LDFLAGS="-s -xarch=native64 $(OMPFLAGS) -lrt" \
+		LDFLAGS=" -xarch=native64 $(OMPFLAGS) -lrt" \
 		OPT_NORMAL="" \
 		OPT_INLINE="-xinline=s1,s2,s3,s4,s5,s6,s7,s8"
 
@@ -494,7 +494,7 @@
 		CC=cc \
 		CFLAGS="-c -fast -xarch=native -DHAVE_CRYPT $(OMPFLAGS)" \
 		ASFLAGS="-c -xarch=native $(OMPFLAGS)" \
-		LDFLAGS="-s -xarch=native $(OMPFLAGS) -lrt" \
+		LDFLAGS=" -xarch=native $(OMPFLAGS) -lrt" \
 		OPT_NORMAL="" \
 		OPT_INLINE="-xinline=s1,s2,s3,s4,s5,s6,s7,s8"
 
@@ -512,7 +512,7 @@
 		CC=cc \
 		CFLAGS="-c -fast -xarch=native -DHAVE_CRYPT $(OMPFLAGS)" \
 		ASFLAGS="-c -xarch=native $(OMPFLAGS)" \
-		LDFLAGS="-s -xarch=native $(OMPFLAGS) -lrt" \
+		LDFLAGS=" -xarch=native $(OMPFLAGS) -lrt" \
 		OPT_NORMAL="" \
 		OPT_INLINE="-xinline=s1,s2,s3,s4,s5,s6,s7,s8"
 
@@ -532,7 +532,7 @@
 		CC=cc \
 		CFLAGS="-c -xO4 -DHAVE_CRYPT $(OMPFLAGS)" \
 		ASFLAGS="-c $(OMPFLAGS)" \
-		LDFLAGS="-s $(OMPFLAGS) -lc -lrt" \
+		LDFLAGS=" $(OMPFLAGS) -lc -lrt" \
 		OPT_NORMAL="" \
 		OPT_INLINE="-xinline=s1,s2,s3,s4,s5,s6,s7,s8"
 
@@ -599,7 +599,7 @@
 	$(MAKE) $(PROJ) \
 		CC=cc \
 		CFLAGS="-c -qunroll=2 -qarch=ppc -qchars=signed" \
-		LDFLAGS="-s -lbsd" \
+		LDFLAGS=" -lbsd" \
 		OPT_NORMAL="-O2" \
 		OPT_INLINE="-O3 -Q=99 -w"
 
@@ -744,7 +744,7 @@
 	$(MAKE) $(PROJ) \
 		CC=cc \
 		CFLAGS="-c -Aa -D_HPUX_SOURCE -DANSI_CPP" \
-		LDFLAGS="-s" \
+		LDFLAGS="" \
 		OPT_NORMAL="+O2" \
 		OPT_INLINE="+O3 +Oinline"
 
@@ -753,7 +753,7 @@
 	$(MAKE) $(PROJ) \
 		CC=cc \
 		CFLAGS="-c -O2 -64 -mips4 -r10000 -signed" \
-		LDFLAGS="-s -64 -mips4 -r10000" \
+		LDFLAGS=" -64 -mips4 -r10000" \
 		OPT_NORMAL="-LNO:opt=1 -OPT:Olimit=2194" \
 		OPT_INLINE="-INLINE:all"
 
@@ -762,7 +762,7 @@
 	$(MAKE) $(PROJ) \
 		CC=cc \
 		CFLAGS="-c -O2 -64 -mips3 -signed" \
-		LDFLAGS="-s -64 -mips3" \
+		LDFLAGS=" -64 -mips3" \
 		OPT_NORMAL="-LNO:opt=1 -OPT:Olimit=2194" \
 		OPT_INLINE="-INLINE:all"
 
@@ -771,7 +771,7 @@
 	$(MAKE) $(PROJ) \
 		CC=cc \
 		CFLAGS="-c -O2 -32 -signed" \
-		LDFLAGS="-s -32" \
+		LDFLAGS=" -32" \
 		OPT_NORMAL="-LNO:opt=1 -OPT:Olimit=2194" \
 		OPT_INLINE="-INLINE:all"
 
