diff -up ppl-1.0/src/mp_std_bits.cc.orig ppl-1.0/src/mp_std_bits.cc
--- ppl-1.0/src/mp_std_bits.cc.orig     2013-01-30 00:54:04.362716243 -0200
+++ ppl-1.0/src/mp_std_bits.cc  2013-01-30 00:56:18.260721371 -0200
@@ -25,6 +25,9 @@ site: http://bugseng.com/products/ppl/ .
 #include "ppl-config.h"
 #include "mp_std_bits.defs.hh"
 
+#if __GNU_MP_VERSION < 5 \
+  || (__GNU_MP_VERSION == 5 && __GNU_MP_VERSION_MINOR < 1)
+
 const bool std::numeric_limits<mpz_class>::is_specialized;
 const int std::numeric_limits<mpz_class>::digits;
 const int std::numeric_limits<mpz_class>::digits10;
@@ -70,3 +73,6 @@ const bool std::numeric_limits<mpq_class
 const bool std::numeric_limits<mpq_class>::traps;
 const bool std::numeric_limits<mpq_class>::tininess_before;
 const std::float_round_style std::numeric_limits<mpq_class>::round_style;
+
+#endif // __GNU_MP_VERSION < 5
+       // || (__GNU_MP_VERSION == 5 && __GNU_MP_VERSION_MINOR < 1)
diff -up ppl-1.0/src/mp_std_bits.defs.hh.orig ppl-1.0/src/mp_std_bits.defs.hh
--- ppl-1.0/src/mp_std_bits.defs.hh.orig        2013-01-30 00:54:10.339716472 -0200
+++ ppl-1.0/src/mp_std_bits.defs.hh     2013-01-30 00:55:45.852720130 -0200
@@ -38,6 +38,9 @@ void swap(mpz_class& x, mpz_class& y);
 #endif // defined(PPL_DOXYGEN_INCLUDE_IMPLEMENTATION_DETAILS)
 void swap(mpq_class& x, mpq_class& y);
 
+#if __GNU_MP_VERSION < 5 \
+  || (__GNU_MP_VERSION == 5 && __GNU_MP_VERSION_MINOR < 1)
+
 namespace std {
 
 #ifdef PPL_DOXYGEN_INCLUDE_IMPLEMENTATION_DETAILS
@@ -164,6 +167,9 @@ public:
 
 } // namespace std
 
+#endif // __GNU_MP_VERSION < 5
+       // || (__GNU_MP_VERSION == 5 && __GNU_MP_VERSION_MINOR < 1)
+
 #include "mp_std_bits.inlines.hh"
 
 #endif // !defined(PPL_mp_std_bits_defs_hh) 
 
