diff -ur tcp_wrappers_7.6-orig/Makefile tcp_wrappers_7.6/Makefile
--- tcp_wrappers_7.6-orig/Makefile	2005-09-24 02:43:14.000000000 +0300
+++ tcp_wrappers_7.6/Makefile	2005-09-24 14:24:31.972087120 +0300
@@ -91,31 +91,31 @@
 # This is good for many BSD+SYSV hybrids with NIS (formerly YP).
 generic aix osf alpha dynix:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ=setenv.o \
+	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	NETGROUP=-DNETGROUP TLI= all
 
 # Ditto, with vsyslog
 sunos4:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ=setenv.o \
+	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	NETGROUP=-DNETGROUP VSYSLOG= TLI= all
 
 # Generic with resolver library.
 generic-resolver:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS=-lresolv RANLIB=ranlib ARFLAGS=rv AUX_OBJ=setenv.o \
+	LIBS=-lresolv RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	NETGROUP=-DNETGROUP TLI= all
 
 # The NeXT loader needs "-m" or it barfs on redefined library functions.
 next:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS=-m RANLIB=ranlib ARFLAGS=rv AUX_OBJ=environ.o \
+	LIBS=-m RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	NETGROUP=-DNETGROUP TLI= all
 
 # SunOS for the 386 was frozen at release 4.0.x.
 sunos40:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ="setenv.o strcasecmp.o" \
+	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	NETGROUP=-DNETGROUP VSYSLOG= TLI= all
 
 # Ultrix is like aix, next, etc., but has miscd and setenv().
@@ -127,61 +127,61 @@
 # This works on EP/IX 1.4.3 and will likely work on Mips (reggers@julian.uwo.ca)
 epix:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ=environ.o \
+	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	NETGROUP=-DNETGROUP TLI= SYSTYPE="-systype bsd43" all
 
 # Freebsd and linux by default have no NIS.
 386bsd netbsd bsdos:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ= NETGROUP= TLI= \
+	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	EXTRA_CFLAGS=-DSYS_ERRLIST_DEFINED VSYSLOG= all
 
 freebsd:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ= NETGROUP= TLI= \
+	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	EXTRA_CFLAGS=-DSYS_ERRLIST_DEFINED VSYSLOG= all
 
 linux:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ=setenv.o \
+	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	NETGROUP= TLI= EXTRA_CFLAGS="-DBROKEN_SO_LINGER" all
 
 # This is good for many SYSV+BSD hybrids with NIS, probably also for HP-UX 7.x.
 hpux hpux8 hpux9 hpux10:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS= RANLIB=echo ARFLAGS=rv AUX_OBJ=setenv.o \
+	LIBS= RANLIB=echo ARFLAGS=rv AUX_OBJ= \
 	NETGROUP=-DNETGROUP TLI= all
 
 # ConvexOS-10.x with UltraNet support (ukkonen@csc.fi).
 convex-ultranet:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS=-lulsock RANLIB=ranlib ARFLAGS=rv AUX_OBJ=environ.o \
+	LIBS=-lulsock RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	NETGROUP=-DNETGROUP TLI= all
 
 # Generic support for the Dynix/PTX version of TLI.
 ptx-generic:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-lsocket -linet -lnsl" RANLIB=echo ARFLAGS=rv \
-	AUX_OBJ="setenv.o strcasecmp.o ptx.o" NETGROUP= TLI=-DPTX all
+	AUX_OBJ= \
 
 # With UDP support optimized for PTX 2.x (timw@sequent.com).
 ptx-2.x:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-lsocket -linet -lnsl" RANLIB=echo ARFLAGS=rv \
-	AUX_OBJ="setenv.o strcasecmp.o tli-sequent.o" NETGROUP= \
+	AUX_OBJ= \
 	TLI=-DTLI_SEQUENT all
 
 # IRIX 4.0.x has a special ar(1) flag.
 irix4:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS="-lc -lsun" RANLIB=echo ARFLAGS=rvs AUX_OBJ=setenv.o \
+	LIBS="-lc -lsun" RANLIB=echo ARFLAGS=rvs AUX_OBJ= \
 	NETGROUP=-DNETGROUP TLI= all
 
 # IRIX 5.2 is SYSV4 with several broken things (such as -lsocket -lnsl).
 irix5:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS=-lsun RANLIB=echo ARFLAGS=rv VSYSLOG= \
-	NETGROUP=-DNETGROUP AUX_OBJ=setenv.o TLI= all
+	NETGROUP=-DNETGROUP AUX_OBJ= \
 
 # IRIX 6.2 (tucker@math.unc.edu). Must find a better value than 200000.
 irix6:
@@ -193,101 +193,101 @@
 sunos5:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-lsocket -lnsl" RANLIB=echo ARFLAGS=rv VSYSLOG= \
-	NETGROUP=-DNETGROUP AUX_OBJ=setenv.o TLI=-DTLI \
+	NETGROUP=-DNETGROUP AUX_OBJ= \
 	BUGS="$(BUGS) -DSOLARIS_24_GETHOSTBYNAME_BUG" all
 
 # Generic SYSV40
 esix sysv4:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-lsocket -lnsl" RANLIB=echo ARFLAGS=rv \
-	NETGROUP=-DNETGROUP AUX_OBJ=setenv.o TLI=-DTLI all
+	NETGROUP=-DNETGROUP AUX_OBJ= \
 
 # DG/UX 5.4.1 and 5.4.2 have an unusual inet_addr() interface.
 dgux:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS=-lnsl RANLIB=echo ARFLAGS=rv \
-	NETGROUP=-DNETGROUP AUX_OBJ=setenv.o TLI=-DTLI \
+	NETGROUP=-DNETGROUP AUX_OBJ= \
 	BUGS="$(BUGS) -DINET_ADDR_BUG" all
 
 dgux543:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS=-lnsl RANLIB=echo ARFLAGS=rv \
-	NETGROUP=-DNETGROUP AUX_OBJ=setenv.o TLI=-DTLI all
+	NETGROUP=-DNETGROUP AUX_OBJ= \
 
 # NCR UNIX 02.02.01 and 02.03.00 (Alex Chircop, msu@unimt.mt)
 ncrsvr4:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-lresolv -lnsl -lsocket" RANLIB=echo ARFLAGS=rv \
-	AUX_OBJ="setenv.o strcasecmp.o" NETGROUP= TLI=-DTLI \
+	AUX_OBJ= \
 	EXTRA_CFLAGS="" FROM_OBJ=ncr.o all
 
 # Tandem SYSV4 (eqawas@hedgehog.ac.cowan.edu.au)
 tandem:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-lsocket -lnsl" RANLIB=echo ARFLAGS=rv \
-	NETGROUP= AUX_OBJ="setenv.o strcasecmp.o" TLI=-DTLI all
+	NETGROUP= AUX_OBJ= \
 
 # Amdahl UTS 2.1.5 (Richard.Richmond@bridge.bst.bls.com)
 uts215:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \   
 	LIBS="-lsocket" RANLIB=echo \
-	ARFLAGS=rv AUX_OBJ=setenv.o NETGROUP=-DNO_NETGROUP TLI= all
+	ARFLAGS=rv AUX_OBJ= \
 
 # UXP/DS System V.4 clone (vic@uida0.uida.es).
 uxp:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-L/usr/ucblib -lsocket -lnsl -lucb" \
 	RANLIB=echo ARFLAGS=rv NETGROUP=-DNETGROUP \
-	AUX_OBJ=setenv.o TLI="-DTLI -DDRS_XTI" all
+	AUX_OBJ= \
 
 # DELL System V.4 Issue 2.2 using gcc (kim@tac.nyc.ny.us, jurban@norden1.com)
 dell-gcc:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-lsocket -lnsl" RANLIB=ranlib ARFLAGS=rv CC=gcc \
-	AUX_OBJ="setenv.o strcasecmp.o" TLI=-DTLI all
+	AUX_OBJ= \
 
 # SCO 3.2v4.1 no frills (jedwards@sol1.solinet.net).
 sco:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-lsocket -lnsl_s" RANLIB=echo ARFLAGS=rv \
-	NETGROUP= AUX_OBJ=setenv.o TLI= all
+	NETGROUP= AUX_OBJ= \
 
 # SCO OpenDesktop 2.0, release 3.2 (peter@midnight.com). Please simplify.
 sco-od2:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-lrpcsvc -lrpc -lyp -lrpc -lrpcsvc -lsocket" \
-	RANLIB=echo ARFLAGS=rv AUX_OBJ=setenv.o \
+	RANLIB=echo ARFLAGS=rv AUX_OBJ= \
 	NETGROUP=-DNETGROUP TLI= all
 
 # SCO 3.2v4.2 with TCP/IP 1.2.1 (Eduard.Vopicka@vse.cz). Please simplify.
 sco-nis:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-lyp -lrpc -lsocket -lyp -lc_s -lc" \
-	RANLIB=echo ARFLAGS=rv AUX_OBJ=setenv.o \
+	RANLIB=echo ARFLAGS=rv AUX_OBJ= \
 	NETGROUP=-DNETGROUP TLI= EXTRA_CFLAGS="-nointl -DNO_NETGRENT" all
 
 # SCO 3.2v5.0.0 OpenServer 5 (bob@odt.handy.com, bill@razorlogic.com)
 sco-os5:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-lrpcsvc -lsocket" RANLIB=echo ARFLAGS=rv VSYSLOG= \
-	AUX_OBJ=setenv.o NETGROUP=-DNETGROUP TLI= all
+	AUX_OBJ= \
 
 # sinix 5.42 setjmp workaround (szrzs023@ub3.ub.uni-kiel.de)
 sinix:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-lsocket -lnsl -L/usr/ccs/lib -lc -L/usr/ucblib -lucb" \
-	RANLIB=echo ARFLAGS=rv AUX_OBJ=setenv.o TLI=-DTLI all
+	RANLIB=echo ARFLAGS=rv AUX_OBJ= \
 
 # Domain SR10.4. Build under bsd, run under either sysv3 or bsd43.
 apollo:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ=setenv.o \
+	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	NETGROUP=-DNETGROUP TLI= SYSTYPE="-A run,any -A sys,any" all
 
 # Pyramid OSx 5.1, using the BSD universe.
 pyramid:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ="environ.o vfprintf.o" \
+	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	STRINGS="-Dstrchr=index -Dstrrchr=rindex -Dmemcmp=bcmp -Dno_memcpy" \
 	NETGROUP="-DNETGROUP -DUSE_GETDOMAIN" TLI= all
 
@@ -295,7 +295,7 @@
 mips:
 	@echo "Warning: some definitions may be wrong."
 	make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ=environ.o \
+	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	NETGROUP=-DNETGROUP TLI= SYSTYPE="-sysname bsd43" all
 
 # Cray (tested with UNICOS 7.0.4).
@@ -303,60 +303,60 @@
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS=-lnet RANLIB=echo ARFLAGS=rv \
 	EXTRA_CFLAGS=-DINADDR_NONE="\"((unsigned long) -1)\"" \
-	AUX_OBJ="setenv.o strcasecmp.o" NETGROUP= TLI= all
+	AUX_OBJ= \
 
 # Unicos 8.x, Cray-YMP (Bruce Kelly).
 unicos8:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS= RANLIB=echo AR=bld ARFLAGS=rv \
-	AUX_OBJ= NETGROUP= TLI= all
+	AUX_OBJ= \
 
 # Power_UNIX 2.1.1 (amantel@lerc.nasa.gov)
 power_unix_211:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-lnsl -lsocket -lgen -lresolv" RANLIB=echo ARFLAGS=rv \
-	NETGROUP= AUX_OBJ=setenv.o TLI=-DTLI BUGS="$(BUGS)" all
+	NETGROUP= AUX_OBJ= \
 
 # ISC (fc@all.net)
 isc:
 	make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-linet -lnsl_s -ldbm" RANLIB=echo ARFLAGS=rv \
-	AUX_OBJ="setenv.o strcasecmp.o" EXTRA_CFLAGS="-DENOTCONN=ENAVAIL" \
+	AUX_OBJ= \
 	NETGROUP= TLI= all
 
 # Interactive UNIX R3.2 version 4.0 (Bobby D. Wright).
 iunix:
 	make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-linet -lnsl_s -ldbm" RANLIB=echo ARFLAGS=rv \
-	AUX_OBJ=environ.o strcasecmp.o NETGROUP= TLI= all
+	AUX_OBJ= \
 
 # RTU 6.0 on a Masscomp 5400 (ben@piglet.cr.usgs.gov). When using the
 # advanced installation, increment argv before actually looking at it.
 rtu:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ=environ.o \
+	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	NETGROUP= TLI= all
 
 # Unixware sans NIS (mc@telebase.com). Compiler dislikes strcasecmp.c.
 unixware1:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-lsocket -lnsl -lc -L/usr/ucblib -lucb" RANLIB=echo ARFLAGS=rv \
-	NETGROUP=$(NETGROUP) AUX_OBJ=environ.o TLI=-DTLI all
+	NETGROUP=$(NETGROUP) AUX_OBJ= \
 
 unixware2:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-lsocket -lnsl -lgen -lc -L/usr/ucblib -lucb" RANLIB=echo \
-	ARFLAGS=rv NETGROUP=$(NETGROUP) AUX_OBJ=environ.o TLI=-DTLI all
+	ARFLAGS=rv NETGROUP=$(NETGROUP) AUX_OBJ= \
 
 u6000:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
 	LIBS="-lsocket -lnsl" RANLIB=echo ARFLAGS=rv \
-	NETGROUP=-DNETGROUP AUX_OBJ="setenv.o strcasecmp.o" TLI=-DTLI all
+	NETGROUP=-DNETGROUP AUX_OBJ= \
 
 # MachTen
 machten:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ=environ.o \
+	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	NETGROUP= TLI= all
 
 ###############################################################
@@ -391,9 +391,9 @@
 # the ones provided with this source distribution. The environ.c module
 # implements setenv(), getenv(), and putenv().
 
-AUX_OBJ= setenv.o
-#AUX_OBJ= environ.o
-#AUX_OBJ= environ.o strcasecmp.o
+AUX_OBJ= \
+#AUX_OBJ= \
+#AUX_OBJ= \
 
 # Uncomment the following if your C library does not provide the
 # strchr/strrchr/memcmp routines, but comes with index/rindex/bcmp.
@@ -659,7 +659,7 @@
 SHELL	= /bin/sh
 .c.o:;	$(CC) $(CFLAGS) -c $*.c
 
-CFLAGS	= -O -DFACILITY=$(FACILITY) $(ACCESS) $(PARANOID) $(NETGROUP) \
+CFLAGS	= ${PARDUS_CFLAGS} -DFACILITY=$(FACILITY) $(ACCESS) $(PARANOID) $(NETGROUP) \
 	$(BUGS) $(SYSTYPE) $(AUTH) $(UMASK) \
 	-DREAL_DAEMON_DIR=\"$(REAL_DAEMON_DIR)\" $(STYLE) $(KILL_OPT) \
 	-DSEVERITY=$(SEVERITY) -DRFC931_TIMEOUT=$(RFC931_TIMEOUT) \
