diff -uNr sphinx2-0.3/include/lm_3g.h sphinx2-0.3.new/include/lm_3g.h
--- sphinx2-0.3/include/lm_3g.h	Wed Dec 13 20:09:08 2000
+++ sphinx2-0.3.new/include/lm_3g.h	Mon Aug 26 19:01:09 2002
@@ -94,6 +94,7 @@
 			   unigram entry.  If mapped, the ID is interpreted depending
 			   on the range it falls into (see lm_3g.c); and it is -ve if
 			   there is no applicable mapping. */
+    int32 wid;
     log_t prob1;
     log_t bo_wt1;
     int32 bigrams;	/* index of 1st entry in lm_t.bigrams[] */ 
