diff -ur disktype-9.orig/linux.c disktype-9/linux.c
--- disktype-9.orig/linux.c	2006-06-04 23:05:06.000000000 +0300
+++ disktype-9/linux.c	2007-02-14 14:34:36.749572288 +0200
@@ -43,11 +43,20 @@
 
   if (get_le_short(buf + 56) == 0xEF53) {
     if (get_le_long(buf + 96) & 0x0008)       /* JOURNAL_DEV flag */
+    {
       print_line(level, "Ext3 external journal");
+      print_line(level, "KERNELMODULE: ext3");
+    }
     else if (get_le_long(buf + 92) & 0x0004)  /* HAS_JOURNAL flag */
+    {
       print_line(level, "Ext3 file system");
+      print_line(level, "KERNELMODULE: ext3");
+    }
     else
+    {
       print_line(level, "Ext2 file system");
+      print_line(level, "KERNELMODULE: ext2");
+    }
 
     get_string(buf + 120, 16, s);
     if (s[0])
@@ -93,19 +102,24 @@
     /* check signature */
     if (memcmp(buf + 52, "ReIsErFs", 8) == 0) {
       print_line(level, "ReiserFS file system (old 3.5 format, standard journal, starts at %d KiB)", at);
+      print_line(level, "KERNELMODULE: reiserfs");
       newformat = 0;
     } else if (memcmp(buf + 52, "ReIsEr2Fs", 9) == 0) {
       print_line(level, "ReiserFS file system (new 3.6 format, standard journal, starts at %d KiB)", at);
+      print_line(level, "KERNELMODULE: reiserfs");
       newformat = 1;
     } else if (memcmp(buf + 52, "ReIsEr3Fs", 9) == 0) {
       newformat = get_le_short(buf + 72);
       if (newformat == 0) {
 	print_line(level, "ReiserFS file system (old 3.5 format, non-standard journal, starts at %d KiB)", at);
+      print_line(level, "KERNELMODULE: reiserfs");
       } else if (newformat == 2) {
 	print_line(level, "ReiserFS file system (new 3.6 format, non-standard journal, starts at %d KiB)", at);
+      print_line(level, "KERNELMODULE: reiserfs");
 	newformat = 1;
       } else {
 	print_line(level, "ReiserFS file system (v3 magic, but unknown version %d, starts at %d KiB)", newformat, at);
+      print_line(level, "KERNELMODULE: reiserfs");
 	continue;
       }
     } else
diff -ur disktype-9.orig/Makefile disktype-9/Makefile
--- disktype-9.orig/Makefile	2006-01-12 19:55:15.000000000 +0200
+++ disktype-9/Makefile	2007-02-14 14:24:30.917995693 +0200
@@ -3,7 +3,7 @@
 ###
 
 RM = rm -f
-CC = gcc
+CC = klcc
 
 OBJS   = main.o lib.o \
          buffer.o file.o cdaccess.o cdimage.o vpc.o compressed.o \
diff -ur disktype-9.orig/unix.c disktype-9/unix.c
--- disktype-9.orig/unix.c	2003-12-30 12:10:41.000000000 +0200
+++ disktype-9/unix.c	2007-02-14 14:27:48.529848766 +0200
@@ -49,6 +49,7 @@
   /* tell the user */
   version = get_le_long(buf + 4);
   print_line(level, "JFS file system, version %d", version);
+  print_line(level, "KERNELMODULE: jfs");
 
   get_string(buf + 101, 11, s);
   print_line(level + 1, "Volume name \"%s\"", s);
@@ -82,6 +83,7 @@
   raw_version = get_be_short(buf + 0x64);
   version = raw_version & 0x000f;
   print_line(level, "XFS file system, version %d", version);
+  print_line(level, "KERNELMODULE: xfs");
 
   get_string(buf + 0x6c, 12, s);
   print_line(level + 1, "Volume name \"%s\"", s);
