--- ext/faad/gstfaad.c~	2009-05-11 01:24:19.701465766 +0300
+++ ext/faad/gstfaad.c	2009-05-11 01:24:19.701465766 +0300
@@ -63,10 +63,6 @@
 #undef faacDecInit2
 #endif
 
-extern long faacDecInit (faacDecHandle, guint8 *, guint32, guint32 *, guint8 *);
-extern gint8 faacDecInit2 (faacDecHandle, guint8 *, guint32,
-    guint32 *, guint8 *);
-
 #endif /* FAAD2_MINOR_VERSION < 7 */
 
 GST_DEBUG_CATEGORY_STATIC (faad_debug);
