Do not register a driver twice.

diff -ur libmikmod-3.2.0-beta2.orig/playercode/mdriver.c libmikmod-3.2.0-beta2/playercode/mdriver.c
--- libmikmod-3.2.0-beta2.orig/playercode/mdriver.c	2004-02-18 15:29:19.000000000 +0200
+++ libmikmod-3.2.0-beta2/playercode/mdriver.c	2009-07-23 20:46:13.000000000 +0300
@@ -202,7 +202,13 @@
 	/* don't register a MISSING() driver */
 	if ((drv->Name) && (drv->Version)) {
 		if (cruise) {
-			while (cruise->next) cruise = cruise->next;
+			// Check if the drv is already there.
+			// Discard it in that case
+			if (cruise==drv) return;
+			while (cruise->next) {
+				cruise = cruise->next;
+				if (cruise==drv) return;
+			}
 			cruise->next = drv;
 		} else
 			firstdriver = drv; 
