diff -ur libmikmod-3.2.0-beta2.orig/include/mikmod.h.in libmikmod-3.2.0-beta2/include/mikmod.h.in
--- libmikmod-3.2.0-beta2.orig/include/mikmod.h.in	2004-02-18 16:10:39.000000000 +0200
+++ libmikmod-3.2.0-beta2/include/mikmod.h.in	2009-07-23 20:38:39.000000000 +0300
@@ -85,7 +85,8 @@
 
 @DOES_NOT_HAVE_SIGNED@
 
-#if defined(__arch64__) || defined(__alpha)
+#include <stdint.h>
+#if __WORDSIZE == 64
 /* 64 bit architectures */
 
 typedef signed char     SBYTE;      /* 1 byte, signed */
diff -ur libmikmod-3.2.0-beta2.orig/include/mikmod_internals.h libmikmod-3.2.0-beta2/include/mikmod_internals.h
--- libmikmod-3.2.0-beta2.orig/include/mikmod_internals.h	2004-02-18 15:29:17.000000000 +0200
+++ libmikmod-3.2.0-beta2/include/mikmod_internals.h	2009-07-23 20:38:39.000000000 +0300
@@ -50,7 +50,8 @@
 /*========== More type definitions */
 
 /* SLONGLONG: 64bit, signed */
-#if defined (__arch64__) || defined(__alpha)
+#include <stdint.h>
+#if __WORDSIZE == 64
 typedef long		SLONGLONG;
 #define NATIVE_64BIT_INT
 #elif defined(__WATCOMC__)
