diff -Nur gksu-2.0.0/gksu/gksu.c gksu-2.0.0.new/gksu/gksu.c
--- gksu-2.0.0/gksu/gksu.c	2006-09-15 04:45:04.000000000 +0200
+++ gksu-2.0.0.new/gksu/gksu.c	2007-03-05 09:09:47.000000000 +0100
@@ -55,6 +55,7 @@
     {"prompt", optional_argument, NULL, 'P'},
     {"desktop", required_argument, NULL, 'D'},
     {"description", required_argument, NULL, 'D'},
+    {"always-ask-pass", no_argument, NULL, 'A'},
     {0, 0, 0, 0}
 };
 
@@ -119,6 +120,9 @@
     N_("  --su-mode, -w\n"
        "    Make GKSu use su, instead of using libgksu's\n"
        "    default.\n"),
+    N_("  --always-ask-pass\n"
+       "    Make GKSu always ask for a password, even if it is cached\n"
+       "    by sudo.\n"),
   };
 
   help_trans = g_strconcat(_(help_text[0]), _(help_text[1]),
@@ -129,6 +133,7 @@
 			   _(help_text[10]), _(help_text[11]),
 			   _(help_text[12]), _(help_text[13]),
 			   _(help_text[14]), _(help_text[15]),
+			   _(help_text[16]), _(help_text[17]),
 			   NULL);
   g_print (_(help_trans), PACKAGE_VERSION, cmdname);
   g_free (help_trans);
@@ -501,7 +506,7 @@
   gtk_init (&newargc, &newargv);
 
   context = gksu_context_new ();
-  while ((c = getopt_long(newargc, newargv, "?hu:lpm:kt:i:gdsSwP::aD:", long_opts, NULL))
+  while ((c = getopt_long(newargc, newargv, "?hu:lpm:kt:i:gdsSwP::aD:A", long_opts, NULL))
 	 != EOF)
     {
       switch (c)
@@ -564,6 +569,9 @@
 	case 'w':
 	  run_mode = SU_MODE;
 	  break;
+	case 'A':
+	  gksu_context_set_always_ask_password(context, TRUE);
+	  break;
 	case 'P':
 	  prompt = TRUE;
 
