diff -Nur -x '*.orig' -x '*~' gksu-2.0.0/nautilus-gksu/libnautilus-gksu.c gksu-2.0.0.new/nautilus-gksu/libnautilus-gksu.c
--- gksu-2.0.0/nautilus-gksu/libnautilus-gksu.c	2008-12-09 22:29:02.000000000 +0100
+++ gksu-2.0.0.new/nautilus-gksu/libnautilus-gksu.c	2008-12-09 22:29:33.000000000 +0100
@@ -246,6 +246,13 @@
   if (tmp)
     *tmp = '\0';
 
+  /* 
+   * Sometimes 'cmd' contains extra blank space at the end which leads
+   * to problem with some programs(e.g. file-roller opens extra window
+   * and steals focus).
+   */
+  cmd = g_strstrip(cmd);
+  
   if (is_desktop)
     full_cmd = cmd;
   else
