diff -Naur Supermodel_0.2a_Src.orig/Makefiles/Makefile.SDL.UNIX.GCC Supermodel_0.2a_Src/Makefiles/Makefile.SDL.UNIX.GCC
--- Supermodel_0.2a_Src.orig/Makefiles/Makefile.SDL.UNIX.GCC	2011-09-23 00:18:42.000000000 +0300
+++ Supermodel_0.2a_Src/Makefiles/Makefile.SDL.UNIX.GCC	2012-01-08 17:22:40.533000094 +0200
@@ -66,10 +66,10 @@
 
 CC = gcc
 LD = gcc
-COMPILER_FLAGS = `sdl-config --cflags` -ISrc/ -ISrc/OSD/SDL/ -ISrc/Pkgs/ -c -Wall -O3
+COMPILER_FLAGS = `sdl-config --cflags` -ISrc/ -ISrc/OSD/SDL/ -c -Wall -O3 `pkg-config --cflags minizip glew`
 CFLAGS = $(COMPILER_FLAGS)
 CPPFLAGS = $(COMPILER_FLAGS)
-LFLAGS = -o $(OUTFILE) -L$(SDL_LIBPATH) -lGL -lGLU -lSDL -lz -lm -lstdc++
+LFLAGS = -o $(OUTFILE) -L$(SDL_LIBPATH) -lGL -lGLU -lSDL -lz `pkg-config --libs minizip glew` -lm -lstdc++
 
 #
 # Build options...
@@ -90,7 +90,7 @@
 #
 HEADERS = Src/Supermodel.h Src/Games.h Src/OSD/SDL/Types.h
 OBJ = $(OBJ_DIR)/PPCDisasm.o $(OBJ_DIR)/Games.o $(OBJ_DIR)/Config.o $(OBJ_DIR)/INIFile.o $(OBJ_DIR)/BlockFile.o $(OBJ_DIR)/93C46.o \
-	$(OBJ_DIR)/ROMLoad.o $(OBJ_DIR)/unzip.o $(OBJ_DIR)/ioapi.o $(OBJ_DIR)/Error.o $(OBJ_DIR)/glew.o $(OBJ_DIR)/Shader.o \
+	$(OBJ_DIR)/ROMLoad.o $(OBJ_DIR)/Error.o $(OBJ_DIR)/Shader.o \
 	$(OBJ_DIR)/Real3D.o $(OBJ_DIR)/Render3D.o $(OBJ_DIR)/Models.o $(OBJ_DIR)/Render2D.o $(OBJ_DIR)/TileGen.o \
 	$(OBJ_DIR)/Model3.o $(OBJ_DIR)/ppc.o $(OBJ_DIR)/Main.o $(OBJ_DIR)/Audio.o $(OBJ_DIR)/Thread.o $(OBJ_DIR)/SoundBoard.o \
 	$(OBJ_DIR)/SCSP.o $(OBJ_DIR)/SCSPDSP.o $(OBJ_DIR)/68K.o $(OBJ_DIR)/m68kcpu.o $(OBJ_DIR)/m68kopnz.o $(OBJ_DIR)/m68kopdm.o \
@@ -203,12 +203,6 @@
 $(OBJ_DIR)/%.o:	Src/OSD/SDL/%.cpp $(HEADERS)
 	$(CC) $< $(CPPFLAGS) -o $(OBJ_DIR)/$(*F).o
 	
-$(OBJ_DIR)/%.o:	Src/Pkgs/%.c Src/Pkgs/%.h
-	$(CC) $< $(CFLAGS) -o $(OBJ_DIR)/$(*F).o
-
-$(OBJ_DIR)/%.o:	Src/Pkgs/%.c
-	$(CC) $< $(CFLAGS) -o $(OBJ_DIR)/$(*F).o
-
 #
 # AMP MPEG decoder library
 #
diff -Naur Supermodel_0.2a_Src.orig/Src/Graphics/Render2D.cpp Supermodel_0.2a_Src/Src/Graphics/Render2D.cpp
--- Supermodel_0.2a_Src.orig/Src/Graphics/Render2D.cpp	2011-09-14 12:08:44.000000000 +0300
+++ Supermodel_0.2a_Src/Src/Graphics/Render2D.cpp	2012-01-08 17:23:17.334000093 +0200
@@ -37,7 +37,7 @@
  */
 
 #include <string.h>
-#include "Pkgs/glew.h"
+#include "GL/glew.h"
 #include "Supermodel.h"
 #include "Graphics/Shaders2D.h"	// fragment and vertex shaders
 
@@ -816,7 +816,7 @@
 	
 	// Load shaders
 	if (OKAY != LoadShaderProgram(&shaderProgram,&vertexShader,&fragmentShader,NULL,NULL,vertexShaderSource,fragmentShaderSource))
-		return FAIL;
+		return FAIL;
 	
 	// Get locations of the uniforms
 	glUseProgram(shaderProgram);	// bind program
@@ -879,7 +879,7 @@
 }
 
 CRender2D::~CRender2D(void)
-{
+{
 	DestroyShaderProgram(shaderProgram,vertexShader,fragmentShader);
 	glDeleteTextures(2, texID);
 	
diff -Naur Supermodel_0.2a_Src.orig/Src/Graphics/Render2D.h Supermodel_0.2a_Src/Src/Graphics/Render2D.h
--- Supermodel_0.2a_Src.orig/Src/Graphics/Render2D.h	2011-09-14 12:08:44.000000000 +0300
+++ Supermodel_0.2a_Src/Src/Graphics/Render2D.h	2012-01-08 17:23:32.879000094 +0200
@@ -28,7 +28,7 @@
 #ifndef INCLUDED_RENDER2D_H
 #define INCLUDED_RENDER2D_H
 
-#include "Pkgs/glew.h"
+#include "GL/glew.h"
 
 
 // Dirty rectangle size, must be multiples of 4 because masking code relies on this
diff -Naur Supermodel_0.2a_Src.orig/Src/Graphics/Render3D.h Supermodel_0.2a_Src/Src/Graphics/Render3D.h
--- Supermodel_0.2a_Src.orig/Src/Graphics/Render3D.h	2011-09-14 12:08:44.000000000 +0300
+++ Supermodel_0.2a_Src/Src/Graphics/Render3D.h	2012-01-08 17:23:51.118000094 +0200
@@ -28,7 +28,7 @@
 #ifndef INCLUDED_RENDER3D_H
 #define INCLUDED_RENDER3D_H
 
-#include "Pkgs/glew.h"
+#include "GL/glew.h"
 
 /******************************************************************************
  Internal Definitions and Data Structures
diff -Naur Supermodel_0.2a_Src.orig/Src/Graphics/Shader.cpp Supermodel_0.2a_Src/Src/Graphics/Shader.cpp
--- Supermodel_0.2a_Src.orig/Src/Graphics/Shader.cpp	2011-09-23 18:49:50.000000000 +0300
+++ Supermodel_0.2a_Src/Src/Graphics/Shader.cpp	2012-01-08 17:24:08.442000094 +0200
@@ -25,18 +25,18 @@
  * OpenGL shader management.
  *
  * To-Do List
- * ----------
- * - Mesa crashes because, evidently, the function pointers are invalid. Mesa
- *   returns the following information:
- *		                   Vendor: Mesa project: www.mesa3d.org
- *		                 Renderer: Mesa GLX Indirect
- *		                  Version: 1.2 (1.5 Mesa 6.5.1)
- *		 Shading Language Version: (null)
- *		Maximum Vertex Array Size: -1 vertices
- *		     Maximum Texture Size: 2048 texels
- *		Maximum Vertex Attributes: 16
- *		  Maximum Vertex Uniforms: 16
- * - Check for OpenGL 2.0 and perhaps check some of the function pointers,
+ * ----------
+ * - Mesa crashes because, evidently, the function pointers are invalid. Mesa
+ *   returns the following information:
+ *		                   Vendor: Mesa project: www.mesa3d.org
+ *		                 Renderer: Mesa GLX Indirect
+ *		                  Version: 1.2 (1.5 Mesa 6.5.1)
+ *		 Shading Language Version: (null)
+ *		Maximum Vertex Array Size: -1 vertices
+ *		     Maximum Texture Size: 2048 texels
+ *		Maximum Vertex Attributes: 16
+ *		  Maximum Vertex Uniforms: 16
+ * - Check for OpenGL 2.0 and perhaps check some of the function pointers,
  *   which will be NULL, if GL 2.0 and shaders are not supported.
  * - Keep in mind that all these checks should probably go somewhere else...
  * - Turn this into a class.
@@ -44,7 +44,7 @@
 
 #include <new>
 #include <stdio.h>
-#include "Pkgs/glew.h"
+#include "GL/glew.h"
 #include "Supermodel.h"
 
 
@@ -89,7 +89,7 @@
 	GLuint		shaderProgram, vertexShader, fragmentShader;
 	GLint		result, len;
 	bool		ret = OKAY;
-	
+	
 	// Load shaders from files if specified
 	if (vsFile != NULL)
 		vsSource = LoadShaderSource(vsFile);
@@ -103,14 +103,14 @@
 	{
 		ret = FAIL;
 		goto Quit;
-	}
-
-	// Ensure that shader support exists
-	if ((glCreateProgram==NULL) || (glCreateShader==NULL) || (glShaderSource==NULL) || (glCompileShader==NULL))
-	{
-		ret = FAIL;
-		ErrorLog("OpenGL 2.x does not appear to be present. Unable to proceed.");
-		goto Quit;
+	}
+
+	// Ensure that shader support exists
+	if ((glCreateProgram==NULL) || (glCreateShader==NULL) || (glShaderSource==NULL) || (glCompileShader==NULL))
+	{
+		ret = FAIL;
+		ErrorLog("OpenGL 2.x does not appear to be present. Unable to proceed.");
+		goto Quit;
 	}
 	
 	// Create the shaders and shader program
@@ -169,10 +169,10 @@
 }
 
 void DestroyShaderProgram(GLuint shaderProgram, GLuint vertexShader, GLuint fragmentShader)
-{
-	// In case LoadShaderProgram() failed above due to lack of OpenGL 2.0+ functions...
-	if ((glUseProgram==NULL) || (glDeleteShader==NULL) || (glDeleteProgram==NULL))
-		return;
+{
+	// In case LoadShaderProgram() failed above due to lack of OpenGL 2.0+ functions...
+	if ((glUseProgram==NULL) || (glDeleteShader==NULL) || (glDeleteProgram==NULL))
+		return;
 
 	glUseProgram(0);	// return to fixed function pipeline
 	glDeleteShader(vertexShader);
diff -Naur Supermodel_0.2a_Src.orig/Src/Graphics/Shader.h Supermodel_0.2a_Src/Src/Graphics/Shader.h
--- Supermodel_0.2a_Src.orig/Src/Graphics/Shader.h	2011-09-14 12:08:44.000000000 +0300
+++ Supermodel_0.2a_Src/Src/Graphics/Shader.h	2012-01-08 17:24:24.232000093 +0200
@@ -28,7 +28,7 @@
 #ifndef INCLUDED_SHADER_H
 #define INCLUDED_SHADER_H
 
-#include "Pkgs/glew.h"
+#include "GL/glew.h"
 
 /*
  * LoadShaderProgram(shaderProgramPtr, vertexShaderPtr, fragmentShaderPtr, 
diff -Naur Supermodel_0.2a_Src.orig/Src/OSD/SDL/Main.cpp Supermodel_0.2a_Src/Src/OSD/SDL/Main.cpp
--- Supermodel_0.2a_Src.orig/Src/OSD/SDL/Main.cpp	2011-09-23 18:49:50.000000000 +0300
+++ Supermodel_0.2a_Src/Src/OSD/SDL/Main.cpp	2012-01-08 17:25:48.569000094 +0200
@@ -37,7 +37,7 @@
 #include <cstdio>
 #include <cstring>
 #include <cstdarg>
-#include "Pkgs/glew.h"
+#include "GL/glew.h"
 #ifdef SUPERMODEL_OSX
 #include <SDL/SDL.h>
 #else
diff -Naur Supermodel_0.2a_Src.orig/Src/Pkgs/glew.c Supermodel_0.2a_Src/Src/Pkgs/glew.c
--- Supermodel_0.2a_Src.orig/Src/Pkgs/glew.c	2011-04-23 18:14:02.000000000 +0300
+++ Supermodel_0.2a_Src/Src/Pkgs/glew.c	2012-01-08 17:26:22.618000095 +0200
@@ -29,11 +29,11 @@
 ** THE POSSIBILITY OF SUCH DAMAGE.
 */
 
-#include "Pkgs/glew.h"
+#include "GL/glew.h"
 #if defined(_WIN32)
-#  include "Pkgs/wglew.h"
+#  include "GL/wglew.h"
 #elif !defined(__APPLE__) || defined(GLEW_APPLE_GLX)
-#  include "Pkgs/glxew.h"
+#  include "GL/glxew.h"
 #endif
 
 /*
diff -Naur Supermodel_0.2a_Src.orig/Src/Pkgs/glxew.h Supermodel_0.2a_Src/Src/Pkgs/glxew.h
--- Supermodel_0.2a_Src.orig/Src/Pkgs/glxew.h	2011-04-23 18:14:02.000000000 +0300
+++ Supermodel_0.2a_Src/Src/Pkgs/glxew.h	2012-01-08 17:26:59.425000094 +0200
@@ -97,7 +97,7 @@
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 #include <X11/Xmd.h>
-#include "Pkgs/glew.h"
+#include "GL/glew.h"
 
 #ifdef __cplusplus
 extern "C" {
diff -Naur Supermodel_0.2a_Src.orig/Src/ROMLoad.cpp Supermodel_0.2a_Src/Src/ROMLoad.cpp
--- Supermodel_0.2a_Src.orig/Src/ROMLoad.cpp	2011-09-17 16:41:14.000000000 +0300
+++ Supermodel_0.2a_Src/Src/ROMLoad.cpp	2012-01-08 17:27:19.284000093 +0200
@@ -28,7 +28,7 @@
 #include <new>
 #include <cstring>
 #include "Supermodel.h"
-#include "Pkgs/unzip.h"
+#include "unzip.h"
 
 
 /*
