Index: kexec-tools-2.0.2_20110617/Makefile.in
===================================================================
--- kexec-tools-2.0.2_20110617.orig/Makefile.in
+++ kexec-tools-2.0.2_20110617/Makefile.in
@@ -162,11 +162,6 @@ include $(srcdir)/kdump/Makefile
 #
 include $(srcdir)/vmcore-dmesg/Makefile
 
-#
-# kexec_test (test program)
-#
-include $(srcdir)/kexec_test/Makefile
-
 SPEC=$(PACKAGE_NAME).spec
 GENERATED_SRCS:= $(SPEC)
 TARBALL=$(PACKAGE_NAME)-$(PACKAGE_VERSION).tar
