--- kdepim-4.7.2/kmail/kmkernel.cpp~	2011-10-07 18:31:03.000000000 +0400
+++ kdepim-4.7.2/kmail/kmkernel.cpp	2011-10-14 15:21:12.000000000 +0400
@@ -276,11 +276,10 @@ void KMKernel::migrateFromKMail1()
       if ( choice == KMessageBox::Cancel )
         exit( 1 );
 
-      // we only will make one attempt at this
-      migrationCfg.writeEntry( "Version", targetVersion );
-      migrationCfg.sync();
-
       if ( choice != KMessageBox::Yes ) {
+        // disable this round of migration
+        migrationCfg.writeEntry( "Version", targetVersion );
+        migrationCfg.sync();
         return;
       }
 
@@ -296,6 +295,9 @@ void KMKernel::migrateFromKMail1()
       }
       if ( result && proc.exitCode() == 0 ) {
         kDebug() << "Akonadi migration has been successful";
+        // we only will make one attempt at this
+        migrationCfg.writeEntry( "Version", targetVersion );
+        migrationCfg.sync();
       } else {
         // exit code 1 means it is already running, so we are probably called by a migrator instance
         kError() << "Akonadi migration failed!";
