diff -Nuar sonata-1.6.2.orig/setup.py sonata-1.6.2/setup.py
--- sonata-1.6.2.orig/setup.py	2009-07-25 13:52:10.920302334 +0300
+++ sonata-1.6.2/setup.py	2009-07-25 14:02:51.900303463 +0300
@@ -73,8 +73,8 @@
         package_dir={"sonata": "sonata/"},
         ext_modules=[Extension(
         "mmkeys", ["mmkeys/mmkeyspy.c", "mmkeys/mmkeys.c", "mmkeys/mmkeysmodule.c"],
-        extra_compile_args=capture("pkg-config --cflags gtk+-2.0 pygtk-2.0").split(),
-        extra_link_args=capture("pkg-config --libs gtk+-2.0 pygtk-2.0").split()
+        extra_compile_args=capture("pkg-config --cflags gtk+-2.0 pygtk-2.0 x11").split(),
+        extra_link_args=capture("pkg-config --libs gtk+-2.0 pygtk-2.0 x11").split()
          ),],
         scripts = ['sonata/sonata'],
         data_files=[('share/sonata', ['README', 'CHANGELOG', 'TODO', 'TRANSLATORS']),
