Index: sonata-1.6.2.1/sonata.desktop
===================================================================
--- sonata-1.6.2.1.orig/sonata.desktop
+++ sonata-1.6.2.1/sonata.desktop
@@ -1,7 +1,9 @@
-
 [Desktop Entry]
 Name=Sonata
+GenericName=Sonata Music Player
+GenericName[tr]=Müzik Oynatıcı
 Comment=An elegant GTK+ MPD client
+Comment[tr]=Müzik arşivinizi dinleyin
 Exec=sonata
 Terminal=false
 Type=Application
