From 316dc3122e3c85dad7ca773f4cc8d0529a9f3694 Mon Sep 17 00:00:00 2001
From: Frederic Crozat <fcrozat@mandriva.com>
Date: Thu, 15 Oct 2009 19:52:09 +0200
Subject: [PATCH] remove unneeded calls

---
 src/unicode_stop |    5 -----
 1 files changed, 0 insertions(+), 5 deletions(-)

diff --git a/src/unicode_stop b/src/unicode_stop
index 07a8e4b..f6760fa 100755
--- a/src/unicode_stop
+++ b/src/unicode_stop
@@ -17,8 +17,3 @@ if test -t ; then
 fi
 stty -iutf8
 
-uid="`id -u 2>/dev/null`" ||:
-if [ "$uid" = '0' ]; then
-	[ ! -r "$HOME/.kbd/.keymap_sv" ] ||
-		loadkeys "$HOME/.kbd/.keymap_sv"
-fi
-- 
1.6.4.4

