--- src/cddb.c~	2004-04-04 00:40:37.000000000 +0200
+++ src/cddb.c	2005-06-24 15:50:55.160952588 +0200
@@ -472,8 +472,11 @@
    
   snprintf(inbuffer, 108, "%s/.cdserverrc", getenv("HOME"));
   if(stat(inbuffer, &st) < 0) {
-    free(inbuffer);
-    return 0;
+    strcpy(inbuffer, "/etc/cdserver");
+    if(stat(inbuffer, &st) < 0) {
+      free(inbuffer);
+      return 0;
+    }
   }
    
   if((cddbconf = fopen(inbuffer, "r")) == NULL) {
