Index: comar-3.0.0/CMakeLists.txt
===================================================================
--- comar-3.0.0.orig/CMakeLists.txt
+++ comar-3.0.0/CMakeLists.txt
@@ -10,11 +10,11 @@ SET (APP_SUFFIX "3")
 
 # Uncomment this for production releases.
 SET (VERSION_SUFFIX "")
-SET (FLAGS "-g -O0 -Werror -Wcast-align -Wno-uninitialized -Wall -fstrict-aliasing")
+#SET (FLAGS "-g -O0 -Werror -Wcast-align -Wno-uninitialized -Wall -fstrict-aliasing")
 
 # Uncomment this for gprof profiling
 # SET (FLAGS "-g -O0 -Werror -Wcast-align -Wno-uninitialized -Wall -fstrict-aliasing -fprofile-arcs -ftest-coverage")
-SET (CMAKE_C_FLAGS "${FLAGS}")
+#SET (CMAKE_C_FLAGS "${FLAGS}")
 # Uncomment this for sparse building
 # SET (CMAKE_C_COMPILER cgcc)
 
