diff -up bash-4.2/variables.h.size_type bash-4.2/variables.h
--- bash-4.2/variables.h.size_type	2012-11-29 10:33:25.109036844 +0100
+++ bash-4.2/variables.h	2012-11-29 10:46:12.718530162 +0100
@@ -95,8 +95,8 @@ typedef struct variable {
 
 typedef struct _vlist {
   SHELL_VAR **list;
-  int list_size;	/* allocated size */
-  int list_len;		/* current number of entries */
+  size_t list_size;	/* allocated size */
+  size_t list_len;	/* current number of entries */
 } VARLIST;
 
 /* The various attributes that a given variable can have. */
