diff -up bash-4.1/examples/loadables/Makefile.in.examples bash-4.1/examples/loadables/Makefile.in
--- bash-4.1/examples/loadables/Makefile.in.examples	2010-06-22 16:20:02.000000000 +0200
+++ bash-4.1/examples/loadables/Makefile.in	2010-06-22 16:20:41.000000000 +0200
@@ -43,7 +43,7 @@ host_os = @host_os@
 host_cpu = @host_cpu@
 host_vendor = @host_vendor@
 
-CFLAGS = @CFLAGS@
+CFLAGS = -O2 -g
 LOCAL_CFLAGS = @LOCAL_CFLAGS@
 DEFS = @DEFS@
 LOCAL_DEFS = @LOCAL_DEFS@
diff -up bash-4.1/examples/loadables/perl/Makefile.in.examples bash-4.1/examples/loadables/perl/Makefile.in
--- bash-4.1/examples/loadables/perl/Makefile.in.examples	2010-06-22 16:20:46.000000000 +0200
+++ bash-4.1/examples/loadables/perl/Makefile.in	2010-06-22 16:21:04.000000000 +0200
@@ -42,7 +42,7 @@ SHELL = @MAKE_SHELL@
 
 PERL5 = perl5
 
-CFLAGS = @CFLAGS@
+CFLAGS = -O2 -g
 
 #
 # These values are generated for configure by ${topdir}/support/shobj-conf.
