Index: gftp-2.0.19/lib/pty.c
===================================================================
--- gftp-2.0.19.orig/lib/pty.c
+++ gftp-2.0.19/lib/pty.c
@@ -61,7 +61,7 @@ _gftp_ptys_open (int fdm, int fds, char
 
 #elif HAVE_GRANTPT
 
-#if !(defined(__FreeBSD__) || defined(__NetBSD__) || defined(__APPLE__))
+#if !(defined(__FreeBSD__) || defined(__NetBSD__) || defined(__APPLE__)|| defined(__linux__))
 #include <stropts.h>
 #endif
 
