diff -Nur libsndfile-1.0.20-old/Makefile.am libsndfile-1.0.20/Makefile.am
--- libsndfile-1.0.20-old/Makefile.am	2009-06-01 15:18:00.043146693 +0300
+++ libsndfile-1.0.20/Makefile.am	2009-06-01 15:18:12.023269283 +0300
@@ -1,6 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
 DISTCHECK_CONFIGURE_FLAGS = --enable-gcc-werror
+ACLOCAL_AMFLAGS = -I M4
 
 if BUILD_OCTAVE_MOD
 octave_dir = Octave
