diff -Naur kdelibs-4.9.5.org/kdecore/auth/kauthhelpersupport.cpp kdelibs-4.9.5/kdecore/auth/kauthhelpersupport.cpp
--- kdelibs-4.9.5.org/kdecore/auth/kauthhelpersupport.cpp	2012-11-30 11:17:37.000000000 +0400
+++ kdelibs-4.9.5/kdecore/auth/kauthhelpersupport.cpp	2013-01-08 17:28:56.979989460 +0400
@@ -22,6 +22,9 @@
 
 #include <cstdlib>
 #include <syslog.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <pwd.h>
 
 #include <QCoreApplication>
 #include <QTimer>
@@ -40,8 +43,26 @@
 
 static bool remote_dbg = false;
 
+static int fixEnviroment(int argc, char **argv)
+{
+    //try correct HOME
+    char *home = "HOME";
+    if(getenv(home)==NULL)
+    {
+	struct passwd *pw = getpwuid(getuid());
+	int overwrite = 0;
+
+	if(pw!=NULL)
+	    setenv(home, pw->pw_dir, overwrite);
+    }
+
+    return 0;
+}
+
 int HelperSupport::helperMain(int argc, char **argv, const char *id, QObject *responder)
 {
+    fixEnviroment(argc, argv);
+
     openlog(id, 0, LOG_USER);
     qInstallMsgHandler(&HelperSupport::helperDebugHandler);
 
