Index: xdm-1.1.8_20090308/config/xdm-config.cpp
===================================================================
--- xdm-1.1.8_20090308.orig/config/xdm-config.cpp
+++ xdm-1.1.8_20090308/config/xdm-config.cpp
@@ -30,9 +30,9 @@ DisplayManager*session:		XDMSCRIPTDIR/Xs
 DisplayManager*reset:		XDMSCRIPTDIR/Xreset
 DisplayManager*authComplain:	true
 ! The following three resources set up display :0 as the console.
-DisplayManager._0.setup:	XDMSCRIPTDIR/Xsetup_0
-DisplayManager._0.startup:	XDMSCRIPTDIR/GiveConsole
-DisplayManager._0.reset:	XDMSCRIPTDIR/TakeConsole
+!DisplayManager._0.setup:	XDMSCRIPTDIR/Xsetup_0
+!DisplayManager._0.startup:	XDMSCRIPTDIR/GiveConsole
+!DisplayManager._0.reset:	XDMSCRIPTDIR/TakeConsole
 #ifdef XPM
 DisplayManager*loginmoveInterval:	10
 #endif /* XPM */
