Index: xdm-1.1.9/config/Xresources.cpp
===================================================================
--- xdm-1.1.9.orig/config/Xresources.cpp
+++ xdm-1.1.9/config/Xresources.cpp
@@ -86,3 +86,5 @@ Chooser*label.font:		*-new century schoo
 Chooser*label.label:		XDMCP Host Menu from CLIENTHOST
 Chooser*list.font:		-*-*-medium-r-normal-*-*-230-*-*-c-*-iso8859-1
 Chooser*Command.font:		*-new century schoolbook-bold-r-normal-*-180-*
+
+XHASHinclude "/etc/X11/Xresources"
