Index: xdm-1.1.8_20090308/config/Xsession.cpp
===================================================================
--- xdm-1.1.8_20090308.orig/config/Xsession.cpp
+++ xdm-1.1.8_20090308/config/Xsession.cpp
@@ -41,47 +41,4 @@ XCOMM	done
 #endif
 fi
 
-case $# in
-1)
-	case $1 in
-	failsafe)
-		exec BINDIR/xterm -geometry 80x24-0-0
-		;;
-	esac
-esac
-
-XCOMM The startup script is not intended to have arguments.
-
-startup=$HOME/.xsession
-resources=$HOME/.Xresources
-
-if [ -s "$startup" ]; then
-	if [ -x "$startup" ]; then
-		exec "$startup"
-	else
-		exec /bin/sh "$startup"
-	fi
-else
-	if [ -r "$resources" ]; then
-		BINDIR/xrdb -load "$resources"
-	fi
-#if defined(__SCO__) || defined(__UNIXWARE__)
-        [ -r /etc/default/xdesktops ] && {
-                . /etc/default/xdesktops
-        }
-
-        [ -r /etc/default/xdm ] && {
-                . /etc/default/xdm
-        }
-
-        XCOMM Allow the user to over-ride the system default desktop
-        [ -r $HOME/.xdmdesktop ] && {
-                . $HOME/.xdmdesktop
-        }
-
-        [ -n "$XDESKTOP" ] && {
-                exec `eval $XDESKTOP`
-        }
-#endif
-	exec BINDIR/xsm
-fi
+. /usr/lib/X11/xinit/Xsession
