diff -Nuar dangerdeep-0.4.0_pre3376.OSmanOS/src/faulthandler.h dangerdeep-0.4.0_pre3376/src/faulthandler.h
--- dangerdeep-0.4.0_pre3376.OSmanOS/src/faulthandler.h	2011-03-01 11:31:00.000000000 +0200
+++ dangerdeep-0.4.0_pre3376/src/faulthandler.h	2013-05-28 08:26:49.261387796 +0300
@@ -119,6 +119,8 @@
 #include <signal.h>
 #include <string>
 #include <sstream>
+#include <sys/types.h>
+#include <unistd.h>
 
 // Note: use --export-dynamic as linker option or you won't get function names here.
 
diff -Nuar dangerdeep-0.4.0_pre3376.OSmanOS/src/matrix3.h dangerdeep-0.4.0_pre3376/src/matrix3.h
--- dangerdeep-0.4.0_pre3376.OSmanOS/src/matrix3.h	2011-03-01 11:31:00.000000000 +0200
+++ dangerdeep-0.4.0_pre3376/src/matrix3.h	2013-05-28 08:27:16.577386121 +0300
@@ -25,6 +25,7 @@
 #define MATRIX3_H
 
 #include "matrix.h"
+#include <string.h>
 
 /// a 3x3 matrix, reimplemented for 3x3 case for speed issues
 template<class D>
