diff -Nur dangerdeep-0.3.0-old/src/system.cpp dangerdeep-0.3.0/src/system.cpp
--- dangerdeep-0.3.0-old/src/system.cpp	2007-07-02 12:31:34.257670839 +0300
+++ dangerdeep-0.3.0/src/system.cpp	2007-07-02 12:32:03.659201129 +0300
@@ -318,7 +318,7 @@
 
 void system::write_console(bool fileonly) const
 {
-	FILE* f = fopen("console_log.txt", "wb");
+	FILE* f = stderr;
 	if (!f) return;
 	const char* rtn = "\n";
 	for (list<string>::const_iterator it = console_text.begin(); it != console_text.end(); ++it) {
