--- sysvinit-2.86/src/init.c.loginshell	2004-07-30 08:16:20.000000000 -0400
+++ sysvinit-2.86/src/init.c	2005-12-19 15:41:29.000000000 -0500
@@ -1153,7 +1153,7 @@
 		for(old = newFamily; old; old = old->next)
 			if (strpbrk(old->rlevel, "S")) break;
 		if (old == NULL)
-			snprintf(buf, sizeof(buf), "~~:S:wait:%s\n", SULOGIN);
+			snprintf(buf, sizeof(buf), "~~:S:wait:%s -l\n", SHELL);
 		else
 			continue;
 	}
