--- lib/xmlparse.c	2007-05-08 05:25:35.000000000 +0300
+++ lib/xmlparse.c	2008-01-08 13:17:07.000000000 +0200
@@ -2563,6 +2563,8 @@
                               (int)(dataPtr - (ICHAR *)dataBuf));
               if (s == next)
                 break;
+              if (ps_parsing == XML_FINISHED || ps_parsing == XML_SUSPENDED)
+                break;
               *eventPP = s;
             }
           }
