Index: indywiki/src/indywiki/indywiki.py
===================================================================
--- indywiki.orig/src/indywiki/indywiki.py	2009-02-22 14:37:21.000000000 +0000
+++ indywiki/src/indywiki/indywiki.py	2009-02-22 14:37:56.000000000 +0000
@@ -192,7 +192,7 @@
         self.actionNonNativeColour = QtGui.QAction(MainWindow)
         self.actionNonNativeColour.setObjectName("actionNonNativeColour")
         self.actionNonNativeColour.setStatusTip(
-                'Change the background colour of non-native images' +
+                'Change the background colour of nearby images' +
                 '(ie, images in pages pointed to/from the current page)')
         QtCore.QObject.connect(self.actionNonNativeColour, QtCore.SIGNAL(
                 'triggered()'), self.setNNimageBgColor)
@@ -394,7 +394,7 @@
             "MainWindow", "&Toggle Progress bar", None,
             QtGui.QApplication.UnicodeUTF8))
         self.actionNonNativeColour.setText(QtGui.QApplication.translate(
-            "MainWindow", "&Change background colour of non-native images", None,
+            "MainWindow", "&Change background colour of nearby images", None,
             QtGui.QApplication.UnicodeUTF8))
         self.actionLoadLatestPage.setText(QtGui.QApplication.translate(
             "MainWindow", "&Load latest page on startup", None,
