Index: git-1.6.3/gitk-git/gitk
===================================================================
--- git-1.6.3.orig/gitk-git/gitk
+++ git-1.6.3/gitk-git/gitk
@@ -10931,9 +10931,9 @@ if {[tk windowingsystem] eq "aqua"} {
     set textfont {Monaco 9}
     set uifont {{Lucida Grande} 9 bold}
 } else {
-    set mainfont {Helvetica 9}
-    set textfont {Courier 9}
-    set uifont {Helvetica 9 bold}
+    set mainfont {{DejaVu Sans} 9}
+    set textfont {{DejaVu Sans Mono} 9}
+    set uifont {{DejaVu Sans} 9 bold}
 }
 set tabstop 8
 set findmergefiles 0
