--- gitweb/gitweb.perl	2007-04-04 11:40:33.000000000 +0300
+++ gitweb/gitweb.perl	2007-04-08 21:30:31.000000000 +0300
@@ -32,8 +32,14 @@
 our $GIT = "++GIT_BINDIR++/git";
 
 # absolute fs-path which will be prepended to the project path
-#our $projectroot = "/pub/scm";
-our $projectroot = "++GITWEB_PROJECTROOT++";
+our $projectroot = "/usr/local/scm";
+open(CONF, '/etc/conf.d/gitweb');
+
+while (<CONF>) {
+    do { $projectroot = $1; last } if /^PROJECT_ROOT="([^"]+)"/;
+}
+
+close(CONF);
 
 # target of the home link on top of all pages
 our $home_link = $my_uri || "/";
