Index: gpsd-2.95/Makefile.am
===================================================================
--- gpsd-2.95.orig/Makefile.am
+++ gpsd-2.95/Makefile.am
@@ -94,12 +94,14 @@ gpxlogger_LDADD = $(DBUS_GLIB_LIBS) libg
 #
 gpsd_c_sources = gpsd_dbus.c gpsd.c
 gpsd_SOURCES = $(gpsd_c_sources) gpsd_dbus.h
+gpsd_LDFLAGS = $(LDFLAGS)
 gpsd_LDADD = $(DBUS_LIBS) $(LIBM) libgpsd.la libgps.la -lm $(LIBPTHREAD) $(LIBUSB_LIBS)
 
 #
 # Build gpsctl
 #
 gpsctl_SOURCES = gpsctl.c
+gpsctl_LDFLAGS = $(LDFLAGS)
 gpsctl_LDADD = $(LIBM) libgpsd.la libgps.la -lm $(LIBPTHREAD) $(LIBUSB_LIBS)
 
 #
@@ -120,12 +122,14 @@ lcdgps_LDADD = $(LIBM) libgps.la -lm
 gpsmon_SOURCES = gpsmon.c monitor_nmea.c monitor_sirf.c \
 	monitor_italk.c monitor_ubx.c monitor_superstar2.c \
 	monitor_oncore.c monitor_tnt.c
+gpsmon_LDFLAGS = $(LDFLAGS)
 gpsmon_LDADD = $(LIBM) $(NCURSES_LIBS) libgpsd.la libgps.la -lm $(LIBPTHREAD)  $(LIBUSB_LIBS)
 
 #
 # Build gpsdecode
 #
 gpsdecode_SOURCES = gpsdecode.c
+gpsdecode_LDFLAGS = $(LDFLAGS)
 gpsdecode_LDADD = $(LIBM) libgpsd.la libgps.la -lm $(LIBPTHREAD)  $(LIBUSB_LIBS)
 
 #
@@ -138,6 +142,7 @@ libgps_VERSION__REVISION = 0
 libgps_VERSION_AGE = 0
 libgps_VERSION_NUMBER = $(libgps_VERSION_AGE):$(libgps_VERSION__REVISION):$(libgps_VERSION_AGE)
 libgps_la_LDFLAGS = -version-number $(libgps_VERSION_CURRENT):$(libgps_VERSION__REVISION):$(libgps_VERSION_AGE)
+libgps_la_LIBADD = $(LIBUSB_LIBS)
 lib_LTLIBRARIES = libgps.la libgpsd.la
 
 libgps_SONAME = $(shell expr $(libgps_VERSION_CURRENT) - $(libgps_VERSION_AGE))
@@ -249,18 +254,19 @@ libgps_la_SOURCES = $(libgps_c_sources)
 
 libgpsd_la_SOURCES = $(libgpsd_c_sources) $(libgpsd_h_sources) \
 				driver_rtcm2.h packet_states.h
+#libgpsd_la_LIBADD = $(LIBUSB_LIBS)
 
 # Warning: This overrides autoconf's normal link-line generation process
 if LIBGPSMM_ENABLE
 libgps_la_SOURCES += libgpsmm.cpp
-libgps_la_LINK = /bin/sh ./libtool --tag=CXX --mode=link g++ $(libgps_la_LDFLAGS) -o $@
+libgps_la_LINK = /bin/sh ./libtool --tag=CXX --mode=link $(CXX) $(LDFLAGS) $(libgps_la_LDFLAGS) -o $@
 else
-libgps_la_LINK = /bin/sh ./libtool --tag=CC --mode=link gcc $(libgps_la_LDFLAGS) -o $@
+libgps_la_LINK = /bin/sh ./libtool --tag=CC --mode=link $(CC) $(LDFLAGS) $(libgps_la_LDFLAGS) -o $@
 endif
 
 nodist_libgpsd_la_SOURCES = packet_names.h ais_json.i
 libgps_la_LIBADD = $(LIBM) $(LIBC) $(LIBNSL) $(LIBSOCKET) $(LIBPTHREAD)
-libgpsd_la_LIBADD = $(LIBM) $(LIBC) $(LIBNSL) $(LIBSOCKET) $(LIBPTHREAD) $(BLUEZ_LIBS) libgps.la
+libgpsd_la_LIBADD = $(LIBM) $(LIBC) $(LIBNSL) $(LIBSOCKET) $(LIBPTHREAD) $(BLUEZ_LIBS) $(LIBUSB_LIBS) libgps.la
 
 noinst_SCRIPTS =
 
@@ -401,6 +407,7 @@ if LIBGPSMM_ENABLE
 # Build test_gpsmm
 #
 test_gpsmm_SOURCES = test_gpsmm.cpp
+test_gpsmm_LDFLAGS = $(LDFLAGS)
 test_gpsmm_LDADD = $(LIBC) libgps.la -lm  $(LIBUSB_LIBS)
 endif
 
@@ -409,7 +416,7 @@ if LIB_Q_GPSMM_ENABLE
 # Build test_qgpsmm
 #
 test_qgpsmm_SOURCES = test_gpsmm.cpp
-test_qgpsmm_LDFLAGS = -Wl,-rpath,$(srcdir)/libQgpsmm/binaries
+test_qgpsmm_LDFLAGS = $(LDFLAGS) -Wl,-rpath,$(srcdir)/libQgpsmm/binaries
 test_qgpsmm_LDADD = $(LIBC) $(LIBUSB) $(QtNetwork_LIBS) -LlibQgpsmm/binaries -lQgpsmm
 test_qgpsmm_DEPENDENCIES = libQgpsmm/binaries/libQgpsmm.so
 endif
@@ -418,12 +425,14 @@ endif
 # Build test_bits tester
 #
 test_bits_SOURCES = test_bits.c
+test_bits_LDFLAGS = $(LDFLAGS)
 test_bits_LDADD = $(LIBC) libgpsd.la libgps.la  $(LIBUSB_LIBS)
 
 #
 # Build packets tester
 #
 test_packet_SOURCES = test_packet.c
+test_packet_LDFLAGS = $(LDFLAGS)
 test_packet_LDADD = $(LIBC) libgpsd.la libgps.la -lm  $(LIBUSB_LIBS)
 
 #
