diff -up gpsd-2.95/gpsd.hotplug.wrapper.hotplugvars gpsd-2.95/gpsd.hotplug.wrapper
--- gpsd-2.95/gpsd.hotplug.wrapper.hotplugvars	2010-07-12 02:40:22.000000000 +0200
+++ gpsd-2.95/gpsd.hotplug.wrapper	2010-07-15 16:48:08.000000000 +0200
@@ -3,20 +3,25 @@
 # This file is Copyright (c) 2010 by the GPSD project
 # BSD terms apply: see the file COPYING in the distribution root for details.
 
+PATH=/usr/sbin:$PATH
+export PATH
+
 if [ -r /etc/default/gpsd ]; then
   . /etc/default/gpsd
 elif [ -r /etc/sysconfig/gpsd ]; then
   . /etc/sysconfig/gpsd
+  GPSD_OPTIONS=$OPTIONS
+  GPSD_SOCKET=$CONTROL_SOCKET
 fi
 
-if [ -n $GPSD_OPTIONS ]; then
+if [ -n "$GPSD_OPTIONS" ]; then
   export GPSD_OPTIONS
 fi
-if [ -n $GPSD_SOCKET ]; then
+if [ -n "$GPSD_SOCKET" ]; then
   export GPSD_SOCKET
 fi
 
-if [ -n $USBAUTO ]; then
+if [ -n "$USBAUTO" ]; then
   [ "$USBAUTO" = "true" ] || exit 0
 fi
 
