diff -ru suspend-0.8.20080612/plymouth_funcs.c suspend-0.8.20080612.new/plymouth_funcs.c
--- suspend-0.8.20080612/plymouth_funcs.c	2010-05-26 15:41:08.000000000 +0200
+++ suspend-0.8.20080612.new/plymouth_funcs.c	2010-05-26 15:39:28.000000000 +0200
@@ -94,7 +94,6 @@
 
 
 
-	plymouth_update ("quit");
 	buffer = ply_buffer_new ();
 
 	key_file = ply_key_file_new (PLYMOUTH_CONFDIR "plymouthd.conf");
diff -ru suspend-0.8.20080612/splash.c suspend-0.8.20080612.new/splash.c
--- suspend-0.8.20080612/splash.c	2010-05-26 15:41:08.000000000 +0200
+++ suspend-0.8.20080612.new/splash.c	2010-05-26 15:43:57.000000000 +0200
@@ -204,6 +204,10 @@
 	splash->key_pressed	= simple_key_pressed;
 	splash->set_caption	= splash_dummy_set_caption;
 
+#ifdef CONFIG_PLYMOUTH
+	plymouth_update ("quit");
+#endif
+
 	if (!mode)
 		return;
 
