Index: cdecl-2.5/Makefile
===================================================================
--- cdecl-2.5.orig/Makefile
+++ cdecl-2.5/Makefile
@@ -15,22 +15,25 @@
 #
 # add -DUSE_READLINE	To compile in support for the GNU readline library.
 
-CFLAGS= -s -O2 -DUSE_READLINE
-CC= gcc
-LIBS= -lreadline -ltermcap
-ALLFILES= makefile cdgram.y cdlex.l cdecl.c cdecl.1 testset testset++
-BINDIR= /usr/bin
-MANDIR= /usr/man/man1
-CATDIR= /usr/man/cat1
-INSTALL= install -c
-INSTALL_DATA= install -c -m 644
-
-cdecl: c++decl
-	ln c++decl cdecl
-
-c++decl: cdgram.c cdlex.c cdecl.c
-	$(CC) $(CFLAGS) -o c++decl cdecl.c $(LIBS)
-	rm -f cdecl
+# For traditional cross-build
+CROSS_COMPILE ?= 
+
+CFLAGS = -s -O2 -DUSE_READLINE
+LDFLAGS =
+CC = $(CROSS_COMPILE)gcc
+LIBS = -lreadline -lncurses
+ALLFILES = Makefile cdgram.y cdlex.l cdecl.c cdecl.1 testset testset++
+INSTALL = install -c
+INSTALL_DATA = install -c -m 644
+
+# Pisi Linux default directories
+prefix = /usr
+bindir = ${prefix}/bin
+mandir = ${prefix}/share/man/man1
+
+cdecl: cdgram.c cdlex.c cdecl.c
+	$(CC) $(CFLAGS) -o cdecl cdecl.c $(LIBS) $(LDFLAGS)
+	ln -s cdecl c++decl
 
 cdlex.c: cdlex.l
 	lex cdlex.l && mv lex.yy.c cdlex.c
@@ -43,18 +46,17 @@ test:
 	./c++decl < testset++
 
 install: cdecl
-	$(INSTALL) cdecl $(BINDIR)
-	ln $(BINDIR)/cdecl $(BINDIR)/c++decl
-	$(INSTALL_DATA) cdecl.1 $(MANDIR)
-	$(INSTALL_DATA) c++decl.1 $(MANDIR)
+	mkdir -pv $(DESTDIR){$(bindir)/,$(mandir)/}
+	$(INSTALL) cdecl c++decl $(DESTDIR)$(bindir)
+	$(INSTALL_DATA) cdecl.1 $(DESTDIR)$(mandir)
+	$(INSTALL_DATA) c++decl.1 $(DESTDIR)$(mandir)
 
 clean:
 	rm -f cdgram.c cdlex.c cdecl y.output c++decl
 
 clobber: clean
-	rm -f $(BINDIR)/cdecl $(BINDIR)/c++decl
-	rm -f $(MANDIR)/cdecl.1 $(MANDIR)/c++decl.1
-	rm -f $(CATDIR)/cdecl.1.gz
+	rm -f $(bindir)/cdecl $(bindir)/c++decl
+	rm -f $(mandir)/cdecl.1 $(mandir)/c++decl.1
 
 cdecl.cpio: $(ALLFILES)
 	ls $(ALLFILES) | cpio -ocv > cdecl.cpio
