--- a/lib/x11/Makefile.x11.in	2012-10-13 12:27:24.894764407 +0200
+++ b/lib/x11/Makefile.x11.in	2012-10-13 12:27:42.063763780 +0200
@@ -100,7 +100,7 @@
 # Rule for building libglfw.pc
 ##########################################################################
 libglfw.pc: libglfw.pc.in
-	$(SED) -e 's,\@PREFIX\@,$(PREFIX),' libglfw.pc.in > libglfw.pc
+	$(SED) -e 's,\@PREFIX\@,$(PREFIX),' -e 's,/usr/lib,$(LIBDIR),' libglfw.pc.in > libglfw.pc
 
 
 ##########################################################################
