diff -Naur pisi~/operations/build.py pisi/operations/build.py
--- pisi~/operations/build.py	2014-06-15 16:33:39.000000000 +0200
+++ pisi/operations/build.py	2014-07-06 18:04:24.648575441 +0200
@@ -402,6 +402,8 @@
         if self.build_type == "emul32":
             env["CC"] = "%s -m32" % os.getenv("CC")
             env["CXX"] = "%s -m32" % os.getenv("CXX")
+            env["CFLAGS"] = os.getenv("CFLAGS").replace("-fPIC", "")
+            env["CXXFLAGS"] = os.getenv("CXXFLAGS").replace("-fPIC", "")
             env["PKG_CONFIG_PATH"] = "/usr/lib32/pkgconfig"
         os.environ.update(env)
 
