diff -Nuar pisi-2.6-OSmanOS/pisi/actionsapi/pisitools.py pisi-2.6/pisi/actionsapi/pisitools.py
--- pisi-2.6-OSmanOS/pisi/actionsapi/pisitools.py	2014-03-08 00:03:46.000000000 +0200
+++ pisi-2.6/pisi/actionsapi/pisitools.py	2014-12-07 20:32:05.111207119 +0200
@@ -41,6 +41,11 @@
     ''' example call: pisitools.dobin("bin/xloadimage", "/bin", "xload") '''
     executable_insinto(join_path(get.installDIR(), destinationDirectory), sourceFile)
 
+def dopixmaps(sourceFile, destinationDirectory = '/usr/share/pixmaps'):
+    '''insert a data file into /usr/share/pixmaps'''
+    ''' example call: pisitools.dopixmaps("/usr/share/pixmaps/firefox", "firefox") '''
+    readable_insinto(join_path(get.installDIR(), destinationDirectory), sourceFile)
+
 def dodir(destinationDirectory):
     '''creates a directory tree'''
     makedirs(join_path(get.installDIR(), destinationDirectory))
