--- pisi~/util.py	2011-07-27 10:17:49.000000000 +0200
+++ pisi/util.py	2013-07-12 11:28:43.861790423 +0200
@@ -666,6 +666,11 @@
 
     return name, "%s-%s" % (version, release)
 
+def parse_package_dir_path(package_name):
+    name = parse_package_name(package_name)[0]
+    if name.split("-").pop() in ["devel", "32bit", "doc", "docs", "userspace"]: name = name[:-1 - len(name.split("-").pop())]
+    return "%s/%s" % (name[0:4].lower() if name.startswith("lib") and len(name) > 3 else name.lower()[0], name.lower())
+
 def parse_delta_package_name_legacy(package_name):
     """Separate delta package name and release infos for package formats <= 1.1.
 
--- pisi~/operations/history.py	2011-05-26 19:17:29.000000000 +0200
+++ pisi/operations/history.py	2013-07-12 11:22:01.869805111 +0200
@@ -72,7 +72,9 @@
     ctx.ui.info(_("Package %s found in repository %s") % (pkg, reponame))
 
     #return _possible_ url for this package
-    return os.path.join(os.path.dirname(repourl), package)
+    return os.path.join(os.path.dirname(repourl),
+                        pisi.util.parse_package_dir_path(package),
+                        package)
 
 def fetch_remote_file(package, errors):
     try:
--- pisi~/index.py	2013-06-21 23:45:57.000000000 +0200
+++ pisi/index.py	2013-07-12 11:34:35.045777591 +0200
@@ -93,11 +93,8 @@
 
         for fn in os.walk(repo_uri).next()[2]:
             if fn.endswith(ctx.const.delta_package_suffix) or fn.endswith(ctx.const.package_suffix):
-                name, version = util.parse_package_name(fn)
-                if name.split("-").pop() in ["devel", "32bit", "doc", "docs", "userspace"]: name = name[:-1 - len(name.split("-").pop())] 
                 pkgpath = os.path.join(repo_uri,
-                                       name[0:4].lower() if name.startswith("lib") and len(name) > 3 else name.lower()[0],
-                                       name.lower())
+                                       util.parse_package_dir_path(fn))
                 if not os.path.isdir(pkgpath): os.makedirs(pkgpath)
                 ctx.ui.info("%-80.80s\r" % (_('Sorting: %s ') %
                     fn), noln = False if ctx.config.get_option("verbose") else True)
--- pisi~/operations/build.py	2013-06-21 23:45:57.000000000 +0200
+++ pisi/operations/build.py	2013-07-12 11:35:22.046775874 +0200
@@ -1214,12 +1214,9 @@
                 if os.path.exists(path):
                     return path
                 else:
-                    name, version = util.parse_package_name(filename)
-                    if name.split("-").pop() in ["devel", "32bit", "doc", "docs", "userspace"]: name = name[:-1 - len(name.split("-").pop())] 
                     path = os.path.join(package_dir,
-                                           name[0:4].lower() if name.startswith("lib") and len(name) > 3 else name.lower()[0],
-                                           name.lower(),
-                                           filename)
+                                        util.parse_package_dir_path(filename),
+                                        filename)
                     if os.path.exists(path):
                         return path
 
