--- pisi/constants.py~	2013-02-14 18:56:41.014485399 +0100
+++ pisi/constants.py	2013-02-14 20:18:11.340306716 +0100
@@ -97,6 +97,12 @@
         self.__c.needs_reboot = "needsreboot"
         self.__c.files_db = "files.db"
         self.__c.repos = "repos"
+        self.__c.devel_package_end = "-devel"
+        self.__c.doc_package_end = "-docs?$"
+        self.__c.assign_to_system_devel = ["system.base", "system.devel"]
+        self.__c.system_devel_component = "system.devel"
+        self.__c.devels_component = "programming.devel"
+        self.__c.docs_component = "programming.docs"
 
         #file/directory permissions
         self.__c.umask = 0022
--- pisi/operations/build.py~	2013-02-14 18:48:42.384502887 +0100
+++ pisi/operations/build.py	2013-02-14 20:30:13.158280342 +0100
@@ -1008,6 +1008,8 @@
         """Build each package defined in PSPEC file. After this process there
         will be .pisi files hanging around, AS INTENDED ;)"""
 
+        doc_ptrn = re.compile(ctx.const.doc_package_end)
+
         self.fetch_component()  # bug 856
 
         # Operations and filters for package files
@@ -1081,7 +1083,15 @@
             if not package.description:
                 package.description = self.spec.source.description
             if not package.partOf:
-                package.partOf = self.spec.source.partOf
+                if package.name.endswith(ctx.const.devel_package_end):
+                    if self.spec.source.partOf in ctx.const.assign_to_system_devel:
+                        package.partOf = ctx.const.system_devel_component
+                    else:
+                        package.partOf = ctx.const.devels_component
+                elif re.search(doc_ptrn, package.name):
+                    package.partOf = ctx.const.docs_component
+                else:
+                    package.partOf = self.spec.source.partOf
             if not package.license:
                 package.license = self.spec.source.license
             if not package.icon:
