diff -Naur pisi.old/cli/search.py pisi/cli/search.py
--- pisi.old/cli/search.py	2011-06-15 11:03:43.000000000 +0200
+++ pisi/cli/search.py	2014-06-01 20:05:00.702709380 +0200
@@ -49,6 +49,8 @@
                                default=False, help=_("Search in installdb"))
         group.add_option("-s", "--sourcedb", action="store_true",
                                default=False, help=_("Search in sourcedb"))
+        group.add_option("-c", "--case-sensitive", action="store_true",
+                               default=False, help=_("Case sensitive search"))
         group.add_option("--name", action="store_true",
                                default=False, help=_('Search in the package name'))
         group.add_option("--summary", action="store_true",
@@ -65,7 +67,8 @@
             self.help()
             return
 
-        replace = re.compile("(%s)" % "|".join(self.args), re.I)
+        cs = ctx.get_option("case_sensitive")
+        replace = re.compile("(%s)" % "|".join(self.args), 0 if cs else re.I)
         lang = ctx.get_option('language')
         repo = ctx.get_option('repository')
         name = ctx.get_option('name')
@@ -77,17 +80,17 @@
 
         if ctx.get_option('installdb'):
             db = pisi.db.installdb.InstallDB()
-            pkgs = db.search_package(self.args, lang, fields)
+            pkgs = db.search_package(self.args, lang, fields, cs)
             get_info = db.get_package
             get_name_sum = lambda pkg:(pkg.name, pkg.summary)
         elif ctx.get_option('sourcedb'):
             db = pisi.db.sourcedb.SourceDB()
-            pkgs = db.search_spec(self.args, lang, repo, fields)
+            pkgs = db.search_spec(self.args, lang, repo, fields, cs)
             get_info = db.get_spec
             get_name_sum = lambda pkg:(pkg.source.name, pkg.source.summary)
         else:
             db = pisi.db.packagedb.PackageDB()
-            pkgs = db.search_package(self.args, lang, repo, fields)
+            pkgs = db.search_package(self.args, lang, repo, fields, cs)
             get_info = db.get_package
             get_name_sum = lambda pkg:(pkg.name, pkg.summary)
 
diff -Naur pisi.old/db/installdb.py pisi/db/installdb.py
--- pisi.old/db/installdb.py	2014-06-01 19:43:22.125756828 +0200
+++ pisi/db/installdb.py	2014-06-01 19:56:23.749728269 +0200
@@ -178,7 +178,7 @@
         files = self.get_files(package)
         return filter(lambda x: x.type == 'config', files.list)
 
-    def search_package(self, terms, lang=None, fields=None):
+    def search_package(self, terms, lang=None, fields=None, cs=False):
         """
         fields (dict) : looks for terms in the fields which are marked as True
         If the fields is equal to None this method will search in all fields
@@ -200,9 +200,9 @@
             if terms == filter(lambda term: (fields['name'] and \
                     re.compile(term, re.I).search(name)) or \
                     (fields['summary'] and \
-                    re.compile(resum % (lang, term), re.I).search(xml)) or \
+                    re.compile(resum % (lang, term), 0 if cs else re.I).search(xml)) or \
                     (fields['desc'] and \
-                    re.compile(redesc % (lang, term), re.I).search(xml)), terms):
+                    re.compile(redesc % (lang, term), 0 if cs else re.I).search(xml)), terms):
                 found.append(name)
         return found
 
diff -Naur pisi.old/db/packagedb.py pisi/db/packagedb.py
--- pisi.old/db/packagedb.py	2011-05-26 19:17:29.000000000 +0200
+++ pisi/db/packagedb.py	2014-06-01 19:56:15.958728554 +0200
@@ -98,7 +98,7 @@
                 found.append(name)
         return found
 
-    def search_package(self, terms, lang=None, repo=None, fields=None):
+    def search_package(self, terms, lang=None, repo=None, fields=None, cs=False):
         """
         fields (dict) : looks for terms in the fields which are marked as True
         If the fields is equal to None the method will search on all fields
@@ -119,9 +119,9 @@
             if terms == filter(lambda term: (fields['name'] and \
                     re.compile(term, re.I).search(name)) or \
                     (fields['summary'] and \
-                    re.compile(resum % (lang, term), re.I).search(xml)) or \
+                    re.compile(resum % (lang, term), 0 if cs else re.I).search(xml)) or \
                     (fields['desc'] and \
-                    re.compile(redesc % (lang, term), re.I).search(xml)), terms):
+                    re.compile(redesc % (lang, term), 0 if cs else re.I).search(xml)), terms):
                 found.append(name)
         return found
 
diff -Naur pisi.old/db/sourcedb.py pisi/db/sourcedb.py
--- pisi.old/db/sourcedb.py	2011-06-15 11:03:43.000000000 +0200
+++ pisi/db/sourcedb.py	2014-06-01 19:56:31.085728001 +0200
@@ -79,7 +79,7 @@
         spec, repo = self.get_spec_repo(name, repo)
         return spec
 
-    def search_spec(self, terms, lang=None, repo=None, fields=None):
+    def search_spec(self, terms, lang=None, repo=None, fields=None, cs=False):
         """
         fields (dict) : looks for terms in the fields which are marked as True
         If the fields is equal to None this method will search in all fields
@@ -100,9 +100,9 @@
             if terms == filter(lambda term: (fields['name'] and \
                     re.compile(term, re.I).search(name)) or \
                     (fields['summary'] and \
-                    re.compile(resum % (lang, term), re.I).search(xml)) or \
+                    re.compile(resum % (lang, term), 0 if cs else re.I).search(xml)) or \
                     (fields['desc'] and \
-                    re.compile(redesc % (lang, term), re.I).search(xml)), terms):
+                    re.compile(redesc % (lang, term), 0 if cs else re.I).search(xml)), terms):
                 found.append(name)
         return found
 
