Index: pisi-2.4_alpha4/pisi/operations/build.py
===================================================================
--- pisi-2.4_alpha4.orig/pisi/operations/build.py
+++ pisi-2.4_alpha4/pisi/operations/build.py
@@ -686,7 +686,8 @@ class Builder:
 
         if func in self.actionLocals:
             if ctx.get_option('ignore_sandbox') or \
-                    not ctx.config.values.build.enablesandbox:
+                    not ctx.config.values.build.enablesandbox or \
+                    "emul32" in self.build_type:
                 self.actionLocals[func]()
             else:
                 import catbox
