--- pisi~/operations/build.py	2013-08-01 10:48:12.000000000 +0200
+++ pisi/operations/build.py	2013-08-01 15:39:31.858867810 +0200
@@ -389,6 +389,7 @@
 
         env = {"PKG_DIR": self.pkg_dir(),
                "WORK_DIR": self.pkg_work_dir(),
+               "HOME": self.pkg_work_dir(),
                "INSTALL_DIR": self.pkg_install_dir(),
                "PISI_BUILD_TYPE": self.build_type,
                "SRC_NAME": self.spec.source.name,
