--- pisi/util.py.orig   2013-11-12 18:54:31.415128663 +0200
+++ pisi/util.py        2013-11-12 18:57:15.760133064 +0200
@@ -578,7 +578,7 @@
         run_strip(filepath, "--strip-debug")
         return True
 
-    elif "SB executable" in fileinfo:
+    elif "SB  executable" in fileinfo:
         if ctx.config.values.build.generatedebug:
             ensure_dirs(os.path.dirname(outpath))
             save_elf_debug(filepath, outpath)
@@ -587,7 +587,7 @@
         # run_chrpath(filepath)
         return True
 
-    elif "SB shared object" in fileinfo:
+    elif "SB  shared object" in fileinfo:
         if ctx.config.values.build.generatedebug:
             ensure_dirs(os.path.dirname(outpath))
             save_elf_debug(filepath, outpath)
 
