--- pisi/util.py~	2013-11-24 16:58:39.830920163 +0100
+++ pisi/util.py	2013-11-24 17:01:35.767913735 +0100
@@ -15,6 +15,7 @@
 # standard python modules
 
 import os
+import re
 import sys
 import fcntl
 import shutil
@@ -578,7 +579,7 @@
         run_strip(filepath, "--strip-debug")
         return True
 
-    elif "SB executable" in fileinfo:
+    elif re.search("SB\s+executable", fileinfo):
         if ctx.config.values.build.generatedebug:
             ensure_dirs(os.path.dirname(outpath))
             save_elf_debug(filepath, outpath)
@@ -587,7 +588,7 @@
         # run_chrpath(filepath)
         return True
 
-    elif "SB shared object" in fileinfo:
+    elif re.search("SB\s+shared object", fileinfo):
         if ctx.config.values.build.generatedebug:
             ensure_dirs(os.path.dirname(outpath))
             save_elf_debug(filepath, outpath)
