--- pisi/operations/build.py~	2013-06-16 23:55:19.000000000 +0200
+++ pisi/operations/build.py	2013-06-16 23:39:09.000000000 +0200
@@ -398,6 +398,7 @@
         if self.build_type == "emul32":
             env["CC"] = "%s -m32" % os.getenv("CC")
             env["CXX"] = "%s -m32" % os.getenv("CXX")
+            env["PKG_CONFIG_PATH"] = "/usr/lib32/pkgconfig"
         os.environ.update(env)
 
         # First check icecream, if not found use ccache
@@ -689,7 +690,7 @@
                     break
             if not os.path.exists(src_dir):
                 src_dir = util.join_path(self.pkg_work_dir(), [d for d in os.walk(self.pkg_work_dir()).next()[1] if not d.startswith(".")][0])
-                if self.get_state() == "unpack": ctx.ui.debug("Using %s as WorkDir" % src_dir)
+                if self.get_state() == "unpack": ctx.ui.debug("Using %s as  WorkDir" % src_dir)
 
         return src_dir
 
--- pisi/actionsapi/autotools.py~	2013-06-16 23:55:19.000000000 +0200
+++ pisi/actionsapi/autotools.py	2013-06-16 23:53:33.000000000 +0200
@@ -109,13 +109,6 @@
     if can_access_file(infoDir):
         unlink(infoDir)
 
-def fixpc():
-    ''' fix .pc files in installDIR()/usr/lib32/pkgconfig'''
-    path = "%s/usr/lib32/pkgconfig" % get.installDIR()
-    if isDirectory(path):
-        for f in ls("%s/*.pc" % path):
-            dosed(f, get.emul32prefixDIR(), get.defaultprefixDIR())
-
 def install(parameters = '', argument = 'install'):
     '''install source into install directory with given parameters'''
     args = 'make prefix=%(prefix)s/%(defaultprefix)s \
@@ -143,7 +136,6 @@
         fixInfoDir()
 
     if get.buildTYPE() == "emul32":
-        fixpc()
         if isDirectory("%s/emul32" % get.installDIR()): removeDir("/emul32")
 
 def rawInstall(parameters = '', argument = 'install'):
@@ -154,7 +146,6 @@
         fixInfoDir()
 
     if get.buildTYPE() == "emul32":
-        fixpc()
         if isDirectory("%s/emul32" % get.installDIR()): removeDir("/emul32")
 
 def aclocal(parameters = ''):
