diff -aur pisi~/pxml/xmlfile.py pisi/pxml/xmlfile.py
--- pisi~/pxml/xmlfile.py	2011-05-26 19:17:29.000000000 +0200
+++ pisi/pxml/xmlfile.py	2013-08-25 15:10:51.403427435 +0200
@@ -71,6 +71,10 @@
         if uri.is_local_file() and not compressed:
             # this is a local file
             localpath = uri.path()
+            if not tmpDir == '/tmp':
+                # copy file for autocompletion
+                import shutil
+                shutil.copy(localpath, tmpDir)
         else:
             # this is a remote file, first download it into tmpDir
             localpath = pisi.file.File.download(uri, tmpDir, sha1sum=sha1sum, 
