diff -aur /usr/lib/pardus/pisi~/operations/build.py /usr/lib/pardus/pisi/operations/build.py
--- /usr/lib/pardus/pisi~/operations/build.py	2014-03-07 23:03:46.000000000 +0100
+++ /usr/lib/pardus/pisi/operations/build.py	2014-04-04 19:57:56.576631165 +0200
@@ -140,7 +140,10 @@
         if name in keeplist:
             continue
 
-        if re.match(pattern, fileinfo):
+        if fileinfo == None:
+            ctx.ui.warning(_("Removing special file skipped for: %s") % filepath)
+            return
+        elif re.match(pattern, fileinfo):
             ctx.ui.debug("Removing special %s file: %s" % (name, filepath))
             os.unlink(filepath)
             # Remove dir if it becomes empty (Bug #11588)
diff -aur /usr/lib/pardus/pisi~/util.py /usr/lib/pardus/pisi/util.py
--- /usr/lib/pardus/pisi~/util.py	2014-03-07 23:03:46.000000000 +0100
+++ /usr/lib/pardus/pisi/util.py	2014-04-04 20:03:27.866630082 +0200
@@ -184,13 +184,13 @@
             return full_path
     return None
 
-def run_batch(cmd):
+def run_batch(cmd, ui_debug=True):
     """Run command and report return value and output."""
     ctx.ui.info(_('Running ') + cmd, verbose=True)
     p = subprocess.Popen(cmd, shell=True,
                          stdout=subprocess.PIPE, stderr=subprocess.PIPE)
     out, err = p.communicate()
-    ctx.ui.debug(_('return value for "%s" is %s') % (cmd, p.returncode))
+    if ui_debug: ctx.ui.debug(_('return value for "%s" is %s') % (cmd, p.returncode))
     return (p.returncode, out, err)
 
 # TODO: it might be worthwhile to try to remove the
@@ -575,7 +575,13 @@
         if ret:
             ctx.ui.warning(_("objcopy (add-debuglink) command failed for file '%s'!") % f)
 
-    if "current ar archive" in fileinfo:
+    if fileinfo == None:        
+        ret, out, err = run_batch("file %s" % filepath, ui_debug=False)
+        if ret:
+            ctx.ui.warning(_("file command failed with return code %s for file: %s") % (ret, filepath))
+            ctx.ui.info(_("Output:\n%s") % out, verbose=True)
+
+    elif "current ar archive" in fileinfo:
         run_strip(filepath, "--strip-debug")
         return True
 
