--- pisi~/actionsapi/kerneltools.py	2013-06-21 23:45:57.000000000 +0200
+++ pisi/actionsapi/kerneltools.py	2013-07-25 22:05:04.983483681 +0200
@@ -82,7 +82,7 @@
         # if successful, this is something like:
         # .1 for 2.6.30.1
         # _rc8 for 2.6.30_rc8
-        extraversion = re.split("2.[0-9].[0-9]{2}([._].*)", get.srcVERSION())[1]
+        extraversion = re.split("3.[0-9].[0-9]{2}([._].*)", get.srcVERSION())[1]
     except IndexError:
         # e.g. if version == 2.6.30
         pass
@@ -181,7 +181,8 @@
 
     # Create extra/ and updates/ subdirectories
     for _dir in ("extra", "updates"):
-        pisitools.dodir("/lib/modules/%s/%s" % (suffix, _dir))
+        pisitools.dodir("/lib/modules/%s/%s" % (re.sub("\.\d+$", "", suffix), _dir))
+        pisitools.dosym("../%s/%s" % (re.sub("\.\d+$", "", suffix), _dir), "/lib/modules/%s/%s" % (suffix, dir))
 
 
 def installHeaders(extraHeaders=None):
