diff -ur a/pisi/actionsapi/kerneltools.py b/pisi/actionsapi/kerneltools.py
--- a/pisi/actionsapi/kerneltools.py	2011-05-26 20:17:29.000000000 +0300
+++ b/pisi/actionsapi/kerneltools.py	2013-07-04 17:17:25.677562199 +0300
@@ -172,8 +172,8 @@
                          "DEPMOD=/bin/true modules_install mod-fw=")
 
     # Remove symlinks first
-    pisitools.remove("/lib/modules/%s/source" % suffix)
-    pisitools.remove("/lib/modules/%s/build" % suffix)
+    #pisitools.remove("/lib/modules/%s/source" % suffix)
+    #pisitools.remove("/lib/modules/%s/build" % suffix)
 
     # Install Module.symvers and System.map here too
     shutil.copy("Module.symvers", "%s/lib/modules/%s/" % (get.installDIR(), suffix))
@@ -238,8 +238,8 @@
     shutil.copy(".config", "%s/" % destination)
 
     # Settle the correct build symlink to this headers
-    pisitools.dosym("/%s" % headersDirectoryName, "/lib/modules/%s/build" % suffix)
-    pisitools.dosym("build", "/lib/modules/%s/source" % suffix)
+    #pisitools.dosym("/%s" % headersDirectoryName, "/lib/modules/%s/build" % suffix)
+    #pisitools.dosym("build", "/lib/modules/%s/source" % suffix)
 
 
 def installLibcHeaders(excludes=None):
