diff -ur pisi/actionsapi/kerneltools.py pisi-2.4/pisi/actionsapi/kerneltools.py
--- pisi/actionsapi/kerneltools.py	2011-05-26 20:17:29.000000000 +0300
+++ pisi/actionsapi/kerneltools.py	2012-10-01 23:06:46.678699641 +0300
@@ -254,7 +254,18 @@
     # Create directories
     shelltools.makedirs(headers_tmp)
     shelltools.makedirs(headers_dir)
-
+    
+    ###################Workaround begins here ...
+    #Workaround information -- http://patches.openembedded.org/patch/33433/
+    cpy_src="%s/linux-*/arch/x86/include/generated" % (get.workDIR())
+    cpy_tgt="%s/arch/x86/include" % (headers_tmp)
+    shelltools.makedirs(cpy_tgt)
+    
+    copy_cmd ="cp -Rv %s %s " % (cpy_src, cpy_tgt)
+    
+    shelltools.system(copy_cmd)
+    #######################Workaround ends here ...
+    
     # make defconfig and install the headers
     autotools.make("%s defconfig" % make_cmd)
     autotools.rawInstall(make_cmd, "headers_install")
diff -ur pisi-2.4-orig/pisi/actionsapi/kerneltools.py pisi-2.4/pisi/actionsapi/kerneltools.py
--- pisi/actionsapi/kerneltools.py	2011-05-26 20:17:29.000000000 +0300
+++ pisi/actionsapi/kerneltools.py	2012-12-01 19:26:08.821064613 +0200
@@ -187,9 +187,10 @@
 def installHeaders(extraHeaders=None):
     """ Install the files needed to build out-of-tree kernel modules. """
 
-    extras = ["drivers/media/dvb/dvb-core",
-              "drivers/media/dvb/frontends",
-              "drivers/media/video"]
+    extras = ["drivers/media/dvb-core",
+              "drivers/media/dvb-frontends",
+              "drivers/media/tuners",
+              "drivers/media/platform"]
 
     if extraHeaders:
         extras.extend(extraHeaders)
