--- pisi/db/filesldb.py~        2014-06-15 16:33:39.000000000 +0200
+++ pisi/db/filesldb.py 2014-06-17 18:29:52.356729861 +0200
@@ -27,7 +27,7 @@
         self.files_ldb_path = os.path.join(ctx.config.info_dir(), ctx.const.files_ldb)
         self.filesdb = plyvel.DB(self.files_ldb_path, create_if_missing=True)
         if not [f for f in os.listdir(self.files_ldb_path) if f.endswith('.ldb')]:
-            self.destroy()
+            if ctx.comar: self.destroy()
             self.create_filesdb()
 
     def __del__(self):
