--- pisi~/cli/command.py	2011-05-26 19:17:29.000000000 +0200
+++ pisi/cli/command.py	2014-06-06 20:58:57.137778684 +0200
@@ -162,7 +162,8 @@
         else:
             ui = pisi.cli.CLI()
 
-        if write and not os.access(pisi.context.config.packages_dir(), os.W_OK):
+        if (write and not os.access(pisi.context.config.packages_dir(), os.W_OK) or \
+            ('sf' in self.get_name() and not os.access(os.path.join(ctx.config.info_dir(), ctx.const.files_ldb), os.W_OK))):
             raise pisi.cli.Error(_("You have to be root for this operation."))
 
         pisi.api.set_userinterface(ui)
