--- pisi~/actionsapi/pisitools.py	2013-08-01 10:48:12.000000000 +0200
+++ pisi/actionsapi/pisitools.py	2013-08-01 15:12:34.424926908 +0200
@@ -306,3 +306,28 @@
 
     for directory in destdirGlob:
         unlinkDir(directory)
+
+class Flags:
+    def __init__(self, *evars):
+        self.evars = evars
+
+    def add(self, *flags):
+        for evar in self.evars:
+            os.environ[evar] = " ".join(os.environ[evar].split() + [f.strip() for f in flags])            
+
+    def remove(self, *flags):
+        for evar in self.evars:
+            os.environ[evar] = " ".join([v for v in os.environ[evar].split() if v not in [f.strip() for f in flags]])
+
+    def replace(self, old_val, new_val):
+        for evar in self.evars:
+            os.environ[evar] = " ".join([new_val if v == old_val else v for v in os.environ[evar].split()])
+
+    def sub(self, pattern, repl, count=0, flags=0):
+        for evar in self.evars:
+            os.environ[evar] = re.sub(pattern, repl, os.environ[evar], count, flags)
+
+cflags = Flags("CFLAGS")
+ldflags = Flags("LDFLAGS")
+cxxflags = Flags("CXXFLAGS")
+flags = Flags("CFLAGS", "CXXFLAGS")
