diff -aur pisi.old/metadata.py pisi/metadata.py
--- pisi.old/metadata.py	2011-05-26 19:17:29.000000000 +0200
+++ pisi/metadata.py	2014-05-25 20:49:04.426267836 +0200
@@ -53,6 +53,7 @@
     t_InstalledSize = [ autoxml.Long, autoxml.mandatory]
     t_PackageSize = [ autoxml.Long, autoxml.optional]
     t_PackageHash = [ autoxml.String, autoxml.optional, "SHA1Sum" ]
+    t_InstallTarHash = [ autoxml.String, autoxml.optional, "SHA1Sum" ]
     t_PackageURI = [ autoxml.String, autoxml.optional]
     t_DeltaPackages = [ [Delta], autoxml.optional]
     t_PackageFormat = [ autoxml.String, autoxml.optional]
@@ -85,6 +86,8 @@
             size = "%.2f %s" % (p_size[0], p_size[1])
             s += _(', Package Size: %s') % size
 
+        s += _(', install.tar.xz sha1sum: %s') % self.installTarHash
+
         return s
 
 class MetaData(xmlfile.XmlFile):
diff -aur pisi.old/operations/build.py pisi/operations/build.py
--- pisi.old/operations/build.py	2014-05-25 14:24:45.759676603 +0200
+++ pisi/operations/build.py	2014-05-25 20:46:43.152934965 +0200
@@ -1124,8 +1124,6 @@
 
             self.gen_metadata_xml(package)
 
-            self.metadata.write(util.join_path(self.pkg_dir(), ctx.const.metadata_xml))
-
             name = self.package_filename(self.metadata.package)
 
             outdir = ctx.get_option('output_dir')
@@ -1154,8 +1152,6 @@
 
             # add xmls and files
             os.chdir(self.pkg_dir())
-
-            pkg.add_metadata_xml(ctx.const.metadata_xml)
             pkg.add_files_xml(ctx.const.files_xml)
 
             # Sort the files in-place according to their path for an ordered
@@ -1169,6 +1165,10 @@
                     orgname = util.join_path("debug", finfo.path)
                 pkg.add_to_install(orgname, finfo.path)
 
+            self.metadata.package.installTarHash = util.sha1_file("%s/install.tar.xz" % self.pkg_dir())
+            self.metadata.write(util.join_path(self.pkg_dir(), ctx.const.metadata_xml))
+            pkg.add_metadata_xml(ctx.const.metadata_xml)
+
             os.chdir(c)
 
             # FIXME Remove this hack
