--- pisi/operations/build.py~	2013-03-12 21:29:07.000000000 +0100
+++ pisi/operations/build.py	2013-03-20 11:38:33.065222645 +0100
@@ -19,6 +19,7 @@
 import pwd
 import grp
 import fnmatch
+import time
 
 import gettext
 __trans = gettext.translation('pisi', fallback=True)
@@ -560,6 +561,7 @@
         install_dir = self.pkg_install_dir()
         abandoned_files = []
         all_paths_in_packages = []
+        skip_paths = []
 
         for package in self.spec.packages:
             for path in package.files:
@@ -568,9 +570,10 @@
 
         def is_included(path1, path2):
             "Return True if path2 includes path1"
+            additional_check = fnmatch.fnmatch(path1, util.join_path(path2, "*")) if os.path.isdir(path2) else False
             return path1 == path2 \
                     or fnmatch.fnmatch(path1, path2) \
-                    or fnmatch.fnmatch(path1, util.join_path(path2, "*"))
+                    or additional_check
 
         for root, dirs, files in os.walk(install_dir):
             if not dirs and not files:
@@ -580,12 +583,24 @@
                 else:
                     abandoned_files.append(root)
 
+            if root in all_paths_in_packages: 
+                skip_paths.append(root)
+                continue
+
+            skip = False 
+            for skip_path in skip_paths:
+                if root.startswith(skip_path):
+                    skip = True
+                    break
+            if skip: continue
+
             for file_ in files:
                 fpath = util.join_path(root, file_)
                 for _path in all_paths_in_packages:
                     if is_included(fpath, _path):
+                        if os.path.isfile(_path):
+                            all_paths_in_packages.pop(all_paths_in_packages.index(_path))
                         break
-
                 else:
                     abandoned_files.append(fpath)
 
