Description: Support Cython 0.14
Author: Sachiel
Origin: backport, http://trac.enlightenment.org/e/changeset/56327/trunk/BINDINGS/python/python-elementary
Forwarded: not-needed
Last-Update: 2011-08-17

Index: b/elementary/elementary.c_elementary_genlist.pxi
===================================================================
--- a/elementary/elementary.c_elementary_genlist.pxi
+++ b/elementary/elementary.c_elementary_genlist.pxi
@@ -126,7 +126,7 @@
     cdef readonly object _state_get_func
     cdef readonly object _del_func
 
-    def __new__(self, *a, **ka):
+    def __cinit__(self, *a, **ka):
         self._item_style = "default"
         self._label_get_func = None
         self._icon_get_func = None
Index: b/elementary/elementary.c_elementary_list.pxi
===================================================================
--- a/elementary/elementary.c_elementary_list.pxi
+++ b/elementary/elementary.c_elementary_list.pxi
@@ -49,7 +49,7 @@
     cdef Elm_List_Item *item
     cdef object cbt
 
-    def __new__(self):
+    def __cinit__(self):
         self.item = NULL
 
     def __del_cb(self):
