Description: Dead function gone so the bindings do work
Author: Sachiel
Origin: upstream, http://trac.enlightenment.org/e/changeset/53448/trunk/BINDINGS/python/python-elementary
Forwarded: not-needed
Last-Update: 2011-08-18

Index: b/elementary/elementary.c_elementary_toolbar.pxi
===================================================================
--- a/elementary/elementary.c_elementary_toolbar.pxi
+++ b/elementary/elementary.c_elementary_toolbar.pxi
@@ -264,20 +264,6 @@
         Object.__init__(self, parent.evas)
         self._set_obj(elm_toolbar_add(parent.obj))
 
-
-    def scrollable_set(self, scrollable):
-        """
-        Set the scrollable property
-
-        @parm: L{scrollable}
-        """
-        elm_toolbar_scrollable_set(self.obj, scrollable)
-
-    property scrollable:
-        def __set__(self, scrollable):
-            elm_toolbar_scrollable_set(self.obj, scrollable)
-
-
     def menu_parent_set(self, c_evas.Object parent):
         elm_toolbar_menu_parent_set(self.obj, parent.obj)
 
Index: b/include/elementary/c_elementary.pxd
===================================================================
--- a/include/elementary/c_elementary.pxd
+++ b/include/elementary/c_elementary.pxd
@@ -541,12 +541,10 @@
     void  elm_toolbar_item_del(Elm_Toolbar_Item *item)
     void  elm_toolbar_item_del_cb_set(Elm_Toolbar_Item *it, void (*func)(void *data, evas.c_evas.Evas_Object *obj, void *event_info))
     void  elm_toolbar_item_select(Elm_Toolbar_Item *item)
-    void  elm_toolbar_scrollable_set(evas.c_evas.Evas_Object *obj, evas.c_evas.Eina_Bool scrollable)
     evas.c_evas.Eina_Bool elm_toolbar_item_disabled_get(Elm_Toolbar_Item *item)
     void  elm_toolbar_item_disabled_set(Elm_Toolbar_Item *item, evas.c_evas.Eina_Bool disabled)
     void  elm_toolbar_item_separator_set(Elm_Toolbar_Item *item, evas.c_evas.Eina_Bool separator)
     evas.c_evas.Eina_Bool elm_toolbar_item_separator_get(Elm_Toolbar_Item *item)
-    void  elm_toolbar_scrollable_set(evas.c_evas.Evas_Object *obj, evas.c_evas.Eina_Bool scrollable)
     void  elm_toolbar_homogenous_set(evas.c_evas.Evas_Object *obj, evas.c_evas.Eina_Bool homogenous)
     void  elm_toolbar_menu_parent_set(evas.c_evas.Evas_Object *obj, evas.c_evas.Evas_Object *parent)
     void  elm_toolbar_item_unselect_all(evas.c_evas.Evas_Object *obj)
