Description: Accommodating for list API changes
Author: glima
Origin: upstream, http://trac.enlightenment.org/e/changeset/53714/trunk/BINDINGS/python/python-elementary
Forwarded: not-needed
Last-Update: 2011-08-18

Index: b/elementary/elementary.c_elementary_list.pxi
===================================================================
--- a/elementary/elementary.c_elementary_list.pxi
+++ b/elementary/elementary.c_elementary_list.pxi
@@ -422,15 +422,22 @@
         def __set__(self, multi):
             elm_list_multi_select_set(self.obj, multi)
 
-    def horizontal_mode_set(self, Elm_List_Mode mode):
-        elm_list_horizontal_mode_set(self.obj, mode)
+    def mode_set(self, Elm_List_Mode mode):
+        elm_list_mode_set(self.obj, mode)
 
-    property horizontal_mode:
+    property mode:
         def __get__(self):
-            return elm_list_horizontal_mode_get(self.obj)
+            return elm_list_mode_get(self.obj)
 
         def __set__(self, Elm_List_Mode mode):
-            elm_list_horizontal_mode_set(self.obj, mode)
+            elm_list_mode_set(self.obj, mode)
+
+    property horizontal:
+        def __get__(self):
+            return elm_list_horizontal_get(self.obj)
+
+        def __set__(self, horizontal):
+            elm_list_horizontal_set(self.obj, horizontal)
 
     def always_select_mode_set(self, always_select):
         elm_list_always_select_mode_set(self.obj, always_select)
Index: b/include/elementary/c_elementary.pxd
===================================================================
--- a/include/elementary/c_elementary.pxd
+++ b/include/elementary/c_elementary.pxd
@@ -100,6 +100,7 @@
         ELM_LIST_COMPRESS
         ELM_LIST_SCROLL
         ELM_LIST_LIMIT
+        ELM_LIST_EXPAND
 
     ctypedef enum Elm_Flip_Mode:
         ELM_FLIP_ROTATE_Y_CENTER_AXIS
@@ -573,8 +574,10 @@
     void         elm_list_go(evas.c_evas.Evas_Object *obj)
     void      elm_list_multi_select_set(evas.c_evas.Evas_Object *obj, evas.c_evas.Eina_Bool multi)
     evas.c_evas.Eina_Bool      elm_list_multi_select_get(evas.c_evas.Evas_Object *obj)
-    void         elm_list_horizontal_mode_set(evas.c_evas.Evas_Object *obj, Elm_List_Mode mode)
-    Elm_List_Mode       elm_list_horizontal_mode_get(evas.c_evas.Evas_Object *obj)
+    void         elm_list_mode_set(evas.c_evas.Evas_Object *obj, Elm_List_Mode mode)
+    Elm_List_Mode       elm_list_mode_get(evas.c_evas.Evas_Object *obj)
+    void         elm_list_horizontal_set(evas.c_evas.Evas_Object *obj, evas.c_evas.Eina_Bool horizontal)
+    evas.c_evas.Eina_Bool       elm_list_horizontal_get(evas.c_evas.Evas_Object *obj)
     void      elm_list_always_select_mode_set(evas.c_evas.Evas_Object *obj, evas.c_evas.Eina_Bool multi)
     evas.c_evas.Eina_List     *elm_list_items_get(evas.c_evas.Evas_Object *obj)
     Elm_List_Item *elm_list_selected_item_get(evas.c_evas.Evas_Object *obj)
