Description: Remove elm_notepad from py-elm
Author: Bruno Dilly <bdilly@profusion.mobi>
Origin: upstream, http://trac.enlightenment.org/e/changeset/54148/trunk/BINDINGS/python/python-elementary
Forwarded: not-needed
Last-Update: 2011-08-18

Index: b/Makefile.am
===================================================================
--- a/Makefile.am
+++ b/Makefile.am
@@ -37,7 +37,6 @@
 	elementary/elementary.c_elementary_layout.pxi \
 	elementary/elementary.c_elementary_list.pxi \
 	elementary/elementary.c_elementary_menu.pxi \
-	elementary/elementary.c_elementary_notepad.pxi \
 	elementary/elementary.c_elementary_notify.pxi \
 	elementary/elementary.c_elementary_object.pxi \
 	elementary/elementary.c_elementary_pager.pxi \
@@ -77,7 +76,6 @@
 	tests/test_layout.py \
 	tests/test_list.py \
 	tests/test_menu.py \
-	tests/test_notepad.py \
 	tests/test_notify.py \
 	tests/test_pager.py \
 	tests/test_panel.py \
Index: b/Makefile.in
===================================================================
--- a/Makefile.in
+++ b/Makefile.in
@@ -304,7 +304,6 @@
 	elementary/elementary.c_elementary_layout.pxi \
 	elementary/elementary.c_elementary_list.pxi \
 	elementary/elementary.c_elementary_menu.pxi \
-	elementary/elementary.c_elementary_notepad.pxi \
 	elementary/elementary.c_elementary_notify.pxi \
 	elementary/elementary.c_elementary_object.pxi \
 	elementary/elementary.c_elementary_pager.pxi \
@@ -345,7 +344,6 @@
 	tests/test_layout.py \
 	tests/test_list.py \
 	tests/test_menu.py \
-	tests/test_notepad.py \
 	tests/test_notify.py \
 	tests/test_pager.py \
 	tests/test_panel.py \
Index: b/elementary/elementary.c_elementary.pyx
===================================================================
--- a/elementary/elementary.c_elementary.pyx
+++ b/elementary/elementary.c_elementary.pyx
@@ -160,7 +160,6 @@
 include "elementary.c_elementary_radio.pxi"
 include "elementary.c_elementary_check.pxi"
 include "elementary.c_elementary_image.pxi"
-include "elementary.c_elementary_notepad.pxi"
 include "elementary.c_elementary_genlist.pxi"
 include "elementary.c_elementary_spinner.pxi"
 include "elementary.c_elementary_notify.pxi"
Index: b/elementary/elementary.c_elementary_notepad.pxi
===================================================================
--- a/elementary/elementary.c_elementary_notepad.pxi
+++ /dev/null
@@ -1,31 +0,0 @@
-# Copyright (c) 2008-2009 Simon Busch
-#
-# This file is part of python-elementary.
-#
-# python-elementary is free software: you can redistribute it and/or modify
-# it under the terms of the GNU Lesser General Public License as published by
-# the Free Software Foundation, either version 3 of the License, or
-# (at your option) any later version.
-#
-# python-elementary is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU Lesser General Public License for more details.
-#
-# You should have received a copy of the GNU Lesser General Public License
-# along with python-elementary.  If not, see <http://www.gnu.org/licenses/>.
-#
-
-cdef class Notepad(Object):
-    def __init__(self, c_evas.Object parent):
-        Object.__init__(self, parent.evas)
-        self._set_obj(elm_notepad_add(parent.obj))
-
-    def file_set(self, file, format):
-        elm_notepad_file_set(self.obj, file, format)
-
-    def bounce_set(self, h, v):
-        elm_notepad_bounce_set(self.obj, h, v)
-
-
-_elm_widget_type_register("notepad", Notepad)
Index: b/include/elementary/c_elementary.pxd
===================================================================
--- a/include/elementary/c_elementary.pxd
+++ b/include/elementary/c_elementary.pxd
@@ -472,11 +472,6 @@
 
     # Composite objects
 
-    # Notepad object
-    evas.c_evas.Evas_Object *elm_notepad_add(evas.c_evas.Evas_Object *parent)
-    void elm_notepad_file_set(evas.c_evas.Evas_Object *obj, char *file, Elm_Text_Format format)
-    void elm_notepad_bounce_set(evas.c_evas.Evas_Object *obj, evas.c_evas.Eina_Bool h_bounce, evas.c_evas.Eina_Bool v_bounce)
-
     # Anchorview object
     evas.c_evas.Evas_Object *elm_anchorview_add(evas.c_evas.Evas_Object *parent)
     void elm_anchorview_text_set(evas.c_evas.Evas_Object *obj, char *text)
Index: b/tests/test.py
===================================================================
--- a/tests/test.py
+++ b/tests/test.py
@@ -14,7 +14,6 @@
 from test_clock import clock_clicked
 from test_hover import hover_clicked, hover2_clicked
 from test_entry import entry_clicked, entry_scrolled_clicked
-from test_notepad import notepad_clicked
 from test_anchorview import anchorview_clicked
 from test_anchorblock import anchorblock_clicked
 from test_toolbar import toolbar_clicked
@@ -89,7 +88,6 @@
              ("Entry", entry_clicked),
              ("Entry Scrolled", entry_scrolled_clicked),
              ("Flip", flip_clicked),
-             ("Notepad", notepad_clicked),
              ("Anchorview", anchorview_clicked),
              ("Anchorblock", anchorblock_clicked),
              ("Toolbar", toolbar_clicked),
Index: b/tests/test_notepad.py
===================================================================
--- a/tests/test_notepad.py
+++ /dev/null
@@ -1,32 +0,0 @@
-#!/usr/bin/env python
-import os
-import elementary
-import edje
-import ecore
-import evas
-
-#----- Notepad -{{{-
-def notepad_clicked(obj, it):
-    win = elementary.Window("notepad", elementary.ELM_WIN_BASIC)
-    win.title_set("Notepad")
-    win.autodel_set(True)
-
-    bg = elementary.Background(win)
-    win.resize_object_add(bg)
-    bg.size_hint_weight_set(evas.EVAS_HINT_EXPAND, evas.EVAS_HINT_EXPAND)
-    bg.show()
-
-    bx = elementary.Box(win)
-    win.resize_object_add(bx)
-    bx.size_hint_weight_set(evas.EVAS_HINT_EXPAND, evas.EVAS_HINT_EXPAND)
-    bx.show()
-
-    l = elementary.Label(win)
-    l.label_set("TODO")
-    bx.pack_end(l)
-    l.show()
-
-    win.resize(320, 300)
-    win.show()
-# }}}
-# vim:foldmethod=marker
