Description: Fix build failure with Cython 0.15
Author: billiob
Origin: upstream, http://trac.enlightenment.org/e/changeset/62423/trunk/BINDINGS/python/python-elementary
Forwarded: not-needed
Last-Update: 2012-01-24

Index: b/elementary/elementary.c_elementary_object.pxi
===================================================================
--- a/elementary/elementary.c_elementary_object.pxi
+++ b/elementary/elementary.c_elementary_object.pxi
@@ -296,15 +296,18 @@
         try:
             lst = self._elmcallbacks[event]
         except KeyError, e:
-            raise ValueError("Unknown event %r" % e)
+            raise ValueError("Unknown event %r" % event)
 
         i = -1
+        ec = None
+        f = None
         for i, (ec, f, a, k) in enumerate(lst):
             if event_conv == ec and func == f:
                 break
-        else:
+
+        if f != func or ec != event_conv:
             raise ValueError("Callback %s was not registered with event %r" %
-                             (func, e))
+                             (func, event))
 
         lst.pop(i)
         if lst:
