Index: python-elementary-0.7.3/configure.ac
===================================================================
--- python-elementary-0.7.3.orig/configure.ac
+++ python-elementary-0.7.3/configure.ac
@@ -63,7 +63,7 @@ fi
 AC_SUBST(pkgconfig_requires_private)
 
 python_evas_version="0.7.3"
-pkgconfig_elementary_requires_libs="elementary >= 0.7.0.52890 evas >= 1.0.0 eina >= 1.0.0 python-evas >= ${python_evas_version}"
+pkgconfig_elementary_requires_libs="elementary >= 0.7.0.0 evas >= 1.0.0 eina >= 1.0.0 python-evas >= ${python_evas_version}"
 AC_SUBST(pkgconfig_elementary_requires_libs)
 PKG_CHECK_MODULES([ELEMENTARY], [$pkgconfig_elementary_requires_libs])
 CYTHON_ELEMENTARY_INCLUDEDIR=`$PKG_CONFIG --variable=includedir "python-evas >= ${python_evas_version}"`
