Index: qedje-mainline/src/CMakeLists.txt
===================================================================
--- qedje-mainline.orig/src/CMakeLists.txt
+++ qedje-mainline/src/CMakeLists.txt
@@ -35,5 +35,5 @@ install(FILES ${qedje_HEADERS} DESTINATI
 
 if (NOT WIN32)
     configure_file(qedje.pc.cmake qedje.pc @ONLY)
-    install(FILES ${CMAKE_BINARY_DIR}/qedje.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig/)
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/qedje.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig/)
 endif (NOT WIN32)
