--- urlview-0.9/url_handler.sh.default	2000-07-04 12:14:30.000000000 +0200
+++ urlview-0.9/url_handler.sh	2007-06-27 17:21:31.000000000 +0200
@@ -28,15 +28,15 @@
 # VT: Launch in the same terminal
 
 # The lists of programs to be executed are
-https_prgs="/usr/X11R6/bin/netscape:XW /usr/bin/lynx:XT"
-http_prgs="/usr/bin/lynx:XT /usr/X11R6/bin/netscape:XW"
+https_prgs="/usr/bin/xdg-open:VT /usr/bin/firefox:XW /usr/bin/chromium-browser:XW /usr/bin/konqueror:XW /usr/bin/htmlview:XW /usr/bin/seamonkey:XW /usr/bin/epiphany:XW /usr/bin/links:XT /usr/bin/lynx:XT"
+http_prgs="/usr/bin/xdg-open:VT /usr/bin/firefox:XW /usr/bin/chromium-browser:XW /usr/bin/konqueror:XW /usr/bin/htmlview:XW /usr/bin/seamonkey:XW /usr/bin/epiphany:XW /usr/bin/links:XT /usr/bin/lynx:XT"
 mailto_prgs="/usr/bin/mutt:VT /usr/bin/elm:VT /usr/bin/pine:VT /usr/bin/mail:VT"
-gopher_prgs="/usr/bin/lynx:XT /usr/bin/gopher:XT"
-ftp_prgs="/usr/bin/lynx:XT /usr/bin/ncftp:XT"
+gopher_prgs="/usr/bin/lynx:XT"
+ftp_prgs="/usr/bin/ncftp:XT /usr/bin/wget:XT /usr/bin/lynx:XT"
 
 # Program used as an xterm (if it doesn't support -T you'll need to change
 # the command line in getprg)
-XTERM=/usr/X11R6/bin/xterm
+XTERM=/usr/bin/xterm
 
 
 ###########################################################################
@@ -100,9 +100,7 @@ gopher)
     prg=`getprg $gopher_prgs`
     ;;
 *)
-    echo "Unknown URL type.  Please report URL and viewer to:"
-    echo "joey@debian.org."
-    echo -n "Press enter to continue."; read x
+    echo -n "Unknown URL type.  Press enter to continue."; read x
     exit
     ;;
 esac
