From 23f86082eab392f4eddf9d49a4f1ced81a074494 Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Fri, 18 Apr 2014 02:13:50 +0200
Subject: [PATCH 01/28] Add vim swap files to the list of exclude filters

BUG: 333558
FIXED-IN: 4.13.1
---
 src/file/fileexcludefilters.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/file/fileexcludefilters.cpp b/src/file/fileexcludefilters.cpp
index 6f31a06..e1f2ed7 100644
--- a/src/file/fileexcludefilters.cpp
+++ b/src/file/fileexcludefilters.cpp
@@ -58,6 +58,7 @@ const char* s_defaultFileExcludeFilters[] = {
     "*.vm*",
     "*.nvram",
     "*.rcore",
+    "*.swap",
     "lzo",
     "litmain.sh",
     "*.orig",
-- 
1.9.0

