From 4f547dbc33aef72592c028273f58b3b235db526c Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Fri, 25 Apr 2014 17:12:49 +0200
Subject: [PATCH 09/28] XapianSearchStore: Remove extra debug info

---
 src/xapian/xapiansearchstore.cpp | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/xapian/xapiansearchstore.cpp b/src/xapian/xapiansearchstore.cpp
index 76e3b1d..a0529ce 100644
--- a/src/xapian/xapiansearchstore.cpp
+++ b/src/xapian/xapiansearchstore.cpp
@@ -215,9 +215,6 @@ int XapianSearchStore::exec(const Query& query)
                 return 0;
             }
 
-            QTime queryGenerationTimer;
-            queryGenerationTimer.start();
-
             Xapian::Query xapQ = toXapianQuery(query.term());
             if (query.searchString().size()) {
                 QString str = query.searchString();
@@ -231,18 +228,12 @@ int XapianSearchStore::exec(const Query& query)
             xapQ = finalizeQuery(xapQ);
 
             Xapian::Enquire enquire(*m_db);
-            kDebug() << xapQ.get_description().c_str();
             enquire.set_query(xapQ);
 
-            kDebug() << "Query Generation" << queryGenerationTimer.elapsed();
-
             Result& res = m_queryMap[m_nextId++];
-            QTime timer;
-            timer.start();
             res.mset = enquire.get_mset(query.offset(), query.limit());
             res.it = res.mset.begin();
 
-            kDebug() << "Exec" << timer.elapsed() << "msecs";
             return m_nextId-1;
         }
         catch (const Xapian::DatabaseModifiedError&) {
-- 
1.9.0

