From dc83e25d304b5339eb390981aa959906cd18a8b3 Mon Sep 17 00:00:00 2001
From: Patrick Spendrin <ps_ml@gmx.de>
Date: Tue, 22 Apr 2014 00:38:29 +0200
Subject: [PATCH 11/28] exclude check for extended attributes for Windows

---
 ConfigureChecks.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ConfigureChecks.cmake b/ConfigureChecks.cmake
index 34db1e4..6b34a24 100644
--- a/ConfigureChecks.cmake
+++ b/ConfigureChecks.cmake
@@ -4,6 +4,7 @@ include(CheckIncludeFiles)
 # least some file systems supported by the operating system. We currently
 # assume the functionality is in libc itself.
 
+if (NOT WIN32)
 # Linux and OS X.
 check_include_files("sys/types.h;sys/xattr.h" HAVE_SYS_XATTR_H)
 
@@ -13,3 +14,4 @@ check_include_files("sys/types.h;sys/extattr.h" HAVE_SYS_EXTATTR_H)
 if (NOT (HAVE_SYS_XATTR_H OR HAVE_SYS_EXTATTR_H))
     message(FATAL_ERROR "Baloo requires extended attributes support in the operating system, but no headers have been found.")
 endif ()
+endif ()
-- 
1.9.0

