From 859f9fc594fe9f2aee5d8a5472c3ff7242c0d6e2 Mon Sep 17 00:00:00 2001
From: Patrick Spendrin <ps_ml@gmx.de>
Date: Mon, 28 Apr 2014 10:23:05 +0200
Subject: [PATCH 13/28] malloc_trim is a gnu extension

---
 src/xapian/xapiandatabase.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/xapian/xapiandatabase.cpp b/src/xapian/xapiandatabase.cpp
index 0478139..737b9a6 100644
--- a/src/xapian/xapiandatabase.cpp
+++ b/src/xapian/xapiandatabase.cpp
@@ -133,7 +133,9 @@ void XapianDatabase::commit()
     m_docsToAdd.clear();
     m_docsToRemove.clear();
 
+#ifdef __GNUC__
     malloc_trim(0);
+#endif
 }
 
 XapianDocument XapianDatabase::document(uint id)
-- 
1.9.0

