From 386d6233691eb84a7f460cc90b006bc0ac558c5a Mon Sep 17 00:00:00 2001
From: Patrick Spendrin <ps_ml@gmx.de>
Date: Mon, 28 Apr 2014 22:26:06 +0200
Subject: [PATCH 22/28] use a more standard way

---
 src/file/autotest/fileindexerconfigutils.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/file/autotest/fileindexerconfigutils.h b/src/file/autotest/fileindexerconfigutils.h
index 15926c5..7669400 100644
--- a/src/file/autotest/fileindexerconfigutils.h
+++ b/src/file/autotest/fileindexerconfigutils.h
@@ -68,7 +68,7 @@ KTempDir* createTmpFolders(const QStringList& folders)
             }
 #ifdef Q_OS_WIN
             if(sf.startsWith(".")) {
-                if(!SetFileAttributesW(reinterpret_cast<WCHAR*>((dir.path() + "/" + sf).toUtf16()), FILE_ATTRIBUTE_HIDDEN)) {
+                if(!SetFileAttributesW(reinterpret_cast<const WCHAR*>((dir.path() + "/" + sf).utf16()), FILE_ATTRIBUTE_HIDDEN)) {
                     qWarning("failed to set 'hidden' attribute!");
                 }
             }
@@ -99,7 +99,7 @@ KTempDir* createTmpFilesAndFolders(const QStringList& list)
                 }
 #ifdef Q_OS_WIN
                 if(sf.startsWith(".")) {
-                    if(!SetFileAttributesW(reinterpret_cast<WCHAR*>((dir.path() + "/" + sf).toUtf16()), FILE_ATTRIBUTE_HIDDEN)) {
+                    if(!SetFileAttributesW(reinterpret_cast<const WCHAR*>((dir.path() + "/" + sf).utf16()), FILE_ATTRIBUTE_HIDDEN)) {
                         qWarning("failed to set 'hidden' attribute!");
                     }
                 }
-- 
1.9.0

