From 833da7c7d5c367dd258be4fcca7c71058fe432f2 Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Tue, 29 Apr 2014 12:06:00 +0200
Subject: [PATCH 24/28] Baloo Search Runner: Show the url in the subtext

---
 src/runner/baloosearchrunner.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/runner/baloosearchrunner.cpp b/src/runner/baloosearchrunner.cpp
index b62cb8b..68c7a80 100644
--- a/src/runner/baloosearchrunner.cpp
+++ b/src/runner/baloosearchrunner.cpp
@@ -24,6 +24,7 @@
 #include <KIcon>
 #include <KRun>
 #include <Plasma/QueryMatch>
+#include <QDir>
 
 #include "query.h"
 
@@ -62,6 +63,12 @@ void SearchRunner::match(Plasma::RunnerContext& context)
         match.setData(it.url());
         match.setType(Plasma::QueryMatch::PossibleMatch);
 
+        QString url = it.url().toLocalFile();
+        if (url.startsWith(QDir::homePath())) {
+            url.replace(0, QDir::homePath().length(), QLatin1String("~"));
+        }
+        match.setSubtext(url);
+
         context.addMatch(context.query(), match);
     }
 
-- 
1.9.0

