From be3389b57a709dbbfd0aaf571731c6639cd4c38a Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Tue, 29 Apr 2014 12:06:15 +0200
Subject: [PATCH 25/28] Remove unused function

---
 src/file/tests/fileindexingqueuetest.cpp | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/src/file/tests/fileindexingqueuetest.cpp b/src/file/tests/fileindexingqueuetest.cpp
index 3e131f3..ad84965 100644
--- a/src/file/tests/fileindexingqueuetest.cpp
+++ b/src/file/tests/fileindexingqueuetest.cpp
@@ -33,16 +33,6 @@
 #include "../fileindexerconfig.h"
 #include "../lib/filemapping.h"
 
-namespace {
-    QString contents(const QString& url) {
-        QFile file(url);
-        file.open(QIODevice::ReadOnly);
-
-        QTextStream stream(&file);
-        return stream.readAll();
-    }
-}
-
 int main(int argc, char** argv)
 {
     KTempDir tempDir;
-- 
1.9.0

