From fc88fdd71c8411ab7383979e04f756469ddcc0ab Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Tue, 29 Apr 2014 14:58:45 +0200
Subject: [PATCH 27/28] FileIndexingJob: Do not allow suspending if already
 suspended

---
 src/file/fileindexingjob.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/file/fileindexingjob.cpp b/src/file/fileindexingjob.cpp
index e76bcbe..758d280 100644
--- a/src/file/fileindexingjob.cpp
+++ b/src/file/fileindexingjob.cpp
@@ -157,12 +157,18 @@ void FileIndexingJob::setTimeoutInterval(int msec)
 
 bool FileIndexingJob::doSuspend()
 {
+    if (m_suspended)
+        return false;
+
     m_suspended = true;
     return true;
 }
 
 bool FileIndexingJob::doResume()
 {
+    if (!m_suspended)
+        return false;
+
     m_suspended = false;
     start(m_args);
     return true;
-- 
1.9.0

