diff -Naur zip30.orig/crc_i386.S zip30/crc_i386.S
--- zip30.orig/crc_i386.S	2007-01-06 17:02:58.000000000 +0200
+++ zip30/crc_i386.S	2010-08-24 14:02:57.828003731 +0300
@@ -302,3 +302,7 @@
 #endif /* i386 || _i386 || _I386 || __i386 */
 
 #endif /* !USE_ZLIB && !CRC_TABLE_ONLY */
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",@progbits
+#endif
diff -Naur zip30.orig/match.S zip30/match.S
--- zip30.orig/match.S	2005-01-28 11:40:14.000000000 +0200
+++ zip30/match.S	2010-08-24 14:03:18.308001773 +0300
@@ -405,3 +405,7 @@
 #endif /* i386 || _I386 || _i386 || __i386  */
 
 #endif /* !USE_ZLIB */
+
+#ifdef __ELF__
+.section .note.GNU-stack,"",@progbits
+#endif
