Index: devede-3.16.9/install.sh
===================================================================
--- devede-3.16.9.orig/install.sh
+++ devede-3.16.9/install.sh
@@ -36,6 +36,7 @@ install_locales()	# arg1=datadir.
 	install_lang "${1}"	ru	ru		# Russian
 	install_lang "${1}"     sk      sk              # Slovak
 	install_lang "${1}"	sv	sv		#
+	install_lang "${1}"	tr	tr		# Turkish
 	install_lang "${1}"	zh_TW	zh_TW		# Taiwanese
 	install_lang "${1}"     zh_CN   zh_CN           # Simplified Chinese
 }
Index: devede-3.16.9/po/update_mo_files
===================================================================
--- devede-3.16.9.orig/po/update_mo_files
+++ devede-3.16.9/po/update_mo_files
@@ -36,6 +36,8 @@ echo sk
 msgfmt sk.po -o sk.mo
 echo sv
 msgfmt sv.po -o sv.mo
+echo tr
+msgfmt tr.po -o tr.mo
 echo "zh_TW (Chinese, Taiwan)"
 msgfmt zh_TW.po -o zh_TW.mo
 echo "zh_CN (Simplified Chinese)"
