diff -r -U1 fotoxx-11.06.orig/zfuncs.cc fotoxx-11.06/zfuncs.cc
--- fotoxx-11.06.orig/zfuncs.cc 2011-06-04 20:53:41.000000000 +0700
+++ fotoxx-11.06/zfuncs.cc  2011-06-07 03:25:04.000000000 +0700
@@ -2999,3 +2999,3 @@
    strncatv(zlocalesdir,199,work,"/share/",zappname,"/locales",null);      //  /installoc/share/appname/locales
-   strncatv(zdocdir,199,work,"/share/doc/",zappname,null);                 //  /installoc/share/doc/appname
+   strncatv(zdocdir,199,work,"/share/doc/",PF,null);                       //  /installoc/share/doc/appname

