--- Makefile.OSmanOS	2013-02-15 20:41:38.000000000 +0200
+++ Makefile	2013-02-16 09:39:00.795498827 +0200
@@ -24,7 +24,7 @@
            f.xform.o f.retouch.o f.retouch2.o f.art.o f.comp.o zfuncs.o
 
 fotoxx: $(ALLFILES)
-	$(CXX) $(LDFLAGS) -o fotoxx $(ALLFILES) $(LIBS) -ltiff -lpng -llcms2
+	$(CXX) $(LDFLAGS) -o fotoxx $(ALLFILES) $(LIBS) -ltiff -lpng -llcms2 -lpthread
 
 fotoxx.o: $(FOTOXX) fotoxx.h
 	$(CXX) $(CFLAGS) -o fotoxx.o $(FOTOXX)
