diff -r -U1 fotoxx-11.08.1.orig/Makefile fotoxx-11.08.1/Makefile
--- fotoxx-11.08.1.orig/Makefile    2011-08-03 16:11:43.000000000 +0700
+++ fotoxx-11.08.1/Makefile 2011-08-04 14:59:24.000000000 +0700
@@ -5,4 +5,4 @@
 # defaults for parameters that may be pre-defined
-CXXFLAGS ?= -O3 -g -Wall
-LDFLAGS ?= -O3 -g -Wall -rdynamic
+CXXFLAGS += -Wall
+LDFLAGS += -rdynamic
 PREFIX ?= /usr
@@ -15,3 +15,3 @@
 LOCALESDIR = $(SHAREDIR)/locales
-DOCDIR = $(PREFIX)/share/doc/fotoxx
+DOCDIR = $(PREFIX)/share/doc/$(PF)
 MANDIR = $(PREFIX)/share/man/man1
@@ -20,3 +20,3 @@
 
-CFLAGS = $(CXXFLAGS) -c `pkg-config --cflags gtk+-2.0`
+CFLAGS = $(CXXFLAGS) -c -DPF=\"$(PF)\" `pkg-config --cflags gtk+-2.0`
 LIBS = `pkg-config --libs gtk+-2.0 gthread-2.0`

