Index: fotoxx-11.08.1/Makefile
===================================================================
--- fotoxx-11.08.1.orig/Makefile
+++ fotoxx-11.08.1/Makefile
@@ -73,7 +73,6 @@ install: fotoxx
 	cp -f data/desktop $(DESTDIR)$(MENUFILE)
 	echo Exec=/usr/bin/fotoxx >> $(DESTDIR)$(MENUFILE)
 	echo Icon=/usr/share/fotoxx/icons/fotoxx.png >> $(DESTDIR)$(MENUFILE)
-	xdg-desktop-menu install $(DESTDIR)$(MENUFILE)
 
 uninstall:
 	rm -f  $(DESTDIR)$(BINDIR)/fotoxx
