From 805a17ac786f47990473c5afa7f2c4a416c6169b Mon Sep 17 00:00:00 2001
From: Jeffrey C. Ollie <jeff@ocjtech.us>
Date: Fri, 27 Feb 2009 09:09:17 -0600
Subject: [PATCH] Include <cstdio> for snprintf.

---
 src/rtpdefines.h |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

Index: jrtplib-3.7.1/src/rtpdefines.h
===================================================================
--- jrtplib-3.7.1.orig/src/rtpdefines.h
+++ jrtplib-3.7.1/src/rtpdefines.h
@@ -76,6 +76,7 @@
 	#if (!defined(_WIN32_WCE)) && (defined(_MSC_VER) && _MSC_VER >= 1400 )
 		#define RTP_SNPRINTF _snprintf_s
 	#else
+        #include <cstdio>
 		#define RTP_SNPRINTF _snprintf
 	#endif
 #else
Index: jrtplib-3.7.1/src/rtperrors.h
===================================================================
--- jrtplib-3.7.1.orig/src/rtperrors.h
+++ jrtplib-3.7.1/src/rtperrors.h
@@ -39,6 +39,7 @@
 #define RTPERRORS_H
 
 #include <string>
+#include <cstdio>
 
 /** Returns a string describing the error code \c errcode. */
 std::string RTPGetErrorString(int errcode);
