--- id3lib-3.8.3/src/Makefile.am~	2003-03-02 08:23:00.000000000 +0800
+++ id3lib-3.8.3/src/Makefile.am	2013-06-15 09:49:10.740855455 +0800
@@ -73,8 +73,11 @@
 libid3_la_SOURCES = $(id3lib_sources)
 
 if ID3_NEEDZLIB
-LDADD        = $(top_builddir)/zlib/src/libz.la
+libid3_la_LIBADD = $(top_builddir)/zlib/src/libz.la
+else
+libid3_la_LIBADD = -lz
 endif
+libid3_la_LIBADD += -lstdc++
 
 libid3_la_LDFLAGS = \
   -version-info $(LT_VERSION) \
