diff -Nuar Masaüstü/edgar-0.0.1_55225.orig//module.desktop.in edgar-0.0.1_55225//module.desktop.in
--- Masaüstü/edgar-0.0.1_55225.orig//module.desktop.in	2011-02-09 15:00:18.000000000 +0200
+++ edgar-0.0.1_55225//module.desktop.in	2011-02-23 11:59:55.951032944 +0200
@@ -5,4 +5,5 @@
 Icon=e-module-edgar
 Comment=<title>Edgar Gadget Loader</title><br>Load simple edje files as E17 gadgets.
 Comment[cs]=<title>Loader gadgetů Edgar</title><br>Umožňuje použít edje soubory jako E17 gadgety.
+Comment[tr]=<title>Edgar Araç Yükleyici</title><br>Basit edje dosyalarını yüklemek ve yönetmek için kullanılan E17 modülüdür.
 X-Enlightenment-ModuleType=system
