diff -Nuar acpica-unix-20100428.orig//compiler/Makefile acpica-unix-20100428//compiler/Makefile
--- acpica-unix-20100428.orig//compiler/Makefile	2010-04-28 19:32:35.000000000 +0000
+++ acpica-unix-20100428//compiler/Makefile	2010-05-04 03:15:17.498007404 +0000
@@ -1,7 +1,7 @@
  
 PROG=	iasl
 SRCS=	\
-	aslcompilerparse.c \
+	aslcompiler.y.c \
 	aslcompilerlex.c \
 	aslanalyze.c \
 	aslcodegen.c \
@@ -117,6 +117,7 @@
 	../tables/tbutils.c \
 	../tables/tbxface.c \
 	../osunixxf.c
+OBJS = $(patsubst %.c,%.o, $(SRCS))
 
 NOMAN=	YES
 CFLAGS+= -Wall -O2 -Wstrict-prototypes -D_LINUX -DACPI_ASL_COMPILER -I../include 
@@ -135,17 +136,19 @@
 
 LDLIBS = -lpthread -lrt
 
-aslmain : $(patsubst %.c,%.o, $(SRCS))
+aslmain : $(OBJS)
 	$(CC) $(LDFLAGS) $(patsubst %.c,%.o, $(SRCS)) \
 		$(LOADLIBES) $(LDLIBS) -o iasl
 
++$(OBJS): aslcompiler.y.h
+
 CLEANFILES= y.output y.tab.c y.tab.h aslcompiler.y.h \
-	aslcompilerparse.c aslcompilerlex.c iasl
+	aslcompiler.y.c aslcompilerlex.c iasl
+
+aslcompiler.y.h: aslcompiler.y.c
 
-aslcompilerparse.c: aslcompiler.y
-	${YACC} ${YFLAGS} aslcompiler.y
-	cp y.tab.c aslcompilerparse.c
-	cp y.tab.h aslcompiler.y.h
+aslcompiler.y.c: aslcompiler.y
+	${YACC} ${YFLAGS} aslcompiler.y -o aslcompiler.y.c
 
 aslcompilerlex.c: aslcompiler.l
 	${LEX} ${LFLAGS} -PAslCompiler -oaslcompilerlex.c aslcompiler.l
