Index: work/gpodder-2.15/data/gpodder.desktop
===================================================================
--- gpodder-2.15/data/gpodder.desktop
+++ gpodder-2.15/data/gpodder.desktop
@@ -8,6 +8,7 @@ Name[fr]=Client de podcasts gPodder
 Name[gl]=Cliente Podcast gPodder
 Name[it]=Client Podcast gPodder
 Name[ru]=gPodder Подкаст-Клиент
+Name[tr]=gPodder Podcast istemcisi
 Name[uk]=gPodder Подкаст-Клієнт
 Name[cs]=gPodder podcast čtečka
 Name[pt]=Cliente de Podcasts gPodder
@@ -21,6 +22,7 @@ GenericName[fr]=Agrégateur de média
 GenericName[gl]=Agregador de medios
 GenericName[it]=Aggregatore di media
 GenericName[ru]=Загрузка, прослушивание подкастов
+GenericName[tr]=Medya Kaynağı
 GenericName[uk]=Завантаження та прослуховування подкастів
 GenericName[cs]=Agregátor zpráv
 GenericName[pt]=Agregador de média
@@ -34,6 +36,7 @@ Comment[fr]=Télécharger des podcasts a
 Comment[gl]=Descargue contidos audio/vídeo desde podcasts
 Comment[it]=Scarica contenuti audio e video dai podcast
 Comment[ru]=Загрузка и воспроизведение аудио и видео подкастов, синхронизация с iPod
+Comment[tr]=Podcastlerden ses ve video içeriği indirin
 Comment[uk]=Завантаження та прослуховування аудіо та відео подкастів, синхронізація з iPod
 Comment[cs]=Umožňuje stahování audio/video podcastu a synchornizaci s MP3 přehrávačem
 Comment[pt]=Descarrega conteúdos de podcasts.
