diff -up ntp-4.2.4p7/ntpd/ntp_proto.c.getprecision ntp-4.2.4p7/ntpd/ntp_proto.c
--- ntp-4.2.4p7/ntpd/ntp_proto.c.getprecision	2009-09-29 14:16:22.000000000 +0200
+++ ntp-4.2.4p7/ntpd/ntp_proto.c	2009-09-29 14:18:13.000000000 +0200
@@ -3099,7 +3099,7 @@ peer_unfit(
 /*
  * Find the precision of this particular machine
  */
-#define MINSTEP 100e-9		/* minimum clock increment (s) */
+#define MINSTEP 10e-9		/* minimum clock increment (s) */
 #define MAXSTEP 20e-3		/* maximum clock increment (s) */
 #define MINLOOPS 5		/* minimum number of step samples */
 
