diff -up ntp-4.2.6p2/lib/isc/unix/interfaceiter.c.tentative ntp-4.2.6p2/lib/isc/unix/interfaceiter.c
--- ntp-4.2.6p2/lib/isc/unix/interfaceiter.c.tentative	2010-06-12 08:13:53.000000000 +0200
+++ ntp-4.2.6p2/lib/isc/unix/interfaceiter.c	2010-07-12 16:25:43.000000000 +0200
@@ -45,6 +45,8 @@
 #include <isc/types.h>
 #include <isc/util.h>
 
+#include <linux/rtnetlink.h>
+
 /* Must follow <isc/net.h>. */
 #ifdef HAVE_NET_IF6_H
 #include <net/if6.h>
@@ -252,6 +254,10 @@ linux_if_inet6_current(isc_interfaceiter
 		iter->current.flags |= INTERFACE_F_MULTICAST;
 #endif
 
+	/* ignore tentative address */
+	if (flags & IFA_F_TENTATIVE)
+		iter->current.flags &= ~INTERFACE_F_UP;
+
 	isc_netaddr_fromin6(&iter->current.address, &addr6);
 	if (isc_netaddr_islinklocal(&iter->current.address)) {
 		isc_netaddr_setzone(&iter->current.address,
