diff -up ntp-4.2.4p7/ntpstat-0.2/ntpstat.c.ntpstat ntp-4.2.4p7/ntpstat-0.2/ntpstat.c
--- ntpstat-0.2/ntpstat.c.ntpstat	2002-06-10 08:02:12.000000000 +0200
+++ ntpstat-0.2/ntpstat.c	2009-07-20 12:22:35.000000000 +0200
@@ -151,7 +151,7 @@ int main (void) {
   /* For the reply message to be valid, the first byte should be as sent, 
      and the second byte should be the same, with the response bit set */
   byte1ok = ((ntpmsg.byte1&0x3F) == B1VAL);
-  byte2ok = (ntpmsg.byte2 == (B2VAL|RMASK));
+  byte2ok = ((ntpmsg.byte2 & ~MMASK) == (B2VAL|RMASK));
   if (!(byte1ok && byte2ok)) {
     fprintf (stderr,"status word is 0x%02x%02x\n", ntpmsg.byte1,ntpmsg.byte2 );
     die ("return data appears to be invalid based on status word");
