Description: Fix gcc-4.7 FTBFS

Index: libcommoncpp2-1.8.1/src/applog.cpp
===================================================================
--- libcommoncpp2-1.8.1.orig/src/applog.cpp	2010-11-01 10:11:55.000000000 +1100
+++ libcommoncpp2-1.8.1/src/applog.cpp	2012-05-19 10:11:19.000000000 +1000
@@ -45,6 +45,9 @@
 #include <cstdlib>
 #include <stdarg.h>
 #include <errno.h>
+#include <fcntl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 
 // TODO sc: test if has to move up now that it is into commoncpp
 // NOTE: the order of inclusion is important do not move following include line
@@ -297,7 +300,7 @@
     else
     {
       // create pipe
-      int err = mkfifo(_nomeFile.c_str(), S_IREAD | S_IWRITE);
+      int err = mkfifo(_nomeFile.c_str(), S_IRUSR | S_IWUSR);
       if (err == 0 || errno == EEXIST)
       {
         // and open it
@@ -342,7 +345,7 @@
     else
     {
       // create pipe
-      int err = mkfifo(_nomeFile.c_str(), S_IREAD | S_IWRITE);
+      int err = mkfifo(_nomeFile.c_str(), S_IRUSR | S_IWUSR);
       if (err == 0 || errno == EEXIST)
       {
         // and open it
@@ -456,7 +459,7 @@
     else
     {
       // create pipe
-      int err = mkfifo(d->_nomeFile.c_str(), S_IREAD | S_IWRITE);
+      int err = mkfifo(d->_nomeFile.c_str(), S_IRUSR | S_IWUSR);
       if (err == 0 || errno == EEXIST)
       {
         // and open it
@@ -562,7 +565,7 @@
     else
     {
       // create pipe
-      int err = mkfifo(d->_nomeFile.c_str(), S_IREAD | S_IWRITE);
+      int err = mkfifo(d->_nomeFile.c_str(), S_IRUSR | S_IWUSR);
       if (err == 0 || errno == EEXIST)
       {
         // and open it
