diff -Nur commoncpp2-1.8.1.orig/src/applog.cpp commoncpp2-1.8.1/src/applog.cpp
--- commoncpp2-1.8.1.orig/src/applog.cpp	2010-10-31 17:11:55.000000000 -0600
+++ commoncpp2-1.8.1/src/applog.cpp	2012-01-22 19:23:13.639292789 -0700
@@ -45,6 +45,7 @@
 #include <cstdlib>
 #include <stdarg.h>
 #include <errno.h>
+#include <sys/stat.h>
 
 // TODO sc: test if has to move up now that it is into commoncpp
 // NOTE: the order of inclusion is important do not move following include line
@@ -297,7 +298,7 @@
     else
     {
       // create pipe
-      int err = mkfifo(_nomeFile.c_str(), S_IREAD | S_IWRITE);
+      int err = mkfifo(_nomeFile.c_str(), S_IRUSR | S_IWUSR);
       if (err == 0 || errno == EEXIST)
       {
         // and open it
@@ -342,7 +343,7 @@
     else
     {
       // create pipe
-      int err = mkfifo(_nomeFile.c_str(), S_IREAD | S_IWRITE);
+      int err = mkfifo(_nomeFile.c_str(), S_IRUSR | S_IWUSR);
       if (err == 0 || errno == EEXIST)
       {
         // and open it
@@ -456,7 +457,7 @@
     else
     {
       // create pipe
-      int err = mkfifo(d->_nomeFile.c_str(), S_IREAD | S_IWRITE);
+      int err = mkfifo(d->_nomeFile.c_str(), S_IRUSR | S_IWUSR);
       if (err == 0 || errno == EEXIST)
       {
         // and open it
@@ -562,7 +563,7 @@
     else
     {
       // create pipe
-      int err = mkfifo(d->_nomeFile.c_str(), S_IREAD | S_IWRITE);
+      int err = mkfifo(d->_nomeFile.c_str(), S_IRUSR | S_IWUSR);
       if (err == 0 || errno == EEXIST)
       {
         // and open it