=== modified file 'configure.in'
--- configure.in	2010-04-02 04:52:17 +0000
+++ configure.in	2010-04-02 04:52:22 +0000
@@ -291,7 +291,12 @@
 dnl XXX This could be done for cross-compiling, but for now it's not.
 dnl
 if test -z "$with_pcap" && test "$cross_compiling" = yes; then
-	AC_MSG_ERROR(pcap type not determined when cross-compiling; use --with-pcap=...)
+	if test -z "$with_pcap" ; then
+		case $host in
+			*-linux*) with_pcap="linux";;
+			*) AC_MSG_ERROR(pcap type not determined when cross-compiling; use --with-pcap=...);;
+		esac
+	fi
 fi
 AC_ARG_WITH(pcap,
 AC_HELP_STRING([--with-pcap=TYPE],[use packet capture TYPE]))
@@ -412,8 +417,10 @@
 linux)
 	AC_MSG_CHECKING(Linux kernel version)
  	if test "$cross_compiling" = yes; then
+		dnl we could check linux/version.h here, but who runs
+		dnl versions of linux older than 2.0.x anymore to bother ?
  		AC_CACHE_VAL(ac_cv_linux_vers,
- 		    ac_cv_linux_vers=unknown)
+		    ac_cv_linux_vers=2)
  	else
  		AC_CACHE_VAL(ac_cv_linux_vers,
  		    ac_cv_linux_vers=`uname -r 2>&1 | \

