diff -p -up plymouth-0.7.2/src/libply/Makefile.am.lib_subst~ plymouth-0.7.2/src/libply/Makefile.am
--- plymouth-0.7.2/src/libply/Makefile.am.lib_subst~	2009-12-07 00:32:33.815065434 -0500
+++ plymouth-0.7.2/src/libply/Makefile.am	2009-12-07 00:33:08.951819357 -0500
@@ -4,7 +4,7 @@ INCLUDES = -I$(top_srcdir)              
            -DPLYMOUTH_TIME_DIRECTORY=\"$(localstatedir)/lib/plymouth/\"       
 
 if WITH_SYSTEM_ROOT_INSTALL
-librarydir = $(libdir:/usr%=%)
+librarydir = $(subst /usr/lib,/lib,$(libdir))
 else
 librarydir = $(libdir)
 endif
