diff -p -up plymouth-0.7.2/src/libply/ply-utils.h.orig~ plymouth-0.7.2/src/libply/ply-utils.h
--- plymouth-0.7.2/src/libply/ply-utils.h.orig~	2009-12-04 19:43:43.295048415 -0500
+++ plymouth-0.7.2/src/libply/ply-utils.h	2009-12-04 19:43:46.498548701 -0500
@@ -25,6 +25,7 @@
 #include <stdint.h>
 #include <stdbool.h>
 #include <unistd.h>
+#include <sys/types.h>
 
 #ifndef MIN
 #define MIN(a,b) ((a) <= (b)? (a) : (b))
