diff -Nuar clamtk-5.18/lib/GUI.pm clamtk-5.18_new/lib/GUI.pm
--- clamtk-5.18/lib/GUI.pm	2015-05-10 13:13:16.000000000 +0300
+++ clamtk-5.18_new/lib/GUI.pm	2015-05-19 18:10:10.978867585 +0300
@@ -261,17 +261,17 @@
         },
         {   link        => _( 'Whitelist' ),
             description => _( 'View or update scanning whitelist' ),
-            image       => 'gtk-new',
+            image       => 'document-new',
             button      => FALSE,
         },
         {   link        => _( 'Network' ),
             description => _( 'Edit proxy settings' ),
-            image       => 'gtk-network',
+            image       => 'network-server',
             button      => FALSE,
         },
         {   link        => _( 'Scheduler' ),
             description => _( 'Schedule a scan or signature update' ),
-            image       => 'gtk-properties',
+            image       => 'document-properties',
             button      => FALSE,
         },
     );
@@ -342,7 +342,7 @@
     my @data = (
         {   link        => _( 'Update' ),
             description => _( 'Update antivirus signatures' ),
-            image       => 'gtk-goto-bottom',
+            image       => 'go-bottom',
             button      => FALSE,
         },
         {   link        => _( 'Update Assistant' ),
@@ -423,7 +423,7 @@
         },
         {   link        => _( 'Quarantine' ),
             description => _( 'Manage quarantined files' ),
-            image       => 'gtk-refresh',
+            image       => 'document-revert',
             button      => FALSE,
         },
     );
@@ -495,12 +495,12 @@
     my @centos_data = (
         {   link        => _( 'Scan a file' ),
             description => _( 'Scan a file' ),
-            image       => 'gtk-file',
+            image       => 'x-office-document',
             button      => FALSE,
         },
         {   link        => _( 'Scan a directory' ),
             description => _( 'Scan a directory' ),
-            image       => 'gtk-directory',
+            image       => 'folder',
             button      => FALSE,
         },
     );
@@ -521,7 +521,7 @@
     my @data = (
         {   link        => _( 'Analysis' ),
             description => _( "View a file's reputation" ),
-            image       => 'gtk-find',
+            image       => 'system-search',
             button      => FALSE,
         },
     );
