diff -Naur perl-5.12.1.orig/cpan/Module-Build/t/compat.t perl-5.12.1/cpan/Module-Build/t/compat.t
--- perl-5.12.1.orig/cpan/Module-Build/t/compat.t	2010-04-26 12:03:52.000000000 +0300
+++ perl-5.12.1/cpan/Module-Build/t/compat.t	2010-08-18 11:28:27.619000563 +0300
@@ -7,6 +7,8 @@
 use IO::File;
 use Config;
 
+plan skip_all => "Skip for sandbox violation";
+
 # Don't let our own verbosity/test_file get mixed up with our subprocess's
 my @makefile_keys = qw(TEST_VERBOSE HARNESS_VERBOSE TEST_FILES MAKEFLAGS);
 local  @ENV{@makefile_keys};
