--- perl-5.16.0-RC2/utils/perlbug.PL.fedora	2012-05-16 16:15:51.000000000 +0200
+++ perl-5.16.0-RC2/utils/perlbug.PL	2012-05-16 16:18:36.018894464 +0200
@@ -271,17 +271,6 @@ sub Init {
     $ok = '';
     if ($opt{o}) {
 	if ($opt{o} eq 'k' or $opt{o} eq 'kay') {
-	    my $age = time - $patchlevel_date;
-	    if ($opt{o} eq 'k' and $age > 60 * 24 * 60 * 60 ) {
-		my $date = localtime $patchlevel_date;
-		print <<"EOF";
-"perlbug -ok" and "perlbug -nok" do not report on Perl versions which
-are more than 60 days old.  This Perl version was constructed on
-$date.  If you really want to report this, use
-"perlbug -okay" or "perlbug -nokay".
-EOF
-		exit();
-	    }
 	    # force these options
 	    unless ($opt{n}) {
 		$opt{S} = 1; # don't prompt for send