diff -Naur lib/util.py lib.modified/util.py
--- lib/util.py	2007-12-09 19:37:16.000000000 +0200
+++ lib.modified/util.py	2011-10-01 10:40:31.238000001 +0300
@@ -29,7 +29,7 @@
 class Util:
 
   pygame.font.init()
-  smallfont = pygame.font.Font(data.filepath(os.path.join("misc", "Vera.ttf")), FONT_SIZE)
+  smallfont = pygame.font.Font("/usr/share/fonts/bitstream-vera/Vera.ttf", FONT_SIZE)
   cached_text_images = {}
   cached_images = {}
   cached_images["key_z"] = pygame.image.load(data.picpath("key", "z"))
