From e6ed635e07bd485c6518894bd9883ae9df64f63f Mon Sep 17 00:00:00 2001
From: Jasmine Hassan <jasmine.aura@gmail.com>
Date: Sat, 15 Dec 2012 06:21:06 +0000
Subject: [keyboard] fix g_format_size_for_display deprecation (GLIB 2.30)

---
diff --git a/plugins/housekeeping/msd-ldsm-dialog.c b/plugins/housekeeping/msd-ldsm-dialog.c
index ea95540..0624448 100644
--- a/plugins/housekeeping/msd-ldsm-dialog.c
+++ b/plugins/housekeeping/msd-ldsm-dialog.c
@@ -17,11 +17,16 @@
  * with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <glib.h>
 #include <glib/gi18n.h>
 #include <gio/gio.h>
 
 #include "msd-ldsm-dialog.h"
 
+#if GLIB_CHECK_VERSION (2, 30, 0)
+#define g_format_size_for_display g_format_size
+#endif
+
 #define SETTINGS_SCHEMA "org.mate.SettingsDaemon.plugins.housekeeping"
 #define SETTINGS_IGNORE_PATHS "ignore-paths"
 
--
cgit v0.9.0.3-65-g4555
