diff -up libnl-1.1/doc/Doxyfile.in.srclink libnl-1.1/doc/Doxyfile.in
--- libnl-1.1/doc/Doxyfile.in.srclink	2008-06-24 16:37:11.000000000 +0900
+++ libnl-1.1/doc/Doxyfile.in	2008-06-24 16:37:39.000000000 +0900
@@ -580,7 +580,7 @@ SOURCE_BROWSER         = YES
 # Setting the INLINE_SOURCES tag to YES will include the body 
 # of functions and classes directly in the documentation.
 
-INLINE_SOURCES         = NO
+INLINE_SOURCES         = YES
 
 # Setting the STRIP_CODE_COMMENTS tag to YES (the default) will instruct 
 # doxygen to hide any special comment blocks from generated source code 
