diff -up libnl-1.0-pre8/lib/Makefile.orig libnl-1.0-pre8/lib/Makefile
--- libnl-1.0-pre8/lib/Makefile.orig	2007-11-21 12:25:39.000000000 -0500
+++ libnl-1.0-pre8/lib/Makefile	2007-12-03 14:08:43.000000000 -0500
@@ -32,6 +32,7 @@ DEPS     := $(CIN:%.c=%.d)
 OBJ      := $(CIN:%.c=%.o)
 CFLAGS   += -fPIC
 OUT_SLIB := $(PACKAGE_NAME).so.$(PACKAGE_VERSION)
+OUT_AR   := $(PACKAGE_NAME).a
 LN_SLIB  := $(PACKAGE_NAME).so
 LN1_SLIB := $(LN_SLIB).1
 
@@ -42,7 +43,8 @@ export
 
 all:
 	@echo "  MAKE $(OUT_SLIB)"; \
-	$(MAKE) $(OUT_SLIB)
+	$(MAKE) $(OUT_SLIB); \
+	$(MAKE) $(OUT_AR)
 
 $(OUT_SLIB): ../Makefile.opts $(OBJ)
 	@echo "  LD $(OUT_SLIB)"; \
@@ -52,6 +54,10 @@ $(OUT_SLIB): ../Makefile.opts $(OBJ)
 	@echo "  LN $(LN1_SLIB) $(LN_SLIB)"; \
 	rm -f $(LN_SLIB) ; $(LN) -s $(LN1_SLIB) $(LN_SLIB)
 
+$(OUT_AR):  ../Makefile.opts $(OBJ)
+	@echo "  AR $(OUT_AR)"; \
+	$(AR) crus $(OUT_AR) $(OBJ)
+
 clean:
 	@echo "  CLEAN lib"; \
 	$(RM) -f $(OBJ) $(OUT_SLIB) $(LN_SLIB) $(LN1_SLIB); \
@@ -68,6 +74,7 @@ install:
 	$(LN) -s $(OUT_SLIB) $(DESTDIR)$(libdir)/$(LN1_SLIB)
 	rm -f $(DESTDIR)$(libdir)/$(LN_SLIB)
 	$(LN) -s $(LN1_SLIB) $(DESTDIR)$(libdir)/$(LN_SLIB)
+	install -m 0644 $(OUT_AR) $(DESTDIR)$(libdir)
 
 $(DEPS): ../Makefile.opts
 
