--- src/main/org/apache/tools/ant/util/JavaEnvUtils.java.orig
+++ src/main/org/apache/tools/ant/util/JavaEnvUtils.java
@@ -73,6 +73,9 @@
     /** Whether this is the Kaffe VM */
     private static boolean kaffeDetected;
 
+    /** Wheter this is a GNU Classpath based VM */
+    private static boolean classpathDetected;
+
     /** array of packages in the runtime */
     private static Vector jrePackages;
 
@@ -119,6 +122,13 @@
         } catch (Throwable t) {
             // swallow as this simply doesn't seem to be Kaffe
         }
+	classpathDetected = false;
+	try {
+	    Class.forName("gnu.classpath.Configuration");
+	    classpathDetected = true;
+	} catch (Throwable t) {
+            // swallow as this simply doesn't seem to be GNU classpath based.
+	}
     }
 
     /**
@@ -179,6 +189,10 @@
         return kaffeDetected;
     }
 
+    public static boolean isClasspathBased() {
+        return classpathDetected;
+    }
+
     /**
      * Finds an executable that is part of a JRE installation based on
      * the java.home system property.
--- src/main/org/apache/tools/ant/taskdefs/optional/native2ascii/KaffeNative2Ascii.java.orig
+++ src/main/org/apache/tools/ant/taskdefs/optional/native2ascii/KaffeNative2Ascii.java
@@ -32,6 +32,7 @@
 
     // sorted by newest Kaffe version first
     private static final String[] N2A_CLASSNAMES = new String[] {
+        "gnu.classpath.tools.native2ascii.Native2ASCII",
         "gnu.classpath.tools.native2ascii.Native2Ascii",
         // pre Kaffe 1.1.5
         "kaffe.tools.native2ascii.Native2Ascii",
--- src/main/org/apache/tools/ant/taskdefs/optional/native2ascii/Native2AsciiAdapterFactory.java.orig
+++ src/main/org/apache/tools/ant/taskdefs/optional/native2ascii/Native2AsciiAdapterFactory.java
@@ -38,7 +38,7 @@
      * vendor
      */
     public static String getDefault() {
-        if (JavaEnvUtils.isKaffe()) {
+        if (JavaEnvUtils.isKaffe() || JavaEnvUtils.isClasspathBased()) {
             return KaffeNative2Ascii.IMPLEMENTATION_NAME;
         }
         return SunNative2Ascii.IMPLEMENTATION_NAME;
@@ -57,7 +57,7 @@
     public static Native2AsciiAdapter getAdapter(String choice,
                                                  ProjectComponent log)
         throws BuildException {
-        if ((JavaEnvUtils.isKaffe() && choice == null)
+        if (((JavaEnvUtils.isKaffe() || JavaEnvUtils.isClasspathBased()) && choice == null)
             || KaffeNative2Ascii.IMPLEMENTATION_NAME.equals(choice)) {
             return new KaffeNative2Ascii();
         } else if (SunNative2Ascii.IMPLEMENTATION_NAME.equals(choice)) {
