diff -Nuar a/Makefile.in b/Makefile.in
--- a/Makefile.in	2011-04-13 18:43:32.000000000 +0300
+++ b/Makefile.in	2013-10-29 20:28:28.582611922 +0200
@@ -297,7 +297,7 @@
 
 libmxml.so.1.5:	$(LIBOBJS)
 	echo Creating $@...
-	$(DSO) $(DSOFLAGS) -o libmxml.so.1.5 $(LIBOBJS)
+	$(DSO) $(DSOFLAGS) -o libmxml.so.1.5 $(LIBOBJS) $(LIBS)
 	$(RM) libmxml.so libmxml.so.1
 	$(LN) libmxml.so.1.5 libmxml.so
 	$(LN) libmxml.so.1.5 libmxml.so.1
