diff -up xf86-video-v4l-0.2.0/src/v4l.c~ xf86-video-v4l-0.2.0/src/v4l.c
--- xf86-video-v4l-0.2.0/src/v4l.c~	2015-02-11 17:42:55.000000000 +0100
+++ xf86-video-v4l-0.2.0/src/v4l.c	2015-02-11 17:44:33.432706357 +0100
@@ -42,6 +42,10 @@
 # define DEBUG(x)
 #endif
 
+#ifndef VIDEO_INVERT_CLIPLIST
+#define VIDEO_INVERT_CLIPLIST 0x00000002
+#endif
+
 /***************************************************************************/
 
 static void     V4LIdentify(int flags);
