--- Examples/Client/svn_cmd.py.orig	2010-11-11 16:18:10.419000184 +0200
+++ Examples/Client/svn_cmd.py	2010-11-11 16:18:30.060000183 +0200
@@ -70,7 +70,7 @@
             locale.setlocale( locale.LC_ALL, '%s.%s' % (language_code, encoding) )
         except locale.Error:
             # force a locale that will work
-            locale.setlocale( locale.LC_ALL, 'en_GB.UTF-8' )
+            locale.setlocale( locale.LC_ALL, 'en_US.UTF-8' )
 
 def fmtDateTime( t ):
     return time.strftime( '%d-%b-%Y %H:%M:%S', time.localtime( t ) )
