--- xfmpc.desktop.in.orig	2008-07-06 20:41:36.000000000 +0300
+++ xfmpc.desktop.in	2008-07-06 20:42:32.000000000 +0300
@@ -1,8 +1,11 @@
 [Desktop Entry]
 Version=1.0
-_Name=Xfmpc
-_Comment=MPD client written in GTK+ for Xfce
-_GenericName=MPD client
+Name=Xfmpc
+Name[tr]=Xfmpc
+Comment=MPD client written in GTK+ for Xfce
+Comment[tr]=Xfce için, GTK+ ile yazılmış MPD istemcisi
+GenericName=MPD client
+GenericName[tr]=MPD istemcisi
 Exec=xfmpc
 Icon=stock_volume
 Terminal=false
