--- gen.c.orig	2008-12-17 09:36:40.000000000 +0200
+++ gen.c	2008-12-17 09:38:00.000000000 +0200
@@ -1891,7 +1891,7 @@
 			outn ("\t\t{ \\");
 			outn ("\t\tint c = '*'; \\");
 			outn ("\t\tint n; \\");
-			outn ("\t\tfor ( n = 0; n < max_size && \\");
+			outn ("\t\tfor ( n = 0; n < (size_t) max_size && \\");
 			outn ("\t\t\t     (c = getc( yyin )) != EOF && c != '\\n'; ++n ) \\");
 			outn ("\t\t\tbuf[n] = (char) c; \\");
 			outn ("\t\tif ( c == '\\n' ) \\");
