diff -Nur flex-2.5.35-old/Makefile.in flex-2.5.35/Makefile.in
--- flex-2.5.35-old/Makefile.in	2009-11-15 21:49:05.186179663 +0200
+++ flex-2.5.35/Makefile.in	2009-11-15 21:49:12.023181070 +0200
@@ -1056,7 +1056,9 @@
 scanflags.o: scanflags.c flexdef.h flexint.h
 gen.o: gen.c flexdef.h flexint.h tables.h tables_shared.h
 libmain.o: libmain.c
+	$(COMPILE) -fPIC -DPIC -o libmain.o -c libmain.c
 libyywrap.o: libyywrap.c
+	$(COMPILE) -fPIC -DPIC -o libyywrap.o -c libyywrap.c
 main.o: main.c flexdef.h flexint.h version.h options.h scanopt.h \
  tables.h tables_shared.h
 misc.o: misc.c flexdef.h flexint.h tables.h tables_shared.h
