--- pcmanfm-qt-desktop-pref.desktop.in	2015-02-05 11:58:53.000000000 +0200
+++ pcmanfm/pcmanfm-qt-desktop-pref.desktop.in	2015-03-28 01:30:15.327778850 +0200
@@ -1,9 +1,12 @@
 [Desktop Entry]
 Type=Application
-Name=Desktop
-GenericName=Desktop Settings
-Comment=Change wallpapers and behavior of desktop manager
-Exec=pcmanfm-qt --desktop-pref=general
 Icon=user-desktop
-Categories=Qt;DesktopSettings;
+Name=Desktop Preferences
+Name[tr]= Masaüstü Tercihleri
+Comment=Change desktop wallpapers and behavior of desktop manager
+Comment[tr]=Masaüstü yöneticisinin davranışların ve duvar kağıdını değiştirin
+Categories=Settings;Qt;DesktopSettings;
+Exec=/usr/bin/pcmanfm-qt --desktop-pref=general
 StartupNotify=true
+Terminal=false
+NotShowIn=GNOME;XFCE;KDE;
