--- pcmanfm-qt.desktop.in	2015-02-05 11:58:53.000000000 +0200
+++ pcmanfm/pcmanfm-qt.desktop.in	2015-03-28 01:24:09.874791027 +0200
@@ -1,10 +1,16 @@
 [Desktop Entry]
+Encoding=UTF-8
 Type=Application
+Icon=system-file-manager
 Name=PCManFM File Manager
+Name[tr]=PCManFM Dosya Yöneticisi
 GenericName=File Manager
+GenericName[tr]=Dosya Yöneticisi
 Comment=Browse the file system and manage the files
-Exec=pcmanfm-qt %U
-MimeType=inode/directory;
-Icon=system-file-manager
+Comment[tr]=Dosya sistemine göz atın ve dosyaları yönetin
-Categories=FileManager;Utility;Core;Qt;
+Categories=FileManager;System;Core;Qt;
+Exec=/usr/bin/pcmanfm-qt
 StartupNotify=true
+Terminal=false
+MimeType=inode/directory;
+NotShowIn=GNOME;XFCE;KDE;
