diff -up esound-0.2.41/esd_config.c.nospawn esound-0.2.41/esd_config.c
--- esound-0.2.41/esd_config.c.nospawn	2009-03-31 23:57:35.860368388 -0400
+++ esound-0.2.41/esd_config.c	2009-03-31 23:57:51.032658859 -0400
@@ -6,7 +6,7 @@
 #include <ctype.h>
 #include <string.h>
 
-int esd_no_spawn=0; /* If we can't even find the system config file,
+int esd_no_spawn=1; /* If we can't even find the system config file,
 		       things are screwed up - don't try to make things
 		       worse. */
 int esd_spawn_wait_ms=100; /* Time to wait trying to connect to an
