Index: xdg-utils-1.0.2.20100618/scripts/Makefile.in
===================================================================
--- xdg-utils-1.0.2.20100618.orig/scripts/Makefile.in
+++ xdg-utils-1.0.2.20100618/scripts/Makefile.in
@@ -5,7 +5,7 @@ INSTALL_PROGRAM = @INSTALL_PROGRAM@ $(IN
 INSTALL_LIBRARY = @INSTALL_PROGRAM@ $(INSTALL_PROGRAM_FLAGS)
 INSTALL_SCRIPT  = @INSTALL_SCRIPT@ $(INSTALL_SCRIPT_FLAGS)
 INSTALL_DATA    = @INSTALL_DATA@ $(INSTALL_DATA_FLAGS)
-XMLTO           = @XMLTO@
+XMLTO           = @XMLTO@ --skip-validation
 
 prefix          = @prefix@
 exec_prefix     = @exec_prefix@
