Index: xdg-utils-1.0.2.20100709/scripts/xdg-terminal.in
===================================================================
--- xdg-utils-1.0.2.20100709.orig/scripts/xdg-terminal.in
+++ xdg-utils-1.0.2.20100709/scripts/xdg-terminal.in
@@ -32,7 +32,7 @@ terminal_kde()
 {
     terminal=`kreadconfig --file kdeglobals --group General --key TerminalApplication --default konsole`
 
-    terminal_exec=`which $terminal >/dev/null 2>/dev/null`
+    terminal_exec=`which $terminal 2>/dev/null`
 
     if [ -x "$terminal_exec" ]; then
         if [ x"$1" == x"" ]; then
