Index: xdg-utils-1.0.2/scripts/xdg-su
===================================================================
--- xdg-utils-1.0.2.orig/scripts/xdg-su
+++ xdg-utils-1.0.2/scripts/xdg-su
@@ -351,6 +351,26 @@ su_gnome()
             exit_failure_operation_failed
         fi
     else
+        su_gk
+    fi
+}
+
+su_gk()
+{
+    GKSU=`which gksu 2>/dev/null`
+    if [ $? -eq 0 ] ; then
+        if [ -z "$user" ] ; then
+            $GKSU "$cmd"
+        else
+            $GKSU -u "$user" "$cmd"
+        fi
+
+        if [ $? -eq 0 ]; then
+            exit_success
+        else
+            exit_failure_operation_failed
+        fi
+    else
         su_generic
     fi
 }
@@ -427,6 +447,10 @@ case "$DE" in
     su_gnome
     ;;
 
+    xfce)
+    su_gk
+    ;;
+
     generic)
     su_generic
     ;;
