Index: plplot-5.9.7/examples/python/CMakeLists.txt
=================================================================== 
--- plplot-5.9.7.orig/examples/python/CMakeLists.txt
+++ plplot-5.9.7/examples/python/CMakeLists.txt
@@ -233,7 +233,7 @@ if(CORE_BUILD)
     @ONLY
     )
 
-  set(python_location1 ${PYTHON_INSTDIR})
+  set(python_location1)
   set(python_location2)
   set(python_location3)
   set(python_location4)
