--- cmake/modules/FindQHULL.cmake.qhull	2012-10-09 17:52:40.081163335 +0800
+++ cmake/modules/FindQHULL.cmake	2012-10-09 17:52:53.431517116 +0800
@@ -9,7 +9,7 @@
 
 find_path(
   QHULL_INCLUDE_DIR
-    qhull/qhull_a.h
+    libqhull/qhull_a.h
   /usr/local/include
   /usr/include
 )
--- lib/nn/delaunay.c.qhull	2012-10-09 17:53:40.662769066 +0800
+++ lib/nn/delaunay.c	2012-10-09 17:53:48.552978244 +0800
@@ -34,7 +34,7 @@
 #include <limits.h>
 #include <float.h>
 #ifdef USE_QHULL
-#include <qhull/qhull_a.h>
+#include <libqhull/qhull_a.h>
 #else
 #include "triangle.h"
 #endif
--- src/plgridd.c.qhull	2012-10-09 17:53:54.033123572 +0800
+++ src/plgridd.c	2012-10-09 17:54:01.763328542 +0800
@@ -32,7 +32,7 @@
 
 #ifdef PL_HAVE_QHULL
 #include "../lib/nn/nn.h"
-#include <qhull/qhull_a.h>
+#include <libqhull/qhull_a.h>
 #endif
 
 // forward declarations
