--- cmake/modules/ada.cmake.gcc	2011-12-06 10:04:37.154783027 +0800
+++ cmake/modules/ada.cmake	2011-12-06 10:06:05.426687398 +0800
@@ -43,7 +43,7 @@
 endif(ENABLE_ada)
 
 if(ENABLE_ada)
-  find_library(GNAT_LIB NAMES gnat gnat-4.5 gnat-4.4 gnat-4.3 gnat-4.2 gnat-4.1)
+  find_library(GNAT_LIB NAMES gnat gnat-4.6 gnat-4.5 gnat-4.4 gnat-4.3 gnat-4.2 gnat-4.1)
   if(NOT GNAT_LIB)
     message(STATUS "WARNING: "
       "gnat library not found. Disabling ada bindings")
