Index: plplot-5.9.7/examples/c/Makefile.examples.in
===================================================================
--- plplot-5.9.7.orig/examples/c/Makefile.examples.in
+++ plplot-5.9.7/examples/c/Makefile.examples.in
@@ -25,8 +25,6 @@ CC = @CC@
 EXEEXT = @EXEEXT@
 
 @pkg_config_true@LIB_TAG = @LIB_TAG@
-@pkg_config_true@PKG_CONFIG_ENV = @PKG_CONFIG_ENV@
-@pkg_config_true@RPATHCMD = @RPATHCMD@
 
 @gcw_true@PLPLOTCANVAS_EXECUTABLES_list = \
 @gcw_true@		 plplotcanvas_demo$(EXEEXT) \
@@ -81,29 +79,29 @@ clean:
 	rm -f $(EXECUTABLES_list)
 
 @gcw_true@plplotcanvas_demo$(EXEEXT): plplotcanvas_demo.c
-@gcw_true@@pkg_config_true@	$(CC) $< -o $@ $(RPATHCMD) \
-@gcw_true@@pkg_config_true@			`$(PKG_CONFIG_ENV) pkg-config --cflags --libs plplot$(LIB_TAG)-gnome2`
+@gcw_true@@pkg_config_true@	$(CC) $< -o $@ \
+@gcw_true@@pkg_config_true@			`pkg-config --cflags --libs plplot$(LIB_TAG)-gnome2`
 @gcw_true@@pkg_config_false@	$(CC) $< -o $@ \
 @gcw_true@@pkg_config_false@			`plplot-config --cflags --libs --with-gcw`
 
 @gcw_true@plplotcanvas_animation$(EXEEXT): plplotcanvas_animation.c
-@gcw_true@@pkg_config_true@	$(CC) $< -o $@ $(RPATHCMD) \
-@gcw_true@@pkg_config_true@			`$(PKG_CONFIG_ENV) pkg-config --cflags --libs plplot$(LIB_TAG)-gnome2` \
+@gcw_true@@pkg_config_true@	$(CC) $< -o $@ \
+@gcw_true@@pkg_config_true@			`pkg-config --cflags --libs plplot$(LIB_TAG)-gnome2` \
 @gcw_true@@pkg_config_true@			`pkg-config --cflags --libs gthread-2.0`			
 @gcw_true@@pkg_config_false@	$(CC) $< -o \
 @gcw_true@@pkg_config_false@			$@ `plplot-config --cflags --libs --with-gcw` \
 @gcw_true@@pkg_config_false@			@GCWTHREAD_CFLAGS@ @GCWTHREAD_LIBS@
 
 @extXdrawable_true@@pkg_config_true@extXdrawable_demo$(EXEEXT): extXdrawable_demo.c
-@extXdrawable_true@@pkg_config_true@	$(CC) $< -o $@ $(RPATHCMD) \
-@extXdrawable_true@@pkg_config_true@	`$(PKG_CONFIG_ENV) pkg-config --cflags --libs plplot$(LIB_TAG) gtk+-x11-2.0`
+@extXdrawable_true@@pkg_config_true@	$(CC) $< -o $@ \
+@extXdrawable_true@@pkg_config_true@	`pkg-config --cflags --libs plplot$(LIB_TAG) gtk+-x11-2.0`
 
 @extcairo_true@@pkg_config_true@ext-cairo-test$(EXEEXT): ext-cairo-test.c
-@extcairo_true@@pkg_config_true@	$(CC) $< -o $@ $(RPATHCMD) \
-@extcairo_true@@pkg_config_true@	`$(PKG_CONFIG_ENV) pkg-config --cflags --libs plplot$(LIB_TAG) cairo`
+@extcairo_true@@pkg_config_true@	$(CC) $< -o $@ \
+@extcairo_true@@pkg_config_true@	`pkg-config --cflags --libs plplot$(LIB_TAG) cairo`
 
 .c$(EXEEXT):
-@pkg_config_true@	$(CC) $< -o $@ $(RPATHCMD) `$(PKG_CONFIG_ENV) pkg-config --cflags --libs plplot$(LIB_TAG)`
+@pkg_config_true@	$(CC) $< -o $@ `pkg-config --cflags --libs plplot$(LIB_TAG)`
 @pkg_config_false@	$(CC) $< -o $@ `plplot-config --cflags --libs`
 
 .SUFFIXES: .c $(EXEEXT)
Index: plplot-5.9.7/examples/c++/Makefile.examples.in
===================================================================
--- plplot-5.9.7.orig/examples/c++/Makefile.examples.in
+++ plplot-5.9.7/examples/c++/Makefile.examples.in
@@ -25,10 +25,7 @@ CXX = @CXX@
 EXEEXT = @EXEEXT@
 
 @pkg_config_true@LIB_TAG = @LIB_TAG@
-@pkg_config_true@PKG_CONFIG_ENV = @PKG_CONFIG_ENV@
-@pkg_config_true@RPATHCMD = @RPATHCMD@
 @qt_gui_true@QT_MOC_EXECUTABLE = @QT_MOC_EXECUTABLE@
-@qt_gui_true@@pkg_config_true@qt_RPATHCMD = @qt_RPATHCMD@
 
 @wxwidgets_true@@pkg_config_true@PLPLOTWXWIDGETS_EXECUTABLES_list = \
 @wxwidgets_true@@pkg_config_true@	wxPLplotDemo$(EXEEXT)
@@ -79,18 +76,18 @@ clean:
 	rm -f $(EXECUTABLES_list)
 
 @wxwidgets_true@@pkg_config_true@wxPLplotDemo$(EXEEXT): wxPLplotDemo.cpp
-@wxwidgets_true@@pkg_config_true@	$(CXX) $< -o $@ $(RPATHCMD) `$(PKG_CONFIG_ENV) pkg-config --cflags --libs plplot$(LIB_TAG)-wxwidgets`
+@wxwidgets_true@@pkg_config_true@	$(CXX) $< -o $@ `pkg-config --cflags --libs plplot$(LIB_TAG)-wxwidgets`
 
 # Note, must filter out -p options which confuse $(QT_MOC_EXECUTABLE).  Such
 # options (e.g., -pthread) tend to appear in the static driver case.
 @qt_gui_true@@pkg_config_true@moc_qt_PlotWindow.cpp: qt_PlotWindow.h
-@qt_gui_true@@pkg_config_true@	$(QT_MOC_EXECUTABLE) `$(PKG_CONFIG_ENV) pkg-config --cflags-only-I plplot$(LIB_TAG)-qt` $< -o $@
+@qt_gui_true@@pkg_config_true@	$(QT_MOC_EXECUTABLE) `pkg-config --cflags-only-I plplot$(LIB_TAG)-qt` $< -o $@
 
 @qt_gui_true@@pkg_config_true@qt_example$(EXEEXT): qt_example.cpp qt_PlotWindow.cpp moc_qt_PlotWindow.cpp
-@qt_gui_true@@pkg_config_true@	$(CXX) qt_example.cpp qt_PlotWindow.cpp moc_qt_PlotWindow.cpp -o $@ $(qt_RPATHCMD) `$(PKG_CONFIG_ENV) pkg-config --cflags --libs plplot$(LIB_TAG)-qt`
+@qt_gui_true@@pkg_config_true@	$(CXX) qt_example.cpp qt_PlotWindow.cpp moc_qt_PlotWindow.cpp -o $@ $(qt_RPATHCMD) `pkg-config --cflags --libs plplot$(LIB_TAG)-qt`
 
 .cc$(EXEEXT):
-@pkg_config_true@	$(CXX) $< -o $@ $(RPATHCMD) `$(PKG_CONFIG_ENV) pkg-config --cflags --libs plplot$(LIB_TAG)-c++`
+@pkg_config_true@	$(CXX) $< -o $@ `pkg-config --cflags --libs plplot$(LIB_TAG)-c++`
 @pkg_config_false@	$(CXX) $< -o $@ `plplot-config --cflags --libs --with-c++`
 
 .SUFFIXES: .cc $(EXEEXT)
Index: plplot-5.9.7/examples/f77/Makefile.examples.in
===================================================================
--- plplot-5.9.7.orig/examples/f77/Makefile.examples.in
+++ plplot-5.9.7/examples/f77/Makefile.examples.in
@@ -25,8 +25,6 @@ F77 = @F77@
 EXEEXT = @EXEEXT@
 
 @pkg_config_true@LIB_TAG = @LIB_TAG@
-@pkg_config_true@PKG_CONFIG_ENV = @PKG_CONFIG_ENV@
-@pkg_config_true@RPATHCMD = @RPATHCMD@
 
 EXECUTABLES_list = \
 		 x01f$(EXEEXT) \
@@ -69,7 +67,7 @@ clean:
 	rm -f $(EXECUTABLES_list)
 
 .f$(EXEEXT):
-@pkg_config_true@	$(F77) $< -o $@ $(RPATHCMD) `$(PKG_CONFIG_ENV) pkg-config --cflags --libs plplot$(LIB_TAG)-f77`
+@pkg_config_true@	$(F77) $< -o $@ `pkg-config --cflags --libs plplot$(LIB_TAG)-f77`
 @pkg_config_false@	$(F77) $< -o $@ `plplot-config --cflags --libs --with-f77`
 
 .SUFFIXES: .f $(EXEEXT)
Index: plplot-5.9.7/examples/f95/Makefile.examples.in
===================================================================
--- plplot-5.9.7.orig/examples/f95/Makefile.examples.in
+++ plplot-5.9.7/examples/f95/Makefile.examples.in
@@ -25,8 +25,6 @@ F95 = @FC@
 EXEEXT = @EXEEXT@
 
 @pkg_config_true@LIB_TAG = @LIB_TAG@
-@pkg_config_true@PKG_CONFIG_ENV = @PKG_CONFIG_ENV@
-@pkg_config_true@RPATHCMD = @RPATHCMD@
 
 EXECUTABLES_list = \
 		 x01f$(EXEEXT) \
@@ -68,7 +66,7 @@ clean:
 	rm -f $(EXECUTABLES_list)
 
 .f90$(EXEEXT):
-@pkg_config_true@	$(F95) @MODULESINCCMD@ $< -o $@ $(RPATHCMD) `$(PKG_CONFIG_ENV) pkg-config --cflags --libs plplot$(LIB_TAG)-f95`
+@pkg_config_true@	$(F95) @MODULESINCCMD@ $< -o $@ `pkg-config --cflags --libs plplot$(LIB_TAG)-f95`
 @pkg_config_false@	$(F95) @MODULESINCCMD@ $< -o $@ `plplot-config --cflags --libs --with-f95`
 
 .SUFFIXES: .f90 $(EXEEXT)
Index: plplot-5.9.7/examples/tk/Makefile.examples.in
===================================================================
--- plplot-5.9.7.orig/examples/tk/Makefile.examples.in
+++ plplot-5.9.7/examples/tk/Makefile.examples.in
@@ -24,8 +24,6 @@ CC = @CC@
 EXEEXT = @EXEEXT@
 
 @pkg_config_true@LIB_TAG = @LIB_TAG@
-@pkg_config_true@PKG_CONFIG_ENV = @PKG_CONFIG_ENV@
-@pkg_config_true@RPATHCMD = @RPATHCMD@
 
 EXECUTABLES_list = xtk01$(EXEEXT)
 # Second and fourth examples depend on itk.
@@ -37,7 +35,7 @@ clean:
 	rm -f $(EXECUTABLES_list) $(itk_EXECUTABLES_list)
 
 .c$(EXEEXT):
-@pkg_config_true@	$(CC) $< -o $@ $(RPATHCMD) `$(PKG_CONFIG_ENV) pkg-config --cflags --libs plplot$(LIB_TAG)-tcl`
+@pkg_config_true@	$(CC) $< -o $@ `pkg-config --cflags --libs plplot$(LIB_TAG)-tcl`
 @pkg_config_false@	$(CC) $< -o $@ `plplot-config --cflags --libs --with-tcl`
 
 .SUFFIXES: .c $(EXEEXT)
