From 849ea066426b8f5f6e6d1aea9804e095edca3b49 Mon Sep 17 00:00:00 2001
From: Martin Szulecki
Date: Thu, 17 May 2012 14:57:17 +0000
Subject: configure: Check if libplist Cython bindings are installed at configure time

---
diff --git a/configure.ac b/configure.ac
index 140c9bd..bab4512 100644
--- a/configure.ac
+++ b/configure.ac
@@ -83,10 +83,20 @@ else
             CYTHON=false
 fi
 if [test "x$CYTHON" != "xfalse"]; then
-            CYTHON_SUB=cython
+            AC_MSG_CHECKING([for libplist Cython bindings])
             CYTHON_PLIST_INCLUDE_DIR=$($PKG_CONFIG --variable=includedir libplist)/plist/cython
-            AC_SUBST([CYTHON_PLIST_INCLUDE_DIR])
-            cython_python_bindings=yes
+            if [test ! -d "$CYTHON_PLIST_INCLUDE_DIR"]; then
+                CYTHON=false
+                CYTHON_SUB=
+                cython_python_bindings=no
+                AC_MSG_RESULT([no])
+                AC_MSG_WARN([cannot find libplist Cython bindings. You should install your distribution specific libplist Cython bindings package.])
+            else
+                AC_SUBST([CYTHON_PLIST_INCLUDE_DIR])
+                AC_MSG_RESULT([$CYTHON_PLIST_INCLUDE_DIR])
+                CYTHON_SUB=cython
+                cython_python_bindings=yes
+            fi
 else
             CYTHON_SUB=
             cython_python_bindings=no
--
cgit v0.9.1-1-gc6f5
