Index: tools/tools.pro
===================================================================
--- tools/tools.pro.orig
+++ tools/tools.pro
@@ -37,6 +37,8 @@ mac {
 
 embedded:SUBDIRS += kmap2qmap
 
+SUBDIRS += qvfb
+
 contains(QT_CONFIG, dbus):SUBDIRS += qdbus
 # We don't need these command line utilities on embedded platforms.
 !wince*:!symbian:contains(QT_CONFIG, xmlpatterns): SUBDIRS += xmlpatterns xmlpatternsvalidator
