diff -ur gedit-2.30.4.orig//configure.ac gedit-2.30.4/configure.ac
--- gedit-2.30.4.orig//configure.ac	2010-09-27 22:25:12.000000000 +0300
+++ gedit-2.30.4/configure.ac	2010-11-23 17:14:01.834000001 +0200
@@ -155,8 +155,6 @@
 
 IT_PROG_INTLTOOL([0.40.0])
 AC_SUBST([GETTEXT_PACKAGE], [gedit])
-AM_GNU_GETTEXT_VERSION([0.17])
-AM_GLIB_GNU_GETTEXT
 AC_DEFINE([GETTEXT_PACKAGE],[PACKAGE_TARNAME], [Define to the gettext package name.])
 
 dnl ================================================================
