Index: libfprint-0.3.0/libfprint/Makefile.am
===================================================================
--- libfprint-0.3.0.orig/libfprint/Makefile.am
+++ libfprint-0.3.0/libfprint/Makefile.am
@@ -77,7 +77,7 @@ fprint_list_udev_rules_CFLAGS = -fvisibi
 fprint_list_udev_rules_LDADD = $(builddir)/libfprint.la
 
 udev_rules_DATA = 60-fprint-autosuspend.rules
-udev_rulesdir = $(sysconfdir)/udev/rules.d/
+udev_rulesdir = /lib/udev/rules.d/
 
 $(udev_rules_DATA): fprint-list-udev-rules
 	$(builddir)/fprint-list-udev-rules > $@
Index: libfprint-0.3.0/libfprint/Makefile.in
===================================================================
--- libfprint-0.3.0.orig/libfprint/Makefile.in
+++ libfprint-0.3.0/libfprint/Makefile.in
@@ -410,7 +410,7 @@ fprint_list_udev_rules_SOURCES = fprint-
 fprint_list_udev_rules_CFLAGS = -fvisibility=hidden -I$(srcdir)/nbis/include $(LIBUSB_CFLAGS) $(GLIB_CFLAGS) $(IMAGEMAGICK_CFLAGS) $(CRYPTO_CFLAGS) $(AM_CFLAGS)
 fprint_list_udev_rules_LDADD = $(builddir)/libfprint.la
 udev_rules_DATA = 60-fprint-autosuspend.rules
-udev_rulesdir = $(sysconfdir)/udev/rules.d/
+udev_rulesdir = /lib/udev/rules.d/
 libfprint_la_SOURCES = \
 	fp_internal.h	\
 	async.c		\
