diff -up openal-soft-1.14/OpenAL32/Include/alu.h.non-x86 openal-soft-1.14/OpenAL32/Include/alu.h
--- openal-soft-1.14/OpenAL32/Include/alu.h.x86	2012-04-12 13:03:01.000000000 +0200
+++ openal-soft-1.14/OpenAL32/Include/alu.h	2012-04-12 13:07:20.000000000 +0200
@@ -227,7 +227,7 @@ static __inline ALfloat cubic(ALfloat va
 
 static __inline int SetMixerFPUMode(void)
 {
-#if defined(_FPU_GETCW) && defined(_FPU_SETCW)
+#if defined(_FPU_GETCW) && defined(_FPU_SETCW) && (defined(__i386__) || defined(__x86_64__))
     fpu_control_t fpuState, newState;
     _FPU_GETCW(fpuState);
     newState = fpuState&~(_FPU_EXTENDED|_FPU_DOUBLE|_FPU_SINGLE |
@@ -249,7 +249,7 @@ static __inline int SetMixerFPUMode(void
 
 static __inline void RestoreFPUMode(int state)
 {
-#if defined(_FPU_GETCW) && defined(_FPU_SETCW)
+#if defined(_FPU_GETCW) && defined(_FPU_SETCW) && (defined(__i386__) || defined(__x86_64__))
     fpu_control_t fpuState = state;
     _FPU_SETCW(fpuState);
 #elif defined(HAVE__CONTROLFP)