diff -Nur stk11xx-2.1.0-old//stk11xx-usb.c stk11xx-2.1.0/stk11xx-usb.c
--- stk11xx-2.1.0-old//stk11xx-usb.c	2011-01-14 23:21:17.463000077 +0200
+++ stk11xx-2.1.0/stk11xx-usb.c	2011-01-14 23:22:00.103999746 +0200
@@ -790,7 +790,7 @@
 	}
 
 	// Init mutexes, spinlock, etc.
-	init_MUTEX(&dev->mutex);
+	sema_init(&dev->mutex, 1);
 	mutex_init(&dev->modlock);
 	spin_lock_init(&dev->spinlock);
 	init_waitqueue_head(&dev->wait_frame);
