diff -Nuar stk11xx-2.1.0.orig/stk11xx-v4l.c stk11xx-2.1.0/stk11xx-v4l.c
--- stk11xx-2.1.0.orig/stk11xx-v4l.c	2009-12-30 12:15:45.925583135 +0200
+++ stk11xx-2.1.0/stk11xx-v4l.c	2009-12-30 12:16:52.388334924 +0200
@@ -1733,7 +1733,7 @@
 	.poll = v4l_stk11xx_poll,
 	.mmap = v4l_stk11xx_mmap,
 	.ioctl = v4l_stk11xx_ioctl,
-#ifdef CONFIG_COMPAT
+#if defined(CONFIG_COMPAT) && defined(v4l_compat_ioctl32)
 	.compat_ioctl = v4l_compat_ioctl32,
 #endif
 };
