--- lftp-4.2.0/src/Makefile.am.link	2011-03-05 13:59:13.000000000 +0100
+++ lftp-4.2.0/src/Makefile.am	2011-03-05 13:59:33.000000000 +0100
@@ -55,8 +55,8 @@
 cmd_mirror_la_LDFLAGS = -module -avoid-version -rpath $(pkgverlibdir)
 cmd_sleep_la_LDFLAGS  = -module -avoid-version -rpath $(pkgverlibdir)
 cmd_torrent_la_LDFLAGS= -module -avoid-version -rpath $(pkgverlibdir)
-liblftp_pty_la_LDFLAGS     = -avoid-version -rpath $(pkgverlibdir)
-liblftp_network_la_LDFLAGS = -avoid-version -rpath $(pkgverlibdir) $(OPENSSL_LDFLAGS)
+liblftp_pty_la_LDFLAGS     = -module -avoid-version -rpath $(pkgverlibdir)
+liblftp_network_la_LDFLAGS = -module -avoid-version -rpath $(pkgverlibdir) $(OPENSSL_LDFLAGS)
 liblftp_network_la_LIBADD  = $(SOCKSLIBS) $(OPENSSL_LIBS) $(LIBGNUTLS_LIBS) $(GNULIB)
 
 proto_ftp_la_LIBADD  = -L$(DESTDIR)$(pkgverlibdir) liblftp-network.la
--- lftp-4.2.0/src/Makefile.in.link	2011-03-05 13:59:48.000000000 +0100
+++ lftp-4.2.0/src/Makefile.in	2011-03-05 14:00:18.000000000 +0100
@@ -1286,8 +1286,8 @@
 cmd_mirror_la_LDFLAGS = -module -avoid-version -rpath $(pkgverlibdir)
 cmd_sleep_la_LDFLAGS = -module -avoid-version -rpath $(pkgverlibdir)
 cmd_torrent_la_LDFLAGS = -module -avoid-version -rpath $(pkgverlibdir)
-liblftp_pty_la_LDFLAGS = -avoid-version -rpath $(pkgverlibdir)
-liblftp_network_la_LDFLAGS = -avoid-version -rpath $(pkgverlibdir) $(OPENSSL_LDFLAGS)
+liblftp_pty_la_LDFLAGS = -module -avoid-version -rpath $(pkgverlibdir)
+liblftp_network_la_LDFLAGS = -module -avoid-version -rpath $(pkgverlibdir) $(OPENSSL_LDFLAGS)
 liblftp_network_la_LIBADD = $(SOCKSLIBS) $(OPENSSL_LIBS) $(LIBGNUTLS_LIBS) $(GNULIB)
 proto_ftp_la_LIBADD = -L$(DESTDIR)$(pkgverlibdir) liblftp-network.la
 proto_http_la_LIBADD = -L$(DESTDIR)$(pkgverlibdir) liblftp-network.la $(EXPAT_LIBS)
