--- xf86-video-voodoo-1.2.5.orig/src/voodoo.h	2012-07-17 08:21:17.000000000 +0300
+++ xf86-video-voodoo-1.2.5/src/voodoo.h	2015-05-10 17:57:53.748577130 +0300
@@ -23,7 +23,9 @@
   
   Bool		      Voodoo2;		/* Set if Voodoo2 */
   pciVideoPtr	      PciInfo;		/* PCI data */
+#ifndef XSERVER_LIBPCIACCESS
   PCITAG	      PciTag;
+#endif
   CARD32	      PhysBase;
   
   CARD32	      Width;		/* Current width */
