diff -Naur puddletag-0.10.3.orig//puddletag.desktop puddletag-0.10.3/puddletag.desktop
--- puddletag-0.10.3.orig//puddletag.desktop	2010-10-30 14:11:45.000000000 +0300
+++ puddletag-0.10.3/puddletag.desktop	2011-03-26 23:49:20.396000030 +0200
@@ -6,4 +6,5 @@
 Exec=puddletag
 Icon=puddletag
 GenericName=Audio Tag Editor
-Categories=AudioVideo;Qt;
\ Dosya sonunda yenisatır yok.
+GenericName[tr]=Ses Etiket Dönüştürücü
+Categories=AudioVideo;Qt;
