diff -Nur scummvm-1.2.0-old/configure scummvm-1.2.0/configure
--- scummvm-1.2.0-old/configure	2010-12-13 19:54:27.926247043 +0200
+++ scummvm-1.2.0/configure	2010-12-13 19:54:34.337246946 +0200
@@ -2478,7 +2478,7 @@
 		for path_dir in $_nasmpath; do
 			if test -x "$path_dir/nasm$NATIVEEXEEXT" ; then
 				NASM="$path_dir/nasm$NATIVEEXEEXT"
-				NASMFLAGS="-O1 -g"
+				NASMFLAGS="-O1 "
 				echo $NASM
 				break
 			fi
