diff -p -up ImageMagick-6.4.8-9/configure.ac.docdir ImageMagick-6.4.8-9/configure.ac
--- ImageMagick-6.4.8-9/configure.ac.docdir	2009-01-14 18:27:35.000000000 +0100
+++ ImageMagick-6.4.8-9/configure.ac	2009-01-28 16:24:45.000000000 +0100
@@ -2464,7 +2464,7 @@ AC_SUBST(FILTER_PATH)
 
 #
 # Path to ImageMagick documentation files
-DOCUMENTATION_RELATIVE_PATH="${PACKAGE_NAME}-${PACKAGE_VERSION}"
+DOCUMENTATION_RELATIVE_PATH="imagemagick-doc"
 DOCUMENTATION_PATH="${DATA_DIR}/doc/${DOCUMENTATION_RELATIVE_PATH}"
 DEFINE_DOCUMENTATION_PATH="${DATA_DIR}/doc/${DOCUMENTATION_RELATIVE_PATH}/"
 case "${build_os}" in
