--- magick/Makefile.am	2008-06-05 05:08:39.000000000 +0200
+++ magick/Makefile.am.oden	2008-07-20 10:50:30.000000000 +0200
@@ -232,7 +232,17 @@
 	magick/xml-tree.c \
 	magick/xml-tree.h \
 	magick/xwindow.c \
-	magick/xwindow.h 
+	magick/xwindow.h \
+	wand/drawing-wand.c \
+	wand/drawing-wand.h \
+	wand/magick-image.c \
+	wand/magick-image.h \
+	wand/magick-wand.c \
+	wand/magick-wand.h \
+	wand/pixel-wand.c \
+	wand/pixel-wand.h \
+	wand/wand.c \
+	wand/wand.h
 
 if WIN32_NATIVE_BUILD
 MAGICK_PLATFORM_SRCS = \
