diff -p -up ImageMagick-6.4.8-9/magick/gem.c.ppc ImageMagick-6.4.8-9/magick/gem.c
--- ImageMagick-6.4.8-9/magick/gem.c.ppc	2008-11-25 19:00:48.000000000 +0100
+++ ImageMagick-6.4.8-9/magick/gem.c	2009-01-28 16:28:53.000000000 +0100
@@ -371,6 +371,10 @@ MagickExport void ConvertRGBToHSB(const 
   *saturation=(double) (delta/max);
   *brightness=(double) (QuantumScale*max);
   if (delta == 0.0)
+    {
+      *hue=0.0;
+      return;
+    }
     return;
   if ((MagickRealType) red == max)
     *hue=(double) ((green-(MagickRealType) blue)/delta);
