--- ImageMagick-6.4.2/configure.ac	2008-08-17 08:10:46.000000000 -0700
+++ ImageMagick-6.4.2/configure.ac.new	2008-08-19 12:08:29.000000000 -0700
@@ -3038,7 +3038,7 @@
 
 # Determine path to pick up MagickCore library from for use with building PerlMagick
 MAGICKCORE_PATH="${LIB_DIR}"
-if test $with_perl_static = 'yes'; then
+if test $with_perl_static = 'yes' -o $with_perl_dynamic = 'yes'; then
     # Find out where libtool hides its uninstalled libraries (as libtool_objdir)
     libtool_objdir=$objdir
 
