--- configure~	2010-10-04 01:26:11.000000000 +0800
+++ configure	2013-01-19 16:31:36.823632412 +0800
@@ -1103,7 +1103,7 @@
 fi
 
 # Add all extralibs provided by the pkgconfig files
-[ -n "$pkgconfig_requires" ] && add_extralibs `pkg-config $pkgconfig_requires --libs`
+[ -n "$pkgconfig_requires" ] && add_extralibs `pkg-config $pkgconfig_requires --libs` && add_cflags `pkg-config $pkgconfig_requires --cflags`
 
 
 #################################################
 
