Index: aufs-utils-0.0_20100924/aubrsync
===================================================================
--- aufs-utils-0.0_20100924.orig/aubrsync
+++ aufs-utils-0.0_20100924/aubrsync
@@ -37,13 +37,13 @@ test ! -x $f && EEcho $f is not installe
 # special handling for backward compatibility.
 #
 # aufs in the donated eeepc is aufs1 20080211 without CONFIG_AUFS_COMPAT.
-# /etc/default/aufs was introduced in aufs1 20080922.
+# /usr/share/aufs-progs/aufs was introduced in aufs1 20080922.
 # shwh/noshwh was introduced in aufs1 20080310 with CONFIG_AUFS_SHWH.
 # noshwh became always available regardless CONFIG_AUFS_SHWH in aufs1 20081117.
 
 noshwh=1
 AUFS_VERSION=20080211
-f=/etc/default/aufs
+f=/usr/share/aufs-progs/aufs
 if [ -s $f ]
 then
 	. $f
Index: aufs-utils-0.0_20100924/auchk
===================================================================
--- aufs-utils-0.0_20100924.orig/auchk
+++ aufs-utils-0.0_20100924/auchk
@@ -24,7 +24,7 @@ EEcho() # str
 	echo $0: $@ 1>&2
 }
 
-f=/etc/default/aufs
+f=/usr/share/aufs-progs/aufs
 . $f
 
 Usage()
Index: aufs-utils-0.0_20100924/Makefile
===================================================================
--- aufs-utils-0.0_20100924.orig/Makefile
+++ aufs-utils-0.0_20100924/Makefile
@@ -73,7 +73,7 @@ install_sbin install_ubin: ${File}
 	install -d ${Tgt}
 	${Install} -m 755 ${File} ${Tgt}
 install_etc: File = etc_default_aufs
-install_etc: Tgt = ${DESTDIR}/etc/default/aufs
+install_etc: Tgt = ${DESTDIR}/usr/share/aufs-progs/aufs
 install_etc: ${File}
 	install -d $(dir ${Tgt})
 	${Install} -m 644 -T ${File} ${Tgt}
Index: aufs-utils-0.0_20100924/README
===================================================================
--- aufs-utils-0.0_20100924.orig/README
+++ aufs-utils-0.0_20100924/README
@@ -50,7 +50,7 @@ o /usr/bin/aubrsync
   Kindly they agreed that I keep my aufs work as free software as it has
   been.
 
-o /etc/default/aufs
+o /usr/share/aufs-progs/aufs
   A library for shell scripts.
 
 o /usr/lib/libau.so
