--- GConf-2.32.0.orig//configure.in	2010-09-27 21:04:39.000000000 +0300
+++ GConf-2.31.91/configure.in	2010-08-29 00:11:39.000000000 +0300
@@ -303,7 +303,7 @@ AM_GLIB_GNU_GETTEXT
 
 AC_CHECK_FUNCS(bind_textdomain_codeset)
 
-GOBJECT_INTROSPECTION_CHECK([0.9.5])
+GOBJECT_INTROSPECTION_CHECK([0.6.7])
 
 AC_SUBST(CFLAGS)
 AC_SUBST(CPPFLAGS)
--- GConf-2.32.0.orig//gconf/gconf-engine.h	2010-09-26 21:50:02.000000000 +0300
+++ GConf-2.31.91/gconf/gconf-engine.h	2009-05-06 20:39:25.000000000 +0300
@@ -25,13 +25,6 @@
 
 G_BEGIN_DECLS
 
-/* Skipped from introspection because it's not an object or registered as boxed */
-/**
- * GConfEngine: (skip)
- *
- * An opaque data type representing one or more configuration sources.
- */
-
 /* A configuration engine (stack of config sources); normally there's
  * just one of these on the system.  
  */
--- GConf-2.32.0.orig//gconf/gconf-listeners.h	2010-09-26 21:50:02.000000000 +0300
+++ GConf-2.31.91/gconf/gconf-listeners.h	2009-05-06 20:39:25.000000000 +0300
@@ -24,17 +24,6 @@
 
 G_BEGIN_DECLS
 
-/* Skipped from introspection because it's not registered as boxed */
-/**
- * GConfListeners: (skip)
- *
- * The #GConfListeners structure contains nothing other than a dummy pointer. Internally 
- * the data about listeners is maintained through a listener table structure,
- * LTable which contains data like the namespace, an array to hold the listeners, count of
- * active listeners,value to be given to the next connection and the list of connection indices
- * to be recycled. There is also a Listener structure maintaining data pertaining to listeners.
- */
-
 /*
  * The listeners object is used to store listeners who want notification
  * of changes in a namespace section.
--- GConf-2.32.0.orig//gconf/gconf-schema.h	2010-09-26 21:50:02.000000000 +0300
+++ GConf-2.31.91/gconf/gconf-schema.h	2009-05-06 20:39:25.000000000 +0300
@@ -27,13 +27,6 @@
 
 G_BEGIN_DECLS
 
-/* Skipped from introspection because it's not registered as boxed */
-/**
- * GConfSchema: (skip)
- *
- * An opaque data type representing a description of a key-value pair.
- */
-
 /*
  *  A "schema" is a value that describes a key-value pair.
  *  It might include the type of the pair, documentation describing 
--- GConf-2.32.0.orig//gconf/Makefile.am	2010-09-26 21:50:02.000000000 +0300
+++ GConf-2.31.91/gconf/Makefile.am	2010-08-18 12:40:47.000000000 +0300
@@ -124,7 +124,7 @@ libgconf_2_la_LIBADD = $(INTLLIBS) $(DEP
 
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS =
-INTROSPECTION_SCANNER_ARGS = --warn-all --add-include-path=$(srcdir)
+INTROSPECTION_SCANNER_ARGS = --add-include-path=$(srcdir)
 INTROSPECTION_COMPILER_ARGS= --includedir=$(srcdir)
 
 if HAVE_INTROSPECTION
@@ -137,7 +137,6 @@ GConf_2_0_gir_CFLAGS = \
 	$(INCLUDES)
 GConf_2_0_gir_LIBS = libgconf-2.la
 GConf_2_0_gir_FILES = $(addprefix $(srcdir)/, $(introspection_files))
-GConf_2_0_gir_SCANNERFLAGS = --identifier-prefix=GConf --symbol-prefix=gconf
 INTROSPECTION_GIRS += GConf-2.0.gir
 
 girdir = $(datadir)/gir-1.0/
