Index: focuswriter-1.3.5/resources/unix/focuswriter.desktop
===================================================================
--- focuswriter-1.3.5.orig/resources/unix/focuswriter.desktop
+++ focuswriter-1.3.5/resources/unix/focuswriter.desktop
@@ -1,8 +1,11 @@
 [Desktop Entry]
 Type=Application
 Name=FocusWriter
+Name[tr]=FocusWriter
 GenericName=Fullscreen Word Processor
+GenericName[tr]=Tam Ekran Kelime İşlemci
 Comment=Write without distractions
+Comment[tr]=Dikkatiniz dağılmadan yazılarınızı yazın
 Icon=focuswriter
 TryExec=focuswriter
 Exec=focuswriter %F
