diff -up ./src/config_manager.cc.fixogg ./src/config_manager.cc
--- ./src/config_manager.cc.fixogg	2010-09-11 16:37:55.952709762 -0400
+++ ./src/config_manager.cc	2010-09-11 16:39:14.832710263 -0400
@@ -622,6 +622,8 @@ String ConfigManager::createDefaultConfi
                                            _(CONTENT_TYPE_MP3)));
     mtcontent->appendElementChild(treat_as(_("application/ogg"), 
                                            _(CONTENT_TYPE_OGG)));
+    mtcontent->appendElementChild(treat_as(_("audio/ogg"),
+                                           _(CONTENT_TYPE_OGG)));
     mtcontent->appendElementChild(treat_as(_("audio/x-flac"), 
                                            _(CONTENT_TYPE_FLAC)));
     mtcontent->appendElementChild(treat_as(_("image/jpeg"), 
