diff -r -U3 mediatomb-0.12.1.orig/build/Makefile.am mediatomb-0.12.1/build/Makefile.am
--- mediatomb-0.12.1.orig/build/Makefile.am	2013-01-30 11:37:04.170323353 -0600
+++ mediatomb-0.12.1/build/Makefile.am	2013-02-12 12:01:35.929475276 -0600
@@ -6,9 +6,6 @@
 
 libmediatomb_a_CXXFLAGS = \
                      -I$(top_srcdir)/src \
-                     -I$(top_srcdir)/tombupnp/ixml/inc \
-                     -I$(top_srcdir)/tombupnp/threadutil/inc \
-                     -I$(top_srcdir)/tombupnp/upnp/inc \
                      -I$(top_srcdir) \
                      $(SQLITE3_CFLAGS) \
                      $(MYSQL_CFLAGS) \
@@ -33,9 +30,6 @@
 bin_PROGRAMS = mediatomb
 mediatomb_SOURCES = $(top_srcdir)/src/main.cc
 mediatomb_CXXFLAGS = -I$(top_srcdir)/src \
-                     -I$(top_srcdir)/tombupnp/ixml/inc \
-                     -I$(top_srcdir)/tombupnp/threadutil/inc \
-                     -I$(top_srcdir)/tombupnp/upnp/inc \
                      -I$(top_srcdir) \
                      $(SQLITE3_CFLAGS) \
                      $(MYSQL_CFLAGS) \
@@ -59,7 +53,6 @@
 
 mediatomb_LDADD = \
     libmediatomb.a \
-    $(top_srcdir)/tombupnp/build/libtombupnp.a \
     $(LIBEXIF_LDFLAGS) \
     $(ZLIB_LDFLAGS) \
     $(EXPAT_LDFLAGS) \
diff -r -U3 mediatomb-0.12.1.orig/configure.ac mediatomb-0.12.1/configure.ac
--- mediatomb-0.12.1.orig/configure.ac	2013-01-30 11:37:04.201323352 -0600
+++ mediatomb-0.12.1/configure.ac	2013-02-12 12:01:08.947475878 -0600
@@ -30,7 +30,7 @@
 
 AC_PREREQ(2.61)
 AC_INIT([MediaTomb], [0.12.1], [jin@mediatomb.cc])
-AC_CONFIG_HEADERS([autoconfig.h tombupnp/upnp/inc/upnpconfig.h])
+AC_CONFIG_HEADERS([autoconfig.h])
 AC_CONFIG_AUX_DIR(configure_aux)
 AC_CONFIG_SRCDIR([src/common.h])
 AM_INIT_AUTOMAKE([1.9 -Wall])
@@ -1953,8 +1953,6 @@
     scripts/Makefile
     scripts/js/Makefile
     scripts/mediatomb-service-optware
-    tombupnp/Makefile
-    tombupnp/build/Makefile
     web/Makefile
     config/Makefile
     artwork/Makefile
diff -r -U3 mediatomb-0.12.1.orig/Makefile.am mediatomb-0.12.1/Makefile.am
--- mediatomb-0.12.1.orig/Makefile.am	2010-03-22 05:39:51.000000000 -0500
+++ mediatomb-0.12.1/Makefile.am	2013-02-12 12:01:08.947475878 -0600
@@ -1,4 +1,4 @@
-SUBDIRS = tombupnp build scripts web config doc artwork
+SUBDIRS = build scripts web config doc artwork
 
 EXTRA_DIST = mediatomb.spec README.UTF_8
 
diff -r -U3 mediatomb-0.12.1.orig/src/action_request.h mediatomb-0.12.1/src/action_request.h
--- mediatomb-0.12.1.orig/src/action_request.h	2010-03-25 09:58:11.000000000 -0500
+++ mediatomb-0.12.1/src/action_request.h	2013-02-12 12:01:08.948475878 -0600
@@ -34,7 +34,7 @@
 #define __ACTION_REQUEST_H__
 
 #include "common.h"
-#include "upnp.h"
+#include <upnp/upnp.h>
 #include "mxml/mxml.h"
 
 /// \brief This class represents the Upnp_Action_Request type from the SDK.
diff -r -U3 mediatomb-0.12.1.orig/src/buffered_io_handler.cc mediatomb-0.12.1/src/buffered_io_handler.cc
--- mediatomb-0.12.1.orig/src/buffered_io_handler.cc	2010-03-25 09:58:11.000000000 -0500
+++ mediatomb-0.12.1/src/buffered_io_handler.cc	2013-02-12 12:01:08.948475878 -0600
@@ -35,6 +35,7 @@
 
 #include "buffered_io_handler.h"
 #include "tools.h"
+#include <assert.h>
 
 using namespace zmm;
 
diff -r -U3 mediatomb-0.12.1.orig/src/buffered_io_handler.h mediatomb-0.12.1/src/buffered_io_handler.h
--- mediatomb-0.12.1.orig/src/buffered_io_handler.h	2010-03-25 09:58:11.000000000 -0500
+++ mediatomb-0.12.1/src/buffered_io_handler.h	2013-02-12 12:01:08.949475878 -0600
@@ -33,7 +33,7 @@
 #define __BUFFERED_IO_HANDLER_H__
 
 #include "common.h"
-#include "upnp.h"
+#include <upnp/upnp.h>
 #include "io_handler_buffer_helper.h"
 
 /// \brief a IOHandler with buffer support
diff -r -U3 mediatomb-0.12.1.orig/src/config_options.h mediatomb-0.12.1/src/config_options.h
--- mediatomb-0.12.1.orig/src/config_options.h	2010-03-25 09:58:11.000000000 -0500
+++ mediatomb-0.12.1/src/config_options.h	2013-02-12 12:01:08.949475878 -0600
@@ -37,6 +37,7 @@
 #include "exceptions.h"
 #include "autoscan.h"
 #include "object_dictionary.h"
+#include <assert.h>
 
 #ifdef EXTERNAL_TRANSCODING
     #include "transcoding/transcoding.h"
diff -r -U3 mediatomb-0.12.1.orig/src/curl_io_handler.h mediatomb-0.12.1/src/curl_io_handler.h
--- mediatomb-0.12.1.orig/src/curl_io_handler.h	2010-03-25 09:58:11.000000000 -0500
+++ mediatomb-0.12.1/src/curl_io_handler.h	2013-02-12 12:01:08.950475878 -0600
@@ -36,7 +36,7 @@
 
 #include <curl/curl.h>
 #include "common.h"
-#include "upnp.h"
+#include <upnp/upnp.h>
 #include "io_handler_buffer_helper.h"
 
 class CurlIOHandler : public IOHandlerBufferHelper
diff -r -U3 mediatomb-0.12.1.orig/src/file_io_handler.cc mediatomb-0.12.1/src/file_io_handler.cc
--- mediatomb-0.12.1.orig/src/file_io_handler.cc	2010-03-25 09:58:11.000000000 -0500
+++ mediatomb-0.12.1/src/file_io_handler.cc	2013-02-12 12:01:08.950475878 -0600
@@ -44,7 +44,7 @@
 #include "cds_objects.h"
 #include "process.h"
 #include "update_manager.h"
-#include "ixml.h"
+#include <upnp/ixml.h>
 #include "file_io_handler.h"
 #include "dictionary.h"
 
diff -r -U3 mediatomb-0.12.1.orig/src/file_request_handler.cc mediatomb-0.12.1/src/file_request_handler.cc
--- mediatomb-0.12.1.orig/src/file_request_handler.cc	2013-01-30 11:37:04.195323352 -0600
+++ mediatomb-0.12.1/src/file_request_handler.cc	2013-02-12 12:01:08.951475878 -0600
@@ -45,7 +45,7 @@
 #include "process.h"
 #include "update_manager.h"
 #include "session_manager.h"
-#include "ixml.h"
+#include <upnp/ixml.h>
 #include "file_io_handler.h"
 #include "dictionary.h"
 #include "file_request_handler.h"
@@ -193,7 +193,7 @@
     tr_profile = dict->get(_(URL_PARAM_TRANSCODE_PROFILE_NAME));
 #endif
 
-    info->http_header = NULL;
+    //info->http_header = NULL;
     // for transcoded resourecs res_id will always be negative
     log_debug("fetching resource id %d\n", res_id);
     String rh = dict->get(_(RESOURCE_HANDLER));
@@ -365,8 +365,8 @@
     header = getDLNAtransferHeader(mimeType, header);
 #endif
 
-    if (string_ok(header))
-        info->http_header = ixmlCloneDOMString(header.c_str());
+    //if (string_ok(header))
+    //    info->http_header = ixmlCloneDOMString(header.c_str());
 
     info->last_modified = statbuf.st_mtime;
     info->is_directory = S_ISDIR(statbuf.st_mode);
@@ -579,7 +579,7 @@
     }
 #endif
 
-    info->http_header = NULL;
+    //info->http_header = NULL;
     // Per default and in case of a bad resource ID, serve the file
     // itself
 
@@ -616,8 +616,8 @@
         header = getDLNAtransferHeader(mimeType, header);
 #endif
 
-        if (string_ok(header))
-                info->http_header = ixmlCloneDOMString(header.c_str());
+        //if (string_ok(header))
+        //        info->http_header = ixmlCloneDOMString(header.c_str());
 
         info->content_type = ixmlCloneDOMString(mimeType.c_str());
         Ref<IOHandler> io_handler = h->serveContent(item, res_id, &(info->file_length));
@@ -733,8 +733,8 @@
 #ifdef EXTEND_PROTOCOLINFO
             header = getDLNAtransferHeader(mimeType, header);
 #endif
-            if (string_ok(header))
-                info->http_header = ixmlCloneDOMString(header.c_str());
+            //if (string_ok(header))
+            //    info->http_header = ixmlCloneDOMString(header.c_str());
 
 
             Ref<IOHandler> io_handler(new FileIOHandler(path));
diff -r -U3 mediatomb-0.12.1.orig/src/io_handler_buffer_helper.h mediatomb-0.12.1/src/io_handler_buffer_helper.h
--- mediatomb-0.12.1.orig/src/io_handler_buffer_helper.h	2010-03-25 09:58:11.000000000 -0500
+++ mediatomb-0.12.1/src/io_handler_buffer_helper.h	2013-02-12 12:01:08.952475877 -0600
@@ -35,7 +35,7 @@
 #include <pthread.h>
 
 #include "common.h"
-#include "upnp.h"
+#include <upnp/upnp.h>
 #include "io_handler.h"
 #include "sync.h"
 
diff -r -U3 mediatomb-0.12.1.orig/src/io_handler.cc mediatomb-0.12.1/src/io_handler.cc
--- mediatomb-0.12.1.orig/src/io_handler.cc	2010-03-25 09:58:11.000000000 -0500
+++ mediatomb-0.12.1/src/io_handler.cc	2013-02-12 12:01:08.952475877 -0600
@@ -46,7 +46,7 @@
 #include "cds_objects.h"
 #include "process.h"
 #include "update_manager.h"
-#include "ixml.h"
+#include <upnp/ixml.h>
 #include "io_handler.h"
 
 using namespace zmm;
diff -r -U3 mediatomb-0.12.1.orig/src/io_handler.h mediatomb-0.12.1/src/io_handler.h
--- mediatomb-0.12.1.orig/src/io_handler.h	2010-03-25 09:58:11.000000000 -0500
+++ mediatomb-0.12.1/src/io_handler.h	2013-02-12 12:01:08.953475877 -0600
@@ -33,7 +33,7 @@
 #define __IO_HANDLER_H__
 
 #include "common.h"
-#include "upnp.h"
+#include <upnp/upnp.h>
 
 class IOHandler : public zmm::Object
 {
diff -r -U3 mediatomb-0.12.1.orig/src/mem_io_handler.cc mediatomb-0.12.1/src/mem_io_handler.cc
--- mediatomb-0.12.1.orig/src/mem_io_handler.cc	2010-03-25 09:58:11.000000000 -0500
+++ mediatomb-0.12.1/src/mem_io_handler.cc	2013-02-12 12:01:08.953475877 -0600
@@ -39,7 +39,7 @@
 #include <unistd.h>
 #include <string.h>
 #include <stdio.h>
-#include "ixml.h"
+#include <upnp/ixml.h>
 #include <time.h>
 #include "common.h"
 #include "storage.h"
diff -r -U3 mediatomb-0.12.1.orig/src/mxml/element.cc mediatomb-0.12.1/src/mxml/element.cc
--- mediatomb-0.12.1.orig/src/mxml/element.cc	2010-03-25 09:58:07.000000000 -0500
+++ mediatomb-0.12.1/src/mxml/element.cc	2013-02-12 12:01:08.954475877 -0600
@@ -37,6 +37,7 @@
 #include "tools.h"
 
 #include <string.h>
+#include <assert.h>
 
 using namespace zmm;
 using namespace mxml;
diff -r -U3 mediatomb-0.12.1.orig/src/mxml/xml_to_json.cc mediatomb-0.12.1/src/mxml/xml_to_json.cc
--- mediatomb-0.12.1.orig/src/mxml/xml_to_json.cc	2010-03-25 09:58:07.000000000 -0500
+++ mediatomb-0.12.1/src/mxml/xml_to_json.cc	2013-02-12 12:01:08.954475877 -0600
@@ -35,6 +35,7 @@
 
 #include "xml_to_json.h"
 #include "tools.h"
+#include <assert.h>
 
 using namespace zmm;
 using namespace mxml;
diff -r -U3 mediatomb-0.12.1.orig/src/server.cc mediatomb-0.12.1/src/server.cc
--- mediatomb-0.12.1.orig/src/server.cc	2010-03-25 09:58:11.000000000 -0500
+++ mediatomb-0.12.1/src/server.cc	2013-02-12 12:01:08.955475877 -0600
@@ -166,7 +166,7 @@
     if (storage->threadCleanupRequired())
         cb = (void *)static_cleanup_callback;
 
-    ret = UpnpInit(ip.c_str(), port, 0, cb);
+    ret = UpnpInit(ip.c_str(), (unsigned short) port);
 
     if (ret != UPNP_E_SUCCESS)
     {
@@ -201,7 +201,7 @@
     }
 
     log_debug("webroot: %s\n", web_root.c_str()); 
-
+/*
         Ref<Array<StringBase> > arr = config->getStringArrayOption(CFG_SERVER_CUSTOM_HTTP_HEADERS);
 
         if (arr != nil)
@@ -221,7 +221,7 @@
                 }
             }
         }
-
+*/
     ret = UpnpAddVirtualDir(virtual_directory.c_str());
     if (ret != UPNP_E_SUCCESS)
     {
diff -r -U3 mediatomb-0.12.1.orig/src/serve_request_handler.cc mediatomb-0.12.1/src/serve_request_handler.cc
--- mediatomb-0.12.1.orig/src/serve_request_handler.cc	2013-01-30 11:37:04.163323353 -0600
+++ mediatomb-0.12.1/src/serve_request_handler.cc	2013-02-12 12:01:08.955475877 -0600
@@ -44,7 +44,7 @@
 #include "cds_objects.h"
 #include "process.h"
 #include "update_manager.h"
-#include "ixml.h"
+#include <upnp/ixml.h>
 #include "file_io_handler.h"
 #include "dictionary.h"
 #include "serve_request_handler.h"
diff -r -U3 mediatomb-0.12.1.orig/src/subscription_request.h mediatomb-0.12.1/src/subscription_request.h
--- mediatomb-0.12.1.orig/src/subscription_request.h	2010-03-25 09:58:11.000000000 -0500
+++ mediatomb-0.12.1/src/subscription_request.h	2013-02-12 12:01:08.956475877 -0600
@@ -34,7 +34,7 @@
 #define __SUBSCRIPTION_REQUEST_H__
 
 #include "common.h"
-#include "upnp.h"
+#include <upnp/upnp.h>
 
 /// \brief This class represents the Upnp_Subscription_Request type from the SDK.
 ///
diff -r -U3 mediatomb-0.12.1.orig/src/transcoding/transcode_ext_handler.cc mediatomb-0.12.1/src/transcoding/transcode_ext_handler.cc
--- mediatomb-0.12.1.orig/src/transcoding/transcode_ext_handler.cc	2013-01-30 11:37:04.158323353 -0600
+++ mediatomb-0.12.1/src/transcoding/transcode_ext_handler.cc	2013-02-12 12:01:08.956475877 -0600
@@ -51,7 +51,7 @@
 #include "process.h"
 #include "update_manager.h"
 #include "session_manager.h"
-#include "ixml.h"
+#include <upnp/ixml.h>
 #include "process_io_handler.h"
 #include "buffered_io_handler.h"
 #include "dictionary.h"
@@ -124,12 +124,12 @@
     
     log_debug("Adding TimeSeekRange response HEADERS: %s\n", header.c_str());
     header = getDLNAtransferHeader(mimeType, header);
-    if (string_ok(header))
-        info->http_header = ixmlCloneDOMString(header.c_str());
+    //if (string_ok(header))
+    //    info->http_header = ixmlCloneDOMString(header.c_str());
 #endif
    
     info->file_length = UNKNOWN_CONTENT_LENGTH;
-    info->force_chunked = (int)profile->getChunked();
+    //info->force_chunked = (int)profile->getChunked();
 
     Ref<ConfigManager> cfg = ConfigManager::getInstance();
    
diff -r -U3 mediatomb-0.12.1.orig/src/transcoding/transcode_ext_handler.h mediatomb-0.12.1/src/transcoding/transcode_ext_handler.h
--- mediatomb-0.12.1.orig/src/transcoding/transcode_ext_handler.h	2013-01-30 11:37:04.158323353 -0600
+++ mediatomb-0.12.1/src/transcoding/transcode_ext_handler.h	2013-02-12 12:01:08.957475877 -0600
@@ -34,7 +34,7 @@
 
 #include "common.h"
 #include "transcode_handler.h"
-#include "upnp.h"
+#include <upnp/upnp.h>
 
 class TranscodeExternalHandler : public TranscodeHandler 
 {
diff -r -U3 mediatomb-0.12.1.orig/src/transcoding/transcode_handler.h mediatomb-0.12.1/src/transcoding/transcode_handler.h
--- mediatomb-0.12.1.orig/src/transcoding/transcode_handler.h	2013-01-30 11:37:04.158323353 -0600
+++ mediatomb-0.12.1/src/transcoding/transcode_handler.h	2013-02-12 12:01:08.957475877 -0600
@@ -35,7 +35,7 @@
 #include "common.h"
 #include "io_handler.h"
 #include "transcoding.h"
-#include "upnp.h"
+#include <upnp/upnp.h>
 #include "cds_objects.h"
 
 class TranscodeHandler : public zmm::Object
diff -r -U3 mediatomb-0.12.1.orig/src/upnp_cds_subscriptions.cc mediatomb-0.12.1/src/upnp_cds_subscriptions.cc
--- mediatomb-0.12.1.orig/src/upnp_cds_subscriptions.cc	2010-03-25 09:58:11.000000000 -0500
+++ mediatomb-0.12.1/src/upnp_cds_subscriptions.cc	2013-02-12 12:01:08.958475877 -0600
@@ -35,7 +35,7 @@
 
 #include "upnp_cds.h"
 #include "upnp_xml.h"
-#include "ixml.h"
+#include <upnp/ixml.h>
 #include "server.h"
 #include "storage.h"
 
diff -r -U3 mediatomb-0.12.1.orig/src/upnp_cm_subscriptions.cc mediatomb-0.12.1/src/upnp_cm_subscriptions.cc
--- mediatomb-0.12.1.orig/src/upnp_cm_subscriptions.cc	2010-03-25 09:58:11.000000000 -0500
+++ mediatomb-0.12.1/src/upnp_cm_subscriptions.cc	2013-02-12 12:01:08.958475877 -0600
@@ -37,7 +37,7 @@
 #include "upnp_cm.h"
 #include "server.h"
 #include "upnp_xml.h"
-#include "ixml.h"
+#include <upnp/ixml.h>
 #include "storage.h"
 
 using namespace zmm;
diff -r -U3 mediatomb-0.12.1.orig/src/upnp_mrreg_subscriptions.cc mediatomb-0.12.1/src/upnp_mrreg_subscriptions.cc
--- mediatomb-0.12.1.orig/src/upnp_mrreg_subscriptions.cc	2010-03-25 09:58:11.000000000 -0500
+++ mediatomb-0.12.1/src/upnp_mrreg_subscriptions.cc	2013-02-12 12:01:08.958475877 -0600
@@ -39,7 +39,7 @@
 #include "upnp_mrreg.h"
 #include "server.h"
 #include "upnp_xml.h"
-#include "ixml.h"
+#include <upnp/ixml.h>
 #include "storage.h"
 
 using namespace zmm;
diff -r -U3 mediatomb-0.12.1.orig/src/url_request_handler.cc mediatomb-0.12.1/src/url_request_handler.cc
--- mediatomb-0.12.1.orig/src/url_request_handler.cc	2013-01-30 11:37:04.164323353 -0600
+++ mediatomb-0.12.1/src/url_request_handler.cc	2013-02-12 12:01:08.959475877 -0600
@@ -38,7 +38,7 @@
 #include "server.h"
 #include "common.h"
 #include "storage.h"
-#include "ixml.h"
+#include <upnp/ixml.h>
 #include "buffered_io_handler.h"
 #include "dictionary.h"
 #include "url_request_handler.h"
@@ -159,10 +159,10 @@
     info->last_modified = 0;
     info->is_directory = 0;
 
-    if (string_ok(header))
-        info->http_header = ixmlCloneDOMString(header.c_str());
-    else
-        info->http_header = NULL;
+//    if (string_ok(header))
+//        info->http_header = ixmlCloneDOMString(header.c_str());
+//    else
+//        info->http_header = NULL;
 
     info->content_type = ixmlCloneDOMString(mimeType.c_str());
     log_debug("web_get_info(): end\n");
@@ -233,7 +233,7 @@
     info->is_readable = 1;
     info->last_modified = 0;
     info->is_directory = 0;
-    info->http_header = NULL;
+    //info->http_header = NULL;
 
 #ifdef EXTERNAL_TRANSCODING
     tr_profile = dict->get(_(URL_PARAM_TRANSCODE_PROFILE_NAME));
@@ -271,8 +271,8 @@
         info->content_type = ixmlCloneDOMString(mimeType.c_str());
     }
 
-    if (string_ok(header))
-        info->http_header = ixmlCloneDOMString(header.c_str());
+//    if (string_ok(header))
+//        info->http_header = ixmlCloneDOMString(header.c_str());
 
     ///\todo make curl io handler configurable for url request handler
     Ref<IOHandler> io_handler(new CurlIOHandler(url, NULL, 1024*1024, 0));
diff -r -U3 mediatomb-0.12.1.orig/src/web_callbacks.cc mediatomb-0.12.1/src/web_callbacks.cc
--- mediatomb-0.12.1.orig/src/web_callbacks.cc	2013-01-30 11:37:04.162323353 -0600
+++ mediatomb-0.12.1/src/web_callbacks.cc	2013-02-12 12:01:08.960475877 -0600
@@ -45,7 +45,7 @@
 #include "cds_objects.h"
 #include "process.h"
 #include "update_manager.h"
-#include "ixml.h"
+#include <upnp/ixml.h>
 #include "io_handler.h"
 #include "request_handler.h"
 #include "file_request_handler.h"
diff -r -U3 mediatomb-0.12.1.orig/src/web_request_handler.cc mediatomb-0.12.1/src/web_request_handler.cc
--- mediatomb-0.12.1.orig/src/web_request_handler.cc	2013-01-30 11:37:04.164323353 -0600
+++ mediatomb-0.12.1/src/web_request_handler.cc	2013-02-12 12:01:08.960475877 -0600
@@ -112,7 +112,7 @@
     contentType = mimetype + "; charset=" + DEFAULT_INTERNAL_CHARSET;
     
     info->content_type = ixmlCloneDOMString(contentType.c_str());
-    info->http_header = ixmlCloneDOMString("Cache-Control: no-cache, must-revalidate");
+    //info->http_header = ixmlCloneDOMString("Cache-Control: no-cache, must-revalidate");
 }
 
 Ref<IOHandler> WebRequestHandler::open(IN enum UpnpOpenFileMode mode)
