diff -Nuar mozplugger-1.14.2.orig//Makefile mozplugger-1.14.2/Makefile
--- mozplugger-1.14.2.orig//Makefile	2011-01-12 10:24:05.014000012 +0200
+++ mozplugger-1.14.2/Makefile	2011-01-12 12:04:16.952000029 +0200
@@ -1,5 +1,5 @@
 # Where is your Xwindows located?
-X11=/usr/X11R6
+X11=/usr
 
 #
 # For building rpms
@@ -146,7 +146,7 @@
 	$(CC) -c $(CFLAGS) -o mozplugger-helper.o mozplugger-helper.c
 
 mozplugger.so: mozplugger.o npunix.o mozplugger-common.o debug.o Makefile
-	$(LDSHARED) $(LDFLAGS) -o mozplugger.so mozplugger.o mozplugger-common.o npunix.o debug.o $(EXTRA_LIBS)
+	$(LDSHARED) $(LDFLAGS) -o mozplugger.so mozplugger.o mozplugger-common.o npunix.o debug.o  $(XLIBS) $(EXTRA_LIBS)
 
 mozplugger-common.o: mozplugger-common.c mozplugger.h Makefile
 	$(CC) -c $(CFLAGS) -o mozplugger-common.o mozplugger-common.c
