Index: mozplugger-1.14.3/Makefile
===================================================================
--- mozplugger-1.14.3.orig/Makefile
+++ mozplugger-1.14.3/Makefile
@@ -9,7 +9,7 @@ root=
 # For installing
 prefix=/usr
 sysconfdir=/etc
-exec_prefix=
+exec_prefix=$(prefix)
 
 # On 64 bit arch change libprefix to lib64
 libprefix=/lib
@@ -201,16 +201,16 @@ localinstall_opera:
 	make localinstall BROWSERDIR=.opera
 
 install:
-	-@mkdir -p $(exec_prefix)/bin
-	-@mkdir -p $(prefix)$(libprefix)/mozilla/plugins
-	-@mkdir -p $(exec_prefix)/share/man/man7
-	-@mkdir -p ${sysconfdir}
-	cp mozplugger-helper $(exec_prefix)/bin/
-	cp mozplugger-controller $(exec_prefix)/bin/
-	cp mozplugger-linker $(exec_prefix)/bin/
-	cp mozplugger.so $(prefix)$(libprefix)/mozilla/plugins/
-	cp mozpluggerrc ${sysconfdir}/
-	cp mozplugger.7 $(exec_prefix)/share/man/man7/
+	-@mkdir -p $(root)$(exec_prefix)/bin
+	-@mkdir -p $(root)$(prefix)$(libprefix)/mozilla/plugins
+	-@mkdir -p $(root)$(exec_prefix)/share/man/man7
+	-@mkdir -p $(root)${sysconfdir}
+	cp mozplugger-helper $(root)$(exec_prefix)/bin/
+	cp mozplugger-controller $(root)$(exec_prefix)/bin/
+	cp mozplugger-linker $(root)$(exec_prefix)/bin/
+	cp mozplugger.so $(root)$(prefix)$(libprefix)/mozilla/plugins/
+	cp mozpluggerrc $(root)${sysconfdir}/
+	cp mozplugger.7 $(root)$(exec_prefix)/share/man/man7/
 
 mozplugger.tar.gz: $(BASE_FILES) $(SOURCE_FILES)
 	@( DIR=`pwd`;\
