diff -up mozplugger-1.13.3/mozpluggerrc.path mozplugger-1.13.3/mozpluggerrc
--- mozplugger-1.13.3/mozpluggerrc.path	2010-04-17 00:05:05.000000000 +0200
+++ mozplugger-1.13.3/mozpluggerrc	2010-04-20 12:07:57.000000000 +0200
@@ -352,7 +352,7 @@ application/vnd.oasis.opendocument.prese
 chemical/x-pdb:pdb: Protein Data Bank file
 model/x-pdb:pdb: Protein Data Bank file
 	swallow(rasmol) fill: rasmol "$file"
-        swallow(molecule) fill: /usr/X11R6/lib/xscreensaver/molecule -delay 20000 -geometry +9000+9000 -no-spin -molecule "$file"
+        swallow(molecule) fill: /usr/libexec/xscreensaver/molecule -delay 20000 -geometry +9000+9000 -no-spin -molecule "$file"
 
 application/bge:blend:Blender Game Engine
         swallow(%f): blenderplayer $file
