diff -Nur bk-client2.0-old/bkf.c bk-client2.0/bkf.c
--- bk-client2.0-old/bkf.c	2011-02-07 15:52:15.146000101 +0200
+++ bk-client2.0/bkf.c	2011-02-07 15:52:56.273000102 +0200
@@ -31,7 +31,7 @@
 int	changes(int ac, char **av);
 int	chomp(char *s);
 int	clone(int ac, char **av);
-void	getline(int sock, char *buf, unsigned int len);
+void	bk_getline(int sock, char *buf, unsigned int len);
 int	scanurl(url *u, char *p);
 int	tcp_connect(char *host, int port);
 int	parent(int ac, char **av);
@@ -148,7 +148,7 @@
 	}
 	sprintf(buf, "bkf_tarball %s\n", rev);
 	unless (write(sock, buf, strlen(buf)) == strlen(buf)) exit(1);
-line:	getline(sock, buf, sizeof(buf));
+line:	bk_getline(sock, buf, sizeof(buf));
 	switch (buf[0]) {
 	    case 'I':
 		unless (quiet) fprintf(stderr, "%s", buf);
@@ -182,7 +182,7 @@
 				break;
 			}
 		} while (bytes > 0);
-		getline(sock, buf, sizeof(buf));
+		bk_getline(sock, buf, sizeof(buf));
 	}
 	pclose(f);
 	exit(1);
@@ -449,7 +449,7 @@
 }
 
 void
-getline(int sock, char *buf, unsigned int len)
+bk_getline(int sock, char *buf, unsigned int len)
 {
 	unsigned int i;
 
