Index: pulseaudio-0.9.14/src/daemon/daemon.conf.in
===================================================================
--- pulseaudio-0.9.14.orig/src/daemon/daemon.conf.in	2009-03-07 05:01:11.000000000 -0500
+++ pulseaudio-0.9.14/src/daemon/daemon.conf.in	2009-03-07 05:01:54.000000000 -0500
@@ -72,5 +72,5 @@
 ; default-sample-rate = 44100
 ; default-sample-channels = 2
 
-; default-fragments = 4
-; default-fragment-size-msec = 25
+default-fragments = 8
+default-fragment-size-msec = 10
