Index: pulseaudio-0.9.21/configure
===================================================================
--- pulseaudio-0.9.21.orig/configure
+++ pulseaudio-0.9.21/configure
@@ -2792,7 +2792,6 @@ as_fn_append ac_func_list " strsignal"
 as_fn_append ac_func_list " sig2str"
 as_fn_append ac_func_list " strtof_l"
 as_fn_append ac_func_list " open64"
-as_fn_append ac_header_list " valgrind/memcheck.h"
 # Check that the precious variables saved in the cache have kept the same
 # value.
 ac_cache_corrupted=false
Index: pulseaudio-0.9.21/configure.ac
===================================================================
--- pulseaudio-0.9.21.orig/configure.ac
+++ pulseaudio-0.9.21/configure.ac
@@ -556,10 +556,6 @@ if test "x${with_caps}" != "xno"; then
                     fi])
 fi
 
-#### Valgrind (optional) ####
-
-AC_CHECK_HEADERS_ONCE([valgrind/memcheck.h])
-
 #### Sound file ####
 
 PKG_CHECK_MODULES(LIBSNDFILE, [ sndfile >= 1.0.20 ])
