Index: pulseaudio-0.9.14/src/daemon/default.pa.in
===================================================================
--- pulseaudio-0.9.14.orig/src/daemon/default.pa.in	2009-03-08 00:18:26.000000000 -0500
+++ pulseaudio-0.9.14/src/daemon/default.pa.in	2009-03-08 00:19:30.000000000 -0500
@@ -26,6 +26,7 @@
 #load-sample-lazy pulse-hotplug /usr/share/sounds/startup3.wav
 #load-sample-lazy pulse-coldplug /usr/share/sounds/startup3.wav
 #load-sample-lazy pulse-access /usr/share/sounds/generic.wav
+load-sample-dir-lazy /usr/share/sounds
 
 .fail
 
