diff -ur gsoap-2.8.orig/gsoap/samples/calc_vs2005/calc_vs2005/stdsoap2.h gsoap-2.8/gsoap/samples/calc_vs2005/calc_vs2005/stdsoap2.h
--- gsoap-2.8.orig/gsoap/samples/calc_vs2005/calc_vs2005/stdsoap2.h	2013-08-26 01:10:06.000000000 +0200
+++ gsoap-2.8/gsoap/samples/calc_vs2005/calc_vs2005/stdsoap2.h	2013-10-16 09:29:02.181454071 +0200
@@ -2130,6 +2130,7 @@
   struct sockaddr_storage peer;	/* IPv6: set by soap_accept and by UDP recv */
 #else
   struct sockaddr_in peer;	/* IPv4: set by soap_connect/soap_accept and by UDP recv */
+  unsigned char pad[sizeof(struct sockaddr_storage) - sizeof(struct sockaddr_in)];
 #endif
 #endif
   size_t peerlen;
diff -ur gsoap-2.8.orig/gsoap/stdsoap2.h gsoap-2.8/gsoap/stdsoap2.h
--- gsoap-2.8.orig/gsoap/stdsoap2.h	2013-08-26 01:10:07.000000000 +0200
+++ gsoap-2.8/gsoap/stdsoap2.h	2013-10-16 09:29:02.181454071 +0200
@@ -2130,6 +2130,7 @@
   struct sockaddr_storage peer;	/* IPv6: set by soap_accept and by UDP recv */
 #else
   struct sockaddr_in peer;	/* IPv4: set by soap_connect/soap_accept and by UDP recv */
+  unsigned char pad[sizeof(struct sockaddr_storage) - sizeof(struct sockaddr_in)];
 #endif
 #endif
   size_t peerlen;
diff -ur gsoap-2.8.orig/gsoap/VisualStudio2005/wsdl2h/wsdl2h/stdsoap2.h gsoap-2.8/gsoap/VisualStudio2005/wsdl2h/wsdl2h/stdsoap2.h
--- gsoap-2.8.orig/gsoap/VisualStudio2005/wsdl2h/wsdl2h/stdsoap2.h	2013-08-26 01:10:07.000000000 +0200
+++ gsoap-2.8/gsoap/VisualStudio2005/wsdl2h/wsdl2h/stdsoap2.h	2013-10-16 09:29:02.181454071 +0200
@@ -2130,6 +2130,7 @@
   struct sockaddr_storage peer;	/* IPv6: set by soap_accept and by UDP recv */
 #else
   struct sockaddr_in peer;	/* IPv4: set by soap_connect/soap_accept and by UDP recv */
+  unsigned char pad[sizeof(struct sockaddr_storage) - sizeof(struct sockaddr_in)];
 #endif
 #endif
   size_t peerlen;
