--- server/gam_node.h.orig	2012-10-02 19:46:52.000000000 +0300
+++ server/gam_node.h		2012-10-02 19:48:22.978078187 +0300
@@ -58,7 +58,7 @@
 void                  gam_node_set_is_dir          (GamNode         *node,
 						   gboolean        is_dir);
 	
-G_CONST_RETURN char  *gam_node_get_path            (GamNode         *node);
+const char           *gam_node_get_path            (GamNode         *node);
 
 GList                *gam_node_get_subscriptions   (GamNode         *node);
 
