diff -Nuar screengrab-0.9.1.orig/screengrab.desktop screengrab-0.9.1/screengrab.desktop
--- screengrab-0.9.1.orig/screengrab.desktop	2011-06-18 21:02:08.969000069 +0300
+++ screengrab-0.9.1/screengrab.desktop	2011-06-18 21:03:22.858000075 +0300
@@ -2,9 +2,11 @@
 Name=ScreenGrab
 Comment=Qt4-based Screen Capture Program
 Comment[ru]=Программа  для создания снимков экрана на базе Qt4
+Comment[tr]=Qt4 tabanlı Ekran Görüntüsü Yakalama Programı
 Comment[uk]=Програма  для створення знімків екрану на базі Qt4
 GenericName=Screen Capture Program
 GenericName[ru]=Создание снимков экрана
+GenericName[tr]=Ekran Görüntüsü Yakalama Programı
 GenericName[uk]=Створення знімків екрану
 Exec=screengrab
 Icon=screengrab
