diff -up phonon-4.4.3/gstreamer/backend.cpp.flac_mimetype phonon-4.4.3/gstreamer/backend.cpp
--- phonon-4.4.3/gstreamer/backend.cpp.flac_mimetype	2010-11-24 12:26:19.000000000 -0600
+++ phonon-4.4.3/gstreamer/backend.cpp	2010-11-30 07:29:31.005424613 -0600
@@ -265,6 +265,11 @@ QStringList Backend::availableMimeTypes(
         if (!availableMimeTypes.contains("audio/ogg")) /* *.oga */
             availableMimeTypes.append("audio/ogg");
     }
+    if (availableMimeTypes.contains("audio/x-flac")) {
+        if (!availableMimeTypes.contains("audio/flac")) {
+            availableMimeTypes.append("audio/flac");
+        }
+    }
     availableMimeTypes.sort();
     return availableMimeTypes;
 }
