diff -Nuar gnome-menus-2.30.5.orig/libmenu/Makefile.am gnome-menus-2.30.5/libmenu/Makefile.am
--- gnome-menus-2.30.5.orig/libmenu/Makefile.am	2010-11-17 13:07:53.000000000 +0200
+++ gnome-menus-2.30.5/libmenu/Makefile.am	2011-02-02 15:59:10.586000049 +0200
@@ -37,7 +37,7 @@
 libgnome_menu_la_LDFLAGS =			\
 	-version-info $(LIB_MENU_LT_VERSION)	\
 	-no-undefined				\
-	-export-symbols-regex gmenu_tree
+	-export-symbols-regex gmenu_tree $(LDFLAGS)
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libgnome-menu.pc
