diff -Nur dsniff-2.4-old/msgsnarf.c dsniff-2.4/msgsnarf.c
--- dsniff-2.4-old/msgsnarf.c	2009-09-15 00:11:17.157475747 +0300
+++ dsniff-2.4/msgsnarf.c	2009-09-15 00:11:20.194202497 +0300
@@ -23,6 +23,7 @@
 #include <nids.h>
 #include <pcap.h>
 #include <pcaputil.h>
+#include <time.h>
 
 #include "buf.h"
 #include "decode.h"
diff -Nur dsniff-2.4-old/sshow.c dsniff-2.4/sshow.c
--- dsniff-2.4-old/sshow.c	2009-09-15 00:11:17.161202828 +0300
+++ dsniff-2.4/sshow.c	2009-09-15 00:11:20.195203044 +0300
@@ -15,6 +15,7 @@
 
 #include <sys/types.h>
 #include <sys/times.h>
+#include <time.h>
 
 #include <netinet/in_systm.h>
 #include <netinet/in.h>
