diff -Nur dsniff-2.4-old/urlsnarf.c dsniff-2.4/urlsnarf.c
--- dsniff-2.4-old/urlsnarf.c	2009-09-15 00:11:20.321202268 +0300
+++ dsniff-2.4/urlsnarf.c	2009-09-15 00:11:20.407202402 +0300
@@ -68,7 +68,7 @@
 		 t->tm_hour - gmt.tm_hour);
 	tz = hours * 60 + t->tm_min - gmt.tm_min;
 	
-	len = strftime(tstr, sizeof(tstr), "%e/%b/%Y:%X", t);
+	len = strftime(tstr, sizeof(tstr), "%d/%b/%Y:%X", t);
 	if (len < 0 || len > sizeof(tstr) - 5)
 		return (NULL);
 	
