diff -Nur dsniff-2.4-old/urlsnarf.c dsniff-2.4/urlsnarf.c
--- dsniff-2.4-old/urlsnarf.c	2009-09-15 00:11:20.446201575 +0300
+++ dsniff-2.4/urlsnarf.c	2009-09-15 00:11:20.450202713 +0300
@@ -200,6 +200,7 @@
 	extern char *optarg;
 	extern int optind;
 	int c;
+	struct nids_chksum_ctl chksum_ctl;
 	
 	while ((c = getopt(argc, argv, "i:p:nvh?V")) != -1) {
 		switch (c) {
@@ -260,6 +261,12 @@
                 }
         }
 
+        chksum_ctl.netaddr = 0;
+        chksum_ctl.mask = 0;
+        chksum_ctl.action = NIDS_DONT_CHKSUM;
+
+        nids_register_chksum_ctl(&chksum_ctl, 1);
+
 	nids_run();
 	
 	/* NOTREACHED */
