diff -Nur dsniff-2.4-old/decode_aim.c dsniff-2.4/decode_aim.c
--- dsniff-2.4-old/decode_aim.c	2009-09-15 00:11:17.159202783 +0300
+++ dsniff-2.4/decode_aim.c	2009-09-15 00:11:20.660451639 +0300
@@ -14,6 +14,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <arpa/inet.h>
 
 #include "hex.h"
 #include "buf.h"
diff -Nur dsniff-2.4-old/decode_mmxp.c dsniff-2.4/decode_mmxp.c
--- dsniff-2.4-old/decode_mmxp.c	2009-09-15 00:11:17.168204279 +0300
+++ dsniff-2.4/decode_mmxp.c	2009-09-15 00:11:20.660451639 +0300
@@ -21,6 +21,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <arpa/inet.h>
 
 #include "buf.h"
 #include "decode.h"
diff -Nur dsniff-2.4-old/decode_pptp.c dsniff-2.4/decode_pptp.c
--- dsniff-2.4-old/decode_pptp.c	2009-09-15 00:11:17.162203445 +0300
+++ dsniff-2.4/decode_pptp.c	2009-09-15 00:11:20.660451639 +0300
@@ -16,6 +16,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <arpa/inet.h>
 
 #include "buf.h"
 #include "decode.h"
diff -Nur dsniff-2.4-old/decode_tds.c dsniff-2.4/decode_tds.c
--- dsniff-2.4-old/decode_tds.c	2009-09-15 00:11:20.655451840 +0300
+++ dsniff-2.4/decode_tds.c	2009-09-15 00:11:20.661451906 +0300
@@ -19,6 +19,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <strlcat.h>
+#include <arpa/inet.h>
 
 #include "decode.h"
 
diff -Nur dsniff-2.4-old/decode_vrrp.c dsniff-2.4/decode_vrrp.c
--- dsniff-2.4-old/decode_vrrp.c	2009-09-15 00:11:17.164203350 +0300
+++ dsniff-2.4/decode_vrrp.c	2009-09-15 00:11:20.661451906 +0300
@@ -15,6 +15,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <arpa/inet.h>
 
 #include "buf.h"
 #include "decode.h"
diff -Nur dsniff-2.4-old/ssh.c dsniff-2.4/ssh.c
--- dsniff-2.4-old/ssh.c	2009-09-15 00:11:20.497202277 +0300
+++ dsniff-2.4/ssh.c	2009-09-15 00:11:20.661451906 +0300
@@ -23,6 +23,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <arpa/inet.h>
 #include <unistd.h>
 
 #include "hex.h"
