diff -Nur dsniff-2.4-old/sshow.c dsniff-2.4/sshow.c
--- dsniff-2.4-old/sshow.c	2009-09-15 00:11:20.578452154 +0300
+++ dsniff-2.4/sshow.c	2009-09-15 00:11:20.762451576 +0300
@@ -224,7 +224,7 @@
 	if (debug)
 		printf("- %s -> %s: DATA (%s bytes, %.2f seconds)\n",
 			s_saddr(ts), s_daddr(ts), s_range(plain_range),
-			(float)delay / CLK_TCK);
+			(float)delay / CLOCKS_PER_SEC);
 	if (debug > 1)
 		print_data(&ts->server, cipher_size);
 
@@ -273,7 +273,7 @@
 	if (debug)
 		printf("- %s <- %s: DATA (%s bytes, %.2f seconds)\n",
 		       s_saddr(ts), s_daddr(ts), s_range(plain_range),
-		       (float)delay / CLK_TCK);
+		       (float)delay / CLOCKS_PER_SEC);
 	if (debug > 1)
 		print_data(&ts->client, cipher_size);
 	
@@ -302,7 +302,7 @@
 	
 	if (session->state == 1 &&
 #ifdef USE_TIMING
-	    now - get_history(session, 2)->timestamp >= CLK_TCK &&
+	    now - get_history(session, 2)->timestamp >= CLOCKS_PER_SEC &&
 #endif
 	    session->protocol == 1 &&
 	    (session->history.directions & 7) == 5 &&
diff -Nur dsniff-2.4-old/sshow.c.orig dsniff-2.4/sshow.c.orig
--- dsniff-2.4-old/sshow.c.orig	2009-09-15 00:11:20.579452073 +0300
+++ dsniff-2.4/sshow.c.orig	2009-09-15 00:11:20.561451664 +0300
@@ -217,6 +217,7 @@
 {
 	clock_t delay;
 	int payload;
+    long CLK_TCK= sysconf(_SC_CLK_TCK);
 
 	delay = add_history(session, 0, cipher_size, plain_range);
 
@@ -265,6 +266,7 @@
 	clock_t delay;
 	int skip;
 	range string_range;
+    long CLK_TCK= sysconf(_SC_CLK_TCK);
 	
 	delay = add_history(session, 1, cipher_size, plain_range);
 	
