diff -Nur dsniff-2.4-old/dnsspoof.8 dsniff-2.4/dnsspoof.8
--- dsniff-2.4-old/dnsspoof.8	2009-09-15 00:11:17.163202943 +0300
+++ dsniff-2.4/dnsspoof.8	2009-09-15 00:11:20.815452672 +0300
@@ -31,7 +31,7 @@
 address queries on the LAN with an answer of the local machine's IP
 address.
 .SH FILES
-.IP \fI/usr/local/lib/dnsspoof.hosts\fR
+.IP \fI/usr/share/dsniff/dnsspoof.hosts\fR
 Sample hosts file.
 .SH "SEE ALSO"
 dsniff(8), hosts(5)
diff -Nur dsniff-2.4-old/dsniff.8 dsniff-2.4/dsniff.8
--- dsniff-2.4-old/dsniff.8	2009-09-15 00:11:20.318202445 +0300
+++ dsniff-2.4/dsniff.8	2009-09-15 00:11:20.815452672 +0300
@@ -68,9 +68,9 @@
 On a hangup signal \fBdsniff\fR will dump its current trigger table to
 \fIdsniff.services\fR.
 .SH FILES
-.IP \fI/usr/local/lib/dsniff.services\fR
+.IP \fI/usr/share/dsniff/dsniff.services\fR
 Default trigger table
-.IP \fI/usr/local/lib/dsniff.magic\fR
+.IP \fI/usr/share/dsniff/dsniff.magic\fR
 Network protocol magic
 .SH "SEE ALSO"
 arpspoof(8), libnids(3), services(5), magic(5)
diff -Nur dsniff-2.4-old/Makefile.in dsniff-2.4/Makefile.in
--- dsniff-2.4-old/Makefile.in	2009-09-15 00:11:17.167204361 +0300
+++ dsniff-2.4/Makefile.in	2009-09-15 00:11:20.814453103 +0300
@@ -11,7 +11,7 @@
 install_prefix  =
 prefix          = @prefix@
 exec_prefix	= @exec_prefix@
-libdir		= @libdir@
+libdir		= $(prefix)/share/dsniff
 sbindir         = @sbindir@
 mandir		= @mandir@
 
@@ -37,8 +37,7 @@
 X11INC	= @X_CFLAGS@
 X11LIB	= @X_LIBS@ @X_PRE_LIBS@ -lXmu -lX11 @X_EXTRA_LIBS@
 
-INCS	= -I. $(NIDSINC) $(PCAPINC) $(LNETINC) $(DBINC) $(SSLINC) $(X11INC) \
-	  -I$(srcdir)/missing
+INCS	= -I. $(X11INC) -I$(srcdir)/missing 
 LIBS	= @LIBS@ -L$(srcdir) -lmissing
 
 INSTALL	= @INSTALL@
diff -Nur dsniff-2.4-old/pathnames.h dsniff-2.4/pathnames.h
--- dsniff-2.4-old/pathnames.h	2009-09-15 00:11:17.162203445 +0300
+++ dsniff-2.4/pathnames.h	2009-09-15 00:11:20.815452672 +0300
@@ -12,7 +12,7 @@
 #define PATHNAMES_H
 
 #ifndef DSNIFF_LIBDIR
-#define DSNIFF_LIBDIR		"/usr/local/lib/"
+#define DSNIFF_LIBDIR		"/usr/share/dsniff/"
 #endif
 
 #define DSNIFF_SERVICES		"dsniff.services"
