diff --git a/setup.py b/setup.py
index 45af61f..8ea5de5 100644
--- a/setup.py
+++ b/setup.py
@@ -183,11 +183,11 @@ for f in glob.glob(os.path.join('lib', '*')):
         pygame_data_files.append(f)
 
 #tests/fixtures
-add_datafiles(data_files, 'pygame/tests',
-              ['test',
-                  [['fixtures',
-                      [['xbm_cursors',
-                          ['*.xbm']]]]]])
+#add_datafiles(data_files, 'pygame/tests',
+#              ['test',
+#                  [['fixtures',
+#                      [['xbm_cursors',
+#                          ['*.xbm']]]]]])
 
 #examples
 add_datafiles(data_files, 'pygame/examples',
@@ -461,25 +461,25 @@ date_files = [(path, files) for path, files in data_files if files]
 PACKAGEDATA = {
        "cmdclass":    cmdclass,
        "packages":    ['pygame', 'pygame.gp2x', 'pygame.threads',
-                       'pygame.tests',
-                       'pygame.tests.test_utils',
-                       'pygame.tests.run_tests__tests',
-                       'pygame.tests.run_tests__tests.all_ok',
-                       'pygame.tests.run_tests__tests.failures1',
-                       'pygame.tests.run_tests__tests.incomplete',
-                       'pygame.tests.run_tests__tests.infinite_loop',
-                       'pygame.tests.run_tests__tests.print_stderr',
-                       'pygame.tests.run_tests__tests.print_stdout',
-                       'pygame.tests.run_tests__tests.incomplete_todo',
-                       'pygame.tests.run_tests__tests.exclude',
-                       'pygame.tests.run_tests__tests.timeout',
-                       'pygame.tests.run_tests__tests.everything',
+#                       'pygame.tests',
+#                       'pygame.tests.test_utils',
+#                       'pygame.tests.run_tests__tests',
+#                       'pygame.tests.run_tests__tests.all_ok',
+#                       'pygame.tests.run_tests__tests.failures1',
+#                       'pygame.tests.run_tests__tests.incomplete',
+#                       'pygame.tests.run_tests__tests.infinite_loop',
+#                       'pygame.tests.run_tests__tests.print_stderr',
+#                       'pygame.tests.run_tests__tests.print_stdout',
+#                       'pygame.tests.run_tests__tests.incomplete_todo',
+#                       'pygame.tests.run_tests__tests.exclude',
+#                       'pygame.tests.run_tests__tests.timeout',
+#                       'pygame.tests.run_tests__tests.everything',
                        'pygame.docs',
                        'pygame.examples'],
        "package_dir": {'pygame': 'lib',
                        'pygame.threads': 'lib/threads',
                        'pygame.gp2x': 'lib/gp2x',
-                       'pygame.tests': 'test',
+#                       'pygame.tests': 'test',
                        'pygame.docs': 'docs',
                        'pygame.examples': 'examples'},
        "headers":     headers,
