Index: libvpx/configure
===================================================================
--- libvpx.orig/configure	2011-08-09 10:06:56.059395856 +0200
+++ libvpx/configure	2011-08-09 10:09:23.928129091 +0200
@@ -429,12 +429,6 @@
 }
 
 process_detect() {
-    if enabled shared; then
-        # Can only build shared libs on a subset of platforms. Doing this check
-        # here rather than at option parse time because the target auto-detect
-        # magic happens after the command line has been parsed.
-        enabled linux || die "--enable-shared only supported on ELF for now"
-    fi
     if [ -z "$CC" ]; then
         echo "Bypassing toolchain for environment detection."
         enable external_build
