--- libvpx-v0.9.7/examples.mk.no-static-lib	2011-08-04 19:30:15.000000000 +0200
+++ libvpx-v0.9.7/examples.mk	2011-08-14 11:34:37.000000000 +0200
@@ -169,8 +169,6 @@ BINS-$(NOT_MSVS)           += $(addprefi
 CODEC_LIB=$(if $(CONFIG_DEBUG_LIBS),vpx_g,vpx)
 CODEC_LIB_SUF=$(if $(CONFIG_SHARED),.so,.a)
 $(foreach bin,$(BINS-yes),\
-    $(if $(BUILD_OBJS),$(eval $(bin):\
-        $(LIB_PATH)/lib$(CODEC_LIB)$(CODEC_LIB_SUF)))\
     $(if $(BUILD_OBJS),$(eval $(call linker_template,$(bin),\
         $(call objs,$($(notdir $(bin)).SRCS)) \
         -l$(CODEC_LIB) $(addprefix -l,$(CODEC_EXTRA_LIBS))\
