diff -Nuar glibc-2.19.orig/sysdeps/ieee754/dbl-64/s_sin.c glibc-2.19/sysdeps/ieee754/dbl-64/s_sin.c
--- glibc-2.19.orig/sysdeps/ieee754/dbl-64/s_sin.c	2014-02-07 11:04:38.000000000 +0200
+++ glibc-2.19/sysdeps/ieee754/dbl-64/s_sin.c	2014-03-23 01:37:31.914697336 +0200
@@ -447,19 +447,21 @@
 	    }
 	  else
 	    {
+          double t;
 	      if (a > 0)
 		{
 		  m = 1;
+          t = a;
 		  db = da;
 		}
 	      else
 		{
 		  m = 0;
-		  a = -a;
+		  t = -a;
 		  db = -da;
 		}
-	      u.x = big + a;
-	      y = a - (u.x - big);
+		  u.x = big + t;
+          y = t - (u.x - big);
 	      res = do_sin (u, y, db, &cor);
 	      cor = (cor > 0) ? 1.035 * cor + eps : 1.035 * cor - eps;
 	      retval = ((res == res + cor) ? ((m) ? res : -res)
@@ -670,20 +672,22 @@
 	      retval = (res == res + cor) ? res : bsloww (a, da, x, n);
 	    }
 	  else
-	    {
+	    { 
+          double t;
 	      if (a > 0)
 		{
 		  m = 1;
+          t = a ;
 		  db = da;
 		}
 	      else
 		{
 		  m = 0;
-		  a = -a;
+		  t = -a;
 		  db = -da;
 		}
-	      u.x = big + a;
-	      y = a - (u.x - big);
+	      u.x = big + t;
+	      y = t - (u.x - big);
 	      res = do_sin (u, y, db, &cor);
 	      cor = (cor > 0) ? 1.035 * cor + eps : 1.035 * cor - eps;
 	      retval = ((res == res + cor) ? ((m) ? res : -res)
