--- glibc-2.3.4/localedata/Makefile.orig	2004-08-07 16:38:13.000000000 -0700
+++ glibc-2.3.4/localedata/Makefile	2005-03-15 14:07:36.000000000 -0800
@@ -206,7 +206,7 @@ INSTALL-SUPPORTED-LOCALES=$(addprefix in
 
 # Sometimes the whole collection of locale files should be installed.
 LOCALEDEF=I18NPATH=. GCONV_PATH=$(common-objpfx)iconvdata LC_ALL=C \
-$(common-objpfx)elf/ld.so --library-path $(rpath-link) $(common-objpfx)locale/localedef
+$(common-objpfx)elf/ld.so --library-path $(rpath-link) $(common-objpfx)locale/localedef --no-archive
 install-locales: $(INSTALL-SUPPORTED-LOCALES)
 
 install-locales-dir:
