
Adapt for kernel 3.19

Signed-off-by: Thomas Backlund <tmb@mageia.org>

diff -Nurp NVIDIA-Linux-x86_64-304.125.orig/kernel/nv.c NVIDIA-Linux-x86_64-304.125/kernel/nv.c
--- NVIDIA-Linux-x86_64-304.125.orig/kernel/nv.c	2014-12-02 05:58:34.000000000 +0200
+++ NVIDIA-Linux-x86_64-304.125/kernel/nv.c	2015-01-25 19:14:21.009109856 +0200
@@ -2026,7 +2026,11 @@ long nv_kern_unlocked_ioctl(
     unsigned long i_arg
 )
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 19, 0)
+    return nv_kern_ioctl(file->f_path.dentry->d_inode, file, cmd, i_arg);
+#else
     return nv_kern_ioctl(file->f_dentry->d_inode, file, cmd, i_arg);
+#endif
 }
 
 long nv_kern_compat_ioctl(
@@ -2035,7 +2039,11 @@ long nv_kern_compat_ioctl(
     unsigned long i_arg
 )
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 19, 0)
+    return nv_kern_ioctl(file->f_path.dentry->d_inode, file, cmd, i_arg);
+#else
     return nv_kern_ioctl(file->f_dentry->d_inode, file, cmd, i_arg);
+#endif
 }
 
 /*
