Index: Makefile
===================================================================
--- Makefile.orig
+++ Makefile
@@ -6,13 +6,13 @@ GCJ?=gcj
 CC?=gcc
 LD?=gcc
 JPPFLAGS+=-C -P
-CFLAGS+=-Wall -Os -pedantic -Werror
+CFLAGS+=-Wall -pedantic
 CSTD?=-std=c99
 CSHAREFLAG+=-fpic -fno-stack-protector
 GCJJNIFLAG=-fjni
 JVERCFLAGS+=-source 1.5
 JCFLAGS+=
-INCLUDES+=-I$(JAVA_HOME)/include -I$(JAVA_HOME)/include/linux
+INCLUDES=-I$(JAVA_HOME)/include -I$(JAVA_HOME)/include/linux
 JAVADOCFLAGS?=-quiet -author -link http://java.sun.com/j2se/1.4.2/docs/api/
 
 LDVER?=$(shell ld -v | cut -d' ' -f1)
@@ -24,7 +24,7 @@ else
 LDSHAREFLAGS+=-lc
 endif
 
-PREFIX?=/usr/local
+PREFIX?=/usr
 JARDIR?=$(PREFIX)/share/java
 DOCDIR?=$(PREFIX)/share/doc/libmatthew-java/
 LIBDIR?=$(PREFIX)/lib/jni
@@ -62,12 +62,7 @@ cgi-$(CGIVER).jar: .classes
 io-$(IOVER).jar: .classes
 	(cd classes; $(JAR) cf ../$@ cx/ath/matthew/io/*class)
 unix-$(UNIXVER).jar: .classes
-ifeq ($(DEBUG),enable)
-	echo "Class-Path: $(JARDIR)/debug-$(DEBUG).jar" > Manifest
-else
-	echo "Class-Path: " > Manifest
-endif
-	(cd classes; $(JAR) cfm ../$@ ../Manifest cx/ath/matthew/unix/*class)
+	(cd classes; $(JAR) cf ../$@ cx/ath/matthew/unix/*class)
 
 hexdump-$(HEXVER).jar: .classes
 	(cd classes; $(JAR) cf ../$@ cx/ath/matthew/utils/Hexdump.class)
@@ -75,7 +70,7 @@ hexdump-$(HEXVER).jar: .classes
 %.o: %.c %.h
 	$(CC) $(CFLAGS) $(CSTD) $(CSHAREFLAG) $(INCLUDES) -c -o $@ $<
 lib%.so: %.o
-	$(CC) $(LDFLAGS) $(LDSHAREFLAGS) -o $@ $<
+	$(CC) $(LDFLAGS) $(LDSHAREFLAGS) -o $@ $< $(LDLIBS)
 unix-java.h: .classes
 	$(JAVAH) -classpath classes -o $@ cx.ath.matthew.unix.UnixServerSocket cx.ath.matthew.unix.UnixSocket cx.ath.matthew.unix.USInputStream cx.ath.matthew.unix.USOutputStream
 cgi-java.h: .classes
@@ -90,11 +85,11 @@ libmatthew-java-$(MATTVER).tar.gz: Makef
 	tar zcf $@ libmatthew-java-$(MATTVER)
 
 debug-enable-$(DEBUGVER).jar: cx/ath/matthew/debug/Debug.jpp
-	make .enabledebug
+	$(MAKE) .enabledebug
 	echo "Class-Path: $(JARDIR)/hexdump.jar" > Manifest
 	(cd classes;jar cfm ../$@ ../Manifest cx/ath/matthew/debug/*.class)
 debug-disable-$(DEBUGVER).jar: cx/ath/matthew/debug/Debug.jpp
-	make .disabledebug
+	$(MAKE) .disabledebug
 	echo "Class-Path: $(JARDIR)/hexdump.jar" > Manifest
 	(cd classes;jar cfm ../$@ ../Manifest cx/ath/matthew/debug/*.class)
 .enabledebug: cx/ath/matthew/debug/Debug.jpp 
