Index: kdelibs-4.7.1/kfile/kfilepreviewgenerator.cpp
===================================================================
--- kdelibs-4.7.1.orig/kfile/kfilepreviewgenerator.cpp
+++ kdelibs-4.7.1/kfile/kfilepreviewgenerator.cpp
@@ -484,7 +484,8 @@ KFilePreviewGenerator::Private::Private(
     m_enabledPlugins = globalConfig.readEntry("Plugins", QStringList()
                                                          << "directorythumbnail"
                                                          << "imagethumbnail"
-                                                         << "jpegthumbnail");
+                                                         << "svgthumbnail"
+                                                         << "ffmpegthumbs");
 
     // If the user is upgrading from KDE <= 4.6, we must check if he had the 'jpegrotatedthumbnail' plugin enabled.
     // This plugin does not exist any more in KDE >= 4.7, so we have to replace it with the 'jpegthumbnail' plugin.
