Index: kdelibs-4.6.1/kded/kbuildservicefactory.cpp
===================================================================
--- kdelibs-4.6.1.orig/kded/kbuildservicefactory.cpp
+++ kdelibs-4.6.1/kded/kbuildservicefactory.cpp
@@ -258,7 +258,7 @@ void KBuildServiceFactory::populateServi
                         // Create those on demand
                         m_mimeTypeFactory->createFakeMimeType(stName);
                     } else {
-                        kDebug(7021) << service->entryPath() << "specifies undefined mimetype/servicetype" << stName;
+                        //kDebug(7021) << service->entryPath() << "specifies undefined mimetype/servicetype" << stName;
                         // technically we could call addServiceOffer here, 'mime' isn't used. But it
                         // would be useless, since the loops for writing out the offers iterate
                         // over all known servicetypes and mimetypes. Unknown -> never written out.
