Index: gimp-2.7.2/app/actions/windows-actions.c
===================================================================
--- gimp-2.7.2.orig/app/actions/windows-actions.c
+++ gimp-2.7.2/app/actions/windows-actions.c
@@ -110,7 +110,7 @@ static const GimpToggleActionEntry windo
     NC_("windows-action", "Single-Window Mode"), NULL,
     NC_("windows-action", "When enabled GIMP is in a single-window mode. Far from completely implemented!"),
     G_CALLBACK (windows_use_single_window_mode_cmd_callback),
-    FALSE,
+    TRUE,
     GIMP_HELP_WINDOWS_USE_SINGLE_WINDOW_MODE }
 };
 
