diff -ur clutter-gst-1.6.0/clutter-gst/clutter-gst-video-sink.c clutter-gst-1.6.0.new/clutter-gst/clutter-gst-video-sink.c
--- clutter-gst-1.6.0/clutter-gst/clutter-gst-video-sink.c	2012-04-27 11:11:08.000000000 +0000
+++ clutter-gst-1.6.0.new/clutter-gst/clutter-gst-video-sink.c	2013-04-26 10:09:18.000000000 +0000
@@ -71,6 +71,7 @@
 
 #include <glib.h>
 #include <string.h>
+#include <GL/gl.h>
 
 /* Flags to give to cogl_texture_new(). Since clutter 1.1.10 put NO_ATLAS to
  * be sure the frames don't end up in an atlas */
