diff -Nur goonies-1.4.1528-old/src/SoundManager.cpp goonies-1.4.1528/src/SoundManager.cpp
--- goonies-1.4.1528-old/src/SoundManager.cpp	2009-07-22 14:36:23.388530992 +0300
+++ goonies-1.4.1528/src/SoundManager.cpp	2009-07-22 14:37:35.786530589 +0300
@@ -92,7 +92,7 @@
         char filename[256];
         Mix_Chunk *t;
 
-        sprintf(filename, "sound/%s", name->get());
+        sprintf(filename, "/usr/share/goonies/sound/%s", name->get());
         t = load_sound(filename);
         n = new SoundManagerNode();
         n->m_sfx = t;
@@ -150,7 +150,7 @@
     // FIXME: unfortunately the entire game has a hardcoded use of the sound/ dir...
     char path[256];
     char *filename;
-    sprintf(path, "sound/%s", localpath);
+    sprintf(path, "/usr/share/goonies/sound/%s", localpath);
 
 #ifdef _WIN32
 
