diff -Nur goonies-1.4.1528-old/src/TheGooniesApp.cpp goonies-1.4.1528/src/TheGooniesApp.cpp
--- goonies-1.4.1528-old/src/TheGooniesApp.cpp	2009-07-22 14:33:26.088530264 +0300
+++ goonies-1.4.1528/src/TheGooniesApp.cpp	2009-07-22 14:34:24.196530171 +0300
@@ -99,7 +99,7 @@
 
     m_screen_dx = SCREEN_X;
     m_screen_dy = SCREEN_Y;
-    fullscreen = true;
+    fullscreen = false;
 
 	screen_shake = true;
 	water_reflection = true;
