--- dbus-sharp-glib-0.5.0.orig/debian/changelog
+++ dbus-sharp-glib-0.5.0/debian/changelog
@@ -0,0 +1,25 @@
+dbus-sharp-glib (0.5.0-4) unstable; urgency=low
+
+  * Upload rebuild to unstable for CLR 4.0 transition
+  * [5697490] Stop import deprecated cli.make
+  * [d6a7c8b] Standards-Version → 3.9.2, no changes required
+
+ -- Iain Lane <laney@debian.org>  Tue, 24 Jan 2012 22:59:55 +0000
+
+dbus-sharp-glib (0.5.0-3) unstable; urgency=low
+
+  * Upload to Sid
+
+ -- Jo Shields <directhex@apebox.org>  Tue, 15 Feb 2011 22:08:21 +0000
+
+dbus-sharp-glib (0.5.0-2) experimental; urgency=low
+
+  * Fix substitution which FTBFSed with dash as sh
+
+ -- Iain Lane <laney@ubuntu.com>  Tue, 16 Nov 2010 23:28:27 +0000
+
+dbus-sharp-glib (0.5.0-1) experimental; urgency=low
+
+  * Initial release. (Closes: #598946)
+
+ -- Mirco Bauer <meebey@debian.org>  Sun, 03 Oct 2010 21:32:47 +0200
--- dbus-sharp-glib-0.5.0.orig/debian/compat
+++ dbus-sharp-glib-0.5.0/debian/compat
@@ -0,0 +1 @@
+8
--- dbus-sharp-glib-0.5.0.orig/debian/libdbus-glib1.0-cil.install
+++ dbus-sharp-glib-0.5.0/debian/libdbus-glib1.0-cil.install
@@ -0,0 +1 @@
+/usr/lib/cli/
--- dbus-sharp-glib-0.5.0.orig/debian/libdbus-glib1.0-cil.installcligac
+++ dbus-sharp-glib-0.5.0/debian/libdbus-glib1.0-cil.installcligac
@@ -0,0 +1 @@
+/usr/lib/cli/dbus-sharp-glib-1.0/dbus-sharp-glib.dll
--- dbus-sharp-glib-0.5.0.orig/debian/libdbus-glib1.0-cil-dev.install
+++ dbus-sharp-glib-0.5.0/debian/libdbus-glib1.0-cil-dev.install
@@ -0,0 +1 @@
+/usr/lib/pkgconfig/
--- dbus-sharp-glib-0.5.0.orig/debian/rules
+++ dbus-sharp-glib-0.5.0/debian/rules
@@ -0,0 +1,44 @@
+#!/usr/bin/make -f
+#export DH_VERBOSE=1
+
+DEB_API_VERSION = 0.5
+DEB_ABI_VERSION = 1.0
+DEB_CLI_DIR = debian/tmp/usr/lib/cli/dbus-sharp-glib-$(DEB_ABI_VERSION)/
+DEB_DOC_DIR = $(CURDIR)/doc
+DEB_LIB_FILES = $(CURDIR)/src/dbus-sharp-glib.dll
+DEB_LIBCONF_FILES = $(CURDIR)/src/dbus-sharp-glib.dll.config
+DEB_CLEAN_FILES = $(CURDIR)/src/dbus-sharp-glib.zip $(CURDIR)/src/dbus-sharp-glib.tree $(DEB_DOC_DIR)
+
+override_dh_auto_configure:
+	dh_auto_configure -- GMCS=/usr/bin/mono-csc
+
+override_dh_auto_build:
+	dh_auto_build
+	mkdir -p $(DEB_DOC_DIR);
+	for LIB in $(DEB_LIB_FILES); do \
+		mdoc update \
+			-fno-assembly-versions \
+			--out=$(DEB_DOC_DIR) \
+			$$LIB; \
+		mdoc assemble \
+			--format ecma \
+			--out $${LIB%.dll} \
+			$(DEB_DOC_DIR); \
+	done
+
+override_dh_auto_install:
+	dh_auto_install
+	mkdir -p $(DEB_CLI_DIR)
+	for FILE in $(DEB_LIB_FILES) $(DEB_LIBCONF_FILES); do \
+		install $$FILE $(CURDIR)/$(DEB_CLI_DIR); \
+	done
+
+override_dh_auto_clean:
+	dh_auto_clean
+	rm -rf $(DEB_CLEAN_FILES)
+
+override_dh_makeclilibs:
+	dh_makeclilibs -m$(DEB_API_VERSION)
+
+%:
+	dh $@ --with cli
--- dbus-sharp-glib-0.5.0.orig/debian/watch
+++ dbus-sharp-glib-0.5.0/debian/watch
@@ -0,0 +1,2 @@
+version=3
+http://github.com/mono/dbus-sharp/downloads .+/dbus-sharp-glib-([\d\.]+)\.tar\.gz
--- dbus-sharp-glib-0.5.0.orig/debian/control
+++ dbus-sharp-glib-0.5.0/debian/control
@@ -0,0 +1,45 @@
+Source: dbus-sharp-glib
+Section: cli-mono
+Priority: optional
+Maintainer: Debian CLI Libraries Team <pkg-cli-libs-team@lists.alioth.debian.org>
+Uploaders: Mirco Bauer <meebey@debian.org>
+Build-Depends: debhelper (>= 8)
+Build-Depends-Indep: cli-common-dev (>= 0.5.7),
+ mono-devel (>= 2.4.3),
+ pkg-config,
+ libdbus1.0-cil-dev (>= 0.7),
+ monodoc-base
+Standards-Version: 3.9.2
+Homepage: http://github.com/mono/dbus-sharp-glib
+Vcs-Git: git://git.debian.org/pkg-cli-libs/packages/dbus-sharp-glib.git
+Vcs-Browser: http://git.debian.org/?p=pkg-cli-libs/packages/dbus-sharp-glib.git
+
+Package: libdbus-glib1.0-cil
+Architecture: all
+Depends: ${cli:Depends}, ${misc:Depends}
+Description: CLI implementation of D-Bus (GLib mainloop integration)
+ dbus-sharp is a fork of ndesk-dbus, which is a C# implementation of
+ D-Bus. It's often referred to as "managed D-Bus" to avoid confusion
+ with existing bindings (which wrap libdbus).
+ .
+ D-Bus is a message bus, used for sending messages between
+ applications.  Conceptually, it fits somewhere in between raw sockets
+ and CORBA in terms of complexity.
+ .
+ This package provides integration into the GLib mainloop and contains
+ the dbus-sharp-glib library itself.
+
+Package: libdbus-glib1.0-cil-dev
+Architecture: all
+Depends: libdbus-glib1.0-cil (= ${binary:Version}), libdbus1.0-cil-dev, ${misc:Depends}
+Description: CLI implementation of D-Bus (GLib mainloop integration) - development files
+ dbus-sharp is a fork of ndesk-dbus, which is a C# implementation of
+ D-Bus. It's often referred to as "managed D-Bus" to avoid confusion
+ with existing bindings (which wrap libdbus).
+ .
+ D-Bus is a message bus, used for sending messages between
+ applications.  Conceptually, it fits somewhere in between raw sockets
+ and CORBA in terms of complexity.
+ .
+ This package contains development files for dbus-sharp-glib, and
+ should be used for compilation.
--- dbus-sharp-glib-0.5.0.orig/debian/copyright
+++ dbus-sharp-glib-0.5.0/debian/copyright
@@ -0,0 +1,46 @@
+Format-Specification: http://wiki.debian.org/Proposals/CopyrightFormat
+Upstream-Name: dbus-sharp-glib
+Upstream-Maintainer: dbus-sharp team
+Upstream-Source: http://github.com/mono/dbus-sharp/downloads
+Upstream-Vcs-Browser: http://github.com/mono/dbus-sharp-glib/tree/master
+
+Files: *
+Copyright: 2006-2009 Alp Toker <alp@atoker.com>
+License: MIT/X11
+ Permission is hereby granted, free of charge, to any person obtaining
+ a copy of this software and associated documentation files (the
+ "Software"), to deal in the Software without restriction, including
+ without limitation the rights to use, copy, modify, merge, publish,
+ distribute, sublicense, and/or sell copies of the Software, and to
+ permit persons to whom the Software is furnished to do so, subject to
+ the following conditions:
+
+ The above copyright notice and this permission notice shall be
+ included in all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+Files: debian/*
+Copyright: 2010 Mirco Bauer <meebey@debian.org>
+License: GPL
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,  
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ On Debian systems the full text of the GNU General Public License can be
+ found in the `/usr/share/common-licenses/GPL' file.
