--- scripts/mumble.desktop~	2009-02-26 16:15:57.000000000 +0200
+++ scripts/mumble.desktop	2009-02-26 16:15:58.000000000 +0200
@@ -1,7 +1,9 @@
 [Desktop Entry]
 Name=Mumble
 GenericName=Voice Chat
+GenericName=Oyunlar İçin Sesli Konuşma İstemcisi
 Comment=A low-latency, high quality voice chat program for gaming
+Comment[tr]=Oyunlar için düşük gecikmeli ve yüksek kaliteli sesli konuşma yazılımı
 Exec=mumble
 Icon=mumble
 Terminal=false
