diff -aur mumble-1.2.3.orig/src/murmur/murmur.pro mumble-1.2.3/src/murmur/murmur.pro
--- mumble-1.2.3.orig/src/murmur/murmur.pro	2011-02-19 22:35:19.000000000 +0100
+++ mumble-1.2.3/src/murmur/murmur.pro	2011-03-30 19:29:46.614774630 +0200
@@ -69,7 +69,7 @@
 	win32 {
 		slice.commands = slice2cpp --checksum -I\"$$ICE_PATH/slice\" Murmur.ice
 	} else {
-		slice.commands = slice2cpp --checksum -I/usr/local/share/Ice -I/usr/share/Ice/slice -I/usr/share/slice -I/usr/share/Ice-3.4.1/slice/ -I/usr/share/Ice-3.3.1/slice/ Murmur.ice
+		slice.commands = slice2cpp --checksum -I/usr/local/share/Ice -I/usr/share/Ice/slice -I/usr/share/slice -I/usr/share/Ice-3.4.1/slice/ -I/usr/share/Ice-3.3.1/slice/ -I/usr/share/Ice-3.4.2/slice/ -I/usr/share/Ice-3.5.0/slice/ Murmur.ice
 	}
 
 	slice.depends = Murmur.ice