diff -uNr mumble-1.2.3.old/scripts/murmur.logrotate mumble-1.2.3/scripts/murmur.logrotate
--- mumble-1.2.3.old/scripts/murmur.logrotate	2011-02-19 22:35:16.000000000 +0100
+++ mumble-1.2.3/scripts/murmur.logrotate	2012-05-29 01:04:47.591309905 +0200
@@ -5,6 +5,6 @@
 	delaycompress
 	missingok
 	postrotate
-		start-stop-daemon --quiet --oknodo --stop --signal 1 --exec /usr/sbin/murmurd --user mumble-server --pidfile /var/run/mumble-server/mumble-server.pid
+		/bin/kill -HUP `cat /var/run/mumble-server/mumble-server.pid 2>/dev/null`  2> /dev/null || true
 	endscript
 }