diff --git a/kernel/module.c b/kernel/module.c
index 921bed4..382414e 100644
--- a/kernel/module.c
+++ b/kernel/module.c
@@ -2873,6 +2873,10 @@ static int check_module_license_and_versions(struct module *mod)
 	if (strcmp(mod->name, "ndiswrapper") == 0)
 		add_taint(TAINT_PROPRIETARY_MODULE, LOCKDEP_NOW_UNRELIABLE);
 
+	/* vbox is garbage. */
+	if (strcmp(mod->name, "vboxdrv") == 0)
+		add_taint(TAINT_CRAP, LOCKDEP_NOW_UNRELIABLE);
+
 	/* driverloader was caught wrongly pretending to be under GPL */
 	if (strcmp(mod->name, "driverloader") == 0)
 		add_taint_module(mod, TAINT_PROPRIETARY_MODULE,
