--- lynx2-8-8/lynx.cfg.orig	2012-11-18 19:20:40.000000000 -0500
+++ lynx2-8-8/lynx.cfg	2012-11-30 10:50:55.589585463 -0500
@@ -1,7 +1,8 @@
 # $LynxId: lynx.cfg,v 1.222 2012/11/14 01:18:19 tom Exp $
 # lynx.cfg file.
-# The default placement for this file is /usr/local/lib/lynx.cfg (Unix)
-#                                     or Lynx_Dir:lynx.cfg (VMS)
+#
+# Please don't edit this file directly (it is updated with every package
+# update, overwriting your changes). Instead, edit /etc/lynx-site.cfg.
 #
 # $Format: "#PRCS LYNX_VERSION \"$ProjectVersion$\""$
 #PRCS LYNX_VERSION "2.8.8dev.15"
@@ -90,7 +91,7 @@
 #
 # Normally we expect you will connect to a remote site, e.g., the Lynx starting
 # site:
-STARTFILE:http://lynx.isc.org/
+STARTFILE:http://www.mageia.org/
 #
 # As an alternative, you may want to use a local URL.  A good choice for this is
 # the user's home directory:
@@ -114,9 +115,9 @@
 # This definition will be overridden if the "LYNX_HELPFILE" environment
 # variable has been set.
 #
-HELPFILE:http://lynx.isc.org/release/lynx2-8-7/lynx_help/lynx_help_main.html
+#HELPFILE:http://lynx.isc.org/release/lynx2-8-7/lynx_help/lynx_help_main.html
 .ex
-#HELPFILE:file://localhost/PATH_TO/lynx_help/lynx_help_main.html
+HELPFILE:file://localhost//usr/share/lynx_help/lynx_help_main.html
 
 .h2 DEFAULT_INDEX_FILE
 # DEFAULT_INDEX_FILE is the default file retrieved when the
@@ -557,7 +558,7 @@
 # Older browser versions will fail so keep NCR_IN_BOOKMARKS:FALSE if you
 # plan to use them.
 #
-#NCR_IN_BOOKMARKS:FALSE
+NCR_IN_BOOKMARKS:TRUE
 
 .h2 FORCE_8BIT_TOUPPER
 # FORCE_8BIT_TOUPPER overrides locale settings and uses internal 8-bit
@@ -2252,7 +2253,7 @@
 # Make this empty (but not commented out) if you don't have such a viewer or
 # want to disable the built-in default viewer mappings for image types.
 #
-#XLOADIMAGE_COMMAND:xv %s
+XLOADIMAGE_COMMAND:xli %s
 #
 # Unix:
 # =====
@@ -2319,11 +2320,11 @@
 #
 .ex 7
 #VIEWER:application/postscript:ghostview %s&:XWINDOWS
-#VIEWER:image/gif:xli %s&:XWINDOWS
-#VIEWER:image/x-xbm:xli %s&:XWINDOWS
-#VIEWER:image/png:xli %s&:XWINDOWS
-#VIEWER:image/tiff:xli %s&:XWINDOWS
-#VIEWER:image/jpeg:xli %s&:XWINDOWS
+VIEWER:image/gif:xli %s&:XWINDOWS
+VIEWER:image/x-xbm:xli %s&:XWINDOWS
+VIEWER:image/png:xli %s&:XWINDOWS
+VIEWER:image/tiff:xli %s&:XWINDOWS
+VIEWER:image/jpeg:xli %s&:XWINDOWS
 #VIEWER:video/mpeg:mpeg_play %s &:XWINDOWS
 
 .h2 GLOBAL_MAILCAP
@@ -3650,3 +3651,6 @@
 #
 # See also BLAT_MAIL and SYSTEM_MAIL flags.
 #ALT_BLAT_MAIL:FALSE
+
+# included last to allow overriding configuration above
+INCLUDE:/etc/lynx-site.cfg
