diff -ur liblzf-1.6.orig/Makefile.in liblzf-1.6/Makefile.in
--- liblzf-1.6.orig/Makefile.in	2006-07-07 18:31:41.000000000 +0300
+++ liblzf-1.6/Makefile.in	2007-01-13 03:48:02.992320097 +0200
@@ -37,12 +37,12 @@
 	$(RANLIB) $@
 
 install: all
-	$(INSTALL) -d $(bindir)
-	$(INSTALL) -m 755 lzf $(bindir)
-	$(INSTALL) -d $(includedir)
-	$(INSTALL_DATA) lzf.h $(includedir)
-	$(INSTALL) -d $(libdir)
-	$(INSTALL_DATA) liblzf.a $(libdir)
+	$(INSTALL) -d $(DESTDIR)/$(bindir)
+	$(INSTALL) -m 755 lzf $(DESTDIR)/$(bindir)
+	$(INSTALL) -d $(DESTDIR)/$(includedir)
+	$(INSTALL_DATA) lzf.h $(DESTDIR)/$(includedir)
+	$(INSTALL) -d $(DESTDIR)/$(libdir)
+	$(INSTALL_DATA) liblzf.a $(DESTDIR)/$(libdir)
 
 dist:
 	mkdir liblzf-$(VERSION)
