Index: liblzf-3.5/lzfP.h
===================================================================
--- liblzf-3.5.orig/lzfP.h
+++ liblzf-3.5/lzfP.h
@@ -52,7 +52,7 @@
  * For best compression, use 15 or 16 (or more, up to 23).
  */
 #ifndef HLOG
-# define HLOG 16
+# define HLOG 13
 #endif
 
 /*
@@ -72,7 +72,7 @@
  * possibly disable this for text data.
  */
 #ifndef ULTRA_FAST
-# define ULTRA_FAST 0
+# define ULTRA_FAST 1
 #endif
 
 /*
