Index: kde-baseapps-4.7.1/dolphin/src/settings/general/previewssettingspage.cpp
===================================================================
--- kde-baseapps-4.7.1.orig/dolphin/src/settings/general/previewssettingspage.cpp
+++ kde-baseapps-4.7.1/dolphin/src/settings/general/previewssettingspage.cpp
@@ -205,7 +205,8 @@ void PreviewsSettingsPage::loadSettings(
     m_enabledPreviewPlugins = globalConfig.readEntry("Plugins", QStringList()
                                                      << QLatin1String("directorythumbnail")
                                                      << QLatin1String("imagethumbnail")
-                                                     << QLatin1String("jpegthumbnail"));
+                                                     << QLatin1String("svgthumbnail")
+                                                     << QLatin1String("ffmpegthumbs"));
 
     // If the user is upgrading from KDE <= 4.6, we must check if he had the 'jpegrotatedthumbnail' plugin enabled.
     // This plugin does not exist any more in KDE >= 4.7, so we have to replace it with the 'jpegthumbnail' plugin.
