Index: kdebase-4.6.3/dolphin/src/dolphinmainwindow.cpp
===================================================================
--- kdebase-4.6.3.orig/dolphin/src/dolphinmainwindow.cpp
+++ kdebase-4.6.3/dolphin/src/dolphinmainwindow.cpp
@@ -1663,7 +1663,6 @@ void DolphinMainWindow::setupDockWidgets
 
     const bool firstRun = DolphinSettings::instance().generalSettings()->firstRun();
     if (firstRun) {
-        infoDock->hide();
         foldersDock->hide();
 #ifndef Q_OS_WIN
         terminalDock->hide();
