Index: recordmydesktop-0.3.8.1/src/Makefile.am
===================================================================
--- recordmydesktop-0.3.8.1.orig/src/Makefile.am
+++ recordmydesktop-0.3.8.1/src/Makefile.am
@@ -76,7 +76,7 @@ recordmydesktop_SOURCES = \
 	skeleton.h
 
 recordmydesktop_CPPFLAGS = -D_THREAD_SAFE -pthread -Wall
-recordmydesktop_LDFLAGS  = @X_LIBS@ @X_EXTRA_LIBS@ @X_PRE_LIBS@
+recordmydesktop_LDFLAGS  =
 
 
 # RectInsert test
