diff --git a/configure.ac b/configure.ac
index 2de2f9e..c817753 100644
--- a/configure.ac
+++ b/configure.ac
@@ -95,7 +95,7 @@ AC_CHECK_LIB([pthread],[pthread_mutex_lock],,
              AC_MSG_ERROR([Can't find libpthread]))
 
 if test x$jack = xtrue ; then
-AC_CHECK_LIB([jack],[jack_activate],jack_lib_found=true,)
+AC_CHECK_LIB([jack],[jack_activate],,)
 fi
 
 if test x$oss = xfalse; then
@@ -119,18 +119,12 @@ AC_OUTPUT
 if test x$audio_backend != xOSS; then
     audio_backend="ALSA"
 fi
-if test x$jack_lib_found = xtrue ; then
-    jack_support="Enabled"
-else
-    jack_support="Disabled"
-fi
 echo ""
 echo ""
 echo "****************************************"
 echo ""
 echo "Audio driver that will be used: $audio_backend"
 echo ""
-echo "Compile with Jack support: $jack_support"
 echo ""
 echo "****************************************"
 echo ""
