diff -up recordmydesktop-0.3.8.1/src/rmd_types.h.ALSA-default recordmydesktop-0.3.8.1/src/rmd_types.h
--- recordmydesktop-0.3.8.1/src/rmd_types.h.ALSA-default	2010-01-15 10:23:33.774880012 +0100
+++ recordmydesktop-0.3.8.1/src/rmd_types.h	2010-01-15 10:23:55.973840192 +0100
@@ -39,7 +39,7 @@
 #ifdef HAVE_LIBASOUND
     #include <alsa/asoundlib.h>
 
-    #define DEFAULT_AUDIO_DEVICE "hw:0,0"
+    #define DEFAULT_AUDIO_DEVICE "default"
 #else
     #include <sys/ioctl.h>
     #include <sys/soundcard.h>
