--- gtk+extra-2.1.2.orig/gtkextra/gtkcolorcombo.c	2004-09-16 20:48:05.000000000 +0300
+++ gtk+extra-2.1.2/gtkextra/gtkcolorcombo.c	2010-12-27 11:37:11.044000024 +0200
@@ -146,8 +146,8 @@
                                       GTK_CLASS_TYPE(object_class),
                                       GTK_SIGNAL_OFFSET(GtkColorComboClass,
                                       changed),
-                                      gtkextra_VOID__INT_BOXED,
-                                      GTK_TYPE_NONE,  
+                                      gtkextra_VOID__BOXED_BOXED,
+                                      GTK_TYPE_NONE,
                                       2, GTK_TYPE_INT, GDK_TYPE_COLOR);
 
   klass->changed = NULL;
