--- gtk+extra-2.1.2.orig/gtkextra/Makefile.am	2005-06-16 01:54:34.000000000 +0300
+++ gtk+extra-2.1.2/gtkextra/Makefile.am	2010-12-27 11:35:46.995000028 +0200
@@ -197,13 +197,13 @@
 $(libgtkextra_x11_2_0_la_OBJECTS) $(libgtkextra_linux_fb_2_0_la_OBJECTS) $(libgtkextra_win32_2_0_la_OBJECTS): ${gtkextra_built_public_sources} ${gtkextra_built_private_headers}
 
 $(srcdir)/gtkextra-marshal.h: gtkextra-marshal.list
-        ( @GLIB_GENMARSHAL@ --prefix=gtkextra gtkextra-marshal.list --header > gtkextra-marshal.tmp \
-        && mv gtkextra-marshal.tmp gtkextra-marshal.h ) \
-        || ( rm -f gtkextra-marshal.tmp && exit 1 )
+		( @GLIB_GENMARSHAL@ --prefix=gtkextra gtkextra-marshal.list --header > gtkextra-marshal.tmp \
+		&& mv gtkextra-marshal.tmp gtkextra-marshal.h ) \
+		|| ( rm -f gtkextra-marshal.tmp && exit 1 )
 $(srcdir)/gtkextra-marshal.c: gtkextra-marshal.h
-        ( @GLIB_GENMARSHAL@ --prefix=gtkextra gtkextra-marshal.list --body > gtkextra-marshal.tmp \
-        && mv gtkextra-marshal.tmp gtkextra-marshal.c ) \
-        || ( rm -f gtkextra-marshal.tmp && exit 1 )
+		( @GLIB_GENMARSHAL@ --prefix=gtkextra gtkextra-marshal.list --body > gtkextra-marshal.tmp \
+		&& mv gtkextra-marshal.tmp gtkextra-marshal.c ) \
+		|| ( rm -f gtkextra-marshal.tmp && exit 1 )
 
 
 $(srcdir)/gtkextratypebuiltins.h: $(gtkextra_public_h_sources)
@@ -281,7 +281,7 @@
 	done
 
 test: testgtksheet testgtkiconlist testgtkfilesel testgtkplot testgtkfont testrealtime testpolar testgtkplot3d testcontour testbubbles testflux testboxes testcandles testimage testpixmap testcharsel testdt testgtkplotdt testiterator
-        builddir=`pwd`; cd $(top_builddir); top_builddir=`pwd`; export HAVE_PROGRESSBAR=1; \
+	builddir=`pwd`; cd $(top_builddir); top_builddir=`pwd`; export HAVE_PROGRESSBAR=1; \
 	cd $$builddir; cd $(srcdir); \
 	$(SHELL) $$top_builddir/libtool --mode=execute gdb $$builddir/testgtksheet;  \
 	$(SHELL) $$top_builddir/libtool --mode=execute gdb $$builddir/testgtkplot;  \
