Index: src/CMakeLists.txt
===================================================================
--- src/CMakeLists.txt
+++ src/CMakeLists.txt	2010-03-17 16:48:22.000000000 +0100
@@ -29,6 +29,6 @@
 install(FILES ${qzion_HEADERS} DESTINATION include/)
 
 if (NOT WIN32)
-    configure_file(common/qzion.pc.cmake qzion.pc @ONLY)
-    install(FILES ${CMAKE_BINARY_DIR}/qzion.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig/)
+  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/common/qzion.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/qzion.pc @ONLY) 
+  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/qzion.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig/) 
 endif (NOT WIN32)
