Index: qzion-mainline/python/qzion/qzionobject.sip
===================================================================
--- qzion-mainline/python/qzion/qzionobject.sip
+++ qzion-mainline/python/qzion/qzionobject.sip	2009-05-26 08:33:48.000000000 -0400
@@ -10,7 +10,7 @@
 
 %ConvertToSubClassCode
     static struct classes {
-        char *name;
+        const char *name;
         sipWrapperType **type;
     } cls[] = {
         {sipName_QZionText, &sipClass_QZionText},
