diff -ur qzion-mainline/python/CMakeLists.txt qzion-mainline/python/CMakeLists.txt
--- qzion-mainline/python/CMakeLists.txt	2009-02-18 21:38:44.000000000 +0000
+++ qzion-mainline/python/CMakeLists.txt	2009-03-05 20:41:10.159038865 +0000
@@ -16,9 +16,6 @@
 set(SIP_TAGS ALL WS_X11 ${PYQT4_VERSION_TAG})
 set(SIP_EXTRA_OPTIONS -g)
 
-set(PYTHON_SITE_PACKAGES_DIR
-    ${CMAKE_INSTALL_PREFIX}/lib/python${PYTHON_SHORT_VERSION}/site-packages)
-
 # Putting inside the "concepts" package until we release.
 file(GLOB qzion_files_sip qzion/*.sip)
 set(SIP_EXTRA_FILES_DEPEND ${qzion_files_sip})
