diff -Nuar glfw-2.7.5.orig/lib/x11/Makefile.x11.in glfw-2.7.5/lib/x11/Makefile.x11.in
--- glfw-2.7.5.orig/lib/x11/Makefile.x11.in	2012-02-07 20:05:06.000000000 +0200
+++ glfw-2.7.5/lib/x11/Makefile.x11.in	2012-06-01 16:59:43.000000000 +0300
@@ -2,7 +2,7 @@
 ##########################################################################
 # Installation prefix (default to /usr/local)
 ##########################################################################
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 
 
 ##########################################################################
@@ -20,19 +20,20 @@
 INSTALL  = install
 ARFLAGS  = -rcs
 HEADERS     = ../../include/GL/glfw.h ../internal.h platform.h
-
+LN_S     = ln -s
 
 ##########################################################################
 # Install GLFW static library
 ##########################################################################
-install: libglfw.a libglfw.pc
-	$(INSTALL) -d $(PREFIX)/lib
-	$(INSTALL) -c -m 644 libglfw.a $(PREFIX)/lib/libglfw.a
-	$(INSTALL) -d $(PREFIX)/include/GL
-	$(INSTALL) -c -m 644 ../../include/GL/glfw.h $(PREFIX)/include/GL/glfw.h
-	$(INSTALL) -d $(PREFIX)/lib/pkgconfig
-	$(INSTALL) -c -m 644 libglfw.pc $(PREFIX)/lib/pkgconfig/libglfw.pc
-
+install: libglfw.pc
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/lib
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/include/GL
+	$(INSTALL) -c -m 755 ../../lib/x11/libglfw.so.0.0 $(DESTDIR)$(PREFIX)/lib
+	$(LN_S) $(PREFIX)/lib/libglfw.so.0.0 $(DESTDIR)$(PREFIX)/lib/libglfw.so.0
+	$(LN_S) $(PREFIX)/lib/libglfw.so.0.0 $(DESTDIR)$(PREFIX)/lib/libglfw.so
+	$(INSTALL) -c -m 644 ../../include/GL/glfw.h $(DESTDIR)$(PREFIX)/include/GL/glfw.h
+	$(INSTALL) -d $(DESTDIR)$(PREFIX)/lib/pkgconfig
+	$(INSTALL) -c -m 644 libglfw.pc $(DESTDIR)$(PREFIX)/lib/pkgconfig/libglfw.pc
 
 ##########################################################################
 # Install GLFW static and shared libraries
@@ -113,8 +114,9 @@
 # Rule for building shared library
 ##########################################################################
 libglfw.so: $(SHARED_OBJS)
-	$(CC) $(SOFLAGS) -o $@ $(SHARED_OBJS) $(LFLAGS) $(LIBS)
-
+	$(CC) $(SOFLAGS)-Wl,-soname=libglfw.so.0 -o libglfw.so.0.0 $(SHARED_OBJS) $(LFLAGS) $(LIBS)
+	$(LN_S) libglfw.so.0.0 libglfw.so.0
+	$(LN_S) libglfw.so.0.0 libglfw.so
 
 ##########################################################################
 # Rules for building static library object files
