Index: tepsonic-0.98.2/tepsonic.desktop
===================================================================
--- tepsonic-0.98.2.orig/tepsonic.desktop
+++ tepsonic-0.98.2/tepsonic.desktop
@@ -1,11 +1,14 @@
 [Desktop Entry]
 Name=TepSonic
+Name[tr]=TepSonic
 Comment=A simple lightweight audioplayer
+Comment[tr]=Basit, hafif bir müzikçalar
 Comment[cs]=Malý jednoduchý hudební přehrávač
 GenericName=A simple audioplayer
+GenericName[tr]=Basit bir müzikçalar
 Exec=tepsonic
 Terminal=false
 Type=Application
 Icon=tepsonic
-Categories=Application;Qt;Audio
+Categories=Application;Qt;AudioVideo;Audio;Player;
 StartupNotify=true
