diff -Nuar medit-1.0.3.orig/moo/medit-app/medit.desktop.in medit-1.0.3/moo/medit-app/medit.desktop.in
--- medit-1.0.3.orig/moo/medit-app/medit.desktop.in	2011-06-20 09:45:47.771000116 +0300
+++ medit-1.0.3/moo/medit-app/medit.desktop.in	2011-06-20 09:47:16.536000117 +0300
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Name=medit
 _Comment=Text editor
+Comment[tr]=Metin Düzenleyici
 Exec=medit %F
 Terminal=false
 Type=Application
