diff -Nuar medit-1.0.3.orig/configure.ac medit-1.0.3/configure.ac
--- medit-1.0.3.orig/configure.ac	2011-06-20 09:45:47.736000116 +0300
+++ medit-1.0.3/configure.ac	2011-06-20 10:04:24.646000116 +0300
@@ -37,7 +37,7 @@
 AC_SUBST(MOO_COPYRIGHT,"_moo_copyright_")
 
 # keep in sync with po/maintain
-AC_SUBST(MOO_PACKAGE_NAME,"medit-1")
+AC_SUBST(MOO_PACKAGE_NAME,"medit")
 
 MOO_PREFS_XML_FILE_NAME="prefs.xml"
 MOO_STATE_XML_FILE_NAME="state.xml"
