Index: gnome-vfs-2.24.1/doc/Makefile.am
===================================================================
--- gnome-vfs-2.24.1.orig/doc/Makefile.am
+++ gnome-vfs-2.24.1/doc/Makefile.am
@@ -78,18 +78,6 @@ content_files = 		\
 
 extra_files = version.xml.in
 
-# CFLAGS and LDFLAGS for compiling scan program. Only needed
-# if $(DOC_MODULE).types is non-empty.
-GTKDOC_CFLAGS = \
-        -I$(top_srcdir)         \
-	$(VFS_CFLAGS)		\
-	$(LIBGNOMEVFS_CFLAGS)
-
-GTKDOC_LIBS = $(LIBGNOMEVFS_LIBS) \
-	$(top_builddir)/libgnomevfs/libgnomevfs-2.la
-
-include $(top_srcdir)/gtk-doc.make
-
 dist-hook-local:
 #	mkdir $(distdir)/TEXT;          \
 #	for f in $(srcdir)/TEXT/* ; do  \
Index: gnome-vfs-2.24.1/configure.in
===================================================================
--- gnome-vfs-2.24.1.orig/configure.in
+++ gnome-vfs-2.24.1/configure.in
@@ -1237,17 +1237,6 @@ if test "x$enable_selinux" != "xno"; the
 fi
 AC_SUBST(SELINUX_LIBS)
 
-dnl **************************
-dnl *** Checks for gtk-doc ***
-dnl **************************
-
-GTK_DOC_CHECK([1.0])
-
-dnl ******************************
-dnl *** End checks for gtk-doc ***
-dnl ******************************
-
-
 dnl borrowed from gnu findutils
 dnl and relicensed as LGPL by the FSF
 
