Index: mixxx-1.9.0~release-1.9.x~bzr2720/src/SConscript
===================================================================
--- mixxx-1.9.0~release-1.9.x~bzr2720.orig/src/SConscript
+++ mixxx-1.9.0~release-1.9.x~bzr2720/src/SConscript
@@ -32,8 +32,8 @@ from build import util
 #
 
 mixxx_version = util.get_mixxx_version()
-branch_name = util.get_bzr_branch_name()
-bazaar_revision = util.get_bzr_revision()
+#branch_name = util.get_bzr_branch_name()
+#bazaar_revision = util.get_bzr_revision()
 print "WE ARE IN:", os.getcwd()
 
 plugins = []
Index: mixxx-1.9.0~release-1.9.x~bzr2720/build/util.py
===================================================================
--- mixxx-1.9.0~release-1.9.x~bzr2720.orig/build/util.py
+++ mixxx-1.9.0~release-1.9.x~bzr2720/build/util.py
@@ -6,6 +6,7 @@ def get_bzr_revision():
     return os.popen("bzr revno").readline().strip()
 
 def get_bzr_branch_name():
+    return ""
     output_lines = os.popen("bzr info").read().splitlines()
 
     matcher = re.compile(
