--- share/applications/shutter.desktop.orig	2012-08-15 18:48:54.000000000 +0300
+++ share/applications/shutter.desktop	2012-08-29 08:23:44.000000000 +0300
@@ -4,16 +4,18 @@
 Name[de_DE]=Shutter
 Name[pt_BR]=Shutter
 GenericName=Screenshot Tool
+GenericName[tr]=Ekran Görüntüsü Yakalama Aracı
 GenericName[de_DE]=Anwendung für Bildschirmfotos
 GenericName[pt_BR]=Captura de tela
 Comment=Capture, edit and share screenshots
+Comment[tr]=Ekran görüntüleri yakalatın, üzerinde düzeltmeler yapın ve paylaşın
 Comment[de_DE]=Bildschirmfotos aufnehmen, bearbeiten und mit Anderen teilen
 Comment[pt_BR]=Aplicativo avançado para capturar imagens da tela
 Exec=shutter %F
-Icon=shutter
+Icon=/usr/share/icons/hicolor/32x32/apps/shutter.png
 Terminal=false
 Type=Application
-Categories=Utility;
+Categories=Utility;Application;Graphics;
 MimeType=image/bmp;image/jpeg;image/gif;image/png;image/tiff;image/x-bmp;image/x-ico;image/x-png;image/x-pcx;image/x-tga;image/xpm;image/svg+xml;
 X-Ayatana-Desktop-Shortcuts=Select;Screen;Window;Active
 
