--- a/src/addins/MonoDevelop.Gettext/Makefile.in.orig	2012-07-19 16:34:55.325000114 +0200
+++ b/src/addins/MonoDevelop.Gettext/Makefile.in	2012-07-19 16:35:12.027000115 +0200
@@ -274,7 +274,7 @@
 	templates/Makefile.template \
 	templates/TranslationProject.xpt.xml
 
-assemblydir = $(MD_ADDIN_DIR)/DisplayBindings/MonoDevelop.Gettext
+assemblydir = $(MD_ADDIN_DIR)/DisplayBindings/Gettext
 assembly_DATA = $(ASSEMBLY) $(ASSEMBLY).mdb
 CLEANFILES = $(ASSEMBLY) $(ASSEMBLY).mdb
 EXTRA_DIST = $(FILES) $(RES)
