diff -p -up monodevelop-1.9.2/monodevelop.in.libxul monodevelop-1.9.2/monodevelop.in
--- monodevelop-1.9.2/monodevelop.in.libxul	2009-02-04 22:51:09.000000000 +0100
+++ monodevelop-1.9.2/monodevelop.in	2009-02-12 17:41:43.000000000 +0100
@@ -11,7 +11,7 @@ MD_REDIRECT_LOG="$HOME/.config/MonoDevel
 #############################
 
 #(NOTE: any changes to this block should be kept in sync with the one in configure.in)
-MOZFILE="libgtkembedmoz.so"
+MOZFILE="libxul.so"
 find_libgtkembedmoz ()
 {
 	MOZDIR=$MOZILLA_FIVE_HOME
diff -p -up monodevelop-1.9.2/configure.in~ monodevelop-1.9.2/configure.in
--- monodevelop-1.9.2/configure.in~	2009-02-09 22:35:53.000000000 +0100
+++ monodevelop-1.9.2/configure.in	2009-02-12 17:44:41.000000000 +0100
@@ -228,7 +228,7 @@ AM_CONDITIONAL(ENABLE_C, [test x$enable_
 # detection code in the MonoDevelop core so that MD can set LD_LIBRARY_PATH 
 # MOZILLA_FIVE_HOME before starting.
 ###############################################################################
-MOZFILE="libgtkembedmoz.so"
+MOZFILE="libxul.so"
 
 #Check pkg-config first
 MOZILLA_HOME="`$PKG_CONFIG --variable=libdir mozilla-gtkmozembed`"
