diff -ur a/external/maccore/src/Foundation/NSHttpCookie.cs b/external/maccore/src/Foundation/NSHttpCookie.cs
--- a/external/maccore/src/Foundation/NSHttpCookie.cs	2012-06-17 22:22:27.000000000 +0300
+++ b/external/maccore/src/Foundation/NSHttpCookie.cs	2013-05-07 04:10:27.257487356 +0300
@@ -108,12 +108,6 @@
 					properties.Add (NSHttpCookie.KeySecure, new NSString ("1"));
 				if (version.HasValue)
 					properties.Add (NSHttpCookie.KeyVersion, new NSString (version.Value.ToString ()));
-				
-				if (IsDirectBinding) {
-					Handle = Messaging.IntPtr_objc_msgSend_IntPtr (this.Handle, selInitWithProperties_, properties.Handle);
-				} else {
-					Handle = Messaging.IntPtr_objc_msgSendSuper_IntPtr (this.SuperHandle, selInitWithProperties_, properties.Handle);
-				}
 			}
 		}
 	}
