diff -up nut-2.6.4/scripts/systemd/nut-monitor.service.in.tmpfiles nut-2.6.4/scripts/systemd/nut-monitor.service.in
--- nut-2.6.4/scripts/systemd/nut-monitor.service.in.tmpfiles	2012-05-02 23:35:36.000000000 +0200
+++ nut-2.6.4/scripts/systemd/nut-monitor.service.in	2012-06-14 11:38:42.408504102 +0200
@@ -3,6 +3,7 @@ Description=Network UPS Tools - power de
 After=local-fs.target network.target nut-server.service
 
 [Service]
+ExecStartPre=-/usr/bin/systemd-tmpfiles --create /etc/tmpfiles.d/nut-client.conf
 ExecStart=@SBINDIR@/upsmon
 PIDFile=@PIDPATH@/upsmon.pid
 Type=forking

