diff -Nur nut-2.4.1-old/scripts/udev/nut-usbups.rules.in nut-2.4.1/scripts/udev/nut-usbups.rules.in
--- nut-2.4.1-old/scripts/udev/nut-usbups.rules.in	2010-01-28 14:04:19.834214165 +0200
+++ nut-2.4.1/scripts/udev/nut-usbups.rules.in	2010-01-28 14:05:08.137214880 +0200
@@ -89,4 +89,10 @@
 #  Ablerex 625L USB  - blazer_usb
 SYSFS{idVendor}=="ffff", SYSFS{idProduct}=="0000", MODE="664", GROUP="@RUN_AS_GROUP@"
 
+
+# Powercom - unknown driver
+SYSFS{idVendor}=="0d9f", SYSFS{idProduct}=="0001", MODE="664", GROUP="@RUN_AS_GROUP@"
+SYSFS{idVendor}=="0d9f", SYSFS{idProduct}=="0002", MODE="664", GROUP="@RUN_AS_GROUP@"
+
+
 LABEL="nut-usbups_rules_end"
