diff -Nur supertuxkart-0.7.1-old//configure.ac supertuxkart-0.7.1/configure.ac
--- supertuxkart-0.7.1-old//configure.ac	2011-04-16 16:53:38.744000125 +0300
+++ supertuxkart-0.7.1/configure.ac	2011-04-16 16:54:49.166000145 +0300
@@ -107,6 +107,7 @@
    if test "x$ac_cv_search_glNewList" = "x-lGLcore"; then
        # if GLcore found, then also check for GL
        AC_SEARCH_LIBS(glXCreateContext, GL)
+       AC_SEARCH_LIBS(gluLookAt, GLU)
    fi
    ;;
 esac
@@ -402,7 +403,6 @@
   enet_LIBS="-Lenet -lenet"
   ;;
 esac
-ENETTREE="src/enet"
 
 # =================
 # Disable/enable addons support
@@ -462,7 +462,6 @@
  src/ide/Makefile                   \
  src/bullet/Makefile                \
  src/bullet/src/Makefile            \
- src/enet/Makefile                  \
 ])
 AC_OUTPUT
 echo -e $SUMMARY
diff -Nur supertuxkart-0.7.1-old//data/challenges/Makefile.am supertuxkart-0.7.1/data/challenges/Makefile.am
--- supertuxkart-0.7.1-old//data/challenges/Makefile.am	2011-04-16 16:53:38.643000125 +0300
+++ supertuxkart-0.7.1/data/challenges/Makefile.am	2011-04-16 16:58:44.857999947 +0300
@@ -1,5 +1,5 @@
 # challenges/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/challenges
+pkgdatadir = $(datadir)/$(PACKAGE)/data/challenges
 
 nobase_dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.challenge") 
diff -Nur supertuxkart-0.7.1-old//data/fonts/Makefile.am supertuxkart-0.7.1/data/fonts/Makefile.am
--- supertuxkart-0.7.1-old//data/fonts/Makefile.am	2011-04-16 16:53:38.648000125 +0300
+++ supertuxkart-0.7.1/data/fonts/Makefile.am	2011-04-16 16:54:04.933000015 +0300
@@ -1,6 +1,6 @@
 # data/fonts/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/fonts
+pkgdatadir = $(datadir)/$(PACKAGE)/data/fonts
 
 dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.png") \
                      $(shell find $(srcdir) -name "*.xml")
diff -Nur supertuxkart-0.7.1-old//data/gfx/Makefile.am supertuxkart-0.7.1/data/gfx/Makefile.am
--- supertuxkart-0.7.1-old//data/gfx/Makefile.am	2011-04-16 16:53:38.497000125 +0300
+++ supertuxkart-0.7.1/data/gfx/Makefile.am	2011-04-16 16:58:33.794999858 +0300
@@ -1,5 +1,5 @@
 #data/gfx
 
-pkgdatadir   = $(datadir)/games/$(PACKAGE)/data/gfx
+pkgdatadir   = $(datadir)/$(PACKAGE)/data/gfx
 dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.xml")
 
diff -Nur supertuxkart-0.7.1-old//data/grandprix/Makefile.am supertuxkart-0.7.1/data/grandprix/Makefile.am
--- supertuxkart-0.7.1-old//data/grandprix/Makefile.am	2011-04-16 16:53:38.703000125 +0300
+++ supertuxkart-0.7.1/data/grandprix/Makefile.am	2011-04-16 16:54:04.934000032 +0300
@@ -1,6 +1,6 @@
 # data/music
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/grandprix
+pkgdatadir = $(datadir)/$(PACKAGE)/data/grandprix
 
 dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.grandprix")  \
                     $(shell find $(srcdir) -name "*.challenge")  \
diff -Nur supertuxkart-0.7.1-old//data/gui/Makefile.am supertuxkart-0.7.1/data/gui/Makefile.am
--- supertuxkart-0.7.1-old//data/gui/Makefile.am	2011-04-16 16:53:38.507000125 +0300
+++ supertuxkart-0.7.1/data/gui/Makefile.am	2011-04-16 16:54:04.934000032 +0300
@@ -1,6 +1,6 @@
 # data/gui/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/gui
+pkgdatadir = $(datadir)/$(PACKAGE)/data/gui
 
 nobase_dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.png"      ) \
                             $(shell find $(srcdir) -name "*.jpg"      ) \
diff -Nur supertuxkart-0.7.1-old//data/karts/Makefile.am supertuxkart-0.7.1/data/karts/Makefile.am
--- supertuxkart-0.7.1-old//data/karts/Makefile.am	2011-04-16 16:53:38.665000125 +0300
+++ supertuxkart-0.7.1/data/karts/Makefile.am	2011-04-16 16:54:04.934000032 +0300
@@ -1,6 +1,6 @@
 # data/karts
 
-pkgdatadir= $(datadir)/games/$(PACKAGE)/data/karts
+pkgdatadir= $(datadir)/$(PACKAGE)/data/karts
 
 nobase_dist_pkgdata_DATA = $(shell find $(srcdir) -name "materials.xml") \
                            $(shell find $(srcdir) -name "*.png")         \
diff -Nur supertuxkart-0.7.1-old//data/Makefile.am supertuxkart-0.7.1/data/Makefile.am
--- supertuxkart-0.7.1-old//data/Makefile.am	2011-04-16 16:53:38.647000125 +0300
+++ supertuxkart-0.7.1/data/Makefile.am	2011-04-16 16:55:16.688000119 +0300
@@ -1,7 +1,7 @@
 # data/
 
 SUBDIRS    = challenges fonts gfx gui karts models music po sfx textures tracks grandprix
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data
+pkgdatadir = $(datadir)/$(PACKAGE)/data
 
 dist_pkgdata_DATA = \
  $(shell find $(srcdir) -name "*.data")                  \
@@ -13,16 +13,7 @@
  stk_config.xml  powerup.xml  items.xml                  \
  CREDITS
 
-desktopdir = $(prefix)/share/applications
-desktop_DATA = supertuxkart.desktop
 
-icondir = $(prefix)/share/pixmaps
-dist_icon_DATA = supertuxkart_32.xpm supertuxkart_64.xpm
 
 
-# The desktop file needs the absolute path to the binary
 # since e.g. /usr/games might not be in the standard path
-EXTRA_DIST = supertuxkart_desktop.template
-CLEANFILES = supertuxkart.desktop
-supertuxkart.desktop: Makefile supertuxkart_desktop.template
-	sed 's#PREFIX#$(prefix)#' $(srcdir)/supertuxkart_desktop.template >supertuxkart.desktop
diff -Nur supertuxkart-0.7.1-old//data/models/Makefile.am supertuxkart-0.7.1/data/models/Makefile.am
--- supertuxkart-0.7.1-old//data/models/Makefile.am	2011-04-16 16:53:38.639000125 +0300
+++ supertuxkart-0.7.1/data/models/Makefile.am	2011-04-16 16:54:04.934000032 +0300
@@ -1,6 +1,6 @@
 # data/models/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/models
+pkgdatadir = $(datadir)/$(PACKAGE)/data/models
 
 dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.b3d") \
                      $(shell find $(srcdir) -name "*.png") \
diff -Nur supertuxkart-0.7.1-old//data/music/Makefile.am supertuxkart-0.7.1/data/music/Makefile.am
--- supertuxkart-0.7.1-old//data/music/Makefile.am	2011-04-16 16:53:38.474000125 +0300
+++ supertuxkart-0.7.1/data/music/Makefile.am	2011-04-16 16:54:04.935000047 +0300
@@ -1,6 +1,6 @@
 # data/music
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/music
+pkgdatadir = $(datadir)/$(PACKAGE)/data/music
 
 dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.ogg")  \
                     $(shell find $(srcdir) -name "*.music") \
diff -Nur supertuxkart-0.7.1-old//data/po/Makefile.am supertuxkart-0.7.1/data/po/Makefile.am
--- supertuxkart-0.7.1-old//data/po/Makefile.am	2011-04-16 16:53:38.645000125 +0300
+++ supertuxkart-0.7.1/data/po/Makefile.am	2011-04-16 16:54:04.935000047 +0300
@@ -1,6 +1,6 @@
 # data/po/
 
-pkgdatadir = $(datadir)/games/@PACKAGE@/data/po
+pkgdatadir = $(datadir)/@PACKAGE@/data/po
 
 dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.po") \
                     supertuxkart.pot
diff -Nur supertuxkart-0.7.1-old//data/sfx/Makefile.am supertuxkart-0.7.1/data/sfx/Makefile.am
--- supertuxkart-0.7.1-old//data/sfx/Makefile.am	2011-04-16 16:53:38.679000125 +0300
+++ supertuxkart-0.7.1/data/sfx/Makefile.am	2011-04-16 16:54:04.935000047 +0300
@@ -1,6 +1,6 @@
 # sfx/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/sfx
+pkgdatadir = $(datadir)/$(PACKAGE)/data/sfx
 
 nobase_dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.ogg")      \
                             $(shell find $(srcdir) -name "sfx.xml")    \
diff -Nur supertuxkart-0.7.1-old//data/textures/Makefile.am supertuxkart-0.7.1/data/textures/Makefile.am
--- supertuxkart-0.7.1-old//data/textures/Makefile.am	2011-04-16 16:53:38.692000125 +0300
+++ supertuxkart-0.7.1/data/textures/Makefile.am	2011-04-16 16:54:04.936000054 +0300
@@ -1,6 +1,6 @@
 # data/textures
 
-pkgdatadir   = $(datadir)/games/$(PACKAGE)/data/textures
+pkgdatadir   = $(datadir)/$(PACKAGE)/data/textures
 dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.png") \
                     $(shell find $(srcdir) -name "*.jpg") \
                     materials.xml License.txt
diff -Nur supertuxkart-0.7.1-old//data/tracks/Makefile.am supertuxkart-0.7.1/data/tracks/Makefile.am
--- supertuxkart-0.7.1-old//data/tracks/Makefile.am	2011-04-16 16:53:38.575000125 +0300
+++ supertuxkart-0.7.1/data/tracks/Makefile.am	2011-04-16 16:54:04.936000054 +0300
@@ -1,6 +1,6 @@
 # data/tracks
 
-pkgdatadir= $(datadir)/games/$(PACKAGE)/data/tracks
+pkgdatadir= $(datadir)/$(PACKAGE)/data/tracks
 nobase_dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.jpg")           \
                            $(shell find $(srcdir) -name "*.JPG")           \
                            $(shell find $(srcdir) -name "*.png")           \
diff -Nur supertuxkart-0.7.1-old//Makefile.am supertuxkart-0.7.1/Makefile.am
--- supertuxkart-0.7.1-old//Makefile.am	2011-04-16 16:53:38.744000125 +0300
+++ supertuxkart-0.7.1/Makefile.am	2011-04-16 16:54:04.931999996 +0300
@@ -1,8 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = $(BULLETTREE) $(ENETTREE) src doc data
+SUBDIRS = $(BULLETTREE) $(ENETTREE) src data
 
-pkgdatadir=$(datadir)/doc/$(PACKAGE)
-dist_pkgdata_DATA=AUTHORS ChangeLog COPYING README
 # NEWS only contains "see changelog"
 # CHANGES only contains "see NEW"
diff -Nur supertuxkart-0.7.1-old//src/Makefile.am supertuxkart-0.7.1/src/Makefile.am
--- supertuxkart-0.7.1-old//src/Makefile.am	2011-04-16 16:53:38.734000125 +0300
+++ supertuxkart-0.7.1/src/Makefile.am	2011-04-16 16:54:04.936000054 +0300
@@ -2,10 +2,9 @@
 
 SUBDIRS = ide
 
-bindir=$(prefix)/games
 bin_PROGRAMS = supertuxkart
 
-AM_CPPFLAGS = -DSUPERTUXKART_DATADIR="\"$(datadir)/games/$(PACKAGE)/\""  \
+AM_CPPFLAGS = -DSUPERTUXKART_DATADIR="\"$(datadir)/$(PACKAGE)/\""  \
  -I$(srcdir)/bullet/src/ -I$(srcdir)/enet/include/
 
 supertuxkart_SOURCES = \
