diff -Nuar a/data/supertuxkart_desktop.template b/data/supertuxkart_desktop.template
--- a/data/supertuxkart_desktop.template	2012-03-06 02:18:51.000000000 +0200
+++ b/data/supertuxkart_desktop.template	2013-08-31 19:58:06.850860804 +0300
@@ -1,15 +1,16 @@
 [Desktop Entry]
+Type=Application
 Encoding=UTF-8
 Name=SuperTuxKart
-Icon=@PREFIX@/share/pixmaps/supertuxkart_64.xpm
+Icon=supertuxkart
 GenericName=A kart racing game
-GenericName[de_DE]=Ein Kart-Rennspiel
-GenericName[fr_FR]=Un jeu de karting
+GenericName[de]=Ein Kart-Rennspiel
+GenericName[fr]=Un jeu de karting
 GenericName[gl]=Xogo de carreiras con karts
-GenericName[ro_RO]=Un joc de curse cu carturi
-Exec=@PREFIX@/games/supertuxkart --log=file
+GenericName[ro]=Un joc de curse cu carturi
+GenericName[tr]=Tux Gokart Yarışı
+Exec=supertuxkart --log=file
 Terminal=false
 StartupNotify=false
 Type=Application
-TryExec=@PREFIX@/games/supertuxkart
-Categories=Game;ArcadeGame;
+Categories=Application;Game;ArcadeGame;
