diff -Nur supertuxkart-0.7-old//src/network/network_manager.cpp supertuxkart-0.7/src/network/network_manager.cpp
--- supertuxkart-0.7-old//src/network/network_manager.cpp	2011-03-11 13:59:16.314001414 +0200
+++ supertuxkart-0.7/src/network/network_manager.cpp	2011-03-11 14:12:47.665001432 +0200
@@ -83,7 +83,8 @@
      m_host = enet_host_create (& address     /* the address to bind the server host to */, 
                                 stk_config->m_max_karts /* number of connections */,
                                 0             /* incoming bandwidth */,
-                                0             /* outgoing bandwidth */     );
+                                0             /* outgoing bandwidth */,
+                                0             /* channel limit */     );
     if (m_host == NULL)
     {
         fprintf (stderr, 
@@ -108,7 +109,8 @@
     m_host = enet_host_create (NULL /* create a client host */,
                                1    /* only allow 1 outgoing connection */,
                                0    /* downstream bandwidth unlimited   */,
-                               0    /*  upstream bandwidth unlimited    */ );
+                               0    /*  upstream bandwidth unlimited    */,
+                               0    /* channel limit */     );
     
     if (m_host == NULL)
     {
