diff -Nur supertuxkart-0.7.1-old//configure.ac supertuxkart-0.7.1/configure.ac
--- supertuxkart-0.7.1-old//configure.ac	2011-04-16 17:04:09.634999488 +0300
+++ supertuxkart-0.7.1/configure.ac	2011-04-16 17:15:52.404000066 +0300
@@ -233,7 +233,7 @@
 
 case "${host}" in
 *-*-linux* )
-  irrlicht_LIBS="-L/$with_irrlicht/lib/Linux -L/$with_irrlicht/lib -lIrrlicht"
+  irrlicht_LIBS="-L/$with_irrlicht/lib/Linux -L/$with_irrlicht/lib -lIrrlicht -lIrrXML"
   ;;
 *darwin*|*macosx*)
   irrlicht_LIBS="-L/$with_irrlicht/lib/ -lIrrlicht -framework Cocoa"
