Index: linux-atm-2.5.1/src/config/Makefile.in
===================================================================
--- linux-atm-2.5.1.orig/src/config/Makefile.in
+++ linux-atm-2.5.1/src/config/Makefile.in
@@ -413,10 +413,10 @@ uninstall-am: uninstall-local uninstall-
 
 
 install-exec-local:
-	-cp hosts.atm /etc
+	-cp hosts.atm $(DESTDIR)$(sysconfdir)
 
 uninstall-local:
-	-rm /etc/hosts.atm
+	-rm $(DESTDIR)$(sysconfdir)/hosts.atm
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
