From b11f0b72976ad3c757539291a22edf4750dac095 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <cguthrie@mandriva.org>
Date: Tue, 20 Apr 2010 19:46:30 +0100
Subject: [PATCH 101/105] streamwidget: Fix a compile warning.

---
 src/streamwidget.cc |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/streamwidget.cc b/src/streamwidget.cc
index 12c7d6d..c4ad15f 100644
--- a/src/streamwidget.cc
+++ b/src/streamwidget.cc
@@ -31,8 +31,8 @@
 /*** StreamWidget ***/
 StreamWidget::StreamWidget(BaseObjectType* cobject, const Glib::RefPtr<Gnome::Glade::Xml>& x) :
     MinimalStreamWidget(cobject, x),
-    mpMainWindow(NULL),
-    peak(NULL) {
+    peak(NULL),
+    mpMainWindow(NULL) {
 
     x->get_widget("lockToggleButton", lockToggleButton);
     x->get_widget("muteToggleButton", muteToggleButton);
-- 
1.7.0.5

