From 9fe20201f2a462cb51680cb939169c020d9bbdc3 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <cguthrie@mandriva.org>
Date: Tue, 20 Apr 2010 22:03:45 +0100
Subject: [PATCH 106/106] main: Cleanup labels after connection rework

---
 src/pavucontrol.cc |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/pavucontrol.cc b/src/pavucontrol.cc
index 0ad020d..b7bfa35 100644
--- a/src/pavucontrol.cc
+++ b/src/pavucontrol.cc
@@ -568,7 +568,7 @@ int main(int argc, char *argv[]) {
     Gtk::Main::run(*mainWindow);
     delete mainWindow;
 
-finish:
-    pa_context_unref(context);
+    if (context)
+        pa_context_unref(context);
     pa_glib_mainloop_free(m);
 }
-- 
1.7.0.5

