From 95e48d5c9e7d3fa3aa24d6e3b8326145712e33de Mon Sep 17 00:00:00 2001
From: Colin Guthrie <cguthrie@mandriva.org>
Date: Wed, 21 Apr 2010 00:44:01 +0100
Subject: [PATCH 107/110] mainwindow: Compact iterator decls

---
 src/mainwindow.cc |   37 ++++++++++++-------------------------
 1 files changed, 12 insertions(+), 25 deletions(-)

diff --git a/src/mainwindow.cc b/src/mainwindow.cc
index 25456a7..c84ca7f 100644
--- a/src/mainwindow.cc
+++ b/src/mainwindow.cc
@@ -917,31 +917,18 @@ void MainWindow::removeClient(uint32_t index) {
 }
 
 void MainWindow::removeAllWidgets() {
-    {
-        std::map<uint32_t, SinkInputWidget*>::iterator it;
-        for (it = sinkInputWidgets.begin(); it != sinkInputWidgets.end(); ++it)
-            removeSinkInput(it->first);
-    }{
-        std::map<uint32_t, SourceOutputWidget*>::iterator it;
-        for (it = sourceOutputWidgets.begin(); it != sourceOutputWidgets.end(); ++it)
-            removeSourceOutput(it->first);
-    }{
-        std::map<uint32_t, SinkWidget*>::iterator it;
-        for (it = sinkWidgets.begin(); it != sinkWidgets.end(); ++it)
-            removeSink(it->first);
-    }{
-        std::map<uint32_t, SourceWidget*>::iterator it;
-        for (it = sourceWidgets.begin(); it != sourceWidgets.end(); ++it)
-            removeSource(it->first);
-    }{
-        std::map<uint32_t, CardWidget*>::iterator it;
-        for (it = cardWidgets.begin(); it != cardWidgets.end(); ++it)
-           removeCard(it->first);
-    }{
-        std::map<uint32_t, char*>::iterator it;
-        for (it = clientNames.begin(); it != clientNames.end(); ++it)
-            removeSourceOutput(it->first);
-    }
+    for (std::map<uint32_t, SinkInputWidget*>::iterator it = sinkInputWidgets.begin(); it != sinkInputWidgets.end(); ++it)
+        removeSinkInput(it->first);
+    for (std::map<uint32_t, SourceOutputWidget*>::iterator it = sourceOutputWidgets.begin(); it != sourceOutputWidgets.end(); ++it)
+        removeSourceOutput(it->first);
+    for (std::map<uint32_t, SinkWidget*>::iterator it = sinkWidgets.begin(); it != sinkWidgets.end(); ++it)
+        removeSink(it->first);
+    for (std::map<uint32_t, SourceWidget*>::iterator it = sourceWidgets.begin(); it != sourceWidgets.end(); ++it)
+        removeSource(it->first);
+    for (std::map<uint32_t, CardWidget*>::iterator it = cardWidgets.begin(); it != cardWidgets.end(); ++it)
+       removeCard(it->first);
+    for (std::map<uint32_t, char*>::iterator it = clientNames.begin(); it != clientNames.end(); ++it)
+        removeSourceOutput(it->first);
     deleteEventRoleWidget();
 }
 
-- 
1.7.0.5

