From 870c3ddb2e5fbe78a16621c61952a31874864e4e Mon Sep 17 00:00:00 2001
From: Colin Guthrie <cguthrie@mandriva.org>
Date: Thu, 22 Apr 2010 00:08:39 +0100
Subject: [PATCH 109/110] mainwindow: Fix clearing out of clients

---
 src/mainwindow.cc |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/mainwindow.cc b/src/mainwindow.cc
index c7008be..4995b53 100644
--- a/src/mainwindow.cc
+++ b/src/mainwindow.cc
@@ -983,7 +983,7 @@ void MainWindow::removeAllWidgets() {
     for (std::map<uint32_t, CardWidget*>::iterator it = cardWidgets.begin(); it != cardWidgets.end(); ++it)
        removeCard(it->first);
     for (std::map<uint32_t, char*>::iterator it = clientNames.begin(); it != clientNames.end(); ++it)
-        removeSourceOutput(it->first);
+        removeClient(it->first);
     deleteEventRoleWidget();
 }
 
-- 
1.7.0.5

