From f70ff92ecd9062522a09c8d31d0e25043b177dc5 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <cguthrie@mandriva.org>
Date: Thu, 22 Apr 2010 02:04:19 +0100
Subject: [PATCH 111/111] main: Format string fixes

---
 src/pavucontrol.cc |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/pavucontrol.cc b/src/pavucontrol.cc
index 714844a..aa71b3e 100644
--- a/src/pavucontrol.cc
+++ b/src/pavucontrol.cc
@@ -595,7 +595,7 @@ int main(int argc, char *argv[]) {
             pa_context_unref(context);
         pa_glib_mainloop_free(m);
     } catch ( const Glib::OptionError & e ) {
-        fprintf(stderr, options.get_help().c_str());
+        fprintf(stderr, "%s", options.get_help().c_str());
         return 1;
     }
 
-- 
1.7.0.5

