Index: pavucontrol-0.9.10/src/pavucontrol.glade
===================================================================
--- pavucontrol-0.9.10.orig/src/pavucontrol.glade
+++ pavucontrol-0.9.10/src/pavucontrol.glade
@@ -608,7 +608,7 @@ Monitors</property>
                             <child>
                               <widget class="GtkImage" id="image18">
                                 <property name="visible">True</property>
-                                <property name="icon_name">stock_lock</property>
+                                <property name="icon_name">object-locked</property>
                                 <property name="icon-size">1</property>
                               </widget>
                             </child>
@@ -765,7 +765,7 @@ Monitors</property>
                             <child>
                               <widget class="GtkImage" id="image18">
                                 <property name="visible">True</property>
-                                <property name="icon_name">stock_lock</property>
+                                <property name="icon_name">object-locked</property>
                                 <property name="icon-size">1</property>
                               </widget>
                             </child>
@@ -785,7 +785,7 @@ Monitors</property>
                             <child>
                               <widget class="GtkImage" id="image2">
                                 <property name="visible">True</property>
-                                <property name="icon_name">emblem-default</property>
+                                <property name="icon_name">go-bottom</property>
                               </widget>
                             </child>
                           </widget>
