diff --git a/dlls/mshtml/mshtml.inf b/dlls/mshtml/mshtml.inf
index 8186c2d..d24cfaf 100644
--- a/dlls/mshtml/mshtml.inf
+++ b/dlls/mshtml/mshtml.inf
@@ -211,6 +211,12 @@ HKCR,".mpe","Content Type",,"video/mpeg"
 HKCR,".mpeg","Content Type",,"video/mpeg"
 HKCR,".mpg","Content Type",,"video/mpeg"
 
+;; PDF
+HKCR,"MIME\Database\Content Type\application/pdf","Extension",,".pdf"
+HKCR,".pdf",,2,"pdffile"
+HKCR,".pdf","Content Type",,"application/pdf"
+HKCR,"pdffile\shell\open\command",,,"winebrowser ""%%1"""
+
 ;; PJPG
 HKCR,"MIME\Database\Content Type\image/pjpeg","CLSID",,"%CLSID_HTMLDocument%"
 HKCR,"MIME\Database\Content Type\image/pjpeg","Extension",,".jpg"
