Index: wine-1.3.13/configure.ac
===================================================================
--- wine-1.3.13.orig/configure.ac
+++ wine-1.3.13/configure.ac
@@ -204,9 +204,9 @@ then
     enable_fonts=${enable_fonts:-no}
     enable_server=${enable_server:-no}
     enable_tools=${enable_tools:-no}
-elif test "x$enable_win64" = "xyes"
+elif test "x$enable_win64" = "xno"
 then
-    test "x$libdir" != "x\${exec_prefix}/lib" || libdir="\${exec_prefix}/lib64"
+    test "x$libdir" != "x\${exec_prefix}/lib" || libdir="\${exec_prefix}/lib32"
 fi
 
 AC_CACHE_CHECK([for the directory containing the Wine tools], wine_cv_toolsdir,
@@ -1299,7 +1299,7 @@ then
         for i in `$gphoto2_devel --libs` `$gphoto2port_devel --libs`
         do
             case "$i" in
-            -L/usr/lib|-L/usr/lib64) ;;
+            -L/usr/lib|-L/usr/lib32) ;;
             -L*|-l*) ac_gphoto2_libs="$ac_gphoto2_libs $i";;
             esac
         done
