--- Makefile
+++ Makefile
@@ -10,4 +10,4 @@
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-Wall -Winline -O2 -g $(BIGFILES)
+CFLAGS+=-Wall -Winline $(BIGFILES)
 
 # Where you want it installed when you do 'make install'
--- Makefile-libbz2_so.orig	2004-12-26 16:32:23.972195408 -0500
+++ Makefile-libbz2_so	2004-12-26 16:32:34.835543928 -0500
@@ -11,4 +11,4 @@
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-fpic -fPIC -Wall -Winline -O2 -g $(BIGFILES)
+CFLAGS+=-fpic -fPIC -Wall -Winline $(BIGFILES)
 
 OBJS= blocksort.o  \
