diff -p -up pygobject-2.16.1/pygtk.py.fixdetection pygobject-2.16.1/pygtk.py
--- pygobject-2.16.1/pygtk.py.fixdetection	2009-02-20 22:27:14.000000000 +0100
+++ pygobject-2.16.1/pygtk.py	2009-02-23 09:44:55.000000000 +0100
@@ -57,6 +57,9 @@ def _get_available_versions():
             # skip empty directories
             if not os.listdir(pathname):
                 continue
+            # only accept directories containing gtk.py or gobject.so
+            if not glob.glob(os.path.join(pathname, "gtk.py")) and not glob.glob(os.path.join(pathname,"gobject.so")):
+                continue
             
             if not versions.has_key(filename[-3:]):
                 versions[filename[-3:]] = pathname
