--- docs/Makefile.am.orig	2006-10-05 10:28:28.000000000 +0200
+++ docs/Makefile.am		2012-07-03 09:59:46.000000000 +0200
@@ -18,9 +18,9 @@
 	  mkdir -lash-manual-html-split; \
 	fi
 	cd lash-manual-html-one-page && \
-	  texi2html --number ../$(top_srcdir)/docs/lash-manual.texi
+	  texi2html ../$(top_srcdir)/docs/lash-manual.texi
 	cd lash-manual-html-split && \
-	  texi2html --number --split=chapter ../$(top_srcdir)/docs/lash-manual.texi
+	  texi2html --split=chapter ../$(top_srcdir)/docs/lash-manual.texi
 	touch html-manual-stamp
 
 CLEANFILES = html-manual-stamp lash-manual.cps