--- cvs-1.11.22/src/diff.c	2007-09-17 16:55:18.000000000 +0200
+++ cvs-1.11.22/src/diff.c.usercheck	2007-09-17 16:55:28.000000000 +0200
@@ -502,13 +502,8 @@ diff_fileproc (callerdat, finfo)
 
     user_file_rev = 0;
     vers = Version_TS (finfo, NULL, NULL, NULL, 1, 0);
-
-    if (diff_rev2 != NULL || diff_date2 != NULL)
-    {
-	/* Skip all the following checks regarding the user file; we're
-	   not using it.  */
-    }
-    else if (vers->vn_user == NULL)
+    
+    if (vers->vn_user == NULL)
     {
 	/* The file does not exist in the working directory.  */
 	if ((diff_rev1 != NULL || diff_date1 != NULL)
