Index: libvdpau-0.3/src/Makefile.am
===================================================================
--- libvdpau-0.3.orig/src/Makefile.am
+++ libvdpau-0.3/src/Makefile.am
@@ -11,7 +11,7 @@ libvdpau_la_SOURCES = \
 libvdpau_la_LIBADD = \
     $(DLOPEN_LIBS)
 
-libvdpau_la_LDFLAGS = -version-info 1:0:0 -no-undefined
+libvdpau_la_LDFLAGS = -version-info 1:0:0 -no-undefined -ldl
 
 libvdpauincludedir = $(includedir)/vdpau
 libvdpauinclude_HEADERS = \
Index: libvdpau-0.3/trace/Makefile.am
===================================================================
--- libvdpau-0.3.orig/trace/Makefile.am
+++ libvdpau-0.3/trace/Makefile.am
@@ -10,7 +10,7 @@ libvdpau_trace_la_SOURCES = \
 libvdpau_trace_la_LIBADD = \
     $(DLOPEN_LIBS)
 
-libvdpau_trace_la_LDFLAGS = -version-info 1:0:0 -module -no-undefined
+libvdpau_trace_la_LDFLAGS = -version-info 1:0:0 -module -no-undefined -ldl
 
 libvdpau_traceincludedir = $(includedir)/vdpau
 libvdpau_traceinclude_HEADERS = \
