--- snack2.2.10/generic/jkFormatMP3.c~  2005-12-14 11:29:38.000000000 +0000
+++ snack2.2.10/generic/jkFormatMP3.c   2009-10-27 17:31:02.000000000 +0000
@@ -31,7 +31,7 @@
 #include <string.h>
 #define FRAS2(is,a) ((is) > 0 ? t_43[(is)]*(a):-t_43[-(is)]*(a))
 #define MAXFRAMESIZE 2106  /* frame size starting at header */
-#define roundf(x) (floor((x)+(float )0.5f))
+//#define roundf(x) (floor((x)+(float )0.5f))
 static char *gblOutputbuf;
 static char *gblReadbuf;
 static int gblBufind = 0;
