--- snack2.2.10/unix/Makefile.in	2005-12-14 03:29:39.000000000 -0800
+++ snack2.2.10/unix/Makefile.in.new	2008-10-15 19:23:46.000000000 -0700
@@ -33,7 +33,7 @@
 
 CFLAGS    = -O @CFLAGS@ @AFLAG@ ${INCLUDES} @TCLAPI@ -I${GENERIC_DIR} @DEFS@
 
-LIBSO     = -lc @ALIB@ @TCL_LIB_SPEC@
+LIBSO     = -lc -lm @ALIB@ @TCL_LIB_SPEC@
 
 LIBSN     = -lc @TK_LIBS@ @ALIB@ @TCL_LIB_SPEC@ @TK_LIB_SPEC@
 
@@ -293,7 +293,7 @@
 libsnacksphere${SHLIB_SUFFIX}: ${OBJNIST}
 	${SHLIB_LD} ${OBJNIST} ${LIBNIST} -o libsnacksphere${SHLIB_SUFFIX}
 
-LIBOGG = @OGGLIBS@ -lc @TCL_LIB_SPEC@ -L. @SNACK_STUB_LIB_FLAG@
+LIBOGG = @OGGLIBS@ -lc -lm @TCL_LIB_SPEC@ -L. @SNACK_STUB_LIB_FLAG@
 OBJOGG = SnackOgg.o
 
 SnackOgg.o: $(GENERIC_DIR)/SnackOgg.c
