--- Imaging-1.1.6/PIL/Image.py.orig	2009-03-20 18:19:52.184274983 +0000
+++ Image-1.1.6/PIL/Image.py	2009-03-20 18:20:48.605088504 +0000
@@ -2065,9 +2065,7 @@
     else:
         format = None
         if not command:
-            command = "xv"
-            if title:
-                command = command + " -name \"%s\"" % title
+            command = "xdg-open"
 
     if image.mode == "I;16":
         # @PIL88 @PIL101
