--- PIL/Image.py	2006-12-03 14:08:22.000000000 +0200
+++ PIL/Image.py	2007-08-24 13:16:57.000000000 +0300
@@ -2065,9 +2065,9 @@
     else:
         format = None
         if not command:
-            command = "xv"
+            command = "display"
             if title:
-                command = command + " -name \"%s\"" % title
+                command = command + " -title \"%s\"" % title
 
     if image.mode == "I;16":
         # @PIL88 @PIL101
