--- autoconf-2.62/lib/autoconf/status.m4.multiline	2008-04-06 01:04:48.000000000 +0200
+++ autoconf-2.62/lib/autoconf/status.m4	2008-04-19 03:33:27.721188004 +0200
@@ -1360,16 +1360,16 @@ _ACEOF
 cat >>$CONFIG_STATUS <<_ACEOF || ac_write_fail=1
 # Files that config.status was made for.
 m4_ifdef([_AC_SEEN_CONFIG(FILES)],
-[config_files="$ac_config_files"
+[config_files="`echo $ac_config_files`"
 ])dnl
 m4_ifdef([_AC_SEEN_CONFIG(HEADERS)],
-[config_headers="$ac_config_headers"
+[config_headers="`echo $ac_config_headers`"
 ])dnl
 m4_ifdef([_AC_SEEN_CONFIG(LINKS)],
-[config_links="$ac_config_links"
+[config_links="`echo $ac_config_links`"
 ])dnl
 m4_ifdef([_AC_SEEN_CONFIG(COMMANDS)],
-[config_commands="$ac_config_commands"
+[config_commands="`echo $ac_config_commands`"
 ])dnl
 
 _ACEOF
