diff -ur autoconf-2.62.orig/man/Makefile.in autoconf-2.62/man/Makefile.in
--- autoconf-2.62.orig/man/Makefile.in	2008-04-06 01:57:19.000000000 +0300
+++ autoconf-2.62/man/Makefile.in	2008-06-26 19:08:46.000000000 +0300
@@ -394,20 +394,6 @@
 
 .x.1:
 	@echo "Updating man page $@"
-	PATH="$(top_builddir)/tests$(PATH_SEPARATOR)$(top_srcdir)/build-aux$(PATH_SEPARATOR)$$PATH"; \
-	export PATH; \
-	$(HELP2MAN) \
-	    --include=$*.x \
-	    --include=$(srcdir)/common.x \
-	    --source='$(PACKAGE_STRING)' \
-	    --output=$@.t `echo '$*' | sed 's,.*/,,'`
-	if sed $(remove_time_stamp) $@ >$@a.t 2>/dev/null && \
-	   sed $(remove_time_stamp) $@.t | cmp $@a.t - >/dev/null 2>&1; then \
-		touch $@; \
-	else \
-		mv $@.t $@; \
-	fi
-	rm -f $@*.t
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
