diff --git a/libs/internals/connection.cpp b/libs/internals/connection.cpp
index 7085b1b..20b60d8 100644
--- a/libs/internals/connection.cpp
+++ b/libs/internals/connection.cpp
@@ -112,10 +112,10 @@ QString Connection::scopeAsString(Connection::Scope scope)
     QString scopeString;
     switch (scope) {
         case User:
-            scopeString = QLatin1String("User");
+            scopeString = i18n("User");
             break;
         case System:
-            scopeString = QLatin1String("System");
+            scopeString = i18n("System");
             break;
         default:
             break;
@@ -125,7 +125,7 @@ QString Connection::scopeAsString(Connection::Scope scope)
 
 Connection::Scope Connection::scopeFromString(const QString & scopeString)
 {
-    if (scopeString == QLatin1String("User")) {
+    if (scopeString == i18n("User")) {
         return Connection::User;
     }
     return Connection::System;
diff --git a/libs/internals/connection.h b/libs/internals/connection.h
index bb7e21b..3b3491a 100644
--- a/libs/internals/connection.h
+++ b/libs/internals/connection.h
@@ -26,6 +26,8 @@ License along with this library.  If not, see <http://www.gnu.org/licenses/>.
 #include <QString>
 #include <QUuid>
 
+#include <klocalizedstring.h>
+
 #include <solid/control/networkinterface.h>
 
 #include "setting.h"
