Index: networkmanagement/CMakeLists.txt
===================================================================
--- networkmanagement.orig/CMakeLists.txt
+++ networkmanagement/CMakeLists.txt
@@ -75,6 +75,7 @@ if (NETWORKMANAGER_FOUND)
         add_subdirectory(vpnplugins)
         add_subdirectory(solidcontrolfuture)
         add_subdirectory(kded)
+        add_subdirectory(po)
 
         # This is unmaintained, do not use it.
         #add_subdirectory(monolithic)
