diff --git a/backends/NetworkManager/settings/802-1xdbus.cpp b/backends/NetworkManager/settings/802-1xdbus.cpp
index 171f4f4..f995a2b 100644
--- a/backends/NetworkManager/settings/802-1xdbus.cpp
+++ b/backends/NetworkManager/settings/802-1xdbus.cpp
@@ -30,9 +30,15 @@ void Security8021xDbus::fromMap(const QVariantMap & map)
             setting->setAnonymousidentity(map.value(QLatin1String(NM_SETTING_802_1X_ANONYMOUS_IDENTITY)).value<QString>());
         }
         if (map.contains(QLatin1String(NM_SETTING_802_1X_CA_CERT))) {
-            setting->setCacert(map.value(QLatin1String(NM_SETTING_802_1X_CA_CERT)).value<QByteArray>());
+            QByteArray certBa = map.value(QLatin1String(NM_SETTING_802_1X_CA_CERT)).value<QByteArray>();
+            QString certString = QString::fromUtf8(certBa);
+
+            if (certString.startsWith("file://"))
+                setting->setCapath(certString.mid(7));
+            else
+                setting->setCacert(map.value(QLatin1String(NM_SETTING_802_1X_CA_CERT)).value<QByteArray>());
         }
-        if (map.contains(QLatin1String(NM_SETTING_802_1X_CA_PATH))) {
+        if (map.contains(QLatin1String(NM_SETTING_802_1X_CA_PATH)) && setting->capath().isEmpty()) {
             setting->setCapath(map.value(QLatin1String(NM_SETTING_802_1X_CA_PATH)).value<QString>());
         }
         if (map.contains(QLatin1String(NM_SETTING_802_1X_CLIENT_CERT))) {
@@ -105,11 +111,11 @@ QVariantMap Security8021xDbus::toMap()
         if (!setting->anonymousidentity().isEmpty()) {
             map.insert(QLatin1String(NM_SETTING_802_1X_ANONYMOUS_IDENTITY), setting->anonymousidentity());
         }
-        if (!setting->cacert().isEmpty()) {
-            map.insert(QLatin1String(NM_SETTING_802_1X_CA_CERT), setting->cacert());
-        }
         if (!setting->capath().isEmpty()) {
-            map.insert(QLatin1String(NM_SETTING_802_1X_CA_PATH), setting->capath());
+            map.insert(QLatin1String(NM_SETTING_802_1X_CA_CERT), QString(QString("file://") + setting->capath()).toUtf8().append('\0'));
+        }
+        if (!setting->cacert().isEmpty() && !map.contains(QLatin1String(NM_SETTING_802_1X_CA_CERT))) {
+            map.insert(QLatin1String(NM_SETTING_802_1X_CA_CERT), setting->cacert());
         }
         if (!setting->clientcert().isEmpty()) {
             map.insert(QLatin1String(NM_SETTING_802_1X_CLIENT_CERT), setting->clientcert());
