diff -uNBwp qtwitter-app/x11/qtwitter.desktop.orig qtwitter-app/x11/qtwitter.desktop
--- qtwitter-app/x11/qtwitter.desktop.orig	2009-10-06 11:26:57.605576005 +0300
+++ qtwitter-app/x11/qtwitter.desktop	2009-10-06 11:33:23.353576411 +0300
@@ -3,7 +3,7 @@ Exec=qtwitter
 Name=qTwitter
 Icon=qtwitter
 Type=Application
-Categories=Qt;Network;Instant Messaging;
+Categories=Network;Instant Messaging;
 Comment=A microblogging client
 Comment[ca]=El client de Twitter i Identi.ca
 Comment[de]=Ein Twitter und Identi.ca Kunde
@@ -12,6 +12,7 @@ Comment[fr]=Un client de Twitter et Iden
 Comment[ja]=ｷｭｰトゥウィータ - Qttによるトゥウィータクライアント
 Comment[pl]=Klient mikroblogów
 Comment[pt_BR]=Um cliente Twitter e Identi.ca
+Comment[tr]=Için Twitter istemcisi
 GenericName=qTwitter
 Terminal=false
 X-KDE-StartupNotify=false
