diff -uNBwp common.pri.orig common.pri
--- common.pri.orig	2011-02-07 23:23:16.572298874 +0100
+++ common.pri	2011-02-07 23:28:14.712305712 +0100
@@ -12,10 +12,6 @@ isEmpty(COMMON_PRI) {
     DEFINES += SHARE_DIR='\\\"$${SHARE_DIR}\\\"'
     DEFINES += HAVE_OAUTH
 
-    unix:!mac {
-        UNAME = $$system(uname -m)
-        contains( UNAME, x86_64 ): LIB_SUFFIX = 64
-    }
 
     UI_DIR = $${TOP}/tmp
     MOC_DIR = $${TOP}/tmp
