diff -ur gnome-panel-2.31.90.orig/gnome-panel/libpanel-util/Makefile.am gnome-panel-2.31.90/gnome-panel/libpanel-util/Makefile.am
--- gnome-panel-2.31.90.orig/gnome-panel/libpanel-util/Makefile.am	2010-03-17 08:45:18.000000000 +0200
+++ gnome-panel-2.31.90/gnome-panel/libpanel-util/Makefile.am	2010-08-28 06:16:37.675000069 +0300
@@ -39,4 +39,6 @@
 	panel-xdg.h			\
 	$(NULL)
 
+libpanel_util_la_LIBADD = -ldbus-glib-1
+
 -include $(top_srcdir)/git.mk
