From 76dc5a00044de3076fe7117ff9944e4c5660e5ba Mon Sep 17 00:00:00 2001
From: Sebastien Bacher <seb128@ubuntu.com>
Date: Wed, 17 Nov 2010 17:41:17 +0000
Subject: build: list the objects before the libraries to fix build with --as-needed

---
diff --git a/applets/notification_area/Makefile.am b/applets/notification_area/Makefile.am
index c70ce9a..70d3937 100644
--- a/applets/notification_area/Makefile.am
+++ b/applets/notification_area/Makefile.am
@@ -29,17 +29,18 @@ libtray_la_SOURCES =		\
 NOTIFICATION_AREA_SOURCES = main.c
 
 NOTIFICATION_AREA_LDADD =				\
+	libtray.la    				        \                               
 	../../libpanel-applet/libpanel-applet-3.la	\
 	$(X_LIBS)					\
 	$(NOTIFICATION_AREA_LIBS)			\
-	$(LIBPANEL_APPLET_LIBS)				\
-	libtray.la
+	$(LIBPANEL_APPLET_LIBS)
 
 testtray_SOURCES = testtray.c
 testtray_LDADD =			\
+	libtray.la			\
 	$(X_LIBS)			\
-	$(NOTIFICATION_AREA_LIBS)	\
-	libtray.la
+	$(NOTIFICATION_AREA_LIBS)
+
 
 if NOTIFICATION_AREA_INPROCESS
 APPLET_IN_PROCESS = true
--
cgit v0.8.3.1
