diff -ur gnome-panel-2.32.0.2.orig//gnome-panel/panel-default-setup.entries gnome-panel-2.32.0.2/gnome-panel/panel-default-setup.entries
--- gnome-panel-2.32.0.2.orig//gnome-panel/panel-default-setup.entries	2010-09-07 02:00:14.000000000 +0300
+++ gnome-panel-2.32.0.2/gnome-panel/panel-default-setup.entries	2010-10-30 16:01:12.849000063 +0300
@@ -29,12 +29,6 @@
           <value>
             <string>menu_bar</string>
           </value>
-          <value>
-            <string>browser_launcher</string>
-          </value>
-          <value>
-            <string>email_launcher</string>
-          </value>
         </list>
       </value>
     </entry>
@@ -220,114 +214,6 @@
     <entry><key>objects/menu_bar/launcher_location</key><schema_key>/schemas/apps/panel/objects/launcher_location</schema_key></entry>
     <entry><key>objects/menu_bar/action_type</key><schema_key>/schemas/apps/panel/objects/action_type</schema_key></entry>
 
-  <!-- Browser Launcher -->
-
-    <entry>
-      <key>objects/browser_launcher/object_type</key>
-      <schema_key>/schemas/apps/panel/objects/object_type</schema_key>
-      <value>
-        <string>launcher-object</string>
-      </value>
-    </entry>
-    <entry>
-      <key>objects/browser_launcher/toplevel_id</key>
-      <schema_key>/schemas/apps/panel/objects/toplevel_id</schema_key>
-      <value>
-        <string>top_panel</string>
-      </value>
-    </entry>
-    <entry>
-      <key>objects/browser_launcher/position</key>
-      <schema_key>/schemas/apps/panel/objects/position</schema_key>
-      <value>
-        <int>1</int>
-      </value>
-    </entry>
-    <entry>
-      <key>objects/browser_launcher/panel_right_stick</key>
-      <schema_key>/schemas/apps/panel/objects/panel_right_stick</schema_key>
-      <value>
-        <bool>false</bool>
-      </value>
-    </entry>
-    <entry>
-      <key>objects/browser_launcher/locked</key>
-      <schema_key>/schemas/apps/panel/objects/locked</schema_key>
-      <value>
-        <bool>true</bool>
-      </value>
-    </entry>
-    <entry>
-      <key>objects/browser_launcher/launcher_location</key>
-      <schema_key>/schemas/apps/panel/objects/launcher_location</schema_key>
-      <value>
-        <string>epiphany.desktop</string>
-      </value>
-    </entry>
-
-    <entry><key>objects/browser_launcher/applet_iid</key><schema_key>/schemas/apps/panel/objects/applet_iid</schema_key></entry>
-    <entry><key>objects/browser_launcher/attached_toplevel_id</key><schema_key>/schemas/apps/panel/objects/attached_toplevel_id</schema_key> </entry>
-    <entry><key>objects/browser_launcher/tooltip</key><schema_key>/schemas/apps/panel/objects/tooltip</schema_key></entry>
-    <entry><key>objects/browser_launcher/use_custom_icon</key><schema_key>/schemas/apps/panel/objects/use_custom_icon</schema_key></entry>
-    <entry><key>objects/browser_launcher/custom_icon</key><schema_key>/schemas/apps/panel/objects/custom_icon</schema_key></entry>
-    <entry><key>objects/browser_launcher/use_menu_path</key><schema_key>/schemas/apps/panel/objects/use_menu_path</schema_key></entry>
-    <entry><key>objects/browser_launcher/menu_path</key><schema_key>/schemas/apps/panel/objects/menu_path</schema_key></entry>
-    <entry><key>objects/browser_launcher/action_type</key><schema_key>/schemas/apps/panel/objects/action_type</schema_key></entry>
-
-  <!-- Email Launcher -->
-
-    <entry>
-      <key>objects/email_launcher/object_type</key>
-      <schema_key>/schemas/apps/panel/objects/object_type</schema_key>
-      <value>
-        <string>launcher-object</string>
-      </value>
-    </entry>
-    <entry>
-      <key>objects/email_launcher/toplevel_id</key>
-      <schema_key>/schemas/apps/panel/objects/toplevel_id</schema_key>
-      <value>
-        <string>top_panel</string>
-      </value>
-    </entry>
-    <entry>
-      <key>objects/email_launcher/position</key>
-      <schema_key>/schemas/apps/panel/objects/position</schema_key>
-      <value>
-        <int>2</int>
-      </value>
-    </entry>
-    <entry>
-      <key>objects/email_launcher/panel_right_stick</key>
-      <schema_key>/schemas/apps/panel/objects/panel_right_stick</schema_key>
-      <value>
-        <bool>false</bool>
-      </value>
-    </entry>
-    <entry>
-      <key>objects/email_launcher/locked</key>
-      <schema_key>/schemas/apps/panel/objects/locked</schema_key>
-      <value>
-        <bool>true</bool>
-      </value>
-    </entry>
-    <entry>
-      <key>objects/email_launcher/launcher_location</key>
-      <schema_key>/schemas/apps/panel/objects/launcher_location</schema_key>
-      <value>
-        <string>evolution.desktop</string>
-      </value>
-    </entry>
-
-    <entry><key>objects/email_launcher/applet_iid</key><schema_key>/schemas/apps/panel/objects/applet_iid</schema_key></entry>
-    <entry><key>objects/email_launcher/attached_toplevel_id</key><schema_key>/schemas/apps/panel/objects/attached_toplevel_id</schema_key> </entry>
-    <entry><key>objects/email_launcher/tooltip</key><schema_key>/schemas/apps/panel/objects/tooltip</schema_key></entry>
-    <entry><key>objects/email_launcher/use_custom_icon</key><schema_key>/schemas/apps/panel/objects/use_custom_icon</schema_key></entry>
-    <entry><key>objects/email_launcher/custom_icon</key><schema_key>/schemas/apps/panel/objects/custom_icon</schema_key></entry>
-    <entry><key>objects/email_launcher/use_menu_path</key><schema_key>/schemas/apps/panel/objects/use_menu_path</schema_key></entry>
-    <entry><key>objects/email_launcher/menu_path</key><schema_key>/schemas/apps/panel/objects/menu_path</schema_key></entry>
-    <entry><key>objects/email_launcher/action_type</key><schema_key>/schemas/apps/panel/objects/action_type</schema_key></entry>
-
   <!-- Window Menu Applet -->
 
     <entry>
