Index: libproxy-0.4.6/bindings/perl/src/CMakeLists.txt
===================================================================
--- libproxy-0.4.6.orig/bindings/perl/src/CMakeLists.txt
+++ libproxy-0.4.6/bindings/perl/src/CMakeLists.txt
@@ -1,5 +1,5 @@
 include_directories( ${PERL_INCLUDE_PATH} ${CMAKE_SOURCE_DIR}/libproxy )
-link_directories(${LIBPROXY_LIBRARY_DIRS})
+link_directories(${LIBPROXY_LIBRARY_DIRS} ${MOZJS_LIBRARY_DIRS})
 
 # Run doxsubpp.pl to run xsubpp on Libproxy.xs
 add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/Libproxy.c
Index: libproxy-0.4.6/utils/CMakeLists.txt
===================================================================
--- libproxy-0.4.6.orig/utils/CMakeLists.txt
+++ libproxy-0.4.6/utils/CMakeLists.txt
@@ -1,5 +1,5 @@
 include_directories("../libproxy")
-link_directories(${LIBPROXY_LIBRARY_DIRS})
+link_directories(${LIBPROXY_LIBRARY_DIRS} ${MOZJS_LIBRARY_DIRS})
 
 add_executable(proxy proxy.c)
 target_link_libraries(proxy libproxy)
