Index: libproxy-0.4.0/bindings/CMakeLists.txt
===================================================================
--- libproxy-0.4.0.orig/bindings/CMakeLists.txt
+++ libproxy-0.4.0/bindings/CMakeLists.txt
@@ -3,7 +3,7 @@ find_package(PythonInterp)
 if(PYTHONINTERP_FOUND)
   execute_process(COMMAND
                   ${PYTHON_EXECUTABLE}
-                  -c "import distutils.sysconfig ; print distutils.sysconfig.get_python_lib(plat_specific=1)"
+                  -c "import distutils.sysconfig ; print distutils.sysconfig.get_python_lib(plat_specific=0)"
                   OUTPUT_VARIABLE pysitepkgdir
                   OUTPUT_STRIP_TRAILING_WHITESPACE)
   install(FILES libproxy.py DESTINATION ${pysitepkgdir})
