--- curl-7.37.1/configure.ac.orig	2014-07-19 10:27:07.972697344 +0200
+++ curl-7.37.1/configure.ac	2014-07-19 10:28:38.279360218 +0200
@@ -1195,7 +1195,7 @@
      if test -n "$host_alias" -a -f "$GSSAPI_ROOT/bin/$host_alias-krb5-config"; then
         GSSAPI_INCS=`$GSSAPI_ROOT/bin/$host_alias-krb5-config --cflags gssapi`
      elif test -f "$GSSAPI_ROOT/bin/krb5-config"; then
-        GSSAPI_INCS=`$GSSAPI_ROOT/bin/krb5-config --cflags gssapi`
+        GSSAPI_INCS=`CFLAGS= $GSSAPI_ROOT/bin/krb5-config --cflags gssapi`
      elif test -f "$KRB5CONFIG"; then
         GSSAPI_INCS=`$KRB5CONFIG --cflags gssapi`
      elif test "$GSSAPI_ROOT" != "yes"; then
@@ -1285,7 +1285,7 @@
         elif test -f "$KRB5CONFIG"; then
            dnl krb5-config doesn't have --libs-only-L or similar, put everything
            dnl into LIBS
-           gss_libs=`$KRB5CONFIG --libs gssapi`
+           gss_libs=`CFLAGS= $KRB5CONFIG --libs gssapi`
            LIBS="$gss_libs $LIBS"
         else
            case $host in