Index: bangarang-bangarang/data/bangarang.desktop
===================================================================
--- bangarang-bangarang.orig/data/bangarang.desktop
+++ bangarang-bangarang/data/bangarang.desktop
@@ -4,6 +4,7 @@ Version=1.0
 Name=Bangarang
 GenericName=Media Player
 GenericName[nl]=Mediaspeler
+GenericName[tr]=Çokluortam Oynatıcı
 GenericName[de]=Medienwiedergabe
 GenericName[zh_CN]=媒体播放器
 Icon=bangarang
Index: bangarang-bangarang/data/bangarang_play_cd.desktop
===================================================================
--- bangarang-bangarang.orig/data/bangarang_play_cd.desktop
+++ bangarang-bangarang/data/bangarang_play_cd.desktop
@@ -9,6 +9,7 @@ X-KDE-Solid-Predicate=[ OpticalDisc.avai
 Exec=bangarang --play-cd %u
 Icon=bangarang
 Name=Play CD with Bangarang
+Name[tr]=CD'yi Bangarang ile Oynat
 Name[nl]=CD afspelen met Bangarang
 Name[de]=CD mit Bangarang wiedergeben
 Name[zh_CN]=使用 Bangarang 播放 CD
Index: bangarang-bangarang/data/bangarang_play_dvd.desktop
===================================================================
--- bangarang-bangarang.orig/data/bangarang_play_dvd.desktop
+++ bangarang-bangarang/data/bangarang_play_dvd.desktop
@@ -9,6 +9,7 @@ X-KDE-Solid-Predicate=[ OpticalDisc.avai
 Exec=bangarang --play-dvd %u
 Icon=bangarang
 Name=Play DVD with Bangarang
+Name[tr]=DVD'yi Bangarang ile Oynat
 Name[nl]=DVD afspelen met Bangarang
 Name[de]=DVD mit Bangarang wiedergeben
 Name[zh_CN]=使用 Bangarang 播放 DVD
