Index: qd-2.3.4/configure.ac
===================================================================
--- qd-2.3.4.orig/configure.ac	2008-02-15 09:21:51.000000000 +0100
+++ qd-2.3.4/configure.ac	2008-02-15 09:21:57.000000000 +0100
@@ -312,7 +312,7 @@
 
 # libtool stuff
 # AC_DISABLE_SHARED
-# AC_PROG_LIBTOOL
+AC_PROG_LIBTOOL
 AC_PROG_RANLIB
 
 # Output
Index: qd-2.3.4/src/Makefile.am
===================================================================
--- qd-2.3.4.orig/src/Makefile.am	2008-02-15 09:22:06.000000000 +0100
+++ qd-2.3.4/src/Makefile.am	2008-02-15 09:22:16.000000000 +0100
@@ -1,9 +1,9 @@
 SRC = c_dd.cpp c_qd.cpp dd_real.cpp dd_const.cpp \
       fpu.cpp qd_real.cpp qd_const.cpp util.cpp bits.cpp util.h
 
-lib_LIBRARIES = libqd.a
+lib_LTLIBRARIES = libqd.la
 
-libqd_a_SOURCES = $(SRC)
+libqd_la_SOURCES = $(SRC)
 
 AM_CPPFLAGS = -I$(top_builddir) -I$(top_builddir)/include -I$(top_srcdir)/include
 
