diff -Nuar sexy-python-0.1.9.orig/sexy/Makefile.in sexy-python-0.1.9/sexy/Makefile.in
--- sexy-python-0.1.9.orig/sexy/Makefile.in	2009-07-02 14:00:23.000000000 +0300
+++ sexy-python-0.1.9/sexy/Makefile.in	2009-07-02 14:06:45.000000000 +0300
@@ -198,7 +198,7 @@
 defs_DATA = sexy.defs
 pkgpyexec_LTLIBRARIES = sexy.la
 sexy_la_CFLAGS = $(LIBSEXY_CFLAGS)
-sexy_la_LDFLAGS = -module -avoid-version -export-symbols-regex initsexy
+sexy_la_LDFLAGS = -module -avoid-version -export-symbols-regex initsexy `python-config --libs`
 sexy_la_LIBADD = $(LIBSEXY_LIBS)
 sexy_la_SOURCES = sexymodule.c 
 nodist_sexy_la_SOURCES = sexy.c
