--- time-1.7/Makefile.in.destdir	2007-02-27 12:13:10.000000000 +0100
+++ time-1.7/Makefile.in	2007-02-27 12:18:07.000000000 +0100
@@ -114,10 +114,10 @@
 maintainer-clean-binPROGRAMS:
 
 install-binPROGRAMS: $(bin_PROGRAMS)
-	$(mkinstalldirs) $(bindir)
+	$(mkinstalldirs) $(DESTDIR)/$(bindir)
 	list="$(bin_PROGRAMS)"; for p in $$list; do \
 	  if test -f $$p; then \
-	    $(INSTALL_PROGRAM) $$p $(bindir)/`echo $$p|sed '$(transform)'`; \
+	    $(INSTALL_PROGRAM) $$p $(DESTDIR)/$(bindir)/`echo $$p|sed '$(transform)'`; \
 	  else :; fi; \
 	done
 
@@ -176,10 +176,10 @@
 	TEXINPUTS=$(srcdir):$$TEXINPUTS $(TEXI2DVI) $<
 
 install-info: $(INFO_DEPS)
-	$(mkinstalldirs) $(infodir)
+	$(mkinstalldirs) $(DESTDIR)/$(infodir)
 	for file in $(INFO_DEPS); do		\
 	  for ifile in `cd $(srcdir) && echo $$file*`; do \
-	    $(INSTALL_DATA) $(srcdir)/$$ifile $(infodir)/$$ifile; \
+	    $(INSTALL_DATA) $(srcdir)/$$ifile $(DESTDIR)/$(infodir)/$$ifile; \
 	  done;					\
 	done
 
@@ -279,7 +279,7 @@
 install-strip:
 	$(MAKE) INSTALL_PROGRAM='$(INSTALL_PROGRAM) -s' install
 installdirs:
-	$(mkinstalldirs)  $(bindir) $(infodir)
+	$(mkinstalldirs)  $(DESTDIR)/$(bindir) $(DESTDIR)/$(infodir)
 
 
 mostlyclean-generic:
