diff -Nur disktype-9-old/dos.c disktype-9/dos.c
--- disktype-9-old/dos.c	2008-07-04 11:06:22.000000000 +0300
+++ disktype-9/dos.c	2008-07-04 11:22:59.000000000 +0300
@@ -485,6 +485,7 @@
     strcpy(s, ", ATARI ST bootable");
   print_line(level, "%s file system (hints score %d of %d%s)",
 	     fatnames[fattype], score, 5, s);
+  print_line(level, "KERNELMODULE: vfat");
 
   if (sectsize > 512)
     print_line(level + 1, "Unusual sector size %lu bytes", sectsize);
@@ -551,6 +552,7 @@
 
   /* tell the user */
   print_line(level, "NTFS file system");
+  print_line(level, "KERNELMODULE: ntfs");
 
   format_blocky_size(s, sectcount, sectsize, "sectors", NULL);
   print_line(level + 1, "Volume size %s", s);
