--- quassel-0.5.2/src/common/aliasmanager.cpp.orig	2010-02-27 20:42:00.609842752 +0200
+++ quassel-0.5.2/src/common/aliasmanager.cpp	2010-02-27 20:42:51.397850978 +0200
@@ -98,6 +98,8 @@
 #ifdef Q_OS_LINUX
   // let's add aliases for scripts that only run on linux
   aliases << Alias("inxi", "/exec inxi $0")
+          << Alias("amarok", "/exec mpris amarok")
+          << Alias("qmmp", "/exec mpris qmmp")
           << Alias("sysinfo", "/exec inxi -d");
 #endif
 
