diff -Nuar a/source/glest_game/facilities/components.h b/source/glest_game/facilities/components.h
--- a/source/glest_game/facilities/components.h	2013-06-13 04:37:15.000000000 +0300
+++ b/source/glest_game/facilities/components.h	2013-10-04 08:00:39.889939066 +0300
@@ -256,8 +256,8 @@
 
     void pushBackItem(string item, string translated_item="");
     void setItems(const vector<string> &items, const vector<string> translated_items=vector<string>());
-	void setSelectedItemIndex(int index, bool errorOnMissing=true);
-    void setSelectedItem(string item, bool errorOnMissing=true);
+	void setSelectedItemIndex(int index, bool errorOnMissing=false);
+    void setSelectedItem(string item, bool errorOnMissing=false);
     void setEditable(bool editable);
 
     bool hasItem(string item) const;
