diff -up megaglest-3.6.0.3/mk/cmake/Modules/FindXercesC.cmake.orig megaglest-3.6.0.3/mk/cmake/Modules/FindXercesC.cmake
--- megaglest-3.6.0.3/mk/cmake/Modules/FindXercesC.cmake.orig	2012-04-28 17:25:05.688259468 -0300
+++ megaglest-3.6.0.3/mk/cmake/Modules/FindXercesC.cmake	2012-04-28 17:25:18.598259855 -0300
@@ -70,6 +70,7 @@ IF (NOT XERCESC_INCLUDE)
 	 /usr/local/include
 	 /usr/include
 	)
+    SET(XERCESC_INCLUDE_DIR ${XERCESC_INCLUDE})
 ENDIF()
 
 #MESSAGE(STATUS "Current Xerces-C lib [${XERCESC_LIBRARY}] XERCESC_INCLUDE [${XERCESC_INCLUDE}")
