Index: kde-workspace-4.7.1/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
===================================================================
--- kde-workspace-4.7.1.orig/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
+++ kde-workspace-4.7.1/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
@@ -5,7 +5,8 @@ if (panelIds.length == 1) {
 }
 
 panel.height = screenGeometry(panel.screen).height > 1024 ? 35 : 30
-panel.addWidget("launcher")
+var launcher = panel.addWidget("launcher")
+launcher.globalShortcut = "Alt+F1";
 panel.addWidget("showdesktop")
 panel.addWidget("org.kde.showActivityManager")
 panel.addWidget("pager")
