Index: kdm/backend/client.c
===================================================================
--- kdm/backend/client.c.orig	2010-11-19 11:51:17.000000000 +0100
+++ kdm/backend/client.c	2010-11-21 20:47:03.608717921 +0100
@@ -462,6 +462,9 @@ isNoPassAllowed(struct passwd *pw)
     if (cursource != PWSRC_MANUAL)
         return True;
 
+	if (td->noPassAllUsers)
+		return True;
+
 #if defined(USE_PAM) || defined(_AIX)
     /* Give nss_ldap, etc. a chance to normalize (uppercase) the name. */
     if (!(pw = getpwnam(un)) ||
Index: kdm/config.def
===================================================================
--- kdm/config.def.orig	2010-11-21 20:47:03.509842471 +0100
+++ kdm/config.def	2010-11-21 20:47:28.785842630 +0100
@@ -1890,6 +1890,15 @@ Description:
  (and any other user with UID = 0).
  <emphasis>Never</emphasis> list <systemitem class="username">root</systemitem>.
 
+Key: NoPassAllUsers
+Type: bool
+Default: false
+User: core
+Instance: #:0/true
+Comment: &
+Description:
+ All users can login without password
+
 Key: AutoLoginEnable
 Type: bool
 Default: false
