Index: kdm/kfrontend/kdm_greet.c
===================================================================
--- kdm/kfrontend/kdm_greet.c.orig	2010-07-06 08:54:31.000000000 +0200
+++ kdm/kfrontend/kdm_greet.c	2010-11-21 20:47:03.694718280 +0100
@@ -458,7 +458,6 @@ secureInputs(Display *dpy)
 {
     debug("secureInputs %s\n", dname);
     secureKeyboard(dpy);
-    securePointer(dpy);
     XSetInputFocus(dpy, None, None, CurrentTime);
     debug("secureInputs %s done\n", dname);
 }
@@ -469,7 +468,6 @@ unsecureInputs(Display *dpy)
     debug("unsecureInputs %s\n", dname);
     XSetInputFocus(dpy, PointerRoot, PointerRoot, CurrentTime);
     XUngrabKeyboard(dpy, CurrentTime);
-    XUngrabPointer(dpy, CurrentTime);
     XSync(dpy, False);
 }
 
