--- plasma/containments/plaindesktop/CMakeLists.txt	2009-01-11 11:37:56.000000000 +0100
+++ plasma/containments/plaindesktop/CMakeLists.txt	2009-05-15 14:07:28.000000000 +0200
@@ -9,13 +9,12 @@
 set(screensaver_xml ${KDEBASE_WORKSPACE_SOURCE_DIR}/krunner/dbus/org.freedesktop.ScreenSaver.xml)
 QT4_ADD_DBUS_INTERFACE(desktop_SRCS ${screensaver_xml} screensaver_interface)
 
-kde4_add_plugin(plasma_containment_desktop ${desktop_SRCS})
+kde4_add_plugin(plasma_containment_plaindesktop ${desktop_SRCS})
 if(WIN32)
-target_link_libraries(plasma_containment_desktop ${KDE4_PLASMA_LIBS} ${KDE4_KIO_LIBS} ${KDE4_KFILE_LIBS})
+target_link_libraries(plasma_containment_plaindesktop ${KDE4_PLASMA_LIBS} ${KDE4_KIO_LIBS} ${KDE4_KFILE_LIBS})
 else(WIN32)
-target_link_libraries(plasma_containment_desktop ${KDE4_PLASMA_LIBS} kworkspace ${KDE4_KIO_LIBS} ${KDE4_KFILE_LIBS} )
+target_link_libraries(plasma_containment_plaindesktop ${KDE4_PLASMA_LIBS} kworkspace ${KDE4_KIO_LIBS} ${KDE4_KFILE_LIBS} )
 endif(WIN32)
 
-install(TARGETS plasma_containment_desktop DESTINATION ${PLUGIN_INSTALL_DIR})
-install(FILES plasma-containment-desktop.desktop DESTINATION ${SERVICES_INSTALL_DIR})
-install(FILES plasma_desktop_containment.kwinrules DESTINATION ${DATA_INSTALL_DIR}/kwin/default_rules)
+install(TARGETS plasma_containment_plaindesktop DESTINATION ${PLUGIN_INSTALL_DIR})
+install(FILES plasma-containment-plaindesktop.desktop DESTINATION ${SERVICES_INSTALL_DIR})
--- plasma/containments/plaindesktop/desktop.cpp	2009/11/20 20:40:57	1.1
+++ plasma/containments/plaindesktop/desktop.cpp	2009/11/20 20:41:10
@@ -59,7 +59,7 @@
 using namespace Plasma;
 
 DefaultDesktop::DefaultDesktop(QObject *parent, const QVariantList &args)
-    : Containment(parent, args),
+    : Containment(parent, args, false),
       m_addPanelsMenu(0),
       m_addPanelAction(0),
       m_runCommandAction(0),
--- plasma/containments/plaindesktop/plasma-containment-plaindesktop.desktop	2009-05-15 14:15:33.000000000 +0200
+++ plasma/containments/plaindesktop/plasma-containment-plaindesktop.desktop	2009-05-15 14:07:28.000000000 +0200
@@ -1,17 +1,17 @@
 [Desktop Entry]
 # ctxt: plasma containment
-Name=Desktop
-Comment=Default desktop containment
+Name=Plain Desktop
+Comment=Desktop without toolbox
 Icon=user-desktop
 Type=Service
 X-KDE-ServiceTypes=Plasma/Applet,Plasma/Containment
 X-Plasma-ContainmentCategories=desktop
 NoDisplay=true
 
-X-KDE-Library=plasma_containment_desktop
+X-KDE-Library=plasma_containment_plaindesktop
 X-KDE-PluginInfo-Author=The Plasma Team
 X-KDE-PluginInfo-Email=panel-devel@kde.org
-X-KDE-PluginInfo-Name=desktop
+X-KDE-PluginInfo-Name=plaindesktop
 X-KDE-PluginInfo-Version=1.0
 X-KDE-PluginInfo-Website=http://plasma.kde.org/
 X-KDE-PluginInfo-Category=Containments

Index: plasma/containments/CMakeLists.txt
===================================================================
--- plasma/containments/CMakeLists.txt	2009/11/20 20:40:06	1.1
+++ plasma/containments/CMakeLists.txt	2009/11/20 20:41:10
@@ -1,3 +1,4 @@
 add_subdirectory(desktop)
 add_subdirectory(panel)
 add_subdirectory(screensaver)
+add_subdirectory(plaindesktop)
