diff --git a/plasma/generic/dataengines/notifications/notificationsengine.cpp b/plasma/generic/dataengines/notifications/notificationsengine.cpp
index 898b4f6..b75f2bd 100644
--- a/plasma/generic/dataengines/notifications/notificationsengine.cpp
+++ b/plasma/generic/dataengines/notifications/notificationsengine.cpp
@@ -211,6 +211,7 @@
         killTimer(event->timerId());
         m_sourceTimers.remove(source);
         m_timeouts.remove(event->timerId());
+        killTimer(event->timerId());
         removeSource(source);
         emit NotificationClosed(source.split(" ").last().toInt(), 1);
         return;
