--- plasma/applets/systemtray/ui/applet.cpp.sav	2009-06-23 13:15:05.000000000 +0200
+++ plasma/applets/systemtray/ui/applet.cpp	2009-10-07 16:48:47.851910189 +0200
@@ -117,6 +117,7 @@
     static SystemTray::Manager *s_manager;
     static int s_managerUsage;
     int autoHideTimeout;
+    int autoHideJobsTimeout;
     int timerId;
 
     Ui::ProtocolsConfig notificationUi;
@@ -214,6 +215,11 @@
         d->autoHideTimeout = 0;
     }
 
+    d->autoHideJobsTimeout = d->autoHideTimeout;
+    if (config().readEntry("DisableAutoHideJobs", true)) {
+        d->autoHideJobsTimeout = 0;
+    }
+
     d->shownCategories.insert(Task::UnknownCategory);
 
     //destroy any item in the systray, that doesn't belong to the completedJobsGroup, since running
@@ -534,6 +540,10 @@
     } else {
         d->autoHideTimeout = 0;
     }
+    d->autoHideJobsTimeout = d->autoHideTimeout;
+    if (cg.readEntry("DisableAutoHideJobs", true)) {
+        d->autoHideJobsTimeout = 0;
+    }
 
     KConfigGroup globalCg = globalConfig();
     globalCg.writeEntry("ShowJobs", d->notificationUi.showJobs->isChecked());
@@ -610,7 +620,7 @@
     extenderItem->config().writeEntry("type", "job");
     extenderItem->setWidget(new JobWidget(job, extenderItem));
 
-    showPopup(d->autoHideTimeout);
+    showPopup(d->autoHideJobsTimeout);
 
     extenderItem->setGroup(extender()->group("jobGroup"));
 }
