Index: plasma/shells/desktop/plasmaapp.cpp
===================================================================
--- plasma/shells/desktop/plasmaapp.cpp	(revision 1011046)
+++ plasma/shells/desktop/plasmaapp.cpp	(working copy)
@@ -558,6 +558,11 @@
         activityAction->setShortcut(KShortcut("alt+d, alt+a"));
         activityAction->setShortcutContext(Qt::ApplicationShortcut);
 
+        KAction *configureAction = c->addAction("configure plasma");
+        configureAction->setText(i18n("Configure Plasma..."));
+        configureAction->setIcon(KIcon("configure"));
+        connect(configureAction, SIGNAL(triggered()), this, SLOT(createConfigurationInterface()));
+
         KAction *zoomAction = c->addAction("zoom out");
         zoomAction->setText(i18n("Zoom Out"));
         zoomAction->setIcon(KIcon("zoom-out"));
@@ -755,6 +760,7 @@
     if (containment->containmentType() == Plasma::Containment::DesktopContainment
             && m_zoomLevel == Plasma::DesktopZoom) {
         foreach (QAction *action, m_corona->actions()) {
+            containment->removeToolBoxAction(action);
             containment->addToolBoxAction(action);
         }
     }
@@ -948,12 +954,6 @@
                 layout->addWidget(actionButton);
             }
 
-            ToolButton *actionButton = new ToolButton(m_controllerDialog);
-            actionButton->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
-            actionButton->setIcon(KIcon("configure"));
-            actionButton->setText(i18n("Configure Plasma..."));
-            layout->addWidget(actionButton);
-            connect(actionButton, SIGNAL(clicked()), this, SLOT(createConfigurationInterface()));
         }
 
         m_controllerDialog->show();
@@ -969,7 +969,7 @@
     m_configUi.setupUi(widget);
     KConfigSkeleton *nullManager = new KConfigSkeleton(0);
     KConfigDialog *dialog = new KConfigDialog(0, "Plasma settings", nullManager);
-    dialog ->addPage(widget, i18n("Plasma settings"));
+    dialog ->addPage(widget, i18n("Plasma Settings"));
     dialog->setAttribute(Qt::WA_DeleteOnClose, true);
     dialog->setFaceType(KPageDialog::Auto);
     dialog->showButton(KDialog::Apply, false);
