Index: plasma/desktop/shell/panelcontroller.cpp
===================================================================
--- plasma/desktop/shell/panelcontroller.cpp.orig	2010-10-05 13:39:06.000000000 +0200
+++ plasma/desktop/shell/panelcontroller.cpp	2010-11-21 20:47:03.782718456 +0100
@@ -627,6 +627,8 @@ void PanelController::mouseMoveFilter(QM
         newX = qMin(newX, screenGeom.left() + screenGeom.width()/3);
         move(newX, pos().y());
         resizeFrameHeight(geometry().left() - screenGeom.left());
+        m_sizeTool->setText(i18n("Width: %1", geometry().left() - screenGeom.left()));
+
         break;
     }
     case Plasma::RightEdge: {
@@ -635,6 +637,8 @@ void PanelController::mouseMoveFilter(QM
         newX = qMax(newX, screenGeom.left() + 2*(screenGeom.width()/3) - width());
         move(newX, pos().y());
         resizeFrameHeight(screenGeom.right() - geometry().right());
+        m_sizeTool->setText(i18n("Width: %1", screenGeom.right() - geometry().right()));
+
         break;
     }
     case Plasma::TopEdge: {
@@ -643,6 +647,9 @@ void PanelController::mouseMoveFilter(QM
         newY = qMin(newY, screenGeom.top() + screenGeom.height()/3);
         move(pos().x(), newY);
         resizeFrameHeight(geometry().top() - screenGeom.top());
+        m_sizeTool->setText(i18n("Height: %1", geometry().top() - screenGeom.top()));
+
+
         break;
     }
     case Plasma::BottomEdge:
@@ -652,6 +659,8 @@ void PanelController::mouseMoveFilter(QM
         newY = qMax(newY, screenGeom.top() + 2*(screenGeom.height()/3) - height());
         move(pos().x(), newY);
         resizeFrameHeight(screenGeom.bottom() - geometry().bottom());
+        m_sizeTool->setText(i18n("Height: %1", screenGeom.bottom() - geometry().bottom()));
+
         break;
     }
     }
