Index: plasma/desktop/shell/dashboardview.cpp
===================================================================
--- plasma/desktop/shell/dashboardview.cpp.orig	2010-07-24 11:43:13.000000000 +0200
+++ plasma/desktop/shell/dashboardview.cpp	2010-11-21 20:47:03.765718635 +0100
@@ -225,6 +225,15 @@ bool DashboardView::eventFilter(QObject
         widgetExplorer->setPos(0, containment()->geometry().height() - widgetExplorer->geometry().height());
     }
 
+    if (watched != (QObject*)m_widgetExplorer.data()) {
+        if (event->type() == QEvent::MouseButtonPress) {
+            QMouseEvent *me = static_cast<QMouseEvent *>(event);
+            if (me->button() == Qt::LeftButton) {
+                hideView();
+            }
+        }
+    }
+
     return false;
 }
 
