Index: plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
===================================================================
--- plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js.orig
+++ plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
@@ -4,8 +4,9 @@ if (panelIds.length == 1) {
     panel.location = 'bottom'
 }
 
-panel.height = screenGeometry(panel.screen).height > 1024 ? 35 : 27
+panel.height = screenGeometry(panel.screen).height > 1024 ? 35 : 30
 panel.addWidget("launcher")
+panel.addWidget("showdesktop")
 panel.addWidget("org.kde.showActivityManager")
 panel.addWidget("pager")
 tasks = panel.addWidget("tasks")
