diff -up foomatic-db-20100819/db/source/PPD/Ricoh/PS/Ricoh-Aficio_SP_C420DN_PS.ppd.device-ids-20100506 foomatic-db-20100819/db/source/PPD/Ricoh/PS/Ricoh-Aficio_SP_C420DN_PS.ppd
--- foomatic-db-20100819/db/source/PPD/Ricoh/PS/Ricoh-Aficio_SP_C420DN_PS.ppd.device-ids-20100506	2010-08-19 09:41:02.000000000 +0200
+++ foomatic-db-20100819/db/source/PPD/Ricoh/PS/Ricoh-Aficio_SP_C420DN_PS.ppd	2010-08-20 19:03:39.000000000 +0200
@@ -42,6 +42,7 @@
 *PSVersion: "(3017.104) 2"
 *ShortNickName: "Ricoh Aficio SP C420DN PS"
 *NickName: "Ricoh Aficio SP C420DN PS"
+*1284DeviceID: "MFG:RICOH;MDL:Aficio SP C420DN;CMD:PJL,RCS,PCL,PCLXL,POSTSCRIPT,POSTSCRIPT,PCL,PCLXL;"
 
 *%PNP_NAME: "Aficio SP C420DN"
 *%LOADDLL_NAME32: "MFRICRES.dll"
@@ -1629,4 +1630,4 @@ cleartomark\n"
 *CloseGroup: JobLog/Job Log
 
 
-*% end of Printer Description file
\ No newline at end of file
+*% end of Printer Description file
