--- SVK-v2.0.1/lib/SVK/Resolve/Emacs.pm.pix	2006-12-28 14:16:52.000000000 +0100
+++ SVK-v2.0.1/lib/SVK/Resolve/Emacs.pm	2007-06-29 11:10:27.000000000 +0200
@@ -55,30 +55,32 @@ use SVK::I18N;
 use SVK::Util qw( devnull );
 use SVK::Logger;
 
-sub commands { 'gnuclient-emacs' }
+sub commands { 'emacsclient' }
 
 sub arguments {
     my $self = shift;
-    my $lisp = "(require 'svk-ediff)";
+
+    my @lisp;
+    push @lisp, "(require 'svk-ediff)";
 
     # set up the signal handlers
     $self->{signal} ||= 'USR1';
 
     if ($self->{base} eq devnull()) {
-        $lisp .= qq(
+        push @lisp, qq(
 (ediff-files-internal
  "$self->{yours}" "$self->{theirs}" nil
  nil 'ediff-merge-files)
 );
     } else {
-        $lisp .= qq(
+        push @lisp, qq(
 (ediff-files-internal
  "$self->{yours}" "$self->{theirs}" "$self->{base}"
  nil 'ediff-merge-files-with-ancestor)
 )
     }
 
-    $lisp .= qq(
+    push @lisp, qq(
 (svk-merge-startup '((working-file . "$self->{yours}")
                        (selected-file . "$self->{theirs}")
                        (common-file . "$self->{base}")
@@ -91,7 +93,7 @@ sub arguments {
 'OK!
 );
 
-    return ('--eval' => $lisp);
+    return ('--eval', @lisp);
 }
 
 sub run_resolver {
