Index: gcc/Makefile.in
===================================================================
--- gcc/Makefile.in.orig	2011-11-03 17:00:43.000000000 +0100
+++ gcc/Makefile.in	2011-11-03 17:01:28.000000000 +0100
@@ -2145,7 +2145,7 @@ dumpvers: dumpvers.c
 
 CFLAGS-version.o += -DBASEVER=$(FULLVER_s) -DDATESTAMP=$(DATESTAMP_s) \
 	-DREVISION=$(REVISION_s) \
-	-DDEVPHASE=$(DEVPHASE_s) -DPKGVERSION=$(PKGVERSION_s) \
+	-DDEVPHASE="" -DPKGVERSION=$(PKGVERSION_s) \
 	-DBUGURL=$(BUGURL_s)
 version.o: version.c version.h $(REVISION) $(DATESTAMP) $(FULLVER) $(DEVPHASE)
 
