Index: libgcc/Makefile.in
===================================================================
--- libgcc/Makefile.in.orig	2011-11-03 16:29:25.000000000 +0100
+++ libgcc/Makefile.in	2011-11-03 17:21:33.000000000 +0100
@@ -279,7 +279,7 @@ INTERNAL_CFLAGS = $(CFLAGS) $(LIBGCC2_CF
 CRTSTUFF_CFLAGS = -O2 $(GCC_CFLAGS) $(INCLUDES) $(MULTILIB_CFLAGS) -g0 \
   -finhibit-size-directive -fno-inline -fno-exceptions \
   -fno-zero-initialized-in-bss -fno-toplevel-reorder -fno-tree-vectorize \
-  -fno-stack-protector \
+  -fno-stack-protector -fno-unwind-tables -fno-asynchronous-unwind-tables \
   $(INHIBIT_LIBC_CFLAGS)
 
 # Extra flags to use when compiling crt{begin,end}.o.
