--- netkit-ftp-0.17/configure.orig	2005-02-11 19:04:49.000000000 -0500
+++ netkit-ftp-0.17/configure	2005-02-11 19:05:02.000000000 -0500
@@ -120,7 +120,6 @@
          $CC -O2 __conftest.c -o __conftest
        ) >/dev/null 2>&1; then
          echo 'yes'
-         CFLAGS="$CFLAGS -O2"
     else
          echo 'no'
          echo -n "Checking if $CC accepts -O... "
@@ -128,7 +127,6 @@
               $CC -O __conftest.c -o __conftest
             ) >/dev/null 2>&1; then
               echo 'yes'
-              CFLAGS="$CFLAGS -O"
          else
               echo 'no'
          fi
