diff -Nur netkit-ftp-0.17-old/ftp/main.c netkit-ftp-0.17/ftp/main.c
--- netkit-ftp-0.17-old/ftp/main.c	2010-04-08 15:23:20.498571189 +0300
+++ netkit-ftp-0.17/ftp/main.c	2010-04-08 15:23:59.443446852 +0300
@@ -111,7 +111,7 @@
 
 	tick = 0;
 
-	setlocale (LC_ALL, "");
+	setlocale (LC_ALL, "en_US.UTF-8");
 	sp = getservbyname("ftp", "tcp");
 	if (sp == 0) {
 		fprintf(stderr, "ftp: ftp/tcp: unknown service\n");
