--- config-scripts/cups-network.m4~	2014-01-18 14:05:51.353295869 +0100
+++ config-scripts/cups-network.m4	2014-01-18 14:07:25.179962227 +0100
@@ -45,7 +45,7 @@ if test x$enable_domainsocket != xno -a
 		case "$uname" in
 			Darwin*)
 				# Darwin and MaxOS X do their own thing...
-				CUPS_DEFAULT_DOMAINSOCKET="$localstatedir/run/cupsd"
+				CUPS_DEFAULT_DOMAINSOCKET="/run/cupsd"
 				;;
 			*)
 				# All others use FHS standard...
--- config-scripts/cups-directories.m4~	2014-01-18 14:05:59.609962507 +0100
+++ config-scripts/cups-directories.m4	2014-01-18 14:07:13.516628933 +0100
@@ -420,7 +420,7 @@ AC_ARG_WITH(rundir, [  --with-rundir
 			;;
 		*)
 			# All others
-			CUPS_STATEDIR="$localstatedir/run/cups"
+			CUPS_STATEDIR="/run/cups"
 			;;
 	esac])
 AC_DEFINE_UNQUOTED(CUPS_STATEDIR, "$CUPS_STATEDIR")
