diff -up ncurses-5.9/misc/ncurses-config.in.config ncurses-5.9/misc/ncurses-config.in
--- ncurses-5.9/misc/ncurses-config.in.config	2011-07-18 15:58:01.606366364 +0200
+++ ncurses-5.9/misc/ncurses-config.in	2011-07-18 16:06:32.572122421 +0200
@@ -83,12 +82,7 @@ while test $# -gt 0; do
 ENDECHO
 		;;
 	--libs)
-		if test "$libdir" = /usr/lib
-		then
 			LIBDIR=
-		else
-			LIBDIR=-L$libdir
-		fi
 		if test @TINFO_NAME@ = @LIB_NAME@ ; then
 		sed -e 's,^[ ]*,,' -e 's, [ ]*, ,g' -e 's,[ ]*$,,' <<-ENDECHO
 			$LIBDIR @EXTRA_LDFLAGS@ -l${THIS} @LIBS@
@@ -120,7 +114,6 @@ ENDECHO
 		echo "${includedir}"
 		;;
 	--libdir)
-		echo "${libdir}"
 		;;
 	--mandir)
 		echo "${mandir}"
