diff -Naur subtitleripper.orig/spudec.c subtitleripper/spudec.c
--- subtitleripper.orig/spudec.c	2003-12-08 11:28:02.000000000 +0200
+++ subtitleripper/spudec.c	2009-12-23 15:41:50.954828002 +0200
@@ -36,14 +36,14 @@
 #include <zlib.h>
 #endif
 
-#ifdef _HAVE_LIB_PPM_
-#include <ppm.h>
-#endif
-
 #ifdef _HAVE_PNG_
 #include "png.h"
 #endif
 
+#ifdef _HAVE_LIB_PPM_
+#include <ppm.h>
+#endif
+
 static void spudec_handle_rest(spudec_handle_t *this);
 
 // read a big endian 16 bit value
