Index: ghostscript-8.70/base/devs.mak
===================================================================
--- ghostscript-8.70.orig/base/devs.mak
+++ ghostscript-8.70/base/devs.mak
@@ -524,7 +524,7 @@ $(GLOBJ)gdevxalt.$(OBJ) : $(GLSRC)gdevxa
 ### NON PORTABLE, ONLY UNIX WITH GCC SUPPORT
 
 $(GLOBJ)X11.so : $(x11alt_) $(x11_)
-	$(CCLD) $(LDFLAGS) -shared -o $(GLOBJ)X11.so $(x11alt_) $(x11_) -L/usr/X11R6/lib -lXt -lSM -lICE -lXext -lX11 -lgs -lm $(XLIBDIRS)
+	$(CCLD) $(LDFLAGS) -shared -o $(GLOBJ)X11.so $(x11alt_) $(x11_) -L/usr/X11R6/lib -L./sobin -lXt -lSM -lICE -lXext -lX11 -lgs -lm $(XLIBDIRS)
 
 ###### --------------- Memory-buffered printer devices --------------- ######
 
