Index: ghostscript-8.70/base/gdevpsu.c
===================================================================
--- ghostscript-8.70.orig/base/gdevpsu.c
+++ ghostscript-8.70/base/gdevpsu.c
@@ -327,6 +327,11 @@ psw_write_page_header(stream *s, const g
 	    else
 		++p;
 	}
+
+	/* If no match then decrement pointer back to /letter */
+	if (p->width == 0 && p->height == 0)
+	    --p;
+
 	pprintd2(s, "%d %d ", width, height);
 	pprints1(s, "%s setpagesize\n", p->size_name);
     }
