diff -Nur pam_skey-1.1.5-old//Makefile.in pam_skey-1.1.5/Makefile.in
--- pam_skey-1.1.5-old//Makefile.in	2010-09-12 16:35:06.214000719 +0300
+++ pam_skey-1.1.5/Makefile.in	2010-09-12 16:35:27.270001484 +0300
@@ -24,7 +24,7 @@
 all: $(PAM_FILES)
 
 pam_skey.so: pam_skey.o
-	$(CC) $(CFLAGS) -o $@ $< $(LIBS) $(LDFLAGS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< $(LIBS)
 
 lint-pam_skey:
 	lclint $(CFLAGS) pam_skey.c
