Author: Daniel Baumann <daniel@debian.org>
Description: Fixes name of the desktop icon file (Closes: #378842).

diff -Naurp xarchiver.orig/xarchiver.desktop.in xarchiver/xarchiver.desktop.in
--- xarchiver.orig/xarchiver.desktop.in	2008-06-27 05:24:12.000000000 +0000
+++ xarchiver/xarchiver.desktop.in	2009-01-09 16:02:15.000000000 +0000
@@ -5,7 +5,7 @@ _Name=Xarchiver
 _Comment=A GTK+2 only archive manager
 _GenericName=Archive manager
 Exec=xarchiver
-Icon=xarchiver
+Icon=xarchiver.png
 Terminal=false
 Type=Application
 X-MultipleArgs=false
