Author: Daniel Baumann <daniel@debian.org>
Description: Disable html icon (Closes: #505778).

diff -Naurp xarchiver.orig/pixmaps/Makefile.am xarchiver/pixmaps/Makefile.am
--- xarchiver.orig/pixmaps/Makefile.am	2009-06-08 15:53:58.000000000 +0000
+++ xarchiver/pixmaps/Makefile.am	2009-07-26 19:30:08.000000000 +0000
@@ -3,7 +3,6 @@
 pixmapsdir = $(datadir)/pixmaps/xarchiver
 
 pixmaps_DATA = 	xarchiver-add.png \
-		xarchiver-extract.png \
-		xarchiver-html.png
+		xarchiver-extract.png
 
 EXTRA_DIST = $(pixmaps_DATA)
diff -Naurp xarchiver.orig/pixmaps/Makefile.in xarchiver/pixmaps/Makefile.in
--- xarchiver.orig/pixmaps/Makefile.in	2009-06-08 15:53:58.000000000 +0000
+++ xarchiver/pixmaps/Makefile.in	2009-07-26 19:30:08.000000000 +0000
@@ -205,8 +205,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 pixmapsdir = $(datadir)/pixmaps/xarchiver
 pixmaps_DATA = xarchiver-add.png \
-		xarchiver-extract.png \
-		xarchiver-html.png
+		xarchiver-extract.png
 
 EXTRA_DIST = $(pixmaps_DATA)
 all: all-am
diff -Naurp xarchiver.orig/src/interface.c xarchiver/src/interface.c
--- xarchiver.orig/src/interface.c	2009-06-08 15:53:59.000000000 +0000
+++ xarchiver/src/interface.c	2009-07-26 19:30:08.000000000 +0000
@@ -522,7 +522,7 @@ void xa_create_main_window (GtkWidget *x
 	g_signal_connect ((gpointer) new1,"activate",G_CALLBACK (xa_new_archive),NULL);
 	g_signal_connect ((gpointer) open1,"activate",G_CALLBACK (xa_open_archive),NULL);
 	g_signal_connect ((gpointer) listing_text,"activate",G_CALLBACK (xa_list_archive),(gpointer) 0);
-	g_signal_connect ((gpointer) listing_html,"activate",G_CALLBACK (xa_list_archive),(gpointer) 1);
+	/* g_signal_connect ((gpointer) listing_html,"activate",G_CALLBACK (xa_list_archive),(gpointer) 1); */
 	g_signal_connect ((gpointer) save1,"activate",G_CALLBACK (xa_save_archive),NULL);
 	g_signal_connect ((gpointer) check_menu,"activate",G_CALLBACK (xa_test_archive),NULL);
 	g_signal_connect ((gpointer) properties,"activate",G_CALLBACK (xa_archive_properties),NULL);
