diff -Naur xarchiver-0.5.0rc1.orig/xarchiver.desktop.in xarchiver-0.5.0rc1/xarchiver.desktop.in
--- xarchiver-0.5.0rc1.orig/xarchiver.desktop.in	2008-06-27 08:24:12.000000000 +0300
+++ xarchiver-0.5.0rc1/xarchiver.desktop.in	2008-10-22 19:59:37.000000000 +0300
@@ -3,7 +3,9 @@
 Version=1.0
 _Name=Xarchiver
 _Comment=A GTK+2 only archive manager
+_Comment[tr]=GTK+2 Arşiv Yöneticisi
 _GenericName=Archive manager
+_GenericName[tr]=Arşiv Yöneticisi
 Exec=xarchiver
 Icon=xarchiver
 Terminal=false
