diff -Naur xarchiver-0.5.2/src/window.c xarchiver-0.5.2.tpg/src/window.c
--- xarchiver-0.5.2/src/window.c	2008-11-11 08:31:45.000000000 +0000
+++ xarchiver-0.5.2.tpg/src/window.c	2009-03-08 21:15:36.000000000 +0000
@@ -231,9 +231,9 @@
 {
 	int response;
 
-	dialog = gtk_message_dialog_new (window,mode,type,button,message1);
+	dialog = gtk_message_dialog_new (window,mode,type,button,message1, "%s");
 	gtk_dialog_set_default_response (GTK_DIALOG (dialog),GTK_RESPONSE_NO);
-	gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (dialog),message2);
+	gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (dialog),message2, "%s");
 	response = gtk_dialog_run (GTK_DIALOG (dialog));
 	gtk_widget_destroy (GTK_WIDGET (dialog));
 	return response;
@@ -509,7 +509,7 @@
 			g_fprintf (stream,_("Comment:\n"));
 			if (bp)
 				g_fprintf(stream,"</b><pre>");
-			g_fprintf (stream,archive[idx]->comment->str);
+			g_fprintf (stream,archive[idx]->comment->str, "%s");
 			if (bp)
 				g_fprintf(stream,"</pre>");
 			g_fprintf (stream,"\n");
