diff -Naur Pyro-3.7.old/bin/pyro-nsd Pyro-3.7/bin/pyro-nsd
--- Pyro-3.7.old/bin/pyro-nsd	2008-02-13 11:00:57.946239290 +0000
+++ Pyro-3.7/bin/pyro-nsd	2008-02-13 11:04:24.484840650 +0000
@@ -17,11 +17,11 @@
 from Pyro.ext.daemonizer import Daemonizer
 
 class NSD(Daemonizer):
-    def __init__(self):
-        Daemonizer.__init__(self)
+    def __init__(self, pidfile):
+        Daemonizer.__init__(self, pidfile)
 
     def main_loop(self):
         Pyro.naming.main(sys.argv[1:])
 
 if __name__ == "__main__":
-    NSD().process_command_line(sys.argv)
+    NSD("/run/pyro-nsd.pid").process_command_line(sys.argv)
