Index: pinentry-0.7.6/qt/main.cpp
===================================================================
--- pinentry-0.7.6.orig/qt/main.cpp
+++ pinentry-0.7.6/qt/main.cpp
@@ -71,23 +71,23 @@ qt_cmd_handler (pinentry_t pe)
       PinEntryDialog pinentry (parent, NULL, true, !!pe->quality_bar);
 
       pinentry.setPinentryInfo (pe);
-      pinentry.setPrompt (QString::fromUtf8 (pe->prompt));
-      pinentry.setDescription (QString::fromUtf8 (pe->description));
+      pinentry.setPrompt (QString::fromLocal8Bit (pe->prompt));
+      pinentry.setDescription (QString::fromLocal8Bit (pe->description));
       /* If we reuse the same dialog window.  */
 #if 0
       pinentry.setText (SecQString::null);
 #endif
 
       if (pe->ok)
-	pinentry.setOkText (QString::fromUtf8 (pe->ok));
+	pinentry.setOkText (QString::fromLocal8Bit (pe->ok));
       if (pe->cancel)
-	pinentry.setCancelText (QString::fromUtf8 (pe->cancel));
+	pinentry.setCancelText (QString::fromLocal8Bit (pe->cancel));
       if (pe->error)
-	pinentry.setError (QString::fromUtf8 (pe->error));
+	pinentry.setError (QString::fromLocal8Bit (pe->error));
       if (pe->quality_bar)
-	pinentry.setQualityBar (QString::fromUtf8 (pe->quality_bar));
+	pinentry.setQualityBar (QString::fromLocal8Bit (pe->quality_bar));
       if (pe->quality_bar_tt)
-	pinentry.setQualityBarTT (QString::fromUtf8 (pe->quality_bar_tt));
+	pinentry.setQualityBarTT (QString::fromLocal8Bit (pe->quality_bar_tt));
 
       bool ret = pinentry.exec ();
       if (!ret)
@@ -113,9 +113,9 @@ qt_cmd_handler (pinentry_t pe)
     }
   else
     {
-      QString desc = QString::fromUtf8 (pe->description? pe->description :"");
-      QString ok   = QString::fromUtf8 (pe->ok ? pe->ok : "OK");
-      QString can  = QString::fromUtf8 (pe->cancel ? pe->cancel : "Cancel");
+      QString desc = QString::fromLocal8Bit (pe->description? pe->description :"");
+      QString ok   = QString::fromLocal8Bit (pe->ok ? pe->ok : "OK");
+      QString can  = QString::fromLocal8Bit (pe->cancel ? pe->cancel : "Cancel");
       bool ret;
       
       ret = QMessageBox::information (parent, "", desc, ok, can );
