--- pam_cap/Makefile	2008-10-17 06:31:01.000000000 +0200
+++ pam_cap/Makefile.oden	2008-12-18 12:48:41.000000000 +0100
@@ -1,6 +1,6 @@
 # simple make file for the pam_cap module
 
-topdir=$(shell pwd)/..
+topdir=..
 include ../Make.Rules
 
 LDLIBS += -lcap
@@ -14,7 +14,7 @@
 	install -m 0755 pam_cap.so $(LIBDIR)/security
 
 pam_cap.so: pam_cap.o
-	$(LD) $(LDFLAGS) -o pam_cap.so $< $(LDLIBS)
+	$(LD) $(LDFLAGS) -o pam_cap.so $< $(LDLIBS) -lpam 
 
 pam_cap.o: pam_cap.c
 	$(CC) $(CFLAGS) -c $< -o $@
