--- htdig-3.2.0b4-011302/db/hash.c.h_hash	Tue Oct 10 05:15:27 2000
+++ htdig-3.2.0b4-011302/db/hash.c	Fri Jan 25 15:56:09 2002
@@ -245,6 +245,11 @@
 		need_sync = 1;
 	}
 
+        /* Make sure we always have a valid hashp->h_hash function. */
+	if (hashp->h_hash == NULL)
+		hashp->h_hash = hcp->hdr->dbmeta.version < 5
+		? CDB___ham_func4 : CDB___ham_func5;
+
 err2:	/* Release the meta data page */
 	if ((t_ret = CDB___ham_release_meta(dbc)) != 0 && ret == 0)
 		ret = t_ret;
