diff -Nur secure_delete-3.1-old//Makefile secure_delete-3.1/Makefile
--- secure_delete-3.1-old//Makefile	2011-06-17 12:22:25.383001395 +0300
+++ secure_delete-3.1/Makefile	2011-06-17 12:22:19.135001323 +0300
@@ -21,13 +21,13 @@
 sdel-lib.o: sdel-lib.c
 	$(CC) ${OPT} -c sdel-lib.c
 
-srm: srm.c
+srm: srm.c sdel-lib.o
 	$(CC) ${OPT} -o srm srm.c sdel-lib.o
-sfill: sfill.c
+sfill: sfill.c sdel-lib.o
 	$(CC) ${OPT} -o sfill sfill.c sdel-lib.o
-sswap: sswap.c
+sswap: sswap.c sdel-lib.o
 	$(CC) ${OPT} -o sswap sswap.c sdel-lib.o
-smem: smem.c
+smem: smem.c sdel-lib.o
 	$(CC) ${OPT} -o smem smem.c sdel-lib.o
 
 clean:
