diff -Nur kmplayer-0.11.2b-old/src/kmplayerprocess.cpp kmplayer-0.11.2b/src/kmplayerprocess.cpp
--- kmplayer-0.11.2b-old/src/kmplayerprocess.cpp	2010-04-30 12:03:57.724911912 +0300
+++ kmplayer-0.11.2b/src/kmplayerprocess.cpp	2010-04-30 12:04:43.718911875 +0300
@@ -543,6 +543,9 @@
     if (exe.isEmpty ())
         exe = "mplayer";
 
+    args << "-nomsgmodule";
+    args << "-nomsgcolor";
+
     args << "-wid" << QString::number (widget ());
     args << "-slave";
 
