diff -Nur kmplayer-0.11.2b-old/src/kmplayerconfig.cpp kmplayer-0.11.2b/src/kmplayerconfig.cpp
--- kmplayer-0.11.2b-old/src/kmplayerconfig.cpp	2010-04-30 12:03:57.723911222 +0300
+++ kmplayer-0.11.2b/src/kmplayerconfig.cpp	2010-04-30 15:43:02.508661597 +0300
@@ -58,7 +58,7 @@
 using namespace KMPlayer;
 
 static OutputDriver _ads[] = {
-    { "alsa,oss,sdl,arts", i18n ("Auto") },
+    { "", i18n ("Auto") },
     { "oss", i18n ("Open Sound System") },
     { "sdl", i18n ("Simple DirectMedia Layer") },
     { "alsa", i18n ("Advanced Linux Sound Architecture") },
@@ -74,7 +74,7 @@
 };
 
 static OutputDriver _vds [] = {
-    { "xv,sdl,x11", i18n ("Auto") },
+    { "", i18n ("Auto") },
     { "x11", i18n ("X11Shm") },
     { "xvidix", i18n ("XVidix") },
     { "xvmc,xv", i18n ("XvMC") },
