diff -Nur spatialindex-src-1.8.0/src/CMakeLists.txt spatialindex-src-1.8.0-soname/src/CMakeLists.txt
--- src/CMakeLists.txt	2012-10-24 22:08:28.000000000 +0200
+++ src/CMakeLists.txt	2012-12-13 20:47:07.668114280 +0100
@@ -203,12 +203,13 @@
 )
 
 
+#TODO: Major-Version wo definieren
 
 set_target_properties(${SIDX_LIB_NAME}
-  PROPERTIES SOVERSION "${SIDX_LIB_SOVERSION}" )
+    PROPERTIES VERSION "${SIDX_LIB_SOVERSION}" SOVERSION 3)
 
 set_target_properties(${SIDX_C_LIB_NAME} 
-  PROPERTIES SOVERSION "${SIDX_LIB_SOVERSION}" )
+    PROPERTIES VERSION "${SIDX_LIB_SOVERSION}" SOVERSION 3)
 
 
 
