Index: etherape-0.9.12/etherape.desktop.in
===================================================================
--- etherape-0.9.12.orig/etherape.desktop.in
+++ etherape-0.9.12/etherape.desktop.in
@@ -2,11 +2,13 @@
 Version=1.0
 Encoding=UTF-8
 Name=EtherApe
+Name[tr]=EtherApe
 GenericName=EtherApe
 Comment=Graphical Network Monitor
 Comment[es]=Monitor Gráfico de Red
-Exec=etherape
+Comment[tr]=Grafiksel Ağ İzleyici
+Exec=xdg-su -c "etherape"
 Terminal=false
 Type=Application
 Icon=etherape
-Categories=GNOME;System;Network;
+Categories=System;Monitor;Network;
