diff -up marble-4.8.0/src/CMakeLists.txt.opt marble-4.8.0/src/CMakeLists.txt
--- marble-4.8.0/src/CMakeLists.txt.opt	2011-12-20 13:19:11.000000000 -0600
+++ marble-4.8.0/src/CMakeLists.txt	2012-02-03 20:33:12.895703342 -0600
@@ -91,10 +91,8 @@ ENDMACRO (QT4_ADD_RESOURCES2)
 ENDIF(MINGW)
 
 if (QTONLY)
-  # ce: don't know why this is needed here - on win32 
-  # 'O2' is activated by default in release mode
   if(NOT WIN32)
-    add_definitions (${QT_DEFINITIONS} -O3)
+    add_definitions (${QT_DEFINITIONS})
   endif(NOT WIN32)
 
   include_directories(
@@ -102,7 +100,7 @@ if (QTONLY)
   )
 else (QTONLY)
   if(NOT WIN32)
-    add_definitions (${QT_DEFINITIONS} ${KDE4_DEFINITIONS} -O3) 
+    add_definitions (${QT_DEFINITIONS} ${KDE4_DEFINITIONS}) 
   endif(NOT WIN32)
 
   include_directories(
diff -up marble-4.8.0/src/lib/CMakeLists.txt.opt marble-4.8.0/src/lib/CMakeLists.txt
--- marble-4.8.0/src/lib/CMakeLists.txt.opt	2011-12-20 13:19:11.000000000 -0600
+++ marble-4.8.0/src/lib/CMakeLists.txt	2012-02-03 20:33:40.822354206 -0600
@@ -31,9 +31,8 @@ set(GENERIC_LIB_VERSION "0.13.0")
 set(GENERIC_LIB_SOVERSION "13")
 
 if (QTONLY)
-  # ce: don't know why this is needed here - on win32 'O2' is activated by default in release mode
   if(NOT WIN32)
-    add_definitions (${QT_DEFINITIONS} -O3)
+    add_definitions (${QT_DEFINITIONS})
   endif(NOT WIN32)
 
   include_directories(
@@ -41,7 +40,7 @@ if (QTONLY)
   )
 else (QTONLY)
   if(NOT WIN32)
-    add_definitions (${QT_DEFINITIONS} ${KDE4_DEFINITIONS} -O3)
+    add_definitions (${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
   endif(NOT WIN32)
 
   include_directories(
diff -up marble-4.8.0/tests/CMakeLists.txt.opt marble-4.8.0/tests/CMakeLists.txt
--- marble-4.8.0/tests/CMakeLists.txt.opt	2011-12-20 13:19:11.000000000 -0600
+++ marble-4.8.0/tests/CMakeLists.txt	2012-02-03 20:32:54.705930743 -0600
@@ -71,10 +71,8 @@ endif( CMAKE_COMMAND AND CMAKE_CTEST_COM
 
 # GeoPolygon test
 if (QTONLY)
-  # ce: don't know why this is needed here - on win32 
-  # 'O2' is activated by default in release mode
   if(NOT WIN32)
-    add_definitions (${QT_DEFINITIONS} -O3)
+    add_definitions (${QT_DEFINITIONS})
   endif(NOT WIN32)
 
   include_directories(
@@ -82,7 +80,7 @@ if (QTONLY)
   )
 else (QTONLY)
   if(NOT WIN32)
-    add_definitions (${QT_DEFINITIONS} ${KDE4_DEFINITIONS} -O3) 
+    add_definitions (${QT_DEFINITIONS} ${KDE4_DEFINITIONS}) 
   endif(NOT WIN32)
 
   include_directories(
