Index: trigger-0.5.2.1-src/src/pengine/terrain.cpp
===================================================================
--- trigger-0.5.2.1-src.orig/src/pengine/terrain.cpp
+++ trigger-0.5.2.1-src/src/pengine/terrain.cpp
@@ -561,7 +561,7 @@ void PTerrain::render(const vec3f &campo
 		ind.bind();
 		glVertexPointer(3, GL_FLOAT, sizeof(vec3f), (*t)->vert.getPointer(0));
 
-		glDrawRangeElements(GL_TRIANGLE_STRIP, 0, (*t)->numverts,
+		glDrawRangeElements(GL_TRIANGLE_STRIP, 0, (*t)->numverts - 1,
 			numinds, GL_UNSIGNED_SHORT, ind.getPointer(0));
 	}
 
@@ -595,7 +595,7 @@ void PTerrain::render(const vec3f &campo
 				glVertexPointer(3, GL_FLOAT, sizeof(PVert_tv), (*t)->foliage[b].buff[0].getPointer(sizeof(float)*2));
 				
 				glDrawRangeElements(GL_TRIANGLES,
-					0,(*t)->foliage[b].numvert,(*t)->foliage[b].numelem,
+					0,(*t)->foliage[b].numvert - 1,(*t)->foliage[b].numelem,
 					GL_UNSIGNED_INT,(*t)->foliage[b].buff[1].getPointer(0));
 			}
 			
