Index: cuneiform-linux-1.1.0/install_files.cmake
===================================================================
--- cuneiform-linux-1.1.0.orig/install_files.cmake
+++ cuneiform-linux-1.1.0/install_files.cmake
@@ -4,11 +4,7 @@ set(INSTALL_DATADIR "${CMAKE_INSTALL_PRE
 
 include(CheckTypeSize)
 check_type_size("void*" PTRSIZE)
-if(PTRSIZE EQUAL 8 AND CMAKE_SYSTEM_NAME STREQUAL "Linux")
-    set(LIBDIR "lib64")
-else()
-    set(LIBDIR "lib")
-endif()
+set(LIBDIR "lib")
 
 # Library versioning done as instructed in
 # http://techbase.kde.org/index.php?title=Development/Tutorials/Creating_Libraries
