Index: kdebase-workspace-4.3.90/plasma/desktop/applets/kickoff/ui/launcher.cpp
===================================================================
--- kdebase-workspace-4.3.90.orig/plasma/desktop/applets/kickoff/ui/launcher.cpp	2010-01-06 20:08:40.378006494 -0500
+++ kdebase-workspace-4.3.90/plasma/desktop/applets/kickoff/ui/launcher.cpp	2010-01-06 20:08:41.070005684 -0500
@@ -26,6 +26,8 @@
 
 // Qt
 #include <QApplication>
+#include <QDir>
+#include <QFileInfo>
 #include <QKeyEvent>
 #include <QLabel>
 #include <QMouseEvent>
@@ -37,6 +39,7 @@
 // KDE
 #include <KDebug>
 #include <KIcon>
+#include <KIconLoader>
 #include <kuser.h>
 #include <Plasma/Theme>
 #include <Plasma/Delegate>
@@ -515,6 +518,14 @@
     // Add status information footer
     d->footer = new QWidget;
 
+    QLabel *userAvatar = new QLabel(this);
+    const QFileInfo fi(QDir(QDir::homePath()), ".face.icon");
+    if (fi.exists()) {
+        userAvatar->setPixmap(QPixmap(fi.absoluteFilePath()).scaled(KIconLoader::SizeMedium, KIconLoader::SizeMedium, Qt::KeepAspectRatio));
+    } else {
+        userAvatar->setPixmap(KIcon("user-identity").pixmap(KIconLoader::SizeMedium, KIconLoader::SizeMedium));
+    }
+
     char hostname[256];
     hostname[0] = '\0';
     if (!gethostname(hostname, sizeof(hostname))) {
@@ -538,11 +549,10 @@
 
     QHBoxLayout *brandingLayout = new QHBoxLayout;
     brandingLayout->setMargin(3);
-    brandingLayout->addSpacing(ItemDelegate::ITEM_LEFT_MARGIN - 3);
+    brandingLayout->addWidget(userAvatar);
     brandingLayout->addWidget(d->userinfo);
     brandingLayout->addStretch(2);
     brandingLayout->addWidget(branding);
-    brandingLayout->addSpacing(rightHeaderMargin);
     d->footer->setLayout(brandingLayout);
 
     layout->addWidget(d->footer);
Index: kdebase-workspace-4.3.90/plasma/desktop/applets/kickoff/ui/searchbar.cpp
===================================================================
--- kdebase-workspace-4.3.90.orig/plasma/desktop/applets/kickoff/ui/searchbar.cpp	2009-10-15 12:33:38.000000000 -0400
+++ kdebase-workspace-4.3.90/plasma/desktop/applets/kickoff/ui/searchbar.cpp	2010-01-06 20:08:41.070005684 -0500
@@ -22,8 +22,6 @@
 
 // Qt
 #include <QCoreApplication>
-#include <QDir>
-#include <QFileInfo>
 #include <QHBoxLayout>
 #include <QKeyEvent>
 #include <QLabel>
@@ -70,12 +68,7 @@
     d->searchLabel = new QLabel(i18nc("Label of the search bar textedit", "Search:"), this);
     QLabel *searchIcon = new QLabel(this);
 
-    const QFileInfo fi(QDir(QDir::homePath()), ".face.icon");
-    if (fi.exists()) {
-        searchIcon->setPixmap(QPixmap(fi.absoluteFilePath()).scaled(KIconLoader::SizeMedium, KIconLoader::SizeMedium, Qt::KeepAspectRatio));
-    } else {
-        searchIcon->setPixmap(KIcon("system-search").pixmap(KIconLoader::SizeMedium, KIconLoader::SizeMedium));
-    }
+    searchIcon->setPixmap(KIcon("system-search").pixmap(KIconLoader::SizeSmall, KIconLoader::SizeSmall));
 
     d->editWidget = new KLineEdit(this);
     d->editWidget->installEventFilter(this);
