Index: kdebase-workspace-4.5.80/krunner/screensaver/saverengine.cpp
===================================================================
--- kdebase-workspace-4.5.80.orig/krunner/screensaver/saverengine.cpp	2010-10-27 19:54:59.000000000 +0000
+++ kdebase-workspace-4.5.80/krunner/screensaver/saverengine.cpp	2010-11-24 11:53:49.516641003 +0000
@@ -206,7 +206,7 @@
     if (mState != Waiting)
         return false;
 
-    if (e)
+    if (e && KScreenSaverSettings::screenSaverEnabled() )
     {
         if (!mXAutoLock)
         {
