Index: kdebase-workspace-4.3.90/startkde.cmake
===================================================================
--- kdebase-workspace-4.3.90.orig/startkde.cmake	2009-10-06 17:37:24.000000000 -0400
+++ kdebase-workspace-4.3.90/startkde.cmake	2010-01-06 20:10:08.018004945 -0500
@@ -88,6 +88,35 @@
 kdehome=$HOME/@KDE_DEFAULT_HOME@
 test -n "$KDEHOME" && kdehome=`echo "$KDEHOME"|sed "s,^~/,$HOME/,"`
 
+# Read LANG and write the country part to kdeglobals
+# de_AT.UTF-8 => Country=at
+# This is necessary because klocale.cpp (kdelibs) does only try to read the value
+# from kdeglobals and then falls back to default instead of triyng to detect it
+
+if [ ! -e $kdehome/share/config/kdeglobals ] || [ ! `grep "Country=" $kdehome/share/config/kdeglobals` ]; then
+  if [ ! $LANG ]; then
+    if [ -e /etc/mudur/locale ]; then
+      locale=`cat /etc/mudur/locale`
+    elif [ -e /etc/default/locale ]; then
+      localefile='/etc/default/locale'
+    fi
+    if [ $localefile ]; then
+      locale=`sed -ne 's/^LANG="\(.*\)"$/\1/p' $localefile`
+    fi
+  else
+    locale=$LANG
+  fi
+  if [ $locale ]; then
+    country=`echo $locale | sed -ne 's/^[A-Za-z]*_\([A-Za-z]*\).*$/\1/p'|tr 'A-Z' 'a-z'`
+    if [ ! $country = "" ]; then
+cat >>$kdehome/share/config/kdeglobals <<EOF
+[Locale]
+Country=$country
+EOF
+    fi
+  fi
+fi
+
 # see kstartupconfig source for usage
 mkdir -m 700 -p $kdehome
 mkdir -m 700 -p $kdehome/share
@@ -104,6 +133,7 @@
 kcmrandrrc [Screen3]
 kcmfonts General forceFontDPI 0
 kdeglobals Locale Language '' # trigger requesting languages from KLocale
+kdeglobals Locale Country ''
 EOF
 kstartupconfig4
 returncode=$?
