--- data/gajim.desktop.in.in.orig	2008-12-21 19:33:04.000000000 +0200
+++ data/gajim.desktop.in.in	2008-12-21 19:38:15.000000000 +0200
@@ -1,8 +1,11 @@
 [Desktop Entry]
 Categories=Network;InstantMessaging;GTK;GNOME;
-_Name=Gajim Instant Messenger
+_Name=Gajim
+Name[tr]=Gajim
 _GenericName=Jabber IM Client
+GenericName[tr]=Jabber İstemcisi
 _Comment=A GTK+ Jabber client
+Comment[tr]=GTK+ tabanlı Jabber istemcisi
 Version=1.0
 Exec=gajim
 Icon=gajim
