diff -Naur --exclude '*.orig' bash-4.0/config-top.h bash-4.0-profile/config-top.h
--- bash-4.0/config-top.h	2009-01-04 14:32:23.000000000 -0500
+++ bash-4.0-profile/config-top.h	2009-02-24 13:13:23.000000000 -0500
@@ -82,7 +82,7 @@
 
 /* Define this to make non-interactive shells begun with argv[0][0] == '-'
    run the startup files when not in posix mode. */
-/* #define NON_INTERACTIVE_LOGIN_SHELLS */
+#define NON_INTERACTIVE_LOGIN_SHELLS
 
 /* Define this if you want bash to try to check whether it's being run by
    sshd and source the .bashrc if so (like the rshd behavior).  This checks
