--- bash-3.2/examples/complete/complete-examples~	2002-03-13 14:41:54.000000000 +0100
+++ bash-3.2/examples/complete/complete-examples	2007-07-11 10:50:06.000000000 +0200
@@ -463,10 +463,12 @@ complete -f strip
 
 complete -f -X '*.gz' gzip
 complete -f -X '*.bz2' bzip2
+complete -f -X '*.lzma' lzma
 complete -f -X '*.Z' compress
 complete -f -X '!*.+(gz|tgz|Gz)' gunzip gzcat zcat zmore
 complete -f -X '!*.Z' uncompress zmore zcat
 complete -f -X '!*.bz2' bunzip2 bzcat
+complete -f -X '!*.lzma' unlzma lzcat
 complete -f -X '!*.zip' unzip
 complete -f -X '!*.+(gif|jpg|jpeg|GIF|JPG|JPEG|bmp)' xv
 
