diff -p -up bash-3.2/print_cmd.c.format-security bash-3.2/print_cmd.c
--- bash-3.2/print_cmd.c.format-security	2008-12-16 18:35:32.000000000 +0100
+++ bash-3.2/print_cmd.c	2008-12-16 18:37:30.000000000 +0100
@@ -1149,7 +1149,7 @@ indent (amount)
   for (i = 0; amount > 0; amount--)
     indentation_string[i++] = ' ';
   indentation_string[i] = '\0';
-  cprintf (indentation_string);
+  cprintf ("%s", indentation_string);
 }
 
 static void
