*** parse.y
--- parse.y	Tue Jan 10 11:36:59 2006
***************
*** 2842,2848 ****
  		  ttrans = ansiexpand (nestret, 0, nestlen - 1, &ttranslen);
  		  xfree (nestret);
  
! 		  if ((rflags & P_DQUOTE) == 0)
  		    {
  		      nestret = sh_single_quote (ttrans);
  		      free (ttrans);
--- 2842,2848 ----
  		  ttrans = ansiexpand (nestret, 0, nestlen - 1, &ttranslen);
  		  xfree (nestret);
  
! 		  if (extended_quote || (rflags & P_DQUOTE) == 0)
  		    {
  		      nestret = sh_single_quote (ttrans);
  		      free (ttrans);
