--- makefiles/makefile.linux.orig	2005-01-30 11:28:00.000000000 +0100
+++ makefiles/makefile.linux		2007-07-19 13:39:28.000000000 +0200
@@ -11,7 +11,7 @@
 
 # default build options (this forces shared library compatibility!!)
 #OPTIONS = -DMNG_BUILD_SO
-OPTIONS = -DMNG_BUILD_SO -DMNG_FULL_CMS
+OPTIONS = -DMNG_BUILD_SO -DMNG_NO_INCLUDE_JNG
 
 # where "make install" puts libmng.a,libmng.so*,libmng.h,libmng_conf.h,libmng_types.h
 prefix=/usr/local
@@ -50,7 +50,7 @@
 	-L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) \
 	-L$(JPEGLIB) -Wl,-rpath,$(JPEGLIB) \
 	-L$(LCMSLIB) -Wl,-rpath,$(LCMSLIB) \
-	-lmng -lz -ljpeg -llcms -lm
+	-lmng -lz -lm
 
 RANLIB=ranlib
 #RANLIB=echo
@@ -107,7 +107,7 @@
 #	$(CC) -shared -Wl,-soname,libmng.so.$(MNGMAJ) -o libmng.so.$(MNGVER) \
 #	$(OBJSDLL) -L$(ZLIBLIB) -L$(JPEGLIB) -L$(LCMSLIB) -lz -lm -lc
 	$(CC) -shared -Wl,-soname,libmng.so.$(MNGMAJ) -o libmng.so.$(MNGVER) \
-	$(OBJSDLL) -L$(ZLIBLIB) -L$(JPEGLIB) -ljpeg -L$(LCMSLIB) -llcms \
+	$(OBJSDLL) -L$(ZLIBLIB) \
     -lz -lm -lc
 
 install: libmng.a libmng.so.$(MNGVER)
