--- makefiles/makefile.linux	2005-01-30 11:28:00.000000000 +0100
+++ makefiles/makefile.linux	2007-07-19 13:35:44.000000000 +0200
@@ -14,25 +14,25 @@
 OPTIONS = -DMNG_BUILD_SO -DMNG_FULL_CMS
 
 # where "make install" puts libmng.a,libmng.so*,libmng.h,libmng_conf.h,libmng_types.h
-prefix=/usr/local
+prefix=/usr
 
 # Where the zlib library and include files are located
 #ZLIBLIB=../zlib
 #ZLIBINC=../zlib
-ZLIBLIB=/usr/local/lib
-ZLIBINC=/usr/local/include
+ZLIBLIB=/usr/$(_LIB)
+ZLIBINC=/usr/include
 
 # Where the jpeg library and include files are located
 #JPEGLIB=../jpgsrc
 #JPEGINC=../jpgsrc
-JPEGLIB=/usr/local/lib
-JPEGINC=/usr/local/include
+JPEGLIB=/usr/$(_LIB)
+JPEGINC=/usr/include
 
 # Where the lcms library and include files are located
 #LCMSLIB=../lcms/lib
 #LCMSINC=../lcms/source
-LCMSLIB=/usr/local/lib
-LCMSINC=/usr/local/include
+LCMSLIB=/usr/$(_LIB)
+LCMSINC=/usr/include
 
 ALIGN=
 # for i386:
@@ -45,7 +45,7 @@
 # for pgcc version 2.95.1, -O3 is buggy; don't use it.
 
 CFLAGS=-I$(ZLIBINC) -I$(JPEGINC) -I$(LCMSINC) -Wall -O3 -funroll-loops \
-	$(OPTIONS) $(ALIGN) # $(WARNMORE) -g
+	$(OPTIONS) $(ALIGN) $(RPM_OPT_FLAGS) -fPIC # $(WARNMORE) -g
 LDFLAGS=-L. -Wl,-rpath,. \
 	-L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) \
 	-L$(JPEGLIB) -Wl,-rpath,$(JPEGLIB) \
@@ -57,11 +57,11 @@
 
 # current version numbers
 MNGMAJ = 1
-MNGMIN = 1.0.9
+MNGMIN = 1.0.10
 MNGVER = $(MNGMAJ).$(MNGMIN)
 
 INCPATH=$(prefix)/include
-LIBPATH=$(prefix)/lib
+LIBPATH=$(prefix)/$(_LIB)
 
 OBJS = \
 	libmng_callback_xs.o \
@@ -111,13 +111,11 @@
     -lz -lm -lc
 
 install: libmng.a libmng.so.$(MNGVER)
-	-@mkdir $(INCPATH) $(LIBPATH)
-	cp libmng.h libmng_conf.h libmng_types.h $(INCPATH)
-	chmod 644 $(INCPATH)/libmng.h $(INCPATH)/libmng_conf.h $(INCPATH)/libmng_types.h
-	cp libmng.a libmng.so.$(MNGVER) $(LIBPATH)
-	chmod 755 $(LIBPATH)/libmng.so.$(MNGVER)
-	-@/bin/rm -f $(LIBPATH)/libmng.so.$(MNGMAJ) $(LIBPATH)/libmng.so
-	(cd $(LIBPATH); ln -sf libmng.so.$(MNGVER) libmng.so.$(MNGMAJ); \
+	-@mkdir -p $(DESTDIR)$(INCPATH) $(DESTDIR)$(LIBPATH)
+	install -m 0644 libmng.h libmng_conf.h libmng_types.h $(DESTDIR)$(INCPATH)
+	install -m 0755 libmng.a libmng.so.$(MNGVER) $(DESTDIR)$(LIBPATH)
+	-@/bin/rm -f $(DESTDIR)$(LIBPATH)/libmng.so.$(MNGMAJ) $(DESTDIR)$(LIBPATH)/libmng.so
+	(cd $(DESTDIR)$(LIBPATH); ln -sf libmng.so.$(MNGVER) libmng.so.$(MNGMAJ); \
 	 ln -sf libmng.so.$(MNGMAJ) libmng.so)
 
 clean:
