diff -Nur rpm2targz-9.0.0.4g-old/Makefile rpm2targz-9.0.0.4g/Makefile
--- rpm2targz-9.0.0.4g-old/Makefile	2010-08-05 09:24:46.756543758 +0300
+++ rpm2targz-9.0.0.4g/Makefile	2010-08-05 09:26:03.828420059 +0300
@@ -30,7 +30,7 @@
 	$(dodir) $(DESTDIR)$(bindir)
 	$(dobin) rpm2targz rpmoffset $(DESTDIR)$(bindir)
 	set -e ; \
-	for t in tar tarbz2 tbz2 tarlzma tgz tarxz txz ; do \
+	for t in tar tarbz2 tarlzma tarxz; do \
 		$(dosym) rpm2targz $(DESTDIR)$(bindir)/rpm2$$t ; \
 	done
 	$(dosym) rpm2targz $(DESTDIR)$(bindir)/rpmunpack
