--- hacks/fireworkx.man
+++ hacks/fireworkx.man
@@ -8,8 +8,8 @@
 [\-window]
 [\-root]
 [\-verbose]
-[\-noflash]
-[\-noglow]
+[\-no-flash]
+[\-no-glow]
 [\-shoot]
 [\-delay \fInumber\fP]
 [\-maxlife \fInumber\fP]
@@ -28,10 +28,10 @@
 .B \-root
 Draw on the root window.
 .TP 8
-.B \-noflash 
+.B \-no-flash 
 Turn off light effect (runs faster)
 .TP 8
-.B \-noglow
+.B \-no-glow
 Turn off self glow (dull blasts..!)
 .TP 8
 .B \-shoot
