Index: driver/timers.c
===================================================================
--- driver/timers.c.orig	2009-12-24 17:06:04.000000000 +0100
+++ driver/timers.c	2009-12-24 17:06:20.000000000 +0100
@@ -950,17 +950,17 @@ sleep_until_idle (saver_info *si, Bool u
 	 * screen)
 	 */
 	  case MapNotify:
-	if( (si->screen_blanked_p) && event.xmap.override_redirect ) 
+	if( (si->screen_blanked_p) && event.x_event.xmap.override_redirect ) 
 	  {
 		int ii,rr=TRUE;
 		for (ii = 0; ii < si->nscreens; ii++) {
-			if (si->screens[ii].screensaver_window==event.xmap.window) {
+			if (si->screens[ii].screensaver_window==event.x_event.xmap.window) {
 				rr=FALSE;
 				break;
 			}
 		}
 		if (p->debug_p) {
-			fprintf (stderr, "MapNotify w/ override redirect from 0x%x received",event.xmap.window);
+			fprintf (stderr, "MapNotify w/ override redirect from 0x%x received",event.x_event.xmap.window);
 		}
 		if(rr==TRUE) { 
 			if (p->debug_p) { fprintf (stderr, ", raising unconditionally.\n"); }
@@ -974,17 +974,17 @@ sleep_until_idle (saver_info *si, Bool u
 	break;
 
 	  case ConfigureNotify:
-	if( (si->screen_blanked_p) && event.xconfigure.override_redirect ) 
+	if( (si->screen_blanked_p) && event.x_event.xconfigure.override_redirect ) 
 	  {
 		int ii,rr=TRUE;
 		for (ii = 0; ii < si->nscreens; ii++) {
-			if(si->screens[ii].screensaver_window==event.xconfigure.window) {
+			if(si->screens[ii].screensaver_window==event.x_event.xconfigure.window) {
 				rr=FALSE;
 				break;
 			}
 		}
 		if (p->debug_p) {
-			fprintf (stderr, "ConfigureNotify event w/ override redirect from 0x%x received",event.xconfigure.window);
+			fprintf (stderr, "ConfigureNotify event w/ override redirect from 0x%x received",event.x_event.xconfigure.window);
 		}
 		if(rr==TRUE) { 
 			if (p->debug_p) { fprintf (stderr, ", raising unconditionally.\n"); }
