--- configure.in
+++ configure.in
@@ -2266,12 +2266,7 @@
       # absolute path
       set dummy $with_login_manager_req ; login_manager_tmp=$2
       AC_MSG_CHECKING([for $login_manager_tmp])
-      if test -x "$login_manager_tmp" ; then
-        AC_MSG_RESULT(yes)
-      else
-        AC_MSG_RESULT(no)
-        with_login_manager=""
-      fi
+      AC_MSG_RESULT(yes)
     ;;
     *)
       # relative path
