Index: driver/XScreenSaver.ad.in
===================================================================
--- driver/XScreenSaver.ad.in.orig
+++ driver/XScreenSaver.ad.in
@@ -404,7 +404,7 @@ XScreenSaver.bourneShell:		/bin/sh
 *passwd.login.label:		New Login
 *passwd.user.label:		Username:
 *passwd.thermometer.width:	8
-*passwd.asterisks:              True
+*passwd.asterisks:              False
 *passwd.uname:                  True
 
 *splash.heading.label:		XScreenSaver %s
Index: driver/XScreenSaver_ad.h
===================================================================
--- driver/XScreenSaver_ad.h.orig
+++ driver/XScreenSaver_ad.h
@@ -281,7 +281,7 @@
 "*passwd.login.label:		New Login",
 "*passwd.user.label:		Username:",
 "*passwd.thermometer.width:	8",
-"*passwd.asterisks:              True",
+"*passwd.asterisks:              False",
 "*passwd.uname:                  True",
 "*splash.heading.label:		XScreenSaver %s",
 "*splash.body.label:		Copyright © 1991-2010 by",
