change default background to black

================================================================================
--- utils/resources.c
+++ utils/resources.c
@@ -164,7 +164,7 @@
   if (s) free (s);
 
   {
-    Bool black_p = (strlen(res_class) >= 10 &&
+    Bool black_p = ! (strlen(res_class) >= 10 &&
                     !strcasecmp ("Background",
                                  res_class + strlen(res_class) - 10));
     if (!ok)
