--- hacks/Makefile.in
+++ hacks/Makefile.in
@@ -17,8 +17,8 @@
 datadir		= @datadir@
 mandir		= @mandir@
 libexecdir	= @libexecdir@
-mansuffix	= 6
-manNdir		= $(mandir)/man$(mansuffix)
+mansuffix	= 6x
+manNdir		= $(mandir)/man6
 
 HACKDIR		= @HACKDIR@
 HACK_CONF_DIR	= @HACK_CONF_DIR@
--- hacks/glx/Makefile.in
+++ hacks/glx/Makefile.in
@@ -18,8 +18,8 @@
 datadir		= @datadir@
 mandir		= @mandir@
 libexecdir	= @libexecdir@
-mansuffix	= 6
-manNdir		= $(mandir)/man$(mansuffix)
+mansuffix	= 6x
+manNdir		= $(mandir)/man6
 
 HACKDIR		= @HACKDIR@
 HACK_CONF_DIR	= @HACK_CONF_DIR@
