Index: gamin-0.1.10/python/Makefile.am
===================================================================
--- gamin-0.1.10.orig/python/Makefile.am
+++ gamin-0.1.10/python/Makefile.am
@@ -11,7 +11,7 @@ python_LTLIBRARIES = _gamin.la
 
 _gamin_la_SOURCES = gamin.c
 _gamin_la_LIBADD = $(top_builddir)/libgamin/libgamin-1.la
-_gamin_la_LDFLAGS = -module -avoid-version
+_gamin_la_LDFLAGS = -module -avoid-version `python-config --libs`
 
 python_PYTHON = gamin.py
 
