diff -Nuar qt-everywhere-opensource-src-5.3.0.orig/qtbase/mkspecs/common/g++-base.conf qt-everywhere-opensource-src-5.3.0/qtbase/mkspecs/common/g++-base.conf
--- qt-everywhere-opensource-src-5.3.0.orig/qtbase/mkspecs/common/g++-base.conf	2014-05-15 20:12:18.000000000 +0300
+++ mkspecs/common/g++-base.conf	2014-05-23 00:06:08.726631340 +0300
@@ -10,17 +10,23 @@
 
 QMAKE_COMPILER          = gcc
 
-QMAKE_CC                = gcc
+QMAKE_CC                = PISILINUX_CC
 
 QMAKE_LINK_C            = $$QMAKE_CC
 QMAKE_LINK_C_SHLIB      = $$QMAKE_CC
 
-QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO += $$QMAKE_CFLAGS_OPTIMIZE -g
+QMAKE_CFLAGS_RELEASE    += PISILINUX_CFLAGS
+QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO += PISILINUX_CFLAGS -g
 
-QMAKE_CXX               = g++
+QMAKE_CXX               = PISILINUX_CXX
 
-QMAKE_LINK              = $$QMAKE_CXX
-QMAKE_LINK_SHLIB        = $$QMAKE_CXX
+
+QMAKE_LINK     = PISILINUX_CXX
+QMAKE_LINK_SHLIB   = PISILINUX_CXX
+QMAKE_LINK_C       = PISILINUX_CC
+QMAKE_LINK_C_SHLIB = PISILINUX_CC
+
+QMAKE_LFLAGS_RELEASE    += PISILINUX_LDFLAGS
 
 QMAKE_CXXFLAGS_RELEASE_WITH_DEBUGINFO += $$QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO
 
diff -Nuar qt-everywhere-opensource-src-5.3.0.orig/qtbase/mkspecs/common/g++-unix.conf qt-everywhere-opensource-src-5.3.0/qtbase/mkspecs/common/g++-unix.conf
--- qt-everywhere-opensource-src-5.3.0.orig/qtbase/mkspecs/common/g++-unix.conf	2014-05-15 20:12:18.000000000 +0300
+++ mkspecs/common/g++-unix.conf	2014-05-23 00:06:52.586630942 +0300
@@ -10,5 +10,5 @@
 
 include(g++-base.conf)
 
-QMAKE_LFLAGS_RELEASE   += -Wl,-O1
+QMAKE_LFLAGS_RELEASE += PISILINUX_LDFLAGS
 QMAKE_LFLAGS_NOUNDEF   += -Wl,--no-undefined
diff -Nuar qt-everywhere-opensource-src-5.3.0.orig/qtbase/mkspecs/common/linux.conf qt-everywhere-opensource-src-5.3.0/qtbase/mkspecs/common/linux.conf
--- qt-everywhere-opensource-src-5.3.0.orig/qtbase/mkspecs/common/linux.conf	2014-05-15 20:12:18.000000000 +0300
+++ mkspecs/common/linux.conf	2014-05-23 00:07:32.366630581 +0300
@@ -48,7 +48,7 @@
 QMAKE_NM                = nm -P
 QMAKE_RANLIB            =
 
-QMAKE_STRIP             = strip
+QMAKE_STRIP             = 
 QMAKE_STRIPFLAGS_LIB   += --strip-unneeded
 QMAKE_INSTALL_FILE      = install -m 644 -p
 QMAKE_INSTALL_PROGRAM   = install -m 755 -p
