--- linux.conf	2014-12-05 18:24:38.000000000 +0200
+++ qtbase-opensource-src-5.4.0/mkspecs/common/linux.conf	2015-02-16 12:36:52.547052606 +0200
@@ -49,7 +49,7 @@
 QMAKE_NM                = nm -P
 QMAKE_RANLIB            =
 
-QMAKE_STRIP             = strip
+QMAKE_STRIP             = 
 QMAKE_STRIPFLAGS_LIB   += --strip-unneeded
 QMAKE_INSTALL_FILE      = install -m 644 -p
 QMAKE_INSTALL_PROGRAM   = install -m 755 -p
