diff -up qtbase-opensource-src-5.4.0/src/tools/qdoc/qdocindexfiles.cpp.QTBUG-43057 qtbase-opensource-src-5.4.0/src/tools/qdoc/qdocindexfiles.cpp
--- qtbase-opensource-src-5.4.0/src/tools/qdoc/qdocindexfiles.cpp.QTBUG-43057	2014-12-05 10:24:31.000000000 -0600
+++ src/tools/qdoc/qdocindexfiles.cpp	2014-12-17 14:47:19.393037164 -0600
@@ -1327,7 +1327,7 @@ void QDocIndexFiles::generateIndexSectio
             const InnerNode* inner = static_cast<const InnerNode*>(node);
 
             NodeList cnodes = inner->childNodes();
-            std::sort(cnodes.begin(), cnodes.end(), compareNodes);
+            qSort(cnodes.begin(), cnodes.end(), compareNodes);
 
             foreach (Node* child, cnodes) {
                 generateIndexSections(writer, child, generateInternalNodes);
 
