Index: telepathy-glib-0.11.13/telepathy-glib/telepathy-glib.pc.in
===================================================================
--- telepathy-glib-0.11.13.orig/telepathy-glib/telepathy-glib.pc.in
+++ telepathy-glib-0.11.13/telepathy-glib/telepathy-glib.pc.in
@@ -6,7 +6,7 @@ includedir=@includedir@
 Name: Telepathy-GLib
 Description: GLib utility library for the Telepathy framework
 Version: @VERSION@
-Requires: pkg-config >= 0.21
+Requires: 
 Requires.private: dbus-glib-1 >= 0.82, glib-2.0 >= 2.24, gobject-2.0 >= 2.24, gio-2.0 >= 2.24
 Libs: -L${libdir} -ltelepathy-glib
 Cflags: -I${includedir}/telepathy-1.0
Index: telepathy-glib-0.11.13/telepathy-glib/telepathy-glib-uninstalled.pc.in
===================================================================
--- telepathy-glib-0.11.13.orig/telepathy-glib/telepathy-glib-uninstalled.pc.in
+++ telepathy-glib-0.11.13/telepathy-glib/telepathy-glib-uninstalled.pc.in
@@ -6,7 +6,7 @@ abs_top_builddir=@abs_top_builddir@
 Name: Telepathy-GLib (uninstalled copy)
 Description: GLib utility library for the Telepathy framework
 Version: @VERSION@
-Requires: pkg-config >= 0.21
+Requires: 
 Requires.private: dbus-glib-1 >= 0.82, glib-2.0 >= 2.24, gobject-2.0 >= 2.24, gio-2.0 >= 2.24
 Libs: ${abs_top_builddir}/telepathy-glib/libtelepathy-glib.la
 Cflags: -I${abs_top_srcdir} -I${abs_top_builddir}
