diff -Nuar egoboo-2.8.1.OSmanOS/src/game/Makefile egoboo-2.8.1/src/game/Makefile
--- egoboo-2.8.1.OSmanOS/src/game/Makefile	2010-11-25 01:50:33.000000000 +0200
+++ egoboo-2.8.1/src/game/Makefile	2013-05-27 22:32:52.121573293 +0300
@@ -33,17 +33,17 @@
 # the compiler options
 
 CC      := gcc
-INC     := -I. -I.. -I../enet/include ${SDLCONF_I} -I./extensions -I./file_formats -I./platform
-LDFLAGS := ${SDLCONF_L} -L../enet/lib -lSDL_ttf -lSDL_mixer -lGL -lGLU -lSDL_image -lphysfs -lenet
+INC     := -I. -I..  ${SDLCONF_I} -I./extensions -I./file_formats -I./platform
+LDFLAGS := ${SDLCONF_L} -L../enet/lib -lSDL_ttf -lSDL_mixer -lGL -lGLU -lSDL_image -lphysfs -lenet -lm
 
 # use different options if the environmental variable PREFIX is defined
 ifdef ($(PREFIX),"")
 	OPT := -Os -Wall
 else
 	OPT := -Os -Wall -DPREFIX=\"${PREFIX}\" -D_NIX_PREFIX
-endif
-
-CFLAGS  := ${OPT} ${INC}
+endif
+
+CFLAGS  := ${OPT} ${INC} -DENET11
 
 #------------------------------------
 # definitions of the target projects
diff -Nuar egoboo-2.8.1.OSmanOS/src/Makefile egoboo-2.8.1/src/Makefile
--- egoboo-2.8.1.OSmanOS/src/Makefile	2010-11-30 20:05:29.000000000 +0200
+++ egoboo-2.8.1/src/Makefile	2013-05-27 22:30:40.642581356 +0300
@@ -10,11 +10,11 @@
 endif
 
 
-PROJ_NAME := egoboo-2.x
+PROJ_NAME := egoboo
 
 .PHONY: all clean
 
-all: enet egoboo
+all: egoboo
 
 clean:
 	make -C ./enet clean
@@ -25,7 +25,7 @@
 
 enet:   ./enet/lib/libenet.a
 
-egoboo: enet
+egoboo:
 	make -C ./game all PREFIX=$(PREFIX) PROJ_NAME=$(PROJ_NAME)
 	
 egoboo_lua: enet
@@ -49,13 +49,13 @@
 	#
 
 #	copy the binary to the games folder
-	mkdir -p ${PREFIX}/games
-	install -m 755 ./game/${PROJ_NAME} ${PREFIX}/games
+	mkdir -p ${PREFIX}
+	install -m 755 ./${PROJ_NAME} ${PREFIX}/
 
 #	copy the data to the games folder
-	mkdir -p ${PREFIX}/share/games/${PROJ_NAME}
-	cp -rdf ./basicdat ${PREFIX}/share/games/${PROJ_NAME}
-	cp -rdf ./modules ${PREFIX}/share/games/${PROJ_NAME}
+	mkdir -p ${PREFIX}/share/${PROJ_NAME}
+	cp -rdf ./basicdat ${PREFIX}/share/${PROJ_NAME}
+	cp -rdf ./modules ${PREFIX}/share/${PROJ_NAME}
 
 #	copy the players to the user's data folder
 	mkdir -p ${HOME}/.${PROJ_NAME}
