diff -Nuar e-module-moon-0.0.1_58042.orig//module.desktop.in e-module-moon-0.0.1_58042/module.desktop.in
--- e-module-moon-0.0.1_58042.orig//module.desktop.in	2011-02-16 22:39:52.000000000 +0200
+++ e-module-moon-0.0.1_58042/module.desktop.in	2011-03-31 15:39:42.376179790 +0300
@@ -8,6 +8,7 @@
 Name[fr]=Lune
 Name[hu]=Hold
 Nome[pt]=Lua
+Name[tr]=Ay
 GenericName=gadget
 Comment=Gadget for e17 which displays the current phase of the moon.
 Comment[ru]=Гаджет для Е17, которыей показывает текущую фазу луны.
@@ -18,5 +19,6 @@
 Comment[it]=Modulo per e17 che mostra l'attuale fase lunare.
 Comment[fr]=Ce gadget pour e17 affiche la phase lunaire actuelle.
 Comment[pt]=Um "gadget" para exibir a fase lunar
+Comment[tr]=Masaüstünde ay saatini gösteren bir modüldür.
 Icon=e-module-moon
 X-Enlightenment-ModuleType=utils
