--- efax-gtk.desktop.orig	2010-08-03 01:29:03.469379685 +0300
+++ efax-gtk.desktop	2010-08-03 01:41:18.640382194 +0300
@@ -4,12 +4,14 @@
 Name=Efax-gtk
 Name[zh_TW]=傳真
 Name[he_IL]=תוכנת פקסים efax-gtk
+Name[tr]=Efax-gtk
+GenericName[tr]=Fax istemcisi
 Comment=GUI front end for the efax fax program
 Comment[zh_TW]=efax 傳真軟體的 GUI 前端程式
 Comment[he_IL]=ממשק גרפי עבור תוכנת הפקסים efax
 TryExec=efax-gtk
 Exec=efax-gtk
-Icon=efax-gtk.png
+Icon=efax.png
 Terminal=false
 Type=Application
 Categories=Application;Office;GTK;
