Index: efax-gtk-3.2.8/efax-gtk-faxfilter/Makefile.am
===================================================================
--- efax-gtk-3.2.8.orig/efax-gtk-faxfilter/Makefile.am
+++ efax-gtk-3.2.8/efax-gtk-faxfilter/Makefile.am
@@ -7,7 +7,7 @@ spool_PROGRAMS = efax-gtk-socket-client
 efax_gtk_socket_client_SOURCES = efax-gtk-socket-client.cpp
 
 install-data-hook:
-	-chown lp $(spooldir) 2> /dev/null
-	-chgrp lp $(spooldir) 2> /dev/null
+	-chown lp $(DESTDIR)/$(spooldir) 2> /dev/null
+	-chgrp lp $(DESTDIR)/$(spooldir) 2> /dev/null
 
 EXTRA_DIST = efax-gtk-faxfilter
