Index: work/xproxymanagementprotocol-1.0.3/Makefile.am
===================================================================
--- work.orig/xproxymanagementprotocol-1.0.3/Makefile.am
+++ work/xproxymanagementprotocol-1.0.3/Makefile.am
@@ -6,7 +6,8 @@ xproxymng_HEADERS = \
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = xproxymngproto.pc
 
-dist_doc_DATA = PM_spec
+xproxymngdocdir = $(datadir)/doc/$(PACKAGE)
+xproxymngdoc_DATA = PM_spec
 
 EXTRA_DIST = xproxymngproto.pc.in
 
