Index: fpc-2.4.0/compiler/utils/samplecfg
===================================================================
--- fpc-2.4.0.orig/compiler/utils/samplecfg
+++ fpc-2.4.0/compiler/utils/samplecfg
@@ -223,7 +223,7 @@ esac
 
 
 # set right path to FPC with $fpcversion
-FPCPATH=`dirname "$1"`/\$fpcversion
+FPCPATH=/usr/lib/fpc/\$fpcversion
 
 # Write (.)fpc.cfg
 echo Writing sample configuration file to $fpccfgfile
