Index: mozilla-release/browser/locales/Makefile.in
===================================================================
--- mozilla-release.orig/browser/locales/Makefile.in
+++ mozilla-release/browser/locales/Makefile.in
@@ -215,7 +215,7 @@ endif
 clobber-zip:
 	$(RM) $(STAGEDIST)/chrome/$(AB_CD).jar \
 	  $(STAGEDIST)/chrome/$(AB_CD).manifest \
-	  $(STAGEDIST)/defaults/pref/firefox-l10n.js
+	  $(STAGEDIST)/defaults/preferences/firefox-l10n.js
 	$(RM) -rf  $(STAGEDIST)/searchplugins \
 	  $(STAGEDIST)/dictionaries \
 	  $(STAGEDIST)/hyphenation \
