diff -Nuar a/extensions/spellcheck/hunspell/src/mozHunspell.h b/extensions/spellcheck/hunspell/src/mozHunspell.h
--- a/extensions/spellcheck/hunspell/src/mozHunspell.h	2013-03-27 00:17:43.000000000 +0200
+++ b/extensions/spellcheck/hunspell/src/mozHunspell.h	2013-04-03 12:24:40.647930523 +0300
@@ -73,7 +73,7 @@
 #include "nsWeakReference.h"
 #include "nsCycleCollectionParticipant.h"
 
-#define MOZ_HUNSPELL_CONTRACTID "@mozilla.org/spellchecker/engine;1"
+#define MOZ_HUNSPELL_CONTRACTID "@mozilla.org/spellchecker/hunspell;1"
 #define MOZ_HUNSPELL_CID         \
 /* 56c778e4-1bee-45f3-a689-886692a97fe7 */   \
 { 0x56c778e4, 0x1bee, 0x45f3, \
diff -Nuar a/extensions/spellcheck/Makefile.in b/extensions/spellcheck/Makefile.in
--- a/extensions/spellcheck/Makefile.in	2013-03-27 00:17:42.000000000 +0200
+++ b/extensions/spellcheck/Makefile.in	2013-04-03 12:26:24.943926912 +0300
@@ -10,7 +10,7 @@
 include $(DEPTH)/config/autoconf.mk
 
 MODULE		= spellchecker
-DIRS		= idl locales hunspell src
+DIRS		= idl locales hunspell zemberek src
 
 TEST_DIRS += tests/chrome
 
diff -Nuar a/extensions/spellcheck/src/Makefile.in b/extensions/spellcheck/src/Makefile.in
--- a/extensions/spellcheck/src/Makefile.in	2013-03-27 00:17:43.000000000 +0200
+++ b/extensions/spellcheck/src/Makefile.in	2013-04-03 12:30:26.715918541 +0300
@@ -30,11 +30,13 @@
 		$(NULL)
 
 SHARED_LIBRARY_LIBS += ../hunspell/src/$(LIB_PREFIX)hunspell_s.$(LIB_SUFFIX)
+SHARED_LIBRARY_LIBS += ../zemberek/src/$(LIB_PREFIX)zemberek_s.$(LIB_SUFFIX)
 
 include $(topsrcdir)/config/rules.mk
 
 LOCAL_INCLUDES += \
 	-I$(srcdir)/../hunspell/src \
+	-I$(srcdir)/../zemberek/src \
 	-I$(topsrcdir)/content/base/src \
 	-I$(topsrcdir)/editor/libeditor/base \
 	$(NULL)
diff -Nuar a/extensions/spellcheck/src/mozSpellChecker.cpp b/extensions/spellcheck/src/mozSpellChecker.cpp
--- a/extensions/spellcheck/src/mozSpellChecker.cpp	2013-03-27 00:17:43.000000000 +0200
+++ b/extensions/spellcheck/src/mozSpellChecker.cpp	2013-04-03 12:44:37.256889092 +0300
@@ -10,7 +10,7 @@
 #include "nsICategoryManager.h"
 #include "nsISupportsPrimitives.h"
 
-#define DEFAULT_SPELL_CHECKER "@mozilla.org/spellchecker/engine;1"
+#define DEFAULT_SPELL_CHECKER "@mozilla.org/spellchecker/hunspell;1"
 
 NS_IMPL_CYCLE_COLLECTING_ADDREF(mozSpellChecker)
 NS_IMPL_CYCLE_COLLECTING_RELEASE(mozSpellChecker)
diff -Nuar a/extensions/spellcheck/src/mozSpellCheckerFactory.cpp b/extensions/spellcheck/src/mozSpellCheckerFactory.cpp
--- a/extensions/spellcheck/src/mozSpellCheckerFactory.cpp	2013-03-27 00:17:43.000000000 +0200
+++ b/extensions/spellcheck/src/mozSpellCheckerFactory.cpp	2013-04-03 12:51:45.892874252 +0300
@@ -6,7 +6,9 @@
 
 #include "mozilla/ModuleUtils.h"
 #include "mozHunspell.h"
+#include "mozZemberek.h"
 #include "mozHunspellDirProvider.h"
+#include "mozZemberekDirProvider.h"
 #include "mozSpellChecker.h"
 #include "mozInlineSpellChecker.h"
 #include "nsTextServicesCID.h"
@@ -24,14 +26,18 @@
 { 0xa0, 0x1a, 0x66, 0x40, 0x2e, 0xa2, 0x86, 0x57} }
 
 NS_GENERIC_FACTORY_CONSTRUCTOR_INIT(mozHunspell, Init)
+NS_GENERIC_FACTORY_CONSTRUCTOR_INIT(mozZemberek, Init)
 NS_GENERIC_FACTORY_CONSTRUCTOR(mozHunspellDirProvider)
+NS_GENERIC_FACTORY_CONSTRUCTOR(mozZemberekDirProvider)
 NS_GENERIC_FACTORY_CONSTRUCTOR_INIT(mozSpellChecker, Init)
 NS_GENERIC_FACTORY_CONSTRUCTOR_INIT(mozPersonalDictionary, Init)
 NS_GENERIC_FACTORY_CONSTRUCTOR(mozSpellI18NManager)
 NS_GENERIC_FACTORY_CONSTRUCTOR(mozInlineSpellChecker)
 
 NS_DEFINE_NAMED_CID(MOZ_HUNSPELL_CID);
+NS_DEFINE_NAMED_CID(MOZ_ZEMBEREK_CID);
 NS_DEFINE_NAMED_CID(HUNSPELLDIRPROVIDER_CID);
+NS_DEFINE_NAMED_CID(ZEMBEREKDIRPROVIDER_CID);
 NS_DEFINE_NAMED_CID(NS_SPELLCHECKER_CID);
 NS_DEFINE_NAMED_CID(MOZ_PERSONALDICTIONARY_CID);
 NS_DEFINE_NAMED_CID(MOZ_SPELLI18NMANAGER_CID);
@@ -40,6 +46,8 @@
 static const mozilla::Module::CIDEntry kSpellcheckCIDs[] = {
     { &kMOZ_HUNSPELL_CID, false, NULL, mozHunspellConstructor },
     { &kHUNSPELLDIRPROVIDER_CID, false, NULL, mozHunspellDirProviderConstructor },
+    { &kMOZ_ZEMBEREK_CID, false, NULL, mozZemberekConstructor },
+    { &kZEMBEREKDIRPROVIDER_CID, false, NULL, mozZemberekDirProviderConstructor },
     { &kNS_SPELLCHECKER_CID, false, NULL, mozSpellCheckerConstructor },
     { &kMOZ_PERSONALDICTIONARY_CID, false, NULL, mozPersonalDictionaryConstructor },
     { &kMOZ_SPELLI18NMANAGER_CID, false, NULL, mozSpellI18NManagerConstructor },
@@ -50,6 +58,8 @@
 static const mozilla::Module::ContractIDEntry kSpellcheckContracts[] = {
     { MOZ_HUNSPELL_CONTRACTID, &kMOZ_HUNSPELL_CID },
     { mozHunspellDirProvider::kContractID, &kHUNSPELLDIRPROVIDER_CID },
+    { MOZ_ZEMBEREK_CONTRACTID, &kMOZ_ZEMBEREK_CID },
+    { mozZemberekDirProvider::kContractID, &kZEMBEREKDIRPROVIDER_CID },
     { NS_SPELLCHECKER_CONTRACTID, &kNS_SPELLCHECKER_CID },
     { MOZ_PERSONALDICTIONARY_CONTRACTID, &kMOZ_PERSONALDICTIONARY_CID },
     { MOZ_SPELLI18NMANAGER_CONTRACTID, &kMOZ_SPELLI18NMANAGER_CID },
@@ -59,6 +69,7 @@
 
 static const mozilla::Module::CategoryEntry kSpellcheckCategories[] = {
     { XPCOM_DIRECTORY_PROVIDER_CATEGORY, "spellcheck-directory-provider", mozHunspellDirProvider::kContractID },
+    { XPCOM_DIRECTORY_PROVIDER_CATEGORY, "spellcheck-directory-provider", mozZemberekDirProvider::kContractID },
     { NULL }
 };
 
diff -Nuar a/toolkit/library/Makefile.in b/toolkit/library/Makefile.in
--- a/toolkit/library/Makefile.in	2013-03-27 00:18:05.000000000 +0200
+++ b/toolkit/library/Makefile.in	2013-04-03 12:35:23.054908281 +0300
@@ -382,6 +382,10 @@
 EXTRA_DSO_LDOPTS += $(MOZ_HUNSPELL_LIBS)
 endif
 
+ifdef MOZ_NATIVE_ZEMBEREK
+EXTRA_DSO_LDOPTS += $(MOZ_ZEMBEREK_LIBS)
+endif
+
 ifdef MOZ_NATIVE_LIBEVENT
 EXTRA_DSO_LDOPTS += $(MOZ_LIBEVENT_LIBS)
 endif
