--- linux-2.6.35-rc6-git-mnb0.1/3rdparty/viahss/viahss.c.orig	2010-07-30 21:17:30.000000000 +0300
+++ linux-2.6.35-rc6-git-mnb0.1/3rdparty/viahss/viahss.c	2010-07-30 23:10:21.324978822 +0300
@@ -25,7 +25,7 @@ static int __init viahss_init(void)
 {
         struct pci_dev *pcidev = NULL;
         unsigned char confval,val;
-        pcidev = pci_find_device (PCI_VENDOR_ID_VIA,PCI_DEVICE_ID_VIA_82C686,NULL);
+        pcidev = pci_get_device (PCI_VENDOR_ID_VIA,PCI_DEVICE_ID_VIA_82C686,NULL);
         if (pcidev) {
                 spin_lock_irq(&driver_lock);
                 /* start config */
@@ -55,7 +55,7 @@ static void __exit viahss_exit(void)
 {
         struct pci_dev *pcidev = NULL;
         unsigned char confval,val;
-        pcidev = pci_find_device (PCI_VENDOR_ID_VIA,PCI_DEVICE_ID_VIA_82C686,NULL);
+        pcidev = pci_get_device (PCI_VENDOR_ID_VIA,PCI_DEVICE_ID_VIA_82C686,NULL);
         if (pcidev) {
                 spin_lock_irq(&driver_lock);
                 /* start config */
