--- linux-3.8.13.alx/drivers/net/ethernet/atheros/alx/alx_main.c.orig	2013-05-27 20:07:10.000000000 +0300
+++ linux-3.8.13.alx/drivers/net/ethernet/atheros/alx/alx_main.c	2013-05-27 20:15:53.702575611 +0300
@@ -1026,6 +1026,7 @@ static int alx_identify_hw(struct alx_ad
 		if (rev < ALX_REV_C0) {
 			hw->ptrn_ofs = 0x600;
 			hw->max_ptrns = 8;
+			pdev->dev_flags |= PCI_DEV_FLAGS_MSI_INTX_DISABLE_BUG;
 		} else {
 			hw->ptrn_ofs = 0x14000;
 			hw->max_ptrns = 16;
