diff -up aspell-0.60.6.1/configure.fc aspell-0.60.6.1/configure
--- aspell-0.60.6.1/configure.fc	2011-07-04 10:58:50.000000000 +0200
+++ aspell-0.60.6.1/configure	2011-08-16 11:28:58.626771599 +0200
@@ -839,6 +839,7 @@ MAINTAINER_MODE_FALSE
 MAINT
 pkgdocdir
 pkgdatadir
+pkgdatadir2
 pkglibdir
 CXX
 CXXFLAGS
@@ -2634,18 +2635,21 @@ pkgdatadir=undef
 # Check whether --enable-pkgdatadir was given.
 if test "${enable_pkgdatadir+set}" = set; then
   enableval=$enable_pkgdatadir; pkgdatadir=$enable_pkgdatadir
+  pkgdatadir2=$enable_pkgdatadir
 fi
 
 
 # Check whether --enable-pkgdata-dir was given.
 if test "${enable_pkgdata_dir+set}" = set; then
   enableval=$enable_pkgdata_dir; pkgdatadir=$enable_dict_dir
+  pkgdatadir2=$enable_dict_dir
 fi
 
 
 if test "$pkgdatadir" = "undef"
 then
   pkgdatadir=\${libdir}/aspell-0.60
+  pkgdatadir2=${exec_prefix}/lib/aspell-0.60:${exec_prefix}/lib64/aspell-0.60
 fi
 
 
@@ -20119,6 +20123,7 @@ MAINTAINER_MODE_FALSE!$MAINTAINER_MODE_F
 MAINT!$MAINT$ac_delim
 pkgdocdir!$pkgdocdir$ac_delim
 pkgdatadir!$pkgdatadir$ac_delim
+pkgdatadir2!$pkgdatadir2$ac_delim
 pkglibdir!$pkglibdir$ac_delim
 CXX!$CXX$ac_delim
 CXXFLAGS!$CXXFLAGS$ac_delim
@@ -20142,7 +20147,7 @@ ac_ct_CC!$ac_ct_CC$ac_delim
 CCDEPMODE!$CCDEPMODE$ac_delim
 _ACEOF
 
-  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 97; then
+  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 98; then
     break
   elif $ac_last_try; then
     { { echo "$as_me:$LINENO: error: could not make $CONFIG_STATUS" >&5
diff -up aspell-0.60.6.1/Makefile.in.fc aspell-0.60.6.1/Makefile.in
--- aspell-0.60.6.1/Makefile.in.fc	2011-07-04 10:58:49.000000000 +0200
+++ aspell-0.60.6.1/Makefile.in	2011-08-16 11:20:09.030887258 +0200
@@ -344,6 +344,7 @@ distcleancheck_listfiles = find . -type 
 
 # These are needed due to a bug in Automake
 pkgdatadir = @pkgdatadir@
+pkgdatadir2 = @pkgdatadir2@
 pkglibdir = @pkglibdir@
 ACLOCAL = @ACLOCAL@
 AMTAR = @AMTAR@
@@ -1932,7 +1933,7 @@ gen/dirs.h: gen/mk-dirs_h.pl
 	cd gen; perl mk-dirs_h.pl ${prefix} ${pkgdatadir} ${pkglibdir}  ${sysconfdir} >  dirs.h
 
 scripts/run-with-aspell: scripts/run-with-aspell.create
-	sh ${srcdir}/scripts/run-with-aspell.create ${pkgdatadir} > scripts/run-with-aspell
+	sh ${srcdir}/scripts/run-with-aspell.create ${pkgdatadir2} > scripts/run-with-aspell
 	chmod 755 scripts/run-with-aspell
 @PSPELL_COMPATIBILITY_TRUE@scripts/pspell-config: scripts/mkconfig
 @PSPELL_COMPATIBILITY_TRUE@	sh ${srcdir}/scripts/mkconfig ${VERSION} ${datadir} ${pkgdatadir}
