Index: QTfrontend/gameuiconfig.cpp
===================================================================
--- QTfrontend/gameuiconfig.cpp.orig
+++ QTfrontend/gameuiconfig.cpp
@@ -52,8 +52,8 @@ GameUIConfig::GameUIConfig(HWForm * Form
     }
     else Form->ui.pageOptions->CBResolution->setCurrentIndex(t);
     Form->ui.pageOptions->CBResolution->setCurrentIndex((t < 0) ? 1 : t);
-    Form->ui.pageOptions->CBFullscreen->setChecked(value("video/fullscreen", false).toBool());
-    bool ffscr=value("frontend/fullscreen", false).toBool();
+    Form->ui.pageOptions->CBFullscreen->setChecked(value("video/fullscreen", true).toBool());
+    bool ffscr=value("frontend/fullscreen", true).toBool();
     Form->ui.pageOptions->CBFrontendFullscreen->setChecked(ffscr);
 
     Form->ui.pageOptions->SLQuality->setValue(value("video/quality", 5).toUInt());
