Index: netcdf-4.0.1/libsrc4/Makefile.in
===================================================================
--- netcdf-4.0.1.orig/libsrc4/Makefile.in
+++ netcdf-4.0.1/libsrc4/Makefile.in
@@ -639,7 +639,7 @@ clean-libLTLIBRARIES:
 	  rm -f "$${dir}/so_locations"; \
 	done
 libnetcdf.la: $(libnetcdf_la_OBJECTS) $(libnetcdf_la_DEPENDENCIES) 
-	$(libnetcdf_la_LINK) -rpath $(libdir) $(libnetcdf_la_OBJECTS) $(libnetcdf_la_LIBADD) $(LIBS)
+	$(libnetcdf_la_LINK) -rpath $(libdir) $(libnetcdf_la_OBJECTS) $(libnetcdf_la_LIBADD) -lhdf5_hl -lhdf5 $(LIBS)
 
 clean-checkPROGRAMS:
 	@list='$(check_PROGRAMS)'; for p in $$list; do \
Index: netcdf-4.0.1/fortran/Makefile.in
===================================================================
--- netcdf-4.0.1.orig/fortran/Makefile.in
+++ netcdf-4.0.1/fortran/Makefile.in
@@ -182,7 +182,7 @@ INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_P
 LD = @LD@
 LDFLAGS = @LDFLAGS@
 LIBOBJS = @LIBOBJS@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lgfortran
 LIBTOOL = @LIBTOOL@
 LIPO = @LIPO@
 LN_S = @LN_S@
