Index: chkconfig-1.3.47/alternatives.c
===================================================================
--- chkconfig-1.3.47.orig/alternatives.c
+++ chkconfig-1.3.47/alternatives.c
@@ -370,11 +370,15 @@ static int removeLinks(struct linkSet *
 
 static int makeLinks(struct linkSet * l, const char * altDir, int flags) {
     char * sl;
+    char * ftarget;
     struct stat sb;
 
     sl = alloca(strlen(altDir) + strlen(l->title) + 2);
     sprintf(sl, "%s/%s", altDir, l->title);
-    if (isLink(l->facility)) {
+    ftarget = alloca(strlen(sl) + 1);
+    if (isLink(l->facility) &&
+		readlink(l->facility, ftarget, strlen(sl)) &&
+		strncmp(ftarget, sl, strlen(sl))) {
 	    if (FL_TEST(flags)) {
 		    printf(_("would link %s -> %s\n"), l->facility, sl);
 	    } else {
