Index: chkconfig-1.3.47/alternatives.c
===================================================================
--- chkconfig-1.3.47.orig/alternatives.c
+++ chkconfig-1.3.47/alternatives.c
@@ -798,8 +798,8 @@ int main(int argc, const char ** argv) {
     struct stat sb;
 
     setlocale(LC_ALL, ""); 
-    bindtextdomain("chkconfig","/usr/share/locale"); 
-    textdomain("chkconfig"); 
+    bindtextdomain("alternatives", "/usr/share/locale"); 
+    textdomain("alternatives"); 
 
     if (!argv[1])
 	return usage(2);
Index: chkconfig-1.3.47/po/Makefile
===================================================================
--- chkconfig-1.3.47.orig/po/Makefile
+++ chkconfig-1.3.47/po/Makefile
@@ -5,12 +5,12 @@ INSTALLNLSDIR=$(DESTDIR)/usr/share/local
 
 MSGMERGE = msgmerge
 
-NLSPACKAGE = chkconfig
+NLSPACKAGE = alternatives
 
 POFILES = $(shell ls *.po)
 CATALOGS = $(shell ls *.po | sed 's/\.po/\.mo/')
 
-POTFILES  = ../chkconfig.c ../ntsysv.c ../leveldb.c ../alternatives.c
+POTFILES  = ../alternatives.c
 
 all: $(NLSPACKAGE).pot $(CATALOGS)
 
