--- a/designer-Qt4/designer.pro	2011-06-13 14:16:23.000000000 +0200
+++ b/designer-Qt4/designer.pro	2012-10-28 18:05:40.000000000 +0100
@@ -4,6 +4,8 @@
 TEMPLATE = lib
 TARGET = qscintillaplugin
 
+INCLUDEPATH = ../Qt4Qt5
+
 CONFIG += designer release plugin
 
 HEADERS = qscintillaplugin.h
@@ -12,4 +14,4 @@
 target.path = $$[QT_INSTALL_PLUGINS]/designer
 INSTALLS += target
 
-LIBS += -lqscintilla2
+LIBS += -L../Qt4Qt5 -lqscintilla2
