--- libgnomevfs/gnome-vfs-unix-mounts.c.orig	2004-09-29 20:01:35.682949864 +0200
+++ libgnomevfs/gnome-vfs-unix-mounts.c	2004-09-29 20:02:25.530371904 +0200
@@ -591,6 +591,7 @@ _gnome_vfs_get_unix_mount_table (GList *
 			&& hasmntopt (mntent, "user") != hasmntopt (mntent, "user_xattr"))
 		    || hasmntopt (mntent, "users") != NULL
 		    || hasmntopt (mntent, "owner") != NULL
+		    || hasmntopt (mntent, "pamconsole") != NULL
 #endif
 		    ) {
 			mount_entry->is_user_mountable = TRUE;
@@ -669,6 +670,7 @@ _gnome_vfs_get_unix_mount_table (GList *
 			&& hasmntopt (&mntent, "user") != hasmntopt (&mntent, "user_xattr"))
 		    || hasmntopt (&mntent, "users") != NULL
 		    || hasmntopt (&mntent, "owner") != NULL
+		    || hasmntopt (mntent, "pamconsole") != NULL
 #endif
 		    ) {
 			mount_entry->is_user_mountable = TRUE;
