--- src/tools/moc/moc.cpp
+++ src/tools/moc/moc.cpp
@@ -732,9 +732,6 @@
 
 void Moc::generate(FILE *out)
 {
-
-    QDateTime dt = QDateTime::currentDateTime();
-    QByteArray dstr = dt.toString().toLatin1();
     QByteArray fn = filename;
     int i = filename.length()-1;
     while (i>0 && filename[i-1] != '/' && filename[i-1] != '\\')
@@ -743,8 +740,8 @@
         fn = filename.mid(i);
     fprintf(out, "/****************************************************************************\n"
             "** Meta object code from reading C++ file '%s'\n**\n" , (const char*)fn);
-    fprintf(out, "** Created: %s\n"
-            "**      by: The Qt Meta Object Compiler version %d (Qt %s)\n**\n" , dstr.data(), mocOutputRevision, QT_VERSION_STR);
+    fprintf(out, "** Created:\n"
+            "**      by: The Qt Meta Object Compiler version %d (Qt %s)\n**\n" , mocOutputRevision, QT_VERSION_STR);
     fprintf(out, "** WARNING! All changes made in this file will be lost!\n"
             "*****************************************************************************/\n\n");
 
--- src/tools/rcc/rcc.cpp
+++ src/tools/rcc/rcc.cpp
@@ -712,7 +712,6 @@
         writeString("** Resource object code\n");
         writeString("**\n");
         writeString("** Created: ");
-        writeByteArray(QDateTime::currentDateTime().toString().toLatin1());
         writeString("\n**      by: The Resource Compiler for Qt version ");
         writeByteArray(QT_VERSION_STR);
         writeString("\n**\n");
--- src/tools/uic/uic.cpp
+++ src/tools/uic/uic.cpp
@@ -139,7 +139,7 @@
         out << "/********************************************************************************\n";
         out << "** Form generated from reading UI file '" << QFileInfo(opt.inputFile).fileName() << "'\n";
         out << "**\n";
-        out << "** Created: " << QDateTime::currentDateTime().toString() << "\n";
+        out << "** Created\n";
         out << "**      " << QString::fromLatin1("by: Qt User Interface Compiler version %1\n").arg(QLatin1String(QT_VERSION_STR));
         out << "**\n";
         out << "** WARNING! All changes made in this file will be lost when recompiling UI file!\n";
--- tools/qdbus/qdbusxml2cpp/qdbusxml2cpp.cpp
+++ tools/qdbus/qdbusxml2cpp/qdbusxml2cpp.cpp
@@ -525,9 +525,11 @@
     } else {
         includeGuard = QLatin1String("QDBUSXML2CPP_PROXY");
     }
+#if 0
     includeGuard = QString(QLatin1String("%1_%2"))
                    .arg(includeGuard)
                    .arg(QDateTime::currentDateTime().toTime_t());
+#endif
     hs << "#ifndef " << includeGuard << endl
        << "#define " << includeGuard << endl
        << endl;
@@ -831,9 +833,11 @@
     } else {
         includeGuard = QLatin1String("QDBUSXML2CPP_ADAPTOR");
     }
+#if 0
     includeGuard = QString(QLatin1String("%1_%2"))
                    .arg(includeGuard)
                    .arg(QDateTime::currentDateTime().toTime_t());
+#endif
     hs << "#ifndef " << includeGuard << endl
        << "#define " << includeGuard << endl
        << endl;
