Index: configure
===================================================================
--- configure.orig
+++ configure
@@ -65,7 +65,6 @@ OPT_CMDLINE=`echo $@ | sed "s,-v ,,g; s,
 
 # initialize global variables
 QMAKE_SWITCHES=
-QMAKE_VARS=
 QMAKE_CONFIG=
 QTCONFIG_CONFIG=
 QT_CONFIG=
@@ -1324,7 +1323,7 @@ while [ "$#" -gt 0 ]; do
 	    QMakeVar add QMAKE_CFLAGS -pg
 	    QMakeVar add QMAKE_CXXFLAGS -pg
 	    QMakeVar add QMAKE_LFLAGS -pg
-            QMAKE_VARS="$QMAKE_VARS CONFIG+=nostrip"
+	    QMakeVar add CONFIG nostrip
         else
             UNKNOWN_OPT=yes
         fi
@@ -2823,6 +2822,8 @@ if [ "$OPT_VERBOSE" = "yes" ]; then
     fi
 fi
 
+QMakeVar add CONFIG nostrip
+
 #-------------------------------------------------------------------------------
 # tests that don't need qmake (must be run before displaying help)
 #-------------------------------------------------------------------------------
