--- Makefile~	2007-01-16 17:18:52.000000000 +0100
+++ Makefile	2007-01-16 17:28:12.000000000 +0100
@@ -249,6 +249,9 @@
 		$(DESTDIR)$(libdir)/libgsm.a		\
 		$(DESTDIR)$(libdir)/libgsm.so		\
 		$(DESTDIR)$(includedir)/gsm/gsm.h	\
+		$(DESTDIR)$(includedir)/gsm/private.h   \
+		$(DESTDIR)$(includedir)/gsm/unproto.h   \
+		$(DESTDIR)$(includedir)/gsm/proto.h   \
 		$(DESTDIR)$(mandir)/man3/gsm.3		\
 		$(DESTDIR)$(mandir)/man3/gsm_explode.3	\
 		$(DESTDIR)$(mandir)/man3/gsm_option.3	\
@@ -380,6 +383,20 @@
 		-rm $(RMFLAGS) $@
 		-mkdir -p $(DESTDIR)$(includedir)/gsm/
 		install -m0444 $? $@
+$(DESTDIR)$(includedir)/gsm/private.h:      $(INC)/private.h
+		-rm $(RMFLAGS) $@
+		-mkdir -p $(DESTDIR)$(includedir)/gsm/
+		-install -m0444 $? $@
+
+$(DESTDIR)$(includedir)/gsm/proto.h:      $(INC)/proto.h
+		-rm $(RMFLAGS) $@
+		-mkdir -p $(DESTDIR)$(includedir)/gsm/
+		-install -m0444 $? $@
+$(DESTDIR)$(includedir)/gsm/unproto.h:      $(INC)/unproto.h
+		-rm $(RMFLAGS) $@
+		-mkdir -p $(DESTDIR)$(includedir)/gsm/
+		-install -m0444 $? $@
+
 
 $(DESTDIR)$(libdir)/libgsm.a:	$(LIBGSM)
 		-rm $(RMFLAGS) $@
