diff -Nur gsm-1.0-pl13-old/Makefile gsm-1.0-pl13/Makefile
--- gsm-1.0-pl13-old/Makefile	2010-02-15 15:25:07.747416137 +0200
+++ gsm-1.0-pl13/Makefile	2010-02-15 15:26:45.211541984 +0200
@@ -47,8 +47,8 @@
 # CC		= /usr/lang/acc
 # CCFLAGS 	= -c -O
 
-CC		= gcc -ansi -pedantic
-CCFLAGS 	= -c $(RPM_OPT_FLAGS) -fPIC -D_REENTRANT -DNeedFunctionPrototypes=1
+CC		= pardusCC -ansi -pedantic
+CCFLAGS 	= -c pardusCFLAGS -fPIC -D_REENTRANT -DNeedFunctionPrototypes=1
 
 LD 		= $(CC)
 
