diff -up cups-1.6.1/scheduler/colorman.c.str4276 cups-1.6.1/scheduler/colorman.c
--- cups-1.6.1/scheduler/colorman.c.str4276	2013-02-15 17:09:23.053544061 +0000
+++ cups-1.6.1/scheduler/colorman.c	2013-02-15 17:10:07.972783360 +0000
@@ -1128,7 +1128,7 @@ colord_delete_device(
   message = COLORD_DBUS_MSG(COLORD_DBUS_PATH, "DeleteDevice");
 
   dbus_message_iter_init_append(message, &args);
-  dbus_message_iter_append_basic(&args, DBUS_TYPE_STRING, &device_id);
+  dbus_message_iter_append_basic(&args, DBUS_TYPE_OBJECT_PATH, &device_path);
 
  /*
   * Send the DeleteDevice request synchronously...
