diff -up cups-1.7b1/cups/globals.c.use-ipp1.1 cups-1.7b1/cups/globals.c
--- cups-1.7b1/cups/globals.c.use-ipp1.1	2013-06-26 15:05:54.615238227 +0100
+++ cups-1.7b1/cups/globals.c	2013-06-26 15:06:11.662299971 +0100
@@ -219,7 +219,7 @@ cups_globals_alloc(void)
   cg->any_root       = 1;
   cg->expired_certs  = 1;
   cg->expired_root   = 1;
-  cg->server_version = 20;
+  cg->server_version = 11;
 
 #ifdef DEBUG
  /*
