diff -up cups-1.7b1/backend/usb-libusb.c~ cups-1.7b1/backend/usb-libusb.c
--- cups-1.7b1/backend/usb-libusb.c~	2013-04-18 15:18:34.000000000 +0100
+++ cups-1.7b1/backend/usb-libusb.c	2013-06-25 09:28:21.325699866 +0100
@@ -158,6 +158,8 @@ static const struct quirk_printer_struct
 			    Printer, https://bugs.launchpad.net/bugs/1032456 */
 	{ 0x04a9, 0x1717, USBLP_QUIRK_BIDIR }, /* Canon, Inc. MP510
 			    Printer, https://bugs.launchpad.net/bugs/1050009 */
+	{ 0x04a9, 0x1730, USBLP_QUIRK_BIDIR }, /* Canon, Inc. MP540
+			    Printer, https://bugzilla.redhat.com/967873 */
 	{ 0x04a9, 0x173d, USBLP_QUIRK_BIDIR }, /* Canon, Inc. MP550
 			    Printer, http://www.cups.org/str.php?L4155 */
 	{ 0x04a9, 0x173e, USBLP_QUIRK_BIDIR }, /* Canon, Inc. MP560
