--- utils/encoder.c
+++ utils/encoder.c
@@ -61,7 +61,7 @@
 int chanf(int fd, int frequency);
 void usage(void);
 
-int splice(unsigned char *, int, int);
+int myth_splice(unsigned char *, int, int);
 #define SPLICE_START 1
 #define SPLICE_END   2
 #define GOP_COUNTER  3
@@ -853,7 +853,7 @@
 }
 
 /* Code borrowed from MythTV */
-int splice(unsigned char *buf, int len, int mode)
+int myth_splice(unsigned char *buf, int len, int mode)
 {
 	unsigned char *bufptr = buf;
 	unsigned int state = 0xFFFFFFFF, v = 0;
