diff -p -up neverball-1.5.2/Makefile~ neverball-1.5.2/Makefile
--- neverball-1.5.2/Makefile~	2009-08-16 20:10:21.000000000 +0200
+++ neverball-1.5.2/Makefile	2009-08-28 09:10:33.000000000 +0200
@@ -75,7 +75,7 @@ ALL_CPPFLAGS := $(SDL_CPPFLAGS) $(PNG_CP
 ALL_CPPFLAGS += \
     -DCONFIG_USER=\"$(USERDIR)\" \
     -DCONFIG_DATA=\"$(DATADIR)\" \
-    -DCONFIG_LOCALE=\"$(LOCALEDIR)\"
+    -DCONFIG_LOCALE=\"/usr/share/locale\"
 
 ifeq ($(ENABLE_NLS),0)
     ALL_CPPFLAGS += -DENABLE_NLS=0
 
