--- Makefile.orig	2010-04-19 10:04:10.000000000 -0700
+++ Makefile	2010-04-19 10:38:02.000000000 -0700
@@ -131,14 +131,14 @@
     OGL_LIBS  := -framework OpenGL
 endif
 
-BASE_LIBS := -ljpeg $(PNG_LIBS) $(FS_LIBS)
+BASE_LIBS := -ljpeg $(PNG_LIBS) $(FS_LIBS) -lm
 
 ifdef DARWIN
     BASE_LIBS += -L/opt/local/lib
 endif
 
 ALL_LIBS := $(SDL_LIBS) $(BASE_LIBS) $(TILT_LIBS) $(INTL_LIBS) -lSDL_ttf \
-    -lvorbisfile $(OGL_LIBS)
+    -lvorbisfile $(OGL_LIBS) -lX11
 
 #------------------------------------------------------------------------------
 
 
