diff -ur hylafax-4.2.4.orig/Makefile.in hylafax-4.2.4/Makefile.in
--- hylafax-4.2.4.orig/Makefile.in	2003-04-14 05:44:42.000000000 +0300
+++ hylafax-4.2.4/Makefile.in	2006-01-11 09:18:10.000000000 +0200
@@ -247,11 +247,7 @@
 	    -F ${SPOOL} sendq doneq docq tmp pollq archive
 makeDirs: makeClientDirs makeServerDirs
 
-makeDevices::
-	${INSTALL} -u ${FAXUSER} -g ${FAXGROUP} -m 600 -fifo \
-	    -idb hylafax.sw.server -F ${SPOOL} FIFO
-
-install: makeDirs makeDevices
+install: makeDirs
 	${INSTALL} -m 444 -F ${SPOOL} -idb hylafax.sw.server \
 	    -src ${SRCDIR}/COPYRIGHT -O COPYRIGHT
 	@if [ "@REGEX@" = yes ]; then \
