--- hylafax-4.3.1.orig/faxd/UUCPLock.c++	2000-09-21 12:24:54.000000000 +0300
+++ hylafax-4.3.1/faxd/UUCPLock.c++	2007-02-16 18:17:12.000000000 +0200
@@ -142,7 +142,7 @@
 UUCPLock::setupIDs()
 {
     if (UUCPuid == (uid_t) -1) {
-	const passwd *pwd = getpwnam("uucp");
+	const passwd *pwd = getpwnam("dialout");
 	if (!pwd)
 	    faxApp::fatal("Can not deduce identity of UUCP");
 	UUCPuid = pwd->pw_uid;
