Index: mozplugger-1.14.3/mozpluggerrc
===================================================================
--- mozplugger-1.14.3.orig/mozpluggerrc
+++ mozplugger-1.14.3/mozpluggerrc
@@ -69,9 +69,9 @@ define(TM_LINKS,[stream links nokill noe
 
 define(TM_AUDIO_STREAM,[controls stream noisy ignore_errors: TM_CMD("$file")])
 
-### OpenOffice
-define([OO],[swallow(VCLSalFrame) fill: ooffice2.0 -nologo -norestore -view $1 "$file"
-	swallow(VCLSalFrame) fill: ooffice -nologo -norestore -view $1 "$file"
+### LibreOffice
+define([LO],[swallow(VCLSalFrame) fill: libreoffice -nologo -norestore -view $1 "$file"
+	swallow(VCLSalFrame) fill: libreoffice -nologo -norestore -view $1 "$file"
 	swallow(VCLSalFrame) fill: soffice -nologo $1 "$file"])
 
 ### Acrobat Reader
@@ -299,56 +299,56 @@ application/postscript:ps:PostScript fil
 application/x-rtf:rtf:Rich Text Format
 application/rtf:rtf:Rich Text Format
 text/rtf:rtf:Rich Text Format
-	OO()
+	LO()
 	repeat noisy swallow(AbiWord) fill: abiword --nosplash --geometry +9000+9000 "$file"
 	repeat noisy swallow(kword): kword "$file"
 	repeat noisy swallow(Ted) fill: Ted "$file"
 
 application/x-msword:doc,dot:Microsoft Word Document
 application/msword:doc,dot:Microsoft Word Document
-	OO()
+	LO()
 	repeat noisy swallow(kword): kword "$file"
 	repeat noisy swallow(AbiWord) fill: abiword --nosplash --geometry +9000+9000 "$file"
 
 application/vnd.ms-excel:xls,xlb:Microsoft Excel Document
-	OO()
+	LO()
 	repeat swallow(Gnumeric) fill: gnumeric "$file"
 
-# OpenOffice MimeTypes (http://framework.openoffice.org/documentation/mimetypes/mimetypes.html)
-application/vnd.sun.xml.writer:sxw:OpenOffice Writer 6.0 documents
-application/so7_vnd.sun.xml.writer:sxw:OpenOffice Writer 7.0 documents
-application/vnd.sun.xml.writer.template:stw:OpenOffice Writer 6.0 templates
-application/vnd.sun.xml.writer.global:sxg:OpenOffice Writer 6.0 global documents
+# LibreOffice MimeTypes (http://framework.openoffice.org/documentation/mimetypes/mimetypes.html)
+application/vnd.sun.xml.writer:sxw:LibreOffice Writer 6.0 documents
+application/so7_vnd.sun.xml.writer:sxw:LibreOffice Writer 7.0 documents
+application/vnd.sun.xml.writer.template:stw:LibreOffice Writer 6.0 templates
+application/vnd.sun.xml.writer.global:sxg:LibreOffice Writer 6.0 global documents
 application/vnd.stardivision.writer:sdw:StarWriter 5.x documents
 application/vnd.stardivision.writer-global:sgl:StarWriter 5.x global documents
 application/x-starwriter:sdw:StarWriter 4.x documents
-application/vnd.sun.xml.calc:sxc:OpenOffice Calc 6.0 spreadsheets
-application/so7_vnd.sun.xml.calc:sxc:OpenOffice Calc 7.0 spreadsheets
-application/vnd.sun.xml.calc.template:stc:OpenOffice Calc 6.0 templates
+application/vnd.sun.xml.calc:sxc:LibreOffice Calc 6.0 spreadsheets
+application/so7_vnd.sun.xml.calc:sxc:LibreOffice Calc 7.0 spreadsheets
+application/vnd.sun.xml.calc.template:stc:LibreOffice Calc 6.0 templates
 application/vnd.stardivision.calc:sdc:StarCalc 5.x spreadsheets
 application/x-starcalc:sdc:StarCalc 4.x spreadsheets
 application/vnd.lotus-1-2-3: 123, wk1: Lotus 1-2-3 Document
-application/vnd.sun.xml.draw:sxd:OpenOffice Draw 6.0 documents
+application/vnd.sun.xml.draw:sxd:LibreOffice Draw 6.0 documents
 application/so7_vnd.sun.xml.draw:sxc:StarOffice Draw 7.0 documents
-application/vnd.sun.xml.draw.template:std:OpenOffice Draw 6.0 templates
+application/vnd.sun.xml.draw.template:std:LibreOffice Draw 6.0 templates
 application/vnd.stardivision.draw:sda:StarDraw 5.x documents
 application/x-stardraw:sda:StarDraw 4.x documents
-application/vnd.sun.xml.impress:sxi:OpenOffice Impress 6.0 presentations
+application/vnd.sun.xml.impress:sxi:LibreOffice Impress 6.0 presentations
 application/so7_vnd.sun.xml.impress:sxi:StarOffice 7.0 Impress presentations
-application/vnd.sun.xml.impress.template:sti:OpenOffice Impress 6.0 templates
+application/vnd.sun.xml.impress.template:sti:LibreOffice Impress 6.0 templates
 application/vnd.stardivision.impress:sdd:StarImpress 5.x presentations
 application/vnd.stardivision.impress-packed:sdp:StarImpress Packed 5.x files
 application/x-starimpress:sdd:StarImpress 4.x presentations
 application/vnd.ms-powerpoint:ppt:PowerPoint Slideshow
 application/mspowerpoint:ppt,ppz,pps,pot:PowerPoint Slideshow
-application/vnd.sun.xml.math:sxm:OpenOffice Math 6.0 documents
+application/vnd.sun.xml.math:sxm:LibreOffice Math 6.0 documents
 application/so7_vnd.sun.xml.math:sxm:StarOffice 7.0 Math documents
 application/vnd.stardivision.math:smf:StarMath 5.x documents
 application/x-starmath:smf:StarMath 4.x documents
 application/vnd.oasis.opendocument.text:odt,ODT:OASIS OpenDocument Text
 application/vnd.oasis.opendocument.spreadsheet:ods,ODS:OASIS OpenDocument SpreadSheet
 application/vnd.oasis.opendocument.presentation:odp,ODP:OASIS OpenDocument Presentation
-	OO()
+	LO()
 
 chemical/x-pdb:pdb: Protein Data Bank file
 model/x-pdb:pdb: Protein Data Bank file
