Index: gstreamer-0.11.94/tools/gst-inspect.c
===================================================================
--- gstreamer-0.11.94.orig/tools/gst-inspect.c
+++ gstreamer-0.11.94/tools/gst-inspect.c
@@ -1350,9 +1350,225 @@ print_element_info (GstElementFactory *
   return 0;
 }
 
+static void
+print_gst_structure_append_field (GList * strings, const char *field)
+{
+  GList *s;
+
+  //g_message ("adding '%s' to the string", field);
+
+  for (s = strings; s != NULL; s = s->next) {
+    g_string_append (s->data, field);
+  }
+}
 
 static void
@@ -1447,7 +1682,7 @@ print_plugin_automatic_install_info_prot
 }
 
 static void
-print_plugin_automatic_install_info (GstPlugin * plugin)
+print_plugin_automatic_install_info (GstPlugin * plugin, gboolean rpm_format)
 {
   GList *features, *l;
 
@@ -1492,7 +1731,7 @@ print_all_plugin_automatic_install_info
     plugin = (GstPlugin *) (plugins->data);
     plugins = g_list_next (plugins);
 
-    print_plugin_automatic_install_info (plugin);
+    print_plugin_automatic_install_info (plugin, FALSE);
   }
   gst_plugin_list_free (orig_plugins);
 }
