diff --git a/configure.in b/configure.in
index ae762a2..729e7f1 100644
--- a/configure.in
+++ b/configure.in
@@ -3,7 +3,8 @@ AC_INIT
 AC_CONFIG_SRCDIR([conf/pam_conv1/pam_conv_y.y])
 AC_CONFIG_AUX_DIR([build-aux])
 AM_INIT_AUTOMAKE("Linux-PAM", 1.1.6)
-AC_PREREQ(2.61)
+LT_INIT
+AC_PREREQ([2.68])
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CANONICAL_HOST
@@ -82,7 +83,6 @@ AM_PROG_LEX
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
-AC_PROG_LIBTOOL
 AM_PROG_CC_C_O
 PAM_LD_AS_NEEDED
 PAM_LD_NO_UNDEFINED
@@ -532,7 +532,7 @@ AC_CHECK_FUNCS(unshare, [UNSHARE=yes], [UNSHARE=no])
 AM_CONDITIONAL([HAVE_UNSHARE], [test "$UNSHARE" = yes])
 
 AC_ARG_ENABLE([regenerate-docu],
-  AC_HELP_STRING([--disable-regenerate-docu], [Don't re-build documentation from XML souces]),
+  AS_HELP_STRING([--disable-regenerate-docu],[Don't re-build documentation from XML sources]),
   [enable_docu=$enableval], [enable_docu=yes])
 dnl
 dnl Check for xsltproc
