diff -up Linux-PAM-1.1.5/modules/pam_limits/limits.conf.limits Linux-PAM-1.1.5/modules/pam_limits/limits.conf
--- Linux-PAM-1.1.5/modules/pam_limits/limits.conf.limits	2011-06-21 11:04:56.000000000 +0200
+++ Linux-PAM-1.1.5/modules/pam_limits/limits.conf	2011-12-21 09:09:17.000000000 +0100
@@ -1,5 +1,8 @@
 # /etc/security/limits.conf
 #
+#This file sets the resource limits for the users logged in via PAM.
+#It does not affect resource limits of the system services.
+#
 #Each line describes a limit for a user in the form:
 #
 #<domain>        <type>  <item>  <value>
