Index: mdbtools-0.6pre1/src/sql/Makefile.am
===================================================================
--- mdbtools-0.6pre1.orig/src/sql/Makefile.am	2005-01-15 06:02:06.000000000 +0100
+++ mdbtools-0.6pre1/src/sql/Makefile.am	2010-06-07 08:05:16.652462537 +0200
@@ -9,3 +9,6 @@ YACC = @YACC@ -d
 
 dist-hook:
 	rm -f $(distdir)/parser.c $(distdir)/parser.h $(distdir)/lexer.c
+
+lexer.o: parser.h
+
