diff -Nur pyclutter-gtk-old/Makefile.am pyclutter-gtk/Makefile.am
--- pyclutter-gtk-old/Makefile.am	2009-10-09 22:04:52.846898362 +0300
+++ pyclutter-gtk/Makefile.am	2009-10-09 22:05:26.937877985 +0300
@@ -1,5 +1,7 @@
 SUBDIRS = build clutter-gtk examples
 
+ACLOCAL_AMFLAGS = -I build/autotools
+
 pyclutter-gtk-$(PYCLUTTERGTK_MAJOR_VERSION).$(PYCLUTTERGTK_MINOR_VERSION).pc: pyclutter-gtk.pc
 	@cp -f $< $(@F)
 
