diff -Nur pyclutter-gtk-old/configure.ac pyclutter-gtk/configure.ac
--- pyclutter-gtk-old/configure.ac	2009-11-07 00:49:54.248247217 +0200
+++ pyclutter-gtk/configure.ac	2009-11-07 00:50:27.595247272 +0200
@@ -4,8 +4,8 @@
 # An odd micro number indicates in-progress development, (eg. from svn)
 # An even micro number indicates a released version.
 m4_define(pycluttergtk_version_major, 0)
-m4_define(pycluttergtk_version_minor, 9)
-m4_define(pycluttergtk_version_micro, 3)
+m4_define(pycluttergtk_version_minor, 90)
+m4_define(pycluttergtk_version_micro, 1)
 m4_define(pycluttergtk_version, pycluttergtk_version_major.pycluttergtk_version_minor.pycluttergtk_version_micro)
 
 m4_define(clutter_required_version,      1.0.0)
@@ -102,7 +102,7 @@
 )
 
 dnl check for clutter-gtk
-PKG_CHECK_MODULES(CLUTTERGTK, clutter-gtk-0.10 >= cluttergtk_required_version)
+PKG_CHECK_MODULES(CLUTTERGTK, clutter-gtk-0.90 >= cluttergtk_required_version)
 AC_SUBST(CLUTTERGTK_CFLAGS)
 AC_SUBST(CLUTTERGTK_LIBS)
 AS_IF([test -n "$export_dynamic"],
