diff -up pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs.no-viewport pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs
--- pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs.no-viewport	2010-09-29 12:07:40.124923613 -0400
+++ pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs	2010-09-29 12:07:50.313924546 -0400
@@ -210,34 +210,6 @@
 ;;  )
 ;;)
 
-;; From gtk-clutter-viewport.h
-
-(define-function gtk_clutter_viewport_get_type
-  (c-name "gtk_clutter_viewport_get_type")
-  (return-type "GType")
-)
-
-(define-function gtk_clutter_viewport_new
-  (c-name "gtk_clutter_viewport_new")
-  (is-constructor-of "GtkClutterViewport")
-  (return-type "ClutterActor*")
-  (properties
-    '("hadjustment" (optional))
-    '("vadjustment" (optional))
-  )
-)
-
-(define-method get_origin
-  (of-object "GtkClutterViewport")
-  (c-name "gtk_clutter_viewport_get_origin")
-  (return-type "none")
-  (parameters
-    '("gfloat*" "x")
-    '("gfloat*" "y")
-    '("gfloat*" "z")
-  )
-)
-
 ;; From gtk-clutter-window.h
 
 (define-function gtk_clutter_window_get_type
diff -up pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs.no-viewport pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs
--- pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs.no-viewport	2010-09-29 12:08:04.227924821 -0400
+++ pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs	2010-09-29 12:08:10.869924552 -0400
@@ -16,13 +16,6 @@
   (gtype-id "GTK_CLUTTER_TYPE_EMBED")
 )
 
-(define-object Viewport
-  (in-module "GtkClutter")
-  (parent "ClutterActor")
-  (c-name "GtkClutterViewport")
-  (gtype-id "GTK_CLUTTER_TYPE_VIEWPORT")
-)
-
 (define-object Window
   (in-module "GtkClutter")
   (parent "GtkWindow")
