--- pyclutter-gtk-0.10.0/clutter-gtk/Makefile.am.link	2010-11-06 03:08:58.000000000 +0000
+++ pyclutter-gtk-0.10.0/clutter-gtk/Makefile.am	2010-11-06 03:09:42.000000000 +0000
@@ -18,7 +18,7 @@
 						 -avoid-version \
 						 -export-symbols-regex init_cluttergtk
 
-_cluttergtk_la_LIBADD = $(CLUTTERGTK_LIBS)
+_cluttergtk_la_LIBADD = $(CLUTTERGTK_LIBS) -lpython$(PYTHON_VERSION)
 _cluttergtk_la_SOURCES = cluttergtkmodule.c
 nodist__cluttergtk_la_SOURCES = cluttergtk.c
 
--- pyclutter-gtk-0.10.0/clutter-gtk/Makefile.in.link	2010-11-06 03:09:10.000000000 +0000
+++ pyclutter-gtk-0.10.0/clutter-gtk/Makefile.in	2010-11-06 03:10:13.000000000 +0000
@@ -250,7 +250,7 @@
 						 -avoid-version \
 						 -export-symbols-regex init_cluttergtk
 
-_cluttergtk_la_LIBADD = $(CLUTTERGTK_LIBS)
+_cluttergtk_la_LIBADD = $(CLUTTERGTK_LIBS) -lpython$(PYTHON_VERSION)
 _cluttergtk_la_SOURCES = cluttergtkmodule.c
 nodist__cluttergtk_la_SOURCES = cluttergtk.c
 defsdir = $(PYCLUTTER_DEFSDIR)
