diff -up pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs.no-zoomable pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs
--- pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs.no-zoomable	2010-09-29 12:03:34.331923847 -0400
+++ pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs	2010-09-29 12:03:56.974924345 -0400
@@ -289,39 +289,3 @@
   (parameters
   )
 )
-
-;; From gtk-clutter-zoomable.h
-
-(define-function gtk_clutter_zoomable_get_type
-  (c-name "gtk_clutter_zoomable_get_type")
-  (return-type "GType")
-)
-
-(define-method set_adjustment
-  (of-object "GtkClutterZoomable")
-  (c-name "gtk_clutter_zoomable_set_adjustment")
-  (return-type "none")
-  (parameters
-    '("GtkAdjustment*" "z_adjust")
-  )
-)
-
-(define-virtual set_adjustment
-  (of-object "GtkClutterZoomable")
-  (return-type "none")
-  (parameters
-    '("GtkAdjustment*" "z_adjust")
-  )
-)
-
-(define-method get_adjustment
-  (of-object "GtkClutterZoomable")
-  (c-name "gtk_clutter_zoomable_get_adjustment")
-  (return-type "GtkAdjustment*")
-)
-
-(define-virtual get_adjustment
-  (of-object "GtkClutterZoomable")
-  (return-type "GtkAdjustment*")
-)
-
diff -up pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs.no-zoomable pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs
--- pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs.no-zoomable	2010-09-29 12:04:18.439925620 -0400
+++ pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs	2010-09-29 12:04:28.169923563 -0400
@@ -36,10 +36,3 @@
   (c-name "GtkClutterWindow")
   (gtype-id "GTK_CLUTTER_TYPE_WINDOW")
 )
-
-(define-interface Zoomable
-  (in-module "GtkClutter")
-  (c-name "GtkClutterZoomable")
-  (gtype-id "GTK_CLUTTER_TYPE_ZOOMABLE")
-  (vtable "GtkClutterZoomableIface")
-)
