diff -urN pyclutter-gtk-0.10.0/Makefile.am pyclutter-gtk-0.10.0-patched/Makefile.am
--- pyclutter-gtk-0.10.0/Makefile.am	2009-08-28 23:00:53.000000000 +1100
+++ pyclutter-gtk-0.10.0-patched/Makefile.am	2012-09-18 22:25:11.000000000 +1100
@@ -7,6 +7,8 @@
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = $(pkgfiles)
 
+ACLOCAL_AMFLAGS = -I build/autotools
+
 DIST_SUBDIRS = \
 	build \
 	clutter-gtk \
