diff -up pyclutter-gtk-0.10.0/configure.ac.BAD pyclutter-gtk-0.10.0/configure.ac
--- pyclutter-gtk-0.10.0/configure.ac.BAD	2010-09-29 11:29:06.181923341 -0400
+++ pyclutter-gtk-0.10.0/configure.ac	2010-09-29 11:29:10.735925579 -0400
@@ -102,7 +102,7 @@ AS_IF([test -n "$export_dynamic"],
 )
 
 dnl check for clutter-gtk
-PKG_CHECK_MODULES(CLUTTERGTK, clutter-gtk-0.10 >= cluttergtk_required_version)
+PKG_CHECK_MODULES(CLUTTERGTK, clutter-gtk-1.0 >= cluttergtk_required_version)
 AC_SUBST(CLUTTERGTK_CFLAGS)
 AC_SUBST(CLUTTERGTK_LIBS)
 AS_IF([test -n "$export_dynamic"],
diff -up pyclutter-gtk-0.10.0/configure.BAD pyclutter-gtk-0.10.0/configure
--- pyclutter-gtk-0.10.0/configure.BAD	2010-09-29 11:29:29.990924071 -0400
+++ pyclutter-gtk-0.10.0/configure	2010-09-29 11:30:17.282924001 -0400
@@ -12301,12 +12301,12 @@ if test -n "$PKG_CONFIG"; then
         pkg_cv_CLUTTERGTK_CFLAGS="$CLUTTERGTK_CFLAGS"
     else
         if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"clutter-gtk-0.10 >= 0.10.2\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "clutter-gtk-0.10 >= 0.10.2") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"clutter-gtk-1.0 >= 0.10.2\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "clutter-gtk-1.0 >= 0.10.2") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_CLUTTERGTK_CFLAGS=`$PKG_CONFIG --cflags "clutter-gtk-0.10 >= 0.10.2" 2>/dev/null`
+  pkg_cv_CLUTTERGTK_CFLAGS=`$PKG_CONFIG --cflags "clutter-gtk-1.0 >= 0.10.2" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -12319,12 +12319,12 @@ if test -n "$PKG_CONFIG"; then
         pkg_cv_CLUTTERGTK_LIBS="$CLUTTERGTK_LIBS"
     else
         if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"clutter-gtk-0.10 >= 0.10.2\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "clutter-gtk-0.10 >= 0.10.2") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"clutter-gtk-1.0 >= 0.10.2\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "clutter-gtk-1.0 >= 0.10.2") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_CLUTTERGTK_LIBS=`$PKG_CONFIG --libs "clutter-gtk-0.10 >= 0.10.2" 2>/dev/null`
+  pkg_cv_CLUTTERGTK_LIBS=`$PKG_CONFIG --libs "clutter-gtk-1.0 >= 0.10.2" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -12343,14 +12343,14 @@ else
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        CLUTTERGTK_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "clutter-gtk-0.10 >= 0.10.2"`
+	        CLUTTERGTK_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "clutter-gtk-1.0 >= 0.10.2"`
         else
-	        CLUTTERGTK_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "clutter-gtk-0.10 >= 0.10.2"`
+	        CLUTTERGTK_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "clutter-gtk-1.0 >= 0.10.2"`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$CLUTTERGTK_PKG_ERRORS" >&5
 
-	as_fn_error "Package requirements (clutter-gtk-0.10 >= 0.10.2) were not met:
+	as_fn_error "Package requirements (clutter-gtk-1.0 >= 0.10.2) were not met:
 
 $CLUTTERGTK_PKG_ERRORS
 
