diff -Nur pyclutter-gtk-old/clutter-gtk/Makefile.am pyclutter-gtk/clutter-gtk/Makefile.am
--- pyclutter-gtk-old/clutter-gtk/Makefile.am	2009-10-09 22:11:42.578150801 +0300
+++ pyclutter-gtk/clutter-gtk/Makefile.am	2009-10-09 22:12:36.340878017 +0300
@@ -18,7 +18,7 @@
 						 -avoid-version \
 						 -export-symbols-regex init_cluttergtk
 
-_cluttergtk_la_LIBADD = $(CLUTTERGTK_LIBS)
+_cluttergtk_la_LIBADD = $(CLUTTERGTK_LIBS) -lpardusPythonVersion
 _cluttergtk_la_SOURCES = cluttergtkmodule.c
 nodist__cluttergtk_la_SOURCES = cluttergtk.c
 
