--- ntop/configure.in.orig	2007-03-05 14:36:41.000000000 -0700
+++ ntop/configure.in	2007-03-05 14:54:46.000000000 -0700
@@ -2299,7 +2299,7 @@
 AC_DEFINE_UNQUOTED(CFG_PLUGIN_DIR,"$CFG_PLUGIN_DIR", [Plugin installation directory])
 echo "    Plugin files are in   $CFG_PLUGIN_DIR"
 
-CFG_DBFILE_DIR=$localstatedir/ntop
+CFG_DBFILE_DIR=$localstatedir/lib/ntop
 CFG_DBFILE_DIR=`(
   test "x$prefix" = xNONE && prefix=${ac_default_prefix}
   eval echo "$CFG_DBFILE_DIR"
