Index: linphone-3.2.1/configure.in
===================================================================
--- linphone-3.2.1.orig/configure.in
+++ linphone-3.2.1/configure.in
@@ -129,7 +129,7 @@ fi
 dnl os-specific problems not handled by existing macros.
 case "$host_os" in
 	*freebsd*)
-		LDFLAGS="$LDFLAGS -pthread"
+		LDFLAGS="$LDFLAGS -lpthread"
 		;;
 esac
 
Index: linphone-3.2.1/configure
===================================================================
--- linphone-3.2.1.orig/configure
+++ linphone-3.2.1/configure
@@ -15303,7 +15303,7 @@ fi
 
 case "$host_os" in
 	*freebsd*)
-		LDFLAGS="$LDFLAGS -pthread"
+		LDFLAGS="$LDFLAGS -lpthread"
 		;;
 esac
 
Index: linphone-3.2.1/mediastreamer2/configure.ac
===================================================================
--- linphone-3.2.1.orig/mediastreamer2/configure.ac
+++ linphone-3.2.1/mediastreamer2/configure.ac
@@ -142,10 +142,10 @@ dnl ORTP_STATIC to tell ortp not to expo
 		mingw_found=yes
 	;;
   *)
-     MSPLUGINS_CFLAGS="-pthread"
-     MSPLUGINS_LIBS="-shared -pthread"
-     CFLAGS="$CFLAGS -pthread -D_REENTRANT"
-     LIBS="$LIBS -pthread -lpthread"
+     MSPLUGINS_CFLAGS="-lpthread"
+     MSPLUGINS_LIBS="-shared -lpthread"
+     CFLAGS="$CFLAGS -lpthread -D_REENTRANT"
+     LIBS="$LIBS -lpthread -lpthread"
      ;;
 esac
 
Index: linphone-3.2.1/oRTP/configure.ac
===================================================================
--- linphone-3.2.1.orig/oRTP/configure.ac
+++ linphone-3.2.1/oRTP/configure.ac
@@ -96,7 +96,7 @@ case "$target_os" in
 		;;
 	*freebsd*)
 		AC_DEFINE(NOCONNECT,1,[Defined if we should not use connect() on udp sockets])
-		PTHREAD_LDFLAGS="-pthread"
+		PTHREAD_LDFLAGS="-lpthread"
 		;;
 	*mingw*)
 		CFLAGS="$CFLAGS -D_WIN32_WINNT=0x0501 -DORTP_STATIC"
