Index: linphone-3.2.1/configure
===================================================================
--- linphone-3.2.1.orig/configure
+++ linphone-3.2.1/configure
@@ -1036,7 +1036,7 @@ sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
 includedir='${prefix}/include'
 oldincludedir='/usr/include'
-docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
+docdir='${datarootdir}/doc/${PACKAGE_NAME}'
 infodir='${datarootdir}/info'
 htmldir='${docdir}'
 dvidir='${docdir}'
Index: linphone-3.2.1/mediastreamer2/configure
===================================================================
--- linphone-3.2.1.orig/mediastreamer2/configure
+++ linphone-3.2.1/mediastreamer2/configure
@@ -1059,7 +1059,7 @@ sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
 includedir='${prefix}/include'
 oldincludedir='/usr/include'
-docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
+docdir='${datarootdir}/doc/${PACKAGE_NAME}'
 infodir='${datarootdir}/info'
 htmldir='${docdir}'
 dvidir='${docdir}'
Index: linphone-3.2.1/mediastreamer2/help/Makefile.in
===================================================================
--- linphone-3.2.1.orig/mediastreamer2/help/Makefile.in
+++ linphone-3.2.1/mediastreamer2/help/Makefile.in
@@ -258,7 +258,7 @@ EXTRA_DIST = Doxyfile.in doxygen.dox.in
 
 #html doc
 @HAVE_DOXYGEN_TRUE@SOURCES = $(top_srcdir)/include/mediastreamer2/*.h $(top_srcdir)/src/*.c $(top_srcdir)/src/*.h
-@HAVE_DOXYGEN_TRUE@pkgdocdir = $(docdir)/$(PACKAGE)-$(VERSION)
+@HAVE_DOXYGEN_TRUE@pkgdocdir = $(docdir)/$(PACKAGE)
 @HAVE_DOXYGEN_TRUE@doc_htmldir = $(pkgdocdir)/html
 @HAVE_DOXYGEN_TRUE@doc_html_DATA = $(top_builddir)/help/doc/html/html.tar
 all: all-am
Index: linphone-3.2.1/oRTP/configure
===================================================================
--- linphone-3.2.1.orig/oRTP/configure
+++ linphone-3.2.1/oRTP/configure
@@ -951,7 +951,7 @@ sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
 includedir='${prefix}/include'
 oldincludedir='/usr/include'
-docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
+docdir='${datarootdir}/doc/${PACKAGE_NAME}'
 infodir='${datarootdir}/info'
 htmldir='${docdir}'
 dvidir='${docdir}'
Index: linphone-3.2.1/oRTP/Makefile.am
===================================================================
--- linphone-3.2.1.orig/oRTP/Makefile.am
+++ linphone-3.2.1/oRTP/Makefile.am
@@ -14,7 +14,7 @@ if HAVE_DOXYGEN
 
 # doxdir & pkgdocdir are not always defined by automake
 docdir=$(datadir)/doc
-pkgdocdir=$(docdir)/$(PACKAGE)-$(VERSION)
+pkgdocdir=$(docdir)/$(PACKAGE)
 doc_htmldir=$(pkgdocdir)/html
 
 doc_html_DATA = $(top_builddir)/doc/html/html.tar
Index: linphone-3.2.1/mediastreamer2/help/Makefile.am
===================================================================
--- linphone-3.2.1.orig/mediastreamer2/help/Makefile.am
+++ linphone-3.2.1/mediastreamer2/help/Makefile.am
@@ -8,7 +8,7 @@ SOURCES=$(top_srcdir)/include/mediastrea
 
 # doxdir & pkgdocdir are not always defined by automake
 docdir=$(datadir)/doc
-pkgdocdir=$(docdir)/$(PACKAGE)-$(VERSION)
+pkgdocdir=$(docdir)
 doc_htmldir=$(pkgdocdir)/html
 
 doc_html_DATA = $(top_builddir)/help/doc/html/html.tar
@@ -29,4 +29,4 @@ uninstall-hook:
 endif
 
 clean-local:
-	rm -rf doc
\ Dosya sonunda yenisatır yok.
+	rm -rf doc
