Index: linphone-3.3.2/share/linphone.desktop.in
===================================================================
--- linphone-3.3.2.orig/share/linphone.desktop.in
+++ linphone-3.3.2/share/linphone.desktop.in
@@ -1,10 +1,13 @@
 [Desktop Entry]
 Name=Linphone
+GenericName=VoIP application
+GenericName[tr]=VoIP uygulaması
 Comment=Linphone is a web-phone
 Comment[fr]=Linphone est un web-phone.
 Comment[de]=Linphone ist ein web-phone.
+Comment[tr]=Linphone bir ağ telefonudur.
 Type=Application
 Exec=linphone-3
 Icon=@prefix@/share/pixmaps/linphone/linphone.png
 Terminal=false
-Categories=Network;Telephony;
\ Dosya sonunda yenisatır yok.
+Categories=Application;Network;
