diff -urN linphone-3.5.2.orig/mediastreamer2/configure.ac linphone-3.5.2/mediastreamer2/configure.ac
--- linphone-3.5.2.orig/mediastreamer2/configure.ac	2012-10-11 09:22:13.545254214 +0700
+++ linphone-3.5.2/mediastreamer2/configure.ac	2012-10-11 09:23:08.812253473 +0700
@@ -625,8 +625,8 @@
 AC_SUBST(PACKAGE_DATA_DIR)
 
 dnl check for video4linux headers
-AC_CHECK_HEADERS(linux/videodev.h linux/videodev2.h)
-if test "${ac_cv_header_linux_videodev_h}" = "yes" ; then
+AC_CHECK_HEADERS(libv4l1-videodev.h linux/videodev2.h)
+if test "${ac_cv_header_libv4l1_videodev_h}" = "yes" ; then
 	found_v4l1=yes
 else
 	found_v4l1=no
diff -urN linphone-3.5.2.orig/mediastreamer2/src/msv4l.c linphone-3.5.2/mediastreamer2/src/msv4l.c
--- linphone-3.5.2.orig/mediastreamer2/src/msv4l.c	2012-10-11 09:22:13.540254214 +0700
+++ linphone-3.5.2/mediastreamer2/src/msv4l.c	2012-10-11 09:24:25.505252447 +0700
@@ -30,7 +30,7 @@
 #include <string.h>
 #include <sys/mman.h>
 
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #ifdef HAVE_LINUX_VIDEODEV2_H
 #include <linux/videodev2.h>
 #endif
@@ -48,6 +48,9 @@
 /* Set enable workaround for bugs, bitfield */
 #define VIDIOCQCSCOMPATIBLE _IOWR('v',QC_IOCTLBASE+10,int)
 
+#ifndef BASE_VIDIOCPRIVATE
+#define BASE_VIDIOCPRIVATE 192
+#endif
 #ifndef VIDIOSFPS
 #define VIDIOSFPS _IOW('v',BASE_VIDIOCPRIVATE+20, int)
 #endif
