--- gtk-glade/Makefile.am.orig	2009-09-07 15:55:22.000000000 +0200
+++ gtk-glade/Makefile.am	2009-09-19 03:29:08.000000000 +0200
@@ -41,7 +41,7 @@
 			loginframe.c \
 			linphone.h 
 
-linphone_3_LDADD=$(top_builddir)/oRTP/src/libortp.la \
+linphone_3_LDADD=$(ORTP_LIBS) \
 		$(top_builddir)/mediastreamer2/src/libmediastreamer.la \
 		$(top_builddir)/coreapi/liblinphone.la \
 			$(LIBGTK_LIBS) $(INTLLIBS) \
