Index: ImageMagick-6.5.4-10/configure.ac
===================================================================
--- ImageMagick-6.5.4-10.orig/configure.ac
+++ ImageMagick-6.5.4-10/configure.ac
@@ -103,9 +103,6 @@ AM_SANITY_CHECK
 
 AM_INIT_AUTOMAKE($PACKAGE_NAME,"${PACKAGE_VERSION}${PACKAGE_LIB_SUBVERSION}", ' ')
 
-# Enable support for silent build rules
-AM_SILENT_RULES([yes])
-
 MAGICK_LIB_VERSION="0x"
 if test ${MAGICK_LIBRARY_CURRENT} -lt 10 ; then
   MAGICK_LIB_VERSION=${MAGICK_LIB_VERSION}0
