diff -up xsane-0.996/src/xsane.c.no-eula xsane-0.996/src/xsane.c
--- xsane-0.996/src/xsane.c.no-eula	2009-07-21 15:33:00.927455229 +0200
+++ xsane-0.996/src/xsane.c	2009-07-21 15:39:28.661456472 +0200
@@ -3524,10 +3524,13 @@ static void xsane_about_dialog(GtkWidget
   snprintf(buf, sizeof(buf), "XSane %s %s\n"
                              "%s %s\n"
                              "\n"
+                             "%s\n%s"
+                             "\n\n"
                              "%s %s\n"
                              "%s %s\n",
                              TEXT_VERSION, XSANE_VERSION,
                              XSANE_COPYRIGHT_SIGN, XSANE_COPYRIGHT_TXT,
+                             TEXT_MODIFIED_BLURB, XSANE_BUGTRACKER_URL,
                              TEXT_HOMEPAGE, XSANE_HOMEPAGE,
                              TEXT_EMAIL_ADR, XSANE_EMAIL_ADR);
 
@@ -5714,6 +5717,7 @@ static int xsane_init(int argc, char **a
 
         case 'v': /* --version */
           g_print("%s-%s %s %s\n", xsane.prog_name, XSANE_VERSION, XSANE_COPYRIGHT_SIGN, XSANE_COPYRIGHT_TXT);
+          g_print("\n%s\n%s\n\n", TEXT_MODIFIED_BLURB, XSANE_BUGTRACKER_URL);
           g_print("  %s %s\n", TEXT_EMAIL_ADR, XSANE_EMAIL_ADR);
           g_print("  %s %s\n", TEXT_PACKAGE, XSANE_PACKAGE_VERSION);
           g_print("  %s%d.%d.%d\n", TEXT_GTK_VERSION, GTK_MAJOR_VERSION, GTK_MINOR_VERSION, GTK_MICRO_VERSION);
@@ -5840,17 +5844,9 @@ static int xsane_init(int argc, char **a
   }
 
 
-  if (xsane_pref_restore()) /* restore preferences, returns TRUE if license is not accpted yet */
+  if (xsane_pref_restore()) /* restore preferences, returns TRUE if the version is different from the last run */
   {
-    if (xsane_display_eula(1)) /* show license and ask for accept/not accept */
-    {
-      DBG(DBG_info, "user did not accept eula, we abort\n");
-      return 1; /* User did not accept eula */
-    }
-    else /* User did accept eula */
-    {
-      xsane_pref_save();
-    }
+    xsane_pref_save();
   }
 
   xsane_pref_restore_media();
diff -up xsane-0.996/src/xsane.h.no-eula xsane-0.996/src/xsane.h
--- xsane-0.996/src/xsane.h.no-eula	2009-07-21 15:33:00.921470546 +0200
+++ xsane-0.996/src/xsane.h	2009-07-21 16:08:01.398707123 +0200
@@ -98,6 +98,9 @@
 #define XSANE_EMAIL_ADR		"Oliver.Rauch@xsane.org"
 #define XSANE_HOMEPAGE		"http://www.xsane.org"
 #define XSANE_COPYRIGHT_TXT	XSANE_DATE " " XSANE_COPYRIGHT
+#ifndef XSANE_BUGTRACKER_URL
+#define XSANE_BUGTRACKER_URL             "(no bug tracker configured)"
+#endif
 
 /* ---------------------------------------------------------------------------------------------------------------------- */
 
diff -up xsane-0.996/src/xsane-text.h.no-eula xsane-0.996/src/xsane-text.h
--- xsane-0.996/src/xsane-text.h.no-eula	2007-08-13 09:16:43.000000000 +0200
+++ xsane-0.996/src/xsane-text.h	2009-07-21 15:42:00.609707360 +0200
@@ -230,6 +230,8 @@
                                           "This program is distributed in the hope that it will be useful, but\n" \
                                           "WITHOUT ANY WARRANTY; without even the implied warranty of\n" \
                                           "MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.\n")
+#define TEXT_MODIFIED_BLURB             _("This package is modified from the original version.\n" \
+                                          "Please contact your vendor or report problems at")
 #define TEXT_EMAIL_ADR					_("E-mail:")
 #define TEXT_HOMEPAGE					_("Homepage:")
 #define TEXT_FILE					_("File:")
