diff -ur kdebase-4.4.0/apps/konqueror/src/konqmainwindow.cpp kdebase-4.4.0-konqueror-kde#228593/apps/konqueror/src/konqmainwindow.cpp
--- kdebase-4.4.0/apps/konqueror/src/konqmainwindow.cpp	2010-01-20 00:18:15.000000000 +0100
+++ kdebase-4.4.0-konqueror-kde#228593/apps/konqueror/src/konqmainwindow.cpp	2010-02-26 12:04:27.000000000 +0100
@@ -1824,9 +1824,17 @@
 
 void KonqMainWindow::slotConfigureSpellChecking()
 {
-    Sonnet::ConfigDialog dialog( KGlobal::config().data(), this);
+    KSharedConfig::Ptr config = KGlobal::config();
+    KConfigGroup htmlSettings = config->group( "HTML Settings" );
+    KConfigGroup spellingSettings = config->group( "Spelling" );
+    // Copy KHTML setting to Sonnet setting for the dialog
+    spellingSettings.writeEntry( "checkerEnabledByDefault", htmlSettings.readEntry( "AutoSpellCheck", true ) );
+    Sonnet::ConfigDialog dialog( config.data(), this);
     dialog.setWindowIcon( KIcon( "konqueror" ));
     dialog.exec();
+    // Copy Sonnet setting from the dialog to KHTML setting
+    htmlSettings.writeEntry( "AutoSpellCheck", spellingSettings.readEntry( "checkerEnabledByDefault", false ) );
+    reparseConfiguration();
 }
 
 void KonqMainWindow::slotConfigureToolbars()
