diff -Naupr zziplib-0.13.59.orig/zzip/_config.h zziplib-0.13.59/zzip/_config.h
--- zziplib-0.13.59.orig/zzip/_config.h	2010-02-14 22:29:34.000000000 +0100
+++ zziplib-0.13.59/zzip/_config.h	2010-12-04 16:22:19.809534085 +0100
@@ -133,7 +133,11 @@
 
 /* whether the system defaults to 32bit off_t but can do 64bit when requested
    */
-/* #undef LARGEFILE_SENSITIVE */
+#if __WORDSIZE == 32
+#ifndef ZZIP_LARGEFILE_SENSITIVE 
+#define ZZIP_LARGEFILE_SENSITIVE  1 
+#endif
+#endif
 
 /* Define to the sub-directory in which libtool stores uninstalled libraries.
    */
@@ -178,8 +181,12 @@
 
 /* The number of bytes in type long */
 #ifndef ZZIP_SIZEOF_LONG 
+#if __WORDSIZE == 32
+#define ZZIP_SIZEOF_LONG  4
+#elif __WORDSIZE == 64
 #define ZZIP_SIZEOF_LONG  8 
 #endif
+#endif
 
 /* The number of bytes in type short */
 #ifndef ZZIP_SIZEOF_SHORT 
@@ -209,7 +216,11 @@
 #endif
 
 /* Number of bits in a file offset, on hosts where this is settable. */
-/* #undef _FILE_OFFSET_BITS */
+#if __WORDSIZE == 32
+#ifndef ZZIP__FILE_OFFSET_BITS 
+#define ZZIP__FILE_OFFSET_BITS  64 
+#endif
+#endif
 
 /* Define for large files, on AIX-style hosts. */
 /* #undef _LARGE_FILES */
