--- a/ext/html/html.defs	2008-08-17 16:40:39.000000000 +0200
+++ b/ext/html/html.defs	2008-08-17 16:41:10.000000000 +0200
@@ -103,18 +103,6 @@
   (return-type "none")
 )
 
-(define-method request_paste
-  (of-object "GtkHTML")
-  (c-name "gtk_html_request_paste")
-  (return-type "int")
-  (parameters
-    '("GdkAtom" "selection")
-    '("gint" "type")
-    '("gint32" "time")
-    '("gboolean" "as_cite")
-  )
-)
-
 (define-method begin
   (of-object "GtkHTML")
   (c-name "gtk_html_begin")
--- a/ext/html/config.m4	2008-08-17 16:44:57.000000000 +0200
+++ b/ext/html/config.m4	2008-08-17 16:45:05.000000000 +0200
@@ -7,7 +7,7 @@
 [  --with-html             Enable GtkHtml support],no)
 
 if test "$PHP_GTK_HTML" != "no"; then
-  PKG_CHECK_MODULES(HTML, [libgtkhtml-3.8 >= html_required_version],
+  PKG_CHECK_MODULES(HTML, [libgtkhtml-3.14 >= html_required_version],
   have_html=yes, have_html=no)
   if test "$have_html" != "yes"; then
     AC_MSG_RESULT([Unable to locate libgtkhtml version html_required_version or higher: not building])
