Index: meterbridge-0.9.2/src/Makefile.am
===================================================================
--- meterbridge-0.9.2.orig/src/Makefile.am
+++ meterbridge-0.9.2/src/Makefile.am
@@ -1,5 +1,5 @@
 #CFLAGS = -O6 `sdl-config --cflags` -DPKG_DATA_DIR=\"$(pkgdatadir)\"
-CFLAGS = -Wall -g $(SDL_CFLAGS) $(JACK_CFLAGS) -DPKG_DATA_DIR=\"$(pkgdatadir)\"
+AM_CFLAGS = -Wall -g $(SDL_CFLAGS) $(JACK_CFLAGS) -DPKG_DATA_DIR=\"$(pkgdatadir)\"
 
 bin_PROGRAMS = meterbridge
 
