commit 16864159654cb6c77b3db36dd218e4103e3bd0c2
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Thu Jun 3 09:23:45 2010 +0200

    Don't install test programs.

diff --git a/test/basic/Makefile.am b/test/basic/Makefile.am
index a2519a2..8022a36 100644
--- a/test/basic/Makefile.am
+++ b/test/basic/Makefile.am
@@ -20,7 +20,7 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-bin_PROGRAMS = test_01 test_02 test_03 test_04 test_05 test_06 \
+check_PROGRAMS = test_01 test_02 test_03 test_04 test_05 test_06 \
 		test_07 test_08 test_09 test_10 test_11
 
 AM_CFLAGS = -I$(top_srcdir)/va -I$(top_srcdir)/src/x11
diff --git a/test/decode/Makefile.am b/test/decode/Makefile.am
index ea30643..23a086f 100644
--- a/test/decode/Makefile.am
+++ b/test/decode/Makefile.am
@@ -20,7 +20,7 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-bin_PROGRAMS = mpeg2vldemo
+check_PROGRAMS = mpeg2vldemo
 
 AM_CFLAGS = -I$(top_srcdir)/va -I$(top_srcdir)/src/x11
 
diff --git a/test/encode/Makefile.am b/test/encode/Makefile.am
index 80c882a..98b2bc4 100644
--- a/test/encode/Makefile.am
+++ b/test/encode/Makefile.am
@@ -20,7 +20,7 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-bin_PROGRAMS = h264encode
+check_PROGRAMS = h264encode
 
 AM_CFLAGS = -I$(top_srcdir)/va -I$(top_srcdir)/src/x11
 
diff --git a/test/putsurface/Makefile.am b/test/putsurface/Makefile.am
index fe18388..d06006c 100644
--- a/test/putsurface/Makefile.am
+++ b/test/putsurface/Makefile.am
@@ -20,7 +20,7 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-bin_PROGRAMS = putsurface
+check_PROGRAMS = putsurface
 
 AM_CFLAGS = -I$(top_srcdir)/va -I$(top_srcdir)/src/x11
 
