commit c8668dc9fb4836c64fdc79eb1c340cb30e6c36ae
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Thu Jan 27 11:03:18 2011 +0100

    Propagate VA-API version to the driver.

diff --git a/va/va.c b/va/va.c
index 9492ac0..17f6779 100644
--- a/va/va.c
+++ b/va/va.c
@@ -240,6 +240,9 @@ static VAStatus va_openDriver(VADisplay dpy, char *driver_name)
                 }
                 ctx->vtable = vtable;
 
+                ctx->version_major = VA_MAJOR_VERSION;
+                ctx->version_minor = VA_MINOR_VERSION;
+
                 if (VA_STATUS_SUCCESS == vaStatus)
                     vaStatus = (*init_func)(ctx);
 
