diff -Naur drumstick-0.5.0.orig/drumstick.xml drumstick-0.5.0/drumstick.xml
--- drumstick-0.5.0.orig/drumstick.xml	2010-09-09 13:38:56.000000000 +0300
+++ drumstick-0.5.0/drumstick.xml	2011-01-26 15:30:09.748000036 +0200
@@ -6,6 +6,7 @@
     <comment>Cakewalk project file</comment>
     <comment xml:lang="en_GB">Cakewalk project file</comment>
     <comment xml:lang="es">Archivo de proyecto Cakewalk</comment>
+    <comment xml:lang="tr">Cakewalk proje dosyası</comment>
     <magic priority="50">
       <match value="CAKEWALK\x1a" type="string" offset="0"/>
     </magic>
@@ -17,6 +18,7 @@
     <comment>Overture project file</comment>
     <comment xml:lang="en_GB">Overture project file</comment>
     <comment xml:lang="es">Archivo de proyecto Overture</comment>
+    <comment xml:lang="tr">Overture proje dosyası</comment>
     <magic priority="50">
       <match value="OVSC\0\0\0" type="string" offset="0"/>
     </magic>
diff -Naur drumstick-0.5.0.orig/utils/drumgrid/drumstick-drumgrid.desktop drumstick-0.5.0/utils/drumgrid/drumstick-drumgrid.desktop
--- drumstick-0.5.0.orig/utils/drumgrid/drumstick-drumgrid.desktop	2010-09-09 13:38:53.000000000 +0300
+++ drumstick-0.5.0/utils/drumgrid/drumstick-drumgrid.desktop	2011-01-26 15:19:34.756000036 +0200
@@ -1,9 +1,15 @@
 [Desktop Entry]
+Type=Application
+Version=1.0
+Encoding=UTF-8
 Name=Drumstick Drum Grid
-Exec=drumstick-drumgrid
+Name[tr]=Drumstick Davul Izgarası
+GenericName=Drum Grid
+GenericName[tr]=Davul Izgarası
+Comment=Drumstick Drum Grid
+Comment[es]=Caja de Ritmos de Drumstick
+Comment[es]=Drumstick Davul Izgarası
 Icon=drumstick
+Exec=drumstick-drumgrid
 Terminal=false
-Type=Application
 Categories=AudioVideo;Audio;Midi;Education;Music;
-Comment=Drumstick Drum Grid
-Comment[es]=Caja de Ritmos de Drumstick 
diff -Naur drumstick-0.5.0.orig/utils/guiplayer/drumstick-guiplayer.desktop drumstick-0.5.0/utils/guiplayer/drumstick-guiplayer.desktop
--- drumstick-0.5.0.orig/utils/guiplayer/drumstick-guiplayer.desktop	2010-09-09 13:38:53.000000000 +0300
+++ drumstick-0.5.0/utils/guiplayer/drumstick-guiplayer.desktop	2011-01-26 15:53:41.833000037 +0200
@@ -1,10 +1,16 @@
 [Desktop Entry]
+Type=Application
+Version=1.0
+Encoding=UTF-8
 Name=Drumstick MIDI Player
-Exec=drumstick-guiplayer
+Name[tr]=Drumstick MIDI Çalar
+GenericName=MIDI Player
+GenericName[tr]=MIDI Çalar
+Comment=Drumstick MIDI Player
+Comment[es]=Reproductor MIDI de Drumstick
+Comment[tr]=Drumstick MIDI Çalar
 Icon=drumstick
+Exec=drumstick-guiplayer
 Terminal=false
-Type=Application
-Categories=AudioVideo;Audio;Midi;Education;Music;
 MimeType=audio/midi;audio/x-midi;audio/cakewalk;audio/overture;
-Comment=Drumstick MIDI Player
-Comment[es]=Reproductor MIDI de Drumstick  
+Categories=AudioVideo;Audio;Midi;Education;Music;
diff -Naur drumstick-0.5.0.orig/utils/vpiano/drumstick-vpiano.desktop drumstick-0.5.0/utils/vpiano/drumstick-vpiano.desktop
--- drumstick-0.5.0.orig/utils/vpiano/drumstick-vpiano.desktop	2010-09-09 13:38:54.000000000 +0300
+++ drumstick-0.5.0/utils/vpiano/drumstick-vpiano.desktop	2011-01-26 15:28:11.092000035 +0200
@@ -1,9 +1,15 @@
 [Desktop Entry]
+Type=Application
+Version=1.0
+Encoding=UTF-8
 Name=Drumstick Virtual Piano
-Exec=drumstick-vpiano
+Name[tr]=Drumstick Sanal Piyano
+GenericName=Virtual Piano
+GenericName[tr]=Sanal Piyano
+Comment=Drumstick Virtual Piano
+Comment[es]=Piano virtual de Drumstick
+Comment[tr]=Drumstick Sanal Piyano
 Icon=drumstick
+Exec=drumstick-vpiano
 Terminal=false
-Type=Application
 Categories=AudioVideo;Audio;Midi;Education;Music;
-Comment=Drumstick Virtual Piano
-Comment[es]=Piano virtual de Drumstick  
