--- lynx2-8-7/userdefs.h.orig	2009-06-07 02:30:35.000000000 +0200
+++ lynx2-8-7/userdefs.h	2010-01-03 14:41:42.000000000 +0100
@@ -115,8 +115,8 @@
  * Mappings in these global and personal files override any SUFFIX
  * definitions in lynx.cfg and built-in defaults from src/HTInit.c.
  */
-#define GLOBAL_EXTENSION_MAP "Lynx_Dir:mime.types"
-#define PERSONAL_EXTENSION_MAP "mime.types"
+#define GLOBAL_EXTENSION_MAP "/etc/mime.types"
+#define PERSONAL_EXTENSION_MAP ".mime.types"
 
 /**************************
  * The MAILCAP file allows you to map file MIME types to
@@ -125,7 +125,7 @@
  * Mappings in these global and personal files override any VIEWER
  * definitions in lynx.cfg and built-in defaults from src/HTInit.c.
  */
-#define GLOBAL_MAILCAP "Lynx_Dir:mailcap"
+#define GLOBAL_MAILCAP "/etc/mailcap"
 #define PERSONAL_MAILCAP ".mailcap"
 
 /**************************
--- lynx2-8-7/lynx.cfg.orig	2009-06-24 01:53:58.000000000 +0200
+++ lynx2-8-7/lynx.cfg	2010-01-03 14:41:42.000000000 +0100
@@ -12,7 +12,7 @@
 # Definition pairs are of the form  VARIABLE:DEFINITION
 # NO spaces are allowed between the pair items.
 #
-#  If you do not have write access to /usr/local/lib you may change
+#  If you do not have write access to /usr/bin you may change
 #  the default location of this file in the userdefs.h file and recompile,
 #  or specify its location on the command line with the "-cfg"
 #  command line option.
