--- lynx2-8-6/WWW/Library/Implementation/HTTelnet.c.pix	2006-05-30 00:44:54.000000000 +0200
+++ lynx2-8-6/WWW/Library/Implementation/HTTelnet.c	2006-11-16 17:27:19.000000000 +0100
@@ -64,6 +64,12 @@
       strcmp(acc_method, "rlogin") == 0 ? rlogin :
       strcmp(acc_method, "tn3270") == 0 ? tn3270 : telnet;
 
+	/* don't accept command line args to telnet */
+	/* notting@redhat.com, 8-25-99 */
+	for (; *host && *host =='-' ; host ++) {
+		*host = '\0';
+	}
+
     /*
      * Modified to allow for odd chars in a username only if exists.
      * 05-28-94 Lynx 2-3-1 Garrett Arch Blythe
