diff -Nur pyclutter-gst-old/clutter-gst/Makefile.am pyclutter-gst/clutter-gst/Makefile.am
--- pyclutter-gst-old/clutter-gst/Makefile.am	2009-10-09 22:18:02.980898681 +0300
+++ pyclutter-gst/clutter-gst/Makefile.am	2009-10-09 22:19:56.988878377 +0300
@@ -18,7 +18,7 @@
 						 -avoid-version \
 						 -export-symbols-regex init_cluttergst
 
-_cluttergst_la_LIBADD = $(CLUTTERGST_LIBS)
+_cluttergst_la_LIBADD = $(CLUTTERGST_LIBS) -lpardusPythonVersion
 _cluttergst_la_SOURCES = cluttergstmodule.c
 nodist__cluttergst_la_SOURCES = cluttergst.c
 
