diff -ur imposter-0.3.orig/src/ui.c imposter-0.3/src/ui.c
--- imposter-0.3.orig/src/ui.c	2005-04-01 16:41:44.000000000 +0300
+++ imposter-0.3/src/ui.c	2005-04-08 13:39:33.000000000 +0300
@@ -4,6 +4,9 @@
 ** modify it under the terms of GNU General Public License.
 */
 
+ /* Ignore some modifier keys in keyboard handling. */
+ #define imposter_ignore_modifiers(a) ((a) &(~(GDK_LOCK_MASK|GDK_MOD2_MASK)))
+
 #include "common.h"
 
 static GtkWidget *menu_bar, *window, *area;
@@ -43,7 +46,7 @@
 	char *key;
 	int i;
 
-	if (ev->state) return FALSE;
+	if (imposter_ignore_modifiers(ev->state)) return FALSE;
 	key = gdk_keyval_name (ev->keyval);
 	if (key) {
 		for (i = 0; defkeys[i].key; i++) {
