diff -Nuar src.OSmanOS/src/kdesvn.desktop src/src/kdesvn.desktop
--- src.OSmanOS/src/kdesvn.desktop	2010-01-27 23:30:47.000000000 +0200
+++ src/src/kdesvn.desktop	2013-12-28 12:15:34.747206591 +0200
@@ -4,6 +4,7 @@
 GenericName[de]=Graphische Oberfläche für SVN
 GenericName[es]=Cliente SVN
 GenericName[lt]=SVN klientas
+GenericName[tr]=SVN İstemcisi
 Exec=kdesvn -caption "%c"
 Icon=kdesvn
 Type=Application
@@ -13,6 +14,7 @@
 Comment[nl]=Een Subversion-client voor KDE
 Comment[es]=Un cliente de Subversion para KDE
 Comment[lt]=Subversion klientas skirtas KDE
+Comment[tr]=KDE için Subversion İstemcisi
 Terminal=false
 MimeType=inode/directory;
 Categories=Qt;KDE;Development;RevisionControl;
