diff -p -up ./configure.ac.tv2 ./configure.ac
--- ./configure.ac.tv2	2012-08-21 11:50:07.844725573 +0200
+++ ./configure.ac	2012-08-21 11:50:18.249052116 +0200
@@ -263,8 +263,8 @@ if test "x$gc_ok" = "xyes" && test "x$cr
 			# include <gc.h>
 			#endif
 			#include <stdio.h>
-			extern unsigned GC_version;
 			int main(void){
+				unsigned GC_version = GC_get_version();
 				unsigned min = ((6 << 16) | (4 << 8) | 0);
 				printf("%d.%d.%d ",GC_version >> 16, (GC_version >> 8) & 0xFF, GC_version & 0xFF);
 				if (GC_version>=min) return 0;
diff -p -up ./configure.tv2 ./configure
--- ./configure.tv2	2012-08-21 11:50:10.157022067 +0200
+++ ./configure	2012-08-21 11:50:30.233540384 +0200
@@ -8933,8 +8933,8 @@ else
 			# include <gc.h>
 			#endif
 			#include <stdio.h>
-			extern unsigned GC_version;
 			int main(void){
+				unsigned GC_version = GC_get_version();
 				unsigned min = ((6 << 16) | (4 << 8) | 0);
 				printf("%d.%d.%d ",GC_version >> 16, (GC_version >> 8) & 0xFF, GC_version & 0xFF);
 				if (GC_version>=min) return 0;

