From: Lubomir Rintel <lkundrak@v3.sk>

https://bugzilla.redhat.com/show_bug.cgi?id=458845

Index: inkscape-0.48.2/inkscape.desktop.in
===================================================================
--- inkscape-0.48.2.orig/inkscape.desktop.in
+++ inkscape-0.48.2/inkscape.desktop.in
@@ -6,7 +6,7 @@ _X-GNOME-FullName=Inkscape Vector Graphi
 _Comment=Create and edit Scalable Vector Graphics images
 Type=Application
 Categories=Graphics;VectorGraphics;GTK;
-MimeType=image/svg+xml;image/svg+xml-compressed;application/vnd.corel-draw;application/pdf;application/postscript;image/x-eps;application/illustrator;
+MimeType=image/svg+xml;image/svg+xml-compressed;application/vnd.corel-draw;application/pdf;application/postscript;image/x-eps;application/illustrator;image/cgm;image/x-wmf;application/x-xccx;application/x-xcgm;application/x-xcdt;application/x-xsk1;application/x-xcmx;image/x-xcdr;
 Exec=inkscape %F
 TryExec=inkscape
 Terminal=false
