*** src/iop/watermark.c~	2012-11-14 09:27:37.876647504 +0100
--- src/iop/watermark.c	2012-11-14 09:30:37.987679299 +0100
***************
*** 38,44 ****
--- 38,47 ----
  #include <inttypes.h>
  
  #include <librsvg/rsvg.h>
+ // ugh, ugly hack. why do people break stuff all the time? - http://www.darktable.org/redmine/issues/8931
+ #ifndef RSVG_CAIRO_H
  #include <librsvg/rsvg-cairo.h>
+ #endif
  
  #include "common/metadata.h"
  #include "common/utility.h"
