# Glade wants psppire libs under /usr/lib :)
--- pspp-0.6.0/Makefile.in.org     2008-08-09 02:42:03.000000000 +0300
+++ pspp-0.6.0/Makefile.in 2008-08-09 02:42:19.000000000 +0300
@@ -21,7 +21,7 @@

 VPATH = @srcdir@
 pkgdatadir = $(datadir)/@PACKAGE@
-pkglibdir = $(libdir)/@PACKAGE@
+pkglibdir = $(libdir)
 pkgincludedir = $(includedir)/@PACKAGE@
 am__cd = CDPATH="$${ZSH_VERSION+.}$(PATH_SEPARATOR)" && cd
 install_sh_DATA = $(install_sh) -c -m 644

