From c0261181eba678bf042191e034434f7e7545437b Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Thu, 23 Sep 2010 13:44:03 +0400
Subject: [PATCH 1/1] Remove gtk_entry_reset_im_context entirely

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 lib/gtksheet/gtkitementry.c |   11 -----------
 1 files changed, 0 insertions(+), 11 deletions(-)

diff --git a/lib/gtksheet/gtkitementry.c b/lib/gtksheet/gtkitementry.c
index 703215c..45818c2 100644
--- a/lib/gtksheet/gtkitementry.c
+++ b/lib/gtksheet/gtkitementry.c
@@ -145,7 +145,6 @@ static void         gtk_entry_draw_cursor              (GtkEntry       *entry,
 static PangoLayout *gtk_entry_ensure_layout            (GtkEntry       *entry,
                                                         gboolean        include_preedit);
 static void         gtk_entry_queue_draw               (GtkEntry       *entry);
-static void         gtk_entry_reset_im_context         (GtkEntry       *entry);
 static void         gtk_entry_recompute                (GtkEntry       *entry);
 static void         gtk_entry_get_cursor_locations     (GtkEntry       *entry,
 							CursorType      type,
@@ -1777,16 +1776,6 @@ gtk_entry_queue_draw (GtkEntry *entry)
 }
 
 static void
-gtk_entry_reset_im_context (GtkEntry *entry)
-{
-  if (entry->need_im_reset)
-    {
-      entry->need_im_reset = 0;
-      gtk_im_context_reset (entry->im_context);
-    }
-}
-
-static void
 gtk_entry_get_cursor_locations (GtkEntry   *entry,
 				CursorType  type,
 				gint       *strong_x,
-- 
1.7.2.3

