--- setup.py.orig	2010-03-04 15:32:04.000000000 +0200
+++ setup.py	2010-03-18 21:55:36.994040152 +0200
@@ -19,17 +19,17 @@
                  ('datas/fonts',glob.glob('datas/fonts/*.*')),
                  ('datas/sounds',glob.glob('datas/sounds/*.*')),
                  ('datas/locale',glob.glob('datas/locale/*.*')),
-                 ('datas/locale/tr/LC_MESSAGES',glob.glob('datas/locale/tr/LC_MESSAGES/*.*')),
-                 ('datas/locale/az/LC_MESSAGES',glob.glob('datas/locale/az/LC_MESSAGES/*.*')),
-                 ('datas/locale/de/LC_MESSAGES',glob.glob('datas/locale/de/LC_MESSAGES/*.*')),
-                 ('datas/locale/es/LC_MESSAGES',glob.glob('datas/locale/es/LC_MESSAGES/*.*')),
-                 ('datas/locale/fr/LC_MESSAGES',glob.glob('datas/locale/fr/LC_MESSAGES/*.*')),
-                 ('datas/locale/ind/LC_MESSAGES',glob.glob('datas/locale/ind/LC_MESSAGES/*.*')),
-                 ('datas/locale/ir/LC_MESSAGES',glob.glob('datas/locale/ir/LC_MESSAGES/*.*')),
-                 ('datas/locale/nl/LC_MESSAGES',glob.glob('datas/locale/nl/LC_MESSAGES/*.*')),
-                 ('datas/locale/pl/LC_MESSAGES',glob.glob('datas/locale/pl/LC_MESSAGES/*.*')),
-                 ('datas/locale/ru/LC_MESSAGES',glob.glob('datas/locale/ru/LC_MESSAGES/*.*')),
-                 ('datas/locale/ro/LC_MESSAGES',glob.glob('datas/locale/ro/LC_MESSAGES/*.*')),
-                 ('datas/locale/en/LC_MESSAGES',glob.glob('datas/locale/en/LC_MESSAGES/*.*'))
+                 ('datas/locale/tr/LC_MESSAGES',glob.glob('datas/locale/tr/LC_MESSAGES/*.mo')),
+                 ('datas/locale/az/LC_MESSAGES',glob.glob('datas/locale/az/LC_MESSAGES/*.mo')),
+                 ('datas/locale/de/LC_MESSAGES',glob.glob('datas/locale/de/LC_MESSAGES/*.mo')),
+                 ('datas/locale/es/LC_MESSAGES',glob.glob('datas/locale/es/LC_MESSAGES/*.mo')),
+                 ('datas/locale/fr/LC_MESSAGES',glob.glob('datas/locale/fr/LC_MESSAGES/*.mo')),
+                 ('datas/locale/ind/LC_MESSAGES',glob.glob('datas/locale/ind/LC_MESSAGES/*.mo')),
+                 ('datas/locale/ir/LC_MESSAGES',glob.glob('datas/locale/ir/LC_MESSAGES/*.mo')),
+                 ('datas/locale/nl/LC_MESSAGES',glob.glob('datas/locale/nl/LC_MESSAGES/*.mo')),
+                 ('datas/locale/pl/LC_MESSAGES',glob.glob('datas/locale/pl/LC_MESSAGES/*.mo')),
+                 ('datas/locale/ru/LC_MESSAGES',glob.glob('datas/locale/ru/LC_MESSAGES/*.mo')),
+                 ('datas/locale/ro/LC_MESSAGES',glob.glob('datas/locale/ro/LC_MESSAGES/*.mo')),
+                 ('datas/locale/en/LC_MESSAGES',glob.glob('datas/locale/en/LC_MESSAGES/*.mo'))
                  ]
-)
\ Dosya sonunda yenisatır yok.
+)
