diff -Nur libpst-0.6.45.orig/html/Makefile.am libpst-0.6.45/html/Makefile.am
--- libpst-0.6.45.orig/html/Makefile.am	2010-02-13 19:14:49.628780864 +0200
+++ libpst-0.6.45/html/Makefile.am	2010-02-13 19:15:25.282524485 +0200
@@ -1,5 +1,5 @@
-htmldir        = ${datadir}/doc/@PACKAGE@-@VERSION@
-htmldeveldir   = ${datadir}/doc/@PACKAGE@-@VERSION@/devel
+htmldir        = ${datadir}/doc/@PACKAGE@
+htmldeveldir   = ${datadir}/doc/@PACKAGE@/devel
 html_DATA      = $(wildcard *.html)
 htmldevel_DATA = $(wildcard devel/*)
 EXTRA_DIST     = $(html_DATA) $(htmldevel_DATA)
diff -Nur libpst-0.6.45.orig/html/Makefile.in libpst-0.6.45/html/Makefile.in
--- libpst-0.6.45.orig/html/Makefile.in	2010-02-13 19:14:49.628780864 +0200
+++ libpst-0.6.45/html/Makefile.in	2010-02-13 19:15:40.953564780 +0200
@@ -188,8 +188,8 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-htmldir = ${datadir}/doc/@PACKAGE@-@VERSION@
-htmldeveldir = ${datadir}/doc/@PACKAGE@-@VERSION@/devel
+htmldir = ${datadir}/doc/@PACKAGE@
+htmldeveldir = ${datadir}/doc/@PACKAGE@/devel
 html_DATA = $(wildcard *.html)
 htmldevel_DATA = $(wildcard devel/*)
 EXTRA_DIST = $(html_DATA) $(htmldevel_DATA)
diff -Nur libpst-0.6.45.orig/Makefile.am libpst-0.6.45/Makefile.am
--- libpst-0.6.45.orig/Makefile.am	2010-02-13 19:14:49.613785587 +0200
+++ libpst-0.6.45/Makefile.am	2010-02-13 19:15:07.681542019 +0200
@@ -1,5 +1,5 @@
 SUBDIRS = src python man html debian
-htmldir = ${datadir}/doc/@PACKAGE@-@VERSION@
+htmldir = ${datadir}/doc/@PACKAGE@
 html_DATA = AUTHORS COPYING ChangeLog NEWS README
 CLEANFILES = xml/libpst xml/Makefile
 
diff -Nur libpst-0.6.45.orig/Makefile.in libpst-0.6.45/Makefile.in
--- libpst-0.6.45.orig/Makefile.in	2010-02-13 19:14:49.612793491 +0200
+++ libpst-0.6.45/Makefile.in	2010-02-13 19:14:58.980781619 +0200
@@ -215,7 +215,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 SUBDIRS = src python man html debian
-htmldir = ${datadir}/doc/@PACKAGE@-@VERSION@
+htmldir = ${datadir}/doc/@PACKAGE@
 html_DATA = AUTHORS COPYING ChangeLog NEWS README
 CLEANFILES = xml/libpst xml/Makefile
 EXTRA_DIST = config.rpath Doxyfile libpst.html.tar.gz libpst.spec $(wildcard xml/M*) $(wildcard xml/h*) $(wildcard xml/lib*)
