diff -ur gsoap-2.8.orig/gsoapck.pc.in gsoap-2.8/gsoapck.pc.in
--- gsoap-2.8.orig/gsoapck.pc.in	2013-08-26 01:10:08.000000000 +0200
+++ gsoap-2.8/gsoapck.pc.in	2013-10-31 09:35:27.786580230 +0100
@@ -7,5 +7,6 @@
 Description: SOAP C Web Services with Cookies Enabled
 Version: @VERSION@
 Requires:
-Libs: -L${libdir} -lgsoapck @SAMPLE_EXTRA_LIBS@
+Libs: -L${libdir} -lgsoapck
+Libs.private: @SAMPLE_EXTRA_LIBS@
 Cflags: @SOAPCPP2_IPV6@ -DWITH_COOKIES -DWITH_DOM -I${includedir}
diff -ur gsoap-2.8.orig/gsoapck++.pc.in gsoap-2.8/gsoapck++.pc.in
--- gsoap-2.8.orig/gsoapck++.pc.in	2013-08-26 01:10:08.000000000 +0200
+++ gsoap-2.8/gsoapck++.pc.in	2013-10-31 09:35:12.459766167 +0100
@@ -7,5 +7,6 @@
 Description: SOAP C++ Web Services with Cookies Enabled
 Version: @VERSION@
 Requires:
-Libs: -L${libdir} -lgsoapck++ @SAMPLE_EXTRA_LIBS@
+Libs: -L${libdir} -lgsoapck++
+Libs.private: @SAMPLE_EXTRA_LIBS@
 Cflags: @SOAPCPP2_IPV6@ -DWITH_COOKIES -DWITH_DOM -I${includedir}
diff -ur gsoap-2.8.orig/gsoap.pc.in gsoap-2.8/gsoap.pc.in
--- gsoap-2.8.orig/gsoap.pc.in	2013-08-26 01:10:08.000000000 +0200
+++ gsoap-2.8/gsoap.pc.in	2013-10-31 09:34:52.043013855 +0100
@@ -7,5 +7,6 @@
 Description: SOAP C Web Services
 Version: @VERSION@
 Requires:
-Libs: -L${libdir} -lgsoap @SAMPLE_EXTRA_LIBS@
+Libs: -L${libdir} -lgsoap
+Libs.private: @SAMPLE_EXTRA_LIBS@
 Cflags: @SOAPCPP2_IPV6@ -DWITH_DOM -I${includedir}
diff -ur gsoap-2.8.orig/gsoap++.pc.in gsoap-2.8/gsoap++.pc.in
--- gsoap-2.8.orig/gsoap++.pc.in	2013-08-26 01:10:08.000000000 +0200
+++ gsoap-2.8/gsoap++.pc.in	2013-10-31 09:34:20.155400703 +0100
@@ -7,5 +7,6 @@
 Description: SOAP C++ Web Services
 Version: @VERSION@
 Requires:
-Libs: -L${libdir} -lgsoap++ @SAMPLE_EXTRA_LIBS@
+Libs: -L${libdir} -lgsoap++
+Libs.private: @SAMPLE_EXTRA_LIBS@
 Cflags: @SOAPCPP2_IPV6@ -DWITH_DOM -I${includedir}
