--- modules/codec/dmo/dmo.c~	2012-09-14 00:02:04.000000000 +0800
+++ modules/codec/dmo/dmo.c	2012-10-16 07:10:41.145463290 +0800
@@ -546,7 +546,7 @@
         }
 
         p_bih = &p_vih->bmiHeader;
-        p_bih->biCompression = i_chroma == VLC_CODEC_RGB24 ? BI_RGB : i_chroma;
+        p_bih->biCompression = i_chroma == VLC_CODEC_RGB24 ? 0L : i_chroma;
         p_bih->biHeight *= -1;
         p_bih->biBitCount = p_dec->fmt_out.video.i_bits_per_pixel;
         p_bih->biSizeImage = p_dec->fmt_in.video.i_width *
