diff --git a/config.c b/config.c
index 76c0df6..399fdf4 100644
--- a/config.c
+++ b/config.c
@@ -956,18 +956,17 @@ static int readConfigFile(const char *configFile, struct logInfo *defConfig)
 					free(key);
 					if ((key = isolateValue(configFile, lineNum, "tabooext", &start,
 							&buf, length)) != NULL) {
-
-						if (*key == '+') {
-							key++;
-							while (isspace(*key) && *key)
-								key++;
+						endtag = key;
+						if (*endtag == '+') {
+							endtag++;
+							while (isspace(*endtag) && *endtag)
+								endtag++;
 						} else {
 							free_2d_array(tabooExts, tabooCount);
 							tabooCount = 0;
 							tabooExts = malloc(1);
 						}
 
-						endtag = key;
 						while (*endtag) {
 							chptr = endtag;
 							while (!isspace(*chptr) && *chptr != ',' && *chptr)
