--- suspend-0.8.20080612/bootsplash.c.printf_format	2009-02-04 09:45:15.000000000 +0100
+++ suspend-0.8.20080612/bootsplash.c	2009-02-04 10:03:01.000000000 +0100
@@ -118,7 +118,7 @@ int bootsplash_dialog(const char *prompt
 {
 	int ret;
 	bootsplash_to_verbose();
-	printf(prompt);
+	printf("%s", prompt);
 	ret = getchar();
 	bootsplash_to_silent();
 
--- suspend-0.8.20080612/splash.c.printf_format	2009-02-04 09:45:15.000000000 +0100
+++ suspend-0.8.20080612/splash.c	2009-02-04 09:56:30.000000000 +0100
@@ -53,7 +53,7 @@ static void splash_dummy_readpass(char *
 
 static int splash_dialog(const char *prompt) 
 {
-	printf(prompt);
+	printf("%s", prompt);
 	return getchar();
 }
 
