--- xvattr.pod.in.org	2002-03-03 22:08:28.000000000 +0200
+++ xvattr.pod.in	2014-03-14 11:05:05.093407336 +0200
@@ -103,7 +103,7 @@
 
 =head1 AUTHOR
 
-Björn Englund. Send bug reports to C<d4bjorn@dtek.chalmers.se>
+Bjorn Englund. Send bug reports to C<d4bjorn@dtek.chalmers.se>
 C<rocky@panix.com> turned the README into this document.
 
 
@@ -114,7 +114,7 @@
 
 =head1 COPYRIGHT
 
-  Copyright (C) 2001-2002 Björn Englund
+  Copyright (C) 2001-2002 Bjorn Englund
 
   This program is free software; you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
