--- less-436/less.nro.old-bot	2010-01-05 15:20:00.690985900 +0100
+++ less-436/less.nro	2010-01-05 15:16:30.500112556 +0100
@@ -555,6 +555,10 @@ This can cause 
 to run somewhat faster than the default.
 .IP "\-G or \-\-HILITE-SEARCH"
 The \-G option suppresses all highlighting of strings found by search commands.
+.IP "\-\-old-bot"
+Reverts to the old bottom of screen behavior.  This can be sometimes  
+desirable  if  the  long lines are not wrapped  correctly
+when  reaching  the  bottom  of  the  terminal,  while scrolling forward.
 .IP "\-h\fIn\fP or \-\-max-back-scroll=\fIn\fP"
 Specifies a maximum number of lines to scroll backward.
 If it is necessary to scroll backward more than \fIn\fP lines,
