diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/msfcli msf3-fhs/msfcli
--- msf3/msfcli	2011-05-03 16:24:09.000000000 +0200
+++ msf3-fhs/msfcli	2011-05-10 11:59:59.000000000 +0200
@@ -9,12 +9,7 @@
 # $Revision: 10719 $
 #
 
-msfbase = __FILE__
-while File.symlink?(msfbase)
-	msfbase = File.expand_path(File.readlink(msfbase), File.dirname(msfbase))
-end
-
-$:.unshift(File.join(File.dirname(msfbase), 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 $:.unshift(ENV['MSF_LOCAL_LIB']) if ENV['MSF_LOCAL_LIB']
 
 require 'rex'
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/msfconsole msf3-fhs/msfconsole
--- msf3/msfconsole	2011-05-03 16:24:09.000000000 +0200
+++ msf3-fhs/msfconsole	2011-05-10 11:59:59.000000000 +0200
@@ -8,12 +8,7 @@
 # $Revision: 11748 $
 #
 
-msfbase = __FILE__
-while File.symlink?(msfbase)
-	msfbase = File.expand_path(File.readlink(msfbase), File.dirname(msfbase))
-end
-
-$:.unshift(File.join(File.expand_path(File.dirname(msfbase)), 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 $:.unshift(ENV['MSF_LOCAL_LIB']) if ENV['MSF_LOCAL_LIB']
 
 require 'optparse'
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/msfd msf3-fhs/msfd
--- msf3/msfd	2011-05-03 16:24:09.000000000 +0200
+++ msf3-fhs/msfd	2011-05-10 11:59:59.000000000 +0200
@@ -10,12 +10,7 @@
 # $Revision: 12499 $
 #
 
-msfbase = __FILE__
-while File.symlink?(msfbase)
-	msfbase = File.expand_path(File.readlink(msfbase), File.dirname(msfbase))
-end
-
-$:.unshift(File.join(File.dirname(msfbase), 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 $:.unshift(ENV['MSF_LOCAL_LIB']) if ENV['MSF_LOCAL_LIB']
 
 require 'msf/base'
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/msfelfscan msf3-fhs/msfelfscan
--- msf3/msfelfscan	2011-05-03 16:24:09.000000000 +0200
+++ msf3-fhs/msfelfscan	2011-05-10 11:59:59.000000000 +0200
@@ -4,12 +4,7 @@
 # $Revision: 9212 $
 #
 
-msfbase = __FILE__
-while File.symlink?(msfbase)
-	msfbase = File.expand_path(File.readlink(msfbase), File.dirname(msfbase))
-end
-
-$:.unshift(File.join(File.dirname(msfbase), 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 $:.unshift(ENV['MSF_LOCAL_LIB']) if ENV['MSF_LOCAL_LIB']
 
 require 'rex/elfparsey'
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/msfencode msf3-fhs/msfencode
--- msf3/msfencode	2011-05-03 16:24:09.000000000 +0200
+++ msf3-fhs/msfencode	2011-05-10 11:59:59.000000000 +0200
@@ -4,12 +4,7 @@
 # $Revision: 11752 $
 #
 
-msfbase = __FILE__
-while File.symlink?(msfbase)
-	msfbase = File.expand_path(File.readlink(msfbase), File.dirname(msfbase))
-end
-
-$:.unshift(File.join(File.dirname(msfbase), 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 $:.unshift(ENV['MSF_LOCAL_LIB']) if ENV['MSF_LOCAL_LIB']
 
 require 'rex'
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/msfgui msf3-fhs/msfgui
--- msf3/msfgui	2011-05-03 16:24:09.000000000 +0200
+++ msf3-fhs/msfgui	2011-05-10 12:32:09.000000000 +0200
@@ -7,13 +7,10 @@
 # $Revision: 10806 $
 #
 
-msfbase = __FILE__
-while File.symlink?(msfbase)
-	msfbase = File.expand_path(File.readlink(msfbase), File.dirname(msfbase))
-end
+msfbase = '/usr/share/metasploit'
 
 if RUBY_PLATFORM =~ /mswin|mingw/i
-	exec "javaw -jar #{File.dirname(msfbase)}/data/gui/msfgui.jar"
+	exec "javaw -jar #{msfbase}/data/gui/msfgui.jar"
 else
-	exec "java -jar #{File.dirname(msfbase)}/data/gui/msfgui.jar"
+	exec "java -jar #{msfbase}/data/gui/msfgui.jar"
 end
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/msfmachscan msf3-fhs/msfmachscan
--- msf3/msfmachscan	2011-05-03 16:24:09.000000000 +0200
+++ msf3-fhs/msfmachscan	2011-05-10 11:59:59.000000000 +0200
@@ -4,12 +4,7 @@
 # $Revision: 9212 $
 #
 
-msfbase = __FILE__
-while File.symlink?(msfbase)
-	msfbase = File.expand_path(File.readlink(msfbase), File.dirname(msfbase))
-end
-
-$:.unshift(File.join(File.dirname(msfbase), 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 $:.unshift(ENV['MSF_LOCAL_LIB']) if ENV['MSF_LOCAL_LIB']
 
 require 'rex/machparsey'
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/msfopcode msf3-fhs/msfopcode
--- msf3/msfopcode	2011-05-03 16:24:09.000000000 +0200
+++ msf3-fhs/msfopcode	2011-05-10 11:59:59.000000000 +0200
@@ -9,12 +9,7 @@
 # $Revision: 9212 $
 #
 
-msfbase = __FILE__
-while File.symlink?(msfbase)
-	msfbase = File.expand_path(File.readlink(msfbase), File.dirname(msfbase))
-end
-
-$:.unshift(File.join(File.dirname(msfbase), 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 $:.unshift(ENV['MSF_LOCAL_LIB']) if ENV['MSF_LOCAL_LIB']
 
 require 'rex'
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/msfpayload msf3-fhs/msfpayload
--- msf3/msfpayload	2011-05-03 17:16:30.000000000 +0200
+++ msf3-fhs/msfpayload	2011-05-10 11:59:59.000000000 +0200
@@ -4,12 +4,7 @@
 # $Revision: 12533 $
 #
 
-msfbase = __FILE__
-while File.symlink?(msfbase)
-	msfbase = File.expand_path(File.readlink(msfbase), File.dirname(msfbase))
-end
-
-$:.unshift(File.join(File.dirname(msfbase), 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 $:.unshift(ENV['MSF_LOCAL_LIB']) if ENV['MSF_LOCAL_LIB']
 
 require 'rex'
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/msfpescan msf3-fhs/msfpescan
--- msf3/msfpescan	2011-05-03 16:24:09.000000000 +0200
+++ msf3-fhs/msfpescan	2011-05-10 11:59:59.000000000 +0200
@@ -4,12 +4,7 @@
 # $Revision: 9212 $
 #
 
-msfbase = __FILE__
-while File.symlink?(msfbase)
-	msfbase = File.expand_path(File.readlink(msfbase), File.dirname(msfbase))
-end
-
-$:.unshift(File.join(File.dirname(msfbase), 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 $:.unshift(ENV['MSF_LOCAL_LIB']) if ENV['MSF_LOCAL_LIB']
 
 require 'rex/peparsey'
@@ -80,7 +75,7 @@
 
 opt.on('-f', '--fingerprint', 'Attempt to identify the packer/compiler') do |t|
 	worker = Rex::PeScan::Analyze::Fingerprint
-	param['database'] = File.join(File.dirname(msfbase), 'data', 'msfpescan', 'identify.txt')
+	param['database'] = '/usr/share/metasploit/data/msfpescan/identify.txt'
 end
 
 opt.on('-i', '--info', 'Display detailed information about the image') do |t|
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/msfrpc msf3-fhs/msfrpc
--- msf3/msfrpc	2011-05-03 16:24:09.000000000 +0200
+++ msf3-fhs/msfrpc	2011-05-10 11:59:59.000000000 +0200
@@ -8,12 +8,7 @@
 # $Revision: 10774 $
 #
 
-msfbase = __FILE__
-while File.symlink?(msfbase)
-	msfbase = File.expand_path(File.readlink(msfbase), File.dirname(msfbase))
-end
-
-$:.unshift(File.join(File.dirname(msfbase), 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 $:.unshift(ENV['MSF_LOCAL_LIB']) if ENV['MSF_LOCAL_LIB']
 
 require 'rex/parser/arguments'
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/msfrpcd msf3-fhs/msfrpcd
--- msf3/msfrpcd	2011-05-03 16:24:09.000000000 +0200
+++ msf3-fhs/msfrpcd	2011-05-10 11:59:59.000000000 +0200
@@ -8,12 +8,7 @@
 # $Revision: 12498 $
 #
 
-msfbase = __FILE__
-while File.symlink?(msfbase)
-	msfbase = File.expand_path(File.readlink(msfbase), File.dirname(msfbase))
-end
-
-$:.unshift(File.join(File.dirname(msfbase), 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 $:.unshift(ENV['MSF_LOCAL_LIB']) if ENV['MSF_LOCAL_LIB']
 
 require 'rex/parser/arguments'
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/msfupdate msf3-fhs/msfupdate
--- msf3/msfupdate	2011-05-04 10:11:15.000000000 +0200
+++ msf3-fhs/msfupdate	2011-05-10 11:59:59.000000000 +0200
@@ -1,10 +1,6 @@
 #!/usr/bin/env ruby
 
-msfbase = __FILE__
-while File.symlink?(msfbase)
-	msfbase = File.expand_path(File.readlink(msfbase), File.dirname(msfbase))
-end
-
+msfbase = '/usr/share/metasploit'
 
 Dir.chdir(File.dirname(msfbase))
 
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/tools/exe2vba.rb msf3-fhs/tools/exe2vba.rb
--- msf3/tools/exe2vba.rb	2011-05-03 16:24:00.000000000 +0200
+++ msf3-fhs/tools/exe2vba.rb	2011-05-10 11:59:59.000000000 +0200
@@ -8,8 +8,7 @@
 # $Revision: 9212 $
 #
 
-msfbase = File.symlink?(__FILE__) ? File.readlink(__FILE__) : __FILE__
-$:.unshift(File.join(File.dirname(msfbase), '..', 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 
 require 'rex'
 require 'msf/base'
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/tools/exe2vbs.rb msf3-fhs/tools/exe2vbs.rb
--- msf3/tools/exe2vbs.rb	2011-05-03 16:24:00.000000000 +0200
+++ msf3-fhs/tools/exe2vbs.rb	2011-05-10 11:59:59.000000000 +0200
@@ -7,8 +7,7 @@
 # $Revision: 9212 $
 #
 
-msfbase = File.symlink?(__FILE__) ? File.readlink(__FILE__) : __FILE__
-$:.unshift(File.join(File.dirname(msfbase), '..', 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 
 require 'rex'
 require 'msf/base'
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/tools/halflm_second.rb msf3-fhs/tools/halflm_second.rb
--- msf3/tools/halflm_second.rb	2011-05-03 16:24:00.000000000 +0200
+++ msf3-fhs/tools/halflm_second.rb	2011-05-10 11:59:59.000000000 +0200
@@ -10,8 +10,7 @@
 # $Revision: 11893 $
 #
 
-msfbase = File.symlink?(__FILE__) ? File.readlink(__FILE__) : __FILE__
-$:.unshift(File.join(File.dirname(msfbase), '..', 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 
 require 'rex'
 
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/tools/lm2ntcrack.rb msf3-fhs/tools/lm2ntcrack.rb
--- msf3/tools/lm2ntcrack.rb	2011-05-03 16:24:00.000000000 +0200
+++ msf3-fhs/tools/lm2ntcrack.rb	2011-05-10 11:59:59.000000000 +0200
@@ -8,8 +8,7 @@
 # $Revision: 11893 $
 #
 
-msfbase = File.symlink?(__FILE__) ? File.readlink(__FILE__) : __FILE__
-$:.unshift(File.join(File.dirname(msfbase), '..', 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 
 require 'rex'
 require 'rex/proto/ntlm/crypt'
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/tools/module_author.rb msf3-fhs/tools/module_author.rb
--- msf3/tools/module_author.rb	2011-05-03 16:24:00.000000000 +0200
+++ msf3-fhs/tools/module_author.rb	2011-05-10 11:59:59.000000000 +0200
@@ -7,8 +7,7 @@
 # the number of modules per author
 #
 
-msfbase = File.symlink?(__FILE__) ? File.readlink(__FILE__) : __FILE__
-$:.unshift(File.join(File.dirname(msfbase), '..', 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 
 require 'rex'
 require 'msf/ui'
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/tools/module_license.rb msf3-fhs/tools/module_license.rb
--- msf3/tools/module_license.rb	2011-05-03 16:24:00.000000000 +0200
+++ msf3-fhs/tools/module_license.rb	2011-05-10 11:59:59.000000000 +0200
@@ -7,8 +7,7 @@
 # $Revision: 10652 $
 #
 
-msfbase = File.symlink?(__FILE__) ? File.readlink(__FILE__) : __FILE__
-$:.unshift(File.join(File.dirname(msfbase), '..', 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 
 require 'rex'
 require 'msf/ui'
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/tools/module_reference.rb msf3-fhs/tools/module_reference.rb
--- msf3/tools/module_reference.rb	2011-05-03 16:24:00.000000000 +0200
+++ msf3-fhs/tools/module_reference.rb	2011-05-10 11:59:59.000000000 +0200
@@ -7,8 +7,7 @@
 # $Revision: 10911 $
 #
 
-msfbase = File.symlink?(__FILE__) ? File.readlink(__FILE__) : __FILE__
-$:.unshift(File.join(File.dirname(msfbase), '..', 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 
 require 'rex'
 require 'msf/ui'
diff -Naur -x '*~' -x '*.rej' -x '*.orig' --ignore-space-change msf3/tools/msf_irb_shell.rb msf3-fhs/tools/msf_irb_shell.rb
--- msf3/tools/msf_irb_shell.rb	2011-05-03 16:24:00.000000000 +0200
+++ msf3-fhs/tools/msf_irb_shell.rb	2011-05-10 11:59:59.000000000 +0200
@@ -4,8 +4,7 @@
 # $Revision: 9212 $
 #
 
-msfbase = File.symlink?(__FILE__) ? File.readlink(__FILE__) : __FILE__
-$:.unshift(File.join(File.dirname(msfbase), '..', 'lib'))
+$:.unshift('/usr/share/metasploit/lib')
 
 require 'rex'
 require 'msf/core'
