Index: netcat-openbsd-1.89/socks.c
===================================================================
--- netcat-openbsd-1.89.orig/socks.c	2008-06-19 16:30:45.000000000 -0400
+++ netcat-openbsd-1.89/socks.c	2008-06-19 16:30:36.000000000 -0400
@@ -53,6 +53,8 @@
 #define SOCKS_DOMAIN	3
 #define SOCKS_IPV6	4
 
+extern int b64_ntop(unsigned char const *, size_t, char *, size_t);
+
 int	remote_connect(const char *, const char *, struct addrinfo);
 int	socks_connect(const char *, const char *, struct addrinfo,
 	    const char *, const char *, struct addrinfo, int,
