diff -urN a/configure.ac b/configure.ac
--- a/configure.ac	2012-03-11 20:23:37.000000000 +0000
+++ b/configure.ac	2012-06-16 21:11:34.189922666 +0100
@@ -74,6 +74,8 @@
 
 dnl pkg-config dependency checks
 
+PKG_CHECK_MODULES(GMODULE, [gmodule-2.0])
+
 PKG_CHECK_MODULES(EGG_SMCLIENT, ice sm gtk+-2.0)
 
 PKG_CHECK_MODULES(PANEL, gdk-pixbuf-2.0 >= $GDK_PIXBUF_REQUIRED pango >= $PANGO_REQUIRED gtk+-2.0 >= $GTK_REQUIRED glib-2.0 >= $GLIB_REQUIRED gio-2.0 >= $GLIB_REQUIRED gio-unix-2.0 >= $GLIB_REQUIRED mate-desktop-2.0 >= $LIBMATE_DESKTOP_REQUIRED mateconf-2.0 >= $MATECONF_REQUIRED libmate-menu >= $LIBMATE_MENU_REQUIRED libcanberra-gtk dbus-glib-1 >= $DBUS_GLIB_REQUIRED)
diff -urN a/mate-panel/Makefile.am b/mate-panel/Makefile.am
--- a/mate-panel/Makefile.am	2012-03-11 20:23:37.000000000 +0000
+++ b/mate-panel/Makefile.am	2012-06-16 21:18:57.305920858 +0100
@@ -137,6 +137,7 @@
 	$(PANEL_LIBS) \
 	$(XRANDR_LIBS) \
 	$(X_LIBS) \
+        $(GMODULE_LIBS) \
 	-lm
 
 mate_panel_LDFLAGS = -export-dynamic
@@ -151,6 +152,7 @@
 mate_desktop_item_edit_LDADD = \
 	$(top_builddir)/mate-panel/libpanel-util/libpanel-util.la \
 	$(PANEL_LIBS) \
+        $(GMODULE_LIBS) \
 	-lX11
 
 mate_panel_test_applets_SOURCES = \
@@ -168,6 +170,7 @@
 mate_panel_test_applets_LDADD = \
 	$(top_builddir)/mate-panel/libmate-panel-applet-private/libmate-panel-applet-private-mini.la \
 	$(top_builddir)/mate-panel/libpanel-util/libpanel-util.la \
+        $(GMODULE_LIBS) \
 	$(PANEL_LIBS)
 
 mate_panel_test_applets_LDFLAGS = -export-dynamic
 
