--- src/internet/podcasts/gpoddersync.h	2015-02-07 15:40:34.000000000 +0200
+++ src/internet/podcasts/gpoddersync_new.h	2015-02-08 18:25:50.195871410 +0200
@@ -29,7 +29,7 @@
 #include <QSharedPointer>
 #include <QUrl>
 
-#include <ApiRequest.h>
+#include <mygpo-qt/ApiRequest.h>
 
 class Application;
 class Podcast;

--- src/internet/podcasts/gpoddersearchpage.h	2015-02-07 15:40:34.000000000 +0200
+++ src/internet/podcasts/gpoddersearchpage_new.h	2015-02-08 18:46:28.338816208 +0200
@@ -22,7 +22,7 @@
 
 #include "addpodcastpage.h"
 
-#include <ApiRequest.h>
+#include <mygpo-qt/ApiRequest.h>
 
 class QNetworkAccessManager;

--- src/internet/podcasts/gpoddertoptagsmodel.cpp	2015-02-07 15:40:34.000000000 +0200
+++ src/internet/podcasts/gpoddertoptagsmodel_new.cpp	2015-02-08 18:50:57.038804228 +0200
@@ -19,7 +19,7 @@
 
 #include "gpoddertoptagsmodel.h"
 
-#include <ApiRequest.h>
+#include <mygpo-qt/ApiRequest.h>
 #include <QMessageBox>
 
 #include "gpoddertoptagspage.h"

--- src/internet/podcasts/gpoddertoptagspage.h	2015-02-07 15:40:34.000000000 +0200
+++ src/internet/podcasts/gpoddertoptagspage_new.h	2015-02-08 19:10:58.561750659 +0200
@@ -22,7 +22,7 @@
 
 #include <QScopedPointer>
 
-#include <ApiRequest.h>
+#include <mygpo-qt/ApiRequest.h>
 
 #include "addpodcastpage.h"
 
--- src/internet/podcasts/podcast.cpp	2015-02-07 15:40:34.000000000 +0200
+++ src/internet/podcasts/podcast_new.cpp	2015-02-08 19:56:26.687629028 +0200
@@ -21,7 +21,7 @@
 
 #include <QDataStream>
 #include <QDateTime>
-#include <Podcast.h>
+#include <mygpo-qt/Podcast.h>
 
 #include "core/utilities.h"
