Index: scidavis-0.2.4/scidavis/src/ApplicationWindow.cpp
===================================================================
--- scidavis-0.2.4.orig/scidavis/src/ApplicationWindow.cpp
+++ scidavis-0.2.4/scidavis/src/ApplicationWindow.cpp
@@ -135,7 +135,7 @@
 #include <QVarLengthArray>
 #include <QList>
 #include <QUrl>
-#include <QAssistantClient>
+#include <QAssistantClient/QAssistantClient>
 #include <QStatusBar>
 #include <QToolButton>
 #include <QSignalMapper>
