diff -urp manual.org/docbook-en/index.docbook manual/docbook-en/index.docbook
--- manual.org/docbook-en/index.docbook	2010-05-02 15:21:59.102806908 +0800
+++ manual/docbook-en/index.docbook	2010-05-02 15:22:53.221057241 +0800
@@ -8,7 +8,7 @@
 
 *************************************************************************
 -->
-<!DOCTYPE book PUBLIC "-//OASIS//DTD DocBook XML V4.5//EN" "/usr/share/xml/docbook/xml-dtd-4.5/docbookx.dtd" [
+<!DOCTYPE book PUBLIC "-//OASIS//DTD DocBook XML V4.5//EN" "/usr/share/sgml/docbook/xml-dtd-4.5-VER-REL/docbookx.dtd" [
 <!--
 		general terms for the application
 -->
diff -urp manual.org/Makefile manual/Makefile
--- manual.org/Makefile	2010-05-02 15:21:59.104807302 +0800
+++ manual/Makefile	2010-05-02 15:23:22.012056778 +0800
@@ -12,3 +12,5 @@ html-packages: html-en
 
 pdf-en:
 	dblatex -o ${BASE}.pdf -c dblatex.conf -p dblatex-param.xsl docbook-en/index.docbook
+
+install:
diff -urp manual.org/scidavis.dsl manual/scidavis.dsl
--- manual.org/scidavis.dsl	2010-05-02 15:21:59.104807302 +0800
+++ manual/scidavis.dsl	2010-05-02 15:24:31.091806836 +0800
@@ -6,7 +6,7 @@
 <!DOCTYPE style-sheet PUBLIC "-//James Clark//DTD DSSSL Style Sheet//EN" [
 
 <!-- Path to main Docbook->html stylesheet. You have to adapt this to your installation. -->
-<!ENTITY docbook-html-stylesheet SYSTEM "/usr/share/sgml/docbook/stylesheet/dsssl/modular/html/docbook.dsl" CDATA DSSSL>
+<!ENTITY docbook-html-stylesheet SYSTEM "/usr/share/sgml/docbook/dsssl-stylesheets-1.79/html/docbook.dsl" CDATA DSSSL>
 <!--ENTITY docbook-html-stylesheet SYSTEM "/usr/share/sgml/docbook/dsssl-stylesheets-1.79/html/docbook.dsl" CDATA DSSSL-->
 
 <!-- Stylesheet function library for creating Windows HTML help and Qt Assistant profiles. -->
diff -urp manual.org/scidavis_html.xsl manual/scidavis_html.xsl
--- manual.org/scidavis_html.xsl	2010-05-02 15:21:59.103806896 +0800
+++ manual/scidavis_html.xsl	2010-05-02 15:24:03.020808680 +0800
@@ -1,5 +1,5 @@
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version='1.0'>
-<xsl:import href="/usr/share/xml/docbook/xsl-stylesheets-1.75.2/html/chunk.xsl"/>
+<xsl:import href="file:////usr/share/sgml/docbook/xsl-stylesheets-1.75.2/html/chunk.xsl"/>
 <xsl:import href="docbook2adp.xsl"/>
 
 <xsl:param name="chunker.output.encoding">UTF-8</xsl:param>
