diff -Nur pnm2ppa-1.12-old/pnm2ppa.conf pnm2ppa-1.12/pnm2ppa.conf
--- pnm2ppa-1.12-old/pnm2ppa.conf	2005-12-31 01:54:54.000000000 +0200
+++ pnm2ppa-1.12/pnm2ppa.conf	2005-12-31 01:55:36.000000000 +0200
@@ -15,8 +15,8 @@
 # will be used.   The printer version can also be set with the command line
 # option e.g., "-v 720".
 
-version 0       # REMOVE THIS ENTRY to make 720 the default printer model!!!
-#version  720	# 710, 712, 722 also acceptable
+# version 0       # REMOVE THIS ENTRY to make 720 the default printer model!!!
+version  720	# 710, 712, 722 also acceptable
 #version  820
 #version 1000
 
