diff -Nur xine-lib-1.1.19-old//m4/pthreads.m4 xine-lib-1.1.19/m4/pthreads.m4
--- xine-lib-1.1.19-old//m4/pthreads.m4	2010-09-04 18:26:52.719999251 +0300
+++ xine-lib-1.1.19/m4/pthreads.m4	2010-09-04 18:28:15.110999847 +0300
@@ -15,7 +15,7 @@
   AC_ARG_VAR([PTHREAD_CFLAGS], [C compiler flags for Pthread support])
   AC_ARG_VAR([PTHREAD_LIBS], [linker flags for Pthread support])
 
-  dnl if PTHREAD_* are not set, default to -pthread (GCC)
+  dnl if PTHREAD_* are not set, default to -lpthread (GCC)
   if test "${PTHREAD_CFLAGS-unset}" = "unset"; then
      case $host in
        *-mingw*)  PTHREAD_CFLAGS=""		;;
@@ -25,7 +25,7 @@
                   dnl Handle Sun Studio compiler (also on Linux)
                   CC_CHECK_CFLAGS([-mt], [PTHREAD_CFLAGS="-mt"]);;
 
-       *)	  PTHREAD_CFLAGS="-pthread"	;;
+       *)	  PTHREAD_CFLAGS="-lpthread"	;;
      esac
   fi
   if test "${PTHREAD_LIBS-unset}" = "unset"; then
@@ -36,7 +36,7 @@
        *-solaris*)
                   dnl Use the same libraries for gcc and Sun Studio cc
                   PTHREAD_LIBS="-lpthread -lposix4 -lrt";;
-       *)	  PTHREAD_LIBS="-pthread"	;;
+       *)	  PTHREAD_LIBS="-lpthread"	;;
      esac
 
      dnl Again, handle Sun Studio compiler
