diff -up xine-lib-1.1.17/autogen.sh.no_autopoint xine-lib-1.1.17/autogen.sh
--- xine-lib-1.1.17/autogen.sh.no_autopoint	2009-10-09 09:33:31.000000000 -0500
+++ xine-lib-1.1.17/autogen.sh	2009-12-02 09:15:48.534555467 -0600
@@ -229,10 +229,10 @@ run_aclocal () {
     echo
   fi
   
-  echo $_echo_n " + Running autopoint: $_echo_c"
+  #echo $_echo_n " + Running autopoint: $_echo_c"
   
-  autopoint
-  echo "done." 
+  #autopoint
+  #echo "done." 
 
   echo $_echo_n " + Running aclocal: $_echo_c"
 
