--- linux-2.6.35/3rdparty/acerhk/acerhk.c.orig	2010-10-03 12:50:33.000000000 +0000
+++ linux-2.6.35/3rdparty/acerhk/acerhk.c	2010-10-03 13:13:22.093868410 +0000
@@ -2730,7 +2730,7 @@ static void acerhk_proc_cleanup(void)
 
 /* {{{ file operations */
 
-static int acerhk_ioctl( struct inode *inode, struct file *file,
+static long acerhk_ioctl( struct inode *inode, struct file *file,
                          unsigned int cmd, unsigned long arg )
 {
   int retval;
@@ -2827,7 +2827,7 @@ static int acerhk_release( struct inode
 
 static struct file_operations acerhk_fops = {
   owner:        THIS_MODULE,
-  ioctl:        acerhk_ioctl,
+  unlocked_ioctl: acerhk_ioctl,
   open:         acerhk_open,
 #ifdef ACERDEBUG
   write:        acerhk_write,
