
 net/ipv4/netfilter/ipt_IFWLOG.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.37-rc3-git1-tmb0.3/net/ipv4/netfilter/ipt_IFWLOG.c.orig	2010-11-24 21:58:36.000000000 +0200
+++ linux-2.6.37-rc3-git1-tmb0.3/net/ipv4/netfilter/ipt_IFWLOG.c	2010-11-25 13:08:55.719379646 +0200
@@ -141,7 +141,7 @@ static unsigned int ipt_IFWLOG_target(st
 
 	ipt_IFWLOG_packet(skb, target_param->in, target_param->out, info);
 
-	return IPT_CONTINUE;
+	return XT_CONTINUE;
 }
 
 static int ipt_IFWLOG_checkentry(const struct xt_tgchk_param *tgchk_param)
