--- linux/drivers/usb/storage/unusual_devs.h.orig	2012-12-06 18:26:48.639813048 +0200
+++ linux/drivers/usb/storage/unusual_devs.h	2012-12-06 18:27:16.570440504 +0200
@@ -2046,16 +2046,16 @@ UNUSUAL_DEV( 0xed10, 0x7636, 0x0001, 0x0
 UNUSUAL_DEV( 0x0584, 0x0001, 0x0100, 0x0102,
 		"RATOCSystems",
 		"Compact Flash Adapter",
-		US_SC_SCSI, US_PR_BULK, NULL,
+		USB_SC_SCSI, USB_PR_BULK, NULL,
 		US_FL_SINGLE_LUN | US_FL_MAX_SECTORS_MIN),
 
 /*
  * Panasonic/OEMs compact USB CDROMs status
  * KXL-840(CD-ROM11): usb_stor_Bulk_max_lun() is danger, need US_FL_SINGLE_LUN
  * KXL-RW11(CDRRW02): usb_stor_Bulk_max_lun() is danger, need US_FL_SINGLE_LUN
- * KXL-RW20(CDRRW03): original IClass is 0xFF, use US_PR_CB and need init reset
+ * KXL-RW20(CDRRW03): original IClass is 0xFF, use USB_PR_CB and need init reset
  * KXL-RW31(CDRRW05): work fine with current code
- * KXL-RW21(CDRRW06): original IClass is 0xFF, use US_PR_CB and need init reset
+ * KXL-RW21(CDRRW06): original IClass is 0xFF, use USB_PR_CB and need init reset
  * KXL-RW32(CDRRW09): work fine with current code
  * KXL-RW40(CDRRW10): work fine with current code
  *      Checked: Go Taniguchi <go@turbolinux.co.jp>
@@ -2063,19 +2063,19 @@ UNUSUAL_DEV( 0x0584, 0x0001, 0x0100, 0x0
 UNUSUAL_DEV( 0x04da, 0x0d01, 0x0000, 0xffff,
 		"Panasonic",
 		"CD-ROM11",
-		US_SC_8020, US_PR_BULK, NULL, US_FL_SINGLE_LUN),
+		USB_SC_8020, USB_PR_BULK, NULL, US_FL_SINGLE_LUN),
 UNUSUAL_DEV( 0x04da, 0x0d02, 0x0000, 0xffff,
 		"Panasonic",
 		"CDRRW02",
-		US_SC_8020, US_PR_BULK, NULL, US_FL_SINGLE_LUN),
+		USB_SC_8020, USB_PR_BULK, NULL, US_FL_SINGLE_LUN),
 UNUSUAL_DEV( 0x04da, 0x0d03, 0x0000, 0xffff,
 		"Panasonic",
 		"CDRRW03",
-		US_SC_8020, US_PR_CB, NULL, US_FL_MAX_SECTORS_MIN),
+		USB_SC_8020, USB_PR_CB, NULL, US_FL_MAX_SECTORS_MIN),
 UNUSUAL_DEV( 0x04da, 0x0d06, 0x0000, 0xffff,
 		"Panasonic",
 		"CDRRW06",
-		US_SC_8020, US_PR_CB, NULL, US_FL_MAX_SECTORS_MIN),
+		USB_SC_8020, USB_PR_CB, NULL, US_FL_MAX_SECTORS_MIN),
 
 /* Control/Bulk transport for all SubClass values */
 USUAL_DEV(USB_SC_RBC, USB_PR_CB),
