diff -p -up ./hdspconf/src/Makefile.am.tv ./hdspconf/src/Makefile.am
--- ./hdspconf/src/Makefile.am.tv	2008-06-10 11:15:09.000000000 +0200
+++ ./hdspconf/src/Makefile.am	2008-06-10 11:15:01.000000000 +0200
@@ -1,4 +1,5 @@
 bin_PROGRAMS = hdspconf
+hdspconf_LDADD = @LDFLAGS@
 hdspconf_SOURCES = \
 	hdspconf.cxx \
 	defines.h \
diff -p -up ./src/Makefile.am.tv ./src/Makefile.am
--- ./hdspmixer/src/Makefile.am.tv	2008-06-10 11:51:47.000000000 +0200
+++ ./hdspmixer/src/Makefile.am	2008-06-10 11:52:06.000000000 +0200
@@ -1,4 +1,5 @@
 bin_PROGRAMS = hdspmixer
+hdspmixer_LDADD = @LDFLAGS@
 hdspmixer_SOURCES = \
 	hdspmixer.cxx \
 	HDSPMixerWindow.cxx \
