diff -Naur alsa-tools-1.0.18/hdsploader/Makefile.in alsa-tools-1.0.18-patched/hdsploader/Makefile.in
--- alsa-tools-1.0.18/hdsploader/Makefile.in	2008-10-29 14:48:40.000000000 +0200
+++ alsa-tools-1.0.18-patched/hdsploader/Makefile.in	2008-10-31 11:36:40.000000000 +0200
@@ -158,7 +158,7 @@
 
 # # Process this file with automake to produce Makefile.in.
 MYNAME = hdsploader
-AM_CFLAGS = -DDATAPATH=\"$(datadir)/alsa/firmware/$(MYNAME)\"
+AM_CFLAGS = -DDATAPATH=\"/lib/firmware/$(MYNAME)\"
 hdsploader_SOURCES = hdsploader.c
 EXTRA_DIST = gitcompile depcomp
 all: all-am
diff -Naur alsa-tools-1.0.18/mixartloader/Makefile.in alsa-tools-1.0.18-patched/mixartloader/Makefile.in
--- alsa-tools-1.0.18/mixartloader/Makefile.in	2008-10-29 14:48:56.000000000 +0200
+++ alsa-tools-1.0.18-patched/mixartloader/Makefile.in	2008-10-31 11:36:58.000000000 +0200
@@ -159,7 +159,7 @@
 # # Process this file with automake to produce Makefile.in.
 AUTOMAKE_OPTIONS = 1.3 foreign
 MYNAME = mixartloader
-AM_CFLAGS = -DDATAPATH=\"$(datadir)/alsa/firmware/$(MYNAME)\"
+AM_CFLAGS = -DDATAPATH=\"/lib/firmware/$(MYNAME)\"
 mixartloader_SOURCES = mixartloader.c
 EXTRA_DIST = gitcompile depcomp
 all: all-am
diff -Naur alsa-tools-1.0.18/pcxhrloader/Makefile.in alsa-tools-1.0.18-patched/pcxhrloader/Makefile.in
--- alsa-tools-1.0.18/pcxhrloader/Makefile.in	2008-10-29 14:49:00.000000000 +0200
+++ alsa-tools-1.0.18-patched/pcxhrloader/Makefile.in	2008-10-31 11:37:27.000000000 +0200
@@ -159,7 +159,7 @@
 # # Process this file with automake to produce Makefile.in.
 AUTOMAKE_OPTIONS = 1.3 foreign
 MYNAME = pcxhrloader
-AM_CFLAGS = -DDATAPATH=\"$(datadir)/alsa/firmware/$(MYNAME)\"
+AM_CFLAGS = -DDATAPATH=\"/lib/firmware/$(MYNAME)\"
 pcxhrloader_SOURCES = pcxhrloader.c
 EXTRA_DIST = gitcompile depcomp
 all: all-am
diff -Naur alsa-tools-1.0.18/usx2yloader/Makefile.in alsa-tools-1.0.18-patched/usx2yloader/Makefile.in
--- alsa-tools-1.0.18/usx2yloader/Makefile.in	2008-10-29 14:49:27.000000000 +0200
+++ alsa-tools-1.0.18-patched/usx2yloader/Makefile.in	2008-10-31 11:37:46.000000000 +0200
@@ -163,7 +163,7 @@
 # # Process this file with automake to produce Makefile.in.
 AUTOMAKE_OPTIONS = 1.3 foreign
 MYNAME = usx2yloader
-AM_CFLAGS = -DDATAPATH=\"$(datadir)/alsa/firmware/$(MYNAME)\"
+AM_CFLAGS = -DDATAPATH=\"/lib/firmware/$(MYNAME)\"
 usx2yloader_SOURCES = usx2yloader.c
 EXTRA_DIST = gitcompile depcomp tascam_fw.usermap tascam_fw.in tascam_fpga.in
 hotplug_files = tascam_fw.usermap tascam_fw tascam_fpga
diff -Naur alsa-tools-1.0.18/vxloader/Makefile.in alsa-tools-1.0.18-patched/vxloader/Makefile.in
--- alsa-tools-1.0.18/vxloader/Makefile.in	2008-10-29 14:49:31.000000000 +0200
+++ alsa-tools-1.0.18-patched/vxloader/Makefile.in	2008-10-31 11:38:09.000000000 +0200
@@ -159,7 +159,7 @@
 # # Process this file with automake to produce Makefile.in.
 AUTOMAKE_OPTIONS = 1.3 foreign
 MYNAME = vxloader
-AM_CFLAGS = -DDATAPATH=\"$(datadir)/alsa/firmware/$(MYNAME)\"
+AM_CFLAGS = -DDATAPATH=\"/lib/firmware/$(MYNAME)\"
 vxloader_SOURCES = vxloader.c
 EXTRA_DIST = gitcompile depcomp
 all: all-am
