--- a/man-pages-posix-2003-a/man1p/printf.1p	2008-06-05 11:22:12.000000000 +0200
+++ b/man-pages-posix-2003-a/man1p/printf.1p	2010-10-22 15:33:14.000000000 +0200
@@ -144,7 +144,8 @@ octal number that shall be converted to 
 numeric value specified by the octal number
 .LP
 .IP " *" 3
-\fB'\\c'\fP, which shall not be written and shall cause \fIprintf\fP
+sequence \fB'\\c'\fP
+which shall not be written and shall cause \fIprintf\fP
 to ignore any remaining characters in the string
 operand containing it, any remaining string operands, and any additional
 characters in the \fIformat\fP operand
