diff -ur man-pages-3.42.orig/man-pages-posix-2003-a/man1p/man.1p man-pages-3.42/man-pages-posix-2003-a/man1p/man.1p
--- man-pages-3.42.orig/man-pages-posix-2003-a/man1p/man.1p	2008-06-05 11:22:12.000000000 +0200
+++ man-pages-3.42/man-pages-posix-2003-a/man1p/man.1p	2012-09-11 10:55:29.899852315 +0200
@@ -44,10 +44,7 @@
 .RS
 .nf
 
-\fBgrep -Ei '
-\fP\fIname
-name\fP\fB...
-'\fP \fIsummary-database\fP
+\fBgrep -Ei '\fP\fIname name\fP\fB...' \fP \fIsummary-database\fP
 .fi
 .RE
 .LP
