--- a/man-pages-posix-2003-a/man1p/awk.1p	2008-06-05 11:22:12.000000000 +0200
+++ b/man-pages-posix-2003-a/man1p/awk.1p	2010-10-22 15:45:44.000000000 +0200
@@ -2197,7 +2197,7 @@ necessary to use parentheses to enforce 
 .LP
 The \fIawk\fP program specified in the command line is most easily
 specified within single-quotes (for example,
-'\fIprogram\fP') for applications using \fIsh\fP, because \fIawk\fP
+\fI'program'\fP) for applications using \fIsh\fP, because \fIawk\fP
 programs commonly contain
 characters that are special to the shell, including double-quotes.
 In the cases where an \fIawk\fP program contains single-quote
