--- configure.ac.OSmanOS	2011-07-25 12:51:20.000000000 +0300
+++ configure.ac	2012-11-20 13:21:36.921307899 +0200
@@ -163,27 +163,6 @@
 fi
 
 
-AC_MSG_CHECKING(how to redirect Ghostscript output to fd 3)
-if test "$DEVFD3" = "" ; then
-	v=`( if test -d /dev/fd ; then echo 3 >/dev/fd/3; fi) 3>&1 1>&2`
-	if test "$v" != "" ; then
-		DEVFD3=/dev/fd/3
-		DEVFD0=/dev/fd/0
-	else
-		DEVFD3="|$CAT >&3"
-		DEVFD0=-
-	fi
-	if test "$GS" != "GHOSTSCRIPT_NOT_FOUND" -a "$GSCHECK" != "no" ; then
-	   rm -f gs_out
-	   gs -q -dBATCH -dNOPAUSE -dSAFER -dPARANOIDSAFER -sDEVICE=laserjet "-sOutputFile=$DEVFD3" ${srcdir}/gsPrint 3>gs_out
-	   if test "!" -s gs_out ; then
-		AC_MSG_ERROR("ERROR - gs cannot handle -sOutputFile=$OUT" )
-	   fi
-	fi
-fi
-AC_MSG_RESULT(using $DEVFD3)
-AC_SUBST(DEVFD3)
-AC_SUBST(DEVFD0)
 
 # Check for Perl
 AC_PATH_PROGS(PERL,perl,/usr/bin/perl,$BINSEARCHPATH)
