--- foomatic-db-engine-4.0.7/foomatic-ppdfile.in.fix	2011-02-18 15:30:22.000000000 +0300
+++ foomatic-db-engine-4.0.7/foomatic-ppdfile.in	2011-08-09 17:52:11.296266684 +0400
@@ -195,9 +195,9 @@ sub generateppd {
 	    unshift(@drivercomponents, pop(@poidcomponents));
 	    $driver = join('-', @drivercomponents);
 	    $printer = join('-', @poidcomponents);
-	    next if (!Foomatic::DB::member($driver, @drivers) or
+	    next if (!Foomatic::DB::member($driver, @drivers) and
 		     !Foomatic::DB::member($printer, @printers));
-	    $poid = $printer;
+	    $poid = join('-', @poidcomponents);
 	    last;
 	}
 	die "ERROR: Could not determine driver name for $ppdname.ppd!\n"
