--- foomatic-db-engine-3.0-20031217/configure.ac.libdir	2003-11-19 18:31:07.000000000 +0000
+++ foomatic-db-engine-3.0-20031217/configure.ac	2003-12-17 13:23:23.975555441 +0000
@@ -43,8 +43,8 @@
 SYSCONFSEARCHPATH=/etc:/usr/etc:/usr/local/etc:/var/etc
 SBINSEARCHPATH=/usr/sbin:/sbin:/usr/local/sbin:/etc/sbin
 BINSEARCHPATH=/usr/bin:/bin:/usr/local/bin
-DATASEARCHPATH=/usr/share:/usr/local/share:/usr/lib:/usr/local/lib:/opt
-LIBSEARCHPATH=/usr/lib:/lib:/usr/local/lib:/usr/local/libexec:/opt
+DATASEARCHPATH=/usr/share:/usr/local/share:$libdir:/usr/local/lib:/opt
+LIBSEARCHPATH=$libdir:/lib:/usr/local/lib:/usr/local/libexec:/opt
 PTALPIPESSEARCHPATH=/var/run:/var:/var/ptal:/usr/var/run:/usr/var:/usr/var/ptal:/usr/local/var/run:/usr/local/var:/usr/local/var/ptal:/dev:/dev/ptal
 MTINKPIPESSEARCHPATH=/var:/var/run:/usr/var/:/usr/var/run:/usr/local/var:/usr/local/var/run:/dev
 
@@ -236,15 +236,15 @@
 AC_PATH_PROGS(CUPS_LPINFO,lpinfo,/usr/sbin/lpinfo,$SBINSEARCHPATH)
 
 dnl Paths for PDQ
-AC_PATH_DIRS(PDQ_CONF,pdq,/usr/lib/pdq,$SYS_LIB_PATH)
+AC_PATH_DIRS(PDQ_CONF,pdq,$libdir/pdq,$SYS_LIB_PATH)
 AC_PATH_PROGS(PDQ_PRINTRC,pdq/printrc,$PDQ_CONF/printrc,$SYS_LIB_PATH)
 dnl PDQ_FOOMATIC=$PDQ_CONF/drivers/foomatic
 AC_PATH_PROGS(PDQ_PRINT,pdq lpr-pdq,/usr/bin/pdq,$BINSEARCHPATH)
 dnl PDQ_JOBDIR=~/.printjobs
 
 dnl Paths for PPR
-AC_PATH_PROGS(PPR_PPRD,ppr/bin/pprd,/usr/lib/ppr/bin/pprd,$LIBSEARCHPATH)
-AC_PATH_DIRS(PPR_INTERFACES,ppr/interfaces,/usr/lib/ppr/interfaces,$LIBSEARCHPATH)
+AC_PATH_PROGS(PPR_PPRD,ppr/bin/pprd,$libdir/ppr/bin/pprd,$LIBSEARCHPATH)
+AC_PATH_DIRS(PPR_INTERFACES,ppr/interfaces,$libdir/ppr/interfaces,$LIBSEARCHPATH)
 AC_PATH_DIRS(PPR_PPDFILES,ppr/PPDFiles,/usr/share/ppr/PPDFiles,$DATASEARCHPATH)
 AC_PATH_DIRS(PPR_ETC,ppr,/etc/ppr,$SYSCONFSEARCHPATH)
 AC_PATH_PROGS(PPR_PPR,ppr lpr-ppr,/usr/bin/ppr,$BINSEARCHPATH)
