diff -Nur foomatic-db-engine-3.0-20061031-old/Makefile.in foomatic-db-engine-3.0-20061031/Makefile.in
--- foomatic-db-engine-3.0-20061031-old/Makefile.in	2006-11-08 10:33:59.000000000 +0200
+++ foomatic-db-engine-3.0-20061031/Makefile.in	2006-11-08 10:38:44.000000000 +0200
@@ -42,7 +42,7 @@
 FOOMATIC_RIP=@FOOMATIC_RIP@
 
 # foomatic-rip path for a "make inplace" installation
-LOCAL_FOOMATIC_RIP=@LOCAL_FOOMATIC_RIP@
+LOCAL_FOOMATIC_RIP=@FOOMATIC_RIP@
 
 # Locations for installing the components of this package
 BINDIR=$(bindir)
@@ -186,10 +186,10 @@
 install-bin: check-config lib/Makefile
 #	This is the only way to get the Perl-generated Makefile using
 #	$(DESTDIR) properly
-	( cd lib && \
-	  $(MAKE) PREFIX=$(PERLPREFIX) && \
-	  $(MAKE) PREFIX=$(DESTDIR)$(PERLPREFIX) pure_install \
-	)
+#	( cd lib && \
+#	  $(MAKE) PREFIX=$(PERLPREFIX) && \
+#	  $(MAKE) PREFIX=$(DESTDIR)$(PERLPREFIX) pure_install \
+#	)
 	${SRC}/mkinstalldirs $(DESTDIR)$(BINDIR)
 	${SRC}/mkinstalldirs $(DESTDIR)$(SBINDIR)
 	${SRC}/mkinstalldirs $(DESTDIR)$(ETCDIR)
