diff -Nuar a/desktoptheme/air/metadata.desktop b/desktoptheme/air/metadata.desktop
--- a/desktoptheme/air/metadata.desktop	2014-06-03 12:32:16.000000000 +0300
+++ b/desktoptheme/air/metadata.desktop	2014-08-06 19:46:04.978996607 +0300
@@ -160,7 +160,7 @@
 X-KDE-PluginInfo-EnabledByDefault=true
 
 [Wallpaper]
-defaultWallpaperTheme=Elarun
+defaultWallpaperTheme=no
 defaultFileSuffix=.png
-defaultWidth=2560
-defaultHeight=1600
+defaultWidth=1920
+defaultHeight=1080
