diff -p -up atool-0.35.0/extra/bash-completion-atool_0.1-1.bash-completion atool-0.35.0/extra/bash-completion-atool_0.1-1
--- atool-0.35.0/extra/bash-completion-atool_0.1-1.bash-completion	2007-12-10 23:01:38.000000000 +0100
+++ atool-0.35.0/extra/bash-completion-atool_0.1-1	2009-01-16 11:27:15.000000000 +0100
@@ -15,12 +15,12 @@ _atool() 
                 --quiet --verbose --page --null --explain --simulate --config=' -- $cur ) )
 		;;
 	    *)
-		_filedir '@(rar|RAR|zip|ZIP|tar.gz|tgz|tar.bz|tbz|tar.Z|tZ|tar.lzo|tzo|tar.7z|t7z|tar|jar|JAR|war|lha|lzh|7z|ace|ACE|arj|ARJ|a|arc|ARC|rpm|deb|gz|bz|bz2|Z|lzo|cpio)'
+		_filedir '@(rar|RAR|zip|ZIP|tar.gz|tgz|tar.bz|tbz|tar.Z|tZ|tar.lzo|tzo|tar.7z|t7z|tar|jar|JAR|war|lha|lzh|7z|ace|ACE|arj|ARJ|a|arc|ARC|rpm|deb|gz|bz|bz2|Z|lzo|cpio|lzma|xz)'
 		;;
 	esac
   
 	return 0
 
 }
-complete -F _atool -o filenames atool
+complete -F _atool -o filenames atool acat arepack als aunpack adiff
 
