diff -Naur kvirc.orig/data/icons/128x128/CMakeLists.txt kvirc/data/icons/128x128/CMakeLists.txt
--- kvirc.orig/data/icons/128x128/CMakeLists.txt	2010-04-17 19:23:22.000000000 +0300
+++ kvirc/data/icons/128x128/CMakeLists.txt	2010-04-19 13:00:04.000000000 +0300
@@ -15,9 +15,14 @@
 		INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/Contents/Resources/icons/128x128/apps/)
 		INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/Contents/Resources/icons/128x128/mimetypes/)
 	ELSE()
-		# Assume linux
-		INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/128x128/apps/)
-		INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/128x128/mimetypes/)
+        # Assume linux
+        IF(CMAKE_STATUS_KDE4_SUPPORT)
+            INSTALL(FILES ${iconapps} DESTINATION ${ICON_INSTALL_DIR}/hicolor/128x128/apps/)
+            INSTALL(FILES ${iconmime} DESTINATION ${ICON_INSTALL_DIR}/hicolor/128x128/mimetypes/)
+        ELSE()
+            INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/128x128/apps/)
+            INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/128x128/mimetypes/)
+        ENDIF()
 	ENDIF()
 ELSEIF(WIN32)
 	INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/icons/128x128/apps/)
diff -Naur kvirc.orig/data/icons/16x16/CMakeLists.txt kvirc/data/icons/16x16/CMakeLists.txt
--- kvirc.orig/data/icons/16x16/CMakeLists.txt	2010-04-17 19:23:21.000000000 +0300
+++ kvirc/data/icons/16x16/CMakeLists.txt	2010-04-19 12:59:43.000000000 +0300
@@ -17,8 +17,13 @@
 		INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/Contents/Resources/icons/16x16/mimetypes/)
 	ELSE()
 		# Assume linux
-		INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/16x16/apps/)
-		INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/16x16/mimetypes/)
+        IF(CMAKE_STATUS_KDE4_SUPPORT)
+            INSTALL(FILES ${iconapps} DESTINATION ${ICON_INSTALL_DIR}/hicolor/16x16/apps/)
+            INSTALL(FILES ${iconmime} DESTINATION ${ICON_INSTALL_DIR}/hicolor/16x16/mimetypes/)
+        ELSE()
+            INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/16x16/apps/)
+            INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/16x16/mimetypes/)
+        ENDIF()
 	ENDIF()
 ELSEIF(WIN32)
 	INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/icons/16x16/apps/)
diff -Naur kvirc.orig/data/icons/32x32/CMakeLists.txt kvirc/data/icons/32x32/CMakeLists.txt
--- kvirc.orig/data/icons/32x32/CMakeLists.txt	2010-04-17 19:23:21.000000000 +0300
+++ kvirc/data/icons/32x32/CMakeLists.txt	2010-04-19 13:00:34.000000000 +0300
@@ -17,8 +17,13 @@
 		INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/Contents/Resources/icons/32x32/mimetypes/)
 	ELSE()
 		# Assume linux
-		INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/32x32/apps/)
-		INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/32x32/mimetypes/)
+        IF(CMAKE_STATUS_KDE4_SUPPORT)
+            INSTALL(FILES ${iconapps} DESTINATION ${ICON_INSTALL_DIR}/hicolor/32x32/apps/)
+            INSTALL(FILES ${iconmime} DESTINATION ${ICON_INSTALL_DIR}/hicolor/32x32/mimetypes/)
+        ELSE()
+            INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/32x32/apps/)
+            INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/32x32/mimetypes/)
+        ENDIF()
 	ENDIF()
 ELSEIF(WIN32)
 	INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/icons/32x32/apps/)
diff -Naur kvirc.orig/data/icons/48x48/CMakeLists.txt kvirc/data/icons/48x48/CMakeLists.txt
--- kvirc.orig/data/icons/48x48/CMakeLists.txt	2010-04-17 19:23:22.000000000 +0300
+++ kvirc/data/icons/48x48/CMakeLists.txt	2010-04-19 13:02:48.000000000 +0300
@@ -17,10 +17,15 @@
 		INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/Contents/Resources/icons/48x48/mimetypes/)
 	ELSE()
 		# Assume linux
-		INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/48x48/apps/)
-		INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/48x48/mimetypes/)
-		#this is needed for .desktop files to find the icon
-		INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/pixmaps/)
+        IF(CMAKE_STATUS_KDE4_SUPPORT)
+            INSTALL(FILES ${iconapps} DESTINATION ${ICON_INSTALL_DIR}/hicolor/48x48/apps/)
+            INSTALL(FILES ${iconmime} DESTINATION ${ICON_INSTALL_DIR}/hicolor/48x48/mimetypes/)
+        ELSE()
+            INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/48x48/apps/)
+            INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/48x48/mimetypes/)
+        ENDIF()
+        #this is needed for .desktop files to find the icon
+        INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/pixmaps/)
 	ENDIF()
 ELSEIF(WIN32)
 	INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/icons/48x48/apps/)
diff -Naur kvirc.orig/data/icons/64x64/CMakeLists.txt kvirc/data/icons/64x64/CMakeLists.txt
--- kvirc.orig/data/icons/64x64/CMakeLists.txt	2010-04-17 19:23:22.000000000 +0300
+++ kvirc/data/icons/64x64/CMakeLists.txt	2010-04-19 13:03:15.000000000 +0300
@@ -17,8 +17,13 @@
 		INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/Contents/Resources/icons/64x64/mimetypes/)
 	ELSE()
 		# Assume linux
-		INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/64x64/apps/)
-		INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/64x64/mimetypes/)
+        IF(CMAKE_STATUS_KDE4_SUPPORT)
+            INSTALL(FILES ${iconapps} DESTINATION ${ICON_INSTALL_DIR}/hicolor/64x64/apps/)
+            INSTALL(FILES ${iconmime} DESTINATION ${ICON_INSTALL_DIR}/hicolor/64x64/mimetypes/)
+        ELSE()
+            INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/64x64/apps/)
+            INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/64x64/mimetypes/)
+        ENDIF()
 	ENDIF()
 ELSEIF(WIN32)
 	INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/icons/64x64/apps/)
diff -Naur kvirc.orig/data/icons/scalable/CMakeLists.txt kvirc/data/icons/scalable/CMakeLists.txt
--- kvirc.orig/data/icons/scalable/CMakeLists.txt	2010-04-17 19:23:22.000000000 +0300
+++ kvirc/data/icons/scalable/CMakeLists.txt	2010-04-19 13:03:52.000000000 +0300
@@ -17,8 +17,13 @@
 		INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/Contents/Resources/icons/scalable/mimetypes/)
 	ELSE()
 		# Assume linux
-		INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/scalable/apps/)
-		INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/scalable/mimetypes/)
+        IF(CMAKE_STATUS_KDE4_SUPPORT)
+            INSTALL(FILES ${iconapps} DESTINATION ${ICON_INSTALL_DIR}/hicolor/scalable/apps/)
+            INSTALL(FILES ${iconmime} DESTINATION ${ICON_INSTALL_DIR}/hicolor/scalable/mimetypes/)
+        ELSE()
+            INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/scalable/apps/)
+            INSTALL(FILES ${iconmime} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/scalable/mimetypes/)
+        ENDIF()
 	ENDIF()
 ELSEIF(WIN32)
 	INSTALL(FILES ${iconapps} DESTINATION ${CMAKE_INSTALL_PREFIX}/icons/scalable/apps/)
