Index: asciidoc-8.6.5/Makefile.in
===================================================================
--- asciidoc-8.6.5.orig/Makefile.in
+++ asciidoc-8.6.5/Makefile.in
@@ -137,7 +137,7 @@ fixconfpath:
 	done
 
 install-vim:
-	@for d in $(DESTDIR)/$(vimdir) /etc/vim; do \
+	@for d in $(DESTDIR)/$(vimdir); do \
 		if ! test -d $$d; then continue; fi ; \
 		echo "installing Vim files in $$d" ; \
 		$(INSTALL) -d $$d/syntax ; \
