Index: asciidoc-8.6.1/Makefile.in
===================================================================
--- asciidoc-8.6.1.orig/Makefile.in
+++ asciidoc-8.6.1/Makefile.in
@@ -149,7 +149,7 @@ uninstall-vim:
 
 build: fixconfpath
 
-install: all $(PROGTARGETS) $(DATATARGETS) progsymlink install-vim
+install: all $(PROGTARGETS) $(DATATARGETS) progsymlink
 
 uninstall: uninstall-vim
 	rm -f $(DESTDIR)/$(progdir)/asciidoc
