diff -uNr clamav-0.98.3/configure.ac clamav-0.98.3-ac2.68/configure.ac
--- clamav-0.98.3/configure.ac	2014-05-06 20:42:41.000000000 +0200
+++ clamav-0.98.3-ac2.68/configure.ac	2014-05-09 10:51:30.508280355 +0200
@@ -42,12 +42,12 @@
 AC_DEFINE([PACKAGE], PACKAGE_NAME, [Name of package])
 
 m4_include([m4/reorganization/version.m4])
+m4_include([m4/reorganization/build_tools.m4])
 
 LT_CONFIG_LTDL_DIR([libltdl])
 LT_INIT([dlopen disable-static])
 LTDL_INIT([recursive])
 
-m4_include([m4/reorganization/build_tools.m4])
 m4_include([m4/reorganization/headers.m4])
 
 AC_CONFIG_FILES([libltdl/Makefile])
diff -uNr clamav-0.98.3/m4/reorganization/build_tools.m4 clamav-0.98.3-ac2.68/m4/reorganization/build_tools.m4
--- clamav-0.98.3/m4/reorganization/build_tools.m4	2014-05-06 20:39:56.000000000 +0200
+++ clamav-0.98.3-ac2.68/m4/reorganization/build_tools.m4	2014-05-09 10:53:30.800262620 +0200
@@ -1,3 +1,5 @@
+AM_PROG_AR
+AM_PROG_CC_C_O
 AC_PROG_CC
 AC_PROG_INSTALL
 AC_PROG_LN_S
diff -uNr clamav-0.98.3/m4/reorganization/clamdtop.m4 clamav-0.98.3-ac2.68/m4/reorganization/clamdtop.m4
--- clamav-0.98.3/m4/reorganization/clamdtop.m4	2014-05-06 20:39:56.000000000 +0200
+++ clamav-0.98.3-ac2.68/m4/reorganization/clamdtop.m4	2014-05-09 10:51:06.963907781 +0200
@@ -6,8 +6,8 @@
 if test "$enable_clamdtop" != "no"; then
 
 AC_LIB_FIND([ncurses], [ncurses/ncurses.h],
-	    AC_LANG_PROGRAM([#include <ncurses/ncurses.h>],
-			    [initscr(); KEY_RESIZE;]),
+	    [AC_LANG_PROGRAM([#include <ncurses/ncurses.h>],
+			    [initscr(); KEY_RESIZE;])],
 	    [CURSES_CPPFLAGS="$INCNCURSES"; CURSES_LIBS="$LTLIBNCURSES";
 	     CURSES_INCLUDE="<ncurses/ncurses.h>"],
 	    [])
@@ -15,8 +15,8 @@
 if test "X$HAVE_LIBNCURSES" != "Xyes"; then
     HAVE_LIBNCURSES=
     AC_LIB_FIND([ncurses], [ncurses.h],
-	    AC_LANG_PROGRAM([#include <ncurses.h>],
-			    [initscr(); KEY_RESIZE;]),
+	    [AC_LANG_PROGRAM([#include <ncurses.h>],
+			    [initscr(); KEY_RESIZE;])],
 	    [CURSES_CPPFLAGS="$INCNCURSES"; CURSES_LIBS="$LTLIBNCURSES";
 	     CURSES_INCLUDE="<ncurses.h>"],
 	    [])
@@ -24,8 +24,8 @@
 
 if test "X$HAVE_LIBNCURSES" != "Xyes"; then
     AC_LIB_FIND([pdcurses],[curses.h],
-			    AC_LANG_PROGRAM([#include <curses.h>],
-					    [initscr(); KEY_RESIZE;]),
+			    [AC_LANG_PROGRAM([#include <curses.h>],
+					    [initscr(); KEY_RESIZE;])],
 			    [CURSES_CPPFLAGS="$INCPDCURSES";
 			     CURSES_LIBS="$LTLIBPDCURSES";
 			     CURSES_INCLUDE="<curses.h>"],
diff -uNr clamav-0.98.3/m4/reorganization/code_checks/unit_tests.m4 clamav-0.98.3-ac2.68/m4/reorganization/code_checks/unit_tests.m4
--- clamav-0.98.3/m4/reorganization/code_checks/unit_tests.m4	2014-05-06 20:39:56.000000000 +0200
+++ clamav-0.98.3-ac2.68/m4/reorganization/code_checks/unit_tests.m4	2014-05-09 10:51:06.963907781 +0200
@@ -12,7 +12,7 @@
 esac
 
 	AC_LIB_FIND([check],[check.h],
-			AC_LANG_PROGRAM([#include <check.h>],[srunner_create(0)]),
+			[AC_LANG_PROGRAM([#include <check.h>],[srunner_create(0)])],
 			[CHECK_CPPFLAGS="$INCCHECK"; CHECK_LIBS="$LTLIBCHECK $LDFLAGS"],
 			[]) 
