diff -uNr clamav-0.98.3/m4/reorganization/libs/openssl.m4 clamav-0.98.3-openssl/m4/reorganization/libs/openssl.m4
--- clamav-0.98.3/m4/reorganization/libs/openssl.m4	2014-05-06 20:39:56.000000000 +0200
+++ clamav-0.98.3-openssl/m4/reorganization/libs/openssl.m4	2014-05-09 13:51:47.581181238 +0200
@@ -22,12 +22,12 @@
     AC_MSG_ERROR([OpenSSL not found.])
 fi
 
-SSL_LDFLAGS="-L$LIBSSL_HOME/lib"
+SSL_LDFLAGS=""
 SSL_LIBS="-lssl -lcrypto"
 SSL_CPPFLAGS="-I$LIBSSL_HOME/include"
 
 save_LDFLAGS="$LDFLAGS"
-LDFLAGS="-L$LIBSSL_HOME/lib $SSL_LIBS"
+LDFLAGS="$SSL_LDFLAGS $SSL_LIBS"
 
 save_CFLAGS="$CFLAGS"
 CFLAGS="$SSL_CPPFLAGS" 
