--- gpac/modules/directfb_out/Makefile.link	2012-04-13 03:16:03.000000000 +0000
+++ gpac/modules/directfb_out/Makefile	2012-04-13 03:16:16.000000000 +0000
@@ -31,7 +31,7 @@
 
 
 $(LIB): $(OBJS)
-	$(CC) $(SHFLAGS) -o ../../bin/gcc/$@ $(OBJS) $(LDFLAGS) -L../../bin/gcc -lgpac
+	$(CC) $(SHFLAGS) -o ../../bin/gcc/$@ $(OBJS) $(LDFLAGS) -L../../bin/gcc -lgpac $(DIRECTFB_LIB)
                                                                                                
 %.o: %.c
 	$(CC) $(CFLAGS) -c -o $@ $< 
