diff -Nur irda-utils-0.9.18-old/etc/Makefile irda-utils-0.9.18/etc/Makefile
--- irda-utils-0.9.18-old/etc/Makefile	2009-05-04 12:45:28.000000000 +0300
+++ irda-utils-0.9.18/etc/Makefile	2009-05-04 12:46:05.000000000 +0300
@@ -1,6 +1,6 @@
 include ../output.mak
 
-CC = gcc $(RPM_OPT_FLAGS)
+CC = pardusCC $(RPM_OPT_FLAGS)
 LD = ld
 RM = rm -f
 
diff -Nur irda-utils-0.9.18-old/findchip/Makefile irda-utils-0.9.18/findchip/Makefile
--- irda-utils-0.9.18-old/findchip/Makefile	2009-05-04 12:45:28.000000000 +0300
+++ irda-utils-0.9.18/findchip/Makefile	2009-05-04 12:46:05.000000000 +0300
@@ -31,8 +31,8 @@
 ######################################################################
 include ../output.mak
 
-CC      = gcc
-CFLAGS= $(RPM_OPT_FLAGS) $(SYS_INCLUDES) -O2 -Wall -Wstrict-prototypes
+CC      = pardusCC
+CFLAGS= $(RPM_OPT_FLAGS) $(SYS_INCLUDES) pardusCFLAGS -Wall -Wstrict-prototypes
 
 SRCS = findchip.c smc.c nsc.c winbond.c
 OBJS = $(SRCS:.c=.o)
diff -Nur irda-utils-0.9.18-old/irattach/Makefile irda-utils-0.9.18/irattach/Makefile
--- irda-utils-0.9.18-old/irattach/Makefile	2009-05-04 12:45:28.000000000 +0300
+++ irda-utils-0.9.18/irattach/Makefile	2009-05-04 12:46:05.000000000 +0300
@@ -25,10 +25,10 @@
 ######################################################################
 include ../output.mak
 
-CC           = gcc
+CC           = pardusCC
 LD           = ld
 RM           = rm -f
-CFLAGS       = -O2 -W -Wall -Wstrict-prototypes -Wshadow $(SYS_INCLUDES)
+CFLAGS       = pardusCFLAGS -W -Wall -Wstrict-prototypes -Wshadow $(SYS_INCLUDES)
 
 SYS_INCLUDES = -I/usr/include
 SYS_LIBPATH  = -L/usr/lib -L/usr/local/lib 
diff -Nur irda-utils-0.9.18-old/irdadump/Makefile irda-utils-0.9.18/irdadump/Makefile
--- irda-utils-0.9.18-old/irdadump/Makefile	2009-05-04 12:45:28.000000000 +0300
+++ irda-utils-0.9.18/irdadump/Makefile	2009-05-04 12:46:05.000000000 +0300
@@ -1,6 +1,6 @@
 include ../output.mak
 
-CC = gcc
+CC = pardusCC
 LD = ld
 AR = ar
 RANLIB = ranlib
@@ -19,7 +19,7 @@
 SYS_LIBRARIES = 
 SYS_LIBPATH = -L/usr/lib -L/usr/local/lib 
 
-CFLAGS= -O2 -W -Wall -I. `pkg-config --cflags glib-2.0`
+CFLAGS= pardusCFLAGS -W -Wall -I. `pkg-config --cflags glib-2.0`
 
 LIBIRDADUMP_OBJS = irdadump.o irlap.o irlmp.o obex.o ircomm.o netbuf.o capture.o
 IRDADUMP_OBJS = main.o
diff -Nur irda-utils-0.9.18-old/irdaping/Makefile irda-utils-0.9.18/irdaping/Makefile
--- irda-utils-0.9.18-old/irdaping/Makefile	2009-05-04 12:45:28.000000000 +0300
+++ irda-utils-0.9.18/irdaping/Makefile	2009-05-04 12:46:05.000000000 +0300
@@ -25,7 +25,7 @@
 ######################################################################
 include ../output.mak
 
-CC = gcc
+CC = pardusCC
 LD = ld
 RM = rm -f
 
diff -Nur irda-utils-0.9.18-old/irnetd/Makefile irda-utils-0.9.18/irnetd/Makefile
--- irda-utils-0.9.18-old/irnetd/Makefile	2009-05-04 12:45:28.000000000 +0300
+++ irda-utils-0.9.18/irnetd/Makefile	2009-05-04 12:46:05.000000000 +0300
@@ -25,7 +25,7 @@
 ######################################################################
 include ../output.mak
 
-CC = gcc
+CC = pardusCC
 LD = ld
 RM = rm -f
 
@@ -35,7 +35,7 @@
 LIBRARIES = $(SYS_LIBRARIES)
 LIBPATH   = $(SYS_LIBPATH)
 
-CFLAGS= $(RPM_OPT_FLAGS) -O2 -Wall -Wstrict-prototypes
+CFLAGS= $(RPM_OPT_FLAGS) pardusCFLAGS -Wall -Wstrict-prototypes
 
 OBJS = irnetd.o
 TARGETS = irnetd
diff -Nur irda-utils-0.9.18-old/irsockets/Makefile irda-utils-0.9.18/irsockets/Makefile
--- irda-utils-0.9.18-old/irsockets/Makefile	2009-05-04 12:45:28.000000000 +0300
+++ irda-utils-0.9.18/irsockets/Makefile	2009-05-04 12:46:05.000000000 +0300
@@ -25,7 +25,7 @@
 ######################################################################
 include ../output.mak
 
-CC = gcc
+CC = pardusCC
 LD = ld
 RM = rm -f
 
@@ -42,7 +42,7 @@
 SYS_LIBRARIES = 
 SYS_LIBPATH = -L/usr/lib -L/usr/local/lib 
 
-CFLAGS= $(SYS_INCLUDES) $(RPM_OPT_FLAGS) -O2 -Wall -Wstrict-prototypes
+CFLAGS= $(SYS_INCLUDES) $(RPM_OPT_FLAGS) pardusCFLAGS -Wall -Wstrict-prototypes
 
 SRCS = irprintf.c irprintfx.c irscanf.c irscanfx.c irdaspray.c ias_query.c send_ultra.c recv_ultra.c
 OBJS = $(SRCS:.c=.o)
diff -Nur irda-utils-0.9.18-old/Makefile irda-utils-0.9.18/Makefile
--- irda-utils-0.9.18-old/Makefile	2009-05-04 12:45:28.000000000 +0300
+++ irda-utils-0.9.18/Makefile	2009-05-04 12:46:05.000000000 +0300
@@ -28,7 +28,7 @@
 TARGET = test
 
 DIRS = irattach irdaping etc man irnetd psion tekram findchip irdadump smcinit
-CFLAGS= -O2 -W -Wall
+CFLAGS= pardusCFLAGS -W -Wall
 
 all:
 	@-(set -e ; for d in $(DIRS) ; do $(MAKE) $(MAKE_OUTPUT) -C $$d $@ ; done)
diff -Nur irda-utils-0.9.18-old/psion/Makefile irda-utils-0.9.18/psion/Makefile
--- irda-utils-0.9.18-old/psion/Makefile	2009-05-04 12:45:28.000000000 +0300
+++ irda-utils-0.9.18/psion/Makefile	2009-05-04 12:46:05.000000000 +0300
@@ -1,6 +1,6 @@
 include ../output.mak
 
-CC=gcc
+CC=pardusCC
 
 PSION_SRC= irpsion5.c
 PSION_OBJS= irpsion5.o
diff -Nur irda-utils-0.9.18-old/smcinit/Makefile irda-utils-0.9.18/smcinit/Makefile
--- irda-utils-0.9.18-old/smcinit/Makefile	2009-05-04 12:45:28.000000000 +0300
+++ irda-utils-0.9.18/smcinit/Makefile	2009-05-04 12:46:05.000000000 +0300
@@ -1,6 +1,6 @@
 include ../output.mak
 
-CC = gcc
+CC = pardusCC
 RM = rm -f
 
 RM_CMD = $(ECMD)$(RM) *.BAK *.bak *.o ,* *~ *.a
@@ -16,7 +16,7 @@
 SYS_LIBRARIES = 
 SYS_LIBPATH = -L/usr/lib -L/usr/local/lib 
 
-CFLAGS= -O2 -W -Wall -I.
+CFLAGS= pardusCFLAGS -W -Wall -I.
 
 SMCINIT_OBJS = smcinit.o tosh1800-smcinit.o tosh2450-smcinit.o 
 
diff -Nur irda-utils-0.9.18-old/tekram/Makefile irda-utils-0.9.18/tekram/Makefile
--- irda-utils-0.9.18-old/tekram/Makefile	2009-05-04 12:45:28.000000000 +0300
+++ irda-utils-0.9.18/tekram/Makefile	2009-05-04 12:46:05.000000000 +0300
@@ -25,7 +25,7 @@
 ######################################################################
 include ../output.mak
 
-CC = gcc
+CC = pardusCC
 LD = ld
 RM = rm -f
 
