--- gtk2_ardour/ardour_ui.h~	2008-06-17 07:58:37.000000000 -0700
+++ gtk2_ardour/ardour_ui.h	2008-09-21 08:19:26.000000000 -0700
@@ -53,6 +53,9 @@
 #include <gtkmm/textbuffer.h>
 #include <gtkmm/adjustment.h>
 #include <gtkmm2ext/gtk_ui.h>
+#ifdef HOST_NOT_FOUND
+#undef HOST_NOT_FOUND
+#endif
 #include <gtkmm2ext/click_box.h>
 #include <gtkmm2ext/stateful_button.h>
 #include <gtkmm2ext/bindable_button.h>
