diff -rupN ardour-2.8.11.old/gtk2_ardour/ardour.sh.in ardour-2.8.11/gtk2_ardour/ardour.sh.in
--- ardour-2.8.11.old/gtk2_ardour/ardour.sh.in	2009-02-24 07:37:42.000000000 -0500
+++ ardour-2.8.11/gtk2_ardour/ardour.sh.in	2010-09-29 15:23:27.000000000 -0400
@@ -1,8 +1,8 @@
 #!/bin/sh
 
-export GTK_PATH=%INSTALL_PREFIX%/%LIBDIR%/ardour2:$GTK_PATH
+export GTK_PATH=%INSTALL_PREFIX%/%LIBDIR%/ardour2${GTK_PATH:+:$GTK_PATH}
 
-export LD_LIBRARY_PATH=%INSTALL_PREFIX%/%LIBDIR%/ardour2:$LD_LIBRARY_PATH 
+export LD_LIBRARY_PATH=%INSTALL_PREFIX%/%LIBDIR%/ardour2${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
 # DYLD_LIBRARY_PATH is for Darwin
 export DYLD_LIBRARY_PATH=$LD_LIBRARY_PATH
 
