diff -rupN ardour-2.8.3.old/libs/midi++2/SConscript ardour-2.8.3/libs/midi++2/SConscript
--- ardour-2.8.3.old/libs/midi++2/SConscript	2009-03-19 13:52:43.000000000 -0400
+++ ardour-2.8.3/libs/midi++2/SConscript	2009-11-12 04:01:48.000000000 -0500
@@ -42,7 +42,7 @@ if env['SYSMIDI'] == 'CoreMIDI':
 else:
    sysdep_src = [ 'alsa_sequencer_midiport.cc' ]
    midi2.Append (CCFLAGS="-DWITH_ALSA")
-
+   midi2.Append (LIBS = "-lasound")
 midi2.Append(CCFLAGS="-D_REENTRANT -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE")
 midi2.Append(CCFLAGS="-DLIBSIGC_DISABLE_DEPRECATED")
 
diff -rupN ardour-2.8.3.old/libs/vamp-sdk/SConscript ardour-2.8.3/libs/vamp-sdk/SConscript
--- ardour-2.8.3.old/libs/vamp-sdk/SConscript	2009-03-26 11:20:21.000000000 -0400
+++ ardour-2.8.3/libs/vamp-sdk/SConscript	2009-11-12 04:04:32.000000000 -0500
@@ -27,7 +27,7 @@ vampsdk.Merge ([libraries['fftw3'], libr
 # HAVE_FFTW3 is used to help improve some performance aspects of VAMP's InputDomainAdapter
 
 vampsdk.Append (CPPATH='#libs/vamp-sdk/vamp', CXXFLAGS="-Ilibs/vamp-sdk -DHAVE_FFTW3")
-
+vampsdk.Append (LIBS = "-ldl")
 libvampsdk = vampsdk.SharedLibrary('vampsdk', vampsdk_files)
 libvamphostsdk = vampsdk.SharedLibrary('vamphostsdk', vamphostsdk_files)
 
