diff -Naur ardour-2.8.2-orig/gtk2_ardour/ardour2.desktop.in ardour-2.8.2/gtk2_ardour/ardour2.desktop.in
--- ardour-2.8.2-orig/gtk2_ardour/ardour2.desktop.in	2009-07-30 11:58:00.601454407 +0300
+++ ardour-2.8.2/gtk2_ardour/ardour2.desktop.in	2009-07-30 11:58:55.248454557 +0300
@@ -1,6 +1,9 @@
 [Desktop Entry]
 Name=Ardour
 Comment=Ardour Digital Audio Workstation
+Comment[tr]=Ardour Sayısal Ses Merkezi
+GenericName=Ardour Digital Audio Workstation
+GenericName[tr]=Ardour Sayısal Ses Merkezi
 Exec=ardour2
 Icon=ardour2
 Terminal=false
diff -Naur ardour-2.8.2-orig/gtk2_ardour/ardour2.xml ardour-2.8.2/gtk2_ardour/ardour2.xml
--- ardour-2.8.2-orig/gtk2_ardour/ardour2.xml	2009-07-30 11:57:57.421453998 +0300
+++ ardour-2.8.2/gtk2_ardour/ardour2.xml	2009-07-30 12:11:48.546328939 +0300
@@ -3,6 +3,7 @@
   <mime-type type="application/x-ardour2">
   <sub-class-of type="application/xml"/>
   <comment>Ardour session file</comment>
+  <comment xml:lang="tr">Ardour oturum dosyası</comment>
   <glob pattern="*.ardour"/>
   </mime-type>
 </mime-info>
