Index: ardour-2.8.9/libs/surfaces/wiimote/SConscript
===================================================================
--- ardour-2.8.9.orig/libs/surfaces/wiimote/SConscript
+++ ardour-2.8.9/libs/surfaces/wiimote/SConscript
@@ -29,7 +29,6 @@ wiimote.Append(CCFLAGS="-D_REENTRANT -D_
 wiimote.Append(CXXFLAGS="-DDATA_DIR=\\\""+final_prefix+"/share\\\"")
 wiimote.Append(CXXFLAGS="-DCONFIG_DIR=\\\""+final_config_prefix+"\\\"")
 wiimote.Append(CXXFLAGS="-DLOCALEDIR=\\\""+final_prefix+"/share/locale\\\"")
-wiimote.Append(LINKFLAGS="-lcwiid")
 #wiimote.Append(CPPPATH = libraries['jack'].get('CPPPATH', []))
 
 wiimote.Merge ([
@@ -42,7 +41,8 @@ wiimote.Merge ([
 	libraries['usb'],
 	libraries['xml'],
 	libraries['glib2'],
-	libraries['glibmm2']
+	libraries['glibmm2'],
+	libraries['cwiid'],
 	])
 
 libardour_wiimote = wiimote.SharedLibrary('ardour_wiimote', wiimote_files)
Index: ardour-2.8.9/SConstruct
===================================================================
--- ardour-2.8.9.orig/SConstruct
+++ ardour-2.8.9/SConstruct
@@ -1080,6 +1080,8 @@ if env['SYSLIBS']:
     libraries['libgnomecanvasmm'] = LibraryInfo()
     libraries['libgnomecanvasmm'].ParseConfig ('pkg-config --cflags --libs libgnomecanvasmm-2.6')
 
+    libraries['cwiid'] = LibraryInfo(LIBS="cwiid")
+
 #    libraries['libglademm'] = LibraryInfo()
 #    libraries['libglademm'].ParseConfig ('pkg-config --cflags --libs libglademm-2.4')
 
