Index: ardour-2.8.11/gtk2_ardour/SConscript
===================================================================
--- ardour-2.8.11.orig/gtk2_ardour/SConscript
+++ ardour-2.8.11/gtk2_ardour/SConscript
@@ -523,10 +523,10 @@ if env['FREEDESKTOP']:
 	env.Alias('install', env.InstallAs(os.path.join(desktop_icon_install_prefix, '22x22', 'mimetypes', 'application-x-ardour2.png'), 'icons/application-x-ardour_22px.png'))
 	env.Alias('install', env.InstallAs(os.path.join(desktop_icon_install_prefix, '32x32', 'mimetypes', 'application-x-ardour2.png'), 'icons/application-x-ardour_32px.png'))
 	env.Alias('install', env.InstallAs(os.path.join(desktop_icon_install_prefix, '48x48', 'mimetypes', 'application-x-ardour2.png'), 'icons/application-x-ardour_48px.png'))
-	env.Alias('install', env.Command (os.path.join(install_prefix, 'share', 'mime'), [], 'update-mime-database $TARGET'))
+	# env.Alias('install', env.Command (os.path.join(install_prefix, 'share', 'mime'), [], 'update-mime-database $TARGET'))
 	# Update the icon cache #
 	env.Alias('install', env.Command (desktop_icon_install_prefix, [], 'touch --no-create $TARGET'))
-	env.Alias('install', env.Command (desktop_icon_install_prefix, [], 'gtk-update-icon-cache $TARGET'))
+	#env.Alias('install', env.Command (desktop_icon_install_prefix, [], 'gtk-update-icon-cache $TARGET'))
 	# Make the ardour2.desktop file and install it #
 	env.Alias('install', env.Command ('ardour2.desktop', 'ardour2.desktop.in', 'cat $SOURCES  > $TARGET'))
 	env.Alias('install', env.Install(os.path.join(install_prefix, 'share', 'applications'), 'ardour2.desktop'))
@@ -542,10 +542,7 @@ if env['FREEDESKTOP']:
 			       Delete (desktop_icon_install_prefix + '/16x16/mimetypes/application-x-ardour2.png'),
 			       Delete (desktop_icon_install_prefix + '/22x22/mimetypes/application-x-ardour2.png'),
 			       Delete (desktop_icon_install_prefix + '/32x32/mimetypes/application-x-ardour2.png'),
-			       Delete (desktop_icon_install_prefix + '/48x48/mimetypes/application-x-ardour2.png'),
-                               Action ('update-mime-database ' + install_prefix + '/share/mime'),
-                               Action ('gtk-update-icon-cache ' + desktop_icon_install_prefix),
-                               Action ('update-desktop-database ' + install_prefix + '/share/applications')])
+			       Delete (desktop_icon_install_prefix + '/48x48/mimetypes/application-x-ardour2.png')])
 	env.Alias('uninstall', remove_desktop_files)
 
 #dist
