diff -Nur shadow-4.1.4.2-old//libmisc/copydir.c shadow-4.1.4.2/libmisc/copydir.c
--- shadow-4.1.4.2-old//libmisc/copydir.c	2010-09-02 23:33:08.830999791 +0300
+++ shadow-4.1.4.2/libmisc/copydir.c	2010-09-02 23:54:15.030000304 +0300
@@ -204,6 +204,7 @@
 	bool set_orig = false;
 	struct DIRECT *ent;
 	DIR *dir;
+	int len = 0;
 
 	/*
 	 * Make certain both directories exist.  This routine is called
@@ -243,6 +244,12 @@
 			char *dst_name;
 			size_t src_len = strlen (ent->d_name) + 2;
 			size_t dst_len = strlen (ent->d_name) + 2;
+
+			/* Skip copying of .newconfig files */
+			len = strlen (ent->d_name);
+			if (len > 10 && strcmp ((ent->d_name+len-10), ".newconfig") == 0)
+				continue;
+
 			src_len += strlen (src_root);
 			dst_len += strlen (dst_root);
 
