diff -up ypserv-2.24/configure.in.aliases ypserv-2.24/configure.in
--- ypserv-2.24/configure.in.aliases	2011-05-06 11:57:05.745509195 +0200
+++ ypserv-2.24/configure.in	2011-05-06 11:57:16.327509192 +0200
@@ -149,7 +149,7 @@ fi
 AC_SUBST(MAKE)
 
 dnl Check for aliases path
-AC_PATH_PROG(ALIASES, aliases, /etc/mail/aliases, /etc/mail:/etc)
+AC_PATH_PROG(ALIASES, aliases, /etc/aliases, /etc/mail:/etc)
 
 
 
