diff -ur gnome-keyring-2.30.3.orig/pkcs11/gck/Makefile.am gnome-keyring-2.30.3/pkcs11/gck/Makefile.am
--- gnome-keyring-2.30.3.orig/pkcs11/gck/Makefile.am	2010-06-21 16:51:38.000000000 +0300
+++ gnome-keyring-2.30.3/pkcs11/gck/Makefile.am	2010-08-22 15:24:40.986008640 +0300
@@ -68,7 +68,8 @@
 	$(top_builddir)/egg/libegg.la \
 	$(GOBJECT_LIBS) \
     	$(LIBGCRYPT_LIBS) \
-	$(GLIB_LIBS) 
+	$(GLIB_LIBS) \
+	$(LIBTASN1_LIBS)
 	
 gck-marshal.h: gck-marshal.list $(GLIB_GENMARSHAL)
 	$(GLIB_GENMARSHAL) $< --header --prefix=gck_marshal > $@
