--- tests/test_smime.py	2010-02-09 17:21:02.428817088 +0100
+++ /home/mitr/m2crypto-tests/current-tests/tests/test_smime.py	2010-02-10 18:07:33.785788459 +0100
@@ -37,7 +37,7 @@
         buf = BIO.MemoryBuffer(self.cleartext)
         s = SMIME.SMIME()
         s.load_key('tests/signer_key.pem', 'tests/signer.pem')
-        p7 = s.sign(buf)
+        p7 = s.sign(buf, SMIME.PKCS7_DETACHED)
         assert len(buf) == 0
         assert p7.type() == SMIME.PKCS7_SIGNED, p7.type()
         assert isinstance(p7, SMIME.PKCS7), p7
@@ -73,9 +73,8 @@
         
         p7, data = SMIME.smime_load_pkcs7_bio(self.signed)
         
-        assert data.read() == self.cleartext
         assert isinstance(p7, SMIME.PKCS7), p7
-        v = s.verify(p7)
+        v = s.verify(p7, data)
         assert v == self.cleartext
     
         t = p7.get0_signers(sk)
@@ -95,7 +94,6 @@
         s.set_x509_store(st)
         
         p7, data = SMIME.smime_load_pkcs7_bio(self.signed)
-        assert data.read() == self.cleartext
         assert isinstance(p7, SMIME.PKCS7), p7
         self.assertRaises(SMIME.PKCS7_Error, s.verify, p7) # Bad signer
 
@@ -169,7 +167,7 @@
         s.set_cipher(SMIME.Cipher('des_ede3_cbc'))
         
         tmp = BIO.MemoryBuffer()
-        s.write(tmp, p7, buf)
+        s.write(tmp, p7)
 
         p7 = s.encrypt(tmp)
         
@@ -211,6 +209,7 @@
         assert p7.write(f) == 1
         f.close()
 
+        p7 = s.sign(BIO.MemoryBuffer('some text'), SMIME.PKCS7_DETACHED)
         self.filenameSmime = 'tests/sig.p7s'
         f = BIO.openfile(self.filenameSmime, 'wb')
         assert s.write(f, p7, BIO.MemoryBuffer('some text')) == 1, Err.get_error()
@@ -220,7 +219,7 @@
         buf = BIO.MemoryBuffer()
         assert SMIME.load_pkcs7(self.filename).write_der(buf) == 1
         s = buf.read()
-        assert len(s) == 1204, len(s)
+        assert len(s) in (1204, 1243), len(s)
         
     def test_load_pkcs7(self):
         assert SMIME.load_pkcs7(self.filename).type() == SMIME.PKCS7_SIGNED
