diff -up ./joe-3.7/main.c.pom ./joe-3.7/main.c
--- joe-3.7/main.c.pom	2008-10-27 04:01:11.000000000 +0100
+++ joe-3.7/main.c	2010-02-09 13:40:14.000000000 +0100
@@ -431,7 +431,7 @@ int main(int argc, char **real_argv, cha
 				b->orphan = 1;
 				b->oldcur = pdup(b->bof, USTR "main");
 				pline(b->oldcur, get_file_pos(b->name));
-				p_goto_bol(bw->cursor);
+				p_goto_bol(b->oldcur);
 				line = b->oldcur->line - (maint->h - 1) / 2;
 				if (line < 0)
 					line = 0;
