diff -ur ltrace-0.5.orig/sysdeps/linux-gnu/trace.c ltrace-0.5/sysdeps/linux-gnu/trace.c
--- ltrace-0.5.orig/sysdeps/linux-gnu/trace.c	2006-04-24 19:25:37.000000000 +0300
+++ ltrace-0.5/sysdeps/linux-gnu/trace.c	2008-01-10 16:56:45.000000000 +0200
@@ -5,7 +5,7 @@
 #include <unistd.h>
 #include <sys/types.h>
 #include "ptrace.h"
-#include <asm/unistd.h>
+#include <asm/unistd_32.h>
 
 #include "ltrace.h"
 #include "options.h"
