Index: xorg-server-1.6.1.901/hw/xfree86/modes/xf86EdidModes.c
===================================================================
--- xorg-server-1.6.1.901.orig/hw/xfree86/modes/xf86EdidModes.c
+++ xorg-server-1.6.1.901/hw/xfree86/modes/xf86EdidModes.c
@@ -175,6 +175,10 @@ static Bool quirk_detailed_use_maximum_s
 	DDC->vendor.prod_id == 6400)
 	return TRUE;
 
+    if (memcmp (DDC->vendor.name, "APP", 4) == 0 &&
+	DDC->vendor.prod_id == 40069)
+	return TRUE;
+
     return FALSE;
 }
 
