Index: build-5.2/packages/kchmviewer.desktop
===================================================================
--- build-5.2.orig/packages/kchmviewer.desktop
+++ build-5.2/packages/kchmviewer.desktop
@@ -7,5 +7,7 @@ Encoding=UTF-8
 Exec=kchmviewer %f
 Icon=kchmviewer
 GenericName=CHM file viewer
+GenericName[tr]=CHM Dosya Gösterici
 Comment=A viewer of CHM (MS-Help) files
+Comment[tr]=MS-Yardım dosyaları için bir gösterici
 MimeType=application/chm;application/x-chm;
