--- setup.py.orig	2008-01-14 04:20:11.000000000 +0200
+++ setup.py	2008-08-28 09:58:37.000000000 +0300
@@ -6,7 +6,7 @@
 to install this package.
 """
 
-from distutils.core import setup
+from setuptools import setup
 from distutils.command.install import INSTALL_SCHEMES
 import sys
 import os
@@ -36,18 +36,10 @@
 cp_license="BSD"
 packages=[
     "cherrypy", "cherrypy.lib", "cherrypy.lib.filter",
-    "cherrypy.tutorial", "cherrypy.test", "cherrypy.filters",
+     "cherrypy.test", "cherrypy.filters",
 ]
 download_url="http://download.cherrypy.org/cherrypy/2.3.0/"
 data_files=[
-    ('cherrypy/tutorial',
-        [
-            'cherrypy/tutorial/tutorial.conf',
-            'cherrypy/tutorial/README.txt',
-            'cherrypy/tutorial/pdf_file.pdf',
-            'cherrypy/tutorial/custom_error.html',
-        ]
-    ),
     ('cherrypy', ['cherrypy/favicon.ico',]),
     ('cherrypy/test', ['cherrypy/test/style.css',]),
     ('cherrypy/test/static', ['cherrypy/test/static/index.html',
