diff -up libtunepimp-0.5.3/lib/fileio.cpp.gcc44 libtunepimp-0.5.3/lib/fileio.cpp
--- libtunepimp-0.5.3/lib/fileio.cpp.gcc44	2006-11-18 04:52:33.000000000 -0600
+++ libtunepimp-0.5.3/lib/fileio.cpp	2009-03-06 13:33:44.000000000 -0600
@@ -122,7 +124,8 @@ int taccess(const char *pathname, int mo
 
 void tmktempname(const char *path, char *newPath, int newPathLen)
 {
-    char *ptr, *temp;
+    const char *ptr;
+    char *temp;
 
     temp = (char *)malloc(strlen(path) + 32);
     ptr = strrchr(path, dirSepChar);
