--- src/gtk-gui/callbacks.cxx.str	2009-04-07 10:15:55.000000000 +0200
+++ src/gtk-gui/callbacks.cxx	2009-04-07 10:18:05.000000000 +0200
@@ -113,6 +113,7 @@
                               GTK_DIALOG_DESTROY_WITH_PARENT,
                               GTK_MESSAGE_ERROR,
                               GTK_BUTTONS_CLOSE,
+			      "%s",
                               message);
     gtk_dialog_run (GTK_DIALOG (dialog));
     gtk_widget_destroy (dialog);
@@ -160,7 +161,7 @@
 	buffer = gtk_text_view_get_buffer(txtOutput);
 	gtk_text_buffer_get_end_iter(buffer, &iter);
 	g_io_channel_read_line(source, &out, &length, NULL, NULL);
-	g_print(out);
+	g_print("%s",out);
 	gtk_text_buffer_insert(buffer, &iter, out, length);
 	g_free(out);
 	return true;
