diff -up crack-attack-1.1.14/src/TextureLoader.h~ crack-attack-1.1.14/src/TextureLoader.h
--- crack-attack-1.1.14/src/TextureLoader.h~	2008-01-04 23:07:02.000000000 +0100
+++ crack-attack-1.1.14/src/TextureLoader.h	2008-01-04 23:07:02.000000000 +0100
@@ -28,6 +28,7 @@
 #define TEXTURELOADER_H
 
 #include <GL/glut.h>
+#include <string.h>
 
 #include "glext.h"
 #include "sstream.h"
Index: crack-attack-1.1.14/src/Communicator.cxx
===================================================================
--- crack-attack-1.1.14/src/Communicator.cxx
+++ crack-attack-1.1.14/src/Communicator.cxx	2009-05-26 09:50:00.000000000 -0400
@@ -30,6 +30,7 @@
 #include <cassert>
 #include <iostream>
 #include <sys/types.h>
+#include <cstdio>
 #include <cstring>
 #include <errno.h>
 
