Index: opentts-0.1/configure.ac
===================================================================
--- opentts-0.1.orig/configure.ac
+++ opentts-0.1/configure.ac
@@ -236,7 +236,7 @@ AM_CONDITIONAL(espeak_support, test $wit
 AS_IF([test $with_flite != "no"],
 	[AC_CHECK_LIB([flite],
 		[flite_init],
-		[flite_basic="-lflite"],
+		[flite_basic="-lflite -lasound"],
 		[AS_IF([test $with_flite = "yes"],
 			[AC_MSG_FAILURE([libflite is not available])])],
 		[-lm])
