--- /home/martin/debian/clanbomber-1.05/clanbomber/ClanBomber.cpp	2004-02-16 19:42:09.000000000 +0100
+++ clanbomber-1.05/clanbomber/ClanBomber.cpp	2004-02-18 12:40:47.000000000 +0100
@@ -82,8 +82,6 @@
 
 	try
 	{
-		init_modules();
-
 		bool fullscreen = false;
 #ifdef WIN32
 		fullscreen = true;
@@ -91,8 +89,16 @@
 
 		if (argc > 1) {
 			if (strcmp (argv[1], "-datafile") == 0) {
+				CL_SetupCore::init();
+				CL_SetupDisplay::init(true);
+				CL_SetupSound::init(true);
+
 				CL_DatafileCompiler::write("clanbomber.scr", "clanbomber.dat");
-				quit_app(0);
+
+				CL_SetupSound::deinit();
+				CL_SetupDisplay::deinit();
+				CL_SetupCore::deinit();
+				return 0;
 			}
 			else if (strcmp (argv[1], "-fullscreen") == 0) {
 				fullscreen = true;
@@ -107,10 +113,11 @@
 					strcmp (argv[1], "/?") == 0 ||
 					strcmp (argv[1], "/h") == 0) {
 				cout << "usage: clanbomber [-fullscreen | -window]" << endl;
-				quit_app(0);
+				return 0;
 			}
 		}
 
+		init_modules();
 		srand( (long)time(NULL) );
 
 		CL_Display::set_videomode(800, 600, 16, fullscreen);
