diff -Nuar clanbomber-1.05.orig//clanbomber/Makefile.am clanbomber-1.05/clanbomber/Makefile.am
--- clanbomber-1.05.orig//clanbomber/Makefile.am	2011-02-02 15:03:47.495000015 +0200
+++ clanbomber-1.05/clanbomber/Makefile.am	2011-02-02 15:04:03.176000030 +0200
@@ -31,10 +31,10 @@
 .PHONY: files
 
 files:
-        @files=s $(DISTFILES) 2> /dev/null ; for p in $$files; do \
+	@files=s $(DISTFILES) 2> /dev/null ; for p in $$files; do \
 	  echo $$p; \
         done
-        @for subdir in $(SUBDIRS); do \
+	    @for subdir in $(SUBDIRS); do \
 	  files=d $$subdir; $(MAKE) files | grep -v "make\[[1-9]\]"; \
           for file in $$files; do \
             echo $$subdir/$$file; \
diff -Nuar clanbomber-1.05.orig//clanbomber/maps/Makefile.am clanbomber-1.05/clanbomber/maps/Makefile.am
--- clanbomber-1.05.orig//clanbomber/maps/Makefile.am	2011-02-02 15:03:47.495000015 +0200
+++ clanbomber-1.05/clanbomber/maps/Makefile.am	2011-02-02 15:04:38.263000041 +0200
@@ -106,10 +106,10 @@
 .PHONY: files
 
 files:
-        @files=s $(DISTFILES) 2> /dev/null ; for p in $$files; do \
+	@files=s $(DISTFILES) 2> /dev/null ; for p in $$files; do \
 	  echo $$p; \
         done
-        @for subdir in $(SUBDIRS); do \
+	    @for subdir in $(SUBDIRS); do \
 	  files=d $$subdir; $(MAKE) files | grep -v "make\[[1-9]\]"; \
           for file in $$files; do \
             echo $$subdir/$$file; \
diff -Nuar clanbomber-1.05.orig//clanbomber/music/Makefile.am clanbomber-1.05/clanbomber/music/Makefile.am
--- clanbomber-1.05.orig//clanbomber/music/Makefile.am	2011-02-02 15:03:47.504000015 +0200
+++ clanbomber-1.05/clanbomber/music/Makefile.am	2011-02-02 15:04:19.671000014 +0200
@@ -10,10 +10,10 @@
 .PHONY: files
 
 files:
-        @files=s $(DISTFILES) 2> /dev/null ; for p in $$files; do \
+	@files=s $(DISTFILES) 2> /dev/null ; for p in $$files; do \
 	  echo $$p; \
         done
-        @for subdir in $(SUBDIRS); do \
+	    @for subdir in $(SUBDIRS); do \
 	  files=d $$subdir; $(MAKE) files | grep -v "make\[[1-9]\]"; \
           for file in $$files; do \
             echo $$subdir/$$file; \
diff -Nuar clanbomber-1.05.orig//clanbomber/pics/Makefile.am clanbomber-1.05/clanbomber/pics/Makefile.am
--- clanbomber-1.05.orig//clanbomber/pics/Makefile.am	2011-02-02 15:03:47.497000015 +0200
+++ clanbomber-1.05/clanbomber/pics/Makefile.am	2011-02-02 15:04:57.687000017 +0200
@@ -3,10 +3,10 @@
 .PHONY: files
 
 files:
-        @files=s $(DISTFILES) 2> /dev/null ; for p in $$files; do \
+	@files=s $(DISTFILES) 2> /dev/null ; for p in $$files; do \
 	  echo $$p; \
         done
-        @for subdir in $(SUBDIRS); do \
+	    @for subdir in $(SUBDIRS); do \
 	  files=d $$subdir; $(MAKE) files | grep -v "make\[[1-9]\]"; \
           for file in $$files; do \
             echo $$subdir/$$file; \
diff -Nuar clanbomber-1.05.orig//clanbomber/wavs/Makefile.am clanbomber-1.05/clanbomber/wavs/Makefile.am
--- clanbomber-1.05.orig//clanbomber/wavs/Makefile.am	2011-02-02 15:03:47.505000015 +0200
+++ clanbomber-1.05/clanbomber/wavs/Makefile.am	2011-02-02 15:05:13.141000092 +0200
@@ -3,10 +3,10 @@
 .PHONY: files
 
 files:
-        @files=s $(DISTFILES) 2> /dev/null ; for p in $$files; do \
+	@files=s $(DISTFILES) 2> /dev/null ; for p in $$files; do \
 	  echo $$p; \
         done
-        @for subdir in $(SUBDIRS); do \
+	    @for subdir in $(SUBDIRS); do \
 	  files=d $$subdir; $(MAKE) files | grep -v "make\[[1-9]\]"; \
           for file in $$files; do \
             echo $$subdir/$$file; \
