Index: data/gaupol.desktop.in
===================================================================
--- data.orig/gaupol.desktop.in
+++ data/gaupol.desktop.in
@@ -1,7 +1,10 @@
 [Desktop Entry]
 _Name=Gaupol
+_Name[tr]=Gaupol
 _GenericName=Subtitle Editor
+_GenericName[tr]=Altyazı Düzenleyici
 _Comment=Edit subtitle files
+_Comment[tr]=Altyazı dosyalarını düzenleme aracı
 Exec=gaupol %F
 Terminal=false
 StartupNotify=true
