## 40_perl-path_examples.dpatch by  <fh@debian.org>
@DPATCH@
diff -urNad libnet-dns-perl~/contrib/check_zone libnet-dns-perl/contrib/check_zone
--- libnet-dns-perl~/contrib/check_zone	2007-07-01 17:25:20.000000000 +0200
+++ libnet-dns-perl/contrib/check_zone	2007-07-01 17:55:04.436946249 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # $Id: check_zone 638 2007-05-15 18:59:26Z olaf $
 
 =head1 NAME
diff -urNad libnet-dns-perl~/contrib/loc2earth.fcgi libnet-dns-perl/contrib/loc2earth.fcgi
--- libnet-dns-perl~/contrib/loc2earth.fcgi	2007-07-01 17:25:20.000000000 +0200
+++ libnet-dns-perl/contrib/loc2earth.fcgi	2007-07-01 17:25:41.733892017 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -T
+#!/usr/bin/perl -T
 
 # loc2earth.cgi - generates a redirect to Earth Viewer based on LOC record
 # [ see <URL: http://www.kei.com/homepages/ckd/dns-loc/ > or RFC 1876 ]
diff -urNad libnet-dns-perl~/demo/axfr libnet-dns-perl/demo/axfr
--- libnet-dns-perl~/demo/axfr	2007-07-01 17:25:20.000000000 +0200
+++ libnet-dns-perl/demo/axfr	2007-07-01 17:25:41.733892017 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # $Id: axfr 264 2005-04-06 09:16:15Z olaf $
 
 use strict;
diff -urNad libnet-dns-perl~/demo/check_soa libnet-dns-perl/demo/check_soa
--- libnet-dns-perl~/demo/check_soa	2007-07-01 17:25:20.000000000 +0200
+++ libnet-dns-perl/demo/check_soa	2007-07-01 17:25:41.733892017 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # $Id: check_soa 264 2005-04-06 09:16:15Z olaf $
 
 =head1 NAME
diff -urNad libnet-dns-perl~/demo/check_zone libnet-dns-perl/demo/check_zone
--- libnet-dns-perl~/demo/check_zone	2007-07-01 17:25:20.000000000 +0200
+++ libnet-dns-perl/demo/check_zone	2007-07-01 17:25:41.733892017 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # $Id: check_zone 264 2005-04-06 09:16:15Z olaf $
 
 =head1 NAME
diff -urNad libnet-dns-perl~/demo/example_recurse.pl libnet-dns-perl/demo/example_recurse.pl
--- libnet-dns-perl~/demo/example_recurse.pl	2007-07-01 17:25:20.000000000 +0200
+++ libnet-dns-perl/demo/example_recurse.pl	2007-07-01 17:25:41.733892017 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 
 # Example usage for Net::DNS::Resolver::Recurse
 # Performs recursion for a query.
diff -urNad libnet-dns-perl~/demo/mresolv libnet-dns-perl/demo/mresolv
--- libnet-dns-perl~/demo/mresolv	2007-07-01 17:25:20.000000000 +0200
+++ libnet-dns-perl/demo/mresolv	2007-07-01 17:25:41.733892017 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # $Id: mresolv 264 2005-04-06 09:16:15Z olaf $
 
 =head1 NAME
diff -urNad libnet-dns-perl~/demo/mx libnet-dns-perl/demo/mx
--- libnet-dns-perl~/demo/mx	2007-07-01 17:25:20.000000000 +0200
+++ libnet-dns-perl/demo/mx	2007-07-01 17:25:41.733892017 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # $Id: mx 264 2005-04-06 09:16:15Z olaf $
 
 =head1 NAME
diff -urNad libnet-dns-perl~/demo/perldig libnet-dns-perl/demo/perldig
--- libnet-dns-perl~/demo/perldig	2007-07-01 17:25:20.000000000 +0200
+++ libnet-dns-perl/demo/perldig	2007-07-01 17:25:41.733892017 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # $Id: perldig 264 2005-04-06 09:16:15Z olaf $
 
 =head1 NAME
diff -urNad libnet-dns-perl~/demo/trace_dns.pl libnet-dns-perl/demo/trace_dns.pl
--- libnet-dns-perl~/demo/trace_dns.pl	2007-07-01 17:25:20.000000000 +0200
+++ libnet-dns-perl/demo/trace_dns.pl	2007-07-01 17:25:41.733892017 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 
 use strict;
 use warnings;
