diff -p -up live/BasicUsageEnvironment/Makefile.tail.pic live/BasicUsageEnvironment/Makefile.tail
--- live/BasicUsageEnvironment/Makefile.tail.pic	2009-01-26 20:18:41.000000000 +0100
+++ live/BasicUsageEnvironment/Makefile.tail	2009-02-09 10:52:56.000000000 +0100
@@ -13,10 +13,10 @@ libBasicUsageEnvironment.$(LIB_SUFFIX): 
 		$(OBJS)
 
 .$(C).$(OBJ):
-	$(C_COMPILER) -c $(C_FLAGS) $<       
+	$(C_COMPILER) -c $(C_FLAGS) $(PIC_FLAGS) $<       
 
 .$(CPP).$(OBJ):
-	$(CPLUSPLUS_COMPILER) -c $(CPLUSPLUS_FLAGS) $<
+	$(CPLUSPLUS_COMPILER) -c $(CPLUSPLUS_FLAGS) $(PIC_FLAGS) $<
 
 BasicUsageEnvironment0.$(CPP):	include/BasicUsageEnvironment0.hh
 include/BasicUsageEnvironment0.hh:	include/BasicUsageEnvironment_version.hh include/DelayQueue.hh
diff -p -up live/config.linux.pic live/config.linux
--- live/config.linux.pic	2009-02-09 10:52:56.000000000 +0100
+++ live/config.linux	2009-02-09 10:52:56.000000000 +0100
@@ -2,6 +2,7 @@ COMPILE_OPTS =		$(INCLUDES) -I. $(RPM_OP
 C =			c
 C_COMPILER =		cc
 C_FLAGS =		$(COMPILE_OPTS)
+PIC_FLAGS =		-fPIC
 CPP =			cpp
 CPLUSPLUS_COMPILER =	c++
 CPLUSPLUS_FLAGS =	$(COMPILE_OPTS) -Wall -DBSD=1
diff -p -up live/groupsock/Makefile.tail.pic live/groupsock/Makefile.tail
--- live/groupsock/Makefile.tail.pic	2009-01-26 20:18:41.000000000 +0100
+++ live/groupsock/Makefile.tail	2009-02-09 10:52:56.000000000 +0100
@@ -4,10 +4,10 @@ ALL = libgroupsock.$(LIB_SUFFIX)
 all:	$(ALL)
 
 .$(C).$(OBJ):
-	$(C_COMPILER) -c $(C_FLAGS) $<       
+	$(C_COMPILER) -c $(C_FLAGS) $(PIC_FLAGS) $<       
 
 .$(CPP).$(OBJ):
-	$(CPLUSPLUS_COMPILER) -c $(CPLUSPLUS_FLAGS) $<
+	$(CPLUSPLUS_COMPILER) -c $(CPLUSPLUS_FLAGS) $(PIC_FLAGS) $<
 
 GROUPSOCK_LIB_OBJS = GroupsockHelper.$(OBJ) GroupEId.$(OBJ) inet.$(OBJ) Groupsock.$(OBJ) NetInterface.$(OBJ) NetAddress.$(OBJ) IOHandlers.$(OBJ)
 
diff -p -up live/liveMedia/Makefile.tail.pic live/liveMedia/Makefile.tail
--- live/liveMedia/Makefile.tail.pic	2009-01-26 20:18:41.000000000 +0100
+++ live/liveMedia/Makefile.tail	2009-02-09 10:53:51.000000000 +0100
@@ -5,10 +5,10 @@ ALL = $(LIVEMEDIA_LIB)
 all:	$(ALL)
 
 .$(C).$(OBJ):
-	$(C_COMPILER) -c $(C_FLAGS) $<       
+	$(C_COMPILER) -c $(C_FLAGS) $(PIC_FLAGS) $<       
 
 .$(CPP).$(OBJ):
-	$(CPLUSPLUS_COMPILER) -c $(CPLUSPLUS_FLAGS) $<
+	$(CPLUSPLUS_COMPILER) -c $(CPLUSPLUS_FLAGS) $(PIC_FLAGS) $<
 
 MP3_SOURCE_OBJS = MP3FileSource.$(OBJ) MP3HTTPSource.$(OBJ) MP3Transcoder.$(OBJ) MP3ADU.$(OBJ) MP3ADUdescriptor.$(OBJ) MP3ADUinterleaving.$(OBJ) MP3ADUTranscoder.$(OBJ) MP3StreamState.$(OBJ) MP3Internals.$(OBJ) MP3InternalsHuffman.$(OBJ) MP3InternalsHuffmanTable.$(OBJ) MP3ADURTPSource.$(OBJ)
 MPEG_SOURCE_OBJS = MPEG1or2Demux.$(OBJ) MPEG1or2DemuxedElementaryStream.$(OBJ) MPEGVideoStreamFramer.$(OBJ) MPEG1or2VideoStreamFramer.$(OBJ) MPEG1or2VideoStreamDiscreteFramer.$(OBJ) MPEG4VideoStreamFramer.$(OBJ) MPEG4VideoStreamDiscreteFramer.$(OBJ) H264VideoStreamFramer.$(OBJ) MPEGVideoStreamParser.$(OBJ) MPEG1or2AudioStreamFramer.$(OBJ) MPEG1or2AudioRTPSource.$(OBJ) MPEG4LATMAudioRTPSource.$(OBJ) MPEG4ESVideoRTPSource.$(OBJ) MPEG4GenericRTPSource.$(OBJ) $(MP3_SOURCE_OBJS) MPEG1or2VideoRTPSource.$(OBJ) MPEG2TransportStreamMultiplexor.$(OBJ) MPEG2TransportStreamFromPESSource.$(OBJ) MPEG2TransportStreamFromESSource.$(OBJ) MPEG2TransportStreamFramer.$(OBJ) ADTSAudioFileSource.$(OBJ)
diff -p -up live/UsageEnvironment/Makefile.tail.pic live/UsageEnvironment/Makefile.tail
--- live/UsageEnvironment/Makefile.tail.pic	2009-01-26 20:18:41.000000000 +0100
+++ live/UsageEnvironment/Makefile.tail	2009-02-09 10:52:56.000000000 +0100
@@ -10,10 +10,10 @@ $(USAGE_ENVIRONMENT_LIB): $(OBJS)
 	$(LIBRARY_LINK)$@ $(LIBRARY_LINK_OPTS) $(OBJS)
 
 .$(C).$(OBJ):
-	$(C_COMPILER) -c $(C_FLAGS) $<       
+	$(C_COMPILER) -c $(C_FLAGS) $(PIC_FLAGS) $<       
 
 .$(CPP).$(OBJ):
-	$(CPLUSPLUS_COMPILER) -c $(CPLUSPLUS_FLAGS) $<
+	$(CPLUSPLUS_COMPILER) -c $(CPLUSPLUS_FLAGS) $(PIC_FLAGS) $<
 
 UsageEnvironment.$(CPP):	include/UsageEnvironment.hh
 include/UsageEnvironment.hh:	include/UsageEnvironment_version.hh include/strDup.hh
