diff -up amarok-2.6.90/src/core-impl/collections/ipodcollection/CMakeLists.txt.dso amarok-2.6.90/src/core-impl/collections/ipodcollection/CMakeLists.txt
--- amarok-2.6.90/src/core-impl/collections/ipodcollection/CMakeLists.txt.dso	2012-12-13 06:54:53.000000000 -0600
+++ amarok-2.6.90/src/core-impl/collections/ipodcollection/CMakeLists.txt	2012-12-14 11:16:15.201095371 -0600
@@ -61,6 +61,7 @@ if(IPOD_FOUND AND WITH_IPOD)
     ${QT_QTGUI_LIBRARY}
     ${GLIB2_LIBRARIES}
     ${GDKPIXBUF_LIBRARY}
+    ${GOBJECT_LIBRARIES}
     ${IPOD_LIBRARIES}
   )
 
