--- dmraid/lib/Makefile.in	2006-07-30 22:48:53.000000000 +0000
+++ dmraid.as-needed/lib/Makefile.in	2006-07-30 22:52:10.000000000 +0000
@@ -53,6 +53,24 @@
   endif
 endif
 
+ifeq ("@KLIBC@", "yes")
+	DMRAIDLIBS += -ldevmapper_klibc
+else
+	ifeq ("@DIETLIBC@", "yes")
+		DMRAIDLIBS += -ldevmapper_dietc
+	else
+		DMRAIDLIBS += -ldevmapper -lz
+	endif
+
+#	ifeq ("@LIBSELINUX@", "yes")
+#		DMRAIDLIBS += -lselinux
+#	endif
+#
+#	ifeq ("@LIBSEPOL@", "yes")
+#		DMRAIDLIBS += -lsepol
+#	endif
+endif
+
 all:
 	@echo "$(TARGETS) $(INSTALL_TARGETS)"
 
--- dmraid/make.tmpl.in	2006-07-30 22:55:01.000000000 +0000
+++ dmraid.as-needed/make.tmpl.in	2006-07-30 22:57:38.000000000 +0000
@@ -129,7 +129,7 @@
 
 %.so: $(OBJECTS)
 	$(CC) -shared -Wl,-soname,$(notdir $@).$(DMRAID_LIB_VERSION) \
-	-Wl,--version-script,.export.sym $(OBJECTS) -o $@
+	-Wl,--version-script,.export.sym $(LDFLAGS) -o $@ $(OBJECTS) $(DMRAIDLIBS)
 
 $(LIB_STATIC): $(OBJECTS)
 	$(RM) $@
