diff -Naur fslint-2.40.orig/fslint.desktop fslint-2.40/fslint.desktop
--- fslint-2.40.orig/fslint.desktop	2008-08-27 02:23:42.000000000 +0300
+++ fslint-2.40/fslint.desktop	2009-08-06 21:44:52.498384351 +0300
@@ -1,8 +1,11 @@
 [Desktop Entry]
+Encoding=UTF-8
 Name=FSlint
+Name[tr]=FSlint
 Comment=File System Lint
+Comment[tr]=Dosya Sistemini Temizleyici
 Exec=fslint-gui
 Terminal=false
 Type=Application
 Icon=fslint_icon
-Categories=System;Filesystem;GTK;
+Categories=System;Utilities;Filesystem;GTK;
