--- src/data/qpitch.desktop.orig	2009-07-14 22:57:16.887528571 +0300
+++ src/data/qpitch.desktop	2009-07-14 22:58:19.799403119 +0300
@@ -1,10 +1,14 @@
 [Desktop Entry]
 Version=1.0
 Encoding=UTF-8
+Name[tr]=QPitch
 Name=QPitch
+GenericName[tr]=Müzik aleti akort yazılımı
+GenericName=Musical instrument tuner
+Comment[tr]=Müzik aletleri için basit bir kromatik akort yazılımı
 Comment=Simple chromatic tuner for musical instruments
 Exec=qpitch
-Icon=/usr/share/pixmaps/qpitch.xpm
+Icon=qpitch.png
 Terminal=false
 Type=Application
-Categories=AudioVideo;Audio;Music;Qt
+Categories=AudioVideo;Audio;Music;Qt;
