diff -Nur hydra-6.4-src-old//Makefile.am hydra-6.4-src/Makefile.am
--- hydra-6.4-src-old//Makefile.am	2011-07-01 23:46:01.563000024 +0300
+++ hydra-6.4-src/Makefile.am	2011-07-01 23:47:17.163000089 +0300
@@ -4,7 +4,7 @@
 
 OPTS=-I. -Wall -O2
 # -ggdb -pedantic
-LIBS=-lm
+LIBS=
 DIR=/bin
 
 SRC = hydra-vnc.c hydra-pcnfs.c hydra-rexec.c hydra-nntp.c hydra-socks5.c \
@@ -37,7 +37,7 @@
 	@echo Now type "make install"
 
 hydra:	$(OBJ)
-	$(CC) $(OPTS) $(LIBS) -o hydra $(HYDRA_LOGO) $(OBJ) $(LIBS) $(XLIBS) $(XLIBPATHS)
+	$(CC) $(OPTS) $(LDFLAGS) $(LIBS) -o hydra $(HYDRA_LOGO) $(OBJ) $(LIBS) $(XLIBS) $(XLIBPATHS)
 	@echo
 	@echo If men could get pregnant, abortion would be a sacrament
 	@echo
@@ -46,7 +46,7 @@
 	-cd hydra-gtk && sh ./make_xhydra.sh
 
 pw-inspector: pw-inspector.c
-	-$(CC) $(OPTS) -o pw-inspector $(PWI_LOGO) pw-inspector.c
+	-$(CC) $(OPTS) $(LDFLAGS) -o pw-inspector $(PWI_LOGO) pw-inspector.c
 
 .c.o:	
 	$(CC) $(OPTS) -c $< $(XDEFINES) $(XIPATHS)
