--- ncompress-4.2.4/compress42.c	1992-10-28 06:10:53.000000000 -0500
+++ ../compress42.c	2004-04-13 16:57:11.325373656 -0400
@@ -171,9 +171,9 @@
 #ifndef NOFUNCDEF
 	extern	void	*malloc	LARGS((int));
 	extern	void	free	LARGS((void *));
-#ifndef _IBMR2
+/* #ifndef _IBMR2
 	extern	int		open	LARGS((char const *,int,...));
-#endif
+#endif */
 	extern	int		close	LARGS((int));
 	extern	int		read	LARGS((int,void *,int));
 	extern	int		write	LARGS((int,void const *,int));
@@ -184,7 +184,7 @@
 	extern	char	*strcpy	LARGS((char *,char const *));
 	extern	char	*strcat	LARGS((char *,char const *));
 	extern	int		strcmp	LARGS((char const *,char const *));
-	extern	unsigned strlen	LARGS((char const *));
+/*	extern	unsigned strlen	LARGS((char const *)); */
 	extern	void	*memset	LARGS((void *,char,unsigned int));
 	extern	void	*memcpy	LARGS((void *,void const *,unsigned int));
 	extern	int		atoi	LARGS((char const *));
