Index: editje-0.1_49588/data/editje.desktop
===================================================================
--- editje-0.1_49588.orig/data/editje.desktop
+++ editje-0.1_49588/data/editje.desktop
@@ -1,6 +1,9 @@
 [Desktop Entry]
 Type=Application
 Name=Editje
+GenericName=Edje Editor
+GenericName[tr]=Edje Dosyası Düzenleyici
 Comment=Editje Edje Editor
 Comment[fr]=Éditeur de fichiers edj
+Comment[tr]=Edje Dosyası Düzenleyici
 Comment[it]=Editor per i file di Edje
