diff -up ./data/editje.desktop.orig ./data/editje.desktop
--- ./data/editje.desktop.orig	2011-10-13 00:13:17.131486512 +0200
+++ ./data/editje.desktop	2011-10-13 00:13:22.211532776 +0200
@@ -8,5 +8,5 @@ Comment[pt]=Editor de ficheiros Edje
 Icon=editje
 Exec=editje-bin %F
 Terminal=false
-Categories=Development;
+Categories=Development;GUIDesigner;
 MimeType=text/edje;application/edje;
+GenericName=Edje Editor
+GenericName[tr]=Edje Dosyası Düzenleyici
 Comment=Editje Edje Editor
 Comment[fr]=Éditeur de fichiers edj
+Comment[tr]=Edje Dosyası Düzenleyici
 Comment[it]=Editor per i file di Edje
