From 5a4feefe61733d7b650d6f6e1995a6d1555530d3 Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Wed, 26 Jun 2013 18:00:43 +0200
Subject: [PATCH] make the curl tool link SSL libraries also used by src/tool_metalink.c

---
 configure    |   10 +++-------
 configure.ac |   18 +++++++++++-------
 2 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/configure b/configure
index a466175..cb63075 100755
--- a/configure
+++ b/configure
@@ -15485,13 +15485,9 @@ fi
 
 
 
-# Determine whether all dependent libraries must be specified when linking
-if test "X$enable_shared" = "Xyes" -a "X$link_all_deplibs" = "Xno"
-then
-    REQUIRE_LIB_DEPS=no
-else
-    REQUIRE_LIB_DEPS=yes
-fi
+REQUIRE_LIB_DEPS=yes
+
+
 
  if test x$REQUIRE_LIB_DEPS = xyes; then
   USE_EXPLICIT_LIB_DEPS_TRUE=
diff --git a/configure.ac b/configure.ac
index 49309e6..90c56b5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -243,13 +243,17 @@ fi
 AC_SUBST([CPPFLAG_CURL_STATICLIB])
 
 
-# Determine whether all dependent libraries must be specified when linking
-if test "X$enable_shared" = "Xyes" -a "X$link_all_deplibs" = "Xno"
-then
-    REQUIRE_LIB_DEPS=no
-else
-    REQUIRE_LIB_DEPS=yes
-fi
+dnl make the curl tool link SSL libraries also used by src/tool_metalink.c
+REQUIRE_LIB_DEPS=yes
+
+dnl # Determine whether all dependent libraries must be specified when linking
+dnl if test "X$enable_shared" = "Xyes" -a "X$link_all_deplibs" = "Xno"
+dnl then
+dnl     REQUIRE_LIB_DEPS=no
+dnl else
+dnl     REQUIRE_LIB_DEPS=yes
+dnl fi
+
 AC_SUBST(REQUIRE_LIB_DEPS)
 AM_CONDITIONAL(USE_EXPLICIT_LIB_DEPS, test x$REQUIRE_LIB_DEPS = xyes)
 
-- 
1.7.1

