diff -Nuar Firebird-2.1.0.17798-0.orig/extern/editline/src/makelist Firebird-2.1.0.17798-0/extern/editline/src/makelist
--- Firebird-2.1.0.17798-0.orig/extern/editline/src/makelist	2007-04-09 15:57:41.000000000 +0300
+++ Firebird-2.1.0.17798-0/extern/editline/src/makelist	2008-07-12 14:02:29.000000000 +0300
@@ -140,7 +140,7 @@
 #
 -fh)
     cat $FILES | $AWK '/el_action_t/ { print $3 }' | \
-    sort | tr '[:lower:]' '[:upper:]' | $AWK '
+    sort | tr abcdefghijklmnopqrstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ | $AWK '
 	BEGIN {
 	    printf("/* Automatically generated file, do not edit */\n");
 	    printf("#ifndef _h_fcns_c\n#define _h_fcns_c\n");
