--- src/remote/inet.cpp.old	2006-12-12 23:48:13.000000000 +0200
+++ src/remote/inet.cpp	2006-12-12 23:49:05.000000000 +0200
@@ -134,7 +134,7 @@
 #error "vms implementation must be completed"
 #else // VMS
 const char* PROXY_FILE	= "/etc/gds_proxy";
-const char* HOSTS_FILE	= "/etc/hosts.equiv";
+const char* HOSTS_FILE	= "/etc/firebird/hosts.equiv";
 #ifdef MULTI_THREAD
 const USHORT MAX_PTYPE	= ptype_batch_send;
 #else
