From 1b7065f25457fea9f9ee7100437b12815b92a38c Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Wed, 11 May 2011 16:46:57 +0200
Subject: [PATCH 4/4] findutils-4.5.7-warnings.patch

---
 find/pred.c   |    6 +++---
 xargs/xargs.c |    5 +++--
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/find/pred.c b/find/pred.c
index 3acfe9f..ee3c245 100644
--- a/find/pred.c
+++ b/find/pred.c
@@ -561,7 +561,7 @@ impl_pred_exec (const char *pathname,
 		struct predicate *pred_ptr)
 {
   struct exec_val *execp = &pred_ptr->args.exec_vec;
-  char *target;
+  const char *target;
   bool result;
   const bool local = is_exec_in_local_dir (pred_ptr->pred_func);
   char *prefix;
@@ -660,7 +660,7 @@ impl_pred_exec (const char *pathname,
   if (target != pathname)
     {
       assert (local);
-      free (target);
+      free ((char *) target);
     }
   return result;
 }
@@ -2106,7 +2106,7 @@ launch (struct buildcmd_control *ctl, void *usercontext, int argc, char **argv)
 	    }
 	}
 
-      if (bc_args_exceed_testing_limit (argv))
+      if (bc_args_exceed_testing_limit ((const char **) argv))
 	errno = E2BIG;
       else
 	execvp (argv[0], argv);
diff --git a/xargs/xargs.c b/xargs/xargs.c
index 3cc1832..35f6822 100644
--- a/xargs/xargs.c
+++ b/xargs/xargs.c
@@ -1132,7 +1132,7 @@ xargs_do_exec (struct buildcmd_control *ctl, void *usercontext, int argc, char *
 
 	    prep_child_for_exec ();
 
-	    if (bc_args_exceed_testing_limit (argv))
+	    if (bc_args_exceed_testing_limit ((const char **) argv))
 	      errno = E2BIG;
 	    else
 	      execvp (argv[0], argv);
@@ -1150,7 +1150,8 @@ xargs_do_exec (struct buildcmd_control *ctl, void *usercontext, int argc, char *
 		 * utility if we run it, for POSIX compliance on the
 		 * handling of exit values.
 		 */
-		write (fd[1], &errno, sizeof (int));
+		int sink = write (fd[1], &errno, sizeof (int));
+		(void) sink;
 	      }
 
 	    close (fd[1]);
-- 
1.7.4.4

