Index: m4/c-stack.m4
===================================================================
--- m4/c-stack.m4.orig	2007-09-24 11:55:43.000000000 +0200
+++ m4/c-stack.m4	2010-02-13 14:56:24.000000000 +0100
@@ -12,6 +12,8 @@ AC_DEFUN([AC_SYS_XSI_STACK_OVERFLOW_HEUR
    AC_REQUIRE([AC_FUNC_ALLOCA])
    AC_CHECK_FUNCS(setrlimit)
 
+   saved_CFLAGS=$CFLAGS
+   CFLAGS=-O0
    AC_CACHE_CHECK([for working C stack overflow detection],
      ac_cv_sys_xsi_stack_overflow_heuristic,
      [AC_TRY_RUN(
@@ -113,6 +115,7 @@ AC_DEFUN([AC_SYS_XSI_STACK_OVERFLOW_HEUR
 	[ac_cv_sys_xsi_stack_overflow_heuristic=yes],
 	[ac_cv_sys_xsi_stack_overflow_heuristic=no],
 	[ac_cv_sys_xsi_stack_overflow_heuristic=cross-compiling])])
+   CFLAGS=$saved_CFLAGS
 
    if test $ac_cv_sys_xsi_stack_overflow_heuristic = yes; then
      AC_DEFINE(HAVE_XSI_STACK_OVERFLOW_HEURISTIC, 1,
