--- diffutils-2.8.4/src/sdiff.c.orig	2002-06-17 00:55:42.000000000 -0500
+++ diffutils-2.8.4/src/sdiff.c	2003-01-08 03:04:02.000000000 -0600
@@ -692,7 +692,7 @@
       ck_fclose (out);
 
       {
-	int wstatus;
+	int wstatus = 0;
 	int werrno = 0;
 
 #if ! (HAVE_WORKING_FORK || HAVE_WORKING_VFORK)
@@ -701,11 +701,6 @@
 	  werrno = errno;
 #else
 	ck_fclose (diffout);
-	while (waitpid (diffpid, &wstatus, 0) < 0)
-	  if (errno == EINTR)
-	    checksigs ();
-	  else
-	    perror_fatal ("waitpid");
 	diffpid = 0;
 #endif
 
