diff -up celestia-1.6.1/src/celestia/imagecapture.cpp.zlib celestia-1.6.1/src/celestia/imagecapture.cpp
--- celestia-1.6.1/src/celestia/imagecapture.cpp.zlib	2012-03-22 13:02:36.968386333 -0400
+++ celestia-1.6.1/src/celestia/imagecapture.cpp	2012-03-22 13:02:43.201336755 -0400
@@ -31,6 +31,8 @@ extern "C" {
 #include "png.h"
 #endif
 
+#include <zlib.h>
+
 // Define png_jmpbuf() in case we are using a pre-1.0.6 version of libpng
 #ifndef png_jmpbuf
 #define png_jmpbuf(png_ptr) png_ptr->jmpbuf