Index: celestia-1.6.0/src/celestia/gtk/data/celestia.desktop
===================================================================
--- celestia-1.6.0.orig/src/celestia/gtk/data/celestia.desktop
+++ celestia-1.6.0/src/celestia/gtk/data/celestia.desktop
@@ -1,12 +1,15 @@
 [Desktop Entry]
 Encoding=UTF-8
-Version=1.4.0
+Version=1.0
 Type=Application
 Exec=celestia
 Icon=celestia
 X-GNOME-DocPath=
 Terminal=false
 Name=Celestia
-Comment=Space Simulator
+Comment=3D space simulator to explore the universe
+Comment[tr]=Evreni keşfetmek için 3B uzay simülatörü
 StartupNotify=true
-Categories=Astronomy;Science;Application;
+Categories=Education;Astronomy;Science;Application;
+GenericName=Space Simulator
+GenericName[tr]=Uzay Simulatörü
