diff -up GD-2.44/Makefile.PL.orig GD-2.44/Makefile.PL
--- GD-2.44/Makefile.PL.orig	2011-06-18 02:28:12.000000000 +0200
+++ GD-2.44/Makefile.PL	2011-06-18 02:28:50.000000000 +0200
@@ -246,7 +246,7 @@ WriteMakefile(
     'PL_FILES'  => {'bdf_scripts/bdf2gdfont.PLS' => 'bdf_scripts/bdf2gdfont.pl',
 		    'GD/Image.pm.PLS' => 'GD/Image.pm',
 		   },
-    'CCFLAGS'    => '-Wformat=0',
+    'CCFLAGS'    => "-Wformat=0 $Config{ccflags}",
     'EXE_FILES' => ['bdf_scripts/bdf2gdfont.pl'],
     'AUTHOR'    => 'Lincoln Stein (lstein@cshl.org)',
     'ABSTRACT'  => 'Interface to Gd Graphics Library',
