--- hydrogen.desktop.in.org	2006-12-04 15:27:58.000000000 +0200
+++ hydrogen.desktop.in	2006-12-04 15:42:28.000000000 +0200
@@ -1,10 +1,14 @@
 [Desktop Entry]
-Name=Hydrogen Drum Machine
-Name[de]=Hydrogen Drumsequencer
-Name[ru]=Драм-машина Hydrogen
+Name=Hydrogen
 Comment=Create drum sequences
 Comment[de]=Schlagzeugsequenzen erstellen
 Comment[ru]=Создание партий ударных инструментов
+GenericName=Drum Machine
+GenericName[de]=Drum-Maschine
+GenericName[it]=drum machine
+GenericName[nl]=drum machine
+GenericName[ru]=Драм-машина
+GenericName[tr]=Davul Seti
 
 Encoding=UTF-8
 Version=1.0
