Index: hydrogen-0.9.4.2/Sconstruct
===================================================================
--- hydrogen-0.9.4.2.orig/Sconstruct
+++ hydrogen-0.9.4.2/Sconstruct
@@ -265,7 +265,6 @@ def get_hydrogen_gui( lib_hydrogen , opt
 	if str(env['portaudio']) == "1": env.Append( LIBS = [ "portaudio" ] )
 	if str(env['portmidi']) == "1":
 		env.Append( LIBS = [ "portmidi" ] )
-		env.Append( LIBS = [ "porttime" ] )
 	if str(env['lash']) == "1":
 	        env.ParseConfig('pkg-config --cflags --libs lash-1.0')
 
