diff -Nur ppracer-0.5alpha-old//configure.in ppracer-0.5alpha/configure.in
--- ppracer-0.5alpha-old//configure.in	2010-12-28 15:52:49.160000026 +0200
+++ ppracer-0.5alpha/configure.in	2010-12-28 15:53:00.946000104 +0200
@@ -523,7 +523,7 @@
 
 
 
-AC_OUTPUT(Makefile src/Makefile src/squirrel/Makefile src/sqstdlib/Makefile src/ppogl/Makefile data/Makefile data/config.nut)
+AC_OUTPUT(Makefile src/Makefile src/ppogl/Makefile data/Makefile data/config.nut)
 
 echo 
 echo PlanetPenguin Racer $VERSION
diff -Nur ppracer-0.5alpha-old//src/Makefile.am ppracer-0.5alpha/src/Makefile.am
--- ppracer-0.5alpha-old//src/Makefile.am	2010-12-28 15:52:49.155000026 +0200
+++ ppracer-0.5alpha/src/Makefile.am	2010-12-28 15:54:28.576000246 +0200
@@ -1,14 +1,16 @@
 
-SUBDIRS = squirrel sqstdlib ppogl
+SUBDIRS = ppogl
 
 AM_CPPFLAGS = -DPP_DATADIR=\"$(ppdatadir)\"
 
 bin_PROGRAMS = ppracer
 
+
+INCLUDES = -I/usr/include/squirrel
 LDADD = \
 		ppogl/libppogl.a	\
-		sqstdlib/libsqstdlib.a	\
-		squirrel/libsquirrel.a	
+		-lsqstdlib	\
+		-lsquirrel
 
 
 ppracer_SOURCES =				\
diff -Nur ppracer-0.5alpha-old//src/ppogl/Makefile.am ppracer-0.5alpha/src/ppogl/Makefile.am
--- ppracer-0.5alpha-old//src/ppogl/Makefile.am	2010-12-28 15:56:00.962000098 +0200
+++ ppracer-0.5alpha/src/ppogl/Makefile.am	2010-12-28 15:56:56.096000030 +0200
@@ -129,8 +129,7 @@
 	font/FTSize.cpp	
 
 INCLUDES = \
-	-I$(top_srcdir)/src/squirrel/ \
-	-I$(top_srcdir)/src/sqstdlib/
+	-I/usr/include/squirrel/
 
 EXTRA_DIST = \
 	font/COPYING.txt \
