diff -Nur ppracer-0.5alpha-old//src/ppogl/ppogl_script.cpp ppracer-0.5alpha/src/ppogl/ppogl_script.cpp
--- ppracer-0.5alpha-old//src/ppogl/ppogl_script.cpp	2010-12-28 15:58:33.702000094 +0200
+++ ppracer-0.5alpha/src/ppogl/ppogl_script.cpp	2010-12-28 16:02:33.818000029 +0200
@@ -253,7 +253,7 @@
 	sq_rawset(m_vm,-3);	
 }
 
-int
+SQInteger
 Script::defaultCb(HSQUIRRELVM vm)
 {
 	Script *script;
diff -Nur ppracer-0.5alpha-old//src/ppogl/ppogl_script.h ppracer-0.5alpha/src/ppogl/ppogl_script.h
--- ppracer-0.5alpha-old//src/ppogl/ppogl_script.h	2010-12-28 15:58:33.691000312 +0200
+++ ppracer-0.5alpha/src/ppogl/ppogl_script.h	2010-12-28 16:02:31.714999998 +0200
@@ -27,11 +27,7 @@
 #include "base/vec4f.h"
 
 extern "C"{
-// this is copied from squirrel.h so we don't need to include it here
-typedef struct SQVM* HSQUIRRELVM;
-typedef int SQInteger;
-typedef SQInteger SQRESULT;
-typedef SQInteger (*SQFUNCTION)(HSQUIRRELVM);
+#include <squirrel.h>
 }
 
 namespace ppogl{
@@ -63,7 +59,7 @@
 
 	void registerLib(const std::string& name, const Script::Lib *callbacks);
 
-	static int defaultCb(HSQUIRRELVM vm);
+	static SQInteger defaultCb(HSQUIRRELVM vm);
 
 
 	bool isKeyInTable(const std::string& key, int index=-1);
