diff -Nur skey-1.1.5-old//skeyprune.pl skey-1.1.5/skeyprune.pl
--- skey-1.1.5-old//skeyprune.pl	2010-09-12 15:03:35.695999732 +0300
+++ skey-1.1.5/skeyprune.pl	2010-09-12 15:06:03.817000568 +0300
@@ -14,7 +14,7 @@
 die "Usage: $0 [days]\n" if $#ARGC > 0;
 
 # Pathnames
-$keyfile = '/etc/skeykeys';
+$keyfile = '/etc/skey/skeykeys';
 $temp = "$keyfile.tmp$$";
 
 # Quick mapping of month name -> number
