Fix for bug #538234


diff -Naur MySQL-python-1.2.3c1.orig/_mysql.c MySQL-python-1.2.3c1/_mysql.c
--- MySQL-python-1.2.3c1.orig/_mysql.c	2009-03-30 18:59:28.000000000 -0400
+++ MySQL-python-1.2.3c1/_mysql.c	2009-11-23 12:58:41.000000000 -0500
@@ -1714,7 +1714,7 @@
 {
 	unsigned long pid;
 	int r;
-	if (!PyArg_ParseTuple(args, "i:kill", &pid)) return NULL;
+	if (!PyArg_ParseTuple(args, "l:kill", &pid)) return NULL;
 	check_connection(self);
 	Py_BEGIN_ALLOW_THREADS
 	r = mysql_kill(&(self->connection), pid);
