diff -ur kde-workspace-4.10.0.orig/libs/plasmagenericshell/backgrounddialog.cpp kde-workspace-4.10.0/libs/plasmagenericshell/backgrounddialog.cpp
--- kde-workspace-4.10.0.orig/libs/plasmagenericshell/backgrounddialog.cpp	2013-01-23 23:46:44.000000000 +0200
+++ kde-workspace-4.10.0/libs/plasmagenericshell/backgrounddialog.cpp	2012-12-17 16:21:54.000000000 +0200
@@ -17,7 +17,6 @@
 #include <QFile>
 #include <QPainter>
 #include <QStandardItemModel>
-#include <QUiLoader>
 
 #include <KColorScheme>
 #include <KDebug>
@@ -28,7 +27,6 @@
 #include <Plasma/Context>
 #include <Plasma/Corona>
 #include <Plasma/FrameSvg>
-#include <Plasma/Package>
 #include <Plasma/Wallpaper>
 #include <Plasma/View>
 
@@ -175,9 +173,8 @@
 class BackgroundDialogPrivate
 {
 public:
-    BackgroundDialogPrivate(BackgroundDialog* dialog, Plasma::Containment* c, Plasma::View* v)
-     : q(dialog),
-       containmentModel(0),
+    BackgroundDialogPrivate(Plasma::Containment* c, Plasma::View* v)
+     : containmentModel(0),
        wallpaper(0),
        view(v),
        containment(c),
@@ -190,8 +187,6 @@
     {
     }
 
-    BackgroundDialog *q;
-
     Ui::BackgroundDialog backgroundDialogUi;
 
     QStandardItemModel* containmentModel;
@@ -203,36 +198,12 @@
     KPageWidgetItem *appearanceItem;
     KPageWidgetItem *mouseItem;
     bool modified;
-
-    void createConfigurationInterfaceForPackage()
-    {
-        const QString uiFile = containment.data()->package()->filePath("mainconfigui");
-        if (uiFile.isEmpty()) {
-            kWarning() << "No ui file found for containment";
-            return;
-        }
-        Plasma::ConfigLoader *configScheme = containment.data()->configScheme();
-        if (!configScheme) {
-            kWarning() << "No configuration scheme found for containment";
-            return;
-        }
-
-        QFile f(uiFile);
-        QUiLoader loader;
-        QWidget *widget = loader.load(&f);
-        if (widget) {
-            q->addPage(widget, configScheme,
-                i18n("Settings"), containment.data()->icon(), i18n("%1 Settings", containment.data()->name()));
-        } else {
-            kWarning() << "Failed to load widget from" << uiFile;
-        }
-    }
 };
 
 BackgroundDialog::BackgroundDialog(const QSize& res, Plasma::Containment *c, Plasma::View* view,
                                    QWidget* parent, const QString &id, KConfigSkeleton *s)
     : KConfigDialog(parent, id, s),
-      d(new BackgroundDialogPrivate(this, c, view))
+      d(new BackgroundDialogPrivate(c, view))
 {
     setWindowIcon(KIcon("preferences-desktop-wallpaper"));
     setCaption(i18n("Desktop Settings"));
@@ -273,11 +244,7 @@
     d->mouseItem = addPage(m, i18n("Mouse Actions"), "input-mouse");
 
     if (d->containment && d->containment.data()->hasConfigurationInterface()) {
-        if (d->containment.data()->package()) {
-            d->createConfigurationInterfaceForPackage();
-        } else {
-            d->containment.data()->createConfigurationInterface(this);
-        }
+        d->containment.data()->createConfigurationInterface(this);
         connect(this, SIGNAL(applyClicked()), d->containment.data(), SLOT(configDialogFinished()));
         connect(this, SIGNAL(okClicked()), d->containment.data(), SLOT(configDialogFinished()));
     }
@@ -564,11 +531,7 @@
 
             //add the new containment's config
             if (d->containment.data()->hasConfigurationInterface()) {
-                if (d->containment.data()->package()) {
-                    d->createConfigurationInterfaceForPackage();
-                } else {
-                    d->containment.data()->createConfigurationInterface(this);
-                }
+                d->containment.data()->createConfigurationInterface(this);
                 connect(this, SIGNAL(applyClicked()), d->containment.data(), SLOT(configDialogFinished()));
                 connect(this, SIGNAL(okClicked()), d->containment.data(), SLOT(configDialogFinished()));
             }
diff -ur kde-workspace-4.10.0.orig/libs/plasmagenericshell/CMakeLists.txt kde-workspace-4.10.0/libs/plasmagenericshell/CMakeLists.txt
--- kde-workspace-4.10.0.orig/libs/plasmagenericshell/CMakeLists.txt	2013-01-23 23:46:44.000000000 +0200
+++ kde-workspace-4.10.0/libs/plasmagenericshell/CMakeLists.txt	2012-12-17 16:21:54.000000000 +0200
@@ -34,7 +34,7 @@
 
 kde4_add_library(plasmagenericshell SHARED ${plasmagenericshell_LIB_SRCS})
 
-target_link_libraries(plasmagenericshell ${KDE4_PLASMA_LIBS} ${KDE4_KDEUI_LIBS} ${KDE4_KFILE_LIBS} ${QT_QTSCRIPT_LIBRARY} ${QT_QTDECLARATIVE_LIBRARY} ${QT_QTUITOOLS_LIBRARY} kworkspace kephal)
+target_link_libraries(plasmagenericshell ${KDE4_PLASMA_LIBS} ${KDE4_KDEUI_LIBS} ${KDE4_KFILE_LIBS} ${QT_QTSCRIPT_LIBRARY} ${QT_QTDECLARATIVE_LIBRARY} kworkspace kephal)
 if(UNIX AND Q_WS_X11)
     target_link_libraries(plasmagenericshell ${X11_LIBRARIES})
 endif(UNIX AND Q_WS_X11)
diff -ur kde-workspace-4.10.0.orig/libs/plasmagenericshell/widgetsexplorer/package/metadata.desktop kde-workspace-4.10.0/libs/plasmagenericshell/widgetsexplorer/package/metadata.desktop
--- kde-workspace-4.10.0.orig/libs/plasmagenericshell/widgetsexplorer/package/metadata.desktop	2013-01-30 22:30:42.000000000 +0200
+++ kde-workspace-4.10.0/libs/plasmagenericshell/widgetsexplorer/package/metadata.desktop	2013-01-01 17:47:41.000000000 +0200
@@ -4,7 +4,6 @@
 Name=Widget explorer for the desktop
 Name[bs]=Miniaplikacija istraživač za desktop
 Name[ca]=Explorador d'estris per l'escriptori
-Name[ca@valencia]=Explorador d'estris per l'escriptori
 Name[cs]=Průzkumník widgetů plochy
 Name[da]=Gennemse widgets til desktoppen
 Name[el]=Εξερευνητής γραφικών συστατικών για την επιφάνεια εργασίας
@@ -18,14 +17,12 @@
 Name[kk]=Үстел виджеттерін шолу
 Name[km]=កម្មវិធី​រុករក​ធាតុក្រាហ្វិក​សម្រាប់​ផ្ទៃតុ
 Name[ko]=데스크톱 위젯 탐색기
-Name[mr]=डेस्कटॉप करिता विजेट एक्सप्लोरर
 Name[nb]=Elementutforsker for skrivebordet
 Name[nl]=Widget-verkenner voor het bureaublad
 Name[pa]=ਡੈਸਕਟਾਪ ਲਈ ਵਿਦਜੈੱਟ ਐਕਸਪਲੋਰਰ
 Name[pl]=Eksplorator w postaci elementu interfejsu na pulpit
 Name[pt]=Explorador de elementos para o ecrã
 Name[pt_BR]=Explorador de widgets para a área de trabalho
-Name[ro]=Explorator de controale pentru birou
 Name[ru]=Обозреватель виджетов для рабочего стола
 Name[sk]=Prehliadač widgetov pre plochu
 Name[sl]=Raziskovalnik gradnikov za namizje
