Index: uClibc-0.9.30.2/extra/gcc-uClibc/gcc-uClibc.c
===================================================================
--- uClibc-0.9.30.2.orig/extra/gcc-uClibc/gcc-uClibc.c
+++ uClibc-0.9.30.2/extra/gcc-uClibc/gcc-uClibc.c
@@ -412,6 +412,10 @@ int main(int argc, char **argv)
 					} else if (strcmp("-fpic",argv[i]) == 0) {
 						use_pic = 1;
 					} 
+					/* Filter out stack-protector */
+					else if (strcmp("-fstack-protector", argv[i]) == 0) {
+						argv[i] = '\0';
+					}
 #ifdef __UCLIBC_PROFILING__
 					else if (strcmp("-fprofile-arcs",argv[i]) == 0) {
 						profile = 1;
