Index: uClibc-0.9.31/Makefile.in
===================================================================
--- uClibc-0.9.31.orig/Makefile.in
+++ uClibc-0.9.31/Makefile.in
@@ -287,7 +287,7 @@ HEADERS_RM-$(UCLIBC_SUSV4_LEGACY)
 	### ucontext.h
 
 ifneq ($(findstring install,$(MAKECMDGOALS)),)
-$(PREFIX)$(RUNTIME_PREFIX)$(MULTILIB_DIR) $(addprefix $(PREFIX)$(DEVEL_PREFIX),include lib):
+$(PREFIX)$(RUNTIME_PREFIX)$(MULTILIB_DIR) $(addprefix $(PREFIX)$(DEVEL_PREFIX),include $(MULTILIB_DIR)):
 	$(do_mkdir)
 endif
 
