Index: uClibc-0.9.30.2/extra/gcc-uClibc/gcc-uClibc.c
===================================================================
--- uClibc-0.9.30.2.orig/extra/gcc-uClibc/gcc-uClibc.c
+++ uClibc-0.9.30.2/extra/gcc-uClibc/gcc-uClibc.c
@@ -653,6 +653,8 @@ crash_n_burn:
 			if (gcc_argument[l]) gcc_argv[i++] = gcc_argument[l];
 		}
 	}
+	/* Pass -muclibc to GCC to fix undefined sincos problems */
+	gcc_argv[i++] = "-muclibc";
 	gcc_argv[i++] = NULL;
 
 	if (verbose) {
