From 3fe01af27849c45c36c4c88eab8e24ddfeae307e Mon Sep 17 00:00:00 2001
From: Tedd Ho-Jeong An <tedd.an@intel.com>
Date: Fri, 23 Aug 2013 00:49:32 +0000
Subject: Fix filename for Intel Bluetooth firmware

This patch fixes incorrect filename described in the WHENCE

Signed-off-by: Tedd Ho-Jeong An <tedd.an@intel.com>
Signed-off-by: David Woodhouse <David.Woodhouse@intel.com>
---
diff --git a/WHENCE b/WHENCE
index ed7de43..e66fea5 100644
--- a/WHENCE
+++ b/WHENCE
@@ -2152,7 +2152,7 @@ Found also in alsa-firmware package.
 
 Driver: btusb - Bluetooth USB driver
 
-File: intel/ibt-default-37.7.bseq
+File: intel/ibt-hw-37.7.bseq
 File: intel/ibt-hw-37.7.10-fw-1.80.2.3.d.bseq
 
 Licence: Redistributable. See LICENCE.ibt_firmware for details
--
cgit v0.9.2
