From e982abf07f7986ca42b8cfdeb3374d009c18902a Mon Sep 17 00:00:00 2001
From: Emmanuel Grumbach <emmanuel.grumbach@intel.com>
Date: Mon, 24 Jun 2013 12:05:52 +0000
Subject: iwlwifi: add firmware for 7260 / 3160 devices

Add new firmware for new devices: 7260 and 3160
version:
22.0.7.0

Cc: Johannes Berg <johannes.berg@intel.com>
Signed-off-by: Emmanuel Grumbach <emmanuel.grumbach@intel.com>
Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
diff --git a/LICENCE.iwlwifi_firmware b/LICENCE.iwlwifi_firmware
index c235164..01c32a0 100644
--- a/LICENCE.iwlwifi_firmware
+++ b/LICENCE.iwlwifi_firmware
@@ -1,4 +1,4 @@
-Copyright (c) 2006-2012, Intel Corporation.
+Copyright (c) 2006-2013, Intel Corporation.
 All rights reserved.
 
 Redistribution.  Redistribution and use in binary form, without
diff --git a/WHENCE b/WHENCE
index dd0c5b7..d7fba9d 100644
--- a/WHENCE
+++ b/WHENCE
@@ -801,6 +801,12 @@ Version: 18.168.6.1
 File: iwlwifi-2000-6.ucode
 Version: 18.168.6.1
 
+File: iwlwifi-7260-7.ucode
+Version: 22.0.7.0
+
+File: iwlwifi-3160-7.ucode
+Version: 22.0.7.0
+
 Licence: Redistributable. See LICENCE.iwlwifi_firmware for details
 
 Also available from http://intellinuxwireless.org/?n=Downloads
diff --git a/iwlwifi-3160-7.ucode b/iwlwifi-3160-7.ucode
new file mode 100644
index 0000000..f6646d5
--- /dev/null
+++ b/iwlwifi-3160-7.ucode
Binary files differ
diff --git a/iwlwifi-7260-7.ucode b/iwlwifi-7260-7.ucode
new file mode 100644
index 0000000..324c7f2
--- /dev/null
+++ b/iwlwifi-7260-7.ucode
Binary files differ
--
cgit v0.9.2
