From 600caefd83a406540b2a789be6415e44c9b87add Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben@decadent.org.uk>
Date: Thu, 08 Aug 2013 20:58:35 +0000
Subject: Merge branch 'moxa' of https://github.com/lunn/linux-firmware

---
diff --git a/WHENCE b/WHENCE
index 1ee1545..ed7de43 100644
--- a/WHENCE
+++ b/WHENCE
@@ -2222,3 +2222,35 @@ Version: 2013-07-10
 License: Redistributable. See LICENSE.amd-ucode for details
 
 --------------------------------------------------------------------------
+
+Driver: mxuport - MOXA UPort USB Serial hub driver
+
+File: moxa/moxa-04e2.fw
+File: moxa/moxa-04e3.fw
+File: moxa/moxa-0582.fw
+File: moxa/moxa-05aa.fw
+File: moxa/moxa-05ab.fw
+File: moxa/moxa-064d.fw
+File: moxa/moxa-0652.fw
+File: moxa/moxa-0675.fw
+File: moxa/moxa-067a.fw
+
+License:
+The software accompanying this license statement (the “Software”)
+is the property of Moxa Inc. (the “Moxa”), and is protected by
+United States and International Copyright Laws and International
+treaty provisions. No ownership rights are granted by this
+Agreement or possession of the Software. Therefore, you must treat
+the Licensed Software like any other copyrighted material. Your
+rights and obligations in its use are described as follows:
+
+1. You may freely redistribute this software under this license.
+2. You may freely download and use this software on Moxa's device.
+3. You may not modify or attempt to reverse engineer the software, or
+   make any attempt to change or even examine the source code of the
+   software.
+4. You may not re-license or sub-license the software to any person or
+   business, using any other license.
+5. Moxa(r) is worldwide registered trademark.
+
+--------------------------------------------------------------------------
diff --git a/moxa/moxa-04e2.fw b/moxa/moxa-04e2.fw
new file mode 100644
index 0000000..61e7ea5
--- /dev/null
+++ b/moxa/moxa-04e2.fw
Binary files differ
diff --git a/moxa/moxa-04e3.fw b/moxa/moxa-04e3.fw
new file mode 100644
index 0000000..7f40650
--- /dev/null
+++ b/moxa/moxa-04e3.fw
Binary files differ
diff --git a/moxa/moxa-0582.fw b/moxa/moxa-0582.fw
new file mode 100644
index 0000000..e370bef
--- /dev/null
+++ b/moxa/moxa-0582.fw
Binary files differ
diff --git a/moxa/moxa-05aa.fw b/moxa/moxa-05aa.fw
new file mode 100644
index 0000000..9ba8374
--- /dev/null
+++ b/moxa/moxa-05aa.fw
Binary files differ
diff --git a/moxa/moxa-05ab.fw b/moxa/moxa-05ab.fw
new file mode 100644
index 0000000..79164d3
--- /dev/null
+++ b/moxa/moxa-05ab.fw
Binary files differ
diff --git a/moxa/moxa-064d.fw b/moxa/moxa-064d.fw
new file mode 100644
index 0000000..8466269
--- /dev/null
+++ b/moxa/moxa-064d.fw
Binary files differ
diff --git a/moxa/moxa-0652.fw b/moxa/moxa-0652.fw
new file mode 100644
index 0000000..89eda87
--- /dev/null
+++ b/moxa/moxa-0652.fw
Binary files differ
diff --git a/moxa/moxa-0675.fw b/moxa/moxa-0675.fw
new file mode 100644
index 0000000..e619689
--- /dev/null
+++ b/moxa/moxa-0675.fw
Binary files differ
diff --git a/moxa/moxa-067a.fw b/moxa/moxa-067a.fw
new file mode 100644
index 0000000..57cfb48
--- /dev/null
+++ b/moxa/moxa-067a.fw
Binary files differ
--
cgit v0.9.2
