From e2b52bb02560829051b2f2069d2bd056690c9458 Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexander.deucher@amd.com>
Date: Sun, 14 Jul 2013 14:32:21 +0000
Subject: radeon: add smc ucode for radeon GPUs

Add new ucode files and update WHENCE entry.

Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
diff --git a/WHENCE b/WHENCE
index 7679357..81f989f 100644
--- a/WHENCE
+++ b/WHENCE
@@ -1444,19 +1444,27 @@ Driver: radeon - ATI Radeon
 File: radeon/R600_rlc.bin
 File: radeon/R700_rlc.bin
 File: radeon/RV710_uvd.bin
+File: radeon/RV710_smc.bin
+File: radeon/RV730_smc.bin
+File: radeon/RV740_smc.bin
+File: radeon/RV770_smc.bin
 File: radeon/CEDAR_me.bin
 File: radeon/CEDAR_pfp.bin
 File: radeon/CEDAR_rlc.bin
+File: radeon/CEDAR_smc.bin
 File: radeon/CYPRESS_me.bin
 File: radeon/CYPRESS_pfp.bin
 File: radeon/CYPRESS_rlc.bin
 File: radeon/CYPRESS_uvd.bin
+File: radeon/CYPRESS_smc.bin
 File: radeon/JUNIPER_me.bin
 File: radeon/JUNIPER_pfp.bin
 File: radeon/JUNIPER_rlc.bin
+File: radeon/JUNIPER_smc.bin
 File: radeon/REDWOOD_me.bin
 File: radeon/REDWOOD_pfp.bin
 File: radeon/REDWOOD_rlc.bin
+File: radeon/REDWOOD_smc.bin
 File: radeon/PALM_me.bin
 File: radeon/PALM_pfp.bin
 File: radeon/SUMO_rlc.bin
@@ -1464,17 +1472,21 @@ File: radeon/SUMO_uvd.bin
 File: radeon/BARTS_mc.bin
 File: radeon/BARTS_me.bin
 File: radeon/BARTS_pfp.bin
+File: radeon/BARTS_smc.bin
 File: radeon/BTC_rlc.bin
 File: radeon/CAICOS_mc.bin
 File: radeon/CAICOS_me.bin
 File: radeon/CAICOS_pfp.bin
+File: radeon/CAICOS_smc.bin
 File: radeon/TURKS_mc.bin
 File: radeon/TURKS_me.bin
 File: radeon/TURKS_pfp.bin
+File: radeon/TURKS_smc.bin
 File: radeon/CAYMAN_mc.bin
 File: radeon/CAYMAN_me.bin
 File: radeon/CAYMAN_pfp.bin
 File: radeon/CAYMAN_rlc.bin
+File: radeon/CAYMAN_smc.bin
 File: radeon/SUMO_pfp.bin
 File: radeon/SUMO_me.bin
 File: radeon/SUMO2_pfp.bin
@@ -1487,27 +1499,32 @@ File: radeon/PITCAIRN_mc.bin
 File: radeon/PITCAIRN_me.bin
 File: radeon/PITCAIRN_pfp.bin
 File: radeon/PITCAIRN_rlc.bin
+File: radeon/PITCAIRN_smc.bin
 File: radeon/TAHITI_ce.bin
 File: radeon/TAHITI_mc.bin
 File: radeon/TAHITI_me.bin
 File: radeon/TAHITI_pfp.bin
 File: radeon/TAHITI_rlc.bin
 File: radeon/TAHITI_uvd.bin
+File: radeon/TAHITI_smc.bin
 File: radeon/VERDE_ce.bin
 File: radeon/VERDE_mc.bin
 File: radeon/VERDE_me.bin
 File: radeon/VERDE_pfp.bin
 File: radeon/VERDE_rlc.bin
+File: radeon/VERDE_smc.bin
 File: radeon/OLAND_ce.bin
 File: radeon/OLAND_mc.bin
 File: radeon/OLAND_me.bin
 File: radeon/OLAND_pfp.bin
 File: radeon/OLAND_rlc.bin
+File: radeon/OLAND_smc.bin
 File: radeon/HAINAN_ce.bin
 File: radeon/HAINAN_mc.bin
 File: radeon/HAINAN_me.bin
 File: radeon/HAINAN_pfp.bin
 File: radeon/HAINAN_rlc.bin
+File: radeon/HAINAN_smc.bin
 
 Licence: Redistributable. See LICENSE.radeon for details.
 
diff --git a/radeon/BARTS_smc.bin b/radeon/BARTS_smc.bin
new file mode 100644
index 0000000..a4d1e08
--- /dev/null
+++ b/radeon/BARTS_smc.bin
Binary files differ
diff --git a/radeon/CAICOS_smc.bin b/radeon/CAICOS_smc.bin
new file mode 100644
index 0000000..bc8840c
--- /dev/null
+++ b/radeon/CAICOS_smc.bin
Binary files differ
diff --git a/radeon/CAYMAN_smc.bin b/radeon/CAYMAN_smc.bin
new file mode 100644
index 0000000..5e1c2e5
--- /dev/null
+++ b/radeon/CAYMAN_smc.bin
Binary files differ
diff --git a/radeon/CEDAR_smc.bin b/radeon/CEDAR_smc.bin
new file mode 100644
index 0000000..1ad1f4e
--- /dev/null
+++ b/radeon/CEDAR_smc.bin
Binary files differ
diff --git a/radeon/CYPRESS_smc.bin b/radeon/CYPRESS_smc.bin
new file mode 100644
index 0000000..69376fa
--- /dev/null
+++ b/radeon/CYPRESS_smc.bin
Binary files differ
diff --git a/radeon/HAINAN_smc.bin b/radeon/HAINAN_smc.bin
new file mode 100644
index 0000000..6bf5b36
--- /dev/null
+++ b/radeon/HAINAN_smc.bin
Binary files differ
diff --git a/radeon/JUNIPER_smc.bin b/radeon/JUNIPER_smc.bin
new file mode 100644
index 0000000..7c2c06a
--- /dev/null
+++ b/radeon/JUNIPER_smc.bin
Binary files differ
diff --git a/radeon/OLAND_smc.bin b/radeon/OLAND_smc.bin
new file mode 100644
index 0000000..44c07f9
--- /dev/null
+++ b/radeon/OLAND_smc.bin
Binary files differ
diff --git a/radeon/PITCAIRN_smc.bin b/radeon/PITCAIRN_smc.bin
new file mode 100644
index 0000000..c918c88
--- /dev/null
+++ b/radeon/PITCAIRN_smc.bin
Binary files differ
diff --git a/radeon/REDWOOD_smc.bin b/radeon/REDWOOD_smc.bin
new file mode 100644
index 0000000..5c835de
--- /dev/null
+++ b/radeon/REDWOOD_smc.bin
Binary files differ
diff --git a/radeon/RV710_smc.bin b/radeon/RV710_smc.bin
new file mode 100644
index 0000000..5679b82
--- /dev/null
+++ b/radeon/RV710_smc.bin
Binary files differ
diff --git a/radeon/RV730_smc.bin b/radeon/RV730_smc.bin
new file mode 100644
index 0000000..4daf985
--- /dev/null
+++ b/radeon/RV730_smc.bin
Binary files differ
diff --git a/radeon/RV740_smc.bin b/radeon/RV740_smc.bin
new file mode 100644
index 0000000..8f416d9
--- /dev/null
+++ b/radeon/RV740_smc.bin
Binary files differ
diff --git a/radeon/RV770_smc.bin b/radeon/RV770_smc.bin
new file mode 100644
index 0000000..50955ca
--- /dev/null
+++ b/radeon/RV770_smc.bin
Binary files differ
diff --git a/radeon/TAHITI_smc.bin b/radeon/TAHITI_smc.bin
new file mode 100644
index 0000000..a6ef51e
--- /dev/null
+++ b/radeon/TAHITI_smc.bin
Binary files differ
diff --git a/radeon/TURKS_smc.bin b/radeon/TURKS_smc.bin
new file mode 100644
index 0000000..d644ab6
--- /dev/null
+++ b/radeon/TURKS_smc.bin
Binary files differ
diff --git a/radeon/VERDE_smc.bin b/radeon/VERDE_smc.bin
new file mode 100644
index 0000000..78a154e
--- /dev/null
+++ b/radeon/VERDE_smc.bin
Binary files differ
--
cgit v0.9.2
