From 79fb9b9c0613d74ef6b83d378abd80a75976fe5d Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexander.deucher@amd.com>
Date: Sun, 14 Jul 2013 14:36:38 +0000
Subject: radeon: add ucode for BONAIRE GPUs

Add new ucode files and update the WHENCE entry.

Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
diff --git a/WHENCE b/WHENCE
index 81f989f..3bed594 100644
--- a/WHENCE
+++ b/WHENCE
@@ -1525,6 +1525,14 @@ File: radeon/HAINAN_me.bin
 File: radeon/HAINAN_pfp.bin
 File: radeon/HAINAN_rlc.bin
 File: radeon/HAINAN_smc.bin
+File: radeon/BONAIRE_ce.bin
+File: radeon/BONAIRE_mc.bin
+File: radeon/BONAIRE_me.bin
+File: radeon/BONAIRE_mec.bin
+File: radeon/BONAIRE_pfp.bin
+File: radeon/BONAIRE_rlc.bin
+File: radeon/BONAIRE_sdma.bin
+File: radeon/BONAIRE_uvd.bin
 
 Licence: Redistributable. See LICENSE.radeon for details.
 
diff --git a/radeon/BONAIRE_ce.bin b/radeon/BONAIRE_ce.bin
new file mode 100644
index 0000000..7d40fbf
--- /dev/null
+++ b/radeon/BONAIRE_ce.bin
Binary files differ
diff --git a/radeon/BONAIRE_mc.bin b/radeon/BONAIRE_mc.bin
new file mode 100644
index 0000000..a3c7047
--- /dev/null
+++ b/radeon/BONAIRE_mc.bin
Binary files differ
diff --git a/radeon/BONAIRE_me.bin b/radeon/BONAIRE_me.bin
new file mode 100644
index 0000000..75bff89
--- /dev/null
+++ b/radeon/BONAIRE_me.bin
Binary files differ
diff --git a/radeon/BONAIRE_mec.bin b/radeon/BONAIRE_mec.bin
new file mode 100644
index 0000000..65fd6fb
--- /dev/null
+++ b/radeon/BONAIRE_mec.bin
Binary files differ
diff --git a/radeon/BONAIRE_pfp.bin b/radeon/BONAIRE_pfp.bin
new file mode 100644
index 0000000..e12a1af
--- /dev/null
+++ b/radeon/BONAIRE_pfp.bin
Binary files differ
diff --git a/radeon/BONAIRE_rlc.bin b/radeon/BONAIRE_rlc.bin
new file mode 100644
index 0000000..7e32235
--- /dev/null
+++ b/radeon/BONAIRE_rlc.bin
Binary files differ
diff --git a/radeon/BONAIRE_sdma.bin b/radeon/BONAIRE_sdma.bin
new file mode 100644
index 0000000..4766a2a
--- /dev/null
+++ b/radeon/BONAIRE_sdma.bin
Binary files differ
diff --git a/radeon/BONAIRE_uvd.bin b/radeon/BONAIRE_uvd.bin
new file mode 100644
index 0000000..0876de9
--- /dev/null
+++ b/radeon/BONAIRE_uvd.bin
Binary files differ
--
cgit v0.9.2
