diff -up notification-daemon-0.4.0/data/notification-daemon.schemas.in.position notification-daemon-0.4.0/data/notification-daemon.schemas.in
--- notification-daemon-0.4.0/data/notification-daemon.schemas.in.position	2009-08-21 23:59:46.831206308 -0400
+++ notification-daemon-0.4.0/data/notification-daemon.schemas.in	2009-08-22 00:00:15.556195107 -0400
@@ -6,7 +6,7 @@
    <applyto>/apps/notification-daemon/popup_location</applyto>
    <owner>notification-daemon</owner>
    <type>string</type>
-   <default>bottom_right</default>
+   <default>top_right</default>
    <locale name="C">
     <short>Popup location</short>
     <long>Default popup location on the workspace for stack notifications. Allowed values: &quot;top_left&quot;,&quot;top_right&quot;,&quot;bottom_left&quot; and &quot;bottom_right&quot;</long>
