--- notification-daemon-0.4.1/src/daemon/stack.c	2009-09-23 01:47:50.000000000 -0400
+++ hacked-foo/src/daemon/stack.c	2009-10-15 21:26:03.269041656 -0400
@@ -199,17 +199,21 @@
 static void
 notify_stack_shift_notifications(NotifyStack *stack,
 								 GtkWindow *nw,
-								 GSList **nw_l,
+								  GSList **nw_l,
 								 gint init_width,
 								 gint init_height,
 								 gint *nw_x,
 								 gint *nw_y)
 {
 	GdkRectangle workarea;
+        GdkRectangle monitor;
 	GSList *l;
 	gint x, y, shiftx = 0, shifty = 0, index = 1;
 
 	get_work_area(GTK_WIDGET(nw), &workarea);
+        gdk_screen_get_monitor_geometry (stack->screen, stack->monitor, &monitor);
+	gdk_rectangle_intersect (&monitor, &workarea, &workarea);
+
 	get_origin_coordinates(stack->location, &workarea, &x, &y,
 						   &shiftx, &shifty, init_width, init_height);
 
