diff -up notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c.version-check notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c
--- notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c.version-check	2008-11-23 13:20:45.000000000 +0100
+++ notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c	2008-11-23 13:21:50.000000000 +0100
@@ -614,8 +614,7 @@ theme_check_init(unsigned int major_ver,
 				 unsigned int micro_ver)
 {
 	return major_ver == 0 &&
-	       minor_ver == 3 &&
-	       micro_ver >= 2;
+	       (minor_ver > 3 || (minor_ver == 3 && micro_ver >= 2));
 }
 
 /* Sending theme info to the notification daemon */
