diff -up notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c.window-type notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c
--- notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c.window-type	2009-09-18 17:43:08.399713271 -0400
+++ notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c	2009-09-18 17:45:47.364444776 -0400
@@ -705,6 +705,9 @@ create_notification(UrlClickedCb url_cli
 	}
 
 	gtk_window_set_title(GTK_WINDOW(win), "Notification");
+	gtk_window_set_type_hint(GTK_WINDOW(win),
+				 GDK_WINDOW_TYPE_HINT_NOTIFICATION);
+
 	gtk_widget_add_events(win, GDK_BUTTON_PRESS_MASK | GDK_BUTTON_RELEASE_MASK);
 	gtk_widget_realize(win);
 //	gtk_widget_set_size_request(win, WIDTH, -1);
