diff -up notification-daemon-engine-nodoka-0.1.0/configure.ac.sexy notification-daemon-engine-nodoka-0.1.0/configure.ac
--- notification-daemon-engine-nodoka-0.1.0/configure.ac.sexy	2008-04-20 05:46:42.000000000 -0400
+++ notification-daemon-engine-nodoka-0.1.0/configure.ac	2009-07-02 00:45:44.654266038 -0400
@@ -40,12 +40,6 @@ PKG_CHECK_MODULES(GTK, gtk+-2.0 >= 2.10.
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
 
-PKG_CHECK_MODULES(LIBSEXY, libsexy,,
-                   AC_MSG_ERROR([libsexy is required to compile nodoka]))
-
-AC_SUBST(LIBSEXY_CFLAGS)
-AC_SUBST(LIBSEXY_LIBS)
-
 AC_CONFIG_HEADERS([src/config.h])
 AC_CONFIG_FILES([Makefile src/Makefile])
 
diff -up notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c.sexy notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c
--- notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c.sexy	2009-07-02 00:45:44.647264442 -0400
+++ notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c	2009-07-21 10:55:05.596047665 -0400
@@ -25,7 +25,6 @@
 #include "config.h"
 
 #include <gtk/gtk.h>
-#include <libsexy/sexy-url-label.h>
 
 /* Define basic nodoka types */
 typedef void (*ActionInvokedCb)(GtkWindow *nw, const char *key);
@@ -660,6 +659,14 @@ get_theme_info(char **theme_name,
 	*homepage = g_strdup("https://nodoka.fedorahosted.org/");
 }
 
+static gboolean
+activate_link (GtkLabel *label, const char *url, WindowData *windata)
+{
+  windata->url_clicked (windata->win, url);
+
+  return TRUE;
+
+}
 /* Create new notification */
 GtkWindow *
 create_notification(UrlClickedCb url_clicked)
@@ -805,12 +812,12 @@ create_notification(UrlClickedCb url_cli
 	gtk_widget_show(vbox);
 	gtk_box_pack_start(GTK_BOX(windata->content_hbox), vbox, TRUE, TRUE, 0);
 
-	windata->body_label = sexy_url_label_new();
+	windata->body_label = gtk_label_new(NULL);
 	gtk_box_pack_start(GTK_BOX(vbox), windata->body_label, TRUE, TRUE, 0);
 	gtk_misc_set_alignment(GTK_MISC(windata->body_label), 0, 0);
 	gtk_label_set_line_wrap(GTK_LABEL(windata->body_label), TRUE);
-	g_signal_connect_swapped(G_OBJECT(windata->body_label), "url_activated",
-							 G_CALLBACK(windata->url_clicked), win);
+	g_signal_connect(G_OBJECT(windata->body_label), "activate-link",
+                         G_CALLBACK(activate_link), windata);
 
 	atkobj = gtk_widget_get_accessible(windata->body_label);
 	atk_object_set_description(atkobj, "Notification body text.");
@@ -837,7 +844,7 @@ set_notification_text(GtkWindow *nw, con
 	gtk_label_set_markup(GTK_LABEL(windata->summary_label), str);
 	g_free(str);
 
-	sexy_url_label_set_markup(SEXY_URL_LABEL(windata->body_label), body);
+	gtk_label_set_markup (GTK_LABEL (windata->body_label), body);
 
 	if (body == NULL || *body == '\0')
 		gtk_widget_hide(windata->body_label);
