Index: notification-daemon-0.5.0/data/notification-daemon.schemas.in
===================================================================
--- notification-daemon-0.5.0.orig/data/notification-daemon.schemas.in
+++ notification-daemon-0.5.0/data/notification-daemon.schemas.in
@@ -18,7 +18,7 @@
    <applyto>/apps/notification-daemon/theme</applyto>
    <owner>notification-daemon</owner>
    <type>string</type>
-   <default>slider</default>
+   <default>nodoka</default>
    <locale name="C">
     <short>Current theme</short>
     <long>The theme used when displaying notifications.</long>
