diff -Nuar cups-1.6.1.orig/config-scripts/cups-directories.m4 cups-1.6.1/config-scripts/cups-directories.m4
--- cups-1.6.1.orig/config-scripts/cups-directories.m4	2012-04-23 20:26:57.000000000 +0300
+++ cups-1.6.1/config-scripts/cups-directories.m4	2013-05-19 19:36:33.103653088 +0300
@@ -107,7 +107,7 @@
 			;;
 		Linux*)
 			if test -d /usr/lib64; then
-				libdir="$exec_prefix/lib64"
+				libdir="$exec_prefix/lib"
 			fi
 			;;
 		HP-UX*)
diff -Nuar cups-1.6.1.orig/configure cups-1.6.1/configure
--- cups-1.6.1.orig/configure	2012-07-27 23:57:23.000000000 +0300
+++ cups-1.6.1/configure	2013-05-19 19:37:27.586654608 +0300
@@ -6016,9 +6016,7 @@
 			libdir="$exec_prefix/lib32"
 			;;
 		Linux*)
-			if test -d /usr/lib64; then
-				libdir="$exec_prefix/lib64"
-			fi
+
 			;;
 		HP-UX*)
 			if test -d /usr/lib/hpux32; then
diff -Nuar cups-1.6.1.orig/cups-config.in cups-1.6.1/cups-config.in
--- cups-1.6.1.orig/cups-config.in	2011-08-27 12:23:01.000000000 +0300
+++ cups-1.6.1/cups-config.in	2013-05-19 19:34:14.470649221 +0300
@@ -50,7 +50,7 @@
 	CFLAGS="$CFLAGS -I$includedir"
     fi
 
-    if test $libdir != /usr/lib -a $libdir != /usr/lib32 -a $libdir != /usr/lib64; then
+    if test $libdir != /usr/lib -a $libdir != /usr/lib -a $libdir != /usr/lib; then
 	LDFLAGS="$LDFLAGS -L$libdir"
     fi
 fi
