--- cups-config.in~	2014-01-18 13:56:12.423297761 +0100
+++ cups-config.in	2014-01-18 13:56:59.459964275 +0100
@@ -50,7 +50,7 @@ else
 	CFLAGS="$CFLAGS -I$includedir"
     fi
 
-    if test $libdir != /usr/lib -a $libdir != /usr/lib32 -a $libdir != /usr/lib64; then
+    if test $libdir != /usr/lib -a $libdir != /usr/lib32 -a $libdir != /usr/lib; then
 	LDFLAGS="$LDFLAGS -L$libdir"
     fi
 fi
--- config-scripts/cups-directories.m4~	2014-01-18 13:56:23.529964393 +0100
+++ config-scripts/cups-directories.m4	2014-01-18 13:57:40.706630807 +0100
@@ -103,8 +103,8 @@ dnl Fix "libdir" variable...
 if test "$libdir" = "\${exec_prefix}/lib"; then
 	case "$uname" in
 		Linux*)
-			if test -d /usr/lib64 -a ! -d /usr/lib64/fakeroot; then
-				libdir="$exec_prefix/lib64"
+			if test -d /usr/lib -a ! -d /usr/lib/fakeroot; then
+				libdir="$exec_prefix/lib"
 			fi
 			;;
 		HP-UX*)
--- configure~	2014-01-18 13:56:32.346631030 +0100
+++ configure	2014-01-18 13:57:12.076630898 +0100
@@ -5893,8 +5893,8 @@ fi
 if test "$libdir" = "\${exec_prefix}/lib"; then
 	case "$uname" in
 		Linux*)
-			if test -d /usr/lib64 -a ! -d /usr/lib64/fakeroot; then
-				libdir="$exec_prefix/lib64"
+			if test -d /usr/lib -a ! -d /usr/lib/fakeroot; then
+				libdir="$exec_prefix/lib"
 			fi
 			;;
 		HP-UX*)
