Index: libcanberra-0.12/acinclude.m4
===================================================================
--- libcanberra-0.12.orig/acinclude.m4
+++ libcanberra-0.12/acinclude.m4
@@ -112,7 +112,7 @@ acx_pthread_flags="pthreads none -Kthrea
 # pthread-config: use pthread-config program (for GNU Pth library)
 
 case "${host_cpu}-${host_os}" in
-        *solaris*)
+        *solaris*|*linux*)
 
         # On Solaris (at least, for some versions), libc contains stubbed
         # (non-functional) versions of the pthreads routines, so link-based
