diff -Nuar pisiyap-0.2.orig/PisiYap/templates/actions.py pisiyap-0.2/PisiYap/templates/actions.py
--- pisiyap-0.2.orig/PisiYap/templates/actions.py	2011-01-28 19:05:57.000000000 +0200
+++ pisiyap-0.2/PisiYap/templates/actions.py	2013-10-29 13:12:07.000000000 +0200
@@ -13,7 +13,7 @@
 # -*- coding: utf-8 -*-
 #
 # Licensed under the GNU General Public License, version 3.
-# See the file http://www.gnu.org/copyleft/gpl.txt
+# See the file http://www.gnu.org/licenses/gpl.txt
 
 from pisi.actionsapi import autotools
 from pisi.actionsapi import pisitools
@@ -35,7 +35,7 @@
 # -*- coding: utf-8 -*-
 #
 # Licensed under the GNU General Public License, version 3.
-# See the file http://www.gnu.org/copyleft/gpl.txt
+# See the file http://www.gnu.org/licenses/gpl.txt
 
 from pisi.actionsapi import cmaketools
 from pisi.actionsapi import pisitools
@@ -57,7 +57,7 @@
 # -*- coding: utf-8 -*-
 #
 # Licensed under the GNU General Public License, version 3.
-# See the file http://www.gnu.org/copyleft/gpl.txt
+# See the file http://www.gnu.org/licenses/gpl.txt
 
 from pisi.actionsapi import pisitools
 from pisi.actionsapi import kde4
@@ -78,7 +78,7 @@
 # -*- coding: utf-8 -*-
 #
 # Licensed under the GNU General Public License, version 3.
-# See the file http://www.gnu.org/copyleft/gpl.txt
+# See the file http://www.gnu.org/licenses/gpl.txt
 
 from pisi.actionsapi import pisitools
 from pisi.actionsapi import qt4
@@ -99,7 +99,7 @@
 # -*- coding: utf-8 -*-
 #
 # Licensed under the GNU General Public License, version 3.
-# See the file http://www.gnu.org/copyleft/gpl.txt
+# See the file http://www.gnu.org/licenses/gpl.txt
 
 from pisi.actionsapi import pythonmodules
 from pisi.actionsapi import pisitools
@@ -117,7 +117,7 @@
 # -*- coding: utf-8 -*-
 #
 # Licensed under the GNU General Public License, version 3.
-# See the file http://www.gnu.org/copyleft/gpl.txt
+# See the file http://www.gnu.org/licenses/gpl.txt
 
 from pisi.actionsapi import pisitools
 from pisi.actionsapi import scons
diff -Nuar pisiyap-0.2.orig/PisiYap/templates/pspec.py pisiyap-0.2/PisiYap/templates/pspec.py
--- pisiyap-0.2.orig/PisiYap/templates/pspec.py	2011-01-19 13:04:31.000000000 +0200
+++ pisiyap-0.2/PisiYap/templates/pspec.py	2013-10-29 13:09:17.000000000 +0200
@@ -10,7 +10,7 @@
 #
 
 pspecXml = u'''<?xml version="1.0" ?>
-<!DOCTYPE PISI SYSTEM "http://www.pardus.org.tr/projeler/pisi/pisi-spec.dtd">
+<!DOCTYPE PISI SYSTEM "http://www.pisilinux.org/projeler/pisi/pisi-spec.dtd">
 <PISI>
     <Source>
         <Name>%(packageName)s</Name>
