Index: misc/desktops/Makefile.am
===================================================================
--- misc/desktops/Makefile.am
+++ misc/desktops/Makefile.am
@@ -1,5 +1,4 @@
 EXTRA_DIST = $(pixmap_DATA) xine.wmconfig \
-	xine.desktop.in \
 	mime.default \
 	mime.extra \
 	$(icon16_DATA) \
@@ -7,7 +6,7 @@ EXTRA_DIST = $(pixmap_DATA) xine.wmconfi
 	$(icon32_DATA) \
 	$(icon48_DATA) 
 
-BUILT_SOURCES = xine.desktop
+BUILT_SOURCES =
 CLEANFILES = $(BUILT_SOURCES)
 
 pixmapdir = $(datadir)/pixmaps
@@ -22,20 +21,3 @@ icon32_DATA = icons/32/xine.png
 icon48_DATA = icons/48/xine.png
 
 pixmap_DATA = xine.xpm
-
-desktopdir = $(datadir)/applications
-desktop_DATA = xine.desktop
-
-if XINE_LIST
-xine.desktop: xine.desktop.in @XINE_LIST@
-	exec > $@; cat $<; \
-	@XINE_LIST@; \
-	awk '{ printf "%s;", $$1 }' < $(srcdir)/mime.extra
-else
-XINE_MIME = $(if $(shell test "$(XINE_CONFIG)" = '' && echo y),$(shell pkg-config --variable=plugindir libxine),$(shell $(XINE_CONFIG) --plugindir))/mime.types
-MIME = $(if $(shell test -f $(XINE_MIME) && echo y),$(XINE_MIME),$(srcdir)/mime.default)
-xine.desktop: xine.desktop.in $(MIME) $(srcdir)/mime.extra
-	exec > $@; \
-	cat $<; \
-	cat $(MIME) $(srcdir)/mime.extra | awk '{ printf "%s;", $$1 }'
-endif
