Index: xine-ui-0.99.6_20110320/src/xitk/kbindings_common.c
===================================================================
--- xine-ui-0.99.6_20110320.orig/src/xitk/kbindings_common.c
+++ xine-ui-0.99.6_20110320/src/xitk/kbindings_common.c
@@ -908,7 +908,7 @@ kbinding_t *kbindings_init_kbinding(void
   _kbinding_load_config(kbt, gGui->keymap_file);
   
   /* Just to check is there redundant entries, and inform user */
-  _kbindings_check_redundancy(kbt);
+  /* _kbindings_check_redundancy(kbt); */
 
   gGui->kbindings_enabled = 1;
 
