diff -urp xine-ui-0.99.5.orig/src/xitk/help.c xine-ui-0.99.5/src/xitk/help.c
--- xine-ui-0.99.5.orig/src/xitk/help.c	2009-07-19 18:27:09.000000000 +0100
+++ xine-ui-0.99.5/src/xitk/help.c	2009-07-19 18:27:25.000000000 +0100
@@ -182,7 +182,7 @@ static void help_sections(void) {
 	  help_add_section(locale_readme, lang->doc_encoding, order_num, section_name);
 	}
 	else {
-	  help_add_section(default_readme, "ISO-8859-1", order_num, section_name);
+	  help_add_section(default_readme, "UTF-8", order_num, section_name);
 	}
       }
     }
