--- configure.ac_orig	2010-11-15 11:54:58.428167033 -0800
+++ configure.ac	2010-11-15 14:26:50.556166647 -0800
@@ -229,14 +229,15 @@
 		CXXFLAGS=`echo ${CXXFLAGS} | sed 's/-g//g'`;
 	fi
 fi
-AC_F77_NOFORMAIN
+# AC_F77_NOFORMAIN
 FFLAGS="${FLIBS} ${FFLAGS}"
-if test "${CC}" != "icl"; then
-  AC_F77_FCN_MANGLE()
-fi
+# this macro is broken, so be sure to set -DVF77_ONEUNDERSCORE
+#if test "${CC}" != "icl"; then
+#  AC_F77_FCN_MANGLE()
+#fi
 AC_LANG_C
 AC_PROG_CPP
-AC_DISABLE_SHARED
+# AC_DISABLE_SHARED
 AM_PROG_LIBTOOL
 FFLAGS="${FFLAGS}"
 
@@ -308,7 +309,7 @@
 dnl # HANDLE COMPILER FLOATING POINT ERROR
 dnl # ------------------------------------
 
-AC_FPERROR
+# AC_FPERROR
 
 dnl ##########################################################################
 dnl 8. checks for library functions
@@ -816,7 +817,7 @@
 fi
 LDFLAGS=${LDFLAGS_SAVE};
 CPPFLAGS=${CPPFLAGS_SAVE};
-AC_SUBST(rl_lib)
+# AC_SUBST(rl_lib)
 
 dnl # BEGIN FETK LIBRARY/HEADER TEST BLOCK
 dnl # -----------------------
@@ -1280,7 +1282,9 @@
 else
   apbs_build_libraries="${top_apbsdir}/bin/libapbsmainroutines.la \
 			${top_apbsdir}/src/aaa_lib/libapbs.la \
-			${maloc_lib} ${lib_list}"
+			${top_apbsdir}/contrib/pmgZ/libapbspmgz.la \
+			${top_apbsdir}/contrib/aqua/libapbsaqua.la \
+			${maloc_lib} ${blas_lib} ${lib_list}"
 fi
 AC_SUBST(apbs_build_libraries)
 
