--- ./bin/Makefile.am_orig	2010-11-15 12:07:02.487166613 -0800
+++ ./bin/Makefile.am	2010-11-15 12:10:49.109166514 -0800
@@ -12,34 +12,31 @@
 
 build_libraries = @apbs_build_libraries@
 
-bindir = ${prefix}/bin
+# bindir = ${prefix}/bin
 bin_PROGRAMS     = apbs
 apbs_SOURCES     = main.c 
-apbs_LDADD       = ${build_libraries}
+apbs_LDADD       = ${build_libraries} ./libapbsmainroutines.la
 
 # This is a hack I suspect, but needed to force apbs to be
 # rebuilt anytime the make command is issued
 # Problem before is that there is no dependency between source file
 # recompilation and relinking the binary. There is probably
 # a cleaner way to do this, I can't find it, however.
-BUILT_SOURCES	 = rm.apbs
+# BUILT_SOURCES	 = rm.apbs
 
-rm.apbs: Makefile
-	-rm -f apbs
-
-libdir = ${prefix}/lib
+# libdir = ${prefix}/lib
 lib_LTLIBRARIES = libapbsmainroutines.la
 libapbsmainroutines_la_SOURCES = routines.c
-libapbsmainroutines_la_LDFLAGS = -version-info ${APBS_VERSION}
+libapbsmainroutines_la_LDFLAGS = @maloc_lib@ ../src/aaa_lib/libapbs.la -version-info ${APBS_VERSION}
 
 INCLUDES = @inc_list@
 AM_CFLAGS= @wall_pedantic@
 AM_LDFLAGS       = @LDFLAGS@ @FFLAGS@
  
-includedir = ${prefix}/include/apbs
-include_HEADERS = routines.h
+# includedir = ${prefix}/include/apbs
+# include_HEADERS = routines.h
 	
-install-data-local:
-	-cp -p ${top_apbsdir}/bin/ApbsClient.py ${prefix}/bin
-	-chmod 755 ${prefix}/bin/ApbsClient.py
-	-rm -f ${prefix}/bin/wsdl2py 
\ No newline at end of file
+#install-data-local:
+#	-cp -p ${top_apbsdir}/bin/ApbsClient.py ${prefix}/bin
+#	-chmod 755 ${prefix}/bin/ApbsClient.py
+#	-rm -f ${prefix}/bin/wsdl2py 
--- ./src/aaa_lib/Makefile.am_orig	2010-11-15 12:01:13.975197621 -0800
+++ ./src/aaa_lib/Makefile.am	2010-11-15 12:01:54.567274233 -0800
@@ -18,12 +18,12 @@
 AQUA_LIBS = ${top_builddir}/contrib/aqua/libapbsaqua.la
 BLAS_LIBS = @blas_lib@
 
-libdir = ${prefix}/lib
+# libdir = ${prefix}/lib
 lib_LTLIBRARIES = libapbs.la
 
 libapbs_la_SOURCES = apbs_link.c
 libapbs_la_LIBADD  = ${GEN_LIBS} ${FEM_LIBS} ${MG_LIBS} ${PMGZ_LIBS} ${AQUA_LIBS} ${BLAS_LIBS}
-libapbs_la_LDFLAGS = -version-info ${APBS_VERSION}
+libapbs_la_LDFLAGS = @maloc_lib@ -version-info ${APBS_VERSION}
 
 INCLUDES         = -I${top_srcdir}/src/generic -I${top_srcdir}/src/mg -I${top_srcdir}/src/fem @inc_list@
 AM_CFLAGS        = @wall_pedantic@
--- ./tools/python/vgrid/Makefile.am_orig	2010-11-15 12:02:55.717416873 -0800
+++ ./tools/python/vgrid/Makefile.am	2010-11-15 12:04:33.565198239 -0800
@@ -16,5 +16,6 @@
    noinst_PROGRAMS     = _vgrid.so
    _vgrid_so_SOURCES   = vgridlib.c
    _vgrid_so_LINK      = $(PY_CC) $(PY_LINKFLAGS) -o $@ $(PY_SHARED)
-   _vgrid_so_LDADD     = ${top_builddir}/bin/.libs/libapbsmainroutines.a ${top_builddir}/src/aaa_lib/.libs/libapbs.a $(PY_LDFLAGS) ${top_builddir}/contrib/lib/libmaloc.a @lib_list@
+   _vgrid_so_LDFLAGS   = $(PY_LDFLAGS) @maloc_lib@ @lib_list@
+   _vgrid_so_LDADD     = ${top_builddir}/bin/.libs/libapbsmainroutines.so ${top_builddir}/src/aaa_lib/.libs/libapbs.so
 endif
--- ./tools/python/Makefile.am_orig	2010-11-15 12:04:54.918478810 -0800
+++ ./tools/python/Makefile.am	2010-11-15 14:22:08.880166492 -0800
@@ -16,4 +16,5 @@
 noinst_PROGRAMS       = _apbslib.so
 _apbslib_so_SOURCES   = apbslib.c
 _apbslib_so_LINK      = $(PY_CC) $(PY_LINKFLAGS) -o $@ $(PY_SHARED)
-_apbslib_so_LDADD     = ${top_builddir}/bin/.libs/libapbsmainroutines.a ${top_builddir}/src/aaa_lib/.libs/libapbs.a ${top_builddir}/contrib/lib/libmaloc.a ${top_builddir}/contrib/blas/.libs/libapbsblas.a $(PY_LDFLAGS) ${z_lib} 
+_apbslib_so_LDFLAGS   = @maloc_lib@ @blas_lib@ $(PY_LDFLAGS)
+_apbslib_so_LDADD     = ${top_builddir}/bin/.libs/libapbsmainroutines.so ${top_builddir}/src/aaa_lib/.libs/libapbs.so
--- ./tools/arpack/Makefile.am_orig	2010-11-15 12:02:08.657198191 -0800
+++ ./tools/arpack/Makefile.am	2010-11-15 12:02:26.559231032 -0800
@@ -12,7 +12,7 @@
 
 noinst_PROGRAMS  = driver
 driver_SOURCES   = driver.c readhb.f ardriv.f
-driver_LDADD     = ../../src/aaa_lib/libapbs.la @arpack_lib@ @apbs_build_libraries@
+driver_LDADD     = ../../src/aaa_lib/libapbs.la @arpack_lib@ @apbs_build_libraries@ -latlas
 INCLUDES         = @inc_list@
 AM_CFLAGS           = @CFLAGS@
 AM_LDFLAGS           = @LDFLAGS@ 
--- ./contrib/opal/Makefile.am_orig	2010-11-15 12:00:27.148447572 -0800
+++ ./contrib/opal/Makefile.am	2010-11-15 12:00:51.998166490 -0800
@@ -9,7 +9,7 @@
 AUTOMAKE_OPTIONS = no-dependencies
 SUFFIXES = .H .h .C .c .f .o
 
-always_built_SUBDIRS = ZSI
+# always_built_SUBDIRS = ZSI
 
 SUBDIRS = $(always_built_SUBDIRS) .
 
