--- configure.ac.OSmanOS        2013-01-31 02:46:24.000000000 +0200
+++ configure.ac                2013-02-16 22:32:00.764755892 +0200
@@ -34,7 +34,7 @@
 
 AM_INIT_AUTOMAKE([1.11.2 no-dist-gzip dist-xz color-tests parallel-tests
                   subdir-objects])
-AM_SILENT_RULES([yes]) # make --enable-silent-rules the default.
+#~ AM_SILENT_RULES([yes]) # make --enable-silent-rules the default.
 
 dnl POSIXCHECK is worthwhile for maintainers, but adds several seconds
 dnl (more than 10% execution time) to ./configure, with no benefit for
