--- a/Makefile.am
+++ b/Makefile.am
@@ -8,9 +8,7 @@
 #SUBDIRS += docs tests
 SUBDIRS += tests
 
-ACLOCAL_AMFLAGS =			\
-	-I m4macros			\
-	$(NULL)
+ACLOCAL_AMFLAGS = -I m4macros
 
 CONFIGURE_DEPENDENCIES =		\
 	requires			\
--- a/liblangtag-gobject/Makefile.am
+++ b/liblangtag-gobject/Makefile.am
@@ -171,6 +171,8 @@
 	$(NULL)
 LangTag_0_4_gir_CFLAGS =		\
 	$(INCLUDES)			\
+	$(NULL)
+LangTag_0_4_gir_SCANNERFLAGS =		\
 	--symbol-prefix=lt		\
 	--identifier-prefix=Lt		\
 	-v				\
