diff -p -up ./liblangtag/Makefile.am.tv ./liblangtag/Makefile.am
--- ./liblangtag/Makefile.am.tv	2013-04-14 14:56:19.425732717 +0100
+++ ./liblangtag/Makefile.am	2013-04-14 14:56:28.003388062 +0100
@@ -182,7 +182,7 @@ liblangtag_la_DEPENDENCIES =					\
 	$(included_modules)					\
 	$(NULL)
 liblangtag_la_LDFLAGS =						\
-	$(LDFLAGS)						\
+	$(LDFLAGS) -ldl						\
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)	\
 	$(NULL)
 
diff -p -up ./liblangtag/Makefile.in.tv ./liblangtag/Makefile.in
--- ./liblangtag/Makefile.in.tv	2013-04-14 14:56:21.265874069 +0100
+++ ./liblangtag/Makefile.in	2013-04-14 14:56:43.765589395 +0100
@@ -546,7 +546,7 @@ liblangtag_la_DEPENDENCIES = \
 	$(NULL)
 
 liblangtag_la_LDFLAGS = \
-	$(LDFLAGS)						\
+	$(LDFLAGS) -ldl						\
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)	\
 	$(NULL)
  
