 liblangtag/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/liblangtag/Makefile.am b/liblangtag/Makefile.am
index a0bbaa3..921ccb4 100644
--- a/liblangtag/Makefile.am
+++ b/liblangtag/Makefile.am
@@ -203,6 +203,7 @@ liblangtag_la_CFLAGS =				\
 liblangtag_la_LIBADD =						\
 	$(LIBS)							\
 	$(included_modules)					\
+	@MODULE_LIBS@ 							\
 	$(NULL)
 liblangtag_la_DEPENDENCIES =					\
 	$(included_modules)					\
