From cde50b7dc80567d6475ee229649c8aa984c018c3 Mon Sep 17 00:00:00 2001
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Wed, 3 Apr 2013 01:48:19 +0200
Subject: [PATCH] fixes for -Wformat

--- liblangtag-0.4.0.orig/liblangtag/lt-ext-module.c
+++ liblangtag-0.4.0/liblangtag/lt-ext-module.c
@@ -258,7 +258,7 @@ lt_ext_module_load(lt_ext_module_t *modu
 					       (lt_destroy_func_t)dlclose);
 				func = dlsym(module->module, "module_get_version");
 				if (!func) {
-					lt_warning(dlerror());
+					lt_warning("%s", dlerror());
 					break;
 				}
 				if (((lt_ext_module_version_func_t)func)() != LT_EXT_MODULE_VERSION) {
@@ -268,7 +268,7 @@ lt_ext_module_load(lt_ext_module_t *modu
 				}
 				func = dlsym(module->module, "module_get_funcs");
 				if (!func) {
-					lt_warning(dlerror());
+					lt_warning("%s", dlerror());
 					break;
 				}
 				if (!(module->funcs = ((lt_ext_module_get_funcs_func_t)func)())) {
--- liblangtag-0.4.0.orig/tests/main.c
+++ liblangtag-0.4.0/tests/main.c
@@ -37,7 +37,7 @@ logger(lt_message_type_t      type,
        const char            *message,
        lt_pointer_t           user_data)
 {
-	lt_error_set(&error, type, message);
+	lt_error_set(&error, type, "%s", message);
 }
 
 static void
