Index: configure.in
===================================================================
--- configure.in
+++ configure.in	2009-05-27 10:33:24.000000000 +0200
@@ -6,6 +6,17 @@
 
 AM_INIT_AUTOMAKE([1.9 check-news dist-bzip2])
 AM_CONFIG_HEADER(config.h)
+
+dnl AM_ACLOCAL_INCLUDE(macrodir)
+AC_DEFUN([AM_ACLOCAL_INCLUDE],
+[
+    AM_CONDITIONAL(INSIDE_GNOME_COMMON, test x = y)
+
+    test -n "$ACLOCAL_FLAGS" && ACLOCAL="$ACLOCAL $ACLOCAL_FLAGS"
+
+    for k in $1 ; do ACLOCAL="$ACLOCAL -I $k" ; done
+])
+
 AM_ACLOCAL_INCLUDE(m4)
 AM_MAINTAINER_MODE
 
