http://bugs.gentoo.org/333621

--- contrib/ntsc-cc.c
+++ contrib/ntsc-cc.c
@@ -34,6 +34,7 @@
 #include <locale.h>
 #include <sys/ioctl.h>
 #include <sys/types.h>
+#include <sys/stat.h>
 #include <sys/time.h>
 #ifdef HAVE_GETOPT_LONG
 #  include <getopt.h>
--- src/io-dvb.c
+++ src/io-dvb.c
@@ -29,6 +29,7 @@
 #include <errno.h>
 #include <sys/select.h>
 #include <sys/ioctl.h>
+#include <sys/stat.h>
 
 #ifndef HAVE_S64_U64
   /* Linux 2.6.x asm/types.h defines __s64 and __u64 only
--- src/io-v4l.c
+++ src/io-v4l.c
@@ -42,6 +42,7 @@
 #include <assert.h>
 #include <sys/time.h>		/* timeval */
 #include <sys/types.h>		/* fd_set, uid_t */
+#include <sys/stat.h>
 #include <sys/ioctl.h>		/* for (_)videodev.h */
 #include <pthread.h>
 
