Index: python-ethumb-0.2_49588/setup.py
===================================================================
--- python-ethumb-0.2_49588.orig/setup.py
+++ python-ethumb-0.2_49588/setup.py
@@ -62,7 +62,7 @@ ethumbmodule = Extension('ethumb.c_ethum
                                    ],
                           depends=['include/ethumb/c_ethumb.pxd',
                                    ],
-                          **pkgconfig('"ethumb >= 0.1"'))
+                          **pkgconfig('"ethumb >= 0.1"', 'ecore', 'eina-0'))
 
 
 trove_classifiers = [
