--- libs/ardour/session.cc~	2006-05-14 09:53:33.000000000 -0700
+++ libs/ardour/session.cc	2006-05-14 10:20:11.000000000 -0700
@@ -3218,7 +3218,8 @@
 #endif			
 		_passthru_buffers.push_back (p);
 
-		*p = 0;
+		if (current_block_size > 0)
+		  *p = 0;
 		
 #ifdef NO_POSIX_MEMALIGN
 		p =  (Sample *) malloc(current_block_size * sizeof(Sample));
@@ -3228,7 +3229,8 @@
 		memset (p, 0, sizeof (Sample) * current_block_size);
 		_silent_buffers.push_back (p);
 
-		*p = 0;
+		if (current_block_size > 0)
+		  *p = 0;
 		
 #ifdef NO_POSIX_MEMALIGN
 		p =  (Sample *) malloc(current_block_size * sizeof(Sample));
