--- libglademm-2.6.7/Makefile.in.orig	2009-09-07 10:21:28.220635805 +0300
+++ libglademm-2.6.7/Makefile.in	2009-09-07 10:22:10.602510838 +0300
@@ -200,7 +200,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = tools libglade docs examples MSVC_Net2005
+SUBDIRS = tools libglade examples MSVC_Net2005
 DIST_SUBDIRS = $(SUBDIRS) scripts 
 EXTRA_DIST = \
 		  build_shared/Makefile_build.am_fragment \
