Index: configure.ac
===================================================================
--- configure.ac.orig	2008-05-16 16:40:37.000000000 +0200
+++ configure.ac	2008-05-16 16:40:52.265115841 +0200
@@ -95,7 +95,7 @@ AC_SUBST(MCM_KEY_VALID_LEN)
 
 
 dnl CFLAGS crap
-CFLAGS="-Wall -pipe"
+CFLAGS="${CFLAGS} -Wall -pipe"
 DEBUG_cflags="-g -Wunused -Wshadow -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wwrite-strings -Wsign-compare -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wmissing-noreturn -Wformat -Wredundant-decls -Wnested-externs -Winline -pedantic"
 OPTIMIZE="-O3"
 PROFILE="-g -pg"

