--- src/buffer.c.orig	2006-02-21 08:58:59.000000000 +0100
+++ src/buffer.c	2008-05-16 16:40:40.808416442 +0200
@@ -259,7 +259,7 @@ mcm_buf_free(struct memcache_ctxt *ctxt,
 }
 
 
-inline u_int32_t
+u_int32_t
 mcm_buf_len(const struct memcache_ctxt *ctxt, const struct memcache_buf *s) {
   return s->len;
 }
@@ -425,13 +425,13 @@ mcm_buf_realloc(struct memcache_ctxt *ct
 }
 
 
-inline size_t
+size_t
 mcm_buf_remain(const struct memcache_ctxt *ctxt, const struct memcache_buf *buf) {
   return mcm_buf_size(ctxt, buf) - mcm_buf_len(ctxt, buf);
 }
 
 
-inline size_t
+size_t
 mcm_buf_remain_off(const struct memcache_ctxt *ctxt, const struct memcache_buf *buf) {
   return mcm_buf_len(ctxt, buf) - buf->off;
 }
