From 6404a98fb79e7bb4c3e9c5ca9919e12c946679d7 Mon Sep 17 00:00:00 2001
From: Nelson Marques <nmo.marques@gmail.com>
Date: Sun, 11 Nov 2012 16:58:20 +0000
Subject: [PATCH] fix startup rendering effect with composite enabled

---
 src/compositor/compositor-xrender.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/compositor/compositor-xrender.c b/src/compositor/compositor-xrender.c
index 004dfd8..f68d5c4 100644
--- a/src/compositor/compositor-xrender.c
+++ b/src/compositor/compositor-xrender.c
@@ -3059,6 +3059,8 @@ meta_compositor_xrender_new (MetaDisplay *display)
   xrc->atom_net_wm_window_type_toolbar = atoms[12];
   xrc->atom_net_wm_window_type_dropdown_menu = atoms[13];
   xrc->atom_net_wm_window_type_tooltip = atoms[14];
+  xrc->show_redraw = FALSE;
+  xrc->debug = FALSE;
 
 #ifdef USE_IDLE_REPAINT
   meta_verbose ("Using idle repaint\n");
-- 
1.7.11.7

