diff -uNrp mate-window-manager-1.5.2/distro/archlinux/mate-window-manager.install mate-window-manager/distro/archlinux/mate-window-manager.install
--- mate-window-manager-1.5.2/distro/archlinux/mate-window-manager.install	1969-12-31 16:00:00.000000000 -0800
+++ mate-window-manager/distro/archlinux/mate-window-manager.install	2013-01-11 15:55:34.480170341 -0800
@@ -0,0 +1,17 @@
+ pkgname=mate-window-manager
+
+post_install() {
+	usr/sbin/mateconfpkg --install ${pkgname}
+}
+
+pre_upgrade() {
+	pre_remove $1
+}
+
+post_upgrade() {
+	post_install $1
+}
+
+pre_remove() {
+	usr/sbin/mateconfpkg --uninstall ${pkgname}
+}
diff -uNrp mate-window-manager-1.5.2/distro/archlinux/PKGBUILD mate-window-manager/distro/archlinux/PKGBUILD
--- mate-window-manager-1.5.2/distro/archlinux/PKGBUILD	1969-12-31 16:00:00.000000000 -0800
+++ mate-window-manager/distro/archlinux/PKGBUILD	2013-01-11 15:55:34.480170341 -0800
@@ -0,0 +1,37 @@
+pkgname=mate-window-manager
+pkgver=1.4.0
+pkgrel=1
+pkgdesc="A window manager for MATE"
+arch=(i686 x86_64)
+license=('GPL')
+depends=('mate-conf' 'mate-dialogs' 'libgtop' 'libwnck') #  'libcanberra' 'startup-notification'
+makedepends=('intltool' 'pkgconfig' 'mate-doc-utils')
+options=('!emptydirs' '!libtool')
+install=mate-window-manager.install
+url="http://matsusoft.com.ar/projects/mate/"
+groups=('mate')
+source=(metacity-restartstyle.patch)
+sha256sums=('ea2450ca3e4fb59166ec18ff4844cc61cb16865880dc6b125078a3ab47bf2caf')
+
+build() {
+	cd "${srcdir}/../../../"
+    ./autogen.sh \
+        --prefix=/usr \
+        --sysconfdir=/etc \
+        --localstatedir=/var \
+        --libexecdir=/usr/lib/mate-window-manager \
+        --disable-static \
+        --disable-startup-notification \
+        --disable-scrollkeeper || return 1
+    make || return 1
+}
+
+package() {
+	cd "${srcdir}/../../../"
+
+    make MATECONF_DISABLE_MAKEFILE_SCHEMA_INSTALL=1 DESTDIR="${pkgdir}" install || return 1
+
+    install -m755 -d "${pkgdir}/usr/share/mateconf/schemas"
+    mateconf-merge-schema "${pkgdir}/usr/share/mateconf/schemas/${pkgname}.schemas" --domain ${pkgname} ${pkgdir}/etc/mateconf/schemas/*.schemas
+    rm -f ${pkgdir}/etc/mateconf/schemas/*.schemas
+}
diff -uNrp mate-window-manager-1.5.2/src/compositor/compositor.c mate-window-manager/src/compositor/compositor.c
--- mate-window-manager-1.5.2/src/compositor/compositor.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/compositor/compositor.c	2013-01-11 15:55:34.535171007 -0800
@@ -15,8 +15,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/compositor/compositor-private.h mate-window-manager/src/compositor/compositor-private.h
--- mate-window-manager-1.5.2/src/compositor/compositor-private.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/compositor/compositor-private.h	2013-01-11 15:55:34.534170995 -0800
@@ -15,8 +15,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_COMPOSITOR_PRIVATE_H
diff -uNrp mate-window-manager-1.5.2/src/compositor/compositor-xrender.c mate-window-manager/src/compositor/compositor-xrender.c
--- mate-window-manager-1.5.2/src/compositor/compositor-xrender.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/compositor/compositor-xrender.c	2013-01-11 15:55:34.535171007 -0800
@@ -17,8 +17,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #define _GNU_SOURCE
@@ -62,6 +62,12 @@
 
 #define USE_IDLE_REPAINT 1
 
+#ifdef __GNUC__
+#define UNUSED_VARIABLE __attribute__ ((unused))
+#else
+#define UNUSED_VARIABLE
+#endif
+
 #ifdef HAVE_COMPOSITE_EXTENSIONS
 static inline gboolean
 composite_at_least_version (MetaDisplay *display,
@@ -1131,8 +1137,8 @@ paint_windows (MetaScreen   *screen,
   MetaCompScreen *info = meta_screen_get_compositor_data (screen);
   Display *xdisplay = meta_display_get_xdisplay (display);
   GList *index, *last;
-  int screen_width, screen_height, screen_number;
-  Window xroot;
+  int screen_width, screen_height, UNUSED_VARIABLE screen_number;
+  Window UNUSED_VARIABLE xroot;
   MetaCompWindow *cw;
   XserverRegion paint_region, desktop_region;
 
@@ -3059,6 +3065,8 @@ meta_compositor_xrender_new (MetaDisplay
   xrc->atom_net_wm_window_type_toolbar = atoms[12];
   xrc->atom_net_wm_window_type_dropdown_menu = atoms[13];
   xrc->atom_net_wm_window_type_tooltip = atoms[14];
+  xrc->show_redraw = FALSE;
+  xrc->debug = FALSE;
 
 #ifdef USE_IDLE_REPAINT
   meta_verbose ("Using idle repaint\n");
diff -uNrp mate-window-manager-1.5.2/src/compositor/compositor-xrender.h mate-window-manager/src/compositor/compositor-xrender.h
--- mate-window-manager-1.5.2/src/compositor/compositor-xrender.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/compositor/compositor-xrender.h	2013-01-11 15:55:34.535171007 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_COMPOSITOR_XRENDER_H_
diff -uNrp mate-window-manager-1.5.2/src/core/atomnames.h mate-window-manager/src/core/atomnames.h
--- mate-window-manager-1.5.2/src/core/atomnames.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/atomnames.h	2013-01-11 15:55:34.535171007 -0800
@@ -19,8 +19,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 /**
diff -uNrp mate-window-manager-1.5.2/src/core/bell.c mate-window-manager/src/core/bell.c
--- mate-window-manager-1.5.2/src/core/bell.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/bell.c	2013-01-11 15:55:34.535171007 -0800
@@ -18,8 +18,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 /**
diff -uNrp mate-window-manager-1.5.2/src/core/bell.h mate-window-manager/src/core/bell.h
--- mate-window-manager-1.5.2/src/core/bell.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/bell.h	2013-01-11 15:55:34.535171007 -0800
@@ -27,8 +27,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <X11/Xlib.h>
diff -uNrp mate-window-manager-1.5.2/src/core/boxes.c mate-window-manager/src/core/boxes.c
--- mate-window-manager-1.5.2/src/core/boxes.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/boxes.c	2013-01-11 15:55:34.535171007 -0800
@@ -22,8 +22,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include "boxes.h"
@@ -311,7 +311,6 @@ meta_rectangle_resize_with_gravity (cons
    */
 
   /* First, the x direction */
-  int adjust = 0;
   switch (gravity)
     {
     case NorthWestGravity:
@@ -344,7 +343,6 @@ meta_rectangle_resize_with_gravity (cons
   rect->width = new_width;
 
   /* Next, the y direction */
-  adjust = 0;
   switch (gravity)
     {
     case NorthWestGravity:
diff -uNrp mate-window-manager-1.5.2/src/core/constraints.c mate-window-manager/src/core/constraints.c
--- mate-window-manager-1.5.2/src/core/constraints.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/constraints.c	2013-01-11 15:55:34.536171019 -0800
@@ -19,8 +19,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/core/constraints.h mate-window-manager/src/core/constraints.h
--- mate-window-manager-1.5.2/src/core/constraints.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/constraints.h	2013-01-11 15:55:34.536171019 -0800
@@ -18,8 +18,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_CONSTRAINTS_H
diff -uNrp mate-window-manager-1.5.2/src/core/core.c mate-window-manager/src/core/core.c
--- mate-window-manager-1.5.2/src/core/core.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/core.c	2013-01-11 15:55:34.536171019 -0800
@@ -19,8 +19,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/core/delete.c mate-window-manager/src/core/delete.c
--- mate-window-manager-1.5.2/src/core/delete.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/delete.c	2013-01-11 15:55:34.536171019 -0800
@@ -18,8 +18,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #define _GNU_SOURCE
diff -uNrp mate-window-manager-1.5.2/src/core/display.c mate-window-manager/src/core/display.c
--- mate-window-manager-1.5.2/src/core/display.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/display.c	2013-01-11 15:55:34.538171045 -0800
@@ -18,8 +18,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 /**
@@ -2805,7 +2805,11 @@ key_event_description (Display *xdisplay
   KeySym keysym;
   const char *str;
   
+#ifdef HAVE_XKB
+  keysym = XkbKeycodeToKeysym (xdisplay, event->xkey.keycode, 0, 0);  
+#else
   keysym = XKeycodeToKeysym (xdisplay, event->xkey.keycode, 0);  
+#endif
 
   str = XKeysymToString (keysym);
   
diff -uNrp mate-window-manager-1.5.2/src/core/display-private.h mate-window-manager/src/core/display-private.h
--- mate-window-manager-1.5.2/src/core/display-private.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/display-private.h	2013-01-11 15:55:34.537171032 -0800
@@ -18,8 +18,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_DISPLAY_PRIVATE_H
diff -uNrp mate-window-manager-1.5.2/src/core/edge-resistance.c mate-window-manager/src/core/edge-resistance.c
--- mate-window-manager-1.5.2/src/core/edge-resistance.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/edge-resistance.c	2013-01-11 15:55:34.538171045 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/core/edge-resistance.h mate-window-manager/src/core/edge-resistance.h
--- mate-window-manager-1.5.2/src/core/edge-resistance.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/edge-resistance.h	2013-01-11 15:55:34.538171045 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_EDGE_RESISTANCE_H
diff -uNrp mate-window-manager-1.5.2/src/core/effects.c mate-window-manager/src/core/effects.c
--- mate-window-manager-1.5.2/src/core/effects.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/effects.c	2013-01-11 15:55:34.538171045 -0800
@@ -51,8 +51,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/core/effects.h mate-window-manager/src/core/effects.h
--- mate-window-manager-1.5.2/src/core/effects.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/effects.h	2013-01-11 15:55:34.538171045 -0800
@@ -38,8 +38,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_EFFECTS_H
diff -uNrp mate-window-manager-1.5.2/src/core/errors.c mate-window-manager/src/core/errors.c
--- mate-window-manager-1.5.2/src/core/errors.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/errors.c	2013-01-11 15:55:34.538171045 -0800
@@ -18,8 +18,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
@@ -29,6 +29,12 @@
 #include <stdlib.h>
 #include <gdk/gdk.h>
 
+#ifdef __GNUC__
+#define UNUSED_VARIABLE __attribute__ ((unused))
+#else
+#define UNUSED_VARIABLE
+#endif
+
 static int x_error_handler    (Display     *display,
                                XErrorEvent *error);
 static int x_io_error_handler (Display     *display);
@@ -121,7 +127,7 @@ meta_error_trap_pop_internal  (MetaDispl
        * assumes that there are no pending GDK traps from GDK itself
        */
       
-      int (* restored_error_handler) (Display     *,
+      int UNUSED_VARIABLE (* restored_error_handler) (Display     *,
                                       XErrorEvent *);
 
       restored_error_handler = XSetErrorHandler (x_error_handler);
diff -uNrp mate-window-manager-1.5.2/src/core/eventqueue.c mate-window-manager/src/core/eventqueue.c
--- mate-window-manager-1.5.2/src/core/eventqueue.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/eventqueue.c	2013-01-11 15:55:34.538171045 -0800
@@ -18,8 +18,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.  */
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.  */
 
 #include "eventqueue.h"
 #include <X11/Xlib.h>
diff -uNrp mate-window-manager-1.5.2/src/core/eventqueue.h mate-window-manager/src/core/eventqueue.h
--- mate-window-manager-1.5.2/src/core/eventqueue.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/eventqueue.h	2013-01-11 15:55:34.538171045 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_EVENT_QUEUE_H
diff -uNrp mate-window-manager-1.5.2/src/core/frame.c mate-window-manager/src/core/frame.c
--- mate-window-manager-1.5.2/src/core/frame.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/frame.c	2013-01-11 15:55:34.539171058 -0800
@@ -19,8 +19,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/core/frame-private.h mate-window-manager/src/core/frame-private.h
--- mate-window-manager-1.5.2/src/core/frame-private.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/frame-private.h	2013-01-11 15:55:34.539171058 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_FRAME_PRIVATE_H
diff -uNrp mate-window-manager-1.5.2/src/core/group.c mate-window-manager/src/core/group.c
--- mate-window-manager-1.5.2/src/core/group.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/group.c	2013-01-11 15:55:34.539171058 -0800
@@ -18,8 +18,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/core/group.h mate-window-manager/src/core/group.h
--- mate-window-manager-1.5.2/src/core/group.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/group.h	2013-01-11 15:55:34.539171058 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_GROUP_H
diff -uNrp mate-window-manager-1.5.2/src/core/group-private.h mate-window-manager/src/core/group-private.h
--- mate-window-manager-1.5.2/src/core/group-private.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/group-private.h	2013-01-11 15:55:34.539171058 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_GROUP_PRIVATE_H
diff -uNrp mate-window-manager-1.5.2/src/core/group-props.c mate-window-manager/src/core/group-props.c
--- mate-window-manager-1.5.2/src/core/group-props.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/group-props.c	2013-01-11 15:55:34.539171058 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/core/group-props.h mate-window-manager/src/core/group-props.h
--- mate-window-manager-1.5.2/src/core/group-props.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/group-props.h	2013-01-11 15:55:34.539171058 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_GROUP_PROPS_H
diff -uNrp mate-window-manager-1.5.2/src/core/iconcache.c mate-window-manager/src/core/iconcache.c
--- mate-window-manager-1.5.2/src/core/iconcache.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/iconcache.c	2013-01-11 15:55:34.539171058 -0800
@@ -17,8 +17,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/core/iconcache.h mate-window-manager/src/core/iconcache.h
--- mate-window-manager-1.5.2/src/core/iconcache.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/iconcache.h	2013-01-11 15:55:34.539171058 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_ICON_CACHE_H
diff -uNrp mate-window-manager-1.5.2/src/core/keybindings.c mate-window-manager/src/core/keybindings.c
--- mate-window-manager-1.5.2/src/core/keybindings.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/keybindings.c	2013-01-11 15:55:34.540171070 -0800
@@ -19,8 +19,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #define _GNU_SOURCE
@@ -1273,7 +1273,11 @@ meta_display_process_key_event (MetaDisp
   
   /* window may be NULL */
   
+#ifdef HAVE_XKB
+  keysym = XkbKeycodeToKeysym (display->xdisplay, event->xkey.keycode, 0, 0);
+#else
   keysym = XKeycodeToKeysym (display->xdisplay, event->xkey.keycode, 0);
+#endif
 
   str = XKeysymToString (keysym);
   
diff -uNrp mate-window-manager-1.5.2/src/core/keybindings.h mate-window-manager/src/core/keybindings.h
--- mate-window-manager-1.5.2/src/core/keybindings.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/keybindings.h	2013-01-11 15:55:34.540171070 -0800
@@ -22,8 +22,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_KEYBINDINGS_H
diff -uNrp mate-window-manager-1.5.2/src/core/main.c mate-window-manager/src/core/main.c
--- mate-window-manager-1.5.2/src/core/main.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/main.c	2013-01-11 15:55:34.540171070 -0800
@@ -18,8 +18,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 /**
@@ -69,6 +69,12 @@
 #include <time.h>
 #include <unistd.h>
 
+#ifdef __GNUC__
+#define UNUSED_VARIABLE __attribute__ ((unused))
+#else
+#define UNUSED_VARIABLE
+#endif
+
 /**
  * The exit code we'll return to our parent process when we eventually die.
  */
@@ -376,7 +382,7 @@ sigterm_handler (int signum)
 {
   if (sigterm_pipe_fds[1] >= 0)
     {
-      int dummy;
+      int UNUSED_VARIABLE dummy;
 
       dummy = write (sigterm_pipe_fds[1], "", 1);
       close (sigterm_pipe_fds[1]);
@@ -415,8 +421,12 @@ main (int argc, char **argv)
   guint i;
   GIOChannel *channel;
 
+#if GLIB_CHECK_VERSION (2, 32, 0)
+  /* g_thread_init () deprecated */
+#else
   if (!g_thread_supported ())
     g_thread_init (NULL);
+#endif
   
   if (setlocale (LC_ALL, "") == NULL)
     meta_warning ("Locale not understood by C library, internationalization will not work\n");
diff -uNrp mate-window-manager-1.5.2/src/core/place.c mate-window-manager/src/core/place.c
--- mate-window-manager-1.5.2/src/core/place.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/place.c	2013-01-11 15:55:34.540171070 -0800
@@ -20,8 +20,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/core/place.h mate-window-manager/src/core/place.h
--- mate-window-manager-1.5.2/src/core/place.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/place.h	2013-01-11 15:55:34.540171070 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_PLACE_H
diff -uNrp mate-window-manager-1.5.2/src/core/prefs.c mate-window-manager/src/core/prefs.c
--- mate-window-manager-1.5.2/src/core/prefs.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/prefs.c	2013-01-11 15:55:34.541171082 -0800
@@ -19,8 +19,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
@@ -31,7 +31,7 @@
 #include <string.h>
 #include <stdlib.h>
 
-#define MAX_REASONABLE_WORKSPACES 36
+#define MAX_REASONABLE_WORKSPACES 16
 
 #define MAX_COMMANDS (32 + NUM_EXTRA_COMMANDS)
 #define NUM_EXTRA_COMMANDS 2
@@ -143,7 +143,10 @@ static char* settings_key_for_workspace_
 
 static void queue_changed (MetaPreference  pref);
 
+#if 0
 static void     cleanup_error             (GError **error);
+#endif
+
 static void maybe_give_disable_workarounds_warning (void);
 
 static void titlebar_handler (MetaPreference, const gchar*, gboolean*);
@@ -275,37 +278,37 @@ static MetaEnumPreference preferences_en
     { "focus-new-windows",
       KEY_GENERAL_SCHEMA,
       META_PREF_FOCUS_NEW_WINDOWS,
-      &focus_new_windows,
+      (gint *) &focus_new_windows,
     },
     { "focus-mode",
       KEY_GENERAL_SCHEMA,
       META_PREF_FOCUS_MODE,
-      &focus_mode,
+      (gint *) &focus_mode,
     },
     { "wrap-style",
       KEY_GENERAL_SCHEMA,
       META_PREF_WRAP_STYLE,
-      &wrap_style,
+      (gint *) &wrap_style,
     },
     { "visual-bell-type",
       KEY_GENERAL_SCHEMA,
       META_PREF_VISUAL_BELL_TYPE,
-      &visual_bell_type,
+      (gint *) &visual_bell_type,
     },
     { "action-double-click-titlebar",
       KEY_GENERAL_SCHEMA,
       META_PREF_ACTION_DOUBLE_CLICK_TITLEBAR,
-      &action_double_click_titlebar,
+      (gint *) &action_double_click_titlebar,
     },
     { "action-middle-click-titlebar",
       KEY_GENERAL_SCHEMA,
       META_PREF_ACTION_MIDDLE_CLICK_TITLEBAR,
-      &action_middle_click_titlebar,
+      (gint *) &action_middle_click_titlebar,
     },
     { "action-right-click-titlebar",
       KEY_GENERAL_SCHEMA,
       META_PREF_ACTION_RIGHT_CLICK_TITLEBAR,
-      &action_right_click_titlebar,
+      (gint *) &action_right_click_titlebar,
     },
     { NULL, NULL, 0, NULL },
   };
@@ -592,7 +595,7 @@ handle_preference_update_enum (const gch
   old_value = * ((gint *) cursor->target);
   
   /* Now look it up... */
-  *cursor->target = g_settings_get_int (settings, key);
+  *cursor->target = g_settings_get_enum (settings, key);
 
   /* Did it change?  If so, tell the listeners about it. */
 
@@ -898,7 +901,7 @@ meta_prefs_init (void)
 /* Updates.                                                                 */
 /****************************************************************************/
 
-gboolean (*preference_update_handler[]) (const gchar*, GVariant*) = {
+gboolean (*preference_update_handler[]) (const gchar*, GSettings*) = {
   handle_preference_update_enum,
   handle_preference_update_bool,
   handle_preference_update_string,
@@ -975,6 +978,7 @@ change_notify (GSettings *settings,
   g_free (schema_name);
 }
 
+#if 0
 static void
 cleanup_error (GError **error)
 {
@@ -986,6 +990,7 @@ cleanup_error (GError **error)
       *error = NULL;
     }
 }
+#endif
 
 /**
  * Special case: give a warning the first time disable_workarounds
@@ -1566,9 +1571,9 @@ init_bindings (void)
     NULL
   };
   int i;
-  gchar **list;
-  gchar *str_val;
-  GSettings *bindings_settings;
+  gchar **list = NULL;
+  gchar *str_val = NULL;
+  GSettings *bindings_settings = NULL;
 
   for (i = 0; prefix[i]; i++)
     {
@@ -1589,8 +1594,8 @@ init_bindings (void)
 static void
 init_commands (void)
 {
-  gchar **list;
-  gchar *str_val;
+  gchar **list = NULL;
+  gchar *str_val = NULL;
 
   list = g_settings_list_keys (settings_command);
   while (*list != NULL)
@@ -1605,8 +1610,8 @@ init_commands (void)
 static void
 init_workspace_names (void)
 {
-  gchar **list;
-  gchar *str_val;
+  gchar **list = NULL;
+  gchar *str_val = NULL;
 
   list = g_settings_list_keys (settings_workspace_names);
   while (*list != NULL)
diff -uNrp mate-window-manager-1.5.2/src/core/schema-bindings.c mate-window-manager/src/core/schema-bindings.c
--- mate-window-manager-1.5.2/src/core/schema-bindings.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/schema-bindings.c	2013-01-11 15:55:34.541171082 -0800
@@ -15,8 +15,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 /** \file  Schema bindings generator.
diff -uNrp mate-window-manager-1.5.2/src/core/screen.c mate-window-manager/src/core/screen.c
--- mate-window-manager-1.5.2/src/core/screen.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/screen.c	2013-01-11 15:55:34.541171082 -0800
@@ -22,8 +22,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
@@ -1272,8 +1272,8 @@ meta_screen_ensure_tab_popup (MetaScreen
     {
       MetaWindow *window;
       MetaRectangle r;
-      GdkPixbuf *win_pixbuf;
-      int width, height;
+      GdkPixbuf *win_pixbuf = NULL;
+      int width = 0, height = 0;
 
       window = tmp->data;
       
diff -uNrp mate-window-manager-1.5.2/src/core/screen-private.h mate-window-manager/src/core/screen-private.h
--- mate-window-manager-1.5.2/src/core/screen-private.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/screen-private.h	2013-01-11 15:55:34.541171082 -0800
@@ -26,8 +26,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_SCREEN_PRIVATE_H
diff -uNrp mate-window-manager-1.5.2/src/core/session.c mate-window-manager/src/core/session.c
--- mate-window-manager-1.5.2/src/core/session.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/session.c	2013-01-11 15:55:34.542171094 -0800
@@ -19,8 +19,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/core/session.h mate-window-manager/src/core/session.h
--- mate-window-manager-1.5.2/src/core/session.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/session.h	2013-01-11 15:55:34.542171094 -0800
@@ -24,8 +24,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_SESSION_H
diff -uNrp mate-window-manager-1.5.2/src/core/stack.c mate-window-manager/src/core/stack.c
--- mate-window-manager-1.5.2/src/core/stack.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/stack.c	2013-01-11 15:55:34.542171094 -0800
@@ -22,8 +22,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/core/stack.h mate-window-manager/src/core/stack.h
--- mate-window-manager-1.5.2/src/core/stack.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/stack.h	2013-01-11 15:55:34.542171094 -0800
@@ -34,8 +34,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_STACK_H
diff -uNrp mate-window-manager-1.5.2/src/core/testasyncgetprop.c mate-window-manager/src/core/testasyncgetprop.c
--- mate-window-manager-1.5.2/src/core/testasyncgetprop.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/testasyncgetprop.c	2013-01-11 15:55:34.542171094 -0800
@@ -81,6 +81,12 @@ print_backtrace (void)
 }
 #endif
 
+#ifdef __GNUC__
+#define UNUSED_VARIABLE __attribute__ ((unused))
+#else
+#define UNUSED_VARIABLE
+#endif
+
 static int error_trap_depth = 0;
 
 static int
@@ -416,7 +422,7 @@ run_speed_comparison (Display *xdisplay,
       
       while ((task = ag_get_next_completed_task (xdisplay)))
         {
-          int result;
+          int UNUSED_VARIABLE result;
           Atom actual_type;
           int actual_format;
           unsigned long n_items;
diff -uNrp mate-window-manager-1.5.2/src/core/testboxes.c mate-window-manager/src/core/testboxes.c
--- mate-window-manager-1.5.2/src/core/testboxes.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/testboxes.c	2013-01-11 15:55:34.542171094 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include "boxes.h"
diff -uNrp mate-window-manager-1.5.2/src/core/util.c mate-window-manager/src/core/util.c
--- mate-window-manager-1.5.2/src/core/util.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/util.c	2013-01-11 15:55:34.542171094 -0800
@@ -18,8 +18,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #define _GNU_SOURCE
diff -uNrp mate-window-manager-1.5.2/src/core/window.c mate-window-manager/src/core/window.c
--- mate-window-manager-1.5.2/src/core/window.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/window.c	2013-01-11 15:55:34.544171118 -0800
@@ -20,8 +20,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
@@ -53,6 +53,12 @@
 #include <X11/extensions/shape.h>
 #endif
 
+#ifdef __GNUC__
+#define UNUSED_VARIABLE __attribute__ ((unused))
+#else
+#define UNUSED_VARIABLE
+#endif
+
 static int destroying_windows_disallowed = 0;
 
 
@@ -5204,7 +5210,7 @@ meta_window_client_message (MetaWindow *
   else if (event->xclient.message_type ==
            display->atom__NET_MOVERESIZE_WINDOW)
     {
-      int gravity, source;
+      int gravity, UNUSED_VARIABLE source;
       guint value_mask;
 
       gravity = (event->xclient.data.l[0] & 0xff);
@@ -5252,7 +5258,7 @@ meta_window_client_message (MetaWindow *
   else if (event->xclient.message_type ==
            display->atom__NET_WM_FULLSCREEN_MONITORS)
     {
-      MetaClientType source_indication;
+      MetaClientType UNUSED_VARIABLE source_indication;
       gulong top, bottom, left, right;
 
       meta_verbose ("_NET_WM_FULLSCREEN_MONITORS request for window '%s'\n",
@@ -5463,7 +5469,7 @@ static gboolean
 process_property_notify (MetaWindow     *window,
                          XPropertyEvent *event)
 {
-  Window xid = window->xwindow;
+  Window UNUSED_VARIABLE xid = window->xwindow;
 
   if (meta_is_verbose ()) /* avoid looking up the name if we don't have to */
     {
@@ -6705,7 +6711,7 @@ meta_window_titlebar_is_onscreen (MetaWi
 {
   MetaRectangle  titlebar_rect;
   GList         *onscreen_region;
-  int            titlebar_size;
+  int            UNUSED_VARIABLE titlebar_size;
   gboolean       is_onscreen;
 
   const int min_height_needed  = 8;
diff -uNrp mate-window-manager-1.5.2/src/core/window-private.h mate-window-manager/src/core/window-private.h
--- mate-window-manager-1.5.2/src/core/window-private.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/window-private.h	2013-01-11 15:55:34.542171094 -0800
@@ -27,8 +27,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_WINDOW_PRIVATE_H
diff -uNrp mate-window-manager-1.5.2/src/core/window-props.c mate-window-manager/src/core/window-props.c
--- mate-window-manager-1.5.2/src/core/window-props.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/window-props.c	2013-01-11 15:55:34.543171106 -0800
@@ -31,8 +31,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #define _GNU_SOURCE
diff -uNrp mate-window-manager-1.5.2/src/core/window-props.h mate-window-manager/src/core/window-props.h
--- mate-window-manager-1.5.2/src/core/window-props.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/window-props.h	2013-01-11 15:55:34.543171106 -0800
@@ -25,8 +25,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_WINDOW_PROPS_H
diff -uNrp mate-window-manager-1.5.2/src/core/workspace.c mate-window-manager/src/core/workspace.c
--- mate-window-manager-1.5.2/src/core/workspace.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/workspace.c	2013-01-11 15:55:34.545171130 -0800
@@ -19,8 +19,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/core/workspace.h mate-window-manager/src/core/workspace.h
--- mate-window-manager-1.5.2/src/core/workspace.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/workspace.h	2013-01-11 15:55:34.545171130 -0800
@@ -26,8 +26,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_WORKSPACE_H
diff -uNrp mate-window-manager-1.5.2/src/core/xprops.c mate-window-manager/src/core/xprops.c
--- mate-window-manager-1.5.2/src/core/xprops.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/core/xprops.c	2013-01-11 15:55:34.545171130 -0800
@@ -23,8 +23,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 /***********************************************************
diff -uNrp mate-window-manager-1.5.2/src/include/all-keybindings.h mate-window-manager/src/include/all-keybindings.h
--- mate-window-manager-1.5.2/src/include/all-keybindings.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/all-keybindings.h	2013-01-11 15:55:34.545171130 -0800
@@ -15,8 +15,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 /**
diff -uNrp mate-window-manager-1.5.2/src/include/boxes.h mate-window-manager/src/include/boxes.h
--- mate-window-manager-1.5.2/src/include/boxes.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/boxes.h	2013-01-11 15:55:34.545171130 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_BOXES_H
diff -uNrp mate-window-manager-1.5.2/src/include/common.h mate-window-manager/src/include/common.h
--- mate-window-manager-1.5.2/src/include/common.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/common.h	2013-01-11 15:55:34.545171130 -0800
@@ -18,8 +18,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_COMMON_H
diff -uNrp mate-window-manager-1.5.2/src/include/compositor.h mate-window-manager/src/include/compositor.h
--- mate-window-manager-1.5.2/src/include/compositor.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/compositor.h	2013-01-11 15:55:34.545171130 -0800
@@ -15,8 +15,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_COMPOSITOR_H
diff -uNrp mate-window-manager-1.5.2/src/include/core.h mate-window-manager/src/include/core.h
--- mate-window-manager-1.5.2/src/include/core.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/core.h	2013-01-11 15:55:34.545171130 -0800
@@ -18,8 +18,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_CORE_H
diff -uNrp mate-window-manager-1.5.2/src/include/display.h mate-window-manager/src/include/display.h
--- mate-window-manager-1.5.2/src/include/display.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/display.h	2013-01-11 15:55:34.545171130 -0800
@@ -15,8 +15,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_DISPLAY_H
diff -uNrp mate-window-manager-1.5.2/src/include/errors.h mate-window-manager/src/include/errors.h
--- mate-window-manager-1.5.2/src/include/errors.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/errors.h	2013-01-11 15:55:34.546171142 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_ERRORS_H
diff -uNrp mate-window-manager-1.5.2/src/include/frame.h mate-window-manager/src/include/frame.h
--- mate-window-manager-1.5.2/src/include/frame.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/frame.h	2013-01-11 15:55:34.546171142 -0800
@@ -15,8 +15,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_FRAME_H
diff -uNrp mate-window-manager-1.5.2/src/include/main.h mate-window-manager/src/include/main.h
--- mate-window-manager-1.5.2/src/include/main.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/main.h	2013-01-11 15:55:34.546171142 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_MAIN_H
diff -uNrp mate-window-manager-1.5.2/src/include/prefs.h mate-window-manager/src/include/prefs.h
--- mate-window-manager-1.5.2/src/include/prefs.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/prefs.h	2013-01-11 15:55:34.546171142 -0800
@@ -18,8 +18,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_PREFS_H
diff -uNrp mate-window-manager-1.5.2/src/include/resizepopup.h mate-window-manager/src/include/resizepopup.h
--- mate-window-manager-1.5.2/src/include/resizepopup.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/resizepopup.h	2013-01-11 15:55:34.546171142 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_RESIZEPOPUP_H
diff -uNrp mate-window-manager-1.5.2/src/include/screen.h mate-window-manager/src/include/screen.h
--- mate-window-manager-1.5.2/src/include/screen.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/screen.h	2013-01-11 15:55:34.546171142 -0800
@@ -15,8 +15,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_SCREEN_H
diff -uNrp mate-window-manager-1.5.2/src/include/tabpopup.h mate-window-manager/src/include/tabpopup.h
--- mate-window-manager-1.5.2/src/include/tabpopup.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/tabpopup.h	2013-01-11 15:55:34.546171142 -0800
@@ -18,8 +18,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_TABPOPUP_H
diff -uNrp mate-window-manager-1.5.2/src/include/types.h mate-window-manager/src/include/types.h
--- mate-window-manager-1.5.2/src/include/types.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/types.h	2013-01-11 15:55:34.546171142 -0800
@@ -15,8 +15,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_TYPES_H
diff -uNrp mate-window-manager-1.5.2/src/include/ui.h mate-window-manager/src/include/ui.h
--- mate-window-manager-1.5.2/src/include/ui.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/ui.h	2013-01-11 15:55:34.546171142 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_UI_H
diff -uNrp mate-window-manager-1.5.2/src/include/util.h mate-window-manager/src/include/util.h
--- mate-window-manager-1.5.2/src/include/util.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/util.h	2013-01-11 15:55:34.546171142 -0800
@@ -18,8 +18,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_UTIL_H
diff -uNrp mate-window-manager-1.5.2/src/include/window.h mate-window-manager/src/include/window.h
--- mate-window-manager-1.5.2/src/include/window.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/window.h	2013-01-11 15:55:34.546171142 -0800
@@ -15,8 +15,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_WINDOW_H
diff -uNrp mate-window-manager-1.5.2/src/include/xprops.h mate-window-manager/src/include/xprops.h
--- mate-window-manager-1.5.2/src/include/xprops.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/include/xprops.h	2013-01-11 15:55:34.546171142 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_XPROPS_H
diff -uNrp mate-window-manager-1.5.2/src/Makefile.am mate-window-manager/src/Makefile.am
--- mate-window-manager-1.5.2/src/Makefile.am	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/Makefile.am	2013-01-11 15:55:34.534170995 -0800
@@ -175,6 +175,9 @@ gsettings_SCHEMAS = org.mate.marco.gsche
 
 @INTLTOOL_XML_NOMERGE_RULE@
 
+convertdir = $(datadir)/MateConf/gsettings
+convert_DATA = marco.convert
+
 xmldir = @MATE_KEYBINDINGS_KEYSDIR@
 xml_in_files = \
 	50-marco-desktop-key.xml.in \
@@ -197,6 +200,7 @@ EXTRA_DIST=$(desktopfiles_files) \
 	$(wmproperties_files) \
 	$(IMAGES) \
 	$(gsettings_SCHEMAS) \
+	$(convert_DATA) \
 	$(desktopfiles_in_files) \
 	$(wmproperties_in_files) \
 	$(xml_in_files) \
diff -uNrp mate-window-manager-1.5.2/src/marco.convert mate-window-manager/src/marco.convert
--- mate-window-manager-1.5.2/src/marco.convert	1969-12-31 16:00:00.000000000 -0800
+++ mate-window-manager/src/marco.convert	2013-01-11 15:55:34.546171142 -0800
@@ -0,0 +1,153 @@
+[org.mate.Marco.general]
+mouse-button-modifier = /apps/marco/general/mouse_button_modifier
+resize-with-right-button = /apps/marco/general/resize_with_right_button
+button-layout = /apps/marco/general/button_layout
+focus-mode = /apps/marco/general/focus_mode
+focus-new-windows = /apps/marco/general/focus_new_windows
+raise-on-click = /apps/marco/general/raise_on_click
+action-double-click-titlebar = /apps/marco/general/action_double_click_titlebar
+action-middle-click-titlebar = /apps/marco/general/action_middle_click_titlebar
+action-right-click-titlebar = /apps/marco/general/action_right_click_titlebar
+auto-raise = /apps/marco/general/auto_raise
+auto-raise-delay = /apps/marco/general/auto_raise_delay
+theme = /apps/marco/general/theme
+titlebar-uses-system-font = /apps/marco/general/titlebar_uses_system_font
+titlebar-font = /apps/marco/general/titlebar_font
+num-workspaces = /apps/marco/general/num_workspaces
+wrap-style = /apps/marco/general/wrap_style
+visual-bell = /apps/marco/general/visual_bell
+audible-bell = /apps/marco/general/audible_bell
+visual-bell-type = /apps/marco/general/visual_bell_type
+compositing-manager = /apps/marco/general/compositing_manager
+compositing-fast-alt-tab = /apps/marco/general/compositing_fast_alt_tab
+reduced-resources = /apps/marco/general/reduced_resources
+application-based = /apps/marco/general/application_based
+disable-workarounds = /apps/marco/general/disable_workarounds
+
+[org.mate.Marco.workspace-names]
+name-1 = /apps/marco/workspace_names/name_1
+name-2 = /apps/marco/workspace_names/name_2
+name-3 = /apps/marco/workspace_names/name_3
+name-4 = /apps/marco/workspace_names/name_4
+name-5 = /apps/marco/workspace_names/name_5
+name-6 = /apps/marco/workspace_names/name_6
+name-7 = /apps/marco/workspace_names/name_7
+name-8 = /apps/marco/workspace_names/name_8
+name-9 = /apps/marco/workspace_names/name_9
+name-10 = /apps/marco/workspace_names/name_10
+name-11 = /apps/marco/workspace_names/name_11
+name-12 = /apps/marco/workspace_names/name_12
+name-13 = /apps/marco/workspace_names/name_13
+name-14 = /apps/marco/workspace_names/name_14
+name-15 = /apps/marco/workspace_names/name_15
+name-16 = /apps/marco/workspace_names/name_16
+
+[org.mate.Marco.global-keybindings]
+run-command-1 = /apps/marco/global_keybindings/run_command_1
+run-command-2 = /apps/marco/global_keybindings/run_command_2
+run-command-3 = /apps/marco/global_keybindings/run_command_3
+run-command-4 = /apps/marco/global_keybindings/run_command_4
+run-command-5 = /apps/marco/global_keybindings/run_command_5
+run-command-6 = /apps/marco/global_keybindings/run_command_6
+run-command-7 = /apps/marco/global_keybindings/run_command_7
+run-command-8 = /apps/marco/global_keybindings/run_command_8
+run-command-9 = /apps/marco/global_keybindings/run_command_9
+run-command-10 = /apps/marco/global_keybindings/run_command_10
+run-command-11 = /apps/marco/global_keybindings/run_command_11
+run-command-12 = /apps/marco/global_keybindings/run_command_12
+switch-to-workspace-1 = /apps/marco/global_keybindings/switch_to_workspace_1
+switch-to-workspace-2 = /apps/marco/global_keybindings/switch_to_workspace_2
+switch-to-workspace-3 = /apps/marco/global_keybindings/switch_to_workspace_3
+switch-to-workspace-4 = /apps/marco/global_keybindings/switch_to_workspace_4
+switch-to-workspace-5 = /apps/marco/global_keybindings/switch_to_workspace_5
+switch-to-workspace-6 = /apps/marco/global_keybindings/switch_to_workspace_6
+switch-to-workspace-7 = /apps/marco/global_keybindings/switch_to_workspace_7
+switch-to-workspace-8 = /apps/marco/global_keybindings/switch_to_workspace_8
+switch-to-workspace-9 = /apps/marco/global_keybindings/switch_to_workspace_9
+switch-to-workspace-10 = /apps/marco/global_keybindings/switch_to_workspace_10
+switch-to-workspace-11 = /apps/marco/global_keybindings/switch_to_workspace_11
+switch-to-workspace-12 = /apps/marco/global_keybindings/switch_to_workspace_12
+switch-to-workspace-left = /apps/marco/global_keybindings/switch_to_workspace_left
+switch-to-workspace-right = /apps/marco/global_keybindings/switch_to_workspace_right
+switch-to-workspace-up = /apps/marco/global_keybindings/switch_to_workspace_up
+switch-to-workspace-down = /apps/marco/global_keybindings/switch_to_workspace_down
+switch-group = /apps/marco/global_keybindings/switch_group
+switch-group-backward = /apps/marco/global_keybindings/switch_group_backward
+switch-windows = /apps/marco/global_keybindings/switch_windows
+switch-windows-backward = /apps/marco/global_keybindings/switch_windows_backward
+switch-windows-all = /apps/marco/global_keybindings/switch_windows_all
+switch-windows-all-backward = /apps/marco/global_keybindings/switch_windows_all_backward
+switch-panels = /apps/marco/global_keybindings/switch_panels
+switch-panels-backward = /apps/marco/global_keybindings/switch_panels_backward
+cycle-group = /apps/marco/global_keybindings/cycle_group
+cycle-group-backward = /apps/marco/global_keybindings/cycle_group_backward
+cycle-windows = /apps/marco/global_keybindings/cycle_windows
+cycle-windows-backward = /apps/marco/global_keybindings/cycle_windows_backward
+cycle-panels = /apps/marco/global_keybindings/cycle_panels
+cycle-panels-backward = /apps/marco/global_keybindings/cycle_panels_backward
+show-desktop = /apps/marco/global_keybindings/show_desktop
+panel-main-menu = /apps/marco/global_keybindings/panel_main_menu
+panel-run-dialog = /apps/marco/global_keybindings/panel_run_dialog
+run-command-screenshot = /apps/marco/global_keybindings/run_command_screenshot
+run-command-window-screenshot = /apps/marco/global_keybindings/run_command_window_screenshot
+run-command-terminal = /apps/marco/global_keybindings/run_command_terminal
+
+[org.mate.Marco.keybinding-commands]
+command-1 = /apps/marco/keybinding_commands/command_1
+command-2 = /apps/marco/keybinding_commands/command_2
+command-3 = /apps/marco/keybinding_commands/command_3
+command-4 = /apps/marco/keybinding_commands/command_4
+command-5 = /apps/marco/keybinding_commands/command_5
+command-6 = /apps/marco/keybinding_commands/command_6
+command-7 = /apps/marco/keybinding_commands/command_7
+command-8 = /apps/marco/keybinding_commands/command_8
+command-9 = /apps/marco/keybinding_commands/command_9
+command-10 = /apps/marco/keybinding_commands/command_10
+command-11 = /apps/marco/keybinding_commands/command_11
+command-12 = /apps/marco/keybinding_commands/command_12
+command-screenshot = /apps/marco/keybinding_commands/command_screenshot
+command-window-screenshot = /apps/marco/keybinding_commands/command_window_screenshot
+
+[org.mate.Marco.window-keybindings]
+activate-window-menu = /apps/marco/window_keybindings/activate_window_menu
+toggle-fullscreen = /apps/marco/window_keybindings/toggle_fullscreen
+toggle-maximized = /apps/marco/window_keybindings/toggle_maximized
+toggle-above = /apps/marco/window_keybindings/toggle_above
+maximize = /apps/marco/window_keybindings/maximize
+unmaximize = /apps/marco/window_keybindings/unmaximize
+toggle-shaded = /apps/marco/window_keybindings/toggle_shaded
+minimize = /apps/marco/window_keybindings/minimize
+close = /apps/marco/window_keybindings/close
+begin-move = /apps/marco/window_keybindings/begin_move
+begin-resize = /apps/marco/window_keybindings/begin_resize
+toggle-on-all-workspaces = /apps/marco/window_keybindings/toggle_on_all_workspaces
+move-to-workspace-1 = /apps/marco/window_keybindings/move_to_workspace_1
+move-to-workspace-2 = /apps/marco/window_keybindings/move_to_workspace_2
+move-to-workspace-3 = /apps/marco/window_keybindings/move_to_workspace_3
+move-to-workspace-4 = /apps/marco/window_keybindings/move_to_workspace_4
+move-to-workspace-5 = /apps/marco/window_keybindings/move_to_workspace_5
+move-to-workspace-6 = /apps/marco/window_keybindings/move_to_workspace_6
+move-to-workspace-7 = /apps/marco/window_keybindings/move_to_workspace_7
+move-to-workspace-8 = /apps/marco/window_keybindings/move_to_workspace_8
+move-to-workspace-9 = /apps/marco/window_keybindings/move_to_workspace_9
+move-to-workspace-10 = /apps/marco/window_keybindings/move_to_workspace_10
+move-to-workspace-11 = /apps/marco/window_keybindings/move_to_workspace_11
+move-to-workspace-12 = /apps/marco/window_keybindings/move_to_workspace_12
+move-to-workspace-left = /apps/marco/window_keybindings/move_to_workspace_left
+move-to-workspace-right = /apps/marco/window_keybindings/move_to_workspace_right
+move-to-workspace-up = /apps/marco/window_keybindings/move_to_workspace_up
+move-to-workspace-down = /apps/marco/window_keybindings/move_to_workspace_down
+raise-or-lower = /apps/marco/window_keybindings/raise_or_lower
+raise = /apps/marco/window_keybindings/raise
+lower = /apps/marco/window_keybindings/lower
+maximize-vertically = /apps/marco/window_keybindings/maximize_vertically
+maximize-horizontally = /apps/marco/window_keybindings/maximize_horizontally
+move-to-corner-nw = /apps/marco/window_keybindings/move_to_corner_nw
+move-to-corner-ne = /apps/marco/window_keybindings/move_to_corner_ne
+move-to-corner-sw = /apps/marco/window_keybindings/move_to_corner_sw
+move-to-corner-se = /apps/marco/window_keybindings/move_to_corner_se
+move-to-side-n = /apps/marco/window_keybindings/move_to_side_n
+move-to-side-s = /apps/marco/window_keybindings/move_to_side_s
+move-to-side-e = /apps/marco/window_keybindings/move_to_side_e
+move-to-side-w = /apps/marco/window_keybindings/move_to_side_w
+move-to-center = /apps/marco/window_keybindings/move_to_center
diff -uNrp mate-window-manager-1.5.2/src/org.mate.marco.gschema.xml mate-window-manager/src/org.mate.marco.gschema.xml
--- mate-window-manager-1.5.2/src/org.mate.marco.gschema.xml	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/org.mate.marco.gschema.xml	2013-01-11 15:55:34.546171142 -0800
@@ -111,6 +111,7 @@
       <description>A font description string describing a font for window titlebars. The size from the description will only be used if the titlebar_font_size option is set to 0. Also, this option is disabled if the titlebar_uses_desktop_font option is set to true.</description>
     </key>
     <key name="num-workspaces" type="i">
+      <range min="1" max="16"/>
       <default>4</default>
       <summary>Number of workspaces</summary>
       <description>Number of workspaces. Must be more than zero, and has a fixed maximum to prevent making the desktop unusable by accidentally asking for too many workspaces.</description>
diff -uNrp mate-window-manager-1.5.2/src/tools/marco-grayscale.c mate-window-manager/src/tools/marco-grayscale.c
--- mate-window-manager-1.5.2/src/tools/marco-grayscale.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/tools/marco-grayscale.c	2013-01-11 15:55:34.550171190 -0800
@@ -15,8 +15,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <gdk-pixbuf/gdk-pixbuf.h>
diff -uNrp mate-window-manager-1.5.2/src/tools/marco-mag.c mate-window-manager/src/tools/marco-mag.c
--- mate-window-manager-1.5.2/src/tools/marco-mag.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/tools/marco-mag.c	2013-01-11 15:55:34.550171190 -0800
@@ -15,8 +15,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #define _GNU_SOURCE
diff -uNrp mate-window-manager-1.5.2/src/tools/marco-message.c mate-window-manager/src/tools/marco-message.c
--- mate-window-manager-1.5.2/src/tools/marco-message.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/tools/marco-message.c	2013-01-11 15:55:34.550171190 -0800
@@ -15,8 +15,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/tools/marco-window-demo.c mate-window-manager/src/tools/marco-window-demo.c
--- mate-window-manager-1.5.2/src/tools/marco-window-demo.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/tools/marco-window-demo.c	2013-01-11 15:55:34.551171202 -0800
@@ -15,8 +15,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <gtk/gtk.h>
diff -uNrp mate-window-manager-1.5.2/src/ui/draw-workspace.c mate-window-manager/src/ui/draw-workspace.c
--- mate-window-manager-1.5.2/src/ui/draw-workspace.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/draw-workspace.c	2013-01-11 15:55:34.551171202 -0800
@@ -21,8 +21,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include "draw-workspace.h"
diff -uNrp mate-window-manager-1.5.2/src/ui/draw-workspace.h mate-window-manager/src/ui/draw-workspace.h
--- mate-window-manager-1.5.2/src/ui/draw-workspace.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/draw-workspace.h	2013-01-11 15:55:34.551171202 -0800
@@ -21,8 +21,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef WNCK_DRAW_WORKSPACE_H
diff -uNrp mate-window-manager-1.5.2/src/ui/fixedtip.c mate-window-manager/src/ui/fixedtip.c
--- mate-window-manager-1.5.2/src/ui/fixedtip.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/fixedtip.c	2013-01-11 15:55:34.551171202 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/ui/fixedtip.h mate-window-manager/src/ui/fixedtip.h
--- mate-window-manager-1.5.2/src/ui/fixedtip.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/fixedtip.h	2013-01-11 15:55:34.551171202 -0800
@@ -15,8 +15,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 /**
diff -uNrp mate-window-manager-1.5.2/src/ui/frames.c mate-window-manager/src/ui/frames.c
--- mate-window-manager-1.5.2/src/ui/frames.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/frames.c	2013-01-11 15:55:34.551171202 -0800
@@ -19,8 +19,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/ui/frames.h mate-window-manager/src/ui/frames.h
--- mate-window-manager-1.5.2/src/ui/frames.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/frames.h	2013-01-11 15:55:34.551171202 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_FRAMES_H
diff -uNrp mate-window-manager-1.5.2/src/ui/gradient.c mate-window-manager/src/ui/gradient.c
--- mate-window-manager-1.5.2/src/ui/gradient.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/gradient.c	2013-01-11 15:55:34.552171215 -0800
@@ -19,8 +19,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.  */
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.  */
 
 #include "gradient.h"
 #include "util.h"
diff -uNrp mate-window-manager-1.5.2/src/ui/gradient.h mate-window-manager/src/ui/gradient.h
--- mate-window-manager-1.5.2/src/ui/gradient.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/gradient.h	2013-01-11 15:55:34.552171215 -0800
@@ -18,8 +18,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.  */
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.  */
 
 #ifndef META_GRADIENT_H
 #define META_GRADIENT_H
diff -uNrp mate-window-manager-1.5.2/src/ui/menu.c mate-window-manager/src/ui/menu.c
--- mate-window-manager-1.5.2/src/ui/menu.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/menu.c	2013-01-11 15:55:34.552171215 -0800
@@ -17,8 +17,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/ui/menu.h mate-window-manager/src/ui/menu.h
--- mate-window-manager-1.5.2/src/ui/menu.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/menu.h	2013-01-11 15:55:34.552171215 -0800
@@ -15,8 +15,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_MENU_H
diff -uNrp mate-window-manager-1.5.2/src/ui/metaaccellabel.c mate-window-manager/src/ui/metaaccellabel.c
--- mate-window-manager-1.5.2/src/ui/metaaccellabel.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/metaaccellabel.c	2013-01-11 15:55:34.552171215 -0800
@@ -20,8 +20,8 @@
  *
  * You should have received a copy of the GNU Lesser General Public
  * License along with this library; if not, write to the
- * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
- * Boston, MA 02111-1307, USA.
+ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
  */
 
 /*
diff -uNrp mate-window-manager-1.5.2/src/ui/metaaccellabel.h mate-window-manager/src/ui/metaaccellabel.h
--- mate-window-manager-1.5.2/src/ui/metaaccellabel.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/metaaccellabel.h	2013-01-11 15:55:34.552171215 -0800
@@ -20,8 +20,8 @@
  *
  * You should have received a copy of the GNU Lesser General Public
  * License along with this library; if not, write to the
- * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
- * Boston, MA 02111-1307, USA.
+ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
  */
 
 /*
diff -uNrp mate-window-manager-1.5.2/src/ui/preview-widget.c mate-window-manager/src/ui/preview-widget.c
--- mate-window-manager-1.5.2/src/ui/preview-widget.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/preview-widget.c	2013-01-11 15:55:34.552171215 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #define _GNU_SOURCE
diff -uNrp mate-window-manager-1.5.2/src/ui/preview-widget.h mate-window-manager/src/ui/preview-widget.h
--- mate-window-manager-1.5.2/src/ui/preview-widget.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/preview-widget.h	2013-01-11 15:55:34.552171215 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include "theme.h"
diff -uNrp mate-window-manager-1.5.2/src/ui/resizepopup.c mate-window-manager/src/ui/resizepopup.c
--- mate-window-manager-1.5.2/src/ui/resizepopup.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/resizepopup.c	2013-01-11 15:55:34.552171215 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/ui/tabpopup.c mate-window-manager/src/ui/tabpopup.c
--- mate-window-manager-1.5.2/src/ui/tabpopup.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/tabpopup.c	2013-01-11 15:55:34.552171215 -0800
@@ -19,8 +19,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/ui/testgradient.c mate-window-manager/src/ui/testgradient.c
--- mate-window-manager-1.5.2/src/ui/testgradient.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/testgradient.c	2013-01-11 15:55:34.552171215 -0800
@@ -17,12 +17,18 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.  */
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.  */
 
 #include "gradient.h"
 #include <gtk/gtk.h>
 
+#ifdef __GNUC__
+#define UNUSED_VARIABLE __attribute__ ((unused))
+#else
+#define UNUSED_VARIABLE
+#endif
+
 typedef void (* RenderGradientFunc) (GdkDrawable *drawable,
                                      cairo_t     *cr,
                                      int          width,
@@ -294,7 +300,7 @@ create_gradient_window (const char *titl
 static void
 meta_gradient_test (void)
 {
-  GtkWidget *window;
+  GtkWidget UNUSED_VARIABLE *window;
 
   window = create_gradient_window ("Simple vertical",
                                    render_vertical_func);
diff -uNrp mate-window-manager-1.5.2/src/ui/theme.c mate-window-manager/src/ui/theme.c
--- mate-window-manager-1.5.2/src/ui/theme.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/theme.c	2013-01-11 15:55:34.573171468 -0800
@@ -17,8 +17,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 /**
diff -uNrp mate-window-manager-1.5.2/src/ui/theme.h mate-window-manager/src/ui/theme.h
--- mate-window-manager-1.5.2/src/ui/theme.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/theme.h	2013-01-11 15:55:34.573171468 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #ifndef META_THEME_H
diff -uNrp mate-window-manager-1.5.2/src/ui/theme-parser.c mate-window-manager/src/ui/theme-parser.c
--- mate-window-manager-1.5.2/src/ui/theme-parser.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/theme-parser.c	2013-01-11 15:55:34.553171228 -0800
@@ -17,8 +17,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/ui/theme-parser.h mate-window-manager/src/ui/theme-parser.h
--- mate-window-manager-1.5.2/src/ui/theme-parser.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/theme-parser.h	2013-01-11 15:55:34.553171228 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include "theme.h"
diff -uNrp mate-window-manager-1.5.2/src/ui/theme-viewer.c mate-window-manager/src/ui/theme-viewer.c
--- mate-window-manager-1.5.2/src/ui/theme-viewer.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/theme-viewer.c	2013-01-11 15:55:34.553171228 -0800
@@ -17,8 +17,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <config.h>
diff -uNrp mate-window-manager-1.5.2/src/ui/themewidget.c mate-window-manager/src/ui/themewidget.c
--- mate-window-manager-1.5.2/src/ui/themewidget.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/themewidget.c	2013-01-11 15:55:34.574171479 -0800
@@ -17,13 +17,19 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include "themewidget.h"
 #include <math.h>
 
+#ifdef __GNUC__
+#define UNUSED_VARIABLE __attribute__ ((unused))
+#else
+#define UNUSED_VARIABLE
+#endif
+
 static void meta_area_class_init   (MetaAreaClass  *klass);
 static void meta_area_init         (MetaArea       *area);
 static void meta_area_size_request (GtkWidget      *widget,
@@ -64,7 +70,7 @@ static void
 meta_area_class_init (MetaAreaClass *class)
 {
   GObjectClass *gobject_class = G_OBJECT_CLASS (class);
-  GtkObjectClass *object_class;
+  GtkObjectClass UNUSED_VARIABLE *object_class;
   GtkWidgetClass *widget_class;
 
   object_class = (GtkObjectClass*) class;
diff -uNrp mate-window-manager-1.5.2/src/ui/themewidget.h mate-window-manager/src/ui/themewidget.h
--- mate-window-manager-1.5.2/src/ui/themewidget.h	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/themewidget.h	2013-01-11 15:55:34.574171479 -0800
@@ -17,8 +17,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include "theme.h"
diff -uNrp mate-window-manager-1.5.2/src/ui/ui.c mate-window-manager/src/ui/ui.c
--- mate-window-manager-1.5.2/src/ui/ui.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/ui/ui.c	2013-01-11 15:55:34.574171479 -0800
@@ -16,8 +16,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include "prefs.h"
diff -uNrp mate-window-manager-1.5.2/src/wm-tester/main.c mate-window-manager/src/wm-tester/main.c
--- mate-window-manager-1.5.2/src/wm-tester/main.c	2012-10-27 07:55:51.000000000 -0700
+++ mate-window-manager/src/wm-tester/main.c	2013-01-11 15:55:34.574171479 -0800
@@ -15,8 +15,8 @@
  * 
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
  */
 
 #include <gtk/gtk.h>
