Index: bitlbee-3.0.3/doc/bitlbee.xinetd
===================================================================
--- bitlbee-3.0.3.orig/doc/bitlbee.xinetd
+++ bitlbee-3.0.3/doc/bitlbee.xinetd
@@ -10,15 +10,16 @@ service ircd
 	wait            = no
 	
 	## You most likely want to change these two
-	user            = nobody
-	server          = /usr/local/sbin/bitlbee -I
+	user            = bitlbee
+	server          = /usr/sbin/bitlbee
 	
 	## You might want to limit access to localhost only:
-	# bind            = 127.0.0.1
+	bind            = 127.0.0.1
 	
 	## Thanks a lot to friedman@splode.com for telling us about the type
 	## argument, so now this file can be used without having to edit
 	## /etc/services too.
 	type            = UNLISTED
 	port            = 6667
+	disable         = yes
 }
