--- data/pcb.desktop.in.orig	2008-01-11 18:28:42.000000000 +0200
+++ data/pcb.desktop.in	2008-07-11 11:17:56.000000000 +0300
@@ -3,9 +3,10 @@
 Encoding=UTF-8
 _Name=PCB Designer
 _GenericName=PCB Design
+GenericName[tr]=Baskı Devre Tasarlayıcısı
 _Comment=Create and edit printed circuit board designs
 Type=Application
 Exec=pcb %f
 Icon=pcb
 MimeType=application/x-pcb-layout;application/x-pcb-footprint;
-Categories=Engineering;Electronics;
+Categories=Application;Science;Engineering;Electronics;
