diff -ur libgsf-1.14.25-orig/python/Makefile.am libgsf-1.14.25/python/Makefile.am
--- libgsf-1.14.25-orig/python/Makefile.am	2011-12-08 01:05:18.000000000 +0200
+++ libgsf-1.14.25/python/Makefile.am	2013-02-23 18:17:40.346593878 +0200
@@ -14,7 +14,7 @@
 _gsfmodule_la_CFLAGS = 		\
 	-I$(top_srcdir)		\
 	$(LIBGSF_CFLAGS)
-_gsfmodule_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_gsf
+_gsfmodule_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_gsf `python-config --libs`
 _gsfmodule_la_LIBADD = $(top_builddir)/gsf/libgsf-1.la $(LIBGSF_LIBS)
 _gsfmodule_la_SOURCES = gsfmodule.c
 nodist__gsfmodule_la_SOURCES = gsf.c
