--- src/sna/sna_accel.c.orig	2014-03-19 16:32:58.000000000 +0200
+++ src/sna/sna_accel.c	2014-05-09 00:45:15.692783288 +0300
@@ -2496,7 +2496,10 @@
 		flags |= MOVE_INPLACE_HINT;
 	}
 
-	if (flags & MOVE_WHOLE_HINT && priv->gpu_damage == NULL)
+   if ((flags & (MOVE_WHOLE_HINT | MOVE_READ)) == (MOVE_WHOLE_HINT | MOVE_READ))
+       return _sna_pixmap_move_to_cpu(pixmap, flags);
+
+   if (flags & MOVE_WHOLE_HINT && priv->gpu_damage == NULL && priv->cpu_damage == NULL)
 		return _sna_pixmap_move_to_cpu(pixmap, flags);
 
 	if (priv->gpu_damage == NULL &&
@@ -2520,22 +2523,18 @@
 		       pixmap->drawable.height));
 		if (dx | dy)
 			RegionTranslate(region, -dx, -dy);
-		return _sna_pixmap_move_to_cpu(pixmap, flags);
+		return _sna_pixmap_move_to_cpu(pixmap, flags | MOVE_READ);
 	}
 
-	if ((flags & MOVE_READ) == 0 &&
-	    priv->gpu_damage &&
-	    region_subsumes_damage(region, priv->gpu_damage)) {
-		DBG(("%s: region [(%d, %d), (%d, %d)] subsumes damage [(%d,%d), (%d, %d)]\n",
+   if (flags & MOVE_WHOLE_HINT) {
+       DBG(("%s: region (%d, %d), (%d, %d) marked with WHOLE hint, pixmap %dx%d\n",
 		       __FUNCTION__,
 		       region->extents.x1,
 		       region->extents.y1,
 		       region->extents.x2,
 		       region->extents.y2,
-		       priv->gpu_damage->extents.x1,
-		       priv->gpu_damage->extents.y1,
-		       priv->gpu_damage->extents.x2,
-		       priv->gpu_damage->extents.y2));
+              pixmap->drawable.width,
+              pixmap->drawable.height));
 		if (dx | dy)
 			RegionTranslate(region, -dx, -dy);
 		return _sna_pixmap_move_to_cpu(pixmap, flags);
@@ -4454,17 +4453,19 @@
 	if (wedged(sna))
 		return false;
 
-	DBG(("%s: bo? %d, can map? %d\n", __FUNCTION__,
-	     priv->gpu_bo != NULL,
-	     priv->gpu_bo ? kgem_bo_can_map__cpu(&sna->kgem, priv->gpu_bo, true) : 0));
-
 	replaces = region->data == NULL &&
 		w >= pixmap->drawable.width &&
 		h >= pixmap->drawable.height;
+    DBG(("%s: bo? %d, can map? %d, replaces? %d\n", __FUNCTION__,
+       priv->gpu_bo != NULL,
+        priv->gpu_bo ? kgem_bo_can_map__cpu(&sna->kgem, priv->gpu_bo, true) : 0,
+        replaces));
+
 
 	if (priv->gpu_bo && (replaces || priv->gpu_bo->proxy)) {
 		DBG(("%s: discarding cached upload proxy\n", __FUNCTION__));
 		sna_pixmap_free_gpu(sna, priv);
+        replaces = true; /* Mark it all GPU damaged afterwards */
 	}
 	assert(priv->gpu_bo == NULL || priv->gpu_bo->proxy == NULL);
 
@@ -4599,9 +4600,12 @@
 		return true;
 
 	hint = MOVE_WRITE;
-	if (w == pixmap->drawable.width && h*stride > 4096)
+	   if (w == pixmap->drawable.width && (h+1)*stride > 65536) {
+       DBG(("%s: large upload (%d bytes), marking WHOLE_HINT\n",
+            __FUNCTION__, h*stride));
 		hint |= MOVE_WHOLE_HINT;
-
+   }
+ 
 	if (!sna_drawable_move_region_to_cpu(&pixmap->drawable, region, hint))
 		return false;
 
