--- ClanLib-0.6.5/Sources/Display/Display/Generic/blit_transparent.cpp.bounds	2001-09-02 04:24:43.000000000 +0200
+++ ClanLib-0.6.5/Sources/Display/Display/Generic/blit_transparent.cpp	2003-10-08 16:07:52.000000000 +0200
@@ -278,6 +278,7 @@ void SubBlitter_Transparent<T>::blt_nocl
 	unsigned int dest_pitch = target->get_pitch();
 
 	unsigned char *dest = (unsigned char *) target->get_data();
+	if (y == 0) y = 1;
 	dest += x*sizeof(T) + (y-1)*dest_pitch;
 
 //	T *dd;
