--- ClanLib-0.6.5/Sources/API/Network/world_template.h.orig	2007-08-28 21:52:38.000000000 +0100
+++ ClanLib-0.6.5/Sources/API/Network/world_template.h	2007-08-28 21:52:49.000000000 +0100
@@ -254,7 +254,7 @@
 	//: Update
 	void update(float delta_time)
 	{
-		std::list<GameObject *>::iterator it;
+		typename std::list<GameObject *>::iterator it;
 		std::list<GameObject *> remove_list;
 		for (it = objects.begin(); it != objects.end(); it++)
 		{
