diff -p -up ClanLib-0.6.5/Sources/MikMod/setupmikmod.cpp~ ClanLib-0.6.5/Sources/MikMod/setupmikmod.cpp
--- ClanLib-0.6.5/Sources/MikMod/setupmikmod.cpp~	2002-01-28 12:08:03.000000000 +0100
+++ ClanLib-0.6.5/Sources/MikMod/setupmikmod.cpp	2008-08-14 12:47:05.000000000 +0200
@@ -32,6 +32,7 @@ MDRIVER drv_clanlib = {
 	255,
 	255, 
 	"clanmikmod",
+	"", // CmdLineHelp
 	NULL, // Command line
 	CL_Streamed_MikModSample_Session::clanMikMod_IsPresent, // IsPresent
 	VC_SampleLoad,
