Index: tdb-1.1.5/tdb.mk
===================================================================
--- tdb-1.1.5.orig/tdb.mk
+++ tdb-1.1.5/tdb.mk
@@ -37,10 +37,9 @@ clean:: 
 build-python:: tdb.$(SHLIBEXT) 
 
 pytdb.o: $(tdbdir)/pytdb.c
-	$(CC) $(PICFLAG) -c $(tdbdir)/pytdb.c $(CFLAGS) `$(PYTHON_CONFIG) --cflags`
-
+	$(CC) $(CFLAGS) $(PICFLAG) -c $(tdbdir)/pytdb.c `$(PYTHON_CONFIG) --cflags`
 tdb.$(SHLIBEXT): libtdb.$(SHLIBEXT) pytdb.o
-	$(SHLD) $(SHLD_FLAGS) -o $@ pytdb.o -L. -ltdb `$(PYTHON_CONFIG) --ldflags`
+	$(SHLD) $(LDFLAGS) $(SHLD_FLAGS) pytdb.o `$(PYTHON_CONFIG) --ldflags` -L. -ltdb -o $@
 
 install:: installdirs installbin installheaders installlibs \
 		  $(PYTHON_INSTALL_TARGET)
