--- libedit-20090923-3.0/src/read.c.sigwinch	2009-09-23 22:04:26.000000000 +0100
+++ libedit-20090923-3.0/src/read.c	2010-03-19 20:47:46.000000000 +0000
@@ -301,6 +301,9 @@ read_char(EditLine *el, char *cp)
 			sig_set(el);
 			el_set(el, EL_REFRESH);
 			goto again;
+		} else if (el->el_signal->sig_no == SIGWINCH) {
+			sig_set(el);
+			goto again;
 		}
 		if (!tried && read__fixio(el->el_infd, errno) == 0)
 			tried = 1;
