Index: kdecore/localization/klocale_kde.cpp
===================================================================
--- kdecore/localization/klocale_kde.cpp
+++ kdecore/localization/klocale_kde.cpp	2011-03-08 10:06:00.070649049 +0100
@@ -315,6 +315,9 @@
         m_catalogNames.append(KCatalogName(QString::fromLatin1("kdeqt")));
         m_catalogNames.append(KCatalogName(QString::fromLatin1("solid_qt")));
         m_catalogNames.append(KCatalogName(QString::fromLatin1("kdecalendarsystems")));
+        // Catalogs added by Mandriva
+        KGlobal::dirs()->addResourceDir("locale", QString::fromLatin1("/usr/share/locale"));
+        m_catalogNames.append(KCatalogName(QString::fromLatin1("xdg-user-dirs")));
         m_numberOfSysCatalogs = m_catalogNames.size() - numberOfCatalogs;
 
         updateCatalogs(); // evaluate this for all languages
