diff -Naurp imap-2007e/src/osdep/unix/env_unix.c imap-2007e.oden/src/osdep/unix/env_unix.c
--- imap-2007e/src/osdep/unix/env_unix.c	2008-06-04 20:46:10.000000000 +0200
+++ imap-2007e.oden/src/osdep/unix/env_unix.c	2009-01-14 00:01:54.000000000 +0100
@@ -24,6 +24,7 @@
  * Last Edited:	15 May 2008
  */
 
+#include <time.h>
 #include <grp.h>
 #include <signal.h>
 #include <sys/wait.h>
diff -Naurp imap-2007e/src/osdep/unix/mh.c imap-2007e.oden/src/osdep/unix/mh.c
--- imap-2007e/src/osdep/unix/mh.c	2008-06-04 20:18:34.000000000 +0200
+++ imap-2007e.oden/src/osdep/unix/mh.c	2009-01-14 00:01:54.000000000 +0100
@@ -36,11 +36,11 @@ extern int errno;		/* just in case */
 #include <pwd.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <time.h>
 #include "misc.h"
 #include "dummy.h"
 #include "fdstring.h"
 
-
 /* Build parameters */
 
 #define MHINBOX "#mhinbox"	/* corresponds to namespace in env_unix.c */
diff -Naurp imap-2007e/src/osdep/unix/mx.c imap-2007e.oden/src/osdep/unix/mx.c
--- imap-2007e/src/osdep/unix/mx.c	2008-06-04 20:18:34.000000000 +0200
+++ imap-2007e.oden/src/osdep/unix/mx.c	2009-01-14 00:01:54.000000000 +0100
@@ -36,6 +36,7 @@ extern int errno;		/* just in case */
 #include <pwd.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <time.h>
 #include "misc.h"
 #include "dummy.h"
 #include "fdstring.h"
diff -Naurp imap-2007e/src/osdep/unix/news.c imap-2007e.oden/src/osdep/unix/news.c
--- imap-2007e/src/osdep/unix/news.c	2008-06-04 20:18:34.000000000 +0200
+++ imap-2007e.oden/src/osdep/unix/news.c	2009-01-14 00:01:54.000000000 +0100
@@ -35,6 +35,7 @@ extern int errno;		/* just in case */
 #include "osdep.h"
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <time.h>
 #include "misc.h"
 #include "newsrc.h"
 #include "fdstring.h"
diff -Naurp imap-2007e/src/osdep/unix/phile.c imap-2007e.oden/src/osdep/unix/phile.c
--- imap-2007e/src/osdep/unix/phile.c	2008-06-04 20:18:34.000000000 +0200
+++ imap-2007e.oden/src/osdep/unix/phile.c	2009-01-14 00:01:54.000000000 +0100
@@ -37,6 +37,7 @@ extern int errno;		/* just in case */
 #include <pwd.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <time.h>
 #include "rfc822.h"
 #include "misc.h"
 #include "dummy.h"
