--- src/osdep/unix/Makefile	2007-12-17 23:10:24.000000000 +0100
+++ src/osdep/unix/Makefile.oden	2008-05-18 19:06:50.000000000 +0200
@@ -42,7 +42,7 @@
 # -O3 seems to be safe, but empirical observation from our local expert
 # indicates that in some (many?) cases -O3 code runs slower than -O2.
 
-GCCOPTLEVEL= -O2
+GCCOPTLEVEL=
 
 
 # Try to have some consistency in GCC builds.  We want optimization, but we
@@ -96,19 +96,18 @@
 LOGINPW=std
 SIGTYPE=bsd
 CRXTYPE=std
-ACTIVEFILE=/usr/lib/news/active
-SPOOLDIR=/usr/spool
+ACTIVEFILE=/var/lib/news/active
+SPOOLDIR=/var/spool
 MAILSPOOL=$(SPOOLDIR)/mail
 NEWSSPOOL=$(SPOOLDIR)/news
-RSHPATH=/usr/ucb/rsh
+RSHPATH=/usr/bin/rsh
 MD5PWD=/etc/cram-md5.pwd
 # Tries one of the test alternatives below if not specified.
-LOCKPGM=
+#LOCKPGM=/etc/mlock
 # Test alternatives if LOCKPGM not specified
-LOCKPGM1=/usr/libexec/mlock
-LOCKPGM2=/usr/sbin/mlock
-LOCKPGM3=/etc/mlock
-
+#LOCKPGM1=/usr/libexec/mlock
+#LOCKPGM2=/usr/sbin/mlock
+#LOCKPGM3=/etc/mlock
 
 # Default formats for creating new mailboxes and for empty mailboxes in the
 # default namespace; must be set to the associated driver's prototype.

