--- src/osdep/unix/Makefile.gss	2006-08-31 03:32:16.000000000 +0200
+++ src/osdep/unix/Makefile.gss.oden	2006-10-26 16:40:37.000000000 +0200
@@ -26,7 +26,7 @@
 
 # Extended flags needed for additional authenticators.  You may need to modify.
 
-GSSDIR=/usr/local
+GSSDIR=/usr
 GSSINCLUDE=$(GSSDIR)/include
 GSSLIB=$(GSSDIR)/lib
 GSSCFLAGS= -I$(GSSINCLUDE) -DGSS_C_NT_HOSTBASED_SERVICE=gss_nt_service_name
