--- deluge-0.9.05/deluge/ui/gtkui/gtkui.py~	2008-08-02 17:06:11.000000000 +0200
+++ deluge-0.9.05/deluge/ui/gtkui/gtkui.py	2008-08-07 07:00:43.000000000 +0200
@@ -76,7 +76,7 @@
     "start_in_tray": False,
     "lock_tray": False,
     "tray_password": "",
-    "check_new_releases": True,
+    "check_new_releases": False,
     "default_load_path": None,
     "window_maximized": False,
     "window_x_pos": 0,
