diff -Nuar libgringotts-1.2.1.orig/docs/Makefile.in libgringotts-1.2.1/docs/Makefile.in
--- libgringotts-1.2.1.orig/docs/Makefile.in	2011-07-06 15:19:30.778000115 +0300
+++ libgringotts-1.2.1/docs/Makefile.in	2011-07-07 08:41:12.635000117 +0300
@@ -131,7 +131,7 @@
 target_alias = @target_alias@
 EXTRA_DIST = manual.htm
 
-libgringottsdocdir = $(datadir)/doc/libgringotts-@VERSION@
+libgringottsdocdir = $(datadir)/doc/libgringotts
 
 libgringottsdoc_DATA = manual.htm
 subdir = docs
diff -Nuar libgringotts-1.2.1.orig/Makefile.in libgringotts-1.2.1/Makefile.in
--- libgringotts-1.2.1.orig/Makefile.in	2011-07-06 15:19:30.764000115 +0300
+++ libgringotts-1.2.1/Makefile.in	2011-07-07 08:37:43.229000116 +0300
@@ -131,7 +131,7 @@
 target_alias = @target_alias@
 SUBDIRS = src docs
 
-libgringottsdocdir = $(datadir)/doc/libgringotts-@VERSION@
+libgringottsdocdir = $(datadir)/doc/libgringotts
 libgringottsdoc_DATA = TODO NEWS AUTHORS README COPYING ChangeLog
 
 EXTRA_DIST = libgringotts.spec.in libgringotts.pc.in $(libgringottsdoc_DATA)
