diff -up swig-2.0.3/Lib/guile/guile_gh_run.swg.rh706140 swig-2.0.3/Lib/guile/guile_gh_run.swg
--- swig-2.0.3/Lib/guile/guile_gh_run.swg.rh706140	2011-05-20 15:16:42.300658987 +0200
+++ swig-2.0.3/Lib/guile/guile_gh_run.swg	2011-05-20 15:17:08.609658991 +0200
@@ -15,7 +15,11 @@
 extern "C" {
 #endif
 
+#if SCM_MAJOR_VERSION >= 2
+typedef scm_t_subr swig_guile_proc;
+#else
 typedef SCM (*swig_guile_proc)();
+#endif
 
 #define SWIG_malloc(size) \
   SCM_MUST_MALLOC(size)
diff -up swig-2.0.3/Lib/guile/guile_scm_run.swg.rh706140 swig-2.0.3/Lib/guile/guile_scm_run.swg
--- swig-2.0.3/Lib/guile/guile_scm_run.swg.rh706140	2011-05-20 15:17:22.057658991 +0200
+++ swig-2.0.3/Lib/guile/guile_scm_run.swg	2011-05-20 15:17:46.183658997 +0200
@@ -12,7 +12,11 @@
 extern "C" {
 #endif
 
+#if SCM_MAJOR_VERSION >= 2
+typedef scm_t_subr swig_guile_proc;
+#else
 typedef SCM (*swig_guile_proc)();
+#endif
 typedef SCM (*guile_destructor)(SCM);
 
 typedef struct swig_guile_clientdata {
