--- SWIG-1.3.23/configure.in.pylib	2004-11-02 00:45:57.000000000 +0100
+++ SWIG-1.3.23/configure.in	2004-11-23 12:20:12.552926982 +0100
@@ -574,9 +574,11 @@
     
     # Set the include directory
 
+    PYLIB=`$PYTHON -c "import sys; print sys.path[[1]]"`
+
     AC_MSG_CHECKING(for Python header files)
     if test -r $PYPREFIX/include/$PYVERSION/Python.h; then
-        PYINCLUDE="-I$PYPREFIX/include/$PYVERSION -I$PYEPREFIX/$PYLIBDIR/$PYVERSION/config"
+        PYINCLUDE="-I$PYPREFIX/include/$PYVERSION -I$PYLIB/config"
     fi
     if test -z "$PYINCLUDE"; then
         if test -r $PYPREFIX/include/Py/Python.h; then
@@ -587,13 +589,6 @@
 
     # Set the library directory blindly.   This probably won't work with older versions
     AC_MSG_CHECKING(for Python library)
-    dirs="$PYVERSION/config $PYVERSION/$PYLIBDIR python/$PYLIBDIR"
-    for i in $dirs; do
-        if test -d $PYEPREFIX/$PYLIBDIR/$i; then
-           PYLIB="$PYEPREFIX/$PYLIBDIR/$i"
-           break
-        fi
-    done
     if test -z "$PYLIB"; then
         AC_MSG_RESULT(Not found)
     else
