diff -Naur xplanet-1.2.1-orig/configure xplanet-1.2.1/configure
--- xplanet-1.2.1-orig/configure	2009-12-04 10:34:21.766146687 +0200
+++ xplanet-1.2.1/configure	2009-12-04 10:34:36.868516259 +0200
@@ -7974,7 +7974,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
-#include <pnm.h>
+#include <netpbm/pnm.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (ac_try="$ac_compile"
diff -Naur xplanet-1.2.1-orig/src/libimage/pnm.c xplanet-1.2.1/src/libimage/pnm.c
--- xplanet-1.2.1-orig/src/libimage/pnm.c	2009-12-04 10:34:21.768391483 +0200
+++ xplanet-1.2.1/src/libimage/pnm.c	2009-12-04 10:35:13.152141234 +0200
@@ -21,7 +21,7 @@
 #include <stdio.h>
 #include <string.h>
 
-#include <pnm.h>
+#include <netpbm/pnm.h>
 
 int
 read_pnm(const char *filename, int *width, int *height, char **rgb)
diff -Naur xplanet-1.2.1-orig/src/libimage/WriteImage.cpp xplanet-1.2.1/src/libimage/WriteImage.cpp
--- xplanet-1.2.1-orig/src/libimage/WriteImage.cpp	2009-12-04 10:34:21.768391483 +0200
+++ xplanet-1.2.1/src/libimage/WriteImage.cpp	2009-12-04 10:35:00.145228715 +0200
@@ -31,7 +31,7 @@
 #endif
     
 #ifdef HAVE_LIBPNM
-#include <pnm.h>
+#include <netpbm/pnm.h>
     int
     write_pnm(FILE *outfile, int width, int height, unsigned char *rgb,
               int maxv, int format, int forceplain);
