Index: CImg.h
===================================================================
--- CImg.h.orig
+++ CImg.h
@@ -291,9 +291,9 @@ extern "C" {
 #define __STDC_CONSTANT_MACROS // ...or stdint.h doesn't define UINT64_C, needed for libavutil
 #endif
 extern "C" {
-#include "avformat.h"
-#include "avcodec.h"
-#include "swscale.h"
+#include <libavformat/avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libswscale/swscale.h>
 }
 #endif
 
@@ -319,7 +319,7 @@ extern "C" {
 // to get a builtin support of various image formats (PNG,JPEG,TIFF,...).
 // Using Magick++ is not mandatory.
 #ifdef cimg_use_magick
-#include "Magick++.h"
+#include <ImageMagick/Magick++.h>
 #endif
 
 // FFTW3 configuration.
@@ -358,11 +358,11 @@ extern "C" {
 //
 // OpenEXR can be used to read/write .exr file formats.
 #ifdef cimg_use_openexr
-#include "ImfRgbaFile.h"
-#include "ImfInputFile.h"
-#include "ImfChannelList.h"
-#include "ImfMatrixAttribute.h"
-#include "ImfArray.h"
+#include <OpenEXR/ImfRgbaFile.h>
+#include <OpenEXR/ImfInputFile.h>
+#include <OpenEXR/ImfChannelList.h>
+#include <OpenEXR/ImfMatrixAttribute.h>
+#include <OpenEXR/ImfArray.h>
 #endif
 
 // Lapack configuration.
