--- vmpk.pro.orig	2012-07-30 22:45:42.000000000 +0300
+++ vmpk.pro	2012-08-28 15:24:49.030348804 +0300
@@ -57,6 +57,7 @@
         translations/vmpk_es.qm \
         translations/vmpk_fr.qm \
         translations/vmpk_ru.qm \
+        translations/vmpk_tr.qm \
         translations/vmpk_sv.qm
     DEPLOYMENT += addFiles
 }
@@ -97,6 +98,7 @@
         data/help_fr.html \
         data/help_nl.html \
         data/help_ru.html \
+        data/help_tr.html \
         data/gmgsxg.ins \
         data/spanish.xml \
         data/german.xml \
@@ -111,6 +113,7 @@
         $$[QT_INSTALL_TRANSLATIONS]/qt_fr.qm \
         $$[QT_INSTALL_TRANSLATIONS]/qt_ru.qm \
         $$[QT_INSTALL_TRANSLATIONS]/qt_sv.qm \
+        $$[QT_INSTALL_TRANSLATIONS]/qt_tr.qm \
         #$$[QT_INSTALL_TRANSLATIONS]/qt_zh_CN.qm \
         vmpk_cs.qm \
         vmpk_de.qm \
@@ -217,6 +220,7 @@
     translations/vmpk_fr.ts \
     #translations/vmpk_nl.ts \
     translations/vmpk_ru.ts \
+    translations/vmpk_tr.ts \
     translations/vmpk_sv.ts
     #translations/vmpk_zh_CN.ts
 
