--- kde-baseapps-4.8.1/dolphin/src/settings/dolphin_generalsettings.kcfg.orig	2011-12-02 23:17:41.000000000 +0200
+++ kde-baseapps-4.8.1/dolphin/src/settings/dolphin_generalsettings.kcfg	2012-03-09 16:42:06.934684169 +0200
@@ -38,7 +38,7 @@
         </entry>
         <entry name="FilterBar" type="Bool">
             <label>Should the filter bar be shown</label>
-            <default>false</default>
+            <default>true</default>
         </entry>
         <entry name="GlobalViewProps" type="Bool">
             <label>Should the view properties be used for all directories</label>
@@ -50,7 +50,7 @@
         </entry>
         <entry name="ConfirmClosingMultipleTabs" type="Bool">
             <label>Ask for confirmation when closing windows with multiple tabs.</label>
-            <default>true</default>
+            <default>false</default>
         </entry>
         <entry name="ShowSelectionToggle" type="Bool">
             <label>Show selection toggle</label>
@@ -58,11 +58,11 @@
         </entry>
         <entry name="ShowToolTips" type="Bool">
             <label>Show tooltips</label>
-            <default>false</default>
+            <default>true</default>
         </entry>
         <entry name="ShowCopyMoveMenu" type="Bool">
             <label>Show 'Copy To' and 'Move To' commands in context menu</label>
-            <default>false</default>
+            <default>true</default>
         </entry>
         <entry name="ViewPropsTimestamp" type="DateTime" >
             <label>Timestamp since when the view properties are valid</label>
@@ -77,7 +77,7 @@
         </entry>
         <entry name="ShowSpaceInfo" type="Bool">
             <label>Show the space information in the statusbar</label>
-            <default>false</default>
+            <default>true</default>
         </entry>
         <entry name="LockPanels" type="Bool">
             <label>Lock the layout of the panels</label>
