Index: kgraphviewer/src/kgrapheditor.desktop
===================================================================
--- kgraphviewer.orig/src/kgrapheditor.desktop
+++ kgraphviewer/src/kgrapheditor.desktop
@@ -49,6 +49,8 @@ Icon=kgraphviewer
 Type=Application
 Categories=Qt;KDE;Graphics;2DGraphics;DataVisualization;
 X-DocPath=kgrapheditor/kgrapheditor.html
+GeneticName=DOT File Editor
+GeneticName[tr]=DOT Dosyası Düzenleyici
 Comment=A Graphviz dot graph editor for KDE
 Comment[bg]=Редактиране на Graphviz-графики в KDE
 Comment[ca]=Un editor de gràfiques de punts de Graphviz pel KDE
Index: kgraphviewer/src/kgraphviewer.desktop
===================================================================
--- kgraphviewer.orig/src/kgraphviewer.desktop
+++ kgraphviewer/src/kgraphviewer.desktop
@@ -50,6 +50,8 @@ Icon=kgraphviewer
 Type=Application
 Categories=Qt;KDE;Graphics;Viewer;2DGraphics;DataVisualization;
 X-DocPath=kgraphviewer/index.html
+GenericName=Graph Viewer
+GenericName[tr]=Graf Gösterici
 Comment=A Graphviz dot graph viewer for KDE
 Comment[bg]=Преглед на Graphviz-графики в KDE
 Comment[ca]=Un visor de gràfiques de punts de Graphviz pel KDE
