--- src/part/DotGraphParsingHelper.cpp.orig
+++ src/part/DotGraphParsingHelper.cpp
@@ -26,10 +26,10 @@
 #include "graphedge.h"

 #include <boost/throw_exception.hpp>
-#include <boost/spirit/core.hpp>
-#include <boost/spirit/utility/distinct.hpp>
-#include <boost/spirit/utility/loops.hpp>
-#include <boost/spirit/utility/confix.hpp>
+#include <boost/spirit/include/classic_core.hpp>
+#include <boost/spirit/include/classic_distinct.hpp>
+#include <boost/spirit/include/classic_loops.hpp>
+#include <boost/spirit/include/classic_confix.hpp>


 #include <iostream>