diff -Nuar osmo-0.2.10.orig/data/osmo.desktop osmo-0.2.10/data/osmo.desktop
--- osmo-0.2.10.orig/data/osmo.desktop	2011-06-30 14:15:06.999000116 +0300
+++ osmo-0.2.10/data/osmo.desktop	2011-06-30 14:15:44.330000116 +0300
@@ -7,6 +7,7 @@
 Comment[pl]=Organizer osobisty
 Comment[tr]=Kişisel Düzenleyici
 GenericName=Osmo
+GenericName[tr]=Osmo
 Exec=osmo
 Icon=osmo
 Categories=GTK;Office;Calendar;ContactManagement;
