diff -p -up mono-2.6/mono/metadata/assembly.c.selfexe mono-2.6/mono/metadata/assembly.c
--- mono-2.6/mono/metadata/assembly.c.selfexe	2009-09-25 16:58:10.000000000 +0200
+++ mono-2.6/mono/metadata/assembly.c	2009-10-01 22:38:25.000000000 +0200
@@ -568,6 +568,7 @@ mono_set_rootdir (void)
 #elif defined(DISABLE_MONO_AUTODETECTION)
 	fallback ();
 #else
+#if 0
 	char buf [4096];
 	int  s;
 	char *str;
@@ -590,6 +591,7 @@ mono_set_rootdir (void)
 		set_dirs (buf);
 		return;
 	} 
+#endif
 	fallback ();
 #endif
 }
