diff -Naur lmms-0.4.0-rc3.orig/data/lmms.desktop lmms-0.4.0-rc3/data/lmms.desktop
--- lmms-0.4.0-rc3.orig/data/lmms.desktop	2008-10-20 13:15:28.000000000 +0300
+++ lmms-0.4.0-rc3/data/lmms.desktop	2008-10-20 20:29:20.000000000 +0300
@@ -1,12 +1,15 @@
 [Desktop Entry]
 Name=Linux MultiMedia Studio
-GenericName=music production suite
+Name[tr]=Linux MultiMedia Studio
+GenericName=Music Production Suite
+GenericName[tr]=Muzik Prodüksiyon Araçları
 GenericName[ca]=Programari de producció musical
 GenericName[de]=Software zur Musik-Produktion
 Comment=easy music production for everyone!
+Comment[tr]=kolayca müzik yapın!
 Comment[ca]=Producció fàcil de música per a tothom!
-Icon=/usr/share/lmms/themes/default/icon.png
-Exec=/usr/bin/lmms
+Icon=lmms.png
+Exec=lmms
 Terminal=false
 Type=Application
 Encoding=UTF-8
