diff -up lmms-0.4.4/CMakeLists.txt.orig lmms-0.4.4/CMakeLists.txt
--- lmms-0.4.4/CMakeLists.txt.orig	2009-05-05 01:13:12.000000000 +0200
+++ lmms-0.4.4/CMakeLists.txt	2009-05-19 14:55:26.000000000 +0200
@@ -321,6 +321,9 @@ IF(LMMS_BUILD_WIN32)
 	SET(LMMS_SUPPORT_VST TRUE)
 	SET(STATUS_VST "OK")
 ENDIF(LMMS_BUILD_WIN32)
+IF(WANT_VST_NOWINE)
+	SET(LMMS_SUPPORT_VST TRUE)
+ENDIF(WANT_VST_NOWINE)
 
 
 # check for libsamplerate
