diff -Nur kiki-old/src/gui/KikiMenu.cpp kiki/src/gui/KikiMenu.cpp
--- kiki-old/src/gui/KikiMenu.cpp	2006-11-18 16:19:58.215998055 +0200
+++ kiki/src/gui/KikiMenu.cpp	2006-11-18 16:20:13.513672455 +0200
@@ -347,6 +347,7 @@
         else
         {
             KikiText::colors[list_color].glColor();
+            list_color = -1;
         }
         
         menu_items[index]->item_text->display();
@@ -364,6 +365,7 @@
             else
             {
                 KikiText::colors[list_color].glColor();
+                list_color = -1;
             }
             menu_items[index]->value_text->display();
 
diff -Nur kiki-old/src/gui/KikiPageText.cpp kiki/src/gui/KikiPageText.cpp
--- kiki-old/src/gui/KikiPageText.cpp	2006-11-18 16:19:58.215998055 +0200
+++ kiki/src/gui/KikiPageText.cpp	2006-11-18 16:20:13.514672303 +0200
@@ -133,6 +133,7 @@
     else
     {
         KikiText::colors[list_color].glColor();
+        list_color = -1;
     }
     
     //KikiText::colors[list_color].glColor();
diff -Nur kiki-old/src/gui/KikiScreenText.cpp kiki/src/gui/KikiScreenText.cpp
--- kiki-old/src/gui/KikiScreenText.cpp	2006-11-18 16:19:58.215998055 +0200
+++ kiki/src/gui/KikiScreenText.cpp	2006-11-18 16:20:18.822865335 +0200
@@ -106,6 +106,7 @@
     if (fade_value != 1.0)
     {
         color.setAlpha (fade_value);
+        list_color = -1;
     }
     
     KikiText::display();
