diff -Nur kiki-old/src/main/KikiController.cpp kiki/src/main/KikiController.cpp
--- kiki-old/src/main/KikiController.cpp	2006-09-02 20:32:35.802504968 +0300
+++ kiki/src/main/KikiController.cpp	2006-09-02 20:33:31.853983848 +0300
@@ -62,13 +62,13 @@
     }
     else
     {
-       KConsole::printf("WARNING :: environment variable KIKI_HOME not set ...");
-       KConsole::printf("           ... assuming resources in current directory");
+       
+       
 
 	   kiki_home = kFileGetCurrentPath();
     }
 #ifndef WIN32
-    kiki_home += "/";
+    kiki_home = "/usr/share/kiki/";
 #else
     kiki_home += "\\";
 #endif
