--- kiki/src/main/main.cpp.orig	2010-12-20 14:59:54.155000050 +0200
+++ kiki/src/main/main.cpp	2010-12-20 15:00:52.307999991 +0200
@@ -9,6 +9,7 @@
 #include <math.h>
 
 #include <SDL.h>
+#include <GL/glut.h>
 #include "KikiController.h"
 
 // __________________________________________________________________________________________________
@@ -19,6 +20,8 @@
     int width, height;
     SDL_Surface * screen;
 
+    glutInit(&argc, argv);
+
     // initialize SDL
     if ( SDL_Init(SDL_INIT_VIDEO | SDL_INIT_VIDEO) < 0 ) // SDL_INIT_TIMER
     {
