diff -Naur kiki.orig/kodilib/src/handler/KFileHandler.h kiki/kodilib/src/handler/KFileHandler.h
--- kiki.orig/kodilib/src/handler/KFileHandler.h	2003-04-09 05:56:01.000000000 +0300
+++ kiki/kodilib/src/handler/KFileHandler.h	2008-04-22 22:16:10.000000000 +0300
@@ -9,6 +9,8 @@
 #include <string>
 #include <vector>
 
+#include <cstring>
+
 class KFileHandler
 {
     protected:
diff -Naur kiki.orig/kodilib/src/handler/KPickable.h kiki/kodilib/src/handler/KPickable.h
--- kiki.orig/kodilib/src/handler/KPickable.h	2003-04-09 05:56:01.000000000 +0300
+++ kiki/kodilib/src/handler/KPickable.h	2008-04-22 22:16:27.000000000 +0300
@@ -9,6 +9,9 @@
 #include "KIntrospection.h"
 #include <vector>
 
+#include <cstring>
+#include <climits>
+
 #define DEBUG_PICKING		false
 
 class KPickable;
diff -Naur kiki.orig/kodilib/src/handler/KPickHandler.h kiki/kodilib/src/handler/KPickHandler.h
--- kiki.orig/kodilib/src/handler/KPickHandler.h	2003-04-09 05:56:02.000000000 +0300
+++ kiki/kodilib/src/handler/KPickHandler.h	2008-04-22 22:16:51.000000000 +0300
@@ -6,6 +6,8 @@
 #ifndef __KPickHandler
 #define __KPickHandler
 
+#include <cstring>
+
 #include "KPickable.h"
 #include "KNotificationObject.h"
 
diff -Naur kiki.orig/kodilib/src/tools/KIncludeTools.h kiki/kodilib/src/tools/KIncludeTools.h
--- kiki.orig/kodilib/src/tools/KIncludeTools.h	2004-06-30 19:03:38.000000000 +0300
+++ kiki/kodilib/src/tools/KIncludeTools.h	2008-04-22 22:17:08.000000000 +0300
@@ -6,6 +6,8 @@
 #ifndef __KIncludeTools
 #define __KIncludeTools
 
+#include <cstring>
+
 #ifdef WIN32
 #pragma warning ( disable : 4800 4305 4267 ) // Disable warnings
 #endif
diff -Naur kiki.orig/kodilib/src/tools/KIntrospection.h kiki/kodilib/src/tools/KIntrospection.h
--- kiki.orig/kodilib/src/tools/KIntrospection.h	2003-04-09 05:56:03.000000000 +0300
+++ kiki/kodilib/src/tools/KIntrospection.h	2008-04-22 22:17:25.000000000 +0300
@@ -11,6 +11,7 @@
 #endif
 
 #include <string>
+#include <cstring>
 
 // --------------------------------------------------------------------------------------------------------
 class KClassInfo
diff -Naur kiki.orig/kodilib/src/tools/KStringTools.cpp kiki/kodilib/src/tools/KStringTools.cpp
--- kiki.orig/kodilib/src/tools/KStringTools.cpp	2003-04-14 00:42:13.000000000 +0300
+++ kiki/kodilib/src/tools/KStringTools.cpp	2008-04-22 22:17:42.000000000 +0300
@@ -9,6 +9,8 @@
 #include <sys/types.h> 	// INT_MAX
 #include <stdio.h>
 
+#include <climits>
+
 // --------------------------------------------------------------------------------------------------------
 void kStringInsertStringBehindTags ( std::string & str, const std::string & insertString, 
                                      const std::string & tag )
diff -Naur kiki.orig/kodilib/src/tools/KStringTools.h kiki/kodilib/src/tools/KStringTools.h
--- kiki.orig/kodilib/src/tools/KStringTools.h	2003-04-09 05:56:04.000000000 +0300
+++ kiki/kodilib/src/tools/KStringTools.h	2008-04-22 22:17:51.000000000 +0300
@@ -10,6 +10,8 @@
 #include <vector>
 #include <stdarg.h>
 
+#include <cstring>
+
 // --------------------------------------------------------------------------------------------------------
 
 void		kStringInsertStringBehindTags	( std::string & str, const std::string & insertString,
diff -Naur kiki.orig/kodilib/src/tools/KXMLTools.cpp kiki/kodilib/src/tools/KXMLTools.cpp
--- kiki.orig/kodilib/src/tools/KXMLTools.cpp	2003-04-14 00:42:13.000000000 +0300
+++ kiki/kodilib/src/tools/KXMLTools.cpp	2008-04-22 22:17:59.000000000 +0300
@@ -8,6 +8,8 @@
 #include "KSeparatedMatrix.h"
 #include <stdio.h>
 
+#include <cstdlib>
+
 // --------------------------------------------------------------------------------------------------------
 std::string kXMLTag ( const std::string & name, const std::string & attributes, int depth )
 {
