diff -Nuar enlightenment-0.17.1-orig/data/themes/edc/about-e.edc enlightenment-0.17.1/data/themes/edc/about-e.edc
--- enlightenment-0.17.1-orig/data/themes/edc/about-e.edc	2013-01-30 15:23:36.000000000 +0200
+++ enlightenment-0.17.1/data/themes/edc/about-e.edc	2013-04-01 11:25:22.268992131 +0300
@@ -42,7 +42,7 @@
             image.normal: "bg_radgrad.png";
             image.scale_hint: STATIC;
             fill.smooth: 0;
-            ASPECT_FILL((1024/653))
+            ASPECT_FILL((2560/1440))
          }
       }
       part { name: "pat";
diff -Nuar enlightenment-0.17.1-orig/data/themes/edc/about-theme.edc enlightenment-0.17.1/data/themes/edc/about-theme.edc
--- enlightenment-0.17.1-orig/data/themes/edc/about-theme.edc	2013-01-30 15:23:36.000000000 +0200
+++ enlightenment-0.17.1/data/themes/edc/about-theme.edc	2013-04-01 11:26:00.215990817 +0300
@@ -19,7 +19,7 @@
             image.normal: "bg_radgrad.png";
             image.scale_hint: STATIC;
             fill.smooth: 0;
-            ASPECT_FILL((1024/653))
+            ASPECT_FILL((2560/1440))
          }
       }
       part { name: "name"; type: TEXT; mouse_events: 0;
diff -Nuar enlightenment-0.17.1-orig/data/themes/edc/background.edc enlightenment-0.17.1/data/themes/edc/background.edc
--- enlightenment-0.17.1-orig/data/themes/edc/background.edc	2013-01-30 15:23:36.000000000 +0200
+++ enlightenment-0.17.1/data/themes/edc/background.edc	2013-04-01 11:27:38.175987425 +0300
@@ -7,7 +7,7 @@
             image.normal: "bg_radgrad.png";
             image.scale_hint: STATIC;
             fill.smooth: 0;
-            ASPECT_FILL((1024/653))
+            ASPECT_FILL((2560/1440))
          }
       }
       part { name: "pat";
diff -Nuar enlightenment-0.17.1-orig/data/themes/edc/desklock.edc enlightenment-0.17.1/data/themes/edc/desklock.edc
--- enlightenment-0.17.1-orig/data/themes/edc/desklock.edc	2013-01-30 15:23:36.000000000 +0200
+++ enlightenment-0.17.1/data/themes/edc/desklock.edc	2013-04-01 11:28:08.169986387 +0300
@@ -7,7 +7,7 @@
             image.normal: "bg_radgrad.png";
             image.scale_hint: STATIC;
             fill.smooth: 0;
-            ASPECT_FILL((1024/653))
+            ASPECT_FILL((2560/1440))
          }
       }
       part { name: "pat";
diff -Nuar enlightenment-0.17.1-orig/data/themes/edc/init.edc enlightenment-0.17.1/data/themes/edc/init.edc
--- enlightenment-0.17.1-orig/data/themes/edc/init.edc	2013-01-30 15:23:36.000000000 +0200
+++ enlightenment-0.17.1/data/themes/edc/init.edc	2013-04-01 11:29:17.716983979 +0300
@@ -23,7 +23,7 @@
             image.normal: "bg_radgrad.png";
             image.scale_hint: STATIC;
             fill.smooth: 0;
-            ASPECT_FILL((1024/653))
+            ASPECT_FILL((2560/1440))
          }
       }
       part { name: "pat";
@@ -534,7 +534,7 @@
             image.normal: "bg_radgrad.png";
             image.scale_hint: STATIC;
             fill.smooth: 0;
-            ASPECT_FILL((1024/653))
+            ASPECT_FILL((2560/1440))
          }
       }
       part { name: "pat";
diff -Nuar enlightenment-0.17.1-orig/data/themes/edc/wizard.edc enlightenment-0.17.1/data/themes/edc/wizard.edc
--- enlightenment-0.17.1-orig/data/themes/edc/wizard.edc	2013-01-30 15:23:36.000000000 +0200
+++ enlightenment-0.17.1/data/themes/edc/wizard.edc	2013-04-01 11:30:24.388981670 +0300
@@ -16,7 +16,7 @@
             image.normal: "bg_radgrad.png";
             image.scale_hint: STATIC;
             fill.smooth: 0;
-            ASPECT_FILL((1024/653))
+            ASPECT_FILL((2560/1440))
          }
       }
       part { name: "pat";
@@ -328,7 +328,7 @@
             image.normal: "bg_radgrad.png";
             image.scale_hint: STATIC;
             fill.smooth: 0;
-            ASPECT_FILL((1024/653))
+            ASPECT_FILL((2560/1440))
          }
       }
       part { name: "pat";
