Index: enlightenment-0.16.999.73893/src/modules/cpufreq/Makefile.am
===================================================================
--- enlightenment-0.16.999.73893.orig/src/modules/cpufreq/Makefile.am
+++ enlightenment-0.16.999.73893/src/modules/cpufreq/Makefile.am
@@ -24,6 +24,8 @@ module_la_LIBADD       = @e_libs@ @dlope
 module_la_LDFLAGS      = -module -avoid-version
 module_la_DEPENDENCIES = $(top_builddir)/config.h
 
+freqset_CFLAGS = @e_cflags@ @SUID_CFLAGS@
+freqset_LDFLAGS = @SUID_LDFLAGS@
 freqsetdir = $(pkgdir)
 
 setuid_root_mode = a=rx,u+xs
