diff -up GD-2.41/Makefile.PL.BAD GD-2.41/Makefile.PL
--- GD-2.41/Makefile.PL.BAD	2009-03-16 09:00:18.000000000 -0400
+++ GD-2.41/Makefile.PL	2009-03-16 09:00:34.000000000 -0400
@@ -232,6 +232,7 @@ WriteMakefile(
 		    'Math::Trig' => 0,
 		    },
     'PM'        => { 'GD.pm' => '$(INST_LIBDIR)/GD.pm',
+                     'GD/Group.pm' => '$(INST_LIBDIR)/GD/Group.pm',
                      'GD/Polyline.pm' => '$(INST_LIBDIR)/GD/Polyline.pm',
                      'GD/Polygon.pm' => '$(INST_LIBDIR)/GD/Polygon.pm',
                      'GD/Simple.pm' => '$(INST_LIBDIR)/GD/Simple.pm',
