--- Django-1.0-orig/django/conf/project_template/settings.py	2008-09-04 05:47:30.000000000 +0300
+++ Django-1.0/django/conf/project_template/settings.py	2008-09-04 05:50:38.000000000 +0300
@@ -59,6 +59,7 @@
 
 MIDDLEWARE_CLASSES = (
     'django.middleware.common.CommonMiddleware',
+    'django.contrib.csrf.middleware.CsrfMiddleware',
     'django.contrib.sessions.middleware.SessionMiddleware',
     'django.contrib.auth.middleware.AuthenticationMiddleware',
 )
