--- Django-1.0-orig/django/conf/project_template/settings.py	2008-09-04 05:47:30.000000000 +0300
+++ Django-1.0/django/conf/project_template/settings.py	2008-09-04 05:50:38.000000000 +0300
@@ -21,11 +21,11 @@
 # although not all choices may be available on all operating systems.
 # If running in a Windows environment this must be set to the same as your
 # system time zone.
-TIME_ZONE = 'America/Chicago'
+TIME_ZONE = 'Europe/Istanbul'
 
 # Language code for this installation. All choices can be found here:
 # http://www.i18nguy.com/unicode/language-identifiers.html
-LANGUAGE_CODE = 'en-us'
+LANGUAGE_CODE = 'tr'
 
 SITE_ID = 1
