Index: paprefs-0.9.9/src/paprefs.desktop.in
===================================================================
--- paprefs-0.9.9.orig/src/paprefs.desktop.in
+++ paprefs-0.9.9/src/paprefs.desktop.in
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Version=1.0
 Encoding=UTF-8
-_Name=PulseAudio Preferences
-_GenericName=Sound Server Preferences
+_Name=paprefs
+_GenericName=PulseAudio Preferences
 _Comment=View and modify the configuration of the local sound server
 Exec=paprefs
 Icon=paprefs
