--- a/include/linux/fs.h	2013-04-22 00:38:45.000000000 +0300
+++ b/include/linux/fs.h	2013-04-26 19:17:13.143318654 +0300
@@ -1573,6 +1573,10 @@ struct inode_operations {
 	int (*atomic_open)(struct inode *, struct dentry *,
 			   struct file *, unsigned open_flag,
 			   umode_t create_mode, int *opened);
+	int (*may_create) (struct inode *, int);
+	int (*may_delete) (struct inode *, struct inode *, int);
+
+
 } ____cacheline_aligned;
 
 ssize_t rw_copy_check_uvector(int type, const struct iovec __user * uvector,
