From: David Sterba <dsterba@suse.cz>
Date: Mon, 02 Jan 2012 13:40:28 +0100
Subject: [PATCH] btrfs: use correct device for maps
Reference: bnc#672923 bnc#769545
Patch-mainline: no

Signed-off-by: David Sterba <dsterba@suse.cz>
---

Index: linux-3.1-openSUSE-12.1/fs/proc/task_mmu.c
===================================================================
diff -Nurp a/fs/proc/task_mmu.c b/fs/proc/task_mmu.c
--- a/fs/proc/task_mmu.c	2013-04-22 00:38:45.000000000 +0300
+++ b/fs/proc/task_mmu.c	2013-04-26 17:40:10.850520241 +0300
@@ -272,7 +272,15 @@ show_map_vma(struct seq_file *m, struct
 
 	if (file) {
 		struct inode *inode = file_inode(vma->vm_file);
-		dev = inode->i_sb->s_dev;
+
+		if (inode->i_sb->s_magic == BTRFS_SUPER_MAGIC) {
+			struct kstat stat;
+
+			vfs_getattr(file->f_path.mnt, file->f_path.dentry, &stat);
+			dev = stat.dev;
+		} else {
+			dev = inode->i_sb->s_dev;
+		}
 		ino = inode->i_ino;
 		pgoff = ((loff_t)vma->vm_pgoff) << PAGE_SHIFT;
 	}
diff -Nurp a/fs/proc/task_nommu.c b/fs/proc/task_nommu.c
--- a/fs/proc/task_nommu.c	2013-04-22 00:38:45.000000000 +0300
+++ b/fs/proc/task_nommu.c	2013-04-26 17:41:15.061518018 +0300
@@ -150,7 +150,15 @@ static int nommu_vma_show(struct seq_fil
 
 	if (file) {
 		struct inode *inode = file_inode(vma->vm_file);
-		dev = inode->i_sb->s_dev;
+
+		if (inode->i_sb->s_magic == BTRFS_SUPER_MAGIC) {
+			struct kstat stat;
+
+			vfs_getattr(file->f_path.mnt, file->f_path.dentry, &stat);
+			dev = stat.dev;
+		} else {
+			dev = inode->i_sb->s_dev;
+		}
 		ino = inode->i_ino;
 		pgoff = (loff_t)vma->vm_pgoff << PAGE_SHIFT;
 	}
