--- tk8.5.10.orig/unix/Makefile.in	2011-08-01 14:24:06.821000004 +0300
+++ tk8.5.10/unix/Makefile.in	2011-08-01 14:52:25.367000004 +0300
@@ -722,7 +722,7 @@
 	    fi
 	@echo "Installing $(LIB_FILE) to $(LIB_INSTALL_DIR)/"
 	@@INSTALL_LIB@
-	@chmod 555 "$(LIB_INSTALL_DIR)"/$(LIB_FILE)
+	@chmod 755 "$(LIB_INSTALL_DIR)"/$(LIB_FILE)
 	@echo "Installing wish@EXEEXT@ as $(BIN_INSTALL_DIR)/wish$(VERSION)@EXEEXT@"
 	@$(INSTALL_PROGRAM) wish@EXEEXT@ "$(BIN_INSTALL_DIR)"/wish$(VERSION)@EXEEXT@
 	@echo "Installing tkConfig.sh to $(CONFIG_INSTALL_DIR)/"
