Index: kdeadmin-4.7.1/system-config-printer-kde/system-config-printer.ui
===================================================================
--- kdeadmin-4.7.1.orig/system-config-printer-kde/system-config-printer.ui
+++ kdeadmin-4.7.1/system-config-printer-kde/system-config-printer.ui
@@ -6,7 +6,7 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>826</width>
+    <width>828</width>
     <height>509</height>
    </rect>
   </property>
@@ -16,22 +16,26 @@
   </property>
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="0" column="0">
-    <widget class="QTreeWidget" name="mainlist">
-     <property name="sizePolicy">
-      <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
-       <horstretch>0</horstretch>
-       <verstretch>0</verstretch>
-      </sizepolicy>
-     </property>
-     <property name="rootIsDecorated">
-      <bool>true</bool>
-     </property>
-     <column>
-      <property name="text">
-       <string/>
-      </property>
-     </column>
-    </widget>
+    <layout class="QVBoxLayout" name="verticalLayout_7">
+     <item>
+      <widget class="QTreeWidget" name="mainlist">
+       <property name="sizePolicy">
+        <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
+         <horstretch>0</horstretch>
+         <verstretch>0</verstretch>
+        </sizepolicy>
+       </property>
+       <property name="rootIsDecorated">
+        <bool>true</bool>
+       </property>
+       <column>
+        <property name="text">
+         <string/>
+        </property>
+       </column>
+      </widget>
+     </item>
+    </layout>
    </item>
    <item row="0" column="1">
     <widget class="QStackedWidget" name="ntbkMain">
@@ -877,8 +881,8 @@ print to.</string>
                  <rect>
                   <x>0</x>
                   <y>0</y>
-                  <width>532</width>
-                  <height>427</height>
+                  <width>504</width>
+                  <height>381</height>
                  </rect>
                 </property>
                 <layout class="QGridLayout" name="gridLayout_5">
@@ -912,8 +916,8 @@ print to.</string>
                  <rect>
                   <x>0</x>
                   <y>0</y>
-                  <width>515</width>
-                  <height>1026</height>
+                  <width>553</width>
+                  <height>1036</height>
                  </rect>
                 </property>
                 <layout class="QHBoxLayout" name="horizontalLayout_3">
@@ -1765,6 +1769,19 @@ if they are not already set by the appli
      <widget class="QWidget" name="page_4"/>
     </widget>
    </item>
+   <item row="1" column="0">
+    <widget class="QLabel" name="label_2">
+     <property name="text">
+      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;a href=&quot;http://localhost:631&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;Open CUPS Web Interface&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
+     </property>
+     <property name="alignment">
+      <set>Qt::AlignCenter</set>
+     </property>
+     <property name="openExternalLinks">
+      <bool>true</bool>
+     </property>
+    </widget>
+   </item>
   </layout>
  </widget>
  <customwidgets>
@@ -1774,11 +1791,6 @@ if they are not already set by the appli
    <header>kcombobox.h</header>
   </customwidget>
   <customwidget>
-   <class>KIntSpinBox</class>
-   <extends>QSpinBox</extends>
-   <header>knuminput.h</header>
-  </customwidget>
-  <customwidget>
    <class>KLineEdit</class>
    <extends>QLineEdit</extends>
    <header>klineedit.h</header>
@@ -1788,6 +1800,11 @@ if they are not already set by the appli
    <extends>QPushButton</extends>
    <header>kpushbutton.h</header>
   </customwidget>
+  <customwidget>
+   <class>KIntSpinBox</class>
+   <extends>QSpinBox</extends>
+   <header>knuminput.h</header>
+  </customwidget>
  </customwidgets>
  <resources/>
  <connections/>
