diff -up kdeadmin-4.6.2/system-config-printer-kde/system-config-printer-kde.desktop.root_privs kdeadmin-4.6.2/system-config-printer-kde/system-config-printer-kde.desktop
--- kdeadmin-4.6.2/system-config-printer-kde/system-config-printer-kde.desktop.root_privs	2011-02-25 15:46:02.000000000 -0600
+++ kdeadmin-4.6.2/system-config-printer-kde/system-config-printer-kde.desktop	2011-04-08 14:17:40.365597452 -0500
@@ -5,6 +5,10 @@ Type=Service
 X-KDE-ServiceTypes=KCModule
 
 X-KDE-Library=kpythonpluginfactory
+X-KDE-ModuleType=Library
+X-KDE-SubstituteUID=false
+X-KDE-HasReadOnlyMode=true
+X-KDE-RootOnly=true
 X-KDE-ParentApp=kcontrol
 X-KDE-PluginKeyword=system-config-printer-kde/system-config-printer-kde.py
 
