Index: teeworlds-b177-r50edfd37-source/bam.lua
===================================================================
--- teeworlds-b177-r50edfd37-source.orig/bam.lua
+++ teeworlds-b177-r50edfd37-source/bam.lua
@@ -9,6 +9,7 @@ config = NewConfig()
 config:Add(OptCCompiler("compiler"))
 config:Add(OptTestCompileC("stackprotector", "int main(){return 0;}", "-fstack-protector -fstack-protector-all"))
 config:Add(OptLibrary("zlib", "zlib.h", false))
+config:Add(OptLibrary("pnglite", "pnglite.h", false))
 config:Add(SDL.OptFind("sdl", true))
 config:Add(FreeType.OptFind("freetype", true))
 config:Finalize("config.lua")
@@ -178,7 +179,18 @@ function build(settings)
 
 	-- build the small libraries
 	wavpack = Compile(settings, Collect("src/engine/external/wavpack/*.c"))
-	pnglite = Compile(settings, Collect("src/engine/external/pnglite/*.c"))
+
+	-- compile pnglite if needed
+	if config.pnglite.value == 1 then
+		settings.link.libs:Add("pnglite")
+		if config.pnglite.include_path then
+			settings.cc.includes:Add(config.pnglite.include_path)
+		end
+		pnglite = {}
+	else
+		pnglite = Compile(settings, Collect("src/engine/external/pnglite/*.c"))
+		settings.cc.includes:Add("src/engine/external/pnglite")
+	end
 
 	-- build game components
 	engine_settings = settings:Copy()
Index: teeworlds-b177-r50edfd37-source/src/engine/client/graphics.cpp
===================================================================
--- teeworlds-b177-r50edfd37-source.orig/src/engine/client/graphics.cpp
+++ teeworlds-b177-r50edfd37-source/src/engine/client/graphics.cpp
@@ -20,7 +20,7 @@
 #endif
 
 #include <base/system.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <pnglite.h>
 
 #include <engine/shared/config.h>
 #include <engine/graphics.h>
Index: teeworlds-b177-r50edfd37-source/src/tools/dilate.cpp
===================================================================
--- teeworlds-b177-r50edfd37-source.orig/src/tools/dilate.cpp
+++ teeworlds-b177-r50edfd37-source/src/tools/dilate.cpp
@@ -2,7 +2,10 @@
 /* If you are missing that file, acquire a complete release at teeworlds.com.                */
 #include <base/system.h>
 #include <base/math.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <sys/types.h>
+#include <string.h>
+#include <stdio.h>
+#include <pnglite.h>
 
 typedef struct
 {
Index: teeworlds-b177-r50edfd37-source/src/tools/tileset_borderfix.cpp
===================================================================
--- teeworlds-b177-r50edfd37-source.orig/src/tools/tileset_borderfix.cpp
+++ teeworlds-b177-r50edfd37-source/src/tools/tileset_borderfix.cpp
@@ -1,7 +1,10 @@
 /* (c) Magnus Auvinen. See licence.txt in the root of the distribution for more information. */
 /* If you are missing that file, acquire a complete release at teeworlds.com.                */
 #include <base/system.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <sys/types.h>
+#include <string.h>
+#include <stdio.h>
+#include <pnglite.h>
 
 typedef struct
 {
