Index: teeworlds-b177-r50edfd37-source/bam.lua
===================================================================
--- teeworlds-b177-r50edfd37-source.orig/bam.lua
+++ teeworlds-b177-r50edfd37-source/bam.lua
@@ -133,6 +133,12 @@ end
 function build(settings)
 	--settings.objdir = Path("objs")
 	settings.cc.Output = Intermediate_Output
+	settings.cc.flags:Add(os.getenv ('CFLAGS'))
+	settings.cc.flags:Add(os.getenv ('CPPFLAGS'))
+	settings.cc.flags:Add(os.getenv ('LDFLAGS'))
+	settings.link.libs:Add("z")
+	settings.link.libs:Add("pnglite")
+
 
 	if config.compiler.driver == "cl" then
 		settings.cc.flags:Add("/wd4244")
