diff -Nuar kile.orig/src/data/kilestdtools.rc kile/src/data/kilestdtools.rc
--- kile.orig/src/data/kilestdtools.rc	2009-07-15 15:44:01.986788507 +0300
+++ kile/src/data/kilestdtools.rc	2009-07-15 15:48:48.222544677 +0300
@@ -697,9 +697,9 @@
 state=Viewer
 type=Part
 
-[Tool/ViewPDF/Acroread]
+[Tool/ViewPDF/Okular]
 class=View
-command=acroread
+command=okular
 options='%target'
 state=Viewer
 from=pdf
diff -Nuar kile.orig/src/data/kilestdtools-win.rc kile/src/data/kilestdtools-win.rc
--- kile.orig/src/data/kilestdtools-win.rc	2009-07-15 15:44:01.907788569 +0300
+++ kile/src/data/kilestdtools-win.rc	2009-07-15 15:48:20.914543697 +0300
@@ -717,9 +717,9 @@
 state=Viewer
 type=Part
 
-[Tool/ViewPDF/Acroread]
+[Tool/ViewPDF/Okular]
 class=View
-command=acroread
+command=okular
 options="%target"
 state=Viewer
 from=pdf
diff -Nuar kile.orig/src/test/runTests.sh kile/src/test/runTests.sh
--- kile.orig/src/test/runTests.sh	2009-07-15 15:44:01.409539142 +0300
+++ kile/src/test/runTests.sh	2009-07-15 15:47:41.547543171 +0300
@@ -181,11 +181,11 @@
 setKey executable okular
 setKey where `which okular`
 
-echo "starting test: Acroread"
-setTool Acroread
+echo "starting test: Okular"
+setTool Okular
 setKey mustpass ""
-setKey executable acroread
-setKey where `which acroread`
+setKey executable okular
+setKey where `which okular`
 
 echo "starting test: DVItoPNG"
 setTool DVItoPNG
