diff -Naur xfce-utils-4.6.0/scripts/startxfce4.in xfce-utils-4.6.0.tpg/scripts/startxfce4.in
--- xfce-utils-4.6.0/scripts/startxfce4.in	2009-02-24 06:15:25.000000000 +0000
+++ xfce-utils-4.6.0.tpg/scripts/startxfce4.in	2009-02-28 18:01:40.000000000 +0000
@@ -28,9 +28,9 @@
 
 if test "x$XDG_DATA_DIRS" = "x"
 then
-  XDG_DATA_DIRS="/usr/local/share:/usr/share:@_datadir_@"
+  XDG_DATA_DIRS="/etc/xdg:/usr/local/share:/usr/share:@_datadir_@"
 else
-  XDG_DATA_DIRS="$XDG_DATA_DIRS:@_datadir_@"
+  XDG_DATA_DIRS="/etc/xdg:$XDG_DATA_DIRS:@_datadir_@"
 fi
 export XDG_DATA_DIRS
 
