diff -Naur xfce-utils-4.6.1/scripts/xinitrc.in.in xfce-utils-4.6.1.tpg/scripts/xinitrc.in.in
--- xfce-utils-4.6.1/scripts/xinitrc.in.in	2009-04-09 23:21:13.000000000 +0000
+++ xfce-utils-4.6.1.tpg/scripts/xinitrc.in.in	2009-04-22 18:15:31.000000000 +0000
@@ -32,32 +32,39 @@
     xdg-user-dirs-update
 fi
 
-if test -f "$XDG_CONFIG_HOME/user-dirs.dirs"; then
-    . "$XDG_CONFIG_HOME/user-dirs.dirs"
+#if test -f "$XDG_CONFIG_HOME/user-dirs.dirs"; then
+#    . "$XDG_CONFIG_HOME/user-dirs.dirs"
     # i'm deliberately not 'export'-ing the XDG_ vars, because you shouldn't
     # rely on the env vars inside apps, since the file could be changed at
     # any time by the user.  this is solely here for migration purposes.
 
     # a bit of user dir migration...
-    if test -d "$HOME/Desktop" -a ! -L "$HOME/Desktop" \
-            -a "$XDG_DESKTOP_DIR" \
-            -a "$HOME/Desktop" != "$XDG_DESKTOP_DIR"
-    then
-        echo "Migrating $HOME/Desktop to $XDG_DESKTOP_DIR..."
-        test -d "$XDG_DESKTOP_DIR" && rmdir "$XDG_DESKTOP_DIR"
-        mv "$HOME/Desktop" "$XDG_DESKTOP_DIR" || echo "Desktop migration failed" >&2
-    fi
-
-    if test -d "$HOME/Templates" -a ! -L "$HOME/Templates" \
-            -a "$XDG_TEMPLATES_DIR" \
-            -a "$HOME/Templates" != "$XDG_TEMPLATES_DIR"
-    then
-        echo "Migrating $HOME/Templates to $XDG_TEMPLATES_DIR..."
-        test -d "$XDG_TEMPLATES_DIR" && rmdir "$XDG_TEMPLATES_DIR"
-        mv "$HOME/Templates" "$XDG_TEMPLATES_DIR" || echo "Templates migration failed" >&2
-    fi
-fi
-
+#    if test -d "$HOME/Desktop" -a ! -L "$HOME/Desktop" \
+#            -a "$XDG_DESKTOP_DIR" \
+#            -a "$HOME/Desktop" != "$XDG_DESKTOP_DIR"
+#    then
+#        echo "Migrating $HOME/Desktop to $XDG_DESKTOP_DIR..."
+#        test -d "$XDG_DESKTOP_DIR" && rmdir "$XDG_DESKTOP_DIR"
+#        mv "$HOME/Desktop" "$XDG_DESKTOP_DIR" || echo "Desktop migration failed" >&2
+#    fi
+
+#    if test -d "$HOME/Templates" -a ! -L "$HOME/Templates" \
+#            -a "$XDG_TEMPLATES_DIR" \
+#            -a "$HOME/Templates" != "$XDG_TEMPLATES_DIR"
+#    then
+#        echo "Migrating $HOME/Templates to $XDG_TEMPLATES_DIR..."
+#        test -d "$XDG_TEMPLATES_DIR" && rmdir "$XDG_TEMPLATES_DIR"
+#        mv "$HOME/Templates" "$XDG_TEMPLATES_DIR" || echo "Templates migration failed" >&2
+#    fi
+#fi
+
+# Create Xfce user preferences if they are missing
+for i in volstatus panel; do
+    if [ ! -d $XDG_CONFIG_HOME/xfce4/$i ]; then
+       mkdir -p $XDG_CONFIG_HOME/xfce4/$i
+       cp -r /etc/X11/xdg/xfce4/$i $XDG_CONFIG_HOME/xfce4
+    fi ;
+done
 
 # Modify libglade and glade environment variables so that
 # it will find the files installed by Xfce
@@ -108,22 +115,24 @@
     fi
 fi 
 
+# (tpg) useless
 # Use ssh-agent if installed and not already running.  Run it separately
 # so it populates the environment here, so we can clean it up later.
-sshagent=`which ssh-agent`
-kill_sshagent=0
-if test -z "$SSH_AGENT_PID" -a "$sshagent" -a "x$sshagent" != "xno"; then
-	eval `$sshagent -s`
-	kill_sshagent=1
-fi
+#sshagent=`which ssh-agent`
+#kill_sshagent=0
+#if test -z "$SSH_AGENT_PID" -a "$sshagent" -a "x$sshagent" != "xno"; then
+#	eval `$sshagent -s`
+#	kill_sshagent=1
+#fi
 
+# (tpg) this is done as X start
 # Use dbus-launch if installed.
-if test x"$DBUS_SESSION_BUS_ADDRESS" = x""; then
-	dbuslaunch=`which dbus-launch`
-	if test x"$dbuslaunch" != x"" -a x"$dbuslaunch" != x"no"; then
-		eval `$dbuslaunch --sh-syntax --exit-with-session`
-	fi
-fi
+#if test x"$DBUS_SESSION_BUS_ADDRESS" = x""; then
+#	dbuslaunch=`which dbus-launch`
+#	if test x"$dbuslaunch" != x"" -a x"$dbuslaunch" != x"no"; then
+#		eval `$dbuslaunch --sh-syntax --exit-with-session`
+#	fi
+#fi
 
 # Run xfce4-session if installed
 xfcesm=`which xfce4-session`
@@ -132,10 +141,10 @@
 		;;
 	*)
 		$xfcesm
-
-		if test $kill_sshagent -eq 1; then
-			eval `$sshagent -k`
-		fi
+# (tpg) useless
+#		if test $kill_sshagent -eq 1; then
+#			eval `$sshagent -k`
+#		fi
 		exit 0
 		;;
 esac
