diff -Nuar lxc-1.0.4.orig/configure.ac lxc-1.0.4/configure.ac
--- lxc-1.0.4.orig/configure.ac	2014-06-13 20:09:56.000000000 +0300
+++ lxc-1.0.4/configure.ac	2014-06-30 01:06:04.080012105 +0300
@@ -162,7 +162,7 @@
 AM_CONDITIONAL([USE_DOCBOOK2X], [test "x$db2xman" != "xdocbook2man"])
 
 if test "x$db2xman" = "xdocbook2man"; then
-	docdtd="\"-//Davenport//DTD DocBook V3.0//EN\""
+	docdtd="\"-//OASIS//DTD DocBook XML\" \"http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd\""
 else
 	docdtd="\"-//OASIS//DTD DocBook XML\" \"http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd\""
 fi
