Index: lxc-0.7.2/src/lxc/Makefile.am
===================================================================
--- lxc-0.7.2.orig/src/lxc/Makefile.am
+++ lxc-0.7.2/src/lxc/Makefile.am
@@ -59,7 +59,7 @@ liblxc_so_LDFLAGS = \
 	-shared \
 	-Wl,-soname,liblxc.so.$(firstword $(subst ., ,$(VERSION)))
 
-liblxc_so_LDADD = -lutil
+liblxc_so_LDADD = -lutil -lcap
 
 bin_SCRIPTS = \
 	lxc-ps \
@@ -93,7 +93,7 @@ pkglib_PROGRAMS = \
 	lxc-init
 
 AM_LDFLAGS=-Wl,-E -Wl,-rpath -Wl,$(libdir)
-LDADD=liblxc.so @CAP_LIBS@
+LDADD=liblxc.so
 
 lxc_attach_SOURCES = lxc_attach.c
 lxc_cgroup_SOURCES = lxc_cgroup.c
