--- sylpheed-3.1.1/src/compose.c	2011-03-16 06:04:50.000000000 +0200
+++ sylpheed-3.1.0/src/compose.c	2011-01-07 08:35:22.000000000 +0200
@@ -576,7 +576,7 @@
 static GtkItemFactoryEntry compose_entries[] =
 {
 	{N_("/_File"),				NULL, NULL, 0, "<Branch>"},
-	{N_("/_File/_Send"),			"<shift><control>E",
+	{N_("/_File/_Send"),			"<control>Return",
 						compose_send_cb, 0, NULL},
 	{N_("/_File/Send _later"),		"<shift><control>S",
 						compose_send_later_cb,  0, NULL},
