--- Python-3.3.2/setup.py.orig	2013-07-01 15:23:24.377711044 +0200
+++ Python-3.3.2/setup.py	2013-07-01 15:23:34.094676496 +0200
@@ -1882,7 +1882,8 @@
                 if not line:
                     ffi_inc = None
                     break
-                if line.startswith('#define LIBFFI_H'):
+                if line.startswith('#define LIBFFI_H') or \
+                   line.startswith('#define ffi_wrapper_h'):
                     break
         ffi_lib = None
         if ffi_inc is not None:
