--- Lib/site.py	2005-10-17 15:38:18.000000000 +0300
+++ Lib/site.py	2005-10-17 15:41:02.000000000 +0300
@@ -349,7 +349,7 @@
     """Set the string encoding used by the Unicode implementation.  The
     default is 'ascii', but if you're willing to experiment, you can
     change this."""
-    encoding = "ascii" # Default value set by _PyUnicode_Init()
+    encoding = "utf-8" # Default value set by _PyUnicode_Init()
     if 0:
         # Enable to support locale aware default string encodings.
         import locale
Yalnızca Python-2.4.2/Lib'da: site.py~
