diff -aur Lib~/sre_compile.py Lib/sre_compile.py
--- Lib~/sre_compile.py	2013-05-12 05:32:44.000000000 +0200
+++ Lib/sre_compile.py	2013-07-16 16:56:01.203830712 +0200
@@ -13,7 +13,7 @@
 import _sre, sys
 import sre_parse
 from sre_constants import *
-from _sre import MAXREPEAT
+#from _sre import MAXREPEAT
 
 assert _sre.MAGIC == MAGIC, "SRE module mismatch"
 
diff -aur Lib~/sre_constants.py Lib/sre_constants.py
--- Lib~/sre_constants.py	2013-05-12 05:32:44.000000000 +0200
+++ Lib/sre_constants.py	2013-07-16 16:55:18.441832274 +0200
@@ -15,7 +15,8 @@
 
 MAGIC = 20031017
 
-from _sre import MAXREPEAT
+MAXREPEAT = 65535
+#from _sre import MAXREPEAT
 
 # SRE standard exception (access as sre.error)
 # should this really be here?
diff -aur Lib~/sre_parse.py Lib/sre_parse.py
--- Lib~/sre_parse.py	2013-05-12 05:32:44.000000000 +0200
+++ Lib/sre_parse.py	2013-07-16 16:55:49.402831143 +0200
@@ -15,7 +15,7 @@
 import sys
 
 from sre_constants import *
-from _sre import MAXREPEAT
+#from _sre import MAXREPEAT
 
 SPECIAL_CHARS = ".\\[{()*+?^$|"
 REPEAT_CHARS = "*+?{"
