diff -Naur Python-2.7.orig/Python/codecs.c Python-2.7/Python/codecs.c
--- Python-2.7.orig/Python/codecs.c	2010-05-09 17:46:46.000000000 +0300
+++ Python-2.7/Python/codecs.c	2010-08-10 14:53:35.520888909 +0300
@@ -45,6 +45,12 @@
     return -1;
 }
 
+/* isupper() forced into the ASCII Locale */
+#define ascii_isupper(x) (((x) >= 0x41) && ((x) <= 0x5A))
+/* tolower() forced into the ASCII Locale */
+#define ascii_tolower(x) (ascii_isupper(x) ? ((x) + 0x20) : (x))
+
+
 /* Convert a string to a normalized Python string: all characters are
    converted to lower case, spaces are replaced with underscores. */
 
@@ -70,7 +76,7 @@
         if (ch == ' ')
             ch = '-';
         else
-            ch = tolower(Py_CHARMASK(ch));
+            ch = ascii_tolower(Py_CHARMASK(ch));
         p[i] = ch;
     }
     return v;
