diff -Naur -x '*~' Python-2.7.1/configure.in Python-2.7.1-skip-shm-test/configure.in
--- Python-2.7.1/configure.in	2010-11-01 02:47:19.000000000 +0100
+++ Python-2.7.1-skip-shm-test/configure.in	2011-02-02 22:28:42.000000000 +0100
@@ -3609,7 +3609,7 @@
 }
 ]])],
 [ac_cv_posix_semaphores_enabled=yes],
-[ac_cv_posix_semaphores_enabled=no],
+[ac_cv_posix_semaphores_enabled=yes],
 [ac_cv_posix_semaphores_enabled=yes])
 )
 AC_MSG_RESULT($ac_cv_posix_semaphores_enabled)
@@ -3645,8 +3645,8 @@
 }
 ]])],
 [ac_cv_broken_sem_getvalue=no],
-[ac_cv_broken_sem_getvalue=yes],
-[ac_cv_broken_sem_getvalue=yes])
+[ac_cv_broken_sem_getvalue=no],
+[ac_cv_broken_sem_getvalue=no])
 )
 AC_MSG_RESULT($ac_cv_broken_sem_getvalue)
 if test $ac_cv_broken_sem_getvalue = yes
