diff -Naur Python-2.7.orig/configure.in Python-2.7/configure.in
--- Python-2.7.orig/configure.in	2010-05-28 18:28:47.000000000 +0300
+++ Python-2.7/configure.in	2010-08-09 13:56:32.995736702 +0300
@@ -1281,7 +1281,7 @@
 AC_MSG_CHECKING(whether $CC accepts -pthread)
 AC_CACHE_VAL(ac_cv_thread,
 [ac_save_cc="$CC"
-CC="$CC -pthread"
+LIBS="$LIBS -lpthread"
 AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #include <pthread.h>
 
@@ -1317,7 +1317,7 @@
   ac_cv_cxx_thread=yes
 elif test "$ac_cv_pthread" = "yes"
 then 
-  CXX="$CXX -pthread"
+  LIBS="$LIBS -lpthread"
   ac_cv_cxx_thread=yes
 fi
 
@@ -1558,7 +1558,7 @@
 elif test "$ac_cv_kthread" = "yes"
 then CC="$CC -Kthread"
 elif test "$ac_cv_pthread" = "yes"
-then CC="$CC -pthread"
+then LIBS="$LIBS -lpthread"
 fi
 AC_MSG_CHECKING(for pthread_t)
 have_pthread_t=no
@@ -2207,10 +2207,7 @@
     THREADOBJ="Python/thread.o"
 elif test "$ac_cv_pthread" = "yes"
 then
-    CC="$CC -pthread"
-    if test "$ac_cv_cxx_thread" = "yes"; then
-        CXX="$CXX -pthread"
-    fi
+    LIBS="$LIBS -lpthread"
     AC_DEFINE(WITH_THREAD)
     posix_threads=yes
     THREADOBJ="Python/thread.o"
diff -Naur Python-2.7.orig/setup.py Python-2.7/setup.py
--- Python-2.7.orig/setup.py	2010-06-27 15:36:16.000000000 +0300
+++ Python-2.7/setup.py	2010-08-09 13:59:22.780544957 +0300
@@ -578,7 +578,7 @@
         # Operations on audio samples
         # According to #993173, this one should actually work fine on
         # 64-bit platforms.
-        exts.append( Extension('audioop', ['audioop.c']) )
+        exts.append( Extension('audioop', ['audioop.c'], library_dirs=lib_dirs, libraries=['m']) )
 
         # Disabled on 64-bit platforms
         if sys.maxint != 9223372036854775807L:
@@ -1376,7 +1376,7 @@
             # This requires sizeof(int) == sizeof(long) == sizeof(char*)
             dl_inc = find_file('dlfcn.h', [], inc_dirs)
             if (dl_inc is not None) and (platform not in ['atheos']):
-                exts.append( Extension('dl', ['dlmodule.c']) )
+                exts.append( Extension('dl', ['dlmodule.c'], library_dirs=lib_dirs, libraries=['dl']) )
             else:
                 missing.append('dl')
         else:
@@ -1435,6 +1435,7 @@
         if sysconfig.get_config_var('WITH_THREAD'):
             exts.append ( Extension('_multiprocessing', multiprocessing_srcs,
                                     define_macros=macros.items(),
+                                    libraries=['pthread'],
                                     include_dirs=["Modules/_multiprocessing"]))
         else:
             missing.append('_multiprocessing')
@@ -1894,10 +1895,13 @@
                         include_dirs=include_dirs,
                         extra_compile_args=extra_compile_args,
                         extra_link_args=extra_link_args,
-                        libraries=[],
+            library_dirs=lib_dirs,
+                        libraries=['dl'],
                         sources=sources,
                         depends=depends)
         ext_test = Extension('_ctypes_test',
+                libraries=['m'],
+                library_dirs=lib_dirs,
                              sources=['_ctypes/_ctypes_test.c'])
         self.extensions.extend([ext, ext_test])
 
