diff -Naur Python-2.7.orig/Lib/ctypes/util.py Python-2.7/Lib/ctypes/util.py
--- Python-2.7.orig/Lib/ctypes/util.py	2010-03-15 15:42:23.000000000 +0200
+++ Python-2.7/Lib/ctypes/util.py	2010-08-06 16:53:48.326335113 +0300
@@ -140,7 +140,9 @@
             dump = f.read()
             rv = f.close()
             if rv == 10:
-                raise OSError, 'objdump command not found'
+                return os.path.basename(f) #  This is good for GLibc, I think,
+                                           # and a dep on binutils is big (for
+                                           # live CDs).
             f = os.popen(cmd)
             try:
                 data = f.read()
