diff -up Python-2.5.1/Lib/gettext.py.plural Python-2.5.1/Lib/gettext.py
--- Python-2.5.1/Lib/gettext.py.plural	2007-09-10 11:38:57.000000000 -0400
+++ Python-2.5.1/Lib/gettext.py	2007-09-10 11:39:00.000000000 -0400
@@ -299,6 +299,8 @@ class GNUTranslations(NullTranslations):
                     item = item.strip()
                     if not item:
                         continue
+                    if item.startswith("#"):
+                        continue
                     if ':' in item:
                         k, v = item.split(':', 1)
                         k = k.strip().lower()
