diff -up Python-2.6/configure.in.rpath Python-2.6/configure.in
--- Python-2.6/configure.in.rpath	2008-11-24 02:51:06.000000000 -0500
+++ Python-2.6/configure.in	2008-11-24 02:51:21.000000000 -0500
@@ -729,7 +729,7 @@ if test $enable_shared = "yes"; then
 	  ;;
     OSF*)
 	  LDLIBRARY='libpython$(VERSION).so'
-	  BLDLIBRARY='-rpath $(LIBDIR) -L. -lpython$(VERSION)'
+	  BLDLIBRARY='-L. -lpython$(VERSION)'
 	  RUNSHARED=LD_LIBRARY_PATH=`pwd`:${LD_LIBRARY_PATH}
 	  ;;
     atheos*)
