Index: Makefile
===================================================================
--- Makefile.orig
+++ Makefile
@@ -19,21 +19,17 @@
 SHELL = /bin/sh
 
 # The C compiler and linker to use
-CC	= gcc
-CFLAGS	= -g -O2 -Wall
-LD	= gcc
-LDFLAGS =
+LD	= $(CC)
+
 
 # Where the executables should be put
-BINDIR	= /usr/local/bin
+BINDIR	= /usr/bin
 
 # Where the man page should be put
-MANDIR	= /usr/local/man
+MANDIR	= /usr/share/man
 
 # Where figlet will search first for fonts (the ".flf" files).
-DEFAULTFONTDIR = /usr/local/share/figlet
-# Use this definition if you can't put things in /usr/local/share/figlet
-DEFAULTFONTDIR = fonts
+DEFAULTFONTDIR = /usr/share/figlet/fonts
 
 # The filename of the font to be used if no other is specified
 #   (standard.flf is recommended, but any other can be used).
