--- grsync.glade.orig	2010-05-01 18:11:46.078650232 +0300
+++ grsync.glade	2010-05-01 18:07:34.611649224 +0300
@@ -1776,22 +1776,6 @@
               </packing>
             </child>
             <child>
-              <object class="GtkButton" id="button_pause">
-                <property name="label">gtk-media-pause</property>
-                <property name="visible">True</property>
-                <property name="can_focus">True</property>
-                <property name="receives_default">False</property>
-                <property name="tooltip_text" translatable="yes">Pause/resume rsync run</property>
-                <property name="use_stock">True</property>
-                <signal name="clicked" handler="on_button_pause_clicked"/>
-              </object>
-              <packing>
-                <property name="expand">False</property>
-                <property name="fill">False</property>
-                <property name="position">1</property>
-              </packing>
-            </child>
-            <child>
               <object class="GtkButton" id="close">
                 <property name="label">gtk-stop</property>
                 <property name="visible">True</property>
