diff -Naur kdesvn-1.4.1.orig/src/kdesvn.desktop kdesvn-1.4.1/src/kdesvn.desktop
--- kdesvn-1.4.1.orig/src/kdesvn.desktop	2009-12-21 18:20:12.491789574 +0200
+++ kdesvn-1.4.1/src/kdesvn.desktop	2009-12-21 18:24:46.142789775 +0200
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Name=kdesvn
 GenericName=SVN client
+GenericName[tr]=SVN İstemcisi
 GenericName[de]=Graphische Oberfläche für SVN
 GenericName[es]=Cliente SVN
 GenericName[lt]=SVN klientas
@@ -9,6 +10,7 @@
 Type=Application
 DocPath=kdesvn/index.html
 Comment=A Subversion client for KDE
+Comment[tr]=KDE için Subversion İstemcisi
 Comment[de]=Ein Subversionclient für KDE
 Comment[nl]=Een Subversion-client voor KDE
 Comment[es]=Un cliente de Subversion para KDE
