Index: mailcap-2.1.37/mailcap
===================================================================
--- mailcap-2.1.37.orig/mailcap
+++ mailcap-2.1.37/mailcap
@@ -3,11 +3,9 @@
 ###
 
 audio/*; /usr/bin/xdg-open %s
-
+video/*; /usr/bin/xdg-open %s
 image/*; /usr/bin/xdg-open %s
 
-application/msword; /usr/bin/xdg-open %s
-application/pdf; /usr/bin/xdg-open %s
-application/postscript ; /usr/bin/xdg-open %s
+application/*; /usr/bin/xdg-open %s
 
-text/html; /usr/bin/xdg-open %s ; copiousoutput
+text/*; /usr/bin/xdg-open %s ; copiousoutput
