commit 82cd0125fe0e74c4171a86d685fb553516f8d792
Author: Ville Skyttä <ville.skytta@iki.fi>
Date:   Wed Mar 30 19:35:14 2011 +0300

    Sync with IANA as of 2011-03-30.

diff --git a/NEWS b/NEWS
index 5317e1f..b1435c2 100644
--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,8 @@
+mailcap 2.1.xx (201x-xx-xx)
+
+  Ville Skyttä:
+  - mime.types: Sync with IANA as of 2011-03-30.
+
 mailcap 2.1.37 (2011-03-29)
 
   Ville Skyttä:
diff --git a/mime.types b/mime.types
index 6872443..0b72ac8 100644
--- a/mime.types
+++ b/mime.types
@@ -860,6 +860,7 @@ application/vnd.syncml.dm+wbxml			bdm
 application/vnd.syncml.dm+xml			xdm
 application/vnd.syncml+xml			xsm
 application/vnd.tao.intent-module-archive	tao
+application/vnd.tcpdump.pcap			pcap cap dmp
 application/vnd.tmobile-livetv			tmo
 application/vnd.trid.tpt			tpt
 application/vnd.triscape.mxs			mxs

commit 18975ebed85555a7c165efad8c757167771b52d2
Author: Ville Skyttä <ville.skytta@iki.fi>
Date:   Tue Apr 12 09:19:48 2011 +0300

    Sync with IANA as of 2011-04-12.

diff --git a/NEWS b/NEWS
index b1435c2..7049075 100644
--- a/NEWS
+++ b/NEWS
@@ -1,7 +1,7 @@
 mailcap 2.1.xx (201x-xx-xx)
 
   Ville Skyttä:
-  - mime.types: Sync with IANA as of 2011-03-30.
+  - mime.types: Sync with IANA as of 2011-04-12.
 
 mailcap 2.1.37 (2011-03-29)
 
diff --git a/mime.types b/mime.types
index 0b72ac8..32c9041 100644
--- a/mime.types
+++ b/mime.types
@@ -630,6 +630,7 @@ application/vnd.oasis.opendocument.text-master			odm
 application/vnd.oasis.opendocument.text-template		ott
 application/vnd.oasis.opendocument.text-web			oth
 application/vnd.obn
+application/vnd.oftn.l10n+json
 application/vnd.oipf.contentaccessdownload+xml
 application/vnd.oipf.contentaccessstreaming+xml
 application/vnd.oipf.cspg-hexbinary
@@ -936,6 +937,7 @@ application/vnd.yamaha.openscoreformat.osfpvg+xml
 application/vnd.yamaha.remote-setup
 application/vnd.yamaha.smaf-audio		saf
 application/vnd.yamaha.smaf-phrase		spf
+application/vnd.yamaha.through-ngn
 application/vnd.yamaha.tunnel-udpencap
 application/vnd.yellowriver-custom-menu		cmp
 application/vnd.zul				zir zirz

commit d463badf8a48c9d0ed9c2dcc37249335ab84757c
Author: Ville Skyttä <ville.skytta@iki.fi>
Date:   Sun May 8 14:37:50 2011 +0300

    Sync with IANA as of 2011-05-08.

diff --git a/NEWS b/NEWS
index 7049075..a212af5 100644
--- a/NEWS
+++ b/NEWS
@@ -1,7 +1,7 @@
 mailcap 2.1.xx (201x-xx-xx)
 
   Ville Skyttä:
-  - mime.types: Sync with IANA as of 2011-04-12.
+  - mime.types: Sync with IANA as of 2011-05-08.
 
 mailcap 2.1.37 (2011-03-29)
 
diff --git a/mime.types b/mime.types
index 32c9041..ff66143 100644
--- a/mime.types
+++ b/mime.types
@@ -269,6 +269,7 @@ application/vnd.apple.installer+xml		dist distz pkg mpkg
 # m3u: application/x-mpegurl for now
 application/vnd.apple.mpegurl			m3u8
 application/vnd.aristanetworks.swi		swi
+application/vnd.astraea-software.iota		iota
 application/vnd.audiograph			aep
 application/vnd.autopackage			package
 application/vnd.avistar+xml
@@ -664,6 +665,7 @@ application/vnd.oma.dcdc
 application/vnd.oma.dd2+xml			dd2
 application/vnd.oma.drm.risd+xml
 application/vnd.oma.group-usage-list+xml
+application/vnd.oma.pal+xml
 application/vnd.oma.poc.detailed-progress-report+xml
 application/vnd.oma.poc.final-report+xml
 application/vnd.oma.poc.groups+xml

commit fec99d7b5d554d54ce3ce2ed111d30313dc2dbb6
Author: Ville Skyttä <ville.skytta@iki.fi>
Date:   Sat May 14 13:58:18 2011 +0300

    Sync with IANA as of 2011-05-14.

diff --git a/NEWS b/NEWS
index a212af5..203f745 100644
--- a/NEWS
+++ b/NEWS
@@ -1,7 +1,7 @@
 mailcap 2.1.xx (201x-xx-xx)
 
   Ville Skyttä:
-  - mime.types: Sync with IANA as of 2011-05-08.
+  - mime.types: Sync with IANA as of 2011-05-14.
 
 mailcap 2.1.37 (2011-03-29)
 
diff --git a/mime.types b/mime.types
index ff66143..c37e78b 100644
--- a/mime.types
+++ b/mime.types
@@ -1100,7 +1100,8 @@ audio/vnd.dra
 # wav: audio/wav, cpt: application/mac-compactpro
 audio/vnd.dts					dts
 audio/vnd.dts.hd				dtshd
-audio/vnd.dvb.file				dvb
+# dvb: video/vnd.dvb.file
+audio/vnd.dvb.file
 audio/vnd.everad.plj				plj
 # rm: audio/x-pn-realaudio
 audio/vnd.hns.audio
@@ -1312,6 +1313,7 @@ video/vnd.dece.video				uvv uvvv
 video/vnd.directv.mpeg
 video/vnd.directv.mpeg-tts
 video/vnd.dlna.mpeg-tts
+video/vnd.dvb.file				dvb
 video/vnd.fvt					fvt
 # rm: audio/x-pn-realaudio
 video/vnd.hns.video

commit f0cc751453597d05c0fad90ebdef4fefd5434779
Author: Ville Skyttä <ville.skytta@iki.fi>
Date:   Thu May 19 23:07:38 2011 +0300

    Sync with IANA as of 2011-05-19.

diff --git a/NEWS b/NEWS
index 203f745..a17f4ab 100644
--- a/NEWS
+++ b/NEWS
@@ -1,7 +1,7 @@
 mailcap 2.1.xx (201x-xx-xx)
 
   Ville Skyttä:
-  - mime.types: Sync with IANA as of 2011-05-14.
+  - mime.types: Sync with IANA as of 2011-05-19.
 
 mailcap 2.1.37 (2011-03-29)
 
diff --git a/mime.types b/mime.types
index c37e78b..ed2d70f 100644
--- a/mime.types
+++ b/mime.types
@@ -139,6 +139,7 @@ application/octet-stream		bin lha lzh exe class so dll img iso
 application/oda					oda
 application/oebps-package+xml			opf
 application/ogg					ogx
+application/oxps				oxps
 application/parityfec
 # xer: application/xcap-error+xml
 application/patch-ops-error+xml

commit d894da5caaab913e8bed9058cc343c891612f490
Author: Ville Skyttä <ville.skytta@iki.fi>
Date:   Thu May 26 22:56:25 2011 +0300

    Sync with IANA as of 2011-05-26.

diff --git a/NEWS b/NEWS
index a17f4ab..58c68d8 100644
--- a/NEWS
+++ b/NEWS
@@ -1,7 +1,7 @@
 mailcap 2.1.xx (201x-xx-xx)
 
   Ville Skyttä:
-  - mime.types: Sync with IANA as of 2011-05-19.
+  - mime.types: Sync with IANA as of 2011-05-26.
 
 mailcap 2.1.37 (2011-03-29)
 
diff --git a/mime.types b/mime.types
index ed2d70f..37e6769 100644
--- a/mime.types
+++ b/mime.types
@@ -584,6 +584,7 @@ application/vnd.multiad.creator.cif		cif
 application/vnd.music-niff
 application/vnd.musician			mus
 application/vnd.muvee.style			msty
+application/vnd.mynfc				taglet
 application/vnd.ncd.control
 application/vnd.ncd.reference
 application/vnd.nervana				entity request bkm kcm
