diff -Nur pth-2.0.7-orig/configure pth-2.0.7/configure
--- pth-2.0.7-orig/configure	2006-06-08 20:14:48.000000000 +0200
+++ pth-2.0.7/configure	2007-10-21 12:30:48.000000000 +0200
@@ -3054,14 +3054,6 @@
  ;;
 esac
 fi
-case "$CFLAGS" in
-    *-g* ) CFLAGS=`echo "$CFLAGS" |\
-                   sed -e 's/ -g / /g' -e 's/ -g$//' -e 's/^-g //g' -e 's/^-g$//'` ;;
-esac
-case "$CXXFLAGS" in
-    *-g* ) CXXFLAGS=`echo "$CXXFLAGS" |\
-                     sed -e 's/ -g / /g' -e 's/ -g$//' -e 's/^-g //g' -e 's/^-g$//'` ;;
-esac
 msg="disabled"
 
 fi; echo "$as_me:$LINENO: checking for compilation debug mode" >&5
