diff -Nur ettercap-NG-0.7.3-old/share/etter.conf ettercap-NG-0.7.3/share/etter.conf
--- ettercap-NG-0.7.3-old/share/etter.conf	2009-09-15 03:26:07.930452410 +0300
+++ ettercap-NG-0.7.3/share/etter.conf	2009-09-15 03:26:52.367453313 +0300
@@ -137,10 +137,10 @@
 [strings]
 
 # the default encoding to be used for the UTF-8 visualization
-utf8_encoding = "ISO-8859-1"
+utf8_encoding = "ISO-8859-9"
 
 # the command used by the remote_browser plugin
-remote_browser = "mozilla -remote openurl(http://%host%url)"
+remote_browser = "xdg-open http://%host%url"
 
 
 #####################################
