diff -Naur ettercap-NG-0.7.3-orig/src/ec_log.c ettercap-NG-0.7.3-patch/src/ec_log.c
--- ettercap-NG-0.7.3-orig/src/ec_log.c	2004-09-30 16:01:45.000000000 +0000
+++ ettercap-NG-0.7.3-patch/src/ec_log.c	2009-02-16 13:04:23.000000000 +0000
@@ -190,7 +190,7 @@
       if (fd->cfd == NULL)
          SEMIFATAL_ERROR("%s", gzerror(fd->cfd, &zerr));
    } else {
-      fd->fd = open(filename, O_CREAT | O_TRUNC | O_RDWR | O_BINARY);
+      fd->fd = open(filename, O_CREAT | O_TRUNC | O_RDWR | O_BINARY, S_IRUSR | S_IWUSR);
       if (fd->fd == -1)
          SEMIFATAL_ERROR("Can't create %s: %s", filename, strerror(errno));
    }
