--- configure~	2005-05-27 11:11:45.000000000 -0400
+++ configure	2009-02-02 06:47:25.000000000 -0500
@@ -28247,7 +28247,7 @@
       echo "$as_me:$LINENO: checking whether $CC accepts -pthread" >&5
 echo $ECHO_N "checking whether $CC accepts -pthread... $ECHO_C" >&6
       LDFLAGS_store="$LDFLAGS"
-      LDFLAGS="$LDFLAGS -pthread"
+      LIBS="$LIBS -lpthread"
       cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
--- acinclude.m4~	2004-09-17 15:38:34.000000000 -0400
+++ acinclude.m4	2009-02-02 06:48:15.000000000 -0500
@@ -51,7 +51,7 @@
    elif test "$OS" != "MACOSX" -a "$OS" != "WINDOWS"; then
       AC_MSG_CHECKING(whether $CC accepts -pthread)
       LDFLAGS_store="$LDFLAGS"
-      LDFLAGS="$LDFLAGS -pthread"
+      LIBS="$LIBS -lpthread"
       AC_TRY_LINK([
          #include <pthread.h>
          ],
--- src/Makefile.am.link	2005-01-13 04:18:48.000000000 -0500
+++ src/Makefile.am	2009-02-02 07:15:18.000000000 -0500
@@ -140,7 +140,8 @@
 ettercap_LDADD += interfaces/gtk/libec_gtk.a
 endif
 
-ettercap_LDFLAGS = -export-dynamic @EC_LIBS@ 
+ettercap_LDFLAGS = -export-dynamic
+ettercap_LDADD += @EC_LIBS@
 
 CLEANFILES = *.log *.out core* missing/*.o *~
 
--- src/Makefile.in.link	2005-05-27 11:11:54.000000000 -0400
+++ src/Makefile.in	2009-02-02 07:16:39.000000000 -0500
@@ -575,7 +575,8 @@
                  interfaces/daemon/libec_daemon.a \
                  interfaces/text/libec_text.a \
 $(am__append_3) $(am__append_4)
-ettercap_LDFLAGS = -export-dynamic @EC_LIBS@ 
+ettercap_LDFLAGS = -export-dynamic
+ettercap_LDADD += @EC_LIBS@
 CLEANFILES = *.log *.out core* missing/*.o *~
 all: all-recursive
 
