Index: plymouth-0.8.3_20101018/scripts/plymouth-set-default-theme.in
===================================================================
--- plymouth-0.8.3_20101018.orig/scripts/plymouth-set-default-theme.in
+++ plymouth-0.8.3_20101018/scripts/plymouth-set-default-theme.in
@@ -192,5 +192,5 @@ grep -q '^[[]Daemon[]]' ${PLYMOUTH_CONFD
 sed -i -e '/^Theme[[:blank:]]*=.*/d' ${PLYMOUTH_CONFDIR}/plymouthd.conf
 sed -i -e "s/^\([[]Daemon[]]\)\n*/\1\nTheme=${THEME_NAME}/" ${PLYMOUTH_CONFDIR}/plymouthd.conf
 
-[ $DO_INITRD_REBUILD -ne 0 ] && (${PLYMOUTH_LIBEXECDIR}/plymouth/plymouth-update-initrd)
+[ $DO_INITRD_REBUILD -ne 0 ] && /sbin/mkinitramfs
 exit 0
