Index: plymouth-0.8.3_20100929/src/libply-splash-core/ply-boot-splash.c
===================================================================
--- plymouth-0.8.3_20100929.orig/src/libply-splash-core/ply-boot-splash.c
+++ plymouth-0.8.3_20100929/src/libply-splash-core/ply-boot-splash.c
@@ -850,7 +850,7 @@ main (int    argc,
 
   if (!ply_renderer_open (renderer))
     {
-      perror ("could not open renderer /dev/fb");
+      perror ("could not open renderer /dev/fb0");
       ply_renderer_free (renderer);
       return errno;
     }
Index: plymouth-0.8.3_20100929/src/main.c
===================================================================
--- plymouth-0.8.3_20100929.orig/src/main.c
+++ plymouth-0.8.3_20100929/src/main.c
@@ -1452,7 +1452,7 @@ add_default_displays_and_keyboard (state
 
   if (!ply_renderer_open (renderer))
     {
-      ply_trace ("could not open renderer /dev/fb");
+      ply_trace ("could not open renderer /dev/fb0");
       ply_renderer_free (renderer);
       ply_terminal_free (terminal);
 
Index: plymouth-0.8.3_20100929/src/plugins/renderers/drm/plugin.c
===================================================================
--- plymouth-0.8.3_20100929.orig/src/plugins/renderers/drm/plugin.c
+++ plymouth-0.8.3_20100929/src/plugins/renderers/drm/plugin.c
@@ -923,7 +923,7 @@ create_heads_for_active_connectors (ply_
 #ifdef PLY_ENABLE_GDM_TRANSITION
   /* If the driver doesn't support mapping the fb console
    * then we can't get a smooth crossfade transition to
-   * the display manager unless we use the /dev/fb interface
+   * the display manager unless we use the /dev/fb0 interface
    * or the plymouth deactivate interface.
    *
    * In multihead configurations, we'd rather have working
Index: plymouth-0.8.3_20100929/src/plugins/renderers/frame-buffer/plugin.c
===================================================================
--- plymouth-0.8.3_20100929.orig/src/plugins/renderers/frame-buffer/plugin.c
+++ plymouth-0.8.3_20100929/src/plugins/renderers/frame-buffer/plugin.c
@@ -56,7 +56,7 @@
 #include "ply-renderer-plugin.h"
 
 #ifndef PLY_FRAME_BUFFER_DEFAULT_FB_DEVICE_NAME
-#define PLY_FRAME_BUFFER_DEFAULT_FB_DEVICE_NAME "/dev/fb"
+#define PLY_FRAME_BUFFER_DEFAULT_FB_DEVICE_NAME "/dev/fb0"
 #endif
 
 struct _ply_renderer_head
