Description: Add missing settings to fix upstream tests
Origin: vendor
Forwarded: http://bugs.php.net/50796
Last-Update: 2010-01-18

--- a/ext/soap/tests/server009.phpt
+++ b/ext/soap/tests/server009.phpt
@@ -10,6 +10,7 @@ SOAP Server 9: setclass and setpersisten
 --INI--
 session.auto_start=1
 session.save_handler=files
+session.save_path=temp_session_store
 --FILE--
 <?php
 class foo {
--- a/ext/standard/tests/general_functions/phpinfo.phpt
+++ b/ext/standard/tests/general_functions/phpinfo.phpt
@@ -1,5 +1,7 @@
 --TEST--
 phpinfo()
+--SKIPIF--
+<?php die("SKIP phpinfo - test suite's handling of "%s" is incompatible with this test"); ?>
 --FILE--
 <?php
 var_dump(phpinfo());
