--- a/ext/mssql/php_mssql.c
+++ b/ext/mssql/php_mssql.c
@@ -1106,6 +1106,14 @@ static void php_mssql_get_column_content
 			return;
 		}
 
+		if (res_length == 0) {
+			ZVAL_NULL(result);
+			return;
+		} else if (res_length < 0) {
+			ZVAL_FALSE(result);
+			return;
+		}
+
 		res_buf = (unsigned char *) emalloc(res_length+1);
 		bin = ((DBBINARY *)dbdata(mssql_ptr->link, offset));
 		res_buf[res_length] = '\0';
