--- a/ext/session/session.c
+++ b/ext/session/session.c
@@ -2226,6 +2226,12 @@ static PHP_MSHUTDOWN_FUNCTION(session) /
 	ps_serializers[PREDEFINED_SERIALIZERS].name = NULL;
 	memset(&ps_modules[PREDEFINED_MODULES], 0, (MAX_MODULES-PREDEFINED_MODULES)*sizeof(ps_module *));
 
+	// reset the callback pointer to null
+	php_session_rfc1867_orig_callback = NULL;
+	if(php_rfc1867_callback == php_session_rfc1867_callback) {
+		php_rfc1867_callback = NULL;
+	}
+
 	return SUCCESS;
 }
 /* }}} */
