--- a/configure.in
+++ b/configure.in
@@ -136,6 +136,8 @@ AC_DEFUN([PHP_EXT_DIR],[ext/$1])dnl
 AC_DEFUN([PHP_EXT_SRCDIR],[$abs_srcdir/ext/$1])dnl
 AC_DEFUN([PHP_ALWAYS_SHARED],[])dnl
 
+AC_DEFINE([_GNU_SOURCE], 1, [Define to enable GNU C Library extensions])
+
 dnl Setting up the PHP version based on the information above.
 dnl -------------------------------------------------------------------------
 
--- a/ext/interbase/interbase.c
+++ b/ext/interbase/interbase.c
@@ -24,7 +24,6 @@
 #include "config.h"
 #endif
 
-#define _GNU_SOURCE
 
 #include "php.h"
 
--- a/ext/pdo_firebird/firebird_driver.c
+++ b/ext/pdo_firebird/firebird_driver.c
@@ -22,7 +22,6 @@
 #include "config.h"
 #endif
 
-#define _GNU_SOURCE
 
 #include "php.h"
 #ifdef ZEND_ENGINE_2
--- a/ext/standard/file.c
+++ b/ext/standard/file.c
@@ -112,9 +112,6 @@ php_file_globals file_globals;
 #endif
 
 #if defined(HAVE_FNMATCH) && !defined(PHP_WIN32)
-# ifndef _GNU_SOURCE
-#  define _GNU_SOURCE
-# endif
 # include <fnmatch.h>
 #endif
 
--- a/ext/zlib/zlib_fopen_wrapper.c
+++ b/ext/zlib/zlib_fopen_wrapper.c
@@ -19,8 +19,6 @@
 
 /* $Id$ */
 
-#define _GNU_SOURCE
-
 #include "php.h"
 #include "php_zlib.h"
 #include "fopen_wrappers.h"
--- a/main/php.h
+++ b/main/php.h
@@ -30,6 +30,7 @@
 #define PHP_HAVE_STREAMS
 #define YYDEBUG 0
 
+#include "php_config.h"
 #include "php_version.h"
 #include "zend.h"
 #include "zend_qsort.h"
--- a/main/streams/cast.c
+++ b/main/streams/cast.c
@@ -18,7 +18,6 @@
 
 /* $Id$ */
 
-#define _GNU_SOURCE
 #include "php.h"
 #include "php_globals.h"
 #include "php_network.h"
--- a/main/streams/memory.c
+++ b/main/streams/memory.c
@@ -18,7 +18,6 @@
 
 /* $Id$ */
 
-#define _GNU_SOURCE
 #include "php.h"
 
 PHPAPI int php_url_decode(char *str, int len);
--- a/main/streams/streams.c
+++ b/main/streams/streams.c
@@ -21,7 +21,6 @@
 
 /* $Id$ */
 
-#define _GNU_SOURCE
 #include "php.h"
 #include "php_globals.h"
 #include "php_network.h"
--- a/Zend/zend_language_parser.c
+++ b/Zend/zend_language_parser.c
@@ -67,6 +67,8 @@
 #define yydebug zenddebug
 #define yynerrs zendnerrs
 
+#include <string.h>
+
 
 /* Tokens.  */
 #ifndef YYTOKENTYPE
