--- a/ext/session/session.c
+++ b/ext/session/session.c
@@ -2192,8 +2192,10 @@ static PHP_MINIT_FUNCTION(session) /* {{
 #ifdef HAVE_LIBMM
 	PHP_MINIT(ps_mm) (INIT_FUNC_ARGS_PASSTHRU);
 #endif
-	php_session_rfc1867_orig_callback = php_rfc1867_callback;
-	php_rfc1867_callback = php_session_rfc1867_callback;
+	if (php_rfc1867_callback != php_session_rfc1867_callback) {
+		php_session_rfc1867_orig_callback = php_rfc1867_callback;
+		php_rfc1867_callback = php_session_rfc1867_callback;
+	}
 
 	/* Register interface */
 	INIT_CLASS_ENTRY(ce, PS_IFACE_NAME, php_session_iface_functions);
@@ -2384,13 +2386,14 @@ static int php_session_rfc1867_callback(
 	php_session_rfc1867_progress *progress;
 	int retval = SUCCESS;
 
-	if (php_session_rfc1867_orig_callback) {
-		retval = php_session_rfc1867_orig_callback(event, event_data, extra TSRMLS_CC);
-	}
 	if (!PS(rfc1867_enabled)) {
 		return retval;
 	}
 
+	if (php_session_rfc1867_orig_callback) {
+		retval = php_session_rfc1867_orig_callback(event, event_data, extra TSRMLS_CC);
+	}
+
 	progress = PS(rfc1867_progress);
 
 	switch(event) {
