diff -Naur epdfview-0.1.6/src/gtk/MainView.cxx epdfview-0.1.6.tpg/src/gtk/MainView.cxx
--- epdfview-0.1.6/src/gtk/MainView.cxx 2007-02-23 07:59:11.000000000 +0000
+++ epdfview-0.1.6.tpg/src/gtk/MainView.cxx 2009-02-11 16:03:11.000000000 +0000
@@ -592,9 +592,10 @@
             GTK_DIALOG_DESTROY_WITH_PARENT,
             GTK_MESSAGE_ERROR,
             GTK_BUTTONS_OK,
+            "%s",
             title);
     gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG(errorDialog),
-                                              body);
+                                              "%s", body);
     gtk_dialog_run (GTK_DIALOG (errorDialog));
     gtk_widget_destroy (errorDialog);
 }

