diff -Nuar epdfview-0.1.7.orig//data/epdfview.desktop epdfview-0.1.7//data/epdfview.desktop
--- epdfview-0.1.7.orig//data/epdfview.desktop	2011-02-09 11:01:15.313000008 +0200
+++ epdfview-0.1.7//data/epdfview.desktop	2011-02-09 14:44:11.540000001 +0200
@@ -3,14 +3,17 @@
 Comment=Lightweight PDF document viewer
 Comment[ca]=Visor de documents PDF lleuger
 Comment[es]=Visor de documentos PDF ligero
+Comment[tr]=Hafif bir PDF okuyucu
 Exec=epdfview %f
 GenericName=PDF Viewer
 GenericName[ca]=Visor PDF
 GenericName[es]=Visor PDF
-Icon=icon_epdfview-48
+GenericName[tr]=PDF Okuyucu
+Icon=/usr/share/pixmaps/epdfview.png
 Name=ePDFViewer
 Name[ca]=ePDFViewer
 Name[es]=ePDFViewer
+Name[tr]=ePDFViewer
 MimeType=application/pdf;
 Terminal=false
 Type=Application
