diff -Nuar epdfview-0.1.8.orig/src/Config.cxx epdfview-0.1.8/src/Config.cxx
--- epdfview-0.1.8.orig/src/Config.cxx	2011-06-30 13:00:12.039000117 +0300
+++ epdfview-0.1.8/src/Config.cxx	2011-06-30 13:00:38.560000117 +0300
@@ -22,7 +22,7 @@
 using namespace ePDFView;
 
 // Constants
-static const gchar *DEFAULT_EXTERNAL_BROWSER_COMMAND_LINE = "firefox %s";
+static const gchar *DEFAULT_EXTERNAL_BROWSER_COMMAND_LINE = "xdg-open %s";
 static const gchar *DEFAULT_OPEN_FILE_FOLDER = NULL;
 static const gchar *DEFAULT_SAVE_FILE_FOLDER = NULL;
 static const gboolean DEFAULT_SHOW_STATUSBAR = TRUE;
