diff -ur cracklib-2.8.15.orig/python/Makefile.am cracklib-2.8.15/python/Makefile.am
--- cracklib-2.8.15.orig/python/Makefile.am	2009-11-19 00:58:20.000000000 +0100
+++ cracklib-2.8.15/python/Makefile.am	2009-11-20 16:58:54.000000000 +0100
@@ -1,5 +1,5 @@
 if BUILD_PYTHON
-python_PYTHON = cracklib.py
+pyexec_PYTHON = cracklib.py
 pyexec_LTLIBRARIES = _cracklibmodule.la
 AM_CFLAGS = -I$(top_srcdir)/lib
 _cracklibmodule_la_LDFLAGS = -module -avoid-version $(top_builddir)/lib/libcrack.la
Only in cracklib-2.8.15/python: Makefile.am.orig
