diff -Nur cracklib-2.8.13-old/python/Makefile.in cracklib-2.8.13/python/Makefile.in
--- cracklib-2.8.13-old/python/Makefile.in	2009-02-03 02:56:55.000000000 +0200
+++ cracklib-2.8.13/python/Makefile.in	2009-02-03 02:59:47.000000000 +0200
@@ -332,9 +332,9 @@
 	done; \
 	if test -n "$$dlist"; then \
 	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pythondir)" $$dlist; \
+	    echo "Not compiling Python files as it breaks sandbox"; \
 	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pythondir)" $$dlist; \
+	    echo "Not compiling Python files as it breaks sandbox"; \
 	  fi; \
 	else :; fi
 
