diff -Nuar ConsoleKit2-0.9.3.orig/src/ck-session.c ConsoleKit2-0.9.3/src/ck-session.c
--- ConsoleKit2-0.9.3.orig/src/ck-session.c	2015-05-18 06:43:04.000000000 +0300
+++ ConsoleKit2-0.9.3/src/ck-session.c	2015-06-13 22:58:50.560971838 +0300
@@ -836,7 +836,11 @@
 
         g_return_val_if_fail (CK_IS_SESSION (cksession), FALSE);
 
-        console_kit_session_complete_get_login_session_id (cksession, context, session->priv->login_session_id);
+        if (session->priv->login_session_id == NULL) {
+                throw_error (context, CK_SESSION_ERROR_FAILED, "no login session id");
+        } else {
+                console_kit_session_complete_get_login_session_id (cksession, context, session->priv->login_session_id);
+        }
         return TRUE;
 }
 
