--- runtime/ftplugin/xml.vim.orig	2010-05-15 14:04:03.000000000 +0300
+++ runtime/ftplugin/xml.vim	2010-08-21 01:12:43.218840497 +0300
@@ -13,13 +13,6 @@
 set cpo-=C
 
 setlocal commentstring=<!--%s-->
-setlocal comments=s:<!--,m:\ \ \ \ \ ,e:-->
-
-setlocal formatoptions-=t
-if !exists("g:ft_xml_autocomment") || (g:ft_xml_autocomment == 1)
-    setlocal formatoptions+=croql
-endif
-
 
 " XML:  thanks to Johannes Zellner and Akbar Ibrahim
 " - case sensitive
@@ -56,7 +49,7 @@
 endif
 
 " Undo the stuff we changed.
-let b:undo_ftplugin = "setlocal commentstring< comments< formatoptions<" .
+let b:undo_ftplugin = "setlocal cms<" .
 		\     " | unlet! b:match_ignorecase b:match_words b:browsefilter"
 
 " Restore the saved compatibility options.
