--- ktranslatoid.cpp.orig	2009-08-05 23:15:51.000000000 +0300
+++ ktranslatoid.cpp	2009-08-18 20:23:33.232609287 +0300
@@ -105,10 +105,10 @@
 
     Plasma::ToolTipManager::self()->setContent(this, data);
 
-    m_buttonChange->setIcon(KIcon("system-switch-user.png"));
+    m_buttonChange->nativeWidget()->setIcon(KIcon("system-switch-user.png"));
     m_buttonChange->nativeWidget()->setToolTip(i18n("Invert the translation"));
 
-    m_buttonTranslate->setIcon(KIcon("tools-wizard.png"));
+    m_buttonTranslate->nativeWidget()->setIcon(KIcon("tools-wizard.png"));
     m_buttonTranslate->setText(i18n("Translate"));
     m_buttonTranslate->nativeWidget()->setToolTip(i18n("Click here to translate"));
 
