Index: jamin-0.95.0/data/jamin.desktop.in
===================================================================
--- jamin-0.95.0.orig/data/jamin.desktop.in
+++ jamin-0.95.0/data/jamin.desktop.in
@@ -2,8 +2,10 @@
 Encoding=UTF-8
 Name=JAMin
 Name[ru]=JAMin
-GenericName=
+GenericName=JACK Audio Mastering Interface
+GenericName[tr]=JACK Mastering Arayüzü
 Comment=JACK Audio Mastering interface
+Comment[tr]=JACK Mastering Arayüzü
 Comment[ru]=JAMin -- приложение для мастеринга звука
 Categories=Application;AudioVideo;
 Exec=jamin
