--- cellwriter-1.3.4_20110312/src//cellwidget.c	2011-10-07 10:30:56.772000031 +0000
+++ cellwriter-1.3.4_20110312/src//cellwidget.c	2011-10-07 10:30:41.765000030 +0000
@@ -1833,7 +1833,7 @@
         block = unicode_blocks + training_block;
         range = block->end - block->start + 1;
         training = TRUE;
-        pack_cells((range + cell_cols - 1) / cell_cols, cell_cols);
+        pack_cells((range + block->additional[0] + cell_cols - 1) / cell_cols, cell_cols);
 
         /* Preset all of the characters for training */
         for (i = 0, pos = 0; i < range; i++) {
@@ -1846,6 +1846,18 @@
                 cells[pos].alts[0] = NULL;
                 cells[pos++].flags = 0;
         }
+
+        for (i = 0; i < block->additional[0]; i++) {
+                unsigned short ch;
+
+                ch = block->additional[i+1];
+                if (char_disabled(ch))
+                        continue;
+                cells[pos].ch = ch;
+                cells[pos].alts[0] = NULL;
+                cells[pos++].flags = 0;
+        }
+
         range = pos;
         for (; pos < cell_rows * cell_cols; pos++)
                 clear_cell(pos);
--- cellwriter-1.3.4_20110312/src//common.h	2011-03-12 06:52:39.000000000 +0000
+++ cellwriter-1.3.4_20110312/src//common.h	2011-10-07 10:30:41.764000030 +0000
@@ -47,6 +47,7 @@
         short enabled;
         const int start, end;
         const char *name;
+        const int additional[16];
 } UnicodeBlock;
 
 extern UnicodeBlock unicode_blocks[];
--- cellwriter-1.3.4_20110312/src//window.c	2011-10-07 10:30:56.779000031 +0000
+++ cellwriter-1.3.4_20110312/src//window.c	2011-10-07 10:36:05.441000031 +0000
@@ -837,6 +837,7 @@
         { TRUE,  0x0080, 0x00FF, "Latin-1 Supplement" },
         { TRUE,  0x0100, 0x017F, "Latin Extended-A" },
         { FALSE, 0x0180, 0x024F, "Latin Extended-B" },
+        { TRUE,  0x0000, 0x007F, "Turkish", {12, 0x0131, 0x0130, 0x015E, 0x015F, 0x011E, 0x011F, 0x00E7, 0x00C7, 0x00F6, 0x00D6, 0x00FC, 0x00DC} },
         { FALSE, 0x0250, 0x02AF, "IPA Extensions" },
         { FALSE, 0x02B0, 0x02FF, "Spacing Modifier Letters" },
         { FALSE, 0x0300, 0x036F, "Combining Diacritical Marks" },
