--- src/cellwidget.c	2008/10/28 05:30:40	1.1
+++ src/cellwidget.c	2008/10/28 05:41:44
@@ -1474,6 +1474,15 @@
         }
 #endif
 
+#if 0
+	/* This if statement is the reason it quits working when I
+	   accidently go outside the box while writing with the tablet
+	   stylus. It is very annoying to have it do that, because the
+	   only think I can do to get Cellwriter working again is to
+	   kill it and restart it. Disabling Xinput is not the right
+	   solution, so let's try something else. -- karlheg
+	 */
+
         /* If we are getting invalid output from this device with XInput
            enabled, try disabling it */
         if ((x < 0 || x > drawing_area->allocation.width ||
@@ -1484,6 +1493,13 @@
                 gdk_device_set_mode(event->device, GDK_MODE_DISABLED);
                 return TRUE;
         }
+#endif
+
+	/* Constrain the stroke to be inside the box. */
+	if (x < 0) x = 0;
+	if (y < 0) y = 0;
+	if (x > drawing_area->allocation.width) x = drawing_area->allocation.width;
+	if (y > drawing_area->allocation.height) y = drawing_area->allocation.height;
 
         /* Check where the pointer is */
         check_cell(x, y, event->device);
