Index: cellwriter-1.3.4_20110312/src/main.c
===================================================================
--- cellwriter-1.3.4_20110312.orig/src/main.c
+++ cellwriter-1.3.4_20110312/src/main.c
@@ -845,6 +845,11 @@ int main(int argc, char *argv[])
         GError *error;
         const char *token;
 
+        /* Set GDK_NATIVE_WINDOWS to 1 to workaround a nasty KWin bug
+         * avoiding multi-stroke input
+         * https://bugs.kde.org/show_bug.cgi?id=234926 */
+        putenv("GDK_NATIVE_WINDOWS=1");
+
         /* Initialize GTK+ */
         error = NULL;
         if (!gtk_init_with_args(&argc, &argv,
