Index: cellwriter-1.3.4_20110312/src/window.c
===================================================================
--- cellwriter-1.3.4_20110312.orig/src/window.c
+++ cellwriter-1.3.4_20110312/src/window.c
@@ -690,8 +690,9 @@ void window_create(void)
         /* Keys button */
         keys_button = gtk_toggle_button_new_with_label(_("Keys"));
         gtk_button_set_focus_on_click(GTK_BUTTON(keys_button), FALSE);
-        image = gtk_image_new_from_icon_name("keyboard", GTK_ICON_SIZE_BUTTON);
-        gtk_button_set_image(GTK_BUTTON(keys_button), image);
+        gtk_button_set_image(GTK_BUTTON(keys_button),
+                             gtk_image_new_from_stock(GTK_STOCK_BOLD,
+                                                      GTK_ICON_SIZE_BUTTON));
         gtk_button_set_relief(GTK_BUTTON(keys_button), GTK_RELIEF_NONE);
         gtk_box_pack_start(GTK_BOX(bottom_box), keys_button, FALSE, FALSE, 0);
         g_signal_connect(G_OBJECT(keys_button), "toggled",
