--- el.ini.orig	2007-11-14 12:56:25.000000000 +0200
+++ el.ini	2007-11-27 17:10:59.000000000 +0200
@@ -36,7 +36,7 @@
 For Windows users, the default is "c:\Program Files\Eternal Lands\".
 For Gentoo users, the ebuild installs in /usr/share/games/eternal-lands.
 
-#data_dir = "c:\Program Files\Eternal Lands\"
+#data_dir = "/usr/share/eternal-lands/"
 
 The no_adjust_shadows directive, if set on 1, tells the engine not to disable the shadows if
 the frame rate is too low.
@@ -180,7 +180,7 @@
 2: log both chat and server messages to chat_log.txt
 3: log chat to chat_log.txt, and server messages to srv_log.txt
 
-#log_chat = 2
+#log_chat = 3
 
 You can adjust the size of the text used for the players name tags on the screen.  A value
 of 1.0 is the default, and you can make the text bigger or smaller by raising or lowering the values.
@@ -217,7 +217,7 @@
 #browser = ""
 
 Note that, because of Micro$oft's infinite wisdom, invoking a command is done differently
-in Win9* and in Windows NT ® ©, so you have to manually select how to invoke your browser.
+in Win9* and in Windows NT ï¿½ ï¿½, so you have to manually select how to invoke your browser.
 Be sure to have only ONE selected browser, otherwise the current browser thingy is likely not 
 to work. 
 
@@ -234,7 +234,7 @@
 selected. That is, deselect the windows browser
 browser = konqueror
 browser = opera
-browser = mozilla
+#browser = firefox
 browser = netscape
 
 Choose your language - all languages available are located in the ./languages/ folder in your eternal lands directory. 
