Index: work/common/usr/share/applications/amdcccle.desktop
===================================================================
--- work.orig/common/usr/share/applications/amdcccle.desktop
+++ work/common/usr/share/applications/amdcccle.desktop
@@ -1,9 +1,10 @@
 [Desktop Entry]
 Encoding=UTF-8
 Type=Application
-Exec=amdcccle
-Icon=ccc_large
+Exec=amdcccle
+Icon=amdcccle
 Categories=GNOME;KDE;GTK;Qt;HardwareSettings;Settings;
+TryExec=/sys/module/fglrx
 
 
 Name=AMD Catalyst Control Center
@@ -44,7 +45,7 @@ Name[sv]=AMD Catalyst Control Center
 Comment[sv]=Möjliggör konfiguration av inställningar för AMD-grafikmaskinvara.
 Name[th]=AMD Catalyst Control Center
 Comment[th]=ยอมให้ตั้งค่าฮาร์ดแวร์กราฟิก AMD
-Name[tr]=AMD Catalyst Control Center
+Name[tr]=AMD Catalyst Denetim Merkezi
 Comment[tr]=AMD grafik donanım ayarlarının yapılandırılmasına olanak sağlar.
 Name[zh_CN]=AMD Catalyst Control Center
 Comment[zh_CN]=允许配置 AMD 图形硬件设置。
Index: work/common/usr/share/applications/amdccclesu.desktop
===================================================================
--- work.orig/common/usr/share/applications/amdccclesu.desktop
+++ work/common/usr/share/applications/amdccclesu.desktop
@@ -1,9 +1,10 @@
 [Desktop Entry]
 Encoding=UTF-8
 Type=Application
-Exec=amdxdg-su -c amdcccle
+Exec=xdg-su -c amdcccle
 Categories=GNOME;KDE;GTK;Qt;HardwareSettings;Settings;
-Icon=ccc_large
+Icon=amdcccle
+TryExec=/sys/module/fglrx
 
 Name=AMD Catalyst Control Center (Administrative)
 Comment=Allows the configuration of the AMD graphics hardware settings that requires administrative access.
@@ -43,7 +44,7 @@ Name[sv]=AMD Catalyst Control Center (Ad
 Comment[sv]=Möjliggör konfiguration av inställningar för AMD-grafikmaskinvara som kräver administrativ åtkomst.
 Name[th]=AMD Catalyst Control Center (สิทธิ์ผู้ดูแลระบบ)
 Comment[th]=ยอมให้ตั้งค่าฮาร์ดแวร์กราฟิก AMD ที่ต้องใช้สิทธิ์ของผู้ดูแลระบบ
-Name[tr]=AMD Catalyst Control Center (Yönetici)
+Name[tr]=AMD Catalyst Denetim Merkezi (Yönetici)
 Comment[tr]=Yönetici erişimi gerektiren AMD grafik donanım ayarlarının yapılandırılmasına olanak sağlar.
 Name[zh_CN]=AMD Catalyst Control Center（管理）
 Comment[zh_CN]=允许配置需要管理权限才能访问的 AMD 图形硬件设置。
