Index: kdepim-4.7.3/akregator/src/akregator_part.cpp
===================================================================
--- kdepim-4.7.3.orig/akregator/src/akregator_part.cpp
+++ kdepim-4.7.3/akregator/src/akregator_part.cpp
@@ -264,7 +264,7 @@ Part::Part( QWidget *parentWidget, QObje
     connect(m_autosaveTimer, SIGNAL(timeout()), this, SLOT(slotSaveFeedList()));
     m_autosaveTimer->start(5*60*1000); // 5 minutes
 
-    QString useragent = QString( "Akregator/%1; syndication" ).arg( KDEPIM_VERSION );
+    QString useragent = QString( "Akregator/%1; syndication Pardus" ).arg( KDEPIM_VERSION );
 
     if( !Settings::customUserAgent().isEmpty() )
         useragent = Settings::customUserAgent();
