Index: kdepim-4.2.4/akregator/src/akregator_part.cpp
===================================================================
--- kdepim-4.2.4.orig/akregator/src/akregator_part.cpp
+++ kdepim-4.2.4/akregator/src/akregator_part.cpp
@@ -96,6 +96,20 @@ namespace {
         QDomElement body = doc.createElement( "body" );
         root.appendChild(body);
 
+        QDomElement pardusMainFolder = doc.createElement( "outline" );
+        pardusMainFolder.setAttribute("text","Pardus");
+        body.appendChild(pardusMainFolder);
+
+        QDomElement planet = doc.createElement( "outline" );
+        planet.setAttribute("text",i18n("Planet Pardus"));
+        planet.setAttribute("xmlUrl","http://planet.pardus.org.tr/rss20.xml");
+        pardusMainFolder.appendChild(planet);
+
+        QDomElement gezegen = doc.createElement( "outline" );
+        gezegen.setAttribute("text",i18n("Pardus Dünyası"));
+        gezegen.setAttribute("xmlUrl","http://gezegen.pardus.org.tr/rss20.xml");
+        pardusMainFolder.appendChild(gezegen);
+
         QDomElement mainFolder = doc.createElement( "outline" );
         mainFolder.setAttribute("text","KDE");
         body.appendChild(mainFolder);
