Index: kdepim-4.7.3/messagecomposer/messagecomposer.kcfg.cmake
===================================================================
--- kdepim-4.7.3.orig/messagecomposer/messagecomposer.kcfg.cmake
+++ kdepim-4.7.3/messagecomposer/messagecomposer.kcfg.cmake
@@ -50,7 +50,7 @@ xsi:schemaLocation="http://www.kde.org/s
     </entry>
 
     <entry name="PreferredCharsets" key="pref-charsets" type="StringList">
-        <default>us-ascii,iso-8859-1,locale,utf-8</default>
+        <default>locale,utf-8</default>
     </entry>
 
     <entry name="AutoTextSignature" type="String" key="signature">
