diff -Nuar kdepim-4.11.1.orig/messagecomposer/settings/messagecomposer.kcfg.cmake kdepim-4.11.1/messagecomposer/settings/messagecomposer.kcfg.cmake
--- kdepim-4.11.1.orig/messagecomposer/settings/messagecomposer.kcfg.cmake	2013-08-28 20:05:43.415155838 +0300
+++ kdepim-4.11.1/messagecomposer/settings/messagecomposer.kcfg.cmake	2013-09-08 08:13:36.032089309 +0300
@@ -50,7 +50,7 @@
     </entry>
 
     <entry name="PreferredCharsets" key="pref-charsets" type="StringList">
-        <default>us-ascii,iso-8859-1,locale,utf-8</default>
+        <default>locale,utf-8</default>
     </entry>
 
     <entry name="AutoTextSignature" type="String" key="signature">
