Index: kdepim-4.7.3/kmail/kmail_addattachmentservicemenu.desktop
===================================================================
--- kdepim-4.7.3.orig/kmail/kmail_addattachmentservicemenu.desktop
+++ kdepim-4.7.3/kmail/kmail_addattachmentservicemenu.desktop
@@ -1,9 +1,11 @@
 [Desktop Entry]
 Type=Service
-ServiceTypes=KonqPopupMenu/Plugin
+Name=Send File As Attachment
+Name[tr]=Eklenti Olarak Gönder
+X-KDE-ServiceTypes=KonqPopupMenu/Plugin,all/allfiles
+X-KDE-ExcludeServiceTypes=application/x-iso,kdedevice/*,inode/directory
 MimeType=all/all;
 Actions=sendto;
-Icon=kmail
 
 [Desktop Action sendto]
 Name=Send To
@@ -38,9 +40,10 @@ Name[sr@ijekavian]=Пошаљи
 Name[sr@ijekavianlatin]=Pošalji
 Name[sr@latin]=Pošalji
 Name[sv]=Skicka till
-Name[tr]=Alıcı
+Name[tr]=Eklenti Olarak Gönder
 Name[ug]=يوللاش
 Name[uk]=Надіслати до
 Name[x-test]=xxSend Toxx
 Name[zh_TW]=傳送到
-Exec=kmail --attach %F
+Exec=kmail --composer --attach %F `echo attachment=\'file://%F\'|sed 's/ \//,file:\/\/\//g'`
+Icon=kmail
