diff -p -up mono-2.6/mono/metadata/icall.c.format-strings mono-2.6/mono/metadata/icall.c
diff -p -up mono-2.6/mono/metadata/reflection.c.format-strings mono-2.6/mono/metadata/reflection.c
diff -p -up mono-2.6/support/syslog.c.format-strings mono-2.6/support/syslog.c
--- mono-2.6/support/syslog.c.format-strings	2009-09-25 16:58:14.000000000 +0200
+++ mono-2.6/support/syslog.c	2009-10-01 22:57:11.000000000 +0200
@@ -34,7 +34,7 @@ Mono_Posix_Syscall_closelog (void)
 int
 Mono_Posix_Syscall_syslog (int priority, const char* message)
 {
-	syslog (priority, message);
+	syslog (priority, "%s", message);
 	return 0;
 }
 
