diff -p -up mono-2.2/data/config.in.dllmap mono-2.2/data/config.in
--- mono-2.2/data/config.in.dllmap	2008-09-02 17:42:56.000000000 +0200
+++ mono-2.2/data/config.in	2009-01-14 09:56:39.000000000 +0100
@@ -5,12 +5,14 @@
 	<dllmap dll="intl" name="bind_textdomain_codeset" target="@LIBC@" os="solaris"/>
 	<dllmap dll="libintl" name="bind_textdomain_codeset" target="@LIBC@" os="solaris"/>
 	<dllmap dll="libintl" target="@INTL@" os="!windows"/>
-	<dllmap dll="i:libxslt.dll" target="libxslt@libsuffix@" os="!windows"/>
-	<dllmap dll="i:odbc32.dll" target="libodbc@libsuffix@" os="!windows"/>
+	<dllmap dll="i:libxslt.dll" target="libxslt@libsuffix@.1" os="!windows"/>
+	<dllmap dll="i:odbc32.dll" target="libodbc@libsuffix@.1" os="!windows"/>
 	<dllmap dll="i:odbc32.dll" target="libiodbc.dylib" os="osx"/>
 	<dllmap dll="oci" target="libclntsh@libsuffix@" os="!windows"/>
 	<dllmap dll="db2cli" target="libdb2_36@libsuffix@" os="!windows"/>
 	<dllmap dll="MonoPosixHelper" target="libMonoPosixHelper@libsuffix@" os="!windows" />
+	<dllmap dll="gdiplus.dll" target="libgdiplus.so.0"/>
+	<dllmap dll="gluezilla" target="libgluezilla.so.0"/> 
 	<dllmap dll="i:msvcrt" target="@LIBC@" os="!windows"/>
 	<dllmap dll="i:msvcrt.dll" target="@LIBC@" os="!windows"/>
 	<dllmap dll="sqlite" target="@SQLITE@" os="!windows"/>
