--- ./Makefile_orig	2007-09-30 23:24:38.000000000 +0300
+++ ./Makefile	2009-11-19 19:58:04.632340440 +0200
@@ -2,13 +2,13 @@
 
 GIMPTOOL=gimptool-2.0
 
-GIMP_LDFLAGS=`$(GIMPTOOL) --libs`
+GIMP_LDFLAGS=-lgimp-2.0 -lgimpui-2.0 -lgimpwidgets-2.0
 GIMP_CFLAGS=`$(GIMPTOOL) --cflags`
 
 UMFPACK_CFLAGS=-I/usr/include/suitesparse -I/usr/include/umfpack
-UMFPACK_LDFLAGS=-lumfpack -lamd -lcholmod -lm
+UMFPACK_LDFLAGS=-lumfpack
 
-CFLAGS=-g -Wall -O3 $(GIMP_CFLAGS) $(UMFPACK_CFLAGS)
+CFLAGS=-g -Wall pardus_cflags -O3 $(GIMP_CFLAGS) $(UMFPACK_CFLAGS)
 LDFLAGS=$(GIMP_LDFLAGS) $(UMFPACK_LDFLAGS)
 
 all: colorize
