diff -Nur nmap-5.30BETA1-old/zenmap/install_scripts/unix/zenmap.desktop nmap-5.30BETA1/zenmap/install_scripts/unix/zenmap.desktop
--- nmap-5.30BETA1-old/zenmap/install_scripts/unix/zenmap.desktop	2010-07-07 21:05:45.140507533 +0300
+++ nmap-5.30BETA1/zenmap/install_scripts/unix/zenmap.desktop	2010-07-07 21:09:03.588380333 +0300
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Name=Zenmap
 GenericName=GUI Port Scanner
+GenericName[tr]=Port Tarayıcı
 TryExec=zenmap
 Exec=zenmap %F
 Terminal=false
@@ -9,3 +10,4 @@
 Type=Application
 Categories=Application;Network;Security
 Comment=A cross-platform GUI for the Nmap Security Scanner.
+Comment[tr]=Nmap Güvenlik Tarayıcı için grafik arabirim.
diff -Nur nmap-5.30BETA1-old/zenmap/install_scripts/unix/zenmap-root.desktop nmap-5.30BETA1/zenmap/install_scripts/unix/zenmap-root.desktop
--- nmap-5.30BETA1-old/zenmap/install_scripts/unix/zenmap-root.desktop	2010-07-07 21:05:45.140507533 +0300
+++ nmap-5.30BETA1/zenmap/install_scripts/unix/zenmap-root.desktop	2010-07-07 21:09:04.924385518 +0300
@@ -1,7 +1,9 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=Zenmap (as root)
+Name[tr]=Zenmap (root olarak)
 GenericName=GUI Port Scanner
+GenericName[tr]=Port Tarayıcı
 TryExec=su-to-zenmap.sh
 Exec=su-to-zenmap.sh %F
 Terminal=false
@@ -9,3 +11,4 @@
 Type=Application
 Categories=Application;Network;Security
 Comment=A cross-platform GUI for the Nmap Security Scanner.
+Comment[tr]=Nmap Güvenlik Tarayıcı Grafik Arayüzü.
