diff -Nur nmap-5.30BETA1-old/zenmap/install_scripts/unix/su-to-zenmap.sh nmap-5.30BETA1/zenmap/install_scripts/unix/su-to-zenmap.sh
--- nmap-5.30BETA1-old/zenmap/install_scripts/unix/su-to-zenmap.sh	2010-07-07 19:50:14.888382490 +0300
+++ nmap-5.30BETA1/zenmap/install_scripts/unix/su-to-zenmap.sh	2010-07-07 19:59:18.998510487 +0300
@@ -3,6 +3,7 @@
 #Joost Witteveen <joostje@debian.org>
 #Morten Brix Pedersen
 #Bill Allombert <ballombe@debian.org>
+# force usage of xdg-su, it is safer to leave assumption to xdg, Onur Küçük <onur@pardus.org.tr>
 
 PRIV=root
 COMMAND="zenmap $@"
@@ -12,40 +13,6 @@
 if test "$euid" = "$privid"; then
     $COMMAND
 else
-    if test -z "$SU_TO_ROOT_X"; then
-      if which gksu >/dev/null 2>&1 ; then
-        SU_TO_ROOT_X=gksu
-        if test "X$KDE_FULL_SESSION" = "Xtrue" ; then
-          if which kdesu >/dev/null 2>&1 ; then
-            SU_TO_ROOT_X=kdesu
-          elif test -x /usr/lib/kde4/libexec/kdesu ; then
-            SU_TO_ROOT_X=kde4su
-          fi;
-        fi;
-      elif which kdesu >/dev/null 2>&1 ; then 
-        SU_TO_ROOT_X=kdesu
-      elif test -x /usr/lib/kde4/libexec/kdesu ; then
-        SU_TO_ROOT_X=kde4su
-      elif which ktsuss >/dev/null 2>&1 ; then
-        SU_TO_ROOT_X=ktsuss
-      elif which xterm>/dev/null 2>&1 ;then
-        if which sudo>/dev/null 2>&1 ;then
-          SU_TO_ROOT_X=sdterm
-        else
-          SU_TO_ROOT_X=sterm
-        fi;
-      else
-        SU_TO_ROOT_X=su-to-root
-      fi
-    fi
-    case $SU_TO_ROOT_X in
-      gksu) gksu -u "$PRIV" "$COMMAND";;
-      kdesu) kdesu -u "$PRIV" "$COMMAND";;
-      kde4su) /usr/lib/kde4/libexec/kdesu -u "$PRIV" "$COMMAND";;
-      ktsuss) ktsuss -u "$PRIV" "$COMMAND";;
-  # As a last resort, open a new xterm use sudo/su
-      sdterm) xterm -e "sudo -u $PRIV $COMMAND";;
-      sterm) xterm -e "su -l $PRIV -c $COMMAND";;
-    esac;
+    xdg-su -u "$PRIV" -c "$COMMAND"
 fi
 
