Index: pychecker/Config.py
===================================================================
--- pychecker/Config.py
+++ pychecker/Config.py
@@ -33,7 +33,7 @@ _WARNING_LEVELS = get_warning_levels()
 
 _RC_FILE = ".pycheckrc"
 CHECKER_VAR = '__pychecker__'
-_VERSION = '0.8.18'
+_VERSION = '0.8.19'
 
 _DEFAULT_BLACK_LIST = [ "Tkinter", "wxPython", "gtk", "GTK", "GDK", ]
 _DEFAULT_VARIABLE_IGNORE_LIST = [ '__version__', '__warningregistry__', 
