--- bzflag-2.0.10/include/common.h	2007-11-15 23:11:33.000000000 +0100
+++ bzflag-2.0.10/include/common.h	2008-01-04 16:42:06.000000000 +0100
@@ -33,9 +33,12 @@
 
 #include <stdio.h>
 #include <stdlib.h> /* needed for bzfrand */
-#include <math.h>
 #ifdef __cplusplus
 #  include <cmath>
+#  include <cstring>
+#else
+#  include <math.h>
+#  include <string.h>
 #endif
 
 
--- bzflag-2.0.10/include/bzfsAPI.h	2007-11-16 00:11:33.000000000 +0200
+++ bzflag-2.0.10/include/bzfsAPI.h	2008-01-24 17:25:53.000000000 +0200
@@ -16,6 +16,9 @@
 #define _BZFS_API_H_
 
 /* system interface headers */
+#include <algorithm>
+#include <cstdlib>
+#include <cstring>
 #include <string>
 #include <vector>
 
