Index: shell/okular.desktop
===================================================================
--- shell/okular.desktop.orig
+++ shell/okular.desktop
@@ -124,5 +124,5 @@ Icon=okular
 Type=Application
 X-DocPath=okular/index.html
 InitialPreference=7
-Categories=Qt;KDE;Graphics;Office;Viewer;
+Categories=Qt;KDE;Graphics;2DGraphics;VectorGraphics;Viewer;
 MimeType=application/vnd.kde.okular-archive;
