diff -Nuar util-linux-2.21.2.orig/libblkid/src/superblocks/ntfs.c util-linux-2.21.2/libblkid/src/superblocks/ntfs.c
--- util-linux-2.21.2.orig/libblkid/src/superblocks/ntfs.c	2012-05-24 13:22:56.000000000 +0300
+++ util-linux-2.21.2/libblkid/src/superblocks/ntfs.c	2012-06-05 14:21:57.967967774 +0300
@@ -160,7 +160,7 @@
 
 const struct blkid_idinfo ntfs_idinfo =
 {
-	.name		= "ntfs",
+	.name		= "ntfs-3g",
 	.usage		= BLKID_USAGE_FILESYSTEM,
 	.probefunc	= probe_ntfs,
 	.magics		=
