Index: playonlinux/python/mainwindow.py
===================================================================
--- playonlinux/python/mainwindow.py
+++ playonlinux/python/mainwindow.py	2012-08-19 19:21:40.087260267 +0200
@@ -1091,29 +1091,6 @@
             wx.MessageBox(_("{0} is not supposed to be run as root. Sorry").format(os.environ["APPLICATION_TITLE"]),_("Error"))
             os._exit(0)
 
-        #### 32 bits OpenGL check
-        try:
-            returncode=subprocess.call([os.environ["PLAYONLINUX"]+"/bash/check_gl","x86"])
-        except:
-            returncode=255
-        if(os.environ["POL_OS"] == "Linux" and returncode != 0):
-            wx.MessageBox(_("{0} is unable to find 32bits OpenGL libraries.\n\nYou might encounter problem with your games").format(os.environ["APPLICATION_TITLE"]),_("Error"))
-            os.environ["OpenGL32"] = "0"
-        else:
-            os.environ["OpenGL32"] = "1"
-
-        #### 64 bits OpenGL check
-        if(os.environ["AMD64_COMPATIBLE"] == "True"):
-            try:
-                returncode=subprocess.call([os.environ["PLAYONLINUX"]+"/bash/check_gl","amd64"])
-            except:
-                returncode=255
-        if(os.environ["AMD64_COMPATIBLE"] == "True" and os.environ["POL_OS"] == "Linux" and returncode != 0):
-            wx.MessageBox(_("{0} is unable to find 64bits OpenGL libraries.\n\nYou might encounter problem with your games").format(os.environ["APPLICATION_TITLE"]),_("Error"))
-            os.environ["OpenGL64"] = "0"
-        else:
-            os.environ["OpenGL64"] = "1"
-
         #### Filesystem check
         if(os.environ["POL_OS"] == "Linux"):
             try:
 
