Index: libxcb-1.7/configure.ac
===================================================================
--- libxcb-1.7.orig/configure.ac
+++ libxcb-1.7/configure.ac
@@ -18,17 +18,7 @@ AC_CONFIG_HEADERS([src/config.h])
 AC_PROG_LIBTOOL
 AC_PROG_CC
 
-AC_PATH_PROG(XSLTPROC, xsltproc, no)
-if test "$XSLTPROC" = "no"; then
-   AC_MSG_ERROR([XCB requires xsltproc.])
-fi
-
 HTML_CHECK_RESULT=false
-if test x"$HAVE_CHECK" = xyes; then
-	if test x"$XSLTPROC" != xno; then
-		HTML_CHECK_RESULT=true
-	fi
-fi
 AC_SUBST(HTML_CHECK_RESULT)
 
 # Checks for pkg-config packages
