--- src/knights.desktop.OSmanOS	2012-08-15 16:59:40.000000000 +0300
+++ src/knights.desktop		2013-01-12 10:28:41.249880380 +0200
@@ -36,6 +36,7 @@
 Name[sr@ijekavianlatin]=Konji
 Name[sr@latin]=Konji
 Name[sv]=Knights
+Name[tr]=Knights
 Name[ug]=Knights
 Name[uk]=Knights
 Name[x-test]=xxKnightsxx
@@ -83,6 +84,7 @@
 GenericName[sr@ijekavianlatin]=Šahovska igra
 GenericName[sr@latin]=Šahovska igra
 GenericName[sv]=Schackspel
+GenericName[tr]=Satranç
 GenericName[ug]=شاھمات ئويۇنى
 GenericName[uk]=Гра у шахи
 GenericName[x-test]=xxChess gamexx
