Index: zaz-1.0.0/extra/zaz.desktop
===================================================================
--- zaz-1.0.0.orig/extra/zaz.desktop
+++ zaz-1.0.0/extra/zaz.desktop
@@ -2,6 +2,9 @@
 Version=1.0
 Type=Application
 Name=Zaz
+Name[tr]=Zaz
+GenericName=Zaz
+GenericName[tr]=Zaz
 Comment=Zaz ain't Z***
 Comment[ru]=Zaz не клон Z***
 Comment[tr]=Top Yok Etme Oyunu
@@ -9,4 +12,3 @@ TryExec=zaz
 Exec=zaz
 Categories=Game;LogicGame;
 Icon=zaz
-MimeType=
