diff -Nur cddlib-094f-old/configure.in cddlib-094f/configure.in
--- cddlib-094f-old/configure.in	2009-09-03 19:16:04.516767767 +0300
+++ cddlib-094f/configure.in	2009-09-03 19:18:00.045768053 +0300
@@ -7,7 +7,7 @@
 dnl Checks for programs.
 AC_PROG_CC
 AC_PROG_INSTALL
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 
 dnl Checks for libraries.
 dnl Replace `main' with a function in -lg:
diff -Nur cddlib-094f-old/lib-src/Makefile.am cddlib-094f/lib-src/Makefile.am
--- cddlib-094f-old/lib-src/Makefile.am	2009-09-03 19:16:04.518768162 +0300
+++ cddlib-094f/lib-src/Makefile.am	2009-09-03 19:17:15.821769747 +0300
@@ -1,6 +1,6 @@
-lib_LIBRARIES = libcdd.a
+lib_LTLIBRARIES = libcdd.la
 
-libcdd_a_SOURCES = \
+libcdd_la_SOURCES = \
 cddcore.c \
 cddlp.c	\
 cddmp.c \
diff -Nur cddlib-094f-old/lib-src-gmp/Makefile.am cddlib-094f/lib-src-gmp/Makefile.am
--- cddlib-094f-old/lib-src-gmp/Makefile.am	2009-09-03 19:16:04.503039551 +0300
+++ cddlib-094f/lib-src-gmp/Makefile.am	2009-09-03 19:17:28.177018576 +0300
@@ -1,6 +1,6 @@
-lib_LIBRARIES = libcddgmp.a
+lib_LTLIBRARIES = libcddgmp.la
 
-libcddgmp_a_SOURCES = \
+libcddgmp_la_SOURCES = \
 cddcore.c \
 cddlp.c	\
 cddmp.c \
