Index: pinentry-0.8.0/configure.ac
===================================================================
--- pinentry-0.8.0.orig/configure.ac
+++ pinentry-0.8.0/configure.ac
@@ -434,11 +434,11 @@ AM_CONDITIONAL(BUILD_PINENTRY_W32, test 
 # Please change the order only after verifying that the preferred
 # pinentry really is better (more feature-complete and more secure).
 
-if test "$pinentry_gtk_2" = "yes"; then
-   PINENTRY_DEFAULT=pinentry-gtk-2
+if test "$pinentry_qt4" = "yes"; then
+   PINENTRY_DEFAULT=pinentry-qt4
 else
-  if test "$pinentry_gtk" = "yes"; then
-    PINENTRY_DEFAULT=pinentry-gtk
+  if test "$pinentry_gtk2" = "yes"; then
+    PINENTRY_DEFAULT=pinentry-gtk-2
   else
     if test "$pinentry_qt" = "yes"; then
       PINENTRY_DEFAULT=pinentry-qt
