--- plugins/zynaddsubfx/fltk/CMakeLists.txt.orig
+++ plugins/zynaddsubfx/fltk/CMakeLists.txt
@@ -46,7 +46,7 @@
 # Search for modules in the FLTK source dir first
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${FLTK_SOURCE_DIR}/CMake)
 
-include_directories(${FLTK_SOURCE_DIR} ${FLTK_BINARY_DIR})
+include_directories(${FLTK_SOURCE_DIR} ${FLTK_BINARY_DIR} /usr/include/freetype2)
 
 # If this is out-of-source build, then copy FL directory
 if(NOT ${FLTK_SOURCE_DIR} STREQUAL ${FLTK_BINARY_DIR})
