Index: gftp-2.0.19/docs/sample.gftp/bookmarks
===================================================================
--- gftp-2.0.19.orig/docs/sample.gftp/bookmarks
+++ gftp-2.0.19/docs/sample.gftp/bookmarks
@@ -2,6 +2,26 @@
 # <masneyb@gftp.org>. Warning: Any comments that you add to this file WILL be
 # overwritten
 
+[Pardus Sites/Ulakbim]
+hostname=ftp.ulakbim.gov.tr
+port=21
+protocol=FTP
+remote directory=/linux/pardus/Installation
+local directory=
+username=anonymous
+password=@EMAIL@
+account=
+
+[Pardus Sites/Pardus]
+hostname=ftp.pardus.org.tr
+port=21
+protocol=FTP
+remote directory=/pub/ISO/Installation
+local directory=
+username=anonymous
+password=@EMAIL@
+account=
+
 [BSD Sites/FreeBSD]
 hostname=ftp.freebsd.org
 port=21
@@ -42,21 +62,11 @@ username=anonymous
 password=@EMAIL@
 account=
 
-[Debian Sites/Debian non-US]
-hostname=non-us.debian.org
-port=21
-protocol=FTP
-remote directory=/
-local directory=
-username=anonymous
-password=@EMAIL@
-account=
-
-[Debian Sites/Freshmeat DEBs]
-hostname=ftp.freshmeat.net
+[Fedora Sites/Fedora]
+hostname=download.fedora.redhat.com
 port=21
 protocol=FTP
-remote directory=/pub/debs
+remote directory=/pub/fedora/linux
 local directory=
 username=anonymous
 password=@EMAIL@
@@ -132,16 +142,6 @@ username=anonymous
 password=@EMAIL@
 account=
 
-[General Sites/Source Forge]
-hostname=ftp.sourceforge.net
-port=21
-protocol=FTP
-remote directory=/pub
-local directory=
-username=anonymous
-password=@EMAIL@
-account=
-
 [General Sites/Tux.Org]
 hostname=ftp.tux.org
 port=21
@@ -152,16 +152,6 @@ username=anonymous
 password=@EMAIL@
 account=
 
-[General Sites/TSX-11]
-hostname=tsx-11.mit.edu
-port=21
-protocol=FTP
-remote directory=/pub/linux
-local directory=
-username=anonymous
-password=@EMAIL@
-account=
-
 [General Sites/Wcarchive]
 hostname=ftp.cdrom.com
 port=21
@@ -282,78 +272,118 @@ username=anonymous
 password=@EMAIL@
 account=
 
-[RedHat Sites/FalseHope RPMs]
-hostname=ftp.falsehope.com
+[Mandriva Sites/Community SRPMS (California)]
+hostname=mirrors.usc.edu
 port=21
 protocol=FTP
-remote directory=/pub
+remote directory=/pub/linux/distributions/mandrakelinux/devel/community/SRPMS/
 local directory=
 username=anonymous
 password=@EMAIL@
 account=
 
-[RedHat Sites/Freshmeat RPMs]
-hostname=ftp.freshmeat.net
+[Mandriva Sites/Devel iso (California)]
+hostname=mirrors.usc.edu
 port=21
 protocol=FTP
-remote directory=/pub/rpms
+remote directory=/pub/linux/distributions/mandrakelinux/devel/iso/
 local directory=
 username=anonymous
 password=@EMAIL@
 account=
 
-[RedHat Sites/RH Contrib]
-hostname=contrib.redhat.com
+[Mandriva Sites/Official iso (California)]
+hostname=mirrors.usc.edu
 port=21
 protocol=FTP
-remote directory=/pub
+remote directory=/pub/linux/distributions/mandrakelinux/official/iso/
 local directory=
 username=anonymous
 password=@EMAIL@
 account=
 
-[RedHat Sites/RH Main]
-hostname=ftp.redhat.com
+[Mandriva Sites/Mandriva Old (California)]
+hostname=mirrors.usc.edu
 port=21
 protocol=FTP
-remote directory=/pub
+remote directory=/pub/linux/distributions/mandrakelinux/old/
 local directory=
 username=anonymous
 password=@EMAIL@
 account=
 
-[RedHat Sites/RH Rawhide]
-hostname=rawhide.redhat.com
+[Mandriva Sites/Community SRPMS (France)]
+hostname=ftp.ciril.fr
 port=21
 protocol=FTP
-remote directory=/
+remote directory=/pub/linux/mandrakelinux/devel/community/SRPMS/
 local directory=
 username=anonymous
 password=@EMAIL@
 account=
 
-[RedHat Sites/RH Updates]
-hostname=updates.redhat.com
+[Mandriva Sites/Devel iso (France)]
+hostname=ftp.ciril.fr
 port=21
 protocol=FTP
-remote directory=/
+remote directory=/pub/linux/mandrakelinux/devel/iso/
 local directory=
 username=anonymous
 password=@EMAIL@
 account=
 
-[RedHat Sites/RPM.org]
-hostname=ftp.rpm.org
+[Mandriva Sites/Official iso (France)]
+hostname=ftp.ciril.fr
 port=21
 protocol=FTP
-remote directory=/pub
+remote directory=/pub/linux/mandrakelinux/official/iso/
 local directory=
 username=anonymous
 password=@EMAIL@
 account=
 
-[RedHat Sites/Fresh RPMS]
-hostname=ftp.freshrpms.net
+[Mandriva Sites/Community SRPMS (Taiwan)]
+hostname=ftp.isu.edu.tw
+port=21
+protocol=FTP
+remote directory=/pub/Linux/Mandrakelinux/devel/community/SRPMS/
+local directory=
+username=anonymous
+password=@EMAIL@
+account=
+
+[Mandriva Sites/Devel iso (Taiwan)]
+hostname=ftp.isu.edu.tw
+port=21
+protocol=FTP
+remote directory=/pub/Linux/Mandrakelinux/devel/iso/
+local directory=
+username=anonymous
+password=@EMAIL@
+account=
+
+[Mandriva Sites/Official iso (Taiwan)]
+hostname=ftp.isu.edu.tw
+port=21
+protocol=FTP
+remote directory=/pub/Linux/Mandrakelinux/official/iso/
+local directory=
+username=anonymous
+password=@EMAIL@
+account=
+
+[Mandriva Sites/Mandriva Old (Taiwan)]
+hostname=ftp.isu.edu.tw
+port=21
+protocol=FTP
+remote directory=/pub/Linux/Mandrakelinux/old/
+local directory=
+username=anonymous
+password=@EMAIL@
+account=
+
+[Red Hat Sites/RH Main]
+hostname=ftp.redhat.com
 port=21
 protocol=FTP
 remote directory=/pub
@@ -362,21 +392,31 @@ username=anonymous
 password=@EMAIL@
 account=
 
-[Slackware Sites/slackware.com]
-hostname=ftp.slackware.com
+[Red Hat Sites/RH Updates]
+hostname=updates.redhat.com
 port=21
 protocol=FTP
-remote directory=/pub/slackware/slackware
+remote directory=/
 local directory=
 username=anonymous
 password=@EMAIL@
 account=
 
-[Slackware Sites/linuxmafia.org]
-hostname=packages.linuxmafia.org
+[Red Hat Sites/Fresh RPMS]
+hostname=ftp.freshrpms.net
 port=21
 protocol=FTP
-remote directory=/pub/slackware-7.0
+remote directory=/pub
+local directory=
+username=anonymous
+password=@EMAIL@
+account=
+
+[Slackware Sites/slackware.com]
+hostname=ftp.slackware.com
+port=21
+protocol=FTP
+remote directory=/pub/slackware/slackware
 local directory=
 username=anonymous
 password=@EMAIL@
@@ -402,11 +442,11 @@ username=anonymous
 password=@EMAIL@
 account=
 
-[SuSE Sites/SuSE Update]
-hostname=ftp.suse.com
+[SuSE Sites/OpenSuSE Update]
+hostname=ftp.hosteurope.de
 port=21
 protocol=FTP
-remote directory=/suse_update
+remote directory=/mirror/ftp.opensuse.org/update
 local directory=
 username=anonymous
 password=@EMAIL@
@@ -450,16 +490,6 @@ remote directory=/pub
 local directory=
 username=anonymous
 password=@EMAIL@
-account=
-
-[X11 Sites/LessTif]
-hostname=ftp.hungry.com
-port=21
-protocol=FTP
-remote directory=/pub/hungry/lesstif
-local directory=
-username=anonymous
-password=@EMAIL@
 account=
 
 [X11 Sites/Qt]
