Index: gftp-2.0.19/docs/gftp.desktop
===================================================================
--- gftp-2.0.19.orig/docs/gftp.desktop
+++ gftp-2.0.19/docs/gftp.desktop
@@ -3,13 +3,14 @@ Encoding=UTF-8
 Name=gFTP
 Comment=Download and upload files using multiple file transfer protocols
 Comment[fr]=Télécharge des fichiers en utilisant le protocole FTP
+Comment[tr]=Dosyaları çoklu dosya aktarım protokolü ile indirin ve yükleyin
 GenericName=gFTP
 Exec=gftp %u
 Terminal=false
 X-MultipleArgs=false
 Type=Application
 Icon=gftp.png
-Categories=Application;Network;
+Categories=GTK;GNOME;Application;Network;FileTransfer;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gftp
 X-GNOME-Bugzilla-Component=general
