diff -Nuar inxi-1.8.34.orig/inxi inxi-1.8.34/inxi
--- inxi-1.8.34.orig/inxi	2013-04-20 14:06:01.586242908 +0300
+++ inxi-1.8.34/inxi	2013-04-20 14:11:08.959251481 +0300
@@ -334,7 +334,7 @@
 FILE_MEMINFO='/proc/meminfo'
 FILE_MODULES='/proc/modules'
 FILE_MOUNTS='/proc/mounts'
-FILE_OS_RELEASE='/etc/os-release'
+FILE_OS_RELEASE='/etc/pisilinux-release'
 FILE_PARTITIONS='/proc/partitions'
 FILE_SCSI='/proc/scsi/scsi'
 FILE_XORG_LOG='/var/log/Xorg.0.log' # if not found, search and replace with actual location
@@ -512,7 +512,7 @@
 DISTROS_DERIVED="antix-version aptosid-version kanotix-version knoppix-version mandrake-release pardus-release sabayon-release siduction-version sidux-version turbolinux-release zenwalk-version"
 # debian_version excluded from DISTROS_PRIMARY so Debian can fall through to /etc/issue detection. Same goes for Ubuntu.
 DISTROS_EXCLUDE_LIST="debian_version ubuntu_version"
-DISTROS_PRIMARY="arch-release gentoo-release redhat-release slackware-version SuSE-release"
+DISTROS_PRIMARY="arch-release gentoo-release redhat-release slackware-version SuSE-release pisilinux-release"
 DISTROS_LSB_GOOD="mandrake-release mandriva-release mandrakelinux-release"
 # this is being used both by core distros and derived distros now, eg, solusos uses it for solusos id, while
 # debian, solusos base, uses it as well, so we have to know which it is.
