diff -Naupr easytag-2.1.6.orig/configure easytag-2.1.6/configure
--- easytag-2.1.6.orig/configure	2008-07-12 16:57:44.000000000 +0200
+++ easytag-2.1.6/configure	2009-09-12 16:47:34.117176630 +0200
@@ -26034,7 +26034,7 @@ ac_config_headers="$ac_config_headers co
 
 
 
-ac_config_files="$ac_config_files Makefile easytag.spec doc/easytag.1 src/Makefile src/id3lib/Makefile src/libapetag/Makefile src/libmpg123/Makefile po/Makefile.in"
+ac_config_files="$ac_config_files Makefile easytag.spec doc/easytag.1 src/Makefile src/id3lib/Makefile src/libapetag/Makefile po/Makefile.in"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -26637,7 +26637,6 @@ do
     "src/Makefile") CONFIG_FILES="$CONFIG_FILES src/Makefile" ;;
     "src/id3lib/Makefile") CONFIG_FILES="$CONFIG_FILES src/id3lib/Makefile" ;;
     "src/libapetag/Makefile") CONFIG_FILES="$CONFIG_FILES src/libapetag/Makefile" ;;
-    "src/libmpg123/Makefile") CONFIG_FILES="$CONFIG_FILES src/libmpg123/Makefile" ;;
     "po/Makefile.in") CONFIG_FILES="$CONFIG_FILES po/Makefile.in" ;;
 
   *) { { echo "$as_me:$LINENO: error: invalid argument: $ac_config_target" >&5
diff -Naupr easytag-2.1.6.orig/configure.in easytag-2.1.6/configure.in
--- easytag-2.1.6.orig/configure.in	2008-05-07 22:45:32.000000000 +0200
+++ easytag-2.1.6/configure.in	2009-09-12 16:47:42.421053700 +0200
@@ -473,7 +473,6 @@ AC_OUTPUT([ Makefile
             src/Makefile
             src/id3lib/Makefile
             src/libapetag/Makefile
-            src/libmpg123/Makefile
             po/Makefile.in
          ])
 
diff -Naupr easytag-2.1.6.orig/src/Makefile.in easytag-2.1.6/src/Makefile.in
--- easytag-2.1.6.orig/src/Makefile.in	2008-07-12 16:57:43.000000000 +0200
+++ easytag-2.1.6/src/Makefile.in	2009-09-12 16:48:30.102176910 +0200
@@ -59,7 +59,7 @@ am_easytag_OBJECTS = about.$(OBJEXT) ape
 	wavpack_tag.$(OBJEXT)
 easytag_OBJECTS = $(am_easytag_OBJECTS)
 am__DEPENDENCIES_1 =
-easytag_DEPENDENCIES = $(am__DEPENDENCIES_1) libmpg123/libmpg123.a \
+easytag_DEPENDENCIES = $(am__DEPENDENCIES_1) \
 	libapetag/libapetag.a id3lib/libid3bugfix.a
 DEFAULT_INCLUDES = -I. -I$(top_builddir)@am__isrc@
 depcomp = $(SHELL) $(top_srcdir)/depcomp
@@ -226,7 +226,7 @@ sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = libmpg123 libapetag id3lib
+SUBDIRS = libapetag id3lib
 easytag_SOURCES = \
  about.c \
  about.h \
@@ -308,7 +308,7 @@ EXTRA_DIST = \
  win32/nsis/easytag-install.ico \
  win32/nsis/easytag-uninstall.bmp 
 
-easytag_LDADD = $(GTK_LIBS) libmpg123/libmpg123.a libapetag/libapetag.a  id3lib/libid3bugfix.a
+easytag_LDADD = $(GTK_LIBS) libapetag/libapetag.a  id3lib/libid3bugfix.a
 PACKAGE_DATA_DIR = $(datadir)/$(PACKAGE)
 INCLUDES = -DLOCALEDIR=\"$(localedir)\" \
            -DPACKAGE_DATA_DIR=\"$(PACKAGE_DATA_DIR)\"
diff -Naupr easytag-2.1.6.orig/src/mpeg_header.c easytag-2.1.6/src/mpeg_header.c
--- easytag-2.1.6.orig/src/mpeg_header.c	2007-11-28 09:54:27.000000000 +0100
+++ easytag-2.1.6/src/mpeg_header.c	2009-09-12 16:48:50.266129581 +0200
@@ -36,7 +36,7 @@
 // Set to :
 // - 1 to use ID3lib for reading headers
 // - 0 to use mpeg123 for reading headers
-#define USE_ID3LIB_4_HEADER 0
+#define USE_ID3LIB_4_HEADER 1
 
 #if USE_ID3LIB_4_HEADER
 #   include <id3.h>
