diff -Naupr easytag-2.1.6.orig/src/misc.c easytag-2.1.6/src/misc.c
--- easytag-2.1.6.orig/src/misc.c	2008-05-24 22:59:34.000000000 +0200
+++ easytag-2.1.6/src/misc.c	2010-04-30 16:58:34.465162065 +0200
@@ -3327,7 +3327,7 @@ void Load_Filename_Set_Filenames (void)
         g_free(list_text);
     }
 
-    g_free(currentPath);
+    gtk_tree_path_free(currentPath);
 
     Browser_List_Refresh_Whole_List();
     ET_Display_File_Data_To_UI(ETCore->ETFileDisplayed);
