diff -Naupr easytag-2.1.6.orig/easytag.desktop easytag-2.1.6/easytag.desktop
--- easytag-2.1.6.orig/easytag.desktop	2007-11-28 09:54:05.000000000 +0100
+++ easytag-2.1.6/easytag.desktop	2010-04-08 11:52:42.475478051 +0200
@@ -22,5 +22,5 @@ Terminal=false
 Type=Application
 Categories=AudioVideo;AudioVideoEditing;GTK;
 X-Desktop-File-Install-Version=0.14
-MimeType=audio/x-mp3;application/ogg;x-directory/normal;
+MimeType=audio/x-mp3;application/ogg;
 
