diff -Naupr easytag-2.1.6.orig/src/picture.c easytag-2.1.6/src/picture.c
--- easytag-2.1.6.orig/src/picture.c	2008-05-18 00:43:28.000000000 +0200
+++ easytag-2.1.6/src/picture.c	2010-04-30 16:56:16.144157741 +0200
@@ -570,7 +570,7 @@ void Picture_Properties_Button_Clicked (
                     pic->description = buffer;
                 }else
                 {
-                    pic->description = 0;
+                    pic->description = strdup( "" );
                     g_free(buffer);
                 }
 
