diff -Naur easytag-2.1.6.orig/src/mp4_header.c easytag-2.1.6/src/mp4_header.c
--- easytag-2.1.6.orig/src/mp4_header.c	2007-11-28 10:54:35.000000000 +0200
+++ easytag-2.1.6/src/mp4_header.c	2010-03-09 12:15:37.205378766 +0200
@@ -44,7 +44,7 @@
 #undef PACKAGE_STRING
 #undef PACKAGE_TARNAME
 #undef PACKAGE_VERSION
-#include <mp4.h>
+#include <mp4v2/mp4v2.h>
 
 
 /****************
diff -Naur easytag-2.1.6.orig/src/mp4_tag.c easytag-2.1.6/src/mp4_tag.c
--- easytag-2.1.6.orig/src/mp4_tag.c	2007-12-11 00:08:11.000000000 +0200
+++ easytag-2.1.6/src/mp4_tag.c	2010-03-09 13:45:42.497525183 +0200
@@ -50,7 +50,7 @@
 #undef PACKAGE_STRING
 #undef PACKAGE_TARNAME
 #undef PACKAGE_VERSION
-#include <mp4.h>
+#include <mp4v2/mp4v2.h>
 
 
 /****************
@@ -193,7 +193,7 @@
     for (pic_num = 0; (MP4GetMetadataCoverArt( mp4file, &coverArt, &coverSize,pic_num )); pic_num++)
 #else
     // There version handle only one picture!
-    if ( MP4GetMetadataCoverArt( mp4file, &coverArt, &coverSize ) )
+    if ( MP4GetMetadataCoverArt( mp4file, &coverArt, &coverSize, 0) )
 #endif
     {
         Picture *pic;
