diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur d-feet-0.1.8-orig/setup.py d-feet-0.1.8/setup.py
--- d-feet-0.1.8-orig/setup.py	2009-02-13 12:20:48.000000000 +0200
+++ d-feet-0.1.8/setup.py	2009-02-13 12:21:12.000000000 +0200
@@ -94,11 +94,6 @@
         call(["scrollkeeper-preinstall", HELPURI[fileNum],
             installedOmfFiles[fileNum], installedOmfFiles[fileNum]])
         
-    #Update the scrollkeeper catalog
-    if os.geteuid() == 0:
-        print "Updating the scrollkeeper index..."
-        call(["scrollkeeper-update", "-o", installroot + omfdir])
-
 print "\nInstallation finished! You can now run d-feet by typing 'd-feet' or through your applications menu icon."
     
 ## To uninstall manually delete these files/folders:
