diff -p -up ptlib-2.5.2/src/ptlib/unix/svcproc.cxx~ ptlib-2.5.2/src/ptlib/unix/svcproc.cxx
--- ptlib-2.5.2/src/ptlib/unix/svcproc.cxx~	2009-01-03 07:59:30.000000000 +0100
+++ ptlib-2.5.2/src/ptlib/unix/svcproc.cxx	2009-01-06 11:28:39.000000000 +0100
@@ -742,7 +742,7 @@ void PServiceProcess::PXOnAsyncSignal(in
 #ifdef P_VXWORKS
   logMsg((char *)msg,0,0,0,0,0,0);
 #else
-    syslog(LOG_CRIT, msg); 
+  syslog(LOG_CRIT, "%s", msg); 
     closelog();
 #endif // !P_VXWORKS
   }
