diff -Nur klibc-1.5.20-old//scripts/Kbuild.include klibc-1.5.20/scripts/Kbuild.include
--- klibc-1.5.20-old//scripts/Kbuild.include	2010-10-12 15:52:28.054000111 +0300
+++ klibc-1.5.20/scripts/Kbuild.include	2010-10-12 15:53:37.588000098 +0300
@@ -127,7 +127,7 @@
 # >'< substitution is for echo to work, >$< substitution to preserve $ when reloading .cmd file
 # note: when using inline perl scripts [perl -e '...$$t=1;...'] in $(cmd_xxx) double $$ your perl vars
 #
-if_changed = $(if $(strip $(filter-out $(PHONY),$?)          \
+if_changed = $(if $(strip $(filter-out $(PHONY) FORCE,$?)          \
 		$(call arg-check, $(cmd_$(1)), $(cmd_$@)) ), \
 	@set -e; \
 	$(echo-cmd) $(cmd_$(1)); \
@@ -135,7 +135,7 @@
 
 # execute the command and also postprocess generated .d dependencies
 # file
-if_changed_dep = $(if $(strip $(filter-out $(PHONY),$?)  \
+if_changed_dep = $(if $(strip $(filter-out $(PHONY) FORCE,$?)  \
 		$(filter-out FORCE $(wildcard $^),$^)    \
 	$(call arg-check, $(cmd_$(1)), $(cmd_$@)) ),     \
 	@set -e; \
@@ -147,7 +147,7 @@
 # Usage: $(call if_changed_rule,foo)
 # will check if $(cmd_foo) changed, or any of the prequisites changed,
 # and if so will execute $(rule_foo)
-if_changed_rule = $(if $(strip $(filter-out $(PHONY),$?)            \
+if_changed_rule = $(if $(strip $(filter-out $(PHONY) FORCE,$?)            \
 			$(call arg-check, $(cmd_$(1)), $(cmd_$@)) ),\
 			@set -e; \
 			$(rule_$(1)))
