diff -Nur klibc-1.4.27-old/klcc/Kbuild klibc-1.4.27/klcc/Kbuild
--- klibc-1.4.27-old/klcc/Kbuild	2006-11-16 18:59:02.679031947 +0200
+++ klibc-1.4.27/klcc/Kbuild	2006-11-16 19:00:47.608080305 +0200
@@ -14,7 +14,7 @@
 	$(Q)echo 'KCROSS=$(KCROSS)' >> $@
 	$(Q)echo 'CC=$(KLIBCCC)' >> $@
 	$(Q)echo 'LD=$(KLIBCLD)' >> $@
-	$(Q)echo 'REQFLAGS=$(filter-out -I%,$(KLIBCDEFS) $(KLIBCREQFLAGS) $(KLIBCARCHREQFLAGS) $(KLIBCCPPFLAGS))' >> $@
+	$(Q)echo 'REQFLAGS=$(NOSTDINC_FLAGS) $(filter-out -I%,$(KLIBCDEFS) $(KLIBCREQFLAGS) $(KLIBCARCHREQFLAGS) $(KLIBCCPPFLAGS))' >> $@
 	$(Q)echo 'OPTFLAGS=$(KLIBCOPTFLAGS)' >> $@
 	$(Q)echo 'LDFLAGS=$(KLIBCLDFLAGS)' >> $@
 	$(Q)echo 'STRIP=$(KLIBCSTRIP)' >> $@
diff -Nur klibc-1.4.27-old/Makefile klibc-1.4.27/Makefile
--- klibc-1.4.27-old/Makefile	2006-11-16 18:59:02.679031947 +0200
+++ klibc-1.4.27/Makefile	2006-11-16 18:59:57.845645347 +0200
@@ -22,7 +22,7 @@
 export OBJCOPY  := $(KLIBCROSS)objcopy
 export OBJDUMP  := $(KLIBCROSS)objdump
 
-NOSTDINC_FLAGS := -nostdlib -nostdinc -isystem $(shell $(CC) -print-file-name=include)
+export NOSTDINC_FLAGS := -nostdlib -nostdinc -isystem $(shell $(CC) -print-file-name=include)
 
 ARCH	          := $(shell uname -m | sed -e s/i.86/i386/ -e s/parisc64/parisc/ -e s/sun4u/sparc64/ -e s/arm.*/arm/ -e s/sa110/arm/)
 export KLIBCARCH  ?= $(ARCH)
