diff -up ./src/tools.cc.fixbufferoverrun ./src/tools.cc
--- ./src/tools.cc.fixbufferoverrun	2010-11-22 00:57:31.387188000 -0500
+++ ./src/tools.cc	2010-11-22 00:57:44.865188000 -0500
@@ -665,6 +665,9 @@ String secondsToHMS(int seconds)
     h = seconds / 60;
 
     // XXX:XX:XX
+		// This fails if h goes over 999
+		if (h > 999)
+			h = 999;
     char *str = (char *)malloc(10);
     sprintf(str, "%02d:%02d:%02d", h, m, s);
     return String::take(str);
