--- src/mail-files.in.orig	2010-04-30 17:27:58.273393629 +0300
+++ src/mail-files.in	2010-04-30 17:28:11.506392089 +0300
@@ -81,7 +81,7 @@
       fi
       echo "Mailing $string to $destin"
       [ $total -ne 0 ] && sleep $SLEEP
-      @MAILER@ -s "$string: $subject" $destin < $file
+      sendmail -s "$string: $subject" $destin < $file
       total=`expr $total + 1`
       [ $count -lt $maxcount ] && sleep $SLEEP
     done
