--- /var/pisi/redeclipse-1.3.1-4/work/redeclipse/redeclipse.sh  2012-05-31 08:52:45.000000000 +0300
+++ redeclipse.sh       2013-01-14 22:14:39.604315562 +0200
@@ -3,7 +3,10 @@
 #RE_PATH=~/redeclipse
 #RE_PATH=/usr/local/redeclipse
 #RE_PATH=.
-RE_PATH="$(cd "$(dirname "$0")" && pwd)"
+RE_PATH=/usr/bin/src
+
+# RE_BIN should refer to the directory in which Red Eclipse executable files are placed.
+RE_BIN=/usr/bin
 
 # RE_OPTIONS contains any command line options you would like to start Red Eclipse with.
 RE_OPTIONS=""
@@ -36,6 +39,7 @@
     BINARY_PATH="bin64"
     ;;
 *)
+    
     SYSTEM_SUFFIX=""
     BINARY_PATH="bin"
     ;;
@@ -47,10 +51,10 @@
     BINARY_PATH="bin"
 fi
 
-if [ -x ${RE_PATH}/${BINARY_PATH}/reclient${SYSTEM_SUFFIX} ]
+if [ -x ${RE_BIN}/reclient ]
 then
     cd ${RE_PATH} || exit 1
-    exec ${RE_PATH}/${BINARY_PATH}/reclient${SYSTEM_SUFFIX} ${RE_OPTIONS} "$@"
+    exec ${RE_BIN}/reclient ${RE_OPTIONS} "$@"
 else
     echo "Your platform does not have a pre-compiled Red Eclipse client."
     echo -n "Would you like to build one now? [Yn] "
