Index: kdeartwork-4.7.1/kscreensaver/kpartsaver/kpartsaver.desktop
===================================================================
--- kdeartwork-4.7.1.orig/kscreensaver/kpartsaver/kpartsaver.desktop
+++ kdeartwork-4.7.1/kscreensaver/kpartsaver/kpartsaver.desktop
@@ -58,7 +58,7 @@ Name[sv]=Mediaskärmsläckare
 Name[ta]=ஊடக திரை காப்பாளர்
 Name[tg]=Аз назар гузарондани мултимедия
 Name[th]=รักษาจอภาพโดยใช้แฟ้มสื่อ
-Name[tr]=Ekran Koruyucu
+Name[tr]=Çokluortam Ekran Koruyucu
 Name[ug]=ۋاسىتە ئېكران قوغدىغۇچ
 Name[uk]=Мультимедійний зберігач екрана
 Name[vi]=Nêm=Hình ảnh Bảo vệ Màn hình
