--- ./gdb/testsuite/gdb.base/annota1.exp	10 Jan 2007 03:23:04 -0000	1.23
+++ ./gdb/testsuite/gdb.base/annota1.exp	10 May 2007 12:54:11 -0000
@@ -57,6 +57,8 @@ if [target_info exists gdb_stub] {
     gdb_step_for_stub;
 }
 
+gdb_test "set breakpoint pending off" "" "Avoid lockup on nonexisting functions"
+
 #
 # the line at which break main will put the breakpoint
 #
--- ./gdb/testsuite/gdb.base/annota3.exp	9 Jan 2007 17:59:09 -0000	1.12
+++ ./gdb/testsuite/gdb.base/annota3.exp	10 May 2007 12:54:11 -0000
@@ -56,6 +56,8 @@ if [target_info exists gdb_stub] {
     gdb_step_for_stub;
 }
 
+gdb_test "set breakpoint pending off" "" "Avoid lockup on nonexisting functions"
+
 #
 # the line at which break main will put the breakpoint
 #
--- gdb-6.6/gdb/testsuite/gdb.threads/step-thread-exit.exp-orig	2007-05-10 15:03:15.000000000 +0200
+++ gdb-6.6/gdb/testsuite/gdb.threads/step-thread-exit.exp	2007-05-10 15:04:24.000000000 +0200
@@ -58,6 +58,9 @@ gdb_test "continue" "Break.*thread_funct
 # thread to be stopped and a message printed to tell us we have stepped
 # over the thread exit.
 set test "step over thread exit 1"
+# ppc64 is currently failing:
+set timeout_old $timeout
+set timeout 60
 gdb_test_multiple "next" "$test" {
   -re "\}.*$gdb_prompt $" {
      send_gdb "next\n"
@@ -71,6 +74,7 @@ gdb_test_multiple "next" "$test" {
      exp_continue
   }
 }
+set timeout $timeout_old
 
 # Without this fixup we could end up in:
 # #0  0x00110416 in __kernel_vsyscall ()
