Index: gdb-7.1/gdb/elfread.c
===================================================================
*** gdb-7.1.orig/gdb/elfread.c	2010-04-15 14:46:14.000000000 +0200
--- gdb-7.1/gdb/elfread.c	2010-04-15 14:50:17.000000000 +0200
*************** debug_print_missing (const char *binary,
*** 1625,1633 ****
  	fprintf_unfiltered (gdb_stdlog,
  			    _("Missing separate debuginfo for %s\n"), binary);
          if (debug != NULL)
! 	  fprintf_unfiltered (gdb_stdlog, _("Try: %s %s\n"),
! 			      "yum --disablerepo='*' --enablerepo='*-debuginfo'"
! 			      " install", debug);
        }
  }
  
--- 1625,1636 ----
  	fprintf_unfiltered (gdb_stdlog,
  			    _("Missing separate debuginfo for %s\n"), binary);
          if (debug != NULL)
! 	  {
! 	    const char *p = strrchr (debug, '/');
! 	    fprintf_unfiltered (gdb_stdlog, _("Try: %s%.2s%.38s\"\n"),
! 				"zypper install -C \"debuginfo(build-id)=",
! 				p - 2, p + 1);
! 	  }
        }
  }
  
