--- ./gdb/breakpoint.c	2010-04-26 02:35:53.000000000 +0200
+++ ./gdb/breakpoint.c	2010-04-26 02:33:26.000000000 +0200
@@ -8499,6 +8499,11 @@ update_breakpoint_locations (struct brea
 	{
 	  struct gdb_exception e;
 
+	  /* Temporary workaround before the non-intrusive ifunc patch gets in.
+	     exp parsing can now call inferior function invalidating
+	     EXISTING_LOCATIONS.  */
+	  existing_locations = NULL;
+
 	  s = b->cond_string;
 	  TRY_CATCH (e, RETURN_MASK_ERROR)
 	    {
