--- gdb/tui/tui-regs.c	2010-01-01 08:32:07.000000000 +0100
+++ gdb/tui/tui-regs.c	2010-04-20 09:23:59.000000000 +0200
@@ -541,7 +541,12 @@
       int i;
 
       if (data->highlight)
-	wstandout (win_info->handle);
+	/* We ignore the return value, casting it to void in order to avoid
+	   a compiler warning.  The warning itself was introduced by a patch
+	   to ncurses 5.7 dated 2009-08-29, changing this macro to expand
+	   to code that causes the compiler to generate an unused-value
+	   warning.  */
+	(void) wstandout (win_info->handle);
       
       wmove (win_info->handle, 0, 0);
       for (i = 1; i < win_info->width; i++)
@@ -551,7 +556,12 @@
         waddstr (win_info->handle, data->content);
 
       if (data->highlight)
-	wstandend (win_info->handle);
+	/* We ignore the return value, casting it to void in order to avoid
+	   a compiler warning.  The warning itself was introduced by a patch
+	   to ncurses 5.7 dated 2009-08-29, changing this macro to expand
+	   to code that causes the compiler to generate an unused-value
+	   warning.  */
+	(void) wstandend (win_info->handle);
       tui_refresh_win (win_info);
     }
 }
--- gdb/tui/tui-stack.c	2010-01-01 08:32:07.000000000 +0100
+++ gdb/tui/tui-stack.c	2010-04-20 09:23:59.000000000 +0200
@@ -256,10 +256,15 @@
 
       string = tui_make_status_line (&element->which_element.locator);
       wmove (locator->handle, 0, 0);
-      wstandout (locator->handle);
+      /* We ignore the return value from wstandout and wstandend, casting
+	 them to void in order to avoid a compiler warning.  The warning
+	 itself was introduced by a patch to ncurses 5.7 dated 2009-08-29,
+	 changing these macro to expand to code that causes the compiler
+	 to generate an unused-value warning.  */
+      (void) wstandout (locator->handle);
       waddstr (locator->handle, string);
       wclrtoeol (locator->handle);
-      wstandend (locator->handle);
+      (void) wstandend (locator->handle);
       tui_refresh_win (locator);
       wmove (locator->handle, 0, 0);
       xfree (string);
