Index: gdb-6.8.50.20090811/gdb/infrun.c
===================================================================
--- gdb-6.8.50.20090811.orig/gdb/infrun.c	2009-08-12 06:56:29.000000000 +0200
+++ gdb-6.8.50.20090811/gdb/infrun.c	2009-08-12 07:13:08.000000000 +0200
@@ -1075,7 +1075,7 @@ static const char *scheduler_enums[] = {
   schedlock_step,
   NULL
 };
-static const char *scheduler_mode = schedlock_off;
+static const char *scheduler_mode = schedlock_step;
 static void
 show_scheduler_mode (struct ui_file *file, int from_tty,
 		     struct cmd_list_element *c, const char *value)
Index: gdb-6.8.50.20090811/gdb/testsuite/gdb.mi/mi-console.exp
===================================================================
--- gdb-6.8.50.20090811.orig/gdb/testsuite/gdb.mi/mi-console.exp	2009-08-12 06:56:29.000000000 +0200
+++ gdb-6.8.50.20090811/gdb/testsuite/gdb.mi/mi-console.exp	2009-08-12 06:58:45.000000000 +0200
@@ -47,6 +47,9 @@ if  { [gdb_compile "${srcdir}/${subdir}/
 
 mi_run_to_main
 
+# thread-id=\"all\" vs. thread-id=\"1\" below:
+mi_gdb_test "210-gdb-set scheduler-locking off" "210\\^done" "set scheduler-locking off"
+
 # Next over the hello() call which will produce lots of output
 mi_gdb_test "220-exec-next" \
 	    "220\\^running(\r\n\\*running,thread-id=\"all\")?" \
Index: gdb-6.8.50.20090811/gdb/testsuite/gdb.mi/mi2-console.exp
===================================================================
--- gdb-6.8.50.20090811.orig/gdb/testsuite/gdb.mi/mi2-console.exp	2009-08-12 06:56:29.000000000 +0200
+++ gdb-6.8.50.20090811/gdb/testsuite/gdb.mi/mi2-console.exp	2009-08-12 06:58:45.000000000 +0200
@@ -47,6 +47,9 @@ if  { [gdb_compile "${srcdir}/${subdir}/
 
 mi_run_to_main
 
+# thread-id=\"all\" vs. thread-id=\"1\" below:
+mi_gdb_test "210-gdb-set scheduler-locking off" "210\\^done" "set scheduler-locking off"
+
 # Next over the hello() call which will produce lots of output
 send_gdb "220-exec-next\n"
 gdb_expect {
Index: gdb-6.8.50.20090811/gdb/testsuite/gdb.mi/mi-cli.exp
===================================================================
--- gdb-6.8.50.20090811.orig/gdb/testsuite/gdb.mi/mi-cli.exp	2009-06-09 19:12:42.000000000 +0200
+++ gdb-6.8.50.20090811/gdb/testsuite/gdb.mi/mi-cli.exp	2009-08-12 07:13:32.000000000 +0200
@@ -173,7 +173,7 @@ mi_execute_to "exec-continue" "breakpoin
 # Test that the token is output even for CLI commands
 # Also test that *stopped includes frame information.
 mi_gdb_test "34 next" \
-    ".*34\\\^running.*\\*running,thread-id=\"all\"" \
+    ".*34\\\^running.*\\*running,thread-id=\"1\"" \
     "34 next: run"
 
 if {!$async} {
