https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=218379


Index: gdb-6.8.50.20090803/gdb/symtab.c
===================================================================
--- gdb-6.8.50.20090803.orig/gdb/symtab.c	2009-08-03 16:09:36.000000000 +0200
+++ gdb-6.8.50.20090803/gdb/symtab.c	2009-08-03 16:13:13.000000000 +0200
@@ -2261,6 +2261,13 @@ find_pc_sect_line (CORE_ADDR pc, struct 
 	  /* See above comment about why warning is commented out */
 	  /* warning ("In stub for %s; unable to find real function/line info", SYMBOL_LINKAGE_NAME (msymbol)) */ ;
 	/* fall through */
+	/* `msymbol' trampoline may be located before its .text symbol
+	   but this text symbol may be the address we were looking for.
+	   Avoid `find_pc_sect_line'<->`find_pc_line' infinite loop.
+	   Red Hat Bug 218379.  */
+	else if (SYMBOL_VALUE (mfunsym) == pc)
+	  warning ("In stub for %s (0x%s); interlocked, please submit the binary to http://bugzilla.redhat.com", SYMBOL_LINKAGE_NAME (msymbol), paddress (target_gdbarch, pc));
+	/* fall through */
 	else
 	  return find_pc_line (SYMBOL_VALUE_ADDRESS (mfunsym), 0);
       }
