diff -up squid-3.1.9/aclocal.m4.ltdl squid-3.1.9/aclocal.m4
--- squid-3.1.9/aclocal.m4.ltdl	2010-10-25 13:34:41.000000000 +0200
+++ squid-3.1.9/aclocal.m4	2010-11-01 14:14:55.297324980 +0100
@@ -1117,7 +1117,7 @@ _LT_SET_OPTIONS([$0], [$1])
 LIBTOOL_DEPS="$ltmain"
 
 # Always use our own libtool.
-LIBTOOL='$(SHELL) $(top_builddir)/libtool'
+LIBTOOL='$(SHELL) libtool'
 AC_SUBST(LIBTOOL)dnl
 
 _LT_SETUP
diff -up squid-3.1.9/configure.ltdl squid-3.1.9/configure
--- squid-3.1.9/configure.ltdl	2010-10-25 13:35:32.000000000 +0200
+++ squid-3.1.9/configure	2010-11-01 14:14:55.320299060 +0100
@@ -9066,7 +9066,7 @@ fi
 LIBTOOL_DEPS="$ltmain"
 
 # Always use our own libtool.
-LIBTOOL='$(SHELL) $(top_builddir)/libtool'
+LIBTOOL='$(SHELL) libtool'
 
 
 
diff -up squid-3.1.9/Makefile.am.ltdl squid-3.1.9/Makefile.am
--- squid-3.1.9/Makefile.am.ltdl	2010-11-02 08:16:07.165707548 +0100
+++ squid-3.1.9/Makefile.am	2010-11-02 08:16:52.977565320 +0100
@@ -2,11 +2,8 @@
 #
 
 AUTOMAKE_OPTIONS = dist-bzip2 subdir-objects 1.5 foreign
-DIST_SUBDIRS	= compat lib libltdl snmplib scripts src icons errors contrib doc helpers test-suite tools
+DIST_SUBDIRS	= compat lib snmplib scripts src icons errors contrib doc helpers test-suite tools
 SUBDIRS		= compat lib $(makesnmplib)
-if USE_LOADABLE_MODULES
-SUBDIRS += libltdl
-endif
 SUBDIRS += scripts src icons errors doc helpers test-suite tools
 
 DISTCLEANFILES = include/stamp-h include/stamp-h[0-9]*
diff -up squid-3.1.9/Makefile.in.ltdl squid-3.1.9/Makefile.in
--- squid-3.1.9/Makefile.in.ltdl	2010-11-02 08:16:13.630458557 +0100
+++ squid-3.1.9/Makefile.in	2010-11-02 08:17:26.543714199 +0100
@@ -33,7 +33,7 @@ PRE_UNINSTALL = :
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@USE_LOADABLE_MODULES_TRUE@am__append_1 = libltdl
+# @USE_LOADABLE_MODULES_TRUE@am__append_1 = libltdl
 subdir = .
 DIST_COMMON = README $(am__configure_deps) $(srcdir)/Makefile.am \
 	$(srcdir)/Makefile.in $(top_srcdir)/configure \
@@ -272,7 +272,7 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = dist-bzip2 subdir-objects 1.5 foreign
-DIST_SUBDIRS = compat lib libltdl snmplib scripts src icons errors contrib doc helpers test-suite tools
+DIST_SUBDIRS = compat lib snmplib scripts src icons errors contrib doc helpers test-suite tools
 SUBDIRS = compat lib $(makesnmplib) $(am__append_1) scripts src icons \
 	errors doc helpers test-suite tools
 DISTCLEANFILES = include/stamp-h include/stamp-h[0-9]*
