Index: squid-3.1.12/src/cf.data.pre
===================================================================
--- squid-3.1.12.orig/src/cf.data.pre
+++ squid-3.1.12/src/cf.data.pre
@@ -918,7 +918,7 @@ http_access allow localnet
 http_access allow localhost
 
 # And finally deny all other access to this proxy
-http_access deny all
+http_access allow localhost
 NOCOMMENT_END
 DOC_END
 
@@ -2774,7 +2774,7 @@ DOC_END
 
 NAME: logfile_rotate
 TYPE: int
-DEFAULT: 10
+DEFAULT: 0
 LOC: Config.Log.rotateNumber
 DOC_START
 	Specifies the number of logfile rotations to make when you
@@ -4138,11 +4138,11 @@ COMMENT_END
 
 NAME: cache_mgr
 TYPE: string
-DEFAULT: webmaster
+DEFAULT: squid
 LOC: Config.adminEmail
 DOC_START
 	Email-address of local cache manager who will receive
-	mail if the cache dies.  The default is "webmaster."
+	mail if the cache dies.  The default is "squid."
 DOC_END
 
 NAME: mail_from
@@ -5275,10 +5275,10 @@ COMMENT_END
 NAME: icon_directory
 TYPE: string
 LOC: Config.icons.directory
-DEFAULT: @DEFAULT_ICON_DIR@
+DEFAULT: /usr/share/squid/icons
 DOC_START
 	Where the icons are stored. These are normally kept in
-	@DEFAULT_ICON_DIR@
+	/usr/share/squid/icons
 DOC_END
 
 NAME: global_internal_static
@@ -5320,8 +5320,10 @@ DEFAULT: none
 DOC_START
 	If you wish to create your own versions of the default
 	error files to customize them to suit your company copy
-	the error/template files to another directory and point
-	this tag at them.
+	/usr/share/squid/errors contains sets of error files
+        in different languages. The default error directory
+        is /etc/squid/errors, which is a link to one of these
+        error sets.
 
 	WARNING: This option will disable multi-language support
 	         on error pages if used.
