diff --git a/common/Makefile.am b/common/Makefile.am
index 7ea4107..aeae3d0 100644
--- a/common/Makefile.am
+++ b/common/Makefile.am
@@ -26,7 +26,6 @@ ntrackmodulesdir = $(pkglibdir)/modules/
 
 AM_CFLAGS = \
 	-Wall \
-	-Werror \
 	-pedantic
 
 AM_LDFLAGS =
diff --git a/common/Makefile.in b/common/Makefile.in
index f55d3bf..4168711 100644
--- a/common/Makefile.in
+++ b/common/Makefile.in
@@ -321,7 +321,6 @@ NTRACK_LT_VERSION = 3:3:3
 ntrackmodulesdir = $(pkglibdir)/modules/
 AM_CFLAGS = \
 	-Wall \
-	-Werror \
 	-pedantic
 
 AM_LDFLAGS = 
diff --git a/common/test/Makefile.am b/common/test/Makefile.am
index bfaf919..695da3a 100644
--- a/common/test/Makefile.am
+++ b/common/test/Makefile.am
@@ -25,7 +25,6 @@ test_libs = \
 AM_CFLAGS = \
 	-pedantic \
 	-Wall \
-	-Werror \
 	-I$(srcdir)/..
 
 check_PROGRAMS = \
diff --git a/common/test/Makefile.in b/common/test/Makefile.in
index 4d5c8f4..f0baa16 100644
--- a/common/test/Makefile.in
+++ b/common/test/Makefile.in
@@ -269,7 +269,6 @@ test_libs = \
 AM_CFLAGS = \
 	-pedantic \
 	-Wall \
-	-Werror \
 	-I$(srcdir)/..
 
 test_ntracklist_SOURCES = \
diff --git a/glib/Makefile.am b/glib/Makefile.am
index 6db288e..0c6133f 100644
--- a/glib/Makefile.am
+++ b/glib/Makefile.am
@@ -25,7 +25,6 @@ NTRACK_GLIB_LT_VERSION = 2:1:0
 
 AM_CFLAGS = \
 	-Wall \
-	-Werror \
 	-pedantic \
 	-I$(srcdir)/../common/
 
diff --git a/glib/Makefile.in b/glib/Makefile.in
index 243a184..3c926ed 100644
--- a/glib/Makefile.in
+++ b/glib/Makefile.in
@@ -324,7 +324,6 @@ CLEANFILES =
 NTRACK_GLIB_LT_VERSION = 2:1:0
 AM_CFLAGS = \
 	-Wall \
-	-Werror \
 	-pedantic \
 	-I$(srcdir)/../common/
 
diff --git a/glib/test/Makefile.am b/glib/test/Makefile.am
index fa0de94..d313c64 100644
--- a/glib/test/Makefile.am
+++ b/glib/test/Makefile.am
@@ -29,7 +29,6 @@ test_libs = \
 AM_CFLAGS = \
 	-pedantic \
 	-Wall \
-	-Werror \
 	-I$(srcdir)/.. \
 	-I$(srcdir)/../../common \
 	$(GLIB_CFLAGS)
diff --git a/glib/test/Makefile.in b/glib/test/Makefile.in
index 4013750..519b28e 100644
--- a/glib/test/Makefile.in
+++ b/glib/test/Makefile.in
@@ -262,7 +262,6 @@ test_libs = \
 AM_CFLAGS = \
 	-pedantic \
 	-Wall \
-	-Werror \
 	-I$(srcdir)/.. \
 	-I$(srcdir)/../../common \
 	$(GLIB_CFLAGS)
diff --git a/gobject/Makefile.am b/gobject/Makefile.am
index f4b1b12..8b8bb7b 100644
--- a/gobject/Makefile.am
+++ b/gobject/Makefile.am
@@ -25,7 +25,6 @@ NTRACK_GOBJECT_LT_VERSION = 1:0:0
 
 AM_CFLAGS = \
 	-Wall \
-	-Werror \
 	-pedantic \
 	-I$(srcdir)/../common/ \
 	-I$(srcdir)/../glib/
diff --git a/gobject/Makefile.in b/gobject/Makefile.in
index 1dddb27..65f2122 100644
--- a/gobject/Makefile.in
+++ b/gobject/Makefile.in
@@ -346,7 +346,6 @@ CLEANFILES = ntrack-gmarshal.h ntrack-enumtypes.h ntrack-enumtypes.c \
 NTRACK_GOBJECT_LT_VERSION = 1:0:0
 AM_CFLAGS = \
 	-Wall \
-	-Werror \
 	-pedantic \
 	-I$(srcdir)/../common/ \
 	-I$(srcdir)/../glib/
diff --git a/gobject/test/Makefile.am b/gobject/test/Makefile.am
index 44c5270..43e85c6 100644
--- a/gobject/test/Makefile.am
+++ b/gobject/test/Makefile.am
@@ -30,7 +30,6 @@ test_libs = \
 AM_CFLAGS = \
 	-pedantic \
 	-Wall \
-	-Werror \
 	-I$(srcdir)/.. \
 	-I$(srcdir)/../../common \
 	-I$(srcdir)/../../glib \
diff --git a/gobject/test/Makefile.in b/gobject/test/Makefile.in
index 56e5629..52add81 100644
--- a/gobject/test/Makefile.in
+++ b/gobject/test/Makefile.in
@@ -269,7 +269,6 @@ test_libs = \
 AM_CFLAGS = \
 	-pedantic \
 	-Wall \
-	-Werror \
 	-I$(srcdir)/.. \
 	-I$(srcdir)/../../common \
 	-I$(srcdir)/../../glib \
diff --git a/modules/Makefile.am b/modules/Makefile.am
index 6d564e8..e81809d 100644
--- a/modules/Makefile.am
+++ b/modules/Makefile.am
@@ -24,7 +24,6 @@ ntrackmodulesdir = $(pkglibdir)/modules
 
 AM_CFLAGS = \
 	-Wall \
-	-Werror \
 	-Wno-variadic-macros \
 	-pedantic \
 	-I$(top_srcdir)/common/
diff --git a/modules/Makefile.in b/modules/Makefile.in
index 425bfa3..d741631 100644
--- a/modules/Makefile.in
+++ b/modules/Makefile.in
@@ -350,7 +350,6 @@ NTRACK_LT_VERSION = 2:0:2
 ntrackmodulesdir = $(pkglibdir)/modules
 AM_CFLAGS = \
 	-Wall \
-	-Werror \
 	-Wno-variadic-macros \
 	-pedantic \
 	-I$(top_srcdir)/common/
diff --git a/qt4/Makefile.am b/qt4/Makefile.am
index 4373e2f..d0d0252 100644
--- a/qt4/Makefile.am
+++ b/qt4/Makefile.am
@@ -27,7 +27,6 @@ NTRACK_QT4_LT_VERSION = 1:1:0
 
 AM_CFLAGS = \
 	-Wall \
-	-Werror \
 	-pedantic \
 	-isystem $(shell pkg-config --variable=includedir QtCore)
 
diff --git a/qt4/Makefile.in b/qt4/Makefile.in
index 735ee02..c3a10d4 100644
--- a/qt4/Makefile.in
+++ b/qt4/Makefile.in
@@ -325,7 +325,6 @@ qt4pkgincludedir = $(pkgincludedir)/qt4
 NTRACK_QT4_LT_VERSION = 1:1:0
 AM_CFLAGS = \
 	-Wall \
-	-Werror \
 	-pedantic \
 	-isystem $(shell pkg-config --variable=includedir QtCore)
 
diff --git a/qt4/test/Makefile.am b/qt4/test/Makefile.am
index 725691d..8a3a825 100644
--- a/qt4/test/Makefile.am
+++ b/qt4/test/Makefile.am
@@ -26,7 +26,6 @@ test_libs = \
 
 AM_CXXFLAGS = \
 	-Wall \
-	-Werror \
 	-I$(srcdir)/.. \
 	-I$(srcdir)/../../common \
 	$(QTCORE_CFLAGS)
diff --git a/qt4/test/Makefile.in b/qt4/test/Makefile.in
index 3a5067c..822b96c 100644
--- a/qt4/test/Makefile.in
+++ b/qt4/test/Makefile.in
@@ -256,7 +256,6 @@ test_libs = \
 
 AM_CXXFLAGS = \
 	-Wall \
-	-Werror \
 	-I$(srcdir)/.. \
 	-I$(srcdir)/../../common \
 	$(QTCORE_CFLAGS)
