--- ogre/Samples/Common/CEGUIRenderer/src/Makefile.in.orig	2009-09-27 16:01:34.000000000 -0500
+++ ogre/Samples/Common/CEGUIRenderer/src/Makefile.in	2009-11-23 16:14:22.000000000 -0600
@@ -237,8 +237,8 @@
 INCLUDES = $(STLPORT_CFLAGS) -I$(top_srcdir)/Samples/Common/CEGUIRenderer/include -I$(top_srcdir)/OgreMain/include @CEGUI_CFLAGS@ $(CFLAGS_VISIBILITY)
 lib_LTLIBRARIES = libCEGUIOgreRenderer.la
 libCEGUIOgreRenderer_la_SOURCES = OgreCEGUIRenderer.cpp OgreCEGUITexture.cpp OgreCEGUIResourceProvider.cpp
-libCEGUIOgreRenderer_la_LIBADD = $(STLPORT_LIBS)
-libCEGUIOgreRenderer_la_LDFLAGS = -release @PACKAGE_VERSION@
+libCEGUIOgreRenderer_la_LIBADD = $(STLPORT_LIBS) @CEGUI_LIBS@ -lOgreMain
+libCEGUIOgreRenderer_la_LDFLAGS = -release @PACKAGE_VERSION@ -L$(top_builddir)/OgreMain/src
 all: all-am
 
 .SUFFIXES:
