diff -Nur xosd-2.2.14-old/libxosd.m4 xosd-2.2.14/libxosd.m4
--- xosd-2.2.14-old/libxosd.m4	2007-03-10 18:55:36.273899027 +0200
+++ xosd-2.2.14/libxosd.m4	2007-03-10 18:56:43.137734180 +0200
@@ -6,7 +6,7 @@
 dnl AM_PATH_LIBXOSD([ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]])
 dnl Test to see if libxosd is installed, and define LIBXOSD_CFLAGS, LIBXOSDLIBS
 dnl
-AC_DEFUN(AM_PATH_LIBXOSD,
+AC_DEFUN([AM_PATH_LIBXOSD],
 [dnl
 dnl Get the cflags and libraries from the xosd-config script
 dnl
