Index: projectM-2.0.1-Source/projectM.cpp
===================================================================
--- projectM-2.0.1-Source.orig/projectM.cpp
+++ projectM-2.0.1-Source/projectM.cpp
@@ -40,6 +40,7 @@
 #include "PipelineContext.hpp"
 //#include <xmms/plugin.h>
 #include <iostream>
+#include <locale.h>
 #include "projectM.hpp"
 #include "BeatDetect.hpp"
 #include "Preset.hpp"
@@ -468,6 +469,9 @@ static void *thread_callback(void *prjm)
     void projectM::projectM_init ( int gx, int gy, int fps, int texsize, int width, int height )
     {
 
+        /** ugly hack to fix locale issue **/
+        setlocale(LC_ALL, "C");
+
         /** Initialise start time */
         timeKeeper = new TimeKeeper(_settings.presetDuration,_settings.smoothPresetDuration, _settings.easterEgg);
 
