diff -Nur vim72-old/runtime/filetype.vim vim72/runtime/filetype.vim
--- vim72-old/runtime/filetype.vim	2008-10-18 01:14:23.000000000 +0300
+++ vim72/runtime/filetype.vim	2008-10-18 01:15:01.000000000 +0300
@@ -1677,6 +1677,7 @@
 " Shell scripts (sh, ksh, bash, bash2, csh); Allow .profile_foo etc.
 " Gentoo ebuilds are actually bash scripts
 au BufNewFile,BufRead .bashrc*,bashrc,bash.bashrc,.bash_profile*,.bash_logout*,*.bash,*.ebuild call SetFileTypeSH("bash")
+au BufNewFile,BufRead */etc/bash*completion*/* call SetFileTypeSH("bash")
 au BufNewFile,BufRead .kshrc*,*.ksh call SetFileTypeSH("ksh")
 au BufNewFile,BufRead /etc/profile,.profile*,*.sh,*.env call SetFileTypeSH(getline(1))
 
