diff -p -up shared-mime-info-0.30/freedesktop.org.xml.in~ shared-mime-info-0.30/freedesktop.org.xml.in
--- shared-mime-info-0.30/freedesktop.org.xml.in~	2008-05-02 12:57:22.000000000 +0200
+++ shared-mime-info-0.30/freedesktop.org.xml.in	2008-05-19 09:10:33.000000000 +0200
@@ -2613,6 +2613,7 @@ command to generate the output files.
   <mime-type type="application/x-x509-ca-cert">
     <_comment>DER/PEM/Netscape-encoded X.509 certificate</_comment>
     <generic-icon name="text-x-generic"/>
+    <sub-class-of type="text/plain"/>
     <glob pattern="*.der"/>
     <glob pattern="*.cer"/>
     <glob pattern="*.crt"/>
@@ -3813,9 +3814,6 @@ command to generate the output files.
     <acronym>VHDL</acronym>
     <expanded-acronym>Very-High-Speed Integrated Circuit Hardware Description Language</expanded-acronym>
     <sub-class-of type="text/plain"/>
-    <magic priority="20">
-      <match type="string" value="--" offset="0"/>
-    </magic>
     <glob pattern="*.vhd"/>
     <glob pattern="*.vhdl"/>
   </mime-type>
