diff -Naur libqalculate-0.9.6.old/po/POTFILES.in libqalculate-0.9.6/po/POTFILES.in
--- libqalculate-0.9.6.old/po/POTFILES.in	2005-06-01 14:13:50.000000000 -0400
+++ libqalculate-0.9.6/po/POTFILES.in	2007-07-08 20:21:17.000000000 -0400
@@ -1,6 +1,7 @@
 # List of source files containing translatable strings.
 [encoding: UTF-8]
 src/qalc.cc
+src/defs2doc.cc
 libqalculate/BuiltinFunctions.cc
 libqalculate/Calculator.cc
 libqalculate/DataSet.cc
@@ -12,3 +13,10 @@
 libqalculate/Unit.cc
 libqalculate/Variable.cc
 libqalculate/util.cc
+data/currencies.xml.in
+data/datasets.xml.in
+data/elements.xml.in
+data/functions.xml.in
+data/planets.xml.in
+data/units.xml.in
+data/variables.xml.in
