Index: qtsoap-2.7_1-opensource/config.pri
===================================================================
--- /dev/null
+++ qtsoap-2.7_1-opensource/config.pri
@@ -0,0 +1 @@
+SOLUTIONS_LIBRARY = yes
Index: qtsoap-2.7_1-opensource/common.pri
===================================================================
--- qtsoap-2.7_1-opensource.orig/common.pri
+++ qtsoap-2.7_1-opensource/common.pri
@@ -4,3 +4,5 @@ QTSOAP_LIBNAME = $$qtLibraryTarget(QtSol
 TEMPLATE -= fakelib
 QTSOAP_LIBDIR = $$PWD/lib
 unix:qtsoap-uselib:!qtsoap-buildlib:QMAKE_RPATHDIR += $$QTSOAP_LIBDIR
+QTSOAP_LIBNAME = $$qtLibraryTarget(qtsoap)
+VERSION=2.7.1
