Index: colorsvn-0.3.2/colorsvn-original
===================================================================
--- colorsvn-0.3.2.orig/colorsvn-original
+++ colorsvn-0.3.2/colorsvn-original
@@ -115,7 +115,7 @@ our %commandmap = ( "add" => ["add"],
 					);
 					
 # never colorize this commands
-our @preventcolor = ("commit", "help", "import", "lock", "switch", "propedit", "resolved", "revert", "unlock", "mkdir");
+our @preventcolor = ("commit", "help", "import", "lock", "switch", "propedit", "resolved", "revert", "unlock", "mkdir", "copy");
 					
 #
 # set default values
