Index: chromium-13.0.782.107/chrome/browser/download/download_extensions.cc
===================================================================
--- chromium-13.0.782.107.orig/chrome/browser/download/download_extensions.cc
+++ chromium-13.0.782.107/chrome/browser/download/download_extensions.cc
@@ -181,9 +181,11 @@ static const struct Executables {
   { "deb", AllowOnUserGesture },
   { "exe", AllowOnUserGesture },
   { "ksh", AllowOnUserGesture },
+  { "pisi", AllowOnUserGesture },
   { "rpm", AllowOnUserGesture },
   { "sh", AllowOnUserGesture },
   { "tcsh", AllowOnUserGesture },
+  { "zsh", AllowOnUserGesture },
 #endif
 };
 
