--- build/linux/system.gyp.orig	2013-01-28 16:03:43.709477316 +0000
+++ build/linux/system.gyp	2013-01-28 16:06:14.210254835 +0000
@@ -175,9 +175,6 @@
                          'gps_open',
                          'gps_close',
                          'gps_read',
-                         # We don't use gps_shm_read() directly, just to make
-                         # sure that libgps has the shared memory support.
-                         'gps_shm_read',
               ],
               'message': 'Generating libgps library loader.',
               'process_outputs_as_sources': 1,
