--- webkit/support/webkit_support_gfx.cc.orig	2012-02-01 11:00:50.000000000 +0100
+++ webkit/support/webkit_support_gfx.cc	2012-02-01 11:04:21.000000000 +0100
@@ -14,6 +14,12 @@
 #else
 #include "third_party/libpng/png.h"
 #endif
+
+#if defined(USE_SYSTEM_ZLIB)
+#include <zlib.h>
+#else
+#include "third_party/zlib/zlib.h"
+#endif
 }
 
 namespace webkit_support {
--- webkit/support/webkit_support.gypi.orig	2012-02-01 11:05:23.000000000 +0100
+++ webkit/support/webkit_support.gypi	2012-02-01 11:06:31.000000000 +0100
@@ -140,6 +140,7 @@
       'variables': { 'enable_wexit_time_destructors': 1, },
       'dependencies': [
         '<(DEPTH)/third_party/libpng/libpng.gyp:libpng',
+        '<(DEPTH)/third_party/zlib/zlib.gyp:zlib',
       ],
       'sources': [
         'webkit_support_gfx.h',
--- webkit/support/webkit_support.gypi.orig	2012-02-01 12:46:29.000000000 +0100
+++ webkit/support/webkit_support.gypi	2012-02-01 12:47:09.000000000 +0100
@@ -89,6 +89,7 @@
         '<(DEPTH)/crypto/crypto.gyp:crypto',
         '<(DEPTH)/net/net.gyp:net',
         '<(DEPTH)/skia/skia.gyp:skia',
+        '<(DEPTH)/third_party/zlib/zlib.gyp:zlib',
         '<(DEPTH)/ui/ui.gyp:ui',
         'glue',
         'webkit_support_gfx',
--- webkit/tools/test_shell/simple_clipboard_impl.cc.orig	2012-02-01 14:02:32.000000000 +0100
+++ webkit/tools/test_shell/simple_clipboard_impl.cc	2012-02-01 14:03:10.000000000 +0100
@@ -11,7 +11,11 @@
 #include "base/string16.h"
 #include "googleurl/src/gurl.h"
 #include "third_party/skia/include/core/SkBitmap.h"
+#if defined(USE_SYSTEM_ZLIB)
+#include <zlib.h>
+#else
 #include "third_party/zlib/zlib.h"
+#endif
 #include "ui/base/clipboard/clipboard.h"
 #include "ui/gfx/codec/png_codec.h"
 #include "ui/gfx/size.h"
