--- third_party/smhasher/src/City.cpp.orig	2013-03-13 23:48:29.232407565 +0000
+++ third_party/smhasher/src/City.cpp	2013-03-13 23:49:24.092714006 +0000
@@ -351,7 +351,7 @@
   }
 }
 
-#ifdef __SSE4_2__
+#if defined(__SSE4_2__) && defined(__x86_64__)
 #include <nmmintrin.h>
 
 // Requires len >= 240.
--- third_party/smhasher/src/City.h.orig	2013-03-13 23:49:50.122859412 +0000
+++ third_party/smhasher/src/City.h	2013-03-13 23:50:18.043015349 +0000
@@ -90,7 +90,7 @@
 
 // Conditionally include declarations for versions of City that require SSE4.2
 // instructions to be available.
-#ifdef __SSE4_2__
+#if defined(__SSE4_2__) && defined(__x86_64__)
 
 // Hash function for a byte array.
 uint128 CityHashCrc128(const char *s, size_t len);
