diff --git a/media/filters/ffmpeg_glue.h b/media/filters/ffmpeg_glue.h
index 17241b9..8a92312 100644
--- a/media/filters/ffmpeg_glue.h
+++ b/media/filters/ffmpeg_glue.h
@@ -28,9 +28,9 @@
 #include "base/basictypes.h"
 #include "base/memory/scoped_ptr.h"
 #include "media/base/media_export.h"
+#include "media/ffmpeg/ffmpeg_common.h"
 
 struct AVFormatContext;
-struct AVIOContext;
 
 namespace media {
 
diff --git a/media/media.gyp b/media/media.gyp
index df217d2..fde3830 100644
--- a/media/media.gyp
+++ b/media/media.gyp
@@ -363,6 +363,9 @@
           'dependencies': [
             '../third_party/ffmpeg/ffmpeg.gyp:ffmpeg',
           ],
+          'export_dependent_settings': [
+            '../third_party/ffmpeg/ffmpeg.gyp:ffmpeg',
+          ],
         }, {  # media_use_ffmpeg == 0
           # Exclude the sources that depend on ffmpeg.
           'sources!': [
--- a/media/ffmpeg/ffmpeg_common.cc.orig	2013-01-17 00:07:51.635057013 +0000
+++ b/media/ffmpeg/ffmpeg_common.cc	2013-01-17 00:15:50.867406811 +0000
@@ -10,6 +10,8 @@
 #include "media/base/video_frame.h"
 #include "media/base/video_util.h"
 
+#undef SampleFormat
+
 namespace media {
 
 // Why FF_INPUT_BUFFER_PADDING_SIZE? FFmpeg assumes all input buffers are
