Index: llvm-2.9/tools/clang-2.9/lib/Frontend/InitHeaderSearch.cpp
===================================================================
--- llvm-2.9.orig/tools/clang-2.9/lib/Frontend/InitHeaderSearch.cpp
+++ llvm-2.9/tools/clang-2.9/lib/Frontend/InitHeaderSearch.cpp
@@ -724,6 +724,12 @@ AddDefaultCPlusPlusIncludePaths(const ll
 
     //===------------------------------------------------------------------===//
 
+    // Pardus
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++",
+                                "i686-pc-linux-gnu", "", "", triple);
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++",
+                                "x86_64-pc-linux-gnu", "32", "", triple);
+
     // Exherbo (2010-01-25)
     AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.4.3",
                                 "x86_64-pc-linux-gnu", "32", "", triple);
