Index: llvm-2.9/tools/clang-2.9/lib/Driver/ToolChains.cpp
===================================================================
--- llvm-2.9.orig/tools/clang-2.9/lib/Driver/ToolChains.cpp
+++ llvm-2.9/tools/clang-2.9/lib/Driver/ToolChains.cpp
@@ -1449,7 +1449,8 @@ Linux::Linux(const HostInfo &Host, const
       GccTriple = "i586-suse-linux";
   }
 
-  const char* GccVersions[] = {"4.5.2", "4.5.1", "4.5", "4.4.5", "4.4.4",
+  const char* GccVersions[] = {"4.6.1", "4.6.0",
+                               "4.5.3", "4.5.2", "4.5.1", "4.5", "4.4.5", "4.4.4",
                                "4.4.3", "4.4", "4.3.4", "4.3.3", "4.3.2",
                                "4.3", "4.2.4", "4.2.3", "4.2.2", "4.2.1",
                                "4.2"};
