Index: llvm-2.7/bindings/ocaml/Makefile.ocaml
===================================================================
--- llvm-2.7.orig/bindings/ocaml/Makefile.ocaml
+++ llvm-2.7/bindings/ocaml/Makefile.ocaml
@@ -165,7 +165,7 @@ clean-deplibs:
 install-deplibs:
 	$(Verb) $(MKDIR) $(PROJ_libocamldir)
 	$(Verb) for i in $(DestLibs:$(PROJ_libocamldir)/%=%); do \
-	  ln -sf "$(PROJ_libdir)/$$i" "$(PROJ_libocamldir)/$$i"; \
+	  ln -sf "../llvm/$$i" "$(PROJ_libocamldir)/$$i"; \
 	done
 
 uninstall-deplibs:
