Index: smplayer-0.6.9_20101208/smplayer_enqueue.desktop
===================================================================
--- smplayer-0.6.9_20101208.orig/smplayer_enqueue.desktop
+++ smplayer-0.6.9_20101208/smplayer_enqueue.desktop
@@ -14,6 +14,7 @@ Name[pt]=Em fila no SMPlayer
 Name[ru]=Добавить в очередь SMPlayer
 Name[uk]=Додати до черги SMPlayer
 Name[de]=In SMPlayer einreihen
+Name[tr]=SMPlayer kuyruğuna ekle
 Comment=Add file(s) to the SMPlayer playlist
 Comment[cs]=Přidá soubor(y) do přehrávací fronty SMPlayeru
 Comment[es]=Añade ficheros a la lista de reproducción de SMPlayer
@@ -25,6 +26,7 @@ Comment[pt]=Adicionar ficheiro(s) à lis
 Comment[ru]=Добавить файл(ы) в конец списка воспроизведения SMPlayer
 Comment[uk]=Додати файл(и) до переліку програвання SMPlayer
 Comment[de]=Datei(en) zur SMPlayer Wiedergabeliste hinzufügen
+Comment[tr]=SMPlayer oynatma listesine dosya ekle
 Type=Application
 X-KDE-StartupNotify=false
 NoDisplay=true
