Index: smplayer-0.6.7_20090703/src/preferences.cpp
===================================================================
--- smplayer-0.6.7_20090703.orig/src/preferences.cpp
+++ smplayer-0.6.7_20090703/src/preferences.cpp
@@ -179,7 +179,7 @@ void Preferences::reset() {
 	font_file = "";
 	font_name = "";
 	use_fontconfig = false;
-	subcp = "ISO-8859-1";
+	subcp = "cp1254";
 	use_enca = false;
 	enca_lang = QString(QLocale::system().name()).section("_",0,0);
 	font_autoscale = 1;
