Index: smplayer-0.6.7_20090703/src/assstyles.cpp
===================================================================
--- smplayer-0.6.7_20090703.orig/src/assstyles.cpp
+++ smplayer-0.6.7_20090703/src/assstyles.cpp
@@ -23,7 +23,7 @@
 #include "colorutils.h"
 
 AssStyles::AssStyles() {
-	fontname = "Arial";
+	fontname = "Sans";
 	fontsize = 20;
 	primarycolor = 0xFFFFFF;
 	backcolor = 0;
Index: smplayer-0.6.7_20090703/src/videopreview/videopreview.cpp
===================================================================
--- smplayer-0.6.7_20090703.orig/src/videopreview/videopreview.cpp
+++ smplayer-0.6.7_20090703/src/videopreview/videopreview.cpp
@@ -339,7 +339,7 @@ bool VideoPreview::addPicture(const QStr
 	// Add current time text
 	if (prop.display_osd) {
 		QString stime = QTime().addSecs(time).toString("hh:mm:ss");
-		QFont font("Arial");
+		QFont font("Sans");
 		font.setBold(true);
 		QPainter painter(&scaled_picture);
 		painter.setPen( Qt::white );
