diff -urNp tar-1.22-orig/src/create.c tar-1.22/src/create.c
--- tar-1.22-orig/src/create.c	2009-05-15 10:50:38.000000000 +0200
+++ tar-1.22/src/create.c	2009-05-15 10:51:52.000000000 +0200
@@ -1827,7 +1827,8 @@ dump_file0 (struct tar_stat_info *st, co
 	    }
 	  else if (atime_preserve_option == replace_atime_preserve
		   && fd && (is_dir || original_size != 0)
-		   && set_file_atime (fd, parentfd, name, st->atime) != 0)
+		   && set_file_atime (fd, parentfd, name, st->atime) != 0
+		   && errno != EROFS )
 	    utime_error (p);
 	}
 
