diff -urNp tar-1.26-orig/gnu/stdio.in.h tar-1.26/gnu/stdio.in.h
--- tar-1.26-orig/gnu/stdio.in.h
+++ tar-1.26/gnu/stdio.in.h
@@ -139,7 +139,9 @@ _GL_WARN_ON_USE (fflush, "fflush is not 
    so any use of gets warrants an unconditional warning.  Assume it is
    always declared, since it is required by C89.  */
 #undef gets
+#if HAVE_RAW_DECL_GETS
 _GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
+#endif
 
 #if @GNULIB_FOPEN@
 # if @REPLACE_FOPEN@
