diff -urNp tar-1.23-orig/src/tar.c tar-1.23/src/tar.c
--- tar-1.23-orig/src/tar.c	2010-06-01 13:55:03.792173060 +0200
+++ tar-1.23/src/tar.c	2010-06-01 14:01:40.494172986 +0200
@@ -1911,6 +1911,10 @@ parse_opt (int key, char *arg, struct ar
 			  _("Invalid number")));
 	}
       break;
+
+    case OLD_ARCHIVE_OPTION:
+      set_archive_format ("v7");
+      break;
 
     case OVERWRITE_DIR_OPTION:
       old_files_option = DEFAULT_OLD_FILES;
