Index: kicad-0.0.20110816.orig/kicad/resources/linux/mime/applications/eeschema.desktop
===================================================================
--- kicad-0.0.20110816.orig.orig/kicad/resources/linux/mime/applications/eeschema.desktop
+++ kicad-0.0.20110816.orig/kicad/resources/linux/mime/applications/eeschema.desktop
@@ -1,10 +1,12 @@
 [Desktop Entry]
-Categories=Development;Electronics
+Categories=Qt;KDE;Education;Science;
 Comment=Design an electronic schematic
 Comment[fr]=Dessiner des schémas électroniques
+Comment[tr]=Elektronik şematik dizaynı yapın
 Exec=eeschema
 GenericName=Electronic schematic design
 GenericName[fr]=Saisie de schéma électronique
+GenericName[tr]=Elektronik şematik ve PCB dizayn araçları
 Icon=eeschema
 MimeType=application/x-kicad-schematic;
 Name=Eeschema
Index: kicad-0.0.20110816.orig/kicad/resources/linux/mime/applications/kicad.desktop
===================================================================
--- kicad-0.0.20110816.orig.orig/kicad/resources/linux/mime/applications/kicad.desktop
+++ kicad-0.0.20110816.orig/kicad/resources/linux/mime/applications/kicad.desktop
@@ -1,10 +1,12 @@
 [Desktop Entry]
-Categories=Development;Electronics
+Categories=Qt;KDE;Education;Science;
 Comment=Design a printed circuit board
 Comment[fr]=Concevoir un circuit imprimé
+Comment[tr]=Baskı devre dizayn edin
 Exec=kicad
 GenericName=EDA Suite
 GenericName[fr]=Suite logicielle de conception électronique
+GenericName[tr]=Elektronik şematik ve PCB dizayn araçları
 Icon=kicad
 MimeType=application/x-kicad-project;
 Name=KiCad
Index: kicad-0.0.20110816.orig/kicad/resources/linux/mime/mimelnk/application/x-kicad-project.desktop
===================================================================
--- kicad-0.0.20110816.orig.orig/kicad/resources/linux/mime/mimelnk/application/x-kicad-project.desktop
+++ kicad-0.0.20110816.orig/kicad/resources/linux/mime/mimelnk/application/x-kicad-project.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
-Type=MimeType
-MimeType=application/x-kicad-project
+Name=KiCad
+MimeType=application/x-kicad-project;
 Icon=kicad
-Patterns=*.pro
 Comment=KiCad Project
 Comment[fr]=Projet KiCad
+Type=Application
Index: kicad-0.0.20110816.orig/kicad/resources/linux/mime/mimelnk/application/x-kicad-schematic.desktop
===================================================================
--- kicad-0.0.20110816.orig.orig/kicad/resources/linux/mime/mimelnk/application/x-kicad-schematic.desktop
+++ kicad-0.0.20110816.orig/kicad/resources/linux/mime/mimelnk/application/x-kicad-schematic.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
-Type=MimeType
-MimeType=application/x-kicad-schematic
+MimeType=application/x-kicad-schematic;
 Icon=eeschema
-Patterns=*.sch
 Comment=KiCad Schematic
 Comment[fr]=Schéma électronique KiCad
+Name=KiCad
+Type=Application
