--- a/scripts/kicad-get-rss
+++ b/scripts/kicad-get-rss
@@ -26,7 +26,7 @@
 
 # Get feed and save
 EXEC="/usr/bin/curl -s" #Path to curl
-`$EXEC $URI &> $FEEDFILE`
+`$EXEC $URI > $FEEDFILE 2>&1`
 
 # Get commit description
 cat $FEEDFILE                | \
@@ -49,7 +49,7 @@
 sed -e '/export/d'    | \
 sed -e 's/[ \t]*//'   | \
 sed -e 's/<link>//'   | \
-sed -e 's/<\/link>//' &> $URLFILE
+sed -e 's/<\/link>//' > $URLFILE 2>&1
 
 # Get commit messages from urls
 exec < $URLFILE
