--- a/include/common.h
+++ b/include/common.h
@@ -46,11 +46,11 @@
 #   error "You must use '--enable-printarch' in your wx library configuration."
 #endif
 
-#if defined( __WXGTK__ )
-#   if !wxUSE_LIBGNOMEPRINT && !wxUSE_GTKPRINT && !SWIG
-#       error "You must use '--with-gnomeprint' or '--with-gtkprint' in your wx library configuration."
-#   endif
-#endif
+//#if defined( __WXGTK__ )
+//#   if !wxUSE_LIBGNOMEPRINT && !wxUSE_GTKPRINT && !SWIG
+//#       error "You must use '--with-gnomeprint' or '--with-gtkprint' in your wx library configuration."
+//#   endif
+//#endif
 
 
 class wxAboutDialogInfo;
