--- a/common/basicframe.cpp
+++ b/common/basicframe.cpp
@@ -310,6 +310,12 @@
      * else the specific doc file (its name is in wxGetApp().GetHelpFileName())
      * The document for beginners is the same for all KiCad utilities
      */
+
+    /* as the sourceless file /Getting_Started_in_KiCad.pdf was
+       not installed due to DFSG policy, implement a workaround,
+       for example using
+       http://www.kicad-pcb.org/download/attachments/1212538/Getting_Started_in_KiCad.pdf
+    */
     if( event.GetId() == wxID_INDEX )
     {
         // Temporary change the help filename
