Index: kicad-0.0.20110228.orig/kicad/common/gestfich.cpp
===================================================================
--- kicad-0.0.20110228.orig/kicad/common/gestfich.cpp
+++ kicad-0.0.20110228.orig/kicad/common/gestfich.cpp
@@ -617,6 +617,7 @@ bool OpenPDF( const wxString& file )
             /* here is a list of PDF viewers candidates */
             const static wxString tries[] =
             {
+                wxT( "/usr/bin/xdg-open" ),
                 wxT( "/usr/bin/evince" ),
                 wxT( "/usr/bin/gpdf" ),
                 wxT( "/usr/bin/konqueror" ),
