diff -Nuar namebench-1.3.1.orig/libnamebench/config.py namebench-1.3.1/libnamebench/config.py
--- namebench-1.3.1.orig/libnamebench/config.py	2010-06-06 19:19:28.000000000 +0300
+++ namebench-1.3.1/libnamebench/config.py	2010-07-05 12:39:43.382484770 +0300
@@ -34,7 +34,7 @@
 SANITY_REFERENCE_URL = 'http://namebench.googlecode.com/svn/trunk/config/hostname_reference.cfg'
 
 
-def GetConfiguration(filename='config/namebench.cfg'):
+def GetConfiguration(filename='/etc/namebench/namebench.cfg'):
   """Get all of our configuration setup, args and config file."""
   (options, args) = DefineAndParseOptions(filename=filename)
   (configured_options, global_ns, regional_ns) = ProcessConfigurationFile(options)
@@ -131,7 +131,7 @@
     except:
       pass
 
-  ref_file = util.FindDataFile('config/hostname_reference.cfg')
+  ref_file = util.FindDataFile('/etc/namebench/hostname_reference.cfg')
   local_config = ConfigParser.ConfigParser()
   local_config.read(ref_file)
 
diff -Nuar namebench-1.3.1.orig/libnamebench/data_sources.py namebench-1.3.1/libnamebench/data_sources.py
--- namebench-1.3.1.orig/libnamebench/data_sources.py	2010-06-06 19:19:28.000000000 +0300
+++ namebench-1.3.1/libnamebench/data_sources.py	2010-07-05 13:16:10.706692714 +0300
@@ -38,7 +38,7 @@
 
 GLOBAL_DATA_CACHE = {}
 
-DEFAULT_CONFIG_PATH = 'config/data_sources.cfg'
+DEFAULT_CONFIG_PATH = '/etc/namebench/data_sources.cfg'
 MAX_NON_UNIQUE_RECORD_COUNT = 500000
 MAX_FILE_MTIME_AGE_DAYS = 45
 MIN_FILE_SIZE = 10000
diff -Nuar namebench-1.3.1.orig/setup.py namebench-1.3.1/setup.py
--- namebench-1.3.1.orig/setup.py	2010-06-06 19:19:39.000000000 +0300
+++ namebench-1.3.1/setup.py	2010-07-05 12:45:11.544731715 +0300
@@ -27,25 +27,7 @@
     pass
 
 # If you don't want 3rd party libraries included, set this in your environment.  
-if os.getenv('NO_THIRD_PARTY', None):
-  packages=['libnamebench']
-else:
-  packages = [
-      'libnamebench',
-      'nb_third_party',
-      'nb_third_party/dns',
-      'nb_third_party/dns/rdtypes',
-      'nb_third_party/dns/rdtypes/ANY',
-      'nb_third_party/dns/rdtypes/IN',
-      'nb_third_party/graphy',
-      'nb_third_party/jinja2',
-      'nb_third_party/httplib2',
-      'nb_third_party/simplejson',
-      'nb_third_party/graphy/backends',
-      'nb_third_party/graphy/backends/google_chart_api'
-  ]
-
-
+packages=['libnamebench']
 
 RT_MANIFEST = 24
 
@@ -114,19 +96,19 @@
       license='Apache 2.0',
       scripts=['namebench.py'],
       data_files=[
-          ('namebench/config',
+          ('/etc/namebench',
            ['config/namebench.cfg',
             'config/hostname_reference.cfg',
             'config/data_sources.cfg']
           ),
-          ('namebench/templates',
+          ('share/namebench/templates',
            ['templates/ascii.tmpl',
             'templates/html.tmpl',
             'templates/resolv.conf.tmpl',
             'templates/style.css'
            ]
           ),
-          ('namebench/data',
+          ('share/namebench/data',
            ['data/alexa-top-2000-domains.txt',
             'data/cache-hit.txt',
             'data/cache-miss.txt',
