diff -Nuar a/desktoptheme/air/metadata.desktop b/desktoptheme/air/metadata.desktop
--- a/desktoptheme/air/metadata.desktop	2013-06-28 20:12:33.529795308 +0300
+++ b/desktoptheme/air/metadata.desktop	2013-08-27 09:51:26.169868217 +0300
@@ -160,7 +160,7 @@
 X-KDE-PluginInfo-EnabledByDefault=true
 
 [Wallpaper]
-defaultWallpaperTheme=Elarun
+defaultWallpaperTheme=aslanon2
 defaultFileSuffix=.png
-defaultWidth=2560
-defaultHeight=1600
+defaultWidth=1600
+defaultHeight=1011
