diff -Nuar mod_perl-2.0.7.OSmanOS/src/modules/perl/modperl_apache_compat.h mod_perl-2.0.7/src/modules/perl/modperl_apache_compat.h
--- mod_perl-2.0.7.OSmanOS/src/modules/perl/modperl_apache_compat.h	2011-02-08 04:00:10.000000000 +0200
+++ mod_perl-2.0.7/src/modules/perl/modperl_apache_compat.h	2013-12-28 15:12:56.886624621 +0200
@@ -65,6 +65,10 @@
 
 #define MP_HTTPD_OVERRIDE_HTACCESS (OR_LIMIT|OR_OPTIONS|OR_FILEINFO|OR_AUTHCFG|OR_INDEXES)
 
+#ifndef OPT_INCNOEXEC
+#define OPT_INCNOEXEC OPT_INCLUDES
+#endif
+
 #define MP_HTTPD_OVERRIDE_OPTS_UNSET (-1)
 #define MP_HTTPD_OVERRIDE_OPTS_DEFAULT (OPT_UNSET | \
                                         OPT_ALL | \
diff -Nuar mod_perl-2.0.7.OSmanOS/src/modules/perl/modperl_config.c mod_perl-2.0.7/src/modules/perl/modperl_config.c
--- mod_perl-2.0.7.OSmanOS/src/modules/perl/modperl_config.c	2011-02-08 04:00:10.000000000 +0200
+++ mod_perl-2.0.7/src/modules/perl/modperl_config.c	2013-12-28 15:14:17.099957577 +0200
@@ -479,7 +479,7 @@
     PerlInterpreter *perl;
 } svav_param_t;
 
-static void *svav_getstr(void *buf, size_t bufsiz, void *param)
+static apr_status_t svav_getstr(void *buf, apr_size_t bufsiz, void *param)
 {
     svav_param_t *svav_param = (svav_param_t *)param;
     dTHXa(svav_param->perl);
@@ -488,7 +488,7 @@
     STRLEN n_a;
 
     if (svav_param->ix > AvFILL(av)) {
-        return NULL;
+        return APR_FROM_OS_ERROR(ENOMEM);
     }
 
     sv = AvARRAY(av)[svav_param->ix++];
@@ -496,7 +496,7 @@
 
     apr_cpystrn(buf, SvPVX(sv), bufsiz);
 
-    return buf;
+    return APR_SUCCESS;
 }
 
 const char *modperl_config_insert(pTHX_ server_rec *s,
