--- Makefile~	2013-04-06 23:49:41.000000000 +0200
+++ Makefile	2014-02-26 19:21:51.000000000 +0100
@@ -57,7 +57,7 @@
 ifdef DEFAULT_LANG
 	@echo $(DEFAULT_LANG) >bootlogo.dir/lang
 endif
-	@sh -c 'cd bootlogo.dir; chmod +t * ; chmod -t init languages'
+	@sh -c 'cd bootlogo.dir; chmod +t * ; chmod -t init languages gfxboot.cfg pabout.txt lang'
 	@sh -c 'cd bootlogo.dir; echo * | sed -e "s/ /\n/g" | cpio --quiet -o >../bootlogo'
 
 clean:
