--- a/src/polkitgnomeauthenticationdialog.c	2010-03-23 14:39:30.258242167 +0200
+++ b/src/polkitgnomeauthenticationdialog.c	2010-03-23 14:39:21.003240995 +0200
@@ -717,6 +724,19 @@
       gtk_widget_set_sensitive (dialog->priv->prompt_label, FALSE);
       gtk_widget_set_sensitive (dialog->priv->password_entry, FALSE);
       gtk_widget_set_sensitive (dialog->priv->auth_button, FALSE);
+
+      int n;
+      struct passwd *passwd;
+      for (n = 0; dialog->priv->users[n] != NULL; n++)
+      {
+        passwd = getpwnam (dialog->priv->users[n]);
+        if (getuid() == passwd->pw_uid)
+        {
+            GtkComboBox *combo = GTK_COMBO_BOX (dialog->priv->user_combobox);
+            gtk_combo_box_set_active (GTK_COMBO_BOX (combo), n+1);
+            break;
+        }
+      }
     }
   else
     {
