Index: polkit-gnome-0.96/configure.ac
===================================================================
--- polkit-gnome-0.96.orig/configure.ac
+++ polkit-gnome-0.96/configure.ac
@@ -23,8 +23,6 @@ AC_ISC_POSIX
 AC_HEADER_STDC
 AC_PROG_LIBTOOL
 
-GTK_DOC_CHECK([1.3])
-
 #### gcc warning flags
 
 if test "x$GCC" = "xyes"; then
@@ -189,8 +187,6 @@ po/Makefile.in
 polkitgtk/Makefile
 polkitgtk/polkit-gtk-1.pc
 src/Makefile
-docs/version.xml
-docs/Makefile
 ])
 
 AC_OUTPUT
Index: polkit-gnome-0.96/Makefile.am
===================================================================
--- polkit-gnome-0.96.orig/Makefile.am
+++ polkit-gnome-0.96/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = polkitgtk src po docs
+SUBDIRS = polkitgtk src po
 
 EXTRA_DIST = \
 	autogen.sh		\
