Index: polkit-gnome-0.96/src/polkitgnomeauthenticationdialog.c
===================================================================
--- polkit-gnome-0.96.orig/src/polkitgnomeauthenticationdialog.c
+++ polkit-gnome-0.96/src/polkitgnomeauthenticationdialog.c
@@ -201,7 +201,7 @@ create_user_combobox (PolkitGnomeAuthent
       if (passwd->pw_dir != NULL)
         {
           gchar *path;
-          path = g_strdup_printf ("%s/.face", passwd->pw_dir);
+          path = g_strdup_printf ("%s/.face.icon", passwd->pw_dir);
           /* TODO: we probably shouldn't hard-code the size to 16x16 */
           pixbuf = gdk_pixbuf_new_from_file_at_scale (path, 16, 16, TRUE, NULL);
           g_free (path);
