diff -Naur src-orig/polkitgnomeauthenticationdialog.c src/polkitgnomeauthenticationdialog.c
--- src-orig/polkitgnomeauthenticationdialog.c	2009-06-22 19:19:24.000000000 +0300
+++ src/polkitgnomeauthenticationdialog.c	2010-02-02 09:53:33.768652074 +0200
@@ -676,19 +676,24 @@
   /* --- */
 
   label = gtk_label_new (NULL);
+
+  /*
   gtk_label_set_use_markup (GTK_LABEL (label), TRUE);
   s = g_strdup_printf ("<small><a href=\"%s\">%s</a></small>",
                        dialog->priv->action_id,
                        dialog->priv->action_id);
   gtk_label_set_markup (GTK_LABEL (label), s);
   g_free (s);
-  gtk_misc_set_alignment (GTK_MISC (label), 0, 1.0);
-  add_row (table, rows++, _("<small><b>Action:</b></small>"), label);
   g_signal_connect (label, "activate-link", G_CALLBACK (action_id_activated), NULL);
 
   s = g_strdup_printf (_("Click to edit %s"), dialog->priv->action_id);
   gtk_widget_set_tooltip_markup (label, s);
   g_free (s);
+  */
+
+  gtk_label_set_text(GTK_LABEL (label), dialog->priv->action_id);
+  gtk_misc_set_alignment (GTK_MISC (label), 0, 1.0);
+  add_row (table, rows++, _("<small><b>Action:</b></small>"), label);
 
   /* --- */
 
