Index: polkit-gnome-0.96/src/polkitgnomeauthenticator.c
===================================================================
--- polkit-gnome-0.96.orig/src/polkitgnomeauthenticator.c
+++ polkit-gnome-0.96/src/polkitgnomeauthenticator.c
@@ -279,16 +279,7 @@ session_request (PolkitAgentSession *ses
 
   /* Fix up, and localize, password prompt if it's password auth */
   if (g_ascii_strncasecmp (request, "password:", 9) == 0)
-    {
-      if (strcmp (g_get_user_name (), authenticator->selected_user) != 0)
-        {
-          modified_request = g_strdup_printf (_("_Password for %s:"), authenticator->selected_user);
-        }
-      else
-        {
-          modified_request = g_strdup (_("_Password:"));
-        }
-    }
+      modified_request = g_strdup_printf (_("_Password:"), authenticator->selected_user);
   else
     {
       modified_request = g_strdup (request);
