diff -up pigment-0.3.17/configure.ac.automake-1_13 pigment-0.3.17/configure.ac
--- pigment-0.3.17/configure.ac.automake-1_13	2013-01-09 02:11:46.080468260 +0100
+++ pigment-0.3.17/configure.ac	2013-01-09 02:13:48.452313596 +0100
@@ -11,7 +11,7 @@ AG_GST_INIT
 
 AC_CONFIG_SRCDIR([pgm/pgm.c])
 AC_CONFIG_AUX_DIR([build-aux])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 AM_INIT_AUTOMAKE([1.7 dist-bzip2])
 AS_VERSION
 AS_NANO(PGM_SVN="no", PGM_SVN="yes")
