Index: wxPython/config.py
===================================================================
--- wxPython.orig/config.py
+++ wxPython/config.py
@@ -990,8 +990,6 @@ elif os.name == 'posix' or COMPILER == '
     if debug:
         cflags.append('-g')
         cflags.append('-O0')
-    else:
-        cflags.append('-O3')
 
     lflags = os.popen(WX_CONFIG + ' --libs', 'r').read()[:-1]
     MONOLITHIC = (lflags.find("_xrc") == -1)
Index: wxPython/wx/build/config.py
===================================================================
--- wxPython.orig/wx/build/config.py
+++ wxPython/wx/build/config.py
@@ -990,8 +990,6 @@ elif os.name == 'posix' or COMPILER == '
     if debug:
         cflags.append('-g')
         cflags.append('-O0')
-    else:
-        cflags.append('-O3')
 
     lflags = os.popen(WX_CONFIG + ' --libs', 'r').read()[:-1]
     MONOLITHIC = (lflags.find("_xrc") == -1)
