diff -ur gdk-pixbuf-2.22.1.orig//gdk-pixbuf/Makefile.am gdk-pixbuf-2.22.1/gdk-pixbuf/Makefile.am
--- gdk-pixbuf-2.22.1.orig//gdk-pixbuf/Makefile.am	2010-09-21 19:42:39.000000000 +0300
+++ gdk-pixbuf-2.22.1/gdk-pixbuf/Makefile.am	2010-12-03 00:28:55.794000145 +0200
@@ -677,7 +677,7 @@
 if HAVE_INTROSPECTION
 
 GdkPixbuf-2.0.gir: libgdk_pixbuf-2.0.la Makefile
-GdkPixbuf_2_0_gir_SCANNERFLAGS = --warn-all --strip-prefix=Gdk
+GdkPixbuf_2_0_gir_SCANNERFLAGS = --warn-all --identifier-prefix=Gdk --symbol-prefix=gdk
 GdkPixbuf_2_0_gir_INCLUDES = GModule-2.0 Gio-2.0
 GdkPixbuf_2_0_gir_LIBS = libgdk_pixbuf-2.0.la
 GdkPixbuf_2_0_gir_CFLAGS = $(INCLUDES)
