diff -up njam-1.25-src/src/njamedit.cpp~ njam-1.25-src/src/njamedit.cpp
--- njam-1.25-src/src/njamedit.cpp~	2010-09-10 09:33:46.000000000 +0200
+++ njam-1.25-src/src/njamedit.cpp	2010-09-10 09:40:08.140574002 +0200
@@ -114,7 +114,7 @@ void NjamEngine::LevelEditor()
 				key = SDLK_a;
 			else if (CheckForSave())
 			{
-				LogFile::LogFile("Saving maps");
+				LogFile("Saving maps");
 				m_Maps.Save(filename);
 				level_type_was = level_type;
 				changed = false;
@@ -139,7 +139,7 @@ void NjamEngine::LevelEditor()
 						"levels/%s.%s",
 #endif					
 						filename, types[level_type]);
-					LogFile::LogFile("Saving maps");
+					LogFile("Saving maps");
 					m_Maps.Save(buf);
 					level_type_was = level_type;
 					changed = false;
