diff -Naur libshout-2.2.2.orig/m4/acx_pthread.m4 libshout-2.2.2/m4/acx_pthread.m4
--- libshout-2.2.2.orig/m4/acx_pthread.m4	2009-04-23 03:28:22.000000000 +0300
+++ libshout-2.2.2/m4/acx_pthread.m4	2009-04-23 04:01:37.000000000 +0300
@@ -62,7 +62,7 @@
 # pthread-config: use pthread-config program (for GNU Pth library)
 
 case "${host_cpu}-${host_os}" in
-        *solaris*)
+        *solaris*|*linux*)
 
         # On Solaris (at least, for some versions), libc contains stubbed
         # (non-functional) versions of the pthreads routines, so link-based
