diff -Naur dbus-1.2.1.orig/bus/policy.c dbus-1.2.1/bus/policy.c
--- dbus-1.2.1.orig/bus/policy.c	2008-06-09 23:28:03.000000000 +0300
+++ dbus-1.2.1/bus/policy.c	2008-06-09 23:32:05.000000000 +0300
@@ -28,6 +28,7 @@
 #include <dbus/dbus-list.h>
 #include <dbus/dbus-hash.h>
 #include <dbus/dbus-internals.h>
+#include <sys/types.h>
 
 BusPolicyRule*
 bus_policy_rule_new (BusPolicyRuleType type,
@@ -1236,6 +1237,12 @@
                                  DBusConnection   *connection,
                                  const DBusString *service_name)
 {
+  // Always allow root user to own all names
+  uid_t uid;
+  if (dbus_connection_get_unix_user(connection, &uid) && uid == 0) {
+      return TRUE;
+  }
+  
   DBusList *link;
   dbus_bool_t allowed;
   
