diff -up pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs.no-standin pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs
--- pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs.no-standin 2010-09-29 12:05:47.434923955 -0400
+++ pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs    2010-09-29 12:06:00.840923486 -0400
@@ -65,37 +65,6 @@
   )
 )
 
-;; From gtk-clutter-standin.h
-
-(define-function standin_get_type
-  (c-name "gtk_clutter_standin_get_type")
-  (return-type "GType")
-)
-
-(define-function standin_new
-  (c-name "gtk_clutter_standin_new")
-  (is-constructor-of "GtkClutterStandin")
-  (return-type "GtkWidget*")
-  (properties
-    '("actor" (optional))
-  )
-)
-
-(define-method set_actor
-  (of-object "GtkClutterStandin")
-  (c-name "gtk_clutter_standin_set_actor")
-  (return-type "none")
-  (parameters
-    '("ClutterActor*" "actor")
-  )
-)
-
-(define-method get_actor
-  (of-object "GtkClutterStandin")
-  (c-name "gtk_clutter_standin_get_actor")
-  (return-type "ClutterActor*")
-)
-
 ;; From gtk-clutter-util.h
 
 
diff -up pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs.no-standin pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs
--- pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs.no-standin   2010-09-29 12:06:12.967924058 -0400
+++ pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs      2010-09-29 12:06:25.660924319 -0400
@@ -16,13 +16,6 @@
   (gtype-id "GTK_CLUTTER_TYPE_EMBED")
 )
 
-(define-object Standin
-  (in-module "GtkClutter")
-  (parent "GtkWidget")
-  (c-name "GtkClutterStandin")
-  (gtype-id "GTK_CLUTTER_TYPE_STANDIN")
-)
-
 (define-object Viewport
   (in-module "GtkClutter")
   (parent "ClutterActor")
