diff -urP ../BUILD/pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs pyclutter-gtk/clutter-gtk/cluttergtk.defs
--- ../BUILD/pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs	2010-01-25 08:30:27.000000000 -0500
+++ pyclutter-gtk/clutter-gtk/cluttergtk.defs	2010-09-29 11:42:19.753924000 -0400
@@ -2,6 +2,50 @@
 
 (include "cluttergtk-types.defs")
 
+;; From gtk-clutter-actor.h
+
+(define-function actor_get_type
+  (c-name "gtk_clutter_actor_get_type")
+  (return-type "GType")
+)
+
+(define-function actor_new
+  (c-name "gtk_clutter_actor_new")
+  (is-constructor-of "GtkClutterActor")
+  (return-type "ClutterActor*")
+)
+
+(define-function actor_new_with_contents
+  (c-name "gtk_clutter_actor_new_with_contents")
+  (return-type "ClutterActor*")
+  (parameters
+    '("GtkWidget*" "contents")
+  )
+)
+
+(define-method set_receives_events
+  (of-object "GtkClutterActor")
+  (c-name "gtk_clutter_actor_set_receives_events")
+  (return-type "none")
+  (parameters
+    '("gboolean" "receives_events")
+  )
+)
+
+(define-method get_receives_events
+  (of-object "GtkClutterActor")
+  (c-name "gtk_clutter_actor_get_receives_events")
+  (return-type "gboolean")
+)
+
+(define-method get_widget
+  (of-object "GtkClutterActor")
+  (c-name "gtk_clutter_actor_get_widget")
+  (return-type "GtkWidget*")
+)
+
+;; From gtk-clutter-embed.h
+
 (define-function clutter_embed_get_type
   (c-name "gtk_clutter_embed_get_type")
   (return-type "GType")
@@ -66,6 +110,37 @@
   )
 )
 
+;; From gtk-clutter-standin.h
+
+(define-function standin_get_type
+  (c-name "gtk_clutter_standin_get_type")
+  (return-type "GType")
+)
+
+(define-function standin_new
+  (c-name "gtk_clutter_standin_new")
+  (is-constructor-of "GtkClutterStandin")
+  (return-type "GtkWidget*")
+  (properties
+    '("actor" (optional))
+  )
+)
+
+(define-method set_actor
+  (of-object "GtkClutterStandin")
+  (c-name "gtk_clutter_standin_set_actor")
+  (return-type "none")
+  (parameters
+    '("ClutterActor*" "actor")
+  )
+)
+
+(define-method get_actor
+  (of-object "GtkClutterStandin")
+  (c-name "gtk_clutter_standin_get_actor")
+  (return-type "ClutterActor*")
+)
+
 ;; From gtk-clutter-util.h
 
 
@@ -239,6 +314,27 @@
   )
 )
 
+;; From gtk-clutter-window.h
+
+(define-function gtk_clutter_window_get_type
+  (c-name "gtk_clutter_window_get_type")
+  (return-type "GType")
+)
+
+(define-function gtk_clutter_window_new
+  (c-name "gtk_clutter_window_new")
+  (is-constructor-of "GtkClutterWindow")
+  (return-type "GtkWidget*")
+)
+
+(define-method get_stage
+  (of-object "GtkClutterWindow")
+  (c-name "gtk_clutter_window_get_stage")
+  (return-type "ClutterActor*")
+  (parameters
+  )
+)
+
 ;; From gtk-clutter-zoomable.h
 
 (define-function gtk_clutter_zoomable_get_type
diff -urP ../BUILD/pyclutter-gtk-0.10.0/clutter-gtk/cluttergtkmodule.c pyclutter-gtk/clutter-gtk/cluttergtkmodule.c
--- ../BUILD/pyclutter-gtk-0.10.0/clutter-gtk/cluttergtkmodule.c	2010-01-25 10:39:00.000000000 -0500
+++ pyclutter-gtk/clutter-gtk/cluttergtkmodule.c	2010-09-29 11:42:19.754924000 -0400
@@ -86,12 +86,6 @@
   m = Py_InitModule ("_cluttergtk", pycluttergtk_functions);
   d = PyModule_GetDict (m);
 
-  PyModule_AddObject (m, "__version__",
-                      Py_BuildValue ("(iii)",
-                                     PYCLUTTERGTK_MAJOR_VERSION,
-                                     PYCLUTTERGTK_MINOR_VERSION,
-                                     PYCLUTTERGTK_MICRO_VERSION));
-
   init_clutter_gtk ();
   pycluttergtk_register_classes (d);
 
diff -urP ../BUILD/pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.override pyclutter-gtk/clutter-gtk/cluttergtk.override
--- ../BUILD/pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.override	2010-01-25 08:30:27.000000000 -0500
+++ pyclutter-gtk/clutter-gtk/cluttergtk.override	2010-09-29 11:42:19.753924000 -0400
@@ -14,10 +14,14 @@
 import gtk.Widget as PyGtkWidget_Type
 import gtk.Adjustment as PyGtkAdjustment_Type
 import gtk.Container as PyGtkContainer_Type
+import gtk.Window as PyGtkWindow_Type
 %%
 ignore-glob
   *_get_type
 %%
+ignore
+  gtk_clutter_actor_new_with_contents
+%%
 body
 
 static PyObject *
@@ -432,3 +436,30 @@
 
     return Py_BuildValue ("(NN)", py_h_adjust, py_v_adjust);
 }
+%%
+new-constructor GTK_CLUTTER_TYPE_ACTOR
+%%
+override gtk_clutter_actor_new kwargs
+static int
+_wrap_gtk_clutter_actor_new (PyGObject *self,
+                             PyObject  *args,
+                             PyObject  *kwargs)
+{
+  static char *kwlist[] = { "contents", NULL };
+  PyGObject *py_contents = NULL;
+  GtkWidget *bin;
+
+  if (!PyArg_ParseTupleAndKeywords (args, kwargs,
+                                    "|O!:cluttergtk.Actor.__init__",
+                                    kwlist,
+                                    &PyGtkWidget_Type, &py_contents))
+    return -1;
+
+  pygobject_construct (self, NULL);
+  bin = gtk_clutter_actor_get_widget (GTK_CLUTTER_ACTOR (self->obj));
+
+  if (py_contents)
+    gtk_container_add (GTK_CONTAINER (bin), GTK_WIDGET (py_contents->obj));
+
+  return 0;
+}
diff -urP ../BUILD/pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs pyclutter-gtk/clutter-gtk/cluttergtk-types.defs
--- ../BUILD/pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs	2010-01-25 08:30:27.000000000 -0500
+++ pyclutter-gtk/clutter-gtk/cluttergtk-types.defs	2010-09-29 11:42:19.753924000 -0400
@@ -2,6 +2,13 @@
 
 ; object definitions ...
 
+(define-object Actor
+  (in-module "GtkClutter")
+  (parent "ClutterActor")
+  (c-name "GtkClutterActor")
+  (gtype-id "GTK_CLUTTER_TYPE_ACTOR")
+)
+
 (define-object Embed
   (in-module "GtkClutter")
   (parent "GtkContainer")
@@ -16,6 +23,13 @@
   (vtable "GtkClutterScrollableIface")
 )
 
+(define-object Standin
+  (in-module "GtkClutter")
+  (parent "GtkWidget")
+  (c-name "GtkClutterStandin")
+  (gtype-id "GTK_CLUTTER_TYPE_STANDIN")
+)
+
 (define-object Viewport
   (in-module "GtkClutter")
   (parent "ClutterActor")
@@ -23,6 +37,13 @@
   (gtype-id "GTK_CLUTTER_TYPE_VIEWPORT")
 )
 
+(define-object Window
+  (in-module "GtkClutter")
+  (parent "GtkWindow")
+  (c-name "GtkClutterWindow")
+  (gtype-id "GTK_CLUTTER_TYPE_WINDOW")
+)
+
 (define-interface Zoomable
   (in-module "GtkClutter")
   (c-name "GtkClutterZoomable")
diff -urP ../BUILD/pyclutter-gtk-0.10.0/clutter-gtk/__init__.py pyclutter-gtk/clutter-gtk/__init__.py
--- ../BUILD/pyclutter-gtk-0.10.0/clutter-gtk/__init__.py	2010-01-25 10:39:00.000000000 -0500
+++ pyclutter-gtk/clutter-gtk/__init__.py	2010-09-29 11:42:19.753924000 -0400
@@ -35,4 +35,3 @@
     from cluttergtk import _cluttergtk
 
 from cluttergtk._cluttergtk import *
-__version__ = _cluttergtk.__version__
Only in ../BUILD/pyclutter-gtk-0.10.0/clutter-gtk/: Makefile.in
