diff -up pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs.no-scrollable pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs
--- pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs.no-scrollable	2010-09-29 11:59:57.406921372 -0400
+++ pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk.defs	2010-09-29 12:00:12.029923777 -0400
@@ -65,51 +65,6 @@
   )
 )
 
-;; From gtk-clutter-scrollable.h
-
-(define-function gtk_clutter_scrollable_get_type
-  (c-name "gtk_clutter_scrollable_get_type")
-  (return-type "GType")
-)
-
-(define-method set_adjustments
-  (of-object "GtkClutterScrollable")
-  (c-name "gtk_clutter_scrollable_set_adjustments")
-  (return-type "none")
-  (parameters
-    '("GtkAdjustment*" "h_adjust")
-    '("GtkAdjustment*" "v_adjust")
-  )
-)
-
-(define-virtual set_adjustments
-  (of-object "GtkClutterScrollable")
-  (return-type "none")
-  (parameters
-    '("GtkAdjustment*" "h_adjust")
-    '("GtkAdjustment*" "v_adjust")
-  )
-)
-
-(define-method get_adjustments
-  (of-object "GtkClutterScrollable")
-  (c-name "gtk_clutter_scrollable_get_adjustments")
-  (return-type "none")
-  (parameters
-    '("GtkAdjustment**" "h_adjust")
-    '("GtkAdjustment**" "v_adjust")
-  )
-)
-
-(define-virtual get_adjustments
-  (of-object "GtkClutterScrollable")
-  (return-type "none")
-  (parameters
-    '("GtkAdjustment**" "h_adjust")
-    '("GtkAdjustment**" "v_adjust")
-  )
-)
-
 ;; From gtk-clutter-standin.h
 
 (define-function standin_get_type
diff -up pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs.no-scrollable pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs
--- pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs.no-scrollable	2010-09-29 12:00:26.478924172 -0400
+++ pyclutter-gtk-0.10.0/clutter-gtk/cluttergtk-types.defs	2010-09-29 12:00:38.008924005 -0400
@@ -16,13 +16,6 @@
   (gtype-id "GTK_CLUTTER_TYPE_EMBED")
 )
 
-(define-interface Scrollable
-  (in-module "GtkClutter")
-  (c-name "GtkClutterScrollable")
-  (gtype-id "GTK_CLUTTER_TYPE_SCROLLABLE")
-  (vtable "GtkClutterScrollableIface")
-)
-
 (define-object Standin
   (in-module "GtkClutter")
   (parent "GtkWidget")
