From 3bbcb9a3a908d1aceb9f69118c1ed17a8d6d1871 Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 28 Feb 2009 13:32:45 +0100
Subject: [PATCH] build some files without stack-protector

---
 Makefile |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

Index: dietlibc-0.32/Makefile
===================================================================
--- dietlibc-0.32.orig/Makefile
+++ dietlibc-0.32/Makefile
@@ -143,6 +143,8 @@ LIBPTHREAD_OBJS=$(patsubst libpthread/%.
 
 LIBGMON_OBJS=$(OBJDIR)/__mcount.o $(OBJDIR)/monitor.o $(OBJDIR)/profil.o
 
+NO_STACK_PROTECTOR=stackgap.o __get_elf_aux_value.o
+
 include $(ARCH)/Makefile.add
 
 LIBMATHOBJ=$(patsubst %,$(OBJDIR)/%,$(LIBMATH))
@@ -189,6 +191,8 @@ $(OBJDIR)/stack_smash_handler2.o:	XCFLAG
 
 $(OBJDIR)/%.o: %.c
 	$(CROSS)$(CC) $(INC) $(CFLAGS) $(XCFLAGS) -c $< -o $@ -D__dietlibc__
+
+$(addprefix $(OBJDIR)/,$(NO_STACK_PROTECTOR)):	XCFLAGS+=-fno-stack-protector
 endif
 
 ifeq ($(shell $(CC) -v 2>&1 | grep "gcc version"),gcc version 4.0.0)
