From b993cb8bebea6870771bb66eae8b6d41f9e6e9e6 Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 25 Jul 2009 15:34:44 +0200
Subject: [PATCH 17/18] minor cleanups in stackgap.c

---
 lib/stackgap.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/stackgap.c b/lib/stackgap.c
index 5d51c5d..f30b58a 100644
--- a/lib/stackgap.c
+++ b/lib/stackgap.c
@@ -26,9 +26,9 @@ void* __tdataptr;
 
 static void findtlsdata(long* auxvec) {
 #if (__WORDSIZE == 64)
-  Elf64_Phdr* x=0;
+  Elf64_Phdr const * x=0;
 #else
-  Elf32_Phdr* x=0;
+  Elf32_Phdr const * x=0;
 #endif
   size_t i,n;
 #ifndef WANT_ELFINFO
@@ -125,9 +125,9 @@ int stackgap(int argc,char* argv[],char* envp[]) {
 #endif
 #ifdef WANT_STACKGAP
   unsigned short s;
+  volatile char* gap;
 #endif
 #if defined(WANT_STACKGAP) || defined(WANT_SSP)
-  volatile char* gap;
 #ifndef WANT_ELFINFO
   rand=find_rand(auxvec);
 #else
-- 
1.6.2.5

