Index: cheese-2.32.0/tests/Makefile.am
===================================================================
--- cheese-2.32.0.orig/tests/Makefile.am
+++ cheese-2.32.0/tests/Makefile.am
@@ -25,16 +25,16 @@ cheese_test_monitor_SOURCES = cheese-tes
 test_webcam_button_SOURCES = test-webcam-button.c
 
 cheese_test_widget_LDADD = $(CHEESE_LIBS) \
-	$(top_builddir)/libcheese/libcheesecommon.la \
 	$(top_builddir)/libcheese/libcheese-gtk.la \
+	$(top_builddir)/libcheese/libcheesecommon.la \
 	$(NULL)
 cheese_test_monitor_LDADD = $(CHEESE_LIBS) \
-	$(top_builddir)/libcheese/libcheesecommon.la \
 	$(top_builddir)/libcheese/libcheese-gtk.la \
+	$(top_builddir)/libcheese/libcheesecommon.la \
 	$(NULL)
 test_webcam_button_LDADD = $(XTEST_LIBS)
 cheese_test_chooser_SOURCES = cheese-test-chooser.c
 cheese_test_chooser_LDADD = $(CHEESE_LIBS) \
-	$(top_builddir)/libcheese/libcheesecommon.la \
 	$(top_builddir)/libcheese/libcheese-gtk.la \
+	$(top_builddir)/libcheese/libcheesecommon.la \
 	$(NULL)
