diff -up hplip-3.9.8/ui4/setupdialog.py.discovery-method hplip-3.9.8/ui4/setupdialog.py
--- hplip-3.9.8/ui4/setupdialog.py.discovery-method	2009-08-04 22:37:22.000000000 +0100
+++ hplip-3.9.8/ui4/setupdialog.py	2009-09-16 15:34:36.509431266 +0100
@@ -171,6 +171,7 @@ class SetupDialog(QDialog, Ui_Dialog):
             QTimer.singleShot(0, self.showRemovePage)
         else:
             if self.skip_discovery:
+                self.discovery_method = 1 # mDNS
                 QTimer.singleShot(0, self.showDevicesPage)
             else:
                 QTimer.singleShot(0, self.showDiscoveryPage)
