diff -up hplip-3.10.9/toolbox.py.dbus-threads hplip-3.10.9/toolbox.py
--- hplip-3.10.9/toolbox.py.dbus-threads	2010-12-15 10:54:17.178739842 +0000
+++ hplip-3.10.9/toolbox.py	2010-12-15 10:57:09.923712662 +0000
@@ -100,8 +100,11 @@ if ui_toolkit == 'qt3':
     try:
         from dbus import SessionBus
         import dbus.service
-        from dbus.mainloop.glib import DBusGMainLoop
+        from dbus.mainloop.glib import DBusGMainLoop, threads_init
         from gobject import MainLoop
+        import glib
+        glib.threads_init();
+        dbus.mainloop.glib.threads_init()
     except ImportError:
         log.error("Unable to load dbus - Automatic status updates in HPLIP Device Manager will be disabled.")
         disable_dbus = True
