diff -up hplip-3.11.1/ui/setupform.py.fax-ppd hplip-3.11.1/ui/setupform.py
--- hplip-3.11.1/ui/setupform.py.fax-ppd	2011-01-19 06:18:26.000000000 +0100
+++ hplip-3.11.1/ui/setupform.py	2011-01-24 12:00:48.000000000 +0100
@@ -886,16 +886,16 @@ class SetupForm(SetupForm_base):
         QApplication.setOverrideCursor(QApplication.waitCursor)
 
         if self.mq.get('fax-type', FAX_TYPE_NONE) == FAX_TYPE_MARVELL:
-            fax_ppd_name = "HP-Fax3-hplip" # Fixed width (2528 pixels) and 300dpi rendering
+            fax_ppd_name = "HP-Fax3-hpcups" # Fixed width (2528 pixels) and 300dpi rendering
             nick = "HP Fax 3"
         if self.mq.get('fax-type', FAX_TYPE_NONE) == FAX_TYPE_SOAP or self.mq.get('fax-type', FAX_TYPE_NONE) == FAX_TYPE_LEDMSOAP:
-            fax_ppd_name = "HP-Fax2-hplip" # Fixed width (2528 pixels) and 300dpi rendering
+            fax_ppd_name = "HP-Fax2-hpcups" # Fixed width (2528 pixels) and 300dpi rendering
             nick = "HP Fax 2"
         if self.mq.get('fax-type', FAX_TYPE_NONE) == FAX_TYPE_LEDM:
-            fax_ppd_name = "HP-Fax4-hplip" # Fixed width (1728 pixels) and 200dpi rendering
+            fax_ppd_name = "HP-Fax4-hpcups" # Fixed width (1728 pixels) and 200dpi rendering
             nick = "HP Fax 4"
         else:
-            fax_ppd_name = "HP-Fax-hplip" # Standard
+            fax_ppd_name = "HP-Fax-hpcups" # Standard
             nick = "HP Fax"
 
         ppds = []
