diff -up hplip-3.9.6b/base/utils.py.ui-optional hplip-3.9.6b/base/utils.py
--- hplip-3.9.6b/base/utils.py.ui-optional	2009-06-25 20:05:53.000000000 +0100
+++ hplip-3.9.6b/base/utils.py	2009-07-28 10:10:57.984991878 +0100
@@ -597,6 +597,13 @@ def checkPyQtImport(): # qt3
         if os.getenv('DISPLAY') and os.getenv('STARTED_FROM_MENU'):
             no_qt_message_gtk()
 
+        # hplip-gui sub-package (Fedora)
+        try:
+            import ui
+        except ImportError:
+            log.error("hplip-gui not installed.  GUI not available. Exiting.")
+            return False
+
         log.error("PyQt not installed. GUI not available. Exiting.")
         return False
 
@@ -642,7 +649,10 @@ def checkPyQtImport(): # qt3
 def checkPyQtImport4():
     try:
         import PyQt4
+        import ui4
     except ImportError:
+        # hplip-gui sub-package (Fedora) requires PyQt4
+        log.error("Install the hplip-gui package for graphical support.")
         return False
     else:
         return True
