diff -up hplip-3.11.5/prnt/hpcups/HPCupsFilter.cpp.newline hplip-3.11.5/prnt/hpcups/HPCupsFilter.cpp
--- hplip-3.11.5/prnt/hpcups/HPCupsFilter.cpp.newline	2011-05-12 16:10:02.000000000 +0200
+++ hplip-3.11.5/prnt/hpcups/HPCupsFilter.cpp	2011-05-12 16:13:06.000000000 +0200
@@ -618,7 +618,7 @@ int HPCupsFilter::processRasterData(cups
             dbglog("10......\n");
         }
 
-        fprintf(stderr, "PAGE: %d %s", current_page_number, m_argv[4]);
+        fprintf(stderr, "PAGE: %d %s\n", current_page_number, m_argv[4]);
         // Iterating through the raster per page
         for (int y = 0; y < (int) cups_header.cupsHeight; y++) {
             cupsRasterReadPixels (cups_raster, m_pPrinterBuffer, cups_header.cupsBytesPerLine);
