diff -up hplip-3.10.6/prnt/drv/hpcups.drv.in.snmp-quirks hplip-3.10.6/prnt/drv/hpcups.drv.in
--- hplip-3.10.6/prnt/drv/hpcups.drv.in.snmp-quirks	2010-08-20 18:03:46.026041759 +0100
+++ hplip-3.10.6/prnt/drv/hpcups.drv.in	2010-08-20 18:04:26.210041811 +0100
@@ -5318,6 +5318,8 @@ Manufacturer "HP"
       Attribute "NickName" "" "HP Officejet Pro 8500 a909g, $Version"
       Attribute "ShortNickName" "" "HP Officejet Pro 8500 a909g"
       Attribute "1284DeviceID" "" "MFG:HP;MDL:officejet pro 8500 a909g;DES:officejet pro 8500 a909g;"
+      // This device lies about its supplies capacity.
+      Attribute "cupsSNMPQuirks" "" "capacity"
       PCFileName "hp-officejet_pro_8500_a909g.ppd"
       Attribute "Product" "" "(HP Officejet Pro 8500 Wireless All-in-one Printer - a909g)"
     }
@@ -5326,6 +5328,8 @@ Manufacturer "HP"
       Attribute "NickName" "" "HP Officejet Pro 8500 a909a, $Version"
       Attribute "ShortNickName" "" "HP Officejet Pro 8500 a909a"
       Attribute "1284DeviceID" "" "MFG:HP;MDL:officejet pro 8500 a909a;DES:officejet pro 8500 a909a;"
+      // This device lies about its supplies capacity.
+      Attribute "cupsSNMPQuirks" "" "capacity"
       PCFileName "hp-officejet_pro_8500_a909a.ppd"
       Attribute "Product" "" "(HP Officejet Pro 8500 All-in-one Printer - a909a)"
     }
