diff -up hplip-3.10.2/prnt/hpcups/HPCupsFilter.cpp.mucks-with-spooldir hplip-3.10.2/prnt/hpcups/HPCupsFilter.cpp
--- hplip-3.10.2/prnt/hpcups/HPCupsFilter.cpp.mucks-with-spooldir	2010-02-26 12:07:47.170265651 +0000
+++ hplip-3.10.2/prnt/hpcups/HPCupsFilter.cpp	2010-02-26 12:09:13.647265807 +0000
@@ -443,26 +443,6 @@ int HPCupsFilter::StartPrintJob(int  arg
 
     getLogLevel();
     m_JA.job_id = atoi(argv[1]);
-    FILE    *fp;
-    char    dFileName[32];
-    memset(dFileName, 0, sizeof(dFileName));
-    m_JA.job_id = atoi(argv[1]);
-    snprintf (dFileName, sizeof(dFileName), "/var/spool/cups/d%05d-001", m_JA.job_id);
-    if ((fp = fopen (dFileName, "r")))
-    {
-        char    line[258];
-        for (int i = 0; i < 10; i++)
-        {
-            fgets (line, 256, fp);
-            if (!strncmp (line, "%%Pages:", 8))
-            {
-                sscanf (line+9, "%d", &m_JA.total_pages);
-                break;
-            }
-        }
-        fclose (fp);
-    }
-
     m_ppd = ppdOpenFile(getenv("PPD"));
     if (m_ppd == NULL) {
         dbglog("DEBUG: ppdOpenFile failed for %s\n", getenv("PPD"));
