diff -up hplip-3.10.9/base/device.py.raw_deviceID-traceback hplip-3.10.9/base/device.py
--- hplip-3.10.9/base/device.py.raw_deviceID-traceback	2010-10-18 17:22:55.138868192 +0100
+++ hplip-3.10.9/base/device.py	2010-10-18 17:24:02.965869220 +0100
@@ -1368,6 +1368,8 @@ class Device(object):
 
     def getDeviceID(self):
         needs_close = False
+        self.raw_deviceID = ''
+        self.deviceID = {}
         if self.io_state != IO_STATE_HP_OPEN:
            try:
                self.open()
@@ -1377,10 +1379,7 @@ class Device(object):
 
         result_code, data = hpmudext.get_device_id(self.device_id)
 
-        if result_code != hpmudext.HPMUD_R_OK:
-            self.raw_deviceID = ''
-            self.deviceID = {}
-        else:
+        if result_code == hpmudext.HPMUD_R_OK:
             self.raw_deviceID = data
             self.deviceID = parseDeviceID(data)
 
