diff -up hplip-3.10.9/ui4/systemtray.py.notification-exception hplip-3.10.9/ui4/systemtray.py
--- hplip-3.10.9/ui4/systemtray.py.notification-exception	2011-01-05 10:45:57.000000000 +0100
+++ hplip-3.10.9/ui4/systemtray.py	2011-01-05 12:34:18.000000000 +0100
@@ -27,6 +27,7 @@ import os
 import signal
 import os.path
 import time
+import gobject
 
 
 # Local
@@ -662,7 +663,10 @@ class SystemTrayApp(QApplication):
                                     else:
                                         n.set_timeout(TRAY_MESSAGE_DELAY)
 
-                                    n.show()
+                                    try:
+                                        n.show()
+                                    except gobject.GError:
+                                        log.error("Failed to show notification!")
 
                                 else: # Use "standard" message bubbles
                                     icon = ERROR_STATE_TO_ICON.get(error_state, QSystemTrayIcon.Information)
