diff -up hplip-3.10.9/ui4/sendfaxdialog.py~ hplip-3.10.9/ui4/sendfaxdialog.py
--- hplip-3.10.9/ui4/sendfaxdialog.py~	2010-09-26 18:07:19.000000000 +0100
+++ hplip-3.10.9/ui4/sendfaxdialog.py	2010-12-13 16:44:56.857648866 +0000
@@ -915,8 +915,12 @@ class SendFaxDialog(QDialog, Ui_Dialog):
     def CheckTimer_timeout(self):
         if not self.busy:
             #log.debug("Checking for incoming faxes...")
-            device_uri, printer_name, event_code, username, job_id, title, timedate, fax_file = \
-                self.service.CheckForWaitingFax(self.device_uri, prop.username, self.last_job_id)
+            try:
+                device_uri, printer_name, event_code, username, job_id, title, timedate, fax_file = \
+                    self.service.CheckForWaitingFax(self.device_uri, prop.username, self.last_job_id)
+            except Exception, e:
+                log.debug("Exception caught: %s" % e)
+                fax_file = None
 
             if fax_file:
                 self.last_job_id = 0
