diff -up hplip-3.11.1/ui4/wifisetupdialog.py.wifisetup hplip-3.11.1/ui4/wifisetupdialog.py
--- hplip-3.11.1/ui4/wifisetupdialog.py.wifisetup	2011-01-19 06:18:20.000000000 +0100
+++ hplip-3.11.1/ui4/wifisetupdialog.py	2011-03-01 15:39:02.000000000 +0100
@@ -301,7 +301,7 @@ class WifiSetupDialog(QDialog, Ui_Dialog
             self.dev.close()
             endWaitCursor()
 
-        self.num_networks = self.networks['numberofscanentries']
+        self.num_networks = self.networks.get('numberofscanentries')
         self.clearNetworksTable()
 
         if self.num_networks:
