diff -up hplip-3.10.9/io/hpmud/hpmud.c.hpaio-segfault hplip-3.10.9/io/hpmud/hpmud.c
--- hplip-3.10.9/io/hpmud/hpmud.c.hpaio-segfault	2010-12-13 16:53:16.470306314 +0000
+++ hplip-3.10.9/io/hpmud/hpmud.c	2010-12-13 16:53:17.423332321 +0000
@@ -305,6 +305,7 @@ static int new_device(const char *uri, e
    {
       BUG("invalid uri %s\n", uri);
       *result = HPMUD_R_INVALID_URI;
+      index = 0;
       goto bugout;
    }
    msp->device[i].io_mode = mode;
