diff -up hplip-3.10.2/prnt/hpcups/LJColor.h.skip-blank-lines hplip-3.10.2/prnt/hpcups/LJColor.h
--- hplip-3.10.2/prnt/hpcups/LJColor.h.skip-blank-lines	2010-02-24 23:04:09.000000000 +0000
+++ hplip-3.10.2/prnt/hpcups/LJColor.h	2010-04-15 21:46:48.375754707 +0100
@@ -48,6 +48,7 @@ public:
     DRIVER_ERROR    Configure(Pipeline **pipeline);
     DRIVER_ERROR    EndJob();
     DRIVER_ERROR    FormFeed();
+    bool            CanSkipRasters() {return false;}
 protected:
     virtual DRIVER_ERROR addJobSettings();
     void    configureRasterData();
