diff -up hplip-3.11.1/prnt/cupsext/cupsext.c.addprinter hplip-3.11.1/prnt/cupsext/cupsext.c
--- hplip-3.11.1/prnt/cupsext/cupsext.c.addprinter	2011-01-19 06:15:27.000000000 +0100
+++ hplip-3.11.1/prnt/cupsext/cupsext.c	2011-01-24 12:24:01.000000000 +0100
@@ -510,6 +510,7 @@ PyObject * addPrinter( PyObject * self, 
         strcpy( info, name );
 
     /* Connect to the HTTP server */
+    cupsSetUser ("root");
     if ( ( http = httpConnectEncrypt( cupsServer(), ippPort(), cupsEncryption() ) ) == NULL )
     {
         r = 0;
