diff -up hplip-3.11.10/prnt/hpcups/HPCupsFilter.cpp.top-margins-not-respected hplip-3.11.10/prnt/hpcups/HPCupsFilter.cpp
--- hplip-3.11.10/prnt/hpcups/HPCupsFilter.cpp.top-margins-not-respected	2011-10-02 05:03:50.000000000 +0200
+++ hplip-3.11.10/prnt/hpcups/HPCupsFilter.cpp	2011-11-14 16:49:39.073220429 +0100
@@ -544,18 +544,10 @@ bool HPCupsFilter::isBlankRaster(BYTE *i
         return true;
     }
 
-    if(header->cupsColorSpace == CUPS_CSPACE_K){
-	if (*input_raster == 0x00 &&
-            !(memcmp(input_raster + 1, input_raster, length_in_bytes - 1))) {
-        return true;
-        }
-    }
-    else{
         if (*input_raster == 0xFF &&
               !(memcmp(input_raster + 1, input_raster, length_in_bytes - 1))) {
         return true;
         }
-    }
     return false;
 }
 
