diff -up hplip-3.10.5/ui4/systemtray.py.no-system-tray hplip-3.10.5/ui4/systemtray.py
--- hplip-3.10.5/ui4/systemtray.py.no-system-tray	2010-05-13 01:45:04.000000000 +0200
+++ hplip-3.10.5/ui4/systemtray.py	2010-05-17 15:03:12.000000000 +0200
@@ -74,6 +74,7 @@ HIDE_INACTIVE_DELAY = 5000
 BLIP_DELAY = 2000
 SET_MENU_DELAY = 1000
 MAX_MENU_EVENTS = 10
+TRAY_AVAILABLE_TIMEOUT = 180
 
 ERROR_STATE_TO_ICON = {
     ERROR_STATE_CLEAR:        QSystemTrayIcon.Information,
@@ -706,7 +707,7 @@ def run(read_pipe):
     app.setQuitOnLastWindowClosed(False) # If not set, settings dlg closes app
 
     i = 0
-    while i < 10:
+    while i < TRAY_AVAILABLE_TIMEOUT:
         if QSystemTrayIcon.isSystemTrayAvailable():
             break
         time.sleep(1.0)
