diff -up hplip-3.10.9/ui4/printsettingstoolbox.py.UnicodeDecodeError hplip-3.10.9/ui4/printsettingstoolbox.py
--- hplip-3.10.9/ui4/printsettingstoolbox.py.UnicodeDecodeError	2010-09-26 19:07:19.000000000 +0200
+++ hplip-3.10.9/ui4/printsettingstoolbox.py	2010-10-22 15:01:13.000000000 +0200
@@ -459,7 +459,7 @@ class PrintSettingsToolbox(QToolBox):
 
                     self.beginControlGroup(g, QString(text))
 
-                    log.debug("  Text: %s" % unicode(text))
+                    log.debug("  Text: %s" % repr(text))
                     log.debug("Num subgroups: %d" % num_subgroups)
 
                     options = cups.getOptionList(g)
