diff -up hplip-3.11.7/data/rules/56-hpmud_support.rules.udev-rules hplip-3.11.7/data/rules/56-hpmud_support.rules
--- hplip-3.11.7/data/rules/56-hpmud_support.rules.udev-rules	2011-07-24 21:04:13.000000000 +0200
+++ hplip-3.11.7/data/rules/56-hpmud_support.rules	2011-07-25 13:52:09.379058787 +0200
@@ -6,10 +6,11 @@ SUBSYSTEM!="usb_device", GOTO="hpmud_rul
 
 LABEL="pid_test"
 
+ENV{ID_USB_INTERFACES}=="", IMPORT{program}="usb_id --export %p"
 # Check for LaserJet products (0x03f0xx17).
-SYSFS{idVendor}=="03f0", SYSFS{idProduct}=="??17", ENV{hp_model}="$sysfs{product}", ENV{hp_test}="yes"
+ATTRS{idVendor}=="03f0", ATTRS{idProduct}=="??17", ENV{hp_model}="$attr{product}", ENV{hp_test}="yes"
 # Check for LaserJet products (0x03f0xx2a).
-SYSFS{idVendor}=="03f0", SYSFS{idProduct}=="??2a", ENV{hp_model}="$sysfs{product}", ENV{hp_test}="yes"
+ATTRS{idVendor}=="03f0", ATTRS{idProduct}=="??2a", ENV{hp_model}="$attr{product}", ENV{hp_test}="yes"
 ENV{hp_test}=="yes", RUN+="/bin/sh -c '/usr/bin/hp-mkuri -c &'"
 
 LABEL="hpmud_rules_end"
diff -up hplip-3.11.10/Makefile.in.udev-rules hplip-3.11.10/Makefile.in
--- hplip-3.11.10/Makefile.in.udev-rules	2011-10-02 05:11:49.000000000 +0200
+++ hplip-3.11.10/Makefile.in	2011-10-24 14:23:17.336002785 +0200
@@ -4093,7 +4093,7 @@ cups_drv = prnt/drv/hpcups.drv
 @HPLIP_BUILD_TRUE@hp_mkuri_CFLAGS = -DCONFDIR=\"$(hplip_confdir)\"
 
 # hpmud.rules
-@HPLIP_BUILD_TRUE@rulesdir = /etc/udev/rules.d
+@HPLIP_BUILD_TRUE@rulesdir = /lib/udev/rules.d
 @HPLIP_BUILD_TRUE@dist_rules_DATA = data/rules/56-hpmud_support.rules \
 @HPLIP_BUILD_TRUE@	$(am__append_2) $(am__append_3)
 @HPLIP_BUILD_TRUE@halpredir = /usr/share/hal/fdi/preprobe/10osvendor
