diff -up hplip-3.10.2/prnt/hpcups/HPCupsFilter.cpp.clear-old-state-reasons hplip-3.10.2/prnt/hpcups/HPCupsFilter.cpp
--- hplip-3.10.2/prnt/hpcups/HPCupsFilter.cpp.clear-old-state-reasons	2010-04-28 12:23:08.296730566 +0100
+++ hplip-3.10.2/prnt/hpcups/HPCupsFilter.cpp	2010-04-28 12:25:25.493730293 +0100
@@ -479,6 +479,14 @@ int HPCupsFilter::StartPrintJob(int  arg
 
     signal(SIGTERM, HPCancelJob);
 
+/*
+ *  Prior to the re-write of hpcups, this filter managed the
+ *  marker-supply-low-warning printer state reason.  Make sure to
+ *  clear that state reason so that upgrades work correctly.
+ */
+
+    fputs ("STATE: -marker-supply-low-warning\n", stderr);
+
     cups_raster = cupsRasterOpen(fd, CUPS_RASTER_READ);
 
     if (cups_raster == NULL) {
