diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_2500-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_2500-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_2500-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:37.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_2500-ps.ppd	2011-11-21 15:56:33.120335351 +0000
@@ -44,6 +44,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 2500;"
 *Manufacturer:	"HP"
 *Product: "(hp color LaserJet 2500)"
 *Product: "(Hewlett-Packard hp color LaserJet 2500)"
diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_2500_series-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_2500_series-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_2500_series-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:37.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_2500_series-ps.ppd	2011-11-21 15:56:33.123335298 +0000
@@ -44,6 +44,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 2500;"
 *Manufacturer:	"HP"
 *Product: "(hp color LaserJet 2500)"
 *Product: "(Hewlett-Packard hp color LaserJet 2500)"
diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_2550_series-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_2550_series-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_2550_series-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:37.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_2550_series-ps.ppd	2011-11-21 15:56:33.128335209 +0000
@@ -44,6 +44,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 2550 series;"
 *Manufacturer:	"HP"
 *Product: "(hp color LaserJet 2550 series)"
 *Product: "(Hewlett-Packard hp color LaserJet 2550 series)"
diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_2605dn-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_2605dn-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_2605dn-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:37.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_2605dn-ps.ppd	2011-11-21 15:56:33.014337244 +0000
@@ -50,6 +50,7 @@
 
 *% Product Name and Version Information
 *Manufacturer:	"HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Color LaserJet 2605dn;DES:Hewlett-Packard Color LaserJet 2605dn;"
 *Product: "(HP Color LaserJet 2605)"
 *Product: "(Hewlett-Packard HP Color LaserJet 2605)"
 *Product: "(HP Color LaserJet 2605dn)"
diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_2840-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_2840-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_2840-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:37.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_2840-ps.ppd	2011-11-21 15:56:33.019337154 +0000
@@ -46,6 +46,7 @@
 
 *% Product Name and Version Information
 *Manufacturer:	"HP"
+*1284DeviceID:  "MFG:Hewlett-Packard;CMD:PJL,PML,MLC,PCL,POSTSCRIPT,PCLXL;MDL:HP Color LaserJet 2840;DES:Hewlett-Packard Color LaserJet 2840;"
 *Product: "(HP Color LaserJet 2820)"
 *Product: "(Hewlett-Packard HP Color LaserJet 2820)"
 *Product: "(HP Color LaserJet 2830)"
diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_3700n-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_3700n-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_3700n-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:37.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_3700n-ps.ppd	2011-11-21 15:56:33.134335101 +0000
@@ -43,6 +43,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 3700;"
 *Manufacturer:	"HP"
 *Product: 	"(hp color LaserJet 3700)"
 *Product: "(Hewlett-Packard hp color LaserJet 3700)"
diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_3700-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_3700-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_3700-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:37.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_3700-ps.ppd	2011-11-21 15:56:33.140334995 +0000
@@ -43,6 +43,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 3700;"
 *Manufacturer:	"HP"
 *Product: 	"(hp color LaserJet 3700)"
 *Product: "(Hewlett-Packard hp color LaserJet 3700)"
diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_3800-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_3800-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_3800-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:37.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_3800-ps.ppd	2011-11-21 15:56:33.027337012 +0000
@@ -51,6 +51,7 @@
 
 *% Product Name and Version Information
 *Manufacturer:	"HP"
+*1284DeviceID:  "MFG:Hewlett-Packard;MDL:HP Color LaserJet 3800;CMD:PJL,MLC,BIDI-ECP,PCLXL,PCL,PDF,PJL,POSTSCRIPT;"
 *Product: 	"(HP Color LaserJet 3800)"
 *Product: "(Hewlett-Packard HP Color LaserJet 3800)"
 *ModelName:	"HP Color LaserJet 3800"
diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_4550-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_4550-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_4550-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:37.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_4550-ps.ppd	2011-11-21 15:56:33.145334905 +0000
@@ -43,6 +43,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Color LaserJet 4550 ;"
 *Manufacturer:	"HP"
 *Product: 	"(HP Color LaserJet 4550 )"
 *Product: "(Hewlett-Packard HP Color LaserJet 4550 )"
diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_4600-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_4600-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_4600-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:37.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_4600-ps.ppd	2011-11-21 15:56:33.151334799 +0000
@@ -44,6 +44,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 4600;"
 *Manufacturer:	"HP"
 *Product: 	"(hp color LaserJet 4600)"
 *Product: "(Hewlett-Packard hp color LaserJet 4600)"
diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_4600_series-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_4600_series-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_4600_series-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:37.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_4600_series-ps.ppd	2011-11-21 15:56:33.156334709 +0000
@@ -44,6 +44,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 4600;"
 *Manufacturer:	"HP"
 *Product: 	"(hp color LaserJet 4600)"
 *Product: "(Hewlett-Packard hp color LaserJet 4600)"
diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_4650-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_4650-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_4650-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:37.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_4650-ps.ppd	2011-11-21 15:56:33.164334567 +0000
@@ -47,6 +47,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 4650;"
 *Manufacturer:	"HP"
 *Product: 	"(hp color LaserJet 4650)"
 *Product: "(Hewlett-Packard hp color LaserJet 4650)"
diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_4700-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_4700-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_4700-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:37.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_4700-ps.ppd	2011-11-21 15:56:33.172334423 +0000
@@ -49,6 +49,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Color LaserJet 4700;"
 *Manufacturer:	"HP"
 *Product: 	"(HP Color LaserJet 4700)"
 *Product: "(Hewlett-Packard HP Color LaserJet 4700)"
diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_5550-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_5550-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_5550-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:37.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_5550-ps.ppd	2011-11-21 15:56:33.183334227 +0000
@@ -46,6 +46,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 5550 ;"
 *Manufacturer:	"HP"
 *Product: "(hp color LaserJet 5550 )"
 *Product: "(Hewlett-Packard hp color LaserJet 5550 )"
diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_cm3530_mfp-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_cm3530_mfp-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_cm3530_mfp-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:38.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_cm3530_mfp-ps.ppd	2011-11-21 15:56:33.346331319 +0000
@@ -51,6 +51,7 @@
 *%== Mac OS X specific end ==
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Color LaserJet CM3530 MFP;"
 *Manufacturer:	"HP"
 *Product: 	"(HP Color LaserJet CM3530 MFP)"
 *Product: "(Hewlett-Packard HP Color LaserJet CM3530 MFP)"
diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_cm4730_mfp-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_cm4730_mfp-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_cm4730_mfp-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:38.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_cm4730_mfp-ps.ppd	2011-11-21 15:56:33.194334031 +0000
@@ -49,6 +49,7 @@
 *%== Mac OS X specific end ==
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Color LaserJet CM4730 MFP;"
 *Manufacturer:	"HP"
 *Product: 	"(HP Color LaserJet CM4730 MFP)"
 *Product: "(Hewlett-Packard HP Color LaserJet CM4730 MFP)"
diff -up hplip-3.11.10/prnt/ps/hp-color_laserjet_cp3525-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-color_laserjet_cp3525-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-color_laserjet_cp3525-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:38.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-color_laserjet_cp3525-ps.ppd	2011-11-21 15:56:33.202333888 +0000
@@ -51,6 +51,7 @@
 *%== Mac OS X specific end ==
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Color LaserJet CP3525;"
 *Manufacturer:	"HP"
 *Product: 	"(HP Color LaserJet CP3525)"
 *Product: "(Hewlett-Packard HP Color LaserJet CP3525)"
diff -up hplip-3.11.10/prnt/ps/hp-designjet_t770_postscript-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-designjet_t770_postscript-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-designjet_t770_postscript-ps.ppd.deviceIDs-ppd	2011-11-21 15:58:21.095199630 +0000
+++ hplip-3.11.10/prnt/ps/hp-designjet_t770_postscript-ps.ppd	2011-11-21 16:01:46.968516936 +0000
@@ -38,6 +38,7 @@
 *ShortNickName: "HP Designjet T770 PostScript"
 *NickName: "HP Designjet T770ps 44in"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Designjet T770;DES:HP Designjet T770;CMD:POSTSCRIPT;"
 
 
 *% =============== Device Capabilities And Keywords for supporting Alert and Status Mechanism =============== 
diff -up hplip-3.11.10/prnt/ps/hp-designjet_t770ps_24in-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-designjet_t770ps_24in-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-designjet_t770ps_24in-ps.ppd.deviceIDs-ppd	2011-11-21 15:58:27.126091749 +0000
+++ hplip-3.11.10/prnt/ps/hp-designjet_t770ps_24in-ps.ppd	2011-11-21 16:02:01.227261873 +0000
@@ -38,6 +38,7 @@
 *ShortNickName: "HP Designjet T770ps 24in"
 *NickName: "HP Designjet T770ps 24in"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Designjet T770;DES:HP Designjet T770;CMD:POSTSCRIPT;"
 
 
 *% =============== Device Capabilities And Keywords for supporting Alert and Status Mechanism =============== 
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_1200n-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_1200n-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_1200n-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_1200n-ps.ppd	2011-11-21 15:56:33.032336922 +0000
@@ -56,6 +56,7 @@
 *Product: "(HP LaserJet 1200)"
 *Product: "(Hewlett-Packard HP LaserJet 1200)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 1200;DES:Hewlett-Packard LaserJet 1200;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_1200-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_1200-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_1200-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_1200-ps.ppd	2011-11-21 15:56:33.036336850 +0000
@@ -56,6 +56,7 @@
 *Product: "(HP LaserJet 1200)"
 *Product: "(Hewlett-Packard HP LaserJet 1200)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 1200;DES:Hewlett-Packard LaserJet 1200;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_1300n-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_1300n-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_1300n-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_1300n-ps.ppd	2011-11-21 15:56:33.039336798 +0000
@@ -60,6 +60,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1300n)"
 *Product: "(HP LaserJet 1300n)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1300;DES:Hewlett-Packard LaserJet 1300;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_1300-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_1300-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_1300-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_1300-ps.ppd	2011-11-21 15:56:33.043336726 +0000
@@ -60,6 +60,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1300n)"
 *Product: "(HP LaserJet 1300n)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1300;DES:Hewlett-Packard LaserJet 1300;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_1300xi-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_1300xi-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_1300xi-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_1300xi-ps.ppd	2011-11-21 15:56:33.047336654 +0000
@@ -60,6 +60,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1300n)"
 *Product: "(HP LaserJet 1300n)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1300;DES:Hewlett-Packard LaserJet 1300;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_1320n-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_1320n-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_1320n-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_1320n-ps.ppd	2011-11-21 15:56:33.051336583 +0000
@@ -57,6 +57,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1320 series)"
 *Product: "(HP LaserJet 1320 series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1320 series;DES:Hewlett-Packard LaserJet 1320 series;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_1320nw-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_1320nw-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_1320nw-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_1320nw-ps.ppd	2011-11-21 15:56:33.056336493 +0000
@@ -57,6 +57,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1320 series)"
 *Product: "(HP LaserJet 1320 series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1320 series;DES:Hewlett-Packard LaserJet 1320 series;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_1320-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_1320-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_1320-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_1320-ps.ppd	2011-11-21 15:56:33.061336404 +0000
@@ -57,6 +57,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1320 series)"
 *Product: "(HP LaserJet 1320 series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1320 series;DES:Hewlett-Packard LaserJet 1320 series;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_1320_series-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_1320_series-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_1320_series-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_1320_series-ps.ppd	2011-11-21 15:56:33.066336315 +0000
@@ -57,6 +57,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1320 series)"
 *Product: "(HP LaserJet 1320 series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1320 series;DES:Hewlett-Packard LaserJet 1320 series;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_1320tn-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_1320tn-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_1320tn-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_1320tn-ps.ppd	2011-11-21 15:56:33.081336049 +0000
@@ -57,6 +57,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1320 series)"
 *Product: "(HP LaserJet 1320 series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1320 series;DES:Hewlett-Packard LaserJet 1320 series;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_2100-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_2100-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_2100-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_2100-ps.ppd	2011-11-21 15:56:33.212333710 +0000
@@ -56,6 +56,7 @@
 *Product: "(HP LaserJet 2100 Series)"
 *Product: "(Hewlett-Packard HP LaserJet 2100 Series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 2100 Series;"
 *PSVersion: "(2014.111) 1"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_2100_series-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_2100_series-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_2100_series-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_2100_series-ps.ppd	2011-11-21 15:56:33.215333656 +0000
@@ -56,6 +56,7 @@
 *Product: "(HP LaserJet 2100 Series)"
 *Product: "(Hewlett-Packard HP LaserJet 2100 Series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 2100 Series;"
 *PSVersion: "(2014.111) 1"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_2300-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_2300-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_2300-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_2300-ps.ppd	2011-11-21 15:56:33.085335977 +0000
@@ -61,6 +61,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 2300L)"
 *Product: "(HP LaserJet 2300L)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 2300 series;DES:Hewlett-Packard LaserJet 2300 series"
 *PSVersion: "(3010.107) 0"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_2300_series-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_2300_series-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_2300_series-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_2300_series-ps.ppd	2011-11-21 15:56:33.094335816 +0000
@@ -61,6 +61,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 2300L)"
 *Product: "(HP LaserJet 2300L)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 2300 series;DES:Hewlett-Packard LaserJet 2300 series"
 *PSVersion: "(3010.107) 0"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_2420-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_2420-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_2420-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_2420-ps.ppd	2011-11-21 15:56:33.221333549 +0000
@@ -60,6 +60,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 2420)"
 *Product: "(HP LaserJet 2420)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 2420;"
 *PSVersion: "(3010.107) 0"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_3390-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_3390-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_3390-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_3390-ps.ppd	2011-11-21 15:56:33.109335547 +0000
@@ -62,6 +62,7 @@
 *Product: "(HP LaserJet 3390)"
 *Product: "(Hewlett-Packard HP LaserJet 3390)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 3390;DES:Hewlett-Packard LaserJet 3390;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_4000_series-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_4000_series-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_4000_series-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_4000_series-ps.ppd	2011-11-21 15:56:33.114335459 +0000
@@ -51,6 +51,7 @@
 
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 4000 Series;DES:Hewlett-Packard LaserJet 4000 Series"
 *Product: "(HP LaserJet 4000 Series)"
 *Product: "(Hewlett-Packard HP LaserJet 4000 Series)"
 *ModelName: "HP LaserJet 4000 Series"
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_4050_series-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_4050_series-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_4050_series-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_4050_series-ps.ppd	2011-11-21 15:56:33.226333460 +0000
@@ -52,6 +52,7 @@
 
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 4050 Series ;"
 *Product: "(HP LaserJet 4050 Series )"
 *Product: "(Hewlett-Packard HP LaserJet 4050 Series )"
 *Product: "(HP LaserJet 4050 Series)"
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_4100_series-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_4100_series-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_4100_series-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_4100_series-ps.ppd	2011-11-21 15:56:33.232333352 +0000
@@ -51,6 +51,7 @@
 
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 4100 Series ;"
 *Product: "(HP LaserJet 4100 Series )"
 *Product: "(Hewlett-Packard HP LaserJet 4100 Series )"
 *Product: "(HP LaserJet 4100 Series)"
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_4200-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_4200-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_4200-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_4200-ps.ppd	2011-11-21 15:56:33.238333246 +0000
@@ -52,6 +52,7 @@
 
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 4200;"
 *Product: "(hp LaserJet 4200)"
 *Product: "(Hewlett-Packard hp LaserJet 4200)"
 *Product: "(HP LaserJet 4200)"
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_4250-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_4250-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_4250-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_4250-ps.ppd	2011-11-21 15:56:33.008337350 +0000
@@ -53,6 +53,7 @@
 *HPPDEPanel: "HPImagingOptions"
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 4250;DES:Hewlett-Packard LaserJet 4250;"
 *Product: "(hp LaserJet 4250)"
 *Product: "(Hewlett-Packard hp LaserJet 4250)"
 *Product: "(HP LaserJet 4250)"
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_4300-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_4300-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_4300-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_4300-ps.ppd	2011-11-21 15:56:33.244333138 +0000
@@ -52,6 +52,7 @@
 
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 4300;"
 *Product: "(hp LaserJet 4300)"
 *Product: "(Hewlett-Packard hp LaserJet 4300)"
 *Product: "(HP LaserJet 4300)"
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_4350-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_4350-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_4350-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:39.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_4350-ps.ppd	2011-11-21 15:56:33.257332906 +0000
@@ -53,6 +53,7 @@
 *HPPDEPanel: "HPImagingOptions"
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 4350;"
 *Product: "(hp LaserJet 4350)"
 *Product: "(Hewlett-Packard hp LaserJet 4350)"
 *Product: "(HP LaserJet 4350)"
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_5100_series-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_5100_series-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_5100_series-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:40.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_5100_series-ps.ppd	2011-11-21 15:56:33.264332782 +0000
@@ -53,6 +53,7 @@
 
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 5100 Series;"
 *Product: "(HP LaserJet 5100 Series)"
 *Product: "(Hewlett-Packard HP LaserJet 5100 Series)"
 *ModelName: "HP LaserJet 5100 Series"
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_8000-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_8000-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_8000-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:40.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_8000-ps.ppd	2011-11-21 15:56:33.270332674 +0000
@@ -55,6 +55,7 @@
 *%=================================================
 *% 		 Product Version Information
 *%=================================================
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 8000 Series;"
 *ModelName: "HP LaserJet 8000 Series"
 *ShortNickName: "HP LaserJet 8000 Series"
 *NickName: "HP LaserJet 8000 Series Postscript (recommended)"
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_8000_series-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_8000_series-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_8000_series-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:40.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_8000_series-ps.ppd	2011-11-21 15:56:33.281332478 +0000
@@ -55,6 +55,7 @@
 *%=================================================
 *% 		 Product Version Information
 *%=================================================
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 8000 Series;"
 *ModelName: "HP LaserJet 8000 Series"
 *ShortNickName: "HP LaserJet 8000 Series"
 *NickName: "HP LaserJet 8000 Series Postscript (recommended)"
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_m3027_mfp-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_m3027_mfp-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_m3027_mfp-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:40.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_m3027_mfp-ps.ppd	2011-11-21 15:56:33.290332317 +0000
@@ -56,6 +56,7 @@
 *%=================================================
 *% 		 Product Version Information
 *%=================================================
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet M3027 MFP;"
 *ModelName: "HP LaserJet M3027 MFP"
 *ShortNickName: "HP LaserJet M3027 MFP"
 *NickName: "HP LaserJet M3027 MFP Postscript (recommended)"
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_m3035_mfp-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_m3035_mfp-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_m3035_mfp-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:40.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_m3035_mfp-ps.ppd	2011-11-21 15:56:33.298332175 +0000
@@ -56,6 +56,7 @@
 *%=================================================
 *% 		 Product Version Information
 *%=================================================
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet M3035 MFP;"
 *ModelName: "HP LaserJet M3035 MFP"
 *ShortNickName: "HP LaserJet M3035 MFP"
 *NickName: "HP LaserJet M3035 MFP Postscript (recommended)"
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_p2015_series-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_p2015_series-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_p2015_series-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:40.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_p2015_series-ps.ppd	2011-11-21 15:56:33.101335690 +0000
@@ -64,6 +64,7 @@
 *Product: "(HP LaserJet P2015 Series)"
 *Product: "(Hewlett-Packard HP LaserJet P2015 Series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet P2015 Series;DES:Hewlett-Packard LaserJet P2015 series;"
 *PSVersion: "(3010.107) 0"
 
 *%=================================================
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_p3005-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_p3005-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_p3005-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:40.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_p3005-ps.ppd	2011-11-21 15:56:33.309331978 +0000
@@ -57,6 +57,7 @@
 *%=================================================
 *% 		 Product Version Information
 *%=================================================
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet P3005;"
 *ModelName: "HP LaserJet P3005"
 *ShortNickName: "HP LaserJet P3005"
 *NickName: "HP LaserJet P3005 Postscript (recommended)"
diff -up hplip-3.11.10/prnt/ps/hp-laserjet_p3010_series-ps.ppd.deviceIDs-ppd hplip-3.11.10/prnt/ps/hp-laserjet_p3010_series-ps.ppd
--- hplip-3.11.10/prnt/ps/hp-laserjet_p3010_series-ps.ppd.deviceIDs-ppd	2011-10-02 04:11:40.000000000 +0100
+++ hplip-3.11.10/prnt/ps/hp-laserjet_p3010_series-ps.ppd	2011-11-21 15:56:33.329331621 +0000
@@ -51,6 +51,7 @@
 *%== Mac OS X specific end ==
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet P3010 Series;"
 *Product: "(HP LaserJet P3015)"
 *Product: "(Hewlett-Packard HP LaserJet P3015)"
 *Product: "(HP LaserJet P3011)"
