diff -up hplip-3.10.9/base/utils.py.addgroup hplip-3.10.9/base/utils.py
--- hplip-3.10.9/base/utils.py.addgroup	2010-10-14 10:52:34.000000000 +0200
+++ hplip-3.10.9/base/utils.py	2010-10-14 11:35:17.000000000 +0200
@@ -65,8 +65,8 @@ def addgroup():
         fp=open('/etc/cups/cupsd.conf')
     except IOError:
         try:
-            if "root" != grp.getgrgid(os.stat('/etc/cups/cupsd.conf').st_gid)[0]:
-                return grp.getgrgid(os.stat('/etc/cups/cupsd.conf').st_gid)[0]
+            if "root" != grp.getgrgid(os.stat('/etc/cups/cupsd.conf').st_gid).gr_name:
+                return [grp.getgrgid(os.stat('/etc/cups/cupsd.conf').st_gid).gr_name]
         except OSError:
             return lis
 
