diff -up makehuman-0.9.1-rc1a/src/BodyPanel.h.gcc43 makehuman-0.9.1-rc1a/src/BodyPanel.h
--- makehuman-0.9.1-rc1a/src/BodyPanel.h.gcc43	2008-01-04 18:33:48.000000000 +0100
+++ makehuman-0.9.1-rc1a/src/BodyPanel.h	2008-01-04 18:40:16.000000000 +0100
@@ -32,6 +32,7 @@
 #endif
 
 #include <vector>
+#include <algorithm>
 #include <mhgui/Panel.h>
 #include <mhgui/Image.h>
 #include "util.h"
diff -up makehuman-0.9.1-rc1a/src/UtilitybarPanel.h.gcc43 makehuman-0.9.1-rc1a/src/UtilitybarPanel.h
--- makehuman-0.9.1-rc1a/src/UtilitybarPanel.h.gcc43	2008-01-04 18:49:25.000000000 +0100
+++ makehuman-0.9.1-rc1a/src/UtilitybarPanel.h	2008-01-04 18:49:35.000000000 +0100
@@ -32,6 +32,7 @@
 #endif
 
 #include <vector>
+#include <algorithm>
 #include <mhgui/Image.h>
 #include <mhgui/Panel.h>
 #include "ImageListener.h"
diff -up makehuman-0.9.1-rc1a/src/FacePanel.h.gcc43 makehuman-0.9.1-rc1a/src/FacePanel.h
--- makehuman-0.9.1-rc1a/src/FacePanel.h.gcc43	2008-01-04 18:44:21.000000000 +0100
+++ makehuman-0.9.1-rc1a/src/FacePanel.h	2008-01-04 18:44:33.000000000 +0100
@@ -32,6 +32,7 @@
 #endif
 
 #include <vector>
+#include <algorithm>
 #include <mhgui/Panel.h>
 #include <mhgui/Image.h>
 #include "TargetSelectionListener.h"
diff -up makehuman-0.9.1-rc1a/src/HandsPanel.h.gcc43 makehuman-0.9.1-rc1a/src/HandsPanel.h
--- makehuman-0.9.1-rc1a/src/HandsPanel.h.gcc43	2008-01-04 18:47:27.000000000 +0100
+++ makehuman-0.9.1-rc1a/src/HandsPanel.h	2008-01-04 18:47:35.000000000 +0100
@@ -32,6 +32,7 @@
 #endif
 
 #include <vector>
+#include <algorithm>
 #include <mhgui/Panel.h>
 #include <mhgui/Image.h>
 #include "util.h"
diff -up makehuman-0.9.1-rc1a/src/UtilitiesPanel.h.gcc43 makehuman-0.9.1-rc1a/src/UtilitiesPanel.h
--- makehuman-0.9.1-rc1a/src/UtilitiesPanel.h.gcc43	2008-01-04 19:11:07.000000000 +0100
+++ makehuman-0.9.1-rc1a/src/UtilitiesPanel.h	2008-01-04 19:11:25.000000000 +0100
@@ -32,6 +32,7 @@
 #endif
 
 #include <vector>
+#include <algorithm>
 #include <mhgui/Panel.h>
 #include <mhgui/ImageSlider.h>
 #include <mhgui/Image.h>
diff -up makehuman-0.9.1-rc1a/src/FileTools.h.gcc43 makehuman-0.9.1-rc1a/src/FileTools.h
--- makehuman-0.9.1-rc1a/src/FileTools.h.gcc43	2008-01-04 19:04:04.000000000 +0100
+++ makehuman-0.9.1-rc1a/src/FileTools.h	2008-01-04 19:08:48.000000000 +0100
@@ -32,7 +32,10 @@
  * It works well on Mac OS X (BSD) and Linux */
 #ifndef _WIN32
 
+#include <sys/types.h>
 #include <string>
+#include <cstring>
+#include <cstdlib>
 #include <list>
 #include <cassert>
 
diff -up makehuman-0.9.1-rc1a/src/CharacterSettingPanel.h.gcc43 makehuman-0.9.1-rc1a/src/CharacterSettingPanel.h
--- makehuman-0.9.1-rc1a/src/CharacterSettingPanel.h.gcc43	2008-01-04 18:57:23.000000000 +0100
+++ makehuman-0.9.1-rc1a/src/CharacterSettingPanel.h	2008-01-04 18:57:42.000000000 +0100
@@ -32,6 +32,7 @@
 #endif
 
 #include <vector>
+#include <algorithm>
 #include <mhgui/Panel.h>
 #include <mhgui/ImageSlider.h>
 #include <mhgui/Image.h>
diff -up makehuman-0.9.1-rc1a/src/TeethPanel.h.gcc43 makehuman-0.9.1-rc1a/src/TeethPanel.h
--- makehuman-0.9.1-rc1a/src/TeethPanel.h.gcc43	2008-01-04 18:45:59.000000000 +0100
+++ makehuman-0.9.1-rc1a/src/TeethPanel.h	2008-01-04 18:46:08.000000000 +0100
@@ -32,6 +32,7 @@
 #endif
 
 #include <vector>
+#include <algorithm>
 #include <mhgui/Panel.h>
 #include <mhgui/Image.h>
 #include "util.h"
diff -up makehuman-0.9.1-rc1a/src/ToolbarPanel.h.gcc43 makehuman-0.9.1-rc1a/src/ToolbarPanel.h
--- makehuman-0.9.1-rc1a/src/ToolbarPanel.h.gcc43	2008-01-04 18:52:39.000000000 +0100
+++ makehuman-0.9.1-rc1a/src/ToolbarPanel.h	2008-01-04 18:52:47.000000000 +0100
@@ -32,6 +32,7 @@
 #endif
 
 #include <vector>
+#include <algorithm>
 #include <mhgui/Panel.h>
 #include <mhgui/Image.h>
 #include "ImageListener.h"
diff -up makehuman-0.9.1-rc1a/src/ClothesPanel.h.gcc43 makehuman-0.9.1-rc1a/src/ClothesPanel.h
--- makehuman-0.9.1-rc1a/src/ClothesPanel.h.gcc43	2008-01-04 19:00:50.000000000 +0100
+++ makehuman-0.9.1-rc1a/src/ClothesPanel.h	2008-01-04 19:00:59.000000000 +0100
@@ -32,6 +32,7 @@
 #endif
 
 #include <vector>
+#include <algorithm>
 #include <mhgui/Panel.h>
 #include <mhgui/Image.h>
 #include "util.h"
diff -up makehuman-0.9.1-rc1a/src/PosesBodyPanel.h.gcc43 makehuman-0.9.1-rc1a/src/PosesBodyPanel.h
--- makehuman-0.9.1-rc1a/src/PosesBodyPanel.h.gcc43	2008-01-04 18:54:37.000000000 +0100
+++ makehuman-0.9.1-rc1a/src/PosesBodyPanel.h	2008-01-04 18:54:47.000000000 +0100
@@ -32,6 +32,7 @@
 #endif
 
 #include <vector>
+#include <algorithm>
 #include <mhgui/Panel.h>
 #include <mhgui/Image.h>
 #include "util.h"
