diff -Nur cdparanoia-III-10pre0-old/configure.in cdparanoia-III-10pre0/configure.in
--- cdparanoia-III-10pre0-old/configure.in	2008-04-02 22:30:49.000000000 +0300
+++ cdparanoia-III-10pre0/configure.in	2008-04-02 22:30:52.000000000 +0300
@@ -1,7 +1,5 @@
 AC_INIT(interface/interface.c)
 
-cp $srcdir/configure.guess $srcdir/config.guess
-cp $srcdir/configure.sub $srcdir/config.sub
 
 AC_CANONICAL_HOST
 
