diff -rupN jack-1.9.5.old/doxyfile jack-1.9.5/doxyfile
--- jack-1.9.5.old/doxyfile	2010-02-12 05:57:30.000000000 -0500
+++ jack-1.9.5/doxyfile	2010-07-19 23:18:57.000000000 -0400
@@ -413,7 +413,7 @@ HTML_HEADER            = 
 # each generated HTML page. If it is left blank doxygen will generate a 
 # standard footer.
 
-HTML_FOOTER            = 
+HTML_FOOTER            = no_date_footer.html
 
 # The HTML_STYLESHEET tag can be used to specify a user defined cascading 
 # style sheet that is used by each HTML page. It can be used to 
diff -rupN jack-1.9.5.old/no_date_footer.html jack-1.9.5/no_date_footer.html
--- jack-1.9.5.old/no_date_footer.html	1969-12-31 19:00:00.000000000 -0500
+++ jack-1.9.5/no_date_footer.html	2010-07-19 23:19:19.000000000 -0400
@@ -0,0 +1,6 @@
+<hr size="1"><address style="text-align: right;"><small>
+Generated for $projectname by <a href="http://www.doxygen.org/
+index.html"><img src="doxygen.png" alt="doxygen" align="middle" border="0"></a>
+$doxygenversion</small></address>
+</body>
+</html>
