diff -rupN jack-1.9.5.old/common/JackNetOneDriver.cpp jack-1.9.5/common/JackNetOneDriver.cpp
--- jack-1.9.5.old/common/JackNetOneDriver.cpp	2010-02-12 05:57:48.000000000 -0500
+++ jack-1.9.5/common/JackNetOneDriver.cpp	2010-07-20 01:26:44.000000000 -0400
@@ -709,9 +709,9 @@ JackNetOneDriver::render_payload_to_jack
 
 	    CELTDecoder *decoder = (CELTDecoder *)src_node->data;
 	    if( !packet_payload )
-		celt_decode_float( decoder, NULL, net_period_down, buf );
+                celt_decode_float( decoder, NULL, net_period_down, buf, NULL );
 	    else
-		celt_decode_float( decoder, packet_bufX, net_period_down, buf );
+                celt_decode_float( decoder, packet_bufX, net_period_down, buf, NULL );
 
 	    src_node = jack_slist_next (src_node);
         }
diff -rupN jack-1.9.5.old/common/netjack_packet.c jack-1.9.5/common/netjack_packet.c
--- jack-1.9.5.old/common/netjack_packet.c	2010-02-12 05:57:47.000000000 -0500
+++ jack-1.9.5/common/netjack_packet.c	2010-07-20 01:12:42.000000000 -0400
@@ -1428,9 +1428,9 @@ render_payload_to_jack_ports_celt (void 
 
 	    CELTDecoder *decoder = src_node->data;
 	    if( !packet_payload )
-		celt_decode_float( decoder, NULL, net_period_down, buf );
+	      celt_decode_float( decoder, NULL, net_period_down, buf, NULL );
 	    else
-		celt_decode_float( decoder, packet_bufX, net_period_down, buf );
+	      celt_decode_float( decoder, packet_bufX, net_period_down, buf, NULL );
 
 	    src_node = jack_slist_next (src_node);
         }
