diff -uNr espeak-1.23-source/src/Makefile espeak-1.23-source-mod/src/Makefile
--- espeak-1.23-source/src/Makefile	2007-04-21 19:05:23.000000000 +0200
+++ espeak-1.23-source-mod/src/Makefile	2007-04-24 07:50:46.000000000 +0200
@@ -86,7 +86,7 @@
 	# Install shared library
 	$(INSTALL) -m 755 $(LIB_NAME).$(LIBTAG) $(DESTDIR)$(LIBDIR)
 	# Install static library
-	$(INSTALL) -m 755 $(STATIC_LIB_NAME) $(DESTDIR)$(LIBDIR)
+	#$(INSTALL) -m 755 $(STATIC_LIB_NAME) $(DESTDIR)$(LIBDIR)
 	$(LN_SF) $(LIB_NAME).$(LIBTAG) $(DESTDIR)$(LIBDIR)/$(LIB_NAME).$(LIB_VERSION)
 	$(LN_SF) $(LIB_NAME).$(LIB_VERSION) $(DESTDIR)$(LIBDIR)/$(LIB_NAME)
 	# Install development headers
