Only in indent-2.2.9/doc: all_opts
diff -Bburpd indent-2.2.9-orig/doc/indent.texinfo indent-2.2.9/doc/indent.texinfo
--- indent-2.2.9-orig/doc/indent.texinfo	2006-07-16 07:25:25.000000000 -0400
+++ indent-2.2.9/doc/indent.texinfo	2006-07-16 07:57:33.000000000 -0400
@@ -754,7 +754,7 @@ if (x > 0)
 
 @kindex -ce
 @kindex --cuddle-else
-@kindex -dce
+@kindex -nce
 @kindex --dont-cuddle-else
 If you are using the @option{-br} option, you probably want to also use
 the @option{-ce} option.  This causes the @code{else} in an if-then-else
@@ -1666,6 +1666,11 @@ Line up continued lines at parentheses.@
 Leave space between @samp{#} and preprocessor directive.@*
 @xref{Indentation}.
 
+@item -nlps
+@itemx --remove-preprocessor-space
+Remove space between @samp{#} and preprocessor directive.@*
+@xref{Indentation}.
+
 @item -nbad
 @itemx --no-blank-lines-after-declarations
 Do not force blank lines after declarations.@*
@@ -1979,6 +1989,7 @@ the corresponding short option.
 \line{ --preprocessor-indentation \leaderfill      -ppi@var{n}\ }
 \line{ --preserve-mtime \leaderfill                -pmt\ }
 \line{ --procnames-start-lines \leaderfill         -psl\ }  
+\line{ --remove-preprocessor-space \leaderfill     -nlps\ }
 \line{ --space-after-cast \leaderfill              -cs\ \ } 
 \line{ --space-after-for \leaderfill               -saf\ }
 \line{ --space-after-if \leaderfill                -sai\ }
@@ -2063,6 +2075,7 @@ the corresponding short option.
 --preserve-mtime				-pmt
 --preprocessor-indentation                      -ppi@var{n}
 --procnames-start-lines                         -psl            
+--remove-preprocessor-space                     -nlps
 --space-after-cast                              -cs             
 --space-after-for				-saf
 --space-after-if				-sai
