diff -burp indent-2.2.9/src/indent.c indent-2.2.9-lcall/src/indent.c
--- indent-2.2.9/src/indent.c	2006-02-01 14:20:25.000000000 +0100
+++ indent-2.2.9-lcall/src/indent.c	2006-02-01 14:08:52.000000000 +0100
@@ -3070,7 +3070,7 @@ int main (
     exit_values_ty exit_status;
 
 #if defined (HAVE_SETLOCALE) && defined (HAVE_LC_MESSAGES) && defined (HAVE_LCCTYPES)
-    setlocale(LC_MESSAGES, "");
+    setlocale(LC_ALL, "");
 #endif
     bindtextdomain(PACKAGE, LOCALEDIR);
     textdomain(PACKAGE);
