From c0001598a2b5b4cc8739beeb183f0d3f9c6a1d84 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Thu, 2 Feb 2012 17:12:44 +0100
Subject: [PATCH] Return non-zero exit code on tests failure

---
 regression/TEST |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/regression/TEST b/regression/TEST
index c860ef2..620e77f 100755
--- a/regression/TEST
+++ b/regression/TEST
@@ -441,6 +441,7 @@ then
   cat  $ERR   
   echo '****  Errors occured: ./output not removed'
   echo
+  exit 1
 else
   cd ..
     ls -l output
-- 
1.7.7.6

