--- include/mpakogre.h.old	2012-11-24 00:08:29.483485776 +0100
+++ include/mpakogre.h	2012-11-24 00:09:09.840160157 +0100
@@ -54,7 +54,7 @@
 	FileInfoListPtr listFileInfo(bool recursive = true, bool dirs = false);
 
 	StringVectorPtr find(const String &pattern, bool recursive = true, bool dirs = false);
-	FileInfoListPtr findFileInfo(const String &pattern, bool recursive, bool dirs = false);
+	FileInfoListPtr findFileInfo(const String &pattern, bool recursive, bool dirs = false) const;
 
 	bool exists(const String &filename);
 
--- src/mpakogre.cpp.old	2012-11-24 00:04:48.801730942 +0100
+++ src/mpakogre.cpp	2012-11-26 11:08:04.114559127 +0100
@@ -180,10 +180,10 @@
 	return ret;
 }
 
-FileInfoListPtr MPakArchive::findFileInfo(const String &pattern, bool recursive, bool dirs) {
+FileInfoListPtr MPakArchive::findFileInfo(const String &pattern, bool recursive, bool dirs) const {
 	FileInfoListPtr ret = FileInfoListPtr(new FileInfoList());
 
-	FileInfoList::iterator i, iend;
+	FileInfoList::const_iterator i, iend;
 	iend = mFileList.end();
 	for(i = mFileList.begin(); i != iend; ++i) {
 		if(StringUtil::match(i->filename, pattern, true))
--- src/openalsoundsystem.cpp.orig	2012-11-26 11:25:48.798301526 +0100
+++ src/openalsoundsystem.cpp	2012-11-26 11:29:53.556485128 +0100
@@ -35,16 +35,16 @@
 #include "playlist.h"
 #include "mpakogre.h"
 
-template<> OpenALSoundSystem* Singleton<OpenALSoundSystem>::ms_Singleton = 0;
+template<> OpenALSoundSystem* Singleton<OpenALSoundSystem>::msSingleton = 0;
 
 OpenALSoundSystem* OpenALSoundSystem::getSingletonPtr() {
-	return ms_Singleton;
+	return msSingleton;
 }
 
 
 OpenALSoundSystem& OpenALSoundSystem::getSingleton() {
-	assert(ms_Singleton);
-	return *ms_Singleton;
+	assert(msSingleton);
+	return *msSingleton;
 }
 
 
--- src/objectsystem.cpp.orig	2012-11-26 11:27:14.562919664 +0100
+++ src/objectsystem.cpp	2012-11-26 11:29:48.549625145 +0100
@@ -26,7 +26,7 @@
 #include "objectsystem.h"
 #include "game.h"
 
-template<> ObjectSystem* Singleton<ObjectSystem>::ms_Singleton = 0;
+template<> ObjectSystem* Singleton<ObjectSystem>::msSingleton = 0;
 
 
 // ObjectSystem constructor
@@ -135,11 +135,11 @@
 
 
 ObjectSystem *ObjectSystem::getSingletonPtr() {
-	return ms_Singleton;
+	return msSingleton;
 }
 
 ObjectSystem &ObjectSystem::getSingleton() {
-	assert(ms_Singleton);
-	return *ms_Singleton;
+	assert(msSingleton);
+	return *msSingleton;
 }
 
--- src/scriptsystem.cpp.orig	2012-11-26 11:31:47.781284474 +0100
+++ src/scriptsystem.cpp	2012-11-26 11:32:04.409817569 +0100
@@ -38,7 +38,7 @@
 #include "mpakogre.h"
 #include <vector>
 
-template<> ScriptSystem* Singleton<ScriptSystem>::ms_Singleton = 0;
+template<> ScriptSystem* Singleton<ScriptSystem>::msSingleton = 0;
 
 
 // Log error messages
@@ -564,12 +564,12 @@
 
 
 ScriptSystem *ScriptSystem::getSingletonPtr() {
-	return ms_Singleton;
+	return msSingleton;
 }
 
 ScriptSystem &ScriptSystem::getSingleton() {
-	assert(ms_Singleton);
-	return *ms_Singleton;
+	assert(msSingleton);
+	return *msSingleton;
 }
 
 
