diff -up funguloids/src/scriptsystem.cpp.lua52 funguloids/src/scriptsystem.cpp
--- funguloids/src/scriptsystem.cpp.lua52	2013-05-29 12:41:07.394520691 -0400
+++ funguloids/src/scriptsystem.cpp	2013-05-29 12:43:12.591522570 -0400
@@ -38,6 +38,10 @@
 #include "mpakogre.h"
 #include <vector>

+#ifndef lua_open
+#define lua_open()     luaL_newstate()
+#endif

 template<> ScriptSystem* Singleton<ScriptSystem>::msSingleton = 0;


@@ -517,6 +521,7 @@ void ScriptSystem::destroyScriptLights()
 // Execute a single string of script
 bool ScriptSystem::executeString(const char *str) {
 	assert(mLuaVM);
+	luaL_openlibs(mLuaVM);
 	if(luaL_dostring(mLuaVM, str))
 		return false;
 	return true;
