Index: bin/Makefile.am
===================================================================
--- bin/Makefile.am.orig
+++ bin/Makefile.am
@@ -4,7 +4,7 @@ EXTRA_DIST = plugins.cfg.in resources.cf
 dist_pixmapinstall_DATA = funguloids.png
 dist_gameinstall_DATA = gamesettings.cfg
 dist_docinstall_DATA = readme.html
-gameinstall_DATA = plugins.cfg resources.cfg
+gameinstall_DATA = plugins.cfg resources.cfg bootstrap.mpk funguloids.mpk
 
 plugins.cfg: plugins.cfg.in
 	cat plugins.cfg.in | sed 's,%%ogreplugindir%%,@ogreplugindir@/,' > plugins.cfg
@@ -22,9 +22,7 @@ install-data-hook:
 		if test ! -e "@highscoredir@/hiscore.lst"; then cp hiscore.lst.in "@highscoredir@/hiscore.lst"; fi; \
 		chgrp -R @highscoregroup@ "@highscoredir@"; \
 		chmod -R g+w "@highscoredir@"; \
-	fi; \
-	if test -r bootstrap.mpk; then cp bootstrap.mpk "@gameinstalldir@"; fi; \
-	if test -r funguloids.mpk; then cp funguloids.mpk "@gameinstalldir@"; fi
+	fi;
 
 uninstall-hook:
 	rm -rf "@docinstalldir@"
Index: bin/music/Makefile.am
===================================================================
--- bin/music/Makefile.am.orig
+++ bin/music/Makefile.am
@@ -1,13 +1,10 @@
-musicinstall_DATA = playlist.lua
+musicinstall_DATA = playlist.lua MarylandInMay.ogg
 
 EXTRA_DIST = playlist.lua.in
 
 playlist.lua: playlist.lua.in
 	cat playlist.lua.in | sed 's,%%musicinstalldir%%,@musicinstalldir@,' > playlist.lua
 
-install-data-hook:
-	if test -r MarylandInMay.ogg; then cp MarylandInMay.ogg "@musicinstalldir@"; fi;
-
 
 distclean: clean
 
