Index: src/openalsoundsystem.cpp
===================================================================
--- src/openalsoundsystem.cpp.orig
+++ src/openalsoundsystem.cpp
@@ -266,14 +266,14 @@ int OpenALSoundSystem::initialise( unsig
 
 	int attributes[] = { 0 };
 	mContext = alcCreateContext(mDevice, attributes);
-	if ( (err = alGetError()) != AL_NO_ERROR) {
+	if ( (err = alcGetError(mDevice)) != ALC_NO_ERROR) {
 		LogManager::getSingleton().logMessage("OpanAL: error creating context: " + lastALError(err) );
 		return 1;
 	}
 //	check_alc_error("Couldn't create audio context: ");
 	alcMakeContextCurrent(mContext);
 //	check_alc_error("Couldn't select audio context: ");
-	if ( (err = alGetError()) != AL_NO_ERROR) {
+	if ( (err = alcGetError(mDevice)) != ALC_NO_ERROR) {
 		LogManager::getSingleton().logMessage("OpanAL: error making context current: " + lastALError(err) );
 		return 2;
 	}
