--- rubygems-1.8.5/lib/rubygems/commands/setup_command.rb.rdoc_v	2011-05-26 13:01:53.000000000 +0900
+++ rubygems-1.8.5/lib/rubygems/commands/setup_command.rb	2011-06-02 00:11:14.000000000 +0900
@@ -329,7 +329,8 @@
 
     require 'rdoc/rdoc'
 
-    args << '--main' << 'README.rdoc' << '--quiet'
+    args << '--main' << 'README.rdoc'
+    args << '--quiet' unless Gem.configuration.really_verbose
     args << '.'
     args << 'README.rdoc' << 'UPGRADING.rdoc'
     args << 'LICENSE.txt' << 'MIT.txt' << 'History.txt'
--- rubygems-1.8.5/lib/rubygems/doc_manager.rb.rdoc_v	2011-05-19 08:36:37.000000000 +0900
+++ rubygems-1.8.5/lib/rubygems/doc_manager.rb	2011-06-02 00:09:15.000000000 +0900
@@ -165,7 +165,7 @@
     args << @spec.require_paths.clone
     args << @spec.extra_rdoc_files
     args << '--title' << "#{@spec.full_name} Documentation"
-    args << '--quiet'
+    args << '--quiet' unless Gem.configuration.really_verbose
     args = args.flatten.map do |arg| arg.to_s end
 
     if self.class.rdoc_version >= Gem::Version.new('2.4.0') then
