diff -Nur klavaro-1.9.5-orig/Makefile.in klavaro-1.9.5/Makefile.in
--- klavaro-1.9.5-orig/Makefile.in	2012-06-03 16:51:36.000000000 +0200
+++ klavaro-1.9.5/Makefile.in	2012-06-27 22:50:40.840095032 +0200
@@ -148,7 +148,7 @@
 INTLTOOL_V_MERGE_OPTIONS = @INTLTOOL_V_MERGE_OPTIONS@
 INTLTOOL__v_MERGE_ = @INTLTOOL__v_MERGE_@
 INTLTOOL__v_MERGE_0 = @INTLTOOL__v_MERGE_0@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ -lgmodule-2.0
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
 LTLIBOBJS = @LTLIBOBJS@
diff -Nur klavaro-1.9.5-orig/src/Makefile.in klavaro-1.9.5/src/Makefile.in
--- klavaro-1.9.5-orig/src/Makefile.in	2012-06-03 16:51:36.000000000 +0200
+++ klavaro-1.9.5/src/Makefile.in	2012-06-27 22:51:03.644267073 +0200
@@ -113,7 +113,7 @@
 INTLTOOL_V_MERGE_OPTIONS = @INTLTOOL_V_MERGE_OPTIONS@
 INTLTOOL__v_MERGE_ = @INTLTOOL__v_MERGE_@
 INTLTOOL__v_MERGE_0 = @INTLTOOL__v_MERGE_0@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ -lgmodule-2.0
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
 LTLIBOBJS = @LTLIBOBJS@
