diff -Nur MemonixSourceCode-old/CMakeLists.txt MemonixSourceCode/CMakeLists.txt
--- MemonixSourceCode-old/CMakeLists.txt	2009-10-20 12:30:15.860003062 +0300
+++ MemonixSourceCode/CMakeLists.txt	2009-10-20 12:31:54.425967566 +0300
@@ -12,6 +12,9 @@
 # into current directory and run ./Memonix
 #
 
+CMAKE_MINIMUM_REQUIRED(VERSION 2.4)
+
+
 # project name
 PROJECT(Memonix)
 
@@ -31,9 +34,11 @@
 FIND_PACKAGE(SDL_mixer REQUIRED)
 
 # sources
-FILE(GLOB_RECURSE memonix_SRCS *.cpp)
+FILE(GLOB_RECURSE memonix_SRCS src/*.cpp)
 
 # targets
-ADD_DEFINITIONS(`${SDL_CONFIG} --cflags`)
+EXEC_PROGRAM(${SDL_CONFIG} ARGS "--cflags" OUTPUT_VARIABLE SDL_CFLAGS)
+EXEC_PROGRAM(${SDL_CONFIG} ARGS "--libs" OUTPUT_VARIABLE SDL_LIBS)
+ADD_DEFINITIONS(${SDL_CFLAGS})
 ADD_EXECUTABLE(Memonix ${memonix_SRCS})
-TARGET_LINK_LIBRARIES(Memonix `${SDL_CONFIG} --libs` ${SDLIMAGE_LIBRARY} ${SDLMIXER_LIBRARY} ${OPENGL_gl_LIBRARY} ${GLU_LIBRARY})
+TARGET_LINK_LIBRARIES(Memonix ${SDL_LIBS} ${SDLIMAGE_LIBRARY} ${SDLMIXER_LIBRARY} ${OPENGL_gl_LIBRARY} ${GLU_LIBRARY})
