Index: pm-utils-1.4.1/pm/module.d/uswsusp
===================================================================
--- pm-utils-1.4.1.orig/pm/module.d/uswsusp
+++ pm-utils-1.4.1/pm/module.d/uswsusp
@@ -82,6 +82,7 @@ fi
 if [ -z "$HIBERNATE_MODULE" ] && \
 	[ -f /sys/power/disk ] && \
 	grep -q disk /sys/power/state && \
+	grep -q "resume=" /proc/cmdline && \
 	[ -c /dev/snapshot ] &&
 	command_exists s2disk; then
 	HIBERNATE_MODULE="uswsusp"
Index: pm-utils-1.4.1/pm/pm-functions.in
===================================================================
--- pm-utils-1.4.1.orig/pm/pm-functions.in
+++ pm-utils-1.4.1/pm/pm-functions.in
@@ -306,7 +306,8 @@ fi
 
 if [ -z "$HIBERNATE_MODULE" ] && \
 	[ -f /sys/power/disk ] && \
-	grep -q disk /sys/power/state; then
+	grep -q disk /sys/power/state && \
+	grep -q "resume=" /proc/cmdline; then
 	HIBERNATE_MODULE="kernel"
 	do_hibernate()
 	{
