Index: pm-utils-1.4.1/configure.ac
===================================================================
--- pm-utils-1.4.1.orig/configure.ac
+++ pm-utils-1.4.1/configure.ac
@@ -41,7 +41,6 @@ man/Makefile
 src/Makefile
 pm/Makefile
 pm/sleep.d/Makefile
-pm/power.d/Makefile
 pm/module.d/Makefile
 ])
 
Index: pm-utils-1.4.1/man/Makefile.am
===================================================================
--- pm-utils-1.4.1.orig/man/Makefile.am
+++ pm-utils-1.4.1/man/Makefile.am
@@ -2,7 +2,6 @@ pm_mans =			\
 	on_ac_power.1		\
 	pm-is-supported.1	\
 	pm-pmu.8		\
-	pm-powersave.8		\
 	pm-action.8
 
 if HAVE_XMLTO
@@ -13,7 +12,6 @@ EXTRA_DIST =			\
 	on_ac_power.xml		\
 	pm-pmu.xml		\
 	pm-is-supported.xml	\
-	pm-powersave.xml	\
 	pm-action.xml		\
 	$(pm_mans)
 
Index: pm-utils-1.4.1/pm/sleep.d/Makefile.am
===================================================================
--- pm-utils-1.4.1.orig/pm/sleep.d/Makefile.am
+++ pm-utils-1.4.1/pm/sleep.d/Makefile.am
@@ -2,8 +2,7 @@ sleepdir = $(libdir)/pm-utils/sleep.d
 
 sleep_SCRIPTS =			\
 	00logging		\
-	00powersave		\
-	01grub			\
+	01grub 			\
 	49bluetooth		\
 	55NetworkManager	\
 	75modules		\
Index: pm-utils-1.4.1/pm/Makefile.am
===================================================================
--- pm-utils-1.4.1.orig/pm/Makefile.am
+++ pm-utils-1.4.1/pm/Makefile.am
@@ -1,6 +1,5 @@
 SUBDIRS =		\
 	sleep.d 	\
-	power.d		\
 	module.d
 
 pm_libdir = $(libdir)/pm-utils
Index: pm-utils-1.4.1/pm/HOWTO.hooks
===================================================================
--- pm-utils-1.4.1.orig/pm/HOWTO.hooks
+++ pm-utils-1.4.1/pm/HOWTO.hooks
@@ -24,12 +24,6 @@ The actual sleep method being used will
 if your hook needs to handle suspend-hybrid (or any other platform-specific
 sleep method), it should examine the second parameter.
 
-For hooks in power.d, the potential values of that parameter are:
-true -- the hook MUST perform whatever action is appropriate when the system 
-	transitions TO battery power.
-false -- The hook MUST perform whatever action is appropriate when the system
-	transitions FROM battery power.
-
 NAMING SCHEME
 
 All hooks are run in lexical sort order according to the C locale.
@@ -44,8 +38,7 @@ Any other return code is interpreted by
 from the hook that it should abort whatever it is doing. When running sleep.d
 hooks, that means that pm-utils stops running hooks, aborts the suspend/resume
 process, calls any hooks that ran successfully prior to this one with the 
-appropriate wakeup options, and exits with a non-zero exit code. When running
-power.d hooks, any hooks after this one will be skipped.
+appropriate wakeup options, and exits with a non-zero exit code.
 
 SLEEP.D SPECIFIC NOTES
 
Index: pm-utils-1.4.1/src/Makefile.am
===================================================================
--- pm-utils-1.4.1.orig/src/Makefile.am
+++ pm-utils-1.4.1/src/Makefile.am
@@ -16,9 +16,7 @@ bin_SCRIPTS = pm-is-supported
 
 dist_bin_SCRIPTS = on_ac_power
 
-sbin_SCRIPTS = pm-powersave
-
-script_in_files = pm-action.in pm-is-supported.in pm-powersave.in service
+script_in_files = pm-action.in pm-is-supported.in service
 
 CLEANFILES =  $(script_in_files:.in=)
 
