diff -p -up xvidcap-1.1.7/src/codecs.c~ xvidcap-1.1.7/src/codecs.c
--- xvidcap-1.1.7/src/codecs.c~	2008-05-26 00:01:53.000000000 +0200
+++ xvidcap-1.1.7/src/codecs.c	2008-10-27 11:07:30.000000000 +0100
@@ -43,8 +43,8 @@
 #include <locale.h>
 
 #ifdef USE_FFMPEG
-#include <ffmpeg/avcodec.h>
-#include <ffmpeg/avformat.h>
+#include <avcodec.h>
+#include <avformat.h>
 #endif     // USE_FFMPEG
 
 #include "app_data.h"
diff -p -up xvidcap-1.1.7/src/main.c~ xvidcap-1.1.7/src/main.c
--- xvidcap-1.1.7/src/main.c~	2008-05-26 00:01:54.000000000 +0200
+++ xvidcap-1.1.7/src/main.c	2008-10-27 11:11:11.000000000 +0100
@@ -45,7 +45,7 @@
 #include <locale.h>
 
 #ifdef USE_FFMPEG
-#include <ffmpeg/avcodec.h>
+#include <avcodec.h>
 #endif     // USE_FFMPEG
 
 #include "control.h"
diff -p -up xvidcap-1.1.7/src/xtoffmpeg.c~ xvidcap-1.1.7/src/xtoffmpeg.c
--- xvidcap-1.1.7/src/xtoffmpeg.c~	2008-05-26 00:01:54.000000000 +0200
+++ xvidcap-1.1.7/src/xtoffmpeg.c	2008-10-27 11:11:54.000000000 +0100
@@ -56,13 +56,13 @@
 #include "xvidcap-intl.h"
 
 // ffmpeg stuff
-#include <ffmpeg/avcodec.h>
-#include <ffmpeg/avformat.h>
-#include <ffmpeg/avdevice.h>
+#include <avcodec.h>
+#include <avformat.h>
+#include <avdevice.h>
 //#include <ffmpeg/dsputil.h>
-#include <ffmpeg/swscale.h>
-#include <ffmpeg/rgb2rgb.h>
-#include <ffmpeg/fifo.h>
+#include <swscale.h>
+#include <rgb2rgb.h>
+#include <fifo.h>
 #define swscale_isRGB(x) ((x)==PIX_FMT_BGR32 || (x)==PIX_FMT_RGB24 \
                         || (x)==PIX_FMT_RGB565 || (x)==PIX_FMT_RGB555 \
                         || (x)==PIX_FMT_RGB8 || (x)==PIX_FMT_RGB4 \
