--- ppm2mpeg.sh	2007-04-24 15:15:42.000000000 +0000
+++ ppm2mpeg.sh.tpg	2007-04-29 20:37:33.000000000 +0000
@@ -18,7 +18,7 @@
 LOC_TRANSCODE=`which transcode`
 LOC_FFMPEG=`which ffmpeg`
 LOC_MENCODER=`which mencoder`
-EXTENSION=`echo "${1}" | nawk -F "." '{print tolower($NF)}'`
+EXTENSION=`echo "${1}" | awk -F "." '{print tolower($NF)}'`
 FILE=`echo "${1}" | sed 's/%[\.]*[0-9]*[i|d]/*/g'`
 OUTPUTFILE="/tmp/output.avi"
 
