diff -Naur xvidcap-1.1.7.orig/configure.in xvidcap-1.1.7/configure.in
--- xvidcap-1.1.7.orig/configure.in	2012-01-27 22:54:17.000000000 +0100
+++ xvidcap-1.1.7/configure.in	2012-01-27 22:55:06.569962858 +0100
@@ -248,6 +248,14 @@
 # Checks for libraries.
 #########################################################
 
+
+#########################################################
+# Zlib
+#########################################################
+
+AC_CHECK_LIB(z,[main])
+
+
 #########################################################
 # X11 libraries
 #########################################################
@@ -266,6 +274,13 @@
 # first check for Xdamage without tweaking, then in X11 paths
 AC_CHECK_LIB(Xdamage,XDamageSubtract,,[unset ac_cv_lib_Xdamage_XDamageSubtract; echo "Couldn't find libXdamage in LD_LIBRARY_PATH, checking X11 paths"; AC_CHECK_LIB(Xdamage,XDamageSubtract,LDFLAGS="${LDFLAGS} -L${ac_x_libraries} -Xlinker -R${ac_x_libraries}"; LIBS="${LIBS} -lXdamage",[echo "libXdamage not available, cannot use delta screenshots"],[-L${ac_x_libraries}])])
 
+# X11
+AC_CHECK_LIB(X11,[main])
+
+# Xext
+AC_CHECK_LIB(Xext,[main])
+
+
 ## libice test is present, but we need to bail out if not there
 
 if test "x${ac_cv_lib_ICE_IceConnectionNumber+set}" = "x"; then
