diff -Nur xine-lib-1.1.18.1-old/src/demuxers/demux_qt.c xine-lib-1.1.18.1/src/demuxers/demux_qt.c
--- xine-lib-1.1.18.1-old/src/demuxers/demux_qt.c	2010-04-15 17:34:34.856980362 +0300
+++ xine-lib-1.1.18.1/src/demuxers/demux_qt.c	2010-04-15 17:41:59.561982714 +0300
@@ -1245,10 +1245,24 @@
           /* fetch audio parameters */
           trak->stsd_atoms[k].audio.codec_fourcc =
             _X_ME_32(&trak_atom[atom_pos + 0x0]);
-          trak->stsd_atoms[k].audio.sample_rate =
-            _X_BE_16(&trak_atom[atom_pos + 0x1C]);
-          trak->stsd_atoms[k].audio.channels = trak_atom[atom_pos + 0x15];
-          trak->stsd_atoms[k].audio.bits = trak_atom[atom_pos + 0x17];
+          switch(trak->stsd_atoms[k].audio.codec_fourcc) {
+              case 0x726D6173: /* samr */
+                  /* amr narrowband */
+                  trak->stsd_atoms[k].audio.sample_rate = 8000;
+                  trak->stsd_atoms[k].audio.channels = 1;
+                  trak->stsd_atoms[k].audio.bits = 8;
+                  break;
+              case 0x62776173: /* sawb */
+                  /* amr wideband */
+                  trak->stsd_atoms[k].audio.sample_rate = 16000;
+                  trak->stsd_atoms[k].audio.channels = 1;
+                  trak->stsd_atoms[k].audio.bits = 8;
+                  break;
+              default:
+                  trak->stsd_atoms[k].audio.sample_rate = _X_BE_16(&trak_atom[atom_pos + 0x1C]);
+                  trak->stsd_atoms[k].audio.channels = trak_atom[atom_pos + 0x15];
+                  trak->stsd_atoms[k].audio.bits = trak_atom[atom_pos + 0x17];
+          }
 
           /* assume uncompressed audio parameters */
           trak->stsd_atoms[k].audio.bytes_per_sample =
@@ -3207,6 +3221,8 @@
     if (strncasecmp (ending, ".mov", 4) &&
         strncasecmp (ending, ".qt", 3) &&
         strncasecmp (ending, ".qtl", 4) &&
+        strncasecmp (ending, ".3gpp", 5) &&
+        strncasecmp (ending, ".3gp", 4) &&
         strncasecmp (ending, ".mp4", 4)) {
       free (this);
       return NULL;
@@ -3255,7 +3271,7 @@
 }
 
 static const char *get_extensions (demux_class_t *this_gen) {
-  return "mov qt qtl mp4 m4a m4b";
+  return "mov qt qtl 3gpp 3gp mp4 m4a m4b";
 }
 
 static const char *get_mimetypes (demux_class_t *this_gen) {
@@ -3263,6 +3279,9 @@
          "video/x-quicktime: mov,qt: Quicktime animation;"
          "audio/x-m4a: m4a,m4b: MPEG-4 audio;"
          "application/x-quicktimeplayer: qtl: Quicktime list;"
+         "video/3gpp: 3gpp,3gp: 3GPP multimedia;"
+         "video/mp4: mp4: MPEG-4 video;"
+         "audio/mp4: mp4: MPEG-4 audio;"
          "video/mp4: mp4,mpg4: MPEG-4 video;"
          "audio/mp4: mp4,mpg4: MPEG-4 audio;";
 }
