diff -Nur xine-lib-1.1.13-old/src/xine-engine/audio_decoder.c xine-lib-1.1.13/src/xine-engine/audio_decoder.c
--- xine-lib-1.1.13-old/src/xine-engine/audio_decoder.c	2008-06-25 18:27:09.784349979 +0300
+++ xine-lib-1.1.13/src/xine-engine/audio_decoder.c	2008-06-25 18:28:05.515475058 +0300
@@ -465,7 +465,7 @@
   int                  err;
 
   if (stream->audio_out == NULL) {
-    stream->audio_fifo = _x_dummy_fifo_buffer_new (5, 8192);
+    stream->audio_fifo = _x_dummy_fifo_buffer_new (5, 16384);
     return 1;
   } else {
     int num_buffers;
@@ -488,7 +488,7 @@
 							"also increased latency and memory consumption."),
                                                       20, NULL, NULL);
   
-    stream->audio_fifo = _x_fifo_buffer_new (num_buffers, 8192);
+    stream->audio_fifo = _x_fifo_buffer_new (num_buffers, 16384);
     stream->audio_channel_user = -1;
     stream->audio_channel_auto = -1;
     stream->audio_track_map_entries = 0;
