diff -urN xine-lib-1.1.2.orig/configure.ac xine-lib-1.1.2/configure.ac
--- xine-lib-1.1.2.orig/configure.ac	2006-07-09 17:36:29 +0300
+++ xine-lib-1.1.2/configure.ac	2006-11-15 00:32:56 +0200
@@ -1213,6 +1213,9 @@
   AC_MSG_RESULT([libmad support disabled])
 elif test x"$external_libmad" = "xyes"; then
   PKG_CHECK_MODULES(LIBMAD, [mad], have_mad=yes, have_mad=no)
+  if test x"$have_mad" = "xno"; then
+    PKG_CHECK_MODULES(LIBMAD, [libmad], have_mad=yes, have_mad=no)
+  fi
   AC_CHECK_HEADERS([mad.h])
   AC_SUBST(LIBMAD_LIBS)
   AC_SUBST(LIBMAD_CFLAGS)
