diff -urN xine-lib-1.1.5.orig/configure.ac xine-lib-1.1.5/configure.ac
--- xine-lib-1.1.5.orig/configure.ac	2007-04-08 19:31:32 +0300
+++ xine-lib-1.1.5/configure.ac	2007-04-10 17:53:49 +0300
@@ -1250,7 +1250,7 @@
   AC_MSG_RESULT([a52dec support disabled])
 elif test "x$external_a52dec" = "xyes"; then
   have_a52="yes"
-  AC_CHECK_HEADERS([a52dec/a52.h a52dec/a52_internal.h],, have_a52="no",
+  AC_CHECK_HEADERS([a52dec/a52.h],, have_a52="no",
 [
   #ifdef HAVE_SYS_TYPES_H
   # include <sys/types.h>
diff -urN xine-lib-1.1.5.orig/src/liba52/xine_a52_decoder.c xine-lib-1.1.5/src/liba52/xine_a52_decoder.c
--- xine-lib-1.1.5.orig/src/liba52/xine_a52_decoder.c	2007-04-08 18:40:59 +0300
+++ xine-lib-1.1.5/src/liba52/xine_a52_decoder.c	2007-04-10 17:53:49 +0300
@@ -55,12 +55,6 @@
 # include "a52.h"
 #endif
 
-#ifdef HAVE_A52DEC_A52_INTERNAL_H
-# include <a52dec/a52_internal.h>
-#else
-# include "a52_internal.h"
-#endif
-
 #include "buffer.h"
 #include "xineutils.h"
 
