diff -urN xine-lib-1.1.7.orig/configure.ac xine-lib-1.1.7/configure.ac
--- xine-lib-1.1.7.orig/configure.ac	2007-06-05 21:44:38 +0300
+++ xine-lib-1.1.7/configure.ac	2007-10-06 00:27:30 +0300
@@ -1571,7 +1571,7 @@
    AS_HELP_STRING([--disable-gdkpixbuf], [do not build gdk-pixbuf support]))
 
 if test "x$enable_gdkpixbuf" != "xno"; then
-  PKG_CHECK_MODULES(GDK_PIXBUF, gdk-pixbuf-2.0,
+  PKG_CHECK_MODULES(GDK_PIXBUF, gdk-pixbuf-2.0 gobject-2.0 glib-2.0,
                 no_gdkpixbuf=no,
 		no_gdkpixbuf=yes)
   AC_SUBST(GDK_PIXBUF_CFLAGS)
