diff -Nur xine-lib-1.1.18.1-old/src/xine-engine/xine.c xine-lib-1.1.18.1/src/xine-engine/xine.c
--- xine-lib-1.1.18.1-old/src/xine-engine/xine.c	2010-04-15 17:34:34.920980153 +0300
+++ xine-lib-1.1.18.1/src/xine-engine/xine.c	2010-04-15 17:36:18.330983403 +0300
@@ -1710,7 +1710,7 @@
   _x_scan_plugins(this);
 
 #ifdef HAVE_SETLOCALE
-  if (!setlocale(LC_CTYPE, ""))
+  if (!setlocale(LC_CTYPE, "en_US.UTF-8"))
     xprintf(this, XINE_VERBOSITY_LOG, _("xine: locale not supported by C library\n"));
 #endif
 
diff -Nur xine-lib-1.1.18.1-old/src/xine-utils/utils.c xine-lib-1.1.18.1/src/xine-utils/utils.c
--- xine-lib-1.1.18.1-old/src/xine-utils/utils.c	2010-04-15 17:34:34.915980632 +0300
+++ xine-lib-1.1.18.1/src/xine-utils/utils.c	2010-04-15 17:36:04.272983486 +0300
@@ -564,7 +564,7 @@
   char *codeset = NULL;
 
 #ifdef HAVE_NL_LANGINFO
-  setlocale(LC_CTYPE, "");
+  setlocale(LC_CTYPE, "en_US.UTF-8");
   codeset = nl_langinfo(CODESET);
 #endif
   /*
