author: Fathi Boudra <fabo@debian.org>

workaround as LDFLAGS isn't honored by configure script.
Bug reported to Trolltech.

--- a/configure
+++ b/configure
@@ -650,6 +650,8 @@ mkdir -p "$outpath/config.tests"
 rm -f "$outpath/config.tests/.qmake.cache"
 cp "$QMAKE_VARS_FILE" "$outpath/config.tests/.qmake.cache"
 
+QMakeVar add QMAKE_LFLAGS -Wl,--as-needed
+
 QMakeVar add styles "cde mac motif plastique cleanlooks windows"
 QMakeVar add decorations "default windows styled"
 QMakeVar add mouse-drivers "pc"
