From ae23adec742ef61a599b11b6429dd0b56b2e0e72 Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@nokia.com>
Date: Mon, 20 Dec 2010 19:38:11 +0100
Subject: [PATCH 2/2] Change the D-Bus signal filter to return NOT_YET_HANDLED for signals

---
 src/dbus/qdbusintegrator.cpp |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

Index: qt-everywhere-opensource-src-4.7.2/src/dbus/qdbusintegrator.cpp
===================================================================
--- qt-everywhere-opensource-src-4.7.2.orig/src/dbus/qdbusintegrator.cpp	2011-02-22 12:03:59.000000000 +0000
+++ qt-everywhere-opensource-src-4.7.2/src/dbus/qdbusintegrator.cpp	2011-03-02 18:14:51.417309722 +0000
@@ -558,8 +558,9 @@
     switch (amsg.type()) {
     case QDBusMessage::SignalMessage:
         handleSignal(amsg);
-        return true;
-        break;
+        // if there are any other filters in this DBusConnection,
+        // let them see the signal too
+        return false;
     case QDBusMessage::MethodCallMessage:
         handleObjectCall(amsg);
         return true;
