Index: qt-everywhere-opensource-src-4.7.1/src/gui/itemviews/qtreeview.cpp
===================================================================
--- qt-everywhere-opensource-src-4.7.1.orig/src/gui/itemviews/qtreeview.cpp	2010-11-15 17:30:21.938553080 +0000
+++ qt-everywhere-opensource-src-4.7.1/src/gui/itemviews/qtreeview.cpp	2010-11-15 17:36:14.582553082 +0000
@@ -3439,6 +3439,10 @@
         q->doItemsLayout();
     }
 
+    if (viewItems.isEmpty()) {
+        q->doItemsLayout();
+    }
+
     int itemsInViewport = 0;
     if (uniformRowHeights) {
         if (defaultItemHeight <= 0)
