--- Makefile.in.orig	2010-07-21 22:32:03.662799405 +0300
+++ Makefile.in	2010-07-21 22:32:16.914799444 +0300
@@ -780,7 +780,7 @@
 	$(MKDIR_P) -m 2755 $(DESTDIR)$(pkglocalstatedir)
 	$(MKDIR_P) -m 2755 $(DESTDIR)$(pkglocalstatedir)/words
 #	chown root:games $(DESTDIR)$(pkglocalstatedir)
-	chown root:games $(DESTDIR)$(pkglocalstatedir)/words
+#	chown root:games $(DESTDIR)$(pkglocalstatedir)/words
 
 uninstall-hook:
 	rm -rf $(DESTDIR)$(pkglocalstatedir)/words
