--- assets/scourge.desktop.orig	2007-02-18 13:12:22.000000000 +0200
+++ assets/scourge.desktop	2007-02-18 13:20:16.000000000 +0200
@@ -2,10 +2,11 @@
 Version=0.10
 Type=Application
 Encoding=UTF-8
-Name=S.C.O.U.R.G.E.: Heroes of Lesser Renown
-Comment=A 3D, roguelike game of not quite epic proportions.
+Name=S.C.O.U.R.G.E.
+Name[tr]=S.C.O.U.R.G.E.
+GenericName=Heroes of Lesser Renown
+GenericName[tr]=Az Tanınan Kahramanlar
 TryExec=scourge
 Exec=scourge
-Categories=Game;RolePlaying
-Icon=scourge.png
-
+Categories=Applications;Game;RolePlaying
+Icon=scourge.png
\ Dosya sonunda yenisatır yok.
