Index: scourge/src/common/constants.h
===================================================================
--- scourge.orig/src/common/constants.h
+++ scourge/src/common/constants.h
@@ -82,7 +82,7 @@ typedef void (APIENTRY * PFNGLMULTITEXCO
 #define _(String) String
 #define gettext_noop(String) String
 #define N_(String) gettext_noop (String)
-#else
+#elif !defined(_)
 #include <libintl.h>
 #define _(String) gettext (String)
 #define gettext_noop(String) String
Index: scourge/src/tools/page.h
===================================================================
--- scourge.orig/src/tools/page.h
+++ scourge/src/tools/page.h
@@ -3,8 +3,9 @@
 
 #include <string>
 
+#include <wx/notebook.h>
+
 /** Forward Declarations **/
-class wxNotebook;
 class wxWindow;
 class DF;
 
