--- Makefile.in.org	2006-05-22 15:10:39.000000000 +0300
+++ Makefile.in	    2006-06-25 00:21:25.000000000 +0300
@@ -191,7 +191,7 @@
 target_alias = @target_alias@
 
 # Build, install and distribute the following subdirectories:
-SUBDIRS = admin src include examples test_suite
+SUBDIRS = admin src include test_suite
 
 # We keep our local autoconf macros in a separate 'm4'subdirectory.
 ACLOCAL_AMFLAGS = -I admin/autotools/m4
