--- xbmc-Frodo_rc3/tools/codegenerator/Helper.groovy	2013-01-03 05:29:19.000000000 +0200
+++ xbmc-Frodo_rc3/tools/codegenerator/Helper.groovy.new	2013-01-09 15:35:38.788236593 +0200
@@ -101,7 +101,8 @@
     }
 
     Node docspec = (new XmlParser().parse(doxygenXmlFile))
-    if (methodOrClass.name() == 'class')
+/* hack to make it build
+     if (methodOrClass.name() == 'class')
       doc = docspec.compounddef[0].detaileddescription[0]
     else // it's a method of some sort ... or it better be
     {
@@ -112,6 +113,8 @@
 
       doc = memberdef != null ? memberdef.detaileddescription[0] : null
     }
+*/
+    doc = null
 
     if (doc != null)
     { 
