Index: qt-everywhere-opensource-src-4.7.2/tools/qdbus/qdbusxml2cpp/qdbusxml2cpp.cpp
===================================================================
--- qt-everywhere-opensource-src-4.7.2.orig/tools/qdbus/qdbusxml2cpp/qdbusxml2cpp.cpp
+++ qt-everywhere-opensource-src-4.7.2/tools/qdbus/qdbusxml2cpp/qdbusxml2cpp.cpp
@@ -513,7 +513,7 @@ static void writeProxy(const QString &fi
     // include guards:
     QString includeGuard;
     if (!headerName.isEmpty() && headerName != QLatin1String("-")) {
-        includeGuard = headerName.toUpper().replace(QLatin1Char('.'), QLatin1Char('_'));
+        includeGuard = headerName.toUpper().replace(QLatin1Char('.'), QLatin1Char('_')).replace(QLatin1Char('-'), QLatin1Char('_'));
         int pos = includeGuard.lastIndexOf(QLatin1Char('/'));
         if (pos != -1)
             includeGuard = includeGuard.mid(pos + 1);
@@ -821,7 +821,7 @@ static void writeAdaptor(const QString &
     // include guards:
     QString includeGuard;
     if (!headerName.isEmpty() && headerName != QLatin1String("-")) {
-        includeGuard = headerName.toUpper().replace(QLatin1Char('.'), QLatin1Char('_'));
+        includeGuard = headerName.toUpper().replace(QLatin1Char('.'), QLatin1Char('_')).replace(QLatin1Char('-'), QLatin1Char('_'));
         int pos = includeGuard.lastIndexOf(QLatin1Char('/'));
         if (pos != -1)
             includeGuard = includeGuard.mid(pos + 1);
