Index: qt-everywhere-opensource-src-4.7.0-beta2/tools/assistant/tools/assistant/assistant.pro
===================================================================
--- a/tools/assistant/tools/assistant/assistant.pro
+++ a/tools/assistant/tools/assistant/assistant.pro
@@ -13,6 +13,8 @@ target.path = $$[QT_INSTALL_BINS]
 INSTALLS += target
 DEPENDPATH += ../shared
 
+LIBS += -lphonon -lQtDBus -lQtCLucene
+
 # ## Work around a qmake issue when statically linking to
 # ## not-yet-installed plugins
 QMAKE_LIBDIR += $$QT_BUILD_TREE/plugins/sqldrivers
--- a/tools/assistant/tools/qhelpgenerator/qhelpgenerator.pro.backup	2010-08-20 16:40:31.460000182 +0300
+++ a/tools/assistant/tools/qhelpgenerator/qhelpgenerator.pro	2010-08-20 16:40:41.903000402 +0300
@@ -5,6 +5,8 @@
 CONFIG -= app_bundle
 QT += network
 
+LIBS += -lQtCLucene
+
 target.path=$$[QT_INSTALL_BINS]
 INSTALLS += target
 
--- a/tools/assistant/tools/qcollectiongenerator/qcollectiongenerator.pro.backup	2010-08-20 16:42:41.020002930 +0300
+++ a/tools/assistant/tools/qcollectiongenerator/qcollectiongenerator.pro	2010-08-20 16:43:00.997997639 +0300
@@ -7,6 +7,9 @@
     warn_on \
     help \
     console
+
+LIBS += -lQtCLucene
+
 CONFIG -= app_bundle
 target.path = $$[QT_INSTALL_BINS]
 INSTALLS += target
--- a/examples/xmlpatterns/qobjectxmlmodel/qobjectxmlmodel.pro.backup	2010-08-21 01:41:27.946000766 +0300
+++ a/examples/xmlpatterns/qobjectxmlmodel/qobjectxmlmodel.pro	2010-08-21 01:42:04.249002847 +0300
@@ -5,6 +5,7 @@
 HEADERS     += qobjectxmlmodel.h  mainwindow.h ../shared/xmlsyntaxhighlighter.h
 RESOURCES   = queries.qrc
 INCLUDEPATH += ../shared/
+LIBS += -lQtWebKit -lphonon
 
 # install
 target.path = $$[QT_INSTALL_EXAMPLES]/xmlpatterns/qobjectxmlmodel
--- a/examples/help/contextsensitivehelp/contextsensitivehelp.pro.backup	2010-08-21 01:58:29.798007184 +0300
+++ a/examples/help/contextsensitivehelp/contextsensitivehelp.pro	2010-08-21 01:58:46.392000131 +0300
@@ -10,6 +10,7 @@
            helpbrowser.h
 
 FORMS   += wateringconfigdialog.ui
+LIBS += -lQtCLucene
 
 # install
 target.path = $$[QT_INSTALL_EXAMPLES]/help/contextsensitivehelp
--- a/demos/qtdemo/qtdemo.pro.backup	2010-08-21 02:01:03.626002035 +0300
+++ a/demos/qtdemo/qtdemo.pro	2010-08-21 02:01:23.134000728 +0300
@@ -5,6 +5,7 @@
 DESTDIR = $$DEMO_DESTDIR/bin
 INSTALLS += target sources
 
+LIBS += -lQtCLucene
 
 QT += xml network declarative
 
--- a/demos/browser/browser.pro.backup	2010-08-21 02:02:59.518998433 +0300
+++ a/demos/browser/browser.pro	2010-08-21 02:03:23.957999916 +0300
@@ -1,6 +1,7 @@
 TEMPLATE = app
 TARGET = browser
 QT += webkit network
+LIBS += -lphonon
 
 CONFIG += qt warn_on
 contains(QT_BUILD_PARTS, tools):!embedded: CONFIG += uitools
