diff -ur qt-x11-opensource-src-4.3.4/src/gui/text/qfont.cpp qt-x11-opensource-src-4.3.4-no-hardcoded-font-aliases/src/gui/text/qfont.cpp
--- qt-x11-opensource-src-4.3.4/src/gui/text/qfont.cpp	2008-02-19 18:47:20.000000000 +0100
+++ qt-x11-opensource-src-4.3.4-no-hardcoded-font-aliases/src/gui/text/qfont.cpp	2008-05-19 15:36:46.000000000 +0200
@@ -1433,7 +1433,7 @@
     // default substitutions
     static const char *initTbl[] = {
 
-#if defined(Q_WS_X11)
+#if 0
         "arial",        "helvetica",
         "times new roman", "times",
         "courier new",  "courier",
