Index: qt-everywhere-opensource-src-4.6.2/src/corelib/kernel/qabstractitemmodel.cpp
===================================================================
--- qt-everywhere-opensource-src-4.6.2.orig/src/corelib/kernel/qabstractitemmodel.cpp
+++ qt-everywhere-opensource-src-4.6.2/src/corelib/kernel/qabstractitemmodel.cpp
@@ -532,7 +532,8 @@ bool QAbstractItemModelPrivate::variantL
     case 1: //floating point
         return v1.toReal() < v2.toReal();
     default:
-        return v1.toString() < v2.toString();
+        return QString::localeAwareCompare(v1.toString(), v2.toString()) < 0;
+
     }
 }
 
