--- qt-x11-opensource-src-4.2.0-snapshot-20060830/tools/designer/src/uitools/uitools.pro.orig	2006-08-30 17:54:48.000000000 -0300
+++ qt-x11-opensource-src-4.2.0-snapshot-20060830/tools/designer/src/uitools/uitools.pro	2006-08-30 17:56:53.000000000 -0300
@@ -1,7 +1,7 @@
 TEMPLATE = lib
 TARGET = QtUiTools
 QT += xml
-CONFIG += qt staticlib
+CONFIG += qt dll
 DESTDIR = ../../../../lib
 DLLDESTDIR = ../../../../bin
 
