diff -Nur trigger-0.5.2-data-old/trigger.config.defs trigger-0.5.2-data/trigger.config.defs
--- trigger-0.5.2-data-old/trigger.config.defs	2007-03-13 21:48:03.951517095 +0200
+++ trigger-0.5.2-data/trigger.config.defs	2007-03-13 21:48:36.710536967 +0200
@@ -2,16 +2,16 @@
 <config>
 	
 	<video
-		width="800"
-		height="600"
+		width="1024"
+		height="768"
 		bpp="0"
 		
 		fullscreen="yes"
 		
 		requirergb="no"
-		requirealpha="no"
+		requirealpha="yes"
 		requiredepth="yes"
-		requirestencil="no"
+		requirestencil="yes"
 		
 		stereo="none"
 		stereoeyeseparation="0.07"
