diff -up mhash-0.9.9.9/lib/whirlpool.c.BAD mhash-0.9.9.9/lib/whirlpool.c
--- mhash-0.9.9.9/lib/whirlpool.c.BAD	2009-07-02 16:59:50.885279180 -0400
+++ mhash-0.9.9.9/lib/whirlpool.c	2009-07-02 17:00:12.189279257 -0400
@@ -970,6 +970,8 @@ void whirlpool_digest(__const struct whi
                     mutils_word8 * digest)
 {
    mutils_word32 i;
+
+   if(!digest) return;
     /*
      * return the completed message digest:
      */
