diff -up mhash-0.9.9.9/configure.in.fix-autotool-stomping mhash-0.9.9.9/configure.in
--- mhash-0.9.9.9/configure.in.fix-autotool-stomping	2007-04-04 22:22:28.000000000 -0400
+++ mhash-0.9.9.9/configure.in	2009-07-02 17:02:39.099044520 -0400
@@ -6,6 +6,7 @@ AC_CONFIG_SRCDIR([lib/mhash.c])
 AM_INIT_AUTOMAKE
 
 AC_DEFINE([MHASH_VERSION], PROGRAM_VERSION, "MHash Version")
+AC_CONFIG_HEADER([include/mutils/config.h])
 AC_CONFIG_HEADER([include/mutils/mhash_config.h])
 
 
diff -up /dev/null mhash-0.9.9.9/include/mutils/config.h.in
--- /dev/null	2009-07-01 18:40:45.228272777 -0400
+++ mhash-0.9.9.9/include/mutils/config.h.in	2009-07-02 17:02:39.100044508 -0400
@@ -0,0 +1,22 @@
+/* Name of package */
+#undef PACKAGE
+
+/* Define to the address where bug reports for this package should be sent. */
+#undef PACKAGE_BUGREPORT
+
+/* Define to the full name of this package. */
+#undef PACKAGE_NAME
+
+/* Define to the full name and version of this package. */
+#undef PACKAGE_STRING
+
+/* Define to the one symbol short name of this package. */
+#undef PACKAGE_TARNAME
+
+/* Define to the version of this package. */
+#undef PACKAGE_VERSION
+
+/* Version number of package */
+#undef VERSION
+
+
diff -up mhash-0.9.9.9/include/mutils/mhash_config.h.in.fix-autotool-stomping mhash-0.9.9.9/include/mutils/mhash_config.h.in
--- mhash-0.9.9.9/include/mutils/mhash_config.h.in.fix-autotool-stomping	2008-12-07 18:33:50.000000000 -0500
+++ mhash-0.9.9.9/include/mutils/mhash_config.h.in	2009-07-02 17:04:30.453049610 -0400
@@ -181,24 +181,6 @@
 /* Define to 1 if your C compiler doesn't accept -c and -o together. */
 #undef NO_MINUS_C_MINUS_O
 
-/* Name of package */
-#undef PACKAGE
-
-/* Define to the address where bug reports for this package should be sent. */
-#undef PACKAGE_BUGREPORT
-
-/* Define to the full name of this package. */
-#undef PACKAGE_NAME
-
-/* Define to the full name and version of this package. */
-#undef PACKAGE_STRING
-
-/* Define to the one symbol short name of this package. */
-#undef PACKAGE_TARNAME
-
-/* Define to the version of this package. */
-#undef PACKAGE_VERSION
-
 /* Define to 1 if the C compiler supports function prototypes. */
 #undef PROTOTYPES
 
@@ -208,9 +190,6 @@
 /* dmalloc */
 #undef USE_DMALLOC
 
-/* Version number of package */
-#undef VERSION
-
 /* Define if using the dmalloc debugging malloc package */
 #undef WITH_DMALLOC
 
