diff -up mhash-0.9.9.9/lib/snefru.c.BAD mhash-0.9.9.9/lib/snefru.c
--- mhash-0.9.9.9/lib/snefru.c.BAD	2009-07-02 16:54:58.973279449 -0400
+++ mhash-0.9.9.9/lib/snefru.c	2009-07-02 16:55:04.609279072 -0400
@@ -859,6 +859,8 @@ static void snefru_digest(__const struct
 {
 	mutils_word32 i;
 
+	if(!digest) return;
+
 	for (i = 0; i < len; i++, digest += 4)
 	{
 		*(mutils_word32 *)digest = mutils_bend2sys32(ctx->hash[i]);
