diff -up kdelibs-4.10.0/cmake/modules/FindKDE4Internal.cmake.no_rpath kdelibs-4.10.0/cmake/modules/FindKDE4Internal.cmake
--- kdelibs-4.10.0/cmake/modules/FindKDE4Internal.cmake.no_rpath	2013-01-31 07:47:15.581960702 -0600
+++ kdelibs-4.10.0/cmake/modules/FindKDE4Internal.cmake	2013-01-31 07:47:15.583960677 -0600
@@ -1050,7 +1050,7 @@ if (UNIX)
 
       set(CMAKE_SKIP_BUILD_RPATH FALSE)
       set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
-      set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+      set(CMAKE_INSTALL_RPATH_USE_LINK_PATH FALSE)
    endif (APPLE)
 endif (UNIX)
 
diff -up kdelibs-4.10.0/kdewidgets/CMakeLists.txt.no_rpath kdelibs-4.10.0/kdewidgets/CMakeLists.txt
--- kdelibs-4.10.0/kdewidgets/CMakeLists.txt.no_rpath	2013-01-31 07:47:15.556961014 -0600
+++ kdelibs-4.10.0/kdewidgets/CMakeLists.txt	2013-01-31 07:47:15.583960677 -0600
@@ -46,14 +46,14 @@ if(QT_QTDESIGNER_FOUND)
   kde4_add_plugin(kdewidgets ${kdewidgets_PART_SRCS})
 
   target_link_libraries(kdewidgets  ${KDE4_KIO_LIBS})
-  if(NOT WIN32)
-    set_target_properties(kdewidgets PROPERTIES
-                                     INSTALL_RPATH_USE_LINK_PATH TRUE
-                                     SKIP_BUILD_RPATH TRUE
-                                     BUILD_WITH_INSTALL_RPATH TRUE
-                                     INSTALL_RPATH ${LIB_INSTALL_DIR}
-                         )
-  endif(NOT WIN32)
+#  if(NOT WIN32)
+#    set_target_properties(kdewidgets PROPERTIES
+#                                     INSTALL_RPATH_USE_LINK_PATH TRUE
+#                                     SKIP_BUILD_RPATH TRUE
+#                                     BUILD_WITH_INSTALL_RPATH TRUE
+#                                     INSTALL_RPATH ${LIB_INSTALL_DIR}
+#                         )
+#  endif(NOT WIN32)
 
   install(TARGETS kdewidgets  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
 
@@ -76,14 +76,14 @@ if(QT_QTDESIGNER_FOUND)
     kde4_add_plugin(kdedeprecated ${kdedeprecated_PART_SRCS})
 
     target_link_libraries(kdedeprecated  ${KDE4_KIO_LIBS})
-    if(NOT WIN32)
-      set_target_properties(kdedeprecated PROPERTIES
-                                          INSTALL_RPATH_USE_LINK_PATH TRUE
-                                          SKIP_BUILD_RPATH TRUE
-                                          BUILD_WITH_INSTALL_RPATH TRUE
-                                          INSTALL_RPATH ${LIB_INSTALL_DIR}
-                           )
-    endif(NOT WIN32)
+#    if(NOT WIN32)
+#      set_target_properties(kdedeprecated PROPERTIES
+#                                          INSTALL_RPATH_USE_LINK_PATH TRUE
+#                                          SKIP_BUILD_RPATH TRUE
+#                                          BUILD_WITH_INSTALL_RPATH TRUE
+#                                          INSTALL_RPATH ${LIB_INSTALL_DIR}
+#                           )
+#    endif(NOT WIN32)
 
     install(TARGETS kdedeprecated  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
   endif(NOT KDE_NO_DEPRECATED)
@@ -111,14 +111,14 @@ if(QT_QTDESIGNER_FOUND)
   kde4_add_plugin(kdewebkitwidgets ${kdewebkitwidgets_PART_SRCS})
 
   target_link_libraries(kdewebkitwidgets ${KDE4_KDEUI_LIBS} ${KDE4_KDEWEBKIT_LIBS} ${QT_QTWEBKIT_LIBRARY})
-  if(NOT WIN32)
-    set_target_properties(kdewebkitwidgets PROPERTIES
-                                     INSTALL_RPATH_USE_LINK_PATH TRUE
-                                     SKIP_BUILD_RPATH TRUE
-                                     BUILD_WITH_INSTALL_RPATH TRUE
-                                     INSTALL_RPATH ${LIB_INSTALL_DIR}
-                         )
-  endif(NOT WIN32)
+#  if(NOT WIN32)
+#    set_target_properties(kdewebkitwidgets PROPERTIES
+#                                     INSTALL_RPATH_USE_LINK_PATH TRUE
+#                                     SKIP_BUILD_RPATH TRUE
+#                                     BUILD_WITH_INSTALL_RPATH TRUE
+#                                     INSTALL_RPATH ${LIB_INSTALL_DIR}
+#                         )
+#  endif(NOT WIN32)
 
   install(TARGETS kdewebkitwidgets  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
 
@@ -147,14 +147,14 @@ if(QT_QTDESIGNER_FOUND)
      kde4_add_plugin(kde3supportwidgets ${kde3supportwidgets_PART_SRCS})
 
      target_link_libraries(kde3supportwidgets ${KDE4_KDE3SUPPORT_LIBS} ${KDE4_KIO_LIBS})
-     if(NOT WIN32)
-        set_target_properties(kde3supportwidgets PROPERTIES
-                                     INSTALL_RPATH_USE_LINK_PATH TRUE
-                                     SKIP_BUILD_RPATH TRUE
-                                     BUILD_WITH_INSTALL_RPATH TRUE
-                                     INSTALL_RPATH ${LIB_INSTALL_DIR}
-                             )
-     endif(NOT WIN32)
+#     if(NOT WIN32)
+#        set_target_properties(kde3supportwidgets PROPERTIES
+#                                     INSTALL_RPATH_USE_LINK_PATH TRUE
+#                                     SKIP_BUILD_RPATH TRUE
+#                                     BUILD_WITH_INSTALL_RPATH TRUE
+#                                     INSTALL_RPATH ${LIB_INSTALL_DIR}
+#                             )
+#     endif(NOT WIN32)
 
      install(TARGETS kde3supportwidgets  DESTINATION ${PLUGIN_INSTALL_DIR}/plugins/designer )
   endif (QT_QT3SUPPORT_FOUND)
