Index: kde4libs-4.6.1/kio/bookmarks/kbookmarkmanager.cc
===================================================================
--- kde4libs-4.6.1.orig/kio/bookmarks/kbookmarkmanager.cc	2011-03-03 17:43:14.912715216 +0000
+++ kde4libs-4.6.1/kio/bookmarks/kbookmarkmanager.cc	2011-03-03 17:43:47.252715207 +0000
@@ -446,8 +446,8 @@
                              "which is most likely a full hard drive.",
                          filename, file.errorString());
 
-        if (d->m_dialogAllowed && qApp->type() != QApplication::Tty && QThread::currentThread() == qApp->thread())
-            KMessageBox::error( QApplication::activeWindow(), err );
+        //if (d->m_dialogAllowed && qApp->type() != QApplication::Tty && QThread::currentThread() == qApp->thread())
+        //    KMessageBox::error( QApplication::activeWindow(), err );
 
         kError() << QString("Unable to save bookmarks in %1. File reported the following error-code: %2.").arg(filename).arg(file.error());
         emit const_cast<KBookmarkManager*>(this)->error(err);
