diff -up boswars-2.6-src/SConstruct.orig boswars-2.6-src/SConstruct
--- boswars-2.6-src/SConstruct.orig	2010-04-11 14:13:04.000000000 +0200
+++ boswars-2.6-src/SConstruct	2010-04-16 16:06:09.693718391 +0200
@@ -265,7 +265,8 @@ addBosWarsPaths(env)
 
 # define the different build environments (variants)
 release = env.Clone()
-release.Append(CCFLAGS = Split('-O2 -pipe -fomit-frame-pointer -fexpensive-optimizations -ffast-math'))
+release.Append(CCFLAGS = Split('-ffast-math'))
+release.Append(LIBS = '-ltolua++')
 
 if mingw['extrapath']:
   mingw.Tool('crossmingw', toolpath = ['tools/scons/'])

