diff -up boswars-2.7-src/SConstruct~ boswars-2.7-src/SConstruct
--- boswars-2.7-src/SConstruct~	2013-08-03 13:56:39.000000000 +0200
+++ boswars-2.7-src/SConstruct	2013-08-03 14:10:15.589042624 +0200
@@ -170,11 +170,11 @@ def CheckOpenGL(env, conf):
 def CheckLuaLib(env, conf):
   if not 'USE_WIN32' in env['CPPDEFINES']:
      if env.WhereIs('pkg-config'):
-        for packagename in ['lua5.1', 'lua51', 'lua']:
+        for packagename in ['lua-5.1', 'lua51', 'lua']:
            exitcode,_ = ParseConfig(env, 'pkg-config --cflags --libs ' + packagename)
            if exitcode == 0:
               break
-  if conf.CheckLibWithHeader('lua51', 'lua.h', 'c'):
+  if conf.CheckLibWithHeader('lua-5.1', 'lua.h', 'c'):
     return 1
   if conf.CheckLibWithHeader('lua5.1', 'lua.h', 'c'):
     return 1

