diff -up thunderbird-31.0/comm-esr31/config/baseconfig.mk.dir thunderbird-31.0/comm-esr31/config/baseconfig.mk
--- thunderbird-31.0/comm-esr31/config/baseconfig.mk.dir	2014-07-29 13:51:58.204106461 +0200
+++ thunderbird-31.0/comm-esr31/config/baseconfig.mk	2014-07-29 13:54:26.868555504 +0200
@@ -1,6 +1,6 @@
 includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
 idldir = $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+installdir = $(libdir)/$(MOZ_APP_NAME)
 sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
 MOZILLA_SRCDIR = $(topsrcdir)/mozilla
 MOZDEPTH = $(DEPTH)/mozilla
diff -up thunderbird-31.0/comm-esr31/mozilla/config/baseconfig.mk.dir thunderbird-31.0/comm-esr31/mozilla/config/baseconfig.mk
--- thunderbird-31.0/comm-esr31/mozilla/config/baseconfig.mk.dir	2014-07-18 02:05:10.000000000 +0200
+++ thunderbird-31.0/comm-esr31/mozilla/config/baseconfig.mk	2014-07-29 13:51:58.204106461 +0200
@@ -4,7 +4,7 @@
 # whether a normal build is happening or whether the check is running.
 includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
 idldir = $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+installdir = $(libdir)/$(MOZ_APP_NAME)
 sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
 ifndef TOP_DIST
 TOP_DIST = dist
