From 2ffbca210033be262fc75e6a73742e0f0e6d5242 Mon Sep 17 00:00:00 2001
From: Marc-André Lureau <marcandre.lureau@redhat.com>
Date: Mon, 11 Apr 2011 10:44:10 +0000
Subject: controller: add missing spice/types include

---
diff --git a/spice/controller_prot.h b/spice/controller_prot.h
index 38329c5..697baa0 100644
--- a/spice/controller_prot.h
+++ b/spice/controller_prot.h
@@ -18,6 +18,8 @@
 #ifndef _H_CONTROLLER_PROT
 #define _H_CONTROLLER_PROT
 
+#include <spice/types.h>
+
 #define CONTROLLER_MAGIC      (*(uint32_t*)"CTRL")
 #define CONTROLLER_VERSION    1
 
--
cgit v0.8.3-6-g21f6
