Index: liblrdf-0.4.0/examples/Makefile.am
===================================================================
--- liblrdf-0.4.0.orig/examples/Makefile.am
+++ liblrdf-0.4.0/examples/Makefile.am
@@ -1,7 +1,9 @@
-noinst_PROGRAMS = showdefaults showtaxonomy dumpstatements export_test \
-	subclass_test add_test instances_test remove_test multi_test \
+TESTS = showtaxonomy dumpstatements export_test \
+	subclass_test add_test instances_test multi_test \
 	setting_test scale_test
 
+check_PROGRAMS = showdefaults $(TESTS) remove_test 
+
 LIBS = ../src/liblrdf.la
 
 pkgdata_DATA = ladspa.rdfs
