Index: testdisk-6.11/Makefile.am
===================================================================
--- testdisk-6.11.orig/Makefile.am
+++ testdisk-6.11/Makefile.am
@@ -15,13 +15,13 @@ AUTOMAKE_OPTIONS = gnits
 man_MANS		= doc_src/testdisk.1 doc_src/photorec.1
 
 install-data-hook:
-	$(mkinstalldirs) $(DESTDIR)$(datadir)/doc/$(PACKAGE)-$(VERSION)
-	$(mkinstalldirs) $(DESTDIR)$(datadir)/doc/$(PACKAGE)-$(VERSION)/html
-	$(INSTALL_DATA) $(top_srcdir)/AUTHORS $(top_srcdir)/ChangeLog $(top_srcdir)/NEWS $(top_srcdir)/README $(top_srcdir)/THANKS $(DESTDIR)$(datadir)/doc/$(PACKAGE)-$(VERSION)/
-	$(INSTALL_DATA) $(top_srcdir)/doc/* $(DESTDIR)$(datadir)/doc/$(PACKAGE)-$(VERSION)/html
+	$(mkinstalldirs) $(DESTDIR)$(datadir)/doc/$(PACKAGE)
+	$(mkinstalldirs) $(DESTDIR)$(datadir)/doc/$(PACKAGE)/html
+	$(INSTALL_DATA) $(top_srcdir)/AUTHORS $(top_srcdir)/ChangeLog $(top_srcdir)/NEWS $(top_srcdir)/README $(top_srcdir)/THANKS $(DESTDIR)$(datadir)/doc/$(PACKAGE)/
+	$(INSTALL_DATA) $(top_srcdir)/doc/* $(DESTDIR)$(datadir)/doc/$(PACKAGE)/html
 
 uninstall-hook:
-	rm -rf $(DESTDIR)$(datadir)/doc/$(PACKAGE)-$(VERSION)
+	rm -rf $(DESTDIR)$(datadir)/doc/$(PACKAGE)
 
 static:
 	rm -f src/testdisk src/photorec src/testdisk.exe src/photorec.exe
