diff -urNp wvstreams-4.6-orig/configure.ac wvstreams-4.6/configure.ac
--- wvstreams-4.6-orig/configure.ac	2009-05-20 20:01:18.000000000 +0200
+++ wvstreams-4.6/configure.ac	2009-06-27 18:18:52.000000000 +0200
@@ -397,7 +397,7 @@ if test "$with_dbus" != "no"; then      
     if test "$with_dbus" != "no"; then
         AC_DEFINE(WITH_DBUS,,
                   [Define to enable DBUS support.])
-        LIBS_DBUS=$DBUS_LIBDIR/libdbus-1.a
+        LIBS_DBUS=$DBUS_LIBDIR/libdbus-1.so
     fi
 fi
 
diff -urNp wvstreams-4.6-orig/configure wvstreams-4.6/configure
--- wvstreams-4.6-orig/configure	2009-06-10 18:03:36.000000000 +0200
+++ wvstreams-4.6/configure	2009-06-27 18:21:55.000000000 +0200
@@ -8351,7 +8351,7 @@ cat >>confdefs.h <<\_ACEOF
 #define WITH_DBUS /**/
 _ACEOF
 
-        LIBS_DBUS=$DBUS_LIBDIR/libdbus-1.a
+        LIBS_DBUS=$DBUS_LIBDIR/libdbus-1.so
     fi
 fi
 
