diff -up kphotoalbum-4.3/MainWindow/FeatureDialog.cpp.mplayer_warning kphotoalbum-4.3/MainWindow/FeatureDialog.cpp
--- kphotoalbum-4.3/MainWindow/FeatureDialog.cpp.mplayer_warning	2012-09-25 10:40:43.000000000 -0500
+++ kphotoalbum-4.3/MainWindow/FeatureDialog.cpp	2012-11-28 14:53:58.509642624 -0600
@@ -97,12 +97,12 @@ FeatureDialog::FeatureDialog( QWidget* p
     else
         text += i18n("<p>Phonon is capable of playing movies of these mime types:<ul><li>%1</ul></p>", mimeTypes.join(QString::fromLatin1( "<li>" ) ) );
 
-    text += i18n("<h1><a name=\"videoPreview\">Video thumbnail support</a></h1>"
-                 "<p>KPhotoAlbum uses <tt>mplayer</tt> to extract thumbnails from videos. These thumbnails are used to preview "
-                 "videos in the thumbnail viewer.</p>"
-                 "<p>If at all possible you should install the <b>mplayer2</b> package rather than the <b>mplayer</b> package, as it has important "
-                 "improvements over the mplayer package. mplayer (in contrast to mplayer2) often has problems extracting the length "
-                 "of videos and also often fails to extract the thumbnails used for cycling video thumbnails.");
+    //text += i18n("<h1><a name=\"videoPreview\">Video thumbnail support</a></h1>"
+    //             "<p>KPhotoAlbum uses <tt>mplayer</tt> to extract thumbnails from videos. These thumbnails are used to preview "
+    //             "videos in the thumbnail viewer.</p>"
+    //             "<p>If at all possible you should install the <b>mplayer2</b> package rather than the <b>mplayer</b> package, as it has important "
+    //             "improvements over the mplayer package. mplayer (in contrast to mplayer2) often has problems extracting the length "
+    //             "of videos and also often fails to extract the thumbnails used for cycling video thumbnails.");
 
     edit->setText( text );
 
diff -up kphotoalbum-4.3/MainWindow/Window.cpp.mplayer_warning kphotoalbum-4.3/MainWindow/Window.cpp
--- kphotoalbum-4.3/MainWindow/Window.cpp.mplayer_warning	2012-11-28 14:45:54.077418657 -0600
+++ kphotoalbum-4.3/MainWindow/Window.cpp	2012-11-28 14:57:49.069788142 -0600
@@ -1733,20 +1733,20 @@ void MainWindow::Window::executeStartupA
 void MainWindow::Window::checkIfMplayerIsInstalled()
 {
     if ( FeatureDialog::mplayerBinary().isNull() ) {
-        KMessageBox::error( this,
-                i18n("<p>Unable to find mplayer on the system</p>"
-                     "<p>KPhotoAlbum needs mplayer to extract video thumbnails among other things. "
-                     "Please install the mplayer2 package</p>") );
+        //KMessageBox::error( this,
+        //        i18n("<p>Unable to find mplayer on the system</p>"
+        //             "<p>KPhotoAlbum needs mplayer to extract video thumbnails among other things. "
+        //             "Please install the mplayer2 package</p>") );
         exit(-1);
     }
 
-    if ( !FeatureDialog::isMplayer2() ) {
-        KMessageBox::information( this,
-                                  i18n("<p>You have mplayer installed on your system, but it is unfortunately not version 2. "
-                                       "mplayer2 is on most systems a separate package, please install that if at all possible, "
-                                       "as that version has much better support for extracting thumbnails from videos."),
-                                  i18n("mplayer is too old"), QString::fromLatin1("mplayerVersionTooOld"));
-    }
+    //if ( !FeatureDialog::isMplayer2() ) {
+    //    KMessageBox::information( this,
+    //                              i18n("<p>You have mplayer installed on your system, but it is unfortunately not version 2. "
+    //                                   "mplayer2 is on most systems a separate package, please install that if at all possible, "
+    //                                   "as that version has much better support for extracting thumbnails from videos."),
+    //                              i18n("mplayer is too old"), QString::fromLatin1("mplayerVersionTooOld"));
+    //}
 }
 
 bool MainWindow::Window::anyVideosSelected() const 
