--- lib/utf8/utf8util.cpp.orig
+++ lib/utf8/utf8util.cpp
@@ -26,6 +26,7 @@
 ----------------------------------------------------------------------------*/
 
 #include <stdio.h>
+#include <cstdlib>
 #include "utf8util.h"
 #include "utf8.h"
 #ifdef WIN32
--- include/tunepimp-0.5/metadata.h.orig
+++ include/tunepimp-0.5/metadata.h
@@ -28,6 +28,7 @@
 #define __METADATA_H__
 
 #include <string>
+#include <cstring>
 #include <cstdio>
 
 #include "defs.h"
--- lib/metadata.cpp.orig
+++ lib/metadata.cpp
@@ -25,6 +25,7 @@
 
 ----------------------------------------------------------------------------*/
 #include <math.h>
+#include <cstdlib>
 #include <stdio.h>
 #include "metadata.h"
 #include "../config.h"
--- lib/protocol.cpp.orig
+++ lib/protocol.cpp
@@ -9,10 +9,13 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string>
+#include <cstring>
 #include <map>
 #include <expat.h>
 #include <curl/curl.h>
+#ifdef HAVE_CURL_TYPES_H
 #include <curl/types.h>
+#endif
 #include <curl/easy.h>
 using namespace std;
 
--- lib/tunepimp.cpp.orig
+++ lib/tunepimp.cpp
@@ -32,6 +32,7 @@
 
 #include <stdio.h>
 #include <map>
+#include <cstdlib>
 using namespace std;
 
 #ifdef WIN32
--- lib/c_wrapper.cpp.orig
+++ lib/c_wrapper.cpp
@@ -28,6 +28,7 @@
 #include "mutex.h"
 #include "tp_c.h"
 #include "astrcmp.h"
+#include <cstdlib>
 
 #define DB printf("%s:%d\n", __FILE__, __LINE__);
 
--- lib/readmeta.cpp.orig
+++ lib/readmeta.cpp
@@ -31,6 +31,7 @@
 #endif
 
 #include <stdio.h>
+#include <cstdlib>
 #ifndef WIN32
 #include <unistd.h>
 #endif
--- lib/fileio.cpp.orig
+++ lib/fileio.cpp
@@ -28,6 +28,8 @@
 #include <assert.h>
 #include <errno.h>
 #include <string>
+#include <cstring>
+#include <cstdlib>
 #ifndef WIN32
 #include <unistd.h>
 #endif
@@ -122,9 +124,9 @@ int taccess(const char *pathname, int mo
 
 void tmktempname(const char *path, char *newPath, int newPathLen)
 {
-    char *ptr, *temp;
+    const char *ptr; 
 
-    temp = (char *)malloc(strlen(path) + 32);
+    char *temp = (char *)malloc(strlen(path) + 32);
     ptr = strrchr(path, dirSepChar);
     if (ptr)
     {
--- plugins/wma/wmafile.cpp.orig
+++ plugins/wma/wmafile.cpp
@@ -20,6 +20,7 @@
  ***************************************************************************/
 
 #include <tstring.h>
+#include <cstring>
 #include <wmafile.h>
 #include <wmatag.h>
 #include <wmaproperties.h>
--- plugins/wma/wma.cpp.orig
+++ plugins/wma/wma.cpp
@@ -27,6 +27,7 @@
 
 #include <string.h>
 #include <stdio.h>
+#include <cstdlib>
 #include <wmafile.h>
 #include "metadata.h"
 #include "plugin.h"
--- configure.in.orig
+++ configure.in
@@ -32,7 +32,9 @@ AM_CONDITIONAL(FREEBSD, test x$os = xfre
 
 dnl Checks for programs.
 AC_PREREQ(2.52)
-AC_PROG_CC
+AC_PROG_CC_STDC
+AC_USE_SYSTEM_EXTENSIONS
+AC_SYS_LARGEFILE
 AC_LIBLTDL_CONVENIENCE
 AC_SUBST(INCLTDL)
 AC_SUBST(LIBLTDL)
@@ -87,6 +89,8 @@ AC_CHECK_LIB(curl, curl_global_init,
 	echo "*"
 	AC_MSG_ERROR("Cannot build. Stop.")], -lcurl)
 
+AC_CHECK_HEADERS([curl/types.h])
+
 AC_CHECK_LIB(ofa, ofa_get_version,,
         [
         echo "*"
