diff -Naur libtunepimp-0.5.3/configure.in libtunepimp-0.5.3.new/configure.in
--- libtunepimp-0.5.3/configure.in	2006-11-28 21:28:38.000000000 +0100
+++ libtunepimp-0.5.3.new/configure.in	2008-06-13 21:56:12.832195847 +0200
@@ -33,7 +33,7 @@
 dnl Checks for programs.
 AC_PREREQ(2.52)
 AC_PROG_CC
-AC_LIBLTDL_CONVENIENCE
+AC_WITH_LTDL
 AC_SUBST(INCLTDL)
 AC_SUBST(LIBLTDL)
 AC_LIBTOOL_DLOPEN
@@ -52,6 +52,14 @@
   prefix="/usr/local" 
 fi
 
+infodir='${prefix}/info' 
+mandir='${prefix}/man'
+
+if test "${LANG+set}"   = set; then LANG=C;   export LANG;   fi
+if test "${LC_ALL+set}" = set; then LC_ALL=C; export LC_ALL; fi
+if test "${LC_MESSAGES+set}" = set; then LC_MESSAGES=C; export LC_MESSAGES; fi
+if test "${LC_CTYPE+set}"    = set; then LC_CTYPE=C;    export LC_CTYPE;    fi
+
 AC_ARG_ENABLE([lgpl],
               [  --enable--lgpl          Use LGPL license (no mp3 support)], 
               [use_lgpl=yes], 
@@ -283,7 +291,6 @@
 AC_DEFINE_UNQUOTED(PREFIX, "${prefix}", "PREFIX") 
 AS_AC_EXPAND(LIBDIR, $libdir)
 AC_DEFINE_UNQUOTED(PLUGIN_DIR, "$LIBDIR/tunepimp/plugins", "PLUGIN_DIR") 
-AC_CONFIG_SUBDIRS(libltdl)
 AC_OUTPUT(
     Makefile
     lib/Makefile
