Index: iscan-2.26.3/iscan-data-1.8.1/policy/Makefile.am
===================================================================
--- iscan-2.26.3.orig/iscan-data-1.8.1/policy/Makefile.am
+++ iscan-2.26.3/iscan-data-1.8.1/policy/Makefile.am
@@ -22,15 +22,11 @@
 
 
 EXTRA_DIST = \
-	fdi.xsl \
-	sled10.custom.fdi \
 	initramfs-hook.in
 
 iscandatadir = $(datadir)/iscan-data
 
-dist_iscandata_DATA = \
-	sled10.custom.fdi \
-	fdi.xsl
+dist_iscandata_DATA =
 
 ##  Created by $(top_srcdir)/configure because the make-policy-file
 ##  script does not refer to any of the directory variables.
Index: iscan-2.26.3/iscan-data-1.8.1/policy/make-policy-file.in
===================================================================
--- iscan-2.26.3.orig/iscan-data-1.8.1/policy/make-policy-file.in
+++ iscan-2.26.3/iscan-data-1.8.1/policy/make-policy-file.in
@@ -429,19 +429,16 @@ case "$MODE" in
         sed -i -e "s|%ISCAN_LIBSANE_FDI_INPUT_FILE_PATH%|$TEMPLATE|g" "$OUT_FILE"
     ;;
     udev)
-        header="`get_header $TEMPLATE | sed -e 's|%|%%|g' -e 's|\(GOTO.*\)libsane_\(.*\)|\1iscan_\2|g' -e 's|\(LABEL.*\)libsane_\(.*\)|\1iscan_\2|g'`"
-        footer="`get_footer $TEMPLATE | sed -e 's|%|%%|g' -e 's|\(GOTO.*\)libsane_\(.*\)|\1iscan_\2|g' -e 's|\(LABEL.*\)libsane_\(.*\)|\1iscan_\2|g'`"
+        #header="`get_header $TEMPLATE | sed -e 's|%|%%|g' -e 's|\(GOTO.*\)libsane_\(.*\)|\1iscan_\2|g' -e 's|\(LABEL.*\)libsane_\(.*\)|\1iscan_\2|g'`"
+        #footer="`get_footer $TEMPLATE | sed -e 's|%|%%|g' -e 's|\(GOTO.*\)libsane_\(.*\)|\1iscan_\2|g' -e 's|\(LABEL.*\)libsane_\(.*\)|\1iscan_\2|g'`"
         stanza="`get_stanza $TEMPLATE | sed -e 's|%|%%|g' -e 's|\(GOTO.*\)libsane_\(.*\)|\1iscan_\2|g' -e 's|\(LABEL.*\)libsane_\(.*\)|\1iscan_\2|g'`"
 
-        printf "$header\n" >> "$OUT_FILE"
-        echo >> "$OUT_FILE"
         for usb_id in `get_usb_id "$DESC_FILE" "$USB_FILE"`; do
             printf "$stanza\n" \
                 | sed "s|\"\([^#]*\)####\"|\"\1$usb_id\"|" \
                 >> "$OUT_FILE"
         done
         echo >> "$OUT_FILE"
-        printf "$footer\n" >> "$OUT_FILE"
     ;;
     *)
         mesg "internal error: inconsistent $MODE handling" >&2
