diff -up kdebase-workspace-4.4.92/kdm/backend/client.c.xsession_errors_O_APPEND kdebase-workspace-4.4.92/kdm/backend/client.c
--- kdebase-workspace-4.4.92/kdm/backend/client.c.xsession_errors_O_APPEND	2010-07-06 01:54:30.000000000 -0500
+++ kdebase-workspace-4.4.92/kdm/backend/client.c	2010-07-07 12:27:24.901922619 -0500
@@ -1223,7 +1223,7 @@ createClientLog(const char *log)
         if (!(lname = expandMacros(log, macros)))
             exit(1);
         unlink(lname);
-        if ((lfd = open(lname, O_WRONLY|O_CREAT|O_EXCL, 0600)) >= 0) {
+        if ((lfd = open(lname, O_WRONLY|O_CREAT|O_EXCL|O_APPEND, 0600)) >= 0) {
             dup2(lfd, 1);
             dup2(lfd, 2);
             close(lfd);
