Index: glibc-2.14-394-g8f3b1ff/elf/ldd.bash.in
===================================================================
--- glibc-2.14-394-g8f3b1ff.orig/elf/ldd.bash.in
+++ glibc-2.14-394-g8f3b1ff/elf/ldd.bash.in
@@ -171,22 +171,6 @@ for file do
       fi
     done
     case $ret in
-    0)
-      # If the program exits with exit code 5, it means the process has been
-      # invoked with __libc_enable_secure.  Fall back to running it through
-      # the dynamic linker.
-      if test -x "$file"; then
-	try_trace "$file"
-      else
-	try_trace "${RTLD}" "$file"
-      fi
-      rc=$?
-      if [ $rc = 5 ]; then
-	try_trace "$RTLD" "$file"
-	rc=$?
-      fi
-      [ $rc = 0 ] || result=1
-      ;;
     1)
       # This can be a non-ELF binary or no binary at all.
       nonelf "$file" || {
