diff -Naur qca-ossl-2.0.0-beta3/qca-ossl.cpp /home/gokcen/.tmp/kde/qca-ossl/qca-ossl.cpp
--- qca-ossl-2.0.0-beta3/qca-ossl.cpp	2009-10-13 14:12:41.170719762 +0300
+++ qca-ossl/qca-ossl.cpp	2009-07-15 20:13:37.010050000 +0300
@@ -42,6 +42,15 @@
 #define OSSL_097
 #endif
 
+#if defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x10000000L
+// OpenSSL 1.0.0 makes a few changes that aren't very C++ friendly...
+// Among other things, CHECKED_PTR_OF returns a void*, but is used in
+// contexts requiring STACK pointers.
+#undef CHECKED_PTR_OF
+#define CHECKED_PTR_OF(type, p) \
+	            ((_STACK*) (1 ? p : (type*)0))
+#endif
+
 using namespace QCA;
 
 namespace opensslQCAPlugin {
@@ -327,7 +336,7 @@
 	X509V3_CTX ctx;
 	X509V3_set_ctx_nodb(&ctx);
 	X509V3_set_ctx(&ctx, NULL, cert, NULL, NULL, 0);
-	X509_EXTENSION *ex = X509V3_EXT_conf_nid(NULL, &ctx, NID_subject_key_identifier, "hash");
+	X509_EXTENSION *ex = X509V3_EXT_conf_nid(NULL, &ctx, NID_subject_key_identifier, (char *)"hash");
 	return ex;
 }
 
@@ -1182,6 +1191,7 @@
 	{
 		pkey = from.pkey;
 		CRYPTO_add(&pkey->references, 1, CRYPTO_LOCK_EVP_PKEY);
+		raw_type = false;
 		state = Idle;
 	}
 
@@ -1226,6 +1236,7 @@
 		}
 		else
 		{
+			raw_type = false;
 			EVP_MD_CTX_init(&mdctx);
 			if(!EVP_VerifyInit_ex(&mdctx, type, NULL))
 				state = VerifyError;
@@ -1775,6 +1786,14 @@
 			md = EVP_md2();
 		else if(alg == EMSA3_RIPEMD160)
 			md = EVP_ripemd160();
+		else if(alg == EMSA3_SHA224)
+			md = EVP_sha224();
+		else if(alg == EMSA3_SHA256)
+			md = EVP_sha256();
+		else if(alg == EMSA3_SHA384)
+			md = EVP_sha384();
+		else if(alg == EMSA3_SHA512)
+			md = EVP_sha512();
 		else if(alg == EMSA3_Raw)
 		{
 			// md = 0
@@ -1793,6 +1812,14 @@
 			md = EVP_md2();
 		else if(alg == EMSA3_RIPEMD160)
 			md = EVP_ripemd160();
+		else if(alg == EMSA3_SHA224)
+			md = EVP_sha224();
+		else if(alg == EMSA3_SHA256)
+			md = EVP_sha256();
+		else if(alg == EMSA3_SHA384)
+			md = EVP_sha384();
+		else if(alg == EMSA3_SHA512)
+			md = EVP_sha512();
 		else if(alg == EMSA3_Raw)
 		{
 			// md = 0
@@ -3394,6 +3421,18 @@
 		case NID_dsaWithSHA1:
 		    p.sigalgo = QCA::EMSA1_SHA1;
 		    break;
+		case NID_sha224WithRSAEncryption:
+		    p.sigalgo = QCA::EMSA3_SHA224;
+		    break;
+		case NID_sha256WithRSAEncryption:
+		    p.sigalgo = QCA::EMSA3_SHA256;
+		    break;
+		case NID_sha384WithRSAEncryption:
+		    p.sigalgo = QCA::EMSA3_SHA384;
+		    break;
+		case NID_sha512WithRSAEncryption:
+		    p.sigalgo = QCA::EMSA3_SHA512;
+		    break;
 		default:
 		    qDebug() << "Unknown signature value: " << OBJ_obj2nid(x->cert_info->signature->algorithm);
 		    p.sigalgo = QCA::SignatureUnknown;
@@ -5128,6 +5167,16 @@
 		v_eof = false;
 	}
 
+	// dummy verification function for SSL_set_verify()
+	static int ssl_verify_callback(int preverify_ok, X509_STORE_CTX *x509_ctx)
+	{
+		Q_UNUSED(preverify_ok);
+		Q_UNUSED(x509_ctx);
+
+		// don't terminate handshake in case of verification failure
+		return 1;
+	}
+
 	virtual QStringList supportedCipherSuites(const TLS::Version &version) const
 	{
 		OpenSSL_add_ssl_algorithms();
@@ -5692,6 +5741,14 @@
 			}
 		}
 
+		// request a certificate from the client, if in server mode
+		if(serv)
+		{
+			SSL_set_verify(ssl,
+				SSL_VERIFY_PEER|SSL_VERIFY_CLIENT_ONCE,
+				ssl_verify_callback);
+		}
+
 		return true;
 	}
 
@@ -6155,6 +6212,7 @@
 				i2d_PKCS7_bio(bo, p7);
 				//PEM_write_bio_PKCS7(bo, p7);
 				out = bio2ba(bo);
+				PKCS7_free(p7);
 			}
 			else
 			{
@@ -6673,7 +6731,7 @@
 	{
 	}
 
-	Context *clone() const
+	Provider::Context *clone() const
 	{
 		return new opensslInfoContext(*this);
 	}
@@ -6694,6 +6752,34 @@
 	}
 };
 
+class opensslRandomContext : public RandomContext
+{
+public:
+	opensslRandomContext(QCA::Provider *p) : RandomContext(p)
+	{
+	}
+
+	Context *clone() const
+	{
+		return new opensslRandomContext(*this);
+	}
+
+	QCA::SecureArray nextBytes(int size)
+	{
+		QCA::SecureArray buf(size);
+		int r;
+		// FIXME: loop while we don't have enough random bytes.
+		while (true) {
+			r = RAND_bytes((unsigned char*)(buf.data()), size);
+			if (r == 1) break; // success
+			r = RAND_pseudo_bytes((unsigned char*)(buf.data()),
+						size);
+			if (r >= 0) break; // accept insecure random numbers
+		}
+		return buf;
+	}
+};
+
 }
 
 using namespace opensslQCAPlugin;
@@ -6713,11 +6799,14 @@
 		OpenSSL_add_all_algorithms();
 		ERR_load_crypto_strings();
 
-		srand(time(NULL));
-		char buf[128];
-		for(int n = 0; n < 128; ++n)
-			buf[n] = rand();
-		RAND_seed(buf, 128);
+		// seed the RNG if it's not seeded yet
+		if (RAND_status() == 0) {
+			qsrand(time(NULL));
+			char buf[128];
+			for(int n = 0; n < 128; ++n)
+				buf[n] = qrand();
+			RAND_seed(buf, 128);
+		}
 
 		openssl_initted = true;
 	}
@@ -6756,6 +6845,7 @@
 	QStringList features() const
 	{
 		QStringList list;
+		list += "random";
 		list += all_hash_types();
 		list += all_mac_types();
 		list += all_cipher_types();
@@ -6782,7 +6872,9 @@
 	Context *createContext(const QString &type)
 	{
 		//OpenSSL_add_all_digests();
-		if ( type == "info" )
+		if ( type == "random" )
+			return new opensslRandomContext(this);
+		else if ( type == "info" )
 			return new opensslInfoContext(this);
 		else if ( type == "sha1" )
 			return new opensslHashContext( EVP_sha1(), this, type);
diff -Naur qca-ossl-2.0.0-beta3/qca-ossl.pro /home/gokcen/.tmp/kde/qca-ossl/qca-ossl.pro
--- qca-ossl-2.0.0-beta3/qca-ossl.pro	2007-11-03 01:06:12.000000000 +0200
+++ qca-ossl/qca-ossl.pro	2008-02-19 20:52:40.615632000 +0200
@@ -24,7 +24,11 @@
 
 	INCLUDEPATH += $$OPENSSL_PREFIX/include
 	LIBS += -L$$OPENSSL_PREFIX/lib
-	LIBS += -llibeay32 -lssleay32
+	LINKAGE = -llibeay32 -lssleay32
+	CONFIG(debug, debug|release) {
+		windows:LINKAGE = -llibeay32d -lssleay32d
+	}
+	LIBS += $$LINKAGE
 	LIBS += -lgdi32 -lwsock32
 }
 
