--- pds/environments.py.orig	2011-07-04 16:43:22.000000000 +0300
+++ pds/environments.py	2014-01-13 23:48:50.387199790 +0200
@@ -64,7 +64,7 @@
     ConfigPath           = '$HOME/.config/xfce4/'
     ConfigFile           = 'xfconf/xfce-perchannel-xml/xsettings.xml'
     ConfigType           = 'xml'
-    DefaultIconTheme     = 'hicolor'
+    DefaultIconTheme     = 'Faenza'
     DefaultConfigPath    = '/etc/xdg/xfce4/%s' % ConfigFile
     IconKey              = 'IconThemeName'
 
@@ -74,7 +74,7 @@
     ConfigPath           = '$HOME/.e/e/'
     ConfigFile           = 'config/standard/e.cfg'
     ConfigType           = 'env'
-    DefaultIconTheme     = 'hicolor'
+    DefaultIconTheme     = 'Faenza'
     IconKey              = 'E_ICON_THEME'
 
 class LXDE(DefaultDe):
@@ -83,9 +83,9 @@
     ConfigPath           = '$HOME/.config'
     ConfigFile           = ''
     ConfigType           = None
-    DefaultIconTheme     = 'oxygen'
+    DefaultIconTheme     = 'Faenza'
     IconKey              = 'theme/name'
-    DefaultIconFile      = '/usr/share/icons/lxde_pardus2011_menu.png'
+    DefaultIconFile      = '/usr/share/lxde/images/pisilinuxLogo.png'
 
 class Fluxbox(DefaultDe):
     Name                 = 'fluxbox'
@@ -93,7 +93,7 @@
     ConfigPath           = '$HOME/.config'
     ConfigFile           = ''
     ConfigType           = None
-    DefaultIconTheme     = 'oxygen'
+    DefaultIconTheme     = 'Faenza'
 
 class Gnome(DefaultDe):
     Name                 = 'gnome'
