Index: xdg-user-dirs-0.12/user-dirs.defaults
===================================================================
--- xdg-user-dirs-0.12.orig/user-dirs.defaults
+++ xdg-user-dirs-0.12/user-dirs.defaults
@@ -4,8 +4,8 @@
 # will be translated on a per-path-element basis into the users locale
 DESKTOP=Desktop
 DOWNLOAD=Downloads
-TEMPLATES=Templates
-PUBLICSHARE=Public
+#TEMPLATES=Templates
+#PUBLICSHARE=Public
 DOCUMENTS=Documents
 MUSIC=Music
 PICTURES=Pictures
