diff -Naur comar-3.0.3.orig/CMakeLists.txt comar-3.0.3/CMakeLists.txt
--- comar-3.0.3.orig/CMakeLists.txt	2014-08-19 19:56:34.057899701 +0200
+++ comar-3.0.3/CMakeLists.txt	2014-08-19 20:00:40.078890712 +0200
@@ -31,7 +31,7 @@
 
 # Define version, config dir, data dir and log file.
 ADD_DEFINITIONS (
-                 -D'PID_FILE="/var/run/${APP_NAME}.pid"'
+                 -D'PID_FILE="/run/${APP_NAME}.pid"'
                  -D'DATA_DIR="/var/lib/${APP_NAME}"'
                  -D'LOG_DIR="/var/log/${APP_NAME}"')
 
diff -Naur comar-3.0.3.orig/include/config.h comar-3.0.3/include/config.h
--- comar-3.0.3.orig/include/config.h	2011-05-12 13:30:26.000000000 +0200
+++ comar-3.0.3/include/config.h	2014-08-19 20:01:17.734889336 +0200
@@ -32,7 +32,7 @@
 #endif
 
 #ifndef DBUS_SERVER_ADDRESS
-#define DBUS_SERVER_ADDRESS "unix:path=/var/run/dbus/system_bus_socket"
+#define DBUS_SERVER_ADDRESS "unix:path=/run/dbus/system_bus_socket"
 #endif
 
 #ifndef DBUS_SERVICE_NAME
@@ -56,7 +56,7 @@
 #endif
 
 #ifndef FILE_PID
-#define FILE_PID "/var/run/comar3.pid"
+#define FILE_PID "/run/comar3.pid"
 #endif
 
 #ifndef WWW_BUGS
