Index: zaz-1.0.0/po/Makefile.in.in
===================================================================
--- zaz-1.0.0.orig/po/Makefile.in.in
+++ zaz-1.0.0/po/Makefile.in.in
@@ -8,8 +8,8 @@
 # Please note that the actual code of GNU gettext is covered by the GNU
 # General Public License and is *not* in the public domain.
 #
-# Origin: gettext-0.17
-GETTEXT_MACRO_VERSION = 0.17
+# Origin: gettext-0.18
+GETTEXT_MACRO_VERSION = 0.18
 
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
