--- mate-vfs-1.2.1/schemas/desktop_mate_url_handlers.schemas.in.orig	2012-03-26 02:14:16.000000000 +0300
+++ mate-vfs-1.2.1/schemas/desktop_mate_url_handlers.schemas.in	2012-07-23 01:46:15.146258793 +0300
@@ -153,7 +153,7 @@
       <applyto>/desktop/mate/url-handlers/http/command</applyto>
       <owner>mate</owner>
       <type>string</type>
-      <default>epiphany %s</default>
+      <default>firefox %s</default>
       <locale name="C">
         <short>The handler for "http" URLs</short>
         <long>The command used to handle "http" URLs, if enabled.</long>
@@ -187,7 +187,7 @@
       <applyto>/desktop/mate/url-handlers/https/command</applyto>
       <owner>mate</owner>
       <type>string</type>
-      <default>epiphany %s</default>
+      <default>firefox %s</default>
       <locale name="C">
         <short>The handler for "https" URLs</short>
         <long>The command used to handle "https" URLs, if enabled.</long>
