--- mate-vfs-1.2.1/schemas/desktop_mate_url_handlers.schemas.in.orig	2012-03-26 02:14:16.000000000 +0300
+++ mate-vfs-1.2.1/schemas/desktop_mate_url_handlers.schemas.in	2012-07-23 01:49:37.710238052 +0300
@@ -221,7 +221,7 @@
       <applyto>/desktop/mate/url-handlers/mailto/command</applyto>
       <owner>mate</owner>
       <type>string</type>
-      <default>evolution %s</default>
+      <default>evolution --component=mail %s</default>
       <locale name="C">
         <short>The handler for "mailto" URLs</short>
         <long>The command used to handle "mailto" URLs, if enabled.</long>
