diff -urN desmume-0.9.10/src/gtk/glx_3Demu.cpp desmume-0.9.10-patched/src/gtk/glx_3Demu.cpp
--- desmume-0.9.10/src/gtk/glx_3Demu.cpp	2013-11-28 11:36:53.000000000 +1100
+++ desmume-0.9.10-patched/src/gtk/glx_3Demu.cpp	2013-11-30 00:22:57.000000000 +1100
@@ -41,11 +41,7 @@
         glXDestroyContext(dpy, ctx);
 
         XCloseDisplay(dpy);
-
-        return true;
     }
-
-    return false;
 }
 
 int init_glx_3Demu(void) 