Index: mscore/packaging/mscore.desktop
===================================================================
--- mscore.orig/packaging/mscore.desktop
+++ mscore/packaging/mscore.desktop
@@ -1,14 +1,14 @@
 [Desktop Entry]
 Version=1.0
-Comment=WYSIWYG Music score typesetter
-Comment[ru]=Визуальный редактор нотных партитур
+Comment=Compose music in a cost-effective way
+Comment[tr]=Hesaplı bir şekilde besteler yapın
 Exec=mscore
-GenericName=mscore
-GenericName[ru]=MuseScore
+GenericName=Music Notation Program
+GenericName[tr]=Müzik Notasyon Programı
 Icon=mscore
 Name=MuseScore
 StartupNotify=true
 Terminal=false
 Type=Application
-Categories=Qt;Audio;Sequencer;Midi;AudioVideoEditing;Music;AudioVideo;
+Categories=Qt;Audio;Sequencer;Midi;AudioVideoEditing;Music;AudioVideo;Education;
 MimeType=application/x-musescore;application/x-musescore+xml;application/vnd.recordare.musicxml;application/vnd.recordare.musicxml+xml;
