Index: mscore/mscore/preferences.cpp
===================================================================
--- mscore.orig/mscore/preferences.cpp
+++ mscore/mscore/preferences.cpp
@@ -221,7 +221,7 @@ void Preferences::init()
       importCharset           = "GBK";
 
       //update
-      checkUpdateStartup      = 0;
+      checkUpdateStartup      = -1;
       
       firstStartWeb = true;
       };
@@ -438,7 +438,7 @@ void Preferences::read()
       
       firstStartWeb = s.value("firstStartWeb", true).toBool();
 
-      checkUpdateStartup = s.value("checkUpdateStartup", UpdateChecker::defaultPeriod()).toInt();
+      checkUpdateStartup = s.value("checkUpdateStartup", -1).toInt();
       if (checkUpdateStartup == 0){
           checkUpdateStartup = UpdateChecker::defaultPeriod();
       }
