--- mscore.old/CMakeLists.txt	2011-04-18 20:42:53.730340051 -0300
+++ mscore/CMakeLists.txt	2011-04-18 20:43:04.982963551 -0300
@@ -31,7 +31,7 @@
 set(GCC_VISIBILITY TRUE)                  # requires gcc 4.x?
 set(HAS_AUDIOFILE TRUE)                   # requires libsndfile
 
-set(BUILD_SCRIPTGEN TRUE)                 #Generate Qt script bindings. (collides with qtscriptgenerator project on several distrib)
+set(BUILD_SCRIPTGEN FALSE)                #Generate Qt script bindings. (collides with qtscriptgenerator project on several distrib)
 set(STATIC_SCRIPT_BINDINGS FALSE)
 set(USE_SSE FALSE)
