--- cmus-v2.4.3/ffmpeg.c~	2011-11-30 16:57:20.000000000 +0800
+++ cmus-v2.4.3/ffmpeg.c	2013-07-15 17:56:10.302927483 +0800
@@ -107,7 +107,7 @@
 {
 	struct ffmpeg_output *output = xnew(struct ffmpeg_output, 1);
 
-	output->buffer_malloc = xnew(uint8_t, AVCODEC_MAX_AUDIO_FRAME_SIZE + 15);
+	output->buffer_malloc = xnew(uint8_t, 192000 + 15);
 	output->buffer = output->buffer_malloc;
 	/* align to 16 bytes so avcodec can SSE/Altivec/etc */
 	while ((intptr_t) output->buffer % 16)
@@ -309,7 +309,7 @@
 	while (1) {
 		/* frame_size specifies the size of output->buffer for
 		 * avcodec_decode_audio2. */
-		int frame_size = AVCODEC_MAX_AUDIO_FRAME_SIZE;
+		int frame_size = 192000;
 		int len;
 
 		if (input->curr_pkt_size <= 0) {
