Index: cmus-v2.4.1/Makefile
===================================================================
--- cmus-v2.4.1.orig/Makefile
+++ cmus-v2.4.1/Makefile
@@ -230,6 +230,7 @@ install-data: man
 	$(INSTALL) -m644 $(mandir)/man1 $(man1)
 	$(INSTALL) -m644 $(mandir)/man7 $(man7)
 	$(INSTALL) -m755 $(exampledir) cmus-status-display
+	$(INSTALL) -m755 $(exampledir) last-cmus.py
 
 install: all install-main install-plugins install-data
 
