--- a/doc/Doxyfile.in~  2002-11-23 19:23:00.000000000 -0500
+++ b/doc/Doxyfile.in   2002-11-25 03:58:14.000000000 -0500
@@ -263,8 +263,10 @@
 
 INPUT                  = ../src \
                          ../include/id3/field.h \
-                         ../include/id3/frame.h \
                          ../include/id3/globals.h \
+                        ../include/id3/io_decorators.h \
+                        ../include/id3/io_helpers.h \
+                        ../include/id3/io_strings.h \
                          ../include/id3/misc_support.h \
                          ../include/id3/reader.h \
                          ../include/id3/readers.h \
