Index: id3lib-3.8.3/src/Makefile.in
===================================================================
--- id3lib-3.8.3.orig/src/Makefile.in
+++ id3lib-3.8.3/src/Makefile.in
@@ -192,7 +192,7 @@ CONFIG_HEADER = $(top_builddir)/config.h
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(lib_LTLIBRARIES)
 
-libid3_la_LIBADD =
+libid3_la_LIBADD = -lstdc++ -lz
 am__objects_1 = c_wrapper.lo field.lo field_binary.lo field_integer.lo \
 	field_string_ascii.lo field_string_unicode.lo frame.lo \
 	frame_impl.lo frame_parse.lo frame_render.lo globals.lo \
