--- init.c.orig	2007-11-16 19:52:10.000000000 +0200
+++ init.c	2007-11-27 08:43:53.000000000 +0200
@@ -119,11 +119,11 @@
 int item_window_on_drop=1;
 int compass_direction=1;
 int buddy_log_notice=1;
-char configdir[256]="./";
+char configdir[256]="/usr/share/eternal-lands";
 #ifdef DATA_DIR
 char datadir[256]=DATA_DIR;
 #else
-char datadir[256]="./";
+char datadir[256]="/usr/share/eternal-lands";
 #endif //DATA_DIR
 
 char lang[10] = "en";
