diff -Naur csmash-0.6.6.orig/images/Makefile.am csmash-0.6.6/images/Makefile.am
--- csmash-0.6.6.orig/images/Makefile.am	2003-08-02 19:45:57.000000000 +0300
+++ csmash-0.6.6/images/Makefile.am	2011-04-24 16:44:43.866000032 +0300
@@ -1,4 +1,4 @@
-pkgdatadir = $(datadir)/games/@PACKAGE@/images
+pkgdatadir = $(datadir)/@PACKAGE@/images
 
 pkgdata_ppm = Bad.ppm Keyboard.ppm Nice.ppm Title.ppm \
 		eight.ppm five.ppm four.ppm nine.ppm one.ppm seven.ppm \
diff -Naur csmash-0.6.6.orig/images/Makefile.in csmash-0.6.6/images/Makefile.in
--- csmash-0.6.6.orig/images/Makefile.in	2003-08-16 08:55:01.000000000 +0300
+++ csmash-0.6.6/images/Makefile.in	2011-04-24 16:47:11.655000031 +0300
@@ -35,7 +35,7 @@
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
 host_triplet = @host@
-pkgdatadir = $(datadir)/games/@PACKAGE@/images
+pkgdatadir = $(datadir)/@PACKAGE@/images
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur csmash-0.6.6.orig/Makefile.am csmash-0.6.6/Makefile.am
--- csmash-0.6.6.orig/Makefile.am	2003-01-10 17:02:49.000000000 +0200
+++ csmash-0.6.6/Makefile.am	2011-04-24 16:44:29.116000033 +0300
@@ -1,4 +1,4 @@
-pkgdatadir = $(datadir)/games/@PACKAGE@
+pkgdatadir = $(datadir)/@PACKAGE@
 localedir = $(prefix)/share/locale
 
 bin_PROGRAMS = csmash
diff -Naur csmash-0.6.6.orig/Makefile.in csmash-0.6.6/Makefile.in
--- csmash-0.6.6.orig/Makefile.in	2003-08-16 08:54:30.000000000 +0300
+++ csmash-0.6.6/Makefile.in	2011-04-24 16:47:01.384000032 +0300
@@ -35,7 +35,7 @@
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
 host_triplet = @host@
-pkgdatadir = $(datadir)/games/@PACKAGE@
+pkgdatadir = $(datadir)/@PACKAGE@
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur csmash-0.6.6.orig/Parts/Bcut/Makefile.am csmash-0.6.6/Parts/Bcut/Makefile.am
--- csmash-0.6.6.orig/Parts/Bcut/Makefile.am	2003-08-02 19:08:39.000000000 +0300
+++ csmash-0.6.6/Parts/Bcut/Makefile.am	2011-04-24 16:45:55.767000033 +0300
@@ -1,4 +1,4 @@
-pkgdatadir = $(datadir)/games/@PACKAGE@/Parts/Bcut
+pkgdatadir = $(datadir)/@PACKAGE@/Parts/Bcut
 
 pkgdata_src = Bcut-Lankle.affine Bcut-Lankle01.dat \
 		Bcut-Larm.affine Bcut-Larm01.dat \
diff -Naur csmash-0.6.6.orig/Parts/Bcut/Makefile.in csmash-0.6.6/Parts/Bcut/Makefile.in
--- csmash-0.6.6.orig/Parts/Bcut/Makefile.in	2003-08-16 08:54:53.000000000 +0300
+++ csmash-0.6.6/Parts/Bcut/Makefile.in	2011-04-24 16:48:18.335000033 +0300
@@ -35,7 +35,7 @@
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
 host_triplet = @host@
-pkgdatadir = $(datadir)/games/@PACKAGE@/Parts/Bcut
+pkgdatadir = $(datadir)/@PACKAGE@/Parts/Bcut
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur csmash-0.6.6.orig/Parts/Bnormal/Makefile.am csmash-0.6.6/Parts/Bnormal/Makefile.am
--- csmash-0.6.6.orig/Parts/Bnormal/Makefile.am	2003-08-02 19:09:01.000000000 +0300
+++ csmash-0.6.6/Parts/Bnormal/Makefile.am	2011-04-24 16:46:09.607000031 +0300
@@ -1,4 +1,4 @@
-pkgdatadir = $(datadir)/games/@PACKAGE@/Parts/Bnormal
+pkgdatadir = $(datadir)/@PACKAGE@/Parts/Bnormal
 
 pkgdata_src = Bnormal-Lankle.affine Bnormal-Lankle01.dat \
 		Bnormal-Larm.affine Bnormal-Larm01.dat \
diff -Naur csmash-0.6.6.orig/Parts/Bnormal/Makefile.in csmash-0.6.6/Parts/Bnormal/Makefile.in
--- csmash-0.6.6.orig/Parts/Bnormal/Makefile.in	2003-08-16 08:54:54.000000000 +0300
+++ csmash-0.6.6/Parts/Bnormal/Makefile.in	2011-04-24 16:48:26.175000033 +0300
@@ -35,7 +35,7 @@
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
 host_triplet = @host@
-pkgdatadir = $(datadir)/games/@PACKAGE@/Parts/Bnormal
+pkgdatadir = $(datadir)/@PACKAGE@/Parts/Bnormal
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur csmash-0.6.6.orig/Parts/Bpeck/Makefile.am csmash-0.6.6/Parts/Bpeck/Makefile.am
--- csmash-0.6.6.orig/Parts/Bpeck/Makefile.am	2003-08-02 19:09:13.000000000 +0300
+++ csmash-0.6.6/Parts/Bpeck/Makefile.am	2011-04-24 16:46:35.476000031 +0300
@@ -1,4 +1,4 @@
-pkgdatadir = $(datadir)/games/@PACKAGE@/Parts/Bpeck
+pkgdatadir = $(datadir)/@PACKAGE@/Parts/Bpeck
 
 pkgdata_src = Bpeck-Larm.affine Bpeck-Larm01.dat \
 		Bpeck-Lelbow.affine Bpeck-Lelbow01.dat \
diff -Naur csmash-0.6.6.orig/Parts/Bpeck/Makefile.in csmash-0.6.6/Parts/Bpeck/Makefile.in
--- csmash-0.6.6.orig/Parts/Bpeck/Makefile.in	2003-08-16 08:54:55.000000000 +0300
+++ csmash-0.6.6/Parts/Bpeck/Makefile.in	2011-04-24 16:48:47.595000032 +0300
@@ -35,7 +35,7 @@
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
 host_triplet = @host@
-pkgdatadir = $(datadir)/games/@PACKAGE@/Parts/Bpeck
+pkgdatadir = $(datadir)/@PACKAGE@/Parts/Bpeck
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur csmash-0.6.6.orig/Parts/Fcut/Makefile.am csmash-0.6.6/Parts/Fcut/Makefile.am
--- csmash-0.6.6.orig/Parts/Fcut/Makefile.am	2003-08-02 19:10:00.000000000 +0300
+++ csmash-0.6.6/Parts/Fcut/Makefile.am	2011-04-24 16:46:44.987000033 +0300
@@ -1,4 +1,4 @@
-pkgdatadir = $(datadir)/games/@PACKAGE@/Parts/Fcut
+pkgdatadir = $(datadir)/@PACKAGE@/Parts/Fcut
 
 pkgdata_src = Fcut-Lankle.affine Fcut-Lankle01.dat \
 		Fcut-Larm.affine Fcut-Larm01.dat \
diff -Naur csmash-0.6.6.orig/Parts/Fcut/Makefile.in csmash-0.6.6/Parts/Fcut/Makefile.in
--- csmash-0.6.6.orig/Parts/Fcut/Makefile.in	2003-08-16 08:54:56.000000000 +0300
+++ csmash-0.6.6/Parts/Fcut/Makefile.in	2011-04-24 16:48:56.250000033 +0300
@@ -35,7 +35,7 @@
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
 host_triplet = @host@
-pkgdatadir = $(datadir)/games/@PACKAGE@/Parts/Fcut
+pkgdatadir = $(datadir)/@PACKAGE@/Parts/Fcut
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur csmash-0.6.6.orig/Parts/Fdrive/Makefile.am csmash-0.6.6/Parts/Fdrive/Makefile.am
--- csmash-0.6.6.orig/Parts/Fdrive/Makefile.am	2003-08-02 19:10:08.000000000 +0300
+++ csmash-0.6.6/Parts/Fdrive/Makefile.am	2011-04-24 16:45:25.777000033 +0300
@@ -1,4 +1,4 @@
-pkgdatadir = $(datadir)/games/@PACKAGE@/Parts/Fdrive
+pkgdatadir = $(datadir)/@PACKAGE@/Parts/Fdrive
 
 pkgdata_src = Fdrive-Lankle.affine Fdrive-Lankle01.dat \
 		Fdrive-Larm.affine Fdrive-Larm01.dat \
diff -Naur csmash-0.6.6.orig/Parts/Fdrive/Makefile.in csmash-0.6.6/Parts/Fdrive/Makefile.in
--- csmash-0.6.6.orig/Parts/Fdrive/Makefile.in	2003-08-16 08:54:57.000000000 +0300
+++ csmash-0.6.6/Parts/Fdrive/Makefile.in	2011-04-24 16:47:44.404000031 +0300
@@ -35,7 +35,7 @@
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
 host_triplet = @host@
-pkgdatadir = $(datadir)/games/@PACKAGE@/Parts/Fdrive
+pkgdatadir = $(datadir)/@PACKAGE@/Parts/Fdrive
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur csmash-0.6.6.orig/Parts/Fnormal/Makefile.am csmash-0.6.6/Parts/Fnormal/Makefile.am
--- csmash-0.6.6.orig/Parts/Fnormal/Makefile.am	2003-08-02 19:10:19.000000000 +0300
+++ csmash-0.6.6/Parts/Fnormal/Makefile.am	2011-04-24 16:46:23.287000033 +0300
@@ -1,4 +1,4 @@
-pkgdatadir = $(datadir)/games/@PACKAGE@/Parts/Fnormal
+pkgdatadir = $(datadir)/@PACKAGE@/Parts/Fnormal
 
 pkgdata_src = Fnormal-Lankle.affine Fnormal-Lankle01.dat \
 		Fnormal-Larm.affine Fnormal-Larm01.dat \
diff -Naur csmash-0.6.6.orig/Parts/Fnormal/Makefile.in csmash-0.6.6/Parts/Fnormal/Makefile.in
--- csmash-0.6.6.orig/Parts/Fnormal/Makefile.in	2003-08-16 08:54:58.000000000 +0300
+++ csmash-0.6.6/Parts/Fnormal/Makefile.in	2011-04-24 16:48:35.514000033 +0300
@@ -35,7 +35,7 @@
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
 host_triplet = @host@
-pkgdatadir = $(datadir)/games/@PACKAGE@/Parts/Fnormal
+pkgdatadir = $(datadir)/@PACKAGE@/Parts/Fnormal
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur csmash-0.6.6.orig/Parts/Fpeck/Makefile.am csmash-0.6.6/Parts/Fpeck/Makefile.am
--- csmash-0.6.6.orig/Parts/Fpeck/Makefile.am	2003-08-02 19:10:31.000000000 +0300
+++ csmash-0.6.6/Parts/Fpeck/Makefile.am	2011-04-24 16:45:37.265000033 +0300
@@ -1,4 +1,4 @@
-pkgdatadir = $(datadir)/games/@PACKAGE@/Parts/Fpeck
+pkgdatadir = $(datadir)/@PACKAGE@/Parts/Fpeck
 
 pkgdata_src = Fpeck-Lankle.affine Fpeck-Lankle01.dat \
 		Fpeck-Larm.affine Fpeck-Larm01.dat \
diff -Naur csmash-0.6.6.orig/Parts/Fpeck/Makefile.in csmash-0.6.6/Parts/Fpeck/Makefile.in
--- csmash-0.6.6.orig/Parts/Fpeck/Makefile.in	2003-08-16 08:54:59.000000000 +0300
+++ csmash-0.6.6/Parts/Fpeck/Makefile.in	2011-04-24 16:47:58.955000031 +0300
@@ -35,7 +35,7 @@
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
 host_triplet = @host@
-pkgdatadir = $(datadir)/games/@PACKAGE@/Parts/Fpeck
+pkgdatadir = $(datadir)/@PACKAGE@/Parts/Fpeck
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur csmash-0.6.6.orig/Parts/Fsmash/Makefile.am csmash-0.6.6/Parts/Fsmash/Makefile.am
--- csmash-0.6.6.orig/Parts/Fsmash/Makefile.am	2003-08-02 19:10:38.000000000 +0300
+++ csmash-0.6.6/Parts/Fsmash/Makefile.am	2011-04-24 16:45:46.323000032 +0300
@@ -1,4 +1,4 @@
-pkgdatadir = $(datadir)/games/@PACKAGE@/Parts/Fsmash
+pkgdatadir = $(datadir)/@PACKAGE@/Parts/Fsmash
 
 pkgdata_src = Fsmash-Larm.affine Fsmash-Larm01.dat \
 		Fsmash-Lelbow.affine Fsmash-Lelbow01.dat \
diff -Naur csmash-0.6.6.orig/Parts/Fsmash/Makefile.in csmash-0.6.6/Parts/Fsmash/Makefile.in
--- csmash-0.6.6.orig/Parts/Fsmash/Makefile.in	2003-08-16 08:55:00.000000000 +0300
+++ csmash-0.6.6/Parts/Fsmash/Makefile.in	2011-04-24 16:48:08.295000032 +0300
@@ -35,7 +35,7 @@
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
 host_triplet = @host@
-pkgdatadir = $(datadir)/games/@PACKAGE@/Parts/Fsmash
+pkgdatadir = $(datadir)/@PACKAGE@/Parts/Fsmash
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur csmash-0.6.6.orig/wav/Makefile.am csmash-0.6.6/wav/Makefile.am
--- csmash-0.6.6.orig/wav/Makefile.am	2002-01-21 19:31:27.000000000 +0200
+++ csmash-0.6.6/wav/Makefile.am	2011-04-24 16:44:54.764000033 +0300
@@ -1,3 +1,3 @@
-pkgdatadir = $(datadir)/games/@PACKAGE@/wav
+pkgdatadir = $(datadir)/@PACKAGE@/wav
 pkgdata_DATA = $(wildcard *.wav)
 EXTRA_DIST = $(pkgdata_DATA)
diff -Naur csmash-0.6.6.orig/wav/Makefile.in csmash-0.6.6/wav/Makefile.in
--- csmash-0.6.6.orig/wav/Makefile.in	2003-08-16 08:55:02.000000000 +0300
+++ csmash-0.6.6/wav/Makefile.in	2011-04-24 16:47:23.856000033 +0300
@@ -35,7 +35,7 @@
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
 host_triplet = @host@
-pkgdatadir = $(datadir)/games/@PACKAGE@/wav
+pkgdatadir = $(datadir)/@PACKAGE@/wav
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naur csmash-0.6.6.orig/win32/Makefile.am csmash-0.6.6/win32/Makefile.am
--- csmash-0.6.6.orig/win32/Makefile.am	2002-01-21 19:31:28.000000000 +0200
+++ csmash-0.6.6/win32/Makefile.am	2011-04-24 16:45:12.984000032 +0300
@@ -1,3 +1,3 @@
-#pkgdatadir = $(datadir)/games/@PACKAGE@/win32
+#pkgdatadir = $(datadir)/@PACKAGE@/win32
 #pkgdata_DATA = GetArgs.cpp GetArgs.h Makefile.nt getopt.c getopt.h minwin32.h wsaerror.cpp wsaerror.h wsainit.cpp csmash.rc kuro.ico orange.ico scream.ico resource.h
 EXTRA_DIST = GetArgs.cpp GetArgs.h Makefile.nt getopt.c getopt.h minwin32.h wsaerror.cpp wsaerror.h wsainit.cpp csmash.rc kuro.ico orange.ico scream.ico resource.h new_exception.cpp
diff -Naur csmash-0.6.6.orig/win32/Makefile.in csmash-0.6.6/win32/Makefile.in
--- csmash-0.6.6.orig/win32/Makefile.in	2003-08-16 08:55:03.000000000 +0300
+++ csmash-0.6.6/win32/Makefile.in	2011-04-24 16:47:34.136000032 +0300
@@ -143,7 +143,7 @@
 target_os = @target_os@
 target_vendor = @target_vendor@
 
-#pkgdatadir = $(datadir)/games/@PACKAGE@/win32
+#pkgdatadir = $(datadir)/@PACKAGE@/win32
 #pkgdata_DATA = GetArgs.cpp GetArgs.h Makefile.nt getopt.c getopt.h minwin32.h wsaerror.cpp wsaerror.h wsainit.cpp csmash.rc kuro.ico orange.ico scream.ico resource.h
 EXTRA_DIST = GetArgs.cpp GetArgs.h Makefile.nt getopt.c getopt.h minwin32.h wsaerror.cpp wsaerror.h wsainit.cpp csmash.rc kuro.ico orange.ico scream.ico resource.h new_exception.cpp
 subdir = win32
