Index: obexftp-0.24-Source/swig/python/CMakeLists.txt
===================================================================
--- obexftp-0.24-Source.orig/swig/python/CMakeLists.txt
+++ obexftp-0.24-Source/swig/python/CMakeLists.txt
@@ -36,13 +36,13 @@ swig_link_libraries ( obexftp-python
 #
 execute_process (
   COMMAND ${PYTHON_EXECUTABLE} -c
-  	"import site, sys; sys.stdout.write(site.PREFIXES[-1])"
+  	"import site, sys; sys.stdout.write(site.PREFIXES[0])"
   OUTPUT_VARIABLE PYTHON_PREFIX
 )
 file ( TO_CMAKE_PATH "${PYTHON_PREFIX}" PYTHON_PREFIX )
 execute_process (
   COMMAND ${PYTHON_EXECUTABLE} -c
-  	"import site, sys; sys.stdout.write(site.getsitepackages()[-1])"
+  	"import site, sys; sys.stdout.write(site.getsitepackages()[0])"
   OUTPUT_VARIABLE PYTHON_SITE_DIR
 )
 file ( TO_CMAKE_PATH "${PYTHON_SITE_DIR}" PYTHON_SITE_DIR )
