--- obexftp-0.22/swig/perl/Makefile.am.p	2006-10-22 15:31:38.000000000 +0200
+++ obexftp-0.22/swig/perl/Makefile.am	2007-03-23 00:39:29.000000000 +0100
@@ -9,7 +9,7 @@
 endif
 
 Makefile.perl: Makefile.PL
-	$(PERL) $< PREFIX=$(prefix) MAKEFILE=$@
+	$(PERL) $< PREFIX=$(prefix) MAKEFILE=$@ INSTALLDIRS="vendor"
 
 OBEXFTP.so: perl_wrap.c Makefile.perl
 	if test ! -f perl_wrap.c; then cp $(srcdir)/perl_wrap.c . ; fi
