Index: qbittorrent-2.6.4/src/mainwindow.cpp
===================================================================
--- qbittorrent-2.6.4.orig/src/mainwindow.cpp
+++ qbittorrent-2.6.4/src/mainwindow.cpp
@@ -428,7 +428,7 @@ void MainWindow::on_actionDocumentation_
 }
 
 void MainWindow::on_actionBugReport_triggered() const {
-  QDesktopServices::openUrl(QUrl(QString::fromUtf8("http://bugs.qbittorrent.org")));
+  QDesktopServices::openUrl(QUrl(QString::fromUtf8("http://bugs.pardus.org.tr")));
 }
 
 void MainWindow::tab_changed(int new_tab) {
