diff -Nuar qbittorrent-2.9.0.orig/src/preferences/preferences.h qbittorrent-2.9.0/src/preferences/preferences.h
--- qbittorrent-2.9.0.orig/src/preferences/preferences.h	2011-10-21 13:32:49.200000094 +0300
+++ qbittorrent-2.9.0/src/preferences/preferences.h	2011-10-21 13:55:39.479000104 +0300
@@ -144,7 +144,7 @@
   }
 
   bool closeToTray() const {
-    return value(QString::fromUtf8("Preferences/General/CloseToTray"), false).toBool();
+    return value(QString::fromUtf8("Preferences/General/CloseToTray"), true).toBool();
   }
 
   void setCloseToTray(bool b) {
@@ -160,7 +160,7 @@
   }
 
   bool isSlashScreenDisabled() const {
-    return value(QString::fromUtf8("Preferences/General/NoSplashScreen"), false).toBool();
+    return value(QString::fromUtf8("Preferences/General/NoSplashScreen"), true).toBool();
   }
 
   void setSplashScreenDisabled(bool b) {
@@ -1236,7 +1236,7 @@
   }
 
   TrayIcon::Style trayIconStyle() const {
-    return TrayIcon::Style(value(QString::fromUtf8("Preferences/Advanced/TrayIconStyle"), TrayIcon::NORMAL).toInt());
+    return TrayIcon::Style(value(QString::fromUtf8("Preferences/Advanced/TrayIconStyle"), TrayIcon::MONO_DARK).toInt());
   }
   void setTrayIconStyle(TrayIcon::Style style) {
     setValue(QString::fromUtf8("Preferences/Advanced/TrayIconStyle"), style);
