--- src/lang/qbittorrent_tr.ts.orig	2012-08-09 13:40:07.000000000 +0300
+++ src/lang/qbittorrent_tr.ts	2012-08-15 23:44:46.162203704 +0300
@@ -49,7 +49,7 @@
     </message>
     <message>
         <source>chris@qbittorrent.org</source>
-        <translation></translation>
+        <translation>chris@qbittorrent.org</translation>
     </message>
     <message>
         <source>Thanks to</source>
@@ -260,11 +260,11 @@
     </message>
     <message>
         <source>Disk write cache size</source>
-        <translation>Yazılabilir disk ön arabellek boyutu</translation>
+        <translation>Kullanılacak önbellek boyutu</translation>
     </message>
     <message>
         <source> MiB</source>
-        <translation>MB</translation>
+        <translation>MiB</translation>
     </message>
     <message>
         <source>Outgoing ports (Min) [0: Disabled]</source>
@@ -276,7 +276,7 @@
     </message>
     <message>
         <source>Recheck torrents on completion</source>
-        <translation>Tamamlanmış torentleri yeniden denetle</translation>
+        <translation>Tamamlanmış torrentleri yeniden denetle</translation>
     </message>
     <message>
         <source>Transfer list refresh interval</source>
@@ -285,12 +285,12 @@
     <message>
         <source> ms</source>
         <comment> milliseconds</comment>
-        <translatorcomment>milisaniye</translatorcomment>
-        <translation>ms</translation>
+        <translatorcomment> milisaniye</translatorcomment>
+        <translation> ms</translation>
     </message>
     <message>
         <source>Resolve peer countries (GeoIP)</source>
-        <translation>Eş ülkelerini çözümle (GeoIP)</translation>
+        <translation>Eş ülkeleri çözümle (GeoIP)</translation>
     </message>
     <message>
         <source>Resolve peer host names</source>
@@ -302,7 +302,7 @@
     </message>
     <message>
         <source>Strict super seeding</source>
-        <translation>Süper gönderme kipi</translation>
+        <translation>Süper gönderimi kısıtla</translation>
     </message>
     <message>
         <source>Network Interface (requires restart)</source>
@@ -311,7 +311,7 @@
     <message>
         <source>Any interface</source>
         <comment>i.e. Any network interface</comment>
-        <translatorcomment>örn. Herhangi bir ağ arayüzü</translatorcomment>
+        <translatorcomment>Örn. Herhangi bir ağ arayüzü</translatorcomment>
         <translation>Herhangi bir arayüz</translation>
     </message>
     <message>
@@ -1293,7 +1293,7 @@
     <message>
         <source>qBittorrent is not the default application to open torrent files or Magnet links.
 Do you want to associate qBittorrent to torrent files and Magnet links?</source>
-        <translation type="obsolete">qBittorrent torent dosyaranı ve benzer bağlantıları açmak için varsayılan uygulama değil.
+        <translation type="obsolete">qBittorrent torrent dosyalarını ve magnet bağlantıları açmak için varsayılan uygulama değil.
 qBittorrent&apos;u bunlarla ilişkilendirmek ister misiniz?</translation>
     </message>
     <message>
@@ -1610,7 +1610,7 @@
     </message>
     <message>
         <source>Edit</source>
-        <translation>Düzenle</translation>
+        <translation>Düzen</translation>
     </message>
     <message>
         <source>Help</source>
@@ -1622,7 +1622,7 @@
     </message>
     <message>
         <source>Download Torrents from their URL or Magnet link</source>
-        <translation>Torrentleri bir adresten ya da ilgili bağlantıdan indir</translation>
+        <translation>Torrentleri bir adresten ya da magnet bağlantıdan indir</translation>
     </message>
     <message>
         <source>Only one link per line</source>
@@ -1745,9 +1745,9 @@
         <source>qBittorrent is a file sharing program. When you run a torrent, its data will be made available to others by means of upload. Any content you share is your sole responsibility.
 
 No further notices will be issued.</source>
-        <translation>qBittorrent bir dosya paylaşım programıdır. Bir torent çalıştırdığınızda başkaları tarafından kopyalanabilecektir. Paylaştığınız tüm içerik sizin sorumluluğunuzdadır.
-
-Başka bir bildiri yayınlanmayacaktır.</translation>
+        <translation>qBittorrent bir dosya paylaşım programıdır. Bir torrent çalıştırdığınızda başkaları tarafından kopyalanabilecektir. Paylaştığınız tüm içerik sizin sorumluluğunuzdadır.
+ 
+ Başka bir bildiri yayınlanmayacaktır.</translation>
     </message>
     <message>
         <source>Press %1 key to accept and continue...</source>
@@ -1824,7 +1824,7 @@
     </message>
     <message>
         <source>Torrent &amp;creator</source>
-        <translation>Torent &amp;oluşturucu</translation>
+        <translation>Torrent &amp;oluşturucu</translation>
     </message>
     <message>
         <source>Set upload limit...</source>
@@ -1953,12 +1953,12 @@
     </message>
     <message>
         <source>Torrent file association</source>
-        <translation>Torrent dosyası ilişkisi</translation>
+        <translation type="unfinished">Torrent dosyası ilişkisi</translation>
     </message>
     <message>
         <source>qBittorrent is not the default application to open torrent files or Magnet links.
 Do you want to associate qBittorrent to torrent files and Magnet links?</source>
-        <translation>qBittorrent torrent dosyalarını ve benzer bağlantıları açmak için varsayılan uygulama değil.
+        <translation type="unfinished">qBittorrent torrent dosyalarını ve benzer bağlantıları açmak için varsayılan uygulama değil.
 qBittorrent&apos;u bunlarla ilişkilendirmek ister misiniz?</translation>
     </message>
     <message>
@@ -2037,7 +2037,7 @@
     </message>
     <message>
         <source>The torrent %1 contains torrent files, do you want to proceed with their download?</source>
-        <translation>%1 , torrent dosyaları içeriyor, bunların indirilmesini birlikte yürütmek istiyor musunuz?</translation>
+        <translation type="unfinished">%1 , torrent dosyaları içeriyor, bunların indirilmesini birlikte yürütmek istiyor musunuz?</translation>
     </message>
     <message>
         <source>Yes</source>
@@ -2372,7 +2372,7 @@
     </message>
     <message>
         <source>Torrent queueing</source>
-        <translation type="obsolete">Torent kuyruğu</translation>
+        <translation type="obsolete">Torrent kuyruğu</translation>
     </message>
     <message>
         <source>Maximum active downloads:</source>
@@ -2384,15 +2384,15 @@
     </message>
     <message>
         <source>Maximum active torrents:</source>
-        <translation>Azami etkin torent:</translation>
+        <translation>Azami etkin torrent:</translation>
     </message>
     <message>
         <source>When adding a torrent</source>
-        <translation>Bir torent eklerken</translation>
+        <translation>Bir torrent eklerken</translation>
     </message>
     <message>
         <source>Display torrent content and some options</source>
-        <translation>Torent içeriğini ve bazı seçenekleri göster</translation>
+        <translation>Torrent içeriğini ve bazı seçenekleri göster</translation>
     </message>
     <message>
         <source>Listening port</source>
@@ -2424,11 +2424,11 @@
     </message>
     <message>
         <source>Maximum number of connections per torrent:</source>
-        <translation>Torent başına azami bağlantı sayısı:</translation>
+        <translation>Torrent başına azami bağlantı sayısı:</translation>
     </message>
     <message>
         <source>Maximum number of upload slots per torrent:</source>
-        <translation>Torent başına azami gönderme yuvası sayısı:</translation>
+        <translation>Torrent başına azami gönderme yuvası sayısı:</translation>
     </message>
     <message>
         <source>Upload:</source>
@@ -2594,7 +2594,7 @@
     </message>
     <message>
         <source>Downloading torrents:</source>
-        <translation>Torent indiriliyor:</translation>
+        <translation>Torrent indiriliyor:</translation>
     </message>
     <message>
         <source>Start / Stop</source>
@@ -2606,7 +2606,7 @@
     </message>
     <message>
         <source>Completed torrents:</source>
-        <translation>Tamamlanan torentler:</translation>
+        <translation>Tamamlanan torrentler:</translation>
     </message>
     <message>
         <source>Desktop</source>
@@ -2637,7 +2637,7 @@
     <message>
         <source>Do not start the download automatically</source>
         <comment>The torrent will be added to download list in pause state</comment>
-        <translatorcomment>Torent, indirme listesine duraklatılmış durumda eklenir</translatorcomment>
+        <translatorcomment>Torrent, indirme listesine duraklatılmış durumda eklenir</translatorcomment>
         <translation>İndirmeyi kendiliğinden olarak başlatma</translation>
     </message>
     <message>
@@ -2654,7 +2654,7 @@
     </message>
     <message>
         <source>Keep incomplete torrents in:</source>
-        <translation>Tamamlanmamış torentlerin tutulacağı yer:</translation>
+        <translation>Tamamlanmamış torrentlerin tutulacağı yer:</translation>
     </message>
     <message>
         <source>Append .!qB extension to incomplete files&apos; names</source>
@@ -2662,7 +2662,7 @@
     </message>
     <message>
         <source>Automatically add torrents from:</source>
-        <translation>Torentlerin kendiliğinden ekleneceği yer:</translation>
+        <translation>Torrentlerin kendiliğinden ekleneceği yer:</translation>
     </message>
     <message>
         <source>Add folder...</source>
@@ -2703,7 +2703,7 @@
     </message>
     <message>
         <source>Seed torrents until their ratio reaches</source>
-        <translation>Torentleri paylaşım oranlarına ulaşıncaya kadar gönder</translation>
+        <translation>Torrentleri paylaşım oranlarına ulaşıncaya kadar gönder</translation>
     </message>
     <message>
         <source>then</source>
@@ -3290,11 +3290,11 @@
     </message>
     <message>
         <source>Removing torrent %1...</source>
-        <translation>Torrent kaldırılıyor: %1...</translation>
+        <translation type="unfinished">Torrent kaldırılıyor: %1...</translation>
     </message>
     <message>
         <source>Pausing torrent %1...</source>
-        <translation>Torrent duraklatılıyor: %1...</translation>
+        <translation type="unfinished">Torrent duraklatılıyor: %1...</translation>
     </message>
     <message>
         <source>qBittorrent is bound to port: TCP/%1</source>
@@ -3397,7 +3397,7 @@
     </message>
     <message>
         <source>&apos;%1&apos; is not a valid magnet URI.</source>
-        <translation>&apos;%1&apos; geçerli bir adres değil.</translation>
+        <translation type="unfinished">&apos;%1&apos; geçerli magnet adresi değil.</translation>
     </message>
     <message>
         <source>&apos;%1&apos; is already in download list.</source>
@@ -3425,15 +3425,15 @@
     </message>
     <message>
         <source>Error: The torrent %1 does not contain any file.</source>
-        <translation>Hata: %1 torrenti herhangi bir dosya içermiyor.</translation>
+        <translation type="unfinished">Hata: %1 torrenti herhangi bir dosya içermiyor.</translation>
     </message>
     <message>
         <source>Note: new trackers were added to the existing torrent.</source>
-        <translation>Not: yeni izleyiciler varolan torrente eklendi.</translation>
+        <translation type="unfinished">Not: yeni izleyiciler varolan torrente eklendi.</translation>
     </message>
     <message>
         <source>Note: new URL seeds were added to the existing torrent.</source>
-        <translation>Not: yeni URL eşleri varolan torrente eklendi.</translation>
+        <translation type="unfinished">Not: yeni URL eşleri varolan torrente eklendi.</translation>
     </message>
     <message>
         <source>&lt;font color=&apos;red&apos;&gt;%1&lt;/font&gt; &lt;i&gt;was blocked due to your IP filter&lt;/i&gt;</source>
@@ -3448,11 +3448,11 @@
     <message>
         <source>Recursive download of file %1 embedded in torrent %2</source>
         <comment>Recursive download of test.torrent embedded in torrent test2</comment>
-        <translation>%1 dosyasının özyineli indirmesi %2 torenti içine gömülü</translation>
+        <translation type="unfinished">%1 dosyasının özyineli indirmesi %2 torrenti içine gömülü</translation>
     </message>
     <message>
         <source>Unable to decode %1 torrent file.</source>
-        <translation>%1 torrent dosyası çözümlenemiyor.</translation>
+        <translation type="unfinished">%1 torrent dosyası çözümlenemiyor.</translation>
     </message>
     <message>
         <source>Torrent name: %1</source>
@@ -3497,7 +3497,7 @@
     </message>
     <message>
         <source>File sizes mismatch for torrent %1, pausing it.</source>
-        <translation>%1 torrentinin dosya boyutu eşleşmiyor, duraklatılıyor.</translation>
+        <translation type="unfinished">%1 torrentinin dosya boyutu eşleşmiyor, duraklatılıyor.</translation>
     </message>
     <message>
         <source>Fast resume data was rejected for torrent %1, checking again...</source>
@@ -3602,7 +3602,7 @@
     </message>
     <message>
         <source>Download torrent</source>
-        <translation>Torenti indir</translation>
+        <translation>Torrenti indir</translation>
     </message>
     <message>
         <source>Open news URL</source>
@@ -3772,7 +3772,7 @@
     <name>RssFeed</name>
     <message>
         <source>Automatically downloading %1 torrent from %2 RSS feed...</source>
-        <translation>%2 beslemesinden %1 torrent kendiliğinden indiriliyor...</translation>
+        <translation type="unfinished">%2 beslemesinden %1 torrent kendiliğinden indiriliyor...</translation>
     </message>
 </context>
 <context>
@@ -4610,7 +4610,7 @@
     </message>
     <message>
         <source>Copy magnet link</source>
-        <translation>Çeken bağlantıyı kopyala</translation>
+        <translation>Magnet bağlantıyı kopyala</translation>
     </message>
     <message>
         <source>Down Speed</source>
@@ -4862,7 +4862,7 @@
     </message>
     <message>
         <source>[files or urls]: downloads the torrents passed by the user (optional)</source>
-        <translation>[dosya ya da url]: kullanıcı tarafından seçilen torentleri indirir (seçime bağlı)</translation>
+        <translation>[dosya ya da url]: kullanıcı tarafından seçilen torrentleri indirir (seçime bağlı)</translation>
     </message>
 </context>
 <context>
@@ -4939,7 +4939,7 @@
     </message>
     <message>
         <source>Torrent size:</source>
-        <translation type="obsolete">Torent boyutu:</translation>
+        <translation>Torrent boyutu:</translation>
     </message>
     <message>
         <source>Unknown</source>
@@ -5811,7 +5811,7 @@
     <message>
         <source>(%1 left after torrent download)</source>
         <comment>e.g. (100MiB left after torrent download)</comment>
-        <translation type="obsolete">(torentin indirilmesinden sonra %1 kalacak)</translation>
+        <translation>(torrentin indirilmesinden sonra %1 kalacak)</translation>
     </message>
     <message>
         <source>(%1 more are required to download)</source>
@@ -5852,11 +5852,11 @@
     </message>
     <message>
         <source>Unable to decode magnet link:</source>
-        <translation type="obsolete">İlişik bağlantı kodu çözümlenemedi:</translation>
+        <translation>Magnet bağlantı kodu çözümlenemedi:</translation>
     </message>
     <message>
         <source>Magnet Link</source>
-        <translation type="obsolete">İlişik Bağlantı</translation>
+        <translation>Magnet Bağlantı</translation>
     </message>
     <message>
         <source>Invalid label name</source>
