Index: src/sql/Makefile.am
===================================================================
--- src/sql/Makefile.am.orig
+++ src/sql/Makefile.am
@@ -1,5 +1,6 @@
 lib_LTLIBRARIES	=	libmdbsql.la
 libmdbsql_la_SOURCES=	mdbsql.c parser.y lexer.l 
+libmdbsql_la_LIBADD = ../libmdb/libmdb.la 
 EXTRA_DIST = main.c
 DISTCLEANFILES = parser.c parser.h lexer.c
 AM_CPPFLAGS	=	-I$(top_srcdir)/include $(GLIB_CFLAGS)
