Index: wxcam-1.0.7/src/setting.cpp
===================================================================
--- wxcam-1.0.7.orig/src/setting.cpp
+++ wxcam-1.0.7/src/setting.cpp
@@ -24,6 +24,9 @@
 #include <wx/filename.h>
 #include <wx/dir.h>
 
+/* for using xdg-dirs Desktop path */
+#include <cstdio>
+
 using namespace std;
 
 Setting* Setting::setting = NULL;
@@ -481,13 +484,21 @@ void Setting::WriteDefaults()
 {
     wxMutexLocker mLocker( mutex );
     wxString s;
+    char buffer[128];
+    FILE *xdg_user_dir_func = popen("xdg-user-dir DESKTOP", "r");
+    if(!xdg_user_dir_func){
+        return;
+    }
+    char *xdg_desktop_dir = fgets(buffer, sizeof(buffer), xdg_user_dir_func);
+    pclose(xdg_user_dir_func);
+
     cfile->writeValueReplace( "Device:", "/dev/video0" );
     cfile->writeValueReplace( "AudioDevice:", "/dev/dsp" );
     cfile->writeValueReplace( "DriverAPI:", "AUTO" );
     cfile->writeValueReplace( "Format:", "AUTO" );
-    s = wxFileName::GetHomeDir() + wxString("/Desktop/image.tif", wxConvUTF8);
+    s = wxString(xdg_desktop_dir, wxConvUTF8).Trim() + wxString("/image.tif", wxConvUTF8);
     cfile->writeValueReplace( "Snapshot:", (const char*)s.mb_str(wxConvUTF8) );
-    s = wxFileName::GetHomeDir() + wxString("/Desktop/video.avi", wxConvUTF8);
+    s = wxString(xdg_desktop_dir, wxConvUTF8).Trim() + wxString("/video.avi", wxConvUTF8);
     cfile->writeValueReplace( "Video:", (const char*)s.mb_str(wxConvUTF8) );
     cfile->writeValueReplace( "Resolution:", "160x120" );
     cfile->writeValueReplace( "Frame_Rate:", "15" );
