--- make-3.80/doc/make.texi.gfortran	2002-09-18 22:51:58.000000000 +0200
+++ make-3.80/doc/make.texi	2005-09-09 16:36:54.000000000 +0200
@@ -7634,7 +7634,7 @@ with the command @samp{$(PC) -c $(PFLAGS
 @item Compiling Fortran and Ratfor programs
 @cindex Fortran, rule to compile
 @cindex Ratfor, rule to compile
-@pindex f77
+@pindex gfortran 
 @pindex .f
 @pindex .r
 @pindex .F
--- make-3.80/default.c.gfortran	2000-03-26 08:56:54.000000000 +0200
+++ make-3.80/default.c	2005-09-09 16:34:53.000000000 +0200
@@ -414,7 +414,7 @@ static char *default_variables[] =
 #ifdef	__convex__
     "FC", "fc",
 #else
-    "FC", "f77",
+    "FC", "gfortran",
 #endif /* __convex__ */
 #endif /* _IBMR2 */
     /* System V uses these, so explicit rules using them should work.
