Index: mscore/mscore/CMakeLists.txt
===================================================================
--- mscore.orig/mscore/CMakeLists.txt
+++ mscore/mscore/CMakeLists.txt
@@ -32,7 +32,6 @@ if (NOT MINGW)
 #     set library search path for runtime linker to load the same
 #     qt libraries as we used at compile time
 #
-set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wl,-rpath,${QT_LIBRARY_DIR}")
 endif (NOT MINGW)
 
 
--- mscore.orig/CMakeLists.txt
+++ mscore/CMakeLists.txt
@@ -56,7 +56,7 @@ if (APPLE)
 endif (APPLE)
 
 set(CMAKE_INCLUDE_CURRENT_DIR TRUE)
-set(CMAKE_BUILD_WITH_INSTALL_RPATH ON)
+set(CMAKE_BUILD_WITH_INSTALL_RPATH OFF)
 set(CMAKE_SKIP_RULE_DEPENDENCY TRUE)
 
 # The Mscore version number.
