--- gui2/gretl.c.orig	2008-02-23 19:21:06.000000000 +0200
+++ gui2/gretl.c	2008-02-29 23:39:05.000000000 +0200
@@ -141,13 +141,13 @@
 char viewdvi[MAXSTR] = "xdvi";
 char Rcommand[MAXSTR] = "xterm -e R";
 #else
-char Browser[MAXSTR] = "mozilla";
+char Browser[MAXSTR] = "xdg-open";
 char calculator[MAXSTR] = "xcalc";
 char latex[MAXSTR] = "latex";
 char viewdvi[MAXSTR] = "xdvi";
-char viewpdf[MAXSTR] = "acroread";
-char viewps[MAXSTR] = "gv";
-char Rcommand[MAXSTR] = "xterm -e R";
+char viewpdf[MAXSTR] = "xdg-open";
+char viewps[MAXSTR] = "xdg-open";
+char Rcommand[MAXSTR] = "xterm -e R";
 #endif
 
 static void spreadsheet_edit (gpointer p, guint u, GtkWidget *w) 
