diff -Nur gfxtheme-pardus-0.11.3-old/data-boot/languages gfxtheme-pardus-0.11.3/data-boot/languages
--- gfxtheme-pardus-0.11.3-old/data-boot/languages	2010-12-29 18:21:52.440554688 +0200
+++ gfxtheme-pardus-0.11.3/data-boot/languages	2010-12-29 18:22:17.889682362 +0200
@@ -3,9 +3,11 @@
 en_US
 es_ES
 fr_FR
+hu_HU
 it_IT
 nl_NL
 pl_PL
 pt_BR
+ru_RU
 sv_SE
 tr_TR
diff -Nur gfxtheme-pardus-0.11.3-old/data-install/languages gfxtheme-pardus-0.11.3/data-install/languages
--- gfxtheme-pardus-0.11.3-old/data-install/languages	2010-12-29 18:21:52.429681233 +0200
+++ gfxtheme-pardus-0.11.3/data-install/languages	2010-12-29 18:22:32.876682628 +0200
@@ -3,9 +3,11 @@
 en_US
 es_ES
 fr_FR
+hu_HU
 it_IT
 nl_NL
 pl_PL
 pt_BR
+ru_RU
 sv_SE
 tr_TR
diff -Nur gfxtheme-pardus-0.11.3-old/Makefile gfxtheme-pardus-0.11.3/Makefile
--- gfxtheme-pardus-0.11.3-old/Makefile	2010-12-29 18:21:52.440554688 +0200
+++ gfxtheme-pardus-0.11.3/Makefile	2010-12-29 18:25:30.277564772 +0200
@@ -14,7 +14,7 @@
 SUBDIRS      = fonts help-boot help-install po src
 
 DEFAULT_LANG = tr_TR
-DEFAULT_LANGUAGES = ca de en es fr it nl pl pt_BR sv tr
+DEFAULT_LANGUAGES = ca de en es fr hu it nl pl pt_BR ru sv tr
 
 .PHONY: all clean distclean themes $(SUBDIRS)
 
diff -Nur gfxtheme-pardus-0.11.3-old/src/common.inc gfxtheme-pardus-0.11.3/src/common.inc
--- gfxtheme-pardus-0.11.3-old/src/common.inc	2010-12-29 18:21:52.437554584 +0200
+++ gfxtheme-pardus-0.11.3/src/common.inc	2010-12-29 18:32:42.292682387 +0200
@@ -417,6 +417,10 @@
           "mudur=language:fr " bc.cmd dup length add sprintf
         } if
 
+        config.lang "hu_HU" eq {
+          "mudur=language:hu " bc.cmd dup length add sprintf
+        } if
+
         config.lang "it_IT" eq {
           "mudur=language:it " bc.cmd dup length add sprintf
         } if
@@ -433,6 +437,10 @@
           "mudur=language:pt_BR " bc.cmd dup length add sprintf
         } if
 
+        config.lang "ru_RU" eq {
+          "mudur=language:ru " bc.cmd dup length add sprintf
+        } if
+
         config.lang "sv_SE" eq {
           "mudur=language:sv " bc.cmd dup length add sprintf
         } if
@@ -473,6 +481,10 @@
             "mudur=language:fr " bc.cmd dup length add sprintf
           } if
 
+          config.lang "hu_HU" eq {
+            "mudur=language:hu " bc.cmd dup length add sprintf
+          } if
+
           config.lang "it_IT" eq {
             "mudur=language:it " bc.cmd dup length add sprintf
           } if
@@ -489,6 +501,10 @@
             "mudur=language:pt_BR " bc.cmd dup length add sprintf
           } if
 
+          config.lang "ru_RU" eq {
+            "mudur=language:ru " bc.cmd dup length add sprintf
+          } if
+
           config.lang "sv_SE" eq {
             "mudur=language:sv " bc.cmd dup length add sprintf
           } if
