diff -Nur gfxtheme-pardus-0.11.3-old/src/common.inc gfxtheme-pardus-0.11.3/src/common.inc
--- gfxtheme-pardus-0.11.3-old/src/common.inc	2011-01-18 12:00:15.042246134 +0200
+++ gfxtheme-pardus-0.11.3/src/common.inc	2011-01-18 12:04:01.714270474 +0200
@@ -346,6 +346,12 @@
     } { pop } ifelse
   } if
 
+  xmenu.kerneloptsinstall {
+    kerneloptsinstall.options xmenu.kerneloptsinstall .xm_current get get dup "" ne {
+      "%s " bc.cmd dup length add sprintf
+    } { pop } ifelse
+  } if
+
   xmenu.zorg {
     zorg.options xmenu.zorg .xm_current get get dup "" ne {
       "%s " bc.cmd dup length add sprintf
diff -Nur gfxtheme-pardus-0.11.3-old/src/dia_kerneloptsinstall.inc gfxtheme-pardus-0.11.3/src/dia_kerneloptsinstall.inc
--- gfxtheme-pardus-0.11.3-old/src/dia_kerneloptsinstall.inc	1970-01-01 02:00:00.000000000 +0200
+++ gfxtheme-pardus-0.11.3/src/dia_kerneloptsinstall.inc	2011-01-18 12:07:17.848248435 +0200
@@ -0,0 +1,98 @@
+% - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+%
+% Kernel options dialog.
+%
+% - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+
+% - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+% Some global vars.
+%
+/kerneloptsinstall.default 0 def
+
+/kerneloptsinstall.options [
+  "splash quiet"
+  ""
+  "apm=off acpi=off mce=off barrier=off ide=nodma idewait=50 i8042.nomux psmouse.proto=bare irqpoll pci=nommconf edd=off"
+  "acpi=off"
+  "nolapic"
+  "bootmem_debug debug debugpat dynamic_printk earlyprintk=vga initcall_debug loglevel=7 mminit_loglevel=4 pnp.debug sched_debug apic=debug"
+] def
+
+
+% - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+% Build kerneloptsinstall list.
+%
+% ( ) ==> ( )
+%
+/kerneloptsinstall.init {
+  /xmenu.kerneloptsinstall .xm_size array def
+
+  /xmenu xmenu.kerneloptsinstall def
+
+  xmenu .xm_current kerneloptsinstall.default put
+  xmenu .xm_list [
+    /txt_kernel_default
+    /txt_zorg_off
+    /txt_kernel_safe
+    /txt_kernel_noacpi
+    /txt_kernel_nolapic
+    /txt_kernel_debug
+  ] put
+  xmenu .xm_title /txt_kernel put
+
+} def
+
+
+% - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+% Update kerneloptsinstall mode.
+%
+% ( ) ==> ( )
+%
+/kerneloptsinstall.update {
+  /xmenu xmenu.kerneloptsinstall def
+
+  /window.action actRedrawPanel def
+} def
+
+
+% - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+% Show kerneloptsinstall menu.
+%
+% ( ) => ( )
+%
+/panel.kerneloptsinstall {
+  "main" help.setcontext
+
+  window.xmenu
+  dup .xmenu xmenu.kerneloptsinstall put
+  dup .xmenu.update /kerneloptsinstall.update put
+  dup window.init
+      window.show
+} def
+
+
+% - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+% Return width of panel entry.
+%
+% ( ) => ( width )
+%
+/panel.kerneloptsinstall.width {
+  /xmenu xmenu.kerneloptsinstall def
+
+  pmenu.width
+} def
+
+
+% - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+% Redraw panel entry.
+%
+% ( panel ) => ( )
+%
+/panel.kerneloptsinstall.update {
+  /xmenu xmenu.kerneloptsinstall def
+
+  pmenu.panel.update
+} def
+
+
diff -Nur gfxtheme-pardus-0.11.3-old/src/main.bc gfxtheme-pardus-0.11.3/src/main.bc
--- gfxtheme-pardus-0.11.3-old/src/main.bc	2011-01-18 12:00:15.043246032 +0200
+++ gfxtheme-pardus-0.11.3/src/main.bc	2011-01-18 12:04:13.010310900 +0200
@@ -19,6 +19,7 @@
 %% include dia_bits.inc
 %% include dia_about.inc
 %% include dia_kernelopts.inc
+%% include dia_kerneloptsinstall.inc
 %% include dia_zorg.inc
 %% include dia_mudur.inc
 %% include panel.inc
diff -Nur gfxtheme-pardus-0.11.3-old/src/panel.inc gfxtheme-pardus-0.11.3/src/panel.inc
--- gfxtheme-pardus-0.11.3-old/src/panel.inc	2011-01-18 12:00:15.043246032 +0200
+++ gfxtheme-pardus-0.11.3/src/panel.inc	2011-01-18 12:05:37.334309992 +0200
@@ -52,7 +52,7 @@
         %[ keyF3 0 "F3" /panel.video /panel.video.width /panel.video.update /video.init ]
         % [ keyF4 0 "F4" /panel.install /panel.install.width /panel.install.update /install.init ]
         [ keyF4 0 "F4" /panel.zorg /panel.zorg.width /panel.zorg.update /zorg.init ]
-        [ keyF5 0 "F5" /panel.kernelopts /panel.kernelopts.width /panel.kernelopts.update /kernelopts.init ]
+        [ keyF5 0 "F5" /panel.kerneloptsinstall /panel.kerneloptsinstall.width /panel.kerneloptsinstall.update /kerneloptsinstall.init ]
         % [ keyF6 0 "F6" /panel.dud /panel.dud.width /panel.dud.update /dud.init ]
 
         %32+64bit_source {
