diff -ur trousers-0.3.6.orig/configure trousers-0.3.6/configure
--- trousers-0.3.6.orig/configure	2011-04-08 09:09:38.000000000 -0400
+++ trousers-0.3.6/configure	2011-04-08 09:41:49.000000000 -0400
@@ -14379,7 +14379,7 @@
 
 
 if test "x${GCC}" = "xyes"; then
-	CFLAGS="$CFLAGS -W -Wall -Werror -Wno-unused-parameter -Wsign-compare"
+	CFLAGS="$CFLAGS -W -Wall -Wno-unused-parameter -Wsign-compare"
 fi
 
 CFLAGS="$CFLAGS -I../include \
