diff -up normalize-0.7.7/configure.ac.automake-1_13 normalize-0.7.7/configure.ac
--- normalize-0.7.7/configure.ac.automake-1_13	2013-01-06 16:14:40.292124401 +0100
+++ normalize-0.7.7/configure.ac	2013-01-06 16:15:55.907413877 +0100
@@ -8,7 +8,7 @@ AC_CONFIG_AUX_DIR(config)
 AC_CONFIG_FILES(config/Makefile)
 
 AM_INIT_AUTOMAKE
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_CANONICAL_HOST
 AC_PROG_CC
