--- old-python-bugzilla-0.7.0/bin/bugzilla	2012-06-07 01:41:50.000000000 +0300
+++ python-bugzilla-0.7.0/bin//bugzilla	2012-11-11 15:34:52.734000001 +0200
@@ -22,7 +22,7 @@
 from tempfile import NamedTemporaryFile
 
 version = '0.7.0'
-default_bz = 'https://bugzilla.redhat.com/xmlrpc.cgi'
+default_bz = 'https://bugzilla.pardus.org.tr/xmlrpc.cgi'
 
 # Initial simple logging stuff
 logging.basicConfig()
@@ -490,9 +490,8 @@
     if global_opt.bztype == 'auto':
         log.info('Autodetecting Bugzilla type')
         # Cheat a little, for the sake of speed
-        if 'bugzilla.redhat.com' in global_opt.bugzilla:
-            log.info('Using RHBugzilla4 for URL containing bugzilla.redhat.com')
-            bzclass = bugzilla.RHBugzilla4
+        if 'pardus.org.tr' in global_opt.bugzilla:
+            bzclass = bugzilla.Bugzilla36
         else:
             bzclass = bugzilla.Bugzilla
     elif global_opt.bztype in bugzilla.classlist:
