Index: python-bugzilla-0.6.2/bin/bugzilla
===================================================================
--- python-bugzilla-0.6.2.orig/bin/bugzilla
+++ python-bugzilla-0.6.2/bin/bugzilla
@@ -84,12 +84,12 @@ def setup_action_parser(action):
                 help="REQUIRED: bug summary")
         p.add_option('-o','--os',default='Linux',dest='op_sys',
                 help="OPTIONAL: operating system (default: Linux)")
-        p.add_option('-a','--arch',default='All',dest='rep_platform',
+        p.add_option('-a','--arch',default='x86',dest='rep_platform',
                 help="OPTIONAL: arch this bug occurs on (default: All)")
-        p.add_option('--severity',default='medium',dest='bug_severity',
-                help="OPTIONAL: bug severity (default: medium)")
-        p.add_option('--priority',default='medium',dest='priority',
-                help="OPTIONAL: bug priority (default: medium)")
+        p.add_option('--severity',default='normal',dest='bug_severity',
+                help="OPTIONAL: bug severity (default: normal)")
+        p.add_option('--priority',default='P5',dest='priority',
+                help="OPTIONAL: bug priority (default: P5)")
         p.add_option('-u','--url',dest='bug_file_loc',default='http://',
                 help="OPTIONAL: URL for further bug info")
         p.add_option('--cc',
Index: python-bugzilla-0.6.2/bugzilla.1
===================================================================
--- python-bugzilla-0.6.2.orig/bugzilla.1
+++ python-bugzilla-0.6.2/bugzilla.1
@@ -54,11 +54,11 @@ REQUIRED: bug summary
 .IP "--os=OP_SYS, -o OP_SYS"
 OPTIONAL: operating system (default: Linux)
 .IP "--arch=REP_PLATFORM, -a REP_PLATFORM"
-OPTIONAL: arch this bug occurs on (default: All)
+OPTIONAL: arch this bug occurs on (default: x86)
 .IP "--severity=BUG_SEVERITY"
-OPTIONAL: bug severity (default: medium)
+OPTIONAL: bug severity (default: normal)
 .IP "--priority=PRIORITY"
-OPTIONAL: bug priority (default: medium)
+OPTIONAL: bug priority (default: P5)
 .IP "--url=BUG_FILE_LOC, -u BUG_FILE_LOC"
 OPTIONAL: URL for further bug info
 .IP "--cc=CC"
Index: python-bugzilla-0.6.2/bugzilla/base.py
===================================================================
--- python-bugzilla-0.6.2.orig/bugzilla/base.py
+++ python-bugzilla-0.6.2/bugzilla/base.py
@@ -738,13 +738,13 @@ class BugzillaBase(object):
 
         DEFAULTED:
           platform: Hardware type where this bug was experienced.
-            Ex: i386
+            Ex: x86
             See querydefaults['rep_platform_list'] for accepted values.
           severity: Bug severity.
-            Ex: medium
+            Ex: normal
             See querydefaults['bug_severity_list'] for accepted values.
           priority: Bug priority.
-            Ex: medium
+            Ex: P5
             See querydefaults['priority_list'] for accepted values.
           op_sys: Operating system bug occurs on.
             Ex: Linux
