Index: gkrellm-2.3.5/server/gkrellmd.conf
===================================================================
--- gkrellm-2.3.5.orig/server/gkrellmd.conf
+++ gkrellm-2.3.5/server/gkrellmd.conf
@@ -4,7 +4,7 @@
 # the client update frequency.  Values may be from 1 to 10 and should be
 # smaller values to reduce network traffic.
 #
-#update-hz 3
+update-hz 3
 
 # Limit number of simultaneous clients allowed to connect.
 #
@@ -17,7 +17,7 @@
 
 # Specify the port to listen on for connections.
 #
-#port 19150
+port 19150
 
 # List of hosts allowed to connect.  If no hosts are specified in a
 # gkrellmd.conf file or on the command line, all hosts will be allowed.
@@ -38,17 +38,17 @@ group	nogroup
 # Create a PID file for the running gkrellmd.  Default is no PID file.
 # NOTE: Option ignored on Windows
 #
-#pidfile /var/run/gkrellmd.pid
+pidfile /run/gkrellmd.pid
 
 # Run in background and detach from the controlling terminal
 # NOTE: Option ignored on Windows
 #
-#detach
+detach
 
 # Enable writing logging message to the system syslog file
 # NOTE: On windows this enables logging to the windows event log
 #
-#syslog
+syslog
 
 # Time interval between checks for various monitors.  If nfs-interval
 # is <= 0 then gkrellmd will not read data for nfs file system types.
