--- Makefile.am~	2013-05-15 20:34:26.015505251 +0200
+++ Makefile.am	2013-05-15 20:34:45.183504551 +0200
@@ -123,14 +123,10 @@
 DOCS= install.txt coderules.txt filelist.txt jconfig.txt change.log \
 	rdrle.c wrrle.c BUILDING.txt ChangeLog.txt
 
-docdir = $(datadir)/doc
+docdir = $(datadir)/doc/libjpeg-turbo
 dist_doc_DATA = README README-turbo.txt libjpeg.txt structure.txt usage.txt \
 	wizard.txt 
 
-exampledir = $(datadir)/doc
-dist_example_DATA = example.c
-
-
 EXTRA_DIST = win release $(DOCS) testimages CMakeLists.txt \
 	sharedlib/CMakeLists.txt cmakescripts libjpeg.map.in doc doxygen.config \
 	jccolext.c jdcolext.c jdmrgext.c
