diff -uNr curl-7.28.1/configure.ac curl-7.28.1p/configure.ac
--- curl-7.28.1/configure.ac	2012-09-08 23:39:18.000000000 +0300
+++ curl-7.28.1p/configure.ac	2013-01-07 18:46:02.407508366 +0200
@@ -35,8 +35,8 @@
 terms of the curl license; see COPYING for more details])
 
 AC_CONFIG_SRCDIR([lib/urldata.h])
-AM_CONFIG_HEADER(lib/curl_config.h include/curl/curlbuild.h)
-AC_CONFIG_MACRO_DIR([m4])
+AC_CONFIG_HEADER(lib/curl_config.h include/curl/curlbuild.h)
+#AC_CONFIG_MACRO_DIR([m4])
 AM_MAINTAINER_MODE
 
 CURL_CHECK_OPTION_DEBUG
