--- curl-7.26.0/curl-config.in.orig	2012-03-16 21:42:16.000000000 +0000
+++ curl-7.26.0/curl-config.in	2012-05-27 22:37:10.000000000 +0000
@@ -74,7 +74,7 @@
 	;;
 
     --cc)
-	echo "@CC@"
+	echo "gcc"
 	;;
 
     --prefix)
@@ -136,16 +136,7 @@
        	;;
 
     --libs)
-	if test "X@libdir@" != "X/usr/lib" -a "X@libdir@" != "X/usr/lib64"; then
-	   CURLLIBDIR="-L@libdir@ "
-	else
-	   CURLLIBDIR=""
-	fi
-	if test "X@REQUIRE_LIB_DEPS@" = "Xyes"; then
-	  echo ${CURLLIBDIR}-lcurl @LIBCURL_LIBS@ @LIBS@
-	else
-	  echo ${CURLLIBDIR}-lcurl @LIBS@
-	fi
+	pkg-config libcurl --libs
 	;;
 
     --static-libs)
