--- curl-7.27.0/libcurl.pc.in.libs	2012-05-26 05:56:17.000000000 +0800
+++ curl-7.27.0/libcurl.pc.in	2012-07-28 12:13:41.000000000 +0800
@@ -34,6 +34,6 @@
 URL: http://curl.haxx.se/
 Description: Library to transfer files with ftp, http, etc.
 Version: @CURLVERSION@
-Libs: -L${libdir} -lcurl @LIBS@
+Libs: -L${libdir} -lcurl
 Libs.private: @LIBCURL_LIBS@ @LIBS@
 Cflags: -I${includedir} @CPPFLAG_CURL_STATICLIB@
