--- curl-7.26.0/docs/Makefile.am.orig	2012-04-27 18:25:35.000000000 +0000
+++ curl-7.26.0/docs/Makefile.am	2012-05-27 22:40:55.000000000 +0000
@@ -28,7 +28,7 @@
 
 HTMLPAGES = $(GENHTMLPAGES) index.html
 
-SUBDIRS = examples libcurl
+SUBDIRS = libcurl
 
 CLEANFILES = $(GENHTMLPAGES) $(PDFPAGES)
 
--- curl-7.26.0/docs/Makefile.in.orig	2012-05-22 21:46:41.000000000 +0000
+++ curl-7.26.0/docs/Makefile.in	2012-05-27 22:41:37.000000000 +0000
@@ -302,7 +302,7 @@
 GENHTMLPAGES = curl.html curl-config.html mk-ca-bundle.html
 PDFPAGES = curl.pdf curl-config.pdf mk-ca-bundle.pdf
 HTMLPAGES = $(GENHTMLPAGES) index.html
-SUBDIRS = examples libcurl
+SUBDIRS = libcurl
 CLEANFILES = $(GENHTMLPAGES) $(PDFPAGES)
 EXTRA_DIST = MANUAL BUGS CONTRIBUTE FAQ FEATURES INTERNALS SSLCERTS	 \
  README.win32 RESOURCES TODO TheArtOfHttpScripting THANKS VERSIONS	 \
