Index: curl-7.21.1/lib/urldata.h
===================================================================
--- curl-7.21.1.orig/lib/urldata.h
+++ curl-7.21.1/lib/urldata.h
@@ -55,7 +55,7 @@
 #define DICT_DEFINE3 "/LOOKUP:"
 
 #define CURL_DEFAULT_USER "anonymous"
-#define CURL_DEFAULT_PASSWORD "ftp@example.com"
+#define CURL_DEFAULT_PASSWORD "-curl@"
 
 /* length of longest IPv6 address string including the trailing null */
 #define MAX_IPADR_LEN sizeof("ffff:ffff:ffff:ffff:ffff:ffff:255.255.255.255")
Index: curl-7.21.1/tests/data/test100
===================================================================
--- curl-7.21.1.orig/tests/data/test100
+++ curl-7.21.1/tests/data/test100
@@ -45,7 +45,7 @@ ftp://%HOSTIP:%FTPPORT/
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE A
Index: curl-7.21.1/tests/data/test1000
===================================================================
--- curl-7.21.1.orig/tests/data/test1000
+++ curl-7.21.1/tests/data/test1000
@@ -33,7 +33,7 @@ ftp://%HOSTIP:%FTPPORT/1000/ -I
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD 1000
 QUIT
Index: curl-7.21.1/tests/data/test1003
===================================================================
--- curl-7.21.1.orig/tests/data/test1003
+++ curl-7.21.1/tests/data/test1003
@@ -35,7 +35,7 @@ ftp://%HOSTIP:%FTPPORT/path/1003
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 EPSV
Index: curl-7.21.1/tests/data/test1005
===================================================================
--- curl-7.21.1.orig/tests/data/test1005
+++ curl-7.21.1/tests/data/test1005
@@ -35,7 +35,7 @@ ftp://%HOSTIP:%FTPPORT/path/1005
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 EPSV
Index: curl-7.21.1/tests/data/test1006
===================================================================
--- curl-7.21.1.orig/tests/data/test1006
+++ curl-7.21.1/tests/data/test1006
@@ -36,7 +36,7 @@ ftp://%HOSTIP:%FTPPORT/path/1006
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 EPSV
Index: curl-7.21.1/tests/data/test101
===================================================================
--- curl-7.21.1.orig/tests/data/test101
+++ curl-7.21.1/tests/data/test101
@@ -47,7 +47,7 @@ ftp://%HOSTIP:%FTPPORT/ -P %CLIENTIP
 </strip>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 PORT 127,0,0,1,243,212
 TYPE A
Index: curl-7.21.1/tests/data/test1010
===================================================================
--- curl-7.21.1.orig/tests/data/test1010
+++ curl-7.21.1/tests/data/test1010
@@ -45,7 +45,7 @@ ftp://%HOSTIP:%FTPPORT//list/this/path/1
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE A
Index: curl-7.21.1/tests/data/test102
===================================================================
--- curl-7.21.1.orig/tests/data/test102
+++ curl-7.21.1/tests/data/test102
@@ -39,7 +39,7 @@ ftp://%HOSTIP:%FTPPORT/102
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 PASV
Index: curl-7.21.1/tests/data/test1028
===================================================================
--- curl-7.21.1.orig/tests/data/test1028
+++ curl-7.21.1/tests/data/test1028
@@ -65,7 +65,7 @@ Host: %HOSTIP:%HTTPPORT
 Accept: */*
 
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test103
===================================================================
--- curl-7.21.1.orig/tests/data/test103
+++ curl-7.21.1/tests/data/test103
@@ -40,7 +40,7 @@ ftp://%HOSTIP:%FTPPORT/a/path/103 -P -
 </strip>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD a
 CWD path
Index: curl-7.21.1/tests/data/test1036
===================================================================
--- curl-7.21.1.orig/tests/data/test1036
+++ curl-7.21.1/tests/data/test1036
@@ -42,7 +42,7 @@ This is the start!!
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 PASV
Index: curl-7.21.1/tests/data/test1037
===================================================================
--- curl-7.21.1.orig/tests/data/test1037
+++ curl-7.21.1/tests/data/test1037
@@ -41,7 +41,7 @@ ftp://%HOSTIP:%FTPPORT/1037 -C -
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 PASV
Index: curl-7.21.1/tests/data/test1038
===================================================================
--- curl-7.21.1.orig/tests/data/test1038
+++ curl-7.21.1/tests/data/test1038
@@ -36,7 +36,7 @@ worx?
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test1039
===================================================================
--- curl-7.21.1.orig/tests/data/test1039
+++ curl-7.21.1/tests/data/test1039
@@ -36,7 +36,7 @@ worx?
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test104
===================================================================
--- curl-7.21.1.orig/tests/data/test104
+++ curl-7.21.1/tests/data/test104
@@ -29,7 +29,7 @@ ftp://%HOSTIP:%FTPPORT/a/path/104 --head
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD a
 CWD path
Index: curl-7.21.1/tests/data/test1044
===================================================================
--- curl-7.21.1.orig/tests/data/test1044
+++ curl-7.21.1/tests/data/test1044
@@ -40,7 +40,7 @@ ftp://%HOSTIP:%FTPPORT/blalbla/1044 -I
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD blalbla
 MDTM 1044
Index: curl-7.21.1/tests/data/test1047
===================================================================
--- curl-7.21.1.orig/tests/data/test1047
+++ curl-7.21.1/tests/data/test1047
@@ -47,7 +47,7 @@ ftp://%HOSTIP:%FTPPORT/ --interface %CLI
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE A
Index: curl-7.21.1/tests/data/test1048
===================================================================
--- curl-7.21.1.orig/tests/data/test1048
+++ curl-7.21.1/tests/data/test1048
@@ -55,7 +55,7 @@ perl -e "print 'Test requires default te
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE A
Index: curl-7.21.1/tests/data/test1050
===================================================================
--- curl-7.21.1.orig/tests/data/test1050
+++ curl-7.21.1/tests/data/test1050
@@ -55,7 +55,7 @@ s/^(EPRT \|2\|::1\|)(.*)/$1/
 </strippart>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPRT |2|::1|
 TYPE A
Index: curl-7.21.1/tests/data/test1055
===================================================================
--- curl-7.21.1.orig/tests/data/test1055
+++ curl-7.21.1/tests/data/test1055
@@ -78,7 +78,7 @@ the
    PUT
       feature
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test1057
===================================================================
--- curl-7.21.1.orig/tests/data/test1057
+++ curl-7.21.1/tests/data/test1057
@@ -38,7 +38,7 @@ FTP retrieve a byte-range relative to en
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test106
===================================================================
--- curl-7.21.1.orig/tests/data/test106
+++ curl-7.21.1/tests/data/test106
@@ -37,7 +37,7 @@ FTP GET with type=A style ASCII URL usin
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD /path with  spaces
 CWD and things2
Index: curl-7.21.1/tests/data/test1062
===================================================================
--- curl-7.21.1.orig/tests/data/test1062
+++ curl-7.21.1/tests/data/test1062
@@ -36,7 +36,7 @@ ftp://%HOSTIP:%FTPPORT/path/1062
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 EPSV
Index: curl-7.21.1/tests/data/test107
===================================================================
--- curl-7.21.1.orig/tests/data/test107
+++ curl-7.21.1/tests/data/test107
@@ -40,7 +40,7 @@ works
 </upload>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test108
===================================================================
--- curl-7.21.1.orig/tests/data/test108
+++ curl-7.21.1/tests/data/test108
@@ -38,7 +38,7 @@ Moooooooooooo
 </strip>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD CWD
 CWD STOR
Index: curl-7.21.1/tests/data/test1086
===================================================================
--- curl-7.21.1.orig/tests/data/test1086
+++ curl-7.21.1/tests/data/test1086
@@ -98,7 +98,7 @@ ftp://%HOSTIP:%FTPPORT/1086 -m 7
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test109
===================================================================
--- curl-7.21.1.orig/tests/data/test109
+++ curl-7.21.1/tests/data/test109
@@ -33,7 +33,7 @@ Moooooooooooo
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test1091
===================================================================
--- curl-7.21.1.orig/tests/data/test1091
+++ curl-7.21.1/tests/data/test1091
@@ -32,7 +32,7 @@ FTP URL with type=i
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD /tmp
 CWD moo
Index: curl-7.21.1/tests/data/test1096
===================================================================
--- curl-7.21.1.orig/tests/data/test1096
+++ curl-7.21.1/tests/data/test1096
@@ -34,7 +34,7 @@ ftp://%HOSTIP:%FTPPORT/dir/1096 ftp://%H
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD dir
 EPSV
Index: curl-7.21.1/tests/data/test110
===================================================================
--- curl-7.21.1.orig/tests/data/test110
+++ curl-7.21.1/tests/data/test110
@@ -38,7 +38,7 @@ ftp://%HOSTIP:%FTPPORT/110 -C 20
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 PASV
Index: curl-7.21.1/tests/data/test1102
===================================================================
--- curl-7.21.1.orig/tests/data/test1102
+++ curl-7.21.1/tests/data/test1102
@@ -36,7 +36,7 @@ ftp://%HOSTIP:%FTPPORT/1102
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 SYST
 SITE NAMEFMT 1
Index: curl-7.21.1/tests/data/test1103
===================================================================
--- curl-7.21.1.orig/tests/data/test1103
+++ curl-7.21.1/tests/data/test1103
@@ -35,7 +35,7 @@ ftp://%HOSTIP:%FTPPORT/1103
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 SYST
 EPSV
Index: curl-7.21.1/tests/data/test1107
===================================================================
--- curl-7.21.1.orig/tests/data/test1107
+++ curl-7.21.1/tests/data/test1107
@@ -40,7 +40,7 @@ ftp://%HOSTIP:%FTPPORT/1107 --ftp-pret
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 PRET RETR 1107
 EPSV
Index: curl-7.21.1/tests/data/test1108
===================================================================
--- curl-7.21.1.orig/tests/data/test1108
+++ curl-7.21.1/tests/data/test1108
@@ -33,7 +33,7 @@ ftp://%HOSTIP:%FTPPORT/1108 --ftp-pret
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 PRET RETR 1108
 </protocol>
Index: curl-7.21.1/tests/data/test111
===================================================================
--- curl-7.21.1.orig/tests/data/test111
+++ curl-7.21.1/tests/data/test111
@@ -34,7 +34,7 @@ ftp://%HOSTIP:%FTPPORT/111 -C 2000
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test1112
===================================================================
--- curl-7.21.1.orig/tests/data/test1112
+++ curl-7.21.1/tests/data/test1112
@@ -101,7 +101,7 @@ FTPS download with strict timeout and sl
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PBSZ 0
 PROT C
 PWD
Index: curl-7.21.1/tests/data/test112
===================================================================
--- curl-7.21.1.orig/tests/data/test112
+++ curl-7.21.1/tests/data/test112
@@ -33,7 +33,7 @@ worx?
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test114
===================================================================
--- curl-7.21.1.orig/tests/data/test114
+++ curl-7.21.1/tests/data/test114
@@ -32,7 +32,7 @@ ftp://%HOSTIP:%FTPPORT/114
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 </protocol>
 </verify>
 </testcase>
Index: curl-7.21.1/tests/data/test115
===================================================================
--- curl-7.21.1.orig/tests/data/test115
+++ curl-7.21.1/tests/data/test115
@@ -34,7 +34,7 @@ ftp://%HOSTIP:%FTPPORT/115
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 PASV
Index: curl-7.21.1/tests/data/test116
===================================================================
--- curl-7.21.1.orig/tests/data/test116
+++ curl-7.21.1/tests/data/test116
@@ -42,7 +42,7 @@ s/^(EPRT \|1\|\d{1,3}\.\d{1,3}\.\d{1,3}\
 </strippart>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPRT |1|1.2.3.4|
 PORT 1,2,3,4,
Index: curl-7.21.1/tests/data/test117
===================================================================
--- curl-7.21.1.orig/tests/data/test117
+++ curl-7.21.1/tests/data/test117
@@ -33,7 +33,7 @@ ftp://%HOSTIP:%FTPPORT/117
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 PASV
Index: curl-7.21.1/tests/data/test118
===================================================================
--- curl-7.21.1.orig/tests/data/test118
+++ curl-7.21.1/tests/data/test118
@@ -35,7 +35,7 @@ ftp://%HOSTIP:%FTPPORT/118
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 PASV
Index: curl-7.21.1/tests/data/test119
===================================================================
--- curl-7.21.1.orig/tests/data/test119
+++ curl-7.21.1/tests/data/test119
@@ -39,7 +39,7 @@ ftp://%HOSTIP:%FTPPORT/119 -P -
 </strip>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 TYPE I
 SIZE 119
Index: curl-7.21.1/tests/data/test120
===================================================================
--- curl-7.21.1.orig/tests/data/test120
+++ curl-7.21.1/tests/data/test120
@@ -39,7 +39,7 @@ ftp://%HOSTIP:%FTPPORT/120 -Q "-DELE fil
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 PASV
Index: curl-7.21.1/tests/data/test121
===================================================================
--- curl-7.21.1.orig/tests/data/test121
+++ curl-7.21.1/tests/data/test121
@@ -37,7 +37,7 @@ ftp://%HOSTIP:%FTPPORT/121 -Q "-DELE aft
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 DELE before_transfer
 EPSV
Index: curl-7.21.1/tests/data/test122
===================================================================
--- curl-7.21.1.orig/tests/data/test122
+++ curl-7.21.1/tests/data/test122
@@ -33,7 +33,7 @@ ftp://%HOSTIP:%FTPPORT/122 -C 5
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 PASV
Index: curl-7.21.1/tests/data/test123
===================================================================
--- curl-7.21.1.orig/tests/data/test123
+++ curl-7.21.1/tests/data/test123
@@ -30,7 +30,7 @@ ftp://%HOSTIP:%FTPPORT/123 -T log/upload
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test124
===================================================================
--- curl-7.21.1.orig/tests/data/test124
+++ curl-7.21.1/tests/data/test124
@@ -34,7 +34,7 @@ ftp://%HOSTIP:%FTPPORT/124
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 PASV
Index: curl-7.21.1/tests/data/test125
===================================================================
--- curl-7.21.1.orig/tests/data/test125
+++ curl-7.21.1/tests/data/test125
@@ -32,7 +32,7 @@ ftp://%HOSTIP:%FTPPORT/path/to/file/125
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 QUIT
Index: curl-7.21.1/tests/data/test126
===================================================================
--- curl-7.21.1.orig/tests/data/test126
+++ curl-7.21.1/tests/data/test126
@@ -34,7 +34,7 @@ ftp://%HOSTIP:%FTPPORT/blalbla/lululul/1
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD blalbla
 CWD lululul
Index: curl-7.21.1/tests/data/test127
===================================================================
--- curl-7.21.1.orig/tests/data/test127
+++ curl-7.21.1/tests/data/test127
@@ -31,7 +31,7 @@ ftp://%HOSTIP:%FTPPORT/path/to/file/127 
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 CWD to
Index: curl-7.21.1/tests/data/test128
===================================================================
--- curl-7.21.1.orig/tests/data/test128
+++ curl-7.21.1/tests/data/test128
@@ -37,7 +37,7 @@ the
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test129
===================================================================
--- curl-7.21.1.orig/tests/data/test129
+++ curl-7.21.1/tests/data/test129
@@ -40,7 +40,7 @@ FTP GET with type=A style ASCII URL and 
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE A
Index: curl-7.21.1/tests/data/test135
===================================================================
--- curl-7.21.1.orig/tests/data/test135
+++ curl-7.21.1/tests/data/test135
@@ -40,7 +40,7 @@ FTP retrieve a byte-range
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test137
===================================================================
--- curl-7.21.1.orig/tests/data/test137
+++ curl-7.21.1/tests/data/test137
@@ -33,7 +33,7 @@ ftp://%HOSTIP:%FTPPORT/blalbla/lululul/1
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD blalbla
 CWD lululul
Index: curl-7.21.1/tests/data/test138
===================================================================
--- curl-7.21.1.orig/tests/data/test138
+++ curl-7.21.1/tests/data/test138
@@ -35,7 +35,7 @@ ftp://%HOSTIP:%FTPPORT/blalbla/lululul/1
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD blalbla
 CWD lululul
Index: curl-7.21.1/tests/data/test139
===================================================================
--- curl-7.21.1.orig/tests/data/test139
+++ curl-7.21.1/tests/data/test139
@@ -33,7 +33,7 @@ ftp://%HOSTIP:%FTPPORT/blalbla/139 -z "1
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD blalbla
 MDTM 139
Index: curl-7.21.1/tests/data/test140
===================================================================
--- curl-7.21.1.orig/tests/data/test140
+++ curl-7.21.1/tests/data/test140
@@ -32,7 +32,7 @@ ftp://%HOSTIP:%FTPPORT/blalbla/140 -z "1
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD blalbla
 MDTM 140
Index: curl-7.21.1/tests/data/test141
===================================================================
--- curl-7.21.1.orig/tests/data/test141
+++ curl-7.21.1/tests/data/test141
@@ -34,7 +34,7 @@ ftp://%HOSTIP:%FTPPORT/blalbla/141 -I
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD blalbla
 MDTM 141
Index: curl-7.21.1/tests/data/test142
===================================================================
--- curl-7.21.1.orig/tests/data/test142
+++ curl-7.21.1/tests/data/test142
@@ -28,7 +28,7 @@ ftp://%HOSTIP:%FTPPORT/part1/part2/part3
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD part1
 CWD part2
Index: curl-7.21.1/tests/data/test143
===================================================================
--- curl-7.21.1.orig/tests/data/test143
+++ curl-7.21.1/tests/data/test143
@@ -30,7 +30,7 @@ FTP URL with type=a
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD /tmp
 CWD moo
Index: curl-7.21.1/tests/data/test144
===================================================================
--- curl-7.21.1.orig/tests/data/test144
+++ curl-7.21.1/tests/data/test144
@@ -38,7 +38,7 @@ ftp://%HOSTIP:%FTPPORT/ -P - -l
 </strip>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 PORT 127,0,0,1,243,212
 TYPE A
Index: curl-7.21.1/tests/data/test145
===================================================================
--- curl-7.21.1.orig/tests/data/test145
+++ curl-7.21.1/tests/data/test145
@@ -40,7 +40,7 @@ ftp://%HOSTIP:%FTPPORT/ -P - -l
 </strip>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 PORT 127,0,0,1,243,212
 TYPE A
Index: curl-7.21.1/tests/data/test146
===================================================================
--- curl-7.21.1.orig/tests/data/test146
+++ curl-7.21.1/tests/data/test146
@@ -36,7 +36,7 @@ ftp://%HOSTIP:%FTPPORT/first/dir/here/14
 </strip>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD first
 CWD dir
Index: curl-7.21.1/tests/data/test147
===================================================================
--- curl-7.21.1.orig/tests/data/test147
+++ curl-7.21.1/tests/data/test147
@@ -32,7 +32,7 @@ ftp://%HOSTIP:%FTPPORT/first/dir/here/14
 </strip>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD first
 MKD first
Index: curl-7.21.1/tests/data/test148
===================================================================
--- curl-7.21.1.orig/tests/data/test148
+++ curl-7.21.1/tests/data/test148
@@ -32,7 +32,7 @@ ftp://%HOSTIP:%FTPPORT/attempt/to/get/th
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD attempt
 MKD attempt
Index: curl-7.21.1/tests/data/test149
===================================================================
--- curl-7.21.1.orig/tests/data/test149
+++ curl-7.21.1/tests/data/test149
@@ -28,7 +28,7 @@ send away this contents
 </strip>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD dir1
 EPSV
Index: curl-7.21.1/tests/data/test161
===================================================================
--- curl-7.21.1.orig/tests/data/test161
+++ curl-7.21.1/tests/data/test161
@@ -27,7 +27,7 @@ ftp://%HOSTIP:%FTPPORT/161
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test182
===================================================================
--- curl-7.21.1.orig/tests/data/test182
+++ curl-7.21.1/tests/data/test182
@@ -25,7 +25,7 @@ ftp://%HOSTIP:%FTPPORT/182
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test190
===================================================================
--- curl-7.21.1.orig/tests/data/test190
+++ curl-7.21.1/tests/data/test190
@@ -30,7 +30,7 @@ ftp://%HOSTIP:%FTPPORT/path/to/file/190 
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 </protocol>
Index: curl-7.21.1/tests/data/test195
===================================================================
--- curl-7.21.1.orig/tests/data/test195
+++ curl-7.21.1/tests/data/test195
@@ -26,7 +26,7 @@ ftp://%HOSTIP:%FTPPORT/195
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 </protocol>
 </verify>
 </testcase>
Index: curl-7.21.1/tests/data/test196
===================================================================
--- curl-7.21.1.orig/tests/data/test196
+++ curl-7.21.1/tests/data/test196
@@ -27,9 +27,9 @@ ftp://%HOSTIP:%FTPPORT/196 --retry 1
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 </protocol>
 </verify>
 </testcase>
Index: curl-7.21.1/tests/data/test2000
===================================================================
--- curl-7.21.1.orig/tests/data/test2000
+++ curl-7.21.1/tests/data/test2000
@@ -48,7 +48,7 @@ moo
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test2001
===================================================================
--- curl-7.21.1.orig/tests/data/test2001
+++ curl-7.21.1/tests/data/test2001
@@ -72,7 +72,7 @@ Host: %HOSTIP:%HTTPPORT
 Accept: */*
 
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test2002
===================================================================
--- curl-7.21.1.orig/tests/data/test2002
+++ curl-7.21.1/tests/data/test2002
@@ -81,7 +81,7 @@ Host: %HOSTIP:%HTTPPORT
 Accept: */*
 
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test2003
===================================================================
--- curl-7.21.1.orig/tests/data/test2003
+++ curl-7.21.1/tests/data/test2003
@@ -81,7 +81,7 @@ Host: %HOSTIP:%HTTPPORT
 Accept: */*
 
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test210
===================================================================
--- curl-7.21.1.orig/tests/data/test210
+++ curl-7.21.1/tests/data/test210
@@ -29,7 +29,7 @@ data blobb
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD a
 CWD path
Index: curl-7.21.1/tests/data/test211
===================================================================
--- curl-7.21.1.orig/tests/data/test211
+++ curl-7.21.1/tests/data/test211
@@ -30,7 +30,7 @@ data blobb
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD a
 CWD path
Index: curl-7.21.1/tests/data/test212
===================================================================
--- curl-7.21.1.orig/tests/data/test212
+++ curl-7.21.1/tests/data/test212
@@ -39,7 +39,7 @@ s/^(PORT )\d{1,3},\d{1,3},\d{1,3},\d{1,3
 </strippart>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD a
 CWD path
Index: curl-7.21.1/tests/data/test215
===================================================================
--- curl-7.21.1.orig/tests/data/test215
+++ curl-7.21.1/tests/data/test215
@@ -36,7 +36,7 @@ ftp://%HOSTIP:%FTPPORT/a/path/215/ ftp:/
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD a
 CWD path
Index: curl-7.21.1/tests/data/test216
===================================================================
--- curl-7.21.1.orig/tests/data/test216
+++ curl-7.21.1/tests/data/test216
@@ -23,7 +23,7 @@ upload this file twice
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD a
 CWD path
Index: curl-7.21.1/tests/data/test227
===================================================================
--- curl-7.21.1.orig/tests/data/test227
+++ curl-7.21.1/tests/data/test227
@@ -40,7 +40,7 @@ ftp://%HOSTIP:%FTPPORT/227 -Q "NOOP 1" -
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 NOOP 1
 FAIL
Index: curl-7.21.1/tests/data/test228
===================================================================
--- curl-7.21.1.orig/tests/data/test228
+++ curl-7.21.1/tests/data/test228
@@ -39,7 +39,7 @@ ftp://%HOSTIP:%FTPPORT/228 --ftp-account
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 ACCT one count
 PWD
 EPSV
Index: curl-7.21.1/tests/data/test229
===================================================================
--- curl-7.21.1.orig/tests/data/test229
+++ curl-7.21.1/tests/data/test229
@@ -31,7 +31,7 @@ ftp://%HOSTIP:%FTPPORT/229 --ftp-account
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 ACCT one count
 </protocol>
 <errorcode>
Index: curl-7.21.1/tests/data/test235
===================================================================
--- curl-7.21.1.orig/tests/data/test235
+++ curl-7.21.1/tests/data/test235
@@ -25,7 +25,7 @@ worx?
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test236
===================================================================
--- curl-7.21.1.orig/tests/data/test236
+++ curl-7.21.1/tests/data/test236
@@ -30,7 +30,7 @@ Test data
 
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test237
===================================================================
--- curl-7.21.1.orig/tests/data/test237
+++ curl-7.21.1/tests/data/test237
@@ -34,7 +34,7 @@ ftp://%HOSTIP:%FTPPORT/237 --disable-eps
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 PASV
 </protocol>
Index: curl-7.21.1/tests/data/test238
===================================================================
--- curl-7.21.1.orig/tests/data/test238
+++ curl-7.21.1/tests/data/test238
@@ -32,7 +32,7 @@ ftp://%HOSTIP:%FTPPORT/238
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 PASV
Index: curl-7.21.1/tests/data/test247
===================================================================
--- curl-7.21.1.orig/tests/data/test247
+++ curl-7.21.1/tests/data/test247
@@ -32,7 +32,7 @@ ftp://%HOSTIP:%FTPPORT/247 -T log/test24
 </upload>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 MDTM 247
 QUIT
Index: curl-7.21.1/tests/data/test248
===================================================================
--- curl-7.21.1.orig/tests/data/test248
+++ curl-7.21.1/tests/data/test248
@@ -38,7 +38,7 @@ works
 </upload>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 MDTM 248
 EPSV
Index: curl-7.21.1/tests/data/test250
===================================================================
--- curl-7.21.1.orig/tests/data/test250
+++ curl-7.21.1/tests/data/test250
@@ -41,7 +41,7 @@ ftp://%HOSTIP:%FTPPORT/
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE A
Index: curl-7.21.1/tests/data/test251
===================================================================
--- curl-7.21.1.orig/tests/data/test251
+++ curl-7.21.1/tests/data/test251
@@ -43,7 +43,7 @@ ftp://%HOSTIP:%FTPPORT/ -P %CLIENTIP
 </strip>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 PORT 127,0,0,1,243,212
 TYPE A
Index: curl-7.21.1/tests/data/test252
===================================================================
--- curl-7.21.1.orig/tests/data/test252
+++ curl-7.21.1/tests/data/test252
@@ -48,7 +48,7 @@ FTP IPv6 dir list PASV
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE A
Index: curl-7.21.1/tests/data/test253
===================================================================
--- curl-7.21.1.orig/tests/data/test253
+++ curl-7.21.1/tests/data/test253
@@ -51,7 +51,7 @@ s/^(EPRT \|2\|::1\|)(.*)/$1/
 </strippart>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPRT |2|::1|
 TYPE A
Index: curl-7.21.1/tests/data/test254
===================================================================
--- curl-7.21.1.orig/tests/data/test254
+++ curl-7.21.1/tests/data/test254
@@ -49,7 +49,7 @@ FTP IPv6 dir list PASV and --disable-eps
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE A
Index: curl-7.21.1/tests/data/test255
===================================================================
--- curl-7.21.1.orig/tests/data/test255
+++ curl-7.21.1/tests/data/test255
@@ -52,7 +52,7 @@ s/^(EPRT \|2\|::1\|)(.*)/$1/
 </strippart>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPRT |2|::1|
 TYPE A
Index: curl-7.21.1/tests/data/test261
===================================================================
--- curl-7.21.1.orig/tests/data/test261
+++ curl-7.21.1/tests/data/test261
@@ -36,7 +36,7 @@ ftp://%HOSTIP:%FTPPORT/261
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test270
===================================================================
--- curl-7.21.1.orig/tests/data/test270
+++ curl-7.21.1/tests/data/test270
@@ -38,7 +38,7 @@ ftp://%HOSTIP:%FTPPORT/270 --ftp-skip-pa
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 PASV
 TYPE I
Index: curl-7.21.1/tests/data/test272
===================================================================
--- curl-7.21.1.orig/tests/data/test272
+++ curl-7.21.1/tests/data/test272
@@ -31,7 +31,7 @@ ftp://%HOSTIP:%FTPPORT/272 -z "2004 jan 
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 MDTM 272
 QUIT
Index: curl-7.21.1/tests/data/test280
===================================================================
--- curl-7.21.1.orig/tests/data/test280
+++ curl-7.21.1/tests/data/test280
@@ -51,7 +51,7 @@ ftp://%HOSTIP:%FTPPORT/280/ --ftp-altern
 <protocol>
 USER anonymous
 USER replacement
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD 280
 EPSV
Index: curl-7.21.1/tests/data/test290
===================================================================
--- curl-7.21.1.orig/tests/data/test290
+++ curl-7.21.1/tests/data/test290
@@ -32,7 +32,7 @@ ftp://%HOSTIP:%FTPPORT/290 --max-filesiz
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test291
===================================================================
--- curl-7.21.1.orig/tests/data/test291
+++ curl-7.21.1/tests/data/test291
@@ -35,7 +35,7 @@ ftp://%HOSTIP:%FTPPORT/291 --max-filesiz
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test294
===================================================================
--- curl-7.21.1.orig/tests/data/test294
+++ curl-7.21.1/tests/data/test294
@@ -51,7 +51,7 @@ ftp://%HOSTIP:%FTPPORT/294/ --ftp-accoun
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 ACCT data for acct
 PWD
 CWD 294
Index: curl-7.21.1/tests/data/test295
===================================================================
--- curl-7.21.1.orig/tests/data/test295
+++ curl-7.21.1/tests/data/test295
@@ -39,7 +39,7 @@ ftp://%HOSTIP:%FTPPORT/295/
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 </protocol>
 </verify>
 </testcase>
Index: curl-7.21.1/tests/data/test296
===================================================================
--- curl-7.21.1.orig/tests/data/test296
+++ curl-7.21.1/tests/data/test296
@@ -33,7 +33,7 @@ FTP CWD with --ftp-method multicwd
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD first
 CWD second
Index: curl-7.21.1/tests/data/test297
===================================================================
--- curl-7.21.1.orig/tests/data/test297
+++ curl-7.21.1/tests/data/test297
@@ -33,7 +33,7 @@ FTP CWD with --ftp-method singlecwd
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD first/second/third
 EPSV
Index: curl-7.21.1/tests/data/test298
===================================================================
--- curl-7.21.1.orig/tests/data/test298
+++ curl-7.21.1/tests/data/test298
@@ -33,7 +33,7 @@ FTP CWD with --ftp-method nocwd
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test350
===================================================================
--- curl-7.21.1.orig/tests/data/test350
+++ curl-7.21.1/tests/data/test350
@@ -45,7 +45,7 @@ ftp://%HOSTIP:%FTPPORT// --ftp-method mu
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD /
 EPSV
Index: curl-7.21.1/tests/data/test351
===================================================================
--- curl-7.21.1.orig/tests/data/test351
+++ curl-7.21.1/tests/data/test351
@@ -45,7 +45,7 @@ ftp://%HOSTIP:%FTPPORT// --ftp-method no
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE A
Index: curl-7.21.1/tests/data/test352
===================================================================
--- curl-7.21.1.orig/tests/data/test352
+++ curl-7.21.1/tests/data/test352
@@ -45,7 +45,7 @@ ftp://%HOSTIP:%FTPPORT// --ftp-method si
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD /
 EPSV
Index: curl-7.21.1/tests/data/test353
===================================================================
--- curl-7.21.1.orig/tests/data/test353
+++ curl-7.21.1/tests/data/test353
@@ -45,7 +45,7 @@ ftp://%HOSTIP:%FTPPORT/ --ftp-method sin
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE A
Index: curl-7.21.1/tests/data/test400
===================================================================
--- curl-7.21.1.orig/tests/data/test400
+++ curl-7.21.1/tests/data/test400
@@ -48,7 +48,7 @@ FTPS dir list PASV unencrypted data
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PBSZ 0
 PROT C
 PWD
Index: curl-7.21.1/tests/data/test401
===================================================================
--- curl-7.21.1.orig/tests/data/test401
+++ curl-7.21.1/tests/data/test401
@@ -43,7 +43,7 @@ works
 </upload>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PBSZ 0
 PROT C
 PWD
Index: curl-7.21.1/tests/data/test403
===================================================================
--- curl-7.21.1.orig/tests/data/test403
+++ curl-7.21.1/tests/data/test403
@@ -50,7 +50,7 @@ FTPS with CCC not supported by server
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PBSZ 0
 PROT C
 CCC
Index: curl-7.21.1/tests/data/test406
===================================================================
--- curl-7.21.1.orig/tests/data/test406
+++ curl-7.21.1/tests/data/test406
@@ -53,7 +53,7 @@ FTPS dir list, PORT with specified IP
 </strip>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PBSZ 0
 PROT C
 PWD
Index: curl-7.21.1/tests/data/test407
===================================================================
--- curl-7.21.1.orig/tests/data/test407
+++ curl-7.21.1/tests/data/test407
@@ -40,7 +40,7 @@ data blobb
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PBSZ 0
 PROT C
 PWD
Index: curl-7.21.1/tests/data/test408
===================================================================
--- curl-7.21.1.orig/tests/data/test408
+++ curl-7.21.1/tests/data/test408
@@ -41,7 +41,7 @@ Moooooooooooo
 </strip>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PBSZ 0
 PROT C
 PWD
Index: curl-7.21.1/tests/data/test409
===================================================================
--- curl-7.21.1.orig/tests/data/test409
+++ curl-7.21.1/tests/data/test409
@@ -43,7 +43,7 @@ works
 </upload>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PBSZ 0
 PROT C
 PWD
Index: curl-7.21.1/tests/data/test505
===================================================================
--- curl-7.21.1.orig/tests/data/test505
+++ curl-7.21.1/tests/data/test505
@@ -47,7 +47,7 @@ works?
 </upload>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test511
===================================================================
--- curl-7.21.1.orig/tests/data/test511
+++ curl-7.21.1/tests/data/test511
@@ -35,7 +35,7 @@ ftp://%HOSTIP:%FTPPORT/511
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 MDTM 511
 QUIT
Index: curl-7.21.1/tests/data/test520
===================================================================
--- curl-7.21.1.orig/tests/data/test520
+++ curl-7.21.1/tests/data/test520
@@ -34,7 +34,7 @@ ftp://%HOSTIP:%FTPPORT/520
 
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 MDTM 520
 EPSV
Index: curl-7.21.1/tests/data/test524
===================================================================
--- curl-7.21.1.orig/tests/data/test524
+++ curl-7.21.1/tests/data/test524
@@ -34,7 +34,7 @@ ftp://%HOSTIP:%FTPPORT/path/to/
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 </protocol>
 
Index: curl-7.21.1/tests/data/test525
===================================================================
--- curl-7.21.1.orig/tests/data/test525
+++ curl-7.21.1/tests/data/test525
@@ -43,7 +43,7 @@ Moooooooooooo
 </strip>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 PORT 127,0,0,1,5,109
Index: curl-7.21.1/tests/data/test526
===================================================================
--- curl-7.21.1.orig/tests/data/test526
+++ curl-7.21.1/tests/data/test526
@@ -41,7 +41,7 @@ ftp://%HOSTIP:%FTPPORT/path/526
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 EPSV
Index: curl-7.21.1/tests/data/test527
===================================================================
--- curl-7.21.1.orig/tests/data/test527
+++ curl-7.21.1/tests/data/test527
@@ -41,7 +41,7 @@ ftp://%HOSTIP:%FTPPORT/path/527
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 EPSV
Index: curl-7.21.1/tests/data/test529
===================================================================
--- curl-7.21.1.orig/tests/data/test529
+++ curl-7.21.1/tests/data/test529
@@ -43,7 +43,7 @@ Moooooooooooo
 </strip>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 PORT 127,0,0,1,5,109
Index: curl-7.21.1/tests/data/test531
===================================================================
--- curl-7.21.1.orig/tests/data/test531
+++ curl-7.21.1/tests/data/test531
@@ -46,7 +46,7 @@ don't upload this
 </strip>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 PORT 127,0,0,1,5,109
Index: curl-7.21.1/tests/data/test532
===================================================================
--- curl-7.21.1.orig/tests/data/test532
+++ curl-7.21.1/tests/data/test532
@@ -41,7 +41,7 @@ ftp://%HOSTIP:%FTPPORT/path/532
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 EPSV
Index: curl-7.21.1/tests/data/test533
===================================================================
--- curl-7.21.1.orig/tests/data/test533
+++ curl-7.21.1/tests/data/test533
@@ -39,7 +39,7 @@ ftp://%HOSTIP:%FTPPORT/path/533 ftp://%H
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 EPSV
Index: curl-7.21.1/tests/data/test534
===================================================================
--- curl-7.21.1.orig/tests/data/test534
+++ curl-7.21.1/tests/data/test534
@@ -40,7 +40,7 @@ ftp://non-existing-host.haxx.se/path/534
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 EPSV
Index: curl-7.21.1/tests/data/test538
===================================================================
--- curl-7.21.1.orig/tests/data/test538
+++ curl-7.21.1/tests/data/test538
@@ -40,7 +40,7 @@ ftp://%HOSTIP:%FTPPORT/538
 </errorcode>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 </protocol>
 </verify>
 </testcase>
Index: curl-7.21.1/tests/data/test539
===================================================================
--- curl-7.21.1.orig/tests/data/test539
+++ curl-7.21.1/tests/data/test539
@@ -45,7 +45,7 @@ ftp://%HOSTIP:%FTPPORT/path/to/the/file/
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path/to/the/file
 EPSV
Index: curl-7.21.1/tests/data/test541
===================================================================
--- curl-7.21.1.orig/tests/data/test541
+++ curl-7.21.1/tests/data/test541
@@ -38,7 +38,7 @@ works?
 </upload>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE I
Index: curl-7.21.1/tests/data/test542
===================================================================
--- curl-7.21.1.orig/tests/data/test542
+++ curl-7.21.1/tests/data/test542
@@ -46,7 +46,7 @@ ftp://%HOSTIP:%FTPPORT/542
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 TYPE I
 SIZE 542
Index: curl-7.21.1/tests/data/test546
===================================================================
--- curl-7.21.1.orig/tests/data/test546
+++ curl-7.21.1/tests/data/test546
@@ -46,7 +46,7 @@ ftp://%HOSTIP:%FTPPORT/path/546 ftp://%H
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 EPSV
Index: curl-7.21.1/tests/data/test562
===================================================================
--- curl-7.21.1.orig/tests/data/test562
+++ curl-7.21.1/tests/data/test562
@@ -41,7 +41,7 @@ FTP a type=A URL and CURLOPT_PORT set
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE A
Index: curl-7.21.1/tests/data/test564
===================================================================
--- curl-7.21.1.orig/tests/data/test564
+++ curl-7.21.1/tests/data/test564
@@ -42,7 +42,7 @@ ftp://%HOSTIP:%FTPPORT/path/564 %HOSTIP:
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 CWD path
 EPSV
Index: curl-7.21.1/tests/data/test706
===================================================================
--- curl-7.21.1.orig/tests/data/test706
+++ curl-7.21.1/tests/data/test706
@@ -48,7 +48,7 @@ FTP dir list PASV via SOCKS4
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE A
Index: curl-7.21.1/tests/data/test707
===================================================================
--- curl-7.21.1.orig/tests/data/test707
+++ curl-7.21.1/tests/data/test707
@@ -48,7 +48,7 @@ FTP dir list PASV via SOCKS5
 <verify>
 <protocol>
 USER anonymous
-PASS ftp@example.com
+PASS -curl@
 PWD
 EPSV
 TYPE A
