diff -rup a/manual/Makefile b/manual/Makefile
--- a/manual/Makefile	2012-05-20 19:47:38.000000000 -0600
+++ b/manual/Makefile	2012-05-29 22:23:33.920428631 -0600
@@ -129,7 +129,8 @@ $(objpfx)%.c.texi: examples/%.c
 	mv -f $@.new $@
 
 $(objpfx)%.info: %.texinfo
-	LANGUAGE=C LC_ALL=C $(MAKEINFO) -P $(objpfx) --output=$@ $<
+	LANGUAGE=C LC_ALL=C $(MAKEINFO) -P $(objpfx) --output=`basename $@` $<
+	mv `basename $@`* $(objpfx)
 
 $(objpfx)%.dvi: %.texinfo
 	cd $(objpfx);$(TEXI2DVI) -I $(shell cd $(<D) && pwd) --output=$@ \
