diff -rup a/sysdeps/unix/sysv/linux/check_pf.c b/sysdeps/unix/sysv/linux/check_pf.c
--- a/sysdeps/unix/sysv/linux/check_pf.c	2012-02-16 22:51:17.148797741 -0700
+++ b/sysdeps/unix/sysv/linux/check_pf.c	2012-02-17 10:55:26.513139941 -0700
@@ -209,7 +209,8 @@ make_request (int fd, pid_t pid)
 		    }
 		  else
 		    {
-		      if (!IN6_IS_ADDR_LOOPBACK (address))
+		      if (!IN6_IS_ADDR_LOOPBACK (address)
+			  && !IN6_IS_ADDR_LINKLOCAL (address))
 			seen_ipv6 = true;
 		    }
 		}
Only in b/sysdeps/x86_64: .dl-trampoline.S.swp
