--- pmtools-20071116/acpidump/Makefile~	2008-03-19 22:33:40.000000000 -0400
+++ pmtools-20071116/acpidump/Makefile	2008-03-19 22:33:50.000000000 -0400
@@ -1,7 +1,7 @@
 PROG= acpidump
 SRCS=	acpidump.c
 KERNEL_INCLUDE := ../include
-CFLAGS += -Wall -Wstrict-prototypes -Wdeclaration-after-statement -Os -s -D_LINUX -DDEFINE_ALTERNATE_TYPES -I$(KERNEL_INCLUDE) 
+CFLAGS += -Wall -Wstrict-prototypes -Wdeclaration-after-statement -s -D_LINUX -DDEFINE_ALTERNATE_TYPES -I$(KERNEL_INCLUDE) 
 
 all: acpidump
 $(PROG) : $(SRCS)
