--- ./src/check_print.c.orig	2012-10-22 08:03:34.000000000 -0600
+++ ./src/check_print.c	2012-10-22 12:51:11.343652481 -0600
@@ -179,10 +179,10 @@ void tr_xmlprint (FILE *file, TestResult
           tr->duration < 0 ? 0 : tr->duration % 1000000);
   fprintf(file, "      <description>");
   fprint_xml_esc(file, tr->tcname);
-  fprintf(file,"</description>\n", tr->tcname);
-  fprintf(file, "      <message>", tr->msg);
+  fprintf(file,"</description>\n");
+  fprintf(file, "      <message>");
   fprint_xml_esc(file, tr->msg);
-  fprintf(file,"</message>\n", tr->msg);
+  fprintf(file,"</message>\n");
   fprintf(file, "    </test>\n");
   
   if (slash != NULL) {
--- ./src/check_log.c.orig	2012-10-22 08:03:34.000000000 -0600
+++ ./src/check_log.c	2012-10-22 12:49:29.626726729 -0600
@@ -257,7 +257,7 @@ void xml_lfun (SRunner *sr CK_ATTRIBUTE_
       gettimeofday(&now, NULL);
       timersub(&now, &inittv, &now);
 
-      fprintf(file, "  <duration>%d.%06d</duration>\n",
+      fprintf(file, "  <duration>%ld.%06ld</duration>\n",
               now.tv_sec, now.tv_usec);
       fprintf(file, "</testsuites>\n");
     }
@@ -267,9 +267,9 @@ void xml_lfun (SRunner *sr CK_ATTRIBUTE_
   case CLSTART_S:
     s = obj;
     fprintf(file, "  <suite>\n");
-    fprintf(file, "    <title>", s->name);
+    fprintf(file, "    <title>");
     fprint_xml_esc(file, s->name);
-    fprintf(file,"</title>\n", s->name);
+    fprintf(file,"</title>\n");
     break;
   case CLEND_SR:
     break; 
