--- libfame-0.9.1/src/half_mmx.h.old	2004-09-26 17:20:09.424203697 -0700
+++ libfame-0.9.1/src/half_mmx.h	2004-09-26 17:21:22.612577356 -0700
@@ -28,10 +28,10 @@
   short _mmx_rc[4]; /* rounding control */
   register short *rc = _mmx_rc;
 
-  _mmx_rc[0] = 1 - rounding;
-  _mmx_rc[1] = 1 - rounding;
-  _mmx_rc[2] = 1 - rounding;
-  _mmx_rc[3] = 1 - rounding;
+  _mmx_rc[0] = _mmx_one[0] - rounding;
+  _mmx_rc[1] = _mmx_one[1] - rounding;
+  _mmx_rc[2] = _mmx_one[2] - rounding;
+  _mmx_rc[3] = _mmx_one[3] - rounding;
 
   asm volatile ("movl (%0), %3\n"          /* %3 = ref[0] */
 		"pxor %%mm7, %%mm7\n"      /* mm7 = 0 */
