diff --git a/test/attr.test b/test/attr.test
index d394562..e8f134c 100644
--- a/test/attr.test
+++ b/test/attr.test
@@ -264,7 +264,7 @@ Tests for proper path recursion reported by Tony Ernst <tee@sgi.com> bnc#457660
 	$ setfattr -n "user.9" 1/2/3
 	$ setfattr -n "user.a" 1/2/3
 
-	$ getfattr -h -L -R -m '.' -e hex 1 | ./sort-getfattr-output
+	$ getfattr -h -L -R -m '.' -e hex 1 | grep -v ^security\\.selinux | ./sort-getfattr-output
 	> # file: 1
 	> user.9
 	> user.a
@@ -278,7 +278,7 @@ Tests for proper path recursion reported by Tony Ernst <tee@sgi.com> bnc#457660
 	> user.a
 	>
 
-	$ getfattr -h -P -R -m '.' -e hex 1/2 | ./sort-getfattr-output
+	$ getfattr -h -P -R -m '.' -e hex 1/2 | grep -v ^security\\.selinux | ./sort-getfattr-output
 	> # file: 1/2
 	> user.9
 	> user.a
