Index: gyachi-1.2.11/gyachi.desktop
===================================================================
--- gyachi-1.2.11.orig/gyachi.desktop
+++ gyachi-1.2.11/gyachi.desktop
@@ -1,8 +1,11 @@
 [Desktop Entry]
 Name=GYachE Improved
 Name[sr]=Усавршени GYachE
+GenericName=GYachE Improved
+GenericName[tr]=Gelişmiş GYachE
 Comment[de]=Yahoo! Klient
 Comment[sr]=Програм за Yahoo!
+Comment[tr]=Yahoo! İstemcisi
 Comment=Yahoo! Client
 Exec=gyachi
 Icon=gyachi
