diff -Naur gyachi-1.2.2.orig/Makefile.am gyachi-1.2.2/Makefile.am
--- gyachi-1.2.2.orig/Makefile.am	2009-08-15 19:13:34.042864816 +0300
+++ gyachi-1.2.2/Makefile.am	2009-08-15 19:38:11.018739527 +0300
@@ -58,37 +58,37 @@
 	    fi \
 	  done \
 	fi
-	if test -d $(srcdir)/doc; then \
-	  $(mkinstalldirs) $(DESTDIR)$(PACKAGE_DOC_DIR); \
-	  for document in $(srcdir)/doc/*.txt; do \
-	    if test -f $$document; then \
-	      $(INSTALL_DATA) $$document $(DESTDIR)$(PACKAGE_DOC_DIR); \
-	    fi \
-	  done \
-	fi
-	if test -d $(srcdir)/doc/txt; then \
-	  $(mkinstalldirs) $(DESTDIR)$(PACKAGE_DOC_DIR)/txt; \
-	  for document in $(srcdir)/doc/txt/*; do \
-	    if test -f $$document; then \
-	      $(INSTALL_DATA) $$document $(DESTDIR)$(PACKAGE_DOC_DIR)/txt; \
-	    fi \
-	  done \
-	fi
-	if test -d $(srcdir)/doc/html; then \
-	  $(mkinstalldirs) $(DESTDIR)$(PACKAGE_DOC_DIR)/html; \
-	  for document in $(srcdir)/doc/html/*; do \
-	    if test -f $$document; then \
-	      $(INSTALL_DATA) $$document $(DESTDIR)$(PACKAGE_DOC_DIR)/html; \
-	    fi \
-	  done \
-	fi
-	$(INSTALL_DATA) ChangeLog $(DESTDIR)$(PACKAGE_DOC_DIR)
-	$(INSTALL_DATA) ABOUT-NLS $(DESTDIR)$(PACKAGE_DOC_DIR)
+# 	if test -d $(srcdir)/doc; then \
+# 	  $(mkinstalldirs) $(DESTDIR)$(PACKAGE_DOC_DIR); \
+# 	  for document in $(srcdir)/doc/*.txt; do \
+# 	    if test -f $$document; then \
+# 	      $(INSTALL_DATA) $$document $(DESTDIR)$(PACKAGE_DOC_DIR); \
+# 	    fi \
+# 	  done \
+# 	fi
+# 	if test -d $(srcdir)/doc/txt; then \
+# 	  $(mkinstalldirs) $(DESTDIR)$(PACKAGE_DOC_DIR)/txt; \
+# 	  for document in $(srcdir)/doc/txt/*; do \
+# 	    if test -f $$document; then \
+# 	      $(INSTALL_DATA) $$document $(DESTDIR)$(PACKAGE_DOC_DIR)/txt; \
+# 	    fi \
+# 	  done \
+# 	fi
+# 	if test -d $(srcdir)/doc/html; then \
+# 	  $(mkinstalldirs) $(DESTDIR)$(PACKAGE_DOC_DIR)/html; \
+# 	  for document in $(srcdir)/doc/html/*; do \
+# 	    if test -f $$document; then \
+# 	      $(INSTALL_DATA) $$document $(DESTDIR)$(PACKAGE_DOC_DIR)/html; \
+# 	    fi \
+# 	  done \
+# 	fi
+# 	$(INSTALL_DATA) ChangeLog $(DESTDIR)$(PACKAGE_DOC_DIR)
+# 	$(INSTALL_DATA) ABOUT-NLS $(DESTDIR)$(PACKAGE_DOC_DIR)
 	$(INSTALL_DATA) gyvoice/voice_servers $(DESTDIR)$(pkgdatadir)
 
 uninstall-local:
 	rm -rf $(DESTDIR)$(pkgdatadir)
-	rm -rf $(DESTDIR)$(PACKAGE_DOC_DIR)
+# 	rm -rf $(DESTDIR)$(PACKAGE_DOC_DIR)
 
 dist-hook:
 	if test -d themes; then \
