diff -Naur gyachi-1.2.2.orig/configure.ac.tmpl gyachi-1.2.2/configure.ac.tmpl
--- gyachi-1.2.2.orig/configure.ac.tmpl	2009-08-15 19:13:34.036873132 +0300
+++ gyachi-1.2.2/configure.ac.tmpl	2009-08-15 19:28:13.034867887 +0300
@@ -198,7 +198,7 @@
 
 dnl checks for plugin-gpgme
 if test "$enable_plugin_gpgme" != "no"; then
-	AC_CHECK_HEADER([gpgme.h], ,[AC_MSG_ERROR(cannot find include file gpgme.h. Perhaps you need to install the gpgme development package?)])
+	AC_CHECK_HEADER([gpgme/gpgme.h], ,[AC_MSG_ERROR(cannot find include file gpgme.h. Perhaps you need to install the gpgme development package?)])
 	PLUGIN_DIRS="$PLUGIN_DIRS gpgme"
 else
 	AC_MSG_NOTICE(gpgme plugin is disabled)
diff -Naur gyachi-1.2.2.orig/plugins/gpgme/gyachi-gpgme.c gyachi-1.2.2/plugins/gpgme/gyachi-gpgme.c
--- gyachi-1.2.2.orig/plugins/gpgme/gyachi-gpgme.c	2009-08-15 19:13:34.048864811 +0300
+++ gyachi-1.2.2/plugins/gpgme/gyachi-gpgme.c	2009-08-15 19:30:52.145742536 +0300
@@ -24,7 +24,7 @@
  *****************************************************************************/
 
 #include "plugin_api.h"
-#include <gpgme.h>
+#include <gpgme/gpgme.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
