diff -Naur gyachi-1.1.63.orig/client/main.c gyachi-1.1.63/client/main.c
--- gyachi-1.1.63.orig/client/main.c	2009-04-10 00:50:19.000000000 +0300
+++ gyachi-1.1.63/client/main.c	2009-04-10 01:06:24.000000000 +0300
@@ -1218,7 +1218,7 @@
 	}
 
 	if ( ! browser_command ) {
-		browser_command = strdup( "htmlview '%s' &" );
+		browser_command = strdup( "xdg-open '%s' &" );
 	}
 
 	/* an app-specific theme can be stored in ~/.yahoorc/gyach/gtkrc  */  
