Index: krb5-1.7/src/configure.in
===================================================================
--- krb5-1.7.orig/src/configure.in
+++ krb5-1.7/src/configure.in
@@ -1026,9 +1026,9 @@ AC_SUBST(LDAP)
 
 dnl We really should look for and use python-config.
 PYTHON_LIB=
-AC_CHECK_HEADERS(Python.h python2.3/Python.h python2.5/Python.h)
-AC_CHECK_LIB(python2.3,main,[PYTHON_LIB=-lpython2.3],
-  AC_CHECK_LIB(python2.5,main,[PYTHON_LIB=-lpython2.5]))
+AC_CHECK_HEADERS(Python.h python2.5/Python.h python2.7/Python.h)
+AC_CHECK_LIB(python2.5,main,[PYTHON_LIB=-lpython2.5],
+  AC_CHECK_LIB(python2.7,main,[PYTHON_LIB=-lpython2.7]))
 AC_SUBST(PYTHON_LIB)
 
 dnl
