--- configure.in.orig	2014-01-16 02:44:15.000000000 +0200
+++ src/configure.in	2014-05-17 17:05:05.317924895 +0300
@@ -1206,9 +1206,9 @@
 # default, so it's not a big deal that it isn't very good.  We should
 # use python-config instead.
 PYTHON_LIB=
-AC_CHECK_HEADERS(Python.h python2.3/Python.h python2.5/Python.h)
-AC_CHECK_LIB(python2.3,main,[PYTHON_LIB=-lpython2.3],
-  AC_CHECK_LIB(python2.5,main,[PYTHON_LIB=-lpython2.5]))
+AC_CHECK_HEADERS(Python.h python2.5/Python.h python2.7/Python.h)
+AC_CHECK_LIB(python2.5,main,[PYTHON_LIB=-lpython2.5],
+  AC_CHECK_LIB(python2.7,main,[PYTHON_LIB=-lpython2.7]))
 AC_SUBST(PYTHON_LIB)
 
 
