Index: krb5-1.7/src/plugins/kdb/db2/libdb2/test/run.test
===================================================================
--- krb5-1.7.orig/src/plugins/kdb/db2/libdb2/test/run.test
+++ krb5-1.7/src/plugins/kdb/db2/libdb2/test/run.test
@@ -413,9 +413,9 @@ test8()
 	awk 'BEGIN {
 		for (i = 1; i <= 10; ++i) {
 			printf("p\nkkey1\nD/bin/sh\n");
-			printf("p\nkkey2\nD/bin/csh\n");
+			printf("p\nkkey2\nD/bin/bash\n");
 			if (i % 8 == 0) {
-				printf("c\nkkey2\nD/bin/csh\n");
+				printf("c\nkkey2\nD/bin/bash\n");
 				printf("c\nkkey1\nD/bin/sh\n");
 				printf("e\t%d of 10 (comparison)\n", i);
 			} else
