--- scripts/checkelf.old	2014-04-25 17:57:56.000000000 +0200
+++ scripts/checkelf	2014-06-15 15:29:21.932687493 +0200
@@ -150,7 +150,11 @@
     for elf_files, paths_and_deps in result_dependencies.items():
         for data in paths_and_deps:
             if not data[1] == "broken" and not data[1] == package_name and not data[1] == "glibc":
-                result_deps.append(data[1])
+                if (data[1].startswith("xorg-video-nvidia") or data[1].startswith("xorg-video-fglrx")) \
+                    and not (package_name.startswith("xorg-video-nvidia") or package_name.startswith("xorg-video-fglrx")):
+                    result_deps.append("mesa")
+                else:
+                    result_deps.append(data[1])
 
     # remove packages that belong to system.base component
     # when -s is used, systembase is set to true
