diff -ur Makefile~ Makefile
--- Makefile~	2012-10-01 11:52:39.000000000 +0300
+++ Makefile	2013-01-13 20:51:28.636173224 +0200
@@ -1,15 +1,8 @@
 CC	 = gcc
 CFLAGS	 = -g -Wall -Wno-pointer-sign -O2 -fomit-frame-pointer
 
-GIT2LOG := $(shell if [ -x ./git2log ] ; then echo ./git2log --update ; else echo true ; fi)
-GITDEPS := $(shell [ -d .git ] && echo .git/HEAD .git/refs/heads .git/refs/tags)
 
-VERSION := $(shell $(GIT2LOG) --version VERSION ; cat VERSION)
-
-# THEMES	 = $(wildcard themes/*)
-THEMES	 = themes/upstream themes/openSUSE themes/SLES themes/SLED themes/KDE
-
-.PHONY: all clean distclean doc install installsrc themes
+.PHONY: all clean distclean doc install installsrc
 
 all:	changelog bin2c gfxboot-compile bincode gfxboot-font addblack
 
@@ -53,20 +46,9 @@
 	install -m 755 gfxboot~ $(DESTDIR)/usr/sbin/gfxboot
 	install -m 755 gfxtest $(DESTDIR)/usr/sbin
 	install -m 755 gfxboot-compile gfxboot-font $(DESTDIR)/usr/sbin
-	@for i in $(THEMES) ; do \
-	  install -d -m 755 $(DESTDIR)/etc/bootsplash/$$i/{bootloader,cdrom} ; \
-	  cp $$i/bootlogo $(DESTDIR)/etc/bootsplash/$$i/cdrom ; \
-	  bin/unpack_bootlogo $(DESTDIR)/etc/bootsplash/$$i/cdrom ; \
-          install -m 644 $$i/{message,po/*.tr,help-boot/*.hlp} $(DESTDIR)/etc/bootsplash/$$i/bootloader ; \
-	  bin/2hl --link --quiet $(DESTDIR)/etc/bootsplash/$$i/* ; \
-	done
-
+	
 installsrc:
 	install -d -m 755 $(DESTDIR)/usr/share/gfxboot/themes
-	@for i in $(THEMES) ; do \
-	  cp -a $$i $(DESTDIR)/usr/share/gfxboot/themes ; \
-	done
-	cp -a themes/example* $(DESTDIR)/usr/share/gfxboot/themes
 	cp -a bin test $(DESTDIR)/usr/share/gfxboot
 
 clean: themes doc
@@ -76,9 +58,6 @@
 distclean: clean
 	@for i in themes/example* ; do make -C $$i clean || break ; done
 
-themes:
-	@for i in $(THEMES) ; do make -C $$i $(MAKECMDGOALS) || break ; done
-
 doc:
 	make -C doc $(MAKECMDGOALS)
 
