diff -Nur gfxboot-4.1.41-old/Makefile gfxboot-4.1.41/Makefile
--- gfxboot-4.1.41-old/Makefile	2010-01-14 10:55:35.586381078 +0200
+++ gfxboot-4.1.41/Makefile	2010-01-14 10:55:53.513130907 +0200
@@ -3,10 +3,8 @@
 
 CC	 = gcc
 CFLAGS	 = -g -Wall -Wno-pointer-sign -O2 -fomit-frame-pointer
-# THEMES	 = $(wildcard themes/*)
-THEMES	 = themes/upstream themes/openSUSE themes/SLES themes/SLED
 
-.PHONY: all clean distclean doc install installsrc themes
+.PHONY: all clean distclean doc install installsrc
 
 all:	bin2c gfxboot-compile bincode gfxboot-font addblack
 
@@ -47,32 +45,18 @@
 	install -m 755 gfxboot~ $(DESTDIR)/usr/sbin/gfxboot
 	install -m 755 gfxtest $(DESTDIR)/usr/sbin
 	install -m 755 gfxboot-compile gfxboot-font $(DESTDIR)/usr/sbin
-	@for i in $(THEMES) ; do \
-	  install -d -m 755 $(DESTDIR)/etc/bootsplash/$$i/{bootloader,cdrom} ; \
-	  cp $$i/bootlogo $(DESTDIR)/etc/bootsplash/$$i/cdrom ; \
-	  bin/unpack_bootlogo $(DESTDIR)/etc/bootsplash/$$i/cdrom ; \
-          install -m 644 $$i/{message,po/*.tr,help-boot/*.hlp} $(DESTDIR)/etc/bootsplash/$$i/bootloader ; \
-	  bin/2hl --link --quiet $(DESTDIR)/etc/bootsplash/$$i/* ; \
-	done
 
 installsrc:
 	install -d -m 755 $(DESTDIR)/usr/share/gfxboot/themes
-	@for i in $(THEMES) ; do \
-	  cp -a $$i $(DESTDIR)/usr/share/gfxboot/themes ; \
-	done
-	cp -a themes/example* $(DESTDIR)/usr/share/gfxboot/themes
 	cp -a bin test $(DESTDIR)/usr/share/gfxboot
 
-clean: themes doc
+clean: doc
 	@rm -f gfxboot-compile bincode gfxboot-font addblack bincode.h bin2c *.lst *.map vocabulary.inc vocabulary.h *.o *~
 	@rm -rf tmp
 
 distclean: clean
 	@for i in themes/example* ; do make -C $$i clean || break ; done
 
-themes:
-	@for i in $(THEMES) ; do make -C $$i $(MAKECMDGOALS) || break ; done
-
 doc:
 	make -C doc $(MAKECMDGOALS)
 
