--- airstrike-0.99+1.0pre6a/src/sprite/refcount.h	2003-01-18 11:48:05.000000000 -0800
+++ airstrike-0.99+1.0pre6a/src/sprite/refcount.h	2007-04-09 08:31:32.000000000 -0700
@@ -9,7 +9,7 @@
 
 /* freefn will be called with x as argument if the refcount goes below 1 */
 #define DEREF(x,freefn) (x)->refcount -= 2; \
-if ((x)->refcount <= 1) {(freefn)(x);}
+if ((x)->refcount == 1) {(freefn)(x);}
 
 /* Marks the referenced obj as dead */
 #define REFKILL(x) ((x)->refcount |= 1)
