--- airstrike-0.99+1.0pre6a.orig/src/config.h
+++ airstrike-0.99+1.0pre6a/src/config.h
@@ -6,7 +6,7 @@
 /* absolute path to main config file (for now it's airstrikerc in the
    current dir)*/
 #ifndef ROOT_CONFIG_FILE
-#define ROOT_CONFIG_FILE "airstrikerc"
+#define ROOT_CONFIG_FILE "/etc/airstrikerc"
 #endif
 /* path to file in users $HOME */
 #ifndef USER_CONFIG_FILE
--- airstrike-0.99+1.0pre6a.orig/src/airstrike.c
+++ airstrike-0.99+1.0pre6a/src/airstrike.c
@@ -74,7 +74,7 @@
   console_set_pos(9,254);
   console_load_bg(path_to_data("console-bg.png"));
   sprite_types_setup();
-  sprite_background_load("data/bg.png","data/bgmask.png");
+  sprite_background_load("/usr/share/airstrike/bg.png","/usr/share/airstrike/bgmask.png");
 
   level_setup();
   winds_setup();
--- airstrike-0.99+1.0pre6a.orig/Makefile
+++ airstrike-0.99+1.0pre6a/Makefile
@@ -29,3 +29,15 @@
 # packages new files listed in new-files.txt
 new:
 	tar -zcvf new.tgz -T new-files.txt
+
+install:
+	install -d $(DESTDIR)/usr/bin
+	install airstrike $(DESTDIR)/usr/bin/airstrike
+	install -d $(DESTDIR)/usr/share/airstrike
+	install -d $(DESTDIR)/usr/share/airstrike/sound
+	install -m 644 data/*.png $(DESTDIR)/usr/share/airstrike
+	install -m 644 data/*.txt $(DESTDIR)/usr/share/airstrike
+#	install -m 644 data/sound/*.wav $(DESTDIR)/usr/share/airstrike/sound
+	install -d $(DESTDIR)/etc
+	install -m 644 airstrikerc $(DESTDIR)/etc/airstrikerc
+
