--- src/Makefile.orig	2003-01-18 22:02:32.000000000 +0200
+++ src/Makefile	2011-01-23 20:15:04.920000058 +0200
@@ -5,10 +5,10 @@
 OBJECTS:= $(CFILES:.c=.o)
 
 airstrike: $(OBJECTS)
-	$(CC) -o $@ $(CFLAGS) $(OBJECTS) `sdl-config --libs` -lSDL_image
+	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $(OBJECTS) `sdl-config --libs` -lSDL_image -lm
 
 airstrike-sound: $(OBJECTS)
-	$(CC) -o $@ $(CFLAGS) $(OBJECTS) `sdl-config --libs` -lSDL_image -lSDL_mixer
+	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $(OBJECTS) `sdl-config --libs` -lSDL_image -lSDL_mixer -lm
 
 # Use the rule below when coding new sprite types
 #sprites_autogen.h: sprite_types/*.c
