diff -Naur kgmailnotifier-0.5.0.1.orig/src/kgmailnotifier.desktop kgmailnotifier-0.5.0.1/src/kgmailnotifier.desktop
--- kgmailnotifier-0.5.0.1.orig/src/kgmailnotifier.desktop	2009-09-11 16:55:52.172215467 +0300
+++ kgmailnotifier-0.5.0.1/src/kgmailnotifier.desktop	2009-09-11 16:58:33.521213385 +0300
@@ -1,6 +1,8 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=KGmailNotifier
+GenericName=Gmail Notifier
+GenericName[tr]=Gmail Uygulaması
 Exec=kgmailnotifier
 Icon=kgmailnotifier
 Type=Application
@@ -21,7 +23,7 @@
 Comment[sl]=Gmail obveščevalec za KDE
 Comment[sr]=Аплет за GMail обавештавање у КДЕ
 Comment[sr@Latn]=Aplet za GMail obaveštavanje u KDE
-Comment[tr]=KDE için bir GMail uyarıcı parçacığı
+Comment[tr]=KDE için GMail bildirim uygulaması
 
 X-DCOP-ServiceType=Unique
 Categories=Qt;KDE;Network;X-KDE-Internet;
