Index: gscan2pdf-0.9.32/gscan2pdf.desktop
===================================================================
--- gscan2pdf-0.9.32.orig/gscan2pdf.desktop
+++ gscan2pdf-0.9.32/gscan2pdf.desktop
@@ -1,8 +1,10 @@
 [Desktop Entry]
 Name=gscan2pdf
 GenericName=gscan2pdf
+GenericName[tr]=Tarayıcı çıktılarını PDF'ye dönüştürücü
 Comment=A GUI to aid the scan-to-PDF process
 Comment[de]=Machen Sie PDF-Dateien direkt von Ihrem Scanner
+Comment[tr]=Taramaları PDF'ye dönüştürmek için bir Grafik Arayüz
 Exec=gscan2pdf
 Icon=gscan2pdf
 Terminal=false
