diff -up gscan2pdf-0.9.32/bin/gscan2pdf.chdorb gscan2pdf-0.9.32/bin/gscan2pdf
--- gscan2pdf-0.9.32/bin/gscan2pdf.chdorb	2011-03-25 08:38:49.874013074 +0100
+++ gscan2pdf-0.9.32/bin/gscan2pdf	2011-03-26 12:48:07.388794297 +0100
@@ -11091,16 +11091,16 @@ sub add_tess_languages {
 
  my $tessdata = `tesseract '' '' -l '' 2>&1`;
  chomp $tessdata;
- $tessdata =~ s/^Unable to load unicharset file //;
- $tessdata =~ s/\/\.unicharset$//;
+ $tessdata =~ s/.*?\//\//;
+ $tessdata =~ s/\/\..*$//;
  $logger->info("Using tessdata at $tessdata");
  my @tesslang;
- for ( glob "$tessdata/*.unicharset" ) {
+ for ( glob "$tessdata/*.*" ) {
 
   # Weed out the empty language files
   if ( not -z $_ ) {
    my $code;
-   $code = $1 if ( $_ =~ /([\w\-]*)\.unicharset$/ );
+   $code = $1 if ( $_ =~ /([\w\-]*)\..*$/ );
    $logger->info("Found tesseract language $code");
    if ( defined $iso639{$code} ) {
     push @tesslang, [ $code, $iso639{$code} ];
