diff -Naur gscan2pdf-0.9.31-pristine/bin/gscan2pdf gscan2pdf-0.9.31/bin/gscan2pdf
--- gscan2pdf-0.9.31-pristine/bin/gscan2pdf	2010-07-12 07:37:05.000000000 -0600
+++ gscan2pdf-0.9.31/bin/gscan2pdf	2010-08-08 09:05:21.294732961 -0600
@@ -4224,7 +4224,7 @@
     and $ENV{GNOME_DESKTOP_SESSION_ID} ne '' )
    {
     $client =
-`gconftool --get /desktop/gnome/url-handlers/mailto/command | cut -d ' ' -f 1`;
+`gconftool-2 --get /desktop/gnome/url-handlers/mailto/command | cut -d ' ' -f 1`;
    }
 
    my $mua_string;
