diff -Nuar a/clutter-qt/Makefile.am b/clutter-qt/Makefile.am
--- a/clutter-qt/Makefile.am	2013-10-11 19:39:49.000000000 +0300
+++ b/clutter-qt/Makefile.am	2013-10-11 19:43:53.376779990 +0300
@@ -23,7 +23,7 @@
 libclutter_qt_@CLUTTER_QT_API_VERSION@_la_LIBADD  = $(CLUTTER_LIBS) $(QT_LIBS)
 libclutter_qt_@CLUTTER_QT_API_VERSION@_la_LDFLAGS = $(CLUTTER_LT_LDFLAGS)
 
-clutterqtheadersdir = $(includedir)/clutter-@CLUTTER_QT_API_VERSION@/clutter-qt
+clutterqtheadersdir = $(includedir)/clutter-1.0/clutter-qt
 clutterqtheaders_HEADERS = clutter-qt.h
 
 moc_clutter-qt.cc: clutter-qt.h
diff -Nuar a/clutter-qt.pc.in b/clutter-qt.pc.in
--- a/clutter-qt.pc.in	2013-10-11 19:39:49.000000000 +0300
+++ b/clutter-qt.pc.in	2013-10-11 19:48:00.884771421 +0300
@@ -7,5 +7,5 @@
 Description: QT widget for Clutter
 Version: @VERSION@
 Libs: -L${libdir} -lclutter-qt-@CLUTTER_QT_API_VERSION@
-Cflags: -I${includedir}/clutter-@CLUTTER_QT_API_VERSION@
-Requires: QtGui clutter-x11-@CLUTTER_QT_API_VERSION@
+Cflags: -I${includedir}/clutter-1.0/clutter-qt
+Requires: QtGui clutter-x11-1.0
