diff -up network-manager-applet-0.9.3.997/configure.ac.foo network-manager-applet-0.9.3.997/configure.ac
--- network-manager-applet-0.9.3.997/configure.ac.foo	2011-11-10 15:10:45.268107655 -0600
+++ network-manager-applet-0.9.3.997/configure.ac	2011-11-10 15:10:51.902024720 -0600
@@ -179,8 +179,6 @@ dnl
 dnl Compiler flags
 dnl
 NM_COMPILER_WARNINGS
-# Use --enable-maintainer-mode to disabled deprecated symbols
-GNOME_MAINTAINER_MODE_DEFINES
 
 
 AC_CONFIG_FILES([
