Index: network-manager-applet-0.8.2/src/connection-editor/nm-connection-editor.c
===================================================================
--- network-manager-applet-0.8.2.orig/src/connection-editor/nm-connection-editor.c
+++ network-manager-applet-0.8.2/src/connection-editor/nm-connection-editor.c
@@ -729,21 +729,21 @@ nm_connection_editor_set_connection (NMC
 
 	connection_type = nm_setting_connection_get_connection_type (s_con);
 	if (!strcmp (connection_type, NM_SETTING_WIRED_SETTING_NAME)) {
+		if (!add_page (editor, ce_page_ip4_new, editor->connection, error))
+			goto out;
 		if (!add_page (editor, ce_page_wired_new, editor->connection, error))
 			goto out;
 		if (!add_page (editor, ce_page_wired_security_new, editor->connection, error))
 			goto out;
-		if (!add_page (editor, ce_page_ip4_new, editor->connection, error))
-			goto out;
 		if (!add_page (editor, ce_page_ip6_new, editor->connection, error))
 			goto out;
 	} else if (!strcmp (connection_type, NM_SETTING_WIRELESS_SETTING_NAME)) {
+		if (!add_page (editor, ce_page_ip4_new, editor->connection, error))
+			goto out;
 		if (!add_page (editor, ce_page_wireless_new, editor->connection, error))
 			goto out;
 		if (!add_page (editor, ce_page_wireless_security_new, editor->connection, error))
 			goto out;
-		if (!add_page (editor, ce_page_ip4_new, editor->connection, error))
-			goto out;
 		if (!add_page (editor, ce_page_ip6_new, editor->connection, error))
 			goto out;
 	} else if (!strcmp (connection_type, NM_SETTING_VPN_SETTING_NAME)) {
