diff -up network-manager-applet-0.9.3.997/nm-applet.schemas.in.foo network-manager-applet-0.9.3.997/nm-applet.schemas.in
--- network-manager-applet-0.9.3.997/nm-applet.schemas.in.foo	2009-10-21 17:23:29.480267635 -0700
+++ network-manager-applet-0.9.3.997/nm-applet.schemas.in	2009-10-21 17:39:42.734142996 -0700
@@ -6,7 +6,7 @@
       <applyto>/apps/nm-applet/disable-connected-notifications</applyto>
       <owner>nm-applet</owner>
       <type>bool</type>
-      <default>FALSE</default>
+      <default>TRUE</default>
       <locale name="C">
         <short>Disable connected notifications</short>
         <long>
@@ -20,7 +20,7 @@
       <applyto>/apps/nm-applet/disable-disconnected-notifications</applyto>
       <owner>nm-applet</owner>
       <type>bool</type>
-      <default>FALSE</default>
+      <default>TRUE</default>
       <locale name="C">
         <short>Disable disconnected notifications</short>
         <long>
@@ -34,7 +34,7 @@
       <applyto>/apps/nm-applet/suppress-wireless-networks-available</applyto>
       <owner>nm-applet</owner>
       <type>bool</type>
-      <default>FALSE</default>
+      <default>TRUE</default>
       <locale name="C">
         <short>Suppress networks available notifications</short>
         <long>
