diff --git network-manager-applet-0.9.3.997/src/libnm-gtk/nm-wireless-dialog.c network-manager-applet-0.9.3.997/src/libnm-gtk/nm-wireless-dialog.c
index 2cd75e0..5ae4764 100644
--- network-manager-applet-0.9.3.997/src/libnm-gtk/nm-wireless-dialog.c
+++ network-manager-applet-0.9.3.997/src/libnm-gtk/nm-wireless-dialog.c
@@ -1151,24 +1151,28 @@ internal_init (NMAWirelessDialog *self,
 
 		tmp = g_strdup_printf (_("Passwords or encryption keys are required to access the wireless network '%s'."),
 		                       esc_ssid ? esc_ssid : "<unknown>");
-		gtk_window_set_title (GTK_WINDOW (self), _("Wireless Network Authentication Required"));
 		label = g_strdup_printf ("<span size=\"larger\" weight=\"bold\">%s</span>\n\n%s",
 		                         _("Authentication required by wireless network"),
 		                         tmp);
 		g_free (esc_ssid);
 		g_free (tmp);
 	} else if (priv->adhoc_create) {
-		gtk_window_set_title (GTK_WINDOW (self), _("Create New Wireless Network"));
 		label = g_strdup_printf ("<span size=\"larger\" weight=\"bold\">%s</span>\n\n%s",
 		                         _("New wireless network"),
 		                         _("Enter a name for the wireless network you wish to create."));
 	} else {
-		gtk_window_set_title (GTK_WINDOW (self), _("Connect to Hidden Wireless Network"));
 		label = g_strdup_printf ("<span size=\"larger\" weight=\"bold\">%s</span>\n\n%s",
 		                         _("Hidden wireless network"),
 		                         _("Enter the name and security details of the hidden wireless network you wish to connect to."));
 	}
 
+	/* The primary text should always tell the story, not the title.  And since
+	 * the dialog is always on top, we can skip the taskbar.
+	 */
+	gtk_window_set_title (GTK_WINDOW (self), "");
+	gtk_window_set_keep_above (GTK_WINDOW (self), TRUE);
+	gtk_window_set_skip_taskbar_hint (GTK_WINDOW (self), TRUE);
+
 	widget = GTK_WIDGET (gtk_builder_get_object (priv->builder, "caption_label"));
 	gtk_label_set_markup (GTK_LABEL (widget), label);
 	g_free (label);
