Index: network-manager-applet-0.8.2/src/applet.c
===================================================================
--- network-manager-applet-0.8.2.orig/src/applet.c
+++ network-manager-applet-0.8.2/src/applet.c
@@ -2979,15 +2979,6 @@ exit_cb (GObject *ignored, gpointer user
 	g_main_loop_quit (applet->loop);
 }
 
-static void
-applet_embedded_cb (GObject *object, GParamSpec *pspec, gpointer user_data)
-{
-	gboolean embedded = gtk_status_icon_is_embedded (GTK_STATUS_ICON (object));
-
-	g_message ("applet now %s the notification area",
-	           embedded ? "embedded in" : "removed from");
-}
-
 static GObject *
 constructor (GType type,
              guint n_props,
@@ -3080,10 +3071,10 @@ constructor (GType type,
 
 	/* Track embedding to help debug issues where user has removed the
 	 * notification area applet from the panel, and thus nm-applet too.
-	 */
 	g_signal_connect (applet->status_icon, "notify::embedded",
 	                  G_CALLBACK (applet_embedded_cb), NULL);
 	applet_embedded_cb (G_OBJECT (applet->status_icon), NULL, NULL);
+	 */
 
 	applet->notify_actions = applet_notify_server_has_actions ();
 
