Index: network-manager-applet-0.8.2/src/connection-editor/nm-connection-editor.c
===================================================================
--- network-manager-applet-0.8.2.orig/src/connection-editor/nm-connection-editor.c
+++ network-manager-applet-0.8.2/src/connection-editor/nm-connection-editor.c
@@ -418,7 +418,7 @@ nm_connection_editor_new (NMConnection *
 
 	editor->ok_button = ce_polkit_button_new (_("_Save"),
 	                                          _("Save any changes made to this connection."),
-	                                          _("_Save..."),
+	                                          _("_Save"),
 	                                          _("Authenticate to save this connection for all users of this machine."),
 	                                          GTK_STOCK_APPLY,
 	                                          settings,
Index: network-manager-applet-0.8.2/src/connection-editor/nm-connection-list.c
===================================================================
--- network-manager-applet-0.8.2.orig/src/connection-editor/nm-connection-list.c
+++ network-manager-applet-0.8.2/src/connection-editor/nm-connection-list.c
@@ -1290,7 +1290,7 @@ add_connection_buttons (NMConnectionList
 	info = action_info_new (self, treeview, GTK_WINDOW (self->dialog), NULL);
 	button = ce_polkit_button_new (_("_Edit"),
 	                               _("Edit the selected connection"),
-	                               _("_Edit..."),
+	                               _("_Edit"),
 	                               _("Authenticate to edit the selected connection"),
 	                               GTK_STOCK_EDIT,
 	                               self->system_settings,
@@ -1308,7 +1308,7 @@ add_connection_buttons (NMConnectionList
 	info = action_info_new (self, treeview, GTK_WINDOW (self->dialog), NULL);
 	button = ce_polkit_button_new (_("_Delete"),
 	                               _("Delete the selected connection"),
-	                               _("_Delete..."),
+	                               _("_Delete"),
 	                               _("Authenticate to delete the selected connection"),
 	                               GTK_STOCK_DELETE,
 	                               self->system_settings,
Index: network-manager-applet-0.8.2/src/wireless-dialog.c
===================================================================
--- network-manager-applet-0.8.2.orig/src/wireless-dialog.c
+++ network-manager-applet-0.8.2/src/wireless-dialog.c
@@ -471,7 +471,7 @@ connection_combo_init (NMAWirelessDialog
 
 		gtk_list_store_append (store, &tree_iter);
 		gtk_list_store_set (store, &tree_iter,
-		                    C_NAME_COLUMN, _("New..."),
+		                    C_NAME_COLUMN, _("New"),
 		                    C_NEW_COLUMN, TRUE, -1);
 
 		gtk_list_store_append (store, &tree_iter);
