diff -Nuar enblend-enfuse-4.1.3.orig/CMakeLists.txt enblend-enfuse-4.1.3/CMakeLists.txt
--- enblend-enfuse-4.1.3.orig/CMakeLists.txt	2013-01-18 11:01:29.000000000 +0200
+++ enblend-enfuse-4.1.3/CMakeLists.txt	2014-06-01 09:45:16.180117386 +0300
@@ -124,7 +124,7 @@
 
 # Compiler specific tweaks and optimizations
 IF(CMAKE_COMPILER_IS_GNUCXX)
-  SET(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -ffast-math -DNDEBUG -Wall")
+  SET(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}  -ffast-math -DNDEBUG -Wall")
   IF(APPLE AND CMAKE_MAJOR_VERSION EQUAL 2 AND CMAKE_MINOR_VERSION LESS 5)
     ## Add needed Compiler and Linker flags for OSX
     SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -flat_namespace -undefined suppress")
diff -Nuar enblend-enfuse-4.1.3.orig/doc/CMakeLists.txt enblend-enfuse-4.1.3/doc/CMakeLists.txt
--- enblend-enfuse-4.1.3.orig/doc/CMakeLists.txt	2013-01-01 11:49:53.000000000 +0200
+++ enblend-enfuse-4.1.3/doc/CMakeLists.txt	2014-06-01 09:45:29.373117879 +0300
@@ -8,7 +8,7 @@
 IF(WIN32)
 set(DocumentsDirectory "doc")
 ELSE(WIN32)
-set(DocumentsDirectory "doc/enblend")
+set(DocumentsDirectory "doc/")
 ENDIF(WIN32)
 
 include(${TOP_SRC_DIR}/doc/macros.cmake)
