diff -up libEMF-1.0.3/include/libEMF/wine/winnt.h.axp libEMF-1.0.3/include/libEMF/wine/winnt.h
--- libEMF-1.0.3/include/libEMF/wine/winnt.h.axp	2007-12-03 22:35:49.000000000 +0100
+++ libEMF-1.0.3/include/libEMF/wine/winnt.h	2007-12-03 22:37:10.000000000 +0100
@@ -671,7 +671,7 @@ typedef struct _CONTEXT CONTEXT;
 #endif
 
 /* Alpha context definitions */
-#ifdef _ALPHA_
+#if defined(__alpha__)
 
 #define CONTEXT_ALPHA   0x00020000
  
