Index: rekonq-0.6.80/src/rekonq.kcfg
===================================================================
--- rekonq-0.6.80.orig/src/rekonq.kcfg
+++ rekonq-0.6.80/src/rekonq.kcfg
@@ -39,10 +39,10 @@
 <!-- New Tab Page Settings -->
 <group name="NewTabPage">
     <entry name="previewNames" type="StringList">
-        <default>KDE Homepage,UserBase,rekonq site</default>
+        <default>Pardus Homepage,Pardus Users Portal</default>
     </entry>
     <entry name="previewUrls" type="StringList">
-        <default>http://www.kde.org,http://userbase.kde.org,http://rekonq.kde.org</default>
+        <default>http://www.pardus-anka.org,http://bugs.pardus-anka.org</default>
     </entry>
 </group>
 
