Index: src/cut.c
===================================================================
--- src/cut.c.orig	2010-05-06 15:16:26.851859241 +0200
+++ src/cut.c	2010-05-06 15:16:27.095859170 +0200
@@ -878,7 +878,10 @@ cut_fields_mb (FILE *stream)
   c = getc (stream);
   empty_input = (c == EOF);
   if (c != EOF)
-    ungetc (c, stream);
+    {
+      ungetc (c, stream);
+      wc = 0;
+    }
   else
     wc = WEOF;
 
