--- parse.c.orig	2007-11-12 22:42:20.000000000 +0200
+++ parse.c	2008-09-20 06:05:03.000000000 +0300
@@ -415,6 +415,8 @@
 		fgets(buffer, bsize, fp);
 	else if((fp = fopen("/etc/turbolinux-release", "r")) != NULL)
 		fgets(buffer, bsize, fp);
+    else if((fp = fopen("/etc/pisilinux-release", "r")) != NULL)
+        fgets(buffer, bsize, fp);
 	else
 		snprintf(buffer, bsize, "Unknown Distro");
 	if(fp != NULL) fclose(fp);
