Index: xchat-xsys-2.2.0/xsys.c
===================================================================
--- xchat-xsys-2.2.0.orig/xsys.c	2008-05-14 22:14:00.000000000 +0200
+++ xchat-xsys-2.2.0/xsys.c	2008-05-14 22:15:15.000000000 +0200
@@ -153,8 +153,11 @@
 	snprintf(buffer, bsize, "%s/xsys2.conf", xchat_get_info(ph, "xchatdirfs"));
 	fp = fopen(buffer, "r");
 	if(fp == NULL)
+	{
 		save_config();
-	
+		fp = fopen(buffer, "r");
+	}
+
 	while(fgets(buffer, bsize, fp) != NULL)
 	{
 		find_match_char(buffer, "format", format);
