Index: xchat-xsys-2.2.0/match.c
===================================================================
--- xchat-xsys-2.2.0.orig/match.c	2008-04-29 20:39:26.000000000 +0200
+++ xchat-xsys-2.2.0/match.c	2008-04-29 20:40:16.000000000 +0200
@@ -69,11 +69,13 @@
 void remove_leading_whitespace(char *buffer)
 {
 	char *pos;
-        while((pos = memchr(buffer, 0x20, 1)))
-	{
-		pos += 1;
-		strcpy(buffer, pos);
-	}
+
+	pos = buffer;
+
+	while (*pos && isspace(*pos))
+		pos++;
+
+	strcpy(buffer, pos);
 }
 
 char *decruft_filename(char *buffer)
