--- /dev/null	2007-12-10 11:21:17.293769627 +0200
+++ tmpwatch.cron.example	2007-12-10 16:56:27.000000000 +0200
@@ -0,0 +1,46 @@
+#
+# ---=== tmpwatch cron file example ===---
+#
+# tmpwatch delete files and directories based on when they were last accessed
+# Please read the directory descriptions carefully, for your safety each of those dirs are disabled in daily cron file.
+# To activate tmpwatch please edit /etc/cron.daily/tmpwatch.cron file.
+
+# ---------------------------------------------------------------------------------------------------------------------
+#   Path variable                                 Description
+# ---------------------------------------------------------------------------------------------------------------------
+
+TMPWATCH="/usr/sbin/tmpwatch"                   # binary file path of tmpwatch
+TMP="/tmp"                                      # tmp directory path
+tmpx11="$TMP/.X11-unix"                         # Where local unix sockets created for screens
+tmpice="$TMP/.ICE-unix"                         # Where X-windows session informations saved
+PISI_ARCH="/var/cache/pisi/archives"            # If you build pisi packages, pisi downloads source code of the package here
+PISI_PKG="/var/cache/pisi/packages"             # When you update or install packages into your system, pisi downloads the packages here
+PISI_TMP="/var/pisi"                            # Where pisi builds the packages
+
+
+# Delete everything in /tmp that hasn't been accessed in 5 days
+if [[ -d ${TMP} ]]; then
+   ${TMPWATCH} -x ${tmpx11} -x ${tmpice} -atime 120 ${TMP}
+fi
+# atime (access time): Make the decision about deleting a file based on the file’s access time
+# 1 day = 24 hrs and 24*5= 120, if you edit this option please know that time value is in HOURS!
+
+
+# Delete everything in PISI_ARCH that hasn't been modified in 2 weeks
+if [[ -d ${PISI_ARCH} ]]; then
+   ${TMPWATCH} --atime 336 ${PISI_ARCH}
+fi
+
+
+# Delete everything in PISI_PKG that hasn't been accessed in 1 months
+if [[ -d $PISI_PKG ]]; then
+   ${TMPWATCH} --atime --mtime 722 ${PISI_PKG}
+fi
+# mtime (modification time): Make the decision about deleting a file based on the file’s modification time instead of the acess time
+
+
+# Delete everything in PISI_TMP that hasn't been accessed in a week
+if [[ -d ${PISI_TMP} ]]; then
+   ${TMPWATCH} --atime --all 168 ${PISI_TMP}
+fi
+# all: Remove all file types, not just regular files, symbolic links and directories
