Index: qwt-5.2.2/qwtconfig.pri
===================================================================
--- qwt-5.2.2.orig/qwtconfig.pri
+++ qwt-5.2.2/qwtconfig.pri
@@ -7,10 +7,14 @@ VER_MIN      = 2
 VER_PAT      = 2
 VERSION      = $${VER_MAJ}.$${VER_MIN}.$${VER_PAT}
 
+CONFIG(install-qt) {
+target.path    = $$[QT_INSTALL_LIBS] 
+headers.path   = $$[QT_INSTALL_HEADERS]/qwt
+doc.path       = $$[QT_INSTALL_DOCS] 
+} else {
 unix {
-    INSTALLBASE    = /usr/local/qwt-$$VERSION
+    INSTALLBASE    = /usr
 }
-
 win32 {
     INSTALLBASE    = C:/Qwt-$$VERSION
 }
@@ -18,6 +22,7 @@ win32 {
 target.path    = $$INSTALLBASE/lib
 headers.path   = $$INSTALLBASE/include
 doc.path       = $$INSTALLBASE/doc
+}
 
 ######################################################################
 # qmake internal options
Index: qwt-5.2.2/src/src.pro
===================================================================
--- qwt-5.2.2.orig/src/src.pro
+++ qwt-5.2.2/src/src.pro
@@ -220,9 +220,17 @@ contains(CONFIG, QwtWidgets) {
 # Install directives
 
 headers.files  = $$HEADERS
-doc.files      = $${QWT_ROOT}/doc/html $${QWT_ROOT}/doc/qwt-5.2.0.qch
-unix {
-    doc.files      += $${QWT_ROOT}/doc/man
-}
+INSTALLS      += target headers
+
+html.files     = $${QWT_ROOT}/doc/html
+html.path      = $$[QT_INSTALL_DOCS]/html/
+INSTALLS      += html
+
+qch.files      = $${QWT_ROOT}/doc/qwt-5.2.0.qch
+qch.path       = $$[QT_INSTALL_DOCS]/qch
+INSTALLS      += qch 
+
+unix:man.files += $${QWT_ROOT}/doc/man
+unix:man.path  = /usr/share
+unix:INSTALLS  += man
 
-INSTALLS       = target headers doc
