--- ufoai-2.4-source/build/install.mk.missing-shebang	2012-05-04 01:40:19.494863757 +0200
+++ ufoai-2.4-source/build/install.mk	2012-05-04 01:41:15.509859142 +0200
@@ -77,6 +77,7 @@
 	@echo "#!/bin/sh" > ufoded.sh
 	@echo "cd $(PKGDATADIR); ./ufoded \$$*; exit \$$?" >> ufoded.sh
 	$(Q)$(INSTALL_SCRIPT) ufoded.sh $(DESTDIR)$(PKGBINDIR)/ufoded
+	@echo "#!/bin/sh" > uforadiant.sh
 	@echo "cd $(PKGDATADIR)/radiant; ./uforadiant \$$*; exit \$$?" >> uforadiant.sh
 	$(Q)$(INSTALL_SCRIPT) uforadiant.sh $(DESTDIR)$(PKGBINDIR)/uforadiant
 	$(Q)rm ufoded.sh ufo.sh uforadiant.sh
