--- build/modules/game.mk.link	2012-09-27 11:16:00.000000000 +0800
+++ build/modules/game.mk	2012-09-27 11:16:20.000000000 +0800
@@ -8,7 +8,7 @@
 
 $(TARGET)_LINKER   := $(CC)
 $(TARGET)_CFLAGS   += -DCOMPILE_UFO $(SO_CFLAGS)
-$(TARGET)_LDFLAGS  += $(SO_LDFLAGS) -lm
+$(TARGET)_LDFLAGS  += $(SO_LDFLAGS) -lm -ldl
 $(TARGET)_FILE     := base/$(TARGET).$(SO_EXT)
 
 # Lua apicheck adds asserts to make sure stack is sane
