--- tvtime/src/videoinput.c	2006-01-06 10:04:14.000000000 +0100
+++ tvtime/src/videoinput.c	2006-01-06 10:03:59.000000000 +0100
@@ -377,6 +377,7 @@
         wait_for_frame_v4l2( vidin );
  
         cur_buf.type = vidin->capbuffers[ 0 ].vidbuf.type;
+        cur_buf.memory = vidin->capbuffers[ 0 ].vidbuf.memory;
         if( ioctl( vidin->grab_fd, VIDIOC_DQBUF, &cur_buf ) < 0 ) {
             /* some drivers return EIO when there is no signal */
             if( errno != EIO ) {
