--- a/src/videoinput.c~	2012-12-09 15:37:11.450681356 +0100
+++ b/src/videoinput.c	2012-12-09 15:37:37.649693293 +0100
@@ -35,8 +35,9 @@
 #ifdef HAVE_CONFIG_H
 # include "config.h"
 #endif
-#include "videodev.h"
-#include "videodev2.h"
+#include <libv4l1-videodev.h>
+#include <linux/videodev2.h>
+#define BASE_VIDIOCPRIVATE      192             /* 192-255 are private */
 #include "videoinput.h"
 #include "mixer.h"
 
