--- kiwi-1.9.26/kiwi/dist.py.orig	2009-12-14 23:37:15.000000000 +0100
+++ kiwi-1.9.26/kiwi/dist.py	2009-12-14 23:41:53.000000000 +0100
@@ -156,14 +156,10 @@ def get_site_packages_dir(*dirs):
 
     @param dirs: directory names to be appended
     """
-    python_version = sys.version_info[:2]
+
     libdir = get_python_lib(plat_specific=False,
                             standard_lib=True, prefix='')
-    if python_version < (2, 6):
-        site = 'site-packages'
-    else:
-        site = 'dist-packages'
-    return os.path.join(libdir, site, *dirs)
+    return os.path.join(libdir, 'site-packages', *dirs)
 
 def listfiles(*dirs):
     """
