Index: kiwi-1.9.29/setup.py
===================================================================
--- kiwi-1.9.29.orig/setup.py     2012-11-13 21:35:41.454383631 +0200
+++ kiwi-1.9.29/setup.py          2012-11-13 21:35:41.473383589 +0200
@@ -56,12 +56,8 @@
                   ('$libdir/glade3/modules', ['kiwiwidgets.py']),
                   ('share/glade3/pixmaps', pixmaps),
                   # Documentation
-                  ('share/doc/kiwi',
+                  ('share/doc/python-kiwi',
                    ('AUTHORS', 'NEWS', 'README')),
-                  ('share/doc/kiwi/howto',
-                   listfiles('doc/howto/', '*')),
-                  ('share/doc/kiwi/api',
-                   listfiles('doc/api/', '*')),
                   ],
       scripts=['bin/kiwi-i18n',
                'bin/kiwi-ui-test'],
