diff -Naur dssi-1.0.0.orig/examples/Makefile.am dssi-1.0.0/examples/Makefile.am
--- dssi-1.0.0.orig/examples/Makefile.am	2009-06-06 16:09:46.000000000 +0300
+++ dssi-1.0.0/examples/Makefile.am	2009-06-06 18:02:15.000000000 +0300
@@ -45,7 +45,7 @@
 trivial_sampler_la_CFLAGS = -I$(top_srcdir)/dssi $(AM_CFLAGS) $(ALSA_CFLAGS) $(SNDFILE_CFLAGS) $(SRC_CFLAGS)
 karplong_la_CXXFLAGS = -I$(top_srcdir)/dssi $(AM_CFLAGS) $(ALSA_CFLAGS)
 
-trivial_synth_la_LDFLAGS = -module -avoid-version
+trivial_synth_la_LDFLAGS = -module -avoid-version -lm
 less_trivial_synth_la_LDFLAGS = -module -avoid-version
 trivial_sampler_la_LDFLAGS = -module -avoid-version
 karplong_la_LDFLAGS = -module -avoid-version
