--- libplot/g_range.c
+++ libplot/g_range.c	2006-05-22 14:59:13.000000000 +0200
@@ -172,7 +172,7 @@
 void
 _set_line_join_bbox (plOutbuf *bufp, double xleft, double yleft, double x, double y, double xright, double yright, double linewidth, int joinstyle, double miterlimit, double m[6])
 {
-  plVector v1, v2, vsum;
+  plVector v1 = {0.0, 0.0}, v2 = {0.0, 0.0}, vsum;
   double v1len, v2len;
   double halfwidth;
   double mitrelen;
