diff -ur avogadro-1.0.3/CMakeLists.txt avogadro-1.0.3-no-strip/CMakeLists.txt
--- avogadro-1.0.3/CMakeLists.txt	2011-04-25 07:29:58.000000000 +0200
+++ avogadro-1.0.3-no-strip/CMakeLists.txt	2011-04-27 18:44:02.000000000 +0200
@@ -98,7 +98,7 @@
 	endif (NOT APPLE)
   # Set up additional build flags for particular build types. These will be added to CMAKE_CXX_FLAGS,
   # they will not replace them.
-  set(CMAKE_CXX_FLAGS_RELEASE "-O2 -DNDEBUG -DQT_NO_DEBUG_OUTPUT -Wl,-s")
+  set(CMAKE_CXX_FLAGS_RELEASE "-O2 -DNDEBUG -DQT_NO_DEBUG_OUTPUT")
   set(CMAKE_CXX_FLAGS_RELWITHDEBINFO
     "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} ${CMAKE_CXX_FLAGS_WARN}")
   set(CMAKE_CXX_FLAGS_DEBUG   "-g3 -fno-inline ${CMAKE_CXX_FLAGS_WARN}")
