--- amsn-0.98.4.orig/plugins.tcl	2010-12-29 23:12:57.660000249 +0200
+++ amsn-0.98.4/plugins.tcl	2010-12-29 23:52:39.175000248 +0200
@@ -1313,7 +1313,7 @@
 			#logic: if this is the first time the user logged in, then 
 			#the configuration list is empty which means load_config
 			#makes loadedplugins empty
-			set loadedplugins [list "Nudge" "Winks" "Cam Shooter" "remind" "Notes" "ColoredNicks" "Inkdraw"]
+			set loadedplugins [list "Nudge" "Winks" "Cam Shooter" "remind" "Notes" "ColoredNicks" "Inkdraw" "Desktop Integration"]
 
 			status_log "Plugins System: load_config: No plugins.xml]\n" red
 		}
