--- amsn-0.98.4.orig/Makefile.in	2010-12-29 23:12:57.664000249 +0200
+++ amsn-0.98.4/Makefile.in	2010-12-30 00:32:16.461000256 +0200
@@ -142,7 +142,7 @@
 ifeq (@FOUND_OS@,mac)
   LDLIBS  := @LDLIBS@ -lz -lm
 else
-  LDLIBS  := @LDLIBS@ -lpng -lz -lm -ljpeg
+  LDLIBS  := @LDLIBS@ -lpng -lz -lm -ljpeg -ldl
 endif
 LDFLAGS := @LDFLAGS@
 
@@ -165,7 +160,7 @@
 LDLIBS	 += @LIBV4L_LIBS@
 
 ifeq (@HAVE_GUPNP@,yes)
-  LDLIBS += @GUPNP_LIBS@
+  LDLIBS += @GUPNP_LIBS@ -lgthread-2.0
   FILES_TO_INSTALL += utils/gupnp/gupnp.$(SHLIB_EXTENSION) utils/gupnp/pkgIndex.tcl
 endif
 
--- amsn-0.98.4.orig/configure.ac	2010-12-29 23:12:57.561000249 +0200
+++ amsn-0.98.4/configure.ac	2010-12-30 00:39:28.642000243 +0200
@@ -237,7 +237,7 @@
 	if test "x${framework}" = "x"; then
 	TCL_LIB_SPEC="-L${TCL_LIB_DIR} -l${TCL_BASE}"
 	fi
-	TCL_STUB_LIB_SPEC="-L${TCL_LIB_DIR} -l${TCL_STUB_BASE}"
+	TCL_STUB_LIB_SPEC="-L${TCL_LIB_DIR} -l${TCL_BASE}"
 	eval "TCL_UNSHARED_LIB_FILE=lib${TCL_BASE}${TCL_UNSHARED_LIB_SUFFIX}"
 	eval "TCL_SHARED_LIB_FILE=lib${TCL_BASE}${TCL_SHARED_LIB_SUFFIX}"
 	
@@ -310,6 +310,7 @@
 	
 	file=$TK_LIB_DIR/tkConfig.sh
 	. $file
+    TK_STUB_LIB_SPEC="${TK_LIB_SPEC}"
 	AC_SUBST(TK_VERSION)
 	AC_SUBST(TK_LIB)
 	AC_SUBST(TK_LIBS)

--- amsn-0.98.4.orig/utils/TkCximage/src/Rules.mk	2010-12-29 23:12:57.652000249 +0200
+++ amsn-0.98.4/utils/TkCximage/src/Rules.mk	2010-12-30 00:44:03.718000254 +0200
@@ -11,7 +11,7 @@
 
 TARGETS-TkCximage := $(tkcximage_dir)/src/TkCximage.cpp.$(SHLIB_EXTENSION)
 
-$(TARGETS-TkCximage):: $(OBJS-TkCximage) $(EXTRAOBJS-TkCximage)
+$(TARGETS-TkCximage):: $(OBJS-TkCximage) $(EXTRAOBJS-TkCximage) -lX11
 
 all:: $(TARGETS-TkCximage)
 
--- amsn-0.98.4.orig/utils/linux/capture/libng/plugins/Rules.mk	2010-12-29 23:12:57.634000249 +0200
+++ amsn-0.98.4/utils/linux/capture/libng/plugins/Rules.mk	2010-12-30 00:48:36.329000249 +0200
@@ -23,21 +23,23 @@
 clean-plugins:
 	rm -f $(TARGETS-plugins) $(capture_dir)/libng/plugins/*.o
 
-$(capture_dir)/libng/plugins/sn9c10x.so: $(capture_dir)/libng/plugins/sn9c10x.o
+$(capture_dir)/libng/plugins/sn9c10x.so: $(capture_dir)/libng/plugins/sn9c10x.o $(capture_dir)/libng/libng.a
 
-$(capture_dir)/libng/plugins/conv-mjpeg.so: $(capture_dir)/libng/plugins/conv-mjpeg.o
+$(capture_dir)/libng/plugins/conv-mjpeg.so: $(capture_dir)/libng/plugins/conv-mjpeg.o $(capture_dir)/libng/libng.a
 
-$(capture_dir)/libng/plugins/drv0-bsd.so:   $(capture_dir)/libng/plugins/drv0-bsd.o
+$(capture_dir)/libng/plugins/drv0-bsd.so:   $(capture_dir)/libng/plugins/drv0-bsd.o $(capture_dir)/libng/libng.a
 
 $(capture_dir)/libng/plugins/drv0-v4l2.so: \
 	$(capture_dir)/libng/plugins/drv0-v4l2.o \
 	$(capture_dir)/libng/plugins/struct-v4l2.o \
-	$(capture_dir)/libng/plugins/struct-dump.o
+	$(capture_dir)/libng/plugins/struct-dump.o \
+	$(capture_dir)/libng/libng.a
 
 $(capture_dir)/libng/plugins/drv1-v4l.so: \
 	$(capture_dir)/libng/plugins/drv1-v4l.o \
 	$(capture_dir)/libng/plugins/struct-v4l.o \
-	$(capture_dir)/libng/plugins/struct-dump.o
+	$(capture_dir)/libng/plugins/struct-dump.o \
+	$(capture_dir)/libng/libng.a
 
 $(capture_dir)/libng/plugins/struct-dump.o: $(capture_dir)/structs/struct-dump.c
 	@$(echo_compile_c)
