diff -Nuar asymptote-1.82.orig/settings.cc asymptote-1.82/settings.cc
--- asymptote-1.82.orig/settings.cc	2009-07-14 02:50:26.000000000 +0300
+++ asymptote-1.82/settings.cc	2009-08-01 10:24:27.759078168 +0300
@@ -82,14 +82,14 @@
 bool msdos=false;
 string HOME="HOME";
 const char pathSeparator=':';
-string defaultPSViewer="gv";
+string defaultPSViewer="xdg-open";
 #ifdef __APPLE__
 string defaultPDFViewer="open";
 #else  
-string defaultPDFViewer="acroread";
+string defaultPDFViewer="xdg-open";
 #endif  
 string defaultGhostscript="gs";
-string defaultDisplay="display";
+string defaultDisplay="xdg-open";
 string defaultAnimate="animate";
 string systemDir=ASYMPTOTE_SYSDIR;
 string docdir=ASYMPTOTE_DOCDIR;
