diff -Nur asymptote-1.43.orig/doc/Makefile.in asymptote-1.43/doc/Makefile.in
--- asymptote-1.43.orig/doc/Makefile.in	2008-06-12 10:41:19.000000000 +0300
+++ asymptote-1.43/doc/Makefile.in	2008-07-15 21:44:39.000000000 +0300
@@ -8,7 +8,7 @@
 infodir = $(DESTDIR)@infodir@
 datarootdir = @datarootdir@
 INSTALL = @INSTALL@
-TEXI2DVI = @TEXI2DVI@
+TEXI2DVI = TEX=tex @TEXI2DVI@
 
 export docdir infodir INSTALL
 
