diff -Nuar asymptote-2.24.orig/Makefile.in asymptote-2.24/Makefile.in
--- asymptote-2.24.orig/Makefile.in	2013-07-12 20:34:21.000000000 +0300
+++ asymptote-2.24/Makefile.in	2013-10-20 19:06:56.578022663 +0300
@@ -215,15 +215,13 @@
 	fi
 
 install-asy: asy sty
-	${INSTALL} -d $(bindir) $(asydir) $(GUIdir) $(exampledir) \
+	${INSTALL} -d $(bindir) $(asydir) $(exampledir) \
           $(animationsdir)
 	-${INSTALL} -d $(latexdir)
 	-${INSTALL} -d $(contextdir)
 	${INSTALL} -p -m 755 $(NAME) $(bindir)
 	${INSTALL} -p -m 644 base/*.asy $(addprefix base/,$(EXTRA)) \
 		asy-keywords.el $(asydir)
-	${INSTALL} -p -m 755 GUI/*.py $(GUIdir)
-	ln -sf @datadir@/asymptote/GUI/xasy.py $(bindir)/$(XNAME)
 	${INSTALL} -p -m 644 examples/*.asy $(addprefix examples/,$(EXEXTRA)) \
 	  doc/extra/*.asy $(addprefix doc/,$(DOCEXTRA)) $(exampledir)
 	${INSTALL} -p -m 644 examples/animations/*.asy \
