From 26762c712b0836d923b4ee4c03bdef59b4604fad Mon Sep 17 00:00:00 2001
From: Paulo Cesar Pereira de Andrade <pcpa@mandriva.com.br>
Date: Fri, 14 Dec 2007 18:01:37 -0200
Subject: [PATCH] Import existing patches, that were originally Red Hat fixes for driver problems.

---
 src/ct_driver.c |   26 ++++++++++++++++++++++++++
 1 files changed, 26 insertions(+), 0 deletions(-)

diff --git a/src/ct_driver.c b/src/ct_driver.c
index e905603..bef8ac6 100644
--- a/src/ct_driver.c
+++ b/src/ct_driver.c
@@ -1495,6 +1495,20 @@ chipsPreInitHiQV(ScrnInfoPtr pScrn, int flags)
 			   "rgb bits %d\n", val);
 	}
     }
+    /* FIXME: Disable 2D acceleration on C&T 69000 by default, since it is
+     * reported to be broken, but nobody who has this hardware has narrowed
+     * it down to individual acceleration primitives yet.  This is a Red Hat
+     * workaround for a bug reported in bugzilla at:
+     * https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=74841
+     * Mike A. Harris <mharris@redhat.com>
+     */
+    if (cPtr->Chipset == CHIPS_CT69000 && (cPtr->Flags & ChipsAccelSupport) ) {
+	cPtr->Flags &= ~ChipsAccelSupport;
+	xf86DrvMsg(pScrn->scrnIndex, X_WARNING,
+	    "Acceleration is disabled by default on C&T 69000 as it has been reported\n"
+	    "to be broken: https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=74841\n");
+    }
+
     if ((cPtr->Flags & ChipsAccelSupport) &&
 	(xf86ReturnOptValBool(cPtr->Options, OPTION_NOACCEL, FALSE))) {
 	cPtr->Flags &= ~ChipsAccelSupport;
@@ -1508,6 +1522,18 @@ chipsPreInitHiQV(ScrnInfoPtr pScrn, int flags)
     } else {
 	cPtr->Accel.UseHWCursor = TRUE;
     }
+    /* FIXME: Default to SWcursor on CT65550 as hardware cursor is reported
+     * broken on this chip in Red Hat bugzilla as of XFree86 4.2.99.4:
+     * http://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=82438
+     * Mike A. Harris <mharris@redhat.com>
+     */
+    if (cPtr->Chipset == CHIPS_CT65550) {
+	cPtr->Accel.UseHWCursor = FALSE;
+	xf86DrvMsg(pScrn->scrnIndex,X_INFO,
+            "Defaulting to SWcursor as there are known issues with HWcursor on C&T 65550\n"
+            "You can use Option \"HWcursor\" in the X server config file if desired to\n"
+	    "force hardware cursors to be used anyway\n");
+    }
     if (xf86GetOptValBool(cPtr->Options, OPTION_HW_CURSOR,
 			  &cPtr->Accel.UseHWCursor))
 	from = X_CONFIG;
-- 
1.5.3.5

