Index: vlc-1.1.10/modules/misc/freetype.c
===================================================================
--- vlc-1.1.10.orig/modules/misc/freetype.c
+++ vlc-1.1.10/modules/misc/freetype.c
@@ -55,7 +55,7 @@
 # define DEFAULT_FONT_FILE "/usr/share/fonts/nokia/nosnb.ttf"
 # define DEFAULT_FAMILY "Nokia Sans Bold"
 #else
-# define DEFAULT_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf"
+# define DEFAULT_FONT_FILE "/usr/share/fonts/dejavu/DejaVuSans.ttf"
 # define DEFAULT_FAMILY "Serif Bold"
 #endif
 
