diff -Naur vlc-1.0.0.orig/share/osdmenu/default.cfg vlc-1.0.0/share/osdmenu/default.cfg
--- vlc-1.0.0.orig/share/osdmenu/default.cfg	2009-05-11 02:03:59.000000000 +0300
+++ vlc-1.0.0/share/osdmenu/default.cfg	2009-07-09 10:08:20.826303964 +0300
@@ -1,4 +1,4 @@
-dir share/osdmenu/default
+dir /usr/share/osdmenu/default
 action	key-play-pause (0,0)
 	unselect unselected.png
 	select   selection/play_pause.png
