--- share/vlc.desktop.in.OSmanOS	2013-06-24 21:00:39.000000000 +0300
+++ share/vlc.desktop.in	        2013-11-16 18:49:06.848928443 +0200
@@ -73,6 +73,7 @@
 Name[sv]=Mediaspelaren VLC
 GenericName[sv]=Mediaspelaren
 Comment[sv]=Allmän uppspelare av film och musik
+Comment[tr]=VLC medya oynatıcı
 Name[te]=VLC మాధ్యమ ప్రదర్శకం
 GenericName[te]=మాధ్యమ ప్రదర్శకం
 Comment[te]=మీ బహుళమాధ్యమ ప్రవాహాలను చదువు, బంధించు మరియు ప్రసారం చేయి
