diff -p -up vlc-2.0.4/modules/access/dtv/linux.c.dvb vlc-2.0.4/modules/access/dtv/linux.c
--- vlc-2.0.4/modules/access/dtv/linux.c.dvb	2012-08-16 12:41:35.000000000 +0200
+++ vlc-2.0.4/modules/access/dtv/linux.c	2012-12-12 12:54:48.251713053 +0100
@@ -916,7 +916,12 @@ int dvb_set_isdbs (dvb_device_t *d, uint
         return -1;
     return dvb_set_props (d, 5, DTV_CLEAR, 0, DTV_DELIVERY_SYSTEM, SYS_ISDBS,
                           DTV_FREQUENCY, freq,
-                          DTV_ISDBS_TS_ID, (uint32_t)ts_id);
+#if DVBv5(8)
+                          DTV_STREAM_ID,
+#else
+                          DTV_ISDBS_TS_ID,
+#endif
+                          (uint32_t)ts_id);
 #else
 # warning ISDB-S needs Linux DVB version 5.1 or later.
     msg_Err (d->obj, "ISDB-S support not compiled-in");
