diff -Naur hmmer-3.0b3.orig/src/Makefile.in hmmer-3.0b3/src/Makefile.in
--- hmmer-3.0b3.orig/src/Makefile.in	2009-11-15 16:03:02.000000000 +0200
+++ hmmer-3.0b3/src/Makefile.in	2009-11-17 16:19:13.000000000 +0200
@@ -32,7 +32,7 @@
 	    -I. \
 	    -I${srcdir}
 
-LIBS  += @GROUPHMMERLIBS@ -leasel
+LIBS  += @GROUPHMMERLIBS@ -leasel -lm
 LIBDEP = ./libhmmer.a @HMMERIMPLLIB@ ${ESLDIR}/libeasel.a
 
 PROGS = hmmalign\
