diff -Naur imap-2007e/Makefile imap-2007e.oden/Makefile
--- imap-2007e/Makefile	2008-06-04 20:43:35.000000000 +0200
+++ imap-2007e.oden/Makefile	2009-01-13 23:59:31.000000000 +0100
@@ -144,7 +144,7 @@
 # mbox	if file "mbox" exists on the home directory, automatically moves mail
 #	 from the spool directory to "mbox" and uses "mbox" as INBOX.
 
-EXTRADRIVERS=mbox
+EXTRADRIVERS=
 
 
 # Plaintext password type.  Defines how plaintext password authentication is
diff -Naur imap-2007e/src/osdep/unix/Makefile imap-2007e.oden/src/osdep/unix/Makefile
--- imap-2007e/src/osdep/unix/Makefile	2009-01-13 23:59:14.000000000 +0100
+++ imap-2007e.oden/src/osdep/unix/Makefile	2009-01-13 23:59:31.000000000 +0100
@@ -27,7 +27,7 @@
 # Command line build parameters
 
 EXTRAAUTHENTICATORS=
-EXTRADRIVERS=mbox
+EXTRADRIVERS=
 PASSWDTYPE=std
 SSLTYPE=nopwd
 IP=4
