diff -Naur imap-2007e.orig/src/osdep/unix/Makefile imap-2007e/src/osdep/unix/Makefile
--- imap-2007e.orig/src/osdep/unix/Makefile	2011-02-01 10:25:01.865074982 +0200
+++ imap-2007e/src/osdep/unix/Makefile	2011-02-01 10:23:35.650187032 +0200
@@ -985,6 +985,9 @@
 	 -DLOCKPGM3=\"$(LOCKPGM3)\" > OSCFLAGS
 	echo $(BASELDFLAGS) $(EXTRALDFLAGS) > LDFLAGS
 	echo "$(ARRC) $(ARCHIVE) $(BINARIES);$(RANLIB) $(ARCHIVE)" > ARCHIVE
+	echo "`$(CAT) CCTYPE` `$(CAT) CFLAGS` `$(CAT) OSFLAGS` -shared \
+	 -Wl,-soname,libc-client.so.1 -o libc-client.so.1.0.0 $(BINARIES)" \
+	 >> ARCHIVE
 	echo $(OS) > OSTYPE
 	./drivers $(EXTRADRIVERS) $(DEFAULTDRIVERS) dummy
 	./mkauths $(EXTRAAUTHENTICATORS) $(DEFAULTAUTHENTICATORS)
