diff -up gnutls-2.8.6/doc/examples/Makefile.am.link gnutls-2.8.6/doc/examples/Makefile.am
--- gnutls-2.8.6/doc/examples/Makefile.am.link	2010-01-24 11:06:21.000000000 +0100
+++ gnutls-2.8.6/doc/examples/Makefile.am	2010-05-12 21:22:51.000000000 +0200
@@ -30,7 +30,7 @@ LDADD = libexamples.la				\
 	../../lib/libgnutls.la			\
 	../../libextra/libgnutls-extra.la	\
 	../../gl/libgnu.la			\
-	$(LIBSOCKET) $(INET_NTOP_LIB) $(INET_PTON_LIB)
+	$(LTLIBGCRYPT) $(LIBSOCKET) $(INET_NTOP_LIB) $(INET_PTON_LIB)
 
 CXX_LDADD = $(LDADD) \
 	../../lib/libgnutlsxx.la
diff -up gnutls-2.8.6/doc/examples/Makefile.in.link gnutls-2.8.6/doc/examples/Makefile.in
--- gnutls-2.8.6/doc/examples/Makefile.in.link	2010-03-15 11:29:19.000000000 +0100
+++ gnutls-2.8.6/doc/examples/Makefile.in	2010-05-12 21:23:25.000000000 +0200
@@ -827,7 +827,7 @@ LDADD = libexamples.la				\
 	../../lib/libgnutls.la			\
 	../../libextra/libgnutls-extra.la	\
 	../../gl/libgnu.la			\
-	$(LIBSOCKET) $(INET_NTOP_LIB) $(INET_PTON_LIB)
+	$(LTLIBGCRYPT) $(LIBSOCKET) $(INET_NTOP_LIB) $(INET_PTON_LIB)
 
 CXX_LDADD = $(LDADD) \
 	../../lib/libgnutlsxx.la
