--- binutils-2.16.91.0.3/bfd/elflink.c.orig	2005-08-22 21:27:41.000000000 +0200
+++ binutils-2.16.91.0.3/bfd/elflink.c	2005-09-04 11:19:23.781924920 +0200
@@ -7089,7 +7089,7 @@
 		      BFD_ASSERT (r_symndx != 0);
 		      if (action & COMPLAIN)
 			(*finfo->info->callbacks->einfo)
-			  (_("%X`%s' referenced in section `%A' of %B: "
+			  (_("`%s' referenced in section `%A' of %B: "
 			     "defined in discarded section `%A' of %B\n"),
 			   sym_name, o, input_bfd, sec, sec->owner);
 
