diff -Nur binutils-2.20.51.0.10-old/bfd/Makefile.in binutils-2.20.51.0.10/bfd/Makefile.in
--- binutils-2.20.51.0.10-old/bfd/Makefile.in	2010-07-20 16:18:43.559557724 +0300
+++ binutils-2.20.51.0.10/bfd/Makefile.in	2010-07-20 16:20:38.592620133 +0300
@@ -324,7 +324,7 @@
 CSEARCH = -I. -I$(srcdir) -I$(INCDIR)
 SUBDIRS = doc po
 bfddocdir = doc
-libbfd_la_LDFLAGS = $(am__append_1) -release `cat libtool-soversion` \
+libbfd_la_LDFLAGS = -Wl,-Bsymbolic-functions $(am__append_1) -release `cat libtool-soversion` \
 	@SHARED_LDFLAGS@ $(am__empty)
 @INSTALL_LIBBFD_TRUE@bfdlib_LTLIBRARIES = libbfd.la
 @INSTALL_LIBBFD_TRUE@bfdinclude_HEADERS = $(BFD_H) $(INCDIR)/ansidecl.h $(INCDIR)/symcat.h $(INCDIR)/bfdlink.h
diff -Nur binutils-2.20.51.0.10-old/opcodes/Makefile.in binutils-2.20.51.0.10/opcodes/Makefile.in
--- binutils-2.20.51.0.10-old/opcodes/Makefile.in	2010-07-20 16:18:44.441682617 +0300
+++ binutils-2.20.51.0.10/opcodes/Makefile.in	2010-07-20 16:20:50.270619981 +0300
@@ -298,7 +298,7 @@
 LINK_FOR_BUILD = $(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) \
 	$(LDFLAGS_FOR_BUILD) -o $@
 
-libopcodes_la_LDFLAGS = $(am__append_1) -release `cat \
+libopcodes_la_LDFLAGS = -Wl,-Bsymbolic-functions $(am__append_1) -release `cat \
 	../bfd/libtool-soversion` @SHARED_LDFLAGS@
 @INSTALL_LIBBFD_TRUE@bfdlib_LTLIBRARIES = libopcodes.la
 @INSTALL_LIBBFD_TRUE@bfdinclude_DATA = $(INCDIR)/dis-asm.h
