Index: binutils-2.20.0/ld/ld.texinfo
===================================================================
--- binutils-2.20.0.orig/ld/ld.texinfo	2009-11-23 14:02:48.000000000 +0100
+++ binutils-2.20.0/ld/ld.texinfo	2009-11-23 14:03:08.000000000 +0100
@@ -2026,7 +2026,7 @@ This linker can create the new dynamic t
 systems may not understand them. If you specify
 @option{--enable-new-dtags}, the dynamic tags will be created as needed.
 If you specify @option{--disable-new-dtags}, no new dynamic tags will be
-created. By default, the new dynamic tags are not created. Note that
+created. By default, the new dynamic tags are created. Note that
 those options are only available for ELF systems.
 
 @kindex --hash-size=@var{number}
Index: binutils-2.20.0/ld/ldmain.c
===================================================================
--- binutils-2.20.0.orig/ld/ldmain.c	2009-11-23 14:02:48.000000000 +0100
+++ binutils-2.20.0/ld/ldmain.c	2009-11-23 14:03:08.000000000 +0100
@@ -262,6 +262,7 @@ main (int argc, char **argv)
 
   link_info.allow_undefined_version = TRUE;
   link_info.keep_memory = TRUE;
+  link_info.new_dtags = TRUE;
   link_info.combreloc = TRUE;
   link_info.strip_discarded = TRUE;
   link_info.emit_hash = TRUE;
