diff -Nur binutils-2.20.51.0.10-old/bfd/config.bfd binutils-2.20.51.0.10/bfd/config.bfd
--- binutils-2.20.51.0.10-old/bfd/config.bfd	2010-07-20 16:18:43.559557724 +0300
+++ binutils-2.20.51.0.10/bfd/config.bfd	2010-07-20 16:21:52.908682710 +0300
@@ -57,7 +57,7 @@
  a29k-* | \
  hppa*-*-rtems* | \
  *-go32-rtems* | \
- i[3-7]86*-*-rtemscoff* | \
+ i[34567]86*-*-rtemscoff* | \
  mips*el-*-rtems* | \
  powerpcle-*-rtems* | \
  sparc*-*-rtemsaout* | \
@@ -83,7 +83,7 @@
 dlx*)		 targ_archs=bfd_dlx_arch ;;
 fido*)		 targ_archs=bfd_m68k_arch ;;
 hppa*)		 targ_archs=bfd_hppa_arch ;;
-i[3-7]86)	 targ_archs=bfd_i386_arch ;;
+i[34567]86)	 targ_archs=bfd_i386_arch ;;
 i370)		 targ_archs=bfd_i370_arch ;;
 lm32)	         targ_archs=bfd_lm32_arch ;;
 m6811*|m68hc11*) targ_archs="bfd_m68hc11_arch bfd_m68hc12_arch" ;;
@@ -485,17 +485,17 @@
     targ_defvec=bfd_elf32_i370_vec
     targ_selvecs="bfd_elf32_i370_vec"
     ;;
-  i[3-7]86-*-sco3.2v5*coff)
+  i[34567]86-*-sco3.2v5*coff)
     targ_defvec=i386coff_vec
     targ_selvecs=bfd_elf32_i386_vec
     ;;
-  i[3-7]86-*-sysv4* | i[3-7]86-*-unixware* | \
-  i[3-7]86-*-elf | i[3-7]86-*-sco3.2v5* | \
-  i[3-7]86-*-dgux* | i[3-7]86-*-sysv5*)
+  i[34567]86-*-sysv4* | i[34567]86-*-unixware* | \
+  i[34567]86-*-elf | i[34567]86-*-sco3.2v5* | \
+  i[34567]86-*-dgux* | i[34567]86-*-sysv5*)
     targ_defvec=bfd_elf32_i386_vec
     targ_selvecs=i386coff_vec
     ;;
-  i[3-7]86-*-solaris2*)
+  i[34567]86-*-solaris2*)
     targ_defvec=bfd_elf32_i386_sol2_vec
     targ_selvecs="i386coff_vec"
     targ64_selvecs="bfd_elf64_x86_64_sol2_vec bfd_elf64_l1om_vec"
@@ -508,21 +508,21 @@
     want64=true
     ;;
 #endif
-  i[3-7]86-*-kaos*)
+  i[34567]86-*-kaos*)
     targ_defvec=bfd_elf32_i386_vec
     targ_selvecs=bfd_elf32_i386_vec
     ;;
-  i[3-7]86-*-nto*)
+  i[34567]86-*-nto*)
     targ_defvec=bfd_elf32_i386_vec
     targ_selvecs=i386coff_vec
     ;;
-  i[3-7]86-*-aros*)
+  i[34567]86-*-aros*)
     targ_defvec=bfd_elf32_i386_vec
     ;;
-  i[3-7]86-*-chorus*)
+  i[34567]86-*-chorus*)
     targ_defvec=bfd_elf32_i386_vec
     ;;
-  i[3-7]86-*-dicos*)
+  i[34567]86-*-dicos*)
     targ_defvec=bfd_elf32_i386_vec
     targ64_selvecs="bfd_elf64_x86_64_vec bfd_elf64_l1om_vec"
     ;;
@@ -530,72 +530,72 @@
     targ_defvec=go32coff_vec
     targ_selvecs="go32stubbedcoff_vec i386aout_vec"
     ;;
-  i[3-7]86-*-sysv* | i[3-7]86-*-isc* | i[3-7]86-*-sco* | i[3-7]86-*-coff | \
-  i[3-7]86-*-aix*)
+  i[34567]86-*-sysv* | i[34567]86-*-isc* | i[34567]86-*-sco* | i[34567]86-*-coff | \
+  i[34567]86-*-aix*)
     targ_defvec=i386coff_vec
     ;;
-  i[3-7]86-*-rtems*)
+  i[34567]86-*-rtems*)
     targ_defvec=bfd_elf32_i386_vec
     targ_selvecs="i386coff_vec i386aout_vec"
     ;;
-  i[3-7]86-*-darwin* | i[3-7]86-*-macos10* | i[3-7]86-*-rhapsody*)
+  i[34567]86-*-darwin* | i[34567]86-*-macos10* | i[34567]86-*-rhapsody*)
     targ_defvec=mach_o_i386_vec
     targ_selvecs="mach_o_le_vec mach_o_be_vec mach_o_fat_vec pef_vec pef_xlib_vec sym_vec"
     targ_archs="$targ_archs bfd_powerpc_arch bfd_rs6000_arch"
     ;;
- i[3-7]86-sequent-bsd*)
+ i[34567]86-sequent-bsd*)
     targ_defvec=i386dynix_vec
     targ_underscore=yes
     ;;
-  i[3-7]86-*-bsd*)
+  i[34567]86-*-bsd*)
     targ_defvec=i386bsd_vec
     targ_underscore=yes
     ;;
-  i[3-7]86-*-freebsdaout* | i[3-7]86-*-freebsd[12].* | \
-  i[3-7]86-*-freebsd[12])
+  i[34567]86-*-freebsdaout* | i[34567]86-*-freebsd[12].* | \
+  i[34567]86-*-freebsd[12])
     targ_defvec=i386freebsd_vec
     targ_selvecs=i386bsd_vec
     targ_underscore=yes
     ;;
-  i[3-7]86-*-freebsd* | i[3-7]86-*-kfreebsd*-gnu | i[3-7]86-*-dragonfly*)
+  i[34567]86-*-freebsd* | i[34567]86-*-kfreebsd*-gnu | i[34567]86-*-dragonfly*)
     targ_defvec=bfd_elf32_i386_freebsd_vec
     targ_selvecs="bfd_elf32_i386_vec i386pei_vec i386coff_vec"
     targ64_selvecs="bfd_elf64_x86_64_freebsd_vec bfd_elf64_x86_64_vec x86_64pei_vec bfd_elf64_l1om_vec bfd_elf64_l1om_freebsd_vec"
     # FreeBSD <= 4.0 supports only the old nonstandard way of ABI labelling.
     case "${targ}" in
-      i[3-7]86-*-freebsd3* | i[3-7]86-*-freebsd4 | i[3-7]86-*-freebsd4.0*)
+      i[34567]86-*-freebsd3* | i[34567]86-*-freebsd4 | i[34567]86-*-freebsd4.0*)
 	targ_cflags=-DOLD_FREEBSD_ABI_LABEL ;;
     esac
     ;;
-  i[3-7]86-*-netbsdelf* | i[3-7]86-*-netbsd*-gnu* | i[3-7]86-*-knetbsd*-gnu)
+  i[34567]86-*-netbsdelf* | i[34567]86-*-netbsd*-gnu* | i[34567]86-*-knetbsd*-gnu)
     targ_defvec=bfd_elf32_i386_vec
     targ_selvecs=i386netbsd_vec
     targ64_selvecs="bfd_elf64_x86_64_vec bfd_elf64_l1om_vec"
     ;;
-  i[3-7]86-*-netbsdpe*)
+  i[34567]86-*-netbsdpe*)
     targ_defvec=i386pe_vec
     targ_selvecs="i386pe_vec i386pei_vec bfd_elf32_i386_vec"
     ;;
-  i[3-7]86-*-netbsdaout* | i[3-7]86-*-netbsd* | \
-  i[3-7]86-*-openbsd[0-2].* | i[3-7]86-*-openbsd3.[0-3])
+  i[34567]86-*-netbsdaout* | i[34567]86-*-netbsd* | \
+  i[34567]86-*-openbsd[0-2].* | i[34567]86-*-openbsd3.[0-3])
     targ_defvec=i386netbsd_vec
     targ_selvecs="bfd_elf32_i386_vec i386bsd_vec"
     targ_underscore=yes
     ;;
-  i[3-7]86-*-openbsd*)
+  i[34567]86-*-openbsd*)
     targ_defvec=bfd_elf32_i386_vec
     targ_selvecs=i386netbsd_vec
     ;;
-  i[3-7]86-*-netware*)
+  i[34567]86-*-netware*)
     targ_defvec=bfd_elf32_i386_vec
     targ_selvecs="nlm32_i386_vec i386coff_vec i386aout_vec"
     ;;
-  i[3-7]86-*-linux*aout*)
+  i[34567]86-*-linux*aout*)
     targ_defvec=i386linux_vec
     targ_selvecs=bfd_elf32_i386_vec
     targ_underscore=yes
     ;;
-  i[3-7]86-*-linux-*)
+  i[34567]86-*-linux-*)
     targ_defvec=bfd_elf32_i386_vec
     targ_selvecs="i386linux_vec i386pei_vec"
     targ64_selvecs="bfd_elf64_x86_64_vec bfd_elf64_l1om_vec"
@@ -639,63 +639,63 @@
     targ_underscore=no
     ;;
 #endif
-  i[3-7]86-*-lynxos*)
+  i[34567]86-*-lynxos*)
     targ_defvec=bfd_elf32_i386_vec
     targ_selvecs="i386lynx_coff_vec i386lynx_aout_vec"
     ;;
-  i[3-7]86-*-gnu*)
+  i[34567]86-*-gnu*)
     targ_defvec=bfd_elf32_i386_vec
     ;;
-  i[3-7]86-*-mach* | i[3-7]86-*-osf1mk*)
+  i[34567]86-*-mach* | i[34567]86-*-osf1mk*)
     targ_defvec=i386mach3_vec
     targ_cflags=-DSTAT_FOR_EXEC
     targ_underscore=yes
     ;;
-  i[3-7]86-*-os9k)
+  i[34567]86-*-os9k)
     targ_defvec=i386os9k_vec
     ;;
-  i[3-7]86-*-msdos*)
+  i[34567]86-*-msdos*)
     targ_defvec=i386aout_vec
     targ_selvecs=i386msdos_vec
     ;;
-  i[3-7]86-*-moss*)
+  i[34567]86-*-moss*)
     targ_defvec=bfd_elf32_i386_vec
     targ_selvecs="i386msdos_vec i386aout_vec"
     ;;
-  i[3-7]86-*-beospe*)
+  i[34567]86-*-beospe*)
     targ_defvec=i386pe_vec
     targ_selvecs="i386pe_vec i386pei_vec"
     ;;
-  i[3-7]86-*-beoself* | i[3-7]86-*-beos*)
+  i[34567]86-*-beoself* | i[34567]86-*-beos*)
     targ_defvec=bfd_elf32_i386_vec
     targ_selvecs="i386pe_vec i386pei_vec"
     ;;
-  i[3-7]86-*-interix*)
+  i[34567]86-*-interix*)
     targ_defvec=i386pei_vec
     targ_selvecs="i386pe_vec"
     # FIXME: This should eventually be checked at runtime.
     targ_cflags=-DSTRICT_PE_FORMAT
     ;;
-  i[3-7]86-*-rdos*)
+  i[34567]86-*-rdos*)
     targ_defvec=bfd_elf32_i386_vec
     targ_selvecs=i386coff_vec
     ;;
-  i[3-7]86-*-mingw32* | i[3-7]86-*-cygwin* | i[3-7]86-*-winnt | i[3-7]86-*-pe)
+  i[34567]86-*-mingw32* | i[34567]86-*-cygwin* | i[34567]86-*-winnt | i[34567]86-*-pe)
     targ_defvec=i386pe_vec
     targ_selvecs="i386pe_vec i386pei_vec bfd_elf32_i386_vec"
     targ_underscore=yes
     ;;
-  i[3-7]86-none-*)
+  i[34567]86-none-*)
     targ_defvec=i386coff_vec
     ;;
-  i[3-7]86-*-aout* | i[3-7]86*-*-vsta*)
+  i[34567]86-*-aout* | i[34567]86*-*-vsta*)
     targ_defvec=i386aout_vec
     ;;
-  i[3-7]86-*-vxworks*)
+  i[34567]86-*-vxworks*)
     targ_defvec=bfd_elf32_i386_vxworks_vec
     targ_underscore=yes
     ;;
-  i[3-7]86-*-chaos)
+  i[34567]86-*-chaos)
     targ_defvec=bfd_elf32_i386_vec
     targ_selfvecs=i386chaos_vec
     ;;
diff -Nur binutils-2.20.51.0.10-old/bfd/configure binutils-2.20.51.0.10/bfd/configure
--- binutils-2.20.51.0.10-old/bfd/configure	2010-07-20 16:18:43.555557717 +0300
+++ binutils-2.20.51.0.10/bfd/configure	2010-07-20 16:21:52.918682437 +0300
@@ -11781,7 +11781,7 @@
 
 
 case "${host}" in
-  sparc-*-solaris*|i[3-7]86-*-solaris*)
+  sparc-*-solaris*|i[34567]86-*-solaris*)
     # On native 32bit sparc and ia32 solaris, large-file and procfs support
     # are mutually exclusive; and without procfs support, the bfd/ elf module
     # cannot provide certain routines such as elfcore_write_prpsinfo
@@ -13868,50 +13868,50 @@
 	TRAD_HEADER='"hosts/i370linux.h"'
 	;;
 
-  i[3-7]86-sequent-bsd*)
+  i[34567]86-sequent-bsd*)
 	COREFILE=trad-core.lo
 	TRAD_HEADER='"hosts/symmetry.h"'
 	;;
-  i[3-7]86-sequent-sysv4*) ;;
-  i[3-7]86-sequent-sysv*)
+  i[34567]86-sequent-sysv4*) ;;
+  i[34567]86-sequent-sysv*)
 	COREFILE=trad-core.lo
 	TRAD_HEADER='"hosts/symmetry.h"'
 	;;
-  i[3-7]86-*-bsdi)
+  i[34567]86-*-bsdi)
 	COREFILE=
 	;;
-  i[3-7]86-*-bsd* | i[3-7]86-*-freebsd[123] | i[3-7]86-*-freebsd[123]\.* | i[3-7]86-*-freebsd4\.[01234] | i[3-7]86-*-freebsd4\.[01234]\.* | i[3-7]86-*-freebsd*aout*)
+  i[34567]86-*-bsd* | i[34567]86-*-freebsd[123] | i[34567]86-*-freebsd[123]\.* | i[34567]86-*-freebsd4\.[01234] | i[34567]86-*-freebsd4\.[01234]\.* | i[34567]86-*-freebsd*aout*)
 	COREFILE=trad-core.lo
 	TRAD_HEADER='"hosts/i386bsd.h"'
 	;;
-  i[3-7]86-*-freebsd* | i[3-7]86-*-kfreebsd*-gnu | i[3-7]86-*-dragonfly*)
+  i[34567]86-*-freebsd* | i[34567]86-*-kfreebsd*-gnu | i[34567]86-*-dragonfly*)
 	COREFILE=''
 	TRAD_HEADER='"hosts/i386bsd.h"'
 	;;
-  i[3-7]86-*-netbsd* | i[3-7]86-*-knetbsd*-gnu | i[3-7]86-*-openbsd*)
+  i[34567]86-*-netbsd* | i[34567]86-*-knetbsd*-gnu | i[34567]86-*-openbsd*)
 	COREFILE=netbsd-core.lo
 	;;
-  i[3-7]86-esix-sysv3*)
+  i[34567]86-esix-sysv3*)
 	COREFILE=trad-core.lo
 	TRAD_HEADER='"hosts/esix.h"'
 	;;
-  i[3-7]86-*-sco3.2v5*)
+  i[34567]86-*-sco3.2v5*)
 	COREFILE=sco5-core.lo
 	;;
-  i[3-7]86-*-sco* | i[3-7]86-*-isc*)
+  i[34567]86-*-sco* | i[34567]86-*-isc*)
 	COREFILE=trad-core.lo
 	TRAD_HEADER='"hosts/i386sco.h"'
 	;;
-  i[3-7]86-*-mach3*)
+  i[34567]86-*-mach3*)
 	COREFILE=trad-core.lo
 	TRAD_HEADER='"hosts/i386mach3.h"'
 	;;
-  i[3-7]86-*-linux-*)
+  i[34567]86-*-linux-*)
 	COREFILE=trad-core.lo
 	TRAD_HEADER='"hosts/i386linux.h"'
 	;;
-  i[3-7]86-*-isc*)	COREFILE=trad-core.lo ;;
-  i[3-7]86-*-aix*)	COREFILE=aix386-core.lo ;;
+  i[34567]86-*-isc*)	COREFILE=trad-core.lo ;;
+  i[34567]86-*-aix*)	COREFILE=aix386-core.lo ;;
   i860-*-mach3* | i860-*-osf1*)
 	COREFILE=trad-core.lo
 	TRAD_HEADER='"hosts/i860mach3.h"'
diff -Nur binutils-2.20.51.0.10-old/bfd/configure.host binutils-2.20.51.0.10/bfd/configure.host
--- binutils-2.20.51.0.10-old/bfd/configure.host	2010-07-20 16:18:43.564558106 +0300
+++ binutils-2.20.51.0.10/bfd/configure.host	2010-07-20 16:21:52.918682437 +0300
@@ -37,11 +37,11 @@
 
 # Workaround for limitations on win9x where file contents are
 # not zero'd out if you seek past the end and then write.
-i[3-7]86-*-mingw32*)	HDEFINES=-D__USE_MINGW_FSEEK;;
+i[34567]86-*-mingw32*)	HDEFINES=-D__USE_MINGW_FSEEK;;
 
-i[3-7]86-sequent-bsd*)	HDEFINES=-Dshared=genshared ;;
-i[3-7]86-sequent-sysv4*) ;;
-i[3-7]86-sequent-sysv*)	HDEFINES=-Dshared=genshared ;;
+i[34567]86-sequent-bsd*)	HDEFINES=-Dshared=genshared ;;
+i[34567]86-sequent-sysv4*) ;;
+i[34567]86-sequent-sysv*)	HDEFINES=-Dshared=genshared ;;
 
 mips*-*-netbsd*)	;;
 mips*-*-openbsd*)	;;
diff -Nur binutils-2.20.51.0.10-old/bfd/configure.in binutils-2.20.51.0.10/bfd/configure.in
--- binutils-2.20.51.0.10-old/bfd/configure.in	2010-07-20 16:18:43.556558159 +0300
+++ binutils-2.20.51.0.10/bfd/configure.in	2010-07-20 16:21:52.919682747 +0300
@@ -257,72 +257,72 @@
 	;;
 
 changequote(,)dnl
-  i[3-7]86-sequent-bsd*)
+  i[34567]86-sequent-bsd*)
 changequote([,])dnl
 	COREFILE=trad-core.lo
 	TRAD_HEADER='"hosts/symmetry.h"'
 	;;
 changequote(,)dnl
-  i[3-7]86-sequent-sysv4*) ;;
-  i[3-7]86-sequent-sysv*)
+  i[34567]86-sequent-sysv4*) ;;
+  i[34567]86-sequent-sysv*)
 changequote([,])dnl
 	COREFILE=trad-core.lo
 	TRAD_HEADER='"hosts/symmetry.h"'
 	;;
 changequote(,)dnl
-  i[3-7]86-*-bsdi)
+  i[34567]86-*-bsdi)
 changequote([,])dnl
 	COREFILE=
 	;;
 changequote(,)dnl
-  i[3-7]86-*-bsd* | i[3-7]86-*-freebsd[123] | i[3-7]86-*-freebsd[123]\.* | i[3-7]86-*-freebsd4\.[01234] | i[3-7]86-*-freebsd4\.[01234]\.* | i[3-7]86-*-freebsd*aout*)
+  i[34567]86-*-bsd* | i[34567]86-*-freebsd[123] | i[34567]86-*-freebsd[123]\.* | i[34567]86-*-freebsd4\.[01234] | i[34567]86-*-freebsd4\.[01234]\.* | i[34567]86-*-freebsd*aout*)
 changequote([,])dnl
 	COREFILE=trad-core.lo
 	TRAD_HEADER='"hosts/i386bsd.h"'
 	;;
 changequote(,)dnl
-  i[3-7]86-*-freebsd* | i[3-7]86-*-kfreebsd*-gnu | i[3-7]86-*-dragonfly*)
+  i[34567]86-*-freebsd* | i[34567]86-*-kfreebsd*-gnu | i[34567]86-*-dragonfly*)
 changequote([,])dnl
 	COREFILE=''
 	TRAD_HEADER='"hosts/i386bsd.h"'
 	;;
 changequote(,)dnl
-  i[3-7]86-*-netbsd* | i[3-7]86-*-knetbsd*-gnu | i[3-7]86-*-openbsd*)
+  i[34567]86-*-netbsd* | i[34567]86-*-knetbsd*-gnu | i[34567]86-*-openbsd*)
 changequote([,])dnl
 	COREFILE=netbsd-core.lo
 	;;
 changequote(,)dnl
-  i[3-7]86-esix-sysv3*)
+  i[34567]86-esix-sysv3*)
 changequote([,])dnl
 	COREFILE=trad-core.lo
 	TRAD_HEADER='"hosts/esix.h"'
 	;;
 changequote(,)dnl
-  i[3-7]86-*-sco3.2v5*)
+  i[34567]86-*-sco3.2v5*)
 changequote([,])dnl
 	COREFILE=sco5-core.lo
 	;;
 changequote(,)dnl
-  i[3-7]86-*-sco* | i[3-7]86-*-isc*)
+  i[34567]86-*-sco* | i[34567]86-*-isc*)
 changequote([,])dnl
 	COREFILE=trad-core.lo
 	TRAD_HEADER='"hosts/i386sco.h"'
 	;;
 changequote(,)dnl
-  i[3-7]86-*-mach3*)
+  i[34567]86-*-mach3*)
 changequote([,])dnl
 	COREFILE=trad-core.lo
 	TRAD_HEADER='"hosts/i386mach3.h"'
 	;;
 changequote(,)dnl
-  i[3-7]86-*-linux-*)
+  i[34567]86-*-linux-*)
 changequote([,])dnl
 	COREFILE=trad-core.lo
 	TRAD_HEADER='"hosts/i386linux.h"'
 	;;
 changequote(,)dnl
-  i[3-7]86-*-isc*)	COREFILE=trad-core.lo ;;
-  i[3-7]86-*-aix*)	COREFILE=aix386-core.lo ;;
+  i[34567]86-*-isc*)	COREFILE=trad-core.lo ;;
+  i[34567]86-*-aix*)	COREFILE=aix386-core.lo ;;
 changequote([,])dnl
   i860-*-mach3* | i860-*-osf1*)
 	COREFILE=trad-core.lo
diff -Nur binutils-2.20.51.0.10-old/binutils/configure binutils-2.20.51.0.10/binutils/configure
--- binutils-2.20.51.0.10-old/binutils/configure	2010-07-20 16:18:43.588558007 +0300
+++ binutils-2.20.51.0.10/binutils/configure	2010-07-20 16:21:52.927682469 +0300
@@ -4927,7 +4927,7 @@
 
 
 case "${host}" in
-  sparc-*-solaris*|i[3-7]86-*-solaris*)
+  sparc-*-solaris*|i[34567]86-*-solaris*)
     # On native 32bit sparc and ia32 solaris, large-file and procfs support
     # are mutually exclusive; and without procfs support, the bfd/ elf module
     # cannot provide certain routines such as elfcore_write_prpsinfo
@@ -13806,7 +13806,7 @@
 	BUILD_DLLWRAP='$(DLLWRAP_PROG)$(EXEEXT)'
     else
 	case $targ in
-	i[3-7]86*-*-netware*)
+	i[34567]86*-*-netware*)
 	  BUILD_NLMCONV='$(NLMCONV_PROG)$(EXEEXT)'
 	  NLMCONV_DEFS="$NLMCONV_DEFS -DNLMCONV_I386"
 	  ;;
@@ -13873,7 +13873,7 @@
 	  BUILD_WINDMC='$(WINDMC_PROG)$(EXEEXT)'
 	  BUILD_DLLWRAP='$(DLLWRAP_PROG)$(EXEEXT)'
 	  ;;
-	i[3-7]86-*-pe* | i[3-7]86-*-cygwin* | i[3-7]86-*-mingw32** | i[3-7]86-*-netbsdpe*)
+	i[34567]86-*-pe* | i[34567]86-*-cygwin* | i[34567]86-*-mingw32** | i[34567]86-*-netbsdpe*)
   	  BUILD_DLLTOOL='$(DLLTOOL_PROG)$(EXEEXT)'
 	  if test -z "$DLLTOOL_DEFAULT"; then
 	    DLLTOOL_DEFAULT="-DDLLTOOL_DEFAULT_I386"
@@ -13883,7 +13883,7 @@
 	  BUILD_WINDMC='$(WINDMC_PROG)$(EXEEXT)'
 	  BUILD_DLLWRAP='$(DLLWRAP_PROG)$(EXEEXT)'
 	  ;;
-	i[3-7]86-*-interix)
+	i[34567]86-*-interix)
 	  BUILD_DLLTOOL='$(DLLTOOL_PROG)'
 	  if test -z "$DLLTOOL_DEFAULT"; then
 	    DLLTOOL_DEFAULT="-DDLLTOOL_DEFAULT_I386"
diff -Nur binutils-2.20.51.0.10-old/binutils/configure.in binutils-2.20.51.0.10/binutils/configure.in
--- binutils-2.20.51.0.10-old/binutils/configure.in	2010-07-20 16:18:43.588558007 +0300
+++ binutils-2.20.51.0.10/binutils/configure.in	2010-07-20 16:21:52.928682549 +0300
@@ -199,7 +199,7 @@
     else
 	case $targ in
 changequote(,)dnl
-	i[3-7]86*-*-netware*) 
+	i[34567]86*-*-netware*) 
 changequote([,])dnl
 	  BUILD_NLMCONV='$(NLMCONV_PROG)$(EXEEXT)'
 	  NLMCONV_DEFS="$NLMCONV_DEFS -DNLMCONV_I386"
@@ -268,7 +268,7 @@
 	  BUILD_DLLWRAP='$(DLLWRAP_PROG)$(EXEEXT)'
 	  ;;
 changequote(,)dnl
-	i[3-7]86-*-pe* | i[3-7]86-*-cygwin* | i[3-7]86-*-mingw32** | i[3-7]86-*-netbsdpe*)
+	i[34567]86-*-pe* | i[34567]86-*-cygwin* | i[34567]86-*-mingw32** | i[34567]86-*-netbsdpe*)
 changequote([,])dnl
   	  BUILD_DLLTOOL='$(DLLTOOL_PROG)$(EXEEXT)'
 	  if test -z "$DLLTOOL_DEFAULT"; then
@@ -280,7 +280,7 @@
 	  BUILD_DLLWRAP='$(DLLWRAP_PROG)$(EXEEXT)'
 	  ;;
 changequote(,)dnl
-	i[3-7]86-*-interix)
+	i[34567]86-*-interix)
 changequote([,])dnl
 	  BUILD_DLLTOOL='$(DLLTOOL_PROG)'
 	  if test -z "$DLLTOOL_DEFAULT"; then
diff -Nur binutils-2.20.51.0.10-old/etc/configure.texi binutils-2.20.51.0.10/etc/configure.texi
--- binutils-2.20.51.0.10-old/etc/configure.texi	2010-07-20 16:18:43.544557944 +0300
+++ binutils-2.20.51.0.10/etc/configure.texi	2010-07-20 16:21:52.931682568 +0300
@@ -1453,7 +1453,7 @@
 
 @smallexample
 case "$@{host@}" in
-i[3-7]86-*-linux-gnu*) do something ;;
+i[34567]86-*-linux-gnu*) do something ;;
 sparc*-sun-solaris2.[56789]*) do something ;;
 sparc*-sun-solaris*) do something ;;
 mips*-*-elf*) do something ;;
@@ -1467,7 +1467,7 @@
 In most cases you must be careful to match a range of processor types.
 For most processor families, a trailing @samp{*} suffices, as in
 @samp{mips*} above.  For the i386 family, something along the lines of
-@samp{i[3-7]86} suffices at present.  For the m68k family, you will
+@samp{i[34567]86} suffices at present.  For the m68k family, you will
 need something like @samp{m68*}.  Of course, if you do not need to match
 on the processor, it is simpler to just replace the entire field by a
 @samp{*}, as in @samp{*-*-irix*}.
diff -Nur binutils-2.20.51.0.10-old/gas/configure binutils-2.20.51.0.10/gas/configure
--- binutils-2.20.51.0.10-old/gas/configure	2010-07-20 16:18:43.406682826 +0300
+++ binutils-2.20.51.0.10/gas/configure	2010-07-20 16:21:52.940682563 +0300
@@ -4915,7 +4915,7 @@
 
 
 case "${host}" in
-  sparc-*-solaris*|i[3-7]86-*-solaris*)
+  sparc-*-solaris*|i[34567]86-*-solaris*)
     # On native 32bit sparc and ia32 solaris, large-file and procfs support
     # are mutually exclusive; and without procfs support, the bfd/ elf module
     # cannot provide certain routines such as elfcore_write_prpsinfo
diff -Nur binutils-2.20.51.0.10-old/gas/configure.tgt binutils-2.20.51.0.10/gas/configure.tgt
--- binutils-2.20.51.0.10-old/gas/configure.tgt	2010-07-20 16:18:43.405683704 +0300
+++ binutils-2.20.51.0.10/gas/configure.tgt	2010-07-20 16:21:52.941682731 +0300
@@ -40,7 +40,7 @@
   crx*)			cpu_type=crx endian=little ;;
   fido)			cpu_type=m68k ;;
   hppa*)		cpu_type=hppa ;;
-  i[3-7]86)		cpu_type=i386 arch=i386;;
+  i[34567]86)		cpu_type=i386 arch=i386;;
   ia64)			cpu_type=ia64 ;;
   ip2k)			cpu_type=ip2k endian=big ;; 
   iq2000)		cpu_type=iq2000 endian=big ;; 
diff -Nur binutils-2.20.51.0.10-old/gprof/configure binutils-2.20.51.0.10/gprof/configure
--- binutils-2.20.51.0.10-old/gprof/configure	2010-07-20 16:18:43.573558107 +0300
+++ binutils-2.20.51.0.10/gprof/configure	2010-07-20 16:21:52.953682533 +0300
@@ -4844,7 +4844,7 @@
 
 
 case "${host}" in
-  sparc-*-solaris*|i[3-7]86-*-solaris*)
+  sparc-*-solaris*|i[34567]86-*-solaris*)
     # On native 32bit sparc and ia32 solaris, large-file and procfs support
     # are mutually exclusive; and without procfs support, the bfd/ elf module
     # cannot provide certain routines such as elfcore_write_prpsinfo
diff -Nur binutils-2.20.51.0.10-old/ld/configure binutils-2.20.51.0.10/ld/configure
--- binutils-2.20.51.0.10-old/ld/configure	2010-07-20 16:18:43.598557785 +0300
+++ binutils-2.20.51.0.10/ld/configure	2010-07-20 16:21:52.966682524 +0300
@@ -5339,7 +5339,7 @@
 
 
 case "${host}" in
-  sparc-*-solaris*|i[3-7]86-*-solaris*)
+  sparc-*-solaris*|i[34567]86-*-solaris*)
     # On native 32bit sparc and ia32 solaris, large-file and procfs support
     # are mutually exclusive; and without procfs support, the bfd/ elf module
     # cannot provide certain routines such as elfcore_write_prpsinfo
diff -Nur binutils-2.20.51.0.10-old/ld/configure.host binutils-2.20.51.0.10/ld/configure.host
--- binutils-2.20.51.0.10-old/ld/configure.host	2010-07-20 16:18:43.672557809 +0300
+++ binutils-2.20.51.0.10/ld/configure.host	2010-07-20 16:21:52.966682524 +0300
@@ -95,31 +95,31 @@
   HOSTING_LIBS='--start-group `if [ -f ../gcc/libgcc.a ]; then echo ../gcc/libgcc.a; else if test "$GCC" = yes; then ${CC} --print-libgcc-file-name; else gcc --print-libgcc-file-name; fi fi` -lc --end-group /usr/lib/pa20_64/milli.a'
   ;;
 
-i[3-7]86-*-bsd* | i[3-7]86-*-freebsd[12] | i[3-7]86-*-freebsd[12]\.* | i[3-7]86-*-freebsd*aout*)
+i[34567]86-*-bsd* | i[34567]86-*-freebsd[12] | i[34567]86-*-freebsd[12]\.* | i[34567]86-*-freebsd*aout*)
   HOSTING_CRT0=/usr/lib/crt0.o
   ;;
 
-i[3-7]86-*-sysv4*)
+i[34567]86-*-sysv4*)
   HOSTING_CRT0='/usr/ccs/lib/crt1.o /usr/ccs/lib/crti.o /usr/ccs/lib/values-Xa.o `if [ -f ../gcc/crtbegin.o ]; then echo ../gcc/crtbegin.o; else ${CC} -print-file-name=crtbegin.o; fi`'
   HOSTING_LIBS="$HOSTING_LIBS"' `if [ -f ../gcc/crtend.o ]; then echo ../gcc/crtend.o; else ${CC} -print-file-name=crtend.o; fi` /usr/ccs/lib/crtn.o'
   ;;
 
-i[3-7]86-sequent-ptx* | i[3-7]86-sequent-sysv*)
+i[34567]86-sequent-ptx* | i[34567]86-sequent-sysv*)
   HOSTING_CRT0='/lib/crt0.o `if [ -f ../gcc/crtbegin.o ]; then echo ../gcc/crtbegin.o; else ${CC} -print-file-name=crtbegin.o; fi`'
   HOSTING_LIBS="$HOSTING_LIBS"' `if [ -f ../gcc/crtend.o ]; then echo ../gcc/crtend.o; else ${CC} -print-file-name=crtend.o; fi`'
   ;;
 
-i[3-7]86-*-sysv*)
+i[34567]86-*-sysv*)
   HOSTING_CRT0='/lib/crt1.o `if [ -f ../gcc/crtbegin.o ]; then echo ../gcc/crtbegin.o; fi`'
   HOSTING_LIBS="$HOSTING_LIBS"' `if [ -f ../gcc/crtend.o ]; then echo ../gcc/crtend.o; fi` /lib/crtn.o'
   ;;
 
-i[3-7]86-*-solaris*)
+i[34567]86-*-solaris*)
   HOSTING_CRT0='`if [ -f ../gcc/crt1.o ]; then echo ../gcc/crt1.o; else ${CC} -print-file-name=crt1.o; fi` `if [ -f ../gcc/crti.o ]; then echo ../gcc/crti.o; else ${CC} -print-file-name=crti.o; fi` /usr/ccs/lib/values-Xa.o `if [ -f ../gcc/crtbegin.o ]; then echo ../gcc/crtbegin.o; else ${CC} -print-file-name=crtbegin.o; fi`'
   HOSTING_LIBS="$HOSTING_LIBS"' `if [ -f ../gcc/crtend.o ]; then echo ../gcc/crtend.o; else ${CC} -print-file-name=crtend.o; fi` `if [ -f ../gcc/crtn.o ]; then echo ../gcc/crtn.o; else ${CC} -print-file-name=crtn.o; fi`'
   ;;
 
-i[3-7]86-*-sco* | i[3-7]86-*-isc*)
+i[34567]86-*-sco* | i[34567]86-*-isc*)
   # In some configurations gcc does not use crtbegin.o and crtend.o.
   # In that case gcc -print-file-name=crtbegin.o will simply print
   # crtbegin.o.  We create dummy crtbegin.o and crtend.o files to
@@ -134,16 +134,16 @@
   HOSTING_LIBS="$HOSTING_LIBS"' `if [ -f ../gcc/crtend.o ]; then echo ../gcc/crtend.o; else ${CC} -print-file-name=crtend.o; fi` /lib/crtn.o'
   ;;
 
-i[3-7]86-pc-interix*)
+i[34567]86-pc-interix*)
   HOSTING_CRT0='$$INTERIX_ROOT/usr/lib/crt0.o'
   HOSTING_LIBS='-L $$X/local_bin -L $$INTERIX_ROOT/usr/lib '"$HOSTING_LIBS"' -lcpsx -lc -lcpsx $$INTERIX_ROOT/usr/lib/psxdll.a $$INTERIX_ROOT/usr/lib/psxdll2.a'
   ;;
 
-i[3-7]86-*-cygwin*)
+i[34567]86-*-cygwin*)
   HOSTING_LIBS="$HOSTING_LIBS"' -lcygwin -L/usr/lib/w32api -luser32 -lkernel32 -ladvapi32 -lshell32 `if [ -f ../gcc/libgcc.a ] ; then echo ../gcc/libgcc.a ; else ${CC} -print-libgcc-file-name; fi`'
   ;;
 
-i[3-7]86-*-mingw*)
+i[34567]86-*-mingw*)
   #We only support msvcrt.dll, crtid == 2.
   HOSTING_CRT0='/mingw/lib/crt2.o'
   HOSTING_LIBS='-L/mingw/lib -lmingw32 -lmoldname -lmingwex -lmsvcrt -luser32 -lkernel32 -ladvapi32 -lshell32 -lmingw32 -lmoldname -lmingwex -lmsvcrt `if [ -f ../gcc/libgcc.a ] ; then echo ../gcc/libgcc.a ; else ${CC} -print-libgcc-file-name; fi`'
diff -Nur binutils-2.20.51.0.10-old/ld/configure.tgt binutils-2.20.51.0.10/ld/configure.tgt
--- binutils-2.20.51.0.10-old/ld/configure.tgt	2010-07-20 16:18:43.595557783 +0300
+++ binutils-2.20.51.0.10/ld/configure.tgt	2010-07-20 16:21:52.967682743 +0300
@@ -177,27 +177,27 @@
 			;;
 i370-*-elf* | i370-*-linux-*) targ_emul=elf32i370
 			;;
-i[3-7]86-*-nto-qnx*)	targ_emul=i386nto ;;
-i[3-7]86-*-vsta)	targ_emul=vsta ;;
-i[3-7]86-*-go32)	targ_emul=i386go32 ;;
-i[3-7]86-*-msdosdjgpp*) targ_emul=i386go32 ;;
-i[3-7]86-*-aix*)	targ_emul=i386coff ;;
-i[3-7]86-*-sco*)	targ_emul=i386coff ;;
-i[3-7]86-*-isc*)	targ_emul=i386coff ;;
-i[3-7]86-*-lynxos*)	targ_emul=i386lynx ;;
-i[3-7]86-*-coff)	targ_emul=i386coff ;;
-i[3-7]86-*-rtems*)	targ_emul=elf_i386 ;;
-i[3-7]86-*-aros*)	targ_emul=elf_i386 ;;
-i[3-7]86-*-rdos*)	targ_emul=elf_i386 ;;
-i[3-7]86-*-bsd)		targ_emul=i386bsd ;;
-i[3-7]86-*-bsd386)	targ_emul=i386bsd ;;
-i[3-7]86-*-bsdi*)	targ_emul=i386bsd ;;
-i[3-7]86-*-aout)	targ_emul=i386aout ;;
-i[3-7]86-*-linux*aout*)	targ_emul=i386linux
+i[34567]86-*-nto-qnx*)	targ_emul=i386nto ;;
+i[34567]86-*-vsta)	targ_emul=vsta ;;
+i[34567]86-*-go32)	targ_emul=i386go32 ;;
+i[34567]86-*-msdosdjgpp*) targ_emul=i386go32 ;;
+i[34567]86-*-aix*)	targ_emul=i386coff ;;
+i[34567]86-*-sco*)	targ_emul=i386coff ;;
+i[34567]86-*-isc*)	targ_emul=i386coff ;;
+i[34567]86-*-lynxos*)	targ_emul=i386lynx ;;
+i[34567]86-*-coff)	targ_emul=i386coff ;;
+i[34567]86-*-rtems*)	targ_emul=elf_i386 ;;
+i[34567]86-*-aros*)	targ_emul=elf_i386 ;;
+i[34567]86-*-rdos*)	targ_emul=elf_i386 ;;
+i[34567]86-*-bsd)		targ_emul=i386bsd ;;
+i[34567]86-*-bsd386)	targ_emul=i386bsd ;;
+i[34567]86-*-bsdi*)	targ_emul=i386bsd ;;
+i[34567]86-*-aout)	targ_emul=i386aout ;;
+i[34567]86-*-linux*aout*)	targ_emul=i386linux
 			targ_extra_emuls=elf_i386
 			tdir_elf_i386=`echo ${targ_alias} | sed -e 's/aout//'` ;;
-i[3-7]86-*-linux*oldld)	targ_emul=i386linux; targ_extra_emuls=elf_i386 ;;
-i[3-7]86-*-linux-*)	targ_emul=elf_i386
+i[34567]86-*-linux*oldld)	targ_emul=i386linux; targ_extra_emuls=elf_i386 ;;
+i[34567]86-*-linux-*)	targ_emul=elf_i386
 			targ_extra_emuls=i386linux
 			targ64_extra_emuls="elf_x86_64 elf_l1om"
 			targ64_extra_libpath=elf_x86_64
@@ -207,8 +207,8 @@
 			targ_extra_libpath=elf_i386
 			tdir_i386linux=`echo ${targ_alias}aout | sed -e 's/x86_64/i386/'`
 			tdir_elf_i386=`echo ${targ_alias} | sed -e 's/x86_64/i386/'` ;;
-i[3-7]86-*-sysv[45]*)	targ_emul=elf_i386 ;;
-i[3-7]86-*-solaris2*)	targ_emul=elf_i386_sol2
+i[34567]86-*-sysv[45]*)	targ_emul=elf_i386 ;;
+i[34567]86-*-solaris2*)	targ_emul=elf_i386_sol2
                         targ_extra_emuls="elf_i386_ldso elf_i386 elf_x86_64_sol2 elf_x86_64 elf_l1om"
 			targ_extra_libpath=$targ_extra_emuls
                         ;;
@@ -217,19 +217,19 @@
 			targ_extra_emuls="elf_x86_64 elf_i386_sol2 elf_i386_ldso elf_i386 elf_l1om"
 			targ_extra_libpath=elf_i386
 			tdir_elf_i386=`echo ${targ_alias} | sed -e 's/x86_64/i386/'` ;;
-i[3-7]86-*-unixware)	targ_emul=elf_i386 ;;
-i[3-7]86-*-solaris*)	targ_emul=elf_i386_ldso
+i[34567]86-*-unixware)	targ_emul=elf_i386 ;;
+i[34567]86-*-solaris*)	targ_emul=elf_i386_ldso
                         targ_extra_emuls="elf_i386"
 			targ_extra_libpath=$targ_extra_emuls
                         ;;
-i[3-7]86-*-netbsdelf* | \
-i[3-7]86-*-netbsd*-gnu* | \
-i[3-7]86-*-knetbsd*-gnu)
+i[34567]86-*-netbsdelf* | \
+i[34567]86-*-netbsd*-gnu* | \
+i[34567]86-*-knetbsd*-gnu)
 			targ_emul=elf_i386
 			targ_extra_emuls=i386nbsd ;;
-i[3-7]86-*-netbsdpe*)	targ_emul=i386pe
+i[34567]86-*-netbsdpe*)	targ_emul=i386pe
 			targ_extra_ofiles="deffilep.o pe-dll.o" ;;
-i[3-7]86-*-netbsd*)	targ_emul=i386nbsd
+i[34567]86-*-netbsd*)	targ_emul=i386nbsd
 			targ_extra_emuls=elf_i386 ;;
 x86_64-*-netbsd*)	targ_emul=elf_x86_64
 			targ_extra_emuls="elf_i386 i386nbsd elf_l1om"
@@ -240,15 +240,15 @@
 			*)		tdir_elf_i386=`echo ${tdir_elf_i386} | \
 				        sed -e 's/netbsd/netbsdelf/'`;;
 			esac ;;
-i[3-7]86-*-netware)	targ_emul=i386nw ;;
-i[3-7]86-*-elf*)	targ_emul=elf_i386 ;;
+i[34567]86-*-netware)	targ_emul=i386nw ;;
+i[34567]86-*-elf*)	targ_emul=elf_i386 ;;
 x86_64-*-elf*)		targ_emul=elf_x86_64
 			targ_extra_emuls="elf_i386 elf_l1om"
 			;;
-i[3-7]86-*-kaos*)	targ_emul=elf_i386 ;;
-i[3-7]86-*-freebsdaout* | i[3-7]86-*-freebsd[12].* | i[3-7]86-*-freebsd[12])
+i[34567]86-*-kaos*)	targ_emul=elf_i386 ;;
+i[34567]86-*-freebsdaout* | i[34567]86-*-freebsd[12].* | i[34567]86-*-freebsd[12])
 			targ_emul=i386bsd ;;
-i[3-7]86-*-freebsd* | i[3-7]86-*-kfreebsd*-gnu | i[3-7]86-*-dragonfly*)
+i[34567]86-*-freebsd* | i[34567]86-*-kfreebsd*-gnu | i[34567]86-*-dragonfly*)
 		        targ_emul=elf_i386_fbsd
 			targ_extra_emuls="elf_i386 i386bsd" ;;
 x86_64-*-freebsd* | x86_64-*-kfreebsd*-gnu)
@@ -259,29 +259,29 @@
 			    | sed -e 's/x86_64/i386/'`
 			tdir_elf_i386=`echo ${targ_alias} \
 			    | sed -e 's/x86_64/i386/'` ;;
-i[3-7]86-*-sysv*)	targ_emul=i386coff ;;
-i[3-7]86-*-ptx*)	targ_emul=i386coff ;;
-i[3-7]86-*-mach*)	targ_emul=i386mach ;;
-i[3-7]86-*-gnu*)	targ_emul=elf_i386 ;;
-i[3-7]86-*-msdos*)	targ_emul=i386msdos; targ_extra_emuls=i386aout ;;
-i[3-7]86-*-moss*)	targ_emul=i386moss; targ_extra_emuls=i386msdos ;;
-i[3-7]86-*-winnt*)	targ_emul=i386pe ;
+i[34567]86-*-sysv*)	targ_emul=i386coff ;;
+i[34567]86-*-ptx*)	targ_emul=i386coff ;;
+i[34567]86-*-mach*)	targ_emul=i386mach ;;
+i[34567]86-*-gnu*)	targ_emul=elf_i386 ;;
+i[34567]86-*-msdos*)	targ_emul=i386msdos; targ_extra_emuls=i386aout ;;
+i[34567]86-*-moss*)	targ_emul=i386moss; targ_extra_emuls=i386msdos ;;
+i[34567]86-*-winnt*)	targ_emul=i386pe ;
 			targ_extra_ofiles="deffilep.o pe-dll.o" ;;
-i[3-7]86-*-pe)		targ_emul=i386pe ;
+i[34567]86-*-pe)		targ_emul=i386pe ;
 			targ_extra_ofiles="deffilep.o pe-dll.o" ;;
-i[3-7]86-*-cygwin*)	targ_emul=i386pe ;
+i[34567]86-*-cygwin*)	targ_emul=i386pe ;
 			targ_extra_ofiles="deffilep.o pe-dll.o"
 			test "$targ" != "$host" && LIB_PATH='${tooldir}/lib/w32api' ;;
-i[3-7]86-*-mingw32*)	targ_emul=i386pe ;
+i[34567]86-*-mingw32*)	targ_emul=i386pe ;
 			targ_extra_ofiles="deffilep.o pe-dll.o" ;;
 x86_64-*-mingw*)	targ_emul=i386pep ;
 			targ_extra_ofiles="deffilep.o pep-dll.o" ;;
-i[3-7]86-*-interix*)	targ_emul=i386pe_posix;
+i[34567]86-*-interix*)	targ_emul=i386pe_posix;
  			targ_extra_ofiles="deffilep.o pe-dll.o" ;;
-i[3-7]86-*-beospe*)	targ_emul=i386beos ;;
-i[3-7]86-*-beos*)	targ_emul=elf_i386_be ;;
-i[3-7]86-*-vxworks*)	targ_emul=elf_i386_vxworks ;;
-i[3-7]86-*-chaos)	targ_emul=elf_i386_chaos
+i[34567]86-*-beospe*)	targ_emul=i386beos ;;
+i[34567]86-*-beos*)	targ_emul=elf_i386_be ;;
+i[34567]86-*-vxworks*)	targ_emul=elf_i386_vxworks ;;
+i[34567]86-*-chaos)	targ_emul=elf_i386_chaos
 			;;
 i860-*-coff)		targ_emul=coff_i860 ;;
 i860-stardent-sysv4* | i860-stardent-elf*)
@@ -672,15 +672,15 @@
   NATIVE_LIB_DIRS=/usr/lib/pa20_64
   ;;
 
-i[3-7]86-*-sysv4*)
+i[34567]86-*-sysv4*)
   NATIVE_LIB_DIRS='/usr/local/lib /usr/ccs/lib /lib /usr/lib'
   ;;
 
-i[3-7]86-*-solaris*)
+i[34567]86-*-solaris*)
   NATIVE_LIB_DIRS='/usr/local/lib /usr/ccs/lib /lib /usr/lib'
   ;;
 
-i[3-7]86-pc-interix*)
+i[34567]86-pc-interix*)
   NATIVE_LIB_DIRS='/usr/local/lib $$INTERIX_ROOT/usr/lib /lib /usr/lib'
   ;;
 
diff -Nur binutils-2.20.51.0.10-old/opcodes/Makefile.am binutils-2.20.51.0.10/opcodes/Makefile.am
--- binutils-2.20.51.0.10-old/opcodes/Makefile.am	2010-07-20 16:18:44.434682945 +0300
+++ binutils-2.20.51.0.10/opcodes/Makefile.am	2010-07-20 16:22:09.640620901 +0300
@@ -275,8 +275,8 @@
 # old version of libbfd, or to pick up libbfd for the wrong architecture
 # if host != build. So for building with shared libraries we use a
 # hardcoded path to libbfd.so instead of relying on the entries in libbfd.la.
-libopcodes_la_DEPENDENCIES = $(OFILES) @SHARED_DEPENDENCIES@
-libopcodes_la_LIBADD = $(OFILES) @SHARED_LIBADD@
+libopcodes_la_DEPENDENCIES = $(OFILES) @SHARED_DEPENDENCIES@ ../bfd/libbfd.la
+libopcodes_la_LIBADD = $(OFILES) @SHARED_LIBADD@ ../bfd/libbfd.la
 libopcodes_la_LDFLAGS += -release `cat ../bfd/libtool-soversion` @SHARED_LDFLAGS@
 # Allow dependency tracking to work on all the source files.
 EXTRA_libopcodes_la_SOURCES = $(LIBOPCODES_CFILES)
diff -Nur binutils-2.20.51.0.10-old/opcodes/Makefile.in binutils-2.20.51.0.10/opcodes/Makefile.in
--- binutils-2.20.51.0.10-old/opcodes/Makefile.in	2010-07-20 16:21:14.512683371 +0300
+++ binutils-2.20.51.0.10/opcodes/Makefile.in	2010-07-20 16:22:09.655745707 +0300
@@ -532,8 +532,8 @@
 # old version of libbfd, or to pick up libbfd for the wrong architecture
 # if host != build. So for building with shared libraries we use a
 # hardcoded path to libbfd.so instead of relying on the entries in libbfd.la.
-libopcodes_la_DEPENDENCIES = $(OFILES) @SHARED_DEPENDENCIES@
-libopcodes_la_LIBADD = $(OFILES) @SHARED_LIBADD@
+libopcodes_la_DEPENDENCIES = $(OFILES) @SHARED_DEPENDENCIES@ ../bfd/libbfd.la
+libopcodes_la_LIBADD = $(OFILES) @SHARED_LIBADD@ ../bfd/libbfd.la
 # Allow dependency tracking to work on all the source files.
 EXTRA_libopcodes_la_SOURCES = $(LIBOPCODES_CFILES)
 
