--- binutils-2.17.50.0.15.orig/bfd/elflink.c	2007-04-19 17:16:29.000000000 +0300
+++ binutils-2.17.50.0.15/bfd/elflink.c	2007-04-20 00:28:00.569581754 +0300
@@ -10057,7 +10057,7 @@
 	      if (dyn.d_tag == DT_TEXTREL)
 		{
 		 info->callbacks->einfo
-		    (_("%P: warning: creating a DT_TEXTREL in a shared object.\n"));
+		    (_("%P: warning: creating a DT_TEXTREL in object.\n"));
 		  break;
 		}
 	    }
--- binutils-2.17.50.0.15.orig/ld/testsuite/lib/ld-lib.exp	2007-03-16 17:48:30.000000000 +0200
+++ binutils-2.17.50.0.15/ld/testsuite/lib/ld-lib.exp	2007-04-20 00:28:30.065082031 +0300
@@ -191,6 +191,9 @@
     # symbol, since the default linker script might use ENTRY.
     regsub -all "(^|\n)(\[^\n\]*: warning: cannot find entry symbol\[^\n\]*\n?)" $exec_output "\\1" exec_output
 
+    # We want to ignore TEXTREL warnings since we force enable them by default
+    regsub -all "^lt-ld-new: warning: creating a DT_TEXTREL in object\." $exec_output "\\1" exec_output
+
     if [string match "" $exec_output] then {
 	return 1
     } else {
--- binutils-2.17.50.20070522/ld/ldmain.c	2007-05-22 11:24:17.000000000 +0300
+++ binutils-2.17.50.20070522/ld/ldmain.c	2007-05-22 11:25:51.000000000 +0300
@@ -266,6 +266,7 @@
   link_info.new_dtags = TRUE;
   link_info.combreloc = TRUE;
   link_info.strip_discarded = TRUE;
+  link_info.warn_shared_textrel = TRUE;
   link_info.emit_hash = TRUE;
   link_info.callbacks = &link_callbacks;
   link_info.input_bfds_tail = &link_info.input_bfds;
