--- docutils-0.3.7/setup.py	2004-07-29 16:57:28.000000000 +0200
+++ docutils-0.3.7/setup.py.new	2004-09-23 10:06:06.263010624 +0200
@@ -82,20 +82,15 @@
 List of (module name, minimum __version__ string, [attribute names])."""
 
 def get_extras():
-    extras = []
-    for module_name, version, attributes in extra_modules:
-        try:
-            module = __import__(module_name)
-            if version and module.__version__ < version:
-                raise ValueError
-            for attribute in attributes or []:
-                getattr(module, attribute)
-            print ('"%s" module already present; ignoring extras/%s.py.'
-                   % (module_name, module_name))
-        except (ImportError, AttributeError, ValueError):
-            extras.append(module_name)
-    return extras
-
+   # old method does not work for upgrading/downgrading docutils.
+   # so we use a simple rule, 'optparse' and 'textwrap' are not installed
+   # for >=python-2.3
+
+   pyver = sys.version_info
+   if pyver[0] == 2 and pyver[1] > 2:
+      return ["roman"]
+   else:
+      return ["optparse","textwrap","roman"]
 
 class dual_build_py(build_py):
 
