diff -up ypserv-2.21/scripts/Makefile.in.path ypserv-2.21/scripts/Makefile.in
--- ypserv-2.21/scripts/Makefile.in.path	2009-06-12 11:32:02.000000000 +0200
+++ ypserv-2.21/scripts/Makefile.in	2010-01-13 16:09:15.056375015 +0100
@@ -168,7 +168,7 @@ AUTOMAKE_OPTIONS = 1.7 gnits
 #
 CLEANFILES = *~ ${SCRIPTS}
 EXTRA_DIST = ${MANS} ${XMLS}
-varypdir = /var/yp
+varypdir = ${prefix}/../var/yp
 man_MANS = pwupdate.8 ypinit.8
 XMLS = pwupdate.8.xml ypinit.8.xml
 libexec_SCRIPTS = ypxfr_1perhour ypxfr_1perday ypxfr_2perday \
