diff -up ypserv-2.19/etc/ypserv.conf.slp-warning ypserv-2.19/etc/ypserv.conf
--- ypserv-2.19/etc/ypserv.conf.slp-warning	2004-04-01 12:06:38.000000000 +0200
+++ ypserv-2.19/etc/ypserv.conf	2008-09-25 13:12:59.000000000 +0200
@@ -17,9 +17,9 @@ dns: no
 files: 30
 
 # Should we register ypserv with SLP ?
-slp: no
+# slp: no
 # After how many seconds we should re-register ypserv with SLP ?
-slp_timeout: 3600
+# slp_timeout: 3600
 
 # xfr requests are only allowed from ports < 1024
 xfr_check_port: yes
