diff -up ypserv-2.19/ypxfr/ypxfr.c.respzero ypserv-2.19/ypxfr/ypxfr.c
--- ypserv-2.19/ypxfr/ypxfr.c.respzero	2005-09-02 16:22:11.000000000 +0200
+++ ypserv-2.19/ypxfr/ypxfr.c	2008-09-25 13:10:39.000000000 +0200
@@ -364,6 +364,7 @@ ypxfr (char *map, char *source_host, cha
   struct hostent *h;
   int sock, result;
 
+  memset(&resp_val, '\0', sizeof(resp_val));
   /* Name of the map file */
   if (strlen (path_ypdb) + strlen (target_domain) + strlen (map) + 3 < MAXPATHLEN)
     sprintf (dbName_orig, "%s/%s/%s", path_ypdb, target_domain, map);
