--- setup.py.orig	2010-11-02 01:23:03.000000000 +0200
+++ setup.py	2010-11-24 22:09:54.424473959 +0200
@@ -31,5 +31,4 @@
         'Topic :: Software Development :: Libraries :: Python Modules'],
     long_description = "\n".join(pydot.__doc__.split('\n')),
     py_modules = ['pydot', 'dot_parser'],
-    install_requires = ['pyparsing', 'setuptools'],
-    data_files = [('.', ['LICENSE', 'README'])] )
+    install_requires = ['pyparsing', 'setuptools'])
