diff -Nuar qtwayland-opensource-src-5.4.1/src/hardwareintegration/client/wayland-egl/qwaylandeglwindow.cpp qtwayland-opensource-src-5.4.11/src/hardwareintegration/client/wayland-egl/qwaylandeglwindow.cpp
--- qtwayland-opensource-src-5.4.1/src/hardwareintegration/client/wayland-egl/qwaylandeglwindow.cpp	2015-02-17 06:57:14.000000000 +0200
+++ src/hardwareintegration/client/wayland-egl/qwaylandeglwindow.cpp	2015-05-03 02:41:51.451646667 +0300
@@ -72,6 +72,13 @@
     // RasterGLSurface window which may have pure raster content.  In this case, where the
     // window is never actually made current, creating a wl_egl_window and EGL surface
     // should be avoided.
+    QSurfaceFormat sf = window->requestedFormat();
+    sf.setRedBufferSize(8);
+    sf.setGreenBufferSize(8);
+    sf.setBlueBufferSize(8);
+    sf.setAlphaBufferSize(8);
+    m_eglConfig = q_configFromGLFormat(m_clientBufferIntegration->eglDisplay(), sf);
+    m_format = q_glFormatFromConfig(m_clientBufferIntegration->eglDisplay(), m_eglConfig);
 }
 
 QWaylandEglWindow::~QWaylandEglWindow()
diff -Nuar qtwayland-opensource-src-5.4.1/src/hardwareintegration/client/wayland-egl/qwaylandglcontext.cpp qtwayland-opensource-src-5.4.11/src/hardwareintegration/client/wayland-egl/qwaylandglcontext.cpp
--- qtwayland-opensource-src-5.4.1/src/hardwareintegration/client/wayland-egl/qwaylandglcontext.cpp	2015-02-17 06:57:14.000000000 +0200
+++ src/hardwareintegration/client/wayland-egl/qwaylandglcontext.cpp	2015-05-03 02:49:20.404631708 +0300
@@ -226,6 +226,16 @@
     , m_blitter(0)
     , mUseNativeDefaultFbo(false)
 {
+    QSurfaceFormat sf = format;
+    sf.setRedBufferSize(8);
+    sf.setGreenBufferSize(8);
+    sf.setBlueBufferSize(8);
+    sf.setAlphaBufferSize(8);
+
+    m_config = q_configFromGLFormat(m_eglDisplay, sf);
+
+    m_format = q_glFormatFromConfig(m_eglDisplay, m_config);
+
     QSurfaceFormat fmt = format;
     if (static_cast<QWaylandIntegration *>(QGuiApplicationPrivate::platformIntegration())->display()->supportsWindowDecoration())
         fmt.setAlphaBufferSize(8);
