diff -ur ocaml-3.12.0/configure ocaml-3.12.0.tk-fix/configure
--- ocaml-3.12.0/configure	2010-07-28 15:18:22.000000000 +0200
+++ ocaml-3.12.0.tk-fix/configure	2010-08-27 09:51:07.000000000 +0200
@@ -1485,6 +1485,7 @@
     8.3) tclmaj=8 tclmin=3 tkmaj=8 tkmin=3 ;;
     8.4) tclmaj=8 tclmin=4 tkmaj=8 tkmin=4 ;;
     8.5) tclmaj=8 tclmin=5 tkmaj=8 tkmin=5 ;;
+    8.6) tclmaj=8 tclmin=6 tkmaj=8 tkmin=6 ;;
     *) echo "This version is not known."; has_tk=false ;;
     esac
   else
diff -ur ocaml-3.12.0/otherlibs/labltk/support/cltkDMain.c ocaml-3.12.0.tk-fix/otherlibs/labltk/support/cltkDMain.c
--- ocaml-3.12.0/otherlibs/labltk/support/cltkDMain.c	2010-04-03 08:43:51.000000000 +0200
+++ ocaml-3.12.0.tk-fix/otherlibs/labltk/support/cltkDMain.c	2010-08-27 09:49:23.000000000 +0200
@@ -16,6 +16,7 @@
 
 /* $Id: cltkDMain.c 10230 2010-04-03 06:43:51Z furuse $ */
 
+#define USE_INTERP_RESULT
 #include <unistd.h>
 #include <fcntl.h>
 #include <tcl.h>
diff -ur ocaml-3.12.0/otherlibs/labltk/support/cltkEval.c ocaml-3.12.0.tk-fix/otherlibs/labltk/support/cltkEval.c
--- ocaml-3.12.0/otherlibs/labltk/support/cltkEval.c	2010-04-03 08:43:51.000000000 +0200
+++ ocaml-3.12.0.tk-fix/otherlibs/labltk/support/cltkEval.c	2010-08-27 09:49:31.000000000 +0200
@@ -16,6 +16,7 @@
 
 /* $Id: cltkEval.c 10230 2010-04-03 06:43:51Z furuse $ */
 
+#define USE_INTERP_RESULT
 #include <stdlib.h>
 #include <string.h>
 
diff -ur ocaml-3.12.0/otherlibs/labltk/support/cltkMain.c ocaml-3.12.0.tk-fix/otherlibs/labltk/support/cltkMain.c
--- ocaml-3.12.0/otherlibs/labltk/support/cltkMain.c	2010-04-03 08:43:51.000000000 +0200
+++ ocaml-3.12.0.tk-fix/otherlibs/labltk/support/cltkMain.c	2010-08-27 09:49:37.000000000 +0200
@@ -16,6 +16,7 @@
 
 /* $Id: cltkMain.c 10230 2010-04-03 06:43:51Z furuse $ */
 
+#define USE_INTERP_RESULT
 #include <string.h>
 #include <tcl.h>
 #include <tk.h>
diff -ur ocaml-3.12.0/otherlibs/labltk/support/cltkMisc.c ocaml-3.12.0.tk-fix/otherlibs/labltk/support/cltkMisc.c
--- ocaml-3.12.0/otherlibs/labltk/support/cltkMisc.c	2010-04-03 08:43:51.000000000 +0200
+++ ocaml-3.12.0.tk-fix/otherlibs/labltk/support/cltkMisc.c	2010-08-27 09:49:40.000000000 +0200
@@ -16,6 +16,7 @@
 
 /* $Id: cltkMisc.c 10230 2010-04-03 06:43:51Z furuse $ */
 
+#define USE_INTERP_RESULT
 #include <string.h>
 #include <tcl.h>
 #include <tk.h>
diff -ur ocaml-3.12.0/otherlibs/labltk/support/cltkVar.c ocaml-3.12.0.tk-fix/otherlibs/labltk/support/cltkVar.c
--- ocaml-3.12.0/otherlibs/labltk/support/cltkVar.c	2010-04-03 08:43:51.000000000 +0200
+++ ocaml-3.12.0.tk-fix/otherlibs/labltk/support/cltkVar.c	2010-08-27 09:49:48.000000000 +0200
@@ -17,6 +17,7 @@
 /* $Id: cltkVar.c 10230 2010-04-03 06:43:51Z furuse $ */
 
 /* Alternative to tkwait variable */
+#define USE_INTERP_RESULT
 #include <string.h>
 #include <tcl.h>
 #include <tk.h>
diff -ur ocaml-3.12.0/otherlibs/labltk/support/cltkWait.c ocaml-3.12.0.tk-fix/otherlibs/labltk/support/cltkWait.c
--- ocaml-3.12.0/otherlibs/labltk/support/cltkWait.c	2010-04-03 08:43:51.000000000 +0200
+++ ocaml-3.12.0.tk-fix/otherlibs/labltk/support/cltkWait.c	2010-08-27 09:49:53.000000000 +0200
@@ -16,6 +16,7 @@
 
 /* $Id: cltkWait.c 10230 2010-04-03 06:43:51Z furuse $ */
 
+#define USE_INTERP_RESULT
 #include <tcl.h>
 #include <tk.h>
 #include <mlvalues.h>
