diff -ur ocaml-3.12.0.old/tools/Makefile.shared ocaml-3.12.0/tools/Makefile.shared
--- ocaml-3.12.0.old/tools/Makefile.shared	2010-06-07 07:58:41.000000000 +0100
+++ ocaml-3.12.0/tools/Makefile.shared	2011-01-04 21:56:13.023974253 +0000
@@ -108,9 +108,6 @@
 	sed -e "s|%%BINDIR%%|$(BINDIR)|" \
 	    -e "s|%%SUPPORTS_SHARED_LIBRARIES%%|$(SUPPORTS_SHARED_LIBRARIES)|" \
 	    -e "s|%%MKSHAREDLIB%%|$(MKSHAREDLIB)|" \
-	    -e "s|%%BYTECCRPATH%%|$(BYTECCRPATH)|" \
-	    -e "s|%%NATIVECCRPATH%%|$(NATIVECCRPATH)|" \
-	    -e "s|%%MKSHAREDLIBRPATH%%|$(MKSHAREDLIBRPATH)|" \
 	    -e "s|%%RANLIB%%|$(RANLIB)|" \
 	  ocamlmklib.mlp >> ocamlmklib.ml
 
