--- Makefile.orig	2011-01-25 08:33:19.000000000 -0700
+++ Makefile		2011-04-25 16:07:52.441724570 -0600
@@ -308,11 +308,11 @@

 all: $(MAIN)

-$(BINDIR): $(PREFIX)
-	$(MKDIR) $(BINDIR)
+$(DESTDIR)$(BINDIR): $(DESTDIR)$(PREFIX)
+	$(MKDIR) $(DESTDIR)$(BINDIR)

-$(PREFIX):
-	$(MKDIR) $(PREFIX)
+$(DESTDIR)$(PREFIX):
+	$(MKDIR) -p $(DESTDIR)$(PREFIX)

 default: $(OBJECTS) $(ADDITIONALOBJECTS) $(EXECS)

@@ -419,11 +419,11 @@

 clean:
 	/bin/rm -f *.o $(EXECS) $(MAIN)
-install: $(BINDIR)
-	cp $(EXECS) $(BINDIR)
+install: $(DESTDIR)$(BINDIR)
+	cp $(EXECS) $(DESTDIR)$(BINDIR)
 #	cp $(EXECS) /usr/local/bin
 #	./gfan installlinks --path $(BINDIR)/
-	cd $(BINDIR) && ./gfan installlinks
+	cd $(DESTDIR)$(BINDIR) && ./gfan installlinks
 library:
 	/bin/rm -f gfanlib/*.a
 	/bin/rm -f gfanlib/*.o