From 91a9ca62b70ded9ef6c624b1d7ca8333957ed9db Mon Sep 17 00:00:00 2001
From: Stefano Karapetsas <stefano@karapetsas.com>
Date: Wed, 05 Dec 2012 10:44:36 +0000
Subject: only glib.h can be included directly

---
diff --git a/src/tomboykeybinder.h b/src/tomboykeybinder.h
index 97c1887..ea70ef4 100644
--- a/src/tomboykeybinder.h
+++ b/src/tomboykeybinder.h
@@ -18,7 +18,7 @@
 #ifndef __TOMBOY_KEY_BINDER_H__
 #define __TOMBOY_KEY_BINDER_H__
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
--
cgit v0.9.0.3-65-g4555
