--- gutenprint-5.0.0/src/gimp2/print-print.c.menu	2007-03-02 17:08:51.000000000 +0000
+++ gutenprint-5.0.0/src/gimp2/print-print.c	2007-03-02 17:09:10.000000000 +0000
@@ -43,7 +43,7 @@
 			  (BAD_CONST_CHAR) copy,
 			  (BAD_CONST_CHAR) VERSION " - " RELEASE_DATE,
 			  /* Do not translate the prefix "<Image>" */
-			  (BAD_CONST_CHAR) N_("<Image>/File/Print..."),
+			  (BAD_CONST_CHAR) N_("<Image>/File/Send/Print..."),
 			  (BAD_CONST_CHAR) types,
 			  GIMP_PLUGIN,
 			  n_args, 0,
