Index: unzip60/unix/Makefile
===================================================================
--- unzip60.orig/unix/Makefile
+++ unzip60/unix/Makefile
@@ -280,13 +280,13 @@ unzipsdocs:	unzips docs
 # make utilities if default:  change "unzip$E:" to "unzip$E:&"
 
 unzip$E:	$(OBJS) $(LIBBZ2)	# add `&' for parallel makes
-	$(LD) $(LF) -L$(IZ_BZIP2) $(LOBJS) $(L_BZ2) $(LF2)
+	$(LD) $(LDFLAGS) $(LF) -L$(IZ_BZIP2) $(LOBJS) $(L_BZ2) $(LF2)
 
 unzipsfx$E:	$(OBJX)			# add `&' for parallel makes
-	$(LD) $(SL) $(LOBJX) $(SL2)
+	$(LD) $(LDFLAGS) $(SL) $(LOBJX) $(SL2)
 
 funzip$E:	$(OBJF)			# add `&' for parallel makes
-	$(LD) $(FL) $(OBJF) $(FL2)
+	$(LD) $(LDFLAGS) $(FL) $(OBJF) $(FL2)
 
 zipinfo$E:	unzip$E			# `&' is pointless here...
 	@echo\
@@ -545,7 +545,7 @@ generic:	flags	   # now try autoconfigur
 #	make $(MAKEF) unzips CF="${CF} `cat flags`"
 
 generic_gcc:
-	$(MAKE) $(MAKEF) generic CC=gcc IZ_BZIP2="$(IZ_BZIP2)"
+	$(MAKE) $(MAKEF) generic CC=$(CC) IZ_BZIP2="$(IZ_BZIP2)"
 
 # extensions to perform SVR4 package-creation after compilation
 generic_pkg:	generic svr4package
