diff -Nur pidgin-libnotify-0.13/src/pidgin-libnotify.c pidgin-libnotify-0.13.new/src/pidgin-libnotify.c
--- pidgin-libnotify-0.13/src/pidgin-libnotify.c	2007-09-04 12:05:27.000000000 -0300
+++ pidgin-libnotify-0.13.new/src/pidgin-libnotify.c	2007-09-04 12:06:05.000000000 -0300
@@ -312,6 +312,7 @@
 	g_hash_table_insert (buddy_hash, contact, notification);
 
 	g_object_set_data (G_OBJECT(notification), "contact", contact);
+	g_object_set_data (G_OBJECT(notification), "buddy", buddy);
 
 	g_signal_connect (notification, "closed", G_CALLBACK(closed_cb), NULL);
 
