--- pidgin-libnotify-0.14.orign/src/pidgin-libnotify.c	2008-12-14 18:45:51.000000000 +0100
+++ pidgin-libnotify-0.14.orign/src/pidgin-libnotify.c	2010-02-07 14:49:45.469825060 +0100
@@ -313,6 +313,8 @@ notify (const gchar *title,
 
 	g_object_set_data (G_OBJECT(notification), "contact", contact);
 
+        g_object_set_data (G_OBJECT(notification), "buddy", buddy);
+
 	g_signal_connect (notification, "closed", G_CALLBACK(closed_cb), NULL);
 
 	notify_notification_set_urgency (notification, NOTIFY_URGENCY_NORMAL); 
