Index: busybox-1.17.2/networking/udhcp/common.c
===================================================================
--- busybox-1.17.2.orig/networking/udhcp/common.c
+++ busybox-1.17.2/networking/udhcp/common.c
@@ -33,7 +33,7 @@ const struct dhcp_optflag dhcp_optflags[
 	{ OPTION_U16                              , 0x0d }, /* DHCP_BOOT_SIZE     */
 	{ OPTION_STRING               | OPTION_REQ, 0x0f }, /* DHCP_DOMAIN_NAME   */
 	{ OPTION_IP                               , 0x10 }, /* DHCP_SWAP_SERVER   */
-	{ OPTION_STRING                           , 0x11 }, /* DHCP_ROOT_PATH     */
+	{ OPTION_STRING               | OPTION_REQ, 0x11 }, /* DHCP_ROOT_PATH     */
 	{ OPTION_U8                               , 0x17 }, /* DHCP_IP_TTL        */
 	{ OPTION_U16                              , 0x1a }, /* DHCP_MTU           */
 	{ OPTION_IP                   | OPTION_REQ, 0x1c }, /* DHCP_BROADCAST     */
