--- avidemux/prefs.in.orig	2009-07-17 11:30:44.000000000 +0300
+++ avidemux/prefs.in	2009-07-17 11:31:39.000000000 +0300
@@ -39,7 +39,7 @@
 #ifdef __WIN32
 filters.subtitle.fontname,		FILENAME, "C:\\Windows\\Fonts\\arial.ttf"
 #else
-filters.subtitle.fontname,		FILENAME, "/usr/share/fonts/truetype/DejaVuSans.ttf"
+filters.subtitle.fontname,		FILENAME, "/usr/share/fonts/dejavu/DejaVuSans.ttf"
 #endif
 filters.subtitle.charset,		STRING, "ISO-8859-1"
 filters.subtitle.fontsize,		UINT,	24,	1,	576
