--- lxsession-logout.c	2013-05-31 22:38:58.000000000 +0300
+++ lxsession-logout/lxsession-logout.c	2014-09-15 13:04:04.103168913 +0300
@@ -689,7 +689,7 @@
             output[strlen ( output ) - 1] = '\0';
         }
 
-        prompt = g_strdup_printf(_("<b><big>Logout %s %s session ?</big></b>"), session_name, output);
+        prompt = g_strdup_printf(_("<b><big>PisiLinux 1.0 Oturumunu Kapat!</big></b>"), session_name, output);
     }
     gtk_label_set_markup(GTK_LABEL(label), prompt);
     gtk_box_pack_start(GTK_BOX(controls), label, FALSE, FALSE, 4);
