diff -Nuar R-2.12.1.orig//config.site R-2.12.1/config.site
--- R-2.12.1.orig//config.site	2010-11-14 01:05:06.000000000 +0200
+++ R-2.12.1/config.site	2010-12-22 07:11:10.429000021 +0200
@@ -28,7 +28,7 @@
 ## The paper size for the local (PostScript) printer.
 ## It must either be left blank or set to one of 'a4' or 'letter'.
 ## If left blank the system will use 'a4'.
-## R_PAPERSIZE=a4
+ R_PAPERSIZE=a4
 
 ## Set the default behavior of R when ending a session
 ## Set this to one of '--save' or '--no-save' depending whether you
@@ -207,23 +207,26 @@
 ## If unset configure searches in turn for (currently)
 ## firefox mozilla galeon opera xdg-open kfmclient gnome-moz-remote open
 ## and uses the full path.
-## R_BROWSER=
+R_BROWSER=xdg-open
+
+## Default pdf viewer
+R_PDFVIEWER=xdg-open
 
 ## BLAS and LAPACK settings
 ## Use BLAS_LIBS for all '-L' and '-l' options needed for linking
 ## against an external BLAS implementation.
-## BLAS_LIBS=
+## BLAS_LIBS=/usr/lib
 ## Use LAPACK_LIBS for all '-L' and '-l' options needed for linking
 ## against an external LAPACK implementation.  
 ## Note that (see R-admin) that our main intention is to allow a
 ## LAPACK-containing BLAS to be used, so this is rarely needed, and
 ## it is not used if the BLAS already contains LAPACK.
-## LAPACK_LIBS=
+## LAPACK_LIBS="-L/usr/lib -llapack -lblas"
 
 ## Make name.
 ## Set this if you want to use a make by another name.
 ## For example, if your GNU make is called 'gmake', use 'MAKE=gmake'.
-## MAKE=
+MAKE=gmake
 
 ## Tar name
 ## Set this to prefer a tar which has the capability to automagically
