diff -Nuar R-2.11.0.orig/configure.ac R-2.11.0/configure.ac
--- R-2.11.0.orig/configure.ac	2010-04-01 01:10:02.000000000 +0300
+++ R-2.11.0/configure.ac	2010-05-02 18:42:50.276871339 +0300
@@ -2508,7 +2508,6 @@
 if test "${acx_blas_ok}" = "yes"; then
   ## Try to figure out which BLAS was used.
   case "${BLAS_LIBS0}" in
-    *-latlas*) r_blas=ATLAS ;;
     *-lsgemm*) r_blas=PhiPack ;;
     *sunperf*) r_blas=SunPerf ;;
     *-lessl*)  r_blas=ESSL ;;
diff -Nuar R-2.11.0.orig/m4/R.m4 R-2.11.0/m4/R.m4
--- R-2.11.0.orig/m4/R.m4	2010-03-28 01:10:03.000000000 +0200
+++ R-2.11.0/m4/R.m4	2010-05-02 18:44:38.987876278 +0300
@@ -2536,19 +2536,6 @@
   fi
 fi
 
-## BLAS linked to by default?  (happens on some supercomputers)
-if test "${acx_blas_ok}" = no; then
-  AC_CHECK_FUNC(${dgemm}, [acx_blas_ok=yes])
-fi
-
-## BLAS in ATLAS library?  (http://math-atlas.sourceforge.net/)
-if test "${acx_blas_ok}" = no; then
-  AC_CHECK_LIB(atlas, ATL_xerbla,
-               [AC_CHECK_LIB(f77blas, ${dgemm},
-                             [acx_blas_ok=yes
-                              BLAS_LIBS="-lf77blas -latlas"],
-			     [], [-latlas])])
-fi
 
 ## BLAS in PhiPACK libraries?  (requires generic BLAS lib, too)
 if test "${acx_blas_ok}" = no; then
@@ -2744,7 +2731,6 @@
   F77_SYMBOL(drotmg)();
   F77_SYMBOL(dsbmv)();
   F77_SYMBOL(dscal)();
-  F77_SYMBOL(dsdot)();
   F77_SYMBOL(dspmv)();
   F77_SYMBOL(dspr)();
   F77_SYMBOL(dspr2)();
