diff -Nuar R-3.0.1.orig/configure.ac R-3.0.1/configure.ac
--- R-3.0.1.orig/configure.ac	2013-03-25 01:05:05.000000000 +0200
+++ R-3.0.1/configure.ac	2013-05-18 23:41:55.803629789 +0300
@@ -137,7 +137,7 @@
   linux*)
     ## Not all distros use this: some choose to march out of step  
     case "${host_cpu}" in
-      x86_64|mips64|ppc64|powerpc64|sparc64|s390x)
+      mips64|ppc64|powerpc64|sparc64|s390x)
         if test -d /usr/lib64; then
           libnn=lib64
 	fi
diff -Nuar R-3.0.1.orig/src/scripts/R.sh.in R-3.0.1/src/scripts/R.sh.in
--- R-3.0.1.orig/src/scripts/R.sh.in	2013-03-06 01:02:41.000000000 +0200
+++ R-3.0.1/src/scripts/R.sh.in	2013-05-18 23:42:28.075630689 +0300
@@ -7,13 +7,13 @@
    linux*)
      run_arch=`uname -m`
      case "$run_arch" in 
-        x86_64|mips64|ppc64|powerpc64|sparc64|s390x)
+        mips64|ppc64|powerpc64|sparc64|s390x)
           libnn=lib64
           libnn_fallback=lib
         ;;
         *)
           libnn=lib
-          libnn_fallback=lib64
+          libnn_fallback=lib
         ;;
      esac
      if [ -x "@prefix@/${libnn}/R/bin/exec/R" ]; then
