
Index: miro-3.5.1/linux/setup.py
===================================================================
--- miro-3.5.1.orig/linux/setup.py
+++ miro-3.5.1/linux/setup.py
@@ -242,7 +242,7 @@ then
         exit;
     fi
 
-    $GDB -ex 'set breakpoint pending on' -ex 'run' --args $PYTHON ./miro.real --sync "$@"
+    $GDB -ex 'set breakpoint pending on' -ex 'run' --args $PYTHON /usr/bin/miro.real --sync "$@"
 else
     miro.real "$@"
 fi
