
Index: lib/startup.py
===================================================================
--- lib/startup.py.orig
+++ lib/startup.py
@@ -368,7 +368,7 @@ def on_frontend_started():
     yield None
     commandline.startup()
     yield None
-    autoupdate.check_for_updates()
+    # autoupdate.check_for_updates()
     yield None
     # Wait a bit before starting the downloader daemon.  It can cause a bunch
     # of disk/CPU load, so try to avoid it slowing other stuff down.
