Index: miro-3.5.1/linux/setup.py
===================================================================
--- miro-3.5.1.orig/linux/setup.py
+++ miro-3.5.1/linux/setup.py
@@ -366,7 +366,7 @@ class install_data(distutils.command.ins
                              APP_REVISION_NUM=revisionnum,
                              APP_REVISION_URL=revisionurl,
                              APP_PLATFORM='linux',
-                             BUILD_MACHINE="%s@%s" % (getlogin(),
+                             BUILD_MACHINE="%s@%s" % ("PisiLinux",
                                                       os.uname()[1]),
                              BUILD_TIME=str(time.time()),
                              MOZILLA_LIB_PATH="")
