Index: miro-4.0.3/lib/item.py
===================================================================
--- miro-4.0.3.orig/lib/item.py
+++ miro-4.0.3/lib/item.py
@@ -159,6 +159,21 @@ class FeedParserValues(object):
 
     def _calc_thumbnail_url(self):
         """Returns a link to the thumbnail of the video.  """
+        # also hqdefault.jpg 0.jpg
+
+        if self._calc_url().startswith("http://www.youtube.com/watch?v=") is True:
+            yt_id = self._calc_url().split("=")[1]
+            if len(yt_id) is 11:
+                return "http://img.youtube.com/vi/" + yt_id + "/default.jpg"
+
+        if self._calc_url().startswith("http://www.youtube.com/v/") is True:
+            if self._calc_url().find("&") is not -1:
+                yt_id = self._calc_url().split("/v/")[1].split("&")[0]
+                if len(yt_id) is 11:
+                    return "http://img.youtube.com/vi/" + yt_id + "/default.jpg"
+
+        if self._calc_url().find("youtube") is not -1:
+            print "YouTube not handled! " + self._calc_url()
         # Try to get the thumbnail specific to the video enclosure
         if self.first_video_enclosure is not None:
             url = self._get_element_thumbnail(self.first_video_enclosure)
