diff -Nuar mutter-3.10.4.OSmanOS/configure.ac mutter-3.10.4/configure.ac
--- mutter-3.10.4.OSmanOS/configure.ac	2014-02-19 20:40:13.000000000 +0200
+++ mutter-3.10.4/configure.ac	2014-06-15 19:37:47.483705619 +0300
@@ -75,7 +75,7 @@
    gsettings-desktop-schemas >= 3.7.3
    xcomposite >= 0.2 xfixes xrender xdamage xi >= 1.6.0
    $CLUTTER_PACKAGE >= 1.15.90
-   cogl-1.0 >= 1.15.6
+   cogl-1.0 >= 1.17.1
    upower-glib > 0.9.11
    gnome-desktop-3.0
 "
diff -Nuar mutter-3.10.4.OSmanOS/src/compositor/meta-texture-rectangle.c mutter-3.10.4/src/compositor/meta-texture-rectangle.c
--- mutter-3.10.4.OSmanOS/src/compositor/meta-texture-rectangle.c	2014-02-19 20:19:50.000000000 +0200
+++ mutter-3.10.4/src/compositor/meta-texture-rectangle.c	2014-06-15 19:39:12.703703101 +0300
@@ -32,10 +32,8 @@
 meta_texture_rectangle_new (unsigned int width,
                             unsigned int height,
                             CoglPixelFormat format,
-                            CoglPixelFormat internal_format,
                             unsigned int rowstride,
-                            const guint8 *data,
-                            GError **error)
+                            const guint8 *data)
 {
   ClutterBackend *backend =
     clutter_get_default_backend ();
@@ -43,10 +41,7 @@
     clutter_backend_get_cogl_context (backend);
   CoglTextureRectangle *tex_rect;
 
-  tex_rect = cogl_texture_rectangle_new_with_size (context,
-                                                   width, height,
-                                                   internal_format,
-                                                   error);
+  tex_rect = cogl_texture_rectangle_new_with_size (context, width, height);
   if (tex_rect == NULL)
     return NULL;
 
diff -Nuar mutter-3.10.4.OSmanOS/src/compositor/meta-texture-rectangle.h mutter-3.10.4/src/compositor/meta-texture-rectangle.h
--- mutter-3.10.4.OSmanOS/src/compositor/meta-texture-rectangle.h	2014-02-19 20:19:50.000000000 +0200
+++ mutter-3.10.4/src/compositor/meta-texture-rectangle.h	2014-06-15 19:39:45.209702141 +0300
@@ -34,10 +34,8 @@
 meta_texture_rectangle_new (unsigned int width,
                             unsigned int height,
                             CoglPixelFormat format,
-                            CoglPixelFormat internal_format,
                             unsigned int rowstride,
-                            const guint8 *data,
-                            GError **error);
+                            const guint8 *data);
 
 gboolean
 meta_texture_rectangle_check (CoglTexture *texture);
diff -Nuar mutter-3.10.4.OSmanOS/src/compositor/meta-texture-tower.c mutter-3.10.4/src/compositor/meta-texture-tower.c
--- mutter-3.10.4.OSmanOS/src/compositor/meta-texture-tower.c	2014-02-19 20:19:50.000000000 +0200
+++ mutter-3.10.4/src/compositor/meta-texture-tower.c	2014-06-15 19:40:27.488700892 +0300
@@ -365,13 +365,9 @@
         meta_texture_rectangle_new (width, height,
                                     /* data format */
                                     TEXTURE_FORMAT,
-                                    /* internal cogl format */
-                                    TEXTURE_FORMAT,
                                     /* rowstride */
                                     width * 4,
                                     /* data */
-                                    NULL,
-                                    /* error */
                                     NULL);
     }
   else
diff -Nuar mutter-3.10.4.OSmanOS/src/compositor/meta-window-actor.c mutter-3.10.4/src/compositor/meta-window-actor.c
--- mutter-3.10.4.OSmanOS/src/compositor/meta-window-actor.c	2014-02-19 20:20:26.000000000 +0200
+++ mutter-3.10.4/src/compositor/meta-window-actor.c	2014-06-15 19:41:12.207699570 +0300
@@ -2209,10 +2209,7 @@
     {
       mask_texture = meta_texture_rectangle_new (tex_width, tex_height,
                                                  COGL_PIXEL_FORMAT_A_8,
-                                                 COGL_PIXEL_FORMAT_A_8,
-                                                 stride,
-                                                 mask_data,
-                                                 NULL /* error */);
+                                                 stride, mask_data);
     }
   else
     {
diff -Nuar mutter-3.10.4.OSmanOS/src/core/meta-cursor-tracker.c mutter-3.10.4/src/core/meta-cursor-tracker.c
--- mutter-3.10.4.OSmanOS/src/core/meta-cursor-tracker.c	2014-02-19 20:19:50.000000000 +0200
+++ mutter-3.10.4/src/core/meta-cursor-tracker.c	2014-06-15 19:41:55.950698278 +0300
@@ -207,7 +207,6 @@
                                           cursor_image->width,
                                           cursor_image->height,
                                           CLUTTER_CAIRO_FORMAT_ARGB32,
-                                          COGL_PIXEL_FORMAT_ANY,
                                           cursor_image->width * 4, /* stride */
                                           cursor_data,
                                           NULL);
