Index: rlplot/Makefile
===================================================================
--- rlplot.orig/Makefile
+++ rlplot/Makefile
@@ -18,8 +18,8 @@
 #    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 #
 #
-CC = g++
-X11LIBS = -lX11 -lm
+CC = $(CXX)
+X11LIBS = -lm
 SRCDIR = ./
 
 #######
@@ -40,9 +40,9 @@ QT3LIBS = "-L$(QT3DIR)/lib -L/usr/X11R6/
 #######
 #######   Declarations for Qt4
 QT4DIR = /usr/lib/qt4
-QT4MOC = /usr/lib/qt4/bin/moc-qt4
+QT4MOC = /usr/bin/moc
 MOC4FLAGS = "MOCFLAGS=-DQT_VERSION=0x040000"
-QT4CFLAGS = "-I/usr/include/Qt -pipe -O2"
+QT4CFLAGS = "$(CFLAGS) $(LDFLAGS) -I/usr/lib/qt4 -I/usr/include/Qt -pipe -O2"
 QT4H = QT_Spec.h
 QT4LIBS = "-L$(QT4DIR)/lib -L/usr/X11R6/lib -lQtCore -lQtGui"
 #
