Index: rlplot/QT_Spec.cpp
===================================================================
--- rlplot.orig/QT_Spec.cpp
+++ rlplot/QT_Spec.cpp
@@ -1238,7 +1238,8 @@ bool
 com_GetTextExtentW(w_char  *txt, int *width, int *height, int cb, TextDEF *TxtSet, QPainter *qP)
 {
 	int i;
-	QString wtxt(0);
+	int tmp=0;
+	QString wtxt(tmp);
 
 	if(!txt || !txt[0]) return com_QStringExt(QString("a"), width, height, 1, TxtSet, qP);
 	for(i = 0; txt[i] && i < cb; i++) wtxt.append(QChar(txt[i]));
@@ -1326,7 +1327,8 @@ bool com_TextOut(int x, int y, char *ctx
 bool com_TextOutW(int x, int y, w_char *wtxt, TextDEF *TxtSet, QPainter *qP, anyOutput *o)
 {
 	int i;
-	QString txt(0);
+	int tmp=0;
+	QString txt(tmp);
 
 	if(!wtxt || !wtxt[0] || !TxtSet || !qP || !o) return false;
 	for(i = 0; wtxt[i]; i++) txt.append(QChar(wtxt[i]));
