--- rlplot.orig/Makefile	2008-05-02 21:48:33.000000000 +0600
+++ rlplot/Makefile	2012-05-09 14:04:51.577277180 +0600
@@ -19,13 +19,13 @@
 #
 #
 CC = g++
-X11LIBS = -lX11 -lm
+X11LIBS = -lX11 -lm -lpthread
 SRCDIR = ./
 
 #######
 #######   Declarations for Qt3
-QT3DIR = /usr/lib/qt-3.3
-QT3MOC = /usr/lib/qt-3.3/bin/moc
+QT3DIR = /opt/qt
+QT3MOC = ${QT3DIR}/bin/moc
 MOC3FLAGS =
 QT3CFLAGS = "-I$(QT3DIR)/include -pipe -O2"
 QT3H = QT3_Spec.h
@@ -39,8 +39,8 @@
 
 #######
 #######   Declarations for Qt4
-QT4DIR = /usr/lib/qt4
-QT4MOC = /usr/lib/qt4/bin/moc-qt4
+QT4DIR = /usr
+QT4MOC = ${QT4DIR}/bin/moc
 MOC4FLAGS = "MOCFLAGS=-DQT_VERSION=0x040000"
 QT4CFLAGS = "-I/usr/include/Qt -pipe -O2"
 QT4H = QT_Spec.h
--- rlplot.orig/QT_Spec.cpp	2008-05-03 00:54:14.000000000 +0600
+++ rlplot/QT_Spec.cpp	2012-05-09 14:04:35.270610422 +0600
@@ -1238,7 +1238,7 @@
 com_GetTextExtentW(w_char  *txt, int *width, int *height, int cb, TextDEF *TxtSet, QPainter *qP)
 {
 	int i;
-	QString wtxt(0);
+	QString wtxt;
 
 	if(!txt || !txt[0]) return com_QStringExt(QString("a"), width, height, 1, TxtSet, qP);
 	for(i = 0; txt[i] && i < cb; i++) wtxt.append(QChar(txt[i]));
@@ -1326,7 +1326,7 @@
 bool com_TextOutW(int x, int y, w_char *wtxt, TextDEF *TxtSet, QPainter *qP, anyOutput *o)
 {
 	int i;
-	QString txt(0);
+	QString txt;
 
 	if(!wtxt || !wtxt[0] || !TxtSet || !qP || !o) return false;
 	for(i = 0; wtxt[i]; i++) txt.append(QChar(wtxt[i]));
