diff -Nuar package-manager-4.0.0.orig/src/packagemodel.py package-manager-4.0.0/src/packagemodel.py
--- package-manager-4.0.0.orig/src/packagemodel.py	2013-12-24 23:35:17.000000000 +0200
+++ package-manager-4.0.0/src/packagemodel.py	2014-01-12 16:28:56.483228983 +0200
@@ -47,7 +47,7 @@
 
         if config.USE_APPINFO:
             self.appinfo = AppInfoClient()
-            self.appinfo.setServer('http://appinfo.pardus.org.tr')
+            self.appinfo.setServer('http://appinfo.pisilinux.org')
             if not self.appinfo.initializeLocalDB()[0]:
                 self.appinfo.checkOutDB()
 
diff -Nuar package-manager-4.0.0.orig/src/pmutils.py package-manager-4.0.0/src/pmutils.py
--- package-manager-4.0.0.orig/src/pmutils.py	2013-12-24 23:35:16.000000000 +0200
+++ package-manager-4.0.0/src/pmutils.py	2014-01-12 16:29:17.453228944 +0200
@@ -156,7 +156,7 @@
     QNetworkProxy.setApplicationProxy(QNetworkProxy())
 
 def network_available():
-    return pisi.fetcher.Fetcher('http://appinfo.pardus.org.tr').test()
+    return pisi.fetcher.Fetcher('http://appinfo.pisilinux.org').test()
 
 def parse_proxy(line):
     settings = {'domain':None,'user':None,'pass':None,'host':None,'port':None}
diff -Nuar package-manager-4.0.0.orig/src/statemanager.py package-manager-4.0.0/src/statemanager.py
--- package-manager-4.0.0.orig/src/statemanager.py	2013-12-24 23:35:17.000000000 +0200
+++ package-manager-4.0.0/src/statemanager.py	2014-01-12 16:29:49.533228560 +0200
@@ -158,7 +158,7 @@
         if config.USE_APPINFO:
             if network_available():
                 if not AppInfoClient().checkOutDB()[0]:
-                    AppInfoClient().setServer('http://appinfo.pardus.org.tr')
+                    AppInfoClient().setServer('http://appinfo.pisilinux.org')
                     AppInfoClient().checkOutDB()
 
         return True
diff -Nuar package-manager-4.0.0.orig/src/webdialog.py package-manager-4.0.0/src/webdialog.py
--- package-manager-4.0.0.orig/src/webdialog.py	2013-12-24 23:35:17.000000000 +0200
+++ package-manager-4.0.0/src/webdialog.py	2014-01-12 16:30:21.433228258 +0200
@@ -41,7 +41,7 @@
 
         self.registerFunction(IN, lambda: parent.statusBar().hide())
         self.registerFunction(FINISHED, lambda: parent.statusBar().setVisible(not self.isVisible()))
-        self._as = 'http://appinfo.pardus.org.tr'
+        self._as = 'http://appinfo.pisilinux.org'
         self.cancelButton.clicked.connect(self._hide)
         self.cancelButton.setIcon(KIcon("cancel"))
 
