diff -Nuar package-manager-4.0.0-OSmanOS/src/summarydialog.py package-manager-4.0.0/src/summarydialog.py
--- package-manager-4.0.0-OSmanOS/src/summarydialog.py	2013-12-24 23:35:17.000000000 +0200
+++ package-manager-4.0.0/src/summarydialog.py	2014-12-07 19:07:40.849356751 +0200
@@ -80,6 +80,19 @@
         for desktopFile in desktopFiles:
             self.addApplication(desktopFile)
 
+
+    def checkIcon(self, iconFileName):
+        extensions = ['png', 'jpg','jpeg', 'svg']
+        pixmapsDir = '/usr/share/pixmaps/'
+        if os.path.isfile(iconFileName) == True:
+            return iconFileName
+        else:
+            for ext in extensions:
+                fname = '%s%s.%s' % (pixmapsDir, iconFileName, ext)
+                if os.path.isfile(fname):
+                    return fname
+
+          
     def addApplication(self, desktopFile):
         parser = desktopparser.DesktopParser()
         parser.read("/%s" % str(desktopFile))
@@ -89,7 +102,7 @@
         if nodisplay == "true" or terminal == "true":
             return
 
-        icon = parser.safe_get_locale('Desktop Entry', 'Icon', None)
+        icon = self.checkIcon(parser.safe_get_locale('Desktop Entry', 'Icon', None))
         command = parser.safe_get_locale('Desktop Entry', 'Exec', None)
         if not command:
             return
@@ -117,4 +130,3 @@
     def showSummary(self):
         if self.hasApplication():
             self.show()
-
