--- common/include/lib.mak.in~	2007-06-14 06:16:11.000000000 +0200
+++ common/include/lib.mak.in	2009-01-10 16:51:00.000000000 +0100
@@ -25,8 +25,8 @@
 	ar rcs $(STATIC_LIB) $(OBJECTS)
 
 $(SHARED_LIB): $(OBJECTS)
-	$(CC) -shared -Wl,-soname,$(SO_NAME) $(LDFLAGS) $(LDLIBS) \
-	      -o $(SHARED_LIB) $(OBJECTS)
+	$(CC) -shared -Wl,-soname,$(SO_NAME) $(LDFLAGS) $(OBJECTS) $(LDLIBS) \
+	      -o $(SHARED_LIB)
 
 install: install_header install_static install_shared
 
--- common/include/plugin.mak~	2007-06-14 06:16:11.000000000 +0200
+++ common/include/plugin.mak	2009-01-10 16:55:09.000000000 +0100
@@ -16,7 +16,7 @@
 all: $(LIB_NAME)
 
 $(LIB_NAME): $(OBJECTS)
-	$(CC) -shared $(LDFLAGS) $(LDLIBS) -o $(LIB_NAME) $(OBJECTS)
+	$(CC) -shared $(LDFLAGS) $(OBJECTS) $(LDLIBS) -o $(LIB_NAME)
 
 install: $(LIB_NAME)
 	install -D $(LIB_NAME) $(DEST_INST_DIR)/$(LIB_NAME)
