--- taglib/CMakeLists.txt.orig	2009-09-21 11:53:17.000000000 +0300
+++ taglib/CMakeLists.txt	2009-09-21 11:56:28.000000000 +0300
@@ -10,6 +10,7 @@
     ${CMAKE_CURRENT_SOURCE_DIR}/mpc
     ${CMAKE_CURRENT_SOURCE_DIR}/mp4
     ${CMAKE_CURRENT_SOURCE_DIR}/ogg/vorbis
+    ${CMAKE_CURRENT_SOURCE_DIR}/wma
     ${CMAKE_CURRENT_SOURCE_DIR}/ogg/speex
     ${CMAKE_CURRENT_SOURCE_DIR}/mpeg/id3v2
     ${CMAKE_CURRENT_SOURCE_DIR}/mpeg/id3v1
@@ -36,9 +37,17 @@
 ADD_SUBDIRECTORY( wavpack )
 ADD_SUBDIRECTORY( trueaudio )
 ADD_SUBDIRECTORY( riff )
+ADD_SUBDIRECTORY( wma )
 
 ########### next target ###############
 
+SET(wma_SRCS
+wma/asf.cpp
+wma/wmafile.cpp
+wma/wmaproperties.cpp
+wma/wmatag.cpp
+)
+
 SET(mpeg_SRCS
 mpeg/mpegfile.cpp
 mpeg/mpegproperties.cpp
@@ -175,7 +184,7 @@
 SET(tag_LIB_SRCS ${mpeg_SRCS} ${id3v1_SRCS} ${id3v2_SRCS} ${frames_SRCS} ${ogg_SRCS}
 		 ${vorbis_SRCS} ${oggflacs_SRCS} ${mpc_SRCS} ${ape_SRCS} ${toolkit_SRCS} ${flacs_SRCS}
 		 ${wavpack_SRCS} ${speex_SRCS} ${trueaudio_SRCS} ${riff_SRCS} ${aiff_SRCS} ${wav_SRCS}
-		 ${mp4_SRCS} ${asf_SRCS}
+		 ${mp4_SRCS} ${asf_SRCS} ${wma_SRCS}
 		 tag.cpp
 		 tagunion.cpp
 		 fileref.cpp
--- taglib/fileref.cpp.orig	2009-09-21 11:57:19.000000000 +0300
+++ taglib/fileref.cpp	2009-09-21 12:00:32.000000000 +0300
@@ -43,6 +43,7 @@
 #include "trueaudiofile.h"
 #include "aifffile.h"
 #include "wavfile.h"
+#include "wmafile.h"
 
 using namespace TagLib;
 
@@ -143,6 +144,7 @@
   l.append("aif");
   l.append("aiff");
   l.append("wav");
+  l.append("wma");
 
   return l;
 }
@@ -235,6 +237,8 @@
       return new RIFF::WAV::File(fileName, readAudioProperties, audioPropertiesStyle);
     if(ext == "AIFF")
       return new RIFF::AIFF::File(fileName, readAudioProperties, audioPropertiesStyle);
+    if(ext == "WMA")
+      return new WMA::File(fileName, readAudioProperties, audioPropertiesStyle);
   }
 
   return 0;
