diff -up 4Suite-XML-1.0.2/Ft/Xml/src/expat/lib/xmltok.c.expat-dos 4Suite-XML-1.0.2/Ft/Xml/src/expat/lib/xmltok.c
--- 4Suite-XML-1.0.2/Ft/Xml/src/expat/lib/xmltok.c.expat-dos	2006-04-28 21:54:54.000000000 +0200
+++ 4Suite-XML-1.0.2/Ft/Xml/src/expat/lib/xmltok.c	2009-11-03 14:18:03.994197360 +0100
@@ -328,13 +328,13 @@ utf8_updatePosition(const ENCODING *enc,
                     const char *end,
                     POSITION *pos)
 {
-  while (ptr != end) {
+  while (ptr < end) {
     unsigned char ch = (unsigned char)*ptr;
     if (ch >= 32) {
       if (ch < 128)
         ptr++;
       else
-        ptr += utf8_code_length[ch];
+        ptr += utf8_code_length[ch] ? utf8_code_length[ch] : 1;
       pos->columnNumber++;
     } else if (ch == 10) {
       pos->columnNumber = 0;
diff -up 4Suite-XML-1.0.2/Ft/Xml/src/expat/lib/xmltok_impl.c.expat-dos 4Suite-XML-1.0.2/Ft/Xml/src/expat/lib/xmltok_impl.c
--- 4Suite-XML-1.0.2/Ft/Xml/src/expat/lib/xmltok_impl.c.expat-dos	2006-04-28 21:54:54.000000000 +0200
+++ 4Suite-XML-1.0.2/Ft/Xml/src/expat/lib/xmltok_impl.c	2009-11-03 14:17:55.169954596 +0100
@@ -1742,7 +1742,7 @@ PREFIX(updatePosition)(const ENCODING *e
                        const char *end,
                        POSITION *pos)
 {
-  while (ptr != end) {
+  while (ptr < end) {
     switch (BYTE_TYPE(enc, ptr)) {
 #define LEAD_CASE(n) \
     case BT_LEAD ## n: \
