diff --git a/modules/ntrack-libnl.c b/modules/ntrack-libnl.c
index dbae740..d9475f6 100644
--- a/modules/ntrack-libnl.c
+++ b/modules/ntrack-libnl.c
@@ -530,7 +530,7 @@ update_connectivity (ntrack_monitor_t *self, struct _nl_info *nl_info)
 		op_default_route_disappeared ((struct _ntrack_monitor_arch*) self, nl_info);
 	} else if (nl_info->topmost_route && topmost_route) {
 		int diff_bits;
-		if ((diff_bits = (route_obj_ops.oo_id_attrs | ROUTE_ATTR_OIF
+		if ((diff_bits = (nl_object_get_id_attrs(&route_obj_ops) | ROUTE_ATTR_OIF
 		                  | ROUTE_ATTR_GATEWAY) &
 		                  nl_object_diff (OBJ_CAST (nl_info->topmost_route),
 		                                  OBJ_CAST (topmost_route)))) {
