diff -Nuar package-manager-3.0.3.orig/src/packagemodel.py package-manager-3.0.3/src/packagemodel.py
--- package-manager-3.0.3.orig/src/packagemodel.py	2011-02-17 08:52:06.000000000 +0200
+++ package-manager-3.0.3/src/packagemodel.py	2013-04-07 03:51:43.053544918 +0300
@@ -46,7 +46,7 @@
         self.state = parent.state
 
         self.appinfo = AppInfoClient()
-        self.appinfo.setServer('http://appinfo.pardus.org.tr')
+        self.appinfo.setServer('http://appinfo.pisilinux.org')
         if not self.appinfo.initializeLocalDB()[0]:
             self.appinfo.checkOutDB()
 
diff -Nuar package-manager-3.0.3.orig/src/pmutils.py package-manager-3.0.3/src/pmutils.py
--- package-manager-3.0.3.orig/src/pmutils.py	2011-07-04 15:25:17.000000000 +0300
+++ package-manager-3.0.3/src/pmutils.py	2013-04-07 03:50:55.648543595 +0300
@@ -156,7 +156,7 @@
     QNetworkProxy.setApplicationProxy(QNetworkProxy())
 
 def network_available():
-    return pisi.fetcher.Fetcher('http://appinfo.pardus.org.tr').test()
+    return pisi.fetcher.Fetcher('http://appinfo.pisilinux.org').test()
 
 def parse_proxy(line):
     settings = {'domain':None,'user':None,'pass':None,'host':None,'port':None}
diff -Nuar package-manager-3.0.3.orig/src/statemanager.py package-manager-3.0.3/src/statemanager.py
--- package-manager-3.0.3.orig/src/statemanager.py	2011-02-28 16:08:50.000000000 +0200
+++ package-manager-3.0.3/src/statemanager.py	2013-04-07 03:51:30.529544568 +0300
@@ -162,7 +162,7 @@
 
         if network_available():
             if not AppInfoClient().checkOutDB()[0]:
-                AppInfoClient().setServer('http://appinfo.pardus.org.tr')
+                AppInfoClient().setServer('http://appinfo.pisilinux.org')
                 AppInfoClient().checkOutDB()
 
         return True
diff -Nuar package-manager-3.0.3.orig/src/webdialog.py package-manager-3.0.3/src/webdialog.py
--- package-manager-3.0.3.orig/src/webdialog.py	2011-06-13 15:46:15.000000000 +0300
+++ package-manager-3.0.3/src/webdialog.py	2013-04-07 03:51:10.266544003 +0300
@@ -41,7 +41,7 @@
 
         self.registerFunction(IN, lambda: parent.statusBar().hide())
         self.registerFunction(FINISHED, lambda: parent.statusBar().setVisible(not self.isVisible()))
-        self._as = 'http://appinfo.pardus.org.tr'
+        self._as = 'http://appinfo.pisilinux.org'
         self.cancelButton.clicked.connect(self._hide)
         self.cancelButton.setIcon(KIcon("dialog-close"))
 
