Index: package-manager-3.0.3/src/pmutils.py
===================================================================
--- package-manager-3.0.3.orig/src/pmutils.py
+++ package-manager-3.0.3/src/pmutils.py
@@ -114,7 +114,7 @@ class PM:
 
 def get_real_paths(packages):
     # If packages are not from repo or remote, find their absolute paths
-    return map(lambda x: x if not x.endswith('pisi') or '://' in x else os.path.abspath(x), packages)
+    return map(lambda x: x if not x.endswith('.pisi') or '://' in x else os.path.abspath(x), packages)
 
 def isAllLocal(packages):
     return all(map(lambda x: x.endswith('.pisi') and not '://' in x, packages))
