--- ui/webdialog.ui.old	2014-02-13 00:22:36.000000000 +0400
+++ ui/webdialog.ui	2014-02-13 00:10:47.000000000 +0400
@@ -232,15 +232,21 @@
             </item>
             <item>
              <widget class="QPushButton" name="cancelButton">
+              <property name="sizePolicy">
+               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
+                <horstretch>0</horstretch>
+                <verstretch>0</verstretch>
+               </sizepolicy>
+              </property>
               <property name="minimumSize">
                <size>
-                <width>32</width>
+                <width>36</width>
                 <height>26</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
-                <width>24</width>
+                <width>36</width>
                 <height>26</height>
                </size>
               </property>
@@ -256,7 +262,7 @@
 </string>
               </property>
               <property name="text">
-               <string notr="true"/>
+               <string notr="true">X</string>
               </property>
               <property name="flat">
                <bool>true</bool>
