diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' perl-tk-804.029~/pod/TixGrid.pod perl-tk-804.029/pod/TixGrid.pod
--- perl-tk-804.029~/pod/TixGrid.pod	2007-02-10 08:54:37.000000000 +0000
+++ perl-tk-804.029/pod/TixGrid.pod	2010-05-30 10:01:07.000000000 +0100
@@ -181,15 +181,15 @@
 
 =over 8
 
-=item B<x_region>
+=item B<x_margin>
 
 The horizontal margin.
 
-=item B<y_region>
+=item B<y_margin>
 
 The vertical margin.
 
-=item B<s_region>
+=item B<s_margin>
 
 The area in the top left corner where the horizontal and vertical margins
 are joined.
