diff -Naurp Tk-804.027/Tk/MakeDepend.pm Tk-804.027.new/Tk/MakeDepend.pm
--- Tk-804.027/Tk/MakeDepend.pm	2003-12-13 07:28:22.000000000 -0500
+++ Tk-804.027.new/Tk/MakeDepend.pm	2006-06-13 22:34:21.000000000 -0400
@@ -248,6 +248,7 @@ sub command_line
  # force /usr/include to be last element of @include
  push @include, $Config{'usrinc'}
    if (defined $Config{'usrinc'} and $Config{'usrinc'} ne '');
+ push @include, "/usr/include";
  # warn "Include:@include\n";
  while (@files)
   {
diff -Naurp Tk-804.027/myConfig Tk-804.027.new/myConfig
--- Tk-804.027/myConfig	2004-02-28 11:59:28.000000000 -0500
+++ Tk-804.027.new/myConfig	2006-06-13 22:27:53.000000000 -0400
@@ -309,7 +309,7 @@ if ($win_arch eq 'x') {
 
   die "Cannot find -lX11 anywhere" unless(defined $xlib);
 
-  ($base) = $xlib =~ m#-L(.*)(?:/lib)$#x;
+  ($base) = $xlib =~ m#-L(.*)(?:/lib(64))$#x;
   if (defined $X11INC)
     {
       $xinc = &IX11("$X11INC");
@@ -321,6 +321,7 @@ if ($win_arch eq 'x') {
     {
       warn "Cannot find X include files via $base/include\n";
       $xinc = &IX11(map("$_/include",@xdirs),
+            chooseX11(</usr/include/X11*>),
 		    '/usr/openwin/include',
 		    chooseX11(</usr/X11*/include>),
 		    chooseX11(</usr/include/X11*>),
