--- quota-tools/rquota_svc.c.orig	2005-06-01 03:21:30.000000000 -0400
+++ quota-tools/rquota_svc.c	2006-11-01 15:11:29.161101000 -0500
@@ -140,7 +140,7 @@ static void parse_options(int argc, char
 				break;
 			case 'p': 
 				port = strtol(optarg, &endptr, 0);
-				if (*endptr || port <= 0) {
+				if (*endptr || port <= 0 || port > 0xffff) {
 					errstr(_("Illegal port number: %s\n"), optarg);
 					show_help();
 					exit(1);
