--- a/warnquota.conf
+++ b/warnquota.conf
@@ -27,8 +27,9 @@ MESSAGE         = Your disk usage has exceeded the agreed limits\
 # is created)
 SIGNATURE	= root@example.com
 # Following text is used for mails about group exceeding quotas
-GROUP_MESSAGE	= Hello, a group '%i' you're member of use too much space at %h.|\
-I chose you to do the cleanup.|Delete group files on the following filesystems:|
+GROUP_MESSAGE  = Hello,|\
+your group %i is using too much disk space at %h.|\
+I suggest you to clean up group files on the following filesystems:|
 # Text in the end of the mail to the group (if not specified, default text using SUPPORT
 # and PHONE is created).
 GROUP_SIGNATURE	= See you!|			Your admin|