--- scim-1.4.2/src/scim-restart.in.1-restart	2005-09-02 12:05:35.000000000 +0900
+++ scim-1.4.2/src/scim-restart.in	2005-08-25 22:52:29.000000000 +0900
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+SOCKET_CMD="@SCIM_LIBEXECDIR@/scim-launcher -d -c `scim-config-agent -c global -g /DefaultConfigModule` -e all -f socket --no-stay"
+XIM_CMD="@SCIM_LIBEXECDIR@/scim-launcher -d -c socket -e socket -f x11"
+
+OPT="-u `id -u`"
+
+pkill ${OPT} -f "${SOCKET_CMD}" && ${SOCKET_CMD}
+pkill ${OPT} -f "${XIM_CMD}" && ${XIM_CMD}
+pkill ${OPT} scim-bridge
--- scim-1.4.2/src/Makefile.am.1-restart	2005-07-10 22:32:23.000000000 +0900
+++ scim-1.4.2/src/Makefile.am	2005-09-02 11:42:57.000000000 +0900
@@ -141,6 +141,7 @@
 
 
 bin_PROGRAMS		= scim scim-config-agent
+bin_SCRIPTS		= scim-restart
 
 scim_SOURCES	 	= scim.cpp 
 scim_LDADD		= libscim-1.0.la
--- scim-1.4.2/configure.ac.1-restart	2005-08-16 15:56:59.000000000 +0900
+++ scim-1.4.2/configure.ac	2005-09-02 11:42:57.000000000 +0900
@@ -516,6 +516,7 @@
 		 intl/Makefile
 		 po/Makefile.in
 		 src/Makefile
+		 src/scim-restart
 		 src/scim_types.h
 		 utils/Makefile
 		 data/Makefile
