--- auto/install.orig	2008-06-06 23:32:22.000000000 +0300
+++ auto/install	2008-06-06 23:35:37.000000000 +0300
@@ -54,8 +54,8 @@
 	test -d '\$(DESTDIR)`dirname "$NGX_HTTP_LOG_PATH"`' || \
 		mkdir -p '\$(DESTDIR)`dirname "$NGX_HTTP_LOG_PATH"`'
 
-	test -d '\$(DESTDIR)$NGX_PREFIX/html' \
-		|| cp -r html '\$(DESTDIR)$NGX_PREFIX'
+	test -d '\$(DESTDIR)$NGX_PREFIX/htdocs' \
+		|| mkdir -p '\$(DESTDIR)$NGX_PREFIX/htdocs' && cp -r html/* '\$(DESTDIR)$NGX_PREFIX/htdocs'
 END
 
 
