--- quota-tools/warnquota.conf.orig	2005-07-19 11:51:07.000000000 -0400
+++ quota-tools/warnquota.conf	2007-02-15 14:58:38.000000000 -0500
@@ -4,16 +4,16 @@
 # and even blank lines
 
 # values can be quoted:
-MAIL_CMD 	= "/usr/my/sendmail/instead/sendmail -t"
-FROM 		= "bas@localhost"
+MAIL_CMD 	= "/usr/sbin/sendmail -t"
+FROM 		= 
 # but they don't have to be:
-SUBJECT 	= Hey, user, clean up your account!
-CC_TO 		= "sysadm@localhost"
+SUBJECT 	= NOTE: You are exceeding your allocated disk space limits
+CC_TO 		= "root@localhost"
 # If you set this variable CC will be used only when user has less than
 # specified grace time left (examples of possible times: 5 seconds, 1 minute,
 # 12 hours, 5 days)
 # CC_BEFORE = 2 days
-SUPPORT 	= "support@myhost.com"
+SUPPORT 	= "root@myhost.com"
 PHONE 		= "(123) 456-1111 or (222) 333-4444"
 # Text in the beginning of the mail (if not specified, default text is used)
 # This way text can be split to more lines
@@ -21,11 +21,11 @@ PHONE 		= "(123) 456-1111 or (222) 333-4
 # The expressions %i, %h, %d, and %% are substituted for user/group name,
 # host name, domain name, and '%' respectively. For backward compatibility
 # %s behaves as %i but is deprecated.
-MESSAGE         = Hello user %i, I've noticed you use too much space\
- on my disk in %h.%d.|Delete your files on the following filesystems:|
+MESSAGE         = Your disk usage has exceeded the agreed limits\
+ on this server|Please delete any unnecessary files on following filesystems:|
 # Text in the end of the mail (if not specified, default text using SUPPORT and PHONE
 # is created)
-SIGNATURE	= See you!|			Your admin of %h|
+SIGNATURE	= root@localhost
 # Following text is used for mails about group exceeding quotas
 GROUP_MESSAGE	= Hello, a group '%i' you're member of use too much space at %h.|\
 I chose you to do the cleanup.|Delete group files on the following filesystems:|
@@ -52,10 +52,10 @@ GROUP_SIGNATURE	= See you!|			Your admin
 #    if the attribute isn't found
 # if binddn and bindpw are blank or left out, an anonymous bind is used
 # 
-# LDAP_MAIL = false # or false if you don't want to use it
-# If you have at least LDAP 2.3 installed, you can use LDAP_URI
+# LDAP_MAIL = false # or true if you want to use it
+# If you have at least LDAP 2.3 installed, you should use LDAP_URI
 # LDAP_URI = ldaps://my.server:389
-# Otherwise you can specify LDAP_HOST and LDAP_PORT
+# Otherwise you should specify LDAP_HOST and LDAP_PORT
 # LDAP_HOST = ldap
 # LDAP_PORT = 389
 # LDAP_BINDDN = uid=ReadOnlyUser,o=YourOrg
diff -urNp quota-tools-orig/warnquota.c quota-tools/warnquota.c
--- quota-tools-orig/warnquota.c	2008-10-13 19:29:18.000000000 +0200
+++ quota-tools/warnquota.c	2009-03-13 14:26:33.000000000 +0100
@@ -806,7 +806,7 @@ static int readconfigfile(const char *fi
 				verify_format(config->group_signature, "GROUP_SIGNATURE");
 			}
 			else if (!strcmp(var, "LDAP_MAIL")) {
-				if(strcasecmp(value, "true") == 0) 
+				if(strncasecmp(value, "true", 4) == 0) 
 					config->use_ldap_mail = 1;
 				else
 					config->use_ldap_mail = 0;
