diff -urNp quota-tools-orig/quotaon.c quota-tools/quotaon.c
--- quota-tools-orig/quotaon.c	2008-03-13 15:49:33.000000000 +0100
+++ quota-tools/quotaon.c	2008-11-14 13:05:53.000000000 +0100
@@ -66,7 +66,7 @@ static void usage(void)
 {
 	errstr(_("Usage:\n\t%s [-guvp] [-F quotaformat] [-x state] -a\n\
 \t%s [-guvp] [-F quotaformat] [-x state] filesys ...\n\n\
--a, --all                turn quotas on for all filesystems\n\
+-a, --all                turn quotas %s for all filesystems\n\
 -f, --off                turn quotas off\n\
 -u, --user               operate on user quotas\n\
 -g, --group              operate on group quotas\n\
@@ -75,7 +75,8 @@ static void usage(void)
 -F, --format=formatname  operate on specific quota format\n\
 -v, --verbose            print more messages\n\
 -h, --help               display this help text and exit\n\
--V, --version            display version information and exit\n"), progname, progname);
+-V, --version            display version information and exit\n"),
+ progname, progname, strcmp (progname, "quotaon")?_("off"):_("on"));
 	exit(1);
 }
 
