diff -up quota-tools/Makefile.in.ftbfs quota-tools/Makefile.in
--- quota-tools/Makefile.in.ftbfs	2010-02-23 12:24:13.000000000 +0100
+++ quota-tools/Makefile.in	2010-02-23 12:25:55.000000000 +0100
@@ -125,7 +125,7 @@ quot: quot.o $(LIBOBJS)
 repquota: repquota.o $(LIBOBJS)
 
 warnquota: warnquota.o $(LIBOBJS)
-	$(CC) $(LDFLAGS) -o $@ $^ $(LDAPLIBS)
+	$(CC) $(LDFLAGS) -o $@ $^ $(LDAPLIBS) -llber
 
 quotastats: quotastats.o common.o pot.o
 
