---
 src/opie_comms.c |   18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

--- src/opie_comms.c.orig
+++ src/opie_comms.c
@@ -64,10 +64,10 @@ const char *OPIE_TODO_FILE     = "Applic
 const char *OPIE_CALENDAR_FILE = "Applications/datebook/datebook.xml";
 const char *OPIE_CATEGORY_FILE = "Settings/Categories.xml";
 
-int opie_curl_fwrite(void* buffer, size_t size, size_t nmemb, void* stream);
-int opie_curl_strwrite(void *buffer, size_t size, size_t nmemb, void *stream);
-int opie_curl_nullwrite(void *buffer, size_t size, size_t nmemb, void *stream);
-int opie_curl_strread(void *buffer, size_t size, size_t nmemb, void *stream);
+size_t opie_curl_fwrite(void* buffer, size_t size, size_t nmemb, void* stream);
+size_t opie_curl_strwrite(void *buffer, size_t size, size_t nmemb, void *stream);
+size_t opie_curl_nullwrite(void *buffer, size_t size, size_t nmemb, void *stream);
+size_t opie_curl_strread(void *buffer, size_t size, size_t nmemb, void *stream);
 gboolean ftp_fetch_files(OpieSyncEnv* env, GList* files_to_fetch);
 gboolean scp_fetch_files(OpieSyncEnv* env, GList* files_to_fetch);
 gboolean ftp_put_files(OpieSyncEnv* env, GList* files_to_put);
@@ -1146,7 +1146,7 @@ gboolean scp_put_files(OpieSyncEnv* env,
 /*
  * opie_curl_fwrite
  */
-int opie_curl_fwrite(void *buffer, size_t size, size_t nmemb, void *stream)
+size_t opie_curl_fwrite(void *buffer, size_t size, size_t nmemb, void *stream)
 {
 	size_t written = fwrite(buffer, size, nmemb, (FILE *)stream);
 	return written; 
@@ -1155,7 +1155,7 @@ int opie_curl_fwrite(void *buffer, size_
 /*
  * opie_curl_strwrite
  */
-int opie_curl_strwrite(void *buffer, size_t size, size_t nmemb, void *stream)
+size_t opie_curl_strwrite(void *buffer, size_t size, size_t nmemb, void *stream)
 {
 	g_string_append_len((GString *)stream, buffer, size * nmemb);
 	return size * nmemb;
@@ -1164,7 +1164,7 @@ int opie_curl_strwrite(void *buffer, siz
 /*
  * opie_curl_nullwrite
  */
-int opie_curl_nullwrite(void *buffer, size_t size, size_t nmemb, void *stream)
+size_t opie_curl_nullwrite(void *buffer, size_t size, size_t nmemb, void *stream)
 {
 	return size * nmemb;
 }
@@ -1172,13 +1172,13 @@ int opie_curl_nullwrite(void *buffer, si
 /*
  * opie_curl_strread
  */
-int opie_curl_strread(void *buffer, size_t size, size_t nmemb, void *stream)
+size_t opie_curl_strread(void *buffer, size_t size, size_t nmemb, void *stream)
 {
 	char *str = ((char *)stream) + m_totalwritten;
 	if(str[0] == '\0')
 		return 0;
 	
-	int numbytes = strlen(str);
+	size_t numbytes = strlen(str);
 	if(numbytes >= (nmemb * size))
 		numbytes = (nmemb * size);
 	memcpy(buffer, str, numbytes);
