diff -Nuar a/src/gnokii_contact.c b/src/gnokii_contact.c
--- a/src/gnokii_contact.c	2007-03-27 14:49:51.000000000 +0300
+++ b/src/gnokii_contact.c	2013-05-09 18:41:00.238120041 +0300
@@ -304,10 +304,6 @@
 
 	osync_trace(TRACE_ENTRY, "%s(%s, %p)", __func__, uid, state);
 
-//	int location = 0;
-//	char memory_type_string[3];
-	gn_error error = GN_ERR_NONE;
-
 	gn_phonebook_entry *contact = (gn_phonebook_entry *) malloc(sizeof(gn_phonebook_entry));
 	gn_data *data = (gn_data *) malloc(sizeof(gn_data));
 
@@ -328,7 +324,6 @@
 	osync_trace(TRACE_INTERNAL, "Try to delete entry with Memory Type: %i at Location: %i\n", 
 			contact->memory_type, contact->location);
 
-	error = gn_sm_functions(GN_OP_DeletePhonebook, data, state);
 	/* FIXME - don't check return code - return code is wrong - bug in libgnokii?
 	if (error != GN_ERR_NONE) {
 		osync_trace(TRACE_EXIT_ERROR, "%s(): Couldn't delete contact: %s\n", __func__, gn_error_print(error));
