diff -up gl2ps-1.3.5-source/CMakeLists.txt.r gl2ps-1.3.5-source/CMakeLists.txt
--- gl2ps-1.3.5-source/CMakeLists.txt.r	2009-10-16 21:14:07.000000000 +0200
+++ gl2ps-1.3.5-source/CMakeLists.txt	2009-12-26 17:23:38.000000000 +0100
@@ -116,6 +116,10 @@ if(OPENGL_FOUND)
   install(TARGETS lib shared DESTINATION lib)
 endif(OPENGL_FOUND)
 
+set_target_properties(shared PROPERTIES
+                      VERSION ${GL2PS_MAJOR_VERSION}.${GL2PS_MINOR_VERSION}.${GL2PS_PATCH_VERSION}
+                      SOVERSION ${GL2PS_MAJOR_VERSION}) 
+
 if(WIN32)
   set(GL2PS_DOC .)
 else(WIN32)
