Subject: Use the KDE configuration tool for Compiz instead of the Gtk one
From: Jaroslaw Zachwieja <grok@warwick.ac.uk>
Signed-off-by: Lubos Lunak
Patch-upstream: no (maybe later)
Bug: bnc#609396

Index: phonon/kded-module/phononserver.cpp
===================================================================
--- phonon/kded-module/phononserver.cpp.orig
+++ phonon/kded-module/phononserver.cpp
@@ -1087,8 +1087,17 @@ void PhononServer::askToRemoveDevices(co
     if (!areAudio && !areVideo)
         return;
 
-    const QString &dontAskAgainName = QLatin1String("phonon_forget_devices_") +
-        devList.join(QLatin1String("_"));
+    KSharedConfigPtr config = KGlobal::config();
+    KConfigGroup phononGroup = config->group("phonon");
+    bool alwaysForget = phononGroup.readEntry("phononAlwaysForget", false);
+    QString dontAskAgainName;
+    if (alwaysForget) {
+        dontAskAgainName = QLatin1String("phonon_forget_devices_always");
+        KMessageBox::saveDontShowAgainYesNo(dontAskAgainName, KMessageBox::Yes);
+    } else {
+        dontAskAgainName = QLatin1String("phonon_forget_devices_") +
+            devList.join(QLatin1String("_"));
+    }
 
     KMessageBox::ButtonCode result;
     if (!KMessageBox::shouldBeShownYesNo(dontAskAgainName, result)) {
