--- ttf2pt1-3.4.4/Makefile
+++ ttf2pt1-3.4.4/Makefile
@@ -9,7 +9,7 @@
 #
 # Use GNU C even if it's not the default compiler
 #
-#CC=gcc
+CC=gcc
 #
 # Use the standard ANSI C compiler on HP-UX even if it's not default
 #
@@ -20,11 +20,11 @@
 #
 # Default
 
-CFLAGS_SYS= -O
+# CFLAGS_SYS= -O
 
 # For GNU C
 #
-#CFLAGS_SYS= -O2
+CFLAGS_SYS= -O2 -Wall
 #
 # For GNU C with long options support library (Linux etc.)
 #
@@ -124,7 +124,7 @@
 # for binaries
 BINDIR = $(INSTDIR)/bin
 # for binaries of little general interest
-LIBXDIR = $(INSTDIR)/libexec/ttf2pt1
+LIBXDIR = $(INSTDIR)/lib/ttf2pt1
 # for scripts, maps/encodings etc.
 SHAREDIR = $(INSTDIR)/share/ttf2pt1
 MANDIR = $(INSTDIR)/man
@@ -135,11 +135,11 @@
 
 # After you have configured the Makefile, comment out the following
 # definition:
-warning: docs
-	@echo >&2
-	@echo "  You have to configure the Makefile before running make!" >&2
-	@echo "(or if you are lazy and hope that it will work as is run \`make all')">&2
-	@echo >&2
+#warning: docs
+#	@echo >&2
+#	@echo "  You have to configure the Makefile before running make!" >&2
+#	@echo "(or if you are lazy and hope that it will work as is run \`make all')">&2
+#	@echo >&2
 
 DOCS=CHANGES README FONTS FONTS.hpux encodings/README other/README \
 	app/X11/README app/netscape/README app/TeX/README
@@ -239,8 +239,8 @@
 	scripts/inst_dir $(MANDIR)/man1 $(OWNER) $(GROUP) 0755
 	scripts/inst_dir $(MANDIR)/man5 $(OWNER) $(GROUP) 0755
 	cp -R $(TXTFILES) $(SUBDIRS) $(SHAREDIR)
-	chown -R $(OWNER) $(SHAREDIR)
-	chgrp -R $(GROUP) $(SHAREDIR)
+#	chown -R $(OWNER) $(SHAREDIR)
+#	chgrp -R $(GROUP) $(SHAREDIR)
 	chmod -R go-w $(SHAREDIR)
 	scripts/inst_file ttf2pt1 $(BINDIR)/ttf2pt1 $(OWNER) $(GROUP) 0755
 	[ -f $(BINDIR)/t1asm ] || scripts/inst_file t1asm $(LIBXDIR)/t1asm $(OWNER) $(GROUP) 0755
@@ -255,8 +255,6 @@
 		sed 's|TTF2PT1_BINDIR|$(BINDIR)|;\
 			s|TTF2PT1_LIBXDIR|$(LIBXDIR)|;\
 			s|TTF2PT1_SHAREDIR|$(SHAREDIR)|;' <$$i >$(MANDIR)/man1/$$i \
-		&& chown $(OWNER) $(MANDIR)/man1/$$i \
-		&& chgrp $(GROUP) $(MANDIR)/man1/$$i \
 		&& chmod 0644 $(MANDIR)/man1/$$i \
 		|| exit 1; \
 	} done
--- ttf2pt1-3.4.4/app/netscape/Makefile
+++ ttf2pt1-3.4.4/app/netscape/Makefile
@@ -1,5 +1,5 @@
 nsfix: nsfix.c
-	$(CC) $(CFLAGS) -o nsfix nsfix.c $(LDFLAGS)
+	$(CC) $(CFLAGS) -O2 -s -o nsfix nsfix.c $(LDFLAGS)
 
 clean:
 	rm -f *.o core *.core core.* nsfix
--- ttf2pt1-3.4.4/scripts/convert
+++ ttf2pt1-3.4.4/scripts/convert
@@ -36,14 +36,14 @@
 then
 	CFGFILE=$1
 else
-	CFGFILE=`pwd`/convert.cfg
+	CFGFILE=/etc/ttf2pt1/convert.cfg
 fi
 
 # these setting would be edited during installation
 
-TTF2PT1_BINDIR=
-TTF2PT1_LIBXDIR=
-TTF2PT1_SHAREDIR=
+TTF2PT1_BINDIR=/usr/bin/
+TTF2PT1_LIBXDIR=/usr/lib/ttf2pt1
+TTF2PT1_SHAREDIR=/usr/share/ttf2pt1/
 
 [ -z "$TTF2PT1_BINDIR" ] && {
 	TTF2PT1_BINDIR=`pwd`/..
--- ttf2pt1-3.4.4/scripts/convert.cfg.sample
+++ ttf2pt1-3.4.4/scripts/convert.cfg.sample
@@ -75,26 +75,28 @@
 # unless you really need it, it's very incomplete.
 #
 # The map planes are not supported in the scripts yet.
-
+# 
+# SRCDIRS="
+# /SOME_DIR_WITH_cyrillic_windows_TTF_FONTS cyrillic ibm-1251
+# /SOME_DIR_WITH_cyrillic_koi_TTF_FONTS cyrillic koi8-r
+# /SOME_DIR_WITH_unicode_TTF_FONTS latin4 iso8859-4
+# /SOME_DIR_WITH_unicode_TTF_FONTS latin5 iso8859-9
+# /SOME_DIR_WITH_unicode_TTF_FONTS latin2 iso8859-2
+# /SOME_DIR_WITH_western_TTF_FONTS latin1 iso8859-1
+# /SOME_DIR_WITH_unicode_TTF_FONTS adobestd adobe-std
+# /SOME_DIR_WITH_weird_unicode_TTF_FONTS adobestd adobe-std adobe-standard-encoding.map
+# /SOME_DIR_WITH_unicode_TTF_FONTS unknown my-special some-very-special.map
+# "
 SRCDIRS="
-/SOME_DIR_WITH_cyrillic_windows_TTF_FONTS cyrillic ibm-1251
-/SOME_DIR_WITH_cyrillic_koi_TTF_FONTS cyrillic koi8-r
-/SOME_DIR_WITH_unicode_TTF_FONTS latin4 iso8859-4
-/SOME_DIR_WITH_unicode_TTF_FONTS latin5 iso8859-9
-/SOME_DIR_WITH_unicode_TTF_FONTS latin2 iso8859-2
-/SOME_DIR_WITH_western_TTF_FONTS latin1 iso8859-1
-/SOME_DIR_WITH_unicode_TTF_FONTS adobestd adobe-std
-/SOME_DIR_WITH_weird_unicode_TTF_FONTS adobestd adobe-std adobe-standard-encoding.map
-/SOME_DIR_WITH_unicode_TTF_FONTS unknown my-special some-very-special.map
+/win/C/windows/fonts latin1 iso8859-1
 "
-
 # the directory for converted X11 fonts
 DSTDIR=/usr/X11R6/lib/X11/fonts/fromttf
 
 # The base directory of Ghostscript;
 # set it to empty space if you don't want the Ghostscript fonts installed. 
 # For some systems the directory is /usr/share/ghostscript.
-GSDIR=/usr/local/share/ghostscript
+GSDIR=/usr/share/ghostscript
 
 # The font directory of Ghostscript;
 # should work for all the versions of Ghostscript. Except (as always) the one 
@@ -108,7 +110,7 @@
 # installation stores the Fontmap files in the same directory as fonts
 # (like Debian Linux does - isn't Linux wonderful in its differences?)
 # then set it to the same value as GSFONTDIR.
-GSCONFDIR=$GSDIR/6.0
+GSCONFDIR=$GSDIR/6.51
 
 # The encodings of generated files by languages
 # (see the whole list of possible encodings for each
@@ -183,9 +185,9 @@
 CORRECTWIDTH=YES
 REMOVET1A=YES
 INSTALLFONTMAP=YES
-HINTSUBST=NO
+HINTSUBST=YES
 ENFORCEISO=NO
-ALLGLYPHS=NO
+ALLGLYPHS=YES
 GENUID=NO
 CREATEPFB=YES
 
--- ttf2pt1-3.4.4/scripts/inst_dir
+++ ttf2pt1-3.4.4/scripts/inst_dir
@@ -30,10 +30,10 @@
 	if( ! -d "$prefix" ) {
 		die "Unable to create directory $prefix:\n$!\n"
 			unless mkdir($prefix, $mode);
-		die "Unable to change owner of $prefix to $owner\n"
-			if system("chown $owner $prefix");
-		die "Unable to change group of $prefix to $group\n"
-			if system("chgrp $group $prefix");
+#		die "Unable to change owner of $prefix to $owner\n"
+#			if system("chown $owner $prefix");
+#		die "Unable to change group of $prefix to $group\n"
+#			if system("chgrp $group $prefix");
 	}
 	if($#sl < 0) {
 		last;
--- ttf2pt1-3.4.4/scripts/inst_file
+++ ttf2pt1-3.4.4/scripts/inst_file
@@ -12,6 +12,4 @@
 }
 
 cp -f $1 $2 \
-&& chown $3 $2 \
-&& chgrp $4 $2 \
 && chmod 0$5 $2
