--- ttf2pt1-3.4.4/Makefile
+++ ttf2pt1-3.4.4/Makefile
@@ -60,7 +60,7 @@
 # (if the include and lib directory do not match your installation,
 # modify them), also uncomment LIBS_FT
 #
-#CFLAGS_FT = -DUSE_FREETYPE -I/usr/local/include/freetype2 -I/usr/local/include
+CFLAGS_FT = -DUSE_FREETYPE -I/usr/include/freetype2 -I/usr/local/include
 
 # 
 # The FreeType-2 library flags (disabled by default)
@@ -71,7 +71,7 @@
 # (if the include and lib directory do not match your installation,
 # modify them), also uncomment CFLAGS_FT
 #
-#LIBS_FT= -L/usr/local/lib -lfreetype
+LIBS_FT= -L/usr/lib -lfreetype
 
 #
 # The flags for C compiler for the Autotrace library (disabled by default). 
