Index: qtbindings/qtbindingsbase.pri
===================================================================
--- qtbindings/qtbindingsbase.pri.orig
+++ qtbindings/qtbindingsbase.pri
@@ -6,3 +6,4 @@ QT += script
 CONFIG += debug_and_release
 GENERATEDCPP = $$PWD/../generated_cpp
 TARGET=$$qtLibraryTarget($$TARGET)
+target.path=$$[QT_INSTALL_PLUGINS]/script
Index: tools/qsexec/src/qsexec.pro
===================================================================
--- tools/qsexec/src/qsexec.pro.orig
+++ tools/qsexec/src/qsexec.pro
@@ -9,3 +9,6 @@ win32:CONFIG+=console
 mac:CONFIG-=app_bundle
 
 SOURCES += main.cpp
+
+target.path=$$[QT_INSTALL_BINS]
+
