Index: ppp-2.4.4/linux/Makefile.top
===================================================================
--- ppp-2.4.4.orig/linux/Makefile.top
+++ ppp-2.4.4/linux/Makefile.top
@@ -3,10 +3,10 @@
 DESTDIR = $(INSTROOT)@DESTDIR@
 BINDIR = $(DESTDIR)/sbin
 INCDIR = $(DESTDIR)/include
-MANDIR = $(DESTDIR)/share/man
-ETCDIR = $(INSTROOT)@SYSCONF@/ppp
-RUNDIR = $(DESTDIR)/var/run/ppp
-LOGDIR = $(DESTDIR)/var/log/ppp
+MANDIR = $(DESTDIR)//share/man
+ETCDIR = $(DESTDIR)/../@SYSCONF@/ppp
+RUNDIR = $(DESTDIR)/../var/run/ppp
+LOGDIR = $(DESTDIR)/../var/log/ppp
 
 # uid 0 = root
 INSTALL= install
