diff -up ppp-2.4.4/pppd/plugins/pppoatm/Makefile.linux.atm-make ppp-2.4.4/pppd/plugins/pppoatm/Makefile.linux
--- ppp-2.4.4/pppd/plugins/pppoatm/Makefile.linux.atm-make	2008-08-28 17:36:10.000000000 -0400
+++ ppp-2.4.4/pppd/plugins/pppoatm/Makefile.linux	2008-08-28 17:36:56.000000000 -0400
@@ -1,5 +1,5 @@
 #CC	= gcc
-COPTS	= -O2 -g
+COPTS	= $(RPM_OPT_FLAGS)
 CFLAGS	= $(COPTS) -I../.. -I../../../include -fPIC
 LDFLAGS	= -shared
 INSTALL	= install
