--- ppp-2.4.4/pppd/plugins/rp-pppoe/Makefile.linux.lib64	2006-07-24 20:43:40.000000000 +0200
+++ ppp-2.4.4/pppd/plugins/rp-pppoe/Makefile.linux	2006-07-24 20:44:56.000000000 +0200
@@ -15,7 +15,7 @@
 
 DESTDIR = $(INSTROOT)@DESTDIR@
 BINDIR = $(DESTDIR)/sbin
-LIBDIR = $(DESTDIR)/lib/pppd/$(PPPDVERSION)
+LIBDIR = $(DESTDIR)/lib/$(shell gcc -print-multi-os-directory 2> /dev/null)/pppd/$(PPPDVERSION)
 
 PPPDVERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../../patchlevel.h)
 
--- ppp-2.4.4/pppd/plugins/radius/Makefile.linux.lib64	2006-07-24 20:43:40.000000000 +0200
+++ ppp-2.4.4/pppd/plugins/radius/Makefile.linux	2006-07-24 20:44:45.000000000 +0200
@@ -5,7 +5,7 @@
 
 DESTDIR = $(INSTROOT)@DESTDIR@
 MANDIR = $(DESTDIR)/share/man/man8
-LIBDIR = $(DESTDIR)/lib/pppd/$(VERSION)
+LIBDIR = $(DESTDIR)/lib/$(shell gcc -print-multi-os-directory 2> /dev/null)/pppd/$(VERSION)
 
 VERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../../patchlevel.h)
 
--- ppp-2.4.4/pppd/plugins/pppoatm/Makefile.linux.lib64	2004-11-14 02:12:10.000000000 +0100
+++ ppp-2.4.4/pppd/plugins/pppoatm/Makefile.linux	2006-07-24 20:44:29.000000000 +0200
@@ -6,8 +6,8 @@
 
 #***********************************************************************
 
-DESTDIR = $(INSTROOT)@DESTDIR@
-LIBDIR = $(DESTDIR)/lib/pppd/$(VERSION)
+DESTDIR = $(INSTROOT)@DESTDIR@
+LIBDIR = $(DESTDIR)/lib/$(shell gcc -print-multi-os-directory 2> /dev/null)/pppd/$(VERSION)
 
 VERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../../patchlevel.h)
 
--- ppp-2.4.4/pppd/plugins/Makefile.linux.lib64	2006-07-24 20:43:40.000000000 +0200
+++ ppp-2.4.4/pppd/plugins/Makefile.linux	2006-07-24 20:43:40.000000000 +0200
@@ -7,7 +7,7 @@
 DESTDIR = $(INSTROOT)@DESTDIR@
 BINDIR = $(DESTDIR)/sbin
 MANDIR = $(DESTDIR)/share/man/man8
-LIBDIR = $(DESTDIR)/lib/pppd/$(VERSION)
+LIBDIR = $(DESTDIR)/lib/$(shell $(CC) -print-multi-os-directory 2> /dev/null)/pppd/$(VERSION)
 
 SUBDIRS := rp-pppoe pppoatm pppol2tp
 # Uncomment the next line to include the radius authentication plugin
--- ppp-2.4.4/pppd/Makefile.linux.lib64	2006-07-24 20:43:40.000000000 +0200
+++ ppp-2.4.4/pppd/Makefile.linux	2006-07-24 20:43:40.000000000 +0200
@@ -8,6 +8,7 @@
 BINDIR = $(DESTDIR)/sbin
 MANDIR = $(DESTDIR)/share/man/man8
 INCDIR = $(DESTDIR)/include
+LIBDIR = $(DESTDIR)/lib/$(shell gcc -print-multi-os-directory 2> /dev/null)
 
 TARGETS = pppd
 
@@ -32,7 +33,7 @@
 
 CC = gcc
 #
-COPTS = -Wall $(RPM_OPT_FLAGS)
+COPTS = -Wall $(RPM_OPT_FLAGS) -DLIBDIR=\""$(LIBDIR)"\"
 LIBS = -lutil
 
 # Uncomment the next 2 lines to include support for Microsoft's
--- ppp-2.4.4/pppd/pathnames.h.lib64	2005-08-26 01:59:34.000000000 +0200
+++ ppp-2.4.4/pppd/pathnames.h	2006-07-24 20:43:40.000000000 +0200
@@ -57,7 +57,7 @@
 
 #ifdef PLUGIN
 #ifdef __STDC__
-#define _PATH_PLUGIN	DESTDIR "/lib/pppd/" VERSION
+#define _PATH_PLUGIN	LIBDIR "/pppd/" VERSION
 #else /* __STDC__ */
 #define _PATH_PLUGIN	"/usr/lib/pppd"
 #endif /* __STDC__ */
--- ppp-2.4.5/pppd/plugins/pppol2tp/Makefile.linux.lib64	2009-11-16 23:26:07.000000000 +0100
+++ ppp-2.4.5/pppd/plugins/pppol2tp/Makefile.linux	2010-08-05 16:33:53.964898629 +0200
@@ -6,8 +6,8 @@
 
 #***********************************************************************
 
-DESTDIR = @DESTDIR@
-LIBDIR = $(DESTDIR)/lib/pppd/$(VERSION)
+DESTDIR = $(INSTROOT)@DESTDIR@
+LIBDIR = $(DESTDIR)/lib/$(shell gcc -print-multi-os-directory 2> /dev/null)/pppd/$(VERSION)
 
 VERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../../patchlevel.h)
 
