diff -up ppp-2.4.4/chat/chat.8.dontwriteetc ppp-2.4.4/chat/chat.8
--- ppp-2.4.4/chat/chat.8.dontwriteetc	2004-11-13 13:22:49.000000000 +0100
+++ ppp-2.4.4/chat/chat.8	2008-03-03 15:37:55.000000000 +0100
@@ -200,7 +200,7 @@ The \fBSAY\fR directive allows the scrip
 at the terminal via standard error.  If \fBchat\fR is being run by
 pppd, and pppd is running as a daemon (detached from its controlling
 terminal), standard error will normally be redirected to the file
-/etc/ppp/connect\-errors.
+/var/log/ppp/connect\-errors.
 .LP
 \fBSAY\fR strings must be enclosed in single or double quotes. If
 carriage return and line feed are needed in the string to be output,
diff -up ppp-2.4.4/pppd/pathnames.h.dontwriteetc ppp-2.4.4/pppd/pathnames.h
--- ppp-2.4.4/pppd/pathnames.h.dontwriteetc	2008-03-03 15:26:51.000000000 +0100
+++ ppp-2.4.4/pppd/pathnames.h	2008-03-03 15:26:51.000000000 +0100
@@ -28,9 +28,9 @@
 #define _PATH_AUTHUP	 _ROOT_PATH "/etc/ppp/auth-up"
 #define _PATH_AUTHDOWN	 _ROOT_PATH "/etc/ppp/auth-down"
 #define _PATH_TTYOPT	 _ROOT_PATH "/etc/ppp/options."
-#define _PATH_CONNERRS	 _ROOT_PATH "/etc/ppp/connect-errors"
+#define _PATH_CONNERRS	 _ROOT_PATH "/var/log/ppp/connect-errors"
 #define _PATH_PEERFILES	 _ROOT_PATH "/etc/ppp/peers/"
-#define _PATH_RESOLV	 _ROOT_PATH "/etc/ppp/resolv.conf"
+#define _PATH_RESOLV	 _ROOT_PATH "/var/run/ppp/resolv.conf"
 
 #define _PATH_USEROPT	 ".ppprc"
 #define	_PATH_PSEUDONYM	 ".ppp_pseudonym"
diff -up ppp-2.4.4/linux/Makefile.top.dontwriteetc ppp-2.4.4/linux/Makefile.top
--- ppp-2.4.4/linux/Makefile.top.dontwriteetc	2006-06-04 07:04:14.000000000 +0200
+++ ppp-2.4.4/linux/Makefile.top	2008-03-03 15:42:10.000000000 +0100
@@ -5,6 +5,8 @@ BINDIR = $(DESTDIR)/sbin
 INCDIR = $(DESTDIR)/include
 MANDIR = $(DESTDIR)/share/man
 ETCDIR = $(INSTROOT)@SYSCONF@/ppp
+RUNDIR = $(DESTDIR)/../var/run/ppp
+LOGDIR = $(DESTDIR)/../var/log/ppp
 
 # uid 0 = root
 INSTALL= install
@@ -16,7 +18,7 @@ all:
 	cd pppstats; $(MAKE) $(MFLAGS) all
 	cd pppdump; $(MAKE) $(MFLAGS) all
 
-install: $(BINDIR) $(MANDIR)/man8 install-progs install-devel
+install: $(BINDIR) $(RUNDIR) $(LOGDIR) $(MANDIR)/man8 install-progs install-devel
 
 install-progs:
 	cd chat; $(MAKE) $(MFLAGS) install
@@ -44,6 +46,10 @@ $(MANDIR)/man8:
 	$(INSTALL) -d -m 755 $@
 $(ETCDIR):
 	$(INSTALL) -d -m 755 $@
+$(RUNDIR):
+	$(INSTALL) -d -m 755 $@
+$(LOGDIR):
+	$(INSTALL) -d -m 755 $@
 
 clean:
 	rm -f `find . -name '*.[oas]' -print`
