--- ppp-2.4.3/pppd/Makefile.linux.make	2004-11-13 13:02:22.000000000 +0100
+++ ppp-2.4.3/pppd/Makefile.linux	2004-11-22 15:46:49.871309478 +0100
@@ -30,10 +30,10 @@
 include .depend
 endif
 
-# CC = gcc
+CC = gcc
 #
-COPTS = -O2 -pipe -Wall -g
-LIBS =
+COPTS = -Wall $(RPM_OPT_FLAGS)
+LIBS = -lutil
 
 # Uncomment the next 2 lines to include support for Microsoft's
 # MS-CHAP authentication protocol.  Also, edit plugins/radius/Makefile.linux.
@@ -61,8 +61,8 @@
 USE_TDB=y
 
 HAS_SHADOW=y
-#USE_PAM=y
-#HAVE_INET6=y
+USE_PAM=y
+HAVE_INET6=y
 
 # Enable plugins
 PLUGIN=y
--- ppp-2.4.3/pppd/plugins/Makefile.linux.make	2004-11-14 08:57:35.000000000 +0100
+++ ppp-2.4.3/pppd/plugins/Makefile.linux	2004-11-22 15:46:49.873309203 +0100
@@ -1,5 +1,5 @@
 #CC	= gcc
-COPTS	= -O2 -g
+COPTS	= $(RPM_OPT_FLAGS)
 CFLAGS	= $(COPTS) -I.. -I../../include -fPIC
 LDFLAGS	= -shared
 INSTALL	= install
--- ppp-2.4.3/pppd/plugins/radius/Makefile.linux.make	2004-11-14 08:02:31.000000000 +0100
+++ ppp-2.4.3/pppd/plugins/radius/Makefile.linux	2004-11-22 15:49:08.858233692 +0100
@@ -12,7 +12,7 @@
 INSTALL	= install
 
 PLUGIN=radius.so radattr.so radrealms.so
-CFLAGS=-I. -I../.. -I../../../include -O2 -fPIC -DRC_LOG_FACILITY=LOG_DAEMON
+CFLAGS=-I. -I../.. -I../../../include $(RPM_OPT_FLAGS) -DRC_LOG_FACILITY=LOG_DAEMON
 
 # Uncomment the next line to include support for Microsoft's
 # MS-CHAP authentication protocol.
--- ppp-2.4.3/pppd/plugins/rp-pppoe/Makefile.linux.make	2004-11-14 08:58:37.000000000 +0100
+++ ppp-2.4.3/pppd/plugins/rp-pppoe/Makefile.linux	2004-11-22 15:46:49.875308929 +0100
@@ -25,7 +25,7 @@
 # Version is set ONLY IN THE MAKEFILE!  Don't delete this!
 RP_VERSION=3.8p
 
-COPTS=-O2 -g
+COPTS=$(RPM_OPT_FLAGS)
 CFLAGS=$(COPTS) -I../../../include '-DRP_VERSION="$(RP_VERSION)"'
 all: rp-pppoe.so pppoe-discovery
 
--- ppp-2.4.3/pppdump/Makefile.linux.make	2004-10-31 02:36:52.000000000 +0200
+++ ppp-2.4.3/pppdump/Makefile.linux	2004-11-22 15:48:16.795379237 +0100
@@ -2,7 +2,7 @@
 BINDIR = $(DESTDIR)/sbin
 MANDIR = $(DESTDIR)/share/man/man8
 
-CFLAGS= -O -I../include/net
+CFLAGS= $(RPM_OPT_FLAGS) -I../include/net
 OBJS = pppdump.o bsd-comp.o deflate.o zlib.o
 
 INSTALL= install
--- ppp-2.4.3/pppstats/Makefile.linux.make	2004-10-31 23:09:03.000000000 +0100
+++ ppp-2.4.3/pppstats/Makefile.linux	2004-11-22 15:46:49.872309341 +0100
@@ -10,7 +10,7 @@
 PPPSTATOBJS = pppstats.o
 
 #CC = gcc
-COPTS = -O
+COPTS = $(RPM_OPT_FLAGS)
 COMPILE_FLAGS = -I../include
 LIBS =
 
--- ppp-2.4.3/chat/Makefile.linux.make	2004-11-03 12:51:47.000000000 +0100
+++ ppp-2.4.3/chat/Makefile.linux	2004-11-22 15:47:59.445760450 +0100
@@ -10,7 +10,7 @@
 CDEF4=	-DFNDELAY=O_NDELAY		# Old name value
 CDEFS=	$(CDEF1) $(CDEF2) $(CDEF3) $(CDEF4)
 
-COPTS=	-O2 -g -pipe
+COPTS=	$(RPM_OPT_FLAGS)
 CFLAGS=	$(COPTS) $(CDEFS)
 
 INSTALL= install
