diff -Naur subtitleeditor-0.22.3/share/subtitleeditor.desktop.in subtitleeditor-0.22.3_new/share/subtitleeditor.desktop.in
--- share/subtitleeditor.desktop.in	2008-08-13 16:26:30.000000000 +0300
+++ share/subtitleeditor.desktop.in	2008-08-30 13:52:20.000000000 +0300
@@ -2,7 +2,9 @@
 Encoding=UTF-8
 _Name=Subtitle Editor
 GenericName=Subtitle Editor
+GenericName[tr]=Altyazı Düzenleyici
 _Comment=A subtitle editor based on GStreamer and Gtk+
+Comment[tr]=Altyazı dosyalarını düzenlemeye yarayan GStreamer ve Gtk+ tabanlı uygulama
 Exec=subtitleeditor %F
 Terminal=false
 Type=Application
