diff -Nuar reportlab-2.5.orig//src/rl_addons/renderPM/gt1/gt1-namecontext.c reportlab-2.5/src/rl_addons/renderPM/gt1/gt1-namecontext.c
--- reportlab-2.5.orig//src/rl_addons/renderPM/gt1/gt1-namecontext.c	2001-05-03 18:44:27.000000000 +0300
+++ reportlab-2.5/src/rl_addons/renderPM/gt1/gt1-namecontext.c	2010-11-09 14:58:53.710999613 +0200
@@ -3,9 +3,8 @@
 #include "gt1-misc.h"
 
 #include "gt1-namecontext.h"
-#if defined(_WIN32) || defined(macintosh)
-#	include <string.h>
-#endif
+
+# include <string.h>
 
 /* btw, I do not know who wrote the following comment. I modified this
    file somewhat from gimp's app/procedural_db.c hash function. */
