diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur yakuake-2.9.4-orig/app/config/yakuake.kcfg yakuake-2.9.4/app/config/yakuake.kcfg
--- yakuake-2.9.4-orig/app/config/yakuake.kcfg	2009-02-13 12:40:02.000000000 +0200
+++ yakuake-2.9.4/app/config/yakuake.kcfg	2009-02-13 12:40:23.000000000 +0200
@@ -20,7 +20,7 @@
     <entry name="Height" type="Int">
       <label context="@label">Window height</label>
       <whatsthis context="@info:whatsthis">Height of the application window as a percentage of the screen.</whatsthis>
-      <default>50</default>
+      <default>90</default>
       <min>10</min>
       <max>100</max>
     </entry>
