Index: yakuake-2.9.7/app/mainwindow.cpp
===================================================================
--- yakuake-2.9.7.orig/app/mainwindow.cpp
+++ yakuake-2.9.7/app/mainwindow.cpp
@@ -1016,7 +1016,7 @@ void MainWindow::showStartupPopup()
     QString message(i18nc("@info", "Application successfully started.<nl/>" "Press <shortcut>%1</shortcut> to use it ...", shortcut));
 
 #if KDE_IS_VERSION(4,3,95)
-    KNotification::event(KNotification::Notification, title, message,
+    KNotification::event("startup", title, message,
         KIconLoader::global()->loadIcon("yakuake", KIconLoader::Desktop));
 #else
     KPassivePopup* popup = new KPassivePopup();
