Index: kbd-1.15/src/unicode_start
===================================================================
--- kbd-1.15.orig/src/unicode_start
+++ kbd-1.15/src/unicode_start
@@ -72,6 +72,8 @@ stty iutf8
 # have a Unicode map attached, or explicitly specified, e.g.,
 # by giving `def.uni' as a second argument.
 
+DEFAULT_UNICODE_FONT=iso01.16
+
 case "$#" in
 	2)
 		setfont "$1" -u "$2"
@@ -80,6 +82,7 @@ case "$#" in
 		setfont "$1"
 		;;
 	0)
+		setfont $DEFAULT_UNICODE_FONT
 		;;
 	*)
 		echo "usage: unicode_start [font [unicode map]]"
