diff -Nrbu xawtv-3.95/configure.ac xawtv-3.95-OK/configure.ac
--- xawtv-3.95/configure.ac	2004-08-18 19:48:09.000000000 +0400
+++ xawtv-3.95-OK/configure.ac	2006-12-27 19:01:29.000000000 +0300
@@ -110,7 +110,10 @@
   AC_CHECK_LIB(c_r,pthread_create,LIBPTHREAD="-lc_r")
 fi
 AC_CHECK_LIB(ossaudio,main,LIBOSS="-lossaudio")
-AC_CHECK_LIB(ncurses,initscr,LIBCURSES="-lncurses")
+AC_CHECK_LIB(ncursesw,initscr,LIBCURSES="-lncursesw")
+if test "$LIBCURSES" = ""; then
+  AC_CHECK_LIB(ncurses,initscr,LIBCURSES="-lncurses")
+fi
 if test "$LIBCURSES" = ""; then
   AC_CHECK_LIB(curses,initscr,LIBCURSES="-lcurses")
 fi
@@ -120,6 +123,9 @@
   echo "      you need also the *-devel packages."
   exit 1
 fi
+if test "$LIBCURSES" = "-lncursesw"; then
+  CFLAGS="$CFLAGS -I/usr/include/ncursesw"
+fi
 AC_SUBST(LIBPTHREAD)
 AC_SUBST(LIBOSS)
 AC_SUBST(LIBCURSES)
diff -Nrbu xawtv-3.95/console/radio.c xawtv-3.95-OK/console/radio.c
--- xawtv-3.95/console/radio.c	2003-04-16 19:35:13.000000000 +0400
+++ xawtv-3.95-OK/console/radio.c	2006-12-27 19:13:06.000000000 +0300
@@ -23,6 +23,7 @@
 #include <string.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <locale.h>
 #include <curses.h>
 #include <sys/time.h>
 #include <sys/ioctl.h>
@@ -398,6 +399,8 @@
     int    quit=0, scan=0, arg_mute=0;
     struct video_tuner tuner;
 
+    setlocale(LC_ALL,"");
+
     /* parse args */
     for (;;) {
 	c = getopt(argc, argv, "mhiqdsSf:c:");
diff -Nrbu xawtv-3.95/console/record.c xawtv-3.95-OK/console/record.c
--- xawtv-3.95/console/record.c	2003-04-03 14:51:00.000000000 +0400
+++ xawtv-3.95-OK/console/record.c	2006-12-27 19:13:18.000000000 +0300
@@ -6,6 +6,7 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <errno.h>
+#include <locale.h>
 #include <curses.h>
 #include <signal.h>
 #include <inttypes.h>
@@ -574,6 +575,8 @@
     record = 0;
     nr = 0;
 
+    setlocale(LC_ALL,"");
+
     /* parse options */
     for (;;) {
 	if (-1 == (c = getopt(argc, argv, "vhlci:o:d:m:r:t:s:L:p:n:")))
