diff -ur xawtv-3.94.orig/x11/xawtv.c xawtv-3.94/x11/xawtv.c
--- xawtv-3.94.orig/x11/xawtv.c	2003-04-11 12:42:04.000000000 +0300
+++ xawtv-3.94/x11/xawtv.c	2005-04-29 12:59:51.000000000 +0300
@@ -26,6 +26,8 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 
+#include <locale.h>
+
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 #include <X11/Xproto.h>
@@ -1682,7 +1684,8 @@
     wm_detect(dpy);
     if (debug)
 	fprintf(stderr,"main: creating windows ...\n");
-    create_optwin();
+    setlocale(LC_ALL,"C");
+	create_optwin();
     create_onscreen(labelWidgetClass);
     create_vtx();
     create_chanwin();
@@ -1794,6 +1797,8 @@
     if (!have_config)
 	XtCallActionProc(tv,"Help",NULL,NULL,0);
 
+	setlocale(LC_ALL,"");
+	
     xt_main_loop();
     return 0;
 }
