diff -Nrbu xawtv-3.95/vbistuff/alevt.css.in xawtv-3.95-OK/vbistuff/alevt.css.in
--- xawtv-3.95/vbistuff/alevt.css.in	2003-02-14 17:14:08.000000000 +0300
+++ xawtv-3.95-OK/vbistuff/alevt.css.in	2007-03-02 19:38:26.000000000 +0300
@@ -3,7 +3,7 @@
 pre.vt		{ border: 1pt solid black; padding: 12pt }
 pre.vt		{ background-color: #c0c0c0; width: auto; float: left }
 pre.vt		{ white-space: pre }
-pre.vt		{ font-family: "teletext, monospace"; font-size: 20px; line-height: 20px }
+pre.vt		{ font-family: "teletext"; font-size: 10px; line-height: 10px }
 
 div.quick	{ clear: both }
 
diff -Nrbu xawtv-3.95/x11/Xawtv.ad xawtv-3.95-OK/x11/Xawtv.ad
--- xawtv-3.95/x11/Xawtv.ad	2003-02-18 19:31:33.000000000 +0300
+++ xawtv-3.95-OK/x11/Xawtv.ad	2007-03-02 19:37:26.000000000 +0300
@@ -382,9 +382,7 @@
 xawtv.onscreen*borderColor:		black
 xawtv.onscreen*foreground:		lightgreen
 xawtv.onscreen.label.justify:		left
-xawtv.onscreen.label.font:		-*-ledfixed-medium-r-*--39-*-*-*-c-*-*-*
 xawtv.onscreen.label.fontSet:		\
-     -*-ledfixed-medium-r-semicondensed--39-*-*-*-c-*-*-*, \
  -*-bitstream vera sans-medium-r-normal--39-*-*-*-*-*-*-*, \
            -*-luxi sans-medium-r-normal--39-*-*-*-*-*-*-*, \
                           -*-*-r-normal--39-*-*-*-*-*-*-*, \
