diff -Nrbu xawtv-3.95/common/capture.c xawtv-3.95-OK/common/capture.c
--- xawtv-3.95/common/capture.c	2004-10-12 14:16:34.000000000 +0400
+++ xawtv-3.95-OK/common/capture.c	2006-12-27 19:55:59.000000000 +0300
@@ -527,12 +527,12 @@
 
     if (h->vfmt.fmtid != VIDEO_NONE && h->afmt.fmtid != AUDIO_NONE) {
 	for (frames = 0; frames < 16; frames++) {
-	    stopby = (uint64_t)(h->frames + frames) * (uint64_t)1000000000000 / h->fps;
+	    stopby = (uint64_t)(h->frames + frames) * (uint64_t)1000000000000ULL / h->fps;
 	    if (stopby > h->ats)
 		break;
 	}
 	frames++;
-	h->stopby = (uint64_t)(h->frames + frames) * (uint64_t)1000000000000 / h->fps;
+	h->stopby = (uint64_t)(h->frames + frames) * (uint64_t)1000000000000ULL / h->fps;
 	while (frames) {
 	    movie_grab_put_video(h,NULL);
 	    frames--;
@@ -594,11 +594,11 @@
 	    (h->rdrift > 0) ? '+' : '-',
 	    (int)((abs(h->rdrift) / 1000000000)),
 	    (int)((abs(h->rdrift) % 1000000000) / 10000000),
-	    (int)(h->rdrift * h->fps / (uint64_t)1000000000000),
+	    (int)(h->rdrift * h->fps / (uint64_t)1000000000000ULL),
 	    (h->vdrift > 0) ? '+' : '-',
 	    (int)((abs(h->vdrift) / 1000000000)),
 	    (int)((abs(h->vdrift) % 1000000000) / 10000000),
-	    (int)(h->vdrift * h->fps / (uint64_t)1000000000000));
+	    (int)(h->vdrift * h->fps / (uint64_t)1000000000000ULL));
     rec_status(line);
 }
 
@@ -623,7 +623,7 @@
 #endif
 
     /* rate control */
-    expected = (buf->info.ts - h->vdrift) * h->fps / (uint64_t)1000000000000;
+    expected = (buf->info.ts - h->vdrift) * h->fps / (uint64_t)1000000000000ULL;
     if (expected < h->frames-1) {
 	if (debug > 1)
 	    fprintf(stderr,"rate: ignoring frame [%d %d]\n",
diff -Nrbu xawtv-3.95/console/showqt.c xawtv-3.95-OK/console/showqt.c
--- xawtv-3.95/console/showqt.c	2003-03-27 13:37:08.000000000 +0300
+++ xawtv-3.95-OK/console/showqt.c	2006-12-27 19:52:51.000000000 +0300
@@ -25,14 +25,14 @@
                    (((uint32_t)x<<8)  & (uint32_t)0x00ff0000) |\
                    (((uint32_t)x<<24) & (uint32_t)0xff000000))
 
-# define SWAP8(x) ((((uint64_t)x>>56) & (uint64_t)0x00000000000000ff) |\
-                   (((uint64_t)x>>40) & (uint64_t)0x000000000000ff00) |\
-                   (((uint64_t)x>>24) & (uint64_t)0x0000000000ff0000) |\
-                   (((uint64_t)x>> 8) & (uint64_t)0x00000000ff000000) |\
-                   (((uint64_t)x<< 8) & (uint64_t)0x000000ff00000000) |\
-                   (((uint64_t)x<<24) & (uint64_t)0x0000ff0000000000) |\
-                   (((uint64_t)x<<40) & (uint64_t)0x00ff000000000000) |\
-                   (((uint64_t)x<<56) & (uint64_t)0xff00000000000000))
+# define SWAP8(x) ((((uint64_t)x>>56) & (uint64_t)0x00000000000000ffULL) |\
+                   (((uint64_t)x>>40) & (uint64_t)0x000000000000ff00ULL) |\
+                   (((uint64_t)x>>24) & (uint64_t)0x0000000000ff0000ULL) |\
+                   (((uint64_t)x>> 8) & (uint64_t)0x00000000ff000000ULL) |\
+                   (((uint64_t)x<< 8) & (uint64_t)0x000000ff00000000ULL) |\
+                   (((uint64_t)x<<24) & (uint64_t)0x0000ff0000000000ULL) |\
+                   (((uint64_t)x<<40) & (uint64_t)0x00ff000000000000ULL) |\
+                   (((uint64_t)x<<56) & (uint64_t)0xff00000000000000ULL))
 #else
 # define SWAP2(a) (a)
 # define SWAP4(a) (a)
diff -Nrbu xawtv-3.95/libng/plugins/read-dv.c xawtv-3.95-OK/libng/plugins/read-dv.c
--- xawtv-3.95/libng/plugins/read-dv.c	2003-02-14 17:14:05.000000000 +0300
+++ xawtv-3.95-OK/libng/plugins/read-dv.c	2006-12-27 19:58:49.000000000 +0300
@@ -125,7 +125,7 @@
 		"dv: height=%d width=%d frame_size=%ld\n",
 		h->dec->quality, h->dec->system, h->dec->std,
 		h->dec->sampling, h->dec->num_dif_seqs, h->dec->height,
-		h->dec->width, h->dec->frame_size);
+		h->dec->width, (long) h->dec->frame_size);
 	fprintf(stderr, "dv: audio: %d Hz, %d bits, %d channels,"
 		" emphasis %s\n",
 		h->dec->audio->frequency,
