diff -Nur xawtv-3.95-old/libng/plugins/read-qt.c xawtv-3.95/libng/plugins/read-qt.c
--- xawtv-3.95-old/libng/plugins/read-qt.c	2007-07-08 15:19:44.836638644 +0300
+++ xawtv-3.95/libng/plugins/read-qt.c	2007-07-08 15:20:18.752482651 +0300
@@ -4,9 +4,9 @@
 #include <stdio.h>
 #include <string.h>
 #include <pthread.h>
-#include <quicktime/quicktime.h>
-#include <quicktime/colormodels.h>
-#include <quicktime/lqt.h>
+#include <lqt/quicktime.h>
+#include <lqt/colormodels.h>
+#include <lqt/lqt.h>
 
 #include "grab-ng.h"
 
diff -Nur xawtv-3.95-old/libng/plugins/write-qt.c xawtv-3.95/libng/plugins/write-qt.c
--- xawtv-3.95-old/libng/plugins/write-qt.c	2007-07-08 15:19:44.836638644 +0300
+++ xawtv-3.95/libng/plugins/write-qt.c	2007-07-08 15:20:34.658064637 +0300
@@ -4,9 +4,9 @@
 #include <stdio.h>
 #include <string.h>
 #include <pthread.h>
-#include <quicktime/quicktime.h>
-#include <quicktime/colormodels.h>
-#include <quicktime/lqt.h>
+#include <lqt/quicktime.h>
+#include <lqt/colormodels.h>
+#include <lqt/lqt.h>
 
 #include "grab-ng.h"
 
