diff -Nrbu xawtv-3.95/console/webcam.c xawtv-3.95-OK/console/webcam.c
--- xawtv-3.95/console/webcam.c	2005-09-30 14:07:58.000000000 +0400
+++ xawtv-3.95-OK/console/webcam.c	2007-08-30 16:46:12.000000000 +0400
@@ -813,7 +813,7 @@
     tmpdir = (NULL != getenv("TMPDIR")) ? getenv("TMPDIR") : "/tmp";
     list_for_each(item,&connections) {
 	s = list_entry(item, struct xfer_state, list);
-	s->ops->open(s);
+	(s->ops->open)(s);
     }
 
     /* print config */
diff -Nrbu xawtv-3.95/libng/grab-ng.c xawtv-3.95-OK/libng/grab-ng.c
--- xawtv-3.95/libng/grab-ng.c	2003-03-28 15:19:28.000000000 +0300
+++ xawtv-3.95-OK/libng/grab-ng.c	2007-08-30 16:49:19.000000000 +0400
@@ -572,7 +572,7 @@
 	    continue;
 	if (ng_debug)
 	    fprintf(stderr,"vid-open: trying: %s... \n", drv->name);
-	if (NULL != (*handle = drv->open(device)))
+	if (NULL != (*handle = (drv->open)(device)))
 	    break;
 	if (ng_debug)
 	    fprintf(stderr,"vid-open: failed: %s\n",drv->name);
@@ -603,7 +603,7 @@
 	    continue;
 	if (ng_debug)
 	    fprintf(stderr,"dsp-open: trying: %s... \n", drv->name);
-	if (NULL != (*handle = drv->open(device,fmt,record)))
+	if (NULL != (*handle = (drv->open)(device,fmt,record)))
 	    break;
 	if (ng_debug)
 	    fprintf(stderr,"dsp-open: failed: %s\n", drv->name);
@@ -628,7 +628,7 @@
         drv = list_entry(item, struct ng_mix_driver, list);
 	if (ng_debug)
 	    fprintf(stderr,"mix-init: trying: %s... \n", drv->name);
-	if (NULL != (handle = drv->open(device))) {
+	if (NULL != (handle = (drv->open)(device))) {
 	    if (NULL != (attrs = drv->volctl(handle,channel)))
 		break;
 	    drv->close(handle);
