diff -Nrbu xawtv-3.95/common/vbi-data.c xawtv-3.95-OK/common/vbi-data.c
--- xawtv-3.95/common/vbi-data.c	2003-02-14 17:14:04.000000000 +0300
+++ xawtv-3.95-OK/common/vbi-data.c	2007-03-01 20:33:44.000000000 +0300
@@ -48,6 +48,18 @@
     if (NULL == vbi->dec)
 	goto oops;
 
+    /*
+     * Give the user possibility to change default zvbi region (16, West-Europe)
+     * Sometimes the region value just not reported in the stream etc...
+     */
+    if (1) {
+	char *env = getenv("ALEVTD_REGION");
+	unsigned int region;
+
+	if (env && (region = strtoul(env,NULL,0)) != 0)
+	    vbi_teletext_set_default_region(vbi->dec,region);
+    }
+
     if (vbi->sim) {
 	vbi->par = init_sim(625,services);
         /* simulation for select */
