diff -Nrbu xawtv-3.95/libng/plugins/Subdir.mk xawtv-3.95-OK/libng/plugins/Subdir.mk
--- xawtv-3.95/libng/plugins/Subdir.mk	2008-08-26 19:55:00.000000000 +0400
+++ xawtv-3.95-OK/libng/plugins/Subdir.mk	2008-08-26 19:58:40.000000000 +0400
@@ -21,7 +21,6 @@
 ifeq ($(FOUND_OS),linux)
 TARGETS-plugins += \
 	libng/plugins/drv0-v4l2.so \
-	libng/plugins/drv0-v4l2-old.so \
 	libng/plugins/drv1-v4l.so \
 	libng/plugins/snd-oss.so
 endif
@@ -55,7 +54,6 @@
 
 libng/plugins/conv-mjpeg.so: libng/plugins/conv-mjpeg.o
 libng/plugins/drv0-bsd.so:   libng/plugins/drv0-bsd.o
-libng/plugins/drv0-v4l2-old.so: libng/plugins/drv0-v4l2-old.o
 libng/plugins/flt-debug.so:  libng/plugins/flt-debug.o
 libng/plugins/flt-disor.so:  libng/plugins/flt-disor.o
 libng/plugins/flt-gamma.so:  libng/plugins/flt-gamma.o
