http://bugs.gentoo.org/show_bug.cgi?id=288158

--- xawtv-3.95/libng/plugins/conv-mjpeg.c
+++ xawtv-3.95/libng/plugins/conv-mjpeg.c
@@ -229,6 +229,7 @@
     jpeg_set_quality(&h->mjpg_cinfo, ng_jpeg_quality, TRUE);
 
     h->mjpg_cinfo.raw_data_in = TRUE;
+    h->mjpg_cinfo.do_fancy_downsampling = FALSE;
     jpeg_set_colorspace(&h->mjpg_cinfo,JCS_YCbCr);
 
     h->mjpg_ptrs[0] = malloc(h->fmt.height*sizeof(char*));
@@ -453,6 +454,7 @@
     h->buf = in;
     jpeg_read_header(&h->mjpg_cinfo,1);
     h->mjpg_cinfo.raw_data_out = 1;
+    h->mjpg_cinfo.do_fancy_upsampling = FALSE;
 
     if (ng_debug > 1)
 	fprintf(stderr,"yuv: %dx%d  -  %d %d / %d %d / %d %d\n",
