diff -uNr xawtv-3.100.orig/common/channel.c xawtv-3.100/common/channel.c
--- xawtv-3.100.orig/common/channel.c	2011-02-22 07:56:44.000000000 +0200
+++ xawtv-3.100/common/channel.c	2011-05-07 15:35:34.948000119 +0300
@@ -181,6 +181,9 @@
     {  -1, NULL,     },
 };
 
+extern struct STRTAB booltab[];
+extern struct STRTAB captab[];
+
 /* just malloc memory for a new channel ... */
 struct CHANNEL*
 add_channel(char *name)
diff -uNr xawtv-3.100.orig/common/channel.h xawtv-3.100/common/channel.h
--- xawtv-3.100.orig/common/channel.h	2011-02-22 07:56:44.000000000 +0200
+++ xawtv-3.100/common/channel.h	2011-05-07 15:36:33.398000163 +0300
@@ -90,8 +90,5 @@
 
 /* ----------------------------------------------------------------------- */
 
-extern struct STRTAB booltab[];
-extern struct STRTAB captab[];
-
 int str_to_int(char *str, struct STRTAB *tab);
 const char* int_to_str(int n, struct STRTAB *tab);
diff -uNr xawtv-3.100.orig/common/commands.c xawtv-3.100/common/commands.c
--- xawtv-3.100.orig/common/commands.c	2011-02-22 07:56:44.000000000 +0200
+++ xawtv-3.100/common/commands.c	2011-05-07 15:37:07.600000117 +0300
@@ -162,6 +162,9 @@
 
 static int keypad_state = -1;
 
+extern struct STRTAB booltab[];
+extern struct STRTAB captab[];
+
 /* ----------------------------------------------------------------------- */
 
 void add_attrs(struct ng_attribute *new) 
