diff -Nur xawtv-3.95-old/Makefile.in xawtv-3.95/Makefile.in
--- xawtv-3.95-old/Makefile.in	2008-10-16 12:55:32.000000000 +0300
+++ xawtv-3.95/Makefile.in	2008-10-16 12:57:51.000000000 +0300
@@ -29,7 +29,7 @@
 # for CFLAGS
 WARN_FLAGS	:= -Wall -Wmissing-prototypes -Wstrict-prototypes -Wpointer-arith
 LFS_FLAGS	:= -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
-X11_FLAGS	:= @X_CFLAGS@ @XFT_FLAGS@ -I/usr/X11R6/include/X11/fonts
+X11_FLAGS	:= @X_CFLAGS@ @XFT_FLAGS@ -I/usr/include/X11/fonts
 LIB_FLAGS	:= -I. -I./vbistuff -I./x11 \
 		   -I$(srcdir)/jwz -I$(srcdir)/common -I$(srcdir)/console \
 		   -I$(srcdir)/x11 -I$(srcdir)/structs \
@@ -49,7 +49,7 @@
 VBI_LIBS	:= @LIBZVBI@
 GL_LIBS		:= @LIBGL@
 DV_LIBS		:= @LIBDV@
-FS_LIBS		:= @X_LIBS@ @FSLIB@
+FS_LIBS		:= @FSLIB@
 DLFLAGS		:= @DLFLAGS@
 
 # stuff configure has found
