--- gtest-1.5.0/Makefile.am 2011-03-19 10:54:43.000000000 +0100
+++ gtest-1.5.0/Makefile.am 2011-03-19 10:58:37.000000000 +0100
@@ -234,6 +234,7 @@ TESTS += samples/sample1_unittest
 check_PROGRAMS += samples/sample1_unittest
 samples_sample1_unittest_SOURCES = samples/sample1_unittest.cc
 samples_sample1_unittest_LDADD = lib/libgtest_main.la \
+                                 lib/libgtest.la \
                                  samples/libsamples.la
 
 # Another sample.  It also verifies that libgtest works.
@@ -247,7 +248,8 @@ samples_sample10_unittest_LDADD = lib/li
 TESTS += test/gtest_all_test
 check_PROGRAMS += test/gtest_all_test
 test_gtest_all_test_SOURCES = test/gtest_all_test.cc
-test_gtest_all_test_LDADD = lib/libgtest_main.la
+test_gtest_all_test_LDADD = lib/libgtest_main.la \
+                            lib/libgtest.la
 
 # Tests that fused gtest files compile and work.
 FUSED_GTEST_SRC = \
--- gtest-1.5.0/Makefile.in 2011-03-19 10:54:49.000000000 +0100
+++ gtest-1.5.0/Makefile.in 2011-03-19 10:58:53.000000000 +0100
@@ -444,12 +444,14 @@ TESTS_ENVIRONMENT = GTEST_SOURCE_DIR="$(
 
 samples_sample1_unittest_SOURCES = samples/sample1_unittest.cc
 samples_sample1_unittest_LDADD = lib/libgtest_main.la \
+                                 lib/libgtest.la \
                                  samples/libsamples.la
 
 samples_sample10_unittest_SOURCES = samples/sample10_unittest.cc
 samples_sample10_unittest_LDADD = lib/libgtest.la
 test_gtest_all_test_SOURCES = test/gtest_all_test.cc
-test_gtest_all_test_LDADD = lib/libgtest_main.la
+test_gtest_all_test_LDADD = lib/libgtest_main.la \
+                            lib/libgtest.la
 
 # Tests that fused gtest files compile and work.
 FUSED_GTEST_SRC = \
