diff -Nur gtest-1.5.0-old//build-aux/ltmain.sh gtest-1.5.0/build-aux/ltmain.sh
--- gtest-1.5.0-old//build-aux/ltmain.sh	2008-02-08 02:57:47.000000000 +0200
+++ gtest-1.5.0/build-aux/ltmain.sh	2011-04-05 16:06:44.390001421 +0300
@@ -1647,7 +1647,7 @@
 	continue
 	;;
 
-     -mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe|-threads)
+     -mt|-mthreads|-kthread|-Kthread|-lpthread|-pthread|-pthreads|--thread-safe|-threads)
 	compiler_flags="$compiler_flags $arg"
 	compile_command="$compile_command $arg"
 	finalize_command="$finalize_command $arg"
@@ -2150,7 +2150,7 @@
 	lib=
 	found=no
 	case $deplib in
-	-mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe|-threads)
+	-mt|-mthreads|-kthread|-Kthread|-lpthread|-pthread|-pthreads|--thread-safe|-threads)
 	  if test "$linkmode,$pass" = "prog,link"; then
 	    compile_deplibs="$deplib $compile_deplibs"
 	    finalize_deplibs="$deplib $finalize_deplibs"
diff -Nur gtest-1.5.0-old//m4/acx_pthread.m4 gtest-1.5.0/m4/acx_pthread.m4
--- gtest-1.5.0-old//m4/acx_pthread.m4	2010-04-16 01:02:00.000000000 +0300
+++ gtest-1.5.0/m4/acx_pthread.m4	2011-04-05 16:07:37.184001436 +0300
@@ -99,7 +99,7 @@
 # which indicates that we try without any flags at all, and "pthread-config"
 # which is a program returning the flags for the Pth emulation library.
 
-acx_pthread_flags="pthreads none -Kthread -kthread lthread -pthread -pthreads -mthreads pthread --thread-safe -mt pthread-config"
+acx_pthread_flags="pthreads none -Kthread -kthread lthread -lpthread -pthread -pthreads -mthreads pthread --thread-safe -mt pthread-config"
 
 # The ordering *is* (sometimes) important.  Some notes on the
 # individual items follow:
