diff -Nuar openconnect-5.01.old/configure.ac openconnect-5.01/configure.ac
--- openconnect-5.01.old/configure.ac	2013-06-01 23:21:19.000000000 +0300
+++ openconnect-5.01/configure.ac	2013-07-05 23:52:18.035160836 +0300
@@ -328,19 +328,6 @@
     AC_CHECK_FUNC(gnutls_pkcs11_add_provider,
 		 [PKG_CHECK_MODULES(P11KIT, p11-kit-1, [AC_DEFINE(HAVE_P11KIT)
     			      		  AC_SUBST(P11KIT_PC, p11-kit-1)], [:])], [])
-    LIBS="$oldlibs -ltspi"
-    AC_MSG_CHECKING([for tss library])
-    AC_LINK_IFELSE([AC_LANG_PROGRAM([
-		   #include <trousers/tss.h>
-		   #include <trousers/trousers.h>],[
-		   int err = Tspi_Context_Create((void *)0);
-		   Trspi_Error_String(err);])],
-		  [AC_MSG_RESULT(yes)
-		   AC_SUBST([TSS_LIBS], [-ltspi])
-		   AC_SUBST([TSS_CFLAGS], [])
-		   AC_DEFINE(HAVE_TROUSERS, 1)],
-		  [AC_MSG_RESULT(no)])
-    LIBS="$oldlibs"
     CFLAGS="$oldcflags"
 fi
 if test "$with_openssl" = "yes" || test "$with_openssl" = "" || test "$ssl_library" = "both"; then
@@ -474,6 +461,7 @@
 AM_CONDITIONAL(HAVE_SYMBOL_VERSIONING, [test "${symvers}" != "no"])
 
 PKG_CHECK_MODULES(LIBXML2, libxml-2.0)
+PKG_CHECK_MODULES(ZLIB, zlib)
 
 PKG_CHECK_MODULES(ZLIB, zlib, [AC_SUBST(ZLIB_PC, [zlib])],
 		  [oldLIBS="$LIBS"
