--- data/org.mate.pluma.gschema.xml.in.in.OSmanOS	2014-01-06 16:41:07.000000000 +0200
+++ data/org.mate.pluma.gschema.xml.in.in			2014-02-04 19:14:35.855610967 +0200
@@ -52,7 +52,7 @@
       <_description>Specifies the number of spaces that should be displayed instead of Tab characters.</_description>
     </key>
     <key name="insert-spaces" type="b">
-      <default>false</default>
+      <default>true</default>
       <_summary>Insert spaces</_summary>
       <_description>Whether pluma should insert spaces instead of tabs.</_description>
     </key>
@@ -62,12 +62,12 @@
       <_description>Whether pluma should enable automatic indentation.</_description>
     </key>
     <key name="display-line-numbers" type="b">
-      <default>false</default>
+      <default>true</default>
       <_summary>Display Line Numbers</_summary>
       <_description>Whether pluma should display line numbers in the editing area.</_description>
     </key>
     <key name="highlight-current-line" type="b">
-      <default>false</default>
+      <default>true</default>
       <_summary>Highlight Current Line</_summary>
       <_description>Whether pluma should highlight the current line.</_description>
     </key>
