Index: Makefile
===================================================================
--- Makefile
+++ Makefile	2012-07-19 01:06:50.018903718 +0200
@@ -59,7 +59,7 @@
 	$(MAKE) target=AIX \
 		CC=cc CXX=$(CC) \
 		CFLAGS='-O2' \
-		CXFLAGS='' \
+		CXXFLAGS='' \
 		opt='$(opt)' \
 		crafty-make
 
@@ -67,7 +67,7 @@
 	$(MAKE) target=FreeBSD \
 		CC=gcc CXX=g++ \
 		CFLAGS='-Wall -pipe -O3' \
-		CXFLAGS='-Wall -pipe -O3' \
+		CXXFLAGS='-Wall -pipe -O3' \
 		LDFLAGS=$(LDFLAGS) \
 		LIBS='-lstdc++' \
 		opt='$(opt)' \
@@ -77,16 +77,16 @@
 	$(MAKE) target=LINUX \
 		CC=gcc CXX=g++ \
 		CFLAGS='-Wall -pipe -O3 -mcpu=G5 -mtune=G5 -fomit-frame-pointer -fast' \
-		CXFLAGS='-Wall -pipe -O3 -mcpu=G5 -mtune=G5 -fomit-frame-pointer -fast' \
+		CXXFLAGS='-Wall -pipe -O3 -mcpu=G5 -mtune=G5 -fomit-frame-pointer -fast' \
 		LDFLAGS='$(LDFLAGS) -lstdc++' \
 		opt='$(opt) -DCPUS=4 -DPOWERPC' \
 		crafty-make
-	
+
 freebsd:
 	$(MAKE) target=FreeBSD \
 		CC=gcc CXX='$(CC)' \
 		CFLAGS='-fomit-frame-pointer -m486 -O3 -Wall' \
-		CXFLAGS='-fomit-frame-pointer -m486 -O3 -Wall' \
+		CXXFLAGS='-fomit-frame-pointer -m486 -O3 -Wall' \
 		LDFLAGS=$(LDFLAGS) \
 		opt='$(opt) -DINLINE32' \
 		crafty-make
@@ -95,7 +95,7 @@
 	$(MAKE) target=FreeBSD \
 		CC=gcc CXX='$(CC)' \
 		CFLAGS='-pipe -mpentium -O -Wall' \
-		CXFLAGS='' \
+		CXXFLAGS='' \
 		LDFLAGS=$(LDFLAGS) \
 		opt='$(opt) -DINLINE32' \
 		crafty-make
@@ -104,7 +104,7 @@
 	$(MAKE) target=HP \
 		CC=cc CXX='$(CC)' \
 		CFLAGS='+ESlit -Ae +w1' \
-		CXFLAGS='' \
+		CXXFLAGS='' \
 		LDFLAGS='$(LDFLAGS) +O3 +Onolimit' \
 		crafty-make
 
@@ -112,35 +112,34 @@
 	$(MAKE) target=LINUX \
 		CC=gcc CXX=g++ \
                 CFLAGS='-Wall -pipe -fprofile-arcs -fomit-frame-pointer -O3 -march=k8' \
-		CXFLAGS='' \
-		LDFLAGS='$(LDFLAGS) -lpthread -lnuma -fprofile-arcs -lstdc++' \
-		opt='$(opt) -DINLINE64 -DCPUS=8 -DNUMA -DLIBNUMA' \
+		CXXFLAGS='' \
+		LDFLAGS='$(LDFLAGS) -lpthread -fprofile-arcs -lstdc++' \
+		opt='$(opt) -DINLINE64 -DCPUS=8' \
 		crafty-make
 
 linux-amd64:
 	$(MAKE) target=LINUX \
 		CC=gcc CXX=g++ \
-                CFLAGS='-Wall -pipe -fbranch-probabilities -fomit-frame-pointer -O3 -march=k8' \
-		CXFLAGS='' \
-		LDFLAGS='$(LDFLAGS) -lpthread -lnuma -lstdc++' \
-		opt='$(opt) -DINLINE64 -DCPUS=8 -DNUMA -DLIBNUMA' \
+                CFLAGS='' \
+		CXXFLAGS='' \
+		LDFLAGS='$(LDFLAGS) -lpthread -lstdc++' \
+		opt='$(opt) -DINLINE64 -DCPUS=8' \
 		crafty-make
 
 linux:
 	$(MAKE) target=LINUX \
 		CC=gcc CXX=g++ \
-		CFLAGS='-pg -Wwrite-strings -Wcast-qual -Wshadow \
-		-Wconversion -W -Wall -ansi -pedantic -pipe' \
-		CXFLAGS='' \
+		CFLAGS='' \
+		CXXFLAGS='' \
 		LDFLAGS='$(LDFLAGS) -pg -lstdc++ -lpthread' \
-		opt='$(opt) -DTRACE -DINLINE64 -DCPUS=2' \
+		opt='$(opt) -DTRACE -DCPUS=2' \
 		crafty-make
 
 linux-profile:
 	$(MAKE) target=LINUX \
 		CC=gcc CXX=g++ \
 		CFLAGS='-Wall -pipe -O3 -fprofile-arcs' \
-		CXFLAGS='' \
+		CXXFLAGS='' \
 		LDFLAGS='$(LDFLAGS) -fprofile-arcs -lstdc++ ' \
 		opt='$(opt) -DINLINE32 -DCPUS=2' \
 		crafty-make
@@ -149,7 +148,7 @@
 	$(MAKE) target=LINUX \
 		CC=gcc CXX=gcc \
 		CFLAGS='-O2 -w -g' \
-		CXFLAGS='-O2 -w -g' \
+		CXXFLAGS='-O2 -w -g' \
 		LDFLAGS='$(LDFLAGS) -g -lpthread -lstdc++' \
 		opt='$(opt) -DINLINE64 -DCPUS=2' \
 		crafty-make
@@ -158,7 +157,7 @@
 	$(MAKE) target=LINUX \
 		CC=icc CXX=icc \
 		CFLAGS='-w -xP -O2 -fno-alias -prof-use -prof_dir ./profdir' \
-		CXFLAGS='-w -xP -O2 -prof-use -prof-dir ./profdir' \
+		CXXFLAGS='-w -xP -O2 -prof-use -prof-dir ./profdir' \
 		LDFLAGS='$(LDFLAGS) -lpthread -lstdc++' \
 		opt='$(opt) -DINLINE64 -DCPUS=2' \
 		crafty-make
@@ -167,7 +166,7 @@
 	$(MAKE) target=LINUX \
 		CC=icc CXX=icc \
 		CFLAGS='-w -xP -O2 -fno-alias -prof-gen=srcpos -prof-dir ./profdir' \
-		CXFLAGS='-w -xP -O2 -ip -prof-gen=srcpos -prof-dir ./profdir' \
+		CXXFLAGS='-w -xP -O2 -ip -prof-gen=srcpos -prof-dir ./profdir' \
 		LDFLAGS='$(LDFLAGS) -lpthread -lstdc++ ' \
 		opt='$(opt) -DINLINE64 -DCPUS=2' \
 		crafty-make
@@ -176,7 +175,7 @@
 	$(MAKE) target=LINUX \
 		CC=icc CXX=icc \
 		CFLAGS='-w -O2 -prof_use -prof_dir ./profdir -fno-alias' \
-		CXFLAGS='-w -O2 -prof_use -prof_dir ./profdir' \
+		CXXFLAGS='-w -O2 -prof_use -prof_dir ./profdir' \
 		LDFLAGS='$(LDFLAGS) -lstdc++' \
 		opt='$(opt) -DINLINE32 -DCPUS=1' \
 		crafty-make
@@ -185,7 +184,7 @@
 	$(MAKE) target=LINUX \
 		CC=icc CXX=icc \
 		CFLAGS='-w -O2 -prof_genx -prof_dir ./profdir -fno-alias' \
-		CXFLAGS='-w -O2 -prof_genx -prof_dir ./profdir' \
+		CXXFLAGS='-w -O2 -prof_genx -prof_dir ./profdir' \
 		LDFLAGS='$(LDFLAGS) -lstdc++ ' \
 		opt='$(opt) -DINLINE32 -DCPUS=2' \
 		crafty-make
@@ -194,7 +193,7 @@
 	$(MAKE) target=NetBSD \
 		CC=gcc CXX=g++ \
 		CFLAGS='-O3 -Wall -fomit-frame-pointer' \
-		CXFLAGS='-O3 -Wall -fomit-frame-pointer' \
+		CXXFLAGS='-O3 -Wall -fomit-frame-pointer' \
 		LDFLAGS=$(LDFLAGS) \
 		opt='$(opt) -DINLINE32' \
 		crafty-make
@@ -203,7 +202,7 @@
 	$(MAKE) target=SUN \
 		CC=cc CXX='$(CC)' \
 		CFLAGS='-fast -xO5 -xunroll=20' \
-		CXFLAGS='-fast -xO5 -xunroll=20' \
+		CXXFLAGS='-fast -xO5 -xunroll=20' \
 		LDFLAGS='$(LDFLAGS)' \
 		opt='$(opt)' \
 		crafty-make
@@ -212,7 +211,7 @@
 	$(MAKE) target=SUN \
 		CC=gcc CXX=g++ \
 		CFLAGS='-Wall -pipe -O2 -fforce-mem -fomit-frame-pointer' \
-		CXFLAGS='-Wall -pipe -O2 -fforce-mem -fomit-frame-pointer' \
+		CXXFLAGS='-Wall -pipe -O2 -fforce-mem -fomit-frame-pointer' \
 		LDFLAGS='$(LDFLAGS) -lstdc++' \
 		opt='$(opt)' \
 		crafty-make
@@ -330,15 +329,15 @@
 includes = data.h chess.h
 
 crafty-make:
-	@$(MAKE) opt='$(opt)' CXFLAGS='$(CXFLAGS)' CFLAGS='$(CFLAGS)' crafty
+	@$(MAKE) opt='$(opt)' CXXFLAGS='$(CXXFLAGS)' CFLAGS='$(CFLAGS)' crafty
 
 crafty.o: *.c *.h
 
 crafty:	$(objects) egtb.o
-	$(CC) $(LDFLAGS) -o crafty $(objects) egtb.o -lm  $(LIBS)
+	$(CXX) $(LDFLAGS) -o crafty $(objects) egtb.o -lm  $(LIBS)
 
 egtb.o: egtb.cpp
-	$(CXX) -c $(CXFLAGS) $(opts) egtb.cpp
+	$(CXX) -c $(CXXFLAGS) $(opts) egtb.cpp
 clean:
 	-rm -f *.o crafty
 
