Index: chess.h
===================================================================
--- chess.h
+++ chess.h	2012-07-18 01:30:32.572188264 +0200
@@ -110,16 +110,16 @@
 #    define UNIX        /* system is unix-based                       */
 #  endif
 #  if !defined(BOOKDIR)
-#    define     BOOKDIR        "."
+#    define     BOOKDIR        "/usr/share/games/crafty"
 #  endif
 #  if !defined(LOGDIR)
 #    define      LOGDIR        "."
 #  endif
 #  if !defined(TBDIR)
-#    define       TBDIR     "./TB"
+#    define       TBDIR     "/usr/share/games/crafty/tb"
 #  endif
 #  if !defined(RCDIR)
-#    define       RCDIR        "."
+#    define       RCDIR        "/etc"
 #  endif
 #  define MAXPLY                                  65
 #  define MAX_TC_NODES                       1000000
Index: option.c
===================================================================
--- option.c
+++ option.c	2012-07-18 01:31:12.767685751 +0200
@@ -1093,7 +1093,7 @@
     char *readstat = (char *) -1;
     int lines = 0;
 
-    helpfile = fopen("crafty.hlp", "r");
+    helpfile = fopen("/usr/share/crafty/crafty.hlp", "r");
     if (!helpfile) {
       printf("ERROR.  Unable to open \"crafty.hlp\" -- help unavailable\n");
       return (1);
