Index: crafty-23.4/chess.h
===================================================================
--- crafty-23.4.orig/chess.h
+++ crafty-23.4/chess.h
@@ -110,16 +110,16 @@
 #    define UNIX        /* system is unix-based                       */
 #  endif
 #  if !defined(BOOKDIR)
-#    define     BOOKDIR        "."
+#    define     BOOKDIR        "/usr/share/crafty"
 #  endif
 #  if !defined(LOGDIR)
 #    define      LOGDIR        "."
 #  endif
 #  if !defined(TBDIR)
-#    define       TBDIR     "./TB"
+#    define       TBDIR     "/usr/share/crafty/tb"
 #  endif
 #  if !defined(RCDIR)
-#    define       RCDIR        "."
+#    define       RCDIR        "/etc"
 #  endif
 #  define MAXPLY                                  65
 #  define MAX_TC_NODES                       1000000
