--- openbabel-2.3.1/scripts/ruby/extconf.rb.gcc47	2011-01-11 23:53:32.000000000 +0900
+++ openbabel-2.3.1/scripts/ruby/extconf.rb	2012-04-05 14:15:14.286842274 +0900
@@ -3,11 +3,13 @@
 # Compensate for the fact that Ruby will try to build universal
 # binaries on OS X by default
 require 'rbconfig'
+darwin_ldflags = ""
 if Config::CONFIG["arch"] =~ /universal-darwin/
   ENV['ARCHFLAGS'] = case `uname -smr`.chomp
     when "i386" then '-arch i386'
     when "ppc"  then '-arch ppc'
   end
+  darwin_ldflags = "-flat_namespace"
 end
 
 require 'mkmf'
@@ -36,7 +38,7 @@
 end
 
 if have_library('openbabel')
-  with_ldflags("#$LDFLAGS -dynamic -flat_namespace") do #Enables cc to handle linking better.
+  with_ldflags("#$LDFLAGS -dynamic #$darwin_ldflags") do #Enables cc to handle linking better.
   create_makefile('openbabel')
 end
 else
