--- openbabel-2.3.1/scripts/CMakeLists.txt.ruby19	2012-04-06 00:09:36.000000000 +0900
+++ openbabel-2.3.1/scripts/CMakeLists.txt	2012-04-06 01:03:29.000000000 +0900
@@ -324,6 +324,10 @@
       endif (RUBY_PREFIX)
       add_custom_target(bindings_ruby ALL
         COMMAND ${RUBY_EXECUTABLE} ${openbabel_SOURCE_DIR}/scripts/ruby/extconf.rb --with-openbabel-lib=${openbabel_BINARY_DIR}/lib${LIB_SUFFIX} --with-openbabel-include=${openbabel_BINARY_DIR}/include
+
+        # As same as rubygems does
+        COMMAND sed -i -e \"\\@^RUBYARCHDIR@s|sitearchdir|vendorarchdir|\" ${openbabel_SOURCE_DIR}/scripts/ruby/Makefile
+
         COMMAND ${CMAKE_COMMAND} -E echo "Compiling Ruby bindings"
         COMMAND make
         DEPENDS ${openbabel_SOURCE_DIR}/scripts/ruby/openbabel-ruby.cpp