diff -up openbabel-2.3.1/scripts/CMakeLists.txt.r openbabel-2.3.1/scripts/CMakeLists.txt
--- openbabel-2.3.1/scripts/CMakeLists.txt.r	2011-10-08 22:09:17.000000000 +0200
+++ openbabel-2.3.1/scripts/CMakeLists.txt	2011-10-12 18:57:31.000000000 +0200
@@ -251,7 +251,7 @@ if (DO_PERL_BINDINGS)
             WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts/perl
         )
       add_dependencies(bindings_perl pre_bindings_perl openbabel)
-      install(CODE "execute_process(COMMAND make install DESTDIR=\$ENV{DESTDIR} WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts/perl)")
+      install(CODE "execute_process(COMMAND make install_vendor DESTDIR=\$ENV{DESTDIR} WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts/perl)")
 
     else (NOT MSVC)
       add_library(bindings_perl MODULE ${openbabel_SOURCE_DIR}/scripts/perl/openbabel-perl.cpp)