diff -ru openbabel-2.3.0/CMakeLists.txt openbabel-2.3.0-lg/CMakeLists.txt
--- openbabel-2.3.0/CMakeLists.txt	2010-10-23 02:05:07.000000000 +0400
+++ openbabel-2.3.0-lg/CMakeLists.txt	2011-02-26 18:49:31.000000000 +0300
@@ -218,7 +218,7 @@
     }
    " SCANDIR_NEEDS_CONST)
 
-  set(OB_MODULE_PATH "${CMAKE_INSTALL_PREFIX}/${OB_PLUGIN_INSTALL_DIR}")
+  set(OB_MODULE_PATH "${OB_PLUGIN_INSTALL_DIR}")
   add_definitions(-DOB_MODULE_PATH="\\"${OB_MODULE_PATH}\\"")
 
   # Add some visibility support when using GCC