Index: pyscard-1.6.12/setup.py
===================================================================
--- pyscard-1.6.12.orig/setup.py
+++ pyscard-1.6.12/setup.py
@@ -170,13 +170,6 @@ kw = {'name': "pyscard",
                    ],
       'package_dir': {"": "."},
       'package_data': {
-                         "smartcard": [
-                                        "ACKS",
-                                        "ChangeLog",
-                                        "LICENSE",
-                                        "README",
-                                        "TODO",
-                                        ],
                          "smartcard/wx": ["resources/*.ico"],
                        },
 
