--- tank.hpp.old	2006-01-28 11:27:16.000000000 -0700
+++ tank.hpp	2006-01-28 11:27:26.000000000 -0700
@@ -61,7 +61,7 @@
 
 class TankFire : public Sprite {
 public:
-    TankFire::TankFire(List *list, Tank *tank, float x, float y, float z,float dx, float dz,float rotated);
+    TankFire(List *list, Tank *tank, float x, float y, float z,float dx, float dz,float rotated);
     void draw();
 	void drawShadowable();
     void move();
--- scratcher.hpp.old	2006-01-28 11:29:22.000000000 -0700
+++ scratcher.hpp	2006-01-28 11:29:31.000000000 -0700
@@ -34,7 +34,7 @@
             UNDEF
     } ;
     
-    Scratcher:: Scratcher(List *list, float x, float z, int bench=0);
+    Scratcher(List *list, float x, float z, int bench=0);
     void draw();
     void drawShadowable();
     void move();
