--- myship.cpp.orig	2002-10-13 00:16:16.000000000 +0300
+++ myship.cpp	2008-04-27 23:51:45.000000000 +0300
@@ -21,6 +21,8 @@
 
 #include <GL/glut.h>
 #include <GL/glext.h>
+#define GL_GLEXT_PROTOTYPES
+#include <GL/gl.h>
 
 #include "sprite.hpp"
 #include "explosion.hpp"
@@ -41,6 +43,9 @@
 extern PFNGLFINALCOMBINERINPUTNVPROC glFinalCombinerInputNV;
 #ifdef WIN32
 extern PFNGLACTIVETEXTUREARBPROC glActiveTextureARB;
+#else
+#define glActiveTexture(x) glActiveTextureARB(x)
+extern PFNGLACTIVETEXTUREARBPROC glActiveTextureARB;
 #endif
 int MyShip::list_;
 
--- scene.cpp.orig	2002-10-10 15:57:58.000000000 +0300
+++ scene.cpp	2008-04-27 23:52:41.000000000 +0300
@@ -66,6 +66,11 @@
 #ifdef WIN32
 PFNGLACTIVETEXTUREARBPROC glActiveTextureARB;
 PFNGLMULTITEXCOORD2FARBPROC glMultiTexCoord2fARB; 
+#else
+#define glActiveTexture(x) glActiveTextureARB(x)
+#define glMultiTexCoord2f(x,y,z) glMultiTexCoord2fARB(x,y,z)
+PFNGLACTIVETEXTUREARBPROC glActiveTextureARB;
+PFNGLMULTITEXCOORD2FARBPROC glMultiTexCoord2fARB;
 #endif
 
 GLenum glError;
@@ -1498,6 +1503,9 @@
 #ifdef WIN32
 	glActiveTextureARB=(PFNGLACTIVETEXTUREARBPROC)wglGetProcAddress("glActiveTextureARB"); 
 	glMultiTexCoord2fARB=(PFNGLMULTITEXCOORD2FARBPROC)wglGetProcAddress("glMultiTexCoord2fARB"); 
+#else
+	glActiveTextureARB=(PFNGLACTIVETEXTUREARBPROC)glXGetProcAddressARB((const GLubyte*)"glActiveTextureARB");
+	glMultiTexCoord2fARB=(PFNGLMULTITEXCOORD2FARBPROC)glXGetProcAddressARB((const GLubyte*)"glMultiTexCoord2fARB");
 #endif
 
 #if defined( GL_REGISTER_COMBINERS_NV) && defined(GL_VERTEX_PROGRAM_NV)
@@ -2018,7 +2026,7 @@
 	oldTimer = newTimer;
 	globalList->move()	;
 #ifndef PI
-#define PI=3.151592653589793
+#define PI=3.141592653589793
 #endif
 	benchFrames++;
 
