diff -up minicom-2.4/src/config.c.orig minicom-2.4/src/config.c
--- minicom-2.4/src/config.c.orig	2010-03-10 12:41:42.000000000 +0100
+++ minicom-2.4/src/config.c	2010-03-10 13:16:16.233368251 +0100
@@ -52,6 +52,12 @@ void read_parms(void)
 
   /* Read global parameters */
   if ((fp = fopen(parfile, "r")) == NULL) {
+    if (dosetup) {
+      fputs(_("minicom: WARNING: configuration file not found, using defaults\n"),stderr);
+      sleep(2);
+      return;
+    }
+    
     fprintf(stderr,
             _("minicom: there is no global configuration file %s\n"), parfile);
     fputs(_("Ask your sysadmin to create one (with minicom -s).\n"), stderr);
