diff -up ./minicom-2.4/src/config.c.umask ./minicom-2.4/src/config.c
--- minicom-2.4/src/config.c.umask	2010-03-09 16:32:14.876153489 +0100
+++ minicom-2.4/src/config.c	2010-03-09 16:33:17.789153650 +0100
@@ -1245,6 +1245,8 @@ int dodflsave(void)
   }
   writepars(fp, dosetup);
   fclose(fp);
+  if(dosetup)
+    chmod(fname, (mode_t) 0644);
   werror(_("Configuration saved"));
 
   return domacsave() < 0 ? -1 : 0;
