diff -up minicom-2.4/src/vt100.c.orig minicom-2.4/src/vt100.c
--- minicom-2.4/src/vt100.c.orig	2009-12-12 16:47:47.000000000 +0100
+++ minicom-2.4/src/vt100.c	2010-03-10 10:53:23.338369299 +0100
@@ -1012,7 +1012,7 @@ void vt_out(int ch)
       if (!using_iconv()) {
       c = vt_inmap[c];    /* conversion 04.09.97 / jl */
 #if TRANSLATE
-      if (vt_type == VT100 && vt_asis == 0)
+      if (vt_type == VT100 && vt_trans[vt_charset] && vt_asis == 0)
         c = vt_trans[vt_charset][c];
 #endif
       }
