diff -up webkitgtk-2.4.8/Source/WebCore/GNUmakefile.list.am.plugin_none webkitgtk-2.4.8/Source/WebCore/GNUmakefile.list.am
--- webkitgtk-2.4.8/Source/WebCore/GNUmakefile.list.am.plugin_none	2015-01-08 11:08:03.261853581 +0100
+++ Source/WebCore/GNUmakefile.list.am	2015-01-08 11:08:53.650619552 +0100
@@ -6254,9 +6254,11 @@ endif # END USE_GLX
 endif # END TARGET_X11
 
 if TARGET_WAYLAND
+if ENABLE_WEBKIT2
 webcore_sources += \
 	Source/WebCore/plugins/PluginPackageNone.cpp \
 	Source/WebCore/plugins/PluginViewNone.cpp
+endif # END ENABLE_WEBKIT2
 endif # END TARGET_WAYLAND
 
 if TARGET_X11_OR_WAYLAND
@@ -6305,9 +6307,11 @@ webcoregtk_sources += \
 endif # END TARGET_WIN32
 
 if TARGET_QUARTZ
+if ENABLE_WEBKIT2
 webcore_sources += \
 	Source/WebCore/plugins/PluginPackageNone.cpp \
 	Source/WebCore/plugins/PluginViewNone.cpp
+endif # END ENABLE_WEBKIT2
 platformgtk_sources += \
 	Source/WebCore/platform/cairo/WidgetBackingStoreCairo.h \
 	Source/WebCore/platform/cairo/WidgetBackingStoreCairo.cpp
 
