Index: docbook-utils-0.6.14/backends/man.in
===================================================================
--- docbook-utils-0.6.14.orig/backends/man.in
+++ docbook-utils-0.6.14/backends/man.in
@@ -8,7 +8,7 @@ TMPDIR=`mktemp -d /tmp/man.XXXXXX` || \
  { echo >&2 "man backend: could not create secure temporary directory"; exit 1;}
 trap 'rm -rf "${TMPDIR}"' EXIT
 nsgmls $SGML_FILE > "${TMPDIR}/nsgmls.tmp"
-sgmlspl $HELPER <"${TMPDIR}/nsgmls.tmp" 2>"${TMPDIR}/errs"
+sgmlspl.pl $HELPER <"${TMPDIR}/nsgmls.tmp" 2>"${TMPDIR}/errs"
 if [ $? -ne 0 ]
 then
   cat "${TMPDIR}/errs"
