Index: scidavis/src/fit_gsl.cpp
===================================================================
--- scidavis/src/fit_gsl.cpp.orig
+++ scidavis/src/fit_gsl.cpp
@@ -614,7 +614,7 @@ int lorentz_multi_peak_fdf (const gsl_ve
 
 int user_f(const gsl_vector * x, void *params, gsl_vector * f)
 {
-        static_cast<struct FitData*>(params)->fit->evaluate_f(x, f);
+        return static_cast<struct FitData*>(params)->fit->evaluate_f(x, f);
 }
 
 double user_d(const gsl_vector * x, void *params)
Index: scidavis/src/future/core/datatypes/Double2DateTimeFilter.cpp
===================================================================
--- scidavis/src/future/core/datatypes/Double2DateTimeFilter.cpp.orig
+++ scidavis/src/future/core/datatypes/Double2DateTimeFilter.cpp
@@ -100,5 +100,6 @@ QDateTime Double2DateTimeFilter::dateTim
 				return result.addMSecs(qRound64((input_value - int(floor(input_value))) * 1000.0));
 			}
 	}
+	return QDateTime();
 }
 
