Index: scidavis/sourcefiles.pri
===================================================================
--- scidavis/sourcefiles.pri.orig
+++ scidavis/sourcefiles.pri
@@ -48,7 +48,6 @@ contains( INSTALLS, translationfiles ){
 ###################### DOCUMENTATION ########################################
 
 documentation.files += ../manual/html \
-                       ../INSTALL.html \
                        ../scidavis.css \
                        ../scidavis-logo.png \
                        ../README \
