Index: scidavis/src/Graph.cpp
===================================================================
--- scidavis/src/Graph.cpp.orig
+++ scidavis/src/Graph.cpp
@@ -4931,11 +4931,15 @@ void Graph::guessUniqueCurveLayout(int&
 		}
 	}
 	if (n_curves > 1)
-		colorIndex = (++colorIndex)%16;
+	{
+		colorIndex = colorIndex + 1;
+		colorIndex = colorIndex%16;
+	}
 	if (colorIndex == 13) //avoid white invisible curves
 		colorIndex = 0;
 
-	symbolIndex = (++symbolIndex)%15;
+	symbolIndex = symbolIndex + 1;
+	symbolIndex = symbolIndex%15;
 	if (!symbolIndex)
 		symbolIndex = 1;
 }
