Index: scidavis/src/scidavis.sip
===================================================================
--- scidavis/src/scidavis.sip.orig
+++ scidavis/src/scidavis.sip
@@ -926,7 +926,7 @@ public:
   void removeCurve(const QString&);
   void deleteFitCurves();
   int curves() /PyName=numCurves/;
-  QList<QwtPlotCurve*> curves() const /NoDerived/;
+  QList<QwtPlotCurve*> curves() /NoDerived/;
 %MethodCode
 	sipRes = new QList<QwtPlotCurve*>();
 	for (int i = 0; i<sipCpp->curves(); i++)
@@ -995,7 +995,7 @@ public:
 	sipRes = sipCpp->d_plot->canvas();
 %End
 
-	QPointF pickPoint() const /NoDerived/;
+	QPointF pickPoint() /NoDerived/;
 %MethodCode
 	ApplicationWindow *app = sipscidavis_app();
 	sipRes = new QPointF();
