diff -ur hal-0.5.11.orig/tools/hal-storage-mount.c hal-0.5.11/tools/hal-storage-mount.c
--- hal-0.5.11.orig/tools/hal-storage-mount.c	2008-05-08 02:24:23.000000000 +0300
+++ hal-0.5.11/tools/hal-storage-mount.c	2008-06-16 10:07:41.000000000 +0300
@@ -586,6 +586,9 @@
 		if (strlen (mount_point) == 0) {
 			char *p;
 			
+			/* Really ugly hack to prevent utf8 errors :( */
+			label = g_strsplit_set(device, "/", -1)[2];
+						
 			if (label != NULL) {
 				/* best - use label */
 				g_strlcpy (mount_point, label, sizeof (mount_point));
