diff -ur hal-0.5.9.1.orig/hald/linux/acpi.c hal-0.5.9.1/hald/linux/acpi.c
--- hal-0.5.9.1.orig/hald/linux/acpi.c	2007-04-03 19:38:50.000000000 +0300
+++ hal-0.5.9.1/hald/linux/acpi.c	2007-08-16 12:39:30.000000000 +0300
@@ -50,6 +50,7 @@
 	ACPI_TYPE_SONY_DISPLAY,
 	ACPI_TYPE_OMNIBOOK_DISPLAY,
 	ACPI_TYPE_SONYPI_DISPLAY,
+	ACPI_TYPE_HP_DISPLAY,
 	ACPI_TYPE_BUTTON
 };
 
@@ -833,6 +834,10 @@
 		type = "sonypi";
 		desc = "Sony LCD Panel";
 		br_levels = 256;
+	} else if (acpi_type == ACPI_TYPE_HP_DISPLAY) {
+		type = "hp";
+		desc = "HP LCD Panel";
+		br_levels = 100;
 	} else {
 		type = "unknown";
 		desc = "Unknown LCD Panel";
@@ -1117,6 +1122,7 @@
 	acpi_synthesize_display ("omnibook", "lcd", ACPI_TYPE_OMNIBOOK_DISPLAY);
 	/* sonypi doesn't have an acpi object fd.o#6729 */
 	acpi_synthesize_sonypi_display ();
+	acpi_synthesize_display ("acpi/video/VGA/LCD", "brightness", ACPI_TYPE_HP_DISPLAY);
 
 	/* setup timer for things that we need to poll */
 	g_timeout_add (ACPI_POLL_INTERVAL,
@@ -1269,6 +1275,14 @@
 	.remove      = acpi_generic_remove
 };
 
+static ACPIDevHandler acpidev_handler_laptop_panel_hp = {
+	.acpi_type   = ACPI_TYPE_HP_DISPLAY,
+	.add         = acpi_generic_add,
+	.compute_udi = acpi_generic_compute_udi,
+	.refresh     = laptop_panel_refresh,
+	.remove      = acpi_generic_remove
+};
+
 static ACPIDevHandler acpidev_handler_button = {
 	.acpi_type   = ACPI_TYPE_BUTTON,
 	.add         = acpi_button_add,
@@ -1298,6 +1312,7 @@
 	&acpidev_handler_laptop_panel_sony,
 	&acpidev_handler_laptop_panel_omnibook,
 	&acpidev_handler_laptop_panel_sonypi,
+	&acpidev_handler_laptop_panel_hp,
 	NULL
 };
 
diff -ur hal-0.5.9.1.orig/tools/linux/hal-system-lcd-get-brightness-linux hal-0.5.9.1/tools/linux/hal-system-lcd-get-brightness-linux
--- hal-0.5.9.1.orig/tools/linux/hal-system-lcd-get-brightness-linux	2007-04-03 19:38:50.000000000 +0300
+++ hal-0.5.9.1/tools/linux/hal-system-lcd-get-brightness-linux	2007-08-16 12:39:30.000000000 +0300
@@ -64,6 +64,11 @@
 	# cat /proc/omnibook/lcd
 	#  LCD brightness:  7
 	value="`cat $HAL_PROP_LINUX_ACPI_PATH | awk '{print $3;}'`"
+elif [ "$HAL_PROP_LAPTOP_PANEL_ACCESS_METHOD" = "hp" ]; then
+	# cat /proc/acpi/video/VGA/LCD/brightness
+	# levels:  20 28 36 44 52 60 68 76 84 93 100
+	# current: 0
+	value="`cat $HAL_PROP_LINUX_ACPI_PATH | grep current: | awk '{print $2;}'`"
 else
 	echo "org.freedesktop.Hal.Device.LaptopPanel.NotSupported" >&2
 	echo "No ACPI method found" >&2
diff -ur hal-0.5.9.1.orig/tools/linux/hal-system-lcd-set-brightness-linux hal-0.5.9.1/tools/linux/hal-system-lcd-set-brightness-linux
--- hal-0.5.9.1.orig/tools/linux/hal-system-lcd-set-brightness-linux	2007-04-03 19:38:50.000000000 +0300
+++ hal-0.5.9.1/tools/linux/hal-system-lcd-set-brightness-linux	2007-08-16 12:40:46.000000000 +0300
@@ -59,6 +59,29 @@
 	# echo "{0..7} || {0...11}" > /proc/omnibook/lcd
 	# http://bugzilla.gnome.org/show_bug.cgi?id=331458
 	echo "$value" > $HAL_PROP_LINUX_ACPI_PATH
+elif [ "$HAL_PROP_LAPTOP_PANEL_ACCESS_METHOD" = "hp" ]; then
+	if [ $value -lt 20 ]; then
+		value="20"
+	elif [ $value -lt 30 ]; then
+		value="28"
+	elif [ $value -lt 40 ]; then
+		value="36"
+	elif [ $value -lt 50 ]; then
+		value="44"
+	elif [ $value -lt 60 ]; then
+		value="52"
+	elif [ $value -lt 65 ]; then
+		value="60"
+	elif [ $value -lt 70 ]; then
+		value="68"
+	elif [ $value -lt 80 ]; then
+		value="76"
+	elif [ $value -lt 90 ]; then
+		value="84"
+	else
+		value="93"
+	fi
+	echo "$value" > $HAL_PROP_LINUX_ACPI_PATH
 else
 	echo "org.freedesktop.Hal.Device.LaptopPanel.NotSupported" >&2
 	echo "No ACPI method found" >&2
