diff -ur hal-0.5.8.1_20061101.orig/fdi/policy/10osvendor/20-storage-methods.fdi hal-0.5.8.1_20061101/fdi/policy/10osvendor/20-storage-methods.fdi
--- hal-0.5.8.1_20061101.orig/fdi/policy/10osvendor/20-storage-methods.fdi	2006-10-02 02:05:09.000000000 +0300
+++ hal-0.5.8.1_20061101/fdi/policy/10osvendor/20-storage-methods.fdi	2006-11-16 23:03:13.000000000 +0200
@@ -165,6 +165,18 @@
 	<append key="volume.mount.valid_options" type="strlist">umask=</append>
       </match>
 
+     <!-- allow these mount options for ntfs-3g -->
+     <match key="volume.fstype" string="ntfs-3g">
+	<append key="volume.mount.valid_options" type="strlist">nonempty</append>
+	<append key="volume.mount.valid_options" type="strlist">show_sys_files</append>
+	<append key="volume.mount.valid_options" type="strlist">silent</append>
+	<append key="volume.mount.valid_options" type="strlist">no_def_opts</append>
+	<append key="volume.mount.valid_options" type="strlist">dmask=</append>
+	<append key="volume.mount.valid_options" type="strlist">fmask=</append>
+	<!-- DISABLED for new ntfs-3g <append key="volume.mount.valid_options" type="strlist">uid=</append> -->
+	<append key="volume.mount.valid_options" type="strlist">locale=</append>
+      </match>
+
       <!-- allow these mount options for ext3 -->
       <match key="volume.fstype" string="ext3">
 	<append key="volume.mount.valid_options" type="strlist">data=</append>
