Index: hal-0.5.14/policy/org.freedesktop.hal.device-access.policy
===================================================================
--- hal-0.5.14.orig/policy/org.freedesktop.hal.device-access.policy
+++ hal-0.5.14/policy/org.freedesktop.hal.device-access.policy
@@ -21,7 +21,9 @@ NOTE: Please keep the actions in alpabet
 
   <action id="org.freedesktop.hal.device-access.audio-player">
     <description>Directly access audio players</description>
+    <description xml:lang="tr">Ses oynatıcılarına doğrudan eriş</description>
     <message>System policy prevents access to audio players</message>
+    <message xml:lang="tr">Sistem politikası ses oynatıcılarına erişimi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -30,7 +32,9 @@ NOTE: Please keep the actions in alpabet
 
   <action id="org.freedesktop.hal.device-access.camera">
     <description>Directly access digital cameras</description>
+    <description xml:lang="tr">Sayısal fotoğraf makinelerine doğrudan eriş</description>
     <message>System policy prevents access to digital cameras</message>
+    <message xml:lang="tr">Sistem politikası sayısal fotoğraf makinelerine erişimi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -39,7 +43,9 @@ NOTE: Please keep the actions in alpabet
 
   <action id="org.freedesktop.hal.device-access.cdrom">
     <description>Directly access optical drives</description>
+    <description xml:lang="tr">Optik aygıtlara doğrudan eriş</description>
     <message>System policy prevents access to optical drives</message>
+    <message xml:lang="tr">Sistem politikası optik aygıtlara erişimi engelliyor</message>
     <defaults>
       <allow_inactive>yes</allow_inactive>
       <allow_active>yes</allow_active>
@@ -48,7 +54,9 @@ NOTE: Please keep the actions in alpabet
 
   <action id="org.freedesktop.hal.device-access.dvb">
     <description>Directly access DVB devices</description>
+    <description xml:lang="tr">DVB aygıtlarına doğrudan eriş</description>
     <message>System policy prevents access to DVB devices</message>
+    <message xml:lang="tr">Sistem politikası DVB aygıtlarına erişimi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -57,7 +65,9 @@ NOTE: Please keep the actions in alpabet
 
   <action id="org.freedesktop.hal.device-access.fingerprint-reader">
     <description>Directly access to fingerprint reader devices</description>
+    <description xml:lang="tr">Parmak izi okuyuculara doğrudan eriş</description>
     <message>System policy prevents access to fingerprint readers</message>
+    <message xml:lang="tr">Sistem politikası parmak izi okuyuculara erişimi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -66,7 +76,9 @@ NOTE: Please keep the actions in alpabet
 
   <action id="org.freedesktop.hal.device-access.floppy">
     <description>Directly access Floppy devices</description>
+    <description xml:lang="tr">Disket sürücülere doğrudan eriş</description>
     <message>System policy prevents access to Floppy devices</message>
+    <message xml:lang="tr">Sistem politikası disker sürücülere erişimi engelliyor</message>
     <defaults>
       <allow_inactive>yes</allow_inactive>
       <allow_active>yes</allow_active>
@@ -75,7 +87,9 @@ NOTE: Please keep the actions in alpabet
 
   <action id="org.freedesktop.hal.device-access.ieee1394-avc">
     <description>Directly access Firewire AVC devices</description>
+    <description xml:lang="tr">Firewire AVC aygıtlarına doğrudan eriş</description>
     <message>System policy prevents access to Firewire AVC devices</message>
+    <message xml:lang="tr">Sistem politikası Firewire AVC aygıtlarına erişimi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -84,7 +98,9 @@ NOTE: Please keep the actions in alpabet
 
   <action id="org.freedesktop.hal.device-access.ieee1394-iidc">
     <description>Directly access Firewire IIDC devices</description>
+    <description xml:lang="tr">Firewire IIDC aygıtlarına doğrudan eriş</description>
     <message>System policy prevents access to Firewire IIDC devices</message>
+    <message xml:lang="tr">Sistem politikası Firewire IIDC aygıtlarına erişimi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -138,7 +154,9 @@ NOTE: Please keep the actions in alpabet
 
   <action id="org.freedesktop.hal.device-access.pda">
     <description>Directly access PDA devices</description>
+    <description xml:lang="tr">PDA aygıtlarına doğrudan eriş</description>
     <message>System policy prevents access to PDA devices</message>
+    <message xml:lang="tr">Sistem politikası PDA aygıtlarına erişimi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -147,7 +165,9 @@ NOTE: Please keep the actions in alpabet
 
   <action id="org.freedesktop.hal.device-access.ppdev">
     <description>Directly access parallel port devices</description>
+    <description xml:lang="tr">Paralel port aygıtlarına doğrudan eriş</description>
     <message>System policy prevents access to parallel port devices</message>
+    <message xml:lang="tr">Sistem politikası paralel port aygıtlarına erişimi engelliyor</message>
     <defaults>
       <allow_inactive>auth_admin_keep_always</allow_inactive>
       <allow_active>auth_admin_keep_always</allow_active>
@@ -156,7 +176,9 @@ NOTE: Please keep the actions in alpabet
 
   <action id="org.freedesktop.hal.device-access.printer">
     <description>Directly access printers</description>
+    <description xml:lang="tr">Yazıcılara doğrudan eriş</description>
     <message>System policy prevents access to printers</message>
+    <message xml:lang="tr">Sistem politikası yazıcılara erişimi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -165,7 +187,9 @@ NOTE: Please keep the actions in alpabet
 
   <action id="org.freedesktop.hal.device-access.removable-block">
     <description>Directly access removable block devices</description>
+    <description xml:lang="tr">Çıkarılabilir blok aygıtlara doğrudan eriş</description>
     <message>System policy prevents access to removable block devices</message>
+    <message xml:lang="tr">Sistem politikası çıkarılabilir blok aygıtlara erişimi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>no</allow_active>
@@ -174,7 +198,9 @@ NOTE: Please keep the actions in alpabet
 
   <action id="org.freedesktop.hal.device-access.scanner">
     <description>Directly access scanners</description>
+    <description xml:lang="tr">Tarayıcılara doğrudan eriş</description>
     <message>System policy prevents access to scanners</message>
+    <message xml:lang="tr">Sistem politikası tarayıcılara erişimi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -183,7 +209,9 @@ NOTE: Please keep the actions in alpabet
 
   <action id="org.freedesktop.hal.device-access.sound">
     <description>Directly access sound devices</description>
+    <description xml:lang="tr">Ses aygıtlarına doğrudan eriş</description>
     <message>System policy prevents access to the sound devices</message>
+    <message xml:lang="tr">Sistem politikası ses aygıtlarına erişimi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -192,7 +220,9 @@ NOTE: Please keep the actions in alpabet
 
   <action id="org.freedesktop.hal.device-access.video">
     <description>Directly access Video devices</description>
+    <description xml:lang="tr">Görüntü aygıtlarına doğrudan eriş</description>
     <message>System policy prevents access to Video devices</message>
+    <message xml:lang="tr">Sistem politikası görüntü aygıtlarına erişimi engelliyor</message>
     <defaults>
       <allow_inactive>yes</allow_inactive>
       <allow_active>yes</allow_active>
@@ -201,7 +231,9 @@ NOTE: Please keep the actions in alpabet
 
   <action id="org.freedesktop.hal.device-access.video4linux">
     <description>Directly access video capture devices</description>
+    <description xml:lang="tr">Video yakalama aygıtlarına doğrudan eriş</description>
     <message>System policy prevents access to video capture devices</message>
+    <message xml:lang="tr">Sistem politikası video yakalama aygıtlarına erişimi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
Index: hal-0.5.14/policy/org.freedesktop.hal.dockstation.policy
===================================================================
--- hal-0.5.14.orig/policy/org.freedesktop.hal.dockstation.policy
+++ hal-0.5.14/policy/org.freedesktop.hal.dockstation.policy
@@ -7,7 +7,9 @@
 
  <action id="org.freedesktop.hal.dockstation.undock">
     <description>Undock a docking station</description>
+    <description xml:lang="tr">Yanaşma aygıtından ayrıl</description>
     <message>System policy prevents undocking</message>
+    <message xml:lang="tr">Sistem politikası yanaşma aygıtından ayrılmayı engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
Index: hal-0.5.14/policy/org.freedesktop.hal.killswitch.policy
===================================================================
--- hal-0.5.14.orig/policy/org.freedesktop.hal.killswitch.policy
+++ hal-0.5.14/policy/org.freedesktop.hal.killswitch.policy
@@ -18,7 +18,9 @@ details.
 
   <action id="org.freedesktop.hal.killswitch.bluetooth">
     <description>Turn Bluetooth radio On/Off</description>
+    <description xml:lang="tr">Bluetooth radyosunu aç/kapa</description>
     <message>System policy prevents turning the Bluetooth radio On/Off</message>
+    <message xml:lang="tr">Sistem politikası Bluetooth radyosunu açmayı/kapamayı engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -27,7 +29,9 @@ details.
 
   <action id="org.freedesktop.hal.killswitch.wlan">
     <description>Turn WLAN radio On/Off</description>
+    <description xml:lang="tr">Kablosuz yerel ağ radyosunu aç/kapa</description>
     <message>System policy prevents turning the WLAN radio On/Off</message>
+    <message xml:lang="tr">Sistem politikası kablosuz yerel ağ radyosunu açmayı/kapamayı engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -36,7 +40,9 @@ details.
 
   <action id="org.freedesktop.hal.killswitch.wwan">
     <description>Turn WWAN radio On/Off</description>
+    <description xml:lang="tr">Kablosuz geniş alan ağı radyosunu aç/kapa</description>
     <message>System policy prevents turning the WWAN radio On/Off</message>
+    <message xml:lang="tr">Sistem politikası kablosuz geniş alan ağı radyosunu açmayı/kapamayı engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
Index: hal-0.5.14/policy/org.freedesktop.hal.policy
===================================================================
--- hal-0.5.14.orig/policy/org.freedesktop.hal.policy
+++ hal-0.5.14/policy/org.freedesktop.hal.policy
@@ -18,7 +18,9 @@ details.
 
   <action id="org.freedesktop.hal.lock">
     <description>Acquire locks on devices</description>
+    <description xml:lang="tr">Aygıtları kilitle</description>
     <message>System policy prevents locking devices</message>
+    <message xml:lang="tr">Sistem politikası aygıtları kilitlemeyi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
Index: hal-0.5.14/policy/org.freedesktop.hal.power-management.policy
===================================================================
--- hal-0.5.14.orig/policy/org.freedesktop.hal.power-management.policy
+++ hal-0.5.14/policy/org.freedesktop.hal.power-management.policy
@@ -18,7 +18,9 @@ details.
 
   <action id="org.freedesktop.hal.power-management.shutdown">
     <description>Shut down the system</description>
+    <description xml:lang="tr">Sistemi kapat</description>
     <message>System policy prevents shutting down the system</message>
+    <message xml:lang="tr">Sistem politikası sistemi kapatmayı engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -27,7 +29,9 @@ details.
 
   <action id="org.freedesktop.hal.power-management.shutdown-multiple-sessions">
     <description>Shut down the system when multiple users are logged in</description>
+    <description xml:lang="tr">Birden fazla kullanıcının oturumu açıkken sistemi kapat</description>
     <message>System policy prevents shutting down the system while multiple users are logged in</message>
+    <message xml:lang="tr">Sistem politikası birden fazla kullanıcının oturumu açıkken sistemin kapatılmasını engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>auth_admin</allow_active>
@@ -36,7 +40,9 @@ details.
 
   <action id="org.freedesktop.hal.power-management.reboot">
     <description>Reboot the system</description>
+    <description xml:lang="tr">Sistemi yeniden başlat</description>
     <message>System policy prevents rebooting the system</message>
+    <message xml:lang="tr">Sistem politikası sistemi yeniden başlatmayı engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -45,7 +51,9 @@ details.
 
   <action id="org.freedesktop.hal.power-management.reboot-multiple-sessions">
     <description>Reboot the system when multiple users are logged in</description>
+    <description xml:lang="tr">Birden fazla kullanıcının oturumu açıkken sistemi yeniden başlat</description>
     <message>System policy prevents rebooting the system while multiple users are logged in</message>
+    <message xml:lang="tr">Sistem politikası birden fazla kullanıcının oturumu açıkken sistemi yeniden başlatmayı engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>auth_admin</allow_active>
@@ -53,8 +61,10 @@ details.
   </action>
 
   <action id="org.freedesktop.hal.power-management.set-powersave">
-    <description>Configure to system to prefer power savings</description>
+    <description>Configure the system to prefer power savings</description>
+    <description xml:lang="tr">Sistemi güç koruma kullanacak şekilde yapılandır</description>
     <message>System policy prevents enabling power savings</message>
+    <message xml:lang="tr">Sistem politikası güç korumayı açmayı engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -63,7 +73,9 @@ details.
 
   <action id="org.freedesktop.hal.power-management.suspend">
     <description>Suspend the system</description>
+    <description xml:lang="tr">Sistemi askıya al</description>
     <message>System policy prevents suspending the system</message>
+    <message xml:lang="tr">Sistem politikası sistemi askıya almayı engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -72,7 +84,9 @@ details.
 
   <action id="org.freedesktop.hal.power-management.hibernate">
     <description>Hibernate the system</description>
+    <description xml:lang="tr">Sistemi hazırda beklet</description>
     <message>System policy prevents hibernating the system</message>
+    <message xml:lang="tr">Sistem politikası sistemi hazırda bekletmeyi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -81,7 +95,9 @@ details.
 
   <action id="org.freedesktop.hal.power-management.cpufreq">
     <description>Configure CPU frequency scaling</description>
+    <description xml:lang="tr">MİB frekans ölçeklemeyi yapılandır</description>
     <message>System policy prevents CPU frequency scaling to be configured</message>
+    <message xml:lang="tr">Sistem politikası MİB frekans ölçeklemenin yapılandırılmasını engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -90,7 +106,9 @@ details.
 
   <action id="org.freedesktop.hal.power-management.lcd-panel">
     <description>Set laptop panel brightness</description>
+    <description xml:lang="tr">Dizüstü ekranı parlaklığını ayarla</description>
     <message>System policy prevents adjusting the laptop panel brightness</message>
+    <message xml:lang="tr">Sistem politikası dizüstü ekranının parlaklığını ayarlamayı engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -99,7 +117,9 @@ details.
 
   <action id="org.freedesktop.hal.power-management.light-sensor">
     <description>Detect ambient light using light sensor</description>
+    <description xml:lang="tr">Işık algılayıcısını kullanarak ortam ışığını algıla</description>
     <message>System policy prevents accessing the ambient light sensor</message>
+    <message xml:lang="tr">Sistem politikası ortam ışık algılayıcısına erişimi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -108,7 +128,9 @@ details.
 
   <action id="org.freedesktop.hal.power-management.keyboard-backlight">
     <description>Set keyboard backlight</description>
+    <description xml:lang="tr">Klavye arka ışığını ayarla</description>
     <message>System policy prevents adjusting the keyboard backlight</message>
+    <message xml:lang="tr">Sistem politikası klavye arka ışığını ayarlamayı engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
Index: hal-0.5.14/policy/org.freedesktop.hal.storage.policy
===================================================================
--- hal-0.5.14.orig/policy/org.freedesktop.hal.storage.policy
+++ hal-0.5.14/policy/org.freedesktop.hal.storage.policy
@@ -17,8 +17,10 @@ details.
 <policyconfig>
 
   <action id="org.freedesktop.hal.storage.mount-fixed">
-    <description>Mount file systems from internal drives.</description>
+    <description>Mount file systems from internal drives</description>
+    <description xml:lang="tr">Dahili sürücülerdeki dosya sistemlerini bağla</description>
     <message>System policy prevents mounting internal media</message>
+    <message xml:lang="tr">Sistem politikası dahili ortamın bağlanmasını engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>auth_admin_keep_always</allow_active>
@@ -26,8 +28,10 @@ details.
   </action>
 
   <action id="org.freedesktop.hal.storage.mount-removable">
-    <description>Mount file systems from removable drives.</description>
+    <description>Mount file systems from removable drives</description>
+    <description xml:lang="tr">Ayrılabilir sürücülerdeki dosya sistemlerini bağla</description>
     <message>System policy prevents mounting removable media</message>
+    <message xml:lang="tr">Sistem politikası ayrılabilir ortamların bağlanmasını engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -35,8 +39,10 @@ details.
   </action>
 
   <action id="org.freedesktop.hal.storage.unmount-others">
-    <description>Unmount file systems mounted by other users.</description>
+    <description>Unmount file systems mounted by other users</description>
+    <description xml:lang="tr">Diğer kullanıcılar tarafından bağlanan dosya sistemlerini ayır</description>
     <message>System policy prevents unmounting media mounted by other users</message>
+    <message xml:lang="tr">Sistem politikası başka kullanıcılar tarafından bağlanan ortamın ayrılmasını engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>auth_admin_keep_always</allow_active>
@@ -44,8 +50,10 @@ details.
   </action>
 
   <action id="org.freedesktop.hal.storage.eject">
-    <description>Eject removable media.</description>
+    <description>Eject removable media</description>
+    <description xml:lang="tr">Ayrılabilir ortamı çıkar</description>
     <message>System policy prevents ejecting removable media</message>
+    <message xml:lang="tr">Sistem politikası ayrılabilir ortamı çıkarmayı engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -53,8 +61,10 @@ details.
   </action>
 
   <action id="org.freedesktop.hal.storage.crypto-setup-fixed">
-    <description>Set up decryption for encrypted fixed storage devices.</description>
+    <description>Set up decryption for encrypted fixed storage devices</description>
+    <description xml:lang="tr">Şifrelenmiş sabit depolama aygıtlarının şifre çözümlemesini ayarla</description>
     <message>System policy prevents accessing encrypted fixed media</message>
+    <message xml:lang="tr">Sistem politikası şifrelenmiş sabit ortama erişimi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>auth_admin_keep_always</allow_active>
@@ -62,8 +72,10 @@ details.
   </action>
 
   <action id="org.freedesktop.hal.storage.crypto-setup-removable">
-    <description>Set up decryption for encrypted removable storage devices.</description>
+    <description>Set up decryption for encrypted removable storage devices</description>
+    <description xml:lang="tr">Şifrelenmiş ayrılabilir depolama aygıtlarının şifre çözümlemesini ayarla</description>
     <message>System policy prevents accessing encrypted removable media</message>
+    <message xml:lang="tr">Sistem politikası şifrelenmiş ayrılabilir ortama erişimi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
Index: hal-0.5.14/policy/org.freedesktop.hal.wol.policy
===================================================================
--- hal-0.5.14.orig/policy/org.freedesktop.hal.wol.policy
+++ hal-0.5.14/policy/org.freedesktop.hal.wol.policy
@@ -7,7 +7,9 @@
 
   <action id="org.freedesktop.hal.wol.enabled">
     <description>If Wake on LAN is enabled</description>
+    <description xml:lang="tr">Yerel ağdan uyanmanın açık olup olmadığı</description>
     <message>System policy prevents checking if Wake on LAN is enabled</message>
+    <message xml:lang="tr">Sistem politikası yerel ağdan uyanmanın açık olup olmadığını kontrol etmeyi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -16,7 +18,9 @@
 
   <action id="org.freedesktop.hal.wol.enable">
     <description>Enable or disable Wake on LAN</description>
+    <description xml:lang="tr">Yerel ağdan uyanmayı aç veya kapat</description>
     <message>System policy prevents enabling or disabling Wake on LAN</message>
+    <message xml:lang="tr">Sistem politikası yerel ağdan uyanmayı açmayı ve kapatmayı engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
@@ -25,7 +29,9 @@
 
   <action id="org.freedesktop.hal.wol.supported">
     <description>If Wake on LAN is supported</description>
+    <description xml:lang="tr">Yerel ağdan uyanmanın desteklenip desteklenmediği</description>
     <message>System policy prevents checking if Wake on LAN is supported</message>
+    <message xml:lang="tr">Sistem politikası yerel ağdan uyanmanın desteklenip desteklenmediğini kontrol etmeyi engelliyor</message>
     <defaults>
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
