diff --git a/hald/linux/osspec.c b/hald/linux/osspec.c
index e8bfa7b..cf2648b 100644
--- a/hald/linux/osspec.c
+++ b/hald/linux/osspec.c
@@ -625,7 +625,8 @@ set_suspend_hibernate_keys (HalDevice *d)
 	if (strstr (poweroptions, "mem"))
 		can_suspend = TRUE;
 	if (strstr (poweroptions, "disk"))
-		can_hibernate = TRUE;
+		if (strstr (hal_util_get_string_from_file("/proc", "cmdline"), "resume="))
+			can_hibernate = TRUE;
 
 	if (!can_suspend) {
 		pmtype = hal_device_property_get_string (d, "power_management.type");
