diff -upNr dbus-c++.orign/src/Makefile.am dbus-c++/src/Makefile.am
--- dbus-c++.orign/src/Makefile.am	2009-01-08 21:58:42.000000000 +0100
+++ dbus-c++/src/Makefile.am	2010-02-13 19:33:37.090109108 +0100
@@ -38,7 +38,7 @@ lib_include_HEADERS = $(HEADER_FILES)
 
 lib_LTLIBRARIES = libdbus-c++-1.la
 libdbus_c___1_la_SOURCES = $(HEADER_FILES) interface.cpp object.cpp introspection.cpp debug.cpp types.cpp connection.cpp connection_p.h property.cpp dispatcher.cpp dispatcher_p.h pendingcall.cpp pendingcall_p.h error.cpp internalerror.h message.cpp message_p.h server.cpp server_p.h eventloop.cpp eventloop-integration.cpp $(GLIB_CPP)
-libdbus_c___1_la_LIBADD = $(dbus_LIBS) $(glib_LIBS) $(pthread_LIBS)
+libdbus_c___1_la_LIBADD = $(dbus_LIBS) $(glib_LIBS) $(pthread_LIBS) $(xml_LIBS)
 
 MAINTAINERCLEANFILES = \
 	Makefile.in
