Index: meterbridge-0.9.2/src/Makefile.am
===================================================================
--- meterbridge-0.9.2.orig/src/Makefile.am
+++ meterbridge-0.9.2/src/Makefile.am
@@ -6,4 +6,4 @@ bin_PROGRAMS = meterbridge
 EXTRA_DIST = *.h
 
 meterbridge_SOURCES = main.c linedraw.c vu_meters.c ppm_meters.c dpm_meters.c jf_meters.c scope.c envelopes.c find_image.c
-meterbridge_LDFLAGS = -lm $(JACK_LIBS) $(SDL_LIBS) -lSDL_image
+meterbridge_LDADD = -lm $(JACK_LIBS) $(SDL_LIBS) -lSDL_image
