From a5f5d88a10e90c0dac44c8e845d8ce295a233bbd Mon Sep 17 00:00:00 2001
From: Jeffrey C. Ollie <jeff@ocjtech.us>
Date: Thu, 22 Oct 2009 23:46:22 -0500
Subject: [PATCH 1/2] Fix issues compiling with newer gnutls

---
 configure.ac |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 91e69e3..952bbe6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -45,7 +45,15 @@ AC_SEARCH_LIBS(recv,socket)
 AC_CHECK_FUNCS(getopt_long)
 AC_CHECK_FUNCS(getaddrinfo)
 
-AM_PATH_LIBGNUTLS(,AC_DEFINE(HAVE_GNUTLS,,"Use libgnutls"))
+dnl Check GNU TLS
+PKG_CHECK_MODULES(GNUTLS, gnutls >= 2.0.0, have_gnutls=yes, have_gnutls=no)
+if test "x$have_gnutls" = "xyes"; then
+  LIBGNUTLS_CFLAGS="$GNUTLS_CFLAGS"
+  LIBGNUTLS_LIBS="$GNUTLS_LIBS"
+  AC_SUBST(LIBGNUTLS_CFLAGS)
+  AC_SUBST(LIBGNUTLS_LIBS)
+  AC_DEFINE(HAVE_GNUTLS, 1, [whether to use GnuTSL support.])
+fi
 
 dnl Check -Wall flag of GCC
 if test "x$GCC" = "xyes"; then
-- 
1.6.5.rc2

