--- data/soundconverter.desktop.in.in.orig	2010-03-03 10:09:16.216198542 +0200
+++ data/soundconverter.desktop.in.in	2010-03-03 10:10:03.290198787 +0200
@@ -1,8 +1,10 @@
 [Desktop Entry]
 Version=1.0
-_Name=Sound Converter
+_Name=soundconverter
 _GenericName=Audio Format Converter
+GenericName[tr]=Ses Biçimi Çeviricisi
 _Comment=Convert audio files into other formats
+Comment[tr]=Ses dosyalarını diğer biçimlere çevirin
 Icon=soundconverter
 MimeType=application/x-ogg;application/ogg;audio/x-mp3;audio/x-scpls;audio/x-mp3;audio/x-mpeg;audio/mpeg;audio/x-mpegurl;application/x-flac;audio/x-wav;
 Exec=soundconverter %U
