diff -Nur remotes-old/q-sonic/lircd.qsonic.conf remotes/q-sonic/lircd.qsonic.conf
--- remotes-old/q-sonic/lircd.qsonic.conf	1970-01-01 02:00:00.000000000 +0200
+++ remotes/q-sonic/lircd.qsonic.conf	2005-08-08 08:43:56.000000000 +0300
@@ -0,0 +1,79 @@
+
+# Please make this file available to others
+# by sending it to <lirc@bartelmus.de>
+#
+# this config file was automatically generated
+# using lirc-0.7.1(any) on Sun Jul 31 00:27:28 2005
+#
+# contributed by Gwenole Beauchesne
+#
+# brand:                       Pearl Agency (www.pearl.de) / Q-Sonic Technologies
+# model no. of remote control: Q-Sonic Master Remote PC/TV (Universal RC),
+#                              Art. No.: PE-9999
+#                              Frequenz: 433.92 MHz
+# devices being controlled by this remote:
+# - programmed to IR Code 088 (Philips TV)
+# - programmed to IR Code 324 (Thomson VCR)
+
+begin remote
+
+  name  QSONIC
+  bits           40
+  eps            30
+  aeps          100
+
+  one             0     0
+  zero            0     0
+  gap          203985
+  toggle_bit      0
+
+
+      begin codes
+          POWER                    0x20EE11F00F
+          MOUSE_PAD_N              0x1447720000
+          MOUSE_PAD_NW             0x1449740000
+          MOUSE_PAD_W              0x1445700000
+          MOUSE_PAD_SW             0x144C770000
+          MOUSE_PAD_S              0x1448730000
+          MOUSE_PAD_SE             0x144B760000
+          MOUSE_PAD_E              0x1446710000
+          MOUSE_PAD_NE             0x144A750000
+          L                        0x144D780000
+          R                        0x14517C0000
+          VOL+                     0x20EE81600F
+          VOL-                     0x20EE01E00F
+          CH+                      0x20EE61400F
+          CH-                      0x20EEE1C00F
+          M                        0x20EEC1A00F
+          1                        0x20EEA1820D
+          2                        0x20EE61420D
+          3                        0x20EEE1C20D
+          4                        0x20EE41220D
+          5                        0x20EEC1A20D
+          6                        0x20EE81620D
+          7                        0x20EE01E20D
+          8                        0x20EE31120D
+          9                        0x20EEB1920D
+          0                        0x20EE21020D
+#         -/--                     0x20EE71520D ??? clobbers OK
+          AV                       0x20EED1BA05
+          MENU                     0x20EED1B609
+          EXIT                     0x20EEE1C906
+          UP                       0x20EEF1D50A
+          DOWN                     0x20EEF1D30C
+          LEFT                     0x20EEF1D20D
+          RIGHT                    0x20EEF1D10E
+          OK                       0x20EE71520D
+          REWIND                   0x20EE513807
+          PLAY                     0x20EED1B00F
+          FAST_FORWARD             0x20EED1B807
+          STOP                     0x20EE91700F
+          PAUSE                    0x20EE91720D
+          RECORD                   0x20EE91700F
+#         TELETEXT_ON              0x20EED1B609 ??? clobbers MENU
+#         TELETEXT_OFF             0x20EEE1C906 ??? clobbers EXIT
+          INFO                     0x20EE513A05
+          TELETEXT_HOLD            0x20EEF1D807
+      end codes
+
+end remote
diff -Nur remotes-old/q-sonic/lircmd.qsonic.conf remotes/q-sonic/lircmd.qsonic.conf
--- remotes-old/q-sonic/lircmd.qsonic.conf	1970-01-01 02:00:00.000000000 +0200
+++ remotes/q-sonic/lircmd.qsonic.conf	2005-08-08 08:43:56.000000000 +0300
@@ -0,0 +1,16 @@
+PROTOCOL INTELLIMOUSE
+
+# ACCELERATOR start max multiplier
+ACCELERATOR 2 30 5
+
+MOVE_N * MOUSE_PAD_N
+MOVE_NW * MOUSE_PAD_NW
+MOVE_W * MOUSE_PAD_W
+MOVE_SW * MOUSE_PAD_SW
+MOVE_S * MOUSE_PAD_S
+MOVE_SE * MOUSE_PAD_SE
+MOVE_E * MOUSE_PAD_E
+MOVE_NE * MOUSE_PAD_NE
+
+BUTTON1_CLICK * L
+BUTTON3_CLICK * R
