diff -ur abiword-2.9.3.orig/src/text/ptbl/xp/pd_DocumentRDF.cpp abiword-2.9.3/src/text/ptbl/xp/pd_DocumentRDF.cpp
--- abiword-2.9.3.orig/src/text/ptbl/xp/pd_DocumentRDF.cpp	2012-10-29 02:09:32.000000000 -0400
+++ abiword-2.9.3/src/text/ptbl/xp/pd_DocumentRDF.cpp	2012-10-29 02:10:23.000000000 -0400
@@ -1884,7 +1884,7 @@
 
 #ifdef WITH_EVOLUTION_DATA_SERVER
 extern "C" {
-  #include <libebook/e-book.h>
+  #include <libebook/libebook.h>
 };
 
 static std::string get( EVCard* c, const char* v )
