diff -Nur abe-1.1-old/configure.in abe-1.1/configure.in
--- abe-1.1-old/configure.in	2006-10-01 03:08:22.866256583 +0300
+++ abe-1.1/configure.in	2006-10-01 03:08:45.307844943 +0300
@@ -52,8 +52,6 @@
 dnl General options
 dnl --------------------------------------------------------------------------
 TR_CPPFLAGS=""
-TR_CFLAGS="-O2 -Wall -fomit-frame-pointer -ffast-math -fexpensive-optimizations"
-TR_CXXFLAGS="-O2 -Wall -fomit-frame-pointer -ffast-math -fexpensive-optimizations"
 TR_LIBS=""
 
 case "$host" in
