--- pccts/antlr/fset2.c.str	2009-04-07 09:43:21.000000000 +0200
+++ pccts/antlr/fset2.c	2009-04-07 09:43:38.000000000 +0200
@@ -2210,7 +2210,7 @@
       if (p->ntype != nToken) continue;
       tn=(TokNode *)p;
       if (depth != 0) fprintf(stdout," ");
-      fprintf(stdout,TerminalString(tn->token));
+      fprintf(stdout,"%s",TerminalString(tn->token));
       depth++;
       if (! MR_AmbAidMultiple) {
         if (set_nil(tn->tset)) {
--- pccts/antlr/gen.c.str	2009-04-07 09:42:11.000000000 +0200
+++ pccts/antlr/gen.c	2009-04-07 09:43:04.000000000 +0200
@@ -159,7 +159,7 @@
 #define gen6(s,a,b,c,d,e,f)	{tab(); fprintf(output, s,a,b,c,d,e,f);}
 #define gen7(s,a,b,c,d,e,f,g)	{tab(); fprintf(output, s,a,b,c,d,e,f,g);}
 
-#define _gen(s)			{fprintf(output, s);}
+#define _gen(s)			{fprintf(output, "%s", s);}
 #define _gen1(s,a)		{fprintf(output, s,a);}
 #define _gen2(s,a,b)	{fprintf(output, s,a,b);}
 #define _gen3(s,a,b,c)	{fprintf(output, s,a,b,c);}
--- pccts/antlr/lex.c.str	2009-04-07 09:41:11.000000000 +0200
+++ pccts/antlr/lex.c	2009-04-07 09:42:01.000000000 +0200
@@ -706,7 +706,7 @@
 /* MR26 */			if (! (isalpha(*t) || isdigit(*t) || *t == '_' || *t == '$')) break;
 /* MR26 */		}
 /* MR26 */	}
-/* MR26 */	fprintf(output,strBetween(pSymbol, t, pSeparator));
+/* MR26 */	fprintf(output,"%s",strBetween(pSymbol, t, pSeparator));
 
     *q = p;
     return (*pSeparator  == 0);
@@ -771,7 +771,7 @@
 				  &pValue,
 				  &pSeparator,
 				  &nest);
-	fprintf(f,strBetween(pDataType, pSymbol, pSeparator));
+	fprintf(f,"%s",strBetween(pDataType, pSymbol, pSeparator));
 }
 
 /* check to see if string e is a word in string s */
@@ -852,9 +852,9 @@
 					  &pSeparator,
 					  &nest);
 		fprintf(f,"\t");
-		fprintf(f,strBetween(pDataType, pSymbol, pSeparator));
+		fprintf(f,"%s",strBetween(pDataType, pSymbol, pSeparator));
 		fprintf(f," ");
-		fprintf(f,strBetween(pSymbol, pEqualSign, pSeparator));
+		fprintf(f,"%s",strBetween(pSymbol, pEqualSign, pSeparator));
 		fprintf(f,";\n");
     }
 	fprintf(f,"};\n");
