--- xf86-video-ati-6.14.6/configure.ac.bero	2012-07-04 01:33:29.031728695 +0200
+++ xf86-video-ati-6.14.6/configure.ac	2012-07-04 01:36:40.524729637 +0200
@@ -149,16 +149,20 @@ if test "$DRI" = yes; then
 				AC_DEFINE(RADEON_DRI2, 1,[Enable DRI2 code])
                         else
                                 DRM_MODE=no
+				AC_MSG_ERROR([Correct version of DRI2 not found, refusing to build])
 			fi
 			PKG_CHECK_MODULES(LIBUDEV, [libudev], [LIBUDEV=yes], [LIBUDEV=no])
 			if test "x$LIBUDEV" = xyes; then
 			       AC_DEFINE(HAVE_LIBUDEV, 1,[libudev support])
 			fi
+		else
+			AC_MSG_ERROR([DRM mode not set, refusing to build])
         	fi
 	fi
 	CFLAGS="$save_CFLAGS"
 else
 	DRM_MODE=no
+	AC_MSG_ERROR([DRI disabled, refusing to build])
 fi
 AM_CONDITIONAL(DRM_MODE, test x$DRM_MODE = xyes)
 AM_CONDITIONAL(LIBUDEV, test x$LIBUDEV = xyes)
