--- libnl-1.0-pre5/lib/Makefile.debuginfo	2006-06-15 18:11:21.000000000 -0400
+++ libnl-1.0-pre5/lib/Makefile	2006-06-15 18:11:28.000000000 -0400
@@ -64,7 +64,7 @@
 
 install:
 	mkdir -p $(DESTDIR)$(libdir)/
-	install -m 0644 $(OUT_SLIB) $(DESTDIR)$(libdir)
+	install -m 0755 $(OUT_SLIB) $(DESTDIR)$(libdir)
 	rm -f $(DESTDIR)$(libdir)/$(LN1_SLIB)
 	$(LN) -s $(OUT_SLIB) $(DESTDIR)$(libdir)/$(LN1_SLIB)
 	rm -f $(DESTDIR)$(libdir)/$(LN_SLIB)
 	$(LN) -s $(LN1_SLIB) $(DESTDIR)$(libdir)/$(LN_SLIB)

