--- stream/http.c~	2008-04-19 21:59:18.000000000 +0200
+++ stream/http.c	2008-04-19 21:54:02.000000000 +0200
@@ -822,6 +822,36 @@
				content_type = http_get_field( http_hdr, "Content-Type" );
				if( content_type!=NULL ) {
					mp_msg(MSGT_NETWORK,MSGL_V,"Content-Type: [%s]\n", content_type );
+					// Check for ASF redirection
+					if (!strcasecmp(content_type, "audio/x-ms-wax") || !strcasecmp(content_type, "audio/x-ms-wma") || !strcasecmp(content_type, "video/x-ms-asf") || !strcasecmp(content_type, "video/x-ms-afs") || !strcasecmp(content_type, "video/x-ms-wmv")) {
+						if (http_hdr->body_size > 7) {
+							if (!strncasecmp(http_hdr->body, "http://", 7) || !strncasecmp(http_hdr->body, "mms://", 6)) {
+								char* newurl;
+								char* p;
+								newurl = malloc(http_hdr->body_size + 1);
+								strcpy(newurl, http_hdr->body);
+								p = newurl + strlen(newurl) - 1;
+								while (p > newurl && (*p == ' ' || *p == '\n' || *p == '\r')) {
+									*p = 0;
+									p--;
+								}
+								url = url_new(newurl);
+								stream->streaming_ctrl->url = url;
+								if (!strcasecmp(url->protocol, "mms")) {
+									res = STREAM_REDIRECTED;
+									goto err_out;
+								} else {
+									redirect = 1;
+									break;
+								}
+							// Check for ASX playlist
+							} else if (!strncasecmp(http_hdr->body, "<asx", 4)) {
+								*file_format = DEMUXER_TYPE_PLAYLIST;
+								res = seekable;
+								goto out;
+							}
+						}
+					}
 					// Check in the mime type table for a demuxer type
 					i = 0;
 					while(mime_type_table[i].mime_type != NULL) {



