diff -Nur mplayer-20080625-old/stream/asf_mmst_streaming.c mplayer-20080625/stream/asf_mmst_streaming.c
--- mplayer-20080625-old/stream/asf_mmst_streaming.c	2008-06-25 23:16:38.975676726 +0300
+++ mplayer-20080625/stream/asf_mmst_streaming.c	2008-06-25 23:19:23.817407449 +0300
@@ -413,8 +413,10 @@
       }
       printf("\n");
 #else
-      mp_msg(MSGT_NETWORK,MSGL_WARN,MSGTR_MPDEMUX_MMST_UnknownObject);
+      /*mp_msg(MSGT_NETWORK,MSGL_WARN,MSGTR_MPDEMUX_MMST_UnknownObject);*/
 #endif
+      /* Ignore length to search for more guids that would not be found otherwise */
+      length = 1;
     }
 
 //    mp_msg(MSGT_NETWORK,MSGL_INFO,"length    : %lld\n", length);
