diff -Nur mplayer-20091228-old/vidix/pci_db2c.awk mplayer-20091228/vidix/pci_db2c.awk
--- mplayer-20091228-old/vidix/pci_db2c.awk	2009-12-29 11:41:24.659409458 +0200
+++ mplayer-20091228/vidix/pci_db2c.awk	2009-12-29 11:41:48.825159324 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/awk -f
+#!/bin/awk -f
 # This file converts given pci.db to "C" source and header files
 # For latest version of pci ids see: http://pciids.sf.net
 # Copyright 2002 Nick Kurshev
