diff -Nur mplayer-20081228-old/mplayer.c mplayer-20081228/mplayer.c
--- mplayer-20081228-old/mplayer.c	2008-12-31 17:01:38.000000000 +0200
+++ mplayer-20081228/mplayer.c	2008-12-31 17:00:29.000000000 +0200
@@ -30,6 +30,7 @@
 #include <string.h>
 #include <time.h>
 #include <unistd.h>
+#include <ctype.h>
 #include <sys/stat.h>
 #include <sys/time.h>
 #include <sys/types.h>
@@ -896,6 +897,8 @@
 static void load_per_extension_config (m_config_t* conf, const char *const file)
 {
     char *str;
+    char strtmp[8];
+    int i;
     char extension[strlen (PROFILE_CFG_EXTENSION) + 8];
     m_profile_t *p;
 
@@ -904,8 +907,12 @@
     if (!str)
         return;
 
+    strncpy(strtmp, str + 1, 7);
+    for(i = 0; i < 8; i++)
+        strtmp[i] = tolower(strtmp[i]);
+
     sprintf (extension, PROFILE_CFG_EXTENSION);
-    strncat (extension, ++str, 7);
+    strncat (extension, strtmp, 7);
     p = m_config_get_profile (conf, extension);
     if (p)
     {
