diff -Nur mplayer-20071130-old/gui/cfg.c mplayer-20071130/gui/cfg.c
--- mplayer-20071130-old/gui/cfg.c	2007-12-01 02:35:39.035216766 +0200
+++ mplayer-20071130/gui/cfg.c	2007-12-01 02:36:04.945277840 +0200
@@ -53,8 +53,8 @@
 char * gtkAOESDDevice;
 #endif
 
-int    gtkCacheOn = 0;
-int    gtkCacheSize = 2048;
+int    gtkCacheOn = 1;
+int    gtkCacheSize = 8192;
 
 int    gtkAutoSyncOn = 0;
 int    gtkAutoSync = 0;
