diff -Nur mplayer-old/libmpdemux/demux_mkv.c mplayer/libmpdemux/demux_mkv.c
--- mplayer-old/libmpdemux/demux_mkv.c	2010-06-01 14:35:58.644548262 +0300
+++ mplayer/libmpdemux/demux_mkv.c	2010-06-01 14:36:40.301548529 +0300
@@ -2831,6 +2831,9 @@
             int seek_id = (demuxer->video->id < 0) ?
                 demuxer->audio->id : demuxer->video->id;
 
+            if (target_timecode == 0)
+                index = mkv_d->indexes;
+
             /* let's find the entry in the indexes with the smallest */
             /* difference to the wanted timecode. */
             for (i = 0; i < mkv_d->num_indexes; i++)
