diff -Nur mplayer-20071130-old/gui/interface.c mplayer-20071130/gui/interface.c
--- mplayer-20071130-old/gui/interface.c	2007-12-01 01:27:23.781789919 +0200
+++ mplayer-20071130/gui/interface.c	2007-12-01 01:47:21.158760951 +0200
@@ -171,9 +171,6 @@
 #ifdef HAVE_DXR3
  if ( !gtkDXR3Device ) gtkDXR3Device=strdup( "/dev/em8300-0" );
 #endif
- if ( stream_cache_size > 0 ) { gtkCacheOn=1; gtkCacheSize=stream_cache_size; }
- else if ( stream_cache_size == 0 ) gtkCacheOn = 0;
- if ( autosync && autosync != gtkAutoSync ) { gtkAutoSyncOn=1; gtkAutoSync=autosync; }

 #ifdef USE_ASS
  gtkASS.enabled = ass_enabled;
