diff -Nur mplayer-20071230-old/etc/dvb-menu.conf mplayer-20071230/etc/dvb-menu.conf
--- mplayer-20071230-old/etc/dvb-menu.conf	2007-12-30 16:58:50.356817725 +0200
+++ mplayer-20071230/etc/dvb-menu.conf	2007-12-30 17:03:38.231054189 +0200
@@ -2,7 +2,7 @@
 <txt name="man" file="manpage.txt"/>
 
 <filesel name="open_file"/>
-<filesel name="open_list" file-action="loadlist '%p'" auto-close="yes"
+<filesel name="open_list" file-action="loadlist '%p'" post-cmd="yes"
 	 title="Open a playlist: %p" />
 
 <dvbsel title="Lista canali" name="dvbsel" />
