diff -Nur mplayer-old/libmpdemux/demux_mkv.c mplayer/libmpdemux/demux_mkv.c
--- mplayer-old/libmpdemux/demux_mkv.c	2010-06-01 14:37:23.645798942 +0300
+++ mplayer/libmpdemux/demux_mkv.c	2010-06-01 14:40:45.217673514 +0300
@@ -1651,6 +1651,8 @@
             track->a_formattag = 0x0055;
         else if (!strncmp(track->codec_id, MKV_A_AC3, strlen(MKV_A_AC3)))
             track->a_formattag = 0x2000;
+        else if (!strncmp(track->codec_id, MKV_A_EAC3, strlen(MKV_A_EAC3)))
+            track->a_formattag = 0x2002;
         else if (!strcmp(track->codec_id, MKV_A_DTS))
             track->a_formattag = 0x2001;
         else if (!strcmp(track->codec_id, MKV_A_PCM)
@@ -1722,6 +1724,7 @@
         sh_a->wf->nAvgBytesPerSec = 16000;
         sh_a->wf->nBlockAlign = 1152;
     } else if ((track->a_formattag == 0x2000) ||        /* AC3 */
+               (track->a_formattag == 0x2002) ||        /* EAC3 */
                (track->a_formattag == 0x2001)) {        /* DTS */
         free(sh_a->wf);
         sh_a->wf = NULL;
diff -Nur mplayer-old/libmpdemux/matroska.h mplayer/libmpdemux/matroska.h
--- mplayer-old/libmpdemux/matroska.h	2010-06-01 14:35:58.645548283 +0300
+++ mplayer/libmpdemux/matroska.h	2010-06-01 14:39:05.623923673 +0300
@@ -34,6 +34,7 @@
 #define MKV_A_AAC_4LTP   "A_AAC/MPEG4/LTP"
 #define MKV_A_AAC        "A_AAC"
 #define MKV_A_AC3        "A_AC3"
+#define MKV_A_EAC3       "A_EAC3"
 #define MKV_A_DTS        "A_DTS"
 #define MKV_A_MP2        "A_MPEG/L2"
 #define MKV_A_MP3        "A_MPEG/L3"
