diff -uNr MPlayer-r27054.orig/stream/asf_streaming.c MPlayer-r27054/stream/asf_streaming.c
--- MPlayer-r27054.orig/stream/asf_streaming.c	2008-04-09 03:32:35.000000000 +0300
+++ MPlayer-r27054/stream/asf_streaming.c	2008-06-24 17:46:42.000000000 +0300
@@ -731,6 +731,11 @@
 			}
 			http_response_append( http_hdr, buffer, i );
 		} while( !http_is_header_entire( http_hdr ) );
+		// Also receive first few bytes of the body 
+		// because asf_http_parse_response might need that
+		i = recv( fd, buffer, BUFFER_SIZE, 0 );
+		if ( i>0 )
+			http_response_append( http_hdr, buffer, i );
 		if( mp_msg_test(MSGT_NETWORK,MSGL_V) ) {
 			http_hdr->buffer[http_hdr->buffer_size]='\0';
 			mp_msg(MSGT_NETWORK,MSGL_DBG2,"Response [%s]\n", http_hdr->buffer );


