diff -Nur mplayer-20090910-old/mp_core.h mplayer-20090910/mp_core.h
--- mplayer-20090910-old/mp_core.h	2009-09-11 11:59:47.062372562 +0300
+++ mplayer-20090910/mp_core.h	2009-09-11 12:01:08.692372668 +0300
@@ -24,6 +24,7 @@
 #define INITIALIZED_DEMUXER 512
 #define INITIALIZED_ACODEC  1024
 #define INITIALIZED_VCODEC  2048
+#define INITIALIZED_SAVED_VOL 4096
 #define INITIALIZED_ALL     0xFFFF
 
 
@@ -111,6 +112,8 @@
     unsigned char *nav_start;    ///< pointer to last read video buffer
     int            nav_in_size;  ///< last read size
 #endif
+    float vol_l;
+    float vol_r;
 } MPContext;
 
 
diff -Nur mplayer-20090910-old/mplayer.c mplayer-20090910/mplayer.c
--- mplayer-20090910-old/mplayer.c	2009-09-11 11:59:47.055372896 +0300
+++ mplayer-20090910/mplayer.c	2009-09-11 12:07:57.571376257 +0300
@@ -188,6 +188,8 @@
 #ifdef CONFIG_DVBIN
     .last_dvb_step = 1,
 #endif
+    .vol_l = -1.,
+    .vol_r = -1.,
 };
 
 static MPContext *mpctx = &mpctx_s;
@@ -580,6 +582,11 @@
     mpctx->mixer.afilter = NULL;
   }
 
+  if(mask&INITIALIZED_SAVED_VOL){
+    mpctx->vol_l = -1;
+    mpctx->vol_r = -1;
+  }
+
   if(mask&INITIALIZED_VCODEC){
     initialized_flags&=~INITIALIZED_VCODEC;
     current_module="uninit_vcodec";
@@ -3659,6 +3666,9 @@
     end_at.type = END_AT_NONE;
 }
 
+if (mpctx->vol_l >= 0. && mpctx->vol_r >= 0.)
+  mixer_setvolume(&mpctx->mixer, mpctx->vol_l, mpctx->vol_r);
+
 #ifdef CONFIG_DVDNAV
 mp_dvdnav_context_free(mpctx);
 if (mpctx->stream->type == STREAMTYPE_DVDNAV) {
@@ -3926,6 +3936,9 @@
 #endif
 }
 
+if (mpctx->sh_audio)
+  mixer_getvolume(&mpctx->mixer, &mpctx->vol_l, &mpctx->vol_r);
+
 goto_next_file:  // don't jump here after ao/vo/getch initialization!
 
 mp_msg(MSGT_CPLAYER,MSGL_INFO,"\n");
