diff -Nur mplayer-20060923-old/libmenu/menu_list.h mplayer-20060923/libmenu/menu_list.h
--- mplayer-20060923-old/libmenu/menu_list.h	2006-09-24 19:24:24.533181016 +0300
+++ mplayer-20060923/libmenu/menu_list.h	2006-09-24 19:42:30.685060832 +0300
@@ -56,9 +56,9 @@
   0,0, \
   5, 3, \
   NULL, \
-  0xFF, 0xFF, \
-  0xFF, 0xFF, \
-  0xA4, 0x50 \
+  0xF0, 0xE0, \
+  0xFF, 0xFF, \
+  0xE4, 0x40 \
 }
   
 
