diff -Nur mplayer-20080509-old/help/help_mp-tr.h mplayer-20080509/help/help_mp-tr.h
--- mplayer-20080509-old/help/help_mp-tr.h	2008-05-09 14:43:08.060421295 +0300
+++ mplayer-20080509/help/help_mp-tr.h	2008-05-09 14:44:00.438421736 +0300
@@ -2168,3 +2168,10 @@
 #define MSGTR_TVI_DS_NoVideoCaptureDevice "tvi_dshow: Görüntü yakalama aygıtı bulunamadı.\n"
 #define MSGTR_TVI_DS_NoAudioCaptureDevice "tvi_dshow: Ses yakalama aygıtı bulunamadı.\n"
 #define MSGTR_TVI_DS_GetActualMediatypeFailed "tvi_dshow: Asıl ortam tipi alınamadı (Hata:0x%x). İstek yapılana eşit olduğu varsayılıyor.\n"
+
+#define MSGTR_LIBMENU_Back "Geri"
+#define MSGTR_LIBMENU_Jump_to "Listeden seç"
+#define MSGTR_LIBMENU_Select_a_file "Dosya yükle: %p"
+#define MSGTR_LIBMENU_Loadfile "yükle '%p'"
+#define MSGTR_LIBMENU_Select_chapter "Bölüm seç               "
+
diff -Nur mplayer-20080509-old/libmenu/menu_chapsel.c mplayer-20080509/libmenu/menu_chapsel.c
--- mplayer-20080509-old/libmenu/menu_chapsel.c	2008-05-09 14:43:08.002421287 +0300
+++ mplayer-20080509/libmenu/menu_chapsel.c	2008-05-09 14:43:18.035421540 +0300
@@ -26,6 +26,8 @@
 #include <string.h>
 
 #include "config.h"
+#include "mp_msg.h"
+#include "help_mp.h"
 
 #include "m_struct.h"
 #include "m_option.h"
@@ -54,7 +56,7 @@
 
 static struct menu_priv_s cfg_dflt = {
   MENU_LIST_PRIV_DFLT,
-  "Select chapter",
+  MSGTR_LIBMENU_Select_chapter,
   NULL,
   "${chapter_name}  [${start}]"
 };
diff -Nur mplayer-20080509-old/libmenu/menu_filesel.c mplayer-20080509/libmenu/menu_filesel.c
--- mplayer-20080509-old/libmenu/menu_filesel.c	2008-05-09 14:43:08.002421287 +0300
+++ mplayer-20080509/libmenu/menu_filesel.c	2008-05-09 14:43:18.035421540 +0300
@@ -55,8 +55,8 @@
   NULL,
 
   NULL,
-  "Select a file: %p",
-  "loadfile '%p'",
+  MSGTR_LIBMENU_Select_a_file,
+  MSGTR_LIBMENU_Loadfile,
   NULL,
   NULL,
   NULL
diff -Nur mplayer-20080509-old/libmenu/menu_param.c mplayer-20080509/libmenu/menu_param.c
--- mplayer-20080509-old/libmenu/menu_param.c	2008-05-09 14:43:08.003420894 +0300
+++ mplayer-20080509/libmenu/menu_param.c	2008-05-09 14:43:18.035421540 +0300
@@ -108,7 +108,7 @@
       if(!m)
         mp_msg(MSGT_OSD_MENU,MSGL_WARN,MSGTR_LIBMENU_NoEntryFoundInTheMenuDefinition);
       m = calloc(1,sizeof(struct list_entry_s));
-      m->p.txt = strdup("Back");
+      m->p.txt = strdup(MSGTR_LIBMENU_Back);
       menu_list_add_entry(menu,m);
       return 1;
     }
diff -Nur mplayer-20080509-old/libmenu/menu_pt.c mplayer-20080509/libmenu/menu_pt.c
--- mplayer-20080509-old/libmenu/menu_pt.c	2008-05-09 14:43:08.002421287 +0300
+++ mplayer-20080509/libmenu/menu_pt.c	2008-05-09 14:43:18.035421540 +0300
@@ -37,7 +37,7 @@
 
 static struct menu_priv_s cfg_dflt = {
   MENU_LIST_PRIV_DFLT,
-  "Jump to",
+  MSGTR_LIBMENU_Jump_to,
   NULL,
   0
 };
