diff -Nur mplayer-old/configure mplayer/configure
--- mplayer-old/configure	2010-05-25 16:15:03.298418162 +0300
+++ mplayer/configure	2010-05-25 16:16:06.358293513 +0300
@@ -5496,7 +5496,7 @@
   if test "$_vaapi_old" = no ; then
     def_vaapi_old='#define CONFIG_VAAPI_OLD 0'
     libs_mencoder="$libs_mencoder -lva"
-    libs_mplayer="$libs_mplayer -lva-x11"
+    libs_mplayer="$libs_mplayer -lva -lva-x11"
   else
     def_vaapi_old='#define CONFIG_VAAPI_OLD 1'
     _mencoder="no"
