diff -Nur mplayer-old/mplayer.c mplayer/mplayer.c
--- mplayer-old/mplayer.c	2010-05-27 21:16:03.073258539 +0300
+++ mplayer/mplayer.c	2010-05-27 21:15:33.039297492 +0300
@@ -2752,6 +2752,10 @@
       opt_exit = 1;
     }
 
+    /* vaapi is too slow with menu on */
+    if(video_driver_list && strcmp(video_driver_list[0],"vaapi")==0)
+      use_menu = 0;
+
     if(audio_driver_list && strcmp(audio_driver_list[0],"help")==0){
       list_audio_out();
       opt_exit = 1;
