diff -Nur mplayer-20090429-old/libavcodec/dsputil.c mplayer-20090429/libavcodec/dsputil.c
--- mplayer-20090429-old/libavcodec/dsputil.c	2009-04-29 18:55:34.000000000 +0300
+++ mplayer-20090429/libavcodec/dsputil.c	2009-04-29 18:57:21.000000000 +0300
@@ -4300,7 +4300,7 @@
 
     if((intptr_t)&aligned & 15){
         if(!did_fail){
-#if HAVE_MMX || HAVE_ALTIVEC
+#if 0
             av_log(NULL, AV_LOG_ERROR,
                 "Compiler did not align stack variables. Libavcodec has been miscompiled\n"
                 "and may be very slow or crash. This is not a bug in libavcodec,\n"
