diff -Nur mplayer-20100210-old/ffmpeg/libavcodec/mpeg4videodec.c mplayer-20100210/ffmpeg/libavcodec/mpeg4videodec.c
--- mplayer-20100210-old/ffmpeg/libavcodec/mpeg4videodec.c	2010-02-11 01:36:53.087692337 +0200
+++ mplayer-20100210/ffmpeg/libavcodec/mpeg4videodec.c	2010-02-11 01:39:21.580695682 +0200
@@ -2119,7 +2119,6 @@
     for(;;) {
         if(get_bits_count(gb) >= gb->size_in_bits){
             if(gb->size_in_bits==8 && (s->divx_version>=0 || s->xvid_build>=0)){
-                av_log(s->avctx, AV_LOG_ERROR, "frame skip %d\n", gb->size_in_bits);
                 return FRAME_SKIPPED; //divx bug
             }else
                 return -1; //end of stream
