diff -Nur mplayer-20100114-old/ffmpeg/libavcodec/mpeg4videodec.c mplayer-20100114/ffmpeg/libavcodec/mpeg4videodec.c
--- mplayer-20100114-old/ffmpeg/libavcodec/mpeg4videodec.c	2010-01-14 18:44:53.028930284 +0200
+++ mplayer-20100114/ffmpeg/libavcodec/mpeg4videodec.c	2010-01-14 18:45:55.229814312 +0200
@@ -1851,7 +1851,7 @@
         s->divx_build= build;
         s->divx_packed= e==3 && last=='p';
         if(s->divx_packed && !s->showed_packed_warning) {
-            av_log(s->avctx, AV_LOG_WARNING, "Invalid and inefficient vfw-avi packed B frames detected\n");
+            //av_log(s->avctx, AV_LOG_WARNING, "Invalid and inefficient vfw-avi packed B frames detected\n");
             s->showed_packed_warning=1;
         }
     }
