Index: mp_msg.c
===================================================================
--- mp_msg.c	(revision 28082)
+++ mp_msg.c	(working copy)
@@ -104,6 +104,12 @@
         fprintf(stream, "\033[%d;3%dm", c >> 3, c & 7);
 }
 
+static void reset_msg_color(FILE* stream)
+{
+    if (mp_msg_color)
+        fprintf(stream, "\033[0m");
+}
+
 static void print_msg_module(FILE* stream, int mod)
 {
     static const char *module_text[MSGT_MAX] = {
@@ -221,5 +227,6 @@
     header = tmp[strlen(tmp)-1] == '\n' || tmp[strlen(tmp)-1] == '\r';
  
     fprintf(stream, "%s", tmp);
+    reset_msg_color(stream);
     fflush(stream);
 }
