diff -Nur mplayer-old/libvo/vo_vaapi.c mplayer/libvo/vo_vaapi.c
--- mplayer-old/libvo/vo_vaapi.c	2010-11-03 11:50:20.491526114 +0200
+++ mplayer/libvo/vo_vaapi.c	2010-11-03 11:46:14.034517838 +0200
@@ -27,13 +27,14 @@
 #include "video_out.h"
 #include "video_out_internal.h"
 #include "fastmemcpy.h"
-#include "sub.h"
+#include "sub/sub.h"
+#include "sub/eosd.h"
 #include "x11_common.h"
 #include "libavutil/common.h"
 #include "libavcodec/vaapi.h"
 #include "gui/interface.h"
 #include "stats.h"
-#include "ass_mp.h"
+#include "sub/ass_mp.h"
 #include <stdarg.h>
 
 #if CONFIG_GL
@@ -2351,10 +2352,10 @@
     enable_osd();
 }
 
-static void draw_eosd(mp_eosd_images_t *imgs)
+static void draw_eosd(struct mp_eosd_image_list *imgs)
 {
-    ass_image_t *img = imgs->imgs;
-    ass_image_t *i;
+    struct mp_eosd_image *img = eosd_image_first(imgs);
+    struct mp_eosd_image *i;
     VAStatus status;
 
     if (!va_eosd_draw_alpha)
@@ -2380,7 +2381,7 @@
 
     memset(va_eosd_image_data, 0, va_eosd_image.data_size);
 
-    for (i = img; i; i = i->next)
+    for (i = img; i; i = eosd_image_next(imgs))
         va_eosd_draw_alpha(i->bitmap, i->w, i->h, i->stride,
                            i->dst_x, i->dst_y, i->color);
 
@@ -2658,7 +2659,7 @@
         draw_eosd(data);
         return VO_TRUE;
     case VOCTRL_GET_EOSD_RES: {
-        mp_eosd_res_t *r = data;
+        struct mp_eosd_settings *r = data;
         r->mt = r->mb = r->ml = r->mr = 0;
         r->srcw = g_image_width;
         r->srch = g_image_height;
