diff -Nur configure.orig mplayer/configure
--- configure.orig	2010-07-16 11:30:09.098081470 +0300
+++ configure	2010-07-16 11:30:56.785055835 +0300
@@ -7744,7 +7744,7 @@
   cat > $TMPC << EOF
 #include <inttypes.h>
 #include <x264.h>
-#if X264_BUILD < 98
+#if X264_BUILD < 90
 #error We do not support old versions of x264. Get the latest from git.
 #endif
 int main(void) { x264_encoder_open((void*)0); return 0; }
Index: ffmpeg/libavcodec/libx264.c
===================================================================
--- ffmpeg/libavcodec/libx264.c	(revision 23430)
+++ ffmpeg/libavcodec/libx264.c	(revision 23429)
@@ -89,7 +89,6 @@
     int nnal, i;
     x264_picture_t pic_out;
 
-    x264_picture_init( &x4->pic );
     x4->pic.img.i_csp   = X264_CSP_I420;
     x4->pic.img.i_plane = 3;
 
