--- lxc-lxc-1.0.0.beta2/configure.ac~	2014-01-16 02:07:33.000000000 +0200
+++ lxc-lxc-1.0.0.beta2/configure.ac	2014-02-13 10:09:36.273870378 +0200
@@ -136,7 +136,7 @@
 
 if test "x$enable_doc" = "xyes" -o "x$enable_doc" = "xauto"; then
 	db2xman=""
-	dbparsers="docbook2x-man db2x_docbook2man docbook2man docbook-to-man"
+	dbparsers="docbook2x docbook2x-man db2x_docbook2man docbook2man docbook-to-man"
 
 	AC_MSG_CHECKING(for docbook2x-man)
 	for name in ${dbparsers}; do

