Index: lxc-0.7.2/lxc-management-tools/lxc-shutdown-agent
===================================================================
--- lxc-0.7.2.orig/lxc-management-tools/lxc-shutdown-agent
+++ lxc-0.7.2/lxc-management-tools/lxc-shutdown-agent
@@ -21,7 +21,7 @@
 
 LXC_ETC=/etc/lxc
 LXC_LIB=/var/lib/lxc
-CGROUP_MOUNT=/cgroup
+CGROUP_MOUNT=/sys/fs/cgroup
 LXC_CONF=${LXC_ETC}/lxc.conf
 [[ -s $LXC_CONF ]] && . $LXC_CONF
 
