--- ncftp-3.1.5/sh_util/gpshare.c.pmeter	Thu Dec  5 19:04:42 2002
+++ ncftp-3.1.5/sh_util/gpshare.c	Thu Dec  5 19:06:36 2002
@@ -436,8 +436,8 @@
 #else
 	gIsAtty1 = isatty(1);
 	gIsAtty2 = isatty(2);
-	/* progmeters = ((isatty(2) != 0) && (getppid() > 1)) ? 1 : 0; */
-	progmeters = (getppid() > 1) ? 1 : 0;
+	progmeters = ((gIsAtty2 != 0) && (getppid() > 1)) ? 1 : 0;
+	/* progmeters = (getppid() > 1) ? 1 : 0; */
 #endif
 	return (progmeters);
 }	/* GetDefaultProgressMeterSetting */
