diff -Naupr ncftp-3.2.3.orig/autoconf_local/aclocal.m4 ncftp-3.2.3/autoconf_local/aclocal.m4
--- ncftp-3.2.3.orig/autoconf_local/aclocal.m4	2009-07-28 16:45:58.000000000 +0200
+++ ncftp-3.2.3/autoconf_local/aclocal.m4	2009-12-08 17:06:12.854319959 +0100
@@ -4632,7 +4632,7 @@ AC_MSG_CHECKING([for curses library])
 wi_cv_lib_curses=no
 wi_cv_lib_curses_result=no
 ac_save_LIBS="$LIBS"
-for LIBCURSES in "-lncurses" "-lcurses" "-lcurses -ltermcap" "-ltermcap -lcurses"
+for LIBCURSES in "-lncursesw" "-lncurses" "-lcurses" "-lcurses -ltermcap" "-ltermcap -lcurses"
 do
 	if test "x$LIBCURSES-$wi_cv_ncurses" = "x-lncurses-no" ; then
 		# This should never work
@@ -4705,7 +4705,7 @@ LIBS="$ac_save_LIBS"
 
 if test "$wi_cv_lib_curses_result" != "no" ; then
 	case "$LIBCURSES" in
-		"-lncurses")
+		"-lncurses*")
 			AC_DEFINE(HAVE_LIBNCURSES)
 			;;
 		"-lcurses")
diff -Naupr ncftp-3.2.3.orig/configure ncftp-3.2.3/configure
--- ncftp-3.2.3.orig/configure	2009-07-28 17:01:30.000000000 +0200
+++ ncftp-3.2.3/configure	2009-12-08 17:06:34.850195005 +0100
@@ -5105,7 +5105,7 @@ echo "configure:5104: checking for curse
 wi_cv_lib_curses=no
 wi_cv_lib_curses_result=no
 ac_save_LIBS="$LIBS"
-for LIBCURSES in "-lncurses" "-lcurses" "-lcurses -ltermcap" "-ltermcap -lcurses"
+for LIBCURSES in "-lncursesw" "-lncurses" "-lcurses" "-lcurses -ltermcap" "-ltermcap -lcurses"
 do
 	if test "x$LIBCURSES-$wi_cv_ncurses" = "x-lncurses-no" ; then
 		# This should never work
