--- btrfs-progs-0.19/Makefile.build	2011-08-05 14:06:29.871749619 -0400
+++ btrfs-progs-0.19/Makefile	2011-08-05 14:06:45.419727708 -0400
@@ -17,8 +17,7 @@ bindir = $(prefix)/bin
 LIBS=-luuid
 
 progs = btrfsctl mkfs.btrfs btrfs-debug-tree btrfs-show btrfs-vol btrfsck \
-	btrfs \
-	btrfs-map-logical
+	btrfs btrfs-map-logical btrfstune btrfs-image btrfs-zero-log
 
 # make C=1 to enable sparse
 ifdef C