--- btrfs-progs-0.19/mkfs.c.orig	2012-04-11 10:38:26.825973948 -0400
+++ btrfs-progs-0.19/mkfs.c	2012-04-11 10:39:46.031360540 -0400
@@ -372,7 +372,6 @@ static u64 parse_profile(char *s)
 
 static char *parse_label(char *input)
 {
-	int i;
 	int len = strlen(input);
 
 	if (len >= BTRFS_LABEL_SIZE) {
@@ -380,12 +379,6 @@ static char *parse_label(char *input)
 			BTRFS_LABEL_SIZE - 1);
 		exit(1);
 	}
-	for (i = 0; i < len; i++) {
-		if (input[i] == '/' || input[i] == '\\') {
-			fprintf(stderr, "invalid label %s\n", input);
-			exit(1);
-		}
-	}
 	return strdup(input);
 }
 