From 0465f8ceab739a0f096c97c31d43a934a6617099 Mon Sep 17 00:00:00 2001
From: Mario Ceresa <mrceresa@gmail.com>
Date: Sun, 20 Mar 2011 12:52:41 +0100
Subject: [PATCH 1/6] Added soname information for all targets

---
 CMakeLists.txt |   25 +++++++++++++++++++++++--
 1 files changed, 23 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 446d478..62e7a6f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,12 +20,21 @@ SET(DCMTK_MODULES config ofstd oflog dcmdata dcmimgle dcmjpeg dcmimage dcmtls dc
 SET(DCMTK_PACKAGE_NAME "dcmtk")
 SET(DCMTK_PACKAGE_DATE "2011-01-06")
 #SET(DCMTK_PACKAGE_TARNAME "dcmtk-3.6.0")
-SET(DCMTK_PACKAGE_VERSION "3.6.0")
-SET(DCMTK_PACKAGE_VERSION_NUMBER "360")
+#SET(DCMTK_PACKAGE_VERSION "3.6.0")
+#SET(DCMTK_PACKAGE_VERSION_NUMBER "360")
 SET(DCMTK_PACKAGE_VERSION_SUFFIX "")
 #SET(DCMTK_PACKAGE_STRING "dcmtk 3.6.0")
 #SET(DCMTK_PACKAGE_BUGREPORT "dicom-bugs@offis.de")
 
+SET(DCMTK_MAJOR_VERSION 3)
+SET(DCMTK_MINOR_VERSION 6)
+SET(DCMTK_BUILD_VERSION 0)
+SET(DCMTK_PACKAGE_VERSION "${DCMTK_MAJOR_VERSION}.${DCMTK_MINOR_VERSION}.${DCMTK_BUILD_VERSION}")
+SET(DCMTK_API_VERSION "${DCMTK_MAJOR_VERSION}.${DCMTK_MINOR_VERSION}")
+SET(DCMTK_LIBRARY_PROPERTIES VERSION "${DCMTK_PACKAGE_VERSION}" SOVERSION "${DCMTK_API_VERSION}")
+SET(DCMTK_PACKAGE_VERSION_NUMBER ${DCMTK_MAJOR_VERSION}${DCMTK_MINOR_VERSION}${DCMTK_BUILD_VERSION})
+
+
 # DCMTK build options
 OPTION(DCMTK_WITH_TIFF "Configure DCMTK with support for TIFF" ON)
 OPTION(DCMTK_WITH_PNG "Configure DCMTK with support for PNG" ON)
@@ -295,6 +304,18 @@ ENDIF(WITH_THREADS)
 # Recurse into subdirectories
 FOREACH(module ${DCMTK_MODULES})
   ADD_SUBDIRECTORY(${module})
+  
+# for each shared lib add proper soname information
+    IF(BUILD_SHARED_LIBS)
+	IF(${module} MATCHES config)
+        # Skip config module
+	ELSEIF(${module} MATCHES doxygen)
+        # Skip doxygen module
+        ELSE(${module} MATCHES config)
+          SET_TARGET_PROPERTIES(${module} PROPERTIES ${DCMTK_LIBRARY_PROPERTIES})
+        ENDIF(${module} MATCHES config)
+    ENDIF(BUILD_SHARED_LIBS)
+    
 ENDFOREACH(module)
 
 #-----------------------------------------------------------------------------
-- 
1.7.4

