From 8401d7914ef110d5aca35bf7e1409341c533f1e4 Mon Sep 17 00:00:00 2001
From: Mario Ceresa <mrceresa@gmail.com>
Date: Sun, 20 Mar 2011 12:54:16 +0100
Subject: [PATCH 2/6] Install libs in the correct arch dir

---
 CMakeLists.txt |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 62e7a6f..86b7601 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,7 +82,7 @@ INCLUDE_DIRECTORIES(${DCMTK_INCLUDE_DIR})
 
 SET(INSTALL_BINDIR "/bin")
 SET(INSTALL_INCDIR "/include")
-SET(INSTALL_LIBDIR "/lib")
+SET(INSTALL_LIBDIR "/lib${LIB_SUFFIX}")
 SET(INSTALL_ETCDIR "/etc/dcmtk")
 SET(INSTALL_DATDIR "/share/dcmtk")
 SET(INSTALL_DOCDIR "/share/doc/dcmtk")
-- 
1.7.4

