From 13f216a9ab2c93041cf18283ca4726c3c2e876d9 Mon Sep 17 00:00:00 2001
From: Mario Ceresa <mrceresa@gmail.com>
Date: Tue, 22 Mar 2011 18:06:54 +0100
Subject: [PATCH] Add soname generation for modules which are not in DCMTK_MODULES variable

---
 CMakeLists.txt |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 010c13c..01e6287 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -304,6 +304,7 @@ ENDIF(WITH_THREADS)
 # Start actual compilation tasks
 #-----------------------------------------------------------------------------
 
+
 # Recurse into subdirectories
 FOREACH(module ${DCMTK_MODULES})
   ADD_SUBDIRECTORY(${module})
@@ -321,6 +322,14 @@ FOREACH(module ${DCMTK_MODULES})
     
 ENDFOREACH(module)
 
+# Set target properties for modules which are not in DCMTK_MODULE
+SET_TARGET_PROPERTIES(dcmdsig PROPERTIES ${DCMTK_LIBRARY_PROPERTIES})
+SET_TARGET_PROPERTIES(ijg12 PROPERTIES ${DCMTK_LIBRARY_PROPERTIES})
+SET_TARGET_PROPERTIES(ijg16 PROPERTIES ${DCMTK_LIBRARY_PROPERTIES})
+SET_TARGET_PROPERTIES(ijg8 PROPERTIES ${DCMTK_LIBRARY_PROPERTIES})
+SET_TARGET_PROPERTIES(libi2d PROPERTIES ${DCMTK_LIBRARY_PROPERTIES})
+
+
 #-----------------------------------------------------------------------------
 # Installation tasks
 #-----------------------------------------------------------------------------
-- 
1.7.4

