diff -Nuar cr3.0.56-7.orig/crengine/src/lvfntman.cpp cr3.0.56-7/crengine/src/lvfntman.cpp
--- cr3.0.56-7.orig/crengine/src/lvfntman.cpp	2012-03-21 07:39:19.000000000 +0200
+++ cr3.0.56-7/crengine/src/lvfntman.cpp	2015-01-23 21:45:06.434827539 +0200
@@ -47,9 +47,9 @@
 #include "freetype/freetype.h"
 #else
 
-#include <freetype/config/ftheader.h>
+#include <freetype2/config/ftheader.h>
 //#include FT_FREETYPE_H
-#include <freetype/freetype.h>
+#include <freetype2/freetype.h>
 #endif
 
 #if (USE_FONTCONFIG==1)
