diff --git a/panel-plugin/panel-plugin.c b/panel-plugin/panel-plugin.c
index d41d590..5e1ce7e 100644
--- a/panel-plugin/panel-plugin.c
+++ b/panel-plugin/panel-plugin.c
@@ -737,7 +737,7 @@ modemlights_create_info (GtkWidget *widget,
                                           GTK_DIALOG_DESTROY_WITH_PARENT,
                                           GTK_MESSAGE_INFO,
                                           GTK_BUTTONS_CLOSE,
-                                          text);
+                                          "%s", text);
     gtk_dialog_run (GTK_DIALOG (dialog));
     gtk_widget_destroy (dialog);
 }
 
