diff -Naur -x '*~' libevent-2.0.21-stable/configure.in libevent-2.0.21-stable-automake-fix/configure.in
--- libevent-2.0.21-stable/configure.in	2012-11-18 07:34:13.000000000 +0100
+++ libevent-2.0.21-stable-automake-fix/configure.in	2012-12-31 14:04:30.219499020 +0100
@@ -12,7 +12,7 @@
 AC_CONFIG_MACRO_DIR([m4])
 
 AM_INIT_AUTOMAKE(libevent,2.0.21-stable)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADER(config.h)
 AC_DEFINE(NUMERIC_VERSION, 0x02001500, [Numeric representation of the version])
 
 dnl Initialize prefix.
diff -Naur -x '*~' libevent-2.0.21-stable/test/Makefile.am libevent-2.0.21-stable-automake-fix/test/Makefile.am
--- libevent-2.0.21-stable/test/Makefile.am	2012-05-31 06:08:19.000000000 +0200
+++ libevent-2.0.21-stable-automake-fix/test/Makefile.am	2012-12-31 14:10:49.045538012 +0100
@@ -19,7 +19,7 @@
 EXTRA_PROGRAMS = regress
 noinst_HEADERS = tinytest.h tinytest_macros.h regress.h tinytest_local.h
 
-TESTS = $(top_srcdir)/test/test.sh
+TESTS = test.sh
 
 BUILT_SOURCES =
 if BUILD_REGRESS
