--- samba-3.2.0pre1/source3/winbindd/winbindd_group.c.star	2007-10-03 17:40:31.000000000 -0400
+++ samba-3.2.0pre1/source3/winbindd/winbindd_group.c	2007-10-03 17:40:46.000000000 -0400
@@ -215,7 +215,7 @@
 	/* Group name and password */
 
 	safe_strcpy(gr->gr_name, full_group_name, sizeof(gr->gr_name) - 1);
-	safe_strcpy(gr->gr_passwd, "x", sizeof(gr->gr_passwd) - 1);
+	safe_strcpy(gr->gr_passwd, "*", sizeof(gr->gr_passwd) - 1);
 
 	return True;
 }
