--- Makefile.in	2002-10-08 18:09:12.000000000 +0200
+++ Makefile.in.oden	2008-08-05 17:45:15.000000000 +0200
@@ -161,7 +161,7 @@
 libgdbm_compat.la: $(C_LOBJS) gdbm.h
 	rm -f libgdbm_compat.la
 	$(LIBTOOL) --mode=link $(CC) -o libgdbm_compat.la -rpath $(libdir) \
-		-version-info $(SHLIB_VER) $(C_LOBJS)
+		-version-info $(SHLIB_VER) $(C_LOBJS) libgdbm.la
 
 gdbm.h:	gdbm.proto gdbmerrno.h gdbm.proto2
 	rm -f gdbm.h
