---
 Makefile.in |    4 	2 +	2 -	0 !
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: gdbm-1.8.3/Makefile.in
===================================================================
--- gdbm-1.8.3.orig/Makefile.in
+++ gdbm-1.8.3/Makefile.in
@@ -130,7 +130,7 @@ install: libgdbm.la gdbm.h gdbm.info
 	$(srcdir)/mkinstalldirs $(INSTALL_ROOT)$(libdir) \
 		$(INSTALL_ROOT)$(includedir) $(INSTALL_ROOT)$(man3dir) \
 		$(INSTALL_ROOT)$(infodir)
-	$(LIBTOOL) $(INSTALL) -c libgdbm.la $(INSTALL_ROOT)$(libdir)/libgdbm.la
+	$(LIBTOOL) --mode=install $(INSTALL) -c libgdbm.la $(INSTALL_ROOT)$(libdir)/libgdbm.la
 	$(INSTALL_DATA) -o $(BINOWN) -g $(BINGRP) gdbm.h \
 		$(INSTALL_ROOT)$(includedir)/gdbm.h
 	$(INSTALL_DATA) -o $(BINOWN) -g $(BINGRP) $(srcdir)/gdbm.3 \
@@ -141,7 +141,7 @@ install: libgdbm.la gdbm.h gdbm.info
 install-compat:
 	$(srcdir)/mkinstalldirs $(INSTALL_ROOT)$(libdir) \
 		$(INSTALL_ROOT)$(includedir)
-	$(LIBTOOL) $(INSTALL) -c libgdbm_compat.la \
+	$(LIBTOOL) --mode=install $(INSTALL) -c libgdbm_compat.la \
 		$(INSTALL_ROOT)$(libdir)/libgdbm_compat.la
 	$(INSTALL_DATA) -o $(BINOWN) -g $(BINGRP) $(srcdir)/dbm.h \
 		$(INSTALL_ROOT)$(includedir)/dbm.h
