diff -Naur xchat-2.8.6.orig/src/common/xchat.c xchat-2.8.6/src/common/xchat.c
--- xchat-2.8.6.orig/src/common/xchat.c	2008-04-01 11:21:18.000000000 +0300
+++ xchat-2.8.6/src/common/xchat.c	2009-02-11 19:29:07.000000000 +0200
@@ -602,7 +602,7 @@
 	"NAME WII\n"			"CMD quote WHOIS %2 %2\n\n";
 
 static char defaultconf_urlhandlers[] =
-		"NAME Open Link in Opera\n"		"CMD !opera -remote 'openURL(%s)'\n\n";
+		"NAME Open Link in Browser\n"		"CMD !xdg-open %s\n\n";
 
 #ifdef USE_SIGACTION
 /* Close and open log files on SIGUSR1. Usefull for log rotating */
