diff -ur xchat-2.8.6.orig/src/common/cfgfiles.c xchat-2.8.6/src/common/cfgfiles.c
--- xchat-2.8.6.orig/src/common/cfgfiles.c	2008-02-05 12:02:47.000000000 +0200
+++ xchat-2.8.6/src/common/cfgfiles.c	2010-01-30 21:13:03.886963491 +0200
@@ -668,7 +668,7 @@
 	prefs.autoopendccsendwindow = 1;
 	prefs.autoopendccchatwindow = 1;
 	prefs.userhost = 1;
-	prefs.gui_url_mod = 4;	/* ctrl */
+	prefs.gui_url_mod = 0;	/* ctrl */
 	prefs.gui_tray = 1;
 	prefs.gui_pane_left_size = 100;
 	prefs.gui_pane_right_size = 100;
@@ -707,7 +707,7 @@
 	snprintf (prefs.sounddir, sizeof (prefs.sounddir), "%s/sounds", get_xdir_utf8 ());
 	snprintf (prefs.dccdir, sizeof (prefs.dccdir), "%s/downloads", get_xdir_utf8 ());
 #endif
-	strcpy (prefs.doubleclickuser, "QUOTE WHOIS %s %s");
+	strcpy (prefs.doubleclickuser, "QUERY %s");
 	strcpy (prefs.awayreason, _("I'm busy"));
 	strcpy (prefs.quitreason, _("Leaving"));
 	strcpy (prefs.partreason, prefs.quitreason);
diff -ur xchat-2.8.6.orig/src/common/servlist.c xchat-2.8.6/src/common/servlist.c
--- xchat-2.8.6.orig/src/common/servlist.c	2008-04-01 11:22:34.000000000 +0300
+++ xchat-2.8.6/src/common/servlist.c	2010-01-30 21:24:21.113683231 +0200
@@ -206,8 +206,12 @@
 	{0,			"irc.ggn.net"},
 	{0,			"irc.vendetta.com"},
 
-	{"FreeNode",	0},
+	{"FreeNode",	0, "#pisi-linux"},
 	{0,				"irc.freenode.net"},
+#ifdef USE_OPENSSL
+	{0,			"irc.freenode.net/+7000"},
+	{0,			"irc.freenode.net/+7070"},
+#endif
 
 /*	{"Freeworld",	0},
 	{0,			"kabel.freeworld.nu"},
@@ -900,7 +904,7 @@
 				free (net->encoding);
 				net->encoding = strdup (def[i].charset);
 			}
-			if (g_str_hash (def[i].network) == 0x8e1b96f7)
+			if (g_str_hash (def[i].network) == 0xa82686ae)
 				prefs.slist_select = j;
 			j++;
 		} else
