Index: xchat-2.8.8/src/fe-gtk/menu.c
===================================================================
--- xchat-2.8.8.orig/src/fe-gtk/menu.c
+++ xchat-2.8.8/src/fe-gtk/menu.c
@@ -937,11 +937,8 @@ menu_urlmenu (GdkEventButton *event, cha
 	menu_quick_item (0, 0, menu, XCMENU_SHADED, 0, 0);
 
 	/* Two hardcoded entries */
-	if (strncmp (str_copy, "irc://", 6) == 0 ||
-	    strncmp (str_copy, "ircs://",7) == 0)
+	if (strncmp (str_copy, "irc://", 6) == 0 || strncmp (str_copy, "ircs://",7) == 0)
 		menu_quick_item_with_callback (open_url_cb, _("Connect"), menu, str_copy);
-	else
-		menu_quick_item_with_callback (open_url_cb, _("Open Link in Browser"), menu, str_copy);
 	menu_quick_item_with_callback (copy_to_clipboard_cb, _("Copy Selected Link"), menu, str_copy);
 	/* custom ones from urlhandlers.conf */
 	menu_create (menu, urlhandler_list, str_copy, TRUE);
