diff -Nur qemu-kvm-0.13.0-old//configure qemu-kvm-0.13.0/configure
--- qemu-kvm-0.13.0-old//configure	2010-12-08 18:19:10.593000002 +0200
+++ qemu-kvm-0.13.0/configure	2010-12-08 18:21:07.716000002 +0200
@@ -302,7 +302,7 @@
 EXESUF=""
 prefix="/usr/local"
 mandir="\${prefix}/share/man"
-datadir="\${prefix}/share/qemu"
+datadir="\${prefix}/share/kvm"
 docdir="\${prefix}/share/doc/qemu"
 bindir="\${prefix}/bin"
 sysconfdir="\${prefix}/etc"
diff -Nur qemu-kvm-0.13.0-old//kvm/scripts/run_img qemu-kvm-0.13.0/kvm/scripts/run_img
--- qemu-kvm-0.13.0-old//kvm/scripts/run_img	2010-12-08 18:19:10.506000002 +0200
+++ qemu-kvm-0.13.0/kvm/scripts/run_img	2010-12-08 18:21:07.719000002 +0200
@@ -1,4 +1,4 @@
 sudo /sbin/rmmod kvm
 sudo /sbin/insmod ../kernel/kvm.ko
 sudo chmod a+rw /dev/hvm
-../qemu/x86_64-softmmu/qemu-system-x86_64 -boot c -L /usr/share/qemu -hda /tmp/mkbootdisk/boot.img -m 384 -serial file:/tmp/qemu_serial.out
+../qemu/x86_64-softmmu/qemu-system-x86_64 -boot c -L /usr/share/kvm -hda /tmp/mkbootdisk/boot.img -m 384 -serial file:/tmp/qemu_serial.out
diff -Nur qemu-kvm-0.13.0-old//kvm/test/Makefile qemu-kvm-0.13.0/kvm/test/Makefile
--- qemu-kvm-0.13.0-old//kvm/test/Makefile	2010-12-08 18:19:10.513000002 +0200
+++ qemu-kvm-0.13.0/kvm/test/Makefile	2010-12-08 18:21:07.721000002 +0200
@@ -1,7 +1,7 @@
 
 include config.mak
 
-DESTDIR := $(PREFIX)/share/qemu/tests
+DESTDIR := $(PREFIX)/share/kvm/tests
 
 .PHONY: arch_clean clean
 
diff -Nur qemu-kvm-0.13.0-old//os-posix.c qemu-kvm-0.13.0/os-posix.c
--- qemu-kvm-0.13.0-old//os-posix.c	2010-12-08 18:19:10.517000002 +0200
+++ qemu-kvm-0.13.0/os-posix.c	2010-12-08 18:21:07.728000002 +0200
@@ -83,9 +83,9 @@
 }
 
 /* Find a likely location for support files using the location of the binary.
-   For installed binaries this will be "$bindir/../share/qemu".  When
+   For installed binaries this will be "$bindir/../share/kvm".  When
    running from the build tree this will be "$bindir/../pc-bios".  */
-#define SHARE_SUFFIX "/share/qemu"
+#define SHARE_SUFFIX "/share/kvm"
 #define BUILD_SUFFIX "/pc-bios"
 char *os_find_datadir(const char *argv0)
 {
