--- bash-completion-1.2.orig/bash_completion	2011-06-20 08:33:21.073014002 -0400
+++ bash-completion-1.2/bash_completion	2011-06-20 08:33:54.508014005 -0400
@@ -107,12 +107,12 @@
 complete -f -X '!*.@(zip|ZIP|z|Z|gz|GZ|tgz|TGZ)' bzme
 # konqueror not here on purpose, it's more than a web/html browser
 complete -f -X '!*.@(?([xX]|[sS])[hH][tT][mM]?([lL]))' netscape mozilla lynx opera galeon dillo elinks amaya firefox mozilla-firefox iceweasel google-chrome chromium-browser epiphany
-complete -f -X '!*.@(sxw|stw|sxg|sgl|doc?([mx])|dot?([mx])|rtf|txt|htm|html|odt|ott|odm)' oowriter
-complete -f -X '!*.@(sxi|sti|pps?(x)|ppt?([mx])|pot?([mx])|odp|otp)' ooimpress
-complete -f -X '!*.@(sxc|stc|xls?([bmx])|xlw|xlt?([mx])|[ct]sv|ods|ots)' oocalc
-complete -f -X '!*.@(sxd|std|sda|sdd|odg|otg)' oodraw
-complete -f -X '!*.@(sxm|smf|mml|odf)' oomath
-complete -f -X '!*.odb' oobase
+complete -f -X '!*.@(sxw|stw|sxg|sgl|doc?([mx])|dot?([mx])|rtf|txt|htm|html|odt|ott|odm)' lowriter
+complete -f -X '!*.@(sxi|sti|pps?(x)|ppt?([mx])|pot?([mx])|odp|otp)' loimpress
+complete -f -X '!*.@(sxc|stc|xls?([bmx])|xlw|xlt?([mx])|[ct]sv|ods|ots)' localc
+complete -f -X '!*.@(sxd|std|sda|sdd|odg|otg)' lodraw
+complete -f -X '!*.@(sxm|smf|mml|odf)' lomath
+complete -f -X '!*.odb' lobase
 complete -f -X '!*.rpm' rpm2cpio
 complete -f -X '!*.s@(qlite?(3)|?(3)db)' sqlite3
 complete -f -X '!*.aux' bibtex
