Index: bash_completion
===================================================================
--- bash_completion.orig
+++ bash_completion
@@ -941,24 +941,6 @@ _gids()
     fi
 }
 
-# This function completes on services
-#
-_services()
-{
-    local sysvdir famdir
-    [ -d /etc/rc.d/init.d ] && sysvdir=/etc/rc.d/init.d || sysvdir=/etc/init.d
-    famdir=/etc/xinetd.d
-    COMPREPLY=( $( printf '%s\n' \
-        $sysvdir/!(*.rpm@(orig|new|save)|*~|functions) ) )
-
-    if [ -d $famdir ]; then
-        COMPREPLY=( "${COMPREPLY[@]}" $( printf '%s\n' \
-            $famdir/!(*.rpm@(orig|new|save)|*~) ) )
-    fi
-
-    COMPREPLY=( $( compgen -W '${COMPREPLY[@]#@($sysvdir|$famdir)/}' -- "$cur" ) )
-}
-
 # This function completes on modules
 #
 _modules()
