--- libXau-1.0.3/AuDispose.c.orig	2007-11-23 17:51:27.000000000 -0200
+++ libXau-1.0.3/AuDispose.c	2007-11-23 17:51:34.000000000 -0200
@@ -33,7 +33,7 @@ in this Software without prior written a
 #include <X11/Xauth.h>
 #include <stdlib.h>
 
-void
+_X_EXPORT void
 XauDisposeAuth (Xauth *auth)
 {
     if (auth) {
--- libXau-1.0.3/AuRead.c.orig	2007-11-23 17:52:14.000000000 -0200
+++ libXau-1.0.3/AuRead.c	2007-11-23 17:52:21.000000000 -0200
@@ -69,7 +69,7 @@ read_counted_string (unsigned short *cou
     return 1;
 }
 
-Xauth *
+_X_EXPORT Xauth *
 XauReadAuth (FILE *auth_file)
 {
     Xauth   local;
