diff -Nur enigma-1.01-old/lib-src/enet/protocol.c enigma-1.01/lib-src/enet/protocol.c
--- enigma-1.01-old/lib-src/enet/protocol.c	2007-07-02 15:45:56.801694657 +0300
+++ enigma-1.01/lib-src/enet/protocol.c	2007-07-02 15:46:53.503074737 +0300
@@ -704,6 +704,9 @@
        if (currentData + sizeof (ENetProtocolCommandHeader) > & host -> receivedData [host -> receivedDataLength])
          break;
 
+       if (command -> header.commandLength < 0)
+         return 0;
+
        command -> header.commandLength = ENET_NET_TO_HOST_32 (command -> header.commandLength);
 
        if (command -> header.commandLength <= 0 || 
