--- enigma-1.01/src/client.cc.orig	2010-12-24 12:23:35.842000067 +0200
+++ enigma-1.01/src/client.cc	2010-12-24 12:25:21.008000067 +0200
@@ -128,11 +128,12 @@
 
     m_network_host = enet_host_create (NULL,
                                        1 /* only allow 1 outgoing connection */,
+                                       1 /* only allow 1 outgoing connection */,
                                        57600 / 8 /* 56K modem with 56 Kbps downstream bandwidth */,
                                        14400 / 8 /* 56K modem with 14 Kbps upstream bandwidth */);
 
     if (m_network_host == NULL) {
-        fprintf (stderr, 
+        fprintf (stderr,
                  "An error occurred while trying to create an ENet client host.\n");
         return false;
     }
@@ -148,7 +149,7 @@
     sv_address.port = 12345;
 
     /* Initiate the connection, allocating the two channels 0 and 1. */
-    m_server = enet_host_connect (m_network_host, &sv_address, 2);    
+    m_server = enet_host_connect (m_network_host, &sv_address, 2, 57600);    
     
     if (m_server == NULL) {
        fprintf (stderr, 
