diff -Naur enigma-1.01.orig/lib-src/enigma-core/IMG_SavePNG.c enigma-1.01/lib-src/enigma-core/IMG_SavePNG.c
--- enigma-1.01.orig/lib-src/enigma-core/IMG_SavePNG.c	2007-09-08 08:20:05.000000000 -0400
+++ enigma-1.01/lib-src/enigma-core/IMG_SavePNG.c	2012-07-26 16:38:56.773717657 -0400
@@ -84,7 +84,7 @@
                 png_bytep *row_pointers = 0;
 
                 /* Set error handling. */
-                if (setjmp(png_ptr->jmpbuf)) {
+                if (setjmp(png_jmpbuf(png_ptr))) {
                     IMG_SetError("Error writing the PNG file");
                 }
                 else {
