diff -up enigma-1.01/lib-src/enigma-core/ecl_dict.hh.gcc46 enigma-1.01/lib-src/enigma-core/ecl_dict.hh
--- enigma-1.01/lib-src/enigma-core/ecl_dict.hh.gcc46	2011-03-10 15:34:54.077340295 +0200
+++ enigma-1.01/lib-src/enigma-core/ecl_dict.hh	2011-03-10 15:35:34.204776840 +0200
@@ -20,6 +20,7 @@
 #define ECL_DICT_HH
 
 #include "ecl_error.hh"
+#include <cstddef>
 #include <utility>
 
 namespace ecl
diff -up enigma-1.01/lib-src/zipios++/src/directory.h.gcc46 enigma-1.01/lib-src/zipios++/src/directory.h
--- enigma-1.01/lib-src/zipios++/src/directory.h.gcc46	2007-09-08 15:20:05.000000000 +0300
+++ enigma-1.01/lib-src/zipios++/src/directory.h	2011-03-10 15:34:43.741227679 +0200
@@ -27,6 +27,7 @@
 
 // --------------------------------------------------------------------------
 
+#include <cstddef>
 #include <iterator>
 #include <string>
 #include <ctime>
