--- src/PropertyManager.cc.old	2009-05-22 10:50:04.000000000 +0200
+++ src/PropertyManager.cc	2009-05-22 10:50:30.000000000 +0200
@@ -28,6 +28,7 @@
 #include "XMLtoLocal.hh"
 #include "XMLtoUtf8.hh"
 #include "ecl_system.hh"
+#include <cstdio>
 #include <iostream>
 #include <xercesc/dom/DOM.hpp>
 #include <xercesc/util/XMLDouble.hpp>
--- src/StateManager.cc.old	2009-05-22 10:52:13.000000000 +0200
+++ src/StateManager.cc	2009-05-22 10:52:38.000000000 +0200
@@ -31,6 +31,7 @@
 #include "XMLtoUtf8.hh"
 #include "ecl_system.hh"
 #include "gui/ErrorMenu.hh"
+#include <cstdio>
 #include <iostream>
 #include <sstream>
 #include <xercesc/dom/DOM.hpp>
--- src/gui/LevelPackComposer.cc.old	2009-05-22 10:53:46.000000000 +0200
+++ src/gui/LevelPackComposer.cc	2009-05-22 10:54:24.000000000 +0200
@@ -28,6 +28,7 @@
 #include "lev/Index.hh"
 
 #include "main.hh"
+#include <cstdio>
 
 using namespace ecl;
 using namespace std;
--- src/lev/Index.cc.old	2009-05-22 10:55:16.000000000 +0200
+++ src/lev/Index.cc	2009-05-22 10:55:41.000000000 +0200
@@ -26,6 +26,7 @@
 #include "StateManager.hh"
 #include "lev/ScoreManager.hh"
 #include "lev/RatingManager.hh"
+#include <cstdio>
 
 
 namespace enigma { namespace lev {
--- src/lua.cc.old	2009-05-28 09:43:36.000000000 +0200
+++ src/lua.cc	2009-05-28 09:44:22.000000000 +0200
@@ -574,7 +574,7 @@
     int         y       = round_down<int>(lua_tonumber(L, 2));
     const char *dir     = lua_tostring(L,3);
     const char *allowed = "wsen";
-    char       *found   = strchr(allowed, dir[0]);
+    const char *found   = strchr(allowed, dir[0]);
 
     if (found && found[0]) 
         world::AddScramble(GridPos(x,y), enigma::Direction(found-allowed));
