--- enigma-1.01/src/netgame.cc.orig	2010-12-24 12:30:24.360000068 +0200
+++ enigma-1.01/src/netgame.cc	2010-12-24 12:32:45.669000051 +0200
@@ -155,7 +155,7 @@
     network_address.host = ENET_HOST_ANY;
     network_address.port = 12345;
 
-    network_host = enet_host_create (&network_address, 1, 0, 0);
+    network_host = enet_host_create (&network_address, 1, 0, 0, 0);
     if (network_host == NULL) {
         fprintf (stderr, 
                  "SV: An error occurred while trying to create an ENet server host.\n");
@@ -240,6 +240,7 @@
     ENetHost *m_network_host;
     m_network_host = enet_host_create (NULL,
                                        1 /* only allow 1 outgoing connection */,
+                                       1 /* only allow 1 outgoing connection */,
                                        57600 / 8 /* 56K modem with 56 Kbps downstream bandwidth */,
                                        14400 / 8 /* 56K modem with 14 Kbps upstream bandwidth */);
 
@@ -260,7 +261,7 @@
 
     /* Initiate the connection, allocating the two channels 0 and 1. */
     int numchannels = 2;
-    m_server = enet_host_connect (m_network_host, &sv_address, numchannels);
+    m_server = enet_host_connect (m_network_host, &sv_address, numchannels, 57600);
     
     if (m_server == NULL) {
         fprintf (stderr, 
