diff -up exiv2-0.18-pre2/src/Makefile.deps exiv2-0.18-pre2/src/Makefile
--- exiv2-0.18-pre2/src/Makefile.deps	2008-09-18 05:26:58.000000000 -0500
+++ exiv2-0.18-pre2/src/Makefile	2008-11-29 05:46:14.000000000 -0600
@@ -192,11 +192,19 @@ $(CCOBJ): %.o: %.cpp
 	@$(LIBTOOL) --mode=compile $(COMPILE.cc) -DEXV_BUILDING_LIB=1 -o $@ $<
 	@$(MAKEDEPEND)
 	@$(POSTDEPEND)
+# The dependancies here are wrong.  It generates .lo output,
+# so things are rebuilt *every* time 'make' is called.
+# Let's try a quick-n-dirty hack -- Rex
+	touch $@
 
 $(COBJ): %.o: %.c
 	@$(LIBTOOL) --mode=compile $(COMPILE.c) -DEXV_BUILDING_LIB=1 -o $@ $<
 	@$(MAKEDEPEND)
 	@$(POSTDEPEND)
+# The dependancies here are wrong.  It generates .lo output,
+# so things are rebuilt *every* time 'make' is called.
+# Let's try a quick-n-dirty hack -- Rex
+	touch $@
 
 $(sort $(BINOBJ) $(EXIV2OBJ) $(MCOBJ) path-test.o): %.o: %.cpp
 	$(COMPILE.cc) -o $@ $<
