Index: exiv2-0.19/config/configure.ac
===================================================================
--- exiv2-0.19.orig/config/configure.ac
+++ exiv2-0.19/config/configure.ac
@@ -111,9 +111,9 @@ AC_ARG_ENABLE(visibility,
     GXX_VISIBILITY=$enableval, GXX_VISIBILITY=yes)
 AC_MSG_RESULT($GXX_VISIBILITY)
 if test "$GXX_VISIBILITY" = "yes"; then
-    AX_CXX_CHECK_FLAG(-fvisibility-inlines-hidden,[],[],[GXX_VISIBILITY=yes],[GXX_VISIBILITY=no])
+    AX_CXX_CHECK_FLAG(-fvisibility=hidden,[],[],[GXX_VISIBILITY=yes],[GXX_VISIBILITY=no])
     if test "$GXX_VISIBILITY" = "yes"; then
-        CXXFLAGS="${CXXFLAGS} -fvisibility=hidden -fvisibility-inlines-hidden"
+        CXXFLAGS="${CXXFLAGS} -fvisibility=hidden"
         AC_DEFINE(HAVE_GXXCLASSVISIBILITY,1)
     fi
     if test "$enable_shared" = "yes"; then
Index: exiv2-0.19/configure
===================================================================
--- exiv2-0.19.orig/configure
+++ exiv2-0.19/configure
@@ -16745,7 +16745,7 @@ ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
 
     save_CXXFLAGS="$CXXFLAGS"
-    CXXFLAGS="$CXXFLAGS -fvisibility-inlines-hidden"
+    CXXFLAGS="$CXXFLAGS"
     cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -16798,7 +16798,7 @@ else
 fi
 
     if test "$GXX_VISIBILITY" = "yes"; then
-        CXXFLAGS="${CXXFLAGS} -fvisibility=hidden -fvisibility-inlines-hidden"
+        CXXFLAGS="${CXXFLAGS} -fvisibility=hidden"
         $as_echo "#define HAVE_GXXCLASSVISIBILITY 1" >>confdefs.h
 
     fi
