diff -up superkaramba-4.7.80/src/CMakeLists.txt.namelink_skip superkaramba-4.7.80/src/CMakeLists.txt
--- superkaramba-4.7.80/src/CMakeLists.txt.namelink_skip	2011-08-22 08:26:16.000000000 -0500
+++ superkaramba-4.7.80/src/CMakeLists.txt	2011-11-28 14:06:32.954362948 -0600
@@ -110,7 +110,7 @@ if(CMAKE_SYSTEM_NAME MATCHES "FreeBSD" O
   target_link_libraries(superkaramba kvm)
 endif(CMAKE_SYSTEM_NAME MATCHES "FreeBSD" OR CMAKE_SYSTEM_NAME MATCHES "DragonFly")
 
-install(TARGETS superkaramba ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS superkaramba ${INSTALL_TARGETS_DEFAULT_ARGS} LIBRARY NAMELINK_SKIP)
 
 ########### superkaramba bin ###############
 
