diff -up texinfo-4.13/makeinfo/sectioning.c.orig texinfo-4.13/makeinfo/sectioning.c
--- texinfo-4.13/makeinfo/sectioning.c.orig	2010-11-10 11:24:53.000000000 +0100
+++ texinfo-4.13/makeinfo/sectioning.c	2010-11-10 11:25:28.000000000 +0100
@@ -256,14 +256,14 @@ current_chapter_number (void)
     return xstrdup ("");
   else if (enum_marker == APPENDIX_MAGIC)
     {
-      char s[1];
+      char s[2];
       sprintf (s, "%c", numbers[0] + 64);
       return xstrdup (s);
     }
   else
     {
       char s[5];
-      sprintf (s, "%d", numbers[0]);
+      sprintf (s, "%4d", numbers[0]);
       return xstrdup (s);
     }
 }
