diff -up texinfo-4.13/util/texi2dvi.orig texinfo-4.13/util/texi2dvi
--- texinfo-4.13/util/texi2dvi.orig	2010-10-11 14:08:21.340831908 +0200
+++ texinfo-4.13/util/texi2dvi	2010-10-11 14:12:14.235841752 +0200
@@ -1683,7 +1683,7 @@ input_file_name_decode ()
 
   # If the COMMAND_LINE_FILENAME is not absolute (e.g., --debug.tex),
   # prepend `./' in order to avoid that the tools take it as an option.
-  echo "$command_line_filename" | $EGREP '^(/|[A-z]:/)' >&6 \
+  echo "$command_line_filename" | $EGREP '^(/|[A-Za-z]:/)' >&6 \
   || command_line_filename="./$command_line_filename"
 
   # See if the file exists.  If it doesn't we're in trouble since, even
