diff -urN clementine-1.2.0/src/podcasts/gpoddersearchpage.h clementine-1.2.0-patched/src/podcasts/gpoddersearchpage.h
--- clementine-1.2.0/src/podcasts/gpoddersearchpage.h	2012-04-14 00:44:57.000000000 +1100
+++ clementine-1.2.0-patched/src/podcasts/gpoddersearchpage.h	2013-10-16 14:58:37.266659053 +1100
@@ -20,7 +20,7 @@
 
 #include "addpodcastpage.h"
 
-#include <ApiRequest.h>
+#include <mygpo-qt/ApiRequest.h>
 
 class QNetworkAccessManager;
 
diff -urN clementine-1.2.0/src/podcasts/gpoddersync.h clementine-1.2.0-patched/src/podcasts/gpoddersync.h
--- clementine-1.2.0/src/podcasts/gpoddersync.h	2012-04-14 00:44:57.000000000 +1100
+++ clementine-1.2.0-patched/src/podcasts/gpoddersync.h	2013-10-16 14:58:12.966279384 +1100
@@ -27,7 +27,7 @@
 #include <QSharedPointer>
 #include <QUrl>
 
-#include <ApiRequest.h>
+#include <mygpo-qt/ApiRequest.h>
 
 class Application;
 class Podcast;
diff -urN clementine-1.2.0/src/podcasts/gpoddertoptagsmodel.cpp clementine-1.2.0-patched/src/podcasts/gpoddertoptagsmodel.cpp
--- clementine-1.2.0/src/podcasts/gpoddertoptagsmodel.cpp	2012-04-14 00:44:57.000000000 +1100
+++ clementine-1.2.0-patched/src/podcasts/gpoddertoptagsmodel.cpp	2013-10-16 14:58:53.803917471 +1100
@@ -20,7 +20,7 @@
 #include "podcast.h"
 #include "core/closure.h"
 
-#include <ApiRequest.h>
+#include <mygpo-qt/ApiRequest.h>
 
 #include <QMessageBox>
 
diff -urN clementine-1.2.0/src/podcasts/gpoddertoptagspage.h clementine-1.2.0-patched/src/podcasts/gpoddertoptagspage.h
--- clementine-1.2.0/src/podcasts/gpoddertoptagspage.h	2012-04-14 00:44:57.000000000 +1100
+++ clementine-1.2.0-patched/src/podcasts/gpoddertoptagspage.h	2013-10-16 14:57:47.048874529 +1100
@@ -20,7 +20,7 @@
 
 #include <QScopedPointer>
 
-#include <ApiRequest.h>
+#include <mygpo-qt/ApiRequest.h>
 
 #include "addpodcastpage.h"
 
diff -urN clementine-1.2.0/src/podcasts/podcast.cpp clementine-1.2.0-patched/src/podcasts/podcast.cpp
--- clementine-1.2.0/src/podcasts/podcast.cpp	2012-04-14 00:44:57.000000000 +1100
+++ clementine-1.2.0-patched/src/podcasts/podcast.cpp	2013-10-16 15:04:06.330805937 +1100
@@ -21,7 +21,7 @@
 #include <QDataStream>
 #include <QDateTime>
 
-#include <Podcast.h>
+#include <mygpo-qt/Podcast.h>
 
 const QStringList Podcast::kColumns = QStringList()
     << "url" << "title" << "description" << "copyright" << "link"
