--- clementine-1.0.1-orig/tests/CMakeLists.txt	2011-12-02 23:24:44.000000000 +0200
+++ clementine-1.0.1/tests/CMakeLists.txt	2012-02-11 01:50:16.620000103 +0200
@@ -129,7 +129,6 @@
 add_test_file(songloader_test.cpp false)
 add_test_file(songplaylistitem_test.cpp false)
 add_test_file(song_test.cpp false)
-add_test_file(translations_test.cpp false)
 add_test_file(utilities_test.cpp false)
 add_test_file(xspfparser_test.cpp false)
 
