--- clementine-1.1.1/src/main.cpp	2012-11-18 10:51:27.000000000 +0200
+++ clementine-1.1.1-new/src/main.cpp	2012-12-10 21:57:24.305727739 +0200
@@ -264,18 +264,6 @@
   }
 #endif
 
-  // This makes us show up nicely in gnome-volume-control
-  g_type_init();
-  g_set_application_name(QCoreApplication::applicationName().toLocal8Bit());
-
-  RegisterMetaTypes();
-
-#ifdef HAVE_LIBLASTFM
-  lastfm::ws::ApiKey = LastFMService::kApiKey;
-  lastfm::ws::SharedSecret = LastFMService::kSecret;
-  lastfm::setNetworkAccessManager(new NetworkAccessManager);
-#endif
-
   CommandlineOptions options(argc, argv);
 
   {
