--- libindicate-0.6.92.orig/docs/reference/tmpl/libindicate-unused.sgml
+++ libindicate-0.6.92/docs/reference/tmpl/libindicate-unused.sgml
@@ -35,10 +35,26 @@
 
 @parent_class: 
 
+<!-- ##### SIGNAL IndicateServer::indicator-added ##### -->
+<para>
+
+</para>
+
+@indicateserver: the object which received the signal.
+@arg1: 
+
 <!-- ##### SIGNAL IndicateServer::indicator-not-shown ##### -->
 <para>
 
 </para>
+
+@indicateserver: the object which received the signal.
+@arg1: 
+
+<!-- ##### SIGNAL IndicateServer::indicator-removed ##### -->
+<para>
+
+</para>
 
 @indicateserver: the object which received the signal.
 @arg1: 
