diff -Nru libindicate-12.10.1.orig/docs/reference/libindicate-docs.sgml libindicate-12.10.1/docs/reference/libindicate-docs.sgml
--- libindicate-12.10.1.orig/docs/reference/libindicate-docs.sgml	2012-09-20 18:17:39.924070034 -0400
+++ libindicate-12.10.1/docs/reference/libindicate-docs.sgml	2012-09-20 18:18:34.764020466 -0400
@@ -15,7 +15,6 @@
 
 <part id="subclass">
 	<title>Subclasses for specific indicators</title>
-	<xi:include href="xml/indicator-message.xml" />
 </part>
 
 <part id="listeners">
