diff -up gpsd-2.95/setup.py.silentmake gpsd-2.95/setup.py
--- gpsd-2.95/setup.py.silentmake	2010-06-11 10:38:24.000000000 +0200
+++ gpsd-2.95/setup.py	2010-07-14 17:23:12.000000000 +0200
@@ -57,7 +57,7 @@ if not 'clean' in sys.argv:
 gpspacket_sources = ["gpspacket.c", "packet.c", "isgps.c",
             "driver_rtcm2.c", "strl.c", "hex.c", "crc24q.c"]
 include_dirs = [ os.path.realpath(os.path.dirname(__file__)) ]
-version_out = os.popen(MAKE + " -s version")
+version_out = os.popen(MAKE + " --no-print-directory version")
 version = version_out.read()
 print(version)
 if version_out.close():
