diff --git a/src/compositor/compositor-xrender.c b/src/compositor/compositor-xrender.c
index 805002e..6294e9d 100644
--- a/src/compositor/compositor-xrender.c
+++ b/src/compositor/compositor-xrender.c
@@ -1821,6 +1821,10 @@ add_win (MetaScreen *screen,
 
   if (xwindow == info->output)
     return;
+    
+  /* If already added, ignore */
+  if (find_window_for_screen (screen, xwindow) != NULL)
+    return;
 
   cw = g_new0 (MetaCompWindow, 1);
   cw->screen = screen;

