diff -Nur libgksu-1.9.8/libgksu/libgksu.c libgksu-1.9.8.new/libgksu/libgksu.c
--- libgksu-1.9.8/libgksu/libgksu.c	2006-10-09 19:16:41.000000000 +0200
+++ libgksu-1.9.8.new/libgksu/libgksu.c	2006-10-09 19:17:16.000000000 +0200
@@ -819,7 +795,15 @@
     context->grab = TRUE;
 
   context->sudo_mode = gconf_client_get_bool (gconf_client, BASE_PATH "sudo-mode",
-					      NULL);
+						 NULL);
+  // UBUNTU specific - if the sudo mode is not set, check if we actualy have
+  // a registered schema for it, upgrades sometimes blew it away (lp: #59079)
+  if(context->sudo_mode == FALSE) {
+     GConfValue* v = gconf_client_get_default_from_schema(gconf_client, BASE_PATH "sudo-mode", NULL);
+     if(!v)
+	context->sudo_mode = TRUE;
+     gconf_value_free(v);
+  }
 }
 
 /**
