=== modified file 'libgksu/libgksu.c'
--- libgksu/libgksu.c	2008-06-12 15:41:31 +0000
+++ libgksu/libgksu.c	2008-06-12 15:47:14 +0000
@@ -869,12 +869,12 @@
 				     "The application '%s' lets you "
 				     "modify essential parts of your "
 				     "system."),
-				   command);
+				   g_markup_escape_text(command, -1));
 	  else
 	    msg = g_strdup_printf (_("<b><big>Enter your password to run "
 				     "the application '%s' as user %s"
 				     "</big></b>"),
-				   command, context->user);
+				   g_markup_escape_text(command, -1), context->user);
 	}
       else
 	{

