diff -Nur -x '*.orig' -x '*~' libgksu-2.0.9/libgksu/libgksu.c libgksu-2.0.9.new/libgksu/libgksu.c
--- libgksu-2.0.9/libgksu/libgksu.c	2009-04-08 01:23:29.000000000 +0100
+++ libgksu-2.0.9.new/libgksu/libgksu.c	2009-04-08 01:25:04.000000000 +0100
@@ -2674,6 +2674,12 @@
       cmd[argcount] = g_strdup("-H");
       argcount++;
     }
+  else
+    {
+      /* Preserve the environment, if sudo will let us */
+      cmd[argcount] = g_strdup("-E");
+      argcount++;
+    }
 
   /* Make sudo read from stdin */
   cmd[argcount] = g_strdup("-S");
