diff -Nur -x '*.orig' -x '*~' libgksu-2.0.13~pre1/libgksu/libgksu.c libgksu-2.0.13~pre1.new/libgksu/libgksu.c
--- libgksu-2.0.13~pre1/libgksu/libgksu.c	2010-01-05 11:32:35.000000000 +0100
+++ libgksu-2.0.13~pre1.new/libgksu/libgksu.c	2010-01-05 11:33:55.000000000 +0100
@@ -3095,7 +3095,7 @@
 
       if (WEXITSTATUS(status))
 	{
-          if (g_str_has_prefix(child_stderr, "Sorry, user "))
+          if (child_stderr && g_str_has_prefix(child_stderr, "Sorry, user "))
             {
               g_set_error (error, gksu_quark, GKSU_ERROR_NOT_ALLOWED,
                            _("The underlying authorization mechanism (sudo) "
