diff -ur libgksu-2.0.12.orig//Makefile.am libgksu-2.0.12/Makefile.am
--- libgksu-2.0.12.orig//Makefile.am	2008-05-09 20:39:57.000000000 +0300
+++ libgksu-2.0.12/Makefile.am	2010-11-24 23:18:59.415000011 +0200
@@ -17,11 +17,11 @@
 
 if GCONF_SCHEMAS_INSTALL
 install-data-local:
-        if test -z "$(DESTDIR)" ; then \
-          for p in $(schemas_DATA) ; do \
-            GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(srcdir)/$$p ; \
-          done \
-        fi
+	if test -z "$(DESTDIR)" ; then \
+		for p in $(schemas_DATA) ; do \
+			GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(srcdir)/$$p ; \
+		done \
+	fi
 else
 install-data-local:
 endif
