Index: VirtualBox-2.2.0_OSE/src/VBox/Installer/freebsd/VirtualBox.desktop
===================================================================
--- VirtualBox-2.2.0_OSE.orig/src/VBox/Installer/freebsd/virtualbox.desktop
+++ VirtualBox-2.2.0_OSE/src/VBox/Installer/freebsd/virtualbox.desktop
@@ -3,6 +3,7 @@ Encoding=UTF-8
 Version=1.0
 Name=Oracle VM VirtualBox
 GenericName=Virtual Machine
+GenericName[tr]=Sanal Makine
 Type=Application
 Exec=VirtualBox
 TryExec=VirtualBox
@@ -14,3 +15,4 @@ Comment[de]=Windows und andere Betriebss
 Comment[pl]=Uruchamianie wielu systemów wirtualnych na jednym komputerze gospodarza
 Comment[sv]=Kör flera virtuella system på en enda värddator
 Comment[ko]=가상 머신
+Comment[tr]=Tek bir sunucu bilgisayarda birden fazla sanal sistem çalıştırın
