Index: VirtualBox-2.2.4_OSE/src/VBox/Frontends/Common/VBoxKeyboard/keyboard-layouts.h
===================================================================
--- VirtualBox-2.2.4_OSE.orig/src/VBox/Frontends/Common/VBoxKeyboard/keyboard-layouts.h
+++ VirtualBox-2.2.4_OSE/src/VBox/Frontends/Common/VBoxKeyboard/keyboard-layouts.h
@@ -1836,7 +1836,7 @@ static const char main_key_th_pat[MAIN_L
 /* Turkey */
 static const char main_key_tr[MAIN_LEN][2] =
 {
-"\"\\","1!","2'","3^","4+","5%","6&","7/","8(","9)","0=","*?","-_",
+"\"\xe9","1!","2'","3^","4+","5%","6&","7/","8(","9)","0=","*?","-_",
 "qQ","wW","eE","rR","tT","yY","uU","\xb9I","oO","pP","\xbb\xab","\xfc\xdc",
 "aA","sS","dD","fF","gG","hH","jJ","kK","lL","\xba\xaa","i\xa9",",;",
 "zZ","xX","cC","vV","bB","nN","mM","\xf6\xd6","\xe7\xc7",".:","<>","\x0\x0","\x0\x0"
