Index: VirtualBox-2.2.4_OSE/src/VBox/Installer/linux/VBox.sh
===================================================================
--- VirtualBox-2.2.4_OSE.orig/src/VBox/Installer/linux/VBox.sh
+++ VirtualBox-2.2.4_OSE/src/VBox/Installer/linux/VBox.sh
@@ -32,16 +32,18 @@ fi
 
 if [ "$1" = "shutdown" ]; then
     SHUTDOWN="true"
-elif ! lsmod|grep -q vboxdrv; then
-    cat << EOF
+elif [ -x /usr/libexec/virt-wrapper ]; then
+    /usr/libexec/virt-wrapper virtualbox || exit 1
+
+    if ! lsmod|grep -q vboxdrv; then
+        cat << EOF
 WARNING: The vboxdrv kernel module is not loaded. Either there is no module
          available for the current kernel (`uname -r`) or it failed to
-         load. Please recompile the kernel module and install it by
-
-           sudo /etc/init.d/vboxdrv setup
+         load.
 
          You will not be able to start VMs until this problem is fixed.
 EOF
+    fi
 elif [ ! -c /dev/vboxdrv ]; then
     cat << EOF
 WARNING: The character device /dev/vboxdrv does not exist. Try
