diff -Nuar bl24_lin_src.OSmanOS/ImageLib/src/Makefile bl24_lin_src/ImageLib/src/Makefile
--- bl24_lin_src.OSmanOS/ImageLib/src/Makefile	2008-01-09 20:35:38.000000000 +0200
+++ bl24_lin_src/ImageLib/src/Makefile	2013-05-27 17:13:00.620750239 +0300
@@ -9,27 +9,13 @@
 # All source files needed for the library
 #
 
-SRC = 	gif/gif.c CImage.cpp png/zlib/adler32.c png/zlib/compress.c \
-	png/zlib/crc32.c png/zlib/deflate.c png/zlib/gzio.c png/zlib/infback.c \
-	png/zlib/inffast.c png/zlib/inflate.c png/zlib/inftrees.c png/zlib/trees.c \
-	png/zlib/uncompr.c png/zlib/zutil.c png/png/png.c png/png/pngerror.c \
-	png/png/pnggccrd.c png/png/pngget.c png/png/pngmem.c png/png/pngpread.c \
-	png/png/pngread.c png/png/pngrio.c png/png/pngrtran.c png/png/pngrutil.c \
-	png/png/pngset.c png/png/pngtrans.c png/png/pngvcrd.c png/png/pngwio.c \
-	png/png/pngwrite.c png/png/pngwtran.c png/png/pngwutil.c png/hpng.c \
-	jpg/H2v2.cpp jpg/idct.cpp jpg/jpegdecoder.cpp jpg/jpegdecodermain.cpp
+SRC = 	gif/gif.c CImage.cpp png/hpng.c \
+ 	jpg/H2v2.cpp jpg/idct.cpp jpg/jpegdecoder.cpp jpg/jpegdecodermain.cpp
 
 
-OBJ=    gif/gif.o CImage.o png/zlib/adler32.o png/zlib/compress.o \
-	png/zlib/crc32.o png/zlib/deflate.o png/zlib/gzio.o png/zlib/infback.o \
-	png/zlib/inffast.o png/zlib/inflate.o png/zlib/inftrees.o png/zlib/trees.o \
-	png/zlib/uncompr.o png/zlib/zutil.o png/png/png.o png/png/pngerror.o \
-	png/png/pnggccrd.o png/png/pngget.o png/png/pngmem.o png/png/pngpread.o \
-	png/png/pngread.o png/png/pngrio.o png/png/pngrtran.o png/png/pngrutil.o \
-	png/png/pngset.o png/png/pngtrans.o png/png/pngvcrd.o png/png/pngwio.o \
-	png/png/pngwrite.o png/png/pngwtran.o png/png/pngwutil.o png/hpng.o \
-	jpg/H2v2.o jpg/idct.o jpg/jpegdecoder.o jpg/jpegdecodermain.o
-
+OBJ=    gif/gif.o CImage.o png/hpng.o \
+ 	jpg/H2v2.o jpg/idct.o jpg/jpegdecoder.o jpg/jpegdecodermain.o
+ 	
 #---------------------------------------------------------------------
 
 CXX	= g++
diff -Nuar bl24_lin_src.OSmanOS/ImageLib/src/png/hpng.c bl24_lin_src/ImageLib/src/png/hpng.c
--- bl24_lin_src.OSmanOS/ImageLib/src/png/hpng.c	2008-01-09 20:35:38.000000000 +0200
+++ bl24_lin_src/ImageLib/src/png/hpng.c	2013-05-27 17:05:29.664777894 +0300
@@ -1,6 +1,7 @@
 #include <malloc.h>
 #include <math.h>
-#include "png/png.h"
+#include "png.h"
+#include <string.h>
 #include "hpng.h"
 
 char PngErrorMessage[1024];
@@ -45,10 +46,6 @@
 	// ---------------- Error handling
 
 	png_set_error_fn(png,NULL,my_png_error,NULL);
-  if (setjmp(png->jmpbuf)) {
- 	  png_destroy_read_struct(&png, &info, &endinfo);
-	  return 0;
-	}
 
 	// ----------------- Read the image
 
@@ -154,7 +151,7 @@
    {
     	sprintf(PngErrorMessage,"png_create_info_struct() failed");
       fclose(fp);
-      png_destroy_write_struct(&png_ptr,  png_infopp_NULL);
+      png_destroy_write_struct(&png_ptr, NULL);
       return PngErrorMessage;
    }
 
