diff -Nur blockout-old/BlockOut/Utils.cpp blockout/BlockOut/Utils.cpp
--- blockout-old/BlockOut/Utils.cpp	2007-09-13 09:54:33.861002366 +0300
+++ blockout/BlockOut/Utils.cpp	2007-09-13 09:57:34.344564694 +0300
@@ -165,8 +165,10 @@
 	char *blockoutHome = getenv("BL2_HOME");
 	if( blockoutHome==NULL ) {
 	  printf("BL2_HOME environement variable if not defined !\n");
-	  printf("Please set the BL2_HOME to the BlockOut II installation directory (ex: BL2_HOME=/usr/local/bl2).\n");
-		return FALSE;
+	  // printf("Please set the BL2_HOME to the BlockOut II installation directory (ex: BL2_HOME=/usr/local/bl2).\n");
+		// return FALSE;
+	  printf("Using default datadir.\n");
+      blockoutHome = "/usr/share/blockout2";
 	}
 	strcpy( bl2Home , blockoutHome );
 		
