diff -Nur blockout-old/BlockOut/Makefile blockout/BlockOut/Makefile
--- blockout-old/BlockOut/Makefile	2007-09-13 10:55:31.013030751 +0300
+++ blockout/BlockOut/Makefile	2007-09-13 10:55:56.204201110 +0300
@@ -40,9 +40,9 @@
 CXX		= g++
 
 CXXFLAGS =     -g -D_DEBUG -Dlinux -c \
-               -I$(SDL_ROOT)/include/SDL -I$(IMGLIB_ROOT) -I.
+               -I$(SDL_ROOT)/include/SDL -I$(IMGLIB_ROOT)/src -I.
 
-LIBS	 =     -L$(SDL_ROOT)/lib -L$(IMGLIB_ROOT)
+LIBS	 =     -L$(SDL_ROOT)/lib -L$(IMGLIB_ROOT)/lib/Release
 
 LFLAGS   =      $(LIBS) -l../ImageLib/src/libimagelib.a -lGL -lGLU -lSDL_mixer -lSDL -lpthread -lm -ldl 
 
