diff -Nur blockout-old/BlockOut/Makefile blockout/BlockOut/Makefile
--- blockout-old/BlockOut/Makefile	2007-09-13 10:48:31.391822871 +0300
+++ blockout/BlockOut/Makefile	2007-09-13 10:50:10.536750550 +0300
@@ -44,7 +44,7 @@
 
 LIBS	 =     -L$(SDL_ROOT)/lib -L$(IMGLIB_ROOT)
 
-LFLAGS   =      $(LIBS) -limagelib -lGL -lGLU -lSDL_mixer -lSDL -lpthread -lm -ldl 
+LFLAGS   =      $(LIBS) ../ImageLib/src/libimagelib.a -lGL -lGLU -lSDL_mixer -lSDL -lpthread -lm -ldl 
 
 #--------------------------------------------------------------------
 
