Index: gtk-qt-engine/kcm_gtk/kcmgtk.cpp
===================================================================
--- gtk-qt-engine.orig/kcm_gtk/kcmgtk.cpp
+++ gtk-qt-engine/kcm_gtk/kcmgtk.cpp
@@ -128,7 +128,9 @@ void KcmGtk::save()
 	QTextStream stream(&file);
 	stream << "#!/bin/bash\n\n";
 	stream << "# Make sure our customised gtkrc file is loaded.\n";
-	stream << "export GTK2_RC_FILES=" + k_gtkRcFileName + "\n";
+	stream << "if [ -e \"$(kde4-config --install module)/kcm_gtk4.so\" ]; then\n";
+	stream << "export GTK2_RC_FILES=${GTK2_RC_FILES:+$GTK2_RC_FILES:}" + k_gtkRcFileName + "\n";
+	stream << "fi\n";
 	file.close();
 	
 	// Make it executable
