Index: src/engine.cpp
===================================================================
--- src/engine.cpp.orig
+++ src/engine.cpp
@@ -107,6 +107,7 @@ Engine::Engine()
 	QString sessionEnv(getenv("SESSION_MANAGER"));
 	if (sessionEnv.endsWith(QString::number(getpid())) ||
 	    commandLine.contains("gnome-wm") ||
+	    commandLine.contains("nspluginviewer") ||
 	    commandLine.contains("metacity") ||
 	    commandLine.contains("xfwm4") ||
 	    commandLine.contains("gnome-settings-daemon") ||
