Index: gtk-qt-engine/kcm_gtk/kcmgtk.cpp
===================================================================
--- gtk-qt-engine.orig/kcm_gtk/kcmgtk.cpp
+++ gtk-qt-engine/kcm_gtk/kcmgtk.cpp
@@ -52,7 +52,6 @@ KcmGtk::KcmGtk(QWidget* parent, const QV
 	connect(m_ui.fontChange, SIGNAL(clicked()), SLOT(fontChangeClicked()));
 	connect(m_ui.fontKde, SIGNAL(clicked(bool)), SLOT(fontKdeClicked()));
 	connect(m_ui.styleBox, SIGNAL(activated(int)), SLOT(styleChanged()));
-	connect(m_ui.styleKde, SIGNAL(clicked(bool)), SLOT(styleKdeClicked(bool)));
 	connect(m_ui.firefoxFix, SIGNAL(clicked()), SLOT(firefoxFixClicked()));
 	
 	m_gtkRc = new GtkRcFile(k_gtkRcFileName);
@@ -88,8 +87,6 @@ void KcmGtk::load()
 {
 	m_gtkRc->load();
 	
-	m_ui.styleKde->setChecked(m_gtkRc->themeName() == k_qtThemeName);
-	m_ui.styleOther->setChecked(m_gtkRc->themeName() != k_qtThemeName);
 	m_ui.styleBox->setCurrentIndex(m_themes.keys().indexOf(m_gtkRc->themeName()));
 	
 	QFont defaultFont;
@@ -173,8 +170,8 @@ void KcmGtk::getInstalledThemes()
 	m_ui.styleBox->clear();
 	m_ui.styleBox->addItems(m_themes.keys());
 	
-	bool installed = m_themes.contains(k_qtThemeName);
-	m_ui.styleKde->setEnabled(installed);
+    /* Do not show warning when you couldn't find Qt4 engine, we remove it in the package since it's buggy with latest Firefox */
+	bool installed = true;
 	m_ui.warning1->setHidden(installed);
 	m_ui.warning2->setHidden(installed);
 	m_ui.warning3->setHidden(installed);
@@ -202,7 +199,6 @@ void KcmGtk::fontKdeClicked()
 void KcmGtk::styleChanged()
 {
 	m_gtkRc->setTheme(m_themes[m_ui.styleBox->currentText()]);
-	m_ui.styleOther->setChecked(true);
 	changed(true);
 }
 
Index: gtk-qt-engine/kcm_gtk/kcmgtkwidget.ui
===================================================================
--- gtk-qt-engine.orig/kcm_gtk/kcmgtkwidget.ui
+++ gtk-qt-engine/kcm_gtk/kcmgtkwidget.ui
@@ -1,35 +1,36 @@
-<ui version="4.0" >
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
  <class>KcmGtkWidget</class>
- <widget class="QWidget" name="KcmGtkWidget" >
-  <property name="geometry" >
+ <widget class="QWidget" name="KcmGtkWidget">
+  <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>535</width>
-    <height>587</height>
+    <width>547</width>
+    <height>596</height>
    </rect>
   </property>
-  <property name="windowTitle" >
+  <property name="windowTitle">
    <string>Dialog</string>
   </property>
-  <layout class="QVBoxLayout" >
+  <layout class="QVBoxLayout">
    <item>
-    <widget class="QGroupBox" name="groupBox" >
-     <property name="title" >
+    <widget class="QGroupBox" name="groupBox">
+     <property name="title">
       <string>GTK Styles</string>
      </property>
-     <layout class="QGridLayout" >
-      <item rowspan="5" row="0" column="0" >
-       <layout class="QVBoxLayout" >
+     <layout class="QGridLayout">
+      <item row="0" column="0" rowspan="7">
+       <layout class="QVBoxLayout">
         <item>
-         <widget class="QLabel" name="styleIcon" />
+         <widget class="QLabel" name="styleIcon"/>
         </item>
         <item>
          <spacer>
-          <property name="orientation" >
+          <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
-          <property name="sizeHint" >
+          <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
@@ -39,29 +40,15 @@
         </item>
        </layout>
       </item>
-      <item row="0" column="1" colspan="4" >
-       <widget class="QRadioButton" name="styleKde" >
-        <property name="text" >
-         <string>&amp;Use my KDE style in GTK applications</string>
-        </property>
-       </widget>
-      </item>
-      <item row="1" column="1" >
-       <widget class="QRadioButton" name="styleOther" >
-        <property name="text" >
-         <string>Use &amp;another style:</string>
-        </property>
-       </widget>
-      </item>
-      <item row="1" column="2" colspan="2" >
-       <widget class="QComboBox" name="styleBox" />
+      <item row="1" column="2" colspan="2">
+       <widget class="QComboBox" name="styleBox"/>
       </item>
-      <item row="1" column="4" >
+      <item row="1" column="4">
        <spacer>
-        <property name="orientation" >
+        <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
-        <property name="sizeHint" >
+        <property name="sizeHint" stdset="0">
          <size>
           <width>191</width>
           <height>20</height>
@@ -69,48 +56,48 @@
         </property>
        </spacer>
       </item>
-      <item row="2" column="1" colspan="4" >
-       <widget class="QLabel" name="warning1" >
-        <property name="sizePolicy" >
-         <sizepolicy vsizetype="Minimum" hsizetype="Preferred" >
+      <item row="4" column="1" colspan="4">
+       <widget class="QLabel" name="warning1">
+        <property name="sizePolicy">
+         <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
-        <property name="text" >
-         <string>&lt;html>&lt;head>&lt;meta name="qrichtext" content="1" />&lt;style type="text/css">
+        <property name="text">
+         <string>&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
 p, li { white-space: pre-wrap; }
-&lt;/style>&lt;/head>&lt;body style=" font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;">
-&lt;p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;">&lt;span style=" font-weight:600;">Warning:&lt;/span> The Gtk-Qt theme engine is not installed. This means you will not be able to use your KDE style in GTK applications.&lt;/p>&lt;/body>&lt;/html></string>
+&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
+&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Warning:&lt;/span&gt; The Gtk-Qt theme engine is not installed. This means you will not be able to use your KDE style in GTK applications.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
-        <property name="wordWrap" >
+        <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
-      <item row="3" column="1" colspan="4" >
-       <widget class="KUrlLabel" name="warning2" >
-        <property name="text" >
+      <item row="5" column="1" colspan="4">
+       <widget class="KUrlLabel" name="warning2">
+        <property name="text">
          <string>Gtk-Qt theme engine homepage</string>
         </property>
-        <property name="url" >
+        <property name="url">
          <string>http://gtk-qt.ecs.soton.ac.uk</string>
         </property>
        </widget>
       </item>
-      <item row="4" column="1" colspan="2" >
-       <widget class="QPushButton" name="warning3" >
-        <property name="text" >
+      <item row="6" column="1" colspan="2">
+       <widget class="QPushButton" name="warning3">
+        <property name="text">
          <string>Change search paths...</string>
         </property>
        </widget>
       </item>
-      <item row="4" column="3" colspan="2" >
+      <item row="6" column="3" colspan="2">
        <spacer>
-        <property name="orientation" >
+        <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
-        <property name="sizeHint" >
+        <property name="sizeHint" stdset="0">
          <size>
           <width>211</width>
           <height>20</height>
@@ -118,30 +105,37 @@ p, li { white-space: pre-wrap; }
         </property>
        </spacer>
       </item>
+      <item row="1" column="1">
+       <widget class="QLabel" name="styleGTK">
+        <property name="text">
+         <string>GTK Widget Style:</string>
+        </property>
+       </widget>
+      </item>
      </layout>
     </widget>
    </item>
    <item>
-    <widget class="QGroupBox" name="groupBox_2" >
-     <property name="title" >
+    <widget class="QGroupBox" name="groupBox_2">
+     <property name="title">
       <string>GTK Fonts</string>
      </property>
-     <layout class="QGridLayout" >
-      <item rowspan="4" row="0" column="0" >
-       <layout class="QVBoxLayout" >
+     <layout class="QGridLayout">
+      <item row="0" column="0" rowspan="4">
+       <layout class="QVBoxLayout">
         <item>
-         <widget class="QLabel" name="fontIcon" >
-          <property name="text" >
+         <widget class="QLabel" name="fontIcon">
+          <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item>
          <spacer>
-          <property name="orientation" >
+          <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
-          <property name="sizeHint" >
+          <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
@@ -151,33 +145,33 @@ p, li { white-space: pre-wrap; }
         </item>
        </layout>
       </item>
-      <item row="0" column="1" colspan="3" >
-       <widget class="QRadioButton" name="fontKde" >
-        <property name="text" >
+      <item row="0" column="1" colspan="3">
+       <widget class="QRadioButton" name="fontKde">
+        <property name="text">
          <string>Use m&amp;y KDE fonts in GTK applications</string>
         </property>
        </widget>
       </item>
-      <item row="1" column="1" >
-       <widget class="QRadioButton" name="fontOther" >
-        <property name="text" >
+      <item row="1" column="1">
+       <widget class="QRadioButton" name="fontOther">
+        <property name="text">
          <string>Use another font:</string>
         </property>
        </widget>
       </item>
-      <item row="1" column="2" >
-       <widget class="QPushButton" name="fontChange" >
-        <property name="text" >
+      <item row="1" column="2">
+       <widget class="QPushButton" name="fontChange">
+        <property name="text">
          <string>Change...</string>
         </property>
        </widget>
       </item>
-      <item row="1" column="3" >
+      <item row="1" column="3">
        <spacer>
-        <property name="orientation" >
+        <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
-        <property name="sizeHint" >
+        <property name="sizeHint" stdset="0">
          <size>
           <width>141</width>
           <height>20</height>
@@ -206,26 +200,26 @@ p, li { white-space: pre-wrap; }
     </widget>
    </item>
    <item>
-    <widget class="QGroupBox" name="groupBox_3" >
-     <property name="title" >
+    <widget class="QGroupBox" name="groupBox_3">
+     <property name="title">
       <string>Firefox and Thunderbird</string>
      </property>
-     <layout class="QGridLayout" >
-      <item rowspan="2" row="0" column="0" >
-       <layout class="QVBoxLayout" >
+     <layout class="QGridLayout">
+      <item row="0" column="0" rowspan="2">
+       <layout class="QVBoxLayout">
         <item>
-         <widget class="QLabel" name="firefoxIcon" >
-          <property name="text" >
+         <widget class="QLabel" name="firefoxIcon">
+          <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item>
          <spacer>
-          <property name="orientation" >
+          <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
-          <property name="sizeHint" >
+          <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
@@ -235,35 +229,35 @@ p, li { white-space: pre-wrap; }
         </item>
        </layout>
       </item>
-      <item row="0" column="1" colspan="2" >
-       <widget class="QLabel" name="label_4" >
-        <property name="sizePolicy" >
-         <sizepolicy vsizetype="Minimum" hsizetype="Preferred" >
+      <item row="0" column="1" colspan="2">
+       <widget class="QLabel" name="label_4">
+        <property name="sizePolicy">
+         <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
-        <property name="text" >
+        <property name="text">
          <string>Scrollbar buttons may sometimes be displayed incorrectly in Firefox and Thunderbird.  This can be fixed by installing some files into your Mozilla profile.</string>
         </property>
-        <property name="wordWrap" >
+        <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
-      <item row="1" column="1" >
-       <widget class="QPushButton" name="firefoxFix" >
-        <property name="text" >
+      <item row="1" column="1">
+       <widget class="QPushButton" name="firefoxFix">
+        <property name="text">
          <string>I&amp;nstall scrollbar fix...</string>
         </property>
        </widget>
       </item>
-      <item row="1" column="2" >
+      <item row="1" column="2">
        <spacer>
-        <property name="orientation" >
+        <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
-        <property name="sizeHint" >
+        <property name="sizeHint" stdset="0">
          <size>
           <width>221</width>
           <height>20</height>
@@ -275,27 +269,27 @@ p, li { white-space: pre-wrap; }
     </widget>
    </item>
    <item>
-    <widget class="QLabel" name="label_6" >
-     <property name="sizePolicy" >
-      <sizepolicy vsizetype="Minimum" hsizetype="Preferred" >
+    <widget class="QLabel" name="label_6">
+     <property name="sizePolicy">
+      <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
-     <property name="text" >
+     <property name="text">
       <string>Note that changes made here will only affect GTK applications when KDE is running.  To change the styles and fonts used by applications in GNOME, use the GNOME configuration tools.</string>
      </property>
-     <property name="wordWrap" >
+     <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item>
     <spacer>
-     <property name="orientation" >
+     <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
-     <property name="sizeHint" >
+     <property name="sizeHint" stdset="0">
       <size>
        <width>535</width>
        <height>21</height>
@@ -307,6 +301,11 @@ p, li { white-space: pre-wrap; }
  </widget>
  <customwidgets>
   <customwidget>
+   <class>KSqueezedTextLabel</class>
+   <extends>QLabel</extends>
+   <header>ksqueezedtextlabel.h</header>
+  </customwidget>
+  <customwidget>
    <class>KUrlLabel</class>
    <extends>QLabel</extends>
    <header>kurllabel.h</header>
Index: gtk-qt-engine/po/tr.po
===================================================================
--- gtk-qt-engine.orig/po/tr.po
+++ gtk-qt-engine/po/tr.po
@@ -29,8 +29,8 @@ msgstr "GTK Stilleri"
 #. i18n: file /mnt/lvm/home/david/projects/cmakestuff/gtk-qt-engine/po/../kcm_gtk/kcmgtkwidget.ui line 109
 #: rc.cpp:9
 #, no-c-format
-msgid "&Use my KDE style in GTK applications"
-msgstr "GTK uygulamalarında KDE stilimi kullan"
+msgid "GTK Widget Style:"
+msgstr "GTK Parçacık Stili:"
 
 #. i18n: file /mnt/lvm/home/david/projects/cmakestuff/gtk-qt-engine/po/../kcm_gtk/kcmgtkwidget.ui line 117
 #: rc.cpp:12
