diff -Nuar usbmuxd-1.0.3.orig/CMakeLists.txt usbmuxd-1.0.3/CMakeLists.txt
--- usbmuxd-1.0.5.orig/CMakeLists.txt	2010-05-06 01:00:40.314397839 +0300
+++ usbmuxd-1.0.5/CMakeLists.txt	2010-05-06 01:05:30.158396652 +0300
@@ -12,13 +12,13 @@
 
 message("-- Configuring usbmuxd v${VERSION_TAG}")
 
-if(NOT DEFINED LIB_SUFFIX)
-  if(CMAKE_SIZEOF_VOID_P EQUAL 8 AND ${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-    set(LIB_SUFFIX "64" CACHE STRING "Define suffix of library directory name (32/64)" )
-  else()
-    set(LIB_SUFFIX "" CACHE STRING "Define suffix of library directory name (32/64)" )
-  endif()
-endif()
+#if(NOT DEFINED LIB_SUFFIX)
+#  if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+#    set(LIB_SUFFIX "64" CACHE STRING "Define suffix of library directory name (32/64)" )
+#  else()
+#    set(LIB_SUFFIX "" CACHE STRING "Define suffix of library directory name (32/64)" )
+#  endif()
+#endif()
 
 # let CFLAGS env override this
 if(CMAKE_C_FLAGS STREQUAL "")

diff -Nuar usbmuxd-1.0.3.orig/libusbmuxd.pc.in usbmuxd-1.0.3/libusbmuxd.pc.in
--- usbmuxd-1.0.5.orig/libusbmuxd.pc.in	2010-05-06 01:00:40.317397662 +0300
+++ usbmuxd-1.0.5/libusbmuxd.pc.in	2010-05-06 01:06:02.070396046 +0300
@@ -1,11 +1,11 @@
 prefix=${CMAKE_INSTALL_PREFIX}
 exec_prefix=${CMAKE_INSTALL_PREFIX}
-libdir=${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}
+libdir=${CMAKE_INSTALL_PREFIX}/lib
 includedir=${CMAKE_INSTALL_PREFIX}/include
 
 Name: libusbmuxd
 Description: A library to communicate with the usbmux daemon
 Version: ${USBMUXD_VERSION}
-Libs: -L${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX} -lusbmuxd
+Libs: -L${CMAKE_INSTALL_PREFIX}/lib -lusbmuxd
 Cflags: -I${CMAKE_INSTALL_PREFIX}/include
 
