Index: darktable-1.1.4/tools/create_metadata.sh
===================================================================
--- darktable-1.1.4.orig/tools/create_metadata.sh
+++ darktable-1.1.4/tools/create_metadata.sh
@@ -5,7 +5,7 @@ outputdir=$2
 outputheader=metadata_gen.h
 outputbody=metadata_gen.c
 
-headerdefine=__$(printf $outputheader | tr '[:lower:].' '[:upper:]_')__
+headerdefine=__$(printf $outputheader | tr '[a-z].' '[A-Z]_')__
 # header of the .h file
 cat > $outputdir/$outputheader << EOF
 /** generated file, do not edit! */
@@ -31,7 +31,7 @@ EOF
 # iterate over the input
 first=0
 for line in $(cat $inputfile | grep -v "^#"); do
-    enum=DT_METADATA_$(printf $line | tr '[:lower:].' '[:upper:]_')
+    enum=DT_METADATA_$(printf $line | tr '[a-z].' '[A-Z]_')
     length=$(printf $line | wc -c)
     if [ "$first" -ne 0 ]; then
         printf ",\n" >> $outputdir/$outputheader
