Index: pkcs11-helper-1.08/lib/libpkcs11-helper-1.pc.in
===================================================================
--- pkcs11-helper-1.08.orig/lib/libpkcs11-helper-1.pc.in
+++ pkcs11-helper-1.08/lib/libpkcs11-helper-1.pc.in
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-includedir=@includedir@
+includedir=@includedir@/pkcs11-helper-1.0
 features=@PKCS11H_FEATURES@
 
 Name: pkcs11-helper
