#fixed
diff -ur cfitsio/configure cfitsio-fixed-configure/configure
--- cfitsio/configure	2010-01-08 13:03:29.000000000 -0500
+++ cfitsio-fixed-configure/configure	2010-01-27 15:46:34.000000000 -0500
@@ -4058,7 +4058,8 @@
     SHLIB_SUFFIX=".sl"
     ;;
   lnx)
-    SHLIB_LD=":"
+    SHLIB_LD="cc -shared -Wl,-soname,libcfitsio.so.0"
+    SHLIB_SUFFIX=".so.0"
     ;;
   osf)
     SHLIB_LD="ld -shared -expect_unresolved '*'"
@@ -4079,7 +4080,7 @@
 # Darwin uses gcc (=cc), but needs different flags (see above)
 # if test "x$GCC" = xyes; then
 if test "x$GCC" = xyes && test "x$EXT" != xdarwin && test "x$EXT" != xcygwin; then
-  SHLIB_LD="$CC -shared"
+  SHLIB_LD="cc -shared -Wl,-soname,libcfitsio.so.0"
   lhea_shlib_cflags='-fPIC'
 fi
 if test "x$lhea_shlib_cflags" != x; then

