From a872047b9b04707adb5e26b6e14cc7a1d740421f Mon Sep 17 00:00:00 2001
From: =?utf8?q?Andrew=20Lee=20(=E6=9D=8E=E5=81=A5=E7=A7=8B)?= <ajqlee@debian.org>
Date: Fri, 29 Jul 2011 21:09:33 +0800
Subject: [PATCH] Support user switch with gdm3 in lxsession-logout.

---
 lxsession-logout/lxsession-logout.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/lxsession-logout/lxsession-logout.c b/lxsession-logout/lxsession-logout.c
index 86f906a..a1c26f4 100644
--- a/lxsession-logout/lxsession-logout.c
+++ b/lxsession-logout/lxsession-logout.c
@@ -442,6 +442,13 @@ int main(int argc, char * argv[])
         handler_context.switch_user_GDM = TRUE;
     }
 
+    /* If we are under GDM3, its "Switch User" is available. */
+    if (verify_running("gdm3", "gdmflexiserver"))
+    {
+        handler_context.switch_user_available = TRUE;
+        handler_context.switch_user_GDM = TRUE;
+    }
+
     /* If we are under KDM, its "Switch User" is available. */
     if (verify_running("kdm", "kdmctl"))
     {
-- 
1.7.0.1

