From 466982ce65e6d673c54ba7a0546d463da1fe5bfa Mon Sep 17 00:00:00 2001
From: =?utf8?q?Andrew=20Lee=20(=E6=9D=8E=E5=81=A5=E7=A7=8B)?= <ajqlee@debian.org>
Date: Fri, 29 Jul 2011 21:11:06 +0800
Subject: [PATCH] Support reboot and shutdown in LTST client. Thanks Vagrant Cascadian <vagrant@freegeek.org>.

---
 lxsession-logout/lxsession-logout.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/lxsession-logout/lxsession-logout.c b/lxsession-logout/lxsession-logout.c
index a1c26f4..5bb8add 100644
--- a/lxsession-logout/lxsession-logout.c
+++ b/lxsession-logout/lxsession-logout.c
@@ -458,7 +458,11 @@ int main(int argc, char * argv[])
 
     /* LTSP support */
     if (g_getenv("LTSP_CLIENT"))
+    {
         handler_context.ltsp = TRUE;
+        handler_context.shutdown_available = TRUE;
+        handler_context.reboot_available = TRUE;
+    }
 
     /* Make the button images accessible. */
     gtk_icon_theme_append_search_path(gtk_icon_theme_get_default(), PACKAGE_DATA_DIR "/lxsession/images");
-- 
1.7.0.1

