diff -Nur gfxtheme-pardus-0.11.3-old/src/dia_kernelopts.inc gfxtheme-pardus-0.11.3/src/dia_kernelopts.inc
--- gfxtheme-pardus-0.11.3-old/src/dia_kernelopts.inc	2011-01-18 12:08:37.782246102 +0200
+++ gfxtheme-pardus-0.11.3/src/dia_kernelopts.inc	2011-01-18 12:09:02.274279886 +0200
@@ -12,7 +12,7 @@
 
 /kernelopts.options [
   ""
-  "apm=off acpi=off mce=off barrier=off ide=nodma idewait=50 i8042.nomux psmouse.proto=bare irqpoll pci=nommconf"
+  "apm=off acpi=off mce=off barrier=off ide=nodma idewait=50 i8042.nomux psmouse.proto=bare irqpoll pci=nommconf edd=off"
   "acpi=off"
   "nolapic"
   "bootmem_debug debug debugpat dynamic_printk earlyprintk=vga initcall_debug loglevel=7 mminit_loglevel=4 pnp.debug sched_debug apic=debug"
