diff -Nur pardus-old/data-boot/languages pardus/data-boot/languages
--- pardus-old/data-boot/languages	2008-06-15 22:32:31.000000000 +0300
+++ pardus/data-boot/languages	2008-06-15 22:32:50.000000000 +0300
@@ -5,6 +5,5 @@
 fr_FR
 it_IT
 nl_NL
-pl_PL
 pt_BR
 tr_TR
diff -Nur pardus-old/data-install/languages pardus/data-install/languages
--- pardus-old/data-install/languages	2008-06-15 22:32:31.000000000 +0300
+++ pardus/data-install/languages	2008-06-15 22:32:57.000000000 +0300
@@ -5,6 +5,5 @@
 fr_FR
 it_IT
 nl_NL
-pl_PL
 pt_BR
 tr_TR
diff -Nur pardus-old/Makefile pardus/Makefile
--- pardus-old/Makefile	2008-06-15 22:32:31.000000000 +0300
+++ pardus/Makefile	2008-06-15 22:32:43.000000000 +0300
@@ -14,7 +14,7 @@
 SUBDIRS      = fonts help-boot help-install po src
 
 DEFAULT_LANG = tr_TR
-DEFAULT_LANGUAGES = ca de en es fr it nl pl pt_BR tr
+DEFAULT_LANGUAGES = ca de en es fr it nl pt_BR tr
 
 .PHONY: all clean distclean themes $(SUBDIRS)
 
