Do not treat duplicate certs as an error.

--- openssl-0.9.6/crypto/x509/by_file.c	Wed Sep 27 15:09:05 2000
+++ openssl-0.9.6/crypto/x509/by_file.c	Wed Sep 27 14:21:20 2000
@@ -163,8 +163,12 @@
 					}
 				}
 			i=X509_STORE_add_cert(ctx->store_ctx,x);
-			if (!i) goto err;
-			count++;
+			/* ignore any problems with current certificate 
+			    and continue with the next one */
+			if (i)
+			    count++;
+			else
+			    ERR_clear_error();
 			X509_free(x);
 			x=NULL;
 			}
@@ -179,7 +183,8 @@
 			goto err;
 			}
 		i=X509_STORE_add_cert(ctx->store_ctx,x);
-		if (!i) goto err;
+		if (!i)
+		  ERR_clear_error();
 		ret=i;
 		}
 	else
