Do not treat duplicate certs as an error.

--- openssl/crypto/x509/by_file.c
+++ openssl/crypto/x509/by_file.c
@@ -164,8 +164,12 @@ int X509_load_cert_file(X509_LOOKUP *ctx, const char *file, int type)
 					}
 				}
 			i=X509_STORE_add_cert(ctx->store_ctx,x);
-			if (!i) goto err;
-			count++;
+			/* ignore any problems with current certificate
+			    and continue with the next one */
+			if (i)
+			    count++;
+			else
+			    ERR_clear_error();
 			X509_free(x);
 			x=NULL;
 			}
@@ -180,7 +184,8 @@ int X509_load_cert_file(X509_LOOKUP *ctx, const char *file, int type)
 			goto err;
 			}
 		i=X509_STORE_add_cert(ctx->store_ctx,x);
-		if (!i) goto err;
+		if (!i)
+		  ERR_clear_error();
 		ret=i;
 		}
 	else
