diff -up openssl-1.0.1c/util/perlpath.pl.perlfind openssl-1.0.1c/util/perlpath.pl
--- openssl-1.0.1c/util/perlpath.pl.perlfind	2012-07-11 22:57:33.000000000 +0200
+++ openssl-1.0.1c/util/perlpath.pl	2012-07-12 00:31:12.102156275 +0200
@@ -4,10 +4,10 @@
 # line in all scripts that rely on perl.
 #
 
-require "find.pl";
+use File::Find;
 
 $#ARGV == 0 || print STDERR "usage: perlpath newpath  (eg /usr/bin)\n";
-&find(".");
+find(\&wanted, ".");
 
 sub wanted
 	{
