diff -Naur openssl-0.9.8l-orig/crypto/md5/asm/md5-x86_64.pl openssl-0.9.8l/crypto/md5/asm/md5-x86_64.pl
--- openssl-0.9.8l-orig/crypto/md5/asm/md5-x86_64.pl	2007-11-11 07:34:06.000000000 -0600
+++ openssl-0.9.8l/crypto/md5/asm/md5-x86_64.pl	2009-11-16 14:34:32.000000000 -0600
@@ -146,73 +146,73 @@
 	mov	%ecx,		%r14d
 	mov	%edx,		%r15d
 EOF
-round1_step(-1,'%eax','%ebx','%ecx','%edx', '1','0xd76aa478', '7');
-round1_step( 0,'%edx','%eax','%ebx','%ecx', '2','0xe8c7b756','12');
+round1_step(-1,'%eax','%ebx','%ecx','%edx', '1','-0x28955B88', '7');
+round1_step( 0,'%edx','%eax','%ebx','%ecx', '2','-0x173848AA','12');
 round1_step( 0,'%ecx','%edx','%eax','%ebx', '3','0x242070db','17');
-round1_step( 0,'%ebx','%ecx','%edx','%eax', '4','0xc1bdceee','22');
-round1_step( 0,'%eax','%ebx','%ecx','%edx', '5','0xf57c0faf', '7');
+round1_step( 0,'%ebx','%ecx','%edx','%eax', '4','-0x3E423112','22');
+round1_step( 0,'%eax','%ebx','%ecx','%edx', '5','-0x0A83F051', '7');
 round1_step( 0,'%edx','%eax','%ebx','%ecx', '6','0x4787c62a','12');
-round1_step( 0,'%ecx','%edx','%eax','%ebx', '7','0xa8304613','17');
-round1_step( 0,'%ebx','%ecx','%edx','%eax', '8','0xfd469501','22');
+round1_step( 0,'%ecx','%edx','%eax','%ebx', '7','-0x57CFB9ED','17');
+round1_step( 0,'%ebx','%ecx','%edx','%eax', '8','-0x02B96AFF','22');
 round1_step( 0,'%eax','%ebx','%ecx','%edx', '9','0x698098d8', '7');
-round1_step( 0,'%edx','%eax','%ebx','%ecx','10','0x8b44f7af','12');
-round1_step( 0,'%ecx','%edx','%eax','%ebx','11','0xffff5bb1','17');
-round1_step( 0,'%ebx','%ecx','%edx','%eax','12','0x895cd7be','22');
+round1_step( 0,'%edx','%eax','%ebx','%ecx','10','-0x74BB0851','12');
+round1_step( 0,'%ecx','%edx','%eax','%ebx','11','-0x0000A44F','17');
+round1_step( 0,'%ebx','%ecx','%edx','%eax','12','-0x76A32842','22');
 round1_step( 0,'%eax','%ebx','%ecx','%edx','13','0x6b901122', '7');
-round1_step( 0,'%edx','%eax','%ebx','%ecx','14','0xfd987193','12');
-round1_step( 0,'%ecx','%edx','%eax','%ebx','15','0xa679438e','17');
+round1_step( 0,'%edx','%eax','%ebx','%ecx','14','-0x02678E6D','12');
+round1_step( 0,'%ecx','%edx','%eax','%ebx','15','-0x5986BC72','17');
 round1_step( 1,'%ebx','%ecx','%edx','%eax', '0','0x49b40821','22');
 
-round2_step(-1,'%eax','%ebx','%ecx','%edx', '6','0xf61e2562', '5');
-round2_step( 0,'%edx','%eax','%ebx','%ecx','11','0xc040b340', '9');
+round2_step(-1,'%eax','%ebx','%ecx','%edx', '6','-0x09E1DA9E', '5');
+round2_step( 0,'%edx','%eax','%ebx','%ecx','11','-0x3FBF4CC0', '9');
 round2_step( 0,'%ecx','%edx','%eax','%ebx', '0','0x265e5a51','14');
-round2_step( 0,'%ebx','%ecx','%edx','%eax', '5','0xe9b6c7aa','20');
-round2_step( 0,'%eax','%ebx','%ecx','%edx','10','0xd62f105d', '5');
+round2_step( 0,'%ebx','%ecx','%edx','%eax', '5','-0x16493856','20');
+round2_step( 0,'%eax','%ebx','%ecx','%edx','10','-0x29D0EFA3', '5');
 round2_step( 0,'%edx','%eax','%ebx','%ecx','15', '0x2441453', '9');
-round2_step( 0,'%ecx','%edx','%eax','%ebx', '4','0xd8a1e681','14');
-round2_step( 0,'%ebx','%ecx','%edx','%eax', '9','0xe7d3fbc8','20');
+round2_step( 0,'%ecx','%edx','%eax','%ebx', '4','-0x275E197F','14');
+round2_step( 0,'%ebx','%ecx','%edx','%eax', '9','-0x182C0438','20');
 round2_step( 0,'%eax','%ebx','%ecx','%edx','14','0x21e1cde6', '5');
-round2_step( 0,'%edx','%eax','%ebx','%ecx', '3','0xc33707d6', '9');
-round2_step( 0,'%ecx','%edx','%eax','%ebx', '8','0xf4d50d87','14');
+round2_step( 0,'%edx','%eax','%ebx','%ecx', '3','-0x3CC8F82A', '9');
+round2_step( 0,'%ecx','%edx','%eax','%ebx', '8','-0x0B2AF279','14');
 round2_step( 0,'%ebx','%ecx','%edx','%eax','13','0x455a14ed','20');
-round2_step( 0,'%eax','%ebx','%ecx','%edx', '2','0xa9e3e905', '5');
-round2_step( 0,'%edx','%eax','%ebx','%ecx', '7','0xfcefa3f8', '9');
+round2_step( 0,'%eax','%ebx','%ecx','%edx', '2','-0x561C16FB', '5');
+round2_step( 0,'%edx','%eax','%ebx','%ecx', '7','-0x03105C08', '9');
 round2_step( 0,'%ecx','%edx','%eax','%ebx','12','0x676f02d9','14');
-round2_step( 1,'%ebx','%ecx','%edx','%eax', '0','0x8d2a4c8a','20');
+round2_step( 1,'%ebx','%ecx','%edx','%eax', '0','-0x72D5B376','20');
 
-round3_step(-1,'%eax','%ebx','%ecx','%edx', '8','0xfffa3942', '4');
-round3_step( 0,'%edx','%eax','%ebx','%ecx','11','0x8771f681','11');
+round3_step(-1,'%eax','%ebx','%ecx','%edx', '8','-0x0005C6BE', '4');
+round3_step( 0,'%edx','%eax','%ebx','%ecx','11','-0x788E097F','11');
 round3_step( 0,'%ecx','%edx','%eax','%ebx','14','0x6d9d6122','16');
-round3_step( 0,'%ebx','%ecx','%edx','%eax', '1','0xfde5380c','23');
-round3_step( 0,'%eax','%ebx','%ecx','%edx', '4','0xa4beea44', '4');
+round3_step( 0,'%ebx','%ecx','%edx','%eax', '1','-0x021AC7F4','23');
+round3_step( 0,'%eax','%ebx','%ecx','%edx', '4','-0x5B4115BC', '4');
 round3_step( 0,'%edx','%eax','%ebx','%ecx', '7','0x4bdecfa9','11');
-round3_step( 0,'%ecx','%edx','%eax','%ebx','10','0xf6bb4b60','16');
-round3_step( 0,'%ebx','%ecx','%edx','%eax','13','0xbebfbc70','23');
+round3_step( 0,'%ecx','%edx','%eax','%ebx','10','-0x0944B4A0','16');
+round3_step( 0,'%ebx','%ecx','%edx','%eax','13','-0x41404390','23');
 round3_step( 0,'%eax','%ebx','%ecx','%edx', '0','0x289b7ec6', '4');
-round3_step( 0,'%edx','%eax','%ebx','%ecx', '3','0xeaa127fa','11');
-round3_step( 0,'%ecx','%edx','%eax','%ebx', '6','0xd4ef3085','16');
+round3_step( 0,'%edx','%eax','%ebx','%ecx', '3','-0x155ED806','11');
+round3_step( 0,'%ecx','%edx','%eax','%ebx', '6','-0x2B10CF7B','16');
 round3_step( 0,'%ebx','%ecx','%edx','%eax', '9', '0x4881d05','23');
-round3_step( 0,'%eax','%ebx','%ecx','%edx','12','0xd9d4d039', '4');
-round3_step( 0,'%edx','%eax','%ebx','%ecx','15','0xe6db99e5','11');
+round3_step( 0,'%eax','%ebx','%ecx','%edx','12','-0x262B2FC7', '4');
+round3_step( 0,'%edx','%eax','%ebx','%ecx','15','-0x1924661B','11');
 round3_step( 0,'%ecx','%edx','%eax','%ebx', '2','0x1fa27cf8','16');
-round3_step( 1,'%ebx','%ecx','%edx','%eax', '0','0xc4ac5665','23');
+round3_step( 1,'%ebx','%ecx','%edx','%eax', '0','-0x3B53A99B','23');
 
-round4_step(-1,'%eax','%ebx','%ecx','%edx', '7','0xf4292244', '6');
+round4_step(-1,'%eax','%ebx','%ecx','%edx', '7','-0x0BD6DDBC', '6');
 round4_step( 0,'%edx','%eax','%ebx','%ecx','14','0x432aff97','10');
-round4_step( 0,'%ecx','%edx','%eax','%ebx', '5','0xab9423a7','15');
-round4_step( 0,'%ebx','%ecx','%edx','%eax','12','0xfc93a039','21');
+round4_step( 0,'%ecx','%edx','%eax','%ebx', '5','-0x546BDC59','15');
+round4_step( 0,'%ebx','%ecx','%edx','%eax','12','-0x036C5FC7','21');
 round4_step( 0,'%eax','%ebx','%ecx','%edx', '3','0x655b59c3', '6');
-round4_step( 0,'%edx','%eax','%ebx','%ecx','10','0x8f0ccc92','10');
-round4_step( 0,'%ecx','%edx','%eax','%ebx', '1','0xffeff47d','15');
-round4_step( 0,'%ebx','%ecx','%edx','%eax', '8','0x85845dd1','21');
+round4_step( 0,'%edx','%eax','%ebx','%ecx','10','-0x70F3336E','10');
+round4_step( 0,'%ecx','%edx','%eax','%ebx', '1','-0x00100B83','15');
+round4_step( 0,'%ebx','%ecx','%edx','%eax', '8','-0x7A7BA22F','21');
 round4_step( 0,'%eax','%ebx','%ecx','%edx','15','0x6fa87e4f', '6');
-round4_step( 0,'%edx','%eax','%ebx','%ecx', '6','0xfe2ce6e0','10');
-round4_step( 0,'%ecx','%edx','%eax','%ebx','13','0xa3014314','15');
+round4_step( 0,'%edx','%eax','%ebx','%ecx', '6','-0x01D31920','10');
+round4_step( 0,'%ecx','%edx','%eax','%ebx','13','-0x5CFEBCEC','15');
 round4_step( 0,'%ebx','%ecx','%edx','%eax', '4','0x4e0811a1','21');
-round4_step( 0,'%eax','%ebx','%ecx','%edx','11','0xf7537e82', '6');
-round4_step( 0,'%edx','%eax','%ebx','%ecx', '2','0xbd3af235','10');
+round4_step( 0,'%eax','%ebx','%ecx','%edx','11','-0x08AC817E', '6');
+round4_step( 0,'%edx','%eax','%ebx','%ecx', '2','-0x42C50DCB','10');
 round4_step( 0,'%ecx','%edx','%eax','%ebx', '9','0x2ad7d2bb','15');
-round4_step( 1,'%ebx','%ecx','%edx','%eax', '0','0xeb86d391','21');
+round4_step( 1,'%ebx','%ecx','%edx','%eax', '0','-0x14792C6F','21');
 $code .= <<EOF;
 	# add old values of A, B, C, D
 	add	%r8d,	%eax
diff -Naur openssl-0.9.8l-orig/crypto/sha/asm/sha1-x86_64.pl openssl-0.9.8l/crypto/sha/asm/sha1-x86_64.pl
--- openssl-0.9.8l-orig/crypto/sha/asm/sha1-x86_64.pl	2007-11-11 07:56:47.000000000 -0600
+++ openssl-0.9.8l/crypto/sha/asm/sha1-x86_64.pl	2009-11-16 14:35:51.000000000 -0600
@@ -143,7 +143,7 @@
 sub BODY_20_39 {
 my ($i,$a,$b,$c,$d,$e,$f)=@_;
 my $j=$i+1;
-my $K=($i<40)?0x6ed9eba1:0xca62c1d6;
+my $K=($i<40)?0x6ed9eba1:-0x359D3E2A;
 $code.=<<___ if ($i<79);
 	lea	$K($xi,$e),$f
 	mov	`4*($j%16)`(%rsp),$xi
@@ -180,7 +180,7 @@
 my ($i,$a,$b,$c,$d,$e,$f)=@_;
 my $j=$i+1;
 $code.=<<___;
-	lea	0x8f1bbcdc($xi,$e),$f
+	lea	-0x70E44324($xi,$e),$f
 	mov	`4*($j%16)`(%rsp),$xi
 	mov	$b,$t0
 	mov	$b,$t1
