From: Jan Engelhardt <jengelh@medozas.de>
Date: 2011-06-01 00:52:07+0200

---
 tc/m_xt.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: iproute2-2.6.37/tc/m_xt.c
===================================================================
--- iproute2-2.6.37.orig/tc/m_xt.c
+++ iproute2-2.6.37/tc/m_xt.c
@@ -343,8 +343,8 @@ print_ipt(struct action_util *au,FILE *
 	return 0;
 }
 
-struct action_util ipt_action_util = {
-        .id = "ipt",
+struct action_util xt_action_util = {
+        .id = "xt",
         .parse_aopt = parse_ipt,
         .print_aopt = print_ipt,
 };
