diff -up iproute2-2.6.29/iproute2-2.6.29/Makefile.opt iproute2-2.6.29/iproute2-2.6.29/Makefile
--- iproute2-2.6.29/Makefile.opt	2008-08-12 14:19:58.000000000 +0200
+++ iproute2-2.6.29/Makefile	2008-08-12 14:21:42.000000000 +0200
@@ -22,7 +22,7 @@ ADDLIB+=ipx_ntop.o ipx_pton.o
 
 CC = gcc
 HOSTCC = gcc
-CCOPTS = -D_GNU_SOURCE -O2 -Wstrict-prototypes -Wall
+CCOPTS = -D_GNU_SOURCE -O2 -Wstrict-prototypes -Wall $(RPM_OPT_FLAGS)
 CFLAGS = $(CCOPTS) -I../include $(DEFINES)
 YACCFLAGS = -d -t -v
 
