diff -up iproute-2.6.29/iproute2-2.6.29/tc/tc_util.c.movelib iproute-2.6.29/iproute2-2.6.29/tc/tc_util.c
--- iproute2-2.6.29/tc/tc_util.c.movelib	2009-01-15 21:25:04.000000000 +0100
+++ iproute2-2.6.29/tc/tc_util.c	2009-02-04 08:58:36.000000000 +0100
@@ -24,8 +24,8 @@
 #include "utils.h"
 #include "tc_util.h"
 
-#ifndef LIBDIR
-#define LIBDIR "/usr/lib/"
+#ifndef SHAREDIR
+#define SHAREDIR "/usr/share/"
 #endif
 
 const char *get_tc_lib(void)
@@ -34,7 +34,7 @@ const char *get_tc_lib(void)
 
 	lib_dir = getenv("TC_LIB_DIR");
 	if (!lib_dir)
-		lib_dir = LIBDIR "/tc/";
+		lib_dir = SHAREDIR "/tc/";
 
 	return lib_dir;
 }
