Index: gftp-2.0.19/lib/parse-dir-listing.c
===================================================================
--- gftp-2.0.19.orig/lib/parse-dir-listing.c
+++ gftp-2.0.19/lib/parse-dir-listing.c
@@ -139,7 +139,12 @@ parse_time (char *str, char **endpos)
             curtime.tm_year = loctime->tm_year - 1;
           else
             curtime.tm_year = loctime->tm_year;
-        }
+
+	  if ( (100*curtime.tm_mon + curtime.tm_mday) >
+	       (100*loctime->tm_mon + loctime->tm_mday) )
+	    curtime.tm_year--;
+
+	}
       else
         tmppos = strptime (str, "%h %d %Y", &curtime);
     }
