Index: gftp-2.0.19/src/gtk/misc-gtk.c
===================================================================
--- gftp-2.0.19.orig/src/gtk/misc-gtk.c
+++ gftp-2.0.19/src/gtk/misc-gtk.c
@@ -757,7 +757,8 @@ add_file_listbox (gftp_window_data * wda
     tempstr = insert_commas (fle->size, NULL, 0);
 
   gtk_clist_set_text (GTK_CLIST (wdata->listbox), clist_num, 2, tempstr);
-  g_free (tempstr);
+  if (tempstr)
+    g_free (tempstr);
 
   if (fle->user)
     gtk_clist_set_text (GTK_CLIST (wdata->listbox), clist_num, 3, fle->user);
