diff -ur amodules/pam_unix/support.h modules/pam_unix/support.h
--- amodules/pam_unix/support.h	2012-08-15 14:08:43.000000000 +0300
+++ modules/pam_unix/support.h	2012-09-16 01:27:44.250142812 +0300
@@ -95,6 +95,11 @@
 
 #define UNIX_DES_CRYPT(ctrl)	(off(UNIX_MD5_PASS,ctrl)&&off(UNIX_BIGCRYPT,ctrl)&&off(UNIX_SHA256_PASS,ctrl)&&off(UNIX_SHA512_PASS,ctrl)&&off(UNIX_BLOWFISH_PASS,ctrl))
 
+/* define not defined YPERR_SUCCESS */
+#ifndef	YPERR_SUCCESS
+#define	YPERR_SUCCESS 0
+#endif	/* YPERR_SUCCESS */
+
 static const UNIX_Ctrls unix_args[UNIX_CTRLS_] =
 {
 /* symbol                  token name          ctrl mask             ctrl     *
