Index: Linux-PAM-1.1.1/configure.in
===================================================================
--- Linux-PAM-1.1.1.orig/configure.in
+++ Linux-PAM-1.1.1/configure.in
@@ -24,15 +24,6 @@ dnl If we use /usr as prefix, use /etc f
         then
                 sysconfdir="/etc"
         fi
-	if test ${libdir} = '${exec_prefix}/lib'
-	then
-		case "`uname -m`" in
-		    x86_64|ppc64|s390x|sparc64)
-			libdir="/lib64" ;;
-		    *)
-			libdir="/lib" ;;
-		esac
-	fi
 	if test ${sbindir} = '${exec_prefix}/sbin'
         then
                 sbindir="/sbin"
