--- Linux-PAM-1.1.6-orig/configure	2012-08-17 11:51:46.000000000 +0300
+++ Linux-PAM-1.1.6/configure	2013-01-23 05:11:03.262654565 +0200
@@ -780,8 +780,6 @@
 LIBSELINUX
 NIS_LIBS
 NIS_CFLAGS
-libtirpc_LIBS
-libtirpc_CFLAGS
 PKG_CONFIG
 HAVE_LIBDB_FALSE
 HAVE_LIBDB_TRUE
@@ -976,9 +974,7 @@
 CPP
 YACC
 YFLAGS
-PKG_CONFIG
-libtirpc_CFLAGS
-libtirpc_LIBS'
+PKG_CONFIG'
 
 
 # Initialize some variables set by options.
@@ -1672,10 +1668,6 @@
               This script will default YFLAGS to the empty string to avoid a
               default value of `-d' given by some make applications.
   PKG_CONFIG  path to pkg-config utility
-  libtirpc_CFLAGS
-              C compiler flags for libtirpc, overriding pkg-config
-  libtirpc_LIBS
-              linker flags for libtirpc, overriding pkg-config
 
 Use these variables to override the choices made by `configure' or to help
 it to find libraries and programs with nonstandard names/locations.
--- Linux-PAM-1.1.6-orig/configure.in	2012-08-17 11:48:24.000000000 +0300
+++ Linux-PAM-1.1.6/configure.in	2013-01-23 05:13:10.113650173 +0200
@@ -452,13 +452,6 @@
   CFLAGS=$old_CFLAGS
   LIBS=$old_LIBS
 
-  dnl if there's libtirpc available, prefer that over the system
-  dnl implementation.
-  PKG_CHECK_MODULES([libtirpc], [libtirpc], [
-    CFLAGS="$CFLAGS $libtirpc_CFLAGS"
-    LIBS="$LIBS $libtirpc_LIBS"
-  ], [:;])
-
   AC_SEARCH_LIBS([yp_get_default_domain], [nsl])
 
   AC_CHECK_FUNCS([yp_get_default_domain yperr_string yp_master yp_bind yp_match yp_unbind])
--- Linux-PAM-1.1.6-orig/Makefile.in	2012-08-17 11:51:46.000000000 +0300
+++ Linux-PAM-1.1.6/Makefile.in	2013-01-23 05:16:56.147642347 +0200
@@ -263,8 +263,6 @@
 libc_cv_fpie = @libc_cv_fpie@
 libdir = @libdir@
 libexecdir = @libexecdir@
-libtirpc_CFLAGS = @libtirpc_CFLAGS@
-libtirpc_LIBS = @libtirpc_LIBS@
 localedir = @localedir@
 localstatedir = @localstatedir@
 lt_ECHO = @lt_ECHO@
