diff -p -up nifticlib-2.0.0/nifticdf/CMakeLists.txt.orig nifticlib-2.0.0/nifticdf/CMakeLists.txt
--- nifticlib-2.0.0/nifticdf/CMakeLists.txt.orig	2010-08-09 21:35:03.688379770 -0300
+++ nifticlib-2.0.0/nifticdf/CMakeLists.txt	2010-08-09 21:35:40.003376642 -0300
@@ -7,6 +7,7 @@ SET(NIFTICDFLIB_SRC nifticdf.c)
 SET(NIFTI_CDFLIB_NAME ${PACKAGE_PREFIX}nifticdf)
 
 ADD_LIBRARY(${NIFTI_CDFLIB_NAME} ${NIFTICDFLIB_SRC} )
+TARGET_LINK_LIBRARIES(${NIFTI_CDFLIB_NAME} -lm)
 
 IF (BUILD_SHARED_LIBS)
   SET_TARGET_PROPERTIES(${NIFTI_CDFLIB_NAME} PROPERTIES ${NIFTI_LIBRARY_PROPERTIES})
