Index: util-linux-2.21-rc2/fdisk/Makefile.am
===================================================================
--- util-linux-2.21-rc2.orig/fdisk/Makefile.am
+++ util-linux-2.21-rc2/fdisk/Makefile.am
@@ -64,9 +64,13 @@ sbin_PROGRAMS += cfdisk
 dist_man_MANS += cfdisk.8
 cfdisk_SOURCES = cfdisk.c $(fdisk_common)
 cfdisk_CFLAGS = $(cflags_blkid)
+if HAVE_TINFO
+cfdisk_LDADD = -ltinfo @NCURSES_LIBS@ $(ldadd_blkid)
+else
 cfdisk_LDADD = @NCURSES_LIBS@ $(ldadd_blkid)
 endif
 endif
+endif
 
 endif # !ARCH_SPARC
 endif # !ARCH_M68K
