diff -Nuar util-linux-2.21.2.orig/sys-utils/klogconsole.c util-linux-2.21.2/sys-utils/klogconsole.c
--- util-linux-2.21.2.orig/sys-utils/klogconsole.c	2002-07-17 00:58:21.000000000 +0300
+++ util-linux-2.21.2/sys-utils/klogconsole.c	2012-06-05 14:14:55.507249257 +0300
@@ -59,7 +59,7 @@
     exit(1);
   }
   if( ioctl( vt, TIOCLINUX, &newvt ) ) {
-    perror("ioctl(TIOCLINUX)");
+    //~ perror("ioctl(TIOCLINUX)");
     exit(1);
   }
   close(vt);
