diff -Nuar util-linux-2.21.2.orig/sys-utils/klogconsole.c util-linux-2.21.2/sys-utils/klogconsole.c
--- util-linux-2.21.2.orig/sys-utils/klogconsole.c	2002-07-17 00:58:21.000000000 +0300
+++ util-linux-2.21.2/sys-utils/klogconsole.c	2012-06-05 14:16:23.369150835 +0300
@@ -46,7 +46,7 @@
   char newvt[2];
   int vt;
 
-  if ((new_console < 1) || (new_console > MAX_CONSOLE)) {
+  if ((new_console < 0) || (new_console > MAX_CONSOLE)) {
     fprintf(stderr,"wrong console number\n");
     exit(1);
   }
