Index: util-linux-2.21-rc2/fdisk/fdisk.c
===================================================================
--- util-linux-2.21-rc2.orig/fdisk/fdisk.c
+++ util-linux-2.21-rc2/fdisk/fdisk.c
@@ -1767,6 +1767,7 @@ static void check_consistency(struct par
 /* compute logical ending (c, h, s) */
 	long2chs(get_start_sect(p) + get_nr_sects(p) - 1, &lec, &leh, &les);
 
+#if 0
 /* Same physical / logical beginning? */
 	if (cylinders <= 1024 && (pbc != lbc || pbh != lbh || pbs != lbs)) {
 		printf(_("Partition %d has different physical/logical "
@@ -1783,7 +1784,6 @@ static void check_consistency(struct par
 		printf(_("logical=(%d, %d, %d)\n"),lec, leh, les);
 	}
 
-#if 0
 /* Beginning on cylinder boundary? */
 	if (pbh != !pbc || pbs != 1) {
 		printf(_("Partition %i does not start on cylinder "
