Index: util-linux-ng-2.13.0.1+git20071121/hwclock/rtc.c
===================================================================
--- util-linux-ng-2.13.0.1+git20071121.orig/hwclock/rtc.c
+++ util-linux-ng-2.13.0.1+git20071121/hwclock/rtc.c
@@ -225,7 +225,7 @@ int ret;
   } else {
     int rc;  /* Return code from ioctl */
     /* Turn on update interrupts (one per second) */
-#if defined(__alpha__) || defined(__sparc__)
+#if defined(__alpha__) || defined(__sparc__) || defined(__x86_64__) || defined(__i386__)
     /* Not all alpha kernels reject RTC_UIE_ON, but probably they should. */
     rc = -1;
     errno = EINVAL;
