diff -Nur tachyon-old/unix/Make-arch tachyon/unix/Make-arch
--- tachyon-old/unix/Make-arch	2009-09-16 19:00:58.118675088 +0300
+++ tachyon/unix/Make-arch	2009-09-16 19:04:58.745676351 +0300
@@ -1174,8 +1174,8 @@
 linux-thr-ogl:
 	$(MAKE) all \
 	"ARCH = linux-thr-ogl" \
-	"CC = cc" \
-	"CFLAGS = -m32 -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux -DTHR -D_REENTRANT $(MISCFLAGS) -DUSEOPENGL $(LINUX_GLX_INCS)" \
+	"CC = pardusCC" \
+	"CFLAGS += -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux -DTHR -D_REENTRANT $(MISCFLAGS) -DUSEOPENGL $(LINUX_GLX_INCS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = touch" \
