Index: glew-1.5.8/Makefile
===================================================================
--- glew-1.5.8.orig/Makefile
+++ glew-1.5.8/Makefile
@@ -172,7 +172,7 @@ bin/$(VISUALINFO.BIN): $(VISUALINFO.BIN.
 
 install.all: install install.mx install.bin
 
-install:     install.include install.lib install.pkgconfig
+install:     install.include install.lib install.pkgconfig install.bin
 
 install.mx:  install.include install.lib.mx install.pkgconfig.mx
 
