diff -up glew-1.6.0/config/Makefile.linux.jx glew-1.6.0/config/Makefile.linux
--- glew-1.6.0/config/Makefile.linux.jx	2011-04-27 12:16:34.000000000 -0400
+++ glew-1.6.0/config/Makefile.linux	2011-06-20 13:21:38.144060270 -0400
@@ -3,14 +3,8 @@ CC = cc
 LD = cc
 PICFLAG = -fPIC
 M_ARCH ?= $(shell uname -m)
-ifeq (x86_64,${M_ARCH})
-LDFLAGS.EXTRA = -L/usr/X11R6/lib64
-LIBDIR = $(GLEW_DEST)/lib64
-else
-LDFLAGS.EXTRA = -L/usr/X11R6/lib
-LIBDIR = $(GLEW_DEST)/lib
-endif
-LDFLAGS.GL = -lXmu -lXi -lGLU -lGL -lXext -lX11
+LIBDIR = $(GLEW_DEST)$(libdir)
+LDFLAGS.GL = -lGL -lX11 -lGLU
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
 NAME = GLEW
diff -up glew-1.6.0/Makefile.jx glew-1.6.0/Makefile
--- glew-1.6.0/Makefile.jx	2011-04-27 12:16:35.000000000 -0400
+++ glew-1.6.0/Makefile	2011-06-20 13:23:05.090373848 -0400
@@ -41,9 +41,9 @@ $(error "Platform '$(SYSTEM)' not suppor
 endif
 
 GLEW_DEST ?= /usr
-BINDIR ?= $(GLEW_DEST)/bin
-LIBDIR ?= $(GLEW_DEST)/lib
-INCDIR ?= $(GLEW_DEST)/include/GL
+BINDIR ?= $(GLEW_DEST)$(bindir)
+LIBDIR ?= $(GLEW_DEST)$(libdir)
+INCDIR ?= $(GLEW_DEST)$(includedir)/GL
 SHARED_OBJ_EXT ?= o
 
 TARDIR = ../glew-$(GLEW_VERSION)
@@ -56,7 +56,7 @@ DIST_SRC_TGZ = glew-$(GLEW_VERSION).tgz
 
 AR = ar
 INSTALL = install
-STRIP = strip
+STRIP = true
 RM = rm -f
 LN = ln -sf
 ifeq ($(MAKECMDGOALS), debug)
@@ -186,7 +186,7 @@ ifeq ($(filter-out mingw% cygwin,$(SYSTE
 	$(INSTALL) -m 0755 lib/$(LIB.SHARED) $(BINDIR)/
 else
 	$(STRIP) -x lib/$(LIB.SHARED)
-	$(INSTALL) -m 0644 lib/$(LIB.SHARED) $(LIBDIR)/
+	$(INSTALL) -m 0755 lib/$(LIB.SHARED) $(LIBDIR)/
 	$(LN) $(LIB.SHARED) $(LIBDIR)/$(LIB.SONAME)
 endif
 # development files
@@ -221,7 +221,7 @@ endif
 
 install.bin: glew.bin
 	$(INSTALL) -d -m 0755 $(BINDIR)
-	$(INSTALL) -s -m 0755 bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN) $(BINDIR)/
+	$(INSTALL) -m 0755 bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN) $(BINDIR)/
 
 install.include:
 	$(INSTALL) -d -m 0755 $(INCDIR)
