--- gnome-mime-data-2.4.1/gnome-vfs.keys.in.default-applications	2004-03-19 16:44:45.000000000 +0000
+++ gnome-mime-data-2.4.1/gnome-vfs.keys.in	2004-03-19 16:44:54.000000000 +0000
@@ -29,6 +29,7 @@
 application/msword
 	_description=Microsoft Word document
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,abiword,staroffice,applix
 	short_list_application_ids_for_intermediate_user_level=openoffice,abiword,staroffice,applix
 	short_list_application_ids_for_advanced_user_level=openoffice,abiword,staroffice,applix
@@ -91,6 +92,7 @@
 application/postscript
 	_description=PostScript document
 	default_action_type=component
+        default_application_id=ggv
 	short_list_component_iids=OAFIID:GNOME_GGV_Control
 	short_list_component_iids_for_novice_user_level=OAFIID:GNOME_GGV_Control
 	short_list_component_iids_for_intermediate_user_level=OAFIID:GNOME_GGV_Control
@@ -113,6 +115,7 @@
 application/rtf
 	_description=Rich text document
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,abiword,staroffice,applix
 	short_list_application_ids_for_intermediate_user_level=openoffice,abiword,staroffice,applix
 	short_list_application_ids_for_advanced_user_level=openoffice,abiword,staroffice,applix
@@ -137,6 +140,7 @@
 	_description=Microsoft Excel spreadsheet
 	icon_filename=i-spreadsheet
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,gnumeric,staroffice,applix
 	short_list_application_ids_for_intermediate_user_level=openoffice,gnumeric,staroffice,applix
 	short_list_application_ids_for_advanced_user_level=openoffice,gnumeric,staroffice,applix
@@ -146,6 +150,7 @@
 application/vnd.ms-powerpoint
 	_description=Microsoft PowerPoint document
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -166,6 +171,7 @@
 application/vnd.stardivision.calc
 	_description=StarCalc spreadsheet
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -175,6 +181,7 @@
 application/vnd.stardivision.chart
 	_description=StarChart chart
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -184,6 +191,7 @@
 application/vnd.stardivision.draw
 	_description=StarDraw drawing
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -193,6 +201,7 @@
 application/vnd.stardivision.impress
 	_description=StarImpress presentation
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -202,6 +211,7 @@
 application/vnd.stardivision.mail
 	_description=StarMail file
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -211,6 +221,7 @@
 application/vnd.stardivision.math
 	_description=StarMath document
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -220,6 +231,7 @@
 application/vnd.stardivision.writer
 	_description=StarWriter document
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -231,6 +243,7 @@
 application/vnd.sun.xml.calc
 	_description=OpenOffice.org spreadsheet
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -240,6 +253,7 @@
 application/vnd.sun.xml.calc.template
 	_description=OpenOffice.org spreadsheet template
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -249,6 +263,7 @@
 application/vnd.sun.xml.draw
 	_description=OpenOffice.org drawing
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -258,6 +273,7 @@
 application/vnd.sun.xml.draw.template
 	_description=OpenOffice.org drawing template
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -267,6 +283,7 @@
 application/vnd.sun.xml.impress
 	_description=OpenOffice.org Impress presentation
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -276,6 +293,7 @@
 application/vnd.sun.xml.impress.template
 	_description=OpenOffice.org Impress presentation template
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -285,6 +303,7 @@
 application/vnd.sun.xml.math
 	_description=OpenOffice.org Math document
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -294,6 +313,7 @@
 application/vnd.sun.xml.writer
 	_description=OpenOffice.org Writer document
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -303,6 +323,7 @@
 application/vnd.sun.xml.writer.global
 	_description=OpenOffice.org Writer global document
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -312,6 +333,7 @@
 application/vnd.sun.xml.writer.template
 	_description=OpenOffice.org Writer template
 	default_action_type=application
+        default_application_id=openoffice
 	short_list_application_ids_for_novice_user_level=openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=openoffice,staroffice
@@ -322,7 +344,8 @@
 application/x-abiword
 	_description=AbiWord document
 	default_action_type=application
-	short_list_application_ids_for_novice_user_level=abiword,openoffice,staroffice
+        default_application_id=abiword
+        short_list_application_ids_for_novice_user_level=abiword,openoffice,staroffice
 	short_list_application_ids_for_intermediate_user_level=abiword,openoffice,staroffice
 	short_list_application_ids_for_advanced_user_level=abiword,openoffice,staroffice
 	_category=Documents/Word Processor
@@ -841,6 +864,7 @@
 application/x-ogg
 	_description=Ogg audio
 	default_action_type=application
+	default_application_id=xmms
 	short_list_application_ids_for_novice_user_level=xmms,freeamp
 	short_list_application_ids_for_intermediate_user_level=xmms,freeamp
 	short_list_application_ids_for_advanced_user_level=xmms,freeamp
@@ -1093,6 +1117,7 @@
 	_description=MP3 audio
 	icon_filename=i-music
 	default_action_type=application
+	default_application_id=xmms
 	short_list_application_ids_for_novice_user_level=xmms,freeamp
 	short_list_application_ids_for_intermediate_user_level=xmms,freeamp
 	short_list_application_ids_for_advanced_user_level=xmms,freeamp
@@ -1170,6 +1195,7 @@
 	_description=MP3 audio playlist
 	icon_filename=i-music
 	default_action_type=application
+	default_application_id=xmms
 	short_list_application_ids_for_novice_user_level=xmms-no-uris,freeamp-no-uris
 	short_list_application_ids_for_intermediate_user_level=xmms-no-uris,freeamp-no-uris
 	short_list_application_ids_for_advanced_user_level=xmms-no-uris,freeamp-no-uris
@@ -1188,6 +1214,7 @@
 audio/x-mpegurl
 	_description=Playlist
 	default_action_type=application
+	default_application_id=xmms
 	short_list_application_ids_for_novice_user_level=xmms-no-uris,freeamp-no-uris
 	short_list_application_ids_for_intermediate_user_level=xmms-no-uris,freeamp-no-uris
 	short_list_application_ids_for_advanced_user_level=xmms-no-uris,freeamp-no-uris
@@ -1310,6 +1337,7 @@
 image/gif
 	_description=GIF image
 	default_action_type=component
+	default_application_id=eog
 	short_list_component_iids=OAFIID:GNOME_EOG_Control
 	short_list_component_iids_for_novice_user_level=OAFIID:GNOME_EOG_Control
 	short_list_component_iids_for_intermediate_user_level=OAFIID:GNOME_EOG_Control
@@ -1326,6 +1354,7 @@
 image/jpeg
 	_description=JPEG image
 	default_action_type=component
+	default_application_id=eog
 	short_list_component_iids=OAFIID:GNOME_EOG_Control
 	short_list_component_iids_for_novice_user_level=OAFIID:GNOME_EOG_Control
 	short_list_component_iids_for_intermediate_user_level=OAFIID:GNOME_EOG_Control
@@ -1365,6 +1394,7 @@
 image/png
 	_description=PNG image
 	default_action_type=component
+	default_application_id=eog
 	short_list_component_iids=OAFIID:GNOME_EOG_Control
 	short_list_component_iids_for_novice_user_level=OAFIID:GNOME_EOG_Control
 	short_list_component_iids_for_intermediate_user_level=OAFIID:GNOME_EOG_Control
@@ -1377,6 +1407,7 @@
 image/svg
 	_description=SVG art
 	default_action_type=application
+	default_application_id=eog
 	short_list_application_ids_for_novice_user_level=sodipodi
 	short_list_application_ids_for_intermediate_user_level=sodipodi
 	short_list_application_ids_for_advanced_user_level=sodipodi
@@ -1393,6 +1424,7 @@
 image/tiff
 	_description=TIFF image
 	default_action_type=component
+	default_application_id=eog
 	short_list_component_iids=OAFIID:GNOME_EOG_Control
 	short_list_component_iids_for_novice_user_level=OAFIID:GNOME_EOG_Control
 	short_list_component_iids_for_intermediate_user_level=OAFIID:GNOME_EOG_Control
@@ -1669,14 +1701,16 @@
 image/x-wmf
 	_description=Windows metafile graphics
 	default_action_type=application
-	short_list_application_ids_for_novice_user_level=killustrator
-	short_list_application_ids_for_intermediate_user_level=killustrator
-	short_list_application_ids_for_advanced_user_level=killustrator
+        default_application_id=openoffice
+	short_list_application_ids_for_novice_user_level=openoffice,killustrator
+	short_list_application_ids_for_intermediate_user_level=openoffice,killustrator
+	short_list_application_ids_for_advanced_user_level=openoffice,killustrator
 	_category=Images
 
 image/x-xbitmap
 	_description=X bitmap image
 	default_action_type=application
+	default_application_id=eog
 	short_list_application_ids_for_novice_user_level=eog,gimp
 	short_list_application_ids_for_intermediate_user_level=eog,gimp
 	short_list_application_ids_for_advanced_user_level=eog,gimp
@@ -1800,9 +1834,9 @@
 	short_list_component_iids_for_novice_user_level=OAFIID:nautilus_text_view:fa466311-17c1-435c-8231-c9fc434b6437
 	short_list_component_iids_for_intermediate_user_level=OAFIID:nautilus_text_view:fa466311-17c1-435c-8231-c9fc434b6437
 	short_list_component_iids_for_advanced_user_level=OAFIID:nautilus_text_view:fa466311-17c1-435c-8231-c9fc434b6437
-	short_list_application_ids_for_novice_user_level=gedit,pico
-	short_list_application_ids_for_intermediate_user_level=xemacs,emacs,nedit,gedit,xfte,pico
-	short_list_application_ids_for_advanced_user_level=xemacs,gvim,emacs,vim,vi,nedit,gedit,xfte,pico
+	short_list_application_ids_for_novice_user_level=gedit,pico,openoffice
+	short_list_application_ids_for_intermediate_user_level=xemacs,emacs,nedit,gedit,xfte,pico,openoffice
+	short_list_application_ids_for_advanced_user_level=xemacs,gvim,emacs,vi,nedit,gedit,xfte,pico,openoffice
 	_category=Documents/Plain Text
 
 # also application/x-abiword
@@ -1867,13 +1901,14 @@
 	can_be_executable=TRUE
 	_description=Plain text document
 	default_action_type=component
+	default_application_id=gedit
 	short_list_component_iids=OAFIID:Nautilus_Text_View
 	short_list_component_iids_for_novice_user_level=OAFIID:nautilus_text_view:fa466311-17c1-435c-8231-c9fc434b6437
 	short_list_component_iids_for_intermediate_user_level=OAFIID:nautilus_text_view:fa466311-17c1-435c-8231-c9fc434b6437
 	short_list_component_iids_for_advanced_user_level=OAFIID:nautilus_text_view:fa466311-17c1-435c-8231-c9fc434b6437
-	short_list_application_ids_for_novice_user_level=gedit,pico
-	short_list_application_ids_for_intermediate_user_level=xemacs,emacs,nedit,gedit,xfte,pico
-	short_list_application_ids_for_advanced_user_level=xemacs,gvim,emacs,vim,vi,nedit,gedit,xfte,pico
+	short_list_application_ids_for_novice_user_level=gedit,pico,openoffice
+	short_list_application_ids_for_intermediate_user_level=xemacs,emacs,nedit,gedit,xfte,pico,openoffice
+	short_list_application_ids_for_advanced_user_level=xemacs,gvim,emacs,vi,nedit,gedit,xfte,pico,openoffice
 	_category=Documents/Plain Text
 	use_category_default=yes
 
@@ -1939,9 +1974,10 @@
 	_description=Comma-separated text document
 	icon_filename=i-spreadsheet
 	default_action_type=application
-	short_list_application_ids_for_novice_user_level=gnumeric,kspread
-	short_list_application_ids_for_intermediate_user_level=gnumeric,kspread
-	short_list_application_ids_for_advanced_user_level=gnumeric,kspread
+        default_application_id=openoffice
+	short_list_application_ids_for_novice_user_level=openoffice,gnumeric,kspread
+	short_list_application_ids_for_intermediate_user_level=openoffice,gnumeric,kspread
+	short_list_application_ids_for_advanced_user_level=openoffice,gnumeric,kspread
 	_category=Documents/Spreadsheet
 	use_category_default=yes
 
@@ -2152,7 +2188,8 @@
 
 text/xml
 	_description=XML document
-	default_action_type=component
+	default_action_type=application
+        default_application_id=mozilla
 	short_list_component_iids=OAFIID:nautilus_mozilla_content_view:1ee70717-57bf-4079-aae5-922abdd576b1,OAFIID:Nautilus_Text_View
 	short_list_component_iids_for_novice_user_level=OAFIID:nautilus_mozilla_content_view:1ee70717-57bf-4079-aae5-922abdd576b1,OAFIID:nautilus_text_view:fa466311-17c1-435c-8231-c9fc434b6437
 	short_list_component_iids_for_intermediate_user_level=OAFIID:nautilus_mozilla_content_view:1ee70717-57bf-4079-aae5-922abdd576b1,OAFIID:nautilus_text_view:fa466311-17c1-435c-8231-c9fc434b6437
