diff -up gnome-mime-data-2.18.0/gnome-vfs.keys.in.openoffice gnome-mime-data-2.18.0/gnome-vfs.keys.in
--- gnome-mime-data-2.18.0/gnome-vfs.keys.in.openoffice	2007-04-03 13:24:30.000000000 -0400
+++ gnome-mime-data-2.18.0/gnome-vfs.keys.in	2008-07-23 12:32:52.000000000 -0400
@@ -27,9 +27,9 @@ application/mime-type-test
 application/msword
 	_description=Microsoft Word document
 	default_action_type=application
-	short_list_application_ids_for_novice_user_level=abiword,openoffice,staroffice,applix
-	short_list_application_ids_for_intermediate_user_level=abiword,openoffice,staroffice,applix
-	short_list_application_ids_for_advanced_user_level=abiword,openoffice,staroffice,applix
+	short_list_application_ids_for_novice_user_level=openoffice,abiword,staroffice,applix
+	short_list_application_ids_for_intermediate_user_level=openoffice,abiword,staroffice,applix
+	short_list_application_ids_for_advanced_user_level=openoffice,abiword,staroffice,applix
 	_category=Documents/Word Processor
 	use_category_default=yes
 
@@ -111,9 +111,9 @@ application/qif
 application/rtf
 	_description=Rich text document
 	default_action_type=application
-	short_list_application_ids_for_novice_user_level=abiword,openoffice,staroffice,applix
-	short_list_application_ids_for_intermediate_user_level=abiword,openoffice,staroffice,applix
-	short_list_application_ids_for_advanced_user_level=abiword,openoffice,staroffice,applix
+	short_list_application_ids_for_novice_user_level=openoffice,abiword,staroffice,applix
+	short_list_application_ids_for_intermediate_user_level=openoffice,abiword,staroffice,applix
+	short_list_application_ids_for_advanced_user_level=openoffice,abiword,staroffice,applix
 	_category=Documents/Word Processor
 	use_category_default=yes
 
@@ -135,9 +135,9 @@ application/vnd.ms-excel
 	_description=Microsoft Excel spreadsheet
 	icon_filename=i-spreadsheet
 	default_action_type=application
-	short_list_application_ids_for_novice_user_level=gnumeric,openoffice,staroffice,applix
-	short_list_application_ids_for_intermediate_user_level=gnumeric,openoffice,staroffice,applix
-	short_list_application_ids_for_advanced_user_level=gnumeric,openoffice,staroffice,applix
+	short_list_application_ids_for_novice_user_level=openoffice,gnumeric,staroffice,applix
+	short_list_application_ids_for_intermediate_user_level=openoffice,gnumeric,staroffice,applix
+	short_list_application_ids_for_advanced_user_level=openoffice,gnumeric,staroffice,applix
 	_category=Documents/Spreadsheet
 	use_category_default=yes
 
