--- libvalhalla-2.0.0/configure.orig	2012-06-30 21:15:31.802114548 +0100
+++ libvalhalla-2.0.0/configure	2012-06-30 21:16:07.988482771 +0100
@@ -1057,7 +1057,7 @@
 fi
 
 echolog "Checking for libavformat ..."
-check_lib_pkgcfg libavformat libavformat/avformat.h av_metadata_get -lavformat || die "Error, can't find libavformat !"
+check_lib_pkgcfg libavformat libavformat/avformat.h av_register_all -lavformat || die "Error, can't find libavformat !"
 
 
 #################################################
