Index: libvalhalla-2.0.0/src/grabber_ffmpeg.c
===================================================================
--- libvalhalla-2.0.0/src/grabber_ffmpeg.c
+++ libvalhalla-2.0.0/src/grabber_ffmpeg.c	2014-11-14 17:49:10.531685146 +0100
@@ -64,7 +64,7 @@
   unsigned int i;
   unsigned int audio_streams = 0, video_streams = 0, sub_streams = 0;
 
-  res = av_find_stream_info (ctx);
+  res = avformat_find_stream_info (ctx, NULL);
   if (res < 0)
   {
     vh_log (VALHALLA_MSG_VERBOSE,
@@ -214,7 +214,7 @@
   res = grabber_ffmpeg_properties_get (ffmpeg, ctx, data);
   /* TODO: res = grabber_ffmpeg_snapshot (ctx, data, pos); */
 
-  av_close_input_file (ctx);
+  avformat_close_input (&ctx);
   return res;
 }
 
Index: libvalhalla-2.0.0/src/parser.c
===================================================================
--- libvalhalla-2.0.0/src/parser.c
+++ libvalhalla-2.0.0/src/parser.c	2014-11-14 17:46:28.605374135 +0100
@@ -377,7 +377,7 @@
   data->file.type = parser_stream_info (ctx);
   data->meta_parser = parser_metadata_get (parser, ctx, data->file.path);
 
-  av_close_input_file (ctx);
+  avformat_close_input (&ctx);
 }
 
 static void *
