--- libvalhalla-2.0.0/src/lavf_utils.c.0011	2010-10-03 17:36:50.000000000 +0100
+++ libvalhalla-2.0.0/src/lavf_utils.c	2012-06-30 21:45:22.623004988 +0100
@@ -203,7 +203,6 @@
   int res;
   const char *name;
   AVFormatContext   *ctx;
-  AVFormatParameters ap;
   AVInputFormat     *fmt = NULL;
 
   ctx = avformat_alloc_context ();
@@ -229,10 +228,8 @@
       fmt = NULL;
   }
 
-  memset (&ap, 0, sizeof (ap));
-  ap.prealloced_context = 1;
-
-  res = av_open_input_file (&ctx, file, fmt, 0, &ap);
+  ctx->pb = &ctx;
+  res = avformat_open_input (&ctx, file, fmt, NULL);
   if (res)
   {
     vh_log (VALHALLA_MSG_WARNING,
