Index: giflib-4.1.6/lib/egif_lib.c
===================================================================
--- giflib-4.1.6.orig/lib/egif_lib.c
+++ giflib-4.1.6/lib/egif_lib.c
@@ -64,11 +64,11 @@
 /* #define DEBUG_NO_PREFIX                  Dump only compressed data. */
 
 /* Masks given codes to BitsPerPixel, to make sure all codes are in range: */
-static GifPixelType CodeMask[] = {
+static const GifPixelType CodeMask[] = {
     0x00, 0x01, 0x03, 0x07, 0x0f, 0x1f, 0x3f, 0x7f, 0xff
 };
 
-static char GifVersionPrefix[GIF_STAMP_LEN + 1] = GIF87_STAMP;
+static const char GifVersionPrefix[GIF_STAMP_LEN + 1] = GIF87_STAMP;
 
 #define WRITE(_gif,_buf,_len)   \
   (((GifFilePrivateType*)_gif->Private)->Write ?    \
Index: giflib-4.1.6/lib/gif_font.c
===================================================================
--- giflib-4.1.6.orig/lib/gif_font.c
+++ giflib-4.1.6/lib/gif_font.c
@@ -26,7 +26,7 @@
  * first.  The most significant bit of each constant is the leftmost bit of
  * the scan line.
  */
-unsigned char AsciiTable[][GIF_FONT_WIDTH] = {
+const unsigned char AsciiTable[][GIF_FONT_WIDTH] = {
     {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00},    /* Ascii 0 */
     {0x3c, 0x42, 0xa5, 0x81, 0xbd, 0x42, 0x3c, 0x00},    /* Ascii 1 */
     {0x3c, 0x7e, 0xdb, 0xff, 0xc3, 0x7e, 0x3c, 0x00},    /* Ascii 2 */
--- giflib-4.1.6/lib/gif_lib.h	2005-10-10 09:22:21.000000000 +0300
+++ giflib-4.1.6/lib/gif_lib.h	2008-01-23 19:33:35.000000000 +0200
@@ -307,7 +307,7 @@
 
 #define GIF_FONT_WIDTH  8
 #define GIF_FONT_HEIGHT 8
-extern unsigned char AsciiTable[][GIF_FONT_WIDTH];
+extern const unsigned char AsciiTable[][GIF_FONT_WIDTH];
 
 #ifdef _WIN32
     extern void DrawGifText(SavedImage * Image,
