diff -up dhcp-4.2.0-P2/omapip/protocol.c.omapi dhcp-4.2.0-P2/omapip/protocol.c
--- dhcp-4.2.0-P2/omapip/protocol.c.omapi	2009-11-20 02:49:02.000000000 +0100
+++ dhcp-4.2.0-P2/omapip/protocol.c	2011-01-03 16:30:25.000000000 +0100
@@ -386,7 +386,7 @@ isc_result_t omapi_protocol_signal_handl
 	/* Should only receive these when opening the initial authenticator. */
 	if (!strcmp (name, "status")) {
 		status = va_arg (ap, isc_result_t);
-		if (status != ISC_R_SUCCESS) {
+		if ((status != ISC_R_SUCCESS) && (status >= ISC_RESULTCLASS_DHCP)) {
 			omapi_signal_in (h -> inner, "status", status,
 					 (omapi_object_t *)0);
 			omapi_disconnect (p -> outer, 1);
