diff -p -up wpa_supplicant-0.6.3/wpa_supplicant/config_ssid.h.WEP232 wpa_supplicant-0.6.3/wpa_supplicant/config_ssid.h
--- wpa_supplicant-0.6.3/wpa_supplicant/config_ssid.h.WEP232	2008-02-23 03:45:24.000000000 +0100
+++ wpa_supplicant-0.6.3/wpa_supplicant/config_ssid.h	2008-03-31 22:28:29.000000000 +0200
@@ -189,7 +189,7 @@ struct wpa_ssid {
 #endif /* IEEE8021X_EAPOL */
 
 #define NUM_WEP_KEYS 4
-#define MAX_WEP_KEY_LEN 16
+#define MAX_WEP_KEY_LEN 32
 	/**
 	 * wep_key - WEP keys
 	 */
