--- config.sample  2010-02-01 01:13:20.000000000 +0200
+++ config.sample  2010-07-29 21:05:22.343661754 +0300
@@ -41,6 +41,9 @@
 # socksParentProxy = "localhost:9050"
 # socksProxyType = socks5
 
+# Uncomment this if you want to scrub private information from the log:
+
+# scrubLogs = true
 
 ### Memory
 ### ******
@@ -150,3 +153,8 @@
 # Vary header (this is not a good idea):
 
 # mindlesslyCacheVary = true
+
+# Uncomment this if you want to add a no-transform directive to all
+# outgoing requests.
+
+# alwaysAddNoTransform = true
