Index: pam-face-authentication-0.3/qt-facetrainer-desktop.sh
===================================================================
--- pam-face-authentication-0.3.orig/qt-facetrainer-desktop.sh
+++ pam-face-authentication-0.3/qt-facetrainer-desktop.sh
@@ -1,13 +1,15 @@
 #!/bin/sh
 cat <<EOF
 [Desktop Entry]
-Version=0.3
 Type=Application
 Name=Qt Face Trainer
+Name[tr]=Qt Yüz Eğitici
 GenericName=Face Trainer
+GenericName[tr]=Yüz Eğitici
 Comment=Face Trainer for PAM Face Authentication
+Comment[tr]=PAM Yüz Tanıma için gerekli yüz eğitici
 TryExec=$1/bin/qt-facetrainer
-Exec=$1/bin/qt-facetrainer &
-Categories=Application;
-Icon=$1/share/icons/pfa-logo.png
+Exec=$1/bin/qt-facetrainer
+Categories=Utility;
+Icon=$1/share/icons/pfa-logo
 EOF
