--- data/ui/uifiles/faceTrainer.ui.orig	2011-03-16 15:52:43.336000054 +0200
+++ data/ui/uifiles/faceTrainer.ui	2011-03-16 16:09:26.998000053 +0200
@@ -29,7 +29,7 @@
    <string>Face Trainer</string>
   </property>
   <property name="windowIcon">
-   <iconset resource="../../../graphics.qrc">
+   <iconset resource="../../var/pisi/pam-face-authentication-0.3-1/work/pam-face-authentication-0.3/graphics.qrc">
     <normaloff>:/data/ui/images/pfa-logo.png</normaloff>:/data/ui/images/pfa-logo.png</iconset>
   </property>
   <property name="styleSheet">
@@ -406,7 +406,7 @@
        <string>Capture</string>
       </property>
       <property name="icon">
-       <iconset resource="../../../graphics.qrc">
+       <iconset resource="../../var/pisi/pam-face-authentication-0.3-1/work/pam-face-authentication-0.3/graphics.qrc">
         <normaloff>:/data/ui/images/record.png</normaloff>:/data/ui/images/record.png</iconset>
       </property>
       <property name="iconSize">
@@ -564,7 +564,7 @@
        <string>About this Software !</string>
       </property>
       <property name="icon">
-       <iconset resource="../../../graphics.qrc">
+       <iconset resource="../../var/pisi/pam-face-authentication-0.3-1/work/pam-face-authentication-0.3/graphics.qrc">
         <normaloff>:/data/ui/images/about.png</normaloff>:/data/ui/images/about.png</iconset>
       </property>
      </widget>
@@ -800,8 +800,8 @@
      <widget class="QPushButton" name="pb_finish">
       <property name="geometry">
        <rect>
-        <x>612</x>
-        <y>411</y>
+        <x>610</x>
+        <y>490</y>
         <width>101</width>
         <height>40</height>
        </rect>
@@ -821,7 +821,7 @@
         <x>36</x>
         <y>176</y>
         <width>677</width>
-        <height>201</height>
+        <height>141</height>
        </rect>
       </property>
       <property name="minimumSize">
@@ -833,7 +833,7 @@
       <property name="font">
        <font>
         <family>Serif</family>
-        <pointsize>13</pointsize>
+        <pointsize>12</pointsize>
         <weight>50</weight>
         <italic>true</italic>
         <bold>false</bold>
@@ -846,11 +846,7 @@
        <string notr="true">color: rgb(126, 126, 126);</string>
       </property>
       <property name="text">
-       <string>Thank you for using our application, to be a part of this project , visit http://code.google.com/p/pam-face-authentication/ , we hope to see you there!
-
-
-regards
-Developers</string>
+       <string>Thank you for using our application, to be a part of this project , visit http://code.google.com/p/pam-face-authentication/ , we hope to see you there!</string>
       </property>
       <property name="textFormat">
        <enum>Qt::PlainText</enum>
@@ -868,12 +864,173 @@
        <number>13</number>
       </property>
      </widget>
+     <widget class="QLabel" name="label">
+      <property name="geometry">
+       <rect>
+        <x>50</x>
+        <y>290</y>
+        <width>631</width>
+        <height>211</height>
+       </rect>
+      </property>
+      <property name="palette">
+       <palette>
+        <active>
+         <colorrole role="WindowText">
+          <brush brushstyle="SolidPattern">
+           <color alpha="255">
+            <red>126</red>
+            <green>126</green>
+            <blue>126</blue>
+           </color>
+          </brush>
+         </colorrole>
+         <colorrole role="Midlight">
+          <brush brushstyle="SolidPattern">
+           <color alpha="255">
+            <red>226</red>
+            <green>226</green>
+            <blue>226</blue>
+           </color>
+          </brush>
+         </colorrole>
+         <colorrole role="Text">
+          <brush brushstyle="SolidPattern">
+           <color alpha="255">
+            <red>126</red>
+            <green>126</green>
+            <blue>126</blue>
+           </color>
+          </brush>
+         </colorrole>
+         <colorrole role="ButtonText">
+          <brush brushstyle="SolidPattern">
+           <color alpha="255">
+            <red>126</red>
+            <green>126</green>
+            <blue>126</blue>
+           </color>
+          </brush>
+         </colorrole>
+        </active>
+        <inactive>
+         <colorrole role="WindowText">
+          <brush brushstyle="SolidPattern">
+           <color alpha="255">
+            <red>126</red>
+            <green>126</green>
+            <blue>126</blue>
+           </color>
+          </brush>
+         </colorrole>
+         <colorrole role="Midlight">
+          <brush brushstyle="SolidPattern">
+           <color alpha="255">
+            <red>226</red>
+            <green>226</green>
+            <blue>226</blue>
+           </color>
+          </brush>
+         </colorrole>
+         <colorrole role="Text">
+          <brush brushstyle="SolidPattern">
+           <color alpha="255">
+            <red>126</red>
+            <green>126</green>
+            <blue>126</blue>
+           </color>
+          </brush>
+         </colorrole>
+         <colorrole role="ButtonText">
+          <brush brushstyle="SolidPattern">
+           <color alpha="255">
+            <red>126</red>
+            <green>126</green>
+            <blue>126</blue>
+           </color>
+          </brush>
+         </colorrole>
+        </inactive>
+        <disabled>
+         <colorrole role="WindowText">
+          <brush brushstyle="SolidPattern">
+           <color alpha="255">
+            <red>126</red>
+            <green>126</green>
+            <blue>126</blue>
+           </color>
+          </brush>
+         </colorrole>
+         <colorrole role="Midlight">
+          <brush brushstyle="SolidPattern">
+           <color alpha="255">
+            <red>226</red>
+            <green>226</green>
+            <blue>226</blue>
+           </color>
+          </brush>
+         </colorrole>
+         <colorrole role="Text">
+          <brush brushstyle="SolidPattern">
+           <color alpha="255">
+            <red>126</red>
+            <green>126</green>
+            <blue>126</blue>
+           </color>
+          </brush>
+         </colorrole>
+         <colorrole role="ButtonText">
+          <brush brushstyle="SolidPattern">
+           <color alpha="255">
+            <red>126</red>
+            <green>126</green>
+            <blue>126</blue>
+           </color>
+          </brush>
+         </colorrole>
+        </disabled>
+       </palette>
+      </property>
+      <property name="font">
+       <font>
+        <family>Serif</family>
+        <pointsize>12</pointsize>
+        <italic>true</italic>
+       </font>
+      </property>
+      <property name="styleSheet">
+       <string notr="true">color: rgb(126, 126, 126);</string>
+      </property>
+      <property name="text">
+       <string>If you want to configure KDM to use KDM greeter plugin
+Install kdm-plugin-pam-face-authentication
+Edit /usr/share/kde4/config/kdm/kdmrc
+search for PluginsLogin, uncomment the line and add
+     PluginsLogin=classic, faceauthenticate
+
+
+regards
+Developers</string>
+      </property>
+      <property name="textFormat">
+       <enum>Qt::PlainText</enum>
+      </property>
+      <property name="scaledContents">
+       <bool>true</bool>
+      </property>
+      <property name="wordWrap">
+       <bool>true</bool>
+      </property>
+      <property name="margin">
+       <number>0</number>
+      </property>
+     </widget>
     </widget>
    </widget>
   </widget>
  </widget>
  <resources>
-  <include location="../../../graphics.qrc"/>
+  <include location="../../var/pisi/pam-face-authentication-0.3-1/work/pam-face-authentication-0.3/graphics.qrc"/>
  </resources>
  <connections>
   <connection>
