diff -rupN dssi-1.1.0.old/doc/jack-dssi-host.1 dssi-1.1.0/doc/jack-dssi-host.1
--- dssi-1.1.0.old/doc/jack-dssi-host.1	2010-09-19 02:14:42.000000000 -0400
+++ dssi-1.1.0/doc/jack-dssi-host.1	2010-09-26 00:29:23.000000000 -0400
@@ -94,7 +94,7 @@ by the environment variable
 .B DSSI_PATH,
 which is a colon-separated list of directories. If
 .B DSSI_PATH
-is not set, a default search path of /usr/lib/dssi,
+is not set, a default search path of /usr/lib64/dssi, /usr/local/lib64/dssi, /usr/lib/dssi,
 /usr/local/lib/dssi, and (assuming the environment variable HOME is
 set,) $HOME/.dssi is used.
 .br
diff -rupN dssi-1.1.0.old/examples/dssi_analyse_plugin.c dssi-1.1.0/examples/dssi_analyse_plugin.c
--- dssi-1.1.0.old/examples/dssi_analyse_plugin.c	2010-06-27 15:30:09.000000000 -0400
+++ dssi-1.1.0/examples/dssi_analyse_plugin.c	2010-09-26 00:32:00.000000000 -0400
@@ -408,7 +408,7 @@ main(int argc, char *argv[])
         char *pathtmp, *element;
 
         if (!dssi_path) {
-            dssi_path = "/usr/local/lib/dssi:/usr/lib/dssi";
+            dssi_path = "/usr/local/lib64/dssi:/usr/lib64/dssi:/usr/local/lib/dssi:/usr/lib/dssi";
             fprintf(stderr, "Warning: DSSI_PATH not set, defaulting to '%s'\n", dssi_path);
         } else if (verbose) {
             printf("Searching DSSI_PATH '%s'...\n", dssi_path);
diff -rupN dssi-1.1.0.old/examples/dssi_list_plugins.c dssi-1.1.0/examples/dssi_list_plugins.c
--- dssi-1.1.0.old/examples/dssi_list_plugins.c	2010-06-27 15:30:09.000000000 -0400
+++ dssi-1.1.0/examples/dssi_list_plugins.c	2010-09-26 00:32:20.000000000 -0400
@@ -200,7 +200,7 @@ main(int argc, char *argv[])
 
     path = getenv("DSSI_PATH");
     if (!path) {
-        path = "/usr/local/lib/dssi:/usr/lib/dssi";
+        path = "/usr/local/lib64/dssi:/usr/lib64/dssi:/usr/local/lib/dssi:/usr/lib/dssi";
         fprintf(stderr, "warning: DSSI_PATH not set, defaulting to '%s'\n", path);
     }
 
diff -rupN dssi-1.1.0.old/jack-dssi-host/jack-dssi-host.c dssi-1.1.0/jack-dssi-host/jack-dssi-host.c
--- dssi-1.1.0.old/jack-dssi-host/jack-dssi-host.c	2010-09-19 01:12:15.000000000 -0400
+++ dssi-1.1.0/jack-dssi-host/jack-dssi-host.c	2010-09-26 00:29:23.000000000 -0400
@@ -492,10 +492,10 @@ load(const char *dllName, void **dll, in
 	if (!defaultDssiPath) {
 	    const char *home = getenv("HOME");
 	    if (home) {
-		defaultDssiPath = malloc(strlen(home) + 60);
-		sprintf(defaultDssiPath, "/usr/local/lib/dssi:/usr/lib/dssi:%s/.dssi", home);
+		defaultDssiPath = malloc(strlen(home) + 100);
+		sprintf(defaultDssiPath, "/usr/local/lib64/dssi:/usr/lib64/dssi:/usr/local/lib/dssi:/usr/lib/dssi:%s/.dssi", home);
 	    } else {
-		defaultDssiPath = strdup("/usr/local/lib/dssi:/usr/lib/dssi");
+		defaultDssiPath = strdup("/usr/local/lib64/dssi:/usr/lib64/dssi:/usr/local/lib/dssi:/usr/lib/dssi");
 	    }
 	}
 	dssiPath = defaultDssiPath;