diff -rupN dssi-1.1.0.old/jack-dssi-host/Makefile.am dssi-1.1.0/jack-dssi-host/Makefile.am
--- dssi-1.1.0.old/jack-dssi-host/Makefile.am	2005-05-25 17:41:05.000000000 -0400
+++ dssi-1.1.0/jack-dssi-host/Makefile.am	2010-09-19 12:54:56.000000000 -0400
@@ -22,6 +22,6 @@ jack_dssi_host_CFLAGS = -I$(top_srcdir)/
 if DARWIN
 jack_dssi_host_LDADD = $(AM_LDFLAGS) -lmx $(ALSA_LIBS) $(LIBLO_LIBS) $(JACK_LIBS)
 else
-jack_dssi_host_LDADD = $(AM_LDFLAGS) $(ALSA_LIBS) $(LIBLO_LIBS) $(JACK_LIBS)
+jack_dssi_host_LDADD = $(AM_LDFLAGS) $(ALSA_LIBS) $(LIBLO_LIBS) $(JACK_LIBS) -lm -ldl
 endif
 
diff -rupN dssi-1.1.0.old/jack-dssi-host/Makefile.in dssi-1.1.0/jack-dssi-host/Makefile.in
--- dssi-1.1.0.old/jack-dssi-host/Makefile.in	2010-09-19 11:43:54.000000000 -0400
+++ dssi-1.1.0/jack-dssi-host/Makefile.in	2010-09-19 12:55:22.000000000 -0400
@@ -215,7 +215,7 @@ jack_dssi_host_SOURCES = \
 	../message_buffer/message_buffer.h
 
 jack_dssi_host_CFLAGS = -I$(top_srcdir)/dssi $(AM_CFLAGS) $(ALSA_CFLAGS) $(LIBLO_CFLAGS) $(JACK_CFLAGS)
-@DARWIN_FALSE@jack_dssi_host_LDADD = $(AM_LDFLAGS) $(ALSA_LIBS) $(LIBLO_LIBS) $(JACK_LIBS)
+@DARWIN_FALSE@jack_dssi_host_LDADD = $(AM_LDFLAGS) $(ALSA_LIBS) $(LIBLO_LIBS) $(JACK_LIBS) -lm -ldl
 @DARWIN_TRUE@jack_dssi_host_LDADD = $(AM_LDFLAGS) -lmx $(ALSA_LIBS) $(LIBLO_LIBS) $(JACK_LIBS)
 all: all-am
 