--- Makefile.in	2014-04-11 05:35:45.000000000 +0300
+++ Makefile.in_new	2014-07-05 02:01:43.203912026 +0300
@@ -378,7 +378,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-AM_CFLAGS = -Wall -Werror -Wno-error=deprecated-declarations -Wextra -Wwrite-strings -Wno-unused-parameter -std=gnu99 -D_GNU_SOURCE -D_REENTRANT
+AM_CFLAGS = -Wall -Wextra -Wwrite-strings -Wno-unused-parameter -std=gnu99 -D_GNU_SOURCE -D_REENTRANT
 EXTRA_DIST = README.html README-WIN32.html config.h.win32 doc json-c.vcproj
 SUBDIRS = . tests
 lib_LTLIBRARIES = libjson-c.la 
 
--- tests/Makefile.in	2014-04-11 05:35:45.000000000 +0300
+++ tests/Makefile.in_new	2014-07-05 02:09:34.093891031 +0300
@@ -544,7 +544,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-AM_CFLAGS = -Wall -Werror -Wno-error=deprecated-declarations -Wextra -Wwrite-strings -Wno-unused-parameter -std=gnu99 -D_GNU_SOURCE -D_REENTRANT
+AM_CFLAGS = -Wall -Wextra -Wwrite-strings -Wno-unused-parameter -std=gnu99 -D_GNU_SOURCE -D_REENTRANT
 LDADD = $(LIBJSON_LA)
 LIBJSON_LA = $(top_builddir)/libjson-c.la
 TESTS = test1.test test2.test test4.test testReplaceExisting.test \