Index: kde-workspace-4.7.1/plasma/generic/applets/digital-clock/clock.cpp
===================================================================
--- kde-workspace-4.7.1.orig/plasma/generic/applets/digital-clock/clock.cpp
+++ kde-workspace-4.7.1/plasma/generic/applets/digital-clock/clock.cpp
@@ -174,7 +174,7 @@ void Clock::clockConfigChanged()
     kDebug() << "showTimezone:" << m_showTimezone;
 
     if (cg.hasKey("showDate")) {    //legacy config entry as of 2011-1-4
-        m_dateStyle = cg.readEntry("showDate", false) ? 2 : 0; //short date : no date
+        m_dateStyle = cg.readEntry("showDate", true) ? 2 : 0; //short date : no date
         cg.deleteEntry("showDate");
     }
     else {
