Index: startkde.cmake
===================================================================
--- startkde.cmake.orig
+++ startkde.cmake
@@ -36,6 +36,47 @@ fi
 # we have to unset this for Darwin since it will screw up KDE's dynamic-loading
 unset DYLD_FORCE_FLAT_NAMESPACE
 
+# The user's personal KDE directory is usually ~/.kde, but this setting
+# may be overridden by setting KDEHOME.
+
+kdehome=$HOME/@KDE_DEFAULT_HOME@
+test -n "$KDEHOME" && kdehome=`echo "$KDEHOME"|sed "s,^~/,$HOME/,"`
+
+cd $HOME
+
+if test -d ".kde3.5" -a ! -L ".kde4" ; then
+    rm -rf .kde3.5/cache-*
+    rm -rf .kde3.5/socket-*
+    rm -rf .kde3.5/tmp-*
+
+    if test -d ".kde4" ; then
+
+        if [[ $LANG == tr_TR*  ]] ; then
+            prgmessage="KDE ayarlarınız aktarılıyor..."
+        else
+            prgmessage="Migrating KDE settings..."
+        fi
+
+        dbusName=`kdialog --progressbar "$prgmessage" --title "KDE" 0`
+
+        cp -a .kde4/* .kde3.5
+
+        if test $? -eq 0 ; then
+            rm -rf .kde4
+        fi
+
+        if test -n "$dbusName"; then
+            qdbus $dbusName org.kde.kdialog.ProgressDialog.close
+        fi
+    fi
+
+    ln -s .kde3.5 .kde4
+else
+    mkdir -m 700 -p $kdehome
+fi
+
+cd -
+
 # in case we have been started with full pathname spec without being in PATH
 bindir=`echo "$0" | sed -n 's,^\(/.*\)/[^/][^/]*$,\1,p'`
 if [ -n "$bindir" ]; then
@@ -66,14 +107,7 @@ fi
 #
 # * Then ksmserver is started which takes control of the rest of the startup sequence
 
-# The user's personal KDE directory is usually ~/.kde, but this setting
-# may be overridden by setting KDEHOME.
-
-kdehome=$HOME/@KDE_DEFAULT_HOME@
-test -n "$KDEHOME" && kdehome=`echo "$KDEHOME"|sed "s,^~/,$HOME/,"`
-
 # see kstartupconfig source for usage
-mkdir -m 700 -p $kdehome
 mkdir -m 700 -p $kdehome/share
 mkdir -m 700 -p $kdehome/share/config
 cat >$kdehome/share/config/startupconfigkeys <<EOF
