diff -Naur kdebase-workspace-4.2.4-orig/kcontrol/launch/kcmlaunch.cpp kdebase-workspace-4.2.4/kcontrol/launch/kcmlaunch.cpp
--- kdebase-workspace-4.2.4-orig/kcontrol/launch/kcmlaunch.cpp	2009-06-19 13:56:33.767340041 +0300
+++ kdebase-workspace-4.2.4/kcontrol/launch/kcmlaunch.cpp	2009-06-19 14:00:57.076155111 +0300
@@ -172,7 +172,7 @@
   cb_taskbarButton->setChecked(taskbarButton);
 
   c= conf.group( "BusyCursorSettings" );
-  sb_cursorTimeout->setValue( c.readEntry( "Timeout", 30 ));
+  sb_cursorTimeout->setValue( c.readEntry( "Timeout", 5));
   bool busyBlinking =c.readEntry("Blinking", false);
   bool busyBouncing =c.readEntry("Bouncing", true);
   if ( !busyCursor )
@@ -228,7 +228,7 @@
   cb_busyCursor->setCurrentIndex(2);
   cb_taskbarButton->setChecked((bool)Default & TaskbarButton);
 
-  sb_cursorTimeout->setValue( 30 );
+  sb_cursorTimeout->setValue( 5 );
   sb_taskbarTimeout->setValue( 30 );
 
   slotBusyCursor( 2 );
@@ -250,7 +250,7 @@
     c.readEntry("TaskbarButton", (bool)(Default & TaskbarButton));
 
   c = conf.group("BusyCursorSettings");
-  unsigned int savedCursorTimeout = c.readEntry( "Timeout", 30 );
+  unsigned int savedCursorTimeout = c.readEntry( "Timeout", 5);
   bool savedBusyBlinking =c.readEntry("Blinking", false);
   bool savedBusyBouncing =c.readEntry("Bouncing", true);
 
diff -Naur kdebase-workspace-4.2.4-orig/krunner/kcfg/klaunch.kcfg kdebase-workspace-4.2.4/krunner/kcfg/klaunch.kcfg
--- kdebase-workspace-4.2.4-orig/krunner/kcfg/klaunch.kcfg	2009-06-19 13:56:33.788340118 +0300
+++ kdebase-workspace-4.2.4/krunner/kcfg/klaunch.kcfg	2009-06-19 13:57:05.639086551 +0300
@@ -8,7 +8,7 @@
   
   <group name="BusyCursorSettings">
     <entry key="Timeout" type="Int">
-      <default>30</default>
+      <default>5</default>
       <label></label>
       <whatsthis></whatsthis>
       <!-- startupid.cpp:58 -->
