Index: kdebase-workspace-4.6.1/kwin/scene_opengl.cpp
===================================================================
--- kdebase-workspace-4.6.1.orig/kwin/scene_opengl.cpp
+++ kdebase-workspace-4.6.1/kwin/scene_opengl.cpp
@@ -144,7 +144,7 @@ SceneOpenGL::SceneOpenGL( Workspace* ws
         return; // error
     // Initialize OpenGL
     initGL();
-    if( QString((const char*)glGetString( GL_RENDERER )) == "Software Rasterizer" )
+    if( (QString((const char*)glGetString( GL_RENDERER )) == "Software Rasterizer") || (QString((const char*)glGetString( GL_RENDERER )).endsWith("on llvmpipe")))
         {
         kError( 1212 ) << "OpenGL Software Rasterizer detected. Falling back to XRender.";
         QTimer::singleShot( 0, Workspace::self(), SLOT( fallbackToXRenderCompositing()));
Index: kdebase-workspace-4.6.1/kwin/compositingprefs.cpp
===================================================================
--- kdebase-workspace-4.6.1.orig/kwin/compositingprefs.cpp
+++ kdebase-workspace-4.6.1/kwin/compositingprefs.cpp
@@ -248,9 +248,13 @@ void CompositingPrefs::applyDriverSpecif
     mStrictBinding = !gl->supports( LooseBinding );
     if ( gl->driver() == Driver_Intel )
         mEnableVSync = false;
+
+    if (gl->driver() == Driver_Llvmpipe)
+        mRecommendCompositing = false;
     }
 
 
+
 bool CompositingPrefs::detectXgl()
     { // Xgl apparently uses only this specific X version
     return VendorRelease(display()) == 70000001;
