diff -ur kdebase-workspace-4.2.1.orig/kdm/kfrontend/genkdmconf.c kdebase-workspace-4.2.1/kdm/kfrontend/genkdmconf.c
--- kdebase-workspace-4.2.1.orig/kdm/kfrontend/genkdmconf.c	2009-04-01 14:02:37.000000000 +0200
+++ kdebase-workspace-4.2.1/kdm/kfrontend/genkdmconf.c	2009-04-01 14:08:40.000000000 +0200
@@ -687,6 +687,9 @@
 "[ -f /etc/xprofile ] && . /etc/xprofile\n"
 "[ -f $HOME/.xprofile ] && . $HOME/.xprofile\n"
 "\n"
+"[ -f /etc/X11/Xmodmap ] && xmodmap /etc/X11/Xmodmap\n"
+"[ -f $HOME/.Xmodmap ] && xmodmap $HOME/.Xmodmap\n"
+"\n"
 "case $session in\n"
 "  \"\")\n"
 "    exec xmessage -center -buttons OK:0 -default OK \"Sorry, $DESKTOP_SESSION is no valid session.\"\n"
