--- startkde.cmake.orig	2009-10-26 13:56:45.254640676 +0200
+++ startkde.cmake	2009-10-26 14:06:24.419527045 +0200
@@ -396,6 +396,13 @@
 KDE_SESSION_UID=`id -ru`
 export KDE_SESSION_UID
 
+if [ -e $kdehome/share/config/plasma-appletsrc ] && [ -e `kde4-config --path lib --locate kconf_update_bin/plasma-to-plasma-desktop.py` ]; then
+    python `kde4-config --path lib --locate kconf_update_bin/plasma-to-plasma-desktop.py`
+
+    # Be sure there is no old configuration any more
+    rm -f $kdehome/share/config/plasma-appletsrc
+fi
+
 # We set LD_BIND_NOW to increase the efficiency of kdeinit.
 # kdeinit unsets this variable before loading applications.
 LD_BIND_NOW=true kdeinit4 +kcminit_startup
