--- kdm/backend/client.c.old	2010-08-17 11:43:02.775903757 +0300
+++ kdm/backend/client.c	2010-08-17 11:43:19.548533659 +0300
@@ -474,14 +474,13 @@
 	 * the module needs an own conversation plugin which does not cause
 	 * curuser being set.
 	 */
-	if (!curuser) {
-		debug( " asking PAM for user ...\n" );
-		pam_get_item( pamh, PAM_USER, &pitem );
-		reInitErrorLog();
-		strDup( &curuser, (const char *)pitem );
-		gSendInt( V_PUT_USER );
-		gSendStr( curuser );
-	}
+    /* Always check username from PAM */
+	debug( " asking PAM for user ...\n" );
+	pam_get_item( pamh, PAM_USER, &pitem );
+	reInitErrorLog();
+	strDup( &curuser, (const char *)pitem );
+	gSendInt( V_PUT_USER );
+	gSendStr( curuser );
 	if (pretc != PAM_SUCCESS) {
 	        /* Log the failed login attempt */
 	        log_to_audit_system (curuser, td->remoteHost, td->name, AU_FAILED);
