Index: kdebase-workspace-4.6.1/plasma/generic/applets/battery/battery.cpp
===================================================================
--- kdebase-workspace-4.6.1.orig/plasma/generic/applets/battery/battery.cpp
+++ kdebase-workspace-4.6.1/plasma/generic/applets/battery/battery.cpp
@@ -379,6 +379,19 @@ void Battery::readColors()
     m_boxColor = Plasma::Theme::defaultTheme()->color(Plasma::Theme::BackgroundColor);
 }
 
+void Battery::wheelEvent(QGraphicsSceneWheelEvent *event)
+{
+    int current = m_brightnessSlider->value();
+    int newVal = current + (5 * (event->delta() / 120));
+
+    if (newVal < 0)
+        newVal = 0;
+
+    m_ignoreBrightnessChange = true;
+    m_brightnessSlider->setValue(newVal);
+    m_ignoreBrightnessChange = false;
+}
+
 void Battery::hoverEnterEvent(QGraphicsSceneHoverEvent *event)
 {
     if (!m_showBatteryLabel && !m_isEmbedded && !isConstrained()) {
Index: kdebase-workspace-4.6.1/plasma/generic/applets/battery/battery.h
===================================================================
--- kdebase-workspace-4.6.1.orig/plasma/generic/applets/battery/battery.h
+++ kdebase-workspace-4.6.1/plasma/generic/applets/battery/battery.h
@@ -75,6 +75,7 @@ class Battery : public Plasma::PopupAppl
         void toolTipAboutToShow();
 
     protected Q_SLOTS:
+        virtual void wheelEvent (QGraphicsSceneWheelEvent *event);
         virtual void hoverEnterEvent(QGraphicsSceneHoverEvent *event);
         virtual void hoverLeaveEvent(QGraphicsSceneHoverEvent *event);
         void configAccepted();
