Index: kdebase-workspace-4.5.3/kcontrol/randr/randroutput.cpp
===================================================================
--- kdebase-workspace-4.5.3.orig/kcontrol/randr/randroutput.cpp
+++ kdebase-workspace-4.5.3/kcontrol/randr/randroutput.cpp
@@ -441,13 +441,13 @@ QStringList RandROutput::startupCommands
 		    .arg( m_crtc->rect().y()).arg( m_crtc->rect().width()).arg( m_crtc->rect().height());
 		switch( m_crtc->rotation()) {
 			case RandR::Rotate90:
-				command += " --rotate right";
+				command += " --rotate left";
 				break;
 			case RandR::Rotate180:
 				command += " --rotate inverted";
 				break;
 			case RandR::Rotate270:
-				command += " --rotate left";
+				command += " --rotate right";
 				break;
 		}
 	}
