Index: kde-workspace-4.7.2/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
===================================================================
--- kde-workspace-4.7.2.orig/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
+++ kde-workspace-4.7.2/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
@@ -8,7 +8,6 @@ panel.height = screenGeometry(panel.scre
 var launcher = panel.addWidget("launcher")
 launcher.globalShortcut = "Alt+F1";
 panel.addWidget("showdesktop")
-panel.addWidget("org.kde.showActivityManager")
 panel.addWidget("pager")
 tasks = panel.addWidget("tasks")
 panel.addWidget("systemtray")
