diff -up dhcp-4.2.0/server/dhcpd.c.ifnamsiz dhcp-4.2.0/server/dhcpd.c
--- dhcp-4.2.0/server/dhcpd.c.ifnamsiz	2010-07-21 14:31:39.000000000 +0200
+++ dhcp-4.2.0/server/dhcpd.c	2010-07-21 16:05:49.000000000 +0200
@@ -409,7 +409,7 @@ main(int argc, char **argv) {
 				log_fatal ("Insufficient memory to %s %s: %s",
 					   "record interface", argv [i],
 					   isc_result_totext (result));
-			strcpy (tmp -> name, argv [i]);
+			strncpy (tmp -> name, argv [i], sizeof (tmp -> name));
 			if (interfaces) {
 				interface_reference (&tmp -> next,
 						     interfaces, MDL);
