diff -up dhcp-4.2.0/configure.ac.ldap dhcp-4.2.0/configure.ac
--- dhcp-4.2.0/configure.ac.ldap	2010-07-10 05:25:51.000000000 +0200
+++ dhcp-4.2.0/configure.ac	2010-07-21 13:06:20.000000000 +0200
@@ -521,6 +521,8 @@ AC_ARG_WITH(ldapcrypto,
 if test x$ldap = xyes || test x$ldapcrypto = xyes ; then
     AC_SEARCH_LIBS(ldap_initialize, [ldap], ,
 		   AC_MSG_FAILURE([*** Cannot find ldap_initialize with -lldap - do you need to install an OpenLDAP2 Devel package?]))
+    AC_SEARCH_LIBS(ber_pvt_opt_on, [lber], ,
+		   AC_MSG_FAILURE([*** Cannot find ber_pvt_opt_on with -llber - do you need to install an OpenLDAP2 Devel package?]))
 
     if test x$ldapcrypto = xyes ; then
 	AC_SUBST(LDAP_CFLAGS, ["-DLDAP_CONFIGURATION -DLDAP_USE_SSL"])
