diff -up dhcp-4.2.0-P2/common/Makefile.am.ldap-configuration dhcp-4.2.0-P2/common/Makefile.am
--- dhcp-4.2.0-P2/common/Makefile.am.ldap-configuration	2011-01-13 17:54:14.000000000 +0100
+++ dhcp-4.2.0-P2/common/Makefile.am	2011-01-13 17:54:14.000000000 +0100
@@ -1,5 +1,5 @@
 AM_CPPFLAGS = -I.. -DLOCALSTATEDIR='"@localstatedir@"'
-AM_CFLAGS = -fpic
+AM_CFLAGS = $(LDAP_CFLAGS) -fpic
 
 noinst_LIBRARIES = libdhcp.a
 libdhcp_a_SOURCES = alloc.c bpf.c comapi.c conflex.c ctrace.c discover.c \
