--- tools/GUIEditor/Makefile.old	2012-03-04 20:03:37.069715461 +0100
+++ tools/GUIEditor/Makefile	2012-03-04 20:05:43.271717591 +0100
@@ -2,8 +2,8 @@
 Target = GUIEditor
 Sources = CGUIAttributeEditor.cpp CGUIEditFactory.cpp CGUIEditWindow.cpp CGUIEditWorkspace.cpp CGUIPanel.cpp CGUITextureCacheBrowser.cpp CMemoryReadWriteFile.cpp main.cpp
 
-CPPFLAGS = -I../../include -I/usr/X11R6/include
-CXXFLAGS = -Wall -O3 -ffast-math
+CPPFLAGS += -I../../include -I$(INCLUDEDIR)
+CXXFLAGS += -Wall -O3 -ffast-math
 
 ifeq ($(HOSTTYPE), x86_64)
 LIBSELECT=64
@@ -13,7 +13,7 @@ all: all_linux
 
 # target specific settings
 all_linux: SYSTEM=Linux
-all_linux: LDFLAGS = -L/usr/X11R6/lib$(LIBSELECT) -L../../lib/$(SYSTEM) -lIrrlicht -lGL -lXxf86vm -lXext -lX11
+all_linux: LDFLAGS = -L$(LIBDIR) -L../../lib/$(SYSTEM) -lIrrlicht -lGL -lXxf86vm -lXext -lX11 -ljpeg -lpng -lz -lbz2
 
 all_win32 clean_win32: SYSTEM=Win32-gcc
 all_win32: LDFLAGS = -L../../lib/$(SYSTEM) -lIrrlicht -lopengl32 -lm
@@ -31,7 +31,7 @@ OBJ = $(Sources:.cpp=.o)
 
 all_linux all_win32: $(OBJ)
 	$(warning Building...)
-	$(CXX) $(CPPFLAGS) $(CXXFLAGS) $^ -o $(DESTPATH) $(LDFLAGS)
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) $^ -L../../lib/$(SYSTEM) -lIrrlicht -lGL -lXxf86vm -lXext -lX11 -ljpeg -lpng -lz -lbz2 -o $(DESTPATH) $(LDFLAGS)
 
 clean: clean_linux clean_win32
 	$(warning Cleaning...)
