diff -Nuar cakephp-cakephp-60df687.orig/cake/console/cake cakephp-cakephp-60df687/cake/console/cake
--- cakephp-cakephp-60df687.orig/cake/console/cake	2011-10-31 09:12:34.520000134 +0200
+++ cakephp-cakephp-60df687/cake/console/cake	2011-10-31 09:13:17.151000140 +0200
@@ -18,7 +18,7 @@
 # @license              MIT License (http://www.opensource.org/licenses/mit-license.php)
 #
 ################################################################################
-LIB=$(cd -P -- "$(dirname -- "$0")" && pwd -P) && LIB=$LIB/$(basename -- "$0")
+LIB=/usr/share/php5/cakephp/cake/console/
 
 while [ -h $LIB ]; do
     DIR=$(dirname -- "$LIB")
@@ -31,4 +31,4 @@
 
 exec php -q ${LIB}cake.php -working "${APP}" "$@"
 
-exit;
\ Dosya sonunda yenisatır yok.
+exit;
