diff -Nur fontforge-20100501.orig/fontforge/splineutil.c fontforge-20100501/fontforge/splineutil.c
--- fontforge-20100501.orig/fontforge/splineutil.c	2010-04-21 18:53:25.000000000 -0600
+++ fontforge-20100501/fontforge/splineutil.c	2010-07-16 16:26:34.000000000 -0600
@@ -4629,14 +4629,14 @@
 	if ( r==NULL )
 return( NULL );
 	spl = r->layers[0].splines;
-	r = r->next;
+	do { r = r->next; } while ( r!=NULL && r->layers[0].splines==NULL );
     }
     while ( r!=NULL ) {
 	while ( spl!=NULL && spl->next!=r->layers[0].splines )
 	    spl = spl->next;
 	spl->next = NULL;
 	spl = r->layers[0].splines;
-	r = r->next;
+	do { r = r->next; } while ( r!=NULL && r->layers[0].splines==NULL );
     }
 return( layer->splines );
 }
