Index: fontforge-20090408/pyhook/loadfontforge.h
===================================================================
--- fontforge-20090408.orig/pyhook/loadfontforge.h
+++ fontforge-20090408/pyhook/loadfontforge.h
@@ -11,7 +11,7 @@ PyMODINIT_FUNC ENTRY_POINT(void) {
 
     if ( (lib = dlopen("libgunicode" SO_EXT,RTLD_LAZY))==NULL ) {
 #ifdef PREFIX
-	lib = dlopen( PREFIX "/lib/" "libgunicode" SO_EXT,RTLD_LAZY);
+	lib = dlopen( SOLIBDIR "libgunicode.so.3" ,RTLD_LAZY);
 #endif
     }
     if ( lib==NULL ) {
@@ -21,7 +21,7 @@ return;
 
     if ( (lib = dlopen("libgutils" SO_EXT,RTLD_LAZY))==NULL ) {
 #ifdef PREFIX
-	lib = dlopen( PREFIX "/lib/" "libgutils" SO_EXT,RTLD_LAZY);
+	lib = dlopen( SOLIBDIR "libgutils.so.1" ,RTLD_LAZY);
 #endif
     }
     if ( lib==NULL ) {
@@ -31,7 +31,7 @@ return;
 
     if ( (lib = dlopen("libfontforge" SO_EXT,RTLD_LAZY))==NULL ) {
 #ifdef PREFIX
-	lib = dlopen( PREFIX "/lib/" "libfontforge" SO_EXT,RTLD_LAZY);
+	lib = dlopen( SOLIBDIR "libfontforge.so.1" SO_EXT,RTLD_LAZY);
 #endif
     }
     if ( lib==NULL ) {
