diff -Nur fontforge-20100501.orig/fontforge/ffpython.h fontforge-20100501/fontforge/ffpython.h
--- fontforge-20100501.orig/fontforge/ffpython.h	2010-04-05 14:10:26.000000000 -0600
+++ fontforge-20100501/fontforge/ffpython.h	2010-07-28 12:07:25.000000000 -0600
@@ -25,7 +25,7 @@
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#if PY_MAJOR_VERSION >= 3
+#if PY_MAJOR_VERSION >= 3 || (PY_MAJOR_VERSION == 2 && PY_MINOR_VERSION > 6)
 
 #define PyInt_Check    PyLong_Check
 #define PyInt_AsLong   PyLong_AsLong
