Index: fontforge-20090408/pyhook/setup.py.in
===================================================================
--- fontforge-20090408.orig/pyhook/setup.py.in
+++ fontforge-20090408/pyhook/setup.py.in
@@ -9,7 +9,7 @@ setup(name='fontforge',
     ext_modules=[Extension('fontforge',['fontforgepyhook.c'],
 	    include_dirs=[@PYINCLUDE@],
 	    define_macros=[@PYDEFINES@],
-	    libraries=[@PYLIBS@]),
+	    libraries=[@PYLIBS@, '-ldl']),
 	Extension('psMat',['psMatpyhook.c'],
 	    include_dirs=[@PYINCLUDE@],
 	    define_macros=[@PYDEFINES@],
