diff -burN libdlna-0.2.4.orig/src/profiles.c libdlna-0.2.4/src/profiles.c
--- libdlna-0.2.4.orig/src/profiles.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4/src/profiles.c	2013-04-21 15:11:17.158148687 +0200
@@ -287,7 +287,7 @@
     return NULL;
   }
 
-  if (av_find_stream_info (ctx) < 0)
+  if (avformat_find_stream_info (ctx, NULL) < 0)
   {
     if (dlna->verbosity)
       fprintf (stderr, "can't find stream info\n");
