diff -uNr scidavis-0.2.4.orig/scidavis/src/ApplicationWindow.cpp scidavis-0.2.4/scidavis/src/ApplicationWindow.cpp
--- scidavis-0.2.4.orig/scidavis/src/ApplicationWindow.cpp	2010-10-06 20:00:36.000000000 +0200
+++ scidavis-0.2.4/scidavis/src/ApplicationWindow.cpp	2010-10-06 20:01:28.000000000 +0200
@@ -135,7 +135,7 @@
 #include <QVarLengthArray>
 #include <QList>
 #include <QUrl>
-#include <QAssistantClient>
+#include <QtAssistant/qassistantclient.h>
 #include <QStatusBar>
 #include <QToolButton>
 #include <QSignalMapper>
 