--- scidavis-0.2.4/scidavis/scidavis.pro.orig	2009-10-06 17:24:55.000000000 +0200
+++ scidavis-0.2.4/scidavis/scidavis.pro	2012-02-18 16:58:34.661711076 +0100
@@ -37,7 +37,7 @@ unix: INSTALLBASE = /usr           # thi
 win32: INSTALLBASE = c:/scidavis
 unix: target.path = "$$INSTALLBASE/bin"               # where to install the binary on Linux/MacOS X
 win32: target.path = "$$INSTALLBASE"                  # where to install the exe on Windows
-unix: documentation.path = "$$INSTALLBASE/share/doc/scidavis"      # where to install the documentation files on Linux/MacOS X
+unix: documentation.path = "$$INSTALLBASE/share/doc/packages/scidavis"      # where to install the documentation files on Linux/MacOS X
 win32: documentation.path = "$$INSTALLBASE"                        # ... on Winodws
 ### Usually, the manual will be expected in the "manual" subfolder of "documentation.path" (see above).
 ### You can override this, uncomment and adjust the path behind the '=' in the next line.
@@ -65,7 +65,7 @@ win32: pythonutils.path = $$INSTALLBASE
 ### building without muParser does not work yet (but will in a future version)
 include( muparser.pri )
 ### comment out the following line to deactivate Python scripting support
-include( python.pri )  
+#include( python.pri )  
 
 ################### start of liborigin block 
 ############ liborigin support has been discontinued due to the lack 
@@ -87,11 +87,11 @@ include( python.pri )
 ### are compiled against Qt4), dynamically against everything else.
 #############################################################################
 
-unix:INCLUDEPATH  += ../3rdparty/qwtplot3d/include
-unix:LIBS         += ../3rdparty/qwtplot3d/lib/libqwtplot3d.a
+#unix:INCLUDEPATH  += ../3rdparty/qwtplot3d/include
+#unix:LIBS         += ../3rdparty/qwtplot3d/lib/libqwtplot3d.a
 
-unix:INCLUDEPATH  += ../3rdparty/qwt/src
-unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
+#unix:INCLUDEPATH  += ../3rdparty/qwt/src
+#unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
 
 unix:LIBS         += -L/usr/lib$${libsuff}
 unix:LIBS         += -lgsl -lgslcblas -lz
@@ -102,17 +102,17 @@ unix:INCLUDEPATH  += /usr/include/muPars
 ### Link everything dynamically
 #############################################################################
 
-#unix:INCLUDEPATH  += /usr/include/qwt5
-#unix:LIBS         += -L/usr/lib$${libsuff}
+unix:INCLUDEPATH  += /usr/include/qwt5
+unix:LIBS         += -L/usr/lib$${libsuff}
 ## dynamically link against Qwt(3D) installed system-wide
 ## WARNING: make sure they are compiled against >= Qt4.2
 ## Mixing Qt 4.2 and Qt >= 4.3 compiled stuff may also 
 ## cause problems.
-#unix:INCLUDEPATH  += /usr/include/qwtplot3d
-#unix:LIBS         += -lqwtplot3d
-#unix:LIBS         += -lqwt
+unix:INCLUDEPATH  += /usr/include/qwtplot3d
+unix:LIBS         += -lqwtplot3d
+unix:LIBS         += -lqwt
 ##dynamically link against GSL and zlib installed system-wide
-#unix:LIBS         += -lgsl -lgslcblas -lz -lmuparser
+unix:LIBS         += -lgsl -lgslcblas -lz -lmuparser
 
 #############################################################################
 ### Default settings for Windows
