diff -ur scidavis-0.2.4-old//scidavis/scidavis.pro scidavis-0.2.4//scidavis/scidavis.pro
--- scidavis-0.2.4-old//scidavis/scidavis.pro	2009-10-06 17:24:55.000000000 +0200
+++ scidavis-0.2.4//scidavis/scidavis.pro	2011-04-02 14:00:28.338837482 +0200
@@ -134,6 +134,10 @@
 win32:LIBS        += c:/zlib/lib/libz.a
 win32:LIBS        += c:/muparser/lib/libmuparser.a
 
+# Add include path for the obsolete Qt Assistant Client
+unix:INCLUDEPATH += /usr/include/QtAssistant
+unix:LIBS += -lQtAssistantClient
+
 #############################################################################
 ###                    END OF USER-SERVICEABLE PART                       ###
 #############################################################################

