diff -Naur PDF-API2-2.016-pristine/lib/PDF/API2.pm PDF-API2-2.016/lib/PDF/API2.pm
--- PDF-API2-2.016-pristine/lib/PDF/API2.pm	2011-01-24 08:41:31.000000000 -0700
+++ PDF-API2-2.016/lib/PDF/API2.pm	2011-01-30 15:45:59.300804559 -0700
@@ -53,8 +53,7 @@
 
 no warnings qw[ deprecated recursion uninitialized ];
 
-our @FontDirs = ( (map { "$_/PDF/API2/fonts" } @INC), 
-                  qw[ /usr/share/fonts /usr/local/share/fonts c:/windows/fonts c:/winnt/fonts ] );
+our @FontDirs = ( qw[ /usr/share/fonts/dejavu /usr/share/fonts ] );
 
 =head1 NAME
 
