--- gliv.desktop.tmpl.OSmanOS	2010-11-07 21:22:01.000000000 +0200
+++ gliv.desktop.tmpl	2013-02-09 20:21:55.776261717 +0200
@@ -4,8 +4,10 @@
 Encoding=UTF-8
 Name=GLiv Image Viewer
 Name[fr]=Visualisateur d'images GLiv
+Name[tr]=GLiv Resim Görüntüleyicisi
 GenericName=Image Viewer
 GenericName[fr]=Visualisateur d'images
+GenericName[tr]=Resim Görüntüleyicisi
 Icon=@SHAREDIR@/pixmaps/gliv.png
 TryExec=@BINDIR@/gliv
 Exec=@BINDIR@/gliv %F
