Index: miro-4.0.3/linux/miro.desktop
===================================================================
--- miro-4.0.3.orig/linux/miro.desktop
+++ miro-4.0.3/linux/miro.desktop
@@ -1,12 +1,15 @@
 [Desktop Entry]
 Type=Application
 Name=Miro music and video player
+Name[tr]=Miro müzik ve video oynatıcısı
 GenericName=Media player
+GenericName[tr]=Çokluortam oynatıcısı
 Comment=Fully featured video and music player, podcast client, and bittorrent client
 Comment[de]=Multimedia Player
 Comment[es]=Reproductor multimedia
 Comment[fr]=Lecteur multimedia
 Comment[it]=Lettore multimediale
+Comment[tr]=Tam teşekküllü video ve müzik oynatıcısı, podcast ve bittorent istemcisi
 Icon=miro
 TryExec=miro
 Exec=miro %F
