Index: NetworkManager-pptp-0.8.4/po/POTFILES.in
===================================================================
--- NetworkManager-pptp-0.8.4.orig/po/POTFILES.in
+++ NetworkManager-pptp-0.8.4/po/POTFILES.in
@@ -4,6 +4,6 @@ auth-dialog/gnome-two-password-dialog.c
 auth-dialog/main.c
 nm-pptp.desktop.in
 properties/advanced-dialog.c
-properties/nm-pptp-dialog.ui
+[type: gettext/glade]properties/nm-pptp-dialog.ui
 properties/nm-pptp.c
 src/nm-pptp-service.c
