Index: rootactions_servicemenu_2.5.92/Root_Actions_2.5.92/dolphin-KDE4/10-rootactionsfolders.desktop
===================================================================
--- rootactions_servicemenu_2.5.92.orig/Root_Actions_2.5.92/dolphin-KDE4/10-rootactionsfolders.desktop
+++ rootactions_servicemenu_2.5.92/Root_Actions_2.5.92/dolphin-KDE4/10-rootactionsfolders.desktop
@@ -26,7 +26,7 @@ X-KDE-Submenu[pt_PT]=Ac&ções Administr
 X-KDE-Submenu[ru]=Действия root
 X-KDE-Submenu[sr]=Корене &акције
 X-KDE-Submenu[sr@latin]=Korene &akcije
-X-KDE-Submenu[tr]=&Yetkili Eylemleri
+X-KDE-Submenu[tr]=&Yetkili Kullanıcı Eylemleri
 #X-KDE-Submenu[xx]=Your string 'xx' is the country abbreviation
 
 [Desktop Action OpenInKonsole]
@@ -153,7 +153,7 @@ Name[pt_PT]=Mudar o nome
 Name[ru]=Переместить/переименовать
 Name[sr]=Премести/преименуј
 Name[sr@latin]=Premesti/preimenuj
-Name[tr]=Taşı/Yeniden İsimlendir
+Name[tr]=Taşı/Yeniden adlandır
 #Name[xx]=Your string 'xx' is the country abbreviation
 
 [Desktop Action Delete]
@@ -205,7 +205,7 @@ Name[pt_PT]=Alterar dono para 'Root'
 Name[ru]=Установить владельцем root
 Name[sr]=Постави власништво на кореног корисника
 Name[sr@latin]=Postavi vlasništvo na korenog korisnika
-Name[tr]=Dosyayı yetkili kullanıcının yap
+Name[tr]=Yetkili kullanıcıyı dosya sahibi yap
 #Name[xx]=Your string 'xx' is the country abbreviation
 
 [Desktop Action ChangeUser]
@@ -231,7 +231,7 @@ Name[pt_PT]=Alterar dono para o utilizad
 Name[ru]=Установить владельцем активного пользователя
 Name[sr]=Постави власништво на тренутног корисника
 Name[sr@latin]=Postavi vlasništvo na trenutnog korisnika
-Name[tr]=Dosyayı şu anda oturumu açık kullanıcının yap
+Name[tr]=Oturumu açan kullanıcıyı dosya sahibi yap
 #Name[xx]=Your string 'xx' is the country abbreviation
 
 [Desktop Action ChangeCustom]
Index: rootactions_servicemenu_2.5.92/Root_Actions_2.5.92/dolphin-KDE4/11-rootactionsfiles.desktop
===================================================================
--- rootactions_servicemenu_2.5.92.orig/Root_Actions_2.5.92/dolphin-KDE4/11-rootactionsfiles.desktop
+++ rootactions_servicemenu_2.5.92/Root_Actions_2.5.92/dolphin-KDE4/11-rootactionsfiles.desktop
@@ -26,7 +26,7 @@ X-KDE-Submenu[pt_PT]=Ac&ções Administr
 X-KDE-Submenu[ru]=Действия root
 X-KDE-Submenu[sr]=Корене &акције
 X-KDE-Submenu[sr@latin]=Korene &akcije
-X-KDE-Submenu[tr]=&Yetkili Eylemleri
+X-KDE-Submenu[tr]=&Yetkili Kullanıcı Eylemleri
 #X-KDE-Submenu[xx]=Your string 'xx' is the country abbreviation
 
 [Desktop Action EditAsText]
@@ -231,7 +231,7 @@ Name[pt_PT]=Alterar dono para 'Root'
 Name[ru]=Установить владельцем root
 Name[sr]=Постави власништво на кореног корисника
 Name[sr@latin]=Postavi vlasništvo na korenog korisnika
-Name[tr]=Dosyayı yetkili kullanıcının yap
+Name[tr]=Yetkili kullanıcıyı dosya sahibi yap
 #Name[xx]=Your string 'xx' is the country abbreviation
 
 [Desktop Action ChangeUser]
@@ -257,7 +257,7 @@ Name[pt_PT]=Alterar dono para o utilizad
 Name[ru]=Установить владельцем активного пользователя
 Name[sr]=Постави власништво на тренутног корисника
 Name[sr@latin]=Postavi vlasništvo na trenutnog korisnika
-Name[tr]=Dosyayı şu anda oturumu açık kullanıcının yap
+Name[tr]=Oturumu açan kullanıcıyı dosya sahibi yap
 #Name[xx]=Your string 'xx' is the country abbreviation
 
 [Desktop Action ChangeCustom]
