--- src/audacity.desktop.in.OSmanOS	2013-02-09 15:16:36.000000000 +0200
+++ src/audacity.desktop.in		2013-02-09 15:21:30.793367124 +0200
@@ -5,9 +5,11 @@
 GenericName=Sound Editor
 GenericName[de]=Audio-Editor
 GenericName[ru]=Редактор звуковых файлов
+GenericName[tr]=Ses Düzenleyici
 Comment=Record and edit audio files
 Comment[de]=Audio-Dateien aufnehmen und bearbeiten
 Comment[ru]=Запись и редактирование звуковых файлов
+Comment[tr]=Ses dosyalarını kaydet ve düzenle
 
 Icon=audacity
 
