--- audacity-src-2.0.4/lib-src/portaudio-v19/src/os/unix/pa_unix_hostapis.c.alsa-by-default	2013-09-01 06:03:30.000000000 +0800
+++ audacity-src-2.0.4/lib-src/portaudio-v19/src/os/unix/pa_unix_hostapis.c	2013-09-09 11:29:58.384670574 +0800
@@ -57,17 +57,6 @@
 
 PaUtilHostApiInitializer *paHostApiInitializers[] =
     {
-#ifdef __linux__
-
-#if PA_USE_ALSA
-        PaAlsa_Initialize,
-#endif
-
-#if PA_USE_OSS
-        PaOSS_Initialize,
-#endif
-
-#else   /* __linux__ */
 
 #if PA_USE_OSS
         PaOSS_Initialize,
@@ -77,8 +66,6 @@
         PaAlsa_Initialize,
 #endif
 
-#endif  /* __linux__ */
-
 #if PA_USE_JACK
         PaJack_Initialize,
 #endif
