--- audacity-src-2.0.0/src/Makefile.in.linking	2012-03-16 10:15:30.542427254 +0100
+++ audacity-src-2.0.0/src/Makefile.in	2012-03-16 10:16:20.230359492 +0100
@@ -20,7 +20,7 @@ PRECOMP_CFLAGS = @PRECOMP_CFLAGS@
 LOCAL_LIBS = @LOCAL_LIBS@
 EXTRAOBJS = @EXTRAOBJS@
 EXTRATARGETS = @EXTRATARGETS@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lpthread
 DIRS=blockfile commands effects effects/ladspa effects/nyquist effects/vamp export import prefs toolbars widgets xml
 
 srcdir=@srcdir@
