Index: audacity-src-1.3.8/src/AudacityApp.cpp
===================================================================
--- audacity-src-1.3.8.orig/src/AudacityApp.cpp
+++ audacity-src-1.3.8/src/AudacityApp.cpp
@@ -957,6 +957,11 @@ bool AudacityApp::OnInit()
       return false;
    }
 
+   gPrefs->Write(wxT("/MP3/Enabled"), wxT("1"));
+   gPrefs->Write(wxT("/MP3/MP3LibPath"), wxT("/usr/lib/libmp3lame.so.0"));
+   gPrefs->Write(wxT("/FFmpeg/Enabled"), wxT("1"));
+   gPrefs->Write(wxT("/FFmpeg/FFmpegLibPath"), wxT("/usr/lib/libavformat.so.52"));
+
    // More initialization
    InitCleanSpeech();
 
