--- openssl-1.0.2/crypto/perlasm/x86_64-xlate.pl~	2015-02-27 20:05:00.732776724 +0000
+++ openssl-1.0.2/crypto/perlasm/x86_64-xlate.pl	2015-02-27 20:05:23.327082845 +0000
@@ -199,8 +199,12 @@
 	if ($gas) {
 	    # Solaris /usr/ccs/bin/as can't handle multiplications
 	    # in $self->{value}
+	    if ($self->{value} =~ /0xA040608020C0E000/) {
+		    $self->{value} = "11547335547999543296";
+	    } else {
	    $self->{value} =~ s/(?<![\w\$\.])(0x?[0-9a-f]+)/oct($1)/egi;
	    $self->{value} =~ s/([0-9]+\s*[\*\/\%]\s*[0-9]+)/eval($1)/eg;
+           }
 	    sprintf "\$%s",$self->{value};
 	} else {
 	    $self->{value} =~ s/(0b[0-1]+)/oct($1)/eig;