diff -ruN openssl-1.0.2.orig/crypto/cryptlib.h openssl-1.0.2/crypto/cryptlib.h
--- openssl-1.0.2.orig/crypto/cryptlib.h	2015-01-22 15:58:32.000000000 +0100
+++ openssl-1.0.2/crypto/cryptlib.h	2015-01-27 15:48:31.239123780 +0100
@@ -82,7 +82,7 @@
 # ifndef OPENSSL_SYS_VMS
 #  define X509_CERT_AREA          OPENSSLDIR
 #  define X509_CERT_DIR           OPENSSLDIR "/certs"
-#  define X509_CERT_FILE          OPENSSLDIR "/cert.pem"
+#  define X509_CERT_FILE          "/etc/ssl/certs/ca-certificates.crt"
 #  define X509_PRIVATE_DIR        OPENSSLDIR "/private"
 # else
 #  define X509_CERT_AREA          "SSLROOT:[000000]"