diff -ruNp openssl-1.0.2.orig/crypto/bn/asm/x86_64-gcc.c openssl-1.0.2/crypto/bn/asm/x86_64-gcc.c
--- openssl-1.0.2.orig/crypto/bn/asm/x86_64-gcc.c	2015-01-22 15:58:32.000000000 +0100
+++ openssl-1.0.2/crypto/bn/asm/x86_64-gcc.c	2015-01-27 16:19:21.432252035 +0100
@@ -212,9 +212,9 @@ BN_ULONG bn_add_words(BN_ULONG *rp, cons
     asm volatile ("       subq    %0,%0           \n" /* clear carry */
                   "       jmp     1f              \n"
                   ".p2align 4                     \n"
-                  "1:     movq    (%4,%2,8),%0    \n"
-                  "       adcq    (%5,%2,8),%0    \n"
-                  "       movq    %0,(%3,%2,8)    \n"
+                  "1:     movq    (%q4,%2,8),%0    \n"
+                  "       adcq    (%q5,%2,8),%0    \n"
+                  "       movq    %0,(%q3,%2,8)    \n"
                   "       lea     1(%2),%2        \n"
                   "       loop    1b              \n"
                   "       sbbq    %0,%0           \n":"=&r" (ret), "+c"(n),
@@ -238,9 +238,9 @@ BN_ULONG bn_sub_words(BN_ULONG *rp, cons
     asm volatile ("       subq    %0,%0           \n" /* clear borrow */
                   "       jmp     1f              \n"
                   ".p2align 4                     \n"
-                  "1:     movq    (%4,%2,8),%0    \n"
-                  "       sbbq    (%5,%2,8),%0    \n"
-                  "       movq    %0,(%3,%2,8)    \n"
+                  "1:     movq    (%q4,%2,8),%0    \n"
+                  "       sbbq    (%q5,%2,8),%0    \n"
+                  "       movq    %0,(%q3,%2,8)    \n"
                   "       lea     1(%2),%2        \n"
                   "       loop    1b              \n"
                   "       sbbq    %0,%0           \n":"=&r" (ret), "+c"(n), 
