diff -ruN openssl-1.0.2.orig/Makefile.org openssl-1.0.2/Makefile.org
--- openssl-1.0.2.orig/Makefile.org	2015-01-22 15:58:06.000000000 +0100
+++ openssl-1.0.2/Makefile.org	2015-01-27 16:04:30.564519915 +0100
@@ -360,7 +360,7 @@
 	@ ( echo 'prefix=$(INSTALLTOP)'; \
 	    echo 'exec_prefix=$${prefix}'; \
 	    echo 'libdir=$${exec_prefix}/$(LIBDIR)'; \
-	    echo 'includedir=$${prefix}/include'; \
+	    echo 'includedir=$${prefix}/include/openssl'; \
 	    echo ''; \
 	    echo 'Name: OpenSSL-libcrypto'; \
 	    echo 'Description: OpenSSL cryptography library'; \
@@ -374,7 +374,7 @@
 	@ ( echo 'prefix=$(INSTALLTOP)'; \
 	    echo 'exec_prefix=$${prefix}'; \
 	    echo 'libdir=$${exec_prefix}/$(LIBDIR)'; \
-	    echo 'includedir=$${prefix}/include'; \
+	    echo 'includedir=$${prefix}/include/openssl'; \
 	    echo ''; \
 	    echo 'Name: OpenSSL-libssl'; \
 	    echo 'Description: Secure Sockets Layer and cryptography libraries'; \
@@ -388,7 +388,7 @@
 	@ ( echo 'prefix=$(INSTALLTOP)'; \
 	    echo 'exec_prefix=$${prefix}'; \
 	    echo 'libdir=$${exec_prefix}/$(LIBDIR)'; \
-	    echo 'includedir=$${prefix}/include'; \
+	    echo 'includedir=$${prefix}/include/openssl'; \
 	    echo ''; \
 	    echo 'Name: OpenSSL'; \
 	    echo 'Description: Secure Sockets Layer and cryptography libraries and tools'; \ 
