--- data/filezilla.desktop.orig	2009-03-21 21:32:16.000000000 +0200
+++ data/filezilla.desktop	2009-03-22 00:02:49.000000000 +0200
@@ -3,9 +3,11 @@
 GenericName=FTP client
 GenericName[de]=FTP-Client
 GenericName[fr]=Client FTP
+GenericName[tr]=FTP istemcisi
 Comment=Download and upload files via FTP, FTPS and SFTP
 Comment[de]=Dateien über FTP, FTPS und SFTP übertragen
 Comment[fr]=Transférer des fichiers via FTP, FTPS et SFTP
+Comment[tr]=Dosyaları ftp, sftp, ftps ile gönderin ve indirin
 Exec=filezilla
 Terminal=false
 Icon=filezilla
