Index: libuser-0.56.17/docs/Makefile.am
===================================================================
--- libuser-0.56.17.orig/docs/Makefile.am
+++ libuser-0.56.17/docs/Makefile.am
@@ -1,12 +1,10 @@
-SUBDIRS = reference
-
 man_MANS = libuser.conf.5
 
 CLEANFILES = $(man_MANS) sgml/libuser.html sgml/libuser-1.html \
 	sgml/libuser-2.html sgml/libuser.txt
 EXTRA_DIST = attributes.txt libuser.conf.5.in rfc2307.txt sgml/libuser.sgml
 
-all: sgml/libuser.txt sgml/libuser.html
+all: libuser.conf.5
 
 libuser.conf.5: $(srcdir)/libuser.conf.5.in Makefile
 	sed 's,@sysconfdir\@,$(sysconfdir),g' \
