--- lib/util.c	2010-05-20 15:46:11.000000000 +0200
+++ lib/util.c	2010-05-20 16:23:28.000000000 +0200
@@ -116,14 +116,14 @@
 }
 
 static const struct {
-	const char initial[5];
+	const char initial[8];
 	char separator[2];
 	size_t salt_length;
 	gboolean sha_rounds;
 } salt_type_info[] = {
 	{"$1$", "$", 8, FALSE },
 	/* FIXME: number of rounds, base64 of 128 bits */
-	{"$2a$", "$", 8, FALSE },
+	{"$2a$08$", "$", 22, FALSE },
 	{"$5$", "$", 16, TRUE },
 	{"$6$", "$", 16, TRUE },
 	{ "", "", 2 },
@@ -230,7 +230,7 @@
 	} salt_types[] = {
 		{ "des", "", FALSE },
 		{ "md5", "$1$", FALSE },
-		{ "blowfish", "$2a$", FALSE },
+		{ "blowfish", "$2a$08$", FALSE },
 		{ "sha256", "$5$", TRUE },
 		{ "sha512", "$6$", TRUE },
 	};
