diff -up polkit-qt-1-0.95.1/cmake/modules/CMakeLists.txt.install-cmake-find polkit-qt-1-0.95.1/cmake/modules/CMakeLists.txt
--- polkit-qt-1-0.95.1/cmake/modules/CMakeLists.txt.install-cmake-find	2009-12-23 12:31:13.000000000 +0100
+++ polkit-qt-1-0.95.1/cmake/modules/CMakeLists.txt	2010-01-14 15:22:30.100520720 +0100
@@ -1,9 +1,9 @@
  
 ## install the cmake files
 
-#file( GLOB cmakeFiles FindPolkitQt.cmake )
-#set(module_install_dir ${CMAKE_ROOT}/Modules )
+file( GLOB cmakeFiles FindPolkitQt-1.cmake )
+set(module_install_dir ${CMAKE_ROOT}/Modules )
 
-#install( FILES ${cmakeFiles} 
-#         DESTINATION ${module_install_dir}  )
+install( FILES ${cmakeFiles} 
+         DESTINATION ${module_install_dir}  )
 
