--- binutils-2.23.51.0.8-orig/ld/emulparams/elf_i386.sh	2012-12-21 21:40:41.000000000 +0200
+++ binutils-2.23.51.0.8/ld/emulparams/elf_i386.sh	2013-01-16 18:15:18.834591275 +0200
@@ -15,6 +15,16 @@
 IREL_IN_PLT=
 SHARABLE_SECTIONS=yes
 
+# In Pardus, we install 32bit libs into /lib32 in an
+# ABI setup with amd64/x86
+case "$target" in
+  x86_64*-linux*)
+    case "$EMULATION_NAME" in
+      *i386*) LIBPATH_SUFFIX=32  ;;
+    esac
+  ;;
+esac
+
 # Linux modify the default library search path to first include
 # a 32-bit specific directory.
 case "$target" in
