--- mplex/stream_params.hpp	2010/03/13 13:05:07	1.6
+++ mplex/stream_params.hpp	2011/06/11 14:28:39	1.7
@@ -19,11 +19,10 @@
  *  along with this program; if not, write to the Free Software
  *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
  */
-
+
 #ifndef __STREAM_PARAMS_HH__
 #define __STREAM_PARAMS_HH__
 
-#include <config.h>
 #include <mjpeg_types.h>
 class LpcmParams
 {
