--- mjpegtools-2.0.0/lavtools/jpeg2yuv.c.format-strings	2008-01-06 04:28:00.000000000 +0100
+++ mjpegtools-2.0.0/lavtools/jpeg2yuv.c	2011-05-24 11:16:32.000000000 +0200
@@ -451,7 +451,7 @@ static int generate_YUV4MPEG(parameters_
            
            if (firstjpeg) {
                p = firstjpeg;
-               sprintf(jpegname, firstjpeg);
+               sprintf(jpegname, "%s", firstjpeg);
                firstjpeg = NULL;
            }
            else {
--- mjpegtools-2.0.0/lavtools/lav_io.c.format-strings	2007-09-10 00:16:53.000000000 +0200
+++ mjpegtools-2.0.0/lavtools/lav_io.c	2011-05-24 11:16:32.000000000 +0200
@@ -926,7 +926,7 @@ lav_file_t *lav_open_input_file(char *fi
             strcat(errmsg, ", quicktime");
 #endif
 	    strcat(errmsg, ").\n");
-            fprintf(stderr, errmsg);
+            fprintf(stderr, "%s", errmsg);
 	    free(lav_fd);
 	    internal_error = ERROR_FORMAT; /* Format not recognized */
 	    return 0;
--- mjpegtools-2.0.0/lavtools/lavtrans.c.format-strings	2005-10-02 19:01:30.000000000 +0200
+++ mjpegtools-2.0.0/lavtools/lavtrans.c	2011-05-24 11:16:32.000000000 +0200
@@ -275,7 +275,7 @@ int main(int argc, char ** argv)
    if (process_image_frame != -1)
    {
       nv = el_get_video_frame(vbuff, process_image_frame, &el);
-      sprintf(imgfname,outfile);
+      sprintf(imgfname,"%s", outfile);
       imgfd = fopen(imgfname,"wb");
       if (imgfd==0) system_error("opening image file","fopen");
       res = fwrite(vbuff,nv,1,imgfd);
--- mjpegtools-2.0.0/mplex/main.cpp.format-strings	2010-03-13 14:05:07.000000000 +0100
+++ mjpegtools-2.0.0/mplex/main.cpp	2011-05-24 11:16:32.000000000 +0200
@@ -615,7 +615,7 @@ bool CmdLineMultiplexJob::ParseWorkaroun
                     message += sep;
             }
             while( flag_table[flag].longname != 0 );
-            mjpeg_error( message.c_str() );
+            mjpeg_error( "%s", message.c_str() );
             return false;
         }
 
--- mjpegtools-2.0.0/yuvfilters/yuvfilters.h.format-strings	2005-02-27 23:46:56.000000000 +0100
+++ mjpegtools-2.0.0/yuvfilters/yuvfilters.h	2011-05-24 11:16:32.000000000 +0200
@@ -35,7 +35,7 @@ extern "C" {
 #ifdef MJPEGTOOLS
 # define perror(MSG) mjpeg_error("%s: %s", MSG, strerror(errno))
 # define WERROR(MSG) mjpeg_error(MSG)
-# define WERRORL(MSG) mjpeg_error(MSG)
+# define WERRORL(MSG) mjpeg_error("%s", MSG)
 # define WWARN(MSG) mjpeg_warn(MSG)
 #else
 # define WERROR(MSG) write(2, MSG "\n", sizeof MSG)
--- mjpegtools-2.0.0/yuvfilters/yuvkineco.c.format-strings	2005-12-05 06:06:06.000000000 +0100
+++ mjpegtools-2.0.0/yuvfilters/yuvkineco.c	2011-05-24 11:16:32.000000000 +0200
@@ -48,7 +48,7 @@ buf_debug(char *buf, FILE *fp, const cha
     } else {
 #ifdef MJPEGTOOLS
       buf[n] = '\0';
-      mjpeg_debug(buf);
+      mjpeg_debug("%s", buf);
 #else
       fputs(buf, stderr);
 #endif
