--- mjpegtools-1.9.0rc1/configure.ac.x86_64	2007-01-19 18:40:11.000000000 +0100
+++ mjpegtools-1.9.0rc1/configure.ac	2007-03-02 14:18:11.000000000 +0100
@@ -442,7 +442,7 @@
     fi
     AC_MSG_CHECKING(sub-architecture settings)
     chmod +x $srcdir/cpuinfo.sh
-    if test x$have_x86cpu = xtrue; then
+    if test x$have_x86cpu = xtrue -a x$have_x86_64cpu != xtrue; then
        ARCHFLAGS=`$srcdir/cpuinfo.sh x86`
     elif test x$have_ppccpu = xtrue; then
        ARCHFLAGS=`$srcdir/cpuinfo.sh ppc`
