From 3230d1d0f2afbae4145f11437f623d1611d855a2 Mon Sep 17 00:00:00 2001
From: Linus Walleij <triad@df.lth.se>
Date: Mon, 24 Jan 2011 19:31:25 +0100
Subject: [PATCH] Fix inverted boolean logic...

---
 src/libusb-glue.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/libusb-glue.c b/src/libusb-glue.c
index ca8c47d..a702eed 100644
--- a/src/libusb-glue.c
+++ b/src/libusb-glue.c
@@ -237,9 +237,9 @@ static int probe_device_descriptor(struct usb_device *dev, FILE *dumpfile)
    * contain any MTP interface, update this the day
    * you find some weird combination...
    */
-  if (dev->descriptor.bDeviceClass != USB_CLASS_PER_INTERFACE &&
-      dev->descriptor.bDeviceClass != USB_CLASS_COMM &&
-      dev->descriptor.bDeviceClass != USB_CLASS_PTP &&
+  if (dev->descriptor.bDeviceClass != USB_CLASS_PER_INTERFACE ||
+      dev->descriptor.bDeviceClass != USB_CLASS_COMM ||
+      dev->descriptor.bDeviceClass != USB_CLASS_PTP ||
       dev->descriptor.bDeviceClass != USB_CLASS_VENDOR_SPEC) {
     return 0;
   }
-- 
1.7.3.4 
