From 8270d2853f2962facdf6f9ea6d669922d6c78aba Mon Sep 17 00:00:00 2001
Message-Id: <8270d2853f2962facdf6f9ea6d669922d6c78aba.1316295404.git.kevin.kofler@chello.at>
From: Kevin Kofler <kevin.kofler@chello.at>
Date: Sat, 17 Sep 2011 23:34:23 +0200
Subject: [PATCH] K3b::IsOverburnAllowed: Don't count used capacity twice.

CCBUG: 276002
---
 libk3b/core/k3bglobals.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libk3b/core/k3bglobals.cpp b/libk3b/core/k3bglobals.cpp
index 28c12c6..cdead56 100644
--- a/libk3b/core/k3bglobals.cpp
+++ b/libk3b/core/k3bglobals.cpp
@@ -595,7 +595,7 @@ bool K3b::IsOverburnAllowed( const K3b::Msf& projectSize, const K3b::Msf& capaci
 bool K3b::IsOverburnAllowed( const Msf& projectSize, const Msf& capacity, const Msf& usedCapacity )
 {
     return( k3bcore->globalSettings()->overburn() &&
-        (projectSize + usedCapacity) <= ( capacity.lba() - usedCapacity.lba() + capacity.lba() / 4 ) ); // 25% tolerance in overburn mode
+        (projectSize + usedCapacity) <= ( capacity.lba() + capacity.lba() / 4 ) ); // 25% tolerance in overburn mode
 }
 
 
-- 
1.7.6
