Subject: Make K3B the default handler for .iso files
From: Lubos Lunak
Patch-upstream: no

Otherwise Ark is the default handler.

--- a/src/k3b.desktop     2010-06-16 21:36:44.000000000 +0200
+++ b/src/k3b.desktop     2010-06-18 16:54:12.154664826 +0200
@@ -140,3 +140,4 @@ X-KDE-StartupNotify=true
 X-KDE-NativeMimeType=application/x-k3b;
 Categories=KDE;Qt;AudioVideo;DiscBurning;
 X-DBUS-ServiceType=Unique
+InitialPreference=5
