diff -up ggz-base-libs-snapshot-0.99.5/configure.ac.nss3 ggz-base-libs-snapshot-0.99.5/configure.ac
diff -up ggz-base-libs-snapshot-0.99.5/configure.nss3 ggz-base-libs-snapshot-0.99.5/configure
--- ggz-base-libs-snapshot-0.99.5/configure.nss3	2009-01-02 13:48:05.000000000 -0600
+++ ggz-base-libs-snapshot-0.99.5/configure	2010-01-28 10:47:12.238459707 -0600
@@ -18004,8 +18004,8 @@ if test "$inc" != "" -a "$lib" != ""; th
 fi
 
   if test "$have_nss" = yes; then
-    GGZTLS_INCLUDES="-I $nss_includes"
-    GGZTLS_LDFLAGS="-L$nss_libraries"
+    GGZTLS_INCLUDES="$nss_includes"
+    GGZTLS_LDFLAGS="$nss_libraries"
     LIB_GGZTLS=$nss_lib
     { $as_echo "$as_me:$LINENO: result: using NSS" >&5
 $as_echo "using NSS" >&6; }
diff -up ggz-base-libs-snapshot-0.99.5/libggz/security/ggz_tls_nss.c.nss3 ggz-base-libs-snapshot-0.99.5/libggz/security/ggz_tls_nss.c
--- ggz-base-libs-snapshot-0.99.5/libggz/security/ggz_tls_nss.c.nss3	2009-01-02 13:32:34.000000000 -0600
+++ ggz-base-libs-snapshot-0.99.5/libggz/security/ggz_tls_nss.c	2010-01-28 10:41:01.384460653 -0600
@@ -26,8 +26,8 @@
 #include <unistd.h>
 
 #include <prinit.h>
-#include <nss/nss.h> // FIXME: evil path override!
-#include <nss/ssl.h>
+#include <nss.h> // FIXME: evil path override!
+#include <ssl.h>
 
 #include "ggz.h"
 
diff -up ggz-base-libs-snapshot-0.99.5/m4/tls.m4.nss3 ggz-base-libs-snapshot-0.99.5/m4/tls.m4
--- ggz-base-libs-snapshot-0.99.5/m4/tls.m4.nss3	2009-01-02 13:32:33.000000000 -0600
+++ ggz-base-libs-snapshot-0.99.5/m4/tls.m4	2010-01-28 10:46:40.208460985 -0600
@@ -345,8 +345,8 @@ then
   dnl NSS check
   AC_GGZ_NSS
   if test "$have_nss" = yes; then
-    GGZTLS_INCLUDES="-I $nss_includes"
-    GGZTLS_LDFLAGS="-L$nss_libraries"
+    GGZTLS_INCLUDES="$nss_includes"
+    GGZTLS_LDFLAGS="$nss_libraries"
     LIB_GGZTLS=$nss_lib
     AC_MSG_RESULT([using NSS])
     AC_DEFINE_UNQUOTED([GGZ_TLS_NSS], 1,
