--- qhull-2003.1/src/qset.c.alias	2008-02-27 14:46:42.000000000 -0700
+++ qhull-2003.1/src/qset.c	2008-02-28 16:44:20.000000000 -0700
@@ -126,8 +126,7 @@
     
 */
 void qh_setappend(setT **setp, void *newelem) {
-  int *sizep;
-  void **endp;
+  int *sizep, end_idx;
 
   if (!newelem)
     return;
@@ -135,8 +134,9 @@
     qh_setlarger(setp);
     sizep= SETsizeaddr_(*setp);
   }
-  *(endp= &((*setp)->e[(*sizep)++ - 1].p))= newelem;
-  *(++endp)= NULL;
+  end_idx = (*sizep)++ - 1;
+  (*setp)->e[end_idx].p = newelem;
+  (*setp)->e[end_idx + 1].p = NULL;
 } /* setappend */
 
 /*-<a                             href="qh-set.htm#TOC"
