--- a/include/freetype/config/ftoption.h	5 Jan 2007 14:48:27 -0000	1.118
+++ b/include/freetype/config/ftoption.h	12 Apr 2007 14:06:41 -0000
@@ -605,7 +605,7 @@
    * is recommended to disable the macro since it reduces the library's code
    * size and activates a few memory-saving optimizations as well.
    */
-#define FT_CONFIG_OPTION_OLD_INTERNALS
+#undef FT_CONFIG_OPTION_OLD_INTERNALS
 
 
   /*
