--- src/truetype/ttinterp.c.OSmanOS     2012-12-18 09:02:44.000000000 +0200
+++ src/truetype/ttinterp.c             2013-02-14 20:32:00.538646369 +0200
@@ -7540,7 +7540,6 @@
   {
     FT_Long  K;
 
-
     K = 0;
 
 #ifdef TT_CONFIG_OPTION_SUBPIXEL_HINTING
@@ -7559,7 +7558,8 @@
     else
 #endif /* TT_CONFIG_OPTION_SUBPIXEL_HINTING */
       if ( ( args[0] & 1 ) != 0 )
-        K = 35;
+      /*K = 35;*/
+      K = 37;
 
     /********************************/
     /* GLYPH ROTATED                */
@@ -7575,7 +7575,7 @@
     /* Return Bit(s): 9             */
     /*                              */
     if ( ( args[0] & 4 ) != 0 && CUR.tt_metrics.stretched )
-      K |= 1 << 8;
+      K |= (1 << 8);
 
     /********************************/
     /* HINTING FOR GRAYSCALE        */
@@ -7583,7 +7583,17 @@
     /* Return Bit(s): 12            */
     /*                              */
     if ( ( args[0] & 32 ) != 0 && CUR.grayscale )
-      K |= 1 << 12;
+      K |= (1 << 12);
+#if 1
+    /* If ClearType widths are compatible with mono- and grayscale widths */
+    if ( ( args[0] & (1 << 6) ) != 0 && CUR.grayscale )
+      K |= (1 << 13); 
+#endif
+#if 0
+    /* If ClearType symmetrical smoothing is being used */
+    if ( ( args[0] & (1 << 7) ) )
+      K |= (1 << 14);
+#endif
 
 #ifdef TT_CONFIG_OPTION_SUBPIXEL_HINTING
     if ( CUR.ignore_x_mode && CUR.rasterizer_version >= 35 )
