diff -Nuar pigment-python-0.3.12.orig/pgm/Makefile.am pigment-python-0.3.12/pgm/Makefile.am
--- pigment-python-0.3.12.orig/pgm/Makefile.am	2010-09-06 01:49:15.469098320 +0300
+++ pigment-python-0.3.12/pgm/Makefile.am	2010-09-06 02:08:29.185850148 +0300
@@ -55,6 +55,7 @@
 	@PYPGM_LDFLAGS@ \
 	@PIGMENT_LIBS@ \
 	-module -avoid-version \
+	`python-config --libs` \
 	-export-symbols-regex "^(init_pgm|_PyGObject_API).*"
 
 _pgmmodule_la_CFLAGS = @PYPGM_CFLAGS@
@@ -78,6 +79,7 @@
 	@PYPGM_IMAGING_LDFLAGS@ \
 	@PIGMENT_IMAGING_LIBS@ \
 	-module -avoid-version \
+	`python-config --libs` \
 	-export-symbols-regex "^(init_pgmimaging|_PyGObject_API).*"
 
 _pgmimagingmodule_la_CFLAGS = @PYPGM_IMAGING_CFLAGS@
@@ -103,6 +105,7 @@
 	@PYPGM_GTK_LDFLAGS@ \
 	@PIGMENT_GTK_LIBS@ \
 	-module -avoid-version \
+	`python-config --libs` \
 	-export-symbols-regex "^(init_pgmgtk|_PyGObject_API).*"
 
 _pgmgtkmodule_la_CFLAGS = @PYPGM_GTK_CFLAGS@
