Index: busybox-1.15.3/shell/ash.c
===================================================================
--- busybox-1.15.3.orig/shell/ash.c
+++ busybox-1.15.3/shell/ash.c
@@ -3642,7 +3642,7 @@ setjobctl(int on)
 			pgrp = tcgetpgrp(fd);
 			if (pgrp < 0) {
  out:
-				ash_msg("can't access tty; job control turned off");
+				/*ash_msg("can't access tty; job control turned off");*/
 				mflag = on = 0;
 				goto close;
 			}
