diff -up openexr-1.6.1/OpenEXR.pc.in.pkgconfig openexr-1.6.1/OpenEXR.pc.in
--- openexr-1.6.1/OpenEXR.pc.in.pkgconfig	2007-03-30 12:14:05.000000000 -0500
+++ openexr-1.6.1/OpenEXR.pc.in	2008-01-09 10:08:53.000000000 -0600
@@ -7,5 +7,7 @@ OpenEXR_includedir=@includedir@/OpenEXR
 Name: OpenEXR
 Description: OpenEXR image library
 Version: @OPENEXR_VERSION@
-Libs: -L${libdir} @ILMBASE_LDFLAGS@ -lIlmImf -lz @ILMBASE_LIBS@
-Cflags: @ILMBASE_CXXFLAGS@ -I${OpenEXR_includedir}
+Requires: IlmBase
+Libs: -L${libdir} -lIlmImf
+Libs.private: -lz
+Cflags: -I${OpenEXR_includedir}
