diff -Nur lirc-0.8.4a-old/drivers/lirc_gpio/Makefile.am lirc-0.8.4a/drivers/lirc_gpio/Makefile.am
--- lirc-0.8.4a-old/drivers/lirc_gpio/Makefile.am	2009-04-21 19:07:37.000000000 +0300
+++ lirc-0.8.4a/drivers/lirc_gpio/Makefile.am	2009-04-21 19:07:38.000000000 +0300
@@ -8,6 +8,6 @@
 automake_dummy_SOURCES = lirc_gpio.c
 
 ## there is no *just* object file support in automake.  This is close enough
-module_DATA = lirc_gpio.o
+module_DATA = 
 
 include ../Makefile.common
