diff -Nur lirc-0.9.0-pre1-old//tools/xmode2.c lirc-0.9.0-pre1/tools/xmode2.c
--- lirc-0.9.0-pre1-old//tools/xmode2.c	2010-12-27 23:17:34.673000031 +0200
+++ lirc-0.9.0-pre1/tools/xmode2.c	2010-12-27 23:29:47.825000031 +0200
@@ -69,7 +69,7 @@
 Window w0, w1;			/*w0 = root */
 char w1_wname[] = "xmode2";
 char w1_iname[] = "xmode2";
-char font1_name[] = "-*-Courier-medium-r-*-*-8-*-*-m-*-iso8859-1";
+char font1_name[] = "-misc-fixed-*-r-*-*-12-*-*-*-*-*-iso8859-1";
 
 int w1_x = 0, w1_y = 0;
 unsigned int w1_w = 640, w1_h = 480, w1_border = 0;
