diff -up automake-1.11.1/tests/Makefile.in.kh automake-1.11.1/tests/Makefile.in
--- automake-1.11.1/tests/Makefile.in.kh	2011-12-01 16:16:56.000000000 +0100
+++ automake-1.11.1/tests/Makefile.in	2011-12-01 16:17:57.000000000 +0100
@@ -410,7 +410,6 @@ cond31.test \
 cond32.test \
 cond33.test \
 cond34.test \
-cond35.test \
 cond36.test \
 cond37.test \
 cond38.test \
@@ -600,7 +599,6 @@ ldadd.test \
 ldflags.test \
 lex.test \
 lex2.test \
-lex3.test \
 lex4.test \
 lex5.test \
 libexec.test \
@@ -817,7 +815,6 @@ silent.test \
 silent2.test \
 silent3.test \
 silent4.test \
-silent5.test \
 silent6.test \
 silent7.test \
 sinclude.test \
