diff --git a/t/list-of-tests.mk b/t/list-of-tests.mk
index 591b4f0..20a9398 100644
--- a/t/list-of-tests.mk
+++ b/t/list-of-tests.mk
@@ -130,8 +130,6 @@ t/amhello-cross-compile.sh \
 t/amhello-binpkg.sh \
 t/aminit-moreargs-deprecation.sh \
 t/amassign.sh \
-t/am-config-header-no-more.sh \
-t/am-prog-cc-stdc-no-more.sh \
 t/am-macro-not-found.sh \
 t/amopt.sh \
 t/amopts-location.sh \
@@ -1202,16 +1200,6 @@ t/upc2.sh \
 t/upc3.sh \
 t/vala-configure.sh \
 t/vala-grepping.sh \
-t/vala-headers.sh \
-t/vala-libs.sh \
-t/vala-mix.sh \
-t/vala-mix2.sh \
-t/vala-non-recursive-setup.sh \
-t/vala-parallel.sh \
-t/vala-per-target-flags.sh \
-t/vala-recursive-setup.sh \
-t/vala-vapi.sh \
-t/vala-vpath.sh \
 t/vars.sh \
 t/vars3.sh \
 t/var-recurs.sh \
