diff --git a/m4/obsolete-err.m4 b/m4/obsolete-err.m4
index d8119a4..6a09401 100644
--- a/m4/obsolete-err.m4
+++ b/m4/obsolete-err.m4
@@ -8,19 +8,56 @@
 # gives unlimited permission to copy and/or distribute it,
 # with or without modifications, as long as this notice is preserved.
 
-dnl TODO: Remove in Automake 1.15.
-AC_DEFUN([AM_CONFIG_HEADER],
-[AC_FATAL(['$0': this macro is obsolete.
-    You should use the 'AC][_CONFIG_HEADERS' macro instead.])])
-
-dnl TODO: Remove in Automake 1.15.
-AC_DEFUN([AM_PROG_CC_STDC],
-[AC_FATAL(['$0': this macro is obsolete.
-    You should simply use the 'AC][_PROG_CC' macro instead.
-    Also, your code should no longer depend upon 'am_cv_prog_cc_stdc',
-    but upon 'ac_cv_prog_cc_stdc'.])])
+# dnl TODO: Remove in Automake 1.15.
+# AC_DEFUN([AM_CONFIG_HEADER],
+# [AC_FATAL(['$0': this macro is obsolete.
+#     You should use the 'AC][_CONFIG_HEADERS' macro instead.])])
+#
+# dnl TODO: Remove in Automake 1.15.
+# AC_DEFUN([AM_PROG_CC_STDC],
+# [AC_FATAL(['$0': this macro is obsolete.
+#     You should simply use the 'AC][_PROG_CC' macro instead.
+#     Also, your code should no longer depend upon 'am_cv_prog_cc_stdc',
+#     but upon 'ac_cv_prog_cc_stdc'.])])
 
 dnl TODO: Remove in Automake 1.14.
 AC_DEFUN([AM_C_PROTOTYPES],
          [AC_FATAL([automatic de-ANSI-fication support has been removed])])
 AU_DEFUN([fp_C_PROTOTYPES], [AM_C_PROTOTYPES])
+
+AU_DEFUN([AM_PROG_CC_STDC],
+[AC_PROG_CC
+AC_DIAGNOSE([obsolete], [$0:
+       your code should no longer depend upon 'am_cv_prog_cc_stdc', but upon
+       'ac_cv_prog_cc_stdc'.  Remove this warning and the assignment when
+       you adjust the code.  You can also remove the above call to
+       AC_PROG_CC if you already called it elsewhere.])
+am_cv_prog_cc_stdc=$ac_cv_prog_cc_stdc
+])
+
+# Reenable these macros in fedora until all packages are fixed (unlikely).
+# TODO: Prepare the some fedora-review checker, or rpmlint checker for this.
+
+AU_DEFUN([AM_CONFIG_HEADER], [AC_CONFIG_HEADERS($@)])
+AU_DEFUN([AM_SANITY_CHECK_CC], [AC_PROG_CC])
+AU_DEFUN([fp_PROG_CC_STDC])
+AU_DEFUN([fp_WITH_DMALLOC], [AM_WITH_DMALLOC])
+AU_DEFUN([AM_CONFIG_HEADER], [AC_CONFIG_HEADERS($@)])
+AU_DEFUN([ud_PATH_LISPDIR], [AM_PATH_LISPDIR])
+AU_DEFUN([jm_MAINTAINER_MODE], [AM_MAINTAINER_MODE])
+AU_DEFUN([ud_GNU_GETTEXT], [AM_GNU_GETTEXT])
+AU_DEFUN([gm_PROG_LIBTOOL], [AM_PROG_LIBTOOL])
+AU_DEFUN([AC_FEATURE_CTYPE],     [AC_HEADER_STDC])
+AU_DEFUN([AC_FEATURE_ERRNO],     [AC_REPLACE_FUNCS([strerror])])
+AU_DEFUN([AM_CYGWIN32],                 [AC_CYGWIN])
+AU_DEFUN([AM_EXEEXT],            [AC_EXEEXT])
+AU_DEFUN([AM_FUNC_MKTIME],       [AC_FUNC_MKTIME])
+AU_DEFUN([AM_HEADER_TIOCGWINSZ_NEEDS_SYS_IOCTL],
+                                [AC_HEADER_TIOCGWINSZ])
+AU_DEFUN([AM_MINGW32],           [AC_MINGW32])
+AU_DEFUN([AM_PROG_INSTALL],      [AC_PROG_INSTALL])
+AU_DEFUN([AM_SANITY_CHECK_CC],   [AC_PROG_CC])
+AU_DEFUN([AM_SYS_POSIX_TERMIOS], [AC_SYS_POSIX_TERMIOS])
+AU_DEFUN([fp_FUNC_FNMATCH],      [AC_FUNC_FNMATCH])
+AU_DEFUN([fp_PROG_INSTALL],      [AC_PROG_INSTALL])
+AU_DEFUN([md_TYPE_PTRDIFF_T],    [AC_CHECK_TYPES([ptrdiff_t])])
