--- creator/gluon-creator.desktop.OSmanOS	2011-04-23 20:35:07.000000000 +0300
+++ creator/gluon-creator.desktop	        2013-05-26 20:53:16.271238370 +0300
@@ -11,6 +11,7 @@
 Name[pt_BR]=Criador do Gluon
 Name[sv]=Gluon Creator
 Name[uk]=Gluon Creator
+Name[tr]=Gluon Üreticisi
 Name[x-test]=xxGluon Creatorxx
 GenericName=Game Creation Tool
 GenericName[de]=Werkzeug zum Erstellen von Spielen
@@ -24,6 +25,7 @@
 GenericName[pt_BR]=Ferramenta de criação de jogos
 GenericName[sv]=Verktyg för att skapa spel
 GenericName[uk]=Інструмент для створення ігор
+GenericName[tr]=Oyun Oluşturma Aracı
 GenericName[x-test]=xxGame Creation Toolxx
 Exec=gluoncreator
 Icon=gluon_creator
