diff -urNp w3m-0.5.2/istream.c w3m-0.5.2-glibc/istream.c
--- w3m-0.5.2/istream.c	2011-06-27 09:00:48.753053138 +0530
+++ w3m-0.5.2-glibc/istream.c	2011-06-27 09:13:01.419642857 +0530
@@ -22,8 +22,8 @@
 static void basic_close(int *handle);
 static int basic_read(int *handle, char *buf, int len);
 
-static void file_close(struct file_handle *handle);
-static int file_read(struct file_handle *handle, char *buf, int len);
+static void file_close(struct io_file_handle *handle);
+static int file_read(struct io_file_handle *handle, char *buf, int len);
 
 static int str_read(Str handle, char *buf, int len);
 
@@ -114,7 +114,7 @@ newFileStream(FILE * f, void (*closep) (
     stream = New(union input_stream);
     init_base_stream(&stream->base, STREAM_BUF_SIZE);
     stream->file.type = IST_FILE;
-    stream->file.handle = New(struct file_handle);
+    stream->file.handle = New(struct io_file_handle);
     stream->file.handle->f = f;
     if (closep)
 	stream->file.handle->close = closep;
@@ -658,13 +658,13 @@ basic_read(int *handle, char *buf, int l
 }
 
 static void
-file_close(struct file_handle *handle)
+file_close(struct io_file_handle *handle)
 {
     handle->close(handle->f);
 }
 
 static int
-file_read(struct file_handle *handle, char *buf, int len)
+file_read(struct io_file_handle *handle, char *buf, int len)
 {
     return fread(buf, 1, len, handle->f);
 }
diff -urNp w3m-0.5.2/istream.h w3m-0.5.2-glibc/istream.h
--- w3m-0.5.2/istream.h	2003-10-20 22:11:56.000000000 +0530
+++ w3m-0.5.2-glibc/istream.h	2011-06-27 08:49:14.175900216 +0530
@@ -20,7 +20,7 @@ struct stream_buffer {
 
 typedef struct stream_buffer *StreamBuffer;
 
-struct file_handle {
+struct io_file_handle {
     FILE *f;
     void (*close) ();
 };
@@ -53,7 +53,7 @@ struct base_stream {
 
 struct file_stream {
     struct stream_buffer stream;
-    struct file_handle *handle;
+    struct io_file_handle *handle;
     char type;
     char iseos;
     int (*read) ();
