--- PyXML-0.8.4-old/xml/parsers/xmlproc/xmlutils.py	2008-07-09 15:32:50.000000000 +0300
+++ PyXML-0.8.4/xml/parsers/xmlproc/xmlutils.py	2008-07-09 15:35:41.000000000 +0300
@@ -720,6 +720,8 @@
             # to the recoding.
             try:
                 self.data = self.charset_converter(self.data)
+                # patched by eric4 install script.
+                self.datasize = len(self.data)
             except UnicodeError, e:
                 self._handle_decoding_error(self.data, e)
             self.input_encoding = enc1
