diff -Naur biopython-1.53.orig/setup.py biopython-1.53/setup.py
--- biopython-1.53.orig/setup.py	2009-12-15 15:57:21.000000000 +0200
+++ biopython-1.53/setup.py	2009-12-22 11:20:19.692827628 +0200
@@ -293,13 +293,12 @@
                'Bio/csupport.c'],
               include_dirs=["Bio"]
               ),
-#Commented out due to the build dependency on flex, see Bug 2619
-#   Extension('Bio.PDB.mmCIF.MMCIFlex',
-#              ['Bio/PDB/mmCIF/lex.yy.c',
-#               'Bio/PDB/mmCIF/MMCIFlexmodule.c'],
-#              include_dirs=["Bio"],
-#              libraries=["fl"]
-#              ),
+   Extension('Bio.PDB.mmCIF.MMCIFlex',
+              ['Bio/PDB/mmCIF/lex.yy.c',
+               'Bio/PDB/mmCIF/MMCIFlexmodule.c'],
+              include_dirs=["Bio"],
+              libraries=["fl"]
+              ),
     Extension('Bio.Nexus.cnexus',
               ['Bio/Nexus/cnexus.c']
               ),
