Index: yakuake-2.9.7/app/config/appearancesettings.ui
===================================================================
--- yakuake-2.9.7.orig/app/config/appearancesettings.ui
+++ yakuake-2.9.7/app/config/appearancesettings.ui
@@ -6,7 +6,7 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>307</width>
+    <width>373</width>
     <height>355</height>
    </rect>
   </property>
@@ -70,81 +70,6 @@
         </item>
        </layout>
       </item>
-      <item>
-       <widget class="QCheckBox" name="kcfg_Translucency">
-        <property name="whatsThis">
-         <string comment="@info:whatsthis">Translucency requires support by the selected skin as well as your desktop environment. The default skin supports translucency.</string>
-        </property>
-        <property name="text">
-         <string comment="@option:check">Use skin translucency if supported</string>
-        </property>
-       </widget>
-      </item>
-      <item>
-       <layout class="QHBoxLayout" name="horizontalLayout">
-        <property name="leftMargin">
-         <number>7</number>
-        </property>
-        <item>
-         <spacer name="spacer">
-          <property name="orientation">
-           <enum>Qt::Horizontal</enum>
-          </property>
-          <property name="sizeType">
-           <enum>QSizePolicy::Fixed</enum>
-          </property>
-          <property name="sizeHint" stdset="0">
-           <size>
-            <width>20</width>
-            <height>20</height>
-           </size>
-          </property>
-         </spacer>
-        </item>
-        <item>
-         <widget class="QLabel" name="backgroundColorOpacityLabel">
-          <property name="enabled">
-           <bool>false</bool>
-          </property>
-          <property name="text">
-           <string comment="@label:spinbox">Background color opacity:</string>
-          </property>
-         </widget>
-        </item>
-        <item>
-         <widget class="QSpinBox" name="kcfg_BackgroundColorOpacity">
-          <property name="enabled">
-           <bool>false</bool>
-          </property>
-          <property name="whatsThis">
-           <string comment="@info:whatsthis">When using translucency, the background color will be filled in using the given opacity before the skin elements are painted on top of the background. This allows influencing the opacity and color tint of translucent areas of the skin.</string>
-          </property>
-          <property name="suffix">
-           <string comment="@label Spinbox unit suffix">%</string>
-          </property>
-          <property name="maximum">
-           <number>100</number>
-          </property>
-          <property name="value">
-           <number>50</number>
-          </property>
-         </widget>
-        </item>
-        <item>
-         <spacer name="spacer_2">
-          <property name="orientation">
-           <enum>Qt::Horizontal</enum>
-          </property>
-          <property name="sizeHint" stdset="0">
-           <size>
-            <width>64</width>
-            <height>17</height>
-           </size>
-          </property>
-         </spacer>
-        </item>
-       </layout>
-      </item>
      </layout>
     </widget>
    </item>
@@ -198,59 +123,24 @@
  </widget>
  <customwidgets>
   <customwidget>
-   <class>KColorButton</class>
-   <extends>QPushButton</extends>
-   <header>kcolorbutton.h</header>
-  </customwidget>
-  <customwidget>
    <class>KLineEdit</class>
    <extends>QLineEdit</extends>
    <header>klineedit.h</header>
   </customwidget>
+  <customwidget>
+   <class>KColorButton</class>
+   <extends>QPushButton</extends>
+   <header>kcolorbutton.h</header>
+  </customwidget>
  </customwidgets>
  <tabstops>
   <tabstop>kcfg_TerminalHighlightOnManualActivation</tabstop>
   <tabstop>kcfg_BackgroundColor</tabstop>
-  <tabstop>kcfg_Translucency</tabstop>
-  <tabstop>kcfg_BackgroundColorOpacity</tabstop>
   <tabstop>skinList</tabstop>
   <tabstop>installButton</tabstop>
   <tabstop>removeButton</tabstop>
   <tabstop>kcfg_Skin</tabstop>
  </tabstops>
  <resources/>
- <connections>
-  <connection>
-   <sender>kcfg_Translucency</sender>
-   <signal>toggled(bool)</signal>
-   <receiver>kcfg_BackgroundColorOpacity</receiver>
-   <slot>setEnabled(bool)</slot>
-   <hints>
-    <hint type="sourcelabel">
-     <x>152</x>
-     <y>76</y>
-    </hint>
-    <hint type="destinationlabel">
-     <x>192</x>
-     <y>106</y>
-    </hint>
-   </hints>
-  </connection>
-  <connection>
-   <sender>kcfg_Translucency</sender>
-   <signal>toggled(bool)</signal>
-   <receiver>backgroundColorOpacityLabel</receiver>
-   <slot>setEnabled(bool)</slot>
-   <hints>
-    <hint type="sourcelabel">
-     <x>152</x>
-     <y>76</y>
-    </hint>
-    <hint type="destinationlabel">
-     <x>95</x>
-     <y>106</y>
-    </hint>
-   </hints>
-  </connection>
- </connections>
+ <connections/>
 </ui>
Index: yakuake-2.9.7/app/config/yakuake.kcfg
===================================================================
--- yakuake-2.9.7.orig/app/config/yakuake.kcfg
+++ yakuake-2.9.7/app/config/yakuake.kcfg
@@ -80,23 +80,11 @@
       <whatsthis context="@info:whatsthis">The skin to be used by the application window.</whatsthis>
       <default>default</default>
     </entry>
-    <entry name="Translucency" type="Bool">
-      <label context="@label">Use translucent background</label>
-      <whatsthis context="@info:whatsthis">Whether to make use of XComposite ARGB translucency.</whatsthis>
-      <default>false</default>
-    </entry>
     <entry name="BackgroundColor" type="Color">
       <label context="@label">Window background color</label>
       <whatsthis context="@info:whatsthis">Background color painted below skin elements.</whatsthis>
       <default>#000000</default>
     </entry>
-    <entry name="BackgroundColorOpacity" type="Int">
-      <label context="@label">Background color opacity</label>
-      <whatsthis context="@info:whatsthis">The opacity of the background color fill when using translucency.</whatsthis>
-      <default>50</default>
-      <min>0</min>
-      <max>100</max>
-    </entry>
     <entry name="TerminalHighlightOnManualActivation" type="Bool">
       <label context="@label">Highlight newly focused terminals</label>
       <whatsthis context="@info:whatsthis">Whether to briefly display a colored overlay above the newly focused terminal when moving focus between multiple terminals in a session, as well as over the currently focused terminal when switching to a session with multiple terminals.</whatsthis>
Index: yakuake-2.9.7/app/mainwindow.cpp
===================================================================
--- yakuake-2.9.7.orig/app/mainwindow.cpp
+++ yakuake-2.9.7/app/mainwindow.cpp
@@ -769,9 +769,7 @@ void MainWindow::paintEvent(QPaintEvent*
         painter.setCompositionMode(QPainter::CompositionMode_Source);
         painter.fillRect(rect(), Qt::transparent);
 #endif
-        painter.setOpacity(qreal(Settings::backgroundColorOpacity()) / 100);
         painter.fillRect(rect(), Settings::backgroundColor());
-        painter.setOpacity(1.0);
     }
     else
     {
@@ -1080,17 +1078,13 @@ void MainWindow::updateUseTranslucency()
         ARGB = true;
     }
 
-    if (ARGB && Settings::translucency())
+    if (ARGB)
     {
         m_useTranslucency = KWindowSystem::compositingActive();
     }
     else
 #elif KDE_IS_VERSION(4,2,68)
-    if (Settings::translucency())
-    {
         m_useTranslucency = KWindowSystem::compositingActive();
-    }
-    else
 #endif
     {
         m_useTranslucency = false;
Index: yakuake-2.9.7/app/tabbar.cpp
===================================================================
--- yakuake-2.9.7.orig/app/tabbar.cpp
+++ yakuake-2.9.7/app/tabbar.cpp
@@ -838,9 +838,6 @@ void TabBar::startDrag(int index)
     QPixmap tab(tabWidth, height());
     QColor fillColor(Settings::backgroundColor());
 
-    if (m_mainWindow->useTranslucency())
-        fillColor.setAlphaF(qreal(Settings::backgroundColorOpacity()) / 100);
-
     tab.fill(fillColor);
 
     QPainter painter(&tab);
