diff -Nur enet-1.2-old/configure.in enet-1.2/configure.in
--- enet-1.2-old/configure.in	2009-08-07 12:38:43.213158556 +0300
+++ enet-1.2/configure.in	2009-08-07 12:38:51.322034526 +0300
@@ -2,7 +2,7 @@
 AM_INIT_AUTOMAKE(libenet.a, 1-8-2008)
 
 AC_PROG_CC
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 
 AC_CHECK_FUNC(gethostbyaddr_r, [AC_DEFINE(HAS_GETHOSTBYADDR_R)])
 AC_CHECK_FUNC(gethostbyname_r, [AC_DEFINE(HAS_GETHOSTBYNAME_R)])
diff -Nur enet-1.2-old/Makefile.am enet-1.2/Makefile.am
--- enet-1.2-old/Makefile.am	2009-08-07 12:38:43.213158556 +0300
+++ enet-1.2/Makefile.am	2009-08-07 12:39:19.286164927 +0300
@@ -1,5 +1,24 @@
-lib_LIBRARIES = libenet.a
-libenet_a_SOURCES = host.c list.c callbacks.c packet.c peer.c protocol.c unix.c win32.c
+lib_LTLIBRARIES = libenet.la
+libenet_la_SOURCES = host.c list.c callbacks.c packet.c peer.c protocol.c unix.c win32.c
 INCLUDES = -Iinclude/
 
 SUBDIRS = include
+
+#   Here are a set of rules to help you update your library version
+# information:
+#
+#   1. If the library source code has changed at all since the last
+#      update, then increment REVISION (`C:R:A' becomes `C:r+1:A').
+#
+#   2. If any interfaces have been added, removed, or changed since the
+#      last update, increment CURRENT, and set REVISION to 0.
+#
+#   3. If any interfaces have been added since the last public release,
+#      then increment AGE.
+#
+#   4. If any interfaces have been removed since the last public release,
+#      then set AGE to 0.
+
+libenet_la_LIBADD = @LTLIBOBJS@
+libenet_la_LDFLAGS = -version-info 2 # CURRENT[:REVISION[:AGE]]
+
