--- w3c-libwww-5.3.2/acinclude.m4~	Sat Mar  2 06:30:37 2002
+++ w3c-libwww-5.3.2/acinclude.m4	Sat Mar  2 06:30:37 2002
@@ -1,4 +1,4 @@
-AC_DEFUN(AC_SYS_AFS_SYSNAME, [
+AC_DEFUN([AC_SYS_AFS_SYSNAME], [
 AC_CHECK_PROG(ac_afsws_fs_found, fs, yes)
 AC_MSG_CHECKING(AFS system name)
 AC_CACHE_VAL(ac_cv_sys_afs_sysname, [ 
@@ -28,7 +28,7 @@
 dnl ])
 
 dnl AC_PATH_WAIS(ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND])
-AC_DEFUN(AC_PATH_WAIS, [
+AC_DEFUN([AC_PATH_WAIS], [
 AC_MSG_CHECKING(for WAIS)
 AC_CACHE_VAL(ac_cv_path_wais,
 [
@@ -53,7 +53,7 @@
 ])
 
 dnl AC_PATH_TCL(ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND])
-AC_DEFUN(AC_PATH_TCL, [
+AC_DEFUN([AC_PATH_TCL], [
 AC_MSG_CHECKING(for TCL)
 AC_CACHE_VAL(ac_cv_path_tcl,
 [
@@ -79,7 +79,7 @@
 ])
 
 dnl AC_PATH_SOCKS(ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND])
-AC_DEFUN(AC_PATH_SOCKS, [
+AC_DEFUN([AC_PATH_SOCKS], [
 AC_MSG_CHECKING(for SOCKS)
 AC_CACHE_VAL(ac_cv_path_socks,
 [
@@ -109,7 +109,7 @@
 ])
 
 dnl AC_PROG_CC_STATIC
-AC_DEFUN(AC_PROG_CC_STATIC,
+AC_DEFUN([AC_PROG_CC_STATIC],
 [AC_REQUIRE([AC_PROG_CC])dnl
 AC_MSG_CHECKING(if the C compiler accepts -static)
 AC_CACHE_VAL(ac_cv_prog_cc_static,
@@ -126,7 +126,7 @@
 ])
 
 dnl AC_PROG_CC_DEBUG
-AC_DEFUN(AC_PROG_CC_DEBUG,
+AC_DEFUN([AC_PROG_CC_DEBUG],
 [AC_REQUIRE([AC_PROG_CC])dnl
 AC_MSG_CHECKING(if the C compiler accepts -g)
 AC_CACHE_VAL(ac_cv_prog_cc_debug,
@@ -143,7 +143,7 @@
 ])
 
 dnl AC_TYPE_FD_SET
-AC_DEFUN(AC_TYPE_FD_SET,
+AC_DEFUN([AC_TYPE_FD_SET],
 [AC_MSG_CHECKING(for fd_set)
  AC_CACHE_VAL(ac_cv_type_fd_set,
  [ AC_HEADER_INCLUDE(tmp_inc_select, sys/select.h select.h)
@@ -160,7 +160,7 @@
 ])
 
 dnl AC_C_EXTERN_TYPEDEF_FUNC
-AC_DEFUN(AC_C_EXTERN_TYPEDEF_FUNC,
+AC_DEFUN([AC_C_EXTERN_TYPEDEF_FUNC],
 [AC_MSG_CHECKING(if an extern declaration of a typedefed function may precede its definition)
  AC_CACHE_VAL(ac_cv_c_extern_typedef_func,
  [AC_TRY_COMPILE([
@@ -175,7 +175,7 @@
 ])
 
 dnl AC_C_PTR_UNDEF_STRUCT
-AC_DEFUN(AC_C_PTR_UNDEF_STRUCT,
+AC_DEFUN([AC_C_PTR_UNDEF_STRUCT],
 [AC_MSG_CHECKING(if pointers to undefined structures are allowed)
  AC_CACHE_VAL(ac_cv_c_ptr_undef_struct,
  [AC_TRY_COMPILE([typedef struct _foo foo; extern foo *bar();], bar();,
@@ -188,7 +188,7 @@
 
 dnl AC_PROG_TELNET_L
 dnl Ok, it's a hack of a method.  Got anything better?
-AC_DEFUN(AC_PROG_TELNET_L,
+AC_DEFUN([AC_PROG_TELNET_L],
 [ AC_MSG_CHECKING(if telnet accepts the -l flag)
   AC_CACHE_VAL(ac_cv_prog_telnet_l,
   [ if test -r `which telnet` \
@@ -205,7 +205,7 @@
 
 dnl AC_FILE_RESOLV_CONF 
 dnl Yeah, it should be very general.  So sue me.
-AC_DEFUN(AC_FILE_RESOLV_CONF,
+AC_DEFUN([AC_FILE_RESOLV_CONF],
 [ AC_MSG_CHECKING(for resolver configuration file)
   AC_CACHE_VAL(ac_cv_file_resolv_conf,
   [ if test -f /etc/resolv.conf; then
@@ -218,7 +218,7 @@
 ])
 
 dnl AC_CHECK_EXTERN(EXTERN, ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND])
-AC_DEFUN(AC_CHECK_EXTERN,
+AC_DEFUN([AC_CHECK_EXTERN],
 [ AC_MSG_CHECKING([for $1])
   AC_CACHE_VAL(ac_cv_extern_$1,
   [ AC_TRY_LINK(ifelse(AC_LANG, CPLUSPLUS, [#ifdef __cplusplus
@@ -240,7 +240,7 @@
 ])
 
 dnl AC_CHECK_EXTERNS(EXTERN... [, ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]])
-AC_DEFUN(AC_CHECK_EXTERNS,
+AC_DEFUN([AC_CHECK_EXTERNS],
 [for ac_exter in $1
 do
 AC_CHECK_EXTERN($ac_exter,
@@ -252,7 +252,7 @@
 ])
 
 dnl AC_C_VOLATILE
-AC_DEFUN(AC_C_VOLATILE,
+AC_DEFUN([AC_C_VOLATILE],
 [ AC_MSG_CHECKING([for working volatile])
   AC_CACHE_VAL(ac_cv_c_volatile,
   [ AC_TRY_COMPILE(,[
@@ -265,7 +265,7 @@
 ])
 
 dnl AC_C_NOSHARE
-AC_DEFUN(AC_C_NOSHARE,
+AC_DEFUN([AC_C_NOSHARE],
 [ AC_MSG_CHECKING([for working noshare])
   AC_CACHE_VAL(ac_cv_c_noshare,
   [ AC_TRY_COMPILE(,[
@@ -278,7 +278,7 @@
 ])
 
 dnl AC_HEADER_INCLUDE(VARIABLE, INCLUDES)
-AC_DEFUN(AC_HEADER_INCLUDE,
+AC_DEFUN([AC_HEADER_INCLUDE],
 [ $1=""
   for ac_hdr in $2; do
     AC_TRY_CPP([#include <$ac_hdr>], [ $1="[#]include <$ac_hdr>"
@@ -309,7 +309,7 @@
 dnl strictly my (<roeber@cern.ch>) fault.
 dnl
 
-AC_DEFUN(AC_WEIRD_ACCEPT, 
+AC_DEFUN([AC_WEIRD_ACCEPT], 
 [ AC_MSG_CHECKING(for unusual accept call)
   AC_CACHE_VAL(ac_cv_weird_accept,
   [ AC_HEADER_INCLUDE(tmp_inc_types,  sys/types.h types.h)
@@ -404,7 +404,7 @@
 ])
 
 dnl AC_HEADER_TIOCGWINSZ
-AC_DEFUN(AC_HEADER_TIOCGWINSZ,
+AC_DEFUN([AC_HEADER_TIOCGWINSZ],
 [ AC_MSG_CHECKING([for TIOCGWINSZ get-window-size capability])
   AC_CACHE_VAL(ac_cv_header_tiocgwinsz,
   [ AC_HEADER_INCLUDE(tmp_inc_termios, sys/termios.h termios.h)
@@ -422,7 +422,7 @@
 
 dnl AC_SYS_NEED_SIGNAL_RESET
 dnl If signal handlers need to be reset, define NEED_SIGNAL_RESET
-AC_DEFUN(AC_SYS_NEED_SIGNAL_RESET, 
+AC_DEFUN([AC_SYS_NEED_SIGNAL_RESET], 
 [ AC_MSG_CHECKING(if signal handlers need to be reset)
   AC_CACHE_VAL(ac_cv_sys_need_signal_reset,
   [ AC_TRY_RUN(
@@ -440,7 +440,7 @@
 
 dnl AC_TYPE_STATUS_T
 dnl Define status_t to be whatever wait/wait3/waitpid want.
-AC_DEFUN(AC_TYPE_STATUS_T,
+AC_DEFUN([AC_TYPE_STATUS_T],
 [ AC_MSG_CHECKING(for what the wait calls want)
   AC_CACHE_VAL(ac_cv_type_status_t,
   [ AC_TRY_COMPILE([#include <sys/types.h>
@@ -456,7 +456,7 @@
 dnl  if timezone and/or altzone exist, define HAVE_TIMEZONE and/or
 dnl  HAVE_ALTZONE respectively.  Finally, if the timezone value is
 dnl  backwards, define NEGATIVE_TIMEZONE.
-AC_DEFUN(AC_STRUCT_GMTOFF,
+AC_DEFUN([AC_STRUCT_GMTOFF],
 [ AC_REQUIRE([AC_STRUCT_TM])dnl
   AC_MSG_CHECKING([for tm_gmtoff in struct tm])
   AC_CACHE_VAL(ac_cv_struct_gmtoff,
@@ -617,7 +617,7 @@
 ])
 
 dnl AC_STRUCT_DIRENT_INO
-AC_DEFUN(AC_STRUCT_DIRENT_INO, 
+AC_DEFUN([AC_STRUCT_DIRENT_INO], 
 [ AC_REQUIRE([AC_HEADER_DIRENT])
   AC_MSG_CHECKING(if directory entries have inodes)
   AC_CACHE_VAL(ac_cv_struct_dirent_ino,
@@ -644,7 +644,7 @@
 ])
 
 dnl AC_STRUCT_WINSIZE
-AC_DEFUN(AC_STRUCT_WINSIZE, 
+AC_DEFUN([AC_STRUCT_WINSIZE], 
 [ AC_CACHE_CHECK(for winsize structure, ac_cv_struct_winsize,
   [ AC_TRY_COMPILE([
 #ifdef HAVE_TERMIOS_H
@@ -661,7 +661,7 @@
 ])
 
 dnl AC_CHECK_SIZEOF_TYPE_T
-AC_DEFUN(AC_CHECK_SIZEOF_TIME_T,
+AC_DEFUN([AC_CHECK_SIZEOF_TIME_T],
 [AC_CACHE_CHECK(for whether time_t is long, ac_cv_c_long_time_t,
 [AC_TRY_RUN([
 #include <time.h>
@@ -674,7 +674,7 @@
 ])
 
 dnl AC_CHECK_SIZEOF_SIZE_T
-AC_DEFUN(AC_CHECK_SIZEOF_SIZE_T,
+AC_DEFUN([AC_CHECK_SIZEOF_SIZE_T],
 [AC_CACHE_CHECK(for whether size_t is long, ac_cv_c_long_size_t,
 [AC_TRY_RUN([
 #include <stddef.h>
@@ -687,7 +687,7 @@
 ])
 
 dnl AC_ADDLIB
-AC_DEFUN(AC_ADDLIB,
+AC_DEFUN([AC_ADDLIB],
 [ AC_MSG_CHECKING([for $1])
   LP=`dirname $1`
   LN=`basename $1 | sed 's/lib\(.*\)\.a/\1/g'`
@@ -700,7 +700,7 @@
 dnl but don't declare them.  SunOS, for instance.
 dnl This isn't a perfect test, but we can combine it with HAVE_STRERROR,
 dnl HAVE_SYS_ERRLIST and HAVE_SYS_NERR.
-AC_DEFUN(AC_DECL_NEED_SYS_ERR,
+AC_DEFUN([AC_DECL_NEED_SYS_ERR],
 [ AC_MSG_CHECKING(if sys_errlist must be declared)
   AC_CACHE_VAL(ac_cv_decl_need_sys_errlist, 
   [ AC_HEADER_INCLUDE(tmp_inc_errno, errno.h sys/errno.h)
--- w3c-libwww-5.3.2/configure.in~	Sat Mar  2 06:46:58 2002
+++ w3c-libwww-5.3.2/configure.in	Sat Mar  2 06:46:58 2002
@@ -986,7 +986,8 @@
     ;;
   *) 
     AC_MSG_RESULT(yes)
-    AC_DEFINE(_POSIX_SOURCE) [CFLAGS="$CFLAGS -D_POSIX_SOURCE"]
+    AC_DEFINE(_POSIX_SOURCE,,[Define if you need to in order for stat and other things to work.])
+    [CFLAGS="$CFLAGS -D_POSIX_SOURCE"]
     ;;
   esac ],
   AC_MSG_RESULT(no)
