--- w3c-libwww-5.3.2/modules/Makefile.am.orig	Mon Feb 22 00:56:18 1999
+++ w3c-libwww-5.3.2/modules/Makefile.am	Sat Jan 25 13:51:39 2003
@@ -5,5 +5,5 @@
 EXTRA_DIST = \
 	$(DOCS)
 
-SUBDIRS = expat md5
+SUBDIRS = md5
 
--- w3c-libwww-5.3.2/ComLine/src/Makefile.am.orig	Fri Aug  4 11:48:37 2000
+++ w3c-libwww-5.3.2/ComLine/src/Makefile.am	Sat Jan 25 14:19:18 2003
@@ -25,8 +25,7 @@
 SUBDIRS = windows
 
 INCLUDES = \
-	-I$(srcdir)/../../Library/src \
-	-I$(top_srcdir)/modules/expat/xmlparse
+	-I$(srcdir)/../../Library/src
 
 #	.h files are distributed but originally are made from the
 #	self-documenting hypertext files.
@@ -64,7 +63,7 @@
 	-rm purify; \
 	purify -cache-dir=$(PURIFY_CACHE) \
 	$(CC) $(LDFLAGS) -o purify $(w3c_OBJECTS) \
-	-lm \
+	-lm -lexpat \
 	../../Library/src/.libs/libwwwinit.a \
 	../../Library/src/.libs/libwwwapp.a \
 	../../Library/src/.libs/libwwwxml.a \
@@ -85,7 +84,5 @@
 	../../Library/src/.libs/libwwwutils.a \
 	../../Library/src/.libs/libwwwzip.a \
 	../../modules/md5/.libs/libmd5.a \
-	../../modules/expat/xmlparse/.libs/libxmlparse.a \
-	../../modules/expat/xmltok/.libs/libxmltok.a \
 	$(LIBS)
 
--- w3c-libwww-5.3.2/Robot/src/Makefile.am.orig	Fri Aug  4 11:48:37 2000
+++ w3c-libwww-5.3.2/Robot/src/Makefile.am	Sat Jan 25 14:20:08 2003
@@ -25,8 +25,7 @@
 SUBDIRS = windows
 
 INCLUDES = \
-	-I$(srcdir)/../../Library/src \
-	-I$(top_srcdir)/modules/expat/xmlparse
+	-I$(srcdir)/../../Library/src
 
 #	.h files are distributed but originally are made from the
 #	self-documenting hypertext files.
@@ -65,11 +64,9 @@
 	-rm purify; \
 	purify -cache-dir=$(PURIFY_CACHE) \
 	$(CC) $(LDFLAGS) -o purify $(webbot_OBJECTS) \
-	-lm \
+	-lm -lexpat \
 	../../Library/src/.libs/libwww.a \
 	../../modules/md5/.libs/libmd5.a \
-	../../modules/expat/xmlparse/.libs/libxmlparse.a \
-	../../modules/expat/xmltok/.libs/libxmltok.a \
 	$(LIBS)
 
 
--- w3c-libwww-5.3.2/LineMode/src/Makefile.am.orig	Fri Aug  4 11:48:37 2000
+++ w3c-libwww-5.3.2/LineMode/src/Makefile.am	Sat Jan 25 14:21:47 2003
@@ -27,7 +27,6 @@
 
 INCLUDES = \
 	-I$(srcdir)/../../Library/src \
-	-I$(top_srcdir)/modules/expat/xmlparse \
 	-I$(srcdir)/../../PICS-client/src
 
 #	.h files are distributed but originally are made from the
--- w3c-libwww-5.3.2/Library/src/Makefile.am.orig	Sat Jan 25 13:50:01 2003
+++ w3c-libwww-5.3.2/Library/src/Makefile.am	Sat Jan 25 14:23:51 2003
@@ -353,12 +353,10 @@
 	HTXML.h \
 	HTXML.c
 
-libwwwxml_la_DEPENDENCIES = \
-	../../modules/expat/xmltok/libxmltok.la \
-	../../modules/expat/xmlparse/libxmlparse.la
-
 libwwwxml_la_LDFLAGS = -rpath $(libdir)
 
+libwwwxml_la_LIBADD = -lexpat
+
 libwwwzip_la_SOURCES = \
 	WWWZip.h \
 	HTZip.h \
@@ -379,8 +377,7 @@
 
 INCLUDES = \
 	-I/usr/local/include \
-	-I${top_srcdir}/modules/md5 \
-	-I${top_srcdir}/modules/expat/xmlparse
+	-I${top_srcdir}/modules/md5
 
 BUILT_SOURCES = \
 	wwwsys.h \
--- w3c-libwww-5.3.2/Library/src/SSL/Makefile.am.orig	Tue Jun 20 13:01:34 2000
+++ w3c-libwww-5.3.2/Library/src/SSL/Makefile.am	Sat Jan 25 14:24:36 2003
@@ -47,7 +47,6 @@
 INCLUDES = \
 	-I/usr/local/include \
 	-I${top_srcdir}/modules/md5 \
-	-I${top_srcdir}/modules/expat/xmlparse \
 	-I$(top_srcdir)/Library/src \
 	@SSLINC@
 
--- w3c-libwww-5.3.2/Library/Examples/Makefile.am.orig	Thu Aug 17 21:36:57 2000
+++ w3c-libwww-5.3.2/Library/Examples/Makefile.am	Sat Jan 25 14:22:21 2003
@@ -31,7 +31,6 @@
 
 INCLUDES = \
 	-I$(srcdir)/../src \
-	-I$(top_srcdir)/modules/expat/xmlparse \
 	-I$(srcdir)/../src/SSL
 
 DOCS :=	$(wildcard *.html)
@@ -45,7 +44,7 @@
 	-rm purify; \
 	purify -cache-dir=$(PURIFY_CACHE) \
 	$(CC) $(LDFLAGS) -o purify $(mget_OBJECTS) \
-	-lm \
+	-lm -lexpat \
 	../src/.libs/libwwwinit.a \
 	../src/.libs/libwwwapp.a \
 	../src/.libs/libwwwxml.a \
@@ -66,8 +65,6 @@
 	../src/.libs/libwwwutils.a \
 	../src/.libs/libwwwzip.a \
 	../../modules/md5/.libs/libmd5.a \
-	../../modules/expat/xmlparse/.libs/libxmlparse.a \
-	../../modules/expat/xmltok/.libs/libxmltok.a \
 	$(LIBS)
 
 
--- w3c-libwww-5.4.0/configure.in.orig	2002-06-12 11:26:46.000000000 +0200
+++ w3c-libwww-5.4.0/configure.in	2003-10-23 00:25:01.459776472 +0200
@@ -497,24 +497,24 @@
     AC_MSG_RESULT(yes)
     AC_DEFINE(HT_EXPAT)
     HTXML="HTXML.lo"
-    HTLIBXML='${top_builddir}/modules/expat/xmlparse/libxmlparse.la ${top_builddir}/modules/expat/xmltok/libxmltok.la'
-    XMLTOK="libxmltok.la"
-    XMLPARSE="libxmlparse.la"
+    HTLIBXML='-lexpat'
+    XMLTOK=""
+    XMLPARSE=""
     WWWXML="libwwwxml.la"
-    LWWWXML="-lwwwxml -lxmltok -lxmlparse"
-    LIBWWWXML='${top_builddir}/Library/src/libwwwxml.la ${top_builddir}/modules/expat/xmlparse/libxmlparse.la ${top_builddir}/modules/expat/xmltok/libxmltok.la'
+    LWWWXML="-lwwwxml -lexpat"
+    LIBWWWXML='${top_builddir}/Library/src/libwwwxml.la -lexpat'
     SHOWXML='showxml$(EXEEXT) ptri$(EXEEXT) stri$(EXEEXT) rdf_parse_file$(EXEEXT) rdf_parse_buffer$(EXEEXT)'
     ;;
   esac ],
   AC_MSG_RESULT(yes)
   AC_DEFINE(HT_EXPAT)
   HTXML="HTXML.lo"
-  HTLIBXML='${top_builddir}/modules/expat/xmlparse/libxmlparse.la ${top_builddir}/modules/expat/xmltok/libxmltok.la'
-  XMLTOK="libxmltok.la"
-  XMLPARSE="libxmlparse.la"
+  HTLIBXML='-lexpat'
+  XMLTOK=""
+  XMLPARSE=""
   WWWXML="libwwwxml.la"
-  LWWWXML="-lwwwxml -lxmltok -lxmlparse"
-  LIBWWWXML='${top_builddir}/Library/src/libwwwxml.la ${top_builddir}/modules/expat/xmlparse/libxmlparse.la ${top_builddir}/modules/expat/xmltok/libxmltok.la'
+  LWWWXML="-lwwwxml -lexpat"
+  LIBWWWXML='${top_builddir}/Library/src/libwwwxml.la -lexpat'
   SHOWXML='showxml$(EXEEXT) ptri$(EXEEXT) stri$(EXEEXT) rdf_parse_file$(EXEEXT) rdf_parse_buffer$(EXEEXT)'
 )
 AC_SUBST(HTXML)		# for old style big library
--- w3c-libwww-5.3.2/Library/src/HTXML.h.orig	Wed Dec 20 12:00:07 2000
+++ w3c-libwww-5.3.2/Library/src/HTXML.h	Sat Jan 25 14:33:46 2003
@@ -19,11 +19,7 @@
 of the  W3C Sample Code Library.
 
 We use James Clark's expat
-XML parser which is very neat indeed. As the code doesn't come as a separate
-library, I included it in the libwww CVS code base where I compile is as
-two libraries: libxmltok.a and libxmlparse.a. See
-the external modules that libwww works with for
-details. Thanks so much to John Punin for writing this code!
+XML parser which is very neat indeed.
 */
 
 #ifndef HTXML_H
@@ -36,7 +36,7 @@
 #include "SGML.h"
 #endif /* HT_STRUCT_XML_STREAM */
 
-#include <xmlparse.h>
+#include <expat.h>
 
 /*
 .
