Index: systemtap-1.4/configure.ac
===================================================================
--- systemtap-1.4.orig/configure.ac
+++ systemtap-1.4/configure.ac
@@ -63,8 +63,8 @@ AC_ARG_ENABLE([ssp],
 AS_IF([test "x$enable_ssp" != xno],[
    save_CFLAGS="$CFLAGS"
    save_CXXFLAGS="$CXXFLAGS"
-   CXXFLAGS="$CXXFLAGS -Werror -fstack-protector-all -D_FORTIFY_SOURCE=2"
-   CFLAGS="$CFLAGS -Werror -fstack-protector-all -D_FORTIFY_SOURCE=2"
+   CXXFLAGS="$CXXFLAGS  -fstack-protector-all -D_FORTIFY_SOURCE=2"
+   CFLAGS="$CFLAGS  -fstack-protector-all -D_FORTIFY_SOURCE=2"
    AC_COMPILE_IFELSE([int something ();], [
        AC_MSG_NOTICE([Compiling with gcc -fstack-protector-all et al.])
 	CFLAGS="$save_CFLAGS -fstack-protector-all -D_FORTIFY_SOURCE=2"
@@ -305,18 +305,20 @@ dnl Find the location of nss and nspr he
   ])
   if test "x$nssdir" != "x" -a "x$nsprdir" != "x"; then
     nss_CFLAGS="-I/usr/include/$nssdir"
+    nss_LDFLAGS="-L/usr/lib/nss"
     nspr_CFLAGS="-I/usr/include/$nsprdir"
     AC_SUBST(nss_CFLAGS)
+    AC_SUBST(nss_LDFLAGS)
     AC_SUBST(nspr_CFLAGS)
     save_CPPFLAGS="$CPPFLAGS"
-    CPPFLAGS="$CFLAGS $nss_CFLAGS $nspr_CFLAGS"
+    CPPFLAGS="$CFLAGS $nss_CFLAGS $nspr_CFLAGS $nss_LDFLAGS"
     have_nss_includes=yes
     AC_CHECK_HEADERS(["$nsprdir/nspr.h" "$nsprdir/plgetopt.h" "$nsprdir/prerror.h" "$nssdir/ssl.h" "$nssdir/nss.h" "$nssdir/pk11func.h" "$nssdir/secerr.h"], [], have_nss_includes=no)
-    CPPFLAGS="$save_CPPFLAGS"
     have_nss_libs=no
     AC_CHECK_LIB(nspr4, PR_Connect, [
       AC_CHECK_LIB(ssl3, SSL_ReHandshake, have_nss_libs=yes)
     ])
+    CPPFLAGS="$save_CPPFLAGS"
   fi
   if test "x${have_nss_includes}${have_nss_libs}${have_certutil}" != "xyesyesyes"; then
     if test "$enable_server" == "yes"; then
Index: systemtap-1.4/Makefile.am
===================================================================
--- systemtap-1.4.orig/Makefile.am
+++ systemtap-1.4/Makefile.am
@@ -13,8 +13,8 @@ AM_CPPFLAGS = -DBINDIR='"$(bindir)"' \
 	      -DPKGLIBDIR='"$(pkglibexecdir)"' \
 	      -I$(srcdir)/includes -I$(builddir)/includes/sys
 
-AM_CFLAGS = -D_GNU_SOURCE -fexceptions -Wall -Werror -Wunused -Wformat=2 -W
-AM_CXXFLAGS = -Wall -Werror
+AM_CFLAGS = -D_GNU_SOURCE -fexceptions -Wall  -Wunused -Wformat=2 -W
+AM_CXXFLAGS = -Wall 
 
 man_MANS = stapprobes.3stap stapfuncs.3stap stapvars.3stap stapex.3stap \
 dtrace.1 stap-merge.1\
@@ -118,7 +118,7 @@ stap_SOURCES += stap-client-connect.c ns
 stap_CFLAGS += $(nss_CFLAGS) $(nspr_CFLAGS) -DSTAP
 stap_CXXFLAGS += $(nss_CFLAGS) $(nspr_CFLAGS)
 stap_CPPFLAGS += $(nss_CFLAGS) $(nspr_CFLAGS)
-stap_LDADD += -lssl3 -lnss3 -lnspr4 -lplc4
+stap_LDADD += $(nss_LDFLAGS) -lssl3 -lnss3 -lnspr4 -lplc4
 stap_client_connect_LDFLAGS = $(AM_LDFLAGS)
 stap_server_connect_LDFLAGS = $(AM_LDFLAGS)
 stap_sign_module_LDFLAGS = $(AM_LDFLAGS)
@@ -170,17 +170,17 @@ CLEANFILES +=  $(pkglibexec_PROGRAMS)
 if BUILD_TRANSLATOR
 if HAVE_NSS
 stap_sign_module_SOURCES = modsign.cxx nsscommon.c
-stap_sign_module_CPPFLAGS = -Wall -Werror $(AM_CPPFLAGS) $(nss_CFLAGS) $(nspr_CFLAGS)
-stap_sign_module_LDADD = -lnss3 -lnspr4
+stap_sign_module_CPPFLAGS = -Wall  $(AM_CPPFLAGS) $(nss_CFLAGS) $(nspr_CFLAGS)
+stap_sign_module_LDADD = $(nss_LDFLAGS) -lnss3 -lnspr4
 
 if BUILD_SERVER
 stap_client_connect_SOURCES = stap-client-connect.c nsscommon.c
-stap_client_connect_CFLAGS = -Wall -Werror $(nss_CFLAGS) $(nspr_CFLAGS)
-stap_client_connect_LDADD = -lssl3 -lnss3 -lnspr4 -lplc4
+stap_client_connect_CFLAGS = -Wall  $(nss_CFLAGS) $(nspr_CFLAGS)
+stap_client_connect_LDADD = $(nss_LDFLAGS) -lssl3 -lnss3 -lnspr4 -lplc4
 
 stap_server_connect_SOURCES = stap-server-connect.c nsscommon.c
-stap_server_connect_CFLAGS = -Wall -Werror $(nss_CFLAGS) $(nspr_CFLAGS)
-stap_server_connect_LDADD = -lssl3 -lnss3 -lnspr4 -lplc4
+stap_server_connect_CFLAGS = -Wall  $(nss_CFLAGS) $(nspr_CFLAGS)
+stap_server_connect_LDADD = $(nss_LDFLAGS) -lssl3 -lnss3 -lnspr4 -lplc4
 endif
 endif
 
