Index: systemtap-1.0/stap-report
===================================================================
--- systemtap-1.0.orig/stap-report
+++ systemtap-1.0/stap-report
@@ -13,9 +13,5 @@ run "gcc -v"
 run "uname -a"
 run "dmesg | egrep 'stap|systemtap' | tail -n 10"
 run "cat /proc/cpuinfo | egrep 'processor|vendor_id|model name'"
-if [ -x /usr/bin/dpkg ]; then
-  run "dpkg --list | egrep 'systemtap|elfutils|kernel|gcc' | awk '{print \$2,\$3}' | sort"
-else
-  run "rpm -qa --qf '%{name}-%{version} %{release}.%{arch}\\n' | egrep 'systemtap|elfutils|kernel|gcc' | sort"
-fi
+run "ls /var/lib/pisi/package | egrep 'systemtap|elfutils|kernel|gcc' | sort"
 run "egrep 'PROBE|TRACE|MARKER|_DEBUG_' /lib/modules/`uname -r`/build/.config | grep -v not.set | sort | fmt -w 80"
