diff -ur libgnome-2.28.0.orig/schemas/desktop_gnome_interface.schemas.in.in libgnome-2.28.0/schemas/desktop_gnome_interface.schemas.in.in
--- libgnome-2.28.0.orig/schemas/desktop_gnome_interface.schemas.in.in	2009-08-09 12:48:44.000000000 +0300
+++ libgnome-2.28.0/schemas/desktop_gnome_interface.schemas.in.in	2010-02-06 21:34:28.786917666 +0200
@@ -65,7 +65,7 @@
       <applyto>/desktop/gnome/interface/menus_have_icons</applyto>
       <owner>gnome</owner>
       <type>bool</type>
-      <default>false</default>
+      <default>true</default>
       <locale name="C">
         <short>Menus Have Icons</short>
         <long>Whether menus may display an icon next to a menu entry.</long>
