Index: libopensync-plugin-kdepim-0.22/src/kaddrbook.cpp
===================================================================
--- libopensync-plugin-kdepim-0.22.orig/src/kaddrbook.cpp	2009-08-04 19:36:17.000000000 +0600
+++ libopensync-plugin-kdepim-0.22/src/kaddrbook.cpp	2009-08-04 19:36:25.000000000 +0600
@@ -153,9 +153,9 @@
 
 	// Convert the VCARD data into a string
 	// only vcard3.0 exports Categories
-	QString tmp = converter->createVCard(*a, VCardConverter::v3_0);
+	QByteArray tmp = converter->createVCard(*a, VCardConverter::v3_0);
 
-	char *data = strdup((const char *)tmp.toUtf8());
+	char *data = strdup(tmp.constData());
 
 	osync_trace(TRACE_SENSITIVE,"\n%s", data);
 
