Index: libopensync-plugin-kdepim-0.22/m4/kde.m4
===================================================================
--- libopensync-plugin-kdepim-0.22.orig/m4/kde.m4
+++ libopensync-plugin-kdepim-0.22/m4/kde.m4
@@ -3071,11 +3071,6 @@ AC_DEFUN([AC_CHECK_COMPILERS],
     CXXFLAGS="-Wcast-qual -Wshadow -Wcast-align $CXXFLAGS"
   fi
 
-  AC_ARG_ENABLE(pch,
-     AC_HELP_STRING([--enable-pch],
-                    [enables precompiled header support (currently only KCC or gcc >=3.4+unsermake) [default=no]]),
-    [ kde_use_pch=$enableval ],[ kde_use_pch=no ])
- 
   HAVE_GCC_VISIBILITY=0
   AC_SUBST([HAVE_GCC_VISIBILITY])
 
@@ -3108,7 +3103,6 @@ AC_DEFUN([AC_CHECK_COMPILERS],
         fi
         rm -f conftest.h conftest.h.gch
     fi
-    AM_CONDITIONAL(unsermake_enable_pch, test "$kde_use_pch" = "yes" && test "$kde_gcc_supports_pch" = "yes")
   fi
   if test "$CXX" = "KCC"; then
     dnl unfortunately we currently cannot disable exception support in KCC
