diff -Naur ktorrent-3.2.2-orig/ktorrent/ktorrent.desktop ktorrent-3.2.2/ktorrent/ktorrent.desktop
--- ktorrent-3.2.2-orig/ktorrent/ktorrent.desktop	2009-07-16 11:17:47.268785035 +0300
+++ ktorrent-3.2.2/ktorrent/ktorrent.desktop	2009-07-16 11:18:32.112658861 +0300
@@ -50,6 +50,7 @@
 Type=Application
 DocPath=ktorrent/index.html
 MimeType=application/x-bittorrent;application/x-torrent;
+InitialPreference=15
 X-DBUS-StartupType=Unique
 X-DBUS-ServiceName=org.ktorrent.ktorrent
 Comment=A BitTorrent program for KDE
