Index: slv2-0.6.6/wscript
===================================================================
--- slv2-0.6.6.orig/wscript
+++ slv2-0.6.6/wscript
@@ -1,6 +1,7 @@
 #!/usr/bin/env python
 import autowaf
 import Options
+import os
 
 # Version of this package (even if built as a child)
 SLV2_VERSION = '0.6.6'
@@ -52,6 +53,7 @@ def configure(conf):
 	autowaf.check_pkg(conf, 'redland', uselib_store='REDLAND', atleast_version='1.0.6', mandatory=True)
 	autowaf.check_pkg(conf, 'jack', uselib_store='JACK', atleast_version='0.107.0', mandatory=False)
 	conf.env.append_value('CCFLAGS', '-std=c99')
+	conf.env.append_value('SHLIB_MARKER', os.environ['LDFLAGS'].split())
 	conf.define('SLV2_VERSION', SLV2_VERSION)
 	conf.write_config_header('slv2-config.h')
 
