--- megaglest.desktop.orig	2011-05-09 00:00:39.000000000 +0300
+++ megaglest.desktop	2011-05-11 14:34:42.158000013 +0300
@@ -1,8 +1,13 @@
 [Desktop Entry]
-Categories=Game;StrategyGame;
-Comment=A real time strategy game.
+Type=Application
+Version=1.0
 Name=MegaGlest
+GenericName=Strategy Game
+GenericName[tr]=Strateji Oyunu
+Comment=A real time strategy game
+Comment[tr]=Gerçek zamanlı bir strateji oyunu
 Icon=megaglest
 Exec=megaglest
 Terminal=false
-Type=Application
+StartupNotify=false
+Categories=Application;Game;StrategyGame;
