From: Julian Ospald <hasufell@gentoo.org>
Date: Sat Aug 18 13:41:33 UTC 2012
Subject: build system

fix static build

--- source/shared_lib/CMakeLists.txt
+++ source/shared_lib/CMakeLists.txt
@@ -273,21 +273,9 @@
                         streflop/softfloat)
 	ENDIF()
 
-	IF(NOT WANT_STATIC_LIBS)
-		MESSAGE(STATUS "*** Searching for miniupnpc since WANT_STATIC_LIBS is off ...")
-		FIND_PACKAGE(Miniupnpc)
-	ENDIF()
-
-	if(NOT MINIUPNP_FOUND)
-		MESSAGE(STATUS "*** Using EMBEDDED miniupnpc since dev system does not have it... MINIUPNP_FOUND [${MINIUPNP_FOUND}]")
-
-		SET(DIRS_WITH_SRC ${DIRS_WITH_SRC} platform/miniupnpc)
-	else()
-		MESSAGE(STATUS "*** Using SHARED miniupnpc found in [${MINIUPNP_INCLUDE_DIR}] MINIUPNP_FOUND [${MINIUPNP_FOUND}] MINIUPNP_LIBRARY [${MINIUPNP_LIBRARY}]")
-
-		INCLUDE_DIRECTORIES(${MINIUPNP_INCLUDE_DIR})
-		SET(EXTERNAL_LIBS ${EXTERNAL_LIBS} ${MINIUPNP_LIBRARY})
-	endif()
+	FIND_PACKAGE(Miniupnpc REQUIRED)
+	INCLUDE_DIRECTORIES(${MINIUPNP_INCLUDE_DIR})
+	SET(EXTERNAL_LIBS ${EXTERNAL_LIBS} ${MINIUPNP_LIBRARY})
 
 	IF(MINIUPNPC_VERSION_PRE1_5)
 	        ADD_DEFINITIONS(-DMINIUPNPC_VERSION_PRE1_5)
@@ -322,22 +310,9 @@
 	INCLUDE_DIRECTORIES( ${MG_INCLUDES_ROOT}platform/sdl )
         INCLUDE_DIRECTORIES( ${MG_INCLUDES_ROOT}xml/rapidxml )
 
-	IF(NOT WANT_STATIC_LIBS)
-		MESSAGE(STATUS "*** Searching for libircclient since WANT_STATIC_LIBS is off ...")
-		FIND_PACKAGE(Ircclient)
-	ENDIF()
-
-	if(NOT IRCCLIENT_FOUND)
-		MESSAGE(STATUS "*** Using EMBEDDED libircclient since this system does not have it... IRCCLIENT_FOUND [${IRCCLIENT_FOUND}]")
-
-		SET(MG_SOURCE_FILES ${MG_SOURCE_FILES} ${PROJECT_SOURCE_DIR}/source/shared_lib/sources/libircclient/src/libircclient.c)
-		INCLUDE_DIRECTORIES(${PROJECT_SOURCE_DIR}/source/shared_lib/include/libircclient/include)
-	else()
-		MESSAGE(STATUS "*** Using SHARED libircclient found in [${IRCCLIENT_INCLUDE_DIR}] IRCCLIENT_FOUND [${IRCCLIENT_FOUND}] IRCCLIENT_LIBRARY [${IRCCLIENT_LIBRARY}]")
-
-		INCLUDE_DIRECTORIES(${IRCCLIENT_INCLUDE_DIR})
-		SET(EXTERNAL_LIBS ${EXTERNAL_LIBS} ${IRCCLIENT_LIBRARY})
-	endif()
+	FIND_PACKAGE(Ircclient REQUIRED)
+	INCLUDE_DIRECTORIES(${IRCCLIENT_INCLUDE_DIR})
+	SET(EXTERNAL_LIBS ${EXTERNAL_LIBS} ${IRCCLIENT_LIBRARY})
 
         set(CMAKE_REQUIRED_INCLUDES "${IRCCLIENT_INCLUDE_DIR}")
         set(CMAKE_REQUIRED_LIBRARIES "${IRCCLIENT_LIBRARY}")
--- mk/cmake/Modules/FindMiniupnpc.cmake
+++ mk/cmake/Modules/FindMiniupnpc.cmake
@@ -35,7 +35,11 @@
 
 find_path(MINIUPNP_INCLUDE_DIR miniupnpc.h
    PATH_SUFFIXES miniupnpc)
-find_library(MINIUPNP_LIBRARY miniupnpc)
+IF(WANT_STATIC_LIBS)
+  find_library(MINIUPNP_LIBRARY libminiupnpc.a)
+ELSE(WANT_STATIC_LIBS)
+  find_library(MINIUPNP_LIBRARY libminiupnpc.so)
+ENDIF(WANT_STATIC_LIBS)
 
 if (MINIUPNP_INCLUDE_DIR AND MINIUPNP_LIBRARY)
     set (MINIUPNP_FOUND TRUE)
--- mk/cmake/Modules/FindOGG.cmake
+++ mk/cmake/Modules/FindOGG.cmake
@@ -19,12 +19,12 @@
 
 FIND_PATH(OGG_INCLUDE_DIR ogg/ogg.h)
 
-#IF (OGG_STATIC AND NOT OGG_LIBRARY)
-#	FIND_LIBRARY(OGG_LIBRARY NAMES libogg.a ogg)
-#ELSE()
-	FIND_LIBRARY(OGG_LIBRARY NAMES ogg)
-#ENDIF()
-
+IF (OGG_STATIC)
+	FIND_LIBRARY(OGG_LIBRARY NAMES libogg.a)
+ELSE()
+	FIND_LIBRARY(OGG_LIBRARY NAMES libogg.so)
+ENDIF()
+ 
 #IF (OGG_STATIC AND NOT VORBIS_LIBRARY)
 #	FIND_LIBRARY(VORBIS_LIBRARY NAMES libvorbis.a vorbis)
 #ELSE()
--- mk/cmake/Modules/FindLUAJIT.cmake
+++ mk/cmake/Modules/FindLUAJIT.cmake
@@ -18,13 +18,13 @@
         NAMES luajit.h
         PATH_SUFFIXES luajit-2.0)
 
-if (LUAJIT_FIND_STATIC)
+#if (LUAJIT_FIND_STATIC)
         find_library(LUAJIT_LIBRARIES luajit.a
                 NAMES libluajit-5.1.a)
-else (LUATJIT_FIND_STATIC)
-        find_library(LUAJIT_LIBRARIES luajit
-                NAMES luajit-5.1)
-endif (LUAJIT_FIND_STATIC)
+#else (LUATJIT_FIND_STATIC)
+#        find_library(LUAJIT_LIBRARIES luajit.so
+#                NAMES libluajit-5.1.so)
+#endif (LUAJIT_FIND_STATIC)
 
 mark_as_advanced(LUAJIT_INCLUDE_DIR)
 mark_as_advanced(LUAJIT_LIBRARIES)
