From: Julian Ospald <hasufell@gentoo.org>
Date: Fri Aug 31 23:45:34 UTC 2012
Subject: fix as-needed

reorder linker line (curl after libircclient)

--- source/shared_lib/CMakeLists.txt
+++ source/shared_lib/CMakeLists.txt
@@ -47,9 +47,7 @@
 
 	FIND_PACKAGE(CURL REQUIRED)
 	INCLUDE_DIRECTORIES(${CURL_INCLUDE_DIRS})
-	IF(UNIX)
-		SET(EXTERNAL_LIBS ${EXTERNAL_LIBS} ${CURL_LIBRARIES})
-	ENDIF()
+
 	# check libcurl version #
 	IF(WIN32)
 		file (READ ${PROJECT_SOURCE_DIR}${CURL_INCLUDE_DIR}/curl/curlver.h CURL_VERSION_H_CONTENTS)
@@ -314,6 +312,10 @@
 	INCLUDE_DIRECTORIES(${IRCCLIENT_INCLUDE_DIR})
 	SET(EXTERNAL_LIBS ${EXTERNAL_LIBS} ${IRCCLIENT_LIBRARY})
 
+	IF(UNIX)
+		SET(EXTERNAL_LIBS ${EXTERNAL_LIBS} ${CURL_LIBRARIES})
+	ENDIF()
+
         set(CMAKE_REQUIRED_INCLUDES "${IRCCLIENT_INCLUDE_DIR}")
         set(CMAKE_REQUIRED_LIBRARIES "${IRCCLIENT_LIBRARY}")
         include(CheckCXXSourceRuns)
