--- krusader-2.2.0-beta1/krusader/krservices.cpp.orig	2010-05-26 10:57:08.325576562 +0200
+++ krusader-2.2.0-beta1/krusader/krservices.cpp	2010-05-26 10:58:52.528576196 +0200
@@ -26,7 +26,7 @@
 #include <QtCore/QTextStream>
 
 #include <kdebug.h>
-
+#include <kstandarddirs.h>
 #include "krglobal.h"
 #include "defaults.h"
 
@@ -41,30 +41,9 @@ bool KrServices::cmdExist(QString cmdNam
     return !detectFullPathName(cmdName).isEmpty();
 }
 
-static const QStringList bin_suffixes = QStringList()
-#ifdef Q_WS_WIN
-                                        << ".cmd" << ".exe" << ".bat"
-#else
-                                        << ""
-#endif
-                                        ;
-
 QString KrServices::detectFullPathName(QString name)
 {
-    QStringList path = QString::fromLocal8Bit(qgetenv("PATH")).split(':');
-
-    for (QStringList::Iterator it = path.begin(); it != path.end(); ++it)
-        foreach(const QString &suffix, bin_suffixes) {
-        if (QDir(*it).exists(name + suffix)) {
-            QString dir = *it;
-            if (!dir.endsWith('/'))
-                dir += '/';
-
-            return dir + name;
-        }
-    }
-
-    return "";
+    return KStandardDirs::findExe( name );
 }
 
 QString KrServices::fullPathName(QString name, QString confName)
