--- xf86-video-mga-1.4.5/src/mga_driver.c.jx	2006-11-08 12:16:44.000000000 -0500
+++ xf86-video-mga-1.4.5/src/mga_driver.c	2006-12-04 16:27:38.000000000 -0500
@@ -1329,8 +1329,7 @@
 	    pMga->DualHeadEnabled = TRUE;
 	} else if (xf86IsPrimInitDone(pScrn->entityList[0])) {
 	    xf86DrvMsg(pScrn->scrnIndex, X_ERROR,
-	 "This card requires the \"mga_hal\" module for dual-head operation\n"
-	 "\tIt can be found at the Matrox web site <http://www.matrox.com>\n");
+	    "No open-source support for dualhead on this card, sorry\n");
 	}
     }
 
