diff -Nur egoboo-2.8.1-old//src/game/network.c egoboo-2.8.1/src/game/network.c
--- egoboo-2.8.1-old//src/game/network.c	2011-01-18 16:27:43.796000312 +0200
+++ egoboo-2.8.1/src/game/network.c	2011-01-18 16:27:36.628000167 +0200
@@ -1845,7 +1845,7 @@
         log_info( "cl_joinGame: Creating client network connection... " );
         // Create my host thingamabober
         /// @todo Should I limit client bandwidth here?
-        net_myHost = enet_host_create( NULL, 1, 0, 0 );
+        net_myHost = enet_host_create( NULL, 1, 0, 0, 0 );
         if ( NULL == net_myHost )
         {
             // can't create a network connection at all
@@ -1861,7 +1861,7 @@
         log_info( "cl_joinGame: Attempting to connect to %s:%d\n", hostname, NET_EGOBOO_PORT );
         enet_address_set_host( &address, hostname );
         address.port = NET_EGOBOO_PORT;
-        net_gameHost = enet_host_connect( net_myHost, &address, NET_EGOBOO_NUM_CHANNELS );
+        net_gameHost = enet_host_connect( net_myHost, &address, NET_EGOBOO_NUM_CHANNELS, 0 );
         if ( NULL == net_gameHost )
         {
             log_info( "cl_joinGame: No available peers to create a connection!\n" );
@@ -1904,7 +1904,7 @@
         address.port = NET_EGOBOO_PORT;
 
         log_info( "sv_hostGame: Creating game on port %d\n", NET_EGOBOO_PORT );
-        net_myHost = enet_host_create( &address, MAX_PLAYER, 0, 0 );
+        net_myHost = enet_host_create( &address, MAX_PLAYER, 0, 0, 0 );
         if ( NULL == net_myHost )
         {
             log_info( "sv_hostGame: Could not create network connection!\n" );
