diff -Nur egoboo-2.7.5-old/game/char.c egoboo-2.7.5/game/char.c
--- egoboo-2.7.5-old/game/char.c	2007-12-28 00:30:05.827575090 +0200
+++ egoboo-2.7.5/game/char.c	2007-12-28 00:30:10.325891241 +0200
@@ -19,7 +19,7 @@
 */
 
 #include "mathstuff.h"
-#include "network.h"
+#include "Network.h"
 #include "Client.h"
 #include "Server.h"
 #include "Log.h"
diff -Nur egoboo-2.7.5-old/game/Client.h egoboo-2.7.5/game/Client.h
--- egoboo-2.7.5-old/game/Client.h	2007-12-28 00:30:05.827575090 +0200
+++ egoboo-2.7.5/game/Client.h	2007-12-28 00:30:10.328890785 +0200
@@ -24,7 +24,7 @@
 #define egoboo_Client_h
 
 #include "enet/enet.h"
-#include "network.h"
+#include "Network.h"
 #include "egobootypedef.h"
 #include "egoboo.h"
 
diff -Nur egoboo-2.7.5-old/game/Clock.c egoboo-2.7.5/game/Clock.c
--- egoboo-2.7.5-old/game/Clock.c	2007-12-28 00:30:05.830574633 +0200
+++ egoboo-2.7.5/game/Clock.c	2007-12-28 00:30:10.329890633 +0200
@@ -30,7 +30,7 @@
 #include <string.h> // memcpy & memset
 
 // Clock data
-typedef struct clock_state_t
+struct clock_state_t
 {
   // Clock data
   double(*timeSource)();   // Function that the clock get it's time values from
@@ -47,7 +47,7 @@
   int frameHistorySize;
   int frameHistoryWindow;
   int frameHistoryHead;
-} ClockState;
+};
 
 static void clock_state_init(ClockState * cs)
 {
diff -Nur egoboo-2.7.5-old/game/common-file.c egoboo-2.7.5/game/common-file.c
--- egoboo-2.7.5-old/game/common-file.c	2007-12-28 00:30:05.827575090 +0200
+++ egoboo-2.7.5/game/common-file.c	2007-12-28 00:30:10.330890480 +0200
@@ -156,4 +156,4 @@
   {
     fclose(pfile);
   };
-};
\ Dosya sonunda yenisatır yok.
+};
diff -Nur egoboo-2.7.5-old/game/egoboo.h egoboo-2.7.5/game/egoboo.h
--- egoboo-2.7.5-old/game/egoboo.h	2007-12-28 00:30:05.828574938 +0200
+++ egoboo-2.7.5/game/egoboo.h	2007-12-28 00:30:10.333890026 +0200
@@ -2308,7 +2308,7 @@
 EXTERN char tagname[MAXTAG][TAGSIZE];               // Scancode names
 EXTERN Uint32 tagvalue[MAXTAG];                     // Scancode values
 
-typedef enum control_t
+enum control_t
 {
   KEY_JUMP = 0,
   KEY_LEFT_USE,
@@ -2351,7 +2351,7 @@
   JOB_RIGHT_PACK,
   JOB_CAMERA,
   MAXCONTROL
-} CONTROL;
+};
 
 EXTERN Uint32 controlvalue[MAXCONTROL];             // The scancode or mask
 EXTERN Uint32 controliskey[MAXCONTROL];             // Is it a key?
diff -Nur egoboo-2.7.5-old/game/enchant.c egoboo-2.7.5/game/enchant.c
--- egoboo-2.7.5-old/game/enchant.c	2007-12-28 00:30:05.827575090 +0200
+++ egoboo-2.7.5/game/enchant.c	2007-12-28 00:30:10.335889721 +0200
@@ -525,7 +525,7 @@
   cnt = 0;
   while (cnt < MAXCHR)
   {
-    if (chrstickybutt[cnt] && chron[cnt])
+    if (chrstickybutt[cnt] && chron[cnt] && chronwhichfan[cnt] != OFFEDGE)
     {
       twist = meshtwist[chronwhichfan[cnt]];
       chrturnmaplr[cnt] = maplrtwist[twist];
diff -Nur egoboo-2.7.5-old/game/Font.c egoboo-2.7.5/game/Font.c
--- egoboo-2.7.5-old/game/Font.c	2007-12-28 00:30:05.827575090 +0200
+++ egoboo-2.7.5/game/Font.c	2007-12-28 00:30:10.335889721 +0200
@@ -29,15 +29,17 @@
 #include <SDL_opengl.h>
 #include <SDL_ttf.h>
 
+#ifndef __unix__
 #define vsnprintf _vsnprintf
+#endif
 
-typedef struct font_t
+struct font_t
 {
   TTF_Font *ttfFont;
 
   GLuint texture;
   GLfloat texCoords[4];
-} Font;
+};
 
 // The next two functions are borrowed from the gl_font.c test program from SDL_ttf
 static int powerOfTwo(int input)
diff -Nur egoboo-2.7.5-old/game/game.c egoboo-2.7.5/game/game.c
--- egoboo-2.7.5-old/game/game.c	2007-12-28 00:30:05.830574633 +0200
+++ egoboo-2.7.5/game/game.c	2007-12-28 00:30:10.338889265 +0200
@@ -2255,6 +2255,21 @@
   }
 }
 
+#if SDL_BYTEORDER != SDL_LIL_ENDIAN
+float LoadFloatByteswapped( float *ptr )
+{
+  union {
+    float f;
+    int i;
+  } u;
+
+  u.f = *ptr;
+  u.i = SDL_Swap32(u.i);
+
+  return u.f;
+}
+#endif
+
 //---------------------------------------------------------------------------------------------
 void rip_md2_commands(Uint16 modelindex)
 {
diff -Nur egoboo-2.7.5-old/game/gltexture.c egoboo-2.7.5/game/gltexture.c
--- egoboo-2.7.5-old/game/gltexture.c	2007-12-28 00:30:05.829574785 +0200
+++ egoboo-2.7.5/game/gltexture.c	2007-12-28 00:30:10.339889113 +0200
@@ -305,4 +305,4 @@
     };
   }
 
-};
\ Dosya sonunda yenisatır yok.
+};
diff -Nur egoboo-2.7.5-old/game/graphicmad.c egoboo-2.7.5/game/graphicmad.c
--- egoboo-2.7.5-old/game/graphicmad.c	2007-12-28 00:30:05.829574785 +0200
+++ egoboo-2.7.5/game/graphicmad.c	2007-12-28 00:30:10.341888810 +0200
@@ -22,9 +22,8 @@
 #include "egoboo.h"
 #include "Md2.h"
 #include "id_md2.h"
-#include "log.h"
+#include "Log.h"
 #include <SDL_opengl.h>
-#include "log.h"
 
 /* Storage for blended vertices */
 static GLfloat md2_blendedVertices[MD2_MAX_VERTICES][3];
@@ -887,4 +886,4 @@
 //  glLoadMatrixf(mView.v);
 //  glMultMatrixf(mWorld.v);
 //}
-//#endif
\ Dosya sonunda yenisatır yok.
+//#endif
diff -Nur egoboo-2.7.5-old/game/Makefile.unix egoboo-2.7.5/game/Makefile.unix
--- egoboo-2.7.5-old/game/Makefile.unix	2007-12-28 00:30:05.829574785 +0200
+++ egoboo-2.7.5/game/Makefile.unix	2007-12-28 00:31:25.493464034 +0200
@@ -16,11 +16,11 @@
 SDLCONF_I := $(shell ${SDL_CONF} --cflags)
 SDLCONF_L := $(shell ${SDL_CONF} --libs)
 
-CC      := gcc
-OPT     := -Os
-INC     := -I. -I../enet/include ${SDLCONF_I}
+CC      := gcc
+OPT     := -Os
+INC     := -I.. -I. ${SDLCONF_I}
 CFLAGS  := ${OPT} ${INC}
-LDFLAGS := ${SDLCONF_L} -lSDL_ttf -lSDL_mixer -lGL -lGLU
+LDFLAGS := ${SDLCONF_L} -lSDL_ttf -lSDL_mixer -lSDL_image -lGL -lGLU
 
 EGO_BIN := egoboo
 
diff -Nur egoboo-2.7.5-old/game/network.c egoboo-2.7.5/game/network.c
--- egoboo-2.7.5-old/game/network.c	2007-12-28 00:30:05.829574785 +0200
+++ egoboo-2.7.5/game/network.c	2007-12-28 00:30:10.343888505 +0200
@@ -20,7 +20,7 @@
     along with Egoboo.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#include "network.h"
+#include "Network.h"
 #include "Client.h"
 #include "Server.h"
 #include "Log.h"
diff -Nur egoboo-2.7.5-old/game/Network.h egoboo-2.7.5/game/Network.h
--- egoboo-2.7.5-old/game/Network.h	2007-12-28 00:30:05.829574785 +0200
+++ egoboo-2.7.5/game/Network.h	2007-12-28 00:30:10.344888353 +0200
@@ -183,4 +183,4 @@
 void stop_players_from_joining();
 //int create_player(int host);
 //void turn_on_service(int service);
-#endif
\ Dosya sonunda yenisatır yok.
+#endif
diff -Nur egoboo-2.7.5-old/game/Timer.h egoboo-2.7.5/game/Timer.h
--- egoboo-2.7.5-old/game/Timer.h	2007-12-28 00:30:05.829574785 +0200
+++ egoboo-2.7.5/game/Timer.h	2007-12-28 00:30:10.347887897 +0200
@@ -23,7 +23,8 @@
 #ifndef Egoboo_Timer_h
 #define Egoboo_Timer_h
 
-typedef struct clock_state_t ClockState;
+/* For ClockState */
+#include "Clock.h"
 
 typedef struct timer_t
 {
diff -Nur egoboo-2.7.5-old/game/Ui.c egoboo-2.7.5/game/Ui.c
--- egoboo-2.7.5-old/game/Ui.c	2007-12-28 00:30:05.829574785 +0200
+++ egoboo-2.7.5/game/Ui.c	2007-12-28 00:30:10.348887745 +0200
@@ -31,7 +31,7 @@
 GLfloat ui_hot_color[]    = {0.54f, 0, 0, 1.0f};
 GLfloat ui_normal_color[] = {0.66f, 0, 0, 0.6f};
 
-typedef struct ui_context_t
+struct ui_context_t
 {
   // Tracking control focus stuff
   UI_ID active;
@@ -44,7 +44,7 @@
 
   Font *defaultFont;
   Font *activeFont;
-} UiContext;
+};
 
 static UiContext ui_context;
 
diff -Nur egoboo-2.7.5-old/game/Ui.h egoboo-2.7.5/game/Ui.h
--- egoboo-2.7.5-old/game/Ui.h	2007-12-28 00:30:05.830574633 +0200
+++ egoboo-2.7.5/game/Ui.h	2007-12-28 00:30:10.352887137 +0200
@@ -101,7 +101,7 @@
 /*****************************************************************************/
 
 // Behaviors
-int  ui_buttonBehavior(ui_Widget * pWidget);
+ui_buttonValues ui_buttonBehavior(ui_Widget * pWidget);
 
 // Drawing
 void ui_drawButton(ui_Widget * pWidget);
