--- a/messages/CMakeLists.txt.orig
+++ b/messages/CMakeLists.txt
@@ -16,4 +16,5 @@ add_subdirectory(kdetoys)
 add_subdirectory(kdeutils)
 add_subdirectory(kdewebdev)
 add_subdirectory(qt)
+add_subdirectory(playground-network)
 install(FILES entry.desktop DESTINATION ${LOCALE_INSTALL_DIR}/${CURRENT_LANG}/ )
--- /dev/null
+++ a/messages/playground-network/CMakeLists.txt
@@ -0,0 +1,2 @@
+file(GLOB _po_files *.po)
+GETTEXT_PROCESS_PO_FILES(${CURRENT_LANG} ALL INSTALL_DESTINATION ${LOCALE_INSTALL_DIR} ${_po_files} )

