Index: umplayer-0.0_180/src/favorites.cpp
===================================================================
--- umplayer-0.0_180.orig/src/favorites.cpp
+++ umplayer-0.0_180/src/favorites.cpp
@@ -135,31 +135,53 @@ int Favorites::findFile(QString filename
 void Favorites::next() {
 	qDebug("Favorites::next");
 
+	// Find the index of the current file/tv
+	// -1 means not found
 	int current = findFile(current_file);
-	//if (current < 0) current = 0;
+
+	if (current == -1)
+		return;
 
 	current++;
-	if (current >= f_list.count()) current = 0;
+	if (current >= f_list.count())
+		current = 0;
+
+	int next_file = current + FIRST_MENU_ENTRY;
 
-        QAction * a = _menu->actions().value(current+FIRST_MENU_ENTRY, 0); // Skip "edit" and separator
-	if (a != 0) {
-		a->trigger();
+	if (next_file < _menu->actions().count()) {
+		QAction * a = _menu->actions()[next_file]; // Skip "edit" and separator
+		if (a != 0) {
+			a->trigger();
+		}
 	}
+	else
+		return;
 }
 
 void Favorites::previous() {
 	qDebug("Favorites::previous");
 
+	// Find the index of the current file/tv
+	// -1 means not found
 	int current = findFile(current_file);
-	//if (current < 0) current = 0;
+
+	if (current == -1)
+		return;
 
 	current--;
-	if (current < 0) current = f_list.count()-1;
+	if (current < 0)
+		current = f_list.count() - 1;
+
+	int previous_file = current + FIRST_MENU_ENTRY;
 
-	QAction * a = _menu->actions()[current+FIRST_MENU_ENTRY]; // Skip "edit" and separator
-	if (a != 0) {
-		a->trigger();
+	if (previous_file < _menu->actions().count()) {
+		QAction * a = _menu->actions()[previous_file]; // Skip "edit" and separator
+		if (a != 0) {
+			a->trigger();
+		}
 	}
+	else
+		return;
 }
 
 void Favorites::save() {
