Index: umplayer-0.6.9_20100211/src/myaction.cpp
===================================================================
--- umplayer-0.6.9_20100211.orig/src/myaction.cpp
+++ umplayer-0.6.9_20100211/src/myaction.cpp
@@ -72,14 +72,14 @@ void MyAction::change(const QIcon & icon
 }
 
 void MyAction::change(const QString & text ) {
-	setText( text );
+	setText( text.trimmed() );
 
 	QString accel_text = shortcut().toString();
 
 	QString s = text;
 	s.replace("&","");
 	if (!accel_text.isEmpty()) {
-		setToolTip( s + " ("+ accel_text +")");
+		setToolTip( s.trimmed() + " ("+ tr(accel_text.toAscii().constData()) +")");
 		setIconText( s );
 	}
 
