--- src/tvtimeosd.c.str	2009-04-11 21:40:11.000000000 +0200
+++ src/tvtimeosd.c	2009-04-11 21:40:22.000000000 +0200
@@ -558,7 +558,7 @@
 
 void tvtime_osd_set_audio_mode( tvtime_osd_t *osd, const char *audiomode )
 {
-    snprintf( osd->audiomode_text, sizeof( osd->audiomode_text ), audiomode );
+    snprintf( osd->audiomode_text, sizeof( osd->audiomode_text ), "%s", audiomode );
 }
 
 void tvtime_osd_set_channel_number( tvtime_osd_t *osd, const char *text )
