diff -up tvtime-1.0.2/src/xmltv.c.localedef tvtime-1.0.2/src/xmltv.c
--- tvtime-1.0.2/src/xmltv.c.localedef	2009-05-31 15:33:13.000000000 +0200
+++ tvtime-1.0.2/src/xmltv.c	2009-05-31 15:33:57.000000000 +0200
@@ -118,9 +118,9 @@ const int num_timezones = sizeof( date_m
 typedef struct {
     const char *code;
     const char *name;
-} locale_t;
+} xmltv_locale_t;
 
-static locale_t locale_table[] = {
+static xmltv_locale_t locale_table[] = {
     {"AA", "Afar"},           {"AB", "Abkhazian"},      {"AF", "Afrikaans"},
     {"AM", "Amharic"},        {"AR", "Arabic"},         {"AS", "Assamese"},
     {"AY", "Aymara"},         {"AZ", "Azerbaijani"},    {"BA", "Bashkir"},
@@ -168,7 +168,7 @@ static locale_t locale_table[] = {
     {"XH", "Xhosa"},          {"YO", "Yoruba"},         {"ZH", "Chinese"},
     {"ZU", "Zulu"} };
 
-const int num_locales = sizeof( locale_table ) / sizeof( locale_t );
+const int num_locales = sizeof( locale_table ) / sizeof( xmltv_locale_t );
 
 /**
  * Timezone parsing code based loosely on the algorithm in
