diff -up ecore-1.2.1/src/lib/ecore/ecore_exe.c.linux-priority ecore-1.2.1/src/lib/ecore/ecore_exe.c
--- ecore-1.2.1/src/lib/ecore/ecore_exe.c.linux-priority	2012-08-03 10:04:03.071057501 -0400
+++ ecore-1.2.1/src/lib/ecore/ecore_exe.c	2012-08-03 10:04:40.043058056 -0400
@@ -2,7 +2,7 @@
 # include <config.h>
 #endif
 
-#if defined (__FreeBSD__) || defined (__OpenBSD__) || defined (__NetBSD__)
+#if defined (__FreeBSD__) || defined (__OpenBSD__) || defined (__NetBSD__) || defined (__linux__)
 # include <sys/time.h>
 # include <sys/resource.h>
 #endif
