Description: Don't hide the build process to aid debugging.
Author: Eugene Paskevich <eugene@raptor.kiev.ua>
Forwarded: not-needed
Last-Update: 2010-08-03

--- libaosd-0.2.6.orig/buildsys.mk.in
+++ libaosd-0.2.6/buildsys.mk.in
@@ -73,7 +73,6 @@
 
 MO_FILES = ${LOCALES:.po=.mo}
 
-.SILENT:
 .SUFFIXES:
 .SUFFIXES: .beam .c .cc .cxx .d .dep .erl .lib.o .mo .m .mm .o .plugin.o .po .py .pyc .xpm .S
 .PHONY: all subdirs pre-depend depend install install-extra uninstall uninstall-extra clean distclean
