diff -Naur icecast-2.3.2.orig/m4/acx_pthread.m4 icecast-2.3.2/m4/acx_pthread.m4
--- icecast-2.3.2.orig/m4/acx_pthread.m4	2009-02-22 00:33:42.000000000 +0200
+++ icecast-2.3.2/m4/acx_pthread.m4	2009-05-01 05:27:18.000000000 +0300
@@ -63,7 +63,7 @@
 # pthread-config: use pthread-config program (for GNU Pth library)
 
 case "${host_cpu}-${host_os}" in
-        *solaris*)
+        *solaris*|*linux*)
 
         # On Solaris (at least, for some versions), libc contains stubbed
         # (non-functional) versions of the pthreads routines, so link-based
