Index: kde4libs-4.6.1/solid/solid/backends/hal/halstorageaccess.cpp
===================================================================
--- kde4libs-4.6.1.orig/solid/solid/backends/hal/halstorageaccess.cpp	2011-02-26 13:44:47.400828876 +0100
+++ kde4libs-4.6.1/solid/solid/backends/hal/halstorageaccess.cpp	2011-02-26 13:46:55.220828824 +0100
@@ -25,6 +25,7 @@
 
 #include <QtCore/QLocale>
 #include <QtCore/QDebug>
+#include <QtCore/QTextStream>
 #include <QtCore/QProcess>
 #include <QtCore/QTimer>
 #include <QtDBus/QDBusConnection>
@@ -33,9 +34,11 @@
 #include <QtDBus/QDBusVariant>
 #include <QtGui/QApplication>
 #include <QtGui/QWidget>
+#include <localization/klocalizedstring.h>
 
 #include <unistd.h>
 #include <stdlib.h>
+#include <locale.h>
 
 #ifdef Q_OS_FREEBSD
 #include <langinfo.h>
@@ -247,10 +250,19 @@
 {
     // TODO: Better error reporting here
     if (m_setupInProgress) {
+       if (error.name() == "org.freedesktop.Hal.Device.Volume.PermissionDenied") {
+          callHalPrivilegedVolumeMount();
+          return;
+        }
         m_setupInProgress = false;
         m_device->broadcastActionDone("setup", Solid::UnauthorizedOperation,
                                       QString(error.name()+": "+error.message()));
     } else if (m_teardownInProgress) {
+        if (error.name() == "org.freedesktop.Hal.Device.PermissionDeniedByPolicy") {
+            callHalPrivilegedVolumeUnmount();
+            return;
+        }
+
         m_teardownInProgress = false;
         m_device->broadcastActionDone("teardown", Solid::UnauthorizedOperation,
                                       QString(error.name()+": "+error.message()));
@@ -393,10 +405,17 @@
 #else
     QString uid="uid=";
 #endif
-    if (halOptions.contains(uid)) {
+    QString fsType = m_device->property("volume.fstype").toString();
+
+    if (halOptions.contains("uid=") && !fsType.contains("ntfs")) {
         options << uid+QString::number(::getuid());
     }
 
+    if (fsType.contains("ntfs")) {
+        options << "locale=" + QString(setlocale(LC_ALL, ""));
+    }
+
+
 #ifdef Q_OS_FREEBSD
     char *cType;
     if ( fstype=="vfat" && halOptions.contains("-L=")) {
@@ -476,6 +495,87 @@
     return m_process!=0;
 }
 
+QString sudoCommandName()
+{
+    return "/usr/lib/kde4/libexec/kdesu";
+}
+
+QString dbusSendCommandName()
+{
+    return "dbus-send";
+}
+
+QProcess* callPrivilegedCommand(const QString& command,
+                                QObject* obj, const char* slot,
+                                const QString& comment)
+{
+    QProcess* process = new QProcess(obj);
+
+    QObject::connect(process, SIGNAL(finished(int, QProcess::ExitStatus)),
+                     obj, slot);
+
+    QStringList commandArgs;
+    commandArgs << "-d" << "--noignorebutton";
+    if(!comment.isEmpty())
+        commandArgs << "--comment" << comment;
+    commandArgs << "-c" << command;
+
+    process->start(sudoCommandName(), commandArgs);
+
+    if (process->waitForStarted()) {
+        return process;
+    } else {
+        delete process;
+        return 0;
+    }
+}
+
+bool Solid::Backends::Hal::StorageAccess::callHalPrivilegedVolumeMount()
+{
+    QStringList options;
+    QStringList halOptions = m_device->property("volume.mount.valid_options").toStringList();
+
+    QString fsType = m_device->property("volume.fstype").toString();
+
+    if (halOptions.contains("uid=") && !fsType.contains("ntfs")) {
+        options << "uid="+QString::number(::getuid());
+    }
+
+    if (fsType.contains("ntfs")) {
+        options << "locale=" + QString(setlocale(LC_ALL, ""));
+    }
+
+    QString command;
+    QTextStream(&command) << dbusSendCommandName()
+            << " --system --print-reply --dest=org.freedesktop.Hal " << m_device->udi()
+            << " org.freedesktop.Hal.Device.Volume.Mount string:" << filePath()
+            << " string: array:string:" << options.join(",");
+
+    m_process = callPrivilegedCommand(command,
+                                      this,
+                                      SLOT(slotProcessFinished(int, QProcess::ExitStatus)),
+                                      QObject::tr("Please enter your password to use this device"));
+
+    return (m_process != 0);
+}
+
+bool Solid::Backends::Hal::StorageAccess::callHalPrivilegedVolumeUnmount()
+{
+    QString command;
+    QTextStream(&command) << dbusSendCommandName()
+            << " --system --print-reply --dest=org.freedesktop.Hal " << m_device->udi()
+            << " org.freedesktop.Hal.Device.Volume.Unmount array:string:force";
+
+    m_process = callPrivilegedCommand(command,
+                                      this,
+                                      SLOT(slotProcessFinished(int, QProcess::ExitStatus)),
+                                      QObject::tr("Please enter your password to safely remove this device"));
+
+    return m_process != 0;
+}
+
+
+
 bool Solid::Backends::Hal::StorageAccess::callSystemUnmount()
 {
     const QString device = m_device->prop("block.device").toString();
Index: kde4libs-4.6.1/solid/solid/backends/hal/halstorageaccess.h
===================================================================
--- kde4libs-4.6.1.orig/solid/solid/backends/hal/halstorageaccess.h	2011-02-26 13:44:47.390828876 +0100
+++ kde4libs-4.6.1/solid/solid/backends/hal/halstorageaccess.h	2011-02-26 13:44:51.870828874 +0100
@@ -79,6 +79,9 @@
     bool callHalVolumeUnmount();
     bool callHalVolumeEject();
 
+    bool callHalPrivilegedVolumeMount();
+    bool callHalPrivilegedVolumeUnmount();
+
     bool callSystemMount();
     bool callSystemUnmount();
 
Index: kde4libs-4.6.1/solid/solid/CMakeLists.txt
===================================================================
--- kde4libs-4.6.1.orig/solid/solid/CMakeLists.txt	2011-02-26 13:44:47.410828876 +0100
+++ kde4libs-4.6.1/solid/solid/CMakeLists.txt	2011-02-26 13:44:51.870828874 +0100
@@ -1,6 +1,6 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
 add_subdirectory( ifaces )
-include_directories( ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR} )
+include_directories( ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR} ${PROJECT_SOURCE_DIR}/kdecore )
 
 if(WIN32)
    include_directories( ${KDEWIN_INCLUDES} )
@@ -277,6 +277,9 @@
    )
 endif(APPLE)
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -I${PROJECT_SOURCE_DIR}/kdecore")
+set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -L${PROJECT_BINARY_DIR}/lib")
+
 if(WIN32)
    include(CheckIncludeFileCXX)
    check_include_file_cxx(wbemidl.h HAVE_WBEM)
