diff -up scorched/src/client/client/SecureID.cpp~ scorched/src/client/client/SecureID.cpp
--- scorched/src/client/client/SecureID.cpp~	2010-07-20 02:28:59.000000000 +0200
+++ scorched/src/client/client/SecureID.cpp	2012-01-17 20:26:20.632511395 +0100
@@ -75,6 +75,7 @@ std::string SecureID::GetPrivateKey(void
 
 #include <sys/ioctl.h>
 #include <net/if.h>
+#include <unistd.h>
 
 std::string SecureID::GetPrivateKey(void)
 {
