--- netpbm-10.32/converter/other/giftopnm.c.message	2006-02-15 23:53:45.000000000 +0100
+++ netpbm-10.32/converter/other/giftopnm.c	2006-02-27 14:47:32.000000000 +0100
@@ -1401,7 +1401,7 @@ convertImages(FILE * const ifP, 
                          imageSeq, imageSeq>1?"s":"" );
         } else {
             if (verbose)
-                pm_message("Reading Image Sequence %d", imageSeq);
+                pm_message("Reading Image Sequence %d", imageSeq+1);
             convertImage(ifP, !allImages && (imageSeq != requestedImageSeq), 
                          imageout_file, alphafile, gifScreen, gif89);
         }
