http://bugs.gentoo.org/291987

--- netpbm/converter/other/Makefile
+++ netbpm/converter/other/Makefile
@@ -159,6 +156,7 @@
 	  $(shell $(LIBOPT) $(NETPBMLIB)) \
 	  $(PNGLIB_LIBOPTS) $(MATHLIB) $(LDFLAGS) $(LDLIBS) $(RPATH) $(LADD)
 
+pnmtopng: LADD += -lz # pnmtopng calls zlib_version itself
 pnmtopng: %: %.o pngx.o pngtxt.o $(NETPBMLIB) $(LIBOPT)
 	$(LD) -o $@ $@.o pngtxt.o \
 	  $(shell $(LIBOPT) $(NETPBMLIB)) \
