--- netpbm-10.35/converter/ppm/ppmtompeg/fsize.c.ppmtompeg	2006-09-18 12:20:24.000000000 +0200
+++ netpbm-10.35/converter/ppm/ppmtompeg/fsize.c	2006-09-18 13:35:53.000000000 +0200
@@ -115,8 +115,7 @@ Fsize_Note(int          const id,
     Fsize_Validate(&Fsize_x, &Fsize_y);
 
     if ((Fsize_x == 0) || (Fsize_y == 0)) {
-        fprintf(stderr,"Frame %d:  size is less than the minimum: %d x %d!\n",
-                id, DCTSIZE*2, DCTSIZE*2);
+    	fprintf(stderr,"Frame %d: image dimensions need to be integer multiples of 16 for ppmtompeg!\n",id);
         exit(1);
     }
 
--- netpbm-10.35/converter/ppm/ppmtompeg/parallel.c.ppmtompeg	2006-09-18 13:30:49.000000000 +0200
+++ netpbm-10.35/converter/ppm/ppmtompeg/parallel.c	2006-09-18 13:32:24.000000000 +0200
@@ -1316,8 +1316,6 @@ CombineServer(int          const numFram
   TransmitPortNum(masterHostName, masterPortNum, combinePortNum);
   
   close(outputServerSocket);
-
-  fclose(ofP);
 }
 
 
