--- netpbm-10.24/GNUmakefile.nodoc	2004-08-11 20:41:54.000000000 +0200
+++ netpbm-10.24/GNUmakefile	2004-09-23 16:08:29.724369240 +0200
@@ -275,16 +275,12 @@
 endif
 
 .PHONY: install.manweb
-install.manweb: $(PKGDIR)/man/web/netpbm.url $(PKGDIR)/bin/doc.url
+install.manweb: $(PKGDIR)/man/web/netpbm.url
 
 $(PKGDIR)/man/web/netpbm.url: $(PKGDIR)/man/web
 	echo "$(NETPBM_DOCURL)" > $@
 	chmod $(INSTALL_PERM_MAN) $@
 
-$(PKGDIR)/bin/doc.url: $(PKGDIR)/bin
-	echo "$(NETPBM_DOCURL)" > $@
-	chmod $(INSTALL_PERM_MAN) $@
-
 .PHONY: install-dev
 # Note that you might install the development package and NOT the runtime
 # package.  If you have a special system for building stuff, maybe for 
