Index: openscam_0.0.11/SConstruct
===================================================================
--- openscam_0.0.11.orig/SConstruct
+++ openscam_0.0.11/SConstruct
@@ -12,7 +12,7 @@ major, minor, revision = version.split('
 
 # Setup
 vars = [('install_prefix', 'Instalation directory prefix', '/usr/local/')]
-env = config.make_env(['compiler', 'cbang', 'dist', 'freetype2', 'opengl',
+env = os.environ(['compiler', 'cbang', 'dist', 'freetype2', 'opengl',
                        'resources', 'packager'], vars)
 
 # Configure
