diff -p -up k3b-1.66.0/libk3b/jobs/k3bmetawriter.cpp.growisofs k3b-1.66.0/libk3b/jobs/k3bmetawriter.cpp
--- k3b-1.66.0/libk3b/jobs/k3bmetawriter.cpp.growisofs	2009-06-10 18:01:47.000000000 -0300
+++ k3b-1.66.0/libk3b/jobs/k3bmetawriter.cpp	2009-06-10 18:02:57.000000000 -0300
@@ -310,14 +310,9 @@ bool K3b::MetaWriter::determineUsedAppAn
                 emit infoMessage( i18n("DVD and Blu-Ray tracks can only be written in MODE1."), MessageWarning );
             }
 
-            if( mediaType & (K3b::Device::MEDIA_DVD_PLUS_RW|K3b::Device::MEDIA_DVD_RW_OVWR) &&
-                d->multiSession ) {
-                // we can only do this with growisofs
+            if( mediaType & Device::MEDIA_DVD_ALL ) {
                 d->usedWritingApp = WritingAppGrowisofs;
             }
-            else if( mediaType & Device::MEDIA_DVD_ALL ) {
-                d->usedWritingApp = WritingAppCdrecord;
-            }
             else if( mediaType & Device::MEDIA_BD_ALL ) {
                 if( cdrecordBluRay ) {
                     d->usedWritingApp = WritingAppCdrecord;
