--- 0ad-0.0.16-alpha/build/premake/premake4.lua.orig	2014-05-17 22:21:52.787713008 +0200
+++ 0ad-0.0.16-alpha/build/premake/premake4.lua	2014-05-17 22:22:39.659845979 +0200
@@ -356,26 +356,6 @@
 			defines { "INSTALLED_LIBDIR=" .. _OPTIONS["libdir"] }
 		end
 
-		if os.is("linux") or os.is("bsd") then
-			-- To use our local shared libraries, they need to be found in the
-			-- runtime dynamic linker path. Add their path to -rpath.
-			if _OPTIONS["libdir"] then
-				linkoptions {"-Wl,-rpath," .. _OPTIONS["libdir"] }
-			else
-				-- On FreeBSD we need to allow use of $ORIGIN
-				if os.is("bsd") then
-					linkoptions { "-Wl,-z,origin" }
-				end
-
-				-- Adding the executable path and taking care of correct escaping
-				if _ACTION == "gmake" then
-					linkoptions { "-Wl,-rpath,'$$ORIGIN'" } 
-				elseif _ACTION == "codeblocks" then
-					linkoptions { "-Wl,-R\\\\$$$ORIGIN" }
-				end
-			end
-		end
-
 	end
 end
 