diff -Nuar a/build/resources/0ad.desktop b/build/resources/0ad.desktop
--- a/build/resources/0ad.desktop	2013-04-21 19:47:36.000000000 +0300
+++ b/build/resources/0ad.desktop	2013-09-07 06:28:24.170885044 +0300
@@ -1,8 +1,15 @@
 [Desktop Entry]
+Encoding=UTF-8
 Version=1.0
-Name=0 A.D.
+Name=0AD
+GenericName=0AD strategy game
+GenericName[de]=0AD Strategie Spiel
+GenericName[it]=0AD Videogioco strategico
+GenericName[tr]=0AD strateji oyunu
 Comment=A real-time strategy game of ancient warfare
+Comment[de]= Eine Echtzeit-Strategie-Spiel der antiken Kriegsführung
 Comment[it]=Videogioco strategico in tempo reale di guerre antiche
+Comment[tr]= Antik çağlarda geçen gerçek zamanlı strateji oyunu
 Exec=0ad
 Icon=0ad
 Terminal=false
