diff -up AstroMenaceSourceCode/AstroMenaceSource/Core/Base.h~ AstroMenaceSourceCode/AstroMenaceSource/Core/Base.h
--- AstroMenaceSourceCode/AstroMenaceSource/Core/Base.h~	2007-09-03 12:42:03.000000000 +0200
+++ AstroMenaceSourceCode/AstroMenaceSource/Core/Base.h	2009-08-18 09:22:10.000000000 +0200
@@ -50,13 +50,15 @@
 	#include <GL/gl.h>	// Header File For The OpenGL32 Library
 	#include <GL/glu.h>	// Header File For The GLu32 Library
 	#undef __glext_h_
+	/* We want / need the OpenGL 1.3 function pointer typedefs */
+	#undef GL_VERSION_1_3
+	#include <GL/glext.h>
 #endif
 
 #include <stdlib.h> // rand, ...
 #include <math.h> // math for core Math
 
 
-#include "RendererInterface/GLext.h"
 #include "SDL/SDL.h"
 #include "SDL/SDL_thread.h"
 #include "SDL/SDL_syswm.h"
