diff -Nur urwid-0.9.9.1-old/urwid/raw_display.py urwid-0.9.9.1/urwid/raw_display.py
--- urwid-0.9.9.1-old/urwid/raw_display.py	2010-07-01 20:05:34.100028116 +0300
+++ urwid-0.9.9.1/urwid/raw_display.py	2010-07-01 20:05:49.698028346 +0300
@@ -400,7 +400,7 @@
     def _get_gpm_codes(self):
         codes = []
         try:
-            while self.gpm_mev is not None:
+            while self.gpm_mev is not None and self.gpm_event_pending:
                 codes.extend(self._encode_gpm_event())
         except IOError, e:
             if e.args[0] != 11:
