--- configs/global~	2008-11-09 07:17:11.000000000 +0100
+++ configs/global	2008-11-09 07:19:44.000000000 +0100
@@ -1,4 +1,4 @@
-/SupportedUnicodeLocales = en_US.UTF-8
+/SupportedUnicodeLocales = en_US.UTF-8,zh_CN.UTF-8,zh_HK.UTF-8,zh_TW.UTF-8,jp_JP.UTF-8,ko_KR.UTF-8
 /DefaultPanelProgram = scim-panel-gtk
 /DefaultConfigModule = simple
 /DefaultSocketFrontEndAddress = local:/tmp/scim-socket-frontend
