diff -p -up pidgin-2.7.0/pidgin/pixmaps/emotes/default/24/theme.smiley pidgin-2.7.0/pidgin/pixmaps/emotes/default/24/theme
--- pidgin-2.7.0/pidgin/pixmaps/emotes/default/24/theme.smiley	2010-05-12 06:37:31.000000000 +0200
+++ pidgin-2.7.0/pidgin/pixmaps/emotes/default/24/theme	2010-05-13 21:15:22.000000000 +0200
@@ -6,9 +6,9 @@ Author=Hylke Bons
 
 # Default smileys
 [default]
-happy.png           :)      :-)
+happy.png           :)      :-)     =)
 excited.png         :-D     :-d     :D      :d
-sad.png             :-(     :(
+sad.png             :-(     :(      =(
 wink.png            ;-)     ;)
 tongue.png          :P      :p      :-P     :-p
 shocked.png         =-O     =-o
@@ -16,7 +16,7 @@ kiss.png            :-*
 glasses-cool.png    8-)
 embarrassed.png     :-[
 crying.png          :'(     :'-(
-thinking.png        :-/     :-\\
+thinking.png        :-/     :-\      :\
 angel.png           O:-)    o:-)
 shut-mouth.png      :-X
 moneymouth.png      :-$
