--- lib/Makefile.am~	2013-01-29 18:03:26.420468163 +0100
+++ lib/Makefile.am	2013-01-29 18:04:32.143465762 +0100
@@ -1632,6 +1632,6 @@
 
 install-data-hook:
 	$(CHMOD) 755 $(DESTDIR)$(pkgdatadir)/configure.py
-	for i in $(scriptdir)/*.py ; do \
+	for i in $(scriptdir)/scripts/*.py ; do \
 		$(CHMOD) 755 $(DESTDIR)$(pkgdatadir)/$$i; \
 	done
--- lib/Makefile.in~	2013-01-29 18:03:35.483467832 +0100
+++ lib/Makefile.in	2013-01-29 18:04:59.846697311 +0100
@@ -3429,7 +3429,7 @@
 
 install-data-hook:
 	$(CHMOD) 755 $(DESTDIR)$(pkgdatadir)/configure.py
-	for i in $(scriptdir)/*.py ; do \
+	for i in $(scriptdir)/scripts/*.py ; do \
 		$(CHMOD) 755 $(DESTDIR)$(pkgdatadir)/$$i; \
 	done
 
