diff -up doxygen-1.8.5/src/configoptions.cpp.html_timestamp_default_false doxygen-1.8.5/src/configoptions.cpp
--- doxygen-1.8.5/src/configoptions.cpp.html_timestamp_default_false	2013-08-26 16:07:23.000000000 +0200
+++ doxygen-1.8.5/src/configoptions.cpp	2013-08-26 18:31:40.000000000 +0200
@@ -1499,7 +1499,7 @@ void addConfigOptions(Config *cfg)
               "to NO can help when comparing the output of multiple runs.\n"
               "The default value is: YES.\n"
               "This tag requires that the tag GENERATE_HTML is set to YES.",
-              TRUE
+              FALSE
              );
   cb->addDependency("GENERATE_HTML");
   //----
