diff -up doxygen-1.8.3/src/htmlgen.cpp.multilib doxygen-1.8.3/src/htmlgen.cpp
--- doxygen-1.8.3/src/htmlgen.cpp.multilib	2012-12-25 11:27:52.000000000 +0100
+++ doxygen-1.8.3/src/htmlgen.cpp	2013-01-02 17:08:57.559028518 +0100
@@ -125,6 +125,17 @@ static unsigned char tab_b_png[36] =
     205, 207, 209, 211, 213, 217, 219, 206, 188 
 };
 
+static unsigned char nav_g_png[95] = {
+0x89, 0x50, 0x4e, 0x47, 0x0d, 0x0a, 0x1a, 0x0a, 0x00, 0x00, 0x00, 0x0d, 
+0x49, 0x48, 0x44, 0x52, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x06, 
+0x08, 0x06, 0x00, 0x00, 0x00, 0x02, 0x10, 0xf4, 0x31, 0x00, 0x00, 0x00, 
+0x26, 0x49, 0x44, 0x41, 0x54, 0x78, 0x01, 0xed, 0xdd, 0x31, 0x0d, 0x00, 
+0x00, 0x08, 0x03, 0xc1, 0x4f, 0x48, 0x17, 0x66, 0x84, 0xe1, 0x5f, 0x0d, 
+0x2d, 0x3e, 0x7e, 0xb8, 0xe5, 0x4d, 0x20, 0x69, 0xcb, 0x4d, 0xe8, 0xc0, 
+0x83, 0x07, 0x07, 0x0f, 0x53, 0x00, 0xbd, 0xfc, 0x82, 0x3c, 0x05, 0x00, 
+0x00, 0x00, 0x00, 0x49, 0x45, 0x4e, 0x44, 0xae, 0x42, 0x60, 0x82, 
+};
+
 // hovering tab background luma
 static unsigned char tab_h_png[36] =
 {
@@ -824,6 +835,7 @@ static ColoredImgDataItem colored_tab_da
   { "tab_s.png",    1, 36, tab_s_png,       0 },
   { "nav_h.png",    1, 12, header_png,      0 },
   { "nav_f.png",    1, 56, func_header_png, 0 },
+  { "nav_g.png",    1,  6, nav_g_png,       0 },
   { "bc_s.png",     8, 30, bc_s_png,        bc_s_a_png },
   { "doxygen.png", 104,31, doxygen_png,     doxygen_a_png },
   { "closed.png",   9,  9, closed_png,      closed_a_png },
@@ -1502,13 +1514,6 @@ void HtmlGenerator::writeTabData()
   Doxygen::indexList->addStyleSheetFile("tabs.css");
   QCString dname=Config_getString("HTML_OUTPUT");
   writeColoredImgData(dname,colored_tab_data);
-
-  {
-    unsigned char shadow[6] = { 5, 5, 5, 5, 5, 5 };
-    unsigned char shadow_alpha[6]  = { 80, 60, 40, 20, 10, 0 };
-    ColoredImage img(1,6,shadow,shadow_alpha,0,0,100);
-    img.save(dname+"/nav_g.png");
-  }
 }
 
 void HtmlGenerator::writeSearchData(const char *dir)
