diff -u mysql-5.6.10.orig/storage/perfschema/unittest/CMakeLists.txt mysql-5.6.10/storage/perfschema/unittest/CMakeLists.txt
--- mysql-5.6.10.orig/storage/perfschema/unittest/CMakeLists.txt	2013-01-22 17:54:49.000000000 +0100
+++ mysql-5.6.10/storage/perfschema/unittest/CMakeLists.txt	2013-02-06 16:23:12.620915802 +0100
@@ -63,7 +63,7 @@
 TARGET_LINK_LIBRARIES(pfs_connect_attr-t sql binlog rpl master slave sql)
 TARGET_LINK_LIBRARIES(pfs_connect_attr-t mysys mysys_ssl)
 TARGET_LINK_LIBRARIES(pfs_connect_attr-t vio ${SSL_LIBRARIES})
-TARGET_LINK_LIBRARIES(pfs_connect_attr-t strings dbug regex mysys zlib)
+TARGET_LINK_LIBRARIES(pfs_connect_attr-t strings dbug regex mysys z)
 ADD_TEST(pfs_connect_attr pfs_connect_attr-t)
 
 # On windows, pfs_connect_attr-t may depend on openssl dlls.