Index: freeradius-server-2.2.0/raddb/radiusd.conf.in
===================================================================
--- freeradius-server-2.2.0.orig/raddb/radiusd.conf.in
+++ freeradius-server-2.2.0/raddb/radiusd.conf.in
@@ -66,7 +66,7 @@ name = radiusd
 
 #  Location of config and logfiles.
 confdir = ${raddbdir}
-run_dir = ${localstatedir}/run/${name}
+run_dir = /run/${name}
 
 # Should likely be ${localstatedir}/lib/radiusd
 db_dir = ${raddbdir}
@@ -112,7 +112,7 @@ libdir = @libdir@
 #
 #  This file is written when ONLY running in daemon mode.
 #
-#  e.g.:  kill -HUP `cat /var/run/radiusd/radiusd.pid`
+#  e.g.:  kill -HUP `cat /run/radiusd/radiusd.pid`
 #
 pidfile = ${run_dir}/${name}.pid
 
Index: freeradius-server-2.2.0/raddb/vmpsd.conf.in
===================================================================
--- freeradius-server-2.2.0.orig/raddb/vmpsd.conf.in
+++ freeradius-server-2.2.0/raddb/vmpsd.conf.in
@@ -113,7 +113,7 @@ libdir = @libdir@
 #
 #  This file is written when ONLY running in daemon mode.
 #
-#  e.g.:  kill -HUP `cat /var/run/radiusd/radiusd.pid`
+#  e.g.:  kill -HUP `cat /run/radiusd/radiusd.pid`
 #
 pidfile = ${run_dir}/vmpsd.pid
 
Index: freeradius-server-2.2.0/redhat/freeradius.spec
===================================================================
--- freeradius-server-2.2.0.orig/redhat/freeradius.spec
+++ freeradius-server-2.2.0/redhat/freeradius.spec
@@ -924,7 +924,7 @@ exit 0
 
 * Thu Dec  3 2009 John Dennis <jdennis@redhat.com> - 2.1.7-3
 - bug #522111 non-conformant initscript
-  also change permission of /var/run/radiusd from 0700 to 0755
+  also change permission of /run/radiusd from 0700 to 0755
   so that "service radiusd status" can be run as non-root
 
 * Wed Sep 16 2009 Tomas Mraz <tmraz@redhat.com> - 2.1.7-2
Index: freeradius-server-2.2.0/scripts/freeradius_acct
===================================================================
--- freeradius-server-2.2.0.orig/scripts/freeradius_acct
+++ freeradius-server-2.2.0/scripts/freeradius_acct
@@ -25,7 +25,7 @@
 #%# capabilities=autoconf
 
 RADMIN=radmin
-SOCKETFILE=/var/run/radiusd/radiusd.sock
+SOCKETFILE=/run/radiusd/radiusd.sock
 
 if [ "$1" = "autoconf" ]; then
     #
Index: freeradius-server-2.2.0/scripts/freeradius.monit
===================================================================
--- freeradius-server-2.2.0.orig/scripts/freeradius.monit
+++ freeradius-server-2.2.0/scripts/freeradius.monit
@@ -8,7 +8,7 @@
 #  Totalmem limit should be lowered to 200.0 if none of the 
 #  interpreted language modules or rlm_cache are being used.
 #
-check process radiusd with pidfile /var/run/radiusd/radiusd.pid
+check process radiusd with pidfile /run/radiusd/radiusd.pid
    start program = "/etc/init.d/radiusd start"
    stop program = "/etc/init.d/radiusd stop"
    if failed host 127.0.0.1 port 1812 type udp protocol radius secret testing123 then alert
Index: freeradius-server-2.2.0/scripts/freeradius_proxy_acct
===================================================================
--- freeradius-server-2.2.0.orig/scripts/freeradius_proxy_acct
+++ freeradius-server-2.2.0/scripts/freeradius_proxy_acct
@@ -25,7 +25,7 @@
 #%# capabilities=autoconf
 
 RADMIN=radmin
-SOCKETFILE=/var/run/radiusd/radiusd.sock
+SOCKETFILE=/run/radiusd/radiusd.sock
 
 if [ "$1" = "autoconf" ]; then
     #
Index: freeradius-server-2.2.0/scripts/freeradius_proxy_auth
===================================================================
--- freeradius-server-2.2.0.orig/scripts/freeradius_proxy_auth
+++ freeradius-server-2.2.0/scripts/freeradius_proxy_auth
@@ -25,7 +25,7 @@
 #%# capabilities=autoconf
 
 RADMIN=radmin
-SOCKETFILE=/var/run/radiusd/radiusd.sock
+SOCKETFILE=/run/radiusd/radiusd.sock
 
 if [ "$1" = "autoconf" ]; then
     #
Index: freeradius-server-2.2.0/src/tests/eapsim-03/radiusd-example.txt
===================================================================
--- freeradius-server-2.2.0.orig/src/tests/eapsim-03/radiusd-example.txt
+++ freeradius-server-2.2.0/src/tests/eapsim-03/radiusd-example.txt
@@ -35,7 +35,7 @@ radacctdir = ${logdir}/radacct
 
 #  Location of config and logfiles.
 confdir = ${raddbdir}
-run_dir = ${localstatedir}/run/radiusd
+run_dir = /run/radiusd
 
 #
 #  The logging messages for the server are appended to the
@@ -84,7 +84,7 @@ libdir = ${exec_prefix}/lib
 #
 #  This file is written when ONLY running in daemon mode.
 #
-#  e.g.:  kill -HUP `cat /var/run/radiusd/radiusd.pid`
+#  e.g.:  kill -HUP `cat /run/radiusd/radiusd.pid`
 #
 pidfile = ${run_dir}/radiusd.pid
 
Index: freeradius-server-2.2.0/suse/freeradius-server-rpmlintrc
===================================================================
--- freeradius-server-2.2.0.orig/suse/freeradius-server-rpmlintrc
+++ freeradius-server-2.2.0/suse/freeradius-server-rpmlintrc
@@ -1,4 +1,4 @@
-#freeradius-server.x86_64: E: dir-or-file-in-var-run (Badness: 900) /var/run/radiusd
+#freeradius-server.x86_64: E: dir-or-file-in-var-run (Badness: 900) /run/radiusd
 #A file or directory in the package is located in /var/run. It's not permitted
 #for packages to install files in this directory as it might be created as
 #tmpfs during boot. Modify your package to create the  necessary files during
Index: freeradius-server-2.2.0/suse/freeradius-server-tmpfiles.conf
===================================================================
--- freeradius-server-2.2.0.orig/suse/freeradius-server-tmpfiles.conf
+++ freeradius-server-2.2.0/suse/freeradius-server-tmpfiles.conf
@@ -1 +1 @@
-D /var/run/radiusd 0710 radiusd radiusd -
+D /run/radiusd 0710 radiusd radiusd -
Index: freeradius-server-2.2.0/suse/freeradius.spec
===================================================================
--- freeradius-server-2.2.0.orig/suse/freeradius.spec
+++ freeradius-server-2.2.0/suse/freeradius.spec
@@ -441,7 +441,7 @@ rm -rf $RPM_BUILD_ROOT
 
 %changelog
 * Fri Feb 10 2012 nix@opensuse.org
-- Add a /var/run/radiusd tmpfile.d config
+- Add a /run/radiusd tmpfile.d config
 - Add -fno-strict-aliasing to compiler flags to fix warning
 - Remove -DLDAP_DEPRECATED from compiler flags
 - Disable rlm_sqlhpwippool as there don't seem to be many users and
Index: freeradius-server-2.2.0/suse/radiusd-logrotate
===================================================================
--- freeradius-server-2.2.0.orig/suse/radiusd-logrotate
+++ freeradius-server-2.2.0/suse/radiusd-logrotate
@@ -25,7 +25,7 @@
     missingok
     create
     postrotate
-	kill -HUP `cat /var/run/radiusd/radiusd.pid`
+	kill -HUP `cat /run/radiusd/radiusd.pid`
 }
 
 /var/log/radius/radwatch.log {
Index: freeradius-server-2.2.0/scripts/freeradius_auth
===================================================================
--- freeradius-server-2.2.0.orig/scripts/freeradius_auth
+++ freeradius-server-2.2.0/scripts/freeradius_auth
@@ -25,7 +25,7 @@
 #%# capabilities=autoconf
 
 RADMIN=radmin
-SOCKETFILE=/var/run/radiusd/radiusd.sock
+SOCKETFILE=/run/radiusd/radiusd.sock
 
 if [ "$1" = "autoconf" ]; then
     #
Index: freeradius-server-2.2.0/scripts/logrotate.freeradius
===================================================================
--- freeradius-server-2.2.0.orig/scripts/logrotate.freeradius
+++ freeradius-server-2.2.0/scripts/logrotate.freeradius
@@ -11,6 +11,6 @@
 	missingok
 	notifempty
 	postrotate
-		kill -HUP `cat /var/run/radiusd.pid` > /dev/null 2>&1
+		kill -HUP `cat /run/radiusd.pid` > /dev/null 2>&1
 	endscript
 }
Index: freeradius-server-2.2.0/Make.inc.in
===================================================================
--- freeradius-server-2.2.0.orig/Make.inc.in
+++ freeradius-server-2.2.0/Make.inc.in
@@ -50,7 +50,7 @@ LDFLAGS		= @LDFLAGS@
 
 LOGDIR		= ${logdir}
 RADDBDIR	= ${raddbdir}
-RUNDIR		= ${localstatedir}/run/radiusd
+RUNDIR		= /run/radiusd
 SBINDIR		= ${sbindir}
 RADIR		= ${radacctdir}
 LIBRADIUS	= $(top_builddir)/src/lib/$(LIBPREFIX)freeradius-radius.la
