--- raddb/radiusd.conf.in.orig	2014-06-04 03:02:16.289472961 +0300
+++ raddb/radiusd.conf.in	2014-05-12 22:22:09.000000000 +0300
@@ -415,8 +415,8 @@
 	#  member.  This can allow for some finer-grained access
 	#  controls.
 	#
-    user = radiusd
-    group = radiusd
+#	user = radius
+#	group = radius
 
 	#  Core dumps are a bad thing.  This should only be set to
 	#  'yes' if you're debugging a problem with the server.
