Index: gtk-qt-engine/kcm_gtk/kcmgtk4.desktop
===================================================================
--- gtk-qt-engine.orig/kcm_gtk/kcmgtk4.desktop
+++ gtk-qt-engine/kcm_gtk/kcmgtk4.desktop
@@ -24,4 +24,4 @@ X-KDE-ServiceTypes=KCModule
 X-KDE-Library=kcm_gtk4
 X-KDE-ParentApp=kcontrol
 Categories=Settings;Qt;KDE;X-KDE-settings-looknfeel;
-X-KDE-System-Settings-Parent-Category=appearance
+X-KDE-System-Settings-Parent-Category=application-appearance
