diff -Naur ioPaintball-1.0.orig/code/q3_ui/ui_mods.c ioPaintball-1.0/code/q3_ui/ui_mods.c
--- ioPaintball-1.0.orig/code/q3_ui/ui_mods.c	2007-10-22 15:51:32.000000000 +0300
+++ ioPaintball-1.0/code/q3_ui/ui_mods.c	2007-11-10 22:42:02.000000000 +0200
@@ -125,7 +125,7 @@
 
 	// always start off with baseq3
 	s_mods.list.numitems = 1;
-	s_mods.list.itemnames[0] = s_mods.descriptionList[0] = "Quake III Arena";
+	s_mods.list.itemnames[0] = s_mods.descriptionList[0] = "ioPaintball";
 	s_mods.fs_gameList[0] = "";
 
 	numdirs = trap_FS_GetFileList( "$modlist", "", dirlist, sizeof(dirlist) );
diff -Naur ioPaintball-1.0.orig/code/qcommon/files.c ioPaintball-1.0/code/qcommon/files.c
--- ioPaintball-1.0.orig/code/qcommon/files.c	2007-11-09 22:24:38.000000000 +0200
+++ ioPaintball-1.0/code/qcommon/files.c	2007-11-10 22:13:11.000000000 +0200
@@ -2708,7 +2708,7 @@
 		homePath = fs_basepath->string;
 	}
 	fs_homepath = Cvar_Get ("fs_homepath", homePath, CVAR_INIT );
-	fs_gamedirvar = Cvar_Get ("fs_game", "", CVAR_INIT|CVAR_SYSTEMINFO );
+	fs_gamedirvar = Cvar_Get ("fs_game", "ioPaintball", CVAR_INIT|CVAR_SYSTEMINFO );
 
 	// add search path elements in reverse priority order
 	if (fs_basepath->string[0]) {
diff -Naur ioPaintball-1.0.orig/code/sys/sys_unix.c ioPaintball-1.0/code/sys/sys_unix.c
--- ioPaintball-1.0.orig/code/sys/sys_unix.c	2007-10-22 15:51:59.000000000 +0300
+++ ioPaintball-1.0/code/sys/sys_unix.c	2007-11-10 22:17:13.000000000 +0200
@@ -51,9 +51,9 @@
 		{
 			Q_strncpyz( homePath, p, sizeof( homePath ) );
 #ifdef MACOS_X
-			Q_strcat( homePath, sizeof( homePath ), "/Library/Application Support/Quake3" );
+			Q_strcat( homePath, sizeof( homePath ), "/Library/Application Support/ioPaintball" );
 #else
-			Q_strcat( homePath, sizeof( homePath ), "/.q3a" );
+			Q_strcat( homePath, sizeof( homePath ), "/.ioPaintball" );
 #endif
 			if( mkdir( homePath, 0777 ) )
 			{
