--- code/client/snd_dma.c.orig	2007-11-10 22:38:59.000000000 +0200
+++ code/client/snd_dma.c	2007-11-11 12:08:22.000000000 +0200
@@ -1447,7 +1447,7 @@
 		return qfalse;
 	}
 
-	s_khz = Cvar_Get ("s_khz", "22", CVAR_ARCHIVE);
+	s_khz = Cvar_Get ("s_khz", "48", CVAR_ARCHIVE);
 	s_mixahead = Cvar_Get ("s_mixahead", "0.2", CVAR_ARCHIVE);
 	s_mixPreStep = Cvar_Get ("s_mixPreStep", "0.05", CVAR_ARCHIVE);
 	s_show = Cvar_Get ("s_show", "0", CVAR_CHEAT);
