Index: alienarena-7.50/source/ref_gl/r_ragdoll.c
===================================================================
--- alienarena-7.50.orig/source/ref_gl/r_ragdoll.c
+++ alienarena-7.50/source/ref_gl/r_ragdoll.c
@@ -462,7 +462,9 @@ void R_RagdollBody_Init( int RagDollID,
 			if(!strcmp(&RagDoll[RagDollID].ragDollMesh->jointname[RagDoll[RagDollID].ragDollMesh->joints[i].name], RagDollBinds[j].name))
 			{
 				int object = RagDollBinds[j].object;
+				if ( ! IS_NAN( RagDoll[RagDollID].initframe[i].a[0] ) ) {
 				Matrix3x4_Add(&bindmat[object], bindmat[object], RagDoll[RagDollID].initframe[i]);
+				}
 				bindweight[object]++;
 				break;
 			}
Index: alienarena-7.50/source/client/menu.c
===================================================================
--- alienarena-7.50.orig/source/client/menu.c
+++ alienarena-7.50/source/client/menu.c
@@ -2001,7 +2001,7 @@ static void ControlsSetMenuItemValues( v
 	Cvar_SetValue("cl_explosiondist", ClampCvar(0, 1, cl_explosiondist->value ) );
 	s_options_explosiondist_box.curvalue		= cl_explosiondist->value;
 
-	Cvar_SetValue("r_ragdolls", ClampCvar(0, 1, r_ragdolls->value ) );
+	Cvar_SetValue("r_ragdolls", ClampCvar(0, 0, r_ragdolls->value ) );
 	s_options_ragdoll_box.curvalue		= r_ragdolls->value;
 
 	Cvar_SetValue("cl_noblood", ClampCvar(0, 1, cl_noblood->value ) );
