Index: alienarena-7.50/Makefile.in
===================================================================
--- alienarena-7.50.orig/Makefile.in
+++ alienarena-7.50/Makefile.in
@@ -58,7 +58,7 @@ DIST_COMMON = README $(am__configure_dep
 	$(nobase_dist_pkgdata_DATA) $(srcdir)/Makefile.am \
 	$(srcdir)/Makefile.in $(srcdir)/game_data.am \
 	$(top_srcdir)/config/config.h.in $(top_srcdir)/configure \
-	COPYING INSTALL config/compile config/config.guess \
+	COPYING config/compile config/config.guess \
 	config/config.sub config/depcomp config/install-sh \
 	config/missing
 subdir = .
@@ -3782,7 +3782,7 @@ distcleancheck: distclean
 	       exit 1; } >&2
 check-am: all-am
 check: check-recursive
-all-am: Makefile $(DATA)
+all-am: Makefile
 installdirs: installdirs-recursive
 installdirs-am:
 	for dir in "$(DESTDIR)$(docdir)" "$(DESTDIR)$(pkgdatadir)"; do \
@@ -3834,8 +3834,7 @@ info: info-recursive
 
 info-am:
 
-install-data-am: install-data-local install-dist_docDATA \
-	install-nobase_dist_pkgdataDATA
+install-data-am: install-data-local install-dist_docDATA 
 
 install-dvi: install-dvi-recursive
 
Index: alienarena-7.50/Makefile.am
===================================================================
--- alienarena-7.50.orig/Makefile.am
+++ alienarena-7.50/Makefile.am
@@ -29,7 +29,6 @@ dist_doc_DATA = \
 	docs/README.txt \
 	README \
 	COPYING \
-	INSTALL
 
 # FUSE game server browser and Server management scripts
 EXTRA_DIST = \
