Index: src/opendocumentthumbnail.desktop
===================================================================
--- src/opendocumentthumbnail.desktop.orig
+++ src/opendocumentthumbnail.desktop
@@ -2,6 +2,7 @@
 Type=Service
 Name=ODF (OpenDocument Format) Files
 Name[es]=Archivos ODF (OpenDocument Format)
+Name[tr]=ODF (Açık Doküman Biçimi) Dosyaları
 X-KDE-ServiceTypes=ThumbCreator
 MimeType=application/vnd.oasis.opendocument.text;application/vnd.oasis.opendocument.text-template;application/vnd.oasis.opendocument.text-master;application/vnd.oasis.opendocument.spreadsheet;application/vnd.oasis.opendocument.spreadsheet-template;application/vnd.oasis.opendocument.graphics;application/vnd.oasis.opendocument.graphics-template;application/vnd.oasis.opendocument.presentation;application/vnd.oasis.opendocument.presentation-template;application/vnd.oasis.opendocument.formula;application/vnd.oasis.opendocument.formula-template;application/vnd.oasis.opendocument.chart;application/vnd.oasis.opendocument.chart-template;
 X-KDE-Library=opendocumentthumbnail
