diff -up libmcrypt-2.5.8/Makefile.in.orig libmcrypt-2.5.8/Makefile.in
--- libmcrypt-2.5.8/Makefile.in.orig	2007-02-19 00:32:38.000000000 -0600
+++ libmcrypt-2.5.8/Makefile.in	2007-07-19 21:23:39.000000000 -0500
@@ -209,8 +209,8 @@ target_os = @target_os@
 target_vendor = @target_vendor@
 INCLUDES = $(INCLTDL)
 EXTRA_DIST = KNOWN-BUGS THANKS NEWS libmcrypt.spec.in libmcrypt.spec COPYING.LIB
-DIST_SUBDIRS = libltdl modules include lib src doc
-SUBDIRS = $(LIBLTDL_DIR) modules include lib src doc
+DIST_SUBDIRS = modules include lib src doc
+SUBDIRS = modules include lib src doc
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
diff -up libmcrypt-2.5.8/configure.orig libmcrypt-2.5.8/configure
--- libmcrypt-2.5.8/configure.orig	2007-02-19 00:32:39.000000000 -0600
+++ libmcrypt-2.5.8/configure	2007-07-19 21:21:38.000000000 -0500
@@ -426,7 +426,7 @@ PACKAGE_STRING=
 PACKAGE_BUGREPORT=
 
 ac_unique_file="lib/mcrypt.c"
-ac_subdirs_all="$ac_subdirs_all libltdl"
+ac_subdirs_all="$ac_subdirs_all"
 # Factoring default headers for most tests.
 ac_includes_default="\
 #include <stdio.h>
@@ -4425,7 +4425,7 @@ fi
 
 
 
-subdirs="$subdirs libltdl"
+subdirs="$subdirs"
 
  # Check whether --enable-static or --disable-static was given.
 if test "${enable_static+set}" = set; then
