--- configure	2007-10-08 19:38:57.000000000 +0300
+++ configure	2008-01-24 19:48:58.000000000 +0200
@@ -35855,8 +35855,8 @@
 #	define DB_VERSION_MINOR 0
 #endif
 
-/* require 4.2-4.5 */
-#if (DB_VERSION_MAJOR >= 4) && (DB_VERSION_MINOR >= 2) && (DB_VERSION_MINOR < 6)
+/* require 4.2-4.6 */
+#if (DB_VERSION_MAJOR >= 4) && (DB_VERSION_MINOR >= 2) && (DB_VERSION_MINOR < 7)
 	__db_version_compat
 #endif
 
