--- servers/slapd/slapd.conf.er	2004-06-18 05:49:08.000000000 +0300
+++ servers/slapd/slapd.conf	2006-06-28 11:20:21.000000000 +0300
@@ -10,11 +10,11 @@
 # service AND an understanding of referrals.
 #referral	ldap://root.openldap.org
 
-pidfile		%LOCALSTATEDIR%/run/slapd.pid
-argsfile	%LOCALSTATEDIR%/run/slapd.args
+pidfile		/var/run/openldap/slapd.pid
+argsfile	/var/run/openldap/slapd.args
 
 # Load dynamic backend modules:
-# modulepath	%MODULEDIR%
+modulepath	%MODULEDIR%
 # moduleload	back_bdb.la
 # moduleload	back_ldap.la
 # moduleload	back_ldbm.la
