--- Makefile	2006-10-17 23:56:07.000000000 +0300
+++ Makefile2	2007-07-30 04:48:11.000000000 +0300
@@ -1,7 +1,7 @@
 include Makefile.common
 
-.PHONY: all build-static build-shared check clean distclean
-all: build-static build-shared check
+.PHONY: all build-shared check clean distclean
+all: build-shared check
 build-static:
 	$(MAKE) -C src build-static
 
@@ -19,13 +19,9 @@
 	$(MAKE) -C src distclean
 
 ifneq ($(OS),Windows)
-.PHONY: install install-static
+.PHONY: install
 install:
-	$(MAKE) -C src install-static
 	$(MAKE) -C src install-shared
 	$(MAKE) -C include install
 
-install-static:
-	$(MAKE) -C src install-static
-	$(MAKE) -C include install
 endif
