diff -up ark-4.7.80/kerfuffle/CMakeLists.txt.namelink_skip ark-4.7.80/kerfuffle/CMakeLists.txt
--- ark-4.7.80/kerfuffle/CMakeLists.txt.namelink_skip	2011-11-04 09:36:35.000000000 -0500
+++ ark-4.7.80/kerfuffle/CMakeLists.txt	2011-11-28 14:14:34.361344532 -0600
@@ -33,7 +33,7 @@ endif (NOT WIN32)
 
 set_target_properties(kerfuffle PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION})
 
-install(TARGETS kerfuffle ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kerfuffle ${INSTALL_TARGETS_DEFAULT_ARGS} LIBRARY NAMELINK_SKIP)
 
 install(FILES kerfufflePlugin.desktop DESTINATION ${SERVICETYPES_INSTALL_DIR})
 install(FILES ark.kcfg                DESTINATION ${KCFG_INSTALL_DIR})
