diff -Nua include/opencvWebcam.h /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/include/opencvWebcam.h
--- include/opencvWebcam.h	2009-08-07 05:53:42.000000000 +0300
+++ /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/include/opencvWebcam.h	2011-09-06 16:20:44.060000566 +0300
@@ -1,7 +1,7 @@
 /** @file */
 
-#include "cv.h"
-#include "highgui.h"
+#include "opencv/cv.h"
+#include "opencv/highgui.h"
 #ifndef _INCL_GUARD_WEBCAM
 #define _INCL_GUARD_WEBCAM
 /**
diff -Nua include/qtUtils.h /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/include/qtUtils.h
--- include/qtUtils.h	2009-08-07 05:54:12.000000000 +0300
+++ /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/include/qtUtils.h	2011-09-06 16:20:44.060000566 +0300
@@ -1,8 +1,8 @@
 /** @file */
 
-#include "cv.h"
+#include "opencv/cv.h"
 #include <QtGui>
-#include "highgui.h"
+#include "opencv/highgui.h"
 #ifndef _INCL_QTUTILS
 #define _INCL_QTUTILS
 /**
diff -Nua include/tracker.h /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/include/tracker.h
--- include/tracker.h	2009-08-07 05:56:47.000000000 +0300
+++ /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/include/tracker.h	2011-09-06 16:20:44.059000566 +0300
@@ -1,5 +1,5 @@
 /** @file */
-#include "cv.h"
+#include "opencv/cv.h"
 #ifndef _INCL_GUARD_TRACK
 #define _INCL_GUARD_TRACK
 /**
diff -Nua include/verifier.h /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/include/verifier.h
--- include/verifier.h	2009-08-07 05:56:07.000000000 +0300
+++ /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/include/verifier.h	2011-09-06 16:20:44.060000566 +0300
@@ -1,7 +1,7 @@
 /** @file */
 
-#include "cv.h"
-#include "highgui.h"
+#include "opencv/cv.h"
+#include "opencv/highgui.h"
 #include <unistd.h>
 #include <sys/types.h>
 #include <pwd.h> /* getpwdid */
diff -Nua include/webcamImagePaint.h /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/include/webcamImagePaint.h
--- include/webcamImagePaint.h	2009-08-07 05:54:18.000000000 +0300
+++ /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/include/webcamImagePaint.h	2011-09-06 16:20:44.060000566 +0300
@@ -1,7 +1,7 @@
 /** @file */
 
-#include "cv.h"
-#include "highgui.h"
+#include "opencv/cv.h"
+#include "opencv/highgui.h"
 #ifndef _INCL_GUARD_PAINT
 #define _INCL_GUARD_PAINT
 /**
diff -Nuar src/detector.cpp /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/detector.cpp
--- src/detector.cpp	2009-08-07 05:49:11.000000000 +0300
+++ /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/detector.cpp	2011-09-06 16:20:44.023000566 +0300
@@ -20,7 +20,7 @@
 */
 
 #include "detector.h"
-#include "cv.h"
+#include "opencv/cv.h"
 #include <stdio.h>
 #include <tracker.h>
 #include "utils.h"
diff -Nuar src/eyesDetector.cpp /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/eyesDetector.cpp
--- src/eyesDetector.cpp	2009-08-07 05:49:18.000000000 +0300
+++ /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/eyesDetector.cpp	2011-09-06 16:20:44.022000566 +0300
@@ -21,8 +21,8 @@
 #include "eyesDetector.h"
 #include "pam_face_defines.h"
 #include <stdio.h>
-#include <highgui.h>
-#include <cv.h>
+#include <opencv/highgui.h>
+#include <opencv/cv.h>
 
 double CenterofMass(IplImage* src,int flagXY);
 
diff -Nuar src/faceDetector.cpp /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/faceDetector.cpp
--- src/faceDetector.cpp	2009-08-06 06:45:56.000000000 +0300
+++ /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/faceDetector.cpp	2011-09-06 16:20:44.022000566 +0300
@@ -19,7 +19,7 @@
 */
 
 #include "faceDetector.h"
-#include "cv.h"
+#include "opencv/cv.h"
 #include "tracker.h"
 #include <stdio.h>
 
diff -Nuar src/main.cpp /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/main.cpp
--- src/main.cpp	2009-08-06 06:17:48.000000000 +0300
+++ /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/main.cpp	2011-09-06 16:20:44.023000566 +0300
@@ -23,8 +23,8 @@
 #include <QApplication>
 #include <unistd.h>
 
-#include "cv.h"
-#include "highgui.h"
+#include "opencv/cv.h"
+#include "opencv/highgui.h"
 #include "faceTrainer.h"
 #include "aboutBox.h"
 #include "webcamImagePaint.h"
diff -Nuar src/opencvWebcam.cpp /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/opencvWebcam.cpp
--- src/opencvWebcam.cpp	2009-08-08 03:53:30.000000000 +0300
+++ /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/opencvWebcam.cpp	2011-09-06 16:20:44.023000566 +0300
@@ -18,8 +18,8 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#include "cv.h"
-#include "highgui.h"
+#include "opencv/cv.h"
+#include "opencv/highgui.h"
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -Nuar src/pam_face_authentication.cpp /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/pam_face_authentication.cpp
--- src/pam_face_authentication.cpp	2009-08-08 03:48:50.000000000 +0300
+++ /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/pam_face_authentication.cpp	2011-09-06 16:20:44.023000566 +0300
@@ -34,8 +34,8 @@
 #include <security/_pam_macros.h>
 #include <security/_pam_types.h>
 #include <pwd.h> /* getpwdid */
-#include "cv.h"
-#include "highgui.h"
+#include "opencv/cv.h"
+#include "opencv/highgui.h"
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/ipc.h>
diff -Nuar src/qtUtils.cpp /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/qtUtils.cpp
--- src/qtUtils.cpp	2009-08-07 05:49:55.000000000 +0300
+++ /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/qtUtils.cpp	2011-09-06 16:20:44.023000566 +0300
@@ -21,11 +21,11 @@
 #include <math.h>
 #include <float.h>
 #include <limits.h>
-#include "cv.h"
+#include "opencv/cv.h"
 #include <QtGui>
-#include "highgui.h"
+#include "opencv/highgui.h"
 #include "pam_face_defines.h"
-#include <cxcore.h>
+#include <opencv/cxcore.h>
 #include <stdio.h>
 #include "qtUtils.h"
 
diff -Nuar src/tracker.cpp /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/tracker.cpp
--- src/tracker.cpp	2009-08-07 05:50:09.000000000 +0300
+++ /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/tracker.cpp	2011-09-06 16:20:44.022000566 +0300
@@ -19,7 +19,7 @@
 
 #define NUMBER_OF_GRID_POINTS 25
 #define GRID_SIDE_SIZE 5
-#include "cv.h"
+#include "opencv/cv.h"
 #include <stdio.h>
 #include "tracker.h"
 tracker::tracker(void)
diff -Nuar src/utils.cpp /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/utils.cpp
--- src/utils.cpp	2009-08-07 05:50:19.000000000 +0300
+++ /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/utils.cpp	2011-09-06 16:20:44.023000566 +0300
@@ -21,11 +21,11 @@
 #include <math.h>
 #include <float.h>
 #include <limits.h>
-#include "cv.h"
-#include "highgui.h"
+#include "opencv/cv.h"
+#include "opencv/highgui.h"
 
 #include "pam_face_defines.h"
-#include <cxcore.h>
+#include <opencv/cxcore.h>
 #include <stdio.h>
 
 #include "utils.h"
diff -Nuar src/webcamImagePaint.cpp /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/webcamImagePaint.cpp
--- src/webcamImagePaint.cpp	2009-08-07 05:50:34.000000000 +0300
+++ /var/pisi/pam_face_authentication-changed/work/pam-face-authentication-0.3/src/webcamImagePaint.cpp	2011-09-06 16:20:44.023000566 +0300
@@ -18,8 +18,8 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 #include <stdio.h>
-#include "cv.h"
-#include "highgui.h"
+#include "opencv/cv.h"
+#include "opencv/highgui.h"
 #include "webcamImagePaint.h"
 
 webcamImagePaint::webcamImagePaint()
--- include/detector.h.orig	2011-09-06 16:45:29.129000566 +0300
+++ include/detector.h	2011-09-06 16:45:49.298000566 +0300
@@ -1,6 +1,6 @@
 /** @file */
 
-#include "cv.h"
+#include "opencv/cv.h"
 #ifndef _INCL_GUARD_DET
 #define _INCL_GUARD_DET
 #include "tracker.h"
--- include/faceDetector.h.orig	2011-09-06 16:47:37.731000564 +0300
+++ include/faceDetector.h	2011-09-06 16:47:54.780000564 +0300
@@ -1,7 +1,7 @@
 /** @file */
 
-#include "cv.h"
-#include "highgui.h"
+#include "opencv/cv.h"
+#include "opencv/highgui.h"
 /**
 * Face Structure, Information that can describe the detected Face on the Image
 */
--- include/eyesDetector.h.orig	2011-09-06 16:50:58.445000566 +0300
+++ include/eyesDetector.h	2011-09-06 16:51:13.486000565 +0300
@@ -1,6 +1,6 @@
 /** @file */
 
-#include "cv.h"
+#include "opencv/cv.h"
 
 /**
 * Eye Structure, Information that can describe the detected Eye on the Face Image
