diff -Naur configure configure
--- configure	2012-02-06 11:16:07.000000000 +0100
+++ configure	2012-05-10 13:29:24.604000013 +0200
@@ -1439,7 +1439,7 @@
 if test -n "$MACOSX_APP" ; then
     INSTDIR='${prefix}'"/GRASS-${GRASS_VERSION_MAJOR}.${GRASS_VERSION_MINOR}.app/Contents/MacOS"
 else
-    INSTDIR='${prefix}'"/grass-${GRASS_VERSION_MAJOR}.${GRASS_VERSION_MINOR}.${GRASS_VERSION_RELEASE}"
+    INSTDIR='${prefix}'"/lib/grass"${GRASS_VERSION_MAJOR}${GRASS_VERSION_MINOR}
 fi
 
 
