diff -ur libpgf.orig/libpgf.pc.in libpgf/libpgf.pc.in
--- libpgf.orig/libpgf.pc.in	2009-09-28 14:53:22.000000000 +0300
+++ libpgf/libpgf.pc.in	2009-12-29 21:18:28.000000000 +0200
@@ -1,7 +1,7 @@
 prefix=@prefix@
-exec_prefix=@prefix@
-libdir=@prefix@/lib
-includedir=@prefix@/include
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@prefix@/include/@PACKAGE@
 
 Name: @PACKAGE@
 Description: libpgf - Progressive Graphics File (PGF) library
