diff -Naur igerman98-20090107-orig/Makefile igerman98-20090107/Makefile
--- igerman98-20090107-orig/Makefile	2009-03-19 16:55:26.000000000 +0200
+++ igerman98-20090107/Makefile	2009-03-19 16:55:53.000000000 +0200
@@ -75,7 +75,7 @@
 ZIP_OPTS = a -tzip -mx=9
 BUILDHASH = true
 
-ISPELL = ispell
+ISPELL = ispell-aspell
 HUNSPELL_BIN = hunspell
 # Hunspell starting with 1.1.9 (?) uses automatic charset transcoding,
 # we need to make sure we use latin1:
