--- pidgin-2.10.7/pidgin/plugins/cap/cap.c~	2013-02-11 04:16:54.000000000 -0500
+++ pidgin-2.10.7/pidgin/plugins/cap/cap.c	2013-02-14 10:44:27.743029516 -0500
@@ -43,7 +43,7 @@
 	int threshold = purple_prefs_get_int("/plugins/gtk/cap/threshold");
 	int min_minute = (current_minute - threshold) % 1440;
 	int max_minute = (current_minute + threshold) % 1440;
-	char *sql, sta_id = NULL;
+	char *sql, *sta_id = NULL;
 	sqlite3_stmt *stmt = NULL;
 	const char *tail = NULL;
 	int rc;
