--- conky-1.9.0/src/conky.c.orig	2012-05-04 00:22:21.000000000 +0300
+++ conky-1.9.0/src/conky.c	2012-11-07 03:10:06.000000000 +0200
@@ -2643,7 +2643,14 @@
 					w += specials[special_index].arg;
 				}
 			} else if (specials[special_index].type == VOFFSET) {
-				last_font_height += specials[special_index].arg;
+				if (specials[special_index].arg > 0)
+					last_font_height += specials[special_index].arg;
+				else
+				/* 
+				* reset negative voffset, otherwise it gets
+				* added as a positive value to window size
+				*/
+					last_font_height = 0;
 			} else if (specials[special_index].type == GOTO) {
 				if (specials[special_index].arg > cur_x) {
 					w = (int) specials[special_index].arg;
@@ -2658,7 +2665,7 @@
 				w += step - (cur_x - text_start_x - start) % step;
 			} else if (specials[special_index].type == FONT) {
 				selected_font = specials[special_index].font_added;
-				if (font_height() > last_font_height) {
+				if ( (font_height() > last_font_height) && (last_font_height > 0) ) {
 					last_font_height = font_height();
 				}
 			}
