diff -Nuar hugin-2010.4.0.orig//src/hugin1/hugin/hugin.desktop hugin-2010.4.0/src/hugin1/hugin/hugin.desktop
--- hugin-2010.4.0.orig//src/hugin1/hugin/hugin.desktop	2010-12-28 14:57:50.000000000 +0200
+++ hugin-2010.4.0/src/hugin1/hugin/hugin.desktop	2011-04-15 14:45:59.909000022 +0300
@@ -6,6 +6,7 @@
 Name[it]=Hugin - Assemblatore di panorami 
 Name[ru]=Сшиватель панорам Hugin
 Name[hu]=Hugin panoráma készítő
+Name=Hugin
 Comment=Stitch photographs together
 Comment[cs]=Skládání panoramatických snímků
 Comment[de]=Erzeugt Panoramabildern aus mehreren Einzelbildern.
@@ -20,6 +21,7 @@
 GenericName[cs]=Nástroj pro skládání panoramat
 GenericName[ru]=Сшиватель панорам
 GenericName[hu]=Panoráma összefűző
+GenericName[tr]=Panorama oluşturucu
 Categories=Graphics;
 Exec=hugin
 Icon=hugin
diff -Nuar hugin-2010.4.0.orig//src/hugin1/ptbatcher/PTBatcherGUI.desktop hugin-2010.4.0/src/hugin1/ptbatcher/PTBatcherGUI.desktop
--- hugin-2010.4.0.orig//src/hugin1/ptbatcher/PTBatcherGUI.desktop	2010-12-28 14:57:50.000000000 +0200
+++ hugin-2010.4.0/src/hugin1/ptbatcher/PTBatcherGUI.desktop	2011-04-15 14:47:39.069000022 +0300
@@ -5,6 +5,7 @@
 Name[it]=Hugin - Elaboratore in serie
 Name[de]=Hugin - Stapelverarbeitung
 Name[hu]=Hugin kötegelt feldolgozó
+Name[tr]=Hugin Toplu İşleyici
 Comment=Hugin project stitching queue manager
 Comment[fr]=Gérer la file d'attente des projets Hugin d'assemblage
 Comment[it]=Gestisce la coda dei progetti di Hugin
@@ -16,6 +17,7 @@
 GenericName[it]=Interfaccia grafica per l'elaborazione in serie dei progetti di Hugin
 GenericName[de]=Panorama Stapelverarbeitung
 GenericName[hu]=Panoráma kötegelt feldolgozó GUI
+GenericName[tr]=Toplu panorama görüntüsü işleyici
 Categories=Graphics;
 Exec=PTBatcherGUI %F
 Icon=ptbatcher
