diff -Nur glibc-20080408T0706-old/Makefile glibc-20080408T0706/Makefile
--- glibc-20080408T0706-old/Makefile	2008-11-14 18:53:21.000000000 +0200
+++ glibc-20080408T0706/Makefile	2008-11-14 18:55:38.000000000 +0200
@@ -115,9 +115,6 @@
 	rm -f $(symbolic-link-list)
 
 install:
-	-test ! -x $(common-objpfx)elf/ldconfig || LC_ALL=C LANGUAGE=C \
-	  $(common-objpfx)elf/ldconfig $(addprefix -r ,$(install_root)) \
-				       $(slibdir) $(libdir)
 ifneq (no,$(PERL))
 ifeq (/usr,$(prefix))
 ifeq (,$(install_root))
