diff -up gocr-0.45/man/Makefile.in.perms gocr-0.45/man/Makefile.in
--- gocr-0.45/man/Makefile.in.perms	2005-05-18 10:49:11.000000000 +0200
+++ gocr-0.45/man/Makefile.in	2008-01-28 02:04:04.000000000 +0100
@@ -17,7 +17,7 @@ all:
 .PHONY : clean proper install uninstall
 install: all
 	$(INSTALL) -d $(DESTDIR)$(mandir)/man1
-	$(INSTALL) ../man/man1/$(PROGRAM).1 $(DESTDIR)$(mandir)/man1
+	$(INSTALL) -m 0644 ../man/man1/$(PROGRAM).1 $(DESTDIR)$(mandir)/man1
 
 uninstall:
 	-rm -f $(DESTDIR)$(mandir)/man1/$(PROGRAM).1
