--- opencity.desktop.orig	2007-10-28 20:56:57.000000000 +0200
+++ opencity.desktop	2007-12-11 20:34:02.000000000 +0200
@@ -3,8 +3,10 @@
 Encoding=UTF-8
 Comment=A 3D city simulator
 Comment[fr]=Jeu de simulation de Ville en 3D
+Comment[tr]=3B Şehir Simülasyonu
 Categories=Game;StrategyGame;
 Exec=opencity
 GenericName=City simulator
 GenericName[fr]=Simulateur de ville
+GenericName[tr]=Şehir Simülatörü
 Icon=opencity.png
