--- opencity-0.0.5stable.orig/config/Makefile.in	2007-12-01 23:12:47.000000000 +0200
+++ opencity-0.0.5stable/config/Makefile.in	2007-12-11 19:56:41.000000000 +0200
@@ -79,7 +79,7 @@
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(CFLAGS) @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
--- opencity-0.0.5stable.orig/configure	2007-12-01 23:12:51.000000000 +0200
+++ opencity-0.0.5stable/configure	2007-12-11 20:00:32.000000000 +0200
@@ -4903,7 +4903,7 @@
 if test "x$enable_debug" = "xyes"; then
         enable_debug="$enable_gdb"
 else
-        enable_debug="-DNDEBUG"
+        enable_debug=""
 fi
 
 
--- opencity-0.0.5stable.orig/docs/Makefile.in	2007-12-01 23:16:54.000000000 +0200
+++ opencity-0.0.5stable/docs/Makefile.in	2007-12-11 19:56:41.000000000 +0200
@@ -81,7 +81,7 @@
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(CFLAGS) @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
--- opencity-0.0.5stable.orig/Makefile.in	2007-12-01 23:12:51.000000000 +0200
+++ opencity-0.0.5stable/Makefile.in	2007-12-11 19:56:41.000000000 +0200
@@ -103,7 +103,7 @@
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(CFLAGS) @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
--- opencity-0.0.5stable.orig/src/binreloc/Makefile.in	2007-12-01 23:12:47.000000000 +0200
+++ opencity-0.0.5stable/src/binreloc/Makefile.in	2007-12-11 19:56:41.000000000 +0200
@@ -90,7 +90,7 @@
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(CFLAGS) @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
--- opencity-0.0.5stable.orig/src/enum/Makefile.in	2007-12-01 23:12:48.000000000 +0200
+++ opencity-0.0.5stable/src/enum/Makefile.in	2007-12-11 19:56:41.000000000 +0200
@@ -73,7 +73,7 @@
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(CFLAGS) @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
--- opencity-0.0.5stable.orig/src/Makefile.in	2007-12-01 23:12:47.000000000 +0200
+++ opencity-0.0.5stable/src/Makefile.in	2007-12-11 19:56:41.000000000 +0200
@@ -136,7 +136,7 @@
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(CFLAGS) @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
--- opencity-0.0.5stable.orig/src/mapgen/Makefile.in	2007-12-01 23:12:48.000000000 +0200
+++ opencity-0.0.5stable/src/mapgen/Makefile.in	2007-12-11 19:56:41.000000000 +0200
@@ -102,7 +102,7 @@
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(CFLAGS) @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
--- opencity-0.0.5stable.orig/src/mas/Makefile.in	2007-12-01 23:12:49.000000000 +0200
+++ opencity-0.0.5stable/src/mas/Makefile.in	2007-12-11 19:56:41.000000000 +0200
@@ -93,7 +93,7 @@
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(CFLAGS) @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
--- opencity-0.0.5stable.orig/src/model/Makefile.in	2007-12-01 23:12:49.000000000 +0200
+++ opencity-0.0.5stable/src/model/Makefile.in	2007-12-11 19:56:41.000000000 +0200
@@ -93,7 +93,7 @@
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(CFLAGS) @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
--- opencity-0.0.5stable.orig/src/networking/Makefile.in	2007-12-01 23:12:49.000000000 +0200
+++ opencity-0.0.5stable/src/networking/Makefile.in	2007-12-11 19:56:41.000000000 +0200
@@ -91,7 +91,7 @@
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(CFLAGS) @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
--- opencity-0.0.5stable.orig/src/simulator/Makefile.in	2007-12-01 23:12:49.000000000 +0200
+++ opencity-0.0.5stable/src/simulator/Makefile.in	2007-12-11 19:56:41.000000000 +0200
@@ -94,7 +94,7 @@
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(CFLAGS) @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
--- opencity-0.0.5stable.orig/src/structure/Makefile.in	2007-12-01 23:12:50.000000000 +0200
+++ opencity-0.0.5stable/src/structure/Makefile.in	2007-12-11 19:56:41.000000000 +0200
@@ -93,7 +93,7 @@
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(CFLAGS) @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
--- opencity-0.0.5stable.orig/src/tinyxml/Makefile.in	2007-12-01 23:12:50.000000000 +0200
+++ opencity-0.0.5stable/src/tinyxml/Makefile.in	2007-12-11 19:56:41.000000000 +0200
@@ -92,7 +92,7 @@
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(CFLAGS) @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
--- opencity-0.0.5stable.orig/src/tinyxpath/Makefile.in	2007-12-01 23:12:50.000000000 +0200
+++ opencity-0.0.5stable/src/tinyxpath/Makefile.in	2007-12-11 19:56:41.000000000 +0200
@@ -95,7 +95,7 @@
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(CFLAGS) @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
--- opencity-0.0.5stable.orig/src/triangulation/Makefile.in	2007-12-01 23:12:50.000000000 +0200
+++ opencity-0.0.5stable/src/triangulation/Makefile.in	2007-12-11 19:56:41.000000000 +0200
@@ -91,7 +91,7 @@
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(CFLAGS) @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
--- opencity-0.0.5stable.orig/viewer/Makefile.in	2007-12-01 23:12:50.000000000 +0200
+++ opencity-0.0.5stable/viewer/Makefile.in	2007-12-11 19:56:41.000000000 +0200
@@ -97,7 +97,7 @@
 CXX = @CXX@
 CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = $(CFLAGS) @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
