diff -ur gnome-doc-utils-0.18.1.orig/tools/gnome-doc-utils.make gnome-doc-utils-0.18.1/tools/gnome-doc-utils.make
--- gnome-doc-utils-0.18.1.orig/tools/gnome-doc-utils.make	2009-11-19 18:54:47.000000000 +0200
+++ gnome-doc-utils-0.18.1/tools/gnome-doc-utils.make	2010-01-22 20:11:57.324726926 +0200
@@ -597,7 +597,6 @@
 	done
 	@if test "x$(_ENABLE_SK)" = "xtrue"; then \
 	  echo "scrollkeeper-update -p $(DESTDIR)$(_sklocalstatedir) -o $(DESTDIR)$(OMF_DIR)/$(_doc_install_dir)"; \
-	  scrollkeeper-update -p "$(DESTDIR)$(_sklocalstatedir)" -o "$(DESTDIR)$(OMF_DIR)/$(_doc_install_dir)"; \
 	fi;
 
 install-doc-dsk:
