Index: speech-dispatcher/config/speechd.conf.in
===================================================================
--- speech-dispatcher.orig/config/speechd.conf.in	2010-06-28 10:35:49.000000000 +0200
+++ speech-dispatcher/config/speechd.conf.in	2010-06-28 10:36:53.000000000 +0200
@@ -166,7 +166,7 @@
 # PulseAudio is a sound server for POSIX and WIN32 systems. 
 #
 
-# AudioOutputMethod "pulse,alsa"
+# AudioOutputMethod "pulse"
 
 # What ALSA device to use when Advanced Linux Sound Architecture is
 # chosen for the audio output.
Index: speech-dispatcher/src/server/config.c
===================================================================
--- speech-dispatcher.orig/src/server/config.c	2010-06-28 10:40:14.000000000 +0200
+++ speech-dispatcher/src/server/config.c	2010-06-28 10:40:29.000000000 +0200
@@ -454,7 +454,7 @@
 /* Added by Willie Walker - default to OSS for Solaris */
     GlobalFDSet.audio_output_method = strdup("oss");
 #else
-    GlobalFDSet.audio_output_method = strdup("pulse,alsa");
+    GlobalFDSet.audio_output_method = strdup("pulse");
 #endif /* __SUNPRO_C */
     GlobalFDSet.audio_oss_device = strdup("/dev/dsp");
     GlobalFDSet.audio_alsa_device = strdup("default");
