--- data/default.layout	2014-02-18 20:09:23.000000000 +0200
+++ data/pisi.layout	2014-04-03 21:38:36.000000000 +0300
@@ -1,52 +1,48 @@
-[Toplevel top]
-expand=true
-orientation=top
-size=24
-
 [Toplevel bottom]
 expand=true
 orientation=bottom
 size=24
 
 [Object menu-bar]
-object-type=menu-bar
-toplevel-id=top
+object-type=applet
+applet-iid=GNOMEMainMenuFactory::GNOMEMainMenu
+toplevel-id=bottom
 position=0
 locked=true
 
-[Object notification-area]
+[Object stickynotes]
 object-type=applet
-applet-iid=NotificationAreaAppletFactory::NotificationArea
-toplevel-id=top
+applet-iid=StickyNotesAppletFactory::StickyNotesApplet
+toplevel-id=bottom
 position=10
-panel-right-stick=true
 locked=true
 
-[Object clock]
+[Object window-list]
 object-type=applet
-applet-iid=ClockAppletFactory::ClockApplet
-toplevel-id=top
-position=0
-panel-right-stick=true
+applet-iid=WnckletFactory::WindowListApplet
+toplevel-id=bottom
+position=20
 locked=true
 
-[Object show-desktop]
+[Object notification-area]
 object-type=applet
-applet-iid=WnckletFactory::ShowDesktopApplet
+applet-iid=NotificationAreaAppletFactory::NotificationArea
 toplevel-id=bottom
-position=0
+position=10
+panel-right-stick=true
 locked=true
 
-[Object window-list]
+[Object workspace-switcher]
 object-type=applet
-applet-iid=WnckletFactory::WindowListApplet
+applet-iid=WnckletFactory::WorkspaceSwitcherApplet
 toplevel-id=bottom
-position=20
+position=5
+panel-right-stick=true
 locked=true
 
-[Object workspace-switcher]
+[Object clock]
 object-type=applet
-applet-iid=WnckletFactory::WorkspaceSwitcherApplet
+applet-iid=ClockAppletFactory::ClockApplet
 toplevel-id=bottom
 position=0
 panel-right-stick=true
