--- data/default.layout	2014-06-07 07:42:15.417844985 +0300
+++ data/pisi.layout	2014-06-10 18:33:36.431919679 +0300
@@ -1,40 +1,20 @@
-[Toplevel top]
-expand=true
-orientation=top
-size=24
-
 [Toplevel bottom]
 expand=true
 orientation=bottom
 size=24
 
 [Object menu-bar]
-object-type=menu-bar
-toplevel-id=top
-position=0
-locked=true
-
-[Object notification-area]
-object-type=applet
-applet-iid=NotificationAreaAppletFactory::NotificationArea
-toplevel-id=top
-position=10
-panel-right-stick=true
-locked=true
-
-[Object clock]
 object-type=applet
-applet-iid=ClockAppletFactory::ClockApplet
-toplevel-id=top
+applet-iid=GNOMEMainMenuFactory::GNOMEMainMenu
+toplevel-id=bottom
 position=0
-panel-right-stick=true
 locked=true
 
-[Object show-desktop]
-object-type=applet
-applet-iid=WnckletFactory::ShowDesktopApplet
+[Object web-browser]
+object-type=launcher
+launcher-location=/usr/share/applications/mozillafirefox.desktop
 toplevel-id=bottom
-position=0
+position=10
 locked=true
 
 [Object window-list]
@@ -44,6 +24,14 @@
 position=20
 locked=true
 
+[Object notification-area]
+object-type=applet
+applet-iid=NotificationAreaAppletFactory::NotificationArea
+toplevel-id=bottom
+position=10
+panel-right-stick=true
+locked=true
+
 [Object workspace-switcher]
 object-type=applet
 applet-iid=WnckletFactory::WorkspaceSwitcherApplet
@@ -51,3 +39,11 @@
 position=0
 panel-right-stick=true
 locked=true
+
+[Object clock]
+object-type=applet
+applet-iid=ClockAppletFactory::ClockApplet
+toplevel-id=bottom
+position=10
+panel-right-stick=true
+locked=true
