--- data/panel-default-layout.mate.OSmanOS	2013-12-13 23:24:19.154976000 +0200
+++ data/panel-default-layout.mate	        2013-12-13 23:29:50.125967110 +0200
@@ -14,6 +14,27 @@
 position=0
 locked=true
 
+[Object file-browser]
+object-type=launcher
+launcher-location=/usr/share/applications/caja-browser.desktop
+toplevel-id=top
+position=10
+locked=true
+
+[Object terminal]
+object-type=launcher
+launcher-location=/usr/share/applications/mate-terminal.desktop
+toplevel-id=top
+position=20
+locked=true
+
+[Object web-browser]
+object-type=launcher
+launcher-location=/usr/share/applications/mozillafirefox.desktop
+toplevel-id=top
+position=30
+locked=true
+
 [Object notification-area]
 object-type=applet
 applet-iid=NotificationAreaAppletFactory::NotificationArea
