--- mate-panel-1.5.3/./mate-panel/panel-menu-bar.c~	2012-12-20 00:48:49.000000000 +0200
+++ mate-panel-1.5.3/./mate-panel/panel-menu-bar.c	2013-01-16 12:07:54.282974605 +0200
@@ -116,7 +116,7 @@
 	menubar->priv->info = NULL;
 
 
-	menubar->priv->applications_menu = create_applications_menu("mate-applications.menu", NULL, TRUE);
+	menubar->priv->applications_menu = create_applications_menu("applications.menu", NULL, TRUE);
 
 	menubar->priv->applications_item = panel_image_menu_item_new();
 	gtk_menu_item_set_label(GTK_MENU_ITEM(menubar->priv->applications_item), _("Applications"));
--- mate-panel-1.5.3/./mate-panel/menu.c~	2012-12-20 00:48:49.000000000 +0200
+++ mate-panel-1.5.3/./mate-panel/menu.c	2013-01-16 12:08:04.950133393 +0200
@@ -1793,7 +1793,7 @@
 {
 	GtkWidget* main_menu;
 
-	main_menu = create_applications_menu("mate-applications.menu", NULL, TRUE);
+	main_menu = create_applications_menu("applications.menu", NULL, TRUE);
 
 	g_object_set_data(G_OBJECT(main_menu), "menu_panel", panel);
 	/* FIXME need to update the panel on parent_set */
--- mate-panel-1.5.3/./mate-panel/panel-menu-button.c~	2012-12-20 00:48:49.000000000 +0200
+++ mate-panel-1.5.3/./mate-panel/panel-menu-button.c	2013-01-16 12:08:17.590716557 +0200
@@ -77,7 +77,7 @@
 } MenuPathRootItem;
 
 static MenuPathRootItem root_items[] = {
-	{APPLICATIONS_MENU, "mate-applications", "mate-applications.menu"},
+	{APPLICATIONS_MENU, "applications", "applications.menu"},
 	{SETTINGS_MENU, "mate-settings", "mate-settings.menu"}
 };
 
--- mate-panel-1.5.3/./mate-panel/panel-addto.c~	2012-12-20 00:48:50.000000000 +0200
+++ mate-panel-1.5.3/./mate-panel/panel-addto.c	2013-01-16 12:08:31.741369388 +0200
@@ -692,12 +692,12 @@
 
 	store = gtk_tree_store_new(NUMBER_COLUMNS, GDK_TYPE_PIXBUF, G_TYPE_STRING, G_TYPE_POINTER, G_TYPE_STRING);
 
-	tree = matemenu_tree_lookup("mate-applications.menu", MATEMENU_TREE_FLAGS_NONE);
+	tree = matemenu_tree_lookup("applications.menu", MATEMENU_TREE_FLAGS_NONE);
 	matemenu_tree_set_sort_key(tree, MATEMENU_TREE_SORT_DISPLAY_NAME);
 
 	if ((root = matemenu_tree_get_root_directory (tree)))
 	{
-		panel_addto_make_application_list(&dialog->application_list, root, "mate-applications.menu");
+		panel_addto_make_application_list(&dialog->application_list, root, "applications.menu");
 		panel_addto_populate_application_model(store, NULL, dialog->application_list);
 
 		matemenu_tree_item_unref(root);
--- mate-panel-1.5.3/./mate-panel/panel-run-dialog.c~	2012-12-20 00:48:50.000000000 +0200
+++ mate-panel-1.5.3/./mate-panel/panel-run-dialog.c	2013-01-16 12:08:41.471818295 +0200
@@ -857,7 +857,7 @@
 	MateMenuTreeDirectory* root;
 	GSList* retval;
 
-	tree = matemenu_tree_lookup("mate-applications.menu", MATEMENU_TREE_FLAGS_NONE);
+	tree = matemenu_tree_lookup("applications.menu", MATEMENU_TREE_FLAGS_NONE);
 	matemenu_tree_set_sort_key(tree, MATEMENU_TREE_SORT_DISPLAY_NAME);
 
 	root = matemenu_tree_get_root_directory(tree);
