diff -up ebook-tools-0.2.1/src/libepub/CMakeLists.txt.rmhardcoded ebook-tools-0.2.1/src/libepub/CMakeLists.txt
--- ebook-tools-0.2.1/src/libepub/CMakeLists.txt.rmhardcoded	2010-09-13 14:31:53.000000000 -0500
+++ ebook-tools-0.2.1/src/libepub/CMakeLists.txt	2012-07-10 14:03:37.417272126 -0500
@@ -4,5 +4,5 @@ target_link_libraries (epub ${LIBZIP_LIB
 
 set_target_properties (epub PROPERTIES VERSION 0.2.1 SOVERSION 0)
 
-install ( TARGETS epub RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib )
+install ( TARGETS epub RUNTIME DESTINATION bin LIBRARY DESTINATION lib${LIB_SUFFIX} ARCHIVE DESTINATION lib${LIB_SUFFIX} )
 install ( FILES epub.h epub_shared.h epub_version.h DESTINATION include )
