diff -Nru SQUIRREL2.orig/autogen.sh SQUIRREL2/autogen.sh
--- SQUIRREL2.orig/autogen.sh	1970-01-01 01:00:00.000000000 +0100
+++ SQUIRREL2/autogen.sh	2007-07-07 19:06:16.000000000 +0200
@@ -0,0 +1,7 @@
+#! /bin/sh
+
+libtoolize --copy
+aclocal
+autoheader
+automake --add-missing --copy --foreign
+autoconf
diff -Nru SQUIRREL2.orig/configure.ac SQUIRREL2/configure.ac
--- SQUIRREL2.orig/configure.ac	1970-01-01 01:00:00.000000000 +0100
+++ SQUIRREL2/configure.ac	2007-07-07 20:06:22.000000000 +0200
@@ -0,0 +1,26 @@
+## Bootstrap autoconf/automake
+AC_PREREQ(2.59)
+AC_INIT([squirrel], [2.2.4], [])
+AC_CANONICAL_TARGET
+AC_CONFIG_SRCDIR([configure.ac])
+AM_INIT_AUTOMAKE
+AM_CONFIG_HEADER([config.h])
+
+## Checks for programs.
+AC_PROG_CC
+AC_PROG_CXX
+AM_PROG_CC_STDC
+AC_PROG_INSTALL
+AC_PROG_MAKE_SET
+AC_PROG_LIBTOOL
+
+# Produce output
+AC_CONFIG_FILES([Makefile
+		squirrel/Makefile
+		sqstdlib/Makefile
+		sq/Makefile
+		include/Makefile
+		])
+AC_OUTPUT
+
+echo "Type 'make' to compile"
diff -Nru SQUIRREL2.orig/include/Makefile.am SQUIRREL2/include/Makefile.am
--- SQUIRREL2.orig/include/Makefile.am	1970-01-01 01:00:00.000000000 +0100
+++ SQUIRREL2/include/Makefile.am	2007-07-07 20:08:47.000000000 +0200
@@ -0,0 +1,11 @@
+MAINTAINERCLEANFILES = Makefile.in
+
+sqdir = $(includedir)/squirrel
+
+sq_HEADERS = sqstdaux.h \
+		sqstdblob.h \
+		sqstdio.h \
+		sqstdmath.h \
+		sqstdstring.h \
+		sqstdsystem.h \
+		squirrel.h
diff -Nru SQUIRREL2.orig/Makefile SQUIRREL2/Makefile
--- SQUIRREL2.orig/Makefile	2006-03-05 17:03:53.000000000 +0100
+++ SQUIRREL2/Makefile	1970-01-01 01:00:00.000000000 +0100
@@ -1,18 +0,0 @@
-
-SQUIRREL=.
-MAKE=make
-
-sq32:
-	cd squirrel; $(MAKE) 
-	cd sqstdlib; $(MAKE) 
-	cd sq; $(MAKE) 
-
-sqprof:
-	cd squirrel; $(MAKE) sqprof
-	cd sqstdlib; $(MAKE) sqprof
-	cd sq; $(MAKE) sqprof
-
-sq64:
-	cd squirrel; $(MAKE) sq64
-	cd sqstdlib; $(MAKE) sq64
-	cd sq; $(MAKE) sq64
diff -Nru SQUIRREL2.orig/Makefile.am SQUIRREL2/Makefile.am
--- SQUIRREL2.orig/Makefile.am	1970-01-01 01:00:00.000000000 +0100
+++ SQUIRREL2/Makefile.am	2007-07-07 20:05:54.000000000 +0200
@@ -0,0 +1,18 @@
+SUBDIRS = squirrel sqstdlib sq include
+
+MAINTAINERCLEANFILES =	INSTALL \
+			Makefile.in \
+			aclocal.m4 \
+			config.guess \
+			config.h.in \
+			config.h.in~ \
+			stamp-h.in \
+			config.sub \
+			configure \
+			install-sh \
+			missing \
+			mkinstalldirs \
+			ltmain.sh \
+			ltconfig \
+			compile \
+			depcomp
diff -Nru SQUIRREL2.orig/sq/Makefile SQUIRREL2/sq/Makefile
--- SQUIRREL2.orig/sq/Makefile	2006-03-06 18:01:41.000000000 +0100
+++ SQUIRREL2/sq/Makefile	1970-01-01 01:00:00.000000000 +0100
@@ -1,21 +0,0 @@
-SQUIRREL= ..
-
-
-OUT= $(SQUIRREL)/bin/sq
-INCZ= -I$(SQUIRREL)/include -I. -I$(SQUIRREL)/sqlibs
-LIBZ= -L$(SQUIRREL)/lib 
-LIB= -lsquirrel -lsqstdlib
-
-OBJS= sq.o
-	
-SRCS= sq.c
-	
-	
-sq32:
-	g++ -O2 -fno-rtti -o $(OUT) $(SRCS) $(INCZ) $(LIBZ) $(LIB)
-
-sqprof:
-	g++ -O2 -pg -fno-rtti -pie -gstabs -g3 -o $(OUT) $(SRCS) $(INCZ) $(LIBZ) $(LIB)
-	
-sq64:
-	g++ -O2 -fno-rtti -D_SQ64 -o $(OUT) $(SRCS) $(INCZ) $(LIBZ) $(LIB)
\ Chybí znak konce řádku na konci souboru
diff -Nru SQUIRREL2.orig/sq/Makefile.am SQUIRREL2/sq/Makefile.am
--- SQUIRREL2.orig/sq/Makefile.am	1970-01-01 01:00:00.000000000 +0100
+++ SQUIRREL2/sq/Makefile.am	2007-07-07 19:24:54.000000000 +0200
@@ -0,0 +1,11 @@
+MAINTAINERCLEANFILES = Makefile.in
+
+AM_CXXFLAGS = -fno-rtti -I$(top_srcdir)/include -I.
+
+bin_PROGRAMS = sq
+
+sq_LDADD = $(top_builddir)/sqstdlib/libsqstdlib.la $(top_builddir)/squirrel/libsquirrel.la
+
+sq_SOURCES = sq.cpp
+
+INCLUDES = -I../include
diff -Nru SQUIRREL2.orig/sqstdlib/Makefile SQUIRREL2/sqstdlib/Makefile
--- SQUIRREL2.orig/sqstdlib/Makefile	2006-03-06 18:01:50.000000000 +0100
+++ SQUIRREL2/sqstdlib/Makefile	1970-01-01 01:00:00.000000000 +0100
@@ -1,30 +0,0 @@
-SQUIRREL= ..
-
-
-OUT= $(SQUIRREL)/lib/libsqstdlib.a
-INCZ= -I$(SQUIRREL)/include -I. -Iinclude
-
-SRCS= \
-	sqstdblob.cpp \
-	sqstdio.cpp \
-	sqstdstream.cpp \
-	sqstdmath.cpp \
-	sqstdsystem.cpp \
-	sqstdstring.cpp \
-	sqstdaux.cpp \
-	sqstdrex.cpp
-	
-	
-sq32:
-	gcc -O2  -fno-rtti -Wall -c $(SRCS) $(INCZ)
-	ar rc $(OUT) *.o
-	rm *.o
-
-sqprof:
-	gcc -O2 -pg -fno-rtti -pie -gstabs -g3 -Wall -c $(SRCS) $(INCZ)
-	ar rc $(OUT) *.o
-	rm *.o
-sq64:
-	gcc -O2 -D_SQ64 -fno-rtti -Wall -c $(SRCS) $(INCZ)
-	ar rc $(OUT) *.o
-	rm *.o
diff -Nru SQUIRREL2.orig/sqstdlib/Makefile.am SQUIRREL2/sqstdlib/Makefile.am
--- SQUIRREL2.orig/sqstdlib/Makefile.am	1970-01-01 01:00:00.000000000 +0100
+++ SQUIRREL2/sqstdlib/Makefile.am	2007-07-07 19:24:48.000000000 +0200
@@ -0,0 +1,20 @@
+MAINTAINERCLEANFILES = Makefile.in
+AM_CXXFLAGS = -fno-rtti
+
+lib_LTLIBRARIES = libsqstdlib.la
+
+libsqstdlib_la_LDFLAGS = -release $(VERSION)
+
+libsqstdlib_la_SOURCES = \
+			sqstdaux.cpp \
+			sqstdblob.cpp \
+			sqstdblobimpl.h \
+			sqstdio.cpp \
+			sqstdmath.cpp \
+			sqstdrex.cpp \
+			sqstdstream.cpp \
+			sqstdstream.h \
+			sqstdstring.cpp \
+			sqstdsystem.cpp
+
+INCLUDES = -I$(top_srcdir)/include -I.
diff -Nru SQUIRREL2.orig/squirrel/Makefile SQUIRREL2/squirrel/Makefile
--- SQUIRREL2.orig/squirrel/Makefile	2006-03-06 18:01:31.000000000 +0100
+++ SQUIRREL2/squirrel/Makefile	1970-01-01 01:00:00.000000000 +0100
@@ -1,52 +0,0 @@
-SQUIRREL= ..
-
-
-OUT= $(SQUIRREL)/lib/libsquirrel.a
-INCZ= -I$(SQUIRREL)/include -I. -Iinclude
-DEFS= 
-LIB=	
-
-OBJS= \
-	sqapi.o \
-	sqbaselib.o \
-	sqcompiler.o \
-	sqdebug.o \
-	sqlexer.o \
-	sqobject.o \
-	sqparser.o \
-	sqstate.o \
-	sqtable.o \
-	sqvm.o \
-	sqmem.o \
-	sqclass.o
-	
-SRCS= \
-	sqapi.cpp \
-	sqbaselib.cpp \
-	sqfuncstate.cpp \
-	sqdebug.cpp \
-	sqlexer.cpp \
-	sqobject.cpp \
-	sqcompiler.cpp \
-	sqstate.cpp \
-	sqtable.cpp \
-	sqmem.cpp \
-	sqvm.cpp \
-	sqclass.cpp
-
-	
-	
-sq32:
-	gcc -O2  -fno-rtti -Wall -c $(SRCS) $(INCZ) $(DEFS)
-	ar rc $(OUT) *.o
-	rm *.o
-
-sqprof:
-	gcc -O2 -pg -fno-rtti -pie -gstabs -g3 -Wall -c $(SRCS) $(INCZ) $(DEFS)
-	ar rc $(OUT) *.o
-	rm *.o
-
-sq64:
-	gcc -O2  -D_SQ64 -fno-rtti -Wall -c $(SRCS) $(INCZ) $(DEFS)
-	ar rc $(OUT) *.o
-	rm *.o
diff -Nru SQUIRREL2.orig/squirrel/Makefile.am SQUIRREL2/squirrel/Makefile.am
--- SQUIRREL2.orig/squirrel/Makefile.am	1970-01-01 01:00:00.000000000 +0100
+++ SQUIRREL2/squirrel/Makefile.am	2007-07-07 19:03:05.000000000 +0200
@@ -0,0 +1,38 @@
+MAINTAINERCLEANFILES = Makefile.in
+AM_CXXFLAGS = -fno-rtti -Wall
+
+lib_LTLIBRARIES = libsquirrel.la
+
+libsquirrel_la_LDFLAGS = -release $(VERSION)
+
+libsquirrel_la_SOURCES = \
+		sqapi.cpp \
+		sqarray.h \
+		sqbaselib.cpp \
+		sqclass.cpp \
+		sqclass.h \
+		sqclosure.h \
+		sqcompiler.cpp \
+		sqcompiler.h \
+		sqdebug.cpp \
+		sqfuncproto.h \
+		sqfuncstate.cpp \
+		sqfuncstate.h \
+		sqlexer.cpp \
+		sqlexer.h \
+		sqmem.cpp \
+		sqobject.cpp \
+		sqobject.h \
+		sqopcodes.h \
+		sqpcheader.h \
+		sqstate.cpp \
+		sqstate.h \
+		sqstring.h \
+		sqtable.cpp \
+		sqtable.h \
+		squserdata.h \
+		squtils.h \
+		sqvm.cpp \
+		sqvm.h
+
+INCLUDES = -I$(top_srcdir)/include -I.
