Index: compat-wireless-2011-08-27-p/include/linux/compat-2.6.38.h
===================================================================
--- compat-wireless-2011-08-27-p.orig/include/linux/compat-2.6.38.h
+++ compat-wireless-2011-08-27-p/include/linux/compat-2.6.38.h
@@ -11,6 +11,10 @@
 /* rename member in struct mmc_host in include/linux/mmc/host.h */
 #define max_segs	max_hw_segs
 
+static inline bool pm_runtime_enabled(struct device *dev)
+{
+	return !dev->power.disable_depth;
+}
 
 /* Exponentially weighted moving average (EWMA) */
 
