diff -Nuar k3d-source-0.8.0.1.orig//k3dsdk/gil/boost/gil/extension/io/png_io_private.hpp k3d-source-0.8.0.1/k3dsdk/gil/boost/gil/extension/io/png_io_private.hpp
--- k3d-source-0.8.0.1.orig//k3dsdk/gil/boost/gil/extension/io/png_io_private.hpp	2010-04-18 08:49:33.000000000 +0300
+++ k3d-source-0.8.0.1/k3dsdk/gil/boost/gil/extension/io/png_io_private.hpp	2011-05-18 23:46:06.514000043 +0300
@@ -5,6 +5,10 @@
 */
 
 /*************************************************************************************************/
+// LIBPNG 1.4 PATCH
+#define png_infopp_NULL (png_infopp)NULL
+#define int_p_NULL (int*)NULL
+// PATCH END
 
 #ifndef GIL_PNG_IO_PRIVATE_H
 #define GIL_PNG_IO_PRIVATE_H
