--- wireless_tools.29/Makefile.makefile	2007-04-19 19:42:23.000000000 -0400
+++ wireless_tools.29/Makefile	2007-05-22 17:31:53.000000000 -0400
@@ -101,12 +101,15 @@ ifdef BUILD_WE_ESSENTIAL
   WEDEF_FLAG= -DWE_ESSENTIAL=y
 endif
 
+ifndef OPT_FLAGS
+  OPT_FLAGS=
+endif
 # Other flags
 CFLAGS=-Os -W -Wall -Wstrict-prototypes -Wmissing-prototypes -Wshadow \
 	-Wpointer-arith -Wcast-qual -Winline -I.
 #CFLAGS=-O2 -W -Wall -Wstrict-prototypes -I.
 DEPFLAGS=-MMD
-XCFLAGS=$(CFLAGS) $(DEPFLAGS) $(WARN) $(HEADERS) $(WELIB_FLAG) $(WEDEF_FLAG)
+XCFLAGS=$(CFLAGS) $(DEPFLAGS) $(WARN) $(HEADERS) $(WELIB_FLAG) $(WEDEF_FLAG) $(OPT_FLAGS)
 PICFLAG=-fPIC
 
 # Standard compilation targets
@@ -146,6 +149,7 @@ wireless.h:
 # Compilation of the dynamic library
 $(DYNAMIC): $(OBJS:.o=.so)
 	$(CC) -shared -o $@ -Wl,-soname,$@ $(STRIPFLAGS) $(LIBS) -lc $^
+	ln -s $@ $(DYNAMIC_LINK)
 
 # Compilation of the static library
 $(STATIC): $(OBJS:.o=.so)
