Index: qgit/src/src.pro
===================================================================
--- qgit.orig/src/src.pro
+++ qgit/src/src.pro
@@ -38,7 +38,7 @@ win32 {
 
 unix {
     TARGET = qgit
-    target.path = ~/bin
+    target.path = /usr/bin
     CONFIG += x11
 }
 
@@ -50,8 +50,8 @@ macx {
 }
 
 HAVE_GCC {
-	QMAKE_CXXFLAGS_RELEASE += -s -O2 -Wno-non-virtual-dtor -Wno-long-long -pedantic -Wconversion
-	QMAKE_CXXFLAGS_DEBUG += -g3 -ggdb -O0 -Wno-non-virtual-dtor -Wno-long-long -pedantic -Wconversion
+	QMAKE_CXXFLAGS_RELEASE +=
+	QMAKE_CXXFLAGS_DEBUG +=
 }
 
 ENABLE_CONSOLE_MSG {
