diff -ur gst-plugins-bad-0.10.23.orig/ext/directfb/dfb-example.c gst-plugins-bad-0.10.23/ext/directfb/dfb-example.c
--- ext/directfb/dfb-example.c	2011-12-11 20:49:11.000000000 +0200
+++ ext/directfb/dfb-example.c	2012-11-07 20:57:47.614763209 +0200
@@ -1,7 +1,7 @@
 
-#include <directfb.h>
 #include <stdio.h>
 #include <gst/gst.h>
+#include <directfb.h>
 
 static IDirectFB *dfb = NULL;
 static IDirectFBSurface *primary = NULL;
