--- src/org.gnome.Terminal.gschema.xml.in.OSmanOS	2013-09-24 19:50:11.000000000 +0300
+++ src/org.gnome.Terminal.gschema.xml.in	2013-10-13 19:01:16.931756891 +0300
@@ -149,12 +149,12 @@
       <_description>If true, newly created terminal windows will have custom size specified by default_size_columns and default_size_rows.</_description>
     </key>
     <key name="default-size-columns" type="i">
-      <default>80</default>
+      <default>120</default>
       <_summary>Default number of columns</_summary>
       <_description>Number of columns in newly created terminal windows. Has no effect if use_custom_default_size is not enabled.</_description>
     </key>
     <key name="default-size-rows" type="i">
-      <default>24</default>
+      <default>30</default>
       <_summary>Default number of rows</_summary>
       <_description>Number of rows in newly created terminal windows. Has no effect if use_custom_default_size is not enabled.</_description>
     </key>
@@ -168,17 +168,17 @@
       <_description>Number of scrollback lines to keep around. You can scroll back in the terminal by this number of lines; lines that don't fit in the scrollback are discarded. If scrollback_unlimited is true, this value is ignored.</_description>
     </key>
     <key name="scrollback-unlimited" type="b">
-      <default>false</default>
+      <default>true</default>
       <_summary>Whether an unlimited number of lines should be kept in scrollback</_summary>
       <_description>If true, scrollback lines will never be discarded. The scrollback history is stored on disk temporarily, so this may cause the system to run out of disk space if there is a lot of output to the terminal.</_description>
     </key>
     <key name="scroll-on-keystroke" type="b">
-      <default>true</default>
+      <default>false</default>
       <_summary>Whether to scroll to the bottom when a key is pressed</_summary>
       <_description>If true, pressing a key jumps the scrollbar to the bottom.</_description>
     </key>
     <key name="scroll-on-output" type="b">
-      <default>false</default>
+      <default>true</default>
       <_summary>Whether to scroll to the bottom when there's new output</_summary>
       <_description>If true, whenever there's new output the terminal will scroll to the bottom.</_description>
     </key>
@@ -347,7 +347,7 @@
       <_summary>Keyboard shortcut to create a new profile</_summary>
     </key>
     <key name="save-contents" type="s">
-      <default>'disabled'</default>
+      <default>'&lt;Ctrl&gt;&lt;Shift&gt;s'</default>
       <_summary>Keyboard shortcut to save the current tab contents to file</_summary>
     </key>
     <key name="close-tab" type="s">
@@ -359,11 +359,11 @@
       <_summary>Keyboard shortcut to close a window</_summary>
     </key>
     <key name="copy" type="s">
-      <default>'&lt;Ctrl&gt;&lt;Shift&gt;c'</default>
+      <default>'&lt;Ctrl&gt;c'</default>
       <_summary>Keyboard shortcut to copy text</_summary>
     </key>
     <key name="paste" type="s">
-      <default>'&lt;Ctrl&gt;&lt;Shift&gt;v'</default>
+      <default>'&lt;Ctrl&gt;v'</default>
       <_summary>Keyboard shortcut to paste text</_summary>
     </key>
     <key name="full-screen" type="s">
