Index: kvm-kmod-2.6.34/scripts/65-kvm.rules
===================================================================
--- kvm-kmod-2.6.34.orig/scripts/65-kvm.rules
+++ kvm-kmod-2.6.34/scripts/65-kvm.rules
@@ -1,2 +1,3 @@
-KERNEL=="kvm", MODE="0660", GROUP="kvm"
+# https://fedoraproject.org/wiki/Features/VirtPrivileges
+KERNEL=="kvm", MODE="0666", GROUP="kvm"
 ACTION=="add|change", SUBSYSTEM=="dmi", KERNEL=="id", RUN+="/bin/sh -c 'grep -q vmx /proc/cpuinfo && /sbin/modprobe kvm-intel; grep -q svm /proc/cpuinfo && /sbin/modprobe kvm-amd'"
