Index: hw/xfree86/vnc/vncInit.c
===================================================================
--- hw/xfree86/vnc/vncInit.c.orig
+++ hw/xfree86/vnc/vncInit.c
@@ -573,8 +573,8 @@ static XF86ModuleVersionInfo vncVersRec
 #if 0
 	ABI_EXTENSION_VERSION,
 #else
-        /* Hack to allow module to work with more servers (vs. 0.3 above) */
-        SET_ABI_VERSION(0, 2),
+        /* Hack to allow module to work with more servers */
+        SET_ABI_VERSION(1, 0),
 #endif
 	MOD_CLASS_EXTENSION,
 	{0,0,0,0}
Index: xcliplist/cliplistmod.c
===================================================================
--- xcliplist/cliplistmod.c.orig
+++ xcliplist/cliplistmod.c
@@ -26,8 +26,8 @@ static XF86ModuleVersionInfo VersRec = {
 #if 0
 	ABI_EXTENSION_VERSION,
 #else
-        /* Hack to allow module to work with more servers (vs. 0.3 above) */
-        SET_ABI_VERSION(0, 2),
+        /* Hack to allow module to work with more servers */
+        SET_ABI_VERSION(1, 0),
 #endif
 	MOD_CLASS_EXTENSION,
 	{0,0,0,0}
