Index: hw/dmx/Makefile.am
================================================================================
--- hw/dmx/Makefile.am
+++ hw/dmx/Makefile.am
@@ -1,6 +1,6 @@
 DIST_SUBDIRS = input vnc config glxProxy examples doc
 
-SUBDIRS = input vnc config examples
+SUBDIRS = input config examples
 bin_PROGRAMS = Xdmx
 
 if XINERAMA
--- hw/dmx/dmx-config.h
+++ hw/dmx/dmx-config.h
@@ -72,7 +72,7 @@
 /* Enable the DMX extension */
 #define DMXEXT
 
-/* Enable VNC ability */
-#define DMXVNC 1
+/* Disable VNC ability */
+#undef DMXVNC
 
 #endif /* DMX_CONFIG_H */
--- hw/dmx/input/Makefile.am
+++ hw/dmx/input/Makefile.am
@@ -61,7 +61,6 @@
 AM_CFLAGS = $(DIX_CFLAGS) \
             -I$(top_srcdir)/hw/dmx \
             -DHAVE_DMX_CONFIG_H \
-            -DDMXVNC=1 \
             $(GLX_DEFS) \
             @DMXMODULES_CFLAGS@
 
--- hw/dmx/vnc/Makefile.am
+++ hw/dmx/vnc/Makefile.am
@@ -37,7 +37,6 @@
             -I$(top_srcdir)/hw/xfree86/common \
             -DHAVE_DMX_CONFIG_H \
 	    $(DIX_CFLAGS) \
-            -DDMXVNC=1 \
             @DMXMODULES_CFLAGS@
 
 ###EXTRA_DIST = dmxdetach.c
