Index: xorg-server-1.6.3.901/hw/vnc/sockets.c
===================================================================
--- xorg-server-1.6.3.901.orig/hw/vnc/sockets.c
+++ xorg-server-1.6.3.901/hw/vnc/sockets.c
@@ -448,7 +448,7 @@ WriteExact(sock, buf, len)
     int n;
     fd_set fds;
     struct timeval tv;
-#if 0
+#if 1
     int totalTimeWaited = 0;
 #endif
 
@@ -470,7 +470,7 @@ WriteExact(sock, buf, len)
 		return n;
 	    }
 
-#if 0
+#if 1
 	    /* Retry every 5 seconds until we exceed rfbMaxClientWait.  We
 	       need to do this because select doesn't necessarily return
 	       immediately when the other end has gone away */
@@ -488,7 +488,7 @@ WriteExact(sock, buf, len)
 	    tv.tv_usec = 0;
 #endif
 	    n = select(sock+1, NULL, &fds, NULL, &tv);
-#if 0
+#if 1
 	    if (n < 0) {
 		rfbLogPerror("WriteExact: select");
 		return n;
