Index: hw/vnc/cutpaste.c
===================================================================
--- hw/vnc/cutpaste.c.orig
+++ hw/vnc/cutpaste.c
@@ -58,7 +58,7 @@ rfbSetXCutText(char *str, int len)
 			 8, PropModeReplace, len,
 			 (pointer)str, TRUE);
     
-    if (dixLookupSelection(&pSel, XA_PRIMARY, serverClient, DixSetAttrAccess)) {
+    if (dixLookupSelection(&pSel, XA_PRIMARY, serverClient, DixSetAttrAccess) == Success) {
 	xEvent event;
 
 	if (pSel->client) {
