Index: xorg-server-1.6.3.901/hw/vnc/sockets.c
===================================================================
--- xorg-server-1.6.3.901.orig/hw/vnc/sockets.c
+++ xorg-server-1.6.3.901/hw/vnc/sockets.c
@@ -490,8 +490,10 @@ WriteExact(sock, buf, len)
 	    n = select(sock+1, NULL, &fds, NULL, &tv);
 #if 1
 	    if (n < 0) {
-		rfbLogPerror("WriteExact: select");
-		return n;
+		if (errno != EINTR) {
+		    rfbLogPerror("WriteExact: select");
+		    return n;
+		}
 	    }
 	    if (n == 0) {
 		totalTimeWaited += 5000;
