--- hw/vnc/vncext.c
+++ hw/vnc/vncext.c
@@ -140,7 +140,7 @@
     if (rec) {
         rec->client = client;
         rec->fakeID = FakeClientID(client->index);
-        rec->res = CreateNewResourceType(VncDestroyClientResourceCallback);
+        rec->res = CreateNewResourceType(VncDestroyClientResourceCallback,"DestroyClientResourceCallback");
         if (!AddResource(rec->fakeID, rec->res, rec)) {
             xfree(rec);
         }
@@ -689,7 +689,7 @@
 
     VncResourceGeneration = serverGeneration;
 
-    if (!(VncNotifyList = CreateNewResourceType(VncDestroyNotifyList))) {
+    if (!(VncNotifyList = CreateNewResourceType(VncDestroyNotifyList,"DestroyNotifyList"))) {
         ErrorF("CreateResourceTypes: failed to allocate vnc notify list resource.\n");
         return FALSE;
     }
--- hw/vnc/init.c
+++ hw/vnc/init.c
@@ -984,6 +984,11 @@
 }
 
 void
+CloseInput()
+{
+}
+
+void
 OsVendorInit()
 {
 }
@@ -1047,7 +1052,7 @@
 }
 
 int
-NewInputDeviceRequest(InputOption *options, DeviceIntPtr *pdev)
+NewInputDeviceRequest(InputOption *options, InputAttributes *attrs, DeviceIntPtr *pdev)
 {
     return BadValue;
 }
