diff -Nur quesoglc-0.7.2-old//build/m4/acx_pthread.m4 quesoglc-0.7.2/build/m4/acx_pthread.m4
--- quesoglc-0.7.2-old//build/m4/acx_pthread.m4	2010-09-04 18:18:05.652000646 +0300
+++ quesoglc-0.7.2/build/m4/acx_pthread.m4	2010-09-04 18:18:20.672997446 +0300
@@ -80,7 +80,7 @@
 # which indicates that we try without any flags at all, and "pthread-config"
 # which is a program returning the flags for the Pth emulation library.
 
-acx_pthread_flags="pthreads none -Kthread -kthread lthread -pthread -pthreads -mthreads pthread --thread-safe -mt pthread-config"
+acx_pthread_flags="pthreads none -Kthread -kthread lthread -lpthread -pthread -pthreads -mthreads pthread --thread-safe -mt pthread-config"
 
 # The ordering *is* (sometimes) important.  Some notes on the
 # individual items follow:
