--- mainmenu_tr1.ts	2015-02-08 11:31:27.000000000 +0200
+++ plugin-mainmenu/translations/mainmenu_tr.ts	2015-02-12 11:10:52.222062935 +0200
@@ -6,7 +6,7 @@
     <message>
         <location filename="../lxqtmainmenu.cpp" line="314"/>
         <source>Show/hide main menu</source>
-        <translation type="unfinished"></translation>
+        <translation>Ana menüyü göster/gizle</translation>
     </message>
     <message>
         <location filename="../lxqtmainmenu.cpp" line="339"/>
@@ -36,22 +36,22 @@
     <message>
         <location filename="../lxqtmainmenuconfiguration.ui" line="14"/>
         <source>Main Menu settings</source>
-        <translation type="unfinished"></translation>
+        <translation>Ana Menü Ayarları</translation>
     </message>
     <message>
         <location filename="../lxqtmainmenuconfiguration.ui" line="35"/>
         <source>Button text:</source>
-        <translation type="unfinished"></translation>
+        <translation>Düğme metni</translation>
     </message>
     <message>
         <location filename="../lxqtmainmenuconfiguration.ui" line="52"/>
         <source>Custom font size:</source>
-        <translation type="unfinished"></translation>
+        <translation>Özel yazıtipi boyutu:</translation>
     </message>
     <message>
         <location filename="../lxqtmainmenuconfiguration.ui" line="62"/>
         <source> pt</source>
-        <translation type="unfinished"></translation>
+        <translation></translation>
     </message>
     <message>
         <location filename="../lxqtmainmenuconfiguration.ui" line="81"/>
@@ -61,7 +61,7 @@
     <message>
         <location filename="../lxqtmainmenuconfiguration.ui" line="87"/>
         <source>Menu file:</source>
-        <translation type="unfinished"></translation>
+        <translation>Menü dosyası:</translation>
     </message>
     <message>
         <location filename="../lxqtmainmenuconfiguration.ui" line="101"/>
@@ -81,7 +81,7 @@
     <message>
         <location filename="../lxqtmainmenuconfiguration.cpp" line="55"/>
         <source>Reset</source>
-        <translation type="unfinished"></translation>
+        <translation>Sıfırla</translation>
     </message>
     <message>
         <location filename="../lxqtmainmenuconfiguration.cpp" line="100"/>
@@ -91,7 +91,7 @@
     <message>
         <location filename="../lxqtmainmenuconfiguration.cpp" line="100"/>
         <source>Menu files (*.menu)</source>
-        <translation>Menü dosyaları (*.menü)</translation>
+        <translation type="unfinished">Menü dosyaları (*.menu)</translation>
     </message>
 </context>
 </TS>
