diff -ru scummvm-1.5.0.orig/config.guess scummvm-1.5.0/config.guess
--- scummvm-1.5.0.orig/config.guess	2012-05-31 06:39:11.000000000 -0400
+++ scummvm-1.5.0/config.guess	2012-10-07 21:27:00.510982188 -0400
@@ -855,7 +855,7 @@
 	exit ;;
     *:GNU/*:*:*)
 	# other systems with GNU libc and userland
-	echo ${UNAME_MACHINE}-unknown-`echo ${UNAME_SYSTEM} | sed 's,^[^/]*/,,' | tr '[A-Z]' '[a-z]'``echo ${UNAME_RELEASE}|sed -e 's/[-(].*//'`-gnu
+	echo ${UNAME_MACHINE}-unknown-`echo ${UNAME_SYSTEM} | sed 's,^[^/]*/,,' | tr '[:upper:]' '[:lower:]'``echo ${UNAME_RELEASE}|sed -e 's/[-(].*//'`-gnu
 	exit ;;
     i*86:Minix:*:*)
 	echo ${UNAME_MACHINE}-pc-minix
@@ -955,7 +955,7 @@
 	exit ;;
     parisc:Linux:*:* | hppa:Linux:*:*)
 	# Look for CPU level
-	case `grep '^cpu[^a-z]*:' /proc/cpuinfo 2>/dev/null | cut -d' ' -f2` in
+	case `grep '^cpu[^[:lower:]]*:' /proc/cpuinfo 2>/dev/null | cut -d' ' -f2` in
 	  PA7*) echo hppa1.1-unknown-linux-gnu ;;
 	  PA8*) echo hppa2.0-unknown-linux-gnu ;;
 	  *)    echo hppa-unknown-linux-gnu ;;
diff -ru scummvm-1.5.0.orig/configure scummvm-1.5.0/configure
--- scummvm-1.5.0.orig/configure	2012-10-07 21:21:27.856337073 -0400
+++ scummvm-1.5.0/configure	2012-10-07 21:26:08.689750153 -0400
@@ -3897,7 +3897,7 @@
 	fi
 
 	# Save the settings
-	defname="ENABLE_`echo $engine | tr '[a-z]' '[A-Z]'`"
+	defname="ENABLE_`echo $engine | tr '[:lower:]' '[:upper:]'`"
 	if test "$isbuilt" = "no" ; then
 		add_line_to_config_mk "# $defname"
 	else
