diff -Nur scummvm-1.3.0-old//backends/platform/sdl/sdl.cpp scummvm-1.3.0/backends/platform/sdl/sdl.cpp
--- scummvm-1.3.0-old//backends/platform/sdl/sdl.cpp	2011-06-13 14:31:28.670000000 +0300
+++ scummvm-1.3.0/backends/platform/sdl/sdl.cpp	2011-06-13 14:32:47.545999345 +0300
@@ -347,7 +347,7 @@
 	}
 #else // WIN32
 	// Activating current locale settings
-	const char *locale = setlocale(LC_ALL, "");
+	const char *locale = setlocale(LC_ALL, "en_US.UTF-8");
 
 	// Detect the language from the locale
 	if (!locale) {
