diff -Nuar kitchensync-0.22.0.orig//src/kitchensync.desktop kitchensync-0.22.0//src/kitchensync.desktop
--- kitchensync-0.22.0.orig//src/kitchensync.desktop	2011-02-09 15:12:54.455000020 +0200
+++ kitchensync-0.22.0//src/kitchensync.desktop	2011-02-09 15:19:29.072000028 +0200
@@ -13,6 +13,7 @@
 Name[pt_BR]=KitchenSync
 Name[sv]=Kitchensync
 Name[uk]=KitchenSync
+Name[tr]=KitchenSync
 Name[x-test]=xxKitchenSyncxx
 GenericName=Synchronization
 GenericName[bg]=Синхронизиране
@@ -29,6 +30,7 @@
 GenericName[pt_BR]=Sincronização
 GenericName[sv]=Synkronisering
 GenericName[uk]=Синхронізація
+GenericName[tr]=Eşzamanlama
 GenericName[x-test]=xxSynchronizationxx
 Exec=kitchensync
 Icon=kitchensync
