--- src/CMakeLists.txt	2010-09-21 18:38:55.000000000 +0200
+++ src/CMakeLists.txt.new	2011-12-01 00:15:02.505502319 +0100
@@ -12,7 +12,7 @@
 
 KDE4_ADD_EXECUTABLE(kitchensync ${kitchensync_SRCS})
 
-TARGET_LINK_LIBRARIES(kitchensync kitchensyncprivate ${KDE_KDEUI_LIBS} )
+TARGET_LINK_LIBRARIES(kitchensync kitchensyncprivate ${KDE4_KDEUI_LIBS} )
 
 INSTALL(TARGETS kitchensync ${INSTALL_TARGETS_DEFAULT_ARGS})
 
