diff -ur bzr-2.0.4.orig/setup.py bzr-2.0.4/setup.py
--- bzr-2.0.4.orig/setup.py	2010-01-21 22:11:55.000000000 +0200
+++ bzr-2.0.4/setup.py	2010-01-30 23:36:41.882697092 +0200
@@ -684,7 +684,7 @@
     if not 'bdist_egg' in sys.argv:
         # generate and install bzr.1 only with plain install, not the
         # easy_install one
-        DATA_FILES = [('man/man1', ['bzr.1'])]
+        DATA_FILES = [('share/man/man1', ['bzr.1'])]
 
     # std setup
     ARGS = {'scripts': ['bzr'],
Yalnızca bzr-2.0.4'da: setup.py.orig
