diff -ur exif-0.6.19.orig/exif/main.c exif-0.6.19/exif/main.c
--- exif-0.6.19.orig/exif/main.c	2009-10-27 07:09:44.000000000 +0100
+++ exif-0.6.19/exif/main.c	2009-11-20 15:27:43.000000000 +0100
@@ -131,10 +131,10 @@
 		/* Fall through to EXIF_LOG_CODE_NO_MEMORY */
 	case EXIF_LOG_CODE_NO_MEMORY:
 		put_colorstring (stderr, COL_RED COL_BOLD COL_UNDERLINE);
-		fprintf (stderr, exif_log_code_get_title (code));
+		fprintf (stderr, "%s", exif_log_code_get_title (code));
 		fprintf (stderr, "\n");
 		put_colorstring (stderr, COL_NORMAL COL_RED);
-		fprintf (stderr, exif_log_code_get_message (code));
+		fprintf (stderr, "%s", exif_log_code_get_message (code));
 		fprintf (stderr, "\n");
 		fprintf (stderr, "%s: ", domain);
 		vfprintf (stderr, format, args);
