--- setup.py.orig	2010-08-18 14:58:57.032780619 +0300
+++ setup.py	2010-08-18 14:59:57.798035996 +0300
@@ -68,7 +68,7 @@
     ext_modules = [ ]
 
 exiv2 = Extension('kaa.metadata.image.exiv2', ['src/image/exiv2.cpp'], libraries=[ 'exiv2' ])
-if exiv2.check_cc([], '', '-lexiv2'):
+if exiv2.check_cc([], '', '&_nonexistent'):
     print 'Building experimental exiv2 parser'
     ext_modules.append(exiv2)
     
