diff -Nur nut-2.2.1-pre1-old/drivers/snmp-ups.h nut-2.2.1-pre1/drivers/snmp-ups.h
--- nut-2.2.1-pre1-old/drivers/snmp-ups.h	2007-10-19 14:14:06.000000000 +0300
+++ nut-2.2.1-pre1/drivers/snmp-ups.h	2007-10-19 14:14:22.000000000 +0300
@@ -207,3 +207,8 @@
 int g_pwr_battery;
 int pollfreq; /* polling frequency */
 int input_phases, output_phases;
+
+/* these symbols are referenced but not defined, blah */
+int allow_severity = LOG_DEBUG;
+int deny_severity = LOG_ERR;
+
