diff -Nur nut-2.4.1-old/drivers/Makefile.am nut-2.4.1/drivers/Makefile.am
--- nut-2.4.1-old/drivers/Makefile.am	2009-05-07 17:36:31.000000000 +0300
+++ nut-2.4.1/drivers/Makefile.am	2009-05-07 17:37:55.000000000 +0300
@@ -215,7 +215,7 @@
  mge-shut.h mge-utalk.h mge-xml.h microdowell.h netvisionmib.h netxml-ups.h nut_usb.h oneac.h	\
  powercom.h powerpanel.h powerp-bin.h powerp-txt.h pwmib.h raritan-mib.h	\
  safenet.h serial.h snmp-ups.h solis.h tripplite.h tripplite-hid.h upshandler.h		\
- usb-common.h usbhid-ups.h
+ usb-common.h usbhid-ups.h nibumib.h
 
 # Define a dummy library so that Automake builds rules for the
 # corresponding object files.  This library is not actually built,
diff -Nur nut-2.4.1-old/drivers/Makefile.in nut-2.4.1/drivers/Makefile.in
--- nut-2.4.1-old/drivers/Makefile.in	2009-05-07 17:36:31.000000000 +0300
+++ nut-2.4.1/drivers/Makefile.in	2009-05-07 17:38:12.000000000 +0300
@@ -717,7 +717,7 @@
  mge-shut.h mge-utalk.h mge-xml.h microdowell.h netvisionmib.h netxml-ups.h nut_usb.h oneac.h	\
  powercom.h powerpanel.h powerp-bin.h powerp-txt.h pwmib.h raritan-mib.h	\
  safenet.h serial.h snmp-ups.h solis.h tripplite.h tripplite-hid.h upshandler.h		\
- usb-common.h usbhid-ups.h
+ usb-common.h usbhid-ups.h nibumibs.h
 
 
 # Define a dummy library so that Automake builds rules for the
diff -Nur nut-2.4.1-old/drivers/snmp-ups.c nut-2.4.1/drivers/snmp-ups.c
--- nut-2.4.1-old/drivers/snmp-ups.c	2009-05-07 17:36:31.000000000 +0300
+++ nut-2.4.1/drivers/snmp-ups.c	2009-05-07 17:40:11.000000000 +0300
@@ -40,6 +40,7 @@
 #include "mgemib.h"
 #include "netvisionmib.h"
 #include "pwmib.h"
+#include "nibumib.h"
 
 mib2nut_info_t mib2nut[] = {
 	{ "apcc", APCC_MIB_VERSION, APCC_OID_POWER_STATUS,
@@ -58,6 +59,8 @@
 		APHEL2_OID_MODEL_NAME, eaton_aphel_revelation_mib },
 	{ "raritan", RARITAN_MIB_VERSION, "",
 		RARITAN_OID_MODEL_NAME, raritan_mib },
+    { "nibu", NIBU_MIB_VERSION, NIBU_OID_BATT_STATUS,
+		NIBU_OID_MODEL_NAME, nibu_mib },
 	{ NULL }
 };
 
@@ -166,7 +169,7 @@
 	upsdebugx(1, "entering upsdrv_makevartable()");
 
 	addvar(VAR_VALUE, SU_VAR_MIBS,
-	    "Set MIB compliance (default=ietf, allowed mge,apcc,netvision,pw)");
+	    "Set MIB compliance (default=ietf, allowed mge,apcc,netvision,pw,nibu)");
 	addvar(VAR_VALUE | VAR_SENSITIVE, SU_VAR_COMMUNITY,
 	    "Set community name (default=public)");
 	addvar(VAR_VALUE, SU_VAR_VERSION,
