diff -Nur nut-2.0.5-old/drivers/nibumib.h nut-2.0.5/drivers/nibumib.h
--- nut-2.0.5-old/drivers/nibumib.h	1970-01-01 02:00:00.000000000 +0200
+++ nut-2.0.5/drivers/nibumib.h	2007-06-22 19:49:35.083113150 +0300
@@ -0,0 +1,104 @@
+/*  nibumib.h
+ *
+ *  data to monitor NIBU SNMP UPS (RFC 1215 compliant) with NUT
+ *
+ */
+
+#define NIBU_MIB_VERSION            "0.1"
+
+/* SNMP OIDs set */
+#define NIBU_OID_UPS_MIB            ".1.3.6.1.4.1.935"
+#define NIBU_OID_MODEL_NAME         ".1.3.6.1.4.1.935.1.1.1.1.1.1.0"
+#define NIBU_OID_AGENTREV           ".1.3.6.1.4.1.935.1.1.1.1.2.4.0"
+#define NIBU_OID_FIRMREV            ".1.3.6.1.4.1.935.1.1.1.1.2.1.0"
+
+#define NIBU_OID_BATT_STATUS        ".1.3.6.1.4.1.935.1.1.1.2.1.1.0"
+#define NIBU_OID_BATT_RUNTIME       ".1.3.6.1.4.1.935.1.1.1.2.2.4.0"
+#define NIBU_OID_BATT_RUNTIME_LOW   ".1.3.6.1.4.1.935.1.1.1.5.2.8.0"
+#define NIBU_OID_BATT_CHARGE        ".1.3.6.1.4.1.935.1.1.1.2.2.1.0"
+#define NIBU_OID_BATT_VOLTAGE       ".1.3.6.1.4.1.935.1.1.1.2.2.2.0"
+#define NIBU_OID_BATT_TEMP          ".1.3.6.1.4.1.935.1.1.1.2.2.3.0"
+#define NIBU_OID_BATT_FULL_CHARGE   ".1.3.6.1.4.1.935.1.1.1.2.2.6.0"
+
+#define NIBU_OID_IN_FREQ            ".1.3.6.1.4.1.935.1.1.1.3.2.4.0"
+#define NIBU_OID_IN_VOLTAGE         ".1.3.6.1.4.1.935.1.1.1.3.2.1.0"
+#define NIBU_OID_IN_MAX_VOLTAGE     ".1.3.6.1.4.1.935.1.1.1.3.2.2.0"
+#define NIBU_OID_IN_MIN_VOLTAGE     ".1.3.6.1.4.1.935.1.1.1.3.2.3.0"
+
+#define NIBU_OID_OUT_VOLTAGE        ".1.3.6.1.4.1.935.1.1.1.4.2.1.0"
+#define NIBU_OID_OUT_FREQ           ".1.3.6.1.4.1.935.1.1.1.4.2.2.0"
+#define NIBU_OID_OUT_STATUS         ".1.3.6.1.4.1.935.1.1.1.4.1.1.0"
+#define NIBU_OID_OUT_LOAD           ".1.3.6.1.4.1.935.1.1.1.4.2.3.0"
+
+
+/* Defines for NIBU_OID_BATT_STATUS (2) */
+info_lkp_t nibu_batt_info[] = {
+    { 1, "" },      /* unknown */
+    { 2, "" },      /* battery normal, should not change anything for nut */
+    { 3, "LB" },    /* batteryLow */
+    { 0, "NULL" }
+} ;
+
+/* Defines for NIBU_OID_OUT_STATUS (2) */
+info_lkp_t nibu_out_info[] = {
+    { 1, "" },        /* unknown */
+    { 2, "OL" },      /* on line */
+    { 3, "OB" },      /* on battery */
+    { 4, "BOOST" },   /* on boost */
+    { 5, "SLEEP" },   /* sleeping */
+    { 6, "BYPASS" },  /* on bypass*/
+    { 7, "REBOOT" },  /* rebooting */
+    { 8, "STANDBY" }, /* standby */
+    { 8, "TRIM" },    /* on buck */
+    { 0, "NULL" }
+} ;
+
+#define NIBU_OID_SD_AFTER_DELAY        ".1.3.6.1.4.1.935.1.1.1.5.2.10.0"
+#define NIBU_OFF_DO                    0
+
+
+/* Snmp2NUT lookup table */
+snmp_info_t nibu_mib[] = {
+    /* UPS page */
+    { "ups.mfr", ST_FLAG_STRING, SU_INFOSIZE, NULL, "NIBU SNMP CARD",
+        SU_FLAG_STATIC | SU_FLAG_ABSENT | SU_FLAG_OK, NULL },
+    { "ups.model", ST_FLAG_STRING, SU_INFOSIZE, NIBU_OID_MODEL_NAME,
+        "Generic SNMP UPS", SU_FLAG_STATIC | SU_FLAG_OK, NULL },
+    { "ups.firmware", ST_FLAG_STRING, SU_INFOSIZE, NIBU_OID_FIRMREV, "",
+        SU_FLAG_STATIC | SU_FLAG_OK, NULL },
+    { "ups.firmware.aux", ST_FLAG_STRING, SU_INFOSIZE, NIBU_OID_AGENTREV, "",
+        SU_FLAG_STATIC | SU_FLAG_OK, NULL },
+    { "ups.status", ST_FLAG_STRING, SU_INFOSIZE, NIBU_OID_OUT_STATUS, "",
+        SU_FLAG_OK | SU_STATUS_BATT, &nibu_out_info[0] },
+    { "ups.status", ST_FLAG_STRING, SU_INFOSIZE, NIBU_OID_BATT_STATUS, "",
+        SU_FLAG_OK | SU_STATUS_BATT, &nibu_batt_info[0] },
+    /* get percent of load and print in percent mode */
+    { "ups.load", 0, 0.0001, NIBU_OID_OUT_LOAD, "", SU_FLAG_OK, NULL },
+
+    /* Battery page */
+    { "battery.charge", 0, 1, NIBU_OID_BATT_CHARGE, "", SU_FLAG_OK, NULL },
+    { "battery.runtime", 0, 60, NIBU_OID_BATT_RUNTIME, "", SU_FLAG_OK, NULL },
+    { "battery.runtime.low", ST_FLAG_RW, 1, NIBU_OID_BATT_RUNTIME_LOW, "", SU_FLAG_OK, NULL },
+    { "battery.voltage", 0, 0.1, NIBU_OID_BATT_VOLTAGE, "", SU_FLAG_OK, NULL },
+    { "battery.temperature", 0, 0.1, NIBU_OID_BATT_TEMP, "", SU_FLAG_OK, NULL },
+    { "battery.voltage.nominal", 0,1, NIBU_OID_BATT_FULL_CHARGE, "", SU_FLAG_OK, NULL },
+
+    /* Output page */
+    { "output.voltage", 0, 0.1, NIBU_OID_OUT_VOLTAGE, "", SU_FLAG_OK, NULL },
+    { "output.frequency", 0, 0.1, NIBU_OID_OUT_FREQ, "", SU_FLAG_OK, NULL },
+
+    /* Input page */
+    { "input.voltage", 0, 0.1, NIBU_OID_IN_VOLTAGE, "", SU_FLAG_OK, NULL },
+    { "input.voltage.maximum", 0, 0.1, NIBU_OID_IN_MAX_VOLTAGE, "", SU_FLAG_OK, NULL },
+    { "input.voltage.minimum", 0, 0.1, NIBU_OID_IN_MIN_VOLTAGE, "", SU_FLAG_OK, NULL },
+    { "input.frequency", 0, 0.1, NIBU_OID_IN_FREQ, "", SU_FLAG_OK, NULL },
+
+    /* instant commands. */
+    /*
+    { "load.off", 0, NIBU_OFF_DO, NIBU_OID_SD_AFTER_DELAY, "", SU_TYPE_CMD | SU_FLAG_OK, NULL },
+    */
+/*    { CMD_SHUTDOWN, 0, NIBU_OFF_GRACEFUL, NIBU_OID_OFF, "", SU_FLAG_OK, NULL }, */
+
+    /* end of structure. */
+    { NULL, 0, 0, NULL, NULL, 0, NULL }
+};
