diff -up nut-2.4.3/m4/nut_check_libhal.m4.halpath nut-2.4.3/m4/nut_check_libhal.m4
--- nut-2.4.3/m4/nut_check_libhal.m4.halpath	2010-02-11 22:43:05.000000000 +0100
+++ nut-2.4.3/m4/nut_check_libhal.m4	2010-02-24 15:04:17.455678931 +0100
@@ -86,29 +86,8 @@ if test -z "${nut_have_libhal_seen}"; th
 		dnl As per HAL spec, §5 Callouts and §2 Device Information Files
 		dnl - addon install path: $libdir/hal
 		AC_MSG_CHECKING(for libhal Callouts path)
-		HAL_CALLOUTS_PATH=`pkg-config --silence-errors --variable=libexecdir hal`
-		if test -n "$HAL_CALLOUTS_PATH"; then
-			AC_MSG_RESULT(${HAL_CALLOUTS_PATH})
-		else
-			# fallback to detecting the right path
-			if (test -d "${libdir}/hal"); then
-				# For Debian
-				HAL_CALLOUTS_PATH="${libdir}/hal"
-				AC_MSG_RESULT(${HAL_CALLOUTS_PATH})
-			elif (test -d "/usr/libexec"); then
-				# For RedHat
-				HAL_CALLOUTS_PATH="${libexecdir}"
-				AC_MSG_RESULT(${HAL_CALLOUTS_PATH})
-			elif (test -d "/usr/lib/hal"); then
-				# For OpenSUSE
-				HAL_CALLOUTS_PATH="${libdir}/hal"
-				AC_MSG_RESULT(${HAL_CALLOUTS_PATH})
-			else
-				# FIXME
-				HAL_CALLOUTS_PATH="${libdir}/hal"
-				AC_MSG_RESULT(using default (${HAL_CALLOUTS_PATH}))
-			fi
-		fi
+		HAL_CALLOUTS_PATH="${libexecdir}"
+		AC_MSG_RESULT(${HAL_CALLOUTS_PATH})
 
 		dnl - fdi install path: $datarootdir/hal/fdi/information/20thirdparty
 		AC_MSG_CHECKING(for libhal Device Information path)
