diff -up nut-2.6.5/configure.in.quickfix nut-2.6.5/configure.in
--- nut-2.6.5/configure.in.quickfix	2012-08-10 14:19:55.924431026 +0200
+++ nut-2.6.5/configure.in	2012-08-10 14:20:13.332566508 +0200
@@ -986,9 +986,8 @@ dnl Override installation directory, wit
 dnl prefix. This is needed for 'distcheck*' targets, otherwise
 dnl files will try to get intalled to the actual system directories
 if test -n "${systemdsystemunitdir}"; then
-	systemdsystemshutdowndir="${libdir}/systemd/system-shutdown"
+	systemdsystemshutdowndir="/lib/systemd/system-shutdown"
 	AC_MSG_RESULT(using ${systemdsystemunitdir})
-	systemdsystemunitdir="`echo ${systemdsystemunitdir} | sed 's/\/lib/\${libdir}/'`"
 else
 	AC_MSG_RESULT(no)
 fi