diff -up nut-2.4.3/drivers/libhid.c.bz575334 nut-2.4.3/drivers/libhid.c
--- nut-2.4.3/drivers/libhid.c.bz575334	2010-03-22 17:34:16.251422549 +0100
+++ nut-2.4.3/drivers/libhid.c	2010-03-22 17:34:16.278422151 +0100
@@ -141,7 +141,7 @@ static int refresh_report_buffer(reportb
 {
 	int	id = pData->ReportID;
 	int	r;
-	unsigned char	buf[SMALLBUF];
+	unsigned char	buf[8];
 
 	if (rbuf->ts[id] + age > time(NULL)) {
 		/* buffered report is still good; nothing to do */
@@ -469,7 +469,7 @@ bool_t HIDSetItemValue(hid_dev_handle_t 
  */
 int HIDGetEvents(hid_dev_handle_t udev, HIDData_t **event, int eventsize)
 {
-	unsigned char	buf[SMALLBUF];
+	unsigned char	buf[8];
 	int		itemCount = 0;
 	int		buflen, r, i;
 	HIDData_t	*pData;
