Index: lua-5.1.4/etc/lua.pc
===================================================================
--- lua-5.1.4.orig/etc/lua.pc
+++ lua-5.1.4/etc/lua.pc
@@ -18,14 +18,13 @@ INSTALL_CMOD= ${prefix}/lib/lua/${V}
 
 # canonical vars
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
 includedir=${prefix}/include
 
 Name: Lua
 Description: An Extensible Extension Language
 Version: ${R}
 Requires: 
-Libs: -L${libdir} -llua -lm
+Libs: -llua -lm
 Cflags: -I${includedir}
 
 # (end of lua.pc)
