diff -ruNp docbook-xsl-1.74.0.orig/fo/lists.xsl docbook-xsl-1.74.0/fo/lists.xsl
--- docbook-xsl-1.74.0.orig/fo/lists.xsl	2008-06-01 23:36:39.000000000 +0200
+++ docbook-xsl-1.74.0/fo/lists.xsl	2008-08-06 13:32:46.000000000 +0200
@@ -324,7 +324,7 @@
       </xsl:when>
       <xsl:when test="@termlength">
         <xsl:variable name="termlength.is.number">
-          <xsl:value-of select="@termlength + 0"/>
+          <xsl:value-of select="@termlength"/>
         </xsl:variable>
         <xsl:choose>
           <xsl:when test="string($termlength.is.number) = 'NaN'">
@@ -333,15 +333,7 @@
           </xsl:when>
           <xsl:otherwise>
             <xsl:value-of select="@termlength"/>
-            <xsl:choose>
-              <!-- workaround for passivetex lack of support for non-constant expressions -->
-              <xsl:when test="$passivetex.extensions != 0">
-                <xsl:text>em</xsl:text>
-              </xsl:when>
-              <xsl:otherwise>
-                <xsl:text>em * 0.60</xsl:text>
-              </xsl:otherwise>
-            </xsl:choose>
+            <xsl:text>em</xsl:text>
           </xsl:otherwise>
         </xsl:choose>
       </xsl:when>
@@ -350,15 +342,7 @@
           <xsl:with-param name="terms" select="varlistentry/term"/>
           <xsl:with-param name="maxlength" select="$variablelist.max.termlength"/>
         </xsl:call-template>
-        <xsl:choose>
-          <!-- workaround for passivetex lack of support for non-constant expressions -->
-          <xsl:when test="$passivetex.extensions != 0">
-            <xsl:text>em</xsl:text>
-          </xsl:when>
-          <xsl:otherwise>
-            <xsl:text>em * 0.60</xsl:text>
-          </xsl:otherwise>
-        </xsl:choose>
+        <xsl:text>em</xsl:text>
       </xsl:otherwise>
     </xsl:choose>
   </xsl:variable>
@@ -372,17 +356,7 @@
 
   <xsl:variable name="label-separation">1em</xsl:variable>
   <xsl:variable name="distance-between-starts">
-    <xsl:choose>
-      <!-- workaround for passivetex lack of support for non-constant expressions -->
-      <xsl:when test="$passivetex.extensions != 0">
-        <xsl:value-of select="$termlength"/>
-      </xsl:when>
-      <xsl:otherwise>
-        <xsl:value-of select="$termlength"/>
-        <xsl:text>+</xsl:text>
-        <xsl:value-of select="$label-separation"/>
-      </xsl:otherwise>
-    </xsl:choose>
+    <xsl:value-of select="$termlength"/>
   </xsl:variable>
 
   <xsl:if test="title">
diff -ruNp docbook-xsl-1.74.0.orig/fo/param.xsl docbook-xsl-1.74.0/fo/param.xsl
--- docbook-xsl-1.74.0.orig/fo/param.xsl	2008-06-02 01:06:18.000000000 +0200
+++ docbook-xsl-1.74.0/fo/param.xsl	2008-08-06 13:32:46.000000000 +0200
@@ -133,8 +133,8 @@
 <xsl:attribute-set name="component.title.properties">
   <xsl:attribute name="keep-with-next.within-column">always</xsl:attribute>
   <xsl:attribute name="space-before.optimum"><xsl:value-of select="concat($body.font.master, 'pt')"/></xsl:attribute>
-  <xsl:attribute name="space-before.minimum"><xsl:value-of select="concat($body.font.master, 'pt * 0.8')"/></xsl:attribute>
-  <xsl:attribute name="space-before.maximum"><xsl:value-of select="concat($body.font.master, 'pt * 1.2')"/></xsl:attribute>
+  <xsl:attribute name="space-before.minimum"><xsl:value-of select="concat($body.font.master*0.8, 'pt')"/></xsl:attribute>
+  <xsl:attribute name="space-before.maximum"><xsl:value-of select="concat($body.font.master*1.2, 'pt')"/></xsl:attribute>
   <xsl:attribute name="hyphenate">false</xsl:attribute>
   <xsl:attribute name="text-align">
     <xsl:choose>
