--- liborigin2-06042009.org/liborigin.pro	2009-03-26 15:49:34.000000000 +0800
+++ liborigin2-06042009/liborigin.pro	2009-11-25 15:28:45.003182843 +0800
@@ -7,9 +7,9 @@
 
 DESTDIR      = ./
 
-#INCLUDEPATH += boost_1_33_0
-LIBS        += -lboost_date_time
-LIBS        += -lboost_thread
+INCLUDEPATH += $$BOOSTINC
+LIBS        += -lboost_date_time-mt
+LIBS        += -lboost_thread-mt
 
 HEADERS += endianfstream.hh
 HEADERS += logging.hpp
@@ -23,4 +23,20 @@
 SOURCES += OriginParser.cpp
 SOURCES += OriginDefaultParser.cpp
 SOURCES += Origin750Parser.cpp
-
+# Install directives
+
+INSTALLBASE = /usr
+
+target.path = $$LIBDIR
+headers.path = $$INSTALLBASE/include/liborigin2
+documentation.path = $$ORIDOCDIR
+headers.files = $$HEADERS tree.hh
+documentation.files += doc/html  \
+			copying \
+			readme \
+			FORMAT
+INSTALLS = target headers documentation
+
+# Avoid link against Qt core and gui modules
+
+QT -= core gui
