diff -Nuar a/liborigin2.pro b/liborigin2.pro
--- a/liborigin2.pro	2010-10-29 13:27:52.000000000 +0300
+++ b/liborigin2.pro	2013-05-19 10:03:11.855302287 +0300
@@ -8,7 +8,7 @@
 DESTDIR      = ./
 
 # Path to the folder where the header files of the BOOST C++ libraries are installed
-INCLUDEPATH += ../boost
+INCLUDEPATH += $$BOOSTINC
 
 HEADERS += endianfstream.hh \
 	   tree.hh \
@@ -32,3 +32,21 @@
 	   Origin750Parser.cpp \
 	   Origin800Parser.cpp \
 	   Origin810Parser.cpp
+	   
+# Install directives
+
+INSTALLBASE = /usr
+
+target.path = $$LIBDIR
+headers.path = $$INSTALLBASE/include/liborigin2
+documentation.path = $$ORIDOCDIR
+headers.files = $$HEADERS tree.hh
+documentation.files += doc/html  \
+			copying \
+			readme \
+			FORMAT
+INSTALLS = target headers documentation
+
+# Avoid link against Qt core and gui modules
+
+QT -= core gui
