diff -Nur crda-1.0.1.orig/Makefile crda-1.0.1/Makefile
--- crda-1.0.1.orig/Makefile	2009-01-24 23:39:26.000000000 +0200
+++ crda-1.0.1/Makefile	2009-03-30 18:37:39.000000000 +0300
@@ -14,7 +14,7 @@
 UDEV_LEVEL=$(CRDA_UDEV_LEVEL)-
 # You can customize this if your distributions uses
 # a different location.
-UDEV_RULE_DIR?=/lib/udev/rules.d/
+UDEV_RULE_DIR?=/lib/udev/rules.d
 
 # Used locally to retrieve all pubkeys during build time
 PUBKEY_DIR=pubkeys
@@ -119,8 +119,5 @@
 	$(Q)$(INSTALL) -m 755 -t $(DESTDIR)/$(SBINDIR) regdbdump
 	$(NQ) '  INSTALL  $(UDEV_LEVEL)regulatory.rules'
 	$(Q)$(MKDIR) $(DESTDIR)/$(UDEV_RULE_DIR)/
-	@# This removes the old rule you may have, we were not
-	@# putting it in the right place.
-	$(Q)rm -f $(DESTDIR)/etc/udev/rules.d/regulatory.rules
 	$(Q)sed 's:$$(SBINDIR):$(SBINDIR):' udev/regulatory.rules > udev/regulatory.rules.parsed
 	$(Q)ln -sf regulatory.rules.parsed udev/$(UDEV_LEVEL)regulatory.rules

