Index: crda-1.1.1/udev/regulatory.rules
===================================================================
--- crda-1.1.1.orig/udev/regulatory.rules
+++ crda-1.1.1/udev/regulatory.rules
@@ -2,4 +2,5 @@
 # For more information see:
 # http://wireless.kernel.org/en/developers/Regulatory/CRDA
 
+SUBSYSTEM=="ieee80211", ACTION=="add", RUN+="/sbin/setregdomain"
 KERNEL=="regulatory*", ACTION=="change", SUBSYSTEM=="platform", RUN+="$(SBINDIR)crda"
