Index: wvdial-1.61/wvrules-posix.mk
===================================================================
--- wvdial-1.61.orig/wvrules-posix.mk
+++ wvdial-1.61/wvrules-posix.mk
@@ -30,7 +30,7 @@ INCFLAGS=$(addprefix -I,$(WVSTREAMS_INC)
 CPPFLAGS += $(CPPOPTS)
 CFLAGS += $(COPTS)
 CXXFLAGS += $(CXXOPTS)
-LDFLAGS += $(LDOPTS) -L$(WVSTREAMS_LIB)
+LDFLAGS += $(LDOPTS)
 
 # Default compiler we use for linking
 WVLINK_CC = $(CXX)
@@ -45,15 +45,6 @@ ifneq ("$(enable_warnings)", "no")
   CFLAGS+=-Wall
 endif
 
-DEBUG:=$(filter-out no 0,$(enable_debug))
-ifdef DEBUG
-  CPPFLAGS += -ggdb -DDEBUG=1 $(patsubst %,-DDEBUG_%,$(DEBUG))
-  LDFLAGS += -ggdb
-else
-  CPPFLAGS += -DDEBUG=0
-  LDFLAGS += 
-endif
-
 define wvlink_ar
 	$(LINK_MSG)set -e; rm -f $1 $(patsubst %.a,%.libs,$1); \
 	echo $2 $($1-EXTRA) >$(patsubst %.a,%.libs,$1); \
@@ -93,4 +84,4 @@ CXX: FORCE
 	@CC="$(CXX)" CFLAGS="$(CXXFLAGS)" CPPFLAGS="$(CPPFLAGS)" \
 	  $(WVSTREAMS_SRC)/gen-cc CXX cc
 
-wvlink=$(LINK_MSG)$(WVLINK_CC) $(LDFLAGS) $($1-LDFLAGS) -o $1 $(filter %.o %.a %.so, $2) $($1-LIBS) $(XX_LIBS) $(LDLIBS) $(PRELIBS) $(LIBS)
+wvlink=$(LINK_MSG)$(WVLINK_CC) $(LDFLAGS) $($1-LDFLAGS) $(filter %.o %.a %.so, $2) $($1-LIBS) $(XX_LIBS) $(LDLIBS) $(PRELIBS) $(LIBS) -o $1
