Index: scidavis-0.2.4/scidavis/scidavis.desktop
===================================================================
--- scidavis-0.2.4.orig/scidavis/scidavis.desktop
+++ scidavis-0.2.4/scidavis/scidavis.desktop
@@ -2,10 +2,13 @@
 Type=Application
 Version=1.0
 Name=SciDAVis
+GenericName=Scientific Data Analysis and Visualization
+GenericName[tr]=Bilimsel Veri Analiz Aracı
 Comment=Application for scientific data analysis and visualization
 Comment[de]=Programm zur wissenschaftlichen Datenauswertung und Visualisierung
 Comment[fr]=Outil d'analyse scientifique de données
 Comment[pt_BR]=Programa para visualização e análise da dados científico
+Comment[tr]=Bilimsel Veri Analiz Aracı
 Icon=scidavis
 Exec=scidavis
 Terminal=false
