*** data/artha.desktop~	2012-11-14 17:21:59.992389490 +0100
--- data/artha.desktop	2012-11-14 17:24:35.802438871 +0100
***************
*** 2,11 ****
  Version=1.0
  Name=Artha
  GenericName=Thesaurus
  Comment=Handy thesaurus based on WordNet
  Exec=artha
  Icon=artha
  Terminal=false
  Type=Application
  StartupNotify=true
! Categories=Utility;Languages;Literature;
--- 2,15 ----
  Version=1.0
  Name=Artha
  GenericName=Thesaurus
+ GenericName[pl]=Tezaurus
+ GenericName[tr]=Eşanlamlılar Sözlüğü
  Comment=Handy thesaurus based on WordNet
+ Comment[pl]=Angielski słownik synonimów oparty o WordNet
+ Comment[tr]=Wordnet tabanlı kullanışlı eşanlamlılar sözlüğü
  Exec=artha
  Icon=artha
  Terminal=false
  Type=Application
  StartupNotify=true
! Categories=Utility;Languages;Literature;Education;
