Index: polkit-0.96/src/polkitbackend/polkitbackendlocalauthority.c
===================================================================
--- polkit-0.96.orig/src/polkitbackend/polkitbackendlocalauthority.c
+++ polkit-0.96/src/polkitbackend/polkitbackendlocalauthority.c
@@ -474,7 +474,8 @@ polkit_backend_local_authority_get_admin
         }
       else if (POLKIT_IS_UNIX_GROUP (identity))
         {
-          ret = g_list_concat (ret, get_users_in_group (identity, FALSE));
+          /* Keep 'root' user as well */
+          ret = g_list_concat (ret, get_users_in_group (identity, TRUE));
         }
       else
         {
