Index: polkit-0.99/src/polkit/polkitauthority.c
===================================================================
--- polkit-0.99.orig/src/polkit/polkitauthority.c
+++ polkit-0.99/src/polkit/polkitauthority.c
@@ -23,6 +23,8 @@
 #  include "config.h"
 #endif
 
+#include <locale.h>
+
 #include "polkitauthorizationresult.h"
 #include "polkitcheckauthorizationflags.h"
 #include "polkitauthority.h"
@@ -650,10 +652,13 @@ polkit_authority_enumerate_actions (Polk
 {
   g_return_if_fail (POLKIT_IS_AUTHORITY (authority));
   g_return_if_fail (cancellable == NULL || G_IS_CANCELLABLE (cancellable));
+
+  char *locale = setlocale(LC_MESSAGES, NULL);
+
   g_dbus_proxy_call (authority->proxy,
                      "EnumerateActions",
                      g_variant_new ("(s)",
-                                    ""), /* TODO: use system locale */
+                         locale ? locale : ""),
                      G_DBUS_CALL_FLAGS_NONE,
                      -1,
                      cancellable,
