Index: smplayer-0.6.8/smplayer.desktop
===================================================================
--- smplayer-0.6.8.orig/smplayer.desktop
+++ smplayer-0.6.8/smplayer.desktop
@@ -10,6 +10,7 @@ Comment[nl]=Een uitstekend frontend voor
 Comment[pl]=Świetna nakładka na Mplayer
 Comment[pt]=Um óptimo interface gráfico para o MPlayer
 Comment[ru]=Мощный интерфейс к MPlayer
+Comment[tr]=Gelişmiş bir MPlayer önyüzü
 Comment[uk]=Потужний інтерфейс до MPlayer
 Exec=smplayer %U
 GenericName=Media Player
@@ -18,6 +19,7 @@ GenericName[es]=Reproductor multimedia
 GenericName[hu]=Médialejátszó
 GenericName[pt]=Reprodutor de Média
 GenericName[ru]=Медиаплеер
+GenericName[tr]=Çokluortam Oynatıcı
 GenericName[uk]=Програвач медіа
 Icon=smplayer
 MimeType=audio/ac3;audio/mp4;audio/mpeg;audio/vnd.rn-realaudio;audio/vorbis;audio/x-adpcm;audio/x-matroska;audio/x-mp2;audio/x-mp3;audio/x-ms-wma;audio/x-vorbis;audio/x-wav;audio/mpegurl;audio/x-mpegurl;audio/x-pn-realaudio;audio/x-scpls;video/avi;video/mp4;video/flv;video/mpeg;video/quicktime;video/vnd.rn-realvideo;video/x-matroska;video/x-ms-asf;video/x-msvideo;video/x-ms-wmv;video/x-ogm;video/x-theora;
