diff -Naur smplayer-0.6.7_20090703-orig/smplayer.desktop smplayer-0.6.7_20090703/smplayer.desktop
--- smplayer-0.6.7_20090703-orig/smplayer.desktop	2009-07-17 14:05:03.125530048 +0300
+++ smplayer-0.6.7_20090703/smplayer.desktop	2009-07-17 14:38:51.968427462 +0300
@@ -22,7 +22,8 @@
 GenericName[uk]=Потужний інтерфейс до MPlayer
 GenericName[tr]=Gelişmiş bir MPlayer önyüzü
 Icon=smplayer
-MimeType=audio/ac3;audio/mp4;audio/mpeg;audio/vnd.rn-realaudio;audio/vorbis;audio/x-adpcm;audio/x-matroska;audio/x-mp2;audio/x-mp3;audio/x-ms-wma;audio/x-vorbis;audio/x-wav;audio/mpegurl;audio/x-mpegurl;audio/x-pn-realaudio;audio/x-scpls;video/avi;video/mp4;video/flv;video/mpeg;video/quicktime;video/vnd.rn-realvideo;video/x-matroska;video/x-ms-asf;video/x-msvideo;video/x-ms-wmv;video/x-ogm;video/x-theora;
+MimeType=video/avi;video/mp4;video/flv;video/mpeg;video/quicktime;video/vnd.rn-realvideo;video/x-matroska;video/x-ms-asf;video/x-msvideo;video/x-ms-wmv;video/x-ogm;video/x-theora;
+InitialPreference=14
 Name=SMPlayer
 Type=Application
 X-KDE-StartupNotify=false
