diff -Nur libgksu-1.9.4/gksu.schemas.in libgksu-1.9.4.new/gksu.schemas.in
--- libgksu-1.9.4/gksu.schemas.in	2006-08-01 10:51:39.000000000 +0200
+++ libgksu-1.9.4.new/gksu.schemas.in	2006-08-01 10:52:01.000000000 +0200
@@ -86,7 +86,7 @@
       <applyto>/apps/gksu/save-to-keyring</applyto>
       <owner>gksu</owner>
       <type>bool</type>
-      <default>true</default>
+      <default>false</default>
       <locale name="C">
 	<short>Save password to gnome-keyring</short>
 	<long>
