diff -Nur libgksu-1.9.4/gksu.schemas.in libgksu-1.9.4.new/gksu.schemas.in
--- libgksu-1.9.4/gksu.schemas.in	2006-07-04 15:56:59.000000000 +0200
+++ libgksu-1.9.4.new/gksu.schemas.in	2006-07-04 15:57:46.000000000 +0200
@@ -70,7 +70,7 @@
       <applyto>/apps/gksu/display-no-pass-info</applyto>
       <owner>gksu</owner>
       <type>bool</type>
-      <default>true</default>
+      <default>false</default>
       <locale name="C">
 	<short>Display information message when no password is needed</short>
 	<long>
