--- libgksu/Makefile.am.OSmanOS		2012-11-04 14:51:41.000000000 +0200
+++ libgksu/Makefile.am			2012-11-04 14:54:18.557075082 +0200
@@ -22,8 +22,8 @@
 pkgconfigdir = ${libdir}/pkgconfig
 pkgconfig_DATA = libgksu2.pc

-pkglibdir = ${libdir}/${PACKAGE}
-pkglib_PROGRAMS = gksu-run-helper
+gksulibdir = ${libdir}/${PACKAGE}
+gksulib_PROGRAMS = gksu-run-helper
 gksu_run_helper_LDFLAGS = `pkg-config --libs glib-2.0`
 gksu_run_helper_SOURCES = gksu-run-helper.c