--- Makefile.in.orig	2009-11-10 11:38:25.000000000 +0200
+++ Makefile.in	2009-11-10 11:38:35.000000000 +0200
@@ -189,7 +189,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 gtviewer_SOURCES = gtviewer.c
-INCLUDES = @GTK_CFLAGS@ -DHELP_FILE=\"$(pkgdatadir)/gtviewer.txt\" -Wall
+INCLUDES = @GTK_CFLAGS@ -DHELP_FILE=\"$(docdir)/gtviewer.txt\" -Wall
 LDADD = @GTK_LIBS@
 EXTRA_DIST = gtviewer.desktop gtviewer_logo.png icon.h gtviewer.texi fdl.texi
 icon_DATA = gtviewer_logo.png
