diff -up animorph-0.3/include/animorph/Hotspot.h.gcc43 animorph-0.3/include/animorph/Hotspot.h
--- animorph-0.3/include/animorph/Hotspot.h.gcc43	2007-11-25 10:34:57.000000000 +0100
+++ animorph-0.3/include/animorph/Hotspot.h	2008-01-04 17:40:58.000000000 +0100
@@ -36,6 +36,7 @@
 #include <vector>
 #include <iostream>
 #include <fstream>
+#include <cstring>
 #include "FileReader.h"
 
 namespace Animorph {
diff -up animorph-0.3/include/animorph/BodySettings.h.gcc43 animorph-0.3/include/animorph/BodySettings.h
--- animorph-0.3/include/animorph/BodySettings.h.gcc43	2007-12-02 23:30:40.000000000 +0100
+++ animorph-0.3/include/animorph/BodySettings.h	2008-01-04 17:40:58.000000000 +0100
@@ -34,6 +34,7 @@
 
 #include <iostream>
 #include <fstream>
+#include <cstring>
 #include <map>
 #include <vector>
 #include "FileReader.h"
diff -up animorph-0.3/include/animorph/FaceGroup.h.gcc43 animorph-0.3/include/animorph/FaceGroup.h
--- animorph-0.3/include/animorph/FaceGroup.h.gcc43	2007-12-02 23:30:40.000000000 +0100
+++ animorph-0.3/include/animorph/FaceGroup.h	2008-01-04 17:40:58.000000000 +0100
@@ -29,6 +29,7 @@
 
 #include <map>
 #include <ios>
+#include <cstring>
 #include "Face.h"
 #include "FaceVector.h"
 #include "FileWriter.h"
diff -up animorph-0.3/include/animorph/util.h.gcc43 animorph-0.3/include/animorph/util.h
--- animorph-0.3/include/animorph/util.h.gcc43	2007-11-25 10:34:57.000000000 +0100
+++ animorph-0.3/include/animorph/util.h	2008-01-04 17:40:58.000000000 +0100
@@ -37,6 +37,7 @@
 #include <iomanip>
 #include <vector>
 #include <iostream>
+#include <cstdlib>
 #include "Vector3.h"
 #include "Vertex.h"
 #include "VertexVector.h"
