--- openttd.orig/src/table/misc_settings.ini	2012-04-23 21:18:40.391693579 +0200
+++ openttd/src/table/misc_settings.ini	2012-04-23 21:19:38.821691445 +0200
@@ -128,35 +128,35 @@
 name     = ""small_font""
 type     = SLE_STRB
 var      = _freetype.small_font
-def      = NULL
+def      = ""DejaVu Sans""
 
 [SDTG_STR]
 ifdef    = WITH_FREETYPE
 name     = ""medium_font""
 type     = SLE_STRB
 var      = _freetype.medium_font
-def      = NULL
+def      = ""DejaVu Sans""
 
 [SDTG_STR]
 ifdef    = WITH_FREETYPE
 name     = ""large_font""
 type     = SLE_STRB
 var      = _freetype.large_font
-def      = NULL
+def      = ""DejaVu Sans""
 
 [SDTG_STR]
 ifdef    = WITH_FREETYPE
 name     = ""mono_font""
 type     = SLE_STRB
 var      = _freetype.mono_font
-def      = NULL
+def      = ""DejaVu Sans Mono""
 
 [SDTG_VAR]
 ifdef    = WITH_FREETYPE
 name     = ""small_size""
 type     = SLE_UINT
 var      = _freetype.small_size
-def      = 8
+def      = 10
 min      = 0
 max      = 72
 
@@ -165,7 +165,7 @@
 name     = ""medium_size""
 type     = SLE_UINT
 var      = _freetype.medium_size
-def      = 10
+def      = 12
 min      = 0
 max      = 72
 
