diff -Nuar a/src/table/misc_settings.ini b/src/table/misc_settings.ini
--- a/src/table/misc_settings.ini	2013-07-27 18:09:31.000000000 +0300
+++ b/src/table/misc_settings.ini	2013-09-29 20:33:56.704790331 +0300
@@ -136,35 +136,35 @@
 name     = ""small_font""
 type     = SLE_STRB
 var      = _freetype.small.font
-def      = NULL
+def      = ""DejaVu Sans""
 
 [SDTG_STR]
 ifdef    = WITH_FREETYPE
 name     = ""medium_font""
 type     = SLE_STRB
 var      = _freetype.medium.font
-def      = NULL
+def      = ""DejaVu Sans""
 
 [SDTG_STR]
 ifdef    = WITH_FREETYPE
 name     = ""large_font""
 type     = SLE_STRB
 var      = _freetype.large.font
-def      = NULL
+def      = ""DejaVu Sans""
 
 [SDTG_STR]
 ifdef    = WITH_FREETYPE
 name     = ""mono_font""
 type     = SLE_STRB
 var      = _freetype.mono.font
-def      = NULL
+def      = ""DejaVu Sans Mono""
 
 [SDTG_VAR]
 ifdef    = WITH_FREETYPE
 name     = ""small_size""
 type     = SLE_UINT
 var      = _freetype.small.size
-def      = 0
+def      = 12
 min      = 0
 max      = 72
 
@@ -173,7 +173,7 @@
 name     = ""medium_size""
 type     = SLE_UINT
 var      = _freetype.medium.size
-def      = 0
+def      = 14
 min      = 0
 max      = 72
 
