diff -Nuar lapack-3.2.1.orig/BLAS/SRC/Makefile lapack-3.2.1/BLAS/SRC/Makefile
--- lapack-3.2.1.orig/BLAS/SRC/Makefile	2010-08-16 13:40:39.106209859 +0300
+++ lapack-3.2.1/BLAS/SRC/Makefile	2010-08-16 13:46:46.288212746 +0300
@@ -163,6 +163,12 @@
 
 FRC:
 	@FRC=$(FRC)
+shared:
+	rm -f *.o
+	ar x ../../blas_LINUX.a
+	$(CC) $(CFLAGS) -fPIC $(LDFLAGS) *.o -lgfortran -lc -lm -shared -Wl,-soname,libblas.so.3 -o libblas.so.3.2.1
+	ln -s libblas.so.3.2.1 libblas.so.3
+	ln -s libblas.so.3.2.1 libblas.so
 
 clean:
 	rm -f *.o
diff -Nuar lapack-3.2.1.orig/Makefile lapack-3.2.1/Makefile
--- lapack-3.2.1.orig/Makefile	2010-08-16 13:40:39.130213129 +0300
+++ lapack-3.2.1/Makefile	2010-08-16 13:44:58.463216507 +0300
@@ -8,8 +8,7 @@
 
 all: lapack_install lib lapack_testing blas_testing
 
-lib: lapacklib tmglib
-#lib: blaslib variants lapacklib tmglib
+lib: blaslib variants lapacklib tmglib
 
 clean: cleanlib cleantesting cleanblas_testing 
 
@@ -18,10 +17,10 @@
 	  ./testdlamch; ./testsecond; ./testdsecnd; ./testversion )
 
 blaslib:
-	( cd BLAS/SRC; $(MAKE) )
+	( cd BLAS/SRC; $(MAKE); $(MAKE) shared )
 
 lapacklib:	lapack_install
-	( cd SRC; $(MAKE) )
+	( cd SRC; $(MAKE); $(MAKE) shared )
 
 variants:
 	( cd SRC/VARIANTS ; $(MAKE))
diff -Nuar lapack-3.2.1.orig/SRC/Makefile lapack-3.2.1/SRC/Makefile
--- lapack-3.2.1.orig/SRC/Makefile	2010-08-16 13:40:39.165210958 +0300
+++ lapack-3.2.1/SRC/Makefile	2010-08-16 13:53:41.417213085 +0300
@@ -408,6 +408,13 @@
 FRC:
 	@FRC=$(FRC)
 
+shared:
+	rm -rf *.o
+	ar x ../lapack_LINUX.a
+	$(CC) $(CFLAGS) -fPIC $(LDFLAGS) *.o -L../BLAS/SRC -lblas -lgfortran -lc -lm -shared -Wl,-soname,liblapack.so.3 -o liblapack.so.3.2.1
+	ln -s liblapack.so.3.2.1 liblapack.so.3
+	ln -s liblapack.so.3.2.1 liblapack.so
+
 clean:
 	rm -f *.o
 
