--- lapack-3.3.1/BLAS/SRC/CMakeLists.txt.sover	2011-05-18 08:36:12.000000000 +0200
+++ lapack-3.3.1/BLAS/SRC/CMakeLists.txt	2011-05-18 08:38:58.000000000 +0200
@@ -141,4 +141,8 @@
   target_link_libraries(blas m)
 endif()
 target_link_libraries(blas)
+set_target_properties(blas PROPERTIES
+    VERSION ${LAPACK_VERSION}
+    SOVERSION ${CPACK_PACKAGE_VERSION_MAJOR}
+)
 lapack_install_library(blas)
--- lapack-3.3.1/SRC/CMakeLists.txt.sover	2011-05-18 08:35:33.000000000 +0200
+++ lapack-3.3.1/SRC/CMakeLists.txt	2011-05-18 08:39:42.000000000 +0200
@@ -393,4 +393,8 @@
 endif()
 add_library(lapack ${ALLOBJ} ${ALLXOBJ})
 target_link_libraries(lapack ${BLAS_LIBRARIES})
+set_target_properties(lapack PROPERTIES
+  VERSION ${LAPACK_VERSION}
+  SOVERSION ${CPACK_PACKAGE_VERSION_MAJOR}
+)
 lapack_install_library(lapack)
--- lapack-3.3.1/CMakeLists.txt.sover	2011-05-18 09:17:03.000000000 +0200
+++ lapack-3.3.1/CMakeLists.txt	2011-05-18 09:17:46.000000000 +0200
@@ -1,5 +1,7 @@
 cmake_minimum_required(VERSION 2.6)
 project(LAPACK Fortran)
+set(LAPACK_VERSION 3.3.1)
+set(CPACK_PACKAGE_VERSION_MAJOR 3)
 
 # Configure the warning and code coverage suppression file
 configure_file( 
@@ -115,8 +117,6 @@
 SET(CPACK_PACKAGE_NAME "LAPACK")
 SET(CPACK_PACKAGE_VENDOR "University of Tennessee, Univ. of California Berkeley, Univ. of Colorado Denver and NAG Ltd")
 SET(CPACK_PACKAGE_DESCRIPTION_SUMMARY "LAPACK- Linear Algebra Package")
-set(LAPACK_VERSION 3.3.1)
-set(CPACK_PACKAGE_VERSION_MAJOR 3)
 set(CPACK_PACKAGE_VERSION_MINOR 3)
 set(CPACK_PACKAGE_VERSION_PATCH 1)
 set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE")
