--- Makefile.am~	2012-11-24 19:26:09.000000000 +0800
+++ Makefile.am	2012-12-01 06:31:08.238738370 +0800
@@ -483,7 +483,7 @@
 bin_PROGRAMS += pcretest
 pcretest_SOURCES = pcretest.c
 pcretest_CFLAGS = $(AM_CFLAGS)
-pcretest_LDADD = $(LIBREADLINE)
+pcretest_LDADD = libpcre.la libpcreposix.la $(LIBREADLINE)
 if WITH_PCRE8
 pcretest_SOURCES += pcre_printint.c
 pcretest_LDADD += libpcre.la libpcreposix.la
@@ -510,8 +510,7 @@
 bin_PROGRAMS += pcregrep
 pcregrep_SOURCES = pcregrep.c
 pcregrep_CFLAGS = $(AM_CFLAGS)
-pcregrep_LDADD = $(LIBZ) $(LIBBZ2)
-pcregrep_LDADD += libpcre.la libpcreposix.la
+pcregrep_LDADD = libpcre.la libpcreposix.la $(LIBZ) $(LIBBZ2)
 if WITH_GCOV
 pcregrep_CFLAGS += $(GCOV_CFLAGS)
 pcregrep_LDADD += $(GCOV_LIBS)
