diff -Nur libxml2-2.7.8-old/configure libxml2-2.7.8/configure
--- libxml2-2.7.8-old/configure	2010-12-01 12:10:29.666782165 +0200
+++ libxml2-2.7.8/configure	2010-12-01 12:10:36.271656056 +0200
@@ -11405,6 +11405,8 @@
 
 
 VERSION_SCRIPT_FLAGS=
+echo host $host
+echo lt_cv_prog_gnu_ld $lt_cv_prog_gnu_ld
 # lt_cv_prog_gnu_ld is from libtool 2.+
 if test "$lt_cv_prog_gnu_ld" = yes; then
   VERSION_SCRIPT_FLAGS=-Wl,--version-script=
@@ -11414,7 +11416,7 @@
   esac
 fi
 
- if test -z "$VERSION_SCRIPT_FLAGS"; then
+ if test -n "$VERSION_SCRIPT_FLAGS"; then
   USE_VERSION_SCRIPT_TRUE=
   USE_VERSION_SCRIPT_FALSE='#'
 else
diff -Nur libxml2-2.7.8-old/configure.in libxml2-2.7.8/configure.in
--- libxml2-2.7.8-old/configure.in	2010-12-01 12:10:29.691781723 +0200
+++ libxml2-2.7.8/configure.in	2010-12-01 12:10:36.267657222 +0200
@@ -75,6 +75,8 @@
 dnl then add it
 dnl
 VERSION_SCRIPT_FLAGS=
+echo host $host
+echo lt_cv_prog_gnu_ld $lt_cv_prog_gnu_ld
 # lt_cv_prog_gnu_ld is from libtool 2.+
 if test "$lt_cv_prog_gnu_ld" = yes; then
   VERSION_SCRIPT_FLAGS=-Wl,--version-script=
@@ -82,9 +84,12 @@
   case $host in
   *-*-sunos*) VERSION_SCRIPT_FLAGS="-Wl,-M -Wl,";;
   esac
+  case $host in
+  *-*-linux*) VERSION_SCRIPT_FLAGS="-Wl,--version-script=";;
+  esac
 fi
 AC_SUBST(VERSION_SCRIPT_FLAGS)
-AM_CONDITIONAL([USE_VERSION_SCRIPT], [test -z "$VERSION_SCRIPT_FLAGS"])
+AM_CONDITIONAL([USE_VERSION_SCRIPT], [test -n "$VERSION_SCRIPT_FLAGS"])
 
 dnl
 dnl We process the AC_ARG_WITH first so that later we can modify
