From 4517499deabb935a286cfadb01030aeed4ba89d1 Mon Sep 17 00:00:00 2001
From: Frederic Crozat <fcrozat@mandriva.com>
Date: Sat, 11 Jul 2009 14:31:05 +0200
Subject: [PATCH] fix 64bits compilation

---
 tools/makedep/cppsetup.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/tools/makedep/cppsetup.cpp b/tools/makedep/cppsetup.cpp
index d8d1026..f60c729 100644
--- a/tools/makedep/cppsetup.cpp
+++ b/tools/makedep/cppsetup.cpp
@@ -195,7 +195,7 @@ static long my_eval_variable (IfParser *ip, const char *var, int len)
     }
     while (s);
 
-    eol = strchr (var, 0);
+    eol = strchr ((char *) var, 0);
 
     // Skip parentheses which Microsoft likes so much
     if (*var == '(')
-- 
1.6.3.3

