--- CMakeLists.txt~	2012-06-10 22:19:35.000000000 +0800
+++ CMakeLists.txt	2012-09-08 15:02:31.644054453 +0800
@@ -74,7 +74,7 @@
  
 # define destination dirs (relative to prefix)
 SET(BINDIR bin)
-SET(LIBDIR lib)
+SET(LIBDIR ${CMAKE_INSTALL_LIBDIR})
 SET(INCLUDEDIR include)
 SET(DATADIR share/lensfun)
 SET(DOCDIR share/doc/lensfun-${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_MICRO}.${VERSION_BUGFIX})
