Index: libdbusmenu-qt-0.3.2/CMakeLists.txt
===================================================================
--- libdbusmenu-qt-0.3.2.orig/CMakeLists.txt
+++ libdbusmenu-qt-0.3.2/CMakeLists.txt
@@ -44,6 +44,10 @@ include_directories(
     ${QT_QTGUI_INCLUDE_DIR}
     )
 
+link_directories(
+    ${QT_LIBRARY_DIR}
+    )
+
 configure_file(dbusmenu-qt.pc.in ${CMAKE_BINARY_DIR}/dbusmenu-qt.pc @ONLY)
 
 install(FILES ${CMAKE_BINARY_DIR}/dbusmenu-qt.pc
