--- a/src/utils_glx.h	2012-10-05 17:02:58.000000000 +0200
+++ b/src/utils_glx.h	2012-10-11 14:36:29.000000000 +0200
@@ -139,6 +139,8 @@
 gl_create_texture(GLenum target, GLenum format, unsigned int width, unsigned int height)
     attribute_hidden;
 
+typedef void (APIENTRYP PFNGLMULTITEXCOORD2FPROC) (GLenum target, GLfloat s, GLfloat t);
+
 typedef struct _GLVTable GLVTable;
 struct _GLVTable {
     PFNGLXBINDTEXIMAGEEXTPROC             glx_bind_tex_image;
