diff -Naur rst2pdf-orig/styles/styles.style rst2pdf/styles/styles.style
--- rst2pdf-orig/styles/styles.style	2010-03-26 11:00:24.563880729 +0200
+++ rst2pdf/styles/styles.style	2010-03-26 11:15:17.714973043 +0200
@@ -1,6 +1,8 @@
 
   # List any fonts you would like to embed in the PDF here
   embeddedFonts: []
+    [DejaVuSans.ttf, DejaVuSans-Bold.ttf, DejaVuSans-Oblique.ttf, DejaVuSans-BoldOblique.ttf]
+    [DejaVuSansMono.ttf, DejaVuSansMono-Bold.ttf, DejaVuSansMono-Oblique.ttf, DejaVuSansMono-BoldOblique.ttf]
 
   # Default page setup.  Can be overridden by including other
   # stylesheets with -s
