diff -Nur dangerdeep-0.4.0_pre3327-old//SConstruct dangerdeep-0.4.0_pre3327/SConstruct
--- dangerdeep-0.4.0_pre3327-old//SConstruct	2011-01-06 18:59:36.839000052 +0200
+++ dangerdeep-0.4.0_pre3327/SConstruct	2011-01-06 19:03:53.982000266 +0200
@@ -199,7 +199,7 @@
 
 	# Set libpath accordingly
 	if arch == 'x86_64':
-		archlib = '/usr/lib64'
+		archlib = '/usr/lib'
 		arch = 'x86-64'
 	else:
 		archlib = '/usr/lib'
@@ -210,6 +210,12 @@
 	ccflags += ' -march=' + arch
 
 	env.Append(LINKFLAGS = '`pkg-config --libs-only-L x11`')
+	if os.environ.has_key('CXX'):
+		env.Replace(CXX = os.environ['CXX'])
+	if os.environ.has_key('CXXFLAGS'):
+		ccflags += " %s " % os.environ['CXXFLAGS']
+	if os.environ.has_key('LDFLAGS'):
+		env.Append(LINKFLAGS = os.environ['LDFLAGS'])
 	if (debug >= 4):
 		ccflags += ' -g -pg -O3' # profiling
 	elif (debug == 3):
@@ -220,7 +226,6 @@
 		env.Append(CPPDEFINES = ['DEBUG'])
 	elif (debug == 1):
 		# same as debug==0 but with DEBUG defined to make logging work (and probably other stuff)
-		ccflags += ' -g -O2'
 		env.Append(CPPDEFINES = ['DEBUG'])
 	elif (debug == -1):
 		ccflags += ' -g -O3 -mfpmath=sse -mmmx -msse -m3dnow'
@@ -234,8 +239,6 @@
 			env.Replace(CXX = "icpc")
 			ccflags += ' -ip -w '
 			build_dir = 'linux-icc'
-		else:
-			ccflags += ' -g -O2' # debug symbols will be stripped by the linker for a debian package
 
 	if (useefence >= 1):
 		extralibs += ['efence']
