diff -NrU5 libtommath-0.41.original/makefile.shared libtommath-0.41/makefile.shared
--- libtommath-0.41.original/makefile.shared	2009-07-28 21:59:47.000000000 -0600
+++ libtommath-0.41/makefile.shared	2009-07-28 22:01:32.000000000 -0600
@@ -1,11 +1,11 @@
 #Makefile for GCC
 #
 #Tom St Denis
 VERSION=0:41
 
-CC = libtool --mode=compile --tag=CC gcc
+LTCOMPILE = libtool --mode=compile --tag=CC $(CC)
 
 CFLAGS  +=  -I./ -Wall -W -Wshadow -Wsign-compare
 
 ifndef IGNORE_SPEED
 
@@ -80,23 +80,26 @@
 bn_mp_init_set_int.o bn_mp_invmod_slow.o bn_mp_prime_rabin_miller_trials.o \
 bn_mp_to_signed_bin_n.o bn_mp_to_unsigned_bin_n.o
 
 objs: $(OBJECTS)
 
+.c.o:
+	$(LTCOMPILE) $(CFLAGS) $(LDFLAGS) -o $@ -c $<
+
 $(LIBNAME):  $(OBJECTS)
-	libtool --mode=link --tag=CC gcc $(LDFLAGS) *.lo -o $(LIBNAME) -rpath $(LIBPATH) -version-info $(VERSION)
+	libtool --mode=link --tag=CC $(CC) $(LDFLAGS) *.lo -o $(LIBNAME) -rpath $(LIBPATH) -version-info $(VERSION)
 
 install: $(LIBNAME)
 	install -d -g $(GROUP) -o $(USER) $(DESTDIR)$(LIBPATH)
 	libtool --mode=install install -c $(LIBNAME) $(DESTDIR)$(LIBPATH)/$(LIBNAME)
 	install -d -g $(GROUP) -o $(USER) $(DESTDIR)$(INCPATH)
 	install -m 644 -g $(GROUP) -o $(USER) $(HEADERS) $(DESTDIR)$(INCPATH)
 
 test: $(LIBNAME) demo/demo.o
-	gcc $(CFLAGS) -c demo/demo.c -o demo/demo.o
-	libtool --mode=link gcc $(LDFLAGS) -o test demo/demo.o $(LIBNAME_S)
+	$(CC) $(CFLAGS) -c demo/demo.c -o demo/demo.o
+	libtool --mode=link $(CC) $(LDFLAGS) -o test demo/demo.o $(LIBNAME_S)
 	
 mtest: test	
-	cd mtest ; gcc $(CFLAGS) $(LDFLAGS) mtest.c -o mtest
+	cd mtest ; $(CC) $(CFLAGS) $(LDFLAGS) mtest.c -o mtest
         
 timing: $(LIBNAME)
-	libtool --mode=link gcc $(CFLAGS) $(LDFLAGS) -DTIMER demo/timing.c $(LIBNAME_S) -o ltmtest
+	libtool --mode=link $(CC) $(CFLAGS) $(LDFLAGS) -DTIMER demo/timing.c $(LIBNAME_S) -o ltmtest
