diff -up ATLAS/CONFIG/src/SpewMakeInc.c.melf ATLAS/CONFIG/src/SpewMakeInc.c
--- ATLAS/CONFIG/src/SpewMakeInc.c.melf	2011-05-14 11:33:24.000000000 -0600
+++ ATLAS/CONFIG/src/SpewMakeInc.c	2012-08-09 10:52:28.051926489 -0600
@@ -665,9 +665,9 @@ main(int nargs, char **args)
    if (MachIsX86(mach))
    {
       if (ptrbits == 32)
-         fprintf(fpout, " -melf_i386");
+         fprintf(fpout, " -Wl,-melf_i386");
       else if (ptrbits == 64)
-         fprintf(fpout, " -melf_x86_64");
+         fprintf(fpout, " -Wl,-melf_x86_64");
       if (OS == OSFreeBSD)
          fprintf(fpout, "_fbsd");
    }
