--- configure.ac.orig		2011-08-01 14:25:25.000000000 +0530
+++ configure.ac		2011-08-14 18:55:45.560334275 +0530
@@ -218,8 +218,10 @@
 dnl ---------------------------------------------------------------------------
 dnl - Build VALA support
 dnl ---------------------------------------------------------------------------
-if test x$found_introspection != xno; then
+AC_ARG_ENABLE(vala, AS_HELP_STRING([--enable-vala],[build vala bindings]),
+	      enable_vala=$enableval,enable_vala=no)
+if test x$found_introspection != xno -a x$enable_vala != xno; then
	AC_PATH_PROG([VAPIGEN], [vapigen], [])
 	if test "x$VAPIGEN" = "x"; then
 		has_vapigen="no"
 		AC_MSG_WARN([vapigen not found, will not build Vala binding])
@@ -228,7 +230,9 @@
 	fi
 else
 	has_vapigen="no"
-	AC_MSG_WARN([Not introspection found, will not build Vala binding])
+	if test x$found_introspection = xno; then
+		AC_MSG_WARN([introspection not found, will not build Vala binding])
+	fi
 fi
 AM_CONDITIONAL(HAVE_VAPIGEN, [test "x$has_vapigen" = "xyes"])
 
