--- psutils/Makefile.unix.orig	Wed Mar 12 00:52:59 1997
+++ psutils/Makefile.unix	Wed Jun  4 10:54:59 1997
@@ -25,10 +25,10 @@
 
 OS = UNIX
 
-BINDIR = /usr/local/bin
+BINDIR = $(DESTDIR)/usr/bin
 SCRIPTDIR = $(BINDIR)
-INCLUDEDIR = /usr/local/share/psutils
-PERL = /usr/local/bin/perl
+INCLUDEDIR = $(DESTDIR)/usr/lib/psutils
+PERL = /usr/bin/perl
 
 BINMODE = 0755
 MANMODE = 0644
@@ -36,10 +36,10 @@
 INSTALL = install -c -m $(BINMODE)
 INSTALLMAN = install -c -m $(MANMODE)
 MANEXT = 1
-MANDIR = /usr/local/share/man/man$(MANEXT)
+MANDIR = $(DESTDIR)/usr/man/man$(MANEXT)
 
 CC = gcc
-CFLAGS = -DPAPER=\"$(PAPER)\" -DUNIX -O -Wall
+CFLAGS = -DPAPER=\"$(PAPER)\" -DUNIX $(RPM_OPT_FLAGS) -Wall
 
 BIN = psbook psselect pstops epsffit psnup \
 	psresize
@@ -219,28 +219,28 @@
 install: install.bin install.script install.man install.include
 
 install.bin: $(BIN)
-	-mkdir $(BINDIR)
+	-mkdir -p $(BINDIR)
 	@for i in $(BIN); do \
 		echo Installing $$i; \
-		$(INSTALL) $$i $(BINDIR); \
+		$(INSTALL) $$i -s $(BINDIR); \
 	done
 
 install.script: $(PERLSCRIPTS) $(SHELLSCRIPTS)
-	-mkdir $(SCRIPTDIR)
+	-mkdir -p $(SCRIPTDIR)
 	@for i in $(PERLSCRIPTS) $(SHELLSCRIPTS); do \
 		echo Installing $$i; \
 		$(INSTALL) $$i $(SCRIPTDIR); \
 	done
 
 install.include: $(INCLUDES)
-	-mkdir $(INCLUDEDIR)
+	-mkdir -p $(INCLUDEDIR)
 	@for i in $(INCLUDES); do \
 		echo Installing $$i; \
 		$(INSTALLMAN) $$i $(INCLUDEDIR); \
 	done
 
 install.man: $(MANPAGES)
-	-mkdir $(MANDIR)
+	-mkdir -p $(MANDIR)
 	@for i in $(MANPAGES); do \
 		echo Installing manual page for $$i; \
 		$(INSTALLMAN) $$i $(MANDIR)/$$i; \
