--- ncompress-4.2.4/compress42.c.lfs	2002-06-19 19:19:33.000000000 -0400
+++ ncompress-4.2.4/compress42.c	2002-06-19 19:20:48.000000000 -0400
@@ -130,6 +130,7 @@
  *   Add variable bit length output.
  *
  */
+#include	<unistd.h>
 #include	<stdio.h>
 #include	<fcntl.h>
 #include	<ctype.h>
@@ -168,30 +169,6 @@
 #	define	SIG_TYPE	void (*)()
 #endif
 
-#ifndef NOFUNCDEF
-	extern	void	*malloc	LARGS((int));
-	extern	void	free	LARGS((void *));
-#ifndef _IBMR2
-	extern	int		open	LARGS((char const *,int,...));
-#endif
-	extern	int		close	LARGS((int));
-	extern	int		read	LARGS((int,void *,int));
-	extern	int		write	LARGS((int,void const *,int));
-	extern	int		chmod	LARGS((char const *,int));
-	extern	int		unlink	LARGS((char const *));
-	extern	int		chown	LARGS((char const *,int,int));
-	extern	int		utime	LARGS((char const *,struct utimbuf const *));
-	extern	char	*strcpy	LARGS((char *,char const *));
-	extern	char	*strcat	LARGS((char *,char const *));
-	extern	int		strcmp	LARGS((char const *,char const *));
-	extern	unsigned strlen	LARGS((char const *));
-	extern	void	*memset	LARGS((void *,char,unsigned int));
-	extern	void	*memcpy	LARGS((void *,void const *,unsigned int));
-	extern	int		atoi	LARGS((char const *));
-	extern	void	exit	LARGS((int));
-	extern	int		isatty	LARGS((int));
-#endif
-	
 #define	MARK(a)	{ asm(" .globl M.a"); asm("M.a:"); }
 
 #ifdef	DEF_ERRNO
@@ -535,8 +512,8 @@
 char 			ofname[MAXPATHLEN];	/* Output filename								*/
 int				fgnd_flag = 0;		/* Running in background (SIGINT=SIGIGN)		*/
 
-long 			bytes_in;			/* Total number of byte from input				*/
-long 			bytes_out;			/* Total number of byte to output				*/
+long long bytes_in;			/* Total number of byte from input				*/
+long long bytes_out;			/* Total number of byte to output				*/
 
 /*
  * 8086 & 80286 Has a problem with array bigger than 64K so fake the array
