diff -Nur ladspa_sdk-old/src/makefile ladspa_sdk/src/makefile
--- ladspa_sdk-old/src/makefile	2009-04-25 01:57:34.000000000 +0300
+++ ladspa_sdk/src/makefile	2009-04-25 01:58:09.000000000 +0300
@@ -57,12 +57,12 @@
 	@echo ---------------------------------------------
 	@echo First listen to the white noise input signal:
 	@echo ---------------------------------------------
-	-sndfile-play ../snd/noise.wav
+	@echo Disabled -sndfile-play ../snd/noise.wav
 	@echo -------------------------
 	@echo Compare to plugin output.
 	@echo -------------------------
 	@echo Should be a noise band around 6000Hz, repeated quietly after 1s.
-	-sndfile-play /tmp/test.wav
+	@echo Disabled -sndfile-play /tmp/test.wav
 	@echo Test complete.
 
 install:	targets
