Author: fwang
Subject: Build fix: Fix linking of headers
Applied-Upstream: commit:30629
Forwarded: yes
Bug: http://bugzilla.abisource.com/show_bug.cgi?id=13256

--- ./configure.in.link	2011-11-12 22:48:23.000000000 +0800
+++ ./configure.in	2012-01-31 02:59:11.487363176 +0800
@@ -103,7 +103,7 @@
 opt_pkgs=''
 
 # gtk deps
-cairo_req='cairo-pdf cairo-ps pangocairo'
+cairo_req='cairo-pdf cairo-ps pangocairo cairo-fc'
 if test "x$abi_cv_gtk2" = "xyes"; then
 gtk_req="gtk+-2.0 >= $GTK2_VERSION_REQ gtk+-unix-print-2.0 librsvg-2.0 >= 2.16.0"
 else
@@ -608,10 +608,13 @@
 	SYSTEM_LIBS="-framework Cocoa"
 	AC_CHECK_PROG([CONVERT], [convert], convert, AC_MSG_ERROR(Cannot find ImageMagick convert))
 else
-	deps_pkgs="$deps_pkgs $gtk_pkgs"
+	SYSTEM_CFLAGS=$PNG_CFLAGS
+	SYSTEM_LIBS=$PNG_LIBS
+	deps_pkgs="$deps_pkgs $gtk_pkgs x11"
 fi
 
 PKG_CHECK_MODULES(DEPS,[$deps_pkgs])
+DEPS_LIBS=`$PKG_CONFIG --libs-only-L --libs-only-l $deps_pkgs`
 dnl specific gtk214 test. 	 
 	if test "$TOOLKIT" = "gtk2"; then 	 
 	        PKG_CHECK_MODULES(GTK214, gtk+-2.0 > 2.14, [abi_cv_gtk214=yes], 	 
--- ./src/Makefile.am.link	2011-11-09 01:53:10.000000000 +0800
+++ ./src/Makefile.am	2012-01-31 03:10:13.187362481 +0800
@@ -47,10 +47,9 @@
 	-DABIWORD_DATADIR="\"$(ABIWORD_DATADIR)\""
 
 libabiword_@ABIWORD_SERIES@_la_LDFLAGS = \
-	$(DEPS_LIBS) \
-	-no-undefined \
+	-Wl, --no-undefined \
 	-avoid-version \
-	-export-dynamic
+	-Wl, --export-dynamic
 
 
 if TOOLKIT_COCOA
@@ -58,7 +57,6 @@
 endif	
 
 libabiword_@ABIWORD_SERIES@_la_LIBADD = \
-	$(opt_libs) \
 	plugins/libplugins.la \
 	text/fmt/xp/libxp.la \
 	text/ptbl/xp/libxp.la \
@@ -72,7 +70,8 @@
 	af/util/$(PLATFORM)/lib$(PLATFORM).la \
 	af/util/xp/libxp.la \
 	af/ev/$(TOOLKIT)/lib$(TOOLKIT).la \
-	af/ev/xp/libxp.la
+	af/ev/xp/libxp.la \
+	$(opt_libs) $(DEPS_LIBS)
 
 if TOOLKIT_GTK_ALL
 libabiword_@ABIWORD_SERIES@_la_LIBADD += \
@@ -114,14 +113,14 @@
 	$(libabiword_@ABIWORD_SERIES@_la_CPPFLAGS)
 
 AbiWord_LDFLAGS = \
-	$(DEPS_LIBS) \
-	-no-undefined \
+	-Wl, --no-undefined \
 	-avoid-version \
-	-export-dynamic \
+	-Wl, --export-dynamic \
 	-headerpad_max_install_names 
 
 AbiWord_LDADD = \
-	libabiword-@ABIWORD_SERIES@.la
+	libabiword-@ABIWORD_SERIES@.la \
+	$(DEPS_LIBS)
 
 AbiWord_SOURCES = \
 	$(main_cpp)
