Index: abiword-2.8.6/abiword.desktop
===================================================================
--- abiword-2.8.6.orig/abiword.desktop
+++ abiword-2.8.6/abiword.desktop
@@ -6,7 +6,7 @@ Type=Application
 Categories=Office;WordProcessor;GNOME;GTK;X-Red-Hat-Base;
 StartupNotify=true
 X-Desktop-File-Install-Version=0.9
-MimeType=application/x-abiword;text/x-abiword;text/x-xml-abiword;text/plain;application/msword;application/rtf;application/vnd.plain;application/xhtml+xml;text/html;application/x-crossmark;application/docbook+xml;application/x-t602;application/vnd.oasis.opendocument.text;application/vnd.oasis.opendocument.text-template;application/vnd.oasis.opendocument.text-web;application/vnd.sun.xml.writer;application/vnd.stardivision.writer;text/vnd.wap.wml;application/wordperfect6;application/wordperfect5.1;application/vnd.wordperfect;application/x-abicollab;
+MimeType=application/x-abiword;text/x-abiword;text/x-xml-abiword;text/plain;application/msword;application/rtf;application/vnd.plain;application/xhtml+xml;text/html;application/x-crossmark;application/docbook+xml;application/x-t602;application/vnd.oasis.opendocument.text;application/vnd.oasis.opendocument.text-template;application/vnd.oasis.opendocument.text-web;application/vnd.sun.xml.writer;application/vnd.stardivision.writer;text/vnd.wap.wml;application/wordperfect6;application/wordperfect5.1;application/vnd.wordperfect;application/x-abicollab;application/x-applix-word;application/x-mswrite;application/x-kword;application/x-mif;
 Name=AbiWord
 GenericName=Word Processor
 Comment=Compose, edit, and view documents
@@ -32,3 +32,6 @@ Name[pl]=Edytor tekstu AbiWord
 Comment[pl]=Edytor tekstu AbiWord
 Name[ja]=AbiWord ワードプロセッサ
 Comment[ja]=AbiWord ワードプロセッサ
+Name[tr]=AbiWord
+Comment[tr]=Belgelerinizi görüntüleyin, değiştirin, oluşturun
+GenericName[tr]=Kelime İşlemci
