--- configure.ac.orig	2011-05-06 16:07:43.983001180 +0300
+++ configure.ac	2011-05-06 16:10:59.192001130 +0300
@@ -996,7 +996,7 @@
 # plain C developer headers/libraries which haver been installed in the
 # GNUstep hierarchy.
 #
-CPPFLAGS="$CPPFLAGS -I$GNUSTEP_SYSTEM_HEADERS -I$GNUSTEP_NETWORK_HEADERS -I$GNUSTEP_LOCAL_HEADERS"
+CPPFLAGS="$CPPFLAGS -I`pkg-config --cflags-only-I libffi | sed -e 's/^\-\I//'` -I$GNUSTEP_SYSTEM_HEADERS -I$GNUSTEP_NETWORK_HEADERS -I$GNUSTEP_LOCAL_HEADERS"
 LDFLAGS="$LDFLAGS -L$GNUSTEP_SYSTEM_LIBRARIES -L$GNUSTEP_NETWORK_LIBRARIES -L$GNUSTEP_LOCAL_LIBRARIES"
 
 #--------------------------------------------------------------------
