--- linux-2.6.35/3rdparty/heci/heci_main.c.orig	2010-10-03 14:06:55.394266389 +0300
+++ linux-2.6.35/3rdparty/heci/heci_main.c	2010-10-03 14:09:58.962555344 +0300
@@ -129,7 +129,7 @@ static int heci_release(struct inode *in
 static unsigned int heci_legacy_poll(struct file *file, poll_table * wait);
 static ssize_t heci_read(struct file *file, char __user * ubuf,
 			 size_t length, loff_t * offset);
-static int heci_ioctl(struct inode *inode, struct file *file,
+static long heci_ioctl(struct inode *inode, struct file *file,
 		      unsigned int cmd, unsigned long data);
 static ssize_t heci_write(struct file *file, const char __user * ubuf,
 			  size_t length, loff_t * offset);
@@ -160,7 +160,7 @@ static struct pci_driver heci_driver = {
 static struct file_operations heci_fops = {
 	.owner = THIS_MODULE,
 	.read = heci_read,
-	.ioctl = heci_ioctl,
+	.unlocked_ioctl = heci_ioctl,
 	.open = heci_open,
 	.release = heci_release,
 	.write = heci_write,
@@ -172,7 +172,7 @@ static struct file_operations heci_fops
  */
 static struct file_operations iamt_legacy_fops = {
 	.owner = THIS_MODULE,
-	.ioctl = heci_ioctl,
+	.unlocked_ioctl = heci_ioctl,
 	.open = heci_open,
 	.release = heci_release,
 	.poll = heci_legacy_poll,
@@ -1169,7 +1169,7 @@ fail:
 /**
  * heci_ioctl - the IOCTL function
  */
-static int heci_ioctl(struct inode *inode, struct file *file,
+static long heci_ioctl(struct inode *inode, struct file *file,
 		      unsigned int cmd, unsigned long data)
 {
 
