diff -Nurp a/3rdparty/ndiswrapper/Makefile b/3rdparty/ndiswrapper/Makefile
--- a/3rdparty/ndiswrapper/Makefile
+++ b/3rdparty/ndiswrapper/Makefile
@@ -12,12 +12,13 @@ DISTFILES = \
 # By default, we try to compile the modules for the currently running
 # kernel.  But it's the first approximation, as we will re-read the
 # version from the kernel sources.
-KVERS ?= $(shell uname -r)
+#KVERS ?= $(shell uname -r)
 
 # KBUILD is the path to the Linux kernel build tree.  It is usually the
 # same as the kernel source tree, except when the kernel was compiled in
 # a separate directory.
-KBUILD := $(shell readlink -f /lib/modules/$(KVERS)/build)
+#KBUILD := $(shell readlink -f /lib/modules/$(KVERS)/build)
+KBUILD := $(srctree)
 
 ifeq (,$(KBUILD))
 $(error Kernel tree not found - please set KBUILD to configured kernel)
@@ -33,13 +34,13 @@ ifeq (,$(wildcard $(VERSION_H)))
 VERSION_H := $(KBUILD)/include/linux/version.h
 endif
 ifeq (,$(wildcard $(VERSION_H)))
-$(error Cannot find kernel version in $(KBUILD), is it configured?)
-endif
-
+#$(error Cannot find kernel version in $(KBUILD), is it configured?)
+else
 KVERS := $(shell sed -ne 's/"//g;s/^\#define UTS_RELEASE //p' $(VERSION_H))
+endif
 
 ifeq (,$(KVERS))
-$(error Cannot find UTS_RELEASE in $(VERSION_H), has 'make modules_prepare' been called?)
+#$(error Cannot find UTS_RELEASE in $(VERSION_H), has 'make modules_prepare' been called?)
 endif
 
 INST_DIR = /lib/modules/$(KVERS)/misc
@@ -48,9 +49,10 @@ SRC_DIR=$(shell pwd)
 
 KCONFIG := $(KBUILD)/.config
 ifeq (,$(wildcard $(KCONFIG)))
-$(error No .config found in $(KBUILD), please set KBUILD to configured kernel)
-endif
+#$(error No .config found in $(KBUILD), please set KBUILD to configured kernel)
+else
 include $(KBUILD)/.config
+endif
 
 # returns of structs and unions in registers when possible, like Windows
 EXTRA_CFLAGS += -freg-struct-return
