--- linux-3.7-rc8/3rdparty/ndiswrapper/ntoskernel.h.orig	2012-12-06 19:50:11.000000000 +0200
+++ linux-3.7-rc8/3rdparty/ndiswrapper/ntoskernel.h	2012-12-06 20:32:26.688430042 +0200
@@ -207,7 +207,7 @@ void wrap_flush_wq(struct workqueue_stru
 #else // WRAP_WQ
 
 /* Compatibility for Linux before 2.6.20 where INIT_WORK takes 3 arguments */
-#if !defined(INIT_WORK_NAR) && !defined(INIT_DELAYED_WORK_DEFERRABLE)
+#if !defined(INIT_WORK_NAR) && !defined(INIT_DELAYED_WORK_DEFERRABLE) && !defined(INIT_DEFERRABLE_WORK)
 typedef void (*compat_work_func_t)(void *work);
 typedef void (*work_func_t)(struct work_struct *work);
 static inline void (INIT_WORK)(struct work_struct *work, work_func_t func)
