--- linux-3.7-rc8/net/ipv4/netfilter/ipt_IFWLOG.c.orig	2012-12-06 19:50:10.000000000 +0200
+++ linux-3.7-rc8/net/ipv4/netfilter/ipt_IFWLOG.c	2012-12-06 20:55:23.669152916 +0200
@@ -60,7 +60,7 @@ static void send_packet(const struct nl_
 
 	memcpy(nlmsg_data(nlh), (const void *) msg, sizeof(*msg));
 
-	NETLINK_CB(skb).pid = 0;  /* from kernel */
+	NETLINK_CB(skb).portid = 0;  /* from kernel */
 	NETLINK_CB(skb).dst_group = IFWLOGNLGRP_DEF;
 
 	if (nl) {
@@ -173,7 +173,7 @@ static int __init ipt_ifwlog_init(void)
 		.groups = IFWLOGNLGRP_MAX,
 	};
 
-	nl = netlink_kernel_create(&init_net, NETLINK_IFWLOG, THIS_MODULE, &cfg);
+	nl = netlink_kernel_create(&init_net, NETLINK_IFWLOG, &cfg);
 	if (!nl) {
 		PRINTR(KERN_WARNING "IFWLOG: cannot create netlink socket\n");
 		return -ENOMEM;
