--- linux-2.6.35-rc6-git-mnb0.1/net/ipv4/netfilter/ipt_psd.c.orig	2010-07-30 21:17:30.000000000 +0300
+++ linux-2.6.35-rc6-git-mnb0.1/net/ipv4/netfilter/ipt_psd.c	2010-07-31 13:29:00.623601957 +0300
@@ -98,7 +98,7 @@ static inline int hashfunc(struct in_add
 
 static bool
 ipt_psd_match(const struct sk_buff *pskb,
-	      const struct xt_match_param *match_param)
+	      struct xt_action_param *match_param)
 {
 	struct iphdr *ip_hdr;
 	struct tcphdr *tcp_hdr;
