--- linux-3.7-rc8/drivers/net/ethernet/atheros/alx/alx_main.c.orig	2012-12-06 19:50:10.000000000 +0200
+++ linux-3.7-rc8/drivers/net/ethernet/atheros/alx/alx_main.c	2012-12-06 21:19:05.500688169 +0200
@@ -121,7 +121,7 @@ void alx_hw_printk(const char *level, co
 	vaf.va = &args;
 
 	if (hw && hw->adpt && hw->adpt->netdev)
-		__netdev_printk(level, hw->adpt->netdev, &vaf);
+		netdev_printk(level, hw->adpt->netdev, "%pV", &vaf);
 	else
 		printk("%salx_hw: %pV", level, &vaf);
 
