--- ./drivers/pci/quirks.c.orig	2013-01-16 18:13:57.000000000 +0200
+++ ./drivers/pci/quirks.c	2013-01-16 18:42:24.331008402 +0200
@@ -1083,7 +1083,7 @@ DECLARE_PCI_FIXUP_EARLY(PCI_VENDOR_ID_SE
  * which cannot be supported by the pata_ali.c driver (the old drivers/ide
  * makes a compatibility effort to change the IDE interrupts).
  */
-static void __devinit quirk_ali_ide_compatibility(struct pci_dev *pdev)
+static void quirk_ali_ide_compatibility(struct pci_dev *pdev)
 {
 	u8 tmp;
 
