---
 drivers/scsi/megaraid/megaraid_mbox.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.25/drivers/scsi/megaraid/megaraid_mbox.c
===================================================================
--- linux-2.6.25.orig/drivers/scsi/megaraid/megaraid_mbox.c
+++ linux-2.6.25/drivers/scsi/megaraid/megaraid_mbox.c
@@ -301,7 +301,7 @@ MODULE_DEVICE_TABLE(pci, pci_id_table_g)
 
 
 static struct pci_driver megaraid_pci_driver = {
-	.name		= "megaraid",
+	.name		= "megaraid_mbox",
 	.id_table	= pci_id_table_g,
 	.probe		= megaraid_probe_one,
 	.remove		= __devexit_p(megaraid_detach_one),
