diff --git a/src/supertux/screen_manager.cpp b/src/supertux/screen_manager.cpp
index 493aaec..9bea87f 100644
--- a/src/supertux/screen_manager.cpp
+++ b/src/supertux/screen_manager.cpp
@@ -14,6 +14,8 @@
 //  You should have received a copy of the GNU General Public License
 //  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+#include <cstddef>
+
 #include "supertux/screen_manager.hpp"
 
 #include "audio/sound_manager.hpp"
