--- supertux-0.3.3/CMakeLists.txt.orig	2010-03-23 13:43:10.000000000 +0100
+++ supertux-0.3.3/CMakeLists.txt	2010-03-23 13:45:57.000000000 +0100
@@ -106,6 +106,21 @@ ENDIF(CURL_FOUND)
 #INCLUDE_DIRECTORIES(${ICONV_INCLUDE_DIR})
 #LINK_LIBRARIES(${ICONV_LIBRARY})
 
+find_library(SQUIRRELL_LIB squirrel
+             PATH_SUFFIXES squirrel )
+find_library(SQUIRRELL_STDLIB sqstdlib
+             PATH_SUFFIXES squirrel )
+
+if (SQUIRRELL_LIB AND SQUIRRELL_STDLIB)
+  message (STATUS "SQUIRRELL_LIB    = ${SQUIRRELL_LIB}")
+  message (STATUS "SQUIRRELL_STDLIB = ${SQUIRRELL_STDLIB}")
+else()
+  message (STATUS "SQUIRRELL_LIB    = ${SQUIRRELL_LIB}")
+  message (STATUS "SQUIRRELL_STDLIB = ${SQUIRRELL_STDLIB}")
+  message (FATAL_ERROR "squirrel libraries not found")
+endif()
+
+
 ## Find revision of WC
 MARK_AS_ADVANCED(SVNVERSION_EXECUTABLE)
 FIND_PROGRAM(SVNVERSION_EXECUTABLE svnversion)
@@ -149,17 +164,17 @@ INCLUDE(ConfigureChecks)
 
 ## Also execute instructions in src/squirrel/CMakeLists.txt
 
-ADD_SUBDIRECTORY(external/squirrel)
+#ADD_SUBDIRECTORY(external/squirrel)
 
 ## Add squirrel lib dir to search path
 
-LINK_DIRECTORIES(external/squirrel)
+#LINK_DIRECTORIES(external/squirrel)
 
 ## Some additional include paths
 
 include_directories (${CMAKE_BINARY_DIR}/)
 include_directories (${CMAKE_CURRENT_SOURCE_DIR}/src/)
-include_directories (${CMAKE_CURRENT_SOURCE_DIR}/external/squirrel/include/)
+#include_directories (${CMAKE_CURRENT_SOURCE_DIR}/external/squirrel/include/)
 include_directories (${CMAKE_CURRENT_SOURCE_DIR}/external/tinygettext/)
 include_directories (${CMAKE_CURRENT_SOURCE_DIR}/external/findlocale/)
 include_directories (${CMAKE_CURRENT_SOURCE_DIR}/external/obstack/)
@@ -333,7 +348,7 @@ ADD_DEPENDENCIES(supertux2 svnversion)
 
 ## Link supertux binary with squirrel and other libraries
 
-TARGET_LINK_LIBRARIES(supertux2 squirrel)
+TARGET_LINK_LIBRARIES(supertux2 ${SQUIRRELL_LIB} ${SQUIRRELL_STDLIB})
 TARGET_LINK_LIBRARIES(supertux2 ${SDL_LIBRARY})
 TARGET_LINK_LIBRARIES(supertux2 ${SDLIMAGE_LIBRARY})
 TARGET_LINK_LIBRARIES(supertux2 ${OPENAL_LIBRARY})
