Index: tuxpaint-0.9.21/src/tuxpaint.c
===================================================================
--- tuxpaint-0.9.21.orig/src/tuxpaint.c
+++ tuxpaint-0.9.21/src/tuxpaint.c
@@ -1130,7 +1130,7 @@ static SDL_Surface *render_text_w(TuxPai
 
     utfstr_max = (sizeof(char) * 4 * (wcslen(str) + 1));
     utfstr = (char *) malloc(utfstr_max);
-    memset(utfstr, utfstr_max, 0);
+    memset(utfstr, 0, utfstr_max);
 
     j = 0;
     for (i = 0; i < wcslen(str); i++)
