--- Makefile.orig	2011-01-27 14:07:01.953000058 +0200
+++ Makefile	2011-01-27 14:14:12.796000059 +0200
@@ -139,7 +139,7 @@
 
 # Icons and launchers:
 ICON_PREFIX:=$(DESTDIR)$(PREFIX)/share/pixmaps
-X11_ICON_PREFIX:=$(DESTDIR)$(PREFIX)/X11R6/include/X11/pixmaps
+X11_ICON_PREFIX:=$(DESTDIR)$(PREFIX)/share/pixmaps
 GNOME_PREFIX:=$(shell gnome-config --prefix 2> /dev/null)
 KDE_PREFIX:=$(shell kde-config --install apps --expandvars 2> /dev/null)
 KDE_ICON_PREFIX:=$(shell kde-config --install icon --expandvars 2> /dev/null)
