https://bugzilla.redhat.com/show_bug.cgi?id=486025

diff -up keychain-2.6.8/keychain~ keychain-2.6.8/keychain
--- keychain-2.6.8/keychain~	2006-10-25 01:09:34.000000000 +0300
+++ keychain-2.6.8/keychain	2009-03-01 01:05:18.000000000 +0200
@@ -860,7 +860,7 @@ startagent() {
                 unset start_gpg_timeout
             fi
             # the 1.9.x series of gpg spews debug on stderr
-            start_out=`gpg-agent --daemon $start_gpg_timeout 2>/dev/null`
+            start_out=`gpg-agent --daemon --write-env-file $start_gpg_timeout 2>/dev/null`
         else
             error "I don't know how to start $start_prog-agent (2)"
             return 1
