Index: keychain-2.6.8/keychain
===================================================================
--- keychain-2.6.8.orig/keychain
+++ keychain-2.6.8/keychain
@@ -96,8 +96,7 @@ die() {
 # Display the version information
 versinfo() {
     qprint
-    qprint "${GREEN}KeyChain ${version}; ${BLUE}http://www.gentoo.org/proj/en/keychain/${OFF}"
-    qprint "Copyright 2002-2004 Gentoo Foundation; Distributed under the GPL"
+    qprint "${GREEN}KeyChain ${version};${OFF}"
     qprint
 }
 
