diff -ur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2013-06-22 10:07:40.000000000 +0300
+++ b/CMakeLists.txt	2013-07-30 03:54:52.703666103 +0300
@@ -78,7 +78,7 @@
 #  INSTALLATION
 # ------------------------------------------------------------------------------
 # -- install binary
-# install( TARGETS yarock DESTINATION /usr/local/bin/ )
+# install( TARGETS yarock DESTINATION /usr/bin/ )
 
 # -- install destop file
 install(FILES data/yarock.desktop  DESTINATION /usr/share/applications/)
diff -ur a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2013-05-26 20:05:16.000000000 +0300
+++ b/src/CMakeLists.txt	2013-07-30 03:55:26.449244225 +0300
@@ -330,8 +330,8 @@
 #  INSTALL
 # ------------------------------------------------------------------------------
 # -- install binary file
-install( TARGETS yarock DESTINATION /usr/local/bin/ )
+install( TARGETS yarock DESTINATION /usr/bin/ )
 
 # -- install translations files
-install(FILES ${YAROCK_QM_FILES} DESTINATION /usr/local/share/locale/yarock/)
+install(FILES ${YAROCK_QM_FILES} DESTINATION /usr/share/locale/yarock/)
   
\ Dosya sonunda yenisatır yok.
diff -ur a/src/main.cpp b/src/main.cpp
--- a/src/main.cpp	2013-05-06 16:33:20.000000000 +0300
+++ b/src/main.cpp	2013-07-30 03:55:57.528855676 +0300
@@ -162,7 +162,7 @@
 
     //! setup translations
     LoadTranslation("qt", QLibraryInfo::location(QLibraryInfo::TranslationsPath), language);
-    LoadTranslation("yarock", QLatin1String("/usr/local/share/locale/yarock"), language); // normal instal path = /usr/share/locale/YaRock/
+    LoadTranslation("yarock", QLatin1String("/usr/share/locale/yarock"), language); // normal instal path = /usr/share/locale/YaRock/
     LoadTranslation("yarock", application.applicationDirPath() + "../translation", language); // in case of local running
 
     QTextCodec::setCodecForTr(QTextCodec::codecForName("utf8"));
