http://bugs.gentoo.org/118600
http://bugs.gentoo.org/237600

--- libaudiofile/audiofile.h
+++ libaudiofile/audiofile.h
@@ -29,6 +29,7 @@
 
 #include <sys/types.h>
 #include <aupvlist.h>
+#include <stdint.h>
 
 #define LIBAUDIOFILE_MAJOR_VERSION 0
 #define LIBAUDIOFILE_MINOR_VERSION 2
@@ -45,8 +45,8 @@ typedef struct _AFfilesetup *AFfilesetup
 typedef struct _AFfilehandle *AFfilehandle;
 typedef void (*AFerrfunc)(long, const char *);
 
-typedef off_t AFframecount;
-typedef off_t AFfileoffset;
+typedef int64_t AFframecount;
+typedef int64_t AFfileoffset;
 
 #define AF_NULL_FILESETUP	((struct _AFfilesetup *) 0)
 #define AF_NULL_FILEHANDLE	((struct _AFfilehandle *) 0)
