Index: libvisual-plugins-0.4.0/configure.ac
===================================================================
--- libvisual-plugins-0.4.0.orig/configure.ac
+++ libvisual-plugins-0.4.0/configure.ac
@@ -156,7 +156,7 @@ AC_ARG_ENABLE([gstreamer-plugin],
   [ENABLE_GSTREAMER_PLUGIN=yes])
 
 if test "$ENABLE_GSTREAMER_PLUGIN" = "yes"; then
-  PKG_CHECK_MODULES([GSTREAMER], [gstreamer-0.8 >= gst_required_version],
+  PKG_CHECK_MODULES([GSTREAMER], [gstreamer-0.10 >= gst_required_version],
     [HAVE_GSTREAMER="yes"], [HAVE_GSTREAMER="no"])
 
   if test "$HAVE_GSTREAMER" = "yes"; then
