diff -Nuar e-module-mem-0.0.1_55225.orig//module.desktop.in Masaüstü/e-module-mem-0.0.1_55225//module.desktop.in
--- e-module-mem-0.0.1_55225.orig//module.desktop.in	2011-02-17 10:55:29.000000000 +0200
+++ Masaüstü/e-module-mem-0.0.1_55225//module.desktop.in	2011-02-23 15:03:32.736844269 +0200
@@ -18,3 +18,4 @@
 Comment[fr]=Affiche l'utilisation de la mémoire.
 Comment[el]=Χρησιμοποιείται για την παρατήρηση της μνήμης.
 Comment[pt]=Utilizado para monitorizar a memória utilizada
+Comment[tr]=Bellek kullanımını gözlemlemek için kullanılır.
