--- funguloids/src/openalsoundsystem.cpp.orig	2007-05-22 18:11:46.000000000 +0300
+++ funguloids/src/openalsoundsystem.cpp	2008-05-11 09:33:08.000000000 +0300
@@ -266,14 +266,14 @@
 
 	int attributes[] = { 0 };
 	mContext = alcCreateContext(mDevice, attributes);
-	if ( (err = alGetError()) != AL_NO_ERROR) {
+	if ( (err = alcGetError(mDevice)) != ALC_NO_ERROR) {
 		LogManager::getSingleton().logMessage("OpanAL: error creating context: " + lastALError(err) );
 		return 1;
 	}
 //	check_alc_error("Couldn't create audio context: ");
 	alcMakeContextCurrent(mContext);
 //	check_alc_error("Couldn't select audio context: ");
-	if ( (err = alGetError()) != AL_NO_ERROR) {
+	if ( (err = alcGetError(mDevice)) != ALC_NO_ERROR) {
 		LogManager::getSingleton().logMessage("OpanAL: error making context current: " + lastALError(err) );
 		return 2;
 	}
 
 
