--- funguloids/src/input.cpp.orig	2007-04-21 20:54:18.000000000 +0300
+++ funguloids/src/input.cpp	2008-05-11 09:36:48.000000000 +0300
@@ -37,6 +37,22 @@
 	OIS::ParamList pl;
 	pl.insert(OIS::ParamList::value_type("WINDOW", Ogre::StringConverter::toString(hWnd)));
 
+	// Disable mouse capture for debugging.
+	#ifdef DISABLE_MOUSE_CAPTURE
+		#if defined OIS_WIN32_PLATFORM
+			pl.insert(std::make_pair(std::string("w32_mouse"), std::string("DISCL_FOREGROUND" )));
+			pl.insert(std::make_pair(std::string("w32_mouse"), std::string("DISCL_NONEXCLUSIVE")));
+			pl.insert(std::make_pair(std::string("w32_keyboard"), std::string("DISCL_FOREGROUND")));
+			pl.insert(std::make_pair(std::string("w32_keyboard"), std::string("DISCL_NONEXCLUSIVE")));
+		#elif defined OIS_LINUX_PLATFORM
+			pl.insert(std::make_pair(std::string("x11_mouse_grab"), std::string("false")));
+			pl.insert(std::make_pair(std::string("x11_mouse_hide"), std::string("false")));
+			pl.insert(std::make_pair(std::string("x11_keyboard_grab"), std::string("false")));
+			pl.insert(std::make_pair(std::string("XAutoRepeatOn"), std::string("true")));
+		#endif
+	#endif
+
+
 	mHWnd = hWnd;
 	mInputManager = OIS::InputManager::createInputSystem(pl);
 	mMouse = static_cast<OIS::Mouse*>(mInputManager->createInputObject(OIS::OISMouse, true));
