--- funguloids/configure.ac.orig	2008-04-18 11:27:44.000000000 +0300
+++ funguloids/configure.ac	2008-04-18 11:33:37.000000000 +0300
@@ -250,12 +250,12 @@
 	prefix="/usr/local"
 fi
 
-bininstalldir="${prefix}/games"
-gameinstalldir="${prefix}/share/games/funguloids"
-iconinstalldir="${prefix}/share/games/funguloids/icon"
-musicinstalldir="${prefix}/share/games/funguloids/music"
-docinstalldir="${prefix}/share/docs/funguloids"
-docimageinstalldir="${prefix}/share/docs/funguloids/docs"
+bininstalldir="${prefix}/bin"
+gameinstalldir="${prefix}/share/funguloids"
+iconinstalldir="${prefix}/share/funguloids/icon"
+musicinstalldir="${prefix}/share/funguloids/music"
+docinstalldir="${prefix}/share/docs/funguloids/html"
+docimageinstalldir="${prefix}/share/docs/funguloids/html/docs"
 pixmapinstalldir="${prefix}/share/pixmaps"
 
 AC_SUBST(prefix)
--- funguloids/bin/Makefile.am.orig	2008-04-18 11:42:22.000000000 +0300
+++ funguloids/bin/Makefile.am	2008-04-18 11:43:18.000000000 +0300
@@ -17,14 +17,8 @@
 	cp -a ../src/funguloids .
 
 install-data-hook:
-	if test "x@highscoredir@" != "x"; then \
-		mkdir -p "@highscoredir@"; \
-		if test ! -e "@highscoredir@/hiscore.lst"; then cp hiscore.lst.in "@highscoredir@/hiscore.lst"; fi; \
-		chgrp -R @highscoregroup@ "@highscoredir@"; \
-		chmod -R g+w "@highscoredir@"; \
-	fi; \
-	if test -r bootstrap.mpk; then cp bootstrap.mpk "@gameinstalldir@"; fi; \
-	if test -r funguloids.mpk; then cp funguloids.mpk "@gameinstalldir@"; fi
+	if test -r bootstrap.mpk; then cp bootstrap.mpk "$(DESTDIR)@gameinstalldir@"; fi; \
+	if test -r funguloids.mpk; then cp funguloids.mpk "$(DESTDIR)@gameinstalldir@"; fi
 
 uninstall-hook:
 	rm -rf "@docinstalldir@"
--- funguloids/bin/music/Makefile.am.orig	2008-04-18 11:19:11.000000000 +0300
+++ funguloids/bin/music/Makefile.am	2008-04-18 11:47:57.000000000 +0300
@@ -6,7 +6,7 @@
 	cat playlist.lua.in | sed 's,%%musicinstalldir%%,@musicinstalldir@,' > playlist.lua
 
 install-data-hook:
-	if test -r MarylandInMay.ogg; then cp MarylandInMay.ogg "@musicinstalldir@"; fi;
+	if test -r MarylandInMay.ogg; then cp MarylandInMay.ogg "$(DESTDIR)@musicinstalldir@"; fi;
 
 
 distclean: clean
