Index: funguloids/src/mpakogre.cpp
===================================================================
--- funguloids/src/mpakogre.cpp
+++ funguloids/src/mpakogre.cpp	2009-03-29 21:00:13.000000000 +0200
@@ -198,6 +198,10 @@
 	return (i != -1);
 }
 
+time_t MPakArchive::getModifiedTime(const String &filename) {
+    return 0;
+}
+
 const String &MPakArchiveFactory::getType(void) const {
 	static String name = "MPK";
 	return name;
Index: funguloids/src/scriptsystem.cpp
===================================================================
--- funguloids/src/scriptsystem.cpp
+++ funguloids/src/scriptsystem.cpp	2009-03-29 20:52:12.000000000 +0200
@@ -137,7 +137,6 @@
 	Asteroid *roid = new Asteroid(String(name), ScriptSystem::getSingleton().getSceneMgr(), String(mesh), String(mat), pos);
 	roid->setScale(scale, scale, scale);
 	roid->setMass(scale * 10.0f);
-	roid->getEntity()->setNormaliseNormals(true);
 	return 0;
 }
 
Index: funguloids/include/mpakogre.h
===================================================================
--- funguloids/include/mpakogre.h
+++ funguloids/include/mpakogre.h	2009-03-29 20:50:11.000000000 +0200
@@ -56,6 +56,7 @@
 	FileInfoListPtr findFileInfo(const String &pattern, bool recursive, bool dirs = false);
 
 	bool exists(const String &filename);
+    time_t getModifiedTime(const String &filename);
 };
 
 
