Index: include/SimpleIni/SimpleIni.h
===================================================================
--- include/SimpleIni/SimpleIni.h.orig
+++ include/SimpleIni/SimpleIni.h
@@ -407,7 +407,7 @@ public:
             return *this;
         }
         bool ConvertToStore(const SI_CHAR * a_pszString) {
-            size_t uLen = SizeToStore(a_pszString);
+            size_t uLen = this->SizeToStore(a_pszString);
             if (uLen == (size_t)(-1)) {
                 return false;
             }
