Index: bin/plugins.cfg.in
===================================================================
--- bin/plugins.cfg.in.orig
+++ bin/plugins.cfg.in
@@ -7,6 +7,6 @@ PluginFolder=%%ogreplugindir%%
 Plugin=RenderSystem_GL
 Plugin=Plugin_ParticleFX
 Plugin=Plugin_OctreeSceneManager
-Plugin=Plugin_CgProgramManager
+#Plugin=Plugin_CgProgramManager
 
 
Index: src/ogreapp.cpp
===================================================================
--- src/ogreapp.cpp.orig
+++ src/ogreapp.cpp
@@ -23,6 +23,7 @@
 //
 //***************************************************************************/
 
+#include <sys/stat.h>
 #include "ogreapp.h"
 #include "objectsystem.h"
 #include "effects.h"
@@ -71,6 +72,13 @@ bool OgreApplication::setup() {
 			OGRE_CONFIG_AND_LOG_PATH + "Ogre.log"
 			);
 
+	ConfigFile cfg;
+	cfg.load(String(OGRE_PLUGINS_AND_RESOURCES_PATH) + "plugins.cfg");
+	String pluginDir = cfg.getSetting("PluginFolder");
+	struct stat sb;
+	if (stat((pluginDir + "Plugin_CgProgramManager.so").c_str(), &sb) == 0)
+		mRoot->loadPlugin(pluginDir + "Plugin_CgProgramManager");
+
 	// Random seed
 	srand(time(NULL));
 
