--- funguloids/src/mp3stream.cpp.orig	2010-12-27 12:13:56.515000099 +0200
+++ funguloids/src/mp3stream.cpp	2010-12-27 12:15:06.596000102 +0200
@@ -234,7 +234,7 @@
 //----------------------------------------------------------------------------
 void MP3Stream::logInfo( struct mad_header *Header )
 {
-	stringstream ss;
+        std::stringstream ss;
 
 	/* Convert the layer number to it's printed representation. */
 	ss << "mpeg 1 layer ";
