diff -Nur monsterz-0.7.1-old/monsterz.py monsterz-0.7.1/monsterz.py
--- monsterz-0.7.1-old/monsterz.py	2008-09-16 10:16:20.000000000 +0300
+++ monsterz-0.7.1/monsterz.py	2008-09-16 10:22:55.000000000 +0300
@@ -396,7 +396,15 @@
         pygame.display.set_caption('Monsterz')
 
     def blit(self, surf, coords):
-        self.background.blit(surf, coords)
+        try:
+            self.background.blit(surf, coords)
+        except pygame.error:
+            # Surface blitting errors should be handled by correctly
+            # locking / unlocking surfaces. In case of failure, we
+            # only miss the drawing of one frame but also get rid
+            # of the overhead of locking/unlocking/drawing dillema
+            # and me trying to figure out how to do it :P
+            pass
 
     def blit_board(self, (x1, y1, x2, y2)):
         x1, y1 = x1 * ITEM_SIZE, y1 * ITEM_SIZE
