--- main.c.orig	2009-06-11 10:42:36.000000000 +0300
+++ main.c	2009-06-11 10:42:54.000000000 +0300
@@ -273,7 +273,7 @@
     {"browser", "Browser", XtRString, sizeof(char *),
       XtOffset(appresPtr, browser), XtRString, (caddr_t) "firefox"},
     {"pdfviewer", "Viewer", XtRString, sizeof(char *),
-      XtOffset(appresPtr, pdf_viewer), XtRString, (caddr_t) "acroread"},
+      XtOffset(appresPtr, pdf_viewer), XtRString, (caddr_t) "xdg-open"},
     {"spinner_delay", "spinnerDelay",   XtRInt, sizeof(int),
       XtOffset(appresPtr, spinner_delay), XtRImmediate, (caddr_t) 500},
     {"spinner_rate", "spinnerRate",   XtRInt, sizeof(int),
