--- Fig.ad.orig	2009-06-10 16:50:27.000000000 +0300
+++ Fig.ad	2009-06-10 16:51:13.000000000 +0300
@@ -25,16 +25,16 @@
 ! 		This is for viewing the xfig html reference.
 ! For firefox, this command will open the help pages in a running firefox,
 !     or start a new netscape if one isn't already running
-Fig.browser:	firefox -remote 'openFile(%f)' || firefox %f
+Fig.browser:	xdg-open %f
 
 ! pdfviewer - put your favorite pdf viewer here.  
 !		This is for viewing the xfig how-to guide and man pages
-Fig.pdfviewer:			xpdf %f
+Fig.pdfviewer:			xdg-open %f
 
 ! Spell check program - put your favorite spelling check program here.
 !		It must write the misspelled words to standard output.
 
-Fig.spellcheckcommand:	spell %f
+Fig.spellcheckcommand:	aspell %f
 
 ! delay before spinner automatically counts when pressed (milliseconds)
 Fig.spinner_delay:		1000
