diff -ur libsignon-glib-1.9-orig/libsignon-glib/Makefile.am libsignon-glib-1.9/libsignon-glib/Makefile.am
--- libsignon-glib-1.9-orig/libsignon-glib/Makefile.am	2012-12-10 11:55:50.000000000 +0200
+++ libsignon-glib-1.9/libsignon-glib/Makefile.am	2013-03-22 10:30:19.999087930 +0200
@@ -16,16 +16,13 @@
 nodist_libsignon_glib_la_SOURCES = \
 	signon-marshal.c \
 	signon-marshal.h \
-	signon-enum-types.h \
-	signon-enum-types.c \
 	signon-errors-enum.c \
 	sso-auth-service-gen.c \
 	sso-auth-service-gen.h \
 	sso-auth-session-gen.c \
 	sso-auth-session-gen.h \
 	sso-identity-gen.c \
-	sso-identity-gen.h \
-	stamp-signon-enum-types.h
+	sso-identity-gen.h
 
 BUILT_SOURCES = \
 	$(nodist_libsignon_glib_la_SOURCES) \
@@ -64,7 +61,6 @@
 	signon-identity-info.h \
 	signon-identity.h \
 	signon-errors.h \
-	signon-enum-types.h \
 	signon-glib.h \
 	signon-types.h \
 	$(signon_headers)
@@ -105,39 +101,9 @@
 signon-marshal.c: signon-marshal.list Makefile
 	$(AM_V_GEN)glib-genmarshal --body --prefix=_signon_marshal $< > $@
 
-signon-enum-types.h: stamp-signon-enum-types.h
-	$(AM_V_at)true
-
-stamp-signon-enum-types.h: Makefile \
-	$(libsignon_glib_headers_with_enums) signon-errors.h
-	$(AM_V_GEN)( cd $(srcdir) &&  glib-mkenums \
-	    --fhead "#ifndef __SIGNON_ENUM_TYPES_H__\n#define __SIGNON_ENUM_TYPES_H__\n#include <glib-object.h>\n\nG_BEGIN_DECLS\n" \
-	    --fprod "/* enumerations from \"@filename@\" */\n" \
-	    --vhead "GType @enum_name@_get_type (void) G_GNUC_CONST;\n#define SIGNON_TYPE_@ENUMSHORT@ (@enum_name@_get_type())\n" \
-	    --ftail "G_END_DECLS\n\n#endif /* __SIGNON_ENUM_TYPES_H__ */" \
-	    $(libsignon_glib_headers_with_enums) signon-errors.h ) >> xgen-geth \
-	    && (cmp -s xgen-geth signon-enum-types.h || cp xgen-geth signon-enum-types.h ) \
-	    && rm -f xgen-geth \
-	    && echo timestamp > $(@F)
-
-signon-enum-types.c: Makefile $(libsignon_glib_headers_with_enums)
-	$(AM_V_GEN)( cd $(srcdir) && glib-mkenums \
-	    --fhead "#include \"signon-enum-types.h\"\n" \
-	    --fhead "#include \"signon-identity-info.h\"\n" \
-			--fhead "#include \"signon-auth-session.h\"\n" \
-			--fhead "#define g_intern_static_string(s) (s)\n" \
-	    --fprod "\n/* enumerations from \"@filename@\" */" \
-	    --ftail "\n#define __SIGNON_ENUM_TYPES_C__\n" \
-	    --vhead "GType\n@enum_name@_get_type (void)\n{\n  static GType etype = 0;\n  if (etype == 0) {\n    static const G@Type@Value values[] = {" \
-	    --vprod "      { @VALUENAME@, \"@VALUENAME@\", \"@valuenick@\" }," \
-	    --vtail "      { 0, NULL, NULL }\n    };\n    etype = g_@type@_register_static (g_intern_static_string (\"@EnumName@\"), values);\n  }\n  return etype;\n}\n" \
-	    $(libsignon_glib_headers_with_enums) ) > xgen-getc \
-	    && cp xgen-getc signon-enum-types.c  \
-	    && rm -f xgen-getc
-
 signon-errors-enum.c: Makefile signon-errors.h
 	$(AM_V_GEN)( cd $(srcdir) && glib-mkenums \
-	    --fhead "#include \"signon-enum-types.h\"\n#include \"signon-errors.h\"\n#include \"signoncommon.h\"\n#define g_intern_static_string(s) (s)\n" \
+	    --fhead "#include \"signon-errors.h\"\n#include \"signoncommon.h\"\n#define g_intern_static_string(s) (s)\n" \
 	    --fprod "\n/* enumerations from \"@filename@\" */" \
 	    --ftail "\n#define __SIGNON_ENUM_TYPES_C__\n" \
 	    --vhead "GType\n@enum_name@_get_type (void)\n{\n  static GType etype = 0;\n  if (etype == 0) {\n    static const G@Type@Value values[] = {" \
@@ -157,8 +123,7 @@
 	  echo -e "};\n" ) > signon-errors-map.c
 
 dist_noinst_DATA = \
-	signon-marshal.list \
-	stamp-signon-enum-types.h
+	signon-marshal.list
 
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS =
@@ -171,8 +136,6 @@
 	signon-auth-service.h \
 	signon-auth-session.c \
 	signon-auth-session.h \
-	signon-enum-types.h \
-	signon-enum-types.c \
 	signon-errors.c \
 	signon-errors.h \
 	signon-identity-info.c \
diff -ur libsignon-glib-1.9-orig/libsignon-glib/Makefile.in libsignon-glib-1.9/libsignon-glib/Makefile.in
--- libsignon-glib-1.9-orig/libsignon-glib/Makefile.in	2013-02-05 16:46:23.000000000 +0200
+++ libsignon-glib-1.9/libsignon-glib/Makefile.in	2013-03-22 10:33:02.948082288 +0200
@@ -115,7 +115,6 @@
 	libsignon_glib_la-sso-auth-service.lo
 nodist_libsignon_glib_la_OBJECTS =  \
 	libsignon_glib_la-signon-marshal.lo \
-	libsignon_glib_la-signon-enum-types.lo \
 	libsignon_glib_la-signon-errors-enum.lo \
 	libsignon_glib_la-sso-auth-service-gen.lo \
 	libsignon_glib_la-sso-auth-session-gen.lo \
@@ -333,16 +332,13 @@
 nodist_libsignon_glib_la_SOURCES = \
 	signon-marshal.c \
 	signon-marshal.h \
-	signon-enum-types.h \
-	signon-enum-types.c \
 	signon-errors-enum.c \
 	sso-auth-service-gen.c \
 	sso-auth-service-gen.h \
 	sso-auth-session-gen.c \
 	sso-auth-session-gen.h \
 	sso-identity-gen.c \
-	sso-identity-gen.h \
-	stamp-signon-enum-types.h
+	sso-identity-gen.h
 
 BUILT_SOURCES = \
 	$(nodist_libsignon_glib_la_SOURCES) \
@@ -380,7 +376,6 @@
 	signon-identity-info.h \
 	signon-identity.h \
 	signon-errors.h \
-	signon-enum-types.h \
 	signon-glib.h \
 	signon-types.h \
 	$(signon_headers)
@@ -396,8 +391,7 @@
 # This should come from `pkg-config --variable=interfaces_dir dbus-1`
 DBUS_INTERFACES_DIR = /usr/share/dbus-1/interfaces
 dist_noinst_DATA = \
-	signon-marshal.list \
-	stamp-signon-enum-types.h
+	signon-marshal.list
 
 INTROSPECTION_GIRS = $(am__append_1)
 INTROSPECTION_SCANNER_ARGS = --add-include-path=$(srcdir)
@@ -407,8 +401,6 @@
 @HAVE_INTROSPECTION_TRUE@	signon-auth-service.h \
 @HAVE_INTROSPECTION_TRUE@	signon-auth-session.c \
 @HAVE_INTROSPECTION_TRUE@	signon-auth-session.h \
-@HAVE_INTROSPECTION_TRUE@	signon-enum-types.h \
-@HAVE_INTROSPECTION_TRUE@	signon-enum-types.c \
 @HAVE_INTROSPECTION_TRUE@	signon-errors.c \
 @HAVE_INTROSPECTION_TRUE@	signon-errors.h \
 @HAVE_INTROSPECTION_TRUE@	signon-identity-info.c \
@@ -518,7 +510,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libsignon_glib_la-signon-auth-service.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libsignon_glib_la-signon-auth-session.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libsignon_glib_la-signon-dbus-queue.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libsignon_glib_la-signon-enum-types.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libsignon_glib_la-signon-errors-enum.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libsignon_glib_la-signon-errors.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libsignon_glib_la-signon-identity-info.Plo@am__quote@
@@ -617,13 +608,6 @@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libsignon_glib_la_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o libsignon_glib_la-signon-marshal.lo `test -f 'signon-marshal.c' || echo '$(srcdir)/'`signon-marshal.c
 
-libsignon_glib_la-signon-enum-types.lo: signon-enum-types.c
-@am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libsignon_glib_la_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -MT libsignon_glib_la-signon-enum-types.lo -MD -MP -MF $(DEPDIR)/libsignon_glib_la-signon-enum-types.Tpo -c -o libsignon_glib_la-signon-enum-types.lo `test -f 'signon-enum-types.c' || echo '$(srcdir)/'`signon-enum-types.c
-@am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) $(DEPDIR)/libsignon_glib_la-signon-enum-types.Tpo $(DEPDIR)/libsignon_glib_la-signon-enum-types.Plo
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='signon-enum-types.c' object='libsignon_glib_la-signon-enum-types.lo' libtool=yes @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libsignon_glib_la_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o libsignon_glib_la-signon-enum-types.lo `test -f 'signon-enum-types.c' || echo '$(srcdir)/'`signon-enum-types.c
-
 libsignon_glib_la-signon-errors-enum.lo: signon-errors-enum.c
 @am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libsignon_glib_la_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -MT libsignon_glib_la-signon-errors-enum.lo -MD -MP -MF $(DEPDIR)/libsignon_glib_la-signon-errors-enum.Tpo -c -o libsignon_glib_la-signon-errors-enum.lo `test -f 'signon-errors-enum.c' || echo '$(srcdir)/'`signon-errors-enum.c
 @am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) $(DEPDIR)/libsignon_glib_la-signon-errors-enum.Tpo $(DEPDIR)/libsignon_glib_la-signon-errors-enum.Plo
@@ -987,39 +971,9 @@
 signon-marshal.c: signon-marshal.list Makefile
 	$(AM_V_GEN)glib-genmarshal --body --prefix=_signon_marshal $< > $@
 
-signon-enum-types.h: stamp-signon-enum-types.h
-	$(AM_V_at)true
-
-stamp-signon-enum-types.h: Makefile \
-	$(libsignon_glib_headers_with_enums) signon-errors.h
-	$(AM_V_GEN)( cd $(srcdir) &&  glib-mkenums \
-	    --fhead "#ifndef __SIGNON_ENUM_TYPES_H__\n#define __SIGNON_ENUM_TYPES_H__\n#include <glib-object.h>\n\nG_BEGIN_DECLS\n" \
-	    --fprod "/* enumerations from \"@filename@\" */\n" \
-	    --vhead "GType @enum_name@_get_type (void) G_GNUC_CONST;\n#define SIGNON_TYPE_@ENUMSHORT@ (@enum_name@_get_type())\n" \
-	    --ftail "G_END_DECLS\n\n#endif /* __SIGNON_ENUM_TYPES_H__ */" \
-	    $(libsignon_glib_headers_with_enums) signon-errors.h ) >> xgen-geth \
-	    && (cmp -s xgen-geth signon-enum-types.h || cp xgen-geth signon-enum-types.h ) \
-	    && rm -f xgen-geth \
-	    && echo timestamp > $(@F)
-
-signon-enum-types.c: Makefile $(libsignon_glib_headers_with_enums)
-	$(AM_V_GEN)( cd $(srcdir) && glib-mkenums \
-	    --fhead "#include \"signon-enum-types.h\"\n" \
-	    --fhead "#include \"signon-identity-info.h\"\n" \
-			--fhead "#include \"signon-auth-session.h\"\n" \
-			--fhead "#define g_intern_static_string(s) (s)\n" \
-	    --fprod "\n/* enumerations from \"@filename@\" */" \
-	    --ftail "\n#define __SIGNON_ENUM_TYPES_C__\n" \
-	    --vhead "GType\n@enum_name@_get_type (void)\n{\n  static GType etype = 0;\n  if (etype == 0) {\n    static const G@Type@Value values[] = {" \
-	    --vprod "      { @VALUENAME@, \"@VALUENAME@\", \"@valuenick@\" }," \
-	    --vtail "      { 0, NULL, NULL }\n    };\n    etype = g_@type@_register_static (g_intern_static_string (\"@EnumName@\"), values);\n  }\n  return etype;\n}\n" \
-	    $(libsignon_glib_headers_with_enums) ) > xgen-getc \
-	    && cp xgen-getc signon-enum-types.c  \
-	    && rm -f xgen-getc
-
 signon-errors-enum.c: Makefile signon-errors.h
 	$(AM_V_GEN)( cd $(srcdir) && glib-mkenums \
-	    --fhead "#include \"signon-enum-types.h\"\n#include \"signon-errors.h\"\n#include \"signoncommon.h\"\n#define g_intern_static_string(s) (s)\n" \
+	    --fhead "#include \"signon-errors.h\"\n#include \"signoncommon.h\"\n#define g_intern_static_string(s) (s)\n" \
 	    --fprod "\n/* enumerations from \"@filename@\" */" \
 	    --ftail "\n#define __SIGNON_ENUM_TYPES_C__\n" \
 	    --vhead "GType\n@enum_name@_get_type (void)\n{\n  static GType etype = 0;\n  if (etype == 0) {\n    static const G@Type@Value values[] = {" \
diff -ur libsignon-glib-1.9-orig/libsignon-glib/signon-errors.c libsignon-glib-1.9/libsignon-glib/signon-errors.c
--- libsignon-glib-1.9-orig/libsignon-glib/signon-errors.c	2012-08-13 11:09:44.000000000 +0300
+++ libsignon-glib-1.9/libsignon-glib/signon-errors.c	2013-03-22 10:34:03.283080199 +0200
@@ -24,7 +24,6 @@
  */
 
 #include "signon-errors.h"
-#include "signon-enum-types.h"
 #include "signon-internals.h"
 #include "signoncommon.h"
 #include <gio/gio.h>
diff -ur libsignon-glib-1.9-orig/libsignon-glib/signon-errors.h libsignon-glib-1.9/libsignon-glib/signon-errors.h
--- libsignon-glib-1.9-orig/libsignon-glib/signon-errors.h	2012-08-13 10:25:41.000000000 +0300
+++ libsignon-glib-1.9/libsignon-glib/signon-errors.h	2013-03-22 10:34:19.148079650 +0200
@@ -27,7 +27,6 @@
 
 #include <glib.h>
 #include <glib-object.h>
-#include "signon-enum-types.h"
 
 #define SIGNON_ERROR (signon_error_quark())
 
diff -ur libsignon-glib-1.9-orig/libsignon-glib/signon-glib.h libsignon-glib-1.9/libsignon-glib/signon-glib.h
--- libsignon-glib-1.9-orig/libsignon-glib/signon-glib.h	2011-12-09 16:52:51.000000000 +0200
+++ libsignon-glib-1.9/libsignon-glib/signon-glib.h	2013-03-22 10:34:36.584079046 +0200
@@ -27,7 +27,6 @@
 
 #include <libsignon-glib/signon-auth-service.h>
 #include <libsignon-glib/signon-auth-session.h>
-#include <libsignon-glib/signon-enum-types.h>
 #include <libsignon-glib/signon-errors.h>
 #include <libsignon-glib/signon-identity-info.h>
 #include <libsignon-glib/signon-identity.h>
