diff -Nuar lxdm-0.5.0.orig/data/lxdm.conf.in lxdm-0.5.0/data/lxdm.conf.in
--- lxdm-0.5.0.orig/data/lxdm.conf.in	2012-02-27 13:28:25.000000000 +0200
+++ lxdm-0.5.0/data/lxdm.conf.in	2014-08-18 14:01:59.955134485 +0300
@@ -34,7 +34,7 @@
 gtk_theme=Clearlooks
 
 ## background of the greeter
-bg=/usr/share/backgrounds/default.png
+#bg=/usr/share/backgrounds/default.png
 
 ## if show bottom pane
 bottom_pane=1
@@ -46,7 +46,7 @@
 keyboard=0
 
 ## the theme of greeter
-theme=Industrial
+theme=lxdm-pisilinux-theme
 
 [input]
 
@@ -58,5 +58,5 @@
 white=
 
 ## blacklist user
-black=
+black=ftp
 
