--- scripts/CMakeLists.txt.orig	2009-07-17 00:27:01.000000000 +0300
+++ scripts/CMakeLists.txt	2009-07-17 00:27:27.000000000 +0300
@@ -1,7 +1,7 @@
 if (WIN32)
 	SET(scriptInstallDir ${CMAKE_INSTALL_PREFIX}/scripts)
 else (WIN32)
-	SET(scriptInstallDir ${CMAKE_INSTALL_PREFIX}/share/ADM_scripts)
+	SET(scriptInstallDir ${CMAKE_INSTALL_PREFIX}/share/avidemux/ADM_scripts)
 endif (WIN32)
 
-install(DIRECTORY . DESTINATION "${scriptInstallDir}" FILES_MATCHING PATTERN "*.js" PATTERN ".svn" EXCLUDE)
\ No newline at end of file
+install(DIRECTORY . DESTINATION "${scriptInstallDir}" FILES_MATCHING PATTERN "*.js" PATTERN ".svn" EXCLUDE)
