diff -ru avidemux_2.5.5.orig/cmake/admFFmpegPrepareTar.cmake avidemux_2.5.5/cmake/admFFmpegPrepareTar.cmake
--- avidemux_2.5.5.orig/cmake/admFFmpegPrepareTar.cmake	2011-05-11 19:57:56.000000000 +0200
+++ avidemux_2.5.5/cmake/admFFmpegPrepareTar.cmake	2011-12-21 19:34:19.000000000 +0100
@@ -16,7 +16,7 @@
 					WORKING_DIRECTORY "${LIBRARY_SOURCE_DIR}"
 					${ffmpegExtractOutput})
 
-		execute_process(COMMAND ${CMAKE_COMMAND} -E rename "ffmpeg-0.7-rc1" "ffmpeg"
+		execute_process(COMMAND ${CMAKE_COMMAND} -E copy_directory "ffmpeg-0.7-rc1" "ffmpeg"
 					WORKING_DIRECTORY "${LIBRARY_SOURCE_DIR}")
 
 		set(FFMPEG_PERFORM_PATCH 1)
@@ -37,4 +37,4 @@
 	set(FFMPEG_PREPARED 1)
 	set(LAST_FFMPEG_VERSION "${FFMPEG_VERSION}" CACHE STRING "" FORCE)
 	#set(LAST_SWSCALE_VERSION "${SWSCALE_VERSION}" CACHE STRING "" FORCE)
-endif (EXISTS "${LIBRARY_SOURCE_DIR}/${FFMPEG_SOURCE_ARCHIVE}") # AND EXISTS "${LIBRARY_SOURCE_DIR}/${SWSCALE_SOURCE_ARCHIVE}")
\ Kein Zeilenumbruch am Dateiende.
+endif (EXISTS "${LIBRARY_SOURCE_DIR}/${FFMPEG_SOURCE_ARCHIVE}") # AND EXISTS "${LIBRARY_SOURCE_DIR}/${SWSCALE_SOURCE_ARCHIVE}")
