--- avidemux_2.5.0.orig/avidemux/prefs.in	2009-02-22 08:36:02.000000000 -0500
+++ avidemux_2.5.0/avidemux/prefs.in	2009-07-10 15:07:05.100178665 -0400
@@ -53,9 +53,9 @@
 #ifdef __WIN32
 device.audiodevice,			STRING,	WIN32
 #else
-device.audiodevice,			STRING,	ALSA
+device.audiodevice,			STRING,	PulseAudioS
 #endif
-device.audio.alsa_device,               STRING, "dmix"
+device.audio.alsa_device,               STRING, "pulse"
 device.videodevice,			UINT,	0,	0,	10
 # Posproc
 default.postproc_type,			UINT,	3,	0,	7
--- avidemux_2.5.0.orig/avidemux/ADM_libraries/ADM_utilities/prefs.cpp	2009-02-22 08:36:02.000000000 -0500
+++ avidemux_2.5.0/avidemux/ADM_libraries/ADM_utilities/prefs.cpp	2009-07-10 15:07:21.436178612 -0400
@@ -97,9 +97,9 @@
 #ifdef __WIN32
 	{"device.audiodevice",		STRING,"WIN32",NULL, NULL, NULL },
 #else
-	{"device.audiodevice",		STRING,"ALSA",	NULL, NULL, NULL },
+	{"device.audiodevice",		STRING,"PulseAudioS",	NULL, NULL, NULL },
 #endif
-	{"device.audio.alsa_device",		STRING,"dmix",	NULL, NULL, NULL },
+	{"device.audio.alsa_device",		STRING,"pulse",	NULL, NULL, NULL },
 	{"device.videodevice",		UINT,	"0",	NULL,	"0",	"10"	},
 	{"default.postproc_type",		UINT,	"3",	NULL,	"0",	"7"	},
 	{"default.postproc_value",		UINT,	"3",	NULL,	"0",	"5"	},
