Index: configure.in
===================================================================
--- configure.in.orig
+++ configure.in
@@ -194,8 +194,7 @@ if test "$with_python" != "no" ; then
         echo "PYTHON is pointing at $PYTHON"
         PYTHON_VERSION=`$PYTHON -c "import sys; print sys.version[[0:3]]"`
 	echo Found Python version $PYTHON_VERSION
-	LIBXML2_PYTHON=`$PYTHON -c "try : import libxml2 ; print 1
-except: print 0"`
+	LIBXML2_PYTHON="1"
 	if test "$LIBXML2_PYTHON" = "1"
 	then
 	    echo Found libxml2-python module
Index: python/tests/Makefile.in
===================================================================
--- python/tests/Makefile.in.orig
+++ python/tests/Makefile.in
@@ -208,8 +208,7 @@ EXAMPLE_DIR = $(datadir)/doc/libxslt-pyt
 TESTSPY = \
     basic.py	\
     exslt.py	\
-    extelem.py	\
-    extfunc.py
+    extelem.py
 
 XMLS = \
     test.xml	\
