From 81ec65daf605e44d6b2926adf2ca97f3ca731ca7 Mon Sep 17 00:00:00 2001
From: unknown author <cooker@mandrivalinux.org>
Date: Thu, 17 Sep 2009 16:29:12 +0000
Subject: [PATCH] Fix python linking

---
 configure.in       |    2 ++
 python/Makefile.am |    2 +-
 2 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/configure.in b/configure.in
index da1136d..8cc0622 100644
--- a/configure.in
+++ b/configure.in
@@ -341,6 +341,7 @@ except: print 0"`
 		    PYTHON_SITE_PACKAGES=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib()"`
 	    fi
 	fi
+        PYTHON_LIBS=`python$PYTHON_VERSION-config --libs`
     fi
     if test "$with_python" != ""
     then
@@ -358,6 +359,7 @@ else
 fi
 AC_SUBST(pythondir)
 AC_SUBST(PYTHON_SUBDIR)
+AC_SUBST(PYTHON_LIBS)
 
 AC_ARG_WITH(crypto, [  --with-crypto           Add crypto support to exslt (on)])
 WITH_CRYPTO=0
diff --git a/python/Makefile.am b/python/Makefile.am
index 9591c11..ebe276a 100644
--- a/python/Makefile.am
+++ b/python/Makefile.am
@@ -36,7 +36,7 @@ all-local: libxslt.py
 python_LTLIBRARIES = libxsltmod.la
 
 libxsltmod_la_SOURCES = libxslt.c types.c libxslt-py.c
-libxsltmod_la_LIBADD = $(mylibs) 
+libxsltmod_la_LIBADD = $(mylibs) @PYTHON_LIBS@
 
 libxslt.py: $(srcdir)/libxsl.py $(srcdir)/libxsltclass.py
 	cat $(srcdir)/libxsl.py $(srcdir)/libxsltclass.py > libxslt.py
-- 
1.6.4.3

