diff -Nuar alibxslt-1.1.27/python/tests/basic.py libxslt-1.1.27/python/tests/basic.py
--- alibxslt-1.1.27/python/tests/basic.py	2009-05-12 09:29:34.000000000 +0300
+++ libxslt-1.1.27/python/tests/basic.py	2012-09-16 07:50:04.859196196 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/python -u
+#!/usr/bin/python2.7 -u
 import sys
 import libxml2
 # Memory debug specific
diff -Nuar alibxslt-1.1.27/python/tests/exslt.py libxslt-1.1.27/python/tests/exslt.py
--- alibxslt-1.1.27/python/tests/exslt.py	2009-05-12 09:29:34.000000000 +0300
+++ libxslt-1.1.27/python/tests/exslt.py	2012-09-16 07:50:17.399128261 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/python -u
+#!/usr/bin/python2.7 -u
 import sys
 import libxml2
 # Memory debug specific
diff -Nuar alibxslt-1.1.27/python/tests/extelem.py libxslt-1.1.27/python/tests/extelem.py
--- alibxslt-1.1.27/python/tests/extelem.py	2009-05-12 09:29:34.000000000 +0300
+++ libxslt-1.1.27/python/tests/extelem.py	2012-09-16 07:50:30.115726036 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/python -u
+#!/usr/bin/python2.7 -u
 import sys
 import string
 import StringIO
diff -Nuar alibxslt-1.1.27/python/tests/extfunc.py libxslt-1.1.27/python/tests/extfunc.py
--- alibxslt-1.1.27/python/tests/extfunc.py	2009-05-12 09:29:34.000000000 +0300
+++ libxslt-1.1.27/python/tests/extfunc.py	2012-09-16 07:50:40.252337788 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/python -u
+#!/usr/bin/python2.7 -u
 import sys
 import string
 import libxml2
diff -Nuar alibxslt-1.1.27/python/tests/pyxsltproc.py libxslt-1.1.27/python/tests/pyxsltproc.py
--- alibxslt-1.1.27/python/tests/pyxsltproc.py	2012-09-04 17:26:23.000000000 +0300
+++ libxslt-1.1.27/python/tests/pyxsltproc.py	2012-09-16 07:50:50.868946939 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/python -u
+#!/usr/bin/python2.7 -u
 #
 # The exercise of rewriting xsltproc on top of the python
 # bindings, not complete yet and shows up the things missing
