--- a/src/libMoose/MooseCommon.cpp
+++ b/src/libMoose/MooseCommon.cpp
@@ -47,7 +47,7 @@
 QString
 dataPath( QString file )
 {
-    return QApplication::applicationDirPath() + "/data/" + file;
+    return "/usr/share/lastfm/" + file;
 }
 
 
@@ -112,7 +112,7 @@
         // Hack to get it working with VS2005
         dirPath = qApp->applicationDirPath();
     #else
-        dirPath = qApp->applicationDirPath() + "/services";
+        dirPath = "/usr/lib/lastfm/";
     #endif
 
     #ifndef QT_NO_DEBUG