Fix compilation with gcc >= 4.1
--- a/src/updatewizard.h
+++ b/src/updatewizard.h
@@ -50,7 +50,7 @@
         CAutoUpdater& updater,
         QWidget*      parent = NULL);
 
-    UpdateWizard::UpdateWizard( CComponentInfo* );
+    UpdateWizard( CComponentInfo* );
 
     bool shouldShow() const{ return m_showWizard; }
 
--- a/src/wizardselectupdatespage.h
+++ b/src/wizardselectupdatespage.h
@@ -58,7 +58,7 @@
     GetChecked(
         std::vector<CComponentInfo*>& vecChecked);
     
-    void WizardSelectUpdatesPage::GetMajorUpdateComponent( std::vector<CComponentInfo*>& vecToUpdate );
+    void GetMajorUpdateComponent( std::vector<CComponentInfo*>& vecToUpdate );
 
 public slots:
 
