--- a/src/settingsdialog.cpp
+++ b/src/settingsdialog.cpp
@@ -92,6 +92,8 @@
     ui_mediadevices.setupUi( mediadeviceWidget );
     ui_mediadevices.deviceWidget->header()->setResizeMode( QHeaderView::ResizeToContents );
     ui.pageStack->addWidget( mediadeviceWidget );
+#else
+    ui_scrobbling.groupBox->hide();
 #endif
 
     // Add icons to user icon dropdown
--- a/src/settingsdialog_scrobbling.ui
+++ b/src/settingsdialog_scrobbling.ui
@@ -175,6 +175,19 @@
      </layout>
     </widget>
    </item>
+   <item>
+    <spacer>
+     <property name="orientation" >
+      <enum>Qt::Vertical</enum>
+     </property>
+     <property name="sizeHint" >
+      <size>
+       <width>20</width>
+       <height>0</height>
+      </size>
+     </property>
+    </spacer>
+   </item>
   </layout>
  </widget>
  <customwidgets>
