Index: strigi-0.7.3/libstreamanalyzer/CMakeLists.txt
===================================================================
--- strigi-0.7.3.orig/libstreamanalyzer/CMakeLists.txt
+++ strigi-0.7.3/libstreamanalyzer/CMakeLists.txt
@@ -25,7 +25,7 @@ set(CMAKE_INSTALL_RPATH_USE_LINK_PATH  T
 set(CMAKE_INSTALL_RPATH                ${LIB_DESTINATION} )
 
 set(LIBSTREAMANALYZER_VERSION 0.7.3)
-set(LIBSTREAMANALYZER_SOVERSION 0.7)
+set(LIBSTREAMANALYZER_SOVERSION 0)
 option(BUILD_UTILS "build luceneindexer, xmlindexer, rdfindexer, ontoprint utilities" ON)
 option(BUILD_DEEPTOOLS "build deep find and deepgrep tools" ON)
 option(ENABLE_XINE
