diff -up strigi-0.7.2/src/streamanalyzer/endanalyzers/flacendanalyzer.cpp.kde234398 strigi-0.7.2/src/streamanalyzer/endanalyzers/flacendanalyzer.cpp
--- strigi-0.7.2/src/streamanalyzer/endanalyzers/flacendanalyzer.cpp.kde234398	2010-02-03 13:03:07.000000000 -0600
+++ strigi-0.7.2/src/streamanalyzer/endanalyzers/flacendanalyzer.cpp	2010-06-30 08:39:35.124448418 -0500
@@ -284,7 +284,7 @@ FlacEndAnalyzer::analyze(Strigi::Analysi
     } while( !(blocktype & 0x80) );
 
     int64_t in_size = in->size();
-    if(in_size>=0)
+    if(in_size>=0 && duration>0)
       indexable.addValue(factory->bitRateField,(uint32_t)(8*((in_size-in->position())/duration)));
 
     return 0;
