diff -up strigi-0.7.7/strigidaemon/bin/daemon/eventlistener/eventlistenerqueue.cpp.gcc47 strigi-0.7.7/strigidaemon/bin/daemon/eventlistener/eventlistenerqueue.cpp
--- strigi-0.7.7/strigidaemon/bin/daemon/eventlistener/eventlistenerqueue.cpp.gcc47	2011-07-14 20:29:31.000000000 -0500
+++ strigi-0.7.7/strigidaemon/bin/daemon/eventlistener/eventlistenerqueue.cpp	2012-01-09 10:59:14.061837577 -0600
@@ -27,9 +27,7 @@
 #include <iostream>
 #include <stdio.h>
 
-#if defined(__SUNPRO_CC)
 #include <unistd.h>
-#endif
 
 using namespace std;
 
