Index: strigi-0.7.5/CMakeLists.txt
===================================================================
--- strigi-0.7.5.orig/CMakeLists.txt
+++ strigi-0.7.5/CMakeLists.txt
@@ -40,8 +40,6 @@ macro(check_subdir _subDirectory)
   endif()
 endmacro()
 
-check_subdir(libstreams)
-check_subdir(libstreamanalyzer)
 check_subdir(strigiutils)
 check_subdir(strigidaemon)
 check_subdir(strigiclient)
@@ -61,8 +59,6 @@ if(showError)
                       "  git submodule update --init"
          )
 else()
-  add_subdirectory(libstreams) # Must be first
-  add_subdirectory(libstreamanalyzer) # Must be second
   add_subdirectory(strigiutils)
   if(NOT WIN32)
     add_subdirectory(strigidaemon)
