diff -Nuar poedit-1.4.6.1.orig/install/poedit.desktop poedit-1.4.6.1/install/poedit.desktop
--- poedit-1.4.6.1.orig/install/poedit.desktop	2010-05-15 21:55:56.359049840 +0300
+++ poedit-1.4.6.1/install/poedit.desktop	2010-05-15 21:57:43.525778202 +0300
@@ -2,6 +2,7 @@
 Name=Poedit
 Comment=Gettext catalogs editor
 Comment[fr]=Outil de traduction
+Comment[tr]=Gettext .po düzenleyicisi
 MimeType=application/x-po;application/x-gettext;text/x-gettext-translation;text/x-po;
 Exec=poedit %F
 Icon=poedit
