diff -ur plt-4.2.1.orig/src/mzscheme/gc2/newgc.c plt-4.2.1/src/mzscheme/gc2/newgc.c
--- plt-4.2.1.orig/src/mzscheme/gc2/newgc.c	2009-07-16 14:28:05.000000000 +0200
+++ plt-4.2.1/src/mzscheme/gc2/newgc.c	2009-08-03 13:58:23.000000000 +0200
@@ -2331,7 +2331,7 @@
               sizes[tag] += info->size;
             }
             if (tag == trace_for_tag) {
-              register_traced_object(obj_staart);
+              register_traced_object(obj_start);
               if (for_each_found)
                 for_each_found(obj_start);
             }
