# http://pkgs.fedoraproject.org/gitweb/?p=ntop.git;a=tree
--- ntop-3.2/prefs.c    2005-09-29 10:39:06.000000000 +1200
+++ ntop-3.2/prefs.c.mjk    2006-07-06 17:34:34.000000000 +1200
@@ -772,8 +772,8 @@
       /* We're root */
       char *user;
 
-      pw = getpwnam(user = "nobody");
-      if(pw == NULL) pw = getpwnam(user = "anonymous");
+      pw = getpwnam(user = "ntop");
+      if(pw == NULL) pw = getpwnam(user = "nobody");
 
       if(pw != NULL) {
    myGlobals.userId  = pw->pw_uid;
