--- ntop-5.0/autogen.sh.orig	2012-08-01 23:12:36.375302258 +0200
+++ ntop-5.0/autogen.sh	2012-08-01 23:12:46.909259070 +0200
@@ -216,16 +216,6 @@
   exit 1
 fi
 
-SVN=`find_command 'svn'`
-($SVN --version) < /dev/null > /dev/null 2>&1 ||
-{
-  echo
-  echo "You must have svn/subversion installed to compile $progname."
-  echo "Download the appropriate package for your distribution, or get the"
-  echo "source from http://subversion.tigris.org"
-  GNU_OR_DIE=0
-}
-
 if test "$GNU_OR_DIE" -eq 0; then
   exit 1
 fi
