--- cmake/kvirc4.desktop.cmake.orig	2010-04-19 12:15:32.000000000 +0300
+++ cmake/kvirc4.desktop.cmake	2010-04-19 12:16:23.000000000 +0300
@@ -3,9 +3,11 @@
 GenericName=IRC Client
 GenericName[pl]=Klient IRC
 GenericName[pt]=Cliente IRC
+GenericName[tr]=IRC İstemcisi
 Comment=Connect to Internet Relay Chat
 Comment[pl]=Połącz się z Internet Relay Chat (IRC)
 Comment[pt]=Conectar ao IRC
+Comment[tr]=IRC'e bağlanın
 Exec=${KVIRC_BINARYNAME} -m %u
 Icon=kvirc
 Type=Application
