--- CMakeLists.txt.orig	2009-12-02 18:14:16.000000000 +0200
+++ CMakeLists.txt	2009-12-02 18:17:54.000000000 +0200
@@ -53,19 +53,7 @@
 
 STRING(COMPARE EQUAL "${MANUAL_REVISION}" "" MANUAL_REVISION_UNSET)
 IF(MANUAL_REVISION_UNSET)
-	FIND_PACKAGE(Subversion)
-	IF(Subversion_FOUND)
-		MESSAGE(STATUS "Found Subversion ${Subversion_VERSION_SVN}")
-		Subversion_WC_INFO(${PROJECT_SOURCE_DIR} KVIrc)
- 		IF(DEFINED KVIrc_WC_REVISION)
-			SET(CMAKE_KVIRC_BUILD_REVISION ${KVIrc_WC_REVISION})
-			MESSAGE(STATUS "Current revision is ${KVIrc_WC_REVISION}")
-		ELSE()
-			MESSAGE(STATUS "No Subversion working copy found")
-		ENDIF()
-	ELSE()
-		MESSAGE("Subversion not found. No revision number will be used")
-	ENDIF()
+	MESSAGE("No revision number will be used")
 ELSE()
 	SET(CMAKE_KVIRC_BUILD_REVISION ${MANUAL_REVISION})
 	MESSAGE(STATUS "Current revision has been manually set to ${MANUAL_REVISION}")
