diff -Nur qemu-kvm-0.14.0-old//vl.c qemu-kvm-0.14.0/vl.c
--- qemu-kvm-0.14.0-old//vl.c	2011-03-15 15:39:56.011000075 +0200
+++ qemu-kvm-0.14.0/vl.c	2011-03-15 15:44:13.917000076 +0200
@@ -2931,7 +2931,8 @@
                 }
                 exit(1);
             }
-            fprintf(stderr, "Could not initialize KVM, will disable KVM support\n");
+            fprintf(stderr, "Could not initialize KVM, Do you have kvm-amd or kvm-intel modprobe'd?\nIf you want to use CPU emulation, start with -no-kvm.\n");
+            exit(1);
         }
         kvm_allowed = ret >= 0;
     }
