Index: qemu-kvm-0.11.1/vl.c
===================================================================
--- qemu-kvm-0.11.1.orig/vl.c
+++ qemu-kvm-0.11.1/vl.c
@@ -175,7 +175,7 @@ int main(int argc, char **argv)
 //#define DEBUG_NET
 //#define DEBUG_SLIRP
 
-#define DEFAULT_RAM_SIZE 128
+#define DEFAULT_RAM_SIZE 384
 
 /* Max number of USB devices that can be specified on the commandline.  */
 #define MAX_USB_CMDLINE 8
