diff -Nur qemu-0.12.4-old/Makefile qemu-0.12.4/Makefile
--- qemu-0.12.4-old/Makefile	2010-05-26 15:54:31.129350501 +0300
+++ qemu-0.12.4/Makefile	2010-05-26 15:54:45.424975507 +0300
@@ -288,8 +288,6 @@
 endif
 
 install-doc: $(DOCS)
-	$(INSTALL_DIR) "$(DESTDIR)$(docdir)"
-	$(INSTALL_DATA) qemu-doc.html  qemu-tech.html "$(DESTDIR)$(docdir)"
 ifdef CONFIG_POSIX
 	$(INSTALL_DIR) "$(DESTDIR)$(mandir)/man1"
 	$(INSTALL_DATA) qemu.1 qemu-img.1 "$(DESTDIR)$(mandir)/man1"
