--- scribus/prefsmanager.cpp	2006-07-29 02:29:59.000000000 +0300
+++ scribus/prefsmanager.cpp.new	2006-10-07 22:25:58.000000000 +0300
@@ -117,7 +117,7 @@
 	bool goodFont = false;
 	for ( SCFontsIterator itf(appPrefs.AvailFonts); itf.current(); ++itf)
 	{
-		if ((itf.currentKey() == "Arial Regular") || (itf.currentKey() == "Times New Roman Regular"))
+		if ((itf.currentKey() == "Arial Regular") || (itf.currentKey() == "Times New Roman Regular") || (itf.currentKey() == "Nimbus Roman No9 L Regular"))
 		{
 			appPrefs.toolSettings.defFont = itf.currentKey();
 			goodFont = true;
