diff -ruN dansguardian-2.10.1.1.orig/src/FatController.cpp dansguardian-2.10.1.1/src/FatController.cpp
--- dansguardian-2.10.1.1.orig/src/FatController.cpp	2009-08-04 11:46:41.465902112 +0200
+++ dansguardian-2.10.1.1/src/FatController.cpp	2009-08-04 11:48:43.541651965 +0200
@@ -2074,7 +2074,7 @@
 	int tofind;
 	reloadconfig = false;
 
-	syslog(LOG_INFO, "Started sucessfully.");
+	syslog(LOG_INFO, "Started successfully.");
 
 	while (failurecount < 30 && !ttg && !reloadconfig) {
 
diff -ruN dansguardian-2.10.1.1.orig/src/FDTunnel.cpp dansguardian-2.10.1.1/src/FDTunnel.cpp
--- dansguardian-2.10.1.1.orig/src/FDTunnel.cpp	2009-08-04 11:46:41.464902536 +0200
+++ dansguardian-2.10.1.1/src/FDTunnel.cpp	2009-08-04 11:47:37.011652752 +0200
@@ -114,7 +114,7 @@
 	bool done = false;  // so we get past the first while
 
 	while (!done && (targetthroughput > -1 ? throughput < targetthroughput : true)) {
-		done = true;  // if we don't make a sucessful read and write this
+		done = true;  // if we don't make a successful read and write this
 		// flag will stay true and so the while() will exit
 
 		inset = fdSet;  // as select() can modify the sets we need to take
diff -ruN dansguardian-2.10.1.1.orig/src/FOptionContainer.cpp dansguardian-2.10.1.1/src/FOptionContainer.cpp
--- dansguardian-2.10.1.1.orig/src/FOptionContainer.cpp	2009-08-04 11:46:41.464902536 +0200
+++ dansguardian-2.10.1.1/src/FOptionContainer.cpp	2009-08-04 11:48:03.572653450 +0200
@@ -56,7 +56,7 @@
 	if (inet_aton(ip, &address)) {	// convert to in_addr
 		struct hostent *answer;
 		answer = gethostbyaddr((char *) &address, sizeof(address), AF_INET);
-		if (answer) {	// sucess in reverse dns
+		if (answer) {	// success in reverse dns
 			result->push_back(String(answer->h_name));
 			for (addrptr = (struct in_addr **) answer->h_addr_list; *addrptr; addrptr++) {
 				result->push_back(String(inet_ntoa(**addrptr)));
diff -ruN dansguardian-2.10.1.1.orig/src/ListContainer.cpp dansguardian-2.10.1.1/src/ListContainer.cpp
--- dansguardian-2.10.1.1.orig/src/ListContainer.cpp	2009-08-04 11:46:41.460901992 +0200
+++ dansguardian-2.10.1.1/src/ListContainer.cpp	2009-08-04 11:47:22.300690828 +0200
@@ -246,7 +246,7 @@
 		}
 	}
 	listfile.close();
-	return true;  // sucessful read
+	return true;  // successful read
 }
 
 // for phrase lists - helper function for readPhraseList
@@ -502,7 +502,7 @@
 			addToItemList(temp.toCharArray(), temp.length());  // add to unsorted list
 	}
 	listfile.close();
-	return true;  // sucessful read
+	return true;  // successful read
 }
 
 // for item lists - read nested item lists