--- dansguardian-2.9.9.1.orig/configs/dansguardianf1.conf.in    2007-07-11 12:17:59.000000000 +0300
+++ dansguardian-2.9.9.1/configs/dansguardianf1.conf.in 2007-09-09 22:42:20.000000000 +0300
@@ -71,9 +71,9 @@
 # Supply categorised lists here and the category string shall be logged against
 # matching requests, but matching these lists does not perform any filtering
 # action.
-#logsitelist = '@DGCONFDIR@/lists/logsitelist'
-#logurllist = '@DGCONFDIR@/lists/logurllist'
-#logregexpurllist = '@DGCONFDIR@/lists/logregexpurllist'
+logsitelist = '@DGCONFDIR@/lists/logsitelist'
+logurllist = '@DGCONFDIR@/lists/logurllist'
+logregexpurllist = '@DGCONFDIR@/lists/logregexpurllist'
 
 # Outgoing HTTP header rules:
 # Optional lists for blocking based on, and modification of, outgoing HTTP
