diff -Nur xdg-utils-1.0.2.20100618-old//scripts/Makefile.in xdg-utils-1.0.2.20100618//scripts/Makefile.in
--- xdg-utils-1.0.2.20100618-old//scripts/Makefile.in	2010-10-27 10:39:56.381000014 +0300
+++ xdg-utils-1.0.2.20100618//scripts/Makefile.in	2010-10-27 10:41:44.799000093 +0300
@@ -20,11 +20,11 @@
     xdg-open \
     xdg-email \
     xdg-screensaver \
-    xdg-settings
-#    xdg-su
-#    xdg-copy \
-#    xdg-file-dialog
-#    xdg-terminal
+    xdg-settings \
+    xdg-su \
+    xdg-copy \
+    xdg-file-dialog \
+    xdg-terminal
 
 MANPAGES=	$(SCRIPTS:%=man/%.1)
 WEBPAGES=	$(SCRIPTS:%=%.html)
