--- scripts/xdg-utils-common.in.OSmanOS	        2012-10-08 18:26:46.000000000 +0300
+++ scripts/xdg-utils-common.in	                2013-04-05 18:22:55.816631216 +0300
@@ -280,7 +280,7 @@
       elif [ x"$GNOME_DESKTOP_SESSION_ID" != x"" ]; then DE=gnome;
       elif [ x"$MATE_DESKTOP_SESSION_ID" != x"" ]; then DE=mate;
       elif `dbus-send --print-reply --dest=org.freedesktop.DBus /org/freedesktop/DBus org.freedesktop.DBus.GetNameOwner string:org.gnome.SessionManager > /dev/null 2>&1` ; then DE=gnome;
-      elif xprop -root _DT_SAVE_MODE 2> /dev/null | grep ' = \"xfce4\"$' >/dev/null 2>&1; then DE=xfce;
+      elif xprop -root | grep -qw XFCE_DESKTOP_WINDOW 2>&1; then DE=xfce;
       elif xprop -root 2> /dev/null | grep -i '^xfce_desktop_window' >/dev/null 2>&1; then DE=xfce
       fi
     fi
