Index: php-5.2.13/configure.in
===================================================================
--- php-5.2.13.orig/configure.in
+++ php-5.2.13/configure.in
@@ -1,7 +1,7 @@
 ## $Id: configure.in 295464 2010-02-24 12:23:46Z iliaa $ -*- autoconf -*-
 dnl ## Process this file with autoconf to produce a configure script.
 
-divert(1)
+divert(1001)
 
 dnl ## Diversion 1 is the autoconf + automake setup phase. We also
 dnl ## set the PHP version, deal with platform-specific compile
@@ -257,7 +257,7 @@ sinclude(TSRM/tsrm.m4)
 
 sinclude(main/suhosin_patch.m4)
 
-divert(2)
+divert(1002)
 
 dnl ## Diversion 2 is where we set PHP-specific options and come up
 dnl ## with reasonable default values for them. We check for pthreads here
@@ -296,7 +296,7 @@ if test "$enable_maintainer_zts" = "yes"
   PTHREADS_FLAGS
 fi
 
-divert(3)
+divert(1003)
 
 dnl ## In diversion 3 we check for compile-time options to the PHP
 dnl ## core and how to deal with different system dependencies.
@@ -656,7 +656,7 @@ if test "x$php_crypt_r" = "x1"; then
   PHP_CRYPT_R_STYLE
 fi
 
-divert(4)
+divert(1004)
 
 dnl ## In diversion 4 we check user-configurable general settings.
 
@@ -897,7 +897,7 @@ else
   AC_MSG_RESULT([using system default])
 fi
 
-divert(5)
+divert(1005)
 
 dnl ## In diversion 5 we check which extensions should be compiled.
 dnl ## All of these are normally in the extension directories.
Index: php-5.2.13/ext/standard/config.m4
===================================================================
--- php-5.2.13.orig/ext/standard/config.m4
+++ php-5.2.13/ext/standard/config.m4
@@ -1,6 +1,6 @@
 dnl $Id: config.m4 291364 2009-11-27 23:41:13Z rasmus $ -*- autoconf -*-
 
-divert(3)dnl
+divert(1003)dnl
 
 dnl
 dnl Check if flush should be called explicitly after buffered io
@@ -205,7 +205,7 @@ dnl
 AC_CHECK_FUNCS(getcwd getwd asinh acosh atanh log1p hypot glob strfmon nice fpclass isinf isnan)
 AC_FUNC_FNMATCH	
 
-divert(5)dnl
+divert(1005)dnl
 
 dnl
 dnl Check for regex library type
Index: php-5.2.13/scripts/phpize.m4
===================================================================
--- php-5.2.13.orig/scripts/phpize.m4
+++ php-5.2.13/scripts/phpize.m4
@@ -1,6 +1,6 @@
 dnl This file becomes configure.in for self-contained extensions.
 
-divert(1)
+divert(1001)
 
 AC_PREREQ(2.13)
 AC_INIT(config.m4)
