--- php-5.2.14.orig/main/snprintf.c	2011-03-24 08:14:09.927000101 +0200
+++ php-5.2.14/main/snprintf.c	2011-03-24 08:40:40.576000100 +0200
@@ -675,10 +675,6 @@
 
 				/*
 				 * Check if a precision was specified
-				 *
-				 * XXX: an unreasonable amount of precision may be specified
-				 * resulting in overflow of num_buf. Currently we
-				 * ignore this possibility.
 				 */
 				if (*fmt == '.') {
 					adjust_precision = YES;
@@ -692,6 +688,10 @@
 							precision = 0;
 					} else
 						precision = 0;
+
+ 					if (precision > FORMAT_CONV_MAX_PRECISION) {
+ 						precision = FORMAT_CONV_MAX_PRECISION;
+ 					}
 				} else
 					adjust_precision = NO;
 			} else
--- php-5.2.14.orig/main/snprintf.h	2011-03-24 08:14:09.925000101 +0200
+++ php-5.2.14/main/snprintf.h	2011-03-24 08:28:59.215000101 +0200
@@ -148,6 +148,17 @@
 extern char * ap_php_conv_p2(register u_wide_int num, register int nbits,
 		 char format, char *buf_end, register int *len);
 
+/* The maximum precision that's allowed for float conversion. Does not include
+* decimal separator, exponent, sign, terminator. Currently does not affect
+* the modes e/f, only g/k/H, as those have a different limit enforced at
+* another level (see NDIG in php_conv_fp()).
+* Applies to the formatting functions of both spprintf.c and snprintf.c, which
+* use equally sized buffers of MAX_BUF_SIZE = 512 to hold the result of the
+* call to php_gcvt().
+* This should be reasonably smaller than MAX_BUF_SIZE (I think MAX_BUF_SIZE - 9
+* should be enough, but let's give some more space) */
+#define FORMAT_CONV_MAX_PRECISION 500
+
 #endif /* SNPRINTF_H */
 
 /*
--- php-5.2.14.orig/main/spprintf.c	2011-03-24 08:14:09.926000101 +0200
+++ php-5.2.14/main/spprintf.c	2011-03-24 08:41:53.721000100 +0200
@@ -282,10 +282,6 @@
 
 				/*
 				 * Check if a precision was specified
-				 *
-				 * XXX: an unreasonable amount of precision may be specified
-				 * resulting in overflow of num_buf. Currently we
-				 * ignore this possibility.
 				 */
 				if (*fmt == '.') {
 					adjust_precision = YES;
@@ -299,6 +295,10 @@
 							precision = 0;
 					} else
 						precision = 0;
+
+ 					if (precision > FORMAT_CONV_MAX_PRECISION) {
+ 						precision = FORMAT_CONV_MAX_PRECISION;
+ 					}
 				} else
 					adjust_precision = NO;
 			} else
