--- php-5.2.14.orig/NEWS	2010-07-21 17:19:59.000000000 +0300
+++ php-5.2.14/NEWS	2010-12-21 22:30:54.122578083 +0200
@@ -1,5 +1,28 @@
-PHP                                                                        NEWS
+﻿PHP                                                                        NEWS
 |||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
+?? ??? 2010, PHP 5.2.15
+- Fixed a possible double free in imap extension (Identified by Mateusz 
+  Kocielski). (Ilia)
+- Fixed possible flaw in open_basedir (CVE-2010-3436). (Pierre)
+- Fixed possible crash in mssql_fetch_batch(). (Kalle)
+
+- Fixed NULL pointer dereference in ZipArchive::getArchiveComment.
+  (CVE-2010-3709). (Maksymilian Arciemowicz)
+- Fixed bug #52929 (Segfault in filter_var with FILTER_VALIDATE_EMAIL with
+  large amount of data). (CVE-2010-3709). (Adam)
+- Fixed bug #52879 (Objects unreferenced in __get, __set, __isset or __unset
+  can be freed too early). (mail_ben_schmidt at yahoo dot com dot au, Dmitry)
+- Fixed bug #52772 (var_dump() doesn't check for the existence of 
+  get_class_name before calling it). (Kalle, Gustavo)
+- Fixed bug #52546 (pdo_dblib segmentation fault when iterating MONEY values).
+  (Felipe, Adam)
+- Fixed bug #52436 (Compile error if systems do not have stdint.h)
+  (Sriram Natarajan)
+- Fixed bug #52390 (mysqli_report() should be per-request setting). (Kalle)
+- Fixed bug #51008 (Zend/tests/bug45877.phpt fails). (Dmitry)
+- Fixed bug #44248 (RFC2616 transgression while HTTPS request through proxy
+  with SoapClient object). (Dmitry)
+
 22 Jul 2010, PHP 5.2.14
 - Reverted bug fix #49521 (PDO fetchObject sets values before calling
   constructor). (Felipe)
@@ -28,6 +51,8 @@
   PDOStatement if instantiated directly instead of doing by the PDO methods.
   (Felipe)
 
+- Fixed bug #52944 (Invalid write on second and subsequent reads with an
+  inflate filter fed invalid data). (Gustavo)
 - Fixed bug #52317 (Segmentation fault when using mail() on a rhel 4.x (only 64
   bit)). (Adam)
 - Fixed bug #52238 (Crash when an Exception occured in iterator_to_array).


--- php-5.2.14.orig/ext/imap/php_imap.c	2010-02-07 15:06:54.000000000 +0200
+++ php-5.2.14/ext/imap/php_imap.c	2010-12-21 22:35:02.386608166 +0200
@@ -26,7 +26,7 @@
    | PHP 4.0 updates:  Zeev Suraski <zeev@zend.com>                       |
    +----------------------------------------------------------------------+
  */
-/* $Id: php_imap.c 294699 2010-02-07 13:06:54Z pajoye $ */
+/* $Id$ */
 
 #define IMAP41
 
@@ -794,10 +794,12 @@
 
 	if (IMAPG(imap_user)) {
 		efree(IMAPG(imap_user));
+		IMAPG(imap_user) = 0;
 	}
 
 	if (IMAPG(imap_password)) {
 		efree(IMAPG(imap_password));
+		IMAPG(imap_password) = 0;
 	}
 
 	/* local filename, need to perform open_basedir and safe_mode checks */
