--- php-5.2.14.orig/ext/shmop/shmop.c	2011-03-14 08:48:28.203000094 +0200
+++ php-5.2.14/ext/shmop/shmop.c	2011-03-14 09:20:36.283000094 +0200
@@ -223,7 +223,7 @@
 		RETURN_FALSE;
 	}
 
-	if (start + count > shmop->size || count < 0) {
+	if (count < 0 || start > (INT_MAX - count) || start + count > shmop->size) {
 		php_error_docref(NULL TSRMLS_CC, E_WARNING, "count is out of range");
 		RETURN_FALSE;
 	}
