--- php-5.2.14.orig/ext/calendar/julian.c	2011-03-23 17:29:36.742190107 +0200
+++ php-5.2.14/ext/calendar/julian.c	2011-03-23 17:54:05.791019112 +0200
@@ -146,6 +146,7 @@
  **************************************************************************/
 
 #include "sdncal.h"
+#include <limits.h>
 
 #define JULIAN_SDN_OFFSET         32083
 #define DAYS_PER_5_MONTHS  153
@@ -164,15 +165,22 @@
 	int dayOfYear;
 
 	if (sdn <= 0) {
-		*pYear = 0;
-		*pMonth = 0;
-		*pDay = 0;
-		return;
+        goto fail;
 	}
-	temp = (sdn + JULIAN_SDN_OFFSET) * 4 - 1;
+    /* Check for overflow */
+    if (sdn > (LONG_MAX - JULIAN_SDN_OFFSET * 4 + 1) / 4 || sdn < LONG_MIN / 4) {
+        goto fail;
+    }
+	temp = sdn * 4 + (JULIAN_SDN_OFFSET * 4 - 1);
 
 	/* Calculate the year and day of year (1 <= dayOfYear <= 366). */
-	year = temp / DAYS_PER_4_YEARS;
+    {
+	    long yearl = temp / DAYS_PER_4_YEARS;
+        if (yearl > INT_MAX || yearl < INT_MIN) {
+            goto fail;
+        }
+        year = (int) yearl;
+    }
 	dayOfYear = (temp % DAYS_PER_4_YEARS) / 4 + 1;
 
 	/* Calculate the month and day of month. */
@@ -196,6 +204,12 @@
 	*pYear = year;
 	*pMonth = month;
 	*pDay = day;
+    return;
+
+fail:
+    *pYear = 0;
+    *pMonth = 0;
+    *pDay = 0;
 }
 
 long int JulianToSdn(
