--- ext/mysql/php_mysql.c.orig	2008-04-16 11:54:44.000000000 +0300
+++ ext/mysql/php_mysql.c	2008-05-05 16:18:02.000000000 +0300
@@ -33,6 +33,7 @@
 #include "php_globals.h"
 #include "ext/standard/info.h"
 #include "ext/standard/php_string.h"
+#include "SAPI.h"
 
 #ifdef ZEND_ENGINE_2
 # include "zend_exceptions.h"
@@ -515,7 +516,7 @@
 {
 	char *user=NULL, *passwd=NULL, *host_and_port=NULL, *socket=NULL, *tmp=NULL, *host=NULL;
 	int  user_len, passwd_len, host_len;
-	char *hashed_details=NULL;
+	char *hashed_details=NULL, *option_section=NULL;
 	int hashed_details_length, port = MYSQL_PORT;
 	long client_flags = 0;
 	php_mysql_conn *mysql=NULL;
@@ -644,6 +645,12 @@
 				mysql_options(&mysql->conn, MYSQL_OPT_CONNECT_TIMEOUT, (const char *)&connect_timeout);
 			}
 
+			/* Read settings from my.cnf */
+			option_section = emalloc(strlen("php-%s") + strlen(sapi_module.name));
+			sprintf(option_section, "php-%s", sapi_module.name);
+			mysql_options(&mysql->conn, MYSQL_READ_DEFAULT_GROUP, option_section);
+			efree(option_section);
+
 			if (mysql_real_connect(&mysql->conn, host, user, passwd, NULL, port, socket, client_flags)==NULL) {
 #else
 			if (mysql_connect(&mysql->conn, host, user, passwd)==NULL) {
@@ -687,6 +694,13 @@
 #endif
 				signal(SIGPIPE, handler);
 #endif /* end mysql_ping */
+
+			/* Read settings from my.cnf */
+			option_section = emalloc(strlen("php-%s") + strlen(sapi_module.name));
+			sprintf(option_section, "php-%s", sapi_module.name);
+			mysql_options(le->ptr, MYSQL_READ_DEFAULT_GROUP, option_section);
+			efree(option_section);
+
 #if MYSQL_VERSION_ID > 32199 /* this lets us set the port number */
 				if (mysql_real_connect(le->ptr, host, user, passwd, NULL, port, socket, client_flags)==NULL) {
 #else
@@ -751,6 +765,12 @@
 			mysql_options(&mysql->conn, MYSQL_OPT_CONNECT_TIMEOUT, (const char *)&connect_timeout);
 		}
 
+		/* Read settings from my.cnf */
+		option_section = emalloc(strlen("php-%s") + strlen(sapi_module.name));
+		sprintf(option_section, "php-%s", sapi_module.name);
+		mysql_options(&mysql->conn, MYSQL_READ_DEFAULT_GROUP, option_section);
+		efree(option_section);
+
 		if (mysql_real_connect(&mysql->conn, host, user, passwd, NULL, port, socket, client_flags)==NULL) {
 #else
 		if (mysql_connect(&mysql->conn, host, user, passwd)==NULL) {
