--- ext/mysqli/mysqli_nonapi.c	2006-04-13 16:59:11.000000000 +0200
+++ ext/mysqli/mysqli_nonapi.c	2006-04-13 17:01:50.000000000 +0200
@@ -28,6 +28,7 @@
 #include "php_ini.h"
 #include "ext/standard/info.h"
 #include "php_mysqli.h"
+#include "SAPI.h"
 
 /* {{{ proto object mysqli_connect([string hostname [,string username [,string passwd [,string dbname [,int port [,string socket]]]]]])
    Open a connection to a mysql server */ 
@@ -36,7 +37,7 @@
 	MY_MYSQL 			*mysql;
 	MYSQLI_RESOURCE 	*mysqli_resource;
 	zval  				*object = getThis();
-	char 				*hostname = NULL, *username=NULL, *passwd=NULL, *dbname=NULL, *socket=NULL;
+	char 				*hostname = NULL, *username=NULL, *passwd=NULL, *dbname=NULL, *socket=NULL, *option_section=NULL;
 	unsigned int 		hostname_len = 0, username_len = 0, passwd_len = 0, dbname_len = 0, socket_len = 0;
 	long				port=0;
 
@@ -83,6 +84,12 @@
 	}
 #endif
 
+	/* Read settings from my.cnf */
+	option_section = emalloc(strlen("php-%s") + strlen(sapi_module.name));
+	sprintf(option_section, "php-%s", sapi_module.name);
+	mysql_options(mysql->mysql, MYSQL_READ_DEFAULT_GROUP, option_section);
+	efree(option_section);
+
 	if (!socket) {
 		socket = MyG(default_socket);
 	}
--- ext/mysqli/mysqli_api.c	2006-04-13 14:37:45.000000000 +0200
+++ ext/mysqli/mysqli_api.c	2006-04-13 17:44:32.000000000 +0200
@@ -28,6 +28,7 @@
 #include "php_ini.h"
 #include "ext/standard/info.h"
 #include "php_mysqli.h"
+#include "SAPI.h"
 
 /* {{{ proto mixed mysqli_affected_rows(object link)
    Get number of affected rows in previous MySQL operation */
@@ -1373,7 +1374,7 @@
 PHP_FUNCTION(mysqli_real_connect)
 {
 	MY_MYSQL 		*mysql;
-	char 			*hostname = NULL, *username=NULL, *passwd=NULL, *dbname=NULL, *socket=NULL;
+	char 			*hostname = NULL, *username=NULL, *passwd=NULL, *dbname=NULL, *socket=NULL, *option_section=NULL;
 	unsigned int 	hostname_len = 0, username_len = 0, passwd_len = 0, dbname_len = 0, socket_len = 0;
 	unsigned long 	port=0, flags=0;
 	zval			*mysql_link;
@@ -1411,6 +1412,12 @@
 		flags ^= CLIENT_LOCAL_FILES;
 	}
 
+	/* Read settings from my.cnf */
+	option_section = emalloc(strlen("php-%s") + strlen(sapi_module.name));
+	sprintf(option_section, "php-%s", sapi_module.name);
+	mysql_options(mysql->mysql, MYSQL_READ_DEFAULT_GROUP, option_section);
+	efree(option_section);
+
 	if (!socket) {
 		socket = MyG(default_socket);
 	}
