--- php-5.2.14.orig/ext/sockets/sockets.c	2011-05-27 11:06:13.907000256 +0300
+++ php-5.2.14/ext/sockets/sockets.c	2011-05-27 11:10:55.054000256 +0300
@@ -1176,6 +1176,11 @@
 			break;
 
 		case AF_UNIX:
+			if (addr_len >= sizeof(s_un.sun_path)) {
+				php_error_docref(NULL TSRMLS_CC, E_WARNING, "Path too long", php_sock->type);
+				RETURN_FALSE;
+			}
+
 			memset(&s_un, 0, sizeof(struct sockaddr_un));
 
 			s_un.sun_family = AF_UNIX;
