--- php.ini-dist	2007-08-22 02:24:18.000000000 +0300
+++ php.ini-dist	2007-12-11 18:21:25.000000000 +0200
@@ -1056,14 +1056,14 @@
 session.referer_check =
 
 ; How many bytes to read from the file.
-session.entropy_length = 0
+;session.entropy_length = 0
 
 ; Specified here to create the session id.
-session.entropy_file =
+;session.entropy_file =
 
-;session.entropy_length = 16
+session.entropy_length = 16
 
-;session.entropy_file = /dev/urandom
+session.entropy_file = /dev/urandom
 
 ; Set to {nocache,private,public,} to determine HTTP caching aspects
 ; or leave this empty to avoid sending anti-caching headers.
