--- php-5.2.14.orig/ext/standard/rand.c	2011-04-04 13:10:46.071000101 +0300
+++ php-5.2.14/ext/standard/rand.c	2011-04-04 13:08:02.056000101 +0300
@@ -315,18 +315,19 @@
 	long number;
 	int  argc = ZEND_NUM_ARGS();
 
-	if (argc != 0 && zend_parse_parameters(argc TSRMLS_CC, "ll", &min, &max) == FAILURE)
-		return;
+    if (argc != 0) {
+        if (zend_parse_parameters(argc TSRMLS_CC, "ll", &min, &max) == FAILURE) {
+            return;
+        } else if (max < min) {
+            php_error_docref(NULL TSRMLS_CC, E_WARNING, "max(%ld) is smaller than min(%ld)", max, min);
+            RETURN_FALSE;
+        }
+    }
 
 	if (!BG(mt_rand_is_seeded)) {
 		php_mt_srand(GENERATE_SEED() TSRMLS_CC);
 	}
 
-    if (max < min) {
-        php_error_docref(NULL TSRMLS_CC, E_WARNING, "max(%d) is smaller than min(%d)", max, min);
-        RETURN_FALSE;
-    }
-
 	/*
 	 * Melo: hmms.. randomMT() returns 32 random bits...
 	 * Yet, the previous php_rand only returns 31 at most.
