Index: geogebra-3.2.46.0/geogebra.desktop
===================================================================
--- geogebra-3.2.46.0.orig/geogebra.desktop
+++ geogebra-3.2.46.0/geogebra.desktop
@@ -1,6 +1,7 @@
 #!/usr/bin/env xdg-open
 [Desktop Entry]
 Name=GeoGebra
+Name[tr]=GeoGebra
 GenericName=Dynamic mathematics software
 GenericName[cs]=Dynamický matematický software
 GenericName[de]=Dynamische Mathematiksoftware
@@ -9,6 +10,7 @@ GenericName[fr]=Logiciel de Mathématiqu
 GenericName[hi]=सक्रिय गणितीय सॉफ्टवेर 
 GenericName[it]=Software di matematica dinamica
 GenericName[pt_BR]=Programa de Matemática Dinâmica
+GenericName[tr]=Bilimsel Matematik Uygulaması
 Comment=Create interactive mathematical constructions and applets. 
 Comment[cs]=Vytvořte interaktivní konstrukce a aplety.
 Comment[de]=Erstelle interaktive Konstruktionen und Applets.
@@ -16,11 +18,12 @@ Comment[fr]=Créer des constructions mat
 Comment[hi]=संवादात्मक गणितीय निर्माण और एप्लेट बनाएँ
 Comment[it]=Creazione di costruzioni matematiche e applet interattive.
 Comment[pt_BR]=Crie construções matemáticas interativas e applets.
+Comment[tr]= Geometri, cebir ve hesap işlerini bir araya getiren dinamik bir matematik uygulaması
 Exec=geogebra
 Icon=geogebra
 Terminal=false
 Type=Application
 StartupNotify=true
 StartupWMClass=geogebra-GeoGebra
-Categories=Education;Math
-MimeType=application/vnd.geogebra.file;application/vnd.geogebra.tool;
\ Dosya sonunda yenisatır yok.
+Categories=Education;Math;
+MimeType=application/vnd.geogebra.file;application/vnd.geogebra.tool;
