Index: pTk/Xlib.h
===================================================================
--- pTk/Xlib.h.orig	2010-11-19 11:26:37.983842257 +0100
+++ pTk/Xlib.h	2010-11-19 11:27:23.559717781 +0100
@@ -1,5 +1,8 @@
 #ifndef _XLIB
 #define _XLIB
+#ifdef _X11_XLIB_H_
+#define _XLIB_H_
+#endif
 #ifndef _XLIB_H_
 extern XFontStruct *XLoadQueryFont  _ANSI_ARGS_((Display *, const char *));
 extern XModifierKeymap *XGetModifierMapping  _ANSI_ARGS_((Display *));
