diff -uNr Coin-2.4.6.orig/src/glue/openal_wrapper.c Coin-2.4.6/src/glue/openal_wrapper.c
--- Coin-2.4.6.orig/src/glue/openal_wrapper.c	2007-04-10 11:23:51.000000000 +0200
+++ Coin-2.4.6/src/glue/openal_wrapper.c	2007-04-14 06:31:09.000000000 +0200
@@ -107,7 +107,7 @@
       int idx;
       const char * possiblelibnames[] = {
         NULL, /* is set below */
-        "openal", "openal32", "libopenal.so",
+	@LIBOPENAL_SONAME@,
         NULL
       };
 
