diff -uNr Coin-2.4.6.orig/man/man3/Makefile.am Coin-2.4.6/man/man3/Makefile.am
--- Coin-2.4.6.orig/man/man3/Makefile.am	2007-04-10 11:24:07.000000000 +0200
+++ Coin-2.4.6/man/man3/Makefile.am	2007-04-14 06:27:29.000000000 +0200
@@ -8,6 +8,7 @@
 	"$(sim_ac_doxygen_exe)" docs/coin.doxygen )
 
 filelist.txt: SoDB.3
+	rm -f *_.3
 	ls *.3 >filelist.txt
 
 
diff -uNr Coin-2.4.6.orig/man/man3/Makefile.in Coin-2.4.6/man/man3/Makefile.in
--- Coin-2.4.6.orig/man/man3/Makefile.in	2007-04-10 11:24:07.000000000 +0200
+++ Coin-2.4.6/man/man3/Makefile.in	2007-04-14 06:27:30.000000000 +0200
@@ -421,6 +421,7 @@
 	"$(sim_ac_doxygen_exe)" docs/coin.doxygen )
 
 filelist.txt: SoDB.3
+	rm -f *_.3
 	ls *.3 >filelist.txt
 
 @BUILD_MANPAGES_TRUE@all-local: filelist.txt
