--- eric/UI/UserInterface.py~	2013-01-08 15:38:15.754469257 +0100
+++ eric/UI/UserInterface.py	2013-01-08 15:38:48.518468060 +0100
@@ -1650,6 +1650,7 @@
             """<b>Check for Updates...</b>"""
             """<p>Checks the internet for updates of eric4.</p>"""
                              ))
+        self.checkUpdateAct.setVisible(False)
         self.connect(self.checkUpdateAct, SIGNAL('triggered()'), self.performVersionCheck)
         self.actions.append(self.checkUpdateAct)
     
@@ -1663,6 +1664,7 @@
             """<p>Shows the eric4 versions available for download """
             """from the internet.</p>"""
                              ))
+        self.checkUpdateAct.setVisible(False)
         self.connect(self.showVersionsAct, SIGNAL('triggered()'), 
                 self.showAvailableVersionsInfo)
         self.actions.append(self.showVersionsAct)
