--- eric4-4.3.1/install.py.orig	2009-03-11 13:45:51.000000000 +0100
+++ eric4-4.3.1/install.py	2009-03-11 13:49:36.000000000 +0100
@@ -204,7 +204,7 @@ def createGlobalPluginsDir():
     fname = os.path.join(pdir, "__init__.py")
     if not os.path.exists(fname):
         if not os.path.exists(pdir):
-            os.mkdir(pdir,  0755)
+            os.makedirs(pdir,  0755)
         f = open(fname, "wb")
         f.write(\
 '''# -*- coding: utf-8 -*-
