--- lib-src/lv2/configure_old	2014-11-13 00:14:55.893037393 +0200
+++ lib-src/lv2/configure	2014-11-13 00:10:16.000000000 +0200
@@ -5,7 +5,7 @@
    pkg="$1"
    shift
    pushd >/dev/null ${pkg}
-   $(which python python2 | tail -1) waf --prefix="." --include="." $@ configure || exit 1
+   python waf --prefix="." --include="." $@ configure || exit 1
    popd >/dev/null
 
    if [ "${pkg}" == "lv2" ]

--- lib-src/lv2/build_old	2014-09-22 23:38:14.000000000 +0300
+++ lib-src/lv2/build	2014-11-13 00:38:00.760975649 +0200
@@ -5,7 +5,7 @@
    pkg=$1
    pushd >/dev/null ${pkg}
    shift
-   $(which python python2 | tail -1) waf --prefix="." --include="." $@ build || exit 1
+   python waf --prefix="." --include="." $@ build || exit 1
    popd >/dev/null 
 
    export PKG_CONFIG_PATH="$(pwd)/${pkg}/build:${PKG_CONFIG_PATH}"


    