diff --git a/libs/internals/connection.cpp b/libs/internals/connection.cpp
index c88ea3b..8e85da2 100644
--- a/libs/internals/connection.cpp
+++ b/libs/internals/connection.cpp
@@ -154,7 +154,7 @@ Connection::Connection(const QString & name, const Connection::Type type, const
 }
 
 Connection::Connection(const QUuid & uuid, const Connection::Type type)
-    : m_uuid(uuid), m_type(type), m_autoConnect(true)
+    : m_uuid(uuid), m_type(type), m_autoConnect(true), m_scope(Connection::System)
 {
     init();
 }
diff --git a/libs/ui/connection.ui b/libs/ui/connection.ui
index 7fea761..dcc046c 100644
--- a/libs/ui/connection.ui
+++ b/libs/ui/connection.ui
@@ -60,6 +60,9 @@
        <property name="whatsThis">
         <string>If checked, the connection settings and secrets will be saved in system storage, which allows the connection to work without a KDE session.</string>
        </property>
+       <property name="checked">
+        <bool>true</bool>
+       </property>
       </widget>
      </item>
     </layout>
