--- generic/jkSoundFile.c	2005-12-14 12:29:38.000000000 +0100
+++ generic/jkSoundFile.c.new	2010-01-25 07:53:45.000000000 +0100
@@ -2371,7 +2371,7 @@
   sprintf(&buf[12], "HEDR");
   PutLELong(buf, 16, 32);
   Tcl_GlobalEvalObj(s->interp, Tcl_NewStringObj(CSL_DATECOMMAND, -1));
-  sprintf(&buf[20], Tcl_GetStringResult(s->interp));
+  sprintf(&buf[20], "%s", Tcl_GetStringResult(s->interp));
   
   PutLELong(buf, 40, s->samprate);
   PutLELong(buf, 44, s->length);
