--- snack2.2.10/generic/SnackOgg.c~     2005-12-14 11:29:38.000000000 +0000
+++ snack2.2.10/generic/SnackOgg.c      2009-10-27 18:01:57.000000000 +0000
@@ -2223,7 +2223,7 @@
   int arg, n = 0, ret;
   OggVorbis_File *of = (OggVorbis_File *)s->extHead2;
   Tcl_Obj **listObj;
-  static char *subOptionStrings[] = {
+  const char *subOptionStrings[] = {
     "-comment", "-maxbitrate", "-minbitrate", "-nominalbitrate",
     "-quality", NULL
   };
@@ -2441,7 +2441,7 @@
 {
   int arg, index;
   OggVorbis_File *of = (OggVorbis_File *)s->extHead2;
-  static char *optionStrings[] = {
+  const char *optionStrings[] = {
     "-comment", "-vendor", "-maxbitrate", "-minbitrate", "-nominalbitrate",
     "-quality", NULL
   };
