--- dhcpcd-hooks/50-dhcpcd-compat.OSmanOS	2013-01-28 11:54:03.000000000 +0200
+++ dhcpcd-hooks/50-dhcpcd-compat	        2013-02-14 21:32:04.430539886 +0200
@@ -30,12 +30,12 @@
 esac
 
 if [ "$r" != "down" ]; then
-	rm -f /var/lib/dhcpcd-"$INTERFACE".info
+	rm -f /var/lib/dhcpcd/dhcpcd-"$INTERFACE".info
 	for x in IPADDR INTERFACE NETMASK BROADCAST NETWORK DHCPSID GATEWAYS \
 		DNSSERVERS DNSDOMAIN DNSSEARCH NISDOMAIN NISSERVERS \
 		NTPSERVERS GATEWAY DNS; do
-		eval echo "$x=\'\$$x\'" >> /var/lib/dhcpcd-"$INTERFACE".info
+		eval echo "$x=\'\$$x\'" >> /var/lib/dhcpcd/dhcpcd-"$INTERFACE".info
 	done
 fi
 
-set -- /var/lib/dhcpcd-"$INTERFACE".info "$r"
+set -- /var/lib/dhcpcd/dhcpcd-"$INTERFACE".info "$x"
