Index: dhcpcd-5.1.5/dhcpcd.c
===================================================================
--- dhcpcd-5.1.5.orig/dhcpcd.c
+++ dhcpcd-5.1.5/dhcpcd.c
@@ -110,8 +110,23 @@ static const struct dhcp_op dhcp_ops[] =
 	{ 0, NULL }
 };
 
+static char * noarp_quirk_table[] = {
+    "P660W-TX",
+    NULL
+};
+
 static void send_release(struct interface *);
 
+static int
+noarp_quirked(const char* server_id)
+{
+	int i;
+	for (i = 0; noarp_quirk_table[i] != NULL; ++i)
+		if (strcmp(noarp_quirk_table[i], server_id) == 0)
+			return 1;
+	return 0;
+}
+
 static const char *
 get_dhcp_op(uint8_t type)
 {
@@ -608,7 +623,9 @@ handle_dhcp(struct interface *iface, str
 		/* If the interface already has the address configured
 		 * then we can't ARP for duplicate detection. */
 		addr.s_addr = state->offer->yiaddr;
-		if (has_address(iface->name, &addr, NULL) != 1) {
+		if (dhcp && noarp_quirked(dhcp->servername))
+			syslog(LOG_INFO, "%s: Skipping ARP check for '%s' as it's blacklisted", iface->name, dhcp->servername);
+		else if (has_address(iface->name, &addr, NULL) != 1) {
 			state->claims = 0;
 			state->probes = 0;
 			state->conflicts = 0;
