diff -ur lame-398.old/frontend/Makefile.am lame-398.new/frontend/Makefile.am
--- lame-398.old/frontend/Makefile.am	2006-09-30 11:17:05.000000000 +0200
+++ lame-398.new/frontend/Makefile.am	2008-07-06 12:36:25.000000000 +0200
@@ -52,7 +52,7 @@
 endif
 
 CFLAGS = @CFLAGS@ @GTK_CFLAGS@ @FRONTEND_CFLAGS@ @SNDFILE_CFLAGS@
-LDFLAGS = @LDFLAGS@ @FRONTEND_LDFLAGS@ @SNDFILE_LIBS@ -static
+LDFLAGS = @LDFLAGS@ @FRONTEND_LDFLAGS@ @SNDFILE_LIBS@
 
 INCLUDES = -I$(top_srcdir)/libmp3lame -I$(top_srcdir)/include -I$(top_builddir)
 
diff -ur lame-398.old/frontend/Makefile.in lame-398.new/frontend/Makefile.in
--- lame-398.old/frontend/Makefile.in	2008-06-25 14:17:51.000000000 +0200
+++ lame-398.new/frontend/Makefile.in	2008-07-06 12:36:39.000000000 +0200
@@ -152,7 +152,7 @@
 	$(top_builddir)/libmp3lame/libmp3lame.la \
 	@FRONTEND_LDADD@
 
-LDFLAGS = @LDFLAGS@ @FRONTEND_LDFLAGS@ @SNDFILE_LIBS@ -static
+LDFLAGS = @LDFLAGS@ @FRONTEND_LDFLAGS@ @SNDFILE_LIBS@
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
 LIBTOOL = @LIBTOOL@
