diff -up kvkbd-0.6/src/CMakeLists.txt.DSO-linking-fix kvkbd-0.6/src/CMakeLists.txt
--- kvkbd-0.6/src/CMakeLists.txt.DSO-linking-fix	2010-02-15 13:57:11.000000000 +0100
+++ kvkbd-0.6/src/CMakeLists.txt	2010-02-15 13:57:51.000000000 +0100
@@ -1,6 +1,7 @@
 find_package(KDE4 REQUIRED)
 
 find_package(LibXslt REQUIRED)
+find_package(X11 REQUIRED)
 
 include_directories( ${KDE4_INCLUDES} )
 
@@ -22,7 +23,7 @@ QT4_ADD_DBUS_ADAPTOR(kvkbd_SRCS org.kde.
 
 kde4_add_executable(kvkbd ${kvkbd_SRCS})
 
-target_link_libraries(kvkbd ${KDE4_KDECORE_LIBS} Xtst ${KDE4_KDEUI_LIBS})
+target_link_libraries(kvkbd ${KDE4_KDECORE_LIBS} Xtst ${KDE4_KDEUI_LIBS} ${X11_LIBRARIES})
 
 install(TARGETS kvkbd ${INSTALL_TARGETS_DEFAULT_ARGS})
 
