--- SConstruct
+++ SConstruct
@@ -90,19 +90,9 @@
 		#env.Append(CPPDEFINES = ['NDEBUG']) 
 
 else:
-	if env['gcc_visibility']: 
-		env.Append(CCFLAGS=['-fvisibility=hidden']);
-		env.Append(CXXFLAGS=['-fvisibility-inlines-hidden', '-fvisibility=hidden']);
-
-	if debug:
-		env.Append(CCFLAGS=['-ggdb', '-D_FORTIFY_SOURCE=2'])
-		env.Append(CPPFLAGS=['-ggdb', '-D_FORTIFY_SOURCE=2'])
-	else: 
-		env.Append(CCFLAGS=['-O3'])
-		env.Append(CPPFLAGS=['-O3'])
 		
-	env.Append(CPPFLAGS=['-Wall', '-pedantic', '-Wno-long-long', '-pipe', '-pthread'])
-	env.Append(CCFLAGS=['-Wall', '-pedantic', '-Wno-long-long', '-pipe', '-pthread'])
+	env.Append(CPPFLAGS=['-Wall', '-pedantic', '-Wno-long-long', '-pthread'])
+	env.Append(CCFLAGS=['-Wall', '-pedantic', '-Wno-long-long', '-pthread'])
 
 
 conf_env = env.Clone()
