diff -up lxdream-0.9.1/configure.implicit lxdream-0.9.1/configure
--- lxdream-0.9.1/configure.implicit	2009-06-28 12:29:06.000000000 +0200
+++ lxdream-0.9.1/configure	2012-02-11 00:21:17.719647658 +0100
@@ -9116,6 +9116,8 @@ _ACEOF
 
 fi
 
+             GTK_LIBS="$GTK_LIBS -lX11"
+       LIBS="$LIBS -lX11 -lm -lz"
    { $as_echo "$as_me:$LINENO: checking for gdk_quartz_window_get_nsview" >&5
 $as_echo_n "checking for gdk_quartz_window_get_nsview... " >&6; }
 if test "${ac_cv_func_gdk_quartz_window_get_nsview+set}" = set; then
diff -up lxdream-0.9.1/configure.in.implicit lxdream-0.9.1/configure.in
--- lxdream-0.9.1/configure.in.implicit	2009-06-28 12:29:06.000000000 +0200
+++ lxdream-0.9.1/configure.in	2012-02-11 00:21:17.720647663 +0100
@@ -210,6 +210,9 @@ else
    AC_CHECK_FUNC(gdk_x11_display_get_xdisplay, [ 
       HAVE_GTK_X11='yes'
       AC_DEFINE([HAVE_GTK_X11],1,[Building with GTK+X11]) ], [])
+      dnl Add -lX11 explicitly as some pkg-configs get it wrong
+      GTK_LIBS="$GTK_LIBS -lX11"
+      LIBS="$LIBS -lX11 -lm -lz"
    AC_CHECK_FUNC(gdk_quartz_window_get_nsview, [ 
       HAVE_GTK_OSX='yes'
       AC_DEFINE([HAVE_GTK_OSX],1,[Building with GTK+Cocoa]) ], [])
