
fix build with 3.0

Signed-off-by: Thomas Backlund <tmb@mageia.org>

 3rdparty/rfswitch/Makefile |    2 --
 1 file changed, 2 deletions(-)

--- a/3rdparty/rfswitch/Makefile.orig	2011-07-15 02:37:27.000000000 +0300
+++ a/3rdparty/rfswitch/Makefile	2011-07-15 13:47:10.463914526 +0300
@@ -29,8 +29,6 @@ ifneq ($(PATCHLEVEL),6) # If we are not
 
 O_TARGET := rfswitch.o
 
-include $(TOPDIR)/Rules.make
-
 endif # End if 2.4 specific settings
 
 else 
