--- ./3rdparty/heci/heci_main.c.orig	2013-01-16 18:13:57.000000000 +0200
+++ ./3rdparty/heci/heci_main.c	2013-01-16 18:53:40.846477255 +0200
@@ -121,9 +121,9 @@ MODULE_DEVICE_TABLE(pci, heci_pci_tbl);
  */
 static int __init heci_init_module(void);
 static void __exit heci_exit_module(void);
-static int __devinit heci_probe(struct pci_dev *pdev,
+static int heci_probe(struct pci_dev *pdev,
 				const struct pci_device_id *ent);
-static void __devexit heci_remove(struct pci_dev *pdev);
+static void heci_remove(struct pci_dev *pdev);
 static int heci_open(struct inode *inode, struct file *file);
 static int heci_release(struct inode *inode, struct file *file);
 static unsigned int heci_legacy_poll(struct file *file, poll_table * wait);
@@ -295,7 +295,7 @@ module_exit(heci_exit_module);
  * 0 on success,
  * negative on failure
  */
-static int __devinit heci_probe(struct pci_dev *pdev,
+static int heci_probe(struct pci_dev *pdev,
 				const struct pci_device_id *ent)
 {
 	struct iamt_heci_device *device = NULL;
@@ -423,7 +423,7 @@ end:
  * @return :
  * none;
  */
-static void __devexit heci_remove(struct pci_dev *pdev)
+static void heci_remove(struct pci_dev *pdev)
 {
 	struct iamt_heci_device *device = pci_get_drvdata(pdev);
 	int err = 0;
