
Fix build for 2.6.37

Signed-off-by: Thomas Backlund <tmb@mandriva.org>

--- a/drivers/gpu/drm/mach64/mach64_drv.c.orig	2011-01-02 16:55:48.000000000 +0200
+++ a/drivers/gpu/drm/mach64/mach64_drv.c	2011-01-02 17:53:54.808775632 +0200
@@ -51,8 +51,6 @@ static struct drm_driver driver = {
 	.irq_uninstall = mach64_driver_irq_uninstall,
 	.irq_handler = mach64_driver_irq_handler,
 	.reclaim_buffers = drm_core_reclaim_buffers,
-	.get_map_ofs = drm_core_get_map_ofs,
-	.get_reg_ofs = drm_core_get_reg_ofs,
 	.ioctls = mach64_ioctls,
 	.dma_ioctl = mach64_dma_buffers,
 	.fops = {
@@ -63,6 +61,7 @@ static struct drm_driver driver = {
 		.mmap = drm_mmap,
 		.poll = drm_poll,
 		.fasync = drm_fasync,
+		.llseek = noop_llseek,
 		},
 	.pci_driver = {
 		.name = DRIVER_NAME,
