
reclaim_buffers is gone in 3.6

Signed-off-by: Thomas Backlund <tmb@mageia.org>

--- linux/drivers/gpu/drm/mach64/mach64_drv.c.orig	2012-10-17 22:26:58.000000000 +0300
+++ linux/drivers/gpu/drm/mach64/mach64_drv.c	2012-10-17 23:05:10.212566630 +0300
@@ -62,7 +62,6 @@ static struct drm_driver driver = {
 	.irq_postinstall = mach64_driver_irq_postinstall,
 	.irq_uninstall = mach64_driver_irq_uninstall,
 	.irq_handler = mach64_driver_irq_handler,
-	.reclaim_buffers = drm_core_reclaim_buffers,
 	.ioctls = mach64_ioctls,
 	.dma_ioctl = mach64_dma_buffers,
 	.fops = &mach64_driver_fops,
