---
 drivers/ide/sis5513.c |    9 +++++++++
 1 file changed, 9 insertions(+)

diff -p -up linux-2.6.28/drivers/ide/sis5513.c.orig linux-2.6.28/drivers/ide/sis5513.c
--- linux-2.6.28/drivers/ide/sis5513.c.orig	2008-12-08 12:41:13.000000000 -0500
+++ linux-2.6.28/drivers/ide/sis5513.c	2008-12-08 12:51:32.000000000 -0500
@@ -412,6 +412,15 @@ static int __devinit sis_find_family(str
 						pci_name(dev));
 				}
 			}
+			else if (trueid == 0x180) { /* sis965L */
+				u16 pci_command;
+				pci_read_config_word(dev, PCI_COMMAND, &pci_command);
+				pci_command &= ~PCI_COMMAND_INTX_DISABLE;
+				pci_write_config_word(dev, PCI_COMMAND, pci_command);
+				chipset_family = ATA_133;
+				printk(KERN_INFO DRV_NAME " %s: SiS 965 IDE UDMA133 controller\n",
+					pci_name(dev));
+			}
 	}
 
 	if (!chipset_family) { /* Belongs to pci-quirks */
