--- linux-2.6.35-rc6-git-mnb0.1/net/ipv4/netfilter/ipt_IFWLOG.c.orig	2010-07-30 21:17:30.000000000 +0300
+++ linux-2.6.35-rc6-git-mnb0.1/net/ipv4/netfilter/ipt_IFWLOG.c	2010-07-31 13:46:33.834611944 +0300
@@ -135,7 +135,7 @@ static void ipt_IFWLOG_packet(const stru
 }
 
 static unsigned int ipt_IFWLOG_target(struct sk_buff *skb,
-				      const struct xt_target_param *target_param)
+				      const struct xt_action_param *target_param)
 {
 	const struct ipt_IFWLOG_info *info = target_param->targinfo;
 
@@ -144,17 +144,17 @@ static unsigned int ipt_IFWLOG_target(st
 	return IPT_CONTINUE;
 }
 
-static bool ipt_IFWLOG_checkentry(const struct xt_tgchk_param *tgchk_param)
+static int ipt_IFWLOG_checkentry(const struct xt_tgchk_param *tgchk_param)
 {
 	const struct ipt_IFWLOG_info *info = tgchk_param->targinfo;
 
 	if (info->prefix[sizeof(info->prefix)-1] != '\0') {
 		DEBUGP("IFWLOG: prefix term %i\n",
 		       info->prefix[sizeof(info->prefix)-1]);
-		return false;
+		return -EINVAL;
 	}
 
-	return true;
+	return 0;
 }
 
 static struct xt_target ipt_IFWLOG = {
