--- ./drivers/platform/x86/shuttle-wmi.c.orig	2013-01-16 18:13:57.000000000 +0200
+++ ./drivers/platform/x86/shuttle-wmi.c	2013-01-16 18:41:13.676433242 +0200
@@ -294,7 +294,7 @@ static struct dmi_system_id shuttle_dmi_
 	{}
 };
 
-static struct dmi_system_id __devinitdata shuttle_quirk_bl_dmi_ids[] = {
+static struct dmi_system_id shuttle_quirk_bl_dmi_ids[] = {
 	{
 		.ident = "Positivo M13",
 		.matches = {
@@ -1105,7 +1105,7 @@ fn_init_err:
 	return rc;
 }
 
-static int __devinit shuttle_wmi_probe(struct platform_device *pdev)
+static int shuttle_wmi_probe(struct platform_device *pdev)
 {
 	struct shuttle_wmi *priv;
 	int rc, i;
@@ -1181,7 +1181,7 @@ err_debugfs:
 	return rc;
 }
 
-static int __devexit shuttle_wmi_remove(struct platform_device *pdev)
+static int shuttle_wmi_remove(struct platform_device *pdev)
 {
 	struct shuttle_wmi *priv = platform_get_drvdata(pdev);
 
@@ -1211,7 +1211,7 @@ static struct platform_driver shuttle_wm
 		.pm = &shuttle_wmi_pm_ops,
 	},
 	.probe = shuttle_wmi_probe,
-	.remove = __devexit_p(shuttle_wmi_remove),
+	.remove = shuttle_wmi_remove,
 };
 
 static struct platform_device *shuttle_wmi_device;
