---
 drivers/scsi/megaraid/megaraid_mbox.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-3.8-rc3-mga0.1/drivers/scsi/megaraid/megaraid_mbox.c.scsi-megaraid-new-sysfs-name.orig	2013-01-10 08:17:11.000000000 +0200
+++ linux-3.8-rc3-mga0.1/drivers/scsi/megaraid/megaraid_mbox.c	2013-01-10 08:30:19.908301909 +0200
@@ -302,7 +302,7 @@ MODULE_DEVICE_TABLE(pci, pci_id_table_g)
 
 
 static struct pci_driver megaraid_pci_driver = {
-	.name		= "megaraid",
+	.name		= "megaraid_mbox",
 	.id_table	= pci_id_table_g,
 	.probe		= megaraid_probe_one,
 	.remove		= megaraid_detach_one,
