Index: fancycode-pylzma-3864d48/setup.py
===================================================================
--- fancycode-pylzma-3864d48.orig/setup.py
+++ fancycode-pylzma-3864d48/setup.py
@@ -22,7 +22,7 @@
 #
 # $Id$
 #
-import sys, os
+import sys, os, platform
 from warnings import warn
 from distutils import log
 from distutils.command.build_ext import build_ext as _build_ext
@@ -123,6 +123,8 @@ macros = []
 lzma_files = ('src/sdk/LzFind.c', 'src/sdk/LzmaDec.c', 'src/sdk/LzmaEnc.c', \
     'src/7zip/C/CpuArch.c', 'src/7zip/C/Aes.c', 'src/7zip/C/AesOpt.c', \
     'src/7zip/C/Sha256.c')
+if platform.uname()[-1] == "i686":
+    compile_args.append("-fno-PIC")
 if ENABLE_COMPATIBILITY:
     c_files += ('src/pylzma/pylzma_decompress_compat.c', 'src/pylzma/pylzma_decompressobj_compat.c', )
     lzma_files += ('src/compat/LzmaCompatDecode.c', )
