--- configure.in
+++ configure.in
@@ -41,7 +41,7 @@
 AC_EGREP_HEADER(PFNGLBINDPROGRAMNVPROC, GL/glext.h,  ,echo "Glaxium won't compile: get a newer version of your openGL header GL/glext.h at <http://oss.sgi.com/projects/ogl-sample/registry/>";exit 12)
 
 dnl Expand DATADIR
-FC_EXPAND_DIR(DATADIR, "$datadir/games/glaxium")
+FC_EXPAND_DIR(DATADIR, "$datadir/glaxium")
 AC_DEFINE_UNQUOTED(DATADIR, "$DATADIR")
 
 AC_OUTPUT(Makefile)
--- Makefile.in
+++ Makefile.in
@@ -1,5 +1,5 @@
 
-CC=@CXX@
+CXX=@CXX@
 INCLUDE=@CFLAGS@ @CPPFLAGS@
 LIB=@LIBS@
 
@@ -15,11 +15,11 @@
 
 GLAXIUM_INSTALLDIR=$(bindir)
 DATA_DIRS=samples textures
-GLAXIUM_HOME=$(datadir)/games/glaxium
+GLAXIUM_HOME=$(datadir)/glaxium
 MAN_DIR=$(mandir)
 
 glaxium : $(OBJ)
-	gcc -g -o glaxium $(OBJ) $(LIB)
+	$(CXX) -o glaxium $(OBJ) $(LIB) $(LDFLAGS)
 
 
 
@@ -31,18 +31,17 @@
 
 
 %.o : %.cpp
-	$(CC) -c  $(INCLUDE) $< -o $@ -DDATADIR=\"$(GLAXIUM_HOME)\"
+	$(CXX) -c  $(INCLUDE) $< -o $@ -DDATADIR=\"$(GLAXIUM_HOME)\"
 
 
 install: glaxium
 	@echo Instaling glaxium in $(GLAXIUM_INSTALLDIR) and its data in $(GLAXIUM_HOME)
-	@if [ ! -d $(GLAXIUM_HOME) ] ; then \
-		mkdir -p $(GLAXIUM_HOME) ; \
-	fi
-	@cp -f glaxium $(GLAXIUM_INSTALLDIR)/
-	@cp -R $(DATA_DIRS) $(GLAXIUM_HOME)
-	@mkdir -p $(MAN_DIR)/man6/
-	@cp -f glaxium.6 $(MAN_DIR)/man6/glaxium.6 
+	mkdir -p "$(DESTDIR)$(GLAXIUM_INSTALLDIR)"
+	@cp -f glaxium "$(DESTDIR)$(GLAXIUM_INSTALLDIR)"/
+	mkdir -p "$(DESTDIR)$(GLAXIUM_HOME)"	
+	@cp -R $(DATA_DIRS) "$(DESTDIR)$(GLAXIUM_HOME)"
+	@mkdir -p "$(DESTDIR)$(MAN_DIR)"/man6/
+	@cp -f glaxium.6 "$(DESTDIR)$(MAN_DIR)"/man6/glaxium.6
 
 uninstall: 
 	@echo Uninstaling glaxium
