diff -Naur WordNet-3.0.orig/include/wn.h WordNet-3.0/include/wn.h
--- WordNet-3.0.orig/include/wn.h	2007-01-03 22:51:35.000000000 +0100
+++ WordNet-3.0/include/wn.h	2007-08-04 22:58:44.000000000 +0200
@@ -29,7 +29,7 @@
 #else
 #define DICTDIR         "/dict"
 #ifndef DEFAULTPATH
-#define DEFAULTPATH	"/usr/local/WordNet-3.0/dict"
+#define DEFAULTPATH	"/usr/share/wordnet/dict"
 #endif
 #define DATAFILE	"%s/data.%s"
 #define INDEXFILE	"%s/index.%s"
diff -Naur WordNet-3.0.orig/configure.ac WordNet-3.0/configure.ac
--- WordNet-3.0.orig/configure.ac	2007-01-20 20:59:24.000000000 +0100
+++ WordNet-3.0/configure.ac	2007-08-04 23:21:09.000000000 +0200
@@ -40,7 +40,7 @@
 fi
 AC_SUBST(ac_prefix)
 AH_TEMPLATE([DEFAULTPATH],[The default search path for WordNet data files])
-AC_DEFINE_UNQUOTED(DEFAULTPATH, ["$ac_prefix/dict"], [Default installation prefix.])
+AC_DEFINE_UNQUOTED(DEFAULTPATH, ["$ac_prefix/share/wordnet/dict"], [Default installation prefix.])
 
 #AC_DEFINE_UNQUOTED(DEFAULTPATH,"${prefix}/dict")
 
diff -Naur WordNet-3.0.orig/dict/Makefile.am WordNet-3.0/dict/Makefile.am
--- WordNet-3.0.orig/dict/Makefile.am	2006-12-18 22:30:07.000000000 +0100
+++ WordNet-3.0/dict/Makefile.am	2007-08-04 23:32:46.000000000 +0200
@@ -1,2 +1,2 @@
-dictdir = $(prefix)/dict
+dictdir = $(prefix)/share/wordnet/dict
 dict_DATA = adj.exc adv.exc cntlist cntlist.rev data.adj data.adv data.noun data.verb frames.vrb index.adj index.adv index.noun index.sense index.verb log.grind.3.0 noun.exc sentidx.vrb sents.vrb verb.Framestext verb.exc lexnames
