diff -Naur WordNet-3.0.orig/configure.ac WordNet-3.0/configure.ac
--- WordNet-3.0.orig/configure.ac	2007-08-04 23:57:09.000000000 +0200
+++ WordNet-3.0/configure.ac	2007-08-04 23:58:16.000000000 +0200
@@ -10,6 +10,7 @@
 AC_PROG_CC
 AC_PROG_RANLIB
 AC_PROG_INSTALL
+AC_PROG_LIBTOOL
 
 # Checks for header files.
 AC_HEADER_STDC
diff -Naur WordNet-3.0.orig/lib/Makefile.am WordNet-3.0/lib/Makefile.am
--- WordNet-3.0.orig/lib/Makefile.am	2007-01-20 21:53:39.000000000 +0100
+++ WordNet-3.0/lib/Makefile.am	2007-08-04 23:59:16.000000000 +0200
@@ -1,5 +1,10 @@
 lib_LIBRARIES = libWN.a
 libWN_a_SOURCES = binsrch.c morph.c search.c wnglobal.c wnhelp.c wnrtl.c wnutil.c
 libWN_a_CPPFLAGS = $(INCLUDES)
+
+lib_LTLIBRARIES = libWN.la
+libWN_la_SOURCES = binsrch.c morph.c search.c wnglobal.c wnhelp.c wnrtl.c wnutil.c
+libWN_la_CPPFLAGS = -fPIC $(INCLUDES)
+
 INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/include
 SUBDIRS = wnres
