--- formats/vformats-xml/Makefile.am	2007-03-27 13:49:27.000000000 +0200
+++ formats/vformats-xml/Makefile.am.oden	2008-07-12 14:07:43.000000000 +0200
@@ -46,5 +46,5 @@
 xmldoc_la_LIBADD = @PACKAGE_LIBS@ @LIBS@ libopensync-xml.la
 
 libopensync_xml_la_SOURCES = xml-support.c
-libopensync_xml_la_LDFLAGS = @GCOV_LDFLAGS@
-libopensync_xml_la_LIBADD = @PACKAGE_LIBS@ @LIBS@
+libopensync_xml_la_LDFLAGS = @GCOV_LDFLAGS@ $(top_builddir)/osengine/libosengine.la $(top_builddir)/opensync/libopensync.la
+libopensync_xml_la_LIBADD = @PACKAGE_LIBS@ @LIBS@ $(XML_LIBS)
