diff -ur libopensync-0.22/configure.in libopensync-0.22.new/configure.in
--- libopensync-0.22/configure.in	2007-03-27 13:49:27.000000000 +0200
+++ libopensync-0.22.new/configure.in	2011-05-26 12:44:29.000000000 +0200
@@ -132,7 +132,7 @@
 	    if test -z "$PYTHON" ; then
 	        AC_MSG_WARN([Python not found])
 	    else
-			AC_PROG_SWIG(1.3.17)
+			AC_PROG_SWIG(2.0.3)
 			SWIG_PYTHON
 			if test x$SWIG_LIB = x; then
 				have_python=no
