--- libopensync-0.20/acinclude.m4~	2006-09-07 17:19:40.000000000 +0200
+++ libopensync-0.20/acinclude.m4	2006-11-12 12:49:05.000000000 +0200
@@ -1129,7 +1129,7 @@
 	# Check for Python library path
 	AC_MSG_CHECKING([for Python library path])
 	python_path=`echo $PYTHON | sed "s,/bin.*$,,"`
-	for i in "$python_path/lib/python$PYTHON_VERSION/config/" "$python_path/lib/python$PYTHON_VERSION/" "$python_path/lib/python/config/" "$python_path/lib/python/" "$python_path/" ; do
+	for i in "$python_path/lib*/python$PYTHON_VERSION/config/" "$python_path/lib*/python$PYTHON_VERSION/" "$python_path/lib*/python/config/" "$python_path/lib*/python/" "$python_path/" ; do
 		python_path=`find $i -type f -name libpython$PYTHON_VERSION.* -print | sed "1q"`
 		if test -n "$python_path" ; then
 			break

