diff -Naur noor-1.1.orig/noor/uihelpers.py noor-1.1/noor/uihelpers.py
--- noor-1.1.orig/noor/uihelpers.py	2008-11-22 08:16:47.000000000 +0200
+++ noor-1.1/noor/uihelpers.py	2008-11-26 17:41:57.000000000 +0200
@@ -183,8 +183,8 @@
 # http://noor.sf.net/

 [common]
-# the translation file
-#translation = /path/to/translation.file
+#Turkish Diyanet Translation. Change the path below for another translation
+translation = /usr/share/noor/trans/diyanet.trans.zip
 # if you want to override noor's quan text
 #quran = /path/to/quran.text

@@ -204,8 +204,8 @@

 [gtk]
 # the font to use
-font = Dejavu Sans Mono size:23 bgcolor:#DDDDFF
-translation_font = Dejavu Sans Mono size:14 fgcolor:#444455
+font = Scheherazade size:24 bgcolor:#DDDDFF
+translation_font = Dejavu Sans size:12 fgcolor:#444455
 # should noor ask which sura to show if no argument is given
 ask_sura_at_start = 1

@@ -232,7 +232,7 @@
 #hide_buttons = 1

 # controls whether to show notes box
-#show_notes = 1
+show_notes = 1
 # the location of note files; ~/.noornotes by default
 #notes_dir = ~/.noornotes

@@ -245,11 +245,11 @@
 #bookmark_file = ~/.noor.bmk

 # uncomment to shadow all but the current aya
-#shadow_ayas = 1
+shadow_ayas = 1

 [html]
 # the font to use
-font = Dejavu Sans Mono 7
+font = Scheherazade 8
 translation_font = Dejavu Sans Mono size:3 fgcolor:444455

 [cgi]
