diff -ubrN volpack-1.0b3/examples.orig/classifyvolume.c volpack-1.0b3/examples/classifyvolume.c
--- volpack-1.0b3/examples.orig/classifyvolume.c	1994-12-31 20:53:04.000000000 +0100
+++ volpack-1.0b3/examples/classifyvolume.c	2007-08-04 21:30:11.000000000 +0200
@@ -28,6 +28,7 @@
  * $Revision: 1.5 $
  */
 
+#include <stdlib.h>
 #include "volume.h"
 
 main(argc, argv)
diff -ubrN volpack-1.0b3/examples.orig/makeoctree.c volpack-1.0b3/examples/makeoctree.c
--- volpack-1.0b3/examples.orig/makeoctree.c	1994-12-31 20:53:05.000000000 +0100
+++ volpack-1.0b3/examples/makeoctree.c	2007-08-04 21:30:11.000000000 +0200
@@ -28,6 +28,7 @@
  * $Revision: 1.5 $
  */
 
+#include <stdlib.h>
 #include "volume.h"
 
 main()
diff -ubrN volpack-1.0b3/examples.orig/makevolume.c volpack-1.0b3/examples/makevolume.c
--- volpack-1.0b3/examples.orig/makevolume.c	1994-12-31 20:53:06.000000000 +0100
+++ volpack-1.0b3/examples/makevolume.c	2007-08-04 21:30:11.000000000 +0200
@@ -29,6 +29,7 @@
  */
 
 #include <sys/types.h>
+#include <stdlib.h>
 #include "volume.h"
 
 //AML main()
diff -ubrN volpack-1.0b3/examples.orig/rendervolume.c volpack-1.0b3/examples/rendervolume.c
--- volpack-1.0b3/examples.orig/rendervolume.c	1995-01-01 01:06:26.000000000 +0100
+++ volpack-1.0b3/examples/rendervolume.c	2007-08-04 21:30:11.000000000 +0200
@@ -29,6 +29,7 @@
  */
 
 #include <string.h>
+#include <stdlib.h>
 #include "volume.h"
 
 main(argc, argv)

