--- fluxbox-1.3.5.orig/util/startfluxbox.in	2013-02-13 12:04:05.903812656 +0200
+++ fluxbox-1.3.5/util/startfluxbox.in	2014-01-03 19:07:39.446722496 +0200
@@ -49,6 +49,11 @@
 # wmnd &
 # wmsmixer -w &
 # idesk &
+exec /usr/libexec/polkit-gnome-authentication-agent-1 &
+exec /usr/bin/volumeicon &
+exec /usr/bin/nm-applet &
+exec /usr/bin/Esetroot -scale /usr/share/fluxbox/backgrounds/pisi-reflect.jpg &
+exec /usr/bin/fbmenugen &
 
 # And last but not least we start fluxbox.
 # Because it is the last app you have to run it with ''exec'' before it.
