diff -Nuar a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2013-04-30 04:27:14.000000000 +0300
+++ b/CMakeLists.txt	2013-10-04 05:58:08.695649713 +0300
@@ -47,8 +47,6 @@
 OPTION(BUILD_MEGAGLEST_TESTS "Build MegaGlest Unit Tests" OFF)
 OPTION(WITH_VLC "Use libVLC to play videos" ON)
 
-FIND_PROGRAM(HELP2MAN "help2man")
-
 set(XVFB_EXEC "")
 find_program(XVFB_RUN NAMES "xvfb-run")
 if (NOT ${XVFB_RUN} MATCHES "XVFB_RUN-NOTFOUND")
@@ -239,10 +237,6 @@
 		IF(NOT MEGAGLEST_ICON_INSTALL_PATH)
                 	SET(MEGAGLEST_ICON_INSTALL_PATH "share/pixmaps/" CACHE STRING "The installation path for icon files (this is appended to the CMAKE_INSTALL_PREFIX)")
 		ENDIF()
-		IF(NOT MEGAGLEST_MANPAGE_INSTALL_PATH)
-                	SET(MEGAGLEST_MANPAGE_INSTALL_PATH "share/man/man6/" CACHE STRING "The installation path for manpage files (this is appended to the CMAKE_INSTALL_PREFIX)")
-		ENDIF()
-
 		IF(NOT CUSTOM_DATA_INSTALL_PATH)
 			SET(CUSTOM_DATA_INSTALL_PATH "'\\\"${CMAKE_INSTALL_PREFIX}/${MEGAGLEST_DATA_INSTALL_PATH}\\\"'" CACHE STRING "The FULL installation path for data files (this is build automatically by combining CMAKE_INSTALL_PREFIX and MEGAGLEST_DATA_INSTALL_PATH)")
 		ENDIF()
diff -Nuar a/source/g3d_viewer/CMakeLists.txt b/source/g3d_viewer/CMakeLists.txt
--- a/source/g3d_viewer/CMakeLists.txt	2013-01-15 22:13:18.000000000 +0200
+++ b/source/g3d_viewer/CMakeLists.txt	2013-10-04 06:15:16.405614131 +0300
@@ -134,9 +134,6 @@
 	        ENDIF()
         ENDIF()
 
-        MESSAGE(STATUS "***Note: The final output path for Megaglest G3d Viewer will be [${EXECUTABLE_OUTPUT_PATH}]")
-
-	FIND_PROGRAM(HELP2MAN "help2man")
 
 	SET_SOURCE_FILES_PROPERTIES(${MG_INCLUDE_FILES} PROPERTIES HEADER_FILE_ONLY 1)
 
@@ -153,38 +150,12 @@
 	ENDIF()
 	TARGET_LINK_LIBRARIES(${TARGET_NAME} ${EXTERNAL_LIBS})
 
-	IF(EXISTS ${HELP2MAN})
-		MESSAGE("***-- Found help2man: ${HELP2MAN} about to build manpage in [${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME_MANPAGE}]")
-
-                SET(MG_MANPAGE_DESCRIPTION "G3D Model editor for the popular 3D realtime stragegy game")
-
-		IF(EXISTS ${XVFB_EXEC})
-			MESSAGE("***-- Found xvfb-run: ${XVFB_EXEC} about to build manpage with it.")
-
-		  	add_custom_target(${TARGET_NAME_MANPAGE} ALL
-		    		COMMAND ${XVFB_EXEC} --auto-servernum --server-num=770 ${HELP2MAN} --name='${MG_MANPAGE_DESCRIPTION}' --section=6 -N -o ${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME_MANPAGE} ${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME}
-		    	DEPENDS ${TARGET_NAME})
-		ELSE()
-		  	add_custom_target(${TARGET_NAME_MANPAGE} ALL
-		    		COMMAND ${HELP2MAN} --name='${MG_MANPAGE_DESCRIPTION}' --section=6 -N -o ${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME_MANPAGE} ${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME}
-		    	DEPENDS ${TARGET_NAME})
-		ENDIF()
-	ENDIF()
-
         # Requires an install prefix for the items below to work
         IF(NOT CMAKE_INSTALL_PREFIX STREQUAL "")
 	        # Installation of the program
 	        INSTALL(TARGETS 
 		        ${TARGET_NAME} 
-		        DESTINATION ${MEGAGLEST_BIN_INSTALL_PATH})
-
-	        IF(EXISTS ${HELP2MAN})
-         		MESSAGE(STATUS "**NOTE: Will Install manpage [${TARGET_NAME_MANPAGE}] to [${MEGAGLEST_MANPAGE_INSTALL_PATH}]")	
-
-		        # Installation of the program manpage file
-		        INSTALL(FILES 
-		          "${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME_MANPAGE}"
-		          DESTINATION ${MEGAGLEST_MANPAGE_INSTALL_PATH})
+		        DESTINATION ${MEGAGLEST_BIN_INSTALL_PATH})	
 	        ENDIF()
 
                 # Installation of the program config and image files
@@ -201,6 +172,4 @@
                 INSTALL(FILES
 	          "${MG_DESKTOP_FILE}"
 	          DESTINATION ${MEGAGLEST_DESKTOP_INSTALL_PATH})
-
-        ENDIF()
 ENDIF()
diff -Nuar a/source/glest_game/CMakeLists.txt b/source/glest_game/CMakeLists.txt
--- a/source/glest_game/CMakeLists.txt	2013-05-01 23:56:19.000000000 +0300
+++ b/source/glest_game/CMakeLists.txt	2013-10-04 06:18:07.092608221 +0300
@@ -256,24 +256,6 @@
 	ENDIF()
 	TARGET_LINK_LIBRARIES(${TARGET_NAME} ${EXTERNAL_LIBS})
 
-	IF(EXISTS ${HELP2MAN})
-		MESSAGE("***-- Found help2man: ${HELP2MAN} about to build manpage in [${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME_MANPAGE}]")
-
-                SET(MG_MANPAGE_DESCRIPTION "3D multi-player real time strategy game")
-
-		IF(EXISTS ${XVFB_EXEC})
-			MESSAGE("***-- Found xvfb-run: ${XVFB_EXEC} about to build manpage with it.")
-
-		  	add_custom_target(${TARGET_NAME_MANPAGE} ALL
-		    		COMMAND ${XVFB_EXEC} --auto-servernum --server-num=770 ${HELP2MAN} --name='${MG_MANPAGE_DESCRIPTION}' --section=6 -N -o ${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME_MANPAGE} ${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME}
-		    	DEPENDS ${TARGET_NAME})
-		ELSE()
-		  	add_custom_target(${TARGET_NAME_MANPAGE} ALL
-		    		COMMAND ${HELP2MAN} --name='${MG_MANPAGE_DESCRIPTION}' --section=6 -N -o ${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME_MANPAGE} ${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME}
-		    	DEPENDS ${TARGET_NAME})
-		ENDIF()
-	ENDIF()
-
         # Requires an install prefix for the items below to work
         IF(NOT CMAKE_INSTALL_PREFIX STREQUAL "")
 	        MESSAGE(STATUS "**Source package [${PROJECT_SOURCE_DIR}]")	
@@ -282,14 +264,6 @@
 	        INSTALL(TARGETS 
 		        ${TARGET_NAME} 
 		        DESTINATION ${MEGAGLEST_BIN_INSTALL_PATH})
-
-	        IF(EXISTS ${HELP2MAN})
-         		MESSAGE(STATUS "**NOTE: Will Install manpage [${TARGET_NAME_MANPAGE}] to [${MEGAGLEST_MANPAGE_INSTALL_PATH}]")	
-
-		        # Installation of the program manpage file
-		        INSTALL(FILES 
-		          "${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME_MANPAGE}"
-		          DESTINATION ${MEGAGLEST_MANPAGE_INSTALL_PATH})
 	        ENDIF()
 
 	        # Check if we are running from an SVN folder structure or tarballs
@@ -327,5 +301,4 @@
 		          "${PROJECT_SOURCE_DIR}/g3dviewer.ico"
 		          DESTINATION ${MEGAGLEST_ICON_INSTALL_PATH})
 	        ENDIF()
-        ENDIF()
 ENDIF()
diff -Nuar a/source/glest_map_editor/CMakeLists.txt b/source/glest_map_editor/CMakeLists.txt
--- a/source/glest_map_editor/CMakeLists.txt	2013-01-15 22:13:18.000000000 +0200
+++ b/source/glest_map_editor/CMakeLists.txt	2013-10-04 07:35:13.445991917 +0300
@@ -148,25 +148,6 @@
 	ENDIF()
 	TARGET_LINK_LIBRARIES(${TARGET_NAME} ${EXTERNAL_LIBS})
 
-	IF(EXISTS ${HELP2MAN})
-		MESSAGE("***-- Found help2man: ${HELP2MAN} about to build manpage in [${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME_MANPAGE}]")
-
-                SET(MG_MANPAGE_DESCRIPTION "Map editor for the popular 3D realtime stragegy game")
-
-		IF(EXISTS ${XVFB_EXEC})
-			MESSAGE("***-- Found xvfb-run: ${XVFB_EXEC} about to build manpage with it.")
-
-		  	add_custom_target(${TARGET_NAME_MANPAGE} ALL
-		    		COMMAND ${XVFB_EXEC} --auto-servernum --server-num=770 ${HELP2MAN} --name='${MG_MANPAGE_DESCRIPTION}' --section=6 -N -o ${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME_MANPAGE} ${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME}
-		    	DEPENDS ${TARGET_NAME})
-		ELSE()
-		  	add_custom_target(${TARGET_NAME_MANPAGE} ALL
-		    		COMMAND ${HELP2MAN} --name='${MG_MANPAGE_DESCRIPTION}' --section=6 -N -o ${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME_MANPAGE} ${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME}
-		    	DEPENDS ${TARGET_NAME})
-		ENDIF()
-	ENDIF()
-
-
         # Requires an install prefix for the items below to work
         IF(NOT CMAKE_INSTALL_PREFIX STREQUAL "")
 	        # Installation of the program
@@ -174,15 +155,6 @@
 		        ${TARGET_NAME} 
 		        DESTINATION ${MEGAGLEST_BIN_INSTALL_PATH})
 
-	        IF(EXISTS ${HELP2MAN})
-         		MESSAGE(STATUS "**NOTE: Will Install manpage [${TARGET_NAME_MANPAGE}] to [${MEGAGLEST_MANPAGE_INSTALL_PATH}]")	
-
-		        # Installation of the program manpage file
-		        INSTALL(FILES 
-		          "${EXECUTABLE_OUTPUT_PATH}/${TARGET_NAME_MANPAGE}"
-		          DESTINATION ${MEGAGLEST_MANPAGE_INSTALL_PATH})
-	        ENDIF()
-
 	        # Installation of the program config and image files
 	        INSTALL(FILES 
                   "${PROJECT_SOURCE_DIR}/editor.ico"
