From: Julian Ospald <hasufell@gentoo.org>
Date: Fri Aug 31 23:45:34 UTC 2012
Subject: fix as-needed

reorder linker line (curl after libircclient)

diff -Nuar a/source/shared_lib/CMakeLists.txt b/source/shared_lib/CMakeLists.txt
--- a/source/shared_lib/CMakeLists.txt	2013-06-26 19:58:34.000000000 +0300
+++ b/source/shared_lib/CMakeLists.txt	2013-10-04 07:52:47.131955435 +0300
@@ -50,9 +50,7 @@
 
 	FIND_PACKAGE(CURL REQUIRED)
 	INCLUDE_DIRECTORIES(${CURL_INCLUDE_DIRS})
-	IF(UNIX)
-		SET(EXTERNAL_LIBS ${EXTERNAL_LIBS} ${CURL_LIBRARIES})
-	ENDIF()
+	
 	# check libcurl version #
 	IF(WIN32)
 		file (READ ${PROJECT_SOURCE_DIR}${CURL_INCLUDE_DIR}/curl/curlver.h CURL_VERSION_H_CONTENTS)
@@ -380,7 +378,11 @@
 		        INCLUDE_DIRECTORIES(${IRCCLIENT_INCLUDE_DIR})
 		        SET(EXTERNAL_LIBS ${EXTERNAL_LIBS} ${IRCCLIENT_LIBRARY})
 	        endif()
-
+	        
+	IF(UNIX)
+		SET(EXTERNAL_LIBS ${EXTERNAL_LIBS} ${CURL_LIBRARIES})
+	ENDIF()
+	
                 set(CMAKE_REQUIRED_INCLUDES "${IRCCLIENT_INCLUDE_DIR}")
                 set(CMAKE_REQUIRED_LIBRARIES "${IRCCLIENT_LIBRARY}")
                 include(CheckCXXSourceRuns)
