diff -Naur aubio-0.3.2/python/aubio/aubio_wrap.c aubio-0.3.2.tpg/python/aubio/aubio_wrap.c
--- aubio-0.3.2/python/aubio/aubio_wrap.c	2006-10-10 14:50:23.000000000 +0000
+++ aubio-0.3.2.tpg/python/aubio/aubio_wrap.c	2008-12-28 21:31:48.000000000 +0000
@@ -844,7 +844,7 @@
     Py_DECREF(old_str);
     Py_DECREF(value);
   } else {
-    PyErr_Format(PyExc_RuntimeError, mesg);
+    PyErr_Format(PyExc_RuntimeError, "%s", mesg);
   }
 }
 
