--- configure    2012-02-08 22:01:55.000000000 +0100
+++ configure.oden    2012-02-13 13:37:22.388759837 +0100
@@ -6826,7 +6826,7 @@
             # -Os is broken on gcc 4.1.x 4.2.x, 4.5.0 we need to tweak it to get good results.
             MOZ_OPTIMIZE_SIZE_TWEAK="-finline-limit=50"
         esac
-        MOZ_OPTIMIZE_FLAGS="-Os -freorder-blocks $MOZ_OPTIMIZE_SIZE_TWEAK"
+        MOZ_OPTIMIZE_FLAGS="-O2 -freorder-blocks $MOZ_OPTIMIZE_SIZE_TWEAK"
         MOZ_DEBUG_FLAGS="-g"
     fi

--- configure.in    2012-02-08 21:51:26.000000000 +0100
+++ configure.in.oden    2012-02-13 13:37:38.973760765 +0100
@@ -2068,7 +2068,7 @@
             # -Os is broken on gcc 4.1.x 4.2.x, 4.5.0 we need to tweak it to get good results.
             MOZ_OPTIMIZE_SIZE_TWEAK="-finline-limit=50"
         esac
-        MOZ_OPTIMIZE_FLAGS="-Os -freorder-blocks $MOZ_OPTIMIZE_SIZE_TWEAK"
+        MOZ_OPTIMIZE_FLAGS="-O2 -freorder-blocks $MOZ_OPTIMIZE_SIZE_TWEAK"
         MOZ_DEBUG_FLAGS="-g"
     fi
