diff -Nuar a/build/data.mk b/build/data.mk
--- a/build/data.mk	2012-04-24 08:04:43.000000000 +0300
+++ b/build/data.mk	2013-10-01 08:34:53.241452400 +0300
@@ -3,10 +3,10 @@
 BASE_DIR = base
 PAK_FILES_OUT = $(addprefix $(BASE_DIR)/,$(PAK_FILES))
 
-pk3: $(PAK_FILES_OUT)
+pk3:
 
 clean-pk3:
-	$(Q)rm -f $(PAK_FILES_OUT)
+#	$(Q)rm -f $(PAK_FILES_OUT)
 
 define FIND
 $(shell find $(BASE_DIR)/$(1) -type f -print)
