Index: wxcam-1.0.7/wxcam.desktop
===================================================================
--- wxcam-1.0.7.orig/wxcam.desktop
+++ wxcam-1.0.7/wxcam.desktop
@@ -1,6 +1,11 @@
 [Desktop Entry]
+Encoding=UTF-8
 Name=Webcam Application
+Name[tr]=Webcam Uygulaması
 Comment=wxCam webcam application
+Comment[tr]=wxCam Webcam uygulaması
+GenericName=Webcam Viewer
+GenericName[tr]=Webcam Görüntüleyicisi
 Exec=wxcam
 Icon=wxcam.png
 Terminal=false
