Index: gocr-0.48/src/Makefile.in
===================================================================
--- gocr-0.48.orig/src/Makefile.in
+++ gocr-0.48/src/Makefile.in
@@ -82,7 +82,7 @@ gocr.o: gocr.h Makefile ../include/versi
 $(PROGRAM): $(LIBOBJS) gocr.o
 	# make it conform to ld --as-needed
 	#$(CC) -o $@ $(LDFLAGS) gocr.o ./lib$(PGMASCLIB).a $(LIBS)
-	$(CC) -o $@ $(LDFLAGS) gocr.o $(LIBOBJS) $(LIBS)
+	$(CC) $(LDFLAGS) gocr.o $(LIBOBJS) $(LIBS) -o $@
 	if test -r $(PROGRAM); then cp $@ ../bin; fi
 
 libs: lib$(PGMASCLIB).a lib$(PGMASCLIB).@PACKAGE_VERSION@.so
