--- tcl8.5.10.orig/generic/tclThreadStorage.c	2011-08-01 10:06:33.932000004 +0300
+++ tcl8.5.10/generic/tclThreadStorage.c	2011-08-01 10:11:35.417000004 +0300
@@ -217,6 +217,7 @@
     hashTablePtr = threadStorageCache[index].hashTablePtr;
     if (threadStorageCache[index].id != id) {
 	Tcl_MutexLock(&threadStorageLock);
+    TclInitThreadStorage();
 #else
     Tcl_MutexLock(&threadStorageLock);
     hashTablePtr = threadStorageCache[index].hashTablePtr;
