Index: quilt-0.48/quilt.quiltrc
===================================================================
--- quilt-0.48.orig/quilt.quiltrc
+++ quilt-0.48/quilt.quiltrc
@@ -6,7 +6,7 @@ QUILT_DIFF_OPTS="--show-c-function"
 # Options passed to GNU patch when applying patches.
 # (For appling patches which are not in unified format, remove the "--unified".)
 #QUILT_PATCH_OPTS="--ignore-whitespace"
-QUILT_PATCH_OPTS="--unified"
+#QUILT_PATCH_OPTS="--unified"
 
 # Options passed to diffstat when generating patch statistics
 #QUILT_DIFFSTAT_OPTS="-f0"
