Index: quilt-0.48/Makefile.in
===================================================================
--- quilt-0.48.orig/Makefile.in
+++ quilt-0.48/Makefile.in
@@ -314,7 +314,7 @@ install-main :: scripts
 	$(INSTALL) -m 755 $(patsubst %,quilt/scripts/%,			\
 			  $(filter-out patchfns,$(SCRIPTS)))		\
 		   $(BUILD_ROOT)$(datadir)/$(PACKAGE)/scripts
-	$(INSTALL) -m 644 quilt/scripts/patchfns			\
+	$(INSTALL) -m 755 quilt/scripts/patchfns			\
 		   $(BUILD_ROOT)$(datadir)/$(PACKAGE)/scripts
 ifneq ($(PATCH_WRAPPER),)
 	$(INSTALL) -d $(BUILD_ROOT)$(datadir)/$(PACKAGE)/wrapper
