Index: configure.ac
===================================================================
--- configure.ac.orig
+++ configure.ac
@@ -16,6 +16,7 @@
 AC_CANONICAL_BUILD
 AC_CANONICAL_HOST
 AC_CANONICAL_TARGET
+AC_GNU_SOURCE
 
 MEMCACHE_MAJOR_VERSION=1
 MEMCACHE_MINOR_VERSION=4
Index: src/memcache.c
===================================================================
--- src/memcache.c.orig
+++ src/memcache.c
@@ -39,13 +39,6 @@
 #include <sysexits.h>
 #include <errno.h>
 #include <sys/types.h>
-#ifdef __linux
-# ifndef __USE_POSIX
-#  define __USE_POSIX
-#warning "Working around busted-ass Linux header include problems: use FreeBSD instead"
-#warning "http://www.FreeBSD.org/ - you won't regret it"
-# endif
-#endif
 #include <sys/time.h>
 #include <sys/socket.h>
 #include <sys/uio.h>
