Index: Stage-3.2.2-Source/libstage/CMakeLists.txt
===================================================================
--- Stage-3.2.2-Source.orig/libstage/CMakeLists.txt
+++ Stage-3.2.2-Source/libstage/CMakeLists.txt
@@ -51,7 +51,7 @@ add_library(stage SHARED ${stageSrcs})
 target_link_libraries( stage
 		       ${OPENGL_LIBRARIES}
   				 ${LTDL_LIB}
-  				 ${FLTK_LIBS2}
+                 ${FLTK_LDFLAGS}
 )
 
 # causes the shared library to have a version number
