diff -up ./CMakeLists.txt.fixfltk ./CMakeLists.txt
--- ./CMakeLists.txt.fixfltk	2012-01-16 21:49:52.535208280 -0500
+++ ./CMakeLists.txt	2012-01-16 21:50:15.975190960 -0500
@@ -101,7 +101,8 @@ IF( PNG_FOUND )
 ELSE( PNG_FOUND )
    MESSAGE( ${INDENT} "PNG library not detected" )
 ENDIF( PNG_FOUND )
-   
+ 
+set (FLTK_SKIP_FLUID TRUE) 
 find_package( FLTK REQUIRED )
 IF( FLTK_FOUND )
 #  MESSAGE( STATUS ${INDENT} "FLTK_LIBRARIES=${FLTK_LIBRARIES}" )
