diff -up Stage-4.0.0-src/cmake/internal/FindOS.cmake.s390x-typos Stage-4.0.0-src/cmake/internal/FindOS.cmake
--- Stage-4.0.0-src/cmake/internal/FindOS.cmake.s390x-typos	2011-05-04 12:38:48.000000000 +0200
+++ Stage-4.0.0-src/cmake/internal/FindOS.cmake	2011-05-04 12:38:57.000000000 +0200
@@ -53,8 +53,8 @@ IF (PROJECT_OS_LINUX)
     ELSEIF (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "ppc64")
         MESSAGE (STATUS "Linux ppc64 Detected")
         SET (PROJECT_PROC_64BIT TRUE BOOL INTERNAL)
-    ELSEIF ($CMAKE_SYSTEM_PROCESSOR} STREQUAL "s390x")
-	MESSAGE (STATEUS "Linux s390x Detected")
+    ELSEIF (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "s390x")
+	MESSAGE (STATUS "Linux s390x Detected")
 	SET (PROJECT_PROC_64BIT TRUE BOOL INTERNAL)
     ENDIF (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86_64")
 ENDIF (PROJECT_OS_LINUX)
