diff -up ./libstageplugin/CMakeLists.txt.plugininstall ./libstageplugin/CMakeLists.txt
--- ./libstageplugin/CMakeLists.txt.plugininstall	2010-12-04 16:08:03.231970996 -0500
+++ ./libstageplugin/CMakeLists.txt	2010-12-04 16:08:14.270971005 -0500
@@ -39,6 +39,6 @@ IF (BUILD_LSPTEST)
 ENDIF (BUILD_LSPTEST)
 
 INSTALL(TARGETS stageplugin
-	LIBRARY DESTINATION ${PROJECT_LIB_DIR}
+	LIBRARY DESTINATION ${PROJECT_LIB_DIR}/player
 )
 
