diff -Nur trackballs-1.1.4-old/po/Makefile.in.in trackballs-1.1.4/po/Makefile.in.in
--- trackballs-1.1.4-old/po/Makefile.in.in	2007-05-26 17:34:47.425466944 +0300
+++ trackballs-1.1.4/po/Makefile.in.in	2007-05-26 17:35:42.727059821 +0300
@@ -22,7 +22,7 @@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
 #localedir = $(datadir)/${PACKAGE}/locale
-localedir = $(datadir)/locale
+localedir = /usr/share/locale
 gettextsrcdir = $(datadir)/gettext/po
 
 INSTALL = @INSTALL@
diff -Nur trackballs-1.1.4-old/share/Makefile.in trackballs-1.1.4/share/Makefile.in
--- trackballs-1.1.4-old/share/Makefile.in	2007-05-26 17:34:47.421467551 +0300
+++ trackballs-1.1.4/share/Makefile.in	2007-05-26 17:36:50.436766383 +0300
@@ -196,7 +196,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 with_highscores = @with_highscores@
-SUBDIRS = fonts levels sfx images icons music
+SUBDIRS = fonts levels sfx images music
 pkgdata_DATA = highScores
 man_MANS = trackballs.6
 EXTRA_DIST = $(pkgdata_DATA) trackballs.6
diff -Nur trackballs-1.1.4-old/src/Makefile.in trackballs-1.1.4/src/Makefile.in
--- trackballs-1.1.4-old/src/Makefile.in	2007-05-26 17:34:47.595441103 +0300
+++ trackballs-1.1.4/src/Makefile.in	2007-05-26 17:37:12.716379367 +0300
@@ -206,7 +206,7 @@
 install_sh = @install_sh@
 libdir = @libdir@
 libexecdir = @libexecdir@
-localedir = $(datadir)/locale
+localedir = /usr/share/locale
 localstatedir = @localstatedir@
 mandir = @mandir@
 mkdir_p = @mkdir_p@
