diff -up trackballs-1.1.4/src/enterHighScoreMode.cc~ trackballs-1.1.4/src/enterHighScoreMode.cc
--- trackballs-1.1.4/src/enterHighScoreMode.cc~ 2007-04-07 15:17:17.000000000 +0200
+++ trackballs-1.1.4/src/enterHighScoreMode.cc  2011-01-25 22:46:23.591215695 +0100
@@ -112,7 +112,7 @@ void EnterHighScoreMode::activated() {
    if(!background) { printf("Error: failed to load %s\n",str); exit(0); }  
   }
 
-  snprintf(name,sizeof(str),"%s",Game::current->gamer->name);
+  snprintf(name, sizeof(name), "%s", Game::current->gamer->name);
   if(!HighScore::highScore->isHighScore(Game::current->player1->score)) 
    GameMode::activate(MenuMode::menuMode);
 
