Index: x11-ssh-askpass-1.2.4.1/Imakefile.in
===================================================================
--- x11-ssh-askpass-1.2.4.1.orig/Imakefile.in
+++ x11-ssh-askpass-1.2.4.1/Imakefile.in
@@ -150,7 +150,6 @@ install::
 	  $(MV) $(DESTDIR)$(BINDIR)/ssh-askpass \@@\
 	    $(DESTDIR)$(BINDIR)/ssh-askpass.old; \@@\
 	fi
-	$(LN) $(NAME) $(DESTDIR)$(BINDIR)/ssh-askpass
 
 install.man::
 	@set -x; \@@\
@@ -158,7 +157,6 @@ install.man::
 	  $(MV) $(DESTDIR)$(MANDIR)/ssh-askpass.$(MANSUFFIX) \@@\
 	    $(DESTDIR)$(MANDIR)/ssh-askpass.$(MANSUFFIX).old; \@@\
 	fi
-	$(LN) $(NAME).$(MANSUFFIX) $(DESTDIR)$(MANDIR)/ssh-askpass.$(MANSUFFIX)
 
 .PHONY: depends changelog realclean squeakyclean spotless tar dist
 
