Description: Midori default settings for Pisi Linux
Author: Erdinç Gültekin <erdincgultekin@pisilinux.org>

diff -Nurp a/config/config b/config/config
--- a/config/config	1970-01-01 02:00:00.000000000 +0200
+++ b/config/config	2013-08-25 21:30:03.027828509 +0300
@@ -0,0 +1,8 @@
+[settings]
+show-menubar=true
+toolbar-items=Homepage,Separator,ZoomOut,ZoomIn,TabNew,Back,NextForward,ReloadStop,BookmarkAdd,Location,Search,Trash,Print,
+homepage=http://www.pisilinux.org/
+tabhome=about:dial
+
+[extensions]
+libtoolbar-editor.so=true
