diff -p -up ./midori/midori-settings.vala.orig ./midori/midori-settings.vala
--- ./midori/midori-settings.vala.orig	2012-09-20 13:21:28.635672165 +0000
+++ ./midori/midori-settings.vala	2012-09-20 13:21:44.468015591 +0000
@@ -73,7 +73,7 @@ namespace Midori {
         public bool open_tabs_next_to_current { get; set; default = true; }
         public bool always_show_tabbar { get; set; default = true; }
 
-        public string homepage { get; set; default = "http://www.google.com"; }
+        public string homepage { get; set; default = "http://www.pisilinux.org"; }
         static string default_download_folder () {
             return Environment.get_user_special_dir (UserDirectory.DOWNLOAD)
                 ?? Environment.get_home_dir ();
