diff -up qt-everywhere-opensource-src-4.7.0-beta2/src/corelib/kernel/qcoreapplication.cpp.kde4_plugins qt-everywhere-opensource-src-4.7.0-beta2/src/corelib/kernel/qcoreapplication.cpp
--- qt-everywhere-opensource-src-4.7.0-beta2/src/corelib/kernel/qcoreapplication.cpp.kde4_plugins	2010-06-29 20:53:15.000000000 -0500
+++ qt-everywhere-opensource-src-4.7.0-beta2/src/corelib/kernel/qcoreapplication.cpp	2010-07-08 08:24:12.714604479 -0500
@@ -2262,6 +2262,15 @@ QStringList QCoreApplication::libraryPat
             if (!app_libpaths->contains(installPathPlugins))
                 app_libpaths->append(installPathPlugins);
         }
+
+        // hack in support for kde4 plugin paths -- Rex
+        QString kde4PathPlugins = QLibraryInfo::location(QLibraryInfo::LibrariesPath) + QLatin1String("/kde4/plugins");
+        if (QFile::exists(kde4PathPlugins)) {
+            // Make sure we convert from backslashes to slashes.
+            // kde4PathPlugins = QDir(kde4PathPlugins).canonicalPath();
+            if (!app_libpaths->contains(kde4PathPlugins))
+                app_libpaths->append(kde4PathPlugins);
+        }
 #endif
 
         // If QCoreApplication is not yet instantiated,
