diff -up qt-everywhere-opensource-src-4.8.0/src/corelib/tools/qlocale_icu.cpp.icu_no_spam qt-everywhere-opensource-src-4.8.0/src/corelib/tools/qlocale_icu.cpp
--- qt-everywhere-opensource-src-4.8.0/src/corelib/tools/qlocale_icu.cpp.icu_no_spam	2011-12-07 23:06:03.000000000 -0600
+++ qt-everywhere-opensource-src-4.8.0/src/corelib/tools/qlocale_icu.cpp	2011-12-20 10:20:19.659502966 -0600
@@ -83,7 +83,7 @@ bool qt_initIcu(const QString &localeStr
         // resolve libicui18n
         QLibrary lib(QLatin1String("icui18n"), QLatin1String(U_ICU_VERSION_SHORT));
         if (!lib.load()) {
-            qWarning() << "Unable to load library icui18n" << lib.errorString();
+        //  qWarning() << "Unable to load library icui18n" << lib.errorString();
             status = ErrorLoading;
             return false;
         }
@@ -112,7 +112,7 @@ bool qt_initIcu(const QString &localeStr
         // resolve libicuuc
         QLibrary ucLib(QLatin1String("icuuc"), QLatin1String(U_ICU_VERSION_SHORT));
         if (!ucLib.load()) {
-            qWarning() << "Unable to load library icuuc" << ucLib.errorString();
+        //  qWarning() << "Unable to load library icuuc" << ucLib.errorString();
             status = ErrorLoading;
             return false;
         }
