diff -up qt-everywhere-opensource-src-4.8.0/tools/qvfb/qvfb.pro.qvfb qt-everywhere-opensource-src-4.8.0/tools/qvfb/qvfb.pro
--- qt-everywhere-opensource-src-4.8.0/tools/qvfb/qvfb.pro.qvfb	2011-12-27 21:02:19.765512041 -0600
+++ qt-everywhere-opensource-src-4.8.0/tools/qvfb/qvfb.pro	2011-12-27 21:02:39.271268187 -0600
@@ -38,8 +38,6 @@ SOURCES         = qvfb.cpp \
 
 !embedded {
     DEFINES         += QT_NO_QWS_SIGNALHANDLER
-    HEADERS         += ../../src/gui/embedded/qlock_p.h
-    SOURCES         += ../../src/gui/embedded/qlock.cpp
 }
 
 include(../shared/deviceskin/deviceskin.pri)
