diff -up qt-everywhere-opensource-src-4.7.1/src/3rdparty/webkit/JavaScriptCore/JavaScriptCore.pro.webkit_debug_2 qt-everywhere-opensource-src-4.7.1/src/3rdparty/webkit/JavaScriptCore/JavaScriptCore.pro
--- qt-everywhere-opensource-src-4.7.1/src/3rdparty/webkit/JavaScriptCore/JavaScriptCore.pro.webkit_debug_2	2010-11-05 20:55:20.000000000 -0500
+++ qt-everywhere-opensource-src-4.7.1/src/3rdparty/webkit/JavaScriptCore/JavaScriptCore.pro	2011-02-23 14:34:32.891416772 -0600
@@ -36,7 +36,7 @@ CONFIG(QTDIR_build) {
 contains(QT_CONFIG, reduce_exports):CONFIG += hide_symbols
 unix:contains(QT_CONFIG, reduce_relocations):CONFIG += bsymbolic_functions
 
-CONFIG(QTDIR_build) {
+!CONFIG(webkit-debug):CONFIG(QTDIR_build) {
     # Remove the following 2 lines if you want debug information in JavaScriptCore
     CONFIG -= separate_debug_info
     CONFIG += no_debug_info
