diff -up qt-everywhere-opensource-src-4.7.0/src/gui/image/qpixmap_x11.cpp.me qt-everywhere-opensource-src-4.7.0/src/gui/image/qpixmap_x11.cpp
--- qt-everywhere-opensource-src-4.7.0/src/gui/image/qpixmap_x11.cpp.me	2010-10-15 22:04:09.000000000 +0200
+++ qt-everywhere-opensource-src-4.7.0/src/gui/image/qpixmap_x11.cpp	2010-10-15 22:04:18.000000000 +0200
@@ -900,9 +900,9 @@ void QX11PixmapData::fromImage(const QIm
             case BPP24_888:                        // 24 bit MSB
                 CYCLE(
                     for (int x=0; x<w; x++) {
-                        *dst++ = qRed  (*p);
+                        *dst++ = qBlue  (*p);
                         *dst++ = qGreen(*p);
-                        *dst++ = qBlue (*p++);
+                        *dst++ = qRed (*p++);
                     }
                     )
                     break;
