diff -up qt-everywhere-opensource-src-4.8.0/src/tools/uic3/embed.cpp.uic_multilib qt-everywhere-opensource-src-4.8.0/src/tools/uic3/embed.cpp
--- qt-everywhere-opensource-src-4.8.0/src/tools/uic3/embed.cpp.uic_multilib	2011-12-08 06:06:03.000000000 +0100
+++ qt-everywhere-opensource-src-4.8.0/src/tools/uic3/embed.cpp	2012-02-20 15:17:27.343005746 +0100
@@ -152,8 +152,12 @@ void Ui3Reader::embed(const char *projec
     for ( it = images.begin(); it != images.end(); ++it )
         out << "**      " << *it << "\n";
     out << "**\n";
+#ifdef TIMESTAMP
     out << "** Created: " << QDateTime::currentDateTime().toString() << "\n";
     out << "**      by: The User Interface Compiler for Qt version " << QT_VERSION_STR << "\n";
+#else
+    out << "** Created by: " << QT_VERSION_STR << "\n";
+#endif
     out << "**\n";
     out << "** WARNING! All changes made in this file will be lost!\n";
     out << "****************************************************************************/\n";
diff -up qt-everywhere-opensource-src-4.8.0/src/tools/uic3/uic.cpp.uic_multilib qt-everywhere-opensource-src-4.8.0/src/tools/uic3/uic.cpp
--- qt-everywhere-opensource-src-4.8.0/src/tools/uic3/uic.cpp.uic_multilib	2011-12-08 06:06:03.000000000 +0100
+++ qt-everywhere-opensource-src-4.8.0/src/tools/uic3/uic.cpp	2012-02-20 15:20:13.584706037 +0100
@@ -146,8 +146,12 @@ void Uic::writeCopyrightHeader(DomUI *ui
         out << "/********************************************************************************\n";
         out << "** Form generated from reading UI file '" << QFileInfo(opt.inputFile).fileName() << "'\n";
         out << "**\n";
+#ifdef TIMESTAMP
         out << "** Created: " << QDateTime::currentDateTime().toString() << "\n";
         out << "**      " << QString::fromLatin1("by: Qt User Interface Compiler version %1\n").arg(QLatin1String(QT_VERSION_STR));
+#else
+        out << "** Created by: " << QString::fromLatin1("Qt User Interface Compiler version %1\n").arg(QLatin1String(QT_VERSION_STR));
+#endif
         out << "**\n";
         out << "** WARNING! All changes made in this file will be lost when recompiling UI file!\n";
         out << "********************************************************************************/\n\n";
diff -up qt-everywhere-opensource-src-4.8.0/src/tools/uic/uic.cpp.uic_multilib qt-everywhere-opensource-src-4.8.0/src/tools/uic/uic.cpp
--- qt-everywhere-opensource-src-4.8.0/src/tools/uic/uic.cpp.uic_multilib	2011-12-08 06:06:03.000000000 +0100
+++ qt-everywhere-opensource-src-4.8.0/src/tools/uic/uic.cpp	2012-02-20 15:13:44.175462520 +0100
@@ -139,8 +139,12 @@ void Uic::writeCopyrightHeader(DomUI *ui
         out << "/********************************************************************************\n";
         out << "** Form generated from reading UI file '" << QFileInfo(opt.inputFile).fileName() << "'\n";
         out << "**\n";
+#ifdef TIMESTAMP
         out << "** Created: " << QDateTime::currentDateTime().toString() << "\n";
         out << "**      " << QString::fromLatin1("by: Qt User Interface Compiler version %1\n").arg(QLatin1String(QT_VERSION_STR));
+#else
+        out << "** Created by: " << QString::fromLatin1("Qt User Interface Compiler version %1\n").arg(QLatin1String(QT_VERSION_STR));
+#endif
         out << "**\n";
         out << "** WARNING! All changes made in this file will be lost when recompiling UI file!\n";
         out << "********************************************************************************/\n\n";
diff -up qt-everywhere-opensource-src-4.8.0/src/tools/moc/moc.cpp.me qt-everywhere-opensource-src-4.8.0/src/tools/moc/moc.cpp
--- qt-everywhere-opensource-src-4.8.0/src/tools/moc/moc.cpp.me	2012-02-20 13:22:38.572628193 +0100
+++ qt-everywhere-opensource-src-4.8.0/src/tools/moc/moc.cpp	2012-02-20 14:46:09.981860726 +0100
@@ -767,8 +767,10 @@ void Moc::parse()
 void Moc::generate(FILE *out)
 {
 
+#ifdef TIMESTAMP
     QDateTime dt = QDateTime::currentDateTime();
     QByteArray dstr = dt.toString().toLatin1();
+#endif
     QByteArray fn = filename;
     int i = filename.length()-1;
     while (i>0 && filename[i-1] != '/' && filename[i-1] != '\\')
@@ -777,8 +779,12 @@ void Moc::generate(FILE *out)
         fn = filename.mid(i);
     fprintf(out, "/****************************************************************************\n"
             "** Meta object code from reading C++ file '%s'\n**\n" , (const char*)fn);
+#ifdef TIMESTAMP
     fprintf(out, "** Created: %s\n"
             "**      by: The Qt Meta Object Compiler version %d (Qt %s)\n**\n" , dstr.data(), mocOutputRevision, QT_VERSION_STR);
+#else
+    fprintf(out, "** Created by: The Qt Meta Object Compiler version %d (Qt %s)\n**\n" , mocOutputRevision, QT_VERSION_STR);
+#endif
     fprintf(out, "** WARNING! All changes made in this file will be lost!\n"
             "*****************************************************************************/\n\n");
 
