diff -up qt-everywhere-opensource-src-4.7.4/src/sql/drivers/tds/qsql_tds.pri.tds_no_strict_aliasing qt-everywhere-opensource-src-4.7.4/src/sql/drivers/tds/qsql_tds.pri
--- qt-everywhere-opensource-src-4.7.4/src/sql/drivers/tds/qsql_tds.pri.tds_no_strict_aliasing	2011-09-12 01:49:28.000000000 -0500
+++ qt-everywhere-opensource-src-4.7.4/src/sql/drivers/tds/qsql_tds.pri	2011-11-03 13:04:34.271086323 -0500
@@ -1,6 +1,8 @@
 HEADERS += $$PWD/qsql_tds.h
 SOURCES += $$PWD/qsql_tds.cpp
 
+*-g++*: QMAKE_CXXFLAGS += -fno-strict-aliasing
+
 unix|win32-g++*: {
     LIBS *= $$QT_LFLAGS_TDS
     !contains(LIBS, .*sybdb.*):LIBS += -lsybdb
