diff -up qt-everywhere-opensource-src-4.8.0/src/gui/painting/qpdf.cpp.QTBUG-14724 qt-everywhere-opensource-src-4.8.0/src/gui/painting/qpdf.cpp
--- qt-everywhere-opensource-src-4.8.0/src/gui/painting/qpdf.cpp.QTBUG-14724	2011-10-03 22:44:31.000000000 -0500
+++ qt-everywhere-opensource-src-4.8.0/src/gui/painting/qpdf.cpp	2011-10-17 10:55:49.713265396 -0500
@@ -1686,6 +1686,7 @@ bool QPdfBaseEnginePrivate::openPrintDev
         cupsTempFile = ret.second;
         outDevice = new QFile();
         static_cast<QFile *>(outDevice)->open(ret.first, QIODevice::WriteOnly);
+        fd = ret.first;
 #endif
 #ifndef QT_NO_LPR
     } else {
