diff -ur qt-everywhere-opensource-src-4.8.0/src/tools/moc/main.cpp qt-everywhere-opensource-src-4.8.0-moc-boost148/src/tools/moc/main.cpp
--- qt-everywhere-opensource-src-4.8.0/src/tools/moc/main.cpp	2011-10-04 05:44:32.000000000 +0200
+++ qt-everywhere-opensource-src-4.8.0-moc-boost148/src/tools/moc/main.cpp	2011-12-04 00:11:25.000000000 +0100
@@ -187,6 +187,7 @@
     Moc moc;
     pp.macros["Q_MOC_RUN"];
     pp.macros["__cplusplus"];
+    pp.macros["BOOST_TT_HAS_OPERATOR_HPP_INCLUDED"]; // rh#756395
     QByteArray filename;
     QByteArray output;
     FILE *in = 0;
