diff -up qt-4.8/src/corelib/plugin/qlibrary.cpp.crash_ppc qt-4.8/src/corelib/plugin/qlibrary.cpp
--- qt-4.8/src/corelib/plugin/qlibrary.cpp.crash_ppc	2011-10-04 05:44:32.000000000 +0200
+++ qt-4.8/src/corelib/plugin/qlibrary.cpp	2011-11-30 15:49:44.185001452 +0100
@@ -371,7 +371,7 @@ static bool qt_unix_query(const QString
     long pos = 0;
     const char pattern[] = "pattern=QT_PLUGIN_VERIFICATION_DATA";
     const ulong plen = qstrlen(pattern);
-#if defined (Q_OF_ELF) && defined(Q_CC_GNU)
+#if defined (Q_OF_ELF) && defined(Q_CC_GNU) && !(defined(__powerpc64__) || defined(__ppc64__))
     int r = QElfParser().parse(filedata, fdlen, library, lib, &pos, &fdlen);
     if (r == QElfParser::NoQtSection) {
         if (pos > 0) {
@@ -393,7 +393,7 @@ static bool qt_unix_query(const QString
     }
 #else
     pos = qt_find_pattern(filedata, fdlen, pattern, plen);
-#endif // defined(Q_OF_ELF) && defined(Q_CC_GNU)
+#endif // defined(Q_OF_ELF) && defined(Q_CC_GNU) && !(defined(__powerpc64__) || defined(__ppc64__))
     bool ret = false;
     if (pos >= 0)
         ret = qt_parse_pattern(filedata + pos, version, debug, key);
