Index: NetworkManager-vpnc-0.8.5.91/properties/nm-vpnc-dialog.ui
===================================================================
--- NetworkManager-vpnc-0.8.5.91.orig/properties/nm-vpnc-dialog.ui
+++ NetworkManager-vpnc-0.8.5.91/properties/nm-vpnc-dialog.ui
@@ -664,7 +664,7 @@
         <property name="xscale">0</property>
         <child>
           <object class="GtkButton" id="advanced_button">
-            <property name="label">Advanced...</property>
+            <property name="label" translatable="yes">Advanced...</property>
             <property name="visible">True</property>
             <property name="can_focus">True</property>
             <property name="receives_default">True</property>
