Index: NetworkManager-vpnc-0.8.5.91/po/POTFILES.in
===================================================================
--- NetworkManager-vpnc-0.8.5.91.orig/po/POTFILES.in
+++ NetworkManager-vpnc-0.8.5.91/po/POTFILES.in
@@ -4,7 +4,7 @@ auth-dialog/gnome-two-password-dialog.c
 auth-dialog/main.c
 nm-vpnc.desktop.in
 properties/nm-vpnc.c
-properties/nm-vpnc-dialog.ui
+[type: gettext/glade]properties/nm-vpnc-dialog.ui
 src/nm-vpnc-service.c
 
 
