Index: NetworkManager-vpnc-0.8.5.91/properties/nm-vpnc-dialog.ui
===================================================================
--- NetworkManager-vpnc-0.8.5.91.orig/properties/nm-vpnc-dialog.ui
+++ NetworkManager-vpnc-0.8.5.91/properties/nm-vpnc-dialog.ui
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <interface>
-  <requires lib="gtk+" version="2.24"/>
+  <requires lib="gtk+" version="2.22"/>
   <object class="GtkImage" id="image1">
     <property name="visible">True</property>
     <property name="can_focus">False</property>
