--- grub-2.00/util/grub.d/10_linux.in~	2012-04-18 23:24:38.000000000 +0200
+++ grub-2.00/util/grub.d/10_linux.in	2012-11-28 22:27:41.652673833 +0100
@@ -29,7 +29,13 @@
 CLASS="--class gnu-linux --class gnu --class os"
 
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
-  OS=GNU/Linux
+  if [ -f "/etc/pisilinux-release" ] ; then
+    OS="$(sed 's, release .*$,,g' /etc/pisilinux-release)"
+  elif [ -e "/etc/system-release" ] ; then
+    OS="$(sed 's, release .*$,,g' /etc/system-release)"
+  else
+    OS=GNU/Linux
+  fi
 else
   OS="${GRUB_DISTRIBUTOR} GNU/Linux"
   CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr 'A-Z' 'a-z' | cut -d' ' -f1) ${CLASS}"
@@ -193,6 +199,7 @@
   initrd=
   for i in "initrd.img-${version}" "initrd-${version}.img" "initrd-${version}.gz" \
 	   "initrd-${version}" "initramfs-${version}.img" \
+	   "initramfs-${version}.gz" "initramfs-${version}" \
 	   "initrd.img-${alt_version}" "initrd-${alt_version}.img" \
 	   "initrd-${alt_version}" "initramfs-${alt_version}.img" \
 	   "initramfs-genkernel-${version}" \
