diff -Naur noor-1.2.orig/contrib/noor.desktop noor-1.2/contrib/noor.desktop
--- noor-1.2.orig/contrib/noor.desktop	2009-03-28 19:10:13.000000000 +0200
+++ noor-1.2/contrib/noor.desktop	2009-03-28 23:04:03.000000000 +0200
@@ -9,8 +9,8 @@
 Name=Noor
 
 Name[tr]=Nur
-Comment[tr]=Kur\'an Okuma Programı
-GenericName[tr]=Kur\'an Okuma Programı
+Comment[tr]=Kur'an Okuma Programı
+GenericName[tr]=Kur'an Okuma Programı
 
 Name[fa]=نور
 Comment[fa]=برنامه‌ی مرورگر قرآن
