--- configure.old	2010-08-21 16:54:00.918473667 +0300
+++ configure	2010-08-21 17:05:44.390473644 +0300
@@ -23994,7 +23994,7 @@
 $as_echo_n "checking to see if compiler understands -Werror... " >&6; }
 
   save_CFLAGS="$CFLAGS"
-  CFLAGS="$CFLAGS -Werror"
+  CFLAGS="$CFLAGS"
 
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
@@ -24016,7 +24016,7 @@
   CFLAGS="$save_CFLAGS"
 
   if test "X$flag_ok" = Xyes ; then
-    ERROR_CFLAGS="$ERROR_CFLAGS -Werror"
+    ERROR_CFLAGS="$ERROR_CFLAGS"
     true
   else
 
@@ -24348,7 +24348,7 @@
 $as_echo_n "checking to see if c++ compiler understands -Werror... " >&6; }
 
   save_CPPFLAGS="$CPPFLAGS"
-  CPPFLAGS="$CPPFLAGS -Werror"
+  CPPFLAGS="$CPPFLAGS"
 
   ac_ext=cpp
 ac_cpp='$CXXCPP $CPPFLAGS'
@@ -24377,7 +24377,7 @@
   CPPFLAGS="$save_CPPFLAGS"
 
   if test "X$flag_ok" = Xyes ; then
-    ERROR_CXXFLAGS="$ERROR_CXXFLAGS -Werror"
+    ERROR_CXXFLAGS="$ERROR_CXXFLAGS"
     true
   else
 
