diff -p -up gst-plugins-bad-0.10.7/ext/timidity/gstwildmidi.c~ gst-plugins-bad-0.10.7/ext/timidity/gstwildmidi.c
--- gst-plugins-bad-0.10.7/ext/timidity/gstwildmidi.c~	2008-03-13 15:25:20.000000000 +0100
+++ gst-plugins-bad-0.10.7/ext/timidity/gstwildmidi.c	2008-05-16 16:31:45.000000000 +0200
@@ -59,7 +59,7 @@
 #include "gstwildmidi.h"
 
 #ifndef WILDMIDI_CFG
-#define WILDMIDI_CFG "/etc/timidity.cfg"
+#define WILDMIDI_CFG "/etc/timidity/timidity.cfg"
 #endif
 
 GST_DEBUG_CATEGORY_STATIC (gst_wildmidi_debug);
