diff -Nur AC1.1.0.3-old/source/src/Makefile AC1.1.0.3/source/src/Makefile
--- AC1.1.0.3-old/source/src/Makefile	2010-11-10 00:12:41.241927382 +0200
+++ AC1.1.0.3/source/src/Makefile	2010-11-10 00:14:26.169812158 +0200
@@ -165,13 +165,13 @@
 
 else
 client: libenet $(CLIENT_OBJS)
-	$(CXX) $(CXXFLAGS) -o ac_client $(CLIENT_OBJS) $(CLIENT_LIBS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o ac_client $(CLIENT_OBJS) $(CLIENT_LIBS)
 
 server: libenet $(SERVER_OBJS)
-	$(CXX) $(CXXFLAGS) -o ac_server $(SERVER_OBJS) $(SERVER_LIBS)  
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o ac_server $(SERVER_OBJS) $(SERVER_LIBS)  
 
 master: libenet $(MASTER_OBJS)
-	$(CXX) $(CXXFLAGS) -o ac_master $(MASTER_OBJS) $(SERVER_LIBS)  
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o ac_master $(MASTER_OBJS) $(SERVER_LIBS)  
 	
 client_install: client
 	cp ac_client ../../bin_unix/$(PLATFORM_PREFIX)_client
