diff -Nur AC1.1.0.3-old/source/enet/Makefile.am AC1.1.0.3/source/enet/Makefile.am
--- AC1.1.0.3-old/source/enet/Makefile.am	2010-11-10 00:41:36.680926656 +0200
+++ AC1.1.0.3/source/enet/Makefile.am	2010-11-10 00:41:47.153930326 +0200
@@ -10,11 +10,10 @@
 	include/enet/time.h \
 	include/enet/types.h \
 	include/enet/unix.h \
-	include/enet/utility.h \
-	include/enet/win32.h
+	include/enet/utility.h
 
 lib_LTLIBRARIES = libenet.la
-libenet_la_SOURCES = callbacks.c compress.c host.c list.c packet.c peer.c protocol.c unix.c win32.c
+libenet_la_SOURCES = callbacks.c compress.c host.c list.c packet.c peer.c protocol.c unix.c
 # see info '(libtool) Updating version info' before making a release
 libenet_la_LDFLAGS = $(AM_LDFLAGS) -version-info 1:0:0
 INCLUDES = -Iinclude
