diff -Nur AssaultCube_v1.0.4-old/source/src/Makefile AssaultCube_v1.0.4/source/src/Makefile
--- AssaultCube_v1.0.4-old/source/src/Makefile	2009-09-08 16:09:30.716024796 +0300
+++ AssaultCube_v1.0.4/source/src/Makefile	2009-09-08 16:09:42.392048557 +0300
@@ -1,12 +1,12 @@
-CXXOPTFLAGS=-O3 -fomit-frame-pointer
-INCLUDES1=-I/usr/X11R6/include -I../enet/include -I../src
+CXXOPTFLAGS?=-O3 -fomit-frame-pointer
+INCLUDES1=-I../enet/include -I/usr/X11R6/include -I../src
 INCLUDES2=`sdl-config --cflags`
 CXXCUSTOMFLAGS=-fsigned-char -Wall -Wextra -Wno-missing-field-initializers -Wno-unused -Wpointer-arith -Wwrite-strings -Wconversion -rdynamic
 CXXFLAGS=$(CXXOPTFLAGS) $(INCLUDES1) $(INCLUDES2) $(CUSTOMFLAGS)
 CXXFLAGSSERVER=$(CXXOPTFLAGS) $(INCLUDES1) $(CUSTOMFLAGS)
 
 PLATFORM_PREFIX=native
-CLIENT_LIBS=-L/usr/X11R6/lib -L../enet -lenet `sdl-config --libs` -lSDL_image -lz -lGL -lopenal -lvorbisfile
+CLIENT_LIBS=-L../enet -lenet -L/usr/X11R6/lib `sdl-config --libs` -lSDL_image -lz -lGL -lopenal -lvorbisfile -lX11
 CLIENT_OBJS1= \
 	client.o \
 	clientgame.o \
