diff -Nur AC1.1.0.3-old/source/src/Makefile AC1.1.0.3/source/src/Makefile
--- AC1.1.0.3-old/source/src/Makefile	2010-11-10 00:26:58.431926877 +0200
+++ AC1.1.0.3/source/src/Makefile	2010-11-10 00:29:24.742805883 +0200
@@ -5,7 +5,7 @@
 PLATFORM= $(shell uname -s)
 PLATFORM_PREFIX=native
 
-INCLUDES= -I. -Ibot -I../enet/include
+INCLUDES= -I. -Ibot -I/usr/include/enet
 
 STRIP=
 ifeq (,$(findstring -g,$(CXXFLAGS)))
@@ -26,7 +26,7 @@
 override CXXFLAGS+= -rdynamic
 CLIENT_INCLUDES= $(INCLUDES) -I/usr/include `sdl-config --cflags` -idirafter ../include
 # -lSDL_ttf 
-CLIENT_LIBS= -L../enet/.libs -lenet -L$(USRLIB) `sdl-config --libs` -lSDL_image -lz -lGL -lopenal -lvorbisfile
+CLIENT_LIBS= -lenet -L$(USRLIB) `sdl-config --libs` -lSDL_image -lz -lGL -lopenal -lvorbisfile
 endif
 
 CLIENT_OBJS= \
@@ -93,7 +93,7 @@
 SERVER_LIBS= -L../lib -lzdll -lenet -llibintl -lws2_32 -lwinmm
 else
 SERVER_INCLUDES= -DSTANDALONE $(INCLUDES)
-SERVER_LIBS= -L../enet/.libs -lenet -lz
+SERVER_LIBS= -lenet -lz
 endif
 
 SERVER_OBJS= \
@@ -121,15 +121,6 @@
 
 all: client server
 
-../enet/Makefile:
-	cd ../enet; ./configure --enable-shared=no --enable-static=yes
-
-libenet: ../enet/Makefile
-	-$(MAKE) -C ../enet all
-
-clean-enet: ../enet/Makefile
-	$(MAKE) -C ../enet/ clean
-
 clean:
 	-$(RM) $(CLIENT_PCH) $(CLIENT_OBJS) $(SERVER_OBJS) $(MASTER_OBJS) ac_client ac_server ac_master
 #	-$(MAKE) -C ../enet/ clean
@@ -164,13 +155,13 @@
 server_install: server
 
 else
-client: libenet $(CLIENT_OBJS)
+client: $(CLIENT_OBJS)
 	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o ac_client $(CLIENT_OBJS) $(CLIENT_LIBS)
 
-server: libenet $(SERVER_OBJS)
+server: $(SERVER_OBJS)
 	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o ac_server $(SERVER_OBJS) $(SERVER_LIBS)  
 
-master: libenet $(MASTER_OBJS)
+master: $(MASTER_OBJS)
 	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o ac_master $(MASTER_OBJS) $(SERVER_LIBS)  
 	
 client_install: client
