Index: doxygen-1.5.8/configure
===================================================================
--- doxygen-1.5.8.orig/configure
+++ doxygen-1.5.8/configure
@@ -485,6 +485,7 @@ INSTTOOL  = $f_insttool
 DOXYDOCS  = ..
 DOCDIR    = $f_docdir
 QTDIR     = $QTDIR
+MAN1DIR    = share/man/man1
 EOF
 
 if test "$f_dot" != NO; then
Index: doxygen-1.5.8/addon/doxywizard/Makefile.in
===================================================================
--- doxygen-1.5.8.orig/addon/doxywizard/Makefile.in
+++ doxygen-1.5.8/addon/doxywizard/Makefile.in
@@ -29,11 +29,11 @@ distclean: Makefile.doxywizard
 	$(RM) Makefile.doxywizard
 
 install:
-	$(INSTTOOL) -d $(INSTALL)/bin	
-	$(INSTTOOL) -m 755 ../../bin/doxywizard $(INSTALL)/bin	
-	$(INSTTOOL) -d $(INSTALL)/$(MAN1DIR)
+	$(INSTTOOL) -d $(DESTDIR)$(INSTALL)/bin	
+	$(INSTTOOL) -m 755 ../../bin/doxywizard $(DESTDIR)$(INSTALL)/bin	
+	$(INSTTOOL) -d $(DESTDIR)$(INSTALL)/$(MAN1DIR)
 	cat ../../doc/doxywizard.1 | sed -e "s/DATE/$(DATE)/g" -e "s/VERSION/$(VERSION)/g" > doxywizard.1
-	$(INSTTOOL) -m 644 doxywizard.1 $(INSTALL)/$(MAN1DIR)/doxywizard.1
+	$(INSTTOOL) -m 644 doxywizard.1 $(DESTDIR)$(INSTALL)/$(MAN1DIR)/doxywizard.1
 	rm doxywizard.1
 
 FORCE:
Index: doxygen-1.5.8/Makefile.in
===================================================================
--- doxygen-1.5.8.orig/Makefile.in
+++ doxygen-1.5.8/Makefile.in
@@ -44,8 +44,6 @@ distclean: clean
 
 DATE=$(shell date "+%B %Y")
 
-MAN1DIR = man/man1
-
 install: doxywizard_install 
 	$(INSTTOOL) -d $(DESTDIR)/$(INSTALL)/bin
 	$(INSTTOOL) -m 755 bin/doxygen    $(DESTDIR)/$(INSTALL)/bin
Index: doxygen-1.5.8/tmake/lib/linux-g++/tmake.conf
===================================================================
--- doxygen-1.5.8.orig/tmake/lib/linux-g++/tmake.conf
+++ doxygen-1.5.8/tmake/lib/linux-g++/tmake.conf
@@ -7,16 +7,16 @@
 TEMPLATE		= app
 CONFIG			= qt warn_on release
 
-TMAKE_CC		= gcc
+TMAKE_CC		= PARDUS_CC
 TMAKE_CFLAGS		= -pipe
 TMAKE_CFLAGS_WARN_ON	= -Wall -W -fno-exceptions
 TMAKE_CFLAGS_WARN_OFF	=
-TMAKE_CFLAGS_RELEASE	= -O2
+TMAKE_CFLAGS_RELEASE	= PARDUS_CFLAGS
 TMAKE_CFLAGS_DEBUG	= -g
 TMAKE_CFLAGS_SHLIB	= -fPIC
 TMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
 
-TMAKE_CXX		= g++
+TMAKE_CXX		= PARDUS_CXX
 TMAKE_CXXFLAGS		= $$TMAKE_CFLAGS
 TMAKE_CXXFLAGS_WARN_ON	= $$TMAKE_CFLAGS_WARN_ON
 TMAKE_CXXFLAGS_WARN_OFF	= $$TMAKE_CFLAGS_WARN_OFF
@@ -27,17 +27,17 @@ TMAKE_CXXFLAGS_YACC	= $$TMAKE_CFLAGS_YAC
 
 TMAKE_INCDIR		=
 TMAKE_LIBDIR		=
-TMAKE_INCDIR_X11	= /usr/X11R6/include
-TMAKE_LIBDIR_X11	= /usr/X11R6/lib
-TMAKE_INCDIR_QT		= $(QTDIR)/include
-TMAKE_LIBDIR_QT		= $(QTDIR)/lib
-TMAKE_INCDIR_OPENGL	= /usr/X11R6/include
-TMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
+TMAKE_INCDIR_X11	=
+TMAKE_LIBDIR_X11	=
+TMAKE_INCDIR_QT		=
+TMAKE_LIBDIR_QT		=
+TMAKE_INCDIR_OPENGL	=
+TMAKE_LIBDIR_OPENGL	=
 
-TMAKE_LINK		= g++
-TMAKE_LINK_SHLIB	= g++
+TMAKE_LINK		= PARDUS_CXX
+TMAKE_LINK_SHLIB	= PARDUS_CXX
 TMAKE_LFLAGS		=
-TMAKE_LFLAGS_RELEASE	=
+TMAKE_LFLAGS_RELEASE	= PARDUS_LDFLAGS
 TMAKE_LFLAGS_DEBUG	= -g
 TMAKE_LFLAGS_SHLIB	= -shared
 TMAKE_LFLAGS_SONAME	= -Wl,-soname,
