diff -up doxygen-1.7.5/src/configoptions.cpp.timestamp doxygen-1.7.5/src/configoptions.cpp
--- doxygen-1.7.5/src/configoptions.cpp.timestamp	2011-08-03 15:54:50.000000000 +0200
+++ doxygen-1.7.5/src/configoptions.cpp	2011-08-23 12:55:56.000000000 +0200
@@ -1173,6 +1173,14 @@ void addConfigOptions(Config *cfg)
   cs->setWidgetType(ConfigString::File);
   cs->addDependency("GENERATE_HTML");
   //----
+  cb = cfg->addBool(
+                 "HTML_TIMESTAMP",
+                 "If the HTML_TIMESTAMP tag is set to YES then the generated HTML\n"
+                 "documentation will contain the timesstamp.",
+                 FALSE
+                );
+  cb->addDependency("GENERATE_HTML");
+  //----
   cs = cfg->addString(
                  "HTML_STYLESHEET",
                  "The HTML_STYLESHEET tag can be used to specify a user-defined cascading\n"
diff -up doxygen-1.7.5/src/config.xml.timestamp doxygen-1.7.5/src/config.xml
--- doxygen-1.7.5/src/config.xml.timestamp	2011-08-03 15:54:48.000000000 +0200
+++ doxygen-1.7.5/src/config.xml	2011-08-23 12:55:56.000000000 +0200
@@ -819,6 +819,11 @@ The HTML_FOOTER tag can be used to speci
 each generated HTML page. If it is left blank doxygen will generate a 
 standard footer. 
 ' defval='' depends='GENERATE_HTML'/>
+    <option type='bool' id='HTML_TIMESTAMP' docs='
+If the HTML_TIMESTAMP tag is set to YES then the generated HTML
+documentation will contain the timesstamp.
+' defval='' depends='GENERATE_HTML'/>
+
     <option type='string' id='HTML_STYLESHEET' format='file' docs='
 The HTML_STYLESHEET tag can be used to specify a user-defined cascading 
 style sheet that is used by each HTML page. It can be used to 
diff -up doxygen-1.7.5/src/htmlgen.cpp.timestamp doxygen-1.7.5/src/htmlgen.cpp
--- doxygen-1.7.5/src/htmlgen.cpp.timestamp	2011-08-01 22:10:17.000000000 +0200
+++ doxygen-1.7.5/src/htmlgen.cpp	2011-08-23 13:01:16.000000000 +0200
@@ -88,7 +88,7 @@ static const char svgpan_script[]=
 
 static QCString g_header;
 static QCString g_footer;
-
+static bool timestamp=false;
 //------------------------- Pictures for the Tabs ------------------------
 
 // active
@@ -1072,6 +1072,8 @@ void HtmlGenerator::init()
   {
     g_footer = defaultHtmlFooter;
   }
+  if (Config_getBool("HTML_TIMESTAMP"))
+    timestamp=true;
   createSubDirs(d);
 
   QCString fileName=dname+"/tabs.css";
@@ -1285,7 +1287,7 @@ QCString HtmlGenerator::writeLogoAsStrin
   if (timeStamp)
   {
     result += theTranslator->trGeneratedAt(
-               dateToString(TRUE),
+               dateToString(timestamp),
                Config_getString("PROJECT_NAME")
               );
   }
