Index: pdb2pqr-1.7/main.py
===================================================================
--- pdb2pqr-1.7.orig/main.py
+++ pdb2pqr-1.7/main.py
@@ -493,6 +493,9 @@ def mainCommand(argv):
     if package_path != "":
         sys.path.extend(package_path.split(":"))
 
+    from distutils.sysconfig import get_python_lib
+    sys.path += ["%s/pdb2pqr/" % get_python_lib()]
+
     shortOptlist = "h,v"
     longOptlist = ["help","verbose","ff=","ffout=","nodebump","noopt","with-ph=","apbs-input","chain","clean","assign-only", "ligand=", "whitespace","typemap","neutraln","neutralc","userff","usernames"]
 
