diff -up alpine-2.00/pith/reply.c.gcc44 alpine-2.00/pith/reply.c
--- alpine-2.00/pith/reply.c.gcc44	2008-06-03 14:27:23.000000000 -0500
+++ alpine-2.00/pith/reply.c	2009-05-06 08:07:27.000000000 -0500
@@ -357,6 +357,7 @@ set_role_from_msg(struct pine *ps, long 
  * reply_seed - fill in reply header
  * 
  */
+__attribute__((__optimize__(0)))
 void
 reply_seed(struct pine *ps, ENVELOPE *outgoing, ENVELOPE *env,
 	   struct mail_address *saved_from, struct mail_address *saved_to,
