diff -ur alpine-2.00.orig/alpine/mailcmd.c alpine-2.00/alpine/mailcmd.c
--- alpine-2.00.orig/alpine/mailcmd.c	2008-08-22 00:14:45.000000000 +0200
+++ alpine-2.00/alpine/mailcmd.c	2010-10-05 17:48:23.661844699 +0200
@@ -7873,7 +7873,7 @@
 
 	      case 13 :			/* Match my addresses */
 		me++;
-		snprintf(sstring, sizeof(sstring), not ? _(dont_match_me) : _(match_me));
+		snprintf(sstring, sizeof(sstring), "%s", not ? _(dont_match_me) : _(match_me));
 		continue;
 
 	      case 14 :			/* Subject: default */
diff -ur alpine-2.00.orig/pith/ldap.c alpine-2.00/pith/ldap.c
--- alpine-2.00.orig/pith/ldap.c	2008-04-08 00:58:40.000000000 +0200
+++ alpine-2.00/pith/ldap.c	2010-10-05 17:49:29.437312505 +0200
@@ -778,7 +778,7 @@
 	 */
 	switch(args){
 	  case 0:
-	    snprintf(filter, sizeof(filter), filt_format);
+	    snprintf(filter, sizeof(filter), "%s", filt_format);
 	    break;
 	  case 1:
 	    snprintf(filter, sizeof(filter), filt_format, scp);
diff -ur alpine-2.00.orig/imap/src/osdep/unix/flocklnx.c alpine-2.00/imap/src/osdep/unix/flocklnx.c
--- alpine-2.00.orig/imap/src/osdep/unix/flocklnx.c	2010-10-02 10:37:57.000000000 +0200
+++ alpine-2.00/imap/src/osdep/unix/flocklnx.c	2010-10-19 22:34:51.302425816 +0200
@@ -57,7 +57,7 @@
     case ENOLCK:		/* lock table is full */
       sprintf (tmp,"File locking failure: %s",strerror (errno));
       mm_log (tmp,WARN);	/* give the user a warning of what happened */
-      if (!logged++) syslog (LOG_ERR,tmp);
+      if (!logged++) syslog (LOG_ERR,"File locking failure: %s",strerror (errno));
 				/* return failure if non-blocking lock */
       if (op & LOCK_NB) return -1;
       sleep (5);		/* slow down in case it loops */
