--- alpine/Makefile.am.bak	2010-10-02 10:37:57.000000000 +0200
+++ alpine/Makefile.am	2010-10-20 02:12:10.463180499 +0200
@@ -35,7 +35,7 @@
 LDADD = ../pico/libpico.a ../pico/osdep/libpicoosd.a \
 	../pith/libpith.a ../pith/osdep/libpithosd.a  \
 	../pith/charconv/libpithcc.a \
-	osdep/libpineosd.a ../c-client/c-client.a
+	osdep/libpineosd.a ../c-client/c-client.a -lpam -lcrypto
 
 AM_CPPFLAGS = -I@top_builddir@/include -I@top_srcdir@/include -DLOCALEDIR=\"$(localedir)\"
 
--- configure.ac.link	2011-12-15 18:27:12.412081009 +0100
+++ configure.ac	2011-12-15 18:27:20.623650740 +0100
@@ -1608,9 +1608,6 @@
 if test -n "$LDFLAGS" ; then
   alpine_c_client_ldflags="$alpine_c_client_ldflags ${LDFLAGS}"
 fi
-if test -n "$LIBS" ; then
-  alpine_c_client_ldflags="$alpine_c_client_ldflags ${LIBS}"
-fi
 
 if test -n "$alpine_c_client_ldflags" ; then
   AC_SUBST(C_CLIENT_LDFLAGS, EXTRALDFLAGS=\"$alpine_c_client_ldflags\")
