--- ./interfaces/driver/lb-blackbox-data.h.orig	2012-06-07 02:30:26.000000000 -0600
+++ ./interfaces/driver/lb-blackbox-data.h	2012-10-02 13:58:40.725472935 -0600
@@ -148,7 +148,7 @@ public:
 	{
 		Blackbox<DomainSource> *B_source= static_cast<Blackbox<DomainSource> * >  (ptr);
 		Blackbox<DomainTarget> *B_target = NULL; /*  was not init't */
-		typename Blackbox<DomainSource>::template rebind<DomainTarget>()(*B_target, *B_source, *D);
+		typename Blackbox<DomainSource>::template rebind<DomainTarget>()(*B_target, *B_source);
 		delete B_source;
 		ptr = B_target;
 	}
