--- ./tests/Makefile.in.orig	2012-06-08 02:27:28.000000000 -0600
+++ ./tests/Makefile.in	2012-09-26 11:25:17.946134272 -0600
@@ -1260,14 +1260,14 @@ AM_CPPFLAGS = -I$(top_srcdir) -I$(top_sr
 	$(FFLAS_FFPACK_CFLAGS) $(GMP_CFLAGS) $(NTL_CFLAGS) \
 	$(GIVARO_CFLAGS) $(LIDIA_CFLAGS) $(IML_CFLAGS)
 # AM_LDFLAGS = -llapack -lcblas -latlas
-AM_LDFLAGS = $(top_srcdir)/linbox/.libs/liblinbox.a
+AM_LDFLAGS = $(top_srcdir)/linbox/liblinbox.la
 #  SAFER_FLAGS = -g -Wall -Wextra -Wno-unused-parameter
 # bds: no need to optimize for unit/regression tests, and opt adds a lot to compilation time
 #AM_CXXFLAGS = @TESTS_CFLAGS@ -DLinBoxSrcOnly
 #  AM_CXXFLAGS = $(SAFER_FLAGS) -DLinBoxSrcOnly
 #AM_CXXFLAGS = $(SAFER_FLAGS) -DLinBoxSrcOnly -O2
 #  AM_CXXFLAGS = -g -DLinBoxSrcOnly -Wall -DDEBUG -O0
-AM_CXXFLAGS = -DLinBoxSrcOnly  @TESTS_CFLAGS@ @DEBUG_CFLAGS@
+AM_CXXFLAGS = -DLinBoxTestOnly  @TESTS_CFLAGS@ @DEBUG_CFLAGS@
 # AM_CXXFLAGS = -g -DLinBoxSrcOnly -Wall -D__LINBOX_HAVE_DGETRF -D__LINBOX_HAVE_DGETRI
 # AM_CFLAGS = -g -Wall
 SUBDIRS = data matrix
--- ./tests/test-common.h.orig	2012-06-07 02:30:32.000000000 -0600
+++ ./tests/test-common.h	2012-09-26 11:25:17.949139540 -0600
@@ -381,7 +381,7 @@ extern inline double incompleteGamma (do
 
 double chiSquaredCDF (double chi_sqr, double df);
 
-#ifdef LinBoxSrcOnly
+#ifdef LinBoxTestOnly
 #include "test-common.C"
 #endif
 #endif // __LINBOX_test_common_H
