--- ./linbox/field/Modular/modular-int64.h.orig	2012-06-07 02:30:27.000000000 -0600
+++ ./linbox/field/Modular/modular-int64.h	2012-09-26 11:26:12.483580130 -0600
@@ -96,7 +96,7 @@ namespace LinBox
 	public:
 		typedef int64_t Element;
 
-		typedef FFPACK::Modular<int64_t> Father_t
+		typedef FFPACK::Modular<int64_t> Father_t;
 		friend class FieldAXPY<Modular<int64_t> >;
 		friend class DotProductDomain<Modular<int64_t> >;
 		friend class MVProductDomain<Modular<int64_t> >;
