diff -ur evince-2.26.2.orig/data/evince.desktop.in.in evince-2.26.2/data/evince.desktop.in.in
--- evince-2.26.2.orig/data/evince.desktop.in.in	2009-04-26 13:11:39.000000000 +0300
+++ evince-2.26.2/data/evince.desktop.in.in	2009-06-22 13:26:38.761938892 +0300
@@ -8,7 +8,6 @@
 Terminal=false
 Type=Application
 Icon=evince
-NoDisplay=true
 X-GNOME-DocPath=
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=evince
