Index: Falcon-0.9.6.6/CMakeLists.txt
===================================================================
--- Falcon-0.9.6.6.orig/CMakeLists.txt
+++ Falcon-0.9.6.6/CMakeLists.txt
@@ -200,7 +200,7 @@ set( doc_files
 
 install(
    FILES ${doc_files}
-   DESTINATION ${FALCON_SHARE_DIR} )
+   DESTINATION "share/doc/falcon" )
 
 #########################################################à
 # Subdirectories
