diff -up aspell-0.60.6.1/configure.mlib aspell-0.60.6.1/configure
--- aspell-0.60.6.1/configure.mlib	2011-08-16 11:40:48.000000000 +0200
+++ aspell-0.60.6.1/configure	2011-08-16 11:41:44.013663519 +0200
@@ -18989,7 +18989,7 @@ rm -f core conftest.err conftest.$ac_obj
 #                                                                 #
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 
-ac_config_files="$ac_config_files Makefile gen/Makefile common/Makefile lib/Makefile data/Makefile auto/Makefile modules/Makefile modules/tokenizer/Makefile modules/speller/Makefile modules/speller/default/Makefile interfaces/Makefile interfaces/cc/Makefile scripts/Makefile examples/Makefile prog/Makefile manual/Makefile po/Makefile.in m4/Makefile modules/filter/Makefile myspell/Makefile lib5/Makefile"
+ac_config_files="$ac_config_files Makefile gen/Makefile common/Makefile lib/Makefile data/Makefile auto/Makefile modules/Makefile modules/tokenizer/Makefile modules/speller/Makefile modules/speller/default/Makefile interfaces/Makefile interfaces/cc/Makefile aspell.pc scripts/Makefile examples/Makefile prog/Makefile manual/Makefile po/Makefile.in m4/Makefile modules/filter/Makefile myspell/Makefile lib5/Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -19985,7 +19985,7 @@ do
     "modules/filter/Makefile") CONFIG_FILES="$CONFIG_FILES modules/filter/Makefile" ;;
     "myspell/Makefile") CONFIG_FILES="$CONFIG_FILES myspell/Makefile" ;;
     "lib5/Makefile") CONFIG_FILES="$CONFIG_FILES lib5/Makefile" ;;
-
+    "aspell.pc" ) CONFIG_FILES="$CONFIG_FILES aspell.pc" ;;
   *) { { echo "$as_me:$LINENO: error: invalid argument: $ac_config_target" >&5
 echo "$as_me: error: invalid argument: $ac_config_target" >&2;}
    { (exit 1); exit 1; }; };;
diff -up aspell-0.60.6.1/Makefile.in.mlib aspell-0.60.6.1/Makefile.in
--- aspell-0.60.6.1/Makefile.in.mlib	2011-08-16 11:20:09.000000000 +0200
+++ aspell-0.60.6.1/Makefile.in	2011-08-16 11:46:30.643236786 +0200
@@ -816,6 +816,8 @@ clean-filterLTLIBRARIES:
 	done
 install-libLTLIBRARIES: $(lib_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
+	mkdir -p $(libdir)/pkgconfig; \
+	cp aspell.pc $(libdir)/pkgconfig/aspell.pc; \
 	test -z "$(libdir)" || $(MKDIR_P) "$(DESTDIR)$(libdir)"
 	@list='$(lib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
diff -up aspell-0.60.6.1/scripts/mkconfig.mlib aspell-0.60.6.1/scripts/mkconfig
--- aspell-0.60.6.1/scripts/mkconfig.mlib	2004-01-03 13:06:24.000000000 +0100
+++ aspell-0.60.6.1/scripts/mkconfig	2011-08-16 11:42:46.810519200 +0200
@@ -15,7 +15,7 @@ case \$1 in
   echo "$2"
   ;;
 --pkgdatadir | pkgdatadir)
-  echo "$3"
+  pkg-config aspell --variable=pkgdatadir
   ;;
 *)
   echo "usage: pspell-config version|datadir|pkgdatadir"
--- /dev/null	2007-01-02 09:09:01.616000852 +0100
+++ aspell-0.60.6.1/aspell.pc.in	2007-01-02 14:59:04.000000000 +0100
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+pkgdatadir=@pkgdatadir@
+
+Name: Aspell
+Description: A spelling checker.
+Version: @VERSION@
+Requires: 
+Libs: -L${libdir} -laspell
+Cflags: -I${includedir}
