diff -Nur kdebase-workspace-4.5.5-orig/kcontrol/randr/CMakeLists.txt kdebase-workspace-4.5.5/kcontrol/randr/CMakeLists.txt
--- kdebase-workspace-4.5.5-orig/kcontrol/randr/CMakeLists.txt	2011-01-18 16:39:43.745000114 +0200
+++ kdebase-workspace-4.5.5/kcontrol/randr/CMakeLists.txt	2011-01-18 16:40:53.284000115 +0200
@@ -26,6 +26,7 @@
 
 set(kcm_randr_PART_SRCS 
    krandrmodule.cpp 
+   glinfo.cpp
    legacyrandrconfig.cpp
    ${randrinternal_PART_SRCS})
 
@@ -38,7 +39,10 @@
 kde4_add_plugin(kcm_randr ${kcm_randr_PART_SRCS})
 
 
-target_link_libraries(kcm_randr  ${KDE4_KDEUI_LIBS} ${X11_Xrandr_LIB} ${X11_LIBRARIES})
+target_link_libraries(kcm_randr  ${KDE4_KDEUI_LIBS} ${X11_Xrandr_LIB}
+    ${X11_LIBRARIES} ${QT_QTGUI_LIBRARY} ${X11_X11_LIB} ${OPENGL_gl_LIBRARY} ${KDE4_KIO_LIBS})
+
+
 
 install(TARGETS kcm_randr  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
diff -Nur kdebase-workspace-4.5.5-orig/kcontrol/randr/glinfo.cpp kdebase-workspace-4.5.5/kcontrol/randr/glinfo.cpp
--- kdebase-workspace-4.5.5-orig/kcontrol/randr/glinfo.cpp	1970-01-01 02:00:00.000000000 +0200
+++ kdebase-workspace-4.5.5/kcontrol/randr/glinfo.cpp	2011-01-18 16:40:53.287000116 +0200
@@ -0,0 +1,63 @@
+#include <glinfo.h>
+#include <kdebug.h>
+
+#include <X11/Xlib.h>
+#include <X11/Xutil.h>
+
+// OpenGL includes
+#include <GL/gl.h>
+#include <GL/glext.h>
+#include <GL/glx.h>
+
+
+void GlInfo::getGlStrings()
+{
+  GLXContext ctx;
+  char *displayName = NULL;
+  int scrnum = 0;
+
+  const int attribSingle[] = {
+    GLX_RGBA,
+    GLX_RED_SIZE, 1,
+    GLX_GREEN_SIZE, 1,
+    GLX_BLUE_SIZE, 1,
+    None };
+  const int attribDouble[] = {
+    GLX_RGBA,
+    GLX_RED_SIZE, 1,
+    GLX_GREEN_SIZE, 1,
+    GLX_BLUE_SIZE, 1,
+    GLX_DOUBLEBUFFER,
+    None };
+
+  Display *dpy = XOpenDisplay(displayName);
+  unsigned long mask;
+  XVisualInfo *visinfo;
+  Window root, win;
+  XSetWindowAttributes attr;
+
+  root = DefaultRootWindow(dpy);
+
+  visinfo = glXChooseVisual(dpy, scrnum, const_cast<int*>(attribSingle));
+  if (!visinfo) {
+     visinfo = glXChooseVisual(dpy, scrnum, const_cast<int*>(attribDouble));
+     if (!visinfo) {
+       kDebug() << "Error: couldn't find RGB GLX visual\n";
+       return ;
+     }
+  }
+
+  attr.colormap = XCreateColormap(dpy, root, visinfo->visual, AllocNone);
+  attr.event_mask = StructureNotifyMask | ExposureMask;
+  mask = CWBackPixel | CWBorderPixel | CWColormap | CWEventMask;
+  win = XCreateWindow(dpy, root, 0, 0, 600, 600, 0, visinfo->depth, InputOutput, visinfo->visual, mask, &attr);
+
+  ctx = glXCreateContext( dpy, visinfo, NULL, GL_TRUE);
+
+  if (glXMakeCurrent(dpy, win, ctx)) {
+    glVendor = (const char *) glGetString(GL_VENDOR);
+    glRenderer = (const char *) glGetString(GL_RENDERER);
+    glVersion = (const char *) glGetString(GL_VERSION);
+  }
+
+}
diff -Nur kdebase-workspace-4.5.5-orig/kcontrol/randr/glinfo.h kdebase-workspace-4.5.5/kcontrol/randr/glinfo.h
--- kdebase-workspace-4.5.5-orig/kcontrol/randr/glinfo.h	1970-01-01 02:00:00.000000000 +0200
+++ kdebase-workspace-4.5.5/kcontrol/randr/glinfo.h	2011-01-18 16:40:53.287000116 +0200
@@ -0,0 +1,14 @@
+#include <QtCore/QString>
+
+class GlInfo
+
+{
+public:
+
+    void getGlStrings();
+    const char *glVendor;
+    const char *glRenderer;
+    const char *glVersion;
+
+};
+
diff -Nur kdebase-workspace-4.5.5-orig/kcontrol/randr/krandrmodule.cpp kdebase-workspace-4.5.5/kcontrol/randr/krandrmodule.cpp
--- kdebase-workspace-4.5.5-orig/kcontrol/randr/krandrmodule.cpp	2011-01-18 16:39:43.745000114 +0200
+++ kdebase-workspace-4.5.5/kcontrol/randr/krandrmodule.cpp	2011-01-18 16:40:53.302000102 +0200
@@ -24,12 +24,26 @@
 #include "randrdisplay.h"
 #include "randrconfig.h"
 
+// X11 includes
+#include <X11/Xlib.h>
+#include <X11/Xutil.h>
+
+// OpenGL includes
+#include <GL/gl.h>
+#include <GL/glext.h>
+#include <GL/glx.h>
+
 #include <KPluginFactory>
 #include <KPluginLoader>
 #include <KDebug>
+#include <KProcess>
+#include <KMessageBox>
+#include <kdesktopfileactions.h>
 #include <config-randr.h>
 
 #include "randr.h"
+#include "glinfo.h"
+
 
 // DLL Interface for kcontrol
 K_PLUGIN_FACTORY(KSSFactory, registerPlugin<KRandRModule>();)
@@ -82,6 +96,7 @@
 KRandRModule::~KRandRModule(void)
 {
 	delete m_display;
+    delete glInfo;;
 }
 
 void KRandRModule::defaults()
@@ -99,11 +114,45 @@
 
 void KRandRModule::load()
 {
-	kDebug() << "Loading KRandRModule...";
-	
-        if (!m_display->isValid()) {
-                return;
+
+    glInfo = new GlInfo();
+    glInfo->getGlStrings();
+
+    QString vendorText;
+    vendorText = glInfo->glVendor;
+
+    qDebug() << vendorText << endl;
+
+    if (vendorText.startsWith("NVIDIA")){
+        int ret = KMessageBox::questionYesNo(this,
+                                    i18n("You are using the proprietary driver provided by the manufacturer.\n"
+                                         "Do you want to use nvidia-settings for your preferencies?"));
+        if(ret == KMessageBox::Yes){
+
+          KUrl url =  KUrl::fromPath("/usr/share/applications/nvidia-settings.desktop");
+          KDesktopFileActions::run(url, true);
+          qDebug() << "YES";
+
+        }
+    }
+    else if (vendorText.startsWith("ATI")){
+        int ret = KMessageBox::questionYesNo(this,
+                                    i18n("You are using the proprietary driver provided by the manufacturer.\n"
+                                         "Do you want to use ati-control-center for your preferencies ?"));
+        if(ret == KMessageBox::Yes){
+
+          KUrl url =  KUrl::fromPath("/usr/share/applications/amdccclesu.desktop");
+          KDesktopFileActions::run(url, true);
+
+
+            qDebug() << "YES";
         }
+    }
+
+    if (!m_display->isValid()) {
+                return;
+    }
+
 #ifdef HAS_RANDR_1_2
 	if (RandR::has_1_2)
 		m_config->load();
diff -Nur kdebase-workspace-4.5.5-orig/kcontrol/randr/krandrmodule.h kdebase-workspace-4.5.5/kcontrol/randr/krandrmodule.h
--- kdebase-workspace-4.5.5-orig/kcontrol/randr/krandrmodule.h	2011-01-18 16:39:43.746000114 +0200
+++ kdebase-workspace-4.5.5/kcontrol/randr/krandrmodule.h	2011-01-18 16:40:53.308000131 +0200
@@ -28,6 +28,7 @@
 class RandRDisplay;
 class LegacyRandRConfig;
 class RandRConfig;
+class GlInfo;
 
 class KRandRModule : public KCModule
 {
@@ -49,6 +50,7 @@
 	LegacyRandRConfig *m_legacyConfig;
 	RandRConfig *m_config;
 	RandRDisplay *m_display;
+    GlInfo* glInfo;
 };
 
 #endif
