Index: kdebase-workspace-4.5.5/kcontrol/dateandtime/helper.cpp
===================================================================
--- kdebase-workspace-4.5.5.orig/kcontrol/dateandtime/helper.cpp
+++ kdebase-workspace-4.5.5/kcontrol/dateandtime/helper.cpp
@@ -165,8 +165,8 @@ int ClockHelper::tz( const QString& sele
 #else
         QString tz = "/usr/share/zoneinfo/" + selectedzone;
 
-        if( !KStandardDirs::findExe( "zic" ).isEmpty()) {
-            KProcess::execute("zic", QStringList() << "-l" << selectedzone);
+        if(QFile::exists("/usr/sbin/exec")) {
+            KProcess::execute("/usr/sbin/zic", QStringList() << "-l" << selectedzone);
         } else if (!QFile::remove("/etc/localtime")) {
           ret |= TimezoneError;
         } else if (!QFile::copy(tz, "/etc/localtime")) {
