Index: kdebase-workspace-4.5.2/ksmserver/shutdowndlg.cpp
===================================================================
--- kdebase-workspace-4.5.2.orig/ksmserver/shutdowndlg.cpp
+++ kdebase-workspace-4.5.2/ksmserver/shutdowndlg.cpp
@@ -475,6 +475,8 @@ KSMShutdownDlg::KSMShutdownDlg( QWidget*
                 for (QStringList::ConstIterator it = rebootOptions.constBegin(); it != rebootOptions.constEnd(); ++it, ++index) {
                     QString label = (*it);
                     label=label.replace('&',"&&");
+                    if (label == "otheroptions")
+                        continue;
                     QAction* action = new QAction(label, rebootActionGroup);
                     action->setData(index);
                     if (index == cur) {
Index: kdebase-workspace-4.5.2/kdm/kfrontend/kdmshutdown.cpp
===================================================================
--- kdebase-workspace-4.5.2.orig/kdm/kfrontend/kdmshutdown.cpp
+++ kdebase-workspace-4.5.2/kdm/kfrontend/kdmshutdown.cpp
@@ -491,11 +491,14 @@ KDMSlimShutdown::KDMSlimShutdown(QWidget
     int dummy, cur;
     if (getBootOptions(&targetList, &dummy, &cur)) {
         QMenu *targets = new QMenu(this);
-        for (int i = 0; i < targetList.size(); i++)
+		for (int i = 0; i < targetList.size(); i++) {
+            if (targetList[i] == "otheroptions")
+                continue;
             (targets->addAction(i == cur ?
                                 i18nc("current option in boot loader",
                                       "%1 (current)", targetList[i]) :
                                 targetList[i]))->setData(i);
+        }
         btnReboot->setDelayedMenu(targets);
         connect(targets, SIGNAL(triggered(QAction *)),
                 SLOT(slotReboot(QAction *)));
