--- kde-workspace-4.8.1.orig/kcontrol/fonts/kxftconfig.h	2012-03-03 00:11:07.000000000 +0200
+++ kde-workspace-4.8.1/kcontrol/fonts/kxftconfig.h	2012-03-10 16:37:42.101939660 +0200
@@ -61,7 +61,7 @@
         SubPixel(Type t, QDomNode &n) : Item(n), type(t) {}
         SubPixel(Type t=None)         : type(t)          {}
 
-        void reset() { Item::reset(); type=None; }
+        void reset() { Item::reset(); type=Rgb; }
 
         Type type;
     };
@@ -91,7 +91,7 @@
         Hint(Style s, QDomNode &n) : Item(n), style(s) {}
         Hint(Style s=NotSet)       : style(s)          {}
 
-        void reset() { Item::reset(); style=NotSet; }
+        void reset() { Item::reset(); style=Full; }
 
         Style style;
     };
