Index: kdebase-workspace-4.5.2/libs/plasmaclock/clockapplet.cpp
===================================================================
--- kdebase-workspace-4.5.2.orig/libs/plasmaclock/clockapplet.cpp
+++ kdebase-workspace-4.5.2/libs/plasmaclock/clockapplet.cpp
@@ -688,17 +688,22 @@ void ClockApplet::updateClipboardMenu()
     sep2->setSeparator(true);
     d->clipboardMenu->addAction(sep2);
 
+    KMenu *otherMenu = new KMenu(i18n("More"));
+
     QStringList calendars = KCalendarSystem::calendarSystems();
     foreach ( const QString &cal, calendars ) {
         if (cal != calendar()->calendarType()) {
             KCalendarSystem *tempCal = KCalendarSystem::create(cal);
             QString text = tempCal->formatDate(dateTime.date(), KLocale::LongDate) + " (" + KCalendarSystem::calendarLabel(cal) + ')';
-            d->clipboardMenu->addAction(text);
+            otherMenu->addAction(text);
             text = tempCal->formatDate(dateTime.date(), KLocale::ShortDate) + " (" + KCalendarSystem::calendarLabel(cal) + ')';
-            d->clipboardMenu->addAction(text);
+            otherMenu->addAction(text);
             delete tempCal;
         }
     }
+
+    d->clipboardMenu->addMenu(otherMenu);
+    connect(otherMenu, SIGNAL(triggered(QAction*)), this, SLOT(copyToClipboard(QAction*)));
 }
 
 void ClockApplet::copyToClipboard(QAction* action)
