diff -Nuar a/plasma/netbook/applets/currentappcontrol/plasma-applet-currentappcontrol.desktop b/plasma/netbook/applets/currentappcontrol/plasma-applet-currentappcontrol.desktop
--- a/plasma/netbook/applets/currentappcontrol/plasma-applet-currentappcontrol.desktop	2013-08-28 19:59:38.295529142 +0300
+++ b/plasma/netbook/applets/currentappcontrol/plasma-applet-currentappcontrol.desktop	2013-06-28 20:10:45.000000000 +0300
@@ -115,7 +115,7 @@
 Comment[ru]=Элементы управления для активного окна
 Comment[si]=සක්‍රිය කවුළුව සඳහා පාලක
 Comment[sk]=Ovládanie pre aktívne okno
-Comment[sl]=Nadzorniki za dejavno okno
+Comment[sl]=Nadzor za aktivno okno
 Comment[sr]=Управљање активним прозором
 Comment[sr@ijekavian]=Управљање активним прозором
 Comment[sr@ijekavianlatin]=Upravljanje aktivnim prozorom
diff -Nuar a/plasma/netbook/containments/sal/plasma-containment-sal.desktop b/plasma/netbook/containments/sal/plasma-containment-sal.desktop
--- a/plasma/netbook/containments/sal/plasma-containment-sal.desktop	2013-08-28 19:59:38.296529135 +0300
+++ b/plasma/netbook/containments/sal/plasma-containment-sal.desktop	2013-06-28 20:10:44.000000000 +0300
@@ -71,7 +71,7 @@
 Comment[es]=Lanzador de aplicaciones a pantalla completa con una interfaz de búsqueda
 Comment[et]=Täisekraanis rakenduste käivitaja otsinguliidesega
 Comment[eu]=Pantaila betetzen duen aplikazio jaurtitzailea bilaketa interfazearekin
-Comment[fi]=Koko näytön sovelluskäynnistin haulla
+Comment[fi]=Koko näytön sovelluskäynnistin hakukäyttöliittymällä
 Comment[fr]=Lanceur d'applications en plein écran avec interface de recherche
 Comment[gl]=Iniciador de programas a pantalla completa con interface de buscas
 Comment[he]=משגר יישומים במסך מלא עם ממשק חיפוש
diff -Nuar a/plasma/netbook/containments/sal/sal.cpp b/plasma/netbook/containments/sal/sal.cpp
--- a/plasma/netbook/containments/sal/sal.cpp	2013-08-28 19:59:38.296529135 +0300
+++ b/plasma/netbook/containments/sal/sal.cpp	2013-06-28 20:10:45.000000000 +0300
@@ -260,7 +260,8 @@
 
     m_resultsView->setIconSize(config().readEntry("ResultsIconSize", (int)KIconLoader::SizeHuge));
 
-    const QString packageManagerName = config().readEntry("PackageManager", "kpackagekit");
+    QString packageManagerName = config().readEntry("PackageManager", "kpackagekit");
+
     if (!packageManagerName.isEmpty()) {
         m_packageManagerService = KService::serviceByDesktopName(packageManagerName);
 
@@ -274,8 +275,6 @@
             connect(addApplicationsAction, SIGNAL(triggered()), this, SLOT(launchPackageManager()));
         }
     }
-
-    m_stripUninitialized = false;
 }
 
 void SearchLaunch::availableScreenRegionChanged()
diff -Nuar a/plasma/netbook/containments/sal/services/plasma-sal-games.desktop b/plasma/netbook/containments/sal/services/plasma-sal-games.desktop
--- a/plasma/netbook/containments/sal/services/plasma-sal-games.desktop	2013-08-28 19:59:38.296529135 +0300
+++ b/plasma/netbook/containments/sal/services/plasma-sal-games.desktop	2013-06-28 20:10:45.000000000 +0300
@@ -146,7 +146,7 @@
 Comment[ru]=Коллекция весёлых игр
 Comment[si]=විනෝද ක්‍රීඩා එකතුවක්
 Comment[sk]=Kolekcia zábavných hier
-Comment[sl]=Zbirka zabavnih iger
+Comment[sl]=Zbirka zabavnih nalog
 Comment[sr]=Збирка игара за разоноду
 Comment[sr@ijekavian]=Збирка игара за разоноду
 Comment[sr@ijekavianlatin]=Zbirka igara za razonodu
diff -Nuar a/plasma/netbook/containments/sal/services/plasma-sal-multimedia.desktop b/plasma/netbook/containments/sal/services/plasma-sal-multimedia.desktop
--- a/plasma/netbook/containments/sal/services/plasma-sal-multimedia.desktop	2013-08-28 19:59:38.296529135 +0300
+++ b/plasma/netbook/containments/sal/services/plasma-sal-multimedia.desktop	2013-06-28 20:10:45.000000000 +0300
@@ -71,7 +71,7 @@
 Name[se]=Multimedia
 Name[si]=බහුමාධ්‍යය
 Name[sk]=Multimédiá
-Name[sl]=Predstavnost
+Name[sl]=Večpredstavnost
 Name[sr]=Мултимедија
 Name[sr@ijekavian]=Мултимедија
 Name[sr@ijekavianlatin]=Multimedija
@@ -143,7 +143,7 @@
 Comment[ru]=Мультимедийные приложения, в том числе проигрыватели аудио и видео
 Comment[si]=බහුමාධ්‍ය භාවිත යෙදවුම්, ශ්‍රව්‍ය දෘශ්‍ය ධාවක වැනි
 Comment[sk]=Multimediálne aplikácie, ako prehrávače videa a zvuku
-Comment[sl]=Predstavnostni programi, npr. predvajalniki zvoka in videa
+Comment[sl]=Večpredstavnostni programi, npr. predvajalniki videa in zvoka
 Comment[sr]=Мултимедијски програми, попут аудио и видео плејера
 Comment[sr@ijekavian]=Мултимедијски програми, попут аудио и видео плејера
 Comment[sr@ijekavianlatin]=Multimedijski programi, poput audio i video plejera
diff -Nuar a/plasma/netbook/containments/sal/services/plasma-sal-system.desktop b/plasma/netbook/containments/sal/services/plasma-sal-system.desktop
--- a/plasma/netbook/containments/sal/services/plasma-sal-system.desktop	2013-08-28 19:59:38.296529135 +0300
+++ b/plasma/netbook/containments/sal/services/plasma-sal-system.desktop	2013-06-28 20:10:44.000000000 +0300
@@ -142,7 +142,7 @@
 Comment[ru]=Системные параметры и установка программ
 Comment[si]=පද්ධති අභිප්‍රේත හා ස්ථාපක වැඩසටහන්
 Comment[sk]=Nastavenie systému a programov
-Comment[sl]=Sistemske možnosti in nastavitveni programi
+Comment[sl]=Sistemske lastnosti in nastavitveni programi
 Comment[sr]=Програми за подешавање система
 Comment[sr@ijekavian]=Програми за подешавање система
 Comment[sr@ijekavianlatin]=Programi za podešavanje sistema
diff -Nuar a/plasma/netbook/shell/nettoolbox/plasma-toolbox-nettoolbox.desktop b/plasma/netbook/shell/nettoolbox/plasma-toolbox-nettoolbox.desktop
--- a/plasma/netbook/shell/nettoolbox/plasma-toolbox-nettoolbox.desktop	2013-08-28 19:59:38.296529135 +0300
+++ b/plasma/netbook/shell/nettoolbox/plasma-toolbox-nettoolbox.desktop	2013-06-28 20:10:45.000000000 +0300
@@ -39,7 +39,7 @@
 Comment[ro]=Trusă de unelte implicită pentru învelișul de netbook
 Comment[ru]=Стандартные инструменты настройки оболочки для нетбуков
 Comment[sk]=Štandardné nástroje pre netbook shell
-Comment[sl]=Privzeti nabor orodij za Namizno lupino Plasma za male prenosnike
+Comment[sl]=Privzeti nabor orodij za delovno okolje Plasma za male prenosnike
 Comment[sr]=Подразумевана алатница за нетбук шкољку
 Comment[sr@ijekavian]=Подразумијевана алатница за нетбук шкољку
 Comment[sr@ijekavianlatin]=Podrazumijevana alatnica za netbuk školjku
