Index: stepmake/stepmake/po-targets.make
diff --git a/stepmake/stepmake/po-targets.make b/stepmake/stepmake/po-targets.make
index 8919dab8239e69f0378a0d6071fc1d249f8bbff0..8a0dd76053f51e8a7ef98f4e9b289a7e24c45218 100644
--- a/stepmake/stepmake/po-targets.make
+++ b/stepmake/stepmake/po-targets.make
@@ -37,10 +37,10 @@ ifneq ($(strip $(ALL_PO_SOURCES)),)
 	 --keyword=_ --keyword=_f --keyword=_i \
 	 $(XGETTEXT_FLAGS) $(ALL_PO_SOURCES)
 endif
-endif
 	sed -i '1,2d' $(po-outdir)/$(package).po
 	sed -i -e 's/^\# This file is distributed.*/$(sed-header)/' $(po-outdir)/$(package).po
 	sed -i -e 's/^\"Content-Type: text\/plain.*/$(sed-content)/' $(po-outdir)/$(package).po
+endif
 
 
 po-update: po
