diff -ru lilypond-2.12.2.orig/lily/relocate.cc lilypond-2.12.2/lily/relocate.cc
--- lilypond-2.12.2.orig/lily/relocate.cc	2009-03-04 15:27:38.000000000 +0000
+++ lilypond-2.12.2/lily/relocate.cc	2009-03-04 15:28:14.000000000 +0000
@@ -276,13 +276,13 @@
   string out;
   while (ptr < start_ptr + len)
     {
-      char *dollar = strchr (ptr, '$');
+      const char *dollar = strchr (ptr, '$');
       
       if (dollar != NULL)
 	{
-	  char *start_var = dollar + 1;
-	  char *end_var = start_var;
-	  char *start_next = end_var;
+	  const char *start_var = dollar + 1;
+	  const char *end_var = start_var;
+	  const char *start_next = end_var;
 	  
 	  out += string (ptr, dollar - ptr);
 	  ptr = dollar;
