--- pidgin-2.10.0/libpurple/protocols/gg/gg.c.orig	2011-08-18 10:51:03.000000000 +0200
+++ pidgin-2.10.0/libpurple/protocols/gg/gg.c	2011-08-21 09:54:41.000000000 +0200
@@ -533,6 +533,8 @@
 		purple_request_fields_get_string(fields, "firstname"));
 	form->nickname = g_strdup(
 		purple_request_fields_get_string(fields, "nickname"));
+	form->uin = g_strdup(
+		purple_request_fields_get_string(fields, "uin"));
 	form->city = g_strdup(
 		purple_request_fields_get_string(fields, "city"));
 	form->birthyear = g_strdup(
@@ -586,6 +588,11 @@
 	purple_request_field_string_set_masked(field, FALSE);
 	purple_request_field_group_add_field(group, field);
 
+	field = purple_request_field_string_new("uin",
+			_("UIN"), NULL, FALSE);
+	purple_request_field_string_set_masked(field, FALSE);
+	purple_request_field_group_add_field(group, field);
+
 	field = purple_request_field_string_new("city",
 			_("City"), NULL, FALSE);
 	purple_request_field_string_set_masked(field, FALSE);
