--- pidgin-2.8.0/libpurple/network.c.reread-resolvconf	2011-06-09 16:30:04.764927571 +0200
+++ pidgin-2.8.0/libpurple/network.c	2011-06-09 16:31:29.356249181 +0200
@@ -874,8 +874,6 @@
 #else
 		case NM_STATE_CONNECTED:
 #endif
-			/* Call res_init in case DNS servers have changed */
-			res_init();
 			/* update STUN IP in case we it changed (theoretically we could
 			   have gone from IPv4 to IPv6, f.ex. or we were previously
 			   offline */
