--- pidgin-2.8.0/libpurple/protocols/gg/gg.c.orig 2011-06-08 03:10:11.000000000 +0200
+++ pidgin-2.8.0/libpurple/protocols/gg/gg.c 2011-06-08 03:13:44.000000000 +0200
@@ -1816,6 +1816,11 @@
		case GG_EVENT_PUBDIR50_SEARCH_REPLY:
			ggp_pubdir_reply_handler(gc, ev->event.pubdir50);
			break;
+ 		case GG_EVENT_DISCONNECT:
+			purple_connection_error_reason (gc,
+					PURPLE_CONNECTION_ERROR_OTHER_ERROR,
+					_("Server Disconnect."));
+			break;
		case GG_EVENT_TYPING_NOTIFICATION:
			ggp_typing_notification_handler(gc, ev->event.typing_notification.uin,
				ev->event.typing_notification.length);
