only in patch2:
unchanged:
Index: XTerm.ad
===================================================================
--- XTerm.ad.orig
+++ XTerm.ad
@@ -210,3 +210,13 @@
 !*allowTcapOps: false
 !*allowTitleOps: false
 !*allowWindowOps: false
+
+! Pardus Defaults:
+*backarrowKeyIsErase:       true
+*ptyInitialErase:           true
+*VT100*eightBitInput:       false
+*VT100*metaSendsEscape:     true
+*VT100*scrollBar:           true
+*VT100*rightScrollBar:      true
+*VT100*utf8Title:           true
+*VT100*modifyFunctionKeys:  0
Index: XTerm-col.ad
===================================================================
--- XTerm-col.ad.orig
+++ XTerm-col.ad
@@ -7,9 +7,9 @@
 *VT100*boldColors: on
 *VT100*dynamicColors: on
 
-! Uncomment this for "white" text on a dark background.
-!*VT100*foreground: gray90
-!*VT100*background: black
+! Set the default text foreground and background colors.
+*VT100*foreground: gray90
+*VT100*background: black
 
 ! - OR -
 ! Uncomment this for black text on a "white" background.
@@ -114,20 +114,32 @@
 *Form.background: AntiqueWhite
 *form.background: AntiqueWhite
 
+*mainMenu*background: AntiqueWhite
+*mainMenu*foreground: gray15
+*vtMenu*background: AntiqueWhite
+*vtMenu*foreground: gray15
+*fontMenu*background: AntiqueWhite
+*fontMenu*foreground: gray15
+*tekMenu*background: AntiqueWhite
+*tekMenu*foreground: gray15
+
 ! The following two sections take advantage of new features in version 7
-! of the Athena widget library.  Comment them out if you have a shallow
-! color depth.
+! of the Athena widget library; see Xaw(7x).
+
+! This section is commented out because Xaw has no way to dynamically size
+! the gradient.  The dimension parameter may need to be adjusted depending
+! on the font used in the menu.
 !*SimpleMenu*backgroundPixmap:   gradient:vertical?dimension=350&start=gray90&end=gray60
 !*SimpleMenu*foreground:         gray15
 
-!*VT100.scrollbar.thumb:         vlines2
-!*VT100.scrollbar.width:         14
-!*VT100.scrollbar.background:    gray60
-!*VT100.scrollbar.foreground:    rgb:a/5/5
-!*VT100.scrollbar.borderWidth:   0
-!*VT100.scrollbar.displayList:\
-!foreground      gray90;\
-!lines           1,-1,-1,-1,-1,1;\
-!foreground      gray60;\
-!lines           -1,0,0,0,0,-1
+*VT100.scrollbar.thumb:         vlines2
+*VT100.scrollbar.width:         14
+*VT100.scrollbar.background:    gray60
+*VT100.scrollbar.foreground:    rgb:a/5/5
+*VT100.scrollbar.borderWidth:   0
+*VT100.scrollbar.displayList:\
+foreground      gray90;\
+lines           1,-1,-1,-1,-1,1;\
+foreground      gray60;\
+lines           -1,0,0,0,0,-1
 #endif
