Index: xterm-255/xterm.man
===================================================================
--- xterm-255.orig/xterm.man
+++ xterm-255/xterm.man
@@ -702,7 +702,7 @@ xterm -e /bin/bash -l -c "my command her
 .RE
 .IP
 Finally, \fB-ls\fP is not completely ignored,
-because \fIxterm\ -ls\ -e\fP does write a \fI/etc/wtmp\fP entry
+because \fIxterm\ -ls\ -e\fP does write a \fI/var/log/wtmp\fP entry
 (if configured to do so),
 whereas \fIxterm\ -e\fP does not.
 .TP 8
@@ -1701,17 +1701,17 @@ Since X11R6, bitmap fonts have been scal
 The font server claims to provide the bold font that \fIxterm\fP requests,
 but the result is not always readable.
 XFree86 provides a feature which can be used to suppress the scaling.
-In the X server's configuration file (e.g., "/etc/X11/XFree86"), you
+In the X server's configuration file (e.g., __xorgcfgdir__/__xorgcfgfil__), you
 can add ":unscaled" to the end of the directory specification for the
 "misc" fonts, which comprise the fixed-pitch fonts that are used by \fIxterm\fP.
 For example
 .RS
-	FontPath	"/usr/lib/X11/fonts/misc/"
+	FontPath	"__fontpath__"
 .RE
 .IP
 would become
 .RS
-	FontPath	"/usr/lib/X11/fonts/misc/:unscaled"
+	FontPath	"__fontpath__/misc/:unscaled"
 .RE
 .IP
 Depending on your configuration, the font server may have its own configuration
@@ -5320,10 +5320,10 @@ may be defined to a nonstandard location
 .SH FILES
 The actual pathnames given may differ on your system.
 .TP 5
-\fI/etc/utmp\fP
+\fI/var/run/utmp\fP
 the system logfile, which records user logins.
 .TP 5
-\fI/etc/wtmp\fP
+\fI/var/log/wtmp\fP
 the system logfile, which records user logins and logouts.
 .TP 5
 .I __apploaddir__/XTerm
Index: xterm-255/minstall.sh
===================================================================
--- xterm-255.orig/minstall.sh
+++ xterm-255/minstall.sh
@@ -109,6 +109,10 @@ WTMP_PATH=$WTMP_PATH/$WTMP_NAME
 X_MANSECT=`man X 2>&1 | tr '\012' '\020' | sed -e 's/^[^0123456789]*\([^) ][^) ]*\).*/\1/'`
 test -z "$X_MANSECT" && X_MANSECT=$suffix
 
+fontpath=/usr/share/fonts
+xorgcfgdir=/etc/X11
+xorgcfgfil=xorg.conf
+X_MANSECT=7
 sed	-e 's%__vendorversion__%"X Window System"%' \
 	-e s%__apploaddir__%$APPS_DIR% \
 	-e s%__mansuffix__%$MY_MANSECT%g \
@@ -117,6 +121,9 @@ sed	-e 's%__vendorversion__%"X Window Sy
 	-e s%/etc/utmp%$UTMP_PATH%g \
 	-e s%fIwtmp'\\%fI'$WTMP_NAME'\\%g' \
 	-e s%/etc/wtmp%$WTMP_PATH%g \
+	-e s%__fontpath__%$fontpath%g \
+	-e s%__xorgcfgdir__%$xorgcfgdir%g \
+	-e s%__xorgcfgfil__%$xorgcfgfil%g \
 	$OLD_FILE >$NEW_FILE
 
 echo "$MINSTALL $OLD_FILE $END_FILE"
