Index: xterm-250/xterm.man
===================================================================
--- xterm-250.orig/xterm.man
+++ xterm-250/xterm.man
@@ -1494,7 +1494,7 @@ The default is \*(``false.\*(''
 .TP
 .B "allowFontOps (\fPclass\fB AllowFontOps)"
 Specifies whether control sequences that set/query the font should be allowed.
-The default is \*(``true.\*(''
+The default is \*(``false.\*(''
 .TP 8
 .B "allowSendEvents (\fPclass\fB AllowSendEvents)"
 Specifies whether or not synthetic key and button events (generated using
Index: xterm-250/main.h
===================================================================
--- xterm-250.orig/main.h
+++ xterm-250/main.h
@@ -92,7 +92,7 @@
 #endif
 
 #ifndef DEF_ALLOW_FONT
-#define DEF_ALLOW_FONT		True
+#define DEF_ALLOW_FONT		False
 #endif
 
 #ifndef DEF_ALLOW_TCAP
