diff -up qt-mobility-opensource-src-1.2.0/features/deploy.pri.pkgconfig qt-mobility-opensource-src-1.2.0/features/deploy.pri
--- qt-mobility-opensource-src-1.2.0/features/deploy.pri.pkgconfig	2011-09-22 01:59:10.000000000 -0500
+++ qt-mobility-opensource-src-1.2.0/features/deploy.pri	2011-10-29 11:43:52.340550783 -0500
@@ -4,7 +4,7 @@ headers.path = $$QT_MOBILITY_INCLUDE/$$T
 contains(TEMPLATE,.*lib) {
     target.path=$$QT_MOBILITY_LIB
 
-    maemo5|maemo6|meego {
+    linux-*|maemo5|maemo6|meego {
         CONFIG += create_pc create_prl
         QMAKE_PKGCONFIG_NAME = lib$$TARGET
         QMAKE_PKGCONFIG_DESTDIR = pkgconfig
diff -up qt-mobility-opensource-src-1.2.0/src/location/location.pro.pkgconfig qt-mobility-opensource-src-1.2.0/src/location/location.pro
