diff -Nur cdogs-sdl-0.4-old/src/grafx.c cdogs-sdl-0.4/src/grafx.c
--- cdogs-sdl-0.4-old/src/grafx.c	2007-07-09 16:15:05.060226159 +0300
+++ cdogs-sdl-0.4/src/grafx.c	2007-07-09 16:23:05.681160672 +0300
@@ -109,7 +109,7 @@
 int hints[HINT_END] = {
 	0,		// HINT_FULLSCREEN
 	1,		// HINT_WINDOW
-	1,		// HINT_SCALEFACTOR
+	2,		// HINT_SCALEFACTOR
 	320,		// HINT_WIDTH
 	240,		// HINT_HEIGHT
 	0		// HINT_FORCEMODE
