diff -Nur libsmi-0.4.8-old/test/Makefile.am libsmi-0.4.8/test/Makefile.am
--- libsmi-0.4.8-old/test/Makefile.am	2010-05-07 20:13:09.165413184 +0300
+++ libsmi-0.4.8/test/Makefile.am	2010-05-07 20:13:52.566415974 +0300
@@ -20,7 +20,6 @@
 TESTS			= smilint-smiv2.test \
 			  smidump-smiv1.test \
 			  smidump-smiv2.test \
-			  smidump-sming.test \
 			  smidump-yang.test \
 			  smidump-mosy.test \
 			  smidump-tree.test \
