--- airstrike-0.99+1.0pre6a/src/sprite_types/biplane.c	2003-01-18 20:48:05.000000000 +0100
+++ airstrike-0.99+1.0pre6a/src/sprite_types/biplane.c	2007-10-10 01:26:25.000000000 +0200
@@ -198,9 +198,10 @@
 	}
       break;
     case SIGNAL_NUM0: /* create bomb */
-      if (sprite_timer_finished(((struct biplane*)s)->bomb_timer) &&	  
-	  (!(s->state & BIPLANE_CRASHING)))
+      if (sprite_timer_finished(((struct biplane*)s)->bomb_timer) &&
+	  (!(s->state & BIPLANE_CRASHING)) && (((struct biplane*)s)->nr_bombs > 0))
 	{
+	  ((struct biplane*)s)->nr_bombs--;
 	  p = sprite_create(&bomb);
 	  p->anim_p = s->anim_p;
 	  ((mech_sprite_t *)p)->angle = ((mech_sprite_t *)s)->angle;
