--- airstrike-0.99+1.0pre6a/src/mech.c	2003-01-18 16:48:05.000000000 -0300
+++ airstrike-0.99+1.0pre6a/src/mech.c	2007-01-14 16:04:06.000000000 -0300
@@ -335,11 +335,10 @@
     n[2]*((mech_sprite_t *)s1)->ang_vel +
     n[3]*((mech_sprite_t *)s2)->ang_vel;
 
-  if (a > 0)
-    {
-      b = n[0]*delta[0] + n[1]*delta[1] + n[2]*delta[2] 
+  b = n[0]*delta[0] + n[1]*delta[1] + n[2]*delta[2] 
 	- n[0]*delta[3] - n[1]*delta[4] + n[3]*delta[5];
-      
+  if ( (a > 0) && (abs(b) > 0) )
+    {
       c = -1.4*a/b;
       
       s1->vel[0] += c*delta[0];
@@ -479,10 +478,9 @@
   a = n[0]*s1->vel[0] + n[1]*s1->vel[1] + 
     n[2]*((mech_sprite_t *)s1)->ang_vel;
 
-  if (a > 0)
+  b = n[0]*delta[0] + n[1]*delta[1] + n[2]*delta[2];
+  if ( (a > 0) && (abs(b) > 0) )
     {
-      b = n[0]*delta[0] + n[1]*delta[1] + n[2]*delta[2];
-      
       c = -1.2*a/b;
       
       s1->vel[0] += c*delta[0];
