diff -urN old/src/airstrike.c new/src/airstrike.c
--- old/src/airstrike.c	2007-09-13 17:57:23.000000000 +0200
+++ new/src/airstrike.c	2007-09-13 11:11:33.000000000 +0200
@@ -92,8 +92,8 @@
   player_sprite_type[1] = &biplane;
   player_startpos[0][0] = 700;
   player_startpos[0][1] = 500;
-  player_startpos[1][0] = 5;
-  player_startpos[1][1] = 300;
+  player_startpos[1][0] = 100;
+  player_startpos[1][1] = 500;
   player_points[0] = max_points;
   player_points[1] = max_points;
 
