--- airstrike-0.99+1.0pre6a/src/config.c	2003-01-18 20:48:05.000000000 +0100
+++ airstrike-0.99+1.0pre6a/src/config.c	2005-01-14 01:42:55.000000000 +0100
@@ -21,7 +21,7 @@ int config_setup()
     }
   if (getenv("HOME"))
     {
-      sprintf(cbuf,"%s/%s",getenv("HOME"),USER_CONFIG_FILE);      
+      snprintf(cbuf,sizeof(cbuf),"%s/%s",getenv("HOME"),USER_CONFIG_FILE);
       cfile = fopen(cbuf,"r");
       if (cfile)
 	{
