Index: qoauth-1.0.1-src/src/src.pro
===================================================================
--- qoauth-1.0.1-src.orig/src/src.pro
+++ qoauth-1.0.1-src/src/src.pro
@@ -62,7 +62,7 @@ else:unix {
     pkgconfig.files = qoauth.pc
     
     contains(QMAKE_HOST.arch, x86_64) {
-      target.path = $${INSTALL_PREFIX}/lib64
+      target.path = $${INSTALL_PREFIX}/lib
     } else {
       target.path = $${INSTALL_PREFIX}/lib
     }
