From f1130f3409f892391687095a463c143529e91cb1 Mon Sep 17 00:00:00 2001
From: Michael Meeks <michael.meeks@novell.com>
Date: Fri, 24 Sep 2010 09:46:40 +0100
Subject: [PATCH 02/34] try harder with taskstats.

---
 TODO                  |    6 ++++++
 collector/collector.c |    2 +-
 2 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/TODO b/TODO
index 5f2bde2..742e011 100644
--- a/TODO
+++ b/TODO
@@ -73,3 +73,9 @@ Process taskstats in advance
 
 Notes:
 	+ SUSE initrd debugging; use "'shell=1'" - is dead good ...
+
+
+
+#18a303 - Tango / Green 
+
+#2da31b  - to same saturation
diff --git a/collector/collector.c b/collector/collector.c
index 19312db..d66ddf8 100644
--- a/collector/collector.c
+++ b/collector/collector.c
@@ -481,7 +481,7 @@ init_taskstat (void)
 
 	netlink_taskstats_id = get_family_id (netlink_socket);
 
-	return netlink_taskstats_id >= 0;
+	return netlink_taskstats_id > 0;
 error:
 	if (netlink_socket >= 0)
 		close (netlink_socket);
-- 
1.7.3.1

