From 41dbd2102e5fd17f078b82f83f60f1686d3affba Mon Sep 17 00:00:00 2001
From: Riccardo Magliocchetti <riccardo.magliocchetti@gmail.com>
Date: Thu, 4 Nov 2010 21:40:35 +0100
Subject: [PATCH 15/34] Fix cli interface after bf95dee1ab84e490e278c5d187506d9f4566dd41

---
 pybootchartgui/main.py.in |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/pybootchartgui/main.py.in b/pybootchartgui/main.py.in
index d1387f5..42b956c 100644
--- a/pybootchartgui/main.py.in
+++ b/pybootchartgui/main.py.in
@@ -114,10 +114,8 @@ def main(argv=None):
 			print "No path given, trying /var/log/bootchart.tgz"
 			args = [ "/var/log/bootchart.tgz" ]
 
+		res = parsing.Trace(writer, args, options)
 
-		res = parsing.parse(writer, args, options.prune,
-				    options.crop_after, options.annotate)
-		
 		if options.interactive or options.output == None:
 			gui.show(res, options)
 		elif options.boottime:
-- 
1.7.3.1

