From 2ea1b6272fa75f4827d81195858ad89b165d63c8 Mon Sep 17 00:00:00 2001
From: Riccardo Magliocchetti <riccardo.magliocchetti@gmail.com>
Date: Thu, 4 Nov 2010 21:44:05 +0100
Subject: [PATCH 16/34] Plug some small leaks in the collector

---
 collector/collector.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/collector/collector.c b/collector/collector.c
index d66ddf8..57330a6 100644
--- a/collector/collector.c
+++ b/collector/collector.c
@@ -851,6 +851,7 @@ int main (int argc, char *argv[])
 				 path, strerror (errno));
 			exit (1);
 		}
+		free (path);
 	}
 
 	stat_file = buffer_file_new (&map, "proc_stat.log");
-- 
1.7.3.1

