From 0acc28ba1cbd45c470498e0a1b8d0244800df16c Mon Sep 17 00:00:00 2001
From: Riccardo Magliocchetti <riccardo.magliocchetti@gmail.com>
Date: Tue, 9 Nov 2010 21:57:27 +0100
Subject: [PATCH 18/34] Add enlightenment to the list of processes we have to wait for.

---
 bootchartd |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/bootchartd b/bootchartd
index d5d5680..82f8dc0 100755
--- a/bootchartd
+++ b/bootchartd
@@ -79,7 +79,7 @@ wait_boot()
 	local runlevel=$( sed -n '/^ *#/d; /^$/d; s/.*:\(.*\):initdefault:.*/\1/g; p; q' /etc/inittab )
 
 	# The processes we have to wait for
-	local exit_proc="kdm_greet xterm konsole gnome-terminal metacity mutter compiz ldm icewm-session"
+	local exit_proc="kdm_greet xterm konsole gnome-terminal metacity mutter compiz ldm icewm-session enlightenment"
 
 	# Wait for /proc first - without it we have issues
 	while [ ! -e /proc/cmdline ]; do
-- 
1.7.3.1

