From 14dbe616eca066630e378d7a41197797307fefc8 Mon Sep 17 00:00:00 2001
From: Riccardo Magliocchetti <riccardo.magliocchetti@gmail.com>
Date: Tue, 9 Nov 2010 22:36:13 +0100
Subject: [PATCH 21/34] Update NEWS.

---
 NEWS |   15 +++++++++++++++
 1 files changed, 15 insertions(+), 0 deletions(-)

diff --git a/NEWS b/NEWS
index 6182619..6618f3f 100644
--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,18 @@
+bootchart2 UNRELEASED:
+	+ bootchartd
+		+ better check for initrd (Riccardo Magliocchetti)
+		+ code cleanup (Riccardo)
+		+ make the list of processes we are waiting for editable
+		  in config file by EXIT_PROC (Riccardo)
+	+ collector
+		+ try harder with taskstats (Michael)
+		+ plug some small leaks (Riccardo)
+	+ pybootchartgui
+		+ report bootchart version in cli interface (Michael)
+		+ improve rendering performance (Michael)
+		+ GUI improvements (Michael)
+		+ lot of cleanups (Michael)
+
 bootchart2 0.12.4:
 	+ bootchartd
 		+ reduce overhead caused by pidof (Riccardo Magliocchetti)
-- 
1.7.3.1

