From cda1709988d167b25156e7e98aa9fd0a34ec184e Mon Sep 17 00:00:00 2001
From: Riccardo Magliocchetti <riccardo.magliocchetti@gmail.com>
Date: Tue, 9 Nov 2010 23:30:46 +0100
Subject: [PATCH 23/34] Update NEWS.

---
 NEWS |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/NEWS b/NEWS
index 6618f3f..afa8177 100644
--- a/NEWS
+++ b/NEWS
@@ -4,6 +4,7 @@ bootchart2 UNRELEASED:
 		+ code cleanup (Riccardo)
 		+ make the list of processes we are waiting for editable
 		  in config file by EXIT_PROC (Riccardo)
+		+ fix parsing of cmdline for alternative init system (Riccardo)
 	+ collector
 		+ try harder with taskstats (Michael)
 		+ plug some small leaks (Riccardo)
-- 
1.7.3.1

