From 402cf1092d36a9406eba4ed8efe413cb9f23d5ee Mon Sep 17 00:00:00 2001
From: Michael Meeks <michael.meeks@novell.com>
Date: Fri, 19 Nov 2010 14:52:06 +0000
Subject: [PATCH 25/34] add Harald.

---
 MAINTAINERS |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/MAINTAINERS b/MAINTAINERS
index c4ac928..c65e131 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -1,2 +1,3 @@
 Riccardo Magliocchetti <riccardo.magliocchetti@gmail.com>
 Michael Meeks <michael.meeks@novell.com>
+Harald Hoyer <harald@redhat.com>
-- 
1.7.3.1

