From daa15279e8638ff71a77e7d06ac09322cabf3fc6 Mon Sep 17 00:00:00 2001
From: Michael Meeks <michael.meeks@novell.com>
Date: Fri, 19 Nov 2010 15:04:34 +0000
Subject: [PATCH 26/34] mention freenode channel

---
 README |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/README b/README
index 19f2388..b6d89e5 100644
--- a/README
+++ b/README
@@ -73,4 +73,4 @@ hierarchy.
 			   --------------
 
 	For potential future work, see the TODO, and/or mail the authors,
-all patches welcome.
+wander over to #bootchart2 on irc.freenode.net. All patches most welcome.
-- 
1.7.3.1

