From eccac61a5f043c6d0ee9cdf3fb16af496e848747 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 18 Nov 2010 15:26:17 +0100
Subject: [PATCH 28/34] bootchartd: handle /init and /linuxrc in initramfs

---
 bootchartd |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/bootchartd b/bootchartd
index 1b032c3..fc02e78 100755
--- a/bootchartd
+++ b/bootchartd
@@ -139,9 +139,13 @@ if [ $$ -eq 1 ]; then
 	INIT_PROCESS="yes"
 	echo "Starting bootchart logging"
 
+	init="/sbin/init"
+
 	# Are we running in the initrd ?
 	if [ -x /init -o -x /linuxrc ]; then
 		IN_INITRD="yes"
+		[ -x /linuxrc ] && init="/linuxrc"
+		[ -x /init ] && init="/init"
 		start &
 	else # running inside the main system
 		echo "bootchart: no initrd used; starting"
@@ -155,7 +159,6 @@ if [ $$ -eq 1 ]; then
 
 	# Optionally, an alternative init(1) process may be specified using
 	# the kernel command line (e.g. "bootchart_init=/sbin/initng")
-	init="/sbin/init"
 	for i in $@; do
 		if [ "${i%%=*}" = "bootchart_init" ]; then
 			init="${i#*=}"
-- 
1.7.3.1

