From a3228ad609a93a6d68261b585fb53ee22d5ee0ce Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 19 Nov 2010 15:09:17 +0100
Subject: [PATCH 29/34] bootchartd: bail out, if already running

---
 bootchartd |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/bootchartd b/bootchartd
index fc02e78..b988b51 100755
--- a/bootchartd
+++ b/bootchartd
@@ -61,6 +61,8 @@ start()
 
 	# Otherwise, manually launched to profile something
 	else
+		# bail out, if already running
+		pidof bootchart-collector && exit 0
 #		echo "bootchartd started manually" >> kmsg
 		/lib/bootchart/bootchart-collector -r $SAMPLE_HZ &
 
-- 
1.7.3.1

