From 8d4f31dbe79949d1c9bb3af10d50ffff052b951b Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 19 Nov 2010 17:17:20 +0100
Subject: [PATCH 30/34] Makefile: install systemd service files

---
 Makefile |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index 9e46676..6b4ef56 100644
--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,7 @@ BINDIR ?= /usr/bin
 PY_LIBDIR ?= /usr/lib/python2.6
 PY_SITEDIR ?= $(PY_LIBDIR)/site-packages
 LIBC_A_PATH = /usr/lib
-
+SYSTEMD_UNIT_DIR = /lib/systemd/system
 COLLECTOR = \
 	collector/collector.o \
 	collector/output.o \
@@ -44,6 +44,12 @@ install-collector: all install-chroot
 	install -m 644 -D bootchartd.conf $(DESTDIR)/etc/bootchartd.conf
 	install -m 755 -D bootchart-collector $(DESTDIR)/lib/bootchart/bootchart-collector
 
+install-service:
+	install -m 0644 bootchart.service \
+	       bootchart-done.service \
+	       bootchart-done.timer \
+	       $(SYSTEMD_UNIT_DIR)      
+
 install: all py-install-compile install-collector
 
 clean:
-- 
1.7.3.1

