From 01a0130c85048c4305da2c7c064a632cd4990798 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 19 Nov 2010 17:19:37 +0100
Subject: [PATCH 31/34] pybootchartgui/main: delay "import gui" until needed

---
 pybootchartgui/main.py.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/pybootchartgui/main.py.in b/pybootchartgui/main.py.in
index 42b956c..97c8700 100644
--- a/pybootchartgui/main.py.in
+++ b/pybootchartgui/main.py.in
@@ -21,7 +21,6 @@ import os
 import optparse
 
 import parsing
-import gui
 import batch
 
 def _mk_options_parser():
@@ -117,6 +116,7 @@ def main(argv=None):
 		res = parsing.Trace(writer, args, options)
 
 		if options.interactive or options.output == None:
+			import gui
 			gui.show(res, options)
 		elif options.boottime:
 			import math
-- 
1.7.3.1

