From 8816bddc438f2120ee3d23475c9f8b037fb9fa05 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 19 Nov 2010 17:25:23 +0100
Subject: [PATCH 32/34] Makefile: do not python compile if NO_PYTHON_COMPILE is set

---
 Makefile |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 6b4ef56..7147028 100644
--- a/Makefile
+++ b/Makefile
@@ -32,9 +32,9 @@ py-install-compile: pybootchartgui/main.py
 	install -d $(DESTDIR)$(PY_SITEDIR)/pybootchartgui
 	cp pybootchartgui/*.py $(DESTDIR)$(PY_SITEDIR)/pybootchartgui
 	install -D -m 755 pybootchartgui.py $(DESTDIR)$(BINDIR)/pybootchartgui
-	cd $(DESTDIR)$(PY_SITEDIR)/pybootchartgui ; \
+	[ -z "$(NO_PYTHON_COMPILE)" ] && ( cd $(DESTDIR)$(PY_SITEDIR)/pybootchartgui ; \
 		python $(PY_LIBDIR)/py_compile.py *.py ; \
-		PYTHONOPTIMIZE=1 python $(PY_LIBDIR)/py_compile.py *.py
+		PYTHONOPTIMIZE=1 python $(PY_LIBDIR)/py_compile.py *.py ); :
 
 install-chroot:
 	install -d $(DESTDIR)/lib/bootchart/tmpfs
-- 
1.7.3.1

