From f0fcead42b89407c2c48fa43a41078d089508ab1 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 19 Nov 2010 17:36:42 +0100
Subject: [PATCH 33/34] NEWS: update

---
 NEWS |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/NEWS b/NEWS
index afa8177..593a6b3 100644
--- a/NEWS
+++ b/NEWS
@@ -5,6 +5,8 @@ bootchart2 UNRELEASED:
 		+ make the list of processes we are waiting for editable
 		  in config file by EXIT_PROC (Riccardo)
 		+ fix parsing of cmdline for alternative init system (Riccardo)
+		+ fixed calling init in initramfs (Harald)
+		+ exit 0 for start, if the collector is already running (Harald)
 	+ collector
 		+ try harder with taskstats (Michael)
 		+ plug some small leaks (Riccardo)
@@ -13,6 +15,10 @@ bootchart2 UNRELEASED:
 		+ improve rendering performance (Michael)
 		+ GUI improvements (Michael)
 		+ lot of cleanups (Michael)
+	+ Makefile
+		+ do not python compile if NO_PYTHON_COMPILE is set (Harald)
+	+ systemd service files
+		+ added them and install (Harald)
 
 bootchart2 0.12.4:
 	+ bootchartd
-- 
1.7.3.1

