From c94c5e8cedca6e22814826347d6403244ff7c17d Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 19 Nov 2010 17:45:32 +0100
Subject: [PATCH 34/34] Makefile: install service files for target "all" and mkdir

---
 Makefile |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index 7147028..00b860b 100644
--- a/Makefile
+++ b/Makefile
@@ -45,12 +45,13 @@ install-collector: all install-chroot
 	install -m 755 -D bootchart-collector $(DESTDIR)/lib/bootchart/bootchart-collector
 
 install-service:
+	mkdir -p $(SYSTEMD_UNIT_DIR)
 	install -m 0644 bootchart.service \
 	       bootchart-done.service \
 	       bootchart-done.timer \
 	       $(SYSTEMD_UNIT_DIR)      
 
-install: all py-install-compile install-collector
+install: all py-install-compile install-collector install-service
 
 clean:
 	-rm -f bootchart-collector bootchart-collector-dynamic \
-- 
1.7.3.1

