From 078506ea1bdd34fb59c2a35283f69652e5e052f8 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 19 Nov 2010 18:11:19 +0100
Subject: [PATCH 35/40] README: added HOWTO for initramfs with dracut

---
 README |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/README b/README
index b6d89e5..51da2f6 100644
--- a/README
+++ b/README
@@ -35,6 +35,11 @@ configured with CONFIG_PROC_EVENTS=y and CONFIG_TASKSTATS=y, without
 these we are slower, less accurate, and produce an uglier task 
 hierarchy.
 
+	If you want to start bootchart2 in a dracut (version >= 008)
+initramfs, you have to change "init=/sbin/bootchartd" to 
+"rdinit=/sbin/bootchartd" and regenerate the initramfs with bootchart support
+with "# dracut -f -a bootchart".
+
 			    Why bootchart2 ?
 			   ------------------
 
-- 
1.7.3.1

