From 067f10ab830209a52f79dc4780f863c25e5136d3 Mon Sep 17 00:00:00 2001
From: Riccardo Magliocchetti <riccardo.magliocchetti@gmail.com>
Date: Fri, 19 Nov 2010 23:46:11 +0100
Subject: [PATCH 36/40] bootchartd: fixup whitespace

---
 bootchartd |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/bootchartd b/bootchartd
index b988b51..ee5ecc6 100755
--- a/bootchartd
+++ b/bootchartd
@@ -46,9 +46,9 @@ CONF="/etc/bootchartd.conf"
 if [ -f $PWD/bootchartd.conf ]; then
 	. $PWD/bootchartd.conf
 elif [ -f $CONF ]; then
-        . $CONF
+	. $CONF
 else
-        echo "$CONF missing"
+	echo "$CONF missing"
 fi
 
 # Start the boot logger.
-- 
1.7.3.1

