From 9458c9a3bc449400b23a5beabdf86223b31da0d5 Mon Sep 17 00:00:00 2001
From: Michael Meeks <michael.meeks@novell.com>
Date: Mon, 22 Nov 2010 10:08:05 +0000
Subject: [PATCH 39/40] clean trailing whitespace, and redundant //

---
 Makefile |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index a720976..0480b82 100644
--- a/Makefile
+++ b/Makefile
@@ -45,11 +45,11 @@ install-collector: all install-chroot
 	install -m 755 -D bootchart-collector $(DESTDIR)/lib/bootchart/bootchart-collector
 
 install-service:
-	mkdir -p $(DESTDIR)/$(SYSTEMD_UNIT_DIR)
+	mkdir -p $(DESTDIR)$(SYSTEMD_UNIT_DIR)
 	install -m 0644 bootchart.service \
 	       bootchart-done.service \
 	       bootchart-done.timer \
-	       $(DESTDIR)/$(SYSTEMD_UNIT_DIR)      
+	       $(DESTDIR)$(SYSTEMD_UNIT_DIR)
 
 install: all py-install-compile install-collector install-service
 
-- 
1.7.3.1

