From 4abed0eeda87ba6d49bd54723d4c86e0a2b8f163 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 22 Nov 2010 15:32:49 +0100
Subject: [PATCH 40/40] collector/tasks-netlink.c: only bail out on poll error

---
 collector/tasks-netlink.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/collector/tasks-netlink.c b/collector/tasks-netlink.c
index f947297..b6a2dd9 100644
--- a/collector/tasks-netlink.c
+++ b/collector/tasks-netlink.c
@@ -480,7 +480,7 @@ pid_scanner_new_netlink (PidScanEventFn event_fn, void *user_data)
 
 		pr.fd = nls->socket;
 		pr.events = POLLIN;
-		if (poll (&pr, 1, 50 /* ms */) || !(pr.revents & POLLIN)) {
+		if ((poll (&pr, 1, 50 /* ms */) <= 0) || (!(pr.revents & POLLIN))) {
 			fprintf (stderr, "No PROC_EVENTs present\n");
 			goto close_and_exit;
 		}
-- 
1.7.3.1

