Subject: configure.ac: Change AC_LANG_SOURCE() into AC_LANG_PROGRAM()

---
 configure.ac |    8 ++------
 1 files changed, 2 insertions(+), 6 deletions(-)

Index: acct-6.6.1/configure.ac
===================================================================
--- acct-6.6.1.orig/configure.ac        2013-03-09 00:19:37.000000000 +0100
+++ acct-6.6.1/configure.ac     2013-12-21 23:54:54.896897230 +0100
@@ -231,7 +231,7 @@ AC_CHECK_HEADER(sys/acct.h,
 dnl
 dnl find out where utmp/pacct are stored
 dnl
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
+AC_RUN_IFELSE([AC_LANG_PROGRAM([[
 #include <stdio.h>
 #include <stdlib.h>
 #include <sys/types.h>
@@ -310,9 +310,7 @@ AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #    endif
 #  endif
 #endif
-
-main ()
-{
+]], [[
   FILE *fp;
   fp = fopen ("locs", "w");
   fprintf (fp, "WTMP_FILE_LOC=%s\n", WTMP_FILE);
@@ -320,8 +318,6 @@ main ()
   fprintf (fp, "SAVACCT_FILE_LOC=%s\n", SAVACCT_FILE);
   fprintf (fp, "USRACCT_FILE_LOC=%s\n", USRACCT_FILE);
   fclose (fp);
-  exit(0);
-}
 ]])],[. ./locs; rm locs],[echo "Error -- could not locate your wtmp and acct files."; exit 1],[echo "Sorry -- you cannot cross-compile this package (FIXME)."; exit 1 ])
 
 dnl types
