Index: build-5.2/src/mainwindow.cpp
===================================================================
--- build-5.2.orig/src/mainwindow.cpp
+++ build-5.2/src/mainwindow.cpp
@@ -23,6 +23,9 @@
 
 #include "kde-qt.h"
 
+#include <kio/netaccess.h>
+#include <KUrl>
+
 #include "libchmfile.h"
 #include "libchmfileimpl.h"
 #include "libchmurlfactory.h"
@@ -151,8 +154,22 @@ bool MainWindow::loadFile ( const QStrin
 		fileName.remove( 0, 7 );
 			
 	LCHMFile * new_chmfile = new LCHMFile();
+    const KUrl url = KIO::NetAccess::mostLocalUrl(fileName, this);
+    QString localFile;
 	
 	if ( new_chmfile->loadFile( fileName ) )
+    if ( url.isLocalFile() )
+    {
+        localFile = url.path();
+    }
+    else
+    {
+        if ( !KIO::NetAccess::download( url, localFile, this ) )
+        {
+            localFile = fileName;
+        }
+    }
+    if ( new_chmfile->loadFile( localFile ) )
 	{
 		// The new file is opened, so we can close the old one
 		if ( m_chmFile )
@@ -171,7 +188,7 @@ bool MainWindow::loadFile ( const QStrin
 
 		// Make the file name absolute; we'll need it later
 		QDir qd;
-		qd.setPath (fileName);
+		qd.setPath (localFile);
 		m_chmFilename = qd.absolutePath();
 		
 		// Qt's 'dirname' does not work well
@@ -452,6 +469,8 @@ void MainWindow::closeFile( )
 	}
 	
 	pConfig->save();
+
+    KIO::NetAccess::removeTempFile( m_chmFilename );
 }
 
 
