diff -up kchmviewer-4.0/CMakeLists.txt.no_msits kchmviewer-4.0/CMakeLists.txt
--- kchmviewer-4.0/CMakeLists.txt.no_msits	2008-12-01 20:31:22.000000000 -0600
+++ kchmviewer-4.0/CMakeLists.txt	2009-04-27 10:23:02.371525695 -0500
@@ -18,6 +18,6 @@ ADD_DEFINITIONS( -DUSE_KDE )
 
 # Run in those subdirectories
 ADD_SUBDIRECTORY( lib/libchmfile )
-ADD_SUBDIRECTORY( lib/kio-msits )
+#ADD_SUBDIRECTORY( lib/kio-msits )
 ADD_SUBDIRECTORY( src )
 ADD_SUBDIRECTORY( po )
