--- /dev/null	2009-07-09 15:46:44.395160073 +0300
+++ buildscript	2009-07-09 15:59:36.644513933 +0300
@@ -0,0 +1,16 @@
+i = 1
+while (i < $argc)
+  Open($argv[i], 1)
+  fontname = StrJoin(StrSplit($fontname, "O"), "")
+  familyname = StrJoin(StrSplit($familyname, " O"), "")
+  fullname = StrJoin(StrSplit($fullname, " O"), "")
+  SetFontNames(fontname, familyname, fullname)
+  ScaleToEm(2048)
+  RoundToInt()
+  SetFontOrder(2)
+  SelectAll()
+  AutoInstr()
+  Generate($fontname + ".ttf")
+  Close()
+  i++
+endloop
