diff -up blobAndConquer-1.0/src/cplusplus/CPak.cpp.orig blobAndConquer-1.0/src/cplusplus/CPak.cpp
--- blobAndConquer-1.0/src/cplusplus/CPak.cpp.orig	2008-06-21 15:00:34.000000000 +0200
+++ blobAndConquer-1.0/src/cplusplus/CPak.cpp	2009-03-31 16:01:35.000000000 +0200
@@ -76,7 +76,7 @@ void Pak::showPakErrorAndExit()
 
 void Pak::initPakFile()
 {
-	uLongf listPos;
+	int listPos;
 
 	debug(("Pak : Filename set to %s\n", PAKFULLPATH));
 
