diff -up blobAndConquer-1.0/data/gameDefs/defines.h~ blobAndConquer-1.0/data/gameDefs/defines.h
--- blobAndConquer-1.0/data/gameDefs/defines.h~	2008-09-28 20:10:53.000000000 +0200
+++ blobAndConquer-1.0/data/gameDefs/defines.h	2008-09-28 20:10:53.000000000 +0200
@@ -377,7 +377,7 @@ enum {
 	SND_GALDOV_CHARGE,
 	SND_GALDOV_FIRE,
 	SND_PLANT_BOMB,
-	SND_POP
+        SND_POP
 };
 
 /* ############### weapons ############ */
