--- makefile.old	2009-11-10 01:27:35.000000000 +0200
+++ makefile	2009-11-10 01:29:43.000000000 +0200
@@ -7,23 +7,17 @@
 
 VERSION = 1.10
 RELEASE = 1
-USEPAK = 0
+USEPAK = 1
 DEV = 0
 
 PREFIX = $(DESTDIR)/usr
-BINDIR = $(PREFIX)/games/
-DATADIR = $(PREFIX)/share/games/blobAndConquer/
-DOCDIR = $(PREFIX)/share/doc/$(PROG)/
-ICONDIR = $(PREFIX)/share/icons/
+ICONDIR = $(PREFIX)/share/icons/hicolor/
 MENUDIR = $(PREFIX)/share/applications/
 LOCALEDIR = $(PREFIX)/share/locale/
 
 CXXFLAGS = $(CFLAGS)
-CXXFLAGS += -Wformat=2 -Wformat-security -Wstrict-aliasing=2
-CXXFLAGS += -Wmissing-format-attribute -Wmissing-noreturn
-CXXFLAGS += -Wdisabled-optimization
 CXXFLAGS += `sdl-config --cflags` -DVERSION=$(VERSION) -DRELEASE=$(RELEASE) -DUSEPAK=$(USEPAK) -DDEV=$(DEV)
-CXXFLAGS += -DPAKNAME=\"$(PAKNAME)\" -DPAKLOCATION=\"$(DATADIR)\" -DGAMEPLAYMANUAL=\"$(DOCDIR)/index.html\" -Wall -pedantic
+CXXFLAGS += -DPAKNAME=\"$(PAKNAME)\" -DPAKLOCATION=\"$(DATADIR)\" -DGAMEPLAYMANUAL=\"$(DOCDIR)/index.html\"
 CXXFLAGS += -DLOCALEDIR=\"$(LOCALEDIR)\" -g
 LIBPATH = -L/usr/X11/lib -L/usr/X11R6/lib
 LIBS = -lGL -lGLU `sdl-config --libs` -lSDL_mixer -lSDL_image -lSDL_ttf -lz
@@ -113,9 +107,7 @@
 	mkdir -p $(BINDIR)
 	mkdir -p $(DATADIR)
 	mkdir -p $(DOCDIR)
-	mkdir -p $(ICONDIR)/16x16/apps
 	mkdir -p $(ICONDIR)/32x32/apps
-	mkdir -p $(ICONDIR)/64x64/apps
 	mkdir -p $(MENUDIR)
 
 	install -m 755 $(PROG) $(BINDIR)/$(PROG)
