From 7aafaf56fbca97f36c775462c1ceea3e03700c42 Mon Sep 17 00:00:00 2001
From: Alexandr Shadchin <alexandr.shadchin@gmail.com>
Date: Sat, 18 Aug 2012 20:53:34 +0600
Subject: [PATCH 1/2] Fix check function in VESASaveRestore

Signed-off-by: Alexandr Shadchin <Alexandr.Shadchin@gmail.com>
Reviewed-by: Peter Hutterer <peter.hutterer@who-t.net>
Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 src/vesa.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/vesa.c b/src/vesa.c
index 8ac77da..11cd26c 100644
--- a/src/vesa.c
+++ b/src/vesa.c
@@ -1626,7 +1626,7 @@ VESASaveRestore(ScrnInfoPtr pScrn, vbeSaveRestoreFunction function)
 {
     VESAPtr pVesa;
 
-    if (MODE_QUERY < 0 || function > MODE_RESTORE)
+    if (function < MODE_QUERY || function > MODE_RESTORE)
 	return (FALSE);
 
     pVesa = VESAGetRec(pScrn);
-- 
1.8.0.2

