diff -up xf86-video-vesa-2.3.0/src/vesa.c.jx xf86-video-vesa-2.3.0/src/vesa.c
--- xf86-video-vesa-2.3.0/src/vesa.c.jx	2010-01-04 14:16:31.000000000 -0500
+++ xf86-video-vesa-2.3.0/src/vesa.c	2010-10-11 15:29:52.000000000 -0400
@@ -651,10 +651,12 @@ VESAPreInit(ScrnInfoPtr pScrn, int flags
 				    V_MODETYPE_VBE);
 
     /* Preferred order for default depth selection. */
-    if (depths & V_DEPTH_24)
+    if ((depths & V_DEPTH_24) && (flags24 & Support32bppFb))
 	defaultDepth = 24;
     else if (depths & V_DEPTH_16)
 	defaultDepth = 16;
+    else if (depths & V_DEPTH_24)
+	defaultDepth = 24;
     else if (depths & V_DEPTH_15)
 	defaultDepth = 15;
     else if (depths & V_DEPTH_8)
