--- example/fioclient.c.orig	2009-06-18 13:14:09.000000000 +0200
+++ example/fioclient.c	2009-09-08 16:45:43.000000000 +0200
@@ -117,6 +117,6 @@
 	}
 
  usage:
-	fprintf(stderr, usage);
+	fprintf(stderr, "%s", usage);
 	return 1;
 }
--- example/cusexmp.c.orig	2009-06-18 13:17:55.000000000 +0200
+++ example/cusexmp.c	2009-09-08 16:56:43.000000000 +0200
@@ -247,7 +247,7 @@
 	switch (key) {
 	case 0:
 		param->is_help = 1;
-		fprintf(stderr, usage);
+		fprintf(stderr, "%s", usage);
 		return fuse_opt_add_arg(outargs, "-ho");
 	default:
 		return 1;
