--- configure.ac~	2014-01-20 17:57:48.293329571 +0100
+++ configure.ac	2014-01-20 17:58:10.999996162 +0100
@@ -95,22 +95,6 @@
 AM_CONDITIONAL(BSD, test "$arch" = bsd)
 
 util_linux_ok=yes
-if test "$arch" = linux -a "$cross_compiling" != "yes"; then
-	AC_MSG_CHECKING([if umount supports --fake --no-canonicalize])
-	# exit code of umount is 1 if option is unrecognised, 2 otherwise
-	umount --fake --no-canonicalize > /dev/null 2>&1
-	if test $? != 1; then
-		AC_MSG_RESULT([yes])
-	else
-		firstline=`umount --fake --no-canonicalize 2>&1 | head -1`
-		if test "$firstline" = 'umount: only root can use "--fake" option'; then
-			AC_MSG_RESULT([yes])
-		else
-			AC_MSG_RESULT([$firstline])
-			util_linux_ok=no
-		fi
-	fi
-fi
 
 AC_CONFIG_FILES([fuse.pc Makefile lib/Makefile util/Makefile example/Makefile include/Makefile doc/Makefile])
 AC_OUTPUT
