Index: fuse-2.7.4/configure.in
===================================================================
--- fuse-2.7.4.orig/configure.in
+++ fuse-2.7.4/configure.in
@@ -62,7 +62,7 @@ AC_CHECK_FUNCS([fork setxattr fdatasync]
 AC_CHECK_MEMBERS([struct stat.st_atim])
 AC_CHECK_MEMBERS([struct stat.st_atimespec])
 
-libfuse_libs="-pthread"
+libfuse_libs="-lpthread"
 LIBS=
 AC_SEARCH_LIBS(dlopen, [dl])
 AC_SEARCH_LIBS(clock_gettime, [rt])
Index: fuse-2.7.4/util/Makefile.am
===================================================================
--- fuse-2.7.4.orig/util/Makefile.am
+++ fuse-2.7.4/util/Makefile.am
@@ -11,7 +11,7 @@ mount_fuse_SOURCES = mount.fuse.c
 
 ulockmgr_server_SOURCES = ulockmgr_server.c
 ulockmgr_server_CPPFLAGS = -D_FILE_OFFSET_BITS=64 -D_REENTRANT 
-ulockmgr_server_LDFLAGS = -pthread
+ulockmgr_server_LDFLAGS = -lpthread
 
 install-exec-hook:
 	-chown root $(DESTDIR)$(bindir)/fusermount
