diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 3f99e2b..0ada32e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -138,7 +138,7 @@ target_link_libraries(kalzium ${KDE4_KHTML_LIBRARY} ${KDE4_KNEWSTUFF3_LIBS} ${KD
 if (OPENBABEL2_FOUND)
    target_link_libraries(kalzium ${OPENBABEL2_LIBRARIES})
    if (EIGEN2_FOUND AND Avogadro_FOUND)
-      target_link_libraries(kalzium compoundviewer avogadro)
+      target_link_libraries(kalzium ${QT_QTOPENGL_LIBRARY} compoundviewer avogadro)
    endif (EIGEN2_FOUND AND Avogadro_FOUND)
 endif (OPENBABEL2_FOUND)
