--- Makefile	2007-10-29 23:54:15.000000000 +0200
+++ Makefile	2007-10-29 23:55:37.000000000 +0200
@@ -2,8 +2,7 @@
 CC  = gcc
 OBJ = main.o UnAlz.o UnAlzUtils.o UnAlzBz2decompress.o UnAlzBzip2.o UnAlzbzlib.o
 BIN = unalz
-LDFLAGS = 
-CFLAGS = -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 
+OS_CFLAGS = -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 $(CFLAGS)
 
 all:
 	@echo ""
@@ -34,7 +33,7 @@
 	$(CPP) $(OBJ) $(LDFLAGS) -o $(BIN)
 
 linux-utf8: unalz
-	$(CPP) -c UnAlz.cpp -c UnAlzUtils.cpp -c main.cpp -D_UNALZ_ICONV -D_UNALZ_UTF8 $(CFLAGS)
+	$(CPP) -c UnAlz.cpp -c UnAlzUtils.cpp -c main.cpp -D_UNALZ_ICONV -D_UNALZ_UTF8 $(OS_CFLAGS)
 	$(CPP) $(OBJ) $(LDFLAGS) -lz -lbz2 -o $(BIN)
 
 install:
