diff -up iproute2-20091009/Makefile.kernel iproute2-20091009/Makefile
--- iproute2-20091009/Makefile.kernel	2009-09-21 10:26:11.000000000 +0200
+++ iproute2-20091009/Makefile	2009-10-09 12:27:03.687382422 +0200
@@ -6,6 +6,7 @@ CONFDIR=/etc/iproute2
 DOCDIR=/share/doc/iproute2
 MANDIR=/share/man
 ARPDDIR=/var/lib/arpd
+KERNEL_INCLUDE=/usr/include
 
 # Path to db_185.h include
 DBM_INCLUDE:=$(ROOTDIR)/usr/include
