diff -up iproute2-2.6.31/Makefile.old iproute2-2.6.31/Makefile
--- iproute2-2.6.31/Makefile.old	2010-01-04 12:46:47.000000000 +0100
+++ iproute2-2.6.31/Makefile	2010-01-04 12:50:17.396384644 +0100
@@ -13,7 +13,7 @@ DBM_INCLUDE:=$(ROOTDIR)/usr/include
 
 SHARED_LIBS = y
 
-DEFINES= -DRESOLVE_HOSTNAMES -DLIBDIR=\"$(LIBDIR)\"
+DEFINES= -DRESOLVE_HOSTNAMES -DLIBDIR=\"$(LIBDIR)\" -DIPT_LIB_DIR=\"$(IPT_LIB_DIR)\"
 ifneq ($(SHARED_LIBS),y)
 DEFINES+= -DNO_SHARED_LIBS
 endif
