Index: glest-3.2.1/mk/linux/Jamfile
===================================================================
--- glest-3.2.1.orig/mk/linux/Jamfile	2009-02-17 21:33:09.000000000 +0100
+++ glest-3.2.1/mk/linux/Jamfile	2009-02-17 21:33:12.000000000 +0100
@@ -26,7 +26,7 @@
 }
 
 Library glestlib : $(LIB_SOURCES) ;
-ExternalLibs glestlib : SDL GL GLU XERCES VORBIS VORBISFILE OGG OPENAL LUA ;
+ExternalLibs glestlib : SDL GL GLU XERCES VORBISFILE OPENAL LUA ;
 IncludeDir glestlib : $(LIB_INCLUDE_DIRS) ;
 
 #### Game ####
@@ -55,7 +55,7 @@
 
 Application glest : $(GLEST_SOURCES) ;
 LinkWith glest : glestlib ;
-ExternalLibs glest : SDL GL GLU XERCES VORBIS VORBISFILE OGG OPENAL LUA ;
+ExternalLibs glest : SDL GL GLU XERCES VORBISFILE OPENAL LUA ;
 IncludeDir glest : ../shared_lib/include/$(LIB_INCLUDE_DIRS) $(GLEST_DIRS) ;
 
 #### Editor ####
