diff -Nur glest-source-3.2.2-old/shared_lib/sources/platform/posix/socket.cpp glest-source-3.2.2/shared_lib/sources/platform/posix/socket.cpp
--- glest-source-3.2.2-old/shared_lib/sources/platform/posix/socket.cpp	2009-08-21 13:55:43.452611574 +0300
+++ glest-source-3.2.2/shared_lib/sources/platform/posix/socket.cpp	2009-08-21 13:56:17.834486961 +0300
@@ -10,6 +10,7 @@
 
 #include <cstring>
 #include <cstdlib>
+#include <cstdio>
 #include <stdexcept>
 #include <sstream>
 #if defined(HAVE_SYS_IOCTL_H)
