diff -Naur glest-source-3.1.2.orig/shared_lib/sources/platform/posix/socket.cpp glest-source-3.1.2/shared_lib/sources/platform/posix/socket.cpp
--- glest-source-3.1.2.orig/shared_lib/sources/platform/posix/socket.cpp	2008-02-16 15:34:09.000000000 +0200
+++ glest-source-3.1.2/shared_lib/sources/platform/posix/socket.cpp	2008-04-18 20:05:43.000000000 +0300
@@ -10,6 +10,10 @@
 
 #include <stdexcept>
 #include <sstream>
+
+#include <cstring>
+#include <cstdlib>
+
 #if defined(HAVE_SYS_IOCTL_H)
 #define BSD_COMP /* needed for FIONREAD on Solaris2 */
 #include <sys/ioctl.h>
diff -Naur glest-source-3.1.2.orig/shared_lib/sources/util/conversion.cpp glest-source-3.1.2/shared_lib/sources/util/conversion.cpp
--- glest-source-3.1.2.orig/shared_lib/sources/util/conversion.cpp	2008-01-22 22:10:59.000000000 +0200
+++ glest-source-3.1.2/shared_lib/sources/util/conversion.cpp	2008-04-18 20:11:30.000000000 +0300
@@ -1,7 +1,7 @@
 // ==============================================================
 //	This file is part of Glest Shared Library (www.glest.org)
 //
-//	Copyright (C) 2001-2008 Martiño Figueroa
+//	Copyright (C) 2001-2008 Martiï¿½o Figueroa
 //
 //	You can redistribute this code and/or modify it under 
 //	the terms of the GNU General Public License as published 
@@ -14,6 +14,8 @@
 #include <stdexcept>
 #include <cstdio>
 
+#include <cstdlib>
+
 #include "leak_dumper.h"
 
 using namespace std;
diff -Naur glest-source-3.1.2.orig/shared_lib/sources/util/properties.cpp glest-source-3.1.2/shared_lib/sources/util/properties.cpp
--- glest-source-3.1.2.orig/shared_lib/sources/util/properties.cpp	2008-02-08 12:43:47.000000000 +0200
+++ glest-source-3.1.2/shared_lib/sources/util/properties.cpp	2008-04-18 20:07:20.000000000 +0300
@@ -1,7 +1,7 @@
 // ==============================================================
 //	This file is part of Glest Shared Library (www.glest.org)
 //
-//	Copyright (C) 2001-2007 Martiño Figueroa
+//	Copyright (C) 2001-2007 Martiï¿½o Figueroa
 //
 //	You can redistribute this code and/or modify it under 
 //	the terms of the GNU General Public License as published 
@@ -14,6 +14,8 @@
 #include <fstream>
 #include <stdexcept>
 
+#include <cstring>
+
 #include "conversion.h"
 #include "leak_dumper.h"
 
diff -Naur glest-source-3.1.2.orig/shared_lib/sources/util/util.cpp glest-source-3.1.2/shared_lib/sources/util/util.cpp
--- glest-source-3.1.2.orig/shared_lib/sources/util/util.cpp	2008-01-22 22:10:59.000000000 +0200
+++ glest-source-3.1.2/shared_lib/sources/util/util.cpp	2008-04-18 20:07:53.000000000 +0300
@@ -1,7 +1,7 @@
 // ==============================================================
 //	This file is part of Glest Shared Library (www.glest.org)
 //
-//	Copyright (C) 2001-2008 Martiño Figueroa
+//	Copyright (C) 2001-2008 Martiï¿½o Figueroa
 //
 //	You can redistribute this code and/or modify it under 
 //	the terms of the GNU General Public License as published 
@@ -15,6 +15,8 @@
 #include <cassert>
 #include <stdexcept>
 
+#include <cstring>
+
 #include "leak_dumper.h"
 
 using namespace std;
