--- Jamrules.orig	2010-02-11 02:45:34.000000000 -0600
+++ Jamrules	2010-02-11 02:45:58.000000000 -0600
@@ -21,9 +21,9 @@
 
 LIBS += -lX11 ;
 
-COMPILER_CFLAGS            += -Wall -W -Wno-unused -Wno-sign-compare ;
-COMPILER_CFLAGS_optimize   += -O3 -DNDEBUG ;
-COMPILER_CXXFLAGS_optimize += -O3 -DNDEBUG ;
+COMPILER_CFLAGS            += ;
+COMPILER_CFLAGS_optimize   += ;
+COMPILER_CXXFLAGS_optimize += ;
 COMPILER_LIBS_optimize     += ;
 COMPILER_CFLAGS_debug      += -DDEBUG -g3 ;
 COMPILER_CXXFLAGS_debug    += -DDEBUG -g3 ;
