Index: smc-1.9/src/core/filesystem/filesystem.cpp
===================================================================
--- smc-1.9.orig/src/core/filesystem/filesystem.cpp
+++ smc-1.9/src/core/filesystem/filesystem.cpp
@@ -142,32 +142,32 @@ vector<std::string> Get_Directory_Files(
 			if( fs::is_directory( *dir_itr ) )
 			{
 				// ignore hidden directories
-				if( dir_itr->path().leaf().find( "." ) == 0 )
+				if( dir_itr->path().filename().string().find( "." ) == 0 )
 				{
 					continue;
 				}
 
 				if( with_directories )
 				{
-					valid_files.push_back( dir + "/" + dir_itr->path().leaf() );
+					valid_files.push_back( dir + "/" + dir_itr->path().filename().string() );
 				}
 
 				// load all items from the sub-directory
 				if( search_in_sub_directories )
 				{
-					vector<std::string> new_valid_files = Get_Directory_Files( dir + "/" + dir_itr->path().leaf(), file_type, with_directories );
+					vector<std::string> new_valid_files = Get_Directory_Files( dir + "/" + dir_itr->path().filename().string(),file_type, with_directories );
 					valid_files.insert( valid_files.end(), new_valid_files.begin(), new_valid_files.end() );
 				}
 			}
 			// valid file
-			else if( file_type.empty() || dir_itr->path().leaf().rfind( file_type ) != std::string::npos )
+			else if( file_type.empty() || dir_itr->path().filename().string().rfind( file_type ) != std::string::npos )
 			{
-				valid_files.push_back( dir + "/" + dir_itr->path().leaf() );
+				valid_files.push_back( dir + "/" + dir_itr->path().filename().string() );
 			}
 		}
 		catch( const std::exception &ex )
 		{
-			printf( "%s %s\n", dir_itr->path().leaf().c_str(), ex.what() );
+			printf( "%s %s\n", dir_itr->path().filename().string().c_str(), ex.what() );
 		}
 	}
 
Index: smc-1.9/src/overworld/world_manager.cpp
===================================================================
--- smc-1.9.orig/src/overworld/world_manager.cpp
+++ smc-1.9/src/overworld/world_manager.cpp
@@ -118,7 +118,7 @@ void cOverworld_Manager :: Load_Dir( con
 	{
 		try
 		{
-			std::string current_dir = dir_itr->path().leaf();
+			std::string current_dir = dir_itr->path().filename().string();
 
 			// only directories with an existing description
 			if( fs::is_directory( *dir_itr ) && File_Exists( dir + "/" + current_dir + "/description.xml" ) )
@@ -145,7 +145,7 @@ void cOverworld_Manager :: Load_Dir( con
 		}
 		catch( const std::exception &ex )
 		{
-			printf( "%s %s\n", dir_itr->path().leaf().c_str(), ex.what() );
+			printf( "%s %s\n", dir_itr->path().filename().string().c_str(), ex.what() );
 		}
 	}
 }
