From 678188bdadef4853b88031b8d5277b81bdfb5f3c Mon Sep 17 00:00:00 2001
From: Perberos <perberos@gmail.com>
Date: Tue, 31 Jul 2012 22:30:42 +0000
Subject: Update src/math-preferences.h

adding missing c++ bracket
---
diff --git a/src/math-preferences.h b/src/math-preferences.h
index e7767f6..c816c31 100644
--- a/src/math-preferences.h
+++ b/src/math-preferences.h
@@ -42,4 +42,6 @@ GType math_preferences_get_type(void);
 
 MathPreferencesDialog* math_preferences_dialog_new(MathEquation* equation);
 
+G_END_DECLS
+
 #endif /* MATH_PREFERENCES_H */
--
cgit v0.9.0.3-65-g4555
