From f6d05dad1665b9490e53f453cdd17294b9efee55 Mon Sep 17 00:00:00 2001
From: Perberos <perberos@gmail.com>
Date: Tue, 31 Jul 2012 23:19:23 +0000
Subject: adding hack to fix bison decls mess

---
diff --git a/src/mp-equation-private.h b/src/mp-equation-private.h
index 319e916..9885a8f 100644
--- a/src/mp-equation-private.h
+++ b/src/mp-equation-private.h
@@ -22,6 +22,12 @@
 
 #include "mp-equation.h"
 
+/* An opaque pointer. */
+#ifndef YY_TYPEDEF_YY_SCANNER_T
+#define YY_TYPEDEF_YY_SCANNER_T
+typedef void* yyscan_t;
+#endif /* YY_TYPEDEF_YY_SCANNER_T */
+
 typedef struct MPEquationParserState MPEquationParserState;
 
 /* State for parser */
--
cgit v0.9.0.3-65-g4555
