--- enscript-1.6.4/afm/Makefile.am.config	2003-03-05 07:26:16.000000000 +0000
+++ enscript-1.6.4/afm/Makefile.am	2005-12-23 09:56:52.000000000 +0000
@@ -36,11 +36,11 @@
 EXTRA_DIST = font.map $(default_afm) $(public_fonts)
 
 install-data-local:
-	$(top_srcdir)/mkinstalldirs $(datadir)/enscript
-	$(top_srcdir)/mkinstalldirs $(datadir)/enscript/afm
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(datadir)/enscript
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(datadir)/enscript/afm
 	for f in $(EXTRA_DIST); do \
-	  $(INSTALL_DATA) $(srcdir)/$$f $(datadir)/enscript/afm/$$f; \
+	  $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)/$(datadir)/enscript/afm/$$f; \
 	done
 
 uninstall-local:
-	rm -rf $(datadir)/enscript/afm
+	rm -rf $(DESTDIR)/$(datadir)/enscript/afm
--- enscript-1.6.4/afm/Makefile.in.config	2003-03-05 07:40:06.000000000 +0000
+++ enscript-1.6.4/afm/Makefile.in	2005-12-23 09:56:52.000000000 +0000
@@ -319,14 +319,14 @@
 
 
 install-data-local:
-	$(top_srcdir)/mkinstalldirs $(datadir)/enscript
-	$(top_srcdir)/mkinstalldirs $(datadir)/enscript/afm
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(datadir)/enscript
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(datadir)/enscript/afm
 	for f in $(EXTRA_DIST); do \
-	  $(INSTALL_DATA) $(srcdir)/$$f $(datadir)/enscript/afm/$$f; \
+	  $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)/$(datadir)/enscript/afm/$$f; \
 	done
 
 uninstall-local:
-	rm -rf $(datadir)/enscript/afm
+	rm -rf $(DESTDIR)/$(datadir)/enscript/afm
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
--- enscript-1.6.4/lib/Makefile.in.config	2003-03-05 07:40:07.000000000 +0000
+++ enscript-1.6.4/lib/Makefile.in	2005-12-23 09:56:52.000000000 +0000
@@ -345,15 +345,16 @@
 all-local: enscript.cfg
 
 install-data-local: enscript.cfg
-	$(top_srcdir)/mkinstalldirs $(sysconfdir)
-	if test -r $(sysconfdir)/enscript.cfg; then \
-	  cp $(sysconfdir)/enscript.cfg $(sysconfdir)/enscript.cfg.old; \
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(sysconfdir)
+	if test -r $(DESTDIR)$(sysconfdir)/enscript.cfg; then \
+	  cp $(DESTDIR)$(sysconfdir)/enscript.cfg \
+		$(DESTDIR)$(sysconfdir)/enscript.cfg.old; \
 	else :; \
 	fi
-	$(INSTALL_DATA) enscript.cfg $(sysconfdir)/enscript.cfg
+	$(INSTALL_DATA) enscript.cfg $(DESTDIR)$(sysconfdir)/enscript.cfg
 
 uninstall-local:
-	rm -f $(sysconfdir)/enscript.cfg
+	rm -f $(DESTDIR)$(sysconfdir)/enscript.cfg
 
 enscript.cfg: $(srcdir)/enscript.cfg.in Makefile
 	sed 's%@DATADIR@%$(datadir)%g; s%@media@%@MEDIA@%g; s%@BINDIR@%$(bindir)%g; s%@spooler@%@SPOOLER@%g; s%@pslevel@%@PSLEVEL@%g' \
--- enscript-1.6.4/states/hl/Makefile.am.config	2005-12-23 09:59:55.000000000 +0000
+++ enscript-1.6.4/states/hl/Makefile.am	2005-12-23 10:00:09.000000000 +0000
@@ -45,10 +45,10 @@
 EXTRA_DIST = $(states)
 
 install-data-local:
-	$(top_srcdir)/mkinstalldirs $(datadir)/enscript
-	$(top_srcdir)/mkinstalldirs $(datadir)/enscript/hl
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(datadir)/enscript
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(datadir)/enscript/hl
 	for f in $(states); do \
-	  $(INSTALL_DATA) $(srcdir)/$$f $(datadir)/enscript/hl/$$f; \
+	  $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)/$(datadir)/enscript/hl/$$f; \
 	done
 
 uninstall-local:
--- enscript-1.6.4/states/hl/Makefile.in.config	2005-12-23 10:00:13.000000000 +0000
+++ enscript-1.6.4/states/hl/Makefile.in	2005-12-23 10:00:34.000000000 +0000
@@ -330,10 +330,10 @@
 
 
 install-data-local:
-	$(top_srcdir)/mkinstalldirs $(datadir)/enscript
-	$(top_srcdir)/mkinstalldirs $(datadir)/enscript/hl
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(datadir)/enscript
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(datadir)/enscript/hl
 	for f in $(states); do \
-	  $(INSTALL_DATA) $(srcdir)/$$f $(datadir)/enscript/hl/$$f; \
+	  $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)/$(datadir)/enscript/hl/$$f; \
 	done
 
 uninstall-local:
