Index: libgssglue-0.1/doc/gssapi_mech.conf
===================================================================
--- libgssglue-0.1.orig/doc/gssapi_mech.conf
+++ libgssglue-0.1/doc/gssapi_mech.conf
@@ -15,7 +15,8 @@
 # library                               initialization function
 # ================================	==========================
 # The MIT K5 gssapi library, use special function for initialization.
-/usr/lib/libgssapi_krb5.so		mechglue_internal_krb5_init
+libgssapi_krb5.so.2            mechglue_internal_krb5_init
 #
 # The SPKM3 gssapi library function.  Use the function spkm3_gss_initialize.
-# /usr/local/gss_mechs/spkm/spkm3/libgssapi_spkm3.so    spkm3_gss_initialize
+# libgssapi_spkm3.so    spkm3_gss_initialize
+
