--- alsa-utils-1.0.28/alsaconf/alsaconf.in~	2014-06-13 19:21:05.000000000 +0200
+++ alsa-utils-1.0.28/alsaconf/alsaconf.in	2014-06-18 08:49:50.065218184 +0200
@@ -304,7 +304,7 @@
   cfgfile="/etc/modprobe.conf"
   if [ -d /etc/modprobe.d ]; then
     cfgout="/etc/modprobe.d/50-sound.conf"
-    cfgoldout="/etc/modprobe.d/sound"
+    cfgoldout="/etc/modprobe.d/sound.conf"
     if [ ! -f $cfgout ]; then
 	case "$(modprobe -V)" in
 	"module-init-tools version "3.[789]*|\
@@ -313,7 +313,7 @@
 	"kmod version "*)
 	  ;;
 	*)
-	  cfgout="/etc/modprobe.d/sound"
+	  cfgout="/etc/modprobe.d/sound.conf"
 	  cfgoldout=""
 	  ;;
 	esac 
