diff -Naur qsvn-0.8.0.orig/src/qsvn.desktop qsvn-0.8.0/src/qsvn.desktop
--- qsvn-0.8.0.orig/src/qsvn.desktop	2008-12-17 00:54:18.000000000 +0200
+++ qsvn-0.8.0/src/qsvn.desktop	2008-12-21 00:13:11.000000000 +0200
@@ -5,7 +5,9 @@
 Icon=qsvn
 Name=qsvn
 GenericName=SVN client
+GenericName[tr]=SVN İstemcisi
 Comment=A graphical Subversion client
+Comment[tr]=Grafik Arayüzlü Svn İstemcisi
 GenericName[de]=Graphische Oberfläche für SVN
 Comment[de]=Ein grafischer Subversionclient
 Type=Application
