diff -Nuar a/cmake/FindICU.cmake b/cmake/FindICU.cmake
--- a/cmake/FindICU.cmake	2013-08-27 00:45:56.000000000 +0300
+++ b/cmake/FindICU.cmake	2013-08-30 20:13:47.048819701 +0300
@@ -76,11 +76,11 @@
 set(${ICU_PRIVATE_VAR_NS}_BIN_SUFFIXES )
 set(${ICU_PRIVATE_VAR_NS}_LIB_SUFFIXES )
 if(CMAKE_SIZEOF_VOID_P EQUAL 8)
-    list(APPEND ${ICU_PRIVATE_VAR_NS}_BIN_SUFFIXES "bin64")
-    list(APPEND ${ICU_PRIVATE_VAR_NS}_LIB_SUFFIXES "lib64")
+    list(APPEND ${ICU_PRIVATE_VAR_NS}_BIN_SUFFIXES "bin")
+    list(APPEND ${ICU_PRIVATE_VAR_NS}_LIB_SUFFIXES "lib")
 endif(CMAKE_SIZEOF_VOID_P EQUAL 8)
-list(APPEND ${ICU_PRIVATE_VAR_NS}_BIN_SUFFIXES "bin")
-list(APPEND ${ICU_PRIVATE_VAR_NS}_LIB_SUFFIXES "lib")
+list(APPEND ${ICU_PRIVATE_VAR_NS}_BIN_SUFFIXES "")
+list(APPEND ${ICU_PRIVATE_VAR_NS}_LIB_SUFFIXES "")
 
 set(${ICU_PRIVATE_VAR_NS}_COMPONENTS )
 # <icu component name> <library name 1> ... <library name N>
diff -Nuar a/cmake/RazorLibSuffix.cmake b/cmake/RazorLibSuffix.cmake
--- a/cmake/RazorLibSuffix.cmake	2013-08-27 00:45:56.000000000 +0300
+++ b/cmake/RazorLibSuffix.cmake	2013-08-30 20:15:20.851816454 +0300
@@ -7,8 +7,8 @@
     # All 32bit system have empty lib suffix
     if(CMAKE_SIZEOF_VOID_P EQUAL 8)
         # If there is lib64 dir, set suffix to 64
-        if(IS_DIRECTORY ${CMAKE_INSTALL_PREFIX}/lib64)
-            set(LIB_SUFFIX 64)
+        if(IS_DIRECTORY ${CMAKE_INSTALL_PREFIX}/lib)
+            set(LIB_SUFFIX "")
         elseif(IS_DIRECTORY ${CMAKE_INSTALL_PREFIX}/lib)
             set(LIB_SUFFIX "")
         else()
diff -Nuar a/libraries/qtxdg/cmake/RazorLibSuffix.cmake b/libraries/qtxdg/cmake/RazorLibSuffix.cmake
--- a/libraries/qtxdg/cmake/RazorLibSuffix.cmake	2013-08-27 00:45:56.000000000 +0300
+++ b/libraries/qtxdg/cmake/RazorLibSuffix.cmake	2013-08-30 20:15:53.148815335 +0300
@@ -7,8 +7,8 @@
     # All 32bit system have empty lib suffix
     if(CMAKE_SIZEOF_VOID_P EQUAL 8)
         # If there is lib64 dir, set suffix to 64
-        if(IS_DIRECTORY ${CMAKE_INSTALL_PREFIX}/lib64)
-            set(LIB_SUFFIX 64)
+        if(IS_DIRECTORY ${CMAKE_INSTALL_PREFIX}/lib)
+            set(LIB_SUFFIX "")
         elseif(IS_DIRECTORY ${CMAKE_INSTALL_PREFIX}/lib)
             set(LIB_SUFFIX "")
         else()
diff -Nuar a/razorqt-resources/config/session.conf b/razorqt-resources/config/session.conf
--- a/razorqt-resources/config/session.conf	2013-08-27 00:45:57.000000000 +0300
+++ b/razorqt-resources/config/session.conf	2013-08-30 20:16:06.320814879 +0300
@@ -1,3 +1,3 @@
 [environment]
 XDG_CURRENT_DESKTOP=Razor
-QT_PLUGIN_PATH=${QT_PLUGIN_PATH}:/usr/lib64/kde4/plugins:/usr/lib/kde4/plugins
+QT_PLUGIN_PATH=${QT_PLUGIN_PATH}:/usr/lib/kde4/plugins
