--- ./razorqt-panel/panel/resources/panel.conf.orig	2012-10-06 12:02:27.950000039 +0200
+++ ./razorqt-panel/panel/resources/panel.conf	2012-10-06 12:34:14.757000039 +0200
@@ -11,11 +11,12 @@
 
 [quicklaunch]
 type=quicklaunch
-apps/size=4
-apps/1/desktop=firefox.desktop
-apps/2/desktop=chromium-browser.desktop
-apps/3/desktop=KMail.desktop
-apps/4/desktop=razor-config.desktop
+apps/size=5
+apps/1/desktop=arora.desktop
+apps/2/desktop=andromeda.desktop
+apps/3/desktop=trojita.desktop
+apps/4/desktop=mageia-drakconf.desktop
+apps/5/desktop=razor-config.desktop
 
 [taskbar]
 type=taskbar
