--- wrapper/Makefile.am.orig	2009-03-25 20:01:41.000000000 +0100
+++ wrapper/Makefile.am	2009-03-25 20:02:21.000000000 +0100
@@ -1,10 +1,10 @@
 
-AM_CFLAGS = -Werror
+AM_CFLAGS =
 
 INCLUDES= @PACKAGE_CFLAGS@ $(PYTHON_INCLUDES) -I$(top_srcdir) -I$(top_srcdir)/opensync @XML_CFLAGS@
 
 opensync_wrap.c: $(srcdir)/opensync.i
-	$(SWIG) -Werror -python -modern -I$(top_srcdir)/opensync $(PYTHON_INCLUDES) -o opensync_wrap.c $(srcdir)/opensync.i
+	$(SWIG) -python -modern -I$(top_srcdir)/opensync $(PYTHON_INCLUDES) -o opensync_wrap.c $(srcdir)/opensync.i
 
 EXTRA_DIST = opensync.i
 
--- wrapper/Makefile.in.orig	2009-03-25 20:01:48.000000000 +0100
+++ wrapper/Makefile.in	2009-03-25 20:02:11.000000000 +0100
@@ -228,7 +228,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-AM_CFLAGS = -Werror
+AM_CFLAGS = 
 INCLUDES = @PACKAGE_CFLAGS@ $(PYTHON_INCLUDES) -I$(top_srcdir) -I$(top_srcdir)/opensync @XML_CFLAGS@
 EXTRA_DIST = opensync.i
 CLEANFILES = \
@@ -548,7 +548,7 @@
 
 
 opensync_wrap.c: $(srcdir)/opensync.i
-	$(SWIG) -Werror -python -modern -I$(top_srcdir)/opensync $(PYTHON_INCLUDES) -o opensync_wrap.c $(srcdir)/opensync.i
+	$(SWIG) -python -modern -I$(top_srcdir)/opensync $(PYTHON_INCLUDES) -o opensync_wrap.c $(srcdir)/opensync.i
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
