Index: rkward-0.4.9a/rkward/rkward.desktop
===================================================================
--- rkward-0.4.9a.orig/rkward/rkward.desktop
+++ rkward-0.4.9a/rkward/rkward.desktop
@@ -5,10 +5,11 @@ Name=RKWard
 Comment=GUI for the R-project
 Comment[pl]=Graficzny interfejs dla projektu R
 Comment[fr]=Interface graphique pour R
+Comment[tr]=R için arayüz
 Icon=rkward
 Exec=rkward -caption "%c" %i %m  
 Terminal=false
 Type=Application
 DocPath=rkward/index.html
-Categories=Education;Science;Math;
+Categories=Applications;Science;
 # vi: encoding=utf-8
