diff -Nuar rkward-0.4.7a.orig/configure rkward-0.4.7a/configure
--- rkward-0.4.7a.orig/configure	2007-05-07 14:10:22.000000000 +0300
+++ rkward-0.4.7a/configure	2007-05-07 17:02:04.000000000 +0300
@@ -31217,7 +31217,7 @@
 if test -f "$R_HOMEDIR/lib/libRlapack.so"; then
    { echo "$as_me:$LINENO: result: yes $R_HOMEDIR/lib/libRlapack.so exists" >&5
 echo "${ECHO_T}yes $R_HOMEDIR/lib/libRlapack.so exists" >&6; }
-   R_LAPACK_FLAG="-lRlapack -lgfortran"
+   R_LAPACK_FLAG="-lRlapack"
 else
    { echo "$as_me:$LINENO: result: no $R_HOMEDIR/lib/libRlapack.so does not exist" >&5
 echo "${ECHO_T}no $R_HOMEDIR/lib/libRlapack.so does not exist" >&6; }
