diff -Nuar rkward-0.5.0pre1.orig/rkward/rbackend/FindR.cmake rkward-0.5.0pre1/rkward/rbackend/FindR.cmake
--- rkward-0.5.0pre1.orig/rkward/rbackend/FindR.cmake	2007-10-22 02:51:58.000000000 +0300
+++ rkward-0.5.0pre1/rkward/rbackend/FindR.cmake	2008-01-16 08:24:22.000000000 +0200
@@ -86,7 +86,7 @@
 	MESSAGE(STATUS "No, ${R_SHAREDLIBDIR}/libRlapack.so does not exist")
 ELSE(NOT LIBR_LAPACK)
 	MESSAGE(STATUS "Yes, ${LIBR_LAPACK} exists")
-	SET(R_USED_LIBS ${R_USED_LIBS} Rlapack gfortran)
+	SET(R_USED_LIBS ${R_USED_LIBS} Rlapack )
 ENDIF(NOT LIBR_LAPACK)
 
 # for at least some versions of R, we seem to have to link against -lRblas. Else loading some
