diff -Nuar rkward-0.4.9a.orig/rkward/rbackend/-m64 rkward-0.4.9a/rkward/rbackend/-m64
--- rkward-0.4.9a.orig/rkward/rbackend/-m64	2007-10-01 15:50:58.000000000 +0300
+++ rkward-0.4.9a/rkward/rbackend/-m64	2008-10-25 22:36:20.000000000 +0300
@@ -139,7 +139,7 @@
   /usr/include/asm-generic/errno-base.h \
   /usr/share/R/include/R_ext/Constants.h \
   /usr/share/R/include/R_ext/Random.h /usr/share/R/include/Rinterface.h \
-  /usr/share/R/include/R_ext/RStartup.h /usr/share/R/include/Rdevices.h \
+  /usr/share/R/include/R_ext/RStartup.h \
   /usr/share/R/include/Rgraphics.h /usr/share/R/include/Rversion.h \
   /usr/share/R/include/R_ext/Parse.h /usr/include/dlfcn.h \
   /usr/include/bits/dlfcn.h /usr/include/sys/resource.h \
@@ -602,8 +602,6 @@
 
 /usr/share/R/include/R_ext/RStartup.h:
 
-/usr/share/R/include/Rdevices.h:
-
 /usr/share/R/include/Rgraphics.h:
 
 /usr/share/R/include/Rversion.h:
diff -Nuar rkward-0.4.9a.orig/rkward/rbackend/rembedinternal.cpp rkward-0.4.9a/rkward/rbackend/rembedinternal.cpp
--- rkward-0.4.9a.orig/rkward/rbackend/rembedinternal.cpp	2008-04-10 16:19:08.000000000 +0300
+++ rkward-0.4.9a/rkward/rbackend/rembedinternal.cpp	2008-10-25 22:34:05.000000000 +0300
@@ -87,7 +87,6 @@
 #include "R.h"
 #include "Rinternals.h"
 #include "Rinterface.h"
-#include "Rdevices.h"
 #include "R_ext/Parse.h"
 
 #ifdef R_2_4
