diff -up dia-0.97.2/configure.in.automake-1_13 dia-0.97.2/configure.in
--- dia-0.97.2/configure.in.automake-1_13	2011-04-25 15:04:41.000000000 +0200
+++ dia-0.97.2/configure.in	2012-12-30 03:35:44.100995644 +0100
@@ -8,7 +8,7 @@ AC_PREREQ(2.50)
 AC_REVISION($Revision: 1.288 $)
 
 dnl Specify a header configuration file
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 dnl do this check now, to keep autogen quiet.
 AC_PROG_CC 
