diff -Nuar dia-0.97.orig/lib/Makefile.in dia-0.97/lib/Makefile.in
--- dia-0.97.orig/lib/Makefile.in	2009-08-21 04:19:30.717981222 +0300
+++ dia-0.97/lib/Makefile.in	2009-08-21 04:31:41.234011951 +0300
@@ -478,7 +478,7 @@
 #		charconv.h 
 #		render.h \
 #		render.c 
-libdia_la_LDFLAGS = -avoid-version --export-dynamic
+libdia_la_LDFLAGS = -avoid-version --export-dynamic `pkg-config --libs gtk+-2.0`
 libdia_la_LIBADD = $(GTK_LIBS)
 AM_CPPFLAGS = \
 	$(LIBART_CFLAGS) \
