diff -Nuar a/source/glest_game/CMakeLists.txt b/source/glest_game/CMakeLists.txt
--- a/source/glest_game/CMakeLists.txt	2013-05-01 23:56:19.000000000 +0300
+++ b/source/glest_game/CMakeLists.txt	2013-10-04 07:42:57.511975849 +0300
@@ -86,7 +86,7 @@
 	FIND_PACKAGE(CURL REQUIRED)
 	INCLUDE_DIRECTORIES(${CURL_INCLUDE_DIRS})
 	IF(UNIX)
-		SET(EXTERNAL_LIBS ${EXTERNAL_LIBS} ${CURL_LIBRARIES})
+		SET(EXTERNAL_LIBS ${EXTERNAL_LIBS} ${CURL_LIBRARIES} ssl)
 	ENDIF()
 
 	FIND_PACKAGE(GLEW REQUIRED)
diff -Nuar a/source/shared_lib/CMakeLists.txt b/source/shared_lib/CMakeLists.txt
--- a/source/shared_lib/CMakeLists.txt	2013-06-26 19:58:34.000000000 +0300
+++ b/source/shared_lib/CMakeLists.txt	2013-10-04 07:43:59.771973694 +0300
@@ -378,7 +378,7 @@
 		        MESSAGE(STATUS "*** Using libircclient found in [${IRCCLIENT_INCLUDE_DIR}] IRCCLIENT_FOUND [${IRCCLIENT_FOUND}] IRCCLIENT_LIBRARY [${IRCCLIENT_LIBRARY}]")
 
 		        INCLUDE_DIRECTORIES(${IRCCLIENT_INCLUDE_DIR})
-		        SET(EXTERNAL_LIBS ${EXTERNAL_LIBS} ${IRCCLIENT_LIBRARY})
+		        SET(EXTERNAL_LIBS ${IRCCLIENT_LIBRARY} ssl crypto ${EXTERNAL_LIBS})
 	        endif()
 
                 set(CMAKE_REQUIRED_INCLUDES "${IRCCLIENT_INCLUDE_DIR}")
