From 6bda8ba424d2dbd2441d9a7e6553608dceaa8454 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Thu, 18 Oct 2012 16:53:12 +0200
Subject: [PATCH] main: Set XDG_MENU_PREFIX

---
 gnome-session/main.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gnome-session/main.c b/gnome-session/main.c
index 1903381..147daec 100644
--- a/gnome-session/main.c
+++ b/gnome-session/main.c
@@ -347,6 +347,11 @@ main (int argc, char **argv)
          */
         gsm_util_setenv ("GNOME_DESKTOP_SESSION_ID", "this-is-deprecated");
 
+        /* We want to use the GNOME menus which has the designed categories,
+         * not the legacy redhat-menus.
+         */
+        gsm_util_setenv ("XDG_MENU_PREFIX", "gnome-");
+
         client_store = gsm_store_new ();
 
         xsmp_server = gsm_xsmp_server_new (client_store);
-- 
1.7.12.1

