Index: player-3.0.2/server/drivers/speech/recognition/CMakeLists.txt
===================================================================
--- player-3.0.2.orig/server/drivers/speech/recognition/CMakeLists.txt
+++ player-3.0.2/server/drivers/speech/recognition/CMakeLists.txt
@@ -1,4 +1,4 @@
-PLAYERDRIVER_OPTION (sphinx2 build_sphinx2 OFF "Disabled by default")
+PLAYERDRIVER_OPTION (sphinx2 build_sphinx2 ON "Sphinx2 enabled")
 PLAYERDRIVER_REQUIRE_HEADER (sphinx2 build_sphinx2 sphinx2/ckd_alloc.h)
 PLAYERDRIVER_ADD_DRIVER (sphinx2 build_sphinx2 LINKFLAGS -lsphinx2 -lsphinx2fe -lsphinx2ad
                          SOURCES sphinx2.cc)
