Index: player-3.0.2/server/drivers/shell/CMakeLists.txt
===================================================================
--- player-3.0.2.orig/server/drivers/shell/CMakeLists.txt
+++ player-3.0.2/server/drivers/shell/CMakeLists.txt
@@ -36,11 +36,7 @@ PLAYERDRIVER_ADD_DRIVER (passthrough bui
 PLAYERDRIVER_OPTION (postlog build_postlog ON)
 PLAYERDRIVER_REQUIRE_PKG (postlog build_postlog libpqxx
     postlog_includeDirs postlog_libDirs postlog_linkLibs postlog_linkFlags postlog_cFlags)
-IF (PLAYER_OS_SOLARIS)
-    SET (postlogLinkFlag "-lpq")
-ELSE (PLAYER_OS_SOLARIS)
-    SET (postlogLinkFlag)
-ENDIF (PLAYER_OS_SOLARIS)
+SET (postlogLinkFlag "-lpq")
 PLAYERDRIVER_ADD_DRIVER (postlog build_postlog
     INCLUDEDIRS ${postlog_includeDirs} LIBDIRS ${postlog_libDirs}
     LINKLIBS ${postlog_linkLibs} LINKFLAGS ${postlog_linkFlags} ${postlogLinkFlag}
