--- cmake/internal/SetupDirectories.cmake.OSmanOS	2012-11-12 19:21:47.000000000 +0200
+++ cmake/internal/SetupDirectories.cmake		2012-11-12 19:28:20.640368660 +0200
@@ -11,13 +11,8 @@
 STRING (TOLOWER ${PROJECT_NAME} projectNameLower)
 SET (PLAYER_INCLUDE_INSTALL_DIR "include/${projectNameLower}-${PLAYER_MAJOR_VERSION}.${PLAYER_MINOR_VERSION}")

-IF (PLAYER_PROC_64BIT)
-    SET (PLAYER_LIBRARY_INSTALL_DIR "lib64")
-    MESSAGE (STATUS "Player libraries will be in lib64")
-ELSE (PLAYER_PROC_64BIT)
-    SET (PLAYER_LIBRARY_INSTALL_DIR "lib")
-    MESSAGE (STATUS "Player libraries will be in lib")
-ENDIF (PLAYER_PROC_64BIT)
+SET (PLAYER_LIBRARY_INSTALL_DIR "lib")
+MESSAGE (STATUS "Player libraries will be in lib")

 IF (PLAYER_OS_WIN)
     SET(PLAYER_PLUGIN_PATH "${CMAKE_INSTALL_PREFIX}\\lib")
