diff -up ./client_libs/libplayerc++/playerclient.cc.boostutc ./client_libs/libplayerc++/playerclient.cc
--- ./client_libs/libplayerc++/playerclient.cc.boostutc	2012-06-30 13:14:02.987701081 -0400
+++ ./client_libs/libplayerc++/playerclient.cc	2012-06-30 13:14:18.530298979 -0400
@@ -168,7 +168,7 @@ void PlayerClient::RunThread()
       Read();
     };
     boost::xtime xt;
-    boost::xtime_get(&xt, boost::TIME_UTC);
+    boost::xtime_get(&xt, TIME_UTC);
     // we sleep for 0 seconds
     boost::thread::sleep(xt);
   }