Index: server/drivers/wifi/CMakeLists.txt
===================================================================
--- server/drivers/wifi/CMakeLists.txt	(revision 8808)
+++ server/drivers/wifi/CMakeLists.txt	(working copy)
@@ -8,5 +8,5 @@
 
 PLAYERDRIVER_OPTION (linuxwifi build_linuxwifi ON)
 PLAYERDRIVER_REJECT_OS (linuxwifi build_linuxwifi PLAYER_OS_WIN)
-PLAYERDRIVER_REQUIRE_HEADER (linuxwifi build_linuxwifi linux/wireless.h)
+PLAYERDRIVER_REQUIRE_HEADER (linuxwifi build_linuxwifi "sys/socket.h;linux/wireless.h")
 PLAYERDRIVER_ADD_DRIVER (linuxwifi build_linuxwifi SOURCES linuxwifi.cc)
