diff -Naur player-3.0.2.orig/server/drivers/health/statgrab/statgrab_health.cpp player-3.0.2/server/drivers/health/statgrab/statgrab_health.cpp
--- player-3.0.2.orig/server/drivers/health/statgrab/statgrab_health.cpp	2009-07-24 19:05:37.000000000 +0200
+++ player-3.0.2/server/drivers/health/statgrab/statgrab_health.cpp	2013-08-17 06:32:44.233361318 +0200
@@ -130,7 +130,7 @@
 int StatGrabDriver::MainSetup()
 {
   // Initialise statgrab
-  sg_init();
+  sg_init(0);
   /* Drop setuid/setgid privileges. */
   if (sg_drop_privileges() != 0)
   {
@@ -171,7 +171,7 @@
   //double receivedCpu;
   float cpuIdle, cpuServer, cpuUser ;
 	//CPU
-	cpu_percent = sg_get_cpu_percents();
+	cpu_percent = sg_get_cpu_percents(NULL);
 
 	cpuIdle =  cpu_percent->idle;
 	mHealth.cpu_usage.idle = cpuIdle;
@@ -183,8 +183,8 @@
 
 
 	//Virtual Memory
-	mem_data     = sg_get_mem_stats();
-	swap_stats   = sg_get_swap_stats();
+	mem_data     = sg_get_mem_stats(NULL);
+	swap_stats   = sg_get_swap_stats(NULL);
 
 	mHealth.mem.total = mem_data->total;
 	mHealth.mem.used =  mem_data->used;
