diff -up player-3.0.2/client_libs/libplayerc\+\+/CMakeLists.txt\~ player-3.0.2/client_libs/libplayerc\+\+/CMakeLists.txt
--- player-3.0.2/client_libs/libplayerc++/CMakeLists.txt~	2014-05-25 15:29:00.407677202 +0200
+++ player-3.0.2/client_libs/libplayerc++/CMakeLists.txt	2014-05-25 15:31:23.880919221 +0200
@@ -104,11 +104,7 @@ IF (BUILD_PLAYERCC)
                     OPTION (Boost_USE_MULTITHREAD "Use the multithreaded versions of the Boost libraries" OFF)
                 ENDIF (PLAYER_OS_OSX)
                 MARK_AS_ADVANCED (Boost_USE_MULTITHREAD)
-                IF (Boost_USE_MULTITHREAD)
-                    SET (BOOST_LIB_SUFFIX "-mt" CACHE STRING "Boost library name suffix")
-                ELSE (Boost_USE_MULTITHREAD)
-                    SET (BOOST_LIB_SUFFIX "" CACHE STRING "Boost library name suffix")
-                ENDIF (Boost_USE_MULTITHREAD)
+                SET (BOOST_LIB_SUFFIX "" CACHE STRING "Boost library name suffix")
                 MARK_AS_ADVANCED (BOOST_LIB_SUFFIX)
 
                 IF (HAVE_BOOST_THREAD)

Diff finished.  Sun May 25 15:31:28 2014
