diff -Nuar a/code/q3_ui/ui_mods.c b/code/q3_ui/ui_mods.c
--- a/code/q3_ui/ui_mods.c	2013-06-29 03:26:06.000000000 +0300
+++ b/code/q3_ui/ui_mods.c	2013-06-29 19:24:10.948864728 +0300
@@ -125,7 +125,7 @@
 
 	// always start off with baseq3
 	s_mods.list.numitems = 1;
-	s_mods.list.itemnames[0] = s_mods.descriptionList[0] = "Quake III Arena";
+	s_mods.list.itemnames[0] = s_mods.descriptionList[0] = "Urban TerrorHD";
 	s_mods.fs_gameList[0] = "";
 
 	numdirs = trap_FS_GetFileList( "$modlist", "", dirlist, sizeof(dirlist) );
diff -Nuar a/code/qcommon/files.c b/code/qcommon/files.c
--- a/code/qcommon/files.c	2013-06-29 03:26:06.000000000 +0300
+++ b/code/qcommon/files.c	2013-06-29 19:25:14.052862543 +0300
@@ -2741,7 +2741,7 @@
 	}
 	
 	fs_homepath = Cvar_Get ("fs_homepath", homePath, CVAR_INIT );
-	fs_gamedirvar = Cvar_Get ("fs_game", "q3ut4", CVAR_INIT|CVAR_SYSTEMINFO );
+	fs_gamedirvar = Cvar_Get ("fs_game", "baseut4", CVAR_INIT|CVAR_SYSTEMINFO );
 
 	// add search path elements in reverse priority order
 	if (fs_basepath->string[0]) {
diff -Nuar a/code/qcommon/q_shared.h b/code/qcommon/q_shared.h
--- a/code/qcommon/q_shared.h	2013-06-29 03:26:06.000000000 +0300
+++ b/code/qcommon/q_shared.h	2013-06-29 19:25:58.043861020 +0300
@@ -36,7 +36,7 @@
 #define CONSOLE_WINDOW_ICON   "iourbanterror console"
 // 1.32 released 7-10-2002
 
-#define BASEGAME              "q3ut4"
+#define BASEGAME              "baseut4"
 
 #define MAX_TEAMNAME 32
 
diff -Nuar a/code/sys/sys_unix.c b/code/sys/sys_unix.c
--- a/code/sys/sys_unix.c	2013-06-29 03:26:06.000000000 +0300
+++ b/code/sys/sys_unix.c	2013-07-09 13:06:28.767650507 +0300
@@ -58,7 +58,7 @@
 #ifdef MACOS_X
 			Q_strcat( homePath, sizeof( homePath ), "/Library/Application Support/Quake3" );
 #else
-			Q_strcat( homePath, sizeof( homePath ), "/.q3a" );
+			Q_strcat( homePath, sizeof( homePath ), "/.urbanterrorHD" );
 #endif
 			if( mkdir( homePath, 0777 ) )
 			{
diff -Nuar a/code/unix/unix_shared.c b/code/unix/unix_shared.c
--- a/code/unix/unix_shared.c	2013-06-29 03:26:06.000000000 +0300
+++ b/code/unix/unix_shared.c	2013-07-09 13:04:23.237654854 +0300
@@ -407,7 +407,7 @@
 #ifdef MACOS_X
 		Q_strcat(homePath, sizeof(homePath), "/Library/Application Support/Quake3");
 #else
-		Q_strcat(homePath, sizeof(homePath), "/.q3a");
+		Q_strcat(homePath, sizeof(homePath), "/.urbanterrorHD");
 #endif
 		if (mkdir(homePath, 0777)) {
 			if (errno != EEXIST)
