diff -Nur vim72-old/runtime/syntax/grub.vim vim72/runtime/syntax/grub.vim
--- vim72-old/runtime/syntax/grub.vim	2008-10-18 01:14:23.000000000 +0300
+++ vim72/runtime/syntax/grub.vim	2008-10-18 01:18:04.000000000 +0300
@@ -27,7 +27,7 @@
 
 syn cluster grubCommands      contains=grubCommand,grubTitleCommand
 
-syn keyword grubCommand       contained default fallback hiddenmenu timeout
+syn keyword grubCommand       contained default fallback hiddenmenu timeout splashimage background gfxmenu
 
 syn keyword grubTitleCommand  contained title nextgroup=grubTitle skipwhite
 
