Index: vim73/runtime/plugin/zipPlugin.vim
===================================================================
--- vim73.orig/runtime/plugin/zipPlugin.vim
+++ vim73/runtime/plugin/zipPlugin.vim
@@ -40,7 +40,7 @@ augroup zip
   au FileWriteCmd zipfile:*/*	call zip#Write(expand("<amatch>"))
  endif
 
- au BufReadCmd   *.zip,*.jar,*.xpi,*.war,*.ear		call zip#Browse(expand("<amatch>"))
+ au BufReadCmd   *.zip,*.jar,*.xpi,*.war,*.ear,*pisi		call zip#Browse(expand("<amatch>"))
 augroup END
 
 " ---------------------------------------------------------------------
