diff -Nur openal-soft-1.13-old//CMakeLists.txt openal-soft-1.13/CMakeLists.txt
--- openal-soft-1.13-old//CMakeLists.txt	2011-04-15 10:39:56.860000093 +0300
+++ openal-soft-1.13/CMakeLists.txt	2011-04-15 11:30:47.593000620 +0300
@@ -246,10 +246,10 @@
         MESSAGE(FATAL_ERROR "No sleep function found!")
     ENDIF()
 
-    CHECK_C_COMPILER_FLAG(-pthread HAVE_PTHREAD)
+    CHECK_C_COMPILER_FLAG(-lpthread HAVE_PTHREAD)
     IF(HAVE_PTHREAD)
-        ADD_DEFINITIONS(-pthread)
-        SET(EXTRA_LIBS ${EXTRA_LIBS} -pthread)
+        ADD_DEFINITIONS(-lpthread)
+        SET(EXTRA_LIBS ${EXTRA_LIBS} -lpthread)
     ENDIF()
 
     # We need pthreads outside of Windows
