--- PokerTH-1.0.1-src/src/net/common/ircthread.cpp	2013-04-07 05:04:27.006209660 -0700
+++ PokerTH-1.0.1-src/src/net/common/ircthread.cpp.new	2013-07-02 16:53:05.109914145 -0700
@@ -32,14 +32,14 @@
 #include <net/socket_helper.h>
 #include <net/ircthread.h>
 #include <net/socket_msg.h>
-#include <libircclient/libircclient.h>
+#include <libircclient.h>
 
 // We need to do the following to handle different versions of libircclient.
 // Sadly, libircclient doesn't have actual definitions for its versions in its headers.
 // However, we can use a definition that appeared in the same version we need
 // to check for. Hacky, but hey, it works.
 #ifdef LIBIRC_OPTION_SSL_NO_VERIFY
-#include <libircclient/libirc_rfcnumeric.h>
+#include <libirc_rfcnumeric.h>
 #endif
 
 #include <boost/algorithm/string/predicate.hpp>
