diff -rup PokerTH-0.9.3-src.orig/pokerth_game.pro PokerTH-0.9.3-src/pokerth_game.pro
--- PokerTH-0.9.3-src.orig/pokerth_game.pro	2012-02-11 17:58:59.000000000 +0100
+++ PokerTH-0.9.3-src/pokerth_game.pro	2012-02-13 21:27:50.379822273 +0100
@@ -497,7 +497,7 @@ mac {
 	QMAKE_CXXFLAGS -= -std=gnu++0x
 
 	# workaround for problems with boost_filesystem exceptions
-	QMAKE_LFLAGS += -no_dead_strip_inits_and_terms
+#	QMAKE_LFLAGS += -no_dead_strip_inits_and_terms
 
 	# for universal-compilation on PPC-Mac uncomment the following line
 	# on Intel-Mac you have to comment this line out or build will fail.
diff -rup PokerTH-0.9.3-src.orig/pokerth_server.pro PokerTH-0.9.3-src/pokerth_server.pro
--- PokerTH-0.9.3-src.orig/pokerth_server.pro	2012-02-11 17:58:59.000000000 +0100
+++ PokerTH-0.9.3-src/pokerth_server.pro	2012-02-13 21:27:50.380822222 +0100
@@ -288,7 +288,7 @@ mac {
 	QMAKE_CXXFLAGS -= -std=gnu++0x
 
 	# workaround for problems with boost_filesystem exceptions
-	QMAKE_LFLAGS += -no_dead_strip_inits_and_terms
+#	QMAKE_LFLAGS += -no_dead_strip_inits_and_terms
 
 	# for universal-compilation on PPC-Mac uncomment the following line
 	# on Intel-Mac you have to comment this line out or build will fail.
diff -rup PokerTH-0.9.3-src.orig/zlib_compress.pro PokerTH-0.9.3-src/zlib_compress.pro
--- PokerTH-0.9.3-src.orig/zlib_compress.pro	2012-02-11 17:58:59.000000000 +0100
+++ PokerTH-0.9.3-src/zlib_compress.pro	2012-02-13 21:27:50.380822222 +0100
@@ -98,7 +98,7 @@ mac{
 	QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.3
 
 	# workaround for problems with boost_filesystem exceptions
-	QMAKE_LFLAGS += -no_dead_strip_inits_and_terms
+#	QMAKE_LFLAGS += -no_dead_strip_inits_and_terms
 
 	# for universal-compilation on PPC-Mac uncomment the following line
 	# on Intel-Mac you have to comment this line out or build will fail.