Index: scorched/src/client/GLEXT/GLState.h
===================================================================
--- scorched.orig/src/client/GLEXT/GLState.h
+++ scorched/src/client/GLEXT/GLState.h
@@ -22,7 +22,7 @@
 #define AFX_GLSTATE_H__32B0E2D0_566D_4438_94E4_B12FE82430B1__INCLUDED_
 
 #define WIN32_LEAN_AND_MEAN		// Exclude rarely-used stuff from Windows headers
-#include "glew.h"
+#include <GL/glew.h>
 
 class GLState  
 {
Index: scorched/src/client/scorchedc/Makefile.am
===================================================================
--- scorched.orig/src/client/scorchedc/Makefile.am
+++ scorched/src/client/scorchedc/Makefile.am
@@ -37,7 +37,6 @@ scorched3dc_SOURCES = \
 	../../common/common/RandomGenerator.cpp \
 	../../common/common/RollingAverage.cpp \
 	../../common/common/sha2.cpp \
-	../../common/common/snprintf.c \
 	../../common/common/SplineCurve.cpp \
 	../../common/common/SplinePath.cpp \
 	../../common/common/StatsLogger.cpp \
@@ -431,7 +430,6 @@ scorched3dc_SOURCES = \
 	../dialogs/TutorialDialog.cpp \
 	../GLEXT/GLCamera.cpp \
 	../GLEXT/GLCameraFrustum.cpp \
-	../GLEXT/glew.c \
 	../GLEXT/GLFont2d.cpp \
 	../GLEXT/GLFont2dFreeType.cpp \
 	../GLEXT/GLFont2dStorage.cpp \
@@ -1168,7 +1166,6 @@ scorched3dc_SOURCES = \
 	../dialogs/TutorialDialog.h \
 	../GLEXT/GLCamera.h \
 	../GLEXT/GLCameraFrustum.h \
-	../GLEXT/glew.h \
 	../GLEXT/GLFont2d.h \
 	../GLEXT/GLFont2dFreeType.h \
 	../GLEXT/GLFont2dStorage.h \
@@ -1194,8 +1191,6 @@ scorched3dc_SOURCES = \
 	../GLEXT/GLTextureStore.h \
 	../GLEXT/GLVertexBufferObject.h \
 	../GLEXT/GLViewPort.h \
-	../GLEXT/glxew.h \
-	../GLEXT/wglew.h \
 	../GLSL/GLSLProgram.h \
 	../GLSL/GLSLShader.h \
 	../GLSL/GLSLShaderSetup.h \
@@ -1482,5 +1477,5 @@ scorched3dc_SOURCES = \
 	../../common/simactions/TankTeamBallanceSimAction.h
 
 AM_CPPFLAGS = -I../../common/porting -I../../common -I../../client -I../../server  @FFTW_CFLAGS@ @AL_CFLAGS@ @FT2_CFLAGS@ @OGG_CFLAGS@ @SDL_CFLAGS@
-LDADD = @FFTW_LIBS@ @AL_LIBS@ @FT2_LIBS@ @GL_LIBS@ @OGG_LIBS@ @SDL_LIBS@
+LDADD = @FFTW_LIBS@ @AL_LIBS@ @FT2_LIBS@ @GL_LIBS@ @OGG_LIBS@ @SDL_LIBS@ -lGLEW
 
Index: scorched/src/client/scorchedc/Makefile.in
===================================================================
--- scorched.orig/src/client/scorchedc/Makefile.in
+++ scorched/src/client/scorchedc/Makefile.in
@@ -67,7 +67,7 @@ am_scorched3dc_OBJECTS = ARGParser.$(OBJ
 	OptionsParameters.$(OBJEXT) OptionsScorched.$(OBJEXT) \
 	OptionsTransient.$(OBJEXT) ProgressCounter.$(OBJEXT) \
 	RandomGenerator.$(OBJEXT) RollingAverage.$(OBJEXT) \
-	sha2.$(OBJEXT) snprintf.$(OBJEXT) SplineCurve.$(OBJEXT) \
+	sha2.$(OBJEXT) SplineCurve.$(OBJEXT) \
 	SplinePath.$(OBJEXT) StatsLogger.$(OBJEXT) \
 	StatsLoggerDatabase.$(OBJEXT) StatsLoggerFile.$(OBJEXT) \
 	StatsLoggerMySQL.$(OBJEXT) StatsLoggerPGSQL.$(OBJEXT) \
@@ -259,7 +259,7 @@ am_scorched3dc_OBJECTS = ARGParser.$(OBJ
 	SkipDialog.$(OBJEXT) SoundDialog.$(OBJEXT) \
 	StartDialog.$(OBJEXT) TextBoxDialog.$(OBJEXT) \
 	TipDialog.$(OBJEXT) TutorialDialog.$(OBJEXT) \
-	GLCamera.$(OBJEXT) GLCameraFrustum.$(OBJEXT) glew.$(OBJEXT) \
+	GLCamera.$(OBJEXT) GLCameraFrustum.$(OBJEXT) \
 	GLFont2d.$(OBJEXT) GLFont2dFreeType.$(OBJEXT) \
 	GLFont2dStorage.$(OBJEXT) GLFrameBufferObject.$(OBJEXT) \
 	GLGlobalState.$(OBJEXT) GLImageItterator.$(OBJEXT) \
@@ -611,7 +611,6 @@ scorched3dc_SOURCES = \
 	../../common/common/RandomGenerator.cpp \
 	../../common/common/RollingAverage.cpp \
 	../../common/common/sha2.cpp \
-	../../common/common/snprintf.c \
 	../../common/common/SplineCurve.cpp \
 	../../common/common/SplinePath.cpp \
 	../../common/common/StatsLogger.cpp \
@@ -1005,7 +1004,6 @@ scorched3dc_SOURCES = \
 	../dialogs/TutorialDialog.cpp \
 	../GLEXT/GLCamera.cpp \
 	../GLEXT/GLCameraFrustum.cpp \
-	../GLEXT/glew.c \
 	../GLEXT/GLFont2d.cpp \
 	../GLEXT/GLFont2dFreeType.cpp \
 	../GLEXT/GLFont2dStorage.cpp \
@@ -1742,7 +1740,6 @@ scorched3dc_SOURCES = \
 	../dialogs/TutorialDialog.h \
 	../GLEXT/GLCamera.h \
 	../GLEXT/GLCameraFrustum.h \
-	../GLEXT/glew.h \
 	../GLEXT/GLFont2d.h \
 	../GLEXT/GLFont2dFreeType.h \
 	../GLEXT/GLFont2dStorage.h \
@@ -1768,8 +1765,6 @@ scorched3dc_SOURCES = \
 	../GLEXT/GLTextureStore.h \
 	../GLEXT/GLVertexBufferObject.h \
 	../GLEXT/GLViewPort.h \
-	../GLEXT/glxew.h \
-	../GLEXT/wglew.h \
 	../GLSL/GLSLProgram.h \
 	../GLSL/GLSLShader.h \
 	../GLSL/GLSLShaderSetup.h \
@@ -2056,7 +2051,7 @@ scorched3dc_SOURCES = \
 	../../common/simactions/TankTeamBallanceSimAction.h
 
 AM_CPPFLAGS = -I../../common/porting -I../../common -I../../client -I../../server  @FFTW_CFLAGS@ @AL_CFLAGS@ @FT2_CFLAGS@ @OGG_CFLAGS@ @SDL_CFLAGS@
-LDADD = @FFTW_LIBS@ @AL_LIBS@ @FT2_LIBS@ @GL_LIBS@ @OGG_LIBS@ @SDL_LIBS@
+LDADD = @FFTW_LIBS@ @AL_LIBS@ @FT2_LIBS@ @GL_LIBS@ @OGG_LIBS@ @SDL_LIBS@ -lglew
 all: all-am
 
 .SUFFIXES:
@@ -2840,7 +2835,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/aseFile.lex.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/aseFile.tab.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/fixed.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/glew.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/igd.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/lapi.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/lauxlib.Po@am__quote@
@@ -2874,7 +2868,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/lzio.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/main.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/sha2.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/snprintf.Po@am__quote@
 
 .c.o:
 @am__fastdepCC_TRUE@	$(COMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $<
@@ -2890,34 +2883,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(COMPILE) -c `$(CYGPATH_W) '$<'`
 
-snprintf.o: ../../common/common/snprintf.c
-@am__fastdepCC_TRUE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -MT snprintf.o -MD -MP -MF $(DEPDIR)/snprintf.Tpo -c -o snprintf.o `test -f '../../common/common/snprintf.c' || echo '$(srcdir)/'`../../common/common/snprintf.c
-@am__fastdepCC_TRUE@	$(am__mv) $(DEPDIR)/snprintf.Tpo $(DEPDIR)/snprintf.Po
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	source='../../common/common/snprintf.c' object='snprintf.o' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o snprintf.o `test -f '../../common/common/snprintf.c' || echo '$(srcdir)/'`../../common/common/snprintf.c
-
-snprintf.obj: ../../common/common/snprintf.c
-@am__fastdepCC_TRUE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -MT snprintf.obj -MD -MP -MF $(DEPDIR)/snprintf.Tpo -c -o snprintf.obj `if test -f '../../common/common/snprintf.c'; then $(CYGPATH_W) '../../common/common/snprintf.c'; else $(CYGPATH_W) '$(srcdir)/../../common/common/snprintf.c'; fi`
-@am__fastdepCC_TRUE@	$(am__mv) $(DEPDIR)/snprintf.Tpo $(DEPDIR)/snprintf.Po
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	source='../../common/common/snprintf.c' object='snprintf.obj' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o snprintf.obj `if test -f '../../common/common/snprintf.c'; then $(CYGPATH_W) '../../common/common/snprintf.c'; else $(CYGPATH_W) '$(srcdir)/../../common/common/snprintf.c'; fi`
-
-glew.o: ../GLEXT/glew.c
-@am__fastdepCC_TRUE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -MT glew.o -MD -MP -MF $(DEPDIR)/glew.Tpo -c -o glew.o `test -f '../GLEXT/glew.c' || echo '$(srcdir)/'`../GLEXT/glew.c
-@am__fastdepCC_TRUE@	$(am__mv) $(DEPDIR)/glew.Tpo $(DEPDIR)/glew.Po
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	source='../GLEXT/glew.c' object='glew.o' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o glew.o `test -f '../GLEXT/glew.c' || echo '$(srcdir)/'`../GLEXT/glew.c
-
-glew.obj: ../GLEXT/glew.c
-@am__fastdepCC_TRUE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -MT glew.obj -MD -MP -MF $(DEPDIR)/glew.Tpo -c -o glew.obj `if test -f '../GLEXT/glew.c'; then $(CYGPATH_W) '../GLEXT/glew.c'; else $(CYGPATH_W) '$(srcdir)/../GLEXT/glew.c'; fi`
-@am__fastdepCC_TRUE@	$(am__mv) $(DEPDIR)/glew.Tpo $(DEPDIR)/glew.Po
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	source='../GLEXT/glew.c' object='glew.obj' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o glew.obj `if test -f '../GLEXT/glew.c'; then $(CYGPATH_W) '../GLEXT/glew.c'; else $(CYGPATH_W) '$(srcdir)/../GLEXT/glew.c'; fi`
-
 .cpp.o:
 @am__fastdepCXX_TRUE@	$(CXXCOMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $<
 @am__fastdepCXX_TRUE@	$(am__mv) $(DEPDIR)/$*.Tpo $(DEPDIR)/$*.Po
Index: scorched/src/launcher/scorched/Makefile.am
===================================================================
--- scorched.orig/src/launcher/scorched/Makefile.am
+++ scorched/src/launcher/scorched/Makefile.am
@@ -36,7 +36,6 @@ scorched3d_SOURCES = \
 	../../common/common/OptionsParameters.cpp \
 	../../common/common/ProgressCounter.cpp \
 	../../common/common/RandomGenerator.cpp \
-	../../common/common/snprintf.c \
 	../../common/common/Vector.cpp \
 	../../common/common/Vector4.cpp \
 	../../common/common/VectorLib.cpp \
Index: scorched/src/launcher/scorched/Makefile.in
===================================================================
--- scorched.orig/src/launcher/scorched/Makefile.in
+++ scorched/src/launcher/scorched/Makefile.in
@@ -67,7 +67,7 @@ am_scorched3d_OBJECTS = main.$(OBJEXT) S
 	LoggerI.$(OBJEXT) NumberParser.$(OBJEXT) OptionEntry.$(OBJEXT) \
 	OptionsGame.$(OBJEXT) OptionsParameters.$(OBJEXT) \
 	ProgressCounter.$(OBJEXT) RandomGenerator.$(OBJEXT) \
-	snprintf.$(OBJEXT) Vector.$(OBJEXT) Vector4.$(OBJEXT) \
+	Vector.$(OBJEXT) Vector4.$(OBJEXT) \
 	VectorLib.$(OBJEXT) XMLFile.$(OBJEXT) XMLNode.$(OBJEXT) \
 	XMLParser.$(OBJEXT) XMLStringBuffer.$(OBJEXT) \
 	OptionsDisplay.$(OBJEXT) ModDirs.$(OBJEXT) \
@@ -273,7 +273,6 @@ scorched3d_SOURCES = \
 	../../common/common/OptionsParameters.cpp \
 	../../common/common/ProgressCounter.cpp \
 	../../common/common/RandomGenerator.cpp \
-	../../common/common/snprintf.c \
 	../../common/common/Vector.cpp \
 	../../common/common/Vector4.cpp \
 	../../common/common/VectorLib.cpp \
@@ -525,7 +524,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/XMLStringBuffer.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/fixed.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/main.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/snprintf.Po@am__quote@
 
 .c.o:
 @am__fastdepCC_TRUE@	$(COMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $<
@@ -541,20 +539,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(COMPILE) -c `$(CYGPATH_W) '$<'`
 
-snprintf.o: ../../common/common/snprintf.c
-@am__fastdepCC_TRUE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -MT snprintf.o -MD -MP -MF $(DEPDIR)/snprintf.Tpo -c -o snprintf.o `test -f '../../common/common/snprintf.c' || echo '$(srcdir)/'`../../common/common/snprintf.c
-@am__fastdepCC_TRUE@	$(am__mv) $(DEPDIR)/snprintf.Tpo $(DEPDIR)/snprintf.Po
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	source='../../common/common/snprintf.c' object='snprintf.o' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o snprintf.o `test -f '../../common/common/snprintf.c' || echo '$(srcdir)/'`../../common/common/snprintf.c
-
-snprintf.obj: ../../common/common/snprintf.c
-@am__fastdepCC_TRUE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -MT snprintf.obj -MD -MP -MF $(DEPDIR)/snprintf.Tpo -c -o snprintf.obj `if test -f '../../common/common/snprintf.c'; then $(CYGPATH_W) '../../common/common/snprintf.c'; else $(CYGPATH_W) '$(srcdir)/../../common/common/snprintf.c'; fi`
-@am__fastdepCC_TRUE@	$(am__mv) $(DEPDIR)/snprintf.Tpo $(DEPDIR)/snprintf.Po
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	source='../../common/common/snprintf.c' object='snprintf.obj' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o snprintf.obj `if test -f '../../common/common/snprintf.c'; then $(CYGPATH_W) '../../common/common/snprintf.c'; else $(CYGPATH_W) '$(srcdir)/../../common/common/snprintf.c'; fi`
-
 .cpp.o:
 @am__fastdepCXX_TRUE@	$(CXXCOMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $<
 @am__fastdepCXX_TRUE@	$(am__mv) $(DEPDIR)/$*.Tpo $(DEPDIR)/$*.Po
Index: scorched/src/server/scorcheds/Makefile.am
===================================================================
--- scorched.orig/src/server/scorcheds/Makefile.am
+++ scorched/src/server/scorcheds/Makefile.am
@@ -53,7 +53,6 @@ scorched3ds_SOURCES = \
 	../../common/common/RandomGenerator.cpp \
 	../../common/common/RollingAverage.cpp \
 	../../common/common/sha2.cpp \
-	../../common/common/snprintf.c \
 	../../common/common/SplineCurve.cpp \
 	../../common/common/SplinePath.cpp \
 	../../common/common/StatsLogger.cpp \
Index: scorched/src/server/scorcheds/Makefile.in
===================================================================
--- scorched.orig/src/server/scorcheds/Makefile.in
+++ scorched/src/server/scorcheds/Makefile.in
@@ -74,7 +74,7 @@ am_scorched3ds_OBJECTS = ComsTankChangeM
 	OptionsMasterListServer.$(OBJEXT) OptionsParameters.$(OBJEXT) \
 	OptionsScorched.$(OBJEXT) OptionsTransient.$(OBJEXT) \
 	ProgressCounter.$(OBJEXT) RandomGenerator.$(OBJEXT) \
-	RollingAverage.$(OBJEXT) sha2.$(OBJEXT) snprintf.$(OBJEXT) \
+	RollingAverage.$(OBJEXT) sha2.$(OBJEXT) \
 	SplineCurve.$(OBJEXT) SplinePath.$(OBJEXT) \
 	StatsLogger.$(OBJEXT) StatsLoggerDatabase.$(OBJEXT) \
 	StatsLoggerFile.$(OBJEXT) StatsLoggerMySQL.$(OBJEXT) \
@@ -489,7 +489,6 @@ scorched3ds_SOURCES = \
 	../../common/common/RandomGenerator.cpp \
 	../../common/common/RollingAverage.cpp \
 	../../common/common/sha2.cpp \
-	../../common/common/snprintf.c \
 	../../common/common/SplineCurve.cpp \
 	../../common/common/SplinePath.cpp \
 	../../common/common/StatsLogger.cpp \
@@ -1902,7 +1901,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/lzio.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/main.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/sha2.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/snprintf.Po@am__quote@
 
 .c.o:
 @am__fastdepCC_TRUE@	$(COMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $<
@@ -1918,20 +1916,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(COMPILE) -c `$(CYGPATH_W) '$<'`
 
-snprintf.o: ../../common/common/snprintf.c
-@am__fastdepCC_TRUE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -MT snprintf.o -MD -MP -MF $(DEPDIR)/snprintf.Tpo -c -o snprintf.o `test -f '../../common/common/snprintf.c' || echo '$(srcdir)/'`../../common/common/snprintf.c
-@am__fastdepCC_TRUE@	$(am__mv) $(DEPDIR)/snprintf.Tpo $(DEPDIR)/snprintf.Po
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	source='../../common/common/snprintf.c' object='snprintf.o' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o snprintf.o `test -f '../../common/common/snprintf.c' || echo '$(srcdir)/'`../../common/common/snprintf.c
-
-snprintf.obj: ../../common/common/snprintf.c
-@am__fastdepCC_TRUE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -MT snprintf.obj -MD -MP -MF $(DEPDIR)/snprintf.Tpo -c -o snprintf.obj `if test -f '../../common/common/snprintf.c'; then $(CYGPATH_W) '../../common/common/snprintf.c'; else $(CYGPATH_W) '$(srcdir)/../../common/common/snprintf.c'; fi`
-@am__fastdepCC_TRUE@	$(am__mv) $(DEPDIR)/snprintf.Tpo $(DEPDIR)/snprintf.Po
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	source='../../common/common/snprintf.c' object='snprintf.obj' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o snprintf.obj `if test -f '../../common/common/snprintf.c'; then $(CYGPATH_W) '../../common/common/snprintf.c'; else $(CYGPATH_W) '$(srcdir)/../../common/common/snprintf.c'; fi`
-
 .cpp.o:
 @am__fastdepCXX_TRUE@	$(CXXCOMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $<
 @am__fastdepCXX_TRUE@	$(am__mv) $(DEPDIR)/$*.Tpo $(DEPDIR)/$*.Po
