diff -Naur Numeric-24.2/Src/arrayobject.c Numeric-24.2-new/Src/arrayobject.c
--- Numeric-24.2/Src/arrayobject.c	2005-11-11 02:30:18.000000000 -0500
+++ Numeric-24.2-new/Src/arrayobject.c	2007-09-01 22:52:50.000000000 -0400
@@ -1866,9 +1866,7 @@
     for (i=0; i<sz; i++) {
 	v=array_item((PyArrayObject *)self, i);
 	PyList_SetItem(lp, i, PyArray_ToList(v));
-	if (((PyArrayObject *)self)->nd>1){
-	    Py_DECREF(v);
-	}
+	Py_DECREF(v);
     }
 
     return lp;
