Fix parallel build

http://bugs.gentoo.org/136373

--- xfsdump-3.0.0/dump/Makefile
+++ xfsdump-3.0.0/dump/Makefile
@@ -106,3 +106,5 @@
 
 $(INVINCL) $(INVCOMMON):
 	$(RM) $@; $(LN_S) ../inventory/$@ $@
+
+$(LOCALS): $(LINKS)
--- xfsdump-3.0.0/invutil/Makefile
+++ xfsdump-3.0.0/invutil/Makefile
@@ -70,3 +70,5 @@
 
 $(INVINCL) $(INVCOMMON):
 	$(RM) $@; $(LN_S) ../inventory/$@ $@
+
+$(LOCALS): $(LINKS)
--- xfsdump-3.0.0/restore/Makefile
+++ xfsdump-3.0.0/restore/Makefile
@@ -116,3 +116,5 @@
 
 $(INVINCL) $(INVCOMMON):
 	$(RM) $@; $(LN_S) ../inventory/$@ $@
+
+$(LOCALS): $(LINKS)
