diff -Nur mingetty-1.0.7s/mingetty.c mingetty-1.0.7s-patched/mingetty.c
--- mingetty-1.0.7s/mingetty.c	2008-07-30 11:06:20.000000000 +0300
+++ mingetty-1.0.7s-patched/mingetty.c	2008-07-30 11:09:39.000000000 +0300
@@ -746,6 +746,10 @@
 	if (priority)
 		nice (priority);
 
+    /* Export LC_ALL to fix the localization problem of /bin/login */
+    if (default_locale)
+        setenv("LC_ALL", default_locale, 1);
+
 	execv (loginprog, logarr);
 	error (_("%s: can't exec %s: %s"), tty, loginprog, strerror (errno));
 	sleep (1);
