diff -Nur telepathy-salut-0.3.11-old/configure.ac telepathy-salut-0.3.11/configure.ac
--- telepathy-salut-0.3.11-old/configure.ac	2010-04-06 13:21:58.592842887 +0300
+++ telepathy-salut-0.3.11/configure.ac	2010-04-06 13:23:33.924588464 +0300
@@ -33,6 +33,10 @@
 AM_PROG_AS
 AM_PROG_MKDIR_P
 
+dnl make sure locale does not mess with upper/lower conversions
+LC_ALL="C"
+AC_SUBST(LC_ALL)
+
 dnl add common command line switches
 SALUT_ARG_DEBUG
 SALUT_ARG_VALGRIND
