--- ppp-2.4.3/pppd/plugins/dhcp/Makefile.linux
+++ ppp-2.4.3/pppd/plugins/dhcp/Makefile.linux
@@ -1,6 +1,6 @@
 
 PLUGIN=dhcpc.so
-CFLAGS=-I../.. -I../../../include -O2
+CFLAGS=$(COPTS) -I../.. -I../../../include -fPIC
 
 all: $(PLUGIN)
 
@@ -9,7 +9,7 @@
 	$(INSTALL) -s -c -m 755 dhcpc.so $(LIBDIR)
 
 dhcpc.so: dhcpc.o clientpacket.o packet.o socket.o options.o 
-	gcc -o dhcpc.so -shared dhcpc.o clientpacket.o packet.o socket.o options.o
+	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ $^
 
 
 clean:
