diff -up notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c.base-color notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c
--- notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c.base-color	2009-06-20 08:51:58.000000000 +0200
+++ notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c	2009-06-20 09:00:55.000000000 +0200
@@ -346,11 +346,26 @@ fill_background(GtkWidget *widget, Windo
 	else
 		alpha = 1.0;
 
+	GtkStyle *style = gtk_widget_get_style(widget);
+	GdkColor *background_color = &style->base[GTK_STATE_NORMAL];
 	cairo_pattern_t *pattern;
 	pattern = cairo_pattern_create_linear (0, 0, 0, windata->height);
-	cairo_pattern_add_color_stop_rgba (pattern, 0, 0.996, 0.996, 0.89, alpha);
-	cairo_pattern_add_color_stop_rgba (pattern, GRADIENT_CENTER, 0.988, 0.988, 0.714, alpha);
-	cairo_pattern_add_color_stop_rgba (pattern, 1, 0.984, 0.984, 0.663, alpha);
+	cairo_pattern_add_color_stop_rgba (pattern, 0, 
+					   background_color->red / 65535.0, 
+					   background_color->green / 65535.0, 
+					   background_color->blue / 65535.0, 
+					   alpha);
+	cairo_pattern_add_color_stop_rgba (pattern, GRADIENT_CENTER, 
+					   background_color->red / 65535.0 * 0.80, 
+					   background_color->green / 65535.0 * 0.80, 
+					   background_color->blue / 65535.0 * 0.80, 
+					   alpha);
+
+	cairo_pattern_add_color_stop_rgba (pattern, 1, 
+					   background_color->red / 65535.0 * 0.74, 
+					   background_color->green / 65535.0 * 0.74, 
+					   background_color->blue / 65535.0 * 0.74, 
+					   alpha);
 	cairo_set_source (cr, pattern);
 	cairo_pattern_destroy (pattern);
 	
@@ -453,10 +468,20 @@ draw_border(GtkWidget *widget, WindowDat
 	else
 		alpha = 1.0;
 
+	GtkStyle *style = gtk_widget_get_style(widget);
+	GdkColor *background_color = &style->base[GTK_STATE_NORMAL];
 	cairo_pattern_t *pattern;
 	pattern = cairo_pattern_create_linear (0, 0, 0, windata->height);
-	cairo_pattern_add_color_stop_rgba (pattern, 0, 0.62, 0.584, 0.341, alpha);
-	cairo_pattern_add_color_stop_rgba (pattern, 1, 0.776, 0.757, 0.596, alpha);
+	cairo_pattern_add_color_stop_rgba (pattern, 0,
+					   background_color->red / 65535.0 * 0.38, 
+					   background_color->green / 65535.0 * 0.38, 
+					   background_color->blue / 65535.0 * 0.38, 
+					   alpha);
+	cairo_pattern_add_color_stop_rgba (pattern, 1,
+					   background_color->red / 65535.0 * 0.67, 
+					   background_color->green / 65535.0 * 0.67, 
+					   background_color->blue / 65535.0 * 0.67, 
+					   alpha);
 	cairo_set_source (cr, pattern);
 	cairo_pattern_destroy (pattern);
 	
