diff -up notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c.rtl notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c
--- notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c.rtl	2008-12-27 13:00:10.000000000 +0100
+++ notification-daemon-engine-nodoka-0.1.0/src/nodoka-theme.c	2008-12-27 13:06:27.000000000 +0100
@@ -73,6 +73,7 @@ typedef struct
 
 	UrlClickedCb url_clicked;
 
+	GtkTextDirection rtl;
 } WindowData;
 
 
@@ -367,7 +368,9 @@ static void
 draw_stripe(GtkWidget *widget, WindowData *windata, cairo_t *cr)
 {
 	cairo_save (cr);
-	cairo_rectangle (cr, 0, 0, STRIPE_WIDTH, windata->height);
+	cairo_rectangle (cr, (windata->rtl == GTK_TEXT_DIR_RTL) ? 
+	                     windata->width-STRIPE_WIDTH : 0, 0, 
+	                 STRIPE_WIDTH, windata->height);
 	cairo_clip (cr);
 
 	GdkColor top_color;
@@ -657,6 +660,7 @@ create_notification(UrlClickedCb url_cli
 	win = gtk_window_new(GTK_WINDOW_POPUP);
 	windata->win = win;
 
+	windata->rtl = gtk_widget_get_direction(win);
 	windata->enable_transparency = FALSE;
 	screen = gtk_window_get_screen(GTK_WINDOW(win));
 	colormap = gdk_screen_get_rgba_colormap(screen);
