diff -up lftp-4.2.1/src/Http.cc.date_fmt lftp-4.2.1/src/Http.cc
--- lftp-4.2.1/src/Http.cc.date_fmt	2011-03-25 15:35:42.000000000 +0100
+++ lftp-4.2.1/src/Http.cc	2011-03-30 14:40:54.170096624 +0200
@@ -29,6 +29,7 @@
 #include <errno.h>
 #include <stdarg.h>
 #include <time.h>
+#include <limits.h>
 #include <fnmatch.h>
 #include <locale.h>
 #include <assert.h>
@@ -565,15 +566,10 @@ void Http::SendRequest(const char *conne
 		     (long long)((limit==FILE_END || limit>entity_size ? entity_size : limit)-1),
 		     (long long)entity_size);
       }
-      if(entity_date!=NO_DATE)
+      if(entity_date!=NO_DATE && entity_date>0L && entity_date<INT_MAX)
       {
-	 static const char weekday_names[][4]={
-	    "Sun","Mon","Tue","Wed","Thu","Fri","Sat"
-	 };
-	 const struct tm *t=gmtime(&entity_date);
-	 const char *d=xstring::format("%s, %2d %s %04d %02d:%02d:%02d GMT",
-	    weekday_names[t->tm_wday],t->tm_mday,month_names[t->tm_mon],
-	    t->tm_year+1900,t->tm_hour,t->tm_min,t->tm_sec);
+         char d[256];
+         strftime(d, sizeof(d), "%a, %d %b %H:%M:%S %Y GMT", gmtime(&entity_date));
 	 Send("Last-Modified: %s\r\n",d);
       }
       break;
