Index: man-db-2.5.2/src/man_db.conf.in
===================================================================
--- man-db-2.5.2.orig/src/man_db.conf.in
+++ man-db-2.5.2/src/man_db.conf.in
@@ -69,6 +69,8 @@ MANDB_MAP	/usr/local/man		/var/cache/man
 MANDB_MAP	/usr/local/share/man	/var/cache/man/local
 MANDB_MAP	/usr/X11R6/man		/var/cache/man/X11R6
 MANDB_MAP	/opt/man		/var/cache/man/opt
+MANDB_MAP	/opt/sun-jre/man	/var/cache/man/sun-jre
+MANDB_MAP	/opt/sun-jdk/man	/var/cache/man/sun-jdk
 #
 #---------------------------------------------------------
 # Program definitions.  These are commented out by default as the value
