diff -up trunk/src/filenames.c.double-free trunk/src/filenames.c
--- trunk/src/filenames.c.double-free	2011-01-10 21:02:05.000000000 +0100
+++ trunk/src/filenames.c	2011-10-04 16:49:42.146071708 +0200
@@ -85,6 +85,8 @@ char *filename_info (const char *file, s
 	char *slash = strrchr (manpage, '/');
 	char *base_name;
 
+	memset (info, 0, sizeof (struct mandata));
+
 #ifdef COMP_SRC
 	struct compression *comp;
 #endif
