diff -up man-db-2.5.9/include/manconfig.h.in.sgr man-db-2.5.9/include/manconfig.h.in
--- man-db-2.5.9/include/manconfig.h.in.sgr	2010-11-17 12:30:36.000000000 +0100
+++ man-db-2.5.9/include/manconfig.h.in	2010-11-24 11:29:57.000000000 +0100
@@ -128,7 +128,7 @@
 
 #ifndef NROFF_MISSING
 #  ifndef NROFF
-#    define NROFF 	"@nroff@"
+#    define NROFF 	"@nroff@ -c"
 #  endif
 #endif
 
diff -up man-db-2.5.9/src/man_db.conf.in.sgr man-db-2.5.9/src/man_db.conf.in
--- man-db-2.5.9/src/man_db.conf.in.sgr	2009-03-15 13:17:24.000000000 +0100
+++ man-db-2.5.9/src/man_db.conf.in	2010-11-24 11:27:45.000000000 +0100
@@ -80,7 +80,7 @@ MANDB_MAP	/opt/man		/var/cache/man/opt
 #DEFINE 	tr	@tr@ '\255\267\264\327' '\055\157\047\170'
 #DEFINE		grep	@grep@
 #DEFINE 	troff 	@troff@
-#DEFINE 	nroff 	@nroff@
+#DEFINE 	nroff 	@nroff@ -c
 #DEFINE 	eqn 	@eqn@
 #DEFINE 	neqn	@neqn@
 #DEFINE 	tbl 	@tbl@
