Index: stlport5.2-5.2.1/build/lib/gcc.mak
===================================================================
--- stlport5.2-5.2.1.orig/build/lib/gcc.mak	2008-12-10 10:56:51.000000000 +0100
+++ stlport5.2-5.2.1/build/lib/gcc.mak	2009-01-11 12:18:34.000000000 +0100
@@ -12,7 +12,7 @@
 include ${SRCROOT}/Makefiles/gmake/subdirs.mak
 
 ifeq ($(OSNAME),linux)
-DEFS += -D_GNU_SOURCE
+DEFS += -D_GNU_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
 endif
 
 # options for build with boost support
Index: stlport5.2-5.2.1/build/test/unit/gcc.mak
===================================================================
--- stlport5.2-5.2.1.orig/build/test/unit/gcc.mak	2008-12-10 10:56:51.000000000 +0100
+++ stlport5.2-5.2.1/build/test/unit/gcc.mak	2009-01-11 12:17:43.000000000 +0100
@@ -25,6 +25,10 @@
 stldbg-shared:	DEFS += -D_STLP_DEBUG_UNINITIALIZED
 endif
 
+ifeq ($(OSNAME), linux)
+DEFS += -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
+endif
+
 ifdef STLP_BUILD_BOOST_PATH
 INCLUDES += -I${STLP_BUILD_BOOST_PATH}
 endif
