diff -Naur dbus-python-0.83.0.orig/dbus/connection.py dbus-python-0.83.0/dbus/connection.py
--- dbus-python-0.83.0.orig/dbus/connection.py	2009-04-22 13:58:33.000000000 +0300
+++ dbus-python-0.83.0/dbus/connection.py	2009-04-22 13:58:40.000000000 +0300
@@ -239,7 +239,7 @@
     ProxyObjectClass = ProxyObject
 
     def __init__(self, *args, **kwargs):
-        super(Connection, self).__init__(*args, **kwargs)
+        super(Connection, self).__init__()
 
         # this if-block is needed because shared bus connections can be
         # __init__'ed more than once
