Index: Docs/sample/gnokiirc
===================================================================
--- Docs/sample/gnokiirc.orig
+++ Docs/sample/gnokiirc
@@ -27,7 +27,10 @@ port = none
 # or similiar.
 #
 # Use this setting also for the Bluetooth connection:
-# port = aa:bb:cc:dd:ee:ff
+
+port = aa:bb:cc:dd:ee:ff
+rfcomm_channel = 14
+
 # when using it with AT driver set it to:
 # port = /dev/rfcomm0
 # or similiar.
@@ -95,7 +98,9 @@ initlength = default
 # you don't use model = AT) you may want to use 'dlr3p' value here.
 # Note that it is recommended and currently the best way to use this
 # cable with nk6510 driver.
-connection = serial
+
+connection = bluetooth
+
 # With DKU-2 cable use the following setting if you want to libusb driver
 # (recommended):
 # connection = dku2libusb
@@ -118,7 +123,8 @@ connection = serial
 # port software (eg. minicom). If you have wrong permissions for the
 # directory, a warning will be generated. If you don't want a lockfile, set
 # it to 'no'.
-use_locking = yes
+
+use_locking = no
 
 # Baudrate to use on serial port connections.
 # Currently used only by models AT and BIP/CIMD. Defaults to 19200.
@@ -181,7 +187,8 @@ allow_breakage = 0
 # permissions 4750, owned by root, group gnokii.  Ensure you
 # are in the gnokii group and that the group exists...
 [gnokiid]
-bindir = /usr/local/sbin/
+
+bindir = /usr/sbin/
 
 # Any entries in the following two sections will be set as environment
 # variables when running the scripts.
