--- xgnokii/xgnokii_cfg.c~	2008-10-06 10:26:55.000000000 +0200
+++ xgnokii/xgnokii_cfg.c	2008-10-06 10:29:42.000000000 +0200
@@ -75,7 +75,7 @@
 	xgnokiiConfig.user.fax = g_strdup("");
 	xgnokiiConfig.user.email = g_strdup("");
 	xgnokiiConfig.user.address = g_strdup("");
-	xgnokiiConfig.helpviewer = g_strdup("firefox");
+	xgnokiiConfig.helpviewer = g_strdup("xdg-open");
 	if ((homedir = g_get_home_dir()) == NULL)
 		homedir = "";
 	xgnokiiConfig.mailbox = g_strdup_printf("%s/Mail/smsbox", homedir);
