diff -Nur pyclutter-1.0.0-old/clutter/Makefile.am pyclutter-1.0.0/clutter/Makefile.am
--- pyclutter-1.0.0-old/clutter/Makefile.am	2009-10-04 04:48:25.706884253 +0300
+++ pyclutter-1.0.0/clutter/Makefile.am	2009-10-04 04:53:22.949144455 +0300
@@ -77,7 +77,7 @@
 
 _clutter_la_CFLAGS  = $(PYCAIRO_CFLAGS) $(CLUTTER_CFLAGS)
 _clutter_la_LDFLAGS = $(pyclutter_common_ldflags) -export-symbols-regex init_clutter
-_clutter_la_LIBADD  = $(PYCAIRO_LIBS) $(PYGOBJECT_LIBS) $(CLUTTER_LIBS)
+_clutter_la_LIBADD  = $(PYCAIRO_LIBS) $(PYGOBJECT_LIBS) $(CLUTTER_LIBS) -lpardusPythonVersion
 _clutter_la_SOURCES = 		\
 	cluttermodule.c		\
         pyclutter-private.h     \
