Index: git-1.7.6.4/gitweb/Makefile
===================================================================
--- git-1.7.6.4.orig/gitweb/Makefile
+++ git-1.7.6.4/gitweb/Makefile
@@ -56,7 +56,7 @@ PERL_PATH  ?= /usr/bin/perl
 # Shell quote;
 bindir_SQ = $(subst ','\'',$(bindir))#'
 gitwebdir_SQ = $(subst ','\'',$(gitwebdir))#'
-gitwebstaticdir_SQ = $(subst ','\'',$(gitwebdir)/static)#'
+gitwebstaticdir_SQ = $(subst ','\'',$(gitwebdir)/gitweb)#'
 SHELL_PATH_SQ = $(subst ','\'',$(SHELL_PATH))#'
 PERL_PATH_SQ  = $(subst ','\'',$(PERL_PATH))#'
 DESTDIR_SQ    = $(subst ','\'',$(DESTDIR))#'
