--- clutter-qt-0.9_20090720/clutter-qt/clutter-qt.h~	2009-07-20 08:25:24.000000000 -0300
+++ clutter-qt-0.9_20090720/clutter-qt/clutter-qt.h	2010-12-03 13:42:30.000000000 -0200
@@ -24,6 +24,12 @@
 
 #include <QWidget>
 #include <QInputEvent>
+
+//work around compile issue with glib >= 2.25
+#ifdef signals
+#undef signals
+#endif
+
 #include <clutter/clutter.h>
 
 class ClutterQtPrivate;
--- clutter-qt-0.9_20090720/clutter-qt/clutter-qt.cc~	2009-07-20 08:25:24.000000000 -0300
+++ clutter-qt-0.9_20090720/clutter-qt/clutter-qt.cc	2010-12-03 13:48:56.000000000 -0200
@@ -26,6 +26,11 @@
 #include <QTime>
 #include <QX11Info>
 
+//work around compile issue with glib >= 2.25
+#ifdef signals
+#undef signals
+#endif
+
 #include <clutter/clutter.h>
 #include <clutter/x11/clutter-x11.h>
 
--- clutter-qt-0.9_20090720/examples/testqt.h~	2009-07-20 08:25:24.000000000 -0300
+++ clutter-qt-0.9_20090720/examples/testqt.h	2010-12-03 13:55:03.000000000 -0200
@@ -26,6 +26,11 @@
 #include <QGridLayout>
 #include <QLineEdit>
 
+//work around compile issue with glib >= 2.25
+#ifdef signals
+#undef signals
+#endif
+
 #include <clutter/clutter.h>
 #include <clutter-qt/clutter-qt.h>
 
--- clutter-qt-0.9_20090720/examples/testqt.cc~	2009-07-20 08:25:24.000000000 -0300
+++ clutter-qt-0.9_20090720/examples/testqt.cc	2010-12-03 13:58:31.000000000 -0200
@@ -28,6 +28,11 @@
 #include <QCheckBox>
 #include <glib.h>
 
+//work around compile issue with glib >= 2.25
+#ifdef signals
+#undef signals
+#endif
+
 #include <clutter/clutter.h>
 #include <clutter-qt/clutter-qt.h>
 #include "testqt.h"
