diff -urN clutter-qt-0.9~20090720.orig/clutter-qt/Makefile.am clutter-qt-0.9~20090720/clutter-qt/Makefile.am
--- clutter-qt-0.9~20090720.orig/clutter-qt/Makefile.am	2009-07-20 12:25:24.000000000 +0100
+++ clutter-qt-0.9~20090720/clutter-qt/Makefile.am	2009-07-29 14:44:27.000000000 +0100
@@ -22,7 +22,7 @@
 libclutter_qt_@CLUTTER_QT_API_VERSION@_la_LIBADD  = $(CLUTTER_LIBS) $(QT_LIBS)
 libclutter_qt_@CLUTTER_QT_API_VERSION@_la_LDFLAGS = $(CLUTTER_LT_LDFLAGS)
 
-clutterqtheadersdir = $(includedir)/clutter-@CLUTTER_QT_API_VERSION@/clutter-qt
+clutterqtheadersdir = $(includedir)/clutter-1.0/clutter-qt
 clutterqtheaders_HEADERS = clutter-qt.h
 
 moc_clutter-qt.cc: clutter-qt.h
diff -urN clutter-qt-0.9~20090720.orig/clutter-qt.pc.in clutter-qt-0.9~20090720/clutter-qt.pc.in
--- clutter-qt-0.9~20090720.orig/clutter-qt.pc.in	2009-07-20 12:25:24.000000000 +0100
+++ clutter-qt-0.9~20090720/clutter-qt.pc.in	2009-07-29 14:45:17.000000000 +0100
@@ -7,5 +7,5 @@
 Description: QT widget for Clutter
 Version: @VERSION@
 Libs: -L${libdir} -lclutter-qt-@CLUTTER_QT_API_VERSION@
-Cflags: -I${includedir}/clutter-@CLUTTER_QT_API_VERSION@/clutter-qt
-Requires: QtGui clutter-x11-@CLUTTER_QT_API_VERSION@
+Cflags: -I${includedir}/clutter-1.0/clutter-qt
+Requires: QtGui clutter-x11-1.0
diff -urN clutter-qt-0.9~20090720.orig/configure.ac clutter-qt-0.9~20090720/configure.ac
--- clutter-qt-0.9~20090720.orig/configure.ac	2009-07-20 12:25:24.000000000 +0100
+++ clutter-qt-0.9~20090720/configure.ac	2009-07-29 14:43:55.000000000 +0100
@@ -81,7 +81,7 @@
 AC_SUBST(CLUTTER_CFLAGS)
 AC_SUBST(CLUTTER_LIBS)
 
-PKG_CHECK_MODULES(QT, QtCore >= 4.2 QtGui >= 4.2 glib-2.0 clutter-0.9)
+PKG_CHECK_MODULES(QT, QtCore >= 4.2 QtGui >= 4.2 glib-2.0 clutter-1.0)
 AC_SUBST(QT_CFLAGS)
 AC_SUBST(QT_LIBS)
 
