diff -Nuar pygtkglext-1.1.0.orig/gtk/gdkgl/Makefile.am pygtkglext-1.1.0/gtk/gdkgl/Makefile.am
--- pygtkglext-1.1.0.orig/gtk/gdkgl/Makefile.am	2010-09-06 12:08:45.174162781 +0300
+++ pygtkglext-1.1.0/gtk/gdkgl/Makefile.am	2010-09-06 12:17:11.667162049 +0300
@@ -24,7 +24,7 @@
 pygdkglexec_LTLIBRARIES = _gdkgl.la
 
 _gdkgl_la_CFLAGS = $(GTKGLEXT_CFLAGS)
-_gdkgl_la_LDFLAGS = $(common_ldflags) -export-symbols-regex init_gdkgl
+_gdkgl_la_LDFLAGS = $(common_ldflags) -export-symbols-regex init_gdkgl `python-config --libs`
 _gdkgl_la_LIBADD = $(GTKGLEXT_LIBS)
 _gdkgl_la_SOURCES = gdkglmodule.c
 nodist__gdkgl_la_SOURCES = gdkglext.c
diff -Nuar pygtkglext-1.1.0.orig/gtk/gtkgl/Makefile.am pygtkglext-1.1.0/gtk/gtkgl/Makefile.am
--- pygtkglext-1.1.0.orig/gtk/gtkgl/Makefile.am	2010-09-06 12:08:45.173162338 +0300
+++ pygtkglext-1.1.0/gtk/gtkgl/Makefile.am	2010-09-06 12:10:14.417162515 +0300
@@ -24,7 +24,7 @@
 pygtkglexec_LTLIBRARIES = _gtkgl.la
 
 _gtkgl_la_CFLAGS = $(GTKGLEXT_CFLAGS)
-_gtkgl_la_LDFLAGS = $(common_ldflags) -export-symbols-regex init_gtkgl
+_gtkgl_la_LDFLAGS = $(common_ldflags) -export-symbols-regex init_gtkgl `python-config --libs`
 _gtkgl_la_LIBADD = $(GTKGLEXT_LIBS)
 _gtkgl_la_SOURCES = gtkglmodule.c
 nodist__gtkgl_la_SOURCES = gtkglext.c
