--- configure.OSmanOS	2013-01-25 09:57:56.000000000 +0200
+++ configure	2013-04-30 23:40:02.639789283 +0300
@@ -19451,7 +19451,7 @@
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_GRAPHVIZ_LIBS=`$PKG_CONFIG --libs ""libgvc"" 2>/dev/null`
+  pkg_cv_GRAPHVIZ_LIBS="-lcdt -lcgraph -lgvc"
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
