diff -Nur kiki-old/kodilib/src/tools/KFileTools.cpp kiki/kodilib/src/tools/KFileTools.cpp
--- kiki-old/kodilib/src/tools/KFileTools.cpp	2006-11-18 14:50:02.196317735 +0200
+++ kiki/kodilib/src/tools/KFileTools.cpp	2006-11-18 14:52:50.992656791 +0200
@@ -44,7 +44,8 @@
 {
 #ifndef WIN32
     char buffer[MAXPATHLEN+1];
-    getwd(buffer);
+    // getwd(buffer);
+    getcwd(buffer, MAXPATHLEN+1);
 #else
 	char buffer[MAX_PATH+1];
 	getcwd(buffer, MAX_PATH+1);
