--- kiki/src/main/KikiPythonWidget.h~	2003-04-09 04:58:18.000000000 +0200
+++ kiki/src/main/KikiPythonWidget.h	2006-08-16 20:53:57.000000000 +0200
@@ -40,7 +40,7 @@ class KikiPythonWidget : public KTextFie
     bool		handleKey		( const KKey & );
     KSize		relayout		( const KSize & );
     
-    KikiPython *	getPythonWindow		() const { return (KikiPython*)getWindow(); }
+    class KikiPython *	getPythonWindow		() const { return (KikiPython*)getWindow(); }
 
     void		render			();
     void		displayStringAtPos	( const std::string &, const KikiPos & );
