diff -Nur kiki-old/src/main/KikiPythonWidget.h kiki/src/main/KikiPythonWidget.h
--- kiki-old/src/main/KikiPythonWidget.h	2006-11-13 16:20:29.000000000 +0200
+++ kiki/src/main/KikiPythonWidget.h	2006-11-13 16:21:03.000000000 +0200
@@ -7,6 +7,7 @@
 #define __KikiPythonWidget
 
 #include "KikiPos.h"
+#include "KikiPython.h"
 #include <KTextField.h>
 
 class KikiPythonWidget : public KTextField
