--- py/lang.py	2006-11-16 11:10:03.000000000 +0200
+++ lang.py	2006-11-16 12:34:00.000000000 +0200
@@ -4,7 +4,7 @@
 
 Controller.language = "english"
 
-lang_list = [  "dutch", "english", "euskara", "german",  "portuguese", "spanish", "swedish", ]
+lang_list = [  "dutch", "english", "euskara", "german",  "portuguese", "spanish", "swedish", "turkish", ]
 lang = {}
 for langName in lang_list:
     execfile (kikipy_path + os.path.sep + "lang" + os.path.sep + langName + ".py")
