diff -uNr clamav-0.98.3/m4/reorganization/code_checks/functions.m4 clamav-0.98.3.nolibs/m4/reorganization/code_checks/functions.m4
--- clamav-0.98.3/m4/reorganization/code_checks/functions.m4	2014-05-06 20:39:56.000000000 +0200
+++ clamav-0.98.3.nolibs/m4/reorganization/code_checks/functions.m4	2014-05-09 10:38:20.376934561 +0200
@@ -1,6 +1,10 @@
 AX_CHECK_UNAME_SYSCALL
+AC_CHECK_FUNC(socket,, [
 AC_CHECK_LIB([socket], [bind], [LIBS="$LIBS -lsocket"; CLAMAV_MILTER_LIBS="$CLAMAV_MILTER_LIBS -lsocket"; FRESHCLAM_LIBS="$FRESHCLAM_LIBS -lsocket"; CLAMD_LIBS="$CLAMD_LIBS -lsocket"])
+])
+AC_CHECK_FUNC(gethostent,, [
 AC_SEARCH_LIBS([gethostent],[nsl], [(LIBS="$LIBS -lnsl"; CLAMAV_MILTER_LIBS="$CLAMAV_MILTER_LIBS -lnsl"; FRESHCLAM_LIBS="$FRESHCLAM_LIBS -lnsl"; CLAMD_LIBS="$CLAMD_LIBS -lnsl")])
+])
 
 AC_CHECK_FUNCS([poll setsid memcpy snprintf vsnprintf strerror_r strlcpy strlcat strcasestr inet_ntop setgroups initgroups ctime_r mkstemp mallinfo madvise])
 AC_FUNC_FSEEKO 
