Index: clamav-0.96.2/clamav-milter/clamav-milter.c
===================================================================
--- clamav-0.96.2.orig/clamav-milter/clamav-milter.c
+++ clamav-0.96.2/clamav-milter/clamav-milter.c
@@ -370,7 +370,7 @@ int main(int argc, char **argv) {
 
     if((opt = optget(opts, "PidFile"))->enabled) {
 	FILE *fd;
-	mode_t old_umask = umask(0002);
+	mode_t old_umask = umask(0022);
 
 	if((fd = fopen(opt->strarg, "w")) == NULL) {
 	    logg("!Can't save PID in file %s\n", opt->strarg);
Index: clamav-0.96.2/shared/output.c
===================================================================
--- clamav-0.96.2.orig/shared/output.c
+++ clamav-0.96.2/shared/output.c
@@ -280,7 +280,7 @@ int logg(const char *str, ...)
 #endif
     if(logg_file) {
 	if(!logg_fp) {
-	    old_umask = umask(0037);
+	    old_umask = umask(0077);
 	    if((logg_fp = fopen(logg_file, "at")) == NULL) {
 		umask(old_umask);
 #ifdef CL_THREAD_SAFE
Index: clamav-0.96.2/freshclam/freshclam.c
===================================================================
--- clamav-0.96.2.orig/freshclam/freshclam.c
+++ clamav-0.96.2/freshclam/freshclam.c
@@ -114,7 +114,7 @@ static void writepid(const char *pidfile
 {
 	FILE *fd;
 	int old_umask;
-    old_umask = umask(0006);
+    old_umask = umask(0022);
     if((fd = fopen(pidfile, "w")) == NULL) {
 	logg("!Can't save PID to file %s: %s\n", pidfile, strerror(errno));
     } else {