Index: security/nss/cmd/modutil/install-ds.c
===================================================================
RCS file: /cvsroot/mozilla/security/nss/cmd/modutil/install-ds.c,v
retrieving revision 1.2
diff -u -p -6 -r1.2 install-ds.c
--- security/nss/cmd/modutil/install-ds.c	25 Apr 2004 15:02:47 -0000	1.2
+++ security/nss/cmd/modutil/install-ds.c	5 Feb 2007 06:57:38 -0000
@@ -249,13 +249,13 @@ Pk11Install_File_Generate(Pk11Install_Fi
 				if(!subval || (subval->type != STRING_VALUE)){
 					errStr = PR_smprintf(errString[BOGUS_FILE_PERMISSIONS],
                                     _this->jarPath);
 					goto loser;
 				}
 				_this->permissions = (int) strtol(subval->string, &endp, 8);
-				if(*endp != '\0' || subval->string == "\0") {
+				if(*endp != '\0' || subval->string[0] == '\0') {
 					errStr = PR_smprintf(errString[BOGUS_FILE_PERMISSIONS],
                                     _this->jarPath);
 					goto loser;
 				}
 				gotPerms = PR_TRUE;
 				Pk11Install_ListIter_delete(subiter);
