--- mozilla/security/coreconf/headers.mk 2012-05-03 12:42:59.000000000 +0300
+++ mozilla/security/coreconf/headers.mk	2013-01-10 14:09:27.207527766 +0200
@@ -12,7 +12,7 @@
 #  (xp) include paths
 #
 
-INCLUDES += -I$(SOURCE_MDHEADERS_DIR) -I$(SOURCE_XPHEADERS_DIR)
+INCLUDES += -I$(SOURCE_MDHEADERS_DIR) -I$(SOURCE_XPHEADERS_DIR) -I/usr/include/nspr -I$(DIST)/include/dbm
 
 #
 #  Only append source-side private cross-platform include paths for
