Index: polkit-qt-1-0.99.0/core/polkitqt1-identity.cpp
===================================================================
--- polkit-qt-1-0.99.0.orig/core/polkitqt1-identity.cpp
+++ polkit-qt-1-0.99.0/core/polkitqt1-identity.cpp
@@ -39,7 +39,10 @@ public:
     }
     ~Data()
     {
-        g_object_unref(identity);
+        if (identity)
+        {
+            g_object_unref(identity);
+        }
     }
 
     PolkitIdentity *identity;
