--- file-4.20/magic/Localstuff.xen	2007-01-12 18:36:27.000000000 +0100
+++ file-4.20/magic/Localstuff	2007-03-10 20:31:10.000000000 +0100
@@ -5,3 +5,9 @@
 # $File: Localstuff,v 1.4 2003/03/23 04:17:27 christos Exp $
 # Add any locally observed files here.  Remember:
 # text if readable, executable if runnable binary, data if unreadable.
+
+# File magic for Xen, the virtual machine monitor for x86
+0	string		LinuxGuestRecord	Xen saved domain
+#>2	regex		\(name\ [^)]*\)		%s
+>20	search/256	(name			(name
+>>&1	string		x			%s...)
