--- src/apprentice.c.OSmanOS	2013-01-04 02:47:35.000000000 +0200
+++ src/apprentice.c	2013-02-16 21:29:37.603866490 +0200
@@ -311,7 +311,7 @@
 	if ((rv = apprentice_map(ms, magic, nmagic, fn)) < 0) {
 		if (rv == -2)
 			return -1;
-		if (ms->flags & MAGIC_CHECK)
+		if (ms->flags & MAGIC_CHECK && strcmp("/etc/magic", fn) != 0)
 			file_magwarn(ms, "using regular magic file `%s'", fn);
 		rv = apprentice_load(ms, magic, nmagic, fn, action);
 		if (rv != 0)
