diff --git a/magic/Magdir/python b/magic/Magdir/python
index e592803..c786469 100644
--- a/magic/Magdir/python
+++ b/magic/Magdir/python
@@ -59,3 +59,8 @@
 !:mime text/x-python
 >&0	search/4096	finally:	Python script text executable
 !:mime text/x-python
+
+# def name(args, args):
+0	regex	 \^(\ |\\t)*def\ +[a-zA-Z]+
+>&0	regex	\ *\\(([a-zA-Z]|,|\ )*\\):$ Python script text executable
+!:mime text/x-python
