From 834831f53398cf2a1cfcd1daaf88c437bbf8d21f Mon Sep 17 00:00:00 2001
From: Christos Zoulas <christos@zoulas.com>
Date: Fri, 4 Jan 2013 17:23:28 +0000
Subject: [PATCH] escape the period in the keywords

---
 magic/Magdir/assembler |   16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/magic/Magdir/assembler b/magic/Magdir/assembler
index 92c6dde..011c08e 100644
--- a/magic/Magdir/assembler
+++ b/magic/Magdir/assembler
@@ -3,16 +3,16 @@
 # make:  file(1) magic for assembler source
 #
-0	regex	\^[\020\t]*\.asciiz\?		assembler source text
+0	regex	\^[\020\t]*\\.asciiz\?		assembler source text
 !:mime	text/x-asm
-0	regex	\^[\020\t]*\.byte		assembler source text
+0	regex	\^[\020\t]*\\.byte		assembler source text
 !:mime	text/x-asm
-0	regex	\^[\020\t]*\.even		assembler source text
+0	regex	\^[\020\t]*\\.even		assembler source text
 !:mime	text/x-asm
-0	regex	\^[\020\t]*\.globl		assembler source text
+0	regex	\^[\020\t]*\\.globl		assembler source text
 !:mime	text/x-asm
-0	regex	\^[\020\t]*\.text		assembler source text
+0	regex	\^[\020\t]*\\.text		assembler source text
 !:mime	text/x-asm
-0	regex	\^[\020\t]*\.file		assembler source text
+0	regex	\^[\020\t]*\\.file		assembler source text
 !:mime	text/x-asm
-0	regex	\^[\020\t]*\.type		assembler source text
+0	regex	\^[\020\t]*\\.type		assembler source text
 !:mime	text/x-asm
-- 
1.7.10

