--- magic/Magdir/apple	2008-02-28 19:57:34.000000000 +0100
+++ magic/Magdir/apple.oden	2008-12-26 00:23:52.000000000 +0100
@@ -26,6 +26,23 @@
 >>0x14	short		x	\b, %u Blocks
 >0xc	byte		02	\b, NIB data
 
+# Apple Emulator 2IMG format
+#
+0	string		2IMG		Apple ][ 2IMG Disk Image
+>4	string		XGS!		\b, XGS
+>4	string		CTKG		\b, Catakig
+>4	string		ShIm		\b, Sheppy's ImageMaker
+>4	string		WOOF		\b, Sweet 16
+>4	string		B2TR		\b, Bernie ][ the Rescue
+>4	string		!nfc		\b, ASIMOV2
+>4	string		x		\b, Unknown Format
+>0xc	byte		00		\b, DOS 3.3 sector order
+>>0x10	byte		00		\b, Volume 254
+>>0x10	byte&0x7f	x		\b, Volume %u
+>0xc	byte		01		\b, ProDOS sector order
+>>0x14	short		x		\b, %u Blocks
+>0xc	byte		02		\b, NIB data
+
 # magic for Newton PDA package formats
 # from Ruda Moura <ruda@helllabs.org>
 0	string	package0	Newton package, NOS 1.x,
