--- file-5.05/magic/Magdir/images.images~	2010-11-25 16:00:12.000000000 +0100
+++ file-5.05/magic/Magdir/images	2011-01-19 09:50:32.000000000 +0100
@@ -626,6 +626,12 @@
 0	string	\211HDF\r\n\032\n	Hierarchical Data Format (version 5) data
 !:mime	application/x-hdf
 
+# The boot loaders syslinux and isolinux use a RLE based image format
+# called SLL16 to store splash screens.
++0	lelong		0x1413f33d	Syslinux SLL16 image data,
+>4	leshort		>0		%hd x
+>6	leshort		>0		%hd
+
 # From: Tobias Burnus <burnus@net-b.de>
 # Xara (for a while: Corel Xara) is a graphic package, see
 # http://www.xara.com/ for Windows and as GPL application for Linux
