diff --git a/magic/Magdir/msdos b/magic/Magdir/msdos
index 560d360..83724c5 100644
--- a/magic/Magdir/msdos
+++ b/magic/Magdir/msdos
@@ -815,6 +815,17 @@
 
 0	string/b	QEVM		QEMU suspend to disk image
 
+# QEMU QED Image
+# http://wiki.qemu.org/Features/QED/Specification
+0	string/b	QED\0		QEMU QED Image
+
+# VDI Image
+64	string/b	\x7f\x10\xda\xbe	VDI Image
+>68	string/b	\x01\x00\x01\x00	version 1.1
+>0	string		>\0			(%s)
+>368	lequad		x			 \b, %lld bytes
+
+
 0	string/b	Bochs\ Virtual\ HD\ Image	Bochs disk image,
 >32	string	x				type %s,
 >48	string	x				subtype %s
