diff --git a/src/magic.c b/src/magic.c
index 7eb0441..d8c5835 100644
--- a/src/magic.c
+++ b/src/magic.c
@@ -101,16 +101,20 @@ get_default_magic(void)
 	if ((home = getenv("HOME")) == NULL)
 		return MAGIC;
 
-	if (asprintf(&hmagicpath, "%s/.magic", home) < 0)
+	if (asprintf(&hmagicpath, "%s/.magic.mgc", home) < 0)
 		return MAGIC;
-	if (stat(hmagicpath, &st) == -1)
-		goto out;
-	if (S_ISDIR(st.st_mode)) {
-		free(hmagicpath);
-		if (asprintf(&hmagicpath, "%s/%s", home, hmagic) < 0)
+	if (stat(hmagicpath, &st) == -1) {
+		if (asprintf(&hmagicpath, "%s/.magic", home) < 0)
 			return MAGIC;
-		if (access(hmagicpath, R_OK) == -1)
+		if (stat(hmagicpath, &st) == -1)
 			goto out;
+		if (S_ISDIR(st.st_mode)) {
+			free(hmagicpath);
+			if (asprintf(&hmagicpath, "%s/%s", home, hmagic) < 0)
+				return MAGIC;
+			if (access(hmagicpath, R_OK) == -1)
+				goto out;
+		}
 	}
 
 	if (asprintf(&default_magic, "%s:%s", hmagicpath, MAGIC) < 0)
