From 4be0945cb85497cb7bdbd978d780a48551351dfb Mon Sep 17 00:00:00 2001
From: Christos Zoulas <christos@zoulas.com>
Date: Fri, 4 Jan 2013 23:31:11 +0000
Subject: [PATCH] remove ?

---
 magic/Magdir/assembler |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/magic/Magdir/assembler b/magic/Magdir/assembler
index 011c08e..a1e577d 100644
--- a/magic/Magdir/assembler
+++ b/magic/Magdir/assembler
@@ -1,8 +1,8 @@
 #------------------------------------------------------------------------------
-# $File: assembler,v 1.2 2012/10/31 18:41:42 christos Exp $
+# $File: assembler,v 1.3 2013/01/04 17:23:28 christos Exp $
 # make:  file(1) magic for assembler source
 #
-0	regex	\^[\020\t]*\\.asciiz\?		assembler source text
+0	regex	\^[\020\t]*\\.asciiz		assembler source text
 !:mime	text/x-asm
 0	regex	\^[\020\t]*\\.byte		assembler source text
 !:mime	text/x-asm
-- 
1.7.10

