--- src/Makefile.in.OSmanOS	2013-02-16 21:30:52.000000000 +0200
+++ src/Makefile.in	        2013-02-16 21:34:32.543857775 +0200
@@ -243,7 +243,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-MAGIC = $(pkgdatadir)/magic
+MAGIC = /etc/magic:$(pkgdatadir)/magic
 lib_LTLIBRARIES = libmagic.la
 include_HEADERS = magic.h
 AM_CPPFLAGS = -DMAGIC='"$(MAGIC)"'
