--- file-5.11/magic/Magdir/commands.php~	2012-02-24 10:29:14.827382671 +0100
+++ file-5.11/magic/Magdir/commands	2012-02-24 10:29:20.645385573 +0100
@@ -67,14 +67,17 @@
 # PHP scripts
 # Ulf Harnhammar <ulfh@update.uu.se>
 0	search/1/c	=<?php			PHP script text
+!:strength + 10
 !:mime	text/x-php
 0	search/1	=<?\n			PHP script text
 !:mime	text/x-php
 0	search/1	=<?\r			PHP script text
 !:mime	text/x-php
 0	search/1/w	#!\ /usr/local/bin/php	PHP script text executable
+!:strength + 10
 !:mime	text/x-php
 0	search/1/w	#!\ /usr/bin/php	PHP script text executable
+!:strength + 10
 !:mime	text/x-php
 # Smarty compiled template, http://www.smarty.net/
 # Elan Ruusamäe <glen@delfi.ee>
