Index: konversation-1.3.1/data/scripts/media
===================================================================
--- konversation-1.3.1.orig/data/scripts/media
+++ konversation-1.3.1/data/scripts/media
@@ -279,6 +279,42 @@ class Amarok2Player(Player):
         self.running=bool(running.strip())
         return self.running
 
+class ClementinePlayer(Player):
+    def __init__(self):
+        Player.__init__(self, 'Clementine', 'audio')
+        self.isRunning()
+
+    def getData(self):
+        playing=os.popen("qdbus org.mpris.clementine /Player PositionGet").readline().strip() != "0"
+        if playing and self.isRunning():
+            title = ''
+            artist = ''
+            album = ''
+            for line in os.popen("qdbus org.mpris.clementine /Player GetMetadata").readlines():
+                m = re.match("^title:\s(.+)", line)
+                if m is not None:
+                    title = m.group(1)
+                m = re.match("^artist:\s(.+)", line)
+                if m is not None:
+                    artist = m.group(1)
+                m = re.match("^album:\s(.+)", line)
+                if m is not None:
+                    album = m.group(1)
+            return (title, artist, album)
+        else:
+            return ''
+
+    def isRunning(self):
+        qdbus_items=subprocess.Popen(['qdbus'], stdout=subprocess.PIPE).communicate()[0]
+        running=re.findall(r'^ org.mpris.clementine\r?$', qdbus_items, re.M)
+        if type(running) is list:
+            try:
+                running=running[0]
+            except IndexError:
+                running=''
+        self.running=bool(running.strip())
+        return self.running
+
 import socket
 
 class JukPlayer(Player):
@@ -507,6 +543,7 @@ StupidPlayer("KPlayer","kplayer","kplaye
 DCOPPlayer("KsCD","kscd","CDPlayer currentTrackTitle","CDPlayer currentArtist","CDPlayer currentAlbum"),
 DCOPPlayer("kdetv","kdetv","KdetvIface channelName",playerType='video'),
 Amarok2Player(),
+ClementinePlayer(),
 DCOPPlayer("Yammi","yammi","YammiPlayer songTitle","YammiPlayer songArtist","YammiPlayer songAlbum"),
 AudaciousPlayer('Audacious'), Audacious2Player('Audacious2'), XmmsPlayer('XMMS'),
 MPD('MPD'),
