diff -Naur konversation-1.3.orig/data/scripts/media konversation-1.3/data/scripts/media
--- konversation-1.3.orig/data/scripts/media	2010-06-08 02:08:17.212769740 +0300
+++ konversation-1.3/data/scripts/media	2010-06-08 02:12:24.760769605 +0300
@@ -44,7 +44,7 @@
 ## If you add a new player, you must add it here or it won't get checked when in audio-only or video-only modes.
 playerRankings= {
     'video' :['kaffeine','kmplayer', 'kplayer', 'noatun', 'kdetv'],
-    'audio' :['amarok', 'amarok1', 'juk', 'noatun', 'kscd', 'kaffeine', 'kmplayer', 'yammi', 'Audacious', 'xmms', 'MPD']
+    'audio' :['amarok', 'amarok1', 'juk', 'noatun', 'kscd', 'kaffeine', 'kmplayer', 'yammi', 'Audacious', 'xmms', 'MPD', 'LastFM']
 }
 
 ## Title, album and artist fields to be quoted depending on contents
@@ -409,6 +409,15 @@
         else:
             return ''
 
+class LastFMPlayer(Player):
+    def __init__(self):
+        Player.__init__(self,"last.fm")
+
+    def getData(self):
+        title = subprocess.Popen(["qdbus","com.lastfm.DBusInterface","/","currentTrack"], stdout=subprocess.PIPE).communicate()[0].rstrip()
+        artist = subprocess.Popen(["qdbus","com.lastfm.DBusInterface","/","currentArtist"], stdout=subprocess.PIPE).communicate()[0].rstrip()
+        album = subprocess.Popen(["qdbus","com.lastfm.DBusInterface","/","currentAlbum"], stdout=subprocess.PIPE).communicate()[0].rstrip()
+        return (title,artist,album)
 
 def playing(playerList, mode=None):
     for i in playerList:
@@ -480,7 +489,8 @@
 Amarok2Player(),
 DCOPPlayer("Yammi","yammi","YammiPlayer songTitle","YammiPlayer songArtist","YammiPlayer songAlbum"),
 AudaciousPlayer('Audacious'), XmmsPlayer('XMMS'),
-MPD('MPD')
+MPD('MPD'),
+LastFMPlayer()
 ]
 
 # Get rid of players that didn't get subclassed so they don't appear in the available players list
