diff -Naur konversation-1.3.orig/data/scripts/media konversation-1.3/data/scripts/media
--- konversation-1.3.orig/data/scripts/media	2010-06-08 02:20:20.000000000 +0300
+++ konversation-1.3/data/scripts/media	2010-06-08 02:21:12.133769635 +0300
@@ -44,7 +44,7 @@
 ## If you add a new player, you must add it here or it won't get checked when in audio-only or video-only modes.
 playerRankings= {
     'video' :['kaffeine','kmplayer', 'kplayer', 'noatun', 'kdetv'],
-    'audio' :['amarok', 'amarok1', 'juk', 'noatun', 'kscd', 'kaffeine', 'kmplayer', 'yammi', 'Audacious', 'xmms', 'MPD', 'LastFM']
+    'audio' :['amarok', 'amarok1', 'juk', 'noatun', 'kscd', 'kaffeine', 'kmplayer', 'yammi', 'Audacious', 'Audacious2', 'xmms', 'MPD', 'LastFM']
 }
 
 ## Title, album and artist fields to be quoted depending on contents
@@ -409,6 +409,26 @@
         else:
             return ''
 
+class Audacious2Player(Player):
+    def __init__(self, display_name):
+        Player.__init__(self, display_name)
+
+    def isRunning(self):
+        self.running = not os.system('audtool2 current-song')
+        return self.running
+
+    def getData(self):
+        if self.isRunning():
+            # get the title of the currently playing track
+            title = os.popen('audtool2 current-song-tuple-data title').read().strip()
+            artist = os.popen('audtool2 current-song-tuple-data artist').read().strip()
+            album = os.popen('audtool2 current-song-tuple-data album').read().strip()
+
+            return (self.reEncodeString(title),self.reEncodeString(artist),self.reEncodeString(album))
+
+        else:
+            return ''
+
 class LastFMPlayer(Player):
     def __init__(self):
         Player.__init__(self,"last.fm")
@@ -488,7 +508,7 @@
 DCOPPlayer("kdetv","kdetv","KdetvIface channelName",playerType='video'),
 Amarok2Player(),
 DCOPPlayer("Yammi","yammi","YammiPlayer songTitle","YammiPlayer songArtist","YammiPlayer songAlbum"),
-AudaciousPlayer('Audacious'), XmmsPlayer('XMMS'),
+AudaciousPlayer('Audacious'), Audacious2Player('Audacious2'), XmmsPlayer('XMMS'),
 MPD('MPD'),
 LastFMPlayer()
 ]
