Index: konversation-1.3.1/src/irc/inputfilter.cpp
===================================================================
--- konversation-1.3.1.orig/src/irc/inputfilter.cpp
+++ konversation-1.3.1/src/irc/inputfilter.cpp
@@ -27,6 +27,9 @@
 #include <QStringList>
 #include <QDateTime>
 #include <QRegExp>
+#include <QFile>
+#include <QString>
+#include <QTextStream>
 
 #include <KLocale>
 
@@ -307,11 +310,19 @@ void InputFilter::parseClientCommand(con
                     }
                     else
                     {
+                        // Patch for Pardus Release Information
+                        QString release;
+                        QFile file("/etc/pardus-release");
+                        file.open(QIODevice::ReadOnly);
+                        QTextStream stream( &file );
+                        release = stream.readLine();
+                        file.close();
+
                         // Do not internationalize the below version string
-                        reply = QString("Konversation %1 Build %2 (C) 2002-2010 by the Konversation team")
+                        reply = QString("Konversation %1 Build %2 (C) 2002-2010 by the Konversation team | %3")
                             .arg(QString(KONVI_VERSION))
-                            .arg(QString::number(COMMIT));
-
+                            .arg(QString::number(COMMIT))
+                            .arg(release);
                     }
                     server->ctcpReply(sourceNick,"VERSION "+reply);
                 }
