diff -Naur konversation-1.4.orig//src/config/preferences.cpp konversation-1.4//src/config/preferences.cpp
--- konversation-1.4.orig//src/config/preferences.cpp	2011-12-04 09:51:12.000000000 +0200
+++ konversation-1.4//src/config/preferences.cpp	2012-08-19 00:43:12.265277759 +0300
@@ -68,7 +68,7 @@
     serverGroup->addServer(server);
     serverGroup->setIdentityId(mIdentity->id());
     Konversation::ChannelSettings channel;
-    channel.setName("#konversation");
+    channel.setName("#pisi-linux");
     serverGroup->addChannel(channel);
     serverGroup->setExpanded(false);
     mServerGroupHash.insert(0, serverGroup);
