diff -Naur konversation-1.2-alpha4.orig/data/scripts/bug konversation-1.2-alpha4/data/scripts/bug
--- konversation-1.2-alpha4.orig/data/scripts/bug	2009-07-04 08:57:15.891280543 +0300
+++ konversation-1.2-alpha4/data/scripts/bug	2009-07-04 09:17:43.853405426 +0300
@@ -12,6 +12,6 @@
   then
     qdbus org.kde.konversation /irc error "You forgot the bug number!"
   else
-    kfmclient openURL https://bugs.kde.org/buglist.cgi?quicksearch=$3
+    kfmclient openURL http://bugs.pardus.org.tr/buglist.cgi?quicksearch=$3
   fi
 fi
diff -Naur konversation-1.2-alpha4.orig/src/config/preferences.cpp konversation-1.2-alpha4/src/config/preferences.cpp
--- konversation-1.2-alpha4.orig/src/config/preferences.cpp	2009-07-04 08:57:15.886280954 +0300
+++ konversation-1.2-alpha4/src/config/preferences.cpp	2009-07-04 09:17:32.214405773 +0300
@@ -121,7 +121,7 @@
 {
     QList<QStringList> defaultList;
     defaultList.append(QStringList() << "1" << "o" << "\\[\\[([^\\s]+)\\]\\]" << "http://en.wikipedia.org/wiki/Special:Search?go=Go&search=%1");
-    defaultList.append(QStringList() << "1" << "o" << "(BUG:|bug:)([0-9]+)" << "https://bugs.kde.org/show_bug.cgi?id=%2");
+    defaultList.append(QStringList() << "1" << "o" << "(BUG:|bug:)([0-9]+)" << "http://bugs.pardus.org.tr/show_bug.cgi?id=%2");
     return defaultList;
 }
 
