diff -urN nspr-4.5.99.old/mozilla/nsprpub/config/Makefile.in nspr-4.5.99/mozilla/nsprpub/config/Makefile.in
--- nspr-4.5.99.old/mozilla/nsprpub/config/Makefile.in	2005-05-03 11:16:45.000000000 +0200
+++ nspr-4.5.99/mozilla/nsprpub/config/Makefile.in	2005-05-18 12:05:53.000000000 +0200
@@ -52,9 +52,9 @@
 
 # autoconf.mk must be deleted last (from the top-level directory)
 # because it is included by every makefile.
-DIST_GARBAGE	= nsprincl.mk nsprincl.sh nspr-config
+DIST_GARBAGE	= nsprincl.mk nsprincl.sh nspr-config nspr.pc
 
-RELEASE_BINS	= nspr-config
+RELEASE_BINS	= nspr-config nspr.pc
 
 include $(topsrcdir)/config/config.mk
 
@@ -139,6 +139,7 @@
 
 export:: $(TARGETS)
 	rm -f $(dist_bindir)/nspr-config
+	rm -f $(dist_bindir)/nspr.pc
 
 ifdef WRAP_SYSTEM_INCLUDES
 export::
diff -urN nspr-4.5.99.old/mozilla/nsprpub/config/nspr.pc.in nspr-4.5.99/mozilla/nsprpub/config/mozilla-nspr.pc.in
--- nspr-4.5.99.old/mozilla/nsprpub/config/nspr.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ nspr-4.5.99/mozilla/nsprpub/config/nspr.pc.in	2005-05-18 12:03:36.000000000 +0200
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: NSPR
+Description: The Netscape Portable Runtime
+Version: @MOD_MAJOR_VERSION@.@MOD_MINOR_VERSION@.@MOD_PATCH_VERSION@
+Libs: -L${libdir} -lplds@MOD_MAJOR_VERSION@ -lplc@MOD_MAJOR_VERSION@ -lnspr@MOD_MAJOR_VERSION@ -lpthread -ldl
+Cflags: -I${includedir}
+
+
diff -urN nspr-4.5.99.old/mozilla/nsprpub/configure nspr-4.5.99/mozilla/nsprpub/configure
--- nspr-4.5.99.old/mozilla/nsprpub/configure	2005-05-13 09:33:07.000000000 +0200
+++ nspr-4.5.99/mozilla/nsprpub/configure	2005-05-18 12:07:00.000000000 +0200
@@ -5718,6 +5718,7 @@
 config/nsprincl.mk
 config/nsprincl.sh
 config/nspr-config
+config/nspr.pc
 lib/Makefile 
 lib/ds/Makefile 
 lib/libc/Makefile 
diff -urN nspr-4.5.99.old/mozilla/nsprpub/configure.in nspr-4.5.99/mozilla/nsprpub/configure.in
--- nspr-4.5.99.old/mozilla/nsprpub/configure.in	2005-05-13 09:33:07.000000000 +0200
+++ nspr-4.5.99/mozilla/nsprpub/configure.in	2005-05-18 12:06:42.000000000 +0200
@@ -2569,6 +2569,7 @@
 config/nsprincl.mk
 config/nsprincl.sh
 config/nspr-config
+config/nspr.pc
 lib/Makefile 
 lib/ds/Makefile 
 lib/libc/Makefile 
