Index: work/xorg-cf-files-1.0.5/host.def
===================================================================
--- /dev/null
+++ work/xorg-cf-files-1.0.5/host.def
@@ -0,0 +1,14 @@
+#define FSUseSyslog             YES
+#define BuildRman               NO
+#define BuildHtmlManPages       NO
+#define HaveLib64               NO
+#define ProjectRoot             /usr
+#define ManPath                 /usr/share/man
+#define XAppLoadDir             /usr/share/X11/app-defaults
+#define ConfigDir               /usr/share/X11/config
+
+#define CcCmd                   _PARDUS_CC_ _PARDUS_LDFLAGS_
+#define CplusplusCmd            _PARDUS_CXX_
+#define AsCmd                   _PARDUS_AS_
+#define LdCmd                   _PARDUS_LD_ _PARDUS_LDFLAGS_ -nostdlib
+#define OptimizedCDebugFlags    _PARDUS_CFLAGS_
