diff -up imake-1.0.4/cleanlinks.jx imake-1.0.4/cleanlinks
--- imake-1.0.4/cleanlinks.jx	2004-12-03 19:42:46.000000000 -0500
+++ imake-1.0.4/cleanlinks	2009-02-08 02:17:08.000000000 -0500
@@ -23,5 +23,5 @@ find . -type l -print |
 
 echo Removing empty directories ...
 #find . -type d -depth -print | xargs rmdir > /dev/null 2>&1
-find . -type d -depth -empty -print -exec rmdir {} \;
+find . -depth -type d -empty -print -exec rmdir {} \;
 exit 0
