diff -up wireshark-1.2.6/ltmain.sh.libpie wireshark-1.2.6/ltmain.sh
--- wireshark-1.2.6/ltmain.sh.libpie	2010-01-27 17:13:14.000000000 +0100
+++ wireshark-1.2.6/ltmain.sh	2010-03-23 17:39:43.000000000 +0100
@@ -2499,7 +2499,6 @@ static const void *lt_preloaded_setup() 
 	symtab_cflags=
 	for arg in $LTCFLAGS; do
 	  case $arg in
-	  -pie | -fpie | -fPIE) ;;
 	  *) symtab_cflags="$symtab_cflags $arg" ;;
 	  esac
 	done
