diff -up wireshark-1.2.4/epan/wslua/template-init.lua.enable_lua wireshark-1.2.4/epan/wslua/template-init.lua
--- wireshark-1.2.4/epan/wslua/template-init.lua.enable_lua	2009-11-16 21:30:58.000000000 +0100
+++ wireshark-1.2.4/epan/wslua/template-init.lua	2009-12-15 09:33:59.000000000 +0100
@@ -39,7 +39,7 @@ if running_superuser then
     local disabled_lib = {}
     setmetatable(disabled_lib,{ __index = function() error("this package has been disabled") end } );
 
-    dofile = function() error("dofile has been disabled") end
+--    dofile = function() error("dofile has been disabled") end
     loadfile = function() error("loadfile has been disabled") end
     loadlib = function() error("loadlib has been disabled") end
     require = function() error("require has been disabled") end
