diff -up mc-4.8.7/src/vfs/cpio/cpio.c.cpiosegfault mc-4.8.7/src/vfs/cpio/cpio.c
--- mc-4.8.7/src/vfs/cpio/cpio.c.cpiosegfault	2012-12-27 09:13:20.000000000 +0100
+++ mc-4.8.7/src/vfs/cpio/cpio.c	2013-03-14 11:55:50.660147302 +0100
@@ -795,7 +795,7 @@ cpio_super_same (const vfs_path_element_
     g_free (archive_name);
 
     /* Has the cached archive been changed on the disk? */
-    if (((cpio_super_data_t *) parc->data)->st.st_mtime < archive_stat->st_mtime)
+    if (parc->data && ((cpio_super_data_t *) parc->data)->st.st_mtime < archive_stat->st_mtime)
     {
         /* Yes, reload! */
         (*vfs_cpiofs_ops.free) ((vfsid) parc);
