diff -ur pisido-2.0b.oig/cpp/mainwindow.cpp pisido-2.0b/cpp/mainwindow.cpp
--- pisido-2.0b.oig/cpp/mainwindow.cpp	2011-11-11 00:56:43.000000000 +0200
+++ pisido-2.0b/cpp/mainwindow.cpp	2013-03-09 23:32:02.297464522 +0200
@@ -194,7 +194,6 @@
     terminal_font.setFamily("Monospace");
     terminal_font.setPointSize(10);
     w_terminal->setTerminalFont(terminal_font);
-    w_terminal->setColorScheme(COLOR_SCHEME_WHITE_ON_BLACK);
     w_terminal->setScrollBarPosition(QTermWidget::ScrollBarRight);
     QHBoxLayout * build_layout = qobject_cast<QHBoxLayout *>(ui->dw_build_contents->layout());
     if(build_layout){
