--- autoconf/configure.orig	2009-05-07 00:09:50.709516510 +0300
+++ autoconf/configure	2009-05-07 00:10:10.640242936 +0300
@@ -16165,6 +16165,7 @@
 #include "confdefs.h"
 
 #if defined(__linux__) || defined(__linux) || defined(linux)
+#include <linux/fs.h>
 #include <linux/ext2_fs.h>
 #endif
 int main() {
@@ -16201,6 +16202,7 @@
 #include "confdefs.h"
 
 #if defined(__linux__) || defined(__linux) || defined(linux)
+#include <linux/fs.h>
 #include <linux/ext2_fs.h>
 #endif
 int main() {
