Index: DEFAULTS/Defaults.linux
===================================================================
--- DEFAULTS/Defaults.linux.orig
+++ DEFAULTS/Defaults.linux
@@ -40,23 +40,23 @@ OSDEFS +=	-D_GNU_SOURCE
 # LINUX_SRC_INCLUDE is used in RULES/*-linux-*cc*.rul
 #
 ###########################################################################
-LINUX_INCL_PATH=	/usr/src/linux/include
+LINUX_INCL_PATH=	
 
 _LINUX_SRC_INCLUDE=	$(_UNIQ)$(MAY_LINUX_SRC_INCLUDE)
 __LINUX_SRC_INCLUDE=	$(_LINUX_SRC_INCLUDE:$(_UNIQ)=$(LINUX_INCL_PATH))
 LINUX_SRC_INCLUDE=      $(__LINUX_SRC_INCLUDE:$(_UNIQ)no%=%)
 
 DEFINCDIRS=	$(SRCROOT)/include
-LDPATH=		-L/opt/schily/lib
+LDPATH=		-L/usr/lib
 #RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib -R$(OLIBSDIR)
-RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib
+RUNPATH=	-R$(INS_BASE)/lib -R/usr/lib
 
 ###########################################################################
 #
 # Installation config stuff
 #
 ###########################################################################
-INS_BASE=	/opt/schily
+INS_BASE=	/usr
 INS_KBASE=	/
 INS_RBASE=	/
 #
Index: librscg/scsi-remote.c
===================================================================
--- librscg/scsi-remote.c.orig
+++ librscg/scsi-remote.c
@@ -568,7 +568,7 @@ rscsigetconn(scgp, host)
 	rscsipeer = host;
 
 	if ((rscsi = getenv("RSCSI")) == NULL)
-		rscsi = "/opt/schily/sbin/rscsi";
+		rscsi = "/usr/sbin/rscsi";
 	rsh = getenv("RSH");
 
 #ifdef	USE_RCMD_RSH
Index: libfind/find.c
===================================================================
--- libfind/find.c.orig
+++ libfind/find.c
@@ -2225,7 +2225,7 @@ main(ac, av)
 
 #ifdef	USE_NLS
 	setlocale(LC_ALL, "");
-	bindtextdomain("SCHILY_FIND", "/opt/schily/lib/locale");
+	bindtextdomain("SCHILY_FIND", "/usr/lib/locale");
 	textdomain("SCHILY_FIND");
 #endif
 	find_argsinit(&fa);
Index: libfind/find_main.c
===================================================================
--- libfind/find_main.c.orig
+++ libfind/find_main.c
@@ -145,7 +145,7 @@ find_main(ac, av, ev, std, quit)
 
 #ifdef	USE_NLS
 	setlocale(LC_ALL, "");
-	bindtextdomain("SCHILY_FIND", "/opt/schily/lib/locale");
+	bindtextdomain("SCHILY_FIND", "/usr/lib/locale");
 	textdomain("SCHILY_FIND");
 #endif
 	find_argsinit(&fa);
