Index: cdrtools-2.01.01/RULES/mk-gmake.id
===================================================================
--- cdrtools-2.01.01.orig/RULES/mk-gmake.id
+++ cdrtools-2.01.01/RULES/mk-gmake.id
@@ -50,10 +50,6 @@ MAKEPROG=	gmake
 
 ifndef		ARCH_DONE
 
-ifndef		GMAKE_NOWARN
-__gmake_warn:=	$(shell cat $(SRCROOT)/$(RULESDIR)/gmake.wrn 1>&2; sleep 2)
-endif
-
 _MACHCMD=	(mach || uname -p || true)	2> /dev/null
 _ARCHCMD=	(arch || /usr/ucb/arch || true)	2> /dev/null
 
