diff -up gle-graphics-4.2.2/src/gui/qgle.pro.in.64bit gle-graphics-4.2.2/src/gui/qgle.pro.in
--- gle-graphics-4.2.2/src/gui/qgle.pro.in.64bit	2010-10-01 15:06:36.000000000 +0200
+++ gle-graphics-4.2.2/src/gui/qgle.pro.in	2010-10-01 15:06:45.000000000 +0200
@@ -160,6 +160,9 @@ DISTFILES += readme.txt \
 linux-g++ {
 	LIBS += -ldl
 }
+linux-g++-64 {
+	LIBS += -ldl
+}
 
 # link against the GLE library (unix -> Linux and Mac OS/X)
 unix {
diff -up gle-graphics-4.2.2/src/gui/qgslibloader.cpp.64bit gle-graphics-4.2.2/src/gui/qgslibloader.cpp
--- gle-graphics-4.2.2/src/gui/qgslibloader.cpp.64bit	2010-10-01 15:07:27.000000000 +0200
+++ gle-graphics-4.2.2/src/gui/qgslibloader.cpp	2010-10-01 15:11:09.000000000 +0200
@@ -246,7 +246,7 @@ int GSLibFunctions::loadLibrary(const QS
 	/* Try to load the library */
 	if (location == "") {
 		#ifdef Q_WS_X11
-		#ifdef __x86_64__
+		#if defined(__x86_64__) || defined(__ppc64__) || defined (__s390x__) || defined (__sparc64__)
 		// try 64 bit libraries on 64 bit system
 		tryLocation("/usr/lib64/libgs.so");
 		tryLocation("/usr/lib64/libgs.so.8");
@@ -257,7 +257,7 @@ int GSLibFunctions::loadLibrary(const QS
 		tryLocation("/usr/lib/libgs.so.8");
 		tryLocation("/usr/local/lib/libgs.so");
 		tryLocation("/usr/local/lib/libgs.so.8");
-		#endif  // __x86_64__
+		#endif  // 64-bit
 		#endif
 		#ifdef Q_WS_MAC
 		tryLocation("/usr/lib/libgs.dylib");
