Index: libcap-ng-0.6.4/src/cap-ng.h
===================================================================
--- libcap-ng-0.6.4.orig/src/cap-ng.h
+++ libcap-ng-0.6.4/src/cap-ng.h
@@ -25,6 +25,7 @@
 
 #include <stdint.h>
 #include <linux/capability.h>
+#include <linux/xattr.h>
 #include <unistd.h>
 
 #ifdef __cplusplus
@@ -35,6 +36,11 @@ extern "C" {
 #define CAP_LAST_CAP CAP_AUDIT_CONTROL
 #endif
 
+#ifndef XATTR_CAPS_SUFFIX
+#define XATTR_CAPS_SUFFIX "capability"
+#define XATTR_NAME_CAPS XATTR_SECURITY_PREFIX XATTR_CAPS_SUFFIX
+#endif
+
 typedef enum {  CAPNG_DROP, CAPNG_ADD } capng_act_t;
 typedef enum {  CAPNG_EFFECTIVE=1, CAPNG_PERMITTED=2,
 		CAPNG_INHERITABLE=4, CAPNG_BOUNDING_SET=8 } capng_type_t;
