diff -Nuar e-module-forecasts-0.0.1_58042.orig//module.desktop.in e-module-forecasts-0.0.1_58042/module.desktop.in
--- e-module-forecasts-0.0.1_58042.orig//module.desktop.in	2011-02-16 22:39:52.000000000 +0200
+++ e-module-forecasts-0.0.1_58042/module.desktop.in	2011-03-31 17:04:07.603043555 +0300
@@ -8,6 +8,7 @@
 Name[fr]=Prévisions
 Name[el]=Προβλέψεις
 Name[pt]=Previsões meteorológicas
+Name[tr]=Hava Tahminleri
 Icon=e-module-forecasts
 X-Enlightenment-ModuleType=utils
 Comment=The forecasts gadget will display the current weather conditions plus a few days forecast.
@@ -19,3 +20,4 @@
 Comment[el]=Το συστατικό προβλέψεων θα εμφανίσει τις τρέχουσες καιρικές συνθήκες και των επόμενων ημερών.
 Comment[fr]=Affiche les conditions météo actuelles et les prévisions sur quelques jours.
 Comment[pt]=Um módulo que exibe as condições atmosféricas para o dia atual e seguintes
+Comment[tr]=O günü ve sonraki günlerle ilgili hava tahminlerini gösteren modüldür.
