--- ./modules/bindings/glibc/linux.lisp.orig	2008-10-08 10:36:19.000000000 -0600
+++ ./modules/bindings/glibc/linux.lisp	2012-07-25 19:05:07.014592097 -0600
@@ -86,7 +86,7 @@
 
 (def-c-type __key_t)            ; int
 
-(c-lines "#include <bits/ipctypes.h>~%")
+(c-lines "#include <sys/ipc.h>~%")
 (def-c-type __ipc_pid_t)        ; ushort
 
 ; --------------------------- <sys/types.h> -----------------------------------
@@ -293,6 +293,8 @@
 ;; for robust mutexes
 (def-c-const EOWNERDEAD (:documentation "Owner died")) ; 130
 (def-c-const ENOTRECOVERABLE (:documentation "State not recoverable")) ; 131
+(def-c-const ERFKILL (:documentation "Operation not possible due to RF-kill")) ; 132
+(def-c-const EHWPOISON (:documentation "Memory page has hardware error")) ; 133
 
 ; -------------------------- <bits/errno.h> -----------------------------------
 
