--- scipy/cluster/setup.py.bak	2011-02-28 09:43:10.567098298 -0500
+++ scipy/cluster/setup.py	2011-02-28 09:44:14.638684188 -0500
@@ -17,11 +17,13 @@
     config.add_extension('_vq',
         sources=[join('src', 'vq_module.c'), join('src', 'vq.c')],
         include_dirs = [get_numpy_include_dirs()],
+        libraries = ['m'],
         define_macros=DEFINE_MACROS)
 
     config.add_extension('_hierarchy_wrap',
         sources=[join('src', 'hierarchy_wrap.c'), join('src', 'hierarchy.c')],
         include_dirs = [get_numpy_include_dirs()],
+        libraries = ['m'],
         define_macros=DEFINE_MACROS)
 
     return config
--- scipy/interpolate/setup.py.bak	2011-02-28 10:06:31.447035839 -0500
+++ scipy/interpolate/setup.py	2011-02-28 10:08:43.512179764 -0500
@@ -12,7 +12,8 @@
                       )
 
     config.add_extension('interpnd',
-                         sources=['interpnd.c'])
+                         sources=['interpnd.c'],
+                         libraries=['m'])
 
     config.add_extension('_fitpack',
                          sources=['src/_fitpackmodule.c'],
@@ -28,6 +29,7 @@
     config.add_extension('_interpolate',
                          sources=['src/_interpolate.cpp'],
                          include_dirs = ['src'],
+                         libraries=['m'],
                          depends = ['src/interpolate.h'])
 
     config.add_data_dir('tests')
--- scipy/optimize/setup.py.bak	2011-02-28 10:34:55.113954541 -0500
+++ scipy/optimize/setup.py	2011-02-28 10:37:49.962810309 -0500
@@ -15,7 +15,8 @@
 
     config.add_library('rootfind',
                        sources=[join('Zeros','*.c')],
-                       headers=[join('Zeros','zeros.h')])
+                       headers=[join('Zeros','zeros.h')],
+                       libraries=['m'])
 
     config.add_extension('_zeros',
                          sources=['zeros.c'],
@@ -30,7 +31,8 @@
     sources=['moduleTNC.c','tnc.c']
     config.add_extension('moduleTNC',
                          sources=[join('tnc',x) for x in sources],
-                         depends=[join('tnc','tnc.h')])
+                         depends=[join('tnc','tnc.h')],
+                         libraries=['m'])
 
     config.add_extension('_cobyla',
                          sources=[join('cobyla',x) for x in ['cobyla.pyf',
--- scipy/signal/setup.py.bak	2011-02-28 10:54:16.018390949 -0500
+++ scipy/signal/setup.py	2011-02-28 10:53:28.736698871 -0500
@@ -12,12 +12,14 @@
                                   'firfilter.c','medianfilter.c', 'lfilter.c.src',
                                   'correlate_nd.c.src'],
                          depends = ['sigtools.h'],
-                         include_dirs=['.']
+			 libraries = ['m'],
+			 include_dirs=['.']
     )
 
     config.add_extension('spline',
         sources = ['splinemodule.c','S_bspline_util.c','D_bspline_util.c',
                    'C_bspline_util.c','Z_bspline_util.c','bspline_util.c'],
+	libraries = ['m']
     )
 
     return config
--- scipy/spatial/setup.py.bak	2011-02-28 11:23:17.732146982 -0500
+++ scipy/spatial/setup.py	2011-02-28 11:50:04.981718334 -0500
@@ -31,14 +31,17 @@
         libs = ['qhull']
     config.add_extension('qhull',
                          sources=['qhull.c'],
-                         libraries=libs,
+                         libraries=libs+['m'],
                          **lapack)
 
-    config.add_extension('ckdtree', sources=['ckdtree.c']) # FIXME: cython
+    config.add_extension('ckdtree', 
+                          sources=['ckdtree.c'],
+                          libraries=['m']) # FIXME: cython
 
     config.add_extension('_distance_wrap',
         sources=[join('src', 'distance_wrap.c'), join('src', 'distance.c')],
-        include_dirs = [get_numpy_include_dirs()])
+        include_dirs = [get_numpy_include_dirs()],
+        libraries=['m'])
 
     return config
 
--- scipy/stats/setup.py.bak	2011-02-28 12:16:53.073623971 -0500
+++ scipy/stats/setup.py	2011-02-28 12:17:14.776488842 -0500
@@ -22,6 +22,7 @@
     # add vonmises_cython module
     config.add_extension('vonmises_cython',
         sources=['vonmises_cython.c'], # FIXME: use cython source
+        libraries=['m']
     )
 
     # add futil module
--- scipy/ndimage/setup.py.bak	2011-02-28 12:52:11.487749623 -0500
+++ scipy/ndimage/setup.py	2011-02-28 12:52:34.238611985 -0500
@@ -12,6 +12,7 @@
                  "src/ni_measure.c",
                  "src/ni_morphology.c","src/ni_support.c"],
         include_dirs=['src']+[get_include()],
+        libraries=['m']
     )
 
     config.add_data_dir('tests')
