Index: sdl-mixer1.2-1.2.8/native_midi_gpl/native_midi_gpl.c
===================================================================
--- sdl-mixer1.2-1.2.8.orig/native_midi_gpl/native_midi_gpl.c	2007-10-07 22:05:21.000000000 +0000
+++ sdl-mixer1.2-1.2.8/native_midi_gpl/native_midi_gpl.c	2007-10-07 22:08:14.000000000 +0000
@@ -77,9 +77,10 @@
     int ret=0;    
 
     /* Open sequencer device */
-    if ((seqfd = open(SEQUENCER_DEV, O_WRONLY, 0)) < 0) 
+    if ((seqfd = open(SEQUENCER_DEV, O_WRONLY, 0)) < 0
+         && (seqfd = open(SEQUENCER_ALTDEV, O_WRONLY, 0)) < 0) 
     {
-	perror("open " SEQUENCER_DEV);
+	perror("open " SEQUENCER_DEV " or " SEQUENCER_ALTDEV);
 	return 0;
     }
 
@@ -253,9 +254,10 @@
     }
     
     /* Open sequencer device */
-    if ((seqfd = open(SEQUENCER_DEV, O_WRONLY, 0)) < 0) 
+    if ((seqfd = open(SEQUENCER_DEV, O_WRONLY, 0)) < 0
+          && (seqfd = open(SEQUENCER_ALTDEV, O_WRONLY, 0)) < 0) 
     {
-	perror("open " SEQUENCER_DEV);
+	perror("open " SEQUENCER_DEV " or " SEQUENCER_ALTDEV);
 	goto eend;
     }
     if(!synth_setup()) { goto end;};
@@ -324,7 +326,7 @@
 {
     if (_seqbufptr)
 	if (write(seqfd, _seqbuf, _seqbufptr) == -1) {
-	    perror("write " SEQUENCER_DEV);
+	    perror("write " SEQUENCER_DEV " or " SEQUENCER_ALTDEV);
 	    return;
 	}
     _seqbufptr = 0;
@@ -354,7 +356,7 @@
 	if (ioctl(seqfd, SNDCTL_SYNTH_INFO, &card_info[i]) == -1) 
 	{
 	    fprintf(stderr, "cannot get info on soundcard\n");
-	    perror(SEQUENCER_DEV);
+	    perror(SEQUENCER_DEV " or " SEQUENCER_ALTDEV);
 	    return 0;
 	}
 	card_info[i].device = i;
Index: sdl-mixer1.2-1.2.8/native_midi_gpl/playmidi.h
===================================================================
--- sdl-mixer1.2-1.2.8.orig/native_midi_gpl/playmidi.h	2007-10-07 22:04:55.000000000 +0000
+++ sdl-mixer1.2-1.2.8/native_midi_gpl/playmidi.h	2007-10-07 22:05:11.000000000 +0000
@@ -35,6 +35,7 @@
 #define MAXTRKS		128
 /* where to find fm patch libraries */
 #define SEQUENCER_DEV	"/dev/sequencer"
+#define SEQUENCER_ALTDEV	"/dev/snd/seq"
 #define O3MELODIC	"/etc/std.o3"
 #define O3DRUMS		"/etc/drums.o3"
 #define SBMELODIC	"/etc/std.sb"
