Index: dragonegg-2.9/Makefile
===================================================================
--- dragonegg-2.9.orig/Makefile
+++ dragonegg-2.9/Makefile
@@ -54,7 +54,7 @@ CPP_OPTIONS+=$(CPPFLAGS) $(shell $(LLVM_
 	     -DGCC_MAJOR=$(GCC_MAJOR) -DGCC_MINOR=$(GCC_MINOR) \
 	     -I$(SRC_DIR) -I$(SRC_DIR)/ADT -I$(GCC_PLUGIN_DIR)/include
 
-LD_OPTIONS+=$(shell $(LLVM_CONFIG) --ldflags) $(LDFLAGS)
+LD_OPTIONS+=$(shell $(LLVM_CONFIG) --ldflags)
 
 # NOTE: The following flags can only be used after TARGET_UTIL has been built.
 TARGET_HEADERS+=-DTARGET_NAME=\"$(shell $(TARGET_UTIL) -t)\" \
@@ -71,7 +71,7 @@ $(TARGET_UTIL_OBJECTS): %.o : $(SRC_DIR)
 
 $(TARGET_UTIL): $(TARGET_UTIL_OBJECTS)
 	@echo Linking $@
-	$(QUIET)$(CXX) -o $@ $^ $(shell $(LLVM_CONFIG) --libs support) \
+	$(QUIET)$(CXX) $(LDFLAGS) -o $@ $^ $(shell $(LLVM_CONFIG) --libs support) \
 	$(LD_OPTIONS)
 
 %.o : $(SRC_DIR)/%.c $(TARGET_UTIL)
@@ -89,7 +89,7 @@ $(TARGET_OBJECT): $(TARGET_UTIL)
 
 $(PLUGIN): $(PLUGIN_OBJECTS) $(TARGET_OBJECT) $(TARGET_UTIL)
 	@echo Linking $@
-	$(QUIET)$(CXX) -o $@ $(LOADABLE_MODULE_OPTIONS) $(CXXFLAGS) \
+	$(QUIET)$(CXX) $(LDFLAGS) -o $@ $(LOADABLE_MODULE_OPTIONS) $(CXXFLAGS) \
 	$(PLUGIN_OBJECTS) $(TARGET_OBJECT) $(shell $(LLVM_CONFIG) --libs \
 	analysis core ipo scalaropts target $(shell $(TARGET_UTIL) -p)) \
 	$(LD_OPTIONS)
